--- origsrc/libbitmask/Makefile.am	2006-05-26 03:13:49.000000000 +0900
+++ src/libbitmask/Makefile.am	2011-06-11 20:40:52.707949000 +0900
@@ -7,7 +7,7 @@ EXTRA_DIST = $(man_MANS) AUTHORS ChangeL
 
 lib_LTLIBRARIES = libbitmask.la
 libbitmask_la_SOURCES = libbitmask.c bitmask.h
-libbitmask_la_LDFLAGS=-version-info 1:1:0
+libbitmask_la_LDFLAGS=-version-info 1:1:0 -no-undefined
 
 htmldir = ${datadir}/doc/packages/libbitmask
 txtdir = ${datadir}/doc/packages/libbitmask
--- origsrc/libbitmask/configure.in	2004-10-23 09:33:48.000000000 +0900
+++ src/libbitmask/configure.in	2011-06-11 20:40:41.574955600 +0900
@@ -3,7 +3,7 @@ AC_INIT(libbitmask, 2.0, pj@sgi.com)
 AM_INIT_AUTOMAKE(libbitmask, 2.0)
 AC_CONFIG_SRCDIR([libbitmask.c])
 AC_PROG_CC
-AM_PROG_LIBTOOL
+LT_INIT
 AC_PROG_INSTALL
 AC_CHECK_HEADERS([stdint.h stdlib.h string.h])
 AC_C_CONST
