--- origsrc/ios-webkit-debug-proxy-1.9.2/Makefile.am	2025-07-01 01:24:40.000000000 +0900
+++ src/ios-webkit-debug-proxy-1.9.2/Makefile.am	2026-01-12 23:45:36.466195400 +0900
@@ -3,3 +3,6 @@
 
 AUTOMAKE_OPTIONS = foreign
 SUBDIRS = src include examples
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = ios-webkit-debug-proxy.pc
--- origsrc/ios-webkit-debug-proxy-1.9.2/configure.ac	2025-07-01 01:24:40.000000000 +0900
+++ src/ios-webkit-debug-proxy-1.9.2/configure.ac	2026-01-12 23:45:36.468203900 +0900
@@ -20,13 +20,7 @@ PKG_CHECK_MODULES(libimobiledevice, libi
 PKG_CHECK_MODULES(libplist, libplist-2.0 >= 2.2.0)
 PKG_CHECK_MODULES(libusbmuxd, libusbmuxd-2.0 >= 2.0.0)
 PKG_CHECK_MODULES(openssl, openssl >= 1.1.0)
-AC_CHECK_LIB([plist-2.0], [plist_to_xml],
-             [ ], [AC_MSG_FAILURE([*** Unable to link with libplist])],
-             [$libplist_LIBS])
 AC_CHECK_LIB([m], [log10])
-AC_CHECK_LIB([imobiledevice-1.0], [idevice_new],
-             [ ], [AC_MSG_FAILURE([*** Unable to link with libimobiledevice])],
-             [$libimobiledevice_LIBS])
 LT_INIT
 
 # Defines versions of required modules
@@ -64,7 +58,7 @@ AC_FUNC_REALLOC
 # Check for operating system
 AC_MSG_CHECKING([whether to enable WIN32 build settings])
 case ${host_os} in
-  *mingw*|*msys*|*cygwin*)
+  *mingw*|*msys*)
     win32=true
     AC_MSG_RESULT([yes])
     AC_DEFINE(WIN32_LEAN_AND_MEAN, 1, [Define to limit the scope of windows.h])
@@ -89,6 +83,5 @@ AC_CHECK_FUNCS([memmove memset regcomp s
 
 AC_CONFIG_FILES([Makefile src/Makefile include/Makefile examples/Makefile])
 
-CFLAGS="${CFLAGS} -Wall -Werror"
-
+AC_CONFIG_FILES([ios-webkit-debug-proxy.pc])
 AC_OUTPUT
--- origsrc/ios-webkit-debug-proxy-1.9.2/examples/Makefile.am	2025-07-01 01:24:40.000000000 +0900
+++ src/ios-webkit-debug-proxy-1.9.2/examples/Makefile.am	2026-01-12 23:45:36.470205000 +0900
@@ -22,13 +22,7 @@ ws_echo2_SOURCES = ws_echo2.c \
     socket_manager.h \
     hash_table.h \
     websocket.h
-ws_echo2_LDADD = \
-    ../src/base64.o \
-    ../src/char_buffer.o \
-    ../src/hash_table.o \
-    ../src/sha1.o \
-    ../src/socket_manager.o \
-    ../src/websocket.o
+ws_echo2_LDADD = $(top_builddir)/src/libios_webkit_debug_proxy.la
 
 wi_client_SOURCES = \
     wi_client.c \
@@ -36,18 +30,11 @@ wi_client_SOURCES = \
     rpc.h \
     idevice_ext.h \
     webinspector.h
-wi_client_LDADD = \
-    ../src/char_buffer.o \
-    ../src/rpc.o \
-    ../src/idevice_ext.o \
-    ../src/webinspector.o
+wi_client_LDADD = $(top_builddir)/src/libios_webkit_debug_proxy.la
 
 dl_client_SOURCES =  \
     dl_client.c \
     char_buffer.h \
     device_listener.h \
     hash_table.h
-dl_client_LDADD = \
-    ../src/char_buffer.o \
-    ../src/device_listener.o \
-    ../src/hash_table.o
+dl_client_LDADD = $(top_builddir)/src/libios_webkit_debug_proxy.la
--- origsrc/ios-webkit-debug-proxy-1.9.2/ios-webkit-debug-proxy.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/ios-webkit-debug-proxy-1.9.2/ios-webkit-debug-proxy.pc.in	2026-01-12 23:45:36.471204000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: ios-webkit-debug-proxy
+Description: iOS WebKit Debug Proxy
+Version: @VERSION@
+Libs: -L${libdir} -lios_webkit_debug_proxy
+Cflags: -I${includedir}
--- origsrc/ios-webkit-debug-proxy-1.9.2/src/Makefile.am	2025-07-01 01:24:40.000000000 +0900
+++ src/ios-webkit-debug-proxy-1.9.2/src/Makefile.am	2026-01-12 23:54:10.028366900 +0900
@@ -8,8 +8,8 @@ AM_LDFLAGS = $(libimobiledevice_LIBS) $(
 
 lib_LTLIBRARIES = libios_webkit_debug_proxy.la
 libios_webkit_debug_proxy_la_LIBADD =
-libios_webkit_debug_proxy_la_LDFLAGS = $(AM_LDFLAGS)
-libios_webkit_debug_proxy_la_SOURCES = ios_webkit_debug_proxy_main.c \
+libios_webkit_debug_proxy_la_LDFLAGS = -no-undefined $(AM_LDFLAGS)
+libios_webkit_debug_proxy_la_SOURCES = \
     base64.c base64.h \
     char_buffer.c char_buffer.h \
     device_listener.c device_listener.h \
@@ -25,19 +25,7 @@ libios_webkit_debug_proxy_la_SOURCES = i
     websocket.c websocket.h
 
 bin_PROGRAMS = ios_webkit_debug_proxy
-ios_webkit_debug_proxy_SOURCES = ios_webkit_debug_proxy_main.c \
-    base64.c base64.h \
-    char_buffer.c char_buffer.h \
-    device_listener.c device_listener.h \
-    hash_table.c hash_table.h \
-    ios_webkit_debug_proxy.c ios_webkit_debug_proxy.h \
-    port_config.c port_config.h \
-    rpc.c rpc.h \
-    sha1.c sha1.h \
-    socket_manager.c socket_manager.h \
-    validate_utf8.h \
-    idevice_ext.c idevice_ext.h \
-    webinspector.c webinspector.h \
-    websocket.c websocket.h
+ios_webkit_debug_proxy_SOURCES = ios_webkit_debug_proxy_main.c
 ios_webkit_debug_proxy_CFLAGS = $(AM_CFLAGS)
 ios_webkit_debug_proxy_LDFLAGS = $(AM_LDFLAGS)
+ios_webkit_debug_proxy_LDADD = libios_webkit_debug_proxy.la
