--- origsrc/ios-webkit-debug-proxy-1.8.5/Makefile.am	2019-06-30 18:10:56.000000000 +0900
+++ src/ios-webkit-debug-proxy-1.8.5/Makefile.am	2020-01-13 16:11:47.007686800 +0900
@@ -3,3 +3,6 @@
 
 AUTOMAKE_OPTIONS = foreign
 SUBDIRS = src include examples
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = ios-webkit-debug-proxy.pc
--- origsrc/ios-webkit-debug-proxy-1.8.5/configure.ac	2019-06-30 18:10:56.000000000 +0900
+++ src/ios-webkit-debug-proxy-1.8.5/configure.ac	2020-01-13 16:08:55.297519400 +0900
@@ -83,7 +83,7 @@ CFLAGS="$CFLAGS $libimobiledevice_CFLAGS
 AC_CHECK_DECL(idevice_connection_get_fd, [AC_DEFINE([HAVE_IDEVICE_CONNECTION_GET_FD], 1, [ ])], [], [#include <libimobiledevice/libimobiledevice.h>])
 CFLAGS=$SAVE_CFLAGS
 
-AC_CONFIG_FILES([Makefile src/Makefile include/Makefile examples/Makefile])
+AC_CONFIG_FILES([ios-webkit-debug-proxy.pc Makefile src/Makefile include/Makefile examples/Makefile])
 
 CFLAGS="${CFLAGS} -Wall -Werror"
 
--- origsrc/ios-webkit-debug-proxy-1.8.5/examples/Makefile.am	2019-06-30 18:10:56.000000000 +0900
+++ src/ios-webkit-debug-proxy-1.8.5/examples/Makefile.am	2020-01-13 15:53:29.838578600 +0900
@@ -22,30 +22,18 @@ ws_echo2_SOURCES = ws_echo2.c \
     socket_manager.h \
     hash_table.h \
     websocket.h
-ws_echo2_LDADD = \
-    ../src/base64.o \
-    ../src/char_buffer.o \
-    ../src/hash_table.o \
-    ../src/sha1.o \
-    ../src/socket_manager.o \
-    ../src/websocket.o
+ws_echo2_LDADD = $(top_builddir)/src/libios_webkit_debug_proxy.la
 
 wi_client_SOURCES = \
     wi_client.c \
     char_buffer.h \
     rpc.h \
     webinspector.h
-wi_client_LDADD = \
-    ../src/char_buffer.o \
-    ../src/rpc.o \
-    ../src/webinspector.o
+wi_client_LDADD = $(top_builddir)/src/libios_webkit_debug_proxy.la
 
 dl_client_SOURCES =  \
     dl_client.c \
     char_buffer.h \
     device_listener.h \
     hash_table.h
-dl_client_LDADD = \
-    ../src/char_buffer.o \
-    ../src/device_listener.o \
-    ../src/hash_table.o
+dl_client_LDADD = $(top_builddir)/src/libios_webkit_debug_proxy.la
--- origsrc/ios-webkit-debug-proxy-1.8.5/ios-webkit-debug-proxy.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/ios-webkit-debug-proxy-1.8.5/ios-webkit-debug-proxy.pc.in	2020-01-13 16:11:26.959753900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: ios-webkit-debug-proxy
+Description: iOS WebKit Debug Proxy
+Version: @VERSION@
+Libs: -L${libdir} -lios_webkit_debug_proxy
+Cflags: -I${includedir}
--- origsrc/ios-webkit-debug-proxy-1.8.5/src/Makefile.am	2019-06-30 18:10:56.000000000 +0900
+++ src/ios-webkit-debug-proxy-1.8.5/src/Makefile.am	2020-01-13 15:51:40.281712900 +0900
@@ -8,7 +8,7 @@ AM_LDFLAGS = $(libimobiledevice_LIBS) $(
 
 lib_LTLIBRARIES = libios_webkit_debug_proxy.la
 libios_webkit_debug_proxy_la_LIBADD =
-libios_webkit_debug_proxy_la_LDFLAGS = $(AM_LDFLAGS)
+libios_webkit_debug_proxy_la_LDFLAGS = -no-undefined $(AM_LDFLAGS)
 libios_webkit_debug_proxy_la_SOURCES = ios_webkit_debug_proxy_main.c \
     base64.c base64.h \
     char_buffer.c char_buffer.h \
