--- origsrc/libuaecore11-0.1.0-rc2/Makefile.am	2014-07-04 00:54:11.000000000 +0900
+++ src/libuaecore11-0.1.0-rc2/Makefile.am	2025-08-21 14:55:09.272365200 +0900
@@ -1,3 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4
 
 SUBDIRS = src
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libuaecore11-0.1.0-rc2/configure.ac	2014-07-04 00:54:11.000000000 +0900
+++ src/libuaecore11-0.1.0-rc2/configure.ac	2025-08-21 14:55:09.272365200 +0900
@@ -10,7 +10,7 @@ LT_INIT
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_SRCDIR([src/WinUAE/build68k.cpp])
 AC_CONFIG_HEADERS([src/config.h])
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([libuaecore11.pc Makefile
                  src/Makefile])
 
 # Checks for programs.
--- origsrc/libuaecore11-0.1.0-rc2/libuaecore11.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libuaecore11-0.1.0-rc2/libuaecore11.pc.in	2025-08-21 14:55:09.277554900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libuaecore11
+Description: Motorola MC68000 emulation core from WinUAE
+Version: @VERSION@
+Libs: -L${libdir} -luaecore11
+Cflags: -I${includedir}
