--- origsrc/libee-0.3.1/configure.ac	2011-04-12 18:43:19.000000000 +0900
+++ src/libee-0.3.1/configure.ac	2011-06-14 21:13:16.442649500 +0900
@@ -15,7 +15,7 @@ if test "$GCC" = "yes"
 then CFLAGS="$CFLAGS -W -Wall -Wformat-security -Wshadow -Wcast-align -Wpointer-arith -Wmissing-format-attribute -g"
 fi
 
-AC_PROG_LIBTOOL
+LT_INIT
 
 # Checks for libraries.
 save_LIBS=$LIBS
--- origsrc/libee-0.3.1/src/Makefile.am	2010-12-17 21:36:34.000000000 +0900
+++ src/libee-0.3.1/src/Makefile.am	2011-06-14 21:13:38.956006900 +0900
@@ -35,7 +35,7 @@ libee_la_SOURCES = \
 
 libee_la_CPPFLAGS = $(LIBXML2_CFLAGS) $(LIBESTR_CFLAGS) $(LIBEE_CFLAGS)
 libee_la_LIBADD = $(LIBXML2_LIBS) $(LIBESTR_LIBS)
-libee_la_LDFLAGS = -version-info 0:0:0
+libee_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 
 convert_SOURCES = convert.c
 convert_CPPFLAGS =  -I$(top_srcdir) $(LIBEE_CFLAGS) $(LIBESTR_CFLAGS) $(LIBXML2_CFLAGS)
