--- origsrc/eris-1.3.20/Eris/Makefile.am	2011-09-15 18:11:06.000000000 +0900
+++ src/eris-1.3.20/Eris/Makefile.am	2012-11-24 21:08:08.557904900 +0900
@@ -2,7 +2,7 @@ AM_CPPFLAGS = -I$(top_builddir) -I$(top_
 
 lib_LTLIBRARIES = liberis-1.3.la
 
-liberis_1_3_la_LDFLAGS = -version-info @ERIS_VERSION_INFO@
+liberis_1_3_la_LDFLAGS = -version-info @ERIS_VERSION_INFO@ -no-undefined
 
 liberis_1_3_la_SOURCES = \
 	Exceptions.cpp \
--- origsrc/eris-1.3.20/eris.dox.in	2011-09-15 18:11:06.000000000 +0900
+++ src/eris-1.3.20/eris.dox.in	2012-11-24 21:08:08.557904900 +0900
@@ -462,7 +462,7 @@ WARN_LOGFILE           =
 # directories like "/usr/src/myproject". Separate the files or directories 
 # with spaces.
 
-INPUT                  = Eris
+INPUT                  = @top_srcdir@/Eris
 
 # This tag can be used to specify the character encoding of the source files that 
 # doxygen parses. Internally doxygen uses the UTF-8 encoding, which is also the default 
@@ -880,7 +880,7 @@ RTF_EXTENSIONS_FILE    =
 # If the GENERATE_MAN tag is set to YES (the default) Doxygen will 
 # generate man pages
 
-GENERATE_MAN           = NO
+GENERATE_MAN           = YES
 
 # The MAN_OUTPUT tag is used to specify where the man pages will be put. 
 # If a relative path is entered the value of OUTPUT_DIRECTORY will be 
