--- origsrc/libbfoverlay-20240505/Makefile.am	2024-05-05 15:48:36.000000000 +0900
+++ src/libbfoverlay-20240505/Makefile.am	2025-05-20 18:57:58.978533600 +0900
@@ -3,20 +3,8 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfvalue \
 	libbfoverlay \
 	bfoverlaytools \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libbfoverlay-20240505/configure.ac	2024-05-05 15:48:36.000000000 +0900
+++ src/libbfoverlay-20240505/configure.ac	2025-05-20 18:58:09.819170000 +0900
@@ -146,17 +146,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
 AC_CONFIG_FILES([libbfoverlay/Makefile])
 AC_CONFIG_FILES([bfoverlaytools/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
