--- origsrc/libspatialaudio-0.4.0/CMakeLists.txt	2025-12-04 16:00:01.000000000 +0900
+++ src/libspatialaudio-0.4.0/CMakeLists.txt	2026-01-21 08:11:11.886148300 +0900
@@ -177,6 +177,10 @@ if(MYSOFA_FOUND)
     target_link_libraries(spatialaudio ${MYSOFA_LIBRARIES})
 endif(MYSOFA_FOUND)
 
+if (DEFINED SOVERSION)
+    set_target_properties(spatialaudio PROPERTIES SOVERSION ${SOVERSION})
+endif()
+
 configure_file(
     "${CMAKE_CURRENT_SOURCE_DIR}/cmake/spatialaudio.pc.cmake"
     "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.pc"
@@ -184,7 +188,7 @@ configure_file(
 )
 
 install(TARGETS spatialaudio
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     FILE_SET HEADERS
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/spatialaudio
@@ -193,8 +197,11 @@ install(TARGETS spatialaudio
 )
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.pc"
     DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
-install(FILES ${PROJECT_BINARY_DIR}/config.h
-    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/spatialaudio)
+
+if (EXISTS ${PROJECT_BINARY_DIR}/config.h)
+    install(FILES ${PROJECT_BINARY_DIR}/config.h
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/spatialaudio)
+endif()
 
 enable_testing()
 add_subdirectory(tests)
