--- origsrc/ncmines-1.0.3/src/draw.c	2020-06-28 04:55:58.000000000 +0900
+++ src/ncmines-1.0.3/src/draw.c	2025-01-11 22:52:42.064222000 +0900
@@ -104,7 +104,7 @@ void drawheader(struct gamestate_t *gs)
     werase(titlewin);
     mvwprintw(titlewin, 0, 0, "%d:%02d", (int)gs->gtime / 60, (int)gs->gtime % 60);
     snprintf(minestr, 8, "%d/%d", (int)gs->mines_found, (int)gs->b->nmines);
-    mvwprintw(titlewin, 0, gs->b->w - strlen(minestr), minestr);
+    mvwprintw(titlewin, 0, gs->b->w - strlen(minestr), "%s", minestr);
     wmove(gamewin, gs->y, gs->x);
     wrefresh(titlewin);
 }
