--- origsrc/vecx-1.1/Makefile	2016-08-20 04:34:16.000000000 +0900
+++ src/vecx-1.1/Makefile	2025-12-28 10:54:17.256175700 +0900
@@ -1,5 +1,5 @@
-
-CFLAGS := -O3 -Wall -Wextra $(shell sdl-config --cflags)
+CFLAGS ?=
+CFLAGS := $(CFLAGS) -O3 -Wall -Wextra $(shell sdl-config --cflags)
 LIBS := $(shell sdl-config --libs) -lSDL_gfx -lSDL_image
 OBJECTS := e6809.o e8910.o osint.o vecx.o
 TARGET := vecx
--- origsrc/vecx-1.1/e6809.c	2016-08-20 04:34:16.000000000 +0900
+++ src/vecx-1.1/e6809.c	2025-12-28 10:53:05.730311100 +0900
@@ -10,7 +10,7 @@
  *    the lower bits with the unused upper bits all set to zero.
  */
 
-#define einline __inline
+#define einline
 
 enum {
 	FLAG_E		= 0x80,
