--- origsrc/libqmi-1.4.0/gobi-api/GobiAPI_1.0.40/GobiConnectionMgmt/Makefile.am	2013-06-08 01:37:57.000000000 +0900
+++ src/libqmi-1.4.0/gobi-api/GobiAPI_1.0.40/GobiConnectionMgmt/Makefile.am	2013-07-28 14:35:45.432914100 +0900
@@ -29,3 +29,4 @@ libGobiConnectionMgmt_la_LIBADD = \
 	$(top_builddir)/Shared/libShared.la \
 	$(top_builddir)/Core/libCore.la
 
+libGobiConnectionMgmt_la_LDFLAGS = -no-undefined
--- origsrc/libqmi-1.4.0/gobi-api/GobiAPI_1.0.40/GobiImageMgmt/Makefile.am	2013-06-08 01:37:57.000000000 +0900
+++ src/libqmi-1.4.0/gobi-api/GobiAPI_1.0.40/GobiImageMgmt/Makefile.am	2013-07-28 14:35:45.448539100 +0900
@@ -29,3 +29,4 @@ libGobiImageMgmt_la_LIBADD = \
 	$(top_builddir)/Shared/libShared.la \
 	$(top_builddir)/Core/libCore.la
 
+ libGobiImageMgmt_la_LDFLAGS = -no-undefined
--- origsrc/libqmi-1.4.0/gobi-api/fixed-GobiAPI-1.0.40/GobiConnectionMgmt/Makefile.am	2013-06-08 01:37:57.000000000 +0900
+++ src/libqmi-1.4.0/gobi-api/fixed-GobiAPI-1.0.40/GobiConnectionMgmt/Makefile.am	2013-07-28 14:35:45.448539100 +0900
@@ -30,3 +30,4 @@ libGobiConnectionMgmt_la_LIBADD = \
 	$(top_builddir)/Core/libCore.la \
 	-lpthread
 
+libGobiConnectionMgmt_la_LDFLAGS = -no-undefined
--- origsrc/libqmi-1.4.0/gobi-api/fixed-GobiAPI-1.0.40/GobiImageMgmt/Makefile.am	2013-06-08 01:37:57.000000000 +0900
+++ src/libqmi-1.4.0/gobi-api/fixed-GobiAPI-1.0.40/GobiImageMgmt/Makefile.am	2013-07-28 14:35:45.448539100 +0900
@@ -30,3 +30,4 @@ libGobiImageMgmt_la_LIBADD = \
 	$(top_builddir)/Core/libCore.la \
 	-lpthread
 
+libGobiImageMgmt_la_LDFLAGS = -no-undefined
--- origsrc/libqmi-1.4.0/libqmi-glib/Makefile.am	2013-06-08 01:37:57.000000000 +0900
+++ src/libqmi-1.4.0/libqmi-glib/Makefile.am	2013-07-28 14:35:45.448539100 +0900
@@ -27,6 +27,8 @@ libqmi_glib_la_SOURCES = \
 	qmi-device.h qmi-device.c \
 	qmi-client.h qmi-client.c
 
+libqmi_glib_la_LDFLAGS = -no-undefined
+
 libqmi_glib_la_LIBADD = \
 	${top_builddir}/libqmi-glib/generated/libqmi-glib-generated.la \
 	$(LIBQMI_GLIB_LIBS)
