--- origsrc/libqmi-1.20.0/src/libqmi-glib/Makefile.am	2018-01-20 22:17:41.000000000 +0900
+++ src/libqmi-1.20.0/src/libqmi-glib/Makefile.am	2018-03-06 09:25:15.344629400 +0900
@@ -46,7 +46,7 @@ libqmi_glib_la_LIBADD = \
 	$(GLIB_LIBS) \
 	$(MBIM_LIBS)
 
-libqmi_glib_la_LDFLAGS = \
+libqmi_glib_la_LDFLAGS = -no-undefined \
 	-version-info $(QMI_GLIB_LT_CURRENT):$(QMI_GLIB_LT_REVISION):$(QMI_GLIB_LT_AGE)
 
 includedir = @includedir@/libqmi-glib
--- origsrc/libqmi-1.20.0/src/libqmi-glib/qmi-version.h	2018-01-20 23:20:09.000000000 +0900
+++ src/libqmi-1.20.0/src/libqmi-glib/qmi-version.h	2018-03-06 09:28:21.285178600 +0900
@@ -92,6 +92,6 @@
  *
  * Since: 1.16
  */
-#define QMI_MBIM_QMUX_SUPPORTED 1
+#define QMI_MBIM_QMUX_SUPPORTED 0
 
 #endif /* _QMI_VERSION_H_ */
--- origsrc/libqmi-1.20.0/utils/swi-update.c	2017-09-14 02:46:37.000000000 +0900
+++ src/libqmi-1.20.0/utils/swi-update.c	2018-03-06 09:25:15.357583900 +0900
@@ -33,7 +33,11 @@
 #include <fcntl.h>
 #include <getopt.h>
 #include <limits.h>
+#if defined(__CYGWIN__)
+#include <asm/types.h>
+#else
 #include <linux/types.h>
+#endif
 #include <malloc.h>
 #include <stdbool.h>
 #include <stdio.h>
