--- origsrc/libqmi-1.18.0/src/libqmi-glib/Makefile.am	2017-03-21 22:26:54.000000000 +0900
+++ src/libqmi-1.18.0/src/libqmi-glib/Makefile.am	2017-03-31 18:03:37.904217000 +0900
@@ -45,7 +45,7 @@ libqmi_glib_la_LIBADD = \
 	$(GLIB_LIBS) \
 	$(MBIM_LIBS)
 
-libqmi_glib_la_LDFLAGS = \
+libqmi_glib_la_LDFLAGS = -no-undefined \
 	-version-info $(QMI_GLIB_LT_CURRENT):$(QMI_GLIB_LT_REVISION):$(QMI_GLIB_LT_AGE)
 
 includedir = @includedir@/libqmi-glib
--- origsrc/libqmi-1.18.0/src/libqmi-glib/qmi-version.h	2017-03-22 20:20:32.000000000 +0900
+++ src/libqmi-1.18.0/src/libqmi-glib/qmi-version.h	2017-03-31 18:12:26.142940500 +0900
@@ -92,6 +92,6 @@
  *
  * Since: 1.16
  */
-#define QMI_MBIM_QMUX_SUPPORTED 1
+#define QMI_MBIM_QMUX_SUPPORTED 0
 
 #endif /* _QMI_VERSION_H_ */
--- origsrc/libqmi-1.18.0/utils/swi-update.c	2017-03-21 22:26:54.000000000 +0900
+++ src/libqmi-1.18.0/utils/swi-update.c	2017-03-31 18:17:02.263304800 +0900
@@ -33,7 +33,11 @@
 #include <fcntl.h>
 #include <getopt.h>
 #include <limits.h>
+#if defined(__CYGWIN__)
+#include <asm/types.h>
+#else
 #include <linux/types.h>
+#endif
 #include <malloc.h>
 #include <stdbool.h>
 #include <stdio.h>
