--- origsrc/libqmi-1.26.4/src/libqmi-glib/Makefile.am	2020-08-23 18:32:04.000000000 +0900
+++ src/libqmi-1.26.4/src/libqmi-glib/Makefile.am	2020-08-24 07:09:05.446030500 +0900
@@ -109,7 +109,7 @@ libqmi_glib_la_LIBADD += \
 
 endif
 
-libqmi_glib_la_LDFLAGS = \
+libqmi_glib_la_LDFLAGS = -no-undefined \
 	-version-info $(QMI_GLIB_LT_CURRENT):$(QMI_GLIB_LT_REVISION):$(QMI_GLIB_LT_AGE) \
 	$(WARN_CFLAGS) \
 	$(GLIB_LIBS) \
--- origsrc/libqmi-1.26.4/utils/swi-update.c	2019-04-17 17:25:22.000000000 +0900
+++ src/libqmi-1.26.4/utils/swi-update.c	2020-08-24 07:09:05.461024900 +0900
@@ -33,7 +33,11 @@
 #include <fcntl.h>
 #include <getopt.h>
 #include <limits.h>
+#if defined(__linux__)
 #include <linux/types.h>
+#else
+#include <asm/types.h>
+#endif
 #include <malloc.h>
 #include <stdbool.h>
 #include <stdio.h>
