--- origsrc/libqmi-1.20.2/src/libqmi-glib/Makefile.am	2018-08-12 21:41:17.000000000 +0900
+++ src/libqmi-1.20.2/src/libqmi-glib/Makefile.am	2018-08-24 12:35:34.349452000 +0900
@@ -46,7 +46,7 @@ libqmi_glib_la_LIBADD = \
 	$(GLIB_LIBS) \
 	$(MBIM_LIBS)
 
-libqmi_glib_la_LDFLAGS = \
+libqmi_glib_la_LDFLAGS = -no-undefined \
 	-version-info $(QMI_GLIB_LT_CURRENT):$(QMI_GLIB_LT_REVISION):$(QMI_GLIB_LT_AGE)
 
 includedir = @includedir@/libqmi-glib
--- origsrc/libqmi-1.20.2/src/libqmi-glib/qmi-version.h	2018-08-12 21:53:44.000000000 +0900
+++ src/libqmi-1.20.2/src/libqmi-glib/qmi-version.h	2018-08-24 12:37:54.792191100 +0900
@@ -92,6 +92,6 @@
  *
  * Since: 1.16
  */
-#define QMI_MBIM_QMUX_SUPPORTED 1
+#define QMI_MBIM_QMUX_SUPPORTED 0
 
 #endif /* _QMI_VERSION_H_ */
--- origsrc/libqmi-1.20.2/utils/swi-update.c	2017-09-14 02:46:37.000000000 +0900
+++ src/libqmi-1.20.2/utils/swi-update.c	2018-08-24 12:35:34.365015100 +0900
@@ -33,7 +33,11 @@
 #include <fcntl.h>
 #include <getopt.h>
 #include <limits.h>
+#if defined(__CYGWIN__)
+#include <asm/types.h>
+#else
 #include <linux/types.h>
+#endif
 #include <malloc.h>
 #include <stdbool.h>
 #include <stdio.h>
