--- origsrc/libqmi-1.6.0/gobi-api/GobiAPI_1.0.40/GobiConnectionMgmt/Makefile.am	2013-09-05 21:59:26.000000000 +0900
+++ src/libqmi-1.6.0/gobi-api/GobiAPI_1.0.40/GobiConnectionMgmt/Makefile.am	2013-09-26 12:09:10.329759000 +0900
@@ -29,3 +29,4 @@ libGobiConnectionMgmt_la_LIBADD = \
 	$(top_builddir)/Shared/libShared.la \
 	$(top_builddir)/Core/libCore.la
 
+libGobiConnectionMgmt_la_LDFLAGS = -no-undefined
--- origsrc/libqmi-1.6.0/gobi-api/GobiAPI_1.0.40/GobiImageMgmt/Makefile.am	2013-09-05 21:59:26.000000000 +0900
+++ src/libqmi-1.6.0/gobi-api/GobiAPI_1.0.40/GobiImageMgmt/Makefile.am	2013-09-26 12:09:10.345384000 +0900
@@ -29,3 +29,4 @@ libGobiImageMgmt_la_LIBADD = \
 	$(top_builddir)/Shared/libShared.la \
 	$(top_builddir)/Core/libCore.la
 
+ libGobiImageMgmt_la_LDFLAGS = -no-undefined
--- origsrc/libqmi-1.6.0/gobi-api/fixed-GobiAPI-1.0.40/GobiConnectionMgmt/Makefile.am	2013-09-05 21:59:26.000000000 +0900
+++ src/libqmi-1.6.0/gobi-api/fixed-GobiAPI-1.0.40/GobiConnectionMgmt/Makefile.am	2013-09-26 12:09:10.345384000 +0900
@@ -30,3 +30,4 @@ libGobiConnectionMgmt_la_LIBADD = \
 	$(top_builddir)/Core/libCore.la \
 	-lpthread
 
+libGobiConnectionMgmt_la_LDFLAGS = -no-undefined
--- origsrc/libqmi-1.6.0/gobi-api/fixed-GobiAPI-1.0.40/GobiImageMgmt/Makefile.am	2013-09-05 21:59:26.000000000 +0900
+++ src/libqmi-1.6.0/gobi-api/fixed-GobiAPI-1.0.40/GobiImageMgmt/Makefile.am	2013-09-26 12:09:10.345384000 +0900
@@ -30,3 +30,4 @@ libGobiImageMgmt_la_LIBADD = \
 	$(top_builddir)/Core/libCore.la \
 	-lpthread
 
+libGobiImageMgmt_la_LDFLAGS = -no-undefined
--- origsrc/libqmi-1.6.0/libqmi-glib/Makefile.am	2013-09-05 21:59:26.000000000 +0900
+++ src/libqmi-1.6.0/libqmi-glib/Makefile.am	2013-09-26 12:15:06.212571500 +0900
@@ -36,7 +36,7 @@ libqmi_glib_la_LIBADD = \
 	${top_builddir}/libqmi-glib/generated/libqmi-glib-generated.la \
 	$(LIBQMI_GLIB_LIBS)
 
-libqmi_glib_la_LDFLAGS = \
+libqmi_glib_la_LDFLAGS = -no-undefined \
 	-version-info $(QMI_GLIB_LT_CURRENT):$(QMI_GLIB_LT_REVISION):$(QMI_GLIB_LT_AGE)
 
 includedir = @includedir@/libqmi-glib
