--- origsrc/libqmi-1.28.4/src/libqmi-glib/Makefile.am	2021-05-14 21:18:13.000000000 +0000
+++ src/libqmi-1.28.4/src/libqmi-glib/Makefile.am	2021-05-28 16:18:41.976646900 +0000
@@ -105,7 +105,7 @@ libqmi_glib_la_SOURCES += \
 	$(NULL)
 endif
 
-libqmi_glib_la_LDFLAGS = \
+libqmi_glib_la_LDFLAGS = -no-undefined \
 	-version-info $(QMI_GLIB_LT_CURRENT):$(QMI_GLIB_LT_REVISION):$(QMI_GLIB_LT_AGE) \
 	$(WARN_CFLAGS) \
 	$(GLIB_LIBS) \
--- origsrc/libqmi-1.28.4/utils/swi-update.c	2021-05-01 19:59:02.000000000 +0000
+++ src/libqmi-1.28.4/utils/swi-update.c	2021-05-28 16:18:41.976646900 +0000
@@ -33,7 +33,11 @@
 #include <fcntl.h>
 #include <getopt.h>
 #include <limits.h>
+#if defined(__linux__)
 #include <linux/types.h>
+#else
+#include <asm/types.h>
+#endif
 #include <malloc.h>
 #include <stdbool.h>
 #include <stdio.h>
