--- origsrc/libotf-0.9.12/configure.ac	2010-09-16 10:09:58.000000000 +0900
+++ src/libotf-0.9.12/configure.ac	2011-02-14 18:26:32.472520000 +0900
@@ -29,7 +29,7 @@ AM_CONFIG_HEADER(src/config.h)
 
 # Checks for programs.
 AC_PROG_CC
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 # Checks for libraries.
 AC_PATH_XTRA
--- origsrc/libotf-0.9.12/example/Makefile.am	2010-04-14 13:48:39.000000000 +0900
+++ src/libotf-0.9.12/example/Makefile.am	2011-02-14 18:26:06.705468800 +0900
@@ -26,6 +26,8 @@ bin_PROGRAMS = otfdump otfview otflist o
 
 CommonLDADD = ${top_builddir}/src/libotf.la
 
+AM_CPPFLAGS = -I$(top_srcdir)/src/
+
 otflist_SOURCE = otflist.c
 otflist_LDADD = ${CommonLDADD}
 
--- origsrc/libotf-0.9.12/src/Makefile.am	2010-04-14 13:48:39.000000000 +0900
+++ src/libotf-0.9.12/src/Makefile.am	2011-02-14 18:26:06.705468800 +0900
@@ -31,4 +31,6 @@ libotf_la_SOURCES = \
 	otfdrive.c \
 	internal.h
 
+libotf_la_LDFLAGS = -no-undefined
+
 include_HEADERS = otf.h
