--- origsrc/libdsm-0.4.3/bin/meson.build	2022-06-02 13:59:44.000000000 +0900
+++ src/libdsm-0.4.3/bin/meson.build	2022-11-17 09:56:06.408621100 +0900
@@ -1,19 +1,23 @@
 executable('dsm', 'dsm.c', 
   include_directories: [includes],
-  link_with: libdsm
+  link_with: libdsm,
+  install: true
 )
 
-executable('discover', 'discover.c',
+executable('dsm_discover', 'discover.c',
   include_directories: [includes],
-  link_with: libdsm
+  link_with: libdsm,
+  install: true
 )
 
-executable('inverse', 'inverse.c',
+executable('dsm_inverse', 'inverse.c',
   include_directories: [includes],
-  link_with: libdsm
+  link_with: libdsm,
+  install: true
 )
 
-executable('lookup', 'lookup.c',
+executable('dsm_lookup', 'lookup.c',
   include_directories: [includes],
-  link_with: libdsm
+  link_with: libdsm,
+  install: true
 )
--- origsrc/libdsm-0.4.3/meson.build	2022-06-02 13:59:44.000000000 +0900
+++ src/libdsm-0.4.3/meson.build	2022-11-17 09:56:06.408621100 +0900
@@ -259,7 +259,7 @@ pkg_mod.generate(
 
 # We only add the doc target if doxygen is found
 doxygen = find_program('doxygen', required: false)
-if doxygen.found()
+if doxygen.found() and get_option('doc')
   subdir('doc')
 else
   warning('Doxygen not found - continuing without Doxygen support')
--- origsrc/libdsm-0.4.3/meson_options.txt	2022-06-02 13:59:44.000000000 +0900
+++ src/libdsm-0.4.3/meson_options.txt	2022-11-17 09:56:06.416544900 +0900
@@ -1 +1,2 @@
 option('binaries', type: 'boolean', value: 'true')
+option('doc', type: 'boolean', value: 'false')
--- origsrc/libdsm-0.4.3/src/smb_ntlm.c	2022-06-02 13:59:44.000000000 +0900
+++ src/libdsm-0.4.3/src/smb_ntlm.c	2022-11-17 09:56:06.419656800 +0900
@@ -36,6 +36,7 @@
 # define _CRT_RAND_S
 #endif
 
+#include <alloca.h>
 #include <assert.h>
 #include <ctype.h>
 #include <wctype.h>
--- origsrc/libdsm-0.4.3/src/smb_share.c	2022-06-02 13:59:44.000000000 +0900
+++ src/libdsm-0.4.3/src/smb_share.c	2022-11-17 09:56:06.422600500 +0900
@@ -32,6 +32,7 @@
 # include "config.h"
 #endif
 
+#include <alloca.h>
 #include <assert.h>
 #include <stdlib.h>
 #include <unistd.h>
