--- origsrc/pure-sfv-0.3/Makefile	2002-08-27 08:42:22.000000000 +0900
+++ src/pure-sfv-0.3/Makefile	2024-10-29 07:04:33.805749500 +0900
@@ -54,13 +54,14 @@ SRCS = ${PAR_SRCS:%.c=${PARS_PATH}${SEP}
 
 VC_CRAP = pure-sfv.ncb pure-sfv.plg
 
-CFLAGS +=	-Wall -Werror -O2 -g -Wno-unused
+CFLAGS ?=	-Wall -Werror -O2 -g -Wno-unused
+CFLAGS += $(CPPFLAGS)
 INSTALL_PREFIX = /usr/local
 INSTALL_PROGRAM = /usr/bin/install -c
 
 all: pure-sfv
 
-CC = gcc
+CC ?= gcc
 
 OBJ = ${SRCS:%.c=%.o}
 
--- origsrc/pure-sfv-0.3/platform.h	2002-08-27 08:17:59.000000000 +0900
+++ src/pure-sfv-0.3/platform.h	2024-10-29 07:05:14.575698000 +0900
@@ -45,7 +45,9 @@
 #define DIR_SEP_PLATFORM DIR_SEP_UNIX
 #define DOS_LINE_END     "\r\n"
 
+#if !defined(O_BINARY)
 #define O_BINARY      0
+#endif
 
 /* End on Unix */
 #else
