--- origsrc/cpprestsdk-2.10.19/CMakeLists.txt	2023-12-05 13:23:31.000000000 +0900
+++ src/cpprestsdk-2.10.19/CMakeLists.txt	2025-09-01 14:39:40.572432800 +0900
@@ -2,3 +2,10 @@ cmake_minimum_required(VERSION 3.9)
 project(cpprestsdk-root NONE)
 enable_testing()
 add_subdirectory(Release)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/cpprestsdk.pc.in ${CMAKE_BINARY_DIR}/cpprestsdk.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/cpprestsdk.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/cpprestsdk-2.10.19/Release/include/cpprest/details/cpprest_compat.h	2023-12-05 13:23:31.000000000 +0900
+++ src/cpprestsdk-2.10.19/Release/include/cpprest/details/cpprest_compat.h	2025-09-01 14:39:40.577043800 +0900
@@ -28,7 +28,9 @@
 
 #else // ^^^ _WIN32 ^^^ // vvv !_WIN32 vvv
 
+#if !defined(__CYGWIN__)
 #define __declspec(x) __attribute__((x))
+#endif
 #define dllimport
 #define novtable /* no novtable equivalent */
 #define __assume(x)                                                                                                    \
--- origsrc/cpprestsdk-2.10.19/Release/src/pch/stdafx.h	2023-12-05 13:23:31.000000000 +0900
+++ src/cpprestsdk-2.10.19/Release/src/pch/stdafx.h	2025-09-01 14:39:40.577043800 +0900
@@ -62,7 +62,9 @@
 #include "boost/thread/mutex.hpp"
 #include <fcntl.h>
 #include <sys/stat.h>
+#if !defined(__CYGWIN__)
 #include <sys/syscall.h>
+#endif
 #include <unistd.h>
 #endif // _WIN32
 
--- origsrc/cpprestsdk-2.10.19/Release/src/pplx/pplxlinux.cpp	2023-12-05 13:23:31.000000000 +0900
+++ src/cpprestsdk-2.10.19/Release/src/pplx/pplxlinux.cpp	2025-09-01 14:39:40.579060000 +0900
@@ -15,7 +15,9 @@
 
 #include "pplx/pplx.h"
 #include "pplx/threadpool.h"
+#if !defined(__CYGWIN__)
 #include "sys/syscall.h"
+#endif
 #include <thread>
 
 #ifdef _WIN32
--- origsrc/cpprestsdk-2.10.19/cpprestsdk.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/cpprestsdk-2.10.19/cpprestsdk.pc.in	2025-09-01 14:39:40.581084500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: cpprestsdk
+Description: C++ libraries for cloud-based client-server communication
+Version: @VERSION@
+Libs: -L${libdir} -lcpprest
+Cflags: -I${includedir}
