--- origsrc/libindicate-12.10.1/bindings/python/Makefile.am	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/bindings/python/Makefile.am	2014-11-03 10:15:40.943545500 +0900
@@ -27,8 +27,8 @@ pkgindicate_PYTHON = __init__.py
 indicatedir = $(pkgpyexecdir)/indicate
 indicate_LTLIBRARIES = _indicate.la
 
-_indicate_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_indicate
-_indicate_la_LIBADD = $(INDICATE_PYTHON_LIBS)  -L$(top_builddir)/libindicate/.libs/ -lindicate -L$(top_builddir)/libindicate-gtk/.libs -lindicate-gtk$(VER) $(LIBINDICATEGTK_LIBS)
+_indicate_la_LDFLAGS = -no-undefined -module -avoid-version -export-symbols-regex init_indicate
+_indicate_la_LIBADD = -lpython2.7 $(INDICATE_PYTHON_LIBS) $(top_builddir)/libindicate/libindicate.la $(top_builddir)/libindicate-gtk/libindicate-gtk$(VER).la $(LIBINDICATEGTK_LIBS)
 _indicate_la_SOURCES = indicatemodule.c
 nodist__indicate_la_SOURCES = indicate.c
 
--- origsrc/libindicate-12.10.1/configure.ac	2012-08-22 13:52:22.000000000 +0900
+++ src/libindicate-12.10.1/configure.ac	2014-11-02 23:36:24.643672900 +0900
@@ -140,6 +140,7 @@ AC_PATH_PROG([VALA_API_GEN], [vapigen])
 # Check for Mono support
 ###########################
 
+if test "$host_os" != "cygwin"; then
 MONO_REQUIRED_VERSION=1.0
 PKG_CHECK_MODULES(MONO_DEPENDENCY, mono >= $MONO_REQUIRED_VERSION, has_mono=true, has_mono=false)
 
@@ -190,6 +191,7 @@ AC_PATH_PROG(GAPI_CODEGEN, gapi2-codegen
 AC_SUBST(GAPI_CODEGEN)
 AC_PATH_PROG(GAPI_FIXUP, gapi2-fixup, no)
 AC_SUBST(GAPI_FIXUP)
+fi
 
 ###########################
 # Check for Python support
@@ -209,11 +211,18 @@ PYGOBJECT_REQUIRED=0.22
 AM_PATH_PYTHON([$PYTHON_REQUIRED])
 AM_CHECK_PYTHON_HEADERS(,[AC_MSG_ERROR(could not find Python headers)])
 
+if test "$host_os" = "cygwin"; then
+PKG_CHECK_MODULES([INDICATE_PYTHON], [pygtk-2.0 >= $PYGTK_REQUIRED
+                                      pygobject-2.0 >= $PYGOBJECT_REQUIRED],
+                                     [have_python=yes]
+)
+else
 PKG_CHECK_MODULES([INDICATE_PYTHON], [python-$PYTHON_VERSION >= $PYTHON_REQUIRED
                                       pygtk-2.0 >= $PYGTK_REQUIRED
                                       pygobject-2.0 >= $PYGOBJECT_REQUIRED],
                                      [have_python=yes]
 )
+fi
 
 # libpyglib-2.0-python* doesn't have a pkgconfig unfortunately
 INDICATE_PYTHON_LIBS+=" -lpyglib-2.0-python$PYTHON_VERSION"
--- origsrc/libindicate-12.10.1/examples/im-client.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/examples/im-client.c	2014-11-03 10:18:17.657412700 +0900
@@ -87,8 +87,6 @@ menuitem_click (DbusmenuMenuitem * menui
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	IndicateServer * server = indicate_server_ref_default();
 	indicate_server_set_type(server, "message.im");
 	indicate_server_set_desktop_file(server, "/usr/share/applications/empathy.desktop");
--- origsrc/libindicate-12.10.1/examples/indicate-alot.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/examples/indicate-alot.c	2014-11-03 10:18:24.751162700 +0900
@@ -29,8 +29,6 @@ with this program.  If not, see <http://
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	IndicateIndicator * indicators[ALOT];
 	int i;
 
--- origsrc/libindicate-12.10.1/examples/indicate-and-crash.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/examples/indicate-and-crash.c	2014-11-03 10:18:30.188662700 +0900
@@ -28,8 +28,6 @@ gboolean crashfunc (gpointer data) { *(i
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	IndicateIndicator * indicator = indicate_indicator_new();
 	indicate_indicator_show(indicator);
 
--- origsrc/libindicate-12.10.1/examples/listen-and-print.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/examples/listen-and-print.c	2014-11-03 10:18:40.854678400 +0900
@@ -128,8 +128,6 @@ server_removed (IndicateListener * liste
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	IndicateListener * listener = indicate_listener_ref_default();
 
 	g_signal_connect(listener, INDICATE_LISTENER_SIGNAL_INDICATOR_ADDED, G_CALLBACK(indicator_added), NULL);
--- origsrc/libindicate-12.10.1/examples/show-hide-server.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/examples/show-hide-server.c	2014-11-03 10:18:46.714053400 +0900
@@ -29,8 +29,6 @@ timeout_cb (gpointer data)
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	IndicateServer * server = indicate_server_ref_default();
 	indicate_server_set_type(server, "message.im");
 	indicate_server_set_desktop_file(server, "/usr/share/applications/empathy.desktop");
--- origsrc/libindicate-12.10.1/tests/Makefile.am	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/Makefile.am	2014-11-02 23:36:24.659297900 +0900
@@ -2,7 +2,7 @@
 TESTS =
 check_PROGRAMS =
 DISTCLEANFILES = $(TESTS)
-DBUS_RUNNER=dbus-test-runner
+DBUS_RUNNER=dbus-test-runner -d /usr/share/dbus-test-runner/session.conf
 
 ##########################
 # test simple
--- origsrc/libindicate-12.10.1/tests/test-icon-theme-client.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-icon-theme-client.c	2014-11-03 10:18:55.115420500 +0900
@@ -16,8 +16,6 @@ done_timeout_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	g_debug("Starting client");
 	IndicateIndicator * indicator = indicate_indicator_new();
 
--- origsrc/libindicate-12.10.1/tests/test-icon-theme-server.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-icon-theme-server.c	2014-11-03 10:19:01.537295500 +0900
@@ -37,8 +37,6 @@ failed_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateListener * listener = indicate_listener_ref_default();
 
 	g_signal_connect(listener, INDICATE_LISTENER_SIGNAL_SERVER_ADDED, G_CALLBACK(server_added), NULL);
--- origsrc/libindicate-12.10.1/tests/test-indicator-display-client.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-indicator-display-client.c	2014-11-03 10:19:08.302920500 +0900
@@ -52,8 +52,6 @@ idle_func (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	g_idle_add(idle_func, NULL);
 	g_timeout_add(MAX_INDICATOR_COUNT * 10, done_timeout_cb, NULL);
 
--- origsrc/libindicate-12.10.1/tests/test-indicator-display-half-client.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-indicator-display-half-client.c	2014-11-03 10:19:13.662295500 +0900
@@ -52,8 +52,6 @@ idle_func (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	g_idle_add(idle_func, NULL);
 	g_timeout_add(MAX_INDICATOR_COUNT * 10, done_timeout_cb, NULL);
 
--- origsrc/libindicate-12.10.1/tests/test-indicator-display-half-server.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-indicator-display-half-server.c	2014-11-03 10:19:19.037295500 +0900
@@ -39,8 +39,6 @@ server_removed (void)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateListener * listener = indicate_listener_ref_default();
 
 	g_signal_connect(listener, INDICATE_LISTENER_SIGNAL_INDICATOR_ADDED, G_CALLBACK(indicator_added), NULL);
--- origsrc/libindicate-12.10.1/tests/test-indicator-display-server.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-indicator-display-server.c	2014-11-03 10:19:24.381045500 +0900
@@ -32,8 +32,6 @@ server_removed (void)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateListener * listener = indicate_listener_ref_default();
 
 	g_signal_connect(listener, INDICATE_LISTENER_SIGNAL_INDICATOR_ADDED, G_CALLBACK(indicator_added), NULL);
--- origsrc/libindicate-12.10.1/tests/test-interests-client.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-interests-client.c	2014-11-03 10:19:29.693545500 +0900
@@ -46,8 +46,6 @@ done_timeout_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateIndicator * indicator = indicate_indicator_new();
 	indicate_indicator_show(indicator);
 
--- origsrc/libindicate-12.10.1/tests/test-interests-server.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-interests-server.c	2014-11-03 10:19:35.490420500 +0900
@@ -30,8 +30,6 @@ failed_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateListener * listener = indicate_listener_ref_default();
 
 	g_signal_connect(listener, INDICATE_LISTENER_SIGNAL_SERVER_ADDED, G_CALLBACK(server_added), NULL);
--- origsrc/libindicate-12.10.1/tests/test-interests-server1.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-interests-server1.c	2014-11-03 10:19:41.459170500 +0900
@@ -26,8 +26,6 @@ failed_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateListener * listener = indicate_listener_ref_default();
 
 	g_signal_connect(listener, INDICATE_LISTENER_SIGNAL_SERVER_ADDED, G_CALLBACK(server_added), NULL);
--- origsrc/libindicate-12.10.1/tests/test-interests-server2.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-interests-server2.c	2014-11-03 10:19:47.396670500 +0900
@@ -26,8 +26,6 @@ failed_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateListener * listener = indicate_listener_ref_default();
 
 	g_signal_connect(listener, INDICATE_LISTENER_SIGNAL_SERVER_ADDED, G_CALLBACK(server_added), NULL);
--- origsrc/libindicate-12.10.1/tests/test-interests-server3.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-interests-server3.c	2014-11-03 10:19:52.084170500 +0900
@@ -26,8 +26,6 @@ failed_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateListener * listener = indicate_listener_ref_default();
 
 	g_signal_connect(listener, INDICATE_LISTENER_SIGNAL_SERVER_ADDED, G_CALLBACK(server_added), NULL);
--- origsrc/libindicate-12.10.1/tests/test-interests-server4.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-interests-server4.c	2014-11-03 10:19:57.271670500 +0900
@@ -26,8 +26,6 @@ failed_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateListener * listener = indicate_listener_ref_default();
 
 	g_signal_connect(listener, INDICATE_LISTENER_SIGNAL_SERVER_ADDED, G_CALLBACK(server_added), NULL);
--- origsrc/libindicate-12.10.1/tests/test-interests-server5.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-interests-server5.c	2014-11-03 10:20:02.037295500 +0900
@@ -26,8 +26,6 @@ failed_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateListener * listener = indicate_listener_ref_default();
 
 	g_signal_connect(listener, INDICATE_LISTENER_SIGNAL_SERVER_ADDED, G_CALLBACK(server_added), NULL);
--- origsrc/libindicate-12.10.1/tests/test-max-indicators-client.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-max-indicators-client.c	2014-11-03 10:20:08.896670500 +0900
@@ -40,8 +40,6 @@ done_timeout_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateIndicator * indicator = indicate_indicator_new();
 	indicate_indicator_show(indicator);
 
--- origsrc/libindicate-12.10.1/tests/test-max-indicators-server-repeat.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-max-indicators-server-repeat.c	2014-11-03 10:20:23.203311200 +0900
@@ -62,8 +62,6 @@ failed_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateListener * listener = indicate_listener_ref_default();
 
 	g_signal_connect(listener, INDICATE_LISTENER_SIGNAL_SERVER_ADDED, G_CALLBACK(server_added), NULL);
--- origsrc/libindicate-12.10.1/tests/test-max-indicators-server.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-max-indicators-server.c	2014-11-03 10:20:18.312686200 +0900
@@ -54,8 +54,6 @@ failed_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateListener * listener = indicate_listener_ref_default();
 
 	g_signal_connect(listener, INDICATE_LISTENER_SIGNAL_SERVER_ADDED, G_CALLBACK(server_added), NULL);
--- origsrc/libindicate-12.10.1/tests/test-menu-client.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-menu-client.c	2014-11-03 10:20:29.282412700 +0900
@@ -16,8 +16,6 @@ done_timeout_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	g_debug("Starting client");
 	IndicateIndicator * indicator = indicate_indicator_new();
 
--- origsrc/libindicate-12.10.1/tests/test-menu-server.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-menu-server.c	2014-11-03 10:20:34.548037700 +0900
@@ -37,8 +37,6 @@ failed_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateListener * listener = indicate_listener_ref_default();
 
 	g_signal_connect(listener, INDICATE_LISTENER_SIGNAL_SERVER_ADDED, G_CALLBACK(server_added), NULL);
--- origsrc/libindicate-12.10.1/tests/test-properties-client.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-properties-client.c	2014-11-03 10:20:41.563662700 +0900
@@ -16,8 +16,6 @@ done_timeout_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateIndicator * indicator = indicate_indicator_new();
 
 	indicate_indicator_set_property(indicator,      "string-value",   "my-value");
--- origsrc/libindicate-12.10.1/tests/test-properties-server.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-properties-server.c	2014-11-03 10:20:48.266787700 +0900
@@ -80,8 +80,6 @@ failed_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateListener * listener = indicate_listener_ref_default();
 
 	g_signal_connect(listener, INDICATE_LISTENER_SIGNAL_INDICATOR_ADDED, G_CALLBACK(indicator_added), NULL);
--- origsrc/libindicate-12.10.1/tests/test-simple-client.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-simple-client.c	2014-11-03 10:20:53.579287700 +0900
@@ -16,8 +16,6 @@ done_timeout_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateIndicator * indicator = indicate_indicator_new();
 	indicate_indicator_show(indicator);
 
--- origsrc/libindicate-12.10.1/tests/test-simple-server.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-simple-server.c	2014-11-03 10:20:58.766787700 +0900
@@ -28,7 +28,6 @@ failed_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
 	// HACK: This just works around a bug
 	indicate_server_ref_default ();
 	
--- origsrc/libindicate-12.10.1/tests/test-thousand-indicators-client.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-thousand-indicators-client.c	2014-11-03 10:21:04.048037700 +0900
@@ -16,8 +16,6 @@ done_timeout_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	int i;
 	for (i = 0; i < 1000; i++) {
 		/* Memory leak :) */
--- origsrc/libindicate-12.10.1/tests/test-thousand-indicators-server.c	2012-07-12 02:59:02.000000000 +0900
+++ src/libindicate-12.10.1/tests/test-thousand-indicators-server.c	2014-11-03 10:21:09.626162700 +0900
@@ -28,8 +28,6 @@ failed_cb (gpointer data)
 int
 main (int argc, char * argv)
 {
-	g_type_init();
-
 	IndicateListener * listener = indicate_listener_ref_default();
 
 	g_signal_connect(listener, INDICATE_LISTENER_SIGNAL_INDICATOR_ADDED, G_CALLBACK(indicator_added), NULL);
