--- origsrc/cpmtools-2.24/Makefile.in	2023-01-31 04:32:25.000000000 +0900
+++ src/cpmtools-2.24/Makefile.in	2023-03-25 11:08:34.882538400 +0900
@@ -41,29 +41,29 @@ all:		$(ALL)
 LibDsk/libdsk.a:
 		cd LibDsk && make
 
-cpmls$(EXEEXT):		cpmls$(OBJEXT) cpmfs$(OBJEXT) getopt$(OBJEXT) getopt1$(OBJEXT) $(DEVICEOBJ) $(LDDEPS)
-		$(CC) $(LDFLAGS) -o $@ cpmls$(OBJEXT) cpmfs$(OBJEXT) getopt$(OBJEXT) getopt1$(OBJEXT) $(DEVICEOBJ) $(LIBS)
+cpmls$(EXEEXT):		cpmls$(OBJEXT) cpmfs$(OBJEXT) $(DEVICEOBJ) $(LDDEPS)
+		$(CC) $(LDFLAGS) -o $@ cpmls$(OBJEXT) cpmfs$(OBJEXT) $(DEVICEOBJ) $(LIBS)
 
-cpmrm$(EXEEXT):		cpmrm$(OBJEXT) cpmfs$(OBJEXT) getopt$(OBJEXT) getopt1$(OBJEXT) $(DEVICEOBJ) $(LDDEPS)
-		$(CC) $(LDFLAGS) -o $@ cpmrm$(OBJEXT) cpmfs$(OBJEXT) getopt$(OBJEXT) getopt1$(OBJEXT) $(DEVICEOBJ) $(LIBS)
+cpmrm$(EXEEXT):		cpmrm$(OBJEXT) cpmfs$(OBJEXT) $(DEVICEOBJ) $(LDDEPS)
+		$(CC) $(LDFLAGS) -o $@ cpmrm$(OBJEXT) cpmfs$(OBJEXT) $(DEVICEOBJ) $(LIBS)
 
-cpmcp$(EXEEXT):		cpmcp$(OBJEXT) cpmfs$(OBJEXT) getopt$(OBJEXT) getopt1$(OBJEXT) $(DEVICEOBJ) filename_posix$(OBJEXT) $(LDDEPS)
-		$(CC) $(LDFLAGS) -o $@ cpmcp$(OBJEXT) cpmfs$(OBJEXT) getopt$(OBJEXT) getopt1$(OBJEXT) $(DEVICEOBJ) filename_posix$(OBJEXT) $(LIBS)
+cpmcp$(EXEEXT):		cpmcp$(OBJEXT) cpmfs$(OBJEXT) $(DEVICEOBJ) filename_posix$(OBJEXT) $(LDDEPS)
+		$(CC) $(LDFLAGS) -o $@ cpmcp$(OBJEXT) cpmfs$(OBJEXT) $(DEVICEOBJ) filename_posix$(OBJEXT) $(LIBS)
 
-cpmchmod$(EXEEXT):	cpmchmod$(OBJEXT) cpmfs$(OBJEXT) getopt$(OBJEXT) getopt1$(OBJEXT) $(DEVICEOBJ) $(LDDEPS)
-		$(CC) $(LDFLAGS) -o $@ cpmchmod$(OBJEXT) cpmfs$(OBJEXT) getopt$(OBJEXT) getopt1$(OBJEXT) $(DEVICEOBJ) $(LIBS)
+cpmchmod$(EXEEXT):	cpmchmod$(OBJEXT) cpmfs$(OBJEXT) $(DEVICEOBJ) $(LDDEPS)
+		$(CC) $(LDFLAGS) -o $@ cpmchmod$(OBJEXT) cpmfs$(OBJEXT) $(DEVICEOBJ) $(LIBS)
 
-cpmchattr$(EXEEXT):	cpmchattr$(OBJEXT) cpmfs$(OBJEXT) getopt$(OBJEXT) getopt1$(OBJEXT) $(DEVICEOBJ) $(LDDEPS)
-		$(CC) $(LDFLAGS) -o $@ cpmchattr$(OBJEXT) cpmfs$(OBJEXT) getopt$(OBJEXT) getopt1$(OBJEXT) $(DEVICEOBJ) $(LIBS)
+cpmchattr$(EXEEXT):	cpmchattr$(OBJEXT) cpmfs$(OBJEXT) $(DEVICEOBJ) $(LDDEPS)
+		$(CC) $(LDFLAGS) -o $@ cpmchattr$(OBJEXT) cpmfs$(OBJEXT) $(DEVICEOBJ) $(LIBS)
 
-mkfs.cpm$(EXEEXT):	mkfs.cpm$(OBJEXT) cpmfs$(OBJEXT) getopt$(OBJEXT) getopt1$(OBJEXT) $(DEVICEOBJ) $(LDDEPS)
-		$(CC) $(LDFLAGS) -o $@ mkfs.cpm$(OBJEXT) cpmfs$(OBJEXT) getopt$(OBJEXT) getopt1$(OBJEXT) $(DEVICEOBJ) $(LIBS)
+mkfs.cpm$(EXEEXT):	mkfs.cpm$(OBJEXT) cpmfs$(OBJEXT) $(DEVICEOBJ) $(LDDEPS)
+		$(CC) $(LDFLAGS) -o $@ mkfs.cpm$(OBJEXT) cpmfs$(OBJEXT) $(DEVICEOBJ) $(LIBS)
 
-fsck.cpm$(EXEEXT):	fsck.cpm$(OBJEXT) cpmfs$(OBJEXT) getopt$(OBJEXT) getopt1$(OBJEXT) $(DEVICEOBJ) $(LDDEPS)
-		$(CC) $(LDFLAGS) -o $@ fsck.cpm$(OBJEXT) cpmfs$(OBJEXT) getopt$(OBJEXT) getopt1$(OBJEXT) $(DEVICEOBJ) $(LIBS)
+fsck.cpm$(EXEEXT):	fsck.cpm$(OBJEXT) cpmfs$(OBJEXT) $(DEVICEOBJ) $(LDDEPS)
+		$(CC) $(LDFLAGS) -o $@ fsck.cpm$(OBJEXT) cpmfs$(OBJEXT) $(DEVICEOBJ) $(LIBS)
 
-fsed.cpm$(EXEEXT):	fsed.cpm$(OBJEXT) cpmfs$(OBJEXT) term_curses$(OBJEXT) getopt$(OBJEXT) getopt1$(OBJEXT) $(DEVICEOBJ) $(LDDEPS)
-		$(CC) $(LDFLAGS) -o $@ fsed.cpm$(OBJEXT) cpmfs$(OBJEXT) term_curses$(OBJEXT) getopt$(OBJEXT) getopt1$(OBJEXT) $(DEVICEOBJ) $(LIBS)
+fsed.cpm$(EXEEXT):	fsed.cpm$(OBJEXT) cpmfs$(OBJEXT) term_curses$(OBJEXT) $(DEVICEOBJ) $(LDDEPS)
+		$(CC) $(LDFLAGS) -o $@ fsed.cpm$(OBJEXT) cpmfs$(OBJEXT) term_curses$(OBJEXT) $(DEVICEOBJ) $(LIBS)
 
 fsck.test:	fsck.cpm
 		-./fsck.cpm -f ibm-3740 -n badfs/status
@@ -115,4 +115,4 @@ tar:		distclean
 depend:
 		$(MAKEDEPEND) $(CPPFLAGS) *.c >.depend
 
-include .depend
+include @srcdir@/.depend
--- origsrc/cpmtools-2.24/configure.in	2023-01-28 06:07:48.000000000 +0900
+++ src/cpmtools-2.24/configure.in	2023-03-25 11:09:15.450575400 +0900
@@ -50,7 +50,7 @@ AC_CHECK_LIB(curses, printw, FSED_CPM=fs
 if test x"$FSED_CPM" = x""; then
   AC_CHECK_LIB(ncurses, printw, FSED_CPM=fsed.cpm LIBS="-lncurses $LIBS", FSED_CPM=)
   if test x"$FSED_CPM" != x""; then
-    AC_DEFINE(NEED_NCURSES)
+    AC_DEFINE([NEED_NCURSES],,[need ncurses])
     AC_CHECK_HEADERS(ncurses/ncurses.h,have_ncurses_ncurses_h=yes)
   fi
 fi
--- origsrc/cpmtools-2.24/cpm.5	2023-02-19 07:59:02.000000000 +0900
+++ src/cpmtools-2.24/cpm.5	2023-03-25 11:15:25.387876000 +0900
@@ -1,7 +1,7 @@
 .\" Believe it or not, reportedly there are nroffs which do not know \(en
 .if n .ds en -
 .if t .ds en \(en
-.TH CPM 5 "January 15, 2023" "CP/M tools" "File formats"
+.TH CPM 5 "January 27, 2023" "CP/M tools" "File formats"
 .SH NAME \"{{{roff}}}\"{{{
 cpm \- CP/M disk and file system format
 .\"}}}
--- origsrc/cpmtools-2.24/cpmchattr.c	2023-01-17 04:24:54.000000000 +0900
+++ src/cpmtools-2.24/cpmchattr.c	2023-03-25 11:09:47.142693900 +0900
@@ -2,11 +2,12 @@
 #include "config.h"
 
 #include <ctype.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
 
-#include "getopt_.h"
+#include <getopt.h>
 #include "cpmfs.h"
 /*}}}*/
 
--- origsrc/cpmtools-2.24/cpmchmod.c	2023-01-17 04:24:54.000000000 +0900
+++ src/cpmtools-2.24/cpmchmod.c	2023-03-25 11:10:08.804283700 +0900
@@ -3,11 +3,12 @@
 
 #include <ctype.h>
 #include <errno.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
 
-#include "getopt_.h"
+#include <getopt.h>
 #include "cpmfs.h"
 /*}}}*/
 
--- origsrc/cpmtools-2.24/cpmcp.c	2023-02-07 02:30:49.000000000 +0900
+++ src/cpmtools-2.24/cpmcp.c	2023-03-25 11:10:27.377914800 +0900
@@ -5,13 +5,14 @@
 #include <sys/types.h>
 #include <ctype.h>
 #include <errno.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <limits.h>
 #include <stdlib.h>
 #include <time.h>
 
-#include "getopt_.h"
+#include <getopt.h>
 #include "cpmfs.h"
 #include "filename.h"
 /*}}}*/
--- origsrc/cpmtools-2.24/cpmfs.c	2023-02-19 07:57:54.000000000 +0900
+++ src/cpmtools-2.24/cpmfs.c	2023-03-25 11:10:54.176602700 +0900
@@ -5,10 +5,13 @@
 #include <assert.h>
 #include <ctype.h>
 #include <errno.h>
+#include <fcntl.h>
+#include <limits.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 #include <time.h>
+#include <unistd.h>
 
 #include "cpmdir.h"
 #include "cpmfs.h"
--- origsrc/cpmtools-2.24/cpmfs.h	2023-01-28 05:51:56.000000000 +0900
+++ src/cpmtools-2.24/cpmfs.h	2023-03-25 11:28:39.543341600 +0900
@@ -5,6 +5,9 @@
 extern "C" {
 #endif
 
+#include <unistd.h>
+#include <utime.h>
+
 #include "device.h"
 
 /* CP/M file attributes */
--- origsrc/cpmtools-2.24/cpmls.c	2023-01-17 04:24:54.000000000 +0900
+++ src/cpmtools-2.24/cpmls.c	2023-03-25 11:11:30.505287900 +0900
@@ -3,12 +3,13 @@
 
 #include <ctype.h>
 #include <errno.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <time.h>
 
-#include "getopt_.h"
+#include <getopt.h>
 #include "cpmfs.h"
 /*}}}*/
 
--- origsrc/cpmtools-2.24/cpmrm.c	2023-01-17 04:24:54.000000000 +0900
+++ src/cpmtools-2.24/cpmrm.c	2023-03-25 11:11:46.330909000 +0900
@@ -3,11 +3,12 @@
 
 #include <ctype.h>
 #include <errno.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
 
-#include "getopt_.h"
+#include <getopt.h>
 #include "cpmfs.h"
 /*}}}*/
 
--- origsrc/cpmtools-2.24/device.h	2023-01-17 04:24:54.000000000 +0900
+++ src/cpmtools-2.24/device.h	2023-03-25 11:12:11.861087000 +0900
@@ -8,6 +8,12 @@
 #define CPMDRV_WINNT 2 /* Windows NT floppy drive accessed via CreateFile */
 #endif
 
+#include <sys/types.h>
+
+#if defined(HAVE_LIBDSK_H)
+#include <libdsk.h>
+#endif
+
 struct Device
 {
   int opened;
--- origsrc/cpmtools-2.24/diskdefs.5	2023-02-19 07:59:02.000000000 +0900
+++ src/cpmtools-2.24/diskdefs.5	2023-03-25 11:15:35.569024700 +0900
@@ -1,7 +1,7 @@
 .\" Believe it or not, reportedly there are nroffs which do not know \(en
 .if n .ds en -
 .if t .ds en \(en
-.TH DISKDEFS 5 "January 15, 2023" "CP/M tools" "File formats"
+.TH DISKDEFS 5 "January 27, 2023" "CP/M tools" "File formats"
 .SH NAME \"{{{roff}}}\"{{{
 diskdefs \- CP/M disk and file system format definitions
 .\"}}}
--- origsrc/cpmtools-2.24/fsck.cpm.c	2023-01-28 05:45:02.000000000 +0900
+++ src/cpmtools-2.24/fsck.cpm.c	2023-03-25 11:12:33.575197600 +0900
@@ -3,12 +3,13 @@
 
 #include <assert.h>
 #include <ctype.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
 #include <errno.h>
 
-#include "getopt_.h"
+#include <getopt.h>
 #include "cpmdir.h"
 #include "cpmfs.h"
 /*}}}*/
--- origsrc/cpmtools-2.24/fsed.cpm.c	2023-01-26 21:03:03.000000000 +0900
+++ src/cpmtools-2.24/fsed.cpm.c	2023-03-25 11:12:51.218182300 +0900
@@ -4,13 +4,14 @@
 #include <assert.h>
 #include <ctype.h>
 #include <errno.h>
+#include <fcntl.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 #include <time.h>
 
 #include "cpmfs.h"
-#include "getopt_.h"
+#include <getopt.h>
 #include "term.h"
 /*}}}*/
 
--- origsrc/cpmtools-2.24/mkfs.cpm.c	2023-01-17 04:24:54.000000000 +0900
+++ src/cpmtools-2.24/mkfs.cpm.c	2023-03-25 11:13:15.345870900 +0900
@@ -3,12 +3,14 @@
 
 #include <ctype.h>
 #include <errno.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <time.h>
+#include <fcntl.h>
 
-#include "getopt_.h"
+#include <getopt.h>
 #include "cpmfs.h"
 /*}}}*/
 /* #defines */ /*{{{*/
