--- origsrc/cpmtools-2.19/Makefile.in	2014-04-09 06:16:37.000000000 +0900
+++ src/cpmtools-2.19/Makefile.in	2014-07-27 10:04:53.260939200 +0900
@@ -21,7 +21,7 @@ EXEEXT =        @EXEEXT@
 OBJEXT =        .@OBJEXT@
 LIBS=           @LIBS@ @LDLIBS@
 LDDEPS=		@LDDEPS@
-CPPFLAGS=	@CPPFLAGS@ -DDISKDEFS=\"$(DISKDEFS)\" -DFORMAT=\"$(DEFFORMAT)\"
+CPPFLAGS=	-I. @CPPFLAGS@ -DDISKDEFS=\"$(DISKDEFS)\" -DFORMAT=\"$(DEFFORMAT)\"
 #
 # I'm only a novice and I haven't worked out how to autoconf this one
 #
@@ -114,4 +114,4 @@ tar:		distclean
 depend:
 		$(MAKEDEPEND) $(CPPFLAGS) *.c >.depend
 
-include .depend
+include @srcdir@/.depend
--- origsrc/cpmtools-2.19/configure.in	2014-04-27 17:11:10.000000000 +0900
+++ src/cpmtools-2.19/configure.in	2014-07-27 10:07:30.104689200 +0900
@@ -37,11 +37,6 @@ DEVICE="posix"
 
 DISKDEFS='${datarootdir}/diskdefs'
 
-if test "$CYGWIN" = "yes"
-then
-  DEVICE="win32"
-#  DISKDEFS='%USERPROFILE%/diskdefs'
-fi
 if test "$MINGW32" = "yes"
 then
   DEVICE="win32"
@@ -58,7 +53,7 @@ AC_ARG_WITH(dmalloc,  [  --with-dmalloc
 		[CPPFLAGS="$CPPFLAGS -I$with_dmalloc/include"
                  LDFLAGS="$LDFLAGS -L$with_dmalloc/lib"
                  LIBS="$LIBS -ldmalloc"
-                 AC_DEFINE(USE_DMALLOC)])
+                 AC_DEFINE([USE_DMALLOC],,[use dmalloc])])
 
 dnl Check for curses. If not found, don't build fsed.cpm
 dnl Try both curses and ncurses
@@ -66,7 +61,7 @@ AC_CHECK_LIB(curses, printw, FSED_CPM=fs
 if test x"$FSED_CPM" = x""; then
   AC_CHECK_LIB(ncurses, printw, FSED_CPM=fsed.cpm LIBS="-lncurses $LIBS", FSED_CPM=)
   if test x"$FSED_CPM" != x""; then
-    AC_DEFINE(NEED_NCURSES)
+    AC_DEFINE([NEED_NCURSES],,[need ncurses])
     AC_CHECK_HEADERS(ncurses/ncurses.h,have_ncurses_ncurses_h=yes)
   fi
 fi
--- origsrc/cpmtools-2.19/cpm.5	2014-04-09 05:56:28.000000000 +0900
+++ src/cpmtools-2.19/cpm.5	2014-07-27 10:07:57.604689200 +0900
@@ -1,7 +1,7 @@
 .\" Believe it or not, reportedly there are nroffs which do not know \(en
 .if n .ds en -
 .if t .ds en \(en
-.TH CPM 5 "April 5, 2014" "CP/M tools" "File formats"
+.TH CPM 5 "April 27, 2014" "CP/M tools" "File formats"
 .SH NAME \"{{{roff}}}\"{{{
 cpm \- CP/M disk and file system format
 .\"}}}
@@ -200,6 +200,7 @@ time stamp.
 1 byte minute in BCD format
 .sp
 .RE
+All time stamps are stored in local time.
 .\"}}}
 .SS "DateStamper time stamps" \"{{{
 The DateStamper software added functions to the BDOS to manage
--- origsrc/cpmtools-2.19/cpmchattr.c	2014-04-09 05:54:24.000000000 +0900
+++ src/cpmtools-2.19/cpmchattr.c	2014-07-27 10:12:39.760939200 +0900
@@ -2,6 +2,7 @@
 #include "config.h"
 
 #include <ctype.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
--- origsrc/cpmtools-2.19/cpmchmod.c	2014-04-06 00:21:54.000000000 +0900
+++ src/cpmtools-2.19/cpmchmod.c	2014-07-27 10:11:56.120314200 +0900
@@ -3,6 +3,7 @@
 
 #include <ctype.h>
 #include <errno.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
--- origsrc/cpmtools-2.19/cpmcp.c	2014-04-09 05:55:09.000000000 +0900
+++ src/cpmtools-2.19/cpmcp.c	2014-07-27 10:11:44.526564200 +0900
@@ -5,6 +5,7 @@
 #include <sys/types.h>
 #include <ctype.h>
 #include <errno.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <limits.h>
--- origsrc/cpmtools-2.19/cpmfs.c	2014-04-11 05:04:05.000000000 +0900
+++ src/cpmtools-2.19/cpmfs.c	2014-07-27 10:10:22.010939200 +0900
@@ -5,6 +5,8 @@
 #include <assert.h>
 #include <ctype.h>
 #include <errno.h>
+#include <fcntl.h>
+#include <limits.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
--- origsrc/cpmtools-2.19/cpmls.c	2014-04-06 00:21:54.000000000 +0900
+++ src/cpmtools-2.19/cpmls.c	2014-07-27 10:08:44.510939200 +0900
@@ -3,6 +3,7 @@
 
 #include <ctype.h>
 #include <errno.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
--- origsrc/cpmtools-2.19/cpmrm.c	2014-04-06 00:21:54.000000000 +0900
+++ src/cpmtools-2.19/cpmrm.c	2014-07-27 10:11:34.885939200 +0900
@@ -3,6 +3,7 @@
 
 #include <ctype.h>
 #include <errno.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
--- origsrc/cpmtools-2.19/device.h	2014-04-06 04:42:52.000000000 +0900
+++ src/cpmtools-2.19/device.h	2014-07-27 10:12:53.401564200 +0900
@@ -1,6 +1,16 @@
 #ifndef DEVICE_H
 #define DEVICE_H
 
+#if defined(__CYGWIN__)
+#define HAVE_WINDOWS_H 0
+#endif
+
+#include <sys/types.h>
+
+#if defined(HAVE_LIBDSK_H)
+#include <libdsk.h>
+#endif
+
 #ifdef _WIN32
 /* The type of device the file system is on: */
 #define CPMDRV_FILE  0 /* Regular file or Unix block device */
--- origsrc/cpmtools-2.19/fsck.cpm.c	2014-04-09 06:10:49.000000000 +0900
+++ src/cpmtools-2.19/fsck.cpm.c	2014-07-27 10:13:36.917189200 +0900
@@ -3,6 +3,7 @@
 
 #include <assert.h>
 #include <ctype.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
--- origsrc/cpmtools-2.19/fsed.cpm.c	2014-04-09 06:10:20.000000000 +0900
+++ src/cpmtools-2.19/fsed.cpm.c	2014-07-27 10:15:13.620314200 +0900
@@ -1,6 +1,10 @@
 /* #includes */ /*{{{C}}}*//*{{{*/
 #include "config.h"
 
+#if !defined(NEED_NCURSES)
+#define NEED_NCURSES 0
+#endif
+
 #include <assert.h>
 #include <ctype.h>
 #if NEED_NCURSES
@@ -13,6 +17,7 @@
 #include <curses.h>
 #endif
 #include <errno.h>
+#include <fcntl.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
--- origsrc/cpmtools-2.19/mkfs.cpm.c	2014-04-09 05:53:22.000000000 +0900
+++ src/cpmtools-2.19/mkfs.cpm.c	2014-07-27 10:16:15.760939200 +0900
@@ -3,10 +3,12 @@
 
 #include <ctype.h>
 #include <errno.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <time.h>
+#include <unistd.h>
 
 #include "getopt_.h"
 #include "cpmfs.h"
