--- origsrc/cpmtools-2.17/Makefile.in	2013-10-09 05:06:28.000000000 +0900
+++ src/cpmtools-2.17/Makefile.in	2013-12-01 19:40:22.712345200 +0900
@@ -21,7 +21,7 @@ EXEEXT =        @EXEEXT@
 OBJEXT =        .@OBJEXT@
 LIBS=           @LIBS@ @LDLIBS@
 LDDEPS=		@LDDEPS@
-CPPFLAGS=	@CPPFLAGS@ -DDISKDEFS=\"$(DISKDEFS)\" -DFORMAT=\"$(DEFFORMAT)\"
+CPPFLAGS=	-I. @CPPFLAGS@ -DDISKDEFS=\"$(DISKDEFS)\" -DFORMAT=\"$(DEFFORMAT)\"
 #
 # I'm only a novice and I haven't worked out how to autoconf this one
 #
@@ -113,4 +113,4 @@ tar:		distclean
 depend:
 		$(MAKEDEPEND) $(CPPFLAGS) *.c >.depend
 
-include .depend
+include @srcdir@/.depend
--- origsrc/cpmtools-2.17/configure.in	2013-11-17 05:02:44.000000000 +0900
+++ src/cpmtools-2.17/configure.in	2013-12-01 19:40:22.712345200 +0900
@@ -37,11 +37,6 @@ DEVICE="posix"
 
 DISKDEFS='${datarootdir}/diskdefs'
 
-if test "$CYGWIN" = "yes"
-then
-  DEVICE="win32"
-#  DISKDEFS='%USERPROFILE%/diskdefs'
-fi
 if test "$MINGW32" = "yes"
 then
   DEVICE="win32"
@@ -58,7 +53,7 @@ AC_ARG_WITH(dmalloc,  [  --with-dmalloc
 		[CPPFLAGS="$CPPFLAGS -I$with_dmalloc/include"
                  LDFLAGS="$LDFLAGS -L$with_dmalloc/lib"
                  LIBS="$LIBS -ldmalloc"
-                 AC_DEFINE(USE_DMALLOC)])
+                 AC_DEFINE([USE_DMALLOC],,[use dmalloc])])
 
 dnl Check for curses. If not found, don't build fsed.cpm
 dnl Try both curses and ncurses
@@ -66,6 +61,9 @@ AC_CHECK_LIB(curses, printw, FSED_CPM=fs
 if test x"$FSED_CPM" = x""; then
   AC_CHECK_LIB(ncurses, printw, FSED_CPM=fsed.cpm LIBS="-lncurses $LIBS", FSED_CPM=)
 fi
+if test x"$FSED_CPM" = x"fsed.cpm"; then
+  CPPFLAGS="$CPPFLAGS -I/usr/include/ncurses"
+fi
 
 dnl If using libdsk, check it's available. 
 if test "$LIBDSK" != ""; then
--- origsrc/cpmtools-2.17/cpmchattr.c	2013-10-09 05:06:28.000000000 +0900
+++ src/cpmtools-2.17/cpmchattr.c	2013-12-01 19:40:22.712345200 +0900
@@ -2,6 +2,7 @@
 #include "config.h"
 
 #include <ctype.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
--- origsrc/cpmtools-2.17/cpmchmod.c	2013-10-09 05:06:28.000000000 +0900
+++ src/cpmtools-2.17/cpmchmod.c	2013-12-01 19:40:22.712345200 +0900
@@ -3,6 +3,7 @@
 
 #include <ctype.h>
 #include <errno.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
--- origsrc/cpmtools-2.17/cpmfs.c	2013-11-17 05:04:10.000000000 +0900
+++ src/cpmtools-2.17/cpmfs.c	2013-12-01 19:40:22.727970200 +0900
@@ -5,9 +5,11 @@
 #include <assert.h>
 #include <ctype.h>
 #include <errno.h>
-#include <stdlib.h>
+#include <fcntl.h>
+#include <limits.h>
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 #include <time.h>
 
 #include "cpmdir.h"
--- origsrc/cpmtools-2.17/cpmls.c	2013-10-09 05:06:28.000000000 +0900
+++ src/cpmtools-2.17/cpmls.c	2013-12-01 19:48:37.732853000 +0900
@@ -3,6 +3,7 @@
 
 #include <ctype.h>
 #include <errno.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
--- origsrc/cpmtools-2.17/cpmrm.c	2013-10-09 05:06:28.000000000 +0900
+++ src/cpmtools-2.17/cpmrm.c	2013-12-01 19:40:22.727970200 +0900
@@ -3,6 +3,7 @@
 
 #include <ctype.h>
 #include <errno.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
--- origsrc/cpmtools-2.17/device.h	2013-10-09 05:06:28.000000000 +0900
+++ src/cpmtools-2.17/device.h	2013-12-01 19:51:37.014103000 +0900
@@ -8,6 +8,12 @@
 #define CPMDRV_WINNT 2 /* Windows NT floppy drive accessed via CreateFile */
 #endif
 
+#include <sys/types.h>
+
+#if defined(HAVE_LIBDSK_H)
+#include <libdsk.h>
+#endif
+
 struct Device
 {
   int opened;
--- origsrc/cpmtools-2.17/device_posix.c	2013-10-09 05:06:28.000000000 +0900
+++ src/cpmtools-2.17/device_posix.c	2013-12-01 19:40:22.727970200 +0900
@@ -3,6 +3,7 @@
 
 #include <assert.h>
 #include <errno.h>
+#include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
--- origsrc/cpmtools-2.17/fsck.cpm.c	2013-10-09 05:06:28.000000000 +0900
+++ src/cpmtools-2.17/fsck.cpm.c	2013-12-01 19:40:22.743595200 +0900
@@ -3,6 +3,7 @@
 
 #include <assert.h>
 #include <ctype.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
--- origsrc/cpmtools-2.17/fsed.cpm.c	2013-10-09 05:06:28.000000000 +0900
+++ src/cpmtools-2.17/fsed.cpm.c	2013-12-01 19:40:22.743595200 +0900
@@ -5,9 +5,11 @@
 #include <ctype.h>
 #include <curses.h>
 #include <errno.h>
+#include <fcntl.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
+#include <unistd.h>
 
 #include "cpmfs.h"
 
--- origsrc/cpmtools-2.17/mkfs.cpm.c	2013-10-09 05:06:28.000000000 +0900
+++ src/cpmtools-2.17/mkfs.cpm.c	2013-12-01 19:40:22.759220200 +0900
@@ -3,6 +3,7 @@
 
 #include <ctype.h>
 #include <errno.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
