--- origsrc/softhsm-2.6.1/configure.ac	2020-04-29 19:10:12.000000000 +0900
+++ src/softhsm-2.6.1/configure.ac	2024-09-21 18:55:19.372342200 +0900
@@ -131,7 +131,7 @@ softhsmtokendir=${full_localstatedir}/li
 # Install the library in a sub-directory
 full_libdir="$full_libdir/softhsm"
 libdir=$full_libdir
-default_softhsm2_lib="$full_libdir/libsofthsm2.so"
+default_softhsm2_lib="$full_libdir/cygsofthsm2.dll"
 
 # For getConfigPath()
 AC_CHECK_FUNCS([getpwuid_r])
--- origsrc/softhsm-2.6.1/src/lib/Makefile.am	2018-09-10 21:00:41.000000000 +0900
+++ src/softhsm-2.6.1/src/lib/Makefile.am	2024-09-21 18:55:19.387967400 +0900
@@ -24,7 +24,7 @@ libsofthsm2_la_LIBADD =		common/libsofth
 				object_store/libsofthsm_objectstore.la \
 				session_mgr/libsofthsm_sessionmgr.la \
 				slot_mgr/libsofthsm_slotmgr.la
-libsofthsm2_la_LDFLAGS =	-version-info @VERSION_INFO@ \
+libsofthsm2_la_LDFLAGS = -no-undefined -version-info @VERSION_INFO@ \
 				-avoid-version -module
 
 # Create a convenience library from all the other convenience library; this is
