--- origsrc/softhsm-1.3.2/configure.ac	2012-03-07 19:43:46.000000000 +0900
+++ src/softhsm-1.3.2/configure.ac	2012-05-03 20:39:09.571086900 +0900
@@ -134,13 +134,20 @@ AC_DEFINE_UNQUOTED(
 # Host specific configuration
 AC_CANONICAL_HOST
 case "${host}" in
-*-*-winnt* | *-*-mingw* | *-*-cygwin*)
+*-*-winnt* | *-*-mingw*)
   AC_DEFINE_UNQUOTED(
     [DEFAULT_PKCS11_LIB],
     ["$full_libdir/libsofthsm.dll"],
     [The default PKCS#11 library]
   )
   ;;
+*-*-cygwin*)
+  AC_DEFINE_UNQUOTED(
+    [DEFAULT_PKCS11_LIB],
+    ["/usr/bin/cygsofthsm.dll"],
+    [The default PKCS#11 library]
+  )
+  ;;
 *)
   AC_DEFINE_UNQUOTED(
     [DEFAULT_PKCS11_LIB],
--- origsrc/softhsm-1.3.2/src/bin/softhsm-keyconv.h	2012-03-07 19:43:46.000000000 +0900
+++ src/softhsm-1.3.2/src/bin/softhsm-keyconv.h	2012-05-03 20:35:46.263469700 +0900
@@ -29,6 +29,8 @@
 #ifndef SOFTHSM_SOFTHSM_KEYCONV_H
 #define SOFTHSM_SOFTHSM_KEYCONV_H 1
 
+#include <cstdio>
+
 #include <botan/init.h>
 #include <botan/base64.h>
 #include <botan/rsa.h>
