--- origsrc/hpenc-3.0/src/aead.cc	2018-12-10 02:51:29.000000000 +0900
+++ src/hpenc-3.0/src/aead.cc	2024-05-05 10:17:20.543529700 +0900
@@ -20,6 +20,7 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
+#include <stdexcept>
 
 #include <openssl/evp.h>
 
--- origsrc/hpenc-3.0/src/kdf.cc	2018-12-10 02:51:29.000000000 +0900
+++ src/hpenc-3.0/src/kdf.cc	2024-05-05 10:17:56.151916000 +0900
@@ -22,6 +22,7 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
+#include <stdexcept>
 #include "kdf.h"
 #include "nonce.h"
 #include "util.h"
--- origsrc/hpenc-3.0/src/main.cc	2018-12-10 02:51:29.000000000 +0900
+++ src/hpenc-3.0/src/main.cc	2024-05-05 10:20:32.704556700 +0900
@@ -23,6 +23,7 @@
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <stdexcept>
 #include "util.h"
 #include "encrypt.h"
 #include "decrypt.h"
--- origsrc/hpenc-3.0/src/util.cc	2018-12-10 02:51:29.000000000 +0900
+++ src/hpenc-3.0/src/util.cc	2024-05-05 10:19:03.705073900 +0900
@@ -23,6 +23,7 @@
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <stdexcept>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
