--- origsrc/libifp-1.0.0.2/configure.in	2005-08-22 11:46:01.000000000 +0900
+++ src/libifp-1.0.0.2/configure.in	2012-04-17 12:09:13.255221400 +0900
@@ -22,7 +22,7 @@ LIBUSB=""
 AC_MSG_CHECKING([libusb])
 AC_ARG_WITH(libusb,
 	AC_HELP_STRING([--with-libusb=PATH],[libusb path (default /usr)]),
-	ac_libusb=$enableval, ac_libusb=no)
+	ac_libusb=$with_libusb, ac_libusb=no)
 if test "x${ac_libusb}" != "xno" ; then
 		case ${with_libusb} in
 			"" | "yes" | "YES")
@@ -131,11 +131,8 @@ AC_SUBST(LIBUSB)
 AC_SUBST(CCFLAGS)
 AC_SUBST(KERNEL_SOURCE)
 AC_CONFIG_FILES(Makefile)
-AC_CONFIG_FILES(docs/Makefile)
-AC_CONFIG_FILES(dpkg/Makefile)
 AC_CONFIG_FILES([docs/doxygen.cfg:docs/doxygen.in])
 AC_CONFIG_FILES(examples/Makefile)
-AC_CONFIG_FILES(kbuild/Makefile)
 AC_CONFIG_FILES(src/Makefile)
 AC_OUTPUT
 
--- origsrc/libifp-1.0.0.2/src/Makefile.am	2005-02-21 05:28:53.000000000 +0900
+++ src/libifp-1.0.0.2/src/Makefile.am	2012-04-17 12:12:03.982760500 +0900
@@ -1,6 +1,7 @@
 lib_LTLIBRARIES = libifp.la
 libifp_la_SOURCES = prim.c comp.c readwrite.c userfile.c ifp_os_libusb.c error_strings.c tuner.c
 libifp_la_LDFLAGS = -version-info 5:4:1 -no-undefined $(LIBUSB)
+libifp_la_LIBADD = -liconv
 
 include_HEADERS = $(top_srcdir)/include/ifp.h
 
