--- origsrc/rudesocket-1.3.0/Makefile.am	2008-02-07 01:03:04.000000000 +0900
+++ src/rudesocket-1.3.0/Makefile.am	2012-03-22 18:47:14.187096200 +0900
@@ -49,8 +49,8 @@ librudesocket_la_SOURCES = 	src/rude_soc
 	src/socket_tcpclient.cpp \
 	src/socket_tcpclient.h
 
-librudesocket_la_LDFLAGS = -version-info 1:3:0
-librudesocket_la_LIBADD = -lssl
+librudesocket_la_LDFLAGS = -version-info 1:3:0 -no-undefined
+librudesocket_la_LIBADD = ${OPENSSL_LIBS}
 
 
 pkginclude_HEADERS = src/socket.h
--- origsrc/rudesocket-1.3.0/configure.ac	2008-02-07 00:57:23.000000000 +0900
+++ src/rudesocket-1.3.0/configure.ac	2012-03-22 18:47:48.765221200 +0900
@@ -54,12 +54,10 @@ AC_ARG_WITH(openssl,
 [  --with-openssl enable SSL support [will use openssl] 
 ],
 [   AC_MSG_RESULT(yes)
-
-	AC_CHECK_LIB(ssl, main,, AC_MSG_ERROR(RudeSocket requires libssl from http://www.openssl.org/))
+PKG_CHECK_MODULES(OPENSSL, openssl)
 	printf "Using openssl";
    CFLAGS="$CFLAGS -DUSING_OPENSSL";
    CXXFLAGS="$CXXFLAGS -DUSING_OPENSSL";
-	LIBS="$LIBS -lssl";
 	USING_OPENSSL=yes
 	AC_SUBST(USING_OPENSSL)
 ],
