--- origsrc/bax2bam-0.0.9/meson.build	2018-06-03 07:12:01.000000000 +0900
+++ src/bax2bam-0.0.9/meson.build	2020-12-15 20:05:39.656672800 +0900
@@ -5,7 +5,6 @@ project(
   default_options : [
     'buildtype=release',
     'warning_level=3',
-    'cpp_std=c++11',
     'b_ndebug=false'],
   license : 'BSD-3',
   meson_version : '>= 0.46.0')
@@ -52,7 +51,10 @@ bax2bam_libblasr_dep = dependency('libbl
 # zlib
 bax2bam_zlib_dep = dependency('zlib', required : true)
 
-bax2bam_deps = [bax2bam_boost_dep, bax2bam_libblasr_dep, bax2bam_zlib_dep]
+# hdf5_cpp
+bax2bam_hdf5_cpp_dep = cpp.find_library('hdf5_cpp', required: true)
+
+bax2bam_deps = [bax2bam_boost_dep, bax2bam_libblasr_dep, bax2bam_zlib_dep, bax2bam_hdf5_cpp_dep]
 
 ########################
 # sources + executable #
