--- origsrc/calcurse-4.8.1/configure.ac	2023-04-15 11:59:10.000000000 +0900
+++ src/calcurse-4.8.1/configure.ac	2023-07-26 00:16:43.892321200 +0900
@@ -2,7 +2,7 @@
 #                                                                           Init
 #-------------------------------------------------------------------------------
 AC_PREREQ([2.71])
-AC_INIT([calcurse],[m4_esyscmd(build-aux/git-version-gen .version)],[bugs@calcurse.org])
+AC_INIT([calcurse],[4.8.1],[bugs@calcurse.org])
 AM_INIT_AUTOMAKE
 #m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
 AM_GNU_GETTEXT([external])
