--- origsrc/calcurse-4.8.2/configure.ac	2025-08-08 05:20:30.000000000 +0900
+++ src/calcurse-4.8.2/configure.ac	2026-01-01 13:04:38.422802400 +0900
@@ -2,7 +2,7 @@
 #                                                                           Init
 #-------------------------------------------------------------------------------
 AC_PREREQ([2.71])
-AC_INIT([calcurse],[m4_esyscmd(build-aux/git-version-gen .version)],[bugs@calcurse.org])
+AC_INIT([calcurse],[4.8.2],[bugs@calcurse.org])
 AM_INIT_AUTOMAKE
 #m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
 AM_GNU_GETTEXT([external])
