--- origsrc/Shallot-shallot-0.0.7/Makefile	2017-09-19 16:30:56.000000000 +0900
+++ src/Shallot-shallot-0.0.7/Makefile	2024-05-08 16:26:46.668730000 +0900
@@ -21,8 +21,8 @@ WARNING_FLAGS:=$(WARNING_FLAGS) -Wall
 DEBUG_FLAGS:=$(DEBUG_FLAGS) -g
 
 
-OBJS:=src/math.o src/error.o src/linux.o src/print.o src/thread.o src/shallot.o
-OBJS_DBG:=src/math.dbg.o src/error.dbg.o src/linux.dbg.o src/print.dbg.o src/thread.dbg.o src/shallot.dbg.o
+OBJS:=src/math.o src/error.o src/linux.o src/print.o src/thread.o src/shallot.o src/globals.o
+OBJS_DBG:=src/math.dbg.o src/error.dbg.o src/linux.dbg.o src/print.dbg.o src/thread.dbg.o src/shallot.dbg.o src/globals.dbg.o
 
 PREFIX ?= /usr/bin
 BINDIR ?= $(DESTDIR)$(PREFIX)
--- origsrc/Shallot-shallot-0.0.7/src/globals.c	1970-01-01 09:00:00.000000000 +0900
+++ src/Shallot-shallot-0.0.7/src/globals.c	2024-05-08 16:29:16.289848100 +0900
@@ -0,0 +1,9 @@
+#include <regex.h>
+#include <stdint.h>
+#include <pthread.h>
+
+uint64_t loop, elim;
+uint8_t found, monitor, maxexectime;
+pthread_t lucky_thread;
+regex_t *regex;
+
--- origsrc/Shallot-shallot-0.0.7/src/globals.h	2017-09-19 16:30:56.000000000 +0900
+++ src/Shallot-shallot-0.0.7/src/globals.h	2024-05-08 16:26:00.648971800 +0900
@@ -10,9 +10,9 @@
 // global variables (saves us the trouble of passing pointers around)
 // TODO: stop being lazy and pass this stuff where it's needed
 // TODO: also put settings in a struct
-uint64_t loop, elim;
-uint8_t found, monitor, maxexectime;
-pthread_t lucky_thread;
-regex_t *regex;
+extern uint64_t loop, elim;
+extern uint8_t found, monitor, maxexectime;
+extern pthread_t lucky_thread;
+extern regex_t *regex;
 
 #endif
