--- origsrc/granite-0.4.1/lib/CMakeLists.txt	2017-06-03 14:31:57.000000000 +0900
+++ src/granite-0.4.1/lib/CMakeLists.txt	2017-08-11 14:41:26.815651000 +0900
@@ -114,7 +114,7 @@ set_target_properties (${PKG_NAME} PROPE
     VERSION ${PKG_SOVERSION}.${PKG_SOMINOR}
     SOVERSION ${PKG_SOVERSION})
 
-install (TARGETS ${PKG_NAME} DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR})
+install (TARGETS ${PKG_NAME} RUNTIME DESTINATION bin ARCHIVE DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR})
 
 ###########################
 # GObject Introspection
