--- origsrc/granite-0.2.1/lib/CMakeLists.txt	2013-07-20 21:40:22.000000000 +0900
+++ src/granite-0.2.1/lib/CMakeLists.txt	2013-07-28 15:44:29.980765600 +0900
@@ -104,7 +104,7 @@ set_target_properties (${PKG_NAME} PROPE
     VERSION ${PKG_SOVERSION}.${PKG_SOMINOR}
     SOVERSION ${PKG_SOVERSION})
 
-install (TARGETS ${PKG_NAME} DESTINATION ${LIB_INSTALL_DIR})
+install (TARGETS ${PKG_NAME} RUNTIME DESTINATION bin ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
 
 ###########################
 # GObject Introspection
