--- origsrc/QHotkey-1.4.2/CMakeLists.txt	2020-10-21 14:10:48.000000000 +0900
+++ src/QHotkey-1.4.2/CMakeLists.txt	2021-08-25 16:30:25.231583000 +0900
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.1)
 
-project(qhotkey VERSION 1.2.2 LANGUAGES CXX)
+project(qhotkey VERSION 1.4.2 LANGUAGES CXX)
 
 option(QHOTKEY_EXAMPLES "Build examples" OFF)
 option(QHOTKEY_INSTALL "Enable install rule" ON)
@@ -71,3 +71,10 @@ if(QHOTKEY_INSTALL)
 
     export(TARGETS qhotkey FILE QHotkeyConfig.cmake)
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/QHotkey.pc.in ${CMAKE_BINARY_DIR}/QHotkey.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/QHotkey.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/QHotkey-1.4.2/QHotkey.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/QHotkey-1.4.2/QHotkey.pc.in	2021-08-25 16:30:33.892800600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: QHotkey
+Description: Global shortcut/hotkey for Desktop Qt-Applications
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lqhotkey
+Cflags: -I${includedir}
