--- origsrc/libvmi-0.12.0/configure.ac	2018-04-23 09:05:43.000000000 +0900
+++ src/libvmi-0.12.0/configure.ac	2020-06-28 05:24:53.710363400 +0900
@@ -178,6 +178,8 @@ AC_PROG_CC
 AM_PROG_CC_C_O
 AM_SANITY_CHECK
 
+AM_ICONV
+
 PKG_CHECK_MODULES([CHECK], [check >= 0.9.4], [have_check="yes"], [have_check="no"])
 PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.16],[],[AC_MSG_ERROR(GLib 2.16 or newer not found. Install missing package and re-run)])
 AC_SUBST([GLIB_CFLAGS])
--- origsrc/libvmi-0.12.0/libvmi/Makefile.am	2018-04-23 09:05:43.000000000 +0900
+++ src/libvmi-0.12.0/libvmi/Makefile.am	2020-06-28 08:08:18.494457500 +0900
@@ -123,7 +123,7 @@ lib_LTLIBRARIES= libvmi.la
 libvmi_la_SOURCES= $(h_public) $(h_private) $(drivers) $(os) $(c_sources)
 libvmi_la_CFLAGS= -fvisibility=hidden $(GLIB_CFLAGS)
 libvmi_la_CFLAGS+= -Wall -Wextra
-libvmi_la_LDFLAGS= $(GLIB_LIBS) -no-undefined
+libvmi_la_LDFLAGS= $(GLIB_LIBS) $(LTLIBICONV) -no-undefined
 libvmi_la_LDFLAGS+= -version-info $(VERSION)
 
 if CONFIGFILE
