--- origsrc/libzc-0.4.1/lib/Makefile.am	2018-06-24 11:04:16.000000000 +0900
+++ src/libzc-0.4.1/lib/Makefile.am	2018-07-07 10:51:51.656350500 +0900
@@ -29,7 +29,7 @@ AM_CFLAGS = \
 	-ffunction-sections \
 	-fdata-sections
 
-AM_LDFLAGS = -lm ${ZLIB_LIBS}
+AM_LDFLAGS = -no-undefined -lm ${ZLIB_LIBS}
 
 # - Start with version information of '0:0:0' for each libtool library.
 # - Update the version information only immediately before a public release
