diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/Libsibow/CMakeLists.txt src/Libsibow/CMakeLists.txt
--- origsrc/Libsibow/CMakeLists.txt	2016-02-05 16:52:19.000000000 +0900
+++ src/Libsibow/CMakeLists.txt	2016-02-05 16:56:48.565192000 +0900
@@ -86,7 +86,7 @@ ADD_CUSTOM_TARGET(uninstall "${CMAKE_COM
 # ----------------------------------------------------------------------------
 
 CONFIGURE_FILE("${PROJECT_SOURCE_DIR}/config.cmake.in" "${PROJECT_BINARY_DIR}/Find${PROJECT_NAME}.cmake")
-INSTALL(FILES "${PROJECT_BINARY_DIR}/Find${PROJECT_NAME}.cmake" DESTINATION lib/cmake/ )
+INSTALL(FILES "${PROJECT_BINARY_DIR}/Find${PROJECT_NAME}.cmake" DESTINATION lib/cmake/${PROJECT_NAME} )
 
 # ----------------------------------------------------------------------------
 #   Program Optimization and debug (Extracted from OpenCV)
@@ -267,3 +267,10 @@ MESSAGE( STATUS )
 MESSAGE( STATUS )
 MESSAGE( STATUS "Change a value with: cmake -D<Variable>=<Value>" )
 MESSAGE( STATUS )
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/${PROJECT_NAME}.pc.in ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/Libsibow/sibow.pc.in src/Libsibow/sibow.pc.in
--- origsrc/Libsibow/sibow.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/Libsibow/sibow.pc.in	2016-02-05 16:56:44.876175500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libsibow
+Description: Simple C++ library for Bag of Words
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lsibow
+Cflags: -I${includedir}/sibow
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/Libsibow/src/CMakeLists.txt src/Libsibow/src/CMakeLists.txt
--- origsrc/Libsibow/src/CMakeLists.txt	2016-02-05 16:52:19.000000000 +0900
+++ src/Libsibow/src/CMakeLists.txt	2016-02-05 16:55:23.605970000 +0900
@@ -3,7 +3,7 @@ INCLUDE (CheckIncludeFiles)
 # Make sure it finds the headers
 include_directories (${PROJECT_SOURCE_DIR}/include)
 
-FILE(GLOB hdrs "*.h*")
+FILE(GLOB hdrs "${PROJECT_SOURCE_DIR}/include/*.h*")
 FILE(GLOB srcs "*.c*")
 
 
@@ -39,4 +39,3 @@ INSTALL(TARGETS ${PROJECT_NAME}
 INSTALL(FILES ${hdrs}
 	DESTINATION include/${PROJECT_NAME}
 	COMPONENT main)
-
