--- origsrc/no-more-secrets-0.3.2/src/nmstermio.c	2017-02-20 03:29:28.000000000 +0900
+++ src/no-more-secrets-0.3.2/src/nmstermio.c	2017-08-24 13:44:47.033000000 +0900
@@ -43,6 +43,10 @@
 #define COLOR_CYAN    6
 #define COLOR_WHITE   7
 
+#if !defined(FIONREAD) && defined(TIOCINQ)
+#define FIONREAD TIOCINQ
+#endif
+
 // Terminal IO settings
 static int clearScr          = 0;                     // clearScr flag
 static int foregroundColor   = COLOR_BLUE;            // Foreground color setting
