--- origsrc/libsodium/configure.ac	2013-03-02 14:39:05.000000000 +0900
+++ src/libsodium/configure.ac	2013-03-02 14:41:28.481499900 +0900
@@ -62,7 +62,9 @@ AX_CHECK_COMPILE_FLAG([-fno-strict-overf
 LIBTOOL_EXTRA_FLAGS="-version-info $SODIUM_LIBRARY_VERSION"
 case $host_os in
   cygwin* | mingw* | pw32* | cegcc*)
-    LIBTOOL_EXTRA_FLAGS="-module -avoid-version"
+    if test "$host_os" != "cygwin"; then
+      LIBTOOL_EXTRA_FLAGS="-module -avoid-version"
+    fi
     AX_CHECK_LINK_FLAG([-Wl,--dynamicbase], [LDFLAGS="$LDFLAGS -Wl,--dynamicbase"])
     AX_CHECK_LINK_FLAG([-Wl,--nxcompat], [LDFLAGS="$LDFLAGS -Wl,--nxcompat"])
     AX_CHECK_COMPILE_FLAG([-fstack-protector-all], [
--- origsrc/libsodium/src/libsodium/Makefile.am	2013-03-02 14:39:05.000000000 +0900
+++ src/libsodium/src/libsodium/Makefile.am	2013-03-02 14:52:14.304591000 +0900
@@ -90,7 +90,8 @@ libsodium_la_LDFLAGS = \
 
 libsodium_la_CPPFLAGS = \
 	$(LTDLINCL) \
-	-I$(top_srcdir)/src/libsodium/include/sodium
+	-I$(top_srcdir)/src/libsodium/include/sodium \
+	-I$(top_builddir)/src/libsodium/include/sodium
 
 SUBDIRS = \
 	include
