--- origsrc/glyr-1.0.10/CMakeLists.txt	2016-11-12 23:51:12.000000000 +0900
+++ src/glyr-1.0.10/CMakeLists.txt	2025-01-03 22:16:07.089807100 +0900
@@ -38,7 +38,7 @@ ENDIF()
 # ------------------------------------------------
 SET(GLYR_VERSION_MAJOR "1")
 SET(GLYR_VERSION_MINOR "0")
-SET(GLYR_VERSION_MICRO "9")
+SET(GLYR_VERSION_MICRO "10")
 SET(GLYR_VERSION_NAME  "Raving Raven")
 # ------------------------------------------------
 
--- origsrc/glyr-1.0.10/lib/CMakeLists.txt	2016-11-12 23:51:12.000000000 +0900
+++ src/glyr-1.0.10/lib/CMakeLists.txt	2025-01-03 22:09:35.219781500 +0900
@@ -36,4 +36,4 @@ INSTALL(FILES misc.h    DESTINATION ${IN
 INSTALL(FILES cache.h   DESTINATION ${INSTALL_INC_DIR}/glyr)
 INSTALL(FILES config.h  DESTINATION ${INSTALL_INC_DIR}/glyr)
 INSTALL(FILES testing.h DESTINATION ${INSTALL_INC_DIR}/glyr)
-INSTALL(TARGETS glyr LIBRARY DESTINATION ${INSTALL_LIB_DIR})
+INSTALL(TARGETS glyr RUNTIME DESTINATION ${INSTALL_BIN_DIR} ARCHIVE DESTINATION ${INSTALL_LIB_DIR})
--- origsrc/glyr-1.0.10/lib/core.c	2016-11-12 23:51:12.000000000 +0900
+++ src/glyr-1.0.10/lib/core.c	2025-01-03 22:09:35.219781500 +0900
@@ -18,6 +18,7 @@
  * along with glyr. If not, see <http://www.gnu.org/licenses/>.
  **************************************************************/
 
+#include <sys/select.h>
 #include <errno.h>
 #include <curl/multi.h>
 
--- origsrc/glyr-1.0.10/lib/core.h	2016-11-12 23:51:12.000000000 +0900
+++ src/glyr-1.0.10/lib/core.h	2025-01-03 22:09:35.219781500 +0900
@@ -26,6 +26,7 @@
 #include "config.h"
 
 /* Global */
+#include <sys/select.h>
 #include <string.h>
 #include <curl/curl.h>
 #include <glib.h>
--- origsrc/glyr-1.0.10/lib/stringlib.c	2016-11-12 23:51:12.000000000 +0900
+++ src/glyr-1.0.10/lib/stringlib.c	2025-01-03 22:09:35.219781500 +0900
@@ -18,6 +18,7 @@
  * along with glyr. If not, see <http://www.gnu.org/licenses/>.
  **************************************************************/
 
+#include <sys/select.h>
 #include <string.h>
 #include <ctype.h>
 
--- origsrc/glyr-1.0.10/src/utils/ping_url.c	2016-11-12 23:51:12.000000000 +0900
+++ src/glyr-1.0.10/src/utils/ping_url.c	2025-01-03 22:09:35.235411800 +0900
@@ -18,6 +18,7 @@
  * along with glyr. If not, see <http://www.gnu.org/licenses/>.
  **************************************************************/
 
+#include <sys/select.h>
 #include <curl/curl.h>
 #include <string.h>
 #include <glib.h>
