diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/glyr-1.0.8/lib/CMakeLists.txt src/glyr-1.0.8/lib/CMakeLists.txt
--- origsrc/glyr-1.0.8/lib/CMakeLists.txt	2015-01-19 08:47:10.000000000 +0900
+++ src/glyr-1.0.8/lib/CMakeLists.txt	2015-06-13 21:03:50.789491900 +0900
@@ -30,4 +30,4 @@ INSTALL(FILES misc.h    DESTINATION ${IN
 INSTALL(FILES cache.h   DESTINATION ${INSTALL_INC_DIR}/glyr)
 INSTALL(FILES config.h  DESTINATION ${INSTALL_INC_DIR}/glyr)
 INSTALL(FILES testing.h DESTINATION ${INSTALL_INC_DIR}/glyr)
-INSTALL(TARGETS glyr LIBRARY DESTINATION ${INSTALL_LIB_DIR})
+INSTALL(TARGETS glyr RUNTIME DESTINATION ${INSTALL_BIN_DIR} ARCHIVE DESTINATION ${INSTALL_LIB_DIR})
