--- origsrc/libgap-4.7.4/src/Makefile.am	2013-12-24 23:31:01.000000000 +0900
+++ src/libgap-4.7.4/src/Makefile.am	2014-05-27 23:05:55.386828100 +0900
@@ -1,4 +1,4 @@
-DEFAULT_INCLUDES = -I@top_builddir@
+DEFAULT_INCLUDES = -I$(top_builddir) -I$(top_builddir)/src
 
 lib_LTLIBRARIES = libgap.la
 
--- origsrc/libgap-4.7.4/test/Makefile.am	2014-02-27 19:30:13.000000000 +0900
+++ src/libgap-4.7.4/test/Makefile.am	2014-05-27 23:10:12.740343700 +0900
@@ -1,4 +1,4 @@
-DEFAULT_INCLUDES = -I@top_srcdir@ -I@top_builddir@
+DEFAULT_INCLUDES = -I$(top_srcdir) -I$(top_builddir) -I$(top_builddir)/src
 
 TESTS = test shell error_handler
 
