--- origsrc/cquery/CMakeLists.txt	2026-02-15 07:21:03.000000000 +0900
+++ src/cquery/CMakeLists.txt	2026-02-15 07:37:04.981503600 +0900
@@ -141,9 +141,7 @@ target_include_directories(cquery PRIVAT
                            src
                            third_party
                            third_party/doctest
-                           third_party/loguru
                            third_party/msgpack-c/include
-                           third_party/pugixml/src
                            third_party/rapidjson/include
                            third_party/sparsepp)
 
@@ -255,9 +253,10 @@ endif()
 
 ### Sources
 
+target_link_libraries(cquery PRIVATE loguru pugixml)
+
 target_sources(cquery PRIVATE
-               third_party/siphash.cc
-               third_party/pugixml/src/pugixml.cpp)
+               third_party/siphash.cc)
 
 target_sources(cquery PRIVATE
                src/c_cpp_properties.cc
--- origsrc/cquery/src/import_manager.cc	2026-02-15 07:21:04.000000000 +0900
+++ src/cquery/src/import_manager.cc	2026-02-15 07:45:33.784657800 +0900
@@ -1,4 +1,5 @@
 #include "import_manager.h"
+#include <mutex>
 
 bool ImportManager::IsInitialImport(const std::string& path) {
   // Try reading the value
--- origsrc/cquery/src/options.h	2026-02-15 07:21:04.000000000 +0900
+++ src/cquery/src/options.h	2026-02-15 07:47:58.698760300 +0900
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <unordered_map>
+#include <string>
 
 std::unordered_map<std::string, std::string> ParseOptions(int argc,
                                                           char** argv);
--- origsrc/cquery/src/third_party_impl.cc	2026-02-15 07:21:04.000000000 +0900
+++ src/cquery/src/third_party_impl.cc	2026-02-15 07:52:44.373262800 +0900
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT
 #include <doctest/doctest.h>
 
-#define LOGURU_IMPLEMENTATION 1
+//#define LOGURU_IMPLEMENTATION 1
 #include <loguru.hpp>
--- origsrc/cquery/third_party/sparsepp/sparsepp/spp_memory.h	2026-02-15 07:21:37.000000000 +0900
+++ src/cquery/third_party/sparsepp/sparsepp/spp_memory.h	2026-02-15 07:41:31.146143600 +0900
@@ -5,7 +5,7 @@
 #include <cstring>
 #include <cstdlib>
 
-#if defined(_WIN32) || defined( __CYGWIN__)
+#if defined(_WIN32)
     #define SPP_WIN
 #endif
 
@@ -14,7 +14,7 @@
     #include <Psapi.h>
     #undef min
     #undef max
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__CYGWIN__)
     #include <sys/types.h>
     #include <sys/sysinfo.h>
 #elif defined(__FreeBSD__)
@@ -35,7 +35,7 @@ namespace spp
         memInfo.dwLength = sizeof(MEMORYSTATUSEX);
         GlobalMemoryStatusEx(&memInfo);
         return static_cast<uint64_t>(memInfo.ullTotalPageFile);
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__CYGWIN__)
         struct sysinfo memInfo;
         sysinfo (&memInfo);
         auto totalVirtualMem = memInfo.totalram;
@@ -74,7 +74,7 @@ namespace spp
         memInfo.dwLength = sizeof(MEMORYSTATUSEX);
         GlobalMemoryStatusEx(&memInfo);
         return static_cast<uint64_t>(memInfo.ullTotalPageFile - memInfo.ullAvailPageFile);
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__CYGWIN__)
         struct sysinfo memInfo;
         sysinfo(&memInfo);
         auto virtualMemUsed = memInfo.totalram - memInfo.freeram;
@@ -115,7 +115,7 @@ namespace spp
         PROCESS_MEMORY_COUNTERS_EX pmc;
         GetProcessMemoryInfo(GetCurrentProcess(), reinterpret_cast<PPROCESS_MEMORY_COUNTERS>(&pmc), sizeof(pmc));
         return static_cast<uint64_t>(pmc.PrivateUsage);
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__CYGWIN__)
         auto parseLine = 
             [](char* line)->int
             {
@@ -165,7 +165,7 @@ namespace spp
         memInfo.dwLength = sizeof(MEMORYSTATUSEX);
         GlobalMemoryStatusEx(&memInfo);
         return static_cast<uint64_t>(memInfo.ullTotalPhys);
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__CYGWIN__)
         struct sysinfo memInfo;
         sysinfo(&memInfo);
 
