--- origsrc/libgwyfile-1.6/examples/Makefile.am	2023-04-18 01:27:51.000000000 +0900
+++ src/libgwyfile-1.6/examples/Makefile.am	2024-12-08 04:08:09.559702300 +0900
@@ -17,7 +17,7 @@ noinst_PROGRAMS = \
 	checkgeneric
 
 AM_CFLAGS = $(ms_bitfields) $(WARNING_CFLAGS) -I$(top_srcdir)/lib
-AM_LDFLAGS = ../lib/libgwyfile.la -lm -all-static
+AM_LDFLAGS = ../lib/libgwyfile.la -lm
 writegwy_SOURCES = writegwy.c
 readgwy_SOURCES = readgwy.c
 readgeneric_SOURCES = readgeneric.c
--- origsrc/libgwyfile-1.6/lib/Makefile.am	2022-02-12 00:40:12.000000000 +0900
+++ src/libgwyfile-1.6/lib/Makefile.am	2024-12-08 04:09:10.923355700 +0900
@@ -1,10 +1,11 @@
 # $Id: Makefile.am 394 2022-02-11 15:40:12Z yeti-dn $
 if OS_MSWIN
-no_undefined = -no-undefined
-export_symbols = -export-symbols gwyfile.def
+export_symbols = -export-symbols $(srcdir)/gwyfile.def
 ms_bitfields = -mms-bitfields
 endif
 
+no_undefined = -no-undefined
+
 gwyfileincludedir = $(includedir)/gwyfile
 pkgconfigdatadir = $(libdir)/pkgconfig
 
