--- origsrc/libgwyfile-1.4/examples/Makefile.am	2020-06-09 01:12:32.000000000 +0900
+++ src/libgwyfile-1.4/examples/Makefile.am	2020-07-09 18:19:20.279176900 +0900
@@ -17,7 +17,7 @@ noinst_PROGRAMS = \
 	checkgeneric
 
 AM_CFLAGS = $(ms_bitfields) $(WARNING_CFLAGS) -I$(top_srcdir)/lib
-AM_LDFLAGS = ../lib/libgwyfile.la -lm -all-static
+AM_LDFLAGS = ../lib/libgwyfile.la -lm
 writegwy_SOURCES = writegwy.c
 readgwy_SOURCES = readgwy.c
 readgeneric_SOURCES = readgeneric.c
--- origsrc/libgwyfile-1.4/lib/Makefile.am	2020-06-09 01:09:41.000000000 +0900
+++ src/libgwyfile-1.4/lib/Makefile.am	2020-07-09 18:19:51.202450900 +0900
@@ -1,7 +1,7 @@
 # $Id: Makefile.am 337 2020-06-08 16:09:41Z yeti-dn $
-if OS_MSWIN
 no_undefined = -no-undefined
-export_symbols = -export-symbols gwyfile.def
+export_symbols = -export-symbols $(srcdir)/gwyfile.def
+if OS_MSWIN
 ms_bitfields = -mms-bitfields
 endif
 
--- origsrc/libgwyfile-1.4/lib/gwyfile.def	2020-06-09 01:13:44.000000000 +0900
+++ src/libgwyfile-1.4/lib/gwyfile.def	2020-07-09 18:23:34.375755200 +0900
@@ -146,6 +146,4 @@ EXPORTS
     gwyfile_object_take_with_type
     gwyfile_object_write_memory
     gwyfile_read_file
-    gwyfile_read_wfile
     gwyfile_write_file
-    gwyfile_write_wfile
