--- origsrc/simde-0.8.2/simde/simde-math.h	2024-05-01 01:35:50.000000000 +0900
+++ src/simde-0.8.2/simde/simde-math.h	2024-07-15 20:20:04.469461000 +0900
@@ -1265,7 +1265,9 @@ simde_math_fpclass(double v, const int i
 #endif
 
 #if !defined(simde_math_roundeven)
-  #if \
+  #if defined(__CYGWIN__)
+    #define simde_math_roundeven(v) (v)
+  #elif \
      ((!defined(HEDLEY_EMSCRIPTEN_VERSION) || HEDLEY_EMSCRIPTEN_VERSION_CHECK(3, 1, 43)) && HEDLEY_HAS_BUILTIN(__builtin_roundeven)) || \
       HEDLEY_GCC_VERSION_CHECK(10,0,0)
     #define simde_math_roundeven(v) __builtin_roundeven(v)
@@ -1285,7 +1287,9 @@ simde_math_fpclass(double v, const int i
 #endif
 
 #if !defined(simde_math_roundevenf)
-  #if \
+  #if defined(__CYGWIN__)
+    #define simde_math_roundevenf(v) (v)
+  #elif \
      ((!defined(HEDLEY_EMSCRIPTEN_VERSION) || HEDLEY_EMSCRIPTEN_VERSION_CHECK(3, 1, 43)) && HEDLEY_HAS_BUILTIN(__builtin_roundevenf)) || \
       HEDLEY_GCC_VERSION_CHECK(10,0,0)
     #define simde_math_roundevenf(v) __builtin_roundevenf(v)
