--- origsrc/libmsym-0.2.3/CMakeLists.txt	2015-11-26 19:15:34.000000000 +0900
+++ src/libmsym-0.2.3/CMakeLists.txt	2018-04-07 10:00:32.423000000 +0900
@@ -10,9 +10,10 @@ include (GenerateExportHeader)
 
 set(LIBMSYM_MAJOR_VERSION 0)
 set(LIBMSYM_MINOR_VERSION 2)
-set(LIBMSYM_PATCH_VERSION 2)
+set(LIBMSYM_PATCH_VERSION 3)
 set(LIBMSYM_VERSION ${LIBMSYM_MAJOR_VERSION}.${LIBMSYM_MINOR_VERSION}.${LIBMSYM_PATCH_VERSION})
 
+set(INSTALL_BIN_DIR bin CACHE PATH "Installation directory for msym executable")
 set(INSTALL_LIB_DIR lib CACHE PATH "Installation directory for msym library")
 set(INSTALL_INCLUDE_DIR include CACHE PATH "Installation directory for header files")
 
@@ -105,3 +106,10 @@ install(EXPORT libmsymTargets DESTINATIO
 if(MSYM_BUILD_EXAMPLES)
 	add_subdirectory(examples)
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libmsym.pc.in ${CMAKE_BINARY_DIR}/libmsym.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libmsym.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libmsym-0.2.3/libmsym.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libmsym-0.2.3/libmsym.pc.in	2018-04-07 09:56:26.671000000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libmsym
+Description: C library dealing with point group symmetry in molecules
+Version: @LIBMSYM_VERSION@
+Libs: -L${libdir} -lmsym
+Cflags: -I${includedir}/libmsym
