--- origsrc/libqb-2.0.6/configure.ac	2022-03-23 18:54:07.000000000 +0900
+++ src/libqb-2.0.6/configure.ac	2022-05-25 03:18:55.354411100 +0900
@@ -29,9 +29,12 @@ LT_PREREQ([2.2.6])
 # --enable-new-dtags: Use RUNPATH instead of RPATH.
 # It is necessary to have this done before libtool does linker detection.
 # See also: https://github.com/kronosnet/kronosnet/issues/107
-AX_CHECK_LINK_FLAG([-Wl,--enable-new-dtags],
-		   [AM_LDFLAGS=-Wl,--enable-new-dtags],
-		   [AC_MSG_ERROR(["Linker support for --enable-new-dtags is required"])])
+AC_CANONICAL_HOST
+if test "$host_os" != "cygwin"; then
+	AX_CHECK_LINK_FLAG([-Wl,--enable-new-dtags],
+			[AM_LDFLAGS=-Wl,--enable-new-dtags],
+			[AC_MSG_ERROR(["Linker support for --enable-new-dtags is required"])])
+fi
 AC_SUBST([AM_LDFLAGS])
 saved_LDFLAGS="$LDFLAGS"
 LDFLAGS="$AM_LDFLAGS $LDFLAGS"
--- origsrc/libqb-2.0.6/lib/Makefile.am	2022-03-23 18:54:07.000000000 +0900
+++ src/libqb-2.0.6/lib/Makefile.am	2022-05-25 03:18:55.357398500 +0900
@@ -97,6 +97,8 @@ else
 endif
 endif
 
+libqb_la_LDFLAGS += -no-undefined
+
 # nested sed expression is also escaping meta character for the outer one
 qblog_script.ld: %.ld: %.ld.in
 	$(AM_V_GEN)$(CPP) -C -D_GNU_SOURCE -P $(AM_CPPFLAGS) -xc $< \
--- origsrc/libqb-2.0.6/tests/Makefile.am	2022-03-23 18:54:07.000000000 +0900
+++ src/libqb-2.0.6/tests/Makefile.am	2022-05-25 03:18:55.361508000 +0900
@@ -175,16 +175,11 @@ tlist_test_CFLAGS = @CHECK_CFLAGS@
 tlist_test_LDADD = $(top_builddir)/lib/libqb.la @CHECK_LIBS@
 
 ipc_test_SOURCES = check_ipc.c
-ipc_test_CFLAGS = @CHECK_CFLAGS@
-ipc_test_LDADD = $(top_builddir)/lib/libqb.la @CHECK_LIBS@
+ipc_test_CFLAGS = @CHECK_CFLAGS@ $(GLIB_CFLAGS)
+ipc_test_LDADD = $(top_builddir)/lib/libqb.la @CHECK_LIBS@ $(GLIB_LIBS)
 if HAVE_FAILURE_INJECTION
 ipc_test_LDADD += _failure_injection.la
 
-if HAVE_GLIB
-ipc_test_CFLAGS += $(GLIB_CFLAGS)
-ipc_test_LDADD += $(GLIB_LIBS)
-endif
-
 check_LTLIBRARIES += _failure_injection.la
 _failure_injection_la_SOURCES = _failure_injection.c _failure_injection.h
 _failure_injection_la_LDFLAGS = -module
--- origsrc/libqb-2.0.6/tests/check_ipc.c	2022-03-23 18:54:07.000000000 +0900
+++ src/libqb-2.0.6/tests/check_ipc.c	2022-05-25 03:18:55.366394900 +0900
@@ -1998,14 +1998,18 @@ test_ipc_server_fail(void)
 	ck_assert(conn != NULL);
 
 	request_server_exit();
+#if defined(HAVE_FAILURE_INJECTION)
 	if (_fi_unlink_inject_failure == QB_TRUE) {
 		_fi_truncate_called = _fi_openat_called = 0;
 	}
+#endif
 	ck_assert_int_eq(QB_FALSE, qb_ipcc_is_connected(conn));
 	qb_ipcc_disconnect(conn);
+#if defined(HAVE_FAILURE_INJECTION)
 	if (_fi_unlink_inject_failure == QB_TRUE) {
 		ck_assert_int_ne(_fi_truncate_called + _fi_openat_called, 0);
 	}
+#endif
 	verify_graceful_stop(pid);
 }
 
--- origsrc/libqb-2.0.6/tests/libstat_wrapper.c	2022-03-23 18:54:07.000000000 +0900
+++ src/libqb-2.0.6/tests/libstat_wrapper.c	2022-05-25 03:32:24.216506200 +0900
@@ -7,14 +7,14 @@
 #include <string.h>
 #include <sys/stat.h>
 #include "../include/config.h"
-#if defined(QB_LINUX) || defined(QB_CYGWIN)
+#if defined(QB_LINUX)
 #include <gnu/lib-names.h>
 #endif
 
 // __xstat for earlier libc
 int __xstat(int __ver, const char *__filename, struct stat *__stat_buf)
 {
-#if defined(QB_LINUX) || defined(QB_CYGWIN)
+#if defined(QB_LINUX)
 	static int opened = 0;
 	static int (*real_xstat)(int __ver, const char *__filename, void *__stat_buf);
 
@@ -37,7 +37,7 @@ int __xstat(int __ver, const char *__fil
 // stat for F35 and later
 int stat(const char *__filename, struct stat *__stat_buf)
 {
-#if defined(QB_LINUX) || defined(QB_CYGWIN)
+#if defined(QB_LINUX)
 	static int opened = 0;
 	static int (*real_stat)(const char *__filename, void *__stat_buf);
 
