--- origsrc/libqb-0.14.2/lib/Makefile.am	2012-09-10 09:53:06.000000000 +0900
+++ src/libqb-0.14.2/lib/Makefile.am	2012-10-04 21:03:52.695148300 +0900
@@ -59,7 +59,7 @@ lib_LTLIBRARIES		= libqb.la
 #    it as -version-info 2:0:0. This release has a new, but backwards
 #    incompatible interface.
 #
-libqb_la_LDFLAGS	= -version-info 14:2:14
+libqb_la_LDFLAGS	= -version-info 14:2:14 -no-undefined
 
 source_to_lint		= util.c hdb.c ringbuffer.c ringbuffer_helper.c \
 			  array.c loop.c loop_poll.c loop_job.c \
