--- origsrc/libksi-3.20.3025/src/example/GNUmakefile.am	2019-10-07 17:49:34.000000000 +0900
+++ src/libksi-3.20.3025/src/example/GNUmakefile.am	2019-11-30 14:48:09.762044700 +0900
@@ -17,7 +17,7 @@
 # reserves and retains all trademark rights.
 #
 
-AM_CFLAGS=-g -Wall -I$(top_builddir)/src/
+AM_CFLAGS=-g -Wall -I$(top_builddir)/src/ -I$(top_srcdir)/src
 AM_LDFLAGS=-L$(top_builddir)/src/ksi -no-install -lksi
 check_PROGRAMS=\
 	ksi_sign \
--- origsrc/libksi-3.20.3025/src/ksi/GNUmakefile.am	2019-10-07 17:49:34.000000000 +0900
+++ src/libksi-3.20.3025/src/ksi/GNUmakefile.am	2019-11-30 14:48:09.762044700 +0900
@@ -161,5 +161,5 @@ otherinclude_HEADERS = \
 	compatibility.h \
 	version.h
 
-libksi_la_LDFLAGS=-version-info @LTVER@
+libksi_la_LDFLAGS=-no-undefined -version-info @LTVER@
 
--- origsrc/libksi-3.20.3025/test/GNUmakefile.am	2019-10-07 17:49:34.000000000 +0900
+++ src/libksi-3.20.3025/test/GNUmakefile.am	2019-11-30 14:48:09.777674300 +0900
@@ -17,7 +17,7 @@
 # reserves and retains all trademark rights.
 #
 
-AM_CFLAGS=-g -Wall -I$(top_builddir)/src/
+AM_CFLAGS=-g -Wall -I$(top_builddir)/src/ -I$(top_srcdir)/src
 AM_LDFLAGS=-L$(top_builddir)/src/ksi -no-install -lksi
 check_PROGRAMS=runner parse-benchmark serialize-benchmark resigner integration-tests async-signer
 
