--- origsrc/yubihsm-shell-2.7.0/CMakeLists.txt	2025-06-11 18:31:09.000000000 +0900
+++ src/yubihsm-shell-2.7.0/CMakeLists.txt	2025-09-08 19:18:52.952588800 +0900
@@ -203,7 +203,7 @@ add_subdirectory (lib)
 if(NOT BUILD_ONLY_LIB)
   add_subdirectory (pkcs11)
 
-  if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+  if(${CMAKE_SYSTEM_NAME} MATCHES "Linux" OR CYGWIN)
     pkg_search_module (LIBPCSC REQUIRED libpcsclite)
   elseif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
     set (LIBPCSC_LDFLAGS "winscard.lib")
