diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*.whl' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x 'cmdline.*' origsrc/yubihsm-shell-2.0.2/CMakeLists.txt src/yubihsm-shell-2.0.2/CMakeLists.txt
--- origsrc/yubihsm-shell-2.0.2/CMakeLists.txt	2019-11-13 20:19:20.000000000 +0900
+++ src/yubihsm-shell-2.0.2/CMakeLists.txt	2020-01-21 08:41:44.360478500 +0900
@@ -64,7 +64,10 @@ if (NOT CMAKE_BUILD_TYPE)
 endif ()
 
 set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pedantic")
-set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Werror")
+set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra")
+if (NOT CYGWIN)
+  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
+endif ()
 set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-format-zero-length -Wno-implicit-fallthrough")
 set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wformat -Wformat-security")
 
@@ -151,7 +154,7 @@ else()
   message("not win32")
 
   set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fstack-protector-all")
-  if (NOT CMAKE_C_COMPILER_ID MATCHES Clang)
+  if (NOT CMAKE_C_COMPILER_ID MATCHES Clang AND NOT CYGWIN)
     set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wl,-z,relro,-z,now")
   endif ()
 
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*.whl' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x 'cmdline.*' origsrc/yubihsm-shell-2.0.2/pkcs11/CMakeLists.txt src/yubihsm-shell-2.0.2/pkcs11/CMakeLists.txt
--- origsrc/yubihsm-shell-2.0.2/pkcs11/CMakeLists.txt	2019-11-13 20:19:20.000000000 +0900
+++ src/yubihsm-shell-2.0.2/pkcs11/CMakeLists.txt	2020-01-21 08:41:44.469624500 +0900
@@ -79,9 +79,7 @@ target_link_libraries(
 
 install(
   TARGETS yubihsm_pkcs11
-  ARCHIVE DESTINATION "${YUBIHSM_INSTALL_LIB_DIR}/pkcs11"
-  LIBRARY DESTINATION "${YUBIHSM_INSTALL_LIB_DIR}/pkcs11"
-  RUNTIME DESTINATION "${YUBIHSM_INSTALL_BIN_DIR}/pkcs11")
+  RUNTIME DESTINATION "${YUBIHSM_INSTALL_LIB_DIR}/pkcs11")
 install(FILES pkcs11.h DESTINATION "${YUBIHSM_INSTALL_INC_DIR}/pkcs11")
 install(FILES pkcs11y.h DESTINATION "${YUBIHSM_INSTALL_INC_DIR}/pkcs11")
 
