--- origsrc/esniper-2-33-0/util.h	2017-03-13 22:21:00.000000000 +0900
+++ src/esniper-2-33-0/util.h	2017-07-03 23:59:46.746688500 +0900
@@ -65,7 +65,7 @@ extern void setPassword(char *password);
 extern char *getPassword(void);
 extern void freePassword(char *password);
 
-#if defined(__CYGWIN__) || defined(WIN32)
+#if defined(WIN32)
 extern char *basename(char *);
 extern char *dirname(char *);
 #else
