--- origsrc/esniper-2-35-0/util.h	2018-06-18 02:26:57.000000000 +0900
+++ src/esniper-2-35-0/util.h	2025-05-03 09:48:45.862245800 +0900
@@ -65,7 +65,7 @@ extern void setPassword(char *password);
 extern char *getPassword(void);
 extern void freePassword(char *password);
 
-#if defined(__CYGWIN__) || defined(WIN32)
+#if defined(WIN32)
 extern char *basename(char *);
 extern char *dirname(char *);
 #else
