--- origsrc/esniper-2-32-0/util.h	2016-10-31 06:16:41.000000000 +0900
+++ src/esniper-2-32-0/util.h	2016-12-08 08:46:46.658278400 +0900
@@ -65,7 +65,7 @@ extern void setPassword(char *password);
 extern char *getPassword(void);
 extern void freePassword(char *password);
 
-#if defined(__CYGWIN__) || defined(WIN32)
+#if defined(WIN32)
 extern char *basename(char *);
 extern char *dirname(char *);
 #else
