--- origsrc/libdigest-0.2/CMakeLists.txt	2008-01-08 02:44:41.000000000 +0900
+++ src/libdigest-0.2/CMakeLists.txt	2015-04-24 07:28:46.703005500 +0900
@@ -12,3 +12,9 @@ ADD_SUBDIRECTORY( test )
 #INSTALL( TARGETS digest /usr/lib )
 #INSTALL( TARGETS testdigest /usr/bin )
 #INSTALL( FILES ${PROJECT_SOURCE_DIR}/lib/digest.h /usr/include )
+
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "0.2")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libdigest.pc.in ${CMAKE_BINARY_DIR}/libdigest.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libdigest.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libdigest-0.2/lib/CMakeLists.txt	2008-01-08 02:44:41.000000000 +0900
+++ src/libdigest-0.2/lib/CMakeLists.txt	2015-04-24 07:25:02.686551400 +0900
@@ -10,5 +10,5 @@ SET_TARGET_PROPERTIES( digest PROPERTIES
 
 TARGET_LINK_LIBRARIES( digest ${APRUTIL_LINK_FLAGS})
 
-INSTALL( TARGETS digest LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/usr/lib )
-INSTALL( FILES ${PROJECT_SOURCE_DIR}/lib/digest.h DESTINATION ${CMAKE_INSTALL_PREFIX}/usr/include )
+INSTALL( TARGETS digest RUNTIME DESTINATION bin ARCHIVE DESTINATION lib )
+INSTALL( FILES ${PROJECT_SOURCE_DIR}/lib/digest.h DESTINATION include/libdigest )
--- origsrc/libdigest-0.2/libdigest.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libdigest-0.2/libdigest.pc.in	2015-04-24 07:26:16.571572000 +0900
@@ -0,0 +1,5 @@
+Name: libdigest
+Description: C library for computing the RFC2617-compliant digest
+Version: @VERSION@
+Libs: -L@libdir@ -ldigest
+Cflags: -I@includedir@
--- origsrc/libdigest-0.2/test/CMakeLists.txt	2008-01-08 02:44:41.000000000 +0900
+++ src/libdigest-0.2/test/CMakeLists.txt	2015-04-24 07:25:02.699007300 +0900
@@ -2,4 +2,4 @@ INCLUDE_DIRECTORIES( ${PROJECT_SOURCE_DI
 SET( testdigest_SRCS testdigest.c )
 ADD_EXECUTABLE( testdigest ${testdigest_SRCS} )
 TARGET_LINK_LIBRARIES( testdigest digest )
-INSTALL( TARGETS testdigest RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/usr/bin )
+INSTALL( TARGETS testdigest RUNTIME DESTINATION bin )
