--- origsrc/libgsignon-glib/libgsignon-glib/Makefile.am	2019-06-06 12:49:46.000000000 +0900
+++ src/libgsignon-glib/libgsignon-glib/Makefile.am	2019-06-06 13:34:13.742400400 +0900
@@ -9,7 +9,7 @@ libgsignon_glib_la_CPPFLAGS = \
 	$(DEPS_CFLAGS) \
 	-Wall -Werror
 libgsignon_glib_la_LIBADD = $(DEPS_LIBS)
-libgsignon_glib_la_LDFLAGS = \
+libgsignon_glib_la_LDFLAGS = -no-undefined \
 	-version-info 1:0:0 \
 	-export-symbols-regex '^signon_'
 
