--- origsrc/miniupnp-miniupnpc_2_2_5/miniupnpc/miniupnpc.pc.in	2023-06-12 08:20:36.000000000 +0900
+++ src/miniupnp-miniupnpc_2_2_5/miniupnpc/miniupnpc.pc.in	2023-08-18 18:19:21.837485500 +0900
@@ -5,14 +5,14 @@
 # Good tutoral for understanding .pc files: 
 # https://people.freedesktop.org/~dbn/pkg-config-guide.html
 
-prefix="@CMAKE_INSTALL_PREFIX@"
-exec_prefix="${prefix}"
-libdir="${prefix}/lib"
-includedir="${prefix}/include"
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${prefix}/lib
+includedir=${prefix}/include
 
 Name: @PROJECT_NAME@
 Description: @PROJECT_DESCRIPTION@
 URL: @PROJECT_HOMEPAGE_URL@
 Version: @PROJECT_VERSION@
-Libs: -L"${libdir}" -lminiupnpc
-Cflags: -I"${includedir}"
+Libs: -L${libdir} -lminiupnpc
+Cflags: -I${includedir}
