--- origsrc/libplacebo/meson.build	2023-08-14 12:15:34.000000000 +0900
+++ src/libplacebo/meson.build	2023-08-14 12:20:05.460394000 +0900
@@ -349,7 +349,7 @@ c_opts = [
   '-Wno-missing-field-initializers', '-Wno-type-limits',
 
   # Warnings to treat as errors
-  '-Werror=implicit-function-declaration',
+  # '-Werror=implicit-function-declaration',
 ]
 
 if cc.has_argument('-Wincompatible-pointer-types')
--- origsrc/libplacebo/src/d3d11/meson.build	2023-08-14 12:15:34.000000000 +0900
+++ src/libplacebo/src/d3d11/meson.build	2023-08-14 12:17:07.890855600 +0900
@@ -36,6 +36,6 @@ if d3d11.allowed()
     'd3d11/swapchain.c',
     'd3d11/utils.c',
   ]
-elif d3d11_header
+elif d3d11_header and host_machine.system() != 'cygwin'
   sources += 'd3d11/stubs.c'
 endif
