diff --strip-trailing-cr -uprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.log -x config.status -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x '*.pc' -x config.h -x testsuites origsrc/libccd-1.4/CMakeLists.txt src/libccd-1.4/CMakeLists.txt
--- origsrc/libccd-1.4/CMakeLists.txt	2013-03-23 12:18:00.126296700 +0900
+++ src/libccd-1.4/CMakeLists.txt	2013-03-23 12:25:31.387038900 +0900
@@ -41,14 +41,20 @@ set_target_properties(ccd PROPERTIES VER
 
 target_link_libraries(ccd ${MATH})
 
+if (NOT CYGWIN)
 add_library(ccd_static STATIC ${SOURCE_FILES})
 set_target_properties(ccd_static PROPERTIES OUTPUT_NAME ccd)
 target_link_libraries(ccd_static ${MATH})
+endif (NOT CYGWIN)
 
 set(pkg_conf_file "${CMAKE_CURRENT_SOURCE_DIR}/ccd.pc")
 configure_file("${pkg_conf_file}.in" "${pkg_conf_file}" @ONLY)
 
+if (CYGWIN)
+install(TARGETS ccd RUNTIME DESTINATION bin ARCHIVE DESTINATION lib${LIB_SUFFIX}/ LIBRARY DESTINATION lib${LIB_SUFFIX}/)
+else (CYGWIN)
 install(TARGETS ccd ccd_static ARCHIVE DESTINATION lib${LIB_SUFFIX}/ LIBRARY DESTINATION lib${LIB_SUFFIX}/)
+endif (CYGWIN)
 install(DIRECTORY src/ccd DESTINATION include FILES_MATCHING PATTERN "*.h")
 install(FILES "${pkg_conf_file}" DESTINATION lib${LIB_SUFFIX}/pkgconfig/ COMPONENT pkgconfig)
 
