--- origsrc/libaccounts-glib-VERSION_1.23-8d14b10652b2fe6c25d8ad8334e2d5023d254313/libaccounts-glib/Makefile.am	2016-09-28 16:51:51.000000000 +0900
+++ src/libaccounts-glib-VERSION_1.23-8d14b10652b2fe6c25d8ad8334e2d5023d254313/libaccounts-glib/Makefile.am	2017-07-08 00:30:37.226280400 +0900
@@ -13,6 +13,7 @@ libaccounts_glib_la_CFLAGS = \
 libaccounts_glib_la_LIBADD = \
 	$(LIBACCOUNTS_LIBS)
 libaccounts_glib_la_LDFLAGS = \
+	-no-undefined \
 	$(COVERAGE_LDFLAGS) \
 	-version-info 2:0:2 \
 	--no-allow-shlib-undefined \
