--- origsrc/grib_api-1.12.3/definitions/Makefile.am	2014-06-26 22:44:08.000000000 +0900
+++ src/grib_api-1.12.3/definitions/Makefile.am	2014-07-26 23:12:28.353441600 +0900
@@ -2626,4 +2626,4 @@ dist_definitionstide_DATA = \
 
 EXTRA_DIST=CMakeLists.txt
 
-include $(DEVEL_RULES)
+include $(srcdir)/$(DEVEL_RULES)
--- origsrc/grib_api-1.12.3/examples/C/Makefile.am	2014-06-26 22:44:21.000000000 +0900
+++ src/grib_api-1.12.3/examples/C/Makefile.am	2014-07-26 23:13:21.825121300 +0900
@@ -6,7 +6,7 @@ TESTS = iterator.sh get.sh print_data.sh
 
 noinst_PROGRAMS        =  nearest set_bitmap iterator get print_data set set_missing keys_iterator \
 							set_data mars_param values_check box nc multi multi_write precision set_pv list sections_copy
-bin_PROGRAMS = points
+bin_PROGRAMS = grib_points
 
 box_SOURCES               =   box.c
 set_data_SOURCES               =   set_data.c
@@ -24,14 +24,21 @@ multi_SOURCES             =   multi.c
 keys_iterator_SOURCES     =   keys_iterator.c
 iterator_SOURCES          =   iterator.c
 set_pv_SOURCES          =   set_pv.c
-points_SOURCES          =   points.c
+grib_points_SOURCES          =   points.c
 list_SOURCES          =   list.c
 sections_copy_SOURCES = sections_copy.c
 
-INCLUDES =  -I$(top_builddir)/src
+INCLUDES =  -I$(top_builddir)/src -I$(top_srcdir)/src
 
 LDADD = $(top_builddir)/src/libgrib_api.la
 DEPENDENCIES = $(LDADD)
 
 EXTRA_DIST =$(TESTS) include.sh CMakeLists.txt include.ctest.sh.in
 
+TESTS_ENVIRONMENT = \
+	GRIB_DEFINITION_PATH=${top_srcdir}/definitions \
+	GRIB_TEMPLATES_PATH=${top_srcdir}/templates \
+	tools_dir=${abs_top_builddir}/tools/ \
+	examples_dir=${abs_top_builddir}/examples/C/ \
+	tigge_dir=${top_srcdir}/tigge/ \
+	data_dir=${top_srcdir}/data
--- origsrc/grib_api-1.12.3/examples/F90/Makefile.am	2014-06-26 22:46:02.000000000 +0900
+++ src/grib_api-1.12.3/examples/F90/Makefile.am	2014-07-26 23:14:22.262621300 +0900
@@ -43,3 +43,10 @@ DEPENDENCIES = $(LDADD)
 
 EXTRA_DIST =$(TESTS) include.sh CMakeLists.txt include.ctest.sh.in
 
+TESTS_ENVIRONMENT = \
+	GRIB_DEFINITION_PATH=${top_srcdir}/definitions \
+	GRIB_SAMPLES_PATH=${top_srcdir}/samples \
+	tools_dir=${abs_top_builddir}/tools/ \
+	examples_dir=${abs_top_builddir}/examples/F90/ \
+	tigge_dir=${top_srcdir}/tigge/ \
+	data_dir=${top_srcdir}/data
--- origsrc/grib_api-1.12.3/fortran/Makefile.am	2014-06-26 22:51:33.000000000 +0900
+++ src/grib_api-1.12.3/fortran/Makefile.am	2014-07-26 23:15:38.988207300 +0900
@@ -7,11 +7,13 @@ include_HEADERS = grib_api_f77.h
 
 libgrib_api_f77_la_SOURCES= grib_fortran.c grib_f77.c 
 libgrib_api_f77_la_DEPENDENCIES = $(top_builddir)/src/libgrib_api.la
-libgrib_api_f77_la_LDFLAGS = -version-info $(GRIB_ABI_CURRENT):$(GRIB_ABI_REVISION):$(GRIB_ABI_AGE)
+libgrib_api_f77_la_LDFLAGS = -no-undefined -version-info $(GRIB_ABI_CURRENT):$(GRIB_ABI_REVISION):$(GRIB_ABI_AGE)
+libgrib_api_f77_la_LIBADD = $(top_builddir)/src/libgrib_api.la
 
 libgrib_api_f90_la_SOURCES= grib_fortran.c grib_f90.f90 
 libgrib_api_f90_la_DEPENDENCIES = $(top_builddir)/src/libgrib_api.la grib_api_externals.h grib_api_visibility.h grib_api_constants.h grib_kinds.h
-libgrib_api_f90_la_LDFLAGS = -version-info $(GRIB_ABI_CURRENT):$(GRIB_ABI_REVISION):$(GRIB_ABI_AGE)
+libgrib_api_f90_la_LDFLAGS = -no-undefined -version-info $(GRIB_ABI_CURRENT):$(GRIB_ABI_REVISION):$(GRIB_ABI_AGE)
+libgrib_api_f90_la_LIBADD = $(top_builddir)/src/libgrib_api.la
 
 libgrib_api_fortran_prototypes= grib_fortran.c
 
@@ -24,7 +26,8 @@ GRIB_API.mod: grib_f90.o
 endif
 
 # set the include path
-INCLUDES=  -I$(top_builddir)/src
+INCLUDES=  -I$(top_builddir)/src -I$(top_srcdir)/src -I$(top_srcdir)/fortran
+AM_FCFLAGS = $(INCLUDES)
 
 ## Make sure these will be cleaned even when they're not built by
 ## default.
@@ -40,7 +43,7 @@ EXTRA_DIST= grib_fortran_prototypes.h gr
             CMakeLists.txt
 
 grib_f90.f90: grib_f90_head.f90 grib_f90_tail.f90 grib_f90_int.f90 grib_f90_long_int.f90 grib_f90_int_size_t.f90 grib_f90_long_size_t.f90 same_int_long same_int_size_t grib_kinds.h
-	./create_grib_f90.sh 
+	srcdir=$(srcdir) $(srcdir)/create_grib_f90.sh
 
 grib_f90.o : grib_kinds.h
 
--- origsrc/grib_api-1.12.3/fortran/create_grib_f90.sh	2014-05-21 23:50:26.000000000 +0900
+++ src/grib_api-1.12.3/fortran/create_grib_f90.sh	2014-07-26 23:16:24.660082300 +0900
@@ -11,19 +11,19 @@ same=`./same_int_long`
 
 if [ $same -eq 1 ]
 then
- long=grib_f90_int.f90
+ long=${srcdir}/grib_f90_int.f90
 else
- long=grib_f90_long_int.f90
+ long=${srcdir}/grib_f90_long_int.f90
 fi
 
 same=`./same_int_size_t`
 
 if [ $same -eq 1 ]
 then
- sizet=grib_f90_int_size_t.f90
+ sizet=${srcdir}/grib_f90_int_size_t.f90
 else
- sizet=grib_f90_long_size_t.f90
+ sizet=${srcdir}/grib_f90_long_size_t.f90
 fi
 
-cat grib_f90_head.f90 $long $sizet grib_f90_tail.f90 > grib_f90.f90
+cat ${srcdir}/grib_f90_head.f90 $long $sizet ${srcdir}/grib_f90_tail.f90 > grib_f90.f90
 
--- origsrc/grib_api-1.12.3/ifs_samples/grib1/Makefile.am	2014-05-23 01:48:04.000000000 +0900
+++ src/grib_api-1.12.3/ifs_samples/grib1/Makefile.am	2014-07-26 23:17:00.976488500 +0900
@@ -77,203 +77,203 @@ nodist_grib1_DATA = \
 	sh_sfc_grib1.tmpl \
 	sh_sfc_grib2.tmpl 
 
-GRIB1.tmpl: ../../samples/GRIB1.tmpl
-	cp -f ../../samples/GRIB1.tmpl .
+GRIB1.tmpl: $(top_srcdir)/samples/GRIB1.tmpl
+	cp -f $(top_srcdir)/samples/GRIB1.tmpl .
 
-GRIB2.tmpl: ../../samples/GRIB2.tmpl
-	cp -f ../../samples/GRIB2.tmpl .
+GRIB2.tmpl: $(top_srcdir)/samples/GRIB2.tmpl
+	cp -f $(top_srcdir)/samples/GRIB2.tmpl .
 
-gg_sfc_grib1.tmpl: ../../samples/gg_sfc_grib1.tmpl
-	cp -f ../../samples/gg_sfc_grib1.tmpl .
+gg_sfc_grib1.tmpl: $(top_srcdir)/samples/gg_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/gg_sfc_grib1.tmpl .
 
-budg.tmpl: ../../samples/budg.tmpl
-	cp -f ../../samples/budg.tmpl .
+budg.tmpl: $(top_srcdir)/samples/budg.tmpl
+	cp -f $(top_srcdir)/samples/budg.tmpl .
 
-gg_sfc_grib2.tmpl: ../../samples/gg_sfc_grib2.tmpl
-	cp -f ../../samples/gg_sfc_grib2.tmpl .
+gg_sfc_grib2.tmpl: $(top_srcdir)/samples/gg_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/gg_sfc_grib2.tmpl .
 
-polar_stereographic_pl_grib1.tmpl: ../../samples/polar_stereographic_pl_grib1.tmpl
-	cp -f ../../samples/polar_stereographic_pl_grib1.tmpl .
+polar_stereographic_pl_grib1.tmpl: $(top_srcdir)/samples/polar_stereographic_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/polar_stereographic_pl_grib1.tmpl .
 
-polar_stereographic_pl_grib2.tmpl: ../../samples/polar_stereographic_pl_grib2.tmpl
-	cp -f ../../samples/polar_stereographic_pl_grib2.tmpl .
+polar_stereographic_pl_grib2.tmpl: $(top_srcdir)/samples/polar_stereographic_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/polar_stereographic_pl_grib2.tmpl .
 
-polar_stereographic_sfc_grib1.tmpl: ../../samples/polar_stereographic_sfc_grib1.tmpl
-	cp -f ../../samples/polar_stereographic_sfc_grib1.tmpl .
+polar_stereographic_sfc_grib1.tmpl: $(top_srcdir)/samples/polar_stereographic_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/polar_stereographic_sfc_grib1.tmpl .
 
-polar_stereographic_sfc_grib2.tmpl: ../../samples/polar_stereographic_sfc_grib2.tmpl
-	cp -f ../../samples/polar_stereographic_sfc_grib2.tmpl .
+polar_stereographic_sfc_grib2.tmpl: $(top_srcdir)/samples/polar_stereographic_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/polar_stereographic_sfc_grib2.tmpl .
 
-reduced_gg_ml_grib1.tmpl: ../../samples/reduced_gg_ml_grib1.tmpl
-	cp -f ../../samples/reduced_gg_ml_grib1.tmpl .
+reduced_gg_ml_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_ml_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_ml_grib1.tmpl .
 
-reduced_gg_ml_grib2.tmpl: ../../samples/reduced_gg_ml_grib2.tmpl
-	cp -f ../../samples/reduced_gg_ml_grib2.tmpl .
+reduced_gg_ml_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_ml_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_ml_grib2.tmpl .
 
-reduced_gg_pl_1024_grib1.tmpl: ../../samples/reduced_gg_pl_1024_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_1024_grib1.tmpl .
+reduced_gg_pl_1024_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_1024_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_1024_grib1.tmpl .
 
-reduced_gg_pl_1024_grib2.tmpl: ../../samples/reduced_gg_pl_1024_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_1024_grib2.tmpl .
+reduced_gg_pl_1024_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_1024_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_1024_grib2.tmpl .
 
-reduced_gg_pl_1280_grib1.tmpl: ../../samples/reduced_gg_pl_1280_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_1280_grib1.tmpl .
+reduced_gg_pl_1280_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_1280_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_1280_grib1.tmpl .
 
-reduced_gg_pl_1280_grib2.tmpl: ../../samples/reduced_gg_pl_1280_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_1280_grib2.tmpl .
+reduced_gg_pl_1280_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_1280_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_1280_grib2.tmpl .
 
-reduced_gg_pl_128_grib1.tmpl: ../../samples/reduced_gg_pl_128_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_128_grib1.tmpl .
+reduced_gg_pl_128_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_128_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_128_grib1.tmpl .
 
-reduced_gg_pl_128_grib2.tmpl: ../../samples/reduced_gg_pl_128_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_128_grib2.tmpl .
+reduced_gg_pl_128_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_128_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_128_grib2.tmpl .
 
-reduced_gg_pl_160_grib1.tmpl: ../../samples/reduced_gg_pl_160_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_160_grib1.tmpl .
+reduced_gg_pl_160_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_160_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_160_grib1.tmpl .
 
-reduced_gg_pl_160_grib2.tmpl: ../../samples/reduced_gg_pl_160_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_160_grib2.tmpl .
+reduced_gg_pl_160_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_160_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_160_grib2.tmpl .
 
-reduced_gg_pl_2000_grib1.tmpl: ../../samples/reduced_gg_pl_2000_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_2000_grib1.tmpl .
+reduced_gg_pl_2000_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_2000_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_2000_grib1.tmpl .
 
-reduced_gg_pl_2000_grib2.tmpl: ../../samples/reduced_gg_pl_2000_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_2000_grib2.tmpl .
+reduced_gg_pl_2000_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_2000_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_2000_grib2.tmpl .
 
-reduced_gg_pl_200_grib1.tmpl: ../../samples/reduced_gg_pl_200_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_200_grib1.tmpl .
+reduced_gg_pl_200_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_200_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_200_grib1.tmpl .
 
-reduced_gg_pl_200_grib2.tmpl: ../../samples/reduced_gg_pl_200_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_200_grib2.tmpl .
+reduced_gg_pl_200_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_200_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_200_grib2.tmpl .
 
-reduced_gg_pl_256_grib1.tmpl: ../../samples/reduced_gg_pl_256_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_256_grib1.tmpl .
+reduced_gg_pl_256_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_256_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_256_grib1.tmpl .
 
-reduced_gg_pl_256_grib2.tmpl: ../../samples/reduced_gg_pl_256_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_256_grib2.tmpl .
+reduced_gg_pl_256_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_256_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_256_grib2.tmpl .
 
-reduced_gg_pl_320_grib1.tmpl: ../../samples/reduced_gg_pl_320_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_320_grib1.tmpl .
+reduced_gg_pl_320_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_320_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_320_grib1.tmpl .
 
-reduced_gg_pl_320_grib2.tmpl: ../../samples/reduced_gg_pl_320_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_320_grib2.tmpl .
+reduced_gg_pl_320_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_320_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_320_grib2.tmpl .
 
-reduced_gg_pl_32_grib1.tmpl: ../../samples/reduced_gg_pl_32_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_32_grib1.tmpl .
+reduced_gg_pl_32_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_32_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_32_grib1.tmpl .
 
-reduced_gg_pl_32_grib2.tmpl: ../../samples/reduced_gg_pl_32_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_32_grib2.tmpl .
+reduced_gg_pl_32_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_32_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_32_grib2.tmpl .
 
-reduced_gg_pl_400_grib1.tmpl: ../../samples/reduced_gg_pl_400_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_400_grib1.tmpl .
+reduced_gg_pl_400_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_400_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_400_grib1.tmpl .
 
-reduced_gg_pl_400_grib2.tmpl: ../../samples/reduced_gg_pl_400_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_400_grib2.tmpl .
+reduced_gg_pl_400_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_400_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_400_grib2.tmpl .
 
-reduced_gg_pl_48_grib1.tmpl: ../../samples/reduced_gg_pl_48_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_48_grib1.tmpl .
+reduced_gg_pl_48_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_48_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_48_grib1.tmpl .
 
-reduced_gg_pl_48_grib2.tmpl: ../../samples/reduced_gg_pl_48_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_48_grib2.tmpl .
+reduced_gg_pl_48_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_48_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_48_grib2.tmpl .
 
-reduced_gg_pl_512_grib1.tmpl: ../../samples/reduced_gg_pl_512_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_512_grib1.tmpl .
+reduced_gg_pl_512_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_512_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_512_grib1.tmpl .
 
-reduced_gg_pl_512_grib2.tmpl: ../../samples/reduced_gg_pl_512_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_512_grib2.tmpl .
+reduced_gg_pl_512_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_512_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_512_grib2.tmpl .
 
-reduced_gg_pl_640_grib1.tmpl: ../../samples/reduced_gg_pl_640_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_640_grib1.tmpl .
+reduced_gg_pl_640_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_640_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_640_grib1.tmpl .
 
-reduced_gg_pl_640_grib2.tmpl: ../../samples/reduced_gg_pl_640_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_640_grib2.tmpl .
+reduced_gg_pl_640_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_640_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_640_grib2.tmpl .
 
-reduced_gg_pl_80_grib1.tmpl: ../../samples/reduced_gg_pl_80_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_80_grib1.tmpl .
+reduced_gg_pl_80_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_80_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_80_grib1.tmpl .
 
-reduced_gg_pl_80_grib2.tmpl: ../../samples/reduced_gg_pl_80_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_80_grib2.tmpl .
+reduced_gg_pl_80_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_80_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_80_grib2.tmpl .
 
-reduced_gg_pl_grib1.tmpl: ../../samples/reduced_gg_pl_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_grib1.tmpl .
+reduced_gg_pl_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_grib1.tmpl .
 
-reduced_gg_pl_grib2.tmpl: ../../samples/reduced_gg_pl_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_grib2.tmpl .
+reduced_gg_pl_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_grib2.tmpl .
 
-reduced_gg_sfc_grib1.tmpl: ../../samples/reduced_gg_sfc_grib1.tmpl
-	cp -f ../../samples/reduced_gg_sfc_grib1.tmpl .
+reduced_gg_sfc_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_sfc_grib1.tmpl .
 
-reduced_gg_sfc_grib2.tmpl: ../../samples/reduced_gg_sfc_grib2.tmpl
-	cp -f ../../samples/reduced_gg_sfc_grib2.tmpl .
+reduced_gg_sfc_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_sfc_grib2.tmpl .
 
-reduced_gg_sfc_jpeg_grib2.tmpl: ../../samples/reduced_gg_sfc_jpeg_grib2.tmpl
-	cp -f ../../samples/reduced_gg_sfc_jpeg_grib2.tmpl .
+reduced_gg_sfc_jpeg_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_sfc_jpeg_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_sfc_jpeg_grib2.tmpl .
 
-reduced_ll_sfc_grib1.tmpl: ../../samples/reduced_ll_sfc_grib1.tmpl
-	cp -f ../../samples/reduced_ll_sfc_grib1.tmpl .
+reduced_ll_sfc_grib1.tmpl: $(top_srcdir)/samples/reduced_ll_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_ll_sfc_grib1.tmpl .
 
-reduced_ll_sfc_grib2.tmpl: ../../samples/reduced_ll_sfc_grib2.tmpl
-	cp -f ../../samples/reduced_ll_sfc_grib2.tmpl .
+reduced_ll_sfc_grib2.tmpl: $(top_srcdir)/samples/reduced_ll_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_ll_sfc_grib2.tmpl .
 
-regular_gg_ml_grib1.tmpl: ../../samples/regular_gg_ml_grib1.tmpl
-	cp -f ../../samples/regular_gg_ml_grib1.tmpl .
+regular_gg_ml_grib1.tmpl: $(top_srcdir)/samples/regular_gg_ml_grib1.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_ml_grib1.tmpl .
 
-regular_gg_ml_grib2.tmpl: ../../samples/regular_gg_ml_grib2.tmpl
-	cp -f ../../samples/regular_gg_ml_grib2.tmpl .
+regular_gg_ml_grib2.tmpl: $(top_srcdir)/samples/regular_gg_ml_grib2.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_ml_grib2.tmpl .
 
-regular_gg_pl_grib1.tmpl: ../../samples/regular_gg_pl_grib1.tmpl
-	cp -f ../../samples/regular_gg_pl_grib1.tmpl .
+regular_gg_pl_grib1.tmpl: $(top_srcdir)/samples/regular_gg_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_pl_grib1.tmpl .
 
-regular_gg_pl_grib2.tmpl: ../../samples/regular_gg_pl_grib2.tmpl
-	cp -f ../../samples/regular_gg_pl_grib2.tmpl .
+regular_gg_pl_grib2.tmpl: $(top_srcdir)/samples/regular_gg_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_pl_grib2.tmpl .
 
-regular_gg_sfc_grib1.tmpl: ../../samples/regular_gg_sfc_grib1.tmpl
-	cp -f ../../samples/regular_gg_sfc_grib1.tmpl .
+regular_gg_sfc_grib1.tmpl: $(top_srcdir)/samples/regular_gg_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_sfc_grib1.tmpl .
 
-regular_gg_sfc_grib2.tmpl: ../../samples/regular_gg_sfc_grib2.tmpl
-	cp -f ../../samples/regular_gg_sfc_grib2.tmpl .
+regular_gg_sfc_grib2.tmpl: $(top_srcdir)/samples/regular_gg_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_sfc_grib2.tmpl .
 
-regular_ll_pl_grib1.tmpl: ../../samples/regular_ll_pl_grib1.tmpl
-	cp -f ../../samples/regular_ll_pl_grib1.tmpl .
+regular_ll_pl_grib1.tmpl: $(top_srcdir)/samples/regular_ll_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/regular_ll_pl_grib1.tmpl .
 
-regular_ll_pl_grib2.tmpl: ../../samples/regular_ll_pl_grib2.tmpl
-	cp -f ../../samples/regular_ll_pl_grib2.tmpl .
+regular_ll_pl_grib2.tmpl: $(top_srcdir)/samples/regular_ll_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/regular_ll_pl_grib2.tmpl .
 
-regular_ll_sfc_grib1.tmpl: ../../samples/regular_ll_sfc_grib1.tmpl
-	cp -f ../../samples/regular_ll_sfc_grib1.tmpl .
+regular_ll_sfc_grib1.tmpl: $(top_srcdir)/samples/regular_ll_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/regular_ll_sfc_grib1.tmpl .
 
-regular_ll_sfc_grib2.tmpl: ../../samples/regular_ll_sfc_grib2.tmpl
-	cp -f ../../samples/regular_ll_sfc_grib2.tmpl .
+regular_ll_sfc_grib2.tmpl: $(top_srcdir)/samples/regular_ll_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/regular_ll_sfc_grib2.tmpl .
 
-rotated_ll_pl_grib1.tmpl: ../../samples/rotated_ll_pl_grib1.tmpl
-	cp -f ../../samples/rotated_ll_pl_grib1.tmpl .
+rotated_ll_pl_grib1.tmpl: $(top_srcdir)/samples/rotated_ll_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/rotated_ll_pl_grib1.tmpl .
 
-rotated_ll_pl_grib2.tmpl: ../../samples/rotated_ll_pl_grib2.tmpl
-	cp -f ../../samples/rotated_ll_pl_grib2.tmpl .
+rotated_ll_pl_grib2.tmpl: $(top_srcdir)/samples/rotated_ll_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/rotated_ll_pl_grib2.tmpl .
 
-rotated_ll_sfc_grib1.tmpl: ../../samples/rotated_ll_sfc_grib1.tmpl
-	cp -f ../../samples/rotated_ll_sfc_grib1.tmpl .
+rotated_ll_sfc_grib1.tmpl: $(top_srcdir)/samples/rotated_ll_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/rotated_ll_sfc_grib1.tmpl .
 
-rotated_ll_sfc_grib2.tmpl: ../../samples/rotated_ll_sfc_grib2.tmpl
-	cp -f ../../samples/rotated_ll_sfc_grib2.tmpl .
+rotated_ll_sfc_grib2.tmpl: $(top_srcdir)/samples/rotated_ll_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/rotated_ll_sfc_grib2.tmpl .
 
-sh_ml_grib1.tmpl: ../../samples/sh_ml_grib1.tmpl
-	cp -f ../../samples/sh_ml_grib1.tmpl .
+sh_ml_grib1.tmpl: $(top_srcdir)/samples/sh_ml_grib1.tmpl
+	cp -f $(top_srcdir)/samples/sh_ml_grib1.tmpl .
 
-sh_ml_grib2.tmpl: ../../samples/sh_ml_grib2.tmpl
-	cp -f ../../samples/sh_ml_grib2.tmpl .
+sh_ml_grib2.tmpl: $(top_srcdir)/samples/sh_ml_grib2.tmpl
+	cp -f $(top_srcdir)/samples/sh_ml_grib2.tmpl .
 
-sh_pl_grib1.tmpl: ../../samples/sh_pl_grib1.tmpl
-	cp -f ../../samples/sh_pl_grib1.tmpl .
+sh_pl_grib1.tmpl: $(top_srcdir)/samples/sh_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/sh_pl_grib1.tmpl .
 
-sh_pl_grib2.tmpl: ../../samples/sh_pl_grib2.tmpl
-	cp -f ../../samples/sh_pl_grib2.tmpl .
+sh_pl_grib2.tmpl: $(top_srcdir)/samples/sh_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/sh_pl_grib2.tmpl .
 
-sh_sfc_grib1.tmpl: ../../samples/sh_sfc_grib1.tmpl
-	cp -f ../../samples/sh_sfc_grib1.tmpl .
+sh_sfc_grib1.tmpl: $(top_srcdir)/samples/sh_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/sh_sfc_grib1.tmpl .
 
-sh_sfc_grib2.tmpl: ../../samples/sh_sfc_grib2.tmpl
-	cp -f ../../samples/sh_sfc_grib2.tmpl .
+sh_sfc_grib2.tmpl: $(top_srcdir)/samples/sh_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/sh_sfc_grib2.tmpl .
 
 
 
--- origsrc/grib_api-1.12.3/ifs_samples/grib1_mlgrib2/Makefile.am	2014-05-23 01:47:57.000000000 +0900
+++ src/grib_api-1.12.3/ifs_samples/grib1_mlgrib2/Makefile.am	2014-07-26 23:17:23.767504100 +0900
@@ -75,203 +75,203 @@ nodist_grib1_mlgrib2_DATA = \
 	sh_sfc_grib1.tmpl \
 	sh_sfc_grib2.tmpl 
 
-GRIB1.tmpl: ../../samples/GRIB1.tmpl
-	cp -f ../../samples/GRIB1.tmpl .
+GRIB1.tmpl: $(top_srcdir)/samples/GRIB1.tmpl
+	cp -f $(top_srcdir)/samples/GRIB1.tmpl .
 
-GRIB2.tmpl: ../../samples/GRIB2.tmpl
-	cp -f ../../samples/GRIB2.tmpl .
+GRIB2.tmpl: $(top_srcdir)/samples/GRIB2.tmpl
+	cp -f $(top_srcdir)/samples/GRIB2.tmpl .
 
-gg_sfc_grib1.tmpl: ../../samples/gg_sfc_grib1.tmpl
-	cp -f ../../samples/gg_sfc_grib1.tmpl .
+gg_sfc_grib1.tmpl: $(top_srcdir)/samples/gg_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/gg_sfc_grib1.tmpl .
 
-budg.tmpl: ../../samples/budg.tmpl
-	cp -f ../../samples/budg.tmpl .
+budg.tmpl: $(top_srcdir)/samples/budg.tmpl
+	cp -f $(top_srcdir)/samples/budg.tmpl .
 
-gg_sfc_grib2.tmpl: ../../samples/gg_sfc_grib2.tmpl
-	cp -f ../../samples/gg_sfc_grib2.tmpl .
+gg_sfc_grib2.tmpl: $(top_srcdir)/samples/gg_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/gg_sfc_grib2.tmpl .
 
-polar_stereographic_pl_grib1.tmpl: ../../samples/polar_stereographic_pl_grib1.tmpl
-	cp -f ../../samples/polar_stereographic_pl_grib1.tmpl .
+polar_stereographic_pl_grib1.tmpl: $(top_srcdir)/samples/polar_stereographic_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/polar_stereographic_pl_grib1.tmpl .
 
-polar_stereographic_pl_grib2.tmpl: ../../samples/polar_stereographic_pl_grib2.tmpl
-	cp -f ../../samples/polar_stereographic_pl_grib2.tmpl .
+polar_stereographic_pl_grib2.tmpl: $(top_srcdir)/samples/polar_stereographic_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/polar_stereographic_pl_grib2.tmpl .
 
-polar_stereographic_sfc_grib1.tmpl: ../../samples/polar_stereographic_sfc_grib1.tmpl
-	cp -f ../../samples/polar_stereographic_sfc_grib1.tmpl .
+polar_stereographic_sfc_grib1.tmpl: $(top_srcdir)/samples/polar_stereographic_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/polar_stereographic_sfc_grib1.tmpl .
 
-polar_stereographic_sfc_grib2.tmpl: ../../samples/polar_stereographic_sfc_grib2.tmpl
-	cp -f ../../samples/polar_stereographic_sfc_grib2.tmpl .
+polar_stereographic_sfc_grib2.tmpl: $(top_srcdir)/samples/polar_stereographic_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/polar_stereographic_sfc_grib2.tmpl .
 
-reduced_gg_ml_grib1.tmpl: ../../samples/reduced_gg_ml_grib1.tmpl
-	cp -f ../../samples/reduced_gg_ml_grib1.tmpl .
+reduced_gg_ml_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_ml_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_ml_grib1.tmpl .
 
-reduced_gg_ml_grib2.tmpl: ../../samples/reduced_gg_ml_grib2.tmpl
-	cp -f ../../samples/reduced_gg_ml_grib2.tmpl .
+reduced_gg_ml_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_ml_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_ml_grib2.tmpl .
 
-reduced_gg_pl_1024_grib1.tmpl: ../../samples/reduced_gg_pl_1024_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_1024_grib1.tmpl .
+reduced_gg_pl_1024_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_1024_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_1024_grib1.tmpl .
 
-reduced_gg_pl_1024_grib2.tmpl: ../../samples/reduced_gg_pl_1024_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_1024_grib2.tmpl .
+reduced_gg_pl_1024_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_1024_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_1024_grib2.tmpl .
 
-reduced_gg_pl_1280_grib1.tmpl: ../../samples/reduced_gg_pl_1280_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_1280_grib1.tmpl .
+reduced_gg_pl_1280_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_1280_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_1280_grib1.tmpl .
 
-reduced_gg_pl_1280_grib2.tmpl: ../../samples/reduced_gg_pl_1280_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_1280_grib2.tmpl .
+reduced_gg_pl_1280_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_1280_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_1280_grib2.tmpl .
 
-reduced_gg_pl_128_grib1.tmpl: ../../samples/reduced_gg_pl_128_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_128_grib1.tmpl .
+reduced_gg_pl_128_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_128_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_128_grib1.tmpl .
 
-reduced_gg_pl_128_grib2.tmpl: ../../samples/reduced_gg_pl_128_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_128_grib2.tmpl .
+reduced_gg_pl_128_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_128_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_128_grib2.tmpl .
 
-reduced_gg_pl_160_grib1.tmpl: ../../samples/reduced_gg_pl_160_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_160_grib1.tmpl .
+reduced_gg_pl_160_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_160_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_160_grib1.tmpl .
 
-reduced_gg_pl_160_grib2.tmpl: ../../samples/reduced_gg_pl_160_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_160_grib2.tmpl .
+reduced_gg_pl_160_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_160_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_160_grib2.tmpl .
 
-reduced_gg_pl_2000_grib1.tmpl: ../../samples/reduced_gg_pl_2000_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_2000_grib1.tmpl .
+reduced_gg_pl_2000_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_2000_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_2000_grib1.tmpl .
 
-reduced_gg_pl_2000_grib2.tmpl: ../../samples/reduced_gg_pl_2000_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_2000_grib2.tmpl .
+reduced_gg_pl_2000_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_2000_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_2000_grib2.tmpl .
 
-reduced_gg_pl_200_grib1.tmpl: ../../samples/reduced_gg_pl_200_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_200_grib1.tmpl .
+reduced_gg_pl_200_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_200_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_200_grib1.tmpl .
 
-reduced_gg_pl_200_grib2.tmpl: ../../samples/reduced_gg_pl_200_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_200_grib2.tmpl .
+reduced_gg_pl_200_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_200_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_200_grib2.tmpl .
 
-reduced_gg_pl_256_grib1.tmpl: ../../samples/reduced_gg_pl_256_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_256_grib1.tmpl .
+reduced_gg_pl_256_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_256_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_256_grib1.tmpl .
 
-reduced_gg_pl_256_grib2.tmpl: ../../samples/reduced_gg_pl_256_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_256_grib2.tmpl .
+reduced_gg_pl_256_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_256_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_256_grib2.tmpl .
 
-reduced_gg_pl_320_grib1.tmpl: ../../samples/reduced_gg_pl_320_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_320_grib1.tmpl .
+reduced_gg_pl_320_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_320_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_320_grib1.tmpl .
 
-reduced_gg_pl_320_grib2.tmpl: ../../samples/reduced_gg_pl_320_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_320_grib2.tmpl .
+reduced_gg_pl_320_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_320_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_320_grib2.tmpl .
 
-reduced_gg_pl_32_grib1.tmpl: ../../samples/reduced_gg_pl_32_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_32_grib1.tmpl .
+reduced_gg_pl_32_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_32_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_32_grib1.tmpl .
 
-reduced_gg_pl_32_grib2.tmpl: ../../samples/reduced_gg_pl_32_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_32_grib2.tmpl .
+reduced_gg_pl_32_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_32_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_32_grib2.tmpl .
 
-reduced_gg_pl_400_grib1.tmpl: ../../samples/reduced_gg_pl_400_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_400_grib1.tmpl .
+reduced_gg_pl_400_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_400_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_400_grib1.tmpl .
 
-reduced_gg_pl_400_grib2.tmpl: ../../samples/reduced_gg_pl_400_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_400_grib2.tmpl .
+reduced_gg_pl_400_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_400_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_400_grib2.tmpl .
 
-reduced_gg_pl_48_grib1.tmpl: ../../samples/reduced_gg_pl_48_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_48_grib1.tmpl .
+reduced_gg_pl_48_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_48_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_48_grib1.tmpl .
 
-reduced_gg_pl_48_grib2.tmpl: ../../samples/reduced_gg_pl_48_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_48_grib2.tmpl .
+reduced_gg_pl_48_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_48_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_48_grib2.tmpl .
 
-reduced_gg_pl_512_grib1.tmpl: ../../samples/reduced_gg_pl_512_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_512_grib1.tmpl .
+reduced_gg_pl_512_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_512_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_512_grib1.tmpl .
 
-reduced_gg_pl_512_grib2.tmpl: ../../samples/reduced_gg_pl_512_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_512_grib2.tmpl .
+reduced_gg_pl_512_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_512_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_512_grib2.tmpl .
 
-reduced_gg_pl_640_grib1.tmpl: ../../samples/reduced_gg_pl_640_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_640_grib1.tmpl .
+reduced_gg_pl_640_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_640_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_640_grib1.tmpl .
 
-reduced_gg_pl_640_grib2.tmpl: ../../samples/reduced_gg_pl_640_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_640_grib2.tmpl .
+reduced_gg_pl_640_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_640_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_640_grib2.tmpl .
 
-reduced_gg_pl_80_grib1.tmpl: ../../samples/reduced_gg_pl_80_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_80_grib1.tmpl .
+reduced_gg_pl_80_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_80_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_80_grib1.tmpl .
 
-reduced_gg_pl_80_grib2.tmpl: ../../samples/reduced_gg_pl_80_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_80_grib2.tmpl .
+reduced_gg_pl_80_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_80_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_80_grib2.tmpl .
 
-reduced_gg_pl_grib1.tmpl: ../../samples/reduced_gg_pl_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_grib1.tmpl .
+reduced_gg_pl_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_grib1.tmpl .
 
-reduced_gg_pl_grib2.tmpl: ../../samples/reduced_gg_pl_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_grib2.tmpl .
+reduced_gg_pl_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_grib2.tmpl .
 
-reduced_gg_sfc_grib1.tmpl: ../../samples/reduced_gg_sfc_grib1.tmpl
-	cp -f ../../samples/reduced_gg_sfc_grib1.tmpl .
+reduced_gg_sfc_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_sfc_grib1.tmpl .
 
-reduced_gg_sfc_grib2.tmpl: ../../samples/reduced_gg_sfc_grib2.tmpl
-	cp -f ../../samples/reduced_gg_sfc_grib2.tmpl .
+reduced_gg_sfc_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_sfc_grib2.tmpl .
 
-reduced_gg_sfc_jpeg_grib2.tmpl: ../../samples/reduced_gg_sfc_jpeg_grib2.tmpl
-	cp -f ../../samples/reduced_gg_sfc_jpeg_grib2.tmpl .
+reduced_gg_sfc_jpeg_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_sfc_jpeg_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_sfc_jpeg_grib2.tmpl .
 
-reduced_ll_sfc_grib1.tmpl: ../../samples/reduced_ll_sfc_grib1.tmpl
-	cp -f ../../samples/reduced_ll_sfc_grib1.tmpl .
+reduced_ll_sfc_grib1.tmpl: $(top_srcdir)/samples/reduced_ll_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_ll_sfc_grib1.tmpl .
 
-reduced_ll_sfc_grib2.tmpl: ../../samples/reduced_ll_sfc_grib2.tmpl
-	cp -f ../../samples/reduced_ll_sfc_grib2.tmpl .
+reduced_ll_sfc_grib2.tmpl: $(top_srcdir)/samples/reduced_ll_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_ll_sfc_grib2.tmpl .
 
-regular_gg_ml_grib1.tmpl: ../../samples/regular_gg_ml_grib1.tmpl
-	cp -f ../../samples/regular_gg_ml_grib1.tmpl .
+regular_gg_ml_grib1.tmpl: $(top_srcdir)/samples/regular_gg_ml_grib1.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_ml_grib1.tmpl .
 
-regular_gg_ml_grib2.tmpl: ../../samples/regular_gg_ml_grib2.tmpl
-	cp -f ../../samples/regular_gg_ml_grib2.tmpl .
+regular_gg_ml_grib2.tmpl: $(top_srcdir)/samples/regular_gg_ml_grib2.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_ml_grib2.tmpl .
 
-regular_gg_pl_grib1.tmpl: ../../samples/regular_gg_pl_grib1.tmpl
-	cp -f ../../samples/regular_gg_pl_grib1.tmpl .
+regular_gg_pl_grib1.tmpl: $(top_srcdir)/samples/regular_gg_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_pl_grib1.tmpl .
 
-regular_gg_pl_grib2.tmpl: ../../samples/regular_gg_pl_grib2.tmpl
-	cp -f ../../samples/regular_gg_pl_grib2.tmpl .
+regular_gg_pl_grib2.tmpl: $(top_srcdir)/samples/regular_gg_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_pl_grib2.tmpl .
 
-regular_gg_sfc_grib1.tmpl: ../../samples/regular_gg_sfc_grib1.tmpl
-	cp -f ../../samples/regular_gg_sfc_grib1.tmpl .
+regular_gg_sfc_grib1.tmpl: $(top_srcdir)/samples/regular_gg_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_sfc_grib1.tmpl .
 
-regular_gg_sfc_grib2.tmpl: ../../samples/regular_gg_sfc_grib2.tmpl
-	cp -f ../../samples/regular_gg_sfc_grib2.tmpl .
+regular_gg_sfc_grib2.tmpl: $(top_srcdir)/samples/regular_gg_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_sfc_grib2.tmpl .
 
-regular_ll_pl_grib1.tmpl: ../../samples/regular_ll_pl_grib1.tmpl
-	cp -f ../../samples/regular_ll_pl_grib1.tmpl .
+regular_ll_pl_grib1.tmpl: $(top_srcdir)/samples/regular_ll_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/regular_ll_pl_grib1.tmpl .
 
-regular_ll_pl_grib2.tmpl: ../../samples/regular_ll_pl_grib2.tmpl
-	cp -f ../../samples/regular_ll_pl_grib2.tmpl .
+regular_ll_pl_grib2.tmpl: $(top_srcdir)/samples/regular_ll_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/regular_ll_pl_grib2.tmpl .
 
-regular_ll_sfc_grib1.tmpl: ../../samples/regular_ll_sfc_grib1.tmpl
-	cp -f ../../samples/regular_ll_sfc_grib1.tmpl .
+regular_ll_sfc_grib1.tmpl: $(top_srcdir)/samples/regular_ll_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/regular_ll_sfc_grib1.tmpl .
 
-regular_ll_sfc_grib2.tmpl: ../../samples/regular_ll_sfc_grib2.tmpl
-	cp -f ../../samples/regular_ll_sfc_grib2.tmpl .
+regular_ll_sfc_grib2.tmpl: $(top_srcdir)/samples/regular_ll_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/regular_ll_sfc_grib2.tmpl .
 
-rotated_ll_pl_grib1.tmpl: ../../samples/rotated_ll_pl_grib1.tmpl
-	cp -f ../../samples/rotated_ll_pl_grib1.tmpl .
+rotated_ll_pl_grib1.tmpl: $(top_srcdir)/samples/rotated_ll_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/rotated_ll_pl_grib1.tmpl .
 
-rotated_ll_pl_grib2.tmpl: ../../samples/rotated_ll_pl_grib2.tmpl
-	cp -f ../../samples/rotated_ll_pl_grib2.tmpl .
+rotated_ll_pl_grib2.tmpl: $(top_srcdir)/samples/rotated_ll_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/rotated_ll_pl_grib2.tmpl .
 
-rotated_ll_sfc_grib1.tmpl: ../../samples/rotated_ll_sfc_grib1.tmpl
-	cp -f ../../samples/rotated_ll_sfc_grib1.tmpl .
+rotated_ll_sfc_grib1.tmpl: $(top_srcdir)/samples/rotated_ll_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/rotated_ll_sfc_grib1.tmpl .
 
-rotated_ll_sfc_grib2.tmpl: ../../samples/rotated_ll_sfc_grib2.tmpl
-	cp -f ../../samples/rotated_ll_sfc_grib2.tmpl .
+rotated_ll_sfc_grib2.tmpl: $(top_srcdir)/samples/rotated_ll_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/rotated_ll_sfc_grib2.tmpl .
 
-sh_ml_grib1.tmpl: ../../samples/sh_ml_grib1.tmpl
-	cp -f ../../samples/sh_ml_grib1.tmpl .
+sh_ml_grib1.tmpl: $(top_srcdir)/samples/sh_ml_grib1.tmpl
+	cp -f $(top_srcdir)/samples/sh_ml_grib1.tmpl .
 
-sh_ml_grib2.tmpl: ../../samples/sh_ml_grib2.tmpl
-	cp -f ../../samples/sh_ml_grib2.tmpl .
+sh_ml_grib2.tmpl: $(top_srcdir)/samples/sh_ml_grib2.tmpl
+	cp -f $(top_srcdir)/samples/sh_ml_grib2.tmpl .
 
-sh_pl_grib1.tmpl: ../../samples/sh_pl_grib1.tmpl
-	cp -f ../../samples/sh_pl_grib1.tmpl .
+sh_pl_grib1.tmpl: $(top_srcdir)/samples/sh_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/sh_pl_grib1.tmpl .
 
-sh_pl_grib2.tmpl: ../../samples/sh_pl_grib2.tmpl
-	cp -f ../../samples/sh_pl_grib2.tmpl .
+sh_pl_grib2.tmpl: $(top_srcdir)/samples/sh_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/sh_pl_grib2.tmpl .
 
-sh_sfc_grib1.tmpl: ../../samples/sh_sfc_grib1.tmpl
-	cp -f ../../samples/sh_sfc_grib1.tmpl .
+sh_sfc_grib1.tmpl: $(top_srcdir)/samples/sh_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/sh_sfc_grib1.tmpl .
 
-sh_sfc_grib2.tmpl: ../../samples/sh_sfc_grib2.tmpl
-	cp -f ../../samples/sh_sfc_grib2.tmpl .
+sh_sfc_grib2.tmpl: $(top_srcdir)/samples/sh_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/sh_sfc_grib2.tmpl .
 
 
 
--- origsrc/grib_api-1.12.3/ifs_samples/grib1_mlgrib2_ieee64/Makefile.am	2014-05-23 01:47:45.000000000 +0900
+++ src/grib_api-1.12.3/ifs_samples/grib1_mlgrib2_ieee64/Makefile.am	2014-07-26 23:17:45.932543200 +0900
@@ -77,203 +77,203 @@ nodist_grib1_mlgrib2_ieee64_DATA = \
 	sh_sfc_grib1.tmpl \
 	sh_sfc_grib2.tmpl 
 
-GRIB1.tmpl: ../../samples/GRIB1.tmpl
-	cp -f ../../samples/GRIB1.tmpl .
+GRIB1.tmpl: $(top_srcdir)/samples/GRIB1.tmpl
+	cp -f $(top_srcdir)/samples/GRIB1.tmpl .
 
-GRIB2.tmpl: ../../samples/GRIB2.tmpl
-	cp -f ../../samples/GRIB2.tmpl .
+GRIB2.tmpl: $(top_srcdir)/samples/GRIB2.tmpl
+	cp -f $(top_srcdir)/samples/GRIB2.tmpl .
 
-gg_sfc_grib1.tmpl: ../../samples/gg_sfc_grib1.tmpl
-	cp -f ../../samples/gg_sfc_grib1.tmpl .
+gg_sfc_grib1.tmpl: $(top_srcdir)/samples/gg_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/gg_sfc_grib1.tmpl .
 
-budg.tmpl: ../../samples/budg.tmpl
-	cp -f ../../samples/budg.tmpl .
+budg.tmpl: $(top_srcdir)/samples/budg.tmpl
+	cp -f $(top_srcdir)/samples/budg.tmpl .
 
-gg_sfc_grib2.tmpl: ../../samples/gg_sfc_grib2.tmpl
-	cp -f ../../samples/gg_sfc_grib2.tmpl .
+gg_sfc_grib2.tmpl: $(top_srcdir)/samples/gg_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/gg_sfc_grib2.tmpl .
 
-polar_stereographic_pl_grib1.tmpl: ../../samples/polar_stereographic_pl_grib1.tmpl
-	cp -f ../../samples/polar_stereographic_pl_grib1.tmpl .
+polar_stereographic_pl_grib1.tmpl: $(top_srcdir)/samples/polar_stereographic_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/polar_stereographic_pl_grib1.tmpl .
 
-polar_stereographic_pl_grib2.tmpl: ../../samples/polar_stereographic_pl_grib2.tmpl
-	cp -f ../../samples/polar_stereographic_pl_grib2.tmpl .
+polar_stereographic_pl_grib2.tmpl: $(top_srcdir)/samples/polar_stereographic_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/polar_stereographic_pl_grib2.tmpl .
 
-polar_stereographic_sfc_grib1.tmpl: ../../samples/polar_stereographic_sfc_grib1.tmpl
-	cp -f ../../samples/polar_stereographic_sfc_grib1.tmpl .
+polar_stereographic_sfc_grib1.tmpl: $(top_srcdir)/samples/polar_stereographic_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/polar_stereographic_sfc_grib1.tmpl .
 
-polar_stereographic_sfc_grib2.tmpl: ../../samples/polar_stereographic_sfc_grib2.tmpl
-	cp -f ../../samples/polar_stereographic_sfc_grib2.tmpl .
+polar_stereographic_sfc_grib2.tmpl: $(top_srcdir)/samples/polar_stereographic_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/polar_stereographic_sfc_grib2.tmpl .
 
-reduced_gg_ml_grib1.tmpl: ../../samples/reduced_gg_ml_grib1.tmpl
-	cp -f ../../samples/reduced_gg_ml_grib1.tmpl .
+reduced_gg_ml_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_ml_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_ml_grib1.tmpl .
 
-reduced_gg_ml_grib2.tmpl: ../../samples/reduced_gg_ml_grib2.tmpl
-	cp -f ../../samples/reduced_gg_ml_grib2.tmpl .
+reduced_gg_ml_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_ml_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_ml_grib2.tmpl .
 
-reduced_gg_pl_1024_grib1.tmpl: ../../samples/reduced_gg_pl_1024_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_1024_grib1.tmpl .
+reduced_gg_pl_1024_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_1024_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_1024_grib1.tmpl .
 
-reduced_gg_pl_1024_grib2.tmpl: ../../samples/reduced_gg_pl_1024_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_1024_grib2.tmpl .
+reduced_gg_pl_1024_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_1024_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_1024_grib2.tmpl .
 
-reduced_gg_pl_1280_grib1.tmpl: ../../samples/reduced_gg_pl_1280_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_1280_grib1.tmpl .
+reduced_gg_pl_1280_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_1280_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_1280_grib1.tmpl .
 
-reduced_gg_pl_1280_grib2.tmpl: ../../samples/reduced_gg_pl_1280_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_1280_grib2.tmpl .
+reduced_gg_pl_1280_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_1280_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_1280_grib2.tmpl .
 
-reduced_gg_pl_128_grib1.tmpl: ../../samples/reduced_gg_pl_128_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_128_grib1.tmpl .
+reduced_gg_pl_128_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_128_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_128_grib1.tmpl .
 
-reduced_gg_pl_128_grib2.tmpl: ../../samples/reduced_gg_pl_128_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_128_grib2.tmpl .
+reduced_gg_pl_128_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_128_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_128_grib2.tmpl .
 
-reduced_gg_pl_160_grib1.tmpl: ../../samples/reduced_gg_pl_160_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_160_grib1.tmpl .
+reduced_gg_pl_160_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_160_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_160_grib1.tmpl .
 
-reduced_gg_pl_160_grib2.tmpl: ../../samples/reduced_gg_pl_160_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_160_grib2.tmpl .
+reduced_gg_pl_160_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_160_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_160_grib2.tmpl .
 
-reduced_gg_pl_2000_grib1.tmpl: ../../samples/reduced_gg_pl_2000_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_2000_grib1.tmpl .
+reduced_gg_pl_2000_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_2000_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_2000_grib1.tmpl .
 
-reduced_gg_pl_2000_grib2.tmpl: ../../samples/reduced_gg_pl_2000_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_2000_grib2.tmpl .
+reduced_gg_pl_2000_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_2000_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_2000_grib2.tmpl .
 
-reduced_gg_pl_200_grib1.tmpl: ../../samples/reduced_gg_pl_200_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_200_grib1.tmpl .
+reduced_gg_pl_200_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_200_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_200_grib1.tmpl .
 
-reduced_gg_pl_200_grib2.tmpl: ../../samples/reduced_gg_pl_200_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_200_grib2.tmpl .
+reduced_gg_pl_200_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_200_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_200_grib2.tmpl .
 
-reduced_gg_pl_256_grib1.tmpl: ../../samples/reduced_gg_pl_256_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_256_grib1.tmpl .
+reduced_gg_pl_256_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_256_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_256_grib1.tmpl .
 
-reduced_gg_pl_256_grib2.tmpl: ../../samples/reduced_gg_pl_256_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_256_grib2.tmpl .
+reduced_gg_pl_256_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_256_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_256_grib2.tmpl .
 
-reduced_gg_pl_320_grib1.tmpl: ../../samples/reduced_gg_pl_320_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_320_grib1.tmpl .
+reduced_gg_pl_320_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_320_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_320_grib1.tmpl .
 
-reduced_gg_pl_320_grib2.tmpl: ../../samples/reduced_gg_pl_320_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_320_grib2.tmpl .
+reduced_gg_pl_320_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_320_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_320_grib2.tmpl .
 
-reduced_gg_pl_32_grib1.tmpl: ../../samples/reduced_gg_pl_32_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_32_grib1.tmpl .
+reduced_gg_pl_32_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_32_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_32_grib1.tmpl .
 
-reduced_gg_pl_32_grib2.tmpl: ../../samples/reduced_gg_pl_32_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_32_grib2.tmpl .
+reduced_gg_pl_32_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_32_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_32_grib2.tmpl .
 
-reduced_gg_pl_400_grib1.tmpl: ../../samples/reduced_gg_pl_400_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_400_grib1.tmpl .
+reduced_gg_pl_400_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_400_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_400_grib1.tmpl .
 
-reduced_gg_pl_400_grib2.tmpl: ../../samples/reduced_gg_pl_400_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_400_grib2.tmpl .
+reduced_gg_pl_400_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_400_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_400_grib2.tmpl .
 
-reduced_gg_pl_48_grib1.tmpl: ../../samples/reduced_gg_pl_48_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_48_grib1.tmpl .
+reduced_gg_pl_48_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_48_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_48_grib1.tmpl .
 
-reduced_gg_pl_48_grib2.tmpl: ../../samples/reduced_gg_pl_48_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_48_grib2.tmpl .
+reduced_gg_pl_48_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_48_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_48_grib2.tmpl .
 
-reduced_gg_pl_512_grib1.tmpl: ../../samples/reduced_gg_pl_512_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_512_grib1.tmpl .
+reduced_gg_pl_512_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_512_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_512_grib1.tmpl .
 
-reduced_gg_pl_512_grib2.tmpl: ../../samples/reduced_gg_pl_512_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_512_grib2.tmpl .
+reduced_gg_pl_512_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_512_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_512_grib2.tmpl .
 
-reduced_gg_pl_640_grib1.tmpl: ../../samples/reduced_gg_pl_640_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_640_grib1.tmpl .
+reduced_gg_pl_640_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_640_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_640_grib1.tmpl .
 
-reduced_gg_pl_640_grib2.tmpl: ../../samples/reduced_gg_pl_640_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_640_grib2.tmpl .
+reduced_gg_pl_640_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_640_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_640_grib2.tmpl .
 
-reduced_gg_pl_80_grib1.tmpl: ../../samples/reduced_gg_pl_80_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_80_grib1.tmpl .
+reduced_gg_pl_80_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_80_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_80_grib1.tmpl .
 
-reduced_gg_pl_80_grib2.tmpl: ../../samples/reduced_gg_pl_80_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_80_grib2.tmpl .
+reduced_gg_pl_80_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_80_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_80_grib2.tmpl .
 
-reduced_gg_pl_grib1.tmpl: ../../samples/reduced_gg_pl_grib1.tmpl
-	cp -f ../../samples/reduced_gg_pl_grib1.tmpl .
+reduced_gg_pl_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_grib1.tmpl .
 
-reduced_gg_pl_grib2.tmpl: ../../samples/reduced_gg_pl_grib2.tmpl
-	cp -f ../../samples/reduced_gg_pl_grib2.tmpl .
+reduced_gg_pl_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_pl_grib2.tmpl .
 
-reduced_gg_sfc_grib1.tmpl: ../../samples/reduced_gg_sfc_grib1.tmpl
-	cp -f ../../samples/reduced_gg_sfc_grib1.tmpl .
+reduced_gg_sfc_grib1.tmpl: $(top_srcdir)/samples/reduced_gg_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_sfc_grib1.tmpl .
 
-reduced_gg_sfc_grib2.tmpl: ../../samples/reduced_gg_sfc_grib2.tmpl
-	cp -f ../../samples/reduced_gg_sfc_grib2.tmpl .
+reduced_gg_sfc_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_sfc_grib2.tmpl .
 
-reduced_gg_sfc_jpeg_grib2.tmpl: ../../samples/reduced_gg_sfc_jpeg_grib2.tmpl
-	cp -f ../../samples/reduced_gg_sfc_jpeg_grib2.tmpl .
+reduced_gg_sfc_jpeg_grib2.tmpl: $(top_srcdir)/samples/reduced_gg_sfc_jpeg_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_gg_sfc_jpeg_grib2.tmpl .
 
-reduced_ll_sfc_grib1.tmpl: ../../samples/reduced_ll_sfc_grib1.tmpl
-	cp -f ../../samples/reduced_ll_sfc_grib1.tmpl .
+reduced_ll_sfc_grib1.tmpl: $(top_srcdir)/samples/reduced_ll_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/reduced_ll_sfc_grib1.tmpl .
 
-reduced_ll_sfc_grib2.tmpl: ../../samples/reduced_ll_sfc_grib2.tmpl
-	cp -f ../../samples/reduced_ll_sfc_grib2.tmpl .
+reduced_ll_sfc_grib2.tmpl: $(top_srcdir)/samples/reduced_ll_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/reduced_ll_sfc_grib2.tmpl .
 
-regular_gg_ml_grib1.tmpl: ../../samples/regular_gg_ml_grib1.tmpl
-	cp -f ../../samples/regular_gg_ml_grib1.tmpl .
+regular_gg_ml_grib1.tmpl: $(top_srcdir)/samples/regular_gg_ml_grib1.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_ml_grib1.tmpl .
 
-regular_gg_ml_grib2.tmpl: ../../samples/regular_gg_ml_grib2.tmpl
-	cp -f ../../samples/regular_gg_ml_grib2.tmpl .
+regular_gg_ml_grib2.tmpl: $(top_srcdir)/samples/regular_gg_ml_grib2.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_ml_grib2.tmpl .
 
-regular_gg_pl_grib1.tmpl: ../../samples/regular_gg_pl_grib1.tmpl
-	cp -f ../../samples/regular_gg_pl_grib1.tmpl .
+regular_gg_pl_grib1.tmpl: $(top_srcdir)/samples/regular_gg_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_pl_grib1.tmpl .
 
-regular_gg_pl_grib2.tmpl: ../../samples/regular_gg_pl_grib2.tmpl
-	cp -f ../../samples/regular_gg_pl_grib2.tmpl .
+regular_gg_pl_grib2.tmpl: $(top_srcdir)/samples/regular_gg_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_pl_grib2.tmpl .
 
-regular_gg_sfc_grib1.tmpl: ../../samples/regular_gg_sfc_grib1.tmpl
-	cp -f ../../samples/regular_gg_sfc_grib1.tmpl .
+regular_gg_sfc_grib1.tmpl: $(top_srcdir)/samples/regular_gg_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_sfc_grib1.tmpl .
 
-regular_gg_sfc_grib2.tmpl: ../../samples/regular_gg_sfc_grib2.tmpl
-	cp -f ../../samples/regular_gg_sfc_grib2.tmpl .
+regular_gg_sfc_grib2.tmpl: $(top_srcdir)/samples/regular_gg_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/regular_gg_sfc_grib2.tmpl .
 
-regular_ll_pl_grib1.tmpl: ../../samples/regular_ll_pl_grib1.tmpl
-	cp -f ../../samples/regular_ll_pl_grib1.tmpl .
+regular_ll_pl_grib1.tmpl: $(top_srcdir)/samples/regular_ll_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/regular_ll_pl_grib1.tmpl .
 
-regular_ll_pl_grib2.tmpl: ../../samples/regular_ll_pl_grib2.tmpl
-	cp -f ../../samples/regular_ll_pl_grib2.tmpl .
+regular_ll_pl_grib2.tmpl: $(top_srcdir)/samples/regular_ll_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/regular_ll_pl_grib2.tmpl .
 
-regular_ll_sfc_grib1.tmpl: ../../samples/regular_ll_sfc_grib1.tmpl
-	cp -f ../../samples/regular_ll_sfc_grib1.tmpl .
+regular_ll_sfc_grib1.tmpl: $(top_srcdir)/samples/regular_ll_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/regular_ll_sfc_grib1.tmpl .
 
-regular_ll_sfc_grib2.tmpl: ../../samples/regular_ll_sfc_grib2.tmpl
-	cp -f ../../samples/regular_ll_sfc_grib2.tmpl .
+regular_ll_sfc_grib2.tmpl: $(top_srcdir)/samples/regular_ll_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/regular_ll_sfc_grib2.tmpl .
 
-rotated_ll_pl_grib1.tmpl: ../../samples/rotated_ll_pl_grib1.tmpl
-	cp -f ../../samples/rotated_ll_pl_grib1.tmpl .
+rotated_ll_pl_grib1.tmpl: $(top_srcdir)/samples/rotated_ll_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/rotated_ll_pl_grib1.tmpl .
 
-rotated_ll_pl_grib2.tmpl: ../../samples/rotated_ll_pl_grib2.tmpl
-	cp -f ../../samples/rotated_ll_pl_grib2.tmpl .
+rotated_ll_pl_grib2.tmpl: $(top_srcdir)/samples/rotated_ll_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/rotated_ll_pl_grib2.tmpl .
 
-rotated_ll_sfc_grib1.tmpl: ../../samples/rotated_ll_sfc_grib1.tmpl
-	cp -f ../../samples/rotated_ll_sfc_grib1.tmpl .
+rotated_ll_sfc_grib1.tmpl: $(top_srcdir)/samples/rotated_ll_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/rotated_ll_sfc_grib1.tmpl .
 
-rotated_ll_sfc_grib2.tmpl: ../../samples/rotated_ll_sfc_grib2.tmpl
-	cp -f ../../samples/rotated_ll_sfc_grib2.tmpl .
+rotated_ll_sfc_grib2.tmpl: $(top_srcdir)/samples/rotated_ll_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/rotated_ll_sfc_grib2.tmpl .
 
-sh_ml_grib1.tmpl: ../../samples/sh_ml_grib1.tmpl
-	cp -f ../../samples/sh_ml_grib1.tmpl .
+sh_ml_grib1.tmpl: $(top_srcdir)/samples/sh_ml_grib1.tmpl
+	cp -f $(top_srcdir)/samples/sh_ml_grib1.tmpl .
 
-sh_ml_grib2.tmpl: ../../samples/sh_ml_grib2.tmpl
-	cp -f ../../samples/sh_ml_grib2.tmpl .
+sh_ml_grib2.tmpl: $(top_srcdir)/samples/sh_ml_grib2.tmpl
+	cp -f $(top_srcdir)/samples/sh_ml_grib2.tmpl .
 
-sh_pl_grib1.tmpl: ../../samples/sh_pl_grib1.tmpl
-	cp -f ../../samples/sh_pl_grib1.tmpl .
+sh_pl_grib1.tmpl: $(top_srcdir)/samples/sh_pl_grib1.tmpl
+	cp -f $(top_srcdir)/samples/sh_pl_grib1.tmpl .
 
-sh_pl_grib2.tmpl: ../../samples/sh_pl_grib2.tmpl
-	cp -f ../../samples/sh_pl_grib2.tmpl .
+sh_pl_grib2.tmpl: $(top_srcdir)/samples/sh_pl_grib2.tmpl
+	cp -f $(top_srcdir)/samples/sh_pl_grib2.tmpl .
 
-sh_sfc_grib1.tmpl: ../../samples/sh_sfc_grib1.tmpl
-	cp -f ../../samples/sh_sfc_grib1.tmpl .
+sh_sfc_grib1.tmpl: $(top_srcdir)/samples/sh_sfc_grib1.tmpl
+	cp -f $(top_srcdir)/samples/sh_sfc_grib1.tmpl .
 
-sh_sfc_grib2.tmpl: ../../samples/sh_sfc_grib2.tmpl
-	cp -f ../../samples/sh_sfc_grib2.tmpl .
+sh_sfc_grib2.tmpl: $(top_srcdir)/samples/sh_sfc_grib2.tmpl
+	cp -f $(top_srcdir)/samples/sh_sfc_grib2.tmpl .
 
 
 
--- origsrc/grib_api-1.12.3/src/Makefile.am	2014-06-26 22:52:33.000000000 +0900
+++ src/grib_api-1.12.3/src/Makefile.am	2014-07-26 23:18:25.678637000 +0900
@@ -295,7 +295,7 @@ libgrib_api_la_SOURCES = $(libgrib_api_l
 
 libgrib_api_la_LIBADD = @LIBOBJS@
 libgrib_api_la_DEPENDENCIES = grib_api.h $(libgrib_api_la_LIBADD)
-libgrib_api_la_LDFLAGS = -version-info $(GRIB_ABI_CURRENT):$(GRIB_ABI_REVISION):$(GRIB_ABI_AGE)
+libgrib_api_la_LDFLAGS = -no-undefined -version-info $(GRIB_ABI_CURRENT):$(GRIB_ABI_REVISION):$(GRIB_ABI_AGE)
 
 # set the include path
 INCLUDES=  -I.
@@ -333,6 +333,6 @@ EXTRA_DIST= dummy.am griby.y gribl.l gri
             grib_bits_fast_big_endian_omp.c grib_bits_any_endian_omp.c \
             CMakeLists.txt grib_api_version.c.in
 
-include $(DEVEL_RULES)
+include $(srcdir)/$(DEVEL_RULES)
 
 
--- origsrc/grib_api-1.12.3/tests/Makefile.am	2014-06-26 22:52:50.000000000 +0900
+++ src/grib_api-1.12.3/tests/Makefile.am	2014-07-26 23:19:07.227465100 +0900
@@ -34,8 +34,16 @@
 
      LDADD = $(top_builddir)/src/libgrib_api.la $(EMOS_LIB)
 
-     INCLUDES   = -I$(top_builddir)/src
+     INCLUDES   = -I$(top_builddir)/src -I$(top_srcdir)/src
 
 EXTRA_DIST = $(TESTS) mf.rules filter_rules include.sh include.ctest.sh.in \
              lamb_az_eq_area.ref  CMakeLists.txt
  
+TESTS_ENVIRONMENT = \
+	GRIB_DEFINITION_PATH=${top_srcdir}/definitions \
+	GRIB_SAMPLES_PATH=${top_srcdir}/samples \
+	tools_dir=${abs_top_builddir}/tools/ \
+	tigge_dir=${top_srcdir}/tigge/ \
+	data_dir=${top_srcdir}/data \
+	test_dir=${abs_top_builddir}/tests \
+	def_dir=${top_srcdir}/definitions
--- origsrc/grib_api-1.12.3/tests/definitions.sh	2014-06-12 21:12:04.000000000 +0900
+++ src/grib_api-1.12.3/tests/definitions.sh	2014-07-26 23:32:17.817308800 +0900
@@ -17,6 +17,6 @@ REDIRECT=/dev/null
 
 for file in `find $GRIB_DEFINITION_PATH -name '*.def' -print`
 do
-  ${tools_dir}parser $file > $REDIRECT
+  ${tools_dir}grib_parser $file > $REDIRECT
 done
 
--- origsrc/grib_api-1.12.3/tigge/Makefile.am	2014-06-26 22:54:00.000000000 +0900
+++ src/grib_api-1.12.3/tigge/Makefile.am	2014-07-26 23:19:27.640551000 +0900
@@ -10,7 +10,7 @@ tigge_name_SOURCES      =  tigge_name.c
 tigge_split_SOURCES      =  tigge_split.c
 tigge_accumulations_SOURCES      =  tigge_accumulations.c
 
-INCLUDES =  -I$(top_builddir)/src
+INCLUDES =  -I$(top_builddir)/src -I$(top_srcdir)/src
 
 LDADD = $(top_builddir)/src/libgrib_api.la
 DEPENDENCIES = $(LDADD)
--- origsrc/grib_api-1.12.3/tools/Makefile.am	2014-06-26 22:54:36.000000000 +0900
+++ src/grib_api-1.12.3/tools/Makefile.am	2014-07-26 23:20:10.626879100 +0900
@@ -16,7 +16,7 @@ bin_PROGRAMS = grib_keys grib_histogram
                grib_debug grib_info grib_filter grib_ls grib_dump \
 			   grib2ppm grib_set grib_get grib_get_data grib_copy grib_repair \
 			   grib_packing grib_cmp grib_convert  grib_distance grib_corruption_check\
-			   grib_compare grib_list_keys parser grib_count grib_index_build grib1to2 \
+			   grib_compare grib_list_keys grib_parser grib_count grib_index_build grib1to2 \
                gg_sub_area_check grib_moments grib_to_netcdf grib_to_json
 
 noinst_PROGRAMS =  gaussian grib_diff mars_request xref all_keys #compile #dumpload grib_compare_old
@@ -36,7 +36,7 @@ grib_packing_SOURCES   =  grib_packing.c
 grib_histogram_SOURCES   =  grib_histogram.c
 grib_distance_SOURCES   =  grib_distance.c
 grib_debug_SOURCES   =  grib_debug.c
-parser_SOURCES      =  parser.c
+grib_parser_SOURCES      =  parser.c
 grib_list_keys_SOURCES      =  list_keys.c
 all_keys_SOURCES      =  all_keys.c
 grib2ppm_SOURCES    =  grib2ppm.c
@@ -63,7 +63,7 @@ gaussian_SOURCES     =  gaussian.c
 gg_sub_area_check_SOURCES     =  gg_sub_area_check.c
 grib1to2_SOURCES = grib1to2.txt
 
-INCLUDES =  -I$(top_builddir)/src
+INCLUDES =  -I$(top_builddir)/src -I$(top_srcdir)/src
 
 LDADD = libgrib_tools.la $(top_builddir)/src/libgrib_api.la
 DEPENDENCIES = $(LDADD)
@@ -76,7 +76,7 @@ CLEANFILES = libgrib_tools.la
 
 grib1to2$(EXEEXT): grib1to2.txt
 	rm -f grib1to2
-	sed "s|toolsdir|$(bindir)|" grib1to2.txt > grib1to2
+	sed "s|toolsdir|$(bindir)|" $(srcdir)/grib1to2.txt > grib1to2
 	chmod +x grib1to2
 
 noinst_HEADERS = grib_tools.h
