diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/stxxl-1.4.1/CMakeLists.txt src/stxxl-1.4.1/CMakeLists.txt
--- origsrc/stxxl-1.4.1/CMakeLists.txt	2014-10-29 18:04:03.000000000 +0900
+++ src/stxxl-1.4.1/CMakeLists.txt	2016-01-09 17:02:16.600938600 +0900
@@ -686,11 +686,11 @@ install(FILES ${PROJECT_BINARY_DIR}/incl
 # prepare pkg-config file
 
 configure_file(misc/cmake/stxxl.pc
-  "${PROJECT_BINARY_DIR}/${STXXL_LIBNAME}.pc" @ONLY)
+"${PROJECT_BINARY_DIR}/stxxl.pc" @ONLY)
 
 # copy the stxxl.pc file into lib/pkgconfig
 if(INSTALL_PKGCONFIG_DIR)
-  install(FILES ${PROJECT_BINARY_DIR}/${STXXL_LIBNAME}.pc
+  install(FILES ${PROJECT_BINARY_DIR}/stxxl.pc
     DESTINATION ${INSTALL_PKGCONFIG_DIR})
 endif()
 
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/stxxl-1.4.1/lib/CMakeLists.txt src/stxxl-1.4.1/lib/CMakeLists.txt
--- origsrc/stxxl-1.4.1/lib/CMakeLists.txt	2014-09-13 02:06:26.000000000 +0900
+++ src/stxxl-1.4.1/lib/CMakeLists.txt	2016-01-09 16:49:18.686747900 +0900
@@ -81,12 +81,16 @@ if(BUILD_SHARED_LIBS)
 
   # build stxxl SHARED library and link all programs again it
   add_library(stxxl SHARED ${LIBSTXXL_SOURCES})
+  if(CYGWIN)
+    set_target_properties(stxxl PROPERTIES SOVERSION 0)
+  else()
   set_target_properties(stxxl PROPERTIES OUTPUT_NAME "${STXXL_LIBNAME}")
+  endif()
   set_target_properties(stxxl PROPERTIES VERSION "${STXXL_VERSION_STRING}")
   install(TARGETS stxxl
     EXPORT stxxl-targets
     ARCHIVE DESTINATION ${INSTALL_LIB_DIR}
-    LIBRARY DESTINATION ${INSTALL_LIB_DIR})
+    RUNTIME DESTINATION ${INSTALL_BIN_DIR})
 
   if(BUILD_STATIC_LIBS)
     # but also build STATIC library
