--- origsrc/stxxl-1.4.1/CMakeLists.txt	2024-12-07 10:50:09.664374000 +0900
+++ src/stxxl-1.4.1/CMakeLists.txt	2024-12-07 12:34:10.093336200 +0900
@@ -686,11 +686,11 @@ install(FILES ${PROJECT_BINARY_DIR}/incl
 # prepare pkg-config file
 
 configure_file(misc/cmake/stxxl.pc
-  "${PROJECT_BINARY_DIR}/${STXXL_LIBNAME}.pc" @ONLY)
+  "${PROJECT_BINARY_DIR}/stxxl.pc" @ONLY)
 
 # copy the stxxl.pc file into lib/pkgconfig
 if(INSTALL_PKGCONFIG_DIR)
-  install(FILES ${PROJECT_BINARY_DIR}/${STXXL_LIBNAME}.pc
+  install(FILES ${PROJECT_BINARY_DIR}/stxxl.pc
     DESTINATION ${INSTALL_PKGCONFIG_DIR})
 endif()
 
--- origsrc/stxxl-1.4.1/lib/CMakeLists.txt	2014-09-13 02:06:26.000000000 +0900
+++ src/stxxl-1.4.1/lib/CMakeLists.txt	2024-12-07 12:35:43.714997300 +0900
@@ -83,10 +83,13 @@ if(BUILD_SHARED_LIBS)
   add_library(stxxl SHARED ${LIBSTXXL_SOURCES})
   set_target_properties(stxxl PROPERTIES OUTPUT_NAME "${STXXL_LIBNAME}")
   set_target_properties(stxxl PROPERTIES VERSION "${STXXL_VERSION_STRING}")
+  if (DEFINED SOVERSION)
+    set_target_properties(stxxl PROPERTIES SOVERSION "${SOVERSION}")
+  endif()
   install(TARGETS stxxl
     EXPORT stxxl-targets
     ARCHIVE DESTINATION ${INSTALL_LIB_DIR}
-    LIBRARY DESTINATION ${INSTALL_LIB_DIR})
+    RUNTIME DESTINATION ${INSTALL_BIN_DIR})
 
   if(BUILD_STATIC_LIBS)
     # but also build STATIC library
