--- origsrc/bppphyview-0.6.1/CMakeLists.txt	2018-08-16 18:36:18.000000000 +0900
+++ src/bppphyview-0.6.1/CMakeLists.txt	2025-06-16 12:42:56.172277400 +0900
@@ -7,9 +7,6 @@
 cmake_minimum_required (VERSION 2.8.11)
 project (bppphyview CXX)
 
-# Compile options
-set (CMAKE_CXX_FLAGS "-std=c++11 -Wall")
-
 IF(NOT CMAKE_BUILD_TYPE)
   SET(CMAKE_BUILD_TYPE RelWithDebInfo CACHE STRING
       "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel."
