--- origsrc/newicktools-0.1/Makefile	2015-01-08 21:11:38.000000000 +0900
+++ src/newicktools-0.1/Makefile	2025-12-23 19:01:00.567409400 +0900
@@ -1,22 +1,22 @@
 .PHONY:all test
 URL=http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=taxonomy&retmode=xml
-CC=g++
-CFLAGS=-O3 -Wall
+CXX?=g++
+CXXFLAGS?=-O3 -Wall
 
 all: taxonomy2newick newick2json newick2dot
 
 newick2json : newick2js.y newick2js.l
 	bison -d newick2js.y
 	flex newick2js.l
-	$(CC) -o $@ $(CFLAGS) newick2js.tab.c lex.yy.c
+	$(CXX) -o $@ $(CXXFLAGS) newick2js.tab.c lex.yy.c
 
 newick2dot : newick2dot.y newick2dot.l
 	bison -d newick2dot.y
 	flex newick2dot.l
-	$(CC) -o $@ $(CFLAGS) newick2dot.tab.c lex.yy.c	
+	$(CXX) -o $@ $(CXXFLAGS) newick2dot.tab.c lex.yy.c	
 
 taxonomy2newick : taxonomy2newick.cpp
-	$(CC) -o $@ $(CFLAGS) $< `xml2-config --cflags --libs`
+	$(CXX) -o $@ $(CXXFLAGS) $< `xml2-config --cflags --libs`
 
 test: taxonomy2newick newick2json newick2dot
 	curl -s '${URL}&id=9606' |./taxonomy2newick | ./newick2json
