--- origsrc/libwords/Makefile.am	2015-03-26 20:28:06.000000000 +0900
+++ src/libwords/Makefile.am	2015-03-26 20:35:26.556944600 +0900
@@ -2,12 +2,9 @@ SUBDIRS = src po tests
 #SUBDIRS = src po
 
 dist_doc_DATA = \
-	README \
+	README.md \
 	COPYING \
-	AUTHORS \
-	ChangeLog \
-	INSTALL \
-	NEWS
+	ChangeLog
 
 
 INTLTOOL_FILES = intltool-extract.in \
--- origsrc/libwords/src/Makefile.am	2015-03-26 20:28:06.000000000 +0900
+++ src/libwords/src/Makefile.am	2015-03-26 20:33:52.665222700 +0900
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = \
+AM_CPPFLAGS = -I$(top_srcdir)/src \
 	-DPACKAGE_LOCALE_DIR=\""$(localedir)"\" \
 	-DPACKAGE_SRC_DIR=\""$(srcdir)"\" \
 	-DPACKAGE_DATA_DIR=\""$(pkgdatadir)"\" \
@@ -17,7 +17,7 @@ libwords_la_SOURCES = \
 	words-radix-tree.c \
 	words-radix-tree.h
 
-libwords_la_LDFLAGS = \
+libwords_la_LDFLAGS = -no-undefined \
   -Wl,--export-dynamic
 
 libwords_la_LIBADD = \
@@ -42,11 +42,3 @@ public-headers.txt: Makefile
 CLEANFILES = public-headers.txt
 
 all-local: public-headers.txt
-
-install-data-local: \
-	@if test -f $(includedir)/libwords.h ; then					\
-	  echo "*** Old headers found in $(includedir). You should remove the" ;	\
-	  echo "*** contents of this directory and type 'make install' again." ;	\
-	  false ;									\
-	fi
-
--- origsrc/libwords/tests/Makefile.am	2015-03-26 20:28:06.000000000 +0900
+++ src/libwords/tests/Makefile.am	2015-03-26 20:32:16.586872000 +0900
@@ -1,8 +1,8 @@
-noinst_PROGRAMS = 
+noinst_PROGRAMS =
 
 common_cppflags =                                             \
   -I$(top_srcdir)                                             \
-  -I$(top_builddir)/src                                       \
+  -I$(top_srcdir)/src                                         \
   $(AM_CPPFLAGS)                                              \
   $(LIBWORDS_CPPFLAGS)
 
