--- origsrc/aws-c-common-0.4.63/CMakeLists.txt	2020-12-09 10:12:46.000000000 +0900
+++ src/aws-c-common-0.4.63/CMakeLists.txt	2020-12-21 11:21:50.729722800 +0900
@@ -235,14 +235,25 @@ else()
    set (TARGET_DIR "static")
 endif()
 
-install(EXPORT "${PROJECT_NAME}-targets"
-    DESTINATION "${LIBRARY_DIRECTORY}/${PROJECT_NAME}/cmake/${TARGET_DIR}"
-    NAMESPACE AWS::
-    COMPONENT Development)
-
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake"
-    DESTINATION "${LIBRARY_DIRECTORY}/${PROJECT_NAME}/cmake"
-    COMPONENT Development)
+if (CYGWIN)
+    install(EXPORT "${PROJECT_NAME}-targets"
+        DESTINATION "${LIBRARY_DIRECTORY}/cmake/${PROJECT_NAME}"
+        NAMESPACE AWS::
+        COMPONENT Development)
+
+    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake"
+        DESTINATION "${LIBRARY_DIRECTORY}/cmake/${PROJECT_NAME}"
+        COMPONENT Development)
+else()
+    install(EXPORT "${PROJECT_NAME}-targets"
+        DESTINATION "${LIBRARY_DIRECTORY}/${PROJECT_NAME}/cmake/${TARGET_DIR}"
+        NAMESPACE AWS::
+        COMPONENT Development)
+
+    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake"
+        DESTINATION "${LIBRARY_DIRECTORY}/${PROJECT_NAME}/cmake"
+        COMPONENT Development)
+endif()
 
 list(APPEND EXPORT_MODULES
     "cmake/AwsCFlags.cmake"
@@ -256,9 +267,15 @@ list(APPEND EXPORT_MODULES
     "cmake/AwsFeatureTests.cmake"
     )
 
-install(FILES ${EXPORT_MODULES}
-        DESTINATION "${LIBRARY_DIRECTORY}/cmake"
-        COMPONENT Development)
+if (CYGWIN)
+    install(FILES ${EXPORT_MODULES}
+            DESTINATION "${LIBRARY_DIRECTORY}/cmake/${PROJECT_NAME}"
+            COMPONENT Development)
+else()
+    install(FILES ${EXPORT_MODULES}
+            DESTINATION "${LIBRARY_DIRECTORY}/cmake"
+            COMPONENT Development)
+endif()
 
 # This should come last, to ensure all variables defined by cmake will be available for export
 configure_file(${CONFIG_HEADER_TEMPLATE}
@@ -272,3 +289,10 @@ if (ALLOW_CROSS_COMPILED_TESTS OR NOT CM
 endif()
 
 include(CPackConfig)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/aws-c-common.pc.in ${CMAKE_BINARY_DIR}/aws-c-common.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/aws-c-common.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/aws-c-common-0.4.63/aws-c-common.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/aws-c-common-0.4.63/aws-c-common.pc.in	2020-12-21 11:22:05.094132400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: aws-c-common
+Description: Core c99 package for AWS SDK for C
+Version: @VERSION@
+Libs: -L${libdir} -laws-c-common
+Cflags: -I${includedir}
--- origsrc/aws-c-common-0.4.63/cmake/AwsCFlags.cmake	2020-12-09 10:12:46.000000000 +0900
+++ src/aws-c-common-0.4.63/cmake/AwsCFlags.cmake	2020-12-21 11:04:40.650463700 +0900
@@ -51,7 +51,10 @@ function(aws_set_common_properties targe
         list(APPEND AWS_C_FLAGS "${_FLAGS}")
 
     else()
-        list(APPEND AWS_C_FLAGS -Wall -Werror -Wstrict-prototypes)
+        list(APPEND AWS_C_FLAGS -Wall -Wstrict-prototypes)
+        if (NOT CYGWIN)
+            list(APPEND AWS_C_FLAGS -Werror)
+        endif()
 
         if(NOT SET_PROPERTIES_NO_WEXTRA)
             list(APPEND AWS_C_FLAGS -Wextra)
--- origsrc/aws-c-common-0.4.63/cmake/aws-c-common-config.cmake	2020-12-09 10:12:46.000000000 +0900
+++ src/aws-c-common-0.4.63/cmake/aws-c-common-config.cmake	2020-12-21 11:04:40.707439800 +0900
@@ -4,9 +4,12 @@ if(WIN32 OR UNIX OR APPLE)
     find_package(Threads REQUIRED)
 endif()
 
-if (BUILD_SHARED_LIBS)
-    include(${CMAKE_CURRENT_LIST_DIR}/shared/@PROJECT_NAME@-targets.cmake)
+if (CYGWIN)
+    include(${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@-targets.cmake)
 else()
-    include(${CMAKE_CURRENT_LIST_DIR}/static/@PROJECT_NAME@-targets.cmake)
+    if (BUILD_SHARED_LIBS)
+        include(${CMAKE_CURRENT_LIST_DIR}/shared/@PROJECT_NAME@-targets.cmake)
+    else()
+        include(${CMAKE_CURRENT_LIST_DIR}/static/@PROJECT_NAME@-targets.cmake)
+    endif()
 endif()
-
--- origsrc/aws-c-common-0.4.63/source/posix/thread.c	2020-12-09 10:12:46.000000000 +0900
+++ src/aws-c-common-0.4.63/source/posix/thread.c	2020-12-21 11:19:02.739325200 +0900
@@ -160,7 +160,7 @@ int aws_thread_launch(
  * NUMA or not is setup in interleave mode.
  * Thread afinity is also not supported on Android systems, and honestly, if you're running android on a NUMA
  * configuration, you've got bigger problems. */
-#if !defined(__MACH__) && !defined(__ANDROID__)
+#if !defined(__MACH__) && !defined(__ANDROID__) && !defined(__CYGWIN__)
         if (options->cpu_id >= 0) {
             AWS_LOGF_INFO(
                 AWS_LS_COMMON_THREAD,
