--- origsrc/aws-c-common/CMakeLists.txt	2018-08-12 15:04:16.000000000 +0900
+++ src/aws-c-common/CMakeLists.txt	2018-08-12 15:20:48.633404100 +0900
@@ -102,6 +102,10 @@ aws_set_common_properties(${CMAKE_PROJEC
 aws_add_sanitizers(${CMAKE_PROJECT_NAME} BLACKLIST "sanitizer-blacklist.txt")
 target_link_libraries(${CMAKE_PROJECT_NAME} PUBLIC ${PLATFORM_LIBS})
 
+if (CYGWIN)
+        set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES SOVERSION 0)
+endif()
+
 target_include_directories(${CMAKE_PROJECT_NAME} PUBLIC
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
         $<INSTALL_INTERFACE:include>)
@@ -116,15 +120,22 @@ endforeach()
 install(
     TARGETS ${CMAKE_PROJECT_NAME} EXPORT ${CMAKE_PROJECT_NAME}-config
         ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
-        LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
+        RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/bin
         COMPONENT library
 )
 
 export(TARGETS ${CMAKE_PROJECT_NAME} FILE ${CMAKE_PROJECT_NAME}-config.cmake)
-install(EXPORT ${CMAKE_PROJECT_NAME}-config DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/${CMAKE_PROJECT_NAME}/cmake/")
+install(EXPORT ${CMAKE_PROJECT_NAME}-config DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/cmake/${CMAKE_PROJECT_NAME}")
 
 file(GLOB EXPORT_MODULES "cmake/*.cmake")
-install(FILES ${EXPORT_MODULES} DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/cmake/")
+install(FILES ${EXPORT_MODULES} DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/cmake/${CMAKE_PROJECT_NAME}")
 
 include(CTest)
 add_subdirectory(tests)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/aws-c-common.pc.in ${CMAKE_BINARY_DIR}/aws-c-common.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/aws-c-common.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/aws-c-common/aws-c-common.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/aws-c-common/aws-c-common.pc.in	2018-08-12 15:21:05.639718400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: aws-c-common
+Description: Core c99 package for AWS SDK for C
+Version: @VERSION@
+Libs: -L${libdir} -laws-c-common
+Cflags: -I${includedir}
