--- origsrc/procbench_0.9.0aubuntu3~intrepid/src/bench/Makefile.am	2008-12-30 22:02:57.000000000 +0900
+++ src/procbench_0.9.0aubuntu3~intrepid/src/bench/Makefile.am	2023-10-08 13:36:34.927236400 +0900
@@ -1,7 +1,6 @@
 lib_LTLIBRARIES= libpbbench.la
 libpbbench_la_SOURCES = ctbench.c  ctest1.c  ctest2.c  ctfibo.c  ctheapsort.c  ctmatrix.c  ctrandom.c
+libpbbench_la_LDFLAGS = -no-undefined -version-info 1
 
 AM_CFLAGS = -O2 -fPIC -c
 
-AM_LDFLAGS = -shared -Wl,-soname,libpbbench.so.1 
-
--- origsrc/procbench_0.9.0aubuntu3~intrepid/src/idriver-i386.cpp	2008-12-30 22:02:57.000000000 +0900
+++ src/procbench_0.9.0aubuntu3~intrepid/src/idriver-i386.cpp	2023-10-08 13:36:34.927236400 +0900
@@ -29,6 +29,8 @@
 #include <stdarg.h>
 #include "cross.h"
 
+#include <cstdio>
+
 const char *host_detect_local_cpu ();
 const char *host_detect_local_cpu_old ();
 
--- origsrc/procbench_0.9.0aubuntu3~intrepid/src/multi_cpu.cpp	2008-12-30 22:02:57.000000000 +0900
+++ src/procbench_0.9.0aubuntu3~intrepid/src/multi_cpu.cpp	2023-10-08 13:36:34.927236400 +0900
@@ -24,8 +24,12 @@
 #include "cross.h"
 #ifdef __WIN32__
 #include <windows.h>
+#elif defined(__CYGWIN__)
+#include <sys/sysinfo.h>
 #endif
 
+#include <cstdio>
+
 #define bit_HTT (1 << 28)
 unsigned int HTTSupported(void)
 {
@@ -82,6 +86,9 @@ void host_detect_multi_cpu ()
     	SYSTEM_INFO info;
 	GetSystemInfo (&info);			//how to detect?!?
 	procs_online = num_procs = (unsigned char) info.dwNumberOfProcessors;	  
+	#elif defined(__CYGWIN__)
+	num_procs = get_nprocs_conf();
+	procs_online = get_nprocs();
 	#else	
 	num_procs = sysconf (_SC_NPROCESSORS_CONF);
 	procs_online = sysconf (_SC_NPROCESSORS_ONLN);
--- origsrc/procbench_0.9.0aubuntu3~intrepid/src/pbbench.cpp	2008-12-30 22:02:57.000000000 +0900
+++ src/procbench_0.9.0aubuntu3~intrepid/src/pbbench.cpp	2023-10-08 13:36:34.927236400 +0900
@@ -61,11 +61,16 @@ int *PBBench(int arg)
 	int x;
 	char *error;
 	printf("\n------=[CPU Benchmarks]=----------------------------------------------\n");
-	lib_handle = dlopen("libpbbench.so", RTLD_LAZY);
+#if defined(__CYGWIN__)
+#define SO_NAME "cygpbbench-1.dll"
+#else
+#define SO_NAME "libpbbench.so"
+#endif
+	lib_handle = dlopen(SO_NAME, RTLD_LAZY);
 	if (!lib_handle) 
 	{
 		fprintf(stderr, "%s\n", dlerror());
-		fprintf(stderr, "probably \"libpbbench.so\" file is missing\n", dlerror());
+		fprintf(stderr, "probably \"" SO_NAME "\" file is missing\n");
 		return 0;
 	}
 
