--- origsrc/procbench_0.9.0aubuntu3~intrepid/configure.ac	2008-12-30 22:02:56.000000000 +0900
+++ src/procbench_0.9.0aubuntu3~intrepid/configure.ac	2010-11-14 13:51:32.902217200 +0900
@@ -7,8 +7,7 @@ AC_CONFIG_SRCDIR([src])
 AC_CONFIG_HEADERS([src/config.h])
 AM_INIT_AUTOMAKE
 
-AC_DISABLE_STATIC
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_LIBTOOL_CXX
 
 
--- origsrc/procbench_0.9.0aubuntu3~intrepid/src/bench/Makefile.am	2008-12-30 22:02:57.000000000 +0900
+++ src/procbench_0.9.0aubuntu3~intrepid/src/bench/Makefile.am	2010-11-14 13:53:09.561206000 +0900
@@ -1,7 +1,8 @@
 lib_LTLIBRARIES= libpbbench.la
 libpbbench_la_SOURCES = ctbench.c  ctest1.c  ctest2.c  ctfibo.c  ctheapsort.c  ctmatrix.c  ctrandom.c
+libpbbench_la_LDFLAGS = -no-undefined
 
 AM_CFLAGS = -O2 -fPIC -c
 
-AM_LDFLAGS = -shared -Wl,-soname,libpbbench.so.1 
+#AM_LDFLAGS = -shared -Wl,-soname,libpbbench.so.1 
 
--- origsrc/procbench_0.9.0aubuntu3~intrepid/src/pbbench.cpp	2008-12-30 22:02:57.000000000 +0900
+++ src/procbench_0.9.0aubuntu3~intrepid/src/pbbench.cpp	2010-11-14 14:34:04.641438000 +0900
@@ -61,11 +61,16 @@ int *PBBench(int arg)
 	int x;
 	char *error;
 	printf("\n------=[CPU Benchmarks]=----------------------------------------------\n");
-	lib_handle = dlopen("libpbbench.so", RTLD_LAZY);
+#if defined(__CYGWIN__)
+#define SO_NAME "cygpbbench-0.dll"
+#else
+#define SO_NAME "libpbbench.so"
+#endif
+	lib_handle = dlopen(SO_NAME, RTLD_LAZY);
 	if (!lib_handle) 
 	{
 		fprintf(stderr, "%s\n", dlerror());
-		fprintf(stderr, "probably \"libpbbench.so\" file is missing\n", dlerror());
+		fprintf(stderr, "probably \"" SO_NAME "\" file is missing\n");
 		return 0;
 	}
 
