--- origsrc/gupnp-av-0.11.2/configure.ac	2012-10-25 19:52:48.000000000 +0900
+++ src/gupnp-av-0.11.2/configure.ac	2012-11-04 23:44:48.290091000 +0900
@@ -45,7 +45,7 @@ GOBJECT_INTROSPECTION_CHECK([0.9.5])
 # vapigen
 AS_IF([test "x$found_introspection" = "xyes"],
       [
-       VALA_PROG_VAPIGEN([0.14])
+       VAPIGEN_CHECK
       ]
 )
 AM_CONDITIONAL([HAVE_VAPIGEN], [test "x$VAPIGEN" != "x"])
