--- origsrc/gupnp-av-0.11.6/configure.ac	2012-12-05 19:22:00.000000000 +0900
+++ src/gupnp-av-0.11.6/configure.ac	2013-01-13 14:19:39.667184900 +0900
@@ -45,7 +45,7 @@ GOBJECT_INTROSPECTION_CHECK([0.9.5])
 # vapigen
 AS_IF([test "x$found_introspection" = "xyes"],
       [
-       VALA_PROG_VAPIGEN([0.14])
+       VAPIGEN_CHECK
       ]
 )
 AM_CONDITIONAL([HAVE_VAPIGEN], [test "x$VAPIGEN" != "x"])
