--- origsrc/bppsuite-2.4.1/CMakeLists.txt	2018-08-16 18:31:28.000000000 +0900
+++ src/bppsuite-2.4.1/CMakeLists.txt	2025-06-16 15:14:56.221375600 +0900
@@ -7,9 +7,6 @@
 cmake_minimum_required (VERSION 2.8.11)
 project (bppsuite CXX)
 
-# Compile options
-SET (CMAKE_CXX_FLAGS "-std=c++11 -Wall -Weffc++ -Wshadow -Wconversion")
-
 IF(NOT CMAKE_BUILD_TYPE)
   SET(CMAKE_BUILD_TYPE RelWithDebInfo CACHE STRING
       "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel."
