--- origsrc/CGNS-3.4.0/CGNS.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/CGNS-3.4.0/CGNS.pc.in	2019-05-30 13:15:30.551756200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: CGNS
+Description: Computational Fluid Dynamics General Notation System
+Version: @VERSION@
+Libs: -L${libdir} -lcgns
+Cflags: -I${includedir}
--- origsrc/CGNS-3.4.0/CMakeLists.txt	2019-03-05 01:10:05.000000000 +0900
+++ src/CGNS-3.4.0/CMakeLists.txt	2019-05-30 13:15:17.861903700 +0900
@@ -389,3 +389,10 @@ endif (CGNS_BUILD_TESTING)
 # Include the src directory
 add_subdirectory(src)
 
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/CGNS.pc.in ${CMAKE_BINARY_DIR}/CGNS.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/CGNS.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/CGNS-3.4.0/src/CMakeLists.txt	2019-03-05 01:10:05.000000000 +0900
+++ src/CGNS-3.4.0/src/CMakeLists.txt	2019-05-30 11:52:24.277763900 +0900
@@ -539,14 +539,15 @@ set(CGNS_BUILD_SHARED "ON" CACHE BOOL "B
 option(CGNS_USE_SHARED "Link programs to the CGNS shared library" "ON")
 
 # Build a static version of the library
-if (CGNS_ENABLE_FORTRAN)
-    add_library(cgns_static STATIC ${cgns_FILES} $<TARGET_OBJECTS:cgns_f2c>)
-else (CGNS_ENABLE_FORTRAN)
-  add_library(cgns_static STATIC ${cgns_FILES})
-endif (CGNS_ENABLE_FORTRAN)
-# Needed to work around a CMake > 3.8 bug on Windows with MSVS and Intel Fortran
-set_property(TARGET cgns_static PROPERTY LINKER_LANGUAGE C)
-
+if (NOT CGNS_BUILD_SHARED)
+  if (CGNS_ENABLE_FORTRAN)
+      add_library(cgns_static STATIC ${cgns_FILES} $<TARGET_OBJECTS:cgns_f2c>)
+  else (CGNS_ENABLE_FORTRAN)
+    add_library(cgns_static STATIC ${cgns_FILES})
+  endif (CGNS_ENABLE_FORTRAN)
+  # Needed to work around a CMake > 3.8 bug on Windows with MSVS and Intel Fortran
+  set_property(TARGET cgns_static PROPERTY LINKER_LANGUAGE C)
+endif()
 
 # Build a shared version of the library
 if(CGNS_BUILD_SHARED)
@@ -575,22 +576,26 @@ else(CGNS_BUILD_SHARED)
 endif(CGNS_BUILD_SHARED)
 
 # Change the output name of the library to be libcgns
-set_target_properties(cgns_static PROPERTIES OUTPUT_NAME cgns)
-set_target_properties(cgns_static PROPERTIES CLEAN_DIRECT_OUTPUT 1)
+if (NOT CGNS_BUILD_SHARED)
+  set_target_properties(cgns_static PROPERTIES OUTPUT_NAME cgns)
+  set_target_properties(cgns_static PROPERTIES CLEAN_DIRECT_OUTPUT 1)
+endif()
 if(CGNS_BUILD_SHARED)
   # for windows we need to change the name of the shared library
   # for both static and shared version to exist
-  if (WIN32 OR CYGWIN)
+  if (WIN32)
     set_target_properties(cgns_shared PROPERTIES OUTPUT_NAME cgnsdll)
-  else (WIN32 OR CYGWIN)
+  else (WIN32)
     set_target_properties(cgns_shared PROPERTIES OUTPUT_NAME cgns)
-  endif (WIN32 OR CYGWIN)
+  endif (WIN32)
   set_target_properties(cgns_shared PROPERTIES CLEAN_DIRECT_OUTPUT 1)
 endif(CGNS_BUILD_SHARED)
 
 # Set the version numbers
-set_target_properties(cgns_static PROPERTIES VERSION "${CGNS_VERSION}")
-set_target_properties(cgns_static PROPERTIES SOVERSION "${CGNS_VERSION}")
+if (NOT CGNS_BUILD_SHARED)
+  set_target_properties(cgns_static PROPERTIES VERSION "${CGNS_VERSION}")
+  set_target_properties(cgns_static PROPERTIES SOVERSION "${CGNS_VERSION}")
+endif()
 
 if(CGNS_BUILD_SHARED)
   set_target_properties(cgns_shared PROPERTIES VERSION "${CGNS_VERSION}")
@@ -598,7 +603,9 @@ if(CGNS_BUILD_SHARED)
 endif(CGNS_BUILD_SHARED)
 
 # Set the install path of the static library
-install(TARGETS cgns_static ARCHIVE DESTINATION lib)
+if (NOT CGNS_BUILD_SHARED)
+  install(TARGETS cgns_static ARCHIVE DESTINATION lib)
+endif()
 # Set the install path of the shared library
 if(CGNS_BUILD_SHARED)
   # for windows, need to install both cgnsdll.dll and cgnsdll.lib
--- origsrc/CGNS-3.4.0/src/Test_UserGuideCode/C_code/CMakeLists.txt	2019-03-05 01:10:05.000000000 +0900
+++ src/CGNS-3.4.0/src/Test_UserGuideCode/C_code/CMakeLists.txt	2019-05-30 11:55:43.533809600 +0900
@@ -4,7 +4,11 @@
 
 # Link all the executables to cgns and hdf5
 link_directories(.)
-link_libraries(cgns_static)
+if (CGNS_BUILD_SHARED)
+	link_libraries(cgns_shared)
+else ()
+	link_libraries(cgns_static)
+endif ()
 
 if (CGNS_ENABLE_HDF5 AND HDF5_LIBRARY)
   link_libraries(${HDF5_LIBRARY})
--- origsrc/CGNS-3.4.0/src/cgnstools/cgnscalc/CMakeLists.txt	2019-03-05 01:10:05.000000000 +0900
+++ src/CGNS-3.4.0/src/cgnstools/cgnscalc/CMakeLists.txt	2019-05-30 11:57:05.084578000 +0900
@@ -5,12 +5,12 @@
 include_directories(${TCL_INCLUDE_PATH} ${TK_INCLUDE_PATH})
 include_directories(../calclib)
 link_directories(.)
-if (CGNS_USE_SHARED)
+link_libraries(calclib ${TCL_LIBRARY} ${TK_LIBRARY})
+if (CGNS_BUILD_SHARED)
   link_libraries(cgns_shared)
-else (CGNS_USE_SHARED)
+else (CGNS_BUILD_SHARED)
   link_libraries(cgns_static)
-endif (CGNS_USE_SHARED)
-link_libraries(calclib ${TCL_LIBRARY} ${TK_LIBRARY})
+endif (CGNS_BUILD_SHARED)
 
 if (CGNS_ENABLE_HDF5 AND HDF5_LIBRARY)
   link_libraries(${HDF5_LIBRARY})
--- origsrc/CGNS-3.4.0/src/cgnstools/cgnsplot/CMakeLists.txt	2019-03-05 01:10:05.000000000 +0900
+++ src/CGNS-3.4.0/src/cgnstools/cgnsplot/CMakeLists.txt	2019-05-30 11:57:56.067918700 +0900
@@ -5,11 +5,11 @@
 include_directories(${TCL_INCLUDE_PATH} ${TK_INCLUDE_PATH} ${OPENGL_INCLUDE_DIR})
 include_directories(../common)
 link_directories(. ../tkogl)
-if (CGNS_USE_SHARED)
+if (CGNS_BUILD_SHARED)
   link_libraries(cgns_shared)
-else (CGNS_USE_SHARED)
+else (CGNS_BUILD_SHARED)
   link_libraries(cgns_static)
-endif (CGNS_USE_SHARED)
+endif (CGNS_BUILD_SHARED)
 link_libraries(tkogl ${TCL_LIBRARY} ${TK_LIBRARY}
 	${OPENGL_gl_LIBRARY} ${OPENGL_glu_LIBRARY})
 
--- origsrc/CGNS-3.4.0/src/cgnstools/utilities/CMakeLists.txt	2019-03-05 01:10:05.000000000 +0900
+++ src/CGNS-3.4.0/src/cgnstools/utilities/CMakeLists.txt	2019-05-30 11:58:28.923289800 +0900
@@ -8,12 +8,12 @@ include_directories(../common)
 
 # cgns_info
 link_directories(.)
-if (CGNS_USE_SHARED)
+link_libraries(calclib)
+if (CGNS_BUILD_SHARED)
   link_libraries(cgns_shared)
-else (CGNS_USE_SHARED)
+else (CGNS_BUILD_SHARED)
   link_libraries(cgns_static)
-endif (CGNS_USE_SHARED)
-link_libraries(calclib)
+endif (CGNS_BUILD_SHARED)
 if (NOT WIN32)
   link_libraries(m)
 endif (NOT WIN32)
--- origsrc/CGNS-3.4.0/src/tests/CMakeLists.txt	2019-03-05 01:10:05.000000000 +0900
+++ src/CGNS-3.4.0/src/tests/CMakeLists.txt	2019-05-30 11:59:03.209281900 +0900
@@ -4,7 +4,11 @@
 
 # Link all the executables to cgns and hdf5
 link_directories(.)
-link_libraries(cgns_static)
+if (CGNS_BUILD_SHARED)
+  link_libraries(cgns_shared)
+else ()
+  link_libraries(cgns_static)
+endif ()
 
 if (CGNS_ENABLE_HDF5 AND HDF5_LIBRARY)
   link_libraries(${HDF5_LIBRARY})
