--- origsrc/octetos-coreutils-0.17.0-alpha/src/Makefile.am	2021-07-26 11:44:51.000000000 +0900
+++ src/octetos-coreutils-0.17.0-alpha/src/Makefile.am	2026-06-19 06:28:30.382417100 +0900
@@ -5,7 +5,7 @@ lib_LTLIBRARIES = liboctetos-coreutils.l
 
 SOURCES = shell.cc ls.cc cd.cc mkdir.cc rm.cc touch.cc rename.cc exists.cc
 
-liboctetos_coreutils_la_LDFLAGS = -shared -fPIC
+liboctetos_coreutils_la_LDFLAGS = -no-undefined -shared -fPIC
 liboctetos_coreutils_la_CPPFLAGS =  $(OCTETOS_CORE_CFLAGS)
 liboctetos_coreutils_la_LIBADD = $(OCTETOS_CORE_LIBS)
 liboctetos_coreutils_la_SOURCES = $(SOURCES)
