--- origsrc/dtcmp-1.1.3/configure.ac	2021-09-30 06:13:13.000000000 +0900
+++ src/dtcmp-1.1.3/configure.ac	2021-10-09 22:52:43.741370400 +0900
@@ -23,7 +23,7 @@ AC_PROG_CC
 AM_PROG_CC_C_O
 AC_PROG_CXX
 LX_FIND_MPI
-X_AC_LWGRP
+PKG_CHECK_MODULES([LWGRP], [liblwgrp])
 
 ##
 # enable libtool
--- origsrc/dtcmp-1.1.3/src/Makefile.am	2021-09-30 06:13:13.000000000 +0900
+++ src/dtcmp-1.1.3/src/Makefile.am	2021-10-09 22:52:43.744404400 +0900
@@ -50,5 +50,5 @@ libdtcmp_la_SOURCES = \
   dtcmp_rankv_sort.c \
   dtcmp_seg_exscan.c
 libdtcmp_la_CFLAGS = $(INCLUDES)
-libdtcmp_la_LIBADD = $(LWGRP_LDFLAGS) $(LWGRP_LIBS)
-libdtcmp_la_LDFLAGS = -avoid-version
+libdtcmp_la_LIBADD = -lmpi $(LWGRP_LDFLAGS) $(LWGRP_LIBS)
+libdtcmp_la_LDFLAGS = -no-undefined
