--- origsrc/libslax-0.16.18/extensions/bit/Makefile.am	2013-08-07 05:28:21.000000000 +0900
+++ src/libslax-0.16.18/extensions/bit/Makefile.am	2013-09-19 00:02:26.438304800 +0900
@@ -35,7 +35,7 @@ LIBS = \
     -lexslt \
     ${LIBXML_LIBS}
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_bit_la_LIBADD = ${top_builddir}/libslax/libslax.la
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
@@ -46,6 +46,7 @@ LIBS += -ledit
 endif
 libext_bit_la_SOURCES = \
     ext_bit.c
+libext_bit_la_LDFLAGS = -no-undefined -module -avoid-version
 
 pkglibdir = ${SLAX_EXTDIR}
 
--- origsrc/libslax-0.16.18/extensions/curl/Makefile.am	2013-08-07 05:28:21.000000000 +0900
+++ src/libslax-0.16.18/extensions/curl/Makefile.am	2013-09-19 00:03:16.221508000 +0900
@@ -39,7 +39,7 @@ LIBS = \
     ${LIBXML_LIBS} \
     ${LIBCURL_LIBS}
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_curl_la_LIBADD = ${top_builddir}/libslax/libslax.la
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
@@ -51,6 +51,7 @@ endif
 
 libext_curl_la_SOURCES = \
     ext_curl.c
+libext_curl_la_LDFLAGS = -no-undefined -module -avoid-version
 
 noinst_HEADERS = \
     ext_curl.h
--- origsrc/libslax-0.16.18/extensions/os/Makefile.am	2013-08-07 05:28:21.000000000 +0900
+++ src/libslax-0.16.18/extensions/os/Makefile.am	2013-09-19 00:03:49.832836100 +0900
@@ -31,7 +31,7 @@ pkglib_LTLIBRARIES = libext_os.la
 LIBS = \
     ${LIBXSLT_LIBS} -lexslt ${LIBXML_LIBS}
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_os_la_LIBADD = ${top_builddir}/libslax/libslax.la
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
@@ -42,6 +42,7 @@ LIBS += -ledit
 endif
 libext_os_la_SOURCES = \
     ext_os.c
+libext_os_la_LDFLAGS = -no-undefined -module -avoid-version
 
 slaxextdir = ${SLAX_EXTDIR}
 pkglibdir = ${slaxextdir}
--- origsrc/libslax-0.16.18/extensions/xutil/Makefile.am	2013-08-07 05:28:21.000000000 +0900
+++ src/libslax-0.16.18/extensions/xutil/Makefile.am	2013-09-19 00:04:23.686351700 +0900
@@ -36,7 +36,7 @@ LIBS = \
     -lexslt \
     ${LIBXML_LIBS}
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_xutil_la_LIBADD = ${top_builddir}/libslax/libslax.la
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
@@ -59,6 +59,7 @@ libext_xutil_la_SOURCES = \
     jsonparser.c \
     jsonlexer.c \
     jsonwriter.c
+libext_xutil_la_LDFLAGS = -no-undefined -module -avoid-version
 
 pkglibdir = ${SLAX_EXTDIR}
 
--- origsrc/libslax-0.16.18/libslax/Makefile.am	2013-08-14 06:41:10.000000000 +0900
+++ src/libslax-0.16.18/libslax/Makefile.am	2013-09-19 00:04:47.203929800 +0900
@@ -28,6 +28,7 @@ AM_CFLAGS += \
 
 lib_LTLIBRARIES = libslax.la
 libslax_la_LIBADD = ${LIBXSLT_LIBS} -lexslt ${LIBXML_LIBS}
+libslax_la_LDFLAGS = -no-undefined
 
 slaxinc_HEADERS = \
      slax.h \
