--- origsrc/libslax-0.17.1/extensions/bit/Makefile.am	2013-09-20 07:34:57.000000000 +0900
+++ src/libslax-0.17.1/extensions/bit/Makefile.am	2013-10-14 00:17:01.309221700 +0900
@@ -35,7 +35,8 @@ LIBS = \
     -lexslt \
     ${LIBXML_LIBS}
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_bit_la_LIBADD = ${top_builddir}/libslax/libslax.la
+libext_bit_la_LDFLAGS = -no-undefined -module -avoid-version
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
--- origsrc/libslax-0.17.1/extensions/curl/Makefile.am	2013-09-20 07:34:57.000000000 +0900
+++ src/libslax-0.17.1/extensions/curl/Makefile.am	2013-10-14 00:17:32.033831100 +0900
@@ -39,7 +39,8 @@ LIBS = \
     ${LIBXML_LIBS} \
     ${LIBCURL_LIBS}
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_curl_la_LIBADD = ${top_builddir}/libslax/libslax.la
+libext_curl_la_LDFLAGS = -no-undefined -module -avoid-version
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
--- origsrc/libslax-0.17.1/extensions/exslt/Makefile.am	2013-09-20 07:34:57.000000000 +0900
+++ src/libslax-0.17.1/extensions/exslt/Makefile.am	2013-10-14 00:18:26.687151400 +0900
@@ -31,7 +31,8 @@ pkglib_LTLIBRARIES = libext_exslt.la
 LIBS = \
     ${LIBXSLT_LIBS} -lexslt ${LIBXML_LIBS}
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_exslt_la_LIBADD = ${top_builddir}/libslax/libslax.la
+libext_exslt_la_LDFLAGS = -no-undefined -module -avoid-version
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
--- origsrc/libslax-0.17.1/extensions/os/Makefile.am	2013-09-20 07:34:57.000000000 +0900
+++ src/libslax-0.17.1/extensions/os/Makefile.am	2013-10-14 00:23:58.413713900 +0900
@@ -31,7 +31,8 @@ pkglib_LTLIBRARIES = libext_os.la
 LIBS = \
     ${LIBXSLT_LIBS} -lexslt ${LIBXML_LIBS}
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_os_la_LIBADD = ${top_builddir}/libslax/libslax.la
+libext_os_la_LDFLAGS = -no-undefined -module -avoid-version
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
--- origsrc/libslax-0.17.1/extensions/xutil/Makefile.am	2013-09-20 07:34:57.000000000 +0900
+++ src/libslax-0.17.1/extensions/xutil/Makefile.am	2013-10-14 00:19:21.308245100 +0900
@@ -34,7 +34,8 @@ LIBS = \
     -lexslt \
     ${LIBXML_LIBS}
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_xutil_la_LIBADD = ${top_builddir}/libslax/libslax.la
+libext_xutil_la_LDFLAGS = -no-undefined -module -avoid-version
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
--- origsrc/libslax-0.17.1/libslax/Makefile.am	2013-09-20 07:34:57.000000000 +0900
+++ src/libslax-0.17.1/libslax/Makefile.am	2013-10-14 00:19:57.551409200 +0900
@@ -27,6 +27,7 @@ AM_CFLAGS += \
 
 lib_LTLIBRARIES = libslax.la
 libslax_la_LIBADD = ${LIBXSLT_LIBS} -lexslt ${LIBXML_LIBS}
+libslax_la_LDFLAGS = -no-undefined
 
 slaxinc_HEADERS = \
      slax.h \
