--- origsrc/libslax-0.20.2/extensions/bit/Makefile.am	2016-08-11 03:51:46.000000000 +0900
+++ src/libslax-0.20.2/extensions/bit/Makefile.am	2016-12-30 23:38:11.849742400 +0900
@@ -40,7 +40,8 @@ LIBS = \
     ${LIBXML_LIBS} \
     -L${top_builddir}/libslax -lslax
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_bit_la_LIBADD = ${top_builddir}/libslax/libslax.la
+libext_bit_la_LDFLAGS = -no-undefined -module -avoid-version
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
@@ -61,7 +62,7 @@ install-exec-hook:
 		if [ x"$$DLNAME" = x ]; \
                     then DLNAME=${LIBNAME}.${SLAX_LIBEXT}; fi ; \
 		if [ "$(build_os)" = "cygwin" ]; \
-		    then DLNAME="../bin/$$DLNAME"; fi ; \
+		    then DLNAME="$$DLNAME"; fi ; \
 		echo Install link $$DLNAME "->" ${UGLY_NAME} "..." ; \
 		mkdir -p ${DESTDIR}${SLAX_EXTDIR} ; \
 		cd ${DESTDIR}${SLAX_EXTDIR} \
--- origsrc/libslax-0.20.2/extensions/curl/Makefile.am	2016-08-11 03:51:46.000000000 +0900
+++ src/libslax-0.20.2/extensions/curl/Makefile.am	2016-12-30 23:39:29.079114700 +0900
@@ -44,7 +44,8 @@ LIBS = \
     ${LIBCURL_LIBS} \
     -L${top_builddir}/libslax -lslax
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_curl_la_LIBADD = ${top_builddir}/libslax/libslax.la
+libext_curl_la_LDFLAGS = -no-undefined -module -avoid-version
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
@@ -69,7 +70,7 @@ install-exec-hook:
 		if [ x"$$DLNAME" = x ]; \
                     then DLNAME=${LIBNAME}.${SLAX_LIBEXT}; fi ; \
 		if [ "$(build_os)" = "cygwin" ]; \
-		    then DLNAME="../bin/$$DLNAME"; fi ; \
+		    then DLNAME="$$DLNAME"; fi ; \
 		echo Install link $$DLNAME "->" ${UGLY_NAME} "..." ; \
 		mkdir -p ${DESTDIR}${SLAX_EXTDIR} ; \
 		cd ${DESTDIR}${SLAX_EXTDIR} \
--- origsrc/libslax-0.20.2/extensions/db/Makefile.am	2016-08-11 03:51:46.000000000 +0900
+++ src/libslax-0.20.2/extensions/db/Makefile.am	2016-12-30 23:39:39.357431200 +0900
@@ -49,7 +49,8 @@ LIBS = \
     ${LIBDB_LIBS} \
     -L${top_builddir}/libslax -lslax
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_db_la_LIBADD = ${top_builddir}/libslax/libslax.la
+libext_db_la_LDFLAGS = -no-undefined -module -avoid-version
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
@@ -74,7 +75,7 @@ install-exec-hook:
 		if [ x"$$DLNAME" = x ]; \
                     then DLNAME=${LIBNAME}.${SLAX_LIBEXT}; fi ; \
 		if [ "$(build_os)" = "cygwin" ]; \
-		    then DLNAME="../bin/$$DLNAME"; fi ; \
+		    then DLNAME="$$DLNAME"; fi ; \
 		echo Install link $$DLNAME "->" ${UGLY_NAME} "..." ; \
 		mkdir -p ${DESTDIR}${SLAX_EXTDIR} ; \
 		cd ${DESTDIR}${SLAX_EXTDIR} \
--- origsrc/libslax-0.20.2/extensions/db/sqlite/Makefile.am	2016-08-11 03:51:46.000000000 +0900
+++ src/libslax-0.20.2/extensions/db/sqlite/Makefile.am	2016-12-30 23:39:48.996522800 +0900
@@ -63,7 +63,8 @@ else
     LIBS += ${SQLITE3_LIBS} -lsqlite3
 endif
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_db_sqlite_la_LIBADD = ${top_builddir}/libslax/libslax.la ../libext_db.la
+libext_db_sqlite_la_LDFLAGS = -no-undefined -module -avoid-version
 
 AM_LDFLAGS = \
     -Wl, --whole-archive, \
@@ -88,7 +89,7 @@ install-exec-hook:
 		if [ x"$$DLNAME" = x ]; \
                     then DLNAME=${LIBNAME}.${SLAX_LIBEXT}; fi ; \
 		if [ "$(build_os)" = "cygwin" ]; \
-		    then DLNAME="../bin/$$DLNAME"; fi ; \
+		    then DLNAME="$$DLNAME"; fi ; \
 		echo Install link $$DLNAME "->" ${UGLY_NAME} "..." ; \
 		mkdir -p ${DESTDIR}${SLAX_EXTDIR}/db ; \
 		cd ${DESTDIR}${SLAX_EXTDIR}/db \
--- origsrc/libslax-0.20.2/extensions/exslt/Makefile.am	2016-08-11 03:51:46.000000000 +0900
+++ src/libslax-0.20.2/extensions/exslt/Makefile.am	2016-12-30 23:40:01.468832800 +0900
@@ -38,7 +38,8 @@ LIBS = \
     ${LIBXML_LIBS} \
     -L${top_builddir}/libslax -lslax
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_exslt_la_LIBADD = ${top_builddir}/libslax/libslax.la
+libext_exslt_la_LDFLAGS = -no-undefined -module -avoid-version
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
@@ -68,7 +69,7 @@ install-exec-hook:
 		if [ x"$$DLNAME" = x ]; \
                     then DLNAME=${LIBNAME}.${SLAX_LIBEXT}; fi ; \
 		if [ "$(build_os)" = "cygwin" ]; \
-		    then DLNAME="../bin/$$DLNAME"; fi ; \
+		    then DLNAME="$$DLNAME"; fi ; \
 		echo Install link $$DLNAME "->" ${UGLY_NAME} "..." ; \
 		mkdir -p ${DESTDIR}${SLAX_EXTDIR} ; \
 		cd ${DESTDIR}${SLAX_EXTDIR} \
--- origsrc/libslax-0.20.2/extensions/os/Makefile.am	2016-08-11 03:51:46.000000000 +0900
+++ src/libslax-0.20.2/extensions/os/Makefile.am	2016-12-30 23:40:10.045630000 +0900
@@ -38,7 +38,8 @@ LIBS = \
     ${LIBXML_LIBS} \
     -L${top_builddir}/libslax -lslax
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_os_la_LIBADD = ${top_builddir}/libslax/libslax.la
+libext_os_la_LDFLAGS = -no-undefined -module -avoid-version
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
@@ -62,7 +63,7 @@ install-exec-hook:
 		if [ x"$$DLNAME" = x ]; \
                     then DLNAME=${LIBNAME}.${SLAX_LIBEXT}; fi ; \
 		if [ "$(build_os)" = "cygwin" ]; \
-		    then DLNAME="../bin/$$DLNAME"; fi ; \
+		    then DLNAME="$$DLNAME"; fi ; \
 		echo Install link $$DLNAME "->" ${UGLY_NAME} "..." ; \
 		mkdir -p ${DESTDIR}${SLAX_EXTDIR} ; \
 		cd ${DESTDIR}${SLAX_EXTDIR} \
--- origsrc/libslax-0.20.2/extensions/os/ext_os.c	2016-08-11 03:51:46.000000000 +0900
+++ src/libslax-0.20.2/extensions/os/ext_os.c	2016-12-30 22:47:55.406419200 +0900
@@ -39,6 +39,11 @@
 #include <libslax/slaxutil.h>
 #include <libslax/slaxnames.h>
 
+#if defined(__CYGWIN__)
+#define GLOB_BRACE      0x0080  /* Expand braces ala csh. */
+#define GLOB_TILDE      0x0800  /* Expand tilde names from the passwd file. */
+#endif
+
 #define XML_FULL_NS "http://xml.libslax.org/os"
 
 typedef struct statOptions_s {
--- origsrc/libslax-0.20.2/extensions/xutil/Makefile.am	2016-08-11 03:51:46.000000000 +0900
+++ src/libslax-0.20.2/extensions/xutil/Makefile.am	2016-12-30 23:40:19.002596700 +0900
@@ -39,7 +39,8 @@ LIBS = \
     ${LIBXML_LIBS} \
     -L${top_builddir}/libslax -lslax
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_xutil_la_LIBADD = ${top_builddir}/libslax/libslax.la
+libext_xutil_la_LDFLAGS = -no-undefined -module -avoid-version
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
@@ -60,7 +61,7 @@ install-exec-hook:
 		if [ x"$$DLNAME" = x ]; \
                     then DLNAME=${LIBNAME}.${SLAX_LIBEXT}; fi ; \
 		if [ "$(build_os)" = "cygwin" ]; \
-		    then DLNAME="../bin/$$DLNAME"; fi ; \
+		    then DLNAME="$$DLNAME"; fi ; \
 		echo Install link $$DLNAME "->" ${UGLY_NAME} "..." ; \
 		mkdir -p ${DESTDIR}${SLAX_EXTDIR} ; \
  		cd ${DESTDIR}${SLAX_EXTDIR} \
--- origsrc/libslax-0.20.2/libslax/Makefile.am	2016-09-27 11:38:54.000000000 +0900
+++ src/libslax-0.20.2/libslax/Makefile.am	2016-12-30 22:47:55.422064600 +0900
@@ -35,6 +35,7 @@ AM_CFLAGS += \
 
 lib_LTLIBRARIES = libslax.la
 libslax_la_LIBADD = ${LIBXSLT_LIBS} -lexslt ${LIBXML_LIBS}
+libslax_la_LDFLAGS = -no-undefined
 
 slaxinc_HEADERS = \
      slax.h \
