--- origsrc/libslax-0.16.15/extensions/bit/Makefile.am	2013-07-16 15:08:07.000000000 +0900
+++ src/libslax-0.16.15/extensions/bit/Makefile.am	2013-07-20 15:09:16.153837500 +0900
@@ -34,7 +34,7 @@ LIBS = \
     -lexslt \
     ${LIBXML_LIBS}
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_bit_la_LIBADD = ${top_builddir}/libslax/libslax.la
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
@@ -45,16 +45,17 @@ LIBS += -ledit
 endif
 libext_bit_la_SOURCES = \
     ext_bit.c
+libext_bit_la_LDFLAGS = -no-undefined -module -avoid-version
 
 pkglibdir = ${SLAX_EXTDIR}
 
 UGLY_NAME = http%3A%2F%2Fxml.libslax.org%2Fbit.ext
 
 install-exec-hook:
-	@DLNAME=`sh -c '. ${pkglibdir}/libext_bit.la ; echo $$dlname'` ; \
+	@DLNAME=`sh -c '. ./libext_bit.la ; echo $$dlname'` ; \
 		if [ x"$$DLNAME" = x ]; then echo error: emtpy dlname; fi ; \
 		echo Install link $$DLNAME "->" ${UGLY_NAME} "..." ; \
-		cd ${SLAX_EXTDIR} \
+		cd ${DESTDIR}${SLAX_EXTDIR} \
 		&& chmod +w . \
 		&& rm -f ${UGLY_NAME} \
 		&& ${LN_S} $$DLNAME ${UGLY_NAME}
--- origsrc/libslax-0.16.15/extensions/curl/Makefile.am	2013-07-16 15:08:07.000000000 +0900
+++ src/libslax-0.16.15/extensions/curl/Makefile.am	2013-07-20 15:10:22.677660100 +0900
@@ -38,7 +38,7 @@ LIBS = \
     ${LIBXML_LIBS} \
     ${LIBCURL_LIBS}
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_curl_la_LIBADD = ${top_builddir}/libslax/libslax.la
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
@@ -50,6 +50,7 @@ endif
 
 libext_curl_la_SOURCES = \
     ext_curl.c
+libext_curl_la_LDFLAGS = -no-undefined -module -avoid-version
 
 noinst_HEADERS = \
     ext_curl.h
@@ -59,10 +60,10 @@ pkglibdir = ${SLAX_EXTDIR}
 UGLY_NAME = http%3A%2F%2Fxml.libslax.org%2Fcurl.ext
 
 install-exec-hook:
-	@DLNAME=`sh -c '. ${pkglibdir}/libext_curl.la ; echo $$dlname'`; \
+	@DLNAME=`sh -c '. ./libext_curl.la ; echo $$dlname'`; \
 		if [ x"$$DLNAME" = x ]; then echo error: emtpy dlname; fi ; \
 		echo Install link $$DLNAME "->" ${UGLY_NAME} "..." ; \
-		cd ${SLAX_EXTDIR} \
+		cd ${DESTDIR}${SLAX_EXTDIR} \
 		&& chmod +w . \
 	        && rm -f ${UGLY_NAME} \
 		&& ${LN_S} $$DLNAME ${UGLY_NAME}
--- origsrc/libslax-0.16.15/extensions/os/Makefile.am	2013-07-16 15:08:07.000000000 +0900
+++ src/libslax-0.16.15/extensions/os/Makefile.am	2013-07-20 15:13:44.342006700 +0900
@@ -30,7 +30,7 @@ pkglib_LTLIBRARIES = libext_os.la
 LIBS = \
     ${LIBXSLT_LIBS} -lexslt ${LIBXML_LIBS}
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_os_la_LIBADD = ${top_builddir}/libslax/libslax.la
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
@@ -41,6 +41,7 @@ LIBS += -ledit
 endif
 libext_os_la_SOURCES = \
     ext_os.c
+libext_os_la_LDFLAGS = -no-undefined -module -avoid-version
 
 slaxextdir = ${SLAX_EXTDIR}
 pkglibdir = ${slaxextdir}
@@ -50,10 +51,10 @@ ext_oslibdir = ${slaxextdir}
 UGLY_NAME = http%3A%2F%2Fxml.libslax.org%2Fos.ext
 
 install-exec-hook:
-	@DLNAME=`sh -c '. ${pkglibdir}/libext_os.la ; echo $$dlname'` ; \
+	@DLNAME=`sh -c '. ./libext_os.la ; echo $$dlname'` ; \
 		if [ x"$$DLNAME" = x ]; then echo error: emtpy dlname; fi ; \
 		echo Install link $$DLNAME "->" ${UGLY_NAME} "..." ; \
-		cd ${slaxextdir} \
+		cd ${DESTDIR}${slaxextdir} \
 		&& chmod +w . \
 		&& rm -f ${UGLY_NAME} \
 		&& ${LN_S} $$DLNAME ${UGLY_NAME}
--- origsrc/libslax-0.16.15/extensions/xutil/Makefile.am	2013-07-16 15:08:07.000000000 +0900
+++ src/libslax-0.16.15/extensions/xutil/Makefile.am	2013-07-20 15:11:39.754116100 +0900
@@ -35,7 +35,7 @@ LIBS = \
     -lexslt \
     ${LIBXML_LIBS}
 
-LDADD = ${top_builddir}/libslax/libslax.la
+libext_xutil_la_LIBADD = ${top_builddir}/libslax/libslax.la
 
 if HAVE_READLINE
 LIBS += -L/opt/local/lib -lreadline
@@ -58,6 +58,7 @@ libext_xutil_la_SOURCES = \
     jsonparser.c \
     jsonlexer.c \
     jsonwriter.c
+libext_xutil_la_LDFLAGS = -no-undefined -module -avoid-version
 
 pkglibdir = ${SLAX_EXTDIR}
 
@@ -84,10 +85,10 @@ CLEANFILES = jsonparser.h jsonparser.c j
 UGLY_NAME = http%3A%2F%2Fxml.libslax.org%2Fxutil.ext
 
 install-exec-hook:
-	@DLNAME=`sh -c '. ${pkglibdir}/libext_xutil.la ; echo $$dlname'`; \
+	@DLNAME=`sh -c '. ./libext_xutil.la ; echo $$dlname'`; \
 		if [ x"$$DLNAME" = x ]; then echo error: emtpy dlname; fi ; \
 		echo Install link $$DLNAME "->" ${UGLY_NAME} "..." ; \
- 		cd ${SLAX_EXTDIR} \
+ 		cd ${DESTDIR}${SLAX_EXTDIR} \
 		&& chmod +w . \
 		&& rm -f ${UGLY_NAME} \
 		&& ${LN_S} $$DLNAME ${UGLY_NAME}
--- origsrc/libslax-0.16.15/libslax/Makefile.am	2013-07-14 13:37:47.000000000 +0900
+++ src/libslax-0.16.15/libslax/Makefile.am	2013-07-20 14:58:10.788609400 +0900
@@ -28,6 +28,7 @@ AM_CFLAGS += \
 
 lib_LTLIBRARIES = libslax.la
 libslax_la_LIBADD = ${LIBXSLT_LIBS} -lexslt ${LIBXML_LIBS}
+libslax_la_LDFLAGS = -no-undefined
 
 slaxinc_HEADERS = \
      slax.h \
