--- origsrc/gwyddion-2.50/Makefile.am	2017-12-24 18:09:57.000000000 +0900
+++ src/gwyddion-2.50/Makefile.am	2018-04-12 17:04:10.856000000 +0900
@@ -17,8 +17,7 @@ SUBDIRS = \
 	ruby \
 	data \
 	utils \
-	thumbnailer \
-	devel-docs
+	thumbnailer
 
 EXTRA_DIST = \
 	autogen.sh \
--- origsrc/gwyddion-2.50/app/Makefile.am	2018-02-01 18:36:00.000000000 +0900
+++ src/gwyddion-2.50/app/Makefile.am	2018-04-12 13:49:57.402000000 +0900
@@ -65,11 +65,12 @@ BUILT_SOURCES = \
 	release.h
 
 if OS_MSWIN
+win32_libs = -lole32
+endif
+
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwyapp2.def
-win32_libs = -lole32
 libgwyapp2_la_DEPENDENCIES = libgwyapp2.def
-endif
 
 # Must init
 EXTRA_DIST = \
@@ -173,8 +174,8 @@ gwyddion_LDADD = @COMMON_LDFLAGS@ @GTK_M
 	$(libgwyddion)
 
 if MAINTAINER_MODE
-authors.h: ${top_srcdir}/AUTHORS $(top_srcdir)/utils/gen-authors.py
-	$(AM_V_GEN)$(PYTHON) $(top_srcdir)/utils/gen-authors.py $(top_srcdir)/AUTHORS header >authors.h
+#authors.h: ${top_srcdir}/AUTHORS $(top_srcdir)/utils/gen-authors.py
+#	$(AM_V_GEN)$(PYTHON) $(top_srcdir)/utils/gen-authors.py $(top_srcdir)/AUTHORS header >authors.h
 
 release.h: ${top_srcdir}/NEWS Makefile $(top_builddir)/config.h
 	$(AM_V_GEN)echo '/* This is a 'GENERATED' file. */' >release.h
--- origsrc/gwyddion-2.50/data/Makefile.am	2017-12-24 18:09:57.000000000 +0900
+++ src/gwyddion-2.50/data/Makefile.am	2018-04-12 13:18:07.451000000 +0900
@@ -105,7 +105,7 @@ endif
 # unexpanded ${prefix} in the schemas file.
 # Depend on Makefile, this may run this rule perhaps too often but ensures
 # the paths are updated every time they should.
-gwyddion-thumbnailer.schemas: gwyddion-thumbnailer.schemas.in Makefile
+gwyddion-thumbnailer.schemas: $(srcdir)/gwyddion-thumbnailer.schemas.in Makefile
 	$(AM_V_GEN)$(SED) -e 's/^\(<gconfschemafile\) .*\(>\)/\1\2/' \
 	  -e 's#[@]bindir[@]#$(bindir)#' \
 	   $(srcdir)/gwyddion-thumbnailer.schemas.in \
@@ -117,18 +117,18 @@ gwyddion-thumbnailer.schemas: gwyddion-t
 # unexpanded ${prefix} in the schemas file.
 # Depend on Makefile, this may run this rule perhaps too often but ensures
 # the paths are updated every time they should.
-gwyddion.desktop: gwyddion.desktop.in $(srcdir)/gwyddion.xml Makefile
+gwyddion.desktop: $(srcdir)/gwyddion.desktop.in $(srcdir)/gwyddion.xml Makefile
 	$(AM_V_GEN)x=`$(SED) -e '/^<mime-type/ s/.*<mime-type type="\(.*\)">.*/\1;/' -e t -e d <$(srcdir)/gwyddion.xml | tr -d '\n'` \
 	&& $(SED) -e "s#@MIME_TYPE_LIST@#$$x#" \
 	  -e 's#[@]datadir[@]#$(datadir)#' \
-	  <gwyddion.desktop.in >gwyddion.desktop
+	  <$(srcdir)/gwyddion.desktop.in >gwyddion.desktop
 
 # Generate gwyddion.thumbnailer from gwyddion.xml to keep the file format list
 # in sync.
-gwyddion.thumbnailer: gwyddion.thumbnailer.in $(srcdir)/gwyddion.xml
+gwyddion.thumbnailer: $(srcdir)/gwyddion.thumbnailer.in $(srcdir)/gwyddion.xml
 	$(AM_V_GEN)x=`$(SED) -e '/^<mime-type/ s/.*<mime-type type="\(.*\)">.*/\1;/' -e t -e d <$(srcdir)/gwyddion.xml | tr -d '\n'` \
 	&& $(SED) -e "s#@MIME_TYPE_LIST@#$$x#" \
-	  <gwyddion.thumbnailer.in >gwyddion.thumbnailer
+	  <$(srcdir)/gwyddion.thumbnailer.in >gwyddion.thumbnailer
 
 # Desktop file installation
 # FIXME: Maybe not according to current standards
--- origsrc/gwyddion-2.50/libdraw/Makefile.am	2017-12-24 18:09:57.000000000 +0900
+++ src/gwyddion-2.50/libdraw/Makefile.am	2018-04-12 13:18:27.054000000 +0900
@@ -23,11 +23,9 @@ libgwydraw2include_HEADERS = \
 
 lib_LTLIBRARIES = libgwydraw2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwydraw2.def
 libgwydraw2_la_DEPENDENCIES = libgwydraw2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.50/libgwyddion/Makefile.am	2018-02-01 18:36:00.000000000 +0900
+++ src/gwyddion-2.50/libgwyddion/Makefile.am	2018-04-12 13:18:45.400000000 +0900
@@ -36,11 +36,9 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwyddion2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwyddion2.def
 libgwyddion2_la_DEPENDENCIES = libgwyddion2.def
-endif
 
 # Must init
 EXTRA_DIST = gwyversion.h.in
--- origsrc/gwyddion-2.50/libgwydgets/Makefile.am	2018-02-01 18:36:00.000000000 +0900
+++ src/gwyddion-2.50/libgwydgets/Makefile.am	2018-04-12 13:19:00.720000000 +0900
@@ -76,11 +76,9 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwydgets2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwydgets2.def
 libgwydgets2_la_DEPENDENCIES = libgwydgets2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.50/libgwymodule/Makefile.am	2017-12-24 18:09:59.000000000 +0900
+++ src/gwyddion-2.50/libgwymodule/Makefile.am	2018-04-12 13:19:15.851000000 +0900
@@ -34,11 +34,9 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwymodule2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwymodule2.def
 libgwymodule2_la_DEPENDENCIES = libgwymodule2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.50/libprocess/Makefile.am	2018-02-01 18:36:00.000000000 +0900
+++ src/gwyddion-2.50/libprocess/Makefile.am	2018-04-12 13:19:33.805000000 +0900
@@ -52,11 +52,9 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwyprocess2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwyprocess2.def
 libgwyprocess2_la_DEPENDENCIES = libgwyprocess2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.50/modules/Makefile.am	2017-12-24 18:09:48.000000000 +0900
+++ src/gwyddion-2.50/modules/Makefile.am	2018-04-12 13:19:59.628000000 +0900
@@ -19,9 +19,7 @@ module_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(plugin_proxy_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.50/modules/file/Makefile.am	2018-01-23 19:25:31.000000000 +0900
+++ src/gwyddion-2.50/modules/file/Makefile.am	2018-04-12 13:20:33.727000000 +0900
@@ -284,9 +284,7 @@ EXTRA_DIST = \
 	$(disabled_sources) \
 	1CHECKLIST
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.50/modules/graph/Makefile.am	2017-12-24 18:09:48.000000000 +0900
+++ src/gwyddion-2.50/modules/graph/Makefile.am	2018-04-12 13:20:46.882000000 +0900
@@ -33,9 +33,7 @@ endif
 graph_LTLIBRARIES = \
 	$(graph_modules)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.50/modules/layer/Makefile.am	2017-12-24 18:09:49.000000000 +0900
+++ src/gwyddion-2.50/modules/layer/Makefile.am	2018-04-12 13:20:58.015000200 +0900
@@ -29,9 +29,7 @@ layer_LTLIBRARIES = \
 noinst_HEADERS = \
 	layer.h
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.50/modules/process/Makefile.am	2018-01-12 23:38:56.000000000 +0900
+++ src/gwyddion-2.50/modules/process/Makefile.am	2018-04-12 13:21:16.749000000 +0900
@@ -148,9 +148,7 @@ process_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.50/modules/pygwy/Makefile.am	2017-12-24 18:09:52.000000000 +0900
+++ src/gwyddion-2.50/modules/pygwy/Makefile.am	2018-04-12 13:23:31.363000000 +0900
@@ -149,8 +149,8 @@ PYGWY_H_FILES = \
 	$(top_srcdir)/libgwydgets/gwystock.h \
 	$(top_srcdir)/libgwydgets/gwyvectorlayer.h \
 	$(top_srcdir)/libgwydgets/gwyvruler.h \
-	wrap_calls.h \
-	macro-prototypes.h
+	$(srcdir)/wrap_calls.h \
+	$(srcdir)/macro-prototypes.h
 
 disabled_sources =
 
@@ -197,9 +197,7 @@ EXTRA_DIST = \
 	pygwy-fix-defs-2.py \
 	gwyutils.py
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
@@ -226,6 +224,8 @@ pygwy_la_LIBADD = $(module_libadd) $(PYT
 pygwy.defs: $(PYGWY_H_FILES) Makefile.am pygwy-fix-defs-1.py pygwy-fix-defs-2.py extra.defs
 	$(AM_V_GEN)$(PYTHON) $(PYGTK_CODEGENDIR)/h2def.py $(PYGWY_H_FILES) \
 		| $(PYTHON) $(srcdir)/pygwy-fix-defs-1.py >pygwy.defs.tmp
+	ln -fs $(srcdir)/extra.defs .
+	ln -fs $(srcdir)/override.defs .
 	$(PYTHON) $(srcdir)/pygwy-fix-defs-2.py --codegendir=$(PYGTK_CODEGENDIR) >pygwy.defs
 	rm pygwy.defs.tmp
 
--- origsrc/gwyddion-2.50/modules/tools/Makefile.am	2017-12-24 18:09:52.000000000 +0900
+++ src/gwyddion-2.50/modules/tools/Makefile.am	2018-04-12 13:21:47.560000000 +0900
@@ -37,9 +37,7 @@ endif
 tools_LTLIBRARIES = \
 	$(tools_modules)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.50/modules/volume/Makefile.am	2017-12-24 18:09:53.000000000 +0900
+++ src/gwyddion-2.50/modules/volume/Makefile.am	2018-04-12 13:21:58.304000000 +0900
@@ -38,9 +38,7 @@ volume_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.50/modules/xyz/Makefile.am	2017-12-24 18:09:53.000000000 +0900
+++ src/gwyddion-2.50/modules/xyz/Makefile.am	2018-04-12 13:22:08.413000200 +0900
@@ -30,9 +30,7 @@ xyz_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.50/pixmaps/Makefile.am	2018-01-24 23:23:24.000000000 +0900
+++ src/gwyddion-2.50/pixmaps/Makefile.am	2018-04-12 16:51:38.603000000 +0900
@@ -915,10 +915,10 @@ gwy_mask_remove-24.png: $(srcdir)/src/gw
 	$(PNGCRUSH) -q -brute "gwy_mask_remove-24.png" "gwy_mask_remove-24.crush.png"
 	if test -s "gwy_mask_remove-24.crush.png"; then mv -f "gwy_mask_remove-24.crush.png" "gwy_mask_remove-24.png"; fi
 
-gwy_mask_set-24.png: $(srcdir)/src/gwy_mask_set-24.svg
-	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_mask_set-24.png" --file="$(srcdir)/src/gwy_mask_set-24.svg"
-	$(PNGCRUSH) -q -brute "gwy_mask_set-24.png" "gwy_mask_set-24.crush.png"
-	if test -s "gwy_mask_set-24.crush.png"; then mv -f "gwy_mask_set-24.crush.png" "gwy_mask_set-24.png"; fi
+#gwy_mask_set-24.png: $(srcdir)/src/gwy_mask_set-24.svg
+#	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_mask_set-24.png" --file="$(srcdir)/src/gwy_mask_set-24.svg"
+#	$(PNGCRUSH) -q -brute "gwy_mask_set-24.png" "gwy_mask_set-24.crush.png"
+#	if test -s "gwy_mask_set-24.crush.png"; then mv -f "gwy_mask_set-24.crush.png" "gwy_mask_set-24.png"; fi
 
 gwy_mask_shrink-24.png: $(srcdir)/src/gwy_mask_shrink-24.svg
 	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_mask_shrink-24.png" --file="$(srcdir)/src/gwy_mask_shrink-24.svg"
@@ -1345,15 +1345,15 @@ gwy_volume_slice-24.png: $(srcdir)/src/g
 	$(PNGCRUSH) -q -brute "gwy_volume_slice-24.png" "gwy_volume_slice-24.crush.png"
 	if test -s "gwy_volume_slice-24.crush.png"; then mv -f "gwy_volume_slice-24.crush.png" "gwy_volume_slice-24.png"; fi
 
-gwy_volumize-24.png: $(srcdir)/src/gwy_volumize-24.svg
-	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_volumize-24.png" --file="$(srcdir)/src/gwy_volumize-24.svg"
-	$(PNGCRUSH) -q -brute "gwy_volumize-24.png" "gwy_volumize-24.crush.png"
-	if test -s "gwy_volumize-24.crush.png"; then mv -f "gwy_volumize-24.crush.png" "gwy_volumize-24.png"; fi
-
-gwy_volumize_layers-24.png: $(srcdir)/src/gwy_volumize_layers-24.svg
-	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_volumize_layers-24.png" --file="$(srcdir)/src/gwy_volumize_layers-24.svg"
-	$(PNGCRUSH) -q -brute "gwy_volumize_layers-24.png" "gwy_volumize_layers-24.crush.png"
-	if test -s "gwy_volumize_layers-24.crush.png"; then mv -f "gwy_volumize_layers-24.crush.png" "gwy_volumize_layers-24.png"; fi
+#gwy_volumize-24.png: $(srcdir)/src/gwy_volumize-24.svg
+#	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_volumize-24.png" --file="$(srcdir)/src/gwy_volumize-24.svg"
+#	$(PNGCRUSH) -q -brute "gwy_volumize-24.png" "gwy_volumize-24.crush.png"
+#	if test -s "gwy_volumize-24.crush.png"; then mv -f "gwy_volumize-24.crush.png" "gwy_volumize-24.png"; fi
+
+#gwy_volumize_layers-24.png: $(srcdir)/src/gwy_volumize_layers-24.svg
+#	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_volumize_layers-24.png" --file="$(srcdir)/src/gwy_volumize_layers-24.svg"
+#	$(PNGCRUSH) -q -brute "gwy_volumize_layers-24.png" "gwy_volumize_layers-24.crush.png"
+#	if test -s "gwy_volumize_layers-24.crush.png"; then mv -f "gwy_volumize_layers-24.crush.png" "gwy_volumize_layers-24.png"; fi
 
 gwy_xy_denoise-24.png: $(srcdir)/src/gwy_xy_denoise-24.svg
 	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_xy_denoise-24.png" --file="$(srcdir)/src/gwy_xy_denoise-24.svg"
