--- origsrc/gwyddion-2.46/app/Makefile.am	2016-10-13 19:35:04.000000000 +0900
+++ src/gwyddion-2.46/app/Makefile.am	2016-11-06 22:11:23.859679500 +0900
@@ -61,12 +61,12 @@ lib_LTLIBRARIES = libgwyapp2.la
 BUILT_SOURCES = \
 	authors.h
 
-if OS_MSWIN
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwyapp2.def
+if OS_MSWIN
 win32_libs = -lole32
-libgwyapp2_la_DEPENDENCIES = libgwyapp2.def
 endif
+libgwyapp2_la_DEPENDENCIES = libgwyapp2.def
 
 # Must init
 EXTRA_DIST = \
--- origsrc/gwyddion-2.46/data/Makefile.am	2016-01-12 20:55:14.000000000 +0900
+++ src/gwyddion-2.46/data/Makefile.am	2016-11-06 22:12:37.266940800 +0900
@@ -117,18 +117,18 @@ gwyddion-thumbnailer.schemas: gwyddion-t
 # unexpanded ${prefix} in the schemas file.
 # Depend on Makefile, this may run this rule perhaps too often but ensures
 # the paths are updated every time they should.
-gwyddion.desktop: gwyddion.desktop.in $(srcdir)/gwyddion.xml Makefile
+gwyddion.desktop: $(srcdir)/gwyddion.desktop.in $(srcdir)/gwyddion.xml Makefile
 	$(AM_V_GEN)x=`$(SED) -e '/^<mime-type/ s/.*<mime-type type="\(.*\)">.*/\1;/' -e t -e d <$(srcdir)/gwyddion.xml | tr -d '\n'` \
 	&& $(SED) -e "s#@MIME_TYPE_LIST@#$$x#" \
 	  -e 's#[@]datadir[@]#$(datadir)#' \
-	  <gwyddion.desktop.in >gwyddion.desktop
+	  <$(srcdir)/gwyddion.desktop.in >gwyddion.desktop
 
 # Generate gwyddion.thumbnailer from gwyddion.xml to keep the file format list
 # in sync.
-gwyddion.thumbnailer: gwyddion.thumbnailer.in $(srcdir)/gwyddion.xml
+gwyddion.thumbnailer: $(srcdir)/gwyddion.thumbnailer.in $(srcdir)/gwyddion.xml
 	$(AM_V_GEN)x=`$(SED) -e '/^<mime-type/ s/.*<mime-type type="\(.*\)">.*/\1;/' -e t -e d <$(srcdir)/gwyddion.xml | tr -d '\n'` \
 	&& $(SED) -e "s#@MIME_TYPE_LIST@#$$x#" \
-	  <gwyddion.thumbnailer.in >gwyddion.thumbnailer
+	  <$(srcdir)/gwyddion.thumbnailer.in >gwyddion.thumbnailer
 
 # Desktop file installation
 # FIXME: Maybe not according to current standards
--- origsrc/gwyddion-2.46/libdraw/Makefile.am	2016-10-13 19:35:04.000000000 +0900
+++ src/gwyddion-2.46/libdraw/Makefile.am	2016-11-06 22:13:31.860089100 +0900
@@ -22,12 +22,9 @@ libgwydraw2include_HEADERS = \
 
 lib_LTLIBRARIES = libgwydraw2.la
 
-
-if OS_MSWIN
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwydraw2.def
 libgwydraw2_la_DEPENDENCIES = libgwydraw2.def
-endif
 
 noinst_HEADERS = \
 	gwydrawinternal.h
@@ -65,4 +62,3 @@ CLEANFILES = libgwydraw2.def
 
 clean-local:
 	rm -f core.* *~
-
--- origsrc/gwyddion-2.46/libgwyddion/Makefile.am	2016-10-13 19:35:04.000000000 +0900
+++ src/gwyddion-2.46/libgwyddion/Makefile.am	2016-11-06 22:13:54.397424300 +0900
@@ -34,11 +34,9 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwyddion2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwyddion2.def
 libgwyddion2_la_DEPENDENCIES = libgwyddion2.def
-endif
 
 # Must init
 EXTRA_DIST = gwyversion.h.in
@@ -104,4 +102,3 @@ libgwyddion2_la_SOURCES = \
 
 clean-local:
 	rm -f core.* *~
-
--- origsrc/gwyddion-2.46/libgwydgets/Makefile.am	2016-10-13 19:35:04.000000000 +0900
+++ src/gwyddion-2.46/libgwydgets/Makefile.am	2016-11-06 22:14:20.218851100 +0900
@@ -75,11 +75,9 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwydgets2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwydgets2.def
 libgwydgets2_la_DEPENDENCIES = libgwydgets2.def
-endif
 
 # Must init
 EXTRA_DIST =
@@ -179,4 +177,3 @@ libgwydgets2_la_SOURCES = \
 
 clean-local:
 	rm -f core.* *~
-
--- origsrc/gwyddion-2.46/libgwymodule/Makefile.am	2016-04-25 16:03:41.000000000 +0900
+++ src/gwyddion-2.46/libgwymodule/Makefile.am	2016-11-06 22:14:32.943040500 +0900
@@ -34,11 +34,9 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwymodule2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwymodule2.def
 libgwymodule2_la_DEPENDENCIES = libgwymodule2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.46/libprocess/Makefile.am	2016-10-13 19:35:04.000000000 +0900
+++ src/gwyddion-2.46/libprocess/Makefile.am	2016-11-06 22:14:47.816403300 +0900
@@ -53,11 +53,9 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwyprocess2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwyprocess2.def
 libgwyprocess2_la_DEPENDENCIES = libgwyprocess2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.46/modules/Makefile.am	2016-02-24 22:40:54.000000000 +0900
+++ src/gwyddion-2.46/modules/Makefile.am	2016-11-06 22:15:12.125457200 +0900
@@ -1,6 +1,6 @@
 # @(#) $Id: Makefile.am 18349 2016-02-24 13:40:54Z yeti-dn $
 
-SUBDIRS = file graph layer process tools volume xyz pygwy
+SUBDIRS = file graph layer process tools volume xyz # pygwy
 
 moduledir = $(pkglibdir)/modules
 
@@ -18,9 +18,7 @@ module_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(plugin_proxy_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.46/modules/file/Makefile.am	2016-07-28 19:44:50.000000000 +0900
+++ src/gwyddion-2.46/modules/file/Makefile.am	2016-11-06 22:15:57.418805900 +0900
@@ -264,9 +264,7 @@ EXTRA_DIST = \
 	$(disabled_sources) \
 	1CHECKLIST
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.46/modules/graph/Makefile.am	2016-02-22 15:57:59.000000000 +0900
+++ src/gwyddion-2.46/modules/graph/Makefile.am	2016-11-06 22:16:42.657329900 +0900
@@ -16,9 +16,7 @@ graph_LTLIBRARIES = \
 	graph_level.la \
 	graph_peaks.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.46/modules/layer/Makefile.am	2016-01-20 00:51:07.000000000 +0900
+++ src/gwyddion-2.46/modules/layer/Makefile.am	2016-11-06 22:16:55.610552600 +0900
@@ -14,9 +14,7 @@ layer_LTLIBRARIES = \
 noinst_HEADERS = \
 	layer.h
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.46/modules/process/Makefile.am	2016-08-22 16:57:46.000000000 +0900
+++ src/gwyddion-2.46/modules/process/Makefile.am	2016-11-06 22:17:17.713295700 +0900
@@ -118,9 +118,7 @@ process_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
@@ -245,4 +243,3 @@ xyzize_la_SOURCES              = xyzize.
 
 clean-local:
 	rm -f core.* *~
-
--- origsrc/gwyddion-2.46/modules/pygwy/Makefile.am	2016-04-25 16:42:57.000000000 +0900
+++ src/gwyddion-2.46/modules/pygwy/Makefile.am	2016-11-06 22:18:08.411341500 +0900
@@ -189,9 +189,7 @@ EXTRA_DIST = \
 	pygwy.h \
 	gwyutils.py
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
@@ -217,7 +215,7 @@ pygwy_la_LIBADD = $(module_libadd) $(PYT
 
 pygwy.defs: $(PYGWY_H_FILES) Makefile.am pygwy-fix-defs.py pygwy-fix-defs.sed
 	$(AM_V_GEN)$(PYTHON) $(PYGTK_CODEGENDIR)/h2def.py $(PYGWY_H_FILES) \
-	| sed -f pygwy-fix-defs.sed >pygwy.defs.tmp
+	| sed -f $(srcdir)/pygwy-fix-defs.sed >pygwy.defs.tmp
 	$(PYTHON) $(srcdir)/pygwy-fix-defs.py $(PYGTK_CODEGENDIR) >pygwy.defs
 	rm pygwy.defs.tmp
 
--- origsrc/gwyddion-2.46/modules/tools/Makefile.am	2015-08-05 18:03:41.000000000 +0900
+++ src/gwyddion-2.46/modules/tools/Makefile.am	2016-11-06 22:21:03.526352200 +0900
@@ -22,9 +22,7 @@ tools_LTLIBRARIES = \
 	spectro.la \
 	stats.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.46/modules/volume/Makefile.am	2015-09-21 16:30:24.000000000 +0900
+++ src/gwyddion-2.46/modules/volume/Makefile.am	2016-11-06 22:21:16.022522900 +0900
@@ -21,9 +21,7 @@ volume_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
@@ -52,4 +50,3 @@ volume_zcal_la_SOURCES      = volume_zca
 
 clean-local:
 	rm -f core.* *~
-
--- origsrc/gwyddion-2.46/modules/xyz/Makefile.am	2016-09-01 17:37:22.000000000 +0900
+++ src/gwyddion-2.46/modules/xyz/Makefile.am	2016-11-06 22:21:31.143732600 +0900
@@ -15,9 +15,7 @@ xyz_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
@@ -40,4 +38,3 @@ xyz_raster_la_SOURCES = xyz_raster.c
 
 clean-local:
 	rm -f core.* *~
-
