--- origsrc/gwyddion-2.67/Makefile.am	2022-10-18 00:16:35.000000000 +0900
+++ src/gwyddion-2.67/Makefile.am	2024-12-09 23:35:29.313138300 +0900
@@ -18,8 +18,11 @@ SUBDIRS = \
 	ruby \
 	data \
 	utils \
-	thumbnailer \
-	devel-docs
+	thumbnailer
+
+if ENABLE_GTK_DOC
+SUBDIRS += devel-docs
+endif
 
 EXTRA_DIST = \
 	autogen.sh \
--- origsrc/gwyddion-2.67/app/Makefile.am	2024-11-06 23:13:15.000000000 +0900
+++ src/gwyddion-2.67/app/Makefile.am	2024-12-09 14:05:40.613741100 +0900
@@ -58,12 +58,11 @@ noinst_HEADERS = \
 lib_LTLIBRARIES = libgwyapp2.la
 
 if OS_MSWIN
-no_undefined = -no-undefined
-export_symbols = -export-symbols libgwyapp2.def
 win32_libs = -lole32
-libgwyapp2_la_DEPENDENCIES = libgwyapp2.def
 endif
 
+no_undefined = -no-undefined
+
 # Must init
 EXTRA_DIST =
 CLEANFILES = libgwyapp2.def
--- origsrc/gwyddion-2.67/data/Makefile.am	2024-02-22 21:21:13.000000000 +0900
+++ src/gwyddion-2.67/data/Makefile.am	2024-12-09 10:53:01.256690900 +0900
@@ -100,21 +100,21 @@ endif
 # unexpanded ${prefix} in the desktop file.
 # Depend on Makefile, this may run this rule perhaps too often but ensures
 # the paths are updated every time they should.
-gwyddion.desktop: gwyddion.desktop.in $(srcdir)/gwyddion.xml Makefile
+gwyddion.desktop: $(srcdir)/gwyddion.desktop.in $(srcdir)/gwyddion.xml Makefile
 	$(AM_V_GEN)x=`$(SED) -e '/^<mime-type/ s/.*<mime-type type="\(.*\)">.*/\1;/' -e t -e d <$(srcdir)/gwyddion.xml | tr -d '\n'` \
 	&& $(SED) -e "s#@MIME_TYPE_LIST@#$$x#" \
 	  -e 's#[@]datadir[@]#$(datadir)#' \
-	  <gwyddion.desktop.in >gwyddion.desktop
+	  < $(srcdir)/gwyddion.desktop.in >gwyddion.desktop
 
 # Generate gwyddion.thumbnailer from gwyddion.xml to keep the file format list
 # in sync.
-gwyddion.thumbnailer: gwyddion.thumbnailer.in $(srcdir)/gwyddion.xml
+gwyddion.thumbnailer: $(srcdir)/gwyddion.thumbnailer.in $(srcdir)/gwyddion.xml
 	$(AM_V_GEN)x=`$(SED) -e '/^<mime-type/ s/.*<mime-type type="\(.*\)">.*/\1;/' -e t -e d <$(srcdir)/gwyddion.xml | tr -d '\n'` \
 	&& $(SED) \
 	     -e "s#@MIME_TYPE_LIST@#$$x#" \
 	     -e "s#[@]EXEEXT[@]#$(EXEEXT)#" \
 	     -e "s#[@]bindir[@]#$(bindir)#" \
-	     <gwyddion.thumbnailer.in >gwyddion.thumbnailer
+	     < $(srcdir)/gwyddion.thumbnailer.in >gwyddion.thumbnailer
 
 # Desktop file installation
 # FIXME: Maybe not according to current standards
--- origsrc/gwyddion-2.67/libdraw/Makefile.am	2024-05-20 23:11:37.000000000 +0900
+++ src/gwyddion-2.67/libdraw/Makefile.am	2024-12-09 13:11:20.913473200 +0900
@@ -23,11 +23,7 @@ libgwydraw2include_HEADERS = \
 
 lib_LTLIBRARIES = libgwydraw2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
-export_symbols = -export-symbols libgwydraw2.def
-libgwydraw2_la_DEPENDENCIES = libgwydraw2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.67/libgwyddion/Makefile.am	2024-11-06 23:13:15.000000000 +0900
+++ src/gwyddion-2.67/libgwyddion/Makefile.am	2024-12-09 11:27:33.077555200 +0900
@@ -38,11 +38,7 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwyddion2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
-export_symbols = -export-symbols libgwyddion2.def
-libgwyddion2_la_DEPENDENCIES = libgwyddion2.def
-endif
 
 # Must init
 EXTRA_DIST = gwyversion.h.in
--- origsrc/gwyddion-2.67/libgwydgets/Makefile.am	2024-11-08 19:37:31.000000000 +0900
+++ src/gwyddion-2.67/libgwydgets/Makefile.am	2024-12-09 14:00:04.204813900 +0900
@@ -77,11 +77,7 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwydgets2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
-export_symbols = -export-symbols libgwydgets2.def
-libgwydgets2_la_DEPENDENCIES = libgwydgets2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.67/libgwymodule/Makefile.am	2024-01-04 19:23:53.000000000 +0900
+++ src/gwyddion-2.67/libgwymodule/Makefile.am	2024-12-09 14:01:36.078999500 +0900
@@ -35,11 +35,7 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwymodule2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
-export_symbols = -export-symbols libgwymodule2.def
-libgwymodule2_la_DEPENDENCIES = libgwymodule2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.67/libprocess/Makefile.am	2024-11-06 23:13:15.000000000 +0900
+++ src/gwyddion-2.67/libprocess/Makefile.am	2024-12-09 13:04:32.218744100 +0900
@@ -56,11 +56,7 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwyprocess2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
-export_symbols = -export-symbols libgwyprocess2.def
-libgwyprocess2_la_DEPENDENCIES = libgwyprocess2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.67/modules/Makefile.am	2022-10-18 00:16:18.000000000 +0900
+++ src/gwyddion-2.67/modules/Makefile.am	2024-12-09 10:56:29.312774600 +0900
@@ -19,9 +19,7 @@ module_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(plugin_proxy_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.67/modules/cmap/Makefile.am	2024-06-17 20:39:07.000000000 +0900
+++ src/gwyddion-2.67/modules/cmap/Makefile.am	2024-12-09 10:56:47.437807000 +0900
@@ -42,9 +42,7 @@ cmap_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.67/modules/file/Makefile.am	2024-10-18 23:09:46.000000000 +0900
+++ src/gwyddion-2.67/modules/file/Makefile.am	2024-12-09 10:57:10.546884600 +0900
@@ -364,9 +364,7 @@ EXTRA_DIST = \
 	$(disabled_sources) \
 	1CHECKLIST
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.67/modules/file/hdf5/Makefile.am	2024-08-08 20:29:20.000000000 +0900
+++ src/gwyddion-2.67/modules/file/hdf5/Makefile.am	2024-12-09 10:57:38.812462000 +0900
@@ -46,9 +46,7 @@ noinst_HEADERS = \
 
 EXTRA_DIST = $(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -I$(srcdir)/.. -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.67/modules/graph/Makefile.am	2023-09-05 22:28:57.000000000 +0900
+++ src/gwyddion-2.67/modules/graph/Makefile.am	2024-12-09 10:57:58.187542800 +0900
@@ -43,9 +43,7 @@ endif
 graph_LTLIBRARIES = \
 	$(graph_modules)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.67/modules/layer/Makefile.am	2022-10-18 00:16:16.000000000 +0900
+++ src/gwyddion-2.67/modules/layer/Makefile.am	2024-12-09 10:58:09.969167200 +0900
@@ -31,9 +31,7 @@ layer_LTLIBRARIES = \
 noinst_HEADERS = \
 	layer.h
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.67/modules/process/Makefile.am	2024-10-03 23:40:24.000000000 +0900
+++ src/gwyddion-2.67/modules/process/Makefile.am	2024-12-09 10:58:26.221241600 +0900
@@ -183,9 +183,7 @@ process_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.67/modules/pygwy/Makefile.am	2024-08-02 18:47:03.000000000 +0900
+++ src/gwyddion-2.67/modules/pygwy/Makefile.am	2024-12-09 10:58:43.827044600 +0900
@@ -233,9 +233,7 @@ EXTRA_DIST = \
 	pygtk-embed/codegen/reversewrapper.py \
 	pygtk-embed/codegen/scmexpr.py
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.67/modules/tools/Makefile.am	2022-10-18 00:16:17.000000000 +0900
+++ src/gwyddion-2.67/modules/tools/Makefile.am	2024-12-09 10:58:59.187173600 +0900
@@ -40,9 +40,7 @@ endif
 tools_LTLIBRARIES = \
 	$(tools_modules)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.67/modules/volume/Makefile.am	2023-10-16 03:25:05.000000000 +0900
+++ src/gwyddion-2.67/modules/volume/Makefile.am	2024-12-09 10:59:08.280407000 +0900
@@ -64,9 +64,7 @@ volume_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.67/modules/xyz/Makefile.am	2024-09-12 16:49:44.000000000 +0900
+++ src/gwyddion-2.67/modules/xyz/Makefile.am	2024-12-09 10:59:15.935937700 +0900
@@ -37,9 +37,7 @@ xyz_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.67/pixmaps/Makefile.am	2024-11-08 19:37:31.000000000 +0900
+++ src/gwyddion-2.67/pixmaps/Makefile.am	2024-12-09 16:21:44.812778700 +0900
@@ -1071,10 +1071,10 @@ gwy_mask_remove-24.png: $(srcdir)/src/gw
 	$(AM_V_at)$(PNGCRUSH) $(PNGCRUSH_SILENCE) -brute "gwy_mask_remove-24.png" "gwy_mask_remove-24.crush.png"
 	$(AM_V_at)if test -s "gwy_mask_remove-24.crush.png"; then mv -f "gwy_mask_remove-24.crush.png" "gwy_mask_remove-24.png"; fi
 
-gwy_mask_set-24.png: $(srcdir)/src/gwy_mask_set-24.svg
-	$(AM_V_GEN)DISPLAY=; unset DISPLAY; $(INKSCAPE_EXPORT) --export-width=24 --export-height=24 $(INKSCAPE_EXPORT_PNGFILE)="gwy_mask_set-24.png" "$(srcdir)/src/gwy_mask_set-24.svg"
-	$(AM_V_at)$(PNGCRUSH) $(PNGCRUSH_SILENCE) -brute "gwy_mask_set-24.png" "gwy_mask_set-24.crush.png"
-	$(AM_V_at)if test -s "gwy_mask_set-24.crush.png"; then mv -f "gwy_mask_set-24.crush.png" "gwy_mask_set-24.png"; fi
+# gwy_mask_set-24.png: $(srcdir)/src/gwy_mask_set-24.svg
+#	$(AM_V_GEN)DISPLAY=; unset DISPLAY; $(INKSCAPE_EXPORT) --export-width=24 --export-height=24 $(INKSCAPE_EXPORT_PNGFILE)="gwy_mask_set-24.png" "$(srcdir)/src/gwy_mask_set-24.svg"
+#	$(AM_V_at)$(PNGCRUSH) $(PNGCRUSH_SILENCE) -brute "gwy_mask_set-24.png" "gwy_mask_set-24.crush.png"
+#	$(AM_V_at)if test -s "gwy_mask_set-24.crush.png"; then mv -f "gwy_mask_set-24.crush.png" "gwy_mask_set-24.png"; fi
 
 gwy_mask_shift-24.png: $(srcdir)/src/gwy_mask_shift-24.svg
 	$(AM_V_GEN)DISPLAY=; unset DISPLAY; $(INKSCAPE_EXPORT) --export-width=24 --export-height=24 $(INKSCAPE_EXPORT_PNGFILE)="gwy_mask_shift-24.png" "$(srcdir)/src/gwy_mask_shift-24.svg"
--- origsrc/gwyddion-2.67/plugins/invert_python.py	2022-10-18 00:16:34.000000000 +0900
+++ src/gwyddion-2.67/plugins/invert_python.py	2024-12-09 23:52:57.657261200 +0900
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python
 # $Id: invert_python.py 22454 2019-08-31 19:53:40Z yeti-dn $
 # A very simple Gwyddion plug-in example in Python.
 # Written by Yeti <yeti@gwyddion.net>.  Public domain.
--- origsrc/gwyddion-2.67/thumbnailer/Makefile.am	2022-10-18 00:16:35.000000000 +0900
+++ src/gwyddion-2.67/thumbnailer/Makefile.am	2024-12-09 11:00:13.719080700 +0900
@@ -9,9 +9,7 @@ libgwyapp = $(top_builddir)/app/libgwyap
 
 kde4dir = $(KDE4_MODULE_DIR)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 # When the thumbnailer is not built, we still must distribute its source code.
 if KDE4_THUMBNAILER
