--- origsrc/gwyddion-2.49/app/Makefile.am	2017-08-14 20:21:07.000000000 +0900
+++ src/gwyddion-2.49/app/Makefile.am	2017-11-13 18:19:58.551000200 +0900
@@ -63,12 +63,12 @@ BUILT_SOURCES = \
 	authors.h \
 	release.h
 
-if OS_MSWIN
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwyapp2.def
+if OS_MSWIN
 win32_libs = -lole32
-libgwyapp2_la_DEPENDENCIES = libgwyapp2.def
 endif
+libgwyapp2_la_DEPENDENCIES = libgwyapp2.def
 
 # Must init
 EXTRA_DIST = \
@@ -170,9 +170,6 @@ gwyddion_LDADD = @COMMON_LDFLAGS@ @GTK_M
 	$(libgwyddion)
 
 if MAINTAINER_MODE
-authors.h: ${top_srcdir}/AUTHORS $(top_srcdir)/utils/gen-authors.py
-	$(AM_V_GEN)$(PYTHON) $(top_srcdir)/utils/gen-authors.py $(top_srcdir)/AUTHORS header >authors.h
-
 release.h: ${top_srcdir}/NEWS Makefile $(top_builddir)/config.h
 	$(AM_V_GEN)echo '/* This is a 'GENERATED' file. */' >release.h
 	$(AM_V_at)echo '/*< private_header >*/' >>release.h
--- origsrc/gwyddion-2.49/data/Makefile.am	2016-01-12 20:55:14.000000000 +0900
+++ src/gwyddion-2.49/data/Makefile.am	2017-11-13 17:56:30.687000000 +0900
@@ -105,7 +105,7 @@ endif
 # unexpanded ${prefix} in the schemas file.
 # Depend on Makefile, this may run this rule perhaps too often but ensures
 # the paths are updated every time they should.
-gwyddion-thumbnailer.schemas: gwyddion-thumbnailer.schemas.in Makefile
+gwyddion-thumbnailer.schemas: $(srcdir)/gwyddion-thumbnailer.schemas.in Makefile
 	$(AM_V_GEN)$(SED) -e 's/^\(<gconfschemafile\) .*\(>\)/\1\2/' \
 	  -e 's#[@]bindir[@]#$(bindir)#' \
 	   $(srcdir)/gwyddion-thumbnailer.schemas.in \
@@ -117,18 +117,18 @@ gwyddion-thumbnailer.schemas: gwyddion-t
 # unexpanded ${prefix} in the schemas file.
 # Depend on Makefile, this may run this rule perhaps too often but ensures
 # the paths are updated every time they should.
-gwyddion.desktop: gwyddion.desktop.in $(srcdir)/gwyddion.xml Makefile
+gwyddion.desktop: $(srcdir)/gwyddion.desktop.in $(srcdir)/gwyddion.xml Makefile
 	$(AM_V_GEN)x=`$(SED) -e '/^<mime-type/ s/.*<mime-type type="\(.*\)">.*/\1;/' -e t -e d <$(srcdir)/gwyddion.xml | tr -d '\n'` \
 	&& $(SED) -e "s#@MIME_TYPE_LIST@#$$x#" \
 	  -e 's#[@]datadir[@]#$(datadir)#' \
-	  <gwyddion.desktop.in >gwyddion.desktop
+	  <$(srcdir)/gwyddion.desktop.in >gwyddion.desktop
 
 # Generate gwyddion.thumbnailer from gwyddion.xml to keep the file format list
 # in sync.
-gwyddion.thumbnailer: gwyddion.thumbnailer.in $(srcdir)/gwyddion.xml
+gwyddion.thumbnailer: $(srcdir)/gwyddion.thumbnailer.in $(srcdir)/gwyddion.xml
 	$(AM_V_GEN)x=`$(SED) -e '/^<mime-type/ s/.*<mime-type type="\(.*\)">.*/\1;/' -e t -e d <$(srcdir)/gwyddion.xml | tr -d '\n'` \
 	&& $(SED) -e "s#@MIME_TYPE_LIST@#$$x#" \
-	  <gwyddion.thumbnailer.in >gwyddion.thumbnailer
+	  <$(srcdir)/gwyddion.thumbnailer.in >gwyddion.thumbnailer
 
 # Desktop file installation
 # FIXME: Maybe not according to current standards
--- origsrc/gwyddion-2.49/libdraw/Makefile.am	2017-08-14 20:21:07.000000000 +0900
+++ src/gwyddion-2.49/libdraw/Makefile.am	2017-11-13 17:57:04.024000000 +0900
@@ -23,11 +23,9 @@ libgwydraw2include_HEADERS = \
 
 lib_LTLIBRARIES = libgwydraw2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwydraw2.def
 libgwydraw2_la_DEPENDENCIES = libgwydraw2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.49/libgwyddion/Makefile.am	2017-08-14 20:21:07.000000000 +0900
+++ src/gwyddion-2.49/libgwyddion/Makefile.am	2017-11-13 17:57:22.096516400 +0900
@@ -35,11 +35,9 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwyddion2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwyddion2.def
 libgwyddion2_la_DEPENDENCIES = libgwyddion2.def
-endif
 
 # Must init
 EXTRA_DIST = gwyversion.h.in
--- origsrc/gwyddion-2.49/libgwydgets/Makefile.am	2017-08-14 20:21:07.000000000 +0900
+++ src/gwyddion-2.49/libgwydgets/Makefile.am	2017-11-13 17:57:40.789957200 +0900
@@ -76,11 +76,9 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwydgets2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwydgets2.def
 libgwydgets2_la_DEPENDENCIES = libgwydgets2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.49/libgwymodule/Makefile.am	2017-08-14 20:21:07.000000000 +0900
+++ src/gwyddion-2.49/libgwymodule/Makefile.am	2017-11-13 17:57:52.634840400 +0900
@@ -34,11 +34,9 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwymodule2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwymodule2.def
 libgwymodule2_la_DEPENDENCIES = libgwymodule2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.49/libprocess/Makefile.am	2017-08-14 20:21:07.000000000 +0900
+++ src/gwyddion-2.49/libprocess/Makefile.am	2017-11-13 17:58:12.305738500 +0900
@@ -52,11 +52,9 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwyprocess2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
 export_symbols = -export-symbols libgwyprocess2.def
 libgwyprocess2_la_DEPENDENCIES = libgwyprocess2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.49/modules/Makefile.am	2016-02-24 22:40:54.000000000 +0900
+++ src/gwyddion-2.49/modules/Makefile.am	2017-11-13 19:22:22.142999800 +0900
@@ -1,6 +1,6 @@
 # @(#) $Id: Makefile.am 18349 2016-02-24 13:40:54Z yeti-dn $
 
-SUBDIRS = file graph layer process tools volume xyz pygwy
+SUBDIRS = file graph layer process tools volume xyz # pygwy
 
 moduledir = $(pkglibdir)/modules
 
@@ -18,9 +18,7 @@ module_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(plugin_proxy_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.49/modules/file/Makefile.am	2017-08-07 23:15:17.000000000 +0900
+++ src/gwyddion-2.49/modules/file/Makefile.am	2017-11-13 17:59:11.753223000 +0900
@@ -282,9 +282,7 @@ EXTRA_DIST = \
 	$(disabled_sources) \
 	1CHECKLIST
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.49/modules/graph/Makefile.am	2017-06-30 19:58:03.000000000 +0900
+++ src/gwyddion-2.49/modules/graph/Makefile.am	2017-11-13 17:59:27.321647100 +0900
@@ -33,9 +33,7 @@ endif
 graph_LTLIBRARIES = \
 	$(graph_modules)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.49/modules/layer/Makefile.am	2017-06-30 19:58:03.000000000 +0900
+++ src/gwyddion-2.49/modules/layer/Makefile.am	2017-11-13 17:59:38.462625700 +0900
@@ -29,9 +29,7 @@ layer_LTLIBRARIES = \
 noinst_HEADERS = \
 	layer.h
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.49/modules/process/Makefile.am	2017-07-30 17:41:00.000000000 +0900
+++ src/gwyddion-2.49/modules/process/Makefile.am	2017-11-13 17:59:57.681179200 +0900
@@ -144,9 +144,7 @@ process_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.49/modules/pygwy/Makefile.am	2017-08-04 00:20:40.000000000 +0900
+++ src/gwyddion-2.49/modules/pygwy/Makefile.am	2017-11-13 19:07:22.464000000 +0900
@@ -149,8 +149,8 @@ PYGWY_H_FILES = \
 	$(top_srcdir)/libgwydgets/gwystock.h \
 	$(top_srcdir)/libgwydgets/gwyvectorlayer.h \
 	$(top_srcdir)/libgwydgets/gwyvruler.h \
-	wrap_calls.h \
-	macro-prototypes.h
+	$(srcdir)/wrap_calls.h \
+	$(srcdir)/macro-prototypes.h
 
 disabled_sources =
 
@@ -197,9 +197,7 @@ EXTRA_DIST = \
 	pygwy-fix-defs-2.py \
 	gwyutils.py
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
@@ -226,6 +224,8 @@ pygwy_la_LIBADD = $(module_libadd) $(PYT
 pygwy.defs: $(PYGWY_H_FILES) Makefile.am pygwy-fix-defs-1.py pygwy-fix-defs-2.py extra.defs
 	$(AM_V_GEN)$(PYTHON) $(PYGTK_CODEGENDIR)/h2def.py $(PYGWY_H_FILES) \
 		| $(PYTHON) $(srcdir)/pygwy-fix-defs-1.py >pygwy.defs.tmp
+	ln -fs $(srcdir)/extra.defs .
+	ln -fs $(srcdir)/override.defs .
 	$(PYTHON) $(srcdir)/pygwy-fix-defs-2.py --codegendir=$(PYGTK_CODEGENDIR) >pygwy.defs
 	rm pygwy.defs.tmp
 
--- origsrc/gwyddion-2.49/modules/tools/Makefile.am	2017-06-30 19:58:03.000000000 +0900
+++ src/gwyddion-2.49/modules/tools/Makefile.am	2017-11-13 18:00:28.806315400 +0900
@@ -37,9 +37,7 @@ endif
 tools_LTLIBRARIES = \
 	$(tools_modules)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.49/modules/volume/Makefile.am	2017-07-07 16:58:48.000000000 +0900
+++ src/gwyddion-2.49/modules/volume/Makefile.am	2017-11-13 18:00:40.885265300 +0900
@@ -38,9 +38,7 @@ volume_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.49/modules/xyz/Makefile.am	2017-06-30 19:58:03.000000000 +0900
+++ src/gwyddion-2.49/modules/xyz/Makefile.am	2017-11-13 18:00:52.025137300 +0900
@@ -30,9 +30,7 @@ xyz_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.49/pixmaps/Makefile.am	2017-08-12 22:52:05.000000000 +0900
+++ src/gwyddion-2.49/pixmaps/Makefile.am	2017-11-13 19:31:36.767000000 +0900
@@ -871,10 +871,10 @@ gwy_mask_remove-24.png: $(srcdir)/src/gw
 	$(PNGCRUSH) -q -brute "gwy_mask_remove-24.png" "gwy_mask_remove-24.crush.png"
 	if test -s "gwy_mask_remove-24.crush.png"; then mv -f "gwy_mask_remove-24.crush.png" "gwy_mask_remove-24.png"; fi
 
-gwy_mask_set-24.png: $(srcdir)/src/gwy_mask_set-24.svg
-	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_mask_set-24.png" --file="$(srcdir)/src/gwy_mask_set-24.svg"
-	$(PNGCRUSH) -q -brute "gwy_mask_set-24.png" "gwy_mask_set-24.crush.png"
-	if test -s "gwy_mask_set-24.crush.png"; then mv -f "gwy_mask_set-24.crush.png" "gwy_mask_set-24.png"; fi
+#gwy_mask_set-24.png: $(srcdir)/src/gwy_mask_set-24.svg
+#	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_mask_set-24.png" --file="$(srcdir)/src/gwy_mask_set-24.svg"
+#	$(PNGCRUSH) -q -brute "gwy_mask_set-24.png" "gwy_mask_set-24.crush.png"
+#	if test -s "gwy_mask_set-24.crush.png"; then mv -f "gwy_mask_set-24.crush.png" "gwy_mask_set-24.png"; fi
 
 gwy_mask_shrink-24.png: $(srcdir)/src/gwy_mask_shrink-24.svg
 	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_mask_shrink-24.png" --file="$(srcdir)/src/gwy_mask_shrink-24.svg"
@@ -1241,15 +1241,15 @@ gwy_volume_slice-24.png: $(srcdir)/src/g
 	$(PNGCRUSH) -q -brute "gwy_volume_slice-24.png" "gwy_volume_slice-24.crush.png"
 	if test -s "gwy_volume_slice-24.crush.png"; then mv -f "gwy_volume_slice-24.crush.png" "gwy_volume_slice-24.png"; fi
 
-gwy_volumize-24.png: $(srcdir)/src/gwy_volumize-24.svg
-	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_volumize-24.png" --file="$(srcdir)/src/gwy_volumize-24.svg"
-	$(PNGCRUSH) -q -brute "gwy_volumize-24.png" "gwy_volumize-24.crush.png"
-	if test -s "gwy_volumize-24.crush.png"; then mv -f "gwy_volumize-24.crush.png" "gwy_volumize-24.png"; fi
-
-gwy_volumize_layers-24.png: $(srcdir)/src/gwy_volumize_layers-24.svg
-	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_volumize_layers-24.png" --file="$(srcdir)/src/gwy_volumize_layers-24.svg"
-	$(PNGCRUSH) -q -brute "gwy_volumize_layers-24.png" "gwy_volumize_layers-24.crush.png"
-	if test -s "gwy_volumize_layers-24.crush.png"; then mv -f "gwy_volumize_layers-24.crush.png" "gwy_volumize_layers-24.png"; fi
+#gwy_volumize-24.png: $(srcdir)/src/gwy_volumize-24.svg
+#	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_volumize-24.png" --file="$(srcdir)/src/gwy_volumize-24.svg"
+#	$(PNGCRUSH) -q -brute "gwy_volumize-24.png" "gwy_volumize-24.crush.png"
+#	if test -s "gwy_volumize-24.crush.png"; then mv -f "gwy_volumize-24.crush.png" "gwy_volumize-24.png"; fi
+
+#gwy_volumize_layers-24.png: $(srcdir)/src/gwy_volumize_layers-24.svg
+#	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_volumize_layers-24.png" --file="$(srcdir)/src/gwy_volumize_layers-24.svg"
+#	$(PNGCRUSH) -q -brute "gwy_volumize_layers-24.png" "gwy_volumize_layers-24.crush.png"
+#	if test -s "gwy_volumize_layers-24.crush.png"; then mv -f "gwy_volumize_layers-24.crush.png" "gwy_volumize_layers-24.png"; fi
 
 gwy_xy_denoise-24.png: $(srcdir)/src/gwy_xy_denoise-24.svg
 	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_xy_denoise-24.png" --file="$(srcdir)/src/gwy_xy_denoise-24.svg"
