--- origsrc/gwyddion-2.56/app/Makefile.am	2019-11-04 17:28:58.000000000 +0900
+++ src/gwyddion-2.56/app/Makefile.am	2020-07-09 21:25:03.675227500 +0900
@@ -50,12 +50,10 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwyapp2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
-export_symbols = -export-symbols libgwyapp2.def
+# export_symbols = -export-symbols $(srcdir)/libgwyapp2.def
 win32_libs = -lole32
 libgwyapp2_la_DEPENDENCIES = libgwyapp2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.56/data/Makefile.am	2019-06-03 17:29:39.000000000 +0900
+++ src/gwyddion-2.56/data/Makefile.am	2020-07-09 20:43:01.470407400 +0900
@@ -99,21 +99,21 @@ endif
 # unexpanded ${prefix} in the desktop file.
 # Depend on Makefile, this may run this rule perhaps too often but ensures
 # the paths are updated every time they should.
-gwyddion.desktop: gwyddion.desktop.in $(srcdir)/gwyddion.xml Makefile
+gwyddion.desktop: $(srcdir)/gwyddion.desktop.in $(srcdir)/gwyddion.xml Makefile
 	$(AM_V_GEN)x=`$(SED) -e '/^<mime-type/ s/.*<mime-type type="\(.*\)">.*/\1;/' -e t -e d <$(srcdir)/gwyddion.xml | tr -d '\n'` \
 	&& $(SED) -e "s#@MIME_TYPE_LIST@#$$x#" \
 	  -e 's#[@]datadir[@]#$(datadir)#' \
-	  <gwyddion.desktop.in >gwyddion.desktop
+	  < $(srcdir)/gwyddion.desktop.in >gwyddion.desktop
 
 # Generate gwyddion.thumbnailer from gwyddion.xml to keep the file format list
 # in sync.
-gwyddion.thumbnailer: gwyddion.thumbnailer.in $(srcdir)/gwyddion.xml
+gwyddion.thumbnailer: $(srcdir)/gwyddion.thumbnailer.in $(srcdir)/gwyddion.xml
 	$(AM_V_GEN)x=`$(SED) -e '/^<mime-type/ s/.*<mime-type type="\(.*\)">.*/\1;/' -e t -e d <$(srcdir)/gwyddion.xml | tr -d '\n'` \
 	&& $(SED) \
 	     -e "s#@MIME_TYPE_LIST@#$$x#" \
 	     -e "s#[@]EXEEXT[@]#$(EXEEXT)#" \
 	     -e "s#[@]bindir[@]#$(bindir)#" \
-	     <gwyddion.thumbnailer.in >gwyddion.thumbnailer
+	     < $(srcdir)/gwyddion.thumbnailer.in >gwyddion.thumbnailer
 
 # Desktop file installation
 # FIXME: Maybe not according to current standards
--- origsrc/gwyddion-2.56/gwyddion/Makefile.am	2019-08-27 22:17:37.000000000 +0900
+++ src/gwyddion-2.56/gwyddion/Makefile.am	2020-07-09 21:27:03.209985400 +0900
@@ -58,8 +58,8 @@ gwyddion_LDADD = @COMMON_LDFLAGS@ @GTK_M
 	$(libgwyddion)
 
 if MAINTAINER_MODE
-authors.h: ${top_srcdir}/AUTHORS $(top_srcdir)/utils/gen-authors.py
-	$(AM_V_GEN)$(PYTHON) $(top_srcdir)/utils/gen-authors.py $(top_srcdir)/AUTHORS header >authors.h
+# authors.h: ${top_srcdir}/AUTHORS $(top_srcdir)/utils/gen-authors.py
+#	$(AM_V_GEN)$(PYTHON) $(top_srcdir)/utils/gen-authors.py $(top_srcdir)/AUTHORS header >authors.h
 
 release.h: ${top_srcdir}/NEWS Makefile $(top_builddir)/config.h
 	$(AM_V_GEN)echo '/* This is a 'GENERATED' file. */' >release.h
--- origsrc/gwyddion-2.56/libdraw/Makefile.am	2019-08-27 20:57:57.000000000 +0900
+++ src/gwyddion-2.56/libdraw/Makefile.am	2020-07-09 21:12:26.883800500 +0900
@@ -23,11 +23,9 @@ libgwydraw2include_HEADERS = \
 
 lib_LTLIBRARIES = libgwydraw2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
-export_symbols = -export-symbols libgwydraw2.def
+# export_symbols = -export-symbols $(srcdir)/libgwydraw2.def
 libgwydraw2_la_DEPENDENCIES = libgwydraw2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.56/libgwyddion/Makefile.am	2020-06-29 17:19:12.000000000 +0900
+++ src/gwyddion-2.56/libgwyddion/Makefile.am	2020-07-09 21:03:21.026760500 +0900
@@ -38,11 +38,9 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwyddion2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
-export_symbols = -export-symbols libgwyddion2.def
+# export_symbols = -export-symbols $(srcdir)/libgwyddion2.def
 libgwyddion2_la_DEPENDENCIES = libgwyddion2.def
-endif
 
 # Must init
 EXTRA_DIST = gwyversion.h.in
--- origsrc/gwyddion-2.56/libgwydgets/Makefile.am	2020-06-29 17:19:12.000000000 +0900
+++ src/gwyddion-2.56/libgwydgets/Makefile.am	2020-07-09 21:17:38.088836700 +0900
@@ -77,11 +77,9 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwydgets2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
-export_symbols = -export-symbols libgwydgets2.def
+# export_symbols = -export-symbols $(srcdir)/libgwydgets2.def
 libgwydgets2_la_DEPENDENCIES = libgwydgets2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.56/libgwymodule/Makefile.am	2018-04-11 21:04:29.000000000 +0900
+++ src/gwyddion-2.56/libgwymodule/Makefile.am	2020-07-09 21:20:15.845624600 +0900
@@ -34,11 +34,9 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwymodule2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
-export_symbols = -export-symbols libgwymodule2.def
+# export_symbols = -export-symbols $(srcdir)/libgwymodule2.def
 libgwymodule2_la_DEPENDENCIES = libgwymodule2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.56/libprocess/Makefile.am	2020-06-29 17:19:12.000000000 +0900
+++ src/gwyddion-2.56/libprocess/Makefile.am	2020-07-09 21:08:52.055710800 +0900
@@ -54,11 +54,9 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwyprocess2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
-export_symbols = -export-symbols libgwyprocess2.def
+# export_symbols = -export-symbols $(srcdir)/libgwyprocess2.def
 libgwyprocess2_la_DEPENDENCIES = libgwyprocess2.def
-endif
 
 # Must init
 EXTRA_DIST =
--- origsrc/gwyddion-2.56/modules/Makefile.am	2018-04-11 21:02:12.000000000 +0900
+++ src/gwyddion-2.56/modules/Makefile.am	2020-07-09 20:46:27.212891500 +0900
@@ -19,9 +19,7 @@ module_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(plugin_proxy_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.56/modules/file/Makefile.am	2020-05-18 20:52:54.000000000 +0900
+++ src/gwyddion-2.56/modules/file/Makefile.am	2020-07-09 20:46:52.221325000 +0900
@@ -319,9 +319,7 @@ EXTRA_DIST = \
 	$(disabled_sources) \
 	1CHECKLIST
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.56/modules/file/gwyzip.h	2019-11-21 21:41:41.000000000 +0900
+++ src/gwyddion-2.56/modules/file/gwyzip.h	2020-07-09 21:55:37.450084200 +0900
@@ -61,6 +61,7 @@ static guchar*    gwyzip_get_file_conten
 #ifdef HAVE_MINIZIP1
 #define HAVE_GWYZIP 1
 #include <unzip.h>
+#include <zip.h>
 
 struct _GwyZipFile {
     unzFile *unzfile;
--- origsrc/gwyddion-2.56/modules/graph/Makefile.am	2019-06-03 21:49:32.000000000 +0900
+++ src/gwyddion-2.56/modules/graph/Makefile.am	2020-07-09 20:47:07.967773700 +0900
@@ -35,9 +35,7 @@ endif
 graph_LTLIBRARIES = \
 	$(graph_modules)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.56/modules/layer/Makefile.am	2019-01-02 07:09:58.000000000 +0900
+++ src/gwyddion-2.56/modules/layer/Makefile.am	2020-07-09 20:47:23.592957200 +0900
@@ -30,9 +30,7 @@ layer_LTLIBRARIES = \
 noinst_HEADERS = \
 	layer.h
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.56/modules/process/Makefile.am	2019-09-05 22:10:45.000000000 +0900
+++ src/gwyddion-2.56/modules/process/Makefile.am	2020-07-09 20:47:44.301229500 +0900
@@ -168,9 +168,7 @@ process_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.56/modules/pygwy/Makefile.am	2020-02-05 22:16:15.000000000 +0900
+++ src/gwyddion-2.56/modules/pygwy/Makefile.am	2020-07-09 20:48:07.598859100 +0900
@@ -226,9 +226,7 @@ EXTRA_DIST = \
 	pygtk-embed/codegen/reversewrapper.py \
 	pygtk-embed/codegen/scmexpr.py
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.56/modules/tools/Makefile.am	2020-04-30 22:03:58.000000000 +0900
+++ src/gwyddion-2.56/modules/tools/Makefile.am	2020-07-09 20:48:22.698800400 +0900
@@ -40,9 +40,7 @@ endif
 tools_LTLIBRARIES = \
 	$(tools_modules)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.56/modules/volume/Makefile.am	2019-10-02 17:36:09.000000000 +0900
+++ src/gwyddion-2.56/modules/volume/Makefile.am	2020-07-09 20:48:39.381252500 +0900
@@ -49,9 +49,7 @@ volume_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.56/modules/xyz/Makefile.am	2019-10-11 00:48:53.000000000 +0900
+++ src/gwyddion-2.56/modules/xyz/Makefile.am	2020-07-09 20:48:54.827398500 +0900
@@ -32,9 +32,7 @@ xyz_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.56/pixmaps/Makefile.am	2020-05-07 01:01:05.000000000 +0900
+++ src/gwyddion-2.56/pixmaps/Makefile.am	2020-07-10 00:29:34.548604100 +0900
@@ -1007,10 +1007,10 @@ gwy_mask_remove-24.png: $(srcdir)/src/gw
 	$(PNGCRUSH) $(PNGCRUSH_SILENCE) -brute "gwy_mask_remove-24.png" "gwy_mask_remove-24.crush.png"
 	if test -s "gwy_mask_remove-24.crush.png"; then mv -f "gwy_mask_remove-24.crush.png" "gwy_mask_remove-24.png"; fi
 
-gwy_mask_set-24.png: $(srcdir)/src/gwy_mask_set-24.svg
-	$(INKSCAPE_EXPORT) --export-width=24 --export-height=24 $(INKSCAPE_EXPORT_PNGFILE)="gwy_mask_set-24.png" "$(srcdir)/src/gwy_mask_set-24.svg"
-	$(PNGCRUSH) $(PNGCRUSH_SILENCE) -brute "gwy_mask_set-24.png" "gwy_mask_set-24.crush.png"
-	if test -s "gwy_mask_set-24.crush.png"; then mv -f "gwy_mask_set-24.crush.png" "gwy_mask_set-24.png"; fi
+# gwy_mask_set-24.png: $(srcdir)/src/gwy_mask_set-24.svg
+#	$(INKSCAPE_EXPORT) --export-width=24 --export-height=24 $(INKSCAPE_EXPORT_PNGFILE)="gwy_mask_set-24.png" "$(srcdir)/src/gwy_mask_set-24.svg"
+#	$(PNGCRUSH) $(PNGCRUSH_SILENCE) -brute "gwy_mask_set-24.png" "gwy_mask_set-24.crush.png"
+#	if test -s "gwy_mask_set-24.crush.png"; then mv -f "gwy_mask_set-24.crush.png" "gwy_mask_set-24.png"; fi
 
 gwy_mask_shrink-24.png: $(srcdir)/src/gwy_mask_shrink-24.svg
 	$(INKSCAPE_EXPORT) --export-width=24 --export-height=24 $(INKSCAPE_EXPORT_PNGFILE)="gwy_mask_shrink-24.png" "$(srcdir)/src/gwy_mask_shrink-24.svg"
