--- origsrc/gwyddion-2.53/app/Makefile.am	2019-02-26 21:03:16.000000000 +0900
+++ src/gwyddion-2.53/app/Makefile.am	2019-06-29 10:48:46.787299800 +0900
@@ -50,8 +50,8 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwyapp2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
+if OS_MSWIN
 export_symbols = -export-symbols libgwyapp2.def
 win32_libs = -lole32
 libgwyapp2_la_DEPENDENCIES = libgwyapp2.def
--- origsrc/gwyddion-2.53/data/Makefile.am	2018-08-22 22:33:53.000000000 +0900
+++ src/gwyddion-2.53/data/Makefile.am	2019-06-29 10:48:46.817705100 +0900
@@ -91,21 +91,21 @@ endif
 # unexpanded ${prefix} in the desktop file.
 # Depend on Makefile, this may run this rule perhaps too often but ensures
 # the paths are updated every time they should.
-gwyddion.desktop: gwyddion.desktop.in $(srcdir)/gwyddion.xml Makefile
+gwyddion.desktop: $(srcdir)/gwyddion.desktop.in $(srcdir)/gwyddion.xml Makefile
 	$(AM_V_GEN)x=`$(SED) -e '/^<mime-type/ s/.*<mime-type type="\(.*\)">.*/\1;/' -e t -e d <$(srcdir)/gwyddion.xml | tr -d '\n'` \
 	&& $(SED) -e "s#@MIME_TYPE_LIST@#$$x#" \
 	  -e 's#[@]datadir[@]#$(datadir)#' \
-	  <gwyddion.desktop.in >gwyddion.desktop
+	  < $(srcdir)/gwyddion.desktop.in >gwyddion.desktop
 
 # Generate gwyddion.thumbnailer from gwyddion.xml to keep the file format list
 # in sync.
-gwyddion.thumbnailer: gwyddion.thumbnailer.in $(srcdir)/gwyddion.xml
+gwyddion.thumbnailer: $(srcdir)/gwyddion.thumbnailer.in $(srcdir)/gwyddion.xml
 	$(AM_V_GEN)x=`$(SED) -e '/^<mime-type/ s/.*<mime-type type="\(.*\)">.*/\1;/' -e t -e d <$(srcdir)/gwyddion.xml | tr -d '\n'` \
 	&& $(SED) \
 	     -e "s#@MIME_TYPE_LIST@#$$x#" \
 	     -e "s#[@]EXEEXT[@]#$(EXEEXT)#" \
 	     -e "s#[@]bindir[@]#$(bindir)#" \
-	     <gwyddion.thumbnailer.in >gwyddion.thumbnailer
+	     < $(srcdir)/gwyddion.thumbnailer.in >gwyddion.thumbnailer
 
 # Desktop file installation
 # FIXME: Maybe not according to current standards
--- origsrc/gwyddion-2.53/gwyddion/Makefile.am	2018-11-21 22:39:11.000000000 +0900
+++ src/gwyddion-2.53/gwyddion/Makefile.am	2019-06-29 10:48:46.832450400 +0900
@@ -58,8 +58,8 @@ gwyddion_LDADD = @COMMON_LDFLAGS@ @GTK_M
 	$(libgwyddion)
 
 if MAINTAINER_MODE
-authors.h: ${top_srcdir}/AUTHORS $(top_srcdir)/utils/gen-authors.py
-	$(AM_V_GEN)$(PYTHON) $(top_srcdir)/utils/gen-authors.py $(top_srcdir)/AUTHORS header >authors.h
+#authors.h: ${top_srcdir}/AUTHORS $(top_srcdir)/utils/gen-authors.py
+#	$(AM_V_GEN)$(PYTHON) $(top_srcdir)/utils/gen-authors.py $(top_srcdir)/AUTHORS header >authors.h
 
 release.h: ${top_srcdir}/NEWS Makefile $(top_builddir)/config.h
 	$(AM_V_GEN)echo '/* This is a 'GENERATED' file. */' >release.h
--- origsrc/gwyddion-2.53/libdraw/Makefile.am	2019-02-26 21:03:16.000000000 +0900
+++ src/gwyddion-2.53/libdraw/Makefile.am	2019-06-29 10:48:46.863688400 +0900
@@ -23,8 +23,8 @@ libgwydraw2include_HEADERS = \
 
 lib_LTLIBRARIES = libgwydraw2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
+if OS_MSWIN
 export_symbols = -export-symbols libgwydraw2.def
 libgwydraw2_la_DEPENDENCIES = libgwydraw2.def
 endif
--- origsrc/gwyddion-2.53/libgwyddion/Makefile.am	2019-02-26 21:03:16.000000000 +0900
+++ src/gwyddion-2.53/libgwyddion/Makefile.am	2019-06-29 10:48:46.881622600 +0900
@@ -37,8 +37,8 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwyddion2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
+if OS_MSWIN
 export_symbols = -export-symbols libgwyddion2.def
 libgwyddion2_la_DEPENDENCIES = libgwyddion2.def
 endif
--- origsrc/gwyddion-2.53/libgwydgets/Makefile.am	2019-02-26 21:03:16.000000000 +0900
+++ src/gwyddion-2.53/libgwydgets/Makefile.am	2019-06-29 10:48:46.894947300 +0900
@@ -77,8 +77,8 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwydgets2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
+if OS_MSWIN
 export_symbols = -export-symbols libgwydgets2.def
 libgwydgets2_la_DEPENDENCIES = libgwydgets2.def
 endif
--- origsrc/gwyddion-2.53/libgwymodule/Makefile.am	2018-04-11 21:04:29.000000000 +0900
+++ src/gwyddion-2.53/libgwymodule/Makefile.am	2019-06-29 10:48:46.894947300 +0900
@@ -34,8 +34,8 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwymodule2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
+if OS_MSWIN
 export_symbols = -export-symbols libgwymodule2.def
 libgwymodule2_la_DEPENDENCIES = libgwymodule2.def
 endif
--- origsrc/gwyddion-2.53/libprocess/Makefile.am	2019-02-26 21:03:16.000000000 +0900
+++ src/gwyddion-2.53/libprocess/Makefile.am	2019-06-29 10:48:46.910563000 +0900
@@ -53,8 +53,8 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libgwyprocess2.la
 
-if OS_MSWIN
 no_undefined = -no-undefined
+if OS_MSWIN
 export_symbols = -export-symbols libgwyprocess2.def
 libgwyprocess2_la_DEPENDENCIES = libgwyprocess2.def
 endif
--- origsrc/gwyddion-2.53/modules/Makefile.am	2018-04-11 21:02:12.000000000 +0900
+++ src/gwyddion-2.53/modules/Makefile.am	2019-06-29 10:48:46.926188900 +0900
@@ -19,9 +19,7 @@ module_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(plugin_proxy_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.53/modules/file/Makefile.am	2018-10-30 19:42:51.000000000 +0900
+++ src/gwyddion-2.53/modules/file/Makefile.am	2019-06-29 10:48:46.926188900 +0900
@@ -300,9 +300,7 @@ EXTRA_DIST = \
 	$(disabled_sources) \
 	1CHECKLIST
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.53/modules/graph/Makefile.am	2018-04-11 21:02:12.000000000 +0900
+++ src/gwyddion-2.53/modules/graph/Makefile.am	2019-06-29 10:48:46.941822200 +0900
@@ -33,9 +33,7 @@ endif
 graph_LTLIBRARIES = \
 	$(graph_modules)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.53/modules/layer/Makefile.am	2019-01-02 07:09:58.000000000 +0900
+++ src/gwyddion-2.53/modules/layer/Makefile.am	2019-06-29 10:48:46.941822200 +0900
@@ -30,9 +30,7 @@ layer_LTLIBRARIES = \
 noinst_HEADERS = \
 	layer.h
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.53/modules/process/Makefile.am	2019-01-25 22:23:09.000000000 +0900
+++ src/gwyddion-2.53/modules/process/Makefile.am	2019-06-29 10:48:46.957441400 +0900
@@ -159,9 +159,7 @@ process_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.53/modules/pygwy/Makefile.am	2018-11-29 18:17:14.000000000 +0900
+++ src/gwyddion-2.53/modules/pygwy/Makefile.am	2019-06-29 10:48:46.957441400 +0900
@@ -211,9 +211,7 @@ EXTRA_DIST = \
 	pygtk-embed/codegen/reversewrapper.py \
 	pygtk-embed/codegen/scmexpr.py
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Module\"
 AM_CFLAGS = @COMMON_CFLAGS@
--- origsrc/gwyddion-2.53/modules/tools/Makefile.am	2019-01-03 00:06:09.000000000 +0900
+++ src/gwyddion-2.53/modules/tools/Makefile.am	2019-06-29 10:48:46.973072300 +0900
@@ -39,9 +39,7 @@ endif
 tools_LTLIBRARIES = \
 	$(tools_modules)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.53/modules/volume/Makefile.am	2018-12-07 03:34:00.000000000 +0900
+++ src/gwyddion-2.53/modules/volume/Makefile.am	2019-06-29 10:48:47.035569700 +0900
@@ -46,9 +46,7 @@ volume_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.53/modules/xyz/Makefile.am	2019-01-30 02:27:53.000000000 +0900
+++ src/gwyddion-2.53/modules/xyz/Makefile.am	2019-06-29 10:48:47.051189300 +0900
@@ -31,9 +31,7 @@ xyz_LTLIBRARIES = \
 EXTRA_DIST = \
 	$(disabled_sources)
 
-if OS_MSWIN
 no_undefined = -no-undefined
-endif
 
 if MODULE_DEPENDENCIES
 module_libadd = \
--- origsrc/gwyddion-2.53/pixmaps/Makefile.am	2019-01-03 00:02:03.000000000 +0900
+++ src/gwyddion-2.53/pixmaps/Makefile.am	2019-06-29 10:48:47.066821800 +0900
@@ -954,10 +954,10 @@ gwy_mask_remove-24.png: $(srcdir)/src/gw
 	$(PNGCRUSH) $(PNGCRUSH_SILENCE) -brute "gwy_mask_remove-24.png" "gwy_mask_remove-24.crush.png"
 	if test -s "gwy_mask_remove-24.crush.png"; then mv -f "gwy_mask_remove-24.crush.png" "gwy_mask_remove-24.png"; fi
 
-gwy_mask_set-24.png: $(srcdir)/src/gwy_mask_set-24.svg
-	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_mask_set-24.png" --file="$(srcdir)/src/gwy_mask_set-24.svg"
-	$(PNGCRUSH) $(PNGCRUSH_SILENCE) -brute "gwy_mask_set-24.png" "gwy_mask_set-24.crush.png"
-	if test -s "gwy_mask_set-24.crush.png"; then mv -f "gwy_mask_set-24.crush.png" "gwy_mask_set-24.png"; fi
+#gwy_mask_set-24.png: $(srcdir)/src/gwy_mask_set-24.svg
+#	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_mask_set-24.png" --file="$(srcdir)/src/gwy_mask_set-24.svg"
+#	$(PNGCRUSH) $(PNGCRUSH_SILENCE) -brute "gwy_mask_set-24.png" "gwy_mask_set-24.crush.png"
+#	if test -s "gwy_mask_set-24.crush.png"; then mv -f "gwy_mask_set-24.crush.png" "gwy_mask_set-24.png"; fi
 
 gwy_mask_shrink-24.png: $(srcdir)/src/gwy_mask_shrink-24.svg
 	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_mask_shrink-24.png" --file="$(srcdir)/src/gwy_mask_shrink-24.svg"
@@ -1414,15 +1414,15 @@ gwy_volume_slice-24.png: $(srcdir)/src/g
 	$(PNGCRUSH) $(PNGCRUSH_SILENCE) -brute "gwy_volume_slice-24.png" "gwy_volume_slice-24.crush.png"
 	if test -s "gwy_volume_slice-24.crush.png"; then mv -f "gwy_volume_slice-24.crush.png" "gwy_volume_slice-24.png"; fi
 
-gwy_volumize-24.png: $(srcdir)/src/gwy_volumize-24.svg
-	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_volumize-24.png" --file="$(srcdir)/src/gwy_volumize-24.svg"
-	$(PNGCRUSH) $(PNGCRUSH_SILENCE) -brute "gwy_volumize-24.png" "gwy_volumize-24.crush.png"
-	if test -s "gwy_volumize-24.crush.png"; then mv -f "gwy_volumize-24.crush.png" "gwy_volumize-24.png"; fi
-
-gwy_volumize_layers-24.png: $(srcdir)/src/gwy_volumize_layers-24.svg
-	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_volumize_layers-24.png" --file="$(srcdir)/src/gwy_volumize_layers-24.svg"
-	$(PNGCRUSH) $(PNGCRUSH_SILENCE) -brute "gwy_volumize_layers-24.png" "gwy_volumize_layers-24.crush.png"
-	if test -s "gwy_volumize_layers-24.crush.png"; then mv -f "gwy_volumize_layers-24.crush.png" "gwy_volumize_layers-24.png"; fi
+#gwy_volumize-24.png: $(srcdir)/src/gwy_volumize-24.svg
+#	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_volumize-24.png" --file="$(srcdir)/src/gwy_volumize-24.svg"
+#	$(PNGCRUSH) $(PNGCRUSH_SILENCE) -brute "gwy_volumize-24.png" "gwy_volumize-24.crush.png"
+#	if test -s "gwy_volumize-24.crush.png"; then mv -f "gwy_volumize-24.crush.png" "gwy_volumize-24.png"; fi
+
+#gwy_volumize_layers-24.png: $(srcdir)/src/gwy_volumize_layers-24.svg
+#	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_volumize_layers-24.png" --file="$(srcdir)/src/gwy_volumize_layers-24.svg"
+#	$(PNGCRUSH) $(PNGCRUSH_SILENCE) -brute "gwy_volumize_layers-24.png" "gwy_volumize_layers-24.crush.png"
+#	if test -s "gwy_volumize_layers-24.crush.png"; then mv -f "gwy_volumize_layers-24.crush.png" "gwy_volumize_layers-24.png"; fi
 
 gwy_xy_denoise-24.png: $(srcdir)/src/gwy_xy_denoise-24.svg
 	$(INKSCAPE_RENDER_PNG) --export-width=24 --export-height=24 --export-png="gwy_xy_denoise-24.png" --file="$(srcdir)/src/gwy_xy_denoise-24.svg"
