--- origsrc/global-6.6.14/configure.ac	2024-12-11 15:44:34.000000000 +0900
+++ src/global-6.6.14/configure.ac	2024-12-24 19:10:25.621422500 +0900
@@ -190,7 +190,7 @@ dnl esac
 dnl #### end of gtags-cscope ###
 
 AM_CPPFLAGS='-I$(top_srcdir)/libparser $(LTDLINCL) -I$(top_srcdir)/libutil -I$(top_srcdir)/libdb -I$(top_srcdir)/libglibc -I../libutil'
-DBLIBRARY='../libdb/libglodb.a'
+DBLIBRARY='../libdb/libglodb.la'
 
 dnl
 dnl This option is for debugging GLOBAL. Not for normal use.
@@ -247,7 +247,7 @@ AC_ARG_WITH(sqlite3,
 			if ! test -r "$withval/include/sqlite3.h"; then
 				AC_MSG_ERROR([header $withval/include/sqlite3.h not found.])
 			fi
-			if ! test -r "$withval/lib/libsqlite3.so" && ! test -r "$withval/lib/libsqlite3.dylib"; then
+			if ! test -r "$withval/lib/libsqlite3.so" && ! test -r "$withval/lib/libsqlite3.dylib" && ! test -r "$withval/lib/libsqlite3.dll.a"; then
 				AC_MSG_ERROR([library $withval/lib/libsqlite3.* not found.])
 			fi
 			AM_CONDITIONAL(USE_SQLITE3_VENDORED, false)
--- origsrc/global-6.6.14/libdb/Makefile.am	2024-12-11 15:44:34.000000000 +0900
+++ src/global-6.6.14/libdb/Makefile.am	2024-12-24 19:10:25.637078200 +0900
@@ -10,7 +10,7 @@
 # WITHOUT ANY WARRANTY, to the extent permitted by law; without even the
 # implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 #
-noinst_LIBRARIES = libglodb.a
+noinst_LTLIBRARIES = libglodb.la
 
 INCS = btree.h db.h extern.h mpool.h queue.h compat.h
 SRCS = bt_close.c bt_conv.c bt_debug.c bt_delete.c bt_get.c bt_open.c bt_overflow.c \
@@ -23,5 +23,5 @@ AM_CPPFLAGS += -DSQLITE_THREADSAFE=0 -DS
 endif
 
 noinst_HEADERS = $(INCS)
-libglodb_a_SOURCES = $(SRCS)
-libglodb_a_DEPENDENCIES = $(libglodb_a_LIBADD)
+libglodb_la_SOURCES = $(SRCS)
+libglodb_la_DEPENDENCIES = $(libglodb_la_LIBADD)
