--- origsrc/global-6.2.7/plugin-factory/Makefile.am	2012-12-22 09:43:58.000000000 +0900
+++ src/global-6.2.7/plugin-factory/Makefile.am	2012-12-31 12:10:50.794091300 +0900
@@ -14,10 +14,10 @@ plugindir = ${libdir}/gtags
 plugin_LTLIBRARIES = exuberant-ctags.la user-custom.la
 
 exuberant_ctags_la_SOURCES = exuberant-ctags.c
-exuberant_ctags_la_LDFLAGS = -module -avoid-version
+exuberant_ctags_la_LDFLAGS = -module -avoid-version -no-undefined
 
 # skeleton
 user_custom_la_SOURCES = user-custom.c
-user_custom_la_LDFLAGS = -module -avoid-version
+user_custom_la_LDFLAGS = -module -avoid-version -no-undefined
 
 AM_CPPFLAGS = @AM_CPPFLAGS@
