--- origsrc/global-6.1/plugin-factory/Makefile.am	2011-10-07 14:02:22.000000000 +0900
+++ src/global-6.1/plugin-factory/Makefile.am	2011-10-10 22:02:36.280126300 +0900
@@ -14,10 +14,10 @@ plugindir = ${libdir}/gtags
 plugin_LTLIBRARIES = exuberant-ctags.la user-custom.la
 
 exuberant_ctags_la_SOURCES = exuberant-ctags.c
-exuberant_ctags_la_LDFLAGS = -module -avoid-version
+exuberant_ctags_la_LDFLAGS = -module -avoid-version -no-undefined
 
 # skeleton
 user_custom_la_SOURCES = user-custom.c
-user_custom_la_LDFLAGS = -module -avoid-version
+user_custom_la_LDFLAGS = -module -avoid-version -no-undefined
 
 INCLUDES = @INCLUDES@
