--- origsrc/global-6.2.4/plugin-factory/Makefile.am	2012-05-30 18:30:15.000000000 +0900
+++ src/global-6.2.4/plugin-factory/Makefile.am	2012-06-08 07:34:53.497241200 +0900
@@ -14,10 +14,10 @@ plugindir = ${libdir}/gtags
 plugin_LTLIBRARIES = exuberant-ctags.la user-custom.la
 
 exuberant_ctags_la_SOURCES = exuberant-ctags.c
-exuberant_ctags_la_LDFLAGS = -module -avoid-version
+exuberant_ctags_la_LDFLAGS = -module -avoid-version -no-undefined
 
 # skeleton
 user_custom_la_SOURCES = user-custom.c
-user_custom_la_LDFLAGS = -module -avoid-version
+user_custom_la_LDFLAGS = -module -avoid-version -no-undefined
 
 INCLUDES = @INCLUDES@
