--- origsrc/global-6.2.9/plugin-factory/Makefile.am	2013-08-26 15:22:17.000000000 +0900
+++ src/global-6.2.9/plugin-factory/Makefile.am	2013-09-18 23:17:29.409844700 +0900
@@ -14,10 +14,10 @@ plugindir = ${libdir}/gtags
 plugin_LTLIBRARIES = exuberant-ctags.la user-custom.la
 
 exuberant_ctags_la_SOURCES = exuberant-ctags.c
-exuberant_ctags_la_LDFLAGS = -module -avoid-version
+exuberant_ctags_la_LDFLAGS = -module -avoid-version -no-undefined
 
 # skeleton
 user_custom_la_SOURCES = user-custom.c
-user_custom_la_LDFLAGS = -module -avoid-version
+user_custom_la_LDFLAGS = -module -avoid-version -no-undefined
 
 AM_CPPFLAGS = @AM_CPPFLAGS@
