--- origsrc/global-5.9.7/plugin-example/Makefile.am	2011-07-01 15:43:34.000000000 +0900
+++ src/global-5.9.7/plugin-example/Makefile.am	2011-07-30 14:02:59.955163200 +0900
@@ -14,6 +14,6 @@ plugindir = ${libdir}/gtags
 plugin_LTLIBRARIES = exuberant-ctags.la
 
 exuberant_ctags_la_SOURCES = exuberant-ctags.c
-exuberant_ctags_la_LDFLAGS = -module -avoid-version
+exuberant_ctags_la_LDFLAGS = -module -avoid-version -no-undefined
 
 INCLUDES = @INCLUDES@
