--- origsrc/global-6.6.5/configure.ac	2020-09-03 15:27:00.000000000 +0900
+++ src/global-6.6.5/configure.ac	2020-09-06 20:23:56.752715000 +0900
@@ -180,7 +180,7 @@ dnl esac
 dnl #### end of gtags-cscope ###
 
 AM_CPPFLAGS='-I$(top_srcdir)/libparser $(LTDLINCL) -I$(top_srcdir)/libutil -I$(top_srcdir)/libdb -I$(top_srcdir)/libglibc -I../libutil'
-DBLIBRARY='../libdb/libglodb.a'
+DBLIBRARY='../libdb/libglodb.la'
 
 dnl
 dnl This option is for debugging GLOBAL. Not for normal use.
--- origsrc/global-6.6.5/libdb/Makefile.am	2020-09-03 15:27:00.000000000 +0900
+++ src/global-6.6.5/libdb/Makefile.am	2020-09-06 20:19:39.420859200 +0900
@@ -10,18 +10,19 @@
 # WITHOUT ANY WARRANTY, to the extent permitted by law; without even the
 # implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 #
-noinst_LIBRARIES = libglodb.a
+noinst_LTLIBRARIES = libglodb.la
 
 INCS = btree.h db.h extern.h mpool.h queue.h compat.h
 SRCS = bt_close.c bt_conv.c bt_debug.c bt_delete.c bt_get.c bt_open.c bt_overflow.c \
         bt_page.c bt_put.c bt_search.c bt_seq.c bt_split.c bt_utils.c db.c mpool.c
 
 if USE_SQLITE3
-INCS += sqlite3.h
-SRCS += sqlite3.c
-AM_CPPFLAGS += -DSQLITE_THREADSAFE=0 -DSQLITE_OMIT_LOAD_EXTENSION
+#INCS += sqlite3.h
+#SRCS += sqlite3.c
+#AM_CPPFLAGS += -DSQLITE_THREADSAFE=0 -DSQLITE_OMIT_LOAD_EXTENSION
+libglodb_la_LIBADD = -lsqlite3
 endif
 
 noinst_HEADERS = $(INCS)
-libglodb_a_SOURCES = $(SRCS)
-libglodb_a_DEPENDENCIES = $(libglodb_a_LIBADD)
+libglodb_la_SOURCES = $(SRCS)
+#libglodb_a_DEPENDENCIES = $(libglodb_a_LIBADD)
