--- origsrc/global-6.2.11/plugin-factory/Makefile.am	2014-03-12 11:49:34.000000000 +0900
+++ src/global-6.2.11/plugin-factory/Makefile.am	2014-03-15 14:18:47.803455800 +0900
@@ -14,11 +14,11 @@ plugindir = ${libdir}/gtags
 plugin_LTLIBRARIES = exuberant-ctags.la user-custom.la
 
 exuberant_ctags_la_SOURCES = exuberant-ctags.c
-exuberant_ctags_la_LDFLAGS = -module -avoid-version
+exuberant_ctags_la_LDFLAGS = -module -avoid-version -no-undefined
 
 # skeleton
 user_custom_la_SOURCES = user-custom.c
-user_custom_la_LDFLAGS = -module -avoid-version
+user_custom_la_LDFLAGS = -module -avoid-version -no-undefined
 
 gtagsdir = ${datadir}/gtags
 gtags_DATA = PLUGIN_HOWTO
