--- origsrc/gsignond-plugin-mail-0.3.0/src/meson.build	2018-10-31 19:12:33.000000000 +0900
+++ src/gsignond-plugin-mail-0.3.0/src/meson.build	2025-12-12 05:27:17.048878000 +0900
@@ -8,7 +8,8 @@ mail_plugin = shared_module(meson.projec
     mail_files,
     dependencies : dependencies,
     install: true,
-    install_dir : gplugins_dir
+    install_dir : gplugins_dir,
+    name_prefix: ''
 )
 
 if get_option('documentation')
