--- origsrc/LiTL-litl-0.2/CMakeLists.txt	2025-12-08 17:12:04.549732300 +0900
+++ src/LiTL-litl-0.2/CMakeLists.txt	2025-12-08 17:42:23.749557300 +0900
@@ -29,7 +29,7 @@ if (NOT libm_exist)
     message(FATAL_ERROR "libm was not found.")
 endif()
 
-set (INSTALL_PKGCONFIG_DIR "${CMAKE_INSTALL_PREFIX}/share/pkgconfig")
+set (INSTALL_PKGCONFIG_DIR "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig")
 
 # Subdirectory
 add_subdirectory (src)
--- origsrc/LiTL-litl-0.2/litl.pc.in	2021-12-17 00:51:17.000000000 +0900
+++ src/LiTL-litl-0.2/litl.pc.in	2025-12-08 17:42:41.311406900 +0900
@@ -6,5 +6,5 @@ includedir=@includedir@
 Name: LiTL
 Description: LiTL -- Lightweight Trace Library
 Version: @VERSION@
-Libs: -L${libdir} -llitl
-Cflags: -I${includedir}
+Libs: -llitl
+Cflags: 
--- origsrc/LiTL-litl-0.2/utils/CMakeLists.txt	2025-12-08 17:12:04.556286800 +0900
+++ src/LiTL-litl-0.2/utils/CMakeLists.txt	2025-12-08 17:14:01.696151000 +0900
@@ -6,7 +6,9 @@ add_executable(litl_split litl_split.c
 
 include_directories(
   ${CMAKE_CURRENT_SOURCE_DIR}/include
-  ${CMAKE_CURRENT_BINARY_DIR}/include>
+  ${CMAKE_CURRENT_BINARY_DIR}/include
+  ${CMAKE_BINARY_DIR}/src
+  ${CMAKE_SOURCE_DIR}/src
   )
 
 target_link_libraries( litl_print  PRIVATE   litl  )
