--- origsrc/terminal-pong-1.0.0/src/ncurses.cpp	2020-06-04 03:40:27.000000000 +0900
+++ src/terminal-pong-1.0.0/src/ncurses.cpp	2025-05-14 19:50:06.092867800 +0900
@@ -34,12 +34,12 @@ void ncurses::erase() { ::erase(); }
 void ncurses::drawBox(ncurses::window_t *win) { box(win, 0, 0); };
 
 void ncurses::draw(uint32_t y, uint32_t x, const std::string &str) {
-  mvprintw(y, x, str.c_str());
+  mvprintw(y, x, "%s", str.c_str());
 }
 
 void ncurses::draw(ncurses::window_t *win, uint32_t y, uint32_t x,
                    const std::string &str) {
-  mvwprintw(win, y, x, str.c_str());
+  mvwprintw(win, y, x, "%s", str.c_str());
 }
 
 void ncurses::refresh() { ::refresh(); }
