--- origsrc/statsrelay-upstream-1.6.8/Makefile.am	2015-03-26 05:11:02.000000000 +0900
+++ src/statsrelay-upstream-1.6.8/Makefile.am	2019-12-12 22:21:15.157740900 +0900
@@ -1,2 +1,2 @@
 SUBDIRS=src
-dist_doc_DATA=README.md LICENSE*
+# dist_doc_DATA=README.md LICENSE*
--- origsrc/statsrelay-upstream-1.6.8/src/log.c	2015-03-26 05:11:02.000000000 +0900
+++ src/statsrelay-upstream-1.6.8/src/log.c	2019-12-12 22:20:15.577471200 +0900
@@ -55,7 +55,7 @@ void stats_vlog(const char *prefix,
 
 	if (g_verbose == 1) {
 		if (prefix != NULL) {
-			fprintf(stderr, prefix);
+			fprintf(stderr, "%s", prefix);
 		}
 		total_written = 0;
 		while (total_written < fmt_len) {
