--- origsrc/homest-1.3/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/homest-1.3/Makefile.am	2011-06-12 14:59:45.452324500 +0900
@@ -0,0 +1,12 @@
+lib_LTLIBRARIES = libhomest.la
+libhomest_la_SOURCES = calc_2Dhomog_coeffs.c homest.c lqs.c linalg.c norm.c buckets.c
+libhomest_la_LDFLAGS = -no-undefined
+libhomest_la_LIBADD = @LAPACK_LIBS@
+
+homest_includedir = $(includedir)/homest
+homest_include_HEADERS = homest.h maplefuncs.h lqs.h util.h
+
+noinst_PROGRAMS = homest_demo
+homest_demo_SOURCES = homest_demo.c
+homest_demo_LDADD = libhomest.la
+
--- origsrc/homest-1.3/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/homest-1.3/configure.ac	2011-06-12 14:33:49.548480200 +0900
@@ -0,0 +1,31 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([homest], [1.3], [http://www.ics.forth.gr/~lourakis/homest/])
+AC_CONFIG_SRCDIR([buckets.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_MAKE_SET
+LT_INIT([disable-static])
+
+# Checks for libraries.
+PKG_CHECK_MODULES(LAPACK, lapack)
+AC_CHECK_LIB([f2c], [main])
+AC_CHECK_LIB([levmar], [dlevmar_der])
+
+# Checks for header files.
+AC_CHECK_HEADERS([float.h limits.h malloc.h stdlib.h string.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_C_INLINE
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([sqrt])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
--- origsrc/homest-1.3/homest.c	2009-07-30 20:10:15.000000000 +0900
+++ src/homest-1.3/homest.c	2011-06-12 14:41:53.853869000 +0900
@@ -27,7 +27,11 @@
 #include "maplefuncs.h"
 
 #include "lqs.h"
+#if defined(__CYGWIN__)
+#include <levmar.h>
+#else
 #include <lm.h>
+#endif
 
 #define USE_BUCKETS
 
