--- origsrc/ValyriaTear/src/CMakeLists.txt	2019-09-14 00:35:59.000000000 +0900
+++ src/ValyriaTear/src/CMakeLists.txt	2026-06-20 16:26:15.261825000 +0900
@@ -112,6 +112,9 @@ IF (NOT DISABLE_TRANSLATIONS)
     ELSEIF (CMAKE_SYSTEM_NAME STREQUAL SunOS)
         # explicit linking to libintl is required on Solaris
         SET(EXTRA_LIBRARIES intl)
+    ELSEIF (CYGWIN)
+        FIND_PACKAGE(LibIntl REQUIRED)
+        FIND_PACKAGE(Iconv REQUIRED)
     ENDIF()
 ENDIF()
 
--- origsrc/ValyriaTear/src/common/app_settings.cpp	2019-07-29 19:49:37.000000000 +0900
+++ src/ValyriaTear/src/common/app_settings.cpp	2026-06-20 16:26:15.275950200 +0900
@@ -50,7 +50,7 @@ static void _CopyOldSaveFiles(const std:
         return;
     }
 
-#if (defined(__linux__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(SOLARIS))
+#if (defined(__linux__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(SOLARIS) || defined(__CYGWIN__))
     // We assume the old save files are in ~/.valyriatear
     passwd *pw = getpwuid(getuid());
     if(!pw)
@@ -104,7 +104,7 @@ static void _CopyOldSettingsFile(const s
         return;
     }
 
-#if (defined(__linux__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(SOLARIS))
+#if (defined(__linux__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(SOLARIS) || defined(__CYGWIN__))
     // We assume the old settings.lua file is in ~/.valyriatear
     passwd *pw = getpwuid(getuid());
     if(!pw)
--- origsrc/ValyriaTear/src/common/global/shop/shop_data.h	2019-07-29 19:49:37.000000000 +0900
+++ src/ValyriaTear/src/common/global/shop/shop_data.h	2026-06-20 16:26:15.288059500 +0900
@@ -11,6 +11,7 @@
 #ifndef __SHOP_DATA_HEADER__
 #define __SHOP_DATA_HEADER__
 
+#include <cstdint>
 #include <map>
 
 namespace vt_global {
--- origsrc/ValyriaTear/src/engine/system.h	2019-07-29 19:49:37.000000000 +0900
+++ src/ValyriaTear/src/engine/system.h	2026-06-20 16:26:15.298581900 +0900
@@ -31,6 +31,10 @@
 #include <set>
 #include <map>
 
+#if defined(DEFAULT_LOCALE)
+#undef DEFAULT_LOCALE
+#endif
+
 namespace vt_mode_manager {
 class GameMode;
 }
--- origsrc/ValyriaTear/src/engine/video/image_base.cpp	2019-07-29 19:49:37.000000000 +0900
+++ src/ValyriaTear/src/engine/video/image_base.cpp	2026-06-20 16:42:25.518505500 +0900
@@ -26,6 +26,7 @@
 #include <SDL2/SDL_image.h>
 #include <SDL2/SDL_endian.h>
 #include <png.h>
+#include <cstring>
 
 using namespace vt_utils;
 
