--- origsrc/libmemcached-1.1.4/CMakeLists.txt	2023-03-06 17:47:30.000000000 +0900
+++ src/libmemcached-1.1.4/CMakeLists.txt	2025-08-13 22:22:19.862142800 +0900
@@ -6,7 +6,7 @@ endif()
 
 include(CMakeVersions.txt)
 
-project(libmemcached-awesome
+project(libmemcached
         VERSION "${LIBMEMCACHED_VERSION}"
         DESCRIPTION "libmemcached-awesome, a C/C++ memcached client library"
         )
--- origsrc/libmemcached-1.1.4/src/p9y/CMakeLists.txt	2023-03-06 17:47:30.000000000 +0900
+++ src/libmemcached-1.1.4/src/p9y/CMakeLists.txt	2025-08-13 19:36:58.780668500 +0900
@@ -20,6 +20,7 @@ target_include_directories(p9y PRIVATE
     ${CMAKE_BINARY_DIR}/include
     )
 
+if (FALSE)
 install(TARGETS p9y EXPORT p9y-targets
         RUNTIME COMPONENT bin DESTINATION ${CMAKE_INSTALL_BINDIR}
         ARCHIVE COMPONENT lib DESTINATION ${CMAKE_INSTALL_LIBDIR}
@@ -30,6 +31,7 @@ export(EXPORT p9y-targets NAMESPACE libm
 install(EXPORT p9y-targets NAMESPACE libmemcached::
         COMPONENT dev
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
+endif()
 
 if(NOT HAVE_GETOPT_H)
     target_sources(p9y PRIVATE getopt.c)
