--- origsrc/gsignond-1.2.0/src/common/gsignond-log.h	2018-10-31 00:57:31.000000000 +0900
+++ src/gsignond-1.2.0/src/common/gsignond-log.h	2025-12-12 05:07:24.703180200 +0900
@@ -27,7 +27,9 @@
 
 #include <glib.h>
 
+#if !defined(__CYGWIN__)
 #include <execinfo.h>
+#endif
 #include <stdio.h>
 #include <stdlib.h>
 
--- origsrc/gsignond-1.2.0/src/gplugind/main.c	2018-10-31 00:57:31.000000000 +0900
+++ src/gsignond-1.2.0/src/gplugind/main.c	2025-12-12 05:07:24.706029800 +0900
@@ -30,7 +30,9 @@
 #include <glib-unix.h>
 #include <glib.h>
 #include <gio/gio.h>
+#if !defined(__CYGWIN__)
 #include <sys/prctl.h>
+#endif
 
 #include <gsignond.h>
 #include "gsignond-dbus.h"
@@ -86,8 +88,10 @@ _install_sighandlers (GMainLoop *main_lo
                            NULL);
     _sig_source_id[2] = g_source_attach (source, ctx);
 
+#if !defined(__CYGWIN__)
     if (prctl(PR_SET_PDEATHSIG, SIGHUP))
         WARN ("failed to set parent death signal");
+#endif
 }
 
 static const gchar* _plugin_path(void)
--- origsrc/gsignond-1.2.0/src/plugins/digest/meson.build	2018-10-31 00:57:31.000000000 +0900
+++ src/gsignond-1.2.0/src/plugins/digest/meson.build	2025-12-12 05:07:24.709061300 +0900
@@ -9,4 +9,5 @@ digest_plugin = shared_module('digest',
     digest_files,
     dependencies : [glib, gio, gio_unix, gmodule, gsignond_common],
     install: true,
-    install_dir : gplugins_dir)
+    install_dir : gplugins_dir,
+    name_prefix : '')
--- origsrc/gsignond-1.2.0/src/plugins/password/meson.build	2018-10-31 00:57:31.000000000 +0900
+++ src/gsignond-1.2.0/src/plugins/password/meson.build	2025-12-12 05:07:24.711071400 +0900
@@ -9,7 +9,8 @@ password_plugin = shared_module('passwor
     password_files,
     dependencies : [glib, gio, gio_unix, gmodule, gsignond_common],
     install: true,
-    install_dir : gplugins_dir)
+    install_dir : gplugins_dir,
+    name_prefix : '')
 
 password_plugin_dep = declare_dependency(
     sources: password_files,
--- origsrc/gsignond-1.2.0/src/plugins/ssotest/meson.build	2018-10-31 00:57:31.000000000 +0900
+++ src/gsignond-1.2.0/src/plugins/ssotest/meson.build	2025-12-12 05:07:24.711071400 +0900
@@ -7,7 +7,8 @@ ssotest_plugin = shared_module('ssotest'
     ssotest_files,
     dependencies : [glib, gio, gio_unix, gmodule, gsignond_common],
     install: true,
-    install_dir : gplugins_dir
+    install_dir : gplugins_dir,
+    name_prefix : ''
 )
 
 ssotest_plugin_dep = declare_dependency(
