--- origsrc/NuML-1.1.6/libnuml/CMakeLists.txt	2022-04-19 17:09:02.000000000 +0900
+++ src/NuML-1.1.6/libnuml/CMakeLists.txt	2022-05-07 20:17:39.193536300 +0900
@@ -271,7 +271,11 @@ endif()
 
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules;${CMAKE_MODULE_PATH}")
 # install find modules, so they can readily be used by others
-install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules/FindLIBNUML.cmake" DESTINATION share/cmake/Modules)
+if (CYGWIN)
+    install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules/FindLIBNUML.cmake" DESTINATION lib/cmake/NuML)
+else()
+    install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules/FindLIBNUML.cmake" DESTINATION share/cmake/Modules)
+endif()
 
 ###############################################################################
 #
@@ -609,8 +613,9 @@ set(DOCUMENTATION_FILES
         VERSION.txt
 )
 
-install(FILES ${DOCUMENTATION_FILES} DESTINATION ${MISC_PREFIX}.)
-
+if (NOT CYGWIN)
+    install(FILES ${DOCUMENTATION_FILES} DESTINATION ${MISC_PREFIX}.)
+endif()
 
 ###############################################################################
 #
