--- origsrc/itl/Makefile.am	2020-02-07 10:26:56.000000000 +0900
+++ src/itl/Makefile.am	2020-02-07 10:27:53.407895100 +0900
@@ -1,7 +1,7 @@
 SUBDIRS=$(AM_SUBDIRS)
 lib_LTLIBRARIES = libitl.la
 libitl_la_CFLAGS = 
-libitl_la_LDFLAGS = -version-info $(LTVERSION)
+libitl_la_LDFLAGS = -no-undefined -version-info $(LTVERSION)
 libitl_la_LIBADD= -lm
 libitl_la_SOURCES = hijri/hijri.c prayertime/astro.c prayertime/prayer.c hijri/umm_alqura.c
 libitl_la_HEADERS = hijri/hijri.h prayertime/prayer.h
@@ -11,9 +11,9 @@ pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = @PACKAGE@.pc
 
 check_PROGRAMS = demo_hijri demo_prayer
-demo_hijri_LDADD= -litl -lm
+demo_hijri_LDADD= libitl.la -lm
 demo_hijri_SOURCES = hijri/demo_hijri.c
-demo_prayer_LDADD= -litl -lm
+demo_prayer_LDADD= libitl.la -lm
 demo_prayer_SOURCES = prayertime/demo_prayer.c
 
 TESTS=demo_prayer demo_hijri
