diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/ribosome/CMakeLists.txt src/ribosome/CMakeLists.txt
--- origsrc/ribosome/CMakeLists.txt	2015-12-28 17:31:39.000000000 +0900
+++ src/ribosome/CMakeLists.txt	2015-12-28 17:33:09.958017400 +0900
@@ -34,5 +34,5 @@ add_subdirectory(src)
 FILE(GLOB headers
 	"${PROJECT_SOURCE_DIR}/include/ribosome/*.hpp"
 )
-install(FILES "${PROJECT_BINARY_DIR}/cmake/RibosomeConfig.cmake" DESTINATION share/ribosome/cmake)
+install(FILES "${PROJECT_BINARY_DIR}/cmake/RibosomeConfig.cmake" DESTINATION lib/cmake/ribosome)
 install(FILES ${headers} DESTINATION include/ribosome)
