--- origsrc/edflib_121/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/edflib_121/Makefile.am	2022-05-17 09:53:22.616320700 +0900
@@ -0,0 +1,23 @@
+
+check_PROGRAMS = test_generator sine_generator sweep_generator test_edflib unit_test
+test_generator_SOURCES = test_generator.c
+test_generator_LDADD = libedf.la
+sine_generator_SOURCES = sine_generator.c
+sine_generator_LDADD = libedf.la
+sweep_generator_SOURCES = sweep_generator.c
+sweep_generator_LDADD = libedf.la
+test_edflib_SOURCES = test_edflib.c
+test_edflib_LDADD = libedf.la
+unit_test_SOURCES = unittest/unittest.c
+unit_test_LDADD = libedf.la
+
+TESTS = unit_test
+
+include_HEADERS = edflib.h
+
+lib_LTLIBRARIES = libedf.la
+libedf_la_SOURCES = edflib.c
+libedf_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/edflib_121/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/edflib_121/configure.ac	2022-05-17 09:59:54.507166000 +0900
@@ -0,0 +1,28 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([edflib], [1.21])
+AC_CONFIG_SRCDIR([edflib.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE([subdir-objects])
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+
+# Checks for library functions.
+AC_FUNC_FSEEKO
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([memset])
+
+AC_CONFIG_FILES([edflib.pc Makefile])
+AC_OUTPUT
--- origsrc/edflib_121/edflib.c	2022-01-14 03:04:48.000000000 +0900
+++ src/edflib_121/edflib.c	2022-05-17 08:51:45.075088500 +0900
@@ -38,7 +38,7 @@
 #define EDFLIB_VERSION  (121)
 #define EDFLIB_MAXFILES  (64)
 
-#if defined(__APPLE__) || defined(__MACH__) || defined(__APPLE_CC__) || defined(__HAIKU__)
+#if defined(__APPLE__) || defined(__MACH__) || defined(__APPLE_CC__) || defined(__HAIKU__) || defined(__CYGWIN__)
 
 #define fopeno fopen
 
--- origsrc/edflib_121/edflib.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/edflib_121/edflib.pc.in	2022-05-17 08:51:45.079123000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: edflib
+Description: Programming library for C/C++ to read/write EDF+/BDF+ files
+Version: @VERSION@
+Libs: -L${libdir} -ledf
+Cflags: -I${includedir}
