--- origsrc/libnetconf2-2.1.11/CMakeLists.txt	2022-05-04 22:50:45.000000000 +0900
+++ src/libnetconf2-2.1.11/CMakeLists.txt	2022-05-27 18:53:39.747251600 +0900
@@ -290,7 +290,7 @@ configure_file(nc_server.h.in nc_server.
 configure_file(nc_version.h.in nc_version.h)
 
 # install library
-install(TARGETS netconf2 DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS netconf2 RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 # install headers
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/nc_client.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
--- origsrc/libnetconf2-2.1.11/CMakeModules/FindLibSSH.cmake	2022-05-04 22:50:45.000000000 +0900
+++ src/libnetconf2-2.1.11/CMakeModules/FindLibSSH.cmake	2022-05-27 18:44:41.465271000 +0900
@@ -55,6 +55,7 @@ else()
         ssh.so
         libssh.so
         libssh.dylib
+        libssh.dll.a
         PATHS
         /usr/lib
         /usr/local/lib
