--- origsrc/libnetconf2-3.5.5/CMakeLists.txt	2025-01-09 23:24:28.000000000 +0900
+++ src/libnetconf2-3.5.5/CMakeLists.txt	2025-03-09 12:55:42.021440200 +0900
@@ -339,7 +339,7 @@ file(COPY ${headers} DESTINATION "${PROJ
 install(DIRECTORY "${PROJECT_SOURCE_DIR}/modules/" DESTINATION ${YANG_MODULE_DIR} FILES_MATCHING PATTERN "*.yang")
 
 # install library
-install(TARGETS netconf2 DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS netconf2 RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 # install headers
 install(FILES ${PROJECT_BINARY_DIR}/include/nc_client.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
--- origsrc/libnetconf2-3.5.5/CMakeModules/FindLibSSH.cmake	2025-01-09 23:24:28.000000000 +0900
+++ src/libnetconf2-3.5.5/CMakeModules/FindLibSSH.cmake	2025-03-09 12:55:42.037122800 +0900
@@ -55,6 +55,7 @@ else()
         ssh.so
         libssh.so
         libssh.dylib
+        libssh.dll.a
         PATHS
         /usr/lib
         /usr/local/lib
--- origsrc/libnetconf2-3.5.5/tests/CMakeLists.txt	2025-01-09 23:24:28.000000000 +0900
+++ src/libnetconf2-3.5.5/tests/CMakeLists.txt	2025-03-09 12:55:42.037122800 +0900
@@ -89,9 +89,11 @@ include_directories(${CMAKE_SOURCE_DIR}/
 configure_file("${PROJECT_SOURCE_DIR}/tests/config.h.in" "${PROJECT_BINARY_DIR}/tests/config.h" ESCAPE_QUOTES @ONLY)
 
 # compile PAM test module
+if (LibPAM_FOUND)
 add_library(pam_netconf SHARED ${CMAKE_SOURCE_DIR}/tests/pam/pam_netconf.c)
 set_target_properties(pam_netconf PROPERTIES PREFIX "")
 target_link_libraries(pam_netconf ${LIBPAM_LIBRARIES})
+endif()
 
 # generate PAM configuration file
 file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/netconf.conf
