--- origsrc/libgsecuredelete-0.3/gsecuredelete/Makefile.am	2016-07-15 05:51:32.000000000 +0900
+++ src/libgsecuredelete-0.3/gsecuredelete/Makefile.am	2017-10-05 10:53:40.937345200 +0900
@@ -15,7 +15,7 @@ libgsecuredelete_la_includedir=$(include
 libgsecuredelete_la_VALAFLAGS = $(AM_VALAFLAGS) \
                                 --library=gsecuredelete -H gsecuredelete.h \
                                 --vapidir=. --pkg=config
-libgsecuredelete_la_LDFLAGS = $(AM_LDFLAGS) \
+libgsecuredelete_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) \
                               -version-info @GSD_LTVERSION@
 libgsecuredelete_la_SOURCES = gsd-async-operation.vala \
                               gsd-delete-operation.vala \
