--- origsrc/libgvectors-0.1/src/Makefile.am	2007-08-22 23:21:45.000000000 +0900
+++ src/libgvectors-0.1/src/Makefile.am	2012-04-28 22:47:48.332024400 +0900
@@ -8,4 +8,4 @@ visualvector.cc vectorstore.cc vstoreite
 vstorefuncs.cc binaryfilewriter.cc xmlfilewriter.cc
 
 libgvectors_la_CXXFLAGS = -I$(top_srcdir)/gvectors $(GFILEIO_CFLAGS) $(LIBXML_CFLAGS) $(AM_CXXFLAGS)
-libgvectors_la_LDFLAGS = $(GFILEIO_LIBS) $(LIBXML_LIBS) $(AM_LDFLAGS)
+libgvectors_la_LDFLAGS = $(GFILEIO_LIBS) $(LIBXML_LIBS) $(AM_LDFLAGS) -no-undefined
