--- origsrc/blitz-1.0.2/src/CMakeLists.txt	2019-10-02 07:50:17.000000000 +0900
+++ src/blitz-1.0.2/src/CMakeLists.txt	2026-05-25 17:09:27.346581300 +0900
@@ -9,7 +9,7 @@ set_target_properties(blitz PROPERTIES
                       CLEAN_DIRECT_OUTPUT 1)
 
 set(BLITZ_LIBS blitz)
-if (NOT WIN32)
+if (NOT WIN32 AND NOT CYGWIN)
     add_library(blitz-static STATIC ${blitz_LIB_SRCS})
     add_dependencies(blitz-static generated-headers)
     set(BLITZ_LIBS ${BLITZ_LIBS} blitz-static)
@@ -21,7 +21,7 @@ endif()
 
 install(TARGETS ${BLITZ_LIBS}
         EXPORT  ${PROJECT_NAME}Targets
-        #RUNTIME DESTINATION bin COMPONENT runtime
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT runtime
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT devel
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT devel
 )
