--- origsrc/sha-1.0.4b/Makefile.am	2012-10-28 01:11:41.000000000 +0900
+++ src/sha-1.0.4b/Makefile.am	2014-12-21 16:43:50.086940400 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = foreign
 bin_PROGRAMS = sha
-noinst_PROGRAMS = shatest
+check_PROGRAMS = shatest
 lib_LTLIBRARIES = libsha.la
 sha_SOURCES = sha.c 
 sha_LDADD = libsha.la
@@ -11,3 +11,6 @@ libsha_la_SOURCES = sha1.c sha256.c sha3
 	version.h
 man_MANS = sha.1
 EXTRA_DIST = $(man_MANS) README.SHA LICENSE
+libsha_la_LDFLAGS = -no-undefined
+TESTS = shatest
+pkginclude_HEADERS = sha1.h sha256.h sha384.h sha512.h version.h
