--- origsrc/sha-1.0.4b/Makefile.am	2012-10-28 01:11:41.000000000 +0900
+++ src/sha-1.0.4b/Makefile.am	2012-11-17 13:47:55.823484100 +0900
@@ -11,3 +11,7 @@ libsha_la_SOURCES = sha1.c sha256.c sha3
 	version.h
 man_MANS = sha.1
 EXTRA_DIST = $(man_MANS) README.SHA LICENSE
+libsha_la_LDFLAGS = -no-undefined
+TESTS = shatest
+sha_includedir = $(includedir)/sha
+sha_include_HEADERS = sha1.h sha256.h sha384.h sha512.h version.h
