--- origsrc/libipt-2.0.1/libipt/CMakeLists.txt	2019-08-13 17:39:44.000000000 +0900
+++ src/libipt-2.0.1/libipt/CMakeLists.txt	2019-10-03 08:44:13.673480100 +0900
@@ -99,10 +99,10 @@ configure_file(
 )
 
 set_target_properties(libipt PROPERTIES
-  PREFIX ""
   PUBLIC_HEADER ${CMAKE_CURRENT_BINARY_DIR}/include/intel-pt.h
   VERSION   ${PT_VERSION}
   SOVERSION ${PT_VERSION_MAJOR}
+  OUTPUT_NAME ipt
 )
 
 install(TARGETS libipt
--- origsrc/libipt-2.0.1/ptdump/CMakeLists.txt	2019-08-13 17:39:44.000000000 +0900
+++ src/libipt-2.0.1/ptdump/CMakeLists.txt	2019-10-03 08:44:13.689106200 +0900
@@ -52,3 +52,5 @@ target_link_libraries(ptdump libipt)
 if (SIDEBAND)
   target_link_libraries(ptdump libipt-sb)
 endif (SIDEBAND)
+
+install(TARGETS ptdump DESTINATION bin)
--- origsrc/libipt-2.0.1/pttc/CMakeLists.txt	2019-08-13 17:39:44.000000000 +0900
+++ src/libipt-2.0.1/pttc/CMakeLists.txt	2019-10-03 08:44:13.704732000 +0900
@@ -66,3 +66,5 @@ target_link_libraries(pttc libipt)
 if (PEVENT)
   target_link_libraries(pttc pevent)
 endif (PEVENT)
+
+install(TARGETS pttc DESTINATION bin)
--- origsrc/libipt-2.0.1/sideband/CMakeLists.txt	2019-08-13 17:39:44.000000000 +0900
+++ src/libipt-2.0.1/sideband/CMakeLists.txt	2019-10-03 08:44:13.720352400 +0900
@@ -55,13 +55,18 @@ configure_file(
 )
 
 set_target_properties(libipt-sb PROPERTIES
-  PREFIX ""
   PUBLIC_HEADER ${CMAKE_CURRENT_BINARY_DIR}/include/libipt-sb.h
   VERSION   ${PT_VERSION}
   SOVERSION ${PT_VERSION_MAJOR}
+  OUTPUT_NAME ipt-sb
 )
 
 target_link_libraries(libipt-sb libipt)
 if (PEVENT)
   target_link_libraries(libipt-sb pevent)
 endif (PEVENT)
+
+install(TARGETS libipt-sb
+  RUNTIME DESTINATION bin
+  ARCHIVE DESTINATION lib
+  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
