--- origsrc/libipt-2.0.5/libipt/CMakeLists.txt	2022-01-25 21:22:01.000000000 +0900
+++ src/libipt-2.0.5/libipt/CMakeLists.txt	2022-05-19 21:14:15.789339900 +0900
@@ -97,10 +97,10 @@ configure_file(
 )
 
 set_target_properties(libipt PROPERTIES
-  PREFIX ""
   PUBLIC_HEADER ${CMAKE_CURRENT_BINARY_DIR}/include/intel-pt.h
   VERSION   ${PT_VERSION}
   SOVERSION ${PT_VERSION_MAJOR}
+  OUTPUT_NAME ipt
 )
 
 install(TARGETS libipt
--- origsrc/libipt-2.0.5/ptdump/CMakeLists.txt	2022-01-25 21:22:01.000000000 +0900
+++ src/libipt-2.0.5/ptdump/CMakeLists.txt	2022-05-19 21:14:15.793307900 +0900
@@ -52,3 +52,5 @@ target_link_libraries(ptdump libipt)
 if (SIDEBAND)
   target_link_libraries(ptdump libipt-sb)
 endif (SIDEBAND)
+
+install(TARGETS ptdump DESTINATION bin)
--- origsrc/libipt-2.0.5/pttc/CMakeLists.txt	2022-01-25 21:22:01.000000000 +0900
+++ src/libipt-2.0.5/pttc/CMakeLists.txt	2022-05-19 21:14:15.799375400 +0900
@@ -66,3 +66,5 @@ target_link_libraries(pttc libipt)
 if (PEVENT)
   target_link_libraries(pttc pevent)
 endif (PEVENT)
+
+install(TARGETS pttc DESTINATION bin)
--- origsrc/libipt-2.0.5/sideband/CMakeLists.txt	2022-01-25 21:22:01.000000000 +0900
+++ src/libipt-2.0.5/sideband/CMakeLists.txt	2022-05-19 21:14:15.806375400 +0900
@@ -55,13 +55,18 @@ configure_file(
 )
 
 set_target_properties(libipt-sb PROPERTIES
-  PREFIX ""
   PUBLIC_HEADER ${CMAKE_CURRENT_BINARY_DIR}/include/libipt-sb.h
   VERSION   ${PT_VERSION}
   SOVERSION ${PT_VERSION_MAJOR}
+  OUTPUT_NAME ipt-sb
 )
 
 target_link_libraries(libipt-sb libipt)
 if (PEVENT)
   target_link_libraries(libipt-sb pevent)
 endif (PEVENT)
+
+install(TARGETS libipt-sb
+  RUNTIME DESTINATION bin
+  ARCHIVE DESTINATION lib
+  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
