--- origsrc/libipt-2.2/libipt/CMakeLists.txt	2025-11-20 19:55:47.000000000 +0900
+++ src/libipt-2.2/libipt/CMakeLists.txt	2026-08-09 08:02:00.526061300 +0900
@@ -98,13 +98,12 @@ configure_file(
 )
 
 set_target_properties(libipt PROPERTIES
-  PREFIX ""
-  IMPORT_PREFIX ""
   PUBLIC_HEADER ${CMAKE_CURRENT_BINARY_DIR}/include/intel-pt.h
   VERSION   ${PT_VERSION}
   SOVERSION ${PT_VERSION_MAJOR}
   C_STANDARD 11
   C_STANDARD_REQUIRED ON
+  OUTPUT_NAME ipt
 )
 
 install(TARGETS libipt
--- origsrc/libipt-2.2/sideband/CMakeLists.txt	2025-11-20 19:55:47.000000000 +0900
+++ src/libipt-2.2/sideband/CMakeLists.txt	2026-08-09 08:14:55.746182800 +0900
@@ -56,13 +56,12 @@ configure_file(
 )
 
 set_target_properties(libipt-sb PROPERTIES
-  PREFIX ""
-  IMPORT_PREFIX ""
   PUBLIC_HEADER ${CMAKE_CURRENT_BINARY_DIR}/include/libipt-sb.h
   VERSION   ${PT_VERSION}
   SOVERSION ${PT_VERSION_MAJOR}
   C_STANDARD 11
   C_STANDARD_REQUIRED ON
+  OUTPUT_NAME ipt-sb
 )
 
 target_link_libraries(libipt-sb libipt)
