--- origsrc/processor-trace-1.6.1/CMakeLists.txt	2017-05-31 20:30:32.000000000 +0900
+++ src/processor-trace-1.6.1/CMakeLists.txt	2017-10-05 12:57:01.506000000 +0900
@@ -261,3 +261,10 @@ endif (PTT)
 if (MAN)
   add_subdirectory(doc/man)
 endif (MAN)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libipt.pc.in ${CMAKE_BINARY_DIR}/libipt.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libipt.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/processor-trace-1.6.1/libipt.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/processor-trace-1.6.1/libipt.pc.in	2017-10-05 12:57:11.842000000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libipt
+Description: Intel(R) Processor Trace decoder library
+Version: @PT_VERSION@
+Libs: -L${libdir} -lipt
+Cflags: -I${includedir}
--- origsrc/processor-trace-1.6.1/ptdump/CMakeLists.txt	2017-05-31 20:30:32.000000000 +0900
+++ src/processor-trace-1.6.1/ptdump/CMakeLists.txt	2017-10-05 12:50:15.936000200 +0900
@@ -49,3 +49,5 @@ add_executable(ptdump
 )
 
 target_link_libraries(ptdump libipt)
+
+install(TARGETS ptdump DESTINATION bin)
--- origsrc/processor-trace-1.6.1/pttc/CMakeLists.txt	2017-05-31 20:30:32.000000000 +0900
+++ src/processor-trace-1.6.1/pttc/CMakeLists.txt	2017-10-05 12:50:56.560000000 +0900
@@ -62,3 +62,5 @@ add_executable(pttc
 )
 
 target_link_libraries(pttc libipt)
+
+install(TARGETS pttc DESTINATION bin)
