--- origsrc/viewglob-2.0.4/Makefile.am	2006-04-27 02:18:05.000000000 +0900
+++ src/viewglob-2.0.4/Makefile.am	2013-10-05 22:16:12.247138700 +0900
@@ -11,19 +11,19 @@ man_MANS = vgseer.1 viewglob.1 $(MANS_FU
 viewglob.1:
 	rm -f viewglob.1.tmp
 	sed -e 's/@RELEASE_DATE\@/$(RELEASE_DATE)/g' \
-		viewglob.1.in > viewglob.1.tmp
+		$(srcdir)/viewglob.1.in > viewglob.1.tmp
 	mv viewglob.1.tmp viewglob.1
 
 vgseer.1:
 	rm -f vgseer.1.tmp
 	sed -e 's/@RELEASE_DATE\@/$(RELEASE_DATE)/g' \
-		vgseer.1.in > vgseer.1.tmp
+		$(srcdir)/vgseer.1.in > vgseer.1.tmp
 	mv vgseer.1.tmp vgseer.1
 
 vgd.1:
 	rm -f vgd.1.tmp
 	sed -e 's/@RELEASE_DATE\@/$(RELEASE_DATE)/g' \
-		vgd.1.in > vgd.1.tmp
+		$(srcdir)/vgd.1.in > vgd.1.tmp
 	mv vgd.1.tmp vgd.1
 
 EXTRA_DIST = COPYING2 HACKING viewglob.1.in vgseer.1.in vgd.1.in
--- origsrc/viewglob-2.0.4/shell/Makefile.am	2006-04-27 02:18:04.000000000 +0900
+++ src/viewglob-2.0.4/shell/Makefile.am	2013-10-05 22:14:50.072334000 +0900
@@ -10,27 +10,27 @@ viewglob:
 		-e 's,@PACKAGE_NAME\@,$(PACKAGE_NAME),g' \
 		-e 's,@PACKAGE_VERSION\@,$(PACKAGE_VERSION),g' \
 		-e 's/@RELEASE_DATE\@/$(RELEASE_DATE)/g' \
-		viewglob.in > viewglob.tmp
+		$(srcdir)/viewglob.in > viewglob.tmp
 	chmod a+x viewglob.tmp
 	mv viewglob.tmp viewglob
 
 getopt.sh:
 	rm -f getopt.sh.tmp
 	sed -e 's,@BASH_FULL_PATH\@,$(BASH_FULL_PATH),g' \
-		getopt.sh.in > getopt.sh.tmp
+		$(srcdir)/getopt.sh.in > getopt.sh.tmp
 	chmod a+x getopt.sh.tmp
 	mv getopt.sh.tmp getopt.sh
 
 .zshrc:
 	rm -f init-viewglob.zshrc.tmp
 	sed -e 's,@pkglibdir\@,$(pkglibdir),g' \
-		init-viewglob.zshrc.in > init-viewglob.zshrc.tmp
+		$(srcdir)/init-viewglob.zshrc.in > init-viewglob.zshrc.tmp
 	mv init-viewglob.zshrc.tmp .zshrc
 
 init-viewglob.bashrc:
 	rm -f init-viewglob.bashrc.tmp
 	sed -e 's,@pkglibdir\@,$(pkglibdir),g' \
-		init-viewglob.bashrc.in > init-viewglob.bashrc.tmp
+		$(srcdir)/init-viewglob.bashrc.in > init-viewglob.bashrc.tmp
 	mv init-viewglob.bashrc.tmp init-viewglob.bashrc
 
 
--- origsrc/viewglob-2.0.4/vgd/Makefile.am	2006-04-27 03:18:59.000000000 +0900
+++ src/viewglob-2.0.4/vgd/Makefile.am	2013-10-05 22:15:19.918037100 +0900
@@ -25,7 +25,7 @@ BUILT_SOURCES = vgd-usage.h
 
 vgd-usage.h: vgd-usage.txt
 	rm -f vgd-usage.tmp
-	sed <vgd-usage.txt >vgd-usage.tmp \
+	sed < $(srcdir)/vgd-usage.txt >vgd-usage.tmp \
 		-e 's/\\/\\\\/g' \
 		-e 's/"/\\"/g' \
 		-e 's/.*/\"&\\n\"/'
--- origsrc/viewglob-2.0.4/vgseer/Makefile.am	2006-04-27 03:13:43.000000000 +0900
+++ src/viewglob-2.0.4/vgseer/Makefile.am	2013-10-05 22:13:47.420966800 +0900
@@ -29,7 +29,7 @@ BUILT_SOURCES = vgseer-usage.h
 
 vgseer-usage.h: vgseer-usage.txt
 	rm -f vgseer-usage.tmp
-	sed <vgseer-usage.txt >vgseer-usage.tmp \
+	sed < $(srcdir)/vgseer-usage.txt >vgseer-usage.tmp \
 		-e 's/\\/\\\\/g' \
 		-e 's/"/\\"/g' \
 		-e 's/.*/\"&\\n\"/'
