--- origsrc/viewglob/Makefile.am	2024-09-26 14:33:39.000000000 +0900
+++ src/viewglob/Makefile.am	2024-09-26 14:38:34.058820600 +0900
@@ -11,19 +11,19 @@ man_MANS = vgseer.1 viewglob.1 $(MANS_FU
 viewglob.1:
 	rm -f viewglob.1.tmp
 	sed -e 's/@RELEASE_DATE\@/$(RELEASE_DATE)/g' \
-		viewglob.1.in > viewglob.1.tmp
+		$(srcdir)/viewglob.1.in > viewglob.1.tmp
 	mv viewglob.1.tmp viewglob.1
 
 vgseer.1:
 	rm -f vgseer.1.tmp
 	sed -e 's/@RELEASE_DATE\@/$(RELEASE_DATE)/g' \
-		vgseer.1.in > vgseer.1.tmp
+		$(srcdir)/vgseer.1.in > vgseer.1.tmp
 	mv vgseer.1.tmp vgseer.1
 
 vgd.1:
 	rm -f vgd.1.tmp
 	sed -e 's/@RELEASE_DATE\@/$(RELEASE_DATE)/g' \
-		vgd.1.in > vgd.1.tmp
+		$(srcdir)/vgd.1.in > vgd.1.tmp
 	mv vgd.1.tmp vgd.1
 
 EXTRA_DIST = COPYING2 HACKING viewglob.1.in vgseer.1.in vgd.1.in
--- origsrc/viewglob/shell/Makefile.am	2024-09-26 14:33:39.000000000 +0900
+++ src/viewglob/shell/Makefile.am	2024-09-26 15:04:14.799009200 +0900
@@ -1,7 +1,8 @@
 
 bin_SCRIPTS = viewglob
-pkglib_SCRIPTS = getopt.sh conf-to-args.sh
-pkglib_DATA = init-viewglob.bashrc .zshrc
+mybindir = $(pkglibdir)
+mybin_SCRIPTS = getopt.sh conf-to-args.sh
+mybin_DATA = init-viewglob.bashrc .zshrc
 
 viewglob:
 	rm -f viewglob.tmp
@@ -10,27 +11,27 @@ viewglob:
 		-e 's,@PACKAGE_NAME\@,$(PACKAGE_NAME),g' \
 		-e 's,@PACKAGE_VERSION\@,$(PACKAGE_VERSION),g' \
 		-e 's/@RELEASE_DATE\@/$(RELEASE_DATE)/g' \
-		viewglob.in > viewglob.tmp
+		$(srcdir)/viewglob.in > viewglob.tmp
 	chmod a+x viewglob.tmp
 	mv viewglob.tmp viewglob
 
 getopt.sh:
 	rm -f getopt.sh.tmp
 	sed -e 's,@BASH_FULL_PATH\@,$(BASH_FULL_PATH),g' \
-		getopt.sh.in > getopt.sh.tmp
+		$(srcdir)/getopt.sh.in > getopt.sh.tmp
 	chmod a+x getopt.sh.tmp
 	mv getopt.sh.tmp getopt.sh
 
 .zshrc:
 	rm -f init-viewglob.zshrc.tmp
 	sed -e 's,@pkglibdir\@,$(pkglibdir),g' \
-		init-viewglob.zshrc.in > init-viewglob.zshrc.tmp
+		$(srcdir)/init-viewglob.zshrc.in > init-viewglob.zshrc.tmp
 	mv init-viewglob.zshrc.tmp .zshrc
 
 init-viewglob.bashrc:
 	rm -f init-viewglob.bashrc.tmp
 	sed -e 's,@pkglibdir\@,$(pkglibdir),g' \
-		init-viewglob.bashrc.in > init-viewglob.bashrc.tmp
+		$(srcdir)/init-viewglob.bashrc.in > init-viewglob.bashrc.tmp
 	mv init-viewglob.bashrc.tmp init-viewglob.bashrc
 
 
--- origsrc/viewglob/vgd/Makefile.am	2024-09-26 14:33:39.000000000 +0900
+++ src/viewglob/vgd/Makefile.am	2024-09-26 14:38:34.058820600 +0900
@@ -25,7 +25,7 @@ BUILT_SOURCES = vgd-usage.h
 
 vgd-usage.h: vgd-usage.txt
 	rm -f vgd-usage.tmp
-	sed <vgd-usage.txt >vgd-usage.tmp \
+	sed < $(srcdir)/vgd-usage.txt >vgd-usage.tmp \
 		-e 's/\\/\\\\/g' \
 		-e 's/"/\\"/g' \
 		-e 's/.*/\"&\\n\"/'
--- origsrc/viewglob/vgdisplay/Makefile.am	2024-09-26 14:33:39.000000000 +0900
+++ src/viewglob/vgdisplay/Makefile.am	2024-09-26 15:13:14.071642400 +0900
@@ -6,7 +6,8 @@ LDADD = \
 	@X_PRE_LIBS@ @X_LIBS@ -lX11 @X_EXTRA_LIBS@ \
 	@GTK_LIBS@
 
-pkglib_PROGRAMS = vgclassic vgmini
+mybindir = $(pkglibdir)
+mybin_PROGRAMS = vgclassic vgmini
 
 vgclassic_SOURCES = \
 	vgclassic.c \
@@ -68,33 +69,35 @@ EXTRA_DIST = \
 app_icons.h:
 	rm -f app_icons.tmp app_icons.h
 	gdk-pixbuf-csource --raw --name=icon_16x16_inline \
-		icon_16x16.png > app_icons.tmp
+		$(srcdir)/icon_16x16.png > app_icons.tmp
 	gdk-pixbuf-csource --raw --name=icon_24x24_inline \
-		icon_24x24.png >> app_icons.tmp
+		$(srcdir)/icon_24x24.png >> app_icons.tmp
 	gdk-pixbuf-csource --raw --name=icon_32x32_inline \
-		icon_32x32.png >> app_icons.tmp
+		$(srcdir)/icon_32x32.png >> app_icons.tmp
 	gdk-pixbuf-csource --raw --name=icon_36x36_inline \
-		icon_36x36.png >> app_icons.tmp
+		$(srcdir)/icon_36x36.png >> app_icons.tmp
 	mv app_icons.tmp app_icons.h
 
 # These are mostly old standard Gnome icons.
 file_icons.h:
 	rm -f file_icons.tmp file_icons.h
 	gdk-pixbuf-csource --raw --name=regular_inline \
-		gnome-fs-regular.png >> file_icons.tmp
+		$(srcdir)/gnome-fs-regular.png >> file_icons.tmp
 	gdk-pixbuf-csource --raw --name=executable_inline \
-		gnome-fs-executable.png >> file_icons.tmp
+		$(srcdir)/gnome-fs-executable.png >> file_icons.tmp
 	gdk-pixbuf-csource --raw --name=directory_inline \
-		gnome-fs-directory.png >> file_icons.tmp
+		$(srcdir)/gnome-fs-directory.png >> file_icons.tmp
 	gdk-pixbuf-csource --raw --name=blockdev_inline \
-		gnome-fs-blockdev.png >> file_icons.tmp
+		$(srcdir)/gnome-fs-blockdev.png >> file_icons.tmp
 	gdk-pixbuf-csource --raw --name=chardev_inline \
-		gnome-fs-chardev.png >> file_icons.tmp
+		$(srcdir)/gnome-fs-chardev.png >> file_icons.tmp
 	gdk-pixbuf-csource --raw --name=fifo_inline \
-		gnome-fs-fifo.png >> file_icons.tmp
+		$(srcdir)/gnome-fs-fifo.png >> file_icons.tmp
 	gdk-pixbuf-csource --raw --name=symlink_inline \
-		gnome-dev-symlink.png >> file_icons.tmp
+		$(srcdir)/gnome-dev-symlink.png >> file_icons.tmp
 	gdk-pixbuf-csource --raw --name=socket_inline \
-		gnome-fs-socket.png >> file_icons.tmp
+		$(srcdir)/gnome-fs-socket.png >> file_icons.tmp
 	mv file_icons.tmp file_icons.h
 
+display-common.o : app_icons.h
+file_box.o: file_icons.h
--- origsrc/viewglob/vgexpand/Makefile.am	2024-09-26 14:33:39.000000000 +0900
+++ src/viewglob/vgexpand/Makefile.am	2024-09-26 15:03:08.118640100 +0900
@@ -2,7 +2,8 @@ COMMON_DIR = $(top_srcdir)/common
 AM_CPPFLAGS = @GLIB_CFLAGS@ -DVG_LIB_DIR="\"$(pkglibdir)\"" -I$(COMMON_DIR)
 LDADD = @GLIB_LIBS@
 
-pkglib_PROGRAMS = vgexpand
+mybindir = $(pkglibdir)
+mybin_PROGRAMS = vgexpand
 
 vgexpand_SOURCES = vgexpand.c
 
--- origsrc/viewglob/vgping/Makefile.am	2024-09-26 14:33:39.000000000 +0900
+++ src/viewglob/vgping/Makefile.am	2024-09-26 15:03:27.200055300 +0900
@@ -1,6 +1,7 @@
 COMMON_DIR = $(top_srcdir)/common
 
-pkglib_PROGRAMS = vgping
+mybindir = $(pkglibdir)
+mybin_PROGRAMS = vgping
 
 vgping_CPPFLAGS = @GLIB_CFLAGS@ -DVG_LIB_DIR="\"$(pkglibdir)\"" -I$(COMMON_DIR)
 vgping_LDADD = @GLIB_LIBS@ @LIBS@
--- origsrc/viewglob/vgseer/Makefile.am	2024-09-26 14:33:39.000000000 +0900
+++ src/viewglob/vgseer/Makefile.am	2024-09-26 14:38:34.058820600 +0900
@@ -29,7 +29,7 @@ BUILT_SOURCES = vgseer-usage.h
 
 vgseer-usage.h: vgseer-usage.txt
 	rm -f vgseer-usage.tmp
-	sed <vgseer-usage.txt >vgseer-usage.tmp \
+	sed < $(srcdir)/vgseer-usage.txt >vgseer-usage.tmp \
 		-e 's/\\/\\\\/g' \
 		-e 's/"/\\"/g' \
 		-e 's/.*/\"&\\n\"/'
--- origsrc/viewglob/vgseer/vgseer.c	2024-09-26 14:33:39.000000000 +0900
+++ src/viewglob/vgseer/vgseer.c	2024-09-26 15:07:38.147002500 +0900
@@ -276,13 +276,13 @@ static gint connect_to_vgd(gchar* server
 	if (STREQ(value, "ERROR")) {
 		/* Print reason for error and exit. */
 		get_param_verify(fd, &param, &value, P_REASON, NULL);
-		g_critical(value);
+		g_critical("%s", value);
 		clean_fail(NULL);
 	}
 	else if (STREQ(value, "WARNING")) {
 		/* Print warning but continue. */
 		get_param_verify(fd, &param, &value, P_REASON, NULL);
-		g_warning(value);
+		g_warning("%s", value);
 	}
 	else if (!STREQ(value, "OK")) {
 		g_critical("Unknown value for P_STATUS: %s", value);
