--- origsrc/mantis-0.1/CMakeLists.txt	2019-06-07 01:11:26.000000000 +0900
+++ src/mantis-0.1/CMakeLists.txt	2026-05-21 21:51:00.130970200 +0900
@@ -4,18 +4,11 @@
 #
 
 cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
-project(mantis VERSION 0.2 LANGUAGES C CXX)
+project(mantis VERSION 0.1 LANGUAGES C CXX)
 if (NOT CMAKE_BUILD_TYPE)
    set (CMAKE_BUILD_TYPE "Release")
 endif()
 
-# We require C++17
-set(CMAKE_CXX_STANDARD 17)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_C_STANDARD 11)
-set(CMAKE_C_STANDARD_REQUIRED ON)
-
-
 set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads REQUIRED)
 
--- origsrc/mantis-0.1/include/canonicalKmer.h	2019-06-07 01:11:26.000000000 +0900
+++ src/mantis-0.1/include/canonicalKmer.h	2026-05-21 23:33:57.981195200 +0900
@@ -5,6 +5,8 @@
 #ifndef MANTIS_CANONICALKMER_H
 #define MANTIS_CANONICALKMER_H
 
+#include <sys/types.h>
+#include <string>
 #include <map>
 
 namespace dna {
--- origsrc/mantis-0.1/include/mantisconfig.hpp	2019-06-07 01:11:26.000000000 +0900
+++ src/mantis-0.1/include/mantisconfig.hpp	2026-05-22 16:31:15.034204200 +0900
@@ -5,9 +5,9 @@
 
 namespace mantis{
     constexpr char major_version[] = "0";
-    constexpr char minor_version[] = "2";
+    constexpr char minor_version[] = "1";
     constexpr char patch_version[] = "0";
-    constexpr char version[] = "0.2.0";
+    constexpr char version[] = "0.1.0";
     constexpr uint32_t index_version = 0;
     constexpr char meta_file_name[] = "/meta_info.json";
     constexpr char CQF_FILE[] = "dbg_cqf.ser";
--- origsrc/mantis-0.1/src/canonicalKmer.cc	2019-06-07 01:11:26.000000000 +0900
+++ src/mantis-0.1/src/canonicalKmer.cc	2026-05-21 23:33:50.666676000 +0900
@@ -5,6 +5,7 @@
 #include <assert.h>
 
 #include "canonicalKmer.h"
+#include <cstdlib>
 
 #define BITMASK(nbits) ((nbits) == 64 ? 0xffffffffffffffff : (1ULL << (nbits)) - 1ULL)
 
