--- origsrc/gupnp-gupnp-1.4.3/meson.build	2022-01-14 02:52:47.000000000 +0900
+++ src/gupnp-gupnp-1.4.3/meson.build	2025-12-13 23:23:53.199694700 +0900
@@ -14,7 +14,6 @@ conf.set('HAVE_IFADDRS_H', ifaddrs_avail
 conf.set('HAVE_LINUX_WIRELESS_H', cc.has_header('linux/wireless.h'))
 
 glib_version = '2.66'
-add_project_arguments(cc.get_supported_arguments('-Werror=deprecated-declarations'), language: 'c')
 conf.set('GLIB_VERSION_MIN_REQUIRED', 'GLIB_VERSION_' + glib_version.underscorify())
 conf.set('GLIB_VERSION_MAX_ALLOWED', 'GLIB_VERSION_' + glib_version.underscorify())
 
--- origsrc/gupnp-gupnp-1.4.3/subprojects/gssdp-1.2.wrap	2022-01-14 02:52:47.000000000 +0900
+++ src/gupnp-gupnp-1.4.3/subprojects/gssdp-1.2.wrap	2025-12-13 23:22:07.103716300 +0900
@@ -3,6 +3,6 @@ url = https://gitlab.gnome.org/GNOME/gss
 revision = gssdp-1.4
 depth = 1
  
-[provides]
+[provide]
 dependency_name = gssdp-1.2
 
