--- origsrc/gupnp-0.20.12/configure.ac	2014-05-29 21:51:43.000000000 +0900
+++ src/gupnp-0.20.12/configure.ac	2014-06-07 10:49:17.950697000 +0900
@@ -31,7 +31,7 @@ LT_INIT
 AC_MSG_CHECKING([operating system])
 os_win32=no
 case "$target_os" in
-     mingw*|cygwin*)
+     mingw*)
         os_win32=yes
         AC_MSG_RESULT(Win32)
         UUID_LIBS=
--- origsrc/gupnp-0.20.12/libgupnp/Makefile.am	2014-05-29 21:51:43.000000000 +0900
+++ src/gupnp-0.20.12/libgupnp/Makefile.am	2014-06-07 10:49:17.950697000 +0900
@@ -5,7 +5,6 @@ CONTEXT_MANAGER_IMPL = gupnp-windows-con
 		       gupnp-windows-context-manager.h
 CONTEXT_MANAGER_CFLAGS =
 CONTEXT_MANAGER_LIBS = -lws2_32 -liphlpapi
-WIN32_LDFLAGS = -no-undefined
 else
 if HAVE_IFADDRS_H
 CONTEXT_MANAGER_IMPL = gupnp-unix-context-manager.c \
@@ -26,6 +25,8 @@ endif
 endif
 endif
 
+WIN32_LDFLAGS = -no-undefined
+
 if HAVE_NETLINK
 CONTEXT_MANAGER_IMPL += \
 	gupnp-linux-context-manager.c \
--- origsrc/gupnp-0.20.12/vala/Makefile.am	2014-05-29 21:47:53.000000000 +0900
+++ src/gupnp-0.20.12/vala/Makefile.am	2014-06-07 10:49:17.966322000 +0900
@@ -8,7 +8,7 @@ gupnp-1.0.stamp: $(top_builddir)/libgupn
 		$(srcdir)/gupnp-1.0-custom.vala
 	$(AM_V_GEN)$(VAPIGEN) --metadatadir=$(srcdir) \
 				  --pkg=libsoup-2.4 \
-			      --library=gupnp-1.0 --pkg=gio-2.0 --pkg=libxml-2.0 \
+			      --library=gupnp-1.0 --pkg=gio-2.0 --pkg=libxml-2.0 --pkg=gssdp-1.0 \
 			      $(top_builddir)/libgupnp/GUPnP-1.0.gir \
 				  $(srcdir)/gupnp-1.0-custom.vala && \
 	echo timestmap > $(@F)
