--- origsrc/gupnp-0.19.4/configure.ac	2012-12-05 22:36:25.000000000 +0900
+++ src/gupnp-0.19.4/configure.ac	2013-01-13 14:05:07.281442700 +0900
@@ -34,7 +34,7 @@ AM_CONDITIONAL(HAVE_GTEST, [test x$gtest
 AC_MSG_CHECKING([operating system])
 os_win32=no
 case "$target_os" in
-     mingw*|cygwin*)
+     mingw*)
         os_win32=yes
         AC_MSG_RESULT(Win32)
         UUID=
@@ -139,7 +139,7 @@ GOBJECT_INTROSPECTION_CHECK([0.6.4])
 # vapigen
 AS_IF([test "x$found_introspection" = "xyes"],
       [
-       VALA_PROG_VAPIGEN([0.14])
+       VAPIGEN_CHECK
       ]
 )
 AM_CONDITIONAL([HAVE_VAPIGEN], [test "x$VAPIGEN" != "x"])
--- origsrc/gupnp-0.19.4/libgupnp/Makefile.am	2012-10-27 22:17:37.000000000 +0900
+++ src/gupnp-0.19.4/libgupnp/Makefile.am	2013-01-13 14:00:15.995309900 +0900
@@ -5,8 +5,8 @@ CONTEXT_MANAGER_IMPL = gupnp-windows-con
 		       gupnp-windows-context-manager.h
 CONTEXT_MANAGER_CFLAGS =
 CONTEXT_MANAGER_LIBS = -lws2_32 -liphlpapi
-WIN32_LDFLAGS = -no-undefined
 else
+WIN32_LDFLAGS = -no-undefined
 CONTEXT_MANAGER_IMPL = gupnp-unix-context-manager.c \
 					   gupnp-unix-context-manager.h
 if USE_NETWORK_MANAGER
--- origsrc/gupnp-0.19.4/vala/Makefile.am	2012-10-27 03:58:55.000000000 +0900
+++ src/gupnp-0.19.4/vala/Makefile.am	2013-01-13 14:00:43.198434900 +0900
@@ -8,7 +8,7 @@ gupnp-1.0.stamp: $(top_builddir)/libgupn
 		$(srcdir)/gupnp-1.0-custom.vala
 	$(AM_V_GEN)$(VAPIGEN) --metadatadir=$(srcdir) \
 				  --pkg=libsoup-2.4 \
-			      --library=gupnp-1.0 --pkg=gio-2.0 --pkg=libxml-2.0 \
+			      --library=gupnp-1.0 --pkg=gio-2.0 --pkg=libxml-2.0 --pkg=gssdp-1.0 \
 			      $(top_builddir)/libgupnp/GUPnP-1.0.gir \
 				  $(srcdir)/gupnp-1.0-custom.vala && \
 	echo timestmap > $(@F)
