--- origsrc/nyan-0.3.1/CMakeLists.txt	2024-11-07 03:25:58.000000000 +0900
+++ src/nyan-0.3.1/CMakeLists.txt	2025-01-11 23:19:35.782467100 +0900
@@ -184,3 +184,9 @@ message("
 ")
 
 set_property(GLOBAL PROPERTY TARGET_MESSAGES OFF)
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/nyan.pc.in ${CMAKE_BINARY_DIR}/nyan.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/nyan.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/nyan-0.3.1/nyan/util.cpp	2024-11-07 03:25:58.000000000 +0900
+++ src/nyan-0.3.1/nyan/util.cpp	2025-01-11 23:11:51.384347300 +0900
@@ -13,7 +13,7 @@
 #include <streambuf>
 #include <string>
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 	#include <array>
 	#include <mutex>
 
@@ -135,7 +135,7 @@ std::string addr_to_string(const void *a
 std::string symbol_name(const void *addr,
                         bool require_exact_addr,
                         bool no_pure_addrs) {
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 	auto symbol_name_result = symbol_name_win(addr, require_exact_addr);
 
 	if (!symbol_name_result.empty()) {
--- origsrc/nyan-0.3.1/nyan.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/nyan-0.3.1/nyan.pc.in	2025-01-11 23:20:03.861139300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: nyan
+Description: Typesafe hierarchical key-value database with inheritance and dynamic patching
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lnyan
+Cflags: -I${includedir}
--- origsrc/nyan-0.3.1/nyan_version	2024-11-07 03:25:58.000000000 +0900
+++ src/nyan-0.3.1/nyan_version	2025-01-11 23:00:41.442921000 +0900
@@ -1 +1 @@
-0.3.0
+0.3.1
