--- origsrc/opkg-0.3.5/libopkg/Makefile.am	2017-06-16 02:13:19.000000000 +0900
+++ src/opkg-0.3.5/libopkg/Makefile.am	2017-11-30 14:56:35.951976900 +0900
@@ -71,4 +71,4 @@ libopkg_la_LIBADD = $(LIBARCHIVE_LIBS) \
 		    $(CURL_LIBS) $(GPGME_LIBS) $(GPGERR_LIBS) $(OPENSSL_LIBS) \
 		    $(PATHFINDER_LIBS) $(SOLVER_LIBS)
 
-libopkg_la_LDFLAGS = -version-info 1:0:0
+libopkg_la_LDFLAGS = -no-undefined -version-info 1:0:0
