--- origsrc/opkg-0.8.0/libopkg/Makefile.am	2025-01-11 02:28:24.000000000 +0900
+++ src/opkg-0.8.0/libopkg/Makefile.am	2025-03-23 04:01:04.566910100 +0900
@@ -67,4 +67,4 @@ libopkg_la_LIBADD = $(LIBARCHIVE_LIBS) \
 		    $(CURL_LIBS) $(GPGME_LIBS) $(GPGERR_LIBS) $(OPENSSL_LIBS) \
 		    $(SOLVER_LIBS)
 
-libopkg_la_LDFLAGS = -version-info 1:0:0
+libopkg_la_LDFLAGS = -no-undefined -version-info 1:0:0
