--- origsrc/opkg/libopkg/Makefile.am	2011-05-26 09:51:50.000000000 +0900
+++ src/opkg/libopkg/Makefile.am	2013-02-07 23:13:35.357891300 +0900
@@ -2,7 +2,7 @@
 AM_CFLAGS=-Wall -DHOST_CPU_STR=\"@host_cpu@\" -DBUILD_CPU=@build_cpu@ -DLIBDIR=\"@libdir@\" -DOPKGLIBDIR=\"@opkglibdir@\" -DOPKGETCDIR=\"@opkgetcdir@\" -DOPKGLOCKFILE=\"@opkglockfile@\" -DDATADIR=\"@datadir@\" -I$(top_srcdir) $(BIGENDIAN_CFLAGS) $(CURL_CFLAGS) $(GPGME_CFLAGS) $(PATHFINDER_CFLAGS)
 
 libopkg_includedir=$(includedir)/libopkg
-libopkg_include_HEADERS= *.h
+libopkg_include_HEADERS= $(srcdir)/*.h
 
 
 opkg_libcore_sources = \
@@ -46,7 +46,7 @@ libopkg_la_SOURCES = \
 
 libopkg_la_LIBADD = $(top_builddir)/libbb/libbb.la $(CURL_LIBS) $(GPGME_LIBS) $(OPENSSL_LIBS) $(PATHFINDER_LIBS)
 
-libopkg_la_LDFLAGS = -version-info 1:0:0
+libopkg_la_LDFLAGS = -version-info 1:0:0 -no-undefined
 
 # make sure we only export symbols that are for public use
 #libopkg_la_LDFLAGS = -export-symbols-regex "^opkg_.*"
