--- origsrc/opkg-0.3.3/libopkg/Makefile.am	2016-07-14 02:05:19.000000000 +0900
+++ src/opkg-0.3.3/libopkg/Makefile.am	2016-12-06 00:07:17.888288700 +0900
@@ -70,4 +70,4 @@ libopkg_la_LIBADD = $(LIBARCHIVE_LIBS) \
 		    $(CURL_LIBS) $(GPGME_LIBS) $(GPGERR_LIBS) $(OPENSSL_LIBS) \
 		    $(PATHFINDER_LIBS) $(SOLVER_LIBS)
 
-libopkg_la_LDFLAGS = -version-info 1:0:0
+libopkg_la_LDFLAGS = -no-undefined -version-info 1:0:0
