--- origsrc/opkg-0.4.0/libopkg/Makefile.am	2018-12-27 04:35:55.000000000 +0900
+++ src/opkg-0.4.0/libopkg/Makefile.am	2019-01-20 09:21:26.489255100 +0900
@@ -71,4 +71,4 @@ libopkg_la_LIBADD = $(LIBARCHIVE_LIBS) \
 		    $(CURL_LIBS) $(GPGME_LIBS) $(GPGERR_LIBS) $(OPENSSL_LIBS) \
 		    $(PATHFINDER_LIBS) $(SOLVER_LIBS)
 
-libopkg_la_LDFLAGS = -version-info 1:0:0
+libopkg_la_LDFLAGS = -no-undefined -version-info 1:0:0
