--- origsrc/opkg-0.2.3/libopkg/Makefile.am	2014-09-06 00:59:41.000000000 +0900
+++ src/opkg-0.2.3/libopkg/Makefile.am	2014-11-10 00:28:36.321821100 +0900
@@ -87,7 +87,7 @@ libopkg_la_SOURCES = \
 
 libopkg_la_LIBADD = $(top_builddir)/libbb/libbb.la $(CURL_LIBS) $(GPGME_LIBS) $(OPENSSL_LIBS) $(PATHFINDER_LIBS)
 
-libopkg_la_LDFLAGS = -version-info 1:0:0
+libopkg_la_LDFLAGS = -no-undefined -version-info 1:0:0
 
 # make sure we only export symbols that are for public use
 #libopkg_la_LDFLAGS = -export-symbols-regex "^opkg_.*"
