--- origsrc/seqio-1.2.2/fmtseq.c	1996-07-10 11:58:39.000000000 +0900
+++ src/seqio-1.2.2/fmtseq.c	2013-07-07 16:16:02.933171000 +0900
@@ -37,8 +37,6 @@
 #include <sys/stat.h>
 #include "seqio.h"
 
-extern char *sys_errlist[];
-
 
 #ifdef WIN32
 char dirch = '\\';
--- origsrc/seqio-1.2.2/grepseq.c	1996-07-10 10:57:06.000000000 +0900
+++ src/seqio-1.2.2/grepseq.c	2013-07-07 16:14:34.261296000 +0900
@@ -18,8 +18,6 @@
 #include <string.h>
 #include "seqio.h"
 
-extern char *sys_errlist[];
-
 #define ASCII 4
 
 /*
--- origsrc/seqio-1.2.2/seqio.c	1996-07-10 11:58:40.000000000 +0900
+++ src/seqio-1.2.2/seqio.c	2013-07-07 16:15:30.245671000 +0900
@@ -147,7 +147,6 @@ static char *get_today();
  * files, and replacement functions for system calls that don't exist on
  * one or more machines.
  */
-extern char *sys_errlist[];
 
 #if defined(__sun) && !defined(FILENAME_MAX)
 #include <sys/param.h>
@@ -158,8 +157,6 @@ extern char *sys_errlist[];
 static int ctype_initflag = 0;
 static char tubuf[384], *tuary;
 
-#define toupper(chr) tuary[(int) (chr)]
-
 static void init_ctype(void)
 {
   int i;
