diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/seqio-1.2.2/fmtseq.c src/seqio-1.2.2/fmtseq.c
--- origsrc/seqio-1.2.2/fmtseq.c	1996-07-10 11:58:39.000000000 +0900
+++ src/seqio-1.2.2/fmtseq.c	2016-03-19 14:20:00.223931200 +0900
@@ -37,8 +37,9 @@
 #include <sys/stat.h>
 #include "seqio.h"
 
+#if !defined(__CYGWIN__)
 extern char *sys_errlist[];
-
+#endif
 
 #ifdef WIN32
 char dirch = '\\';
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/seqio-1.2.2/grepseq.c src/seqio-1.2.2/grepseq.c
--- origsrc/seqio-1.2.2/grepseq.c	1996-07-10 10:57:06.000000000 +0900
+++ src/seqio-1.2.2/grepseq.c	2016-03-19 14:19:13.575485300 +0900
@@ -18,7 +18,9 @@
 #include <string.h>
 #include "seqio.h"
 
+#if !defined(__CYGWIN__)
 extern char *sys_errlist[];
+#endif
 
 #define ASCII 4
 
@@ -1114,4 +1116,3 @@ void *my_malloc(int bufsize)
   memset(s, 0, bufsize);
   return s;
 }
-
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/seqio-1.2.2/seqio.c src/seqio-1.2.2/seqio.c
--- origsrc/seqio-1.2.2/seqio.c	1996-07-10 11:58:40.000000000 +0900
+++ src/seqio-1.2.2/seqio.c	2016-03-19 14:19:36.342687900 +0900
@@ -147,7 +147,9 @@ static char *get_today();
  * files, and replacement functions for system calls that don't exist on
  * one or more machines.
  */
+#if !defined(__CYGWIN__)
 extern char *sys_errlist[];
+#endif
 
 #if defined(__sun) && !defined(FILENAME_MAX)
 #include <sys/param.h>
@@ -158,7 +160,9 @@ extern char *sys_errlist[];
 static int ctype_initflag = 0;
 static char tubuf[384], *tuary;
 
+#if !defined(__CYGWIN__)
 #define toupper(chr) tuary[(int) (chr)]
+#endif
 
 static void init_ctype(void)
 {
@@ -19331,4 +19335,3 @@ static char *get_today()
  
   return buffer;
 }
-  
