diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/png++-0.2.7/config.hpp src/png++-0.2.7/config.hpp
--- origsrc/png++-0.2.7/config.hpp	2015-01-15 22:28:06.000000000 +0900
+++ src/png++-0.2.7/config.hpp	2015-05-28 22:35:44.284604500 +0900
@@ -32,7 +32,7 @@
 #define PNGPP_CONFIG_HPP_INCLUDED
 
 // Endianness test
-#if defined(__GLIBC__)
+#if defined(__GLIBC__) || defined(__CYGWIN__)
 
 #include <endian.h>
 
