--- origsrc/libvisca-1.1.0/visca/Makefile.am	2011-08-21 14:48:08.000000000 +0900
+++ src/libvisca-1.1.0/visca/Makefile.am	2013-01-16 15:34:12.994475700 +0900
@@ -4,7 +4,7 @@ lib_LTLIBRARIES = libvisca.la
 AM_CFLAGS = -I$(top_srcdir)
 EXTRA_DIST = libvisca_avr.c libvisca_win32.c
 
-libvisca_la_LDFLAGS = -version-info @lt_major@:@lt_revision@:@lt_age@
+libvisca_la_LDFLAGS = -version-info @lt_major@:@lt_revision@:@lt_age@ -no-undefined
 
 libvisca_la_SOURCES =  \
 		libvisca.c 		\
--- origsrc/libvisca-1.1.0/visca/libvisca_posix.c	2011-08-21 14:49:13.000000000 +0900
+++ src/libvisca-1.1.0/visca/libvisca_posix.c	2013-01-16 15:37:03.747036200 +0900
@@ -25,6 +25,7 @@
 #include <errno.h>
 #include <unistd.h>
 #include <sys/ioctl.h>
+#include <asm/socket.h>
 
 /* implemented in libvisca.c
  */
