--- origsrc/libvisca-1.0.1/examples/testvisca.c	2003-05-09 00:27:28.000000000 +0900
+++ src/libvisca-1.0.1/examples/testvisca.c	2011-08-03 10:21:03.901699100 +0900
@@ -27,6 +27,10 @@
 #include <termios.h> /* POSIX terminal control definitions */
 #include <sys/ioctl.h>
 
+#if defined(__CYGWIN__)
+#include <asm/socket.h>
+#endif
+
 #include "../src/libvisca.h"
 
 #define EVI_D30
--- origsrc/libvisca-1.0.1/examples/visca_cli.c	2006-04-01 16:31:48.000000000 +0900
+++ src/libvisca-1.0.1/examples/visca_cli.c	2011-08-03 10:20:45.664594700 +0900
@@ -348,6 +348,10 @@ get_md_obj_pos (returns the center posit
 #include <sys/ioctl.h>
 #include <string.h>
 
+#if defined(__CYGWIN__)
+#include <asm/socket.h>
+#endif
+
 #include "../src/libvisca.h"
 
 #define DEBUG 0
--- origsrc/libvisca-1.0.1/src/Makefile.am	2003-05-09 00:27:28.000000000 +0900
+++ src/libvisca-1.0.1/src/Makefile.am	2011-08-03 09:53:17.669737600 +0900
@@ -2,7 +2,7 @@
 MAINTAINERCLEANFILES = Makefile.in
 lib_LTLIBRARIES = libvisca.la
 
-libvisca_la_LDFLAGS = -version-info @lt_major@:@lt_revision@:@lt_age@
+libvisca_la_LDFLAGS = -version-info @lt_major@:@lt_revision@:@lt_age@ -no-undefined
 
 libvisca_la_SOURCES =  \
 		libvisca.c 	\
--- origsrc/libvisca-1.0.1/src/libvisca.c	2007-07-23 20:19:04.000000000 +0900
+++ src/libvisca-1.0.1/src/libvisca.c	2011-08-03 10:19:11.074325100 +0900
@@ -22,6 +22,10 @@
 #include "config.h"
 #include "libvisca.h"
 
+#if defined(__CYGWIN__)
+#include <asm/socket.h>
+#endif
+
 /********************************/
 /*      PRIVATE FUNCTIONS       */
 /********************************/
