--- origsrc/log4cplus/configure.ac	2026-07-27 22:25:32.000000000 +0900
+++ src/log4cplus/configure.ac	2026-07-27 22:26:02.071118800 +0900
@@ -460,6 +460,9 @@ AS_CASE(["$target_os"],
 
 dnl Checks for header files.
 
+m4_ifndef([AC_CHECK_INCLUDES_DEFAULT],
+  [AC_DEFUN([AC_CHECK_INCLUDES_DEFAULT], [])])
+
 AC_CHECK_INCLUDES_DEFAULT
 AC_PROG_EGREP
 
--- origsrc/log4cplus/m4/ax_swig_multi_module_support.m4	2026-07-27 22:25:32.000000000 +0900
+++ src/log4cplus/m4/ax_swig_multi_module_support.m4	2026-07-27 22:26:02.071118800 +0900
@@ -52,5 +52,4 @@
 AU_ALIAS([SWIG_MULTI_MODULE_SUPPORT], [AX_SWIG_MULTI_MODULE_SUPPORT])
 AC_DEFUN([AX_SWIG_MULTI_MODULE_SUPPORT],[
         AC_REQUIRE([AX_PKG_SWIG])
-        SWIG="$SWIG -noruntime"
 ])
