--- origsrc/log4cplus/configure.ac	2024-01-26 12:14:59.000000000 +0900
+++ src/log4cplus/configure.ac	2024-01-26 12:15:17.896455500 +0900
@@ -450,6 +450,9 @@ AS_CASE(["$target_os"],
 
 dnl Checks for header files.
 
+m4_ifndef([AC_CHECK_INCLUDES_DEFAULT],
+  [AC_DEFUN([AC_CHECK_INCLUDES_DEFAULT], [])])
+
 AC_CHECK_INCLUDES_DEFAULT
 AC_PROG_EGREP
 
--- origsrc/log4cplus/m4/ax_swig_multi_module_support.m4	2024-01-26 12:14:59.000000000 +0900
+++ src/log4cplus/m4/ax_swig_multi_module_support.m4	2024-01-26 12:52:40.564693300 +0900
@@ -52,5 +52,5 @@
 AU_ALIAS([SWIG_MULTI_MODULE_SUPPORT], [AX_SWIG_MULTI_MODULE_SUPPORT])
 AC_DEFUN([AX_SWIG_MULTI_MODULE_SUPPORT],[
         AC_REQUIRE([AX_PKG_SWIG])
-        SWIG="$SWIG -noruntime"
+#        SWIG="$SWIG -noruntime"
 ])
