--- origsrc/procinfo-ng-2.0.304/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/procinfo-ng-2.0.304/Makefile.am	2015-05-16 17:34:34.243717000 +0900
@@ -0,0 +1,5 @@
+bin_PROGRAMS = procinfo
+AM_CPPFLAGS = @NCURSES_CFLAGS@
+procinfo_SOURCES = procinfo.cpp
+man_MANS = procinfo.8
+procinfo_LDADD = @NCURSES_LIBS@
--- origsrc/procinfo-ng-2.0.304/configure.in	2009-04-27 12:33:50.000000000 +0900
+++ src/procinfo-ng-2.0.304/configure.in	2015-05-16 17:34:34.259031000 +0900
@@ -4,6 +4,7 @@
 AC_PREREQ(2.61)
 AC_INIT([procinfo-ng], [2.0], [tabrisnet@users.sourceforge.net])
 AC_CONFIG_SRCDIR([procinfo.cpp])
+AM_INIT_AUTOMAKE
 
 # Checks for programs.
 AC_PROG_CC
@@ -34,6 +35,8 @@ AC_FUNC_SELECT_ARGTYPES
 AC_FUNC_STRTOD
 AC_CHECK_FUNCS([select strtol strtoul strtoull snprintf readdir64])
 
+PKG_CHECK_MODULES(NCURSES, ncurses)
+
 AC_ARG_ENABLE(maintainer-mode, [ --enable-maintainer-mode	Enables debug symbols and disables optimizations [default=no]],
 [ enable_maintainer_mode=yes ])
 
