--- origsrc/tagcoll2/configure.ac	2013-10-21 07:05:58.000000000 +0900
+++ src/tagcoll2/configure.ac	2015-02-15 19:19:09.836286500 +0900
@@ -43,6 +43,8 @@ AC_SUBST(LIBTAGCOLL_VERSION_INFO)
 
 dnl TODO: add looking for zlib
 
+PKG_CHECK_MODULES([ZLIB], [zlib])
+
 dnl Add the languages which your application supports here.
 dnl ALL_LINGUAS="fr da es no it"
 dnl ALL_LINGUAS="it"
--- origsrc/tagcoll2/tagcoll/Makefile.am	2013-10-21 07:06:03.000000000 +0900
+++ src/tagcoll2/tagcoll/Makefile.am	2015-02-15 19:20:28.040388000 +0900
@@ -80,8 +80,8 @@ libtagcoll2_la_SOURCES = \
 		SmartHierarchy.cc \
 		TextFormat.cc
 
-libtagcoll2_la_LIBADD = tagexpr/libtagexpr.la
-libtagcoll2_la_LDFLAGS = $(LIBWIBBLE_LIBS) -version-info @LIBTAGCOLL_VERSION_INFO@
+libtagcoll2_la_LIBADD = tagexpr/libtagexpr.la $(ZLIB_LIBS)
+libtagcoll2_la_LDFLAGS = -no-undefined $(LIBWIBBLE_LIBS) -version-info @LIBTAGCOLL_VERSION_INFO@
 
 # Tests
 TESTS_ENVIRONMENT = $(top_srcdir)/tagcoll/runtest
