--- origsrc/libgrapple-0.9.8/configure.in	2009-08-04 20:31:55.000000000 +0900
+++ src/libgrapple-0.9.8/configure.in	2012-12-14 08:43:11.768244700 +0900
@@ -87,12 +87,13 @@ AC_CHECK_HEADER([netdb.h], CFLAGS="$CFLA
 
 AC_CHECK_HEADER([sys/ioctl.h], CFLAGS="$CFLAGS -DHAVE_SYS_IOCTL_H",)
 
+if test "$host_os" != "cygwin"; then
 AC_CHECK_HEADER([winsock2.h], CFLAGS="$CFLAGS -DHAVE_WINSOCK2_H" LIBS="$LIBS -lws2_32",)
 
 AC_CHECK_HEADER([ws2tcpip.h], CFLAGS="$CFLAGS -DHAVE_WS2TCPIP_H",)
 
 AC_CHECK_HEADER([windows.h], CFLAGS="$CFLAGS -DHAVE_WINDOWS_H",)
-
+fi
 AC_CHECK_HEADER([unistd.h], CFLAGS="$CFLAGS -DHAVE_UNISTD_H",)
 
 AC_CHECK_HEADER([ifaddrs.h], CFLAGS="$CFLAGS -DHAVE_IFADDRS_H",)
