--- origsrc/tlsdate-tlsdate-0.0.13/src/util.c	2015-05-29 03:49:40.000000000 +0900
+++ src/tlsdate-tlsdate-0.0.13/src/util.c	2019-10-11 20:28:17.298559600 +0900
@@ -290,7 +290,11 @@ int file_write(int fd, void *buf, size_t
 	ssize_t ret;
 	iov[0].iov_base = buf;
 	iov[0].iov_len = sz;
+#if defined(__CYGWIN__)
+	ret = IGNORE_EINTR (pwrite (fd, buf, sz, 0));
+#else
 	ret = IGNORE_EINTR (pwritev (fd, iov, 1, 0));
+#endif
 	if (ret != sz)
 	{
 		return -1;
@@ -332,7 +336,11 @@ int file_read(int fd, void *buf, size_t
 	struct iovec iov[1];
 	iov[0].iov_base = buf;
 	iov[0].iov_len = sz;
+#if defined(__CYGWIN__)
+  if (pread (fd, buf, sz, 0) != sz)
+#else
 	if (preadv (fd, iov, 1, 0) != sz)
+#endif
 	{
 		/* Returns -1 on read failure */
 		return -1;
