--- origsrc/libvslvm-20240504/Makefile.am	2024-05-04 17:43:21.000000000 +0900
+++ src/libvslvm-20240504/Makefile.am	2025-05-19 22:56:43.089912100 +0900
@@ -3,23 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfvalue \
 	libvslvm \
 	vslvmtools \
 	pyvslvm \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libvslvm-20240504/configure.ac	2024-05-04 17:43:21.000000000 +0900
+++ src/libvslvm-20240504/configure.ac	2025-05-19 22:56:43.089912100 +0900
@@ -170,19 +170,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
 AC_CONFIG_FILES([libvslvm/Makefile])
 AC_CONFIG_FILES([pyvslvm/Makefile])
 AC_CONFIG_FILES([vslvmtools/Makefile])
--- origsrc/libvslvm-20240504/pyvslvm/Makefile.am	2024-05-04 17:43:21.000000000 +0900
+++ src/libvslvm-20240504/pyvslvm/Makefile.am	2025-05-19 22:56:56.749032800 +0900
@@ -49,7 +49,7 @@ pyvslvm_la_LIBADD = \
 	@LIBBFIO_LIBADD@
 
 pyvslvm_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyvslvm_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyvslvm_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
