--- origsrc/cmockery-0.1.2/Makefile.am	2008-09-16 09:18:12.000000000 +0900
+++ src/cmockery-0.1.2/Makefile.am	2014-03-30 23:13:07.486415800 +0900
@@ -46,13 +46,14 @@ noinst_SCRIPTS =
 
 lib_LTLIBRARIES += libcmockery.la
 libcmockery_la_SOURCES = src/config.h src/cmockery.c src/google/cmockery.h
-libcmockery_la_CFLAGS = -Isrc/google
+libcmockery_la_CFLAGS = -I$(top_srcdir)/src/google
+libcmockery_la_LDFLAGS = -no-undefined
 
 noinst_PROGRAMS = calculator
 calculator_SOURCES = src/example/calculator.c src/config.h
-calculator_CFLAGS =
+calculator_CFLAGS =-I$(top_srcdir)/src/google
 
-unit_test_CFLAGS = -Isrc/google -Isrc/example -DUNIT_TESTING=1
+unit_test_CFLAGS = -I$(top_srcdir)/src/google -I$(top_srcdir)/src/example -DUNIT_TESTING=1
 
 noinst_PROGRAMS += calculator_test
 calculator_test_SOURCES = src/example/calculator_test.c \
