--- origsrc/bestline/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/bestline/Makefile.am	2024-11-29 11:11:48.691939900 +0900
@@ -0,0 +1,15 @@
+
+check_PROGRAMS = bestline_example bestline_multi
+bestline_example_SOURCES = example.c
+bestline_example_LDADD = libbestline.la
+bestline_multi_SOURCES = multi.c
+bestline_multi_LDADD = libbestline.la
+
+include_HEADERS = bestline.h
+
+lib_LTLIBRARIES = libbestline.la
+libbestline_la_SOURCES = bestline.c
+libbestline_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/bestline/bestline.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/bestline/bestline.pc.in	2024-11-29 11:11:48.629432000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: bestline
+Description: ANSI Standard X3.64 Teletypewriter Command Session Library
+Version: @VERSION@
+Libs: -L${libdir} -lbestline
+Cflags: -I${includedir}
--- origsrc/bestline/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/bestline/configure.ac	2024-11-29 11:11:48.707539000 +0900
@@ -0,0 +1,34 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.72])
+AC_INIT([bestline], [0])
+AC_CONFIG_SRCDIR([bestline.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_CXX
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([fcntl.h netinet/in.h sys/ioctl.h sys/socket.h sys/time.h termios.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_C_INLINE
+AC_TYPE_MODE_T
+AC_TYPE_SIZE_T
+AC_TYPE_SSIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_MMAP
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([atexit bzero getdelim memchr memmove memset munmap socket strchr strdup strndup strrchr])
+
+AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([bestline.pc])
+AC_OUTPUT
--- origsrc/bestline/multi.c	2024-11-29 09:50:26.000000000 +0900
+++ src/bestline/multi.c	2024-11-29 11:22:36.291134800 +0900
@@ -25,7 +25,11 @@ int udp_open(int port) {
 }
 
 struct sockaddr_in client;
+#if defined(__CYGWIN__)
+socklen_t client_len = sizeof(client);
+#else
 unsigned int client_len = sizeof(client);
+#endif
 
 int udp_read(int fd, void *c, int len) {
     int n = recvfrom(fd, c, len, 0, (struct sockaddr *)&client, &client_len);
