--- origsrc/libfreenect-0.5.6/CMakeLists.txt	2017-07-22 10:58:53.000000000 +0900
+++ src/libfreenect-0.5.6/CMakeLists.txt	2017-10-05 09:02:30.513000000 +0900
@@ -70,6 +70,8 @@ IF(PROJECT_OS_LINUX)
 	OPTION(BUILD_CPACK_TGZ "Build an TGZ using CPack" OFF)
 ENDIF(PROJECT_OS_LINUX)
 
+OPTION(ENABLE_STATIC "Enable static" OFF)
+
 ######################################################################################
 # Dependencies and Definitions
 ######################################################################################
@@ -181,7 +183,7 @@ configure_file(
 
 # --- cmake config file ---
 CONFIGURE_FILE(libfreenectConfig.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/libfreenectConfig.cmake @ONLY)
-INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/libfreenectConfig.cmake DESTINATION share/${PROJECT_NAME})
+INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/libfreenectConfig.cmake DESTINATION lib/cmake/${PROJECT_NAME})
 
 add_custom_target(uninstall
   COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/UninstallTarget.cmake)
--- origsrc/libfreenect-0.5.6/src/CMakeLists.txt	2017-07-22 10:58:53.000000000 +0900
+++ src/libfreenect-0.5.6/src/CMakeLists.txt	2017-10-05 09:03:11.950000000 +0900
@@ -26,18 +26,23 @@ set_target_properties ( freenect PROPERT
   VERSION ${PROJECT_VER}
   SOVERSION ${PROJECT_APIVER})	
 install (TARGETS freenect
-  DESTINATION "${PROJECT_LIBRARY_INSTALL_DIR}")
+  RUNTIME DESTINATION bin
+  ARCHIVE DESTINATION "${PROJECT_LIBRARY_INSTALL_DIR}")
 
-add_library (freenectstatic STATIC ${SRC})
-set_target_properties (freenectstatic PROPERTIES OUTPUT_NAME freenect)
-IF(UNIX AND NOT APPLE)
-  SET_TARGET_PROPERTIES (freenectstatic PROPERTIES COMPILE_FLAGS "-fPIC")
-ENDIF()
-install (TARGETS freenectstatic
-  DESTINATION "${PROJECT_LIBRARY_INSTALL_DIR}")
+if(ENABLE_STATIC)
+  add_library (freenectstatic STATIC ${SRC})
+  set_target_properties (freenectstatic PROPERTIES OUTPUT_NAME freenect)
+  IF(UNIX AND NOT APPLE)
+    SET_TARGET_PROPERTIES (freenectstatic PROPERTIES COMPILE_FLAGS "-fPIC")
+  ENDIF()
+  install (TARGETS freenectstatic
+    DESTINATION "${PROJECT_LIBRARY_INSTALL_DIR}")
+endif()
 
 target_link_libraries (freenect ${LIBUSB_1_LIBRARIES})
-target_link_libraries (freenectstatic ${LIBUSB_1_LIBRARIES})
+if(ENABLE_STATIC)
+  target_link_libraries (freenectstatic ${LIBUSB_1_LIBRARIES})
+endif()
 
 # Install the header files
 install (FILES "../include/libfreenect.h" "../include/libfreenect_registration.h" "../include/libfreenect_audio.h"
--- origsrc/libfreenect-0.5.6/wrappers/c_sync/CMakeLists.txt	2017-07-22 10:58:53.000000000 +0900
+++ src/libfreenect-0.5.6/wrappers/c_sync/CMakeLists.txt	2017-10-05 09:04:41.436000200 +0900
@@ -7,19 +7,26 @@ find_package(Threads REQUIRED)
 include_directories(${THREADS_PTHREADS_INCLUDE_DIR})
 
 add_library (freenect_sync SHARED libfreenect_sync.c)
-add_library (freenect_sync_static STATIC libfreenect_sync.c)
-set_target_properties (freenect_sync_static PROPERTIES OUTPUT_NAME freenect_sync)
+if(ENABLE_STATIC)
+  add_library (freenect_sync_static STATIC libfreenect_sync.c)
+  set_target_properties (freenect_sync_static PROPERTIES OUTPUT_NAME freenect_sync)
+endif()
 
 set_target_properties (freenect_sync PROPERTIES
   VERSION ${PROJECT_VER}
   SOVERSION ${PROJECT_APIVER})
 
 target_link_libraries (freenect_sync freenect ${CMAKE_THREAD_LIBS_INIT})
-target_link_libraries (freenect_sync_static freenect ${CMAKE_THREAD_LIBS_INIT})
+if(ENABLE_STATIC)
+  target_link_libraries (freenect_sync_static freenect ${CMAKE_THREAD_LIBS_INIT})
+endif()
 
 install (TARGETS freenect_sync
-  DESTINATION "${PROJECT_LIBRARY_INSTALL_DIR}")
-install (TARGETS freenect_sync_static
-  DESTINATION "${PROJECT_LIBRARY_INSTALL_DIR}")
+  RUNTIME DESTINATION bin
+  ARCHIVE DESTINATION "${PROJECT_LIBRARY_INSTALL_DIR}")
+if(ENABLE_STATIC)
+  install (TARGETS freenect_sync_static
+    DESTINATION "${PROJECT_LIBRARY_INSTALL_DIR}")
+endif()
 install (FILES "libfreenect_sync.h"
   DESTINATION ${PROJECT_INCLUDE_INSTALL_DIR})
--- origsrc/libfreenect-0.5.6/wrappers/opencv/CMakeLists.txt	2017-07-22 10:58:53.000000000 +0900
+++ src/libfreenect-0.5.6/wrappers/opencv/CMakeLists.txt	2017-10-05 09:05:07.748000000 +0900
@@ -12,7 +12,8 @@ include_directories (../c_sync)
 target_link_libraries (freenect_cv freenect_sync ${OpenCV_LIBS})
 
 install (TARGETS freenect_cv
-  DESTINATION "${PROJECT_LIBRARY_INSTALL_DIR}")
+  RUNTIME DESTINATION bin
+  ARCHIVE DESTINATION "${PROJECT_LIBRARY_INSTALL_DIR}")
 install (FILES "libfreenect_cv.h"
   DESTINATION ${PROJECT_INCLUDE_INSTALL_DIR})
 
