--- origsrc/foliautils-0.23/Makefile.am	2024-12-16 22:12:05.000000000 +0900
+++ src/foliautils-0.23/Makefile.am	2025-01-19 10:25:06.181310400 +0900
@@ -16,3 +16,5 @@ docker-dev:
 deps:
 	./build-deps.sh
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/foliautils-0.23/configure.ac	2024-12-16 22:12:05.000000000 +0900
+++ src/foliautils-0.23/configure.ac	2025-01-19 10:24:41.189911500 +0900
@@ -114,4 +114,5 @@ AC_CONFIG_FILES([
   docs/Makefile
   tests/Makefile
 ])
+AC_CONFIG_FILES([foliautils.pc])
 AC_OUTPUT
--- origsrc/foliautils-0.23/foliautils.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/foliautils-0.23/foliautils.pc.in	2025-01-19 10:24:41.142629800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: foliautils
+Description: Tools for working with the Format for Linguistic Annotation (FoLiA)
+Version: @VERSION@
+Libs: -L${libdir} -lfoliautils
+Cflags: -I${includedir}
--- origsrc/foliautils-0.23/src/Makefile.am	2024-12-16 22:12:05.000000000 +0900
+++ src/foliautils-0.23/src/Makefile.am	2025-01-19 09:52:11.667272000 +0900
@@ -1,5 +1,5 @@
-AM_CPPFLAGS = -I ../include/
-AM_CXXFLAGS = -std=c++17 -W -Wall -pedantic -g -O3
+AM_CPPFLAGS = -I$(top_srcdir)/include/
+AM_CXXFLAGS = -std=gnu++17 -pedantic
 
 bin_PROGRAMS = FoLiA-stats FoLiA-collect FoLiA-idf FoLiA-correct \
 	FoLiA-alto FoLiA-langcat FoLiA-hocr FoLiA-page FoLiA-txt FoLiA-2text \
@@ -31,3 +31,5 @@ FoLiA_merge_SOURCES = FoLiA-merge.cxx
 
 TESTS = tst.sh
 EXTRA_DIST = tst.sh
+
+libfoliautils_la_LDFLAGS += -no-undefined
