--- origsrc/foliautils-0.10/src/Makefile.am	2018-11-29 23:12:55.000000000 +0900
+++ src/foliautils-0.10/src/Makefile.am	2019-03-29 19:12:15.773722100 +0900
@@ -1,4 +1,4 @@
-AM_CXXFLAGS = -std=c++11 -W -Wall -pedantic -g
+AM_CXXFLAGS = -std=gnu++11 -W -Wall -pedantic -g
 
 bin_PROGRAMS = FoLiA-stats FoLiA-collect FoLiA-idf FoLiA-correct \
 	FoLiA-alto FoLiA-langcat FoLiA-hocr FoLiA-page FoLiA-txt FoLiA-2text \
