--- origsrc/libzeitgeist-0.3.10/configure.ac	2011-03-31 21:38:13.000000000 +0900
+++ src/libzeitgeist-0.3.10/configure.ac	2011-05-10 23:01:29.908390400 +0900
@@ -20,7 +20,7 @@ ZEITGEIST_LT_CURRENT=1
 ZEITGEIST_LT_REV=3
 ZEITGEIST_LT_AGE=0
 ZEITGEIST_LT_VERSION="$ZEITGEIST_LT_CURRENT:$ZEITGEIST_LT_REV:$ZEITGEIST_LT_AGE"
-ZEITGEIST_LT_LDFLAGS="-version-info $ZEITGEIST_LT_VERSION -export-symbols-regex '^zeitgeist_.*'"
+ZEITGEIST_LT_LDFLAGS="-no-undefined -version-info $ZEITGEIST_LT_VERSION -export-symbols-regex '^zeitgeist_.*'"
 
 AC_SUBST(ZEITGEIST_LT_VERSION)
 AC_SUBST(ZEITGEIST_LT_LDFLAGS)
@@ -32,7 +32,7 @@ AC_ISC_POSIX
 AC_PROG_CC
 AM_PROG_CC_STDC
 AC_HEADER_STDC
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 
 ####################################################################
--- origsrc/libzeitgeist-0.3.10/src/zeitgeist-log.c	2011-03-31 21:38:13.000000000 +0900
+++ src/libzeitgeist-0.3.10/src/zeitgeist-log.c	2011-05-10 23:02:05.999593900 +0900
@@ -1054,7 +1054,7 @@ _zeitgeist_log_on_zg_vanished (GDBusConn
   self = ZEITGEIST_LOG (user_data);
   priv = ZEITGEIST_LOG_GET_PRIVATE (self);
 
-  g_object_unref (priv->log);
+  if (priv->log) g_object_unref (priv->log);
   priv->log = NULL;
 
   g_object_notify (G_OBJECT (self), "connected");
