--- origsrc/libzeitgeist-0.3.6/configure.ac	2011-03-07 23:25:43.000000000 +0900
+++ src/libzeitgeist-0.3.6/configure.ac	2011-03-17 16:24:13.936794100 +0900
@@ -20,7 +20,7 @@ ZEITGEIST_LT_CURRENT=1
 ZEITGEIST_LT_REV=1
 ZEITGEIST_LT_AGE=0
 ZEITGEIST_LT_VERSION="$ZEITGEIST_LT_CURRENT:$ZEITGEIST_LT_REV:$ZEITGEIST_LT_AGE"
-ZEITGEIST_LT_LDFLAGS="-version-info $ZEITGEIST_LT_VERSION -export-symbols-regex '^zeitgeist_.*'"
+ZEITGEIST_LT_LDFLAGS="-no-undefined -version-info $ZEITGEIST_LT_VERSION -export-symbols-regex '^zeitgeist_.*'"
 
 AC_SUBST(ZEITGEIST_LT_VERSION)
 AC_SUBST(ZEITGEIST_LT_LDFLAGS)
@@ -32,7 +32,7 @@ AC_ISC_POSIX
 AC_PROG_CC
 AM_PROG_CC_STDC
 AC_HEADER_STDC
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 
 ####################################################################
