--- origsrc/UTF8Strings-1.14.0/CMakeLists.txt	2013-05-21 20:17:54.000000000 +0900
+++ src/UTF8Strings-1.14.0/CMakeLists.txt	2014-12-08 09:11:15.806662600 +0900
@@ -5,7 +5,7 @@ add_library (SharedTarget SHARED String.
 
 set_target_properties(SharedTarget PROPERTIES OUTPUT_NAME UTF8Strings-1)
 
-INSTALL(TARGETS SharedTarget DESTINATION lib${LIB_SUFFIX})
+INSTALL(TARGETS SharedTarget RUNTIME DESTINATION bin ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
 INSTALL(FILES Exception.h String.h DESTINATION include/UTF8Strings-1)
 
