--- origsrc/UTF8Strings-1.12.1/CMakeLists.txt	2013-02-23 01:36:08.000000000 +0900
+++ src/UTF8Strings-1.12.1/CMakeLists.txt	2013-02-24 21:25:30.022631900 +0900
@@ -5,7 +5,7 @@ add_library (SharedTarget SHARED String.
 
 set_target_properties(SharedTarget PROPERTIES OUTPUT_NAME UTF8Strings-1)
 
-INSTALL(TARGETS SharedTarget DESTINATION lib${LIB_SUFFIX})
+INSTALL(TARGETS SharedTarget RUNTIME DESTINATION bin ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
 INSTALL(FILES Exception.h String.h DESTINATION include/UTF8Strings-1)
 
