--- origsrc/ZThread-2.3.2/Makefile.am	2014-12-08 09:26:38.943351300 +0900
+++ src/ZThread-2.3.2/Makefile.am	2014-12-08 09:26:41.208976300 +0900
@@ -17,6 +17,8 @@
 ## WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 ## CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
+ACLOCAL_AMFLAGS = -I share
+
 ## extra files
 EXTRA_DIST = \
 BUILDING \
@@ -30,7 +32,7 @@ aclocaldir = $(datadir)/aclocal
 aclocal_DATA = share/zthread.m4 share/pthread.m4
 
 zincludedir = $(includedir)/zthread
-zinclude_HEADERS = include/zthread/*.h
+zinclude_HEADERS = $(top_srcdir)/include/zthread/*.h
 
 distclean-local: 
 	-rm -rf $(top_srcdir)/$(PACKAGE)-$(VERSION).tar.gz
--- origsrc/ZThread-2.3.2/configure.ac	2014-12-08 09:26:38.943351300 +0900
+++ src/ZThread-2.3.2/configure.ac	2014-12-08 09:26:41.208976300 +0900
@@ -19,7 +19,7 @@ dnl CONNECTION WITH THE SOFTWARE OR THE
 
 AC_REVISION($Revision: 1.14 $ZT_VERSION $)
 
-AC_INIT
+AC_INIT([zthread], [2.3.2])
 AC_CONFIG_SRCDIR([README])
 AC_COPYRIGHT([Eric Crahen, 2000-2002])
 
@@ -53,7 +53,7 @@ AC_SUBST(LT_REVISION)
 AC_SUBST(LT_AGE)
 
 
-AM_INIT_AUTOMAKE(ZThread, $ZT_VERSION)
+AM_INIT_AUTOMAKE
 
 AM_CONFIG_HEADER(src/config.h)
 AM_ACLOCAL_INCLUDE(share)
--- origsrc/ZThread-2.3.2/share/atomic-gcc.m4	2005-03-13 10:47:02.000000000 +0900
+++ src/ZThread-2.3.2/share/atomic-gcc.m4	2014-12-08 09:26:41.224601300 +0900
@@ -35,7 +35,7 @@ EOF
 
 atomic_gcc_explicit="no"
 
-AC_DEFUN(AM_ENABLE_ATOMIC_GCC,
+AC_DEFUN([AM_ENABLE_ATOMIC_GCC],
 [AC_ARG_ENABLE(atomic-gcc,
 	AC_HELP_STRING([--enable-atomic-gcc],
 		       [use gcc atomic instructions [default=no]]),
--- origsrc/ZThread-2.3.2/share/atomic-linux.m4	2005-03-13 10:47:13.000000000 +0900
+++ src/ZThread-2.3.2/share/atomic-linux.m4	2014-12-08 09:26:41.224601300 +0900
@@ -39,7 +39,7 @@ EOF
 
 atomic_linux_explicit="no"
 
-AC_DEFUN(AM_ENABLE_ATOMIC_LINUX,
+AC_DEFUN([AM_ENABLE_ATOMIC_LINUX],
 [AC_ARG_ENABLE(atomic-linux,
 	AC_HELP_STRING([--enable-atomic-linux],
 		       [use linux atomic instructions [default=no]]),
--- origsrc/ZThread-2.3.2/share/ftime.m4	2005-03-13 10:47:24.000000000 +0900
+++ src/ZThread-2.3.2/share/ftime.m4	2014-12-08 09:26:41.224601300 +0900
@@ -35,7 +35,7 @@ EOF
 
 have_ftime="detect"
 
-AC_DEFUN(AM_DETECT_FTIME,
+AC_DEFUN([AM_DETECT_FTIME],
 [
 
  AC_ARG_WITH(ftime,
--- origsrc/ZThread-2.3.2/src/Makefile.am	2014-12-08 09:26:38.943351300 +0900
+++ src/ZThread-2.3.2/src/Makefile.am	2014-12-08 09:26:41.224601300 +0900
@@ -24,7 +24,7 @@ SUBDIRS=.
 lib_LTLIBRARIES = libZThread.la
 
 libZThread_la_LIBADD=@LINKER_OPTIONS@ @EXTRA_LINKER_OPTIONS@
-libZThread_la_LDFLAGS= \
+libZThread_la_LDFLAGS= -no-undefined \
         -release $(LT_RELEASE)  \
         -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 
