--- origsrc/disnix-disnix-0.10.2/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/Makefile.am	2023-11-19 23:14:20.902017700 +0900
@@ -1 +1 @@
-SUBDIRS = conf init.d src scripts nix xsl doc maintenance
+SUBDIRS = conf init.d src scripts nix xsl # doc maintenance
--- origsrc/disnix-disnix-0.10.2/scripts/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/scripts/Makefile.am	2023-11-19 23:13:23.614714500 +0900
@@ -9,16 +9,16 @@ bin_SCRIPTS = disnix-instantiate \
 	disnix-convert \
 	disnix-tmpfile
 pkgdata_SCRIPTS = checks
-noinst_DATA = disnix-instantiate.1.xml \
-	disnix-manifest.1.xml \
-	disnix-gendist-roundrobin.1.xml \
-	disnix-delegate.1.xml \
-	disnix-env.1.xml \
-	disnix-ssh-client.1.xml \
-	disnix-reconstruct.1.xml \
-	disnix-normalize-infra.1.xml \
-	disnix-convert.1.xml \
-	disnix-tmpfile.1.xml
+# noinst_DATA = disnix-instantiate.1.xml \
+#	disnix-manifest.1.xml \
+#	disnix-gendist-roundrobin.1.xml \
+#	disnix-delegate.1.xml \
+#	disnix-env.1.xml \
+#	disnix-ssh-client.1.xml \
+#	disnix-reconstruct.1.xml \
+#	disnix-normalize-infra.1.xml \
+#	disnix-convert.1.xml \
+#	disnix-tmpfile.1.xml
 
 disnix-instantiate.1: disnix-instantiate.in
 	$(HELP2MAN) --output=$@ --no-info --name 'Instantiate a distributed derivation from Disnix expressions' "$(SHELL) disnix-instantiate"
--- origsrc/disnix-disnix-0.10.2/src/activate/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/activate/Makefile.am	2023-11-19 22:24:09.740825000 +0900
@@ -6,11 +6,11 @@ disnix-activate.1.xml: disnix-activate.1
 
 bin_PROGRAMS = disnix-activate
 noinst_HEADERS = run-activate.h
-noinst_DATA = disnix-activate.1.xml
+# noinst_DATA = disnix-activate.1.xml
 man1_MANS = disnix-activate.1
 
 disnix_activate_SOURCES = run-activate.c main.c
-disnix_activate_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I../libprocreact -I../libnixxml -I../libinfrastructure -I../libmanifest -I../libmain -I../libmodel -I../libdeploy -I../libmigrate
+disnix_activate_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libinfrastructure -I$(srcdir)/../libmanifest -I$(srcdir)/../libmain -I$(srcdir)/../libmodel -I$(srcdir)/../libdeploy -I$(srcdir)/../libmigrate
 disnix_activate_LDADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la ../libmanifest/libmanifest.la ../libmain/libmain.la ../libdeploy/libdeploy.la
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/build/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/build/Makefile.am	2023-11-19 21:58:23.812689400 +0900
@@ -6,11 +6,11 @@ disnix-build.1.xml: disnix-build.1
 
 bin_PROGRAMS = disnix-build
 noinst_HEADERS = run-build.h
-noinst_DATA = disnix-build.1.xml
+# noinst_DATA = disnix-build.1.xml
 man1_MANS = disnix-build.1
 
 disnix_build_SOURCES = run-build.c main.c
-disnix_build_CFLAGS = $(LIBXML2_CFLAGS) $(GLIB2_CFLAGS) -I../libnixxml -I../libprocreact -I../libdistderivation -I../libmain -I../libmodel -I../libbuild
+disnix_build_CFLAGS = $(LIBXML2_CFLAGS) $(GLIB2_CFLAGS) -I$(srcdir)/../libnixxml -I$(srcdir)/../libprocreact -I$(srcdir)/../libdistderivation -I$(srcdir)/../libmain -I$(srcdir)/../libmodel -I$(srcdir)/../libbuild
 disnix_build_LDADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la ../libdistderivation/libdistderivation.la ../libmain/libmain.la ../libpkgmgmt/libpkgmgmt.la ../libbuild/libbuild.la
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/capture-infra/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/capture-infra/Makefile.am	2023-11-19 23:07:29.352752800 +0900
@@ -6,11 +6,11 @@ disnix-capture-infra.1.xml: disnix-captu
 
 bin_PROGRAMS = disnix-capture-infra
 noinst_HEADERS = capture-infra.h
-noinst_DATA = disnix-capture-infra.1.xml
+# noinst_DATA = disnix-capture-infra.1.xml
 man1_MANS = disnix-capture-infra.1
 
 disnix_capture_infra_SOURCES = capture-infra.c main.c
 disnix_capture_infra_LDADD = ../libprocreact/libprocreact.la ../libinfrastructure/libinfrastructure.la ../libmain/libmain.la ../libstatemgmt/libstatemgmt.la
-disnix_capture_infra_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) $(LIBXSLT_CFLAGS) -I../libprocreact -I../libnixxml -I../libnixxml-glib -I../libinfrastructure -I../libmain -I../libstatemgmt -I../libmodel
+disnix_capture_infra_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) $(LIBXSLT_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libnixxml-glib -I$(srcdir)/../libinfrastructure -I$(srcdir)/../libmain -I$(srcdir)/../libstatemgmt -I$(srcdir)/../libmodel
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/capture-manifest/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/capture-manifest/Makefile.am	2023-11-19 23:08:29.422294100 +0900
@@ -8,11 +8,11 @@ disnix-capture-manifest.1.xml: disnix-ca
 
 bin_PROGRAMS = disnix-capture-manifest
 noinst_HEADERS = aggregated-manifest.h capture-manifest.h
-noinst_DATA = disnix-capture-manifest.1.xml
+# noinst_DATA = disnix-capture-manifest.1.xml
 man1_MANS = disnix-capture-manifest.1
 
 disnix_capture_manifest_SOURCES = aggregated-manifest.c capture-manifest.c main.c
 disnix_capture_manifest_LDADD = ../libprocreact/libprocreact.la ../libinfrastructure/libinfrastructure.la ../libmain/libmain.la ../libprofilemanifest/libprofilemanifest.la ../libpkgmgmt/libpkgmgmt.la ../libmanifest/libmanifest.la
-disnix_capture_manifest_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I../libprocreact -I../libnixxml -I../libinfrastructure -I../libmain -I../libmodel -I../libprofilemanifest -I../libpkgmgmt -I../libmanifest
+disnix_capture_manifest_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libinfrastructure -I$(srcdir)/../libmain -I$(srcdir)/../libmodel -I$(srcdir)/../libprofilemanifest -I$(srcdir)/../libpkgmgmt -I$(srcdir)/../libmanifest
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/clean-snapshots/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/clean-snapshots/Makefile.am	2023-11-19 22:27:54.081924900 +0900
@@ -6,11 +6,11 @@ disnix-clean-snapshots.1.xml: disnix-cle
 
 bin_PROGRAMS = disnix-clean-snapshots
 noinst_HEADERS = clean-snapshots.h
-noinst_DATA = disnix-clean-snapshots.1.xml
+# noinst_DATA = disnix-clean-snapshots.1.xml
 man1_MANS = disnix-clean-snapshots.1
 
 disnix_clean_snapshots_SOURCES = clean-snapshots.c main.c
-disnix_clean_snapshots_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) $(LIBXSLT_CFLAGS) -I../libprocreact -I../libnixxml -I../libinfrastructure -I../libmain -I../libstatemgmt -I../libmodel
+disnix_clean_snapshots_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) $(LIBXSLT_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libinfrastructure -I$(srcdir)/../libmain -I$(srcdir)/../libstatemgmt -I$(srcdir)/../libmodel
 disnix_clean_snapshots_LDADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la ../libinfrastructure/libinfrastructure.la ../libmain/libmain.la ../libstatemgmt/libstatemgmt.la
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/collect-garbage/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/collect-garbage/Makefile.am	2023-11-19 21:52:26.999558000 +0900
@@ -6,11 +6,11 @@ disnix-collect-garbage.1.xml: disnix-col
 
 bin_PROGRAMS = disnix-collect-garbage
 noinst_HEADERS = collect-garbage.h
-noinst_DATA = disnix-collect-garbage.1.xml
+# noinst_DATA = disnix-collect-garbage.1.xml
 man1_MANS = disnix-collect-garbage.1
 
 disnix_collect_garbage_SOURCES = collect-garbage.c main.c
 disnix_collect_garbage_LDADD = ../libprocreact/libprocreact.la ../libinfrastructure/libinfrastructure.la ../libmain/libmain.la ../libpkgmgmt/libpkgmgmt.la
-disnix_collect_garbage_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) $(LIBXSLT_CFLAGS) -I../libprocreact -I../libnixxml -I../libinfrastructure -I../libmain -I../libpkgmgmt -I../libmodel
+disnix_collect_garbage_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) $(LIBXSLT_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libinfrastructure -I$(srcdir)/../libmain -I$(srcdir)/../libpkgmgmt -I$(srcdir)/../libmodel
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/compare-manifest/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/compare-manifest/Makefile.am	2023-11-19 21:51:34.480885100 +0900
@@ -6,11 +6,11 @@ disnix-compare-manifest.1.xml: disnix-co
 
 bin_PROGRAMS = disnix-compare-manifest
 noinst_HEADERS = compare-manifest.h
-noinst_DATA = disnix-compare-manifest.1.xml
+# noinst_DATA = disnix-compare-manifest.1.xml
 man1_MANS = disnix-compare-manifest.1
 
 disnix_compare_manifest_SOURCES = compare-manifest.c main.c
-disnix_compare_manifest_CFLAGS = $(GLIB2_CFLAGS) -I../libprocreact -I../libmanifest -I../libmain -I../libnixxml
+disnix_compare_manifest_CFLAGS = $(GLIB2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libmanifest -I$(srcdir)/../libmain -I$(srcdir)/../libnixxml
 disnix_compare_manifest_LDADD = ../libmain/libmain.la ../libmanifest/libmanifest.la
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/convert-manifest/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/convert-manifest/Makefile.am	2023-11-19 23:12:22.215142500 +0900
@@ -8,11 +8,11 @@ disnix-convert-manifest.1.xml: disnix-co
 
 bin_PROGRAMS = disnix-convert-manifest
 noinst_HEADERS = activationmapping.h convert-manifest.h distributionmapping.h oldmanifest.h oldsnapshotmapping.h
-noinst_DATA = disnix-convert-manifest.1.xml
+# noinst_DATA = disnix-convert-manifest.1.xml
 man1_MANS = disnix-convert-manifest.1
 
 disnix_convert_manifest_SOURCES = activationmapping.c convert-manifest.c distributionmapping.c oldmanifest.c oldsnapshotmapping.c main.c
 disnix_convert_manifest_LDADD = ../libprocreact/libprocreact.la ../libinfrastructure/libinfrastructure.la ../libmain/libmain.la ../libmanifest/libmanifest.la
-disnix_convert_manifest_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I../libprocreact -I../libnixxml -I../libnixxml-glib -I../libinfrastructure -I../libmain -I../libmodel -I../libmanifest
+disnix_convert_manifest_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libnixxml-glib -I$(srcdir)/../libinfrastructure -I$(srcdir)/../libmain -I$(srcdir)/../libmodel -I$(srcdir)/../libmanifest
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/copy-closure/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/copy-closure/Makefile.am	2023-11-19 21:50:07.374967400 +0900
@@ -5,11 +5,11 @@ disnix-copy-closure.1.xml: disnix-copy-c
 	$(SHELL) ../../maintenance/man2docbook.bash $<
 
 bin_PROGRAMS = disnix-copy-closure
-noinst_DATA = disnix-copy-closure.1.xml
+# noinst_DATA = disnix-copy-closure.1.xml
 man1_MANS = disnix-copy-closure.1
 
 disnix_copy_closure_SOURCES = main.c
 disnix_copy_closure_LDADD = ../libmain/libmain.la ../libpkgmgmt/libpkgmgmt.la
-disnix_copy_closure_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) $(LIBXSLT_CFLAGS) -I../libmain -I../libpkgmgmt -I../libprocreact
+disnix_copy_closure_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) $(LIBXSLT_CFLAGS) -I$(srcdir)/../libmain -I$(srcdir)/../libpkgmgmt -I$(srcdir)/../libprocreact
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/copy-snapshots/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/copy-snapshots/Makefile.am	2023-11-19 21:50:51.554456000 +0900
@@ -5,11 +5,11 @@ disnix-copy-snapshots.1.xml: disnix-copy
 	$(SHELL) ../../maintenance/man2docbook.bash $<
 
 bin_PROGRAMS = disnix-copy-snapshots
-noinst_DATA = disnix-copy-snapshots.1.xml
+# noinst_DATA = disnix-copy-snapshots.1.xml
 man1_MANS = disnix-copy-snapshots.1
 
 disnix_copy_snapshots_SOURCES = main.c
 disnix_copy_snapshots_LDADD = ../libmain/libmain.la ../libstatemgmt/libstatemgmt.la
-disnix_copy_snapshots_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) $(LIBXSLT_CFLAGS) -I../libmain -I../libprocreact -I../libstatemgmt
+disnix_copy_snapshots_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) $(LIBXSLT_CFLAGS) -I$(srcdir)/../libmain -I$(srcdir)/../libprocreact -I$(srcdir)/../libstatemgmt
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/dbus-service/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/dbus-service/Makefile.am	2023-11-19 21:57:10.981765300 +0900
@@ -14,23 +14,23 @@ AM_CPPFLAGS=-DLOCALSTATEDIR=\"$(localsta
 
 bin_PROGRAMS = disnix-service disnix-client
 noinst_HEADERS = daemonize.h methods.h signaling.h logging.h locking.h jobmanagement.h disnix-client.h disnix-service.h
-noinst_DATA = disnix-client.1.xml disnix-service.8.xml
+# noinst_DATA = disnix-client.1.xml disnix-service.8.xml
 man1_MANS = disnix-client.1
 man8_MANS = disnix-service.8
 
 disnix_service_SOURCES = daemonize.c methods.c signaling.c logging.c locking.c jobmanagement.c disnix-service.c disnix-service-main.c disnix-dbus.c
-disnix_service_CFLAGS = $(GLIB2_CFLAGS) $(GIO2_CFLAGS) -I../libprocreact -I../libpkgmgmt -I../libstatemgmt -I../libprofilemanifest
+disnix_service_CFLAGS = $(GLIB2_CFLAGS) $(GIO2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libpkgmgmt -I$(srcdir)/../libstatemgmt -I$(srcdir)/../libprofilemanifest
 disnix_service_LDADD = $(GLIB2_LIBS) $(GIO2_LIBS) ../libpkgmgmt/libpkgmgmt.la ../libstatemgmt/libstatemgmt.la ../libprofilemanifest/libprofilemanifest.la
 
 disnix_client_SOURCES = disnix-client.c disnix-client-main.c disnix-dbus.c
-disnix_client_CFLAGS = $(GLIB2_CFLAGS) $(GIO2_CFLAGS) -I../libmain
+disnix_client_CFLAGS = $(GLIB2_CFLAGS) $(GIO2_CFLAGS) -I$(srcdir)/../libmain
 disnix_client_LDADD = $(GLIB2_LIBS) $(GIO2_LIBS) ../libmain/libmain.la
 
 disnix-dbus.h: disnix.xml
-	gdbus-codegen --generate-c-code=disnix-dbus disnix.xml
+	gdbus-codegen --generate-c-code=disnix-dbus $(srcdir)/disnix.xml
 
 disnix-dbus.c: disnix.xml
-	gdbus-codegen --generate-c-code=disnix-dbus disnix.xml
+	gdbus-codegen --generate-c-code=disnix-dbus $(srcdir)/disnix.xml
 
 BUILT_SOURCES = disnix-dbus.c disnix-dbus.h
 
--- origsrc/disnix-disnix-0.10.2/src/delete-state/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/delete-state/Makefile.am	2023-11-19 23:06:28.790045900 +0900
@@ -6,11 +6,11 @@ disnix-delete-state.1.xml: disnix-delete
 
 bin_PROGRAMS = disnix-delete-state
 noinst_HEADERS = run-delete-state.h
-noinst_DATA = disnix-delete-state.1.xml
+# noinst_DATA = disnix-delete-state.1.xml
 man1_MANS = disnix-delete-state.1
 
 disnix_delete_state_SOURCES = run-delete-state.c main.c
-disnix_delete_state_CFLAGS = $(LIBXML2_CFLAGS) $(GLIB2_CFLAGS) -I../libprocreact -I../libnixxml -I../libinfrastructure -I../libmanifest -I../libmain -I../libmodel -I../libmigrate
+disnix_delete_state_CFLAGS = $(LIBXML2_CFLAGS) $(GLIB2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libinfrastructure -I$(srcdir)/../libmanifest -I$(srcdir)/../libmain -I$(srcdir)/../libmodel -I$(srcdir)/../libmigrate
 disnix_delete_state_LDADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la ../libmanifest/libmanifest.la ../libmain/libmain.la ../libmigrate/libmigrate.la
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/deploy/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/deploy/Makefile.am	2023-11-19 23:11:30.399407900 +0900
@@ -6,11 +6,11 @@ disnix-deploy.1.xml: disnix-deploy.1
 
 bin_PROGRAMS = disnix-deploy
 noinst_HEADERS = run-deploy.h
-noinst_DATA = disnix-deploy.1.xml
+# noinst_DATA = disnix-deploy.1.xml
 man1_MANS = disnix-deploy.1
 
 disnix_deploy_SOURCES = run-deploy.c main.c
-disnix_deploy_CFLAGS = $(GLIB2_CFLAGS) -I../libprocreact -I../libnixxml -I../libmanifest -I../libmodel  -I../libmain -I../libmigrate -I../libdeploy
+disnix_deploy_CFLAGS = $(GLIB2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libmanifest -I$(srcdir)/../libmodel  -I$(srcdir)/../libmain -I$(srcdir)/../libmigrate -I$(srcdir)/../libdeploy
 disnix_deploy_LDADD = ../libmain/libmain.la ../libmigrate/libmigrate.la ../libdeploy/libdeploy.la
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/diagnose/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/diagnose/Makefile.am	2023-11-19 22:19:08.434053600 +0900
@@ -6,11 +6,11 @@ disnix-diagnose.1.xml: disnix-diagnose.1
 
 bin_PROGRAMS = disnix-diagnose
 noinst_HEADERS = diagnose.h
-noinst_DATA = disnix-diagnose.1.xml
+# noinst_DATA = disnix-diagnose.1.xml
 man1_MANS = disnix-diagnose.1
 
 disnix_diagnose_SOURCES = diagnose.c main.c
-disnix_diagnose_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I../libprocreact -I../libnixxml -I../libnixxml-glib -I../libinfrastructure -I../libmanifest -I../libmain -I../libstatemgmt -I../libmodel
+disnix_diagnose_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libnixxml-glib -I$(srcdir)/../libinfrastructure -I$(srcdir)/../libmanifest -I$(srcdir)/../libmain -I$(srcdir)/../libstatemgmt -I$(srcdir)/../libmodel
 disnix_diagnose_LDADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la ../libmanifest/libmanifest.la ../libmain/libmain.la ../libstatemgmt/libstatemgmt.la
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/distribute/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/distribute/Makefile.am	2023-11-19 21:59:18.387080100 +0900
@@ -6,11 +6,11 @@ disnix-distribute.1.xml: disnix-distribu
 
 bin_PROGRAMS = disnix-distribute
 noinst_HEADERS = run-distribute.h
-noinst_DATA = disnix-distribute.1.xml
+# noinst_DATA = disnix-distribute.1.xml
 man1_MANS = disnix-distribute.1
 
 disnix_distribute_SOURCES = run-distribute.c main.c
-disnix_distribute_CFLAGS = $(GLIB2_CFLAGS) -I../libprocreact -I../libnixxml -I../libmanifest -I../libmain -I../libmodel -I../libdeploy
+disnix_distribute_CFLAGS = $(GLIB2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libmanifest -I$(srcdir)/../libmain -I$(srcdir)/../libmodel -I$(srcdir)/../libdeploy
 disnix_distribute_LDADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la ../libmanifest/libmanifest.la ../libmain/libmain.la ../libpkgmgmt/libpkgmgmt.la ../libdeploy/libdeploy.la
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/libbuild/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/libbuild/Makefile.am	2023-11-19 21:42:26.771035900 +0900
@@ -2,5 +2,6 @@ pkglib_LTLIBRARIES = libbuild.la
 pkginclude_HEADERS = build.h
 
 libbuild_la_SOURCES = build.c
-libbuild_la_CFLAGS = $(LIBXML2_CFLAGS) $(GLIB2_CFLAGS) -I../libnixxml -I../libprocreact -I../libdistderivation -I../libmodel -I../libpkgmgmt
+libbuild_la_CFLAGS = $(LIBXML2_CFLAGS) $(GLIB2_CFLAGS) -I$(srcdir)/../libnixxml -I$(srcdir)/../libprocreact -I$(srcdir)/../libdistderivation -I$(srcdir)/../libmodel -I$(srcdir)/../libpkgmgmt
 libbuild_la_LIBADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la ../libdistderivation/libdistderivation.la ../libpkgmgmt/libpkgmgmt.la
+libbuild_la_LDFLAGS = -no-undefined
--- origsrc/disnix-disnix-0.10.2/src/libdeploy/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/libdeploy/Makefile.am	2023-11-19 21:41:28.705077200 +0900
@@ -2,5 +2,6 @@ pkglib_LTLIBRARIES = libdeploy.la
 pkginclude_HEADERS = distribute.h locking.h set-profiles.h transition.h activate.h deploy.h deploymentflags.h
 
 libdeploy_la_SOURCES = distribute.c locking.c set-profiles.c transition.c activate.c deploy.c
-libdeploy_la_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I../libprocreact -I../libinfrastructure -I../libmanifest -I../libnixxml -I../libmodel -I../libpkgmgmt -I../libstatemgmt -I../libmigrate
-libdeploy_la_LIBADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la ../libmanifest/libmanifest.la ../libpkgmgmt/libpkgmgmt.la ../libstatemgmt/libstatemgmt.la ../libmigrate/libmigrate.la
+libdeploy_la_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libinfrastructure -I$(srcdir)/../libmanifest -I$(srcdir)/../libnixxml -I$(srcdir)/../libmodel -I$(srcdir)/../libpkgmgmt -I$(srcdir)/../libstatemgmt -I$(srcdir)/../libmigrate
+libdeploy_la_LIBADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la ../libmanifest/libmanifest.la ../libpkgmgmt/libpkgmgmt.la ../libstatemgmt/libstatemgmt.la ../libmigrate/libmigrate.la ../libmain/libmain.la
+libdeploy_la_LDFLAGS = -no-undefined
--- origsrc/disnix-disnix-0.10.2/src/libdistderivation/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/libdistderivation/Makefile.am	2023-11-19 21:34:11.651591500 +0900
@@ -2,5 +2,6 @@ pkglib_LTLIBRARIES = libdistderivation.l
 pkginclude_HEADERS = distributedderivation.h derivationmapping.h derivationmappingarray.h derivationmapping-iterator.h interface.h interfacestable.h
 
 libdistderivation_la_SOURCES = distributedderivation.c derivationmapping.c derivationmappingarray.c derivationmapping-iterator.c interface.c interfacestable.c
-libdistderivation_la_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I../libprocreact -I../libnixxml -I../libnixxml-glib -I../libmodel
+libdistderivation_la_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libnixxml-glib -I$(srcdir)/../libmodel
 libdistderivation_la_LIBADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la ../libmodel/libmodel.la ../libnixxml-glib/libnixxml-glib.la
+libdistderivation_la_LDFLAGS = -no-undefined
--- origsrc/disnix-disnix-0.10.2/src/libinfrastructure/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/libinfrastructure/Makefile.am	2023-11-19 21:33:04.314693600 +0900
@@ -4,5 +4,6 @@ pkglib_LTLIBRARIES = libinfrastructure.l
 pkginclude_HEADERS = target.h targetstable.h targets-iterator.h targetpropertiestable.h containerstable.h
 
 libinfrastructure_la_SOURCES = target.c targetstable.c targets-iterator.c targetpropertiestable.c containerstable.c
-libinfrastructure_la_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) $(LIBXSLT_CFLAGS) -I../libprocreact -I../libnixxml -I../libnixxml-glib -I../libmodel -I../libpkgmgmt
+libinfrastructure_la_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) $(LIBXSLT_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libnixxml-glib -I$(srcdir)/../libmodel -I$(srcdir)/../libpkgmgmt
 libinfrastructure_la_LIBADD = $(GLIB2_LIBS) $(LIBXML2_LIBS) $(LIBXSLT_LIBS) ../libprocreact/libprocreact.la ../libnixxml/libnixxml.la ../libnixxml-glib/libnixxml-glib.la ../libmodel/libmodel.la ../libpkgmgmt/libpkgmgmt.la
+libinfrastructure_la_LDFLAGS = -no-undefined
--- origsrc/disnix-disnix-0.10.2/src/libmain/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/libmain/Makefile.am	2023-11-19 15:58:36.659276200 +0900
@@ -2,3 +2,4 @@ pkglib_LTLIBRARIES = libmain.la
 pkginclude_HEADERS = defaultoptions.h interrupt.h
 
 libmain_la_SOURCES = defaultoptions.c interrupt.c
+libmain_la_LDFLAGS = -no-undefined
--- origsrc/disnix-disnix-0.10.2/src/libmain/interrupt.c	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/libmain/interrupt.c	2023-11-19 15:59:05.945773000 +0900
@@ -22,7 +22,9 @@
 
 #define TRUE 1
 #define FALSE 0
+#if !defined(NULL)
 #define NULL 0
+#endif
 
 volatile int interrupted = FALSE;
 
--- origsrc/disnix-disnix-0.10.2/src/libmanifest/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/libmanifest/Makefile.am	2023-11-19 21:35:12.870723800 +0900
@@ -31,5 +31,6 @@ libmanifest_la_SOURCES = interdependency
 	snapshotmappingarray.c \
 	snapshotmapping-traverse.c
 
-libmanifest_la_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I../libprocreact -I../libnixxml -I../libnixxml-glib -I../libmodel -I../libinfrastructure
+libmanifest_la_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libnixxml-glib -I$(srcdir)/../libmodel -I$(srcdir)/../libinfrastructure
 libmanifest_la_LIBADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la ../libmodel/libmodel.la ../libinfrastructure/libinfrastructure.la
+libmanifest_la_LDFLAGS = -no-undefined
--- origsrc/disnix-disnix-0.10.2/src/libmigrate/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/libmigrate/Makefile.am	2023-11-19 21:38:37.373885300 +0900
@@ -2,5 +2,6 @@ pkglib_LTLIBRARIES = libmigrate.la
 pkginclude_HEADERS = restore.h snapshot.h delete-state.h migrate.h datamigrationflags.h
 
 libmigrate_la_SOURCES = restore.c snapshot.c delete-state.c migrate.c
-libmigrate_la_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I../libprocreact -I../libinfrastructure -I../libmanifest -I../libnixxml -I../libnixxml-glib -I../libmodel -I../libstatemgmt
+libmigrate_la_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libinfrastructure -I$(srcdir)/../libmanifest -I$(srcdir)/../libnixxml -I$(srcdir)/../libnixxml-glib -I$(srcdir)/../libmodel -I$(srcdir)/../libstatemgmt
 libmigrate_la_LIBADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la ../libmanifest/libmanifest.la ../libstatemgmt/libstatemgmt.la
+libmigrate_la_LDFLAGS = -no-undefined
--- origsrc/disnix-disnix-0.10.2/src/libmodel/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/libmodel/Makefile.am	2023-11-19 16:00:14.136240800 +0900
@@ -2,5 +2,6 @@ pkglib_LTLIBRARIES = libmodel.la
 pkginclude_HEADERS = modeliterator.h
 
 libmodel_la_SOURCES = modeliterator.c
-libmodel_la_CFLAGS = $(LIBXML2_CFLAGS) $(GLIB2_CFLAGS) -I../libprocreact
+libmodel_la_CFLAGS = $(LIBXML2_CFLAGS) $(GLIB2_CFLAGS) -I$(srcdir)/../libprocreact
 libmodel_la_LIBADD = $(LIBXML2_LIBS) $(GLIB2_LIBS) ../libprocreact/libprocreact.la
+libmodel_la_LDFLAGS = -no-undefined
--- origsrc/disnix-disnix-0.10.2/src/libnixxml/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/libnixxml/Makefile.am	2023-11-19 15:55:53.901459400 +0900
@@ -27,3 +27,4 @@ libnixxml_la_SOURCES = 	nixxml-generate-
 
 libnixxml_la_CFLAGS = $(LIBXML2_CFLAGS)
 libnixxml_la_LIBADD = $(LIBXML2_LIBS)
+libnixxml_la_LDFLAGS = -no-undefined
--- origsrc/disnix-disnix-0.10.2/src/libnixxml-glib/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/libnixxml-glib/Makefile.am	2023-11-19 15:58:03.304785500 +0900
@@ -2,5 +2,6 @@ pkglib_LTLIBRARIES = libnixxml-glib.la
 pkginclude_HEADERS = nixxml-gptrarray.h nixxml-ghashtable.h nixxml-ghashtable-iter.h nixxml-glib.h
 
 libnixxml_glib_la_SOURCES = nixxml-gptrarray.c nixxml-ghashtable.c nixxml-ghashtable-iter.c nixxml-glib.c
-libnixxml_glib_la_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I../libnixxml
+libnixxml_glib_la_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I$(srcdir)/../libnixxml
 libnixxml_glib_la_LIBADD = $(GLIB2_LIBS) ../libnixxml/libnixxml.la
+libnixxml_glib_la_LDFLAGS = -no-undefined
--- origsrc/disnix-disnix-0.10.2/src/libpkgmgmt/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/libpkgmgmt/Makefile.am	2023-11-19 16:00:55.630294600 +0900
@@ -4,5 +4,6 @@ pkginclude_HEADERS = package-management.
 AM_CPPFLAGS=-DLOCALSTATEDIR=\"$(localstatedir)\"
 
 libpkgmgmt_la_SOURCES = package-management.c remote-package-management.c copy-closure.c
-libpkgmgmt_la_CFLAGS = $(GLIB2_CFLAGS) -I../libprocreact
+libpkgmgmt_la_CFLAGS = $(GLIB2_CFLAGS) -I$(srcdir)/../libprocreact
 libpkgmgmt_la_LIBADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la
+libpkgmgmt_la_LDFLAGS = -no-undefined
--- origsrc/disnix-disnix-0.10.2/src/libprocreact/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/libprocreact/Makefile.am	2023-11-19 15:53:44.693115800 +0900
@@ -2,3 +2,4 @@ pkglib_LTLIBRARIES = libprocreact.la
 pkginclude_HEADERS = procreact_future.h procreact_pid.h procreact_pid_iterator.h procreact_future_iterator.h procreact_signal.h procreact_types.h procreact_util.h
 
 libprocreact_la_SOURCES = procreact_future.c procreact_pid.c procreact_pid_iterator.c procreact_future_iterator.c procreact_signal.c procreact_types.c
+libprocreact_la_LDFLAGS = -no-undefined
--- origsrc/disnix-disnix-0.10.2/src/libprofilemanifest/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/libprofilemanifest/Makefile.am	2023-11-19 21:36:40.551409100 +0900
@@ -2,5 +2,6 @@ pkglib_LTLIBRARIES = libprofilemanifest.
 pkginclude_HEADERS = profilemanifest.h profilemanifesttarget.h profilemanifesttargettable.h profilemanifesttarget-iterator.h profilelocking.h
 
 libprofilemanifest_la_SOURCES = profilemanifest.c profilemanifesttarget.c profilemanifesttargettable.c profilemanifesttarget-iterator.c profilelocking.c
-libprofilemanifest_la_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I../libprocreact -I../libmodel -I../libstatemgmt -I../libnixxml -I../libnixxml-glib -I../libmanifest -I../libinfrastructure
+libprofilemanifest_la_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libmodel -I$(srcdir)/../libstatemgmt -I$(srcdir)/../libnixxml -I$(srcdir)/../libnixxml-glib -I$(srcdir)/../libmanifest -I$(srcdir)/../libinfrastructure
 libprofilemanifest_la_LIBADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la ../libmodel/libmodel.la ../libstatemgmt/libstatemgmt.la ../libnixxml/libnixxml.la ../libnixxml-glib/libnixxml-glib.la ../libmanifest/libmanifest.la
+libprofilemanifest_la_LDFLAGS = -no-undefined
--- origsrc/disnix-disnix-0.10.2/src/libprofilemanifest/profilelocking.c	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/libprofilemanifest/profilelocking.c	2023-11-19 21:37:45.103566300 +0900
@@ -8,7 +8,7 @@
 #include <manifestservicestable.h>
 #include <manifestservice.h>
 
-static int lock_or_unlock_services(int log_fd, ProfileManifest *profile_manifest, gchar *action, pid_t (*notify_function) (gchar *type, gchar *container, gchar *component, int stdout, int stderr))
+static int lock_or_unlock_services(int log_fd, ProfileManifest *profile_manifest, gchar *action, pid_t (*notify_function) (gchar *type, gchar *container, gchar *component, int _stdout, int _stderr))
 {
     unsigned int i;
     int exit_status = TRUE;
--- origsrc/disnix-disnix-0.10.2/src/libstatemgmt/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/libstatemgmt/Makefile.am	2023-11-19 16:01:42.593165100 +0900
@@ -2,5 +2,6 @@ pkglib_LTLIBRARIES = libstatemgmt.la
 pkginclude_HEADERS = state-management.h snapshot-management.h remote-state-management.h remote-snapshot-management.h copy-snapshots.h
 
 libstatemgmt_la_SOURCES = state-management.c snapshot-management.c remote-state-management.c remote-snapshot-management.c copy-snapshots.c
-libstatemgmt_la_CFLAGS = $(GLIB2_CFLAGS) -I../libprocreact
+libstatemgmt_la_CFLAGS = $(GLIB2_CFLAGS) -I$(srcdir)/../libprocreact
 libstatemgmt_la_LIBADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la
+libstatemgmt_la_LDFLAGS = -no-undefined
--- origsrc/disnix-disnix-0.10.2/src/lock/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/lock/Makefile.am	2023-11-19 22:18:14.298510400 +0900
@@ -6,11 +6,11 @@ disnix-lock.1.xml: disnix-lock.1
 
 bin_PROGRAMS = disnix-lock
 noinst_HEADERS = lock-or-unlock.h
-noinst_DATA = disnix-lock.1.xml
+# noinst_DATA = disnix-lock.1.xml
 man1_MANS = disnix-lock.1
 
 disnix_lock_SOURCES = lock-or-unlock.c main.c
-disnix_lock_CFLAGS = $(GLIB2_CFLAGS) -I../libprocreact -I../libnixxml -I../libmanifest -I../libmain -I../libmodel -I../libdeploy
+disnix_lock_CFLAGS = $(GLIB2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libmanifest -I$(srcdir)/../libmain -I$(srcdir)/../libmodel -I$(srcdir)/../libdeploy
 disnix_lock_LDADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la ../libmanifest/libmanifest.la ../libmain/libmain.la ../libpkgmgmt/libpkgmgmt.la ../libdeploy/libdeploy.la
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/migrate/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/migrate/Makefile.am	2023-11-19 23:10:41.598016300 +0900
@@ -6,11 +6,11 @@ disnix-migrate.1.xml: disnix-migrate.1
 
 bin_PROGRAMS = disnix-migrate
 noinst_HEADERS = run-migrate.h
-noinst_DATA = disnix-migrate.1.xml
+# noinst_DATA = disnix-migrate.1.xml
 man1_MANS = disnix-migrate.1
 
 disnix_migrate_SOURCES = run-migrate.c main.c
-disnix_migrate_CFLAGS = $(LIBXML2_CFLAGS) $(GLIB2_CFLAGS) -I../libprocreact -I../libnixxml -I../libinfrastructure -I../libmanifest -I../libmodel  -I../libmain -I../libmigrate
+disnix_migrate_CFLAGS = $(LIBXML2_CFLAGS) $(GLIB2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libinfrastructure -I$(srcdir)/../libmanifest -I$(srcdir)/../libmodel  -I$(srcdir)/../libmain -I$(srcdir)/../libmigrate
 disnix_migrate_LDADD = ../libmain/libmain.la ../libmigrate/libmigrate.la
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/query/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/query/Makefile.am	2023-11-19 21:53:21.055160200 +0900
@@ -6,11 +6,11 @@ disnix-query.1.xml: disnix-query.1
 
 bin_PROGRAMS = disnix-query
 noinst_HEADERS = query-installed.h
-noinst_DATA = disnix-query.1.xml
+# noinst_DATA = disnix-query.1.xml
 man1_MANS = disnix-query.1
 
 disnix_query_SOURCES = query-installed.c main.c
 disnix_query_LDADD = ../libprocreact/libprocreact.la ../libinfrastructure/libinfrastructure.la ../libmain/libmain.la ../libpkgmgmt/libpkgmgmt.la ../libprofilemanifest/libprofilemanifest.la
-disnix_query_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I../libprocreact -I../libnixxml -I../libinfrastructure -I../libmain -I../libpkgmgmt -I../libmodel -I../libprofilemanifest
+disnix_query_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libinfrastructure -I$(srcdir)/../libmain -I$(srcdir)/../libpkgmgmt -I$(srcdir)/../libmodel -I$(srcdir)/../libprofilemanifest
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/restore/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/restore/Makefile.am	2023-11-19 22:27:05.355500500 +0900
@@ -6,11 +6,11 @@ disnix-restore.1.xml: disnix-restore.1
 
 bin_PROGRAMS = disnix-restore
 noinst_HEADERS = run-restore.h
-noinst_DATA = disnix-restore.1.xml
+# noinst_DATA = disnix-restore.1.xml
 man1_MANS = disnix-restore.1
 
 disnix_restore_SOURCES = run-restore.c main.c
-disnix_restore_CFLAGS = $(LIBXML2_CFLAGS) $(GLIB2_CFLAGS) -I../libprocreact -I../libnixxml -I../libinfrastructure -I../libmanifest -I../libmain -I../libmodel -I../libmigrate
+disnix_restore_CFLAGS = $(LIBXML2_CFLAGS) $(GLIB2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libinfrastructure -I$(srcdir)/../libmanifest -I$(srcdir)/../libmain -I$(srcdir)/../libmodel -I$(srcdir)/../libmigrate
 disnix_restore_LDADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la ../libmanifest/libmanifest.la ../libmain/libmain.la ../libmigrate/libmigrate.la
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/run-activity/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/run-activity/Makefile.am	2023-11-19 23:09:47.859449700 +0900
@@ -8,11 +8,11 @@ AM_CPPFLAGS=-DLOCALSTATEDIR=\"$(localsta
 
 bin_PROGRAMS = disnix-run-activity
 noinst_HEADERS = run-activity.h
-noinst_DATA = disnix-run-activity.1.xml
+# noinst_DATA = disnix-run-activity.1.xml
 man1_MANS = disnix-run-activity.1
 
 disnix_run_activity_SOURCES = run-activity.c main.c
-disnix_run_activity_CFLAGS = $(GLIB2_CFLAGS) $(GIO2_CFLAGS) -I../libmain -I../libprocreact -I../libpkgmgmt -I../libstatemgmt -I../libprofilemanifest
+disnix_run_activity_CFLAGS = $(GLIB2_CFLAGS) $(GIO2_CFLAGS) -I$(srcdir)/../libmain -I$(srcdir)/../libprocreact -I$(srcdir)/../libpkgmgmt -I$(srcdir)/../libstatemgmt -I$(srcdir)/../libprofilemanifest
 disnix_run_activity_LDADD = $(GLIB2_LIBS) $(GIO2_LIBS) ../libmain/libmain.la ../libpkgmgmt/libpkgmgmt.la ../libstatemgmt/libstatemgmt.la ../libprofilemanifest/libprofilemanifest.la
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/set/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/set/Makefile.am	2023-11-19 22:20:07.272891200 +0900
@@ -8,11 +8,11 @@ AM_CPPFLAGS = -DLOCALSTATEDIR=\"$(locals
 
 bin_PROGRAMS = disnix-set
 noinst_HEADERS = run-set-profiles.h
-noinst_DATA = disnix-set.1.xml
+# noinst_DATA = disnix-set.1.xml
 man1_MANS = disnix-set.1
 
 disnix_set_SOURCES = run-set-profiles.c main.c
-disnix_set_CFLAGS = $(GLIB2_CFLAGS) -I../libprocreact -I../libnixxml -I../libmanifest -I../libmain -I../libpkgmgmt -I../libmodel -I../libdeploy
+disnix_set_CFLAGS = $(GLIB2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libmanifest -I$(srcdir)/../libmain -I$(srcdir)/../libpkgmgmt -I$(srcdir)/../libmodel -I$(srcdir)/../libdeploy
 disnix_set_LDADD = $(GLIB2_LIBS) ../libprocreact/libprocreact.la ../libmanifest/libmanifest.la ../libmain/libmain.la ../libpkgmgmt/libpkgmgmt.la ../libdeploy/libdeploy.la
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/snapshot/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/snapshot/Makefile.am	2023-11-19 22:26:10.856010500 +0900
@@ -6,11 +6,11 @@ disnix-snapshot.1.xml: disnix-snapshot.1
 
 bin_PROGRAMS = disnix-snapshot
 noinst_HEADERS = run-snapshot.h
-noinst_DATA = disnix-snapshot.1.xml
+# noinst_DATA = disnix-snapshot.1.xml
 man1_MANS = disnix-snapshot.1
 
 disnix_snapshot_SOURCES = run-snapshot.c main.c
-disnix_snapshot_CFLAGS = $(LIBXML2_CFLAGS) $(GLIB2_CFLAGS) -I../libprocreact -I../libnixxml -I../libinfrastructure -I../libmanifest -I../libmodel -I../libmain -I../libmigrate
+disnix_snapshot_CFLAGS = $(LIBXML2_CFLAGS) $(GLIB2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libinfrastructure -I$(srcdir)/../libmanifest -I$(srcdir)/../libmodel -I$(srcdir)/../libmain -I$(srcdir)/../libmigrate
 disnix_snapshot_LDADD = ../libmain/libmain.la ../libmigrate/libmigrate.la
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
--- origsrc/disnix-disnix-0.10.2/src/visualize/Makefile.am	2022-03-05 20:26:56.000000000 +0900
+++ src/disnix-disnix-0.10.2/src/visualize/Makefile.am	2023-11-19 22:25:09.314338700 +0900
@@ -6,11 +6,11 @@ disnix-visualize.1.xml: disnix-visualize
 
 bin_PROGRAMS = disnix-visualize
 noinst_HEADERS = clustertable.h edgestable.h graph.h
-noinst_DATA = disnix-visualize.1.xml
+# noinst_DATA = disnix-visualize.1.xml
 man1_MANS = disnix-visualize.1
 
 disnix_visualize_SOURCES = main.c clustertable.c edgestable.c graph.c
-disnix_visualize_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I../libprocreact -I../libnixxml -I../libinfrastructure -I../libmanifest -I../libmain -I../libmodel
+disnix_visualize_CFLAGS = $(GLIB2_CFLAGS) $(LIBXML2_CFLAGS) -I$(srcdir)/../libprocreact -I$(srcdir)/../libnixxml -I$(srcdir)/../libinfrastructure -I$(srcdir)/../libmanifest -I$(srcdir)/../libmain -I$(srcdir)/../libmodel
 disnix_visualize_LDADD = $(GLIB2_LIBS) ../libmanifest/libmanifest.la ../libmain/libmain.la
 
 EXTRA_DIST = $(man1_MANS) $(noinst_DATA)
