diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/packdude/Makefile src/packdude/Makefile
--- origsrc/packdude/Makefile	2015-07-25 12:47:41.000000000 +0900
+++ src/packdude/Makefile	2015-07-25 12:48:47.082622500 +0900
@@ -3,7 +3,7 @@ CFLAGS ?= -O2 -pipe
 LDFLAGS ?= -s
 PKG_CONFIG ?= pkg-config
 DESTDIR ?=
-BIN_DIR ?= /bin
+BIN_DIR ?= /usr/bin
 MAN_DIR ?= /usr/share/man
 DOC_DIR ?= /usr/share/doc
 VAR_DIR ?= /var
