--- origsrc/libss7-2.0.0/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libss7-2.0.0/Makefile.am	2015-03-17 14:36:09.971371000 +0900
@@ -0,0 +1,13 @@
+
+check_PROGRAMS = parser_debug
+parser_debug_SOURCES = parser_debug.c
+parser_debug_LDADD = libss7.la
+
+include_HEADERS = libss7.h
+
+lib_LTLIBRARIES = libss7.la
+libss7_la_SOURCES = isup.c mtp2.c mtp3.c ss7.c ss7_sched.c version.c
+libss7_la_LDFLAGS = -no-undefined -version-info 2
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = libss7.pc
--- origsrc/libss7-2.0.0/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/libss7-2.0.0/configure.ac	2015-03-17 14:35:55.127621000 +0900
@@ -0,0 +1,31 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([libss7], [2.0.0])
+AC_CONFIG_SRCDIR([isup.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_AWK
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_LN_S
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([fcntl.h netinet/in.h stdlib.h string.h sys/ioctl.h sys/socket.h sys/time.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_C_INLINE
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([gettimeofday memset select strcasecmp strchr strerror])
+
+AC_CONFIG_FILES([Makefile libss7.pc])
+AC_OUTPUT
--- origsrc/libss7-2.0.0/libss7.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libss7-2.0.0/libss7.pc.in	2015-03-17 14:35:07.111996000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libss7
+Description: Implementation of Signaling System 7 (SS7)
+Version: @VERSION@
+Libs: -L${libdir} -lss7
+Cflags: -I${includedir}
--- origsrc/libss7-2.0.0/version.c	1970-01-01 09:00:00.000000000 +0900
+++ src/libss7-2.0.0/version.c	2015-03-17 14:25:03.052425700 +0900
@@ -0,0 +1,14 @@
+/*
+ * version.c
+ * Automatically generated
+ */
+
+#include "libss7.h"
+
+static const char ss7_version[] = "2.0.0";
+
+const char *ss7_get_version(void)
+{
+	return ss7_version;
+}
+
