--- origsrc/h4h5tools-2.2.5/Makefile.am	2020-03-10 23:33:18.000000000 +0900
+++ src/h4h5tools-2.2.5/Makefile.am	2024-10-16 02:29:58.720212700 +0900
@@ -16,3 +16,6 @@
 #############################################################################
 
 SUBDIRS=lib utils
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = h4toh5.pc
--- origsrc/h4h5tools-2.2.5/configure.ac	2024-10-16 02:20:38.253983500 +0900
+++ src/h4h5tools-2.2.5/configure.ac	2024-10-16 02:30:47.844829500 +0900
@@ -28,7 +28,7 @@ AC_CONFIG_SRCDIR([lib/src/h4toh5main.c])
 AC_CONFIG_AUX_DIR([bin])
 AM_INIT_AUTOMAKE([foreign])
 AM_MAINTAINER_MODE
-AC_PROG_RANLIB
+LT_INIT
 
 AC_CANONICAL_HOST
 AC_SUBST([CPPFLAGS])
@@ -663,6 +663,7 @@ AC_CONFIG_FILES([config/depend1
                  utils/h4toh5/testh4toh5.sh
                  utils/h5toh4/Makefile
                  utils/h5toh4/testh5toh4.sh])
+AC_CONFIG_FILES([h4toh5.pc])
 AC_OUTPUT
 no_create=$saved_no_create
 
--- origsrc/h4h5tools-2.2.5/h4toh5.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/h4h5tools-2.2.5/h4toh5.pc.in	2024-10-16 02:31:00.297348500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: h4toh5
+Description: Tools for converting between HDF4 and HDF5 formats
+Version: @VERSION@
+Libs: -L${libdir} -lh4toh5
+Cflags: -I${includedir}
--- origsrc/h4h5tools-2.2.5/lib/examples/Makefile.am	2020-03-10 23:33:18.000000000 +0900
+++ src/h4h5tools-2.2.5/lib/examples/Makefile.am	2024-10-16 02:20:40.628987700 +0900
@@ -26,8 +26,8 @@ AM_CPPFLAGS+=-I$(top_builddir)/lib/src
 noinst_PROGRAMS = h4toh5example
 
 h4toh5example_SOURCES = h4toh5example.c
-h4toh5example_LDADD =  ../src/libh4toh5.a
-h4toh5example_DEPENDENCIES = ../src/libh4toh5.a
+h4toh5example_LDADD =  ../src/libh4toh5.la
+h4toh5example_DEPENDENCIES = ../src/libh4toh5.la
 
 
 
--- origsrc/h4h5tools-2.2.5/lib/src/Makefile.am	2020-03-10 23:33:18.000000000 +0900
+++ src/h4h5tools-2.2.5/lib/src/Makefile.am	2024-10-16 02:20:40.644617100 +0900
@@ -13,13 +13,15 @@
 
 AM_CPPFLAGS=-I$(top_builddir)/lib/src
 
-lib_LIBRARIES = libh4toh5.a
+lib_LTLIBRARIES = libh4toh5.la
                                                                                 
 CSOURCES = H4TOH5I.c h4toh5anno.c h4toh5pal.c h4toh5sds.c h4toh5util.c  \
         h4toh5vdata.c h4toh5vgroup.c h4toh5image.c h4toh5main.c h4toh5eos.c
                                                                                 
 CHEADERS = h4toh5.h H4TOH5api_adpt.h h4toh5vector.h h4toh5apicompatible.h
                                                                                 
-libh4toh5_a_SOURCES = $(CSOURCES)
+libh4toh5_la_SOURCES = $(CSOURCES)
 include_HEADERS = $(CHEADERS)
 
+libh4toh5_la_LDFLAGS = -no-undefined 
+libh4toh5_la_LIBADD = -ldf -lmfhdf -lhdf5_hl
--- origsrc/h4h5tools-2.2.5/lib/test/Makefile.am	2020-03-10 23:33:18.000000000 +0900
+++ src/h4h5tools-2.2.5/lib/test/Makefile.am	2024-10-16 02:20:40.644617100 +0900
@@ -22,73 +22,73 @@ noinst_PROGRAMS = h4toh5test h4h5apitest
 
 ## Information for building the "h4toh5test" program
 h4toh5test_SOURCES = h4toh5test.c
-h4toh5test_LDADD = ../src/libh4toh5.a
-h4toh5test_DEPENDENCIES = ../src/libh4toh5.a
+h4toh5test_LDADD = ../src/libh4toh5.la
+h4toh5test_DEPENDENCIES = ../src/libh4toh5.la
 
 ## Information for building the "h4h5apitestdefault" program
 h4h5apitestdefault_SOURCES = h4h5apitestdefault.c
-h4h5apitestdefault_LDADD = ../src/libh4toh5.a
-h4h5apitestdefault_DEPENDENCIES = ../src/libh4toh5.a
+h4h5apitestdefault_LDADD = ../src/libh4toh5.la
+h4h5apitestdefault_DEPENDENCIES = ../src/libh4toh5.la
 
 ## Information for building the "h4h5apitestname" program
 h4h5apitestname_SOURCES = h4h5apitestname.c
-h4h5apitestname_LDADD = ../src/libh4toh5.a
-h4h5apitestname_DEPENDENCIES = ../src/libh4toh5.a
+h4h5apitestname_LDADD = ../src/libh4toh5.la
+h4h5apitestname_DEPENDENCIES = ../src/libh4toh5.la
 
 ## Information for building the "h4h5apitestadvgroup" program
 h4h5apitestadvgroup_SOURCES = h4h5apitestadvgroup.c
-h4h5apitestadvgroup_LDADD = ../src/libh4toh5.a
-h4h5apitestadvgroup_DEPENDENCIES = ../src/libh4toh5.a
+h4h5apitestadvgroup_LDADD = ../src/libh4toh5.la
+h4h5apitestadvgroup_DEPENDENCIES = ../src/libh4toh5.la
 
 ## Information for building the "h4h5apitestsds" program
 h4h5apitestsds_SOURCES = h4h5apitestsds.c
-h4h5apitestsds_LDADD = ../src/libh4toh5.a
-h4h5apitestsds_DEPENDENCIES = ../src/libh4toh5.a
+h4h5apitestsds_LDADD = ../src/libh4toh5.la
+h4h5apitestsds_DEPENDENCIES = ../src/libh4toh5.la
 
 ## Information for building the "h4h5apitestimage" program
 h4h5apitestimage_SOURCES = h4h5apitestimage.c
-h4h5apitestimage_LDADD = ../src/libh4toh5.a
-h4h5apitestimage_DEPENDENCIES = ../src/libh4toh5.a
+h4h5apitestimage_LDADD = ../src/libh4toh5.la
+h4h5apitestimage_DEPENDENCIES = ../src/libh4toh5.la
 
 ## Information for building the "h4h5apitestvdata" program
 h4h5apitestvdata_SOURCES = h4h5apitestvdata.c
-h4h5apitestvdata_LDADD = ../src/libh4toh5.a
-h4h5apitestvdata_DEPENDENCIES = ../src/libh4toh5.a
+h4h5apitestvdata_LDADD = ../src/libh4toh5.la
+h4h5apitestvdata_DEPENDENCIES = ../src/libh4toh5.la
 
 ## Information for building the "h4h5apitestbasvgroup" program
 h4h5apitestbasvgroup_SOURCES = h4h5apitestbasvgroup.c
-h4h5apitestbasvgroup_LDADD = ../src/libh4toh5.a
-h4h5apitestbasvgroup_DEPENDENCIES = ../src/libh4toh5.a
+h4h5apitestbasvgroup_LDADD = ../src/libh4toh5.la
+h4h5apitestbasvgroup_DEPENDENCIES = ../src/libh4toh5.la
 
 ## Information for building the "h4h5apitestimageattr" program
 h4h5apitestimageattr_SOURCES = h4h5apitestimageattr.c
-h4h5apitestimageattr_LDADD = ../src/libh4toh5.a
-h4h5apitestimageattr_DEPENDENCIES = ../src/libh4toh5.a
+h4h5apitestimageattr_LDADD = ../src/libh4toh5.la
+h4h5apitestimageattr_DEPENDENCIES = ../src/libh4toh5.la
 
 ## Information for building the "h4h5apitestsdsattr" program
 h4h5apitestsdsattr_SOURCES = h4h5apitestsdsattr.c
-h4h5apitestsdsattr_LDADD = ../src/libh4toh5.a
-h4h5apitestsdsattr_DEPENDENCIES = ../src/libh4toh5.a
+h4h5apitestsdsattr_LDADD = ../src/libh4toh5.la
+h4h5apitestsdsattr_DEPENDENCIES = ../src/libh4toh5.la
 
 ## Information for building the "h4h5apitestlonesds" program
 h4h5apitestlonesds_SOURCES = h4h5apitestlonesds.c
-h4h5apitestlonesds_LDADD = ../src/libh4toh5.a
-h4h5apitestlonesds_DEPENDENCIES = ../src/libh4toh5.a
+h4h5apitestlonesds_LDADD = ../src/libh4toh5.la
+h4h5apitestlonesds_DEPENDENCIES = ../src/libh4toh5.la
 
 ## Information for building the "h4h5apitestloneimage" program
 h4h5apitestloneimage_SOURCES = h4h5apitestloneimage.c
-h4h5apitestloneimage_LDADD = ../src/libh4toh5.a
-h4h5apitestloneimage_DEPENDENCIES = ../src/libh4toh5.a
+h4h5apitestloneimage_LDADD = ../src/libh4toh5.la
+h4h5apitestloneimage_DEPENDENCIES = ../src/libh4toh5.la
 
 ## Information for building the "h4h5apitestlonevdata" program
 h4h5apitestlonevdata_SOURCES = h4h5apitestlonevdata.c
-h4h5apitestlonevdata_LDADD = ../src/libh4toh5.a
-h4h5apitestlonevdata_DEPENDENCIES = ../src/libh4toh5.a
+h4h5apitestlonevdata_LDADD = ../src/libh4toh5.la
+h4h5apitestlonevdata_DEPENDENCIES = ../src/libh4toh5.la
 
 ## Information for building the "h4toh5eostest" program
 h4toh5eostest_SOURCES = h4toh5eostest.c
-h4toh5eostest_LDADD = ../src/libh4toh5.a
-h4toh5eostest_DEPENDENCIES = ../src/libh4toh5.a
+h4toh5eostest_LDADD = ../src/libh4toh5.la
+h4toh5eostest_DEPENDENCIES = ../src/libh4toh5.la
 
 #############################################################################
 ##                              Testing                                    ##
--- origsrc/h4h5tools-2.2.5/utils/h4toh5/Makefile.am	2020-03-10 23:33:18.000000000 +0900
+++ src/h4h5tools-2.2.5/utils/h4toh5/Makefile.am	2024-10-16 02:20:40.644617100 +0900
@@ -22,8 +22,8 @@ check_PROGRAMS = h4toh5test
                                                                                 
 ## Information for building the "h4toh5test" program
 h4toh5_SOURCES = h4toh5main.c
-h4toh5_LDADD = ../../lib/src/libh4toh5.a
-h4toh5_DEPENDENCIES = ../../lib/src/libh4toh5.a
+h4toh5_LDADD = ../../lib/src/libh4toh5.la -ldf
+h4toh5_DEPENDENCIES = ../../lib/src/libh4toh5.la
                                                                                 
 #############################################################################
 ##                   Testing -- Here there be dragons.                     ##
--- origsrc/h4h5tools-2.2.5/utils/h4toh5/h4toh5main.c	2020-03-10 23:33:18.000000000 +0900
+++ src/h4h5tools-2.2.5/utils/h4toh5/h4toh5main.c	2024-10-16 02:25:16.305474400 +0900
@@ -42,7 +42,7 @@ Author:  Kent Yang(myang6@hdfgroup.org)
 #include <strings.h>
 #include <string.h>
 #endif
-
+#include <stdlib.h>
 
 /**
 * @brief no comment!
--- origsrc/h4h5tools-2.2.5/utils/h5toh4/Makefile.am	2020-03-10 23:33:18.000000000 +0900
+++ src/h4h5tools-2.2.5/utils/h5toh4/Makefile.am	2024-10-16 02:20:40.644617100 +0900
@@ -20,6 +20,7 @@ AM_CPPFLAGS = -I$(top_builddir)/lib/src/
                                                                                 
 ## Information for building the "h5toh4" program
 h5toh4_SOURCES = h5toh4.c
+h5toh4_LDADD = -ldf -lmfhdf
                                                                                 
 #############################################################################
 ##                   Testing -- Here there be dragons.                     ##
--- origsrc/h4h5tools-2.2.5/utils/h5toh4/h5toh4.c	2020-03-10 23:33:18.000000000 +0900
+++ src/h4h5tools-2.2.5/utils/h5toh4/h5toh4.c	2024-10-16 02:20:40.644617100 +0900
@@ -302,7 +302,7 @@ main(int argc, char *argv[])
   {
     (void) fprintf(stderr, err1, argc);
     PrintOptions_h5toh4();
-    (void) fprintf(stderr, err4);
+    (void) fprintf(stderr, "%s", err4);
     return EXIT_FAILURE;
   }
 
@@ -317,7 +317,7 @@ main(int argc, char *argv[])
     if ((token = gtoken(argv[i])) == TOK_ERR)
     {
       PrintOptions_h5toh4();
-      (void) fprintf(stderr, err4);
+      (void) fprintf(stderr, "%s", err4);
       return EXIT_FAILURE;
     }
 
@@ -359,9 +359,9 @@ main(int argc, char *argv[])
 
       case TOK_ERR:   /* command syntax error */
       default:
-        (void) fprintf(stderr, err2);
+        (void) fprintf(stderr, "%s", err2);
         PrintOptions_h5toh4();
-        (void) fprintf(stderr, err4);
+        (void) fprintf(stderr, "%s", err4);
         return EXIT_FAILURE;
     }
   }
