--- origsrc/cvblob/CMakeLists.txt	2011-01-26 19:48:09.000000000 +0900
+++ src/cvblob/CMakeLists.txt	2019-10-23 17:25:02.910081500 +0900
@@ -85,7 +85,7 @@ configure_file("${CMAKE_CURRENT_SOURCE_D
 
 if(UNIX)
   # For a command "FIND_PACKAGE(FOO)", CMake will look at the directory /usr/share|lib/FOO/FOOConfig.cmake, so:
-  install(FILES ${CMAKE_BINARY_DIR}/unix-install/cvBlobConfig.cmake DESTINATION ${CMAKE_INSTALL_PREFIX}/share/cvblob/)
+  install(FILES ${CMAKE_BINARY_DIR}/unix-install/cvBlobConfig.cmake DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/cmake/cvblob/)
 endif()
 
 # --------------------------------------------------------------------------------------------
--- origsrc/cvblob/cvBlob/CMakeLists.txt	2011-01-10 19:48:48.000000000 +0900
+++ src/cvblob/cvBlob/CMakeLists.txt	2019-10-23 17:24:15.734605600 +0900
@@ -34,6 +34,10 @@ set_source_files_properties(${cvBlob_SRC
 
 add_library(cvblob SHARED ${cvBlob_CVBLOB})
 
+if(CYGWIN)
+        set_target_properties(cvblob PROPERTIES SOVERSION 0)
+endif()
+
 target_link_libraries(cvblob ${OpenCV_LIBS})
 
 install(FILES cvblob.h DESTINATION include)
