--- origsrc/cvblob/CMakeLists.txt	2011-01-26 19:48:09.000000000 +0900
+++ src/cvblob/CMakeLists.txt	2025-11-24 13:25:44.826923900 +0900
@@ -85,7 +85,7 @@ configure_file("${CMAKE_CURRENT_SOURCE_D
 
 if(UNIX)
   # For a command "FIND_PACKAGE(FOO)", CMake will look at the directory /usr/share|lib/FOO/FOOConfig.cmake, so:
-  install(FILES ${CMAKE_BINARY_DIR}/unix-install/cvBlobConfig.cmake DESTINATION ${CMAKE_INSTALL_PREFIX}/share/cvblob/)
+  install(FILES ${CMAKE_BINARY_DIR}/unix-install/cvBlobConfig.cmake DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/cmake/cvblob/)
 endif()
 
 # --------------------------------------------------------------------------------------------
--- origsrc/cvblob/cvBlob/CMakeLists.txt	2011-01-10 19:48:48.000000000 +0900
+++ src/cvblob/cvBlob/CMakeLists.txt	2025-11-24 13:26:21.306108200 +0900
@@ -34,6 +34,10 @@ set_source_files_properties(${cvBlob_SRC
 
 add_library(cvblob SHARED ${cvBlob_CVBLOB})
 
+if (DEFINED SOVERSION)
+    set_target_properties(cvblob PROPERTIES SOVERSION ${SOVERSION})
+endif(DEFINED SOVERSION)
+
 target_link_libraries(cvblob ${OpenCV_LIBS})
 
 install(FILES cvblob.h DESTINATION include)
