--- origsrc/hkl/Documentation/figures/Makefile.am	2022-08-25 15:46:15.000000000 +0900
+++ src/hkl/Documentation/figures/Makefile.am	2022-08-25 15:46:56.183263600 +0900
@@ -16,9 +16,6 @@ AM_CPPFLAGS = -Wextra -D_DEFAULT_SOURCE
 	$(GOBJECT_CFLAGS) \
 	$(GSL_CFLAGS)
 
-# wee need access to the private part until the trajectory is done.
-AM_LDFLAGS = -Wl,--whole-archive,$(top_builddir)/hkl/.libs/libhkl.a,--no-whole-archive
-
 LDADD = \
 	$(top_builddir)/hkl/libhkl.la \
 	$(top_builddir)/hkl/api2/libhkl2.la \
--- origsrc/hkl/configure.ac	2022-08-25 15:46:15.000000000 +0900
+++ src/hkl/configure.ac	2022-08-25 15:58:55.194256500 +0900
@@ -62,7 +62,10 @@ AX_CXXFLAGS_WARN_ALL
 # Checks for libraries.
 AX_PATH_GSL
 AM_PATH_GLIB_2_0
-PKG_CHECK_MODULES([HDF5], [hdf5 >= 1.8.13])
+if test -z "${HDF5_CFLAGS}" -a -z "${HDF5_LIBS}"
+then
+	PKG_CHECK_MODULES([HDF5], [hdf5 >= 1.8.13])
+fi
 
 # Checks for header files.
 AC_HEADER_STDC
--- origsrc/hkl/contrib/petra3/Makefile.am	2022-08-25 15:46:15.000000000 +0900
+++ src/hkl/contrib/petra3/Makefile.am	2022-08-25 15:46:56.187211100 +0900
@@ -5,9 +5,6 @@ AM_CPPFLAGS = -Wextra -D_DEFAULT_SOURCE
 	$(GOBJECT_CFLAGS) \
 	$(GSL_CFLAGS)
 
-# wee need access to the private part until the trajectory is done.
-AM_LDFLAGS = -Wl,--whole-archive,$(top_builddir)/hkl/.libs/libhkl.a,--no-whole-archive
-
 LDADD = \
 	$(top_builddir)/hkl/libhkl.la \
 	$(top_builddir)/hkl/api2/libhkl2.la \
--- origsrc/hkl/hkl/api2/Makefile.am	2022-08-25 15:46:15.000000000 +0900
+++ src/hkl/hkl/api2/Makefile.am	2022-08-25 15:46:56.191135800 +0900
@@ -6,9 +6,6 @@ AM_CPPFLAGS = -Wextra -D_DEFAULT_SOURCE
 	$(GLIB_CFLAGS) \
 	$(GSL_CFLAGS)
 
-# need access to the private part
-AM_LDFLAGS = -Wl,--whole-archive,$(top_builddir)/hkl/.libs/libhkl.a,--no-whole-archive
-
 LDADD = \
 	$(GLIB_LIBS) \
 	$(GSL_LIBS)
--- origsrc/hkl/hkl/ccan/coroutine/coroutine.c	2022-08-25 15:46:15.000000000 +0900
+++ src/hkl/hkl/ccan/coroutine/coroutine.c	2022-08-25 15:46:56.197174300 +0900
@@ -59,9 +59,11 @@ struct coroutine_stack *coroutine_stack_
 	struct coroutine_stack *stack;
 	size_t size = bufsize - sizeof(*stack) - metasize;
 
+#if !defined(__CYGWIN__)
 #ifdef MINSIGSTKSZ
 	BUILD_ASSERT(COROUTINE_MIN_STKSZ >= MINSIGSTKSZ);
 #endif
+#endif
 
 	if (bufsize < (COROUTINE_MIN_STKSZ + sizeof(*stack) + metasize))
 		return NULL;
--- origsrc/hkl/hkl/hkl-macros-private.h	2022-08-25 15:46:16.000000000 +0900
+++ src/hkl/hkl/hkl-macros-private.h	2022-08-25 15:46:56.200206400 +0900
@@ -31,7 +31,7 @@
 #define HKL_EULERIAN_KAPPA_SOLUTION 1
 
 /* the assert method */
-#if !defined(NDEBUG) && !_MSC_VER
+#if !defined(NDEBUG) && !_MSC_VER && !defined(__CYGWIN__)
 # include <execinfo.h>
 # include <assert.h>
 # define hkl_assert(x) do{ if (!(x)) {hkl_printbt(); assert(x); } } while(0)
--- origsrc/hkl/hkl/hkl-macros.c	2022-08-25 15:46:16.000000000 +0900
+++ src/hkl/hkl/hkl-macros.c	2022-08-25 15:46:56.205241600 +0900
@@ -19,10 +19,10 @@
  *
  * Authors: Picca Frédéric-Emmanuel <picca@synchrotron-soleil.fr>
  */
-#include <execinfo.h>                   // for backtrace, etc
 #include "hkl-macros-private.h"
 
-#ifndef _MSC_VER
+#if !defined(_MSC_VER) && !defined(__CYGWIN__)
+#include <execinfo.h>                   // for backtrace, etc
 void hkl_printbt(void)
 {
 	void *array[20];
--- origsrc/hkl/tests/Makefile.am	2022-08-25 15:46:16.000000000 +0900
+++ src/hkl/tests/Makefile.am	2022-08-25 15:46:56.220189100 +0900
@@ -40,12 +40,10 @@ AM_CPPFLAGS = -Wextra -D_DEFAULT_SOURCE
 	$(GOBJECT_CFLAGS) \
 	$(GSL_CFLAGS)
 
-AM_LDFLAGS = -Wl,--whole-archive,$(top_builddir)/hkl/.libs/libhkl.a,--no-whole-archive
-
-LDADD = $(top_builddir)/hkl/libhkl.la \
-	$(top_builddir)/hkl/api2/libhkl2.la \
+LDADD = $(top_builddir)/tests/tap/libtap.a \
 	$(top_builddir)/binoculars/libhkl-binoculars.a \
-	$(top_builddir)/tests/tap/libtap.a \
+	$(top_builddir)/hkl/api2/libhkl2.la \
+	$(top_builddir)/hkl/libhkl.la \
 	$(GLIB_LIBS) \
 	$(GOBJECT_LIBS) \
 	$(GSL_LIBS)
