--- origsrc/libtpms-0.7.3/tests/Makefile.am	2020-07-10 03:26:39.000000000 +0900
+++ src/libtpms-0.7.3/tests/Makefile.am	2020-07-18 12:19:10.829290700 +0900
@@ -9,7 +9,7 @@ TESTS_ENVIRONMENT = \
   abs_top_builddir=`cd '$(top_builddir)'; pwd` \
   abs_top_srcdir=`cd '$(top_srcdir)'; pwd`
 
-AM_CFLAGS = -I$(top_srcdir)/include $(SANITIZERS)
+AM_CFLAGS = -I$(top_srcdir)/include $(SANITIZERS) -I$(top_srcdir)/include/libtpms
 AM_LDFLAGS = -ltpms -L$(top_builddir)/src/.libs $(SANITIZERS)
 
 check_PROGRAMS = \
