--- origsrc/libtpms-0.8.3/configure.ac	2021-06-01 19:39:46.000000000 +0900
+++ src/libtpms-0.8.3/configure.ac	2021-06-10 17:15:27.178928000 +0900
@@ -3,7 +3,7 @@
 #
 # See the LICENSE file for the license associated with this file.
 
-AC_INIT([libtpms], [0.8.2])
+AC_INIT([libtpms], [0.8.3])
 AC_PREREQ(2.12)
 AC_CONFIG_SRCDIR(Makefile.am)
 AC_CONFIG_AUX_DIR([.])
File origsrc/libtpms-0.8.3/include/libtpms/tpm_library.h is a regular file while file src/libtpms-0.8.3/include/libtpms/tpm_library.h is a symbolic link
--- origsrc/libtpms-0.8.3/tests/Makefile.am	2021-06-01 19:39:46.000000000 +0900
+++ src/libtpms-0.8.3/tests/Makefile.am	2021-06-10 16:58:14.288638600 +0900
@@ -9,7 +9,7 @@ TESTS_ENVIRONMENT = \
   abs_top_builddir=`cd '$(top_builddir)'; pwd` \
   abs_top_srcdir=`cd '$(top_srcdir)'; pwd`
 
-AM_CFLAGS = -I$(top_srcdir)/include $(SANITIZERS)
+AM_CFLAGS = -I$(top_srcdir)/include $(SANITIZERS) -I$(top_srcdir)/include/libtpms
 AM_LDFLAGS = -ltpms -L$(top_builddir)/src/.libs $(SANITIZERS)
 
 check_PROGRAMS = \
