--- origsrc/libtpms-0.7.0/tests/Makefile.am	2019-07-20 01:27:37.000000000 +0900
+++ src/libtpms-0.7.0/tests/Makefile.am	2019-07-25 22:54:40.385066300 +0900
@@ -9,7 +9,7 @@ TESTS_ENVIRONMENT = \
   abs_top_builddir=`cd '$(top_builddir)'; pwd` \
   abs_top_srcdir=`cd '$(top_srcdir)'; pwd`
 
-AM_CFLAGS = -I$(top_srcdir)/include $(SANITIZERS)
+AM_CFLAGS = -I$(top_srcdir)/include $(SANITIZERS) -I$(top_srcdir)/include/libtpms
 AM_LDFLAGS = -ltpms -L$(top_builddir)/src/.libs $(SANITIZERS)
 
 check_PROGRAMS = \
