--- origsrc/libmatchbox-1.11/configure.ac	2013-08-14 19:54:28.000000000 +0900
+++ src/libmatchbox-1.11/configure.ac	2014-07-24 09:01:58.071137400 +0900
@@ -156,14 +156,14 @@ fi
 dnl ------ Check for PNG ---------------------------------------------------
 
 if test x$enable_png != xno; then
-  AC_MSG_CHECKING(for libpng12)
-  if test x$PKG_CONFIG != xno && $PKG_CONFIG --exists libpng12; then 
+  AC_MSG_CHECKING(for libpng)
+  if test x$PKG_CONFIG != xno && $PKG_CONFIG --exists libpng; then 
         AC_MSG_RESULT(yes)
-        PNG_LIBS=`$PKG_CONFIG --libs libpng12`
-	PNG_CFLAGS=`$PKG_CONFIG --cflags libpng12`
+        PNG_LIBS=`$PKG_CONFIG --libs libpng`
+	PNG_CFLAGS=`$PKG_CONFIG --cflags libpng`
         AC_DEFINE(USE_PNG, [1], [Use Png])
 	SUPPORTS_PNG=1
-	PNG_REQUIRED="libpng12"
+	PNG_REQUIRED="libpng"
   else
 	AC_MSG_RESULT(no)
         # AC_CHECK_HEADERS(png.h, [ have_png_h="yes" ], [ have_png_h="no" ] )
--- origsrc/libmatchbox-1.11/libmb/Makefile.am	2013-08-14 19:54:28.000000000 +0900
+++ src/libmatchbox-1.11/libmb/Makefile.am	2014-07-24 08:58:16.176797100 +0900
@@ -27,7 +27,7 @@ libmb_la_LIBADD      = @XLIBS_LIBS@ @PAN
 # http://sources.redhat.com/autobook/autobook/autobook_91.html#SEC91
 # current : revision : age
 
-libmb_la_LDFLAGS = -version-info 1:9:0
+libmb_la_LDFLAGS = -no-undefined -version-info 1:9:0
 
 libmbheadersdir = $(includedir)/libmb
 libmbheaders_DATA = $(source_h)
