--- origsrc/tlog-8/Makefile.am	2020-04-20 21:35:56.000000000 +0900
+++ src/tlog-8/Makefile.am	2020-08-12 08:23:55.075384600 +0900
@@ -28,3 +28,6 @@ dist_noinst_DATA = tlog.spec
 clean-local:
 	find '$(top_builddir)' -name '*.gcno' -delete -or -name '*.gcda' -delete
 	-rm -rf clang_output_*
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/tlog-8/configure.ac	2020-04-20 21:35:56.000000000 +0900
+++ src/tlog-8/configure.ac	2020-08-12 08:23:55.082381600 +0900
@@ -79,7 +79,7 @@ LIBCURL_CHECK_CONFIG([yes], [7.15.4], ,
                      AC_MSG_ERROR([libcurl not found]))
 
 # Output
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([tlog.pc Makefile
                  m4/Makefile
                  m4/tlog/Makefile
                  include/Makefile
--- origsrc/tlog-8/include/tlog/misc.h	2020-04-20 21:35:56.000000000 +0900
+++ src/tlog-8/include/tlog/misc.h	2020-08-12 08:23:55.090381000 +0900
@@ -34,6 +34,11 @@
 #include <tlog/errs.h>
 #include <tlog/grc.h>
 
+#if defined(__CYGWIN__)
+#define setresgid(rgid, egid, sgid) (-1)
+#define setresuid(ruid, euid, suid) (-1)
+#endif
+
 /** Tlog version and license information */
 extern const char *tlog_version;
 
--- origsrc/tlog-8/include/tlog/pkt.h	2020-04-20 21:35:56.000000000 +0900
+++ src/tlog-8/include/tlog/pkt.h	2020-08-12 08:23:55.096385900 +0900
@@ -33,6 +33,7 @@
 #include <stdbool.h>
 #include <stdint.h>
 #include <stdlib.h>
+#include <sys/time.h>
 
 /** Packet type */
 enum tlog_pkt_type {
--- origsrc/tlog-8/include/tlog/timestr.h	2020-04-20 21:35:56.000000000 +0900
+++ src/tlog-8/include/tlog/timestr.h	2020-08-12 08:23:55.117403000 +0900
@@ -30,6 +30,7 @@
 
 #include <stdlib.h>
 #include <stdbool.h>
+#include <sys/time.h>
 
 /** The timestamp string parser state */
 struct tlog_timestr_parser {
--- origsrc/tlog-8/lib/tlog/Makefile.am	2020-04-20 21:35:56.000000000 +0900
+++ src/tlog-8/lib/tlog/Makefile.am	2020-08-12 08:23:55.123388400 +0900
@@ -159,3 +159,5 @@ libtlog_la_LIBADD = \
     -lutil \
     -lrt \
     -lm
+
+libtlog_la_LDFLAGS = -no-undefined
--- origsrc/tlog-8/tlog.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/tlog-8/tlog.pc.in	2020-08-12 08:23:55.131395100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: tlog
+Description: Terminal I/O logger
+Version: @VERSION@
+Libs: -L${libdir} -ltlog
+Cflags: -I${includedir}
