diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/validns-0.8/Makefile src/validns-0.8/Makefile
--- origsrc/validns-0.8/Makefile	2014-02-12 05:08:39.000000000 +0900
+++ src/validns-0.8/Makefile	2015-06-13 19:00:55.344307400 +0900
@@ -1,6 +1,6 @@
 # The following options seem to work fine on Linux, FreeBSD, and Darwin
 OPTIMIZE=-O2 -g
-CFLAGS=-Wall -Werror -pthread -fno-strict-aliasing
+CFLAGS=-Wall -pthread -fno-strict-aliasing
 INCPATH=-I/usr/local/include -I/opt/local/include -I/usr/local/ssl/include
 CC?=cc
 
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/validns-0.8/carp.c src/validns-0.8/carp.c
--- origsrc/validns-0.8/carp.c	2014-02-12 05:47:14.000000000 +0900
+++ src/validns-0.8/carp.c	2015-06-13 19:00:41.171821000 +0900
@@ -96,7 +96,7 @@ v(int is_croak, int use_errno, int exit_
 		exit(exit_code);
 }
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__CYGWIN__)
 static char proggy[MAXPATHLEN];
 #endif
 
@@ -108,7 +108,7 @@ const char *thisprogname(void)
 	return getprogname();
 #elif defined(__sun__)
 	return getexecname();
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__CYGWIN__)
 	if (readlink("/proc/self/exe", proggy, MAXPATHLEN) != -1)
 		return proggy;
 	return "";
