--- origsrc/ezstream-1.0.0/src/Makefile.am	2020-01-30 05:29:45.000000000 +0900
+++ src/ezstream-1.0.0/src/Makefile.am	2020-02-13 17:46:14.661357300 +0900
@@ -41,8 +41,8 @@ libezstream_la_SOURCES = \
 libezstream_la_DEPENDENCIES = \
 	$(builddir)/libcommon.la \
 	$(top_builddir)/compat/libcompat.la
-libezstream_la_LIBADD = @EZ_LIBS@ \
-	$(libezstream_la_DEPENDENCIES)
+libezstream_la_LIBADD = \
+	$(libezstream_la_DEPENDENCIES) @EZ_LIBS@
 
 bin_SCRIPTS	 = ezstream-file.sh
 bin_PROGRAMS	 = ezstream ezstream-cfgmigrate
@@ -55,8 +55,8 @@ ezstream_cfgmigrate_SOURCES = ezstream-c
 ezstream_cfgmigrate_DEPENDENCIES = \
 	$(builddir)/libcommon.la \
 	$(top_builddir)/compat/libcompat.la
-ezstream_cfgmigrate_LDADD = @EZ_LIBS@ \
-	$(ezstream_cfgmigrate_DEPENDENCIES)
+ezstream_cfgmigrate_LDADD = \
+	$(ezstream_cfgmigrate_DEPENDENCIES) @EZ_LIBS@
 
 AM_CPPFLAGS	 = @EZ_CPPFLAGS@ -I$(top_srcdir)/compat
 AM_CFLAGS	 = @EZ_CFLAGS@
