--- origsrc/yahttp-0.2.1/Makefile.am	2017-05-18 19:17:33.000000000 +0900
+++ src/yahttp-0.2.1/Makefile.am	2025-05-02 08:48:31.618730200 +0900
@@ -1,3 +1,6 @@
 ACLOCAL_AMFLAGS=-I m4
-SUBDIRS = yahttp docs tests examples
+SUBDIRS = yahttp tests examples
 EXTRA_DIST = LICENSE README.md
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/yahttp-0.2.1/configure.ac	2017-05-18 19:17:33.000000000 +0900
+++ src/yahttp-0.2.1/configure.ac	2025-05-02 09:29:29.760332500 +0900
@@ -1,6 +1,6 @@
 AC_PREREQ([2.61])
 
-AC_INIT([yahttp], [git])
+AC_INIT([yahttp], [0.2.1])
 
 AC_CONFIG_SRCDIR([yahttp/reqresp.cpp])
 AC_CONFIG_MACRO_DIR([m4])
@@ -34,5 +34,5 @@ fi
 
 AC_CHECK_PROG([DOXYGEN], [doxygen], [doxygen], [true])
 
-AC_CONFIG_FILES([Makefile yahttp/Makefile tests/Makefile examples/Makefile docs/Makefile])
+AC_CONFIG_FILES([yahttp.pc Makefile yahttp/Makefile tests/Makefile examples/Makefile docs/Makefile])
 AC_OUTPUT
--- origsrc/yahttp-0.2.1/tests/Makefile.am	2017-05-18 19:17:33.000000000 +0900
+++ src/yahttp-0.2.1/tests/Makefile.am	2025-05-02 08:48:31.618730200 +0900
@@ -3,8 +3,8 @@ EXTRA_DIST=request-get-cookies-ok.txt re
            request-issue-7.txt request-post-ok.txt response-binary.txt response-chunked-body.txt \
            response-chunked-headers.txt response-google-200.txt response-incomplete.txt request-chunked.txt
 
-test_CFLAGS=$(RELRO_CFLASG) $(PIE_CFLAGS) -I.. $(BOOST_CPPFLAGS)
-test_CXXFLAGS=$(RELRO_CFLASG) $(PIE_CFLAGS) -I.. $(BOOST_CPPFLAGS)
+test_CFLAGS=$(RELRO_CFLASG) $(PIE_CFLAGS) -I$(top_srcdir) $(BOOST_CPPFLAGS) -D_GNU_SOURCE
+test_CXXFLAGS=$(RELRO_CFLASG) $(PIE_CFLAGS) -I$(top_srcdir) $(BOOST_CPPFLAGS) -D_GNU_SOURCE
 test_LDADD=$(RELRO_LDFLAGS) $(PIE_LDFLAGS) $(BOOST_UNIT_TEST_FRAMEWORK_LDFLAGS) $(BOOST_UNIT_TEST_FRAMEWORK_LIBS) ../yahttp/libyahttp.la
 test_SOURCES=md5.h md5.c test-main.cpp test-md5.cpp test-utility.cpp test-url.cpp test-cookie.cpp test-request.cpp test-response.cpp test-router.cpp
 
--- origsrc/yahttp-0.2.1/yahttp/Makefile.am	2017-05-18 19:17:33.000000000 +0900
+++ src/yahttp-0.2.1/yahttp/Makefile.am	2025-05-02 08:48:31.618730200 +0900
@@ -1,5 +1,6 @@
 lib_LTLIBRARIES=libyahttp.la
 include_yahttpdir=$(includedir)/yahttp
 include_yahttp_HEADERS=cookie.hpp exception.hpp reqresp.hpp router.hpp url.hpp utility.hpp yahttp.hpp yahttp-config.h
-libyahttp_la_CXXFLAGS=-W -Wall $(RELRO_CFLAGS) $(PIE_CFLAGS) -D__STRICT_ANSI__
+libyahttp_la_CXXFLAGS=-W -Wall $(RELRO_CFLAGS) $(PIE_CFLAGS) -D_GNU_SOURCE
 libyahttp_la_SOURCES=cookie.hpp exception.hpp reqresp.cpp reqresp.hpp router.cpp router.hpp url.hpp utility.hpp yahttp.hpp
+libyahttp_la_LDFLAGS=-no-undefined
--- origsrc/yahttp-0.2.1/yahttp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/yahttp-0.2.1/yahttp.pc.in	2025-05-02 08:48:31.618730200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: yahttp
+Description: Yet Another HTTP Library
+Version: @VERSION@
+Libs: -L${libdir} -lyahttp
+Cflags: -I${includedir}/yahttp
