--- origsrc/flactag-2.0.4/Makefile.am	2012-06-19 21:35:44.000000000 +0900
+++ src/flactag-2.0.4/Makefile.am	2025-02-08 08:47:38.559182600 +0900
@@ -20,7 +20,7 @@ AM_CPPFLAGS += @DEP_ALL_CFLAGS@ @DEP_FLA
 #AM_LDFLAGS = -ldiscid
 
 bin_PROGRAMS = flactag discid
-flactag_LDADD = @DEP_ALL_LIBS@ @DEP_FLACTAG_LIBS@
+flactag_LDADD = @DEP_ALL_LIBS@ @DEP_FLACTAG_LIBS@ $(LTLIBICONV)
 #flactag_CFLAGS = -I/usr/include/neon
 flactag_SOURCES = flactag.cc Album.cc Track.cc AlbumWindow.cc TrackWindow.cc FlacInfo.cc TagName.cc TagsWindow.cc CuesheetTrack.cc Cuesheet.cc DiscIDWrapper.cc base64.cc ScrollableWindow.cc ConfigFile.cc MusicBrainzInfo.cc FileNameBuilder.cc ErrorLog.cc CommandLine.cc CoverArt.cc UTF8Tag.cc WriteInfo.cc
 discid_LDADD = @DEP_ALL_LIBS@
@@ -44,10 +44,10 @@ ripflac:	ripflac.in Makefile
 	chmod +x ripflac
 
 flactag.1:	flactag.1.txt Makefile
-	a2x -f manpage flactag.1.txt
+	a2x -f manpage $(srcdir)/flactag.1.txt
 
 flactag.html:	flactag.txt Makefile
-	asciidoc -a numbered flactag.txt
+	asciidoc -a numbered $(srcdir)/flactag.txt
 
 CLEANFILES = ripflac flactag.1.xml flactag.1 flactag.html
 
--- origsrc/flactag-2.0.4/configure.ac	2012-06-19 21:35:44.000000000 +0900
+++ src/flactag-2.0.4/configure.ac	2025-02-08 08:42:49.984426500 +0900
@@ -23,6 +23,8 @@ AC_PROG_LIBTOOL
 
 AC_PROG_LN_S
 
+AM_ICONV
+
 PKG_CHECK_MODULES([DEP_ALL], [libdiscid])
 PKG_CHECK_MODULES([DEP_FLACTAG], [neon slang libmusicbrainz5 flac++ unac])
 AC_SEARCH_LIBS([jpeg_read_header], jpeg, , [
