--- origsrc/tmx-tmx_1.10.1/CMakeLists.txt	2026-01-26 08:30:22.000000000 +0900
+++ src/tmx-tmx_1.10.1/CMakeLists.txt	2026-02-05 14:59:21.907383500 +0900
@@ -39,6 +39,8 @@ if(NOT STDINT_H)
     message(FATAL_ERROR "error: required header stdint.h not found")
 endif()
 
+find_package(pkgconfig)
+
 if(WANT_ZLIB AND EMSCRIPTEN)
     target_link_options(tmx INTERFACE "SHELL:-s USE_ZLIB=1")
 elseif(WANT_ZLIB)
@@ -52,12 +54,8 @@ endif()
 
 if(WANT_ZSTD)
     target_compile_definitions(tmx PRIVATE WANT_ZSTD)
-    find_package(zstd REQUIRED)
-    if (ZSTD_PREFER_STATIC)
-        target_link_libraries(tmx zstd::libzstd_static)
-    else()
-        target_link_libraries(tmx zstd::libzstd_shared)
-    endif()
+    pkg_check_modules(ZSTD libzstd)
+    target_link_libraries(tmx ${ZSTD_LIBRARIES})
 else()
     message("zstd not wanted")
 endif()
@@ -106,3 +104,10 @@ install(EXPORT tmx_exports
         DESTINATION "lib/cmake/tmx")
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/tmxConfig.cmake" "${CMAKE_CURRENT_BINARY_DIR}/tmxConfigVersion.cmake"
         DESTINATION "lib/cmake/tmx")
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/tmx.pc.in ${CMAKE_BINARY_DIR}/tmx.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/tmx.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/tmx-tmx_1.10.1/tmx.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/tmx-tmx_1.10.1/tmx.pc.in	2026-02-05 14:59:21.912398400 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: tmx
+Description: C tmx map loader
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -ltmx
+Cflags: -I${includedir}
+Requires: libxml-2.0 libzstd zlib
