--- origsrc/js0n-1.0/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/js0n-1.0/Makefile.am	2017-09-01 17:33:08.775517400 +0900
@@ -0,0 +1,15 @@
+
+check_PROGRAMS = js0n_example js0n_test
+js0n_example_SOURCES = test/example.c
+js0n_example_LDADD = libjs0n.la
+js0n_test_SOURCES = test/js0n_test.c
+js0n_test_LDADD = libjs0n.la
+
+include_HEADERS = j0g.h js0n.h
+
+lib_LTLIBRARIES = libjs0n.la
+libjs0n_la_SOURCES = j0g.c js0n.c
+libjs0n_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/js0n-1.0/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/js0n-1.0/configure.ac	2017-09-01 17:33:08.775517400 +0900
@@ -0,0 +1,27 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([js0n], [1.0])
+AC_CONFIG_SRCDIR([j0g.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+
+AC_CONFIG_FILES([js0n.pc Makefile])
+AC_OUTPUT
--- origsrc/js0n-1.0/js0n.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/js0n-1.0/js0n.pc.in	2017-09-01 17:33:26.057000000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: js0n
+Description: Flexible Zero-Footprint JSON Parser in C
+Version: @VERSION@
+Libs: -L${libdir} -ljs0n
+Cflags: -I${includedir}
