--- origsrc/tmate-2.4.0/configure.ac	2019-11-17 07:09:38.000000000 +0900
+++ src/tmate-2.4.0/configure.ac	2024-10-09 00:23:47.429981600 +0900
@@ -201,7 +201,7 @@ fi
 
 PKG_CHECK_MODULES(
   MSGPACK,
-  msgpack >= 1.1.0,
+  msgpack-c >= 1.1.0,
   [
     CPPFLAGS="$MSGPACK_CFLAGS $CPPFLAGS"
     LIBS="$MSGPACK_LIBS $LIBS"
--- origsrc/tmate-2.4.0/tmate-session.c	2019-11-17 07:09:38.000000000 +0900
+++ src/tmate-2.4.0/tmate-session.c	2024-10-09 00:17:29.601143700 +0900
@@ -9,6 +9,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <assert.h>
+#include <errno.h>
 
 #include "tmate.h"
 
--- origsrc/tmate-2.4.0/tmate-ssh-client.c	2019-11-17 07:09:38.000000000 +0900
+++ src/tmate-2.4.0/tmate-ssh-client.c	2024-10-09 00:17:29.601143700 +0900
@@ -5,6 +5,7 @@
 #include <stdlib.h>
 #include <event.h>
 #include <assert.h>
+#include <errno.h>
 
 #include "tmate.h"
 #include "window-copy.h"
