--- origsrc/tmate-2.2.1/tmate-debug.c	2016-03-29 12:30:07.000000000 +0900
+++ src/tmate-2.2.1/tmate-debug.c	2019-02-06 14:38:57.691703600 +0900
@@ -1,4 +1,6 @@
+#if !defined(__CYGWIN__)
 #include <execinfo.h>
+#endif
 #include <stdio.h>
 #include <stdlib.h>
 #include <regex.h>
@@ -55,6 +57,7 @@ static int print_resolved_stack_frame(co
 
 void tmate_print_stack_trace(void)
 {
+#if !defined(__CYGWIN__)
 	void *array[20];
 	size_t size;
 	char **strings;
@@ -73,6 +76,7 @@ void tmate_print_stack_trace(void)
 	}
 
 	free (strings);
+#endif
 }
 
 
