--- origsrc/libiqxmlrpc-0.13/libiqxmlrpc/CMakeLists.txt	2015-02-12 06:08:06.000000000 +0900
+++ src/libiqxmlrpc-0.13/libiqxmlrpc/CMakeLists.txt	2015-02-12 21:25:56.836803100 +0900
@@ -130,7 +130,12 @@ string(REPLACE ";" " " PC_BOOST_LIBRARIE
 string(REPLACE ";" " " PC_OPENSSL_LDFLAGS "${OPENSSL_LDFLAGS}")
 configure_file(../libiqxmlrpc.pc.in libiqxmlrpc.pc @ONLY)
 
-install(TARGETS iqxmlrpc DESTINATION lib)
+if (CYGWIN)
+	install(TARGETS iqxmlrpc RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+	set_target_properties(iqxmlrpc PROPERTIES SOVERSION 0)
+else (CYGWIN)
+	install(TARGETS iqxmlrpc DESTINATION lib)
+endif (CYGWIN)
 install(FILES ${PUBLIC_HEADERS} DESTINATION include/libiqxmlrpc)
 if (UNIX)
 	install(FILES ${CMAKE_BINARY_DIR}/libiqxmlrpc/libiqxmlrpc.pc DESTINATION lib/pkgconfig)
--- origsrc/libiqxmlrpc-0.13/libiqxmlrpc/inet_addr.cc	2015-02-12 06:08:06.000000000 +0900
+++ src/libiqxmlrpc-0.13/libiqxmlrpc/inet_addr.cc	2015-02-12 21:25:56.836803100 +0900
@@ -17,7 +17,7 @@ std::string get_host_name()
 }
 
 
-#if defined(WIN32) || defined(__APPLE__)
+#if defined(WIN32) || defined(__APPLE__) || defined(__CYGWIN__)
 #define IQXMLRPC_GETHOSTBYNAME(_h) \
   hent = ::gethostbyname( _h ); \
   if( !hent ) { \
