--- origsrc/libiqxmlrpc-0.13.6/libiqxmlrpc/CMakeLists.txt	2018-04-05 06:01:21.000000000 +0900
+++ src/libiqxmlrpc-0.13.6/libiqxmlrpc/CMakeLists.txt	2024-05-06 14:59:32.303706900 +0900
@@ -132,7 +132,10 @@ string(REPLACE ";" " " PC_BOOST_LIBRARIE
 string(REPLACE ";" " " PC_OPENSSL_LDFLAGS "${OPENSSL_LDFLAGS}")
 configure_file(../libiqxmlrpc.pc.in libiqxmlrpc.pc @ONLY)
 
-install(TARGETS iqxmlrpc DESTINATION lib)
+install(TARGETS iqxmlrpc RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+if (DEFINED SOVERSION)
+	set_target_properties(iqxmlrpc PROPERTIES SOVERSION "${SOVERSION}")
+endif()
 install(FILES ${PUBLIC_HEADERS} DESTINATION include/libiqxmlrpc)
 if (UNIX)
 	install(FILES ${CMAKE_BINARY_DIR}/libiqxmlrpc/libiqxmlrpc.pc DESTINATION lib/pkgconfig)
--- origsrc/libiqxmlrpc-0.13.6/libiqxmlrpc/inet_addr.cc	2018-04-05 06:01:21.000000000 +0900
+++ src/libiqxmlrpc-0.13.6/libiqxmlrpc/inet_addr.cc	2024-05-06 14:58:09.153006800 +0900
@@ -17,7 +17,7 @@ std::string get_host_name()
 }
 
 
-#if defined(WIN32) || defined(__APPLE__)
+#if defined(WIN32) || defined(__APPLE__) || defined(__CYGWIN__)
 #define IQXMLRPC_GETHOSTBYNAME(_h) \
   hent = ::gethostbyname( _h ); \
   if( !hent ) { \
