diff --strip-trailing-cr -uprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.log -x config.status -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libiqxmlrpc-0.12.2/libiqxmlrpc/CMakeLists.txt src/libiqxmlrpc-0.12.2/libiqxmlrpc/CMakeLists.txt
--- origsrc/libiqxmlrpc-0.12.2/libiqxmlrpc/CMakeLists.txt	2013-03-13 05:41:17.000000000 +0900
+++ src/libiqxmlrpc-0.12.2/libiqxmlrpc/CMakeLists.txt	2013-03-30 22:09:59.793354700 +0900
@@ -130,7 +130,12 @@ string(REPLACE ";" " " PC_BOOST_LIBRARIE
 string(REPLACE ";" " " PC_OPENSSL_LDFLAGS "${OPENSSL_LDFLAGS}")
 configure_file(../libiqxmlrpc.pc.in libiqxmlrpc.pc @ONLY)
 
-install(TARGETS iqxmlrpc DESTINATION lib)
+if (CYGWIN)
+	install(TARGETS iqxmlrpc RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+	set_target_properties(iqxmlrpc PROPERTIES SOVERSION 0)
+else (CYGWIN)
+	install(TARGETS iqxmlrpc DESTINATION lib)
+endif (CYGWIN)
 install(FILES ${PUBLIC_HEADERS} DESTINATION include/libiqxmlrpc)
 if (UNIX)
 	install(FILES ${CMAKE_BINARY_DIR}/libiqxmlrpc/libiqxmlrpc.pc DESTINATION lib/pkgconfig)
diff --strip-trailing-cr -uprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.log -x config.status -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libiqxmlrpc-0.12.2/libiqxmlrpc/inet_addr.cc src/libiqxmlrpc-0.12.2/libiqxmlrpc/inet_addr.cc
--- origsrc/libiqxmlrpc-0.12.2/libiqxmlrpc/inet_addr.cc	2013-03-13 05:41:17.000000000 +0900
+++ src/libiqxmlrpc-0.12.2/libiqxmlrpc/inet_addr.cc	2013-03-30 22:05:50.152729700 +0900
@@ -17,7 +17,7 @@ std::string iqnet::get_host_name()
 }
 
 
-#if defined(_WINDOWS) || defined(__APPLE__)
+#if defined(_WINDOWS) || defined(__APPLE__) || defined(__CYGWIN__)
 #define IQXMLRPC_GETHOSTBYNAME(_h) \
   hent = ::gethostbyname( _h ); \
   if( !hent ) { \
