--- origsrc/kokkos-3.2.00/cmake/kokkos_tribits.cmake	2020-08-26 01:41:19.000000000 +0900
+++ src/kokkos-3.2.00/cmake/kokkos_tribits.cmake	2020-09-08 10:38:28.492678000 +0900
@@ -385,6 +385,9 @@ FUNCTION(KOKKOS_ADD_LIBRARY LIBRARY_NAME
       KOKKOS_SET_LIBRARY_PROPERTIES(${LIBRARY_NAME})
     ENDIF()
   ENDIF()
+  IF (CYGWIN)
+    SET_TARGET_PROPERTIES(${LIBRARY_NAME} PROPERTIES SOVERSION 0)
+  ENDIF()
 ENDFUNCTION()
 
 FUNCTION(KOKKOS_ADD_INTERFACE_LIBRARY NAME)
--- origsrc/kokkos-3.2.00/core/src/Kokkos_Macros.hpp	2020-08-26 01:41:19.000000000 +0900
+++ src/kokkos-3.2.00/core/src/Kokkos_Macros.hpp	2020-09-08 10:39:24.630290600 +0900
@@ -665,9 +665,11 @@
 #if (defined(KOKKOS_COMPILER_GNU) || defined(KOKKOS_COMPILER_CLANG) ||  \
      defined(KOKKOS_COMPILER_INTEL) || defined(KOKKOS_COMPILER_PGI)) && \
     !defined(KOKKOS_COMPILER_MSVC)
+#if !defined(__CYGWIN__)
 #define KOKKOS_IMPL_ENABLE_STACKTRACE
 #define KOKKOS_IMPL_ENABLE_CXXABI
 #endif
+#endif
 
 // WORKAROUND for AMD aomp which apparently defines CUDA_ARCH when building for
 // AMD GPUs with OpenMP Target ???
