--- origsrc/kokkos-3.3.00/cmake/kokkos_tribits.cmake	2020-12-19 11:26:35.000000000 +0900
+++ src/kokkos-3.3.00/cmake/kokkos_tribits.cmake	2021-01-18 13:49:42.547691900 +0900
@@ -490,6 +490,9 @@ FUNCTION(KOKKOS_ADD_LIBRARY LIBRARY_NAME
       KOKKOS_SET_LIBRARY_PROPERTIES(${LIBRARY_NAME})
     ENDIF()
   ENDIF()
+  IF (CYGWIN)
+    SET_TARGET_PROPERTIES(${LIBRARY_NAME} PROPERTIES SOVERSION 0)
+  ENDIF()
 ENDFUNCTION()
 
 FUNCTION(KOKKOS_ADD_INTERFACE_LIBRARY NAME)
--- origsrc/kokkos-3.3.00/core/src/Kokkos_Macros.hpp	2020-12-19 11:26:35.000000000 +0900
+++ src/kokkos-3.3.00/core/src/Kokkos_Macros.hpp	2021-01-18 13:49:42.596716800 +0900
@@ -536,9 +536,11 @@
 #if (defined(KOKKOS_COMPILER_GNU) || defined(KOKKOS_COMPILER_CLANG) ||  \
      defined(KOKKOS_COMPILER_INTEL) || defined(KOKKOS_COMPILER_PGI)) && \
     !defined(KOKKOS_COMPILER_MSVC)
+#if !defined(__CYGWIN__)
 #define KOKKOS_IMPL_ENABLE_STACKTRACE
 #define KOKKOS_IMPL_ENABLE_CXXABI
 #endif
+#endif
 
 // WORKAROUND for AMD aomp which apparently defines CUDA_ARCH when building for
 // AMD GPUs with OpenMP Target ???
