--- origsrc/libu2f-server-libu2f-server-1.1.0/src/Makefile.am	2018-01-04 23:21:38.000000000 +0900
+++ src/libu2f-server-libu2f-server-1.1.0/src/Makefile.am	2019-10-14 09:06:24.195264000 +0900
@@ -40,10 +40,11 @@ cmdline.c cmdline.h: cmdline.ggo Makefil
 BUILT_SOURCES = cmdline.c cmdline.h
 MAINTAINERCLEANFILES = $(BUILT_SOURCES)
 
-dist_man_MANS = $(top_srcdir)/man/u2f-server.1
-DISTCLEANFILES = $(dist_man_MANS)
+man_MANS = $(top_builddir)/man/u2f-server.1
+DISTCLEANFILES = $(man_MANS)
 
-$(top_srcdir)/man/u2f-server.1: $(srcdir)/u2f-server.c $(srcdir)/cmdline.ggo $(top_srcdir)/configure.ac | $(builddir)/u2f-server$(EXEEXT)
+u2f-server.1: $(srcdir)/u2f-server.c $(srcdir)/cmdline.ggo $(top_srcdir)/configure.ac | $(builddir)/u2f-server$(EXEEXT)
+	mkdir -p $(top_builddir)/man
 	$(HELP2MAN) \
 		--output=$(top_builddir)/man/$@ $(builddir)/u2f-server$(EXEEXT) \
 		--name="Yubico Universal 2nd Factor (U2F) Server Tool" \
