--- origsrc/mednafen/include/desa68/Makefile.am	2013-09-01 06:33:42.000000000 +0900
+++ src/mednafen/include/desa68/Makefile.am	2014-04-06 17:06:14.844542700 +0900
@@ -7,7 +7,7 @@
 AUTOMAKE_OPTIONS = subdir-objects
 AM_CFLAGS = @AM_CFLAGS@ -Wno-multichar
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl
 
 noinst_LIBRARIES	=	libdesa68.a
 libdesa68_a_SOURCES	=	desa68.c
--- origsrc/mednafen/mednafen/Makefile.am	2014-01-24 02:58:20.000000000 +0900
+++ src/mednafen/mednafen/Makefile.am	2014-04-06 17:06:14.844542700 +0900
@@ -1,7 +1,7 @@
 SUBDIRS	= trio
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_builddir)/include/blip -I$(top_srcdir)/intl -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_builddir)/include/blip -I$(top_srcdir)/intl -I$(top_srcdir)
 
 bin_PROGRAMS	=	mednafen
 mednafen_SOURCES 	= 	debug.cpp error.cpp mempatcher.cpp settings.cpp endian.cpp mednafen.cpp file.cpp general.cpp md5.cpp memory.cpp netplay.cpp state.cpp movie.cpp player.cpp PSFLoader.cpp tests.cpp qtrecord.cpp cdplay.cpp okiadpcm.cpp FileWrapper.cpp Stream.cpp MemoryStream.cpp FileStream.cpp
--- origsrc/mednafen/mednafen/desa68/Makefile.am	2013-09-01 06:33:42.000000000 +0900
+++ src/mednafen/mednafen/desa68/Makefile.am	2014-04-06 17:06:14.844542700 +0900
@@ -7,7 +7,7 @@
 AUTOMAKE_OPTIONS = subdir-objects
 AM_CFLAGS = @AM_CFLAGS@ -Wno-multichar
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl
 
 noinst_LIBRARIES	=	libdesa68.a
 libdesa68_a_SOURCES	=	desa68.c
--- origsrc/mednafen/mednafen/drivers/Makefile.am	2014-01-23 08:16:04.000000000 +0900
+++ src/mednafen/mednafen/drivers/Makefile.am	2014-04-06 17:06:14.860167700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@  @SDL_CFLAGS@
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir)
 
 noinst_LIBRARIES	=	libmdfnsdl.a
 
--- origsrc/mednafen/mednafen/gb/Makefile.am	2013-09-01 06:33:40.000000000 +0900
+++ src/mednafen/mednafen/gb/Makefile.am	2014-04-06 17:06:14.860167700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@  
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)/src/hw_sound
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)/include/blip -I$(top_srcdir)/src/hw_sound
 
 noinst_LIBRARIES        =       libgb.a
 gb_SOURCES     =	gb.cpp gfx.cpp gbGlobals.cpp memory.cpp sound.cpp z80.cpp
--- origsrc/mednafen/mednafen/gba/Makefile.am	2013-09-01 06:33:40.000000000 +0900
+++ src/mednafen/mednafen/gba/Makefile.am	2014-04-06 17:06:14.875792700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ @GBA_EXTRA_FLAGS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip  -I$(top_srcdir)/src/hw_sound -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)/include/blip -I$(top_srcdir)/src/hw_sound -I$(top_srcdir)
 
 noinst_LIBRARIES        =       libgba.a
 libgba_a_SOURCES	= GBAinline.cpp arm.cpp thumb.cpp bios.cpp eeprom.cpp flash.cpp GBA.cpp Gfx.cpp Globals.cpp Mode0.cpp Mode1.cpp Mode2.cpp Mode3.cpp Mode4.cpp Mode5.cpp RTC.cpp Sound.cpp sram.cpp
--- origsrc/mednafen/mednafen/hw_cpu/Makefile.am	2013-09-01 06:33:34.000000000 +0900
+++ src/mednafen/mednafen/hw_cpu/Makefile.am	2014-04-06 17:06:14.875792700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@  
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)
 
 noinst_LIBRARIES        =       libmdfnhwcpu.a
 
@@ -15,7 +15,7 @@ gen68k_SOURCES = c68k/c68kexec.c c68k/c6
 gen68k_CFLAGS = -DC68K_GEN
 gen68k_LDFLAGS = -DC68K_GEN
 nodist_pkgdata_DATA = c68k/c68k_op0.inc
-c68k/c68k_op0.inc: c68k/*.h c68k/*.c c68k/c68k_ini.inc c68k/c68kmac.inc c68k/gen68k.inc
+c68k/c68k_op0.inc: $(srcdir)/c68k/*.h $(srcdir)/c68k/*.c c68k/c68k_ini.inc c68k/c68kmac.inc c68k/gen68k.inc
 	make gen68k && cd c68k && ../gen68k$(EXEEXT)
 
 libmdfnhwcpu_a_SOURCES 	=       c68k/c68kexec.c c68k/c68k.c
--- origsrc/mednafen/mednafen/hw_cpu/z80-fuse/z80.h	2013-09-01 06:33:36.000000000 +0900
+++ src/mednafen/mednafen/hw_cpu/z80-fuse/z80.h	2014-04-06 17:06:14.891417700 +0900
@@ -117,7 +117,7 @@ static INLINE uint8 Z80_RP_MACRO(uint16
 
 void z80_set_interrupt(int set);
 
-#include "../../mednafen/state.h"
+#include "src/state.h"
 int z80_state_action(StateMem *sm, int load, int data_only, const char *section_name);
 
 #endif			/* #ifndef FUSE_Z80_H */
--- origsrc/mednafen/mednafen/hw_sound/Makefile.am	2013-09-01 06:33:21.000000000 +0900
+++ src/mednafen/mednafen/hw_sound/Makefile.am	2014-04-06 17:06:14.907042700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)/include/blip -I$(top_srcdir)
 
 noinst_LIBRARIES        =       libmdfnhwsound.a
 
--- origsrc/mednafen/mednafen/hw_video/Makefile.am	2013-09-01 06:33:44.000000000 +0900
+++ src/mednafen/mednafen/hw_video/Makefile.am	2014-04-06 17:06:14.907042700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)
 
 noinst_LIBRARIES        =       libmdfnhwvideo.a
 
--- origsrc/mednafen/mednafen/lynx/Makefile.am	2013-09-01 06:33:45.000000000 +0900
+++ src/mednafen/mednafen/lynx/Makefile.am	2014-04-06 17:06:14.922667700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl
 
 noinst_LIBRARIES	=	liblynx.a
 liblynx_a_SOURCES 	= 	cart.cpp c65c02.cpp memmap.cpp mikie.cpp ram.cpp rom.cpp susie.cpp system.cpp
--- origsrc/mednafen/mednafen/md/Makefile.am	2013-09-01 06:33:39.000000000 +0900
+++ src/mednafen/mednafen/md/Makefile.am	2014-04-06 17:06:14.922667700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/src/hw_sound
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/src/hw_sound -I$(top_srcdir)
 
 
 noinst_LIBRARIES	=	libmd.a
--- origsrc/mednafen/mednafen/nes/Makefile.am	2013-09-01 06:33:47.000000000 +0900
+++ src/mednafen/mednafen/nes/Makefile.am	2014-04-06 17:06:14.922667700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl
 
 noinst_LIBRARIES	=	libnes.a
 mednafen_SOURCES 	= 	nes.cpp x6502.cpp cart.cpp fds.cpp ines.cpp input.cpp nsf.cpp nsfe.cpp unif.cpp vsuni.cpp sound.cpp filter.cpp fds-sound.cpp
--- origsrc/mednafen/mednafen/ngp/Makefile.am	2013-09-01 06:33:48.000000000 +0900
+++ src/mednafen/mednafen/ngp/Makefile.am	2014-04-06 17:06:14.938292700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@  -fno-strict-aliasing
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/src/ngp -ITLCS-900h  -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/intl -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/src/ngp -I$(top_srcdir)/src/ngp/TLCS-900h  -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/intl -I$(top_srcdir)
 
 noinst_LIBRARIES	=	libngp.a
 ngp_SOURCES 	=	bios.cpp biosHLE.cpp dma.cpp flash.cpp gfx.cpp T6W28_Apu.cpp	\
--- origsrc/mednafen/mednafen/pce/Makefile.am	2013-11-10 05:08:52.000000000 +0900
+++ src/mednafen/mednafen/pce/Makefile.am	2014-04-06 17:06:14.938292700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir) -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/src/hw_misc -I$(top_srcdir)/src/hw_sound -I$(top_srcdir)/src/hw_video
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir) -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/src/hw_misc -I$(top_srcdir)/src/hw_sound -I$(top_srcdir)/src/hw_video
 
 noinst_LIBRARIES	=	libpce.a
 mednafen_SOURCES 	= 	pce.cpp vce.cpp input.cpp huc.cpp pcecd.cpp hes.cpp tsushin.cpp subhw.cpp mcgenjin.cpp
--- origsrc/mednafen/mednafen/pce_fast/Makefile.am	2013-12-02 07:35:42.000000000 +0900
+++ src/mednafen/mednafen/pce_fast/Makefile.am	2014-04-06 17:06:14.953917700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/src/hw_misc -I$(top_srcdir)/src/hw_sound
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/src/hw_misc -I$(top_srcdir)/src/hw_sound
 
 noinst_LIBRARIES	=	libpce_fast.a
 mednafen_SOURCES 	= 	huc6280.cpp pce.cpp vdc.cpp input.cpp huc.cpp hes.cpp pcecd.cpp pcecd_drive.cpp psg.cpp
--- origsrc/mednafen/mednafen/pcfx/Makefile.am	2013-09-01 06:33:38.000000000 +0900
+++ src/mednafen/mednafen/pcfx/Makefile.am	2014-04-06 17:06:14.953917700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir) -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/src/hw_sound -I$(top_srcdir)/src/hw_video
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir) -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/src/hw_sound -I$(top_srcdir)/src/hw_video
 
 noinst_LIBRARIES	=	libpcfx.a
 libpcfx_a_SOURCES 	= 	king.cpp soundbox.cpp pcfx.cpp interrupt.cpp input.cpp timer.cpp rainbow.cpp jrevdct.cpp huc6273.cpp fxscsi.cpp
--- origsrc/mednafen/mednafen/psx/Makefile.am	2013-09-01 06:33:33.000000000 +0900
+++ src/mednafen/mednafen/psx/Makefile.am	2014-04-06 17:06:14.953917700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@  @MMX_CFLAGS@ @SSE_CFLAGS@ @SSE2_CFLAGS@ -funroll-loops
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir)
 
 noinst_LIBRARIES	=	libpsx.a
 libpsx_a_SOURCES 	= 	psx.cpp irq.cpp timer.cpp dma.cpp frontio.cpp sio.cpp cpu.cpp gte.cpp dis.cpp cdc.cpp spu.cpp gpu.cpp mdec.cpp
--- origsrc/mednafen/mednafen/sms/Makefile.am	2013-09-01 06:33:39.000000000 +0900
+++ src/mednafen/mednafen/sms/Makefile.am	2014-04-06 17:06:14.969542700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/src/hw_sound
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/src/hw_sound -I$(top_srcdir)
 
 noinst_LIBRARIES        =       libsms.a
 
--- origsrc/mednafen/mednafen/snes/Makefile.am	2013-09-01 06:33:26.000000000 +0900
+++ src/mednafen/mednafen/snes/Makefile.am	2014-04-06 17:06:14.969542700 +0900
@@ -2,7 +2,7 @@ AM_CFLAGS = @AM_CFLAGS@ @SNES_EXTRA_FLAG
 AM_CXXFLAGS = @AM_CXXFLAGS@ @SNES_EXTRA_FLAGS@ @SNES_EXTRA_CXXFLAGS@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@  -DNOMINMAX
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/src/snes/src/lib
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/src/snes/src/lib
 
 noinst_LIBRARIES	=	libsnes.a
 
--- origsrc/mednafen/mednafen/sound/Makefile.am	2013-11-10 05:10:00.000000000 +0900
+++ src/mednafen/mednafen/sound/Makefile.am	2014-04-06 17:06:14.985167700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)/include/blip
 
 noinst_LIBRARIES	=	libmdfnsound.a
 
--- origsrc/mednafen/mednafen/trio/Makefile.am	2013-09-01 06:33:48.000000000 +0900
+++ src/mednafen/mednafen/trio/Makefile.am	2014-04-06 17:06:14.985167700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ @TRIO_CFLAGS@
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl
 
 noinst_LIBRARIES	=	libtrio.a
 libtrio_a_SOURCES 	= 	trio.c trionan.c triostr.c
--- origsrc/mednafen/mednafen/vb/Makefile.am	2013-09-01 06:33:26.000000000 +0900
+++ src/mednafen/mednafen/vb/Makefile.am	2014-04-06 17:06:15.000792700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@  -fno-strict-aliasing
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir)
 
 noinst_LIBRARIES	=	libvb.a
 libvb_a_SOURCES 	= 	vb.cpp timer.cpp input.cpp vip.cpp vsu.cpp
--- origsrc/mednafen/mednafen/wswan/Makefile.am	2013-09-01 06:33:40.000000000 +0900
+++ src/mednafen/mednafen/wswan/Makefile.am	2014-04-06 17:06:15.000792700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl
 
 noinst_LIBRARIES	=	libwswan.a
 
--- origsrc/mednafen/src/Makefile.am	2014-01-24 02:58:20.000000000 +0900
+++ src/mednafen/src/Makefile.am	2014-04-06 17:06:14.844542700 +0900
@@ -1,7 +1,7 @@
 SUBDIRS	= trio
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_builddir)/include/blip -I$(top_srcdir)/intl -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_builddir)/include/blip -I$(top_srcdir)/intl -I$(top_srcdir)
 
 bin_PROGRAMS	=	mednafen
 mednafen_SOURCES 	= 	debug.cpp error.cpp mempatcher.cpp settings.cpp endian.cpp mednafen.cpp file.cpp general.cpp md5.cpp memory.cpp netplay.cpp state.cpp movie.cpp player.cpp PSFLoader.cpp tests.cpp qtrecord.cpp cdplay.cpp okiadpcm.cpp FileWrapper.cpp Stream.cpp MemoryStream.cpp FileStream.cpp
--- origsrc/mednafen/src/desa68/Makefile.am	2013-09-01 06:33:42.000000000 +0900
+++ src/mednafen/src/desa68/Makefile.am	2014-04-06 17:06:14.844542700 +0900
@@ -7,7 +7,7 @@
 AUTOMAKE_OPTIONS = subdir-objects
 AM_CFLAGS = @AM_CFLAGS@ -Wno-multichar
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl
 
 noinst_LIBRARIES	=	libdesa68.a
 libdesa68_a_SOURCES	=	desa68.c
--- origsrc/mednafen/src/drivers/Makefile.am	2014-01-23 08:16:04.000000000 +0900
+++ src/mednafen/src/drivers/Makefile.am	2014-04-06 17:06:14.860167700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@  @SDL_CFLAGS@
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir)
 
 noinst_LIBRARIES	=	libmdfnsdl.a
 
--- origsrc/mednafen/src/gb/Makefile.am	2013-09-01 06:33:40.000000000 +0900
+++ src/mednafen/src/gb/Makefile.am	2014-04-06 17:06:14.860167700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@  
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)/src/hw_sound
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)/include/blip -I$(top_srcdir)/src/hw_sound
 
 noinst_LIBRARIES        =       libgb.a
 gb_SOURCES     =	gb.cpp gfx.cpp gbGlobals.cpp memory.cpp sound.cpp z80.cpp
--- origsrc/mednafen/src/gba/Makefile.am	2013-09-01 06:33:40.000000000 +0900
+++ src/mednafen/src/gba/Makefile.am	2014-04-06 17:06:14.875792700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ @GBA_EXTRA_FLAGS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip  -I$(top_srcdir)/src/hw_sound -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)/include/blip -I$(top_srcdir)/src/hw_sound -I$(top_srcdir)
 
 noinst_LIBRARIES        =       libgba.a
 libgba_a_SOURCES	= GBAinline.cpp arm.cpp thumb.cpp bios.cpp eeprom.cpp flash.cpp GBA.cpp Gfx.cpp Globals.cpp Mode0.cpp Mode1.cpp Mode2.cpp Mode3.cpp Mode4.cpp Mode5.cpp RTC.cpp Sound.cpp sram.cpp
--- origsrc/mednafen/src/hw_cpu/Makefile.am	2013-09-01 06:33:34.000000000 +0900
+++ src/mednafen/src/hw_cpu/Makefile.am	2014-04-06 17:06:14.875792700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@  
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)
 
 noinst_LIBRARIES        =       libmdfnhwcpu.a
 
@@ -15,7 +15,7 @@ gen68k_SOURCES = c68k/c68kexec.c c68k/c6
 gen68k_CFLAGS = -DC68K_GEN
 gen68k_LDFLAGS = -DC68K_GEN
 nodist_pkgdata_DATA = c68k/c68k_op0.inc
-c68k/c68k_op0.inc: c68k/*.h c68k/*.c c68k/c68k_ini.inc c68k/c68kmac.inc c68k/gen68k.inc
+c68k/c68k_op0.inc: $(srcdir)/c68k/*.h $(srcdir)/c68k/*.c c68k/c68k_ini.inc c68k/c68kmac.inc c68k/gen68k.inc
 	make gen68k && cd c68k && ../gen68k$(EXEEXT)
 
 libmdfnhwcpu_a_SOURCES 	=       c68k/c68kexec.c c68k/c68k.c
--- origsrc/mednafen/src/hw_cpu/z80-fuse/z80.h	2013-09-01 06:33:36.000000000 +0900
+++ src/mednafen/src/hw_cpu/z80-fuse/z80.h	2014-04-06 17:06:14.891417700 +0900
@@ -117,7 +117,7 @@ static INLINE uint8 Z80_RP_MACRO(uint16
 
 void z80_set_interrupt(int set);
 
-#include "../../mednafen/state.h"
+#include "src/state.h"
 int z80_state_action(StateMem *sm, int load, int data_only, const char *section_name);
 
 #endif			/* #ifndef FUSE_Z80_H */
--- origsrc/mednafen/src/hw_sound/Makefile.am	2013-09-01 06:33:21.000000000 +0900
+++ src/mednafen/src/hw_sound/Makefile.am	2014-04-06 17:06:14.907042700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)/include/blip -I$(top_srcdir)
 
 noinst_LIBRARIES        =       libmdfnhwsound.a
 
--- origsrc/mednafen/src/hw_video/Makefile.am	2013-09-01 06:33:44.000000000 +0900
+++ src/mednafen/src/hw_video/Makefile.am	2014-04-06 17:06:14.907042700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)
 
 noinst_LIBRARIES        =       libmdfnhwvideo.a
 
--- origsrc/mednafen/src/lynx/Makefile.am	2013-09-01 06:33:45.000000000 +0900
+++ src/mednafen/src/lynx/Makefile.am	2014-04-06 17:06:14.922667700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl
 
 noinst_LIBRARIES	=	liblynx.a
 liblynx_a_SOURCES 	= 	cart.cpp c65c02.cpp memmap.cpp mikie.cpp ram.cpp rom.cpp susie.cpp system.cpp
--- origsrc/mednafen/src/md/Makefile.am	2013-09-01 06:33:39.000000000 +0900
+++ src/mednafen/src/md/Makefile.am	2014-04-06 17:06:14.922667700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/src/hw_sound
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/src/hw_sound -I$(top_srcdir)
 
 
 noinst_LIBRARIES	=	libmd.a
--- origsrc/mednafen/src/nes/Makefile.am	2013-09-01 06:33:47.000000000 +0900
+++ src/mednafen/src/nes/Makefile.am	2014-04-06 17:06:14.922667700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl
 
 noinst_LIBRARIES	=	libnes.a
 mednafen_SOURCES 	= 	nes.cpp x6502.cpp cart.cpp fds.cpp ines.cpp input.cpp nsf.cpp nsfe.cpp unif.cpp vsuni.cpp sound.cpp filter.cpp fds-sound.cpp
--- origsrc/mednafen/src/ngp/Makefile.am	2013-09-01 06:33:48.000000000 +0900
+++ src/mednafen/src/ngp/Makefile.am	2014-04-06 17:06:14.938292700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@  -fno-strict-aliasing
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/src/ngp -ITLCS-900h  -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/intl -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/src/ngp -I$(top_srcdir)/src/ngp/TLCS-900h  -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/intl -I$(top_srcdir)
 
 noinst_LIBRARIES	=	libngp.a
 ngp_SOURCES 	=	bios.cpp biosHLE.cpp dma.cpp flash.cpp gfx.cpp T6W28_Apu.cpp	\
--- origsrc/mednafen/src/pce/Makefile.am	2013-11-10 05:08:52.000000000 +0900
+++ src/mednafen/src/pce/Makefile.am	2014-04-06 17:06:14.938292700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir) -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/src/hw_misc -I$(top_srcdir)/src/hw_sound -I$(top_srcdir)/src/hw_video
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir) -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/src/hw_misc -I$(top_srcdir)/src/hw_sound -I$(top_srcdir)/src/hw_video
 
 noinst_LIBRARIES	=	libpce.a
 mednafen_SOURCES 	= 	pce.cpp vce.cpp input.cpp huc.cpp pcecd.cpp hes.cpp tsushin.cpp subhw.cpp mcgenjin.cpp
--- origsrc/mednafen/src/pce_fast/Makefile.am	2013-12-02 07:35:42.000000000 +0900
+++ src/mednafen/src/pce_fast/Makefile.am	2014-04-06 17:06:14.953917700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/src/hw_misc -I$(top_srcdir)/src/hw_sound
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/src/hw_misc -I$(top_srcdir)/src/hw_sound
 
 noinst_LIBRARIES	=	libpce_fast.a
 mednafen_SOURCES 	= 	huc6280.cpp pce.cpp vdc.cpp input.cpp huc.cpp hes.cpp pcecd.cpp pcecd_drive.cpp psg.cpp
--- origsrc/mednafen/src/pcfx/Makefile.am	2013-09-01 06:33:38.000000000 +0900
+++ src/mednafen/src/pcfx/Makefile.am	2014-04-06 17:06:14.953917700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir) -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/src/hw_sound -I$(top_srcdir)/src/hw_video
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir) -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/src/hw_sound -I$(top_srcdir)/src/hw_video
 
 noinst_LIBRARIES	=	libpcfx.a
 libpcfx_a_SOURCES 	= 	king.cpp soundbox.cpp pcfx.cpp interrupt.cpp input.cpp timer.cpp rainbow.cpp jrevdct.cpp huc6273.cpp fxscsi.cpp
--- origsrc/mednafen/src/psx/Makefile.am	2013-09-01 06:33:33.000000000 +0900
+++ src/mednafen/src/psx/Makefile.am	2014-04-06 17:06:14.953917700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@  @MMX_CFLAGS@ @SSE_CFLAGS@ @SSE2_CFLAGS@ -funroll-loops
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir)
 
 noinst_LIBRARIES	=	libpsx.a
 libpsx_a_SOURCES 	= 	psx.cpp irq.cpp timer.cpp dma.cpp frontio.cpp sio.cpp cpu.cpp gte.cpp dis.cpp cdc.cpp spu.cpp gpu.cpp mdec.cpp
--- origsrc/mednafen/src/sms/Makefile.am	2013-09-01 06:33:39.000000000 +0900
+++ src/mednafen/src/sms/Makefile.am	2014-04-06 17:06:14.969542700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/src/hw_sound
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/src/hw_cpu -I$(top_srcdir)/src/hw_sound -I$(top_srcdir)
 
 noinst_LIBRARIES        =       libsms.a
 
--- origsrc/mednafen/src/snes/Makefile.am	2013-09-01 06:33:26.000000000 +0900
+++ src/mednafen/src/snes/Makefile.am	2014-04-06 17:06:14.969542700 +0900
@@ -2,7 +2,7 @@ AM_CFLAGS = @AM_CFLAGS@ @SNES_EXTRA_FLAG
 AM_CXXFLAGS = @AM_CXXFLAGS@ @SNES_EXTRA_FLAGS@ @SNES_EXTRA_CXXFLAGS@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@  -DNOMINMAX
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/src/snes/src/lib
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/src/snes/src/lib
 
 noinst_LIBRARIES	=	libsnes.a
 
--- origsrc/mednafen/src/sound/Makefile.am	2013-11-10 05:10:00.000000000 +0900
+++ src/mednafen/src/sound/Makefile.am	2014-04-06 17:06:14.985167700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_builddir)/include/blip -I$(top_srcdir)/include/blip
 
 noinst_LIBRARIES	=	libmdfnsound.a
 
--- origsrc/mednafen/src/trio/Makefile.am	2013-09-01 06:33:48.000000000 +0900
+++ src/mednafen/src/trio/Makefile.am	2014-04-06 17:06:14.985167700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ @TRIO_CFLAGS@
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl
 
 noinst_LIBRARIES	=	libtrio.a
 libtrio_a_SOURCES 	= 	trio.c trionan.c triostr.c
--- origsrc/mednafen/src/vb/Makefile.am	2013-09-01 06:33:26.000000000 +0900
+++ src/mednafen/src/vb/Makefile.am	2014-04-06 17:06:15.000792700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@  -fno-strict-aliasing
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl -I$(top_srcdir)
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir)
 
 noinst_LIBRARIES	=	libvb.a
 libvb_a_SOURCES 	= 	vb.cpp timer.cpp input.cpp vip.cpp vsu.cpp
--- origsrc/mednafen/src/wswan/Makefile.am	2013-09-01 06:33:40.000000000 +0900
+++ src/mednafen/src/wswan/Makefile.am	2014-04-06 17:06:15.000792700 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = subdir-objects
 DEFS = -DLOCALEDIR=\"$(datadir)/locale\" @DEFS@ 
-DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/intl
+DEFAULT_INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I$(top_srcdir)/intl
 
 noinst_LIBRARIES	=	libwswan.a
 
