diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/Box2D-2.3.1/Box2D/Box2D/CMakeLists.txt src/Box2D-2.3.1/Box2D/Box2D/CMakeLists.txt
--- origsrc/Box2D-2.3.1/Box2D/Box2D/CMakeLists.txt	2014-04-06 09:43:12.000000000 +0900
+++ src/Box2D-2.3.1/Box2D/Box2D/CMakeLists.txt	2016-03-20 20:59:43.275136600 +0900
@@ -145,6 +145,7 @@ if(BOX2D_BUILD_SHARED)
 		OUTPUT_NAME "Box2D"
 		CLEAN_DIRECT_OUTPUT 1
 		VERSION ${BOX2D_VERSION}
+		SOVERSION ${BOX2D_VERSION}
 	)
 endif()
 
@@ -169,6 +170,7 @@ if(BOX2D_BUILD_STATIC)
 	set_target_properties(Box2D PROPERTIES
 		CLEAN_DIRECT_OUTPUT 1
 		VERSION ${BOX2D_VERSION}
+		SOVERSION ${BOX2D_VERSION}
 	)
 endif()
 
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/Box2D-2.3.1/Box2D/CMakeLists.txt src/Box2D-2.3.1/Box2D/CMakeLists.txt
--- origsrc/Box2D-2.3.1/Box2D/CMakeLists.txt	2014-04-06 09:43:12.000000000 +0900
+++ src/Box2D-2.3.1/Box2D/CMakeLists.txt	2016-03-20 21:00:18.904047900 +0900
@@ -14,7 +14,7 @@ option(BOX2D_BUILD_SHARED "Build Box2D s
 option(BOX2D_BUILD_STATIC "Build Box2D static libraries" ON)
 option(BOX2D_BUILD_EXAMPLES "Build Box2D examples" ON)
 
-set(BOX2D_VERSION 2.3.0)
+set(BOX2D_VERSION 2.3.1)
 set(LIB_INSTALL_DIR lib${LIB_SUFFIX})
 
 # The Box2D library.
@@ -26,8 +26,6 @@ if(BOX2D_BUILD_EXAMPLES)
 
   # Testbed and dependencies.
   find_package(OpenGL REQUIRED)
-  add_subdirectory(freeglut)
-  add_subdirectory(glui)
   add_subdirectory(Testbed)
 endif(BOX2D_BUILD_EXAMPLES)
 
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/Box2D-2.3.1/Box2D/HelloWorld/CMakeLists.txt src/Box2D-2.3.1/Box2D/HelloWorld/CMakeLists.txt
--- origsrc/Box2D-2.3.1/Box2D/HelloWorld/CMakeLists.txt	2014-04-06 09:43:12.000000000 +0900
+++ src/Box2D-2.3.1/Box2D/HelloWorld/CMakeLists.txt	2016-03-20 20:47:48.444743600 +0900
@@ -1,4 +1,4 @@
 # Hello World examples
 include_directories (${Box2D_SOURCE_DIR})
 add_executable(HelloWorld HelloWorld.cpp)
-target_link_libraries (HelloWorld Box2D)
+target_link_libraries (HelloWorld Box2D_shared)
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/Box2D-2.3.1/Box2D/Testbed/CMakeLists.txt src/Box2D-2.3.1/Box2D/Testbed/CMakeLists.txt
--- origsrc/Box2D-2.3.1/Box2D/Testbed/CMakeLists.txt	2014-04-06 09:43:12.000000000 +0900
+++ src/Box2D-2.3.1/Box2D/Testbed/CMakeLists.txt	2016-03-20 20:58:40.151323900 +0900
@@ -4,8 +4,8 @@ add_definitions( -DFREEGLUT_EXPORTS -DFR
 # Define the framework files.
 set(Testbed_Framework_SRCS
 	Framework/Main.cpp
-	Framework/Render.cpp
-	Framework/Render.h
+	Framework/RenderGL3.cpp
+	Framework/RenderGL3.h
 	Framework/Test.cpp
 	Framework/Test.h
 )
@@ -84,9 +84,7 @@ add_executable(Testbed
 
 target_link_libraries (
 	Testbed
-	Box2D
-	freeglut_static
-	glui
+	Box2D_shared
 	${ADDITIONAL_LIBRARIES}
 	${OPENGL_LIBRARIES}
 ) 
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/Box2D-2.3.1/Box2D/glfw/glfw3.h src/Box2D-2.3.1/Box2D/glfw/glfw3.h
--- origsrc/Box2D-2.3.1/Box2D/glfw/glfw3.h	2014-04-06 09:43:12.000000000 +0900
+++ src/Box2D-2.3.1/Box2D/glfw/glfw3.h	2016-03-20 20:53:06.254887100 +0900
@@ -79,7 +79,7 @@ extern "C" {
  * to the list of "valid Win32 identifiers", which removes the need for
  * -mwin32)
  */
-#if !defined(_WIN32) && (defined(__WIN32__) || defined(WIN32) || defined(__CYGWIN__))
+#if !defined(_WIN32) && (defined(__WIN32__) || defined(WIN32))
  #define _WIN32
 #endif /* _WIN32 */
 
@@ -2278,4 +2278,3 @@ GLFWAPI GLFWglproc glfwGetProcAddress(co
 #endif
 
 #endif /* _glfw3_h_ */
-
