--- origsrc/box2d-2.4.1/Box2D.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/box2d-2.4.1/Box2D.pc.in	2020-10-18 15:44:14.233216200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: Box2D
+Description: 2D Physics Engine for Games
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lBox2D
+Cflags: -I${includedir}
--- origsrc/box2d-2.4.1/CMakeLists.txt	2020-10-18 09:09:34.000000000 +0900
+++ src/box2d-2.4.1/CMakeLists.txt	2020-10-18 15:41:59.351296300 +0900
@@ -63,3 +63,10 @@ install(
   DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/include/box2d"
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 )
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/Box2D.pc.in ${CMAKE_BINARY_DIR}/Box2D.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/Box2D.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
