--- origsrc/Box2D_v2.3.0/Box2D/Box2D/CMakeLists.txt	2013-10-23 09:15:59.000000000 +0900
+++ src/Box2D_v2.3.0/Box2D/Box2D/CMakeLists.txt	2013-11-25 08:07:45.770516400 +0900
@@ -145,6 +145,7 @@ if(BOX2D_BUILD_SHARED)
 		OUTPUT_NAME "Box2D"
 		CLEAN_DIRECT_OUTPUT 1
 		VERSION ${BOX2D_VERSION}
+		SOVERSION ${BOX2D_VERSION}
 	)
 endif()
 
@@ -195,7 +196,7 @@ if(BOX2D_INSTALL)
 
 	# install libraries
 	if(BOX2D_BUILD_SHARED)
-		install(TARGETS Box2D_shared EXPORT Box2D-targets DESTINATION ${LIB_INSTALL_DIR})
+		install(TARGETS Box2D_shared EXPORT Box2D-targets RUNTIME DESTINATION bin ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
 	endif()
 	if(BOX2D_BUILD_STATIC)
 		install(TARGETS Box2D EXPORT Box2D-targets DESTINATION ${LIB_INSTALL_DIR})
--- origsrc/Box2D_v2.3.0/Box2D/Testbed/CMakeLists.txt	2011-09-07 09:47:46.000000000 +0900
+++ src/Box2D_v2.3.0/Box2D/Testbed/CMakeLists.txt	2013-11-25 08:06:26.876961700 +0900
@@ -84,7 +84,7 @@ add_executable(Testbed
 
 target_link_libraries (
 	Testbed
-	Box2D
+	$<TARGET_FILE:Box2D_shared>
 	freeglut_static
 	glui
 	${ADDITIONAL_LIBRARIES}
--- origsrc/Box2D_v2.3.0/Box2D/freeglut/freeglut_main.c	2011-09-05 12:06:42.000000000 +0900
+++ src/Box2D_v2.3.0/Box2D/freeglut/freeglut_main.c	2013-11-25 07:44:49.079274000 +0900
@@ -38,6 +38,7 @@
 #else
 #    define VFPRINTF(s,f,a)
 #endif
+#include <sys/time.h>
 
 #ifdef _WIN32_WCE
 
