--- origsrc/gtkdatabox-0.9.3.1/examples/Makefile.am	2018-06-30 05:20:11.000000000 +0900
+++ src/gtkdatabox-0.9.3.1/examples/Makefile.am	2019-09-07 17:18:43.291571900 +0900
@@ -52,7 +52,7 @@ CLEANFILES = basics_libglade.h
 basics_libglade.h: basics_libglade.glade
 	echo "#pragma GCC diagnostic ignored \"-Woverlength-strings\"" > $@
 	echo "const char * basics_glade = \"\\" >> $@
-	sed -e 's/"/\\"/g' -e 's/$$/\\/' basics_libglade.glade >> $@
+	sed -e 's/"/\\"/g' -e 's/$$/\\/' $(srcdir)/basics_libglade.glade >> $@
 	echo "\";" >> $@
 
 AM_CPPFLAGS		= -I$(top_srcdir)/gtk
--- origsrc/gtkdatabox-0.9.3.1/gtk/Makefile.am	2018-11-03 02:08:46.000000000 +0900
+++ src/gtkdatabox-0.9.3.1/gtk/Makefile.am	2019-09-07 17:13:32.633922700 +0900
@@ -18,7 +18,7 @@ libgtkdatabox_la_SOURCES= \
             gtkdatabox_grid.c\
 			gtkdatabox_ruler.c
 
-libgtkdatabox_la_LDFLAGS= \
+libgtkdatabox_la_LDFLAGS= -no-undefined \
 		        -version-info $(LT_CURRENT)\
 			-release $(LT_RELEASE)\
 			@GTK_LIBS@
