--- origsrc/https_dns_proxy/CMakeLists.txt	2026-01-11 22:26:45.000000000 +0900
+++ src/https_dns_proxy/CMakeLists.txt	2026-01-11 22:35:07.207540900 +0900
@@ -44,7 +44,9 @@ set(SERVICE_TYPE "simple")
 # It is possible to define external default value like: cmake -DSW_VERSION=1.2-custom
 
 find_package(Git)
-if(Git_FOUND)
+if(DEFINED GIT_VERSION)
+  set(SW_VERSION "${GIT_VERSION}")
+elseif(Git_FOUND AND EXISTS "${CMAKE_SOURCE_DIR}/.git")
   execute_process(
     COMMAND "${GIT_EXECUTABLE}" show --date=format:%Y.%m.%d --format=%ad-%h --no-patch
     WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
--- origsrc/https_dns_proxy/src/stat.h	2026-01-11 23:37:42.447031400 +0900
+++ src/https_dns_proxy/src/stat.h	2026-01-11 23:43:26.328494700 +0900
@@ -10,8 +10,8 @@
 // stat_connection_(open|closed|reused) update the tallies.
 //
 
-#ifndef _STAT_H_
-#define _STAT_H_
+#ifndef _HTTPS_DNS_PROXY_STAT_H_
+#define _HTTPS_DNS_PROXY_STAT_H_
 
 #include <stdint.h>
 #include <ev.h>
@@ -54,4 +54,4 @@ void stat_stop(stat_t *s);
 
 void stat_cleanup(stat_t *s);
 
-#endif // _STAT_H_
+#endif // _HTTPS_DNS_PROXY_STAT_H_
