--- origsrc/https_dns_proxy/CMakeLists.txt	2018-08-19 11:07:55.000000000 +0900
+++ src/https_dns_proxy/CMakeLists.txt	2018-08-19 11:08:45.932452300 +0900
@@ -30,7 +30,9 @@ set(TARGET_NAME "https_dns_proxy")
 aux_source_directory(src SRC_LIST)
 set(SRC_LIST ${SRC_LIST})
 add_executable(${TARGET_NAME} ${SRC_LIST})
-add_subdirectory(lib)
+if(NOT CYGWIN)
+  add_subdirectory(lib)
+endif()
 set(LIBS ${LIBS} cares curl ev resolv nxjson)
 target_link_libraries(${TARGET_NAME} ${LIBS})
 
