--- origsrc/mctc-lib-0.3.2/CMakeLists.txt	2024-08-29 21:42:31.000000000 +0900
+++ src/mctc-lib-0.3.2/CMakeLists.txt	2025-01-10 04:45:56.902268700 +0900
@@ -94,6 +94,7 @@ install(
   "${PROJECT_NAME}-lib"
   EXPORT
   "${PROJECT_NAME}-targets"
+  RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
   LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
   ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
 )
--- origsrc/mctc-lib-0.3.2/config/CMakeLists.txt	2024-08-29 21:42:31.000000000 +0900
+++ src/mctc-lib-0.3.2/config/CMakeLists.txt	2025-01-10 04:45:56.902268700 +0900
@@ -23,7 +23,7 @@ set(
   CACHE STRING
   "Subdirectory to install generated module files to"
 )
-set(
+option(
   module-dir
   "${${PROJECT_NAME}-module-dir}"
 )
