--- origsrc/mctc-lib-0.3.0/CMakeLists.txt	2022-04-13 18:44:42.000000000 +0900
+++ src/mctc-lib-0.3.0/CMakeLists.txt	2022-05-13 20:48:37.844241900 +0900
@@ -86,6 +86,7 @@ install(
   "${PROJECT_NAME}-lib"
   EXPORT
   "${PROJECT_NAME}-targets"
+  RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
   LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
   ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
 )
@@ -100,11 +101,19 @@ set(
   module-id
   "${CMAKE_Fortran_COMPILER_ID}-${CMAKE_Fortran_COMPILER_VERSION}"
 )
-install(
-  DIRECTORY
-  "${PROJECT_BINARY_DIR}/include/"
-  DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}/${module-id}"
-)
+if (CYGWIN)
+  install(
+    DIRECTORY
+    "${PROJECT_BINARY_DIR}/include/"
+    DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${module-dir}"
+  )
+else()
+  install(
+    DIRECTORY
+    "${PROJECT_BINARY_DIR}/include/"
+    DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}/${module-id}"
+  )
+endif()
 # Package license files
 install(
   FILES
--- origsrc/mctc-lib-0.3.0/config/CMakeLists.txt	2022-04-13 18:44:42.000000000 +0900
+++ src/mctc-lib-0.3.0/config/CMakeLists.txt	2022-05-13 17:04:16.975690900 +0900
@@ -17,10 +17,14 @@ option(BUILD_SHARED_LIBS "Whether the li
 option(WITH_OpenMP "Enable support for shared memory parallelisation with OpenMP" TRUE)
 option(WITH_JSON "Enable support for JSON parsing via json-fortran" FALSE)
 
-set(
-  module-dir
-  "${PROJECT_NAME}/${CMAKE_Fortran_COMPILER_ID}-${CMAKE_Fortran_COMPILER_VERSION}"
-)
+if (CYGWIN)
+  set(module-dir "${PROJECT_NAME}")
+else()
+  set(
+    module-dir
+    "${PROJECT_NAME}/${CMAKE_Fortran_COMPILER_ID}-${CMAKE_Fortran_COMPILER_VERSION}"
+  )
+endif()
 set(module-dir "${module-dir}" PARENT_SCOPE)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
