--- origsrc/mctc-lib-0.3.1/CMakeLists.txt	2022-08-01 23:01:52.000000000 +0900
+++ src/mctc-lib-0.3.1/CMakeLists.txt	2022-12-02 17:43:13.618487500 +0900
@@ -86,6 +86,7 @@ install(
   "${PROJECT_NAME}-lib"
   EXPORT
   "${PROJECT_NAME}-targets"
+  RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
   LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
   ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
 )
--- origsrc/mctc-lib-0.3.1/config/CMakeLists.txt	2022-08-01 23:01:52.000000000 +0900
+++ src/mctc-lib-0.3.1/config/CMakeLists.txt	2022-12-02 17:44:56.472334300 +0900
@@ -23,7 +23,7 @@ set(
   CACHE STRING
   "Subdirectory to install generated module files to"
 )
-set(
+option(
   module-dir
   "${${PROJECT_NAME}-module-dir}"
 )
