--- origsrc/elektra-0.8.22/cmake/Modules/LibAddLib.cmake	2018-02-28 04:06:57.000000000 +0900
+++ src/elektra-0.8.22/cmake/Modules/LibAddLib.cmake	2018-04-14 11:18:19.526000000 +0900
@@ -30,7 +30,7 @@ function(add_lib name)
 		target_link_libraries (elektra-${name}
 			${ARG_LINK_LIBRARIES})
 
-		install (TARGETS elektra-${name} DESTINATION lib${LIB_SUFFIX} EXPORT ElektraTargetsLibelektra)
+		install (TARGETS elektra-${name} EXPORT ElektraTargetsLibelektra RUNTIME DESTINATION bin ARCHIVE DESTINATION lib${LIB_SUFFIX})
 	endif (BUILD_SHARED)
 
 endfunction()
--- origsrc/elektra-0.8.22/scripts/CMakeLists.txt	2018-02-28 04:06:57.000000000 +0900
+++ src/elektra-0.8.22/scripts/CMakeLists.txt	2018-04-14 11:20:10.698000000 +0900
@@ -83,6 +83,10 @@ if (INSTALL_SYSTEM_FILES)
 		install (FILES kdb_zsh_completion
 			DESTINATION /usr/local/share/zsh/site-functions
 			RENAME _kdb)
+	elseif (CYGWIN)
+		install (FILES kdb_zsh_completion
+			DESTINATION /usr/share/zsh/site-functions
+			RENAME _kdb)
 	else ()
 		# This Folder is at least the valid folder for Debian
 		install (FILES kdb_zsh_completion
--- origsrc/elektra-0.8.22/src/bindings/glib/CMakeLists.txt	2018-02-28 04:06:57.000000000 +0900
+++ src/elektra-0.8.22/src/bindings/glib/CMakeLists.txt	2018-04-14 15:44:38.283000000 +0900
@@ -36,9 +36,14 @@ else ()
 	target_link_libraries (${GELEKTRA_LIBRARY} PUBLIC elektra-core elektra-kdb)
 	target_link_libraries (${GELEKTRA_LIBRARY} PUBLIC ${GOBJECT_LDFLAGS})
 
+	if (CYGWIN)
+		set_target_properties (${GELEKTRA_LIBRARY} PROPERTIES SOVERSION ${SO_VERSION})
+	endif()
+
 	install (
 		TARGETS ${GELEKTRA_LIBRARY}
-		LIBRARY DESTINATION lib${LIB_SUFFIX}
+		RUNTIME DESTINATION bin
+		ARCHIVE DESTINATION lib${LIB_SUFFIX}
 	)
 
 	install (
--- origsrc/elektra-0.8.22/src/bindings/gsettings/CMakeLists.txt	2018-02-28 04:06:57.000000000 +0900
+++ src/elektra-0.8.22/src/bindings/gsettings/CMakeLists.txt	2018-04-14 11:47:51.309000000 +0900
@@ -65,7 +65,7 @@ if (GELEKTRA_LIBRARY)
 	include_directories (${CMAKE_BINARY_DIR}/src/include/)
 	target_link_libraries (elektrasettings ${GLIB_LIBRARIES} ${GMODULE_LIBRARIES} ${GIO_LIBRARIES} ${GELEKTRA_LIBRARY} elektra-core)
 	if (INSTALL_SYSTEM_FILES)
-		install (TARGETS elektrasettings LIBRARY DESTINATION ${GIO_MODULE_DIR})
+		install (TARGETS elektrasettings RUNTIME DESTINATION ${GIO_MODULE_DIR})
 	endif ()
 else ()
 	if (CMAKE_VERSION VERSION_LESS 3.4)
--- origsrc/elektra-0.8.22/src/bindings/io/uv/CMakeLists.txt	2018-02-28 04:06:57.000000000 +0900
+++ src/elektra-0.8.22/src/bindings/io/uv/CMakeLists.txt	2018-04-14 14:28:36.075000000 +0900
@@ -39,9 +39,12 @@ else ()
 		set_target_properties (${IO_UV_LIBRARY} PROPERTIES APPEND PROPERTY COMPILE_DEFINITIONS "HAVE_LIBUV1")
 	endif ()
 
+	set_target_properties(${IO_UV_LIBRARY} PROPERTIES SOVERSION ${SO_VERSION})
+
 	install (
 		TARGETS ${IO_UV_LIBRARY}
-		LIBRARY DESTINATION lib${LIB_SUFFIX}
+		RUNTIME DESTINATION bin
+		ARCHIVE DESTINATION lib${LIB_SUFFIX}
 	)
 
 	install (
--- origsrc/elektra-0.8.22/src/libs/elektra/CMakeLists.txt	2018-02-28 04:06:57.000000000 +0900
+++ src/elektra-0.8.22/src/libs/elektra/CMakeLists.txt	2018-04-14 11:22:07.542000000 +0900
@@ -134,7 +134,7 @@ if (BUILD_SHARED)
 	set (ELEKTRA_RPATH "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/${TARGET_PLUGIN_FOLDER}")
 	set_target_properties (${elektra-all_LIBRARIES} PROPERTIES INSTALL_RPATH "${ELEKTRA_RPATH}")
 
-	install (TARGETS elektra-core elektra-kdb elektra DESTINATION lib${LIB_SUFFIX} EXPORT ElektraTargetsLibelektra)
+	install (TARGETS elektra-core elektra-kdb elektra EXPORT ElektraTargetsLibelektra RUNTIME DESTINATION bin ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
 	list (APPEND targets_built ${elektra-all_LIBRARIES})
 endif (BUILD_SHARED)
--- origsrc/elektra-0.8.22/src/libs/tools/src/CMakeLists.txt	2018-02-28 04:06:57.000000000 +0900
+++ src/elektra-0.8.22/src/libs/tools/src/CMakeLists.txt	2018-04-14 11:23:00.136000000 +0900
@@ -29,7 +29,7 @@ if (BUILD_SHARED)
 		)
 	endif ()
 
-	install (TARGETS elektratools DESTINATION lib${LIB_SUFFIX})
+	install (TARGETS elektratools RUNTIME DESTINATION bin ARCHIVE DESTINATION lib${LIB_SUFFIX})
 endif (BUILD_SHARED)
 
 if (BUILD_FULL)
--- origsrc/elektra-0.8.22/src/plugins/CMakeLists.txt	2018-02-28 04:06:57.000000000 +0900
+++ src/elektra-0.8.22/src/plugins/CMakeLists.txt	2018-04-14 11:23:42.558000000 +0900
@@ -69,7 +69,7 @@ if (output EQUAL -1)
 endif()
 
 
-if (BUILD_SHARED)
+if (BUILD_SHARED AND NOT CYGWIN)
 	mkdir (${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
 	create_lib_symlink (
 			libelektra-${KDB_DEFAULT_STORAGE}.so
@@ -82,5 +82,5 @@ if (BUILD_SHARED)
 			libelektra-resolver.so
 			PLUGIN
 			)
-endif (BUILD_SHARED)
+endif (BUILD_SHARED AND NOT CYGWIN)
 
--- origsrc/elektra-0.8.22/src/plugins/fstab/testmntent.c	2018-02-28 04:06:57.000000000 +0900
+++ src/elektra-0.8.22/src/plugins/fstab/testmntent.c	2018-04-14 11:24:20.134000000 +0900
@@ -23,6 +23,6 @@ int main (void)
 	opts = m->mnt_opts;
 	freq = m->mnt_freq;
 	passno = m->mnt_passno;
-
+	addmntent(f, m);
 	return 0;
 }
