--- origsrc/libsignon-glib/libsignon-glib/Makefile.am	2018-08-24 15:23:44.000000000 +0900
+++ src/libsignon-glib/libsignon-glib/Makefile.am	2018-08-24 15:27:22.100848000 +0900
@@ -12,6 +12,7 @@ libsignon_glib_la_CFLAGS = \
 	-Wall -Werror -Wno-error=deprecated-declarations
 libsignon_glib_la_LIBADD = $(DEPS_LIBS)
 libsignon_glib_la_LDFLAGS = \
+	-no-undefined \
 	$(COVERAGE_LDFLAGS) \
 	-version-info 1:0:0 \
 	-export-symbols-regex '^signon_'
@@ -216,7 +217,7 @@ VAPIGEN_VAPIS = libsignon-glib.vapi
 
 libsignon_glib_vapi_DEPS = gio-2.0
 libsignon_glib_vapi_METADATADIRS = $(srcdir)
-libsignon_glib_vapi_FILES = Signon-1.0.gir Signon-1.0.metadata
+libsignon_glib_vapi_FILES = Signon-1.0.gir $(srcdir)/Signon-1.0.metadata
 
 vapidir = $(datadir)/vala/vapi
 vapi_DATA = $(VAPIGEN_VAPIS)
