--- origsrc/libsignon-glib-VERSION_1.14-e90302e342bfd27bc8c9132ab9d0ea3d8723fd03/libsignon-glib/Makefile.am	2016-05-20 00:21:34.000000000 +0900
+++ src/libsignon-glib-VERSION_1.14-e90302e342bfd27bc8c9132ab9d0ea3d8723fd03/libsignon-glib/Makefile.am	2017-07-07 07:22:02.733000000 +0900
@@ -12,6 +12,7 @@ libsignon_glib_la_CFLAGS = \
 	-Wall -Werror -Wno-error=deprecated-declarations
 libsignon_glib_la_LIBADD = $(DEPS_LIBS)
 libsignon_glib_la_LDFLAGS = \
+	-no-undefined \
 	$(COVERAGE_LDFLAGS) \
 	-version-info 1:0:0 \
 	-export-symbols-regex '^signon_'
