--- origsrc/MASA-0.50.0/examples/Makefile.am	2017-02-28 01:45:35.000000000 +0900
+++ src/MASA-0.50.0/examples/Makefile.am	2019-01-02 22:23:13.923747800 +0900
@@ -2,7 +2,7 @@ AM_CPPFLAGS   = -I$(top_srcdir)/src -I$(
 AM_FCFLAGS    = -I$(top_srcdir)/src -I$(top_builddir)/src
 LIBS          = -L$(top_builddir)/src -lmasa
 
-exampledir    = @prefix@/examples
+exampledir    = @prefix@/share/MASA/examples
 
 # License Information
 BUILT_SOURCES = .license.stamp
--- origsrc/MASA-0.50.0/src/Makefile.am	2017-02-28 01:45:35.000000000 +0900
+++ src/MASA-0.50.0/src/Makefile.am	2019-01-02 22:08:12.993592600 +0900
@@ -33,7 +33,7 @@ library_include_HEADERS = masa.h
 # MASA C/C++ library
 #-----------------------
 
-libmasa_la_LDFLAGS      = $(all_libraries) -release $(GENERIC_RELEASE)
+libmasa_la_LDFLAGS      = $(all_libraries) -release $(GENERIC_RELEASE) -no-undefined
 libmasa_la_CPPFLAGS     = $(masa_optional_INCLUDES)
 libmasa_la_SOURCES      = $(cc_sources) $(h_sources)
 
@@ -44,9 +44,10 @@ libmasa_la_SOURCES      = $(cc_sources)
 if FORT_ENABLED
   CLEANFILES             += masa.mod 
   lib_LTLIBRARIES        += libfmasa.la
-  libfmasa_la_LDFLAGS     = $(all_libraries) -release $(GENERIC_RELEASE)
+  libfmasa_la_LDFLAGS     = $(all_libraries) -release $(GENERIC_RELEASE) -no-undefined
   libfmasa_la_CPPFLAGS    = $(masa_optional_INCLUDES)
   libfmasa_la_SOURCES     = masa.f90 $(libmasa_la_SOURCES)
+  libfmasa_la_LIBADD      = -lgfortran
 endif
 
 #-----------------------
@@ -61,7 +62,7 @@ if SWIG_ENABLED
   pkgpyexec_LTLIBRARIES = _masa.la
   _masa_la_SOURCES      = masa_wrap.c
   _masa_la_CFLAGS       = $(AX_SWIG_PYTHON_CPPFLAGS) -I$(top_srcdir)/src
-  _masa_la_LDFLAGS      = -module
+  _masa_la_LDFLAGS      = -module -no-undefined
   _masa_la_LIBADD       = libmasa.la
 
   # Produce two built sources from a single masa.i source
