--- origsrc/MASA-0.51.0/examples/Makefile.am	2024-11-05 07:20:01.000000000 +0900
+++ src/MASA-0.51.0/examples/Makefile.am	2025-03-17 10:11:35.776594800 +0900
@@ -2,7 +2,7 @@ AM_CPPFLAGS   = -I$(top_srcdir)/src -I$(
 AM_FCFLAGS    = -I$(top_srcdir)/src -I$(top_builddir)/src
 LIBS          = -L$(top_builddir)/src -lmasa
 
-exampledir    = @prefix@/examples
+exampledir    = @prefix@/share/masa/examples
 
 # License Information
 BUILT_SOURCES = .license.stamp
--- origsrc/MASA-0.51.0/src/Makefile.am	2024-11-05 07:20:01.000000000 +0900
+++ src/MASA-0.51.0/src/Makefile.am	2025-03-17 11:21:49.749081100 +0900
@@ -36,7 +36,7 @@ library_include_HEADERS = masa.h
 # MASA C/C++ library
 #-----------------------
 
-libmasa_la_LDFLAGS      = $(all_libraries) -release $(GENERIC_RELEASE)
+libmasa_la_LDFLAGS      = $(all_libraries) -release $(GENERIC_RELEASE) -no-undefined
 libmasa_la_CPPFLAGS     = $(masa_optional_INCLUDES)
 libmasa_la_SOURCES      = $(cc_sources) $(h_sources)
 
@@ -47,9 +47,10 @@ libmasa_la_SOURCES      = $(cc_sources)
 if FORT_ENABLED
   CLEANFILES             += masa.mod 
   lib_LTLIBRARIES        += libfmasa.la
-  libfmasa_la_LDFLAGS     = $(all_libraries) -release $(GENERIC_RELEASE)
+  libfmasa_la_LDFLAGS     = $(all_libraries) -release $(GENERIC_RELEASE) -no-undefined
   libfmasa_la_CPPFLAGS    = $(masa_optional_INCLUDES)
   libfmasa_la_SOURCES     = masa.f90 $(libmasa_la_SOURCES)
+  libfmasa_la_LIBADD      = -lgfortran
 endif
 
 #-----------------------
@@ -64,7 +65,7 @@ if SWIG_ENABLED
   pkgpyexec_LTLIBRARIES = _masa.la
   _masa_la_SOURCES      = masa_wrap.c
   _masa_la_CFLAGS       = $(AX_SWIG_PYTHON_CPPFLAGS) -I$(top_srcdir)/src
-  _masa_la_LDFLAGS      = -module
+  _masa_la_LDFLAGS      = -module -no-undefined
   _masa_la_LIBADD       = libmasa.la
 
   # Produce two built sources from a single masa.i source
@@ -108,11 +109,11 @@ bin_PROGRAMS                           =
 if FORT_ENABLED
 
 install-data-local: libfmasa.la
-	mkdir -p $(DESTDIR)$(libdir)
-	$(INSTALL_DATA) $(top_builddir)/src/masa.mod $(DESTDIR)$(libdir)/masa.mod
+	mkdir -p $(DESTDIR)$(includedir)
+	$(INSTALL_DATA) $(top_builddir)/src/masa.mod $(DESTDIR)$(includedir)/masa.mod
 
 uninstall-local: libfmasa.la
-	rm -rf $(DESTDIR)$(libdir)/masa.mod
+	rm -rf $(DESTDIR)$(includedir)/masa.mod
 endif
 
 EXTRA_DIST   = lic_utils/update_license.pl .license.stamp masa.i
