--- origsrc/grap-1.49/Makefile.am	2024-01-07 08:56:03.000000000 +0900
+++ src/grap-1.49/Makefile.am	2025-06-08 12:26:06.266761900 +0900
@@ -24,12 +24,12 @@ dist_man1_MANS = grap.doc
 
 # Where to put grap.defines
 definesdir = @DEFINES_DIR@
-dist_defines_DATA = grap*.defines
+dist_defines_DATA = $(top_srcdir)/grap*.defines
 
 # Examples
 exampledir = @EXAMPLE_DIR@
-dist_example_DATA = examples/*.d examples/*.result examples/Makefile \
-		    examples/example.ms
+dist_example_DATA = $(top_srcdir)/examples/*.d $(top_srcdir)/examples/*.result $(top_srcdir)/examples/Makefile \
+		    $(top_srcdir)/examples/example.ms
 
 # Other docs
 docdir = @DOC_DIR@
@@ -40,7 +40,7 @@ AM_CXXFLAGS = @GXXFLAGS@ @CX0FLAGS@
 
 # Rules for y.tab.h and grap.man
 y.tab.h:	grap.yy
-	${YACC} -d grap.yy && rm -f y.tab.c
+	${YACC} -d $(top_srcdir)/grap.yy && rm -f y.tab.c
 
 grap.man:	grap.doc
 	groff -mdoc -Tascii grap.doc > grap.man
