--- origsrc/libstroke-0.5.1/Makefile.am	2023-10-07 14:09:33.558358400 +0900
+++ src/libstroke-0.5.1/Makefile.am	2023-10-07 14:15:40.327914100 +0900
@@ -17,3 +17,5 @@ CLEANFILES = core *~
 DISTCLEANFILES = core *~
 MAINTAINERCLEANFILES = core *~
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libstroke-0.5.1/configure.in	2023-10-07 14:09:33.558358400 +0900
+++ src/libstroke-0.5.1/configure.in	2023-10-07 14:19:21.006713900 +0900
@@ -1,14 +1,8 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(libstroke/stroke.c)
+AC_INIT([libstroke], [0.5.1])
 
 dnl Initialize automake stuff
-PACKAGE=libstroke
-
-dnl version
-LIBSTROKE_VERSION=0.5
-
-dnl Initialize automake stuff
-AM_INIT_AUTOMAKE($PACKAGE, $LIBSTROKE_VERSION, no-define)
+AM_INIT_AUTOMAKE
 
 dnl Create a configuration header
 AM_CONFIG_HEADER(config.h)
@@ -105,7 +99,7 @@ dnl 	  libgstroke/Makefile \
 dnl           tests/Makefile )
 dnl fi
 
-AC_OUTPUT( Makefile
+AC_OUTPUT( Makefile libstroke.pc
   libstroke/Makefile \
   libgstroke/Makefile \
   doc/Makefile \
--- origsrc/libstroke-0.5.1/libstroke/Makefile.am	2023-10-07 14:09:33.230233200 +0900
+++ src/libstroke-0.5.1/libstroke/Makefile.am	2023-10-07 14:10:01.732081900 +0900
@@ -29,7 +29,7 @@ lib_LTLIBRARIES = libstroke.la
 #libstroke_tcl_la_LDFLAGS = -ltcl -version-info 0:5:0
 libstroke_la_SOURCES = stroke.c stroke.h
 
-libstroke_la_LDFLAGS = -version-info 0:5:0
+libstroke_la_LDFLAGS = -no-undefined -version-info 0:5:0
 
 include_HEADERS = stroke.h
 
--- origsrc/libstroke-0.5.1/libstroke.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libstroke-0.5.1/libstroke.pc.in	2023-10-07 14:14:16.700073500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libstroke
+Description: Stroke (mouse gesture) translation library
+Version: @VERSION@
+Libs: -L${libdir} -lstroke
+Cflags: -I${includedir}
