--- origsrc/libstroke-0.5.1/configure.in	2001-08-07 15:08:05.000000000 +0900
+++ src/libstroke-0.5.1/configure.in	2010-11-28 16:11:30.490244500 +0900
@@ -17,10 +17,7 @@ dnl Build time sanity check...
 AM_SANITY_CHECK
 
 dnl Initialize libtool
-AM_PROG_LIBTOOL
-
-dnl Initialize maintainer mode
-AM_MAINTAINER_MODE
+LT_INIT(win32-dll)
 
 dnl Checks for programs.
 AC_PROG_CC
--- origsrc/libstroke-0.5.1/libgstroke/Makefile.am	2010-11-28 16:11:03.261090900 +0900
+++ src/libstroke-0.5.1/libgstroke/Makefile.am	2010-11-28 16:12:36.435068500 +0900
@@ -11,15 +11,14 @@ EXTRA_DIST =
 
 lib_LTLIBRARIES = libgstroke.la
 libgstroke_la_SOURCES = gnome-stroke.c gnome-stroke-draw.c gstroke.h
-
-libgstroke_la_LDFLAGS = -version-info 0:5:0
+libgstroke_la_LIBADD = @GTK_LIBS@
+libgstroke_la_LDFLAGS = -version-info 0:5:0 -no-undefined
 
 include_HEADERS = gstroke.h
 
 noinst_HEADERS = gstroke-internal.h
 
 INCLUDES = @GTK_CFLAGS@
-LDADD    = @GTK_LIBS@
 
 MOSTLYCLEANFILES = core *~ 
 CLEANFILES = core *~ 
--- origsrc/libstroke-0.5.1/libstroke/Makefile.am	2010-11-28 16:11:03.261090900 +0900
+++ src/libstroke-0.5.1/libstroke/Makefile.am	2010-11-28 16:11:49.928194900 +0900
@@ -26,7 +26,7 @@ lib_LTLIBRARIES = libstroke.la
 #libstroke_tcl_la_LDFLAGS = -ltcl -version-info 0:5:0
 libstroke_la_SOURCES = stroke.c stroke.h
 
-libstroke_la_LDFLAGS = -version-info 0:5:0
+libstroke_la_LDFLAGS = -version-info 0:5:0 -no-undefined
 
 include_HEADERS = stroke.h
 
