--- origsrc/librdkafka-1.1.0/CMakeLists.txt	2019-06-24 05:49:32.000000000 +0900
+++ src/librdkafka-1.1.0/CMakeLists.txt	2019-06-25 06:48:08.107058300 +0900
@@ -241,9 +241,15 @@ write_basic_package_version_file(
 )
 
 install(
-    FILES "${project_config}" "${project_version}" "packaging/cmake/Modules/FindLZ4.cmake"
+    FILES "${project_config}" "${project_version}"
     DESTINATION "${config_install_dir}"
 )
+if (NOT CYGWIN)
+  install(
+    FILES "packaging/cmake/Modules/FindLZ4.cmake"
+    DESTINATION "${config_install_dir}"
+  )
+endif()
 
 install(
     EXPORT "${targets_export_name}"
@@ -251,10 +257,12 @@ install(
     DESTINATION "${config_install_dir}"
 )
 
-install(
-    FILES LICENSES.txt
-    DESTINATION "share/licenses/librdkafka"
-)
+if (NOT CYGWIN)
+  install(
+      FILES LICENSES.txt
+      DESTINATION "share/licenses/librdkafka"
+  )
+endif()
 
 # }
 
--- origsrc/librdkafka-1.1.0/tests/CMakeLists.txt	2019-06-24 05:49:32.000000000 +0900
+++ src/librdkafka-1.1.0/tests/CMakeLists.txt	2019-06-25 06:35:26.421195400 +0900
@@ -106,7 +106,7 @@ add_test(NAME RdKafkaTestInParallel COMM
 add_test(NAME RdKafkaTestSequentially COMMAND rdkafka_test -p1)
 add_test(NAME RdKafkaTestBrokerLess COMMAND rdkafka_test -p5 -l)
 
-if(NOT WIN32 AND NOT APPLE)
+if(NOT WIN32 AND NOT APPLE AND NOT CYGWIN)
   set(tests_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
   add_subdirectory(interceptor_test)
 endif()
