--- origsrc/rhonabwy-1.1.6/CMakeLists.txt	2022-05-29 21:42:30.000000000 +0900
+++ src/rhonabwy-1.1.6/CMakeLists.txt	2022-07-01 10:48:56.247454500 +0900
@@ -98,6 +98,18 @@ if (NETTLE_FOUND)
   include_directories(${NETTLE_INCLUDE_DIRS})
 endif ()
 
+pkg_check_modules(HOGWEED hogweed)
+if (HOGWEED_FOUND)
+    set(LIBS ${LIBS} ${HOGWEED_LIBRARIES})
+    include_directories(${HOGWEED_INCLUDE_DIRS})
+endif()
+
+pkg_check_modules(GMP gmp)
+if (GMP_FOUND)
+    set(LIBS ${LIBS} ${GMP_LIBRARIES})
+    include_directories(${GMP_INCLUDE_DIRS})
+endif()
+
 include(FindGnuTLS)
 find_package(GnuTLS REQUIRED)
 if (GNUTLS_FOUND)
