--- origsrc/rhonabwy-1.1.8/CMakeLists.txt	2022-08-21 23:06:45.000000000 +0900
+++ src/rhonabwy-1.1.8/CMakeLists.txt	2022-08-24 13:19:21.801145300 +0900
@@ -98,6 +98,18 @@ if (NETTLE_FOUND)
   include_directories(${NETTLE_INCLUDE_DIRS})
 endif ()
 
+pkg_check_modules(HOGWEED hogweed)
+if (HOGWEED_FOUND)
+    set(LIBS ${LIBS} ${HOGWEED_LIBRARIES})
+    include_directories(${HOGWEED_INCLUDE_DIRS})
+endif()
+
+pkg_check_modules(GMP gmp)
+if (GMP_FOUND)
+    set(LIBS ${LIBS} ${GMP_LIBRARIES})
+    include_directories(${GMP_INCLUDE_DIRS})
+endif()
+
 include(FindGnuTLS)
 find_package(GnuTLS REQUIRED)
 if (GNUTLS_FOUND)
