--- origsrc/rhonabwy-1.1.11/CMakeLists.txt	2023-02-08 23:06:47.000000000 +0900
+++ src/rhonabwy-1.1.11/CMakeLists.txt	2023-03-13 16:11:55.762617000 +0900
@@ -98,6 +98,18 @@ if (NETTLE_FOUND)
   include_directories(${NETTLE_INCLUDE_DIRS})
 endif ()
 
+pkg_check_modules(HOGWEED hogweed)
+if (HOGWEED_FOUND)
+    set(LIBS ${LIBS} ${HOGWEED_LIBRARIES})
+    include_directories(${HOGWEED_INCLUDE_DIRS})
+endif()
+
+pkg_check_modules(GMP gmp)
+if (GMP_FOUND)
+    set(LIBS ${LIBS} ${GMP_LIBRARIES})
+    include_directories(${GMP_INCLUDE_DIRS})
+endif()
+
 include(FindGnuTLS)
 find_package(GnuTLS REQUIRED)
 if (GNUTLS_FOUND)
