--- origsrc/rhonabwy-1.1.13/CMakeLists.txt	2023-11-15 01:38:17.000000000 +0900
+++ src/rhonabwy-1.1.13/CMakeLists.txt	2024-10-10 16:57:24.027708600 +0900
@@ -20,7 +20,7 @@ project(rhonabwy C)
 
 set(CMAKE_C_STANDARD 99)
 if (NOT MSVC)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
 endif()
 
 # library info
@@ -124,6 +124,18 @@ else ()
     set(R_WITH_CURL OFF)
 endif ()
 
+pkg_check_modules(HOGWEED hogweed)
+if (HOGWEED_FOUND)
+    list(APPEND RHONABWY_LIBS ${HOGWEED_LIBRARIES})
+    include_directories(${HOGWEED_INCLUDE_DIRS})
+endif()
+
+pkg_check_modules(GMP gmp)
+if (GMP_FOUND)
+    list(APPEND RHONABWY_LIBS ${GMP_LIBRARIES})
+    include_directories(${GMP_INCLUDE_DIRS})
+endif()
+
 # directories and source
 
 set(INC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include)
