--- origsrc/nfft-3.1.3/Makefile.am	2012-05-02 12:34:53.041323900 +0900
+++ src/nfft-3.1.3/Makefile.am	2012-05-02 12:48:40.099917600 +0900
@@ -22,13 +22,13 @@ lib_LTLIBRARIES = libnfft3.la
 
 libnfft3_la_SOURCES =
 libnfft3_la_LIBADD = 3rdparty/lib3rdparty.la util/libutil.la kernel/libkernel.la @fftw3_LIBS@ -lm
-libnfft3_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ @fftw3_LDFLAGS@
+libnfft3_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ @fftw3_LDFLAGS@ -no-undefined
 
 EXTRA_DIST = bootstrap.sh VERSION doxygen.c
 
 install-data-hook:
 	mkdir -p $(DESTDIR)$(pkgdatadir)
-	cp -R doc $(DESTDIR)$(pkgdatadir)
+	cp -R $(top_srcdir)/doc $(DESTDIR)$(pkgdatadir)
 
 maintainer-clean-local:
 	rm -f doc/api/html/*.html
@@ -46,9 +46,9 @@ dist-hook:
 	doxygen doxygen/doxygen.Doxyfile
 	mkdir -p $(distdir)/doc
 	mkdir -p $(distdir)/doc/api
-	cp -R doc/api/html $(distdir)/doc/api
-	cp -R doc/api/latex $(distdir)/doc/api
-	cp -R doc/tutorial $(distdir)/doc
+	cp -R $(top_srcdir)/doc/api/html $(distdir)/doc/api
+	cp -R $(top_srcdir)/doc/api/latex $(distdir)/doc/api
+	cp -R $(top_srcdir)/doc/tutorial $(distdir)/doc
 	rm -Rf $(distdir)/doc/.svn
 	rm -Rf $(distdir)/doc/tutorial/.svn
 	rm -Rf $(distdir)/doc/api/html/.svn
