--- origsrc/nfft-3.5.1/Makefile.am	2019-08-09 17:11:32.000000000 +0900
+++ src/nfft-3.5.1/Makefile.am	2019-12-23 18:54:11.051205400 +0900
@@ -88,6 +88,7 @@ endif
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = nfft3@PREC_SUFFIX@.pc
 
+if DX_COND_doc
 install-data-hook:
 	if test -f "$(abs_top_builddir)/doc/html/index.html"; then \
 		$(MKDIR_P) $(DESTDIR)$(docdir); \
@@ -100,6 +101,7 @@ install-data-hook:
 uninstall-hook:
 	chmod -Rf u+rwX $(DESTDIR)$(docdir)
 	rm -Rf $(DESTDIR)$(docdir)
+endif
 
 clean-local:
 	rm -f $(abs_top_builddir)/doc/doxygen_sqlite3.db
