--- origsrc/libhsplasma/CMakeLists.txt	2014-09-22 22:58:13.000000000 +0900
+++ src/libhsplasma/CMakeLists.txt	2014-09-22 23:34:54.443867400 +0900
@@ -10,7 +10,7 @@ set(CMAKE_C_FLAGS_DEBUG "-DDEBUG ${CMAKE
 set(CMAKE_CXX_FLAGS_DEBUG "-DDEBUG ${CMAKE_CXX_FLAGS_DEBUG}")
 if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG)
     set(WARNING_FLAGS "-Wall -Wextra -Wno-unused-parameter")
-    set(CMAKE_CXX_FLAGS "-std=c++0x ${WARNING_FLAGS} ${CMAKE_CXX_FLAGS}")
+    set(CMAKE_CXX_FLAGS "-std=gnu++0x ${WARNING_FLAGS} ${CMAKE_CXX_FLAGS}")
     set(CMAKE_C_FLAGS "${WARNING_FLAGS} ${CMAKE_C_FLAGS}")
 endif()
 
--- origsrc/libhsplasma/core/CMakeLists.txt	2014-09-22 22:58:15.000000000 +0900
+++ src/libhsplasma/core/CMakeLists.txt	2014-09-22 23:47:38.907734600 +0900
@@ -790,6 +790,7 @@ target_link_libraries(HSPlasma ${ZLIB_LI
 if(PHYSX_FOUND)
     target_link_libraries(HSPlasma ${PHYSX_COOKING_LIBRARY})
 endif(PHYSX_FOUND)
+set_target_properties(HSPlasma PROPERTIES SOVERSION 0)
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/libhsplasma.pc.in.cmake ${CMAKE_CURRENT_BINARY_DIR}/libhsplasma.pc @ONLY)
 
--- origsrc/libhsplasma/net/CMakeLists.txt	2014-09-22 22:58:18.000000000 +0900
+++ src/libhsplasma/net/CMakeLists.txt	2014-09-22 23:48:01.933125200 +0900
@@ -103,6 +103,7 @@ if(WIN32)
     target_link_libraries(HSPlasmaNet ws2_32)
 endif(WIN32)
 
+set_target_properties(HSPlasmaNet PROPERTIES SOVERSION 0)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/libhsplasmanet.pc.in.cmake ${CMAKE_CURRENT_BINARY_DIR}/libhsplasmanet.pc @ONLY)
 
 # Package installation
