--- origsrc/darkice-1.2/configure.in	2013-07-15 14:50:01.000000000 +0900
+++ src/darkice-1.2/configure.in	2015-05-07 15:20:28.878065300 +0900
@@ -55,7 +55,7 @@ AC_ARG_WITH(lame-prefix,
 
 if test "x${USE_LAME}" = "xyes" ; then
     AC_MSG_CHECKING( [for lame library at ${CONFIG_LAME_PREFIX}] )
-    LA_SEARCH_LIB( LAME_LIB_LOC, LAME_INC_LOC, libmp3lame.a libmp3lame.so, lame/lame.h,
+    LA_SEARCH_LIB( LAME_LIB_LOC, LAME_INC_LOC, libmp3lame.dll.a libmp3lame.a libmp3lame.so, lame/lame.h,
                    ${CONFIG_LAME_PREFIX})
     if test "x${LAME_LIB_LOC}" != "x" ; then
         AC_DEFINE( HAVE_LAME_LIB, 1, [build with lame library] )
@@ -89,9 +89,9 @@ AC_ARG_WITH(vorbis-prefix,
 
 if test "x${USE_VORBIS}" = "xyes" ; then
     AC_MSG_CHECKING( [for vorbis libraries at ${CONFIG_VORBIS_PREFIX}] )
-    LA_SEARCH_LIB( OGG_LIB_LOC, OGG_INC_LOC, libogg.a libogg.so, ogg/ogg.h,
+    LA_SEARCH_LIB( OGG_LIB_LOC, OGG_INC_LOC, libogg.dll.a libogg.a libogg.so, ogg/ogg.h,
                    ${CONFIG_VORBIS_PREFIX})
-    LA_SEARCH_LIB( VORBIS_LIB_LOC, VORBIS_INC_LOC, libvorbis.a libvorbis.so, vorbis/codec.h,
+    LA_SEARCH_LIB( VORBIS_LIB_LOC, VORBIS_INC_LOC, libvorbis.dll.a libvorbis.a libvorbis.so, vorbis/codec.h,
                    ${CONFIG_VORBIS_PREFIX})
     LA_SEARCH_LIB( VORBISENC_LIB_LOC, VORBISENC_INC_LOC,
                    libvorbisenc.a libvorbisenc.so, vorbis/vorbisenc.h,
@@ -132,9 +132,9 @@ AC_ARG_WITH(opus-prefix,
 
 if test "x${USE_OPUS}" = "xyes" ; then
     AC_MSG_CHECKING( [for opus libraries at ${CONFIG_OPUS_PREFIX}] )
-    LA_SEARCH_LIB( OGG_LIB_LOC, OGG_INC_LOC, libogg.a libogg.so, ogg/ogg.h,
+    LA_SEARCH_LIB( OGG_LIB_LOC, OGG_INC_LOC, libogg.dll.a libogg.a libogg.so, ogg/ogg.h,
                    ${CONFIG_OPUS_PREFIX})
-    LA_SEARCH_LIB( OPUS_LIB_LOC, OPUS_INC_LOC, libopus.a libopus.so, opus/opus.h,
+    LA_SEARCH_LIB( OPUS_LIB_LOC, OPUS_INC_LOC, libopus.dll.a libopus.a libopus.so, opus/opus.h,
                    ${CONFIG_OPUS_PREFIX})
 
     if test "x${OGG_LIB_LOC}" != "x" -a \
@@ -170,7 +170,7 @@ AC_ARG_WITH(faac-prefix,
 
 if test "x${USE_FAAC}" = "xyes" ; then
     AC_MSG_CHECKING( [for faac library at ${CONFIG_FAAC_PREFIX}] )
-    LA_SEARCH_LIB( FAAC_LIB_LOC, FAAC_INC_LOC, libfaac.a libfaac.so, faac.h,
+    LA_SEARCH_LIB( FAAC_LIB_LOC, FAAC_INC_LOC, libfaac.dll.a libfaac.a libfaac.so, faac.h,
                    ${CONFIG_FAAC_PREFIX})
     if test "x${FAAC_LIB_LOC}" != "x" ; then
         AC_DEFINE( HAVE_FAAC_LIB, 1, [build with faac library] )
@@ -204,7 +204,7 @@ AC_ARG_WITH(aacplus-prefix,
 
 if test "x${USE_AACPLUS}" = "xyes" ; then
     AC_MSG_CHECKING( [for aacplus library at ${CONFIG_AACPLUS_PREFIX}] )
-    LA_SEARCH_LIB( AACPLUS_LIB_LOC, AACPLUS_INC_LOC, libaacplus.a libaacplus.so, aacplus.h,
+    LA_SEARCH_LIB( AACPLUS_LIB_LOC, AACPLUS_INC_LOC, libaacplus.dll.a libaacplus.a libaacplus.so, aacplus.h,
                    ${CONFIG_AACPLUS_PREFIX})
     if test "x${AACPLUS_LIB_LOC}" != "x" ; then
         AC_DEFINE( HAVE_AACPLUS_LIB, 1, [build with aacplus library] )
@@ -238,7 +238,7 @@ AC_ARG_WITH(twolame-prefix,
 
 if test "x${USE_TWOLAME}" = "xyes" ; then
     AC_MSG_CHECKING( [for twolame library at ${CONFIG_TWOLAME_PREFIX}] )
-    LA_SEARCH_LIB( TWOLAME_LIB_LOC, TWOLAME_INC_LOC, libtwolame.a libtwolame.so, twolame.h,
+    LA_SEARCH_LIB( TWOLAME_LIB_LOC, TWOLAME_INC_LOC, libtwolame.dll.a libtwolame.a libtwolame.so, twolame.h,
                    ${CONFIG_TWOLAME_PREFIX})
     if test "x${TWOLAME_LIB_LOC}" != "x" ; then
         AC_DEFINE( HAVE_TWOLAME_LIB, 1, [build with twolame library] )
@@ -320,7 +320,7 @@ AC_ARG_WITH(pulseaudio-prefix,
 
 if test "x${USE_PULSEAUDIO}" = "xyes" ; then
     AC_MSG_CHECKING( [for pulseaudio libraries at ${CONFIG_PULSEAUDIO_PREFIX}] )
-    LA_SEARCH_LIB( PULSEAUDIO_LIB_LOC, PULSEAUDIO_INC_LOC, libpulse.so, pulse/pulseaudio.h,
+    LA_SEARCH_LIB( PULSEAUDIO_LIB_LOC, PULSEAUDIO_INC_LOC, libpulse.dll.a libpulse.so, pulse/pulseaudio.h,
                    ${CONFIG_PULSEAUDIO_PREFIX})
 
     if test "x${PULSEAUDIO_LIB_LOC}" != "x" ; then
--- origsrc/darkice-1.2/darkice.cfg	2013-07-15 14:50:01.000000000 +0900
+++ src/darkice-1.2/darkice.cfg	2015-05-07 15:20:28.878065300 +0900
@@ -23,7 +23,7 @@ channel         = 2         # channels.
 bitrateMode     = abr       # average bit rate
 format          = vorbis    # format of the stream: ogg vorbis
 bitrate         = 96        # bitrate of the stream sent to the server
-server          = yp.yourserver.com
+server          = 127.0.0.1
                             # host name of the server
 port            = 8000      # port of the IceCast2 server, usually 8000
 password        = hackme    # source password to the IceCast2 server
@@ -32,7 +32,7 @@ name            = DarkIce trial
                             # name of the stream
 description     = This is only a trial
                             # description of the stream
-url             = http://www.yourserver.com
+url             = http://127.0.0.1/
                             # URL related to the stream
 genre           = my own    # genre of the stream
 public          = yes       # advertise this stream?
@@ -43,9 +43,10 @@ localDumpFile	= dump.ogg  # local dump f
 # these can be mixed with [icecast2-x] and [shoutcast-x] sections
 [icecast-0]
 bitrateMode     = cbr       # constant bit rate
+format          = mp3
 bitrate         = 96        # bitrate of the mp3 stream sent to the server
 quality         = 0.8       # encoding quality
-server          = yp.yourserver.com
+server          = 127.0.0.1
                             # host name of the server
 port            = 8000      # port of the IceCast server, usually 8000
 password        = hackme    # source password to the IceCast server
@@ -54,7 +55,7 @@ name            = DarkIce trial
                             # name of the stream
 description     = This is only a trial
                             # description of the stream
-url             = http://www.yourserver.com
+url             = http://127.0.0.1/
                             # URL related to the stream
 genre           = my own    # genre of the stream
 public          = yes       # advertise this stream?
@@ -65,13 +66,13 @@ public          = yes       # advertise
 [shoutcast-0]
 bitrateMode     = vbr       # variable bit rate mode
 quality         = 0.5       # encoding quality
-server          = yp.yourserver.com
+server          = 127.0.0.1
                             # host name of the server
 port            = 8001      # source port of the ShoutCast server, usually 8001
 password        = hackme    # source password to the ShoutCast server
 name            = DarkIce trial
                             # name of the stream
-url             = http://www.yourserver.com
+url             = http://127.0.0.1
                             # URL related to the stream
 genre           = my own    # genre of the stream
 public          = yes       # advertise this stream?
--- origsrc/darkice-1.2/src/FaacEncoder.h	2013-07-15 14:50:42.000000000 +0900
+++ src/darkice-1.2/src/FaacEncoder.h	2015-05-07 15:20:28.895080800 +0900
@@ -40,6 +40,8 @@
 #include "config.h"
 #endif
 
+#include <cstring>
+
 #ifdef HAVE_FAAC_LIB
 #include <faac.h>
 #else
--- origsrc/darkice-1.2/src/FileSink.cpp	2013-07-15 14:50:42.000000000 +0900
+++ src/darkice-1.2/src/FileSink.cpp	2015-05-07 15:22:09.987966500 +0900
@@ -87,6 +87,7 @@
 #error need signal.h
 #endif
 
+#include <sys/select.h>
 
 #include <iostream>
 #include <sstream>
@@ -380,5 +381,3 @@ FileSink :: close ( void )
     ::close( fileDescriptor);
     fileDescriptor = 0;
 }
-
-
--- origsrc/darkice-1.2/src/OpusLibEncoder.h	2013-07-15 14:50:42.000000000 +0900
+++ src/darkice-1.2/src/OpusLibEncoder.h	2015-05-07 15:20:28.895080800 +0900
@@ -40,6 +40,10 @@
 #include "config.h"
 #endif
 
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+
 #ifdef HAVE_OPUS_LIB
 #include <opus/opus.h>
 #include <ogg/ogg.h>
--- origsrc/darkice-1.2/src/OssDspSource.cpp	2013-07-15 14:50:42.000000000 +0900
+++ src/darkice-1.2/src/OssDspSource.cpp	2015-05-07 15:22:36.326458300 +0900
@@ -92,6 +92,7 @@
 #error need sys/soundcard.h
 #endif
 
+#include <sys/select.h>
 
 #include "Util.h"
 #include "Exception.h"
@@ -312,4 +313,3 @@ OssDspSource :: close ( void )
 }
 
 #endif // SUPPORT_OSS_DSP
-
--- origsrc/darkice-1.2/src/PulseAudioDspSource.h	2013-07-15 14:50:42.000000000 +0900
+++ src/darkice-1.2/src/PulseAudioDspSource.h	2015-05-07 15:20:28.912096300 +0900
@@ -44,6 +44,8 @@
 #include "config.h"
 #endif
 
+#include <cstdio>
+
 #include "Reporter.h"
 #include "AudioSource.h"
 
--- origsrc/darkice-1.2/src/SerialUlaw.cpp	2013-07-15 14:50:42.000000000 +0900
+++ src/darkice-1.2/src/SerialUlaw.cpp	2015-05-07 15:23:50.874831400 +0900
@@ -94,11 +94,13 @@
 #error need termios.h
 #endif
 
+#include <sys/select.h>
 
 #include "Util.h"
 #include "Exception.h"
 #include "SerialUlaw.h"
 
+#include <cstdio>
 
 /* ===================================================  local data structures */
 
--- origsrc/darkice-1.2/src/TcpSocket.cpp	2013-07-15 14:50:42.000000000 +0900
+++ src/darkice-1.2/src/TcpSocket.cpp	2015-05-07 15:24:05.602763500 +0900
@@ -87,6 +87,7 @@
 #error need signal.h
 #endif
 
+#include <sys/select.h>
 
 #include "Util.h"
 #include "Exception.h"
@@ -413,5 +414,3 @@ TcpSocket :: close ( void )
     ::close( sockfd);
     sockfd = 0;
 }
-
-
--- origsrc/darkice-1.2/src/Util.cpp	2013-07-15 14:50:42.000000000 +0900
+++ src/darkice-1.2/src/Util.cpp	2015-05-07 15:24:18.088324200 +0900
@@ -87,6 +87,7 @@
 #error need signal.h
 #endif
 
+#include <sys/select.h>
 
 #include "Util.h"
 
