--- origsrc/corgi-0.0.2/src/Makefile.am	2010-10-11 01:15:13.000000000 +0900
+++ src/corgi-0.0.2/src/Makefile.am	2012-02-12 15:49:10.121933600 +0900
@@ -1,10 +1,11 @@
 
 bin_PROGRAMS = corgi
 corgi_SOURCES = main.c
-corgi_LDADD = .libs/libcorgi.a
+corgi_LDADD = libcorgi.la
 lib_LTLIBRARIES = libcorgi.la
 libcorgi_la_SOURCES = corgi.c unicode.c
-DEFAULT_INCLUDES = -I$(top_srcdir)/include
+libcorgi_la_LDFLAGS = -no-undefined
+DEFAULT_INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include
 AM_CFLAGS = -Wall -Werror
 
 TOOLS_DIR = $(top_srcdir)/tools
