--- origsrc/libzia-4.67/bin/Makefile.am	2024-08-23 21:40:37.000000000 +0900
+++ src/libzia-4.67/bin/Makefile.am	2025-04-23 10:20:39.229576600 +0900
@@ -5,4 +5,4 @@ EXTRA_DIST = $(bin_SCRIPTS) zia-config.i
 datarootdir = @datarootdir@
 
 svnversion$(EXEEXT):
-	(cd ..; /bin/sh svnver.sh)
+	(cd $(top_srcdir); /bin/sh svnver.sh)
--- origsrc/libzia-4.67/configure.ac	2025-02-11 16:32:32.000000000 +0900
+++ src/libzia-4.67/configure.ac	2025-04-23 10:20:39.229576600 +0900
@@ -14,6 +14,8 @@ fi
 LT_INIT([dlopen])
 AC_CONFIG_MACRO_DIR([m4])
 
+AM_ICONV
+
 dnl Determine default prefix
 test x$prefix = "xNONE" && prefix="$ac_default_prefix"
                                         
@@ -36,6 +38,9 @@ case "$host_os" in
 mingw*)
     PLATFORM="mingw"
     ;;
+cygwin*)
+    PLATFORM="cygwin"
+    ;;
 *)    
     PLATFORM=$(uname -s | sed 'y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/')
     if test $? != 0 ; then 
@@ -84,7 +89,10 @@ AM_CONDITIONAL([USE_WLMAP], [test "$app_
 
 dnl Check for system header files
 AC_HEADER_STDC
-AC_CHECK_HEADERS(arpa/inet.h bfd.h cygwin/in.h ctype.h dbghelp.h dirent.h dlfcn.h errno.h execinfo.h fcntl.h getopt.h ifaddrs.h linux/i2c-dev.h linux/input.h linux/spi/spidev.h math.h netdb.h netinet/in.h psapi.h pty.h shlobj.h signal.h stdarg.h stdio.h string.h sys/ioctl.h sys/prctl.h sys/select.h sys/socket.h sys/time.h sys/types.h sys/wait.h termios.h time.h unistd.h win32api/ws2tcpip.h windows.h winsock2.h ws2tcpip.h)
+AC_CHECK_HEADERS(arpa/inet.h bfd.h cygwin/in.h ctype.h dirent.h dlfcn.h errno.h execinfo.h fcntl.h getopt.h ifaddrs.h linux/i2c-dev.h linux/input.h linux/spi/spidev.h math.h netdb.h netinet/in.h pty.h shlobj.h signal.h stdarg.h stdio.h string.h sys/ioctl.h sys/prctl.h sys/select.h sys/socket.h sys/time.h sys/types.h sys/wait.h termios.h time.h unistd.h)
+if test "$host_os" != "cygwin"; then
+    AC_CHECK_HEADERS(dbghelp.h psapi.h win32api/ws2tcpip.h windows.h winsock2.h ws2tcpip.h)
+fi
 AC_CHECK_HEADERS([linux/wireless.h], [], [], [
    #include <sys/types.h>
    #include <sys/socket.h>
@@ -563,5 +571,5 @@ echo ";") > src/settings.c
 
 cat config.h |grep '^#define' | sed 's/#define \(.*\)/#define Z_\1/' > include/unix/zconfig.h
 
-chmod +x svnver.sh bin/zia-config
+chmod +x ${srcdir}/svnver.sh bin/zia-config
 
--- origsrc/libzia-4.67/include/libziaint.h	2024-08-23 21:40:34.000000000 +0900
+++ src/libzia-4.67/include/libziaint.h	2025-04-23 10:20:39.229576600 +0900
@@ -56,7 +56,9 @@ msvc10 (vs2010):         WIN32 _MSC_VER=
 #elif defined(__CYGWIN__)
 #define Z_CYGWIN
 #define Z_PLATFORM "cygwin"
+#if !defined(Z_MACHINE)
 #define Z_MACHINE "win32"
+#endif
 #elif defined(ANDROID)
 #define Z_ANDROID
 #define Z_PLATFORM "android"
--- origsrc/libzia-4.67/include/zfiledlg.h	2024-08-23 21:40:34.000000000 +0900
+++ src/libzia-4.67/include/zfiledlg.h	2025-04-23 10:20:39.229576600 +0900
@@ -35,7 +35,7 @@ struct zfiledlg{
     void *arg;
     char *last_dir;
     struct zselect *zsel;
-#ifdef Z_MSC_MINGW_CYGWIN
+#ifdef Z_MSC_MINGW
 	OPENFILENAME ofn;
 #elif defined(Z_HAVE_DLFCN_H)
     GPtrArray *memlist;
--- origsrc/libzia-4.67/src/Makefile.am	2025-04-23 10:20:37.419512600 +0900
+++ src/libzia-4.67/src/Makefile.am	2025-04-23 10:20:39.229576600 +0900
@@ -1,8 +1,11 @@
 datarootdir = @datarootdir@
 
 bindir=$(exec_prefix)/lib/$(DEB_HOST_MULTIARCH)/libzia
+
+bin_PROGRAMS = map2d
+
 if USE_WLMAP
-bin_PROGRAMS = libzia-@VERSION@.so.d #zsdltest gpiotest
+bin_PROGRAMS += libzia-@VERSION@.so.d #zsdltest gpiotest
 endif
 
 lib_LTLIBRARIES = libzia.la
@@ -73,8 +76,9 @@ libzia_la_SOURCES = \
 	zver.c
 
 
-libzia_la_CFLAGS = $(Z_SVNVER) -I../include -I../include/unix 
-libzia_la_LDFLAGS = -release @VERSION@ -version-info 0
+libzia_la_CFLAGS = $(Z_SVNVER) -I../include -I../include/unix -I$(top_srcdir)/include
+libzia_la_LDFLAGS = -no-undefined
+libzia_la_LIBADD = $(LTLIBICONV)
 library_includedir = $(includedir)/libzia
 
 #zsdltest_SOURCES = zsdltest.c
@@ -96,6 +100,3 @@ libzia_@VERSION@_so_d_SOURCES = map2d.c
 
 libzia-@VERSION@.so.d$(EXEEXT): libzia.la map2d
 	./map2d libzia.so.map libzia-@VERSION@.so.d
-
-map2d: map2d.c
-	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) map2d.c -o $@
--- origsrc/libzia-4.67/src/zbfd.c	2024-08-23 21:40:37.000000000 +0900
+++ src/libzia-4.67/src/zbfd.c	2025-04-23 10:20:39.245208400 +0900
@@ -115,7 +115,7 @@ int z_backtrace(void **buffer, int size,
         STACKSTEP(20);STACKSTEP(21);STACKSTEP(22);STACKSTEP(24);STACKSTEP(25);
         STACKSTEP(26);STACKSTEP(27);STACKSTEP(28);STACKSTEP(29);STACKSTEP(30);
     }while(0);*/
-#else
+#elif !defined(__CYGWIN__)
     // Z_UNIX
     i = backtrace(buffer, size);
 
--- origsrc/libzia-4.67/src/zerror.c	2024-08-23 21:40:37.000000000 +0900
+++ src/libzia-4.67/src/zerror.c	2025-04-23 10:20:39.245208400 +0900
@@ -98,7 +98,7 @@ void z_strerror(GString *gs, int err){
 #endif
 }
 
-#ifdef Z_MSC_MINGW_CYGWIN
+#ifdef Z_MSC_MINGW
 void z_lasterror(GString *gs){
 	z_lasterror_e(gs, GetLastError());
 }
--- origsrc/libzia-4.67/src/zfiledlg.c	2024-08-23 21:40:37.000000000 +0900
+++ src/libzia-4.67/src/zfiledlg.c	2025-04-23 10:20:39.245208400 +0900
@@ -58,7 +58,7 @@ void zfiledlg_free(struct zfiledlg *dlg)
 
 
 int zfiledlg_open(struct zfiledlg *dlg, struct zselect *zsel, void (*fce)(void *, char *), void *arg, const char *filename, const char *ext){
-#ifdef Z_MSC_MINGW_CYGWIN
+#ifdef Z_MSC_MINGW
 	int i, l, t;
 	GString *gs;
 	char *c, *ext2;
@@ -148,13 +148,13 @@ int zfiledlg_open(struct zfiledlg *dlg,
 }
 
 static gpointer zfiledlg_thread_func(gpointer data){
-#ifdef Z_MSC_MINGW_CYGWIN
+#ifdef Z_MSC_MINGW
 	struct zfiledlg *dlg = (struct zfiledlg *)data;
 #endif
 
 	zg_thread_set_name("Libzia zfiledlg");
 
-#ifdef Z_MSC_MINGW_CYGWIN
+#ifdef Z_MSC_MINGW
 
     if (!GetOpenFileName(&dlg->ofn)){
 		zselect_msg_send(dlg->zsel, "ZFILEDLG;0x%lx;X;%d", (unsigned long)GPOINTER_TO_SIZE(dlg), CommDlgExtendedError());
--- origsrc/libzia-4.67/src/zselect.c	2024-08-23 21:40:37.000000000 +0900
+++ src/libzia-4.67/src/zselect.c	2025-04-23 10:20:39.245208400 +0900
@@ -609,7 +609,7 @@ void alarm_handler(void *arg, siginfo_t
 void check_for_select_race(void)
 {
     if (critical_section) {
-#ifdef SIGALRM
+#if defined(SIGALRM) && defined(Z_UNIX_ANDROID)
         zselect_signal_set(SIGALRM, alarm_handler, NULL, 1);
 #endif
         pending_alarm = 1;
