diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/cool-retro-term-1.0.0/cool-retro-term.pro src/cool-retro-term-1.0.0/cool-retro-term.pro
--- origsrc/cool-retro-term-1.0.0/cool-retro-term.pro	2015-01-25 03:16:52.000000000 +0900
+++ src/cool-retro-term-1.0.0/cool-retro-term.pro	2015-06-27 00:18:29.122342600 +0900
@@ -2,7 +2,6 @@ TEMPLATE = subdirs
 
 CONFIG += ordered
 
-SUBDIRS += qmltermwidget
 SUBDIRS += app
 
 desktop.files += cool-retro-term.desktop
