--- origsrc/datamash-1.3/configure.ac	2018-01-25 09:24:31.000000000 +0900
+++ src/datamash-1.3/configure.ac	2018-03-28 08:34:17.848000000 +0900
@@ -9,7 +9,7 @@ dnl WITHOUT ANY WARRANTY, to the extent
 dnl implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 
 AC_INIT([GNU datamash],
-	[m4_esyscmd([build-aux/git-version-gen .tarball-version])],
+	[1.3],
 	[assafgordon@gmail.com], [],
 	[https://www.gnu.org/software/datamash])
 
