--- origsrc/datamash-1.0.6/configure.ac	2014-07-30 04:18:16.000000000 +0900
+++ src/datamash-1.0.6/configure.ac	2015-03-19 13:29:27.609943400 +0900
@@ -9,7 +9,7 @@ dnl WITHOUT ANY WARRANTY, to the extent
 dnl implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 
 AC_INIT([GNU datamash],
-        [m4_esyscmd([build-aux/git-version-gen .tarball-version])],
+        [1.0.6],
 	[assafgordon@gmail.com], [],
 	[http://www.gnu.org/software/datamash])
 
