--- origsrc/datamash-1.5/configure.ac	2019-08-27 07:07:01.000000000 +0900
+++ src/datamash-1.5/configure.ac	2019-10-12 22:44:54.615152700 +0900
@@ -9,7 +9,7 @@ dnl WITHOUT ANY WARRANTY, to the extent
 dnl implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 
 AC_INIT([GNU datamash],
-	[m4_esyscmd([build-aux/git-version-gen .tarball-version])],
+	[1.5],
 	[assafgordon@gmail.com], [],
 	[https://www.gnu.org/software/datamash])
 
