--- origsrc/wasm3/CMakeLists.txt	2022-09-09 21:37:40.000000000 +0900
+++ src/wasm3/CMakeLists.txt	2022-09-09 21:58:08.599293700 +0900
@@ -195,21 +195,30 @@ if(BUILD_WASI MATCHES "simple")
 elseif(BUILD_WASI MATCHES "metawasi")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Dd_m3HasMetaWASI")
 elseif(BUILD_WASI MATCHES "uvwasi")
-  include(FetchContent)
-  FetchContent_Declare(
-    uvwasi
-    GIT_REPOSITORY https://github.com/nodejs/uvwasi.git
-    GIT_TAG b599542f7ce001e04cdff9db82b05fee96bb3332
-  )
+  find_package(pkgconfig)
+  pkg_check_modules(UVWASI uvwasi)
+  if (NOT UVWASI_FOUND)
+    include(FetchContent)
+    FetchContent_Declare(
+      uvwasi
+      GIT_REPOSITORY https://github.com/nodejs/uvwasi.git
+      GIT_TAG b599542f7ce001e04cdff9db82b05fee96bb3332
+    )
 
-  FetchContent_GetProperties(uvwasi)
-  if(NOT uvwasi_POPULATED)
-    FetchContent_Populate(uvwasi)
-    include_directories("${uvwasi_SOURCE_DIR}/include")
-    add_subdirectory(${uvwasi_SOURCE_DIR} ${uvwasi_BINARY_DIR} EXCLUDE_FROM_ALL)
+    FetchContent_GetProperties(uvwasi)
+    if(NOT uvwasi_POPULATED)
+      FetchContent_Populate(uvwasi)
+      include_directories("${uvwasi_SOURCE_DIR}/include")
+      add_subdirectory(${uvwasi_SOURCE_DIR} ${uvwasi_BINARY_DIR} EXCLUDE_FROM_ALL)
+    endif()
   endif()
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Dd_m3HasUVWASI")
-  target_link_libraries(${OUT_FILE} uvwasi_a uv_a)
+  if (UVWASI_FOUND)
+    include_directories("/usr/include/uvwasi")
+    target_link_libraries(${OUT_FILE} ${UVWASI_LIBRARIES})
+  else()
+    target_link_libraries(${OUT_FILE} uvwasi_a uv_a)
+  endif()
 endif()
 
 check_ipo_supported(RESULT result)
