--- origsrc/wasm3-0.4.6/source/m3_api_wasi.c	2020-01-31 06:25:18.000000000 +0900
+++ src/wasm3-0.4.6/source/m3_api_wasi.c	2020-02-16 21:04:27.192085700 +0900
@@ -25,7 +25,7 @@ typedef uint32_t __wasi_size_t;
 #include <stdio.h>
 #include <fcntl.h>
 
-#if defined(__wasi__) || defined(__APPLE__) || defined(__ANDROID_API__) || defined(__OpenBSD__) || defined(__linux__)
+#if defined(__wasi__) || defined(__APPLE__) || defined(__ANDROID_API__) || defined(__OpenBSD__) || defined(__linux__) || defined(__CYGWIN__)
 #  include <unistd.h>
 #  include <sys/uio.h>
 #  if defined(__APPLE__)
--- origsrc/wasm3-0.4.6/source/m3_config_platforms.h	2020-01-31 06:25:18.000000000 +0900
+++ src/wasm3-0.4.6/source/m3_config_platforms.h	2020-02-16 21:08:22.954163800 +0900
@@ -172,7 +172,7 @@
 # endif
 
 
-# if defined(M3_COMPILER_MSVC)
+# if defined(M3_COMPILER_MSVC) || defined(__CYGWIN__)
 #  define M3_WEAK
 # else
 #  define M3_WEAK __attribute__((weak))
