--- origsrc/ivykis-0.43/src/Makefile.am	2024-01-30 03:45:10.000000000 +0900
+++ src/ivykis-0.43/src/Makefile.am	2024-04-25 11:03:08.117761500 +0900
@@ -60,7 +60,7 @@ SRC			+= iv_inotify.c
 INC			+= include/iv_inotify.h
 endif
 
-LINKFLAGS	= -version-info 6:0:6
+LINKFLAGS	= -no-undefined -version-info 6:0:6
 if HAVE_VERSIONING
 LINKFLAGS	+= -Wl,--version-script,$(top_srcdir)/libivykis.posix.ver \
 		   -Wl,-undefined-version
