--- origsrc/ivykis-0.37/src/Makefile.am	2012-12-19 09:48:20.000000000 +0900
+++ src/ivykis-0.37/src/Makefile.am	2012-12-23 15:20:03.131872400 +0900
@@ -60,7 +60,7 @@ SRC			+= iv_inotify.c
 INC			+= include/iv_inotify.h
 endif
 
-LINKFLAGS	= -version-info 4:0:4
+LINKFLAGS	= -version-info 4:0:4 -no-undefined
 if HAVE_VERSIONING
 LINKFLAGS	+= -Wl,--version-script,$(top_srcdir)/libivykis.posix.ver
 endif
@@ -76,8 +76,7 @@ SRC			+= iv_event_raw_win32.c		\
 			   iv_time_win32.c
 
 LINKFLAGS = -version-info 0:0:0						\
-	    -Wl,--version-script,$(top_srcdir)/libivykis.win32.ver	\
-	    -no-undefined
+	    -Wl,--version-script,$(top_srcdir)/libivykis.win32.ver
 
 endif
 
