--- origsrc/libheif-1.7.0/gdk-pixbuf/Makefile.am	2019-11-08 07:04:26.000000000 +0900
+++ src/libheif-1.7.0/gdk-pixbuf/Makefile.am	2020-08-10 08:27:10.042005700 +0900
@@ -9,6 +9,7 @@ libpixbufloader_heif_la_CFLAGS = -I$(top
 libpixbufloader_heif_la_LIBADD = ../libheif/libheif.la $(gdkpixbuf_LIBS)
 libpixbufloader_heif_la_LDFLAGS = -avoid-version -module
 libpixbufloader_heif_la_SOURCES = pixbufloader-heif.c
+libpixbufloader_heif_la_LDFLAGS = -no-undefined -avoid-version -module
 endif
 
 EXTRA_DIST = CMakeLists.txt
--- origsrc/libheif-1.7.0/libheif/Makefile.am	2020-05-06 21:00:28.000000000 +0900
+++ src/libheif-1.7.0/libheif/Makefile.am	2020-08-10 08:27:10.081085600 +0900
@@ -34,9 +34,7 @@ libheif_la_CXXFLAGS = \
 libheif_la_LIBADD = $(ADDITIONAL_LIBS)
 
 libheif_la_LDFLAGS = -version-info $(LIBHEIF_CURRENT):$(LIBHEIF_REVISION):$(LIBHEIF_AGE)
-if MINGW
 libheif_la_LDFLAGS += -no-undefined
-endif
 
 libheif_la_SOURCES = \
   bitstream.h \
