--- origsrc/libheif-1.6.2/gdk-pixbuf/Makefile.am	2020-01-17 22:33:13.000000000 +0900
+++ src/libheif-1.6.2/gdk-pixbuf/Makefile.am	2020-02-02 21:47:20.799784200 +0900
@@ -9,6 +9,7 @@ libpixbufloader_heif_la_CFLAGS = -I$(top
 libpixbufloader_heif_la_LIBADD = ../libheif/libheif.la $(gdkpixbuf_LIBS)
 libpixbufloader_heif_la_LDFLAGS = -avoid-version -module
 libpixbufloader_heif_la_SOURCES = pixbufloader-heif.c
+libpixbufloader_heif_la_LDFLAGS = -no-undefined -avoid-version -module
 endif
 
 EXTRA_DIST = CMakeLists.txt
--- origsrc/libheif-1.6.2/libheif/Makefile.am	2020-01-31 20:23:52.000000000 +0900
+++ src/libheif-1.6.2/libheif/Makefile.am	2020-02-02 21:47:20.831071200 +0900
@@ -29,9 +29,7 @@ libheif_la_CXXFLAGS = \
 libheif_la_LIBADD = $(ADDITIONAL_LIBS)
 
 libheif_la_LDFLAGS = -version-info $(LIBHEIF_CURRENT):$(LIBHEIF_REVISION):$(LIBHEIF_AGE)
-if MINGW
 libheif_la_LDFLAGS += -no-undefined
-endif
 
 libheif_la_SOURCES = \
   bitstream.h \
