--- origsrc/ceelog-0.1/Makefile.am	2012-09-12 20:55:35.000000000 +0900
+++ src/ceelog-0.1/Makefile.am	2012-12-13 22:19:39.984958800 +0900
@@ -70,7 +70,7 @@ lib_libceelog_la_SOURCES = lib/event.c l
 	lib/libceelog.c \
 	lib/source.c lib/source_private.h lib/source_static_file.c
 lib_libceelog_la_CPPFLAGS = $(AM_CPPFLAGS) $(LOCALEDIR_CPPFLAGS)
-lib_libceelog_la_LDFLAGS = -version-info 0:0:0
+lib_libceelog_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 lib_libceelog_la_LIBADD = $(glib_LIBS) $(json_LIBS) $(pcre_LIBS)
 
 src_ceelog_SOURCES = src/ceelog.c
