--- origsrc/libqofonoext-1.0.32/plugin/plugin.pro	2024-03-19 21:56:15.000000000 +0900
+++ src/libqofonoext-1.0.32/plugin/plugin.pro	2025-08-14 00:00:41.738137900 +0900
@@ -1,9 +1,11 @@
 TARGET=qofonoextdeclarative
 TEMPLATE = lib
 CONFIG += plugin link_pkgconfig
-PKGCONFIG += qofono-qt$${QT_MAJOR_VERSION}
+PKGCONFIG += qofono
 QMAKE_CXXFLAGS += -Wno-unused-parameter -Wno-psabi
 
+DEFINES += QOFONOEXT_LIBRARY
+
 QT_VERSION=$$[QT_VERSION]
 
 QT += qml dbus
--- origsrc/libqofonoext-1.0.32/src/src.pro	2024-03-19 21:56:15.000000000 +0900
+++ src/libqofonoext-1.0.32/src/src.pro	2025-08-13 23:52:50.177762000 +0900
@@ -1,6 +1,6 @@
 TARGET = qofonoext
 CONFIG += create_pc create_prl no_install_prl link_pkgconfig
-PKGCONFIG += qofono-qt$${QT_MAJOR_VERSION}
+PKGCONFIG += qofono
 
 QT += dbus
 QT -= gui
--- origsrc/libqofonoext-1.0.32/src/version.pri	2024-03-19 21:56:15.000000000 +0900
+++ src/libqofonoext-1.0.32/src/version.pri	2025-08-14 00:08:31.121261600 +0900
@@ -1 +1 @@
-VERSION = 1.0.31
+VERSION = 1.0.32
