--- origsrc/libgtrfc3161-1.1.78/src/example/GNUmakefile.am	2019-05-17 16:35:27.000000000 +0900
+++ src/libgtrfc3161-1.1.78/src/example/GNUmakefile.am	2024-10-13 15:27:24.986597800 +0900
@@ -17,7 +17,7 @@
 # reserves and retains all trademark rights.
 #
 
-AM_CFLAGS = -g -Wall -I$(top_builddir)/src/gtrfc3161
+AM_CFLAGS = -g -Wall -I$(top_builddir)/src/gtrfc3161 -I$(top_srcdir)/src/gtrfc3161
 AM_LDFLAGS = -L$(top_builddir)/src/gtrfc3161 -no-install -lgtrfc3161 -lksi
 check_PROGRAMS = gtrfc3161_convert
 
--- origsrc/libgtrfc3161-1.1.78/src/gtrfc3161/GNUmakefile.am	2019-05-17 16:35:27.000000000 +0900
+++ src/libgtrfc3161-1.1.78/src/gtrfc3161/GNUmakefile.am	2024-10-13 15:27:24.986597800 +0900
@@ -36,7 +36,7 @@ otherinclude_HEADERS = \
 	tsconvert.h\
 	version.h
 
-libgtrfc3161_la_LDFLAGS=-version-info @LTVER@
+libgtrfc3161_la_LDFLAGS=-no-undefined -version-info @LTVER@
 
 if FULL_TEST_COVERAGE
 lib_LTLIBRARIES += libgtrfc3161mlc.la
