--- origsrc/itpp-4.2/configure.ac	2010-09-21 22:46:43.000000000 +0900
+++ src/itpp-4.2/configure.ac	2011-01-03 01:29:50.482126100 +0900
@@ -47,7 +47,7 @@ AC_SUBST(GENERIC_LIBRARY_VERSION)
 AC_CANONICAL_HOST
 case $host_os in
   cygwin* | mingw*)
-    AC_DISABLE_SHARED
+    AC_ENABLE_SHARED
     AC_ENABLE_STATIC
     if test "x$enable_shared" = xyes; then
       NO_UNDEFINED="-no-undefined"
@@ -77,7 +77,7 @@ AC_PROG_CXXCPP
 CXXFLAGS="$save_CXXFLAGS"
 AC_PROG_F77
 AC_PROG_INSTALL
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_PROG_LN_S
 AC_PROG_SED
 
