--- origsrc/h5xx-0.9.2/CMakeLists.txt	2019-06-28 19:16:52.000000000 +0900
+++ src/h5xx-0.9.2/CMakeLists.txt	2019-12-24 14:29:06.852629800 +0900
@@ -37,9 +37,13 @@ include_directories(SYSTEM ${HDF5_INCLUD
 include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
 include_directories(${CMAKE_SOURCE_DIR})
 
-enable_testing()
-include(CTest)
-add_subdirectory(test)
+if (NOT CYGWIN)
+  enable_testing()
+  include(CTest)
+  add_subdirectory(test)
+endif()
 
 set(H5XX_INCLUDE_DIR "${CMAKE_SOURCE_DIR}/h5xx")
 add_subdirectory(example)
+
+install(DIRECTORY ${H5XX_INCLUDE_DIR} DESTINATION include)
