--- origsrc/libshairport/configure.in	2013-02-23 19:07:36.000000000 +0900
+++ src/libshairport/configure.in	2013-02-23 19:07:59.828402100 +0900
@@ -1,8 +1,8 @@
 # -*- Autoconf -*-
 # Process this file with autoconf to produce a configure script.
 
-AC_INIT([shairport.c], [1.2.1], [http://trac.xbmc.org])
-AM_INIT_AUTOMAKE(libshairport, 1.2.1)
+AC_INIT([libshairport], [1.2.1], [http://trac.xbmc.org])
+AM_INIT_AUTOMAKE
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_MACRO_DIR([m4])
 
--- origsrc/libshairport/src/Makefile.am	2013-02-23 19:07:36.000000000 +0900
+++ src/libshairport/src/Makefile.am	2013-02-23 19:17:10.287386400 +0900
@@ -6,5 +6,5 @@ library_include_HEADERS = shairport.h
 noinst_HEADERS = alac.h hairtunes.h socketlib.h shairport_private.h
 
 libshairport_la_SOURCES=shairport.c hairtunes.c socketlib.c alac.c
-libshairport_la_CFLAGS=-Wl,--no-undefined 
-libshairport_la_LDFLAGS=-version-info @LIBSHAIRPORT_LIBTOOL_VERSION@
+libshairport_la_CFLAGS=
+libshairport_la_LDFLAGS=-version-info @LIBSHAIRPORT_LIBTOOL_VERSION@ -no-undefined
