--- origsrc/libite-2.6.1/src/Makefile.am	2023-10-15 17:22:36.000000000 +0900
+++ src/libite-2.6.1/src/Makefile.am	2024-01-05 15:30:20.587679300 +0900
@@ -2,7 +2,7 @@ lib_LTLIBRARIES       = libite.la
 
 libite_la_CPPFLAGS    = -D_GNU_SOURCE
 libite_la_CFLAGS      = -W -Wall -Wextra
-libite_la_LDFLAGS     = $(AM_LDFLAGS) -version-info 9:0:4
+libite_la_LDFLAGS     = $(AM_LDFLAGS) -no-undefined -version-info 9:0:4
 libite_la_SOURCES     = chomp.c		copyfile.c			\
 			conio.c		conio.h		dir.c		\
 			erasef.c	fopenf.c	fremove.c	\
--- origsrc/libite-2.6.1/test/copyfile.c	2023-10-15 17:22:36.000000000 +0900
+++ src/libite-2.6.1/test/copyfile.c	2024-01-05 18:49:32.936254500 +0900
@@ -6,9 +6,9 @@
 #define ERRX(fmt, args...) { printf(fmt "\n", ##args); rc = 1; }
 
 static char *files[] = {
-	"/etc/passwd", "/tmp/mypwd", "/tmp/mypwd",
-	"/etc/passwd", "/tmp/",      "/tmp/passwd",
-	"/etc/passwd", "/tmp",       "/tmp/passwd",
+	"/etc/fstab", "/tmp/mypwd", "/tmp/mypwd",
+	"/etc/fstab", "/tmp/",      "/tmp/fstab",
+	"/etc/fstab", "/tmp",       "/tmp/fstab",
 	NULL
 };
 
--- origsrc/libite-2.6.1/test/fsendfile.c	2023-10-15 17:22:36.000000000 +0900
+++ src/libite-2.6.1/test/fsendfile.c	2024-01-05 18:47:48.286568500 +0900
@@ -6,9 +6,9 @@ int main(void)
 {
 	int i = 0;
 	char *files[] = {
-		"/etc/passwd", "/tmp/tok",
-		"/etc/passwd", "/tmp/passwd",
-		"/etc/passwd", "/tmp/passwd",
+		"/etc/fstab", "/tmp/tok",
+		"/etc/fstab", "/tmp/passwd",
+		"/etc/fstab", "/tmp/passwd",
 		NULL
 	};
 	FILE *src, *dst;
