--- origsrc/libite-2.5.2/src/Makefile.am	2022-04-17 21:19:11.000000000 +0900
+++ src/libite-2.5.2/src/Makefile.am	2022-05-20 08:17:26.505423900 +0900
@@ -20,6 +20,7 @@ libite_la_SOURCES     = chomp.c		copyfil
 			lite.h		strlite.h			\
 			queue.h		tree.h
 
+libite_la_LDFLAGS += -no-undefined
 
 ## Distribute these header files in the <libite/...> namespace
 pkgincludedir       = $(includedir)/libite
--- origsrc/libite-2.5.2/test/copyfile.c	2022-04-17 21:19:11.000000000 +0900
+++ src/libite-2.5.2/test/copyfile.c	2022-05-20 08:19:54.072263800 +0900
@@ -6,9 +6,15 @@ int main(void)
 {
 	int i = 0;
 	char *files[] = {
+#if defined(__CYGWIN__)
+		"/etc/fstab", "/tmp/mypwd", "/tmp/mypwd",
+		"/etc/fstab", "/tmp/", "/tmp/passwd",
+		"/etc/fstab", "/tmp", "/tmp/passwd",
+#else
 		"/etc/passwd", "/tmp/mypwd", "/tmp/mypwd",
 		"/etc/passwd", "/tmp/", "/tmp/passwd",
 		"/etc/passwd", "/tmp", "/tmp/passwd",
+#endif
 		NULL
 	};
 	FILE *src, *dst;
--- origsrc/libite-2.5.2/test/fsendfile.c	2022-04-17 21:19:11.000000000 +0900
+++ src/libite-2.5.2/test/fsendfile.c	2022-05-20 08:20:31.816082200 +0900
@@ -6,9 +6,15 @@ int main(void)
 {
 	int i = 0;
 	char *files[] = {
+#if defined(__CYGWIN__)
+		"/etc/fstab", "/tmp/tok",
+		"/etc/fstab", "/tmp/passwd",
+		"/etc/fstab", "/tmp/passwd",
+#else
 		"/etc/passwd", "/tmp/tok",
 		"/etc/passwd", "/tmp/passwd",
 		"/etc/passwd", "/tmp/passwd",
+#endif
 		NULL
 	};
 	FILE *src, *dst;
