--- origsrc/libite-2.2.0/src/Makefile.am	2020-03-17 01:41:44.000000000 +0900
+++ src/libite-2.2.0/src/Makefile.am	2020-03-24 13:04:08.614668300 +0900
@@ -2,7 +2,7 @@ lib_LTLIBRARIES       = libite.la
 
 libite_la_CPPFLAGS    = -D_GNU_SOURCE
 libite_la_CFLAGS      = -W -Wall -Wextra
-libite_la_LDFLAGS     = $(AM_LDFLAGS) -version-info 6:0:1
+libite_la_LDFLAGS     = $(AM_LDFLAGS) -no-undefined -version-info 6:0:1
 libite_la_SOURCES     = chomp.c		copyfile.c			\
 			conio.c		conio.h				\
 			dir.c		fopenf.c			\
--- origsrc/libite-2.2.0/src/ifconfig.c	2020-03-17 01:41:44.000000000 +0900
+++ src/libite-2.2.0/src/ifconfig.c	2020-03-24 14:37:10.494626300 +0900
@@ -59,13 +59,13 @@ int ifconfig(const char *ifname, const c
 	if (up) {
 		if (addr) {
 			if (inet_pton(AF_INET, addr, &sin->sin_addr) == 1)
-				ret = ioctl(sd, SIOCSIFADDR, &ifr);
+				ret = ioctl(sd, SIOCGIFADDR, &ifr);
 		}
 
 		/* Non-zero IP address */
 		if (mask && addr && strcmp(addr, "0.0.0.0")) {
 			if (inet_pton(AF_INET, mask, &sin->sin_addr) == -1)
-				ret = ioctl(sd, SIOCSIFNETMASK, &ifr);
+				ret = ioctl(sd, SIOCGIFNETMASK, &ifr);
 		}
 	}
 
@@ -75,7 +75,7 @@ int ifconfig(const char *ifname, const c
 		else
 			ifr.ifr_flags &= ~IFF_UP;
 
-		ret = ioctl(sd, SIOCSIFFLAGS, &ifr);
+		ret = ioctl(sd, SIOCGIFFLAGS, &ifr);
 	}
 
 	close(sd);
