--- origsrc/libite-2.5.3/src/Makefile.am	2023-04-08 14:05:58.000000000 +0900
+++ src/libite-2.5.3/src/Makefile.am	2023-05-19 08:17:14.235274700 +0900
@@ -2,7 +2,7 @@ lib_LTLIBRARIES       = libite.la
 
 libite_la_CPPFLAGS    = -D_GNU_SOURCE
 libite_la_CFLAGS      = -W -Wall -Wextra
-libite_la_LDFLAGS     = $(AM_LDFLAGS) -version-info 8:2:3
+libite_la_LDFLAGS     = $(AM_LDFLAGS) -no-undefined -version-info 8:2:3
 libite_la_SOURCES     = chomp.c		copyfile.c			\
 			conio.c		conio.h		dir.c		\
 			erasef.c	fopenf.c	fremove.c	\
--- origsrc/libite-2.5.3/test/copyfile.c	2023-04-08 14:05:58.000000000 +0900
+++ src/libite-2.5.3/test/copyfile.c	2023-05-19 08:48:15.857371600 +0900
@@ -6,9 +6,9 @@
 #define ERRX(fmt, args...) { printf(fmt "\n", ##args); rc = 1; }
 
 static char *files[] = {
-	"/etc/passwd", "/tmp/mypwd", "/tmp/mypwd",
-	"/etc/passwd", "/tmp/",      "/tmp/passwd",
-	"/etc/passwd", "/tmp",       "/tmp/passwd",
+	"/etc/fstab", "/tmp/mypwd", "/tmp/mypwd",
+	"/etc/fstab", "/tmp/",      "/tmp/fstab",
+	"/etc/fstab", "/tmp",       "/tmp/fstab",
 	NULL
 };
 
--- origsrc/libite-2.5.3/test/fsendfile.c	2023-04-08 14:05:58.000000000 +0900
+++ src/libite-2.5.3/test/fsendfile.c	2023-05-19 08:19:46.533459200 +0900
@@ -6,9 +6,9 @@ int main(void)
 {
 	int i = 0;
 	char *files[] = {
-		"/etc/passwd", "/tmp/tok",
-		"/etc/passwd", "/tmp/passwd",
-		"/etc/passwd", "/tmp/passwd",
+		"/etc/fstab", "/tmp/tok",
+		"/etc/fstab", "/tmp/passwd",
+		"/etc/fstab", "/tmp/passwd",
 		NULL
 	};
 	FILE *src, *dst;
