--- origsrc/phat-0.4.1/configure.ac	2007-05-02 09:25:49.000000000 +0900
+++ src/phat-0.4.1/configure.ac	2015-04-30 18:44:31.131698100 +0900
@@ -49,7 +49,7 @@ GTK_DOC_CHECK(1.0)
 PKG_CHECK_MODULES(MISC, libgnomecanvas-2.0)
 
 # set compilation flags
-CFLAGS="$CFLAGS $GTK_CFLAGS $MISC_CFLAGS -Wall -Werror"
+CFLAGS="$CFLAGS $GTK_CFLAGS $MISC_CFLAGS -Wall"
 LIBS="$LIBS $GTK_LIBS $MISC_LIBS"
 
 # print build summary
--- origsrc/phat-0.4.1/demos/Makefile.am	2006-03-30 17:34:30.000000000 +0900
+++ src/phat-0.4.1/demos/Makefile.am	2015-04-30 18:44:31.131698100 +0900
@@ -25,3 +25,4 @@ phatknob_CFLAGS = `pkg-config --cflags g
 phatknob_LIBS = `pkg-config --libs gtk+-2.0`
 phatknob_LDADD = ../phat/libphat.la
 
+INCLUDES = -I$(top_srcdir)
--- origsrc/phat-0.4.1/phat/Makefile.am	2006-03-30 17:24:08.000000000 +0900
+++ src/phat-0.4.1/phat/Makefile.am	2015-04-30 18:44:31.142571900 +0900
@@ -6,7 +6,9 @@ phatprivate.c phatsliderbutton.c phatsli
 phatkeyboard.h phatvkeyboard.c phatvkeyboard.h phathkeyboard.c		\
 phathkeyboard.h phatpad.c phatpad.h phatknob.c phatknob.h
 
-libphat_la_CFLAGS = $(CFLAGS) -I.. -DINSTALL_DIR=\"$(datadir)\"
+libphat_la_CFLAGS = $(CFLAGS) -I$(top_srcdir) -DINSTALL_DIR=\"$(datadir)\"
+
+libphat_la_LDFLAGS = -no-undefined
 
 libphatincludedir = $(includedir)/phat
 
