--- origsrc/solidity-0.7.1/CMakeLists.txt	2020-09-02 21:43:57.000000000 +0900
+++ src/solidity-0.7.1/CMakeLists.txt	2020-09-18 18:25:57.180897300 +0900
@@ -28,7 +28,11 @@ include(EthCcache)
 
 # Let's find our dependencies
 include(EthDependencies)
-include(jsoncpp)
+find_package(PkgConfig)
+pkg_check_modules(JSONCPP jsoncpp)
+if (NOT JSONCPP_FOUND)
+  include(jsoncpp)
+endif()
 include_directories(SYSTEM ${JSONCPP_INCLUDE_DIR})
 
 find_package(Threads)
