--- origsrc/solidity-0.8.4/CMakeLists.txt	2021-04-21 22:09:37.000000000 +0900
+++ src/solidity-0.8.4/CMakeLists.txt	2021-05-10 22:11:28.836034200 +0900
@@ -38,9 +38,14 @@ option(SOLC_STATIC_STDLIBS "Link solc ag
 # Setup cccache.
 include(EthCcache)
 
+find_package(pkgconfig)
+
 # Let's find our dependencies
 include(EthDependencies)
-include(jsoncpp)
+pkg_check_modules(JSONCPP jsoncpp)
+if (NOT JSONCPP_FOUND)
+  include(jsoncpp)
+endif()
 include(range-v3)
 include_directories(SYSTEM ${JSONCPP_INCLUDE_DIR})
 
--- origsrc/solidity-0.8.4/cmake/EthCompilerSettings.cmake	2021-04-21 22:09:37.000000000 +0900
+++ src/solidity-0.8.4/cmake/EthCompilerSettings.cmake	2021-05-10 22:15:53.355644900 +0900
@@ -53,10 +53,7 @@ if (("${CMAKE_CXX_COMPILER_ID}" MATCHES
 	add_compile_options(-Wimplicit-fallthrough)
 	add_compile_options(-Wsign-conversion)
 	add_compile_options(-Wconversion)
-
-	eth_add_cxx_compiler_flag_if_supported(
-		$<$<COMPILE_LANGUAGE:CXX>:-Wextra-semi>
-	)
+	
 	eth_add_cxx_compiler_flag_if_supported(-Wfinal-dtor-non-final-class)
 	eth_add_cxx_compiler_flag_if_supported(-Wnewline-eof)
 	eth_add_cxx_compiler_flag_if_supported(-Wsuggest-destructor-override)
Binary files origsrc/solidity-0.8.4/deps/downloads/range-v3-0.11.0.tar.gz and src/solidity-0.8.4/deps/downloads/range-v3-0.11.0.tar.gz differ
