--- origsrc/solidity-0.5.9/CMakeLists.txt	2019-05-29 01:49:01.000000000 +0900
+++ src/solidity-0.5.9/CMakeLists.txt	2019-06-09 08:43:58.019563500 +0900
@@ -23,7 +23,11 @@ include(EthCcache)
 
 # Let's find our dependencies
 include(EthDependencies)
-include(jsoncpp)
+find_package(PkgConfig)
+pkg_check_modules(JSONCPP jsoncpp)
+if (NOT JSONCPP_FOUND)
+	include(jsoncpp)
+endif()
 include_directories(SYSTEM ${JSONCPP_INCLUDE_DIR})
 
 find_package(Threads)
--- origsrc/solidity-0.5.9/commit_hash.txt	1970-01-01 09:00:00.000000000 +0900
+++ src/solidity-0.5.9/commit_hash.txt	2019-06-09 08:47:29.784336500 +0900
@@ -0,0 +1 @@
+c68bc34e
--- origsrc/solidity-0.5.9/libyul/optimiser/BlockHasher.cpp	2019-05-29 01:49:01.000000000 +0900
+++ src/solidity-0.5.9/libyul/optimiser/BlockHasher.cpp	2019-06-09 09:07:21.622048200 +0900
@@ -27,6 +27,10 @@ using namespace std;
 using namespace dev;
 using namespace yul;
 
+#if defined(_N)
+#undef _N
+#endif
+
 namespace
 {
 static constexpr uint64_t compileTimeLiteralHash(char const* _literal, size_t _N)
