--- origsrc/getht-0.3/Makefile	2010-06-26 03:14:44.000000000 +0900
+++ src/getht-0.3/Makefile	2026-07-03 22:28:03.208068400 +0900
@@ -14,12 +14,10 @@ all: ${TARGETS}
 ${OBJ}: config.mk
 
 .c.o:
-	@echo CC $<
-	@cc -c ${CFLAGS} $<
+	cc -c ${CFLAGS} $<
 
 $(TARGETS): $(OBJ)
-	@echo LD $@
-	@cc -o $@ $(OBJ) $(LDFLAGS)
+	cc -o $@ $(OBJ) $(LDFLAGS)
 
 clean:
 	rm -f -- ${TARGETS} ${OBJ} ${NAME}-${VERSION}.tar.gz ${NAME}-${VERSION}.tar.gz.sig
--- origsrc/getht-0.3/config.mk	2010-06-26 03:14:44.000000000 +0900
+++ src/getht-0.3/config.mk	2026-07-03 22:28:25.255975100 +0900
@@ -16,8 +16,10 @@ XMLINC = $(shell pkg-config --cflags lib
 XMLLIB = $(shell pkg-config --libs libxml-2.0)
 
 # flags
-CFLAGS = -std=c99 -Wall -Werror ${INCS} ${CURLINC} ${XMLINC} -DVERSION=\"${VERSION}\"
-LDFLAGS = ${LIBS} ${CURLLIB} ${XMLLIB}
+CFLAGS ?=
+CFLAGS += -std=c99 -Wall ${INCS} ${CURLINC} ${XMLINC} -DVERSION=\"${VERSION}\"
+LDFLAGS ?=
+LDFLAGS += ${LIBS} ${CURLLIB} ${XMLLIB}
 
 # compiler and linker
-CC = cc
+CC ?= cc
--- origsrc/getht-0.3/download.c	2010-06-26 03:14:44.000000000 +0900
+++ src/getht-0.3/download.c	2026-07-03 22:25:07.948082300 +0900
@@ -12,7 +12,6 @@
 #include <sys/types.h>
 #include <dirent.h>
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 #include "getht.h"
