--- origsrc/libcppdigest/CMakeLists.txt	2008-01-16 01:53:10.000000000 +0900
+++ src/libcppdigest/CMakeLists.txt	2013-10-31 21:24:48.406160500 +0900
@@ -21,5 +21,8 @@ SET( test_SRCs testcppdigest.cpp )
 ADD_EXECUTABLE( testcppdigest ${test_SRCs} )
 TARGET_LINK_LIBRARIES( testcppdigest cppdigest ${LibDigest_LIBRARY} ${PCRECPP_LIBRARIES} )
 
-INSTALL( TARGETS cppdigest LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/usr/lib )
-INSTALL( FILES ${PROJECT_SOURCE_DIR}/cppdigest.h DESTINATION ${CMAKE_INSTALL_PREFIX}/usr/include )
+INSTALL( TARGETS cppdigest RUNTIME DESTINATION bin ARCHIVE DESTINATION lib )
+INSTALL( FILES ${PROJECT_SOURCE_DIR}/cppdigest.h DESTINATION include/libcppdigest )
+
+ENABLE_TESTING()
+ADD_TEST(testcppdigest testcppdigest)
--- origsrc/libcppdigest/cmake/FindLibDigest.cmake	2008-01-15 21:02:05.000000000 +0900
+++ src/libcppdigest/cmake/FindLibDigest.cmake	2013-10-31 21:19:52.405589000 +0900
@@ -12,7 +12,7 @@
 # Redistribution and use is allowed according to the terms of the BSD license.
 # For details see the accompanying COPYING-CMAKE-SCRIPTS file.
 
-FIND_PATH( LibDigest_INCLUDE_DIR NAMES digest.h PATHS ENV PATH PATH_SUFFIXES include digest )
+FIND_PATH( LibDigest_INCLUDE_DIR NAMES libdigest/digest.h PATHS ENV PATH PATH_SUFFIXES include digest)
 
 IF( LibDigest_INCLUDE_DIR )
         FIND_LIBRARY( LibDigest_LIBRARY NAMES digest PATHS PATH PATH_SUFFIXES lib lib-release lib_release )
--- origsrc/libcppdigest/cppdigest.cpp	2008-01-16 02:10:05.000000000 +0900
+++ src/libcppdigest/cppdigest.cpp	2013-10-31 21:23:00.907358000 +0900
@@ -31,7 +31,9 @@
 
 #include <pcrecpp.h>
 #include <fstream>
-#include <digest.h> // From libdigest
+extern "C" {
+#include <libdigest/digest.h> // From libdigest
+}
 #include "cppdigest.h"
 
 #ifdef DEBUG
