--- origsrc/libcppdigest/CMakeLists.txt	2008-01-16 01:53:10.000000000 +0900
+++ src/libcppdigest/CMakeLists.txt	2015-04-24 16:33:33.155885200 +0900
@@ -21,5 +21,14 @@ SET( test_SRCs testcppdigest.cpp )
 ADD_EXECUTABLE( testcppdigest ${test_SRCs} )
 TARGET_LINK_LIBRARIES( testcppdigest cppdigest ${LibDigest_LIBRARY} ${PCRECPP_LIBRARIES} )
 
-INSTALL( TARGETS cppdigest LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/usr/lib )
-INSTALL( FILES ${PROJECT_SOURCE_DIR}/cppdigest.h DESTINATION ${CMAKE_INSTALL_PREFIX}/usr/include )
+INSTALL( TARGETS cppdigest RUNTIME DESTINATION bin ARCHIVE DESTINATION lib )
+INSTALL( FILES ${PROJECT_SOURCE_DIR}/cppdigest.h DESTINATION include/libcppdigest )
+
+ENABLE_TESTING()
+ADD_TEST(testcppdigest testcppdigest)
+
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "0.2")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libcppdigest.pc.in ${CMAKE_BINARY_DIR}/libcppdigest.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libcppdigest.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libcppdigest/cmake/FindLibDigest.cmake	2008-01-15 21:02:05.000000000 +0900
+++ src/libcppdigest/cmake/FindLibDigest.cmake	2015-04-24 16:30:22.263078600 +0900
@@ -12,7 +12,7 @@
 # Redistribution and use is allowed according to the terms of the BSD license.
 # For details see the accompanying COPYING-CMAKE-SCRIPTS file.
 
-FIND_PATH( LibDigest_INCLUDE_DIR NAMES digest.h PATHS ENV PATH PATH_SUFFIXES include digest )
+FIND_PATH( LibDigest_INCLUDE_DIR NAMES libdigest/digest.h PATHS ENV PATH PATH_SUFFIXES include digest)
 
 IF( LibDigest_INCLUDE_DIR )
         FIND_LIBRARY( LibDigest_LIBRARY NAMES digest PATHS PATH PATH_SUFFIXES lib lib-release lib_release )
--- origsrc/libcppdigest/cppdigest.cpp	2008-01-16 02:10:05.000000000 +0900
+++ src/libcppdigest/cppdigest.cpp	2015-04-24 16:30:22.263078600 +0900
@@ -31,7 +31,9 @@
 
 #include <pcrecpp.h>
 #include <fstream>
-#include <digest.h> // From libdigest
+extern "C" {
+#include <libdigest/digest.h> // From libdigest
+}
 #include "cppdigest.h"
 
 #ifdef DEBUG
--- origsrc/libcppdigest/libcppdigest.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libcppdigest/libcppdigest.pc.in	2015-04-24 16:32:16.799810500 +0900
@@ -0,0 +1,5 @@
+Name: libcppdigest
+Description: C++ library for managing htdigest-like file
+Version: @VERSION@
+Libs: -L@libdir@ -lcppdigest
+Cflags: -I@includedir@
