--- origsrc/libgzipf-20240422/Makefile.am	2024-04-22 13:38:48.000000000 +0900
+++ src/libgzipf-20240422/Makefile.am	2025-05-18 14:08:07.410609300 +0900
@@ -3,23 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
 	libgzipf \
 	gzipftools \
 	pygzipf \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libgzipf-20240422/configure.ac	2024-04-22 13:38:48.000000000 +0900
+++ src/libgzipf-20240422/configure.ac	2025-05-18 14:08:07.410609300 +0900
@@ -174,19 +174,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
 AC_CONFIG_FILES([libgzipf/Makefile])
 AC_CONFIG_FILES([pygzipf/Makefile])
 AC_CONFIG_FILES([gzipftools/Makefile])
--- origsrc/libgzipf-20240422/pygzipf/Makefile.am	2024-04-22 13:38:48.000000000 +0900
+++ src/libgzipf-20240422/pygzipf/Makefile.am	2025-05-18 14:08:22.030329100 +0900
@@ -43,7 +43,7 @@ pygzipf_la_LIBADD = \
 	@LIBBFIO_LIBADD@
 
 pygzipf_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pygzipf_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pygzipf_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
--- origsrc/libgzipf-20240422/tests/Makefile.am	2024-04-22 13:38:48.000000000 +0900
+++ src/libgzipf-20240422/tests/Makefile.am	2025-05-18 14:18:52.625718100 +0900
@@ -3,6 +3,7 @@ AUTOMAKE_OPTIONS = subdir-objects
 AM_CPPFLAGS = \
 	-I../include -I$(top_srcdir)/include \
 	-I../common -I$(top_srcdir)/common \
+	-I$(top_builddir)/libgzipf \
 	@LIBCERROR_CPPFLAGS@ \
 	@LIBCTHREADS_CPPFLAGS@ \
 	@LIBCDATA_CPPFLAGS@ \
