--- origsrc/libpagekite-0.91.201110/configure.ac	2020-11-11 02:19:08.000000000 +0900
+++ src/libpagekite-0.91.201110/configure.ac	2023-09-06 03:55:52.047165100 +0900
@@ -5,7 +5,7 @@ AC_PREREQ([2.68])
 
 # libpagekite is Copyright 2011-2016, The Beanstalks Project ehf.
 
-AC_INIT([libpagekite], [0.90.150224], [info@pagekite.net])
+AC_INIT([libpagekite], [0.91.201110], [info@pagekite.net])
 AM_INIT_AUTOMAKE([-Wall -Werror foreign])
 AC_CONFIG_SRCDIR([libpagekite/pkconn.h])
 AC_CONFIG_HEADERS([config.h])
