--- origsrc/libredstone-0.1.0/src/Makefile.am	2013-08-05 10:49:41.000000000 +0900
+++ src/libredstone-0.1.0/src/Makefile.am	2014-06-07 16:41:45.826817700 +0900
@@ -31,6 +31,6 @@ libredstone_la_SOURCES = \
     $(C_FILES)           \
     $(H_FILES)
 
-libredstone_la_LDFLAGS = -version-info $(LIBREDSTONE_LT_VERSION)
+libredstone_la_LDFLAGS = -no-undefined -version-info $(LIBREDSTONE_LT_VERSION)
 
 pkginclude_HEADERS = $(H_FILES)
