--- origsrc/libeconf-0.3.7/lib/Makefile.am	2020-06-06 00:55:16.000000000 +0900
+++ src/libeconf-0.3.7/lib/Makefile.am	2020-06-10 17:38:45.986617600 +0900
@@ -4,7 +4,7 @@ libeconf_la_SOURCES = libeconf.c getfile
 libeconf_la_CFLAGS = -D_REENTRANT=1 @CFLAGS_CHECKS@ @CFLAGS_WARNINGS@
 libeconf_la_CPPFLAGS = -I$(top_srcdir)/include
 libeconf_la_LDFLAGS = @LDFLAGS_CHECKS@ @CFLAGS_WARNINGS@ \
-	-version-info 3:1:3 -Wl,--no-undefined \
+	-version-info 3:1:3 -no-undefined \
 	-Wl,--version-script=$(top_srcdir)/lib/libeconf.map
 
 CLEANFILES = *~
