diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libmaia-0.9.0/maia.pro src/libmaia-0.9.0/maia.pro
--- origsrc/libmaia-0.9.0/maia.pro	2015-08-23 00:19:53.000000000 +0900
+++ src/libmaia-0.9.0/maia.pro	2015-11-10 23:29:46.711430800 +0900
@@ -6,7 +6,8 @@ include("maia.pri")
 
 TEMPLATE = lib #app
 TARGET = maia
-CONFIG += staticlib
+# CONFIG += staticlib
+PREFIX = /usr
 
 DEPENDPATH += .
 INCLUDEPATH += .
@@ -27,4 +28,3 @@ INSTALLS += \
     headers \
 
 # Input
-
