--- origsrc/geda-gaf-1.9.2/libgeda/src/Makefile.am	2015-08-20 00:35:54.000000000 +0900
+++ src/geda-gaf-1.9.2/libgeda/src/Makefile.am	2017-11-13 14:21:32.404000000 +0900
@@ -90,10 +90,10 @@ libgeda_la_CPPFLAGS = -DLOCALEDIR=\"$(lo
 	-I$(srcdir)/../include -I$(srcdir)/../include/libgeda -I$(top_srcdir)
 libgeda_la_CFLAGS = \
 	$(GCC_CFLAGS) $(MINGW_CFLAGS) $(GUILE_CFLAGS) $(GLIB_CFLAGS) \
-	$(GDK_PIXBUF_CFLAGS)
-libgeda_la_LDFLAGS = -version-info $(LIBGEDA_SHLIB_VERSION) \
+	$(GDK_PIXBUF_CFLAGS) $(GIO_CFLAGS)
+libgeda_la_LDFLAGS = -no-undefined -version-info $(LIBGEDA_SHLIB_VERSION) \
 	$(WINDOWS_LIBTOOL_FLAGS) $(MINGW_LDFLAGS) $(GUILE_LIBS) \
-	$(GLIB_LIBS) $(GDK_PIXBUF_LIBS)
+	$(GLIB_LIBS) $(GDK_PIXBUF_LIBS) $(GIO_LIBS)
 LIBTOOL=@LIBTOOL@ --silent
 
 # This is used to generate boilerplate for defining Scheme functions
--- origsrc/geda-gaf-1.9.2/libgedacairo/Makefile.am	2015-08-20 00:35:54.000000000 +0900
+++ src/geda-gaf-1.9.2/libgedacairo/Makefile.am	2017-11-13 14:27:42.565000000 +0900
@@ -23,7 +23,8 @@ AM_CPPFLAGS = \
 AM_CFLAGS = \
 	$(GCC_CFLAGS) $(MINGW_CFLAGS) $(GUILE_CFLAGS) $(GLIB_CFLAGS) \
 	$(GDK_CFLAGS) $(CAIRO_CFLAGS) $(PANGO_CFLAGS) $(GDK_PIXBUF_CFLAGS)
-AM_LDFLAGS = -version-info $(LIBGEDACAIRO_SHLIB_VERSION) \
+AM_LDFLAGS = -no-undefined -version-info $(LIBGEDACAIRO_SHLIB_VERSION) \
+	$(top_builddir)/libgeda/src/libgeda.la \
 	$(WINDOWS_LIBTOOL_FLAGS) $(MINGW_LDFLAGS) $(GUILE_LIBS) \
 	$(GDK_LIBS) $(GLIB_LIBS) $(CAIRO_LIBS) $(PANGO_LIBS) $(GDK_PIXBUF_LIBS)
 LIBTOOL=@LIBTOOL@ --silent
