--- origsrc/geda-gaf-1.10.2/libgeda/libgeda.pc.in	2020-12-13 00:20:18.000000000 +0900
+++ src/geda-gaf-1.10.2/libgeda/libgeda.pc.in	2025-12-09 07:11:39.155945000 +0900
@@ -9,7 +9,7 @@ Name: libgeda
 Description: gEDA/gaf's core library 
 Requires: glib-2.0 gdk-pixbuf-2.0 gio-2.0 @GUILE_PKG_NAME@
 Requires.private:
-Version: @DATE_VERSION@
+Version: @VERSION@
 Libs: -L${libdir} -lgeda
 Libs.private:
 Cflags: -I${includedir}
--- origsrc/geda-gaf-1.10.2/libgeda/src/Makefile.am	2020-12-17 23:33:05.000000000 +0900
+++ src/geda-gaf-1.10.2/libgeda/src/Makefile.am	2025-12-09 02:06:12.770817100 +0900
@@ -91,7 +91,7 @@ libgeda_la_CPPFLAGS = -DLOCALEDIR=\"$(lo
 libgeda_la_CFLAGS = \
 	$(GCC_CFLAGS) $(MINGW_CFLAGS) $(GUILE_CFLAGS) $(GLIB_CFLAGS) \
 	$(GIO_CFLAGS) $(GDK_PIXBUF_CFLAGS)
-libgeda_la_LDFLAGS = -version-info $(LIBGEDA_SHLIB_VERSION) \
+libgeda_la_LDFLAGS = -no-undefined -version-info $(LIBGEDA_SHLIB_VERSION) \
 	$(WINDOWS_LIBTOOL_FLAGS) $(MINGW_LDFLAGS) $(GUILE_LIBS) \
 	$(GLIB_LIBS) $(GIO_LIBS) $(GDK_PIXBUF_LIBS)
 LIBTOOL=@LIBTOOL@ --silent
--- origsrc/geda-gaf-1.10.2/libgedacairo/Makefile.am	2020-12-13 00:20:18.000000000 +0900
+++ src/geda-gaf-1.10.2/libgedacairo/Makefile.am	2025-12-09 02:13:10.540696500 +0900
@@ -23,9 +23,10 @@ AM_CPPFLAGS = \
 AM_CFLAGS = \
 	$(GCC_CFLAGS) $(MINGW_CFLAGS) $(GUILE_CFLAGS) $(GLIB_CFLAGS) \
 	$(GDK_CFLAGS) $(CAIRO_CFLAGS) $(PANGO_CFLAGS) $(GDK_PIXBUF_CFLAGS)
-AM_LDFLAGS = -version-info $(LIBGEDACAIRO_SHLIB_VERSION) \
+AM_LDFLAGS = -no-undefined -version-info $(LIBGEDACAIRO_SHLIB_VERSION) \
 	$(WINDOWS_LIBTOOL_FLAGS) $(MINGW_LDFLAGS) $(GUILE_LIBS) \
-	$(GDK_LIBS) $(GLIB_LIBS) $(CAIRO_LIBS) $(PANGO_LIBS) $(GDK_PIXBUF_LIBS)
+	$(GDK_LIBS) $(GLIB_LIBS) $(CAIRO_LIBS) $(PANGO_LIBS) $(GDK_PIXBUF_LIBS) \
+	$(top_builddir)/libgeda/src/libgeda.la
 LIBTOOL=@LIBTOOL@ --silent
 
 pkgconfigdir = $(libdir)/pkgconfig
--- origsrc/geda-gaf-1.10.2/libgedacairo/libgedacairo.pc.in	2020-12-13 00:20:18.000000000 +0900
+++ src/geda-gaf-1.10.2/libgedacairo/libgedacairo.pc.in	2025-12-09 07:11:51.295269000 +0900
@@ -7,7 +7,7 @@ Name: libgedacairo
 Description: gEDA/gaf schematic renderer library
 Requires: libgeda glib-2.0 gobject-2.0 gdk-2.0 gdk-pixbuf-2.0 cairo pangocairo
 Requires.private:
-Version: @DATE_VERSION@
+Version: @VERSION@
 Libs: -L${libdir} -lgedacairo -lm
 Libs.private:
 Cflags: -I${includedir}
