--- origsrc/geda-gaf-1.8.2/configure.ac	2013-09-26 06:55:18.000000000 +0900
+++ src/geda-gaf-1.8.2/configure.ac	2013-09-28 22:49:34.189630200 +0900
@@ -75,6 +75,9 @@ AX_CHECK_GUILE([1.8.0])
 PKG_CHECK_MODULES(GLIB, [glib-2.0 >= 2.20.0], ,
   AC_MSG_ERROR([GLib 2.20.0 or later is required.]))
 
+PKG_CHECK_MODULES(GIO, [gio-2.0 >= 2.20.0], ,
+  AC_MSG_ERROR([Gio 2.20.0 or later is required.]))
+
 PKG_CHECK_MODULES(GTK, [gtk+-2.0 >= 2.16.0], ,
   AC_MSG_ERROR([GTK+ 2.16.0 or later is required.]))
 
--- origsrc/geda-gaf-1.8.2/libgeda/src/Makefile.am	2013-09-26 05:59:27.000000000 +0900
+++ src/geda-gaf-1.8.2/libgeda/src/Makefile.am	2013-09-28 22:49:34.189630200 +0900
@@ -87,9 +87,9 @@ libgeda_la_CPPFLAGS = -DLOCALEDIR=\"$(lo
 libgeda_la_CFLAGS = \
 	$(GCC_CFLAGS) $(MINGW_CFLAGS) $(GUILE_CFLAGS) $(GLIB_CFLAGS) \
 	$(GDK_PIXBUF_CFLAGS)
-libgeda_la_LDFLAGS = -version-info $(LIBGEDA_SHLIB_VERSION) \
+libgeda_la_LDFLAGS = -no-undefined -version-info $(LIBGEDA_SHLIB_VERSION) \
 	$(WINDOWS_LIBTOOL_FLAGS) $(MINGW_LDFLAGS) $(GUILE_LIBS) \
-	$(GLIB_LIBS) $(GDK_PIXBUF_LIBS)
+	$(GLIB_LIBS) $(GDK_PIXBUF_LIBS) $(GIO_LIBS)
 LIBTOOL=@LIBTOOL@ --silent
 
 # This is used to generate boilerplate for defining Scheme functions
