--- origsrc/nsync-1.30.0/CMakeLists.txt	2025-10-30 06:36:35.000000000 +0900
+++ src/nsync-1.30.0/CMakeLists.txt	2026-01-29 10:32:48.568185700 +0900
@@ -300,6 +300,7 @@ if (NSYNC_ENABLE_TESTS)
 		${NSYNC_TEST_OS_SRC}
 	)
 	add_library (nsync_test ${NSYNC_TEST_SRC})
+	target_link_libraries(nsync_test nsync)
 	set_target_properties (nsync_test PROPERTIES
 		VERSION ${PROJECT_VERSION}
 		SOVERSION ${PROJECT_VERSION_MAJOR})
@@ -322,6 +323,7 @@ if (NSYNC_ENABLE_TESTS)
 	endforeach (t)
 
 	add_library (nsync_test_cpp ${NSYNC_TEST_CPP_SRC})
+	target_link_libraries(nsync_test_cpp nsync_cpp)
 	set_target_properties (nsync_test_cpp PROPERTIES
 		VERSION ${PROJECT_VERSION}
 		SOVERSION ${PROJECT_VERSION_MAJOR})
@@ -400,11 +402,11 @@ endif ()
 set (CMAKE_SKIP_INSTALL_ALL_DEPENDENCY ON)
 
 install (TARGETS nsync EXPORT nsync
-	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT RuntimeLibraries
+	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT RuntimeLibraries
 	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT Development)
 
 install (TARGETS nsync_cpp EXPORT nsync_cpp OPTIONAL
-	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT RuntimeLibraries
+	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT RuntimeLibraries
 	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT Development)
 
 set (NSYNC_INCLUDES
