--- origsrc/pacman-v7.1.0/meson.build	2025-11-01 20:05:29.000000000 +0900
+++ src/pacman-v7.1.0/meson.build	2026-01-30 17:32:40.065838900 +0900
@@ -370,7 +370,7 @@ pacman_bin = executable(
   pacman_sources,
   include_directories : includes,
   link_with : [libalpm, libcommon],
-  dependencies : [libarchive],
+  dependencies : [libarchive, libintl],
   install : true,
 )
 
@@ -379,7 +379,7 @@ executable(
   pacman_conf_sources,
   include_directories : includes,
   link_with : [libalpm, libcommon],
-  dependencies : [libarchive],
+  dependencies : [libarchive, libintl],
   install : true,
 )
 
@@ -388,7 +388,7 @@ executable(
   testpkg_sources,
   include_directories : includes,
   link_with : [libalpm],
-  dependencies : [libarchive],
+  dependencies : [libarchive, libintl],
   install : true,
 )
 
@@ -396,7 +396,7 @@ executable(
   'vercmp',
   vercmp_sources,
   include_directories : includes,
-  link_with : [libalpm_a],
+  link_with : [libalpm],
   install : true,
 )
 
