--- origsrc/liblinear-2.11/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/liblinear-2.11/Makefile.am	2017-08-08 01:58:34.917000000 +0900
@@ -0,0 +1,17 @@
+
+bin_PROGRAMS = train predict
+train_SOURCES = train.c
+train_LDADD = liblinear.la
+predict_SOURCES = predict.c
+predict_LDADD = liblinear.la
+
+linear_includedir = $(includedir)/liblinear
+linear_include_HEADERS = linear.h tron.h
+
+lib_LTLIBRARIES = liblinear.la
+liblinear_la_SOURCES = linear.cpp tron.cpp
+liblinear_la_LIBADD = ${BLAS_LIBS}
+liblinear_la_LDFLAGS = -no-undefined -version-info 3
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/liblinear-2.11/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/liblinear-2.11/configure.ac	2017-08-08 01:58:37.761000000 +0900
@@ -0,0 +1,33 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([liblinear], [2.11])
+AC_CONFIG_SRCDIR([linear.cpp])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+AC_PROG_MAKE_SET
+LT_INIT
+
+# Checks for header files.
+AC_CHECK_HEADERS([locale.h stdlib.h string.h])
+
+PKG_CHECK_MODULES([BLAS], [blas])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+AC_C_INLINE
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_FUNC_STRTOD
+AC_CHECK_FUNCS([floor pow setlocale sqrt strdup strrchr strtol])
+
+AC_CONFIG_FILES([liblinear.pc Makefile])
+AC_OUTPUT
--- origsrc/liblinear-2.11/liblinear.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/liblinear-2.11/liblinear.pc.in	2017-08-08 01:58:31.730131500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: liblinear
+Description: Library for Large Linear Classification
+Version: @VERSION@
+Libs: -L${libdir} -llinear
+Cflags: -I${includedir}
