--- origsrc/liblinear-1.92/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/liblinear-1.92/Makefile.am	2013-01-14 13:36:13.375359400 +0900
@@ -0,0 +1,14 @@
+
+bin_PROGRAMS = train predict
+train_SOURCES = train.c
+train_LDADD = liblinear.la
+predict_SOURCES = predict.c
+predict_LDADD = liblinear.la
+
+linear_includedir = $(includedir)/liblinear
+linear_include_HEADERS = linear.h tron.h
+
+lib_LTLIBRARIES = liblinear.la
+liblinear_la_SOURCES = linear.cpp tron.cpp
+liblinear_la_LIBADD = @BLAS_LIBS@
+liblinear_la_LDFLAGS = -no-undefined -version-info 1:0:0
--- origsrc/liblinear-1.92/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/liblinear-1.92/configure.ac	2013-01-14 13:40:16.259148500 +0900
@@ -0,0 +1,33 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([liblinear], [1.92])
+AC_CONFIG_SRCDIR([linear.cpp])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+AC_PROG_MAKE_SET
+LT_INIT
+
+# Checks for libraries.
+PKG_CHECK_MODULES([BLAS], [blas])
+
+# Checks for header files.
+AC_CHECK_HEADERS([locale.h stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+AC_C_INLINE
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_FUNC_STRTOD
+AC_CHECK_FUNCS([setlocale sqrt strdup strrchr strtol])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
