--- origsrc/dynstr/CMakeLists.txt	2024-04-07 15:53:45.000000000 +0900
+++ src/dynstr/CMakeLists.txt	2024-04-07 16:00:20.712265600 +0900
@@ -12,8 +12,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 cmake_minimum_required(VERSION 3.0)
-project(dynstr C 0.1.0)
+project(dynstr LANGUAGES C VERSION 0.1.0)
 add_library(${PROJECT_NAME} "dynstr.c")
+if (DEFINED SOVERSION)
+    set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION "${SOVERSION}")
+endif()
 target_include_directories(${PROJECT_NAME} PUBLIC "include")
 install(TARGETS ${PROJECT_NAME})
 install(DIRECTORY include/ TYPE INCLUDE)
