--- origsrc/pslib-0.4.6/configure.in	2019-11-02 03:37:29.000000000 +0900
+++ src/pslib-0.4.6/configure.in	2019-12-09 17:01:58.863455900 +0900
@@ -206,6 +206,8 @@ if test "$enableval" = "no" ; then
 		AC_DEFINE(DISABLE_BMP, 1, "Support for bmp images")
 fi
 
+AC_CHECK_LIB(ghthash, ght_create, [GHTHASH_LIBS=-lghthash])
+
 dnl
 dnl Extra flags
 dnl
@@ -245,6 +247,8 @@ AC_SUBST(TIFF_LIBDIR)
 AC_SUBST(TIFF_LIBS)
 AC_SUBST(TIFF_INCLUDEDIR)
 
+AC_SUBST(GHTHASH_LIBS)
+
 AC_SUBST(HTML_DIR)
 dnl AC_SUBST(HAVE_ISNAN)
 dnl AC_SUBST(HAVE_ISINF)
--- origsrc/pslib-0.4.6/src/Makefile.am	2019-11-02 03:31:59.000000000 +0900
+++ src/pslib-0.4.6/src/Makefile.am	2019-12-09 17:03:04.348138900 +0900
@@ -7,7 +7,6 @@ noinst_HEADERS = \
 	ps_intern.h \
 	ps_memory.h \
 	ps_error.h \
-	ght_hash_table.h \
 	hyphen.h \
 	hnjalloc.h \
 	bmp.h \
@@ -17,7 +16,7 @@ noinst_HEADERS = \
 
 lib_LTLIBRARIES = libps.la
 
-libps_la_LDFLAGS = -version-info @LIBPS_VERSION_INFO@ $(LIBHNJ_LIBS) $(PNG_LIBDIR) $(JPEG_LIBDIR) $(GIF_LIBDIR) $(TIFF_LIBDIR)
+libps_la_LDFLAGS = -no-undefined -version-info @LIBPS_VERSION_INFO@ $(LIBHNJ_LIBS) $(PNG_LIBDIR) $(JPEG_LIBDIR) $(GIF_LIBDIR) $(TIFF_LIBDIR)
 
 libps_la_SOURCES = \
 	pslib.c \
@@ -27,8 +26,6 @@ libps_la_SOURCES = \
 	ps_afm.c \
 	ps_list.c \
 	ps_util.c \
-	hash_table.c \
-	hash_functions.c \
 	hyphen.c \
 	hnjalloc.c \
 	bmp.c \
@@ -36,7 +33,7 @@ libps_la_SOURCES = \
 	ps_fontenc.c \
 	ps_strbuf.c
 
-libps_la_LIBADD = $(LIBHNJ_LIBS) $(PNG_LIBS) $(JPEG_LIBS) $(GIF_LIBS) $(TIFF_LIBS)
+libps_la_LIBADD = $(LIBHNJ_LIBS) $(PNG_LIBS) $(JPEG_LIBS) $(GIF_LIBS) $(TIFF_LIBS) $(INTLLIBS) $(GHTHASH_LIBS)
 
 BUILD_LIBS = -lm
 
