--- origsrc/libcineon/configure.ac	2010-08-13 06:11:00.000000000 +0900
+++ src/libcineon/configure.ac	2014-12-27 19:40:12.307859300 +0900
@@ -16,7 +16,7 @@ AM_DISABLE_SHARED
 AM_PROG_LIBTOOL
 
 # Checks for libraries.
-AC_CHECK_LIB(tiff,_TIFFmalloc,have_libtiff='yes',have_libtiff='no')
+PKG_CHECK_MODULES([TIFF], [libtiff-4], [have_libtiff='yes'], [have_libtiff='no'])
 AM_CONDITIONAL(HAVE_LIBTIFF, test "$have_libtiff" != "no")
 
 # Checks for header files.
--- origsrc/libcineon/libcineon/CineonHeader.cpp	2010-08-13 06:11:00.000000000 +0900
+++ src/libcineon/libcineon/CineonHeader.cpp	2014-12-27 19:41:15.090085900 +0900
@@ -364,7 +364,10 @@ void cineon::Header::Reset()
 	IndustryHeader::Reset();
 }
 
-
+int cineon::GenericHeader::ImageElementComponentCount(const int element) const
+{
+	return 1;
+}
 
 int cineon::GenericHeader::ImageElementCount() const
 {
--- origsrc/libcineon/libcineon/Makefile.am	2010-08-11 04:22:46.000000000 +0900
+++ src/libcineon/libcineon/Makefile.am	2014-12-27 19:40:37.324460900 +0900
@@ -1,7 +1,7 @@
 
-lib_LIBRARIES = libcineon.a
+lib_LTLIBRARIES = libcineon.la
 
-libcineon_a_SOURCES = Codec.cpp \
+libcineon_la_SOURCES = Codec.cpp \
                    Cineon.cpp \
                    CineonHeader.cpp \
                    ElementReadStream.cpp \
@@ -10,6 +10,7 @@ libcineon_a_SOURCES = Codec.cpp \
                    Reader.cpp \
 				   TestFunc.cpp \
                    Writer.cpp
+libcineon_la_LDFLAGS = -no-undefined
 
 noinst_HEADERS = BaseTypeConverter.h \
 				 Codec.h \
--- origsrc/libcineon/tools/cineon2tiff/Makefile.am	2010-08-10 06:42:33.000000000 +0900
+++ src/libcineon/tools/cineon2tiff/Makefile.am	2014-12-27 19:45:22.379519200 +0900
@@ -1,10 +1,10 @@
 
-LIBCINEON = $(top_builddir)/libcineon/libcineon.a
+LIBCINEON = $(top_builddir)/libcineon/libcineon.la
 
-INCLUDES = -I$(top_builddir)/libcineon
+INCLUDES = -I$(top_srcdir)/libcineon
 
 bin_PROGRAMS = cineon2tiff
 
 cineon2tiff_SOURCES = cineon2tiff.cpp
-cineon2tiff_LDADD = $(LIBCINEON) -ltiff
+cineon2tiff_LDADD = $(LIBCINEON) $(TIFF_LIBS)
 
--- origsrc/libcineon/tools/cineonheader/Makefile.am	2010-08-10 06:42:33.000000000 +0900
+++ src/libcineon/tools/cineonheader/Makefile.am	2014-12-27 19:45:09.464480100 +0900
@@ -1,7 +1,7 @@
 
-LIBCINEON = $(top_builddir)/libcineon/libcineon.a
+LIBCINEON = $(top_builddir)/libcineon/libcineon.la
 
-INCLUDES = -I$(top_builddir)/libcineon
+INCLUDES = -I$(top_srcdir)/libcineon
 
 bin_PROGRAMS = cineonheader
 
