--- origsrc/GBTOLib-3.0.3/CMakeLists.txt	2021-12-22 02:32:50.000000000 +0900
+++ src/GBTOLib-3.0.3/CMakeLists.txt	2022-08-08 18:30:40.282505200 +0900
@@ -302,6 +302,10 @@ set_target_properties(libGBTO PROPERTIES
     LINK_FLAGS "${OpenMP_Fortran_FLAGS}"
 )
 
+if (DEFINED SOVERSION)
+    set_target_properties(libGBTO PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 add_executable(print_orbital_table "source/programs/print_orbital_table.f90")
 add_executable(basis_read          "source/programs/basis_read.f90")
 add_executable(scatci_integrals    "source/programs/scatci_integrals.f90")
@@ -337,3 +341,10 @@ if (DOXYGEN_FOUND)
     install(DIRECTORY "${CMAKE_CURRENT_LIST_DIR}/doc/html/"
             DESTINATION "${CMAKE_INSTALL_PREFIX}/doc/gbtolib-doxygen")
 endif (DOXYGEN_FOUND)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/GBTOLib.pc.in ${CMAKE_BINARY_DIR}/GBTOLib.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/GBTOLib.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/GBTOLib-3.0.3/GBTOLib.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/GBTOLib-3.0.3/GBTOLib.pc.in	2022-08-08 18:31:12.154649900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: GBTOLib
+Description: High-performance library for evaluation of molecular integrals
+Version: @VERSION@
+Libs: -L${libdir} -lGBTO
+Cflags: -I${includedir}/gbtolib
