--- origsrc/libyang-2.1.111/CMakeLists.txt	2023-08-18 19:23:24.000000000 +0900
+++ src/libyang-2.1.111/CMakeLists.txt	2023-08-28 23:04:52.065302200 +0900
@@ -414,7 +414,7 @@ endforeach()
 install(DIRECTORY "${PROJECT_SOURCE_DIR}/models/" DESTINATION ${YANG_MODULE_DIR} FILES_MATCHING PATTERN "*.yang")
 
 # install all library files
-install(TARGETS yang DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS yang RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(FILES ${headers} ${g_headers} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libyang)
 
 find_package(PkgConfig)
