--- origsrc/libyang-4.2.2/CMakeLists.txt	2025-12-01 21:59:32.000000000 +0900
+++ src/libyang-4.2.2/CMakeLists.txt	2026-01-19 16:42:29.985607200 +0900
@@ -430,7 +430,7 @@ endforeach()
 install(DIRECTORY "${PROJECT_SOURCE_DIR}/modules/" DESTINATION ${YANG_MODULE_DIR} FILES_MATCHING PATTERN "*.yang")
 
 # install all library files
-install(TARGETS yang DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS yang RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(FILES ${headers} ${g_headers} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libyang)
 
 find_package(PkgConfig)
