--- origsrc/z80ex-1.1.21/CMakeLists.txt	2013-07-29 08:45:44.000000000 +0900
+++ src/z80ex-1.1.21/CMakeLists.txt	2013-10-05 11:34:57.434242900 +0900
@@ -32,29 +32,12 @@ if (OPSTEP_FAST_AND_ROUGH)
 endif ()
 
 set (z80ex_sources z80ex.c)
-add_library (z80ex-static STATIC ${z80ex_sources})
-set_target_properties (z80ex-static PROPERTIES OUTPUT_NAME z80ex)
-if (NOT DEFINED Z80EX_STATIC_ONLY)
-    add_library (z80ex SHARED ${z80ex_sources})
-# Affects Win32 only: avoid dynamic/static *.lib files naming conflict 
-    set_target_properties (z80ex-static PROPERTIES PREFIX "lib")
-endif ()
+add_library (z80ex SHARED ${z80ex_sources})
 
 set (z80ex_dasm_sources z80ex_dasm.c)
-add_library (z80ex_dasm-static STATIC ${z80ex_dasm_sources})
-set_target_properties (z80ex_dasm-static PROPERTIES OUTPUT_NAME z80ex_dasm)
-if (NOT DEFINED Z80EX_STATIC_ONLY)
-    add_library (z80ex_dasm SHARED ${z80ex_dasm_sources})
-# Affects Win32 only: avoid dynamic/static *.lib files naming conflict 
-    set_target_properties (z80ex_dasm-static PROPERTIES PREFIX "lib")
-endif ()
+add_library (z80ex_dasm SHARED ${z80ex_dasm_sources})
 
-if (NOT DEFINED Z80EX_STATIC_ONLY)
-    set_target_properties(z80ex z80ex_dasm
-        PROPERTIES VERSION ${VERSION_STR} SOVERSION ${API_REVISION}
-    )
-endif ()
-set_target_properties(z80ex-static z80ex_dasm-static
+set_target_properties(z80ex z80ex_dasm
     PROPERTIES VERSION ${VERSION_STR} SOVERSION ${API_REVISION}
 )
 
@@ -63,8 +46,5 @@ if ("${CMAKE_C_IMPLICIT_LINK_DIRECTORIES
 else ()
     set (LIB_DIR "lib")
 endif ()
-if (NOT DEFINED Z80EX_STATIC_ONLY)
-    install (TARGETS z80ex z80ex_dasm LIBRARY DESTINATION ${LIB_DIR} ARCHIVE DESTINATION ${LIB_DIR})
-endif ()
-install (TARGETS z80ex-static z80ex_dasm-static LIBRARY DESTINATION ${LIB_DIR} ARCHIVE DESTINATION ${LIB_DIR})
+install (TARGETS z80ex z80ex_dasm RUNTIME DESTINATION bin LIBRARY DESTINATION ${LIB_DIR} ARCHIVE DESTINATION ${LIB_DIR})
 install (DIRECTORY include/ DESTINATION include/z80ex PATTERN "*.h" )
