--- origsrc/timblserver-1.12/demos/Makefile.am	2018-05-16 22:48:45.000000000 +0900
+++ src/timblserver-1.12/demos/Makefile.am	2019-03-29 17:24:10.855226100 +0900
@@ -2,7 +2,7 @@
 # $URL: $
 
 AM_CPPFLAGS = -I@top_srcdir@/include
-AM_CXXFLAGS = -std=c++0x
+AM_CXXFLAGS = -std=gnu++0x
 
 if WANT_OMP
 noinst_PROGRAMS = sockettestServer sockettestClient
--- origsrc/timblserver-1.12/src/Makefile.am	2018-05-16 22:48:45.000000000 +0900
+++ src/timblserver-1.12/src/Makefile.am	2019-03-29 17:24:10.855226100 +0900
@@ -9,6 +9,6 @@ timblclient_SOURCES = TimblClient.cxx
 timblserver_SOURCES = TimblServer.cxx
 
 lib_LTLIBRARIES = libtimblserver.la
-libtimblserver_la_LDFLAGS= -version-info 4:0:0
+libtimblserver_la_LDFLAGS= -no-undefined -version-info 4:0:0
 
 libtimblserver_la_SOURCES = ClientBase.cxx
