--- origsrc/Catch2-3.7.1/CMakeLists.txt	2024-09-17 17:45:43.000000000 +0900
+++ src/Catch2-3.7.1/CMakeLists.txt	2024-10-02 16:00:41.724108400 +0900
@@ -165,10 +165,17 @@ if (NOT_SUBPROJECT)
     endif()
 
     ## Provide some pkg-config integration
-    set(PKGCONFIG_INSTALL_DIR
-        "${CMAKE_INSTALL_DATAROOTDIR}/pkgconfig"
-        CACHE PATH "Path where catch2.pc is installed"
-    )
+    if (CYGWIN)
+      set(PKGCONFIG_INSTALL_DIR
+          "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
+          CACHE PATH "Path where catch2.pc is installed"
+      )
+    else()
+      set(PKGCONFIG_INSTALL_DIR
+          "${CMAKE_INSTALL_DATAROOTDIR}/pkgconfig"
+          CACHE PATH "Path where catch2.pc is installed"
+      )
+    endif()
     configure_file(
       ${CMAKE_CURRENT_SOURCE_DIR}/CMake/catch2.pc.in
       ${CMAKE_CURRENT_BINARY_DIR}/catch2.pc
