--- origsrc/libx52-0.3.3/daemon/meson.build	2026-03-13 04:10:26.000000000 +0900
+++ src/libx52-0.3.3/daemon/meson.build	2026-03-29 19:05:06.005749500 +0900
@@ -7,6 +7,8 @@ libx52dcomm_sources = [
 install_headers('x52dcomm.h', subdir: meson.project_name())
 
 lib_libx52dcomm = library('x52dcomm', libx52dcomm_sources,
+  install: true,
+  version: '0',
   dependencies: [dep_intl],
   include_directories: includes)
 
--- origsrc/libx52-0.3.3/meson.build	2026-03-13 04:10:26.000000000 +0900
+++ src/libx52-0.3.3/meson.build	2026-03-29 20:43:29.228294500 +0900
@@ -93,9 +93,7 @@ if dep_systemd.found() and not get_optio
   pinelog_options = ['show-date=false']
 endif
 
-sub_pinelog = subproject('pinelog', required: true,
-  default_options: pinelog_options)
-dep_pinelog = sub_pinelog.get_variable('libpinelog_dep')
+dep_pinelog = dependency('pinelog')
 
 # inih
 # Use system inih
