--- origsrc/editline-1.16.1/src/Makefile.am	2019-06-07 19:04:01.000000000 +0900
+++ src/editline-1.16.1/src/Makefile.am	2019-07-06 00:48:54.461351400 +0900
@@ -3,4 +3,4 @@ libeditline_la_SOURCES  = editline.c edi
 libeditline_la_CFLAGS   = -std=gnu99
 libeditline_la_CFLAGS  += -W -Wall -Wextra -Wundef -Wunused -Wstrict-prototypes
 libeditline_la_CFLAGS  += -Werror-implicit-function-declaration -Wshadow -Wcast-qual
-libeditline_la_LDFLAGS  = $(AM_LDFLAGS) -version-info 1:1:0
+libeditline_la_LDFLAGS  = $(AM_LDFLAGS) -no-undefined -version-info 1:1:0
