--- origsrc/editline-2.1.0/src/Makefile.am	2026-07-04 19:48:33.000000000 +0900
+++ src/editline-2.1.0/src/Makefile.am	2026-07-21 05:51:18.066249400 +0900
@@ -3,4 +3,4 @@ libeditline_la_SOURCES  = editline.c edi
 libeditline_la_CFLAGS   = -std=gnu99
 libeditline_la_CFLAGS  += -W -Wall -Wextra -Wundef -Wunused -Wstrict-prototypes
 libeditline_la_CFLAGS  += -Werror-implicit-function-declaration -Wshadow -Wcast-qual
-libeditline_la_LDFLAGS  = $(AM_LDFLAGS) -version-info 2:1:1
+libeditline_la_LDFLAGS  = $(AM_LDFLAGS) -no-undefined -version-info 2:1:1
