--- origsrc/isam_v1_7/CMakeLists.txt	2013-07-03 10:48:45.000000000 +0900
+++ src/isam_v1_7/CMakeLists.txt	2024-09-29 08:37:17.558470300 +0900
@@ -61,3 +61,10 @@ add_subdirectory(isamlib)
 add_subdirectory(isam)
 add_subdirectory(examples)
 add_subdirectory(misc)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/isam.pc.in ${CMAKE_BINARY_DIR}/isam.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/isam.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/isam_v1_7/include/isam/ChowLiuTree.h	2013-07-03 10:48:46.000000000 +0900
+++ src/isam_v1_7/include/isam/ChowLiuTree.h	2024-09-29 08:25:09.424327900 +0900
@@ -37,6 +37,9 @@
 
 #include "Node.h"
 
+#if defined(_L)
+#undef _L
+#endif
 
 namespace isam {
 
--- origsrc/isam_v1_7/isam.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/isam_v1_7/isam.pc.in	2024-09-29 08:36:46.533354500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: isam
+Description: Incremental Smoothing and Mapping
+Version: @VERSION@
+Libs: -L${libdir} -lisam
+Cflags: -I${includedir}
--- origsrc/isam_v1_7/isamlib/CMakeLists.txt	2013-07-03 10:48:31.000000000 +0900
+++ src/isam_v1_7/isamlib/CMakeLists.txt	2024-09-29 00:55:03.961984800 +0900
@@ -12,10 +12,14 @@ find_package(Cholmod REQUIRED)
 include_directories(${CHOLMOD_INCLUDES})
 target_link_libraries(isamlib ${CHOLMOD_LIBRARIES})
 
+if (DEFINED SOVERSION)
+  set_target_properties(isamlib PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 # install library
 install(TARGETS isamlib
   PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin
   ARCHIVE DESTINATION lib
 )
 # install header files
--- origsrc/isam_v1_7/isamlib/Cholesky.cpp	2013-07-03 10:48:46.000000000 +0900
+++ src/isam_v1_7/isamlib/Cholesky.cpp	2024-09-29 08:21:34.695378900 +0900
@@ -36,6 +36,10 @@
 #include "cs.h"
 #include "cholmod.h"
 
+#if defined(_L)
+#undef _L
+#endif
+
 // use CSparse instead of CHOLMOD (default)
 const bool USE_CSPARSE = false;
 
--- origsrc/isam_v1_7/isamlib/ChowLiuTree.cpp	2013-07-03 10:48:46.000000000 +0900
+++ src/isam_v1_7/isamlib/ChowLiuTree.cpp	2024-09-29 08:23:32.400982100 +0900
@@ -236,4 +236,4 @@ void ChowLiuTree::_build_tree_rec(int id
   tree[new_node.id] = new_node;
 }
 
-} //namespace isam
\ No newline at end of file
+} //namespace isam
