diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x include -x lib -x bamtools_version.h origsrc/bamtools-2.4.0/CMakeLists.txt src/bamtools-2.4.0/CMakeLists.txt
--- origsrc/bamtools-2.4.0/CMakeLists.txt	2015-06-17 01:50:53.000000000 +0900
+++ src/bamtools-2.4.0/CMakeLists.txt	2015-07-19 16:16:43.787699900 +0900
@@ -58,8 +58,19 @@ endif()
 
 # -------------------------------------------
 
+INCLUDE(FindPkgConfig)
+PKG_CHECK_MODULES(JSONCPP REQUIRED jsoncpp)
+
 # add our includes root path
 include_directories( src )
 
 # list subdirectories to build in
 add_subdirectory( src )
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "${BamTools_VERSION_MAJOR}.${BamTools_VERSION_MINOR}.${BamTools_VERSION_BUILD}")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/bamtools.pc.in ${CMAKE_BINARY_DIR}/bamtools.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/bamtools.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x include -x lib -x bamtools_version.h origsrc/bamtools-2.4.0/bamtools.pc.in src/bamtools-2.4.0/bamtools.pc.in
--- origsrc/bamtools-2.4.0/bamtools.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/bamtools-2.4.0/bamtools.pc.in	2015-07-19 16:16:43.787699900 +0900
@@ -0,0 +1,9 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+Name: bamtools
+Description: C++ API & command-line toolkit for working with BAM data
+Version: @VERSION@
+Libs: -L${libdir} -lbamtools
+Cflags: -I${includedir}
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x include -x lib -x bamtools_version.h origsrc/bamtools-2.4.0/src/CMakeLists.txt src/bamtools-2.4.0/src/CMakeLists.txt
--- origsrc/bamtools-2.4.0/src/CMakeLists.txt	2015-06-17 01:50:53.000000000 +0900
+++ src/bamtools-2.4.0/src/CMakeLists.txt	2015-07-19 16:16:43.801482500 +0900
@@ -6,7 +6,6 @@
 # ==========================
 
 add_subdirectory( api )
-add_subdirectory( third_party )
 add_subdirectory( toolkit )
 add_subdirectory( utils )
 
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x include -x lib -x bamtools_version.h origsrc/bamtools-2.4.0/src/api/CMakeLists.txt src/bamtools-2.4.0/src/api/CMakeLists.txt
--- origsrc/bamtools-2.4.0/src/api/CMakeLists.txt	2015-06-17 01:50:53.000000000 +0900
+++ src/bamtools-2.4.0/src/api/CMakeLists.txt	2015-07-19 16:16:43.815265100 +0900
@@ -10,7 +10,9 @@ include_directories( ${BamTools_SOURCE_D
 
 # add compiler definitions 
 add_definitions( -DBAMTOOLS_API_LIBRARY ) # (for proper exporting of library symbols)
+if( NOT CYGWIN )
 add_definitions( -fPIC ) # (attempt to force PIC compiling on CentOS, not being set on shared libs by CMake)
+endif()
 
 # fetch all internal source files
 add_subdirectory( internal )
@@ -38,10 +40,13 @@ set_target_properties( BamTools PROPERTI
                        OUTPUT_NAME "bamtools" )
 
 # create main BamTools API static library
+option( ENABLE_STATIC "enable static" OFF )
+if( ENABLE_STATIC )
 add_library( BamTools-static STATIC ${BamToolsAPISources} )
 set_target_properties( BamTools-static PROPERTIES 
                        OUTPUT_NAME "bamtools" 
                        PREFIX "lib" )
+endif()
 
 # link libraries automatically with zlib (and Winsock2, if applicable)
 if( WIN32 )
@@ -51,11 +56,15 @@ else()
 endif()
 
 target_link_libraries( BamTools        ${APILibs} )
+if( ENABLE_STATIC )
 target_link_libraries( BamTools-static ${APILibs} )
+endif()
 
 # set library install destinations
-install( TARGETS BamTools        LIBRARY DESTINATION "lib/bamtools" RUNTIME DESTINATION "bin")
+install( TARGETS BamTools        ARCHIVE DESTINATION "lib/bamtools" RUNTIME DESTINATION "bin")
+if( ENABLE_STATIC )
 install( TARGETS BamTools-static ARCHIVE DESTINATION "lib/bamtools")
+endif()
 
 # export API headers
 include(../ExportHeader.cmake)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x include -x lib -x bamtools_version.h origsrc/bamtools-2.4.0/src/api/internal/io/TcpSocketEngine_unix_p.cpp src/bamtools-2.4.0/src/api/internal/io/TcpSocketEngine_unix_p.cpp
--- origsrc/bamtools-2.4.0/src/api/internal/io/TcpSocketEngine_unix_p.cpp	2015-06-17 01:50:53.000000000 +0900
+++ src/bamtools-2.4.0/src/api/internal/io/TcpSocketEngine_unix_p.cpp	2015-07-19 16:16:43.815265100 +0900
@@ -21,6 +21,8 @@ using namespace BamTools::Internal;
 #include <iostream>
 using namespace std;
 
+#include <sys/select.h>
+
 // ------------------------
 // static utility methods
 // ------------------------
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x include -x lib -x bamtools_version.h origsrc/bamtools-2.4.0/src/toolkit/bamtools_filter.cpp src/bamtools-2.4.0/src/toolkit/bamtools_filter.cpp
--- origsrc/bamtools-2.4.0/src/toolkit/bamtools_filter.cpp	2015-06-17 01:50:53.000000000 +0900
+++ src/bamtools-2.4.0/src/toolkit/bamtools_filter.cpp	2015-07-19 16:16:43.815265100 +0900
@@ -612,7 +612,6 @@ bool FilterTool::FilterToolPrivate::Pars
   
     // filter object parsing variables
     Json::Value null(Json::nullValue);
-    Json::Value propertyValue;
     
     // store results
     map<string, string> propertyTokens;
@@ -624,7 +623,7 @@ bool FilterTool::FilterToolPrivate::Pars
         const string& propertyName = (*propertyNameIter);
         
         // if property defined in filter, add to token list
-        propertyValue = filterObject.get(propertyName, null);
+        const Json::Value& propertyValue = filterObject.get(propertyName, null);
         if ( propertyValue != null ) 
             propertyTokens.insert( make_pair(propertyName, propertyValue.asString()) );
     }
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x include -x lib -x bamtools_version.h origsrc/bamtools-2.4.0/src/utils/CMakeLists.txt src/bamtools-2.4.0/src/utils/CMakeLists.txt
--- origsrc/bamtools-2.4.0/src/utils/CMakeLists.txt	2015-06-17 01:50:53.000000000 +0900
+++ src/bamtools-2.4.0/src/utils/CMakeLists.txt	2015-07-19 16:16:43.829047700 +0900
@@ -10,7 +10,9 @@ include_directories( ${BamTools_SOURCE_D
 
 # add compiler definitions 
 add_definitions( -DBAMTOOLS_UTILS_LIBRARY ) # (for proper exporting of library symbols)
+if( NOT CYGWIN )
 add_definitions( -fPIC ) # (attempt to force PIC compiling on CentOS, not being set on shared libs by CMake)
+endif()
 
 # create BamTools utils library
 add_library( BamTools-utils STATIC
