--- origsrc/bamtools-2.4.1/src/api/CMakeLists.txt	2017-01-18 21:09:54.774562700 +0900
+++ src/bamtools-2.4.1/src/api/CMakeLists.txt	2017-01-18 21:24:49.875732700 +0900
@@ -43,7 +43,7 @@ set_target_properties( BamTools PROPERTI
                        SOVERSION "2.4.1"
                        OUTPUT_NAME "bamtools" )
 target_link_libraries( BamTools ${APILibs} )
-install( TARGETS BamTools LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" )
+install( TARGETS BamTools ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" )
 
 # create main BamTools API static library
 if( BUILD_STATIC )
