--- origsrc/cglm-0.8.7/cglm.pc.in	2022-10-27 05:15:54.000000000 +0900
+++ src/cglm-0.8.7/cglm.pc.in	2022-12-12 23:23:45.812641100 +0900
@@ -1,7 +1,7 @@
 prefix=@CMAKE_INSTALL_PREFIX@
-exec_prefix="${prefix}"
-libdir="${exec_prefix}/lib"
-includedir="${prefix}/include"
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
 
 Name: @PROJECT_NAME@
 Description: @CMAKE_PROJECT_DESCRIPTION@
--- origsrc/cglm-0.8.7/configure.ac	2022-10-27 05:15:54.000000000 +0900
+++ src/cglm-0.8.7/configure.ac	2022-12-12 23:23:20.491902900 +0900
@@ -70,6 +70,17 @@ AC_TYPE_UINT8_T
 # Checks for library functions.
 AC_FUNC_ERROR_AT_LINE
 
+PROJECT_NAME=$PACKAGE
+AC_SUBST([PROJECT_NAME])
+PROJECT_VERSION=$VERSION
+AC_SUBST([PROJECT_VERSION])
+CMAKE_PROJECT_DESCRIPTION="Highly Optimized Graphics Math (glm) for C"
+AC_SUBST([CMAKE_PROJECT_DESCRIPTION])
+CMAKE_PROJECT_HOMEPAGE_URL="https://github.com/recp/cglm"
+AC_SUBST([CMAKE_PROJECT_HOMEPAGE_URL])
+CMAKE_INSTALL_PREFIX=$prefix
+AC_SUBST([CMAKE_INSTALL_PREFIX])
+
 AC_CONFIG_FILES([Makefile cglm.pc])
 
 AC_OUTPUT
