--- origsrc/cglm-0.6.0/Makefile.am	2019-09-01 06:30:15.000000000 +0900
+++ src/cglm-0.6.0/Makefile.am	2019-10-12 21:23:53.936477800 +0900
@@ -18,13 +18,10 @@ AM_CFLAGS = -Wall \
 lib_LTLIBRARIES = libcglm.la
 libcglm_la_LDFLAGS = -no-undefined -version-info 0:1:0
 
-checkLDFLAGS = -L./.libs \
-               -L./test/lib/cmocka/build/src \
-               -lcmocka \
+checkLDFLAGS = -lcmocka \
                -lm \
-               -lcglm
-checkCFLAGS = -I./test/lib/cmocka/include \
-              -I./include
+               libcglm.la
+checkCFLAGS = -I$(top_srcdir)/include
 
 check_PROGRAMS = test/tests
 TESTS = $(check_PROGRAMS)
