--- origsrc/cglm-0.9.6/Makefile.am	2025-02-14 04:35:16.000000000 +0900
+++ src/cglm-0.9.6/Makefile.am	2025-04-09 09:11:57.104920700 +0900
@@ -23,7 +23,7 @@ checkCFLAGS = $(AM_CFLAGS) \
                -std=gnu11 \
                -O3 \
                -DCGLM_DEFINE_PRINTS \
-               -I./include
+               -I$(top_srcdir)/include
 
 check_PROGRAMS = test/tests
 TESTS = $(check_PROGRAMS)
