--- origsrc/cglm-0.9.2/Makefile.am	2023-12-31 19:49:06.000000000 +0900
+++ src/cglm-0.9.2/Makefile.am	2024-01-30 12:33:44.037664500 +0900
@@ -23,7 +23,7 @@ checkCFLAGS = $(AM_CFLAGS) \
                -std=gnu11 \
                -O3 \
                -DCGLM_DEFINE_PRINTS \
-               -I./include
+               -I$(top_srcdir)/include
 
 check_PROGRAMS = test/tests
 TESTS = $(check_PROGRAMS)
