--- origsrc/lib3mf-1.1.5/CMakeLists.txt	2018-04-24 22:40:48.000000000 +0900
+++ src/lib3mf-1.1.5/CMakeLists.txt	2019-01-02 22:55:03.798501900 +0900
@@ -124,7 +124,7 @@ add_library(${PROJECT_NAME} SHARED ${SRC
 )
 set(STARTUPPROJECT ${PROJECT_NAME})
 SET_TARGET_PROPERTIES(${PROJECT_NAME} PROPERTIES PREFIX "" IMPORT_PREFIX "")
-target_link_libraries(${PROJECT_NAME} ${PROJECT_NAME}_s)
+target_link_libraries(${PROJECT_NAME} ${PROJECT_NAME}_s z zip)
 if (UNIX)
 	# Uncomment the following to but the version info into the .so-file.
 	SET_TARGET_PROPERTIES(${PROJECT_NAME} PROPERTIES VERSION "${LIB3MF_VERSION_MAJOR}.${LIB3MF_VERSION_MINOR}.${LIB3MF_VERSION_MICRO}.${BUILD_NUMBER}")
@@ -173,3 +173,10 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL
 		set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT ${STARTUPPROJECT})
 	ENDIF()
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/lib3mf.pc.in ${CMAKE_BINARY_DIR}/lib3mf.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/lib3mf.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/lib3mf-1.1.5/Source/CMakeLists.txt	2018-04-24 22:40:48.000000000 +0900
+++ src/lib3mf-1.1.5/Source/CMakeLists.txt	2019-01-02 22:47:54.970283100 +0900
@@ -13,7 +13,6 @@ if(NMR_COM_NATIVE)
 else()
 	# sources for COM emulation
 	set(SRCS_COM
-	${LIBZIP_FILES}
 	./Source/Model/COM/NMR_COMInterface_GCC.cpp
 	./Source/Common/Platform/NMR_Platform_GCC.cpp
 	./Source/Common/OPC/NMR_OpcPackageReader.cpp
@@ -94,17 +93,6 @@ ${SRCS_COM}
 ./Source/Common/Platform/NMR_XmlWriter.cpp
 ./Source/Common/Platform/NMR_XmlWriter_GCC.cpp
 ./Source/Common/Platform/NMR_XmlWriter_Native.cpp
-./Source/Libraries/zlib/adler32.c
-./Source/Libraries/zlib/compress.c
-./Source/Libraries/zlib/crc32.c
-./Source/Libraries/zlib/deflate.c
-./Source/Libraries/zlib/infback.c
-./Source/Libraries/zlib/inffast.c
-./Source/Libraries/zlib/inflate.c
-./Source/Libraries/zlib/inftrees.c
-./Source/Libraries/zlib/trees.c
-./Source/Libraries/zlib/uncompr.c
-./Source/Libraries/zlib/zutil.c
 ./Source/Model/Classes/NMR_PackageResourceID.cpp
 ./Source/Model/Classes/NMR_Model.cpp
 ./Source/Model/Classes/NMR_ModelAttachment.cpp
--- origsrc/lib3mf-1.1.5/lib3mf.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/lib3mf-1.1.5/lib3mf.pc.in	2019-01-02 22:59:09.881261400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: lib3mf
+Description: Reference implementation of the 3D Manufacturing Format file standard
+Version: @LIB3MF_VERSION_MAJOR@.@LIB3MF_VERSION_MINOR@.@LIB3MF_VERSION_MICRO@
+Libs: -L${libdir} -l3mf
+Cflags: -I${includedir}
