--- origsrc/libaffa-0.9.6/Makefile.am	2003-04-07 07:45:36.000000000 +0900
+++ src/libaffa-0.9.6/Makefile.am	2020-11-07 00:03:13.915448300 +0900
@@ -1 +1,4 @@
 SUBDIRS = src
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libaffa-0.9.6/configure.in	2006-06-23 16:59:19.000000000 +0900
+++ src/libaffa-0.9.6/configure.in	2020-11-07 00:03:13.971400100 +0900
@@ -1,4 +1,4 @@
-AC_INIT(README)
+AC_INIT([libaffa], [0.9.6])
 
 AAF_MAJOR_VERSION=0
 AAF_MINOR_VERSION=9
@@ -24,10 +24,10 @@ AC_SUBST(LT_CURRENT)
 AC_SUBST(LT_REVISION)
 AC_SUBST(LT_AGE)
 
-AM_INIT_AUTOMAKE(libaffa, $AAF_VERSION)
+AM_INIT_AUTOMAKE
 AC_PROG_CXX
 CXXFLAGS="-O2 -funroll-loops -fomit-frame-pointer -fno-exceptions"
 AC_PROG_LIBTOOL
 AM_PROG_LIBTOOL
 
-AC_OUTPUT(Makefile src/Makefile)
+AC_OUTPUT(libaffa.pc Makefile src/Makefile)
--- origsrc/libaffa-0.9.6/libaffa.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libaffa-0.9.6/libaffa.pc.in	2020-11-07 00:03:14.016419600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libaffa
+Description: C++ Affine Arithmetic Library
+Version: @VERSION@
+Libs: -L${libdir} -laffa
+Cflags: -I${includedir}/libaffa
--- origsrc/libaffa-0.9.6/src/Makefile.am	2005-05-28 19:24:06.000000000 +0900
+++ src/libaffa-0.9.6/src/Makefile.am	2020-11-07 00:03:14.040410400 +0900
@@ -1,8 +1,8 @@
 lib_LTLIBRARIES = libaffa.la
 libaffa_la_SOURCES = aa_rounding.cpp aa_interval.cpp aa_aaftrigo.cpp aa_aafapprox.cpp aa_aafarithm.cpp aa_aafcommon.cpp
-libaffa_la_LDFLAGS =      \
+libaffa_la_LDFLAGS = -no-undefined \
 	-release $(LT_RELEASE)  \
         -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 
-include_HEADERS = aa.h aa_aaf.h aa_interval.h
+pkginclude_HEADERS = aa.h aa_aaf.h aa_interval.h
 noinst_HEADERS = aa_rounding.h
