--- origsrc/libdna-1.3/src/dna4_fill_random_x86.c	2022-10-31 00:18:13.000000000 +0900
+++ src/libdna-1.3/src/dna4_fill_random_x86.c	2022-11-16 18:10:38.790548000 +0900
@@ -31,7 +31,7 @@ dna4_fill_random_select(void)
 
 DNA_PUBLIC
 void
-dna4_fill_random(char *dest, char *end, uint32_t seed)
+dna4_fill_random(char *begin, char *end, uint32_t seed)
 	__attribute__((ifunc("dna4_fill_random_select")));
 
 #elif defined(__APPLE__)
@@ -54,23 +54,23 @@ dna4_fill_random_macho(void)
 // boolean variable.
 
 void
-dna4_fill_random_callonce(char *dest, char *end, uint32_t seed);
+dna4_fill_random_callonce(char *begin, char *end, uint32_t seed);
 
 static dna4_fill_random_fn *dna4_fill_random_fnptr = dna4_fill_random_callonce;
 
 DNA_LOCAL
 void
-dna4_fill_random_callonce(char *dest, char *end, uint32_t seed)
+dna4_fill_random_callonce(char *begin, char *end, uint32_t seed)
 {
 	dna4_fill_random_fnptr = dna4_fill_random_select();
-	return dna4_fill_random_fnptr(begin, end, dest);
+	dna4_fill_random_fnptr(begin, end, seed);
 }
 
 DNA_PUBLIC
 void
-dna4_fill_random(char *dest, char *end, uint32_t seed)
+dna4_fill_random(char *begin, char *end, uint32_t seed)
 {
-	return dna4_fill_random_fnptr(begin, end, dest);
+	dna4_fill_random_fnptr(begin, end, seed);
 }
 
 DNA_LOCAL
