--- origsrc/libghwp-0.1.1/Makefile.am	2012-12-24 00:33:19.000000000 +0900
+++ src/libghwp-0.1.1/Makefile.am	2024-09-27 13:52:35.269572700 +0900
@@ -2,16 +2,6 @@
 
 SUBDIRS = src 
 
-libghwpdocdir = ${prefix}/doc/libghwp
-libghwpdoc_DATA = \
-	README\
-	COPYING\
-	AUTHORS\
-	ChangeLog\
-	INSTALL\
-	NEWS
-
-
 EXTRA_DIST = $(libghwpdoc_DATA)
 
 
--- origsrc/libghwp-0.1.1/configure.ac	2012-12-24 00:33:19.000000000 +0900
+++ src/libghwp-0.1.1/configure.ac	2024-09-27 13:56:01.467201900 +0900
@@ -1,4 +1,4 @@
-AC_INIT([libghwp], [0.1], [https://github.com/cogniti/libghwp/issues], [libghwp])
+AC_INIT([libghwp], [0.1.1], [https://github.com/cogniti/libghwp/issues], [libghwp])
 
 AC_CONFIG_HEADERS([config.h])
 
--- origsrc/libghwp-0.1.1/src/Makefile.am	2012-12-24 00:33:19.000000000 +0900
+++ src/libghwp-0.1.1/src/Makefile.am	2024-09-27 13:48:39.262336200 +0900
@@ -22,7 +22,7 @@ libghwp_la_SOURCES = \
 	ghwp-parse.c \
 	gsf-input-stream.c
 
-libghwp_la_LDFLAGS = 
+libghwp_la_LDFLAGS = -no-undefined
 
 libghwp_la_LIBADD = $(LIBGHWP_LIBS)
 
