--- origsrc/tty-solitaire-1.3.1/src/gui.c	2021-09-04 01:03:20.000000000 +0900
+++ src/tty-solitaire-1.3.1/src/gui.c	2022-12-27 17:43:07.246953600 +0900
@@ -13,8 +13,8 @@ static const char *card_values[13] = {"A
                                       "8", "9", "10", "J", "Q", "K"};
 
 static void draw_value(struct card *card) {
-  mvwprintw(card->frame->window, 0, 0, card_values[card->value]);
-  mvwprintw(card->frame->window, 4, 7 - strlen(card_values[card->value]),
+  mvwprintw(card->frame->window, 0, 0, "%s", card_values[card->value]);
+  mvwprintw(card->frame->window, 4, 7 - strlen(card_values[card->value]), "%s",
             card_values[card->value]);
 }
 
@@ -42,9 +42,9 @@ static void draw_suit(struct card *card)
       break;
     }
   }
-  mvwprintw(card->frame->window, 0, strlen(card_values[card->value]),
+  mvwprintw(card->frame->window, 0, strlen(card_values[card->value]), "%s",
             card_suits[card->suit]);
-  mvwprintw(card->frame->window, 4, 6 - strlen(card_values[card->value]),
+  mvwprintw(card->frame->window, 4, 6 - strlen(card_values[card->value]), "%s",
             card_suits[card->suit]);
   if (card->suit % 2 == 0) {
     wattroff(card->frame->window, COLOR_PAIR(RED_ON_WHITE));
