--- origsrc/ironout-0.0.2/Makefile	2008-09-23 21:06:11.000000000 +0900
+++ src/ironout-0.0.2/Makefile	2025-01-01 12:01:18.057580300 +0900
@@ -2,8 +2,8 @@ CC = cc
 YACC = bison
 LEX = flex
 
-CFLAGS = -std=c89 -pedantic -Wall -O2 -g
-LDFLAGS =
+#CFLAGS = -std=c89 -pedantic -Wall -O2 -g
+#LDFLAGS =
 LFLAGS =
 YFLAGS = -d
 
--- origsrc/ironout-0.0.2/ironout.c	2008-09-23 21:06:11.000000000 +0900
+++ src/ironout-0.0.2/ironout.c	2025-01-01 12:02:09.576128600 +0900
@@ -127,7 +127,7 @@ static char *version =
 
 static void print_version(int argc, char **argv)
 {
-	printf(version);
+	printf("%s", version);
 }
 
 int main(int argc, char **argv)
--- origsrc/ironout-0.0.2/tests/Makefile	2008-09-23 21:06:11.000000000 +0900
+++ src/ironout-0.0.2/tests/Makefile	2025-01-01 12:03:27.071256300 +0900
@@ -1,9 +1,9 @@
-CFLAGS = -std=c89 -pedantic -Wall -O2 -I..
-LDFLAGS = -s
+#CFLAGS = -std=c89 -pedantic -Wall -O2 -I..
+#LDFLAGS = -s
 
 all: runtests test_hash
 .c.o:
-	$(CC) -c $(CFLAGS) $<
+	$(CC) -c -I.. $(CFLAGS) $<
 runtests: runtests.o ../strutils.o
 	$(CC) $(LDFLAGS) -o $@ $^
 test_hash: test_hash.o ../hash.o ../utils.o
