--- origsrc/libcoveb-1.0.4/configure.ac	2008-03-27 23:44:50.000000000 +0900
+++ src/libcoveb-1.0.4/configure.ac	2011-07-03 00:16:26.888943000 +0900
@@ -3,7 +3,7 @@ AC_INIT(libcoveb,1.0.4,cilibrar@cilibrar
 AC_CONFIG_SRCDIR(src/coveb-tree.c)
 AM_INIT_AUTOMAKE
 AC_CANONICAL_HOST
-AC_PROG_LIBTOOL
+LT_INIT
 
 AC_PATH_PROG(DOXYGEN, doxygen, "")
 if test x$DOXYGEN = x ; then
@@ -19,7 +19,7 @@ AM_MAINTAINER_MODE
 
 PRIVATELIBS="$LIBS"
 
-if (uname) | grep -i darwin 2>/dev/null 1>/dev/null ; then
+if (uname) | grep -i -e darwin -e cygwin 2>/dev/null 1>/dev/null ; then
   SONAMEOPTS=""
 else
   SONAMEOPTS="-Wl,-soname -Wl,libcoveb-1.0${shrext_cmds}.0"
--- origsrc/libcoveb-1.0.4/src/Makefile.am	2008-03-28 00:59:52.000000000 +0900
+++ src/libcoveb-1.0.4/src/Makefile.am	2011-07-03 00:26:35.600008600 +0900
@@ -5,7 +5,7 @@ EXTRA_DIST = $(GOBCS) $(GOBHS) $(GOBS) p
 
 GLOBCPPFLAGS= -I@abs_srcdir@ -I@abs_srcdir@/coveb
 
-EXELDFLAGS = -lcoveb
+EXELDFLAGS = libcoveb.la
 lib_LTLIBRARIES   = libcoveb.la
 
 BASEHEADERS=coveb/coveb-tree.h coveb/coveb.h
@@ -19,7 +19,7 @@ endif
 INTERNALHEADERS=bigmac.h
 libcoveb_la_SOURCES= coveb-tree.c coveb-map.c $(BASEHEADERS) $(INTERNALHEADERS)
 libcoveb_la_CPPFLAGS= @EXTCPPFLAGS@ @EXTCFLAGS@ $(GLOBCPPFLAGS)
-libcoveb_la_LDFLAGS= -version-info @libcoveb_version@ @SONAMEOPTS@ @LIBS@
+libcoveb_la_LDFLAGS= -version-info @libcoveb_version@ @SONAMEOPTS@ @LIBS@ -no-undefined
 
 benchcoveb_SOURCES = benchcoveb.c
 benchcoveb_LDADD = $(EXELDFLAGS)
