--- origsrc/sslsplit-0.5.0/GNUmakefile	2016-03-27 22:46:35.000000000 +0900
+++ src/sslsplit-0.5.0/GNUmakefile	2017-08-12 00:03:56.540999800 +0900
@@ -339,7 +339,7 @@ CFLAGS+=	$(PKG_CFLAGS) \
 		-D_FORTIFY_SOURCE=2 -fstack-protector-all
 CPPFLAGS+=	$(PKG_CPPFLAGS) $(CPPDEFS) $(FEATURES)
 TCPPFLAGS+=	$(TPKG_CPPFLAGS)
-LDFLAGS+=	$(PKG_LDFLAGS)
+LDFLAGS+=	$(PKG_LDFLAGS) -fstack-protector-all
 LIBS+=		$(PKG_LIBS)
 
 ifneq ($(shell uname),Darwin)
