--- origsrc/sslsplit-0.5.5/GNUmakefile	2019-08-30 20:09:00.000000000 +0900
+++ src/sslsplit-0.5.5/GNUmakefile	2021-05-11 14:56:45.228785000 +0900
@@ -63,7 +63,7 @@
 # or on platforms without usable libnet/libpcap.  Logging connection content to
 # PCAP files will remain fully functional (-X/-Y/-y options) as it does not
 # make use of libnet and libpcap.
-#FEATURES+=	-DWITHOUT_MIRROR
+FEATURES+=	-DWITHOUT_MIRROR
 
 
 ### OpenSSL tweaking
