--- origsrc/libgamerzilla-release-0.1.0/CMakeLists.txt	2021-07-04 10:34:35.000000000 +0900
+++ src/libgamerzilla-release-0.1.0/CMakeLists.txt	2021-07-07 17:58:49.891531900 +0900
@@ -32,7 +32,7 @@ set(INSTALL_INC_DIR "${CMAKE_INSTALL_PRE
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/gamerzilla.pc.cmakein
 	${CMAKE_CURRENT_BINARY_DIR}/gamerzilla.pc @ONLY)
 
-install(TARGETS ${PROJECT_NAME} DESTINATION "lib${LIB_SUFFIX}")
+install(TARGETS ${PROJECT_NAME} RUNTIME DESTINATION bin ARCHIVE DESTINATION "lib${LIB_SUFFIX}")
 
 # Install library headers
 file(GLOB HEADERS *.h)
--- origsrc/libgamerzilla-release-0.1.0/gamerzilla.pc.cmakein	2021-07-04 10:34:35.000000000 +0900
+++ src/libgamerzilla-release-0.1.0/gamerzilla.pc.cmakein	2021-07-07 17:57:57.322471400 +0900
@@ -5,6 +5,6 @@ includedir=@INSTALL_INC_DIR@
 
 Name: Gamerzilla
 Description: Record achievements and upload them
-Version: @VERSION@
+Version: @GAMERZILLA_VERSION_STRING@
 Libs: -L${libdir} -lgamerzilla
 Cflags: -I${includedir}/gamerzilla
