--- origsrc/libgamerzilla-release-0.0.6/CMakeLists.txt	2020-09-14 06:39:09.000000000 +0900
+++ src/libgamerzilla-release-0.0.6/CMakeLists.txt	2020-09-26 09:38:33.758438300 +0900
@@ -2,8 +2,8 @@ cmake_minimum_required (VERSION 2.8.11)
 project (gamerzilla)
 
 set(GAMERZILLA_VERSION_MAJOR 0)
-set(GAMERZILLA_VERSION_MINOR 1)
-set(GAMERZILLA_VERSION_PATCH 0)
+set(GAMERZILLA_VERSION_MINOR 0)
+set(GAMERZILLA_VERSION_PATCH 6)
 set(GAMERZILLA_VERSION_STRING ${GAMERZILLA_VERSION_MAJOR}.${GAMERZILLA_VERSION_MINOR}.${GAMERZILLA_VERSION_PATCH})
 
 add_library (gamerzilla SHARED gamerzilla.c)
@@ -32,7 +32,7 @@ set(INSTALL_INC_DIR "${CMAKE_INSTALL_PRE
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/gamerzilla.pc.cmakein
 	${CMAKE_CURRENT_BINARY_DIR}/gamerzilla.pc @ONLY)
 
-install(TARGETS ${PROJECT_NAME} DESTINATION "lib${LIB_SUFFIX}")
+install(TARGETS ${PROJECT_NAME} RUNTIME DESTINATION bin ARCHIVE DESTINATION "lib${LIB_SUFFIX}")
 
 # Install library headers
 file(GLOB HEADERS *.h)
--- origsrc/libgamerzilla-release-0.0.6/gamerzilla.pc.cmakein	2020-09-14 06:39:09.000000000 +0900
+++ src/libgamerzilla-release-0.0.6/gamerzilla.pc.cmakein	2020-09-26 09:37:56.550517400 +0900
@@ -5,6 +5,6 @@ includedir=@INSTALL_INC_DIR@
 
 Name: Gamerzilla
 Description: Record achievements and upload them
-Version: @VERSION@
+Version: @GAMERZILLA_VERSION_STRING@
 Libs: -L${libdir} -lgamerzilla
 Cflags: -I${includedir}/gamerzilla
