--- origsrc/libbitcoin-database-3.6.0/Makefile.am	2019-08-29 22:34:14.000000000 +0900
+++ src/libbitcoin-database-3.6.0/Makefile.am	2019-09-22 11:39:58.156775200 +0900
@@ -57,6 +57,7 @@ src_libbitcoin_database_la_SOURCES = \
     src/primitives/slab_manager.cpp \
     src/result/block_result.cpp \
     src/result/transaction_result.cpp
+src_libbitcoin_database_la_LDFLAGS = -no-undefined
 
 # local: test/libbitcoin-database-test
 #------------------------------------------------------------------------------
