--- origsrc/xdag-0.4.0/CMakeLists.txt	2021-02-16 01:16:35.000000000 +0900
+++ src/xdag-0.4.0/CMakeLists.txt	2024-09-23 23:11:07.370297800 +0900
@@ -33,6 +33,9 @@ endif()
 set(RANDOMX_LIBDIR "/usr/lib" "/usr/local/lib")
 set(RANDOMX_LIBRARIES "randomx")
 
+pkg_check_modules(CJSON libcjson libcjson_utils)
+pkg_check_modules(LINENOISE linenoise)
+
 file(GLOB_RECURSE DAGGER_SOURCES
   client/*.c
   dfslib/*.c
@@ -46,6 +49,13 @@ list(FILTER DAGGER_SOURCES EXCLUDE REGEX
 list(FILTER DAGGER_SOURCES EXCLUDE REGEX ".*dvstime.c$")
 list(FILTER DAGGER_SOURCES EXCLUDE REGEX ".*statsdaemon.c$")
 
+if (CJSON_FOUND)
+  list(FILTER DAGGER_SOURCES EXCLUDE REGEX "cJSON.*.c$")
+endif()
+if (LINENOISE_FOUND)
+  list(FILTER DAGGER_SOURCES EXCLUDE REGEX "linenoise.c$")
+endif()
+
 file(GLOB_RECURSE DAGGER_HEADERS
   client/*.h
   dfslib/*.h
@@ -53,6 +63,13 @@ file(GLOB_RECURSE DAGGER_HEADERS
   ldus/*.h
 )
 
+if (CJSON_FOUND)
+  list(FILTER DAGGER_HEADERS EXCLUDE REGEX "cJSON.*.h$")
+endif()
+if (LINENOISE_FOUND)
+  list(FILTER DAGGER_HEADERS EXCLUDE REGEX "linenoise.h$")
+endif()
+
 set(SHA256_LINUX_ASM_SOURCES
   client/algorithms/sha256-mb-x86_64.s
   client/x86_64cpuid.s
@@ -89,6 +106,9 @@ endif()
 include_directories(
   ${OPENSSL_INCLUDE_DIR}
   ${SECP256K1_INCLUDE_DIRS}
+  ${CJSON_INCLUDE_DIRS}
+  ${LINENOISE_INCLUDE_DIRS}
+  ${PROJECT_SOURCE_DIR}/dnet
 )
 
 if(APPLE)
@@ -120,6 +140,8 @@ target_link_libraries(
   -L${RANDOMX_LIBDIR} ${RANDOMX_LIBRARIES}
   m
   -lstdc++
+  ${CJSON_LIBRARIES}
+  ${LINENOISE_LIBRARIES}
 )
 
 add_executable(
--- origsrc/xdag-0.4.0/client/algorithms/sha256-mb-x86_64.s	2021-02-16 01:16:35.000000000 +0900
+++ src/xdag-0.4.0/client/algorithms/sha256-mb-x86_64.s	2024-09-23 22:54:48.546017300 +0900
@@ -3,7 +3,7 @@
 
 
 .globl	xsha256_multi_block
-.type	xsha256_multi_block,@function
+#.type	xsha256_multi_block,@function
 .align	32
 xsha256_multi_block:
 	movq	xOPENSSL_ia32cap_P+4(%rip),%rcx
@@ -2620,8 +2620,8 @@ xsha256_multi_block:
 	leaq	(%rax),%rsp
 .Lepilogue:
 	.byte	0xf3,0xc3
-.size	xsha256_multi_block,.-xsha256_multi_block
-.type	xsha256_multi_block_shaext,@function
+#.size	xsha256_multi_block,.-xsha256_multi_block
+#.type	xsha256_multi_block_shaext,@function
 .align	32
 xsha256_multi_block_shaext:
 _shaext_shortcut:
@@ -3106,8 +3106,8 @@ _shaext_shortcut:
 	leaq	(%rax),%rsp
 .Lepilogue_shaext:
 	.byte	0xf3,0xc3
-.size	xsha256_multi_block_shaext,.-xsha256_multi_block_shaext
-.type	xsha256_multi_block_avx,@function
+#.size	xsha256_multi_block_shaext,.-xsha256_multi_block_shaext
+#.type	xsha256_multi_block_avx,@function
 .align	32
 xsha256_multi_block_avx:
 _avx_shortcut:
@@ -5359,8 +5359,8 @@ _avx_shortcut:
 	leaq	(%rax),%rsp
 .Lepilogue_avx:
 	.byte	0xf3,0xc3
-.size	xsha256_multi_block_avx,.-xsha256_multi_block_avx
-.type	xsha256_multi_block_avx2,@function
+#.size	xsha256_multi_block_avx,.-xsha256_multi_block_avx
+#.type	xsha256_multi_block_avx2,@function
 .align	32
 xsha256_multi_block_avx2:
 _avx2_shortcut:
@@ -7748,7 +7748,7 @@ _avx2_shortcut:
 	leaq	(%rax),%rsp
 .Lepilogue_avx2:
 	.byte	0xf3,0xc3
-.size	xsha256_multi_block_avx2,.-xsha256_multi_block_avx2
+#.size	xsha256_multi_block_avx2,.-xsha256_multi_block_avx2
 .align	256
 K256:
 .long	1116352408,1116352408,1116352408,1116352408
--- origsrc/xdag-0.4.0/client/x86_64cpuid.s	2021-02-16 01:16:35.000000000 +0900
+++ src/xdag-0.4.0/client/x86_64cpuid.s	2024-09-23 17:57:41.461744300 +0900
@@ -1,11 +1,11 @@
 
-.hidden	xOPENSSL_ia32cap_P
+#.hidden	xOPENSSL_ia32cap_P
 .comm	xOPENSSL_ia32cap_P,16,4
 
 .text	
 
 .globl	xOPENSSL_ia32_cpuid
-.type	xOPENSSL_ia32_cpuid,@function
+#.type	xOPENSSL_ia32_cpuid,@function
 .align	16
 xOPENSSL_ia32_cpuid:
 	movq	%rbx,%r8
@@ -134,5 +134,5 @@ xOPENSSL_ia32_cpuid:
 	movq	%r8,%rbx
 	orq	%r9,%rax
 	.byte	0xf3,0xc3
-.size	xOPENSSL_ia32_cpuid,.-xOPENSSL_ia32_cpuid
+#.size	xOPENSSL_ia32_cpuid,.-xOPENSSL_ia32_cpuid
 
