--- origsrc/xdag-0.3.0/automake/Makefile.am	2018-11-16 05:17:18.000000000 +0900
+++ src/xdag-0.3.0/automake/Makefile.am	2020-01-28 23:34:45.774948400 +0900
@@ -2,7 +2,7 @@
 # created by cz.worker@gmail.com on 2018-03-29
 # modify by reymondtu@gmx.com on 2018-05-23
 
-SRCROOT             = ..
+SRCROOT             = $(top_srcdir)/..
 client              = $(SRCROOT)/client
 dnet	            = $(SRCROOT)/dnet
 dfstools            = $(SRCROOT)/dus/programs/dfstools/source
@@ -25,6 +25,7 @@ sources = \
 	$(client)/hash.c                                \
 	$(client)/init.c                                \
 	$(client)/main.c                                \
+	$(client)/math.c                                \
 	$(client)/memory.c                              \
 	$(client)/miner.c                               \
 	$(client)/netdb.c                               \
@@ -33,6 +34,7 @@ sources = \
 	$(client)/sha256.c                              \
 	$(client)/storage.c                             \
 	$(client)/sync.c                                \
+	$(client)/time.c                                \
 	$(client)/transport.c                           \
 	$(client)/wallet.c                              \
 	$(client)/commands.c                            \
@@ -47,10 +49,8 @@ sources = \
 	$(dar)/lib/crc_c.c                              \
 	$(utils)/log.c                                  \
 	$(utils)/utils.c                                \
-	$(utils)/linenoise.c                            \
 	$(utils)/dirname.c                              \
 	$(moving_statistics)/moving_average.c           \
-	$(jsonrpc)/cJSON.c                              \
 	$(jsonrpc)/cJSON_Utils.c                        \
 	$(jsonrpc)/rpc_procedures.c                     \
 	$(jsonrpc)/rpc_procedure.c                      \
@@ -58,8 +58,7 @@ sources = \
 	$(jsonrpc)/rpc_wrapper.c                        \
 	$(jsonrpc)/rpc_commands.c                       \
 	$(http)/url.c                                   \
-	$(http)/http.c                                  \
-	$(secp256k1)/src/secp256k1.c
+	$(http)/http.c
 
 
 headers = \
@@ -104,9 +103,7 @@ headers = \
 	$(ldusinc)/rbtree.h                             \
 	$(utils)/log.h                                  \
 	$(utils)/utils.h                                \
-	$(utils)/linenoise.h                            \
 	$(moving_statistics)/moving_average.h           \
-	$(jsonrpc)/cJSON.h                              \
 	$(jsonrpc)/cJSON_Utils.h                        \
 	$(jsonrpc)/rpc_procedures.h                     \
 	$(jsonrpc)/rpc_procedure.h                      \
@@ -143,6 +140,6 @@ _INCLUDES =
 	
 
 xdag_SOURCES = $(sources)
-#xdag_LDADD = -lpthread -lcrypto -lssl -lm
+xdag_LDADD = -lcjson -llinenoise -lsecp256k1
 xdag_CFLAGS = -std=gnu11 -O3 -g $(_DEFS) $(LIBCRYPTO_CFLAGS) $(_INCLUDES) -Wall -Wmissing-prototypes -Wno-unused-result
 
--- origsrc/xdag-0.3.0/automake/configure.ac	2018-11-16 05:17:18.000000000 +0900
+++ src/xdag-0.3.0/automake/configure.ac	2020-01-28 21:58:49.995325200 +0900
@@ -11,7 +11,11 @@ case "${host_os}" in
         HOST_SUFFIX=""
         HOST_OS="linux"
         ;;
-    cygwin*|mingw*)
+    cygwin*)
+        HOST_SUFFIX=""
+        HOST_OS="linux"
+        ;;
+    mingw*)
         HOST_SUFFIX="-win"
         HOST_OS="win"
         ;;
--- origsrc/xdag-0.3.0/client/math.c	2018-11-16 05:17:18.000000000 +0900
+++ src/xdag-0.3.0/client/math.c	2020-01-28 23:12:47.948802100 +0900
@@ -7,7 +7,10 @@
 #include "utils/log.h"
 
 // convert xdag_amount_t to long double
-inline long double amount2xdags(xdag_amount_t amount)
+#if !defined(__CYGWIN__)
+inline 
+#endif
+long double amount2xdags(xdag_amount_t amount)
 {
 	return xdag_amount2xdag(amount) + (long double)xdag_amount2cheato(amount) / 1000000000;
 }
--- origsrc/xdag-0.3.0/client/sha256-mb-x86_64.s	2018-11-16 05:17:18.000000000 +0900
+++ src/xdag-0.3.0/client/sha256-mb-x86_64.s	2020-01-28 23:27:29.659058400 +0900
@@ -3,7 +3,7 @@
 
 
 .globl	xsha256_multi_block
-.type	xsha256_multi_block,@function
+#.type	xsha256_multi_block,@function
 .align	32
 xsha256_multi_block:
 	movq	xOPENSSL_ia32cap_P+4(%rip),%rcx
@@ -2620,8 +2620,8 @@ xsha256_multi_block:
 	leaq	(%rax),%rsp
 .Lepilogue:
 	.byte	0xf3,0xc3
-.size	xsha256_multi_block,.-xsha256_multi_block
-.type	xsha256_multi_block_shaext,@function
+#.size	xsha256_multi_block,.-xsha256_multi_block
+#.type	xsha256_multi_block_shaext,@function
 .align	32
 xsha256_multi_block_shaext:
 _shaext_shortcut:
@@ -3106,8 +3106,8 @@ _shaext_shortcut:
 	leaq	(%rax),%rsp
 .Lepilogue_shaext:
 	.byte	0xf3,0xc3
-.size	xsha256_multi_block_shaext,.-xsha256_multi_block_shaext
-.type	xsha256_multi_block_avx,@function
+#.size	xsha256_multi_block_shaext,.-xsha256_multi_block_shaext
+#.type	xsha256_multi_block_avx,@function
 .align	32
 xsha256_multi_block_avx:
 _avx_shortcut:
@@ -5359,8 +5359,8 @@ _avx_shortcut:
 	leaq	(%rax),%rsp
 .Lepilogue_avx:
 	.byte	0xf3,0xc3
-.size	xsha256_multi_block_avx,.-xsha256_multi_block_avx
-.type	xsha256_multi_block_avx2,@function
+#.size	xsha256_multi_block_avx,.-xsha256_multi_block_avx
+#.type	xsha256_multi_block_avx2,@function
 .align	32
 xsha256_multi_block_avx2:
 _avx2_shortcut:
@@ -7748,7 +7748,7 @@ _avx2_shortcut:
 	leaq	(%rax),%rsp
 .Lepilogue_avx2:
 	.byte	0xf3,0xc3
-.size	xsha256_multi_block_avx2,.-xsha256_multi_block_avx2
+#.size	xsha256_multi_block_avx2,.-xsha256_multi_block_avx2
 .align	256
 K256:
 .long	1116352408,1116352408,1116352408,1116352408
--- origsrc/xdag-0.3.0/client/x86_64cpuid.s	2018-11-16 05:17:18.000000000 +0900
+++ src/xdag-0.3.0/client/x86_64cpuid.s	2020-01-28 23:25:53.603465200 +0900
@@ -1,11 +1,11 @@
 
-.hidden	xOPENSSL_ia32cap_P
+#.hidden	xOPENSSL_ia32cap_P
 .comm	xOPENSSL_ia32cap_P,16,4
 
 .text	
 
 .globl	xOPENSSL_ia32_cpuid
-.type	xOPENSSL_ia32_cpuid,@function
+#.type	xOPENSSL_ia32_cpuid,@function
 .align	16
 xOPENSSL_ia32_cpuid:
 	movq	%rbx,%r8
@@ -134,5 +134,5 @@ xOPENSSL_ia32_cpuid:
 	movq	%r8,%rbx
 	orq	%r9,%rax
 	.byte	0xf3,0xc3
-.size	xOPENSSL_ia32_cpuid,.-xOPENSSL_ia32_cpuid
+#.size	xOPENSSL_ia32_cpuid,.-xOPENSSL_ia32_cpuid
 
