--- origsrc/covered-0.7.10/src/lxt2_read.h	2008-06-27 23:02:03.000000000 +0900
+++ src/covered-0.7.10/src/lxt2_read.h	2017-08-10 01:37:55.295000000 +0900
@@ -47,7 +47,7 @@ typedef long off_t;
 #include <zlib.h>
 #endif
 
-#ifdef __GNUC__
+#if defined(__GNUC__) && !defined(__CYGWIN__)
 #define _LXT2_RD_INLINE inline
 #else
 #define _LXT2_RD_INLINE
--- origsrc/covered-0.7.10/src/report.c	2010-03-26 15:10:18.000000000 +0900
+++ src/covered-0.7.10/src/report.c	2017-08-10 01:35:27.653000200 +0900
@@ -1163,12 +1163,12 @@ void command_report(
           assert( interp );
 
           if( Tcl_Init( interp ) == TCL_ERROR ) {
-            printf( "ERROR: %s\n", interp->result );
+            printf( "ERROR: %s\n", Tcl_GetObjResult(interp) );
             Throw 0;
           }
 
           if( Tk_SafeInit( interp ) == TCL_ERROR ) {
-            printf( "ERROR: %s\n", interp->result );
+            printf( "ERROR: %s\n", Tcl_GetObjResult(interp) );
             Throw 0;
           }
 
