--- origsrc/octetos-core-2.5.0-alpha/src/Makefile.am	2020-08-31 07:55:02.000000000 +0900
+++ src/octetos-core-2.5.0-alpha/src/Makefile.am	2020-09-11 16:26:28.611857300 +0900
@@ -10,15 +10,14 @@ liboctetos_core_la_CFLAGS = -fPIC
 liboctetos_core_la_CPPFLAGS = -fPIC
 liboctetos_core_la_CXXFLAGS = -fPIC
 liboctetos_core_la_LDFLAGS = -shared -fPIC
+liboctetos_core_la_LIBADD = $(CUNIT_LIBS) $(LIBCONFIG_LIBS)
 if PLATFORM_WIN32
 liboctetos_core_la_LDFLAGS += -no-undefined
 endif
 if NATIVE_WIN32
 liboctetos_core_la_LDFLAGS += -export-dynamic
 
-liboctetos_core_la_LIBADD = $(CUNIT_LIBS) $(LIBCONFIG_LIBS)
 endif
-liboctetos_core_la_LIBDADD =
 liboctetos_core_la_SOURCES = \
 	common.c \
 	Artifact.cc \
@@ -32,12 +31,8 @@ liboctetos_core_la_SOURCES = \
 	semver-lexer.cc \
 	Buffer.cc
 
-libshmem_la_SOURCES = dummy.c
-libshmem_la_CFLAGS = -fPIC
-
 octetos_core_includedir = $(includedir)/octetos/core
 octetos_core_include_HEADERS = semver-lexer.h common.h Version.hh Message.hh Error.hh Object.hh Artifact.hh Licence.hh Buffer.hh
-include_HEADERS = semver-lexer.h common.h Version.hh Message.hh Error.hh Object.hh Artifact.hh Licence.hh Buffer.hh
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = octetos-core.pc
