--- origsrc/YODA-2.1.2/include/YODA/Config/BuildConfig.h	2025-10-24 03:58:57.000000000 +0900
+++ src/YODA-2.1.2/include/YODA/Config/BuildConfig.h	2026-02-09 14:19:04.313321500 +0900
@@ -48,6 +48,6 @@
 #define WITH_HIGHFIVE 1
 
 /* Define to 1 if running on OSX. */
-#define WITH_OSX 1
+/* #undef WITH_OSX */
 
 #endif
--- origsrc/YODA-2.1.2/include/YODA/Makefile.am	2025-10-24 03:57:47.000000000 +0900
+++ src/YODA-2.1.2/include/YODA/Makefile.am	2026-02-09 14:45:36.029443500 +0900
@@ -60,6 +60,6 @@ if ENABLE_H5
    pkginclude_HEADERS += ReaderH5.h WriterH5.h
    nobase_pkginclude_HEADERS += Utils/H5Utils.h
 if !WITH_HIGHFIVE_CPP
-   nobase_pkginclude_HEADERS += highfive/*.hpp highfive/bits/*.hpp
+   nobase_pkginclude_HEADERS += $(srcdir)/highfive/*.hpp $(srcdir)/highfive/bits/*.hpp
 endif
 endif
--- origsrc/YODA-2.1.2/include/YODA/Utils/BinEstimators.h	2025-07-04 19:25:14.000000000 +0900
+++ src/YODA-2.1.2/include/YODA/Utils/BinEstimators.h	2026-02-09 14:24:24.996944500 +0900
@@ -5,6 +5,10 @@
 #include "YODA/Utils/MathUtils.h"
 #include <cstdlib>
 
+#if defined(_N)
+#undef _N
+#endif
+
 namespace YODA {
 
 
--- origsrc/YODA-2.1.2/src/Makefile.am	2025-07-04 19:45:38.000000000 +0900
+++ src/YODA-2.1.2/src/Makefile.am	2026-02-09 14:41:47.542068400 +0900
@@ -18,7 +18,7 @@ libYODA_la_SOURCES = \
     WriterFLAT.cc \
     Counter.cc
 
-libYODA_la_LDFLAGS = -avoid-version
+libYODA_la_LDFLAGS = -no-undefined
 libYODA_la_LIBADD = $(builddir)/binreloc/libyoda-binreloc.la
 if !WITH_YAML_CPP
 libYODA_la_LIBADD += $(builddir)/yamlcpp/libyoda-yaml-cpp.la
