--- origsrc/more-itertools-10.1.0/pyproject.toml	2023-07-24 23:04:08.762728000 +0900
+++ src/more-itertools-10.1.0/pyproject.toml	2023-09-19 13:10:55.587442300 +0900
@@ -7,6 +7,7 @@ name = "more-itertools"
 authors = [{name = "Erik Rose", email = "erikrose@grinchcentral.com"}]
 readme = "README.rst"
 requires-python = ">=3.8"
+description = "More routines for operating on iterables, beyond itertools"
 license = {file = "LICENSE"}
 keywords = [
     "itertools",
@@ -33,7 +34,7 @@ classifiers = [
     "Programming Language :: Python :: Implementation :: PyPy",
     "Topic :: Software Development :: Libraries",
 ]
-dynamic = ["version", "description"]
+dynamic = ["version"]
 
 [project.urls]
 Homepage = "https://github.com/more-itertools/more-itertools"
