--- origsrc/gtick-0.5.4/testsuite/Makefile.am	2014-06-22 23:04:13.000000000 +0900
+++ src/gtick-0.5.4/testsuite/Makefile.am	2016-12-30 21:35:14.344370000 +0900
@@ -62,8 +62,8 @@ localedir = $(datadir)/locale
 DEFS = -DLOCALEDIR=\"$(localedir)\" @DEFS@
 LIBS = @LIBINTL@ @LIBS@
 
-AM_CPPFLAGS = -I../src -I../intl -I$(top_srcdir)/intl @DEPS_CFLAGS@ @SNDFILE_CFLAGS@ @CHECK_CFLAGS@ @DMALLOC_CFLAGS@
-AM_CFLAGS = -DVERSION='"@VERSION@"' -DPACKAGE='"@PACKAGE@"' -DUSE_DMALLOC
+AM_CPPFLAGS = -I$(top_srcdir)/src -I../src -I../intl -I$(top_srcdir)/intl @DEPS_CFLAGS@ @SNDFILE_CFLAGS@ @CHECK_CFLAGS@ @DMALLOC_CFLAGS@
+AM_CFLAGS = -DVERSION='"@VERSION@"' -DPACKAGE='"@PACKAGE@"'
 
 AM_YFLAGS = -d
 AM_LFLAGS=-olex.yy.c
