--- origsrc/mbuffer-20161115/Makefile.in	2017-01-19 00:01:45.683598800 +0900
+++ src/mbuffer-20161115/Makefile.in	2017-01-19 00:04:43.727931600 +0900
@@ -20,7 +20,7 @@ TARGET		= mbuffer$(EXE)
 SOURCES		= log.c network.c mbuffer.c
 OBJECTS		= $(SOURCES:.c=.o)
 
-TESTTREE	= /opt/csw/gcc4
+TESTTREE	= /usr/share/cygport
 
 .PHONY: clean all distclean install check testcleanup
 
@@ -102,4 +102,3 @@ test3: test.md5
 	sync
 	diff $@.md5 test.md5
 	touch $@
-
