--- origsrc/mbuffer-20170515/Makefile.in	2017-08-06 21:19:14.025005900 +0900
+++ src/mbuffer-20170515/Makefile.in	2017-08-06 21:27:53.530000000 +0900
@@ -13,6 +13,7 @@ exec_prefix     = @exec_prefix@
 datarootdir	= @datarootdir@
 bindir          = @bindir@
 mandir		= @mandir@/man1
+sysconfdir = @sysconfdir@
 
 RM		= @RM@
 INSTALL		= @INSTALL@
@@ -21,7 +22,7 @@ TARGET		= mbuffer$(EXE)
 SOURCES		= log.c network.c mbuffer.c
 OBJECTS		= $(SOURCES:.c=.o)
 
-TESTTREE	= /opt/csw/gcc4
+TESTTREE	= /usr/share/cygport
 
 .PHONY: clean all distclean install check testcleanup
 
@@ -39,13 +40,13 @@ distclean: clean
 	-$(RM) $(TARGET) config.h config.log \
 	config.status Makefile mbuffer.1 core
 
-$(DESTDIR)$(prefix)/etc:
+$(DESTDIR)$(sysconfdir):
 	-$(INSTALL) -d -m 755 $@
 
-$(DESTDIR)$(prefix)/etc/mbuffer.rc:
-	$(INSTALL) -m 644 mbuffer.rc $(DESTDIR)$(prefix)/etc/
+$(DESTDIR)$(sysconfdir)/mbuffer.rc:
+	$(INSTALL) -m 644 mbuffer.rc $(DESTDIR)$(sysconfdir)/
 
-install: $(TARGET) $(DESTDIR)$(prefix)/etc $(DESTDIR)$(prefix)/etc/mbuffer.rc
+install: $(TARGET) $(DESTDIR)$(sysconfdir) $(DESTDIR)$(sysconfdir)/mbuffer.rc
 	-$(INSTALL) -d -m 755 $(DESTDIR)$(bindir)
 	$(INSTALL) -m 755 $(TARGET) $(DESTDIR)$(bindir)/
 	-$(INSTALL) -d -m 755 $(DESTDIR)$(mandir)
@@ -109,4 +110,3 @@ test3: test.md5
 	sync
 	diff $@.md5 test.md5
 	touch $@
-
