--- origsrc/vmdktool-1.4/Makefile	2014-08-11 18:44:53.000000000 +0900
+++ src/vmdktool-1.4/Makefile	2014-08-23 15:44:51.071356400 +0900
@@ -1,12 +1,12 @@
 PREFIX?=	/usr/local
 LDLIBS=		-lz
-CFLAGS+=	-Wsystem-headers -Wno-format-y2k -W -Werror \
+CFLAGS+=	-Wsystem-headers -Wno-format-y2k -W \
 		-Wno-unused-parameter -Wstrict-prototypes \
 		-Wmissing-prototypes -Wpointer-arith -Wreturn-type \
 		-Wcast-qual -Wwrite-strings -Wswitch -Wshadow -Wcast-align \
 		-Wunused-parameter -Wchar-subscripts -Winline \
 		-Wnested-externs -Wunused
-CFLAGS+=	-g -O -pipe
+CFLAGS+=	-g -O3 -pipe
 OBJ=		vmdktool.o expand_number.o
 
 all:	vmdktool vmdktool.8.gz
