--- origsrc/libsc-2.8.5/Makefile.am	2023-03-30 15:24:48.000000000 +0900
+++ src/libsc-2.8.5/Makefile.am	2023-08-15 08:21:35.984624400 +0900
@@ -33,17 +33,8 @@ config/Makefile.sc.mk : config/Makefile.
         sed -e 's,{\(.*prefix\)},{sc_\1},g' \
             -e 's,^\(.*prefix *=\),sc_\1,g' > $@
 
-# install libsc m4 macros in a dedicated directory
-scaclocaldir = $(datadir)/aclocal
-dist_scaclocal_DATA = \
-        config/sc_builtin.m4 config/sc_memalign.m4 config/sc_include.m4 \
-        config/ax_prefix_config_h.m4 config/ax_split_version.m4 \
-        config/sc_package.m4 config/sc_mpi.m4 \
-        config/sc_pthread.m4 config/sc_openmp.m4 config/sc_v4l2.m4 \
-        config/sc_qsort.m4
-
 # install example .ini files in a dedicated directory
-scinidir = $(datadir)/ini
+scinidir = $(datadir)/libsc/ini
 dist_scini_DATA =
 
 # handle toplevel directory
@@ -62,8 +53,6 @@ LOG_COMPILER = @SC_MPIRUN@ @SC_MPI_TEST_
 
 # non-recursive build
 include src/Makefile.am
-include iniparser/Makefile.am
-include libb64/Makefile.am
 include test/Makefile.am
 ## include example/bspline/Makefile.am
 ## include example/cuda/Makefile.am
@@ -112,10 +101,10 @@ dist-hook:
 
 install-data-hook:
 	cd $(DESTDIR)$(pkgconfigdir) && \
-              mv -f sc_autotools.pc "libsc-$(VERSION).pc"
+              mv -f sc_autotools.pc libsc.pc
 
 uninstall-hook:
-	cd $(DESTDIR)$(pkgconfigdir) && rm -f "libsc-$(VERSION).pc"
+	cd $(DESTDIR)$(pkgconfigdir) && rm -f libsc.pc
 
 ## From the GNU automake documentation "What gets Cleaned"
 ## -------------------------------------------------------
--- origsrc/libsc-2.8.5/configure.ac	2023-03-30 15:24:48.000000000 +0900
+++ src/libsc-2.8.5/configure.ac	2023-08-15 08:22:07.751515600 +0900
@@ -98,6 +98,9 @@ echo "o---------------------------------
 
 SC_CHECK_LIBRARIES([SC])
 
+PKG_CHECK_MODULES([INIPARSER], [iniparser])
+PKG_CHECK_MODULES([LIBB64], [libb64])
+
 # Print summary.
 
 m4_ifset([SC_CHECK_MPI_F77], [
--- origsrc/libsc-2.8.5/src/Makefile.am	2023-03-30 15:24:48.000000000 +0900
+++ src/libsc-2.8.5/src/Makefile.am	2023-08-15 08:23:05.061230100 +0900
@@ -43,10 +43,11 @@ lib_LTLIBRARIES += src/libsc.la
 src_libsc_la_SOURCES = \
         $(libsc_internal_headers) \
         $(libsc_compiled_sources)
-src_libsc_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBSC_CPPFLAGS)
+src_libsc_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBSC_CPPFLAGS) $(INIPARSER_CFLAGS) $(LIBB64_CFLAGS)
 ## This is the official API versioning scheme of libtool.  Please see:
 ## Read https://www.gnu.org/software/libtool/manual/libtool.html#Versioning
-src_libsc_la_LDFLAGS = -version-info 2:0:0
+src_libsc_la_LDFLAGS = -no-undefined -version-info 2:0:0
+src_libsc_la_LIBADD = $(INIPARSER_LIBS) $(LIBB64_LIBS)
 LDADD += src/libsc.la
 
 nodist_include_HEADERS += $(libsc_generated_headers)
--- origsrc/libsc-2.8.5/src/sc_io.c	2023-03-30 15:24:48.000000000 +0900
+++ src/libsc-2.8.5/src/sc_io.c	2023-08-15 08:38:16.207517400 +0900
@@ -23,7 +23,8 @@
 
 #include <sc_io.h>
 #include <sc_puff.h>
-#include <libb64.h>
+#include <b64/cdecode.h>
+#include <b64/cencode.h>
 #ifdef SC_HAVE_ZLIB
 #include <zlib.h>
 #endif
