--- origsrc/libsc-2.8.7/Makefile.am	2025-03-19 04:55:51.000000000 +0900
+++ src/libsc-2.8.7/Makefile.am	2026-01-20 16:15:33.767884100 +0900
@@ -33,17 +33,8 @@ config/Makefile.sc.mk : config/Makefile.
         sed -e 's,{\(.*prefix\)},{sc_\1},g' \
             -e 's,^\(.*prefix *=\),sc_\1,g' > $@
 
-# install libsc m4 macros in a dedicated directory
-scaclocaldir = $(datadir)/aclocal
-dist_scaclocal_DATA = \
-        config/sc_builtin.m4 config/sc_memalign.m4 config/sc_include.m4 \
-        config/ax_prefix_config_h.m4 config/ax_split_version.m4 \
-        config/sc_package.m4 config/sc_mpi.m4 \
-        config/sc_pthread.m4 config/sc_openmp.m4 config/sc_v4l2.m4 \
-        config/sc_qsort.m4
-
 # install example .ini files in a dedicated directory
-scinidir = $(datadir)/ini
+scinidir = $(datadir)/libsc/ini
 dist_scini_DATA =
 
 # handle toplevel directory
@@ -67,8 +58,6 @@ LOG_COMPILER = @SC_MPIRUN@ @SC_MPI_TEST_
 
 # non-recursive build
 include src/Makefile.am
-include iniparser/Makefile.am
-include libb64/Makefile.am
 include test/Makefile.am
 ## include example/bspline/Makefile.am
 ## include example/cuda/Makefile.am
--- origsrc/libsc-2.8.7/configure.ac	2025-03-19 04:55:51.000000000 +0900
+++ src/libsc-2.8.7/configure.ac	2026-01-20 16:15:33.774203900 +0900
@@ -92,6 +92,9 @@ echo "o---------------------------------
 
 SC_CHECK_LIBRARIES([SC])
 
+PKG_CHECK_MODULES([INIPARSER], [iniparser])
+PKG_CHECK_MODULES([LIBB64], [libb64])
+
 # Print summary.
 echo "o----------------------------------"
 echo "| Results for libsc are"
--- origsrc/libsc-2.8.7/src/Makefile.am	2025-03-19 04:55:51.000000000 +0900
+++ src/libsc-2.8.7/src/Makefile.am	2026-01-20 16:15:33.776214300 +0900
@@ -45,10 +45,11 @@ lib_LTLIBRARIES += src/libsc.la
 src_libsc_la_SOURCES = \
         $(libsc_internal_headers) \
         $(libsc_compiled_sources)
-src_libsc_la_CPPFLAGS = $(AM_CPPFLAGS) $(SC_CPPFLAGS)
+src_libsc_la_CPPFLAGS = $(AM_CPPFLAGS) $(SC_CPPFLAGS) $(INIPARSER_CFLAGS) $(LIBB64_CFLAGS)
 ## This is the official API versioning scheme of libtool.  Please see:
 ## Read https://www.gnu.org/software/libtool/manual/libtool.html#Versioning
-src_libsc_la_LDFLAGS = -version-info $(SC_SOVERSION)
+src_libsc_la_LDFLAGS = -no-undefined -version-info $(SC_SOVERSION)
+src_libsc_la_LIBADD = $(INIPARSER_LIBS) $(LIBB64_LIBS)
 LDADD += src/libsc.la
 
 nodist_include_HEADERS += $(libsc_generated_headers)
--- origsrc/libsc-2.8.7/src/sc_io.c	2025-03-19 04:55:51.000000000 +0900
+++ src/libsc-2.8.7/src/sc_io.c	2026-01-20 16:15:33.776214300 +0900
@@ -23,7 +23,8 @@
 
 #include <sc_io.h>
 #include <sc_puff.h>
-#include <libb64.h>
+#include <b64/cdecode.h>
+#include <b64/cencode.h>
 #ifdef SC_HAVE_ZLIB
 #include <zlib.h>
 #else
