--- origsrc/vcakec/src/Makefile	2009-03-16 02:13:58.000000000 +0900
+++ src/vcakec/src/Makefile	2025-12-23 15:26:31.213514900 +0900
@@ -1,7 +1,7 @@
 #DEBUG =		-g -DDEBUG
 OPT =		-O3 -funroll-loops
 INCLUDE =	-I/usr/include/malloc
-CFLAGS =	-std=gnu99 -Werror -Wall \
+CFLAGS ?=	-std=gnu99 -Wall \
 		${INCLUDE} ${DEBUG} ${OPT}
 CC =		gcc
 OBJ =		main.o readset.o nts.o opt.o seqhash.o shindex.o lookup3.o \
