--- origsrc/libimobiledevice-glue-1.0.0/configure.ac	2023-04-22 00:08:02.000000000 +0900
+++ src/libimobiledevice-glue-1.0.0/configure.ac	2023-08-08 20:01:16.942258600 +0900
@@ -66,7 +66,7 @@ fi
 # Check for operating system
 AC_MSG_CHECKING([for platform-specific build settings])
 case ${host_os} in
-  *mingw32*|*cygwin*)
+  *mingw32*)
     AC_MSG_RESULT([${host_os}])
     win32=true
     AC_DEFINE(WINVER, 0x0501, [minimum Windows version])
