--- origsrc/ccif-0.3.20111209/Makefile.am	2011-12-06 00:25:05.000000000 +0900
+++ src/ccif-0.3.20111209/Makefile.am	2012-05-02 09:29:04.325861900 +0900
@@ -27,6 +27,7 @@ ccif/star_print.c  ccif/star_graph.c   c
 ccif/list.c        ccif/rb.c \
 ccif/cifdic_to_symtab.inc            ccif/cifdic_to_symtab_main.inc \
 ccif/cifdic_to_symtab_ddl.inc        ccif/lexactions.inc
+ccif_libccif_la_LDFLAGS = -no-undefined
 
 # ast.c is #included from star_map.c; the rest from star_walk.c
 noinst_HEADERS = ccif/ast.c ccif/sorcerer.c ccif/astlib.c ccif/s_errsupport.c
