--- origsrc/sollya-7.0/Makefile.am	2018-08-27 17:32:07.000000000 +0900
+++ src/sollya-7.0/Makefile.am	2019-10-30 09:19:17.744771500 +0900
@@ -186,3 +186,6 @@ sollya.1: version.h sollya.h2m
 			-o $@ \
 			$(top_builddir)/sollya
 endif
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/sollya-7.0/configure.ac	2018-09-20 20:11:30.000000000 +0900
+++ src/sollya-7.0/configure.ac	2019-10-30 09:19:17.588519300 +0900
@@ -852,5 +852,5 @@ AM_CPPFLAGS="-I\${abs_top_srcdir} $AM_CP
 AC_SUBST(AM_CPPFLAGS)
 AC_SUBST(AM_CFLAGS)
 AC_SUBST(AM_LDFLAGS)
-AC_CONFIG_FILES([Makefile doc/Makefile tests-tool/Makefile tests-lib/Makefile])
+AC_CONFIG_FILES([sollya.pc Makefile doc/Makefile tests-tool/Makefile tests-lib/Makefile])
 AC_OUTPUT
--- origsrc/sollya-7.0/sollya.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/sollya-7.0/sollya.pc.in	2019-10-30 09:19:16.416652500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: sollya
+Description: Environment for the development of numerical codes
+Version: @VERSION@
+Libs: -L${libdir} -lsollya
+Cflags: -I${includedir}
