--- origsrc/libgig-4.5.0/src/Akai.h	2019-02-21 03:51:24.000000000 +0900
+++ src/libgig-4.5.0/src/Akai.h	2025-06-24 13:04:43.567702300 +0900
@@ -30,7 +30,7 @@
 # include <config.h>
 #endif
 
-#if !defined(_CARBON_) && !defined(__APPLE__) && !defined(WIN32)
+#if !defined(_CARBON_) && !defined(__APPLE__) && !defined(WIN32) && !defined(__CYGWIN__)
 # define LINUX 1
 #endif
 
@@ -46,7 +46,7 @@
 #ifndef _MSC_VER
 # include <sys/fcntl.h>
 #endif
-#if defined(_CARBON_) || defined(__APPLE__) || LINUX
+#if defined(_CARBON_) || defined(__APPLE__) || LINUX || defined(__CYGWIN__)
 # include <sys/ioctl.h>
 # include <unistd.h>
 #elif defined(WIN32)
@@ -147,7 +147,7 @@ public:
 protected:
 #ifdef WIN32
   HANDLE mFile;
-#elif defined _CARBON_ || defined(__APPLE__) || LINUX
+#elif defined _CARBON_ || defined(__APPLE__) || LINUX || defined(__CYGWIN__)
   int mFile;
 #endif
   bool mRegularFile;
--- origsrc/libgig-4.5.0/src/testcases/Makefile.am	2025-02-28 03:15:15.000000000 +0900
+++ src/libgig-4.5.0/src/testcases/Makefile.am	2025-06-24 13:11:31.866911800 +0900
@@ -26,7 +26,7 @@ serializationtests_LDADD = $(top_builddi
 
 EXTRA_DIST = CMakeLists.txt
 
-check: gigtests helpertests rifftests serializationtests
+check: gigtests$(EXEEXT) helpertests$(EXEEXT) rifftests$(EXEEXT) serializationtests$(EXEEXT)
 	@echo
 	@echo '********** RUNNING HELPER TESTS **********'
 	@echo
