--- origsrc/libgig-4.1.0/src/Akai.h	2014-05-23 00:54:02.000000000 +0900
+++ src/libgig-4.1.0/src/Akai.h	2017-12-22 17:26:04.246406500 +0900
@@ -30,7 +30,7 @@
 # include <config.h>
 #endif
 
-#if !defined(_CARBON_) && !defined(__APPLE__) && !defined(WIN32)
+#if !defined(_CARBON_) && !defined(__APPLE__) && !defined(WIN32) &!defined(__CYGWIN__)
 # define LINUX 1
 #endif
 
@@ -146,7 +146,7 @@ public:
 protected:
 #ifdef WIN32
   HANDLE mFile;
-#elif defined _CARBON_ || defined(__APPLE__) || LINUX
+#elif defined _CARBON_ || defined(__APPLE__) || LINUX || defined(__CYGWIN__)
   int mFile;
 #endif
   bool mRegularFile;
--- origsrc/libgig-4.1.0/src/Makefile.am	2017-05-22 02:04:34.000000000 +0900
+++ src/libgig-4.1.0/src/Makefile.am	2017-12-22 17:32:18.259000000 +0900
@@ -1,7 +1,7 @@
 # to prevent compile errors on some systems
 AM_CXXFLAGS = -pedantic -Wreturn-type
 
-pkglib_LTLIBRARIES = libgig.la libakai.la
+lib_LTLIBRARIES = libgig.la libakai.la
 
 libgigincludedir = $(includedir)/libgig
 libgiginclude_HEADERS = RIFF.h DLS.h SF.h gig.h Korg.h Serialization.h
