--- origsrc/TinySVM-0.09/configure.in	2025-06-11 22:47:51.806532300 +0900
+++ src/TinySVM-0.09/configure.in	2025-06-11 22:47:53.265797800 +0900
@@ -1,6 +1,6 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(src/svm_learn.cpp)
-AM_INIT_AUTOMAKE(TinySVM,0.09)
+AC_INIT([tinysvm], [0.09])
+AM_INIT_AUTOMAKE
 
 # Before making a release, the LTVERSION string should be modified.
 # The string is of the form C:R:A.
@@ -69,7 +69,7 @@ else
         fi
 fi
 
-AC_DEFUN(ADD_CC_FLAG, [
+AC_DEFUN([ADD_CC_FLAG], [
   AC_MSG_CHECKING(whether ${CC-cc} accepts $1)
   AC_LANG_SAVE
   AC_LANG_C
@@ -81,7 +81,7 @@ AC_DEFUN(ADD_CC_FLAG, [
   AC_LANG_RESTORE
 ])
 
-AC_DEFUN(ADD_CXX_FLAG, [
+AC_DEFUN([ADD_CXX_FLAG], [
   AC_MSG_CHECKING(whether ${CXX-c++} accepts $1)
   AC_LANG_SAVE
   AC_LANG_CPLUSPLUS
@@ -327,7 +327,6 @@ AC_OUTPUT([Makefile
            tests/Makefile 	   
 	   man/Makefile 
 	   doc/Makefile 
-	   java/Makefile
 	   python/Setup
 	   perl/Makefile.PL
 	   ruby/extconf.rb
