--- origsrc/TinySVM-0.09/Makefile.am	2001-12-07 19:54:21.000000000 +0900
+++ src/TinySVM-0.09/Makefile.am	2018-08-12 15:40:36.456913000 +0900
@@ -61,3 +61,6 @@ dist-all-package:
 	make clean
 	make win
 	make win-dist
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/TinySVM-0.09/TinySVM.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/TinySVM-0.09/TinySVM.pc.in	2018-08-12 15:41:16.211249800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: TinySVM
+Description: Support Vector Machines
+Version: @VERSION@
+Libs: -L${libdir} -ltinysvm
+Cflags: -I${includedir}
--- origsrc/TinySVM-0.09/configure.in	2002-08-20 14:58:13.000000000 +0900
+++ src/TinySVM-0.09/configure.in	2018-08-12 15:41:26.760087800 +0900
@@ -1,6 +1,6 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(src/svm_learn.cpp)
-AM_INIT_AUTOMAKE(TinySVM,0.09)
+AC_INIT([TinySVM], [0.09])
+AM_INIT_AUTOMAKE
 
 # Before making a release, the LTVERSION string should be modified.
 # The string is of the form C:R:A.
@@ -54,7 +54,7 @@ else
         fi
 fi
 
-AC_DEFUN(ADD_CC_FLAG, [
+AC_DEFUN([ADD_CC_FLAG], [
   AC_MSG_CHECKING(whether ${CC-cc} accepts $1)
   AC_LANG_SAVE
   AC_LANG_C
@@ -66,7 +66,7 @@ AC_DEFUN(ADD_CC_FLAG, [
   AC_LANG_RESTORE
 ])
 
-AC_DEFUN(ADD_CXX_FLAG, [
+AC_DEFUN([ADD_CXX_FLAG], [
   AC_MSG_CHECKING(whether ${CXX-c++} accepts $1)
   AC_LANG_SAVE
   AC_LANG_CPLUSPLUS
@@ -301,14 +301,13 @@ AC_SUBST(JAVAINC)
 #------------------------------------------------------
 
 AM_CONFIG_HEADER(config.h)
-AC_OUTPUT([Makefile
+AC_OUTPUT([TinySVM.pc Makefile
            src/Makefile 
 	   src/Makefile.bcc32
 	   src/Makefile.msvc
            tests/Makefile 	   
 	   man/Makefile 
 	   doc/Makefile 
-	   java/Makefile
 	   python/Setup
 	   perl/Makefile.PL
 	   ruby/extconf.rb
