--- origsrc/hesiod-3.1.0/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/hesiod-3.1.0/Makefile.am	2011-09-21 00:40:51.669476900 +0900
@@ -0,0 +1,14 @@
+
+AM_CPPFLAGS = -DSYSCONFDIR=\"@sysconfdir@\"
+
+lib_LTLIBRARIES = libhesiod.la
+libhesiod_la_SOURCES = hesiod.c hesmailhost.c hespwnam.c hesservbyname.c hescompat.c
+libhesiod_la_LDFLAGS = -no-undefined -version-info 0:0:0
+
+noinst_PROGRAMS = hestest
+hestest_SOURCES = hestest.c
+hestest_LDADD = libhesiod.la
+
+man_MANS = hesiod.3 hesiod.conf.5 hesiod_end.3 hesiod_free_list.3 hesiod_free_passwd.3 hesiod_free_postoffice.3 hesiod_free_servent.3 hesiod_free_string.3 hesiod_getmailhost.3 hesiod_getpwnam.3 hesiod_getpwuid.3 hesiod_getservbyname.3 hesiod_init.3 hesiod_parse_result.3 hesiod_resolve.3 hesiod_to_bind.3
+
+include_HEADERS = hesiod.h
--- origsrc/hesiod-3.1.0/configure.in	2011-09-21 00:03:45.342939200 +0900
+++ src/hesiod-3.1.0/configure.in	2011-09-21 00:26:11.982440100 +0900
@@ -1,9 +1,9 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(hesiod.c)
+AC_INIT(hesiod, 3.1.0)
+AM_INIT_AUTOMAKE
 
 AC_PROG_CC
 AC_PROG_INSTALL
-AC_PROG_RANLIB
 AC_PROG_LIBTOOL
 
 AC_EGREP_HEADER(pw_quota, pwd.h, AC_DEFINE(HAVE_PW_QUOTA))
