--- origsrc/libusbmuxd-2.1.1/configure.ac	2025-06-04 22:26:10.000000000 +0900
+++ src/libusbmuxd-2.1.1/configure.ac	2025-10-07 13:19:39.229909600 +0900
@@ -74,7 +74,7 @@ AC_CHECK_FUNCS([strcasecmp strdup strerr
 # Check for operating system
 AC_MSG_CHECKING([for platform-specific build settings])
 case ${host_os} in
-  *mingw32*|*cygwin*)
+  *mingw32*)
     AC_MSG_RESULT([${host_os}])
     win32=true
     AC_DEFINE(WINVER, 0x0501, [minimum Windows version])
