--- origsrc/TECkit_2_5_1/configure.ac	2008-04-07 23:21:12.000000000 +0900
+++ src/TECkit_2_5_1/configure.ac	2011-03-21 17:06:12.516561000 +0900
@@ -13,11 +13,7 @@ AC_PROG_CXX
 AC_PROG_CC
 AM_PROG_CC_C_O
 AC_PROG_INSTALL
-AC_PROG_LIBTOOL
-
-if test "${target_os}" = "mingw32"; then
-AC_PROG_RC
-fi
+LT_INIT(win32-dll)
 
 AC_ARG_WITH(old-lib-names,
 [  --with-old-lib-names    Use old library names (for Windows only)],
