--- origsrc/libMeshb-7.80/CMakeLists.txt	2024-02-27 18:27:56.000000000 +0900
+++ src/libMeshb-7.80/CMakeLists.txt	2024-09-02 01:28:10.645767900 +0900
@@ -60,7 +60,6 @@ add_subdirectory (sources)
 add_subdirectory (examples)
 add_subdirectory (utilities)
 
-install (FILES LICENSE.txt copyright.txt DESTINATION share/libMeshb)
 install (DIRECTORY sample_meshes DESTINATION share/libMeshb)
 
 
--- origsrc/libMeshb-7.80/examples/CMakeLists.txt	2024-02-27 18:27:56.000000000 +0900
+++ src/libMeshb-7.80/examples/CMakeLists.txt	2024-09-02 01:32:22.423003700 +0900
@@ -4,31 +4,31 @@
 ##################
 
 add_executable(test_libmeshb test_libmeshb.c)
-target_link_libraries(test_libmeshb Meshb.7 ${AIO_LIBRARIES})
+target_link_libraries(test_libmeshb Meshb ${AIO_LIBRARIES})
 install (TARGETS test_libmeshb DESTINATION share/libMeshb/examples COMPONENT examples)
 
 add_executable(test_libmeshb_block test_libmeshb_block.c)
-target_link_libraries(test_libmeshb_block Meshb.7 ${AIO_LIBRARIES})
+target_link_libraries(test_libmeshb_block Meshb ${AIO_LIBRARIES})
 install (TARGETS test_libmeshb_block DESTINATION share/libMeshb/examples COMPONENT examples)
 
 add_executable(test_libmeshb_sol test_libmeshb_sol.c)
-target_link_libraries(test_libmeshb_sol Meshb.7 ${AIO_LIBRARIES})
+target_link_libraries(test_libmeshb_sol Meshb ${AIO_LIBRARIES})
 install (TARGETS test_libmeshb_sol DESTINATION share/libMeshb/examples COMPONENT examples)
 
 add_executable(test_libmeshb_HO test_libmeshb_HO.c)
-target_link_libraries(test_libmeshb_HO Meshb.7 ${AIO_LIBRARIES})
+target_link_libraries(test_libmeshb_HO Meshb ${AIO_LIBRARIES})
 install (TARGETS test_libmeshb_HO DESTINATION share/libMeshb/examples COMPONENT examples)
 
 add_executable(test_libmeshb_p2_sol test_libmeshb_p2_sol.c)
-target_link_libraries(test_libmeshb_p2_sol Meshb.7 ${AIO_LIBRARIES})
+target_link_libraries(test_libmeshb_p2_sol Meshb ${AIO_LIBRARIES})
 install (TARGETS test_libmeshb_p2_sol DESTINATION share/libMeshb/examples COMPONENT examples)
 
 add_executable(test_polyhedra test_polyhedra.c ${PROJECT_SOURCE_DIR}/utilities/libmeshb7_helpers.c)
-target_link_libraries(test_polyhedra Meshb.7 ${AIO_LIBRARIES})
+target_link_libraries(test_polyhedra Meshb ${AIO_LIBRARIES})
 install (TARGETS test_polyhedra DESTINATION share/libMeshb/examples COMPONENT examples)
 
 add_executable(mesh2poly mesh2poly.c)
-target_link_libraries(mesh2poly Meshb.7 ${AIO_LIBRARIES})
+target_link_libraries(mesh2poly Meshb ${AIO_LIBRARIES})
 install (TARGETS mesh2poly DESTINATION share/libMeshb/examples COMPONENT examples)
 
 
@@ -39,28 +39,28 @@ install (TARGETS mesh2poly DESTINATION s
 
 if (CMAKE_Fortran_COMPILER)
    add_executable(test_libmeshb_f77 test_libmeshb.f)
-   target_link_libraries(test_libmeshb_f77 Meshb.7 ${AIO_LIBRARIES})
+   target_link_libraries(test_libmeshb_f77 Meshb ${AIO_LIBRARIES})
    install (TARGETS test_libmeshb_f77 DESTINATION share/libMeshb/examples COMPONENT examples)
    
    add_executable(test_libmeshb_f90 test_libmeshb.f90)
-   target_link_libraries(test_libmeshb_f90 Meshb.7 ${AIO_LIBRARIES})
+   target_link_libraries(test_libmeshb_f90 Meshb ${AIO_LIBRARIES})
    install (TARGETS test_libmeshb_f90 DESTINATION share/libMeshb/examples COMPONENT examples)
    
    add_executable(test_libmeshb_block_f77 test_libmeshb_block.f)
-   target_link_libraries(test_libmeshb_block_f77 Meshb.7 ${AIO_LIBRARIES})
+   target_link_libraries(test_libmeshb_block_f77 Meshb ${AIO_LIBRARIES})
    install (TARGETS test_libmeshb_block_f77 DESTINATION share/libMeshb/examples COMPONENT examples)
    
    add_executable(test_libmeshb_block_f90 test_libmeshb_block.f90)
-   target_link_libraries(test_libmeshb_block_f90 Meshb.7 ${AIO_LIBRARIES})
+   target_link_libraries(test_libmeshb_block_f90 Meshb ${AIO_LIBRARIES})
    install (TARGETS test_libmeshb_block_f90 DESTINATION share/libMeshb/examples COMPONENT examples)
    
    add_executable(test_libmeshb_HO_f90 test_libmeshb_HO.f90)
-   target_link_libraries(test_libmeshb_HO_f90 Meshb.7 ${AIO_LIBRARIES})
+   target_link_libraries(test_libmeshb_HO_f90 Meshb ${AIO_LIBRARIES})
    install (TARGETS test_libmeshb_HO_f90 DESTINATION share/libMeshb/examples COMPONENT examples)
 
 
    #add_executable(test_libmeshb_block_bindC test_libmeshb_block_bindC.f90)
-   #target_link_libraries(test_libmeshb_block_bindC Meshb.7 ${AIO_LIBRARIES})
+   #target_link_libraries(test_libmeshb_block_bindC Meshb ${AIO_LIBRARIES})
    #install (TARGETS test_libmeshb_block_bindC DESTINATION share/libMeshb/examples COMPONENT examples)
 
 endif ()
--- origsrc/libMeshb-7.80/sources/CMakeLists.txt	2024-02-27 18:27:56.000000000 +0900
+++ src/libMeshb-7.80/sources/CMakeLists.txt	2024-09-02 01:34:04.428368400 +0900
@@ -4,19 +4,17 @@
 ##########################
 
 if(CMAKE_Fortran_COMPILER)
-  add_library(Meshb.7 libmeshb7.c ../utilities/libmeshb7_helpers.c  libmeshb7_mod.f90)
+  add_library(Meshb libmeshb7.c ../utilities/libmeshb7_helpers.c  libmeshb7_mod.f90)
 else()
-  add_library(Meshb.7 libmeshb7.c ../utilities/libmeshb7_helpers.c)
+  add_library(Meshb libmeshb7.c ../utilities/libmeshb7_helpers.c)
 endif()
+set_target_properties(Meshb PROPERTIES SOVERSION 7)
 
 install (FILES libmeshb7.h libmeshb7.ins ../utilities/libmeshb7_helpers.h DESTINATION include COMPONENT headers)
 target_include_directories(
-  Meshb.7 PUBLIC $<INSTALL_INTERFACE:include>
+  Meshb PUBLIC $<INSTALL_INTERFACE:include>
                  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/sources>)
-install (TARGETS Meshb.7 EXPORT meshb-target DESTINATION lib COMPONENT libraries)
-install (EXPORT meshb-target NAMESPACE ${PROJECT_NAME}::
-         DESTINATION lib/cmake/${PROJECT_NAME})
-install (TARGETS Meshb.7 EXPORT libMeshb-target DESTINATION lib COMPONENT libraries)
+install (TARGETS Meshb EXPORT libMeshb-target RUNTIME DESTINATION bin ARCHIVE DESTINATION lib COMPONENT libraries)
 
 install(DIRECTORY ${CMAKE_Fortran_MODULE_DIRECTORY}/ DESTINATION include)
 
--- origsrc/libMeshb-7.80/utilities/CMakeLists.txt	2024-02-27 18:27:56.000000000 +0900
+++ src/libMeshb-7.80/utilities/CMakeLists.txt	2024-09-02 01:31:37.448499300 +0900
@@ -5,17 +5,17 @@
 
 add_executable(transmesh transmesh.c ../sources/libmeshb7.c)
 target_compile_options (transmesh PRIVATE -DTRANSMESH)
-target_link_libraries (transmesh Meshb.7 ${AIO_LIBRARIES})
+target_link_libraries (transmesh Meshb ${AIO_LIBRARIES})
 install (TARGETS transmesh DESTINATION bin COMPONENT applications)
 
 add_executable(mshinfo mshinfo.c)
-target_link_libraries (mshinfo Meshb.7 ${AIO_LIBRARIES})
+target_link_libraries (mshinfo Meshb ${AIO_LIBRARIES})
 install (TARGETS mshinfo DESTINATION bin COMPONENT applications)
 
 add_executable(speed_test speed_test.c)
-target_link_libraries (speed_test Meshb.7 ${AIO_LIBRARIES})
+target_link_libraries (speed_test Meshb ${AIO_LIBRARIES})
 install (TARGETS speed_test DESTINATION bin COMPONENT applications)
 
 add_executable(ugrid2mesh ugrid2mesh.c)
-target_link_libraries (ugrid2mesh Meshb.7 ${AIO_LIBRARIES})
+target_link_libraries (ugrid2mesh Meshb ${AIO_LIBRARIES})
 install (TARGETS ugrid2mesh DESTINATION bin COMPONENT applications)
