--- origsrc/beanstalkpp/CMakeLists.txt	2013-03-02 10:47:31.000000000 +0900
+++ src/beanstalkpp/CMakeLists.txt	2013-03-02 10:58:05.890670000 +0900
@@ -13,6 +13,10 @@ INCLUDE_DIRECTORIES( ${Boost_INCLUDE_DIR
 ADD_LIBRARY(
   beanstalkpp SHARED tokenizedstream.cpp serverexception.cpp exception.cpp client.cpp job.cpp 
 )
+TARGET_LINK_LIBRARIES(beanstalkpp boost_system-mt)
+SET_TARGET_PROPERTIES(
+  beanstalkpp PROPERTIES SOVERSION 0
+)
 
 ADD_EXECUTABLE(
   test test.cpp
@@ -39,11 +43,15 @@ ADD_EXECUTABLE(
 )
 TARGET_LINK_LIBRARIES(beanspeek ${Boost_LIBRARIES} beanstalkpp pthread)
 
-INSTALL(TARGETS beanstalkpp DESTINATION lib)
+INSTALL(TARGETS beanstalkpp RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 if(APPLE)
   set_target_properties(${PROJECT_NAME} PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
 endif(APPLE)
+if(CYGWIN)
+INSTALL(DIRECTORY . DESTINATION include/beanstalk++ FILES_MATCHING PATTERN "*.h" PATTERN "beanstalkpp.h" EXCLUDE PATTERN "*~" EXCLUDE PATTERN "*.git*" EXCLUDE PATTERN "*build*" EXCLUDE PATTERN "CYGWIN-PATCHES" EXCLUDE)
+else()
 INSTALL(DIRECTORY . DESTINATION include/beanstalk++ FILES_MATCHING PATTERN "*.h" PATTERN "beanstalkpp.h" EXCLUDE PATTERN "*~" EXCLUDE PATTERN "*.git*" EXCLUDE PATTERN "*build*" EXCLUDE)
+endif()
 INSTALL(FILES "beanstalkpp.h" DESTINATION include)
 
 SET(CMAKE_CXX_FLAGS "-Wall -Wextra -pedantic -g -O0")
