--- origsrc/beanstalkpp/CMakeLists.txt	2012-02-04 08:36:31.000000000 +0900
+++ src/beanstalkpp/CMakeLists.txt	2012-02-04 08:44:27.971401200 +0900
@@ -9,33 +9,38 @@ INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINA
 ADD_LIBRARY(
   beanstalkpp SHARED tokenizedstream.cpp serverexception.cpp exception.cpp client.cpp job.cpp 
 )
+TARGET_LINK_LIBRARIES(beanstalkpp boost_system-mt)
+
+SET_TARGET_PROPERTIES(
+  beanstalkpp PROPERTIES VERSION 0 SOVERSION 0
+)
 
 ADD_EXECUTABLE(
   test test.cpp
 )
-TARGET_LINK_LIBRARIES(test boost_iostreams boost_system beanstalkpp boost_regex pthread)
+TARGET_LINK_LIBRARIES(test boost_iostreams-mt boost_system-mt beanstalkpp boost_regex-mt pthread)
 
 ADD_EXECUTABLE(
   beansreserve beansreserve.cpp
 )
-TARGET_LINK_LIBRARIES(beansreserve boost_iostreams boost_system beanstalkpp boost_regex pthread)
+TARGET_LINK_LIBRARIES(beansreserve boost_iostreams-mt boost_system-mt beanstalkpp boost_regex-mt pthread)
 
 ADD_EXECUTABLE(
   beansput beansput.cpp
 )
-TARGET_LINK_LIBRARIES(beansput boost_iostreams boost_system beanstalkpp boost_regex pthread)
+TARGET_LINK_LIBRARIES(beansput boost_iostreams-mt boost_system-mt beanstalkpp boost_regex-mt pthread)
 
 ADD_EXECUTABLE(
   listtubes listtubes.cpp
 )
-TARGET_LINK_LIBRARIES(listtubes boost_iostreams boost_system beanstalkpp boost_regex pthread)
+TARGET_LINK_LIBRARIES(listtubes boost_iostreams-mt boost_system-mt beanstalkpp boost_regex-mt pthread)
 
 ADD_EXECUTABLE(
   beanspeek beanspeek.cpp
 )
-TARGET_LINK_LIBRARIES(beanspeek boost_iostreams boost_system beanstalkpp boost_regex pthread)
+TARGET_LINK_LIBRARIES(beanspeek boost_iostreams-mt boost_system-mt beanstalkpp boost_regex-mt pthread)
 
-INSTALL(TARGETS beanstalkpp DESTINATION lib)
+INSTALL(TARGETS beanstalkpp RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 INSTALL(DIRECTORY . DESTINATION include/beanstalk++ FILES_MATCHING PATTERN "*.h" PATTERN "beanstalkpp.h" EXCLUDE PATTERN "*~" EXCLUDE PATTERN "*.git*" EXCLUDE PATTERN "*build*" EXCLUDE)
 INSTALL(FILES "beanstalkpp.h" DESTINATION include)
 
