--- origsrc/pwsafe-0.2.0/configure.ac	2005-10-01 06:06:37.000000000 +0900
+++ src/pwsafe-0.2.0/configure.ac	2024-09-20 11:57:49.728991600 +0900
@@ -146,7 +146,7 @@ AC_CHECK_TYPE(socklen_t,[AC_DEFINE(HAVE_
 
 
 dnl Checks for library functions.
-AM_SYS_POSIX_TERMIOS
+AC_SYS_POSIX_TERMIOS
 if test "$am_cv_sys_posix_termios" != yes -a "$ac_cv_sys_posix_termios" != yes; then
   AC_MSG_ERROR("POSIX termios operations are required")
 fi
