--- origsrc/zeromq-2.1.7/src/Makefile.am	2011-03-11 05:24:37.000000000 +0900
+++ src/zeromq-2.1.7/src/Makefile.am	2011-08-21 11:14:12.947185200 +0900
@@ -139,10 +139,11 @@ libzmq_la_SOURCES = \
     zmq_listener.cpp
 
 if ON_MINGW
-libzmq_la_LDFLAGS = -no-undefined -avoid-version -version-info @LTVER@ @LIBZMQ_EXTRA_LDFLAGS@
+libzmq_la_LDFLAGS = -avoid-version -version-info @LTVER@ @LIBZMQ_EXTRA_LDFLAGS@
 else
 libzmq_la_LDFLAGS = -version-info @LTVER@ @LIBZMQ_EXTRA_LDFLAGS@
 endif
+libzmq_la_LDFLAGS += -no-undefined
 
 libzmq_la_CXXFLAGS = @LIBZMQ_EXTRA_CXXFLAGS@
 
