--- origsrc/mongo-cxx-driver-r4.1.4/cmake/BsoncxxUtil.cmake	2025-10-09 02:10:42.000000000 +0900
+++ src/mongo-cxx-driver-r4.1.4/cmake/BsoncxxUtil.cmake	2025-12-07 22:25:29.172760300 +0900
@@ -30,7 +30,11 @@ function(bsoncxx_add_library TARGET OUTP
 
         # ABI version number. Only necessary for shared library targets.
         if(LINK_TYPE STREQUAL "SHARED")
-            set(soversion _noabi)
+            if (DEFINED SOVERSION)
+                set(soversion ${SOVERSION})
+            else()
+                set(soversion _noabi)
+            endif()
             set_target_properties(${TARGET} PROPERTIES SOVERSION ${soversion})
             string(APPEND abi_tag "-v${soversion}")
         endif()
--- origsrc/mongo-cxx-driver-r4.1.4/cmake/MongocxxUtil.cmake	2025-10-09 02:10:42.000000000 +0900
+++ src/mongo-cxx-driver-r4.1.4/cmake/MongocxxUtil.cmake	2025-12-07 22:30:33.861163900 +0900
@@ -24,7 +24,11 @@ function(mongocxx_add_library TARGET OUT
 
         # ABI version number. Only necessary for shared library targets.
         if(LINK_TYPE STREQUAL "SHARED")
-            set(soversion _noabi)
+            if (DEFINED SOVERSION)
+                set(soversion ${SOVERSION})
+            else()
+                set(soversion _noabi)
+            endif()
             set_target_properties(${TARGET} PROPERTIES SOVERSION ${soversion})
             string(APPEND abi_tag "-v${soversion}")
         endif()
--- origsrc/mongo-cxx-driver-r4.1.4/src/bsoncxx/include/bsoncxx/v1/detail/compare.hpp	2025-10-09 02:10:42.000000000 +0900
+++ src/mongo-cxx-driver-r4.1.4/src/bsoncxx/include/bsoncxx/v1/detail/compare.hpp	2025-12-07 19:42:40.217635600 +0900
@@ -100,9 +100,15 @@ class strong_ordering {
 
 #pragma push_macro("INLINE_VAR")
 #undef INLINE_VAR
+#if defined(__CYGWIN__)
+// Cygwin uses PE/COFF format which doesn't properly support weak symbols
+// Use inline instead (requires C++17 or later)
+#define INLINE_VAR inline
+#else
 #define INLINE_VAR                             \
     BSONCXX_PRIVATE_IF_GNU_LIKE([[gnu::weak]]) \
     BSONCXX_PRIVATE_IF_MSVC(__declspec(selectany))
+#endif
 
 INLINE_VAR const strong_ordering strong_ordering::less = strong_ordering(strong_ordering::_construct{}, -1);
 INLINE_VAR const strong_ordering strong_ordering::greater = strong_ordering(strong_ordering::_construct{}, 1);
