--- origsrc/roscpp_core-0.7.2/cpp_common/CMakeLists.txt	2020-05-29 13:45:12.000000000 +0900
+++ src/roscpp_core-0.7.2/cpp_common/CMakeLists.txt	2020-07-16 17:57:18.299312100 +0900
@@ -34,6 +34,10 @@ endif(HAVE_GLIBC_BACKTRACE)
 add_library(${PROJECT_NAME} src/debug.cpp src/header.cpp)
 target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES} ${console_bridge_LIBRARIES})
 
+if(CYGWIN)
+  set_target_properties(cpp_common PROPERTIES SOVERSION 0)
+endif()
+
 install(TARGETS ${PROJECT_NAME}
   ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
   LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
