--- origsrc/libtsm-3/Makefile.am	2013-10-29 16:27:36.084000000 +0900
+++ src/libtsm-3/Makefile.am	2015-05-02 10:49:17.218651900 +0900
@@ -70,10 +70,10 @@ AM_CPPFLAGS = \
 	-include $(top_builddir)/config.h \
 	-I $(srcdir)/src
 AM_LDFLAGS = \
-	-Wl,--as-needed \
-	-Wl,--gc-sections \
-	-Wl,-z,relro \
-	-Wl,-z,now
+	--as-needed \
+	--gc-sections \
+	-z relro \
+	-z now
 
 if BUILD_ENABLE_DEBUG
 AM_CFLAGS += -g
@@ -125,7 +125,7 @@ libtsm_la_SOURCES = \
 libtsm_la_CPPFLAGS = $(AM_CPPFLAGS)
 libtsm_la_LIBADD = libshl.la
 EXTRA_libtsm_la_DEPENDENCIES = $(top_srcdir)/docs/libtsm.sym
-libtsm_la_LDFLAGS = \
+libtsm_la_LDFLAGS = -no-undefined \
 	$(AM_LDFLAGS) \
 	-version-info $(LIBTSM_CURRENT):$(LIBTSM_REVISION):$(LIBTSM_AGE) \
 	-Wl,--version-script="$(top_srcdir)/docs/libtsm.sym"
@@ -159,7 +159,7 @@ test_lflags = \
 test_htable_SOURCES = test/test_htable.c $(test_sources)
 test_htable_CPPFLAGS = $(test_cflags)
 test_htable_LDADD = $(test_libs)
-test_htable_LDFLAGS = $(test_lflags)
+test_htable_LDFLAGS = # $(test_lflags)
 
 #
 # Phony targets
