--- origsrc/GKlib/CMakeLists.txt	2019-10-24 23:23:42.000000000 +0900
+++ src/GKlib/CMakeLists.txt	2019-10-24 23:23:46.514454000 +0900
@@ -7,15 +7,28 @@ unset(abs)
 include(GKlibSystem.cmake)
 
 include_directories(".")
-add_library(GKlib STATIC ${GKlib_sources})
+add_library(GKlib SHARED ${GKlib_sources})
 if(UNIX)
   target_link_libraries(GKlib m)
 endif(UNIX)
+if(PCRE)
+  target_link_libraries(GKlib pcreposix)
+endif()
+if(CYGWIN)
+  set_target_properties(GKlib PROPERTIES SOVERSION 0)
+endif()
 
 include_directories("test")
 add_subdirectory("test")
 
 install(TARGETS GKlib
-  ARCHIVE DESTINATION lib/${LINSTALL_PATH}
-  LIBRARY DESTINATION lib/${LINSTALL_PATH})
-install(FILES ${GKlib_includes} DESTINATION include/${HINSTALL_PATH})
+  ARCHIVE DESTINATION lib
+  RUNTIME DESTINATION bin)
+install(FILES ${GKlib_includes} DESTINATION include/GKlib/${HINSTALL_PATH})
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/GKlib.pc.in ${CMAKE_BINARY_DIR}/GKlib.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/GKlib.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/GKlib/GKlib.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/GKlib/GKlib.pc.in	2019-10-24 23:23:46.514454000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: GKlib
+Description: Library of various helper routines and frameworks used by many of the lab's software
+Version: 0
+Libs: -L${libdir} -lGKlib
+Cflags: -I${includedir}/GKlib
