diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libofetion-2.2.2/CMakeLists.txt src/libofetion-2.2.2/CMakeLists.txt
--- origsrc/libofetion-2.2.2/CMakeLists.txt	2012-01-02 02:41:32.000000000 +0900
+++ src/libofetion-2.2.2/CMakeLists.txt	2016-01-03 02:08:10.966004700 +0900
@@ -100,13 +100,21 @@ set_target_properties(ofetion PROPERTIES
     LINK_FLAGS ${_link_flags}
 )
 
+option(ENABLE_STATIC "enable static" OFF)
+if(ENABLE_STATIC)
 add_library(ofetion-static STATIC ${libofetion_SRC})
 SET_TARGET_PROPERTIES(ofetion-static PROPERTIES OUTPUT_NAME "ofetion")
 SET_TARGET_PROPERTIES(ofetion-static PROPERTIES PREFIX "lib")
+endif()
 
-install(TARGETS ofetion ofetion-static
+install(TARGETS ofetion
+	RUNTIME DESTINATION ${BIN_INSTALL_DIR}
+	ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
+if(ENABLE_STATIC)
+install(TARGETS ofetion-static
 	LIBRARY DESTINATION ${LIB_INSTALL_DIR}
 	ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
+endif()
 file(GLOB ofetion_INCLUDES "./*.h")
 install(FILES ${ofetion_INCLUDES} DESTINATION ${INCLUDE_INSTALL_DIR} COMPONENT Devel)
 
