--- origsrc/libofetion-2.2.2/CMakeLists.txt	2012-01-02 02:41:32.000000000 +0900
+++ src/libofetion-2.2.2/CMakeLists.txt	2020-01-07 18:57:34.081852900 +0900
@@ -100,13 +100,21 @@ set_target_properties(ofetion PROPERTIES
     LINK_FLAGS ${_link_flags}
 )
 
+option(ENABLE_STATIC "enable static" OFF)
+if(ENABLE_STATIC)
 add_library(ofetion-static STATIC ${libofetion_SRC})
 SET_TARGET_PROPERTIES(ofetion-static PROPERTIES OUTPUT_NAME "ofetion")
 SET_TARGET_PROPERTIES(ofetion-static PROPERTIES PREFIX "lib")
+endif()
 
-install(TARGETS ofetion ofetion-static
+install(TARGETS ofetion
+	RUNTIME DESTINATION ${BIN_INSTALL_DIR}
+	ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
+if(ENABLE_STATIC)
+install(TARGETS ofetion-static
 	LIBRARY DESTINATION ${LIB_INSTALL_DIR}
 	ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
+endif()
 file(GLOB ofetion_INCLUDES "./*.h")
 install(FILES ${ofetion_INCLUDES} DESTINATION ${INCLUDE_INSTALL_DIR} COMPONENT Devel)
 
