--- origsrc/lastfmlib-0.4.0/Makefile.am	2009-11-28 17:58:07.000000000 +0900
+++ src/lastfmlib-0.4.0/Makefile.am	2010-10-31 19:57:32.899894800 +0900
@@ -49,7 +49,7 @@ nobase_include_HEADERS   =      lastfmli
                                 lastfmlib/utils/log.h
 
 liblastfmlib_la_CXXFLAGS       =   $(AM_CXXFLAGS)
-liblastfmlib_la_LDFLAGS        =   -version-info 4:0:3
+liblastfmlib_la_LDFLAGS        =   -version-info 4:0:3 -no-undefined
 liblastfmlib_la_LIBADD         =   $(CURL_LIBS) -lpthread
 
 ACLOCAL_AMFLAGS = -I m4
@@ -58,6 +58,7 @@ pkgconfigdir = @pkgconfigdir@
 pkgconfig_DATA = liblastfmlib.pc
 
 if ENABLE_UNITTEST
+TESTS = testrunner
 check_PROGRAMS = testrunner
 endif
 
--- origsrc/lastfmlib-0.4.0/configure.ac	2009-11-28 17:46:30.000000000 +0900
+++ src/lastfmlib-0.4.0/configure.ac	2010-10-31 19:26:02.177177200 +0900
@@ -5,7 +5,7 @@ AM_INIT_AUTOMAKE
 AC_LANG_CPLUSPLUS
 AC_PROG_CXX
 AC_PROG_INSTALL
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 AC_CONFIG_MACRO_DIR([m4])
 AC_HEADER_STDC
