--- origsrc/libgpewidget-0.118/Makefile.am	2013-01-14 17:39:05.000000000 +0900
+++ src/libgpewidget-0.118/Makefile.am	2015-02-16 20:24:01.108541000 +0900
@@ -34,7 +34,7 @@ libgpewidget_la_SOURCES = gtkdatecombo.c
                           color-slider.c colordialog.c link-warning.h \
                           colorrenderer.c
                           
-libgpewidget_la_LDFLAGS = Wl,--no-add-needed -lX11 -lm @GPEWIDGET_LIBS@ @HILDON_LIBS@ -version-info 1:0:0 
+libgpewidget_la_LDFLAGS = -no-undefined -Wl,--no-add-needed -lX11 -lm @GPEWIDGET_LIBS@ @HILDON_LIBS@ -version-info 1:0:0 
 
 gpeincludedir = $(includedir)/gpe
 gpeinclude_HEADERS = gpe/init.h gpe/errorbox.h gpe/smallbox.h gpe/pixmaps.h gpe/gtkdatecombo.h \
@@ -56,7 +56,7 @@ pixmap_DATA = pixmaps/clock.png pixmaps/
 bin_PROGRAMS = infoprint
 infoprint_DEPENDENCIES = libgpewidget.la
 infoprint_SOURCES = infoprint-main.c
-infoprint_LDADD = -lX11 -lgpewidget
+infoprint_LDADD = -lX11 libgpewidget.la
 
 EXTRA_DIST = $(pixmap_DATA) \
              intltool-extract.in \
