--- origsrc/idzebra-2.1.4/index/Makefile.am	2018-05-03 20:25:27.000000000 +0900
+++ src/idzebra-2.1.4/index/Makefile.am	2019-06-26 05:38:00.299009700 +0900
@@ -21,37 +21,37 @@ tabdatadir = $(datadir)/$(PACKAGE)$(PACK
 
 # The shared modules 
 mod_grs_regx_la_SOURCES = mod_grs_regx.c
-mod_grs_regx_la_LDFLAGS = -rpath $(modlibdir) -module -avoid-version
+mod_grs_regx_la_LDFLAGS = -no-undefined -rpath $(modlibdir) -module -avoid-version
 mod_grs_regx_la_LADD = $(TCL_LIB)
 mod_grs_regx_la_LIBADD = $(zebralib) $(mod_grs_regx_la_LADD)
 
 mod_grs_xml_la_SOURCES = mod_grs_xml.c
-mod_grs_xml_la_LDFLAGS = -rpath $(modlibdir) -module -avoid-version
+mod_grs_xml_la_LDFLAGS = -no-undefined -rpath $(modlibdir) -module -avoid-version
 mod_grs_xml_la_LADD = $(EXPAT_LIBS)
 mod_grs_xml_la_LIBADD = $(zebralib) $(mod_grs_xml_la_LADD)
 
 mod_grs_marc_la_SOURCES = mod_grs_marc.c marcomp.c marcomp.h inline.c inline.h
-mod_grs_marc_la_LDFLAGS = -rpath $(modlibdir) -module -avoid-version
+mod_grs_marc_la_LDFLAGS = -no-undefined -rpath $(modlibdir) -module -avoid-version
 mod_grs_marc_la_LADD = $(YAZLALIB)
 mod_grs_marc_la_LIBADD = $(zebralib) $(mod_grs_marc_la_LADD)
 
 mod_safari_la_SOURCES = mod_safari.c
-mod_safari_la_LDFLAGS = -rpath $(modlibdir) -module -avoid-version
+mod_safari_la_LDFLAGS = -no-undefined -rpath $(modlibdir) -module -avoid-version
 mod_safari_la_LADD = $(YAZLALIB)
 mod_safari_la_LIBADD = $(zebralib) $(mod_safari_la_LADD)
 
 mod_alvis_la_SOURCES = mod_alvis.c
-mod_alvis_la_LDFLAGS = -rpath $(modlibdir) -module -avoid-version
+mod_alvis_la_LDFLAGS = -no-undefined -rpath $(modlibdir) -module -avoid-version
 mod_alvis_la_LADD = $(YAZLALIB)
 mod_alvis_la_LIBADD = $(zebralib) $(mod_alvis_la_LADD)
 
 mod_dom_la_SOURCES = mod_dom.c
-mod_dom_la_LDFLAGS = -rpath $(modlibdir) -module -avoid-version
+mod_dom_la_LDFLAGS = -no-undefined -rpath $(modlibdir) -module -avoid-version
 mod_dom_la_LADD = $(YAZLALIB)
 mod_dom_la_LIBADD = $(zebralib) $(mod_dom_la_LADD)
 
 mod_text_la_SOURCES = mod_text.c
-mod_text_la_LDFLAGS = -rpath $(modlibdir) -module -avoid-version
+mod_text_la_LDFLAGS = -no-undefined -rpath $(modlibdir) -module -avoid-version
 mod_text_la_LADD = $(YAZLALIB)
 mod_text_la_LIBADD = $(zebralib) $(mod_text_la_LADD)
 
@@ -77,8 +77,8 @@ EXTRA_libidzebra_2_0_la_SOURCES = \
 lib_LTLIBRARIES = $(zebralib)
 
 libidzebra_2_0_la_DEPENDENCIES = $(STATIC_MODULE_OBJ) $(aux_libs)
-libidzebra_2_0_la_LIBADD = $(STATIC_MODULE_OBJ)  $(aux_libs) $(STATIC_MODULE_LADD)
-libidzebra_2_0_la_LDFLAGS=-export-dynamic -version-info $(ZEBRALIBS_VERSION_INFO)
+libidzebra_2_0_la_LIBADD = $(STATIC_MODULE_OBJ)  $(aux_libs) $(STATIC_MODULE_LADD) $(YAZLIB)
+libidzebra_2_0_la_LDFLAGS = -no-undefined -export-dynamic -version-info $(ZEBRALIBS_VERSION_INFO)
 libidzebra_2_0_la_SOURCES = \
   recctrl.c \
   recgrs.c \
