--- origsrc/libyogrt-1.27/.version	1970-01-01 09:00:00.000000000 +0900
+++ src/libyogrt-1.27/.version	2022-03-02 00:47:38.449889100 +0900
@@ -0,0 +1 @@
+UNKNOWN
--- origsrc/libyogrt-1.27/Makefile.am	2022-01-25 07:26:44.000000000 +0900
+++ src/libyogrt-1.27/Makefile.am	2022-03-02 00:39:19.204444800 +0900
@@ -1,6 +1,7 @@
 SUBDIRS = src doc test
 EXTRA_DIST = COPYING COPYING.LESSER DISCLAIMER INSTALL libyogrt.spec README.md
 dist_sysconf_DATA = yogrt.conf.example
+ACLOCAL_AMFLAGS = -I config
 
 EXTRA_DIST += $(top_srcdir)/.version
 BUILT_SOURCES = $(top_srcdir)/.version
