--- origsrc/spdlog-1.10.0/bench/bench.cpp	2022-04-04 22:48:58.000000000 +0900
+++ src/spdlog-1.10.0/bench/bench.cpp	2022-09-10 07:48:37.615051300 +0900
@@ -15,7 +15,12 @@
 #if defined(SPDLOG_USE_STD_FORMAT)
 #    include <format>
 #elif defined(SPDLOG_FMT_EXTERNAL)
-#    include <fmt/locale.h>
+#    include <fmt/core.h>
+#    if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#        include <locale>
+#    else
+#        include <fmt/locale.h>
+#    endif
 #else
 #    include "spdlog/fmt/bundled/format.h"
 #endif
