--- origsrc/cmph-2.0.2/cxxmph/Makefile.am	2018-12-28 12:35:52.000000000 +0900
+++ src/cmph-2.0.2/cxxmph/Makefile.am	2025-06-12 23:49:16.039201800 +0900
@@ -16,7 +16,7 @@ cxxmph_include_HEADERS = mph_bits.h mph_
 noinst_LTLIBRARIES = libcxxmph_bm.la
 lib_LTLIBRARIES = libcxxmph.la
 libcxxmph_la_SOURCES = MurmurHash3.cpp trigraph.cc mph_bits.cc mph_index.cc benchmark.h benchmark.cc string_util.cc
-libcxxmph_la_LDFLAGS = -version-info 0:0:0
+libcxxmph_la_LDFLAGS = -no-undefined -version-info 0:0:0
 libcxxmph_test_la_SOURCES = test.h test.cc
 libcxxmph_test_la_LIBADD = libcxxmph.la
 libcxxmph_bm_la_SOURCES = benchmark.h benchmark.cc bm_common.h bm_common.cc
--- origsrc/cmph-2.0.2/examples/Makefile.am	2018-12-28 11:11:36.000000000 +0900
+++ src/cmph-2.0.2/examples/Makefile.am	2025-06-12 23:49:16.039201800 +0900
@@ -1,6 +1,6 @@
 noinst_PROGRAMS = vector_adapter_ex1 file_adapter_ex2 struct_vector_adapter_ex3 small_set_ex4
 
-AM_CPPFLAGS = -I../src/
+AM_CPPFLAGS = -I$(top_srcdir)/src/
 
 vector_adapter_ex1_LDADD   = ../src/libcmph.la
 vector_adapter_ex1_SOURCES = vector_adapter_ex1.c
--- origsrc/cmph-2.0.2/src/Makefile.am	2014-11-25 12:37:13.000000000 +0900
+++ src/cmph-2.0.2/src/Makefile.am	2025-06-12 23:49:16.039201800 +0900
@@ -28,7 +28,7 @@ libcmph_la_SOURCES =  hash.h hash.c \
 		      cmph_benchmark.h cmph_benchmark.c \
 		      cmph_time.h
 
-libcmph_la_LDFLAGS = -version-info 0:0:0
+libcmph_la_LDFLAGS = -no-undefined -version-info 0:0:0
 
 cmph_SOURCES = 	main.c wingetopt.h wingetopt.c
 cmph_LDADD = libcmph.la
--- origsrc/cmph-2.0.2/tests/Makefile.am	2018-12-28 11:11:36.000000000 +0900
+++ src/cmph-2.0.2/tests/Makefile.am	2025-06-12 23:49:16.039201800 +0900
@@ -2,7 +2,7 @@ TESTS = $(check_PROGRAMS)
 check_PROGRAMS = graph_tests select_tests compressed_seq_tests compressed_rank_tests cmph_benchmark_test
 noinst_PROGRAMS = packed_mphf_tests mphf_tests
 
-AM_CPPFLAGS = -I../src/
+AM_CPPFLAGS = -I$(top_srcdir)/src/
 
 graph_tests_SOURCES = graph_tests.c
 graph_tests_LDADD = ../src/libcmph.la
