--- origsrc/libxcorr/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libxcorr/Makefile.am	2015-02-21 11:33:36.892666600 +0900
@@ -0,0 +1,11 @@
+
+include_HEADERS = include/xcorr.h
+
+lib_LTLIBRARIES = libxcorr.la
+libxcorr_la_CFLAGS = -std=c99
+libxcorr_la_SOURCES = src/xcorr.c
+libxcorr_la_LIBADD = $(FFTW3_LIBS)
+libxcorr_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = libxcorr.pc
--- origsrc/libxcorr/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/libxcorr/configure.ac	2015-02-21 11:31:50.939541600 +0900
@@ -0,0 +1,27 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([libxcorr], [0+git20110506])
+AC_CONFIG_SRCDIR([include/xcorr.h])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE([subdir-objects])
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_INSTALL
+LT_INIT
+
+# Checks for libraries.
+PKG_CHECK_MODULES([FFTW3], [fftw3])
+
+# Checks for header files.
+AC_CHECK_HEADERS([string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+
+# Checks for library functions.
+AC_CHECK_FUNCS([memset])
+
+AC_CONFIG_FILES([Makefile libxcorr.pc])
+AC_OUTPUT
--- origsrc/libxcorr/libxcorr.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libxcorr/libxcorr.pc.in	2015-02-21 11:31:03.392666600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libxcorr
+Description: Cross Correlation function for C using FFTW
+Version: 0+git20110506
+Libs: -L${libdir} -lxcorr
+Cflags: -I${includedir}
