--- origsrc/primesum-1.7/CMakeLists.txt	2019-02-23 19:21:47.000000000 +0900
+++ src/primesum-1.7/CMakeLists.txt	2024-03-10 15:24:35.305901500 +0900
@@ -68,7 +68,7 @@ include(GNUInstallDirs)
 
 # libprimesieve ######################################################
 
-add_subdirectory(lib/primesieve)
+find_package(primesieve REQUIRED)
 
 # Check if int128_t type exists ######################################
 
