--- origsrc/CLHEP-CLHEP_2_4_7_1/clhep-config.in	2023-10-13 03:36:02.000000000 +0900
+++ src/CLHEP-CLHEP_2_4_7_1/clhep-config.in	2024-12-13 10:59:19.174204800 +0900
@@ -89,7 +89,7 @@ while test $# -gt 0; do
        	;;
 
     --libs)
-       	echo -L\"${libdir}\" -lCLHEP-@VERSION@ @LIBS@
+       	echo -L\"${libdir}\" -lCLHEP @LIBS@
        	;;
 
     *)
--- origsrc/CLHEP-CLHEP_2_4_7_1/cmake/Modules/ClhepBuildLibrary.cmake	2023-10-13 03:36:02.000000000 +0900
+++ src/CLHEP-CLHEP_2_4_7_1/cmake/Modules/ClhepBuildLibrary.cmake	2024-12-13 10:40:58.029352500 +0900
@@ -20,39 +20,34 @@ macro(clhep_build_library package)
 
   # check for dependencies
   set( package_library_list )
-  set(package_library_list_static)
   if( ${PACKAGE}_DEPS )
      foreach ( dep ${${PACKAGE}_DEPS} )
         ##message( STATUS "clhep_build_library ${package} will use ${dep}")
 	list(APPEND package_library_list ${dep})
-	list(APPEND package_library_list_static ${dep}S)
      endforeach()
   endif()
 
   # Add the libraries and set properties
   ADD_LIBRARY(${package}  SHARED ${CLHEP_${package}_SOURCES})
-  ADD_LIBRARY(${package}S STATIC ${CLHEP_${package}_SOURCES})
   SET_TARGET_PROPERTIES (${package}
       PROPERTIES 
-        OUTPUT_NAME CLHEP-${package}-${VERSION}
+        OUTPUT_NAME CLHEP-${package}
       )
-  SET_TARGET_PROPERTIES(${package}S
-      PROPERTIES 
-        OUTPUT_NAME CLHEP-${package}-${VERSION}
-      )
-
+  
+  if (DEFINED SOVERSION)
+    set_target_properties(${package} PROPERTIES SOVERSION "${SOVERSION}")
+  endif()
+  
   # Do not add -Dname_EXPORTS to the command-line when building files in this
   # target. Doing so is actively harmful for the modules build because it
   # creates extra module variants, and not useful because we don't use these
   # macros.
   SET_TARGET_PROPERTIES(${package} PROPERTIES DEFINE_SYMBOL "")
-  SET_TARGET_PROPERTIES(${package}S PROPERTIES DEFINE_SYMBOL "")
 
   target_link_libraries(${package}  ${package_library_list} )
-  target_link_libraries(${package}S ${package_library_list_static} )
 
   # Install the libraries
-  INSTALL (TARGETS ${package} ${package}S
+  INSTALL (TARGETS ${package}
       EXPORT CLHEPLibraryDepends
       RUNTIME DESTINATION bin
       LIBRARY DESTINATION lib${LIB_SUFFIX}
@@ -70,25 +65,24 @@ macro(clhep_build_libclhep )
   ##message( STATUS "clheplib source list ${clhep_sources}" )
 
   ADD_LIBRARY (CLHEP  SHARED ${clhep_sources})
-  ADD_LIBRARY (CLHEPS STATIC ${clhep_sources})
 
   SET_TARGET_PROPERTIES(CLHEP
       PROPERTIES 
-        OUTPUT_NAME CLHEP-${VERSION}
-      )
-  SET_TARGET_PROPERTIES(CLHEPS
-      PROPERTIES 
-        OUTPUT_NAME CLHEP-${VERSION}
+        OUTPUT_NAME CLHEP
       )
+  
+  if (DEFINED SOVERSION)
+    set_target_properties(CLHEP PROPERTIES SOVERSION "${SOVERSION}")
+  endif()
+  
   # Do not add -Dname_EXPORTS to the command-line when building files in this
   # target. Doing so is actively harmful for the modules build because it
   # creates extra module variants, and not useful because we don't use these
   # macros.
   SET_TARGET_PROPERTIES(CLHEP PROPERTIES DEFINE_SYMBOL "")
-  SET_TARGET_PROPERTIES(CLHEPS PROPERTIES DEFINE_SYMBOL "")
 
   # export creates library dependency files for CLHEPConfig.cmake
-  INSTALL(TARGETS CLHEP CLHEPS
+  INSTALL(TARGETS CLHEP
       EXPORT CLHEPLibraryDepends
       RUNTIME DESTINATION bin
       LIBRARY DESTINATION lib${LIB_SUFFIX}
@@ -96,21 +90,4 @@ macro(clhep_build_libclhep )
       INCLUDES DESTINATION include
       ) 
 
-  if( ${CMAKE_SYSTEM_NAME} MATCHES "Windows" )
-      # copy
-      file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/clhep_static_symlink 
-        	 "execute_process(COMMAND \"${CMAKE_COMMAND}\" -E copy ${CMAKE_STATIC_LIBRARY_PREFIX}CLHEP-${VERSION}${CMAKE_STATIC_LIBRARY_SUFFIX} ${CMAKE_STATIC_LIBRARY_PREFIX}CLHEP${CMAKE_STATIC_LIBRARY_SUFFIX} WORKING_DIRECTORY \"$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}\" )" )
-      file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/clhep_shared_symlink 
-        	 "execute_process(COMMAND \"${CMAKE_COMMAND}\" -E copy ${CMAKE_SHARED_LIBRARY_PREFIX}CLHEP-${VERSION}${CMAKE_SHARED_LIBRARY_SUFFIX} ${CMAKE_SHARED_LIBRARY_PREFIX}CLHEP${CMAKE_SHARED_LIBRARY_SUFFIX} WORKING_DIRECTORY \"$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/bin\" )" )
-  else()
-      # create the symbolic links
-      file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/clhep_static_symlink 
-        	 "execute_process(COMMAND \"${CMAKE_COMMAND}\" -E create_symlink ${CMAKE_STATIC_LIBRARY_PREFIX}CLHEP-${VERSION}${CMAKE_STATIC_LIBRARY_SUFFIX} ${CMAKE_STATIC_LIBRARY_PREFIX}CLHEP${CMAKE_STATIC_LIBRARY_SUFFIX} WORKING_DIRECTORY \"\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}\" )" )
-      file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/clhep_shared_symlink 
-        	 "execute_process(COMMAND \"${CMAKE_COMMAND}\" -E create_symlink ${CMAKE_SHARED_LIBRARY_PREFIX}CLHEP-${VERSION}${CMAKE_SHARED_LIBRARY_SUFFIX} ${CMAKE_SHARED_LIBRARY_PREFIX}CLHEP${CMAKE_SHARED_LIBRARY_SUFFIX} WORKING_DIRECTORY \"\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}\" )" )
-  endif()
-
-  INSTALL(SCRIPT ${CMAKE_CURRENT_BINARY_DIR}/clhep_static_symlink )
-  INSTALL(SCRIPT ${CMAKE_CURRENT_BINARY_DIR}/clhep_shared_symlink )
-
 endmacro(clhep_build_libclhep )
--- origsrc/CLHEP-CLHEP_2_4_7_1/cmake/Modules/ClhepBuildTest.cmake	2023-10-13 03:36:02.000000000 +0900
+++ src/CLHEP-CLHEP_2_4_7_1/cmake/Modules/ClhepBuildTest.cmake	2024-12-13 10:12:26.346251000 +0900
@@ -26,14 +26,14 @@ macro( clhep_test testname )
   if( CTST_NOLIB )
     set(package_library_list)
   else()
-    set( package_library_list ${PACKAGE}S)
+    set( package_library_list ${PACKAGE})
     #set( package_library_list ${PROJECT_BINARY_DIR}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}CLHEP-${PACKAGE}-${VERSION}${CMAKE_STATIC_LIBRARY_SUFFIX} )
   endif()
 
   if( ${PACKAGE}_DEPS )
      foreach ( dep ${${PACKAGE}_DEPS} )
         ##message( STATUS "clhep_test ${testname} will use ${dep}")
-	list(APPEND package_library_list ${PACKAGE}S)
+	list(APPEND package_library_list ${PACKAGE})
 	#list(APPEND package_library_list  ${PROJECT_BINARY_DIR}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}CLHEP-${dep}-${VERSION}${CMAKE_STATIC_LIBRARY_SUFFIX} )
      endforeach()
   endif()
--- origsrc/CLHEP-CLHEP_2_4_7_1/cmake/Modules/ClhepToolchain.cmake	2023-10-13 03:36:02.000000000 +0900
+++ src/CLHEP-CLHEP_2_4_7_1/cmake/Modules/ClhepToolchain.cmake	2024-12-13 10:57:53.090520600 +0900
@@ -56,11 +56,8 @@ configure_package_config_file(${PROJECT_
 # This file is include()d by CLHEPConfig.cmake
 # First build up list of all modular targets, static and shared
 set(CLHEP_libraries_all ${CLHEP_libraries})
-foreach(_lib ${CLHEP_libraries})
-  list(APPEND CLHEP_libraries_all "${_lib}S")
-endforeach()
 
-export(TARGETS CLHEP CLHEPS ${CLHEP_libraries_all}
+export(TARGETS CLHEP ${CLHEP_libraries_all}
   NAMESPACE "CLHEP::"
   FILE ${PROJECT_BINARY_DIR}/CLHEPLibraryDepends.cmake
   )
@@ -73,9 +70,9 @@ export(TARGETS CLHEP CLHEPS ${CLHEP_libr
 # In the build tree we hardcode all paths, as we never need to relocate
 # a build tree
 #
-set(CLHEP_PKGCONFIG_PREFIX ${PROJECT_BINARY_DIR})
-set(CLHEP_PKGCONFIG_INCLUDEDIR ${CLHEP_INCLUDE_DIR})
-set(CLHEP_PKGCONFIG_LIBDIR ${PROJECT_BINARY_DIR}/lib${LIB_SUFFIX})
+set(CLHEP_PKGCONFIG_PREFIX ${CMAKE_INSTALL_PREFIX})
+set(CLHEP_PKGCONFIG_INCLUDEDIR ${CMAKE_INSTALL_PREFIX}/include)
+set(CLHEP_PKGCONFIG_LIBDIR ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX})
 
 configure_file(${PROJECT_SOURCE_DIR}/cmake/Templates/clhep.pc.in
   ${PROJECT_BINARY_DIR}/clhep.pc
@@ -130,9 +127,6 @@ configure_package_config_file(${PROJECT_
 # to the include and lib dirs.... This make the pc file fully relocatable!!
 # If you later use GNUInstallDirs.cmake, this would need more work to
 # ensure accurate calculation of the relative path.
-set(CLHEP_PKGCONFIG_PREFIX "\${pcfiledir}/../..")
-set(CLHEP_PKGCONFIG_LIBDIR "\${prefix}/lib${LIB_SUFFIX}")
-set(CLHEP_PKGCONFIG_INCLUDEDIR "\${prefix}/include")
 
 configure_file(${PROJECT_SOURCE_DIR}/cmake/Templates/clhep.pc.in
   ${PROJECT_BINARY_DIR}/InstallTreeFiles/clhep.pc
@@ -161,12 +155,12 @@ endforeach()
 install(FILES
   ${PROJECT_BINARY_DIR}/InstallTreeFiles/CLHEPConfigVersion.cmake 
   ${PROJECT_BINARY_DIR}/InstallTreeFiles/CLHEPConfig.cmake 
-  DESTINATION lib${LIB_SUFFIX}/CLHEP-${VERSION}
+  DESTINATION lib${LIB_SUFFIX}/cmake/CLHEP
   )
 
 install(EXPORT CLHEPLibraryDepends 
         NAMESPACE "CLHEP::"
-        DESTINATION lib${LIB_SUFFIX}/CLHEP-${VERSION})
+        DESTINATION lib${LIB_SUFFIX}/cmake/CLHEP)
 
 # Install the pkg-config file. The choice of 'lib${LIB_SUFFIX}/pkgconfig' for the 
 # installation seems fairly standard.
--- origsrc/CLHEP-CLHEP_2_4_7_1/cmake/Modules/ClhepVariables.cmake	2023-10-13 03:36:02.000000000 +0900
+++ src/CLHEP-CLHEP_2_4_7_1/cmake/Modules/ClhepVariables.cmake	2024-12-13 11:05:11.763864700 +0900
@@ -455,7 +455,7 @@ macro( clhep_package_config_no_lib )
   ## don't install <package>-config on Windows
   if( NOT ${CMAKE_SYSTEM_NAME} MATCHES "Windows" )
     install ( PROGRAMS ${CLHEP_BINARY_DIR}/${PACKAGE}/${PACKAGE}-config
-              DESTINATION bin
+              DESTINATION bin RENAME clhep-${PACKAGE}-config
 	    )
   endif()
 endmacro( clhep_package_config_no_lib )
@@ -463,11 +463,11 @@ endmacro( clhep_package_config_no_lib )
 macro( clhep_package_config )
   set( ${PACKAGE}_CPPFLAGS "-I${includedir}" )
   set( ${PACKAGE}_LDFLAGS  "-L\${exec_prefix}/lib${LIB_SUFFIX}" )
-  set( ${PACKAGE}_LIBS     "-lCLHEP-${PACKAGE}-${VERSION}" )
+  set( ${PACKAGE}_LIBS     "-lCLHEP-${PACKAGE}" )
   if( ${PACKAGE}_DEPS )
      foreach ( dep ${${PACKAGE}_DEPS} )
         message( STATUS "${PACKAGE} depends on ${dep}")
-	set( ${PACKAGE}_LIBS     "${${PACKAGE}_LIBS} -lCLHEP-${dep}-${VERSION}" )
+	set( ${PACKAGE}_LIBS     "${${PACKAGE}_LIBS} -lCLHEP-${dep}" )
      endforeach()
   endif()
   configure_file ( ${CLHEP_SOURCE_DIR}/${PACKAGE}/${PACKAGE}-config.in
@@ -477,7 +477,7 @@ macro( clhep_package_config )
   ## don't install <package>-config on Windows
   if( NOT ${CMAKE_SYSTEM_NAME} MATCHES "Windows" )
     install ( PROGRAMS ${CLHEP_BINARY_DIR}/${PACKAGE}/${PACKAGE}-config
-              DESTINATION bin
+              DESTINATION bin RENAME clhep-${PACKAGE}-config
 	    )
   endif()
 endmacro( clhep_package_config )
--- origsrc/CLHEP-CLHEP_2_4_7_1/cmake/Templates/clhep-component.pc.in	2023-10-13 03:36:02.000000000 +0900
+++ src/CLHEP-CLHEP_2_4_7_1/cmake/Templates/clhep-component.pc.in	2024-12-13 10:15:32.246878300 +0900
@@ -9,6 +9,6 @@ Name: CLHEP-@CLHEP_PC_COMPONENT@
 Description: Class Library for High Energy Physics, @CLHEP_PC_COMPONENT@ Component
 Version: @CLHEP_VERSION@
 Requires: @CLHEP_PC_COMPONENT_REQUIRES@
-Libs: -L${libdir} -lCLHEP-@CLHEP_PC_COMPONENT@-@CLHEP_VERSION@
+Libs: -L${libdir} -lCLHEP-@CLHEP_PC_COMPONENT@
 Cflags: -I${includedir}
 
--- origsrc/CLHEP-CLHEP_2_4_7_1/cmake/Templates/clhep.pc.in	2023-10-13 03:36:02.000000000 +0900
+++ src/CLHEP-CLHEP_2_4_7_1/cmake/Templates/clhep.pc.in	2024-12-13 10:15:35.954447600 +0900
@@ -8,6 +8,6 @@ includedir=@CLHEP_PKGCONFIG_INCLUDEDIR@
 Name: CLHEP
 Description: Class Library for High Energy Physics
 Version: @CLHEP_VERSION@
-Libs: -L${libdir} -lCLHEP-@CLHEP_VERSION@
+Libs: -L${libdir} -lCLHEP
 Cflags: -I${includedir}
 
