diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/2.2.0.8/CLHEP/cmake/Modules/ClhepBuildLibrary.cmake src/2.2.0.8/CLHEP/cmake/Modules/ClhepBuildLibrary.cmake
--- origsrc/2.2.0.8/CLHEP/cmake/Modules/ClhepBuildLibrary.cmake	2015-06-19 00:52:58.000000000 +0900
+++ src/2.2.0.8/CLHEP/cmake/Modules/ClhepBuildLibrary.cmake	2015-08-13 22:53:25.757466800 +0900
@@ -5,6 +5,8 @@
 # clhep_build_library( <package> <source code files> )
 # 
 
+option(ENABLE_STATIC "enable static" OFF)
+
 macro(clhep_build_library package)
   set( CLHEP_${package}_SOURCES ${ARGN} )
 
@@ -29,25 +31,29 @@ macro(clhep_build_library package)
 
   # Add the libraries and set properties
   ADD_LIBRARY(${package}  SHARED ${CLHEP_${package}_SOURCES})
+  if(ENABLE_STATIC)
   ADD_LIBRARY(${package}S STATIC ${CLHEP_${package}_SOURCES})
+  endif()
   SET_TARGET_PROPERTIES (${package}
       PROPERTIES 
-        OUTPUT_NAME CLHEP-${package}-${VERSION}
+        OUTPUT_NAME CLHEP-${package}
 	CLEAN_DIRECT_OUTPUT 1
+    SOVERSION 0
       )
+  if(ENABLE_STATIC)
   SET_TARGET_PROPERTIES(${package}S
       PROPERTIES 
         OUTPUT_NAME CLHEP-${package}-${VERSION}
 	CLEAN_DIRECT_OUTPUT 1
       )
-
+  endif()
   target_link_libraries(${package}  ${package_library_list} )
+  if(ENABLE_STATIC)
   target_link_libraries(${package}S ${package_library_list} )
-
+  endif()
   # Install the libraries
-  INSTALL (TARGETS ${package} ${package}S
+  INSTALL (TARGETS ${package}
       RUNTIME DESTINATION bin
-      LIBRARY DESTINATION lib${LIB_SUFFIX}
       ARCHIVE DESTINATION lib${LIB_SUFFIX}
       ) 
 endmacro(clhep_build_library)
@@ -61,24 +67,26 @@ macro(clhep_build_libclhep )
   ##message( STATUS "clheplib source list ${clhep_sources}" )
 
   ADD_LIBRARY (CLHEP  SHARED ${clhep_sources})
+  if(ENABLE_STATIC)
   ADD_LIBRARY (CLHEPS STATIC ${clhep_sources})
-
+  endif()
   SET_TARGET_PROPERTIES(CLHEP
       PROPERTIES 
-        OUTPUT_NAME CLHEP-${VERSION}
+        OUTPUT_NAME CLHEP
         CLEAN_DIRECT_OUTPUT 1
+        SOVERSION 0
       )
+  if(ENABLE_STATIC)
   SET_TARGET_PROPERTIES(CLHEPS
       PROPERTIES 
         OUTPUT_NAME CLHEP-${VERSION}
         CLEAN_DIRECT_OUTPUT 1
       )
-
+  endif()
   # export creates library dependency files for CLHEPConfig.cmake
-  INSTALL(TARGETS CLHEP CLHEPS
+  INSTALL(TARGETS CLHEP
       EXPORT CLHEPLibraryDepends
       RUNTIME DESTINATION bin
-      LIBRARY DESTINATION lib${LIB_SUFFIX}
       ARCHIVE DESTINATION lib${LIB_SUFFIX}
       ) 
 
@@ -96,7 +104,8 @@ macro(clhep_build_libclhep )
         	 "execute_process(COMMAND \"${CMAKE_COMMAND}\" -E create_symlink ${CMAKE_SHARED_LIBRARY_PREFIX}CLHEP-${VERSION}${CMAKE_SHARED_LIBRARY_SUFFIX} ${CMAKE_SHARED_LIBRARY_PREFIX}CLHEP${CMAKE_SHARED_LIBRARY_SUFFIX} WORKING_DIRECTORY \"\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}\" )" )
   endif()
 
+  if(NOT CYGWIN)
   INSTALL(SCRIPT ${CMAKE_CURRENT_BINARY_DIR}/clhep_static_symlink )
   INSTALL(SCRIPT ${CMAKE_CURRENT_BINARY_DIR}/clhep_shared_symlink )
-
+  endif()
 endmacro(clhep_build_libclhep )
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/2.2.0.8/CLHEP/cmake/Modules/ClhepBuildTest.cmake src/2.2.0.8/CLHEP/cmake/Modules/ClhepBuildTest.cmake
--- origsrc/2.2.0.8/CLHEP/cmake/Modules/ClhepBuildTest.cmake	2015-06-19 00:52:58.000000000 +0900
+++ src/2.2.0.8/CLHEP/cmake/Modules/ClhepBuildTest.cmake	2015-08-13 22:37:27.170204000 +0900
@@ -26,14 +26,14 @@ macro( clhep_test testname )
   if( CTST_NOLIB )
     set(package_library_list)
   else()
-    set( package_library_list ${PACKAGE}S)
+    set( package_library_list ${PACKAGE})
     #set( package_library_list ${PROJECT_BINARY_DIR}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}CLHEP-${PACKAGE}-${VERSION}${CMAKE_STATIC_LIBRARY_SUFFIX} )
   endif()
 
   if( ${PACKAGE}_DEPS )
      foreach ( dep ${${PACKAGE}_DEPS} )
         ##message( STATUS "clhep_test ${testname} will use ${dep}")
-	list(APPEND package_library_list ${PACKAGE}S)
+	list(APPEND package_library_list ${PACKAGE})
 	#list(APPEND package_library_list  ${PROJECT_BINARY_DIR}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}CLHEP-${dep}-${VERSION}${CMAKE_STATIC_LIBRARY_SUFFIX} )
      endforeach()
   endif()
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/2.2.0.8/CLHEP/cmake/Modules/ClhepToolchain.cmake src/2.2.0.8/CLHEP/cmake/Modules/ClhepToolchain.cmake
--- origsrc/2.2.0.8/CLHEP/cmake/Modules/ClhepToolchain.cmake	2015-06-19 00:52:58.000000000 +0900
+++ src/2.2.0.8/CLHEP/cmake/Modules/ClhepToolchain.cmake	2015-08-13 22:38:27.176004100 +0900
@@ -52,7 +52,7 @@ configure_file(${PROJECT_SOURCE_DIR}/cma
 
 # We 'export' the main CLHEP library targets from the build tree.
 # This file is include()d by CLHEPConfig.cmake
-export(TARGETS CLHEP CLHEPS 
+export(TARGETS CLHEP
   FILE ${PROJECT_BINARY_DIR}/CLHEPLibraryDepends.cmake
   )
 
@@ -129,10 +129,10 @@ configure_file(${PROJECT_SOURCE_DIR}/cma
 install(FILES
   ${PROJECT_BINARY_DIR}/InstallTreeFiles/CLHEPConfigVersion.cmake 
   ${PROJECT_BINARY_DIR}/InstallTreeFiles/CLHEPConfig.cmake 
-  DESTINATION lib${LIB_SUFFIX}/CLHEP-${VERSION}
+  DESTINATION lib${LIB_SUFFIX}/cmake/CLHEP
   )
 
-install(EXPORT CLHEPLibraryDepends DESTINATION lib${LIB_SUFFIX}/CLHEP-${VERSION})
+install(EXPORT CLHEPLibraryDepends DESTINATION lib${LIB_SUFFIX}/cmake/CLHEP)
 
 # Install the pkg-config file. The choice of 'lib${LIB_SUFFIX}/pkgconfig' for the 
 # installation seems fairly standard.
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/2.2.0.8/CLHEP/cmake/Modules/ClhepVariables.cmake src/2.2.0.8/CLHEP/cmake/Modules/ClhepVariables.cmake
--- origsrc/2.2.0.8/CLHEP/cmake/Modules/ClhepVariables.cmake	2015-06-19 00:52:58.000000000 +0900
+++ src/2.2.0.8/CLHEP/cmake/Modules/ClhepVariables.cmake	2015-08-13 22:39:45.385390000 +0900
@@ -237,12 +237,12 @@ macro( clhep_set_compiler_flags )
   if( ${clhep_c_compiler} STREQUAL "clang" )
     set(CMAKE_C_FLAGS "-O -pedantic -Wall ${CMAKE_C_FLAGS}")
   elseif( CMAKE_COMPILER_IS_GNUCC )
-    set(CMAKE_C_FLAGS "-O -ansi -pedantic -Wall -D_GNU_SOURCE ${CMAKE_C_FLAGS}")
+    set(CMAKE_C_FLAGS "-O -pedantic -Wall -D_GNU_SOURCE ${CMAKE_C_FLAGS}")
   endif()
   if( ${clhep_cxx_compiler} STREQUAL "clang++" )
     set(CMAKE_CXX_FLAGS "-O -pedantic -Wall ${CMAKE_CXX_FLAGS}")
   elseif(CMAKE_COMPILER_IS_GNUCXX)
-    set(CMAKE_CXX_FLAGS "-O -ansi -pedantic -Wall -D_GNU_SOURCE ${CMAKE_CXX_FLAGS}")
+    set(CMAKE_CXX_FLAGS "-O -pedantic -Wall -D_GNU_SOURCE ${CMAKE_CXX_FLAGS}")
   endif()
   if( ${CMAKE_SYSTEM_NAME} MATCHES "Windows" )
     ##message( STATUS "system is Windows" )
@@ -267,7 +267,7 @@ macro( clhep_package_config_no_lib )
   ## don't install <package>-config on Windows
   if( NOT ${CMAKE_SYSTEM_NAME} MATCHES "Windows" )
     install ( PROGRAMS ${CLHEP_BINARY_DIR}/${PACKAGE}/${PACKAGE}-config
-              DESTINATION bin
+              DESTINATION bin RENAME CLHEP-${PACKAGE}-config
 	    )
   endif()
 endmacro( clhep_package_config_no_lib )
@@ -275,11 +275,11 @@ endmacro( clhep_package_config_no_lib )
 macro( clhep_package_config )
   set( ${PACKAGE}_CPPFLAGS "-I${includedir}" )
   set( ${PACKAGE}_LDFLAGS  "-L\${exec_prefix}/lib${LIB_SUFFIX}" )
-  set( ${PACKAGE}_LIBS     "-lCLHEP-${PACKAGE}-${VERSION}" )
+  set( ${PACKAGE}_LIBS     "-lCLHEP-${PACKAGE}" )
   if( ${PACKAGE}_DEPS )
      foreach ( dep ${${PACKAGE}_DEPS} )
         message( STATUS "${PACKAGE} depends on ${dep}")
-	set( ${PACKAGE}_LIBS     "${${PACKAGE}_LIBS} -lCLHEP-${dep}-${VERSION}" )
+	set( ${PACKAGE}_LIBS     "${${PACKAGE}_LIBS} -lCLHEP-${dep}" )
      endforeach()
   endif()
   configure_file ( ${CLHEP_SOURCE_DIR}/${PACKAGE}/${PACKAGE}-config.in
@@ -289,7 +289,7 @@ macro( clhep_package_config )
   ## don't install <package>-config on Windows
   if( NOT ${CMAKE_SYSTEM_NAME} MATCHES "Windows" )
     install ( PROGRAMS ${CLHEP_BINARY_DIR}/${PACKAGE}/${PACKAGE}-config
-              DESTINATION bin
+              DESTINATION bin RENAME CLHEP-${PACKAGE}-config
 	    )
   endif()
 endmacro( clhep_package_config )
