--- origsrc/grepcidr-1.4/Makefile	2013-11-18 13:44:10.000000000 +0900
+++ src/grepcidr-1.4/Makefile	2013-12-08 17:25:41.756009600 +0900
@@ -1,5 +1,5 @@
 DESTDIR=
-PREFIX=/usr/local
+PREFIX=/usr
 DATAROOTDIR=$(PREFIX)/share
 DATADIR=$(DATAROOTDIR)
 EXEC_PREFIX=$(PREFIX)
@@ -9,7 +9,7 @@ MANDIR=$(DATAROOTDIR)/man
 INFODIR=$(DATAROOTDIR)/info
 
 CPPFLAGS=
-CFLAGS=-Wall -Werror
+CFLAGS=-Wall -Werror -O3
 LDFLAGS=
 
 all: grepcidr
