--- origsrc/dpx/libdpx/Makefile.am	2012-04-18 20:23:21.000000000 +0900
+++ src/dpx/libdpx/Makefile.am	2012-04-18 20:24:06.275824400 +0900
@@ -1,7 +1,7 @@
 
-lib_LIBRARIES = libdpx.a
+lib_LTLIBRARIES = libdpx.la
 
-libdpx_a_SOURCES = Codec.cpp \
+libdpx_la_SOURCES = Codec.cpp \
                    DPX.cpp \
                    DPXHeader.cpp \
                    ElementReadStream.cpp \
@@ -12,6 +12,8 @@ libdpx_a_SOURCES = Codec.cpp \
 				   TestFunc.cpp \
                    Writer.cpp
 
+libdpx_la_LDFLAGS = -no-undefined
+
 noinst_HEADERS = BaseTypeConverter.h \
 				 Codec.h \
 				 ElementReadStream.h \
--- origsrc/dpx/tools/dpx2tiff/Makefile.am	2012-04-18 20:23:20.000000000 +0900
+++ src/dpx/tools/dpx2tiff/Makefile.am	2012-04-18 20:24:21.059027500 +0900
@@ -1,7 +1,7 @@
 
-LIBDPX = $(top_builddir)/libdpx/libdpx.a
+LIBDPX = $(top_builddir)/libdpx/libdpx.la
 
-INCLUDES = -I$(top_builddir)/libdpx
+INCLUDES = -I$(top_srcdir)/libdpx
 
 bin_PROGRAMS = dpx2tiff
 
--- origsrc/dpx/tools/dpxheader/Makefile.am	2012-04-18 20:23:20.000000000 +0900
+++ src/dpx/tools/dpxheader/Makefile.am	2012-04-18 20:24:31.843207200 +0900
@@ -1,7 +1,7 @@
 
-LIBDPX = $(top_builddir)/libdpx/libdpx.a
+LIBDPX = $(top_builddir)/libdpx/libdpx.la
 
-INCLUDES = -I$(top_builddir)/libdpx
+INCLUDES = -I$(top_srcdir)/libdpx
 
 bin_PROGRAMS = dpxheader
 
