--- origsrc/dpx/configure.ac	2010-01-10 06:15:33.000000000 +0900
+++ src/dpx/configure.ac	2014-12-27 21:00:12.681576800 +0900
@@ -16,7 +16,7 @@ AM_DISABLE_SHARED
 AM_PROG_LIBTOOL
 
 # Checks for libraries.
-AC_CHECK_LIB(tiff,_TIFFmalloc,have_libtiff='yes',have_libtiff='no')
+PKG_CHECK_MODULES([TIFF], [libtiff-4], [have_libtiff='yes'], [have_libtiff='no'])
 AM_CONDITIONAL(HAVE_LIBTIFF, test "$have_libtiff" != "no")
 
 # Checks for header files.
--- origsrc/dpx/libdpx/Makefile.am	2010-01-10 06:15:33.000000000 +0900
+++ src/dpx/libdpx/Makefile.am	2014-12-27 20:59:07.004321800 +0900
@@ -1,7 +1,7 @@
 
-lib_LIBRARIES = libdpx.a
+lib_LTLIBRARIES = libdpx.la
 
-libdpx_a_SOURCES = Codec.cpp \
+libdpx_la_SOURCES = Codec.cpp \
                    DPX.cpp \
                    DPXHeader.cpp \
                    ElementReadStream.cpp \
@@ -12,6 +12,8 @@ libdpx_a_SOURCES = Codec.cpp \
 				   TestFunc.cpp \
                    Writer.cpp
 
+libdpx_la_LDFLAGS = -no-undefined
+
 noinst_HEADERS = BaseTypeConverter.h \
 				 Codec.h \
 				 ElementReadStream.h \
--- origsrc/dpx/tools/dpx2tiff/Makefile.am	2010-01-10 06:15:33.000000000 +0900
+++ src/dpx/tools/dpx2tiff/Makefile.am	2014-12-27 20:59:07.020498000 +0900
@@ -1,7 +1,7 @@
 
-LIBDPX = $(top_builddir)/libdpx/libdpx.a
+LIBDPX = $(top_builddir)/libdpx/libdpx.la
 
-INCLUDES = -I$(top_builddir)/libdpx
+INCLUDES = -I$(top_srcdir)/libdpx
 
 bin_PROGRAMS = dpx2tiff
 
--- origsrc/dpx/tools/dpxheader/Makefile.am	2010-01-10 06:15:33.000000000 +0900
+++ src/dpx/tools/dpxheader/Makefile.am	2014-12-27 20:59:07.020498000 +0900
@@ -1,7 +1,7 @@
 
-LIBDPX = $(top_builddir)/libdpx/libdpx.a
+LIBDPX = $(top_builddir)/libdpx/libdpx.la
 
-INCLUDES = -I$(top_builddir)/libdpx
+INCLUDES = -I$(top_srcdir)/libdpx
 
 bin_PROGRAMS = dpxheader
 
