--- origsrc/libirecovery-1.1.0/configure.ac	2023-05-13 11:54:38.000000000 +0900
+++ src/libirecovery-1.1.0/configure.ac	2023-08-14 04:16:26.805589700 +0900
@@ -69,13 +69,6 @@ case ${host_os} in
 		GLOBAL_LDFLAGS+=" -static-libgcc -lkernel32 -lsetupapi"
 		win32=true
 	;;
-	cygwin*)
-		AC_MSG_RESULT([${host_os}])
-		CC=gcc-3
-		CFLAGS+=" -mno-cygwin"
-		GLOBAL_LDFLAGS+=" -static-libgcc -lkernel32 -lsetupapi"
-		win32=true
-	;;
 	*)
 		AC_MSG_RESULT([${host_os}])
 	;;
