--- origsrc/libirecovery-0.1.1/configure.ac	2013-09-21 22:12:19.000000000 +0900
+++ src/libirecovery-0.1.1/configure.ac	2014-03-06 07:28:05.839458200 +0900
@@ -48,10 +48,10 @@ case "$host_os" in
 		ACLIBS+="-lkernel32 -lmsvcrt -lsetupapi"
 	;;
 	cygwin*)
-		CC=gcc-3
-		ACCFLAGS+="-O3 -DWIN32 -mno-cygwin -I$(top_srcdir)/include -I/usr/include"
-		ACLDFLAGS+="-static-libgcc -L/lib -L/usr/lib -L/bin"
-		ACLIBS+="-lkernel32 -lmsvcrt -lsetupapi"
+		PKG_CHECK_MODULES([LIBUSB], [libusb-1.0])
+		ACCFLAGS+=$LIBUSB_CFLAGS
+		ACCFLAGS+="-O3 -I$(top_srcdir)/include -I/usr/include"
+		ACLIBS+=$LIBUSB_LIBS
 	;;
 	*)
 		
--- origsrc/libirecovery-0.1.1/src/libirecovery.c	2013-09-21 22:12:19.000000000 +0900
+++ src/libirecovery-0.1.1/src/libirecovery.c	2014-03-06 07:31:45.004210100 +0900
@@ -27,6 +27,7 @@
 #include <libusb-1.0/libusb.h>
 #define _FMT_qX "%qX"
 #define _FMT_016llx "%016llx"
+#undef WIN32
 #else
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
