--- origsrc/libirecovery-1.0.0/configure.ac	2020-06-16 03:18:50.000000000 +0900
+++ src/libirecovery-1.0.0/configure.ac	2020-06-19 15:00:30.471741800 +0900
@@ -66,13 +66,6 @@ case ${host_os} in
 		GLOBAL_LDFLAGS+=" -static-libgcc -lkernel32 -lmsvcrt -lsetupapi"
 		win32=true
 	;;
-	cygwin*)
-		AC_MSG_RESULT([${host_os}])
-		CC=gcc-3
-		CFLAGS+=" -mno-cygwin"
-		GLOBAL_LDFLAGS+=" -static-libgcc -lkernel32 -lmsvcrt -lsetupapi"
-		win32=true
-	;;
 	*)
 		AC_MSG_RESULT([${host_os}])
 		AX_PTHREAD([], [AC_MSG_ERROR([pthread is required to build $PACKAGE])])
--- origsrc/libirecovery-1.0.0/src/libirecovery.c	2020-06-16 03:18:50.000000000 +0900
+++ src/libirecovery-1.0.0/src/libirecovery.c	2020-06-19 15:00:30.487361900 +0900
@@ -36,6 +36,9 @@
 #ifndef WIN32
 #ifndef HAVE_IOKIT
 #include <libusb.h>
+#if defined(__CYGWIN__) && defined(WIN32)
+#undef WIN32
+#endif
 #include <pthread.h>
 #if (defined(LIBUSB_API_VERSION) && (LIBUSB_API_VERSION >= 0x01000102)) || (defined(LIBUSBX_API_VERSION) && (LIBUSBX_API_VERSION >= 0x01000102))
 #define HAVE_LIBUSB_HOTPLUG_API 1
