--- origsrc/flatbuffers-25.9.23/src/util.cpp	2025-09-24 14:18:02.000000000 +0900
+++ src/flatbuffers-25.9.23/src/util.cpp	2025-11-13 18:16:54.127780300 +0900
@@ -18,13 +18,13 @@
 // Dont't remove `format off`, it prevent reordering of win-includes.
 
 #include <cstring>
-#if defined(__MINGW32__) || defined(__MINGW64__) || defined(__CYGWIN__) || \
+#if defined(__MINGW32__) || defined(__MINGW64__) || \
     defined(__QNXNTO__)
 #  define _POSIX_C_SOURCE 200809L
 #  define _XOPEN_SOURCE 700L
 #endif
 
-#if defined(_WIN32) || defined(__MINGW32__) || defined(__MINGW64__) || defined(__CYGWIN__)
+#if defined(_WIN32) || defined(__MINGW32__) || defined(__MINGW64__)
 #  ifndef WIN32_LEAN_AND_MEAN
 #    define WIN32_LEAN_AND_MEAN
 #  endif
@@ -350,7 +350,7 @@ std::string AbsolutePath(const std::stri
   #ifdef FLATBUFFERS_NO_ABSOLUTE_PATH_RESOLUTION
     return filepath;
   #else
-    #if defined(_WIN32) || defined(__MINGW32__) || defined(__MINGW64__) || defined(__CYGWIN__)
+    #if defined(_WIN32) || defined(__MINGW32__) || defined(__MINGW64__)
       char abs_path[MAX_PATH];
       return GetFullPathNameA(filepath.c_str(), MAX_PATH, abs_path, nullptr)
     #else
