--- origsrc/flatbuffers-1.0.3/CMakeLists.txt	2014-10-23 03:58:54.000000000 +0900
+++ src/flatbuffers-1.0.3/CMakeLists.txt	2014-12-21 18:56:32.566000600 +0900
@@ -58,7 +58,7 @@ if(APPLE)
     "${CMAKE_CXX_FLAGS} -std=c++11 -stdlib=libc++ -Wall -pedantic -Werror -Wextra")
 elseif(CMAKE_COMPILER_IS_GNUCXX OR "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
   set(CMAKE_CXX_FLAGS
-    "${CMAKE_CXX_FLAGS} -std=c++0x -Wall -pedantic -Werror -Wextra")
+    "${CMAKE_CXX_FLAGS} -std=gnu++0x -Wall -pedantic -Werror -Wextra")
 endif()
 
 if(FLATBUFFERS_CODE_COVERAGE)
