--- origsrc/flatbuffers-24.3.25/CMakeLists.txt	2024-03-26 14:18:07.000000000 +0900
+++ src/flatbuffers-24.3.25/CMakeLists.txt	2024-04-04 21:21:48.801792400 +0900
@@ -656,7 +656,7 @@ if(FLATBUFFERS_INSTALL)
     install(
       TARGETS flatbuffers_shared EXPORT FlatBuffersSharedTargets
       ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-      RUNTIME DESTINATION ${CMAKE_INSTALL_LIBDIR}
+      RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
       LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
       INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
     )
--- origsrc/flatbuffers-24.3.25/src/util.cpp	2024-03-26 14:18:07.000000000 +0900
+++ src/flatbuffers-24.3.25/src/util.cpp	2024-04-04 21:21:48.817416400 +0900
@@ -18,13 +18,13 @@
 // Dont't remove `format off`, it prevent reordering of win-includes.
 
 #include <cstring>
-#if defined(__MINGW32__) || defined(__MINGW64__) || defined(__CYGWIN__) || \
+#if defined(__MINGW32__) || defined(__MINGW64__) || \
     defined(__QNXNTO__)
 #  define _POSIX_C_SOURCE 200809L
 #  define _XOPEN_SOURCE 700L
 #endif
 
-#if defined(_WIN32) || defined(__MINGW32__) || defined(__MINGW64__) || defined(__CYGWIN__)
+#if defined(_WIN32) || defined(__MINGW32__) || defined(__MINGW64__)
 #  ifndef WIN32_LEAN_AND_MEAN
 #    define WIN32_LEAN_AND_MEAN
 #  endif
@@ -346,7 +346,7 @@ std::string AbsolutePath(const std::stri
   #ifdef FLATBUFFERS_NO_ABSOLUTE_PATH_RESOLUTION
     return filepath;
   #else
-    #if defined(_WIN32) || defined(__MINGW32__) || defined(__MINGW64__) || defined(__CYGWIN__)
+    #if defined(_WIN32) || defined(__MINGW32__) || defined(__MINGW64__)
       char abs_path[MAX_PATH];
       return GetFullPathNameA(filepath.c_str(), MAX_PATH, abs_path, nullptr)
     #else
