--- origsrc/flatbuffers-1.8.0/CMakeLists.txt	2017-11-21 02:46:03.000000000 +0900
+++ src/flatbuffers-1.8.0/CMakeLists.txt	2018-02-22 22:51:47.635141900 +0900
@@ -292,7 +292,7 @@ if(FLATBUFFERS_INSTALL)
     install(
       TARGETS flatbuffers_shared EXPORT FlatbuffersSharedTargets
       ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-      RUNTIME DESTINATION ${CMAKE_INSTALL_LIBDIR}
+      RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
       LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     )
 
@@ -318,3 +318,11 @@ include(CMake/BuildFlatBuffers.cmake)
 if(FLATBUFFERS_PACKAGE_DEBIAN)
     include(CMake/PackageDebian.cmake)
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "1.8.0")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/flatbuffers.pc.in ${CMAKE_BINARY_DIR}/flatbuffers.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/flatbuffers.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/flatbuffers-1.8.0/flatbuffers.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/flatbuffers-1.8.0/flatbuffers.pc.in	2018-02-22 22:51:47.584387000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: flatbuffers
+Description: Memory Efficient Serialization Library
+Version: @VERSION@
+Libs: -L${libdir} -lflatbuffers
+Cflags: -I${includedir}
