--- origsrc/flatbuffers-1.1.0/CMakeLists.txt	2015-03-31 07:38:36.000000000 +0900
+++ src/flatbuffers-1.1.0/CMakeLists.txt	2015-06-18 20:57:09.593092200 +0900
@@ -74,7 +74,7 @@ if(APPLE)
     "${CMAKE_CXX_FLAGS} -std=c++11 -stdlib=libc++ -Wall -pedantic -Werror -Wextra")
 elseif(CMAKE_COMPILER_IS_GNUCXX OR "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
   set(CMAKE_CXX_FLAGS
-    "${CMAKE_CXX_FLAGS} -std=c++0x -Wall -pedantic -Werror -Wextra")
+    "${CMAKE_CXX_FLAGS} -std=gnu++0x -Wall -pedantic -Werror -Wextra")
 endif()
 
 if(FLATBUFFERS_CODE_COVERAGE)
--- origsrc/flatbuffers-1.1.0/src/flathash.cpp	2015-03-31 07:38:36.000000000 +0900
+++ src/flatbuffers-1.1.0/src/flathash.cpp	2015-06-18 20:58:27.757103000 +0900
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+#include <cstdio>
 #include <iostream>
 #include <sstream>
 #include <string>
@@ -99,4 +100,3 @@ int main(int argc, char* argv[]) {
   }
   return 0;
 }
-
