--- origsrc/gerbv-2.6.2/src/pick-and-place.c	2017-02-06 09:18:29.000000000 +0900
+++ src/gerbv-2.6.2/src/pick-and-place.c	2018-06-25 00:42:47.358526700 +0900
@@ -179,6 +179,7 @@ pick_and_place_get_float_unit(char *str)
     } else if(strstr(unit, "mm")) {
 	x /= 25.4;
     } else { /* default to GUI units */
+#if !defined(__CYGWIN__)
 	switch (screen.unit) {
 	case GERBV_INS:
 	    break;
@@ -191,6 +192,7 @@ pick_and_place_get_float_unit(char *str)
 	default:
 	    assert(screen.unit);
 	}
+#endif
     }
 
     return x;
