--- origsrc/gerbv-2.7.0/Makefile.am	2019-01-22 18:00:59.000000000 +0900
+++ src/gerbv-2.7.0/Makefile.am	2019-06-28 22:26:36.778334300 +0900
@@ -25,7 +25,7 @@ SUBDIRS = \
 	share/glib-2.0/schemas \
 	desktop \
 	doc man \
-	intl po \
+	intl \
 	example \
 	test \
 	win32
--- origsrc/gerbv-2.7.0/src/run_gerbv.in	2019-01-22 18:01:00.000000000 +0900
+++ src/gerbv-2.7.0/src/run_gerbv.in	2019-06-29 07:42:13.304350000 +0900
@@ -76,6 +76,12 @@ done
 #
 # This hack came from wcalc.
 
+if test X`uname -o` = XCygwin ; then
+	SONAME=cyggerbv-1.dll
+else
+	SONAME=libgerbv.so
+fi
+
 LIBGERBV_DIR=@BUILDDIR@/.libs
 if test "X${LD_LIBRARY_PATH}" = "X" ; then
 	LD_LIBRARY_PATH=${LIBGERBV_DIR}
@@ -94,9 +100,9 @@ export LD_RUN_PATH
 echo "Set LD_RUN_PATH to ${LD_RUN_PATH}"
 
 if test "X${LD_PRELOAD}" = "X" ; then
-	LD_PRELOAD=${LIBGERBV_DIR}/libgerbv.so
+	LD_PRELOAD=${LIBGERBV_DIR}/${SONAME}
 else
-	LD_PRELOAD=${LIBGERBV_DIR}/libgerbv.so:${LD_PRELOAD}
+	LD_PRELOAD=${LIBGERBV_DIR}/${SONAME}:${LD_PRELOAD}
 fi 
 export LD_PRELOAD
 echo "Set LD_PRELOAD to ${LD_PRELOAD}"
