--- origsrc/gerbv-2.9.6/Makefile.am	2023-01-05 04:31:29.000000000 +0900
+++ src/gerbv-2.9.6/Makefile.am	2023-01-07 14:15:30.950436900 +0900
@@ -27,7 +27,6 @@ SUBDIRS = \
 	share/glib-2.0/schemas \
 	desktop \
 	doc man \
-	po \
 	example \
 	test \
 	win32
--- origsrc/gerbv-2.9.6/configure.ac	2023-01-05 04:31:29.000000000 +0900
+++ src/gerbv-2.9.6/configure.ac	2023-01-07 14:15:31.005854000 +0900
@@ -392,7 +392,9 @@ AC_CHECK_FUNCS(strlwr)
 AC_CHECK_FUNCS(realpath canonicalize_file_name)
 libiberty_NEED_DECLARATION(canonicalize_file_name)
 
-AC_SUBST([GTK_CFLAGS_ISYSTEM], ['$(subst -I/usr/include/gtk-2.0,-isystem /usr/include/gtk-2.0,$(GTK_CFLAGS))'])
+# AC_SUBST([GTK_CFLAGS_ISYSTEM], ['$(subst -I/usr/include/gtk-2.0,-isystem /usr/include/gtk-2.0,$(GTK_CFLAGS))'])
+GTK_CFLAGS_ISYSTEM="${GTK_CFLAGS}"
+AC_SUBST(GTK_CFLAGS_ISYSTEM)
 
 CFLAGS="$CFLAGS $GDK_PIXBUF_CFLAGS $GTK_CFLAGS_ISYSTEM $CAIRO_CFLAGS"
 LIBS="$LIBS $GDK_PIXBUF_LIBS $GTK_LIBS $CAIRO_LIBS -lm"
--- origsrc/gerbv-2.9.6/src/run_gerbv.in	2023-01-05 04:31:29.000000000 +0900
+++ src/gerbv-2.9.6/src/run_gerbv.in	2023-01-07 14:15:31.174284900 +0900
@@ -84,6 +84,12 @@ done
 #
 # This hack came from wcalc.
 
+if test X`uname -o` = XCygwin ; then
+	SONAME=cyggerbv-1.dll
+else
+	SONAME=libgerbv.so
+fi
+
 LIBGERBV_DIR=@BUILDDIR@/.libs
 if test "X${LD_LIBRARY_PATH}" = "X" ; then
 	LD_LIBRARY_PATH=${LIBGERBV_DIR}
@@ -102,9 +108,9 @@ export LD_RUN_PATH
 echo "Set LD_RUN_PATH to ${LD_RUN_PATH}"
 
 if test "X${LD_PRELOAD}" = "X" ; then
-	LD_PRELOAD=${LIBGERBV_DIR}/libgerbv.so
+	LD_PRELOAD=${LIBGERBV_DIR}/${SONAME}
 else
-	LD_PRELOAD=${LIBGERBV_DIR}/libgerbv.so:${LD_PRELOAD}
+	LD_PRELOAD=${LIBGERBV_DIR}/${SONAME}:${LD_PRELOAD}
 fi 
 export LD_PRELOAD
 echo "Set LD_PRELOAD to ${LD_PRELOAD}"
--- origsrc/gerbv-2.9.6/utils/git-version-gen.sh	2023-01-05 04:31:29.000000000 +0900
+++ src/gerbv-2.9.6/utils/git-version-gen.sh	2023-01-07 14:15:31.191247900 +0900
@@ -19,6 +19,10 @@ if [ "" == "${PREFIX}" ]; then
 	exit 1
 fi
 
+if [ ! -d ".git" ]; then
+	echo -n "${PREFIX}"
+	exit 0
+fi
 
 # Validate environment
 GIT=`command -v git`
