--- origsrc/libocrdma-1.0.8/Makefile.am	2016-04-14 15:33:45.000000000 +0900
+++ src/libocrdma-1.0.8/Makefile.am	2025-01-29 15:41:10.827471800 +0900
@@ -10,7 +10,7 @@ else
 endif
 
 src_libocrdma_la_SOURCES = src/ocrdma_main.c src/ocrdma_verbs.c
-src_libocrdma_la_LDFLAGS = -avoid-version -module -release @IBV_DEVICE_LIBRARY_EXTENSION@\
+src_libocrdma_la_LDFLAGS = -no-undefined -avoid-version -release @IBV_DEVICE_LIBRARY_EXTENSION@\
     $(ocrdma_version_script)
 
 ocrdmaconfdir = $(sysconfdir)/libibverbs.d
