--- origsrc/Field3D-1.7.2/CMakeLists.txt	2016-06-19 13:56:00.000000000 +0900
+++ src/Field3D-1.7.2/CMakeLists.txt	2020-02-03 19:00:57.017672100 +0900
@@ -75,7 +75,10 @@ LINK_DIRECTORIES ( ${HDF5_LIBRARY_DIRS}
 LINK_DIRECTORIES ( ${ILMBASE_LIBRARY_DIRS} )
 
 IF ( CMAKE_HOST_UNIX )
-  ADD_DEFINITIONS ( -fPIC -DREQUIRE_IOSTREAM -Wno-invalid-offsetof )
+  ADD_DEFINITIONS ( -DREQUIRE_IOSTREAM -Wno-invalid-offsetof )
+  IF ( NOT CYGWIN )
+    ADD_DEFINITIONS ( -fPIC )
+  ENDIF ( )
 ENDIF ( )
 IF ( CMAKE_HOST_WIN32 )
   ADD_DEFINITIONS (
@@ -149,7 +152,8 @@ IF ( CMAKE_HOST_UNIX )
   LIST ( APPEND Field3D_Libraries_Shared
     Iex Half IlmThread Imath
     pthread dl z )
-  SET ( Field3D_DSO_Libraries ${Field3D_Libraries_Shared} )
+  SET ( Field3D_DSO_Libraries ${Field3D_Libraries_Shared}
+        ${Boost_LIBRARIES} )
   SET ( Field3D_BIN_Libraries Field3D ${Field3D_Libraries_Shared}
         ${Boost_LIBRARIES} )
 ENDIF ( )
@@ -242,23 +246,22 @@ ADD_EXECUTABLE ( create_and_write
 
 TARGET_LINK_LIBRARIES ( create_and_write ${Field3D_BIN_Libraries} )
 
-IF (DOXYGEN_FOUND)
+IF (INSTALL_DOCS AND DOXYGEN_FOUND)
   ADD_CUSTOM_TARGET ( doc
     ALL
     ${DOXYGEN_EXECUTABLE} Field3D.doxyfile
     WORKING_DIRECTORY ${CMAKE_HOME_DIRECTORY}
     )
-  IF (INSTALL_DOCS)
     INSTALL (DIRECTORY 
       ${CMAKE_HOME_DIRECTORY}/docs
       DESTINATION ${CMAKE_INSTALL_PREFIX}
     )
-  ENDIF (INSTALL_DOCS)
-ENDIF (DOXYGEN_FOUND)
+ENDIF (INSTALL_DOCS AND DOXYGEN_FOUND)
 
 INSTALL ( TARGETS
   Field3D
-  DESTINATION lib${LIB_SUFFIX}
+  RUNTIME DESTINATION bin
+  ARCHIVE DESTINATION lib${LIB_SUFFIX}
 )
 
 FILE(GLOB Field3d_Includes  "${CMAKE_CURRENT_SOURCE_DIR}/export/*.h")
@@ -272,3 +275,9 @@ INSTALL ( TARGETS f3dinfo
   RUNTIME DESTINATION bin
 )
 
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/Field3D.pc.in ${CMAKE_BINARY_DIR}/Field3D.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/Field3D.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/Field3D-1.7.2/Field3D.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/Field3D-1.7.2/Field3D.pc.in	2020-02-03 19:00:57.048940400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: Field3D
+Description: Library for storing voxel data on disk and in memory
+Version: @VERSION@
+Libs: -L${libdir} -lField3D
+Cflags: -I${includedir}
