--- origsrc/judy-1.0.5/doc/Makefile.am	2007-05-22 23:38:41.000000000 +0900
+++ src/judy-1.0.5/doc/Makefile.am	2014-07-06 18:14:46.756989600 +0900
@@ -96,10 +96,12 @@ man3_MANS = man/man3/Judy \
 
 
 man/man3/Judy: 
-	../tool/jhton ext/Judy_3.htm | grep -v '^[   ]*$$' | sed -e 's/\.C//' >  man/man3/Judy
+	mkdir -p man/man3
+	../tool/jhton $(top_srcdir)/doc/ext/Judy_3.htm | grep -v '^[   ]*$$' | sed -e 's/\.C//' >  man/man3/Judy
 
 man/man3/Judy1:
-	../tool/jhton ext/Judy1_3.htm | grep -v '^[   ]*$$' | sed -e 's/\.C//' > man/man3/Judy1
+	mkdir -p man/man3
+	../tool/jhton $(top_srcdir)/doc/ext/Judy1_3.htm | grep -v '^[   ]*$$' | sed -e 's/\.C//' > man/man3/Judy1
 	cd man/man3; ln -s Judy J1T
 	cd man/man3; ln -s Judy J1S
 	cd man/man3; ln -s Judy J1U
@@ -117,7 +119,7 @@ man/man3/Judy1:
 	cd man/man3; ln -s Judy J1MU
 
 man/man3/Judy1_funcs:
-	../tool/jhton ext/Judy1_funcs_3.htm | grep -v '^[   ]*$$' | sed -e 's/\.C//' > man/man3/Judy1_funcs
+	../tool/jhton $(top_srcdir)/doc/ext/Judy1_funcs_3.htm | grep -v '^[   ]*$$' | sed -e 's/\.C//' > man/man3/Judy1_funcs
 	cd man/man3; ln -s Judy1_funcs Judy1Test 
 	cd man/man3; ln -s Judy1_funcs Judy1Set 
 	cd man/man3; ln -s Judy1_funcs Judy1Unset 
@@ -135,7 +137,8 @@ man/man3/Judy1_funcs:
 	cd man/man3; ln -s Judy1_funcs Judy1MemUsed
 
 man/man3/JudyL:
-	../tool/jhton ext/JudyL_3.htm | grep -v '^[   ]*$$' | sed -e 's/\.C//' > man/man3/JudyL
+	mkdir -p man/man3
+	../tool/jhton $(top_srcdir)/doc/ext/JudyL_3.htm | grep -v '^[   ]*$$' | sed -e 's/\.C//' > man/man3/JudyL
 	cd man/man3; ln -s JudyL JLG
 	cd man/man3; ln -s JudyL JLI
 	cd man/man3; ln -s JudyL JLD
@@ -153,6 +156,7 @@ man/man3/JudyL:
 	cd man/man3; ln -s JudyL JLMU
 
 man/man3/JudyL_funcs:
+	mkdir -p man/man3
 	../tool/jhton ext/JudyL_funcs_3.htm | grep -v '^[   ]*$$' | sed -e 's/\.C//' > man/man3/JudyL_funcs
 	cd man/man3; ln -s JudyL_funcs JudyLGet
 	cd man/man3; ln -s JudyL_funcs JudyLIns
@@ -171,7 +175,7 @@ man/man3/JudyL_funcs:
 	cd man/man3; ln -s JudyL_funcs JudyLMemUsed
 
 man/man3/JudySL:
-	../tool/jhton ext/JudySL_3.htm | grep -v '^[   ]*$$' | sed -e 's/\.C//' > man/man3/JudySL
+	../tool/jhton $(top_srcdir)/doc/ext/JudySL_3.htm | grep -v '^[   ]*$$' | sed -e 's/\.C//' > man/man3/JudySL
 	cd man/man3; ln -s JudySL JSLG
 	cd man/man3; ln -s JudySL JSLI
 	cd man/man3; ln -s JudySL JSLD
@@ -182,7 +186,8 @@ man/man3/JudySL:
 	cd man/man3; ln -s JudySL JSLFA
 
 man/man3/JudySL_funcs:
-	../tool/jhton ext/JudySL_funcs_3.htm | grep -v '^[   ]*$$' | sed -e 's/\.C//' > man/man3/JudySL_funcs
+	mkdir -p man/man3
+	../tool/jhton $(top_srcdir)/doc/ext/JudySL_funcs_3.htm | grep -v '^[   ]*$$' | sed -e 's/\.C//' > man/man3/JudySL_funcs
 	cd man/man3; ln -s JudySL_funcs JudySLGet
 	cd man/man3; ln -s JudySL_funcs JudySLIns
 	cd man/man3; ln -s JudySL_funcs JudySLDel
@@ -193,14 +198,16 @@ man/man3/JudySL_funcs:
 	cd man/man3; ln -s JudySL_funcs JudySLFreeArray
 
 man/man3/JudyHS:
-	../tool/jhton ext/JudyHS_3.htm | grep -v '^[   ]*$$' | sed -e 's/\.C//' > man/man3/JudyHS
+	mkdir -p man/man3
+	../tool/jhton $(top_srcdir)/doc/ext/JudyHS_3.htm | grep -v '^[   ]*$$' | sed -e 's/\.C//' > man/man3/JudyHS
 	cd man/man3; ln -s JudyHS JHSG 
 	cd man/man3; ln -s JudyHS JHSI
 	cd man/man3; ln -s JudyHS JHSD
 	cd man/man3; ln -s JudyHS JHSFA
 
 man/man3/JudyHS_funcs:
-	../tool/jhton ext/JudyHS_funcs_3.htm | grep -v '^[   ]*$$' | sed -e 's/\.C//' > man/man3/JudyHS_funcs
+	mkdir -p man/man3
+	../tool/jhton $(top_srcdir)/doc/ext/JudyHS_funcs_3.htm | grep -v '^[   ]*$$' | sed -e 's/\.C//' > man/man3/JudyHS_funcs
 	cd man/man3; ln -s JudyHS_funcs JudyHSGet
 	cd man/man3; ln -s JudyHS_funcs JudyHSIns
 	cd man/man3; ln -s JudyHS_funcs JudyHSDel
--- origsrc/judy-1.0.5/src/Judy1/Makefile.am	2007-05-22 23:38:41.000000000 +0900
+++ src/judy-1.0.5/src/Judy1/Makefile.am	2014-07-06 17:09:22.480773500 +0900
@@ -1,4 +1,4 @@
-INCLUDES =  -I. -I.. -I../JudyCommon/ 
+INCLUDES =  -I. -I.. -I../JudyCommon/ -I$(top_srcdir)/src/JudyCommon -I$(top_srcdir)/src -I$(top_srcdir)/src/Judy1
 AM_CFLAGS = -DJUDY1 @WARN_CFLAGS@ 
 
 noinst_LTLIBRARIES = libJudy1.la libnext.la libprev.la libcount.la libinline.la
@@ -20,68 +20,68 @@ libinline_la_CFLAGS = $(AM_CFLAGS) -DJUD
 Judy1Tables.c: Judy1TablesGen.c
 	$(CC) $(INCLUDES) $(AM_CFLAGS) @CFLAGS@ -o Judy1TablesGen Judy1TablesGen.c; ./Judy1TablesGen
 
-Judy1ByCount.c:../JudyCommon/JudyByCount.c
-	cp -f ../JudyCommon/JudyByCount.c      		Judy1ByCount.c   
+Judy1ByCount.c:$(top_srcdir)/src/JudyCommon/JudyByCount.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyByCount.c      		Judy1ByCount.c   
 
-Judy1Cascade.c:../JudyCommon/JudyCascade.c
-	cp -f ../JudyCommon/JudyCascade.c      		Judy1Cascade.c
+Judy1Cascade.c:$(top_srcdir)/src/JudyCommon/JudyCascade.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyCascade.c      		Judy1Cascade.c
 
-Judy1Count.c:../JudyCommon/JudyCount.c
-	cp -f ../JudyCommon/JudyCount.c        		Judy1Count.c
+Judy1Count.c:$(top_srcdir)/src/JudyCommon/JudyCount.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyCount.c        		Judy1Count.c
 
-Judy1CreateBranch.c:../JudyCommon/JudyCreateBranch.c
-	cp -f ../JudyCommon/JudyCreateBranch.c 		Judy1CreateBranch.c
+Judy1CreateBranch.c:$(top_srcdir)/src/JudyCommon/JudyCreateBranch.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyCreateBranch.c 		Judy1CreateBranch.c
 
-Judy1Decascade.c:../JudyCommon/JudyDecascade.c
-	cp -f ../JudyCommon/JudyDecascade.c    		Judy1Decascade.c
+Judy1Decascade.c:$(top_srcdir)/src/JudyCommon/JudyDecascade.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyDecascade.c    		Judy1Decascade.c
 
-Judy1Unset.c:../JudyCommon/JudyDel.c
-	cp -f ../JudyCommon/JudyDel.c          		Judy1Unset.c
+Judy1Unset.c:$(top_srcdir)/src/JudyCommon/JudyDel.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyDel.c          		Judy1Unset.c
 
-Judy1First.c:../JudyCommon/JudyFirst.c
-	cp -f ../JudyCommon/JudyFirst.c        		Judy1First.c
+Judy1First.c:$(top_srcdir)/src/JudyCommon/JudyFirst.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyFirst.c        		Judy1First.c
 
-Judy1FreeArray.c:../JudyCommon/JudyFreeArray.c
-	cp -f ../JudyCommon/JudyFreeArray.c    		Judy1FreeArray.c
+Judy1FreeArray.c:$(top_srcdir)/src/JudyCommon/JudyFreeArray.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyFreeArray.c    		Judy1FreeArray.c
 
-Judy1Test.c:../JudyCommon/JudyGet.c
-	cp -f ../JudyCommon/JudyGet.c          		Judy1Test.c
+Judy1Test.c:$(top_srcdir)/src/JudyCommon/JudyGet.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyGet.c          		Judy1Test.c
 
-j__udy1Test.c:../JudyCommon/JudyGet.c
-	cp -f ../JudyCommon/JudyGet.c          		j__udy1Test.c
+j__udy1Test.c:$(top_srcdir)/src/JudyCommon/JudyGet.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyGet.c          		j__udy1Test.c
 
-Judy1SetArray.c:../JudyCommon/JudyInsArray.c
-	cp -f ../JudyCommon/JudyInsArray.c     		Judy1SetArray.c
+Judy1SetArray.c:$(top_srcdir)/src/JudyCommon/JudyInsArray.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyInsArray.c     		Judy1SetArray.c
 
-Judy1Set.c:../JudyCommon/JudyIns.c
-	cp -f ../JudyCommon/JudyIns.c          		Judy1Set.c
+Judy1Set.c:$(top_srcdir)/src/JudyCommon/JudyIns.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyIns.c          		Judy1Set.c
 
-Judy1InsertBranch.c:../JudyCommon/JudyInsertBranch.c
-	cp -f ../JudyCommon/JudyInsertBranch.c 		Judy1InsertBranch.c
+Judy1InsertBranch.c:$(top_srcdir)/src/JudyCommon/JudyInsertBranch.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyInsertBranch.c 		Judy1InsertBranch.c
 
-Judy1MallocIF.c:../JudyCommon/JudyMallocIF.c
-	cp -f ../JudyCommon/JudyMallocIF.c     		Judy1MallocIF.c
+Judy1MallocIF.c:$(top_srcdir)/src/JudyCommon/JudyMallocIF.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyMallocIF.c     		Judy1MallocIF.c
 
-Judy1MemActive.c:../JudyCommon/JudyMemActive.c
-	cp -f ../JudyCommon/JudyMemActive.c    		Judy1MemActive.c
+Judy1MemActive.c:$(top_srcdir)/src/JudyCommon/JudyMemActive.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyMemActive.c    		Judy1MemActive.c
 
-Judy1MemUsed.c:../JudyCommon/JudyMemUsed.c
-	cp -f ../JudyCommon/JudyMemUsed.c      		Judy1MemUsed.c
+Judy1MemUsed.c:$(top_srcdir)/src/JudyCommon/JudyMemUsed.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyMemUsed.c      		Judy1MemUsed.c
 
-Judy1Next.c:../JudyCommon/JudyPrevNext.c
-	cp -f ../JudyCommon/JudyPrevNext.c     		Judy1Next.c
+Judy1Next.c:$(top_srcdir)/src/JudyCommon/JudyPrevNext.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyPrevNext.c     		Judy1Next.c
 
-Judy1Prev.c:../JudyCommon/JudyPrevNext.c
-	cp -f ../JudyCommon/JudyPrevNext.c     		Judy1Prev.c
+Judy1Prev.c:$(top_srcdir)/src/JudyCommon/JudyPrevNext.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyPrevNext.c     		Judy1Prev.c
 
-Judy1NextEmpty.c:../JudyCommon/JudyPrevNextEmpty.c
-	cp -f ../JudyCommon/JudyPrevNextEmpty.c		Judy1NextEmpty.c
+Judy1NextEmpty.c:$(top_srcdir)/src/JudyCommon/JudyPrevNextEmpty.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyPrevNextEmpty.c		Judy1NextEmpty.c
 
-Judy1PrevEmpty.c:../JudyCommon/JudyPrevNextEmpty.c
-	cp -f ../JudyCommon/JudyPrevNextEmpty.c		Judy1PrevEmpty.c
+Judy1PrevEmpty.c:$(top_srcdir)/src/JudyCommon/JudyPrevNextEmpty.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyPrevNextEmpty.c		Judy1PrevEmpty.c
 
-Judy1TablesGen.c:../JudyCommon/JudyTables.c
-	cp -f ../JudyCommon/JudyTables.c	        Judy1TablesGen.c
+Judy1TablesGen.c:$(top_srcdir)/src/JudyCommon/JudyTables.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyTables.c	        Judy1TablesGen.c
 
 DISTCLEANFILES = .deps Makefile 
 
--- origsrc/judy-1.0.5/src/JudyCommon/Makefile.am	2007-05-22 23:38:41.000000000 +0900
+++ src/judy-1.0.5/src/JudyCommon/Makefile.am	2014-07-06 16:54:57.475890700 +0900
@@ -1,4 +1,4 @@
-INCLUDES =  -I. -I..
+INCLUDES =  -I. -I.. -I$(top_srcdir)/src
 AM_CFLAGS = @CFLAGS@ @WARN_CFLAGS@ 
 
 noinst_LTLIBRARIES = libJudyMalloc.la
--- origsrc/judy-1.0.5/src/JudyHS/Makefile.am	2007-05-22 23:38:41.000000000 +0900
+++ src/judy-1.0.5/src/JudyHS/Makefile.am	2014-07-06 18:11:20.801911500 +0900
@@ -1,4 +1,4 @@
-INCLUDES =  -I. -I.. -I../JudyCommon/ 
+INCLUDES =  -I. -I.. -I../JudyCommon/ -I$(top_srcdir)/src/JudyCommon -I$(top_srcdir)/src
 AM_CFLAGS = @CFLAGS@ @WARN_CFLAGS@ 
 
 noinst_LTLIBRARIES = libJudyHS.la
--- origsrc/judy-1.0.5/src/JudyL/Makefile.am	2007-05-22 23:38:41.000000000 +0900
+++ src/judy-1.0.5/src/JudyL/Makefile.am	2014-07-06 17:01:46.774718800 +0900
@@ -1,4 +1,4 @@
-INCLUDES =  -I. -I.. -I../JudyCommon/ 
+INCLUDES =  -I. -I.. -I../JudyCommon/ -I$(top_srcdir)/src/JudyCommon -I$(top_srcdir)/src -I$(top_srcdir)/src/JudyL
 AM_CFLAGS = -DJUDYL @WARN_CFLAGS@ 
 
 noinst_LTLIBRARIES = libJudyL.la libnext.la libprev.la libcount.la libinline.la
@@ -20,68 +20,68 @@ libinline_la_CFLAGS = $(AM_CFLAGS) -DJUD
 JudyLTables.c: JudyLTablesGen.c
 	$(CC) $(INCLUDES) $(AM_CFLAGS) @CFLAGS@ -o JudyLTablesGen JudyLTablesGen.c; ./JudyLTablesGen 
 
-JudyLByCount.c:   ../JudyCommon/JudyByCount.c
-	cp -f ../JudyCommon/JudyByCount.c      		JudyLByCount.c   
+JudyLByCount.c:   $(top_srcdir)/src/JudyCommon/JudyByCount.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyByCount.c      		JudyLByCount.c   
 
-JudyLCascade.c:../JudyCommon/JudyCascade.c
-	cp -f ../JudyCommon/JudyCascade.c       	JudyLCascade.c
+JudyLCascade.c:$(top_srcdir)/src/JudyCommon/JudyCascade.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyCascade.c       	JudyLCascade.c
 
-JudyLCount.c:../JudyCommon/JudyCount.c
-	cp -f ../JudyCommon/JudyCount.c        		JudyLCount.c
+JudyLCount.c:$(top_srcdir)/src/JudyCommon/JudyCount.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyCount.c        		JudyLCount.c
 
-JudyLCreateBranch.c:../JudyCommon/JudyCreateBranch.c 
-	cp -f ../JudyCommon/JudyCreateBranch.c 		JudyLCreateBranch.c
+JudyLCreateBranch.c:$(top_srcdir)/src/JudyCommon/JudyCreateBranch.c 
+	cp -f $(top_srcdir)/src/JudyCommon/JudyCreateBranch.c 		JudyLCreateBranch.c
 
-JudyLDecascade.c:../JudyCommon/JudyDecascade.c    
-	cp -f ../JudyCommon/JudyDecascade.c    		JudyLDecascade.c
+JudyLDecascade.c:$(top_srcdir)/src/JudyCommon/JudyDecascade.c    
+	cp -f $(top_srcdir)/src/JudyCommon/JudyDecascade.c    		JudyLDecascade.c
 
-JudyLDel.c:../JudyCommon/JudyDel.c       
-	cp -f ../JudyCommon/JudyDel.c          		JudyLDel.c
+JudyLDel.c:$(top_srcdir)/src/JudyCommon/JudyDel.c       
+	cp -f $(top_srcdir)/src/JudyCommon/JudyDel.c          		JudyLDel.c
 
-JudyLFirst.c:../JudyCommon/JudyFirst.c
-	cp -f ../JudyCommon/JudyFirst.c        		JudyLFirst.c
+JudyLFirst.c:$(top_srcdir)/src/JudyCommon/JudyFirst.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyFirst.c        		JudyLFirst.c
 
-JudyLFreeArray.c:../JudyCommon/JudyFreeArray.c    
-	cp -f ../JudyCommon/JudyFreeArray.c    		JudyLFreeArray.c
+JudyLFreeArray.c:$(top_srcdir)/src/JudyCommon/JudyFreeArray.c    
+	cp -f $(top_srcdir)/src/JudyCommon/JudyFreeArray.c    		JudyLFreeArray.c
 
-JudyLGet.c:../JudyCommon/JudyGet.c       
-	cp -f ../JudyCommon/JudyGet.c          		JudyLGet.c
+JudyLGet.c:$(top_srcdir)/src/JudyCommon/JudyGet.c       
+	cp -f $(top_srcdir)/src/JudyCommon/JudyGet.c          		JudyLGet.c
 
-j__udyLGet.c:../JudyCommon/JudyGet.c 
-	cp -f ../JudyCommon/JudyGet.c          		j__udyLGet.c
+j__udyLGet.c:$(top_srcdir)/src/JudyCommon/JudyGet.c 
+	cp -f $(top_srcdir)/src/JudyCommon/JudyGet.c          		j__udyLGet.c
 
-JudyLInsArray.c:../JudyCommon/JudyInsArray.c     
-	cp -f ../JudyCommon/JudyInsArray.c     		JudyLInsArray.c
+JudyLInsArray.c:$(top_srcdir)/src/JudyCommon/JudyInsArray.c     
+	cp -f $(top_srcdir)/src/JudyCommon/JudyInsArray.c     		JudyLInsArray.c
 
-JudyLIns.c:../JudyCommon/JudyIns.c      
-	cp -f ../JudyCommon/JudyIns.c          		JudyLIns.c
+JudyLIns.c:$(top_srcdir)/src/JudyCommon/JudyIns.c      
+	cp -f $(top_srcdir)/src/JudyCommon/JudyIns.c          		JudyLIns.c
 
-JudyLInsertBranch.c:../JudyCommon/JudyInsertBranch.c
-	cp -f ../JudyCommon/JudyInsertBranch.c 		JudyLInsertBranch.c
+JudyLInsertBranch.c:$(top_srcdir)/src/JudyCommon/JudyInsertBranch.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyInsertBranch.c 		JudyLInsertBranch.c
 
-JudyLMallocIF.c:../JudyCommon/JudyMallocIF.c     
-	cp -f ../JudyCommon/JudyMallocIF.c     		JudyLMallocIF.c
+JudyLMallocIF.c:$(top_srcdir)/src/JudyCommon/JudyMallocIF.c     
+	cp -f $(top_srcdir)/src/JudyCommon/JudyMallocIF.c     		JudyLMallocIF.c
 
-JudyLMemActive.c:../JudyCommon/JudyMemActive.c
-	cp -f ../JudyCommon/JudyMemActive.c    		JudyLMemActive.c
+JudyLMemActive.c:$(top_srcdir)/src/JudyCommon/JudyMemActive.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyMemActive.c    		JudyLMemActive.c
 
-JudyLMemUsed.c:../JudyCommon/JudyMemUsed.c   
-	cp -f ../JudyCommon/JudyMemUsed.c      		JudyLMemUsed.c
+JudyLMemUsed.c:$(top_srcdir)/src/JudyCommon/JudyMemUsed.c   
+	cp -f $(top_srcdir)/src/JudyCommon/JudyMemUsed.c      		JudyLMemUsed.c
 
-JudyLNext.c:../JudyCommon/JudyPrevNext.c
-	cp -f ../JudyCommon/JudyPrevNext.c     		JudyLNext.c
+JudyLNext.c:$(top_srcdir)/src/JudyCommon/JudyPrevNext.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyPrevNext.c     		JudyLNext.c
 
-JudyLPrev.c:../JudyCommon/JudyPrevNext.c
-	cp -f ../JudyCommon/JudyPrevNext.c     		JudyLPrev.c
+JudyLPrev.c:$(top_srcdir)/src/JudyCommon/JudyPrevNext.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyPrevNext.c     		JudyLPrev.c
 
-JudyLNextEmpty.c:../JudyCommon/JudyPrevNextEmpty.c
-	cp -f ../JudyCommon/JudyPrevNextEmpty.c		JudyLNextEmpty.c
+JudyLNextEmpty.c:$(top_srcdir)/src/JudyCommon/JudyPrevNextEmpty.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyPrevNextEmpty.c		JudyLNextEmpty.c
 
-JudyLPrevEmpty.c:../JudyCommon/JudyPrevNextEmpty.c
-	cp -f ../JudyCommon/JudyPrevNextEmpty.c		JudyLPrevEmpty.c
+JudyLPrevEmpty.c:$(top_srcdir)/src/JudyCommon/JudyPrevNextEmpty.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyPrevNextEmpty.c		JudyLPrevEmpty.c
 	
-JudyLTablesGen.c:../JudyCommon/JudyTables.c
-	cp -f ../JudyCommon/JudyTables.c       	 	JudyLTablesGen.c
+JudyLTablesGen.c:$(top_srcdir)/src/JudyCommon/JudyTables.c
+	cp -f $(top_srcdir)/src/JudyCommon/JudyTables.c       	 	JudyLTablesGen.c
 
 DISTCLEANFILES = .deps Makefile 
 
--- origsrc/judy-1.0.5/src/JudySL/Makefile.am	2007-05-22 23:38:41.000000000 +0900
+++ src/judy-1.0.5/src/JudySL/Makefile.am	2014-07-06 18:10:51.440583300 +0900
@@ -1,4 +1,4 @@
-INCLUDES =  -I. -I.. -I../JudyCommon/ 
+INCLUDES =  -I. -I.. -I../JudyCommon/ -I$(top_srcdir)/src/JudyCommon -I$(top_srcdir)/src
 AM_CFLAGS = @CFLAGS@ @WARN_CFLAGS@ 
 
 noinst_LTLIBRARIES = libJudySL.la
--- origsrc/judy-1.0.5/src/obj/Makefile.am	2007-05-22 23:38:41.000000000 +0900
+++ src/judy-1.0.5/src/obj/Makefile.am	2014-07-06 16:49:56.175785900 +0900
@@ -10,6 +10,6 @@ libJudy_la_SOURCES =
 
 libJudy_la_LIBADD = ../JudyCommon/*.lo ../JudyL/*.lo ../Judy1/*.lo ../JudyHS/*.lo ../JudySL/*.lo 
 
-libJudy_la_LDFLAGS = @VERSION_INFO@
+libJudy_la_LDFLAGS = -no-undefined @VERSION_INFO@
 
 DISTCLEANFILES = .deps Makefile
