--- origsrc/libbacklight/Makefile.am	2025-03-16 10:28:17.000000000 +0900
+++ src/libbacklight/Makefile.am	2025-03-16 10:28:18.405011200 +0900
@@ -3,7 +3,8 @@ dist_doc_DATA = README COPYING
 include_HEADERS = libbacklight.h
 lib_LTLIBRARIES = libbacklight.la
 libbacklight_la_SOURCES = libbacklight.c libbacklight.h
-libatasmart_la_CFLAGS = $(PCIACCESS_CFLAGS)
+libbacklight_la_CFLAGS = $(PCIACCESS_CFLAGS)
+libbacklight_la_LDFLAGS = -no-undefined
 
 ACLOCAL_AMFLAGS = -I m4
 
--- origsrc/libbacklight/libbacklight.c	2025-03-16 10:28:17.000000000 +0900
+++ src/libbacklight/libbacklight.c	2025-03-16 10:28:18.405011200 +0900
@@ -32,10 +32,12 @@
 #include <libbacklight.h>
 #include <stdio.h>
 #include <sys/types.h>
+#if defined(linux)
 #include <linux/types.h>
+#endif
 #include <dirent.h>
 #include <sys/stat.h>
-#include <drm/drm_mode.h>
+#include <libdrm/drm_mode.h>
 #include <fcntl.h>
 #include <malloc.h>
 #include <string.h>
