--- origsrc/pimmel-0.1.0/src/Makefile.am	2013-01-17 00:05:50.000000000 +0900
+++ src/pimmel-0.1.0/src/Makefile.am	2024-09-26 16:57:59.359755000 +0900
@@ -23,13 +23,13 @@ libpimmel_la_CPPFLAGS = $(AM_CPPFLAGS)
 libpimmel_la_CFLAGS = -fPIC
 libpimmel_la_LDFLAGS = $(AM_LDFLAGS)
 libpimmel_la_LDFLAGS += $(XCCLDFLAGS)
-libpimmel_la_LDFLAGS += -version-info 0:0:0
+libpimmel_la_LDFLAGS += -no-undefined -version-info 0:0:0
 
 bin_PROGRAMS += pimmel-noti
 pimmel_noti_SOURCES = pimmel-noti.c pimmel-noti-clo.ggo
 pimmel_noti_CPPFLAGS = $(AM_CPPFLAGS)
 pimmel_noti_CPPFLAGS += $(libev_CFLAGS)
-pimmel_noti_LDFLAGS = $(AM_LDFLAGS) -static
+pimmel_noti_LDFLAGS = $(AM_LDFLAGS)
 pimmel_noti_LDFLAGS += $(libev_LIBS)
 pimmel_noti_LDADD = libpimmel.la
 BUILT_SOURCES += pimmel-noti-clo.c pimmel-noti-clo.h
@@ -38,7 +38,7 @@ bin_PROGRAMS += pimmel-wait
 pimmel_wait_SOURCES = pimmel-wait.c pimmel-wait-clo.ggo
 pimmel_wait_CPPFLAGS = $(AM_CPPFLAGS)
 pimmel_wait_CPPFLAGS += $(libev_CFLAGS)
-pimmel_wait_LDFLAGS = $(AM_LDFLAGS) -static
+pimmel_wait_LDFLAGS = $(AM_LDFLAGS)
 pimmel_wait_LDFLAGS += $(libev_LIBS)
 pimmel_wait_LDADD = libpimmel.la
 BUILT_SOURCES += pimmel-wait-clo.c pimmel-wait-clo.h
@@ -48,7 +48,7 @@ pimmel_router_SOURCES = pimmel-router.c
 pimmel_router_SOURCES += daemonise.c daemonise.h
 pimmel_router_CPPFLAGS = $(AM_CPPFLAGS)
 pimmel_router_CPPFLAGS += $(libev_CFLAGS)
-pimmel_router_LDFLAGS = $(AM_LDFLAGS) -static
+pimmel_router_LDFLAGS = $(AM_LDFLAGS)
 pimmel_router_LDFLAGS += $(libev_LIBS)
 pimmel_router_LDADD = libpimmel.la
 BUILT_SOURCES += pimmel-router-clo.c pimmel-router-clo.h
@@ -58,7 +58,7 @@ pimmel_dealer_SOURCES = pimmel-dealer.c
 pimmel_dealer_SOURCES += daemonise.c daemonise.h
 pimmel_dealer_CPPFLAGS = $(AM_CPPFLAGS)
 pimmel_dealer_CPPFLAGS += $(libev_CFLAGS)
-pimmel_dealer_LDFLAGS = $(AM_LDFLAGS) -static
+pimmel_dealer_LDFLAGS = $(AM_LDFLAGS)
 pimmel_dealer_LDFLAGS += $(libev_LIBS)
 pimmel_dealer_LDADD = libpimmel.la
 BUILT_SOURCES += pimmel-dealer-clo.c pimmel-dealer-clo.h
@@ -67,7 +67,7 @@ bin_PROGRAMS += pimmel-moni
 pimmel_moni_SOURCES = pimmel-moni.c pimmel-moni-clo.ggo
 pimmel_moni_CPPFLAGS = $(AM_CPPFLAGS)
 pimmel_moni_CPPFLAGS += $(libev_CFLAGS)
-pimmel_moni_LDFLAGS = $(AM_LDFLAGS) -static
+pimmel_moni_LDFLAGS = $(AM_LDFLAGS)
 pimmel_moni_LDFLAGS += $(libev_LIBS)
 pimmel_moni_LDADD = libpimmel.la
 BUILT_SOURCES += pimmel-moni-clo.c pimmel-moni-clo.h
--- origsrc/pimmel-0.1.0/src/pimmel.c	2013-01-17 00:05:50.000000000 +0900
+++ src/pimmel-0.1.0/src/pimmel.c	2024-09-26 16:57:06.571749600 +0900
@@ -172,6 +172,8 @@ fiddle_with_mtu(int MAYBE_UNUSED(s))
 	socklen_t mtuilen = sizeof(mtui);
 #endif	/* IPV6_PATHMTU */
 
+	int opt;
+
 #if defined IPV6_USE_MIN_MTU
 	/* use minimal mtu */
 	opt = 1;
