--- origsrc/wiredtiger-10.0.0/Makefile.am	2021-04-12 11:43:08.000000000 +0900
+++ src/wiredtiger-10.0.0/Makefile.am	2021-11-22 09:36:47.747752700 +0900
@@ -65,7 +65,7 @@ LDADD = $(lib_LTLIBRARIES)
 
 # BEGIN SOURCES, maintained by makemake and dist/filelist
 
-libwiredtiger_la_LDFLAGS = -release @VERSION@
+libwiredtiger_la_LDFLAGS = -no-undefined -release @VERSION@
 libwiredtiger_la_SOURCES =
 libwiredtiger_la_SOURCES += src/block/block_addr.c
 libwiredtiger_la_SOURCES += src/block/block_ckpt.c
--- origsrc/wiredtiger-10.0.0/ext/collators/reverse/Makefile.am	2021-04-12 10:26:28.000000000 +0900
+++ src/wiredtiger-10.0.0/ext/collators/reverse/Makefile.am	2021-11-21 22:37:53.025284100 +0900
@@ -6,5 +6,5 @@ libwiredtiger_reverse_collator_la_SOURCE
 # libtool hack: noinst_LTLIBRARIES turns off building shared libraries as well
 # as installation, it will only build static libraries.  As far as I can tell,
 # the "approved" libtool way to turn them back on is by adding -rpath.
-libwiredtiger_reverse_collator_la_LDFLAGS = \
+libwiredtiger_reverse_collator_la_LDFLAGS = -no-undefined \
 	-avoid-version -module -rpath /nowhere
--- origsrc/wiredtiger-10.0.0/ext/collators/revint/Makefile.am	2021-04-12 10:26:28.000000000 +0900
+++ src/wiredtiger-10.0.0/ext/collators/revint/Makefile.am	2021-11-21 22:38:49.215778700 +0900
@@ -6,5 +6,5 @@ libwiredtiger_revint_collator_la_SOURCES
 # libtool hack: noinst_LTLIBRARIES turns off building shared libraries as well
 # as installation, it will only build static libraries.  As far as I can tell,
 # the "approved" libtool way to turn them back on is by adding -rpath.
-libwiredtiger_revint_collator_la_LDFLAGS = \
+libwiredtiger_revint_collator_la_LDFLAGS = -no-undefined \
 	-avoid-version -module -rpath /nowhere
--- origsrc/wiredtiger-10.0.0/ext/compressors/nop/Makefile.am	2021-04-12 10:26:28.000000000 +0900
+++ src/wiredtiger-10.0.0/ext/compressors/nop/Makefile.am	2021-11-21 22:39:51.458450400 +0900
@@ -6,4 +6,4 @@ libwiredtiger_nop_la_SOURCES = nop_compr
 # libtool hack: noinst_LTLIBRARIES turns off building shared libraries as well
 # as installation, it will only build static libraries.  As far as I can tell,
 # the "approved" libtool way to turn them back on is by adding -rpath.
-libwiredtiger_nop_la_LDFLAGS = -avoid-version -module -rpath /nowhere
+libwiredtiger_nop_la_LDFLAGS = -no-undefined -avoid-version -module -rpath /nowhere
--- origsrc/wiredtiger-10.0.0/ext/encryptors/nop/Makefile.am	2021-04-12 10:26:28.000000000 +0900
+++ src/wiredtiger-10.0.0/ext/encryptors/nop/Makefile.am	2021-11-21 22:41:01.529380400 +0900
@@ -6,4 +6,4 @@ libwiredtiger_nop_la_SOURCES = nop_encry
 # libtool hack: noinst_LTLIBRARIES turns off building shared libraries as well
 # as installation, it will only build static libraries.  As far as I can tell,
 # the "approved" libtool way to turn them back on is by adding -rpath.
-libwiredtiger_nop_la_LDFLAGS = -avoid-version -module -rpath /nowhere
+libwiredtiger_nop_la_LDFLAGS = -no-undefined -avoid-version -module -rpath /nowhere
--- origsrc/wiredtiger-10.0.0/ext/encryptors/rotn/Makefile.am	2021-04-12 10:26:28.000000000 +0900
+++ src/wiredtiger-10.0.0/ext/encryptors/rotn/Makefile.am	2021-11-21 22:49:27.432169000 +0900
@@ -6,4 +6,4 @@ libwiredtiger_rotn_la_SOURCES = rotn_enc
 # libtool hack: noinst_LTLIBRARIES turns off building shared libraries as well
 # as installation, it will only build static libraries.  As far as I can tell,
 # the "approved" libtool way to turn them back on is by adding -rpath.
-libwiredtiger_rotn_la_LDFLAGS = -avoid-version -module -rpath /nowhere
+libwiredtiger_rotn_la_LDFLAGS = -no-undefined -avoid-version -module -rpath /nowhere
--- origsrc/wiredtiger-10.0.0/ext/extractors/csv/Makefile.am	2021-04-12 10:26:28.000000000 +0900
+++ src/wiredtiger-10.0.0/ext/extractors/csv/Makefile.am	2021-11-21 22:50:26.120240200 +0900
@@ -6,5 +6,5 @@ libwiredtiger_csv_extractor_la_SOURCES =
 # libtool hack: noinst_LTLIBRARIES turns off building shared libraries as well
 # as installation, it will only build static libraries.  As far as I can tell,
 # the "approved" libtool way to turn them back on is by adding -rpath.
-libwiredtiger_csv_extractor_la_LDFLAGS = \
+libwiredtiger_csv_extractor_la_LDFLAGS = -no-undefined \
 	-avoid-version -module -rpath /nowhere
--- origsrc/wiredtiger-10.0.0/ext/storage_sources/local_store/Makefile.am	2021-04-12 10:26:28.000000000 +0900
+++ src/wiredtiger-10.0.0/ext/storage_sources/local_store/Makefile.am	2021-11-21 22:51:27.688179000 +0900
@@ -6,4 +6,4 @@ libwiredtiger_local_store_la_SOURCES = l
 # libtool hack: noinst_LTLIBRARIES turns off building shared libraries as well
 # as installation, it will only build static libraries.  As far as I can tell,
 # the "approved" libtool way to turn them back on is by adding -rpath.
-libwiredtiger_local_store_la_LDFLAGS = -avoid-version -module -rpath /nowhere
+libwiredtiger_local_store_la_LDFLAGS = -no-undefined -avoid-version -module -rpath /nowhere
--- origsrc/wiredtiger-10.0.0/ext/test/fail_fs/Makefile.am	2021-04-12 10:26:28.000000000 +0900
+++ src/wiredtiger-10.0.0/ext/test/fail_fs/Makefile.am	2021-11-21 22:54:17.123273100 +0900
@@ -6,4 +6,4 @@ libwiredtiger_fail_fs_la_SOURCES = fail_
 # libtool hack: noinst_LTLIBRARIES turns off building shared libraries as well
 # as installation, it will only build static libraries.  As far as I can tell,
 # the "approved" libtool way to turn them back on is by adding -rpath.
-libwiredtiger_fail_fs_la_LDFLAGS = -avoid-version -module -rpath /nowhere
+libwiredtiger_fail_fs_la_LDFLAGS = -no-undefined -avoid-version -module -rpath /nowhere
--- origsrc/wiredtiger-10.0.0/ext/test/fail_fs/fail_fs.c	2021-04-12 10:26:28.000000000 +0900
+++ src/wiredtiger-10.0.0/ext/test/fail_fs/fail_fs.c	2021-11-21 22:53:51.303296500 +0900
@@ -35,7 +35,9 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <sys/stat.h>
+#if !defined(__CYGWIN__)
 #include <execinfo.h>
+#endif
 
 #include <wiredtiger_ext.h>
 #include "queue.h"
@@ -646,6 +648,7 @@ static int
 fail_fs_simulate_fail(
   FAIL_FILE_HANDLE *fail_fh, WT_SESSION *session, int64_t nops, const char *opkind)
 {
+#if !defined(__CYGWIN__)
     FAIL_FILE_SYSTEM *fail_fs;
     WT_EXTENSION_API *wtext;
 #ifdef __FreeBSD__
@@ -673,6 +676,7 @@ fail_fs_simulate_fail(
             free(btstr);
         }
     }
+#endif
     return (EIO);
 }
 
--- origsrc/wiredtiger-10.0.0/src/include/posix.h	2021-04-12 10:26:28.000000000 +0900
+++ src/wiredtiger-10.0.0/src/include/posix.h	2021-11-21 22:56:03.425253300 +0900
@@ -18,7 +18,9 @@
 #endif
 
 /* Define O_BINARY for Posix systems */
+#if !defined(O_BINARY)
 #define O_BINARY 0
+#endif
 
 /*
  * Define WT threading and concurrency primitives
