--- origsrc/cminpack-1.3.7/cmake/CMakeLists.txt	2020-12-10 12:46:30.000000000 +0900
+++ src/cminpack-1.3.7/cmake/CMakeLists.txt	2020-12-21 19:21:59.260488700 +0900
@@ -13,4 +13,8 @@ configure_file(cminpack.pc.in ${pkg_conf
 install(FILES ${pkg_conf_file}
     DESTINATION ${CMINPACK_LIB_INSTALL_DIR}/pkgconfig/ COMPONENT pkgconfig)
 
-install(EXPORT CMinpackConfig DESTINATION share/cminpack NAMESPACE cminpack::)
\ No newline at end of file
+if (CYGWIN)
+    install(EXPORT CMinpackConfig DESTINATION lib/cmake/cminpack NAMESPACE cminpack::)
+else()
+    install(EXPORT CMinpackConfig DESTINATION share/cminpack NAMESPACE cminpack::)
+endif()
--- origsrc/cminpack-1.3.7/cmake/cminpack.pc.in	2020-12-10 12:46:30.000000000 +0900
+++ src/cminpack-1.3.7/cmake/cminpack.pc.in	2020-12-21 19:24:56.033403300 +0900
@@ -6,7 +6,7 @@ includedir=${prefix}/@CMINPACK_INCLUDE_I
 
 Name: @PROJECT_NAME@
 Description: @PKG_DESC@
-Version: @CMINPACK_VERSION@
+Version: @VERSION@
 Requires: @PKG_EXTERNAL_DEPS@
 Libs: -L${libdir} -lcminpack -lm @PC_CMINPACK_LIBRARIES@ @PC_CMINPACK_LDFLAGS@
 Cflags: -I${includedir} @CMINPACK_CFLAGS@
