diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x '*.f' origsrc/cminpack-1.3.6/cmake/CMakeLists.txt src/cminpack-1.3.6/cmake/CMakeLists.txt
--- origsrc/cminpack-1.3.6/cmake/CMakeLists.txt	2017-02-25 00:09:31.000000000 +0900
+++ src/cminpack-1.3.6/cmake/CMakeLists.txt	2017-03-31 20:09:39.978598500 +0900
@@ -13,4 +13,4 @@ configure_file(cminpack.pc.in ${pkg_conf
 install(FILES ${pkg_conf_file}
     DESTINATION ${CMINPACK_LIB_INSTALL_DIR}/pkgconfig/ COMPONENT pkgconfig)
 
-install(FILES CMinpackConfig.cmake DESTINATION share/cmake/CMinpack)
+install(FILES CMinpackConfig.cmake DESTINATION lib/cmake/CMinpack)
