--- origsrc/cminpack-1.3.14/CMakeLists.txt	2026-07-25 20:11:13.000000000 +0900
+++ src/cminpack-1.3.14/CMakeLists.txt	2026-08-01 12:55:03.867986700 +0900
@@ -225,6 +225,6 @@ foreach (cminpack_lib ${cminpack_libs})
 
 endforeach ()
 
-install (EXPORT CMinpackConfig DESTINATION share/cminpack NAMESPACE cminpack::)
+install (EXPORT CMinpackConfig DESTINATION lib/cmake/cminpack NAMESPACE cminpack::)
 
 add_subdirectory (examples)
