--- origsrc/cole-1.0.0/Makefile.am	1999-06-18 06:37:42.000000000 +0900
+++ src/cole-1.0.0/Makefile.am	2013-10-03 22:12:46.949155200 +0900
@@ -8,7 +8,7 @@ AUTOMAKE_OPTIONS = gnu 1.4
 
 
 lib_LTLIBRARIES = libcole.la
-libcole_la_LDFLAGS = -version-info @VERSION_INFO@
+libcole_la_LDFLAGS = -no-undefined -version-info @VERSION_INFO@
 libcole_la_SOURCES = \
 		olecod.c    \
 		oledecod.c  \
@@ -21,8 +21,8 @@ libcole_la_SOURCES = \
 noinst_PROGRAMS = demo test
 ## Next two lines are a bad hack. I don't know how to do it cleaner.
 ## (don't link with shared library, because it's not compiled by default)
-demo_LDADD = .libs/libcole.a
-test_LDADD = .libs/libcole.a
+demo_LDADD = libcole.la
+test_LDADD = libcole.la
 
 TESTS = test.sh
 
