--- origsrc/libtimezonemap-0.4.5/src/timezone-completion.c	2015-12-24 03:22:54.000000000 +0900
+++ src/libtimezonemap-0.4.5/src/timezone-completion.c	2017-10-11 10:42:48.891000000 +0900
@@ -346,11 +346,11 @@ get_version (void)
 
   if (version == NULL)
     {
-      gchar *stdout = NULL;
-      g_spawn_command_line_sync ("lsb_release -rs", &stdout, NULL, NULL, NULL);
+      gchar *stdout_ = NULL;
+      g_spawn_command_line_sync ("lsb_release -rs", &stdout_, NULL, NULL, NULL);
 
-      if (stdout != NULL)
-        version = g_strstrip (stdout);
+      if (stdout_ != NULL)
+        version = g_strstrip (stdout_);
       else
         version = g_strdup("");
     }
