--- origsrc/libtimezonemap-0.4.6/src/timezone-completion.c	2018-11-28 12:39:27.000000000 +0900
+++ src/libtimezonemap-0.4.6/src/timezone-completion.c	2020-01-16 16:57:18.015803400 +0900
@@ -346,11 +346,11 @@ get_version (void)
 
   if (version == NULL)
     {
-      gchar *stdout = NULL;
-      g_spawn_command_line_sync ("lsb_release -rs", &stdout, NULL, NULL, NULL);
+      gchar *stdout_ = NULL;
+      g_spawn_command_line_sync ("lsb_release -rs", &stdout_, NULL, NULL, NULL);
 
-      if (stdout != NULL)
-        version = g_strstrip (stdout);
+      if (stdout_ != NULL)
+        version = g_strstrip (stdout_);
       else
         version = g_strdup("");
     }
