--- origsrc/violetminer/src/Argon2/src/CMakeLists.txt	2021-05-02 21:04:53.000000000 +0900
+++ src/violetminer/src/Argon2/src/CMakeLists.txt	2021-05-02 21:06:54.606307700 +0900
@@ -25,7 +25,7 @@ endif()
 
 # Add cpu_features if we're going to need them for runtime dispatching
 if("${ARCH}" STREQUAL "x86_64" OR "${ARCH}" STREQUAL "armv8")
-    add_subdirectory(cpu_features)
+#    add_subdirectory(cpu_features)
 endif()
 
 add_subdirectory(Intrinsics)
--- origsrc/violetminer/src/Argon2/src/Intrinsics/X86/ArgonIntrinsics.h	2021-05-02 21:04:53.000000000 +0900
+++ src/violetminer/src/Argon2/src/Intrinsics/X86/ArgonIntrinsics.h	2021-05-02 21:14:37.533777400 +0900
@@ -7,7 +7,7 @@
 #include <cstdint>
 
 #include "Argon2/Argon2.h"
-#include "cpu_features/include/cpuinfo_x86.h"
+#include <cpu_features/cpuinfo_x86.h>
 #include "Intrinsics/X86/IncludeIntrinsics.h"
 
 static const cpu_features::X86Features features = cpu_features::GetX86Info().features;
--- origsrc/violetminer/src/Argon2/src/Intrinsics/X86/BlakeIntrinsics.h	2021-05-02 21:04:53.000000000 +0900
+++ src/violetminer/src/Argon2/src/Intrinsics/X86/BlakeIntrinsics.h	2021-05-02 21:14:06.258774800 +0900
@@ -7,7 +7,7 @@
 #include <cstdint>
 
 #include "Blake2/Blake2b.h"
-#include "cpu_features/include/cpuinfo_x86.h"
+#include <cpu_features/cpuinfo_x86.h>
 #include "Intrinsics/X86/IncludeIntrinsics.h"
 
 static const cpu_features::X86Features features = cpu_features::GetX86Info().features;
--- origsrc/violetminer/src/Miner/GetConfig.cpp	2021-05-02 21:04:47.000000000 +0900
+++ src/violetminer/src/Miner/GetConfig.cpp	2021-05-02 21:23:47.103251500 +0900
@@ -17,7 +17,7 @@
 #include "Utilities/String.h"
 
 #if defined(X86_OPTIMIZATIONS)
-#include "cpu_features/include/cpuinfo_x86.h"
+#include <cpu_features/cpuinfo_x86.h>
 #endif
 
 #if defined(NVIDIA_ENABLED)
--- origsrc/violetminer/src/Miner/main.cpp	2021-05-02 21:04:47.000000000 +0900
+++ src/violetminer/src/Miner/main.cpp	2021-05-02 21:17:15.566887000 +0900
@@ -15,7 +15,7 @@
 #include "Utilities/GetChar.h"
 
 #if defined(X86_OPTIMIZATIONS)
-#include "cpu_features/include/cpuinfo_x86.h"
+#include <cpu_features/cpuinfo_x86.h>
 #endif
 
 #if defined(NVIDIA_ENABLED)
