--- origsrc/supermagic-1.2/Makefile.am	2012-09-18 05:30:34.000000000 +0900
+++ src/supermagic-1.2/Makefile.am	2015-03-07 16:26:33.725126900 +0900
@@ -21,7 +21,7 @@ supermagic.h supermagic.c
 
 supermagic_CFLAGS =
 supermagic_LDFLAGS =
-supermagic_LDADD =
+supermagic_LDADD = $(MPI_LIBS)
 
 if SMGC_BUILD_CELL
 supermagic_CFLAGS += -I$(top_srcdir)/cell
--- origsrc/supermagic-1.2/configure.ac	2012-09-18 06:10:57.000000000 +0900
+++ src/supermagic-1.2/configure.ac	2015-03-07 16:26:44.318876900 +0900
@@ -39,14 +39,7 @@ AC_CONFIG_MACRO_DIR([m4])
 
 dnl check for adequate mpi support
 AC_MSG_CHECKING([if $CC can compile MPI applications])
-AC_LINK_IFELSE([
-    AC_LANG_PROGRAM(
-        [[#include <mpi.h>]],
-        [[MPI_Finalize();]])],
-    [AC_MSG_RESULT([yes])],
-    [AC_MSG_RESULT([no])
-     AC_MSG_ERROR([$CC cannot compile MPI applications. cannot continue.])]
-)
+PKG_CHECK_MODULES([MPI], [ompi])
 
 dnl checks for libraries.
 SMGC_CELL
