--- origsrc/atompaw-4.2.0.1/Makefile.am	2022-03-23 10:50:12.000000000 +0900
+++ src/atompaw-4.2.0.1/Makefile.am	2022-06-23 10:42:40.479069600 +0900
@@ -80,3 +80,5 @@ nobase_doc_DATA = $(atp_xmpl_srcs) $(atp
 EXTRA_DIST += $(atp_xmpl_srcs)
 EXTRA_DIST += $(atp_doc_srcs)
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/atompaw-4.2.0.1/atompaw.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/atompaw-4.2.0.1/atompaw.pc.in	2022-06-23 10:42:32.659788900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: atompaw
+Description: PAW dataset generator for 1st-principle simulations
+Version: @VERSION@
+Libs: -L${libdir} -latompaw
+Cflags: -I${includedir}
--- origsrc/atompaw-4.2.0.1/configure.ac	2022-05-27 01:23:46.000000000 +0900
+++ src/atompaw-4.2.0.1/configure.ac	2022-06-23 10:59:05.363639600 +0900
@@ -243,4 +243,5 @@ AC_CONFIG_FILES([
   src/Makefile
   src/pkginfo.f90
 ])
+AC_CONFIG_FILES([atompaw.pc])
 AC_OUTPUT
--- origsrc/atompaw-4.2.0.1/src/Makefile.am	2022-03-21 06:37:25.000000000 +0900
+++ src/atompaw-4.2.0.1/src/Makefile.am	2022-06-23 10:30:30.402824900 +0900
@@ -61,10 +61,11 @@ atp_built_srcs = \
   pkginfo.f90
 
 # Libraries to install
-noinst_LTLIBRARIES = libatompaw.la
+lib_LTLIBRARIES = libatompaw.la
 libatompaw_la_SOURCES = $(atp_srcs) atompaw_lib.F90
 nodist_libatompaw_la_SOURCES = $(atp_built_srcs)
-#libatompaw_la_LDFLAGS = -version-info 0:0:0
+libatompaw_la_LDFLAGS = -no-undefined -version-info 0:0:0
+pkginclude_HEADERS = $(builddir)/*.mod
 
 # Programs to install
 bin_PROGRAMS = atompaw graphatom
