--- origsrc/sleuthkit-4.0.1/bindings/java/jni/Makefile.am	2012-11-14 04:46:34.000000000 +0900
+++ src/sleuthkit-4.0.1/bindings/java/jni/Makefile.am	2012-12-09 10:22:29.744149400 +0900
@@ -4,6 +4,7 @@ EXTRA_DIST = .indent.pro
 lib_LTLIBRARIES = libtsk_jni.la
 libtsk_jni_la_SOURCES = dataModel_SleuthkitJNI.cpp dataModel_SleuthkitJNI.h 
 libtsk_jni_la_LIBADD = ../../../tsk3/libtsk3.la
+libtsk_jni_la_LDFLAGS = -no-undefined
 
 indent:
 	indent *.cpp *.h
--- origsrc/sleuthkit-4.0.1/samples/Makefile.am	2012-11-14 04:46:34.000000000 +0900
+++ src/sleuthkit-4.0.1/samples/Makefile.am	2012-12-09 10:14:01.320321300 +0900
@@ -1,6 +1,5 @@
 AM_CPPFLAGS = -I.. -I$(srcdir)/.. -Wall 
 LDADD = ../tsk3/libtsk3.la
-AM_LDFLAGS = -static
 EXTRA_DIST = .indent.pro 
 
 noinst_PROGRAMS = posix_style callback_style posix_cpp_style callback_cpp_style
--- origsrc/sleuthkit-4.0.1/tests/Makefile.am	2012-11-14 04:46:34.000000000 +0900
+++ src/sleuthkit-4.0.1/tests/Makefile.am	2012-12-09 10:14:20.947274400 +0900
@@ -1,6 +1,6 @@
 AM_CPPFLAGS = -I.. -I$(srcdir)/.. -Wall $(PTHREAD_CFLAGS)
 LDADD = ../tsk3/libtsk3.la
-LDFLAGS += -static $(PTHREAD_LIBS)
+LDFLAGS += $(PTHREAD_LIBS)
 EXTRA_DIST = .indent.pro 
 
 noinst_PROGRAMS = read_apis fs_fname_apis fs_attrlist_apis fs_thread_test
--- origsrc/sleuthkit-4.0.1/tools/autotools/Makefile.am	2012-11-14 04:46:34.000000000 +0900
+++ src/sleuthkit-4.0.1/tools/autotools/Makefile.am	2012-12-09 10:14:31.864266600 +0900
@@ -1,6 +1,5 @@
 AM_CPPFLAGS = -I../.. -I$(srcdir)/../.. -Wall
 LDADD = ../../tsk3/libtsk3.la
-LDFLAGS += -static
 EXTRA_DIST = .indent.pro
 
 bin_PROGRAMS = tsk_recover tsk_loaddb tsk_comparedir tsk_gettimes
--- origsrc/sleuthkit-4.0.1/tools/fstools/Makefile.am	2012-11-14 04:46:34.000000000 +0900
+++ src/sleuthkit-4.0.1/tools/fstools/Makefile.am	2012-12-09 10:14:42.792977500 +0900
@@ -1,6 +1,5 @@
 AM_CPPFLAGS = -I../.. -I$(srcdir)/../.. -Wall 
 LDADD = ../../tsk3/libtsk3.la
-LDFLAGS += -static
 EXTRA_DIST = .indent.pro fscheck.cpp
 
 bin_PROGRAMS = blkcalc blkcat blkls blkstat ffind fls fcat fsstat icat ifind ils \
--- origsrc/sleuthkit-4.0.1/tools/hashtools/Makefile.am	2012-11-14 04:46:34.000000000 +0900
+++ src/sleuthkit-4.0.1/tools/hashtools/Makefile.am	2012-12-09 10:14:51.935555600 +0900
@@ -1,6 +1,5 @@
 AM_CPPFLAGS = -I../.. -I$(srcdir)/../.. -Wall 
 LDADD = ../../tsk3/libtsk3.la
-LDFLAGS += -static
 EXTRA_DIST = .indent.pro md5.c sha1.c
 
 bin_PROGRAMS = hfind
--- origsrc/sleuthkit-4.0.1/tools/imgtools/Makefile.am	2012-11-14 04:46:34.000000000 +0900
+++ src/sleuthkit-4.0.1/tools/imgtools/Makefile.am	2012-12-09 10:15:01.929696300 +0900
@@ -1,6 +1,5 @@
 AM_CPPFLAGS = -I../.. -I$(srcdir)/../.. -Wall
 LDADD = ../../tsk3/libtsk3.la
-LDFLAGS += -static
 EXTRA_DIST = .indent.pro
 
 bin_PROGRAMS = img_cat img_stat
--- origsrc/sleuthkit-4.0.1/tools/srchtools/Makefile.am	2012-11-14 04:46:34.000000000 +0900
+++ src/sleuthkit-4.0.1/tools/srchtools/Makefile.am	2012-12-09 10:15:15.894540000 +0900
@@ -6,7 +6,6 @@ srch_strings_SOURCES = srch_strings.c
 
 sigfind_SOURCES = sigfind.cpp 
 sigfind_LDADD = ../../tsk3/libtsk3.la
-sigfind_LDFLAGS = -static
 
 indent:
 	indent *.c *.cpp
--- origsrc/sleuthkit-4.0.1/tools/vstools/Makefile.am	2012-11-14 04:46:34.000000000 +0900
+++ src/sleuthkit-4.0.1/tools/vstools/Makefile.am	2012-12-09 10:15:25.240243100 +0900
@@ -1,6 +1,5 @@
 AM_CPPFLAGS = -I../.. -I$(srcdir)/../.. -Wall 
 LDADD = ../../tsk3/libtsk3.la
-LDFLAGS += -static
 EXTRA_DIST = .indent.pro
 
 bin_PROGRAMS = mmls mmstat mmcat
--- origsrc/sleuthkit-4.0.1/tsk3/Makefile.am	2012-11-14 04:47:40.000000000 +0900
+++ src/sleuthkit-4.0.1/tsk3/Makefile.am	2012-12-09 11:12:16.688500000 +0900
@@ -8,6 +8,6 @@ libtsk3_la_LIBADD = base/libtskbase.la i
     vs/libtskvs.la fs/libtskfs.la hashdb/libtskhashdb.la \
     auto/libtskauto.la
 # current:revision:age
-libtsk3_la_LDFLAGS = -version-info 9:0:0
+libtsk3_la_LDFLAGS = -version-info 9:0:0 -no-undefined
 
 EXTRA_DIST = tsk_tools_i.h docs/Doxyfile docs/*.dox docs/*.html
--- origsrc/sleuthkit-4.0.1/unit_tests/base/Makefile.am	2012-11-14 04:46:34.000000000 +0900
+++ src/sleuthkit-4.0.1/unit_tests/base/Makefile.am	2012-12-09 11:41:39.376000000 +0900
@@ -1,6 +1,5 @@
-AM_CPPFLAGS = -I../.. -Wall $(CPPUNIT_CFLAGS) 
+AM_CPPFLAGS = -I$(top_srcdir)/tsk3 -I$(top_srcdir) -Wall $(CPPUNIT_CFLAGS) 
 LDADD = ../../tsk3/libtsk3.la $(CPPUNIT_LIBS)
-LDFLAGS = -static 
 
 noinst_PROGRAMS = test_base
 test_base_SOURCES= test_base.cpp errors_test.cpp errors_test.h
