--- origsrc/sleuthkit-sleuthkit-4.11.1/configure.ac	2021-10-30 06:32:46.000000000 +0900
+++ src/sleuthkit-sleuthkit-4.11.1/configure.ac	2022-08-01 07:26:59.361529700 +0900
@@ -281,7 +281,7 @@ AC_CONFIG_COMMANDS([tsk/tsk_incs.h],
 
 AC_MSG_CHECKING([if libtool needs -no-undefined flag to build shared libraries])
 case "$host" in
-*-*-mingw*)
+*-*-mingw*|*-*-cygwin*)
   dnl Add -no-undefined flag to LDFLAGS to let libtool build DLLs.
   AC_MSG_RESULT([yes])
   LIBTSK_LDFLAGS="-no-undefined"
--- origsrc/sleuthkit-sleuthkit-4.11.1/samples/Makefile.am	2021-10-30 06:32:46.000000000 +0900
+++ src/sleuthkit-sleuthkit-4.11.1/samples/Makefile.am	2022-07-31 20:29:53.646071300 +0900
@@ -1,7 +1,7 @@
 AM_CPPFLAGS = -I.. -I$(srcdir)/..
 AM_CXXFLAGS += -Wno-unused-command-line-argument
 LDADD = ../tsk/libtsk.la
-AM_LDFLAGS = -static
+AM_LDFLAGS = 
 EXTRA_DIST = .indent.pro 
 
 noinst_PROGRAMS = posix_style callback_style posix_cpp_style callback_cpp_style
--- origsrc/sleuthkit-sleuthkit-4.11.1/tests/Makefile.am	2021-10-30 06:32:46.000000000 +0900
+++ src/sleuthkit-sleuthkit-4.11.1/tests/Makefile.am	2022-07-31 20:29:53.654057100 +0900
@@ -2,12 +2,12 @@ AM_CPPFLAGS = -I.. -I$(srcdir)/..
 AM_CFLAGS += $(PTHREAD_CFLAGS)
 AM_CXXFLAGS += -Wno-unused-command-line-argument $(PTHREAD_CFLAGS)
 LDADD = ../tsk/libtsk.la
-LDFLAGS += -static $(PTHREAD_LIBS)
+LDFLAGS += $(PTHREAD_LIBS)
 EXTRA_DIST = .indent.pro runtests.sh
 
-check_SCRIPTS = runtests.sh test_libraries.sh
+check_SCRIPTS = runtests.sh
 
-TESTS = runtests.sh test_libraries.sh
+TESTS = runtests.sh
 
 check_PROGRAMS = read_apis fs_fname_apis fs_attrlist_apis fs_thread_test
 
--- origsrc/sleuthkit-sleuthkit-4.11.1/tools/autotools/Makefile.am	2021-10-30 06:32:46.000000000 +0900
+++ src/sleuthkit-sleuthkit-4.11.1/tools/autotools/Makefile.am	2022-07-31 20:29:53.660024000 +0900
@@ -1,7 +1,7 @@
 AM_CPPFLAGS = -I../.. -I$(srcdir)/../.. 
 AM_CXXFLAGS += -Wno-overloaded-virtual -Wno-unused-command-line-argument
 LDADD = ../../tsk/libtsk.la
-LDFLAGS += -static
+LDFLAGS +=
 EXTRA_DIST = .indent.pro
 
 bin_PROGRAMS = tsk_recover tsk_loaddb tsk_comparedir tsk_gettimes tsk_imageinfo
--- origsrc/sleuthkit-sleuthkit-4.11.1/tools/fiwalk/src/Makefile.am	2021-10-30 06:32:46.000000000 +0900
+++ src/sleuthkit-sleuthkit-4.11.1/tools/fiwalk/src/Makefile.am	2022-07-31 20:29:53.663982900 +0900
@@ -1,5 +1,5 @@
 bin_PROGRAMS = fiwalk 
-AM_CPPFLAGS = -I../../..
+AM_CPPFLAGS = -I$(top_builddir) -I$(top_srcdir)
 AM_CXXFLAGS += -Wno-unused-command-line-argument
 LDADD = ../../../tsk/libtsk.la
 
--- origsrc/sleuthkit-sleuthkit-4.11.1/tools/fstools/Makefile.am	2021-10-30 06:32:46.000000000 +0900
+++ src/sleuthkit-sleuthkit-4.11.1/tools/fstools/Makefile.am	2022-07-31 20:29:53.688947800 +0900
@@ -1,7 +1,7 @@
 AM_CPPFLAGS = -I../.. -I$(srcdir)/../..
 AM_CXXFLAGS += -Wno-unused-command-line-argument
 LDADD = ../../tsk/libtsk.la
-LDFLAGS += -static
+LDFLAGS +=
 EXTRA_DIST = .indent.pro fscheck.cpp
 
 bin_PROGRAMS = blkcalc blkcat blkls blkstat ffind fls fcat fsstat icat ifind ils \
--- origsrc/sleuthkit-sleuthkit-4.11.1/tools/hashtools/Makefile.am	2021-10-30 06:32:46.000000000 +0900
+++ src/sleuthkit-sleuthkit-4.11.1/tools/hashtools/Makefile.am	2022-07-31 20:29:53.709985500 +0900
@@ -1,7 +1,7 @@
 AM_CPPFLAGS = -I../.. -I$(srcdir)/../..
 AM_CXXFLAGS += -Wno-unused-command-line-argument
 LDADD = ../../tsk/libtsk.la
-LDFLAGS += -static
+LDFLAGS +=
 EXTRA_DIST = .indent.pro md5.c sha1.c
 
 bin_PROGRAMS = hfind
--- origsrc/sleuthkit-sleuthkit-4.11.1/tools/imgtools/Makefile.am	2021-10-30 06:32:46.000000000 +0900
+++ src/sleuthkit-sleuthkit-4.11.1/tools/imgtools/Makefile.am	2022-07-31 20:29:53.718011600 +0900
@@ -1,7 +1,7 @@
 AM_CPPFLAGS = -I../.. -I$(srcdir)/../..
 AM_CXXFLAGS += -Wno-unused-command-line-argument
 LDADD = ../../tsk/libtsk.la
-LDFLAGS += -static
+LDFLAGS +=
 EXTRA_DIST = .indent.pro
 
 bin_PROGRAMS = img_cat img_stat
--- origsrc/sleuthkit-sleuthkit-4.11.1/tools/pooltools/Makefile.am	2021-10-30 06:32:46.000000000 +0900
+++ src/sleuthkit-sleuthkit-4.11.1/tools/pooltools/Makefile.am	2022-08-01 07:47:21.575372700 +0900
@@ -1,6 +1,6 @@
 AM_CPPFLAGS = -I../.. -I$(srcdir)/../.. -Wall
 LDADD = ../../tsk/libtsk.la
-LDFLAGS += -static
+LDFLAGS +=
 EXTRA_DIST = .indent.pro
 
 bin_PROGRAMS = pstat
--- origsrc/sleuthkit-sleuthkit-4.11.1/tools/srchtools/Makefile.am	2021-10-30 06:32:46.000000000 +0900
+++ src/sleuthkit-sleuthkit-4.11.1/tools/srchtools/Makefile.am	2022-07-31 20:29:53.725115700 +0900
@@ -8,7 +8,7 @@ srch_strings_SOURCES = srch_strings.c
 
 sigfind_SOURCES = sigfind.cpp 
 sigfind_LDADD = ../../tsk/libtsk.la
-sigfind_LDFLAGS = -static
+sigfind_LDFLAGS =
 
 indent:
 	indent *.c *.cpp
--- origsrc/sleuthkit-sleuthkit-4.11.1/tools/vstools/Makefile.am	2021-10-30 06:32:46.000000000 +0900
+++ src/sleuthkit-sleuthkit-4.11.1/tools/vstools/Makefile.am	2022-07-31 20:29:53.730015500 +0900
@@ -1,7 +1,7 @@
 AM_CPPFLAGS = -I../.. -I$(srcdir)/../..
 AM_CXXFLAGS += -Wno-unused-command-line-argument
 LDADD = ../../tsk/libtsk.la
-LDFLAGS += -static
+LDFLAGS +=
 EXTRA_DIST = .indent.pro
 
 bin_PROGRAMS = mmls mmstat mmcat
--- origsrc/sleuthkit-sleuthkit-4.11.1/tsk/img/ewf.cpp	2021-10-30 06:32:46.000000000 +0900
+++ src/sleuthkit-sleuthkit-4.11.1/tsk/img/ewf.cpp	2022-07-31 20:29:53.735068300 +0900
@@ -67,7 +67,7 @@ ewf_image_read(TSK_IMG_INFO * img_info,
 
     tsk_take_lock(&(ewf_info->read_lock));
 #if defined( HAVE_LIBEWF_V2_API )
-    cnt = libewf_handle_read_random(ewf_info->handle,
+    cnt = libewf_handle_read_buffer_at_offset(ewf_info->handle,
         buf, len, offset, &ewf_error);
     if (cnt < 0) {
         char *errmsg = NULL;
--- origsrc/sleuthkit-sleuthkit-4.11.1/unit_tests/base/Makefile.am	2021-10-30 06:32:46.000000000 +0900
+++ src/sleuthkit-sleuthkit-4.11.1/unit_tests/base/Makefile.am	2022-07-31 20:29:53.766114500 +0900
@@ -1,7 +1,7 @@
-AM_CPPFLAGS = -I../.. $(CPPUNIT_CFLAGS)
+AM_CPPFLAGS = -I../.. -I$(top_srcdir) -I$(top_srcdir)/tsk $(CPPUNIT_CFLAGS)
 AM_CXXFLAGS += -Wno-unused-command-line-argument
 LDADD = ../../tsk/libtsk.la $(CPPUNIT_LIBS)
-LDFLAGS = -static 
+LDFLAGS =
 
 TESTS = test_base
 
