--- origsrc/gzstream/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/gzstream/Makefile.am	2018-06-22 16:38:10.707856200 +0900
@@ -0,0 +1,15 @@
+check_PROGRAMS = test_gzip test_gunzip
+test_gzip_SOURCES = test_gzip.C
+test_gzip_LDADD = libgzstream.la
+test_gunzip_SOURCES = test_gunzip.C
+test_gunzip_LDADD = libgzstream.la
+
+include_HEADERS = gzstream.h
+
+lib_LTLIBRARIES = libgzstream.la
+libgzstream_la_SOURCES = gzstream.C
+libgzstream_la_LDFLAGS = -no-undefined
+libgzstream_la_LIBADD = $(ZLIB_LIBS)
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/gzstream/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/gzstream/configure.ac	2018-06-22 16:38:10.707856200 +0900
@@ -0,0 +1,26 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([gzstream], [1.5])
+AC_CONFIG_SRCDIR([gzstream.C])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+PKG_CHECK_MODULES([ZLIB], [zlib])
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+
+# Checks for library functions.
+
+AC_CONFIG_FILES([gzstream.pc Makefile])
+AC_OUTPUT
--- origsrc/gzstream/gzstream.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/gzstream/gzstream.pc.in	2018-06-22 16:38:10.660982600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: gzstream
+Description: Zlib C-library in a C++ iostream
+Version: @VERSION@
+Libs: -L${libdir} -lgzstream
+Cflags: -I${includedir}
