--- origsrc/libgaiagraphics-0.5/configure.ac	2013-02-18 18:41:00.000000000 +0900
+++ src/libgaiagraphics-0.5/configure.ac	2024-09-27 11:20:14.457911500 +0900
@@ -75,7 +75,7 @@ AC_CHECK_FUNCS([memset sqrt strcasecmp s
 AC_CHECK_LIB(z,inflateInit_,,AC_MSG_ERROR(['libz' is required but it doesn't seems to be installed on this system.]),-lm)
 AC_CHECK_LIB(jpeg,jpeg_start_compress,,AC_MSG_ERROR(['libjpeg' is required but it doesn't seems to be installed on this system.]),-lm)
 AC_CHECK_LIB(tiff,TIFFClientOpen,,AC_MSG_ERROR(['libtiff' is required but it doesn't seems to be installed on this system.]),-lm)
-AC_CHECK_LIB(proj,pj_init_plus,,AC_MSG_ERROR(['libproj' is required but it doesn't seems to be installed on this system.]),-lm)
+PKG_CHECK_MODULES([PROJ], [proj], , AC_MSG_ERROR(['libproj' is required but it doesn't seems to be installed on this system.]))
 AC_CHECK_LIB(geotiff,GTIFSetFromProj4,,AC_MSG_ERROR(['libgeotiff' [>= v.1.2.5] is required but it doesn't seems to be installed on this system.]),-lm)
 
 PKG_CHECK_MODULES([LIBPNG], [libpng], , AC_MSG_ERROR(['libpng' is required but it doesn't seems to be installed on this system.]))
--- origsrc/libgaiagraphics-0.5/gaiagraphics.pc.in	2013-02-18 18:41:00.000000000 +0900
+++ src/libgaiagraphics-0.5/gaiagraphics.pc.in	2024-09-27 12:54:51.443474400 +0900
@@ -7,6 +7,6 @@ includedir=@includedir@
 
 Name: gaiagraphics
 Description: Gaia Common Graphics Support
-Version: 1.0
+Version: @VERSION@
 Libs: -L${libdir} -lgaiagraphics
 Cflags: -I${includedir}
