--- origsrc/activemq-cpp-library-3.8.3/configure.ac	2014-07-12 03:05:36.000000000 +0900
+++ src/activemq-cpp-library-3.8.3/configure.ac	2014-12-30 22:59:26.661481800 +0900
@@ -224,7 +224,7 @@ case "${host_os}" in
 esac
 
 if test "$GCC" = "yes"; then
-   PLAT_CXXFLAGS="-ansi -pedantic"
+   PLAT_CXXFLAGS="-pedantic"
 fi
 
 ## find and configure the Apache Decaf Library
@@ -263,8 +263,8 @@ AC_SUBST([AMQ_CFLAGS])
 AC_SUBST([AMQ_LIBS])
 
 if test "$GCC" = "yes"; then
-   AMQ_CXXFLAGS="$PLAT_CXXFLAGS $DECAF_INCLUDES $DECAF_CXXFLAGS -W -Wall -Wextra -Wconversion -fPIC -fstrict-aliasing -Wstrict-aliasing=2 -Wno-long-long"
-   AMQ_CFLAGS="$PLAT_CFLAGS $DECAF_INCLUDES $DECAF_CFLAGS -W -Wall -fPIC"
+   AMQ_CXXFLAGS="$PLAT_CXXFLAGS $DECAF_INCLUDES $DECAF_CXXFLAGS -W -Wall -Wextra -Wconversion -fstrict-aliasing -Wstrict-aliasing=2 -Wno-long-long"
+   AMQ_CFLAGS="$PLAT_CFLAGS $DECAF_INCLUDES $DECAF_CFLAGS -W -Wall"
 else
    AMQ_CXXFLAGS="$PLAT_CXXFLAGS $DECAF_INCLUDES $DECAF_CXXFLAGS"
    AMQ_CFLAGS="$PLAT_CXXFLAGS $DECAF_INCLUDES $DECAF_CXXFLAGS"
--- origsrc/activemq-cpp-library-3.8.3/src/main/Makefile.am	2014-07-12 03:07:16.000000000 +0900
+++ src/activemq-cpp-library-3.8.3/src/main/Makefile.am	2014-12-30 22:54:02.831403700 +0900
@@ -1326,7 +1326,7 @@ lib_LTLIBRARIES= libactivemq-cpp.la
 libactivemq_cpp_la_SOURCES= $(h_sources) $(cc_sources)
 libactivemq_cpp_la_CXXFLAGS= $(AMQ_CXXFLAGS)
 libactivemq_cpp_la_CFLAGS= $(AMQ_CFLAGS)
-libactivemq_cpp_la_LDFLAGS= -version-info $(ACTIVEMQ_LIBRARY_VERSION)
+libactivemq_cpp_la_LDFLAGS= -no-undefined -version-info $(ACTIVEMQ_LIBRARY_VERSION)
 libactivemq_cpp_la_LIBADD= $(AMQ_LIBS)
 
 ##
--- origsrc/activemq-cpp-library-3.8.3/src/test/Makefile.am	2014-07-12 03:07:15.000000000 +0900
+++ src/activemq-cpp-library-3.8.3/src/test/Makefile.am	2014-12-30 22:54:02.831403700 +0900
@@ -509,7 +509,7 @@ h_sources = \
 check_PROGRAMS = activemq-test
 
 ## Also run the tests as part of make check
-## TESTS = $(check_PROGRAMS)
+TESTS = $(check_PROGRAMS)
 
 ## 
 ## Compiler/Linker Options
