--- origsrc/activemq-cpp-library-3.9.5/src/main/Makefile.am	2019-01-26 05:52:51.000000000 +0900
+++ src/activemq-cpp-library-3.9.5/src/main/Makefile.am	2024-09-19 14:59:10.305014500 +0900
@@ -1370,7 +1370,7 @@ lib_LTLIBRARIES= libactivemq-cpp.la
 libactivemq_cpp_la_SOURCES= $(h_sources) $(cc_sources)
 libactivemq_cpp_la_CXXFLAGS= $(AMQ_CXXFLAGS)
 libactivemq_cpp_la_CFLAGS= $(AMQ_CFLAGS)
-libactivemq_cpp_la_LDFLAGS= -version-info $(ACTIVEMQ_LIBRARY_VERSION)
+libactivemq_cpp_la_LDFLAGS= -no-undefined -version-info $(ACTIVEMQ_LIBRARY_VERSION)
 libactivemq_cpp_la_LIBADD= $(AMQ_LIBS)
 
 ##
