--- origsrc/parasail-2.6.2/Makefile.am	2023-03-11 03:08:17.000000000 +0900
+++ src/parasail-2.6.2/Makefile.am	2024-08-16 18:22:51.544785300 +0900
@@ -36,7 +36,7 @@ else
 if ON_MSYS
 libparasail_la_LDFLAGS = $(AM_LDFLAGS) -no-undefined -avoid-version -version-info @LTVER@
 else
-libparasail_la_LDFLAGS = $(AM_LDFLAGS) -version-info @LTVER@
+libparasail_la_LDFLAGS = $(AM_LDFLAGS) -no-undefined -version-info @LTVER@
 endif
 endif
 
