--- origsrc/libvbucket/configure.ac	2010-12-24 10:54:58.000000000 +0900
+++ src/libvbucket/configure.ac	2010-12-24 11:37:03.011337100 +0900
@@ -11,14 +11,13 @@ AC_CONFIG_SRCDIR([src/vbucket.c])
 AC_CONFIG_AUX_DIR(config)
 AM_CONFIG_HEADER([config.h])
 AC_CONFIG_MACRO_DIR([m4])
-AC_DISABLE_STATIC
 
 # the malloc tests seems to be broken for cross compilation.. ignore them
 ac_cv_func_malloc_0_nonnull=yes
 ac_cv_func_realloc_0_nonnull=yes
 
-
-PANDORA_CANONICAL_TARGET(less-warnings, warnings-always-on, force-gcc42)
+LT_INIT(win32-dll)
+PANDORA_CANONICAL_TARGET(less-warnings, force-gcc42)
 PANDORA_REQUIRE_LIBHASHKIT
 
 AH_TOP([
