--- origsrc/seccure-0.4/Makefile	2009-04-09 21:42:50.000000000 +0900
+++ src/seccure-0.4/Makefile	2011-06-27 10:22:18.496137500 +0900
@@ -28,8 +28,8 @@ rebuild: clean default
 
 
 seccure-key: seccure.o numtheory.o ecc.o serialize.o protocol.o curves.o aes256ctr.o
-	$(CC) $(CFLAGS) -o seccure-key -lgcrypt seccure.o numtheory.o ecc.o \
-	curves.o serialize.o protocol.o aes256ctr.o
+	$(CC) $(CFLAGS) -o seccure-key seccure.o numtheory.o ecc.o \
+	curves.o serialize.o protocol.o aes256ctr.o -lgcrypt
 	strip seccure-key
 
 seccure-encrypt: seccure-key
