--- origsrc/libibumad-1.3.9/Makefile.am	2014-02-06 01:19:48.000000000 +0900
+++ src/libibumad-1.3.9/Makefile.am	2014-07-21 12:02:39.761928300 +0900
@@ -29,7 +29,7 @@ endif
 
 libibumad_la_SOURCES = src/umad.c src/sysfs.c
 libibumad_la_LDFLAGS = -version-info $(ibumad_api_version) \
-    -export-dynamic $(libibumad_version_script)
+    -export-dynamic $(libibumad_version_script) -no-undefined
 libibumad_la_DEPENDENCIES = $(srcdir)/src/libibumad.map
 
 libibumadincludedir = $(includedir)/infiniband
