--- origsrc/libibumad-1.3.10.2/Makefile.am	2014-12-08 23:54:19.000000000 +0900
+++ src/libibumad-1.3.10.2/Makefile.am	2016-04-18 18:36:44.571676600 +0900
@@ -36,7 +36,7 @@ endif
 
 libibumad_la_SOURCES = src/umad.c src/sysfs.c src/umad_str.c
 libibumad_la_LDFLAGS = -version-info $(ibumad_api_version) \
-    -export-dynamic $(libibumad_version_script)
+    -export-dynamic $(libibumad_version_script) -no-undefined
 libibumad_la_DEPENDENCIES = $(srcdir)/src/libibumad.map
 
 libibumadincludedir = $(includedir)/infiniband
@@ -73,3 +73,6 @@ install-data-hook:
 	$(LN_S) umad_init.3 umad_done.3 && \
 	$(LN_S) umad_get_ca.3 umad_release_ca.3 && \
 	$(LN_S) umad_get_port.3 umad_release_port.3
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libibumad-1.3.10.2/configure.in	2015-03-04 06:07:49.000000000 +0900
+++ src/libibumad-1.3.10.2/configure.in	2016-04-18 18:36:37.868110400 +0900
@@ -68,5 +68,5 @@ AC_CACHE_CHECK(whether ld accepts --vers
 
 AM_CONDITIONAL(HAVE_LD_VERSION_SCRIPT, test "$ac_cv_version_script" = "yes")
 
-AC_CONFIG_FILES([Makefile libibumad.spec])
+AC_CONFIG_FILES([libibumad.pc Makefile libibumad.spec])
 AC_OUTPUT
--- origsrc/libibumad-1.3.10.2/libibumad.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libibumad-1.3.10.2/libibumad.pc.in	2016-04-18 18:37:05.636082400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libibumad
+Description: OpenFabrics Alliance InfiniBand umad (user MAD) library
+Version: @VERSION@
+Libs: -L${libdir} -libumad
+Cflags: -I${includedir}
