--- origsrc/libmurmurhash-1.6/Makefile.am	2023-05-18 23:24:56.000000000 +0900
+++ src/libmurmurhash-1.6/Makefile.am	2024-02-15 09:12:13.378961600 +0900
@@ -6,7 +6,7 @@ lib_LTLIBRARIES  = libmurmurhash.la
 libmurmurhashdir=$(includedir)
 libmurmurhash_HEADERS= murmurhash.h
 libmurmurhash_la_SOURCES = PMurHash.c murmurhash.c PMurHash.h
-libmurmurhash_la_LDFLAGS = -version-info @LIB_VERSION@
+libmurmurhash_la_LDFLAGS = -no-undefined -version-info @LIB_VERSION@
 
 libmurmurhash_la_CPPFLAGS = $(INCLUDES)
 
