--- origsrc/wlsuite/Makefile.am	2015-04-20 15:27:21.000000000 +0900
+++ src/wlsuite/Makefile.am	2015-04-20 15:43:55.938726300 +0900
@@ -1,2 +1,5 @@
 SUBDIRS = src
 ACLOCAL_AMFLAGS = -I m4
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/wlsuite/configure.ac	2015-04-20 15:27:21.000000000 +0900
+++ src/wlsuite/configure.ac	2015-04-20 15:43:48.907500300 +0900
@@ -1,6 +1,6 @@
 AC_PREREQ([2.50])
 AC_INIT([wlsuite], [0.1], [k@ailis.de])
-AM_INIT_AUTOMAKE([foreign -Wall -Werror])
+AM_INIT_AUTOMAKE([foreign subdir-objects -Wall])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADERS([config.h])
 
@@ -34,5 +34,6 @@ AC_CONFIG_FILES(
   src/decodecpa/Makefile
   src/unpacktiles/Makefile
   src/unpackpics/Makefile
+  wlsuite.pc
 )
 AC_OUTPUT
--- origsrc/wlsuite/src/libwasteland/Makefile.am	2015-04-20 15:27:21.000000000 +0900
+++ src/wlsuite/src/libwasteland/Makefile.am	2015-04-20 15:32:42.559863400 +0900
@@ -18,7 +18,7 @@ libwasteland_la_SOURCES = \
   pics.c
 libwastelandincludedir = $(includedir)
 libwastelandinclude_HEADERS = wasteland.h
-AM_CFLAGS = -Wall -Werror -O2
+AM_CFLAGS = -Wall -O2
 
 doc:
 	rm -f ../../doc/html/*.html ../../doc/html/*.gif
--- origsrc/wlsuite/src/packsprites/Makefile.am	2015-04-20 15:27:21.000000000 +0900
+++ src/wlsuite/src/packsprites/Makefile.am	2015-04-20 15:33:50.759985500 +0900
@@ -4,4 +4,4 @@ wl_packsprites_SOURCES = \
         ../common/str.c \
 	packsprites.c
 
-AM_CFLAGS = -Wall -Werror -O2
+AM_CFLAGS = -Wall -O2
--- origsrc/wlsuite/wlsuite.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/wlsuite/wlsuite.pc.in	2015-04-20 15:42:42.720179700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: wlsuite
+Description: Library and a suite of programs for converting wasteland files into standard files and vice versa
+Version: @VERSION@
+Libs: -L${libdir} -lwasteland
+Cflags: -I${includedir}
