--- origsrc/level-zero-1.4.1/source/layers/tracing/CMakeLists.txt	2021-06-23 02:10:50.000000000 +0900
+++ src/level-zero-1.4.1/source/layers/tracing/CMakeLists.txt	2021-07-26 11:07:40.709435400 +0900
@@ -37,6 +37,7 @@ set_target_properties(${TARGET_NAME} PRO
 )
 
 install(TARGETS ze_tracing_layer
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     LIBRARY
         DESTINATION ${CMAKE_INSTALL_LIBDIR}
@@ -45,6 +46,7 @@ install(TARGETS ze_tracing_layer
 )
 
 install(TARGETS ze_tracing_layer
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     LIBRARY
         DESTINATION ${CMAKE_INSTALL_LIBDIR}
--- origsrc/level-zero-1.4.1/source/layers/validation/CMakeLists.txt	2021-06-23 02:10:50.000000000 +0900
+++ src/level-zero-1.4.1/source/layers/validation/CMakeLists.txt	2021-07-26 11:07:40.713436100 +0900
@@ -33,6 +33,7 @@ set_target_properties(${TARGET_NAME} PRO
 )
 
 install(TARGETS ze_validation_layer
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     LIBRARY
         DESTINATION ${CMAKE_INSTALL_LIBDIR}
@@ -41,6 +42,7 @@ install(TARGETS ze_validation_layer
 )
 
 install(TARGETS ze_validation_layer
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     LIBRARY
         DESTINATION ${CMAKE_INSTALL_LIBDIR}
