--- origsrc/libdjconsole-0.1.3/Makefile.am	2008-06-11 14:39:41.000000000 +0900
+++ src/libdjconsole-0.1.3/Makefile.am	2011-01-03 03:15:42.485705200 +0900
@@ -43,4 +43,4 @@ EXTRA_DIST = libdjconsole.spec \
 			 06f8-b101.leds \
 			 45-hpdjconsole.rules
 
-libdjconsole_la_LDFLAGS = -version-info @LIB_CURRENT@:@LIB_REVISION@:@LIB_AGE@
+libdjconsole_la_LDFLAGS = -version-info @LIB_CURRENT@:@LIB_REVISION@:@LIB_AGE@ -no-undefined
--- origsrc/libdjconsole-0.1.3/configure.ac	2008-06-11 14:39:41.000000000 +0900
+++ src/libdjconsole-0.1.3/configure.ac	2011-01-03 03:15:30.498468400 +0900
@@ -20,7 +20,7 @@ AC_PROG_CXX
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
 AC_PROG_CPP
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 PKG_CHECK_MODULES(LIBUSB, libusb >= 0.1.11)
 CXXFLAGS="$CXXFLAGS $LIBUSB_CFLAGS"
