--- origsrc/libdjconsole-0.1.3/Makefile.am	2008-06-11 14:39:41.000000000 +0900
+++ src/libdjconsole-0.1.3/Makefile.am	2024-10-28 23:43:49.507329900 +0900
@@ -43,4 +43,4 @@ EXTRA_DIST = libdjconsole.spec \
 			 06f8-b101.leds \
 			 45-hpdjconsole.rules
 
-libdjconsole_la_LDFLAGS = -version-info @LIB_CURRENT@:@LIB_REVISION@:@LIB_AGE@
+libdjconsole_la_LDFLAGS = -no-undefined -version-info @LIB_CURRENT@:@LIB_REVISION@:@LIB_AGE@
--- origsrc/libdjconsole-0.1.3/djconsole.cpp	2008-06-11 14:39:41.000000000 +0900
+++ src/libdjconsole-0.1.3/djconsole.cpp	2024-10-29 00:22:58.375533300 +0900
@@ -23,6 +23,7 @@ libdjconsole; if not, write to the Free
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 #include <usb.h>
 
 DJConsole::DJConsole(bool load_data)
@@ -107,7 +108,7 @@ DJConsole::DJConsole(bool load_data)
 
 	hdev1=usb_open(dev);
 
-	if(hdev1 < 0)
+	if(hdev1 != 0)
 	{
 		printf("Error opening DJ Console device, check permissions\n");
 		return;
@@ -115,7 +116,7 @@ DJConsole::DJConsole(bool load_data)
 
 	hdev2=usb_open(dev);
 
-	if(hdev2 < 0)
+	if(hdev2 != 0)
 	{
 		printf("Error opening DJ Console device, check permissions\n");
 		return;
--- origsrc/libdjconsole-0.1.3/libdjconsole.pc.in	2008-06-11 14:39:41.000000000 +0900
+++ src/libdjconsole-0.1.3/libdjconsole.pc.in	2024-10-29 00:25:26.456167100 +0900
@@ -5,6 +5,6 @@ includedir=@includedir@/libdjconsole
 
 Name: libdjconsole
 Description: Allows access to Hercules DJ Console
-Version: 0.1.0
+Version: @VERSION@
 Libs: -L${libdir} -ldjconsole
 Cflags: -I${includedir}
