--- origsrc/libofono-qt-1.30/ofono-qt.pro	2021-04-11 17:20:10.000000000 +0900
+++ src/libofono-qt-1.30/ofono-qt.pro	2025-08-13 08:31:51.246811100 +0900
@@ -1,7 +1,7 @@
 include(version.pri)
 TEMPLATE = subdirs
 CONFIG += ordered
-SUBDIRS += lib tests
+SUBDIRS += lib # tests
 
 dox.target = doc
 dox.commands = sed 's/@VERSION@/'$$VERSION'/' doxygen.cfg.in > doxygen.cfg; \
@@ -13,7 +13,7 @@ dox.depends = FORCE
 dox.clean_commands = rm -fr doc/
 dox.clean = doxygen.cfg 
 
-INSTALLS += dox
+# INSTALLS += dox
 
 QMAKE_EXTRA_TARGETS += dox
 QMAKE_DISTNAME = ofono-qt-$$VERSION
