--- origsrc/libtermkey-0.22/Makefile	2019-03-14 22:47:10.000000000 +0900
+++ src/libtermkey-0.22/Makefile	2020-01-16 17:30:07.997227000 +0900
@@ -70,7 +70,7 @@ all: $(LIBRARY) $(DEMOS)
 	$(LIBTOOL) --mode=compile --tag=CC $(CC) $(CFLAGS) -o $@ -c $<
 
 $(LIBRARY): $(OBJECTS)
-	$(LIBTOOL) --mode=link --tag=CC $(CC) -rpath $(LIBDIR) -version-info $(VERSION_CURRENT):$(VERSION_REVISION):$(VERSION_AGE) $(LDFLAGS) -o $@ $^
+	$(LIBTOOL) --mode=link --tag=CC $(CC) -rpath $(LIBDIR) -no-undefined -version-info $(VERSION_CURRENT):$(VERSION_REVISION):$(VERSION_AGE) $(LDFLAGS) -o $@ $^
 
 demo: $(LIBRARY) demo.lo
 	$(LIBTOOL) --mode=link --tag=CC $(CC) -o $@ $^
