--- origsrc/libupnp-1.6.18/configure.ac	2013-01-30 00:39:56.000000000 +0900
+++ src/libupnp-1.6.18/configure.ac	2013-02-06 08:01:48.958721200 +0900
@@ -605,7 +605,10 @@ AC_C_CONST
 # so we use a new test.
 #TYPE_SOCKLEN_T
 
-AC_CHECK_HEADERS([sys/types.h sys/socket.h ws2tcpip.h])
+AC_CHECK_HEADERS([sys/types.h sys/socket.h])
+if test "$host_os" != "cygwin"; then
+	AC_CHECK_HEADERS([ws2tcpip.h])
+fi
 AC_MSG_CHECKING(for socklen_t)
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
 #ifdef HAVE_SYS_TYPES_H
@@ -700,8 +703,10 @@ AC_OUTPUT
 #
 # Files copied for windows compilation.
 #
-echo "configure: copying \"autoconfig.h\"          to \"build/inc/autoconfig.h\""
-cp autoconfig.h build/inc/autoconfig.h
-echo "configure: copying \"upnp/inc/upnpconfig.h\" to \"build/inc/upnpconfig.h\""
-cp upnp/inc/upnpconfig.h build/inc/upnpconfig.h
+if test "$host_os" != "cygwin"; then
+	echo "configure: copying \"autoconfig.h\"          to \"build/inc/autoconfig.h\""
+	cp autoconfig.h build/inc/autoconfig.h
+	echo "configure: copying \"upnp/inc/upnpconfig.h\" to \"build/inc/upnpconfig.h\""
+	cp upnp/inc/upnpconfig.h build/inc/upnpconfig.h
+fi
 
--- origsrc/libupnp-1.6.18/ixml/Makefile.am	2010-10-20 02:31:19.000000000 +0900
+++ src/libupnp-1.6.18/ixml/Makefile.am	2013-02-06 08:02:25.583721200 +0900
@@ -7,7 +7,7 @@
 
 SUBDIRS 		= doc
 
-AM_CPPFLAGS		= -I$(srcdir)/inc -I$(srcdir)/src/inc
+AM_CPPFLAGS		= -I$(srcdir)/inc -I$(srcdir)/src/inc -I$(top_srcdir)/upnp/inc
 AM_CFLAGS		= 
 
 LDADD			= libixml.la
@@ -21,7 +21,7 @@ endif
 lib_LTLIBRARIES		= libixml.la
 
 libixml_la_LDFLAGS	= -version-info $(LT_VERSION_IXML) \
-			  -export-symbols-regex '^ixml.*'
+			  -export-symbols-regex '^ixml.*' -no-undefined
 
 libixml_la_SOURCES	= \
 			src/attr.c \
--- origsrc/libupnp-1.6.18/threadutil/Makefile.am	2013-01-29 22:50:42.000000000 +0900
+++ src/libupnp-1.6.18/threadutil/Makefile.am	2013-02-06 08:02:51.849346200 +0900
@@ -4,7 +4,7 @@
 # (C) Copyright 2005 Remi Turboult <r3mi@users.sourceforge.net>
 #
 
-AM_CPPFLAGS		= -I$(srcdir)/inc -I$(srcdir)/src/inc
+AM_CPPFLAGS		= -I$(srcdir)/inc -I$(srcdir)/src/inc -I$(top_srcdir)/upnp/inc
 
 if ENABLE_DEBUG
     AM_CPPFLAGS		+= -DDEBUG -DSTATS
@@ -15,7 +15,7 @@ endif
 
 lib_LTLIBRARIES		= libthreadutil.la
 
-libthreadutil_la_LDFLAGS = -version-info $(LT_VERSION_THREADUTIL)
+libthreadutil_la_LDFLAGS = -version-info $(LT_VERSION_THREADUTIL) -no-undefined
 
 libthreadutil_la_SOURCES = \
 			inc/FreeList.h \
--- origsrc/libupnp-1.6.18/upnp/Makefile.am	2013-01-29 22:50:42.000000000 +0900
+++ src/libupnp-1.6.18/upnp/Makefile.am	2013-02-06 08:03:06.286846200 +0900
@@ -38,7 +38,7 @@ lib_LTLIBRARIES = libupnp.la
 
 libupnp_la_CPPFLAGS = $(AM_CPPFLAGS) -I$(srcdir)/src/inc 
 
-libupnp_la_LDFLAGS = \
+libupnp_la_LDFLAGS = -no-undefined \
 	-version-info $(LT_VERSION_UPNP) \
 	-export-symbols-regex '^Upnp.*' \
 	$(top_builddir)/threadutil/libthreadutil.la \
