--- origsrc/libnmserver-0.0.92/src/Makefile.am	2010-12-10 22:38:07.000000000 +0900
+++ src/libnmserver-0.0.92/src/Makefile.am	2011-08-15 00:24:42.246400300 +0900
@@ -11,7 +11,7 @@ libnmserver_la_CFLAGS = $(DBUS_CFLAGS) $
 libnmserver_la_LIBADD = $(LIBOBJS) $(DBUS_LIBS)
 
 # http://www.gnu.org/software/libtool/manual/html_node/Libtool-versioning.html
-libnmserver_la_LDFLAGS = -version-info 1:0:0
+libnmserver_la_LDFLAGS = -version-info 1:0:0 -no-undefined
 
 libnmserver_la_includedir = $(includedir)/libnmserver/
 libnmserver_la_include_HEADERS = libnmserver.h
