--- origsrc/libnmserver-0.0.91/configure.ac	2010-06-28 23:00:35.000000000 +0900
+++ src/libnmserver-0.0.91/configure.ac	2011-01-10 15:04:10.026368800 +0900
@@ -4,13 +4,12 @@
 AC_PREREQ([2.63])
 AC_INIT([libnmserver], [0.0.91], [https://fedorahosted.org/libnmserver/newticket])
 AM_INIT_AUTOMAKE([foreign -Wall])
-LT_INIT
+LT_INIT(win32-dll)
 
 AC_CONFIG_HEADERS([config.h])
 
 # Checks for programs.
 AC_PROG_CC
-AM_PROG_LIBTOOL
 
 # Checks for libraries.
 PKG_CHECK_MODULES([DBUS], [dbus-1])
--- origsrc/libnmserver-0.0.91/src/Makefile.am	2010-06-03 21:35:16.000000000 +0900
+++ src/libnmserver-0.0.91/src/Makefile.am	2011-01-10 15:04:30.628299400 +0900
@@ -4,6 +4,7 @@ lib_LTLIBRARIES = libnmserver.la
 libnmserver_la_SOURCES = libnmserver-dbus.c libnmserver.c nslist.c
 libnmserver_la_CFLAGS = $(DBUS_CFLAGS) $(AM_CFLAGS)
 libnmserver_la_LIBADD = $(LIBOBJS) $(DBUS_LIBS)
+libnmserver_la_LDFLAGS = -no-undefined
 
 libnmserver_ladir = $(ladir)
 libnmserver_la_includedir = $(includedir)/libnmserver-0/
