--- origsrc/xdrfile-1.1.4/Makefile.am	2009-06-05 15:47:48.000000000 +0900
+++ src/xdrfile-1.1.4/Makefile.am	2024-09-26 12:48:46.319034000 +0900
@@ -137,3 +137,6 @@ DISTCLEANFILES = libtool config.cache co
 
 
 
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/xdrfile-1.1.4/configure.ac	2014-06-12 06:50:19.000000000 +0900
+++ src/xdrfile-1.1.4/configure.ac	2024-09-26 12:49:32.716364100 +0900
@@ -3,7 +3,7 @@
 #######################################################################
  
 AC_PREREQ(2.50)
-AC_INIT(xdrfile, 1.1.2, [gmx-users@gromacs.org])
+AC_INIT(xdrfile, 1.1.4, [gmx-users@gromacs.org])
 AC_CONFIG_SRCDIR(src/xdrfile.c)
 AC_CONFIG_AUX_DIR(config)
 AC_CANONICAL_HOST
@@ -200,8 +200,7 @@ ACX_FLOAT_FORMAT
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
-AC_LIBTOOL_WIN32_DLL
-AC_PROG_LIBTOOL
+LT_INIT
 AC_SYS_LARGEFILE
 AC_FUNC_FSEEKO
 
@@ -411,6 +410,7 @@ AC_CONFIG_FILES([ include/Makefile ])
 AC_CONFIG_FILES([ src/Makefile ])
 AC_CONFIG_FILES([ src/python/Makefile ])
 
+AC_CONFIG_FILES([xdrfile.pc])
 AC_OUTPUT
 
 
--- origsrc/xdrfile-1.1.4/src/Makefile.am	2011-04-08 05:28:03.000000000 +0900
+++ src/xdrfile-1.1.4/src/Makefile.am	2024-09-26 12:46:51.415360200 +0900
@@ -18,7 +18,7 @@ LDADD = $(lib_LTLIBRARIES)
 
 libxdrfile@LIBSUFFIX@_la_LIBADD         = 
 libxdrfile@LIBSUFFIX@_la_DEPENDENCIES   = 
-libxdrfile@LIBSUFFIX@_la_LDFLAGS        = -version-info @SHARED_VERSION_INFO@
+libxdrfile@LIBSUFFIX@_la_LDFLAGS        = -no-undefined -version-info @SHARED_VERSION_INFO@
 
 # Note: you don't have to list sources for "prog" if it is the single file prog.c
 
--- origsrc/xdrfile-1.1.4/xdrfile.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/xdrfile-1.1.4/xdrfile.pc.in	2024-09-26 12:48:46.288432300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: xdrfile
+Description: Library to read gromacs trajectory and topology files
+Version: @VERSION@
+Libs: -L${libdir} -lxdrfile
+Cflags: -I${includedir}
