--- origsrc/hitro-1.0/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/hitro-1.0/Makefile.am	2012-05-15 19:05:27.274225100 +0900
@@ -0,0 +1,11 @@
+
+include_HEADERS = hitro.h
+
+lib_LTLIBRARIES = libhitro.la
+libhitro_la_SOURCES = hitro.c
+libhitro_la_LDFLAGS = -no-undefined
+
+check_PROGRAMS = example
+example_SOURCES = example.c
+example_LDADD = libhitro.la
+
--- origsrc/hitro-1.0/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/hitro-1.0/configure.ac	2012-05-15 19:00:46.275670100 +0900
@@ -0,0 +1,26 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([hitro], [1.0])
+AC_CONFIG_SRCDIR([example.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([float.h stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([pow sqrt])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
