--- origsrc/libdfu-1.0.0/CMakeLists.txt	2025-10-12 06:56:31.000000000 +0900
+++ src/libdfu-1.0.0/CMakeLists.txt	2025-11-16 19:10:44.711993900 +0900
@@ -55,3 +55,10 @@ install(FILES ${CMAKE_CURRENT_SOURCE_DIR
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/quirks.h DESTINATION ${CMAKE_INSTALL_PREFIX}/include/dfu)
 
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/FindDFU.cmake DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATADIR}/cmake-${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}/Modules")
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libdfu.pc.in ${CMAKE_BINARY_DIR}/libdfu.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libdfu.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libdfu-1.0.0/libdfu.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libdfu-1.0.0/libdfu.pc.in	2025-11-16 19:10:58.872579800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libdfu
+Description: DFU (Device Firmware Update) library
+Version: @DFU_VERSION@
+Libs: -L${libdir} -ldfu
+Cflags: -I${includedir}
