--- origsrc/intx-0.14.0/test/benchmarks/CMakeLists.txt	2025-11-14 18:53:48.000000000 +0900
+++ src/intx-0.14.0/test/benchmarks/CMakeLists.txt	2026-01-12 23:13:14.789285000 +0900
@@ -13,6 +13,6 @@ add_executable(intx-bench
     bench_int128.cpp
     benchmarks.cpp
 )
-target_link_libraries(intx-bench PRIVATE intx intx::experimental intx::testutils benchmark::benchmark GMP::gmp)
+target_link_libraries(intx-bench PRIVATE intx intx::experimental intx::testutils benchmark::benchmark ${GMP_LIBRARY})
 target_compile_options(intx-bench PRIVATE $<$<CXX_COMPILER_ID:GNU,Clang>:-falign-functions=32>)
 set_target_properties(intx-bench PROPERTIES RUNTIME_OUTPUT_DIRECTORY ..)
