--- origsrc/yaramod-2.7.0/CMakeLists.txt	2019-05-01 05:43:54.000000000 +0900
+++ src/yaramod-2.7.0/CMakeLists.txt	2019-05-12 19:08:35.932741400 +0900
@@ -48,3 +48,10 @@ endif()
 if(YARAMOD_DOC)
 	add_subdirectory(doc)
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/yaramod.pc.in ${CMAKE_BINARY_DIR}/yaramod.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/yaramod.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/yaramod-2.7.0/deps/CMakeLists.txt	2019-05-01 05:43:54.000000000 +0900
+++ src/yaramod-2.7.0/deps/CMakeLists.txt	2019-05-12 18:50:13.077913200 +0900
@@ -1,4 +1,4 @@
-if(YARAMOD_TESTS)
+if(YARAMOD_TESTS AND NOT CYGWIN)
 	add_subdirectory(googletest)
 endif()
 if(YARAMOD_PYTHON)
--- origsrc/yaramod-2.7.0/src/CMakeLists.txt	2019-05-01 05:43:54.000000000 +0900
+++ src/yaramod-2.7.0/src/CMakeLists.txt	2019-05-12 19:03:32.433663400 +0900
@@ -65,6 +65,10 @@ if(NOT TARGET yaramod)
 	# Library.
 	add_library(yaramod ${SOURCES})
 
+	if(CYGWIN)
+		set_target_properties(yaramod PROPERTIES SOVERSION 0)
+	endif()
+
 	# Library includes and interface.
 	target_include_directories(yaramod PRIVATE "${YARAMOD_INCLUDE_DIR}")
 	target_include_directories(yaramod SYSTEM INTERFACE "${YARAMOD_INCLUDE_DIR}")
@@ -103,7 +107,7 @@ if(NOT TARGET yaramod)
 	install(
 		TARGETS yaramod
 		ARCHIVE DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}"
-		LIBRARY DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}"
+		RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}"
 	)
 	install(
 		DIRECTORY "${YARAMOD_INCLUDE_DIR}/"
@@ -118,10 +122,10 @@ if(NOT TARGET yaramod)
 	)
 	install(
 		DIRECTORY "${YARAMOD_DEPS_DIR}/optional_lite"
-		DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}"
+		DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}/yaramod"
 	)
 	install(
 		DIRECTORY "${YARAMOD_DEPS_DIR}/variant"
-		DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}"
+		DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}/yaramod"
 	)
 endif()
--- origsrc/yaramod-2.7.0/yaramod.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/yaramod-2.7.0/yaramod.pc.in	2019-05-12 19:08:35.838974400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: yaramod
+Description: Parsing of YARA rules into AST and building new rulesets in C++
+Version: @VERSION@
+Libs: -L${libdir} -lyaramod
+Cflags: -I${includedir}
