--- origsrc/wabt-1.0.12/CMakeLists.txt	2019-09-26 06:04:14.000000000 +0900
+++ src/wabt-1.0.12/CMakeLists.txt	2019-10-11 21:24:11.714395600 +0900
@@ -100,7 +100,11 @@ else ()
     -Wuninitialized
   )
 
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wold-style-cast")
+  if (CYGWIN)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wold-style-cast")
+  else()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wold-style-cast")
+  endif()
 
   if (WERROR)
     add_definitions(-Werror)
