--- origsrc/wabt-1.0.6/CMakeLists.txt	2018-10-06 08:10:30.000000000 +0900
+++ src/wabt-1.0.6/CMakeLists.txt	2018-10-27 10:12:53.957643800 +0900
@@ -94,7 +94,11 @@ else ()
     -Wuninitialized
   )
 
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wold-style-cast")
+  if (CYGWIN)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wold-style-cast")
+  else()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wold-style-cast")
+  endif()
 
   if (NOT WITH_EXCEPTIONS)
     add_definitions(-fno-exceptions)
