--- origsrc/wabt/CMakeLists.txt	2020-02-16 20:32:03.000000000 +0900
+++ src/wabt/CMakeLists.txt	2020-02-16 20:33:58.861647200 +0900
@@ -117,7 +117,11 @@ else ()
     -Wuninitialized
   )
 
+  if (CYGWIN)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wold-style-cast")
+  else()
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wold-style-cast")
+  endif()
 
   if (WERROR)
     add_definitions(-Werror)
