--- origsrc/wabt-1.0.11/CMakeLists.txt	2019-05-08 14:30:07.000000000 +0900
+++ src/wabt-1.0.11/CMakeLists.txt	2019-06-09 17:48:16.468047700 +0900
@@ -98,7 +98,11 @@ else ()
     -Wuninitialized
   )
 
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wold-style-cast")
+  if (CYGWIN)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wold-style-cast")
+  else()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wold-style-cast")
+  endif()
 
   if (WERROR)
     add_definitions(-Werror)
