--- origsrc/wabt-1.0.10/CMakeLists.txt	2019-03-07 09:41:26.000000000 +0900
+++ src/wabt-1.0.10/CMakeLists.txt	2019-03-30 12:10:21.949866900 +0900
@@ -99,7 +99,11 @@ else ()
     -Wuninitialized
   )
 
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wold-style-cast")
+  if (CYGWIN)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wold-style-cast")
+  else()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wold-style-cast")
+  endif()
 
   if (WERROR)
     add_definitions(-Werror)
