--- origsrc/libgnunetworker-1.1.1/src/Makefile.am	2025-04-06 10:26:24.000000000 +0900
+++ src/libgnunetworker-1.1.1/src/Makefile.am	2025-04-15 03:19:06.348263100 +0900
@@ -30,10 +30,10 @@ lib@PROJECT_NAME@_la_SOURCES = \
 	worker.c \
 	worker.h
 
-lib@PROJECT_NAME@_la_LDFLAGS = \
+lib@PROJECT_NAME@_la_LDFLAGS = -no-undefined \
 	-version-info $(LTVERSIONINFO)
 
-lib@PROJECT_NAME@_la_LIBADD = \
+lib@PROJECT_NAME@_la_LIBADD = $(LTLIBINTL) \
 	$(PTHREAD_LIBS) \
 	$(GNUNET_WORKER_LIBS)
 
