--- origsrc/libgnunetworker-1.0.0/src/Makefile.am	2022-07-14 01:02:46.000000000 +0900
+++ src/libgnunetworker-1.0.0/src/Makefile.am	2024-03-18 00:43:01.566136500 +0900
@@ -30,9 +30,11 @@ lib@PROJECT_NAME@_la_SOURCES = \
 	worker.h
 
 lib@PROJECT_NAME@_la_LDFLAGS = \
+	-no-undefined \
 	-version-info $(LTVERSIONINFO)
 
 lib@PROJECT_NAME@_la_LIBADD = \
+	$(LTLIBINTL) \
 	$(PTHREAD_LIBS) \
 	$(GNUNET_WORKER_LIBS)
 
--- origsrc/libgnunetworker-1.0.0/src/include/gnunet_worker_lib.h	2022-07-14 01:02:46.000000000 +0900
+++ src/libgnunetworker-1.0.0/src/include/gnunet_worker_lib.h	2024-03-17 19:56:49.232804500 +0900
@@ -38,7 +38,6 @@
 
 #include <time.h>
 #include <stdbool.h>
-#include <gnunet/platform.h>
 #include <gnunet/gnunet_common.h>
 
 
--- origsrc/libgnunetworker-1.0.0/src/worker.c	2022-07-14 01:02:46.000000000 +0900
+++ src/libgnunetworker-1.0.0/src/worker.c	2024-03-18 00:46:36.377121600 +0900
@@ -42,14 +42,17 @@
 #include <errno.h>
 #include <unistd.h>
 #include <pthread.h>
-#include <libintl.h>
-#include <gnunet/platform.h>
+#include <fcntl.h>
+#include <gnunet/gnunet_util_lib.h>
 #include <gnunet/gnunet_scheduler_lib.h>
 #include <gnunet/gnunet_network_lib.h>
 #include "include/gnunet_worker_lib.h"
 #include "requirement.h"
 #include "worker.h"
 
+#if !defined(_)
+#define _(str) gettext(str)
+#endif
 
 /*
 
--- origsrc/libgnunetworker-1.0.0/src/worker.h	2022-07-14 01:02:46.000000000 +0900
+++ src/libgnunetworker-1.0.0/src/worker.h	2024-03-17 19:58:02.267183800 +0900
@@ -40,7 +40,6 @@
 #include <stdatomic.h>
 #include <time.h>
 #include <pthread.h>
-#include <gnunet/platform.h>
 #include <gnunet/gnunet_common.h>
 #include <gnunet/gnunet_scheduler_lib.h>
 #include <gnunet/gnunet_network_lib.h>
