--- origsrc/libgnunetworker-1.0.1/src/Makefile.am	2024-09-12 13:21:28.000000000 +0900
+++ src/libgnunetworker-1.0.1/src/Makefile.am	2024-09-13 17:12:58.763605600 +0900
@@ -30,10 +30,10 @@ lib@PROJECT_NAME@_la_SOURCES = \
 	worker.c \
 	worker.h
 
-lib@PROJECT_NAME@_la_LDFLAGS = \
+lib@PROJECT_NAME@_la_LDFLAGS = -no-undefined \
 	-version-info $(LTVERSIONINFO)
 
-lib@PROJECT_NAME@_la_LIBADD = \
+lib@PROJECT_NAME@_la_LIBADD = $(LTLIBINTL) \
 	$(PTHREAD_LIBS) \
 	$(GNUNET_WORKER_LIBS)
 
