--- origsrc/libprelude-1.0.1/bindings/c++/Makefile.am	2012-04-02 18:41:15.000000000 +0900
+++ src/libprelude-1.0.1/bindings/c++/Makefile.am	2014-06-07 18:44:49.475597300 +0900
@@ -21,7 +21,7 @@ libpreludecpp_la_SOURCES = \
 
 libpreludecpp_la_LDFLAGS = -no-undefined -version-info @LIBPRELUDECPP_SONAME@
 libpreludecpp_la_LIBADD  = $(top_builddir)/src/.libs/libprelude.la
-libpreludecpp_la_CPPFLAGS = -I$(top_builddir) -I$(top_builddir)/src/include -I$(top_srcdir)/src/include -I$(top_builddir)/src/libprelude-error -I$(top_builddir)/bindings/c++/include -I$(top_srcdir)/bindings/c++/include 
+libpreludecpp_la_CPPFLAGS = -I$(top_builddir) -I$(top_builddir)/src/include -I$(top_srcdir)/src/include -I$(top_builddir)/src/libprelude-error -I$(top_srcdir)/src/libprelude-error -I$(top_builddir)/bindings/c++/include -I$(top_srcdir)/bindings/c++/include 
 
 endif
 
--- origsrc/libprelude-1.0.1/bindings/low-level/python/setup.py.in	2012-06-02 00:12:00.000000000 +0900
+++ src/libprelude-1.0.1/bindings/low-level/python/setup.py.in	2014-06-07 19:57:39.573253600 +0900
@@ -105,6 +105,6 @@ setup(name="prelude",
       py_modules=["prelude"],
       ext_modules=[Extension("_prelude",
                              ["_prelude.c"],
-                             extra_compile_args=split_args("-I@top_builddir@ -I@top_srcdir@/src/include -I@top_builddir@/src/include -I@top_builddir@/src/libprelude-error -I@top_srcdir@/bindings"),
+                             extra_compile_args=split_args("-I@top_builddir@ -I@top_srcdir@/src/include -I@top_builddir@/src/include -I@top_builddir@/src/libprelude-error -I@top_srcdir@/src/libprelude-error -I@top_srcdir@/bindings"),
                              library_dirs=[ "@top_builddir@/src/.libs/" ],
                              extra_link_args=split_args("-lprelude @LIBPRELUDE_LIBS@ @LIBADD_DL@ @LTLIBTHREAD@"))])
--- origsrc/libprelude-1.0.1/bindings/lua/Makefile.am	2012-04-02 18:41:15.000000000 +0900
+++ src/libprelude-1.0.1/bindings/lua/Makefile.am	2014-06-07 17:56:05.656174700 +0900
@@ -6,7 +6,7 @@ if HAVE_LUA
 
 AM_CPPFLAGS = -I@top_srcdir@ -I@top_builddir@/src/include -I@top_srcdir@/src/include -I@top_builddir@/src/libprelude-error -I@top_srcdir@/bindings/c++/include @LUA_CFLAGS@ -I@top_srcdir@/libmissing -I@top_builddir@/libmissing
 
-PreludeEasy_la_LDFLAGS = -module -avoid-version @LUA_LIBS@
+PreludeEasy_la_LDFLAGS = -no-undefined -module -avoid-version @LUA_LIBS@
 PreludeEasy_la_LIBADD = $(top_builddir)/bindings/c++/.libs/libpreludecpp.la
 PreludeEasy_la_SOURCES = PreludeEasy.cxx
 PreludeEasydir = $(libdir)
--- origsrc/libprelude-1.0.1/bindings/ruby/Makefile.am	2012-04-02 18:41:15.000000000 +0900
+++ src/libprelude-1.0.1/bindings/ruby/Makefile.am	2014-06-07 17:56:05.656174700 +0900
@@ -5,7 +5,7 @@ EXTRA_DIST = libpreludecpp-ruby.i Prelud
 
 rbexec_LTLIBRARIES = PreludeEasy.la
 PreludeEasy_la_CPPFLAGS = $(RUBY_INCLUDES) $(RUBY_CCFLAGS) -I@top_builddir@ -I$(top_srcdir)/src/include -I$(top_builddir)/src/include -I@top_builddir@/src/libprelude-error -I$(top_srcdir)/bindings/c++/include -I@top_srcdir@/libmissing -I@top_builddir@/libmissing
-PreludeEasy_la_LDFLAGS = -module -avoid-version
+PreludeEasy_la_LDFLAGS = -no-undefined -module -avoid-version
 PreludeEasy_la_LIBADD = $(top_builddir)/bindings/c++/.libs/libpreludecpp.la $(RUBY_LIBS)
 nodist_PreludeEasy_la_SOURCES = PreludeEasy.cxx
 
--- origsrc/libprelude-1.0.1/prelude-admin/Makefile.am	2012-04-02 18:41:15.000000000 +0900
+++ src/libprelude-1.0.1/prelude-admin/Makefile.am	2014-06-07 18:43:55.373533800 +0900
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = @PCFLAGS@ -I$(top_builddir)/src/include -I$(top_srcdir)/src/include -I$(top_builddir)/libmissing -I$(top_srcdir)/libmissing -I$(top_builddir)/src/libprelude-error @LIBPRELUDE_CFLAGS@ @LIBGNUTLS_CFLAGS@ @PRELUDE_ADMIN_CFLAGS@ 
+AM_CPPFLAGS = @PCFLAGS@ -I$(top_builddir)/src/include -I$(top_srcdir)/src/include -I$(top_builddir)/libmissing -I$(top_srcdir)/libmissing -I$(top_builddir)/src/libprelude-error -I$(top_srcdir)/src/libprelude-error @LIBPRELUDE_CFLAGS@ @LIBGNUTLS_CFLAGS@ @PRELUDE_ADMIN_CFLAGS@ 
 
 bin_PROGRAMS = prelude-admin
 bin_SCRIPTS = prelude-adduser
--- origsrc/libprelude-1.0.1/prelude-admin/prelude-admin.c	2012-06-02 00:12:22.000000000 +0900
+++ src/libprelude-1.0.1/prelude-admin/prelude-admin.c	2014-06-07 17:56:05.671799700 +0900
@@ -57,7 +57,6 @@
 #include <errno.h>
 #include <gnutls/gnutls.h>
 #include <gnutls/x509.h>
-#include <gnutls/extra.h>
 
 #include "common.h"
 #include "config-engine.h"
--- origsrc/libprelude-1.0.1/prelude-admin/server.c	2012-06-02 00:12:22.000000000 +0900
+++ src/libprelude-1.0.1/prelude-admin/server.c	2014-06-07 17:56:05.687424700 +0900
@@ -39,7 +39,6 @@
 
 #include <gcrypt.h>
 #include <gnutls/gnutls.h>
-#include <gnutls/extra.h>
 
 #include "prelude-client.h"
 #include "prelude-error.h"
--- origsrc/libprelude-1.0.1/src/idmef-criteria-string.lex.l	2012-06-02 00:12:31.000000000 +0900
+++ src/libprelude-1.0.1/src/idmef-criteria-string.lex.l	2014-06-07 18:27:59.083909000 +0900
@@ -41,7 +41,7 @@ int yylex(void);
 int yyget_lineno(void);
 FILE *yyget_in(void);
 FILE *yyget_out(void);
-int yyget_leng(void);
+yy_size_t yyget_leng(void);
 char *yyget_text(void);
 void yyset_lineno(int line_number);
 void yyset_in(FILE *in_str);
--- origsrc/libprelude-1.0.1/src/include/prelude-plugin.h	2012-06-02 00:12:30.000000000 +0900
+++ src/libprelude-1.0.1/src/include/prelude-plugin.h	2014-06-07 17:56:05.687424700 +0900
@@ -63,7 +63,9 @@ typedef struct {
 # define lt_preloaded_symbols lt__PROGRAM__LTX_preloaded_symbols
 #endif
 
+#if !defined(__CYGWIN__)
 extern const void *lt_preloaded_symbols[];
+#endif
 
 #define PRELUDE_PLUGIN_SET_PRELOADED_SYMBOLS()         \
         prelude_plugin_set_preloaded_symbols(lt_preloaded_symbols)
