--- origsrc/libprelude-1.0.1/bindings/lua/Makefile.am	2012-04-02 18:41:15.000000000 +0900
+++ src/libprelude-1.0.1/bindings/lua/Makefile.am	2012-12-16 22:56:14.583312500 +0900
@@ -6,7 +6,7 @@ if HAVE_LUA
 
 AM_CPPFLAGS = -I@top_srcdir@ -I@top_builddir@/src/include -I@top_srcdir@/src/include -I@top_builddir@/src/libprelude-error -I@top_srcdir@/bindings/c++/include @LUA_CFLAGS@ -I@top_srcdir@/libmissing -I@top_builddir@/libmissing
 
-PreludeEasy_la_LDFLAGS = -module -avoid-version @LUA_LIBS@
+PreludeEasy_la_LDFLAGS = -no-undefined -module -avoid-version @LUA_LIBS@
 PreludeEasy_la_LIBADD = $(top_builddir)/bindings/c++/.libs/libpreludecpp.la
 PreludeEasy_la_SOURCES = PreludeEasy.cxx
 PreludeEasydir = $(libdir)
--- origsrc/libprelude-1.0.1/bindings/ruby/Makefile.am	2012-04-02 18:41:15.000000000 +0900
+++ src/libprelude-1.0.1/bindings/ruby/Makefile.am	2012-12-16 22:56:30.489562500 +0900
@@ -5,7 +5,7 @@ EXTRA_DIST = libpreludecpp-ruby.i Prelud
 
 rbexec_LTLIBRARIES = PreludeEasy.la
 PreludeEasy_la_CPPFLAGS = $(RUBY_INCLUDES) $(RUBY_CCFLAGS) -I@top_builddir@ -I$(top_srcdir)/src/include -I$(top_builddir)/src/include -I@top_builddir@/src/libprelude-error -I$(top_srcdir)/bindings/c++/include -I@top_srcdir@/libmissing -I@top_builddir@/libmissing
-PreludeEasy_la_LDFLAGS = -module -avoid-version
+PreludeEasy_la_LDFLAGS = -no-undefined -module -avoid-version
 PreludeEasy_la_LIBADD = $(top_builddir)/bindings/c++/.libs/libpreludecpp.la $(RUBY_LIBS)
 nodist_PreludeEasy_la_SOURCES = PreludeEasy.cxx
 
--- origsrc/libprelude-1.0.1/prelude-admin/prelude-admin.c	2012-06-02 00:12:22.000000000 +0900
+++ src/libprelude-1.0.1/prelude-admin/prelude-admin.c	2012-12-16 23:27:24.501281200 +0900
@@ -57,7 +57,6 @@
 #include <errno.h>
 #include <gnutls/gnutls.h>
 #include <gnutls/x509.h>
-#include <gnutls/extra.h>
 
 #include "common.h"
 #include "config-engine.h"
--- origsrc/libprelude-1.0.1/prelude-admin/server.c	2012-06-02 00:12:22.000000000 +0900
+++ src/libprelude-1.0.1/prelude-admin/server.c	2012-12-16 23:27:49.658507800 +0900
@@ -39,7 +39,6 @@
 
 #include <gcrypt.h>
 #include <gnutls/gnutls.h>
-#include <gnutls/extra.h>
 
 #include "prelude-client.h"
 #include "prelude-error.h"
--- origsrc/libprelude-1.0.1/src/include/prelude-plugin.h	2012-06-02 00:12:30.000000000 +0900
+++ src/libprelude-1.0.1/src/include/prelude-plugin.h	2012-12-16 23:13:28.986632800 +0900
@@ -63,7 +63,9 @@ typedef struct {
 # define lt_preloaded_symbols lt__PROGRAM__LTX_preloaded_symbols
 #endif
 
+#if !defined(__CYGWIN__)
 extern const void *lt_preloaded_symbols[];
+#endif
 
 #define PRELUDE_PLUGIN_SET_PRELOADED_SYMBOLS()         \
         prelude_plugin_set_preloaded_symbols(lt_preloaded_symbols)
