--- origsrc/libprelude-3.1.0/bindings/c++/Makefile.am	2016-09-15 15:49:10.721000800 +0900
+++ src/libprelude-3.1.0/bindings/c++/Makefile.am	2016-12-01 08:21:22.650674200 +0900
@@ -22,7 +22,7 @@ libpreludecpp_la_SOURCES = \
 
 libpreludecpp_la_LDFLAGS = -no-undefined -version-info @LIBPRELUDECPP_SONAME@
 libpreludecpp_la_LIBADD  = $(top_builddir)/src/.libs/libprelude.la
-libpreludecpp_la_CPPFLAGS = -I$(top_builddir) -I$(top_builddir)/src/include -I$(top_srcdir)/src/include -I$(top_builddir)/src/libprelude-error -I$(top_builddir)/bindings/c++/include -I$(top_srcdir)/bindings/c++/include 
+libpreludecpp_la_CPPFLAGS = -I$(top_builddir) -I$(top_builddir)/src/include -I$(top_srcdir)/src/include -I$(top_builddir)/src/libprelude-error -I$(top_srcdir)/src/libprelude-error -I$(top_builddir)/bindings/c++/include -I$(top_srcdir)/bindings/c++/include 
 
 endif
 
--- origsrc/libprelude-3.1.0/bindings/lua/Makefile.am	2016-09-15 15:49:10.723000800 +0900
+++ src/libprelude-3.1.0/bindings/lua/Makefile.am	2016-12-01 08:55:09.086051600 +0900
@@ -7,15 +7,15 @@ if HAVE_LUA
 swigdatadir = @SWIG_DATA_DIR@/lua
 dist_swigdata_DATA = libpreludecpp-lua.i
 
-AM_CPPFLAGS = -I@top_srcdir@ -I@top_builddir@/src/include -I@top_srcdir@/src/include -I@top_builddir@/src/libprelude-error -I@top_srcdir@/bindings/c++/include @LUA_CFLAGS@ -I@top_srcdir@/libmissing -I@top_builddir@/libmissing
+AM_CPPFLAGS = -I@top_srcdir@ -I@top_builddir@/src/include -I@top_srcdir@/src/include -I@top_builddir@/src/libprelude-error -I@top_srcdir@/src/libprelude-error -I@top_srcdir@/bindings/c++/include @LUA_CFLAGS@ -I@top_srcdir@/libmissing -I@top_builddir@/libmissing
 
-prelude_la_LDFLAGS = -module -avoid-version @LUA_LIBS@
+prelude_la_LDFLAGS = -no-undefined -module -avoid-version @LUA_LIBS@
 prelude_la_LIBADD = $(top_builddir)/bindings/c++/.libs/libpreludecpp.la
 prelude_la_SOURCES = prelude.cxx
 preludedir = $(libdir)/lua/@LUA_VERSION@
 prelude_LTLIBRARIES = prelude.la
 
-prelude.cxx: $(top_srcdir)/bindings/c++/include/*.hxx $(top_srcdir)/bindings/libpreludecpp.i libpreludecpp-lua.i 
+prelude.cxx: $(top_srcdir)/bindings/c++/include/*.hxx $(top_srcdir)/bindings/libpreludecpp.i libpreludecpp-lua.i
 if HAVE_SWIG
 	$(SWIG) -c++ -DSWIG_COMPILE_LIBPRELUDE -I$(top_srcdir)/bindings -I$(top_srcdir)/bindings/c++/include -I$(top_srcdir)/src/include -I$(top_srcdir)/src/libprelude-error -lua -o $@ $(top_srcdir)/bindings/libpreludecpp.i
 endif
--- origsrc/libprelude-3.1.0/bindings/perl/Makefile.PL.in	2016-09-15 15:49:10.725000800 +0900
+++ src/libprelude-3.1.0/bindings/perl/Makefile.PL.in	2016-12-01 09:39:04.850338400 +0900
@@ -25,9 +25,9 @@ use File::Copy;
 use Cwd "abs_path";
 
 my %attributs = (NAME => 'Prelude',
-		 INC => '-I@top_builddir@ -I@top_builddir@/src/include -I@top_srcdir@/src/include -I@top_builddir@/src/libprelude-error -I@top_srcdir@/bindings/c++/include',
+		 INC => '-I@top_builddir@ -I@top_builddir@/src/include -I@top_srcdir@/src/include -I@top_builddir@/src/libprelude-error -I@top_srcdir@/src/libprelude-error -I@top_srcdir@/bindings/c++/include',
 		 LIBS => ["-L@top_builddir@/src/.libs -lprelude -L@top_builddir@/bindings/c++/.libs -lpreludecpp", "-L$ENV{LIBDIR} -lprelude @LIBPRELUDE_LIBS@ @LIBADD_DL@ @LTLIBTHREAD@" ],
-		 LDDLFLAGS => "-L@top_builddir@/src/.libs $Config{lddlflags} @LDFLAGS@,-rpath");
+		 LDDLFLAGS => "-L@top_builddir@/src/.libs $Config{lddlflags} -Wl,-rpath");
 
 if ( abs_path("@top_srcdir@") ne abs_path("@top_builddir@") ) {
     foreach my $filename ( qw/Prelude.cxx Prelude.pm/ ) {
--- origsrc/libprelude-3.1.0/bindings/python/setup.py.in	2016-09-15 15:49:10.729000800 +0900
+++ src/libprelude-3.1.0/bindings/python/setup.py.in	2016-12-01 08:59:56.644147600 +0900
@@ -113,6 +113,6 @@ setup(name="prelude",
       py_modules=["prelude"],
       ext_modules=[Extension("_prelude",
                              ["_prelude.cxx"],
-                             extra_compile_args=split_args("-I@top_builddir@ -I@top_srcdir@/src/include -I@top_builddir@/src/include -I@top_builddir@/src/libprelude-error -I@top_srcdir@/bindings/c++/include"),
+                             extra_compile_args=split_args("-I@top_builddir@ -I@top_srcdir@/src/include -I@top_builddir@/src/include -I@top_builddir@/src/libprelude-error -I@top_srcdir@/src/libprelude-error -I@top_srcdir@/bindings/c++/include"),
                              library_dirs=[ "@top_builddir@/src/.libs/", "@top_builddir@/bindings/c++/.libs/" ],
                              extra_link_args=split_args("-lpreludecpp -lprelude @LIBPRELUDE_LIBS@ @LIBADD_DL@ @LTLIBTHREAD@"))])
--- origsrc/libprelude-3.1.0/bindings/ruby/Makefile.am	2016-09-15 15:49:10.729000800 +0900
+++ src/libprelude-3.1.0/bindings/ruby/Makefile.am	2016-12-01 09:10:10.193993000 +0900
@@ -7,12 +7,12 @@ dist_swigdata_DATA = libpreludecpp-ruby.
 EXTRA_DIST = libpreludecpp-ruby.i Prelude.cxx
 
 rbexec_LTLIBRARIES = Prelude.la
-Prelude_la_CPPFLAGS = $(RUBY_INCLUDES) $(RUBY_CCFLAGS) -I@top_builddir@ -I$(top_srcdir)/src/include -I$(top_builddir)/src/include -I@top_builddir@/src/libprelude-error -I$(top_srcdir)/bindings/c++/include
-Prelude_la_LDFLAGS = -module -avoid-version
+Prelude_la_CPPFLAGS = $(RUBY_INCLUDES) $(RUBY_CCFLAGS) -I@top_builddir@ -I$(top_srcdir)/src/include -I$(top_builddir)/src/include -I@top_builddir@/src/libprelude-error -I@top_srcdir@/src/libprelude-error -I$(top_srcdir)/bindings/c++/include
+Prelude_la_LDFLAGS = -no-undefined -module -avoid-version
 Prelude_la_LIBADD = $(top_builddir)/bindings/c++/.libs/libpreludecpp.la $(RUBY_LIBS)
 nodist_Prelude_la_SOURCES = Prelude.cxx
 
-Prelude.cxx: $(top_srcdir)/bindings/c++/include/*.hxx $(top_srcdir)/bindings/libpreludecpp.i libpreludecpp-ruby.i 
+Prelude.cxx: $(top_srcdir)/bindings/c++/include/*.hxx $(top_srcdir)/bindings/libpreludecpp.i libpreludecpp-ruby.i
 if HAVE_SWIG
 	$(SWIG) -c++ -DSWIG_COMPILE_LIBPRELUDE -I$(top_srcdir)/bindings -I$(top_srcdir)/bindings/c++/include -I$(top_srcdir)/src/include -I$(top_srcdir)/src/libprelude-error -ruby -o $@ $(top_srcdir)/bindings/libpreludecpp.i
 endif
--- origsrc/libprelude-3.1.0/configure.ac	2016-09-15 15:49:10.732000800 +0900
+++ src/libprelude-3.1.0/configure.ac	2016-12-01 08:46:41.629970400 +0900
@@ -71,8 +71,8 @@ dnl * Setup GnuLib, libtool and libltdl.
 dnl **********************************************************
 
 dnl Provide replacement for system missing this GnuLib required macro.
-m4_ifndef([AS_VAR_COPY], [m4_define([AS_VAR_COPY], 
-       [AS_LITERAL_IF([$1[]$2], [$1=$$2], [eval $1=\$$2])])]) 
+m4_ifndef([AS_VAR_COPY], [m4_define([AS_VAR_COPY],
+       [AS_LITERAL_IF([$1[]$2], [$1=$$2], [eval $1=\$$2])])])
 
 gl_EARLY
 LT_CONFIG_LTDL_DIR([libltdl])
--- origsrc/libprelude-3.1.0/prelude-admin/Makefile.am	2016-09-15 15:49:10.785000800 +0900
+++ src/libprelude-3.1.0/prelude-admin/Makefile.am	2016-12-01 08:19:46.956478000 +0900
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = @PCFLAGS@ -I$(top_builddir)/src/include -I$(top_srcdir)/src/include -I$(top_builddir)/libmissing -I$(top_srcdir)/libmissing -I$(top_builddir)/src/libprelude-error @LIBPRELUDE_CFLAGS@ @PRELUDE_ADMIN_CFLAGS@
+AM_CPPFLAGS = @PCFLAGS@ -I$(top_builddir)/src/include -I$(top_srcdir)/src/include -I$(top_builddir)/libmissing -I$(top_srcdir)/libmissing -I$(top_builddir)/src/libprelude-error -I$(top_srcdir)/src/libprelude-error @LIBPRELUDE_CFLAGS@ @PRELUDE_ADMIN_CFLAGS@
 
 bin_PROGRAMS = prelude-admin
 bin_SCRIPTS = prelude-adduser
@@ -13,7 +13,7 @@ prelude_admin_LDADD = $(top_builddir)/sr
 prelude_admin_SOURCES = 		\
 	prelude-admin.c		\
 	server.c			\
-	tls-register.c			
+	tls-register.c
 
 noinst_HEADERS = tls-register.h server.h
 
--- origsrc/libprelude-3.1.0/src/Makefile.am	2016-09-15 15:49:10.786000800 +0900
+++ src/libprelude-3.1.0/src/Makefile.am	2016-12-01 08:46:45.352651900 +0900
@@ -11,7 +11,7 @@ libpreludeincludedir = $(includedir)/
 libprelude_la_LIBADD = @LIBPRELUDE_LIBS@ $(LIBLTDL) $(top_builddir)/libmissing/libmissing.la libprelude-error/libprelude-error.la
 libprelude_la_LDFLAGS = $(GETADDRINFO_LIB) $(GETHOSTNAME_LIB) $(HOSTENT_LIB) $(INET_NTOP_LIB) $(INET_PTON_LIB) $(LIBSOCKET) $(LIB_POLL) $(LIB_SELECT) $(LTLIBINTL) $(LTLIBTHREAD) $(SERVENT_LIB) -no-undefined -version-info @LIBPRELUDE_SONAME@ -export-symbols-regex "^(idmef|_?prelude)_"
 lib_LTLIBRARIES = libprelude.la
-BUILT_SOURCES = $(LTDLDEPS) 
+BUILT_SOURCES = $(LTDLDEPS)
 
 AM_YFLAGS = -d
 LEX_OUTPUT_ROOT = lex.yy
@@ -61,6 +61,6 @@ libprelude_la_SOURCES = 		\
 	idmef-time.c			\
 	idmef-tree-wrap.c		\
 	idmef-value.c			\
-	idmef-value-type.c		
+	idmef-value-type.c
 
 -include $(top_srcdir)/git.mk
