--- origsrc/libprelude-5.1.1/bindings/lua/Makefile.am	2019-10-15 05:33:02.485000000 +0900
+++ src/libprelude-5.1.1/bindings/lua/Makefile.am	2020-02-02 22:10:11.561356400 +0900
@@ -9,7 +9,7 @@ dist_swigdata_DATA = libpreludecpp-lua.i
 
 AM_CPPFLAGS = -I@top_srcdir@ -I@top_builddir@/src/include -I@top_srcdir@/src/include -I@top_builddir@/src/libprelude-error -I@top_srcdir@/bindings/c++/include @LUA_CFLAGS@ -I@top_srcdir@/libmissing -I@top_builddir@/libmissing
 
-prelude_la_LDFLAGS = -module -avoid-version @LUA_LIBS@
+prelude_la_LDFLAGS = -no-undefined -module -avoid-version @LUA_LIBS@
 prelude_la_LIBADD = $(top_builddir)/bindings/c++/.libs/libpreludecpp.la
 prelude_la_SOURCES = prelude.cxx
 preludedir = $(libdir)/lua/@LUA_VERSION@
--- origsrc/libprelude-5.1.1/bindings/perl/Makefile.PL.in	2019-10-15 05:33:03.107000000 +0900
+++ src/libprelude-5.1.1/bindings/perl/Makefile.PL.in	2020-02-02 22:10:11.576988000 +0900
@@ -27,7 +27,7 @@ use Cwd "abs_path";
 my %attributs = (NAME => 'Prelude',
 		 INC => '-I@top_builddir@ -I@top_builddir@/src/include -I@top_srcdir@/src/include -I@top_builddir@/src/libprelude-error -I@top_srcdir@/bindings/c++/include',
 		 LIBS => ["-L@top_builddir@/src/.libs -lprelude -L@top_builddir@/bindings/c++/.libs -lpreludecpp", "-L$ENV{LIBDIR} -lprelude @LIBPRELUDE_LIBS@ @LIBADD_DL@ @LTLIBTHREAD@" ],
-		 LDDLFLAGS => "-L@top_builddir@/src/.libs $Config{lddlflags} @LDFLAGS@,-rpath");
+		 LDDLFLAGS => "-L@top_builddir@/src/.libs $Config{lddlflags} -Wl,-rpath");
 
 if ( abs_path("@top_srcdir@") ne abs_path("@top_builddir@") ) {
     foreach my $filename ( qw/Prelude.cxx Prelude.pm/ ) {
--- origsrc/libprelude-5.1.1/bindings/ruby/Makefile.am	2019-10-15 05:33:02.504000000 +0900
+++ src/libprelude-5.1.1/bindings/ruby/Makefile.am	2020-02-02 22:10:11.576988000 +0900
@@ -8,7 +8,7 @@ EXTRA_DIST = libpreludecpp-ruby.i Prelud
 
 rbexec_LTLIBRARIES = Prelude.la
 Prelude_la_CPPFLAGS = $(RUBY_INCLUDES) $(RUBY_CCFLAGS) -I@top_builddir@ -I$(top_srcdir)/src/include -I$(top_builddir)/src/include -I@top_builddir@/src/libprelude-error -I$(top_srcdir)/bindings/c++/include
-Prelude_la_LDFLAGS = -module -avoid-version
+Prelude_la_LDFLAGS = -no-undefined -module -avoid-version
 Prelude_la_LIBADD = $(top_builddir)/bindings/c++/.libs/libpreludecpp.la $(RUBY_LIBS)
 nodist_Prelude_la_SOURCES = Prelude.cxx
 
