--- origsrc/roundup-0.0.6/roundup.sh	2015-01-27 06:36:11.000000000 +0900
+++ src/roundup-0.0.6/roundup.sh	2025-01-24 13:59:23.901030900 +0900
@@ -35,7 +35,7 @@ set -e
 
 # The current version is set during `make version`.  Do not modify this line in
 # anyway unless you know what you're doing.
-ROUNDUP_VERSION="0.0.5"
+ROUNDUP_VERSION="0.0.6"
 export ROUNDUP_VERSION
 
 # Usage is defined in a specific comment syntax. It is `grep`ed out of this file
