--- origsrc/zn_poly-0.9/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/zn_poly-0.9/Makefile.am	2024-05-03 22:08:34.515459900 +0900
@@ -0,0 +1,16 @@
+# -*- Makefile -*-
+
+AM_CPPFLAGS = -I${top_srcdir}/include
+
+pkginclude_HEADERS = include/zn_poly.h include/wide_arith.h
+noinst_HEADERS = include/support.h include/profiler.h include/zn_poly_internal.h
+libsrcs = src/array.c src/invert.c src/ks_support.c src/mulmid.c src/mulmid_ks.c src/misc.c src/mpn_mulmid.c src/mul.c src/mul_fft.c src/mul_fft_dft.c src/mul_ks.c src/nuss.c src/pack.c src/pmf.c src/pmfvec_fft.c src/tuning.c src/zn_mod.c
+
+lib_LTLIBRARIES = libzn_poly.la
+
+libzn_poly_la_SOURCES = ${libsrcs}
+libzn_poly_la_LDFLAGS = -no-undefined -release ${PACKAGE_VERSION}
+libzn_poly_la_LIBADD  = -lgmp
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/zn_poly-0.9/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/zn_poly-0.9/configure.ac	2024-05-03 22:08:34.531084200 +0900
@@ -0,0 +1,10 @@
+AC_INIT([zn_poly], [0.9])
+AC_CONFIG_AUX_DIR([build-aux])
+AM_INIT_AUTOMAKE([-Wall foreign])
+AC_PROG_INSTALL
+AC_PROG_CC
+AC_DISABLE_STATIC
+m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
+LT_INIT
+AC_CONFIG_FILES([zn_poly.pc Makefile])
+AC_OUTPUT
--- origsrc/zn_poly-0.9/zn_poly.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/zn_poly-0.9/zn_poly.pc.in	2024-05-03 22:08:34.531084200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: zn_poly
+Description: Library for polynomial arithmetic
+Version: @VERSION@
+Libs: -L${libdir} -lzn_poly
+Cflags: -I${includedir}/zn_poly
