--- origsrc/libarms-6.00/Makefile.am	2017-02-01 14:56:21.000000000 +0900
+++ src/libarms-6.00/Makefile.am	2017-11-27 08:59:10.151584200 +0900
@@ -5,10 +5,10 @@
 # revision: increment if source change.
 # age: increment if function add.
 
-libarms_la_LDFLAGS= -version-info 0:3:0
+libarms_la_LDFLAGS= -no-undefined -version-info 0:3:0
 
 ACLOCAL_AMFLAGS=-I m4
-AM_CPPFLAGS= -Wall
+AM_CPPFLAGS= -Wall -I$(top_srcdir)/include
 
 lib_LTLIBRARIES=libarms.la
 libarms_la_SOURCES=$(LIBARMS_SRCS)
@@ -321,3 +321,6 @@ doc/_static/websupport.js \
 doc/.buildinfo \
 doc/objects.inv \
 doc/searchindex.js
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libarms-6.00/configure.ac	2017-07-20 06:20:11.000000000 +0900
+++ src/libarms-6.00/configure.ac	2017-11-27 08:59:10.151584200 +0900
@@ -147,6 +147,6 @@ AC_ARG_ENABLE(debug,
               [enable debugging]),
               CPPFLAGS="$CPPFLAGS -DARMS_DEBUG",)
 
-AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([libarms.pc Makefile])
 AC_OUTPUT_COMMANDS([echo timestamp > stamp-h1])
 AC_OUTPUT
--- origsrc/libarms-6.00/libarms.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libarms-6.00/libarms.pc.in	2017-11-27 08:59:10.151584200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libarms
+Description: Client library for SMFv2 system
+Version: @VERSION@
+Libs: -L${libdir} -larms
+Cflags: -I${includedir}
