--- origsrc/libarms-5.01/Makefile.am	2012-01-31 13:24:02.000000000 +0900
+++ src/libarms-5.01/Makefile.am	2012-08-05 06:55:35.602198500 +0900
@@ -5,7 +5,9 @@
 # revision: increment if source change.
 # age: increment if function add.
 
-libarms_la_LDFLAGS= -version-info 0:1:0
+INCLUDES = -I$(top_srcdir)/include
+
+libarms_la_LDFLAGS= -version-info 0:1:0 -no-undefined
 
 ACLOCAL_AMFLAGS=-I m4
 AM_CPPFLAGS= -Wall
