--- origsrc/mercator-0.3.1/Doxyfile.in	2011-03-25 21:53:24.000000000 +0900
+++ src/mercator-0.3.1/Doxyfile.in	2012-07-26 19:05:33.454056300 +0900
@@ -574,8 +574,8 @@ WARN_LOGFILE           =
 # directories like "/usr/src/myproject". Separate the files or directories
 # with spaces.
 
-INPUT                  = ./ \
-                         Mercator/
+INPUT                  = @top_srcdir@/ \
+                         @top_srcdir@/Mercator/
 
 # This tag can be used to specify the character encoding of the source files
 # that doxygen parses. Internally doxygen uses the UTF-8 encoding, which is
@@ -1146,7 +1146,7 @@ RTF_EXTENSIONS_FILE    =
 # If the GENERATE_MAN tag is set to YES (the default) Doxygen will
 # generate man pages
 
-GENERATE_MAN           = NO
+GENERATE_MAN           = YES
 
 # The MAN_OUTPUT tag is used to specify where the man pages will be put.
 # If a relative path is entered the value of OUTPUT_DIRECTORY will be
--- origsrc/mercator-0.3.1/Mercator/Makefile.am	2011-02-14 19:51:54.000000000 +0900
+++ src/mercator-0.3.1/Mercator/Makefile.am	2012-07-26 19:05:47.922806300 +0900
@@ -2,7 +2,7 @@ AM_CPPFLAGS = -I$(top_builddir) -I$(top_
 
 lib_LTLIBRARIES = libmercator-0.3.la
 
-libmercator_0_3_la_LDFLAGS = -version-info @MERCATOR_VERSION_INFO@
+libmercator_0_3_la_LDFLAGS = -version-info @MERCATOR_VERSION_INFO@ -no-undefined
 
 libmercator_0_3_la_SOURCES = \
 	iround.h \
--- origsrc/mercator-0.3.1/configure.ac	2012-06-27 02:16:50.000000000 +0900
+++ src/mercator-0.3.1/configure.ac	2012-07-26 19:06:00.032181300 +0900
@@ -90,5 +90,6 @@ AC_CONFIG_FILES([
 	mercator.spec
 	mingw32-mercator.spec
 	mercator-0.3.pc
+	Doxyfile
 ])
 AC_OUTPUT
