--- origsrc/mercator-0.3.0/Doxyfile	2009-08-16 11:49:29.000000000 +0900
+++ src/mercator-0.3.0/Doxyfile	2011-05-11 10:57:15.313759700 +0900
@@ -534,8 +534,8 @@ WARN_LOGFILE           =
 # directories like "/usr/src/myproject". Separate the files or directories 
 # with spaces.
 
-INPUT                  = ./ \
-                         Mercator/
+INPUT                  = ../src/mercator-0.3.0/ \
+                         ../src/mercator-0.3.0/Mercator/
 
 # This tag can be used to specify the character encoding of the source files 
 # that doxygen parses. Internally doxygen uses the UTF-8 encoding, which is 
@@ -1012,7 +1012,7 @@ RTF_EXTENSIONS_FILE    =
 # If the GENERATE_MAN tag is set to YES (the default) Doxygen will 
 # generate man pages
 
-GENERATE_MAN           = NO
+GENERATE_MAN           = YES
 
 # The MAN_OUTPUT tag is used to specify where the man pages will be put. 
 # If a relative path is entered the value of OUTPUT_DIRECTORY will be 
--- origsrc/mercator-0.3.0/Makefile.am	2011-01-08 08:57:48.000000000 +0900
+++ src/mercator-0.3.0/Makefile.am	2011-05-11 10:57:34.110788500 +0900
@@ -7,5 +7,5 @@ EXTRA_DIST = mercator.spec Doxyfile auto
 
 docs:
 	@echo "running doxygen..."
-	@doxygen Doxyfile
+	@doxygen $(top_srcdir)/Doxyfile
 	@echo "documentation is in doc/."
--- origsrc/mercator-0.3.0/Mercator/Makefile.am	2011-02-13 04:23:48.000000000 +0900
+++ src/mercator-0.3.0/Mercator/Makefile.am	2011-05-11 10:57:53.438580500 +0900
@@ -2,7 +2,7 @@ AM_CPPFLAGS = -I$(top_builddir) -I$(top_
 
 lib_LTLIBRARIES = libmercator-0.3.la
 
-libmercator_0_3_la_LDFLAGS = -version-info @MERCATOR_VERSION_INFO@
+libmercator_0_3_la_LDFLAGS = -version-info @MERCATOR_VERSION_INFO@ -no-undefined
 
 libmercator_0_3_la_SOURCES = \
 	iround.h \
