--- origsrc/mercator-0.2.9/Doxyfile	2010-09-18 20:42:53.000000000 +0900
+++ src/mercator-0.2.9/Doxyfile	2010-12-30 20:50:37.123714000 +0900
@@ -534,8 +534,8 @@ WARN_LOGFILE           =
 # directories like "/usr/src/myproject". Separate the files or directories 
 # with spaces.
 
-INPUT                  = ./ \
-                         Mercator/
+INPUT                  = ../src/mercator-0.2.9/ \
+                         ../src/mercator-0.2.9/Mercator/
 
 # This tag can be used to specify the character encoding of the source files 
 # that doxygen parses. Internally doxygen uses the UTF-8 encoding, which is 
@@ -1012,7 +1012,7 @@ RTF_EXTENSIONS_FILE    =
 # If the GENERATE_MAN tag is set to YES (the default) Doxygen will 
 # generate man pages
 
-GENERATE_MAN           = NO
+GENERATE_MAN           = YES
 
 # The MAN_OUTPUT tag is used to specify where the man pages will be put. 
 # If a relative path is entered the value of OUTPUT_DIRECTORY will be 
--- origsrc/mercator-0.2.9/Makefile.am	2010-09-18 20:42:53.000000000 +0900
+++ src/mercator-0.2.9/Makefile.am	2010-12-30 20:27:43.371576000 +0900
@@ -7,5 +7,5 @@ EXTRA_DIST = mercator.spec Doxyfile auto
 
 docs:
 	@echo "running doxygen..."
-	@doxygen Doxyfile
+	@doxygen $(top_srcdir)/Doxyfile
 	@echo "documentation is in doc/."
--- origsrc/mercator-0.2.9/Mercator/Makefile.am	2010-11-13 03:35:34.000000000 +0900
+++ src/mercator-0.2.9/Mercator/Makefile.am	2010-12-30 20:26:00.533702400 +0900
@@ -2,7 +2,7 @@ INCLUDES = -I$(top_builddir) -I$(top_src
 
 lib_LTLIBRARIES = libmercator-0.2.la
 
-libmercator_0_2_la_LDFLAGS = -version-info @MERCATOR_VERSION_INFO@
+libmercator_0_2_la_LDFLAGS = -version-info @MERCATOR_VERSION_INFO@ -no-undefined
 
 libmercator_0_2_la_SOURCES = \
 	iround.h \
--- origsrc/mercator-0.2.9/configure.ac	2010-11-13 03:49:20.000000000 +0900
+++ src/mercator-0.2.9/configure.ac	2010-12-30 20:25:41.085737600 +0900
@@ -26,8 +26,7 @@ AC_SUBST(MERCATOR_VERSION_INFO)
 
 dnl Test for GNU libtool
 
-AC_DISABLE_STATIC
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_PROG_CXX
 
 AC_ARG_ENABLE(debug,
