--- origsrc/cwc-1.0.2/src/cwc.cc	2008-06-18 02:43:28.000000000 +0900
+++ src/cwc-1.0.2/src/cwc.cc	2012-04-03 17:56:16.322426700 +0900
@@ -19,6 +19,8 @@
  * 02111-1307, USA.
  **/
 
+#include <cstdlib>
+
 #include <unistd.h>
 #include <fcntl.h>
 #include <stdio.h>
--- origsrc/cwc-1.0.2/src/dict.cc	2008-06-18 02:43:28.000000000 +0900
+++ src/cwc-1.0.2/src/dict.cc	2012-04-03 17:55:42.338051700 +0900
@@ -19,6 +19,8 @@
  * 02111-1307, USA.
  **/
 
+#include <cstring>
+
 #include <string>
 #include <fstream>
 #include <iostream>
--- origsrc/cwc-1.0.2/src/symbol.cc	2008-06-18 02:43:28.000000000 +0900
+++ src/cwc-1.0.2/src/symbol.cc	2012-04-03 17:49:25.791176700 +0900
@@ -19,6 +19,8 @@
  * 02111-1307, USA.
  **/
 
+#include <cstdlib>
+
 #include <iostream>
 
 using std::cout;
