--- origsrc/darts-0.32/darts.h	2008-03-23 01:34:02.000000000 +0900
+++ src/darts-0.32/darts.h	2015-05-08 16:14:59.883774600 +0900
@@ -501,7 +501,7 @@ namespace Darts {
 #define DARTS_ARRAY_SIZE_IS_DEFINED 1
 #endif
 
-#if 4 == 4 && !defined(DARTS_ARRAY_SIZE_IS_DEFINED)
+#if 4 == 8 && !defined(DARTS_ARRAY_SIZE_IS_DEFINED)
   typedef Darts::DoubleArrayImpl<char, unsigned char, long,
                                  unsigned long> DoubleArray;
 #define DARTS_ARRAY_SIZE_IS_DEFINED 1
