--- origsrc/libdynarr-0.1.1/configure.ac	2004-03-04 00:48:48.000000000 +0900
+++ src/libdynarr-0.1.1/configure.ac	2010-12-09 12:08:13.189284100 +0900
@@ -9,7 +9,7 @@ AC_CONFIG_HEADER([config.h])
 
 # Checks for programs.
 AC_PROG_CC
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 # Checks for libraries.
 
--- origsrc/libdynarr-0.1.1/lib/Makefile.am	2004-03-04 00:45:32.000000000 +0900
+++ src/libdynarr-0.1.1/lib/Makefile.am	2010-12-09 12:07:55.431009100 +0900
@@ -1,4 +1,4 @@
 lib_LTLIBRARIES = libdynarr.la
 libdynarr_la_SOURCES = dynarr.c ../include/dynarr.h
-libdynarr_la_LDFLAGS = -version-info 1:0:1
+libdynarr_la_LDFLAGS = -version-info 1:0:1 -no-undefined
 LIBS = -lpthread
