--- origsrc/libosip2-5.3.1/src/osipparser2/osip_port.c	2021-05-04 04:43:11.000000000 +0900
+++ src/libosip2-5.3.1/src/osipparser2/osip_port.c	2024-09-19 18:40:06.463726200 +0900
@@ -877,7 +877,7 @@ int __osip_port_gettimeofday(struct time
   tp->tv_usec = timebuffer.millitm * 1000;
   return OSIP_SUCCESS;
 }
-#elif defined(__linux)
+#elif defined(__linux) || defined(__CYGWIN__)
 #include <sys/time.h>
 #define __osip_port_gettimeofday gettimeofday
 #elif defined(__APPLE__)
--- origsrc/libosip2-5.3.1/src/test/Makefile.am	2020-07-26 01:00:35.000000000 +0900
+++ src/libosip2-5.3.1/src/test/Makefile.am	2024-09-19 18:40:06.463726200 +0900
@@ -65,7 +65,7 @@ check:
 	@echo " *******************************"
 	@echo " ****** starting tests! ********"
 	@echo " *******************************"
-	@./$(top_srcdir)/src/test/tst ./$(top_srcdir)/src/test/res -c
+	@$(top_srcdir)/src/test/tst $(top_srcdir)/src/test/res -c
 
 	@echo ""
 	@echo "In case you have a doubt, send the generated"
