--- origsrc/libosip2-5.1.1/src/test/Makefile.am	2020-01-12 07:39:09.000000000 +0900
+++ src/libosip2-5.1.1/src/test/Makefile.am	2020-01-19 19:23:05.460575500 +0900
@@ -56,7 +56,7 @@ check:
 	@echo " *******************************"
 	@echo " ****** starting tests! ********"
 	@echo " *******************************"
-	@./$(top_srcdir)/src/test/tst ./$(top_srcdir)/src/test/res -c
+	@$(top_srcdir)/src/test/tst $(top_srcdir)/src/test/res -c
 
 	@echo ""
 	@echo "In case you have a doubt, send the generated"
