--- origsrc/libosip2-3.3.0/Makefile.am	2008-11-16 01:48:10.000000000 +0900
+++ src/libosip2-3.3.0/Makefile.am	2010-12-07 00:20:28.147559900 +0900
@@ -24,3 +24,5 @@ indent:
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libosip2.pc
+
+ACLOCAL_AMFLAGS = -I scripts
--- origsrc/libosip2-3.3.0/configure.in	2009-03-04 20:12:47.000000000 +0900
+++ src/libosip2-3.3.0/configure.in	2010-12-07 00:21:18.419847900 +0900
@@ -44,7 +44,7 @@ dnl Initialize automake stuff
 AC_CONFIG_AUX_DIR(scripts)
 AM_CONFIG_HEADER(config.h)
 AM_INIT_AUTOMAKE($PACKAGE, $VERSION)
-AM_ACLOCAL_INCLUDE(scripts)
+dnl AM_ACLOCAL_INCLUDE(scripts)
 
 AC_CANONICAL_HOST                                                               
 
@@ -56,9 +56,8 @@ AC_PROG_INSTALL
 
 
 dnl Initialize libtool
-AC_LIBTOOL_WIN32_DLL
+LT_INIT(win32-dll)
 LT_AC_PROG_SED
-AC_PROG_LIBTOOL
 
 dnl declare --enable-* args and collect ac_help strings
 
--- origsrc/libosip2-3.3.0/src/test/Makefile.am	2008-09-18 20:49:33.000000000 +0900
+++ src/libosip2-3.3.0/src/test/Makefile.am	2010-12-07 00:20:28.157574300 +0900
@@ -49,7 +49,7 @@ check:
 	@echo " *******************************"
 	@echo " ****** starting tests! ********"
 	@echo " *******************************"
-	@./$(top_srcdir)/src/test/tst ./$(top_srcdir)/src/test/res -c
+	@$(top_srcdir)/src/test/tst $(top_srcdir)/src/test/res -c
 
 	@echo ""
 	@echo "In case you have a doubt, send the generated"
