--- origsrc/libosip2-4.1.0/Makefile.am	2012-11-17 03:05:57.000000000 +0900
+++ src/libosip2-4.1.0/Makefile.am	2014-08-01 08:27:55.255918400 +0900
@@ -27,3 +27,5 @@ indent:
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libosip2.pc
+
+ACLOCAL_AMFLAGS = -I scripts
--- origsrc/libosip2-4.1.0/configure.ac	2013-12-19 02:51:42.000000000 +0900
+++ src/libosip2-4.1.0/configure.ac	2014-08-01 08:27:42.123577600 +0900
@@ -44,7 +44,6 @@ dnl Initialize automake stuff
 AC_CONFIG_AUX_DIR(scripts)
 AC_CONFIG_HEADERS(osip-config.h)
 AM_INIT_AUTOMAKE($PACKAGE, $VERSION)
-AM_ACLOCAL_INCLUDE(scripts)
 
 AC_CANONICAL_HOST                                                               
 
--- origsrc/libosip2-4.1.0/src/test/Makefile.am	2012-11-15 19:02:02.000000000 +0900
+++ src/libosip2-4.1.0/src/test/Makefile.am	2014-08-01 08:31:25.507729700 +0900
@@ -49,7 +49,7 @@ check:
 	@echo " *******************************"
 	@echo " ****** starting tests! ********"
 	@echo " *******************************"
-	@./$(top_srcdir)/src/test/tst ./$(top_srcdir)/src/test/res -c
+	@$(top_srcdir)/src/test/tst $(top_srcdir)/src/test/res -c
 
 	@echo ""
 	@echo "In case you have a doubt, send the generated"
