--- origsrc/libindicator-0.3.22/configure.ac	2011-03-24 03:13:23.000000000 +0900
+++ src/libindicator-0.3.22/configure.ac	2011-06-05 18:07:16.310829500 +0900
@@ -12,8 +12,8 @@ m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RU
 AC_ISC_POSIX
 AC_PROG_CC
 AM_PROG_CC_C_O
+LT_INIT
 AC_STDC_HEADERS
-AC_PROG_LIBTOOL
 
 AC_PATH_PROG([GLIB_MKENUMS], [glib-mkenums])
 
--- origsrc/libindicator-0.3.22/libindicator/Makefile.am	2011-02-17 06:37:42.000000000 +0900
+++ src/libindicator-0.3.22/libindicator/Makefile.am	2011-06-05 18:08:20.127549500 +0900
@@ -45,7 +45,7 @@ libindicator_la_SOURCES = \
 libindicator_la_CFLAGS = \
 	$(LIBINDICATOR_CFLAGS) \
 	-DG_LOG_DOMAIN=\"libindicator\" \
-	-Wall -Werror
+	-Wall
 
 libindicator_la_LIBADD = \
 	$(LIBINDICATOR_LIBS)
--- origsrc/libindicator-0.3.22/tests/Makefile.am	2011-02-17 05:46:22.000000000 +0900
+++ src/libindicator-0.3.22/tests/Makefile.am	2011-06-05 18:34:18.392058700 +0900
@@ -35,7 +35,7 @@ test_loader_SOURCES = \
 	test-loader.c
 
 test_loader_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir) \
 	-DBUILD_DIR="\"$(builddir)\""
 
@@ -54,7 +54,7 @@ test_desktop_shortcuts_SOURCES = \
 	test-desktop-shortcuts.c
 
 test_desktop_shortcuts_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	-DSRCDIR="\"$(srcdir)\"" \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir) \
 	-DBUILD_DIR="\"$(abs_builddir)\""
@@ -88,7 +88,7 @@ libdummy_indicator_blank_la_SOURCES = \
 	dummy-indicator-blank.c
 
 libdummy_indicator_blank_la_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 libdummy_indicator_blank_la_LIBADD = \
@@ -97,7 +97,7 @@ libdummy_indicator_blank_la_LIBADD = \
 	$(INDICATOR_LIB)
 
 libdummy_indicator_blank_la_LDFLAGS = \
-	-module \
+	-module -no-undefined \
 	-avoid-version
 
 #############################
@@ -108,7 +108,7 @@ libdummy_indicator_null_la_SOURCES = \
 	dummy-indicator-null.c
 
 libdummy_indicator_null_la_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 libdummy_indicator_null_la_LIBADD = \
@@ -117,7 +117,7 @@ libdummy_indicator_null_la_LIBADD = \
 	$(INDICATOR_LIB)
 
 libdummy_indicator_null_la_LDFLAGS = \
-	-module \
+	-module -no-undefined \
 	-avoid-version
 
 #############################
@@ -128,7 +128,7 @@ libdummy_indicator_signaler_la_SOURCES =
 	dummy-indicator-signaler.c
 
 libdummy_indicator_signaler_la_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 libdummy_indicator_signaler_la_LIBADD = \
@@ -137,7 +137,7 @@ libdummy_indicator_signaler_la_LIBADD =
 	$(INDICATOR_LIB)
 
 libdummy_indicator_signaler_la_LDFLAGS = \
-	-module \
+	-module -no-undefined \
 	-avoid-version
 
 #############################
@@ -148,7 +148,7 @@ libdummy_indicator_simple_la_SOURCES = \
 	dummy-indicator-simple.c
 
 libdummy_indicator_simple_la_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 libdummy_indicator_simple_la_LIBADD = \
@@ -157,7 +157,7 @@ libdummy_indicator_simple_la_LIBADD = \
 	$(INDICATOR_LIB)
 
 libdummy_indicator_simple_la_LDFLAGS = \
-	-module \
+	-module -no-undefined \
 	-avoid-version
 
 #############################
@@ -170,7 +170,7 @@ service_shutdown_timeout_SOURCES = \
 	service-shutdown-timeout.c
 
 service_shutdown_timeout_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_shutdown_timeout_LDADD = \
@@ -196,7 +196,7 @@ service_manager_no_connect_SOURCES = \
 	service-manager-no-connect.c
 
 service_manager_no_connect_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_manager_no_connect_LDADD = \
@@ -228,7 +228,7 @@ service_manager_connect_SOURCES = \
 	service-manager-connect.c
 
 service_manager_connect_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_manager_connect_LDADD = \
@@ -242,7 +242,7 @@ service_manager_connect_service_SOURCES
 	service-manager-connect-service.c
 
 service_manager_connect_service_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_manager_connect_service_LDADD = \
@@ -275,7 +275,7 @@ service_version_manager_SOURCES = \
 	service-version-manager.c
 
 service_version_manager_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_manager_LDADD = \
@@ -290,7 +290,7 @@ service_version_bad_service_SOURCES = \
 	service-version-bad-service.c
 
 service_version_bad_service_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_bad_service_LDADD = \
@@ -305,7 +305,7 @@ service_version_good_service_SOURCES = \
 	service-version-good-service.c
 
 service_version_good_service_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_good_service_LDADD = \
@@ -332,7 +332,7 @@ service_version_manager_SOURCES = \
 	service-version-multiwatch-manager.c
 
 service_version_multiwatch_manager_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_multiwatch_manager_LDADD = \
@@ -347,7 +347,7 @@ service_version_manager_impolite_SOURCES
 	service-version-multiwatch-manager-impolite.c
 
 service_version_multiwatch_manager_impolite_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_multiwatch_manager_impolite_LDADD = \
@@ -362,7 +362,7 @@ service_version_multiwatch_service_SOURC
 	service-version-multiwatch-service.c
 
 service_version_multiwatch_service_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_version_multiwatch_service_LDADD = \
@@ -388,7 +388,7 @@ service_manager_nostart_connect_SOURCES
 	service-manager-nostart-connect.c
 
 service_manager_nostart_connect_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir)
 
 service_manager_nostart_connect_LDADD = \
--- origsrc/libindicator-0.3.22/tools/Makefile.am	2011-01-07 07:02:08.000000000 +0900
+++ src/libindicator-0.3.22/tools/Makefile.am	2011-06-05 18:09:45.068528700 +0900
@@ -17,7 +17,7 @@ indicator_loader_SOURCES = \
 	indicator-loader.c
 
 indicator_loader_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	$(LIBINDICATOR_CFLAGS) -I$(top_srcdir) \
 	-DBUILD_DIR="\"$(builddir)\""
 
