--- origsrc/libstree-0.4.3/Makefile.am	2003-09-10 22:11:10.000000000 +0900
+++ src/libstree-0.4.3/Makefile.am	2014-07-20 14:08:45.291359700 +0900
@@ -1,6 +1,10 @@
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS = src test docs
+SUBDIRS = src test
+
+if ENABLE_GTK_DOC
+SUBDIRS += docs
+endif
 
 EXTRA_DIST = \
 	autogen.sh \
--- origsrc/libstree-0.4.3/src/Makefile.am	2003-09-10 22:29:30.000000000 +0900
+++ src/libstree-0.4.3/src/Makefile.am	2014-07-20 14:08:45.291359700 +0900
@@ -25,4 +25,4 @@ libstree_la_SOURCES = \
 	lst_structs.h lst_macros.h \
 	lst_timestamp.h
 
-libstree_la_LDFLAGS = -module
+libstree_la_LDFLAGS = -no-undefined
--- origsrc/libstree-0.4.3/test/Makefile.am	2003-09-10 22:36:18.000000000 +0900
+++ src/libstree-0.4.3/test/Makefile.am	2014-07-20 14:08:45.306984700 +0900
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
 INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src -W -Wall
-LIBS = $(top_builddir)/src/.libs/libstree.a
+LIBS = $(top_builddir)/src/libstree.la
 
 bin_PROGRAMS = lcstext lcshex lrstext deltest buildtest addtest addremovetest
 
