--- origsrc/fson/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/fson/Makefile.am	2013-10-14 15:35:59.760955700 +0900
@@ -0,0 +1,19 @@
+
+
+check_PROGRAMS = example1
+
+example1_SOURCES = src/examples/example1.f95
+example1_LDADD = libfson.la
+
+
+include_HEADERS = fson.mod fson_path_m.mod fson_string_m.mod fson_value_m.mod
+
+
+lib_LTLIBRARIES = libfson.la
+
+libfson_la_SOURCES = \
+	src/fson_string_m.f95 src/fson_value_m.f95 src/fson_path_m.f95 \
+	src/fson.f95
+libfson_la_LDFLAGS = -no-undefined
+
+
--- origsrc/fson/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/fson/configure.ac	2013-10-14 15:34:55.593963500 +0900
@@ -0,0 +1,21 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([fson], [0+git20130109])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_FC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+
+# Checks for typedefs, structures, and compiler characteristics.
+
+# Checks for library functions.
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
