--- origsrc/fson-1.0.5/meson.build	2020-11-20 13:21:04.000000000 +0900
+++ src/fson-1.0.5/meson.build	2025-04-08 06:43:24.284680200 +0900
@@ -19,7 +19,7 @@ else
   f_args = []
 endif
 
-fson = shared_library('fson', fson_sources, fortran_args: f_args, install: true)
+fson = shared_library('fson', fson_sources, soversion: 1, fortran_args: f_args, install: true)
 
 fson_dep = declare_dependency(link_with : fson)
 
@@ -67,6 +67,7 @@ foreach test_name: test_names
   test_src = join_paths(meson.current_source_dir(),
                         'src', 'tests', test_name + '_zofu.f90')
   test_lib = static_library(test_name + '_lib', test_src,
+                            include_directories: '/usr/include',
                             link_with: fson,
                             dependencies: zofu)
   test_driver_src = custom_target(
@@ -76,6 +77,7 @@ foreach test_name: test_names
     command: [zofu_driver, '@INPUT@', '@OUTPUT@'])
   unit_test = executable(
     test_name, [test_driver_src, test_src],
+    include_directories: '/usr/include',
     link_with: [fson, test_lib],
     dependencies: zofu)
   test(test_name, unit_test, env: test_env)
