--- origsrc/aldo-0.7.7/src/Makefile.am	2012-03-31 20:32:34.000000000 +0900
+++ src/aldo-0.7.7/src/Makefile.am	2015-05-15 16:38:30.212430100 +0900
@@ -12,5 +12,5 @@ bin_PROGRAMS = aldo
 aldo_SOURCES = astream.cc  blocks.cc datafile.cc koch_exc.cc menu.cc qrz.cc random.cc setup.cc textfile_exc.cc\
 		check.cc blocks_exc.cc keyer.cc main.cc option.cc qrz_exc.cc section.cc textfile.cc wave.cc\
 		audioworkspace.cc koch.cc
-aldo_CXXFLAGS = -ansi -Wall -pedantic
+aldo_CXXFLAGS = -Wall -pedantic
 INCLUDES = -I$(top_srcdir)/include
