--- origsrc/CheMPS2-1.8.3/CMakeLists.txt	2016-11-16 03:51:19.000000000 +0900
+++ src/CheMPS2-1.8.3/CMakeLists.txt	2017-01-01 20:51:17.567780000 +0900
@@ -154,7 +154,7 @@ endif()
 # <<<  Export config  >>>
 
     # explicit "share" not "DATADIR" for CMake search path
-set (CMAKECONFIG_INSTALL_DIR "share/cmake/${PROJECT_NAME}")
+set (CMAKECONFIG_INSTALL_DIR "lib/cmake/${PROJECT_NAME}")
 configure_package_config_file (${CheMPS2_SOURCE_DIR}/CMake/${PROJECT_NAME}Config.cmake.in
                                     "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
                                     INSTALL_DESTINATION ${CMAKECONFIG_INSTALL_DIR})
@@ -165,3 +165,9 @@ install (FILES ${CMAKE_CURRENT_BINARY_DI
                ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake
                DESTINATION ${CMAKECONFIG_INSTALL_DIR})
 
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/${PROJECT_NAME}.pc.in ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/CheMPS2-1.8.3/CheMPS2/CMakeLists.txt	2016-11-16 03:51:19.000000000 +0900
+++ src/CheMPS2-1.8.3/CheMPS2/CMakeLists.txt	2017-01-01 20:51:55.735704400 +0900
@@ -99,7 +99,7 @@ target_include_directories (chemps2-bin
 if (NOT STATIC_ONLY)
     install (TARGETS chemps2-shared chemps2-bin
                      EXPORT "${PROJECT_NAME}Targets-shared"
-                     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+                     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
                      RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 endif()
 
@@ -141,7 +141,7 @@ endif()
 # <<<  Export Config  >>>
 
     # explicit "share" not "DATADIR" for CMake search path
-set (CMAKECONFIG_INSTALL_DIR "share/cmake/${PROJECT_NAME}")
+set (CMAKECONFIG_INSTALL_DIR "lib/cmake/${PROJECT_NAME}")
 if (NOT STATIC_ONLY)
     install (EXPORT "${PROJECT_NAME}Targets-shared"
                      NAMESPACE "${PROJECT_NAME}::"
@@ -153,4 +153,3 @@ if (NOT SHARED_ONLY)
                      NAMESPACE "${PROJECT_NAME}::"
                      DESTINATION ${CMAKECONFIG_INSTALL_DIR})
 endif()
-
--- origsrc/CheMPS2-1.8.3/CheMPS2.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/CheMPS2-1.8.3/CheMPS2.pc.in	2017-01-01 20:54:21.907932400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: CheMPS2
+Description: Spin-adapted implementation of DMRG for ab initio quantum chemistry
+Version: @CheMPS2_MMP_VERSION@
+Libs: -L${libdir} -lCheMPS2
+Cflags: -I${includedir}/chemps2
