diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*.whl' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x Doxyfile.in -x version.h origsrc/librepo-1.11.2/CMakeLists.txt src/librepo-1.11.2/CMakeLists.txt
--- origsrc/librepo-1.11.2/CMakeLists.txt	2020-01-19 23:50:24.000000000 +0900
+++ src/librepo-1.11.2/CMakeLists.txt	2020-01-23 11:17:19.766576000 +0900
@@ -8,8 +8,10 @@ OPTION (WITH_ZCHUNK "Build with zchunk s
 INCLUDE (${CMAKE_SOURCE_DIR}/VERSION.cmake)
 SET (VERSION "${LIBREPO_MAJOR}.${LIBREPO_MINOR}.${LIBREPO_PATCH}")
 
-SET (CMAKE_C_FLAGS          "${CMAKE_C_FLAGS} -std=c99 -Wall")
-SET (CMAKE_C_FLAGS_DEBUG    "${CMAKE_C_FLAGS} -ggdb -O0 -Wall -Wextra -DDEBUG")
+if (NOT CYGWIN)
+    SET (CMAKE_C_FLAGS          "${CMAKE_C_FLAGS} -std=c99 -Wall")
+    SET (CMAKE_C_FLAGS_DEBUG    "${CMAKE_C_FLAGS} -ggdb -O0 -Wall -Wextra -DDEBUG")
+endif()
 
 IF(CMAKE_BUILD_TYPE AND (CMAKE_BUILD_TYPE STREQUAL "DEBUG" OR CMAKE_BUILD_TYPE STREQUAL "debug"))
     MESSAGE("Build type is set do DEBUG!")
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*.whl' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x Doxyfile.in -x version.h origsrc/librepo-1.11.2/librepo/CMakeLists.txt src/librepo-1.11.2/librepo/CMakeLists.txt
--- origsrc/librepo-1.11.2/librepo/CMakeLists.txt	2020-01-19 23:50:24.000000000 +0900
+++ src/librepo-1.11.2/librepo/CMakeLists.txt	2020-01-23 11:17:19.783064900 +0900
@@ -64,12 +64,12 @@ SET_TARGET_PROPERTIES(librepo PROPERTIES
 CONFIGURE_FILE("librepo.pc.cmake" "${CMAKE_BINARY_DIR}/librepo.pc" @ONLY)
 CONFIGURE_FILE("version.h.in" "${CMAKE_CURRENT_SOURCE_DIR}/version.h" @ONLY)
 
-IF (CMAKE_SIZEOF_VOID_P MATCHES "8")
+IF (CMAKE_SIZEOF_VOID_P MATCHES "8" AND NOT CYGWIN)
   SET (LIB_SUFFIX "64")
-ENDIF (CMAKE_SIZEOF_VOID_P MATCHES "8")
+ENDIF (CMAKE_SIZEOF_VOID_P MATCHES "8" AND NOT CYGWIN)
 
 INSTALL(FILES ${librepo_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/librepo)
-INSTALL(TARGETS librepo LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+INSTALL(TARGETS librepo RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 INSTALL(FILES "${CMAKE_BINARY_DIR}/librepo.pc"
 	DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*.whl' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x Doxyfile.in -x version.h origsrc/librepo-1.11.2/librepo/python/python3/CMakeLists.txt src/librepo-1.11.2/librepo/python/python3/CMakeLists.txt
--- origsrc/librepo-1.11.2/librepo/python/python3/CMakeLists.txt	2020-01-19 23:50:24.000000000 +0900
+++ src/librepo-1.11.2/librepo/python/python3/CMakeLists.txt	2020-01-23 11:17:19.783064900 +0900
@@ -26,4 +26,4 @@ TARGET_LINK_LIBRARIES(_librepo
                      )
 
 INSTALL(FILES ${pylibrepo_SRCDIR}/__init__.py DESTINATION ${PYTHON_INSTALL_DIR}/librepo)
-INSTALL(TARGETS _librepo LIBRARY DESTINATION ${PYTHON_INSTALL_DIR}/librepo)
+INSTALL(TARGETS _librepo RUNTIME DESTINATION ${PYTHON_INSTALL_DIR}/librepo)
