--- origsrc/librepo/CMakeLists.txt	2014-07-06 11:02:50.000000000 +0900
+++ src/librepo/CMakeLists.txt	2014-07-06 11:17:07.865963500 +0900
@@ -29,6 +29,7 @@ FIND_PACKAGE(CURL REQUIRED)
 FIND_LIBRARY(CHECK_LIBRARY NAMES check)
 FIND_PACKAGE(Gpgme REQUIRED)
 FIND_PACKAGE(Xattr REQUIRED)
+FIND_PACKAGE(OpenSSL REQUIRED)
 
 INCLUDE_DIRECTORIES(${GLIB2_INCLUDE_DIRS})
 
--- origsrc/librepo/librepo/CMakeLists.txt	2014-07-06 11:02:50.000000000 +0900
+++ src/librepo/librepo/CMakeLists.txt	2014-07-06 11:18:50.115963500 +0900
@@ -46,6 +46,7 @@ TARGET_LINK_LIBRARIES(librepo
                         ${CURL_LIBRARY}
                         ${GPGME_VANILLA_LIBRARIES}
                         ${GLIB2_LIBRARIES}
+                        ${OPENSSL_LIBRARIES}
                      )
 SET_TARGET_PROPERTIES(librepo PROPERTIES OUTPUT_NAME "repo")
 SET_TARGET_PROPERTIES(librepo PROPERTIES SOVERSION 0)
@@ -54,13 +55,13 @@ SET_TARGET_PROPERTIES(librepo PROPERTIES
 CONFIGURE_FILE("librepo.pc.cmake" "${CMAKE_BINARY_DIR}/librepo.pc" @ONLY)
 CONFIGURE_FILE("version.h.in" "${CMAKE_CURRENT_SOURCE_DIR}/version.h" @ONLY)
 
-IF (CMAKE_SIZEOF_VOID_P MATCHES "8")
+IF (CMAKE_SIZEOF_VOID_P MATCHES "8" AND NOT CYGWIN)
   SET (LIB_SUFFIX "64")
-ENDIF (CMAKE_SIZEOF_VOID_P MATCHES "8")
+ENDIF (CMAKE_SIZEOF_VOID_P MATCHES "8" AND NOT CYGWIN)
 SET (LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
 
 INSTALL(FILES ${librepo_HEADERS} DESTINATION include/librepo)
-INSTALL(TARGETS librepo LIBRARY DESTINATION ${LIB_INSTALL_DIR})
+INSTALL(TARGETS librepo RUNTIME DESTINATION bin ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
 INSTALL(FILES "${CMAKE_BINARY_DIR}/librepo.pc"
        DESTINATION "${LIB_INSTALL_DIR}/pkgconfig")
 
--- origsrc/librepo/librepo/python/python2/CMakeLists.txt	2014-07-06 11:02:51.000000000 +0900
+++ src/librepo/librepo/python/python2/CMakeLists.txt	2014-07-06 11:23:37.380611900 +0900
@@ -23,4 +23,4 @@ TARGET_LINK_LIBRARIES(_librepomodule
                      )
 
 INSTALL(FILES ${pylibrepo_SRCDIR}/__init__.py DESTINATION ${PYTHON_INSTALL_DIR}/librepo)
-INSTALL(TARGETS _librepomodule LIBRARY DESTINATION ${PYTHON_INSTALL_DIR}/librepo)
+INSTALL(TARGETS _librepomodule RUNTIME DESTINATION ${PYTHON_INSTALL_DIR}/librepo)
--- origsrc/librepo/librepo/yum.c	2014-07-06 11:02:51.000000000 +0900
+++ src/librepo/librepo/yum.c	2014-07-06 11:15:38.827877500 +0900
@@ -18,7 +18,9 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#if !defined(__CYGWIN__)
 #define _POSIX_SOURCE
+#endif
 #define _BSD_SOURCE
 
 #include <stdio.h>
