--- origsrc/react-2.3.1/CMakeLists.txt	2014-04-30 03:43:12.000000000 +0900
+++ src/react-2.3.1/CMakeLists.txt	2025-01-20 15:21:55.598637300 +0900
@@ -57,7 +57,6 @@ add_library(react SHARED
 set_target_properties(react PROPERTIES
 	VERSION ${DEBFULLVERSION}
 	SOVERSION ${REACT_VERSION_ABI}
-	COMPILE_FLAGS "-std=c++0x -W -Wall -Werror -pedantic"
 	LINKER_LANGUAGE CXX
 )
 
@@ -67,6 +66,7 @@ endif()
 
 install(TARGETS react
 	EXPORT ReactTargets
+	RUNTIME DESTINATION bin
 	LIBRARY DESTINATION lib${LIB_SUFFIX}
 	ARCHIVE DESTINATION lib${LIB_SUFFIX}
 	BUNDLE DESTINATION library
@@ -82,3 +82,10 @@ install(DIRECTORY foreign/
 
 set(INSTALL_INCLUDE_DIR "${CMAKE_INSTALL_PREFIX}/include")
 set(INSTALL_LIBRARY_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/react.pc.in ${CMAKE_BINARY_DIR}/react.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/react.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/react-2.3.1/react.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/react-2.3.1/react.pc.in	2025-01-20 15:22:16.056399300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: react
+Description: Realtime call tree for C++
+Version: @DEBFULLVERSION@
+Libs: -L${libdir} -lreact
+Cflags: -I${includedir}
--- origsrc/react-2.3.1/tests/test_actions_set.cpp	2014-04-30 03:43:12.000000000 +0900
+++ src/react-2.3.1/tests/test_actions_set.cpp	2025-01-20 15:18:21.362407800 +0900
@@ -74,7 +74,7 @@ BOOST_AUTO_TEST_CASE( get_action_name_no
 	actions_set_t actions_set;
 
 	BOOST_CHECK_THROW( actions_set.get_action_name(actions_set_t::NO_ACTION),
-					   std::invalid_argument )
+					   std::invalid_argument );
 }
 
 BOOST_AUTO_TEST_SUITE_END()
--- origsrc/react-2.3.1/tests/test_call_tree_updater.cpp	2014-04-30 03:43:12.000000000 +0900
+++ src/react-2.3.1/tests/test_call_tree_updater.cpp	2025-01-20 15:17:30.814230300 +0900
@@ -240,7 +240,7 @@ BOOST_AUTO_TEST_CASE( action_guard_stop_
 	{
 		action_guard_t action_guard(NULL, NO_ACTION);
 		action_guard.stop();
-		BOOST_CHECK_THROW( action_guard.stop(), std::logic_error )
+		BOOST_CHECK_THROW( action_guard.stop(), std::logic_error );
 	}
 
 	{
@@ -251,7 +251,7 @@ BOOST_AUTO_TEST_CASE( action_guard_stop_
 
 		action_guard_t action_guard(&updater, action_code);
 		action_guard.stop();
-		BOOST_CHECK_THROW( action_guard.stop(), std::logic_error )
+		BOOST_CHECK_THROW( action_guard.stop(), std::logic_error );
 	}
 }
 
