--- origsrc/freeradius-client-release_1_1_7/lib/Makefile.am	2015-01-20 01:18:26.000000000 +0900
+++ src/freeradius-client-release_1_1_7/lib/Makefile.am	2024-11-08 16:38:42.635000100 +0900
@@ -27,6 +27,6 @@ if !ENABLE_NETTLE
 libfreeradius_client_la_SOURCES += md5.c md5.h
 endif
 
-libfreeradius_client_la_LDFLAGS = -version-info $(LIBVERSION)
+libfreeradius_client_la_LDFLAGS = -no-undefined -version-info $(LIBVERSION)
 
 libfreeradius_client_la_LIBADD = $(CRYPTO_LIBS)
