--- origsrc/gtkmaskedentry-0.0.2/configure.ac	2006-01-12 01:09:27.000000000 +0900
+++ src/gtkmaskedentry-0.0.2/configure.ac	2010-12-03 18:16:47.547884300 +0900
@@ -7,12 +7,8 @@ AM_INIT_AUTOMAKE
 
 AM_MAINTAINER_MODE
 
-AC_LIBTOOL_WIN32_DLL
-
 AC_PROG_CC
-AC_PROG_LIBTOOL
-LT_INIT
-IT_PROG_INTLTOOL
+LT_INIT(win32-dll)
 
 AC_CANONICAL_SYSTEM
 
--- origsrc/gtkmaskedentry-0.0.2/src/gtkcellrenderermasked.c	2006-01-12 01:44:33.000000000 +0900
+++ src/gtkmaskedentry-0.0.2/src/gtkcellrenderermasked.c	2010-12-03 18:14:49.357935500 +0900
@@ -24,6 +24,10 @@
 
 #include "gtkcellrenderermasked.h"
 
+#if defined(__CYGWIN__) && !defined(GTK_PARAM_READWRITE)
+#define GTK_PARAM_READWRITE G_PARAM_READWRITE|G_PARAM_STATIC_NAME|G_PARAM_STATIC_NICK|G_PARAM_STATIC_BLURB
+#endif /* defined(__CYGWIN__) && !defined(GTK_PARAM_READWRITE) */
+
 static void gtk_cell_renderer_masked_init (GtkCellRendererMasked *cell);
 static void gtk_cell_renderer_masked_class_init (GtkCellRendererMaskedClass *class);
 
--- origsrc/gtkmaskedentry-0.0.2/tests/Makefile.am	2005-08-01 01:10:06.000000000 +0900
+++ src/gtkmaskedentry-0.0.2/tests/Makefile.am	2010-12-03 18:14:49.357935500 +0900
@@ -2,7 +2,8 @@ INCLUDES = \
 	$(WARN_CFLAGS) \
 	$(DISABLE_DEPRECATED_CFLAGS) \
 	$(GTK_CFLAGS) \
-	-I../src
+	-I../src \
+	-I$(top_srcdir)/src
 
 LDADD = ../src/libgtkmaskedentry.la
 
