--- origsrc/zile-2.3.24/configure.ac	2011-04-30 01:50:45.000000000 +0900
+++ src/zile-2.3.24/configure.ac	2011-08-13 22:28:10.437318800 +0900
@@ -90,11 +90,7 @@ gl_INIT
 AC_CHECK_FUNCS_ONCE(fchmod)
 
 dnl Curses
-AX_WITH_CURSES
-if test "$ax_cv_curses" != "yes"; then
-  AC_MSG_ERROR([cannot find curses])
-fi
-AC_ARG_VAR(CURSES_LIB, [linker flags for curses library])
+PKG_CHECK_MODULES(CURSES, ncurses)
 
 dnl Emacs
 AX_WITH_PROG(EMACS, emacs)
--- origsrc/zile-2.3.24/src/Makefile.am	2011-04-29 23:38:31.000000000 +0900
+++ src/zile-2.3.24/src/Makefile.am	2011-08-13 22:29:04.174589200 +0900
@@ -20,8 +20,8 @@
 # MA 02111-1301, USA.
 
 AM_CFLAGS = $(WARN_CFLAGS)
-AM_CPPFLAGS = -I$(top_builddir)/lib -I$(top_srcdir)/lib -DPATH_DATA="\"$(pkgdatadir)\""
-LDADD = $(top_builddir)/lib/libgnu.a $(LIB_EACCESS) $(LIBINTL) $(CURSES_LIB)
+AM_CPPFLAGS = -I$(top_builddir)/lib -I$(top_srcdir)/lib -DPATH_DATA="\"$(pkgdatadir)\"" $(CURSES_CFLAGS)
+LDADD = $(top_builddir)/lib/libgnu.a $(LIB_EACCESS) $(LIBINTL) $(CURSES_LIBS)
 
 bin_PROGRAMS = zile
 
