--- origsrc/libosinfo/Makefile.am	2011-01-10 15:48:29.000000000 +0900
+++ src/libosinfo/Makefile.am	2011-01-10 15:53:23.186215300 +0900
@@ -9,8 +9,6 @@ EXTRA_DIST = \
 
 DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-doc
 
-ACLOCAL_AMFLAGS = -I m4
-
 LCOV = lcov
 GENHTML = genhtml
 
@@ -26,4 +24,4 @@ cov: clean-cov
 
 
 clean-cov:
-	rm -rf $(top_builddir)/coverage
\ No newline at end of file
+	rm -rf $(top_builddir)/coverage
--- origsrc/libosinfo/configure.ac	2011-01-10 15:48:29.000000000 +0900
+++ src/libosinfo/configure.ac	2011-01-10 15:51:33.273168100 +0900
@@ -5,12 +5,10 @@ AM_INIT_AUTOMAKE([1.11.1 foreign color-t
 AC_PREREQ([2.61])
 AC_CONFIG_SRCDIR([osinfo/osinfo_db.c])
 AC_CONFIG_HEADERS([config.h])
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_PROG_CC
 
 GNOME_COMPILE_WARNINGS([maximum])
-GNOME_MAINTAINER_MODE_DEFINES
-AM_MAINTAINER_MODE([enable])
 
 # Use the silent-rules feature when possible.
 m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
--- origsrc/libosinfo/osinfo/Makefile.am	2011-01-10 15:48:29.000000000 +0900
+++ src/libosinfo/osinfo/Makefile.am	2011-01-10 16:20:35.872673900 +0900
@@ -36,7 +36,7 @@ libosinfo_1_0_la_LDFLAGS = \
 	$(GIO_LIBS) \
 	$(VERSION_SCRIPT_FLAGS)$(LIBOSINFO_VERSION_FILE) \
         -version-info $(LIBOSINFO_VERSION_INFO) \
-	$(NO_UNDEFINED_FLAGS)
+	-no-undefined
 
 libosinfo_1_0_includedir = $(includedir)/libosinfo-1.0/osinfo
 
