--- origsrc/libosinfo/Makefile.am	2011-08-18 10:48:42.000000000 +0900
+++ src/libosinfo/Makefile.am	2011-08-18 10:50:26.375928500 +0900
@@ -8,8 +8,6 @@ EXTRA_DIST = \
 
 DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-doc
 
-ACLOCAL_AMFLAGS = -I m4
-
 LCOV = lcov
 GENHTML = genhtml
 
@@ -25,4 +23,4 @@ cov: clean-cov
 
 
 clean-cov:
-	rm -rf $(top_builddir)/coverage
\ No newline at end of file
+	rm -rf $(top_builddir)/coverage
--- origsrc/libosinfo/configure.ac	2011-08-18 10:48:42.000000000 +0900
+++ src/libosinfo/configure.ac	2011-08-18 11:13:21.926380200 +0900
@@ -54,7 +54,7 @@ case "$host" in
     ;;
 
   *-*-cygwin*)
-    NO_UNDEFINED_LDFLAGS="-no-undefined"
+    NO_UNDEFINED_FLAGS="-no-undefined"
     ;;
 
   *)
