--- origsrc/szip-2.1.1/Makefile.am	2017-01-10 00:14:35.000000000 +0900
+++ src/szip-2.1.1/Makefile.am	2017-09-20 21:14:37.296607400 +0900
@@ -3,3 +3,6 @@
 #############################################################################
 
 SUBDIRS=src test
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/szip-2.1.1/configure.ac	2017-01-14 05:01:14.000000000 +0900
+++ src/szip-2.1.1/configure.ac	2017-09-20 21:14:37.296607400 +0900
@@ -195,7 +195,7 @@ if test "X$ENCODING" = "Xyes"; then
    else
      echo "no"
 fi
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([szip.pc Makefile
                  src/Makefile
                  test/Makefile])
 AC_OUTPUT
--- origsrc/szip-2.1.1/src/Makefile.am	2017-01-10 00:14:35.000000000 +0900
+++ src/szip-2.1.1/src/Makefile.am	2017-09-20 21:12:01.718466300 +0900
@@ -9,7 +9,7 @@ CSOURCES = rice.c sz_api.c encoding.c
 CHEADERS =  ricehdf.h szip_adpt.h szlib.h
 
 libsz_la_SOURCES = $(CSOURCES)
-libsz_la_LDFLAGS = -version-info 2:0:0
+libsz_la_LDFLAGS = -no-undefined -version-info 2:0:0
 include_HEADERS = $(CHEADERS)
 
 
--- origsrc/szip-2.1.1/szip.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/szip-2.1.1/szip.pc.in	2017-09-20 21:15:03.187000000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: szip
+Description: Implementation of the extended-Rice lossless compression algorithm
+Version: @VERSION@
+Libs: -L${libdir} -lsz
+Cflags: -I${includedir}
