--- origsrc/szip-2.1/src/Makefile.am	2005-01-06 05:21:56.000000000 +0900
+++ src/szip-2.1/src/Makefile.am	2012-05-07 09:46:29.838087900 +0900
@@ -9,7 +9,7 @@ CSOURCES = rice.c sz_api.c encoding.c
 CHEADERS =  ricehdf.h szip_adpt.h szlib.h
 
 libsz_la_SOURCES = $(CSOURCES)
-libsz_la_LDFLAGS = -version-info 2:0:0
+libsz_la_LDFLAGS = -version-info 2:0:0 -no-undefined
 include_HEADERS = $(CHEADERS)
 
 
