--- origsrc/calceph-3.5.2/pythonapi/src/Makefile.am	2023-03-08 22:43:06.000000000 +0900
+++ src/calceph-3.5.2/pythonapi/src/Makefile.am	2023-07-25 19:54:57.762715100 +0900
@@ -63,8 +63,8 @@
 
 pyexec_LTLIBRARIES = calcephpy.la
 
-calcephpy_la_SOURCES = calcephpy.c 
-calcephpy_la_LDFLAGS = -avoid-version -module $(PYTHON_LIBS)  -shared
+calcephpy_la_SOURCES = $(srcdir)/calcephpy.c 
+calcephpy_la_LDFLAGS = -avoid-version -module $(PYTHON_LIBS)  -shared $(CALCEPHWIN32_LDFLAGS)
 calcephpy_la_LIBADD = $(top_builddir)/src/libcalcephnoinst.la
 calcephpy_la_CPPFLAGS = $(PYTHON_CPPFLAGS) -I$(top_srcdir)/src
 
@@ -122,7 +122,7 @@ calcephpy.c: $(MAYBE_PYTHONAPI_CALCEPHPY
 
 
 calcephpy_c_target :
-	test "$(top_srcdir)" == "$(top_builddir)" || cp -p $(top_srcdir)/pythonapi/src/calcephpy.p*  $(top_builddir)/pythonapi/src/
+	test "$(top_srcdir)" == "$(top_builddir)" || cp -p $(top_srcdir)/pythonapi/src/calcephpy.pxd $(top_builddir)/pythonapi/src/
 	CC="$(CC) -I$(top_srcdir)/src $(CFLAGS)" $(PYTHON)  setup.py build_ext -i
 	sed -i -e 's://.*::g'  $(top_builddir)/pythonapi/src/calcephpy.c
 
