--- origsrc/calceph-4.0.0/src/CMakeLists.txt	2024-04-17 23:59:59.000000000 +0900
+++ src/calceph-4.0.0/src/CMakeLists.txt	2024-06-01 15:48:42.187338100 +0900
@@ -104,7 +104,7 @@ if (CALCEPH_INSTALL)
         install(FILES "${CMAKE_CURRENT_BINARY_DIR}/calceph.mod" COMPONENT dev DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
     endif()
     install(TARGETS  calceph EXPORT calcephTargets ARCHIVE COMPONENT dev  DESTINATION "${CMAKE_INSTALL_LIBDIR}" LIBRARY COMPONENT lib  DESTINATION "${CMAKE_INSTALL_LIBDIR}")
-    if (WIN32)
+    if (WIN32 OR CYGWIN)
 	    if (BUILD_SHARED_LIBS)
 	        install(TARGETS calceph RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
 	    endif()
