--- origsrc/libcintelhex/Makefile.am	2020-03-04 17:58:59.000000000 +0900
+++ src/libcintelhex/Makefile.am	2020-03-04 18:50:59.647628900 +0900
@@ -21,3 +21,6 @@
 ACLOCAL_AMFLAGS = -I m4
 
 SUBDIRS = src include tests
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libcintelhex/configure.ac	2020-03-04 17:58:59.000000000 +0900
+++ src/libcintelhex/configure.ac	2020-03-04 18:50:59.663343700 +0900
@@ -59,7 +59,7 @@ AC_FUNC_MMAP
 AC_CHECK_FUNCS([munmap memset])
 
 AC_CONFIG_HEADERS([config.h])
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([libcintelhex.pc Makefile
                  include/Makefile
                  src/Makefile
                  tests/Makefile])
--- origsrc/libcintelhex/libcintelhex.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libcintelhex/libcintelhex.pc.in	2020-03-04 18:50:59.444495400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libcintelhex
+Description: C library for parsing Intel HEX files
+Version: @VERSION@
+Libs: -L${libdir} -lcintelhex
+Cflags: -I${includedir}
--- origsrc/libcintelhex/src/Makefile.am	2020-03-04 17:58:59.000000000 +0900
+++ src/libcintelhex/src/Makefile.am	2020-03-04 18:36:23.721576900 +0900
@@ -24,4 +24,4 @@ lib_LTLIBRARIES = libcintelhex.la
 
 libcintelhex_la_SOURCES = ihex_copy.c ihex_parse.c ihex_record.c
 libcintelhex_la_CPPFLAGS = -I$(top_srcdir)/include
-libcintelhex_la_LDFLAGS = -version-info 1:0:1
+libcintelhex_la_LDFLAGS = -no-undefined -version-info 1:0:1
