--- origsrc/libcmis-0.3.1/src/libcmis-c/Makefile.am	2013-01-30 05:01:33.000000000 +0900
+++ src/libcmis-0.3.1/src/libcmis-c/Makefile.am	2013-02-24 22:45:03.841816700 +0900
@@ -35,7 +35,7 @@ libcmis_c_@LIBCMIS_API_VERSION@_la_SOURC
 
 libcmis_c_@LIBCMIS_API_VERSION@_la_LDFLAGS = -export-dynamic -no-undefined -version-info 3:1:0
 
-libcmis_c_@LIBCMIS_API_VERSION@_la_LIBADD = \
+libcmis_c_@LIBCMIS_API_VERSION@_la_LIBADD = ../libcmis/libcmis-@LIBCMIS_API_VERSION@.la \
 	$(XML2_LIBS) \
 	$(CURL_LIBS) \
 	$(BOOST_SMART_PTR_LIBS) \
