--- origsrc/cmockery2-1.3.9/Makefile.am	2015-04-14 04:24:28.000000000 +0900
+++ src/cmockery2-1.3.9/Makefile.am	2023-12-19 08:41:14.958168500 +0900
@@ -18,7 +18,7 @@ cmockeryinclude_HEADERS = src/cmockery/c
 						  src/cmockery/cmockery.h\
 						  src/cmockery/pbc.h
 
-docdir = $(prefix)/share/doc/$(PACKAGE)-$(VERSION)
+# docdir = $(prefix)/share/doc/$(PACKAGE)-$(VERSION)
 ## This is for HTML and other documentation you want to install.
 ## Add your documentation files (in doc/) in addition to these
 ## top-level boilerplate files.  Also add a TODO file if you have one.
@@ -52,6 +52,7 @@ libcmockery_la_SOURCES = src/cmockery/co
 						 src/cmockery/pbc.h
 libcmockery_la_CFLAGS = -I$(top_srcdir)/src/cmockery -DHAVE_CONFIG_H -Wall \
                         -Werror
+libcmockery_la_LDFLAGS = -no-undefined
 
 noinst_PROGRAMS = calculator
 nodist_calculator_SOURCES = src/example/calculator.c src/cmockery/config.h
