--- origsrc/vectorscan-vectorscan-5.4.8/tools/hsbench/CMakeLists.txt	2022-09-13 21:57:08.000000000 +0900
+++ src/vectorscan-vectorscan-5.4.8/tools/hsbench/CMakeLists.txt	2022-10-02 19:41:10.805228000 +0900
@@ -58,10 +58,10 @@ if (BUILD_CHIMERA)
     )
     add_executable(hsbench ${hsbench_SOURCES})
     include_directories(${PCRE_INCLUDE_DIRS})
-    target_link_libraries(hsbench hs chimera ${PCRE_LDFLAGS} databaseutil
-        expressionutil ${SQLITE3_LDFLAGS} ${CMAKE_THREAD_LIBS_INIT})
+    target_link_libraries(hsbench chimera ${PCRE_LDFLAGS} databaseutil
+        expressionutil ${SQLITE3_LDFLAGS} ${CMAKE_THREAD_LIBS_INIT} hs)
 else()
     add_executable(hsbench ${hsbench_SOURCES})
-    target_link_libraries(hsbench hs databaseutil expressionutil
-        ${SQLITE3_LDFLAGS} ${CMAKE_THREAD_LIBS_INIT})
+    target_link_libraries(hsbench databaseutil expressionutil
+        ${SQLITE3_LDFLAGS} ${CMAKE_THREAD_LIBS_INIT} hs)
 endif()
--- origsrc/vectorscan-vectorscan-5.4.8/tools/hsbench/main.cpp	2022-09-13 21:57:08.000000000 +0900
+++ src/vectorscan-vectorscan-5.4.8/tools/hsbench/main.cpp	2022-10-02 19:13:07.942545300 +0900
@@ -149,7 +149,10 @@ public:
 
         // The 'clang' compiler complains about an unused result here, so we
         // silence it.
-        (void)CPU_SET(cpu, &cpuset);
+#if !defined(__CYGWIN__)
+        (void)
+#endif
+        CPU_SET(cpu, &cpuset);
 
         int rv = pthread_setaffinity_np(thr.native_handle(), sizeof(cpuset),
                                         &cpuset);
--- origsrc/vectorscan-vectorscan-5.4.8/unit/CMakeLists.txt	2022-09-13 21:57:08.000000000 +0900
+++ src/vectorscan-vectorscan-5.4.8/unit/CMakeLists.txt	2022-10-02 19:00:20.857754900 +0900
@@ -132,7 +132,7 @@ endif(NOT RELEASE_BUILD)
 
 add_executable(unit-internal ${unit_internal_SOURCES})
 set_target_properties(unit-internal PROPERTIES COMPILE_FLAGS "${HS_CXX_FLAGS}")
-target_link_libraries(unit-internal hs corpusomatic)
+target_link_libraries(unit-internal corpusomatic hs)
 endif(NOT FAT_RUNTIME)
 
 if (BUILD_CHIMERA)
