--- origsrc/ezxml/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/ezxml/Makefile.am	2024-10-05 14:54:04.851670200 +0900
@@ -0,0 +1,10 @@
+
+include_HEADERS = ezxml.h
+
+lib_LTLIBRARIES = libezxml.la
+libezxml_la_SOURCES = ezxml.c
+libezxml_la_LDFLAGS = -no-undefined
+
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/ezxml/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/ezxml/configure.ac	2024-10-05 14:54:04.851670200 +0900
@@ -0,0 +1,30 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([ezxml], [0.8.6])
+AC_CONFIG_SRCDIR([ezxml.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([fcntl.h stdlib.h string.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_MMAP
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([memmove memset munmap strchr strcspn strdup strspn strstr strtol])
+
+AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([ezxml.pc])
+AC_OUTPUT
--- origsrc/ezxml/ezxml.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/ezxml/ezxml.pc.in	2024-10-05 14:54:04.836042000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: ezxml
+Description: XML Parsing C Library
+Version: @VERSION@
+Libs: -L${libdir} -lezxml
+Cflags: -I${includedir}
