--- origsrc/libfov-1.0.4/Makefile.am	2009-09-20 09:12:58.000000000 +0900
+++ src/libfov-1.0.4/Makefile.am	2010-12-03 20:46:06.066404300 +0900
@@ -1,8 +1,8 @@
 
-SUBDIRS = fov tests examples docs
+SUBDIRS = fov examples docs #tests
 CLEANFILES = *~
 EXTRA_DIST=autogen.sh
-ACLOCAL_AMFLAGS=-I m4
+#ACLOCAL_AMFLAGS=-I m4
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libfov.pc
--- origsrc/libfov-1.0.4/configure.in	2009-09-20 09:19:04.000000000 +0900
+++ src/libfov-1.0.4/configure.in	2010-12-03 20:11:35.194221700 +0900
@@ -57,7 +57,7 @@ dnl ------------------------------------
 
 AC_PROG_CC
 AC_PROG_CXX
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_LANG_CPLUSPLUS
 
 CFLAGS="-Wall -ansi -O2"
--- origsrc/libfov-1.0.4/examples/Makefile.am	2009-09-20 09:10:30.000000000 +0900
+++ src/libfov-1.0.4/examples/Makefile.am	2010-12-03 20:50:51.456775500 +0900
@@ -21,7 +21,7 @@ noinst_PROGRAMS = $(SDL_PROGS) $(CURSES_
 AM_CFLAGS = @SDL_CFLAGS@ -O2 -ansi -pedantic -Wall -pedantic-errors -Wfloat-equal -Werror -Wno-unused
 AM_CXXFLAGS = $(AM_CFLAGS)
 
-simple_LDADD = @top_srcdir@/fov/libfov.la @SDL_LIBS@
+simple_LDADD = @top_builddir@/fov/libfov.la @SDL_LIBS@
 simple_SOURCES = \
 simple.cc \
 display.h \
--- origsrc/libfov-1.0.4/fov/Makefile.am	2009-09-20 09:10:27.000000000 +0900
+++ src/libfov-1.0.4/fov/Makefile.am	2010-12-03 20:38:46.924949900 +0900
@@ -1,5 +1,5 @@
 CLEANFILES = *~
-AM_CFLAGS = -Wall -O2 -ansi -pedantic -pedantic-errors -Wfloat-equal -Werror
+AM_CFLAGS = -Wall -O2 -ansi -pedantic -pedantic-errors -Wfloat-equal
 
 library_includedir=$(includedir)/$(LIBFOV_LIBRARY_NAME)
 library_include_HEADERS = fov.h
@@ -10,7 +10,7 @@ libfov_config_DATA = config.h
 lib_LTLIBRARIES = libfov.la
 libfov_la_SOURCES = fov.c
 libfov_la_LIBS = $(LIBM)
-libfov_la_LDFLAGS= -version-info $(LIBFOV_LTVERSION)
+libfov_la_LDFLAGS= -version-info $(LIBFOV_LTVERSION) -no-undefined
 
 splint:
 	splint fov.c
