--- origsrc/libfov-1.0.4/Makefile.am	2009-09-20 09:12:58.000000000 +0900
+++ src/libfov-1.0.4/Makefile.am	2014-07-21 20:56:38.881138800 +0900
@@ -1,8 +1,7 @@
 
-SUBDIRS = fov tests examples docs
+SUBDIRS = fov examples docs
 CLEANFILES = *~
 EXTRA_DIST=autogen.sh
-ACLOCAL_AMFLAGS=-I m4
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libfov.pc
--- origsrc/libfov-1.0.4/examples/Makefile.am	2009-09-20 09:10:30.000000000 +0900
+++ src/libfov-1.0.4/examples/Makefile.am	2014-07-21 20:56:38.881138800 +0900
@@ -21,7 +21,7 @@ noinst_PROGRAMS = $(SDL_PROGS) $(CURSES_
 AM_CFLAGS = @SDL_CFLAGS@ -O2 -ansi -pedantic -Wall -pedantic-errors -Wfloat-equal -Werror -Wno-unused
 AM_CXXFLAGS = $(AM_CFLAGS)
 
-simple_LDADD = @top_srcdir@/fov/libfov.la @SDL_LIBS@
+simple_LDADD = $(top_builddir)/fov/libfov.la $(SDL_LIBS)
 simple_SOURCES = \
 simple.cc \
 display.h \
--- origsrc/libfov-1.0.4/fov/Makefile.am	2009-09-20 09:10:27.000000000 +0900
+++ src/libfov-1.0.4/fov/Makefile.am	2014-07-21 20:56:38.881138800 +0900
@@ -1,5 +1,5 @@
 CLEANFILES = *~
-AM_CFLAGS = -Wall -O2 -ansi -pedantic -pedantic-errors -Wfloat-equal -Werror
+AM_CFLAGS = -Wall -O2 -ansi -pedantic -pedantic-errors -Wfloat-equal
 
 library_includedir=$(includedir)/$(LIBFOV_LIBRARY_NAME)
 library_include_HEADERS = fov.h
@@ -10,7 +10,7 @@ libfov_config_DATA = config.h
 lib_LTLIBRARIES = libfov.la
 libfov_la_SOURCES = fov.c
 libfov_la_LIBS = $(LIBM)
-libfov_la_LDFLAGS= -version-info $(LIBFOV_LTVERSION)
+libfov_la_LDFLAGS= -no-undefined -version-info $(LIBFOV_LTVERSION)
 
 splint:
 	splint fov.c
