--- origsrc/libversion-2.8.1/libversion/CMakeLists.txt	2018-10-31 03:16:39.000000000 +0900
+++ src/libversion-2.8.1/libversion/CMakeLists.txt	2019-01-25 12:36:57.045363200 +0900
@@ -28,6 +28,6 @@ INSTALL(FILES
 	${CMAKE_CURRENT_BINARY_DIR}/config.h
 	DESTINATION include/libversion
 )
-INSTALL(TARGETS version version_static LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+INSTALL(TARGETS version version_static RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
 
 INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/libversion.pc DESTINATION ${PKGCONFIGDIR})
--- origsrc/libversion-2.8.1/tests/CMakeLists.txt	2018-10-31 03:16:39.000000000 +0900
+++ src/libversion-2.8.1/tests/CMakeLists.txt	2019-01-25 12:37:27.284931700 +0900
@@ -1,12 +1,12 @@
 ADD_EXECUTABLE(compare_test compare_test.c)
-TARGET_LINK_LIBRARIES(compare_test version_static)
+TARGET_LINK_LIBRARIES(compare_test version)
 SET_TARGET_PROPERTIES(compare_test PROPERTIES COMPILE_DEFINITIONS LIBVERSION_NO_DEPRECATED)
 ADD_TEST(compare_test compare_test)
 
 ADD_EXECUTABLE(signature_test signature_test.c)
-TARGET_LINK_LIBRARIES(signature_test version_static)
+TARGET_LINK_LIBRARIES(signature_test version)
 ADD_TEST(signature_test signature_test)
 
 ADD_EXECUTABLE(compare_fuzzer compare_fuzzer.c)
-TARGET_LINK_LIBRARIES(compare_fuzzer version_static)
+TARGET_LINK_LIBRARIES(compare_fuzzer version)
 SET_TARGET_PROPERTIES(compare_fuzzer PROPERTIES COMPILE_DEFINITIONS LIBVERSION_NO_DEPRECATED)
--- origsrc/libversion-2.8.1/version_compare/CMakeLists.txt	2018-10-31 03:16:39.000000000 +0900
+++ src/libversion-2.8.1/version_compare/CMakeLists.txt	2019-01-25 12:37:39.134866500 +0900
@@ -1,4 +1,4 @@
 ADD_EXECUTABLE(version_compare version_compare.c)
-TARGET_LINK_LIBRARIES(version_compare version_static)
+TARGET_LINK_LIBRARIES(version_compare version)
 SET_TARGET_PROPERTIES(version_compare PROPERTIES COMPILE_DEFINITIONS LIBVERSION_NO_DEPRECATED)
 INSTALL(TARGETS version_compare RUNTIME DESTINATION bin)
