--- origsrc/ememoa/configure.ac	2010-07-25 02:37:11.000000000 +0900
+++ src/ememoa/configure.ac	2010-11-11 22:11:00.443855100 +0900
@@ -22,7 +22,7 @@ AM_INIT_AUTOMAKE([1.9.2 foreign dist-bzi
 AM_CONFIG_HEADER([config.h])
 
 # Libtool
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 AC_ARG_ENABLE([debug],
         [AS_HELP_STRING([--enable-debug], [for debugging with gdb])],
--- origsrc/ememoa/src/lib/ememoa/Makefile.am	2010-07-25 02:37:09.000000000 +0900
+++ src/ememoa/src/lib/ememoa/Makefile.am	2010-11-11 22:10:18.082943100 +0900
@@ -10,7 +10,7 @@ libememoa_la_SOURCES 	=			\
 	mempool_struct.h
 libememoa_la_CFLAGS	= $(PTHREAD_CFLAGS) @COVERAGE_CFLAGS@
 libememoa_la_LIBADD     = @COVERAGE_LIBS@
-libememoa_la_LDFLAGS	= $(PTHREAD_CFLAGS) $(PTHREAD_LIBS) -version-info 0:26:0
+libememoa_la_LDFLAGS	= $(PTHREAD_CFLAGS) $(PTHREAD_LIBS) -version-info 0:26:0 -no-undefined
 CC			= $(PTHREAD_CC)
 INCLUDES		= -I$(top_srcdir)/include
 
