--- origsrc/ememoa/src/lib/ememoa/Makefile.am	2008-07-31 00:27:08.000000000 +0900
+++ src/ememoa/src/lib/ememoa/Makefile.am	2014-11-18 23:27:35.847973900 +0900
@@ -10,7 +10,7 @@ libememoa_la_SOURCES 	=			\
 	mempool_struct.h
 libememoa_la_CFLAGS	= $(PTHREAD_CFLAGS) @COVERAGE_CFLAGS@
 libememoa_la_LIBADD     = @COVERAGE_LIBS@
-libememoa_la_LDFLAGS	= $(PTHREAD_CFLAGS) $(PTHREAD_LIBS) -version-info 0:26:0
+libememoa_la_LDFLAGS	= -no-undefined $(PTHREAD_CFLAGS) $(PTHREAD_LIBS) -version-info 0:26:0
 CC			= $(PTHREAD_CC)
 INCLUDES		= -I$(top_srcdir)/include
 
