--- origsrc/xrmc-6.6.0/src/Makefile.am	2017-04-12 21:06:53.000000000 +0900
+++ src/xrmc-6.6.0/src/Makefile.am	2018-07-10 00:04:55.739741900 +0900
@@ -39,7 +39,7 @@ SUBLIBS = \
 lib_LTLIBRARIES = libxrmc.la
 libxrmc_la_SOURCES = 
 libxrmc_la_LIBADD = $(SUBLIBS) @xraylib_LIBS@ -lm
-libxrmc_la_LDFLAGS = -version-info @LIB_CURRENT@:@LIB_REVISION@:@LIB_AGE@ @OPENMP_CXXFLAGS@ @LDFLAGS_EXTRA@ 
+libxrmc_la_LDFLAGS = -no-undefined -version-info @LIB_CURRENT@:@LIB_REVISION@:@LIB_AGE@ @OPENMP_CXXFLAGS@ @LDFLAGS_EXTRA@ 
 libxrmc_la_LINK = $(LIBTOOL) --tag=CXX $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(libxrmc_la_LDFLAGS) $(LDFLAGS) -o $@
--- origsrc/xrmc-6.6.0/src/xmi-msim/Makefile.am	2017-04-24 19:23:26.000000000 +0900
+++ src/xrmc-6.6.0/src/xmi-msim/Makefile.am	2018-07-10 00:03:56.198274700 +0900
@@ -39,4 +39,4 @@ plugin_LTLIBRARIES = xrmc-xmimsim.la
 xrmc_xmimsim_la_CFLAGS = @xmi_msim_CFLAGS@ @gmodule_CFLAGS@ @OPENMP_CFLAGS@
 xrmc_xmimsim_la_LIBADD = @xmi_msim_LIBS@ @gmodule_LIBS@
 xrmc_xmimsim_la_SOURCES = xrmc-xmimsim.c
-xrmc_xmimsim_la_LDFLAGS = -module -avoid-version -shared @LDFLAGS_EXTRA@
+xrmc_xmimsim_la_LDFLAGS = -no-undefined -module -avoid-version -shared @LDFLAGS_EXTRA@
