--- origsrc/scalapack-2.2.1/CMakeLists.txt	2022-03-02 07:01:06.000000000 +0900
+++ src/scalapack-2.2.1/CMakeLists.txt	2022-06-13 23:20:44.830746500 +0900
@@ -62,6 +62,11 @@ if (MPI_FOUND)
       message(STATUS "--> Fortran Compiler : ${CMAKE_Fortran_COMPILER}")
    endif()
 
+   if (CYGWIN)
+      set(PC_REQUIRES_MPI "ompi")
+   else()
+      set(PC_REQUIRES_MPI "mpi")
+   endif()
 else()
    message(STATUS "Found MPI_LIBRARY : ${MPI_FOUND} ")
    set(MPI_BASE_DIR ${MPI_BASE_DIR} CACHE PATH "MPI Path")
@@ -84,11 +89,18 @@ if (UNIX)
 endif ()
 
 macro(SCALAPACK_install_library lib)
-  install(TARGETS ${lib} EXPORT scalapack-targets
-    ARCHIVE DESTINATION lib${LIB_SUFFIX}
-    LIBRARY DESTINATION lib${LIB_SUFFIX}
-    RUNTIME DESTINATION Testing
-  )
+   if (CYGWIN)
+      install(TARGETS ${lib} EXPORT scalapack-targets
+         ARCHIVE DESTINATION lib
+         RUNTIME DESTINATION bin
+      )
+   else()
+      install(TARGETS ${lib} EXPORT scalapack-targets
+         ARCHIVE DESTINATION lib${LIB_SUFFIX}
+         LIBRARY DESTINATION lib${LIB_SUFFIX}
+         RUNTIME DESTINATION Testing
+      )
+   endif()
 endmacro()
 
 # --------------------------------------------------
@@ -237,7 +249,7 @@ if (UNIX)
    set_target_properties(scalapack PROPERTIES
     VERSION ${SCALAPACK_VERSION}
     SOVERSION ${SCALAPACK_VERSION_MAJOR}.${SCALAPACK_VERSION_MINOR})
-   target_link_libraries( scalapack ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES})
+   target_link_libraries( scalapack ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES} ${MPI_LIBRARIES})
    scalapack_install_library(scalapack)
 else (UNIX) # Need to separate Fortran and C Code
    OPTION(BUILD_SHARED_LIBS "Build shared libraries" ON )
@@ -317,8 +329,8 @@ configure_file(${SCALAPACK_SOURCE_DIR}/C
 install(FILES
   ${SCALAPACK_BINARY_DIR}/CMakeFiles/scalapack-config.cmake
   ${SCALAPACK_BINARY_DIR}/scalapack-config-version.cmake
-  DESTINATION lib/cmake/scalapack-${SCALAPACK_VERSION}
+  DESTINATION lib/cmake/scalapack
   )
 
 install(EXPORT scalapack-targets
-  DESTINATION lib/cmake/scalapack-${SCALAPACK_VERSION})
+  DESTINATION lib/cmake/scalapack)
--- origsrc/scalapack-2.2.1/TESTING/EIG/CMakeLists.txt	2022-03-02 07:01:06.000000000 +0900
+++ src/scalapack-2.2.1/TESTING/EIG/CMakeLists.txt	2022-06-13 23:02:53.321692600 +0900
@@ -94,8 +94,8 @@ target_link_libraries(xdsyevr scalapack
 target_link_libraries(xcheevr scalapack ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES})
 target_link_libraries(xzheevr scalapack ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES})
 
-target_link_libraries(xshseqr scalapack ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES})
-target_link_libraries(xdhseqr scalapack ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES})
+target_link_libraries(xshseqr scalapack ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES} ${MPI_Fortran_LIBRARIES})
+target_link_libraries(xdhseqr scalapack ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES} ${MPI_Fortran_LIBRARIES})
 
 if ("${CMAKE_Fortran_COMPILER_ID}" STREQUAL "GNU" AND CMAKE_Fortran_COMPILER_VERSION VERSION_GREATER_EQUAL 10)
     set( CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -std=legacy" )  # local to this directory
--- origsrc/scalapack-2.2.1/scalapack.pc.in	2022-03-02 07:01:06.000000000 +0900
+++ src/scalapack-2.2.1/scalapack.pc.in	2022-06-13 21:09:37.848762400 +0900
@@ -6,4 +6,4 @@ Description: SCALAPACK reference impleme
 Version: @SCALAPACK_VERSION@
 URL: http://www.netlib.org/scalapack/
 Libs: -L${libdir} -lscalapack
-Requires: mpi lapack blas
+Requires: @PC_REQUIRES_MPI@ lapack blas
