--- origsrc/protobuf-c-rpc/Makefile.am	2022-11-03 16:21:39.000000000 +0900
+++ src/protobuf-c-rpc/Makefile.am	2022-11-03 16:26:41.833690200 +0900
@@ -16,7 +16,7 @@ AM_CPPFLAGS = \
 	-I${top_builddir} \
 	-I${top_srcdir}
 AM_CFLAGS = ${my_CFLAGS}
-AM_LDFLAGS =
+AM_LDFLAGS = -no-undefined
 
 # code coverage
 
@@ -54,6 +54,8 @@ protobuf_c_rpc_libprotobuf_c_rpc_la_LDFL
 	-version-info $(LIBPROTOBUF_C_RPC_CURRENT):$(LIBPROTOBUF_C_RPC_REVISION):$(LIBPROTOBUF_C_RPC_AGE) \
 	-export-symbols-regex "^(protobuf_c_rpc_[a-z].*)"
 
+protobuf_c_rpc_libprotobuf_c_rpc_la_LIBADD = $(libprotobuf_c_LIBS)
+
 pkgconfig_DATA += protobuf-c-rpc/libprotobuf-c-rpc.pc
 CLEANFILES += protobuf-c-rpc/libprotobuf-c-rpc.pc
 EXTRA_DIST += protobuf-c-rpc/libprotobuf-c-rpc.pc.in
