--- origsrc/protobuf-c-rpc/Makefile.am	2017-06-29 21:31:06.000000000 +0900
+++ src/protobuf-c-rpc/Makefile.am	2017-06-29 21:48:37.122000000 +0900
@@ -55,6 +55,8 @@ protobuf_c_rpc_libprotobuf_c_rpc_la_LDFL
 	-export-symbols-regex "^(protobuf_c_rpc_[a-z].*)" \
 	-no-undefined
 
+protobuf_c_rpc_libprotobuf_c_rpc_la_LIBADD = $(libprotobuf_c_LIBS)
+
 pkgconfig_DATA += protobuf-c-rpc/libprotobuf-c-rpc.pc
 CLEANFILES += protobuf-c-rpc/libprotobuf-c-rpc.pc
 EXTRA_DIST += protobuf-c-rpc/libprotobuf-c-rpc.pc.in
