--- origsrc/geohash-0.2.0/CMakeLists.txt	1970-01-01 09:00:00.000000000 +0900
+++ src/geohash-0.2.0/CMakeLists.txt	2025-06-06 15:35:39.446205500 +0900
@@ -0,0 +1,29 @@
+cmake_minimum_required(VERSION 3.28)
+project(geohash-goodcleanfun
+  VERSION 0.2.0
+  DESCRIPTION "Geohash implementation in C with neighors and tile bounds"
+  HOMEPAGE_URL "https://github.com/goodcleanfun/geohash")
+
+include(GNUInstallDirs)
+include(CTest)
+
+add_library(geohash-goodcleanfun SHARED src/geohash.c)
+set_target_properties(geohash-goodcleanfun PROPERTIES SOVERSION 0)
+
+install(TARGETS geohash-goodcleanfun
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+add_executable(geohash-goodcleanfun-test test.c)
+target_include_directories(geohash-goodcleanfun-test PRIVATE src)
+target_link_libraries(geohash-goodcleanfun-test geohash-goodcleanfun)
+add_test(NAME geohash-goodcleanfun-test COMMAND geohash-goodcleanfun-test)
+
+install(FILES src/geohash.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} RENAME geohash-goodcleanfun.h)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/geohash-goodcleanfun.pc.in ${CMAKE_BINARY_DIR}/geohash-goodcleanfun.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/geohash-goodcleanfun.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/geohash-0.2.0/geohash-goodcleanfun.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/geohash-0.2.0/geohash-goodcleanfun.pc.in	2025-06-06 15:39:37.037797800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: geohash-goodcleanfun
+Description: Geohash implementation in C with neighors and tile bounds
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lgeohash-goodcleanfun
+Cflags: -I${includedir}
--- origsrc/geohash-0.2.0/test.c	2024-01-01 18:13:21.000000000 +0900
+++ src/geohash-0.2.0/test.c	2025-06-06 15:36:12.879978900 +0900
@@ -1,7 +1,7 @@
 #include <stdio.h>
 #include <string.h>
 #include "geohash.h"
-#include "greatest/greatest.h"
+#include <greatest.h>
 
 // Test values were computed to exact precision using arbitrary precision
 static const struct {
