--- origsrc/portage-utils-0.64/Makefile.am	2016-11-27 14:20:55.000000000 +0900
+++ src/portage-utils-0.64/Makefile.am	2017-12-01 00:03:22.886856600 +0900
@@ -11,7 +11,8 @@ bin_PROGRAMS = q
 q_SOURCES = main.c
 q_CPPFLAGS = \
 	-I$(top_builddir)/autotools/gnulib \
-	-I$(top_srcdir)/autotools/gnulib
+	-I$(top_srcdir)/autotools/gnulib \
+	-I/usr/include/iniparser
 q_LDADD = \
 	$(top_builddir)/autotools/gnulib/libgnu.a \
 	-liniparser \
