--- origsrc/portage-utils-0.86/Makefile.am	2020-02-21 17:49:28.000000000 +0900
+++ src/portage-utils-0.86/Makefile.am	2020-03-12 16:03:27.499814800 +0900
@@ -69,6 +69,7 @@ q_CPPFLAGS = \
 	-I$(top_srcdir)/libq \
 	-I$(top_builddir)/autotools/gnulib \
 	-I$(top_srcdir)/autotools/gnulib \
+	-I/usr/include/iniparser \
 	$(NULL)
 q_LDADD = \
 	$(top_builddir)/libq/libq.la \
--- origsrc/portage-utils-0.86/libq/Makefile.am	2020-02-21 17:49:28.000000000 +0900
+++ src/portage-utils-0.86/libq/Makefile.am	2020-03-12 16:03:27.499814800 +0900
@@ -34,4 +34,5 @@ libq_la_SOURCES = $(QFILES)
 libq_la_CPPFLAGS = \
 	$(OPENMP_CFLAGS) \
 	-I$(top_builddir)/autotools/gnulib \
-	-I$(top_srcdir)/autotools/gnulib
+	-I$(top_srcdir)/autotools/gnulib \
+	-I$(top_srcdir)
