--- origsrc/portage-utils-0.80/Makefile.am	2019-07-19 03:38:46.000000000 +0900
+++ src/portage-utils-0.80/Makefile.am	2019-08-16 17:30:35.229259400 +0900
@@ -69,6 +69,7 @@ q_CPPFLAGS = \
 	-I$(top_srcdir)/libq \
 	-I$(top_builddir)/autotools/gnulib \
 	-I$(top_srcdir)/autotools/gnulib \
+	-I/usr/include/iniparser \
 	$(NULL)
 q_LDADD = \
 	$(top_builddir)/libq/libq.la \
--- origsrc/portage-utils-0.80/libq/Makefile.am	2019-07-19 03:38:46.000000000 +0900
+++ src/portage-utils-0.80/libq/Makefile.am	2019-08-16 17:14:11.382871000 +0900
@@ -34,4 +34,5 @@ libq_la_SOURCES = $(QFILES)
 libq_la_CPPFLAGS = \
 	$(OPENMP_CFLAGS) \
 	-I$(top_builddir)/autotools/gnulib \
-	-I$(top_srcdir)/autotools/gnulib
+	-I$(top_srcdir)/autotools/gnulib \
+	-I$(top_srcdir)
