--- origsrc/hidapi-hidapi-0.15.0/libusb/CMakeLists.txt	2025-05-19 03:25:13.000000000 +0900
+++ src/hidapi-hidapi-0.15.0/libusb/CMakeLists.txt	2025-08-26 20:37:33.778825400 +0900
@@ -98,7 +98,7 @@ add_library(hidapi-libusb ALIAS hidapi_l
 
 if(HIDAPI_INSTALL_TARGETS)
     install(TARGETS hidapi_libusb EXPORT hidapi
-        LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
         ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
         PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/hidapi"
     )
