--- origsrc/libextractor-1.10/src/plugins/Makefile.am	2018-06-21 05:02:55.000000000 +0900
+++ src/libextractor-1.10/src/plugins/Makefile.am	2020-07-24 00:28:00.134689300 +0900
@@ -581,7 +581,7 @@ libextractor_rpm_la_SOURCES = \
 libextractor_rpm_la_LDFLAGS = \
   $(PLUGINFLAGS)
 libextractor_rpm_la_LIBADD = \
-  -lrpm -lpthread $(XLIB)
+  -lrpm -lrpmio -lpthread $(XLIB)
 
 test_rpm_SOURCES = \
   test_rpm.c
--- origsrc/libextractor-1.10/src/plugins/html_extractor.c	2020-03-03 20:54:56.000000000 +0900
+++ src/libextractor-1.10/src/plugins/html_extractor.c	2020-07-24 00:23:28.200320200 +0900
@@ -27,7 +27,7 @@
 #include "extractor.h"
 #include <magic.h>
 #include <tidy/tidy.h>
-#include <tidy/tidybuffio.h>
+#include <tidy/buffio.h>
 
 /**
  * Mapping of HTML META names to LE types.
