--- origsrc/libextractor-1.13/src/plugins/Makefile.am	2021-05-08 18:10:13.000000000 +0900
+++ src/libextractor-1.13/src/plugins/Makefile.am	2024-05-09 06:41:58.985282500 +0900
@@ -600,7 +600,7 @@ libextractor_rpm_la_SOURCES = \
 libextractor_rpm_la_LDFLAGS = \
   $(PLUGINFLAGS)
 libextractor_rpm_la_LIBADD = \
-  -lrpm -lpthread $(XLIB)
+  -lrpm -lrpmio -lpthread $(XLIB)
 
 test_rpm_SOURCES = \
   test_rpm.c
--- origsrc/libextractor-1.13/src/plugins/html_extractor.c	2021-04-08 20:30:42.000000000 +0900
+++ src/libextractor-1.13/src/plugins/html_extractor.c	2024-05-09 06:41:58.985282500 +0900
@@ -27,7 +27,7 @@
 #include "extractor.h"
 #include <magic.h>
 #include <tidy/tidy.h>
-#include <tidy/tidybuffio.h>
+#include <tidy/buffio.h>
 
 /**
  * Mapping of HTML META names to LE types.
