--- origsrc/libextractor-1.11/src/plugins/Makefile.am	2021-01-30 22:20:23.000000000 +0900
+++ src/libextractor-1.11/src/plugins/Makefile.am	2021-02-04 20:18:20.251495100 +0900
@@ -581,7 +581,7 @@ libextractor_rpm_la_SOURCES = \
 libextractor_rpm_la_LDFLAGS = \
   $(PLUGINFLAGS)
 libextractor_rpm_la_LIBADD = \
-  -lrpm -lpthread $(XLIB)
+  -lrpm -lrpmio -lpthread $(XLIB)
 
 test_rpm_SOURCES = \
   test_rpm.c
--- origsrc/libextractor-1.11/src/plugins/html_extractor.c	2021-01-30 22:20:23.000000000 +0900
+++ src/libextractor-1.11/src/plugins/html_extractor.c	2021-02-04 20:18:20.268524400 +0900
@@ -27,7 +27,7 @@
 #include "extractor.h"
 #include <magic.h>
 #include <tidy/tidy.h>
-#include <tidy/tidybuffio.h>
+#include <tidy/buffio.h>
 
 /**
  * Mapping of HTML META names to LE types.
