--- origsrc/libepp-nicbr-1.13/Makefile.am	2013-05-16 01:49:29.000000000 +0900
+++ src/libepp-nicbr-1.13/Makefile.am	2013-06-02 12:05:54.443054700 +0900
@@ -27,15 +27,15 @@ SUBDIRS = lib $(SHEPP_DIR) unit_test
 DIST_SUBDIRS = lib src/shepp unit_test
 
 libepp_nicbr_includedir = $(includedir)/libepp_nicbr
-dist_libepp_nicbr_include_HEADERS = include/*.H include/libepp_nicbr.H.template
+dist_libepp_nicbr_include_HEADERS = $(top_srcdir)/include/*.H include/libepp_nicbr.H.template
 
 EXTRA_DIST = config/libepp_dox samples
 
 templatesdir = $(prefix)/share/libepp_nicbr/templates
-dist_templates_DATA = docs/templates/*.xml
+dist_templates_DATA = $(top_srcdir)/docs/templates/*.xml
 
 schemasdir = $(prefix)/share/libepp_nicbr/schemas
-dist_schemas_DATA = docs/schemas/*.xsd
+dist_schemas_DATA = $(top_srcdir)/docs/schemas/*.xsd
 
 sslcertdir = $(prefix)/share/libepp_nicbr/ssl
 dist_sslcert_DATA = unit_test/root.pem unit_test/client.pem
--- origsrc/libepp-nicbr-1.13/configure.ac	2013-05-16 01:49:29.000000000 +0900
+++ src/libepp-nicbr-1.13/configure.ac	2013-06-02 12:08:35.664734400 +0900
@@ -43,7 +43,7 @@ AC_PROG_CC
 AC_PROG_CPP
 AC_PROG_CXX
 AC_PROG_CXXCPP
-AC_PROG_RANLIB
+LT_INIT
 
 if test "$CFLAGS" = "-g -O2" || test "$CFLAGS" = "NONE"
 then
--- origsrc/libepp-nicbr-1.13/lib/Makefile.am	2013-05-16 01:49:29.000000000 +0900
+++ src/libepp-nicbr-1.13/lib/Makefile.am	2013-06-02 12:07:12.522156300 +0900
@@ -23,9 +23,9 @@
  ##
 ## $Id: Makefile.tmp 1100 2011-02-02 14:58:12Z rafael $ ## 
 
-lib_LIBRARIES = libepp-nicbr-1.13.a
+lib_LTLIBRARIES = libepp-nicbr-1.13.la
 
-libepp_nicbr_1_13_a_SOURCES = $(top_srcdir)/src/lib/StrUtil.cpp \
+libepp_nicbr_1_13_la_SOURCES = $(top_srcdir)/src/lib/StrUtil.cpp \
 	$(top_srcdir)/src/lib/Login.cpp \
 	$(top_srcdir)/src/lib/TransportTLSCommon.cpp \
 	$(top_srcdir)/src/lib/TransportTLS.cpp 	\
@@ -79,4 +79,6 @@ libepp_nicbr_1_13_a_SOURCES = $(top_srcd
 	$(top_srcdir)/src/lib/AsnInfo.cpp \
 	$(top_srcdir)/src/lib/AsnUpdate.cpp
 
-libepp_nicbr_1_13_a_CPPFLAGS = -I ../include
+libepp_nicbr_1_13_la_CPPFLAGS = -I $(top_srcdir)/include
+libepp_nicbr_1_13_la_LDFLAGS = -no-undefined
+libepp_nicbr_1_13_la_LIBADD = -lssl -lxerces-c
--- origsrc/libepp-nicbr-1.13/src/shepp/Makefile.am	2013-05-16 01:49:29.000000000 +0900
+++ src/libepp-nicbr-1.13/src/shepp/Makefile.am	2013-06-02 12:07:38.837586000 +0900
@@ -41,4 +41,4 @@ shepp_SOURCES = shepp.cpp \
 	SheppObjSet.H \
 	SheppPrint.H \
 	SheppStrUtil.H
-shepp_LDADD = $(top_srcdir)/lib/libepp-nicbr-1.13.a -lxerces-c -lcrypto -lssl
+shepp_LDADD = $(top_builddir)/lib/libepp-nicbr-1.13.la -lxerces-c -lcrypto -lssl
--- origsrc/libepp-nicbr-1.13/unit_test/Makefile.am	2013-05-16 01:49:29.000000000 +0900
+++ src/libepp-nicbr-1.13/unit_test/Makefile.am	2013-06-02 12:08:08.534851600 +0900
@@ -37,7 +37,7 @@ TransportExceptionTestServer_SOURCES = T
 TransportExceptionTestServer_LDADD = -lssl
 
 SessionTestServer_SOURCES = SessionTestServer.cpp
-SessionTestServer_LDADD = $(top_srcdir)/lib/libepp-nicbr-1.13.a -lssl -lxerces-c
+SessionTestServer_LDADD = $(top_builddir)/lib/libepp-nicbr-1.13.la -lssl -lxerces-c
 
 CppUnitTest_SOURCES = CppUnitTestMain.cpp \
 		StrUtilTest.cpp StrUtilTest.H \
@@ -90,4 +90,4 @@ CppUnitTest_SOURCES = CppUnitTestMain.cp
 		AsnInfoTest.cpp AsnInfoTest.H \
 		AsnUpdateTest.cpp AsnUpdateTest.H
 
-CppUnitTest_LDADD = $(top_srcdir)/lib/libepp-nicbr-1.13.a -lcppunit -lssl -lxerces-c
+CppUnitTest_LDADD = $(top_builddir)/lib/libepp-nicbr-1.13.la -lcppunit -lssl -lxerces-c
