--- origsrc/libdap-3.21.1/Makefile.am	2025-01-14 08:03:54.000000000 +0900
+++ src/libdap-3.21.1/Makefile.am	2025-03-01 18:26:36.482105900 +0900
@@ -54,18 +54,18 @@ libparsers_la_LIBADD = $(XML2_LIBS)
 
 libdap_la_SOURCES = $(DAP_SRC) $(GNU_SRC)  $(DAP4_ONLY_SRC)
 
-libdap_la_LDFLAGS = -version-info $(LIBDAP_VERSION)
+libdap_la_LDFLAGS = -no-undefined -version-info $(LIBDAP_VERSION)
 libdap_la_CPPFLAGS = $(AM_CPPFLAGS)
 libdap_la_LIBADD = $(XML2_LIBS) $(PTHREAD_LIBS) gl/libgnu.la d4_ce/libd4_ce_parser.la \
 d4_function/libd4_function_parser.la libparsers.la $(CRYPTO_LIBS)
 
 libdapclient_la_SOURCES = $(CLIENT_SRC) $(DAP4_CLIENT_HDR) $(DAP4_CLIENT_SRC)
-libdapclient_la_LDFLAGS = -version-info $(CLIENTLIB_VERSION)
+libdapclient_la_LDFLAGS = -no-undefined -version-info $(CLIENTLIB_VERSION)
 libdapclient_la_CPPFLAGS = $(AM_CPPFLAGS) $(CURL_CFLAGS)
 libdapclient_la_LIBADD = $(CURL_LIBS) libdap.la $(PTHREAD_LIBS)
 
 libdapserver_la_SOURCES = $(SERVER_SRC)
-libdapserver_la_LDFLAGS = -version-info $(SERVERLIB_VERSION)
+libdapserver_la_LDFLAGS = -no-undefined -version-info $(SERVERLIB_VERSION)
 libdapserver_la_LIBADD = libdap.la $(UUID_LIBS)
 
 pkginclude_HEADERS = $(DAP_HDR) $(GNU_HDR) $(CLIENT_HDR) $(SERVER_HDR) $(DAP4_ONLY_HDR) $(DAP4_CLIENT_HDR)
--- origsrc/libdap-3.21.1/debug.h	2025-01-14 08:03:54.000000000 +0900
+++ src/libdap-3.21.1/debug.h	2025-03-01 18:27:47.687591300 +0900
@@ -55,7 +55,9 @@ using std::string;
 #define DBG(x) FILE_N_LINE, x
 #define DBGN(x) x
 #else
+#if !defined(DBG)
 #define DBG(x)  /* x */
+#endif
 #define DBGN(x) /* x */
 #endif
 
--- origsrc/libdap-3.21.1/getdap4.cc	2025-01-14 08:03:54.000000000 +0900
+++ src/libdap-3.21.1/getdap4.cc	2025-03-01 18:28:26.227115000 +0900
@@ -45,6 +45,10 @@
 #include <string>
 #include <unistd.h> // getopt
 
+#if defined(__CYGWIN__)
+#include </usr/include/getopt.h>
+#endif
+
 #include "D4BaseTypeFactory.h"
 #include "D4Connect.h"
 #include "D4Group.h"
--- origsrc/libdap-3.21.1/unit-tests/run_tests_cppunit.h	2025-01-14 08:03:54.000000000 +0900
+++ src/libdap-3.21.1/unit-tests/run_tests_cppunit.h	2025-03-01 18:28:57.018093100 +0900
@@ -40,6 +40,10 @@
 #include <cppunit/extensions/HelperMacros.h>
 #include <cppunit/extensions/TestFactoryRegistry.h>
 
+#if defined(__CYGWIN__)
+#include </usr/include/getopt.h>
+#endif
+
 bool debug = false;
 
 #undef DBG
--- origsrc/libdap-3.21.1/xdr-datatypes.h	2025-01-14 08:20:31.000000000 +0900
+++ src/libdap-3.21.1/xdr-datatypes.h	2025-03-01 18:30:01.655445500 +0900
@@ -18,14 +18,25 @@
 #include <rpc/xdr.h>
 #endif
 
-
+#if !defined(XDR_INT32)
 #define XDR_INT32 xdr_int32_t
+#endif
+#if !defined(XDR_UINT32)
 #define XDR_UINT32 xdr_u_int32_t
+#endif
 
+#if !defined(XDR_INT16)
 #define XDR_INT16 xdr_int16_t
+#endif
+#if !defined(XDR_UINT16)
 #define XDR_UINT16 xdr_u_int16_t
+#endif
 
+#if !defined(XDR_FLOAT64)
 #define XDR_FLOAT64 xdr_double
+#endif
+#if !defined(XDR_FLOAT32)
 #define XDR_FLOAT32 xdr_float
+#endif
 
 #endif /* __XDR_DATATYPES__ */
