--- origsrc/libdap-3.20.10/Makefile.am	2022-03-16 20:02:01.000000000 +0900
+++ src/libdap-3.20.10/Makefile.am	2022-07-03 21:58:06.332201500 +0900
@@ -63,18 +63,18 @@ libparsers_la_LIBADD = $(XML2_LIBS)
 
 libdap_la_SOURCES = $(DAP_SRC) $(GNU_SRC)  $(DAP4_ONLY_SRC)
 
-libdap_la_LDFLAGS = -version-info $(LIBDAP_VERSION)
+libdap_la_LDFLAGS = -no-undefined -version-info $(LIBDAP_VERSION)
 libdap_la_CPPFLAGS = $(AM_CPPFLAGS)
 libdap_la_LIBADD = $(XML2_LIBS) $(PTHREAD_LIBS) gl/libgnu.la d4_ce/libd4_ce_parser.la \
 d4_function/libd4_function_parser.la libparsers.la $(CRYPTO_LIBS)
 
 libdapclient_la_SOURCES = $(CLIENT_SRC) $(DAP4_CLIENT_HDR) $(DAP4_CLIENT_SRC)
-libdapclient_la_LDFLAGS = -version-info $(CLIENTLIB_VERSION)
+libdapclient_la_LDFLAGS = -no-undefined -version-info $(CLIENTLIB_VERSION)
 libdapclient_la_CPPFLAGS = $(AM_CPPFLAGS) $(CURL_CFLAGS)
 libdapclient_la_LIBADD = $(CURL_LIBS) libdap.la $(PTHREAD_LIBS)
 
 libdapserver_la_SOURCES = $(SERVER_SRC)
-libdapserver_la_LDFLAGS = -version-info $(SERVERLIB_VERSION)
+libdapserver_la_LDFLAGS = -no-undefined -version-info $(SERVERLIB_VERSION)
 libdapserver_la_LIBADD = libdap.la $(UUID_LIBS)
 
 pkginclude_HEADERS = $(DAP_HDR) $(GNU_HDR) $(CLIENT_HDR) $(SERVER_HDR) $(DAP4_ONLY_HDR) $(DAP4_CLIENT_HDR)
--- origsrc/libdap-3.20.10/getdap4.cc	2022-03-16 20:02:01.000000000 +0900
+++ src/libdap-3.20.10/getdap4.cc	2022-07-04 00:37:07.630500300 +0900
@@ -45,6 +45,10 @@
 #include <sstream>
 #include <unistd.h>     // getopt
 
+#if defined(__CYGWIN__)
+#include </usr/include/getopt.h>
+#endif
+
 #include "DMR.h"
 #include "XMLWriter.h"
 #include "D4BaseTypeFactory.h"
