--- origsrc/libdap-3.11.1/AlarmHandler.h	2011-04-29 22:19:43.000000000 +0900
+++ src/libdap-3.11.1/AlarmHandler.h	2011-05-27 03:01:17.078475500 +0900
@@ -27,7 +27,7 @@
 #define alarm_handler_h
 
 #include <cstdio>
-
+#include <cstdlib>
 #include <string>
 
 #include "EventHandler.h"
--- origsrc/libdap-3.11.1/Makefile.am	2011-04-29 22:19:43.000000000 +0900
+++ src/libdap-3.11.1/Makefile.am	2011-05-27 02:12:51.612101800 +0900
@@ -42,17 +42,17 @@ BUILT_SOURCES = $(GRAM_SRC) dods-datatyp
 libparsers_la_SOURCES = $(GRAM_SRC) 
 
 libdap_la_SOURCES = $(DAP_SRC) $(GNU_SRC) 
-libdap_la_LDFLAGS = -version-info $(LIBDAP_VERSION)
+libdap_la_LDFLAGS = -version-info $(LIBDAP_VERSION) -no-undefined
 libdap_la_CPPFLAGS = $(AM_CPPFLAGS) $(XML2_CFLAGS)
 libdap_la_LIBADD = $(XML2_LIBS) $(PTHREAD_LIBS) gl/libgnu.la libparsers.la
 
 libdapclient_la_SOURCES = $(CLIENT_SRC) 
-libdapclient_la_LDFLAGS = -version-info $(CLIENTLIB_VERSION)
+libdapclient_la_LDFLAGS = -version-info $(CLIENTLIB_VERSION) -no-undefined
 libdapclient_la_CPPFLAGS = $(AM_CPPFLAGS) $(CURL_CFLAGS) $(XML2_CFLAGS)
 libdapclient_la_LIBADD = $(CURL_LIBS) libdap.la $(PTHREAD_LIBS)
 
 libdapserver_la_SOURCES = $(SERVER_SRC)
-libdapserver_la_LDFLAGS = -version-info $(SERVERLIB_VERSION)
+libdapserver_la_LDFLAGS = -version-info $(SERVERLIB_VERSION) -no-undefined
 libdapserver_la_LIBADD = libdap.la $(UUID_LIBS)
 
 pkginclude_HEADERS = $(DAP_HDR) $(GNU_HDR) $(CLIENT_HDR) $(SERVER_HDR) 
--- origsrc/libdap-3.11.1/XDRFileUnMarshaller.cc	2011-04-29 22:19:43.000000000 +0900
+++ src/libdap-3.11.1/XDRFileUnMarshaller.cc	2011-05-27 02:55:29.228291500 +0900
@@ -51,6 +51,8 @@
 #include "util.h"
 #include "InternalErr.h"
 
+#include <cstdlib>
+
 namespace libdap {
 
 XDRFileUnMarshaller::XDRFileUnMarshaller( FILE *out )
--- origsrc/libdap-3.11.1/XDRStreamMarshaller.cc	2011-04-29 22:19:43.000000000 +0900
+++ src/libdap-3.11.1/XDRStreamMarshaller.cc	2011-05-27 02:56:44.666766700 +0900
@@ -38,6 +38,8 @@
 #include "util.h"
 #include "debug.h"
 
+#include <cstdlib>
+
 namespace libdap {
 
 char *XDRStreamMarshaller::_buf = 0 ;
--- origsrc/libdap-3.11.1/XDRStreamUnMarshaller.cc	2011-04-29 22:19:43.000000000 +0900
+++ src/libdap-3.11.1/XDRStreamUnMarshaller.cc	2011-05-27 02:57:39.495606700 +0900
@@ -36,6 +36,8 @@
 #include <string>
 #include <sstream>
 
+#include <cstdlib>
+
 #define DODS_DEBUG2 1
 #define DODS_DEBUG 1
 
--- origsrc/libdap-3.11.1/XDRUtils.cc	2011-04-29 22:19:43.000000000 +0900
+++ src/libdap-3.11.1/XDRUtils.cc	2011-05-27 02:54:51.504046700 +0900
@@ -34,6 +34,8 @@
 #include "debug.h"
 #include "Str.h"
 
+#include <cstdlib>
+
 using namespace libdap ;
 
 // This function is used to allocate memory for, and initialize, a new XDR
--- origsrc/libdap-3.11.1/configure.ac	2011-04-29 22:19:42.000000000 +0900
+++ src/libdap-3.11.1/configure.ac	2011-05-27 02:12:26.220539700 +0900
@@ -90,7 +90,7 @@ AM_PROG_LEX
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
-AC_PROG_LIBTOOL
+LT_INIT
 
 dnl We really need bison and not yacc. If you use AC_PROG_YACC, the resulting 
 dnl Makefile will call bison -y which doesn't know how to make the parsers 
--- origsrc/libdap-3.11.1/util.cc	2011-04-29 22:19:43.000000000 +0900
+++ src/libdap-3.11.1/util.cc	2011-05-27 02:52:57.570217900 +0900
@@ -42,6 +42,8 @@ static char rcsid[] not_used =
 #include <cassert>
 #include <cstring>
 
+#include <limits.h>
+
 #include <ctype.h>
 #ifndef TM_IN_SYS_TIME
 #include <time.h>
