--- origsrc/libdap-3.11.0/Makefile.am	2010-08-27 06:49:24.000000000 +0900
+++ src/libdap-3.11.0/Makefile.am	2011-03-09 23:59:18.202509500 +0900
@@ -43,15 +43,15 @@ libdap_la_SOURCES = $(DAP_SRC) $(GNU_SRC
 libdapclient_la_SOURCES = $(CLIENT_SRC) 
 libdapserver_la_SOURCES = $(SERVER_SRC)
 
-libdap_la_LDFLAGS = -version-info $(LIBDAP_VERSION)
+libdap_la_LDFLAGS = -version-info $(LIBDAP_VERSION) -no-undefined
 libdap_la_CPPFLAGS = $(AM_CPPFLAGS) $(XML2_CFLAGS)
 libdap_la_LIBADD = $(XML2_LIBS) $(PTHREAD_LIBS) gl/libgnu.la
 
-libdapclient_la_LDFLAGS = -version-info $(CLIENTLIB_VERSION)
+libdapclient_la_LDFLAGS = -version-info $(CLIENTLIB_VERSION) -no-undefined
 libdapclient_la_CPPFLAGS = $(AM_CPPFLAGS) $(CURL_CFLAGS) $(XML2_CFLAGS)
 libdapclient_la_LIBADD = $(CURL_LIBS) libdap.la $(PTHREAD_LIBS)
 
-libdapserver_la_LDFLAGS = -version-info $(SERVERLIB_VERSION)
+libdapserver_la_LDFLAGS = -version-info $(SERVERLIB_VERSION) -no-undefined
 libdapserver_la_LIBADD = libdap.la $(UUID_LIBS)
 
 pkginclude_HEADERS = $(DAP_HDR) $(GNU_HDR) $(CLIENT_HDR) $(SERVER_HDR) 
--- origsrc/libdap-3.11.0/XDRFileUnMarshaller.cc	2010-08-27 06:49:24.000000000 +0900
+++ src/libdap-3.11.0/XDRFileUnMarshaller.cc	2011-03-10 00:29:39.959058200 +0900
@@ -51,6 +51,8 @@
 #include "util.h"
 #include "InternalErr.h"
 
+#include <cstdlib>
+
 namespace libdap {
 
 XDRFileUnMarshaller::XDRFileUnMarshaller( FILE *out )
--- origsrc/libdap-3.11.0/XDRStreamMarshaller.cc	2010-08-27 06:49:25.000000000 +0900
+++ src/libdap-3.11.0/XDRStreamMarshaller.cc	2011-03-10 00:30:10.241261800 +0900
@@ -38,6 +38,8 @@
 #include "util.h"
 #include "debug.h"
 
+#include <cstdlib>
+
 namespace libdap {
 
 char *XDRStreamMarshaller::_buf = 0 ;
--- origsrc/libdap-3.11.0/XDRStreamUnMarshaller.cc	2010-08-27 06:49:24.000000000 +0900
+++ src/libdap-3.11.0/XDRStreamUnMarshaller.cc	2011-03-10 00:30:39.357829300 +0900
@@ -33,6 +33,7 @@
 #include "XDRStreamUnMarshaller.h"
 
 #include <cstring> // for memcpy
+#include <cstdlib>
 #include <string>
 #include <sstream>
 
--- origsrc/libdap-3.11.0/XDRUtils.cc	2010-08-27 06:49:24.000000000 +0900
+++ src/libdap-3.11.0/XDRUtils.cc	2011-03-10 00:28:52.490802200 +0900
@@ -34,6 +34,8 @@
 #include "debug.h"
 #include "Str.h"
 
+#include <cstdlib>
+
 using namespace libdap ;
 
 // This function is used to allocate memory for, and initialize, a new XDR
--- origsrc/libdap-3.11.0/configure.ac	2010-09-14 07:04:33.000000000 +0900
+++ src/libdap-3.11.0/configure.ac	2011-03-09 23:58:29.712415300 +0900
@@ -77,7 +77,7 @@ AM_PROG_LEX
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 dnl We really need bison and not yacc. If you use AC_PROG_YACC, the resulting 
 dnl Makefile will call bison -y which doesn't know how to make the parsers 
--- origsrc/libdap-3.11.0/util.cc	2010-08-27 06:49:25.000000000 +0900
+++ src/libdap-3.11.0/util.cc	2011-03-10 00:24:48.434016900 +0900
@@ -49,6 +49,8 @@ static char rcsid[] not_used =
 #include <sys/time.h>
 #endif
 
+#include <limits.h>
+
 #ifndef WIN32
 #include <unistd.h>    // for stat
 #else
