--- origsrc/dumb-2.0.3/pkg-config.pc.in	2018-01-30 12:36:51.000000000 +0900
+++ src/dumb-2.0.3/pkg-config.pc.in	2025-06-17 06:09:05.008633700 +0900
@@ -7,6 +7,6 @@ Name: DUMB
 Description: DUMB is a module audio renderer library. 
 Version: @DUMB_VERSION@
 URL: https://github.com/kode54/dumb/
-Libs: -L${libdir} -l@PROJECT_NAME@
+Libs: -L${libdir} -ldumb
 Libs.private: -lm
 Cflags: -I${includedir} 
