--- origsrc/mlpack-mlpack-3.1.0/CMakeLists.txt	2019-04-26 14:07:37.000000000 +0900
+++ src/mlpack-mlpack-3.1.0/CMakeLists.txt	2019-05-02 14:12:53.930082100 +0900
@@ -525,7 +525,7 @@ if (BUILD_CLI_EXECUTABLES AND UNIX)
 
     # Set the rules to install the documentation.
     install(DIRECTORY ${CMAKE_BINARY_DIR}/share/man/
-        DESTINATION ${CMAKE_INSTALL_MANDIR})
+        DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
   endif ()
 endif ()
 
--- origsrc/mlpack-mlpack-3.1.0/src/mlpack/CMakeLists.txt	2019-04-26 14:07:37.000000000 +0900
+++ src/mlpack-mlpack-3.1.0/src/mlpack/CMakeLists.txt	2019-05-02 09:22:09.026086400 +0900
@@ -36,7 +36,7 @@ endif ()
 set_target_properties(mlpack PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
 include(GenerateExportHeader)
 generate_export_header(mlpack EXPORT_FILE_NAME mlpack_export.hpp)
-if (NOT BUILD_SHARED_LIBS)
+if (NOT BUILD_SHARED_LIBS OR CYGWIN)
   add_definitions(-DMLPACK_STATIC_DEFINE)
 endif ()
 
