--- origsrc/mlpack-3.3.1/CMakeLists.txt	2020-04-30 06:36:40.000000000 +0900
+++ src/mlpack-3.3.1/CMakeLists.txt	2020-06-11 20:52:32.951558900 +0900
@@ -594,7 +594,7 @@ if (BUILD_CLI_EXECUTABLES AND UNIX)
 
     # Set the rules to install the documentation.
     install(DIRECTORY "${CMAKE_BINARY_DIR}/share/man/"
-        DESTINATION "${CMAKE_INSTALL_MANDIR}")
+        DESTINATION "${CMAKE_INSTALL_MANDIR}/man1")
   endif ()
 endif ()
 
--- origsrc/mlpack-3.3.1/src/mlpack/CMakeLists.txt	2020-04-30 06:36:40.000000000 +0900
+++ src/mlpack-3.3.1/src/mlpack/CMakeLists.txt	2020-06-11 20:52:32.967182100 +0900
@@ -36,7 +36,7 @@ endif ()
 set_target_properties(mlpack PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
 include(GenerateExportHeader)
 generate_export_header(mlpack EXPORT_FILE_NAME mlpack_export.hpp)
-if (NOT BUILD_SHARED_LIBS)
+if (NOT BUILD_SHARED_LIBS OR CYGWIN)
   add_definitions(-DMLPACK_STATIC_DEFINE)
 endif ()
 
