--- origsrc/gtest_mpi-2.0.0/CMakeLists.txt	2024-03-26 19:14:20.000000000 +0900
+++ src/gtest_mpi-2.0.0/CMakeLists.txt	2026-07-06 10:45:46.495342800 +0900
@@ -36,7 +36,7 @@ configure_file(${PROJECT_SOURCE_DIR}/cma
 	@ONLY
 )
 
-install(TARGETS gtest_mpi DESTINATION ${CMAKE_INSTALL_LIBDIR} EXPORT gtest_mpi-targets)
+install(TARGETS gtest_mpi EXPORT gtest_mpi-targets RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/gtest_mpi DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} FILES_MATCHING PATTERN "*.h" PATTERN "*.hpp")
 install(EXPORT gtest_mpi-targets NAMESPACE gtest_mpi:: DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/gtest_mpi FILE gtest_mpi-targets.cmake)
 install(
@@ -48,3 +48,10 @@ install(
 )
 
 add_library(gtest_mpi::gtest_mpi ALIAS gtest_mpi)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/gtest_mpi.pc.in ${CMAKE_BINARY_DIR}/gtest_mpi.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/gtest_mpi.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/gtest_mpi-2.0.0/gtest_mpi.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/gtest_mpi-2.0.0/gtest_mpi.pc.in	2026-07-06 10:45:57.336297900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: gtest_mpi
+Description: Extension to Googletest with MPI
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lgtest_mpi
+Cflags: -I${includedir}
