--- origsrc/musl-obstack-1.2.2/Makefile.am	2020-10-27 03:08:05.000000000 +0900
+++ src/musl-obstack-1.2.2/Makefile.am	2021-04-25 14:20:42.768792400 +0900
@@ -1,9 +1,8 @@
 ## Makefile.am - procress this file with automake to produce Makefile.in
 lib_LTLIBRARIES = libobstack.la
 libobstack_la_SOURCES = obstack.c obstack_printf.c
-libobstack_la_HEADERS = obstack.h
-libobstack_la_LDFLAGS = -version-info 1
-libobstack_ladir = $(includedir)
+include_HEADERS = obstack.h
+libobstack_la_LDFLAGS = -version-info 1 -no-undefined
 ACLOCAL_AMFLAGS = -Im4
 
 pkgconfig_DATA = musl-obstack.pc
--- origsrc/musl-obstack-1.2.2/configure.ac	2020-10-27 03:08:05.000000000 +0900
+++ src/musl-obstack-1.2.2/configure.ac	2021-04-25 14:21:00.552355900 +0900
@@ -2,7 +2,7 @@ dnl Process this file with autoconf to p
 
 AC_PREREQ(2.69)
 
-AC_INIT([obstack], [1.2], [pullmoll@t-online.de])
+AC_INIT([obstack], [1.2.2], [pullmoll@t-online.de])
 
 AM_INIT_AUTOMAKE([1.16])
 AC_CONFIG_MACRO_DIR([m4])
