--- origsrc/musl-obstack-1.1/Makefile.am	2017-01-14 19:17:44.000000000 +0900
+++ src/musl-obstack-1.1/Makefile.am	2017-11-24 14:41:13.078000000 +0900
@@ -1,8 +1,8 @@
 ## Makefile.am - procress this file with automake to produce Makefile.in
 lib_LTLIBRARIES = libobstack.la
 libobstack_la_SOURCES = obstack.c obstack_printf.c
-libobstack_la_HEADERS = obstack.h
-libobstack_ladir = $(includedir)
+libobstack_la_LDFLAGS = -no-undefined
+include_HEADERS = obstack.h
 ACLOCAL_AMFLAGS = -Im4
 
 pkgconfig_DATA = musl-obstack.pc
--- origsrc/musl-obstack-1.1/configure.ac	2017-01-14 19:17:44.000000000 +0900
+++ src/musl-obstack-1.1/configure.ac	2017-11-24 14:41:21.358000000 +0900
@@ -4,7 +4,7 @@ AC_PREREQ(2.69)
 
 AC_INIT([obstack], [1.1], [pullmoll@t-online.de])
 
-AM_INIT_AUTOMAKE([1.15])
+AM_INIT_AUTOMAKE
 AC_CONFIG_MACRO_DIR([m4])
 
 AC_PROG_CC
