--- origsrc/bozohttpd-20210227/Makefile.boot	2021-02-28 14:20:51.000000000 +0900
+++ src/bozohttpd-20210227/Makefile.boot	2021-05-28 11:18:25.856755100 +0900
@@ -6,14 +6,14 @@
 CC=	cc
 OPT=	-O
 LARGE_CFLAGS=	-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
-LOCAL_CFLAGS=	-DNO_LUA_SUPPORT -DNO_BLOCKLIST_SUPPORT -D_GNU_SOURCE -D_DEFAULT_SOURCE
-CFLAGS=	$(OPT) $(LARGE_CFLAGS) $(LOCAL_CFLAGS)
+LOCAL_CFLAGS=	-DNO_BLOCKLIST_SUPPORT -D_GNU_SOURCE -D_DEFAULT_SOURCE
+CFLAGS=	$(OPT) $(LARGE_CFLAGS) $(LOCAL_CFLAGS) `pkg-config --cflags lua5.3`
 
 GROFF=	groff -Tascii
 CRYPTOLIBDIR=	# -L/usr/local/lib
 CRYPTOLIBS=	$(CRYPTOLIBDIR) -lcrypto -lssl
 
-LIBS=	$(CRYPTOLIBS) $(EXTRALIBS)
+LIBS=	$(CRYPTOLIBS) $(EXTRALIBS) `pkg-config --libs lua5.3`
 
 FILES=	bozohttpd.c auth-bozo.c cgi-bozo.c content-bozo.c daemon-bozo.c \
 	dir-index-bozo.c lua-bozo.c ssl-bozo.c tilde-luzah-bozo.c main.c
