--- origsrc/bozohttpd-20260503/Makefile.boot	2026-05-04 06:54:07.000000000 +0900
+++ src/bozohttpd-20260503/Makefile.boot	2026-05-09 23:28:35.206103400 +0900
@@ -3,17 +3,17 @@
 # very simple makefile to compile bozohttpd, should work with every make.
 # see Makefile for a list of compile options that may be placed in CFLAGS.
 
-CC=	cc
-OPT=	-O
+CC?=	cc
 LARGE_CFLAGS=	-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
-LOCAL_CFLAGS=	-DNO_LUA_SUPPORT -DNO_BLOCKLIST_SUPPORT -D_GNU_SOURCE -D_DEFAULT_SOURCE
-CFLAGS=	$(OPT) $(LARGE_CFLAGS) $(LOCAL_CFLAGS)
+LOCAL_CFLAGS=	-DNO_BLOCKLIST_SUPPORT -D_GNU_SOURCE -D_DEFAULT_SOURCE
+CFLAGS ?=
+CFLAGS +=	$(LARGE_CFLAGS) $(LOCAL_CFLAGS) `pkg-config --cflags lua5.3`
 
 GROFF=	groff -Tascii
 CRYPTOLIBDIR=	# -L/usr/local/lib
 CRYPTOLIBS=	$(CRYPTOLIBDIR) -lcrypto -lssl
 
-LIBS=	$(CRYPTOLIBS) $(EXTRALIBS)
+LIBS=	$(CRYPTOLIBS) $(EXTRALIBS) `pkg-config --libs lua5.3`
 
 FILES=	bozohttpd.c auth-bozo.c cgi-bozo.c content-bozo.c daemon-bozo.c \
 	dir-index-bozo.c lua-bozo.c ssl-bozo.c tilde-luzah-bozo.c main.c
