--- origsrc/apriltag-3.1.1/CMakeLists.txt	2019-06-14 08:21:58.000000000 +0900
+++ src/apriltag-3.1.1/CMakeLists.txt	2019-06-19 04:18:01.375051200 +0900
@@ -39,7 +39,7 @@ set_source_files_properties(SOURCE ${TAG
 
 # install library
 install(TARGETS ${PROJECT_NAME} EXPORT apriltag
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
 )
 
@@ -52,7 +52,7 @@ endforeach()
 
 # export library
 install(EXPORT apriltag
-    DESTINATION share/apriltag/cmake
+    DESTINATION lib/cmake/apriltag
     FILE apriltagConfig.cmake
 )
 
