--- origsrc/apriltag-3.3.0/CMakeLists.txt	2022-09-04 01:35:06.000000000 +0900
+++ src/apriltag-3.3.0/CMakeLists.txt	2023-01-13 17:09:10.605649800 +0900
@@ -74,7 +74,11 @@ set(generated_dir "${CMAKE_CURRENT_BINAR
 set(version_config "${generated_dir}/${PROJECT_NAME}ConfigVersion.cmake")
 set(project_config "${generated_dir}/${PROJECT_NAME}Config.cmake")
 set(targets_export_name "${PROJECT_NAME}Targets")
-set(config_install_dir "share/${PROJECT_NAME}/cmake")
+if (CYGWIN)
+    set(config_install_dir "lib/cmake/${PROJECT_NAME}")
+else()
+    set(config_install_dir "share/${PROJECT_NAME}/cmake")
+endif()
 
 # Include module with fuction 'write_basic_package_version_file'
 include(CMakePackageConfigHelpers)
@@ -96,7 +100,7 @@ write_basic_package_version_file("${vers
 
 # install library
 install(TARGETS ${PROJECT_NAME} EXPORT ${targets_export_name}
-        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
         )
 
@@ -112,9 +116,7 @@ export(TARGETS apriltag
 
 
 # install pkgconfig related files
-FILE(READ apriltag.pc.in PKGC)
-STRING(REGEX REPLACE "^prefix=" "prefix=${CMAKE_INSTALL_PREFIX}" PKGC_CONF "${PKGC}" )
-FILE(WRITE ${PROJECT_BINARY_DIR}/apriltag.pc ${PKGC_CONF})
+configure_file(${PROJECT_SOURCE_DIR}/apriltag.pc.in ${PROJECT_BINARY_DIR}/apriltag.pc @ONLY)
 install(FILES "${PROJECT_BINARY_DIR}/apriltag.pc" DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig/")
 
 
@@ -165,7 +167,11 @@ target_link_libraries(apriltag_demo ${PR
 
 # opencv_demo
 # NB: contrib required for TickMeter in OpenCV 2.4. This is only required for 16.04 backwards compatibility and can be removed in the future.
-find_package(OpenCV COMPONENTS core imgproc videoio highgui contrib QUIET)
+if (CYGWIN)
+    find_package(OpenCV COMPONENTS core imgproc videoio highgui QUIET)
+else()
+    find_package(OpenCV COMPONENTS core imgproc videoio highgui contrib QUIET)
+endif()
 if(OpenCV_FOUND)
     add_executable(opencv_demo example/opencv_demo.cc)
     target_link_libraries(opencv_demo apriltag ${OpenCV_LIBRARIES})
--- origsrc/apriltag-3.3.0/apriltag.pc.in	2022-09-04 01:35:06.000000000 +0900
+++ src/apriltag-3.3.0/apriltag.pc.in	2023-01-13 17:05:08.437688400 +0900
@@ -1,10 +1,10 @@
-prefix=
+prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
 includedir=${prefix}/include
 libdir=${exec_prefix}/lib
 
 Name: apriltag
 Description: AprilTag detector library
-Version: 0.10.0
+Version: @PROJECT_VERSION@
 Cflags: -I${includedir}
 Libs: -L${libdir} -lapriltag
