--- origsrc/apriltag-3.4.0/CMakeLists.txt	2024-03-10 17:59:08.000000000 +0900
+++ src/apriltag-3.4.0/CMakeLists.txt	2024-03-15 14:30:32.581609900 +0900
@@ -42,7 +42,10 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CO
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang")
-    add_compile_options(-Wall -Wextra -Werror)
+    if (NOT CYGWIN)
+        add_compile_options(-Werror)
+    endif()
+    add_compile_options(-Wall -Wextra)
     # add_compile_options(-Wpedantic)
     add_compile_options(-Wno-shift-negative-value)
 endif()
@@ -104,7 +107,7 @@ set(generated_dir "${CMAKE_CURRENT_BINAR
 set(version_config "${generated_dir}/${PROJECT_NAME}ConfigVersion.cmake")
 set(project_config "${generated_dir}/${PROJECT_NAME}Config.cmake")
 set(targets_export_name "${PROJECT_NAME}Targets")
-set(config_install_dir "share/${PROJECT_NAME}/cmake")
+option(config_install_dir "config_install_dir" "share/${PROJECT_NAME}/cmake")
 
 # Include module with fuction 'write_basic_package_version_file'
 include(CMakePackageConfigHelpers)
