--- origsrc/apriltag-3.1.4/CMakeLists.txt	2021-03-04 06:23:26.000000000 +0900
+++ src/apriltag-3.1.4/CMakeLists.txt	2021-07-14 09:02:40.545810700 +0900
@@ -33,7 +33,7 @@ target_include_directories(${PROJECT_NAM
 
 # install library
 install(TARGETS ${PROJECT_NAME} EXPORT apriltagTargets
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
 )
 
@@ -45,15 +45,27 @@ foreach(HEADER ${HEADER_FILES})
 endforeach()
 
 # export library
-install(EXPORT apriltagTargets
-    FILE apriltagTargets.cmake
-    NAMESPACE apriltag::
-    DESTINATION share/apriltag/cmake
-)
-install(FILES CMake/apriltagConfig.cmake DESTINATION share/apriltag/cmake)
-export(TARGETS apriltag
-    NAMESPACE apriltag::
-    FILE apriltagTargets.cmake)
+if (CYGWIN)
+    install(EXPORT apriltagTargets
+        FILE apriltagTargets.cmake
+        NAMESPACE apriltag::
+        DESTINATION lib/cmake/apriltag
+    )
+    install(FILES CMake/apriltagConfig.cmake DESTINATION lib/cmake/apriltag)
+    export(TARGETS apriltag
+        NAMESPACE apriltag::
+        FILE apriltagTargets.cmake)
+else()
+    install(EXPORT apriltagTargets
+        FILE apriltagTargets.cmake
+        NAMESPACE apriltag::
+        DESTINATION share/apriltag/cmake
+    )
+    install(FILES CMake/apriltagConfig.cmake DESTINATION share/apriltag/cmake)
+    export(TARGETS apriltag
+        NAMESPACE apriltag::
+        FILE apriltagTargets.cmake)
+endif()
 
 FILE(READ apriltag.pc.in PKGC)
 STRING(REGEX REPLACE "^prefix=" "prefix=${CMAKE_INSTALL_PREFIX}" PKGC_CONF "${PKGC}" )
