--- origsrc/apriltag-3.4.3/CMakeLists.txt	2025-02-16 22:50:22.000000000 +0900
+++ src/apriltag-3.4.3/CMakeLists.txt	2025-03-12 11:37:07.236425300 +0900
@@ -46,7 +46,7 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CO
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang")
-    add_compile_options(-Wall -Wextra -Werror)
+    add_compile_options(-Wall -Wextra)
     add_compile_options(-Wpedantic)
     if(CMAKE_C_COMPILER_ID MATCHES "Clang")
         add_compile_options(
@@ -115,7 +115,7 @@ set(generated_dir "${CMAKE_CURRENT_BINAR
 set(version_config "${generated_dir}/${PROJECT_NAME}ConfigVersion.cmake")
 set(project_config "${generated_dir}/${PROJECT_NAME}Config.cmake")
 set(targets_export_name "${PROJECT_NAME}Targets")
-set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/${PROJECT_NAME}/cmake")
+set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
 
 # Include module with fuction 'write_basic_package_version_file'
 include(CMakePackageConfigHelpers)
