--- origsrc/libelysium-0.99.9/configure.ac	2003-06-08 05:52:09.000000000 +0900
+++ src/libelysium-0.99.9/configure.ac	2014-11-02 19:13:54.612697000 +0900
@@ -1,10 +1,10 @@
 AC_PREREQ(2.53)
 
-AC_INIT(libelysium, 0.99.9)
+AC_INIT([libelysium], [0.99.9])
 AC_CONFIG_SRCDIR(config.h.in)
 AC_COPYRIGHT([Copyright 2002-2003 Elysium GNU/Linux])
 
-AM_INIT_AUTOMAKE(AC_PACKAGE_NAME, AC_PACKAGE_VERSION)
+AM_INIT_AUTOMAKE
 AM_CONFIG_HEADER(config.h)
 
 AC_ISC_POSIX
--- origsrc/libelysium-0.99.9/src/Makefile.am	2003-06-03 15:13:31.000000000 +0900
+++ src/libelysium-0.99.9/src/Makefile.am	2014-11-02 19:13:26.314845500 +0900
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-INCLUDES = \
+AM_CPPFLAGS = \
 	-DG_DISABLE_DEPRECATED \
 	-DEPIXMAPDIR=\"@datadir@/pixmaps/\" \
 	$(GLIB_CFLAGS)
@@ -20,10 +20,11 @@ libelysium_HEADERS = \
 
 libelysiumdir = $(includedir)/libelysium-2.0/libelysium
 
-libelysium_2_0_la_LDFLAGS = -version-info 0:0:0
+libelysium_2_0_la_LDFLAGS = -no-undefined -version-info 0:0:0
 libelysium_2_0_la_LIBADD = $(GLIB_LIBS)
 
-noinst_PROGRAMS = test-datetok test-cookies
+TESTS = test-datetok test-cookies
+check_PROGRAMS = test-datetok test-cookies
 
 test_datetok_SOURCES = \
 	test-datetok.c
--- origsrc/libelysium-0.99.9/src/test-cookies.c	2003-06-06 11:19:00.000000000 +0900
+++ src/libelysium-0.99.9/src/test-cookies.c	2014-11-02 19:16:22.973048600 +0900
@@ -19,8 +19,6 @@ gint main (gint argc, gchar * argv[]) {
   gchar * header;
   gint i;
 
-  g_type_init ();
-
   cookies = e_cookies_new ();
 
   for (i = 0; i < G_N_ELEMENTS (cookies_list); i++) {
--- origsrc/libelysium-0.99.9/src/test-datetok.c	2003-06-03 15:04:23.000000000 +0900
+++ src/libelysium-0.99.9/src/test-datetok.c	2014-11-02 19:15:41.691798600 +0900
@@ -1,5 +1,7 @@
 #include "e-date-rfc.h"
 
+#include <stdio.h>
+
 typedef struct _dtest_data {
   const char * type_string;
   const char * date_string;
