--- origsrc/libelysium-0.99.9/configure.ac	2003-06-08 05:52:09.000000000 +0900
+++ src/libelysium-0.99.9/configure.ac	2010-12-30 15:56:02.705451100 +0900
@@ -10,7 +10,7 @@ AM_CONFIG_HEADER(config.h)
 AC_ISC_POSIX
 AC_PROG_CC
 AC_STDC_HEADERS
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_PROG_INTLTOOL
 
 GETTEXT_PACKAGE="${PACKAGE}"
--- origsrc/libelysium-0.99.9/src/Makefile.am	2003-06-03 15:13:31.000000000 +0900
+++ src/libelysium-0.99.9/src/Makefile.am	2010-12-30 16:19:59.066467500 +0900
@@ -20,10 +20,11 @@ libelysium_HEADERS = \
 
 libelysiumdir = $(includedir)/libelysium-2.0/libelysium
 
-libelysium_2_0_la_LDFLAGS = -version-info 0:0:0
+libelysium_2_0_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libelysium_2_0_la_LIBADD = $(GLIB_LIBS)
 
-noinst_PROGRAMS = test-datetok test-cookies
+TESTS = test-datetok test-cookies
+check_PROGRAMS = test-datetok test-cookies
 
 test_datetok_SOURCES = \
 	test-datetok.c
