--- origsrc/signon-plugin-oauth2-VERSION_0.25/src/signon-oauth2plugin.pc	2020-11-11 04:43:33.000000000 +0900
+++ src/signon-plugin-oauth2-VERSION_0.25/src/signon-oauth2plugin.pc	2020-11-29 23:10:56.365917500 +0900
@@ -4,6 +4,6 @@ libdir=${prefix}/lib/
 includedir=${prefix}/include
 Name: signon-oauth2plugin
 Description: Signon OAuth 2.0 plugin
-Version: 0.0.1
+Version: 0.25
 Requires: signon-plugins
 Libs.private: -L/usr/lib  -lQtCore
