--- origsrc/liblangtag-0.3/liblangtag/Makefile.am	2012-03-15 20:07:20.000000000 +0900
+++ src/liblangtag-0.3/liblangtag/Makefile.am	2012-09-24 23:07:31.467713500 +0900
@@ -1,7 +1,7 @@
 ##
 # Global definitions
 NULL =
-SUBDIRS =
+SUBDIRS = .
 INCLUDES =							\
 	-I$(top_srcdir)/liblangtag/				\
 	-I$(top_srcdir)						\
@@ -158,6 +158,6 @@ liblangtag_la_CFLAGS =				\
 liblangtag_la_LDFLAGS =						\
 	$(LDFLAGS)						\
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)	\
-	$(NULL)
+	-no-undefined $(NULL)
 
 -include $(top_srcdir)/git.mk
--- origsrc/liblangtag-0.3/liblangtag/extensions/Makefile.am	2012-03-05 18:10:19.000000000 +0900
+++ src/liblangtag-0.3/liblangtag/extensions/Makefile.am	2012-09-24 23:10:16.467713500 +0900
@@ -49,7 +49,8 @@ liblangtag_ext_ldml_t_la_CFLAGS =		\
 liblangtag_ext_ldml_t_la_LDFLAGS =		\
 	-avoid-version				\
 	-module					\
-	$(NULL)
+	-no-undefined $(NULL)
+liblangtag_ext_ldml_t_la_LIBADD = ../liblangtag.la
 #
 liblangtag_ext_ldml_u_la_SOURCES =	\
 	lt-ext-ldml-u.c			\
@@ -60,6 +61,7 @@ liblangtag_ext_ldml_u_la_CFLAGS =		\
 liblangtag_ext_ldml_u_la_LDFLAGS =		\
 	-avoid-version				\
 	-module					\
-	$(NULL)
+	-no-undefined $(NULL)
+liblangtag_ext_ldml_u_la_LIBADD = ../liblangtag.la
 
 -include $(top_srcdir)/git.mk
--- origsrc/liblangtag-0.3/liblangtag-gobject/Makefile.am	2012-02-28 19:26:16.000000000 +0900
+++ src/liblangtag-0.3/liblangtag-gobject/Makefile.am	2012-09-24 22:18:02.256927100 +0900
@@ -137,7 +137,7 @@ liblangtag_gobject_la_CFLAGS =				\
 liblangtag_gobject_la_LDFLAGS =						\
 	$(LDFLAGS)							\
 	-version-info $(LT_G_CURRENT):$(LT_G_REVISION):$(LT_G_AGE)	\
-	$(NULL)
+	-no-undefined $(NULL)
 #
 if HAVE_INTROSPECTION
 lt_headers = $(liblangtag_gobject_private_headers:.gir.h=.h)
