--- origsrc/liblangtag-0.5.1/extensions/Makefile.am	2013-04-22 17:06:56.000000000 +0900
+++ src/liblangtag-0.5.1/extensions/Makefile.am	2013-05-25 13:22:16.353695700 +0900
@@ -60,6 +60,7 @@ liblangtag_ext_ldml_t_la_CFLAGS =		\
 	$(module_t_cflags)			\
 	$(NULL)
 liblangtag_ext_ldml_t_la_LDFLAGS =		\
+	-no-undefined				\
 	-avoid-version				\
 	-module					\
 	$(NULL)
@@ -71,6 +72,7 @@ liblangtag_ext_ldml_u_la_CFLAGS =		\
 	$(module_u_cflags)			\
 	$(NULL)
 liblangtag_ext_ldml_u_la_LDFLAGS =		\
+	-no-undefined				\
 	-avoid-version				\
 	-module					\
 	$(NULL)
--- origsrc/liblangtag-0.5.1/liblangtag/Makefile.am	2013-04-22 17:06:56.000000000 +0900
+++ src/liblangtag-0.5.1/liblangtag/Makefile.am	2013-05-25 13:22:35.369320700 +0900
@@ -1,7 +1,7 @@
 ##
 # Global definitions
 NULL =
-SUBDIRS =
+SUBDIRS = .
 INCLUDES =							\
 	-I$(top_srcdir)/liblangtag/				\
 	-I$(top_srcdir)						\
@@ -209,6 +209,7 @@ liblangtag_la_DEPENDENCIES =					\
 	$(included_modules)					\
 	$(NULL)
 liblangtag_la_LDFLAGS =						\
+	-no-undefined						\
 	$(LDFLAGS)						\
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)	\
 	$(NULL)
--- origsrc/liblangtag-0.5.1/liblangtag/lt-trie.c	2013-04-30 11:37:30.000000000 +0900
+++ src/liblangtag-0.5.1/liblangtag/lt-trie.c	2013-05-25 14:28:19.877958700 +0900
@@ -33,11 +33,11 @@ struct _lt_trie_t {
 	lt_iter_tmpl_t  parent;
 	lt_trie_node_t *root;
 };
-typedef struct _lt_trie_iter_t {
+struct _lt_trie_iter_t {
 	lt_iter_t    parent;
 	lt_list_t   *stack;
 	lt_string_t *pos_str;
-} lt_trie_iter_t;
+};
 
 /*< private >*/
 static lt_trie_node_t *
--- origsrc/liblangtag-0.5.1/liblangtag-gobject/Makefile.am	2013-04-23 12:23:38.000000000 +0900
+++ src/liblangtag-0.5.1/liblangtag-gobject/Makefile.am	2013-05-25 14:29:52.201200900 +0900
@@ -147,6 +147,7 @@ liblangtag_gobject_la_CFLAGS =				\
 	-DG_LOG_DOMAIN="\"LangTag-GObject\""		\
 	$(NULL)
 liblangtag_gobject_la_LDFLAGS =						\
+	-no-undefined							\
 	$(LDFLAGS)							\
 	-version-info $(LT_G_CURRENT):$(LT_G_REVISION):$(LT_G_AGE)	\
 	$(NULL)
--- origsrc/liblangtag-0.5.1/tests/Makefile.am	2013-04-22 17:06:57.000000000 +0900
+++ src/liblangtag-0.5.1/tests/Makefile.am	2013-05-25 14:47:22.070341500 +0900
@@ -5,7 +5,7 @@ SUBDIRS =
 INCLUDES =								\
 	-I$(top_srcdir)/liblangtag					\
 	-I$(top_srcdir)							\
-	-DTEST_DATADIR="\"$(abs_top_builddir)/data\""			\
+	-DTEST_DATADIR="\"$(abs_top_srcdir)/data\""			\
 	-DTEST_MODDIR="\"$(abs_top_builddir)/extensions/.libs\""	\
 	-D__LANGTAG_COMPILATION						\
 	$(CHECK_CFLAGS)							\
