--- origsrc/liblangtag-0.5.2/extensions/Makefile.am	2013-08-27 19:48:53.000000000 +0900
+++ src/liblangtag-0.5.2/extensions/Makefile.am	2013-09-25 10:43:01.902390600 +0900
@@ -60,6 +60,7 @@ liblangtag_ext_ldml_t_la_CFLAGS =		\
 	$(module_t_cflags)			\
 	$(NULL)
 liblangtag_ext_ldml_t_la_LDFLAGS =		\
+	-no-undefined				\
 	-avoid-version				\
 	-module					\
 	$(NULL)
@@ -71,6 +72,7 @@ liblangtag_ext_ldml_u_la_CFLAGS =		\
 	$(module_u_cflags)			\
 	$(NULL)
 liblangtag_ext_ldml_u_la_LDFLAGS =		\
+	-no-undefined				\
 	-avoid-version				\
 	-module					\
 	$(NULL)
--- origsrc/liblangtag-0.5.2/liblangtag/Makefile.am	2013-08-27 19:49:17.000000000 +0900
+++ src/liblangtag-0.5.2/liblangtag/Makefile.am	2013-09-25 10:43:42.677781200 +0900
@@ -209,6 +209,7 @@ liblangtag_la_DEPENDENCIES =					\
 	$(included_modules)					\
 	$(NULL)
 liblangtag_la_LDFLAGS =						\
+	-no-undefined						\
 	$(LDFLAGS)						\
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)	\
 	$(NULL)
--- origsrc/liblangtag-0.5.2/liblangtag-gobject/Makefile.am	2013-08-27 19:56:13.000000000 +0900
+++ src/liblangtag-0.5.2/liblangtag-gobject/Makefile.am	2013-09-25 10:44:18.335984400 +0900
@@ -147,6 +147,7 @@ liblangtag_gobject_la_CFLAGS =				\
 	-DG_LOG_DOMAIN="\"LangTag-GObject\""		\
 	$(NULL)
 liblangtag_gobject_la_LDFLAGS =						\
+	-no-undefined							\
 	$(LDFLAGS)							\
 	-version-info $(LT_G_CURRENT):$(LT_G_REVISION):$(LT_G_AGE)	\
 	$(NULL)
--- origsrc/liblangtag-0.5.2/tests/Makefile.am	2013-08-27 19:49:29.000000000 +0900
+++ src/liblangtag-0.5.2/tests/Makefile.am	2013-09-25 10:44:46.056687500 +0900
@@ -5,7 +5,7 @@ SUBDIRS =
 AM_CPPFLAGS =								\
 	-I$(top_srcdir)/liblangtag					\
 	-I$(top_srcdir)							\
-	-DTEST_DATADIR="\"$(abs_top_builddir)/data\""			\
+	-DTEST_DATADIR="\"$(abs_top_srcdir)/data\""			\
 	-DTEST_MODDIR="\"$(abs_top_builddir)/extensions/.libs\""	\
 	-D__LANGTAG_COMPILATION						\
 	$(CHECK_CFLAGS)							\
