--- origsrc/qpid-cpp-1.39.0/CMakeLists.txt	2018-10-22 07:43:10.000000000 +0900
+++ src/qpid-cpp-1.39.0/CMakeLists.txt	2020-10-08 16:17:02.132769800 +0900
@@ -96,7 +96,9 @@ endif (WIN32)
 set_absolute_install_path (QPIDC_CONF_FILE ${QPID_INSTALL_CONFDIR}/qpidc.conf)
 set_absolute_install_path (QPIDD_CONF_FILE ${QPID_INSTALL_CONFDIR}/qpidd.conf)
 
-install(FILES  LICENSE.txt NOTICE.txt DESTINATION ${QPID_INSTALL_DOCDIR})
+if (NOT CYGWIN)
+  install(FILES  LICENSE.txt NOTICE.txt DESTINATION ${QPID_INSTALL_DOCDIR})
+endif()
 install(FILES  include/qmf/qmf2.i
         DESTINATION ${QPID_INSTALL_INCLUDEDIR}/qmf)
 
