--- origsrc/qpid-cpp-1.38.0/CMakeLists.txt	2018-03-22 08:16:10.000000000 +0900
+++ src/qpid-cpp-1.38.0/CMakeLists.txt	2018-06-19 02:36:52.552790200 +0900
@@ -96,7 +96,9 @@ endif (WIN32)
 set_absolute_install_path (QPIDC_CONF_FILE ${QPID_INSTALL_CONFDIR}/qpidc.conf)
 set_absolute_install_path (QPIDD_CONF_FILE ${QPID_INSTALL_CONFDIR}/qpidd.conf)
 
-install(FILES  LICENSE.txt NOTICE.txt DESTINATION ${QPID_INSTALL_DOCDIR})
+if (NOT CYGWIN)
+  install(FILES  LICENSE.txt NOTICE.txt DESTINATION ${QPID_INSTALL_DOCDIR})
+endif()
 install(FILES  include/qmf/qmf2.i
         DESTINATION ${QPID_INSTALL_INCLUDEDIR}/qmf)
 
--- origsrc/qpid-cpp-1.38.0/src/CMakeLists.txt	2018-03-22 08:16:10.000000000 +0900
+++ src/qpid-cpp-1.38.0/src/CMakeLists.txt	2018-06-19 00:36:54.818835400 +0900
@@ -352,7 +352,7 @@ if (NOT CMAKE_SYSTEM_NAME STREQUAL Windo
   if (HAVE_POLL)
   set(poller_default poll)
   endif (HAVE_POLL)
-  if (HAVE_EPOLL)
+  if (HAVE_EPOLL AND NOT CYGWIN)
   set(poller_default epoll)
   endif (HAVE_EPOLL)
   set(QPID_POLLER ${poller_default} CACHE STRING "Poller implementation (poll/epoll)")
