--- origsrc/rudeshipping/Makefile.am	2024-11-07 12:42:00.000000000 +0900
+++ src/rudeshipping/Makefile.am	2024-11-07 12:42:02.674659200 +0900
@@ -89,7 +89,7 @@ librudeshipping_la_SOURCES =	\
 	src/calculators/UPSCalculator.cpp
 
 
-librudeshipping_la_LDFLAGS = -version-info 3:1:1
+librudeshipping_la_LDFLAGS = -no-undefined -version-info 3:1:1
 librudeshipping_la_LIBADD = -lrudesocket -lrudeconfig
 
 
@@ -100,3 +100,5 @@ pkgincludedir = $(includedir)/rude
 AUTOMAKE_OPTIONS = gnits dist-bzip2 dist-shar dist-zip
 
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/rudeshipping/configure.ac	2024-11-07 12:42:00.000000000 +0900
+++ src/rudeshipping/configure.ac	2024-11-07 12:42:02.674659200 +0900
@@ -49,6 +49,6 @@ AC_TYPE_SIZE_T
 
 # Checks for library functions.
 AC_CHECK_FUNCS([strstr])
-AC_OUTPUT(Makefile )
+AC_OUTPUT(rudeshipping.pc Makefile)
 
 
--- origsrc/rudeshipping/rudeshipping.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/rudeshipping/rudeshipping.pc.in	2024-11-07 12:42:02.674659200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: rudeshipping
+Description: C++ interface for various shipping services like UPS
+Version: @VERSION@
+Libs: -L${libdir} -lrudeshipping
+Cflags: -I${includedir}
--- origsrc/rudeshipping/src/calculators/UPSCalculator.cpp	2024-11-07 12:42:00.000000000 +0900
+++ src/rudeshipping/src/calculators/UPSCalculator.cpp	2024-11-07 12:42:02.674659200 +0900
@@ -22,6 +22,7 @@
 //------------------------------------------------------------------------
 
 
+#include <cstring>
 
 #include "UPSCalculator.h"
 
--- origsrc/rudeshipping/src/calculators/USPSDomesticCalculator.cpp	2024-11-07 12:42:00.000000000 +0900
+++ src/rudeshipping/src/calculators/USPSDomesticCalculator.cpp	2024-11-07 12:42:02.674659200 +0900
@@ -21,7 +21,7 @@
 // 02111-1307, USA.
 //------------------------------------------------------------------------
 
-
+#include <cstring>
 
 #include "USPSDomesticCalculator.h"
 
