--- origsrc/libmonetra-7.13.0/lib/Makefile.am	2014-01-20 03:30:49.000000000 +0900
+++ src/libmonetra-7.13.0/lib/Makefile.am	2015-01-10 08:31:07.357067500 +0900
@@ -1,9 +1,9 @@
 ACLOCAL_AMFLAGS = -I m4
 
 lib_LTLIBRARIES = libmonetra.la
-libmonetra_la_SOURCES = 
+libmonetra_la_SOURCES =
 libmonetra_la_LIBADD = ../src/libmonetra.la
-libmonetra_la_LDFLAGS = -version-info 7:13:0
+libmonetra_la_LDFLAGS = -no-undefined -version-info 7:13:0
 
 include_HEADERS = ../src/monetra.h ../src/libmonetra_compat.h ../src/libmonetra_defines.h ../src/libmonetra_legacy.h
 
