--- origsrc/libmonetra-7.0.5/Makefile.am	2009-09-26 03:23:01.000000000 +0900
+++ src/libmonetra-7.0.5/Makefile.am	2010-12-01 10:57:46.432657800 +0900
@@ -9,7 +9,7 @@ libmonetra_la_SOURCES = libmonetra_atoin
 			proxy.c http.c md5.c base64.c
 
 libmonetra_la_LIBADD = @LIBCRYPTO@
-libmonetra_la_LDFLAGS = -version-info 7:5:0
+libmonetra_la_LDFLAGS = -version-info 7:5:0 -no-undefined
 
 include_HEADERS = monetra.h libmonetra_compat.h libmonetra_config.h libmonetra_defines.h libmonetra_legacy.h
 
--- origsrc/libmonetra-7.0.5/configure.in	2009-09-26 03:34:39.000000000 +0900
+++ src/libmonetra-7.0.5/configure.in	2010-12-01 10:57:28.977558600 +0900
@@ -23,7 +23,7 @@ AC_PROG_CC
 AC_PROG_CC_STDC
 AM_PROG_CC_C_O
 AC_PROG_INSTALL
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 AC_C_CONST
 
