--- origsrc/guacamole-server-1.5.5/README-unit-testing.md	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/README-unit-testing.md	2024-10-10 00:17:10.640951000 +0900
@@ -46,7 +46,7 @@ modified to contain a sections like the
         ...all source files...
 
     test_myproj_CFLAGS =        \
-        -Werror -Wall -pedantic \
+         -Wall -pedantic \
         ...other flags...
 
     test_myproj_LDADD = \
--- origsrc/guacamole-server-1.5.5/configure.ac	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/configure.ac	2024-10-10 00:17:10.640951000 +0900
@@ -20,7 +20,7 @@
 AC_PREREQ([2.61])
 AC_INIT([guacamole-server], [1.5.5])
 AC_CONFIG_AUX_DIR([build-aux])
-AM_INIT_AUTOMAKE([-Wall -Werror foreign subdir-objects])
+AM_INIT_AUTOMAKE([-Wall  foreign subdir-objects])
 AM_SILENT_RULES([yes])
 
 LT_PREREQ([2.2])
@@ -670,7 +670,7 @@ if test "x$with_rdp" != "xno"
 then
     have_freerdp2=yes
     PKG_CHECK_MODULES([RDP], [freerdp2 freerdp-client2 winpr2],
-                      [CPPFLAGS="${RDP_CFLAGS} -Werror $CPPFLAGS"]
+                      [CPPFLAGS="${RDP_CFLAGS}  $CPPFLAGS"]
                       [AS_IF([test "x${FREERDP2_PLUGIN_DIR}" = "x"],
                              [FREERDP2_PLUGIN_DIR="`$PKG_CONFIG --variable=libdir freerdp2`/freerdp2"])],
                       [AC_MSG_WARN([
--- origsrc/guacamole-server-1.5.5/src/common/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/common/Makefile.am	2024-10-10 00:17:10.649427600 +0900
@@ -63,7 +63,7 @@ libguac_common_la_SOURCES = \
     surface.c
 
 libguac_common_la_CFLAGS =  \
-    -Werror -Wall -pedantic \
+     -Wall -pedantic \
     @LIBGUAC_INCLUDE@
 
 libguac_common_la_LIBADD = \
--- origsrc/guacamole-server-1.5.5/src/common/tests/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/common/tests/Makefile.am	2024-10-10 00:17:10.656289200 +0900
@@ -49,7 +49,7 @@ test_common_SOURCES =          \
     string/split.c
 
 test_common_CFLAGS =        \
-    -Werror -Wall -pedantic \
+     -Wall -pedantic \
     @COMMON_INCLUDE@        \
     @LIBGUAC_INCLUDE@
 
--- origsrc/guacamole-server-1.5.5/src/common-ssh/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/common-ssh/Makefile.am	2024-10-10 00:17:10.695685100 +0900
@@ -44,7 +44,7 @@ noinst_HEADERS =            \
     common-ssh/user.h
 
 libguac_common_ssh_la_CFLAGS = \
-    -Werror -Wall -pedantic    \
+     -Wall -pedantic    \
     @COMMON_INCLUDE@           \
     @LIBGUAC_INCLUDE@
 
--- origsrc/guacamole-server-1.5.5/src/common-ssh/tests/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/common-ssh/tests/Makefile.am	2024-10-10 00:17:10.695685100 +0900
@@ -37,7 +37,7 @@ test_common_ssh_SOURCES = \
     sftp/normalize_path.c
 
 test_common_ssh_CFLAGS =    \
-    -Werror -Wall -pedantic \
+     -Wall -pedantic \
     @COMMON_INCLUDE@        \
     @COMMON_SSH_INCLUDE@    \
     @LIBGUAC_INCLUDE@
--- origsrc/guacamole-server-1.5.5/src/guacd/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/guacd/Makefile.am	2024-10-10 00:51:45.600450700 +0900
@@ -54,7 +54,7 @@ guacd_SOURCES =  \
     proc-map.c
 
 guacd_CFLAGS =              \
-    -Werror -Wall -pedantic \
+     -Wall -pedantic \
     @COMMON_INCLUDE@        \
     @LIBGUAC_INCLUDE@
 
@@ -80,8 +80,9 @@ initdir = @init_dir@
 init_SCRIPTS = init.d/guacd
 
 init.d/guacd: init.d/guacd.in
-	sed -e 's,[@]sbindir[@],$(sbindir),g' < init.d/guacd.in > init.d/guacd
-	chmod +x init.d/guacd
+	mkdir -p init.d
+	sed -e 's,[@]sbindir[@],$(sbindir),g' < $< > $@
+	chmod +x $@
 endif
 
 # Systemd service
--- origsrc/guacamole-server-1.5.5/src/guacenc/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/guacenc/Makefile.am	2024-10-10 00:17:10.692864800 +0900
@@ -88,7 +88,7 @@ noinst_HEADERS  += webp.h
 endif
 
 guacenc_CFLAGS =            \
-    -Werror -Wall           \
+     -Wall           \
     @AVCODEC_CFLAGS@        \
     @AVFORMAT_CFLAGS@       \
     @AVUTIL_CFLAGS@         \
--- origsrc/guacamole-server-1.5.5/src/guaclog/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/guaclog/Makefile.am	2024-10-10 00:17:10.692864800 +0900
@@ -48,7 +48,7 @@ guaclog_SOURCES =     \
     state.c
 
 guaclog_CFLAGS =      \
-    -Werror -Wall     \
+     -Wall     \
     @LIBGUAC_INCLUDE@
 
 guaclog_LDADD =     \
--- origsrc/guacamole-server-1.5.5/src/libguac/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/libguac/Makefile.am	2024-10-10 00:17:10.672505600 +0900
@@ -157,7 +157,7 @@ libguacinc_HEADERS += guacamole/socket-w
 endif
 
 libguac_la_CFLAGS = \
-    -Werror -Wall -pedantic
+     -Wall -pedantic
 
 libguac_la_LDFLAGS =     \
     -version-info 24:0:0 \
--- origsrc/guacamole-server-1.5.5/src/libguac/guacamole/plugin-constants.h	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/libguac/guacamole/plugin-constants.h	2024-10-10 09:53:11.218035700 +0900
@@ -29,12 +29,20 @@
 /**
  * String prefix which begins the library filename of all client plugins.
  */
+#if defined(__CYGWIN__)
+#define GUAC_PROTOCOL_LIBRARY_PREFIX "cygguac-client-"
+#else
 #define GUAC_PROTOCOL_LIBRARY_PREFIX "libguac-client-"
+#endif
 
 /**
  * String suffix which ends the library filename of all client plugins.
  */
+#if defined(__CYGWIN__)
+#define GUAC_PROTOCOL_LIBRARY_SUFFIX ".dll"
+#else
 #define GUAC_PROTOCOL_LIBRARY_SUFFIX ".so"
+#endif
 
 /**
  * The maximum number of characters (COUNTING NULL TERMINATOR) to allow
--- origsrc/guacamole-server-1.5.5/src/libguac/tests/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/libguac/tests/Makefile.am	2024-10-10 00:17:10.676497600 +0900
@@ -69,7 +69,7 @@ test_libguac_SOURCES =               \
     unicode/write.c
 
 test_libguac_CFLAGS =       \
-    -Werror -Wall -pedantic \
+     -Wall -pedantic \
     @LIBGUAC_INCLUDE@
 
 test_libguac_LDADD = \
--- origsrc/guacamole-server-1.5.5/src/protocols/kubernetes/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/protocols/kubernetes/Makefile.am	2024-10-10 09:54:42.290611100 +0900
@@ -56,7 +56,7 @@ noinst_HEADERS = \
     user.h
 
 libguac_client_kubernetes_la_CFLAGS = \
-    -Werror -Wall -Iinclude           \
+     -Wall -Iinclude           \
     @LIBGUAC_INCLUDE@                 \
     @TERMINAL_INCLUDE@
 
@@ -65,8 +65,8 @@ libguac_client_kubernetes_la_LIBADD = \
     @LIBGUAC_LTLIB@                   \
     @TERMINAL_LTLIB@
 
-libguac_client_kubernetes_la_LDFLAGS = \
-    -version-info 0:0:0                \
+libguac_client_kubernetes_la_LDFLAGS = -no-undefined \
+    -avoid-version -module                \
     @PTHREAD_LIBS@                     \
     @SSL_LIBS@                         \
     @WEBSOCKETS_LIBS@
--- origsrc/guacamole-server-1.5.5/src/protocols/kubernetes/tests/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/protocols/kubernetes/tests/Makefile.am	2024-10-10 00:17:10.687882900 +0900
@@ -38,7 +38,7 @@ test_kubernetes_SOURCES = \
     url/escape.c
 
 test_kubernetes_CFLAGS =                \
-    -Werror -Wall -pedantic             \
+     -Wall -pedantic             \
     @LIBGUAC_CLIENT_KUBERNETES_INCLUDE@ \
     @LIBGUAC_INCLUDE@
 
--- origsrc/guacamole-server-1.5.5/src/protocols/rdp/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/protocols/rdp/Makefile.am	2024-10-10 09:54:40.366109300 +0900
@@ -132,14 +132,14 @@ noinst_HEADERS =
     user.h
 
 libguac_client_rdp_la_CFLAGS = \
-    -Werror -Wall -Iinclude    \
+     -Wall -Iinclude    \
     @COMMON_INCLUDE@           \
     @COMMON_SSH_INCLUDE@       \
     @LIBGUAC_INCLUDE@          \
     @RDP_CFLAGS@
 
-libguac_client_rdp_la_LDFLAGS = \
-    -version-info 0:0:0         \
+libguac_client_rdp_la_LDFLAGS = -no-undefined \
+    -avoid-version -module         \
     @CAIRO_LIBS@                \
     @PTHREAD_LIBS@              \
     @RDP_LIBS@
@@ -166,11 +166,11 @@ libguac_common_svc_client_la_SOURCES =
     plugins/guac-common-svc/guac-common-svc.c
 
 libguac_common_svc_client_la_CFLAGS = \
-    -Werror -Wall -Iinclude           \
+     -Wall -Iinclude           \
     @LIBGUAC_INCLUDE@                 \
     @RDP_CFLAGS@
 
-libguac_common_svc_client_la_LDFLAGS = \
+libguac_common_svc_client_la_LDFLAGS = -no-undefined \
     -module -avoid-version -shared     \
     @RDP_LIBS@
 
@@ -188,13 +188,13 @@ libguacai_client_la_SOURCES =
     plugins/ptr-string.c
 
 libguacai_client_la_CFLAGS = \
-    -Werror -Wall -Iinclude  \
+     -Wall -Iinclude  \
     @COMMON_INCLUDE@         \
     @COMMON_SSH_INCLUDE@     \
     @LIBGUAC_INCLUDE@        \
     @RDP_CFLAGS@
 
-libguacai_client_la_LDFLAGS =      \
+libguacai_client_la_LDFLAGS = -no-undefined \
     -module -avoid-version -shared \
     @PTHREAD_LIBS@                 \
     @RDP_LIBS@
--- origsrc/guacamole-server-1.5.5/src/protocols/rdp/tests/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/protocols/rdp/tests/Makefile.am	2024-10-10 00:17:10.681872000 +0900
@@ -38,7 +38,7 @@ test_rdp_SOURCES =      \
     fs/normalize_path.c
 
 test_rdp_CFLAGS =                \
-    -Werror -Wall -pedantic      \
+     -Wall -pedantic      \
     @LIBGUAC_CLIENT_RDP_INCLUDE@ \
     @LIBGUAC_INCLUDE@
 
--- origsrc/guacamole-server-1.5.5/src/protocols/ssh/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/protocols/ssh/Makefile.am	2024-10-10 09:54:40.365105900 +0900
@@ -59,7 +59,7 @@ noinst_HEADERS += ssh_agent.h
 endif
 
 libguac_client_ssh_la_CFLAGS = \
-    -Werror -Wall -Iinclude    \
+     -Wall -Iinclude    \
     @COMMON_SSH_INCLUDE@       \
     @LIBGUAC_INCLUDE@          \
     @TERMINAL_INCLUDE@
@@ -70,8 +70,8 @@ libguac_client_ssh_la_LIBADD = \
     @LIBGUAC_LTLIB@            \
     @TERMINAL_LTLIB@
 
-libguac_client_ssh_la_LDFLAGS = \
-    -version-info 0:0:0         \
+libguac_client_ssh_la_LDFLAGS = -no-undefined \
+    -avoid-version -module         \
     @PTHREAD_LIBS@              \
     @SSH_LIBS@                  \
     @SSL_LIBS@
--- origsrc/guacamole-server-1.5.5/src/protocols/telnet/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/protocols/telnet/Makefile.am	2024-10-10 09:54:40.362143800 +0900
@@ -49,7 +49,7 @@ noinst_HEADERS = \
     user.h
 
 libguac_client_telnet_la_CFLAGS = \
-    -Werror -Wall -Iinclude       \
+     -Wall -Iinclude       \
     @LIBGUAC_INCLUDE@             \
     @TERMINAL_INCLUDE@
 
@@ -58,8 +58,8 @@ libguac_client_telnet_la_LIBADD = \
     @LIBGUAC_LTLIB@               \
     @TERMINAL_LTLIB@
 
-libguac_client_telnet_la_LDFLAGS = \
-    -version-info 0:0:0            \
+libguac_client_telnet_la_LDFLAGS = -no-undefined \
+    -avoid-version -module            \
     @PTHREAD_LIBS@                 \
     @TELNET_LIBS@
 
--- origsrc/guacamole-server-1.5.5/src/protocols/vnc/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/protocols/vnc/Makefile.am	2024-10-10 09:54:40.362143800 +0900
@@ -55,18 +55,18 @@ noinst_HEADERS =      \
     vnc.h
 
 libguac_client_vnc_la_CFLAGS =        \
-    -Werror -Wall -pedantic -Iinclude \
+     -Wall -pedantic -Iinclude \
     @COMMON_INCLUDE@                  \
     @COMMON_SSH_INCLUDE@              \
     @LIBGUAC_INCLUDE@                 \
     @PULSE_INCLUDE@
 
-libguac_client_vnc_la_LDFLAGS = \
-    -version-info 0:0:0         \
+libguac_client_vnc_la_LDFLAGS = -no-undefined \
+    -avoid-version -module         \
     @CAIRO_LIBS@                \
     @VNC_LIBS@ 
 
-libguac_client_vnc_la_LIBADD = \
+libguac_client_vnc_la_LIBADD = -lgcrypt \
     @COMMON_LTLIB@             \
     @LIBGUAC_LTLIB@
 
--- origsrc/guacamole-server-1.5.5/src/pulse/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/pulse/Makefile.am	2024-10-10 00:17:10.672505600 +0900
@@ -35,7 +35,7 @@ libguac_pulse_la_SOURCES = \
     pulse.c
 
 libguac_pulse_la_CFLAGS =  \
-    -Werror -Wall -pedantic \
+     -Wall -pedantic \
     @LIBGUAC_INCLUDE@
 
 libguac_pulse_la_LIBADD = \
--- origsrc/guacamole-server-1.5.5/src/terminal/Makefile.am	2024-03-15 01:44:29.000000000 +0900
+++ src/guacamole-server-1.5.5/src/terminal/Makefile.am	2024-10-10 00:17:10.656289200 +0900
@@ -66,7 +66,7 @@ libguac_terminal_la_SOURCES =   \
     xparsecolor.c
 
 libguac_terminal_la_CFLAGS = \
-    -Werror -Wall            \
+     -Wall            \
     @COMMON_INCLUDE@         \
     @LIBGUAC_INCLUDE@        \
     @PANGO_CFLAGS@           \
