--- origsrc/guacamole-server-0.9.1/configure.ac	2014-05-24 04:02:24.000000000 +0900
+++ src/guacamole-server-0.9.1/configure.ac	2014-07-20 22:46:39.095058400 +0900
@@ -43,12 +43,10 @@ AC_CHECK_HEADERS([fcntl.h stdlib.h strin
 AC_DEFINE([_XOPEN_SOURCE], [700], [Uses X/Open and POSIX APIs])
 
 # libpng
-AC_CHECK_LIB([png], [png_write_png], [PNG_LIBS=-lpng],
-             AC_MSG_ERROR("libpng is required for writing png messages"))
+PKG_CHECK_MODULES([PNG], [libpng])
 
 # Cairo
-AC_CHECK_LIB([cairo], [cairo_create], [CAIRO_LIBS=-lcairo],
-             AC_MSG_ERROR("Cairo is required for drawing instructions"))
+PKG_CHECK_MODULES([CAIRO], [cairo])
 
 # libpthread
 AC_CHECK_LIB([pthread], [pthread_create], [PTHREAD_LIBS=-lpthread
@@ -56,16 +54,13 @@ AC_CHECK_LIB([pthread], [pthread_create]
                         [Whether libpthread was found])])
 
 # cunit
-AC_CHECK_LIB([cunit], [CU_run_test], [CUNIT_LIBS=-lcunit])
+PKG_CHECK_MODULES([CUNIT], [cunit])
 
 # WinSock
 AC_CHECK_LIB([wsock32], [main])
 
 AC_SUBST(LIBADD_DLOPEN)
-AC_SUBST(PNG_LIBS)
-AC_SUBST(CAIRO_LIBS)
 AC_SUBST(PTHREAD_LIBS)
-AC_SUBST(CUNIT_LIBS)
 
 # Library functions
 AC_CHECK_FUNCS([clock_gettime gettimeofday memmove memset select strdup nanosleep])
@@ -113,11 +108,7 @@ AC_ARG_ENABLE(ssh_agent,
 # libssl
 #
 
-have_ssl=yes
-SSL_LIBS=
-
-AC_CHECK_HEADER(openssl/ssl.h,, [have_ssl=no])
-AC_CHECK_LIB([ssl], [SSL_CTX_new], [SSL_LIBS="$SSL_LIBS -lssl"], [have_ssl=no])
+PKG_CHECK_MODULES([SSL], [openssl], [have_ssl=yes], [have_ssl=no])
 AM_CONDITIONAL([ENABLE_SSL], [test "x${have_ssl}" = "xyes"])
 
 if test "x${have_ssl}" = "xno"
@@ -131,20 +122,11 @@ else
     AC_DEFINE([ENABLE_SSL],, [Whether SSL-related support is enabled])
 fi
 
-AC_SUBST(SSL_LIBS)
-
-
 #
 # Ogg Vorbis
 #
 
-have_vorbis=yes
-VORBIS_LIBS=
-
-AC_CHECK_HEADER(vorbis/vorbisenc.h,, [have_vorbis=no])
-AC_CHECK_LIB([ogg], [ogg_stream_init], [VORBIS_LIBS="$VORBIS_LIBS -logg"], [have_vorbis=no])
-AC_CHECK_LIB([vorbis], [vorbis_block_init], [VORBIS_LIBS="$VORBIS_LIBS -lvorbis"], [have_vorbis=no])
-AC_CHECK_LIB([vorbisenc], [vorbis_encode_init], [VORBIS_LIBS="$VORBIS_LIBS -lvorbisenc"], [have_vorbis=no])
+PKG_CHECK_MODULES([VORBIS], [ogg vorbis vorbisenc], [have_vorbis=yes], [have_vorbis=no])
 AM_CONDITIONAL([ENABLE_OGG], [test "x${have_vorbis}" = "xyes"])
 
 if test "x${have_vorbis}" = "xno"
@@ -159,16 +141,11 @@ else
               [Whether support for Ogg Vorbis is enabled])
 fi
 
-AC_SUBST(VORBIS_LIBS)
-
 #
 # PulseAudio
 #
 
-have_pulse=yes
-PULSE_LIBS=
-
-AC_CHECK_LIB([pulse], [pa_context_new], [PULSE_LIBS="$PULSE_LIBS -lpulse"], [have_pulse=no])
+PKG_CHECK_MODULES([PULSE], [libpulse], [have_pulse=yes], [have_pulse=no])
 AM_CONDITIONAL([ENABLE_PULSE], [test "x${have_pulse}" = "xyes"])
 
 if test "x${have_pulse}" = "xno"
@@ -182,8 +159,6 @@ else
     AC_DEFINE([ENABLE_PULSE],, [Whether PulseAudio support is enabled])
 fi
 
-AC_SUBST(PULSE_LIBS)
-
 #
 # PANGO
 #
@@ -208,14 +183,9 @@ AM_CONDITIONAL([ENABLE_TERMINAL], [test
 # libVNCServer
 #
 
-have_libvncserver=yes
-VNC_LIBS=
-
-AC_CHECK_LIB([vncclient], [rfbInitClient], [VNC_LIBS="$VNC_LIBS -lvncclient"], [have_libvncserver=no])
+PKG_CHECK_MODULES([VNC], [libvncserver libvncclient], [have_libvncserver=yes], [have_libvncserver=no])
 AM_CONDITIONAL([ENABLE_VNC], [test "x${have_libvncserver}" = "xyes"])
 
-AC_SUBST(VNC_LIBS)
-
 #
 # Repeater support within libVNCServer
 #
@@ -281,55 +251,18 @@ freerdp_interface=unknown
 event_interface=unknown
 RDP_LIBS=
 
-# libfreerdp-cache
-AC_CHECK_LIB([freerdp-cache], [glyph_cache_register_callbacks],
-             [RDP_LIBS="$RDP_LIBS -lfreerdp-cache"],
-             [AC_MSG_WARN([
-  --------------------------------------------
-   Unable to find libfreerdp-cache
-   RDP will be disabled.
-  --------------------------------------------])
-              have_freerdp=no])
-
 # libfreerdp-core
-AC_CHECK_LIB([freerdp-core], [freerdp_new],
-             [RDP_LIBS="$RDP_LIBS -lfreerdp-core"],
-             [AC_MSG_WARN([
-  --------------------------------------------
-   Unable to find libfreerdp-core
-   RDP will be disabled.
-  --------------------------------------------])
-              have_freerdp=no])
-
 # libfreerdp-channels (1.0) / libfreerdp-client (1.1+)
-AC_CHECK_LIB([freerdp-client], [freerdp_channels_new],
-             [RDP_LIBS="$RDP_LIBS -lfreerdp-client"],
-             [AC_CHECK_LIB([freerdp-channels], [freerdp_channels_new],
-                           [RDP_LIBS="$RDP_LIBS -lfreerdp-channels"
-                            legacy_freerdp_extensions=yes],
-                           [AC_MSG_WARN([
-  --------------------------------------------
-   Unable to find libfreerdp-client / libfreerdp-channels
-   RDP will be disabled.
-  --------------------------------------------])
-                            have_freerdp=no])])
-
 # libfreerdp-utils
-AC_CHECK_LIB([freerdp-utils], [svc_plugin_init],
-             [RDP_LIBS="$RDP_LIBS -lfreerdp-utils"],
-             [AC_MSG_WARN([
-  --------------------------------------------
-   Unable to find libfreerdp-utils
-   RDP will be disabled.
-  --------------------------------------------])
-              have_freerdp=no])
-
 # libfreerdp-codec
-AC_CHECK_LIB([freerdp-codec], [freerdp_image_convert],
-             [RDP_LIBS="$RDP_LIBS -lfreerdp-codec"],
+PKG_CHECK_MODULES([RDP], [freerdp], ,[have_freerdp=no])
+
+# libfreerdp-cache
+AC_CHECK_LIB([freerdp-cache], [glyph_cache_register_callbacks],
+             [RDP_LIBS="$RDP_LIBS -lfreerdp-cache"],
              [AC_MSG_WARN([
   --------------------------------------------
-   Unable to find libfreerdp-codec
+   Unable to find libfreerdp-cache
    RDP will be disabled.
   --------------------------------------------])
               have_freerdp=no])
@@ -567,18 +500,11 @@ AC_SUBST(RDP_LIBS)
 # libssh2
 #
 
-have_libssh2=yes
-SSH_LIBS=
-
-AC_CHECK_LIB([ssh2], [libssh2_session_init_ex],
-                     [SSH_LIBS="$SSH_LIBS -lssh2"],
-                     [have_libssh2=no])
+PKG_CHECK_MODULES([SSH], [libssh2], [have_libssh2=yes], [have_libssh2=no])
 AM_CONDITIONAL([ENABLE_SSH], [test "x${have_libssh2}"  = "xyes" \
                                 -a "x${have_terminal}" = "xyes" \
                                 -a "x${have_ssl}"      = "xyes"])
 
-AC_SUBST(SSH_LIBS)
-
 #
 # Agent forwarding support within libssh2
 #
@@ -613,18 +539,10 @@ AM_CONDITIONAL([ENABLE_SSH_AGENT],
 # libtelnet
 #
 
-have_libtelnet=yes
-TELNET_LIBS=
-
-AC_CHECK_LIB([telnet], [telnet_init],
-                       [TELNET_LIBS="$TELNET_LIBS -ltelnet"],
-                       [have_libtelnet=no])
+PKG_CHECK_MODULES([TELNET], [libtelnet], [have_libtelnet=yes], [have_libtelnet=no])
 AM_CONDITIONAL([ENABLE_TELNET], [test "x${have_libtelnet}"  = "xyes" \
                                    -a "x${have_terminal}" = "xyes"])
 
-AC_SUBST(TELNET_LIBS)
-
-
 AC_CONFIG_FILES([Makefile
                  tests/Makefile
                  src/common/Makefile
--- origsrc/guacamole-server-0.9.1/src/guacd/Makefile.am	2014-05-24 04:02:25.000000000 +0900
+++ src/guacamole-server-0.9.1/src/guacd/Makefile.am	2014-07-20 23:16:37.323574000 +0900
@@ -46,8 +46,9 @@ if ENABLE_INIT
 initdir = @init_dir@
 init_SCRIPTS = init.d/guacd
 
-init.d/guacd: init.d/guacd.in
-	sed -e 's,[@]sbindir[@],$(sbindir),g' < init.d/guacd.in > init.d/guacd
+init.d/guacd: $(srcdir)/init.d/guacd.in
+	mkdir -p init.d
+	sed -e 's,[@]sbindir[@],$(sbindir),g' < $(srcdir)/init.d/guacd.in > init.d/guacd
 	chmod +x init.d/guacd
 endif
 
--- origsrc/guacamole-server-0.9.1/src/libguac/Makefile.am	2014-05-24 04:02:25.000000000 +0900
+++ src/guacamole-server-0.9.1/src/libguac/Makefile.am	2014-07-20 21:44:54.441889100 +0900
@@ -22,7 +22,7 @@
 
 AUTOMAKE_OPTIONS = foreign 
 ACLOCAL_AMFLAGS = -I m4
-AM_CFLAGS = -Werror -Wall -pedantic -Iguacamole
+AM_CFLAGS = -Werror -Wall -pedantic -I$(top_srcdir)/src/libguac/guacamole
 
 libguacincdir = $(includedir)/guacamole
 libguacinc_HEADERS =                  \
@@ -88,6 +88,6 @@ noinst_HEADERS += ogg_encoder.h
 endif
 
 lib_LTLIBRARIES = libguac.la
-libguac_la_LDFLAGS = -version-info 7:0:0 @PTHREAD_LIBS@ @CAIRO_LIBS@ @PNG_LIBS@ @VORBIS_LIBS@
+libguac_la_LDFLAGS = -no-undefined -version-info 7:0:0 @PTHREAD_LIBS@ @CAIRO_LIBS@ @PNG_LIBS@ @VORBIS_LIBS@
 libguac_la_LIBADD = @LIBADD_DLOPEN@
 
--- origsrc/guacamole-server-0.9.1/src/protocols/rdp/Makefile.am	2014-05-24 04:02:25.000000000 +0900
+++ src/guacamole-server-0.9.1/src/protocols/rdp/Makefile.am	2014-07-20 21:48:49.297357900 +0900
@@ -105,15 +105,15 @@ guacsnd_sources += compat/winpr-stream.c
 guacdr_sources  += compat/winpr-stream.c
 endif
 
-libguac_client_rdp_la_LDFLAGS = -version-info 0:0:0 @RDP_LIBS@ @PTHREAD_LIBS@ @CAIRO_LIBS@
-guacsvc_ldflags = -module -avoid-version -shared @RDP_LIBS@ @PTHREAD_LIBS@
-guacsnd_ldflags = -module -avoid-version -shared @RDP_LIBS@ @PTHREAD_LIBS@
-guacdr_ldflags = -module -avoid-version -shared @RDP_LIBS@ @PTHREAD_LIBS@
+libguac_client_rdp_la_LDFLAGS = -no-undefined -version-info 0:0:0 @RDP_LIBS@ @PTHREAD_LIBS@ @CAIRO_LIBS@
+guacsvc_ldflags = -no-undefined -module -avoid-version -shared @RDP_LIBS@ @PTHREAD_LIBS@
+guacsnd_ldflags = -no-undefined -module -avoid-version -shared @RDP_LIBS@ @PTHREAD_LIBS@
+guacdr_ldflags = -no-undefined -module -avoid-version -shared @RDP_LIBS@ @PTHREAD_LIBS@
 
 libguac_client_rdp_la_LIBADD = @LIBGUAC_LTLIB@ @COMMON_LTLIB@
 guacsvc_libadd = @LIBGUAC_LTLIB@ @COMMON_LTLIB@
 guacsnd_libadd = @LIBGUAC_LTLIB@
-guacdr_libadd = @LIBGUAC_LTLIB@
+guacdr_libadd = libguac-client-rdp.la @LIBGUAC_LTLIB@
 
 # Autogenerate keymaps
 CLEANFILES = _generated_keymaps.c
--- origsrc/guacamole-server-0.9.1/src/protocols/ssh/Makefile.am	2014-05-24 04:02:25.000000000 +0900
+++ src/guacamole-server-0.9.1/src/protocols/ssh/Makefile.am	2014-07-20 20:58:31.331433700 +0900
@@ -52,5 +52,5 @@ endif
 
 libguac_client_ssh_la_CFLAGS = -Werror -Wall -Iinclude @LIBGUAC_INCLUDE@ @TERMINAL_INCLUDE@
 libguac_client_ssh_la_LIBADD = @LIBGUAC_LTLIB@ @TERMINAL_LTLIB@
-libguac_client_ssh_la_LDFLAGS = -version-info 0:0:0 @SSH_LIBS@ @SSL_LIBS@ @PTHREAD_LIBS@
+libguac_client_ssh_la_LDFLAGS = -no-undefined -version-info 0:0:0 @SSH_LIBS@ @SSL_LIBS@ @PTHREAD_LIBS@
 
--- origsrc/guacamole-server-0.9.1/src/protocols/telnet/Makefile.am	2014-05-24 04:02:25.000000000 +0900
+++ src/guacamole-server-0.9.1/src/protocols/telnet/Makefile.am	2014-07-20 20:58:41.177136800 +0900
@@ -40,5 +40,5 @@ noinst_HEADERS =                \
 
 libguac_client_telnet_la_CFLAGS = -Werror -Wall -Iinclude @LIBGUAC_INCLUDE@ @TERMINAL_INCLUDE@
 libguac_client_telnet_la_LIBADD = @LIBGUAC_LTLIB@ @TERMINAL_LTLIB@
-libguac_client_telnet_la_LDFLAGS = -version-info 0:0:0 @TELNET_LIBS@ @PTHREAD_LIBS@
+libguac_client_telnet_la_LDFLAGS = -no-undefined -version-info 0:0:0 @TELNET_LIBS@ @PTHREAD_LIBS@
 
--- origsrc/guacamole-server-0.9.1/src/protocols/vnc/Makefile.am	2014-05-24 04:02:25.000000000 +0900
+++ src/guacamole-server-0.9.1/src/protocols/vnc/Makefile.am	2014-07-20 22:45:00.876308400 +0900
@@ -47,10 +47,10 @@ libguac_client_vnc_la_SOURCES += pulse.c
 noinst_HEADERS += pulse.h
 endif
 
-libguac_client_vnc_la_LDFLAGS = -version-info 0:0:0 \
+libguac_client_vnc_la_LDFLAGS = -no-undefined -version-info 0:0:0 \
 	@CAIRO_LIBS@                                    \
 	@PULSE_LIBS@                                    \
 	@VNC_LIBS@ 
 
-libguac_client_vnc_la_LIBADD  = @LIBGUAC_LTLIB@ @COMMON_LTLIB@
+libguac_client_vnc_la_LIBADD  = $(VNC_LIBS) @LIBGUAC_LTLIB@ @COMMON_LTLIB@
 
