--- origsrc/libmdf-1.0.29/configure.ac	2024-08-13 22:50:05.000000000 +0900
+++ src/libmdf-1.0.29/configure.ac	2024-09-24 06:01:24.870035100 +0900
@@ -61,7 +61,7 @@ LDFLAGS="$tmp_LDFLAGS"
 AC_CHECK_FUNC(setsockopt, , AC_CHECK_LIB(socket, setsockopt))
 
 case $host_os in 
-	*cygwin*|*mingw* )
+	*mingw* )
 		AC_CHECK_HEADERS([winsock2.h])
 		AC_CHECK_LIB(ws2_32, main, , AC_MSG_ERROR([Winsock2 is needed for Windows MinGW build.]))
 		AC_CHECK_TOOL(WINDRES, windres)
