--- origsrc/libmwaw/configure.ac	2012-03-14 00:02:21.000000000 +0900
+++ src/libmwaw/configure.ac	2012-04-14 20:10:58.512227100 +0900
@@ -89,7 +89,7 @@ AC_SUBST(MWAW2RAW_WIN32_RESOURCE)
 
 AC_MSG_CHECKING([for Win32 platform in general])
 AS_CASE([$host],
-	[*-*-mingw*|*-*-cygwin*], [platform_win32=yes],
+	[*-*-mingw*], [platform_win32=yes],
 	[platform_win32=no]
 )
 AC_MSG_RESULT([$platform_win32])
--- origsrc/libmwaw/src/conv/html/mwaw2html.cpp	2012-03-14 00:02:21.000000000 +0900
+++ src/libmwaw/src/conv/html/mwaw2html.cpp	2012-04-14 20:35:37.853927400 +0900
@@ -26,6 +26,7 @@
  * Corel Corporation or Corel Corporation Limited."
  */
 
+#include <cstring>
 #include <stdio.h>
 #include <libwpd-stream/libwpd-stream.h>
 #include "IMWAWDocument.hxx"
--- origsrc/libmwaw/src/lib/IMWAWContentListener.cxx	2012-03-14 00:02:21.000000000 +0900
+++ src/libmwaw/src/lib/IMWAWContentListener.cxx	2012-04-14 20:15:07.128616700 +0900
@@ -40,6 +40,7 @@
 
 #include <iostream>
 #include <cmath>
+#include <cstring>
 
 IMWAWContentListener::ParsingState::ParsingState() :
   m_textBuffer(), m_numDeferredTabs(0),
--- origsrc/libmwaw/src/lib/IMWAWList.cxx	2012-03-14 00:02:21.000000000 +0900
+++ src/libmwaw/src/lib/IMWAWList.cxx	2012-04-14 20:26:59.411656500 +0900
@@ -36,6 +36,8 @@
 
 #include <iostream>
 
+#include <cstring>
+
 void IMWAWList::Level::addTo(WPXPropertyList &propList, double /*actTextPosition*/, double /*listBeginPos*/, int startVal) const
 {
   propList.insert("text:min-label-width", m_labelWidth);
@@ -81,7 +83,7 @@ void IMWAWList::Level::addTo(WPXProperty
   m_sendToInterface = true;
 }
 
-int IMWAWList::Level::cmp(IMWAWList::Level::Level const &levl) const
+int IMWAWList::Level::cmp(IMWAWList::Level const &levl) const
 {
   int diff = int(m_type)-int(levl.m_type);
   if (diff) return diff;
@@ -98,7 +100,7 @@ int IMWAWList::Level::cmp(IMWAWList::Lev
   return 0;
 }
 
-std::ostream &operator<<(std::ostream &o, IMWAWList::Level::Level const &ft)
+std::ostream &operator<<(std::ostream &o, IMWAWList::Level const &ft)
 {
   o << "ListLevel[";
   switch(ft.m_type) {
--- origsrc/libmwaw/src/lib/TMWAWOleParser.cxx	2012-03-14 00:02:22.000000000 +0900
+++ src/libmwaw/src/lib/TMWAWOleParser.cxx	2012-04-14 20:27:45.521031500 +0900
@@ -70,6 +70,8 @@
  * ------------------------------------------------------------
  */
 
+#include <cstring>
+
 #include <map>
 #include <sstream>
 #include <string>
