--- origsrc/libmwaw-0.1.11/configure.ac	2013-09-04 21:36:24.000000000 +0900
+++ src/libmwaw-0.1.11/configure.ac	2013-09-19 21:43:36.673679200 +0900
@@ -99,7 +99,7 @@ AC_SUBST(MWAWZIP_WIN32_RESOURCE)
 
 AC_MSG_CHECKING([for Win32 platform in general])
 AS_CASE([$host],
-	[*-*-mingw*|*-*-cygwin*], [platform_win32=yes],
+	[*-*-mingw*], [platform_win32=yes],
 	[platform_win32=no]
 )
 AC_MSG_RESULT([$platform_win32])
