--- origsrc/tmxparser-2.1.0/CMakeLists.txt	2018-02-10 15:43:20.000000000 +0900
+++ src/tmxparser-2.1.0/CMakeLists.txt	2025-06-13 14:19:06.650037400 +0900
@@ -35,25 +35,12 @@ if (NOT USE_MINIZ)
     find_package(ZLIB)
 endif (NOT USE_MINIZ)
 
-set(LIB_CFLAGS "-std=c++11 -pedantic -Wall -Werror=strict-prototypes -Werror=old-style-definition -Werror=missing-prototypes")
-
-if (NOT USE_MINIZ)
-    list(APPEND ${LIB_CFLAGS} "-Werror")
-endif (NOT USE_MINIZ)
-
-set(EXAMPLE_CFLAGS "-std=c++11 -pedantic -Werror -Wall -g")
 #include_directories("${PROJECT_SOURCE_DIR}/include")
 
-add_library(tmxparser_static STATIC ${SOURCES} ${HEADERS})
-set_target_properties(tmxparser_static PROPERTIES
-  OUTPUT_NAME tmxparser
-  COMPILE_FLAGS ${LIB_CFLAGS})
-
 add_library(tmxparser SHARED ${SOURCES} ${HEADERS})
 set_target_properties(tmxparser PROPERTIES
   SOVERSION ${VERSION_MAJOR}
-  VERSION ${VERSION}
-  COMPILE_FLAGS ${LIB_CFLAGS})
+  VERSION ${VERSION})
 
 target_link_libraries(tmxparser ${TINYXML2_LIBRARIES})
 include_directories(${TINYXML2_INCLUDE_DIR})
@@ -64,14 +51,12 @@ if (NOT USE_MINIZ)
 endif (NOT USE_MINIZ)
 
 add_executable(run_tests test/test.cpp)
-set_target_properties(run_tests PROPERTIES
-  COMPILE_FLAGS ${EXAMPLE_CFLAGS})
 target_link_libraries(run_tests tmxparser ${TINYXML2_LIBRARIES})
 include_directories("${PROJECT_BINARY_DIR}")
 include_directories("${PROJECT_SOURCE_DIR}/src")
 
 install(FILES ${HEADERS} DESTINATION include/tmxparser)
-install(TARGETS tmxparser tmxparser_static DESTINATION lib)
+install(TARGETS tmxparser RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 
 configure_file(src/tmxparser.pc.in tmxparser.pc @ONLY)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/tmxparser.pc DESTINATION lib/pkgconfig)
