diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/tmxparser-2.0.1/CMakeLists.txt src/tmxparser-2.0.1/CMakeLists.txt
--- origsrc/tmxparser-2.0.1/CMakeLists.txt	2015-06-01 04:52:30.000000000 +0900
+++ src/tmxparser-2.0.1/CMakeLists.txt	2015-09-08 20:15:15.626333900 +0900
@@ -42,10 +42,14 @@ endif (NOT USE_MINIZ)
 set(EXAMPLE_CFLAGS "-pedantic -Werror -Wall")
 #include_directories("${PROJECT_SOURCE_DIR}/include")
 
+option(ENABLE_STATIC "enable static" OFF)
+
+if (ENABLE_STATIC)
 add_library(tmxparser_static STATIC ${SOURCES} ${HEADERS})
 set_target_properties(tmxparser_static PROPERTIES
   OUTPUT_NAME tmxparser
   COMPILE_FLAGS ${LIB_CFLAGS})
+endif()
 
 add_library(tmxparser SHARED ${SOURCES} ${HEADERS})
 set_target_properties(tmxparser PROPERTIES
@@ -69,7 +73,10 @@ include_directories("${PROJECT_BINARY_DI
 include_directories("${PROJECT_SOURCE_DIR}/src")
 
 install(FILES ${HEADERS} DESTINATION include/tmxparser)
-install(TARGETS tmxparser tmxparser_static DESTINATION lib)
+install(TARGETS tmxparser RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+if(ENABLE_STATIC)
+install(TARGETS tmxparser_static DESTINATION lib)
+endif()
 
 configure_file(src/tmxparser.pc.in tmxparser.pc @ONLY)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/tmxparser.pc DESTINATION lib/pkgconfig)
