--- origsrc/libmxp-0.2.4/src/CMakeLists.txt	2009-04-27 00:02:52.000000000 +0900
+++ src/libmxp-0.2.4/src/CMakeLists.txt	2011-09-14 22:20:35.532460700 +0900
@@ -15,7 +15,9 @@ set (mxp_SRCS
 add_library(mxp SHARED ${mxp_SRCS})
 
 set_target_properties(mxp PROPERTIES VERSION 0.0.4 SOVERSION 0)
-install(TARGETS mxp DESTINATION lib${LIB_SUFFIX} )
+install(TARGETS mxp RUNTIME DESTINATION bin
+                    LIBRARY DESTINATION lib${LIB_SUFFIX} 
+                    ARCHIVE DESTINATION lib${LIB_SUFFIX} )
 
 install (FILES libmxp.h DESTINATION include/libmxp)
 
