diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libmxp/CMakeLists.txt src/libmxp/CMakeLists.txt
--- origsrc/libmxp/CMakeLists.txt	2015-09-11 18:54:49.000000000 +0900
+++ src/libmxp/CMakeLists.txt	2015-10-23 23:51:06.384300500 +0900
@@ -8,3 +8,11 @@ SET(CMAKE_CXX_FLAGS "-Wall -O2")
 
 add_subdirectory( src )
 
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "0.2.4")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/${PROJECT_NAME}.pc.in ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libmxp/libmxp.pc.in src/libmxp/libmxp.pc.in
--- origsrc/libmxp/libmxp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libmxp/libmxp.pc.in	2015-10-23 23:51:17.178178100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libmxp
+Description: Parser for the MUD eXtension protocol
+Version: @VERSION@
+Libs: -L${libdir} -lmxp
+Cflags: -I${includedir}/libmxp
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libmxp/src/CMakeLists.txt src/libmxp/src/CMakeLists.txt
--- origsrc/libmxp/src/CMakeLists.txt	2015-09-11 18:54:49.000000000 +0900
+++ src/libmxp/src/CMakeLists.txt	2015-10-23 23:50:11.762706700 +0900
@@ -15,7 +15,9 @@ set (mxp_SRCS
 add_library(mxp SHARED ${mxp_SRCS})
 
 set_target_properties(mxp PROPERTIES VERSION 0.0.4 SOVERSION 0)
-install(TARGETS mxp DESTINATION lib${LIB_SUFFIX} )
+install(TARGETS mxp RUNTIME DESTINATION bin
+                    LIBRARY DESTINATION lib${LIB_SUFFIX} 
+                    ARCHIVE DESTINATION lib${LIB_SUFFIX} )
 
 install (FILES libmxp.h DESTINATION include/libmxp)
 
