--- origsrc/libxmlpatch-0.3.3/diff/Makefile.am	2010-03-03 22:53:58.000000000 +0900
+++ src/libxmlpatch-0.3.3/diff/Makefile.am	2012-06-08 23:24:20.407885500 +0900
@@ -30,6 +30,7 @@ lib_LTLIBRARIES = libxmldiff.la
 
 libxmldiff_la_SOURCES = xml_diff.c
 libxmldiff_la_LIBADD = $(top_builddir)/src/libxmlpatch.la $(DEPS_XML_LIBS) $(DEPS_GLIB_LIBS)
+libxmldiff_la_LDFLAGS = -no-undefined
 
 libxmldiff_includedir = $(includedir)/libxmldiff
 libxmldiff_include_HEADERS = xml_diff.h
--- origsrc/libxmlpatch-0.3.3/src/Makefile.am	2010-03-03 22:53:58.000000000 +0900
+++ src/libxmlpatch-0.3.3/src/Makefile.am	2012-06-08 23:24:37.376635500 +0900
@@ -31,6 +31,7 @@ lib_LTLIBRARIES = libxmlpatch.la
 libxmlpatch_la_CFLAGS = $(DEPS_XML_CFLAGS)
 libxmlpatch_la_SOURCES = xml_patch.c
 libxmlpatch_la_LIBADD = $(DEPS_XML_LIBS)
+libxmlpatch_la_LDFLAGS = -no-undefined
 
 libxmlpatch_includedir = $(includedir)/libxmlpatch
 libxmlpatch_include_HEADERS = xml_patch.h
