--- origsrc/libxmlpatch/diff/Makefile.am	2012-08-20 16:15:21.000000000 +0900
+++ src/libxmlpatch/diff/Makefile.am	2024-10-29 08:34:23.964299200 +0900
@@ -30,6 +30,7 @@ lib_LTLIBRARIES = libxmldiff.la
 
 libxmldiff_la_SOURCES = xml_diff.c
 libxmldiff_la_LIBADD = $(top_builddir)/src/libxmlpatch.la $(DEPS_XML_LIBS) $(DEPS_GLIB_LIBS)
+libxmldiff_la_LDFLAGS = -no-undefined
 
 libxmldiff_includedir = $(includedir)/libxmldiff
 libxmldiff_include_HEADERS = xml_diff.h
--- origsrc/libxmlpatch/diff/libxmldiff.pc.in	2012-08-20 16:15:21.000000000 +0900
+++ src/libxmlpatch/diff/libxmldiff.pc.in	2024-10-29 09:18:43.353278900 +0900
@@ -5,7 +5,7 @@ includedir=@includedir@
 
 Name: libxmldiff
 Description: an xml-diff library
-Version: 0.1.0
+Version: @VERSION@
 Requires: libxml-2.0 glib-2.0
 Libs: -L${libdir} -lxmldiff
 Cflags: -I${includedir}/libxmldiff
--- origsrc/libxmlpatch/src/Makefile.am	2012-08-20 16:15:21.000000000 +0900
+++ src/libxmlpatch/src/Makefile.am	2024-10-29 08:34:23.964299200 +0900
@@ -27,6 +27,7 @@ lib_LTLIBRARIES = libxmlpatch.la
 libxmlpatch_la_CFLAGS = $(DEPS_XML_CFLAGS)
 libxmlpatch_la_SOURCES = xml_patch.c
 libxmlpatch_la_LIBADD = $(DEPS_XML_LIBS)
+libxmlpatch_la_LDFLAGS = -no-undefined
 
 libxmlpatch_includedir = $(includedir)/libxmlpatch
 libxmlpatch_include_HEADERS = xml_patch.h
--- origsrc/libxmlpatch/src/libxmlpatch.pc.in	2012-08-20 16:15:21.000000000 +0900
+++ src/libxmlpatch/src/libxmlpatch.pc.in	2024-10-29 09:18:26.478433300 +0900
@@ -5,7 +5,7 @@ includedir=@includedir@
 
 Name: libxmlpatch
 Description: an xml-patch library
-Version: 0.1.0
+Version: @VERSION@
 Requires: libxml-2.0
 Libs: -L${libdir} -lxmlpatch
 Cflags: -I${includedir}/libxmlpatch
--- origsrc/libxmlpatch/tests/Makefile.am	2012-08-20 16:15:21.000000000 +0900
+++ src/libxmlpatch/tests/Makefile.am	2024-10-29 08:34:23.964299200 +0900
@@ -1,10 +1,9 @@
 TESTS = test-diff
 
-test_PROGRAMS = test-diff
+check_PROGRAMS = test-diff
 test_diff_SOURCES = test_diff.c
 test_diff_CFLAGS = -I$(top_srcdir)/src -I$(top_srcdir)/diff $(DEPS_XML_CFLAGS) $(DEPS_GLIB_CFLAGS)
 test_diff_LDADD = $(top_builddir)/src/libxmlpatch.la $(top_builddir)/diff/libxmldiff.la $(DEPS_XML_LIBS) $(DEPS_GLIB_LIBS)
-testdir = $(libdir)/tests/xml_diff
 
 bin_PROGRAMS = xml-patch xml-diff
 
