diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/pcl-pcl-1.7.2/CMakeLists.txt src/pcl-pcl-1.7.2/CMakeLists.txt
--- origsrc/pcl-pcl-1.7.2/CMakeLists.txt	2014-09-11 06:22:57.000000000 +0900
+++ src/pcl-pcl-1.7.2/CMakeLists.txt	2015-11-08 14:42:08.378261700 +0900
@@ -89,8 +89,7 @@ endif (PCL_ENABLE_SSE)
 
 # ---[ Unix/Darwin/Windows specific flags
 if(CMAKE_COMPILER_IS_GNUCXX)
-  if("${CMAKE_CXX_FLAGS}" STREQUAL "")
-    SET(CMAKE_CXX_FLAGS "-Wall -Wextra -Wno-unknown-pragmas -fno-strict-aliasing -Wno-format-extra-args -Wno-sign-compare -Wno-invalid-offsetof -Wno-conversion ${SSE_FLAGS}")
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wno-unknown-pragmas -fno-strict-aliasing -Wno-format-extra-args -Wno-sign-compare -Wno-invalid-offsetof -Wno-conversion ${SSE_FLAGS}")
 
     # Enable -Wabi for GCC > 4.3, and -Wno-deprecated for GCC < 4.3
     # to disable a lot of warnings which are not fixable
@@ -102,11 +101,10 @@ if(CMAKE_COMPILER_IS_GNUCXX)
       message(STATUS "-- GCC < 4.3 found, enabling -Wno-deprecated")
       SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated")
     endif ()
-  endif()
 
-  if(NOT ANDROID)
+  if(NOT ANDROID AND NOT CYGWIN)
     SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
-  endif(NOT ANDROID)
+  endif(NOT ANDROID AND NOT CYGWIN)
 
   if(WIN32)
     if(PCL_SHARED_LIBS)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/pcl-pcl-1.7.2/cmake/pcl_utils.cmake src/pcl-pcl-1.7.2/cmake/pcl_utils.cmake
--- origsrc/pcl-pcl-1.7.2/cmake/pcl_utils.cmake	2014-09-11 06:22:57.000000000 +0900
+++ src/pcl-pcl-1.7.2/cmake/pcl_utils.cmake	2015-11-08 16:42:28.182073400 +0900
@@ -110,6 +110,8 @@ macro(SET_INSTALL_DIRS)
     set(PKGCFG_INSTALL_DIR "${LIB_INSTALL_DIR}/pkgconfig")
     if(WIN32 AND NOT MINGW)
         set(PCLCONFIG_INSTALL_DIR "cmake")
+    elseif(CYGWIN)
+        set(PCLCONFIG_INSTALL_DIR "lib/cmake/${PROJECT_NAME}")
       else(WIN32 AND NOT MINGW)
         set(PCLCONFIG_INSTALL_DIR "share/${PROJECT_NAME_LOWER}-${PCL_MAJOR_VERSION}.${PCL_MINOR_VERSION}")
       endif(WIN32 AND NOT MINGW)
@@ -444,4 +446,3 @@ function(find_python_module module)
   endif(NOT PY_${module_upper})
   find_package_handle_standard_args(PY_${module} DEFAULT_MSG PY_${module_upper})
 endfunction(find_python_module)
-
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/pcl-pcl-1.7.2/surface/include/pcl/surface/3rdparty/poisson4/multi_grid_octree_data.hpp src/pcl-pcl-1.7.2/surface/include/pcl/surface/3rdparty/poisson4/multi_grid_octree_data.hpp
--- origsrc/pcl-pcl-1.7.2/surface/include/pcl/surface/3rdparty/poisson4/multi_grid_octree_data.hpp	2014-09-11 06:22:57.000000000 +0900
+++ src/pcl-pcl-1.7.2/surface/include/pcl/surface/3rdparty/poisson4/multi_grid_octree_data.hpp	2015-11-08 14:42:08.393886700 +0900
@@ -36,6 +36,12 @@ DAMAGE.
   #endif
 #endif
 
+#if defined(_X)
+#undef _X
+#endif
+#if defined(_B)
+#undef _B
+#endif
 
 #define ITERATION_POWER 1.0/3
 #define MEMORY_ALLOCATOR_BLOCK_SIZE 1<<12
