--- origsrc/bitcoin-seeder-0.1.1/Makefile	2012-01-10 09:20:22.000000000 +0900
+++ src/bitcoin-seeder-0.1.1/Makefile	2024-09-25 10:42:03.435179000 +0900
@@ -1,5 +1,5 @@
 dnsseed: dns.o bitcoin.o netbase.o protocol.o db.o main.o
-	g++ -pthread -lssl -o dnsseed dns.o bitcoin.o netbase.o protocol.o db.o main.o
+	g++ -pthread -o dnsseed dns.o bitcoin.o netbase.o protocol.o db.o main.o -lcrypto -lssl
 	strip -s dnsseed
 
 dnsseed.dbg: dns.o bitcoin.o netbase.o protocol.o db.o main.o
