--- origsrc/logg-0.5.0/CMakeLists.txt	2022-10-31 05:24:16.000000000 +0900
+++ src/logg-0.5.0/CMakeLists.txt	2023-09-25 16:02:13.575619400 +0900
@@ -40,10 +40,14 @@ target_sources(logg
     src/internal/io.cpp
 )
 
+if (DEFINED SOVERSION)
+    set_target_properties(logg PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 # Install in an unix system (not apple)
 if(UNIX AND NOT APPLE)
     include(GNUInstallDirs)
-    install(TARGETS logg DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    install(TARGETS logg RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
     # Public sources
     install(FILES src/logg.hpp DESTINATION
         ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME})
