--- origsrc/m4rie-20150908/Makefile.am	2015-09-04 18:47:02.000000000 +0900
+++ src/m4rie-20150908/Makefile.am	2019-12-20 08:15:11.640112400 +0900
@@ -52,3 +52,5 @@ libm4rie_la_LIBADD = -lm4ri
 SUBDIRS = . tests
 DIST_SUBDIRS = $(SUBDIRS) bench
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/m4rie-20150908/configure.ac	2015-09-04 17:52:09.000000000 +0900
+++ src/m4rie-20150908/configure.ac	2019-12-20 08:15:11.702679000 +0900
@@ -64,6 +64,6 @@ AC_SUBST(RELEASE)
 
 AC_PROG_MAKE_SET
 
-AC_CONFIG_FILES([Makefile tests/Makefile bench/Makefile])
+AC_CONFIG_FILES([m4rie.pc Makefile tests/Makefile bench/Makefile])
 AC_OUTPUT
 
--- origsrc/m4rie-20150908/m4rie.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/m4rie-20150908/m4rie.pc.in	2019-12-20 08:15:11.780852300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: m4rie
+Description: Linear Algebra over F_2^e
+Version: @VERSION@
+Libs: -L${libdir} -lm4rie
+Cflags: -I${includedir}
--- origsrc/m4rie-20150908/tests/Makefile.am	2015-09-04 18:13:03.000000000 +0900
+++ src/m4rie-20150908/tests/Makefile.am	2019-12-20 08:15:12.296450700 +0900
@@ -1,7 +1,7 @@
 AUTOMAKE_OPTIONS = foreign subdir-objects
 
 AM_CFLAGS = ${SIMD_FLAGS} ${OPENMP_CFLAGS} ${DEBUG_FLAGS} ${M4RIE_M4RI_CFLAGS} ${M4RI_CFLAGS}
-CFLAGS = ${AM_CFLAGS} -I.. @CFLAGS@ ${M4RIE_M4RI_CFLAGS} ${M4RI_CFLAGS}
+CFLAGS = ${AM_CFLAGS} -I.. @CFLAGS@ ${M4RIE_M4RI_CFLAGS} ${M4RI_CFLAGS} -I$(top_srcdir)
 LDADD = ${top_builddir}/libm4rie.la -lm4ri -lm
 LDFLAGS = ${M4RIE_M4RI_LDFLAGS} -no-install
 
