--- origsrc/m4rie-20140914/Makefile.am	2014-09-14 01:58:03.000000000 +0900
+++ src/m4rie-20140914/Makefile.am	2015-01-04 12:33:16.459689400 +0900
@@ -50,8 +50,8 @@ libm4rie_la_LDFLAGS = -release 0.0.$(REL
 libm4rie_la_LIBADD = -lm4ri
 
 #testing
-TESTCFLAGS = ${AM_CFLAGS} @CFLAGS@ ${M4RIE_M4RI_CFLAGS} ${M4RI_CFLAGS} -I./tests
-TESTLIBADD = -lm4ri -lm4rie -lm
+TESTCFLAGS = ${AM_CFLAGS} @CFLAGS@ ${M4RIE_M4RI_CFLAGS} ${M4RI_CFLAGS} -I./tests -I$(top_srcdir)/m4rie
+TESTLIBADD = -lm4ri -lm libm4rie.la
 TESTLDADD = 
 TESTLDFLAGS = ${M4RIE_M4RI_LDFLAGS}
 
