--- origsrc/libcsystem-20150629/Makefile.am	2015-06-30 03:19:10.000000000 +0900
+++ src/libcsystem-20150629/Makefile.am	2016-04-04 23:52:42.742832200 +0900
@@ -3,13 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libclocale \
-	libcnotify \
-	libuna \
 	libcsystem \
-	po \
 	manuals \
 	tests \
 	msvscpp
@@ -87,4 +81,3 @@ splint:
 	(cd $(srcdir)/libcsystem && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
-
--- origsrc/libcsystem-20150629/configure.ac	2015-06-30 03:19:10.000000000 +0900
+++ src/libcsystem-20150629/configure.ac	2016-04-04 23:52:42.758300900 +0900
@@ -143,11 +143,6 @@ dnl Have configure make the Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
 AC_CONFIG_FILES([libcsystem/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
@@ -186,4 +181,3 @@ Features:
    Verbose output:              $ac_cv_enable_verbose_output
    Debug output:                $ac_cv_enable_debug_output
 ]);
-
--- origsrc/libcsystem-20150629/libcsystem/Makefile.am	2015-06-30 03:19:10.000000000 +0900
+++ src/libcsystem-20150629/libcsystem/Makefile.am	2016-04-04 23:52:42.758300900 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -56,4 +57,3 @@ distclean: clean
 splint:
 	@echo "Running splint on libcsystem ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(libcsystem_la_SOURCES)
-
