--- origsrc/dc3dd-7.2.646/configure.ac	2012-11-06 21:43:00.000000000 +0900
+++ src/dc3dd-7.2.646/configure.ac	2018-03-28 09:39:28.396000000 +0900
@@ -24,7 +24,7 @@ AC_PREREQ(2.61)
 # indicates that it is built from the 219th delta (in _some_ repository)
 # following the v6.9 tag, and that 58ddd is a prefix of the commit SHA1.
 AC_INIT([dc3dd],
-	m4_esyscmd([build-aux/git-version-gen .version]),
+	[7.2.646],
 	[dc3dd@dc3.mil])
 
 AC_CONFIG_SRCDIR(src/dc3dd.c)
--- origsrc/dc3dd-7.2.646/src/dc3dd.c	2012-12-03 17:29:48.000000000 +0900
+++ src/dc3dd-7.2.646/src/dc3dd.c	2018-03-28 09:57:18.657000000 +0900
@@ -50,7 +50,6 @@
    #include <sys/ioctl.h>
 #elif defined (__CYGWIN__)
    #include <sys/ioctl.h>
-   #include <cygwin/types.h>
    #include <cygwin/fs.h>
 #endif
 
