--- origsrc/libdasm-1.5/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libdasm-1.5/Makefile.am	2012-03-31 13:03:55.357392600 +0900
@@ -0,0 +1,13 @@
+
+include_HEADERS = libdasm.h
+
+lib_LTLIBRARIES = libdasm.la
+libdasm_la_LDFLAGS = -no-undefined
+libdasm_la_SOURCES = libdasm.c
+
+check_PROGRAMS = das simple
+das_LDADD = libdasm.la
+das_SOURCES = examples/das.c
+simple_LDADD = libdasm.la
+simple_SOURCES = examples/simple.c
+
--- origsrc/libdasm-1.5/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/libdasm-1.5/configure.ac	2012-03-31 13:01:26.338191700 +0900
@@ -0,0 +1,33 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([libdasm], [1.5])
+AC_CONFIG_SRCDIR([libdasm.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_MAKE_SET
+LT_INIT([disable-static])
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_INT16_T
+AC_TYPE_INT32_T
+AC_TYPE_INT64_T
+AC_TYPE_INT8_T
+AC_TYPE_SSIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([memset strncasecmp])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
