--- origsrc/libgltf-0.0.2/configure.ac	2014-10-02 03:31:43.000000000 +0900
+++ src/libgltf-0.0.2/configure.ac	2014-10-13 10:41:28.657203100 +0900
@@ -148,7 +148,7 @@ AC_SUBST(DEBUG_CXXFLAGS)
 # ==========
 # Unit tests
 # ==========
-if test "$host_os" = "linux-gnu"; then
+if test "$host_os" = "linux-gnu" -o "$host_os" = "cygwin" ; then
 
 AC_ARG_ENABLE([tests],
  [AS_HELP_STRING([--enable-tests], [Build and run unit tests])],
@@ -158,26 +158,18 @@ AC_ARG_ENABLE([tests],
 AS_IF([test "x$enable_tests" = "xyes"], [
  PKG_CHECK_MODULES([CPPUNIT], [cppunit])
 ], [])
-AC_SUBST([CPPUNIT_CFLAGS])
-AC_SUBST([CPPUNIT_LIBS])
 
 AS_IF([test "x$enable_tests" = "xyes"], [
  PKG_CHECK_MODULES([X11], [x11])
 ], [])
-AC_SUBST([X11_CFLAGS])
-AC_SUBST([X11_LIBS])
 
 AS_IF([test "x$enable_tests" = "xyes"], [
  PKG_CHECK_MODULES([GL], [gl])
 ], [])
-AC_SUBST([GL_CFLAGS])
-AC_SUBST([GL_LIBS])
 
 AS_IF([test "x$enable_tests" = "xyes"], [
  PKG_CHECK_MODULES([LIBPNG], [libpng])
 ], [])
-AC_SUBST([LIBPNG_CFLAGS])
-AC_SUBST([LIBPNG_LIBS])
 
 fi
 
