--- origsrc/libgltf-0.1.0/configure.ac	2016-12-07 04:14:15.000000000 +0900
+++ src/libgltf-0.1.0/configure.ac	2017-10-05 10:17:00.836066100 +0900
@@ -173,7 +173,7 @@ AC_SUBST(DEBUG_CXXFLAGS)
 # ==========
 # Unit tests
 # ==========
-if test "$host_os" = "linux-gnu"; then
+if test "$host_os" = "linux-gnu" -o "$host_os" = "cygwin" ; then
 
 AC_ARG_ENABLE([tests],
  [AS_HELP_STRING([--enable-tests], [Build and run unit tests])],
@@ -183,26 +183,18 @@ AC_ARG_ENABLE([tests],
 AS_IF([test "x$enable_tests" = "xyes"], [
  PKG_CHECK_MODULES([CPPUNIT], [cppunit])
 ], [])
-AC_SUBST([CPPUNIT_CFLAGS])
-AC_SUBST([CPPUNIT_LIBS])
 
 AS_IF([test "x$enable_tests" = "xyes"], [
  PKG_CHECK_MODULES([X11], [x11])
 ], [])
-AC_SUBST([X11_CFLAGS])
-AC_SUBST([X11_LIBS])
 
 AS_IF([test "x$enable_tests" = "xyes"], [
  PKG_CHECK_MODULES([GL], [gl])
 ], [])
-AC_SUBST([GL_CFLAGS])
-AC_SUBST([GL_LIBS])
 
 AS_IF([test "x$enable_tests" = "xyes"], [
  PKG_CHECK_MODULES([LIBPNG], [libpng])
 ], [])
-AC_SUBST([LIBPNG_CFLAGS])
-AC_SUBST([LIBPNG_LIBS])
 
 AS_IF([test "x$enable_tests" = "xyes"], [
  AC_CHECK_HEADER(unistd.h, [], [AC_MSG_ERROR(unistd.h not found.)], [])
