--- origsrc/z80dasm-1.1.5/test/Makefile.am	2017-07-06 04:16:40.000000000 +0900
+++ src/z80dasm-1.1.5/test/Makefile.am	2017-10-13 12:42:22.279000000 +0900
@@ -243,13 +243,13 @@ EXTRA_DIST = 	stress \
 test: cleantest $(TEST_CASES:.bin=.generated.asm) stress_test
 
 stress_test: stress
-	./stress $(top_builddir)/src/z80dasm 20
+	$(srcdir)/stress $(top_builddir)/src/z80dasm 20
 
 %.generated.asm: %.bin %.blocks %.opts %.asm
-	$(top_builddir)/src/z80dasm -g0 -l $(shell cat $(srcdir)/$*.opts) -o $@ -b $(srcdir)/$*.blocks $(srcdir)/$<
+	$(top_builddir)/src/z80dasm -g0 -l $(shell cat $(srcdir)/$*.opts) -o $@ -b $(srcdir)/$*.blocks $<
 	diff -u -I z80dasm $(srcdir)/$*.asm $@
 	z80asm $@ -o $*.generated.bin
-	diff -u $(srcdir)/$< $*.generated.bin
+	diff -u $< $*.generated.bin
 
 cleantest: clean-local
 
