--- origsrc/z80dasm-1.1.3/test/Makefile.am	2012-03-03 19:12:20.000000000 +0900
+++ src/z80dasm-1.1.3/test/Makefile.am	2015-05-06 19:36:36.538795700 +0900
@@ -14,23 +14,23 @@ TEST_CASES =	test.asm
 test: $(TEST_CASES) labelsdd jrwrap stress_test cleantest
 
 stress_test: stress
-	./stress $(top_builddir)/src/z80dasm 20
+	$(srcdir)/stress $(top_builddir)/src/z80dasm 20
 
 %.asm: %.bin %.blocks
-	$(top_builddir)/src/z80dasm -at $(srcdir)/$< -o $@
+	$(top_builddir)/src/z80dasm -at $< -o $@
 	z80asm $@ -o $<-generated
-	diff $<-generated $(srcdir)/$<
-	$(top_builddir)/src/z80dasm -atl $(srcdir)/$< -b $*.blocks -o $@
+	diff $<-generated $<
+	$(top_builddir)/src/z80dasm -atl $< -b $*.blocks -o $@
 	z80asm $@ -o $<-generated
-	diff $<-generated $(srcdir)/$<
+	diff $<-generated $<
 
 labelsdd: labelsdd.bin labelsdd.asm.vetted
-	$(top_builddir)/src/z80dasm -g0 -l labelsdd.bin -o labelsdd.asm
-	diff -I z80dasm labelsdd.asm.vetted labelsdd.asm
+	$(top_builddir)/src/z80dasm -g0 -l $(srcdir)/labelsdd.bin -o labelsdd.asm
+	diff -I z80dasm $(srcdir)/labelsdd.asm.vetted labelsdd.asm
 
 jrwrap: jrwrap.bin jrwrap.asm.vetted
-	$(top_builddir)/src/z80dasm -g0 -l jrwrap.bin -o jrwrap.asm
-	diff -I z80dasm jrwrap.asm.vetted jrwrap.asm
+	$(top_builddir)/src/z80dasm -g0 -l $(srcdir)/jrwrap.bin -o jrwrap.asm
+	diff -I z80dasm $(srcdir)/jrwrap.asm.vetted jrwrap.asm
 
 cleantest:
 	rm -f *.asm *.bin-generated
