--- origsrc/ProjectTox-Core/configure.ac	2013-10-12 23:56:51.000000000 +0900
+++ src/ProjectTox-Core/configure.ac	2013-10-12 23:58:18.479263500 +0900
@@ -162,6 +162,9 @@ case $host_os in
         WIN32="yes"
         EXTRA_LT_LDFLAGS="$EXTRA_LT_LDFLAGS -no-undefined"
     ;;
+    *cygwin*)
+        EXTRA_LT_LDFLAGS="$EXTRA_LT_LDFLAGS -no-undefined"
+    ;;
     *solaris*)
         LIBS="$LIBS -lssp -lsocket -lnsl"
     ;;
