--- origsrc/libydpdict-1.0.4/configure.ac	2012-01-11 05:43:32.000000000 +0900
+++ src/libydpdict-1.0.4/configure.ac	2014-02-16 00:20:54.308938100 +0900
@@ -1,7 +1,7 @@
 dnl $Id: configure.ac 121 2012-01-10 20:43:31Z wojtekka $
 
-AC_INIT(src/ydpdict.c)
-AM_INIT_AUTOMAKE(libydpdict, 1.0.4)
+AC_INIT([libydpdict], [1.0.4])
+AM_INIT_AUTOMAKE
 AC_PREREQ(2.50)
 AC_CONFIG_HEADERS(config.h)
 
--- origsrc/libydpdict-1.0.4/src/Makefile.am	2010-05-15 19:13:57.000000000 +0900
+++ src/libydpdict-1.0.4/src/Makefile.am	2014-02-16 00:24:58.154426100 +0900
@@ -1,4 +1,4 @@
 lib_LTLIBRARIES = libydpdict.la
 libydpdict_la_SOURCES = ydpdict.c
-libydpdict_la_CFLAGS = -Wall -I$(top_srcdir)/include
-libydpdict_la_LDFLAGS = -version-number 2:2
+libydpdict_la_CFLAGS = -Wall -I$(top_builddir)/include
+libydpdict_la_LDFLAGS = -no-undefined -version-number 2:2
