--- origsrc/libydpdict-1.0.4/configure.ac	2018-11-07 04:58:12.000000000 +0900
+++ src/libydpdict-1.0.4/configure.ac	2025-07-12 09:14:06.368860500 +0900
@@ -1,7 +1,7 @@
 dnl $Id$
 
-AC_INIT(src/ydpdict.c)
-AM_INIT_AUTOMAKE(libydpdict, 1.0.4)
+AC_INIT([libydpdict], [1.0.4])
+AM_INIT_AUTOMAKE
 AC_PREREQ(2.50)
 AC_CONFIG_HEADERS(config.h)
 
--- origsrc/libydpdict-1.0.4/src/Makefile.am	2018-11-07 04:58:12.000000000 +0900
+++ src/libydpdict-1.0.4/src/Makefile.am	2025-07-12 09:14:06.370870400 +0900
@@ -1,4 +1,4 @@
 lib_LTLIBRARIES = libydpdict.la
 libydpdict_la_SOURCES = ydpdict.c
-libydpdict_la_CFLAGS = -Wall -I$(top_srcdir)/include
-libydpdict_la_LDFLAGS = -version-number 2:2
+libydpdict_la_CFLAGS = -Wall -I$(top_builddir)/include
+libydpdict_la_LDFLAGS = -no-undefined -version-number 2:2
