--- origsrc/libbitcoin-node-3.6.0/Makefile.am	2019-08-30 00:48:11.000000000 +0900
+++ src/libbitcoin-node-3.6.0/Makefile.am	2019-09-23 09:37:59.064538200 +0900
@@ -55,6 +55,7 @@ src_libbitcoin_node_la_SOURCES = \
     src/utility/performance.cpp \
     src/utility/reservation.cpp \
     src/utility/reservations.cpp
+src_libbitcoin_node_la_LDFLAGS = -no-undefined
 
 # local: test/libbitcoin-node-test
 #------------------------------------------------------------------------------
