--- origsrc/igraph-0.6.5/configure.ac	2013-03-03 05:23:36.000000000 +0900
+++ src/igraph-0.6.5/configure.ac	2013-08-03 11:38:50.792478300 +0900
@@ -216,7 +216,8 @@ if test "$internal_f2c" = "no"; then
      AC_CHECK_LIB([f2c], [f77_alloc], [],
         AC_CHECK_LIB([f2c], [F77_ALLOC_], [],
            AC_CHECK_LIB([f2c], [F77_ALLOC], [],
-	      [AC_MSG_RESULT(not found, trying to use -lf2c anyway.)]))))
+              AC_CHECK_LIB([f2c], [F77_aloc], [],
+                 [AC_MSG_RESULT(not found, trying to use -lf2c anyway.)])))))
   LDFLAGS="${LDFLAGS}"
 else
   AC_DEFINE([INTERNAL_F2C], [1], [Define to 1 if you use the internal F2C library])
