--- origsrc/CLTune-2.7.0/CMakeLists.txt	2017-06-27 04:11:53.000000000 +0900
+++ src/CLTune-2.7.0/CMakeLists.txt	2018-04-15 13:08:11.919000000 +0900
@@ -158,6 +158,10 @@ else(BUILD_SHARED_LIBS)
 endif()
 target_link_libraries(cltune ${FRAMEWORK_LIBRARIES})
 
+if(CYGWIN)
+  set_target_properties(cltune PROPERTIES SOVERSION 0)
+endif()
+
 # Sets the proper __declspec(dllexport) keyword for Visual Studio when the library is built
 if(MSVC)
   if(BUILD_SHARED_LIBS)
@@ -166,7 +170,7 @@ if(MSVC)
 endif()
 
 # Installs the library
-install(TARGETS cltune DESTINATION lib)
+install(TARGETS cltune RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 install(FILES include/cltune.h DESTINATION include)
 
 # Install pkg-config file on Linux
