--- origsrc/CLTune-2.7.0/CMakeLists.txt	2017-06-27 04:11:53.000000000 +0900
+++ src/CLTune-2.7.0/CMakeLists.txt	2025-03-14 07:41:18.109052400 +0900
@@ -158,6 +158,10 @@ else(BUILD_SHARED_LIBS)
 endif()
 target_link_libraries(cltune ${FRAMEWORK_LIBRARIES})
 
+if (DEFINED SOVERSION)
+  set_target_properties(cltune PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 # Sets the proper __declspec(dllexport) keyword for Visual Studio when the library is built
 if(MSVC)
   if(BUILD_SHARED_LIBS)
@@ -166,7 +170,7 @@ if(MSVC)
 endif()
 
 # Installs the library
-install(TARGETS cltune DESTINATION lib)
+install(TARGETS cltune RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 install(FILES include/cltune.h DESTINATION include)
 
 # Install pkg-config file on Linux
--- origsrc/CLTune-2.7.0/src/ml_model.cc	2017-06-27 04:11:53.000000000 +0900
+++ src/CLTune-2.7.0/src/ml_model.cc	2025-03-14 07:44:38.587273400 +0900
@@ -31,6 +31,7 @@
 #include <vector>
 #include <algorithm>
 #include <cmath>
+#include <limits>
 
 namespace cltune {
 // =================================================================================================
