--- origsrc/xdffileio-xdffileio-0.3/Makefile.am	2012-05-01 15:47:27.000000000 +0900
+++ src/xdffileio-xdffileio-0.3/Makefile.am	2025-08-19 17:56:32.118712100 +0900
@@ -1,5 +1,5 @@
 ACLOCAL_AMFLAGS=-I m4
-SUBDIRS = lib src doc tests
+SUBDIRS = src doc tests
 
 EXTRA_DIST=autogen.sh m4/gnulib-cache.m4
 doc_DATA = AUTHORS NEWS README COPYING
--- origsrc/xdffileio-xdffileio-0.3/configure.ac	2025-08-19 17:56:30.747276000 +0900
+++ src/xdffileio-xdffileio-0.3/configure.ac	2025-08-19 17:56:32.118712100 +0900
@@ -34,7 +34,6 @@ AM_PROG_CC_C_O
 
 LT_INIT([win32-dll disable-static])
 AC_SUBST(LIBTOOL_DEPS)
-gl_INIT
 
 AC_DEF_API_EXPORT_ATTRS
 AC_C_BIGENDIAN
@@ -79,7 +78,7 @@ AC_SUBST([CURRENT],[lib_current])
 AC_SUBST([REVISION],[lib_revision])
 AC_SUBST([AGE],[lib_age])
 
-AC_CONFIG_FILES([Makefile lib/Makefile src/Makefile src/xdffileio.pc doc/Makefile tests/Makefile])
+AC_CONFIG_FILES([Makefile src/Makefile src/xdffileio.pc doc/Makefile tests/Makefile])
 AC_OUTPUT
 
 if test "$HAVE_IEEE_FLOAT_ROUND" = -1 
@@ -95,4 +94,3 @@ There are two ways (among others) to enf
   - Do not store floating point variables in registers
           (If gcc is used, add -ffloat-store to compiler flags)"
 fi
-
--- origsrc/xdffileio-xdffileio-0.3/src/Makefile.am	2012-05-01 15:47:27.000000000 +0900
+++ src/xdffileio-xdffileio-0.3/src/Makefile.am	2025-08-19 17:56:32.121946900 +0900
@@ -13,7 +13,7 @@ libxdffileio_la_SOURCES = xdffile.h xdff
 AM_CPPFLAGS=-I../lib -I$(top_srcdir)/lib
 libxdffileio_la_LDFLAGS = $(AM_LDFLAGS) -no-undefined \
 			  -version-info $(CURRENT):$(REVISION):$(AGE) 
-libxdffileio_la_LIBADD = ../lib/libgnu.la
+libxdffileio_la_LIBADD = 
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = xdffileio.pc
--- origsrc/xdffileio-xdffileio-0.3/src/xdfconfig.c	2012-05-01 15:47:27.000000000 +0900
+++ src/xdffileio-xdffileio-0.3/src/xdfconfig.c	2025-08-19 17:56:32.123968800 +0900
@@ -29,7 +29,9 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <unistd.h>
+#if !defined(__CYGWIN__)
 #include <binary-io.h>
+#endif
 #include "xdfio.h"
 #include "xdftypes.h"
 #include "xdffile.h"
@@ -1041,4 +1043,3 @@ int xdf_get_event(struct xdf* xdf, unsig
 	*dur = evt->duration;
 	return 0;
 }
-
