--- origsrc/fastText-0.2.0/CMakeLists.txt	2018-12-18 12:11:13.000000000 +0900
+++ src/fastText-0.2.0/CMakeLists.txt	2019-05-06 15:05:54.339925100 +0900
@@ -44,21 +44,38 @@ set(SOURCE_FILES
     src/vector.cc)
 
 add_library(fasttext-shared SHARED ${SOURCE_FILES} ${HEADER_FILES})
-add_library(fasttext-static STATIC ${SOURCE_FILES} ${HEADER_FILES})
-add_library(fasttext-static_pic STATIC ${SOURCE_FILES} ${HEADER_FILES})
+if (NOT BUILD_SHARED_LIBS)
+    add_library(fasttext-static STATIC ${SOURCE_FILES} ${HEADER_FILES})
+    add_library(fasttext-static_pic STATIC ${SOURCE_FILES} ${HEADER_FILES})
+endif()
 set_target_properties(fasttext-shared PROPERTIES OUTPUT_NAME fasttext)
-set_target_properties(fasttext-static PROPERTIES OUTPUT_NAME fasttext)
-set_target_properties(fasttext-static_pic PROPERTIES OUTPUT_NAME fasttext_pic
-  POSITION_INDEPENDENT_CODE True)
+if (CYGWIN)
+    set_target_properties(fasttext-shared PROPERTIES SOVERSION 0)
+endif()
+if (NOT BUILD_SHARED_LIBS)
+    set_target_properties(fasttext-static PROPERTIES OUTPUT_NAME fasttext)
+    set_target_properties(fasttext-static_pic PROPERTIES OUTPUT_NAME fasttext_pic
+        POSITION_INDEPENDENT_CODE True)
+endif()
 add_executable(fasttext-bin src/main.cc)
-target_link_libraries(fasttext-bin pthread fasttext-static)
+target_link_libraries(fasttext-bin pthread fasttext-shared)
 set_target_properties(fasttext-bin PROPERTIES PUBLIC_HEADER "${HEADER_FILES}" OUTPUT_NAME fasttext)
 install (TARGETS fasttext-shared
-    LIBRARY DESTINATION lib)
-install (TARGETS fasttext-static
-    ARCHIVE DESTINATION lib)
-install (TARGETS fasttext-static_pic
+    RUNTIME DESTINATION bin
     ARCHIVE DESTINATION lib)
+if (NOT BUILD_SHARED_LIBS)
+    install (TARGETS fasttext-static
+        ARCHIVE DESTINATION lib)
+    install (TARGETS fasttext-static_pic
+        ARCHIVE DESTINATION lib)
+endif()
 install (TARGETS fasttext-bin
     RUNTIME DESTINATION bin
  PUBLIC_HEADER DESTINATION include/fasttext)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/fastText.pc.in ${CMAKE_BINARY_DIR}/fastText.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/fastText.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/fastText-0.2.0/fastText.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/fastText-0.2.0/fastText.pc.in	2019-05-06 15:04:45.407118000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: fastText
+Description: Library for fast text representation and classification
+Version: @VERSION@
+Libs: -L${libdir} -lfasttext
+Cflags: -I${includedir}
