--- origsrc/gloox-1.0.6/configure.ac	2013-09-04 00:34:31.000000000 +0900
+++ src/gloox-1.0.6/configure.ac	2013-09-18 20:45:10.666982800 +0900
@@ -94,7 +94,7 @@ AC_MSG_RESULT($libidn)
 
 dnl extra win32 libs
 case "$host_os" in
-  *mingw*|*cygwin*)
+  *mingw*)
       LIBS="${LIBS} -lwsock32 -lgdi32 -ldnsapi";;
   *)
       CXXFLAGS="$CXXFLAGS -ansi";;
@@ -109,13 +109,13 @@ AC_ARG_WITH(schannel, AC_HELP_STRING([--
       schannel=yes, schannel=no)
 if test "$schannel" = "yes"; then
   case "$host_os" in
-    *mingw*|*cygwin*)
+    *mingw*)
         ssl=yes
         LIBS="${LIBS} -lcrypt32 -lsecur32";;
     *)
         schannel=no
         echo "*"
-        echo "* You provided the '--with-schannel' switch. SChannel is only available in MinGW/cygwin. Ignoring."
+        echo "* You provided the '--with-schannel' switch. SChannel is only available in MinGW. Ignoring."
         echo "*";;
   esac
   AC_MSG_CHECKING([if SChannel should be used])
--- origsrc/gloox-1.0.6/src/linklocalclient.cpp	2013-08-30 16:17:35.000000000 +0900
+++ src/gloox-1.0.6/src/linklocalclient.cpp	2013-09-18 20:56:21.641592100 +0900
@@ -20,6 +20,7 @@
 #include "connectiontcpclient.h"
 
 #include <cstdio>
+#include <sys/time.h>
 
 #if ( !defined( _WIN32 ) && !defined( _WIN32_WCE ) ) || defined( __SYMBIAN32__ )
 # include <arpa/inet.h>
--- origsrc/gloox-1.0.6/src/linklocalmanager.cpp	2013-08-30 16:17:35.000000000 +0900
+++ src/gloox-1.0.6/src/linklocalmanager.cpp	2013-09-18 20:55:54.594717100 +0900
@@ -21,6 +21,7 @@
 
 #include <cstdio>
 #include <unistd.h>
+#include <sys/time.h>
 
 #if ( !defined( _WIN32 ) && !defined( _WIN32_WCE ) ) || defined( __SYMBIAN32__ )
 # include <arpa/inet.h>
