--- origsrc/gloox-1.0.10/configure.ac	2014-04-09 07:11:31.000000000 +0900
+++ src/gloox-1.0.10/configure.ac	2014-05-08 16:50:06.791423000 +0900
@@ -20,8 +20,6 @@ dnl Checks for programs.
 AC_PROG_INSTALL
 AC_PROG_CC
 AC_PROG_CXX
-AC_PROG_RANLIB
-AM_PROG_LIBTOOL
 AC_PROG_LIBTOOL
 
 dnl
@@ -94,7 +92,7 @@ AC_MSG_RESULT($libidn)
 
 dnl extra win32 libs
 case "$host_os" in
-  *mingw*|*cygwin*)
+  *mingw*)
       LIBS="${LIBS} -lwsock32 -lgdi32 -ldnsapi";;
   *)
       CXXFLAGS="$CXXFLAGS -ansi";;
@@ -109,13 +107,13 @@ AC_ARG_WITH(schannel, AC_HELP_STRING([--
       schannel=yes, schannel=no)
 if test "$schannel" = "yes"; then
   case "$host_os" in
-    *mingw*|*cygwin*)
+    *mingw*)
         ssl=yes
         LIBS="${LIBS} -lcrypt32 -lsecur32";;
     *)
         schannel=no
         echo "*"
-        echo "* You provided the '--with-schannel' switch. SChannel is only available in MinGW/cygwin. Ignoring."
+        echo "* You provided the '--with-schannel' switch. SChannel is only available in MinGW. Ignoring."
         echo "*";;
   esac
   AC_MSG_CHECKING([if SChannel should be used])
--- origsrc/gloox-1.0.10/src/linklocalclient.cpp	2013-08-30 16:17:35.000000000 +0900
+++ src/gloox-1.0.10/src/linklocalclient.cpp	2014-05-08 16:48:43.810954300 +0900
@@ -20,6 +20,7 @@
 #include "connectiontcpclient.h"
 
 #include <cstdio>
+#include <sys/time.h>
 
 #if ( !defined( _WIN32 ) && !defined( _WIN32_WCE ) ) || defined( __SYMBIAN32__ )
 # include <arpa/inet.h>
--- origsrc/gloox-1.0.10/src/linklocalmanager.cpp	2013-08-30 16:17:35.000000000 +0900
+++ src/gloox-1.0.10/src/linklocalmanager.cpp	2014-05-08 16:48:43.810954300 +0900
@@ -21,6 +21,7 @@
 
 #include <cstdio>
 #include <unistd.h>
+#include <sys/time.h>
 
 #if ( !defined( _WIN32 ) && !defined( _WIN32_WCE ) ) || defined( __SYMBIAN32__ )
 # include <arpa/inet.h>
