--- origsrc/gloox-1.0/configure.ac	2009-10-31 23:05:03.000000000 +0900
+++ src/gloox-1.0/configure.ac	2010-09-25 11:46:20.329300200 +0900
@@ -21,8 +21,7 @@ AC_PROG_INSTALL
 AC_PROG_CC
 AC_PROG_CXX
 AC_PROG_RANLIB
-AM_PROG_LIBTOOL
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 dnl
 dnl extra paths
@@ -71,7 +70,7 @@ AC_MSG_RESULT($libidn)
 
 dnl extra win32 libs
 case "$host_os" in
-  *mingw*|*cygwin*)
+  *mingw*)
       LIBS="${LIBS} -lwsock32 -lgdi32 -ldnsapi";;
   *)
       CXXFLAGS="$CXXFLAGS -ansi";;
@@ -86,13 +85,13 @@ AC_ARG_WITH(schannel, AC_HELP_STRING([--
       schannel=yes, schannel=no)
 if test "$schannel" = "yes"; then
   case "$host_os" in
-    *mingw*|*cygwin*)
+    *mingw*)
         ssl=yes
         LIBS="${LIBS} -lcrypt32 -lsecur32";;
     *)
         schannel=no
         echo "*"
-        echo "* You provided the '--with-schannel' switch. SChannel is only available in MinGW/cygwin. Ignoring."
+        echo "* You provided the '--with-schannel' switch. SChannel is only available in MinGW. Ignoring."
         echo "*";;
   esac
   AC_MSG_CHECKING([if SChannel should be used])
--- origsrc/gloox-1.0/src/examples/e2ee_client.cpp	2009-10-14 03:57:34.000000000 +0900
+++ src/gloox-1.0/src/examples/e2ee_client.cpp	2010-09-25 13:43:36.235550200 +0900
@@ -17,6 +17,7 @@ using namespace gloox;
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 #ifdef WIN32
 #include <windows.h>
--- origsrc/gloox-1.0/src/examples/e2ee_server.cpp	2009-10-14 03:47:21.000000000 +0900
+++ src/gloox-1.0/src/examples/e2ee_server.cpp	2010-09-25 13:45:38.266800200 +0900
@@ -17,6 +17,7 @@ using namespace gloox;
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 #ifdef WIN32
 #include <windows.h>
--- origsrc/gloox-1.0/src/examples/ft_recv.cpp	2009-10-06 22:14:26.000000000 +0900
+++ src/gloox-1.0/src/examples/ft_recv.cpp	2010-09-25 13:47:43.266800200 +0900
@@ -15,6 +15,7 @@ using namespace gloox;
 #include <string>
 
 #include <cstdio> // [s]print[f]
+#include <ctime>
 
 #if defined( WIN32 ) || defined( _WIN32 )
 # include <windows.h>
