--- origsrc/solarus-v1.6.0/cmake/AddInstallTargets.cmake	2018-12-22 23:14:40.000000000 +0900
+++ src/solarus-v1.6.0/cmake/AddInstallTargets.cmake	2019-01-01 20:52:52.253095800 +0900
@@ -19,7 +19,7 @@ set(SOLARUS_HEADERS_INSTALL_DESTINATION
 # Files to install with make install.
 # Install the shared library and the solarus-run executable.
 install(TARGETS solarus solarus-run
-  LIBRARY DESTINATION ${SOLARUS_LIBRARY_INSTALL_DESTINATION}
+  ARCHIVE DESTINATION ${SOLARUS_LIBRARY_INSTALL_DESTINATION}
   RUNTIME DESTINATION ${SOLARUS_EXECUTABLE_INSTALL_DESTINATION}
 )
 # Install headers: useful for projects that use Solarus as a library.
--- origsrc/solarus-v1.6.0/gui/CMakeLists.txt	2018-12-22 23:14:40.000000000 +0900
+++ src/solarus-v1.6.0/gui/CMakeLists.txt	2019-01-01 20:53:19.292817800 +0900
@@ -128,7 +128,7 @@ target_link_libraries(solarus-launcher
 
 # Set files to install.
 install(TARGETS solarus-gui solarus-launcher
-  LIBRARY DESTINATION ${SOLARUS_LIBRARY_INSTALL_DESTINATION}
+  ARCHIVE DESTINATION ${SOLARUS_LIBRARY_INSTALL_DESTINATION}
   RUNTIME DESTINATION ${SOLARUS_EXECUTABLE_INSTALL_DESTINATION}
 )
 install(DIRECTORY
