--- origsrc/solarus-v2.0.3/CMakeLists.txt	2026-01-22 02:00:26.000000000 +0900
+++ src/solarus-v2.0.3/CMakeLists.txt	2026-02-02 21:38:59.619851900 +0900
@@ -18,3 +18,10 @@ include(cmake/AddInstallTargets.cmake)
 include(cmake/AddUninstallTargets.cmake)
 include(cmake/AddUnitTests.cmake)
 include(cmake/AddCLI.cmake)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/solarus.pc.in ${CMAKE_BINARY_DIR}/solarus.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/solarus.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/solarus-v2.0.3/solarus.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/solarus-v2.0.3/solarus.pc.in	2026-02-02 21:39:14.251194100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: solarus
+Description: Free and open-source Action-RPG game engine
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lsolarus
+Cflags: -I${includedir}
--- origsrc/solarus-v2.0.3/src/core/QuestFiles.cpp	2026-01-22 02:00:26.000000000 +0900
+++ src/solarus-v2.0.3/src/core/QuestFiles.cpp	2026-02-02 20:45:26.027874300 +0900
@@ -37,6 +37,8 @@
 #   include "solarus/core/AppleInterface.h"
 #endif
 
+#include <string.h> // strdup()
+
 namespace Solarus {
 
 namespace QuestFiles {
