--- origsrc/libshbeu-1.1.0/configure.ac	2011-08-26 22:43:52.000000000 +0900
+++ src/libshbeu-1.1.0/configure.ac	2015-04-09 16:01:47.767771100 +0900
@@ -226,6 +226,8 @@ else
 AC_SUBST(SHLIB_VERSION_ARG)
 AC_SUBST(SHARED_VERSION_INFO)
 
+AM_CONDITIONAL(IS_LINUX, [test "$host_os" = "linux"])
+
 AC_OUTPUT([
 Makefile
 doc/Makefile
--- origsrc/libshbeu-1.1.0/src/Makefile.am	2011-08-26 22:43:52.000000000 +0900
+++ src/libshbeu-1.1.0/src/Makefile.am	2015-04-09 16:02:34.986134400 +0900
@@ -1 +1,6 @@
-SUBDIRS = libshbeu tools
+
+SUBDIRS = libshbeu
+
+if IS_LINUX
+SUBDIRS += tools
+endif
--- origsrc/libshbeu-1.1.0/src/libshbeu/Makefile.am	2011-08-26 22:43:52.000000000 +0900
+++ src/libshbeu-1.1.0/src/libshbeu/Makefile.am	2015-04-09 15:51:36.172305100 +0900
@@ -17,5 +17,5 @@ libshbeu_la_SOURCES = \
 	beu.c
 
 libshbeu_la_CFLAGS = $(UIOMUX_CFLAGS)
-libshbeu_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
+libshbeu_la_LDFLAGS = -no-undefined -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
 libshbeu_la_LIBADD = $(UIOMUX_LIBS)
