--- origsrc/lis-1.4.61/src/Makefile.am	2014-08-12 19:17:37.000000000 +0900
+++ src/lis-1.4.61/src/Makefile.am	2014-09-28 15:32:10.765433700 +0900
@@ -27,3 +27,4 @@ if ENABLE_FORTRAN
 endif
 endif
 liblis@LIBSUFFIX@_la_SOURCES = 
+liblis@LIBSUFFIX@_la_LDFLAGS = -no-undefined $(FLIBS)
--- origsrc/lis-1.4.61/src/esolver/Makefile.am	2010-07-21 21:51:00.000000000 +0900
+++ src/lis-1.4.61/src/esolver/Makefile.am	2014-09-28 15:32:30.188285300 +0900
@@ -8,16 +8,5 @@ lis_esolver_cg.c	\
 lis_esolver_si.c	\
 lis_esolver_rqi.c	
 
-AM_CFLAGS = -I$(top_srcdir)/include
+AM_CPPFLAGS = -I$(top_srcdir)/include
 CLEANFILES = *.il
-
-PPF77COMPILE = $(F77) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_FFLAGS) $(FFLAGS)
-LTPPF77COMPILE = $(LIBTOOL) --mode=compile --tag=F77 $(F77) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_FFLAGS) $(FFLAGS)
-COMPILE = $(CC) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
-LTCOMPILE = $(LIBTOOL) --mode=compile --tag=CC $(CC) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
--- origsrc/lis-1.4.61/src/fortran/Makefile.am	2014-08-12 19:17:57.000000000 +0900
+++ src/lis-1.4.61/src/fortran/Makefile.am	2014-09-28 15:32:42.908988400 +0900
@@ -12,14 +12,3 @@ lisf_init.F
 
 AM_CPPFLAGS = -I$(top_srcdir)/include
 CLEANFILES = *.il
-
-PPF77COMPILE = $(F77) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_FFLAGS) $(FFLAGS)
-LTPPF77COMPILE = $(LIBTOOL) --mode=compile --tag=F77 $(F77) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_FFLAGS) $(FFLAGS)
-COMPILE = $(CC) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
-LTCOMPILE = $(LIBTOOL) --mode=compile --tag=CC $(CC) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
--- origsrc/lis-1.4.61/src/fortran/amg/Makefile.am	2013-03-21 20:08:00.000000000 +0900
+++ src/lis-1.4.61/src/fortran/amg/Makefile.am	2014-09-28 15:33:00.300590000 +0900
@@ -25,25 +25,6 @@ endif
 
 CLEANFILES = *.mod
 AM_CPPFLAGS = -I$(top_srcdir)/include
-FCCOMPILE = $(FC) $(AMDEFS) $(AM_FCFLAGS) $(FCFLAGS)
-LTFCCOMPILE = $(LIBTOOL) --mode=compile $(FC) $(AMDEFS) $(AM_FCFLAGS) $(FCFLAGS)
-FCLD = $(FC)
-FCLINK = $(LIBTOOL) --mode=link $(FCLD) $(AM_LDFLAGS) $(LDFLAGS) -o $@
-PPFCCOMPILE = $(FC) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-        $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_FCFLAGS) $(FCFLAGS)
-LTPPFCCOMPILE = $(LIBTOOL) --mode=compile $(FC) $(AMDEFS) $(DEFS) \
-        $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-        $(AM_FCFLAGS) $(FCFLAGS)
-PPF77COMPILE = $(F77) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_FFLAGS) $(FFLAGS)
-LTPPF77COMPILE = $(LIBTOOL) --mode=compile --tag=F77 $(F77) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_FFLAGS) $(FFLAGS)
-COMPILE = $(CC) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
-LTCOMPILE = $(LIBTOOL) --mode=compile --tag=CC $(CC) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
 
 .NOTPARALLEL:
 .NO_PARALLEL:
--- origsrc/lis-1.4.61/src/matrix/Makefile.am	2013-01-17 18:19:44.000000000 +0900
+++ src/lis-1.4.61/src/matrix/Makefile.am	2014-09-28 15:33:20.852347800 +0900
@@ -18,16 +18,5 @@ lis_matrix_ops.c           \
 lis_matrix_rco.c           \
 lis_matrix_vbr.c           
 
-AM_CFLAGS = -I$(top_srcdir)/include $(ILIBS)
+AM_CPPFLAGS = -I$(top_srcdir)/include $(ILIBS)
 CLEANFILES = *.il
-
-PPF77COMPILE = $(F77) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_FFLAGS) $(FFLAGS)
-LTPPF77COMPILE = $(LIBTOOL) --mode=compile --tag=F77 $(F77) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_FFLAGS) $(FFLAGS)
-COMPILE = $(CC) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
-LTCOMPILE = $(LIBTOOL) --mode=compile --tag=CC $(CC) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
--- origsrc/lis-1.4.61/src/matvec/Makefile.am	2013-01-17 18:19:22.000000000 +0900
+++ src/lis-1.4.61/src/matvec/Makefile.am	2014-09-28 15:33:36.542777500 +0900
@@ -13,16 +13,5 @@ lis_matvec_msr.c \
 lis_matvec_vbr.c
 
 
-AM_CFLAGS = -I$(top_srcdir)/include
+AM_CPPFLAGS = -I$(top_srcdir)/include
 CLEANFILES = *.il
-
-PPF77COMPILE = $(F77) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_FFLAGS) $(FFLAGS)
-LTPPF77COMPILE = $(LIBTOOL) --mode=compile --tag=F77 $(F77) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_FFLAGS) $(FFLAGS)
-COMPILE = $(CC) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
-LTCOMPILE = $(LIBTOOL) --mode=compile --tag=CC $(CC) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
--- origsrc/lis-1.4.61/src/precision/Makefile.am	2007-10-31 14:58:42.000000000 +0900
+++ src/lis-1.4.61/src/precision/Makefile.am	2014-09-28 15:33:49.794730600 +0900
@@ -4,16 +4,5 @@ lis_precision_vec.c    \
 lis_precision_matvec.c    
 
 
-AM_CFLAGS = -I$(top_srcdir)/include $(MPFLAG)
+AM_CPPFLAGS = -I$(top_srcdir)/include $(MPFLAG)
 CLEANFILES = *.il
-
-PPF77COMPILE = $(F77) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_FFLAGS) $(FFLAGS)
-LTPPF77COMPILE = $(LIBTOOL) --mode=compile --tag=F77 $(F77) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_FFLAGS) $(FFLAGS)
-COMPILE = $(CC) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
-LTCOMPILE = $(LIBTOOL) --mode=compile --tag=CC $(CC) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
--- origsrc/lis-1.4.61/src/precon/Makefile.am	2007-10-31 14:58:42.000000000 +0900
+++ src/lis-1.4.61/src/precon/Makefile.am	2014-09-28 15:34:02.343558700 +0900
@@ -12,16 +12,5 @@ lis_precon_sainv.c  \
 lis_precon_ssor.c  
 
 
-AM_CFLAGS = -I$(top_srcdir)/include
+AM_CPPFLAGS = -I$(top_srcdir)/include
 CLEANFILES = *.il
-
-PPF77COMPILE = $(F77) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_FFLAGS) $(FFLAGS)
-LTPPF77COMPILE = $(LIBTOOL) --mode=compile --tag=F77 $(F77) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_FFLAGS) $(FFLAGS)
-COMPILE = $(CC) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
-LTCOMPILE = $(LIBTOOL) --mode=compile --tag=CC $(CC) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
--- origsrc/lis-1.4.61/src/solver/Makefile.am	2008-10-07 20:06:44.000000000 +0900
+++ src/lis-1.4.61/src/solver/Makefile.am	2014-09-28 15:34:17.064261800 +0900
@@ -17,16 +17,5 @@ lis_solver_qmr.c       \
 lis_solver_sor.c       
 
 
-AM_CFLAGS = -I$(top_srcdir)/include
+AM_CPPFLAGS = -I$(top_srcdir)/include
 CLEANFILES = *.il
-
-PPF77COMPILE = $(F77) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_FFLAGS) $(FFLAGS)
-LTPPF77COMPILE = $(LIBTOOL) --mode=compile --tag=F77 $(F77) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_FFLAGS) $(FFLAGS)
-COMPILE = $(CC) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
-LTCOMPILE = $(LIBTOOL) --mode=compile --tag=CC $(CC) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
--- origsrc/lis-1.4.61/src/system/Makefile.am	2014-07-23 22:02:02.000000000 +0900
+++ src/lis-1.4.61/src/system/Makefile.am	2014-09-28 15:34:30.081840000 +0900
@@ -13,16 +13,5 @@ lis_time.c      \
 mt19937ar.c
 
 
-AM_CFLAGS = -I$(top_srcdir)/include
+AM_CPPFLAGS = -I$(top_srcdir)/include
 CLEANFILES = *.il
-
-PPF77COMPILE = $(F77) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_FFLAGS) $(FFLAGS)
-LTPPF77COMPILE = $(LIBTOOL) --mode=compile --tag=F77 $(F77) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_FFLAGS) $(FFLAGS)
-COMPILE = $(CC) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
-LTCOMPILE = $(LIBTOOL) --mode=compile --tag=CC $(CC) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
--- origsrc/lis-1.4.61/src/vector/Makefile.am	2007-10-31 14:58:46.000000000 +0900
+++ src/lis-1.4.61/src/vector/Makefile.am	2014-09-28 15:34:42.755668100 +0900
@@ -4,16 +4,5 @@ lis_vector_ops.c  \
 lis_vector_opv.c
 
 
-AM_CFLAGS = -I$(top_srcdir)/include
+AM_CPPFLAGS = -I$(top_srcdir)/include
 CLEANFILES = *.il
-
-PPF77COMPILE = $(F77) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_FFLAGS) $(FFLAGS)
-LTPPF77COMPILE = $(LIBTOOL) --mode=compile --tag=F77 $(F77) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_FFLAGS) $(FFLAGS)
-COMPILE = $(CC) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
-LTCOMPILE = $(LIBTOOL) --mode=compile --tag=CC $(CC) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
--- origsrc/lis-1.4.61/test/Makefile.am	2014-09-13 15:06:53.000000000 +0900
+++ src/lis-1.4.61/test/Makefile.am	2014-09-28 15:35:50.117972800 +0900
@@ -5,18 +5,19 @@ AM_LDFLAGS = $(AMDEFS)
 
 EXTRA_DIST = defs test.sh testmat.mtx testmat2.mtx
 
-testdir = ${prefix}/share/examples/lis
+testdir = $(libdir)/lis/examples
 
 test_DATA = testmat.mtx testmat2.mtx
 
 test_SCRIPTS = defs test.sh
 
-test_PROGRAMS = test1 test2 test2b test3 test3b test4 test5 test6 etest1 etest2 etest3 etest4 etest5 etest6 spmvtest1 spmvtest2 spmvtest2b spmvtest3 spmvtest3b spmvtest4 spmvtest5
+check_PROGRAMS = test1 test2 test2b test3 test3b test4 test5 test6 etest1 etest2 etest3 etest4 etest5 etest6 spmvtest1 spmvtest2 spmvtest2b spmvtest3 spmvtest3b spmvtest4 spmvtest5
 if ENABLE_FORTRAN
-  test_PROGRAMS += test1f test4f etest1f etest4f
+  check_PROGRAMS += test1f test4f etest1f etest4f
+  LDADD += $(FLIBS)
 endif
 if ENABLE_F90
-  test_PROGRAMS += test2f
+  check_PROGRAMS += test2f
 endif
 
 test1_SOURCES  = test1.c
@@ -57,33 +58,6 @@ esolve_SOURCES  = etest5.c
 hpcg_kernel_SOURCES    = test3b.c
 hpcg_spmvtest_SOURCES    = spmvtest3b.c
 
-PPF77COMPILE = $(F77) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_FFLAGS) $(FFLAGS)
-LTPPF77COMPILE = $(LIBTOOL) --mode=compile --tag=F77 $(F77) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_FFLAGS) $(FFLAGS)
-COMPILE = $(CC) $(AMDEFS) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
-LTCOMPILE = $(LIBTOOL) --mode=compile --tag=CC $(CC) $(AMDEFS) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
-
-if ENABLE_SAAMG
-  LINK = $(LIBTOOL) --mode=link $(FC) \
-	$(AM_LDFLAGS) $(LDFLAGS) $(FCLDFLAGS) -o $@ $(LIBS) $(CLIBS)
-  F77LINK = $(LIBTOOL) --mode=link $(FC) \
-	$(AM_LDFLAGS) $(LDFLAGS) -o $@ $(LIBS) $(CLIBS)
-else
-if ENABLE_FORTRAN
-  LINK = $(LIBTOOL) --mode=link $(FC) \
-	$(AM_LDFLAGS) $(LDFLAGS) $(FCLDFLAGS) -o $@ $(LIBS) $(CLIBS)
-  F77LINK = $(LIBTOOL) --mode=link $(F77) \
-	$(AM_LDFLAGS) $(LDFLAGS) -o $@ $(LIBS) $(CLIBS)
-else
-  LINK = $(LIBTOOL) --mode=link $(CC) $(AM_LDFLAGS) $(LDFLAGS) -o $@ $(LIBS)
-endif
-endif
-
 CLEANFILES = *.il
 
 
--- origsrc/lis-1.4.61/test/test.sh	2014-09-11 21:04:43.000000000 +0900
+++ src/lis-1.4.61/test/test.sh	2014-09-28 15:39:30.617972800 +0900
@@ -27,25 +27,25 @@ fi
 
 echo ' '
 echo 'checking linear solvers...'
-$MPIRUN $srcdir/test1 $srcdir/testmat.mtx 0 /dev/null /dev/null
+$MPIRUN $top_builddir/test1 $srcdir/testmat.mtx 0 /dev/null /dev/null
 
 echo 'checking eigensolvers...'
-$MPIRUN $srcdir/etest1 $srcdir/testmat.mtx /dev/null /dev/null
+$MPIRUN $top_builddir/etest1 $srcdir/testmat.mtx /dev/null /dev/null
 
 if test "$enable_fortran" = "yes" || test "$enable_f90" = "yes"; then
   echo 'checking Fortran interface...'
-  $MPIRUN $srcdir/test4f
+  $MPIRUN $top_builddir/test4f
   echo ' '
 fi
 
 if test "$enable_quad" = "yes"; then
   echo 'checking double-double precision operations...'
-  $MPIRUN $srcdir/test5 200 2.0 -f double
-  $MPIRUN $srcdir/test5 200 2.0 -f quad
+  $MPIRUN $top_builddir/test5 200 2.0 -f double
+  $MPIRUN $top_builddir/test5 200 2.0 -f quad
 fi
 
 if test "$enable_saamg" = "yes"; then
   echo 'checking SAAMG preconditioner...'
-  $MPIRUN $srcdir/test2 10 10 1 /dev/null /dev/null -i cg -p saamg
+  $MPIRUN $top_builddir/test2 10 10 1 /dev/null /dev/null -i cg -p saamg
 fi
 
