--- origsrc/lis-2.1.7/Makefile.am	2023-05-24 15:31:09.000000000 +0900
+++ src/lis-2.1.7/Makefile.am	2025-03-16 23:07:21.128378900 +0900
@@ -648,3 +648,6 @@ man_MANS =   doc/man/man1/lsolve.1 \
 ACLOCAL_AMFLAGS = -I m4
 
 AUTOMAKE_OPTIONS = dist-zip
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/lis-2.1.7/configure.ac	2024-12-17 11:58:10.000000000 +0900
+++ src/lis-2.1.7/configure.ac	2025-03-16 23:07:21.144006500 +0900
@@ -1305,7 +1305,7 @@ LIBS="-lm $LIBS"
 
 # Output
 
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([lis.pc Makefile
                  src/Makefile
                  src/vector/Makefile
                  src/matrix/Makefile
--- origsrc/lis-2.1.7/lis.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/lis-2.1.7/lis.pc.in	2025-03-16 23:07:21.144006500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: lis
+Description: Library of Iterative Solvers for Linear Systems
+Version: @VERSION@
+Libs: -L${libdir} -llis
+Cflags: -I${includedir}
--- origsrc/lis-2.1.7/src/Makefile.am	2015-01-08 23:07:02.000000000 +0900
+++ src/lis-2.1.7/src/Makefile.am	2025-03-16 23:07:21.159647700 +0900
@@ -19,11 +19,12 @@ liblis@LIBSUFFIX@_la_LIBADD = array/liba
                    system/libsystem.la \
                    vector/libvector.la
 if ENABLE_SAAMG
-  liblis@LIBSUFFIX@_la_LIBADD += fortran/amg/libsaamg.la
-  liblis@LIBSUFFIX@_la_LIBADD += fortran/libfortran.la
+  liblis@LIBSUFFIX@_la_LIBADD += fortran/amg/libsaamg.la -lmpi_mpifh
+  liblis@LIBSUFFIX@_la_LIBADD += fortran/libfortran.la -lgfortran
 else
 if ENABLE_FORTRAN
-  liblis@LIBSUFFIX@_la_LIBADD += fortran/libfortran.la
+  liblis@LIBSUFFIX@_la_LIBADD += fortran/libfortran.la -lgfortran
 endif
 endif
 liblis@LIBSUFFIX@_la_SOURCES = 
+liblis@LIBSUFFIX@_la_LDFLAGS = -no-undefined
--- origsrc/lis-2.1.7/test/Makefile.am	2018-05-24 10:59:55.000000000 +0900
+++ src/lis-2.1.7/test/Makefile.am	2025-03-16 23:07:21.159647700 +0900
@@ -5,7 +5,7 @@ AM_LDFLAGS = $(AMDEFS)
 
 EXTRA_DIST = defs test.sh testmat.mtx testmat2.mtx testmat3.mtx testmat4.mtx testmat0.mtx testvec0.mtx
 
-testdir = ${prefix}/share/examples/lis
+testdir = $(libdir)/lis/examples
 
 test_DATA = testmat.mtx testmat2.mtx testmat3.mtx testmat4.mtx testmat0.mtx testvec0.mtx
 
