--- origsrc/lis-1.2.93/src/Makefile.am	2012-08-02 19:40:27.000000000 +0900
+++ src/lis-1.2.93/src/Makefile.am	2012-08-11 16:44:21.092041500 +0900
@@ -16,13 +16,17 @@ liblis@LIBSUFFIX@_la_LIBADD = matrix/lib
                    solver/libsolver.la \
                    esolver/libesolver.la \
                    system/libsystem.la \
-                   precision/libprecision.la 
+                   precision/libprecision.la
 if ENABLE_SAAMG
-  liblis@LIBSUFFIX@_la_LIBADD += fortran/libfortran.la
+  liblis@LIBSUFFIX@_la_LIBADD += fortran/libfortran.la -lgfortran
   liblis@LIBSUFFIX@_la_LIBADD += fortran/amg/libsaamg.la
 else
 if ENABLE_FORTRAN
-  liblis@LIBSUFFIX@_la_LIBADD += fortran/libfortran.la
+  liblis@LIBSUFFIX@_la_LIBADD += fortran/libfortran.la -lgfortran
 endif
 endif
+if ENABLE_OMP
+  liblis@LIBSUFFIX@_la_LIBADD += -lgomp
+endif
 liblis@LIBSUFFIX@_la_SOURCES = 
+liblis@LIBSUFFIX@_la_LDFLAGS = -no-undefined
--- origsrc/lis-1.2.93/test/test.sh	2012-08-06 16:37:54.000000000 +0900
+++ src/lis-1.2.93/test/test.sh	2012-08-11 16:52:29.390869600 +0900
@@ -25,30 +25,30 @@ fi
 
 echo ' '
 echo 'checking linear solvers...'
-$MPIRUN $srcdir/test1 $srcdir/testmat.mtx 0 $srcdir/sol.txt $srcdir/res.txt -omp_num_threads $OMPNUM
+$MPIRUN $top_builddir/test/test1 $srcdir/testmat.mtx 0 $srcdir/sol.txt $srcdir/res.txt -omp_num_threads $OMPNUM
 
 echo 'checking eigensolvers...'
-$MPIRUN $srcdir/etest1 $srcdir/testmat.mtx $srcdir/sol.txt $srcdir/res.txt -omp_num_threads $OMPNUM
+$MPIRUN $top_builddir/test/etest1 $srcdir/testmat.mtx $srcdir/sol.txt $srcdir/res.txt -omp_num_threads $OMPNUM
 
 if test "$enable_fortran" = "yes"; then
   echo 'checking Fortran interface...'
   echo ' '
 #  $MPIRUN $srcdir/test1f $srcdir/testmat.mtx 0 $srcdir/sol.txt $srcdir/res.txt -omp_num_threads $OMPNUM
-  cd $srcdir; 
+  cd $top_builddir/test; 
   $MPIRUN ./test4f
   echo ' '
 fi
 
 if test "$enable_quad" = "yes"; then
   echo 'checking quad precision operations...'
-  $MPIRUN $srcdir/test5 200 2.0 -f double -omp_num_threads $OMPNUM
-  $MPIRUN $srcdir/test5 200 2.0 -f quad -omp_num_threads $OMPNUM
+  $MPIRUN $top_builddir/test/test5 200 2.0 -f double -omp_num_threads $OMPNUM
+  $MPIRUN $top_builddir/test/test5 200 2.0 -f quad -omp_num_threads $OMPNUM
 fi
 
 OMPNUM=1
 if test "$enable_saamg" = "yes"; then
   echo 'checking SAAMG preconditioner...'
-  $MPIRUN $srcdir/test2 10 10 1 $srcdir/sol.txt $srcdir/res.txt -i cg -p saamg -omp_num_threads $OMPNUM
+  $MPIRUN $top_builddir/test/test2 10 10 1 $srcdir/sol.txt $srcdir/res.txt -i cg -p saamg -omp_num_threads $OMPNUM
 #  $MPIRUN $srcdir/test1 $srcdir/add32.mtx $srcdir/sol.txt $srcdir/res.txt -i bicgstab -p saamg -saamg_unsym true -omp_num_threads $OMPNUM
 fi
 
