--- origsrc/libva-1.8.2/configure.ac	2017-05-23 00:15:09.000000000 +0900
+++ src/libva-1.8.2/configure.ac	2017-07-23 20:42:18.998283400 +0900
@@ -119,7 +119,7 @@ LIBVA_LT_CURRENT=libva_lt_current
 LIBVA_LT_REV=libva_lt_revision
 LIBVA_LT_AGE=libva_lt_age
 LIBVA_LT_VERSION="$LIBVA_LT_CURRENT:$LIBVA_LT_REV:$LIBVA_LT_AGE"
-LIBVA_LT_LDFLAGS="-version-info $LIBVA_LT_VERSION"
+LIBVA_LT_LDFLAGS="-no-undefined -version-info $LIBVA_LT_VERSION"
 AC_SUBST(LIBVA_LT_VERSION)
 AC_SUBST(LIBVA_LT_LDFLAGS)
 
--- origsrc/libva-1.8.2/va/sysdeps.h	2017-05-23 00:15:10.000000000 +0900
+++ src/libva-1.8.2/va/sysdeps.h	2017-07-23 20:42:18.998283400 +0900
@@ -48,12 +48,21 @@
 # include <utils/Log.h>
 #endif
 
-#if defined __GNUC__ && defined HAVE_GNUC_VISIBILITY_ATTRIBUTE
+#if defined(__CYGWIN__)
+#if !defined(DLL_HIDDEN)
+#define DLL_HIDDEN __attribute__((visibility("hidden")))
+#endif
+#if !defined(DLL_EXPORT)
+#define DLL_EXPORT __attribute__((visibility("default")))
+#endif
+#else
+#if (defined __GNUC__ && defined HAVE_GNUC_VISIBILITY_ATTRIBUTE)
 # define DLL_HIDDEN __attribute__((visibility("hidden")))
 # define DLL_EXPORT __attribute__((visibility("default")))
 #else
 # define DLL_HIDDEN
 # define DLL_EXPORT
 #endif
+#endif
 
 #endif /* SYSDEPS_H */
--- origsrc/libva-1.8.2/va/va.c	2017-05-23 00:15:10.000000000 +0900
+++ src/libva-1.8.2/va/va.c	2017-07-23 20:42:18.998283400 +0900
@@ -38,7 +38,11 @@
 #include <dlfcn.h>
 #include <unistd.h>
 
+#if defined(__CYGWIN__)
+#define DRIVER_EXTENSION	"_drv_video.dll"
+#else
 #define DRIVER_EXTENSION	"_drv_video.so"
+#endif
 
 #define CTX(dpy) (((VADisplayContextP)dpy)->pDriverContext)
 #define CHECK_DISPLAY(dpy) if( !vaDisplayIsValid(dpy) ) { return VA_STATUS_ERROR_INVALID_DISPLAY; }
--- origsrc/libva-1.8.2/va/va_backend.h	2017-05-23 00:15:10.000000000 +0900
+++ src/libva-1.8.2/va/va_backend.h	2017-07-23 20:42:19.012292200 +0900
@@ -30,7 +30,9 @@
 #define _VA_BACKEND_H_
 
 #include <va/va.h>
+#if !defined(__CYGWIN__)
 #include <linux/videodev2.h>
+#endif
 
 typedef struct VADriverContext *VADriverContextP;
 typedef struct VADisplayContext *VADisplayContextP;
--- origsrc/libva-1.8.2/va/va_compat.c	2017-05-23 00:15:10.000000000 +0900
+++ src/libva-1.8.2/va/va_compat.c	2017-07-23 20:42:19.012292200 +0900
@@ -40,4 +40,6 @@ vaCreateSurfaces_0_32_0(
     return vaCreateSurfaces(dpy, format, width, height, surfaces, num_surfaces,
                             NULL, 0);
 }
+#if !defined(__CYGWIN__)
 VA_CPP_HELPER_ALIAS(vaCreateSurfaces, 0,32,0);
+#endif
--- origsrc/libva-1.8.2/va/va_trace.c	2017-05-23 00:15:10.000000000 +0900
+++ src/libva-1.8.2/va/va_trace.c	2017-07-23 20:42:19.027079400 +0900
@@ -46,7 +46,12 @@
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/time.h>
+#if defined(__CYGWIN__)
+#define syscall(x) getpid()
+#else
 #include <sys/syscall.h>
+#endif
 #include <pthread.h>
 #include <unistd.h>
 #include <time.h>
