--- origsrc/libva-2.22.0/va/va.c	2024-06-20 15:49:27.000000000 +0900
+++ src/libva-2.22.0/va/va.c	2025-01-07 15:19:21.543744200 +0900
@@ -44,7 +44,11 @@
 #else
 #include <dlfcn.h>
 #include <unistd.h>
+#if defined(_CYGWIN__)
+#define DRIVER_EXTENSION    "_drv_video.dll"
+#else
 #define DRIVER_EXTENSION    "_drv_video.so"
+#endif
 #define DRIVER_PATH_STRING  "%s/%s%s"
 #define ENV_VAR_SEPARATOR ":"
 #endif
--- origsrc/libva-2.22.0/va/va_compat.c	2024-06-20 15:49:27.000000000 +0900
+++ src/libva-2.22.0/va/va_compat.c	2025-01-07 15:20:17.838347700 +0900
@@ -40,6 +40,6 @@ vaCreateSurfaces_0_32_0(
     return vaCreateSurfaces(dpy, format, width, height, surfaces, num_surfaces,
                             NULL, 0);
 }
-#if !defined(_WIN32)
+#if !defined(_WIN32) && !defined(__CYGWIN__)
 VA_CPP_HELPER_ALIAS(vaCreateSurfaces, 0, 32, 0);
 #endif
