--- origsrc/libva-1.0.13/configure.ac	2011-05-31 11:10:12.000000000 +0900
+++ src/libva-1.0.13/configure.ac	2011-06-10 09:50:09.297655400 +0900
@@ -57,7 +57,7 @@ LIBVA_LT_CURRENT=libva_lt_current
 LIBVA_LT_REV=libva_lt_revision
 LIBVA_LT_AGE=libva_lt_age
 LIBVA_LT_VERSION="$LIBVA_LT_CURRENT:$LIBVA_LT_REV:$LIBVA_LT_AGE"
-LIBVA_LT_LDFLAGS="-version-info $LIBVA_LT_VERSION"
+LIBVA_LT_LDFLAGS="-version-info $LIBVA_LT_VERSION -no-undefined"
 AC_SUBST(LIBVA_LT_VERSION)
 AC_SUBST(LIBVA_LT_LDFLAGS)
 
--- origsrc/libva-1.0.13/dummy_drv_video/Makefile.am	2011-05-31 11:10:12.000000000 +0900
+++ src/libva-1.0.13/dummy_drv_video/Makefile.am	2011-06-11 12:37:52.745034000 +0900
@@ -26,7 +26,7 @@ dummy_drv_video_la_LDFLAGS = -module -av
 dummy_drv_video_la_LIBADD = $(top_builddir)/va/$(libvabackendlib)
 dummy_drv_video_la_DEPENDENCIES = $(top_builddir)/va/$(libvabackendlib)
 
-AM_CFLAGS = -I$(top_srcdir)/va -I$(top_srcdir)/../../include/external/ -I$(top_srcdir)/../../include/kmd
+AM_CFLAGS = -I$(top_srcdir) -I$(top_srcdir)/va -I$(top_srcdir)/../../include/external/ -I$(top_srcdir)/../../include/kmd
 dummy_drv_video_la_SOURCES = dummy_drv_video.c object_heap.c
 
 noinst_HEADERS = dummy_drv_video.h object_heap.h
--- origsrc/libva-1.0.13/test/basic/Makefile.am	2011-05-31 11:10:12.000000000 +0900
+++ src/libva-1.0.13/test/basic/Makefile.am	2011-06-11 12:09:36.291226200 +0900
@@ -23,7 +23,7 @@
 bin_PROGRAMS = test_01 test_02 test_03 test_04 test_05 test_06 \
 		test_07 test_08 test_09 test_10 test_11
 
-AM_CFLAGS = -I$(top_srcdir)/va -I$(top_srcdir)/src/x11
+AM_CFLAGS = -I$(top_srcdir) -I$(top_srcdir)/va -I$(top_srcdir)/src/x11
 
 TEST_LIBS = $(top_builddir)/va/$(libvabackendlib)
 
--- origsrc/libva-1.0.13/test/decode/Makefile.am	2011-05-31 11:10:12.000000000 +0900
+++ src/libva-1.0.13/test/decode/Makefile.am	2011-06-11 12:20:29.151957800 +0900
@@ -22,7 +22,7 @@
 
 bin_PROGRAMS = mpeg2vldemo
 
-AM_CFLAGS = -I$(top_srcdir)/va -I$(top_srcdir)/src/x11
+AM_CXXFLAGS = -I$(top_srcdir) -I$(top_srcdir)/va -I$(top_srcdir)/src/x11
 
 TEST_LIBS = $(top_builddir)/va/$(libvabackendlib)
 
--- origsrc/libva-1.0.13/test/encode/Makefile.am	2011-05-31 11:10:12.000000000 +0900
+++ src/libva-1.0.13/test/encode/Makefile.am	2011-06-11 12:22:43.655789900 +0900
@@ -22,7 +22,7 @@
 
 bin_PROGRAMS = h264encode avcenc
 
-AM_CFLAGS = -I$(top_srcdir)/va -I$(top_srcdir)/src/x11
+AM_CFLAGS = -I$(top_srcdir) -I$(top_srcdir)/va -I$(top_srcdir)/src/x11
 
 TEST_LIBS = $(top_builddir)/va/$(libvabackendlib)
 
--- origsrc/libva-1.0.13/test/encode/avcenc.c	2011-05-31 11:10:12.000000000 +0900
+++ src/libva-1.0.13/test/encode/avcenc.c	2011-06-11 12:31:17.917299600 +0900
@@ -16,7 +16,9 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
+#if !defined(__CYGWIN__)
 #include <pciaccess.h>
+#endif
 #include <assert.h>
 #include <time.h>
 
--- origsrc/libva-1.0.13/test/putsurface/Makefile.am	2011-05-31 11:10:12.000000000 +0900
+++ src/libva-1.0.13/test/putsurface/Makefile.am	2011-06-11 12:33:04.130026000 +0900
@@ -22,7 +22,7 @@
 
 bin_PROGRAMS = putsurface
 
-AM_CFLAGS = -I$(top_srcdir)/va -I$(top_srcdir)/src/x11
+AM_CFLAGS = -I$(top_srcdir) -I$(top_srcdir)/va -I$(top_srcdir)/src/x11
 
 TEST_LIBS = $(top_builddir)/va/$(libvabackendlib) -lpthread
 
--- origsrc/libva-1.0.13/test/vainfo/Makefile.am	2011-05-31 11:10:12.000000000 +0900
+++ src/libva-1.0.13/test/vainfo/Makefile.am	2011-06-11 12:35:23.540488400 +0900
@@ -23,7 +23,7 @@
 
 bin_PROGRAMS = vainfo
 
-AM_CFLAGS = -I$(top_srcdir)/va -I$(top_srcdir)/test/basic -I$(top_srcdir)/src/x11
+AM_CFLAGS = -I$(top_srcdir) -I$(top_srcdir)/va -I$(top_srcdir)/test/basic -I$(top_srcdir)/src/x11
 
 vainfo_LDADD = $(top_builddir)/va/$(libvacorelib) $(top_builddir)/va/$(libvabackendlib)
 vainfo_DEPENDENCIES =  $(top_builddir)/va/$(libvacorelib) $(top_builddir)/va/$(libvabackendlib)
--- origsrc/libva-1.0.13/va/Makefile.am	2011-05-31 11:10:12.000000000 +0900
+++ src/libva-1.0.13/va/Makefile.am	2011-06-11 11:58:45.864292100 +0900
@@ -21,7 +21,7 @@
 # SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 INCLUDES = \
-	$(LIBVA_CFLAGS) -I$(top_srcdir)/va/x11 \
+	$(LIBVA_CFLAGS) -I$(top_srcdir) -I$(top_srcdir)/va/x11 \
 	-DVA_DRIVERS_PATH="\"$(LIBVA_DRIVERS_PATH)\""
 
 LDADD = \
--- origsrc/libva-1.0.13/va/egl/Makefile.am	2011-05-31 11:10:12.000000000 +0900
+++ src/libva-1.0.13/va/egl/Makefile.am	2011-06-11 11:56:43.741382800 +0900
@@ -20,7 +20,7 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-AM_CFLAGS = -DLINUX -I$(top_srcdir)/va -I$(top_srcdir)/va/x11
+AM_CFLAGS = -DLINUX -I$(top_srcdir) -I$(top_srcdir)/va -I$(top_srcdir)/va/x11
 
 source_c = \
 	va_egl.c		
--- origsrc/libva-1.0.13/va/glx/Makefile.am	2011-05-31 11:10:12.000000000 +0900
+++ src/libva-1.0.13/va/glx/Makefile.am	2011-06-11 11:54:44.662036200 +0900
@@ -20,7 +20,7 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-AM_CFLAGS = -DLINUX -I$(top_srcdir)/va -I$(top_srcdir)/va/x11
+AM_CFLAGS = -DLINUX -I$(top_srcdir) -I$(top_srcdir)/va -I$(top_srcdir)/va/x11
 
 source_c = \
 	va_glx.c		\
--- origsrc/libva-1.0.13/va/va.c	2011-05-31 11:10:12.000000000 +0900
+++ src/libva-1.0.13/va/va.c	2011-06-11 12:02:46.151474200 +0900
@@ -205,7 +205,7 @@ static VAStatus va_openDriver(VADisplay
         strncat( driver_path, DRIVER_EXTENSION, strlen(DRIVER_EXTENSION) );
         
         va_infoMessage("Trying to open %s\n", driver_path);
-#ifndef ANDROID
+#if !defined(ANDROID) && !defined(__CYGWIN__)
         handle = dlopen( driver_path, RTLD_NOW | RTLD_GLOBAL | RTLD_NODELETE );
 #else
         handle = dlopen( driver_path, RTLD_NOW| RTLD_GLOBAL);
--- origsrc/libva-1.0.13/va/va_backend.h	2011-05-31 11:10:12.000000000 +0900
+++ src/libva-1.0.13/va/va_backend.h	2011-06-11 11:53:39.915667600 +0900
@@ -33,7 +33,9 @@
 #ifndef ANDROID
 #include <X11/Xlib.h>
 #endif
+#if defined(__linux__)
 #include <linux/videodev2.h>
+#endif
 
 typedef struct VADriverContext *VADriverContextP;
 typedef struct VADisplayContext *VADisplayContextP;
--- origsrc/libva-1.0.13/va/va_backend_tpi.h	2011-05-31 11:10:12.000000000 +0900
+++ src/libva-1.0.13/va/va_backend_tpi.h	2011-06-11 12:04:58.892346200 +0900
@@ -32,7 +32,9 @@
 #include <va/va.h>
 #include <va/va_backend.h>
 
+#if defined(__linux__)
 #include <linux/videodev2.h>
+#endif
 
 struct VADriverVTableTPI
 {
--- origsrc/libva-1.0.13/va/x11/Makefile.am	2011-05-31 11:10:12.000000000 +0900
+++ src/libva-1.0.13/va/x11/Makefile.am	2011-06-11 11:48:28.803039200 +0900
@@ -20,7 +20,7 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-AM_CFLAGS = -DLINUX -I$(top_srcdir)/va $(DRM_CFLAGS)
+AM_CFLAGS = -DLINUX -I$(top_srcdir) -I$(top_srcdir)/va $(DRM_CFLAGS)
 
 noinst_LTLIBRARIES = libva_x11.la	
 
