--- origsrc/libinjection-3.10.0/CMakeLists.txt	1970-01-01 09:00:00.000000000 +0900
+++ src/libinjection-3.10.0/CMakeLists.txt	2026-01-26 08:08:55.966757500 +0900
@@ -0,0 +1,97 @@
+cmake_minimum_required(VERSION 3.10)
+project(libinjection VERSION 3.10.0 LANGUAGES C)
+
+set(LIBNAME injection)
+
+set(SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src)
+
+set(HEADERS
+    ${SRC_DIR}/libinjection.h
+    ${SRC_DIR}/libinjection_sqli.h
+    ${SRC_DIR}/libinjection_sqli_data.h
+    ${SRC_DIR}/libinjection_html5.h
+    ${SRC_DIR}/libinjection_xss.h
+)
+
+set(SOURCES
+    ${SRC_DIR}/libinjection_sqli.c
+    ${SRC_DIR}/libinjection_html5.c
+    ${SRC_DIR}/libinjection_xss.c
+)
+
+add_library(${LIBNAME} SHARED ${SOURCES})
+target_include_directories(${LIBNAME} PUBLIC
+    $<BUILD_INTERFACE:${SRC_DIR}>
+    $<INSTALL_INTERFACE:include>
+)
+set_target_properties(${LIBNAME} PROPERTIES
+    VERSION ${PROJECT_VERSION}
+    SOVERSION ${PROJECT_VERSION_MAJOR}
+    PUBLIC_HEADER "${HEADERS}"
+)
+
+add_executable(sqli ${SRC_DIR}/sqli_cli.c)
+target_link_libraries(sqli PRIVATE ${LIBNAME})
+
+add_executable(html5 ${SRC_DIR}/html5_cli.c)
+target_link_libraries(html5 PRIVATE ${LIBNAME})
+
+add_executable(fptool ${SRC_DIR}/fptool.c)
+target_link_libraries(fptool PRIVATE ${LIBNAME})
+
+add_executable(reader ${SRC_DIR}/reader.c)
+target_link_libraries(reader PRIVATE ${LIBNAME})
+
+add_executable(testdriver ${SRC_DIR}/testdriver.c)
+target_link_libraries(testdriver PRIVATE ${LIBNAME})
+
+add_executable(testspeedsqli ${SRC_DIR}/test_speed_sqli.c)
+target_link_libraries(testspeedsqli PRIVATE ${LIBNAME})
+
+add_executable(testspeedxss ${SRC_DIR}/test_speed_xss.c)
+target_link_libraries(testspeedxss PRIVATE ${LIBNAME})
+
+include(GNUInstallDirs)
+
+install(TARGETS ${LIBNAME}
+    EXPORT ${PROJECT_NAME}Targets
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+)
+
+install(EXPORT ${PROJECT_NAME}Targets
+    FILE ${PROJECT_NAME}Targets.cmake
+    NAMESPACE ${PROJECT_NAME}::
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}
+)
+
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file(
+    "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+    VERSION ${PROJECT_VERSION}
+    COMPATIBILITY SameMajorVersion
+)
+
+configure_file(
+    "${CMAKE_CURRENT_SOURCE_DIR}/${PROJECT_NAME}Config.cmake.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+    @ONLY
+)
+
+install(FILES
+    "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+    "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}
+)
+
+configure_file(
+    "${CMAKE_CURRENT_SOURCE_DIR}/${LIBNAME}.pc.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/${LIBNAME}.pc"
+    @ONLY
+)
+
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${LIBNAME}.pc"
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
+)
--- origsrc/libinjection-3.10.0/injection.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libinjection-3.10.0/injection.pc.in	2026-01-26 08:01:39.682309700 +0900
@@ -0,0 +1,10 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
+
+Name: libinjection
+Description: SQL / SQLI tokenizer parser analyzer
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -linjection
+Cflags: -I${includedir}
--- origsrc/libinjection-3.10.0/libinjectionConfig.cmake.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libinjection-3.10.0/libinjectionConfig.cmake.in	2026-01-26 08:01:39.684442700 +0900
@@ -0,0 +1,5 @@
+@PACKAGE_INIT@
+
+include("${CMAKE_CURRENT_LIST_DIR}/libinjectionTargets.cmake")
+
+check_required_components(libinjection)
--- origsrc/libinjection-3.10.0/src/html5_cli.c	2017-05-22 05:44:26.000000000 +0900
+++ src/libinjection-3.10.0/src/html5_cli.c	2026-01-26 08:11:13.065353300 +0900
@@ -96,7 +96,7 @@ const char* h5_type_to_string(enum html5
     case TAG_COMMENT: return "TAG_COMMENT";
     case DOCTYPE: return "DOCTYPE";
     default:
-        assert(0);
+        assert(0); return "UNKNOWN";
     }
 }
 
