--- origsrc/websocketpp-0.8.1/cmake/CMakeHelpers.cmake	2018-07-16 21:40:53.000000000 +0900
+++ src/websocketpp-0.8.1/cmake/CMakeHelpers.cmake	2019-01-01 07:54:57.852537900 +0900
@@ -73,7 +73,7 @@ endmacro ()
 
 # Finalize target for all types
 macro (final_target)
-    if ("${TARGET_LIB_TYPE}" STREQUAL "EXECUTABLE")
+    if (NOT CYGWIN AND "${TARGET_LIB_TYPE}" STREQUAL "EXECUTABLE")
         install (TARGETS ${TARGET_NAME}
                  RUNTIME DESTINATION "bin"
                  CONFIGURATIONS ${CMAKE_CONFIGURATION_TYPES})
