--- origsrc/websocketpp-0.8.2/cmake/CMakeHelpers.cmake	2020-04-20 03:25:17.000000000 +0900
+++ src/websocketpp-0.8.2/cmake/CMakeHelpers.cmake	2020-08-08 12:54:56.153678700 +0900
@@ -73,7 +73,7 @@ endmacro ()
 
 # Finalize target for all types
 macro (final_target)
-    if ("${TARGET_LIB_TYPE}" STREQUAL "EXECUTABLE")
+    if (NOT CYGWIN AND "${TARGET_LIB_TYPE}" STREQUAL "EXECUTABLE")
         install (TARGETS ${TARGET_NAME}
                  RUNTIME DESTINATION "bin"
                  CONFIGURATIONS ${CMAKE_CONFIGURATION_TYPES})
