--- origsrc/gdome2-0.8.1/libgdome/Makefile.am	2002-07-16 23:15:56.000000000 +0900
+++ src/gdome2-0.8.1/libgdome/Makefile.am	2026-05-22 19:09:24.149352600 +0900
@@ -36,7 +36,7 @@ libgdome_la_LIBADD = 						\
         $(GLIB_LIBS)                                            \
 	$(NULL)
         
-libgdome_la_LDFLAGS = -version-info @GDOME_VERSION_INFO@
+libgdome_la_LDFLAGS = -no-undefined -version-info @GDOME_VERSION_INFO@
 
 # what is this include dir ?
 libgdomeincdir = $(includedir)/libgdome
