--- origsrc/libpinyin-0.5.92/src/Makefile.am	2011-11-30 15:17:30.000000000 +0900
+++ src/libpinyin-0.5.92/src/Makefile.am	2012-04-12 15:07:55.305685200 +0900
@@ -44,10 +44,10 @@ noinst_LTLIBRARIES	= libpinyin_internal.
 
 libpinyin_la_SOURCES	= pinyin.cpp
 
-libpinyin_la_LIBADD	= storage/libstorage.la lookup/liblookup.la
+libpinyin_la_LIBADD	= storage/libstorage.la lookup/liblookup.la $(GLIB2_LIBS)
 
 libpinyin_la_LDFLAGS	= -Wl,--version-script=$(srcdir)/libpinyin.ver \
-			  -version-info "0:3:0"
+			  -version-info "0:3:0" -no-undefined
 
 libpinyin_internal_la_SOURCES	= pinyin_internal.cpp
 
@@ -56,4 +56,4 @@ libpinyin_internal_la_LIBADD	= storage/l
 
 ## Note:
 ## As libpinyin internal interface will change, only provides static library
-##   to catch errors when compiling instead of running.
\ No newline at end of file
+##   to catch errors when compiling instead of running.
--- origsrc/libpinyin-0.5.92/tests/storage/Makefile.am	2012-03-23 11:48:02.000000000 +0900
+++ src/libpinyin-0.5.92/tests/storage/Makefile.am	2012-04-12 15:08:40.899435200 +0900
@@ -70,3 +70,5 @@ test_parser2_LDADD      = ../../src/libp
 test_chewing_table_SOURCES    = test_chewing_table.cpp
 
 test_chewing_table_LDADD      = ../../src/libpinyin_internal.la @GLIB2_LDFLAGS@
+
+TESTS = test_flexible_ngram test_ngram test_phrase_index test_phrase_index_logger
