--- origsrc/libpinyin-2.8.1/src/Makefile.am	2023-01-16 13:04:32.000000000 +0900
+++ src/libpinyin-2.8.1/src/Makefile.am	2023-01-22 00:43:19.152256000 +0900
@@ -91,6 +91,8 @@ libpinyin_la_LDFLAGS = -Wl,--version-scr
 			  -version-info @LT_VERSION_INFO@
 endif
 
+libpinyin_la_LDFLAGS += -no-undefined
+
 if ENABLE_LIBZHUYIN
 lib_LTLIBRARIES     += libzhuyin.la
 
@@ -107,6 +109,9 @@ else
 libzhuyin_la_LDFLAGS = -Wl,--version-script=$(srcdir)/libzhuyin.ver \
 			  -version-info @LT_VERSION_INFO@
 endif
+
+libzhuyin_la_LDFLAGS += -no-undefined
+
 endif
 
 libpinyin_internal_a_SOURCES = pinyin_internal.cpp
