--- origsrc/libpinyin-0.8.0/src/Makefile.am	2012-10-17 15:10:34.000000000 +0900
+++ src/libpinyin-0.8.0/src/Makefile.am	2012-11-24 17:01:07.983083200 +0900
@@ -44,10 +44,10 @@ noinst_LTLIBRARIES	= libpinyin_internal.
 
 libpinyin_la_SOURCES	= pinyin.cpp
 
-libpinyin_la_LIBADD	= storage/libstorage.la lookup/liblookup.la
+libpinyin_la_LIBADD	= storage/libstorage.la lookup/liblookup.la $(GLIB2_LIBS)
 
 libpinyin_la_LDFLAGS	= -Wl,--version-script=$(srcdir)/libpinyin.ver \
-			  -version-info @LT_VERSION_INFO@
+			  -version-info @LT_VERSION_INFO@ -no-undefined
 
 libpinyin_internal_la_SOURCES	= pinyin_internal.cpp
 
