--- origsrc/libpinyin-2.4.92/src/Makefile.am	2020-10-15 14:21:39.000000000 +0900
+++ src/libpinyin-2.4.92/src/Makefile.am	2020-11-23 00:34:51.498617800 +0900
@@ -81,7 +81,7 @@ libpinyin_la_SOURCES = $(pinyin_SOURCES)
 
 libpinyin_la_LIBADD	= @GLIB2_LIBS@
 
-libpinyin_la_LDFLAGS = -Wl,--version-script=$(srcdir)/libpinyin.ver \
+libpinyin_la_LDFLAGS = -no-undefined -Wl,--version-script=$(srcdir)/libpinyin.ver \
 			  -version-info @LT_VERSION_INFO@
 
 if ENABLE_LIBZHUYIN
@@ -91,7 +91,7 @@ libzhuyin_la_SOURCES = $(pinyin_SOURCES)
 
 libzhuyin_la_LIBADD	= @GLIB2_LIBS@
 
-libzhuyin_la_LDFLAGS = -Wl,--version-script=$(srcdir)/libzhuyin.ver \
+libzhuyin_la_LDFLAGS = -no-undefined -Wl,--version-script=$(srcdir)/libzhuyin.ver \
 			  -version-info @LT_VERSION_INFO@
 endif
 
