--- origsrc/libpinyin-0.9.92/src/Makefile.am	2013-02-19 13:32:34.000000000 +0900
+++ src/libpinyin-0.9.92/src/Makefile.am	2013-06-15 18:37:01.595406300 +0900
@@ -47,7 +47,7 @@ libpinyin_la_SOURCES	= pinyin.cpp
 libpinyin_la_LIBADD	= storage/libstorage.la lookup/liblookup.la @GLIB2_LIBS@
 
 libpinyin_la_LDFLAGS	= -Wl,--version-script=$(srcdir)/libpinyin.ver \
-			  -version-info @LT_VERSION_INFO@
+			  -version-info @LT_VERSION_INFO@ -no-undefined
 
 libpinyin_internal_la_SOURCES	= pinyin_internal.cpp
 
