--- origsrc/bullet-2.77.2243svn/Demos/AllBulletDemos/Makefile.am	2010-12-09 20:03:40.000000000 +0900
+++ src/bullet-2.77.2243svn/Demos/AllBulletDemos/Makefile.am	2010-12-10 11:34:21.619950600 +0900
@@ -1,33 +1,33 @@
 noinst_PROGRAMS=AllBulletDemo
 
 AllBulletDemo_SOURCES=\
-		      	../SoftDemo/SoftDemo.cpp \
-			../ConstraintDemo/ConstraintDemo.cpp \
-			../ForkLiftDemo/ForkLiftDemo.cpp \
-			../RagdollDemo/RagdollDemo.cpp \
-			../BasicDemo/BasicDemo.cpp \
-			../CcdPhysicsDemo/CcdPhysicsDemo.cpp \
-			../SliderConstraintDemo/SliderConstraintDemo.cpp \
-			../ConcaveRaycastDemo/ConcaveRaycastDemo.cpp \
-			../Raytracer/Raytracer.cpp \
-			../BspDemo/BspDemo.cpp \
-			../BspDemo/BspLoader.cpp \
-			../BspDemo/BspConverter.cpp \
-			../ConcaveConvexcastDemo/ConcaveConvexcastDemo.cpp \
-			../ConvexDecompositionDemo/ConvexDecompositionDemo.cpp \
-			../GimpactTestDemo/GimpactTestDemo.cpp \
-			../GjkConvexCastDemo/LinearConvexCastDemo.cpp \
-			../ConcaveDemo/ConcavePhysicsDemo.cpp \
-			../DynamicControlDemo/MotorDemo.cpp \
+		      	$(top_srcdir)/Demos/SoftDemo/SoftDemo.cpp \
+			$(top_srcdir)/Demos/ConstraintDemo/ConstraintDemo.cpp \
+			$(top_srcdir)/Demos/ForkLiftDemo/ForkLiftDemo.cpp \
+			$(top_srcdir)/Demos/RagdollDemo/RagdollDemo.cpp \
+			$(top_srcdir)/Demos/BasicDemo/BasicDemo.cpp \
+			$(top_srcdir)/Demos/CcdPhysicsDemo/CcdPhysicsDemo.cpp \
+			$(top_srcdir)/Demos/SliderConstraintDemo/SliderConstraintDemo.cpp \
+			$(top_srcdir)/Demos/ConcaveRaycastDemo/ConcaveRaycastDemo.cpp \
+			$(top_srcdir)/Demos/Raytracer/Raytracer.cpp \
+			$(top_srcdir)/Demos/BspDemo/BspDemo.cpp \
+			$(top_srcdir)/Demos/BspDemo/BspLoader.cpp \
+			$(top_srcdir)/Demos/BspDemo/BspConverter.cpp \
+			$(top_srcdir)/Demos/ConcaveConvexcastDemo/ConcaveConvexcastDemo.cpp \
+			$(top_srcdir)/Demos/ConvexDecompositionDemo/ConvexDecompositionDemo.cpp \
+			$(top_srcdir)/Demos/GimpactTestDemo/GimpactTestDemo.cpp \
+			$(top_srcdir)/Demos/GjkConvexCastDemo/LinearConvexCastDemo.cpp \
+			$(top_srcdir)/Demos/ConcaveDemo/ConcavePhysicsDemo.cpp \
+			$(top_srcdir)/Demos/DynamicControlDemo/MotorDemo.cpp \
 			DemoEntries.cpp \
 			DemoEntries.h\
 			Main.cpp
 AllBulletDemo_CXXFLAGS=\
-		       	-I@top_builddir@/src \
-		       	-I@top_builddir@/Extras/GIMPACTUtils \
-		       	-I@top_builddir@/Extras/ConvexDecomposition \
-		       	-I@top_builddir@/Extras \
-			-I@top_builddir@/Demos/OpenGL \
-			-I@top_builddir@/Demos/SoftDemo \
+		       	-I$(top_srcdir)/src \
+		       	-I$(top_srcdir)/Extras/GIMPACTUtils \
+		       	-I$(top_srcdir)/Extras/ConvexDecomposition \
+		       	-I$(top_srcdir)/Extras \
+			-I$(top_srcdir)/Demos/OpenGL \
+			-I$(top_srcdir)/Demos/SoftDemo \
 			$(CXXFLAGS)
 AllBulletDemo_LDADD=-L../OpenGL -lbulletopenglsupport -L../../src -L../../Extras -lgimpactutils -lconvexdecomposition -lBulletSoftBody -lBulletDynamics -lBulletCollision -lLinearMath -lglui @opengl_LIBS@
--- origsrc/bullet-2.77.2243svn/Demos/BasicDemo/Makefile.am	2010-12-09 20:03:32.000000000 +0900
+++ src/bullet-2.77.2243svn/Demos/BasicDemo/Makefile.am	2010-12-09 20:39:22.527372500 +0900
@@ -1,5 +1,5 @@
 noinst_PROGRAMS=BasicDemo
 
 BasicDemo_SOURCES=BasicDemo.cpp BasicDemo.h main.cpp
-BasicDemo_CXXFLAGS=-I@top_builddir@/src -I@top_builddir@/Demos/OpenGL $(CXXFLAGS)
+BasicDemo_CXXFLAGS=-I$(top_srcdir)/src -I$(top_srcdir)/Demos/OpenGL $(CXXFLAGS)
 BasicDemo_LDADD=-L../OpenGL -lbulletopenglsupport -L../../src -lBulletDynamics -lBulletCollision -lLinearMath @opengl_LIBS@
--- origsrc/bullet-2.77.2243svn/Demos/CcdPhysicsDemo/Makefile.am	2010-12-09 20:03:38.000000000 +0900
+++ src/bullet-2.77.2243svn/Demos/CcdPhysicsDemo/Makefile.am	2010-12-09 20:39:47.543343700 +0900
@@ -1,5 +1,5 @@
 noinst_PROGRAMS=CcdPhysicsDemo
 
 CcdPhysicsDemo_SOURCES=CcdPhysicsDemo.cpp CcdPhysicsDemo.h main.cpp
-CcdPhysicsDemo_CXXFLAGS=-I@top_builddir@/src -I@top_builddir@/Demos/OpenGL $(CXXFLAGS)
+CcdPhysicsDemo_CXXFLAGS=-I$(top_srcdir)/src -I$(top_srcdir)/Demos/OpenGL $(CXXFLAGS)
 CcdPhysicsDemo_LDADD=-L../OpenGL -lbulletopenglsupport -L../../src -lBulletDynamics -lBulletCollision -lLinearMath @opengl_LIBS@
--- origsrc/bullet-2.77.2243svn/Demos/MultiThreadedDemo/Makefile.am	2010-12-09 20:03:38.000000000 +0900
+++ src/bullet-2.77.2243svn/Demos/MultiThreadedDemo/Makefile.am	2010-12-11 00:41:44.688100800 +0900
@@ -1,5 +1,5 @@
 noinst_PROGRAMS=MultiThreadedDemo
 
 MultiThreadedDemo_SOURCES=MultiThreadedDemo.cpp MultiThreadedDemo.h main.cpp
-MultiThreadedDemo_CXXFLAGS=-I@top_builddir@/src -I@top_builddir@/Demos/OpenGL -I@top_builddir@/Extras  $(CXXFLAGS)
-MultiThreadedDemo_LDADD=-L../OpenGL -L../../Extras  -L../../src  -lbulletmultithreaded -lbulletopenglsupport -lbulletdynamics -lbulletcollision -lbulletmath @opengl_LIBS@ -lpthread
+MultiThreadedDemo_CXXFLAGS=-I$(top_srcdir)/src -I$(top_srcdir)/Demos/OpenGL -I$(top_srcdir)/Extras  $(CXXFLAGS)
+MultiThreadedDemo_LDADD=-L../OpenGL -L../../Extras  -L../../src  -lbulletmultithreaded -lbulletopenglsupport -lbulletdynamics -lbulletcollision -lLinearMath @opengl_LIBS@ -lpthread
--- origsrc/bullet-2.77.2243svn/Demos/OpenGL/Makefile.am	2010-12-09 20:03:31.000000000 +0900
+++ src/bullet-2.77.2243svn/Demos/OpenGL/Makefile.am	2010-12-09 20:38:30.152060500 +0900
@@ -9,4 +9,4 @@ libbulletopenglsupport_a_SOURCES	= \
 	 DebugCastResult.h  GLDebugDrawer.cpp   \
 	GL_ShapeDrawer.h    GlutStuff.cpp       RenderTexture.h
 
-INCLUDES=-I../../src
+INCLUDES=-I$(top_srcdir)/src
--- origsrc/bullet-2.77.2243svn/Demos/SoftDemo/Makefile.am	2010-12-09 20:03:36.000000000 +0900
+++ src/bullet-2.77.2243svn/Demos/SoftDemo/Makefile.am	2010-12-09 20:40:09.274591700 +0900
@@ -1,5 +1,5 @@
 noinst_PROGRAMS=SoftDemo
 
 SoftDemo_SOURCES=SoftDemo.cpp SoftDemo.h main.cpp
-SoftDemo_CXXFLAGS=-I@top_builddir@/src -I@top_builddir@/Demos/OpenGL $(CXXFLAGS)
+SoftDemo_CXXFLAGS=-I$(top_srcdir)/src -I$(top_srcdir)/Demos/OpenGL $(CXXFLAGS)
 SoftDemo_LDADD=-L../OpenGL -lbulletopenglsupport -L../../src -lBulletSoftBody -lBulletDynamics -lBulletCollision -lLinearMath @opengl_LIBS@
--- origsrc/bullet-2.77.2243svn/Demos/TerrainDemo/Makefile.am	2010-12-09 20:03:40.000000000 +0900
+++ src/bullet-2.77.2243svn/Demos/TerrainDemo/Makefile.am	2010-12-09 20:40:31.406415700 +0900
@@ -2,5 +2,5 @@
 noinst_PROGRAMS=TerrainDemo
 
 TerrainDemo_SOURCES=TerrainDemo.cpp TerrainDemo.h main.cpp
-TerrainDemo_CXXFLAGS=-I@top_builddir@/src -I@top_builddir@/Demos/OpenGL $(CXXFLAGS)
+TerrainDemo_CXXFLAGS=-I$(top_srcdir)/src -I$(top_srcdir)/Demos/OpenGL $(CXXFLAGS)
 TerrainDemo_LDADD=-L../OpenGL -lbulletopenglsupport -L../../src -lBulletDynamics -lBulletCollision -lLinearMath @opengl_LIBS@
--- origsrc/bullet-2.77.2243svn/Demos/VehicleDemo/Makefile.am	2010-12-09 20:03:36.000000000 +0900
+++ src/bullet-2.77.2243svn/Demos/VehicleDemo/Makefile.am	2010-12-09 20:40:57.624114900 +0900
@@ -1,5 +1,5 @@
 noinst_PROGRAMS=VehicleDemo
 
 VehicleDemo_SOURCES=VehicleDemo.cpp VehicleDemo.h main.cpp
-VehicleDemo_CXXFLAGS=-I@top_builddir@/src -I@top_builddir@/Demos/OpenGL $(CXXFLAGS)
+VehicleDemo_CXXFLAGS=-I$(top_srcdir)/src -I$(top_srcdir)/Demos/OpenGL $(CXXFLAGS)
 VehicleDemo_LDADD=-L../OpenGL -lbulletopenglsupport -L../../src -lBulletDynamics -lBulletCollision -lLinearMath @opengl_LIBS@
--- origsrc/bullet-2.77.2243svn/Extras/Makefile.am	2010-12-09 20:03:27.000000000 +0900
+++ src/bullet-2.77.2243svn/Extras/Makefile.am	2010-12-09 20:38:30.342334100 +0900
@@ -41,7 +41,7 @@ libglui_a_SOURCES =\
 		glui/arcball.h\
 		glui/algebra3.h
 
-libconvexdecomposition_a_CXXFLAGS = ${CXXFLAGS} -IConvexDecomposition/ -I../src
+libconvexdecomposition_a_CXXFLAGS = ${CXXFLAGS} -I$(top_srcdir)/Extras/ConvexDecomposition/ -I$(top_srcdir)/src
 libconvexdecomposition_a_SOURCES =\
 		ConvexDecomposition/concavity.cpp\
 		ConvexDecomposition/ConvexDecomposition.cpp\
@@ -73,6 +73,6 @@ libconvexdecomposition_a_SOURCES =\
 		ConvexDecomposition/bestfitobb.h\
 		ConvexDecomposition/meshvolume.h
 
-libgimpactutils_a_CXXFLAGS = ${CXXFLAGS}  -I../src -IGIMPACTUtils -IConvexDecomposition
+libgimpactutils_a_CXXFLAGS = ${CXXFLAGS}  -I$(top_srcdir)/src -I$(top_srcdir)/Extras/GIMPACTUtils -I$(top_srcdir)/Extras/ConvexDecomposition
 libgimpactutils_a_SOURCES = GIMPACTUtils/btGImpactConvexDecompositionShape.cpp GIMPACTUtils/btGImpactConvexDecompositionShape.h
 
--- origsrc/bullet-2.77.2243svn/configure.ac	2010-12-09 20:03:50.000000000 +0900
+++ src/bullet-2.77.2243svn/configure.ac	2010-12-09 20:38:30.362362900 +0900
@@ -16,14 +16,19 @@ AC_CONFIG_SRCDIR([configure.ac])
 AM_INIT_AUTOMAKE
 AM_PROG_CC_C_O
 AC_PROG_CXX
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 case "$host" in
-        *-*-mingw*|*-*-cygwin*)
+        *-*-mingw*)
                 AC_DEFINE(PLATFORM_WIN32, 1, [Platform is Win32])
                 opengl_LIBS="-lunsupported_platform"
                 PLATFORM_STRING="Win32"
                 ;;
+        *-*-cygwin*)
+                AC_DEFINE(PLATFORM_CYGWIN, 1, [Platform is Cygwin])
+                opengl_LIBS="-lGL -lGLU -lglut"
+                PLATFORM_STRING="Cygwin"
+                ;;
         *-*-linux*)
                 AC_DEFINE(PLATFORM_LINUX, 1, [Platform is Linux])
                 opengl_LIBS="-lGL -lGLU -lglut"
--- origsrc/bullet-2.77.2243svn/src/Makefile.am	2010-12-09 20:01:03.000000000 +0900
+++ src/bullet-2.77.2243svn/src/Makefile.am	2010-12-11 00:43:35.328985100 +0900
@@ -6,11 +6,8 @@ nobase_bullet_include_HEADERS = \
 
 if CONDITIONAL_BUILD_MULTITHREADED
 nobase_bullet_include_HEADERS += \
+	BulletMultiThreaded/btParallelConstraintSolver.h \
 	BulletMultiThreaded/PosixThreadSupport.h \
-	BulletMultiThreaded/vectormath/scalar/cpp/mat_aos.h \
-	BulletMultiThreaded/vectormath/scalar/cpp/vec_aos.h \
-	BulletMultiThreaded/vectormath/scalar/cpp/quat_aos.h \
-	BulletMultiThreaded/vectormath/scalar/cpp/vectormath_aos.h \
 	BulletMultiThreaded/PpuAddressSpace.h \
 	BulletMultiThreaded/SpuCollisionTaskProcess.h \
 	BulletMultiThreaded/PlatformDefinitions.h \
@@ -41,6 +38,7 @@ lib_LTLIBRARIES	= libLinearMath.la libBu
 
 libBulletMultiThreaded_la_CXXFLAGS = ${CXXFLAGS} -I./BulletMultiThreaded/vectormath/scalar/cpp
 libBulletMultiThreaded_la_SOURCES =\
+		BulletMultiThreaded/btParallelConstraintSolver.cpp \
 		BulletMultiThreaded/SpuCollisionObjectWrapper.cpp \
 		BulletMultiThreaded/SpuSampleTask/SpuSampleTask.cpp \
 		BulletMultiThreaded/SpuLibspe2Support.cpp \
@@ -83,7 +81,8 @@ libBulletMultiThreaded_la_SOURCES =\
 		BulletMultiThreaded/SpuNarrowPhaseCollisionTask/boxBoxDistance.cpp \
 		BulletMultiThreaded/SpuNarrowPhaseCollisionTask/boxBoxDistance.h \
 		BulletMultiThreaded/SpuNarrowPhaseCollisionTask/Box.h
-
+libBulletMultiThreaded_la_LDFLAGS = -no-undefined
+libBulletMultiThreaded_la_LIBADD = libLinearMath.la libBulletCollision.la libBulletSoftBody.la
 else
 lib_LTLIBRARIES	= libLinearMath.la libBulletCollision.la libBulletDynamics.la libBulletSoftBody.la
 endif
@@ -118,7 +117,7 @@ libLinearMath_la_SOURCES	= \
 		LinearMath/btDefaultMotionState.h \
 		LinearMath/btIDebugDraw.h \
 		LinearMath/btRandom.h
-
+libLinearMath_la_LDFLAGS = -no-undefined
 
 libBulletCollision_la_SOURCES = \
 		BulletCollision/NarrowPhaseCollision/btRaycastCallback.cpp \
@@ -300,6 +299,8 @@ libBulletCollision_la_SOURCES = \
                 BulletCollision/Gimpact/gim_contact.cpp\
                 BulletCollision/Gimpact/gim_memory.cpp\
                 BulletCollision/Gimpact/gim_tri_collision.cpp
+libBulletCollision_la_LDFLAGS = -no-undefined
+libBulletCollision_la_LIBADD = libLinearMath.la
 
 libBulletDynamics_la_SOURCES = \
 		BulletDynamics/Dynamics/btContinuousDynamicsWorld.cpp \
@@ -350,6 +351,8 @@ libBulletDynamics_la_SOURCES = \
 		BulletDynamics/Vehicle/btVehicleRaycaster.h \
 		BulletDynamics/Vehicle/btRaycastVehicle.h \
 		BulletDynamics/Vehicle/btWheelInfo.h
+libBulletDynamics_la_LDFLAGS = -no-undefined
+libBulletDynamics_la_LIBADD = libLinearMath.la libBulletCollision.la
 
 libBulletSoftBody_la_SOURCES = \
 		BulletSoftBody/btDefaultSoftBodySolver.cpp \
@@ -369,8 +372,8 @@ libBulletSoftBody_la_SOURCES = \
 		BulletSoftBody/btSoftBodyConcaveCollisionAlgorithm.h \
 		BulletSoftBody/btSoftRigidDynamicsWorld.h \
 		BulletSoftBody/btSoftBodyHelpers.h
-
-
+libBulletSoftBody_la_LDFLAGS = -no-undefined
+libBulletSoftBody_la_LIBADD = libLinearMath.la libBulletCollision.la libBulletDynamics.la
 
 nobase_bullet_include_HEADERS += \
 	BulletSoftBody/btSoftBodyRigidBodyCollisionConfiguration.h \
