--- origsrc/vorbisgain-0.37/vorbisgain.c	2025-03-02 10:06:18.462317100 +0900
+++ src/vorbisgain-0.37/vorbisgain.c	2025-03-02 10:11:08.346428700 +0900
@@ -26,15 +26,15 @@
 
 #ifdef WIN32
 #include <windows.h>
-#endif
-
+#endif
+
 #ifdef ENABLE_RECURSIVE
 #include "recurse.h"
 #endif
 
 
 /* Some version numbers */
-#define VG_VERSION     "0.36"
+#define VG_VERSION     "0.37"
 #define VORBIS_VERSION "1.0"
 
 
