--- origsrc/libfixc-0.1.1/m4/sxe-compiler.m4	2013-03-21 16:28:58.000000000 +0900
+++ src/libfixc-0.1.1/m4/sxe-compiler.m4	2013-10-16 18:48:43.773476500 +0900
@@ -233,9 +233,9 @@ AC_DEFUN([SXE_FEATFLAGS], [dnl
 	SXE_CHECK_COMPILER_FLAGS([-static-intel], [
 		cflags="${cflags} ${XFLAG} -static-intel"
 		ldflags="${ldflags} ${XCCFLAG} -static-intel"])
-	SXE_CHECK_COMPILER_FLAGS([-static-libgcc], [
-		cflags="${cflags} ${XFLAG} -static-libgcc"
-		ldflags="${ldflags} ${XCCFLAG} -static-libgcc"])
+	dnl SXE_CHECK_COMPILER_FLAGS([-static-libgcc], [
+dnl		cflags="${cflags} ${XFLAG} -static-libgcc"
+dnl		ldflags="${ldflags} ${XCCFLAG} -static-libgcc"])
 ])dnl SXE_FEATFLAGS
 
 
@@ -278,7 +278,7 @@ AC_DEFUN([SXE_CHECK_CFLAGS], [dnl
 	dnl are handled separately, not just the xe_cflags_warning stuff.
 
 	## Use either command line flag, environment var, or autodetection
-	SXE_DEBUGFLAGS
+	dnl SXE_DEBUGFLAGS
 	SXE_WARNFLAGS
 	SXE_OPTIFLAGS
 	SXE_FEATFLAGS
--- origsrc/libfixc-0.1.1/m4/sxe-libtool.m4	2013-03-21 16:28:58.000000000 +0900
+++ src/libfixc-0.1.1/m4/sxe-libtool.m4	2013-10-16 15:56:42.941321900 +0900
@@ -57,7 +57,7 @@ AC_DEFUN([_SXE_CHECK_LT2], [dnl
 	dnl for the `old' libtool2 stuff as this is available through
 	dnl cvs only and we stick with the latest
 	dnl AC_CONFIG_SUBDIRS([libltdl])
-	AC_CONFIG_MACRO_DIR([libltdl/m4])
+	dnl AC_CONFIG_MACRO_DIR([libltdl/m4])
 ])dnl _SXE_CHECK_LT2
 
 AC_DEFUN([SXE_CHECK_LIBLTDL], [dnl
--- origsrc/libfixc-0.1.1/src/Makefile.am	2013-03-21 16:28:58.000000000 +0900
+++ src/libfixc-0.1.1/src/Makefile.am	2013-10-16 15:50:53.740150000 +0900
@@ -1,7 +1,7 @@
 ### Makefile.am
 
 AM_CPPFLAGS = -D_POSIX_C_SOURCE=200112L -D_XOPEN_SOURCE=600 -D_BSD_SOURCE
-AM_LDFLAGS =
+AM_LDFLAGS = -no-undefined
 
 headerdir = $(includedir)/libfixc
 
@@ -77,14 +77,12 @@ bin_PROGRAMS += fixml2fix
 fixml2fix_CPPFLAGS = $(AM_CPPFLAGS)
 fixml2fix_CPPFLAGS += -DSTANDALONE
 fixml2fix_LDFLAGS = $(AM_LDFLAGS)
-fixml2fix_LDFLAGS += -static
 fixml2fix_LDADD = libfixc.la
 
 bin_PROGRAMS += fix2fixml
 fix2fixml_CPPFLAGS = $(AM_CPPFLAGS)
 fix2fixml_CPPFLAGS += -DSTANDALONE
 fix2fixml_LDFLAGS = $(AM_LDFLAGS)
-fix2fixml_LDFLAGS += -static
 fix2fixml_LDADD = libfixc.la
 
 
--- origsrc/libfixc-0.1.1/test/Makefile.am	2013-03-21 16:28:58.000000000 +0900
+++ src/libfixc-0.1.1/test/Makefile.am	2013-10-16 18:27:54.645546800 +0900
@@ -71,13 +71,13 @@ fixml_tests += fixc_to_fixml_01
 fixc_to_fixml_01_LDADD = $(top_builddir)/src/libfixc.la
 fixml_tests += fixc_to_fixml_02
 fixc_to_fixml_02_SOURCES = fixc_to_fixml_01.c
-fixc_to_fixml_02_LDFLAGS = $(AM_LDFLAGS) -static
+fixc_to_fixml_02_LDFLAGS = $(AM_LDFLAGS)
 fixc_to_fixml_02_LDADD = $(top_builddir)/src/libfixc.la
 fixml_tests += fixc_to_fixml_03
 fixc_to_fixml_03_LDADD = $(top_builddir)/src/libfixc.la
 fixml_tests += fixc_to_fixml_04
 fixc_to_fixml_04_SOURCES = fixc_to_fixml_03.c
-fixc_to_fixml_04_LDFLAGS = $(AM_LDFLAGS) -static
+fixc_to_fixml_04_LDFLAGS = $(AM_LDFLAGS)
 fixc_to_fixml_04_LDADD = $(top_builddir)/src/libfixc.la
 
 api_tests += fixc_extr_01
@@ -128,7 +128,7 @@ fixc_from_fix_01_LDADD = $(top_builddir)
 ## same thing but static
 rd_tests += fixc_from_fix_02
 fixc_from_fix_02_SOURCES = fixc_from_fix_01.c
-fixc_from_fix_02_LDFLAGS = $(AM_LDFLAGS) -static
+fixc_from_fix_02_LDFLAGS = $(AM_LDFLAGS)
 fixc_from_fix_02_LDADD = $(top_builddir)/src/libfixc.la
 
 rd_tests += fixc_from_fix_03
@@ -136,7 +136,7 @@ fixc_from_fix_03_LDADD = $(top_builddir)
 ## same thing but static
 rd_tests += fixc_from_fix_04
 fixc_from_fix_04_SOURCES = fixc_from_fix_03.c
-fixc_from_fix_04_LDFLAGS = $(AM_LDFLAGS) -static
+fixc_from_fix_04_LDFLAGS = $(AM_LDFLAGS)
 fixc_from_fix_04_LDADD = $(top_builddir)/src/libfixc.la
 
 rd_tests += fixc_from_fixml_01
@@ -144,7 +144,7 @@ fixc_from_fixml_01_LDADD = $(top_builddi
 ## same thing but static
 rd_tests += fixc_from_fixml_02
 fixc_from_fixml_02_SOURCES = fixc_from_fixml_01.c
-fixc_from_fixml_02_LDFLAGS = $(AM_LDFLAGS) -static
+fixc_from_fixml_02_LDFLAGS = $(AM_LDFLAGS)
 fixc_from_fixml_02_LDADD = $(top_builddir)/src/libfixc.la
 
 rd_tests += fixc_from_fixml_03
@@ -152,7 +152,7 @@ fixc_from_fixml_03_LDADD = $(top_builddi
 ## same thing but static
 rd_tests += fixc_from_fixml_04
 fixc_from_fixml_04_SOURCES = fixc_from_fixml_03.c
-fixc_from_fixml_04_LDFLAGS = $(AM_LDFLAGS) -static
+fixc_from_fixml_04_LDFLAGS = $(AM_LDFLAGS)
 fixc_from_fixml_04_LDADD = $(top_builddir)/src/libfixc.la
 
 rd_tests += fixc_from_fixml_05
@@ -160,7 +160,7 @@ fixc_from_fixml_05_LDADD = $(top_builddi
 ## same thing but static
 rd_tests += fixc_from_fixml_06
 fixc_from_fixml_06_SOURCES = fixc_from_fixml_05.c
-fixc_from_fixml_06_LDFLAGS = $(AM_LDFLAGS) -static
+fixc_from_fixml_06_LDFLAGS = $(AM_LDFLAGS)
 fixc_from_fixml_06_LDADD = $(top_builddir)/src/libfixc.la
 
 rd_tests += fixc_from_fixml_07
@@ -168,7 +168,7 @@ fixc_from_fixml_07_LDADD = $(top_builddi
 ## same thing but static
 rd_tests += fixc_from_fixml_08
 fixc_from_fixml_08_SOURCES = fixc_from_fixml_07.c
-fixc_from_fixml_08_LDFLAGS = $(AM_LDFLAGS) -static
+fixc_from_fixml_08_LDFLAGS = $(AM_LDFLAGS)
 fixc_from_fixml_08_LDADD = $(top_builddir)/src/libfixc.la
 
 rd_tests += fixc_from_fixml_09
@@ -176,7 +176,7 @@ fixc_from_fixml_09_LDADD = $(top_builddi
 ## same thing but static
 rd_tests += fixc_from_fixml_10
 fixc_from_fixml_10_SOURCES = fixc_from_fixml_09.c
-fixc_from_fixml_10_LDFLAGS = $(AM_LDFLAGS) -static
+fixc_from_fixml_10_LDFLAGS = $(AM_LDFLAGS)
 fixc_from_fixml_10_LDADD = $(top_builddir)/src/libfixc.la
 
 rd_tests += fixc_from_fixml_11
@@ -184,7 +184,7 @@ fixc_from_fixml_11_LDADD = $(top_builddi
 ## same thing but static
 rd_tests += fixc_from_fixml_12
 fixc_from_fixml_12_SOURCES = fixc_from_fixml_11.c
-fixc_from_fixml_12_LDFLAGS = $(AM_LDFLAGS) -static
+fixc_from_fixml_12_LDFLAGS = $(AM_LDFLAGS)
 fixc_from_fixml_12_LDADD = $(top_builddir)/src/libfixc.la
 
 rd_tests += fixc_from_fixml_13
@@ -192,7 +192,7 @@ fixc_from_fixml_13_LDADD = $(top_builddi
 ## same thing but static
 rd_tests += fixc_from_fixml_14
 fixc_from_fixml_14_SOURCES = fixc_from_fixml_13.c
-fixc_from_fixml_14_LDFLAGS = $(AM_LDFLAGS) -static
+fixc_from_fixml_14_LDFLAGS = $(AM_LDFLAGS)
 fixc_from_fixml_14_LDADD = $(top_builddir)/src/libfixc.la
 
 TESTS_ENVIRONMENT += top_builddir=$(top_builddir)
@@ -215,7 +215,7 @@ cme_settle_SOURCES = cme-settle.c
 cme_settle_CPPFLAGS = $(AM_CPPFLAGS)
 cme_settle_CPPFLAGS += -D_GNU_SOURCE
 cme_settle_LDFLAGS = $(AM_LDFLAGS)
-cme_settle_LDADD = $(top_builddir)/src/libfixc.la
+cme_settle_LDADD = $(top_builddir)/src/libfixc.la -largp
 
 bin_tests += cme_settle_test_01.sh
 bin_tests += cme_settle_test_02.sh
