--- origsrc/glog-0.3.1/Makefile.am	2010-05-15 06:44:19.000000000 +0900
+++ src/glog-0.3.1/Makefile.am	2010-11-11 23:13:09.072618300 +0900
@@ -192,7 +192,7 @@ libglog_la_SOURCES = $(gloginclude_HEADE
 nodist_libglog_la_SOURCES = $(nodist_gloginclude_HEADERS)
 
 libglog_la_CXXFLAGS = $(PTRHEAD_CFLAGS) $(GFLAGS_CFLAGS) $(AM_CXXFLAGS) -DNDEBUG
-libglog_la_LDFLAGS = $(PTRHEAD_CFLAGS) $(GFLAGS_LDFLAGS)
+libglog_la_LDFLAGS = $(PTRHEAD_CFLAGS) $(GFLAGS_LDFLAGS) -no-undefined
 libglog_la_LIBADD = $(COMMON_LIBS)
 
 ## The location of the windows project file for each binary we make
--- origsrc/glog-0.3.1/configure.ac	2010-06-15 15:29:42.000000000 +0900
+++ src/glog-0.3.1/configure.ac	2010-11-11 23:05:01.869119100 +0900
@@ -20,7 +20,7 @@ AC_PROG_CPP
 AC_PROG_CXX
 AM_CONDITIONAL(GCC, test "$GCC" = yes)   # let the Makefile know if we're gcc
 
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_SUBST(LIBTOOL_DEPS)
 
 # Check whether some low-level functions/files are available
