--- origsrc/cpmredir-1.1.1/configure.in	2003-03-29 05:09:14.000000000 +0900
+++ src/cpmredir-1.1.1/configure.in	2014-04-06 23:54:38.315615200 +0900
@@ -1,21 +1,24 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(lib/cpmredir.c)
+AC_INIT([cpmredir], [1.1.1])
 AC_CONFIG_AUX_DIR(config)
 VERSION=1.1.1
 UPDATED='March 28, 2003'
-AM_INIT_AUTOMAKE(cpmredir, $VERSION)
+AM_INIT_AUTOMAKE
 AM_CONFIG_HEADER(config.h)
 
 dnl Checks for programs.
 AC_PROG_CC
-AC_PROG_RANLIB
+LT_INIT
 
 dnl Checks for libraries.
 
 dnl Checks for header files.
 AC_HEADER_DIRENT
 AC_HEADER_STDC
-AC_CHECK_HEADERS(fcntl.h unistd.h sys/vfs.h windows.h winnt.h utime.h)
+AC_CHECK_HEADERS(fcntl.h unistd.h sys/vfs.h utime.h)
+if test "$host_os" != "cygwin"; then
+  AC_CHECK_HEADERS(windows.h winnt.h)
+fi
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_TYPE_OFF_T
--- origsrc/cpmredir-1.1.1/lib/Makefile.am	2001-07-07 21:49:16.000000000 +0900
+++ src/cpmredir-1.1.1/lib/Makefile.am	2014-04-06 23:54:16.108584000 +0900
@@ -2,7 +2,7 @@
 
 INCLUDES=-I$(top_srcdir)/include
 
-lib_LIBRARIES=libcpmredir.a
-libcpmredir_a_SOURCES = cpmdrv.c   cpmint.h	  cpmredir.c  util.c \
+lib_LTLIBRARIES=libcpmredir.la
+libcpmredir_la_SOURCES = cpmdrv.c   cpmint.h	  cpmredir.c  util.c \
 		   cpmglob.c  cpmparse.c  drdos.c     xlt.c
-
+libcpmredir_la_LDFLAGS = -no-undefined
