--- origsrc/dnstracer/Makefile	2008-06-11 21:51:35.000000000 +0900
+++ src/dnstracer/Makefile	2025-12-28 12:06:04.504304200 +0900
@@ -2,7 +2,7 @@ MANDATE=	June 10, 2008
 MANSEC=		8
 MANTITLE=	Additional Software
 
-VERSION=	1.11
+VERSION=	1.10
 
 PREFIX=		/usr/local/
 BINPREFIX=	${PREFIX}/bin/
@@ -18,7 +18,7 @@ clean:
 	-rm dnstracer.o dnstracer dnstracer.8 dnstracer.core
 
 dnstracer: dnstracer.c dnstracer-missing.h
-	$(CC) -Wall -o dnstracer -DVERSION="${VERSION}" dnstracer.c
+	$(CC) $(CFLAGS) -Wall -o dnstracer -DVERSION="${VERSION}" dnstracer.c
 
 dnstracer.8: dnstracer.pod
 	pod2man --release="${MANDATE}" --date="${MANDATE}" \
