--- origsrc/beef/Makefile.am	2018-02-22 21:00:05.000000000 +0900
+++ src/beef/Makefile.am	2018-02-22 21:17:06.234000000 +0900
@@ -1,7 +1,6 @@
 NULL =
 
 SUBDIRS = \
-	doc \
 	src \
 	$(NULL)
 
--- origsrc/beef/configure.ac	2018-02-22 21:00:05.000000000 +0900
+++ src/beef/configure.ac	2018-02-22 21:17:11.922000000 +0900
@@ -14,7 +14,7 @@ AM_PROG_CC_C_O
 AC_PROG_LIBTOOL
 
 AC_PATH_PROG([PANDOC], [pandoc])
-test "$PANDOC" || AC_MSG_ERROR([pandoc not found])
+# test "$PANDOC" || AC_MSG_ERROR([pandoc not found])
 
 dnl **********************
 dnl *** Check for GLib ***
