--- origsrc/liblognorm-0.3.2/src/Makefile.am	2011-11-17 01:43:09.000000000 +0900
+++ src/liblognorm-0.3.2/src/Makefile.am	2012-01-10 22:04:57.568306800 +0900
@@ -22,7 +22,7 @@ liblognorm_la_SOURCES = \
 
 liblognorm_la_CPPFLAGS = $(LIBEE_CFLAGS) $(LIBESTR_CFLAGS)
 liblognorm_la_LIBADD = $(rt_libs) $(LIBEE_LIBS) $(LIBESTR_LIBS) -lestr
-liblognorm_la_LDFLAGS = -version-info 0:0:0
+liblognorm_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 
 EXTRA_DIST = \
 	internal.h \
