--- origsrc/liblognorm-2.0.3/src/Makefile.am	2017-01-23 23:32:52.000000000 +0900
+++ src/liblognorm-2.0.3/src/Makefile.am	2017-03-31 16:49:38.349076700 +0900
@@ -9,7 +9,7 @@ PTHREADS_CFLAGS = -pthread
 bin_PROGRAMS = lognormalizer
 lognormalizer_SOURCES = lognormalizer.c
 lognormalizer_CPPFLAGS =  -I$(top_srcdir) $(WARN_CFLAGS) $(JSON_C_CFLAGS) $(LIBESTR_CFLAGS)
-lognormalizer_LDADD = $(JSON_C_LIBS) $(LIBLOGNORM_LIBS) $(LIBESTR_LIBS) ../compat/compat.la 
+lognormalizer_LDADD = $(JSON_C_LIBS) $(LIBLOGNORM_LIBS) $(LIBESTR_LIBS) ../compat/compat.la
 lognormalizer_DEPENDENCIES = liblognorm.la
 
 check_PROGRAMS = ln_test
@@ -47,7 +47,7 @@ liblognorm_la_LIBADD = $(rt_libs) $(JSON
 # info on version-info:
 # http://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
 # Note: v2 now starts at version 5, as v1 previously also had 4
-liblognorm_la_LDFLAGS = -version-info 6:0:1
+liblognorm_la_LDFLAGS = -no-undefined -version-info 6:0:1
 
 EXTRA_DIST = \
 	internal.h \
