--- origsrc/liblognorm-2.0.4/src/Makefile.am	2017-01-23 23:32:52.000000000 +0900
+++ src/liblognorm-2.0.4/src/Makefile.am	2017-10-12 20:58:09.151070200 +0900
@@ -47,7 +47,7 @@ liblognorm_la_LIBADD = $(rt_libs) $(JSON
 # info on version-info:
 # http://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
 # Note: v2 now starts at version 5, as v1 previously also had 4
-liblognorm_la_LDFLAGS = -version-info 6:0:1
+liblognorm_la_LDFLAGS = -no-undefined -version-info 6:0:1
 
 EXTRA_DIST = \
 	internal.h \
