--- origsrc/yasr-0.6.9/yasr/main.c	2007-09-23 09:46:58.000000000 +0900
+++ src/yasr-0.6.9/yasr/main.c	2015-05-06 21:47:06.707177900 +0900
@@ -32,6 +32,7 @@
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/wait.h>
+#include <sys/select.h>
 
 static int cpid;
 static int size;
--- origsrc/yasr-0.6.9/yasr/yasr.h	2008-01-31 21:56:45.000000000 +0900
+++ src/yasr-0.6.9/yasr/yasr.h	2015-05-06 21:45:45.770099500 +0900
@@ -308,8 +308,10 @@ extern void opt_set(int num, void *val);
 extern void opt_queue_empty(int ll);
 extern void opt_write(FILE * fp);
 
+#if !defined(__CYGWIN__)
 /* openpty.c prototypes */
 extern int openpty(int *, int *, char *, struct termios *, struct winsize *);
+#endif
 
 /* cfmakeraw.c prototypes */
 extern void cfmakeraw(struct termios *);
@@ -317,8 +319,10 @@ extern void cfmakeraw(struct termios *);
 /* login_tty.c prototypes */
 extern int login_tty(int);
 
+#if !defined(__CYGWIN__)
 /* forkpty.c prototypes */
 extern int forkpty(int *, char *, struct termios *, struct winsize *);
+#endif
 
 /* tbc - Would it be more efficient to ensure that "blank" grids always held
    ascii 0x20 rather than ascii 0x00? */
