--- origsrc/code/CMakeLists.txt	2025-03-27 14:32:19.294801300 +0900
+++ src/code/CMakeLists.txt	2025-03-27 16:49:33.033983900 +0900
@@ -117,3 +117,10 @@ set(CPACK_SOURCE_IGnore_FILES
 )
 
 include(CPack)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libdecaf.pc.in ${CMAKE_BINARY_DIR}/libdecaf.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libdecaf.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/code/libdecaf.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/code/libdecaf.pc.in	2025-03-27 14:43:19.015695200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libdecaf
+Description: Fast, strong elliptic curve cryptography
+Version: @VERSION@
+Libs: -L${libdir} -ldecaf
+Cflags: -I${includedir}
--- origsrc/code/src/CMakeLists.txt	2025-03-27 14:32:15.000000000 +0900
+++ src/code/src/CMakeLists.txt	2025-03-27 14:41:49.436483400 +0900
@@ -95,7 +95,7 @@ endif()
 if(ENABLE_SHARED)
 	add_library(decaf SHARED ${DECAF_HEADER_FILES} ${DECAF_SOURCE_FILES_C} ${DECAF_SOURCE_FILES_CXX} $<TARGET_OBJECTS:p25519> $<TARGET_OBJECTS:p448> $<TARGET_OBJECTS:CURVE25519> $<TARGET_OBJECTS:CURVE448>)
 	add_dependencies(decaf generatedCode)
-	set_target_properties(decaf PROPERTIES VERSION ${DECAF_SO_VERSION})
+	set_target_properties(decaf PROPERTIES VERSION ${DECAF_SO_VERSION} SOVERSION ${DECAF_SO_VERSION})
 	target_include_directories(decaf INTERFACE $<INSTALL_INTERFACE:include/decaf>)
 	target_link_libraries(decaf PRIVATE)
 	if(MSVC)
