--- origsrc/code/CMakeLists.txt	2022-11-16 17:17:32.000000000 +0900
+++ src/code/CMakeLists.txt	2022-11-16 17:52:31.651617000 +0900
@@ -75,7 +75,11 @@ if(ENABLE_TESTS)
 endif()
 
 include(CMakePackageConfigHelpers)
-set(ConfigPackageLocation share/decaf/cmake)
+if(CYGWIN)
+	set(ConfigPackageLocation lib/cmake/decaf)
+else()
+	set(ConfigPackageLocation share/decaf/cmake)
+endif()
 export(EXPORT ${EXPORT_TARGETS_NAME}Targets
 	FILE "${CMAKE_CURRENT_BINARY_DIR}/DecafTargets.cmake"
 )
@@ -117,3 +121,10 @@ set(CPACK_SOURCE_IGnore_FILES
 )
 
 include(CPack)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libdecaf.pc.in ${CMAKE_BINARY_DIR}/libdecaf.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libdecaf.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/code/libdecaf.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/code/libdecaf.pc.in	2022-11-16 17:52:51.278149900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libdecaf
+Description: Fast, strong elliptic curve cryptography
+Version: @VERSION@
+Libs: -L${libdir} -ldecaf
+Cflags: -I${includedir}/decaf
--- origsrc/code/src/CMakeLists.txt	2022-11-16 17:17:32.000000000 +0900
+++ src/code/src/CMakeLists.txt	2022-11-16 17:19:39.252465900 +0900
@@ -95,7 +95,7 @@ endif()
 if(ENABLE_SHARED)
 	add_library(decaf SHARED ${DECAF_HEADER_FILES} ${DECAF_SOURCE_FILES_C} ${DECAF_SOURCE_FILES_CXX} $<TARGET_OBJECTS:p25519> $<TARGET_OBJECTS:p448> $<TARGET_OBJECTS:CURVE25519> $<TARGET_OBJECTS:CURVE448>)
 	add_dependencies(decaf generatedCode)
-	set_target_properties(decaf PROPERTIES VERSION ${DECAF_SO_VERSION})
+	set_target_properties(decaf PROPERTIES VERSION ${DECAF_SO_VERSION} SOVERSION ${DECAF_SO_VERSION})
 	target_include_directories(decaf INTERFACE $<INSTALL_INTERFACE:include/decaf>)
 	target_link_libraries(decaf PRIVATE)
 	if(MSVC)
--- origsrc/code/src/per_curve/eddsa.tmpl.h	2022-11-16 17:17:32.000000000 +0900
+++ src/code/src/per_curve/eddsa.tmpl.h	2022-11-16 17:37:54.086583200 +0900
@@ -143,8 +143,7 @@ void DECAF_API_VIS decaf_ed$(gf_shortnam
     uint8_t context_len
 ) __attribute__((nonnull(1,2,3))) DECAF_NOINLINE
 #if DECAF_EDDSA_NON_KEYPAIR_API_IS_DEPRECATED
-  __attribute__((deprecated("Passing the pubkey and privkey separately is unsafe",
-        "decaf_ed$(gf_shortname)_keypair_sign")))
+  __attribute__((deprecated("Passing the pubkey and privkey separately is unsafe")))
 #endif
 ;
 
@@ -171,8 +170,7 @@ void DECAF_API_VIS decaf_ed$(gf_shortnam
     uint8_t context_len
 ) __attribute__((nonnull(1,2,3,4))) DECAF_NOINLINE
 #if DECAF_EDDSA_NON_KEYPAIR_API_IS_DEPRECATED
-  __attribute__((deprecated("Passing the pubkey and privkey separately is unsafe",
-        "decaf_ed$(gf_shortname)_keypair_sign_prehash")))
+  __attribute__((deprecated("Passing the pubkey and privkey separately is unsafe")))
 #endif
 ;
 
