--- origsrc/libdecaf-1.0.3/CMakeLists.txt	2026-07-30 08:34:48.635353400 +0900
+++ src/libdecaf-1.0.3/CMakeLists.txt	2026-07-30 08:40:05.505774900 +0900
@@ -130,3 +130,10 @@ set(CPACK_SOURCE_IGnore_FILES
 )
 
 include(CPack)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libdecaf.pc.in ${CMAKE_BINARY_DIR}/libdecaf.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libdecaf.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libdecaf-1.0.3/libdecaf.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libdecaf-1.0.3/libdecaf.pc.in	2026-07-30 08:41:09.127260000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libdecaf
+Description: Fast, strong elliptic curve cryptography
+Version: @VERSION@
+Libs: -L${libdir} -ldecaf
+Cflags: -I${includedir}/decaf
--- origsrc/libdecaf-1.0.3/src/CMakeLists.txt	2026-04-30 04:06:31.000000000 +0900
+++ src/libdecaf-1.0.3/src/CMakeLists.txt	2026-07-30 08:35:42.767985900 +0900
@@ -94,7 +94,7 @@ endif()
 if(ENABLE_SHARED)
 	add_library(decaf SHARED ${DECAF_HEADER_FILES} ${DECAF_SOURCE_FILES_C} ${DECAF_SOURCE_FILES_CXX} $<TARGET_OBJECTS:decaf-p25519> $<TARGET_OBJECTS:decaf-p448> $<TARGET_OBJECTS:decaf-curve25519> $<TARGET_OBJECTS:decaf-curve448>)
 	add_dependencies(decaf decaf-generated-code)
-	set_target_properties(decaf PROPERTIES VERSION ${DECAF_SO_VERSION})
+	set_target_properties(decaf PROPERTIES SOVERSION ${DECAF_SO_VERSION})
 	target_include_directories(decaf INTERFACE $<BUILD_INTERFACE:${GSOURCE_PATH}/include> $<INSTALL_INTERFACE:include/decaf>)
 	if (ENABLE_SANITIZER)
 		target_link_options(decaf PRIVATE ${SANITIZER_LINK_FLAGS})
