diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libpsca/CMakeLists.txt src/libpsca/CMakeLists.txt
--- origsrc/libpsca/CMakeLists.txt	2016-02-06 14:12:31.000000000 +0900
+++ src/libpsca/CMakeLists.txt	2016-02-06 14:14:37.108770100 +0900
@@ -24,3 +24,9 @@ endif (DOXYGEN_FOUND)
 add_subdirectory (lib)
 add_subdirectory (examples)
 
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libpsca.pc.in ${CMAKE_BINARY_DIR}/libpsca.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libpsca.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libpsca/examples/CMakeLists.txt src/libpsca/examples/CMakeLists.txt
--- origsrc/libpsca/examples/CMakeLists.txt	2016-02-06 14:12:31.000000000 +0900
+++ src/libpsca/examples/CMakeLists.txt	2016-02-06 14:15:27.542356000 +0900
@@ -9,3 +9,4 @@ add_executable (psca_auto ${PSCA_AUTO_SO
 add_dependencies (psca_auto psca)
 target_link_libraries (psca_auto psca)
 
+install(TARGETS psca_auto DESTINATION bin)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libpsca/lib/CMakeLists.txt src/libpsca/lib/CMakeLists.txt
--- origsrc/libpsca/lib/CMakeLists.txt	2016-02-06 14:12:31.000000000 +0900
+++ src/libpsca/lib/CMakeLists.txt	2016-02-06 14:12:54.989200300 +0900
@@ -13,14 +13,6 @@ include_directories (${PROJECT_BINARY_DI
   set (PSCA_PSCA_HEADERS ${PSCA_VERSION_OUT} ${PSCA_EXPORT_HEADER})
 # }}}
 
-# Build static library {{{
-  add_library (psca_static STATIC ${PSCA_SOURCES} ${PSCA_HEADERS} ${PSCA_PSCA_HEADERS})
-
-  set_target_properties (psca_static
-                         PROPERTIES
-                         OUTPUT_NAME "psca")
-# }}}
-
 # Build shared library {{{
   add_library (psca SHARED ${PSCA_SOURCES} ${PSCA_HEADERS})
 
@@ -31,7 +23,7 @@ include_directories (${PROJECT_BINARY_DI
 # }}}
 
 # Install targets {{{
-  install (TARGETS psca psca_static
+  install (TARGETS psca
            RUNTIME DESTINATION bin
            LIBRARY DESTINATION lib
            ARCHIVE DESTINATION lib)
@@ -42,4 +34,3 @@ include_directories (${PROJECT_BINARY_DI
   install (FILES ${PSCA_PSCA_HEADERS}
            DESTINATION include/psca)
 # }}}
-
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libpsca/libpsca.pc.in src/libpsca/libpsca.pc.in
--- origsrc/libpsca/libpsca.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libpsca/libpsca.pc.in	2016-02-06 14:14:13.377546300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libpsca
+Description: Pool Stack C Allocator
+Version: @PSCA_VERSION_STRING@
+Libs: -L${libdir} -lpsca
+Cflags: -I${includedir}
