--- origsrc/g729/Makefile.am	2012-12-03 04:01:25.000000000 +0900
+++ src/g729/Makefile.am	2013-10-21 09:45:07.733125000 +0900
@@ -1,5 +1,4 @@
 lib_LTLIBRARIES         = libg729b.la
-libg729b_la_CPPFLAGS = -fPIC
 
 libg729b_la_SOURCES  = \
  acelp_ca.c\
@@ -39,5 +38,6 @@ libg729b_la_SOURCES  = \
  vad.c
 	
 libg729b_la_LDFLAGS = $LDFLAGS -no-undefined
-libg729b_la_HEADERS = *.h
-libg729b_ladir = $(includedir)/g729b
\ No newline at end of file
+pkginclude_HEADERS = \
+basic_op.h  dtx.h  ld8a.h  octet.h  oper_32b.h  sid.h  tab_dtx.h  tab_ld8a.h \
+typedef.h   vad.h
--- origsrc/g729/dec_ld8a.c	2010-10-18 21:34:57.000000000 +0900
+++ src/g729/dec_ld8a.c	2013-10-22 10:11:20.026851600 +0900
@@ -136,7 +136,7 @@ void Decod_ld8a(
   Word32  L_temp;
 
   Word16 bad_pitch;             /* bad pitch indicator */
-  extern Word16 bad_lsf;        /* bad LSF indicator   */
+  Word16 bad_lsf;        /* bad LSF indicator   */
 
   /* for G.729B */
   Word16 ftyp;
