--- origsrc/squish-1.11/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/squish-1.11/Makefile.am	2012-01-15 19:52:38.293981300 +0900
@@ -0,0 +1,7 @@
+
+include_HEADERS = squish.h
+
+lib_LTLIBRARIES = libsquish.la
+
+libsquish_la_SOURCES = alpha.cpp clusterfit.cpp colourblock.cpp colourfit.cpp colourset.cpp maths.cpp rangefit.cpp singlecolourfit.cpp squish.cpp
+libsquish_la_LDFLAGS = -no-undefined
--- origsrc/squish-1.11/alpha.cpp	2008-07-13 08:40:16.000000000 +0900
+++ src/squish-1.11/alpha.cpp	2012-01-15 19:51:30.987198900 +0900
@@ -25,6 +25,7 @@
    
 #include "alpha.h"
 #include <algorithm>
+#include <climits>
 
 namespace squish {
 
--- origsrc/squish-1.11/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/squish-1.11/configure.ac	2012-01-15 19:43:35.952221700 +0900
@@ -0,0 +1,30 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([squish], [1.11], [http://code.google.com/p/libsquish/])
+AC_CONFIG_SRCDIR([alpha.cpp])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+AC_PROG_INSTALL
+LT_INIT([disable-static])
+
+# Checks for libraries.
+
+# Checks for header files.
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_HEADER_STDBOOL
+AC_C_INLINE
+
+# Checks for library functions.
+AC_FUNC_ERROR_AT_LINE
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([floor pow sqrt])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
--- origsrc/squish-1.11/singlecolourfit.cpp	2008-07-13 08:40:16.000000000 +0900
+++ src/squish-1.11/singlecolourfit.cpp	2012-01-15 19:56:40.431973500 +0900
@@ -26,6 +26,7 @@
 #include "singlecolourfit.h"
 #include "colourset.h"
 #include "colourblock.h"
+#include <climits>
 
 namespace squish {
 
