--- origsrc/squish/CMakeLists.txt	2012-03-24 19:44:59.000000000 +0900
+++ src/squish/CMakeLists.txt	2015-04-05 00:05:20.427887100 +0900
@@ -82,6 +82,7 @@ IF (BUILD_SQUISH_EXTRA)
     ADD_EXECUTABLE(squishtest ${SQUISHTEST_SRCS})
     SET_TARGET_PROPERTIES(squishtest PROPERTIES DEBUG_POSTFIX "d")
     TARGET_LINK_LIBRARIES(squishtest squish)
+    INSTALL(TARGETS squishtest DESTINATION bin)
 
     SET(SQUISHPNG_SRCS extra/squishpng.cpp)
 
@@ -93,12 +94,22 @@ IF (BUILD_SQUISH_EXTRA)
         ADD_EXECUTABLE(squishpng ${SQUISHPNG_SRCS})
         SET_TARGET_PROPERTIES(squishpng PROPERTIES DEBUG_POSTFIX "d")
         TARGET_LINK_LIBRARIES(squishpng squish ${PNG_LIBRARIES})
+        INSTALL(TARGETS squishpng DESTINATION bin)
     ENDIF (PNG_FOUND)
 ENDIF (BUILD_SQUISH_EXTRA)
 
 INSTALL(
     TARGETS squish
+    RUNTIME DESTINATION bin
     LIBRARY DESTINATION lib
     ARCHIVE DESTINATION lib
     PUBLIC_HEADER DESTINATION include
     )
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "1.11")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/squish.pc.in ${CMAKE_BINARY_DIR}/squish.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/squish.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/squish/squish.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/squish/squish.pc.in	2015-04-05 00:05:31.919518900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: squish
+Description: Open source DXT compression library
+Version: @VERSION@
+Libs: -L${libdir} -lsquish
+Cflags: -I${includedir}
