--- origsrc/nylon-1.21/configure.in	2013-12-22 17:37:55.180211900 +0900
+++ src/nylon-1.21/configure.in	2013-12-22 17:43:46.896508800 +0900
@@ -84,7 +84,6 @@ AC_CHECK_TYPE(u_int32_t, unsigned int)
 AC_CHECK_TYPE(u_int16_t, unsigned short)
 AC_CHECK_TYPE(u_int8_t, unsigned char)
 AC_CHECK_TYPE(in_addr_t, unsigned long)
-AC_CHECK_TYPE(socklen_t, unsigned int)
 
 dnl Checks for library functions.
 AC_PROG_GCC_TRADITIONAL
--- origsrc/nylon-1.21/src/Makefile.am	2013-12-22 17:37:55.180211900 +0900
+++ src/nylon-1.21/src/Makefile.am	2013-12-22 17:48:46.008814300 +0900
@@ -1,5 +1,7 @@
 include $(top_srcdir)/Makefile.am.inc
 
+AM_CPPFLAGS = -I$(top_srcdir)/include
+
 bin_PROGRAMS = nylon
 
 nylon_SOURCES = nylon.c print.c cfg.c expanda.c net.c access.c atomicio.c \
