--- origsrc/rolldice-1.16/main.c	2016-10-02 01:34:53.000000000 +0900
+++ src/rolldice-1.16/main.c	2017-09-01 19:20:05.272322000 +0900
@@ -24,8 +24,10 @@ struct option long_opts[] = {{"help", 0,
                  {"interactive", 0, NULL, 'i'},
                  {NULL, 0, NULL, 0}};
 
+#if !defined(__CYGWIN__)
 /* For getopt usage */
 extern int optind;
+#endif
 
 /* Stores the random number file to use */
 rand_type rand_file = UNDEF;
