diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/rolldice-1.15/main.c src/rolldice-1.15/main.c
--- origsrc/rolldice-1.15/main.c	2015-10-17 02:27:18.000000000 +0900
+++ src/rolldice-1.15/main.c	2015-10-24 12:53:54.797507800 +0900
@@ -22,8 +22,10 @@ struct option long_opts[] = {{"help", 0,
 			     {"urandom", 0, NULL, 'u'},
 			     {"separate", 0, NULL, 's'}};
 
+#if !defined(__CYGWIN__)
 /* For getopt usage */
 extern int optind;
+#endif
 
 /* Stores the random number file to use */
 rand_type rand_file = UNDEF;
