diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/c-blosc-1.7.0/CMakeLists.txt src/c-blosc-1.7.0/CMakeLists.txt
--- origsrc/c-blosc-1.7.0/CMakeLists.txt	2015-07-05 21:31:37.000000000 +0900
+++ src/c-blosc-1.7.0/CMakeLists.txt	2015-11-08 22:38:55.310189000 +0900
@@ -278,3 +278,10 @@ set(CPACK_STRIP_FILES TRUE)
 set(CPACK_SOURCE_STRIP_FILES TRUE)
 
 include(CPack)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/c-blosc.pc.in ${CMAKE_BINARY_DIR}/c-blosc.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/c-blosc.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/c-blosc-1.7.0/bench/plot-speeds.py src/c-blosc-1.7.0/bench/plot-speeds.py
--- origsrc/c-blosc-1.7.0/bench/plot-speeds.py	2015-07-05 21:31:37.000000000 +0900
+++ src/c-blosc-1.7.0/bench/plot-speeds.py	2015-11-08 22:27:57.481727700 +0900
@@ -1,3 +1,4 @@
+#!/usr/bin/env python
 """Script for plotting the results of the 'suite' benchmark.
 Invoke without parameters for usage hints.
 
@@ -211,5 +212,3 @@ if __name__ == '__main__':
     plots.append(plot_)
     show_plot(plots, yaxis, legends, gtitle, xmax=int(options.xmax) if
             options.xmax else None)
-
-
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/c-blosc-1.7.0/blosc/CMakeLists.txt src/c-blosc-1.7.0/blosc/CMakeLists.txt
--- origsrc/c-blosc-1.7.0/blosc/CMakeLists.txt	2015-07-05 21:31:37.000000000 +0900
+++ src/c-blosc-1.7.0/blosc/CMakeLists.txt	2015-11-08 22:28:55.001702100 +0900
@@ -178,7 +178,7 @@ endif(BUILD_STATIC)
 
 # install
 install(FILES blosc.h blosc-export.h DESTINATION include COMPONENT DEV)
-install(TARGETS blosc_shared DESTINATION ${lib_dir} COMPONENT LIB)
+install(TARGETS blosc_shared RUNTIME DESTINATION bin ARCHIVE DESTINATION ${lib_dir} COMPONENT LIB)
 if(BUILD_STATIC)
     install(TARGETS blosc_static DESTINATION ${lib_dir} COMPONENT DEV)
 endif(BUILD_STATIC)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/c-blosc-1.7.0/c-blosc.pc.in src/c-blosc-1.7.0/c-blosc.pc.in
--- origsrc/c-blosc-1.7.0/c-blosc.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/c-blosc-1.7.0/c-blosc.pc.in	2015-11-08 22:38:19.005986400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: c-blosc
+Description: High performance compressor optimized for binary data
+Version: @BLOSC_VERSION_STRING@
+Libs: -L${libdir} -lblosc
+Cflags: -I${includedir}
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/c-blosc-1.7.0/tests/test_common.h src/c-blosc-1.7.0/tests/test_common.h
--- origsrc/c-blosc-1.7.0/tests/test_common.h	2015-07-05 21:31:37.000000000 +0900
+++ src/c-blosc-1.7.0/tests/test_common.h	2015-11-08 22:31:31.102377900 +0900
@@ -67,7 +67,7 @@ static void* blosc_test_malloc(const siz
 #elif _POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600
   /* Platform does have an implementation of posix_memalign */
   res = posix_memalign(&block, alignment, size);
-#elif defined(__APPLE__)
+#elif defined(__APPLE__) || defined(__CYGWIN__)
   /* Mac OS X guarantees 16-byte alignment in small allocs */
   block = malloc(size);
 #else
