diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/c-blosc-1.14.3/blosc/CMakeLists.txt src/c-blosc-1.14.3/blosc/CMakeLists.txt
--- origsrc/c-blosc-1.14.3/blosc/CMakeLists.txt	2018-04-06 18:27:51.000000000 +0900
+++ src/c-blosc-1.14.3/blosc/CMakeLists.txt	2018-05-16 16:51:40.425393500 +0900
@@ -213,7 +213,7 @@ endif(BUILD_STATIC)
 if(BLOSC_INSTALL)
     install(FILES blosc.h blosc-export.h DESTINATION include COMPONENT DEV)
     if(BUILD_SHARED)
-        install(TARGETS blosc_shared DESTINATION ${lib_dir} COMPONENT LIB)
+        install(TARGETS blosc_shared RUNTIME DESTINATION bin ARCHIVE DESTINATION ${lib_dir} COMPONENT LIB)
     endif(BUILD_SHARED)
     if(BUILD_STATIC)
         install(TARGETS blosc_static DESTINATION ${lib_dir} COMPONENT DEV)
