--- origsrc/c-blosc-1.20.0/tests/fuzz/CMakeLists.txt	2020-07-26 03:48:43.000000000 +0900
+++ src/c-blosc-1.20.0/tests/fuzz/CMakeLists.txt	2020-08-30 00:53:34.047677000 +0900
@@ -52,8 +52,8 @@ foreach(source ${SOURCES})
         target_link_libraries(${target} ${FUZZING_ENGINE})
     endif()
 
-    target_link_libraries(${target} blosc_static)
-    add_dependencies(${target} blosc_static)
+    target_link_libraries(${target} blosc)
+    add_dependencies(${target} blosc)
 
     # run standalone fuzzer against each file
     file(GLOB COMPAT_FILES ${PROJECT_SOURCE_DIR}/compat/*.cdata)
