--- origsrc/c-blosc-1.21.5/tests/fuzz/CMakeLists.txt	2023-09-02 16:17:46.000000000 +0900
+++ src/c-blosc-1.21.5/tests/fuzz/CMakeLists.txt	2023-09-20 14:44:52.490648200 +0900
@@ -52,8 +52,8 @@ foreach(source ${SOURCES})
         target_link_libraries(${target} ${FUZZING_ENGINE})
     endif()
 
-    target_link_libraries(${target} blosc_static)
-    add_dependencies(${target} blosc_static)
+    target_link_libraries(${target} blosc)
+    add_dependencies(${target} blosc)
 
     # run standalone fuzzer against each file
     file(GLOB COMPAT_FILES ${PROJECT_SOURCE_DIR}/compat/*.cdata)
