--- origsrc/tvlsim-tvlsim-1.01.0/config/project_config_embeddable.cmake	2013-06-09 05:00:17.000000000 +0900
+++ src/tvlsim-tvlsim-1.01.0/config/project_config_embeddable.cmake	2014-11-08 17:40:27.392472700 +0900
@@ -1692,7 +1692,8 @@ macro (module_library_add_specific
   # Installation of the library
   install (TARGETS ${_lib_target}
     EXPORT ${LIB_DEPENDENCY_EXPORT}
-    LIBRARY DESTINATION "${INSTALL_LIB_DIR}" COMPONENT runtime)
+    RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
+    ARCHIVE DESTINATION "${INSTALL_LIB_DIR}")
 
   # Register, for reporting purpose, the list of libraries to be built
   # and installed for that module
--- origsrc/tvlsim-tvlsim-1.01.0/tvlsim-config.cmake.in	2013-06-09 05:00:17.000000000 +0900
+++ src/tvlsim-tvlsim-1.01.0/tvlsim-config.cmake.in	2014-11-08 17:40:50.503800800 +0900
@@ -18,5 +18,5 @@ set (TVLSIM_LIBRARY_DIRS "@TVLSIM_LIB_DI
 include ("@TVLSIM_CMAKE_DIR@/tvlsim-library-depends.cmake")
 
 # These are the TvlSim IMPORTED targets, created by tvlsim-library-depends.cmake
-set (TVLSIM_LIBRARIES tvlsimlib)
+set (TVLSIM_LIBRARIES tvlsim)
 set (TVLSIM_EXECUTABLES tvlsim)
