--- origsrc/tvlsim-1.01.8/config/project_config_embeddable.cmake	2025-01-28 01:32:39.000000000 +0900
+++ src/tvlsim-1.01.8/config/project_config_embeddable.cmake	2025-11-27 18:36:10.019330500 +0900
@@ -1630,7 +1630,7 @@ macro (add_modules)
   set (_embedded_components ${ARGV})
   set (LIB_DEPENDENCY_EXPORT ${PROJECT_NAME}-library-depends)
   set (PY_LIB_DEPENDENCY_EXPORT ${PROJECT_NAME}-python-library-depends)
-  set (LIB_DEPENDENCY_EXPORT_PATH "${INSTALL_DATA_DIR}/${PROJECT_NAME}/CMake")
+  set (LIB_DEPENDENCY_EXPORT_PATH "${INSTALL_LIB_DIR}/cmake/${PACKAGE}")
   #
   foreach (_embedded_comp ${_embedded_components})
     #
@@ -2160,7 +2160,7 @@ endmacro (module_script_add)
 # can refer to it (for libraries, header files and binaries)
 macro (module_export_install)
   install (EXPORT ${LIB_DEPENDENCY_EXPORT}
-    DESTINATION "${INSTALL_DATA_DIR}/${PACKAGE}/CMake" COMPONENT devel)
+    DESTINATION "${INSTALL_LIB_DIR}/cmake/${PACKAGE}" COMPONENT devel)
 
   if (NEED_PYTHON)
 	install (EXPORT ${PY_LIB_DEPENDENCY_EXPORT}
--- origsrc/tvlsim-1.01.8/man/TvlSimServer.doc	2025-01-28 01:32:39.000000000 +0900
+++ src/tvlsim-1.01.8/man/TvlSimServer.doc	2025-11-27 18:59:50.874584700 +0900
@@ -96,7 +96,7 @@
     Query word list (not used for now)
 
 
-See the output of the <tt>`TvlSimServer --help'</tt> command for default options.
+See the output of the <tt>TvlSimServer --help</tt> command for default options.
 
 
 \section sec_see_also SEE ALSO
--- origsrc/tvlsim-1.01.8/man/simulate.doc	2025-01-28 01:32:39.000000000 +0900
+++ src/tvlsim-1.01.8/man/simulate.doc	2025-11-27 18:59:50.874584700 +0900
@@ -86,7 +86,7 @@
  \b -q, \b --query <travel-query><br>
     Query word list (not used for now)
 
-See the output of the <tt>`simulate --help'</tt> command for default options.
+See the output of the <tt>simulate --help</tt> command for default options.
 
 
 \section sec_see_also SEE ALSO
--- origsrc/tvlsim-1.01.8/man/tvlsim.doc	2025-01-28 01:32:39.000000000 +0900
+++ src/tvlsim-1.01.8/man/tvlsim.doc	2025-11-27 18:59:50.874584700 +0900
@@ -86,7 +86,7 @@
  \b -q, \b --query <travel-query><br>
     Query word list (not used for now)
 
-See the output of the <tt>`tvlsim --help'</tt> command for default options.
+See the output of the <tt>tvlsim --help</tt> command for default options.
 
 
 \section sec_see_also SEE ALSO
--- origsrc/tvlsim-1.01.8/tvlsim-config.cmake.in	2025-01-28 01:32:39.000000000 +0900
+++ src/tvlsim-1.01.8/tvlsim-config.cmake.in	2025-11-27 18:36:10.035166000 +0900
@@ -18,5 +18,5 @@ set (TVLSIM_LIBRARY_DIRS "@TVLSIM_LIB_DI
 include ("@TVLSIM_CMAKE_DIR@/tvlsim-library-depends.cmake")
 
 # These are the TvlSim IMPORTED targets, created by tvlsim-library-depends.cmake
-set (TVLSIM_LIBRARIES tvlsimlib)
+set (TVLSIM_LIBRARIES tvlsim)
 set (TVLSIM_EXECUTABLES tvlsim)
