--- origsrc/libtelnet/Makefile.am	2011-02-21 01:10:59.000000000 +0900
+++ src/libtelnet/Makefile.am	2011-02-21 01:11:43.114162300 +0900
@@ -8,7 +8,7 @@ include_HEADERS = libtelnet.h
 
 lib_LTLIBRARIES = libtelnet.la
 libtelnet_la_SOURCES = libtelnet.c libtelnet.h
-libtelnet_la_LDFLAGS = -version-info 2:0:0
+libtelnet_la_LDFLAGS = -version-info 2:0:0 -no-undefined
 libtelnet_la_LIBADD = $(zlib_LIBS)
 
 pkgconfigdir = $(libdir)/pkgconfig
--- origsrc/libtelnet/util/Makefile.am	2011-02-21 01:10:59.000000000 +0900
+++ src/libtelnet/util/Makefile.am	2011-02-21 01:12:32.094592700 +0900
@@ -1,15 +1,15 @@
 AM_CFLAGS = -I$(top_srcdir)
 
-telnet_client_SOURCES = telnet-client.c ../libtelnet.h
+telnet_client_SOURCES = telnet-client.c $(top_srcdir)/libtelnet.h
 telnet_client_LDADD = ../libtelnet.la
 
-telnet_chatd_SOURCES = telnet-chatd.c ../libtelnet.h
+telnet_chatd_SOURCES = telnet-chatd.c $(top_srcdir)/libtelnet.h
 telnet_chatd_LDADD = ../libtelnet.la
 
-telnet_proxy_SOURCES = telnet-proxy.c ../libtelnet.h
+telnet_proxy_SOURCES = telnet-proxy.c $(top_srcdir)/libtelnet.h
 telnet_proxy_LDADD = ../libtelnet.la
 
-telnet_test_SOURCES = telnet-test.c ../libtelnet.h
+telnet_test_SOURCES = telnet-test.c $(top_srcdir)/libtelnet.h
 telnet_test_LDADD = ../libtelnet.la
 
 if BUILD_UTIL
