--- origsrc/libtelnet-0.23/Makefile.am	2018-10-18 15:18:49.000000000 +0900
+++ src/libtelnet-0.23/Makefile.am	2024-10-10 00:05:13.121969500 +0900
@@ -8,7 +8,7 @@ include_HEADERS = libtelnet.h
 
 lib_LTLIBRARIES = libtelnet.la
 libtelnet_la_SOURCES = libtelnet.c libtelnet.h
-libtelnet_la_LDFLAGS = -version-info 2:0:0
+libtelnet_la_LDFLAGS = -no-undefined -version-info 2:0:0
 libtelnet_la_LIBADD = $(zlib_LIBS)
 
 pkgconfigdir = $(libdir)/pkgconfig
--- origsrc/libtelnet-0.23/configure.ac	2018-10-18 15:18:49.000000000 +0900
+++ src/libtelnet-0.23/configure.ac	2024-10-10 00:05:13.121969500 +0900
@@ -7,7 +7,7 @@ AC_INIT([libtelnet], [0.23], [http://git
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_SRCDIR([libtelnet.c])
-AM_INIT_AUTOMAKE([1.9 foreign -Wall -Werror subdir-objects])
+AM_INIT_AUTOMAKE([1.9 foreign -Wall subdir-objects])
 AM_PROG_AR
 #LT_INIT([win32-dll])
 AC_LIBTOOL_WIN32_DLL
--- origsrc/libtelnet-0.23/util/Makefile.am	2018-10-18 15:18:49.000000000 +0900
+++ src/libtelnet-0.23/util/Makefile.am	2024-10-10 00:05:13.121969500 +0900
@@ -1,15 +1,15 @@
 AM_CFLAGS = -I$(top_srcdir)
 
-telnet_client_SOURCES = telnet-client.c ../libtelnet.h
+telnet_client_SOURCES = telnet-client.c $(top_srcdir)/libtelnet.h
 telnet_client_LDADD = ../libtelnet.la
 
-telnet_chatd_SOURCES = telnet-chatd.c ../libtelnet.h
+telnet_chatd_SOURCES = telnet-chatd.c $(top_srcdir)/libtelnet.h
 telnet_chatd_LDADD = ../libtelnet.la
 
-telnet_proxy_SOURCES = telnet-proxy.c ../libtelnet.h
+telnet_proxy_SOURCES = telnet-proxy.c $(top_srcdir)/libtelnet.h
 telnet_proxy_LDADD = ../libtelnet.la
 
-telnet_test_SOURCES = telnet-test.c ../libtelnet.h
+telnet_test_SOURCES = telnet-test.c $(top_srcdir)/libtelnet.h
 telnet_test_LDADD = ../libtelnet.la
 
 if BUILD_UTIL
