--- origsrc/LogMon-0.4.4/LineManip.cc	2006-01-08 03:47:38.000000000 +0900
+++ src/LogMon-0.4.4/LineManip.cc	2012-11-25 23:21:07.633600000 +0900
@@ -2,6 +2,8 @@
 // $Id: LineManip.cc 72 2005-07-12 00:46:02Z amackenz $
 // 
 
+#include <cstdlib>
+
 #include <unistd.h>
 #include <ncurses.h>
 #include <ctype.h>
--- origsrc/LogMon-0.4.4/Makefile.am	2006-01-08 03:47:38.000000000 +0900
+++ src/LogMon-0.4.4/Makefile.am	2012-11-25 23:21:07.633600000 +0900
@@ -1,3 +1,5 @@
 bin_PROGRAMS = logmon
 logmon_SOURCES = Main.cc Frame.cc Util.cc LineManip.cc 
+logmon_CPPFLAGS = $(NCURSES_CFLAGS)
+logmon_LDADD = $(NCURSES_LIBS)
 AM_CXXFLAGS = -Wall
--- origsrc/LogMon-0.4.4/configure.in	2006-01-08 03:47:38.000000000 +0900
+++ src/LogMon-0.4.4/configure.in	2012-11-25 23:21:07.633600000 +0900
@@ -15,7 +15,7 @@ AC_PROG_LN_S
 
 dnl Checks for libraries.
 dnl Replace `main' with a function in -lncurses:
-AC_CHECK_LIB(ncurses, initscr)
+PKG_CHECK_MODULES([NCURSES], [ncurses])
 
 dnl Checks for header files.
 AC_HEADER_STDC
