--- origsrc/libeastwood-0.3/CMakeLists.txt	2009-10-13 18:47:00.000000000 +0900
+++ src/libeastwood-0.3/CMakeLists.txt	2015-03-20 00:11:46.414329000 +0900
@@ -9,7 +9,7 @@ SET(LIBEASTWOOD_MINOR "0")
 SET(LIBEASTWOOD_PATCH "0")
 SET(VERSION "0.3")
 
-SET( CMAKE_CXX_FLAGS "-Wall -Wextra -Weffc++ -pedantic -O3 -Wno-long-long -Wno-variadic-macros -fPIC -g -O0 ${CMAKE_CXX_FLAGS}" )
+SET( CMAKE_CXX_FLAGS "-Wall -Wextra -Weffc++ -pedantic -O3 -Wno-long-long -Wno-variadic-macros -g -O0 ${CMAKE_CXX_FLAGS}" )
 
 SET( LIBEASTWOOD_VERSION_INFO "${VERSION}" )
 SET( LIBEASTWOOD_SOVERSION_INFO "${LIBEASTWOOD_MAJOR}.${LIBEASTWOOD_MINOR}.${LIBEASTWOOD_PATCH}" )
@@ -73,7 +73,7 @@ ENDIF(LIBEASTWOOD_BUILD_PYTHON)
 CONFIGURE_FILE(${LIBEASTWOOD_SOURCE_DIR}/libeastwood.pc.in ${LIBEASTWOOD_BINARY_DIR}/libeastwood.pc @ONLY)
 INSTALL( FILES ${LIBEASTWOOD_BINARY_DIR}/libeastwood.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig )
 
-INSTALL( FILES ${LIBEASTWOOD_SOURCE_DIR}/cmake/modules/FindLibEastwood.cmake DESTINATION ${CMAKE_INSTALL_PREFIX}/share/cmake/Modules )
+INSTALL( FILES ${LIBEASTWOOD_SOURCE_DIR}/cmake/modules/FindLibEastwood.cmake DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/cmake/libwestwood )
 
 ADD_SUBDIRECTORY( src )
 ADD_SUBDIRECTORY( include )
--- origsrc/libeastwood-0.3/include/eastwood/StdDef.h	2009-10-13 18:47:00.000000000 +0900
+++ src/libeastwood-0.3/include/eastwood/StdDef.h	2015-03-20 00:12:56.247336800 +0900
@@ -11,6 +11,8 @@
 #include <cstring>
 #include <tr1/cinttypes>
 
+#include <unistd.h>
+
 #ifdef _WIN32
 #include <tchar.h>
 #else
--- origsrc/libeastwood-0.3/libeastwood.pc.in	2009-10-13 18:47:00.000000000 +0900
+++ src/libeastwood-0.3/libeastwood.pc.in	2015-03-20 00:15:10.069602400 +0900
@@ -1,6 +1,6 @@
 prefix=@CMAKE_INSTALL_PREFIX@
-exec_prefix=@LIB_INSTALL_DIR@
-libdir=@LIB_INSTALL_DIR@
+exec_prefix=@CMAKE_INSTALL_PREFIX@
+libdir=@CMAKE_INSTALL_PREFIX@/lib
 includedir=@CMAKE_INSTALL_PREFIX@/include
 
 Name: @PACKAGE@
--- origsrc/libeastwood-0.3/src/CMakeLists.txt	2009-10-13 18:47:00.000000000 +0900
+++ src/libeastwood-0.3/src/CMakeLists.txt	2015-03-20 00:11:27.117454000 +0900
@@ -49,7 +49,7 @@ SET( SDL_LIBEASTWOOD_SRCS
 ADD_LIBRARY( eastwood SHARED ${LIBEASTWOOD_SRCS} )
 TARGET_LINK_LIBRARIES( eastwood ${LIBSAMPLERATE_LIBRARIES} )
 SET_TARGET_PROPERTIES( eastwood PROPERTIES SOVERSION "${LIBEASTWOOD_SOVERSION_INFO}" )
-INSTALL(TARGETS eastwood LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} )
+INSTALL(TARGETS eastwood RUNTIME DESTINATION bin LIBRARY DESTINATION lib )
 
 
 IF(LIBEASTWOOD_BUILD_SDL)
@@ -58,6 +58,6 @@ TARGET_LINK_LIBRARIES( SDL_eastwood east
 TARGET_LINK_LIBRARIES( eastwood ${SDL_LIBRARY} )
 TARGET_LINK_LIBRARIES( eastwood ${SDLMIXER_LIBRARY} )
 SET_TARGET_PROPERTIES( SDL_eastwood PROPERTIES SOVERSION "${LIBEASTWOOD_SOVERSION_INFO}" )
-INSTALL(TARGETS SDL_eastwood LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} )
+INSTALL(TARGETS SDL_eastwood RUNTIME DESTINATION bin LIBRARY DESTINATION lib )
 ENDIF(LIBEASTWOOD_BUILD_SDL)
 
