--- origsrc/countly-cpp-17.10/CMakeLists.txt	2017-10-27 23:57:00.000000000 +0900
+++ src/countly-cpp-17.10/CMakeLists.txt	2019-03-29 10:20:32.726966300 +0900
@@ -42,6 +42,9 @@ IF(APPLE)
   # use OpenSSL from Homebrew installation due to Apple deprecated its support
   INCLUDE_DIRECTORIES(/usr/local/opt/openssl/include)
   LINK_DIRECTORIES(/usr/local/opt/openssl/lib)
+ELSEIF(CYGWIN)
+  INCLUDE(FindPkgConfig)
+  PKG_CHECK_MODULES(OPENSSL REQUIRED openssl)
 ENDIF()
 
 SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/bin)
@@ -57,3 +60,15 @@ TARGET_LINK_LIBRARIES(test-countly
   ssl
   crypto
 )
+
+IF(CYGWIN)
+  FILE(GLOB headers "include/countly/*.h")
+  INSTALL(FILES ${headers} DESTINATION include/countly)
+ENDIF()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/countly-cpp.pc.in ${CMAKE_BINARY_DIR}/countly-cpp.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/countly-cpp.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/countly-cpp-17.10/countly-cpp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/countly-cpp-17.10/countly-cpp.pc.in	2019-03-29 10:20:09.017961300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: countly-cpp
+Description: Lightweight C++ library for Countly
+Version: @VERSION@
+Libs: -L${libdir} -lCountly
+Cflags: -I${includedir}
--- origsrc/countly-cpp-17.10/lib/CMakeLists.txt	2017-10-27 23:57:00.000000000 +0900
+++ src/countly-cpp-17.10/lib/CMakeLists.txt	2019-03-29 10:20:09.033773200 +0900
@@ -20,3 +20,10 @@ ADD_LIBRARY(Countly
     Event.cpp
     Version.cpp
     )
+
+IF(CYGWIN)
+  TARGET_LINK_LIBRARIES(Countly ${OPENSSL_LIBRARIES})
+  SET_TARGET_PROPERTIES(Countly PROPERTIES SOVERSION 0)
+
+  INSTALL(TARGETS Countly RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+ENDIF()
--- origsrc/countly-cpp-17.10/lib/Device.cpp	2017-10-27 23:57:00.000000000 +0900
+++ src/countly-cpp-17.10/lib/Device.cpp	2019-03-29 10:20:09.033773200 +0900
@@ -39,7 +39,7 @@
 #include <net/if_dl.h>
 #include <ifaddrs.h>
 #include <net/if_types.h>
-#else
+#elif defined(__linux__)
 #include <linux/if.h>
 #include <linux/sockios.h>
 #endif
