--- origsrc/libnfs-libnfs-1.8.0/examples/Makefile.am	2013-10-31 10:17:49.000000000 +0900
+++ src/libnfs-libnfs-1.8.0/examples/Makefile.am	2013-12-22 10:27:16.122406200 +0900
@@ -1,15 +1,15 @@
-noinst_PROGRAMS = nfsclient-async nfsclient-raw nfsclient-sync nfsclient-bcast nfsclient-listservers nfs-ls nfs-cp
+bin_PROGRAMS = nfsclient-async nfsclient-raw nfsclient-sync nfsclient-bcast nfsclient-listservers nfs-ls nfs-cp
 
 AM_CPPFLAGS = \
 	-I$(abs_top_srcdir)/include \
 	-I$(abs_top_srcdir)/include/nfsc \
-	-I../mount \
-	-I../nfs \
-	-I../rquota \
-	-I../portmap \
+	-I$(top_srcdir)/mount \
+	-I$(top_srcdir)/nfs \
+	-I$(top_srcdir)/rquota \
+	-I$(top_srcdir)/portmap \
 	"-D_U_=__attribute__((unused))"
 
-AM_LDFLAGS = ../lib/.libs/libnfs.la -lpopt
+AM_LDFLAGS = $(top_builddir)/lib/libnfs.la -lpopt
 
 nfs_ls_SOURCES = nfs-ls.c
 
--- origsrc/libnfs-libnfs-1.8.0/include/nfsc/libnfs.h	2013-10-31 10:17:49.000000000 +0900
+++ src/libnfs-libnfs-1.8.0/include/nfsc/libnfs.h	2013-12-22 10:20:16.492523400 +0900
@@ -18,7 +18,7 @@
  * This is the highlevel interface to access NFS resources using a posix-like interface
  */
 #include <stdint.h>
-#if defined(ANDROID)
+#if defined(ANDROID) || defined(__CYGWIN__)
 #include <sys/time.h>
 #endif
 #if defined(AROS)
--- origsrc/libnfs-libnfs-1.8.0/lib/Makefile.am	2013-10-31 10:17:49.000000000 +0900
+++ src/libnfs-libnfs-1.8.0/lib/Makefile.am	2013-12-22 10:26:02.147796900 +0900
@@ -2,11 +2,11 @@ lib_LTLIBRARIES = libnfs.la
 
 libnfs_la_CPPFLAGS = -I$(abs_top_srcdir)/include \
 		     -I$(abs_top_srcdir)/include/nfsc \
-		     -I../mount \
-		     -I../nfs \
-		     -I../nlm \
-		     -I../portmap \
-		     -I../rquota \
+		     -I$(top_srcdir)/mount \
+		     -I$(top_srcdir)/nfs \
+		     -I$(top_srcdir)/nlm \
+		     -I$(top_srcdir)/portmap \
+		     -I$(top_srcdir)/rquota \
 		     "-D_U_=__attribute__((unused))"
 
 libnfs_la_SOURCES = \
@@ -17,7 +17,7 @@ libnfs_la_SOURCES = \
 	pdu.c \
 	socket.c
 
-libnfs_la_LDFLAGS = -version-info 1:8:0
+libnfs_la_LDFLAGS = -no-undefined -version-info 1:8:0
 libnfs_la_LIBADD = \
 	../mount/libmount.la \
 	../nfs/libnfs.la \
--- origsrc/libnfs-libnfs-1.8.0/nfs/libnfs-raw-nfs.h	2013-10-31 10:17:49.000000000 +0900
+++ src/libnfs-libnfs-1.8.0/nfs/libnfs-raw-nfs.h	2013-12-22 10:24:40.472015600 +0900
@@ -21,7 +21,7 @@ extern "C" {
 typedef long long int quad_t;
 typedef long long unsigned u_quad_t;
 #endif
-#if defined(WIN32)
+#if defined(WIN32) || defined(__CYGWIN__)
 typedef long long int quad_t;
 typedef long long unsigned u_quad_t;
 #endif
--- origsrc/libnfs-libnfs-1.8.0/nlm/libnfs-raw-nlm.h	2013-10-31 10:17:49.000000000 +0900
+++ src/libnfs-libnfs-1.8.0/nlm/libnfs-raw-nlm.h	2013-12-22 10:21:33.101898400 +0900
@@ -12,7 +12,7 @@
 extern "C" {
 #endif
 
-#if defined(ANDROID)
+#if defined(ANDROID) || defined(__CYGWIN__)
 typedef long long int quad_t;
 typedef long long unsigned u_quad_t;
 #endif
