--- origsrc/fex-1.20100416.2814/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/fex-1.20100416.2814/Makefile.am	2011-11-03 09:19:56.634212800 +0900
@@ -0,0 +1,2 @@
+bin_PROGRAMS = fex
+fex_SOURCES = fex.c
--- origsrc/fex-1.20100416.2814/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/fex-1.20100416.2814/configure.ac	2011-11-03 09:22:20.340444300 +0900
@@ -0,0 +1,31 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([fex], [1.20100416.2814], [http://www.semicomplete.com/projects/fex/])
+AC_CONFIG_SRCDIR([fex.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_MAKE_SET
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([memchr memset snprintf strchr strcspn strdup strtol])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
