--- origsrc/orcania-1.2.3/CMakeLists.txt	2018-04-13 10:36:07.000000000 +0900
+++ src/orcania-1.2.3/CMakeLists.txt	2018-06-15 16:53:00.504883500 +0900
@@ -161,7 +161,7 @@ option(INSTALL_HEADER "Install the heade
 configure_file(liborcania.pc.in liborcania.pc @ONLY)
 
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/liborcania.pc
-        DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/pkgconfig)
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 
 set(TARGETS orcania)
 if (BUILD_STATIC)
