--- origsrc/cppmyth-2.17.6/cppmyth/CMakeLists.txt	2025-06-13 05:02:57.000000000 +0900
+++ src/cppmyth-2.17.6/cppmyth/CMakeLists.txt	2025-10-23 20:09:23.044729500 +0900
@@ -269,7 +269,6 @@ endif ()
 install (FILES
   ${CMAKE_CURRENT_BINARY_DIR}/include/local_config.h
   ${CMAKE_CURRENT_BINARY_DIR}/include/mythlocked.h
-  ${CMAKE_CURRENT_BINARY_DIR}/include/mythintrinsic.h
   ${CMAKE_CURRENT_BINARY_DIR}/include/mythsharedptr.h
   ${CMAKE_CURRENT_BINARY_DIR}/include/mythtypes.h
   ${CMAKE_CURRENT_BINARY_DIR}/include/mythcontrol.h
