--- origsrc/libgdbus-0.2/src/Makefile.am	2008-08-11 05:14:05.000000000 +0900
+++ src/libgdbus-0.2/src/Makefile.am	2012-04-01 13:48:05.960647400 +0900
@@ -4,7 +4,7 @@ include_HEADERS = gdbus.h
 lib_LTLIBRARIES = libgdbus.la
 
 libgdbus_la_SOURCES = debug.h debug.c mainloop.c object.c watch.c
-libgdbus_la_LDFLAGS = -version-info 0:0:0 -export-symbols-regex g_dbus_.*
+libgdbus_la_LDFLAGS = -version-info 0:0:0 -export-symbols-regex g_dbus_.* -no-undefined
 
 libgdbus_la_LIBADD = @GLIB_LIBS@ @DBUS_LIBS@
 
