--- origsrc/solarpowerlog-solarpowerlog-0.26/configure.ac	2023-06-10 01:44:56.000000000 +0900
+++ src/solarpowerlog-solarpowerlog-0.26/configure.ac	2025-02-07 19:34:13.950441900 +0900
@@ -230,8 +230,10 @@ PKG_CHECK_EXISTS
 AC_LANG(C++)
 
 ### Checks for header files.
+if test "$host_os" != "cygwin"; then
 AC_CHECK_HEADER([w32api/wtypes.h],
  [AC_DEFINE([HAVE_WIN32_API], [1], [Windows API]) ])
+fi
 
 if test "x$ac_cv_header_w32api_wtypes_h" = "xyes"; then
   AC_MSG_RESULT([WIN32 API detected. ])
