--- origsrc/snmpkit-0.9/src/Makefile.am	2001-11-22 09:54:53.000000000 +0900
+++ src/snmpkit-0.9/src/Makefile.am	2014-06-08 20:23:17.533987500 +0900
@@ -7,7 +7,7 @@ libsnmpkit_la_SOURCES=ber.C beroid.C ber
 	snmpkit_except snmpkit_tags ber.h oidseq.h snmpsock.h
 
 
-libsnmpkit_la_LDFLAGS= -version-info 2:0:0
+libsnmpkit_la_LDFLAGS= -no-undefined -version-info 2:0:0
 
 include_HEADERS=snmpkit.h snmpkit snmpkit_except snmpkit_tags
 
@@ -17,4 +17,4 @@ LDADD=libsnmpkit.la
 noinst_PROGRAMS=snmptest1 snmptest2 snmptest3 # snmptest4
 snmptest1_SOURCES=snmptest1.C  
 snmptest2_SOURCES=snmptest2.C 
-snmptest3_SOURCES=snmptest3.C
\ No newline at end of file
+snmptest3_SOURCES=snmptest3.C
--- origsrc/snmpkit-0.9/src/session.C	2001-09-22 05:19:17.000000000 +0900
+++ src/snmpkit-0.9/src/session.C	2014-06-08 20:27:42.098225500 +0900
@@ -530,7 +530,7 @@ std::list<SNMP_session*> &SNMP_sessions(
 std::list<SNMP_session*> &SNMP_sessions(std::list<SNMP_session*> &dest,
 				   std::list<std::string> &hostspecs,
 				   void *(*start_routine)(SNMP_session*),
-				   const std::string &community="public")
+				   const std::string &community)
   throw(std::bad_alloc,SocketNoUDPException,SocketCreateFailException,
 	ReceiverCreateException,SessionHostNotFoundException,
 	JoinerCreateException,SessionWorkerCreateException,
--- origsrc/snmpkit-0.9/src/snmpkit_interface.C	2001-09-22 05:19:17.000000000 +0900
+++ src/snmpkit-0.9/src/snmpkit_interface.C	2014-06-09 20:17:27.693625000 +0900
@@ -30,6 +30,8 @@
 
 #include "errno.h"
 
+#include <cstring>
+
 SNMPERRNO sk_errno;
 char *sk_erroid;
 
@@ -506,7 +508,7 @@ int sk_sfiller_remove(SNMPTABLE *st,cons
 void **sk_table_get(SNMPTABLE *st){
   std::list<void*> vals;
   from_c(st)->get(vals);
-  void **retval=new (void*)[vals.size()+1];
+  void **retval=new void*[vals.size()+1];
   retval[vals.size()]=NULL;
   int i=0;
   for(std::list<void*>::iterator cur=vals.begin();cur!=vals.end();cur++,i++)
--- origsrc/snmpkit-0.9/src/snmpsock.C	2001-09-22 05:19:17.000000000 +0900
+++ src/snmpkit-0.9/src/snmpsock.C	2014-06-08 20:26:03.965822500 +0900
@@ -102,7 +102,7 @@ void *receiver(void *sockp)
   throw(std::bad_alloc,ReceiverSelectException,ReceiverReadException){
   int sock=*(int*)sockp;
   int readcnt;
-  unsigned int fromlen;
+  socklen_t fromlen;
   for(;;){
     sockaddr_in from;
     memset(&from,0,sizeof(from));
--- origsrc/snmpkit-0.9/src/snmptest1.C	2001-08-01 08:37:15.000000000 +0900
+++ src/snmpkit-0.9/src/snmptest1.C	2014-06-09 20:18:25.256125000 +0900
@@ -27,7 +27,11 @@
 #include <snmpkit>
 
 #include <string.h>
-#include <iostream.h>
+#include <iostream>
+
+#include <cstdlib>
+
+using namespace std;
 
 static const char SYSDESC[]="1.3.6.1.2.1.1.1.0";
 
--- origsrc/snmpkit-0.9/src/snmptest2.C	2001-09-22 05:19:17.000000000 +0900
+++ src/snmpkit-0.9/src/snmptest2.C	2014-06-09 20:19:16.834250000 +0900
@@ -27,7 +27,10 @@
 #include <snmpkit>
 
 #include <string.h>
-#include <iostream.h>
+#include <cstdlib>
+#include <iostream>
+
+using namespace std;
 
 static const char SYSDESC[]="1.3.6.1.2.1.1.1.0";
 
--- origsrc/snmpkit-0.9/src/snmptest3.C	2001-09-22 05:19:17.000000000 +0900
+++ src/snmpkit-0.9/src/snmptest3.C	2014-06-09 20:19:56.068625000 +0900
@@ -27,9 +27,12 @@
 #include <snmpkit>
 
 #include <string.h>
-#include <iostream.h>
+#include <cstdlib>
+#include <iostream>
 #include <string>
 
+using namespace std;
+
 const std::string HPSTR("JETDIRECT");
 const std::string LEXMARKSTR("Lexmark");
 const std::string TEKTRONIXSTR("Tektronix");
--- origsrc/snmpkit-0.9/src/structfill.C	2001-09-22 05:19:17.000000000 +0900
+++ src/snmpkit-0.9/src/structfill.C	2014-06-08 20:26:44.223225500 +0900
@@ -31,12 +31,14 @@
 
 /* would prefer <iostream> but some linux systems seem to be 
    missing it */
-#include <iostream.h>
+#include <iostream>
 #include <algorithm>
 
 #include "snmpkit"
 #include "oidseq.h"
 
+using namespace std;
+
 void SNMP_structFiller::append(const std::string &oidstr,Tags tag,
 			       void (*fp)())
   throw(std::bad_alloc,BerOidBadSubOidException,BerNoOidsException){
