--- origsrc/belr-5.3.97/CMakeLists.txt	2024-12-12 14:14:59.832374400 +0900
+++ src/belr-5.3.97/CMakeLists.txt	2024-12-12 14:15:22.507384100 +0900
@@ -23,7 +23,7 @@
 
 cmake_minimum_required(VERSION 3.22)
 
-project(Belr VERSION 5.3.0 LANGUAGES C CXX)
+project(Belr VERSION 5.3.97 LANGUAGES C CXX)
 
 set(BELR_SO_VERSION "1")
 
--- origsrc/belr-5.3.97/src/CMakeLists.txt	2024-07-18 16:33:12.000000000 +0900
+++ src/belr-5.3.97/src/CMakeLists.txt	2024-12-12 16:00:11.631880600 +0900
@@ -40,7 +40,7 @@ if(WIN32 AND NOT CMAKE_SYSTEM_NAME STREQ
 endif()
 
 add_library(belr ${BELR_HEADER_FILES} ${BELR_SOURCE_FILES_C} ${BELR_SOURCE_FILES_CXX})
-set_target_properties(belr PROPERTIES VERSION ${BELR_SO_VERSION})
+set_target_properties(belr PROPERTIES SOVERSION ${BELR_SO_VERSION})
 target_include_directories(belr INTERFACE
 	$<INSTALL_INTERFACE:include>
 	$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
