--- origsrc/belr-5.0.23/CMakeLists.txt	2021-09-08 01:17:31.000000000 +0900
+++ src/belr-5.0.23/CMakeLists.txt	2021-09-17 09:35:09.188391000 +0900
@@ -155,3 +155,6 @@ install(FILES
 if (ENABLE_PACKAGE_SOURCE)
 	add_subdirectory(build)
 endif()
+
+configure_file(${CMAKE_SOURCE_DIR}/belr.pc.in ${CMAKE_BINARY_DIR}/belr.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/belr.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/belr-5.0.23/src/CMakeLists.txt	2021-09-08 01:17:31.000000000 +0900
+++ src/belr-5.0.23/src/CMakeLists.txt	2021-09-17 09:34:28.760211100 +0900
@@ -68,7 +68,7 @@ if(ENABLE_SHARED)
 			PUBLIC_HEADER "${BELR_HEADER_FILES}"
 		)
 	endif()
-	set_target_properties(belr PROPERTIES VERSION ${BELR_SO_VERSION})
+	set_target_properties(belr PROPERTIES SOVERSION ${BELR_SO_VERSION})
 	target_include_directories(belr INTERFACE
 		$<INSTALL_INTERFACE:include>
 		$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
