--- origsrc/belr-5.4.117/CMakeLists.txt	2026-02-25 00:11:45.000000000 +0900
+++ src/belr-5.4.117/CMakeLists.txt	2026-05-27 06:25:01.515325500 +0900
@@ -23,7 +23,7 @@
 
 cmake_minimum_required(VERSION 3.22)
 
-project(Belr VERSION 5.4.0 LANGUAGES C CXX)
+project(Belr VERSION 5.4.117 LANGUAGES C CXX)
 
 set(BELR_SO_VERSION "1")
 
@@ -81,7 +81,7 @@ if(ENABLE_TOOLS)
 endif()
 
 include(CMakePackageConfigHelpers)
-set(CMAKE_MODULES_INSTALL_DIR "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+option(CMAKE_MODULES_INSTALL_DIR "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
 configure_package_config_file("${PROJECT_NAME}Config.cmake.in" "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
 	INSTALL_DESTINATION "${CMAKE_MODULES_INSTALL_DIR}"
 	NO_SET_AND_CHECK_MACRO
@@ -104,3 +104,6 @@ install(EXPORT ${PROJECT_NAME}Targets
 if (ENABLE_PACKAGE_SOURCE)
 	add_subdirectory(build)
 endif()
+
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/belr.pc.in ${CMAKE_CURRENT_BINARY_DIR}/belr.pc)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/belr.pc DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
--- origsrc/belr-5.4.117/src/CMakeLists.txt	2026-02-25 00:11:45.000000000 +0900
+++ src/belr-5.4.117/src/CMakeLists.txt	2026-05-26 23:33:10.302886400 +0900
@@ -38,7 +38,7 @@ if(WIN32 AND NOT CMAKE_SYSTEM_NAME STREQ
 endif()
 
 add_library(belr ${BELR_HEADER_FILES} ${BELR_SOURCE_FILES_CXX})
-set_target_properties(belr PROPERTIES VERSION ${BELR_SO_VERSION})
+set_target_properties(belr PROPERTIES SOVERSION ${BELR_SO_VERSION})
 target_include_directories(belr INTERFACE
 	$<INSTALL_INTERFACE:include>
 	$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
