--- origsrc/libroxml-3.0.2/Makefile.am	2019-06-22 18:51:25.000000000 +0900
+++ src/libroxml-3.0.2/Makefile.am	2025-12-25 13:21:16.566249800 +0900
@@ -70,7 +70,7 @@ libroxml_la_SOURCES += src/roxml_xpath.c
 libroxml_la_CFLAGS += -DCONFIG_XML_XPATH
 endif
 
-libroxml_la_LDFLAGS = $(VERSION_INFO)
+libroxml_la_LDFLAGS = -no-undefined $(VERSION_INFO)
 
 bin_PROGRAMS =
 
--- origsrc/libroxml-3.0.2/src/roxml_mem.h	2019-06-22 18:51:25.000000000 +0900
+++ src/libroxml-3.0.2/src/roxml_mem.h	2025-12-25 13:22:07.808416400 +0900
@@ -14,7 +14,7 @@
 
 #include "roxml_internal.h"
 
-memory_cell_t head_cell;
+extern memory_cell_t head_cell;
 
 /** \brief alloc memory function
  *
