--- origsrc/libroxml-2.2.0/CMakeLists.txt	2012-06-12 06:16:31.000000000 +0900
+++ src/libroxml-2.2.0/CMakeLists.txt	2012-08-11 21:57:52.893407900 +0900
@@ -56,7 +56,7 @@ set_target_properties (roxml-tool
 		OUTPUT_NAME roxml
 	)
 
-install (TARGETS roxml-dynamic roxml-static DESTINATION lib/)
+install (TARGETS roxml-dynamic RUNTIME DESTINATION bin/ ARCHIVE DESTINATION lib/)
 install (TARGETS roxml-tool DESTINATION bin/)
 install (FILES "${TOPDIR}inc/roxml.h" DESTINATION include/)
 
