--- origsrc/libroxml-2.2.2/CMakeLists.txt	2013-06-08 07:26:58.000000000 +0900
+++ src/libroxml-2.2.2/CMakeLists.txt	2013-08-03 12:16:05.527829900 +0900
@@ -56,7 +56,7 @@ set_target_properties (roxml-tool
 		OUTPUT_NAME roxml
 	)
 
-install (TARGETS roxml-dynamic roxml-static DESTINATION lib/)
+install (TARGETS roxml-dynamic RUNTIME DESTINATION bin/ ARCHIVE DESTINATION lib/)
 install (TARGETS roxml-tool DESTINATION bin/)
 install (FILES "${TOPDIR}inc/roxml.h" DESTINATION include/)
 
