--- origsrc/ferite-1.1.17/modules/regexp/regexp.fec	2009-07-16 02:21:34.000000000 +0900
+++ src/ferite-1.1.17/modules/regexp/regexp.fec	2022-06-30 11:21:12.710580500 +0900
@@ -427,7 +427,7 @@ class Regexp {
 		}
 		
 		/* get the number of subparts */
-		captured_str_cnt = pcre_info( rgx->compiled_re, NULL, NULL ) + 1;
+		pcre_fullinfo( rgx->compiled_re, NULL, PCRE_INFO_CAPTURECOUNT, &captured_str_cnt ) + 1;
 		/* create an offset array */
 		size_offsets = (int)(captured_str_cnt * 3);
 		offsets = (int *)fmalloc(size_offsets * sizeof(int));
