--- origsrc/libexecstream/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libexecstream/Makefile.am	2015-02-16 10:06:38.005021500 +0900
@@ -0,0 +1,15 @@
+
+check_PROGRAMS = exec-stream-test
+exec_stream_test_SOURCES = test/exec-stream-test.cpp
+exec_stream_test_LDADD = libexecstream.la
+
+include_HEADERS = exec-stream.h
+
+lib_LTLIBRARIES = libexecstream.la
+libexecstream_la_SOURCES = exec-stream.cpp
+libexecstream_la_LDFLAGS = -no-undefined
+
+TESTS = $(check_PROGRAMS)
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = libexecstream.pc
--- origsrc/libexecstream/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/libexecstream/configure.ac	2015-02-16 10:06:20.458146500 +0900
@@ -0,0 +1,31 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([libexecstream], [0.3])
+AC_CONFIG_SRCDIR([exec-stream.cpp])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE([subdir-objects])
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([fcntl.h stdlib.h string.h sys/time.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+AC_C_INLINE
+AC_TYPE_PID_T
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_FORK
+AC_CHECK_FUNCS([gettimeofday select strerror])
+
+AC_CONFIG_FILES([Makefile libexecstream.pc])
+AC_OUTPUT
--- origsrc/libexecstream/libexecstream.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libexecstream/libexecstream.pc.in	2015-02-16 10:06:06.005021500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libexecstream
+Description: C++ library to run a child process with standard C++ streams
+Version: @VERSION@
+Libs: -L${libdir} -lexecstream
+Cflags: -I${includedir}
--- origsrc/libexecstream/test/exec-stream-test.cpp	2004-04-12 05:17:24.000000000 +0900
+++ src/libexecstream/test/exec-stream-test.cpp	2015-02-16 09:57:40.851265600 +0900
@@ -38,6 +38,10 @@ EVEN IF ADVISED OF THE POSSIBILITY OF SU
 #include <stdlib.h>
 #include <ctype.h>
 
+#include <climits>
+#include <cstdio>
+#include <cstring>
+
 #ifdef _WIN32
 
 #include <windows.h>
