--- origsrc/CBFlib-0.9.2.4/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/CBFlib-0.9.2.4/Makefile.am	2012-03-28 11:20:44.111713800 +0900
@@ -0,0 +1,64 @@
+
+INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/examples
+
+
+bin_PROGRAMS = adscimg2cbf cbf2adscimg adscimg2cbf \
+convert_image convert_minicbf makecbf img2cif cif2cbf cif2c
+
+adscimg2cbf_SOURCES = examples/adscimg2cbf.c examples/adscimg2cbf_sub.c
+adscimg2cbf_LDADD = libcbf.la
+cbf2adscimg_SOURCES = examples/cbf2adscimg.c examples/cbf2adscimg_sub.c
+cbf2adscimg_LDADD = libcbf.la
+convert_image_SOURCES = examples/convert_image.c
+convert_image_LDADD = libcbf.la
+convert_minicbf_SOURCES = examples/convert_minicbf.c
+convert_minicbf_LDADD = libcbf.la
+makecbf_SOURCES = examples/makecbf.c
+makecbf_LDADD = libcbf.la
+img2cif_SOURCES = examples/img2cif.c
+img2cif_LDADD = libcbf.la
+cif2cbf_SOURCES = examples/cif2cbf.c
+cif2cbf_LDADD = libcbf.la
+cif2c_SOURCES = examples/cif2c.c
+cif2c_LDADD = libcbf.la
+
+
+cbf_includedir = $(includedir)/cbf
+cbf_include_HEADERS = examples/img.h \
+include/cbf.h              include/cbf_alloc.h \
+include/cbf_ascii.h        include/cbf_binary.h \
+include/cbf_byte_offset.h  include/cbf_canonical.h \
+include/cbf_codes.h        include/cbf_compress.h \
+include/cbf_context.h      include/cbf_copy.h \
+include/cbf_file.h         include/cbf_getopt.h \
+include/cbf_lex.h          include/cbf_packed.h \
+include/cbf_predictor.h    include/cbf_read_binary.h \
+include/cbf_read_mime.h    include/cbf_simple.h \
+include/cbf_string.h       include/cbf_stx.h \
+include/cbf_tree.h         include/cbf_uncompressed.h \
+include/cbf_write.h        include/cbf_write_binary.h \
+include/cbf_ws.h           include/cbff.h \
+include/global.h           include/md5.h
+
+
+lib_LTLIBRARIES = libcbf.la libfcb.la
+
+libcbf_la_CPPFLAGS = -DCBF_DONT_USE_LONG_LONG -D_USE_XOPEN_EXTENDED
+libcbf_la_LDFLAGS = -no-undefined
+libcbf_la_SOURCES = examples/img.c \
+src/cbf.c               src/cbf_alloc.c        src/cbf_ascii.c \
+src/cbf_binary.c        src/cbf_byte_offset.c  src/cbf_canonical.c \
+src/cbf_codes.c         src/cbf_compress.c     src/cbf_context.c \
+src/cbf_copy.c          src/cbf_file.c         src/cbf_getopt.c \
+src/cbf_lex.c           src/cbf_packed.c       src/cbf_predictor.c \
+src/cbf_read_binary.c   src/cbf_read_mime.c    src/cbf_simple.c \
+src/cbf_string.c        src/cbf_stx.c          src/cbf_tree.c \
+src/cbf_uncompressed.c  src/cbf_write.c        src/cbf_write_binary.c \
+src/cbf_ws.c            src/cbff.c             src/md5c.c
+
+libfcb_la_FCFLAGS = -fno-range-check
+libfcb_la_LDFLAGS = -no-undefined
+libfcb_la_SOURCES = \
+src/fcb_atol_wcnt.f90  src/fcb_ci_strncmparr.f90  src/fcb_nblen_array.f90 \
+src/fcb_read_byte.f90  src/fcb_read_line.f90      src/fcb_skip_whitespace.f90
+
--- origsrc/CBFlib-0.9.2.4/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/CBFlib-0.9.2.4/configure.ac	2012-03-28 11:18:02.033588800 +0900
@@ -0,0 +1,38 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([CBFlib], [0.9.2.4])
+AC_CONFIG_SRCDIR([examples/adscimg2cbf.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_YACC
+AC_PROG_CC
+AC_PROG_CPP
+AC_PROG_FC
+AC_PROG_INSTALL
+AC_PROG_LN_S
+AC_PROG_MAKE_SET
+LT_INIT([disable-static])
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_FUNC_ALLOCA
+AC_CHECK_HEADERS([float.h libintl.h limits.h malloc.h stddef.h stdlib.h string.h unistd.h wchar.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_HEADER_STDBOOL
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_FUNC_STRTOD
+AC_CHECK_FUNCS([floor memmove memset pow regcomp rint sqrt strcasecmp strchr strcspn strerror strncasecmp strpbrk strspn strstr strtol strtoul])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
