--- origsrc/libcdatetime-20240104/Makefile.am	2024-01-04 13:59:55.000000000 +0900
+++ src/libcdatetime-20240104/Makefile.am	2024-02-05 17:45:21.279074900 +0900
@@ -3,9 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
 	libcdatetime \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libcdatetime-20240104/configure.ac	2024-01-04 13:59:55.000000000 +0900
+++ src/libcdatetime-20240104/configure.ac	2024-02-05 17:45:21.279074900 +0900
@@ -91,7 +91,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
 AC_CONFIG_FILES([libcdatetime/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
--- origsrc/libcdatetime-20240104/libcdatetime/Makefile.am	2024-01-04 13:59:55.000000000 +0900
+++ src/libcdatetime-20240104/libcdatetime/Makefile.am	2024-02-05 17:45:21.279074900 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libcdatetime-20240104/tests/Makefile.am	2024-01-04 13:59:55.000000000 +0900
+++ src/libcdatetime-20240104/tests/Makefile.am	2024-02-05 17:45:21.279074900 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
