--- origsrc/libcdatetime-20240414/Makefile.am	2024-04-14 13:11:24.000000000 +0900
+++ src/libcdatetime-20240414/Makefile.am	2025-05-15 20:35:06.072645900 +0900
@@ -3,9 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
 	libcdatetime \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libcdatetime-20240414/configure.ac	2024-04-14 13:11:24.000000000 +0900
+++ src/libcdatetime-20240414/configure.ac	2025-05-15 20:35:06.072645900 +0900
@@ -91,7 +91,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
 AC_CONFIG_FILES([libcdatetime/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
