--- origsrc/ifp-line-0.3/configure.ac	2005-12-04 01:29:03.000000000 +0900
+++ src/ifp-line-0.3/configure.ac	2024-12-20 10:00:06.564963200 +0900
@@ -14,7 +14,7 @@ AC_PROG_INSTALL
 
 # Checks for libraries.
 AM_ICONV
-AM_PATH_LIBUSB(0.1.7, , AC_MSG_ERROR([libusb is required.]))
+PKG_CHECK_MODULES([LIBUSB], [libusb])
 
 # Checks for header files.
 AC_HEADER_STDC
--- origsrc/ifp-line-0.3/ifp_routines.c	2005-12-04 01:55:06.000000000 +0900
+++ src/ifp-line-0.3/ifp_routines.c	2024-12-20 10:02:19.633561800 +0900
@@ -10,6 +10,7 @@ static const char rcsid[] = "$Id: ifp_ro
 #include <errno.h>
 #include <fts.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include "unicodehack.h"
 #include "ifp_routines.h"
