--- origsrc/orgadoc-1.3/Makefile.in	2023-03-27 19:33:58.000000000 +0900
+++ src/orgadoc-1.3/Makefile.in	2024-04-01 01:38:00.086885000 +0900
@@ -23,7 +23,7 @@ MISCFILES   = INSTALL AUTHORS
 #   doc files
 DOCFILES    = ChangeLog COPYING NEWS README
 
-SUBDIRS     = src utils docs
+SUBDIRS     = src utils # docs
 
 all:
 	for subdir in $(SUBDIRS); do \
--- origsrc/orgadoc-1.3/docs/Makefile.in	2023-03-27 19:33:58.000000000 +0900
+++ src/orgadoc-1.3/docs/Makefile.in	2024-04-01 01:34:22.873970900 +0900
@@ -31,6 +31,7 @@ install:
 	texi2html $(DOCUMENT1)
 	texi2html $(DOCUMENT2)
 	mkdir -p $(DESTDIR)$(DOCDIR)
+	mkdir -p $(DESTDIR)$(mandir)
 	-cp $(DOCFILES) $(DESTDIR)$(DOCDIR)
 	-cp $(MANDIR) $(DESTDIR)$(mandir)
 
