--- origsrc/qxmpp-1.2.0/src/CMakeLists.txt	2020-02-06 22:04:09.000000000 +0900
+++ src/qxmpp-1.2.0/src/CMakeLists.txt	2020-02-16 15:41:33.491152700 +0900
@@ -247,8 +247,9 @@ target_link_libraries(qxmpp
 
 install(
     TARGETS qxmpp
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}"
     EXPORT QXmppTarget
+    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
 )
 
 install(
