--- origsrc/faac-1.29.9.2/Makefile.am	2017-09-23 21:44:13.000000000 +0900
+++ src/faac-1.29.9.2/Makefile.am	2019-01-16 15:01:49.984625900 +0900
@@ -1 +1,4 @@
 SUBDIRS = include libfaac common frontend
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/faac-1.29.9.2/configure.ac	2017-11-14 23:50:07.000000000 +0900
+++ src/faac-1.29.9.2/configure.ac	2019-01-16 15:01:49.989767500 +0900
@@ -59,6 +59,7 @@ AC_CHECK_TYPES([in_port_t, socklen_t], ,
 AC_CHECK_MEMBERS(fpos_t.__pos,,, [#include <stdio.h>])
 
 AC_OUTPUT([
+    faac.pc
 common/Makefile
 libfaac/Makefile
 frontend/Makefile
--- origsrc/faac-1.29.9.2/faac.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/faac-1.29.9.2/faac.pc.in	2019-01-16 15:01:49.998578800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: faac
+Description: MPEG-4 and MPEG-2 AAC encoder
+Version: @VERSION@
+Libs: -L${libdir} -lfaac_drm
+Cflags: -I${includedir}
--- origsrc/faac-1.29.9.2/libfaac/Makefile.am	2017-11-02 19:48:24.000000000 +0900
+++ src/faac-1.29.9.2/libfaac/Makefile.am	2019-01-16 15:01:50.016396200 +0900
@@ -12,16 +12,19 @@ libfaac_drm_la_SOURCES = ${common_SOURCE
 EXTRA_libfaac_drm_la_SOURCES = kiss_fft/_kiss_fft_guts.h kiss_fft/kiss_fftr.h kiss_fft/README kiss_fft/README.kiss_fft
 libfaac_drm_la_LIBADD = ${common_LIBADD}
 libfaac_drm_la_CFLAGS = ${common_CFLAGS} -DDRM
+libfaac_drm_la_LDFLAGS = -no-undefined
+if MINGW
+libfaac_drm_la_LDFLAGS += -Wl,--add-stdcall-alias
+endif
 else
 lib_LTLIBRARIES = libfaac.la
 libfaac_la_SOURCES = ${common_SOURCES} ${common_INCLUDES}
 libfaac_la_LIBADD = ${common_LIBADD}
 libfaac_la_CFLAGS = ${common_CFLAGS}
-endif
-
 libfaac_la_LDFLAGS = -no-undefined
 if MINGW
 libfaac_la_LDFLAGS += -Wl,--add-stdcall-alias
 endif
+endif
 
 AM_CPPFLAGS = -I$(top_srcdir)/include
