--- origsrc/oc-1.0/Makefile.am	2009-08-14 00:25:56.000000000 +0900
+++ src/oc-1.0/Makefile.am	2012-02-11 22:26:30.215826900 +0900
@@ -52,7 +52,7 @@ liboc_la_SOURCES = config.h \
 	ocdrno.c ocxdr_stdio.c \
 	${IDLSRC} ${IDLHDRS}
 
-liboc_la_LDFLAGS = -version-info $(LIBOC_VERSION) @LDFLAGS@
+liboc_la_LDFLAGS = -version-info $(LIBOC_VERSION) @LDFLAGS@ -no-undefined
 
 pkginclude_HEADERS = i.h oc.h
 noinst_HEADERS = occontent.h ocdata.h oc.h ocinternal.h ocnode.h \
