--- origsrc/oc-1.0/Makefile.am	2009-08-14 00:25:56.000000000 +0900
+++ src/oc-1.0/Makefile.am	2010-08-01 01:53:29.673418800 +0900
@@ -52,7 +52,7 @@ liboc_la_SOURCES = config.h \
 	ocdrno.c ocxdr_stdio.c \
 	${IDLSRC} ${IDLHDRS}
 
-liboc_la_LDFLAGS = -version-info $(LIBOC_VERSION) @LDFLAGS@
+liboc_la_LDFLAGS = -version-info $(LIBOC_VERSION) @LDFLAGS@ -no-undefined
 
 pkginclude_HEADERS = i.h oc.h
 noinst_HEADERS = occontent.h ocdata.h oc.h ocinternal.h ocnode.h \
--- origsrc/oc-1.0/configure.ac	2009-08-20 21:09:49.000000000 +0900
+++ src/oc-1.0/configure.ac	2010-08-01 01:53:44.126543800 +0900
@@ -17,6 +17,7 @@ AC_SUBST(host)
 
 # Checks for programs.
 AC_PROG_CC
+AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
 AC_PROG_MAKE_SET
 AC_PROG_INSTALL
