--- origsrc/eqc-1.4.12/Makefile.am	2012-07-20 22:28:22.000000000 +0900
+++ src/eqc-1.4.12/Makefile.am	2014-06-14 15:48:37.030331000 +0900
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS = src docs examples
+SUBDIRS = src examples
 
 # pkg-config metadata
 pkgconfigdir = $(libdir)/pkgconfig
--- origsrc/eqc-1.4.12/src/Makefile.am	2014-02-25 04:29:26.000000000 +0900
+++ src/eqc-1.4.12/src/Makefile.am	2014-06-14 15:57:22.496128500 +0900
@@ -7,6 +7,7 @@ compprefix=libeqc_la-
 libeqc_la_SOURCES = eqc.cpp equation.cpp expression.cpp func.cpp message.cpp ncsymbol.cpp\
 			operands.cpp unit.cpp printing.cpp utils.cpp optstack.cpp msgdriver.cpp\
 			stringex.cpp extintegral.cpp differential.cpp
+libeqc_la_LDFLAGS = -no-undefined
 libeqc_HEADERS = eqc.h equation.h expression.h func.h message.h operands.h ncsymbol.h\
 			unit.h printing.h utils.h optstack.h msgdriver.h version.h\
 			stringex.h extintegral.h differential.h
@@ -27,4 +28,4 @@ AM_CPPFLAGS = $(all_includes)
 # the library search path.
 #eqc_LDFLAGS = $(all_libraries) 
 noinst_HEADERS = ltxfilelex.h ltxfileparse.h
-eqc_LDADD = libeqc.la
+eqc_LDADD = libeqc.la -largp
