--- origsrc/eqc-1.4.5/Makefile.am	2011-02-01 22:31:21.000000000 +0900
+++ src/eqc-1.4.5/Makefile.am	2012-03-29 21:10:42.410290600 +0900
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS = src docs examples
+SUBDIRS = src examples
 
 # pkg-config metadata
 pkgconfigdir = $(libdir)/pkgconfig
--- origsrc/eqc-1.4.5/src/Makefile.am	2011-02-23 17:26:17.000000000 +0900
+++ src/eqc-1.4.5/src/Makefile.am	2012-03-29 21:07:54.942517200 +0900
@@ -6,6 +6,7 @@ lib_LTLIBRARIES=libeqc.la
 compprefix=libeqc_la-
 libeqc_la_SOURCES = eqc.cpp equation.cpp expression.cpp func.cpp message.cpp ncsymbol.cpp\
 			operands.cpp unit.cpp printing.cpp utils.cpp optstack.cpp msgdriver.cpp
+libeqc_la_LDFLAGS = -no-undefined
 libeqc_HEADERS = eqc.h equation.h expression.h func.h message.h operands.h ncsymbol.h\
 			unit.h printing.h utils.h optstack.h msgdriver.h version.h
 libeqcdir = $(includedir)/libeqc
@@ -25,4 +26,4 @@ INCLUDES= $(all_includes)
 # the library search path.
 #eqc_LDFLAGS = $(all_libraries) 
 noinst_HEADERS = ltxfilelex.h ltxfileparse.h
-eqc_LDADD = libeqc.la
+eqc_LDADD = libeqc.la -largp
