--- origsrc/eqc/Makefile.am	2014-11-08 22:31:28.000000000 +0900
+++ src/eqc/Makefile.am	2020-11-09 18:07:45.777743300 +0900
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS = src docs examples
+SUBDIRS = src examples
 
 # pkg-config metadata
 pkgconfigdir = $(libdir)/pkgconfig
--- origsrc/eqc/src/Makefile.am	2017-01-16 04:58:41.000000000 +0900
+++ src/eqc/src/Makefile.am	2020-11-09 18:05:45.065368200 +0900
@@ -7,6 +7,7 @@ compprefix=libeqc_la-
 libeqc_la_SOURCES = eqc.cpp equation.cpp expression.cpp func.cpp message.cpp extsymbol.cpp\
 			operands.cpp unit.cpp printing.cpp utils.cpp optstack.cpp msgdriver.cpp\
 			stringex.cpp extintegral.cpp differential.cpp
+libeqc_la_LDFLAGS = -no-undefined
 libeqc_HEADERS = eqc.h equation.h expression.h func.h message.h operands.h extsymbol.h\
 			unit.h printing.h utils.h optstack.h msgdriver.h\
 			stringex.h extintegral.h differential.h
@@ -27,4 +28,4 @@ AM_CPPFLAGS = $(all_includes)
 # the library search path.
 #eqc_LDFLAGS = $(all_libraries) 
 noinst_HEADERS = ltxfilelex.h ltxfileparse.h
-eqc_LDADD = libeqc.la
+eqc_LDADD = libeqc.la -largp
