--- origsrc/tree-sitter-php-0.24.2/CMakeLists.txt	2025-08-18 14:11:55.000000000 +0900
+++ src/tree-sitter-php-0.24.2/CMakeLists.txt	2025-09-13 18:37:39.160850000 +0900
@@ -20,12 +20,14 @@ find_program(TREE_SITTER_CLI tree-sitter
 include(GNUInstallDirs)
 
 macro(add_parser name)
+if (NOT TREE_SITTER_CLI STREQUAL "TREE_SITTER_CLI-NOTFOUND")
     add_custom_command(OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/src/parser.c"
                        DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/src/grammar.json"
                        COMMAND "${TREE_SITTER_CLI}" generate src/grammar.json
                                 --abi=${TREE_SITTER_ABI_VERSION}
                        WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
                        COMMENT "Generating parser.c")
+endif()
 
     add_library(tree-sitter-${name}
                 "${CMAKE_CURRENT_SOURCE_DIR}/src/parser.c"
@@ -46,10 +48,10 @@ macro(add_parser name)
     configure_file("${CMAKE_SOURCE_DIR}/bindings/c/tree-sitter-${name}.pc.in"
                    "${CMAKE_CURRENT_BINARY_DIR}/tree-sitter-${name}.pc" @ONLY)
 
-    install(FILES "${CMAKE_SOURCE_DIR}/bindings/c/tree-sitter-${name}.h"
+    install(FILES "${CMAKE_SOURCE_DIR}/bindings/c/tree_sitter/tree-sitter-${name}.h"
             DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/tree_sitter")
     install(FILES "${CMAKE_CURRENT_BINARY_DIR}/tree-sitter-${name}.pc"
-            DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/pkgconfig")
+            DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
     install(TARGETS tree-sitter-${name}
             LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 endmacro()
