--- origsrc/matio-1.5.23/src/safe-math.h	2022-04-01 06:17:14.000000000 +0900
+++ src/matio-1.5.23/src/safe-math.h	2022-05-29 15:12:20.866921600 +0900
@@ -127,7 +127,9 @@
 #    pragma warning(push)
 #    pragma warning(disable:4005)
 #  endif
-#  include <intsafe.h>
+#  if !defined(__CYGWIN__)
+#    include <intsafe.h>
+#  endif
 #  if defined(_MSC_VER) && (_MSC_VER == 1600)
 #    pragma warning(pop)
 #  endif
