--- origsrc/libdshconfig-0.20.13/Makefile.am	2007-08-15 09:03:09.000000000 +0900
+++ src/libdshconfig-0.20.13/Makefile.am	2025-01-16 22:15:33.148858100 +0900
@@ -7,7 +7,7 @@ INCLUDES = \
 DSHCONFIG_SONAME = 1:0:0
 
 lib_LTLIBRARIES = libdshconfig.la
-libdshconfig_la_LDFLAGS = -export-dynamic -version-info $(DSHCONFIG_SONAME) @vsymldflags@
+libdshconfig_la_LDFLAGS = -export-dynamic -version-info $(DSHCONFIG_SONAME) @vsymldflags@ -no-undefined
 include_HEADERS = libdshconfig.h
 test_dshconfig_SOURCES = test-dshconfig.c
 test_dshconfig_LDADD = libdshconfig.la
@@ -51,3 +51,6 @@ viper: distcheck
 upload-dist-all: distcheck
 	scp @PACKAGE@-@VERSION@.tar.gz aegis.netfort.gr.jp:public_html/software/downloads
 	$(MAKE) ../@PACKAGE@_@VERSION@.orig.tar.gz
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libdshconfig-0.20.13/configure.ac	2007-08-15 08:30:23.000000000 +0900
+++ src/libdshconfig-0.20.13/configure.ac	2025-01-16 22:15:33.148858100 +0900
@@ -31,6 +31,6 @@ vsymldflags="-Wl,--version-script,dsh.ve
 AC_SUBST(vsymldflags)
 AC_MSG_RESULT([${vsymldflags}])
 
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([libdshconfig.pc Makefile
 libdshconfig.dox])
 AC_OUTPUT
--- origsrc/libdshconfig-0.20.13/libdshconfig.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libdshconfig-0.20.13/libdshconfig.pc.in	2025-01-16 22:15:33.148858100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libdshconfig
+Description: Library for parsing dsh-style configuration files
+Version: @VERSION@
+Libs: -L${libdir} -ldshconfig
+Cflags: -I${includedir}
