--- origsrc/libzhuyin-0.9.93/src/Makefile.am	2013-12-20 13:56:37.000000000 +0900
+++ src/libzhuyin-0.9.93/src/Makefile.am	2014-01-05 22:45:01.878011400 +0900
@@ -46,7 +46,7 @@ libzhuyin_la_SOURCES	= zhuyin.cpp
 
 libzhuyin_la_LIBADD	= storage/libstorage.la lookup/liblookup.la @GLIB2_LIBS@
 
-libzhuyin_la_LDFLAGS	= -Wl,--version-script=$(srcdir)/libzhuyin.ver \
+libzhuyin_la_LDFLAGS	= -no-undefined -Wl,--version-script=$(srcdir)/libzhuyin.ver \
 			  -version-info @LT_VERSION_INFO@
 
 libzhuyin_internal_la_SOURCES	= zhuyin_internal.cpp
