--- origsrc/libloc-0.9.17/Makefile.am	2023-08-01 02:00:55.000000000 +0900
+++ src/libloc-0.9.17/Makefile.am	2023-11-24 16:24:13.297575900 +0900
@@ -135,6 +135,7 @@ src_libloc_la_CFLAGS = \
 
 src_libloc_la_LDFLAGS = \
 	$(AM_LDFLAGS) \
+	-no-undefined \
 	-version-info $(LIBLOC_CURRENT):$(LIBLOC_REVISION):$(LIBLOC_AGE)
 
 if HAVE_LD_VERSION_SCRIPT
@@ -217,6 +218,7 @@ src_python__location_la_CFLAGS = \
 
 src_python__location_la_LDFLAGS = \
 	$(AM_LDFLAGS) \
+	-no-undefined \
 	-shared \
 	-module \
 	-avoid-version
@@ -253,7 +255,7 @@ build-perl: src/libloc.la
 
 	cd $(builddir)/src/perl && $(PERL) Makefile.PL NO_PACKLIST=1 NO_PERLLOCAL=1 \
 		INSTALLDIRS=vendor \
-		INC="-I$(abs_srcdir)/src" LIBS="-L$(abs_builddir)/src/.libs -lloc"
+		INC="-I$(abs_srcdir)/src" LIBS="-L$(abs_builddir)/src/.libs -lloc -no-undefined"
 	cd $(builddir)/src/perl && $(MAKE)
 	touch build-perl
 
