--- origsrc/libmirage-3.1.0/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/CMakeLists.txt	2017-07-14 08:39:57.503267300 +0900
@@ -177,7 +177,7 @@ set_target_properties (mirage PROPERTIES
     SOVERSION ${MIRAGE_SOVERSION_MAJOR}
 )
 
-install (TARGETS mirage DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install (TARGETS mirage RUNTIME DESTINATION bin ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install (FILES ${libmirage_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libmirage-${MIRAGE_VERSION_MAJOR}.${MIRAGE_VERSION_MINOR}/mirage)
 
 # Translations; we use the version provided by our FindIntlTool.cmake
@@ -194,7 +194,7 @@ if (INTROSPECTION_ENABLED AND INTROSPECT
         INCLUDES GObject-2.0 Gio-2.0 GLib-2.0 GModule-2.0
         LIBRARY mirage
         QUIET
-        SCANNER_ARGS --add-include-path=${PROJECT_SOURCE_DIR} --warn-all
+        SCANNER_ARGS --add-include-path=${PROJECT_SOURCE_DIR} --warn-all --library-path=${PROJECT_BINARY_DIR}
         COMPILER_ARGS --includedir=${PROJECT_SOURCE_DIR}
         SYMBOL_PREFIXES mirage
         IDENTIFIER_PREFIXES Mirage
--- origsrc/libmirage-3.1.0/filters/filter-cso/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/filters/filter-cso/CMakeLists.txt	2017-07-14 08:39:57.503267300 +0900
@@ -19,7 +19,7 @@ if (ZLIB_FOUND)
         filter-stream.c
         plugin.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/filters/filter-daa/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/filters/filter-daa/CMakeLists.txt	2017-07-14 08:39:57.519102900 +0900
@@ -22,7 +22,7 @@ if (ZLIB_FOUND)
         lzma-sdk/Bra86.c
         lzma-sdk/LzmaDec.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/filters/filter-dmg/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/filters/filter-dmg/CMakeLists.txt	2017-07-14 08:39:57.519102900 +0900
@@ -23,7 +23,7 @@ if (ZLIB_FOUND AND BZIP2_FOUND)
         plugin.c
         resource-fork.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES} ${BZIP2_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES} ${BZIP2_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/filters/filter-ecm/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/filters/filter-ecm/CMakeLists.txt	2017-07-14 08:39:57.519102900 +0900
@@ -10,7 +10,7 @@ if (TRUE)
         filter-stream.c
         plugin.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/filters/filter-gzip/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/filters/filter-gzip/CMakeLists.txt	2017-07-14 08:39:57.519102900 +0900
@@ -19,7 +19,7 @@ if (ZLIB_FOUND)
         filter-stream.c
         plugin.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/filters/filter-isz/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/filters/filter-isz/CMakeLists.txt	2017-07-14 08:39:57.534522400 +0900
@@ -21,7 +21,7 @@ if (ZLIB_FOUND AND BZIP2_FOUND)
         filter-stream.c
         plugin.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES} ${BZIP2_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES} ${BZIP2_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/filters/filter-macbinary/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/filters/filter-macbinary/CMakeLists.txt	2017-07-14 08:39:57.534522400 +0900
@@ -12,7 +12,7 @@ if (TRUE)
         ../filter-dmg/resource-fork.c
         ../filter-dmg/adc.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/filters/filter-sndfile/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/filters/filter-sndfile/CMakeLists.txt	2017-07-14 08:39:57.534522400 +0900
@@ -22,7 +22,7 @@ if (SNDFILE_FOUND AND SAMPLERATE_FOUND)
         filter-stream.c
         plugin.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES} ${SNDFILE_LIBRARIES} ${SAMPLERATE_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES} ${SNDFILE_LIBRARIES} ${SAMPLERATE_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/filters/filter-xz/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/filters/filter-xz/CMakeLists.txt	2017-07-14 08:39:57.534522400 +0900
@@ -19,7 +19,7 @@ if (LIBLZMA_FOUND)
         filter-stream.c
         plugin.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES} ${LIBLZMA_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES} ${LIBLZMA_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/images/image-b6t/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/images/image-b6t/CMakeLists.txt	2017-07-14 08:39:57.534522400 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/images/image-c2d/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/images/image-c2d/CMakeLists.txt	2017-07-14 08:39:57.534522400 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/images/image-ccd/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/images/image-ccd/CMakeLists.txt	2017-07-14 08:39:57.550166500 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/images/image-cdi/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/images/image-cdi/CMakeLists.txt	2017-07-14 08:39:57.550166500 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/images/image-cif/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/images/image-cif/CMakeLists.txt	2017-07-14 08:39:57.550166500 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/images/image-cue/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/images/image-cue/CMakeLists.txt	2017-07-14 08:39:57.550166500 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/images/image-harddisk/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/images/image-harddisk/CMakeLists.txt	2017-07-14 08:39:57.550166500 +0900
@@ -10,7 +10,7 @@ if (TRUE)
         partition-tables.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/images/image-iso/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/images/image-iso/CMakeLists.txt	2017-07-14 08:39:57.565765200 +0900
@@ -10,7 +10,7 @@ if (TRUE)
         plugin.c
         writer.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/images/image-mds/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/images/image-mds/CMakeLists.txt	2017-07-14 08:39:57.565765200 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/images/image-mdx/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/images/image-mdx/CMakeLists.txt	2017-07-14 08:39:57.565765200 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/images/image-nrg/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/images/image-nrg/CMakeLists.txt	2017-07-14 08:39:57.565765200 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/images/image-readcd/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/images/image-readcd/CMakeLists.txt	2017-07-14 08:39:57.565765200 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/images/image-toc/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/images/image-toc/CMakeLists.txt	2017-07-14 08:39:57.581413100 +0900
@@ -10,7 +10,7 @@ if (TRUE)
         plugin.c
         writer.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/images/image-xcdroast/CMakeLists.txt	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/images/image-xcdroast/CMakeLists.txt	2017-07-14 08:39:57.581413100 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # On OS X, we need to explicitly enable dynamic resolving of undefined symbols
     if(APPLE)
--- origsrc/libmirage-3.1.0/mirage/mirage.c	2017-06-11 06:29:17.000000000 +0900
+++ src/libmirage-3.1.0/mirage/mirage.c	2017-07-14 08:39:57.581413100 +0900
@@ -158,7 +158,11 @@ gboolean mirage_initialize (GError **err
 
     /* Check every file in the plugin dir */
     while ((plugin_file = g_dir_read_name(plugins_dir))) {
+#if defined(__CYGWIN__)
+        if (g_str_has_suffix(plugin_file, ".dll")) {
+#else
         if (g_str_has_suffix(plugin_file, ".so")) {
+#endif
             MiragePlugin *plugin;
             gchar *fullpath;
 
