--- origsrc/libmirage-3.0.0/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/CMakeLists.txt	2014-07-24 23:58:41.559831800 +0900
@@ -171,7 +171,7 @@ set_target_properties (mirage PROPERTIES
     SOVERSION ${MIRAGE_SOVERSION_MAJOR}
 )
 
-install (TARGETS mirage DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install (TARGETS mirage RUNTIME DESTINATION bin ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install (FILES ${libmirage_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libmirage-${MIRAGE_VERSION_MAJOR}.${MIRAGE_VERSION_MINOR}/mirage)
 
 # *** Introspection ***
@@ -182,7 +182,7 @@ if (INTROSPECTION_ENABLED AND INTROSPECT
         INCLUDES GObject-2.0 Gio-2.0 GLib-2.0 GModule-2.0
         LIBRARY mirage
         QUIET
-        SCANNER_ARGS --add-include-path=${PROJECT_SOURCE_DIR} --warn-all
+        SCANNER_ARGS --add-include-path=${PROJECT_SOURCE_DIR} --warn-all --library-path=${PROJECT_BINARY_DIR}
         COMPILER_ARGS --includedir=${PROJECT_SOURCE_DIR}
         SYMBOL_PREFIXES mirage
         IDENTIFIER_PREFIXES Mirage
--- origsrc/libmirage-3.0.0/filters/filter-cso/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/filters/filter-cso/CMakeLists.txt	2014-07-24 21:05:03.221317500 +0900
@@ -19,7 +19,7 @@ if (ZLIB_FOUND)
         filter-stream.c
         plugin.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${filter_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/filters/filter-daa/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/filters/filter-daa/CMakeLists.txt	2014-07-24 21:05:53.174688400 +0900
@@ -25,7 +25,7 @@ if (ZLIB_FOUND)
         lzma-sdk/Bra86.c
         lzma-sdk/LzmaDec.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${filter_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/filters/filter-dmg/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/filters/filter-dmg/CMakeLists.txt	2014-07-24 21:05:30.761131200 +0900
@@ -23,7 +23,7 @@ if (ZLIB_FOUND AND BZIP2_FOUND)
         plugin.c
         resource-fork.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES} ${BZIP2_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES} ${BZIP2_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${filter_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/filters/filter-ecm/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/filters/filter-ecm/CMakeLists.txt	2014-07-24 21:06:16.165301500 +0900
@@ -10,7 +10,7 @@ if (TRUE)
         filter-stream.c
         plugin.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${filter_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/filters/filter-gzip/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/filters/filter-gzip/CMakeLists.txt	2014-07-24 21:06:27.198504600 +0900
@@ -19,7 +19,7 @@ if (ZLIB_FOUND)
         filter-stream.c
         plugin.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${filter_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/filters/filter-isz/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/filters/filter-isz/CMakeLists.txt	2014-07-24 21:06:48.925067100 +0900
@@ -21,7 +21,7 @@ if (ZLIB_FOUND AND BZIP2_FOUND)
         filter-stream.c
         plugin.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES} ${BZIP2_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES} ${ZLIB_LIBRARIES} ${BZIP2_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${filter_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/filters/filter-macbinary/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/filters/filter-macbinary/CMakeLists.txt	2014-07-24 21:06:57.677020200 +0900
@@ -12,7 +12,7 @@ if (TRUE)
         ../filter-dmg/resource-fork.c
         ../filter-dmg/adc.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${filter_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/filters/filter-sndfile/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/filters/filter-sndfile/CMakeLists.txt	2014-07-24 21:07:07.053973400 +0900
@@ -22,7 +22,7 @@ if (SNDFILE_FOUND AND SAMPLERATE_FOUND)
         filter-stream.c
         plugin.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES} ${SNDFILE_LIBRARIES} ${SAMPLERATE_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES} ${SNDFILE_LIBRARIES} ${SAMPLERATE_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${filter_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/filters/filter-xz/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/filters/filter-xz/CMakeLists.txt	2014-07-24 21:07:15.118426500 +0900
@@ -19,7 +19,7 @@ if (LIBLZMA_FOUND)
         filter-stream.c
         plugin.c
     )
-    target_link_libraries(${filter_name} ${GLIB_LIBRARIES} ${LIBLZMA_LIBRARIES})
+    target_link_libraries(${filter_name} mirage ${GLIB_LIBRARIES} ${LIBLZMA_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${filter_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/images/image-b6t/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/images/image-b6t/CMakeLists.txt	2014-07-24 21:08:01.882098400 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${image_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/images/image-c2d/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/images/image-c2d/CMakeLists.txt	2014-07-24 21:07:46.797137400 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${image_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/images/image-ccd/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/images/image-ccd/CMakeLists.txt	2014-07-24 21:08:09.252215500 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${image_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/images/image-cdi/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/images/image-cdi/CMakeLists.txt	2014-07-24 21:08:16.894793700 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${image_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/images/image-cif/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/images/image-cif/CMakeLists.txt	2014-07-24 21:08:24.029559300 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${image_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/images/image-cue/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/images/image-cue/CMakeLists.txt	2014-07-24 21:08:31.719012400 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${image_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/images/image-harddisk/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/images/image-harddisk/CMakeLists.txt	2014-07-24 21:08:39.244403000 +0900
@@ -10,7 +10,7 @@ if (TRUE)
         partition-tables.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${image_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/images/image-iso/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/images/image-iso/CMakeLists.txt	2014-07-24 21:08:51.168231200 +0900
@@ -10,7 +10,7 @@ if (TRUE)
         plugin.c
         writer.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${image_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/images/image-mds/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/images/image-mds/CMakeLists.txt	2014-07-24 21:08:58.662371800 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${image_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/images/image-mdx/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/images/image-mdx/CMakeLists.txt	2014-07-24 21:09:06.429949900 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${image_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/images/image-nrg/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/images/image-nrg/CMakeLists.txt	2014-07-24 21:09:14.916278000 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${image_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/images/image-readcd/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/images/image-readcd/CMakeLists.txt	2014-07-24 21:09:22.191668700 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${image_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/images/image-toc/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/images/image-toc/CMakeLists.txt	2014-07-24 21:09:29.748309300 +0900
@@ -10,7 +10,7 @@ if (TRUE)
         plugin.c
         writer.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${image_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/images/image-xcdroast/CMakeLists.txt	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/images/image-xcdroast/CMakeLists.txt	2014-07-24 21:09:37.359637400 +0900
@@ -9,7 +9,7 @@ if (TRUE)
         parser.c
         plugin.c
     )
-    target_link_libraries(${image_name} ${GLIB_LIBRARIES})
+    target_link_libraries(${image_name} mirage ${GLIB_LIBRARIES})
 
     # Disable library prefix
     set_target_properties(${image_name} PROPERTIES PREFIX "")
--- origsrc/libmirage-3.0.0/mirage/mirage.c	2014-06-29 22:25:04.000000000 +0900
+++ src/libmirage-3.0.0/mirage/mirage.c	2014-07-24 23:53:58.521961000 +0900
@@ -152,7 +152,11 @@ gboolean mirage_initialize (GError **err
 
     /* Check every file in the plugin dir */
     while ((plugin_file = g_dir_read_name(plugins_dir))) {
+#if defined(__CYGWIN__)
+        if (g_str_has_suffix(plugin_file, ".dll")) {
+#else
         if (g_str_has_suffix(plugin_file, ".so")) {
+#endif
             MiragePlugin *plugin;
             gchar *fullpath;
 
