--- origsrc/hesinfo-3.1.0/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/hesinfo-3.1.0/Makefile.am	2011-09-22 00:42:47.060684500 +0900
@@ -0,0 +1,7 @@
+
+bin_PROGRAMS = hesinfo
+hesinfo_SOURCES = hesinfo.c
+hesinfo_LDADD = -lhesiod
+
+man_MANS = hesinfo.1
+
--- origsrc/hesinfo-3.1.0/configure.in	2006-03-31 01:22:39.000000000 +0900
+++ src/hesinfo-3.1.0/configure.in	2011-09-22 00:41:30.290294100 +0900
@@ -1,12 +1,12 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(hesinfo.c)
+AC_INIT(hesinfo, 3.1.0)
+AM_INIT_AUTOMAKE
 
 test -z "$lbindir" && lbindir='${bindir}'
 AC_SUBST(lbindir)
 
 AC_PROG_CC
 AC_PROG_INSTALL
-
-ATHENA_HESIOD_REQUIRED
+LT_INIT
 
 AC_OUTPUT(Makefile)
