--- origsrc/irrxml-1.2/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/irrxml-1.2/Makefile.am	2024-10-04 15:15:01.211347300 +0900
@@ -0,0 +1,17 @@
+
+AM_CPPFLAGS = -I$(top_srcdir)/src
+
+check_PROGRAMS = irrxml
+irrxml_SOURCES = example/test.cpp
+irrxml_LDADD = libirrxml.la
+
+pkginclude_HEADERS = \
+src/CXMLReaderImpl.h  src/fast_atof.h  src/heapsort.h  src/irrArray.h \
+src/irrString.h       src/irrTypes.h   src/irrXML.h
+
+lib_LTLIBRARIES = libirrxml.la
+libirrxml_la_SOURCES = src/irrXML.cpp
+libirrxml_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/irrxml-1.2/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/irrxml-1.2/configure.ac	2024-10-04 15:15:01.226969100 +0900
@@ -0,0 +1,29 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([irrxml], [1.2])
+AC_CONFIG_SRCDIR([example/test.cpp])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+AC_PROG_CPP
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+AC_C_INLINE
+
+# Checks for library functions.
+AC_CHECK_FUNCS([pow strtol])
+
+AC_CONFIG_FILES([irrxml.pc Makefile])
+AC_OUTPUT
--- origsrc/irrxml-1.2/example/test.cpp	2005-07-02 09:00:56.000000000 +0900
+++ src/irrxml-1.2/example/test.cpp	2024-10-04 15:15:01.226969100 +0900
@@ -3,6 +3,7 @@ using namespace irr;
 using namespace io;
 
 #include <string> // we use STL strings to store data in this example
+#include <string.h>
 
 int main()
 {
--- origsrc/irrxml-1.2/irrxml.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/irrxml-1.2/irrxml.pc.in	2024-10-04 15:15:01.226969100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: irrxml
+Description: Simple and fast open source xml parser for C++
+Version: @VERSION@
+Libs: -L${libdir} -lirrxml
+Cflags: -I${includedir}/irrxml
