--- origsrc/hoel-1.4.27/CMakeLists.txt	2023-01-06 00:12:03.000000000 +0900
+++ src/hoel-1.4.27/CMakeLists.txt	2023-01-21 12:33:08.423231800 +0900
@@ -103,11 +103,17 @@ set(LIB_SRC
 find_package (Threads)
 set(LIBS ${LIBS} ${CMAKE_THREAD_LIBS_INIT})
 
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(ORCANIA liborcania)
+if (ORCANIA_FOUND)
+  set(LIBS ${ORCANIA_LIBRARIES})
+endif()
+
 include(FindJansson)
 set(JANSSON_MIN_VERSION 2.4)
 find_package(Jansson ${JANSSON_MIN_VERSION} REQUIRED)
 if (JANSSON_FOUND)
-  set(LIBS ${JANSSON_LIBRARIES})
+  set(LIBS ${LIBS} ${JANSSON_LIBRARIES})
 endif ()
 
 option(WITH_SQLITE3 "Use SQLite3 library" ON)
@@ -176,7 +182,11 @@ if (WITH_SQLITE3)
   set (PKGCONF_REQ_PRIVATE "${PKGCONF_REQ_PRIVATE}, sqlite3")
 endif ()
 if (WITH_MARIADB)
-  set (PKGCONF_REQ_PRIVATE "${PKGCONF_REQ_PRIVATE}, mariadb")
+  if (CYGWIN)
+    set (PKGCONF_REQ_PRIVATE "${PKGCONF_REQ_PRIVATE}, libmariadb")
+  else()
+    set (PKGCONF_REQ_PRIVATE "${PKGCONF_REQ_PRIVATE}, mariadb")
+  endif()
 endif ()
 if (WITH_PGSQL)
   set (PKGCONF_REQ_PRIVATE "${PKGCONF_REQ_PRIVATE}, libpq")
