--- origsrc/hoel-1.4.13/CMakeLists.txt	2019-12-29 00:24:42.000000000 +0900
+++ src/hoel-1.4.13/CMakeLists.txt	2020-01-02 14:01:09.701631700 +0900
@@ -80,11 +80,17 @@ set(LIB_SRC
 find_package (Threads)
 set(LIBS ${LIBS} ${CMAKE_THREAD_LIBS_INIT})
 
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(ORCANIA liborcania)
+if (ORCANIA_FOUND)
+  set(LIBS ${ORCANIA_LIBRARIES})
+endif()
+
 include(FindJansson)
 set(JANSSON_MIN_VERSION 2.4)
 find_package(Jansson ${JANSSON_MIN_VERSION} REQUIRED)
 if (JANSSON_FOUND)
-  set(LIBS ${JANSSON_LIBRARIES})
+  set(LIBS ${LIBS} ${JANSSON_LIBRARIES})
 endif ()
 
 option(WITH_SQLITE3 "Use SQLite3 library" ON)
