diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/Pincaster/configure.ac src/Pincaster/configure.ac
--- origsrc/Pincaster/configure.ac	2015-05-27 21:11:41.000000000 +0900
+++ src/Pincaster/configure.ac	2015-05-27 21:18:37.727200200 +0900
@@ -54,9 +54,7 @@ case "$yajl" in
   ;;
   "system")
     AC_MSG_RESULT([system])
-    AC_CHECK_YAJL(,[
-      AC_MSG_ERROR([YAJL library not found.])
-    ])
+    PKG_CHECK_MODULES([YAJL], [yajl], [], AC_MSG_ERROR([YAJL library not found.]))
   ;;
   "embedded")
     AC_MSG_RESULT([embedded])
@@ -119,9 +117,7 @@ AC_CHECK_HEADERS([fcntl.h])
 
 AC_SUBST([MAINT])
 
-LEVENTDIR="levent2"
-AC_CONFIG_SUBDIRS([src/levent2])
-CPPFLAGS="-I$LEVENTDIR -I$LEVENTDIR/include $CPPFLAGS"
+PKG_CHECK_MODULES([LIBEVENT], [libevent libevent_pthreads])
 
 dnl Output
 
@@ -138,5 +134,3 @@ AC_CONFIG_FILES([Makefile
                 ])
 
 AC_OUTPUT
-
-(cd src/levent2 && $am_make)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/Pincaster/src/Makefile.am src/Pincaster/src/Makefile.am
--- origsrc/Pincaster/src/Makefile.am	2015-05-27 21:11:41.000000000 +0900
+++ src/Pincaster/src/Makefile.am	2015-05-27 21:18:11.258405400 +0900
@@ -1,8 +1,5 @@
 
-pincaster_LDADD = \
-        levent2/.libs/libevent_core.a \
-        levent2/.libs/libevent_extra.a \
-        levent2/.libs/libevent_pthreads.a \
+pincaster_LDADD = @LIBEVENT_LIBS@ \
         @YAJL_LDADD@
 
 AM_CFLAGS = \
@@ -66,4 +63,4 @@ pincaster_SOURCES = \
         replication_slave.h
 
 SUBDIRS = \
-        ext levent2 yajl
+        ext yajl
