--- origsrc/SFCGAL-1.0.5/CMakeLists.txt	2014-08-19 17:54:17.000000000 +0900
+++ src/SFCGAL-1.0.5/CMakeLists.txt	2014-09-18 23:28:06.761976200 +0900
@@ -233,7 +233,9 @@ install(DIRECTORY ./include DESTINATION
 
 
 #-- create a libtool file for SFCGAL (needed by PostGIS)
+if( NOT CYGWIN )
 create_libtool_file( SFCGAL /lib )
+endif()
 
 #-- sfcgal-config
 if ( "${CMAKE_BUILD_TYPE}" STREQUAL "Debug" )
--- origsrc/SFCGAL-1.0.5/src/CMakeLists.txt	2014-08-19 17:54:17.000000000 +0900
+++ src/SFCGAL-1.0.5/src/CMakeLists.txt	2014-09-18 23:28:06.761976200 +0900
@@ -55,6 +55,10 @@ if ( ${Use_precompiled_headers} )
   endif()
 endif()
 
+if(CYGWIN)
+  set_target_properties(SFCGAL PROPERTIES SOVERSION 0)
+endif()
+
 # install library
 install(
 	TARGETS
