--- origsrc/SFCGAL-1.0.4/CMakeLists.txt	2013-11-30 01:41:50.000000000 +0900
+++ src/SFCGAL-1.0.4/CMakeLists.txt	2014-07-20 11:52:58.171300900 +0900
@@ -203,7 +203,9 @@ install(DIRECTORY ./include DESTINATION
 
 
 #-- create a libtool file for SFCGAL (needed by PostGIS)
+if( NOT CYGWIN )
 create_libtool_file( SFCGAL /lib )
+endif()
 
 #-- sfcgal-config
 if ( "${CMAKE_BUILD_TYPE}" STREQUAL "Debug" )
--- origsrc/SFCGAL-1.0.4/src/CMakeLists.txt	2013-11-30 01:41:50.000000000 +0900
+++ src/SFCGAL-1.0.4/src/CMakeLists.txt	2014-07-20 00:59:49.610760100 +0900
@@ -46,6 +46,10 @@ if ( ${Use_precompiled_headers} )
   endif()
 endif()
 
+if(CYGWIN)
+  set_target_properties(SFCGAL PROPERTIES SOVERSION 0)
+endif()
+
 # install library
 install(
 	TARGETS
