--- origsrc/libbuspirate-0.5.0/Makefile.am	2021-05-18 16:21:35.000000000 +0900
+++ src/libbuspirate-0.5.0/Makefile.am	2021-12-30 15:16:31.784826700 +0900
@@ -1,2 +1,5 @@
 ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = src
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libbuspirate-0.5.0/configure.ac	2021-05-18 16:21:35.000000000 +0900
+++ src/libbuspirate-0.5.0/configure.ac	2021-12-30 15:16:31.786741800 +0900
@@ -36,5 +36,5 @@ if test "$with_werror" != "no"; then
 fi
 AC_SUBST(CFLAGS)
 
-AC_CONFIG_FILES([Makefile src/Makefile src/dev/Makefile])
+AC_CONFIG_FILES([libbuspirate.pc Makefile src/Makefile src/dev/Makefile])
 AC_OUTPUT
--- origsrc/libbuspirate-0.5.0/libbuspirate.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libbuspirate-0.5.0/libbuspirate.pc.in	2021-12-30 15:16:31.670654500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libbuspirate
+Description: C library to talk with the bus pirate hardware
+Version: @VERSION@
+Libs: -L${libdir} -lbuspirate
+Cflags: -I${includedir}
--- origsrc/libbuspirate-0.5.0/src/dev/Makefile.am	2021-05-18 16:30:10.000000000 +0900
+++ src/libbuspirate-0.5.0/src/dev/Makefile.am	2021-12-30 15:12:07.637434800 +0900
@@ -1,4 +1,5 @@
 AM_CFLAGS = -D__BUSPIRATE__
+AM_CPPFLAGS = -I$(top_srcdir)/src
 
 buspirate_demo_SOURCES = \
 	24c02.c \
