--- origsrc/libacm-libacm_v1.2/Makefile.am	2011-04-02 21:09:03.000000000 +0900
+++ src/libacm-libacm_v1.2/Makefile.am	2018-12-05 16:19:18.003732700 +0900
@@ -64,3 +64,5 @@ zip: acmtool.exe in_libacm.dll
 
 .PHONY: amp exe zip
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libacm-libacm_v1.2/configure.ac	2011-04-02 21:09:03.000000000 +0900
+++ src/libacm-libacm_v1.2/configure.ac	2018-12-05 16:19:08.943907600 +0900
@@ -139,7 +139,7 @@ AM_CONDITIONAL(MAKE_AUDACIOUS_PLUGIN, te
 AM_CONDITIONAL(MAKE_GST10_PLUGIN, test "$gst10_plugin" = "yes")
 AM_CONDITIONAL(USE_LIBAO, test "$have_ao" = "yes")
 
-AC_CONFIG_FILES([Makefile src/Makefile plugins/Makefile])
+AC_CONFIG_FILES([libacm.pc Makefile src/Makefile plugins/Makefile])
 AC_OUTPUT
 
 
--- origsrc/libacm-libacm_v1.2/libacm.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libacm-libacm_v1.2/libacm.pc.in	2018-12-05 16:19:08.900910000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libacm
+Description: Library for InterPlay ACM Audio format
+Version: @VERSION@
+Libs: -L${libdir} -lacm
+Cflags: -I${includedir}
--- origsrc/libacm-libacm_v1.2/src/Makefile.am	2011-04-02 21:09:03.000000000 +0900
+++ src/libacm-libacm_v1.2/src/Makefile.am	2018-12-05 16:16:33.082082900 +0900
@@ -1,10 +1,11 @@
 
 bin_PROGRAMS = acmtool
-noinst_LTLIBRARIES = libacm.la
+lib_LTLIBRARIES = libacm.la
 
-noinst_HEADERS = libacm.h
+include_HEADERS = libacm.h
 
 libacm_la_SOURCES = decode.c util.c
+libacm_la_LDFLAGS = -no-undefined
 
 acmtool_SOURCES = acmtool.c
 
