--- origsrc/libdstr-20080124/Makefile.am	2008-01-25 08:02:34.000000000 +0900
+++ src/libdstr-20080124/Makefile.am	2010-11-28 13:51:19.637733300 +0900
@@ -25,4 +25,4 @@ include_HEADERS = Dstr
 # then set age to 0.
 
 # FIXME: Update libtool version info on each release
-libdstr_la_LDFLAGS = -version-info 2:1:0
+libdstr_la_LDFLAGS = -version-info 2:1:0 -no-undefined
--- origsrc/libdstr-20080124/configure.ac	2008-01-25 08:02:34.000000000 +0900
+++ src/libdstr-20080124/configure.ac	2010-11-28 13:51:33.016971700 +0900
@@ -7,7 +7,7 @@ AM_INIT_AUTOMAKE([-Wall -Werror])
 
 AC_LANG([C++])
 AC_PROG_CXX
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 AC_CONFIG_FILES([Makefile])
 AC_OUTPUT
