--- origsrc/libdstr-1.1/Makefile.am	2026-01-17 06:14:30.000000000 +0900
+++ src/libdstr-1.1/Makefile.am	2026-03-21 18:42:15.376329600 +0900
@@ -27,4 +27,7 @@ ACLOCAL_AMFLAGS = -I m4
 # then set age to 0.
 
 # FIXME: Update libtool version info on each release
-libdstr_la_LDFLAGS = -version-info 4:0:2
+libdstr_la_LDFLAGS = -no-undefined -version-info 4:0:2
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libdstr-1.1/configure.ac	2026-01-17 06:14:30.000000000 +0900
+++ src/libdstr-1.1/configure.ac	2026-03-21 18:42:15.376329600 +0900
@@ -14,4 +14,5 @@ AC_REQUIRE_CPP
 LT_INIT
 
 AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([libdstr.pc])
 AC_OUTPUT
--- origsrc/libdstr-1.1/libdstr.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libdstr-1.1/libdstr.pc.in	2026-03-21 18:42:15.358476100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libdstr
+Description: Dave's string class
+Version: @VERSION@
+Libs: -L${libdir} -ldstr
+Cflags: -I${includedir}
