diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/ismrmrd-1.3.2/CMakeLists.txt src/ismrmrd-1.3.2/CMakeLists.txt
--- origsrc/ismrmrd-1.3.2/CMakeLists.txt	2015-10-25 12:25:41.000000000 +0900
+++ src/ismrmrd-1.3.2/CMakeLists.txt	2016-02-06 10:25:47.440116200 +0900
@@ -182,7 +182,7 @@ set_target_properties(ismrmrd PROPERTIES
 target_link_libraries(ismrmrd ${ISMRMRD_TARGET_LINK_LIBS})
 
 # install the main library
-install(TARGETS ismrmrd DESTINATION lib)
+install(TARGETS ismrmrd RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 
 # install the headers
 install(DIRECTORY include/ismrmrd  DESTINATION include)
@@ -191,7 +191,7 @@ install(DIRECTORY include/ismrmrd  DESTI
 install(FILES schema/ismrmrd.xsd DESTINATION share/ismrmrd/schema)
 
 # install the cmake modules
-install(FILES cmake/FindIsmrmrd.cmake cmake/FindFFTW3.cmake DESTINATION share/ismrmrd/cmake)
+install(FILES cmake/FindIsmrmrd.cmake cmake/FindFFTW3.cmake DESTINATION lib/cmake/ismrmrd)
 
 #  ---   Main Library  (end) ----
 
@@ -220,3 +220,11 @@ if(CPACK_GENERATOR)
   set(CPACK_PROJECT_CONFIG_FILE ${ISMRMRD_CPACK_CFG_FILE})
   include(CPack)
 endif(CPACK_GENERATOR)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "1.3.2")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/ismrmrd.pc.in ${CMAKE_BINARY_DIR}/ismrmrd.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/ismrmrd.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/ismrmrd-1.3.2/ismrmrd.pc.in src/ismrmrd-1.3.2/ismrmrd.pc.in
--- origsrc/ismrmrd-1.3.2/ismrmrd.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/ismrmrd-1.3.2/ismrmrd.pc.in	2016-02-06 10:20:37.325850400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: ismrmrd
+Description: ISMRM Raw Data Format
+Version: @VERSION@
+Libs: -L${libdir} -lismrmrd
+Cflags: -I${includedir}
