--- origsrc/libqes-0.2.8/CMakeLists.txt	2017-02-07 13:03:15.000000000 +0900
+++ src/libqes-0.2.8/CMakeLists.txt	2024-11-18 21:20:19.249070500 +0900
@@ -162,3 +162,10 @@ ADD_SUBDIRECTORY(src)
 if (NOT LIBQES_IS_SUBPROJECT)
     ADD_SUBDIRECTORY(test)
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include/libqes)
+configure_file(${CMAKE_SOURCE_DIR}/libqes.pc.in ${CMAKE_BINARY_DIR}/libqes.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libqes.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libqes-0.2.8/libqes.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libqes-0.2.8/libqes.pc.in	2024-11-18 21:12:02.133022600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libqes
+Description: Small C library, with a bioinformatic focus
+Version: @VERSION@
+Libs: -L${libdir} -lqes
+Cflags: -I${includedir}
--- origsrc/libqes-0.2.8/src/CMakeLists.txt	2017-02-07 13:03:15.000000000 +0900
+++ src/libqes-0.2.8/src/CMakeLists.txt	2024-11-18 21:20:11.749093700 +0900
@@ -2,19 +2,24 @@ file(GLOB LIBQES_SOURCES qes_*.c)
 
 # Targets
 if(LIBQES_DONT_INSTALL)
+    if (BUILD_STATIC_LIBS)
     ADD_LIBRARY(qes_static STATIC EXCLUDE_FROM_ALL ${LIBQES_SOURCES})
+    endif()
     ADD_LIBRARY(qes SHARED EXCLUDE_FROM_ALL ${LIBQES_SOURCES})
 else()
+    if (BUILD_STATIC_LIBS)
     ADD_LIBRARY(qes_static STATIC ${LIBQES_SOURCES})
+    endif()
     ADD_LIBRARY(qes SHARED ${LIBQES_SOURCES})
 endif()
 
-
+if (BUILD_STATIC_LIBS)
 SET_TARGET_PROPERTIES(qes_static PROPERTIES OUTPUT_NAME qes)
 TARGET_LINK_LIBRARIES(qes_static ${LIBQES_DEPENDS_LIBS})
 target_include_directories(qes_static PUBLIC ${LIBQES_DEPENDS_INCLUDE_DIRS} ${CMAKE_CURRENT_SOURCE_DIR})
+endif()
 
-SET_TARGET_PROPERTIES(qes PROPERTIES SONAME_VERSION 0 VERSION 0)
+SET_TARGET_PROPERTIES(qes PROPERTIES SOVERSION 0 VERSION 0)
 TARGET_LINK_LIBRARIES(qes ${LIBQES_DEPENDS_LIBS})
 target_include_directories(qes PUBLIC ${LIBQES_DEPENDS_INCLUDE_DIRS} ${CMAKE_CURRENT_SOURCE_DIR})
 
@@ -22,6 +27,9 @@ CONFIGURE_FILE(qes_config.h.in ${CMAKE_B
 FILE(GLOB LIBQES_HEADERS ${CMAKE_SOURCE_DIR}/src/*.h ${CMAKE_BINARY_DIR}/qes_config.h)
 
 IF (NOT LIBQES_DONT_INSTALL)
-	INSTALL(FILES ${LIBQES_HEADERS} DESTINATION "include")
-	INSTALL(TARGETS qes qes_static DESTINATION "lib")
+	INSTALL(FILES ${LIBQES_HEADERS} DESTINATION "include/libqes")
+	INSTALL(TARGETS qes RUNTIME DESTINATION bin ARCHIVE DESTINATION "lib")
+    if (BUILD_STATIC_LIBS)
+	INSTALL(TARGETS qes_static DESTINATION "lib")
+    endif()
 ENDIF()
