diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libqes-0.1.19/CMakeLists.txt src/libqes-0.1.19/CMakeLists.txt
--- origsrc/libqes-0.1.19/CMakeLists.txt	2015-02-25 15:47:09.000000000 +0900
+++ src/libqes-0.1.19/CMakeLists.txt	2015-06-14 17:58:16.379152800 +0900
@@ -152,3 +152,11 @@ LINK_DIRECTORIES(${CMAKE_BINARY_DIR}/lib
 # Traverse to library source and tests
 ADD_SUBDIRECTORY(src)
 ADD_SUBDIRECTORY(test)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "0.1.19")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/${PROJECT_NAME}.pc.in ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libqes-0.1.19/libqes.pc.in src/libqes-0.1.19/libqes.pc.in
--- origsrc/libqes-0.1.19/libqes.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libqes-0.1.19/libqes.pc.in	2015-06-14 17:58:32.371795400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libqes
+Description: Small C library, with a bioinformatic focus
+Version: @VERSION@
+Libs: -L${libdir} -lqes
+Cflags: -I${includedir}/@PROJECT_NAME@
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libqes-0.1.19/src/CMakeLists.txt src/libqes-0.1.19/src/CMakeLists.txt
--- origsrc/libqes-0.1.19/src/CMakeLists.txt	2015-02-25 15:47:09.000000000 +0900
+++ src/libqes-0.1.19/src/CMakeLists.txt	2015-06-14 17:57:33.938654100 +0900
@@ -1,16 +1,23 @@
 file(GLOB LIBQES_SOURCES qes_*.c)
 
 # Targets
+OPTION(ENABLE_STATIC "enable static" OFF)
+IF(ENABLE_STATIC)
 ADD_LIBRARY(qes_static STATIC ${LIBQES_SOURCES})
 SET_TARGET_PROPERTIES(qes_static PROPERTIES OUTPUT_NAME qes)
 TARGET_LINK_LIBRARIES(qes_static ${LIBQES_DEPENDS_LIBS})
+ENDIF()
 ADD_LIBRARY(qes SHARED ${LIBQES_SOURCES})
 TARGET_LINK_LIBRARIES(qes ${LIBQES_DEPENDS_LIBS})
+SET_TARGET_PROPERTIES(qes PROPERTIES SOVERSION 0)
 
 CONFIGURE_FILE(qes_config.h.in ${CMAKE_BINARY_DIR}/qes_config.h)
 FILE(GLOB LIBQES_HEADERS ${CMAKE_SOURCE_DIR}/src/*.h ${CMAKE_BINARY_DIR}/qes_config.h)
 
 IF (NOT LIBQES_DONT_INSTALL)
-	INSTALL(FILES ${LIBQES_HEADERS} DESTINATION "include")
-	INSTALL(TARGETS qes qes_static DESTINATION "lib")
+	INSTALL(FILES ${LIBQES_HEADERS} DESTINATION "include/${PROJECT_NAME}")
+	INSTALL(TARGETS qes RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+	IF(ENABLE_STATIC)
+		INSTALL(TARGETS qes_static LIBRARY DESTINATION lib)
+	ENDIF()
 ENDIF()
