--- origsrc/libdockapp-0.6.2/configure.in	2005-07-16 20:49:10.000000000 +0900
+++ src/libdockapp-0.6.2/configure.in	2010-12-08 01:28:31.246572200 +0900
@@ -3,8 +3,6 @@ dnl Process this file with autoconf to p
 AC_INIT(src/dockapp.h)
 AM_INIT_AUTOMAKE(libdockapp, 0.6.1)
 
-AC_PROG_LIBTOOL
-
 dnl
 dnl Specify whether to install font and examples
 dnl ============================================
@@ -27,8 +25,7 @@ AM_CONDITIONAL(COND_EXAMPLES, [test "$wi
 
 
 dnl Checks for programs.
-AC_PROG_RANLIB
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_PROG_CC
 AC_PROG_CXX
 AC_PROG_INSTALL
--- origsrc/libdockapp-0.6.2/examples/Makefile.am	2003-08-26 22:08:37.000000000 +0900
+++ src/libdockapp-0.6.2/examples/Makefile.am	2010-12-08 10:09:04.948356500 +0900
@@ -5,18 +5,20 @@
 #	but I don't know how and they don't tell.
 #
 
-SHAREDIR = $(DESTDIR)/usr/local/share
+SHAREDIR = $(DESTDIR)/usr/share
 
 BASICDIR = basic
-BASICSRC = basic/Imakefile basic/README basic/ball_red.xpm basic/basic.c
+BASICSRC_ = basic/Imakefile basic/README basic/ball_red.xpm basic/basic.c
+BASICSRC = $(patsubst %,$(top_srcdir)/examples/%,$(BASICSRC_))
 
 RECTDIR  = rectangles
-RECTSRC  = rectangles/Imakefile rectangles/rectangles.c
+RECTSRC_ = rectangles/Imakefile rectangles/rectangles.c
+RECTSRC  = $(patsubst %,$(top_srcdir)/examples/%,$(RECTSRC_))
 
 #SHAPEDIR = shapes
 #SHAPESRC =
 
-EXAMPLESDIR = /examples/dockapp/
+EXAMPLESDIR = /dockapp/examples/
 
 install-data-local:
 	$(INSTALL) -d $(SHAREDIR)$(EXAMPLESDIR)$(BASICDIR)
--- origsrc/libdockapp-0.6.2/fonts/Makefile.am	2003-08-26 22:08:37.000000000 +0900
+++ src/libdockapp-0.6.2/fonts/Makefile.am	2010-12-08 01:29:26.796449000 +0900
@@ -5,10 +5,11 @@
 #	but I don't know how and they don't tell.
 #
 
-XFONTDIR = $(DESTDIR)/usr/X11R6/lib/X11/fonts
+XFONTDIR = $(DESTDIR)/usr/share/fonts
 
 MYFONTDIR = /dockapp
-MYDATA = luxel-ascii-06x09.pcf.gz seg7-ascii-05x07.pcf.gz fonts.dir fonts.alias
+MYDATA_ = luxel-ascii-06x09.pcf.gz seg7-ascii-05x07.pcf.gz fonts.dir fonts.alias
+MYDATA = $(patsubst %,$(top_srcdir)/fonts/%,$(MYDATA_))
 
 install-data-local:
 	$(INSTALL) -d $(XFONTDIR)$(MYFONTDIR)
--- origsrc/libdockapp-0.6.2/src/Makefile.am	2005-04-09 00:46:33.000000000 +0900
+++ src/libdockapp-0.6.2/src/Makefile.am	2010-12-08 01:29:38.773671400 +0900
@@ -4,7 +4,7 @@ AUTOMAKE_OPTIONS = no-dependencies
 
 lib_LTLIBRARIES = libdockapp.la
 
-libdockapp_la_LDFLAGS = -version-info 2:0:0 @X_LIBS@
+libdockapp_la_LDFLAGS = -version-info 2:0:0 @X_LIBS@ -no-undefined
 
 include_HEADERS = dockapp.h
 
