--- origsrc/libdockapp-0.7.3/src/Makefile.am	2020-05-03 21:16:29.000000000 +0900
+++ src/libdockapp-0.7.3/src/Makefile.am	2024-02-26 16:06:04.328940400 +0900
@@ -3,7 +3,7 @@ AUTOMAKE_OPTIONS = no-dependencies
 
 lib_LTLIBRARIES = libdockapp.la
 
-libdockapp_la_LDFLAGS = -version-info 3:1:0
+libdockapp_la_LDFLAGS = -no-undefined -version-info 3:1:0
 
 otherincludedir = $(includedir)/libdockapp
 otherinclude_HEADERS = dockapp.h \
