--- origsrc/audiowaveform-1.4.0/CMakeLists.txt	2019-11-03 21:53:19.000000000 +0900
+++ src/audiowaveform-1.4.0/CMakeLists.txt	2019-11-19 18:48:43.270866400 +0900
@@ -182,7 +182,9 @@ if(CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CX
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "4.7")
+    if(NOT CYGWIN)
     set(CMAKE_CXX_FLAGS "-std=c++11")
+    endif()
 else()
     # Set GTEST_LANG_CXX11=0 to disable C++11 features when compiling googlemock.
     # Without this compilation fails with g++ 4.6.3 on gmock-matchers.h.
@@ -286,7 +288,10 @@ if(ENABLE_TESTS)
 
     # Use EXCLUDE_FROM_ALL to prevent installing googletest headers as part of
     # 'make install'.
+    option(USE_SYSTEM_GMOCK "use system gmock" ON)
+    if (NOT USE_SYSTEM_GMOCK)
     add_subdirectory(googlemock EXCLUDE_FROM_ALL)
+    endif()
 
     set(TESTS
         test/FileFormatTest.cpp
@@ -310,7 +315,7 @@ if(ENABLE_TESTS)
     )
 
     include_directories(${gtest_SOURCE_DIR}/include ${gmock_SOURCE_DIR}/include)
-    set(TEST_LIBS gmock_main)
+    set(TEST_LIBS gmock gmock_main gtest)
     add_executable(audiowaveform_tests ${MODULES} ${TESTS})
     target_link_libraries(audiowaveform_tests ${LIBS} ${TEST_LIBS})
     add_test(audiowaveform_tests audiowaveform_tests)
