--- origsrc/audiowaveform-1.1.0/CMakeLists.txt	2017-02-02 19:41:03.000000000 +0900
+++ src/audiowaveform-1.1.0/CMakeLists.txt	2017-12-07 22:55:07.067000000 +0900
@@ -172,7 +172,11 @@ if(CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CX
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "4.7")
-    set(CMAKE_CXX_FLAGS "-std=c++11")
+    if(CYGWIN)
+        set(CMAKE_CXX_FLAGS "-std=gnu++11")
+    else()
+        set(CMAKE_CXX_FLAGS "-std=c++11")
+    endif()
 else()
     # Set GTEST_LANG_CXX11=0 to disable C++11 features when compiling googlemock.
     # Without this compilation fails with g++ 4.6.3 on gmock-matchers.h.
@@ -262,7 +266,9 @@ endif()
 if(ENABLE_TESTS)
     enable_testing()
 
-    add_subdirectory(googlemock)
+    if(NOT CYGWIN)
+        add_subdirectory(googlemock)
+    endif()
 
     set(TESTS
         test/AudioFileReaderTest.cpp
@@ -284,7 +290,7 @@ if(ENABLE_TESTS)
     )
 
     include_directories(${gtest_SOURCE_DIR}/include ${gmock_SOURCE_DIR}/include)
-    set(TEST_LIBS gmock_main)
+    set(TEST_LIBS gmock gmock_main gtest)
     add_executable(audiowaveform_tests ${MODULES} ${TESTS})
     target_link_libraries(audiowaveform_tests ${LIBS} ${TEST_LIBS})
     add_test(audiowaveform_tests audiowaveform_tests)
