--- origsrc/apertium-ambiguous/src/CLExec.cpp	2020-07-23 21:35:49.000000000 +0900
+++ src/apertium-ambiguous/src/CLExec.cpp	2020-07-23 21:42:26.926032000 +0900
@@ -14,8 +14,8 @@
 #include <algorithm>
 #include <cfloat>
 #include <sstream>
-#include <apertium-3.5/apertium/string_utils.h>
-#include <apertium-3.5/apertium/utf_converter.h>
+#include <apertium-3.6/apertium/string_utils.h>
+#include <apertium-3.6/apertium/utf_converter.h>
 
 #include "CLExec.h"
 #include "TranElemLiterals.h"
