--- origsrc/med-4.1.0/Makefile.am	2020-01-31 01:04:21.000000000 +0900
+++ src/med-4.1.0/Makefile.am	2020-10-24 12:02:58.227155900 +0900
@@ -7,4 +7,4 @@ EXTRA_DIST = bootstrap COPYING COPYING.L
 MAINTAINERCLEANFILES = Makefile.in aclocal.m4 configure config-h.in stamp-h.in config.cache config.h.in config.log config.status
 
 # Process the following directories
-SUBDIRS = config include src tests tools doc python
+SUBDIRS = config include src tests tools python
--- origsrc/med-4.1.0/python/Makefile.am	2018-10-23 07:56:37.000000000 +0900
+++ src/med-4.1.0/python/Makefile.am	2020-10-24 18:12:51.069750100 +0900
@@ -99,9 +99,9 @@ pkgpyexec_PYTHON            +=  __init__
 nodist_pkgpyexec_PYTHON   +=  medlibrary.py
 pkgpyexec_LTLIBRARIES     += _medlibrary.la
 
-_medlibrary_la_LDFLAGS         = -module -no-undefined
+_medlibrary_la_LDFLAGS         = -module -avoid-version -no-undefined
 #_medlibrary_la_LDFLAGS = -avoid-version -module
-_medlibrary_la_LIBADD          = $(top_builddir)/src/libmed.la
+_medlibrary_la_LIBADD          = $(top_builddir)/src/libmed.la $(PYTHON_LDFLAGS)
 _medlibrary_la_CPPFLAGS        = $(SWIG_PYTHON_CPPFLAGS)
 _medlibrary_la_CPPFLAGS       += $(PYTHON_CPPFLAGS)
 _medlibrary_la_CPPFLAGS       += $(AM_CPPFLAGS)
@@ -121,8 +121,8 @@ medlibrary_@LONG_OR_INT@_wrap.cc : $(MED
 nodist_pkgpyexec_PYTHON   +=  medfile.py
 pkgpyexec_LTLIBRARIES     += _medfile.la
 
-_medfile_la_LDFLAGS         = -module -no-undefined
-_medfile_la_LIBADD          = $(top_builddir)/src/libmed.la
+_medfile_la_LDFLAGS         = -module -avoid-version -no-undefined
+_medfile_la_LIBADD          = $(top_builddir)/src/libmed.la $(PYTHON_LDFLAGS)
 _medfile_la_CPPFLAGS        = $(SWIG_PYTHON_CPPFLAGS)
 _medfile_la_CPPFLAGS       += $(PYTHON_CPPFLAGS)
 _medfile_la_CPPFLAGS       += $(AM_CPPFLAGS)
@@ -142,9 +142,9 @@ medfile_@LONG_OR_INT@_wrap.cc : $(MED_SW
 nodist_pkgpyexec_PYTHON   +=  medenum.py
 pkgpyexec_LTLIBRARIES     += _medenum.la
 
-_medenum_la_LDFLAGS         = -module -no-undefined
+_medenum_la_LDFLAGS         = -module -avoid-version -no-undefined
 
-_medenum_la_LIBADD          = $(top_builddir)/src/libmed.la
+_medenum_la_LIBADD          = $(top_builddir)/src/libmed.la $(PYTHON_LDFLAGS)
 _medenum_la_CPPFLAGS        = $(SWIG_PYTHON_CPPFLAGS)
 _medenum_la_CPPFLAGS       += $(PYTHON_CPPFLAGS)
 _medenum_la_CPPFLAGS       += $(AM_CPPFLAGS)
@@ -165,9 +165,9 @@ medenum_@LONG_OR_INT@_wrap.cc : $(MED_SW
 nodist_pkgpyexec_PYTHON   +=  medmesh.py
 pkgpyexec_LTLIBRARIES     += _medmesh.la
 
-_medmesh_la_LDFLAGS         = -module -no-undefined
+_medmesh_la_LDFLAGS         = -module -avoid-version -no-undefined
 
-_medmesh_la_LIBADD          = $(top_builddir)/src/libmed.la
+_medmesh_la_LIBADD          = $(top_builddir)/src/libmed.la $(PYTHON_LDFLAGS)
 _medmesh_la_CPPFLAGS        = $(SWIG_PYTHON_CPPFLAGS)
 _medmesh_la_CPPFLAGS       += $(PYTHON_CPPFLAGS)
 _medmesh_la_CPPFLAGS       += $(AM_CPPFLAGS)
@@ -188,9 +188,9 @@ medmesh_@LONG_OR_INT@_wrap.cc : $(MED_SW
 nodist_pkgpyexec_PYTHON   +=  medfield.py
 pkgpyexec_LTLIBRARIES     += _medfield.la
 
-_medfield_la_LDFLAGS         = -module -no-undefined
+_medfield_la_LDFLAGS         = -module -avoid-version -no-undefined
 
-_medfield_la_LIBADD          = $(top_builddir)/src/libmed.la
+_medfield_la_LIBADD          = $(top_builddir)/src/libmed.la $(PYTHON_LDFLAGS)
 _medfield_la_CPPFLAGS        = $(SWIG_PYTHON_CPPFLAGS)
 _medfield_la_CPPFLAGS       += $(PYTHON_CPPFLAGS)
 _medfield_la_CPPFLAGS       += $(AM_CPPFLAGS)
@@ -210,9 +210,9 @@ medfield_@LONG_OR_INT@_wrap.cc : $(MED_S
 nodist_pkgpyexec_PYTHON   +=  medfamily.py
 pkgpyexec_LTLIBRARIES     += _medfamily.la
 
-_medfamily_la_LDFLAGS         = -module -no-undefined
+_medfamily_la_LDFLAGS         = -module -avoid-version -no-undefined
 
-_medfamily_la_LIBADD          = $(top_builddir)/src/libmed.la
+_medfamily_la_LIBADD          = $(top_builddir)/src/libmed.la $(PYTHON_LDFLAGS)
 _medfamily_la_CPPFLAGS        = $(SWIG_PYTHON_CPPFLAGS)
 _medfamily_la_CPPFLAGS       += $(PYTHON_CPPFLAGS)
 _medfamily_la_CPPFLAGS       += $(AM_CPPFLAGS)
@@ -232,9 +232,9 @@ medfamily_@LONG_OR_INT@_wrap.cc : $(MED_
 nodist_pkgpyexec_PYTHON   +=  medequivalence.py
 pkgpyexec_LTLIBRARIES     += _medequivalence.la
 
-_medequivalence_la_LDFLAGS         = -module -no-undefined
+_medequivalence_la_LDFLAGS         = -module -avoid-version -no-undefined
 
-_medequivalence_la_LIBADD          = $(top_builddir)/src/libmed.la
+_medequivalence_la_LIBADD          = $(top_builddir)/src/libmed.la $(PYTHON_LDFLAGS)
 _medequivalence_la_CPPFLAGS        = $(SWIG_PYTHON_CPPFLAGS)
 _medequivalence_la_CPPFLAGS       += $(PYTHON_CPPFLAGS)
 _medequivalence_la_CPPFLAGS       += $(AM_CPPFLAGS)
@@ -255,9 +255,9 @@ medequivalence_@LONG_OR_INT@_wrap.cc : $
 nodist_pkgpyexec_PYTHON   +=  medenumtest.py
 pkgpyexec_LTLIBRARIES     += _medenumtest.la
 
-_medenumtest_la_LDFLAGS         = -module -no-undefined
+_medenumtest_la_LDFLAGS         = -module -avoid-version -no-undefined
 
-_medenumtest_la_LIBADD          = $(top_builddir)/src/libmed.la
+_medenumtest_la_LIBADD          = $(top_builddir)/src/libmed.la $(PYTHON_LDFLAGS)
 _medenumtest_la_CPPFLAGS        = $(SWIG_PYTHON_CPPFLAGS)
 _medenumtest_la_CPPFLAGS       += $(PYTHON_CPPFLAGS)
 _medenumtest_la_CPPFLAGS       += $(AM_CPPFLAGS)
@@ -277,9 +277,9 @@ medenumtest_@LONG_OR_INT@_wrap.cc : $(ME
 nodist_pkgpyexec_PYTHON   +=  medprofile.py
 pkgpyexec_LTLIBRARIES     += _medprofile.la
 
-_medprofile_la_LDFLAGS         = -module -no-undefined
+_medprofile_la_LDFLAGS         = -module -avoid-version -no-undefined
 
-_medprofile_la_LIBADD          = $(top_builddir)/src/libmed.la
+_medprofile_la_LIBADD          = $(top_builddir)/src/libmed.la $(PYTHON_LDFLAGS)
 _medprofile_la_CPPFLAGS        = $(SWIG_PYTHON_CPPFLAGS)
 _medprofile_la_CPPFLAGS       += $(PYTHON_CPPFLAGS)
 _medprofile_la_CPPFLAGS       += $(AM_CPPFLAGS)
@@ -299,9 +299,9 @@ medprofile_@LONG_OR_INT@_wrap.cc : $(MED
 nodist_pkgpyexec_PYTHON   +=  medlocalization.py
 pkgpyexec_LTLIBRARIES     += _medlocalization.la
 
-_medlocalization_la_LDFLAGS         = -module -no-undefined
+_medlocalization_la_LDFLAGS         = -module -avoid-version -no-undefined
 
-_medlocalization_la_LIBADD          = $(top_builddir)/src/libmed.la
+_medlocalization_la_LIBADD          = $(top_builddir)/src/libmed.la $(PYTHON_LDFLAGS)
 _medlocalization_la_CPPFLAGS        = $(SWIG_PYTHON_CPPFLAGS)
 _medlocalization_la_CPPFLAGS       += $(PYTHON_CPPFLAGS)
 _medlocalization_la_CPPFLAGS       += $(AM_CPPFLAGS)
@@ -321,9 +321,9 @@ medlocalization_@LONG_OR_INT@_wrap.cc :
 nodist_pkgpyexec_PYTHON   +=  medlink.py
 pkgpyexec_LTLIBRARIES     += _medlink.la
 
-_medlink_la_LDFLAGS         = -module -no-undefined
+_medlink_la_LDFLAGS         = -module -avoid-version -no-undefined
 
-_medlink_la_LIBADD          = $(top_builddir)/src/libmed.la
+_medlink_la_LIBADD          = $(top_builddir)/src/libmed.la $(PYTHON_LDFLAGS)
 _medlink_la_CPPFLAGS        = $(SWIG_PYTHON_CPPFLAGS)
 _medlink_la_CPPFLAGS       += $(PYTHON_CPPFLAGS)
 _medlink_la_CPPFLAGS       += $(AM_CPPFLAGS)
@@ -343,9 +343,9 @@ medlink_@LONG_OR_INT@_wrap.cc : $(MED_SW
 nodist_pkgpyexec_PYTHON   +=  medparameter.py
 pkgpyexec_LTLIBRARIES     += _medparameter.la
 
-_medparameter_la_LDFLAGS         = -module -no-undefined
+_medparameter_la_LDFLAGS         = -module -avoid-version -no-undefined
 
-_medparameter_la_LIBADD          = $(top_builddir)/src/libmed.la
+_medparameter_la_LIBADD          = $(top_builddir)/src/libmed.la $(PYTHON_LDFLAGS)
 _medparameter_la_CPPFLAGS        = $(SWIG_PYTHON_CPPFLAGS)
 _medparameter_la_CPPFLAGS       += $(PYTHON_CPPFLAGS)
 _medparameter_la_CPPFLAGS       += $(AM_CPPFLAGS)
@@ -365,9 +365,9 @@ medparameter_@LONG_OR_INT@_wrap.cc : $(M
 nodist_pkgpyexec_PYTHON   +=  medfilter.py
 pkgpyexec_LTLIBRARIES     += _medfilter.la
 
-_medfilter_la_LDFLAGS         = -module -no-undefined
+_medfilter_la_LDFLAGS         = -module -avoid-version -no-undefined
 
-_medfilter_la_LIBADD          = $(top_builddir)/src/libmed.la
+_medfilter_la_LIBADD          = $(top_builddir)/src/libmed.la $(PYTHON_LDFLAGS)
 _medfilter_la_CPPFLAGS        = $(SWIG_PYTHON_CPPFLAGS)
 _medfilter_la_CPPFLAGS       += $(PYTHON_CPPFLAGS)
 _medfilter_la_CPPFLAGS       += $(AM_CPPFLAGS)
@@ -387,9 +387,9 @@ medfilter_@LONG_OR_INT@_wrap.cc : $(MED_
 nodist_pkgpyexec_PYTHON   +=  medsubdomain.py
 pkgpyexec_LTLIBRARIES     += _medsubdomain.la
 
-_medsubdomain_la_LDFLAGS         = -module -no-undefined
+_medsubdomain_la_LDFLAGS         = -module -avoid-version -no-undefined
 
-_medsubdomain_la_LIBADD          = $(top_builddir)/src/libmed.la
+_medsubdomain_la_LIBADD          = $(top_builddir)/src/libmed.la $(PYTHON_LDFLAGS)
 _medsubdomain_la_CPPFLAGS        = $(SWIG_PYTHON_CPPFLAGS)
 _medsubdomain_la_CPPFLAGS       += $(PYTHON_CPPFLAGS)
 _medsubdomain_la_CPPFLAGS       += $(AM_CPPFLAGS)
@@ -409,9 +409,9 @@ medsubdomain_@LONG_OR_INT@_wrap.cc : $(M
 nodist_pkgpyexec_PYTHON   +=  medinterp.py
 pkgpyexec_LTLIBRARIES     += _medinterp.la
 
-_medinterp_la_LDFLAGS         = -module -no-undefined
+_medinterp_la_LDFLAGS         = -module -avoid-version -no-undefined
 
-_medinterp_la_LIBADD          = $(top_builddir)/src/libmed.la
+_medinterp_la_LIBADD          = $(top_builddir)/src/libmed.la $(PYTHON_LDFLAGS)
 _medinterp_la_CPPFLAGS        = $(SWIG_PYTHON_CPPFLAGS)
 _medinterp_la_CPPFLAGS       += $(PYTHON_CPPFLAGS)
 _medinterp_la_CPPFLAGS       += $(AM_CPPFLAGS)
@@ -431,9 +431,9 @@ medinterp_@LONG_OR_INT@_wrap.cc : $(MED_
 nodist_pkgpyexec_PYTHON   +=  medstructelement.py
 pkgpyexec_LTLIBRARIES     += _medstructelement.la
 
-_medstructelement_la_LDFLAGS         = -module -no-undefined
+_medstructelement_la_LDFLAGS         = -module -avoid-version -no-undefined
 
-_medstructelement_la_LIBADD          = $(top_builddir)/src/libmed.la
+_medstructelement_la_LIBADD          = $(top_builddir)/src/libmed.la $(PYTHON_LDFLAGS)
 _medstructelement_la_CPPFLAGS        = $(SWIG_PYTHON_CPPFLAGS)
 _medstructelement_la_CPPFLAGS       += $(PYTHON_CPPFLAGS)
 _medstructelement_la_CPPFLAGS       += $(AM_CPPFLAGS)
--- origsrc/med-4.1.0/tools/medimport/Makefile.am	2019-09-19 19:41:14.000000000 +0900
+++ src/med-4.1.0/tools/medimport/Makefile.am	2020-10-24 12:02:58.494157400 +0900
@@ -13,7 +13,7 @@ AM_CPPFLAGS+= -I$(srcdir)/include -DMED_
 lib_LTLIBRARIES   = libmedimport.la 
 libmedimport_la_SOURCES = libmedimport.c MAJ_version.c
 nodist_EXTRA_libmedimport_la_SOURCES = dummy1.cxx
-libmedimport_la_LDFLAGS = --no-undefined -version-info @MEDIMPORT_LIBTOOL_MAJ@:@MEDIMPORT_LIBTOOL_MIN@:@MEDIMPORT_LIBTOOL_REL@
+libmedimport_la_LDFLAGS = -no-undefined -version-info @MEDIMPORT_LIBTOOL_MAJ@:@MEDIMPORT_LIBTOOL_MIN@:@MEDIMPORT_LIBTOOL_REL@
 libmedimport_la_LIBADD  = $(top_builddir)/src/libmed.la \
 $(top_builddir)/src/2.3.6/ci/libci.la $(top_builddir)/src/2.3.6/hdfi/libhdfi.la $(top_builddir)/src/2.3.6/misc/libmisc.la 2.1.x/libMaj21To22.la 2.3.1/libMaj231To232.la 2.3.6/libMaj236To300.la 3.0.0/libMaj300To310.la 3.1.0/libMaj310To320.la 3.2.0/libMaj320To330.la 4.0.0/libMaj400To410.la \
 $(top_builddir)/src/hdfi/MEDattrFermer.lo \
