--- origsrc/zycore-c/CMakeLists.txt	2020-11-04 07:26:06.000000000 +0900
+++ src/zycore-c/CMakeLists.txt	2020-11-04 07:38:25.675035900 +0900
@@ -9,6 +9,8 @@ include(CMakePackageConfigHelpers)
 
 project(Zycore VERSION 1.0.0.0 LANGUAGES C CXX)
 
+enable_testing()
+
 # =============================================================================================== #
 # Overridable options                                                                             #
 # =============================================================================================== #
@@ -40,7 +42,7 @@ option(ZYCORE_BUILD_TESTS
 # =============================================================================================== #
 
 # Download and unpack googletest
-if (ZYCORE_BUILD_TESTS)
+if (ZYCORE_BUILD_TESTS AND NOT USE_SYSTEM_GTEST)
     if (NOT DEFINED ZYCORE_DOWNLOADED_GTEST)
         configure_file("CMakeLists.txt.in" "${CMAKE_BINARY_DIR}/gtest/download/CMakeLists.txt")
         execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
@@ -121,6 +123,9 @@ else ()
     add_library("Zycore" STATIC)
 endif ()
 
+if (CYGWIN)
+    set_target_properties(Zycore PROPERTIES SOVERSION 0)
+endif()
 set_target_properties("Zycore" PROPERTIES LINKER_LANGUAGE C)
 target_include_directories("Zycore"
     PUBLIC "include" ${PROJECT_BINARY_DIR}
@@ -195,10 +200,17 @@ configure_package_config_file(cmake/zyco
     "${CMAKE_CURRENT_BINARY_DIR}/zycore-config.cmake"
     INSTALL_DESTINATION "${CMAKE_INSTALL_PREFIX}/cmake"
 )
-install(FILES
-    "${CMAKE_CURRENT_BINARY_DIR}/zycore-config.cmake"
-    DESTINATION "${CMAKE_INSTALL_PREFIX}/cmake"
-)
+if (CYGWIN)
+    install(FILES
+        "${CMAKE_CURRENT_BINARY_DIR}/zycore-config.cmake"
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/zycore-c"
+    )
+else()
+    install(FILES
+        "${CMAKE_CURRENT_BINARY_DIR}/zycore-config.cmake"
+        DESTINATION "${CMAKE_INSTALL_PREFIX}/cmake"
+    )
+endif()
 
 install(TARGETS "Zycore"
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
@@ -253,6 +265,7 @@ function (zyan_add_test test)
     set_target_properties("Test${test}" PROPERTIES FOLDER "Tests")
     target_compile_definitions("Test${test}" PRIVATE "_CRT_SECURE_NO_WARNINGS")
     zyan_maybe_enable_wpo("Test${test}")
+    add_test(NAME "${test}" COMMAND "Test${test}")
 endfunction ()
 
 if (ZYCORE_BUILD_TESTS)
@@ -262,3 +275,10 @@ if (ZYCORE_BUILD_TESTS)
 endif ()
 
 # =============================================================================================== #
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/zycore-c.pc.in ${CMAKE_BINARY_DIR}/zycore-c.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/zycore-c.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/zycore-c/zycore-c.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/zycore-c/zycore-c.pc.in	2020-11-04 07:40:04.871586200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: zycore-c
+Description: Zyan Core Library for C
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lzycore
+Cflags: -I${includedir}
