--- origsrc/zycore-c-1.1.0/CMakeLists.txt	2021-10-26 14:13:51.000000000 +0900
+++ src/zycore-c-1.1.0/CMakeLists.txt	2022-06-22 11:39:15.164926500 +0900
@@ -82,12 +82,12 @@ if (ZYCORE_BUILD_TESTS)
 
         set(ZYCORE_DOWNLOADED_GTEST TRUE CACHE BOOL "")
         mark_as_advanced(ZYCORE_DOWNLOADED_GTEST)
+
+        add_subdirectory("${CMAKE_BINARY_DIR}/gtest/src" "${CMAKE_BINARY_DIR}/gtest/build"
+            EXCLUDE_FROM_ALL)
     endif ()
 
     set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
-
-    add_subdirectory("${CMAKE_BINARY_DIR}/gtest/src" "${CMAKE_BINARY_DIR}/gtest/build"
-        EXCLUDE_FROM_ALL)
 endif ()
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
@@ -244,12 +244,21 @@ function (zyan_add_test test)
     set_target_properties("Test${test}" PROPERTIES FOLDER "Tests")
     target_compile_definitions("Test${test}" PRIVATE "_CRT_SECURE_NO_WARNINGS")
     zyan_maybe_enable_wpo("Test${test}")
+    add_test(NAME "${test}" COMMAND "Test${test}")
 endfunction ()
 
 if (ZYCORE_BUILD_TESTS)
+    enable_testing()
     zyan_add_test("String")
     zyan_add_test("Vector")
     zyan_add_test("ArgParse")
 endif ()
 
 # =============================================================================================== #
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/zycore-c.pc.in ${CMAKE_BINARY_DIR}/zycore-c.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/zycore-c.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/zycore-c-1.1.0/zycore-c.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/zycore-c-1.1.0/zycore-c.pc.in	2022-06-22 11:40:14.228977500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: zycore-c
+Description: Zyan Core Library for C
+Version: @VERSION@
+Libs: -L${libdir} -lZycore
+Cflags: -I${includedir}
