--- origsrc/zycore-c-1.5.2/CMakeLists.txt	2025-02-10 00:19:45.000000000 +0900
+++ src/zycore-c-1.5.2/CMakeLists.txt	2025-09-08 21:59:35.602133500 +0900
@@ -229,7 +229,7 @@ install(DIRECTORY "include/" DESTINATION
 # =============================================================================================== #
 
 find_package(Doxygen)
-if (DOXYGEN_FOUND)
+if (BUILD_DOCS AND DOXYGEN_FOUND)
     set(DOXYGEN_GENERATE_MAN YES)
     doxygen_add_docs(ZycoreDoc "include/Zycore/" ALL)
     install(
@@ -300,3 +300,9 @@ if (ZYCORE_BUILD_TESTS)
 endif ()
 
 # =============================================================================================== #
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/zycore-c.pc.in ${CMAKE_BINARY_DIR}/zycore-c.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/zycore-c.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/zycore-c-1.5.2/zycore-c.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/zycore-c-1.5.2/zycore-c.pc.in	2025-09-08 21:59:35.615570100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: zycore-c
+Description: Zyan Core Library for C
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lZycore
+Cflags: -I${includedir}
