--- origsrc/ticcutils-0.36/configure.ac	2024-12-15 20:49:35.000000000 +0900
+++ src/ticcutils-0.36/configure.ac	2025-12-05 14:36:37.614369300 +0900
@@ -91,7 +91,7 @@ PKG_CHECK_MODULES([XML2], [libxml-2.0 >=
 CXXFLAGS="$CXXFLAGS $XML2_CFLAGS"
 LIBS="$LIBS $XML2_LIBS"
 
-PKG_CHECK_MODULES([ICU], [icu-uc >= 50 icu-io] )
+PKG_CHECK_MODULES([ICU], [icu-uc >= 50 icu-io icu-i18n] )
 CXXFLAGS="$CXXFLAGS $ICU_CFLAGS"
 LIBS="$ICU_LIBS $LIBS"
 
--- origsrc/ticcutils-0.36/src/Makefile.am	2024-12-15 20:49:35.000000000 +0900
+++ src/ticcutils-0.36/src/Makefile.am	2025-12-05 14:28:05.440621600 +0900
@@ -11,6 +11,7 @@ libticcutils_la_SOURCES = LogStream.cxx
 	FileUtils.cxx CommandLine.cxx SocketBasics.cxx ServerBase.cxx \
 	FdStream.cxx Unicode.cxx UniHash.cxx
 
+libticcutils_la_LDFLAGS += -no-undefined
 
 check_PROGRAMS = runtest testlogstream
 runtest_SOURCES = runtest.cxx
