--- origsrc/ticcutils-0.25/src/Makefile.am	2020-11-17 22:40:21.000000000 +0900
+++ src/ticcutils-0.25/src/Makefile.am	2021-01-28 20:23:03.958532800 +0900
@@ -11,6 +11,7 @@ libticcutils_la_SOURCES = LogStream.cxx
 	FileUtils.cxx CommandLine.cxx SocketBasics.cxx ServerBase.cxx \
 	FdStream.cxx Unicode.cxx
 
+libticcutils_la_LDFLAGS += -no-undefined
 
 check_PROGRAMS = runtest testlogstream
 runtest_SOURCES = runtest.cxx
