--- origsrc/ticcutils-0.30/src/Makefile.am	2023-01-02 23:01:44.000000000 +0900
+++ src/ticcutils-0.30/src/Makefile.am	2023-01-19 09:45:00.489148300 +0900
@@ -11,6 +11,7 @@ libticcutils_la_SOURCES = LogStream.cxx
 	FileUtils.cxx CommandLine.cxx SocketBasics.cxx ServerBase.cxx \
 	FdStream.cxx Unicode.cxx UniHash.cxx
 
+libticcutils_la_LDFLAGS += -no-undefined
 
 check_PROGRAMS = runtest testlogstream
 runtest_SOURCES = runtest.cxx
