--- origsrc/ticcutils-0.36/src/Makefile.am	2024-12-15 20:49:35.000000000 +0900
+++ src/ticcutils-0.36/src/Makefile.am	2025-01-18 10:19:01.619237100 +0900
@@ -11,6 +11,7 @@ libticcutils_la_SOURCES = LogStream.cxx
 	FileUtils.cxx CommandLine.cxx SocketBasics.cxx ServerBase.cxx \
 	FdStream.cxx Unicode.cxx UniHash.cxx
 
+libticcutils_la_LDFLAGS += -no-undefined
 
 check_PROGRAMS = runtest testlogstream
 runtest_SOURCES = runtest.cxx
