--- origsrc/ticcutils-0.27/src/Makefile.am	2021-07-13 01:02:54.000000000 +0900
+++ src/ticcutils-0.27/src/Makefile.am	2021-08-25 21:51:09.289806900 +0900
@@ -11,6 +11,7 @@ libticcutils_la_SOURCES = LogStream.cxx
 	FileUtils.cxx CommandLine.cxx SocketBasics.cxx ServerBase.cxx \
 	FdStream.cxx Unicode.cxx
 
+libticcutils_la_LDFLAGS += -no-undefined
 
 check_PROGRAMS = runtest testlogstream
 runtest_SOURCES = runtest.cxx
