--- origsrc/poslib-1.0.6/poslib/Makefile.am	2004-05-18 23:47:45.000000000 +0900
+++ src/poslib-1.0.6/poslib/Makefile.am	2014-11-08 11:36:12.585968700 +0900
@@ -1,4 +1,4 @@
-SUBDIRS = server
+SUBDIRS = . server
 
 #if DEBUG
 #lib_LTLIBRARIES = libposlib.a
@@ -8,6 +8,7 @@ lib_LTLIBRARIES = libposlib.la
 INCLUDES = -I$(top_srcdir)
 
 #libposlib_la_LDFLAGS = -version-info $(VERINFO) -release $(VERSION)
+libposlib_la_LDFLAGS = -no-undefined
 
 libposlib_la_includedir=$(includedir)/$(PACKAGE)-$(VERSION)/poslib
 
@@ -28,7 +29,6 @@ libposlib_la_include_HEADERS = \
   syssocket.h \
   sysstl.h \
   sysstring.h \
-  syssocket.h \
   types.h \
   vsnprintf.h \
   w32poll.h
--- origsrc/poslib-1.0.6/poslib/server/Makefile.am	2004-04-20 04:15:03.000000000 +0900
+++ src/poslib-1.0.6/poslib/server/Makefile.am	2014-11-08 11:36:12.601593700 +0900
@@ -22,6 +22,9 @@ libposserver_la_SOURCES = \
   srvresolver.cpp \
   serverthread.cpp
 
+libposserver_la_LDFLAGS = -no-undefined
+libposserver_la_LIBADD = $(top_builddir)/poslib/libposlib.la
+
 #if DEBUG
 #lib_LIBRARIES = libposserver.a
 #libposserver_a_SOURCES = $(libposserver_la_SOURCES)
--- origsrc/poslib-1.0.6/tools/fileclient.cpp	2003-02-01 23:29:23.000000000 +0900
+++ src/poslib-1.0.6/tools/fileclient.cpp	2014-11-08 11:36:12.601593700 +0900
@@ -18,6 +18,8 @@
     Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 */
 
+#include <limits.h>
+
 #include <poslib/poslib.h>
 
 int main(int argc, char **argv) {
