--- origsrc/mimalloc-2.0.6/CMakeLists.txt	2022-04-15 08:54:04.000000000 +0900
+++ src/mimalloc-2.0.6/CMakeLists.txt	2022-06-02 07:48:33.365402500 +0900
@@ -319,7 +319,7 @@ if(MI_BUILD_SHARED)
     install(FILES "$<TARGET_FILE_DIR:mimalloc>/mimalloc-redirect${MIMALLOC_REDIRECT_SUFFIX}.dll" DESTINATION ${mi_install_libdir})
   endif()
 
-  install(TARGETS mimalloc EXPORT mimalloc DESTINATION ${mi_install_libdir} LIBRARY)  
+  install(TARGETS mimalloc EXPORT mimalloc RUNTIME DESTINATION bin ARCHIVE DESTINATION ${mi_install_libdir})  
   install(EXPORT mimalloc DESTINATION ${mi_install_cmakedir})
 endif()
 
@@ -411,3 +411,12 @@ if (MI_OVERRIDE)
     endif()
   endif()
 endif()
+
+if (CYGWIN)
+  set(prefix ${CMAKE_INSTALL_PREFIX})
+  set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+  set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+  set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+  configure_file(${CMAKE_SOURCE_DIR}/mimalloc.pc.in ${CMAKE_BINARY_DIR}/mimalloc.pc @ONLY)
+  install(FILES ${CMAKE_BINARY_DIR}/mimalloc.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+endif()
--- origsrc/mimalloc-2.0.6/mimalloc.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/mimalloc-2.0.6/mimalloc.pc.in	2022-06-02 07:52:05.651796600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mimalloc
+Description: Compact general purpose allocator with excellent performance
+Version: @mi_version_major@.@mi_version_minor@.@mi_version_patch@
+Libs: -L${libdir} -lmimalloc
+Cflags: -I${includedir}
--- origsrc/mimalloc-2.0.6/src/os.c	2022-04-15 08:54:04.000000000 +0900
+++ src/mimalloc-2.0.6/src/os.c	2022-06-02 07:06:57.123022100 +0900
@@ -1161,7 +1161,7 @@ static void* mi_os_alloc_huge_os_pagesx(
   return VirtualAlloc(addr, size, flags, PAGE_READWRITE);
 }
 
-#elif defined(MI_OS_USE_MMAP) && (MI_INTPTR_SIZE >= 8) && !defined(__HAIKU__)
+#elif defined(MI_OS_USE_MMAP) && (MI_INTPTR_SIZE >= 8) && !defined(__HAIKU__) && !defined(__CYGWIN__)
 #include <sys/syscall.h>
 #ifndef MPOL_PREFERRED
 #define MPOL_PREFERRED 1
