--- origsrc/mimalloc-2.0.9/CMakeLists.txt	2022-12-24 06:31:56.000000000 +0900
+++ src/mimalloc-2.0.9/CMakeLists.txt	2023-02-07 19:38:52.262530400 +0900
@@ -383,7 +383,7 @@ if(MI_BUILD_SHARED)
     install(FILES "$<TARGET_FILE_DIR:mimalloc>/mimalloc-redirect${MIMALLOC_REDIRECT_SUFFIX}.dll" DESTINATION ${mi_install_libdir})
   endif()
 
-  install(TARGETS mimalloc EXPORT mimalloc DESTINATION ${mi_install_libdir} LIBRARY)
+  install(TARGETS mimalloc EXPORT mimalloc RUNTIME DESTINATION bin ARCHIVE DESTINATION ${mi_install_libdir})
   install(EXPORT mimalloc DESTINATION ${mi_install_cmakedir})
 endif()
 
--- origsrc/mimalloc-2.0.9/src/os.c	2022-12-24 06:31:56.000000000 +0900
+++ src/mimalloc-2.0.9/src/os.c	2023-02-07 19:36:58.634354000 +0900
@@ -1197,7 +1197,7 @@ static void* mi_os_alloc_huge_os_pagesx(
   return VirtualAlloc(addr, size, flags, PAGE_READWRITE);
 }
 
-#elif defined(MI_OS_USE_MMAP) && (MI_INTPTR_SIZE >= 8) && !defined(__HAIKU__)
+#elif defined(MI_OS_USE_MMAP) && (MI_INTPTR_SIZE >= 8) && !defined(__HAIKU__) && !defined(__CYGWIN__)
 #include <sys/syscall.h>
 #ifndef MPOL_PREFERRED
 #define MPOL_PREFERRED 1
