--- origsrc/JsonBox-0.6.2/CMakeLists.txt	2015-05-23 05:48:51.000000000 +0900
+++ src/JsonBox-0.6.2/CMakeLists.txt	2019-01-02 13:19:58.595374000 +0900
@@ -34,6 +34,9 @@ set(JSONBOX_HEADERS
 # build library
 set(CMAKE_DEBUG_POSTFIX "_d")
 add_library(JsonBox ${JSONBOX_SOURCES} ${JSONBOX_HEADERS})
+if (CYGWIN)
+  set_target_properties(JsonBox PROPERTIES SOVERSION 0)
+endif()
 
 generate_export_header(JsonBox EXPORT_FILE_NAME Export.h)
 
@@ -73,7 +76,7 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR
   DESTINATION include/JsonBox
 )
 install(EXPORT ${PROJECT_NAME}
-  DESTINATION lib${LIB_SUFFIX}/cmake
+  DESTINATION lib${LIB_SUFFIX}/cmake/JsonBox
   FILE ${PROJECT_NAME}Config.cmake
 )
 
@@ -82,3 +85,10 @@ install(EXPORT ${PROJECT_NAME}
 add_executable(example1 "${CMAKE_CURRENT_SOURCE_DIR}/examples/main.cpp")
 add_dependencies(example1 JsonBox)
 target_link_libraries(example1 JsonBox)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/JsonBox.pc.in ${CMAKE_BINARY_DIR}/JsonBox.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/JsonBox.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/JsonBox-0.6.2/JsonBox.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/JsonBox-0.6.2/JsonBox.pc.in	2019-01-02 13:19:58.516605200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: JsonBox
+Description: 
+Version: @VERSION@
+Libs: -L${libdir} -lJsonBox
+Cflags: -I${includedir}
