--- origsrc/JsonBox-0.6.2/CMakeLists.txt	2015-05-23 05:48:51.000000000 +0900
+++ src/JsonBox-0.6.2/CMakeLists.txt	2025-08-07 11:16:37.032202500 +0900
@@ -35,6 +35,10 @@ set(JSONBOX_HEADERS
 set(CMAKE_DEBUG_POSTFIX "_d")
 add_library(JsonBox ${JSONBOX_SOURCES} ${JSONBOX_HEADERS})
 
+if (DEFINED SOVERSION)
+  set_target_properties(JsonBox PROPERTIES SOVERSION ${SOVERSION})
+endif()
+
 generate_export_header(JsonBox EXPORT_FILE_NAME Export.h)
 
 target_include_directories(JsonBox PRIVATE
@@ -73,7 +77,7 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR
   DESTINATION include/JsonBox
 )
 install(EXPORT ${PROJECT_NAME}
-  DESTINATION lib${LIB_SUFFIX}/cmake
+  DESTINATION lib${LIB_SUFFIX}/cmake/${PROJECT_NAME}
   FILE ${PROJECT_NAME}Config.cmake
 )
 
@@ -82,3 +86,9 @@ install(EXPORT ${PROJECT_NAME}
 add_executable(example1 "${CMAKE_CURRENT_SOURCE_DIR}/examples/main.cpp")
 add_dependencies(example1 JsonBox)
 target_link_libraries(example1 JsonBox)
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/JsonBox.pc.in ${CMAKE_BINARY_DIR}/JsonBox.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/JsonBox.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/JsonBox-0.6.2/JsonBox.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/JsonBox-0.6.2/JsonBox.pc.in	2025-08-07 11:16:36.951630300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: JsonBox
+Description: C++ library used to read and write JSON with ease and speed
+Version: @VERSION@
+Libs: -L${libdir} -lJsonBox
+Cflags: -I${includedir}
