diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/aces_container-1.0.1/CMakeLists.txt src/aces_container-1.0.1/CMakeLists.txt
--- origsrc/aces_container-1.0.1/CMakeLists.txt	2015-10-24 05:52:07.000000000 +0900
+++ src/aces_container-1.0.1/CMakeLists.txt	2015-11-02 20:33:53.559515000 +0900
@@ -53,6 +53,7 @@ set( AcesContainer_MINOR_VERSION 0 )
 set( AcesContainer_PATCH_VERSION 1 )
 set( AcesContainer_VERSION ${AcesContainer_MAJOR_VERSION}.${AcesContainer_MINOR_VERSION}.${AcesContainer_PATCH_VERSION} )
 
+set( INSTALL_BIN_DIR bin CACHE PATH "Install directory for runtimes")
 set( INSTALL_LIB_DIR lib CACHE PATH "Install directory for libraries" )
 set( INSTALL_INCLUDE_DIR include CACHE PATH "Install directory for public header files" )
 
@@ -64,7 +65,7 @@ endif()
 set(INSTALL_CMAKE_DIR ${DEF_INSTALL_CMAKE_DIR} CACHE PATH "Install directory for project CMake files" )
 
 ## convert install paths to absolute
-foreach( p LIB INCLUDE CMAKE )
+foreach( p BIN LIB INCLUDE CMAKE )
   set( var INSTALL_${p}_DIR )
   if( NOT IS_ABSOLUTE "${${var}}" )
     set( ${var} "${CMAKE_INSTALL_PREFIX}/${${var}}" )
@@ -83,7 +84,7 @@ add_library( AcesContainer SHARED
 	aces_md5.cc
 )
 
-install (TARGETS AcesContainer EXPORT AcesContainerTargets DESTINATION ${INSTALL_LIB_DIR})
+install (TARGETS AcesContainer EXPORT AcesContainerTargets RUNTIME DESTINATION ${INSTALL_BIN_DIR} ARCHIVE DESTINATION ${INSTALL_LIB_DIR})
 install (FILES 
 			aces_errors.h
 			aces_genericWriter.h
@@ -100,7 +101,9 @@ install (FILES
 		 DESTINATION 
 		 	${INSTALL_INCLUDE_DIR}/aces
 		 )
-
+if(CYGWIN)
+  set_target_properties(AcesContainer PROPERTIES SOVERSION 0)
+endif()
 
 find_package( PkgConfig )
 if ( PKG_CONFIG_FOUND )
