--- origsrc/e00compr-1.0.1/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/e00compr-1.0.1/Makefile.am	2012-06-23 16:18:37.079929900 +0900
@@ -0,0 +1,14 @@
+
+bin_PROGRAMS = e00conv
+e00conv_SOURCES = e00conv.c
+e00conv_LDADD = libe00compr.la
+
+e00compr_includedir = $(includedir)/e00compr
+e00compr_include_HEADERS = e00compr.h cpl_port.h cpl_conv.h cpl_error.h cpl_vsi.h
+
+lib_LTLIBRARIES=libe00compr.la
+libe00compr_la_SOURCES = e00read.c e00write.c cpl_error.c cpl_conv.c cpl_vsisimple.c
+libe00compr_la_LDFLAGS = -no-undefined
+
+man_MANS = e00conv.1
+
--- origsrc/e00compr-1.0.1/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/e00compr-1.0.1/configure.ac	2012-06-23 16:20:46.224339700 +0900
@@ -0,0 +1,28 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([e00compr], [1.0.1])
+AC_CONFIG_SRCDIR([cpl_conv.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h string.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([strchr strdup strerror strstr])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
