--- origsrc/libestr-0.1.0/configure.ac	2010-11-30 20:26:08.000000000 +0900
+++ src/libestr-0.1.0/configure.ac	2011-04-08 08:51:45.841951900 +0900
@@ -15,7 +15,7 @@ if test "$GCC" = "yes"
 then CFLAGS="$CFLAGS -W -Wall -Wformat-security -Wshadow -Wcast-align -Wpointer-arith -Wmissing-format-attribute -g"
 fi
 
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 # Checks for libraries.
 save_LIBS=$LIBS
--- origsrc/libestr-0.1.0/src/Makefile.am	2010-11-04 23:44:31.000000000 +0900
+++ src/libestr-0.1.0/src/Makefile.am	2011-04-08 08:52:20.001070300 +0900
@@ -12,6 +12,6 @@ libestr_la_SOURCES = \
 
 libestr_la_CPPFLAGS = $(LIBESTR_CFLAGS)
 libestr_la_LIBADD = 
-libestr_la_LDFLAGS = -version-info 0:0:0
+libestr_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 
 include_HEADERS = 
