--- origsrc/libaacs-0.8.1/configure.ac	2015-03-13 18:20:13.000000000 +0900
+++ src/libaacs-0.8.1/configure.ac	2015-05-23 09:51:38.116414000 +0900
@@ -149,7 +149,7 @@ AS_IF([test "x$enable_extra_warnings" !=
 AS_IF([test "x$enable_optimizations" != "xno"], [
   CC_CHECK_CFLAGS_APPEND([-O3 -fomit-frame-pointer])
 ])
-
+CC_CHECK_CFLAGS_APPEND([-std=gnu99])
 
 dnl export library version number
 AACS_VERSION_MAJOR=aacs_major()
