--- origsrc/libaacs-0.8.0/configure.ac	2015-01-23 20:27:45.000000000 +0900
+++ src/libaacs-0.8.0/configure.ac	2015-02-11 11:45:42.296619400 +0900
@@ -183,7 +183,7 @@ AS_IF([test "x$enable_extra_warnings" !=
 AS_IF([test "x$enable_optimizations" != "xno"], [
   CC_CHECK_CFLAGS_APPEND([-O3 -fomit-frame-pointer])
 ])
-
+CC_CHECK_CFLAGS_APPEND([-std=gnu99])
 
 dnl export library version number
 AACS_VERSION_MAJOR=aacs_major()
