--- origsrc/libaacs-0.3.0/src/Makefile.am	2011-12-01 08:08:59.000000000 +0900
+++ src/libaacs-0.3.0/src/Makefile.am	2012-01-28 11:27:51.827901400 +0900
@@ -6,7 +6,7 @@ SET_FEATURES = -D_ISOC99_SOURCE -D_POSIX
 SET_WARNINGS = @SET_WARNINGS@
 SET_OPTIMIZATIONS = @SET_OPTIMIZATIONS@
 SET_DEBUG_OPTS = @SET_DEBUG_OPTS@
-SET_INCLUDES = -I$(top_srcdir) -Ifile
+SET_INCLUDES = -I$(top_srcdir) -Ifile -I$(top_srcdir)/src/file
 
 AM_CFLAGS = -std=c99 $(SET_FEATURES) $(SET_WARNINGS) $(SET_OPTIMIZATIONS) \
 	$(SET_DEBUG_OPTS) $(SET_INCLUDES) $(LIBGCRYPT_CFLAGS)
@@ -50,7 +50,7 @@ endif
 
 libaacs_ladir= $(includedir)/libaacs
 libaacs_la_HEADERS= libaacs/aacs.h file/filesystem.h libaacs/aacs-version.h
-libaacs_la_LDFLAGS= $(LIBGCRYPT_LIBS) -version-info $(LIB_VERSION_INFO)
+libaacs_la_LDFLAGS= $(LIBGCRYPT_LIBS) -version-info $(LIB_VERSION_INFO) -no-undefined
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libaacs.pc
 
