--- origsrc/libaacs-0.7.0/src/Makefile.am	2013-12-18 18:21:16.000000000 +0900
+++ src/libaacs-0.7.0/src/Makefile.am	2014-05-22 21:54:35.107275500 +0900
@@ -81,7 +81,7 @@ parser_test_SOURCES = examples/parser_te
 	file/keydbcfg-lexer.l \
 	util/strutl.c \
 	util/logging.c
-parser_test_CFLAGS = -std=c99 $(SET_FEATURES) $(SET_INCLUDES)
+parser_test_CFLAGS = -std=gnu99 $(SET_FEATURES) $(SET_INCLUDES)
 
 aacs_info_SOURCES = examples/aacs_info.c
 aacs_info_CFLAGS = -std=c99 $(SET_FEATURES) $(SET_INCLUDES)
