--- origsrc/libaacs/src/Makefile.am	2011-08-14 14:21:20.000000000 +0900
+++ src/libaacs/src/Makefile.am	2011-08-14 14:21:26.518824200 +0900
@@ -11,7 +11,7 @@ SET_FEATURES = -D_ISOC99_SOURCE -D_POSIX
 SET_WARNINGS = @SET_WARNINGS@
 SET_OPTIMIZATIONS = @SET_OPTIMIZATIONS@
 SET_DEBUG_OPTS = @SET_DEBUG_OPTS@
-SET_INCLUDES = -I$(top_srcdir) -Ifile
+SET_INCLUDES = -I$(top_srcdir) -Ifile -I$(top_srcdir)/src/file
 
 AM_CFLAGS = -std=c99 $(SET_FEATURES) $(SET_WARNINGS) $(SET_OPTIMIZATIONS) \
 	$(SET_DEBUG_OPTS) $(SET_INCLUDES) $(LIBGCRYPT_CFLAGS)
@@ -53,7 +53,7 @@ endif
 
 libaacs_ladir= $(includedir)/libaacs
 libaacs_la_HEADERS= libaacs/aacs.h file/filesystem.h
-libaacs_la_LDFLAGS= $(LIBGCRYPT_LIBS) -version-info $(LIB_VERSION_INFO)
+libaacs_la_LDFLAGS= $(LIBGCRYPT_LIBS) -version-info $(LIB_VERSION_INFO) -no-undefined
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libaacs.pc
 
