--- origsrc/libaacs-0.7.1/configure.ac	2014-05-27 17:12:00.000000000 +0900
+++ src/libaacs-0.7.1/configure.ac	2014-07-27 09:04:22.564805400 +0900
@@ -180,7 +180,7 @@ AS_IF([test "x$enable_extra_warnings" !=
 AS_IF([test "x$enable_optimizations" != "xno"], [
   CC_CHECK_CFLAGS_APPEND([-O3 -fomit-frame-pointer])
 ])
-
+CC_CHECK_CFLAGS_APPEND([-std=gnu99])
 
 dnl export library version number
 AACS_VERSION_MAJOR=aacs_major()
--- origsrc/libaacs-0.7.1/src/Makefile.am	2014-05-27 17:18:58.000000000 +0900
+++ src/libaacs-0.7.1/src/Makefile.am	2014-07-27 01:06:03.989388900 +0900
@@ -81,7 +81,7 @@ parser_test_SOURCES = examples/parser_te
 	file/keydbcfg-lexer.l \
 	util/strutl.c \
 	util/logging.c
-parser_test_CFLAGS = -std=c99 $(SET_FEATURES) $(SET_INCLUDES)
+parser_test_CFLAGS = -std=gnu99 $(SET_FEATURES) $(SET_INCLUDES)
 
 aacs_info_SOURCES = examples/aacs_info.c
 aacs_info_CFLAGS = -std=c99 $(SET_FEATURES) $(SET_INCLUDES)
