--- origsrc/validator-0.2.2/install.c	2023-11-28 18:49:04.000000000 +0900
+++ src/validator-0.2.2/install.c	2023-12-16 08:18:10.632738400 +0900
@@ -21,6 +21,7 @@
 #include "config.h"
 #include "main.h"
 
+#include <errno.h>
 #include <fcntl.h>
 #include <unistd.h>
 
--- origsrc/validator-0.2.2/utils.h	2023-11-28 18:49:04.000000000 +0900
+++ src/validator-0.2.2/utils.h	2023-12-16 08:19:35.191338000 +0900
@@ -1,3 +1,6 @@
+#include <errno.h>
+#include <unistd.h>
+
 #include <glib.h>
 
 #include <openssl/evp.h>
