--- origsrc/mcache-client-0.2.14/src/Makefile.am	2016-03-11 14:56:33.000000000 +0900
+++ src/mcache-client-0.2.14/src/Makefile.am	2025-01-24 16:11:07.579644300 +0900
@@ -39,8 +39,8 @@ SRC = mcache.cc logger.cc error.cc init.
                                                                                \
       hash/city.cc hash/jenkins.cc hash/murmur3.cc hash/spooky.cc
 
-libmcache_la_LIBADD = @LIBTBB_LIBS@ @ZLIB_LIBS@
-libmcache_la_LDFLAGS = @VERSION_INFO@
+libmcache_la_LIBADD = @LIBTBB_LIBS@ @ZLIB_LIBS@ $(LIBMCACHE_EXTRA_LIBS)
+libmcache_la_LDFLAGS = -no-undefined @VERSION_INFO@
 libmcache_la_SOURCES = $(SRC)
 
 EXTRA_PROGRAMS = test-mcache test-pool test-consistent-hashing                 \
