--- origsrc/pipemeter-1.1.4/pipemeter.c	2020-10-25 01:44:33.000000000 +0900
+++ src/pipemeter-1.1.4/pipemeter.c	2020-11-07 05:22:48.285161600 +0900
@@ -397,7 +397,7 @@ void show_just_rate(int sig) {
     fprintf(stderr,"\n");
     exit(1);
   } else {
-    fprintf(stderr,trailer);
+    fprintf(stderr,"%s", trailer);
   }
 }
  
@@ -487,7 +487,7 @@ void show_progress(int sig) {
   }
 
   strncpy(progressbar+1,progressfill,progress);
-  fprintf(stderr, progressbar);
+  fprintf(stderr, "%s", progressbar);
   fprintf(stderr," %s/s",buf2);
   formatbytes(buf2,bytes);
   fprintf(stderr," %s",buf2);
@@ -497,7 +497,7 @@ void show_progress(int sig) {
     fprintf(stderr,"\n");
     exit(0);
   } else {
-    fprintf(stderr,trailer);
+    fprintf(stderr,"%s",trailer);
   }
 }
 
--- origsrc/pipemeter-1.1.4/testscript.sh	2020-10-25 01:44:33.000000000 +0900
+++ src/pipemeter-1.1.4/testscript.sh	2020-11-07 05:18:01.128910600 +0900
@@ -1,6 +1,6 @@
 #!/bin/sh
 TESTSIZE=200M
-RESULTS="results.txt"
+RESULTS="results.yacp.txt"
 TIME=/usr/bin/time
 if [ ! -x ./pipemeter ] ; then
   echo "You must compile pipemeter before testing it."
