--- origsrc/openobex/acinclude.m4	2010-11-28 01:17:10.000000000 +0900
+++ src/openobex/acinclude.m4	2010-11-28 01:18:14.600862200 +0900
@@ -160,6 +160,13 @@ AC_DEFUN([AC_PATH_USB], [
 		USB_LIBS="-lusb"
 		usb_lib_found=yes
 		;;
+	*-*-cygwin*)
+		AC_CHECK_PROGS([LIBUSB_CONFIG], [libusb-config])
+		if test x != x$LIBUSB_CONFIG; then
+			USB_LIBS=`$LIBUSB_CONFIG --libs`
+			USB_CFLAGS=`$LIBUSB_CONFIG --cflags`
+		fi
+		;;
 	*)
 		PKG_CHECK_MODULES(USB, libusb, usb_lib_found=yes, usb_lib_found=no)
 		AC_CHECK_FILE(${prefix}/lib/pkgconfig/libusb.pc, REQUIRES="libusb")
--- origsrc/openobex/configure.ac	2010-11-28 01:17:10.000000000 +0900
+++ src/openobex/configure.ac	2010-11-28 01:18:45.675545400 +0900
@@ -21,9 +21,7 @@ AC_PROG_INSTALL
 m4_define([_LT_AC_TAGCONFIG], [])
 m4_ifdef([AC_LIBTOOL_TAGS], [AC_LIBTOOL_TAGS([])])
 
-AC_LIBTOOL_WIN32_DLL
-
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 AC_VISIBILITY
 AC_CANONICAL_HOST
