--- origsrc/libhandy-1.8.3/src/meson.build	2024-02-09 22:29:11.000000000 +0900
+++ src/libhandy-1.8.3/src/meson.build	2025-12-11 13:45:51.836382300 +0900
@@ -263,7 +263,7 @@ if cc.links('int main() { return 0; }',
 endif
 
 # set default libdir on win32 for libhandy target to keep MinGW compatibility
-if target_system == 'windows'
+if target_system == 'windows' or target_system == 'cygwin'
   handy_libdir = [true]
 else
   handy_libdir = libdir
