--- origsrc/libhandy-1.0.0/src/meson.build	2020-09-08 21:26:03.000000000 +0900
+++ src/libhandy-1.0.0/src/meson.build	2020-09-26 14:34:50.574126300 +0900
@@ -181,7 +181,7 @@ libhandy_deps = [
   dependency('glib-2.0', version: glib_min_version),
   dependency('gio-2.0', version: glib_min_version),
   dependency('gmodule-2.0', version: glib_min_version),
-  dependency('gtk+-3.0', version: '>= 3.24.1'),
+  dependency('gtk+-3.0', version: host_machine.system() == 'cygwin' ? '>= 3.22.28' : '>= 3.24.1'),
   cc.find_library('m', required: false),
   cc.find_library('rt', required: false),
 ]
@@ -222,7 +222,7 @@ if cc.links('int main() { return 0; }',
 endif
 
 # set default libdir on win32 for libhandy target to keep MinGW compatibility
-if target_system == 'windows'
+if target_system == 'windows' or host_machine.system() == 'cygwin'
   handy_libdir = [true]
 else
   handy_libdir = libdir
