--- origsrc/ompl-1.3.2-Source/CMakeModules/CompilerSettings.cmake	2017-11-11 14:55:33.000000000 +0900
+++ src/ompl-1.3.2-Source/CMakeModules/CompilerSettings.cmake	2017-11-29 20:11:14.899000000 +0900
@@ -6,7 +6,11 @@ if (CMAKE_VERSION VERSION_LESS "3.1")
 else()
     set(CMAKE_CXX_STANDARD 11)
     set(CMAKE_CXX_STANDARD_REQUIRED ON)
-    set(CMAKE_CXX_EXTENSIONS OFF)
+    if(CYGWIN)
+        set(CMAKE_CXX_EXTENSIONS ON)
+    else()
+        set(CMAKE_CXX_EXTENSIONS OFF)
+    endif()
     # this next line shouldn't be necessary, but doesn't always get added by cmake (e.g., for clang++-5)
     add_definitions(-std=c++11)
 endif()
@@ -58,9 +62,9 @@ if(IS_XLC)
     set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -q64")
 endif(IS_XLC)
 
-if((CMAKE_COMPILER_IS_GNUCXX OR IS_ICPC) AND NOT MINGW)
+if((CMAKE_COMPILER_IS_GNUCXX OR IS_ICPC) AND NOT MINGW AND NOT CYGWIN)
     add_definitions(-fPIC)
-endif((CMAKE_COMPILER_IS_GNUCXX OR IS_ICPC) AND NOT MINGW)
+endif((CMAKE_COMPILER_IS_GNUCXX OR IS_ICPC) AND NOT MINGW AND NOT CYGWIN)
 
 option(OMPL_SKIP_RPATH "Don't set RPATH to the OMPL library" OFF)
 if(NOT OMPL_SKIP_RPATH)
--- origsrc/ompl-1.3.2-Source/src/ompl/CMakeLists.txt	2017-07-21 12:39:54.000000000 +0900
+++ src/ompl-1.3.2-Source/src/ompl/CMakeLists.txt	2017-11-29 20:06:53.164604400 +0900
@@ -81,7 +81,8 @@ endif(MSVC)
 
 # install the library
 install(TARGETS ompl
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION bin
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     COMPONENT ompl)
 if(NOT MSVC)
     add_custom_command(TARGET ompl POST_BUILD
--- origsrc/ompl-1.3.2-Source/tests/CMakeLists.txt	2017-10-21 00:39:46.000000000 +0900
+++ src/ompl-1.3.2-Source/tests/CMakeLists.txt	2017-11-29 20:43:26.160000000 +0900
@@ -24,7 +24,7 @@ if (OMPL_BUILD_TESTS)
     add_ompl_test(test_gridb datastructures/gridb.cpp)
     add_ompl_test(test_nearestneighbors datastructures/nearestneighbors.cpp)
     if(FLANN_FOUND)
-        target_link_libraries(test_nearestneighbors ${FLANN_LIBRARIES})
+        target_link_libraries(test_nearestneighbors ${FLANN_LIBRARIES} lz4)
     endif()
     add_ompl_test(test_pdf datastructures/pdf.cpp)
 
