--- origsrc/ompl-0.13.0-Source/src/ompl/CMakeLists.txt	2013-12-10 08:32:26.316419600 +0900
+++ src/ompl-0.13.0-Source/src/ompl/CMakeLists.txt	2013-12-10 08:33:36.732435200 +0900
@@ -68,7 +68,7 @@ else(MSVC)
 endif(MSVC)
 
 # install the library
-install(TARGETS ompl DESTINATION ${OMPL_LIB_INSTALL_DIR})
+install(TARGETS ompl RUNTIME DESTINATION bin ARCHIVE DESTINATION lib${LIB_SUFFIX})
 if(NOT MSVC)
     get_target_property(OMPL_LIBRARY_LOC ompl LOCATION)
     add_custom_command(TARGET ompl POST_BUILD
--- origsrc/ompl-0.13.0-Source/src/ompl/util/Console.h	2013-07-10 09:54:58.000000000 +0900
+++ src/ompl-0.13.0-Source/src/ompl/util/Console.h	2013-12-10 08:38:16.352401400 +0900
@@ -37,6 +37,7 @@
 #ifndef OMPL_UTIL_CONSOLE_
 #define OMPL_UTIL_CONSOLE_
 
+#include <cstdio>
 #include <string>
 
 /** \file Console.h
