diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x ompl.pc -x '*.html' -x '*.md' -x '*.sh' -x config.h origsrc/ompl-1.1.0-Source/CMakeModules/CompilerSettings.cmake src/ompl-1.1.0-Source/CMakeModules/CompilerSettings.cmake
--- origsrc/ompl-1.1.0-Source/CMakeModules/CompilerSettings.cmake	2015-10-29 08:25:56.000000000 +0900
+++ src/ompl-1.1.0-Source/CMakeModules/CompilerSettings.cmake	2016-03-05 19:09:33.768050300 +0900
@@ -43,9 +43,9 @@ if(IS_XLC)
     set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -q64")
 endif(IS_XLC)
 
-if((CMAKE_COMPILER_IS_GNUCXX OR IS_ICPC) AND NOT MINGW)
+if((CMAKE_COMPILER_IS_GNUCXX OR IS_ICPC) AND NOT MINGW AND NOT CYGWIN)
     add_definitions(-fPIC)
-endif((CMAKE_COMPILER_IS_GNUCXX OR IS_ICPC) AND NOT MINGW)
+endif((CMAKE_COMPILER_IS_GNUCXX OR IS_ICPC) AND NOT MINGW AND NOT CYGWIN)
 
 # Set rpath http://www.paraview.org/Wiki/CMake_RPATH_handling
 set(CMAKE_SKIP_BUILD_RPATH FALSE)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x ompl.pc -x '*.html' -x '*.md' -x '*.sh' -x config.h origsrc/ompl-1.1.0-Source/src/ompl/CMakeLists.txt src/ompl-1.1.0-Source/src/ompl/CMakeLists.txt
--- origsrc/ompl-1.1.0-Source/src/ompl/CMakeLists.txt	2015-08-11 23:23:31.000000000 +0900
+++ src/ompl-1.1.0-Source/src/ompl/CMakeLists.txt	2016-03-05 19:09:33.783675300 +0900
@@ -79,7 +79,8 @@ endif(MSVC)
 
 # install the library
 install(TARGETS ompl
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION bin
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     COMPONENT ompl)
 if(NOT MSVC)
     add_custom_command(TARGET ompl POST_BUILD
