--- origsrc/halloc/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/halloc/Makefile.am	2015-02-15 20:07:00.813465100 +0900
@@ -0,0 +1,9 @@
+
+include_HEADERS = halloc.h
+
+lib_LTLIBRARIES = libhalloc.la
+libhalloc_la_SOURCES = src/align.h src/halloc.c src/hlist.h src/macros.h
+libhalloc_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = halloc.pc
--- origsrc/halloc/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/halloc/configure.ac	2015-02-15 20:07:08.831794900 +0900
@@ -0,0 +1,28 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([halloc], [1.2.2])
+AC_CONFIG_SRCDIR([halloc.h])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([malloc.h stddef.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([memset])
+
+AC_CONFIG_FILES([Makefile halloc.pc])
+AC_OUTPUT
--- origsrc/halloc/halloc.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/halloc/halloc.pc.in	2015-02-15 20:07:31.281412400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: halloc
+Description: Hierarchical Allocator for C
+Version: @VERSION@
+Libs: -L${libdir} -lhalloc
+Cflags: -I${includedir}
