--- origsrc/mgcal-0.1.1/Makefile.am	2015-09-08 10:20:23.000000000 +0900
+++ src/mgcal-0.1.1/Makefile.am	2025-08-15 19:46:39.613065200 +0900
@@ -5,20 +5,6 @@
 
 SUBDIRS = include src examples
 
-libmgcaldocdir = ${prefix}/doc/libmgcal
-libmgcaldoc_DATA = \
-	README\
-	COPYING\
-	AUTHORS\
-	ChangeLog\
-	INSTALL\
-	NEWS\
-	TODO
-
-EXTRA_DIST = $(libmgcaldoc_DATA)
-
-
-
 # Copy all the spec files. Of cource, only one is actually used.
 dist-hook:
 	for specfile in *.spec; do \
@@ -27,3 +13,5 @@ dist-hook:
 		fi \
 	done
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/mgcal-0.1.1/examples/Makefile.am	2015-09-08 10:20:23.000000000 +0900
+++ src/mgcal-0.1.1/examples/Makefile.am	2025-08-15 19:46:39.613065200 +0900
@@ -19,7 +19,7 @@ noinst_PROGRAMS=mgcal_examples
 mgcal_examples_SOURCES= example_dipole.c example_prism.c example_kernel.c examples.c
 
 # Linker options for a.out
-mgcal_examples_LDFLAGS = $(top_srcdir)/src/libmgcal.la
+mgcal_examples_LDFLAGS = $(top_builddir)/src/libmgcal.la
 
 # Compiler options for a.out
 mgcal_examples_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/examples
--- origsrc/mgcal-0.1.1/src/Makefile.am	2015-09-08 10:20:23.000000000 +0900
+++ src/mgcal-0.1.1/src/Makefile.am	2025-08-15 19:46:39.613065200 +0900
@@ -27,4 +27,4 @@ libmgcal_la_SOURCES = \
 # to be searched for headers included in the source code.
 libmgcal_la_LIBADD = -L$(top_srcdir)/src
 libmgcal_la_CPPFLAGS = -I$(top_srcdir)/include
-libmgcal_la_LDFLAGS = -version-info $(MGCAL_LT_VERSION)
+libmgcal_la_LDFLAGS = -no-undefined -version-info $(MGCAL_LT_VERSION)
