--- origsrc/SuperLU_4.0/Makefile.am	2010-08-29 19:12:38.954665700 +0900
+++ src/SuperLU_4.0/Makefile.am	2010-08-29 19:56:24.985915700 +0900
@@ -1 +1 @@
-SUBDIRS = SRC
+SUBDIRS = SRC TESTING
--- origsrc/SuperLU_4.0/SRC/Makefile.am	2010-08-29 19:12:38.954665700 +0900
+++ src/SuperLU_4.0/SRC/Makefile.am	2010-08-30 00:01:32.053128400 +0900
@@ -62,7 +62,11 @@ libsuperlu_la_SOURCES = \
 libsuperlu_la_LIBADD = $(BLAS_LIBS)
 libsuperlu_la_LDFLAGS = -no-undefined -version-info 3:0:0
 
-
+superlu_includedir = $(includedir)/superlu
+superlu_include_HEADERS = \
+colamd.h         slu_Cnames.h    slu_scomplex.h  slu_zdefs.h \
+html_mainpage.h  slu_dcomplex.h  slu_sdefs.h     supermatrix.h \
+slu_cdefs.h      slu_ddefs.h     slu_util.h
 
 ##################################
 # Do not optimize these routines #
--- origsrc/SuperLU_4.0/TESTING/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/SuperLU_4.0/TESTING/Makefile.am	2010-08-29 21:58:36.149693900 +0900
@@ -0,0 +1,50 @@
+TESTS = stest.csh dtest.csh ctest.csh ztest.csh
+check_PROGRAMS = stest dtest ctest ztest
+
+INCLUDES = -I$(top_srcdir)/SRC
+
+stest_SOURCES = sp_ienv.c \
+	sdrive.c sp_sconvert.c sgst01.c sgst02.c sgst04.c sgst07.c \
+	MATGEN/lsamen.c \
+	MATGEN/slatm1.c MATGEN/slaran.c MATGEN/slarnd.c \
+	MATGEN/slaruv.c MATGEN/slabad.c MATGEN/slarnv.c \
+	MATGEN/slatb4.c MATGEN/slaset.c MATGEN/slartg.c \
+	MATGEN/slatms.c MATGEN/slatme.c MATGEN/slatmr.c \
+	MATGEN/slagge.c MATGEN/slagsy.c MATGEN/slarge.c \
+	MATGEN/slaror.c MATGEN/slarot.c MATGEN/slatm2.c MATGEN/slatm3.c \
+	MATGEN/r_lg10.c MATGEN/r_sign.c MATGEN/pow_dd.c
+stest_LDADD = $(top_builddir)/SRC/libsuperlu.la $(BLAS_LIBS)
+
+dtest_SOURCES = sp_ienv.c \
+	ddrive.c sp_dconvert.c dgst01.c dgst02.c dgst04.c dgst07.c \
+	MATGEN/lsamen.c \
+	MATGEN/dlatm1.c MATGEN/dlaran.c MATGEN/dlarnd.c \
+	MATGEN/dlaruv.c MATGEN/dlabad.c MATGEN/dlarnv.c \
+	MATGEN/dlatb4.c MATGEN/dlaset.c MATGEN/dlartg.c \
+	MATGEN/dlatms.c MATGEN/dlatme.c MATGEN/dlatmr.c \
+	MATGEN/dlagge.c MATGEN/dlagsy.c MATGEN/dlarge.c \
+	MATGEN/dlaror.c MATGEN/dlarot.c MATGEN/dlatm2.c MATGEN/dlatm3.c \
+	MATGEN/d_lg10.c MATGEN/d_sign.c MATGEN/pow_dd.c
+dtest_LDADD = $(top_builddir)/SRC/libsuperlu.la $(BLAS_LIBS)
+
+ctest_SOURCES = sp_ienv.c \
+	cdrive.c sp_cconvert.c cgst01.c cgst02.c cgst04.c cgst07.c \
+	MATGEN/lsamen.c \
+	MATGEN/clatb4.c MATGEN/claset.c MATGEN/clartg.c \
+	MATGEN/clarnv.c MATGEN/clacgv.c MATGEN/csymv.c \
+	MATGEN/clatms.c MATGEN/clatme.c MATGEN/clatmr.c \
+	MATGEN/clagge.c MATGEN/clagsy.c MATGEN/clarge.c \
+	MATGEN/claror.c MATGEN/clarot.c MATGEN/clatm2.c MATGEN/clatm3.c \
+	MATGEN/claghe.c MATGEN/clarnd.c MATGEN/cdotc.c
+ctest_LDADD = $(top_builddir)/SRC/libsuperlu.la $(BLAS_LIBS)
+
+ztest_SOURCES = sp_ienv.c \
+	zdrive.c sp_zconvert.c zgst01.c zgst02.c zgst04.c zgst07.c \
+	MATGEN/lsamen.c \
+	MATGEN/zlatb4.c MATGEN/zlaset.c MATGEN/zlartg.c \
+	MATGEN/zlarnv.c MATGEN/zlacgv.c MATGEN/zsymv.c \
+	MATGEN/zlatms.c MATGEN/zlatme.c MATGEN/zlatmr.c \
+	MATGEN/zlagge.c MATGEN/zlagsy.c MATGEN/zlarge.c \
+	MATGEN/zlaror.c MATGEN/zlarot.c MATGEN/zlatm2.c MATGEN/zlatm3.c \
+	MATGEN/zlaghe.c MATGEN/zlarnd.c MATGEN/zdotc.c
+ztest_LDADD = $(top_builddir)/SRC/libsuperlu.la $(BLAS_LIBS)
--- origsrc/SuperLU_4.0/configure.ac	2010-08-29 19:12:38.939040700 +0900
+++ src/SuperLU_4.0/configure.ac	2010-08-29 21:10:12.267165700 +0900
@@ -4,11 +4,12 @@ AC_INIT(superlu, 3.0, xsli@lbl.gov)
 AM_INIT_AUTOMAKE([foreign])
 AC_CONFIG_HEADER([config.h])
 AC_PROG_INSTALL
-AC_PROG_LIBTOOL
-sinclude(acx_blas.m4)
-ACX_BLAS
-if test x"$BLAS_LIBS" = x; then
-   AC_MSG_ERROR([Cannot find blas libraries])
-fi
-AC_CONFIG_FILES([Makefile SRC/Makefile])
+LT_INIT(win32-dll)
+#sinclude(acx_blas.m4)
+#ACX_BLAS
+#if test x"$BLAS_LIBS" = x; then
+#   AC_MSG_ERROR([Cannot find blas libraries])
+#fi
+PKG_CHECK_MODULES(BLAS, lapack)
+AC_CONFIG_FILES([Makefile SRC/Makefile TESTING/Makefile])
 AC_OUTPUT
