diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x make.inc origsrc/SuperLU_5.1.1/CMakeLists.txt src/SuperLU_5.1.1/CMakeLists.txt
--- origsrc/SuperLU_5.1.1/CMakeLists.txt	2016-01-23 00:17:32.000000000 +0900
+++ src/SuperLU_5.1.1/CMakeLists.txt	2016-03-09 22:16:11.152119400 +0900
@@ -113,3 +113,9 @@ string (REPLACE ";" " " BLAS_LIB_STR "${
 # configure_file(${CMAKE_SOURCE_DIR}/make.inc.in ${CMAKE_BINARY_DIR}/make.inc)
 configure_file(${SuperLU_SOURCE_DIR}/make.inc.in ${SuperLU_SOURCE_DIR}/make.inc)
  
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/${PROJECT_NAME}.pc.in ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x make.inc origsrc/SuperLU_5.1.1/EXAMPLE/CMakeLists.txt src/SuperLU_5.1.1/EXAMPLE/CMakeLists.txt
--- origsrc/SuperLU_5.1.1/EXAMPLE/CMakeLists.txt	2016-01-22 13:34:21.000000000 +0900
+++ src/SuperLU_5.1.1/EXAMPLE/CMakeLists.txt	2016-03-09 22:09:23.989359300 +0900
@@ -3,5 +3,5 @@ if(enable_single)
   set(SLINEXM slinsol.c)
 
   add_executable(slinsol ${SLINEXM})
-  target_link_libraries(slinsol superlu_${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_BugFix} ${BLAS_LIB} m)
+  target_link_libraries(slinsol superlu ${BLAS_LIB} m)
 endif()
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x make.inc origsrc/SuperLU_5.1.1/SRC/CMakeLists.txt src/SuperLU_5.1.1/SRC/CMakeLists.txt
--- origsrc/SuperLU_5.1.1/SRC/CMakeLists.txt	2016-01-22 13:34:21.000000000 +0900
+++ src/SuperLU_5.1.1/SRC/CMakeLists.txt	2016-03-09 22:14:52.938921700 +0900
@@ -226,10 +226,12 @@ if(enable_complex16)
   )
 endif()
 
-add_library(superlu_${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_BugFix}  ${sources} ${HEADERS})
-
-install(TARGETS superlu_${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_BugFix}
-  DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
+add_library(superlu SHARED ${sources} ${HEADERS})
+target_link_libraries(superlu ${BLAS_blas_LIBRARY})
+set_target_properties(superlu PROPERTIES SOVERSION ${VERSION_MAJOR})
+install(TARGETS superlu
+  RUNTIME DESTINATION bin
+  ARCHIVE DESTINATION lib
 )
 
 install(FILES ${headers}
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x make.inc origsrc/SuperLU_5.1.1/TESTING/CMakeLists.txt src/SuperLU_5.1.1/TESTING/CMakeLists.txt
--- origsrc/SuperLU_5.1.1/TESTING/CMakeLists.txt	2016-01-22 13:34:21.000000000 +0900
+++ src/SuperLU_5.1.1/TESTING/CMakeLists.txt	2016-03-09 22:08:42.886167300 +0900
@@ -60,7 +60,7 @@ if(enable_single)
   set(SLINTST sdrive.c sp_sconvert.c sgst01.c sgst02.c sgst04.c sgst07.c)
 
   add_executable(s_test ${ALINTST} ${SLINTST})
-  target_link_libraries(s_test matgen superlu_${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_BugFix} ${BLAS_LIB} m)
+  target_link_libraries(s_test matgen superlu ${BLAS_LIB} m)
 
   add_superlu_test(s_test.out g20.rua s_test)
 
@@ -71,7 +71,7 @@ if(enable_double)
   set(DLINTST ddrive.c sp_dconvert.c dgst01.c dgst02.c dgst04.c dgst07.c)
 
   add_executable(d_test ${ALINTST} ${DLINTST})
-  target_link_libraries(d_test matgen superlu_${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_BugFix}  ${BLAS_LIB} m)
+  target_link_libraries(d_test matgen superlu ${BLAS_LIB} m)
 
   add_superlu_test(d_test.out g20.rua d_test)
 endif()
@@ -80,7 +80,7 @@ if(enable_complex)
   set(CLINTST cdrive.c sp_cconvert.c cgst01.c cgst02.c cgst04.c cgst07.c)
 
   add_executable(c_test ${ALINTST} ${CLINTST})
-  target_link_libraries(c_test matgen superlu_${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_BugFix} ${BLAS_LIB} m)
+  target_link_libraries(c_test matgen superlu ${BLAS_LIB} m)
 
   add_superlu_test(c_test.out cg20.cua c_test)
 endif()
@@ -90,7 +90,7 @@ if(enable_complex16)
   set(ZLINTST zdrive.c sp_zconvert.c zgst01.c zgst02.c zgst04.c zgst07.c)
 
   add_executable(z_test ${ALINTST} ${ZLINTST})
-  target_link_libraries(z_test matgen superlu_${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_BugFix} ${BLAS_LIB} m)
+  target_link_libraries(z_test matgen superlu ${BLAS_LIB} m)
 
   add_superlu_test(z_test.out cg20.cua z_test)
 endif()
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x make.inc origsrc/SuperLU_5.1.1/superlu.pc.in src/SuperLU_5.1.1/superlu.pc.in
--- origsrc/SuperLU_5.1.1/superlu.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/SuperLU_5.1.1/superlu.pc.in	2016-03-09 22:16:20.123264600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: superlu
+Description: Sparse LU factorization library
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lsuperlu
+Cflags: -I${includedir}
