--- origsrc/nng-1.11/CMakeLists.txt	2025-06-02 23:33:39.000000000 +0900
+++ src/nng-1.11/CMakeLists.txt	2026-01-27 18:29:17.026502400 +0900
@@ -268,6 +268,11 @@ elseif (CMAKE_SYSTEM_NAME MATCHES "QNX")
     add_definitions(-DNNG_PLATFORM_QNX)
     set(NNG_PLATFORM_POSIX ON)
 
+elseif (CMAKE_SYSTEM_NAME MATCHES "CYGWIN")
+    add_definitions (-DNNG_PLATFORM_POSIX)
+    add_definitions (-DNNG_PLATFORM_LINUX)
+    set(NNG_PLATFORM_POSIX ON)
+
 else ()
     message(AUTHOR_WARNING "WARNING: This platform may not be supported: ${CMAKE_SYSTEM_NAME}")
     message(AUTHOR_WARNING "${ISSUE_REPORT_MSG}")
@@ -310,3 +315,9 @@ set(CPACK_PACKAGE_FILE_NAME "${PROJECT_N
 
 add_custom_target(dist COMMAND ${CMAKE_MAKE_PROGRAM} package_source)
 include(CPack)
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/nng.pc.in ${CMAKE_BINARY_DIR}/nng.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/nng.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/nng-1.11/nng.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/nng-1.11/nng.pc.in	2026-01-27 18:52:59.199351900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: nng
+Description: nanomsg-next-generation
+Version: @NNG_ABI_VERSION@
+Libs: -L${libdir} -lnng
+Cflags: -I${includedir}
