--- origsrc/libfortuna/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libfortuna/Makefile.am	2013-08-03 20:51:13.006871100 +0900
@@ -0,0 +1 @@
+SUBDIRS = src
--- origsrc/libfortuna/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/libfortuna/configure.ac	2013-08-03 20:55:32.520543000 +0900
@@ -0,0 +1,28 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([libfortuna], [0+git20130716])
+AC_CONFIG_SRCDIR([src/blf.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_INSTALL
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([fcntl.h stdlib.h string.h sys/param.h sys/time.h unistd.h malloc_np.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([gettimeofday memmove memset strcasecmp strchr])
+
+AC_CONFIG_FILES([Makefile src/Makefile])
+AC_OUTPUT
--- origsrc/libfortuna/src/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libfortuna/src/Makefile.am	2013-08-03 20:52:33.554722700 +0900
@@ -0,0 +1,12 @@
+
+
+fortuna_includedir = $(includedir)/fortuna
+fortuna_include_HEADERS = blf.h c.h fortuna.h internal.h md5.h px.h rijndael.h sha1.h sha2.h
+
+
+lib_LTLIBRARIES = libfortuna.la
+libfortuna_la_CFLAGS = -Wall
+libfortuna_la_LDFLAGS = -no-undefined
+libfortuna_la_SOURCES = blf.c fortuna.c internal.c md5.c px.c random.c rijndael.c sha1.c sha2.c
+
+
--- origsrc/libfortuna/src/internal.h	2013-08-03 20:42:12.000000000 +0900
+++ src/libfortuna/src/internal.h	2013-08-03 20:57:40.568394600 +0900
@@ -33,7 +33,9 @@
 #define __INTERNAL_H
 
 #include <stdlib.h>
+#if defined(HAVE_MALLOC_NP_H)
 #include <malloc_np.h>
+#endif
 #include <time.h>
 
 #include "px.h"
--- origsrc/libfortuna/src/px.h	2013-08-03 20:42:12.000000000 +0900
+++ src/libfortuna/src/px.h	2013-08-03 20:57:13.739293000 +0900
@@ -35,7 +35,9 @@
 #include <stdlib.h>
 #include <sys/types.h>
 #include <sys/param.h>
+#if defined(HAVE_MALLOC_NP_H)
 #include <malloc_np.h>
+#endif
 #include "c.h"
 
 /* keep debug messages? */
