--- origsrc/zim-tools-3.6.0/meson.build	2025-04-18 05:32:19.000000000 +0900
+++ src/zim-tools-3.6.0/meson.build	2026-02-10 14:40:44.419109800 +0900
@@ -26,6 +26,7 @@ find_library_in_compiler = meson.version
 rt_dep = dependency('rt', required:false)
 docopt_dep = dependency('docopt', static:static_linkage)
 icu_dep = dependency('icu-i18n', static:static_linkage)
+icu_uc_dep = dependency('icu-uc', static:static_linkage)
 
 with_writer = host_machine.system() != 'windows'
 
--- origsrc/zim-tools-3.6.0/src/zimcheck/meson.build	2025-04-18 05:32:19.000000000 +0900
+++ src/zim-tools-3.6.0/src/zimcheck/meson.build	2026-02-10 14:40:45.890676800 +0900
@@ -8,7 +8,7 @@ endif
 
 inc = include_directories(extra_include)
 
-zimcheck_deps = [libzim_dep, icu_dep, docopt_dep]
+zimcheck_deps = [libzim_dep, icu_dep, icu_uc_dep, docopt_dep]
 
 # C++ std::thread is implemented using pthread on Linux by GCC, and on FreeBSD
 # for both GCC and LLVM.
--- origsrc/zim-tools-3.6.0/src/zimwriterfs/meson.build	2025-04-18 05:32:19.000000000 +0900
+++ src/zim-tools-3.6.0/src/zimwriterfs/meson.build	2026-02-10 14:40:47.407264000 +0900
@@ -7,7 +7,7 @@ sources = [
   'zimcreatorfs.cpp'
 ]
 
-deps = [thread_dep, libzim_dep, zlib_dep, gumbo_dep, magic_dep, icu_dep]
+deps = [thread_dep, libzim_dep, zlib_dep, gumbo_dep, magic_dep, icu_dep, icu_uc_dep]
 
 zimwriterfs = executable('zimwriterfs',
                          sources,
--- origsrc/zim-tools-3.6.0/test/meson.build	2025-04-18 05:32:19.000000000 +0900
+++ src/zim-tools-3.6.0/test/meson.build	2026-02-10 14:40:50.308631700 +0900
@@ -1,7 +1,7 @@
 gtest_dep = dependency('gtest', main:true, fallback:['gtest', 'gtest_main_dep'], required:false)
 
 
-test_deps = [gtest_dep, libzim_dep, icu_dep, docopt_dep]
+test_deps = [gtest_dep, libzim_dep, icu_dep, icu_uc_dep, docopt_dep]
 tests = [
     'metadata-test',
     'zimcheck-test'
