--- origsrc/zim-tools-3.2.0/meson.build	2023-04-28 18:36:51.000000000 +0900
+++ src/zim-tools-3.2.0/meson.build	2023-09-04 22:02:52.608462600 +0900
@@ -1,7 +1,6 @@
 project('zim-tools', ['c', 'cpp'],
   version : '3.2.0',
-  license : 'GPLv3+',
-  default_options : ['c_std=c11', 'cpp_std=c++11', 'werror=true'])
+  license : 'GPLv3+')
 
 add_global_arguments(['-Werror', '-Wall'], language:'cpp')
 
