--- origsrc/yajl/example/CMakeLists.txt	2012-02-19 01:42:20.000000000 +0900
+++ src/yajl/example/CMakeLists.txt	2012-02-19 01:48:02.886164000 +0900
@@ -20,4 +20,4 @@ LINK_DIRECTORIES(${CMAKE_CURRENT_BINARY_
 
 ADD_EXECUTABLE(parse_config ${SRCS})
 
-TARGET_LINK_LIBRARIES(parse_config yajl_s)
+TARGET_LINK_LIBRARIES(parse_config yajl)
--- origsrc/yajl/perf/CMakeLists.txt	2012-02-19 01:42:20.000000000 +0900
+++ src/yajl/perf/CMakeLists.txt	2012-02-19 01:49:02.029309600 +0900
@@ -20,4 +20,4 @@ LINK_DIRECTORIES(${CMAKE_CURRENT_BINARY_
 
 ADD_EXECUTABLE(perftest ${SRCS})
 
-TARGET_LINK_LIBRARIES(perftest yajl_s)
+TARGET_LINK_LIBRARIES(perftest yajl)
--- origsrc/yajl/reformatter/CMakeLists.txt	2012-02-19 01:42:20.000000000 +0900
+++ src/yajl/reformatter/CMakeLists.txt	2012-02-19 01:48:27.826085600 +0900
@@ -26,7 +26,7 @@ LINK_DIRECTORIES(${CMAKE_CURRENT_BINARY_
 
 ADD_EXECUTABLE(json_reformat ${SRCS})
 
-TARGET_LINK_LIBRARIES(json_reformat yajl_s)
+TARGET_LINK_LIBRARIES(json_reformat yajl)
 
 # copy the binary into the output directory
 GET_TARGET_PROPERTY(binPath json_reformat LOCATION)
--- origsrc/yajl/src/CMakeLists.txt	2012-02-19 01:42:20.000000000 +0900
+++ src/yajl/src/CMakeLists.txt	2012-02-19 02:01:02.178037300 +0900
@@ -35,7 +35,7 @@ SET (shareDir ${CMAKE_CURRENT_BINARY_DIR
 # set the output path for libraries
 SET(LIBRARY_OUTPUT_PATH ${libDir})
 
-ADD_LIBRARY(yajl_s STATIC ${SRCS} ${HDRS} ${PUB_HDRS})
+#ADD_LIBRARY(yajl_s STATIC ${SRCS} ${HDRS} ${PUB_HDRS})
 
 ADD_LIBRARY(yajl SHARED ${SRCS} ${HDRS} ${PUB_HDRS})
 
@@ -78,9 +78,9 @@ INCLUDE_DIRECTORIES(${incDir}/..)
 IF(NOT WIN32)
   # at build time you may specify the cmake variable LIB_SUFFIX to handle
   # 64-bit systems which use 'lib64'
-  INSTALL(TARGETS yajl LIBRARY DESTINATION lib${LIB_SUFFIX})
-  INSTALL(TARGETS yajl_s ARCHIVE DESTINATION lib${LIB_SUFFIX})
+  INSTALL(TARGETS yajl RUNTIME DESTINATION bin ARCHIVE DESTINATION lib${LIB_SUFFIX})
+#  INSTALL(TARGETS yajl_s ARCHIVE DESTINATION lib${LIB_SUFFIX})
   INSTALL(FILES ${PUB_HDRS} DESTINATION include/yajl)
   INSTALL(FILES ${incDir}/yajl_version.h DESTINATION include/yajl)
-  INSTALL(FILES ${shareDir}/yajl.pc DESTINATION share/pkgconfig)
+  INSTALL(FILES ${shareDir}/yajl.pc DESTINATION lib/pkgconfig)
 ENDIF()
--- origsrc/yajl/test/CMakeLists.txt	2012-02-19 01:42:20.000000000 +0900
+++ src/yajl/test/CMakeLists.txt	2012-02-19 01:46:40.549615200 +0900
@@ -20,4 +20,4 @@ LINK_DIRECTORIES(${CMAKE_CURRENT_BINARY_
 
 ADD_EXECUTABLE(yajl_test ${SRCS})
 
-TARGET_LINK_LIBRARIES(yajl_test yajl_s)
+TARGET_LINK_LIBRARIES(yajl_test yajl)
--- origsrc/yajl/verify/CMakeLists.txt	2012-02-19 01:42:20.000000000 +0900
+++ src/yajl/verify/CMakeLists.txt	2012-02-19 01:47:08.972524000 +0900
@@ -26,7 +26,7 @@ LINK_DIRECTORIES(${CMAKE_CURRENT_BINARY_
 
 ADD_EXECUTABLE(json_verify ${SRCS})
 
-TARGET_LINK_LIBRARIES(json_verify yajl_s)
+TARGET_LINK_LIBRARIES(json_verify yajl)
 
 # copy in the binary
 GET_TARGET_PROPERTY(binPath json_verify LOCATION)
