--- origsrc/JAGS-4.2.0/configure.ac	2016-02-09 22:14:58.000000000 +0900
+++ src/JAGS-4.2.0/configure.ac	2016-04-11 16:35:33.062263900 +0900
@@ -56,12 +56,19 @@ R_FUNC_ISFINITE
 fi
 
 case "${host_os}" in
-  mingw*)
+  mingw*|cygwin*)
     win=true ;;
   *)
     win=false ;;
 esac
 AM_CONDITIONAL(WINDOWS, test x$win = xtrue)
+case "${host_os}" in
+  cygwin*)
+    cyg=true ;;
+  *)
+    cyg=false ;;
+esac
+AM_CONDITIONAL(CYGWIN, test x$cyg = xtrue)
 
 dnl fortran stuff
 AC_F77_WRAPPERS
--- origsrc/JAGS-4.2.0/src/terminal/Makefile.am	2015-10-01 17:01:15.000000000 +0900
+++ src/JAGS-4.2.0/src/terminal/Makefile.am	2016-04-11 16:35:33.062263900 +0900
@@ -4,7 +4,7 @@ AM_YFLAGS = -d
 libexec_PROGRAMS = jags-terminal
 
 jags_terminal_SOURCES = parser.yy scanner.ll ReadData.cc 
-if WINDOWS
+if !CYGWIN
 ## 64-bit Windows build fails if we try pre-linking modules (although
 ## 32-bit Windows allows it).  
 jags_terminal_LDADD = $(LIBLTDL) ${top_builddir}/src/lib/libjags.la
@@ -30,7 +30,7 @@ noinst_HEADERS = ReadData.h
 ## The shell script is not required under Windows, so we do not
 ## build or install it. Instead, we install a batch file 
 
-if WINDOWS
+if !CYGWIN
 
 dist_bin_SCRIPTS = jags.bat
 
