--- origsrc/JAGS-3.3.0/configure.ac	2012-03-02 22:40:45.000000000 +0900
+++ src/JAGS-3.3.0/configure.ac	2012-10-21 15:55:07.689453100 +0900
@@ -37,13 +37,21 @@ R_FUNC_ISFINITE
 fi
 
 case "${host_os}" in
-  mingw*)
+  mingw*|cygwin*)
     win=true ;;
   *)
     win=false ;;
 esac
 AM_CONDITIONAL(WINDOWS, test x$win = xtrue)
 
+case "${host_os}" in
+  cygwin*)
+    cyg=true ;;
+  *)
+    cyg=false ;;
+esac
+AM_CONDITIONAL(CYGWIN, test x$cyg = xtrue)
+
 dnl fortran stuff
 AC_F77_WRAPPERS
 AC_F77_LIBRARY_LDFLAGS
--- origsrc/JAGS-3.3.0/src/terminal/Makefile.am	2011-07-18 22:33:22.000000000 +0900
+++ src/JAGS-3.3.0/src/terminal/Makefile.am	2012-10-21 15:55:07.689453100 +0900
@@ -30,7 +30,7 @@ noinst_HEADERS = ReadData.h
 ## The shell script is not required under Windows, so we do not
 ## build or install it. Instead, we install a batch file 
 
-if WINDOWS
+if !CYGWIN
 
 dist_bin_SCRIPTS = jags.bat
 
--- origsrc/JAGS-3.3.0/src/terminal/parser.cc	2012-10-02 16:42:20.000000000 +0900
+++ src/JAGS-3.3.0/src/terminal/parser.cc	2012-10-21 15:55:07.705078100 +0900
@@ -3586,7 +3586,7 @@ int main (int argc, char **argv)
     interactive = true;
   }
 
-#ifndef WIN32
+#if !defined(WIN32) && !defined(__CYGWIN__)
   /* 
      - Allows emulation of dynamic loading on platforms that do not
      support it by preloading modules. 
