--- origsrc/chash-1.0.2/libchash/libchash/Makefile.am	2015-07-21 20:28:22.000000000 +0900
+++ src/chash-1.0.2/libchash/libchash/Makefile.am	2016-04-11 09:37:58.717156500 +0900
@@ -1,6 +1,6 @@
 lib_LTLIBRARIES=libchash.la
 libchash_la_SOURCES=chash.c chash.h
-libchash_la_LDFLAGS=-version-info $(LIBCHASH_VERSION_INFO)
+libchash_la_LDFLAGS=-no-undefined -version-info $(LIBCHASH_VERSION_INFO)
 
 check_PROGRAMS=chash_test
 chash_test_SOURCES=chash_test.c
