--- origsrc/libfvalue-20160325/Makefile.am	2016-03-26 05:11:35.000000000 +0900
+++ src/libfvalue-20160325/Makefile.am	2016-04-05 10:53:13.194698100 +0900
@@ -3,17 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libcnotify \
-	libuna \
-	libfdatetime \
-	libfguid \
-	libfwnt \
 	libfvalue \
-	po \
 	manuals	\
 	msvscpp	\
 	tests
@@ -99,4 +89,3 @@ splint:
 	(cd $(srcdir)/libfvalue && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
-
--- origsrc/libfvalue-20160325/configure.ac	2016-03-26 05:11:35.000000000 +0900
+++ src/libfvalue-20160325/configure.ac	2016-04-05 10:54:01.000609800 +0900
@@ -148,15 +148,6 @@ dnl Have configure make the Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfwnt/Makefile])
 AC_CONFIG_FILES([libfvalue/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
@@ -198,4 +189,3 @@ Features:
    Verbose output:          $ac_cv_enable_verbose_output
    Debug output:            $ac_cv_enable_debug_output
 ]);
-
--- origsrc/libfvalue-20160325/libfvalue/Makefile.am	2016-03-26 05:11:35.000000000 +0900
+++ src/libfvalue-20160325/libfvalue/Makefile.am	2016-04-05 10:53:28.252011400 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -73,4 +74,3 @@ distclean: clean
 splint:
 	@echo "Running splint on libfvalue ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(libfvalue_la_SOURCES)
-
--- origsrc/libfvalue-20160325/tests/Makefile.am	2016-03-26 05:11:35.000000000 +0900
+++ src/libfvalue-20160325/tests/Makefile.am	2016-04-05 10:53:42.639096200 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -35,4 +36,3 @@ MAINTAINERCLEANFILES = \
 
 distclean: clean
 	/bin/rm -f Makefile
-
--- origsrc/libfvalue-20160325/tests/fvalue_test_value.c	2016-03-26 05:11:35.000000000 +0900
+++ src/libfvalue-20160325/tests/fvalue_test_value.c	2016-04-05 10:57:15.663648400 +0900
@@ -19,7 +19,7 @@
  * along with this software.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include <common.h>
+#include <memory.h>
 
 #if defined( HAVE_STDLIB_H ) || defined( WINAPI )
 #include <stdlib.h>
@@ -938,4 +938,3 @@ int main( int argc, char * const argv[]
 	}
 	return( EXIT_SUCCESS );
 }
-
