--- origsrc/epeg-0.9.2/src/lib/Makefile.am	2016-12-21 09:59:47.000000000 +0900
+++ src/epeg-0.9.2/src/lib/Makefile.am	2017-01-02 14:37:37.807299200 +0900
@@ -11,4 +11,4 @@ epeg_private.h
 
 libepeg_la_LIBADD       = @my_libs@
 libepeg_la_DEPENDENCIES = $(top_builddir)/config.h
-libepeg_la_LDFLAGS      = -version-info @version_info@
+libepeg_la_LDFLAGS      = -no-undefined -version-info @version_info@
