--- origsrc/epeg-0.9.1.042/src/lib/Makefile.am	2007-11-13 15:01:32.000000000 +0900
+++ src/epeg-0.9.1.042/src/lib/Makefile.am	2014-06-01 22:53:49.989062600 +0900
@@ -11,4 +11,4 @@ epeg_private.h
 
 libepeg_la_LIBADD       = @my_libs@
 libepeg_la_DEPENDENCIES = $(top_builddir)/config.h
-libepeg_la_LDFLAGS      = -version-info @version_info@
+libepeg_la_LDFLAGS      = -version-info @version_info@ -no-undefined
