--- origsrc/libmspub-0.0.0/src/lib/Makefile.am	2012-05-31 16:46:08.000000000 +0900
+++ src/libmspub-0.0.0/src/lib/Makefile.am	2012-06-08 22:37:31.395190200 +0900
@@ -15,7 +15,7 @@ AM_CXXFLAGS = $(LIBMSPUB_CXXFLAGS) $(ZLI
 
 libmspub_@MSPUB_MAJOR_VERSION@_@MSPUB_MINOR_VERSION@_la_LIBADD  = $(LIBMSPUB_LIBS) $(ZLIB_LIBS) @LIBMSPUB_WIN32_RESOURCE@
 libmspub_@MSPUB_MAJOR_VERSION@_@MSPUB_MINOR_VERSION@_la_DEPENDENCIES = @LIBMSPUB_WIN32_RESOURCE@
-libmspub_@MSPUB_MAJOR_VERSION@_@MSPUB_MINOR_VERSION@_la_LDFLAGS = $(version_info) -export-dynamic -no_undefined
+libmspub_@MSPUB_MAJOR_VERSION@_@MSPUB_MINOR_VERSION@_la_LDFLAGS = $(version_info) -export-dynamic -no-undefined
 libmspub_@MSPUB_MAJOR_VERSION@_@MSPUB_MINOR_VERSION@_la_SOURCES = \
 	MSPUBCollector.cpp \
 	MSPUBDocument.cpp \
