--- origsrc/stan-0.4.1/configure.in	2004-09-15 16:05:19.000000000 +0900
+++ src/stan-0.4.1/configure.in	2024-10-31 08:57:17.312028900 +0900
@@ -2,19 +2,14 @@
 #
 # GNU Configure
 #
-AC_INIT(src/stan.c)
-AM_INIT_AUTOMAKE(stan, 0.4.1)
+AC_INIT([stan], [0.4.1])
+AM_INIT_AUTOMAKE
 AM_CONFIG_HEADER(config.h)
 
 dnl Checks for programs.
 AC_PROG_CC
 AC_C_BIGENDIAN
 
-if test "x$GCC" = "xyes" 
-then
-   CFLAGS="-Wall -O3"
-fi
-
 dnl Checks for libraries.
 AC_CHECK_LIB(m,pow,,
    echo The math library libm.a is required for compilation ; 
