diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/airtsp-1.01.1/airtsp-config.cmake.in src/airtsp-1.01.1/airtsp-config.cmake.in
--- origsrc/airtsp-1.01.1/airtsp-config.cmake.in	2013-06-09 18:27:40.000000000 +0900
+++ src/airtsp-1.01.1/airtsp-config.cmake.in	2015-08-13 20:22:02.020047900 +0900
@@ -19,5 +19,5 @@ include ("@AIRTSP_CMAKE_DIR@/airtsp-libr
 
 # These are the AirTSP IMPORTED targets, created by
 # airtsp-library-depends.cmake
-set (AIRTSP_LIBRARIES airtsplib)
+set (AIRTSP_LIBRARIES airtsp)
 set (AIRTSP_EXECUTABLES airtsp)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/airtsp-1.01.1/config/project_config_embeddable.cmake src/airtsp-1.01.1/config/project_config_embeddable.cmake
--- origsrc/airtsp-1.01.1/config/project_config_embeddable.cmake	2013-06-09 18:27:40.000000000 +0900
+++ src/airtsp-1.01.1/config/project_config_embeddable.cmake	2015-08-13 20:30:17.988447000 +0900
@@ -162,7 +162,7 @@ macro (set_project_options _build_doc _e
     unset (DOC_NEWS_FILE)
   endif (NOT EXISTS ${DOC_NEWS_FILE})
   set (BASICDOC_FILES AUTHORS ${DOC_NEWS_FILE} README ${DOC_INSTALL_FILE})
-  set (BASICDOC_PATH "share/doc/${PACKAGE}-${PACKAGE_VERSION}")
+  set (BASICDOC_PATH "share/doc/${PACKAGE}")
 
 endmacro (set_project_options)
 
@@ -1427,7 +1427,7 @@ endmacro (module_set_name)
 macro (add_modules)
   set (_embedded_components ${ARGV})
   set (LIB_DEPENDENCY_EXPORT ${PROJECT_NAME}-library-depends)
-  set (LIB_DEPENDENCY_EXPORT_PATH "${INSTALL_DATA_DIR}/${PROJECT_NAME}/CMake")
+  set (LIB_DEPENDENCY_EXPORT_PATH "${INSTALL_LIB_DIR}/cmake/${PROJECT_NAME}")
   #
   foreach (_embedded_comp ${_embedded_components})
     #
@@ -1692,7 +1692,8 @@ macro (module_library_add_specific
   # Installation of the library
   install (TARGETS ${_lib_target}
     EXPORT ${LIB_DEPENDENCY_EXPORT}
-    LIBRARY DESTINATION "${INSTALL_LIB_DIR}" COMPONENT runtime)
+    RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
+    ARCHIVE DESTINATION "${INSTALL_LIB_DIR}")
 
   # Register, for reporting purpose, the list of libraries to be built
   # and installed for that module
@@ -1878,7 +1879,7 @@ endmacro (module_script_add)
 # can refer to it (for libraries, header files and binaries)
 macro (module_export_install)
   install (EXPORT ${LIB_DEPENDENCY_EXPORT}
-    DESTINATION "${INSTALL_DATA_DIR}/${PACKAGE}/CMake" COMPONENT devel)
+  DESTINATION "${INSTALL_LIB_DIR}/cmake/${PACKAGE}" COMPONENT devel)
 endmacro (module_export_install)
 
 
