--- origsrc/airtsp-1.01.0/config/project_config_embeddable.cmake	2013-06-09 18:27:40.000000000 +0900
+++ src/airtsp-1.01.0/config/project_config_embeddable.cmake	2014-06-15 17:22:21.176639700 +0900
@@ -1691,8 +1691,8 @@ macro (module_library_add_specific
   ##
   # Installation of the library
   install (TARGETS ${_lib_target}
-    EXPORT ${LIB_DEPENDENCY_EXPORT}
-    LIBRARY DESTINATION "${INSTALL_LIB_DIR}" COMPONENT runtime)
+    RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
+    ARCHIVE DESTINATION "${INSTALL_LIB_DIR}")
 
   # Register, for reporting purpose, the list of libraries to be built
   # and installed for that module
