--- origsrc/airtsp-1.01.16/airtsp-config.cmake.in	2025-12-25 08:05:51.000000000 +0900
+++ src/airtsp-1.01.16/airtsp-config.cmake.in	2025-12-29 21:43:51.554842200 +0900
@@ -21,5 +21,5 @@ include ("${CMAKE_CURRENT_LIST_DIR}/@PRO
 
 # These are the AirTSP IMPORTED targets, created by
 # airtsp-library-depends.cmake
-set (AIRTSP_LIBRARIES airtsplib)
+set (AIRTSP_LIBRARIES airtsp)
 set (AIRTSP_EXECUTABLES airtsp)
--- origsrc/airtsp-1.01.16/config/project_config_embeddable.cmake	2025-12-25 08:05:51.000000000 +0900
+++ src/airtsp-1.01.16/config/project_config_embeddable.cmake	2025-12-29 21:44:14.336595300 +0900
@@ -439,7 +439,7 @@ macro (get_git)
   message (STATUS "Requires Git without specifying any version")
 
   find_package (Git)
-  if (Git_FOUND)
+  if (Git_FOUND AND EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/.git)
     Git_WC_INFO (${CMAKE_CURRENT_SOURCE_DIR} PROJ)
     set (GIT_REVISION ${PROJ_WC_REVISION_HASH})
     message (STATUS "Current Git revision name: ${PROJ_WC_REVISION_NAME}")
