--- origsrc/rz-libdemangle/meson.build	2023-05-17 10:14:48.000000000 +0900
+++ src/rz-libdemangle/meson.build	2023-05-17 10:14:57.407271200 +0900
@@ -16,6 +16,9 @@ libdemangle_src = [
   'src' / 'pascal' / 'pascal.c',
   'src' / 'rust.c',
 ]
+libdemangle_hdr = [
+  'include/rz_libdemangle.h'
+]
 
 tests = [
   'borland',
@@ -52,6 +55,8 @@ endif
 
 libdemangle = library('demangle', libdemangle_src,
     c_args : common_c_args + libdemangle_c_args,
+    soversion: 0,
+    install: true,
     dependencies: [],
     implicit_include_directories: false,
     include_directories: include_directories(['include', 'src']))
@@ -62,6 +67,17 @@ libdemangle_dep = declare_dependency(
     include_directories: include_directories('include'),
 )
 
+install_headers(libdemangle_hdr)
+
+pkgconfig = import('pkgconfig')
+pkgconfig.generate(
+    description: 'Rizin Library to demangle symbols',
+    filebase: 'libdemangle',
+    libraries: libdemangle,
+    name: 'libdemangle',
+    version: '0'
+)
+
 if get_option('enable_tests') and not meson.is_subproject()
   test_dependencies = [libdemangle_dep]
   if get_option('b_sanitize').contains('address')
