--- origsrc/rz-libdemangle/meson.build	2024-10-30 09:03:34.000000000 +0900
+++ src/rz-libdemangle/meson.build	2024-10-30 09:03:35.485504700 +0900
@@ -19,6 +19,9 @@ libdemangle_src = [
   'src' / 'rust' / 'rust_legacy.c',
   'src' / 'rust' / 'rust_v0.c',
 ]
+libdemangle_hdr = [
+  'include/rz_libdemangle.h'
+]
 
 tests = [
   'borland',
@@ -57,6 +60,8 @@ endif
 
 libdemangle = library('demangle', libdemangle_src,
     c_args : common_c_args + libdemangle_c_args,
+    soversion: 0,
+    install: true,
     dependencies: [],
     implicit_include_directories: false,
     install: get_option('install_lib'),
@@ -81,6 +86,17 @@ if get_option('enable_cli')
   )
 endif
 
+install_headers(libdemangle_hdr)
+
+pkgconfig = import('pkgconfig')
+pkgconfig.generate(
+    description: 'Rizin Library to demangle symbols',
+    filebase: 'libdemangle',
+    libraries: libdemangle,
+    name: 'libdemangle',
+    version: '0'
+)
+
 if get_option('enable_tests') and not meson.is_subproject()
   test_dependencies = [libdemangle_dep]
   if get_option('b_sanitize').contains('address')
