--- origsrc/hello-2.12/Makefile.am	2022-01-31 20:29:14.000000000 +0900
+++ src/hello-2.12/Makefile.am	2022-04-20 17:35:01.203569100 +0900
@@ -37,7 +37,7 @@ SUBDIRS = po
 bin_PROGRAMS = hello
 hello_SOURCES = src/hello.c src/system.h
 
-hello_LDADD = $(LIBINTL) $(top_builddir)/lib/lib$(PACKAGE).a
+hello_LDADD = $(top_builddir)/lib/lib$(PACKAGE).a $(LIBINTL) $(LTLIBICONV)
 
 localedir = $(datadir)/locale
 DEFS = -DLOCALEDIR=\"$(localedir)\" @DEFS@
--- origsrc/hello-2.12/configure.ac	2021-03-08 06:42:45.000000000 +0900
+++ src/hello-2.12/configure.ac	2022-04-20 17:27:54.285456500 +0900
@@ -11,7 +11,7 @@ dnl WITHOUT ANY WARRANTY, to the extent
 dnl implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 
 AC_INIT([GNU Hello],
-	[m4_esyscmd([build-aux/git-version-gen .tarball-version])],
+	[2.12],
 	[bug-hello@gnu.org], [],
 	[https://www.gnu.org/software/hello/])
 
@@ -40,6 +40,8 @@ dnl keep everything together.
 gl_EARLY
 gl_INIT
 
+AM_ICONV
+
 dnl Are we building from git checked-out sources, or a tarball ?
 dnl This is used in "Makefile.am" to avoid re-generating the manpage
 dnl when building from tarballs.
