--- origsrc/hello-2.10/configure.ac	2014-10-05 19:19:54.000000000 +0900
+++ src/hello-2.10/configure.ac	2019-12-06 23:03:26.102013400 +0900
@@ -9,7 +9,7 @@ dnl WITHOUT ANY WARRANTY, to the extent
 dnl implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 
 AC_INIT([GNU Hello],
-	[m4_esyscmd([build-aux/git-version-gen .tarball-version])],
+	[2.10],
 	[bug-hello@gnu.org], [],
 	[http://www.gnu.org/software/hello/])
 
