--- origsrc/libproton-1.1.4/Makefile.am	2013-05-13 18:19:43.000000000 +0900
+++ src/libproton-1.1.4/Makefile.am	2015-02-27 01:59:43.129507000 +0900
@@ -2,3 +2,6 @@ ACLOCAL_AMFLAGS = -I m4
 
 SUBDIRS = src test examples
 DIST_SUBDIRS = src test examples
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = libproton.pc
--- origsrc/libproton-1.1.4/configure.ac	2013-05-13 18:19:43.000000000 +0900
+++ src/libproton-1.1.4/configure.ac	2015-02-27 01:59:28.460561700 +0900
@@ -37,5 +37,6 @@ AC_CHECK_HEADERS([ctime cstring cstdio c
 AC_CONFIG_FILES([Makefile
                  src/Makefile
                  examples/Makefile
-                 test/Makefile])
+                 test/Makefile
+                 libproton.pc])
 AC_OUTPUT
--- origsrc/libproton-1.1.4/examples/Makefile.am	2013-05-13 18:19:43.000000000 +0900
+++ src/libproton-1.1.4/examples/Makefile.am	2015-02-27 01:56:23.608022700 +0900
@@ -1,48 +1,48 @@
-TESTS = eg_func eg_vector eg_deque eg_string eg_getopt eg_tuple eg_set eg_unordered_set eg_map eg_unordered_map wc
-check_PROGRAMS = eg_func eg_vector eg_deque eg_string eg_getopt eg_tuple eg_set eg_unordered_set eg_map eg_unordered_map wc
+TESTS = eg_func eg_vector eg_deque eg_string eg_getopt eg_tuple eg_set eg_unordered_set wc
+check_PROGRAMS = eg_func eg_vector eg_deque eg_string eg_getopt eg_tuple eg_set eg_unordered_set  wc
 
 eg_func_SOURCES = functor.cpp
 eg_func_CXXFLAGS = $(BOOST_CPPFLAGS)
-eg_func_LDADD = $(top_srcdir)/src/libproton.la
+eg_func_LDADD = $(top_builddir)/src/libproton.la
 
 eg_string_SOURCES = string.cpp
 eg_string_CXXFLAGS = $(BOOST_CPPFLAGS)
-eg_string_LDADD = $(top_srcdir)/src/libproton.la
+eg_string_LDADD = $(top_builddir)/src/libproton.la
 
 eg_vector_SOURCES = vector.cpp
 eg_vector_CXXFLAGS = $(BOOST_CPPFLAGS)
-eg_vector_LDADD = $(top_srcdir)/src/libproton.la
+eg_vector_LDADD = $(top_builddir)/src/libproton.la
 
 eg_deque_SOURCES = deque.cpp
 eg_deque_CXXFLAGS = $(BOOST_CPPFLAGS)
-eg_deque_LDADD = $(top_srcdir)/src/libproton.la
+eg_deque_LDADD = $(top_builddir)/src/libproton.la
 
 eg_map_SOURCES = map.cpp
 eg_map_CXXFLAGS = $(BOOST_CPPFLAGS)
-eg_map_LDADD = $(top_srcdir)/src/libproton.la
+eg_map_LDADD = $(top_builddir)/src/libproton.la
 
 eg_set_SOURCES = set.cpp
 eg_set_CXXFLAGS = $(BOOST_CPPFLAGS)
-eg_set_LDADD = $(top_srcdir)/src/libproton.la
+eg_set_LDADD = $(top_builddir)/src/libproton.la
 
 eg_unordered_set_SOURCES = unordered_set.cpp
 eg_unordered_set_CXXFLAGS = $(BOOST_CPPFLAGS)
-eg_unordered_set_LDADD = $(top_srcdir)/src/libproton.la
+eg_unordered_set_LDADD = $(top_builddir)/src/libproton.la
 
 eg_unordered_map_SOURCES = unordered_map.cpp
 eg_unordered_map_CXXFLAGS = $(BOOST_CPPFLAGS)
-eg_unordered_map_LDADD = $(top_srcdir)/src/libproton.la
+eg_unordered_map_LDADD = $(top_builddir)/src/libproton.la
 
 eg_getopt_SOURCES = getopt.cpp
 eg_getopt_CXXFLAGS = $(BOOST_CPPFLAGS)
-eg_getopt_LDADD = $(top_srcdir)/src/libproton.la
+eg_getopt_LDADD = $(top_builddir)/src/libproton.la
 
 eg_tuple_SOURCES = tuple.cpp
 eg_tuple_CXXFLAGS = $(BOOST_CPPFLAGS)
-eg_tuple_LDADD = $(top_srcdir)/src/libproton.la
+eg_tuple_LDADD = $(top_builddir)/src/libproton.la
 
 wc_SOURCES = wc.cpp
 wc_CXXFLAGS = $(BOOST_CPPFLAGS)
-wc_LDADD = $(top_srcdir)/src/libproton.la
+wc_LDADD = $(top_builddir)/src/libproton.la
 
 INCLUDES = -I$(top_srcdir)/include
--- origsrc/libproton-1.1.4/libproton.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libproton-1.1.4/libproton.pc.in	2015-02-27 01:59:16.116811700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libproton
+Description: Library to provide Python-like interfaces for C++11
+Version: 1.1.4
+Libs: -L${libdir} -lproton
+Cflags: -I${includedir}
--- origsrc/libproton-1.1.4/test/Makefile.am	2013-05-13 18:19:43.000000000 +0900
+++ src/libproton-1.1.4/test/Makefile.am	2015-02-27 01:53:54.128738200 +0900
@@ -1,24 +1,24 @@
-TESTS = base_test pool_ut ref_ut stl_test own_test
-check_PROGRAMS = base_test pool_ut ref_ut stl_test own_test
+TESTS = base_test ref_ut stl_test
+check_PROGRAMS = base_test ref_ut stl_test
 
 base_test_SOURCES = base_test.cpp
 base_test_CXXFLAGS = $(BOOST_CPPFLAGS)
-base_test_LDADD = $(top_srcdir)/src/libproton.la
+base_test_LDADD = $(top_builddir)/src/libproton.la
 
 pool_ut_SOURCES = pool_ut.cpp pool_types.hpp
 pool_ut_CXXFLAGS = $(BOOST_CPPFLAGS)
-pool_ut_LDADD = $(top_srcdir)/src/libproton.la
+pool_ut_LDADD = $(top_builddir)/src/libproton.la
 
 ref_ut_SOURCES = ref_ut.cpp
 ref_ut_CXXFLAGS = $(BOOST_CPPFLAGS)
-ref_ut_LDADD = $(top_srcdir)/src/libproton.la
+ref_ut_LDADD = $(top_builddir)/src/libproton.la
 
 stl_test_SOURCES = test.cpp
 stl_test_CXXFLAGS = $(BOOST_CPPFLAGS)
-stl_test_LDADD = $(top_srcdir)/src/libproton.la
+stl_test_LDADD = $(top_builddir)/src/libproton.la
 
 own_test_SOURCES = test_.cpp
 own_test_CXXFLAGS = $(BOOST_CPPFLAGS)
-own_test_LDADD = $(top_srcdir)/src/libproton.la
+own_test_LDADD = $(top_builddir)/src/libproton.la
 
 INCLUDES = -I$(top_srcdir)/include
