--- origsrc/liboqs-0.7.2/CMakeLists.txt	2022-08-22 04:54:09.000000000 +0900
+++ src/liboqs-0.7.2/CMakeLists.txt	2022-11-24 10:05:03.188551500 +0900
@@ -209,3 +209,12 @@ if(NOT ${OQS_BUILD_ONLY_LIB})
             USES_TERMINAL)
     endif()
 endif()
+
+if (CYGWIN)
+  set(prefix ${CMAKE_INSTALL_PREFIX})
+  set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+  set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+  set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+  configure_file(${CMAKE_SOURCE_DIR}/liboqs.pc.in ${CMAKE_BINARY_DIR}/liboqs.pc @ONLY)
+  install(FILES ${CMAKE_BINARY_DIR}/liboqs.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+endif()
--- origsrc/liboqs-0.7.2/liboqs.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/liboqs-0.7.2/liboqs.pc.in	2022-11-24 10:05:03.191455400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: liboqs
+Description: C library for prototyping and experimenting with quantum-resistant cryptography
+Version: @OQS_VERSION_TEXT@
+Libs: -L${libdir} -loqs
+Cflags: -I${includedir}
--- origsrc/liboqs-0.7.2/tests/CMakeLists.txt	2022-08-22 04:54:09.000000000 +0900
+++ src/liboqs-0.7.2/tests/CMakeLists.txt	2022-11-24 10:05:03.195563700 +0900
@@ -118,7 +118,7 @@ add_definitions(-DOQS_COMPILE_OPTIONS="[
 # for DLL builds.
 add_custom_target(
     run_tests
-    COMMAND ${CMAKE_COMMAND} -E env OQS_BUILD_DIR=${CMAKE_BINARY_DIR} ${PYTHON3_EXEC} -m pytest --verbose --numprocesses=auto --ignore=scripts/copy_from_upstream/repos
+    COMMAND ${CMAKE_COMMAND} -E env OQS_BUILD_DIR=${CMAKE_BINARY_DIR} ${PYTHON3_EXEC} -m pytest --verbose --ignore=scripts/copy_from_upstream/repos
     WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
     DEPENDS oqs example_kem kat_kem test_kem example_sig kat_sig test_sig test_sig_mem test_kem_mem ${UNIX_TESTS}
     USES_TERMINAL)
