--- origsrc/airinv-airinv-1.00.5/airinv-config.cmake.in	2020-08-10 07:21:11.000000000 +0900
+++ src/airinv-airinv-1.00.5/airinv-config.cmake.in	2020-08-31 16:49:05.109210300 +0900
@@ -18,5 +18,5 @@ set (AIRINV_LIBRARY_DIRS "@AIRINV_LIB_DI
 include ("@AIRINV_CMAKE_DIR@/airinv-library-depends.cmake")
 
 # These are the AirInv IMPORTED targets, created by airinv-library-depends.cmake
-set (AIRINV_LIBRARIES airinvlib)
+set (AIRINV_LIBRARIES airinv)
 set (AIRINV_EXECUTABLES airinv airinv_parseInventory)
--- origsrc/airinv-airinv-1.00.5/config/project_config_embeddable.cmake	2020-08-10 07:21:11.000000000 +0900
+++ src/airinv-airinv-1.00.5/config/project_config_embeddable.cmake	2020-08-31 16:53:15.984685800 +0900
@@ -1635,7 +1635,11 @@ macro (add_modules)
   set (_embedded_components ${ARGV})
   set (LIB_DEPENDENCY_EXPORT ${PROJECT_NAME}-library-depends)
   set (PY_LIB_DEPENDENCY_EXPORT ${PROJECT_NAME}-python-library-depends)
-  set (LIB_DEPENDENCY_EXPORT_PATH "${INSTALL_DATA_DIR}/${PROJECT_NAME}/CMake")
+  if (CYGWIN)
+    set (LIB_DEPENDENCY_EXPORT_PATH "${INSTALL_LIB_DIR}/cmake/${PROJECT_NAME}")
+  else()
+    set (LIB_DEPENDENCY_EXPORT_PATH "${INSTALL_DATA_DIR}/${PROJECT_NAME}/CMake")
+  endif()
   #
   foreach (_embedded_comp ${_embedded_components})
     #
@@ -1949,7 +1953,8 @@ macro (module_library_add_specific
     # Install the library in the standard location
     install (TARGETS ${_lib_target}
       EXPORT ${LIB_DEPENDENCY_EXPORT}
-      LIBRARY DESTINATION "${INSTALL_LIB_DIR}" COMPONENT runtime)
+      RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
+      ARCHIVE DESTINATION "${INSTALL_LIB_DIR}" COMPONENT runtime)
   endif ("${_lib_prefix}" STREQUAL "py")
   
   # Register, for reporting purpose, the list of libraries to be built
@@ -2164,12 +2169,22 @@ endmacro (module_script_add)
 # Installation of the CMake import helper, so that third party projects
 # can refer to it (for libraries, header files and binaries)
 macro (module_export_install)
-  install (EXPORT ${LIB_DEPENDENCY_EXPORT}
-    DESTINATION "${INSTALL_DATA_DIR}/${PACKAGE}/CMake" COMPONENT devel)
+  if (CYGWIN)
+    install (EXPORT ${LIB_DEPENDENCY_EXPORT}
+      DESTINATION "${INSTALL_LIB_DIR}/cmake/${PACKAGE}" COMPONENT devel)
+  else()
+    install (EXPORT ${LIB_DEPENDENCY_EXPORT}
+      DESTINATION "${INSTALL_DATA_DIR}/${PACKAGE}/CMake" COMPONENT devel)
+  endif()
 
   if (NEED_PYTHON)
-	install (EXPORT ${PY_LIB_DEPENDENCY_EXPORT}
-      DESTINATION "${INSTALL_DATA_DIR}/${PACKAGE}/CMake" COMPONENT python-devel)
+    if (CYGWIN)
+      install (EXPORT ${PY_LIB_DEPENDENCY_EXPORT}
+        DESTINATION "${INSTALL_LIB_DIR}/cmake/${PACKAGE}" COMPONENT python-devel)
+    else()
+      install (EXPORT ${PY_LIB_DEPENDENCY_EXPORT}
+        DESTINATION "${INSTALL_DATA_DIR}/${PACKAGE}/CMake" COMPONENT python-devel)
+    endif()
   endif (NEED_PYTHON)
 endmacro (module_export_install)
 
