--- origsrc/libvshadow-20240504/Makefile.am	2024-05-04 15:16:57.000000000 +0900
+++ src/libvshadow-20240504/Makefile.am	2025-05-19 22:34:16.924540300 +0900
@@ -3,22 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfdatetime \
-	libfguid \
 	libvshadow \
 	vshadowtools \
 	pyvshadow \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libvshadow-20240504/configure.ac	2024-05-04 15:16:57.000000000 +0900
+++ src/libvshadow-20240504/configure.ac	2025-05-19 22:34:16.924540300 +0900
@@ -167,18 +167,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
 AC_CONFIG_FILES([libvshadow/Makefile])
 AC_CONFIG_FILES([pyvshadow/Makefile])
 AC_CONFIG_FILES([vshadowtools/Makefile])
--- origsrc/libvshadow-20240504/pyvshadow/Makefile.am	2024-05-04 15:16:57.000000000 +0900
+++ src/libvshadow-20240504/pyvshadow/Makefile.am	2025-05-19 22:34:27.657782000 +0900
@@ -49,7 +49,7 @@ pyvshadow_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pyvshadow_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyvshadow_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyvshadow_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
