--- origsrc/libvshadow-20120511/libbfio/Makefile.am	2012-04-15 21:49:42.000000000 +0900
+++ src/libvshadow-20120511/libbfio/Makefile.am	2012-06-24 18:19:39.877198500 +0900
@@ -2,6 +2,12 @@ if HAVE_LOCAL_LIBBFIO
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcerror \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libclocale \
+	-I$(top_srcdir)/libuna \
+	-I$(top_srcdir)/libcfile \
+	-I$(top_srcdir)/libcpath \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
 	@LIBCLOCALE_CPPFLAGS@ \
--- origsrc/libvshadow-20120511/libcerror/Makefile.am	2012-04-15 16:04:07.000000000 +0900
+++ src/libvshadow-20120511/libcerror/Makefile.am	2012-06-24 10:36:41.777868000 +0900
@@ -2,6 +2,7 @@ if HAVE_LOCAL_LIBCERROR
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcstring \
 	@LIBCSTRING_CPPFLAGS@ \
 	@MEMWATCH_CPPFLAGS@
 
--- origsrc/libvshadow-20120511/libcfile/Makefile.am	2012-04-15 16:04:19.000000000 +0900
+++ src/libvshadow-20120511/libcfile/Makefile.am	2012-06-24 18:16:39.215087200 +0900
@@ -2,6 +2,10 @@ if HAVE_LOCAL_LIBCFILE
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcerror \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libclocale \
+	-I$(top_srcdir)/libuna \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
 	@LIBCLOCALE_CPPFLAGS@ \
--- origsrc/libvshadow-20120511/libclocale/Makefile.am	2012-04-15 16:04:13.000000000 +0900
+++ src/libvshadow-20120511/libclocale/Makefile.am	2012-06-24 18:10:45.190640600 +0900
@@ -3,6 +3,8 @@ AM_CPPFLAGS = \
 	-DLOCALEDIR=\"$(datadir)/locale\" \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcerror \
+	-I$(top_srcdir)/libcstring \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
 	@MEMWATCH_CPPFLAGS@
--- origsrc/libvshadow-20120511/libcnotify/Makefile.am	2012-04-15 16:04:10.000000000 +0900
+++ src/libvshadow-20120511/libcnotify/Makefile.am	2012-06-24 18:11:44.584195300 +0900
@@ -1,6 +1,8 @@
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcerror \
+	-I$(top_srcdir)/libcstring \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
 	@MEMWATCH_CPPFLAGS@
--- origsrc/libvshadow-20120511/libcpath/Makefile.am	2012-04-15 16:04:16.000000000 +0900
+++ src/libvshadow-20120511/libcpath/Makefile.am	2012-06-24 18:18:07.132080500 +0900
@@ -2,6 +2,10 @@ if HAVE_LOCAL_LIBCPATH
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcerror \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libclocale \
+	-I$(top_srcdir)/libuna \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
 	@LIBCLOCALE_CPPFLAGS@ \
--- origsrc/libvshadow-20120511/libcsystem/Makefile.am	2012-05-11 13:49:45.000000000 +0900
+++ src/libvshadow-20120511/libcsystem/Makefile.am	2012-06-24 18:33:21.379602900 +0900
@@ -3,6 +3,11 @@ AM_CPPFLAGS = \
 	-DLOCALEDIR=\"$(datadir)/locale\" \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcerror \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libclocale \
+	-I$(top_srcdir)/libcnotify \
+	-I$(top_srcdir)/libuna \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
 	@LIBCLOCALE_CPPFLAGS@ \
--- origsrc/libvshadow-20120511/libfdatetime/Makefile.am	2012-05-11 13:50:10.000000000 +0900
+++ src/libvshadow-20120511/libfdatetime/Makefile.am	2012-06-24 18:20:37.416172700 +0900
@@ -2,6 +2,8 @@ if HAVE_LOCAL_LIBFDATETIME
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcerror \
+	-I$(top_srcdir)/libcstring \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
 	@MEMWATCH_CPPFLAGS@
--- origsrc/libvshadow-20120511/libfguid/Makefile.am	2012-05-11 13:50:13.000000000 +0900
+++ src/libvshadow-20120511/libfguid/Makefile.am	2012-06-24 18:24:07.870791400 +0900
@@ -2,6 +2,8 @@ if HAVE_LOCAL_LIBFGUID
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcerror \
+	-I$(top_srcdir)/libcstring \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
 	@MEMWATCH_CPPFLAGS@
--- origsrc/libvshadow-20120511/libuna/Makefile.am	2012-04-15 16:03:59.000000000 +0900
+++ src/libvshadow-20120511/libuna/Makefile.am	2012-06-24 18:12:43.613088900 +0900
@@ -2,6 +2,8 @@ if HAVE_LOCAL_LIBUNA
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcerror \
+	-I$(top_srcdir)/libcstring \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
 	@MEMWATCH_CPPFLAGS@
--- origsrc/libvshadow-20120511/libvshadow/Makefile.am	2012-04-15 16:07:39.000000000 +0900
+++ src/libvshadow-20120511/libvshadow/Makefile.am	2012-06-24 18:25:33.161204300 +0900
@@ -1,6 +1,16 @@
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcerror \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libclocale \
+	-I$(top_srcdir)/libcnotify \
+	-I$(top_srcdir)/libuna \
+	-I$(top_srcdir)/libcfile \
+	-I$(top_srcdir)/libcpath \
+	-I$(top_srcdir)/libbfio \
+	-I$(top_srcdir)/libfdatetime \
+	-I$(top_srcdir)/libfguid \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
 	@LIBCLOCALE_CPPFLAGS@ \
--- origsrc/libvshadow-20120511/vshadowtools/Makefile.am	2012-05-11 14:07:36.000000000 +0900
+++ src/libvshadow-20120511/vshadowtools/Makefile.am	2012-06-24 18:34:56.885068300 +0900
@@ -1,6 +1,17 @@
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcerror \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libclocale \
+	-I$(top_srcdir)/libcnotify \
+	-I$(top_srcdir)/libuna \
+	-I$(top_srcdir)/libbfio \
+	-I$(top_srcdir)/libfdatetime \
+	-I$(top_srcdir)/libfguid \
+	-I$(top_srcdir)/libcfile \
+	-I$(top_srcdir)/libcpath \
+	-I$(top_srcdir)/libcsystem \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
 	@LIBCLOCALE_CPPFLAGS@ \
