--- origsrc/libvshadow-20130113/libbfio/Makefile.am	2012-09-16 00:57:39.000000000 +0900
+++ src/libvshadow-20130113/libbfio/Makefile.am	2013-01-27 19:42:47.219343800 +0900
@@ -2,6 +2,13 @@ if HAVE_LOCAL_LIBBFIO
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libcerror \
+	-I$(top_srcdir)/libcdata \
+	-I$(top_srcdir)/libclocale \
+	-I$(top_srcdir)/libuna \
+	-I$(top_srcdir)/libcfile \
+	-I$(top_srcdir)/libcpath \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
 	@LIBCDATA_CPPFLAGS@ \
--- origsrc/libvshadow-20130113/libcdata/Makefile.am	2012-09-16 00:34:13.000000000 +0900
+++ src/libvshadow-20130113/libcdata/Makefile.am	2013-01-27 19:33:13.320906300 +0900
@@ -1,6 +1,8 @@
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libcerror \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@
 
--- origsrc/libvshadow-20130113/libcerror/Makefile.am	2012-09-12 04:29:20.000000000 +0900
+++ src/libvshadow-20130113/libcerror/Makefile.am	2013-01-27 19:31:46.131453200 +0900
@@ -2,7 +2,8 @@ if HAVE_LOCAL_LIBCERROR
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
-	@LIBCSTRING_CPPFLAGS@
+	@LIBCSTRING_CPPFLAGS@ \
+	-I$(top_srcdir)/libcstring
 
 noinst_LTLIBRARIES = libcerror.la
 
--- origsrc/libvshadow-20130113/libcfile/Makefile.am	2012-09-12 04:27:40.000000000 +0900
+++ src/libvshadow-20130113/libcfile/Makefile.am	2013-01-27 19:39:48.269148500 +0900
@@ -2,6 +2,10 @@ if HAVE_LOCAL_LIBCFILE
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libcerror \
+	-I$(top_srcdir)/libclocale \
+	-I$(top_srcdir)/libuna \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
 	@LIBCLOCALE_CPPFLAGS@ \
--- origsrc/libvshadow-20130113/libclocale/Makefile.am	2013-01-13 23:41:41.000000000 +0900
+++ src/libvshadow-20130113/libclocale/Makefile.am	2013-01-27 19:34:09.278914100 +0900
@@ -3,6 +3,8 @@ AM_CPPFLAGS = \
 	-DLOCALEDIR=\"$(datadir)/locale\" \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libcerror \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@
 
--- origsrc/libvshadow-20130113/libcnotify/Makefile.am	2012-09-12 04:29:06.000000000 +0900
+++ src/libvshadow-20130113/libcnotify/Makefile.am	2013-01-27 19:35:48.567976600 +0900
@@ -1,6 +1,8 @@
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libcerror \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@
 
--- origsrc/libvshadow-20130113/libcpath/Makefile.am	2012-09-12 04:33:45.000000000 +0900
+++ src/libvshadow-20130113/libcpath/Makefile.am	2013-01-27 19:41:14.497664100 +0900
@@ -2,6 +2,11 @@ if HAVE_LOCAL_LIBCPATH
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libcerror \
+	-I$(top_srcdir)/libclocale \
+	-I$(top_srcdir)/libcsplit \
+	-I$(top_srcdir)/libuna \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
 	@LIBCLOCALE_CPPFLAGS@ \
--- origsrc/libvshadow-20130113/libcsplit/Makefile.am	2012-09-12 04:31:04.000000000 +0900
+++ src/libvshadow-20130113/libcsplit/Makefile.am	2013-01-27 19:37:35.382429700 +0900
@@ -2,6 +2,8 @@ if HAVE_LOCAL_LIBCSPLIT
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libcerror \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@
 
--- origsrc/libvshadow-20130113/libcsystem/Makefile.am	2013-01-13 23:44:39.000000000 +0900
+++ src/libvshadow-20130113/libcsystem/Makefile.am	2013-01-27 19:49:38.929304700 +0900
@@ -2,6 +2,10 @@ if HAVE_LOCAL_LIBCSYSTEM
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libcerror \
+	-I$(top_srcdir)/libclocale \
+	-I$(top_srcdir)/libcnotify \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
 	@LIBCLOCALE_CPPFLAGS@ \
--- origsrc/libvshadow-20130113/libfdatetime/Makefile.am	2012-09-12 04:30:45.000000000 +0900
+++ src/libvshadow-20130113/libfdatetime/Makefile.am	2013-01-27 19:43:57.025984400 +0900
@@ -2,6 +2,8 @@ if HAVE_LOCAL_LIBFDATETIME
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libcerror \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@
 
--- origsrc/libvshadow-20130113/libfguid/Makefile.am	2012-09-12 04:30:40.000000000 +0900
+++ src/libvshadow-20130113/libfguid/Makefile.am	2013-01-27 19:44:40.745711000 +0900
@@ -2,6 +2,8 @@ if HAVE_LOCAL_LIBFGUID
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libcerror \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@
 
--- origsrc/libvshadow-20130113/libuna/Makefile.am	2012-09-12 04:30:33.000000000 +0900
+++ src/libvshadow-20130113/libuna/Makefile.am	2013-01-27 19:38:30.923445300 +0900
@@ -2,6 +2,8 @@ if HAVE_LOCAL_LIBUNA
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libcerror \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@
 
--- origsrc/libvshadow-20130113/libvshadow/Makefile.am	2013-01-13 23:38:30.000000000 +0900
+++ src/libvshadow-20130113/libvshadow/Makefile.am	2013-01-27 19:48:50.704695300 +0900
@@ -1,6 +1,18 @@
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libcerror \
+	-I$(top_srcdir)/libcdata \
+	-I$(top_srcdir)/libclocale \
+	-I$(top_srcdir)/libcnotify \
+	-I$(top_srcdir)/libcsplit \
+	-I$(top_srcdir)/libuna \
+	-I$(top_srcdir)/libcfile \
+	-I$(top_srcdir)/libcpath \
+	-I$(top_srcdir)/libbfio \
+	-I$(top_srcdir)/libfdatetime \
+	-I$(top_srcdir)/libfguid \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
 	@LIBCDATA_CPPFLAGS@ \
--- origsrc/libvshadow-20130113/pyvshadow/Makefile.am	2013-01-13 23:38:22.000000000 +0900
+++ src/libvshadow-20130113/pyvshadow/Makefile.am	2013-01-27 19:26:15.971027500 +0900
@@ -45,7 +45,7 @@ pyvshadow_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pyvshadow_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyvshadow_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyvshadow_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS) -no-undefined
 
 endif
 
--- origsrc/libvshadow-20130113/vshadowtools/Makefile.am	2013-01-13 23:38:25.000000000 +0900
+++ src/libvshadow-20130113/vshadowtools/Makefile.am	2013-01-27 19:51:01.215437500 +0900
@@ -1,6 +1,20 @@
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libcstring \
+	-I$(top_srcdir)/libcerror \
+	-I$(top_srcdir)/libcdata \
+	-I$(top_srcdir)/libclocale \
+	-I$(top_srcdir)/libcnotify \
+	-I$(top_srcdir)/libcsplit \
+	-I$(top_srcdir)/libuna \
+	-I$(top_srcdir)/libbfio \
+	-I$(top_srcdir)/libfdatetime \
+	-I$(top_srcdir)/libfguid \
+	-I$(top_srcdir)/libcfile \
+	-I$(top_srcdir)/libcpath \
+	-I$(top_srcdir)/libcsystem \
+	-I$(top_srcdir)/libvshadow \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
 	@LIBCDATA_CPPFLAGS@ \
