--- origsrc/airspyhf-1.6.8/libairspyhf/src/CMakeLists.txt	2020-04-09 22:29:36.000000000 +0900
+++ src/airspyhf-1.6.8/libairspyhf/src/CMakeLists.txt	2021-09-17 18:18:31.360152400 +0900
@@ -60,12 +60,6 @@ set_target_properties(airspyhf-static PR
 # Dependencies
 target_link_libraries(airspyhf ${LIBUSB_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
 
-# For cygwin just force UNIX OFF and WIN32 ON
-if( ${CYGWIN} )
-  SET(UNIX OFF)
-  SET(WIN32 ON)
-endif( ${CYGWIN} )
-
 if( ${UNIX} )
    install(TARGETS airspyhf
            LIBRARY DESTINATION lib${LIB_SUFFIX}
