--- origsrc/xmimsim-8.1/src/Makefile.am	2020-02-24 19:07:05.000000000 +0900
+++ src/xmimsim-8.1/src/Makefile.am	2022-06-21 08:07:52.654707900 +0900
@@ -91,7 +91,7 @@ libxmimsim_la_FCFLAGS = @OPENMP_FCFLAGS@
 			@xraylib_f2003_CFLAGS@ \
 			-DXMIMSIM_HDF5_DEFAULT=\"$(datadir)/xmimsim/xmimsimdata.h5\"
 
-libxmimsim_la_LDFLAGS = -version-info @LIB_CURRENT@:@LIB_REVISION@:@LIB_AGE@ @LDFLAGS_XMIMSIM_LIB@
+libxmimsim_la_LDFLAGS = -no-undefined -version-info @LIB_CURRENT@:@LIB_REVISION@:@LIB_AGE@ @LDFLAGS_XMIMSIM_LIB@
 libxmimsim_la_LIBADD = 	@HDF5_LIBS@ \
 			@PTHREAD_LIBS@ \
 			@xslt_LIBS@ \
@@ -113,7 +113,7 @@ xmimsim_cl_la_CFLAGS = @WSTRICT_CFLAGS@
 xmimsim_cl_la_CPPFLAGS = -I${top_srcdir}/include -I${top_builddir}/include @CPPFLAGS_CL@ @gmodule_CFLAGS@ @glib2_CFLAGS@ $(AM_CPPFLAGS)
 xmimsim_cl_la_LIBADD = @LIBS_CL@ libxmimsim.la
 xmimsim_cl_la_SOURCES = xmi_solid_angle_cl.c
-xmimsim_cl_la_LDFLAGS = -module $(LDFLAGS_XMIMSIM_CL) -avoid-version -shared
+xmimsim_cl_la_LDFLAGS = -no-undefined -module $(LDFLAGS_XMIMSIM_CL) -avoid-version -shared
 if ENABLE_MAC_APP
 xmimsim_cl_la_LDFLAGS += -Wl,-headerpad_max_install_names
 endif
