diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/clib-1.4.2/Makefile src/clib-1.4.2/Makefile
--- origsrc/clib-1.4.2/Makefile	2015-01-07 00:03:22.000000000 +0900
+++ src/clib-1.4.2/Makefile	2015-11-01 13:58:10.159155900 +0900
@@ -3,11 +3,19 @@ CC     ?= cc
 PREFIX ?= /usr/local
 
 ifeq ($(OS),Windows_NT)
+ifeq ($(shell uname -o),Cygwin)
+BINS    = clib.exe clib-install.exe clib-search.exe
+LDFLAGS = -lcurl
+CP      = cp -f
+RM      = rm -f
+MKDIR_P = mkdir -p
+else
 BINS    = clib.exe clib-install.exe clib-search.exe
 LDFLAGS = -lcurldll
 CP      = copy /Y
 RM      = del /Q /S
 MKDIR_P = mkdir
+endif
 else
 BINS    = clib clib-install clib-search
 LDFLAGS = -lcurl
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/clib-1.4.2/test.sh src/clib-1.4.2/test.sh
--- origsrc/clib-1.4.2/test.sh	2015-01-07 00:03:22.000000000 +0900
+++ src/clib-1.4.2/test.sh	2015-11-01 13:58:10.159155900 +0900
@@ -1,6 +1,11 @@
 #!/bin/bash
 
+if test x`uname -o` = xCygwin
+then
+TESTS=`find test/*`
+else
 TESTS=`find test/* -type f -perm +111`
+fi
 EXIT_CODE=0
 
 echo -e "\nRunning clib(1) tests\n"
