--- origsrc/clib-2.7.0/src/clib-build.c	2021-01-26 17:41:25.000000000 +0900
+++ src/clib-2.7.0/src/clib-build.c	2021-01-29 18:03:20.870538400 +0900
@@ -62,7 +62,7 @@
 #endif
 
 #if defined(_WIN32) || defined(WIN32) || defined(__MINGW32__) ||               \
-    defined(__MINGW64__) || defined(__CYGWIN__)
+    defined(__MINGW64__)
 #define setenv(k, v, _) _putenv_s(k, v)
 #define realpath(a, b) _fullpath(a, b, strlen(a))
 #endif
--- origsrc/clib-2.7.0/src/clib-configure.c	2021-01-26 17:41:25.000000000 +0900
+++ src/clib-2.7.0/src/clib-configure.c	2021-01-29 18:03:20.902608700 +0900
@@ -54,7 +54,7 @@
 #endif
 
 #if defined(_WIN32) || defined(WIN32) || defined(__MINGW32__) ||               \
-    defined(__MINGW64__) || defined(__CYGWIN__)
+    defined(__MINGW64__)
 #define setenv(k, v, _) _putenv_s(k, v)
 #define realpath(a, b) _fullpath(a, b, strlen(a))
 #endif
--- origsrc/clib-2.7.0/src/clib-init.c	2021-01-26 17:41:25.000000000 +0900
+++ src/clib-2.7.0/src/clib-init.c	2021-01-29 18:03:20.923607700 +0900
@@ -19,7 +19,7 @@
 #include <unistd.h>
 
 #if defined(_WIN32) || defined(WIN32) || defined(__MINGW32__) ||               \
-    defined(__MINGW64__) || defined(__CYGWIN__)
+    defined(__MINGW64__)
 #define setenv(k, v, _) _putenv_s(k, v)
 #define realpath(a, b) _fullpath(a, b, strlen(a))
 #endif
--- origsrc/clib-2.7.0/src/clib-install.c	2021-01-26 17:41:25.000000000 +0900
+++ src/clib-2.7.0/src/clib-install.c	2021-01-29 18:03:20.936628300 +0900
@@ -33,7 +33,7 @@
 #endif
 
 #if defined(_WIN32) || defined(WIN32) || defined(__MINGW32__) ||               \
-    defined(__MINGW64__) || defined(__CYGWIN__)
+    defined(__MINGW64__)
 #define setenv(k, v, _) _putenv_s(k, v)
 #define realpath(a, b) _fullpath(a, b, strlen(a))
 #endif
--- origsrc/clib-2.7.0/src/clib-search.c	2021-01-26 17:41:25.000000000 +0900
+++ src/clib-2.7.0/src/clib-search.c	2021-01-29 18:03:20.970646200 +0900
@@ -29,7 +29,7 @@
 #define CLIB_SEARCH_CACHE_TIME 1 * 24 * 60 * 60
 
 #if defined(_WIN32) || defined(WIN32) || defined(__MINGW32__) ||               \
-    defined(__MINGW64__) || defined(__CYGWIN__)
+    defined(__MINGW64__)
 #define setenv(k, v, _) _putenv_s(k, v)
 #define realpath(a, b) _fullpath(a, b, strlen(a))
 #endif
--- origsrc/clib-2.7.0/src/clib-uninstall.c	2021-01-26 17:41:25.000000000 +0900
+++ src/clib-2.7.0/src/clib-uninstall.c	2021-01-29 18:06:35.788039100 +0900
@@ -22,7 +22,7 @@
 #define CLIB_UNINSTALL_DEFAULT_TARGET "make uninstall"
 
 #if defined(_WIN32) || defined(WIN32) || defined(__MINGW32__) ||               \
-    defined(__MINGW64__) || defined(__CYGWIN__)
+    defined(__MINGW64__)
 #define setenv(k, v, _) _putenv_s(k, v)
 #endif
 
--- origsrc/clib-2.7.0/src/clib-update.c	2021-01-26 17:41:25.000000000 +0900
+++ src/clib-2.7.0/src/clib-update.c	2021-01-29 18:03:20.982611400 +0900
@@ -33,7 +33,7 @@
 #endif
 
 #if defined(_WIN32) || defined(WIN32) || defined(__MINGW32__) ||               \
-    defined(__MINGW64__) || defined(__CYGWIN__)
+    defined(__MINGW64__)
 #define setenv(k, v, _) _putenv_s(k, v)
 #define realpath(a, b) _fullpath(a, b, strlen(a))
 #endif
--- origsrc/clib-2.7.0/src/clib-upgrade.c	2021-01-26 17:41:25.000000000 +0900
+++ src/clib-2.7.0/src/clib-upgrade.c	2021-01-29 18:03:21.017620800 +0900
@@ -35,7 +35,7 @@
 #endif
 
 #if defined(_WIN32) || defined(WIN32) || defined(__MINGW32__) ||               \
-    defined(__MINGW64__) || defined(__CYGWIN__)
+    defined(__MINGW64__)
 #define setenv(k, v, _) _putenv_s(k, v)
 #define realpath(a, b) _fullpath(a, b, strlen(a))
 #endif
--- origsrc/clib-2.7.0/src/common/clib-package.c	2021-01-26 17:41:25.000000000 +0900
+++ src/clib-2.7.0/src/common/clib-package.c	2021-01-29 18:03:21.035614400 +0900
@@ -48,7 +48,7 @@
 #define GITHUB_CONTENT_URL_WITH_TOKEN "https://%s@raw.githubusercontent.com/"
 
 #if defined(_WIN32) || defined(WIN32) || defined(__MINGW32__) ||               \
-    defined(__MINGW64__) || defined(__CYGWIN__)
+    defined(__MINGW64__)
 #define setenv(k, v, _) _putenv_s(k, v)
 #define realpath(a, b) _fullpath(a, b, strlen(a))
 #endif
