--- origsrc/json-fortran-9.0.3/CMakeLists.txt	2025-02-23 01:41:27.000000000 +0900
+++ src/json-fortran-9.0.3/CMakeLists.txt	2025-04-12 21:37:44.347277500 +0900
@@ -128,7 +128,9 @@ endif ()
 
 # Put package export CMake files where they can be found
 # use `find_package ( jsonfortran-${CMAKE_Fortran_COMPILER_ID} <version> REQUIRED )`
-if (USE_GNU_INSTALL_CONVENTION)
+if (CYGWIN)
+  set ( EXPORT_INSTALL_DIR "${INSTALL_LIB_DIR}/cmake/${PROJECT_NAME}" )
+elseif (USE_GNU_INSTALL_CONVENTION)
   set ( EXPORT_INSTALL_DIR "${INSTALL_LIB_DIR}/cmake/${PACKAGE_VERSION}" )
 else ()
   set ( EXPORT_INSTALL_DIR "${PACKAGE_VERSION}/cmake" )
@@ -180,14 +182,12 @@ else()
     add_library ( ${LIB_NAME}-static STATIC ${JF_LIB_SRCS} )
   else()
     add_library ( ${LIB_NAME}        SHARED ${JF_LIB_SRCS} )
-    add_library ( ${LIB_NAME}-static STATIC ${JF_LIB_SRCS} )
   endif()
 endif()
 
 # add an alias so that including json-fortran is agnostic
 # of find_package or being directly compiled through add_subdirectory
 add_library ( ${PACKAGE_NAME}::${LIB_NAME}        ALIAS ${LIB_NAME} )
-add_library ( ${PACKAGE_NAME}::${LIB_NAME}-static ALIAS ${LIB_NAME}-static )
 
 if(JSON_FORTRAN_USE_OpenCoarrays)
   target_link_libraries(${LIB_NAME}
@@ -200,30 +200,6 @@ target_include_directories(${LIB_NAME}
   PUBLIC
   $<BUILD_INTERFACE:${MODULE_DIR}>
   $<INSTALL_INTERFACE:${INSTALL_MOD_DIR}>)
-target_include_directories(${LIB_NAME}-static
-  PUBLIC
-  $<BUILD_INTERFACE:${MODULE_DIR}>
-  $<INSTALL_INTERFACE:${INSTALL_MOD_DIR}>)
-if(CMAKE_Fortran_COMPILER_ID STREQUAL IntelLLVM)
-    set_target_properties ( ${LIB_NAME}-static
-      PROPERTIES
-      if(NOT MSVC_IDE)
-      PREFIX lib
-      endif()
-      VERSION ${PROJECT_VERSION}
-      ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib
-      Fortran_MODULE_DIRECTORY ${MODULE_DIR} )
-else()
-    set_target_properties ( ${LIB_NAME}-static
-      PROPERTIES
-      OUTPUT_NAME ${LIB_NAME}
-      if(NOT MSVC_IDE)
-      PREFIX lib
-      endif()
-      VERSION ${PROJECT_VERSION}
-      ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib
-      Fortran_MODULE_DIRECTORY ${MODULE_DIR} )
-endif()
 set_target_properties ( ${LIB_NAME}
   PROPERTIES
   OUTPUT_NAME ${LIB_NAME}
@@ -331,7 +307,7 @@ if ( ENABLE_TESTS )
   # emulate GNU Autotools `make check`
   add_custom_target(check COMMAND ${CMAKE_CTEST_COMMAND} -C $<CONFIG> --output-on-failure)
   add_custom_target(build_tests) # Make target to build all tests
-  add_dependencies(build_tests ${LIB_NAME} ${LIB_NAME}-static)
+  add_dependencies(build_tests ${LIB_NAME})
 
   find_program ( JSONLINT jsonlint )
 
@@ -453,8 +429,9 @@ endif ()
 # Perform the installation
 #-------------------------
 
-install ( TARGETS ${LIB_NAME} ${LIB_NAME}-static
+install ( TARGETS ${LIB_NAME}
   EXPORT ${PACKAGE_NAME}-targets
+  RUNTIME DESTINATION bin
   LIBRARY DESTINATION "${INSTALL_LIB_DIR}"
   ARCHIVE DESTINATION "${INSTALL_LIB_DIR}" )
 
--- origsrc/json-fortran-9.0.3/json-fortran.pc.cmake.in	2025-02-23 01:41:27.000000000 +0900
+++ src/json-fortran-9.0.3/json-fortran.pc.cmake.in	2025-04-12 21:37:44.347277500 +0900
@@ -6,4 +6,4 @@ Description: @PROJECT_DESCRIPTION@
 URL: @PROJECT_URL@
 Version: @PROJECT_VERSION@
 Libs: -L${libdir} -ljsonfortran
-Cflags: -I${includedir}
+Cflags: -I${includedir}/json-fortran
--- origsrc/json-fortran-9.0.3/packaging/CMakeLists.txt	2025-02-23 01:41:27.000000000 +0900
+++ src/json-fortran-9.0.3/packaging/CMakeLists.txt	2025-04-12 21:37:44.347277500 +0900
@@ -26,7 +26,7 @@ install ( FILES
 #----------------------------------------------
 # Make build tree targets accessible for import
 #----------------------------------------------
-export ( TARGETS ${LIB_NAME} ${LIB_NAME}-static FILE ${PACKAGE_NAME}-targets.cmake )
+export ( TARGETS ${LIB_NAME} FILE ${PACKAGE_NAME}-targets.cmake )
 
 # build tree package config file, NOT installed
 configure_file (
@@ -54,4 +54,4 @@ install(FILES
 if(MSVC_IDE)
    INCLUDE_DIRECTORIES("src")
    SET(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fpp")
-endif()
\ No newline at end of file
+endif()
