--- origsrc/CRF++-0.58/Makefile.am	2012-09-30 01:44:38.000000000 +0900
+++ src/CRF++-0.58/Makefile.am	2013-09-16 20:15:05.875013500 +0900
@@ -9,6 +9,7 @@ libcrfpp_la_SOURCES = crfpp.h thread.h l
                       feature_cache.cpp feature_index.cpp node.cpp path.cpp tagger.cpp \
 		      common.h darts.h encoder.h feature_cache.h feature_index.h \
                       freelist.h lbfgs.h mmap.h node.h path.h tagger.h timer.h winmain.h
+libcrfpp_la_LDFLAGS = -no-undefined
 include_HEADERS = crfpp.h
 
 dist-hook:
--- origsrc/CRF++-0.58/winmain.h	2012-09-30 01:44:38.000000000 +0900
+++ src/CRF++-0.58/winmain.h	2013-09-16 20:15:05.968763500 +0900
@@ -5,7 +5,7 @@
 //
 //  Copyright(C) 2005-2007 Taku Kudo <taku@chasen.org>
 //
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 
 #include <windows.h>
 #include <string>
