--- origsrc/msgpack11-0.1.0/CMakeLists.txt	2022-06-03 06:21:44.000000000 +0900
+++ src/msgpack11-0.1.0/CMakeLists.txt	2024-10-09 09:12:20.978285800 +0900
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.2)
-project(msgpack11 VERSION 0.0.9 LANGUAGES CXX C)
+project(msgpack11 VERSION 0.1.0 LANGUAGES CXX C)
 
 option(MSGPACK11_BUILD_TESTS "Build unit tests" ON)
 
@@ -12,6 +12,9 @@ target_compile_options(msgpack11 PRIVATE
 if(NOT MSVC)
   target_compile_options(msgpack11 PRIVATE -Wall -Wextra -Werror)
 endif()
+if (DEFINED SOVERSION)
+  set_target_properties(msgpack11 PROPERTIES SOVERSION "${SOVERSION}")
+endif()
 configure_file("msgpack11.pc.in" "msgpack11.pc" @ONLY)
 
 if (MSGPACK11_BUILD_TESTS)
@@ -19,6 +22,6 @@ if (MSGPACK11_BUILD_TESTS)
   add_subdirectory(test)
 endif()
 
-install(TARGETS msgpack11 DESTINATION lib)
+install(TARGETS msgpack11 RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/msgpack11.hpp" DESTINATION include)
 install(FILES "${CMAKE_BINARY_DIR}/msgpack11.pc" DESTINATION lib/pkgconfig)
