--- origsrc/dnsdbq-2.6.7/Makefile	2024-05-20 07:29:31.000000000 +0900
+++ src/dnsdbq-2.6.7/Makefile	2024-06-01 04:57:30.616689800 +0900
@@ -32,7 +32,7 @@ CWARN =-W -Wall -Wextra -Wcast-qual -Wpo
 	-Wmissing-prototypes  -Wbad-function-cast -Wnested-externs \
 	-Wunused -Wshadow -Wmissing-noreturn -Wswitch-enum -Wconversion
 # try shipping without any warnings
-CWARN   +=-Werror
+#CWARN   +=-Werror
 # warning about bad indentation, only for clang 6.x+
 #CWARN   +=-Werror=misleading-indentation
 
