--- origsrc/wimlib-1.12.0/configure.ac	2017-07-30 05:50:08.000000000 +0900
+++ src/wimlib-1.12.0/configure.ac	2017-12-26 17:10:53.638047000 +0900
@@ -46,6 +46,12 @@ linux*)
 	WITH_NTFS_3G_DEFAULT="yes"
 	WITH_FUSE_DEFAULT="yes"
 	;;
+cygwin*)
+	# Cygwin
+	WITH_NTFS_3G_DEFAULT="no"
+	WITH_FUSE_DEFAULT="no"
+	PLATFORM_LDFLAGS="-no-undefined"
+	;;
 *)
 	# Other UNIX
 	WITH_NTFS_3G_DEFAULT="yes"
--- origsrc/wimlib-1.12.0/include/wimlib/file_io.h	2016-12-27 02:05:26.000000000 +0900
+++ src/wimlib-1.12.0/include/wimlib/file_io.h	2017-12-26 17:19:02.104000000 +0900
@@ -26,7 +26,7 @@ full_write(struct filedes *fd, const voi
 extern int
 full_pwrite(struct filedes *fd, const void *buf, size_t count, off_t offset);
 
-#ifndef __WIN32__
+#if !defined(O_BINARY)
 #  define O_BINARY 0
 #endif
 
