--- origsrc/wimlib-1.13.1/configure.ac	2019-05-07 11:25:17.000000000 +0900
+++ src/wimlib-1.13.1/configure.ac	2019-06-09 18:16:59.113538600 +0900
@@ -46,6 +46,12 @@ linux*)
 	WITH_NTFS_3G_DEFAULT="yes"
 	WITH_FUSE_DEFAULT="yes"
 	;;
+cygwin*)
+	# Cygwin
+	WITH_NTFS_3G_DEFAULT="no"
+	WITH_FUSE_DEFAULT="no"
+	PLATFORM_LDFLAGS="-no-undefined"
+	;;
 *)
 	# Other UNIX
 	WITH_NTFS_3G_DEFAULT="yes"
--- origsrc/wimlib-1.13.1/include/wimlib/file_io.h	2017-09-20 11:55:16.000000000 +0900
+++ src/wimlib-1.13.1/include/wimlib/file_io.h	2019-06-09 18:16:59.129092700 +0900
@@ -26,7 +26,7 @@ full_write(struct filedes *fd, const voi
 extern int
 full_pwrite(struct filedes *fd, const void *buf, size_t count, off_t offset);
 
-#ifndef __WIN32__
+#if !defined(O_BINARY)
 #  define O_BINARY 0
 #endif
 
