--- origsrc/wimlib-1.14.5/configure.ac	2024-10-27 08:59:19.000000000 +0900
+++ src/wimlib-1.14.5/configure.ac	2026-02-07 11:47:05.488061500 +0900
@@ -48,6 +48,12 @@ linux*)
 	WITH_NTFS_3G_DEFAULT="yes"
 	WITH_FUSE_DEFAULT="yes"
 	;;
+cygwin*)
+	# Cygwin
+	WITH_NTFS_3G_DEFAULT="no"
+	WITH_FUSE_DEFAULT="no"
+	PLATFORM_LDFLAGS="-no-undefined"
+	;;
 *)
 	# Other UNIX
 	WITH_NTFS_3G_DEFAULT="yes"
--- origsrc/wimlib-1.14.5/include/wimlib/file_io.h	2026-01-30 11:08:05.000000000 +0900
+++ src/wimlib-1.14.5/include/wimlib/file_io.h	2026-02-07 11:47:05.503785700 +0900
@@ -26,7 +26,7 @@ full_write(struct filedes *fd, const voi
 int
 full_pwrite(struct filedes *fd, const void *buf, size_t count, off_t offset);
 
-#ifndef _WIN32
+#if !defined(O_BINARY)
 #  define O_BINARY 0
 #endif
 
--- origsrc/wimlib-1.14.5/tests/test-imagex	2024-10-27 08:59:19.000000000 +0900
+++ src/wimlib-1.14.5/tests/test-imagex	2026-02-07 11:47:05.505842200 +0900
@@ -5,9 +5,9 @@
 
 set -e
 cd tests
-srcdir="${srcdir:-.}/.."
-srcdir="$(cd $srcdir; pwd)"
-. "$srcdir/tests/test_utils.sh"
+_srcdir="${_srcdir:-.}/.."
+_srcdir="$(cd $_srcdir; pwd)"
+. "$_srcdir/tests/test_utils.sh"
 
 TEST_SUBDIR=tmpdir_test-imagex
 
@@ -18,7 +18,7 @@ cd $TEST_SUBDIR
 
 # Make test directory
 mkdir dir
-cp $srcdir/src/*.c dir
+cp $_srcdir/src/*.c dir
 mkdir dir/subdir
 echo 'hello' > dir/subdir/hello
 echo 'hello' > dir/subdir/hello2
--- origsrc/wimlib-1.14.5/tests/test-imagex-capture_and_apply	2024-10-27 08:59:19.000000000 +0900
+++ src/wimlib-1.14.5/tests/test-imagex-capture_and_apply	2026-02-07 11:47:05.521772000 +0900
@@ -10,9 +10,9 @@
 
 set -e
 cd tests
-srcdir="${srcdir:-.}/.."
-srcdir="$(cd $srcdir; pwd)"
-. "$srcdir/tests/test_utils.sh"
+_srcdir="${_srcdir:-.}/.."
+_srcdir="$(cd $_srcdir; pwd)"
+. "$_srcdir/tests/test_utils.sh"
 
 TEST_SUBDIR=tmpdir_test-imagex-capture_and_apply
 
@@ -140,7 +140,7 @@ mkdir $TEST_SUBDIR
 cd $TEST_SUBDIR
 mkdir in.dir out.dir
 
-. $srcdir/tests/common_tests.sh
+. $_srcdir/tests/common_tests.sh
 
 # Test the data recovery mode
 __msg "Testing data recovery mode"
@@ -225,7 +225,7 @@ macOS=false
 if [ "$(uname)" = Darwin ]; then
 	macOS=true
 fi
-for t_file in "$srcdir/tests/exclusionlists"/*; do
+for t_file in "$_srcdir/tests/exclusionlists"/*; do
 	if $macOS && [[ $t_file == */case_*sensitive ]]; then
 		# Exclude test cases that fail on case-insensitive filesystem
 		continue
--- origsrc/wimlib-1.14.5/tests/test-imagex-update_and_extract	2024-10-27 08:59:19.000000000 +0900
+++ src/wimlib-1.14.5/tests/test-imagex-update_and_extract	2026-02-07 11:47:05.535392200 +0900
@@ -4,9 +4,9 @@
 
 set -e
 cd tests
-srcdir="${srcdir:-.}/.."
-srcdir="$(cd $srcdir; pwd)"
-. "$srcdir/tests/test_utils.sh"
+_srcdir="${_srcdir:-.}/.."
+_srcdir="$(cd $_srcdir; pwd)"
+. "$_srcdir/tests/test_utils.sh"
 
 TEST_SUBDIR=tmpdir_test-imagex-update_and_extract
 
@@ -38,7 +38,7 @@ do_apply() {
 }
 
 prepare_empty_wim
-cp $srcdir/src/add_image.c file
+cp $_srcdir/src/add_image.c file
 echo 1 > 1
 echo 2 > 2
 
@@ -204,7 +204,7 @@ msg "Testing adding files to WIM image"
 rm -rf in.dir
 mkdir in.dir
 wimappend in.dir test.wim "2"
-cp $srcdir/src/*.c in.dir
+cp $_srcdir/src/*.c in.dir
 wimupdate test.wim 2 << EOF
 add in.dir /
 add file /file
