--- origsrc/libilbm/src/acbm2ilbm/Makefile.am	2024-11-20 13:19:10.000000000 +0900
+++ src/libilbm/src/acbm2ilbm/Makefile.am	2024-11-20 13:22:37.830242200 +0900
@@ -1,7 +1,7 @@
 acbm2ilbm.1: main.c
 	$(HELP2MAN) --output=$@ --no-info --name 'Convert all ACBM images inside an IFF file to ILBM images' --libtool ./acbm2ilbm
 
-AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H)
+AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H) -I$(top_srcdir)/src/libilbm
 
 bin_PROGRAMS = acbm2ilbm
 noinst_HEADERS = acbm2ilbm.h
--- origsrc/libilbm/src/ilbm2acbm/Makefile.am	2024-11-20 13:19:10.000000000 +0900
+++ src/libilbm/src/ilbm2acbm/Makefile.am	2024-11-20 13:22:43.719533500 +0900
@@ -1,7 +1,7 @@
 ilbm2acbm.1: main.c
 	$(HELP2MAN) --output=$@ --no-info --name 'Convert all ILBM images inside an IFF file to ACBM images' --libtool ./ilbm2acbm
 
-AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H)
+AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H) -I$(top_srcdir)/src/libilbm
 
 bin_PROGRAMS = ilbm2acbm
 noinst_HEADERS = ilbm2acbm.h
--- origsrc/libilbm/src/ilbmpack/Makefile.am	2024-11-20 13:19:10.000000000 +0900
+++ src/libilbm/src/ilbmpack/Makefile.am	2024-11-20 13:21:43.288675700 +0900
@@ -1,7 +1,7 @@
 ilbmpack.1: main.c
 	$(HELP2MAN) --output=$@ --no-info --name 'Compress or decompress ILBM images inside an IFF file' --libtool ./ilbmpack
 
-AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H)
+AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H) -I$(top_srcdir)/src/libilbm
 
 bin_PROGRAMS = ilbmpack
 noinst_HEADERS = pack.h
--- origsrc/libilbm/src/ilbmpp/Makefile.am	2024-11-20 13:19:10.000000000 +0900
+++ src/libilbm/src/ilbmpp/Makefile.am	2024-11-20 13:22:13.911278700 +0900
@@ -1,7 +1,7 @@
 ilbmpp.1: main.c
 	$(HELP2MAN) --output=$@ --no-info --name ' Displays a textual representation of an IFF/ILBM file on the standard output' --libtool ./ilbmpp
 
-AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H)
+AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H) -I$(top_srcdir)/src/libilbm
 
 bin_PROGRAMS = ilbmpp
 noinst_HEADERS = pp.h
--- origsrc/libilbm/src/libilbm/Makefile.am	2024-11-20 13:19:10.000000000 +0900
+++ src/libilbm/src/libilbm/Makefile.am	2024-11-20 13:19:44.905051500 +0900
@@ -4,3 +4,4 @@ pkginclude_HEADERS = bitmapheader.h colo
 libilbm_la_SOURCES = bitmapheader.c colormap.c colorrange.c cycleinfo.c destmerge.c dpiheader.c grab.c sprite.c viewport.c byterun.c ilbm.c interleave.c ilbmimage.c drange.c cmykmap.c colornames.c
 libilbm_la_CFLAGS = $(LIBIFF_CFLAGS)
 libilbm_la_LIBADD = $(LIBIFF_LIBS)
+libilbm_la_LDFLAGS = -no-undefined
--- origsrc/libilbm/tests/Makefile.am	2024-11-20 13:19:10.000000000 +0900
+++ src/libilbm/tests/Makefile.am	2024-11-20 13:26:54.655915800 +0900
@@ -45,3 +45,5 @@ byterun_CFLAGS = -I../src/libilbm $(LIBI
 TESTS = writesimpleilbm writesimpleilbm-padded readsimpleilbm writesimplepbm readsimplepbm writesimpleacbm readsimpleacbm interleave-simple.sh interleave-simple-padded.sh byterun-simple.sh byterun-simple-padded.sh
 
 EXTRA_DIST = interleave-simple.sh interleave-simple-padded.sh byterun-simple.sh byterun-simple-padded.sh
+
+AM_CPPFLAGS= -I$(top_srcdir)/src/libilbm
