--- origsrc/libgeodecomp-0.4.0/src/CMakeLists.txt	2014-10-28 18:12:48.000000000 +0900
+++ src/libgeodecomp-0.4.0/src/CMakeLists.txt	2014-11-09 17:19:55.796908700 +0900
@@ -70,6 +70,7 @@ set(PythonLibs_FIND_VERSION_MAJOR 2)
 find_package(CUDA)
 find_package(OpenCL)
 find_package(MPI)
+find_package(PkgConfig)
 find_package(OpenCV)
 find_package(OpenMP)
 find_package(PythonLibs)
@@ -445,6 +446,7 @@ set(ALL_SOURCES ${SOURCES})
 set(ALL_HEADERS ${HEADERS})
 
 add_library(geodecomp ${LIB_LINKAGE_TYPE} ${SOURCES})
+set_property(TARGET geodecomp PROPERTY SOVERSION 0)
 # CMake <2.8.11 doesn't have target_include_directories()
 if(CMAKE_VERSION VERSION_GREATER 2.8.11)
   target_include_directories(geodecomp INTERFACE ${INSTALL_INCLUDE_DIRECTORIES})
@@ -512,8 +514,8 @@ target_link_libraries(geodecomp ${LIBGEO
 install(
   TARGETS geodecomp
   EXPORT ${PACKAGE_NAME}-targets
-  ARCHIVE DESTINATION lib
-  LIBRARY DESTINATION lib)
+  RUNTIME DESTINATION bin
+  ARCHIVE DESTINATION lib)
 
 install(FILES "${CMAKE_BINARY_DIR}/${PACKAGE_NAME}/config.h" DESTINATION include/${PACKAGE_NAME})
 install(FILES libgeodecomp.h DESTINATION include)
