--- origsrc/libfolia-1.10/src/Makefile.am	2017-10-18 00:26:03.000000000 +0900
+++ src/libfolia-1.10/src/Makefile.am	2017-10-21 13:30:29.804050000 +0900
@@ -5,7 +5,7 @@ AM_CXXFLAGS = -std=c++11 -g -O3 -W -Wall
 LDADD = libfolia.la
 
 lib_LTLIBRARIES = libfolia.la
-libfolia_la_LDFLAGS = -version-info 6:3:0
+libfolia_la_LDFLAGS = -no-undefined -version-info 6:3:0
 
 libfolia_la_SOURCES = folia_impl.cxx folia_document.cxx folia_utils.cxx \
 	folia_types.cxx folia_properties.cxx
