--- origsrc/libfolia-2.20/src/Makefile.am	2024-09-12 19:53:11.000000000 +0900
+++ src/libfolia-2.20/src/Makefile.am	2024-09-13 12:58:36.029811400 +0900
@@ -11,6 +11,8 @@ libfolia_la_SOURCES = folia_impl.cxx fol
 	folia_types.cxx folia_properties.cxx folia_provenance.cxx \
 	folia_subclasses.cxx folia_textpolicy.cxx folia_engine.cxx
 
+libfolia_la_LDFLAGS += -no-undefined
+
 bin_PROGRAMS = folialint
 folialint_SOURCES = folialint.cxx
 
