--- origsrc/cmrt-1.0.6/src/Makefile.am	2016-09-08 08:51:38.000000000 +0900
+++ src/cmrt-1.0.6/src/Makefile.am	2017-12-29 18:53:04.455000000 +0900
@@ -46,7 +46,6 @@ cmrt_cflags = \
 	-msse4.1                \
 	-msse4.2                \
 	-msse4                  \
-	-fPIC                  \
 	$(NULL)
 
 cmrt_ldflags = \
--- origsrc/cmrt-1.0.6/src/cm_mem.h	2016-09-08 08:51:38.000000000 +0900
+++ src/cmrt-1.0.6/src/cm_mem.h	2017-12-29 18:59:34.259726300 +0900
@@ -41,7 +41,9 @@
 #include <smmintrin.h>
 
 typedef uintptr_t UINT_PTR;
+#if !defined(__fastcall)
 #define __fastcall
+#endif
 #define __noop
 
 #ifdef __try
--- origsrc/cmrt-1.0.6/src/oscl_impl_linux.c	2016-09-08 08:51:38.000000000 +0900
+++ src/cmrt-1.0.6/src/oscl_impl_linux.c	2017-12-29 19:07:10.659474300 +0900
@@ -71,6 +71,7 @@ BOOL FreeLibrary(HMODULE hLibModule)
 	return SUCCESS;
 }
 
+#if !defined(__CYGWIN__)
 FARPROC GetProcAddress(HMODULE hModule, LPCSTR lpProcName)
 {
 	VOID *pSym;
@@ -79,6 +80,7 @@ FARPROC GetProcAddress(HMODULE hModule,
 	SetErrorNum(ERROR_SUCCESS);
 	return (FARPROC) pSym;
 }
+#endif
 
 static VOID SetErrorNum(DWORD Error)
 {
--- origsrc/cmrt-1.0.6/src/oscl_platform_def.h	2016-09-08 08:51:38.000000000 +0900
+++ src/cmrt-1.0.6/src/oscl_platform_def.h	2017-12-29 19:03:28.897425700 +0900
@@ -31,7 +31,9 @@
 
 #include "os_defs.h"
 
+#if !defined(UINT_MAX)
 #define UINT_MAX            0xffffffff
+#endif
 #define INFINITE            0xFFFFFFFF
 #define MAX_PATH            128
 
