--- origsrc/simbody-Simbody-3.6.1/SimTKcommon/src/ParallelExecutor.cpp	2018-06-12 10:04:33.000000000 +0900
+++ src/simbody-Simbody-3.6.1/SimTKcommon/src/ParallelExecutor.cpp	2019-01-22 10:47:22.992900500 +0900
@@ -184,7 +184,7 @@ void ParallelExecutor::execute(Task& tas
    #include <dlfcn.h>
 #elif _WIN32
    #include <windows.h>
-#elif __linux__
+#elif defined(__linux__) || defined(__CYGWIN__)
    #include <dlfcn.h>
    #include <unistd.h>
 #else
@@ -203,7 +203,7 @@ int ParallelExecutor::getNumProcessors()
        return(1);
     }
 #else
-#ifdef __linux__
+#if defined(__linux__) || defined(__CYGWIN__)
     long nProcessorsOnline     = sysconf(_SC_NPROCESSORS_ONLN);
     if( nProcessorsOnline == -1 )  {
         return(1);
