--- origsrc/cityinfo-0.5.19/src/Makefile.am	2020-10-31 02:14:48.000000000 +0900
+++ src/cityinfo-0.5.19/src/Makefile.am	2021-08-17 17:34:18.249355900 +0900
@@ -5,7 +5,8 @@ libcityinfo0_la_SOURCES = \
 	cityinfo.h
 
 libcityinfo0_la_CFLAGS = $(CITYINFO_CFLAGS)
-libcityinfo0_la_LDFLAGS = -Wl,--as-needed
+libcityinfo0_la_LDFLAGS = -no-undefined -Wl,--as-needed
+libcityinfo0_la_LIBADD = $(CITYINFO_LIBS)
 
 libcityinfoincludedir=$(includedir)
 libcityinfoinclude_HEADERS = \
