--- origsrc/libuecc-5/src/CMakeLists.txt	2015-01-27 03:41:00.000000000 +0900
+++ src/libuecc-5/src/CMakeLists.txt	2015-03-06 18:32:42.679070800 +0900
@@ -11,14 +11,24 @@ set_target_properties(uecc_shared PROPER
   COMPILE_FLAGS -Wall
 )
 
+if (NOT CYGWIN)
 add_library(uecc_static STATIC ${UECC_SRC})
 set_target_properties(uecc_static PROPERTIES
   OUTPUT_NAME uecc
   COMPILE_FLAGS -Wall
 )
+endif()
 
+if (CYGWIN)
+install(TARGETS uecc_shared
+  ARCHIVE DESTINATION ${LIBDIR}
+  LIBRARY DESTINATION ${LIBDIR}
+  RUNTIME DESTINATION bin
+)
+else()
 install(TARGETS uecc_shared uecc_static
   ARCHIVE DESTINATION ${LIBDIR}
   LIBRARY DESTINATION ${LIBDIR}
   RUNTIME DESTINATION bin
 )
+endif()
