--- origsrc/libuecc-7/src/CMakeLists.txt	2016-03-27 09:37:26.000000000 +0900
+++ src/libuecc-7/src/CMakeLists.txt	2024-12-28 14:42:09.768135400 +0900
@@ -11,13 +11,7 @@ set_target_properties(uecc_shared PROPER
   COMPILE_FLAGS -Wall
 )
 
-add_library(uecc_static STATIC ${UECC_SRC})
-set_target_properties(uecc_static PROPERTIES
-  OUTPUT_NAME uecc
-  COMPILE_FLAGS -Wall
-)
-
-install(TARGETS uecc_shared uecc_static
+install(TARGETS uecc_shared
   ARCHIVE DESTINATION ${LIBDIR}
   LIBRARY DESTINATION ${LIBDIR}
   RUNTIME DESTINATION bin
