diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libuecc-6/src/CMakeLists.txt src/libuecc-6/src/CMakeLists.txt
--- origsrc/libuecc-6/src/CMakeLists.txt	2015-10-26 00:01:42.000000000 +0900
+++ src/libuecc-6/src/CMakeLists.txt	2015-11-03 13:49:11.894016200 +0900
@@ -11,14 +11,24 @@ set_target_properties(uecc_shared PROPER
   COMPILE_FLAGS -Wall
 )
 
+if (NOT CYGWIN)
 add_library(uecc_static STATIC ${UECC_SRC})
 set_target_properties(uecc_static PROPERTIES
   OUTPUT_NAME uecc
   COMPILE_FLAGS -Wall
 )
+endif()
 
+if (CYGWIN)
+install(TARGETS uecc_shared
+  ARCHIVE DESTINATION ${LIBDIR}
+  LIBRARY DESTINATION ${LIBDIR}
+  RUNTIME DESTINATION bin
+)
+else()
 install(TARGETS uecc_shared uecc_static
   ARCHIVE DESTINATION ${LIBDIR}
   LIBRARY DESTINATION ${LIBDIR}
   RUNTIME DESTINATION bin
 )
+endif()
