--- origsrc/libuecc-7/src/CMakeLists.txt	2016-03-27 09:37:26.000000000 +0900
+++ src/libuecc-7/src/CMakeLists.txt	2016-04-09 17:02:30.657035900 +0900
@@ -11,14 +11,24 @@ set_target_properties(uecc_shared PROPER
   COMPILE_FLAGS -Wall
 )
 
+if (NOT CYGWIN)
 add_library(uecc_static STATIC ${UECC_SRC})
 set_target_properties(uecc_static PROPERTIES
   OUTPUT_NAME uecc
   COMPILE_FLAGS -Wall
 )
+endif()
 
+if (CYGWIN)
+install(TARGETS uecc_shared
+  ARCHIVE DESTINATION ${LIBDIR}
+  LIBRARY DESTINATION ${LIBDIR}
+  RUNTIME DESTINATION bin
+)
+else()
 install(TARGETS uecc_shared uecc_static
   ARCHIVE DESTINATION ${LIBDIR}
   LIBRARY DESTINATION ${LIBDIR}
   RUNTIME DESTINATION bin
 )
+endif()
