--- origsrc/libbitcoin-blockchain-3.6.0/Makefile.am	2019-08-29 23:57:16.000000000 +0900
+++ src/libbitcoin-blockchain-3.6.0/Makefile.am	2019-09-22 13:49:25.678253500 +0900
@@ -51,6 +51,7 @@ src_libbitcoin_blockchain_la_SOURCES = \
     src/validate/validate_block.cpp \
     src/validate/validate_input.cpp \
     src/validate/validate_transaction.cpp
+src_libbitcoin_blockchain_la_LDFLAGS = -no-undefined
 
 # local: test/libbitcoin-blockchain-test
 #------------------------------------------------------------------------------
--- origsrc/libbitcoin-blockchain-3.6.0/include/bitcoin/blockchain/define.hpp	2019-08-29 23:57:16.000000000 +0900
+++ src/libbitcoin-blockchain-3.6.0/include/bitcoin/blockchain/define.hpp	2019-09-22 13:50:27.580157600 +0900
@@ -44,6 +44,7 @@
 // DLL exports (or does nothing for static build)
 // BCD_INTERNAL is used for non-api symbols.
 
+#if !defined(BCD_API)
 #if defined BCB_STATIC
     #define BCD_API
     #define BCD_INTERNAL
@@ -54,6 +55,7 @@
     #define BCD_API      BC_HELPER_DLL_IMPORT
     #define BCD_INTERNAL BC_HELPER_DLL_LOCAL
 #endif
+#endif
 
 // Log name.
 #define LOG_BLOCKCHAIN "blockchain"
