--- origsrc/libbitcoin-blockchain-3.5.0/Makefile.am	2018-01-02 14:54:27.000000000 +0900
+++ src/libbitcoin-blockchain-3.5.0/Makefile.am	2018-04-05 17:50:18.504480700 +0900
@@ -34,6 +34,7 @@ doc_DATA = \
 lib_LTLIBRARIES = src/libbitcoin-blockchain.la
 src_libbitcoin_blockchain_la_CPPFLAGS = -I${srcdir}/include ${bitcoin_database_CPPFLAGS} ${bitcoin_consensus_CPPFLAGS}
 src_libbitcoin_blockchain_la_LIBADD = ${bitcoin_database_LIBS} ${bitcoin_consensus_LIBS}
+src_libbitcoin_blockchain_la_LDFLAGS = -no-undefined
 src_libbitcoin_blockchain_la_SOURCES = \
     src/settings.cpp \
     src/interface/block_chain.cpp \
--- origsrc/libbitcoin-blockchain-3.5.0/configure.ac	2018-01-02 14:54:27.000000000 +0900
+++ src/libbitcoin-blockchain-3.5.0/configure.ac	2018-04-05 18:10:12.613000000 +0900
@@ -13,7 +13,7 @@
 AC_PREREQ([2.65])
 
 # Process command-line arguments and perform initialization and verification.
-AC_INIT([libbitcoin-blockchain], [3.4.0], [eric@voskuil.org])
+AC_INIT([libbitcoin-blockchain], [3.5.0], [eric@voskuil.org])
 
 # Do compilation tests.
 AC_LANG(C++)
@@ -227,18 +227,19 @@ AS_CASE([${CC}], [*],
     [AX_CHECK_LINK_FLAG([-fstack-protector-all],
         [LDFLAGS="$LDFLAGS -fstack-protector-all"])])
 
-# Hide internal functions from external libs. Enabled in gcc only.
-#------------------------------------------------------------------------------
-AS_CASE([${CC}], [*gcc*],
-    [AX_CHECK_COMPILE_FLAG([-fvisibility-hidden],
-        [CXXFLAGS="$CXXFLAGS -fvisibility-hidden"])])
-
-# Hide inlines from external libs. Enabled in gcc only.
-#------------------------------------------------------------------------------
-AS_CASE([${CC}], [*gcc*],
-    [AX_CHECK_COMPILE_FLAG([-fvisibility-inlines-hidden],
-        [CXXFLAGS="$CXXFLAGS -fvisibility-inlines-hidden"])])
-
+if test "$host_os" != "cygwin"; then
+    # Hide internal functions from external libs. Enabled in gcc only.
+    #------------------------------------------------------------------------------
+    AS_CASE([${CC}], [*gcc*],
+        [AX_CHECK_COMPILE_FLAG([-fvisibility-hidden],
+            [CXXFLAGS="$CXXFLAGS -fvisibility-hidden"])])
+
+    # Hide inlines from external libs. Enabled in gcc only.
+    #------------------------------------------------------------------------------
+    AS_CASE([${CC}], [*gcc*],
+        [AX_CHECK_COMPILE_FLAG([-fvisibility-inlines-hidden],
+            [CXXFLAGS="$CXXFLAGS -fvisibility-inlines-hidden"])])
+fi
 
 # Process outputs into templates.
 #==============================================================================
--- origsrc/libbitcoin-blockchain-3.5.0/include/bitcoin/blockchain/define.hpp	2018-01-02 14:54:27.000000000 +0900
+++ src/libbitcoin-blockchain-3.5.0/include/bitcoin/blockchain/define.hpp	2018-04-05 18:01:41.237000000 +0900
@@ -44,6 +44,7 @@
 // DLL exports (or does nothing for static build)
 // BCD_INTERNAL is used for non-api symbols.
 
+#if !defined(BCD_API)
 #if defined BCB_STATIC
     #define BCD_API
     #define BCD_INTERNAL
@@ -54,6 +55,7 @@
     #define BCD_API      BC_HELPER_DLL_IMPORT
     #define BCD_INTERNAL BC_HELPER_DLL_LOCAL
 #endif
+#endif
 
 // Log name.
 #define LOG_BLOCKCHAIN "blockchain"
