--- origsrc/libbitcoin-blockchain-3.4.0/Makefile.am	2017-12-13 11:16:25.000000000 +0900
+++ src/libbitcoin-blockchain-3.4.0/Makefile.am	2017-12-19 17:00:38.040999800 +0900
@@ -34,6 +34,7 @@ doc_DATA = \
 lib_LTLIBRARIES = src/libbitcoin-blockchain.la
 src_libbitcoin_blockchain_la_CPPFLAGS = -I${srcdir}/include ${bitcoin_database_CPPFLAGS} ${bitcoin_consensus_CPPFLAGS}
 src_libbitcoin_blockchain_la_LIBADD = ${bitcoin_database_LIBS} ${bitcoin_consensus_LIBS}
+src_libbitcoin_blockchain_la_LDFLAGS = -no-undefined
 src_libbitcoin_blockchain_la_SOURCES = \
     src/settings.cpp \
     src/interface/block_chain.cpp \
