--- origsrc/libbitcoin-blockchain-3.3.0/Makefile.am	2017-07-27 10:23:27.000000000 +0900
+++ src/libbitcoin-blockchain-3.3.0/Makefile.am	2017-09-18 13:11:58.767999800 +0900
@@ -34,6 +34,7 @@ doc_DATA = \
 lib_LTLIBRARIES = src/libbitcoin-blockchain.la
 src_libbitcoin_blockchain_la_CPPFLAGS = -I${srcdir}/include ${bitcoin_database_CPPFLAGS} ${bitcoin_consensus_CPPFLAGS}
 src_libbitcoin_blockchain_la_LIBADD = ${bitcoin_database_LIBS} ${bitcoin_consensus_LIBS}
+src_libbitcoin_blockchain_la_LDFLAGS = -no-undefined
 src_libbitcoin_blockchain_la_SOURCES = \
     src/settings.cpp \
     src/interface/block_chain.cpp \
@@ -78,7 +79,7 @@ endif WITH_TESTS
 #------------------------------------------------------------------------------
 if WITH_TOOLS
 
-noinst_PROGRAMS = tools/initchain/initchain
+bin_PROGRAMS = tools/initchain/initchain
 tools_initchain_initchain_CPPFLAGS = -I${srcdir}/include ${bitcoin_database_CPPFLAGS} ${bitcoin_consensus_CPPFLAGS}
 tools_initchain_initchain_LDADD = src/libbitcoin-blockchain.la ${bitcoin_database_LIBS} ${bitcoin_consensus_LIBS}
 tools_initchain_initchain_SOURCES = \
