--- origsrc/fgsl-1.2.0/Makefile.am	2017-01-21 22:34:07.000000000 +0900
+++ src/fgsl-1.2.0/Makefile.am	2017-08-11 10:02:45.226000000 +0900
@@ -1,6 +1,10 @@
-SUBDIRS = . tests doc
+SUBDIRS = . tests
+if INSTALL_DOC
+SUBDIRS += doc
+endif
 ACLOCAL_AMFLAGS = -I m4
 AM_CPPFLAGS = -I${top_srcdir} -I${top_builddir}
+AM_FCFLAGS = -ffree-line-length-none
 
 fgslincludedir=${includedir}/fgsl
 
--- origsrc/fgsl-1.2.0/configure.ac	2017-01-21 22:34:07.000000000 +0900
+++ src/fgsl-1.2.0/configure.ac	2017-08-11 09:43:08.897000000 +0900
@@ -92,7 +92,7 @@ if test "x$C_LOC_TARGET" = xno; then
 fi
 
 case "$host" in
-	*mingw*)
+	*mingw*|*cygwin*)
 		FGSL_LDFLAGS="-no-undefined"
 		AC_SUBST(FGSL_LDFLAGS)
 		;;
@@ -121,7 +121,8 @@ AC_LANG_PUSH(Fortran)
 AC_OPENMP
 AC_LANG_POP(Fortran)
 
-
+AC_ARG_ENABLE([doc], [install doc], [enable_doc=yes], [enable_doc=no])
+AM_CONDITIONAL([INSTALL_DOC], [test x"$enable_doc" = "xyes"])
 
 AC_CONFIG_FILES([Makefile tests/Makefile fgsl.pc fgsl.spec doc/Makefile doc/examples/Makefile])
 AC_CONFIG_HEADERS([config.h])
--- origsrc/fgsl-1.2.0/doc/examples/Makefile.am	2017-01-21 22:34:07.000000000 +0900
+++ src/fgsl-1.2.0/doc/examples/Makefile.am	2017-08-11 09:44:38.646000000 +0900
@@ -2,7 +2,7 @@ AM_CPPFLAGS = -I${top_srcdir} -I${top_bu
 AM_FCFLAGS = -I${top_srcdir} -I${top_builddir}
 LDADD = ${top_builddir}/libfgsl.la $(gsl_LIBS)
 
-fgsl_exampledir = $(datadir)/examples/fgsl
+fgsl_exampledir = $(datadir)/fgsl/examples
 
 
 
