--- origsrc/fgsl-0.9.4/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/fgsl-0.9.4/Makefile.am	2013-07-21 18:24:51.142875500 +0900
@@ -0,0 +1,12 @@
+
+SUBDIRS = . tests
+
+include_HEADERS = fgsl.mod
+
+lib_LTLIBRARIES = libfgsl.la
+
+libfgsl_la_CFLAGS = $(FGSL_CFLAGS)
+libfgsl_la_SOURCES = fgsl.f90 fgsl_utils.c
+libfgsl_la_LDFLAGS = -no-undefined
+libfgsl_la_LIBADD = $(FGSL_LIBS)
+
--- origsrc/fgsl-0.9.4/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/fgsl-0.9.4/configure.ac	2013-07-21 17:48:57.071586500 +0900
@@ -0,0 +1,30 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([fgsl], [0.9.4])
+AC_CONFIG_SRCDIR([checktypes.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_FC
+AC_PROG_INSTALL
+LT_INIT
+
+PKG_CHECK_MODULES([FGSL], [gsl])
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([strerror])
+
+AC_CONFIG_FILES([Makefile
+                 tests/Makefile])
+AC_OUTPUT
--- origsrc/fgsl-0.9.4/tests/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/fgsl-0.9.4/tests/Makefile.am	2013-07-21 18:24:33.884086500 +0900
@@ -0,0 +1,91 @@
+
+TESTS = runtests.sh
+
+check_PROGRAMS = poly interp specfunc permutation integration \
+	sort array linalg eigen fft rng histogram \
+	statistics ntuple siman ode deriv roots \
+	minim multiroots montecarlo bspline ieee_utils \
+	multimin fit nlfit version
+
+AM_FCFLAGS = -I$(top_builddir)
+
+poly_SOURCES = mod_unit.f90 poly.f90
+poly_LDADD = ../libfgsl.la
+
+interp_SOURCES = mod_unit.f90 interp.f90
+interp_LDADD = ../libfgsl.la
+
+specfunc_SOURCES = mod_unit.f90 specfunc.f90
+specfunc_LDADD = ../libfgsl.la
+
+permutation_SOURCES = mod_unit.f90 permutation.f90
+permutation_LDADD = ../libfgsl.la
+
+integration_SOURCES = mod_unit.f90 integration.f90
+integration_LDADD = ../libfgsl.la
+
+sort_SOURCES = mod_unit.f90 sort.f90
+sort_LDADD = ../libfgsl.la
+
+array_SOURCES = mod_unit.f90 array.f90
+array_LDADD = ../libfgsl.la
+
+linalg_SOURCES = mod_unit.f90 linalg.f90
+linalg_LDADD = ../libfgsl.la
+
+eigen_SOURCES = mod_unit.f90 eigen.f90
+eigen_LDADD = ../libfgsl.la
+
+fft_SOURCES = mod_unit.f90 fft.f90
+fft_LDADD = ../libfgsl.la
+
+rng_SOURCES = mod_unit.f90 rng.f90
+rng_LDADD = ../libfgsl.la
+
+histogram_SOURCES = mod_unit.f90 histogram.f90
+histogram_LDADD = ../libfgsl.la
+
+statistics_SOURCES = mod_unit.f90 statistics.f90
+statistics_LDADD = ../libfgsl.la
+
+ntuple_SOURCES = mod_unit.f90 ntuple.f90
+ntuple_LDADD = ../libfgsl.la
+
+siman_SOURCES = mod_unit.f90 siman.f90
+siman_LDADD = ../libfgsl.la
+
+ode_SOURCES = mod_unit.f90 ode.f90
+ode_LDADD = ../libfgsl.la
+
+deriv_SOURCES = mod_unit.f90 deriv.f90
+deriv_LDADD = ../libfgsl.la
+
+roots_SOURCES = mod_unit.f90 roots.f90
+roots_LDADD = ../libfgsl.la
+
+minim_SOURCES = mod_unit.f90 minim.f90
+minim_LDADD = ../libfgsl.la
+
+multiroots_SOURCES = mod_unit.f90 multiroots.f90
+multiroots_LDADD = ../libfgsl.la
+
+montecarlo_SOURCES = mod_unit.f90 montecarlo.f90
+montecarlo_LDADD = ../libfgsl.la
+
+bspline_SOURCES = mod_unit.f90 bspline.f90
+bspline_LDADD = ../libfgsl.la
+
+ieee_utils_SOURCES = mod_unit.f90 ieee_utils.f90
+ieee_utils_LDADD = ../libfgsl.la
+
+multimin_SOURCES = mod_unit.f90 multimin.f90
+multimin_LDADD = ../libfgsl.la
+
+fit_SOURCES = mod_unit.f90 fit.f90
+fit_LDADD = ../libfgsl.la
+
+nlfit_SOURCES = mod_unit.f90 nlfit.f90
+nlfit_LDADD = ../libfgsl.la
+
+version_SOURCES = mod_unit.f90 version.f90
+version_LDADD = ../libfgsl.la
