--- origsrc/fgsl-1.0.0/Makefile.am	2013-09-09 18:12:22.000000000 +0900
+++ src/fgsl-1.0.0/Makefile.am	2015-01-02 16:03:06.128790600 +0900
@@ -1,4 +1,7 @@
-SUBDIRS = . tests api interface doc
+SUBDIRS = . tests api interface
+if INSTALL_DOC
+SUBDIRS += doc
+endif
 ACLOCAL_AMFLAGS = -I m4
 AM_CPPFLAGS = -I${top_srcdir} -I${top_builddir}
 
--- origsrc/fgsl-1.0.0/configure.ac	2014-02-03 19:46:15.000000000 +0900
+++ src/fgsl-1.0.0/configure.ac	2015-01-02 15:56:56.644415600 +0900
@@ -69,7 +69,7 @@ if test "x$IEEE_EXCEPTIONS" = xyes ; the
 fi
 
 case "$host" in
-	*mingw*)
+	*mingw*|*cygwin*)
 		FGSL_LDFLAGS="-no-undefined"
 		AC_SUBST(FGSL_LDFLAGS)
 		;;
@@ -98,7 +98,8 @@ AC_LANG_PUSH(Fortran)
 AC_OPENMP
 AC_LANG_POP(Fortran)
 
-
+AC_ARG_ENABLE([doc], [install doc], [enable_doc=yes], [enable_doc=no])
+AM_CONDITIONAL([INSTALL_DOC], [test x"$enable_doc" = "xyes"])
 
 AC_CONFIG_FILES([Makefile api/Makefile interface/Makefile tests/Makefile fgsl.pc fgsl.spec doc/Makefile doc/examples/Makefile])
 AC_CONFIG_HEADERS([config.h])
--- origsrc/fgsl-1.0.0/doc/examples/Makefile.am	2013-09-09 18:12:22.000000000 +0900
+++ src/fgsl-1.0.0/doc/examples/Makefile.am	2015-01-02 15:53:34.232306300 +0900
@@ -2,7 +2,7 @@ AM_CPPFLAGS = -I${top_srcdir} -I${top_bu
 AM_FCFLAGS = -I${top_srcdir} -I${top_builddir}
 LDADD = ${top_builddir}/libfgsl.la
 
-fgsl_exampledir = $(datadir)/examples/fgsl
+fgsl_exampledir = $(datadir)/fgsl/examples
 
 
 
