--- origsrc/memcached-1.4.20/Makefile.am	2014-05-29 12:43:18.160020100 +0900
+++ src/memcached-1.4.20/Makefile.am	2014-05-29 12:44:51.370957600 +0900
@@ -80,8 +80,8 @@ EXTRA_DIST = doc scripts t memcached.spe
 MOSTLYCLEANFILES = *.gcov *.gcno *.gcda *.tcov
 
 test:	memcached-debug sizes testapp
-	$(srcdir)/sizes
-	$(srcdir)/testapp
+	$(builddir)/sizes
+	$(builddir)/testapp
 	prove $(srcdir)/t
 	@if test `basename $(PROFILER)` = "gcov"; then \
 	  for file in memcached_debug-*.gc??; do \
--- origsrc/memcached-1.4.20/configure.ac	2014-05-29 12:43:18.347520100 +0900
+++ src/memcached-1.4.20/configure.ac	2014-05-29 12:53:48.760606100 +0900
@@ -70,10 +70,6 @@ AC_DEFUN([DETECT_SUNCC],
 ])
 
 DETECT_SUNCC([CFLAGS="-mt $CFLAGS"], [])
-AS_IF([test "$ICC" = "yes" -o "$GCC" = "yes"],
-[
-    AS_IF(test "$CLANG" = "no",[CFLAGS="$CFLAGS -pthread"])
-])
 
 if test "$ICC" = "no"; then
    AC_PROG_CC_C99
@@ -176,7 +172,6 @@ AC_ARG_ENABLE(coverage,
 if test "x$enable_coverage" != "xno"; then
    if test "$GCC" = "yes" -a "$ICC" != "yes" -a "$CLANG" != "yes"
    then
-      CFLAGS="$CFLAGS -pthread"
       AC_PATH_PROG([PROFILER], [gcov], "no", [$PATH])
       if test "x$PROFILER" != "xno"; then
          # Issue 97: The existense of gcov doesn't mean we have -lgcov
@@ -583,7 +578,7 @@ then
 elif test "$GCC" = "yes"
 then
   GCC_VERSION=`$CC -dumpversion`
-  CFLAGS="$CFLAGS -Wall -Werror -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
+  CFLAGS="$CFLAGS -Wall -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
   case $GCC_VERSION in
     4.4.*)
     CFLAGS="$CFLAGS -fno-strict-aliasing"
--- origsrc/memcached-1.4.20/testapp.c	2014-05-29 12:43:18.472520100 +0900
+++ src/memcached-1.4.20/testapp.c	2014-05-29 12:46:06.976426400 +0900
@@ -155,7 +155,13 @@ static enum test_return cache_redzone_te
 
     /* Ignore SIGABORT */
     struct sigaction old_action;
+#if defined(__CYGWIN__)
+    struct sigaction action;
+    action.sa_handler = SIG_IGN;
+    action.sa_flags = 0;
+#else
     struct sigaction action = { .sa_handler = SIG_IGN, .sa_flags = 0};
+#endif
     sigemptyset(&action.sa_mask);
     sigaction(SIGABRT, &action, &old_action);
 
