--- origsrc/memcached-1.4.22/Makefile.am	2015-04-17 21:30:29.991568300 +0900
+++ src/memcached-1.4.22/Makefile.am	2015-04-17 21:34:36.671615800 +0900
@@ -80,8 +80,8 @@ EXTRA_DIST = doc scripts t memcached.spe
 MOSTLYCLEANFILES = *.gcov *.gcno *.gcda *.tcov
 
 test:	memcached-debug sizes testapp
-	$(srcdir)/sizes
-	$(srcdir)/testapp
+	$(builddir)/sizes
+	$(builddir)/testapp
 	prove $(srcdir)/t
 	@if test `basename $(PROFILER)` = "gcov"; then \
 	  for file in memcached_debug-*.gc??; do \
--- origsrc/memcached-1.4.22/configure.ac	2015-04-17 21:30:30.116568300 +0900
+++ src/memcached-1.4.22/configure.ac	2015-04-17 21:32:11.937887600 +0900
@@ -583,7 +583,7 @@ then
 elif test "$GCC" = "yes"
 then
   GCC_VERSION=`$CC -dumpversion`
-  CFLAGS="$CFLAGS -Wall -Werror -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
+  CFLAGS="$CFLAGS -Wall -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
   case $GCC_VERSION in
     4.4.*)
     CFLAGS="$CFLAGS -fno-strict-aliasing"
