--- origsrc/memcached-1.6.0/configure.ac	2020-03-11 18:14:44.557044300 +0900
+++ src/memcached-1.6.0/configure.ac	2020-03-11 18:14:49.228960000 +0900
@@ -760,7 +760,7 @@ then
 elif test "$GCC" = "yes"
 then
   GCC_VERSION=`$CC -dumpversion`
-  CFLAGS="$CFLAGS -Wall -Werror -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
+  CFLAGS="$CFLAGS -Wall -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
   if test "x$enable_asan" = "xyes"; then
     CFLAGS="$CFLAGS -fsanitize=address"
   fi
--- origsrc/memcached-1.6.0/extstore.c	2020-02-15 13:27:29.000000000 +0900
+++ src/memcached-1.6.0/extstore.c	2020-03-11 18:22:21.659647100 +0900
@@ -773,7 +773,7 @@ static void *extstore_io_thread(void *ar
                     }
                     pthread_mutex_unlock(&p->mutex);
                     if (do_op) {
-#ifdef __APPLE__
+#if defined(__APPLE__) || defined(__CYGWIN__)
                         ret = lseek(p->fd, SEEK_SET, p->offset + cur_io->offset);
                         if (ret >= 0) {
                             if (cur_io->iov == NULL) {
