--- origsrc/memcached-1.5.19/Makefile.am	2019-10-04 14:07:29.913666500 +0900
+++ src/memcached-1.5.19/Makefile.am	2019-10-04 14:07:34.835667400 +0900
@@ -132,8 +132,8 @@ test_basic_tls:
 endif
 
 test:	memcached-debug sizes testapp
-	$(srcdir)/sizes
-	$(srcdir)/testapp
+	$(builddir)/sizes
+	$(builddir)/testapp
 if ENABLE_TLS
 	@if test $(SSL_TEST)1 = 1; then \
           $(MAKE) SSL_TEST=1  test_basic_tls; \
--- origsrc/memcached-1.5.19/configure.ac	2019-10-04 14:07:30.382424000 +0900
+++ src/memcached-1.5.19/configure.ac	2019-10-04 14:07:34.866917000 +0900
@@ -787,7 +787,7 @@ then
 elif test "$GCC" = "yes"
 then
   GCC_VERSION=`$CC -dumpversion`
-  CFLAGS="$CFLAGS -Wall -Werror -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
+  CFLAGS="$CFLAGS -Wall -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
   case $GCC_VERSION in
     4.4.*)
     CFLAGS="$CFLAGS -fno-strict-aliasing"
