--- origsrc/memcached-1.4.17/Makefile.am	2014-01-26 15:16:49.735677600 +0900
+++ src/memcached-1.4.17/Makefile.am	2014-01-26 15:17:25.472005700 +0900
@@ -78,8 +78,8 @@ EXTRA_DIST = doc scripts t memcached.spe
 MOSTLYCLEANFILES = *.gcov *.gcno *.gcda *.tcov
 
 test:	memcached-debug sizes testapp
-	$(srcdir)/sizes
-	$(srcdir)/testapp
+	$(builddir)/sizes
+	$(builddir)/testapp
 	prove $(srcdir)/t
 	@if test `basename $(PROFILER)` = "gcov"; then \
 	  for file in memcached_debug-*.gc??; do \
--- origsrc/memcached-1.4.17/configure.ac	2014-01-26 15:16:49.876302600 +0900
+++ src/memcached-1.4.17/configure.ac	2014-01-26 15:17:51.134115100 +0900
@@ -574,7 +574,7 @@ then
 elif test "$GCC" = "yes"
 then
   GCC_VERSION=`$CC -dumpversion`
-  CFLAGS="$CFLAGS -Wall -Werror -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
+  CFLAGS="$CFLAGS -Wall -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
   case $GCC_VERSION in
     4.4.*)
     CFLAGS="$CFLAGS -fno-strict-aliasing"
--- origsrc/memcached-1.4.17/testapp.c	2014-01-26 15:16:49.720052600 +0900
+++ src/memcached-1.4.17/testapp.c	2014-01-26 15:18:29.571615100 +0900
@@ -155,7 +155,13 @@ static enum test_return cache_redzone_te
 
     /* Ignore SIGABORT */
     struct sigaction old_action;
+#if defined(__CYGWIN__)
+    struct sigaction action;
+    action.sa_handler = SIG_IGN;
+    action.sa_flags = 0;
+#else /* defined(__CYGWIN__) */
     struct sigaction action = { .sa_handler = SIG_IGN, .sa_flags = 0};
+#endif /* defined(__CYGWIN__) */
     sigemptyset(&action.sa_mask);
     sigaction(SIGABRT, &action, &old_action);
 
