--- origsrc/memcached-1.5.16/Makefile.am	2019-06-06 16:02:15.387833600 +0900
+++ src/memcached-1.5.16/Makefile.am	2019-06-06 16:03:38.683428300 +0900
@@ -131,8 +131,8 @@ test_basic_tls:
 endif
 
 test:	memcached-debug sizes testapp
-	$(srcdir)/sizes
-	$(srcdir)/testapp
+	$(builddir)/sizes
+	$(builddir)/testapp
 if ENABLE_TLS
 	@if test $(SSL_TEST)1 = 1; then \
           $(MAKE) SSL_TEST=1  test_basic_tls; \
--- origsrc/memcached-1.5.16/configure.ac	2019-06-06 16:02:16.091858600 +0900
+++ src/memcached-1.5.16/configure.ac	2019-06-06 16:04:21.171517500 +0900
@@ -772,7 +772,7 @@ then
 elif test "$GCC" = "yes"
 then
   GCC_VERSION=`$CC -dumpversion`
-  CFLAGS="$CFLAGS -Wall -Werror -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
+  CFLAGS="$CFLAGS -Wall -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
   case $GCC_VERSION in
     4.4.*)
     CFLAGS="$CFLAGS -fno-strict-aliasing"
