--- origsrc/memcached-1.4.6/configure.ac	2011-07-15 17:58:59.000000000 +0900
+++ src/memcached-1.4.6/configure.ac	2011-08-07 12:53:37.025131100 +0900
@@ -493,7 +493,7 @@ then
 elif test "$GCC" = "yes"
 then
   GCC_VERSION=`$CC -dumpversion`
-  CFLAGS="$CFLAGS -Wall -Werror -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
+  CFLAGS="$CFLAGS -Wall -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
   case $GCC_VERSION in
     4.4.*)
     CFLAGS="$CFLAGS -fno-strict-aliasing"
--- origsrc/memcached-1.4.6/testapp.c	2011-07-10 16:46:58.000000000 +0900
+++ src/memcached-1.4.6/testapp.c	2011-08-07 12:53:37.035145500 +0900
@@ -155,7 +155,13 @@ static enum test_return cache_redzone_te
 
     /* Ignore SIGABORT */
     struct sigaction old_action;
+#if defined(__CYGWIN__)
+    struct sigaction action;
+    action.sa_handler = SIG_IGN;
+    action.sa_flags = 0;
+#else /* defined(__CYGWIN__) */
     struct sigaction action = { .sa_handler = SIG_IGN, .sa_flags = 0};
+#endif /* defined(__CYGWIN__) */
     sigemptyset(&action.sa_mask);
     sigaction(SIGABRT, &action, &old_action);
 
