--- origsrc/memcached-1.5.21/configure.ac	2020-01-25 11:48:11.038856700 +0900
+++ src/memcached-1.5.21/configure.ac	2020-01-25 11:49:51.671428200 +0900
@@ -769,7 +769,7 @@ then
 elif test "$GCC" = "yes"
 then
   GCC_VERSION=`$CC -dumpversion`
-  CFLAGS="$CFLAGS -Wall -Werror -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
+  CFLAGS="$CFLAGS -Wall -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
   if test "x$enable_asan" = "xyes"; then
     CFLAGS="$CFLAGS -fsanitize=address"
   fi
