--- origsrc/memcached-1.5.2/Makefile.am	2017-10-04 21:23:08.386966300 +0900
+++ src/memcached-1.5.2/Makefile.am	2017-10-04 21:23:10.387149200 +0900
@@ -96,8 +96,8 @@ EXTRA_DIST = doc scripts t memcached.spe
 MOSTLYCLEANFILES = *.gcov *.gcno *.gcda *.tcov
 
 test:	memcached-debug sizes testapp
-	$(srcdir)/sizes
-	$(srcdir)/testapp
+	$(builddir)/sizes
+	$(builddir)/testapp
 	@if test -n "${PARALLEL}"; then \
 	  prove $(srcdir)/t -j ${PARALLEL}; \
 	else \
--- origsrc/memcached-1.5.2/configure.ac	2017-10-04 21:23:08.574488200 +0900
+++ src/memcached-1.5.2/configure.ac	2017-10-04 21:23:10.387149200 +0900
@@ -612,7 +612,7 @@ then
 elif test "$GCC" = "yes"
 then
   GCC_VERSION=`$CC -dumpversion`
-  CFLAGS="$CFLAGS -Wall -Werror -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
+  CFLAGS="$CFLAGS -Wall -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls"
   case $GCC_VERSION in
     4.4.*)
     CFLAGS="$CFLAGS -fno-strict-aliasing"
