--- origsrc/rudedatabase-4.1.0/Makefile.am	2008-01-09 07:48:53.000000000 +0900
+++ src/rudedatabase-4.1.0/Makefile.am	2010-12-29 23:14:35.897806900 +0900
@@ -28,7 +28,7 @@ librudedatabase_la_SOURCES = 	src/databa
 	src/implementation.cpp \
 	src/implementation.h
 
-librudedatabase_la_LDFLAGS = -version-info 4:1:0
+librudedatabase_la_LDFLAGS = -version-info 4:1:0 -no-undefined
 
 #  -------------------------------------
 librudedatabase_la_LIBADD = -lmysqlclient
--- origsrc/rudedatabase-4.1.0/configure.ac	2008-01-09 07:49:20.000000000 +0900
+++ src/rudedatabase-4.1.0/configure.ac	2010-12-29 23:14:23.880526900 +0900
@@ -121,7 +121,7 @@ fi
 # Checks for programs.
 AC_PROG_CXX
 AC_PROG_CC
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 # Checks for libraries.
 
