--- origsrc/rudedatabase-4.1.0/Makefile.am	2008-01-09 07:48:53.000000000 +0900
+++ src/rudedatabase-4.1.0/Makefile.am	2011-11-23 10:26:53.107131200 +0900
@@ -28,7 +28,7 @@ librudedatabase_la_SOURCES = 	src/databa
 	src/implementation.cpp \
 	src/implementation.h
 
-librudedatabase_la_LDFLAGS = -version-info 4:1:0
+librudedatabase_la_LDFLAGS = -version-info 4:1:0 -no-undefined
 
 #  -------------------------------------
 librudedatabase_la_LIBADD = -lmysqlclient
