--- origsrc/gravity-0.9.8/CMakeLists.txt	2026-08-05 15:22:24.000000000 +0000
+++ src/gravity-0.9.8/CMakeLists.txt	2026-08-18 13:39:18.794222300 +0000
@@ -16,3 +16,10 @@ add_subdirectory(src)
 if(BUILD_CLI)
     add_subdirectory(src/cli)
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/gravity.pc.in ${CMAKE_BINARY_DIR}/gravity.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/gravity.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/gravity-0.9.8/gravity.pc.in	1970-01-01 00:00:00.000000000 +0000
+++ src/gravity-0.9.8/gravity.pc.in	2026-08-18 13:39:18.796243700 +0000
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: gravity
+Description: Gravity Programming Language
+Version: @VERSION@
+Libs: -L${libdir} -lgravityapi
+Cflags: -I${includedir}
--- origsrc/gravity-0.9.8/src/CMakeLists.txt	2026-08-05 15:22:24.000000000 +0000
+++ src/gravity-0.9.8/src/CMakeLists.txt	2026-08-18 13:39:18.796243700 +0000
@@ -74,12 +74,16 @@ endif()
 if (BUILD_SHARED_LIBS)
     add_library(gravityapi SHARED ${SRC_FILES})
     target_compile_definitions(gravityapi PUBLIC BUILD_GRAVITY_API)
-endif ()
-add_library(gravityapi_s STATIC ${SRC_FILES})
+    if (DEFINED SOVERSION)
+        set_target_properties(gravityapi PROPERTIES SOVERSION ${SOVERSION})
+    endif()
+else()
+    add_library(gravityapi_s STATIC ${SRC_FILES})
+endif()
 
 # ----------------------------------------------------------------
 if (BUILD_SHARED_LIBS)
-    set(GRAVITY_TARGETS gravityapi gravityapi_s)
+    set(GRAVITY_TARGETS gravityapi)
 else()
     set(GRAVITY_TARGETS gravityapi_s)
 endif()
--- origsrc/gravity-0.9.8/src/cli/CMakeLists.txt	2026-08-05 15:22:24.000000000 +0000
+++ src/gravity-0.9.8/src/cli/CMakeLists.txt	2026-08-18 13:39:18.798259200 +0000
@@ -5,7 +5,7 @@ SET(GRAVITY_SRC gravity.c)
 
 # Executable
 add_executable(gravity ${GRAVITY_SRC})
-target_link_libraries(gravity gravityapi_s)
+target_link_libraries(gravity gravityapi)
 target_include_directories(gravity PUBLIC ${GRAVITY_INCLUDE_DIR})
 set_target_properties(gravity PROPERTIES
         RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin"
