--- origsrc/numdiff-5.9.0/configure.ac	2017-02-26 02:43:40.000000000 +0900
+++ src/numdiff-5.9.0/configure.ac	2022-02-10 07:09:38.578766000 +0900
@@ -99,17 +99,6 @@ dnl * Debian uses ginstall-info instead
 dnl * to install info files.
 AC_PATH_PROGS(INSTINFO,install-info ginstall-info)
 
-dnl --------------------------------------
-dnl Checks for header files
-dnl --------------------------------------
-dnl
-AC_HEADER_STDC
-if test $ac_cv_header_stdc = yes; then
- AC_MSG_RESULT(Ok! I have just found them.)
-else
-  AC_MSG_ERROR(Cannot find ANSI C header files!)
-fi
-
 AC_HEADER_STDBOOL
 AC_CHECK_HEADERS(fcntl.h locale.h stddef.h \
   stdlib.h string.h sys/file.h time.h unistd.h wchar.h wctype.h)
