--- origsrc/azure-storage-cpplite-0.3.0/CMakeLists.txt	2020-03-13 12:39:47.000000000 +0900
+++ src/azure-storage-cpplite-0.3.0/CMakeLists.txt	2023-10-31 05:40:27.596674900 +0900
@@ -130,6 +130,10 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMA
 
 add_library(azure-storage-lite ${AZURE_STORAGE_LITE_HEADER} ${AZURE_STORAGE_LITE_SOURCE})
 
+if (CYGWIN)
+  set_target_properties(azure-storage-lite PROPERTIES SOVERSION 0)
+endif()
+
 set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
 set(THREADS_PREFER_PTHREAD_FLAG TRUE)
 find_package(Threads REQUIRED)
@@ -173,7 +177,7 @@ target_link_libraries(azure-storage-lite
 if(MSVC)
   target_compile_options(azure-storage-lite PRIVATE /W4 /WX /MP)
 elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-  target_compile_options(azure-storage-lite PRIVATE -Wall -Wextra -Werror -pedantic)
+  target_compile_options(azure-storage-lite PRIVATE -Wall -Wextra -pedantic)
 endif()
 
 if(BUILD_ADLS)
@@ -193,8 +197,15 @@ set(AZURE_STORAGE_LITE_VERSION_MAJOR 0)
 set(AZURE_STORAGE_LITE_VERSION_MINOR 3)
 set(AZURE_STORAGE_LITE_VERSION_REVISION 0)
 
-install(DIRECTORY include/ DESTINATION include)
+install(DIRECTORY include/ DESTINATION include/azure-storage-cpplite)
 install(TARGETS azure-storage-lite
     ARCHIVE DESTINATION lib
     LIBRARY DESTINATION lib
     RUNTIME DESTINATION bin)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/azure-storage-cpplite.pc.in ${CMAKE_BINARY_DIR}/azure-storage-cpplite.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/azure-storage-cpplite.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/azure-storage-cpplite-0.3.0/azure-storage-cpplite.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/azure-storage-cpplite-0.3.0/azure-storage-cpplite.pc.in	2023-10-31 05:35:25.047181600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: azure-storage-cpplite
+Description: Lite version of C++ Client Library for Microsoft Azure Storage
+Version: @AZURE_STORAGE_LITE_VERSION_MAJOR@.@AZURE_STORAGE_LITE_VERSION_MINOR@.@AZURE_STORAGE_LITE_VERSION_REVISION@
+Libs: -L${libdir} -lazure-storage-lite
+Cflags: -I${includedir}/azure-storage-cpplite
--- origsrc/azure-storage-cpplite-0.3.0/src/base64.cpp	2020-03-13 12:39:47.000000000 +0900
+++ src/azure-storage-cpplite-0.3.0/src/base64.cpp	2023-10-31 05:45:59.687582200 +0900
@@ -1,5 +1,6 @@
 #include <array>
 #include <cstring>
+#include <stdexcept>
 
 #include "base64.h"
 
