--- origsrc/pcsc-lite-2.3.0/configure.ac	2024-08-03 23:39:59.000000000 +0900
+++ src/pcsc-lite-2.3.0/configure.ac	2024-09-15 12:57:49.710466700 +0900
@@ -3,7 +3,7 @@
 
 AC_PREREQ([2.69])
 
-AC_INIT([pcsc-lite], [2.2.0])
+AC_INIT([pcsc-lite], [2.3.0])
 AC_CONFIG_SRCDIR(src/pcscdaemon.c)
 AM_INIT_AUTOMAKE(1.8 dist-bzip2 no-dist-gzip foreign subdir-objects)
 AC_CONFIG_HEADERS([config.h])
--- origsrc/pcsc-lite-2.3.0/src/Makefile.am	2024-08-03 23:39:59.000000000 +0900
+++ src/pcsc-lite-2.3.0/src/Makefile.am	2024-09-15 02:19:16.474453100 +0900
@@ -7,7 +7,7 @@ lib_LTLIBRARIES = libpcsclite.la libpcsc
 check_LTLIBRARIES = libpcsclite_fake.la
 
 sbin_PROGRAMS = pcscd
-noinst_PROGRAMS = testpcsc pcsc-wirecheck pcsc-wirecheck-gen
+noinst_PROGRAMS = testpcsc # pcsc-wirecheck pcsc-wirecheck-gen
 
 if ENABLE_SERIAL
 SERIAL_CONFIG = configfile.l
@@ -24,7 +24,7 @@ libpcsclite_la_SOURCES = \
 	error.c \
 	libredirect.c \
 	sys_unix.c
-libpcsclite_la_LDFLAGS = -version-info 1:0:0
+libpcsclite_la_LDFLAGS = -no-undefined -version-info 1:0:0
 
 libpcsclite_real_la_SOURCES = \
 	debug.c \
@@ -33,7 +33,7 @@ libpcsclite_real_la_SOURCES = \
 	sys_unix.c \
 	utils.c \
 	winscard_msg.c
-libpcsclite_real_la_LDFLAGS = -version-info 1:0:0
+libpcsclite_real_la_LDFLAGS = -no-undefined -version-info 1:0:0
 libpcsclite_real_la_CFLAGS = $(PTHREAD_CFLAGS) -DLIBPCSCLITE -DSIMCLIST_NO_DUMPRESTORE
 libpcsclite_real_la_LIBADD = $(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
 
--- origsrc/pcsc-lite-2.3.0/src/spy/Makefile.am	2024-08-03 23:39:59.000000000 +0900
+++ src/pcsc-lite-2.3.0/src/spy/Makefile.am	2024-09-15 02:19:16.474453100 +0900
@@ -7,6 +7,7 @@ dist_doc_DATA = setup_spy.sh
 libpcscspy_la_SOURCES = \
 	libpcscspy.c \
 	../sys_unix.c
+libpcscspy_la_LDFLAGS = -no-undefined
 
 if ENABLE_DOC
 man_MANS = pcsc-spy.1
