--- origsrc/pcsc-lite-1.7.2/configure.in	2011-04-01 04:51:51.000000000 +0900
+++ src/pcsc-lite-1.7.2/configure.in	2011-05-29 03:08:27.437679700 +0900
@@ -85,7 +85,6 @@ CFLAGS="$saved_CFLAGS"
 # Add libtool support
 # Static lib is disabled by default. Use --enable-static if needed
 LT_INIT(disable-static)
-AM_PROG_LIBTOOL
 # Automatically update the libtool script if it becomes out-of-date
 #AC_SUBST(LIBTOOL_DEPS)
 
--- origsrc/pcsc-lite-1.7.2/src/Makefile.am	2011-02-26 02:15:21.000000000 +0900
+++ src/pcsc-lite-1.7.2/src/Makefile.am	2011-05-29 11:20:42.585734100 +0900
@@ -6,7 +6,7 @@ AM_CPPFLAGS = -I$(top_srcdir)/src/PCSC -
 lib_LTLIBRARIES = libpcsclite.la
 
 sbin_PROGRAMS = pcscd
-noinst_PROGRAMS = testpcsc pcsc-wirecheck pcsc-wirecheck-gen
+noinst_PROGRAMS = testpcsc pcsc-wirecheck-gen pcsc-wirecheck 
 
 if ENABLE_SERIAL
 SERIAL_CONFIG = configfile.l
@@ -30,7 +30,7 @@ libpcsclite_la_SOURCES = \
 	sys_unix.c \
 	utils.c \
 	winscard_msg.c
-libpcsclite_la_LDFLAGS = -version-info 1:0:0
+libpcsclite_la_LDFLAGS = -version-info 1:0:0 -no-undefined
 libpcsclite_la_CFLAGS = $(CFLAGS) $(PTHREAD_CFLAGS) -DLIBPCSCLITE -DSIMCLIST_NO_DUMPRESTORE
 libpcsclite_la_LIBADD = $(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
 
@@ -97,9 +97,11 @@ fix-rights: install-sbinPROGRAMS
 
 testpcsc_SOURCES = testpcsc.c
 testpcsc_LDADD = libpcsclite.la
+testpcsc_CPPFLAGS = $(AM_CPPFLAGS)
 
 pcsc_wirecheck_gen_SOURCES = \
 	pcsc-wirecheck-gen.c
+pcsc_wirecheck_gen_CPPFLAGS = $(AM_CPPFLAGS)
 
 # pcsc_wirecheck_gen_CPPFLAGS := $(LIBPCSCLITE_CFLAGS)
 
