--- origsrc/pcsc-lite-1.9.0/src/Makefile.am	2020-06-14 22:43:10.000000000 +0900
+++ src/pcsc-lite-1.9.0/src/Makefile.am	2020-06-16 22:30:09.515897000 +0900
@@ -6,7 +6,7 @@ AM_CPPFLAGS = -I$(top_srcdir)/src/PCSC -
 lib_LTLIBRARIES = libpcsclite.la
 
 sbin_PROGRAMS = pcscd
-noinst_PROGRAMS = testpcsc pcsc-wirecheck pcsc-wirecheck-gen
+noinst_PROGRAMS = testpcsc pcsc-wirecheck-gen pcsc-wirecheck
 
 if ENABLE_SERIAL
 SERIAL_CONFIG = configfile.l
@@ -28,7 +28,7 @@ libpcsclite_la_SOURCES = \
 	sys_unix.c \
 	utils.c \
 	winscard_msg.c
-libpcsclite_la_LDFLAGS = -version-info 1:0:0
+libpcsclite_la_LDFLAGS = -no-undefined -version-info 1:0:0
 libpcsclite_la_CFLAGS = $(CFLAGS) $(PTHREAD_CFLAGS) -DLIBPCSCLITE -DSIMCLIST_NO_DUMPRESTORE
 libpcsclite_la_LIBADD = $(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
 
--- origsrc/pcsc-lite-1.9.0/src/pcscd.h	2020-06-14 22:43:31.000000000 +0900
+++ src/pcsc-lite-1.9.0/src/pcscd.h	2020-06-16 22:34:14.754963900 +0900
@@ -42,7 +42,7 @@
 #define SCARD_INSERTED			0x0002	/**< Card was inserted */
 #define SCARD_REMOVED			0x0004	/**< Card was removed */
 
-#define PCSCLITE_CONFIG_DIR		"/usr/local/etc/reader.conf.d"
+#define PCSCLITE_CONFIG_DIR		"/etc/reader.conf.d"
 
 #define PCSCLITE_IPC_DIR		USE_IPCDIR
 #define PCSCLITE_RUN_PID		PCSCLITE_IPC_DIR "/pcscd.pid"
--- origsrc/pcsc-lite-1.9.0/src/spy/Makefile.am	2019-11-08 23:54:33.000000000 +0900
+++ src/pcsc-lite-1.9.0/src/spy/Makefile.am	2020-06-16 22:30:09.515897000 +0900
@@ -5,6 +5,7 @@ dist_bin_SCRIPTS = pcsc-spy
 
 libpcscspy_la_SOURCES = \
 	libpcscspy.c
+libpcscspy_la_LDFLAGS = -no-undefined
 
 if ENABLE_DOC
 man_MANS = pcsc-spy.1
