--- origsrc/libdrawtext-0.3/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libdrawtext-0.3/Makefile.am	2017-10-05 00:03:14.323732000 +0900
@@ -0,0 +1,16 @@
+AM_CPPFLAGS = -I$(top_srcdir)/src
+
+bin_PROGRAMS = font2glyphmap
+font2glyphmap_SOURCES = tools/font2glyphmap/src/font2glyphmap.c
+font2glyphmap_LDADD = libdrawtext.la
+
+pkginclude_HEADERS = src/drawtext.h
+
+lib_LTLIBRARIES = libdrawtext.la
+libdrawtext_la_CPPFLAGS = $(LIBDRAWTEXT_CFLAGS)
+libdrawtext_la_SOURCES = src/draw.c src/drawgl.c src/drawrast.c src/font.c src/utf8.c
+libdrawtext_la_LDFLAGS = -no-undefined
+libdrawtext_la_LIBADD = $(LIBDRAWTEXT_LIBS)
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libdrawtext-0.3/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/libdrawtext-0.3/configure.ac	2017-10-05 00:03:14.323732000 +0900
@@ -0,0 +1,30 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([libdrawtext], [0.3])
+AC_CONFIG_SRCDIR([examples/nofreetype/simple-noft.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_LN_S
+LT_INIT
+
+PKG_CHECK_MODULES([LIBDRAWTEXT], [freetype2 gl glu glew])
+
+# Checks for header files.
+AC_FUNC_ALLOCA
+AC_CHECK_HEADERS([float.h limits.h malloc.h stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([atexit clock_gettime memset strerror strrchr strtol])
+
+AC_CONFIG_FILES([libdrawtext.pc Makefile])
+AC_OUTPUT
--- origsrc/libdrawtext-0.3/libdrawtext.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libdrawtext-0.3/libdrawtext.pc.in	2017-10-05 00:03:14.289358400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libdrawtext
+Description: OpenGL text rendering library
+Version: @VERSION@
+Libs: -L${libdir} -ldrawtext
+Cflags: -I${includedir}
