--- origsrc/abPOA-1.5.3/CMakeLists.txt	2024-09-19 04:23:26.000000000 +0900
+++ src/abPOA-1.5.3/CMakeLists.txt	2024-11-13 08:03:07.797523000 +0900
@@ -29,7 +29,11 @@ add_library(abpoa
     src/simd_check.c
     src/utils.c)
 
-target_link_libraries(abpoa z pthread m)
+target_link_libraries(abpoa ${ZLIB_LIBRARIES} pthread m)
+
+if (DEFINED SOVERSION)
+    set_target_properties(abpoa PROPERTIES SOVERSION "${SOVERSION}")
+endif()
 
 add_executable(abpoa_bin
     src/abpoa.c)
@@ -42,7 +46,7 @@ target_include_directories(abpoa PUBLIC
     $<INSTALL_INTERFACE:src>)
 
 
-install(TARGETS abpoa DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS abpoa RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(TARGETS abpoa_bin DESTINATION ${CMAKE_INSTALL_BINDIR})
 install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} FILES_MATCHING PATTERN "*.h")
 
@@ -53,3 +57,6 @@ endif()
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/abpoa.pc.in ${CMAKE_CURRENT_BINARY_DIR}/abpoa-1.pc @ONLY)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/abpoa-1.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+
+add_executable(abpoa_example example.c)
+target_link_libraries(abpoa_example abpoa)
