--- origsrc/abPOA/CMakeLists.txt	2023-08-15 11:47:34.000000000 +0900
+++ src/abPOA/CMakeLists.txt	2023-08-15 11:53:36.039216400 +0900
@@ -29,10 +29,16 @@ add_library(abpoa
     src/simd_check.c
     src/utils.c)
 
+target_link_libraries(abpoa ${ZLIB_LIBRARY})
+
+if (DEFINED SOVERSION)
+    set_target_properties(abpoa PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 add_executable(abpoa_bin
     src/abpoa.c)
 
-target_link_libraries(abpoa_bin abpoa z pthread m)
+target_link_libraries(abpoa_bin abpoa pthread m)
 set_target_properties(abpoa_bin PROPERTIES OUTPUT_NAME abpoa)
 
 target_include_directories(abpoa PUBLIC
@@ -40,7 +46,7 @@ target_include_directories(abpoa PUBLIC
     $<INSTALL_INTERFACE:src>)
 
 
-install(TARGETS abpoa DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS abpoa RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(TARGETS abpoa_bin DESTINATION ${CMAKE_INSTALL_BINDIR})
 install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} FILES_MATCHING PATTERN "*.h")
 
