--- origsrc/oath-toolkit-2.4.1/libpskc/Makefile.am	2013-09-18 16:14:18.000000000 +0900
+++ src/oath-toolkit-2.4.1/libpskc/Makefile.am	2014-04-09 08:08:36.012385600 +0900
@@ -17,7 +17,7 @@
 
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
 
-SUBDIRS = gl schemas include . examples tests gtk-doc man
+SUBDIRS = gl schemas include . examples tests gtk-doc
 
 ACLOCAL_AMFLAGS = -I m4 -I gl/m4
 
--- origsrc/oath-toolkit-2.4.1/libpskc/global.c	2013-09-18 16:14:18.000000000 +0900
+++ src/oath-toolkit-2.4.1/libpskc/global.c	2014-04-09 08:08:36.012385600 +0900
@@ -30,6 +30,12 @@
 #include <xmlsec/xmlsec.h>
 #include <xmlsec/crypto.h>
 
+#ifdef XMLSEC_CRYPTO_DYNAMIC_LOADING
+#define QUOTE(str) #str
+#define STRINGIFY(str) QUOTE(str)
+#define STRINGIFIED_XMLSEC_CRYPTO STRINGIFY(XMLSEC_CRYPTO)
+#endif
+
 int _pskc_init = 0;
 
 /**
@@ -72,7 +78,7 @@ pskc_global_init (void)
     }
 
 #ifdef XMLSEC_CRYPTO_DYNAMIC_LOADING
-  if (xmlSecCryptoDLLoadLibrary (BAD_CAST XMLSEC_CRYPTO) < 0)
+  if (xmlSecCryptoDLLoadLibrary (BAD_CAST STRINGIFIED_XMLSEC_CRYPTO) < 0)
     {
       _pskc_debug ("xmlSecCryptoDLLoadLibrary failed");
       return PSKC_XMLSEC_ERROR;
