--- origsrc/oath-toolkit-2.6.7/libpskc/Makefile.am	2021-05-01 09:05:03.000000000 +0000
+++ src/oath-toolkit-2.6.7/libpskc/Makefile.am	2021-05-29 09:51:23.263203900 +0000
@@ -17,7 +17,7 @@
 
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
 
-SUBDIRS = gl schemas include . examples tests gtk-doc man
+SUBDIRS = gl schemas include . examples tests gtk-doc
 
 ACLOCAL_AMFLAGS = -I m4 -I gl/m4
 
--- origsrc/oath-toolkit-2.6.7/libpskc/global.c	2021-05-01 09:14:55.000000000 +0000
+++ src/oath-toolkit-2.6.7/libpskc/global.c	2021-05-29 09:51:23.263203900 +0000
@@ -32,6 +32,12 @@
 #include <xmlsec/crypto.h>
 #endif
 
+#ifdef XMLSEC_CRYPTO_DYNAMIC_LOADING
+#define QUOTE(str) #str
+#define STRINGIFY(str) QUOTE(str)
+#define STRINGIFIED_XMLSEC_CRYPTO STRINGIFY(XMLSEC_CRYPTO)
+#endif
+
 int _pskc_init = 0;
 
 /**
@@ -74,7 +80,7 @@ pskc_global_init (void)
     }
 
 #ifdef XMLSEC_CRYPTO_DYNAMIC_LOADING
-  if (xmlSecCryptoDLLoadLibrary (BAD_CAST XMLSEC_CRYPTO) < 0)
+  if (xmlSecCryptoDLLoadLibrary (BAD_CAST STRINGIFIED_XMLSEC_CRYPTO) < 0)
     {
       _pskc_debug ("xmlSecCryptoDLLoadLibrary failed");
       return PSKC_XMLSEC_ERROR;
