--- origsrc/hyphy-2.5.93/src/core/avllist.cpp	2025-12-24 04:21:54.000000000 +0900
+++ src/hyphy-2.5.93/src/core/avllist.cpp	2026-01-12 11:18:36.645999600 +0900
@@ -37,6 +37,7 @@
 
  */
 
+#include <alloca.h>
 #include <ctype.h>
 #include <limits.h>
 #include <math.h>
--- origsrc/hyphy-2.5.93/src/core/include/dataset_filter.h	2025-12-24 04:21:54.000000000 +0900
+++ src/hyphy-2.5.93/src/core/include/dataset_filter.h	2026-01-12 11:18:11.038918800 +0900
@@ -44,6 +44,8 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN
 #include "stdlib.h"
 #include "string_file_wrapper.h"
 
+#include <alloca.h>
+
 //_________________________________________________________
 enum _hy_dataset_filter_ambiguity_resolution {
   kAmbiguityHandlingResolve,
--- origsrc/hyphy-2.5.93/src/core/polynoml.cpp	2025-12-24 04:21:54.000000000 +0900
+++ src/hyphy-2.5.93/src/core/polynoml.cpp	2026-01-12 11:44:46.128970900 +0900
@@ -37,6 +37,7 @@
 
  */
 
+#include <alloca.h>
 #include <math.h>
 #include <string.h>
 
--- origsrc/hyphy-2.5.93/src/core/simplelist.cpp	2025-12-24 04:21:54.000000000 +0900
+++ src/hyphy-2.5.93/src/core/simplelist.cpp	2026-01-12 11:40:52.408465500 +0900
@@ -37,6 +37,7 @@
 
 */
 
+#include <alloca.h>
 #include <ctype.h>
 #include <limits.h>
 #include <math.h>
