--- origsrc/hyphy-2.5.96/src/core/avllist.cpp	2026-02-28 07:04:00.000000000 +0900
+++ src/hyphy-2.5.96/src/core/avllist.cpp	2026-03-20 10:41:22.371112100 +0900
@@ -37,6 +37,7 @@
 
  */
 
+#include <alloca.h>
 #include <ctype.h>
 #include <limits.h>
 #include <math.h>
--- origsrc/hyphy-2.5.96/src/core/include/dataset_filter.h	2026-02-28 07:04:00.000000000 +0900
+++ src/hyphy-2.5.96/src/core/include/dataset_filter.h	2026-03-20 10:41:22.373121800 +0900
@@ -44,6 +44,8 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN
 #include "stdlib.h"
 #include "string_file_wrapper.h"
 
+#include <alloca.h>
+
 //_________________________________________________________
 enum _hy_dataset_filter_ambiguity_resolution {
   kAmbiguityHandlingResolve,
--- origsrc/hyphy-2.5.96/src/core/polynoml.cpp	2026-02-28 07:04:00.000000000 +0900
+++ src/hyphy-2.5.96/src/core/polynoml.cpp	2026-03-20 10:41:22.373121800 +0900
@@ -37,6 +37,7 @@
 
  */
 
+#include <alloca.h>
 #include <math.h>
 #include <string.h>
 
--- origsrc/hyphy-2.5.96/src/core/simplelist.cpp	2026-02-28 07:04:00.000000000 +0900
+++ src/hyphy-2.5.96/src/core/simplelist.cpp	2026-03-20 10:41:22.375128400 +0900
@@ -37,6 +37,7 @@
 
 */
 
+#include <alloca.h>
 #include <ctype.h>
 #include <limits.h>
 #include <math.h>
