--- origsrc/xxl-1.0.1/Makefile.am	2004-10-14 01:34:52.000000000 +0900
+++ src/xxl-1.0.1/Makefile.am	2017-07-09 13:59:07.147268000 +0900
@@ -10,15 +10,18 @@ dist_libxxl_la_SOURCES		= xxl.c		\
 				  ixxl.h
 nodist_libxxl_la_SOURCES	= xxl.h
 nodist_include_HEADERS		= xxl.h
-libxxl_la_LDFLAGS		= -version-info $(XXL_VERSION_CURRENT):$(XXL_VERSION_REVISION):$(XXL_VERSION_AGE)
+libxxl_la_LDFLAGS		= -version-info $(XXL_VERSION_CURRENT):$(XXL_VERSION_REVISION):$(XXL_VERSION_AGE) -no-undefined
 
 check_PROGRAMS			= xxltest
 dist_xxltest_SOURCES		= xxltest.c
 nodist_xxltest_SOURCES		= xxl.h
-xxltest_LDADD			= -L. -lxxl
+xxltest_LDADD			= $(top_builddir)/libxxl.la
 TESTS				= xxltest
 
 distclean-local:
 	rm -rf autom4te.cache
 
 # vim: noet sts=0 sw=8 ts=8
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/xxl-1.0.1/configure.ac	2005-01-31 02:42:38.000000000 +0900
+++ src/xxl-1.0.1/configure.ac	2017-07-09 13:59:07.131688800 +0900
@@ -1,10 +1,10 @@
 AC_PREREQ(2.57)
 AC_COPYRIGHT([(c) Copyright 2003-2005 Matt Messier, John Viega])
 AC_INIT([xxl], [1.0.1], [xxl@zork.org])
+AC_CONFIG_AUX_DIR(.)
 AM_INIT_AUTOMAKE([1.7 foreign])
 AM_CONFIG_HEADER([config.h xxl.h])
 AC_CONFIG_SRCDIR([xxl.h.in])
-AC_CONFIG_AUX_DIR(.)
 AC_PREFIX_DEFAULT([/usr/local])
 
 XXL_VERSION_CURRENT=2
@@ -50,5 +50,5 @@ if test "$GCC" = "yes"; then
     CFLAGS="$CFLAGS -Wall"
 fi
 
-AC_CONFIG_FILES([Makefile], [chmod -w Makefile])
+AC_CONFIG_FILES([xxl.pc Makefile], [chmod -w Makefile])
 AC_OUTPUT
--- origsrc/xxl-1.0.1/xxl.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/xxl-1.0.1/xxl.pc.in	2017-07-09 13:59:07.116989300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: xxl
+Description: Library for C and C++ that provides exception handling and asset management
+Version: @VERSION@
+Libs: -L${libdir} -lxxl
+Cflags: -I${includedir}
