--- origsrc/xxl-1.0.1/Makefile.am	2004-10-14 01:34:52.000000000 +0900
+++ src/xxl-1.0.1/Makefile.am	2014-12-05 08:40:35.331483300 +0900
@@ -10,12 +10,12 @@ dist_libxxl_la_SOURCES		= xxl.c		\
 				  ixxl.h
 nodist_libxxl_la_SOURCES	= xxl.h
 nodist_include_HEADERS		= xxl.h
-libxxl_la_LDFLAGS		= -version-info $(XXL_VERSION_CURRENT):$(XXL_VERSION_REVISION):$(XXL_VERSION_AGE)
+libxxl_la_LDFLAGS		= -version-info $(XXL_VERSION_CURRENT):$(XXL_VERSION_REVISION):$(XXL_VERSION_AGE) -no-undefined
 
 check_PROGRAMS			= xxltest
 dist_xxltest_SOURCES		= xxltest.c
 nodist_xxltest_SOURCES		= xxl.h
-xxltest_LDADD			= -L. -lxxl
+xxltest_LDADD			= $(top_builddir)/libxxl.la
 TESTS				= xxltest
 
 distclean-local:
--- origsrc/xxl-1.0.1/configure.ac	2005-01-31 02:42:38.000000000 +0900
+++ src/xxl-1.0.1/configure.ac	2014-12-05 08:40:35.347108300 +0900
@@ -1,10 +1,10 @@
 AC_PREREQ(2.57)
 AC_COPYRIGHT([(c) Copyright 2003-2005 Matt Messier, John Viega])
 AC_INIT([xxl], [1.0.1], [xxl@zork.org])
+AC_CONFIG_AUX_DIR(.)
 AM_INIT_AUTOMAKE([1.7 foreign])
 AM_CONFIG_HEADER([config.h xxl.h])
 AC_CONFIG_SRCDIR([xxl.h.in])
-AC_CONFIG_AUX_DIR(.)
 AC_PREFIX_DEFAULT([/usr/local])
 
 XXL_VERSION_CURRENT=2
