--- origsrc/nativefiledialog-extended-1.1.1/CMakeLists.txt	2023-11-22 01:05:54.000000000 +0900
+++ src/nativefiledialog-extended-1.1.1/CMakeLists.txt	2023-12-29 19:22:36.450289400 +0900
@@ -51,3 +51,10 @@ add_subdirectory(src)
 if(${NFD_BUILD_TESTS})
   add_subdirectory(test)
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/nativefiledialog-extended.pc.in ${CMAKE_BINARY_DIR}/nativefiledialog-extended.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/nativefiledialog-extended.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/nativefiledialog-extended-1.1.1/nativefiledialog-extended.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/nativefiledialog-extended-1.1.1/nativefiledialog-extended.pc.in	2023-12-29 19:22:36.450289400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: nativefiledialog-extended
+Description: Cross platform (Windows, Mac, Linux) native file dialog library
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lnfd
+Cflags: -I${includedir}
