--- origsrc/playerctl-0.5.0/playerctl/Makefile.am	2016-03-21 02:39:55.000000000 +0900
+++ src/playerctl-0.5.0/playerctl/Makefile.am	2017-11-30 23:45:30.668420000 +0900
@@ -11,6 +11,8 @@ lib_LTLIBRARIES =
 bin_PROGRAMS =
 
 AM_CPPFLAGS += \
+	-I$(top_srcdir) \
+	-I$(top_builddir) \
 	-DPLAYERCTL_COMPILATION \
 	$(PLAYERCTL_CFLAGS) \
 	$(NULL)
@@ -30,7 +32,7 @@ source_c = \
 DISTCLEANFILES += playerctl-version.h
 
 playerctl-generated.c:
-	$(GDBUS_CODEGEN) --generate-c-code playerctl-generated mpris-dbus-interface.xml
+	$(GDBUS_CODEGEN) --generate-c-code playerctl-generated $(srcdir)/mpris-dbus-interface.xml
 
 BUILT_SOURCES += \
 	playerctl-generated.c \
@@ -42,11 +44,12 @@ CLEANFILES += $(BUILT_SOURCES)
 lib_LTLIBRARIES += libplayerctl-1.0.la
 libplayerctl_1_0_la_LIBADD = $(PLAYERCTL_LIBS)
 libplayerctl_1_0_la_SOURCES = $(source_c) $(source_h) $(BUILT_SOURCES)
+libplayerctl_1_0_la_LDFLAGS = -no-undefined
 
 playerctlincludedir = $(includedir)/playerctl
 playerctlinclude_DATA = \
 	$(source_h) \
-	$(top_srcdir)/playerctl/playerctl-version.h \
+	$(top_builddir)/playerctl/playerctl-version.h \
 	$(top_srcdir)/playerctl/playerctl.h \
 	$(NULL)
 
