--- origsrc/playerctl-2.0.2/playerctl/meson.build	2019-03-27 10:48:21.000000000 +0900
+++ src/playerctl-2.0.2/playerctl/meson.build	2019-05-05 09:09:06.355484200 +0900
@@ -47,7 +47,7 @@ deps = [
   gio_dep,
 ]
 
-playerctl_lib = both_libraries(
+playerctl_shared_lib = shared_library(
   'playerctl',
   playerctl_sources, enums,
   dependencies: deps,
@@ -58,7 +58,7 @@ playerctl_lib = both_libraries(
 
 # Required for linking against the shared lib we just built
 playerctl_shared_link = declare_dependency(
-  link_with: playerctl_lib.get_shared_lib(),
+  link_with: playerctl_shared_lib,
   dependencies: deps,
 )
 
@@ -85,7 +85,7 @@ if get_option('introspection')
   endif
 
   gnome.generate_gir(
-    playerctl_lib.get_shared_lib(),
+    playerctl_shared_lib,
     sources: [
       enums,
       'playerctl-player-name.c',
@@ -104,7 +104,7 @@ if get_option('introspection')
 endif
 
 pkgconfig.generate(
-  libraries: playerctl_lib.get_shared_lib(),
+  libraries: playerctl_shared_lib,
   subdirs: 'playerctl',
   version: meson.project_version(),
   name: 'Playerctl',
