diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/docopt.cpp/CMakeLists.txt src/docopt.cpp/CMakeLists.txt
--- origsrc/docopt.cpp/CMakeLists.txt	2015-09-08 23:17:02.000000000 +0900
+++ src/docopt.cpp/CMakeLists.txt	2015-09-08 23:19:31.098104600 +0900
@@ -53,7 +53,9 @@ endif()
 
 INSTALL(TARGETS
 	docopt
-	DESTINATION lib)
+	RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION lib)
+SET_TARGET_PROPERTIES(docopt PROPERTIES SOVERSION 0)
 if(WITH_STATIC)
 	INSTALL(TARGETS
 		docopt_s
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/docopt.cpp/docopt.cpp src/docopt.cpp/docopt.cpp
--- origsrc/docopt.cpp/docopt.cpp	2015-09-08 23:17:02.000000000 +0900
+++ src/docopt.cpp/docopt.cpp	2015-09-08 23:22:46.477857700 +0900
@@ -21,6 +21,8 @@
 #include <iostream>
 #include <cassert>
 
+#include <boost/lexical_cast.hpp>
+
 using namespace docopt;
 
 DocoptExitHelp::DocoptExitHelp()
@@ -690,7 +692,7 @@ PatternList parse_short(Tokens& tokens,
 		
 		if (similar.size() > 1) {
 			std::string error = shortOpt + " is specified ambiguously "
-			+ std::to_string(similar.size()) + " times";
+			+ boost::lexical_cast<std::string>(similar.size()) + " times";
 			throw Tokens::OptionError(std::move(error));
 		} else if (similar.empty()) {
 			options.emplace_back(shortOpt, "", 0);
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/docopt.cpp/run_testcase.cpp src/docopt.cpp/run_testcase.cpp
--- origsrc/docopt.cpp/run_testcase.cpp	2015-09-08 23:17:03.000000000 +0900
+++ src/docopt.cpp/run_testcase.cpp	2015-09-08 23:23:44.607270900 +0900
@@ -10,6 +10,8 @@
 
 #include <iostream>
 
+#include <cstdlib>
+
 int main(int argc, const char** argv)
 {
 	if (argc < 2) {
