--- origsrc/docopt.cpp-0.6.2/CMakeLists.txt	2016-12-03 04:12:44.000000000 +0900
+++ src/docopt.cpp-0.6.2/CMakeLists.txt	2016-12-21 17:31:22.547949000 +0900
@@ -9,6 +9,7 @@ include(GNUInstallDirs)
 option(WITH_TESTS "Build tests." OFF)
 option(WITH_EXAMPLE "Build example." OFF)
 option(USE_BOOST_REGEX "Replace std::regex with Boost.Regex" OFF)
+option(WITH_STATIC "Build static." OFF)
 
 #============================================================================
 # Internal compiler options
@@ -118,10 +119,13 @@ endif()
 set(export_name "docopt-targets")
 
 # Runtime package
-install(TARGETS docopt EXPORT ${export_name} DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS docopt EXPORT ${export_name} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+set_target_properties(docopt PROPERTIES SOVERSION 0)
 
 # Development package
+if(WITH_STATIC)
 install(TARGETS docopt_s EXPORT ${export_name} DESTINATION ${CMAKE_INSTALL_LIBDIR})
+endif()
 install(FILES ${docopt_HEADERS} DESTINATION include/docopt)
 
 # CMake Package
