--- origsrc/cvechecker-3.9/configure.ac	2018-09-09 19:32:47.000000000 +0900
+++ src/cvechecker-3.9/configure.ac	2019-01-14 20:33:42.718713200 +0900
@@ -35,6 +35,8 @@ AC_CHECK_HEADERS([argp.h],,[AC_MSG_ERROR
 # Checks for library functions.
 AC_CHECK_FUNCS([gethostname memset regcomp strchr strrchr strstr])
 
+AC_CHECK_LIB([argp], [main], [LIBS="$LIBS -largp"])
+
 # Checks for typedefs, structures, and compiler characteristics.
 AC_TYPE_SIZE_T
 
