--- origsrc/libxls/src/Makefile.am	2012-03-25 01:54:51.000000000 +0900
+++ src/libxls/src/Makefile.am	2012-03-29 20:41:07.702473400 +0900
@@ -5,8 +5,7 @@ bin_PROGRAMS = xls2csv
 lib_LTLIBRARIES = libxlsreader.la
 
 xls2csv_SOURCES =	\
-	xls2csv.c		\
-	getopt.c
+	xls2csv.c
 
 xls2csv_LDADD = libxlsreader.la
 
@@ -15,5 +14,5 @@ libxlsreader_la_SOURCES =	\
 	ole.c					\
 	xls.c
 
-libxlsreader_la_LDFLAGS = -version-info $(VERSION_INFO) $(LIBXLS_LIBS)
+libxlsreader_la_LDFLAGS = -no-undefined -version-info $(VERSION_INFO) $(LIBXLS_LIBS)
 
