diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x conf.h origsrc/lamarc-2.1.10/Makefile.am src/lamarc-2.1.10/Makefile.am
--- origsrc/lamarc-2.1.10/Makefile.am	2015-07-29 05:47:53.000000000 +0900
+++ src/lamarc-2.1.10/Makefile.am	2016-04-29 19:45:37.986373600 +0900
@@ -69,18 +69,6 @@ lam_conv_rc.o   :       $(top_srcdir)/re
 ## to this list
 LDADD				=
 
-noinst_LIBRARIES		= libtinyxml.a
-
-libtinyxml_a_CXXFLAGS	= -DTIXML_USE_STL
-
-libtinyxml_a_SOURCES	= \
-			tinyxml/tinyxml.h \
-			tinyxml/tinyxml.cpp \
-			tinyxml/tinyxmlerror.cpp \
-			tinyxml/tinyxmlparser.cpp \
-			tinyxml/tinystr.h \
-			tinyxml/tinystr.cpp
-
 ## ############################################################
 ## setting up compiler flags
 ## ############################################################
@@ -110,10 +98,7 @@ lamarc_includes         = \
 			-I $(lamarc_srcdir)/ui_util \
 			-I $(lamarc_srcdir)/ui_vars \
 			-I $(lamarc_srcdir)/xml \
-			-I $(top_srcdir)/boost \
-			-I $(top_srcdir)/resources \
-			-I $(top_srcdir)/tinyxml
-
+			-I $(top_srcdir)/resources
 
 ## include resource file if needed
 lamarc_resources                =
@@ -604,14 +589,6 @@ xml_sources     =   \
 		    src/xml/toxml.cpp \
 		    $(xml_core_sources)
 
-tinyxml_sources =   \
-		    tinyxml/tinyxml.h \
-		    tinyxml/tinyxml.cpp \
-		    tinyxml/tinyxmlerror.cpp \
-		    tinyxml/tinyxmlparser.cpp \
-		    tinyxml/tinystr.h \
-		    tinyxml/tinystr.cpp
-
 config_sources  =   \
 		    config/conf.h
 
@@ -634,7 +611,7 @@ lamarc_SOURCES  =   $(lamarcgcov_sources
 					$(lamarc_resources) \
 					$(config_sources)
 
-lamarc_LDADD	= libtinyxml.a
+lamarc_LDADD	= -ltinyxml
 
 old_lam_conv_core       = \
 			src/conversion/ConverterIf.cpp \
@@ -942,9 +919,9 @@ lam_conv_SOURCES        = \
 			$(lam_conv_core) \
 			$(lam_conv_resources)
 
-lam_conv_LDADD = @WX_LIBS@ -lX11 libtinyxml.a
+lam_conv_LDADD = @WX_LIBS@ -lX11 -ltinyxml
 
-EXTRA_lam_conv_SOURCES  = $(tinyxml_suite) $(lam_conv_icons)
+EXTRA_lam_conv_SOURCES  = $(lam_conv_icons)
 
 batch_lam_conv_core     = \
 			src/guiconv/batchconverter.h \
@@ -961,9 +938,9 @@ batch_lam_conv_testable = \
 batch_lam_conv_SOURCES	=	\
 			$(batch_lam_conv_testable)
 
-batch_lam_conv_LDADD = @WX_LIBS@ -lX11 libtinyxml.a
+batch_lam_conv_LDADD = @WX_LIBS@ -lX11 -ltinyxml
 
-EXTRA_batch_lam_conv_SOURCES = $(tinyxml_suite)
+EXTRA_batch_lam_conv_SOURCES =
 
 ## ############################################################
 ## To make OSX executables for wxWindows stuff
@@ -1028,7 +1005,7 @@ lamarc.app:     lamarc
 ## html documentation
 ## ############################################################
 
-htmldir = $(prefix)/html/lamarc
+htmldir = $(docdir)/html/
 
 html_DATA = \
 			doc/html/bayes.html \
@@ -1049,7 +1026,6 @@ html_DATA = \
 			doc/html/genotype.html \
 			doc/html/glossary.html \
 			doc/html/growthmenu.html \
-			doc/html/forces.html \
 			doc/html/index.html \
 			doc/html/insumfile.2reg3rep.html \
 			doc/html/insumfile.2reg3rep.xml \
@@ -1079,7 +1055,7 @@ html_DATA = \
 			doc/html/viral_data.html \
 			doc/html/xmlinput.html
 
-htmlimgdir = $(prefix)/html/lamarc/images
+htmlimgdir = $(docdir)/html/images
 
 htmlimg_DATA = \
 			doc/html/images/LamarcAnalysisScreen.png \
@@ -1104,7 +1080,7 @@ htmlimg_DATA = \
 			doc/html/images/uncorrelated.gif \
 			doc/html/images/variably_correlated.gif
 
-htmlbconvdir = $(prefix)/html/lamarc/batch_converter
+htmlbconvdir = $(docdir)/html/batch_converter
 
 htmlbconv_DATA = \
 			doc/html/batch_converter/README.txt \
@@ -1121,7 +1097,7 @@ htmlbconv_DATA = \
 			doc/html/batch_converter/sample-conv-cmd.html \
 			doc/html/batch_converter/sample-conv-cmd.xml
 
-htmlbconvimgdir = $(prefix)/html/lamarc/batch_converter/images
+htmlbconvimgdir = $(docdir)/html/batch_converter/images
 
 htmlbconvimg_DATA = \
 			doc/html/batch_converter/images/CombinedPanels.png \
@@ -1154,7 +1130,7 @@ htmlbconvimg_DATA = \
 			doc/html/batch_converter/images/lam_conv_export_file_mac_expanded.png
 
 
-htmltraitdir = $(prefix)/html/lamarc/trait_mapping
+htmltraitdir = $(docdir)/html/trait_mapping
 
 htmltrait_DATA  = \
 			doc/html/trait_mapping/README.txt \
@@ -1317,4 +1293,3 @@ distclean-local:
 clean-local:
 	rm -f *.gcov *.gcno *.gcda
 	rm -f -r *.app
-
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x conf.h origsrc/lamarc-2.1.10/configure.ac src/lamarc-2.1.10/configure.ac
--- origsrc/lamarc-2.1.10/configure.ac	2015-07-29 05:47:29.000000000 +0900
+++ src/lamarc-2.1.10/configure.ac	2016-04-29 19:29:05.445120300 +0900
@@ -66,7 +66,7 @@ case "$target_os" in
             LAMARC_CODE_OSTYPE=LAMARC_COMPILE_MACOSX
         fi
         ;;
-    freebsd*|linux*)
+    freebsd*|linux*|cygwin*)
         if test "X${lamCONV}" = "Xyes" ; then
             if test "X${lamGUI}" = "Xyes" ; then
                 if test "X${GUI_TOOLKIT}" = "X" ; then
@@ -373,9 +373,10 @@ if test X$user_set_cxxflags != Xyes; the
 fi
 
 # so we can build tinyxml as a library
-AC_PROG_RANLIB
 
 # Checks for libraries.
+BOOST_REQUIRE
+BOOST_SMART_PTR
 
 # Checks for header files.
 AC_HEADER_STDC
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x conf.h origsrc/lamarc-2.1.10/src/control/types.h src/lamarc-2.1.10/src/control/types.h
--- origsrc/lamarc-2.1.10/src/control/types.h	2012-06-30 10:32:39.000000000 +0900
+++ src/lamarc-2.1.10/src/control/types.h	2016-04-29 19:29:38.329893400 +0900
@@ -17,7 +17,7 @@
 #include <memory>
 
 #include "vectorx.h"
-#include "shared_ptr.hpp"
+#include <boost/shared_ptr.hpp>
 
 class Branch;
 class DataType;
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x conf.h origsrc/lamarc-2.1.10/src/force/event.cpp src/lamarc-2.1.10/src/force/event.cpp
--- origsrc/lamarc-2.1.10/src/force/event.cpp	2013-10-26 02:01:34.000000000 +0900
+++ src/lamarc-2.1.10/src/force/event.cpp	2016-04-29 19:30:07.026870300 +0900
@@ -24,7 +24,7 @@
 #include "mathx.h"
 #include "range.h"                      // For Link-related typedefs and constants.
 #include "registry.h"
-#include "shared_ptr.hpp"
+#include <boost/shared_ptr.hpp>
 #include "stringx.h"
 #include "timemanager.h"                // for use in Picktime for coals and disease
 #include "tree.h"
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x conf.h origsrc/lamarc-2.1.10/src/tree/branch.h src/lamarc-2.1.10/src/tree/branch.h
--- origsrc/lamarc-2.1.10/src/tree/branch.h	2012-06-30 10:32:42.000000000 +0900
+++ src/lamarc-2.1.10/src/tree/branch.h	2016-04-29 19:31:42.071321500 +0900
@@ -50,8 +50,8 @@
 #include "range.h"                      // For Link-related typedefs and constants.
 #include "dlcell.h"
 #include "locuscell.h"
-#include "shared_ptr.hpp"               // for Branch_ptr (boost::shared_ptr)
-#include "enable_shared_from_this.hpp"  // for shared_ptr support
+#include <boost/shared_ptr.hpp>               // for Branch_ptr (boost::shared_ptr)
+#include <boost/enable_shared_from_this.hpp>  // for shared_ptr support
 #include "branchtag.h"
 #include "rangex.h"                     // for class Range (and subclass) factory duties
 #include "fc_status.h"                  // for RevalidateRange debug function
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x conf.h origsrc/lamarc-2.1.10/src/tree/treesum.cpp src/lamarc-2.1.10/src/tree/treesum.cpp
--- origsrc/lamarc-2.1.10/src/tree/treesum.cpp	2013-12-10 07:48:52.000000000 +0900
+++ src/lamarc-2.1.10/src/tree/treesum.cpp	2016-04-29 19:32:04.039047300 +0900
@@ -20,7 +20,7 @@
 
 #include "chainmanager.h"
 #include "constants.h"
-#include "enable_shared_from_this.hpp"  // For shared_from_this in RecTreeSummary handling of
+#include <boost/enable_shared_from_this.hpp>  // For shared_from_this in RecTreeSummary handling of
                                         // recombination events for partnerpicks logic.
 #include "force.h"                      // For setting up the recsbypartition vector of intervals
                                         // in a RecTreeSummary.
