--- origsrc/roswell-21.05.14.109/configure.ac	2021-05-12 13:15:11.000000000 +0000
+++ src/roswell-21.05.14.109/configure.ac	2021-05-31 12:47:01.883438700 +0000
@@ -11,7 +11,7 @@ AC_CONFIG_FILES([Makefile src/Makefile])
 AC_CONFIG_SRCDIR([Makefile.am])
 AM_INIT_AUTOMAKE([foreign])
 AM_MAINTAINER_MODE([enable])
-
+AC_CANONICAL_HOST
 
 # Checks for programs.
 AC_PROG_CC
@@ -46,8 +46,9 @@ fi
 
 
 
-
-AC_CHECK_LIB([wininet], [main],[LIBS="-lwininet $LIBS";wwwlib=yes],[])
+if test "$host_os" != "cygwin"; then
+  AC_CHECK_LIB([wininet], [main],[LIBS="-lwininet $LIBS";wwwlib=yes],[])
+fi
 if test "x$wwwlib" = xno; then
   AC_CHECK_LIB([curl], [curl_global_init],[LIBS="-lcurl $LIBS";wwwlib=yes],[])
   AC_CHECK_HEADERS(curl/curl.h)
@@ -59,7 +60,9 @@ fi
 
 AC_CHECK_HEADERS(gnu/libc-version.h)
 
-AC_CHECK_HEADERS(windows.h,[with_windows=true])
+if test "$host_os" != "cygwin"; then
+  AC_CHECK_HEADERS(windows.h,[with_windows=true])
+fi
 AM_CONDITIONAL([WITH_WINDOWS], [test "$with_windows" != ""])
 AC_CHECK_TOOL([WINDRES], [windres], [])
 AM_CONDITIONAL([WITH_WIN_ICON], [test "$WINDRES" != ""])
--- origsrc/roswell-21.05.14.109/src/download.c	2021-05-12 13:15:11.000000000 +0000
+++ src/roswell-21.05.14.109/src/download.c	2021-05-31 12:47:01.883438700 +0000
@@ -1,3 +1,5 @@
+#include <curl/curl.h>
+
 #include "opt.h"
 
 int download_count=0;
