--- origsrc/lttoolbox-3.2.0/configure.ac	2010-03-15 00:14:34.000000000 +0900
+++ src/lttoolbox-3.2.0/configure.ac	2011-03-17 22:16:21.291362700 +0900
@@ -112,7 +112,7 @@ then
 fi
 
 AC_PROG_CXX
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AM_SANITY_CHECK
 AC_LANG_CPLUSPLUS
 
--- origsrc/lttoolbox-3.2.0/lttoolbox/Makefile.am	2009-12-10 22:03:57.000000000 +0900
+++ src/lttoolbox-3.2.0/lttoolbox/Makefile.am	2011-03-17 22:54:11.575871500 +0900
@@ -18,7 +18,7 @@ instdir = lttoolbox
 
 lib_LTLIBRARIES= liblttoolbox3.la
 liblttoolbox3_la_SOURCES= $(h_sources) $(cc_sources)
-liblttoolbox3_la_LDFLAGS= -version-info $(GENERIC_LIBRARY_VERSION) -release $(GENERIC_RELEASE)
+liblttoolbox3_la_LDFLAGS= -version-info $(GENERIC_LIBRARY_VERSION) -release $(GENERIC_RELEASE) -no-undefined
 
 lttoolboxdir = $(prefix)/share/lttoolbox
 lttoolboxinclude = $(prefix)/include
@@ -48,7 +48,7 @@ lt_tmxproc_LDFLAGS = -llttoolbox$(GENERI
 
 man_MANS = lt-comp.1 lt-expand.1 lt-proc.1 lt-tmxcomp.1 lt-tmxproc.1
 
-INCLUDES = -I$(top_srcdir) $(LTTOOLBOX_CFLAGS)
+INCLUDES = -I$(top_builddir) -I$(top_srcdir) $(LTTOOLBOX_CFLAGS)
 CLEANFILES = *~
 
 EXTRA_DIST = dix.dtd CMakeLists.txt $(man_MANS)
