--- origsrc/lttoolbox-3.6.6/lttoolbox/Makefile.am	2022-05-31 14:19:23.000000000 +0900
+++ src/lttoolbox-3.6.6/lttoolbox/Makefile.am	2022-07-02 10:26:27.995921900 +0900
@@ -19,7 +19,7 @@ instdir = lttoolbox
 
 lib_LTLIBRARIES= liblttoolbox3.la
 liblttoolbox3_la_SOURCES= $(h_sources) $(cc_sources)
-liblttoolbox3_la_LDFLAGS= -version-info $(SOVERSION) -release $(VERSION_API)
+liblttoolbox3_la_LDFLAGS= -no-undefined -version-info $(SOVERSION) -release $(VERSION_API)
 liblttoolbox3_la_LIBADD= $(ICU_LIBS)
 
 lttoolboxdir = $(prefix)/share/lttoolbox
@@ -52,7 +52,7 @@ lt_tmxproc_SOURCES = lt_tmxproc.cc
 
 man_MANS = lt-append.1 lt-comp.1 lt-expand.1 lt-proc.1 lt-tmxcomp.1 lt-tmxproc.1 lt-print.1 lt-trim.1
 
-INCLUDES = -I$(top_srcdir) $(LIBXML_CFLAGS) $(ICU_CFLAGS)
+INCLUDES = -I$(top_builddir) -I$(top_srcdir) $(LIBXML_CFLAGS) $(ICU_CFLAGS)
 CLEANFILES = *~
 
 EXTRA_DIST = dix.dtd dix.rng dix.rnc acx.rng xsd/dix.xsd xsd/acx.xsd $(man_MANS)
--- origsrc/lttoolbox-3.6.6/python/setup.py.in	2022-05-31 14:19:23.000000000 +0900
+++ src/lttoolbox-3.6.6/python/setup.py.in	2022-07-02 11:01:27.682143400 +0900
@@ -17,8 +17,8 @@ lttoolbox_module = Extension(
 	language='c++',
     sources=['lttoolbox.i'],
     swig_opts = ["-c++", '-I..', "-I@top_srcdir@", "-Wall"],
-    include_dirs=['@top_srcdir@', '@top_srcdir@/lttoolbox'] + '@LIBXML_CFLAGS@'.replace('-I', '').split() + '@ICU_CFLAGS@'.replace('-I', '').split(),
-    library_dirs=['@top_srcdir@/lttoolbox/.libs'],
+    include_dirs=['@top_srcdir@', '@top_srcdir@/lttoolbox'] + '@LIBXML_CFLAGS@'.replace('-I', '').split() + '@ICU_CFLAGS@'.replace('-I', '').split() + '@CPPFLAGS@'.replace('-I', '').replace('-D_GNU_SOURCE', '').split(),
+    library_dirs=['@top_builddir@/lttoolbox/.libs'],
 	libraries=['lttoolbox@VERSION_MAJOR@', 'xml2', 'icuio', 'icui18n', 'icuuc', 'icudata'],
     extra_compile_args=compile_args,
     extra_link_args=link_args,
