--- origsrc/libproperties-1.1.2/CMakeLists.txt	2020-12-20 23:44:25.000000000 +0900
+++ src/libproperties-1.1.2/CMakeLists.txt	2025-11-17 21:30:55.730736000 +0900
@@ -17,8 +17,6 @@ project(libproperties VERSION ${LIBPROPE
 #   Define GNU standard installation directories
 include(GNUInstallDirs)
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g")
-
 #   Define the target
 add_library(properties buf.c properties.c)
 set_target_properties(properties PROPERTIES
@@ -56,7 +54,7 @@ if(LIBPROPERTIES_INSTALL)
         DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
     )
     install(EXPORT properties NAMESPACE "libproperties::"
-        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECTNAME}"
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
         FILE "libproperties-config.cmake"
     )
 endif()
@@ -83,3 +81,10 @@ if(LIBPROPERTIES_TEST)
         libproproperties_add_test("${casename}")
     endforeach()
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libproperties.pc.in ${CMAKE_BINARY_DIR}/libproperties.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libproperties.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libproperties-1.1.2/libproperties.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libproperties-1.1.2/libproperties.pc.in	2025-11-17 21:31:35.893133600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libproperties
+Description: Library to parse the Java .properties files
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lproperties
+Cflags: -I${includedir}
