--- origsrc/liblzf-3.6/Makefile.am	2025-08-11 23:31:19.780234300 +0900
+++ src/liblzf-3.6/Makefile.am	2025-08-11 23:32:06.277473000 +0900
@@ -72,3 +72,7 @@ EXTRA_DIST = $(docfiles) $(csCLZFfiles)
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = liblzf.pc
 
+check_PROGRAMS = bench
+bench_LDADD = liblzf.la
+bench_SOURCES = bench.c
+TESTS = $(check_PROGRAMS)
--- origsrc/liblzf-3.6/bench.c	2025-08-11 23:31:18.956188600 +0900
+++ src/liblzf-3.6/bench.c	2025-08-11 23:32:39.581872000 +0900
@@ -45,8 +45,6 @@ static void sigu (int signum)
 #define DSIZE 17318440
 //#define DSIZE 32768
 
-#include "lzf_c_best.c"
-
 unsigned char data[DSIZE], data2[DSIZE*2], data3[DSIZE*2];
 
 int main(void)
@@ -86,7 +84,7 @@ int main(void)
       //struct timeval tv; gettimeofday (&tv, 0);
       //void *x = mmap (0, 16384, PROT_READ|PROT_WRITE, MAP_ANONYMOUS|MAP_PRIVATE,-1,0);
 
-      l = lzf_compress_best (data, DSIZE, data2, DSIZE*2);
+      l = lzf_compress (data, DSIZE, data2, DSIZE*2);
       //for (k = 0; k < l; ++k)
         //printf ("1 %2d: %02x\n", k, data2[k]);
       assert(l);
--- origsrc/liblzf-3.6/liblzf.pc.in	2025-08-11 23:31:19.778221600 +0900
+++ src/liblzf-3.6/liblzf.pc.in	2025-08-11 23:32:52.138158300 +0900
@@ -5,7 +5,7 @@ includedir=@includedir@
 
 Name: liblzf
 Description: Small data compression library
-URL: http://oldhome.schmorp.de/marc/liblzf.html
+URL: https://oldhome.schmorp.de/marc/liblzf.html
 Version: @VERSION@
 Libs: -L${libdir} -llzf
 Cflags: -I${includedir}
