--- origsrc/liblzf-3.5/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/liblzf-3.5/Makefile.am	2011-02-05 20:51:33.782211100 +0900
@@ -0,0 +1,14 @@
+lib_LTLIBRARIES = liblzf.la
+liblzf_la_SOURCES = lzf_c.c lzf_d.c
+liblzf_la_LDFLAGS = -no-undefined
+
+bin_PROGRAMS = lzf
+lzf_SOURCES = lzf.c
+lzf_LDADD = liblzf.la
+
+TESTS = bench
+check_PROGRAMS = bench
+bench_SOURCES = bench.c
+bench_LDADD = liblzf.la
+
+include_HEADERS = lzf.h
--- origsrc/liblzf-3.5/configure.ac	2007-11-30 19:33:44.000000000 +0900
+++ src/liblzf-3.5/configure.ac	2011-02-05 22:05:53.110136200 +0900
@@ -1,4 +1,5 @@
-AC_INIT
+AC_INIT(liblzf, 3.5)
+AM_INIT_AUTOMAKE
 AC_CONFIG_SRCDIR([lzfP.h])
 
 AC_CONFIG_HEADER(config.h)
@@ -6,9 +7,9 @@ AC_CONFIG_HEADER(config.h)
 AC_GNU_SOURCE
 AC_SYS_LARGEFILE
 AC_PROG_CC
-AC_PROG_RANLIB
 AC_PROG_INSTALL
 AC_HEADER_STDC
+LT_INIT(win32-dll)
 
 AC_C_CONST
 AC_C_INLINE
