--- origsrc/liblzf-3.6/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/liblzf-3.6/Makefile.am	2014-07-23 21:42:19.388817000 +0900
@@ -0,0 +1,16 @@
+
+bin_PROGRAMS = lzf
+lzf_SOURCES = lzf.c
+lzf_LDADD = liblzf.la
+
+TESTS = bench
+check_PROGRAMS = bench
+bench_SOURCES = bench.c
+bench_LDADD = liblzf.la
+
+include_HEADERS = lzf.h
+
+lib_LTLIBRARIES = liblzf.la
+liblzf_la_SOURCES = lzf_c.c lzf_d.c
+liblzf_la_LDFLAGS = -no-undefined
+
--- origsrc/liblzf-3.6/bench.c	2014-07-23 21:41:50.031395200 +0900
+++ src/liblzf-3.6/bench.c	2014-07-23 21:53:59.111473300 +0900
@@ -45,7 +45,8 @@ static void sigu (int signum)
 #define DSIZE 17318440
 //#define DSIZE 32768
 
-#include "lzf_c_slow.c"
+#define lzf_compress_slow lzf_compress
+#include "lzf_c.c"
 
 unsigned char data[DSIZE], data2[DSIZE*2], data3[DSIZE*2];
 
--- origsrc/liblzf-3.6/configure.ac	2007-11-30 19:33:44.000000000 +0900
+++ src/liblzf-3.6/configure.ac	2014-07-23 21:41:50.656395200 +0900
@@ -1,4 +1,5 @@
-AC_INIT
+AC_INIT(liblzf, 3.6)
+AM_INIT_AUTOMAKE
 AC_CONFIG_SRCDIR([lzfP.h])
 
 AC_CONFIG_HEADER(config.h)
@@ -6,9 +7,9 @@ AC_CONFIG_HEADER(config.h)
 AC_GNU_SOURCE
 AC_SYS_LARGEFILE
 AC_PROG_CC
-AC_PROG_RANLIB
 AC_PROG_INSTALL
 AC_HEADER_STDC
+LT_INIT
 
 AC_C_CONST
 AC_C_INLINE
