--- origsrc/libgnurdf-0.3.1/acinclude.m4	2002-05-14 13:57:48.000000000 +0900
+++ src/libgnurdf-0.3.1/acinclude.m4	2023-10-03 08:55:26.497949600 +0900
@@ -2,7 +2,7 @@ dnl # $Id: acinclude.m4,v 1.4 2002/05/14
 dnl #
 dnl # Check for --with-leakbug option.
 
-AC_DEFUN(AM_WITH_LEAKBUG,
+AC_DEFUN([AM_WITH_LEAKBUG],
 [
 	AC_MSG_CHECKING(if leakbug is wanted)
 	AC_ARG_WITH(leakbug,
@@ -27,7 +27,7 @@ AC_DEFUN(AM_WITH_LEAKBUG,
 
 # serial 1
 
-AC_DEFUN(AM_WITH_DMALLOC,
+AC_DEFUN([AM_WITH_DMALLOC],
 [AC_MSG_CHECKING(if malloc debugging is wanted)
 AC_ARG_WITH(dmalloc,
 [  --with-dmalloc          use dmalloc, as in
@@ -52,7 +52,7 @@ fi], [AC_MSG_RESULT(no)])
 dnl AM_PATH_XML([MINIMUM-VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]]])
 dnl Test for XML, and define XML_CFLAGS and XML_LIBS
 dnl
-AC_DEFUN(AM_PATH_XML,[ 
+AC_DEFUN([AM_PATH_XML],[ 
 AC_ARG_WITH(xml-prefix,
             [  --with-xml-prefix=PFX   Prefix where libxml is installed (optional)],
             xml_config_prefix="$withval", xml_config_prefix="")
@@ -247,7 +247,7 @@ main()
 dnl AM_PATH_XML2([MINIMUM-VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]]])
 dnl Test for XML, and define XML_CFLAGS and XML_LIBS
 dnl
-AC_DEFUN(AM_PATH_XML2,[ 
+AC_DEFUN([AM_PATH_XML2],[ 
 AC_ARG_WITH(xml-prefix,
             [  --with-xml-prefix=PFX   Prefix where libxml is installed (optional)],
             xml_config_prefix="$withval", xml_config_prefix="")
--- origsrc/libgnurdf-0.3.1/libgnurdf/Makefile.am	2002-05-14 14:57:31.000000000 +0900
+++ src/libgnurdf-0.3.1/libgnurdf/Makefile.am	2023-10-03 08:33:14.605011700 +0900
@@ -3,8 +3,6 @@
 libincludedir = $(includedir)/libgnurdf
 
 libinclude_HEADERS = \
-	schema.h \
-	types.h \
 	alt.h \
 	attribute.h \
 	bag.h \
@@ -42,7 +40,7 @@ libgnurdf_la_SOURCES = \
 libgnurdf_la_LIBADD = $(XML_LIBS)
 
 libgnurdf_la_LDFLAGS = \
-	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
+	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -no-undefined
 
 INCLUDES = \
 	-I$(top_srcdir) \
--- origsrc/libgnurdf-0.3.1/libgnurdf.m4.in	2001-11-03 17:06:58.000000000 +0900
+++ src/libgnurdf-0.3.1/libgnurdf.m4.in	2023-10-03 08:33:14.605011700 +0900
@@ -4,7 +4,7 @@
 dnl AM_PATH_GNURDF([MINIMUM-VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]]])
 dnl Test for GNURDF, and define GNURDF_CFLAGS and GNURDF_LIBS
 dnl
-AC_DEFUN(AM_PATH_GNURDF,
+AC_DEFUN([AM_PATH_GNURDF],
 [dnl 
 dnl Get the cflags and libraries from the gnurdf-config script
 dnl
