diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x Doxyfile -x build_config.hpp origsrc/actor-framework-0.15.7/libcaf_core/CMakeLists.txt src/actor-framework-0.15.7/libcaf_core/CMakeLists.txt
--- origsrc/actor-framework-0.15.7/libcaf_core/CMakeLists.txt	2018-03-21 17:56:46.000000000 +0900
+++ src/actor-framework-0.15.7/libcaf_core/CMakeLists.txt	2018-05-11 20:19:32.386347400 +0900
@@ -132,7 +132,7 @@ if (NOT CAF_BUILD_STATIC_ONLY)
                         VERSION ${CAF_VERSION}
                         OUTPUT_NAME caf_core)
   if (CYGWIN)
-    install(TARGETS libcaf_core_shared RUNTIME DESTINATION bin)
+    install(TARGETS libcaf_core_shared RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
   else()
     install(TARGETS libcaf_core_shared LIBRARY DESTINATION lib)
   endif()
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x Doxyfile -x build_config.hpp origsrc/actor-framework-0.15.7/libcaf_io/CMakeLists.txt src/actor-framework-0.15.7/libcaf_io/CMakeLists.txt
--- origsrc/actor-framework-0.15.7/libcaf_io/CMakeLists.txt	2018-03-21 17:56:46.000000000 +0900
+++ src/actor-framework-0.15.7/libcaf_io/CMakeLists.txt	2018-05-11 20:20:08.643372400 +0900
@@ -49,7 +49,7 @@ if (NOT CAF_BUILD_STATIC_ONLY)
                         VERSION ${CAF_VERSION}
                         OUTPUT_NAME caf_io)
   if (CYGWIN)
-    install(TARGETS libcaf_io_shared RUNTIME DESTINATION bin)
+    install(TARGETS libcaf_io_shared RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
   else()
     install(TARGETS libcaf_io_shared LIBRARY DESTINATION lib)
   endif()
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x Doxyfile -x build_config.hpp origsrc/actor-framework-0.15.7/libcaf_opencl/CMakeLists.txt src/actor-framework-0.15.7/libcaf_opencl/CMakeLists.txt
--- origsrc/actor-framework-0.15.7/libcaf_opencl/CMakeLists.txt	2018-03-21 17:56:46.000000000 +0900
+++ src/actor-framework-0.15.7/libcaf_opencl/CMakeLists.txt	2018-05-11 19:19:37.603143700 +0900
@@ -28,7 +28,7 @@ if(NOT CAF_BUILD_STATIC_ONLY)
                         VERSION "${CAF_VERSION}"
                         OUTPUT_NAME caf_opencl)
   if(NOT WIN32)
-    install(TARGETS libcaf_opencl_shared LIBRARY DESTINATION lib)
+    install(TARGETS libcaf_opencl_shared RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
   endif()
 endif()
 # build static library only if --build-static or --build-static-only was set
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x Doxyfile -x build_config.hpp origsrc/actor-framework-0.15.7/libcaf_openssl/CMakeLists.txt src/actor-framework-0.15.7/libcaf_openssl/CMakeLists.txt
--- origsrc/actor-framework-0.15.7/libcaf_openssl/CMakeLists.txt	2018-03-21 17:56:46.000000000 +0900
+++ src/actor-framework-0.15.7/libcaf_openssl/CMakeLists.txt	2018-05-11 20:20:41.668551900 +0900
@@ -35,7 +35,7 @@ if (NOT CAF_BUILD_STATIC_ONLY)
                         VERSION ${CAF_VERSION}
                         OUTPUT_NAME caf_openssl)
   if (CYGWIN)
-    install(TARGETS libcaf_openssl_shared RUNTIME DESTINATION bin)
+    install(TARGETS libcaf_openssl_shared RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
   elseif (NOT WIN32)
     install(TARGETS libcaf_openssl_shared LIBRARY DESTINATION lib)
   endif()
