--- origsrc/actor-framework-0.18.6/libcaf_core/src/scheduled_actor.cpp	2022-03-25 05:29:16.000000000 +0900
+++ src/actor-framework-0.18.6/libcaf_core/src/scheduled_actor.cpp	2022-04-25 10:45:29.335407800 +0900
@@ -173,6 +173,8 @@ bool scheduled_actor::enqueue(mailbox_el
       // enqueued to a running actors' mailbox; nothing to do
       CAF_LOG_ACCEPT_EVENT(false);
       return true;
+    default:
+      return false;
   }
 }
 
