--- origsrc/actor-framework-0.17.3/libcaf_core/CMakeLists.txt	2019-11-12 04:28:41.000000000 +0900
+++ src/actor-framework-0.17.3/libcaf_core/CMakeLists.txt	2020-01-01 01:05:15.656790400 +0900
@@ -186,7 +186,7 @@ if (NOT CAF_BUILD_STATIC_ONLY)
   )
   install(TARGETS libcaf_core_shared
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
   )
   add_dependencies(libcaf_core_shared libcaf_core)
 endif ()
--- origsrc/actor-framework-0.17.3/libcaf_io/CMakeLists.txt	2019-11-12 04:28:41.000000000 +0900
+++ src/actor-framework-0.17.3/libcaf_io/CMakeLists.txt	2020-01-01 01:05:38.673329400 +0900
@@ -68,7 +68,7 @@ if (NOT CAF_BUILD_STATIC_ONLY)
                         OUTPUT_NAME caf_io)
   install(TARGETS libcaf_io_shared
           RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-          LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+          ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
   add_dependencies(libcaf_io_shared libcaf_io)
 endif ()
 
--- origsrc/actor-framework-0.17.3/libcaf_opencl/CMakeLists.txt	2019-11-12 04:28:41.000000000 +0900
+++ src/actor-framework-0.17.3/libcaf_opencl/CMakeLists.txt	2020-01-01 01:06:36.972028900 +0900
@@ -30,7 +30,7 @@ if(NOT CAF_BUILD_STATIC_ONLY)
                         SOVERSION "${CAF_VERSION}"
                         VERSION "${CAF_LIB_VERSION}"
                         OUTPUT_NAME caf_opencl)
-  install(TARGETS libcaf_opencl_shared LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  install(TARGETS libcaf_opencl_shared RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
 
 # build static library only if --build-static or --build-static-only was set
--- origsrc/actor-framework-0.17.3/libcaf_openssl/CMakeLists.txt	2019-11-12 04:28:41.000000000 +0900
+++ src/actor-framework-0.17.3/libcaf_openssl/CMakeLists.txt	2020-01-01 01:07:14.892041100 +0900
@@ -35,7 +35,7 @@ if (NOT CAF_BUILD_STATIC_ONLY)
                         VERSION ${CAF_LIB_VERSION}
                         OUTPUT_NAME caf_openssl)
   if (CYGWIN)
-    install(TARGETS libcaf_openssl_shared RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+    install(TARGETS libcaf_openssl_shared RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
   elseif (NOT WIN32)
     install(TARGETS libcaf_openssl_shared LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
   endif()
