--- origsrc/actor-framework-0.17.6/libcaf_core/CMakeLists.txt	2020-07-25 03:31:40.000000000 +0900
+++ src/actor-framework-0.17.6/libcaf_core/CMakeLists.txt	2020-09-01 01:12:22.625110400 +0900
@@ -190,7 +190,7 @@ if (NOT CAF_BUILD_STATIC_ONLY)
   )
   install(TARGETS libcaf_core_shared
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
   )
   add_dependencies(libcaf_core_shared libcaf_core)
 endif ()
--- origsrc/actor-framework-0.17.6/libcaf_io/CMakeLists.txt	2020-07-25 03:31:40.000000000 +0900
+++ src/actor-framework-0.17.6/libcaf_io/CMakeLists.txt	2020-09-01 01:12:22.673169900 +0900
@@ -69,7 +69,7 @@ if (NOT CAF_BUILD_STATIC_ONLY)
                         OUTPUT_NAME caf_io)
   install(TARGETS libcaf_io_shared
           RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-          LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+          ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
   add_dependencies(libcaf_io_shared libcaf_io)
 endif ()
 
--- origsrc/actor-framework-0.17.6/libcaf_opencl/CMakeLists.txt	2020-07-25 03:31:40.000000000 +0900
+++ src/actor-framework-0.17.6/libcaf_opencl/CMakeLists.txt	2020-09-01 01:12:22.848173900 +0900
@@ -30,7 +30,7 @@ if(NOT CAF_BUILD_STATIC_ONLY)
                         SOVERSION "${CAF_VERSION}"
                         VERSION "${CAF_LIB_VERSION}"
                         OUTPUT_NAME caf_opencl)
-  install(TARGETS libcaf_opencl_shared LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  install(TARGETS libcaf_opencl_shared RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
 
 # build static library only if --build-static or --build-static-only was set
--- origsrc/actor-framework-0.17.6/libcaf_openssl/CMakeLists.txt	2020-07-25 03:31:40.000000000 +0900
+++ src/actor-framework-0.17.6/libcaf_openssl/CMakeLists.txt	2020-09-01 01:12:23.133300300 +0900
@@ -35,7 +35,7 @@ if (NOT CAF_BUILD_STATIC_ONLY)
                         VERSION ${CAF_LIB_VERSION}
                         OUTPUT_NAME caf_openssl)
   if (CYGWIN)
-    install(TARGETS libcaf_openssl_shared RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+    install(TARGETS libcaf_openssl_shared RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
   elseif (NOT WIN32)
     install(TARGETS libcaf_openssl_shared LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
   endif()
