--- origsrc/actor-framework-0.16.3/libcaf_core/CMakeLists.txt	2018-12-28 05:33:32.000000000 +0900
+++ src/actor-framework-0.16.3/libcaf_core/CMakeLists.txt	2019-01-10 08:21:36.447914300 +0900
@@ -159,7 +159,7 @@ if (NOT CAF_BUILD_STATIC_ONLY)
   )
   install(TARGETS libcaf_core_shared
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib
   )
   add_dependencies(libcaf_core_shared libcaf_core)
 endif ()
--- origsrc/actor-framework-0.16.3/libcaf_io/CMakeLists.txt	2018-12-28 05:33:32.000000000 +0900
+++ src/actor-framework-0.16.3/libcaf_io/CMakeLists.txt	2019-01-10 08:21:53.941682200 +0900
@@ -66,7 +66,7 @@ if (NOT CAF_BUILD_STATIC_ONLY)
                         OUTPUT_NAME caf_io)
   install(TARGETS libcaf_io_shared
           RUNTIME DESTINATION bin
-          LIBRARY DESTINATION lib)
+          ARCHIVE DESTINATION lib)
   add_dependencies(libcaf_io_shared libcaf_io)
 endif ()
 
--- origsrc/actor-framework-0.16.3/libcaf_opencl/CMakeLists.txt	2018-12-28 05:33:32.000000000 +0900
+++ src/actor-framework-0.16.3/libcaf_opencl/CMakeLists.txt	2019-01-10 08:22:09.229543900 +0900
@@ -31,7 +31,7 @@ if(NOT CAF_BUILD_STATIC_ONLY)
                         SOVERSION "${CAF_VERSION}"
                         VERSION "${CAF_VERSION}"
                         OUTPUT_NAME caf_opencl)
-  install(TARGETS libcaf_opencl_shared LIBRARY DESTINATION lib)
+  install(TARGETS libcaf_opencl_shared RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 endif()
 
 # build static library only if --build-static or --build-static-only was set
--- origsrc/actor-framework-0.16.3/libcaf_openssl/CMakeLists.txt	2018-12-28 05:33:32.000000000 +0900
+++ src/actor-framework-0.16.3/libcaf_openssl/CMakeLists.txt	2019-01-10 08:22:54.397923200 +0900
@@ -35,11 +35,7 @@ if (NOT CAF_BUILD_STATIC_ONLY)
                         SOVERSION ${CAF_VERSION}
                         VERSION ${CAF_VERSION}
                         OUTPUT_NAME caf_openssl)
-  if (CYGWIN)
-    install(TARGETS libcaf_openssl_shared RUNTIME DESTINATION bin)
-  elseif (NOT WIN32)
-    install(TARGETS libcaf_openssl_shared LIBRARY DESTINATION lib)
-  endif()
+  install(TARGETS libcaf_openssl_shared RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
   add_dependencies(libcaf_openssl_shared libcaf_openssl)
 endif ()
 
