--- origsrc/actor-framework-0.18.7/libcaf_core/src/scheduled_actor.cpp	2023-02-08 16:31:38.000000000 +0900
+++ src/actor-framework-0.18.7/libcaf_core/src/scheduled_actor.cpp	2023-03-02 10:32:43.714407100 +0900
@@ -173,6 +173,8 @@ bool scheduled_actor::enqueue(mailbox_el
       // enqueued to a running actors' mailbox; nothing to do
       CAF_LOG_ACCEPT_EVENT(false);
       return true;
+    default:
+      return false;
   }
 }
 
