--- origsrc/dspkg-validator/CMakeLists.txt	2026-05-30 09:50:49.000000000 +0900
+++ src/dspkg-validator/CMakeLists.txt	2026-05-30 09:52:11.823627000 +0900
@@ -61,6 +61,10 @@ add_library(${PROJECT_NAME}::${PROJECT_N
 find_package(nlohmann_json CONFIG REQUIRED)
 target_link_libraries(${PROJECT_NAME} nlohmann_json::nlohmann_json)
 
+if (DEFINED SOVERSION)
+    set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION ${SOVERSION})
+endif ()
+
 if (DSPKG_VALIDATOR_BUILD_TESTS)
     add_subdirectory(tests)
 endif ()
--- origsrc/dspkg-validator/dspkg-validator.pc.in	2026-05-30 09:50:49.000000000 +0900
+++ src/dspkg-validator/dspkg-validator.pc.in	2026-05-30 09:51:41.340947600 +0900
@@ -1,4 +1,4 @@
-prefix=${pcfiledir}/../..
+prefix=@CMAKE_INSTALL_PREFIX@
 
 # dspkg-validator pkg-config file
 
--- origsrc/dspkg-validator/include/dspkg-validator/LanguageUtil.h	2026-05-30 09:50:49.000000000 +0900
+++ src/dspkg-validator/include/dspkg-validator/LanguageUtil.h	2026-05-30 10:11:16.153835700 +0900
@@ -3,6 +3,7 @@
 
 #include <filesystem>
 #include <set>
+#include <vector>
 
 namespace DspkgVal
 {
