--- origsrc/libfplll-3.1.1/configure.in	2011-04-14 22:57:24.000000000 +0900
+++ src/libfplll-3.1.1/configure.in	2011-05-18 23:48:44.829139900 +0900
@@ -14,7 +14,7 @@ CFLAGS=${CFLAGS:-$DEFAULT_CFLAGS}
 CXXFLAGS=${CXXFLAGS:-$DEFAULT_CFLAGS}
 
 #AC_DISABLE_SHARED
-AC_PROG_LIBTOOL
+LT_INIT
 
 AC_PROG_INSTALL
 
--- origsrc/libfplll-3.1.1/src/Makefile.am	2011-02-04 23:40:39.000000000 +0900
+++ src/libfplll-3.1.1/src/Makefile.am	2011-05-18 23:49:47.849759100 +0900
@@ -41,7 +41,7 @@ libfplll_la_SOURCES=fplll.cpp fplll.h ut
 	svpcvp.cpp svpcvp.h
 EXTRA_libfplll_la_SOURCES=wrapper.cpp wrapper.h
 libfplll_la_LIBADD=-lgmp -lmpfr
-libfplll_la_LDFLAGS=-version-info 1:0:1
+libfplll_la_LDFLAGS=-version-info 1:0:1 -no-undefined
 
 libfplllv_la_SOURCES=$(libfplll_la_SOURCES)
 EXTRA_libfplllv_la_SOURCES=$(EXTRA_libfplll_la_SOURCES)
