--- origsrc/libfplll-3.0.12/Makefile.am	2008-07-17 17:36:09.000000000 +0900
+++ src/libfplll-3.0.12/Makefile.am	2010-09-23 10:24:53.707385500 +0900
@@ -6,6 +6,6 @@ lib_LTLIBRARIES = libfplll.la
 libfplll_la_SOURCES = dummy.cpp
 
 libfplll_la_LIBADD = -lgmp -lmpfr
-libfplll_la_LDFLAGS = -version-info 1:0:1
+libfplll_la_LDFLAGS = -version-info 1:0:1 -no-undefined
 
 
--- origsrc/libfplll-3.0.12/configure.in	2009-05-10 19:04:27.000000000 +0900
+++ src/libfplll-3.0.12/configure.in	2010-09-23 10:24:00.954260500 +0900
@@ -11,7 +11,7 @@ DEFAULT_CFLAGS="-O2 -Wall"
 CFLAGS=${CFLAGS:-$DEFAULT_CFLAGS}
 CXXFLAGS=${CXXFLAGS:-$DEFAULT_CFLAGS}
 
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 AC_PROG_INSTALL
 
