--- origsrc/libecap-0.2.0/src/libecap/Makefile.am	2011-02-17 01:53:27.000000000 +0900
+++ src/libecap-0.2.0/src/libecap/Makefile.am	2012-06-08 07:59:41.991147600 +0900
@@ -11,7 +11,7 @@ libecap_la_SOURCES =
 # Dummy C++ source to cause C++ linking.
 nodist_EXTRA_libecap_la_SOURCES = dummy4lib.cc
 
-libecap_la_LDFLAGS = -version-info $(LIBECAP_LIBRARY_VERSION)
+libecap_la_LDFLAGS = -version-info $(LIBECAP_LIBRARY_VERSION) -no-undefined
 
 # include all libraries
 libecap_la_LIBADD = \
