--- origsrc/hashalot-0.3/hashalot.c	2004-04-29 13:06:23.000000000 +0900
+++ src/hashalot-0.3/hashalot.c	2024-11-01 09:45:11.545501300 +0900
@@ -256,11 +256,13 @@ main(int argc, char *argv[])
 	 * plus a newline, plus a null */
 	passhash = xmalloc(2*hashlen + 2);
 
+#if !defined(__CYGWIN__)
 	/* try to lock memory so it doesn't get swapped out for sure */
 	if (mlockall(MCL_CURRENT | MCL_FUTURE) == -1) {
 		perror("mlockall");
 		fputs("Warning: couldn't lock memory, are you root?\n", stderr);
 	}
+#endif
 
 	/* here we acquire the precious passphrase... */
 	pass = xgetpass("Enter passphrase: ");
