--- origsrc/gresg-0.4.0/configure.ac	2019-01-18 04:26:29.000000000 +0900
+++ src/gresg-0.4.0/configure.ac	2020-12-06 16:39:12.048562600 +0900
@@ -5,7 +5,7 @@ dnl ************************************
 dnl Define Versioning Information
 dnl ***********************************************************************
 m4_define([major_version],[0])
-m4_define([minor_version],[2])
+m4_define([minor_version],[4])
 m4_define([micro_version],[0])
 m4_define([package_version],[major_version.minor_version.micro_version])
 m4_define([bug_report_url],[https://bugzilla.gnome.org/enter_bug.cgi?product=gresg])
