--- origsrc/libcoap-4.3.0/Makefile.am	2021-06-21 17:39:42.000000000 +0900
+++ src/libcoap-4.3.0/Makefile.am	2021-09-04 12:37:25.838931200 +0900
@@ -171,7 +171,7 @@ libcoap_@LIBCOAP_NAME_SUFFIX@_la_LDFLAGS
   -version-info $(LT_LIBCOAP_CURRENT):$(LT_LIBCOAP_REVISION):$(LT_LIBCOAP_AGE)	\
   @libcoap_SYMBOLS@ \
   $(DTLS_LIBS) \
-  -pie
+  -pie -no-undefined
 
 ## Collect symbols here we want to ignore while building the helper files
 ## libcoap-$(LIBCOAP_API_VERSION).{map,sym} for the linker.
File origsrc/libcoap-4.3.0/coap_config.h.lwip is a regular file while file src/libcoap-4.3.0/coap_config.h.lwip is a symbolic link
File origsrc/libcoap-4.3.0/coap_config.h.riot is a regular file while file src/libcoap-4.3.0/coap_config.h.riot is a symbolic link
File origsrc/libcoap-4.3.0/coap_config.h.windows is a regular file while file src/libcoap-4.3.0/coap_config.h.windows is a symbolic link
--- origsrc/libcoap-4.3.0/configure.ac	2021-06-21 17:39:42.000000000 +0900
+++ src/libcoap-4.3.0/configure.ac	2021-09-04 12:37:34.661214400 +0900
@@ -478,19 +478,6 @@ if test "x$build_dtls" = "xyes"; then
         have_tinydtls="no" # don't confuse AC_MSG_RESULT at the end of the script
     fi
 
-    # The user wants to use explicit TinyDTLS if '--with-tinydtls was set'.
-    if test "x$with_tinydtls" = "xyes" ; then
-        if test -d "$srcdir/ext/tinydtls"; then
-           AC_CONFIG_SUBDIRS([ext/tinydtls])
-           have_tinydtls="yes"
-         else
-           have_tinydtls="no" # don't confuse AC_MSG_RESULT at the end of the script
-         fi
-         have_gnutls="no" # don't confuse AC_MSG_RESULT at the end of the script
-         have_openssl="no" # don't confuse AC_MSG_RESULT at the end of the script
-         have_mbedtls="no" # don't confuse AC_MSG_RESULT at the end of the script
-    fi
-
     if test "$TLSCOUNT" -eq 0; then
       # The user hasn't requested the use of a specific cryptography library
       # we try first GnuTLS for usability ...
@@ -804,7 +791,6 @@ case $host in
     *-cygwin*)
     AC_MSG_RESULT([Cygwin])
     ADDITIONAL_CFLAGS="-D_GNU_SOURCE -D_CYGWIN_ENV"
-    LDFLAGS="-no-undefined $LDFLAGS"
     ;;
 
     *-solaris*)
