--- origsrc/pnetcdf-1.11.2/acinclude.m4	2019-05-06 04:43:13.000000000 +0900
+++ src/pnetcdf-1.11.2/acinclude.m4	2019-07-02 17:26:49.824368300 +0900
@@ -2111,7 +2111,7 @@ AC_DEFUN([LT_MPI_CHECK_SHLIB],[
       RM="$RM -f"
    fi
    ac_ltcompile='./libtool --mode=compile $MPICC -c $CFLAGS $CPPFLAGS conftest.$ac_ext -o conftest.lo >&AS_MESSAGE_LOG_FD'
-   ac_ltlink_la='./libtool --mode=link $MPICC -rpath `pwd` $CFLAGS $LDFLAGS -o libconftest.la conftest.lo $LIBS >&AS_MESSAGE_LOG_FD'
+   ac_ltlink_la='./libtool --mode=link $MPICC -rpath `pwd` $CFLAGS $LDFLAGS -no-undefined -o libconftest.la conftest.lo $LIBS >&AS_MESSAGE_LOG_FD'
    AS_IF([AC_TRY_EVAL([ac_ltcompile]) &&
        AC_TRY_EVAL([ac_ltlink_la]) &&
        AC_TRY_COMMAND([test -s libconftest.la])],
--- origsrc/pnetcdf-1.11.2/src/libs/Makefile.am	2019-05-01 08:25:09.000000000 +0900
+++ src/pnetcdf-1.11.2/src/libs/Makefile.am	2019-07-02 17:26:49.855652000 +0900
@@ -11,7 +11,7 @@ lib_LTLIBRARIES = libpnetcdf.la
 # Using libpnetcdf_la_LDFLAGS = $(ABIVERSIONFLAGS) does NOT seem to work,
 # because libpnetcdf_la_LINK is explicitly set below to deal with cross
 # language link.
-AM_LDFLAGS = $(ABIVERSIONFLAGS)
+AM_LDFLAGS = -no-undefined $(ABIVERSIONFLAGS)
 
 libpnetcdf_la_SOURCES =
 libpnetcdf_la_LIBADD  = ../dispatchers/libdispatchers.la
