--- origsrc/libparserutils-0.1.1/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libparserutils-0.1.1/Makefile.am	2012-03-08 23:25:20.172061700 +0900
@@ -0,0 +1,77 @@
+INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/src
+
+lib_LTLIBRARIES = libparserutils.la
+
+libparserutils_la_SOURCES = \
+	src/charset/aliases.c \
+	src/charset/codec.c \
+	src/charset/codecs/codec_8859.c \
+	src/charset/codecs/codec_ascii.c \
+	src/charset/codecs/codec_ext8.c \
+	src/charset/codecs/codec_utf16.c \
+	src/charset/codecs/codec_utf8.c \
+	src/charset/encodings/utf16.c \
+	src/charset/encodings/utf8.c \
+	src/input/filter.c \
+	src/input/inputstream.c \
+	src/utils/buffer.c \
+	src/utils/errors.c \
+	src/utils/stack.c \
+	src/utils/vector.c
+libparserutils_la_CPPFLAGS = \
+	-D_BSD_SOURCE -DNDEBUG \
+	-Wall -W -Wundef -Wpointer-arith -Wcast-align  -Wwrite-strings \
+	-Wstrict-prototypes -Wmissing-prototypes  -Wmissing-declarations \
+	-Wnested-externs -pedantic \
+	-D_ALIGNED="__attribute__((aligned))" -O3 -std=c99
+libparserutils_la_LDFLAGS = -no-undefined
+libparserutils_la_LIBADD = -liconv
+
+libparserutils_includedir = $(includedir)/parserutils
+libparserutils_include_HEADERS = \
+	include/parserutils/errors.h \
+	include/parserutils/functypes.h \
+	include/parserutils/parserutils.h \
+	include/parserutils/types.h
+libparserutils_charset_includedir = $(includedir)/parserutils/charset
+libparserutils_charset_include_HEADERS = \
+	include/parserutils/charset/codec.h \
+	include/parserutils/charset/mibenum.h \
+	include/parserutils/charset/utf16.h \
+	include/parserutils/charset/utf8.h
+libparserutils_input_includedir = $(includedir)/parserutils/input
+libparserutils_input_include_HEADERS = \
+	include/parserutils/input/inputstream.h
+libparserutils_utils_includedir = $(includedir)/parserutils/utils
+libparserutils_utils_include_HEADERS = \
+	include/parserutils/utils/buffer.h \
+	include/parserutils/utils/stack.h \
+	include/parserutils/utils/vector.h
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = libparserutils.pc
+
+check_PROGRAMS = \
+	test/aliases \
+	test/cscodec-8859 \
+	test/cscodec-ext8 \
+	test/cscodec-utf16 \
+	test/cscodec-utf8 \
+	test/filter \
+	test/inputstream \
+	test/regression/filter-badenc-segv \
+	test/regression/filter-segv \
+	test/regression/stream-nomem
+test_aliases_LDADD = libparserutils.la
+test_cscodec_8859_LDADD = libparserutils.la
+test_cscodec_ext8_LDADD = libparserutils.la
+test_cscodec_utf16_LDADD = libparserutils.la
+test_cscodec_utf8_LDADD = libparserutils.la
+test_filter_LDADD = libparserutils.la
+test_inputstream_LDADD = libparserutils.la
+test_regression_filter_badenc_segv_CPPFLAGS = -I$(top_srcdir)/test
+test_regression_filter_badenc_segv_LDADD = libparserutils.la
+test_regression_filter_segv_CPPFLAGS = -I$(top_srcdir)/test
+test_regression_filter_segv_LDADD = libparserutils.la
+test_regression_stream_nomem_CPPFLAGS = -I$(top_srcdir)/test
+test_regression_stream_nomem_LDADD = libparserutils.la
--- origsrc/libparserutils-0.1.1/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/libparserutils-0.1.1/configure.ac	2012-03-08 23:26:50.363789700 +0900
@@ -0,0 +1,36 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([libparserutils], [0.1.1])
+AC_CONFIG_SRCDIR([include/parserutils/errors.h])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AM_PROG_CC_C_O
+LT_INIT([disable-static])
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([arpa/inet.h inttypes.h netinet/in.h stddef.h stdint.h stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_HEADER_STDBOOL
+AC_C_INLINE
+AC_TYPE_INT32_T
+AC_TYPE_SIZE_T
+AC_TYPE_UINT16_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT8_T
+
+# Checks for library functions.
+AC_FUNC_ALLOCA
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([memmove memset strncasecmp])
+
+AC_CONFIG_FILES([Makefile libparserutils.pc])
+AC_OUTPUT
