--- origsrc/libportage-0.1_alpha0/Makefile.am	2009-06-05 21:06:02.000000000 +0900
+++ src/libportage-0.1_alpha0/Makefile.am	2015-05-04 15:51:41.562904500 +0900
@@ -2,3 +2,6 @@ include $(top_srcdir)/aminclude.am
 
 SUBDIRS = replace src include test doc
 EXTRA_DIST = doxygen.cfg
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libportage-0.1_alpha0/configure.ac	2009-06-05 21:06:02.000000000 +0900
+++ src/libportage-0.1_alpha0/configure.ac	2015-05-04 15:51:51.430170000 +0900
@@ -1,6 +1,6 @@
-AC_INIT([libportage], [HEAD], [libportage-bugs@lists.berlios.de])
+AC_INIT([libportage], [0.1_alpha0], [libportage-bugs@lists.berlios.de])
 AC_CONFIG_AUX_DIR(config)
-AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+AM_INIT_AUTOMAKE([-Wall foreign])
 
 # checks for programs
 AC_PROG_CC
@@ -40,7 +40,7 @@ AM_CONDITIONAL(USE_VALGRIND, $USE_VALGRI
 # other stuff
 AC_CONFIG_HEADERS(config.h)
 
-AC_CONFIG_FILES(Makefile replace/Makefile src/Makefile include/Makefile
+AC_CONFIG_FILES(libportage.pc Makefile replace/Makefile src/Makefile include/Makefile
                 test/Makefile doc/Makefile)
 
 # hack to transform replace lib list into libtool format.
--- origsrc/libportage-0.1_alpha0/libportage.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libportage-0.1_alpha0/libportage.pc.in	2015-05-04 15:52:09.838877000 +0900
@@ -0,0 +1,9 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+Name: libportage
+Description: Fast and portable library offering functionality for doing package management in a Gentoo system
+Version: @VERSION@
+Libs: -L${libdir} -lportage
+Cflags: -I${includedir}
--- origsrc/libportage-0.1_alpha0/src/Makefile.am	2009-06-05 21:06:02.000000000 +0900
+++ src/libportage-0.1_alpha0/src/Makefile.am	2015-05-04 15:43:45.595690000 +0900
@@ -2,8 +2,8 @@ lib_LTLIBRARIES = libportage.la
 
 libportage_la_SOURCES = liblpatom.c liblputil.c liblpxpak.c liblparchives.c   \
 			liblpversion.c
-libportage_la_LIBADD = ../replace/libreplace.la
-libportage_la_LDFLAGS = -larchive
+libportage_la_LIBADD = ../replace/libreplace.la -larchive
+libportage_la_LDFLAGS = -no-undefined
 
 AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/replace
 METASOURCES = AUTO
--- origsrc/libportage-0.1_alpha0/test/Makefile.am	2009-06-05 21:06:02.000000000 +0900
+++ src/libportage-0.1_alpha0/test/Makefile.am	2015-05-04 15:50:11.712995600 +0900
@@ -40,7 +40,7 @@ lp_objects = ../src/liblputil.o ../src/l
 05_lparchives_LDFLAGS = $(all_libraries)
 05_lparchives_LDADD = $(lp_objects)
 
-AM_CPPFLAGS = -I$(top_srcdir)/include
+AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/replace
 
 EXTRA_DIST = valgrind-wrapper
 
