--- origsrc/dsdcc-1.8.6/CMakeLists.txt	2019-08-23 07:02:54.000000000 +0900
+++ src/dsdcc-1.8.6/CMakeLists.txt	2019-10-13 07:10:21.926743300 +0900
@@ -14,7 +14,11 @@ option(BUILD_TOOL "Build dsdccx tool" ON
 # use c++11
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
+if(CYGWIN)
+    set(CMAKE_CXX_EXTENSIONS ON)
+else()
+    set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
 
 # use, i.e. don't skip the full RPATH for the build tree
 set(CMAKE_SKIP_BUILD_RPATH  FALSE)
@@ -178,13 +182,13 @@ if (BUILD_DEBIAN)
     target_link_libraries(dsdcc mbelib)
 endif()
 
+set_target_properties(dsdcc PROPERTIES VERSION ${VERSION} SOVERSION ${MAJOR_VERSION})
+
 if(BUILD_TOOL)
 add_executable(dsdccx
     dsd_main.cpp
 )
 
-set_target_properties(dsdccx PROPERTIES VERSION ${VERSION} SOVERSION ${MAJOR_VERSION})
-
 target_include_directories(dsdccx PUBLIC
     ${PROJECT_SOURCE_DIR}
     ${CMAKE_CURRENT_BINARY_DIR}
@@ -222,5 +226,5 @@ INSTALL(
 if(BUILD_TOOL)
     install(TARGETS dsdccx DESTINATION bin)
 endif(BUILD_TOOL)
-install(TARGETS dsdcc DESTINATION lib${LIB_SUFFIX})
+install(TARGETS dsdcc RUNTIME DESTINATION bin ARCHIVE DESTINATION lib${LIB_SUFFIX})
 install(FILES ${dsdcc_HEADERS} DESTINATION include/${PROJECT_NAME})
