--- origsrc/dsdcc-1.8.7/CMakeLists.txt	2020-06-16 09:25:37.000000000 +0900
+++ src/dsdcc-1.8.7/CMakeLists.txt	2020-07-06 15:47:28.680041500 +0900
@@ -14,7 +14,11 @@ option(BUILD_TOOL "Build dsdccx tool" ON
 # use c++11
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
+if(CYGWIN)
+    set(CMAKE_CXX_EXTENSIONS ON)
+else()
+    set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
 
 # use, i.e. don't skip the full RPATH for the build tree
 set(CMAKE_SKIP_BUILD_RPATH  FALSE)
@@ -221,5 +225,5 @@ INSTALL(
 if(BUILD_TOOL)
     install(TARGETS dsdccx DESTINATION bin)
 endif(BUILD_TOOL)
-install(TARGETS dsdcc DESTINATION ${LIB_INSTALL_DIR})
+install(TARGETS dsdcc RUNTIME DESTINATION bin ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
 install(FILES ${dsdcc_HEADERS} DESTINATION include/${PROJECT_NAME})
