--- origsrc/libhttpseverywhere-0.8.3/meson.build	2018-04-14 05:53:47.000000000 +0900
+++ src/libhttpseverywhere-0.8.3/meson.build	2025-03-15 12:22:15.829754800 +0900
@@ -49,8 +49,7 @@ pkgconfig.generate(libraries : httpsever
                    filebase : meson.current_build_dir()+'/httpseverywhere-'+api,
                    requires : 'glib-2.0 gobject-2.0 gio-2.0 json-glib-1.0 libsoup-2.4 gee-0.8 libarchive',
                    subdirs: 'httpseverywhere-'+api,
-                   description : 'A library to rewrite HTTP URLs to HTTPS URLs.',
-                   install: true)
+                   description : 'A library to rewrite HTTP URLs to HTTPS URLs.')
 
 install_data('data/default.rulesets', install_dir: get_option('datadir') + '/libhttpseverywhere')
 install_data('httpseverywhere-'+api+'.deps', install_dir: get_option('datadir') + '/vala/vapi')
