--- origsrc/metaio-8.5.1/src/Makefile.am	2014-03-11 10:30:07.000000000 +0900
+++ src/metaio-8.5.1/src/Makefile.am	2025-11-23 23:03:39.248642200 +0900
@@ -14,7 +14,7 @@ check_PROGRAMS = parse_test parse_test_t
 TESTS = metaio_test.sh
 
 lib_LTLIBRARIES = libmetaio.la
-libmetaio_la_LDFLAGS = -version-info $(LIBVERSION)
+libmetaio_la_LDFLAGS = -no-undefined -version-info $(LIBVERSION)
 
 lwtscan_SOURCES = lwtscan.c metaio.h
 lwtscan_LDADD = libmetaio.la
--- origsrc/metaio-8.5.1/src/lwtprint.c	2013-12-10 06:15:49.000000000 +0900
+++ src/metaio-8.5.1/src/lwtprint.c	2025-11-23 23:03:39.256553600 +0900
@@ -509,7 +509,7 @@ int main( int argc, char **argv )
       for ( i=0; i<ncols; i++ ) {
 	icol = collist[i];
 	if ( i > 0 )
-          printf( delim );
+          fputs( delim, stdout );
 	if ( icol == PSEUDOCOL_ROW )
 	  fprintf( stdout, "%d", irow );
 	else
