--- origsrc/metaio-8.3.0/src/Makefile.am	2012-11-13 04:23:42.000000000 +0900
+++ src/metaio-8.3.0/src/Makefile.am	2015-06-17 21:30:55.044906600 +0900
@@ -11,11 +11,11 @@ bin_SCRIPTS = lwtselect concatMeta
 include_HEADERS = metaio.h ligo_lw_header.h
 
 check_PROGRAMS = parse_test parse_test_table_only
-TESTS_ENVIRONMENT = /bin/sh 
+TESTS_ENVIRONMENT = /bin/sh
 TESTS = metaio_test.sh
 
 lib_LTLIBRARIES = libmetaio.la
-libmetaio_la_LDFLAGS = -version-info $(LIBVERSION)
+libmetaio_la_LDFLAGS = -no-undefined -version-info $(LIBVERSION)
 
 lwtscan_SOURCES = lwtscan.c metaio.h
 lwtscan_LDADD = libmetaio.la
