--- origsrc/cpu_features-0.9.0/CMakeLists.txt	2023-09-14 21:15:33.000000000 +0900
+++ src/cpu_features-0.9.0/CMakeLists.txt	2023-09-22 09:33:01.464642600 +0900
@@ -182,6 +182,10 @@ if(APPLE)
 endif()
 add_library(CpuFeatures::cpu_features ALIAS cpu_features)
 
+if (CYGWIN)
+  set_target_properties(cpu_features PROPERTIES SOVERSION "${CMAKE_PROJECT_VERSION_MAJOR}")
+endif()
+
 #
 # program : list_cpu_features
 #
@@ -206,10 +210,12 @@ endif()
 
 include(CTest)
 if(BUILD_TESTING)
+  find_package(gtest)
   # Automatically incorporate googletest into the CMake Project if target not
   # found.
   enable_language(CXX)
 
+  if (NOT GTEST_FOUND)
   set(CMAKE_CXX_STANDARD 14)
   set(CMAKE_CXX_STANDARD_REQUIRED ON)
   set(CMAKE_CXX_EXTENSIONS OFF) # prefer use of -std14 instead of -gnustd14
@@ -249,6 +255,7 @@ if(BUILD_TESTING)
                      ${CMAKE_BINARY_DIR}/googletest-build
                      EXCLUDE_FROM_ALL)
   endif()
+  endif()
 
   add_subdirectory(test)
 endif()
@@ -300,3 +307,10 @@ if(ENABLE_INSTALL)
     COMPONENT Devel
   )
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/cpu_features.pc.in ${CMAKE_BINARY_DIR}/cpu_features.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/cpu_features.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/cpu_features-0.9.0/cpu_features.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/cpu_features-0.9.0/cpu_features.pc.in	2023-09-22 09:33:22.638862400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: cpu_features
+Description: Cross platform C99 library to get cpu features at runtime
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lcpu_features
+Cflags: -I${includedir}
--- origsrc/cpu_features-0.9.0/src/impl_x86_linux_or_android.c	2023-09-14 21:15:33.000000000 +0900
+++ src/cpu_features-0.9.0/src/impl_x86_linux_or_android.c	2023-09-22 09:22:46.563281800 +0900
@@ -15,7 +15,7 @@
 #include "cpu_features_macros.h"
 
 #ifdef CPU_FEATURES_ARCH_X86
-#if defined(CPU_FEATURES_OS_LINUX) || defined(CPU_FEATURES_OS_ANDROID)
+#if defined(CPU_FEATURES_OS_LINUX) || defined(CPU_FEATURES_OS_ANDROID) || defined(__CYGWIN__)
 
 #include "impl_x86__base_implementation.inl"
 
--- origsrc/cpu_features-0.9.0/test/CMakeLists.txt	2023-09-14 21:15:33.000000000 +0900
+++ src/cpu_features-0.9.0/test/CMakeLists.txt	2023-09-22 09:23:23.300978500 +0900
@@ -16,7 +16,7 @@ target_link_libraries(hwcaps_for_testing
 ##------------------------------------------------------------------------------
 add_library(stack_line_reader ../src/stack_line_reader.c)
 target_compile_definitions(stack_line_reader PUBLIC STACK_LINE_READER_BUFFER_SIZE=1024)
-target_link_libraries(stack_line_reader string_view)
+target_link_libraries(stack_line_reader string_view filesystem_for_testing)
 ##------------------------------------------------------------------------------
 add_library(stack_line_reader_for_test ../src/stack_line_reader.c)
 target_compile_definitions(stack_line_reader_for_test PUBLIC STACK_LINE_READER_BUFFER_SIZE=16)
