--- origsrc/libresource/.version	1970-01-01 09:00:00.000000000 +0900
+++ src/libresource/.version	2012-03-31 23:05:13.410981300 +0900
@@ -0,0 +1 @@
+0.21.0+git20120227
--- origsrc/libresource/configure.ac	2012-03-31 22:51:14.000000000 +0900
+++ src/libresource/configure.ac	2012-03-31 23:03:37.504731300 +0900
@@ -3,7 +3,7 @@
 AC_PREREQ(2.59)
 
 AC_INIT([libresource],
-	m4_esyscmd([build-aux/git-version-gen .tarball-version]),
+	[0.21.0+git20120227],
 	[multimedia at ok.research.nokia.com])
 
 AC_CONFIG_SRCDIR([src])
@@ -21,7 +21,6 @@ AC_PROG_GCC_TRADITIONAL
 AC_GNU_SOURCE
 
 AC_HEADER_STDC
-AC_INCLUDES_DEFAULT
 
 m4_define(version_major, `echo $VERSION | cut -d. -f1 | cut -d- -f1`)
 m4_define(version_minor, `echo $VERSION | cut -d. -f2 | cut -d- -f1`)
--- origsrc/libresource/src/Makefile.am	2012-03-31 22:51:14.000000000 +0900
+++ src/libresource/src/Makefile.am	2012-03-31 23:06:33.888520300 +0900
@@ -8,7 +8,7 @@ libresource_la_CFLAGS = @DBUS_CFLAGS@ -I
 if DEBUG
 libresource_la_CFLAGS += -D__DEBUG__
 endif
-libresource_la_LDFLAGS = -version-info @LIBRESOURCE_VERSION_INFO@
+libresource_la_LDFLAGS = -version-info @LIBRESOURCE_VERSION_INFO@ -no-undefined
 libresource_la_LIBADD = @DBUS_LIBS@
 
 libresource_glib_la_SOURCES = resource.c resource-glib-glue.c
@@ -17,8 +17,8 @@ libresource_glib_la_CFLAGS = @DBUS_CFLAG
 if DEBUG
 libresource_glib_la_CFLAGS += -D__DEBUG__
 endif
-libresource_glib_la_LDFLAGS = -version-info @LIBRESOURCE_VERSION_INFO@
-libresource_glib_la_LIBADD = $(top_builddir)/src/libresource.la @DBUS_LIBS@
+libresource_glib_la_LDFLAGS = -version-info @LIBRESOURCE_VERSION_INFO@ -no-undefined
+libresource_glib_la_LIBADD = libresource.la @DBUS_LIBS@
 
 
 pkgincludedir = $(includedir)/resource
--- origsrc/libresource/tests/Makefile.am	2012-03-31 22:51:14.000000000 +0900
+++ src/libresource/tests/Makefile.am	2012-03-31 22:59:11.803559400 +0900
@@ -13,4 +13,4 @@ memory_leak_test_LDADD   = $(top_builddi
                 		   $(top_builddir)/src/libresource.la \
                            @DBUS_LIBS@
 
-noinst_PROGRAMS = resource_test memory_leak_test
+noinst_PROGRAMS = resource-test memory-leak-test
