--- origsrc/snappy-1.1.7/CMakeLists.txt	2017-08-25 08:54:23.000000000 +0900
+++ src/snappy-1.1.7/CMakeLists.txt	2017-09-28 16:55:59.040000000 +0900
@@ -118,7 +118,7 @@ if(SNAPPY_BUILD_TESTS)
       "${PROJECT_SOURCE_DIR}/snappy-test.cc"
   )
   target_compile_definitions(snappy_unittest PRIVATE -DHAVE_CONFIG_H)
-  target_link_libraries(snappy_unittest snappy ${GFLAGS_LIBRARIES})
+  target_link_libraries(snappy_unittest snappy ${GFLAGS_LIBRARIES} ${GTEST_LIBRARIES})
 
   if(HAVE_LIBZ)
     target_link_libraries(snappy_unittest z)
@@ -172,3 +172,10 @@ install(
     "${PROJECT_BINARY_DIR}/SnappyConfigVersion.cmake"
   DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/Snappy"
 )
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/snappy.pc.in ${CMAKE_BINARY_DIR}/snappy.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/snappy.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/snappy-1.1.7/snappy.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/snappy-1.1.7/snappy.pc.in	2017-09-28 17:00:15.196000000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: snappy
+Description: Fast compressor/decompressor
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lsnappy
+Cflags: -I${includedir}
