--- origsrc/zipkin-cpp-opentracing-0.5.2/zipkin/CMakeLists.txt	2018-08-17 23:32:47.000000000 +0900
+++ src/zipkin-cpp-opentracing-0.5.2/zipkin/CMakeLists.txt	2019-06-25 14:02:20.530107300 +0900
@@ -23,7 +23,8 @@ if (BUILD_SHARED_LIBS)
                                           SOVERSION ${ZIPKIN_VERSION_MAJOR})
   install(TARGETS zipkin 
           COMPONENT DIST
-          LIBRARY DESTINATION lib)
+          RUNTIME DESTINATION bin
+          ARCHIVE DESTINATION lib)
 endif()
 
 if (BUILD_STATIC_LIBS)
--- origsrc/zipkin-cpp-opentracing-0.5.2/zipkin_opentracing/CMakeLists.txt	2018-08-17 23:32:47.000000000 +0900
+++ src/zipkin-cpp-opentracing-0.5.2/zipkin_opentracing/CMakeLists.txt	2019-06-25 14:02:49.488648000 +0900
@@ -30,7 +30,8 @@ if (BUILD_SHARED_LIBS)
                                                       SOVERSION ${ZIPKIN_VERSION_MAJOR})
   install(TARGETS zipkin_opentracing 
           COMPONENT DIST
-          LIBRARY DESTINATION lib)
+          RUNTIME DESTINATION bin
+          ARCHIVE DESTINATION lib)
 endif()
 
 if (BUILD_STATIC_LIBS)
