--- origsrc/buddy-2.4/Makefile.am	2004-07-11 02:12:23.000000000 +0900
+++ src/buddy-2.4/Makefile.am	2020-06-27 19:40:39.466593500 +0900
@@ -3,3 +3,6 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = src examples doc
 
 EXTRA_DIST = CHANGES m4/debug.m4
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/buddy-2.4/buddy.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/buddy-2.4/buddy.pc.in	2020-06-27 19:41:01.862036500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: buddy
+Description: Binary Decision Diagram library
+Version: @VERSION@
+Libs: -L${libdir} -lbdd
+Cflags: -I${includedir}
--- origsrc/buddy-2.4/configure.ac	2004-07-11 02:46:37.000000000 +0900
+++ src/buddy-2.4/configure.ac	2020-06-27 19:40:39.482221300 +0900
@@ -21,7 +21,7 @@ buddy_DEBUG_FLAGS
 
 AC_CONFIG_HEADERS([config.h])
 
-AC_CONFIG_FILES([
+AC_CONFIG_FILES([buddy.pc 
   Makefile
   src/Makefile
   doc/Makefile
--- origsrc/buddy-2.4/src/Makefile.am	2004-07-11 02:12:23.000000000 +0900
+++ src/buddy-2.4/src/Makefile.am	2020-06-27 19:33:18.525243300 +0900
@@ -24,7 +24,7 @@ libbdd_la_SOURCES = \
 
 # See the `Updating version info' node of the Libtool manual before
 # changing this.
-libbdd_la_LDFLAGS = -version-info 0:0:0
+libbdd_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 
 check_PROGRAMS = bddtest
 bddtest_SOURCES = bddtest.cxx
