--- origsrc/mkl-dnn-0.19/CMakeLists.txt	2019-05-15 07:03:33.000000000 +0900
+++ src/mkl-dnn-0.19/CMakeLists.txt	2019-06-07 16:49:51.728123600 +0900
@@ -90,9 +90,18 @@ add_subdirectory(src)
 add_subdirectory(examples)
 add_subdirectory(tests)
 
-if(MKLDNN_INSTALL_MODE STREQUAL "BUNDLE")
-    install(FILES LICENSE DESTINATION ${CMAKE_INSTALL_PREFIX})
-else()
-    # Cannot use CMAKE_INSTALL_DOCDIR since it uses PROJECT_NAME and not LIB_NAME
-    install(FILES LICENSE DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/doc/${LIB_NAME})
+if(NOT CYGWIN)
+    if(MKLDNN_INSTALL_MODE STREQUAL "BUNDLE")
+        install(FILES LICENSE DESTINATION ${CMAKE_INSTALL_PREFIX})
+    else()
+        # Cannot use CMAKE_INSTALL_DOCDIR since it uses PROJECT_NAME and not LIB_NAME
+        install(FILES LICENSE DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/doc/${LIB_NAME})
+    endif()
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/mkl-dnn.pc.in ${CMAKE_BINARY_DIR}/mkl-dnn.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/mkl-dnn.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/mkl-dnn-0.19/cmake/SDL.cmake	2019-05-15 07:03:33.000000000 +0900
+++ src/mkl-dnn-0.19/cmake/SDL.cmake	2019-06-07 15:06:38.586524700 +0900
@@ -52,7 +52,7 @@ if(UNIX)
     if(APPLE)
         append(CMAKE_SHARED_LINKER_FLAGS "-Wl,-bind_at_load")
         append(CMAKE_EXE_LINKER_FLAGS "-Wl,-bind_at_load")
-    else()
+    elseif(NOT CYGWIN)
         append(CMAKE_EXE_LINKER_FLAGS "-pie")
         append(CMAKE_SHARED_LINKER_FLAGS "-Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now")
         append(CMAKE_EXE_LINKER_FLAGS "-Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now")
--- origsrc/mkl-dnn-0.19/mkl-dnn.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/mkl-dnn-0.19/mkl-dnn.pc.in	2019-06-07 16:51:42.202590100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mkl-dnn
+Description: Intel(R) Math Kernel Library for Deep Neural Networks
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lmkldnn
+Cflags: -I${includedir}
--- origsrc/mkl-dnn-0.19/tests/gtests/gtest/CMakeLists.txt	2019-05-15 07:03:33.000000000 +0900
+++ src/mkl-dnn-0.19/tests/gtests/gtest/CMakeLists.txt	2019-06-07 15:08:19.933160800 +0900
@@ -6,6 +6,8 @@ source_group("" FILES ${MAIN_SRC})
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}
                     ${CMAKE_CURRENT_SOURCE_DIR}/..
   )
-find_package(Threads REQUIRED)
+if(NOT CYGWIN)
+  find_package(Threads REQUIRED)
+endif()
 add_library(${TARGET_NAME} STATIC ${MAIN_SRC})
 target_link_libraries(${TARGET_NAME} ${CMAKE_THREAD_LIBS_INIT})
