--- origsrc/mkl-dnn-0.13/CMakeLists.txt	2018-03-04 04:15:32.000000000 +0900
+++ src/mkl-dnn-0.13/CMakeLists.txt	2018-04-09 10:13:20.112000000 +0900
@@ -68,4 +68,13 @@ add_subdirectory(src)
 add_subdirectory(examples)
 add_subdirectory(tests)
 
-install(FILES LICENSE DESTINATION share/doc/${LIB_NAME})
+if(NOT CYGWIN)
+    install(FILES LICENSE DESTINATION share/doc/${LIB_NAME})
+endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/mkl-dnn.pc.in ${CMAKE_BINARY_DIR}/mkl-dnn.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/mkl-dnn.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/mkl-dnn-0.13/cmake/SDL.cmake	2018-03-04 04:15:32.000000000 +0900
+++ src/mkl-dnn-0.13/cmake/SDL.cmake	2018-04-09 09:54:39.821903700 +0900
@@ -42,7 +42,7 @@ if(UNIX OR APPLE)
     if(APPLE)
         set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-bind_at_load")
         set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-bind_at_load")
-    else()
+    elseif(NOT CYGWIN)
         set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -pie")
         set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now")
         set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now")
--- origsrc/mkl-dnn-0.13/mkl-dnn.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/mkl-dnn-0.13/mkl-dnn.pc.in	2018-04-09 10:12:49.233000000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mkl-dnn
+Description: Intel(R) Math Kernel Library for Deep Neural Networks
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lmkldnn
+Cflags: -I${includedir}
--- origsrc/mkl-dnn-0.13/src/CMakeLists.txt	2018-03-04 04:15:32.000000000 +0900
+++ src/mkl-dnn-0.13/src/CMakeLists.txt	2018-04-09 10:14:14.688999800 +0900
@@ -55,5 +55,5 @@ set_property(TARGET ${TARGET_NAME} PROPE
 set_property(TARGET ${TARGET_NAME} PROPERTY VERSION "${PROJECT_VERSION}.0")
 set_property(TARGET ${TARGET_NAME} PROPERTY SOVERSION "0")
 
-install(TARGETS ${TARGET_NAME} DESTINATION lib)
+install(TARGETS ${TARGET_NAME} RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 install(FILES ${HEADERS} DESTINATION include)
--- origsrc/mkl-dnn-0.13/tests/gtests/gtest/CMakeLists.txt	2018-03-04 04:15:32.000000000 +0900
+++ src/mkl-dnn-0.13/tests/gtests/gtest/CMakeLists.txt	2018-04-09 09:26:37.073000000 +0900
@@ -6,6 +6,8 @@ source_group("" FILES ${MAIN_SRC})
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}
                     ${CMAKE_CURRENT_SOURCE_DIR}/..
   )
-find_package(Threads REQUIRED)
+if(NOT CYGWIN)
+  find_package(Threads REQUIRED)
+endif()
 add_library(${TARGET_NAME} STATIC ${MAIN_SRC})
 target_link_libraries(${TARGET_NAME} ${CMAKE_THREAD_LIBS_INIT})
