--- origsrc/mkl-dnn-1.1.1/CMakeLists.txt	2019-10-22 23:23:33.000000000 +0900
+++ src/mkl-dnn-1.1.1/CMakeLists.txt	2019-12-05 08:39:05.333962300 +0900
@@ -117,10 +117,19 @@ add_subdirectory(src)
 add_subdirectory(examples)
 add_subdirectory(tests)
 
-if(DNNL_INSTALL_MODE STREQUAL "BUNDLE")
-    install(FILES LICENSE DESTINATION ${CMAKE_INSTALL_PREFIX})
-    install(FILES README.md DESTINATION ${CMAKE_INSTALL_PREFIX})
-else()
-    # Cannot use CMAKE_INSTALL_DOCDIR since it uses PROJECT_NAME and not LIB_NAME
-    install(FILES LICENSE DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/doc/${LIB_NAME})
+if(NOT CYGWIN)
+    if(DNNL_INSTALL_MODE STREQUAL "BUNDLE")
+        install(FILES LICENSE DESTINATION ${CMAKE_INSTALL_PREFIX})
+        install(FILES README.md DESTINATION ${CMAKE_INSTALL_PREFIX})
+    else()
+        # Cannot use CMAKE_INSTALL_DOCDIR since it uses PROJECT_NAME and not LIB_NAME
+        install(FILES LICENSE DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/doc/${LIB_NAME})
+    endif()
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/mkl-dnn.pc.in ${CMAKE_BINARY_DIR}/mkl-dnn.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/mkl-dnn.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/mkl-dnn-1.1.1/cmake/SDL.cmake	2019-10-22 23:23:33.000000000 +0900
+++ src/mkl-dnn-1.1.1/cmake/SDL.cmake	2019-12-05 08:39:05.333962300 +0900
@@ -52,7 +52,7 @@ if(UNIX)
     if(APPLE)
         append(CMAKE_SHARED_LINKER_FLAGS "-Wl,-bind_at_load")
         append(CMAKE_EXE_LINKER_FLAGS "-Wl,-bind_at_load")
-    else()
+    elseif(NOT CYGWIN)
         append(CMAKE_EXE_LINKER_FLAGS "-pie")
         append(CMAKE_SHARED_LINKER_FLAGS "-Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now")
         append(CMAKE_EXE_LINKER_FLAGS "-Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now")
--- origsrc/mkl-dnn-1.1.1/mkl-dnn.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/mkl-dnn-1.1.1/mkl-dnn.pc.in	2019-12-05 08:39:05.349584500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mkl-dnn
+Description: Intel(R) Math Kernel Library for Deep Neural Networks
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -ldnnl
+Cflags: -I${includedir}
--- origsrc/mkl-dnn-1.1.1/src/CMakeLists.txt	2019-10-22 23:23:33.000000000 +0900
+++ src/mkl-dnn-1.1.1/src/CMakeLists.txt	2019-12-05 08:39:05.349584500 +0900
@@ -140,7 +140,7 @@ if (WIN32)
         DEPENDS ${LIB_NAME})
     install(FILES $<TARGET_FILE_DIR:${LIB_NAME}>/mkldnn.lib
         DESTINATION ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
-else()
+elseif(NOT CYGWIN)
     if(DNNL_LIBRARY_TYPE STREQUAL "SHARED")
         set_ternary(ext APPLE "dylib" "so")
         set(vers ".${DNNL_VERSION_MAJOR};.${DNNL_VERSION_MAJOR}.${DNNL_VERSION_MINOR}")
--- origsrc/mkl-dnn-1.1.1/src/common/tag_traits.hpp	2019-10-22 23:23:33.000000000 +0900
+++ src/mkl-dnn-1.1.1/src/common/tag_traits.hpp	2019-12-05 08:39:05.365206100 +0900
@@ -22,6 +22,10 @@
 #include "c_types_map.hpp"
 #include "utils.hpp"
 
+#if defined(_B)
+#undef _B
+#endif
+
 namespace dnnl {
 namespace impl {
 
