--- origsrc/mkl-dnn-0.17/CMakeLists.txt	2018-11-17 09:02:10.000000000 +0900
+++ src/mkl-dnn-0.17/CMakeLists.txt	2018-11-25 12:40:56.537893900 +0900
@@ -82,4 +82,13 @@ add_subdirectory(src)
 add_subdirectory(examples)
 add_subdirectory(tests)
 
-install(FILES LICENSE DESTINATION share/doc/${LIB_NAME})
+if(NOT CYGWIN)
+    install(FILES LICENSE DESTINATION share/doc/${LIB_NAME})
+endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/mkl-dnn.pc.in ${CMAKE_BINARY_DIR}/mkl-dnn.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/mkl-dnn.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/mkl-dnn-0.17/cmake/SDL.cmake	2018-11-17 09:02:10.000000000 +0900
+++ src/mkl-dnn-0.17/cmake/SDL.cmake	2018-11-25 12:40:56.548881800 +0900
@@ -51,7 +51,7 @@ if(UNIX)
     if(APPLE)
         set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-bind_at_load")
         set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-bind_at_load")
-    else()
+    elseif(NOT CYGWIN)
         set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -pie")
         set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now")
         set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now")
--- origsrc/mkl-dnn-0.17/mkl-dnn.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/mkl-dnn-0.17/mkl-dnn.pc.in	2018-11-25 12:40:56.558808000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mkl-dnn
+Description: Intel(R) Math Kernel Library for Deep Neural Networks
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lmkldnn
+Cflags: -I${includedir}
--- origsrc/mkl-dnn-0.17/src/CMakeLists.txt	2018-11-17 09:02:10.000000000 +0900
+++ src/mkl-dnn-0.17/src/CMakeLists.txt	2018-11-25 12:40:56.569800000 +0900
@@ -103,7 +103,7 @@ set_property(TARGET ${TARGET_NAME} PROPE
 set_property(TARGET ${TARGET_NAME} PROPERTY VERSION "${PROJECT_VERSION}.0")
 set_property(TARGET ${TARGET_NAME} PROPERTY SOVERSION "0")
 
-if(MINGW)
+if(MINGW OR CYGWIN)
     # We need to install *.dll into bin/ and *.a into lib/. 
     install(TARGETS ${TARGET_NAME}
         RUNTIME DESTINATION bin
--- origsrc/mkl-dnn-0.17/tests/gtests/gtest/CMakeLists.txt	2018-11-17 09:02:10.000000000 +0900
+++ src/mkl-dnn-0.17/tests/gtests/gtest/CMakeLists.txt	2018-11-25 12:40:56.582794100 +0900
@@ -6,6 +6,8 @@ source_group("" FILES ${MAIN_SRC})
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}
                     ${CMAKE_CURRENT_SOURCE_DIR}/..
   )
-find_package(Threads REQUIRED)
+if(NOT CYGWIN)
+  find_package(Threads REQUIRED)
+endif()
 add_library(${TARGET_NAME} STATIC ${MAIN_SRC})
 target_link_libraries(${TARGET_NAME} ${CMAKE_THREAD_LIBS_INIT})
