--- origsrc/libbrahe-1.2.0/Makefile.am	2009-11-18 14:34:41.000000000 +0900
+++ src/libbrahe-1.2.0/Makefile.am	2010-11-30 00:21:46.232580300 +0900
@@ -1,4 +1,4 @@
-ACLOCAL_AMFLAGS = -I m4
+#ACLOCAL_AMFLAGS = -I m4
 
 if HAVE_DOXYGEN
 DOC_DIR = docs
@@ -6,6 +6,6 @@ else
 DOC_DIR =
 endif
 
-SUBDIRS = src test $(DOC_DIR)
+SUBDIRS = src test #$(DOC_DIR)
 EXTRA_DIST = LICENSE.GPL LICENSE.POLICY LICENSE.NON-FREE reconf
 
--- origsrc/libbrahe-1.2.0/configure.ac	2009-11-18 09:33:37.000000000 +0900
+++ src/libbrahe-1.2.0/configure.ac	2010-11-29 10:12:12.084602500 +0900
@@ -27,7 +27,7 @@ AM_INIT_AUTOMAKE($PACKAGE, $VERSION, [no
 
 AC_PROG_CC
 AC_PROG_INSTALL
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AM_SANITY_CHECK
 AC_HEADER_STDC
 AC_HEADER_STDBOOL
--- origsrc/libbrahe-1.2.0/src/Makefile.am	2009-11-18 01:34:15.000000000 +0900
+++ src/libbrahe-1.2.0/src/Makefile.am	2010-11-29 10:12:22.348716900 +0900
@@ -7,7 +7,7 @@ c_sources = trig.c rounding.c gcflcm.c p
 lib_LTLIBRARIES = libbrahe.la
 
 libbrahe_la_SOURCES = $(h_sources) $(c_sources)
-libbrahe_la_LDFLAGS= -version-info $(GENERIC_LIBRARY_VERSION) -release $(GENERIC_RELEASE)
+libbrahe_la_LDFLAGS= -version-info $(GENERIC_LIBRARY_VERSION) -release $(GENERIC_RELEASE) -no-undefined
 
 library_includedir=$(includedir)/$(GENERIC_LIBRARY_NAME)
 library_include_HEADERS = $(h_sources)
--- origsrc/libbrahe-1.2.0/test/Makefile.am	2009-11-18 02:26:52.000000000 +0900
+++ src/libbrahe-1.2.0/test/Makefile.am	2010-11-30 00:30:20.411933900 +0900
@@ -8,3 +8,5 @@ brahe_test_fft_SOURCES = brahe_test_fft.
 
 LIBS = -L../src -lbrahe -lm -lrt
 AM_CFLAGS = -std=gnu99 -pedantic -Wall -Wno-format
+
+TESTS = $(bin_PROGRAMS)
