--- origsrc/z3-z3-4.11.2/src/shell/CMakeLists.txt	2022-09-04 07:47:12.000000000 +0900
+++ src/z3-z3-4.11.2/src/shell/CMakeLists.txt	2022-10-29 22:53:13.126270000 +0900
@@ -11,12 +11,14 @@ set (shell_object_files "")
 set(shell_deps api extra_cmds opt sat)
 z3_expand_dependencies(shell_expanded_deps ${shell_deps})
 get_property(Z3_LIBZ3_COMPONENTS_LIST GLOBAL PROPERTY Z3_LIBZ3_COMPONENTS)
+if (NOT CYGWIN)
 foreach (component ${Z3_LIBZ3_COMPONENTS_LIST})
   if (NOT ("${component}" STREQUAL "api_dll"))
     # We don't use the api_dll component in the Z3 executable
     list(APPEND shell_object_files $<TARGET_OBJECTS:${component}>)
   endif()
 endforeach()
+endif()
 add_executable(shell
   datalog_frontend.cpp
   dimacs_frontend.cpp
@@ -50,6 +52,9 @@ target_compile_definitions(shell PRIVATE
 target_compile_options(shell PRIVATE ${Z3_COMPONENT_CXX_FLAGS})
 target_include_directories(shell PRIVATE ${Z3_COMPONENT_EXTRA_INCLUDE_DIRS})
 target_link_libraries(shell PRIVATE ${Z3_DEPENDENT_LIBS})
+if (CYGWIN)
+  target_link_libraries(shell PRIVATE libz3)
+endif()
 z3_add_component_dependencies_to_target(shell ${shell_expanded_deps})
 z3_append_linker_flag_list_to_target(shell ${Z3_DEPENDENT_EXTRA_CXX_LINK_FLAGS})
 install(TARGETS shell
