--- origsrc/hstr-3.1/src/hstr_utils.c	2023-04-19 06:26:18.000000000 +0900
+++ src/hstr-3.1/src/hstr_utils.c	2023-09-29 08:34:26.373013600 +0900
@@ -161,6 +161,7 @@ void fill_terminal_input(char* cmd, bool
 {
     if(cmd && strlen(cmd)>0) {
             if(is_tiocsti) {
+#if !defined(__MS_WSL__) && !defined(__CYGWIN__)
             size_t size = strlen(cmd);
             unsigned i;
             char *c;
@@ -169,6 +170,7 @@ void fill_terminal_input(char* cmd, bool
                 c=(cmd+i);
                 ioctl(0, TIOCSTI, c);
             }
+#endif
             // echo, but don't flush to terminal
             if(padding) printf("\n");
         } else {
