--- origsrc/hstr-1.16/src/hstr_utils.c	2015-02-27 07:38:44.000000000 +0900
+++ src/hstr-1.16/src/hstr_utils.c	2015-03-20 00:46:31.494407100 +0900
@@ -68,9 +68,11 @@ void tiocsti()
 {
     char buf[] = DEFAULT_COMMAND;
     int i;
+#if defined(TIOCSTI)
     for (i = 0; i < sizeof buf - 1; i++) {
         ioctl(0, TIOCSTI, &buf[i]);
     }
+#endif
 }
 
 void fill_terminal_input(char *cmd, bool padding)
@@ -79,11 +81,13 @@ void fill_terminal_input(char *cmd, bool
         size_t size = strlen(cmd);
         unsigned i;
         char *c;
+#if defined(TIOCSTI)
         for (i = 0; i < size; i++) {
             // terminal I/O control, simulate terminal input
             c=(cmd+i);
             ioctl(0, TIOCSTI, c);
         }
+#endif
         // echo, but don't flush to terminal
         if(padding) printf("\n");
     }
