--- origsrc/hstr-1.24/src/hstr_utils.c	2018-01-03 20:41:58.000000000 +0900
+++ src/hstr-1.24/src/hstr_utils.c	2018-04-13 18:31:28.733929600 +0900
@@ -78,9 +78,11 @@ void tiocsti()
 {
     char buf[] = DEFAULT_COMMAND;
     int i;
+#if defined(TIOCSTI)
     for (i = 0; i < sizeof buf - 1; i++) {
         ioctl(0, TIOCSTI, &buf[i]);
     }
+#endif
 }
 #endif
 
@@ -94,11 +96,13 @@ void fill_terminal_input(char *cmd, bool
         size_t size = strlen(cmd);
         unsigned i;
         char *c;
+#if defined(TIOCSTI)
         for (i = 0; i < size; i++) {
             // terminal I/O control, simulate terminal input
             c=(cmd+i);
             ioctl(0, TIOCSTI, c);
         }
+#endif
         // echo, but don't flush to terminal
         if(padding) printf("\n");
 #endif
