--- origsrc/ctemplate-0.98/Makefile.am	2010-09-25 04:00:51.000000000 +0900
+++ src/ctemplate-0.98/Makefile.am	2010-12-12 01:54:47.027502900 +0900
@@ -124,7 +124,7 @@ libctemplate_la_SOURCES = $(nodist_ctemp
 	src/htmlparser/statemachine.h src/htmlparser/statemachine.cc \
 	src/htmlparser/htmlparser_cpp.h
 libctemplate_la_CXXFLAGS = $(PTHREAD_CFLAGS) -DNDEBUG $(AM_CXXFLAGS)
-libctemplate_la_LDFLAGS = $(PTHREAD_CFLAGS) -export-symbols-regex $(CTEMPLATE_SYMBOLS)
+libctemplate_la_LDFLAGS = $(PTHREAD_CFLAGS) -export-symbols-regex $(CTEMPLATE_SYMBOLS) -no-undefined
 libctemplate_la_LIBADD = $(PTHREAD_LIBS)
 
 # For MinGW, we need to bring in the windows port files
@@ -138,7 +138,7 @@ lib_LTLIBRARIES += libctemplate_nothread
 libctemplate_nothreads_la_SOURCES = $(libctemplate_la_SOURCES)
 libctemplate_nothreads_la_DEPENDENCIES = $(libctemplate_la_DEPENDENCIES)
 libctemplate_nothreads_la_CXXFLAGS = -DNDEBUG -DNO_THREADS $(AM_CXXFLAGS)
-libctemplate_nothreads_la_LDFLAGS = -export-symbols-regex $(CTEMPLATE_SYMBOLS)
+libctemplate_nothreads_la_LDFLAGS = -export-symbols-regex $(CTEMPLATE_SYMBOLS) -no-undefined
 
 # For our tests, we want versions of these libraries that include asserts.
 noinst_LTLIBRARIES += libctemplate_debug.la
--- origsrc/ctemplate-0.98/configure.ac	2010-09-23 03:36:57.000000000 +0900
+++ src/ctemplate-0.98/configure.ac	2010-12-12 01:55:15.248082100 +0900
@@ -42,7 +42,7 @@ esac
 AM_CONDITIONAL(MINGW, expr $host : '.*-mingw' >/dev/null 2>&1)
 
 # Uncomment this if you'll be exporting libraries (.so's)
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_SUBST(LIBTOOL_DEPS)
 
 AC_PROG_LN_S
