--- origsrc/ctemplate/Makefile.am	2012-05-22 06:53:01.000000000 +0900
+++ src/ctemplate/Makefile.am	2014-11-14 08:13:29.367372100 +0900
@@ -92,12 +92,15 @@ TESTDATA =
 # These files are auto-generated via generate_fsm.py.  Since we don't want
 # to require python to build ctemplate, we include these in the dist
 src/htmlparser/htmlparser_fsm.h: src/htmlparser/generate_fsm.py src/htmlparser/fsm_config.py src/htmlparser/htmlparser_fsm.config
+	mkdir -p src/htmlparser
 	$(top_srcdir)/src/htmlparser/generate_fsm.py $(top_srcdir)/src/htmlparser/htmlparser_fsm.config > $@
 
 src/htmlparser/jsparser_fsm.h: src/htmlparser/generate_fsm.py src/htmlparser/fsm_config.py src/htmlparser/jsparser_fsm.config
+	mkdir -p src/htmlparser
 	$(top_srcdir)/src/htmlparser/generate_fsm.py $(top_srcdir)/src/htmlparser/jsparser_fsm.config > $@
 
 src/tests/statemachine_test_fsm.h: src/htmlparser/generate_fsm.py src/htmlparser/fsm_config.py src/tests/statemachine_test_fsm.config
+	mkdir -p src/tests
 	$(top_srcdir)/src/htmlparser/generate_fsm.py $(top_srcdir)/src/tests/statemachine_test_fsm.config > $@
 
 # This is a required hack for auto-generated .h files: cf the automake info pages
@@ -405,7 +408,7 @@ TESTDATA += \
 # We want to compile statemachine.cc as a c file as well as a c++ file
 
 statemachine.c: src/htmlparser/statemachine.cc
-	$(LN_S) src/htmlparser/statemachine.cc $@
+	$(LN_S) $(top_srcdir)/src/htmlparser/statemachine.cc $@
 
 BUILT_SOURCES += statemachine.c
 CLEANFILES = statemachine.c
