--- origsrc/ctemplate-0.99/Makefile.am	2011-01-15 06:56:53.000000000 +0900
+++ src/ctemplate-0.99/Makefile.am	2011-05-21 17:13:39.759386500 +0900
@@ -128,7 +128,7 @@ libctemplate_la_CXXFLAGS = $(PTHREAD_CFL
 # -version-info gets passed to libtool
 libctemplate_la_LDFLAGS = $(PTHREAD_CFLAGS) \
                           -export-symbols-regex $(CTEMPLATE_SYMBOLS) \
-                          -version-info @SO_VERSION@
+                          -version-info @SO_VERSION@ -no-undefined
 libctemplate_la_LIBADD = $(PTHREAD_LIBS)
 
 # For MinGW, we need to bring in the windows port files
@@ -143,7 +143,7 @@ libctemplate_nothreads_la_SOURCES = $(li
 libctemplate_nothreads_la_DEPENDENCIES = $(libctemplate_la_DEPENDENCIES)
 libctemplate_nothreads_la_CXXFLAGS = -DNDEBUG -DNO_THREADS $(AM_CXXFLAGS)
 libctemplate_nothreads_la_LDFLAGS = -export-symbols-regex $(CTEMPLATE_SYMBOLS) \
-                                    -version-info @SO_VERSION@
+                                    -version-info @SO_VERSION@ -no-undefined
 
 # For our tests, we want versions of these libraries that include asserts.
 noinst_LTLIBRARIES += libctemplate_debug.la
--- origsrc/ctemplate-0.99/configure.ac	2011-01-22 14:26:37.000000000 +0900
+++ src/ctemplate-0.99/configure.ac	2011-05-21 17:12:58.099482500 +0900
@@ -47,7 +47,7 @@ esac
 AM_CONDITIONAL(MINGW, expr $host : '.*-mingw' >/dev/null 2>&1)
 
 # Uncomment this if you'll be exporting libraries (.so's)
-AC_PROG_LIBTOOL
+LT_INIT
 AC_SUBST(LIBTOOL_DEPS)
 AC_SUBST(SO_VERSION)
 
