--- origsrc/luxio-0.2.2/Makefile.am	2010-02-10 10:44:59.000000000 +0900
+++ src/luxio-0.2.2/Makefile.am	2014-06-29 12:01:28.557471700 +0900
@@ -1,6 +1,9 @@
+SUBDIRS = . test
+
 lib_LTLIBRARIES = libluxio.la
 
 libluxio_la_SOURCES = btree.cpp array.cpp data.h util.h
+libluxio_la_LDFLAGS = -no-undefined
 
 libluxio_includedir = $(includedir)/luxio
 libluxio_include_HEADERS = btree.h array.h dbm.h types.h exception.h luxio-config.h
--- origsrc/luxio-0.2.2/configure.ac	2010-02-10 10:44:59.000000000 +0900
+++ src/luxio-0.2.2/configure.ac	2014-06-29 12:02:20.670752900 +0900
@@ -14,7 +14,7 @@ AC_PROG_CPP
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
-AC_PROG_RANLIB
+LT_INIT
 
 # Checks for libraries.
 AC_PROG_LIBTOOL
@@ -40,5 +40,5 @@ AC_SYS_LARGEFILE
 AC_FUNC_MMAP
 AC_CHECK_FUNCS([ftruncate getpagesize gettimeofday memmove memset mkdir munmap strchr pread pwrite])
 
-AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([Makefile test/Makefile])
 AC_OUTPUT
--- origsrc/luxio-0.2.2/test/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/luxio-0.2.2/test/Makefile.am	2014-06-29 12:01:28.573096700 +0900
@@ -0,0 +1,34 @@
+
+check_PROGRAMS = btree_cluster_test \
+          btree_noncluster_padded_test \
+          btree_noncluster_linked_test \
+          data_test \
+          array_test \
+          btree_thread_test
+
+AM_CPPFLAGS = -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
+AM_LDFLAGS = -fprofile-arcs -ftest-coverage
+
+btree_cluster_test_SOURCES = btree_cluster_test.cpp
+btree_cluster_test_CXXFLAGS = -fprofile-arcs -ftest-coverage
+btree_cluster_test_LDADD = -lgtest $(top_builddir)/libluxio.la
+
+btree_noncluster_padded_test_SOURCES = btree_noncluster_padded_test.cpp
+btree_noncluster_padded_test_CXXFLAGS = -fprofile-arcs -ftest-coverage
+btree_noncluster_padded_test_LDADD = -lgtest $(top_builddir)/libluxio.la
+
+btree_noncluster_linked_test_SOURCES = btree_noncluster_linked_test.cpp
+btree_noncluster_linked_test_CXXFLAGS = -fprofile-arcs -ftest-coverage
+btree_noncluster_linked_test_LDADD = -lgtest $(top_builddir)/libluxio.la
+
+data_test_SOURCES = data_test.cpp
+data_test_CXXFLAGS = -fprofile-arcs -ftest-coverage
+data_test_LDADD = -lgtest $(top_builddir)/libluxio.la
+
+array_test_SOURCES = array_test.cpp
+array_test_CXXFLAGS = -fprofile-arcs -ftest-coverage
+array_test_LDADD = -lgtest $(top_builddir)/libluxio.la
+
+btree_thread_test_SOURCES = btree_thread_test.cpp
+btree_thread_test_CXXFLAGS = -fprofile-arcs -ftest-coverage
+btree_thread_test_LDADD = -lgtest $(top_builddir)/libluxio.la
--- origsrc/luxio-0.2.2/test/btree_cluster_test.cpp	2010-02-10 10:44:59.000000000 +0900
+++ src/luxio-0.2.2/test/btree_cluster_test.cpp	2014-06-29 12:01:28.573096700 +0900
@@ -532,6 +532,7 @@ namespace {
     delete bt;
   }
 
+#if !defined(__CYGWIN__) // hang up
   /* 
    * sequence: ordered
    * key: string
@@ -564,6 +565,7 @@ namespace {
     ASSERT_EQ(true, bt->close());
     delete bt;
   }
+#endif
 
   /* 
    * sequence: ordered
--- origsrc/luxio-0.2.2/util.h	2010-02-10 10:44:59.000000000 +0900
+++ src/luxio-0.2.2/util.h	2014-06-29 12:01:28.573096700 +0900
@@ -26,6 +26,7 @@
 #include <string>
 #include <iostream>
 #include <unistd.h>
+#include <cstdio>
 
 #ifdef DEBUG
 #define error_log(msg) \
