--- origsrc/mini18n-0.2.1/CMakeLists.txt	2025-03-11 16:44:21.059260000 +0900
+++ src/mini18n-0.2.1/CMakeLists.txt	2025-03-11 16:54:13.674903700 +0900
@@ -21,3 +21,10 @@ include(CPack)
 add_subdirectory(doc)
 add_subdirectory(example)
 add_subdirectory(src)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/mini18n.pc.in ${CMAKE_BINARY_DIR}/mini18n.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/mini18n.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/mini18n-0.2.1/mini18n.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/mini18n-0.2.1/mini18n.pc.in	2025-03-11 16:54:24.283658100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mini18n
+Description: Minimal internationalization library
+Version: @MINI18N_VERSION@
+Libs: -L${libdir} -lmini18n
+Cflags: -I${includedir}
--- origsrc/mini18n-0.2.1/src/CMakeLists.txt	2025-03-11 16:44:21.074853500 +0900
+++ src/mini18n-0.2.1/src/CMakeLists.txt	2025-03-11 16:50:03.361593300 +0900
@@ -22,10 +22,7 @@ try_compile(HAVE_WCS_POINTERS ${CMAKE_BI
 
 add_library(mini18n-shared SHARED ${mini18n_SOURCES})
 set_target_properties(mini18n-shared PROPERTIES OUTPUT_NAME mini18n SOVERSION 1 VERSION 1.0.1)
-add_library(mini18n-static STATIC ${mini18n_SOURCES})
-set_target_properties(mini18n-static PROPERTIES OUTPUT_NAME mini18n)
 
-install(TARGETS mini18n-shared DESTINATION "${CMAKE_INSTALL_LIBDIR}")
-install(TARGETS mini18n-static DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+install(TARGETS mini18n-shared RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 install(FILES mini18n.h DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
 install(FILES mini18n-multi.h DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
