--- origsrc/libpaseto-1.0/CMakeLists.txt	2019-07-03 23:51:58.000000000 +0900
+++ src/libpaseto-1.0/CMakeLists.txt	2020-02-07 18:42:53.308257900 +0900
@@ -5,7 +5,7 @@ project(libpaseto
 
 set(CMAKE_C_STANDARD 11)
 set(CMAKE_C_EXTENSIONS OFF)
-add_compile_options(-Wall -Wextra -Wno-unused-parameter -Wno-unused-result -Werror)
+add_compile_options(-Wall -Wextra -Wno-unused-parameter -Wno-unused-result)
 
 find_package(PkgConfig REQUIRED)
 pkg_check_modules(SODIUM REQUIRED libsodium)
@@ -27,8 +27,11 @@ configure_file("paseto.pc.in" "paseto.pc
 add_library(paseto SHARED ${PASETO_SOURCES})
 target_link_libraries(paseto ${SODIUM_LIBRARIES})
 install(TARGETS paseto
-	LIBRARY DESTINATION lib
+	RUNTIME DESTINATION bin
 	ARCHIVE DESTINATION lib)
+if(CYGWIN)
+	set_target_properties(paseto PROPERTIES SOVERSION 0)
+endif()
 install(FILES include/paseto.h
 	DESTINATION include)
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/paseto.pc"
