--- origsrc/libpaseto-1.0/CMakeLists.txt	2019-07-03 23:51:58.000000000 +0900
+++ src/libpaseto-1.0/CMakeLists.txt	2026-03-26 16:44:30.011238400 +0900
@@ -5,7 +5,7 @@ project(libpaseto
 
 set(CMAKE_C_STANDARD 11)
 set(CMAKE_C_EXTENSIONS OFF)
-add_compile_options(-Wall -Wextra -Wno-unused-parameter -Wno-unused-result -Werror)
+add_compile_options(-Wall -Wextra -Wno-unused-parameter -Wno-unused-result)
 
 find_package(PkgConfig REQUIRED)
 pkg_check_modules(SODIUM REQUIRED libsodium)
@@ -27,12 +27,15 @@ configure_file("paseto.pc.in" "paseto.pc
 add_library(paseto SHARED ${PASETO_SOURCES})
 target_link_libraries(paseto ${SODIUM_LIBRARIES})
 install(TARGETS paseto
-	LIBRARY DESTINATION lib
+	RUNTIME DESTINATION bin
 	ARCHIVE DESTINATION lib)
 install(FILES include/paseto.h
 	DESTINATION include)
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/paseto.pc"
 	DESTINATION lib/pkgconfig)
+set_target_properties(paseto PROPERTIES
+	VERSION ${PROJECT_VERSION}
+	SOVERSION ${PROJECT_VERSION_MAJOR})
 
 # test executable
 add_executable(pasetotest
--- origsrc/libpaseto-1.0/paseto.pc.in	2019-07-03 23:51:58.000000000 +0900
+++ src/libpaseto-1.0/paseto.pc.in	2026-03-26 16:45:16.623623300 +0900
@@ -5,7 +5,7 @@ includedir=${prefix}/include
 
 Name: paseto
 Description: C implementation of PASETO (v2 tokens only)
-Version: @VERSION@
+Version: @PROJECT_VERSION@
 
 Requires: libsodium
 Libs: -L${libdir} -lpaseto
