--- origsrc/libfixbuf-1.6.1/src/Makefile.am	2014-10-10 21:28:39.000000000 +0900
+++ src/libfixbuf-1.6.1/src/Makefile.am	2014-11-08 00:25:15.362880300 +0900
@@ -19,7 +19,7 @@ libfixbuf_la_SOURCES =  fbuf.c       fbi
                         fbconnspec.c fbexporter.c  fbcollector.c fblistener.c \
 			fbnetflow.c fbsflow.c
 libfixbuf_la_LIBADD = $(GLIB_LDADD)
-libfixbuf_la_LDFLAGS = $(SPREAD_LDFLAGS) $(SPREAD_LIBS) $(GLIB_LIBS) -version-info $(LIBCOMPAT)
+libfixbuf_la_LDFLAGS = -no-undefined $(SPREAD_LDFLAGS) $(SPREAD_LIBS) $(GLIB_LIBS) -version-info $(LIBCOMPAT)
 libfixbuf_la_CFLAGS = $(WARN_CFLAGS) $(DEBUG_CFLAGS) $(SPREAD_CFLAGS) $(GLIB_CFLAGS)
 
 noinst_HEADERS = fbcollector.h
