--- origsrc/libfixbuf-2.4.0/src/Makefile.am	2019-08-28 02:11:53.000000000 +0900
+++ src/libfixbuf-2.4.0/src/Makefile.am	2019-11-30 09:13:02.509240500 +0900
@@ -45,7 +45,8 @@ libfixbuf_la_SOURCES =  fbuf.c       fbi
                         fbconnspec.c fbexporter.c  fbcollector.c fbcollector.h \
                         fblistener.c fbnetflow.c   fbsflow.c     fbxml.c
 nodist_libfixbuf_la_SOURCES = $(MAKE_INFOMODEL_OUTPUTS)
-libfixbuf_la_LDFLAGS = -version-info $(LIBCOMPAT)
+libfixbuf_la_LDFLAGS = -no-undefined -version-info $(LIBCOMPAT)
+libfixbuf_la_LIBADD = $(GLIB_LIBS)
 
 EXTRA_DIST = xml2fixbuf.xslt make-infomodel
 
