--- origsrc/libfixbuf-1.1.1/src/Makefile.am	2011-09-26 23:37:27.000000000 +0900
+++ src/libfixbuf-1.1.1/src/Makefile.am	2012-03-28 10:40:22.670522700 +0900
@@ -17,7 +17,7 @@ libfixbuf_la_SOURCES =  fbuf.c       fbi
                         fbconnspec.c fbexporter.c  fbcollector.c fblistener.c \
 						fbnetflow.c
 libfixbuf_la_LIBADD = $(GLIB_LDADD)
-libfixbuf_la_LDFLAGS = $(SPREAD_LDFLAGS) $(SPREAD_LIBS) $(GLIB_LIBS) -version-info $(LIBCOMPAT) -release ${VERSION}
+libfixbuf_la_LDFLAGS = $(SPREAD_LDFLAGS) $(SPREAD_LIBS) $(GLIB_LIBS) -version-info $(LIBCOMPAT) -release ${VERSION} -no-undefined
 libfixbuf_la_CFLAGS = $(WARN_FLAGS) $(DEBUG_CFLAGS) $(SPREAD_CFLAGS) $(GLIB_CFLAGS)
 
 noinst_HEADERS = fbcollector.h
