--- origsrc/libfixbuf-1.8.0/src/Makefile.am	2017-03-31 04:07:18.000000000 +0900
+++ src/libfixbuf-1.8.0/src/Makefile.am	2018-03-03 00:17:44.393149000 +0900
@@ -19,7 +19,7 @@ libfixbuf_la_SOURCES =  fbuf.c       fbi
                         fbconnspec.c fbexporter.c  fbcollector.c fblistener.c \
 			fbnetflow.c fbsflow.c
 libfixbuf_la_LIBADD = $(GLIB_LDADD)
-libfixbuf_la_LDFLAGS = $(SPREAD_LDFLAGS) $(SPREAD_LIBS) $(GLIB_LIBS) -version-info $(LIBCOMPAT)
+libfixbuf_la_LDFLAGS = -no-undefined $(SPREAD_LDFLAGS) $(SPREAD_LIBS) $(GLIB_LIBS) -version-info $(LIBCOMPAT)
 libfixbuf_la_CFLAGS = $(WARN_CFLAGS) $(DEBUG_CFLAGS) $(SPREAD_CFLAGS) $(GLIB_CFLAGS)
 
 noinst_HEADERS = fbcollector.h
