--- origsrc/libparamset-1.1.240/GNUmakefile.am	2018-04-12 17:03:38.000000000 +0900
+++ src/libparamset-1.1.240/GNUmakefile.am	2019-01-05 06:47:46.480476400 +0900
@@ -45,3 +45,6 @@ doc: doc/doxyfile
 doc-zip: doc
 	cp -r doc/html ${ZIPDOC_DIR}
 	zip -r ${ZIPDOC_DIR}.zip ${ZIPDOC_DIR}
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libparamset-1.1.240/src/param_set/GNUmakefile.am	2018-04-12 17:03:38.000000000 +0900
+++ src/libparamset-1.1.240/src/param_set/GNUmakefile.am	2019-01-04 23:19:53.593943500 +0900
@@ -48,5 +48,5 @@ otherinclude_HEADERS = \
 	wildcardexpanders.h \
 	version.h
 
-libparamset_la_LDFLAGS=-version-info @LTVER@
+libparamset_la_LDFLAGS=-no-undefined -version-info @LTVER@
 
--- origsrc/libparamset-1.1.240/test/GNUmakefile.am	2018-04-12 17:03:38.000000000 +0900
+++ src/libparamset-1.1.240/test/GNUmakefile.am	2019-01-04 23:20:14.766724900 +0900
@@ -17,7 +17,7 @@
 # reserves and retains all trademark rights.
 #
 
-AM_CFLAGS=-g -Wall -I$(top_builddir)/src/param_set
+AM_CFLAGS=-g -Wall -I$(top_builddir)/src/param_set -I$(top_srcdir)/src/param_set
 AM_LDFLAGS=-L$(top_builddir)/src/param_set -no-install -lparamset
 check_PROGRAMS = test
 
