--- origsrc/metaproxy-1.17.1/src/Makefile.am	2019-10-22 21:50:48.000000000 +0900
+++ src/metaproxy-1.17.1/src/Makefile.am	2019-12-05 06:41:14.597120400 +0900
@@ -41,7 +41,7 @@ filter_src = \
 	filter_zoom.cpp filter_zoom.hpp
 
 lib_LTLIBRARIES = libmetaproxy.la
-libmetaproxy_la_LDFLAGS = -version-info 6:0:0 -export-dynamic
+libmetaproxy_la_LDFLAGS = -no-undefined -version-info 6:0:0 -export-dynamic
 
 libmetaproxy_la_SOURCES = \
 	$(filter_src) \
@@ -88,7 +88,7 @@ metaproxy_SOURCES = metaproxy_prog.cpp
 pkglib_LTLIBRARIES = metaproxy_filter_dl.la
 
 metaproxy_filter_dl_la_SOURCES = filter_dl.cpp
-metaproxy_filter_dl_la_LDFLAGS = -rpath $(pkglibdir) -module -avoid-version
+metaproxy_filter_dl_la_LDFLAGS = -no-undefined -rpath $(pkglibdir) -module -avoid-version
 metaproxy_filter_dl_la_LIBADD = libmetaproxy.la
 
 # Rules for test programs..
