--- origsrc/sagittarius-0.9.8/ext/odbc/odbc.c	2021-09-11 04:33:30.000000000 +0900
+++ src/sagittarius-0.9.8/ext/odbc/odbc.c	2022-05-30 00:17:06.675113900 +0900
@@ -32,6 +32,7 @@
 #define LIBSAGITTARIUS_EXT_BODY
 #include <sagittarius/extend.h>
 #include "odbc.h"
+#include <cstring>
 
 static SgClass *error_cpl[] = {
   SG_ERROR_CONDITION_CPL,
--- origsrc/sagittarius-0.9.8/ext/odbc/odbc.h	2021-09-11 04:33:30.000000000 +0900
+++ src/sagittarius-0.9.8/ext/odbc/odbc.h	2022-05-30 00:16:03.405829200 +0900
@@ -36,7 +36,7 @@
 #endif
 
 #include <sagittarius.h>
-#if __CYGWIN__ || _WIN32
+#if _WIN32
 #include <windows.h>
 #endif
 #include <sql.h>
--- origsrc/sagittarius-0.9.8/src/CMakeLists.txt	2021-09-11 04:33:30.000000000 +0900
+++ src/sagittarius-0.9.8/src/CMakeLists.txt	2022-05-30 04:17:34.151539500 +0900
@@ -189,7 +189,7 @@ ENDIF()
 # for version control
 # default dynamic load path like /usr/local/lib/sagittarius/${version}/${triple}
 # so let's put it there.
-INSTALL(TARGETS sagittarius DESTINATION ${SAGITTARIUS_DYNLIB_PATH})
+INSTALL(TARGETS sagittarius RUNTIME DESTINATION ${SAGITTARIUS_DYNLIB_PATH})
 INSTALL(TARGETS sash DESTINATION ${SAGITTARIUS_DYNLIB_PATH})
 IF (NOT WIN32 AND INSTALL_SYMLINK)
   MESSAGE(STATUS
@@ -251,9 +251,13 @@ EXECUTE_PROCESS (
     "EXECUTE_PROCESS(COMMAND ${CMAKE_COMMAND} -P ${LIB_LINKING_FILE})")
 
   IF (POSIX_EMULATOR)
-    # It seems Cygwin/MSYS can't handle symbolic linked dll.
-    # NB: even CMake doesn't install namelink, so probaby above is true. 
-    INSTALL(TARGETS sagittarius DESTINATION ${CMAKE_INSTALL_PREFIX}/${BIN_DIR})
+    if (CYGWIN)
+      install(TARGETS sagittarius RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    else()
+      # It seems MSYS can't handle symbolic linked dll.
+      # NB: even CMake doesn't install namelink, so probaby above is true. 
+      INSTALL(TARGETS sagittarius DESTINATION ${CMAKE_INSTALL_PREFIX}/${BIN_DIR})
+    endif()
   ENDIF()
   INSTALL(TARGETS sash DESTINATION ${BIN_DIR})
 ENDIF()
