--- origsrc/gn-0.1807/src/util/build_config.h	2020-07-18 08:14:46.000000000 +0900
+++ src/gn-0.1807/src/util/build_config.h	2020-07-27 08:39:25.410559900 +0900
@@ -26,7 +26,7 @@
 #if defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE
 #define OS_IOS 1
 #endif  // defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__CYGWIN__)
 #define OS_LINUX 1
 // include a system header to pull in features.h for glibc/uclibc macros.
 #include <unistd.h>
