--- origsrc/kytea-0.4.3/configure.ac	2013-02-04 23:49:03.000000000 +0900
+++ src/kytea-0.4.3/configure.ac	2013-02-24 14:35:30.727946400 +0900
@@ -58,6 +58,9 @@ AC_HEADER_STDBOOL
 AC_C_INLINE
 AC_TYPE_SIZE_T
 
+AC_CHECK_LIB([linear], [main], [use_external_linear=true], [use_external_linear=false])
+AM_CONDITIONAL(USE_EXTERNAL_LINEAR, test x$use_external_linear = xtrue)
+
 # Check to make sure that we have unordered_map
 AC_LANG([C++])
 AC_CHECK_HEADERS([boost/tr1/unordered_map.hpp tr1/unordered_map ext/hash_map], break)
--- origsrc/kytea-0.4.3/src/lib/Makefile.am	2012-04-02 23:26:42.000000000 +0900
+++ src/kytea-0.4.3/src/lib/Makefile.am	2013-02-24 14:35:49.806071400 +0900
@@ -6,10 +6,16 @@ KYTCPP = kytea.cpp corpus-io.cpp model-i
 
 AM_CPPFLAGS = -I$(srcdir)/../include -DPKGDATADIR='"$(pkgdatadir)"'
 
+if !USE_EXTERNAL_LINEAR
 SUBDIRS = liblinear
+endif
 
 lib_LTLIBRARIES = libkytea.la
 
 libkytea_la_SOURCES = ${KYTCPP}
+if USE_EXTERNAL_LINEAR
+libkytea_la_LIBADD = -llinear
+else
 libkytea_la_LIBADD = ${LLLIBS}
-libkytea_la_LDFLAGS = -version-info 0:0:0
+endif
+libkytea_la_LDFLAGS = -version-info 0:0:0 -no-undefined
