--- origsrc/ranpwd-1.2/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/ranpwd-1.2/Makefile.am	2011-05-31 00:44:08.365856600 +0900
@@ -0,0 +1,4 @@
+bin_PROGRAMS = ranpwd
+man_MANS = ranpwd.1
+ranpwd_SOURCES = ranpwd.c
+TESTS = ranpwd
--- origsrc/ranpwd-1.2/configure.in	2008-01-19 06:04:19.000000000 +0900
+++ src/ranpwd-1.2/configure.in	2011-05-31 00:38:59.101197500 +0900
@@ -12,23 +12,10 @@
 
 AC_INIT(ranpwd, 1.2)
 AC_PREREQ(2.59)
+AM_INIT_AUTOMAKE
 
 AC_PROG_CC
 
-PA_ADD_CFLAGS(-Wall)
-PA_ADD_CFLAGS(-W)
-PA_ADD_CFLAGS(-Wpointer-arith)
-PA_ADD_CFLAGS(-Wbad-function-cast)
-PA_ADD_CFLAGS(-Wcast-equal)
-PA_ADD_CFLAGS(-Wstrict-prototypes)
-PA_ADD_CFLAGS(-Wmissing-prototypes)
-PA_ADD_CFLAGS(-Wmissing-declarations)
-PA_ADD_CFLAGS(-Wnested-externs)
-PA_ADD_CFLAGS(-Winline)
-PA_ADD_CFLAGS(-Wshadow)
-PA_ADD_CFLAGS(-Wcast-align)
-PA_ADD_CFLAGS(-pipe)
-
 AC_CHECK_HEADERS(getopt.h)
 AC_CHECK_FUNCS(getopt_long)
 
