--- origsrc/parmetis-4.0.3/CMakeLists.txt	2013-03-31 01:24:50.000000000 +0900
+++ src/parmetis-4.0.3/CMakeLists.txt	2020-01-25 21:11:56.670618500 +0900
@@ -1,13 +1,6 @@
 cmake_minimum_required(VERSION 2.8)
 project(ParMETIS)
 
-set(GKLIB_PATH METIS/GKlib CACHE PATH "path to GKlib")
-set(METIS_PATH METIS CACHE PATH "path to METIS")
-
-# Symlink ./metis to wherever metis is. This allows files to be
-# included from metis/libmetis/.
-execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ${METIS_PATH} metis)
-
 # Search for MPI.
 # GK commented this out as it seems to be creating problems
 # include(FindMPI)
@@ -23,7 +16,6 @@ else()
   set(ParMETIS_LIBRARY_TYPE STATIC)
 endif()
 
-include(${GKLIB_PATH}/GKlibSystem.cmake)
 
 # List of paths that the compiler will search for header files.
 # i.e., the -I equivalent
@@ -33,10 +25,16 @@ include_directories(${GKLIB_PATH})
 include_directories(${METIS_PATH}/include)
 
 # List of directories that cmake will look for CMakeLists.txt
-add_subdirectory(${METIS_PATH}/libmetis ${CMAKE_BINARY_DIR}/libmetis)
 add_subdirectory(include)
 add_subdirectory(libparmetis)
 add_subdirectory(programs)
 
 # This is for testing during development and is not being distributed
 #add_subdirectory(test)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/parmetis.pc.in ${CMAKE_BINARY_DIR}/parmetis.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/parmetis.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/parmetis-4.0.3/libparmetis/CMakeLists.txt	2013-03-31 01:24:50.000000000 +0900
+++ src/parmetis-4.0.3/libparmetis/CMakeLists.txt	2020-01-25 21:08:24.156525200 +0900
@@ -8,7 +8,11 @@ add_library(parmetis ${ParMETIS_LIBRARY_
 target_link_libraries(parmetis metis ${MPI_LIBRARIES})
 set_target_properties(parmetis PROPERTIES LINK_FLAGS "${MPI_LINK_FLAGS}")
 
+if (CYGWIN)
+  set_target_properties(parmetis PROPERTIES SOVERSION 0)
+endif()
+
 install(TARGETS parmetis
   LIBRARY DESTINATION lib
-  RUNTIME DESTINATION lib
+  RUNTIME DESTINATION bin
   ARCHIVE DESTINATION lib)
--- origsrc/parmetis-4.0.3/parmetis.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/parmetis-4.0.3/parmetis.pc.in	2020-01-25 21:11:50.804505100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: parmetis
+Description: Parallel Graph Partitioning and Fill-reducing Matrix Ordering
+Version: @VERSION@
+Libs: -L${libdir} -lparmetis
+Cflags: -I${includedir}
