--- origsrc/libevhtp/CMakeLists.txt	2012-03-29 21:57:36.000000000 +0900
+++ src/libevhtp/CMakeLists.txt	2012-03-29 23:22:07.542312500 +0900
@@ -120,16 +120,13 @@ find_path(LIBEVENT_INCLUDE_DIR event2/ev
 
 include_directories(
 	${CMAKE_CURRENT_BINARY_DIR}/compat
-	${CMAKE_CURRENT_SOURCE_DIR}/htparse
-	${CMAKE_CURRENT_BINARY_DIR}/oniguruma
-	${CMAKE_CURRENT_SOURCE_DIR}/oniguruma
 	${CMAKE_CURRENT_SOURCE_DIR}
-	${CMAKE_CURRENT_SOURCE_DIR}/evthr
 	${OPENSSL_INCLUDE_DIR}
 	${LIBEVENT_INCLUDE_DIR}
 )
 
 set(LIBEVHTP_EXTERNAL_LIBS
+		htparse onig
 		${LIBEVENT_LIBRARY}
 		${LIBEVENT_PTHREADS_LIBRARY}
 		${LIBEVENT_OPENSSL_LIBRARY}
@@ -143,25 +140,22 @@ if (NOT ${LIBEVENT_OPENSSL_FOUND})
 	set (EVHTP_DISABLE_SSL 1)
 endif(NOT ${LIBEVENT_OPENSSL_FOUND})
 
-set(LIBEVHTP_SOURCES evhtp.c htparse/htparse.c)
+set(LIBEVHTP_SOURCES evhtp.c)
 
 if (NOT EVHTP_DISABLE_EVTHR)
-	set (LIBEVHTP_EXTERNAL_LIBS ${LIBEVHTP_EXTERNAL_LIBS} pthread)
-	set (LIBEVHTP_SOURCES ${LIBEVHTP_SOURCES} evthr/evthr.c)
+	set (LIBEVHTP_EXTERNAL_LIBS ${LIBEVHTP_EXTERNAL_LIBS} pthread evthr)
 endif(NOT EVHTP_DISABLE_EVTHR)
 
 if (NOT ${CMAKE_BUILD_TYPE} STREQUAL "Debug")
   set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DNDEBUG")
 endif (NOT ${CMAKE_BUILD_TYPE} STREQUAL "Debug")
 
-add_library(libevhtp STATIC ${LIBEVHTP_SOURCES} ${ONIG_SOURCES})
-set_target_properties(libevhtp PROPERTIES OUTPUT_NAME "evhtp")
+add_library(libevhtp SHARED ${LIBEVHTP_SOURCES})
+target_link_libraries(libevhtp ${LIBEVHTP_EXTERNAL_LIBS})
+set_target_properties(libevhtp PROPERTIES OUTPUT_NAME "evhtp" SOVERSION 0)
 
-install (TARGETS libevhtp DESTINATION lib)
+install (TARGETS libevhtp RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 install (FILES evhtp.h DESTINATION include)
-install (FILES htparse/htparse.h DESTINATION include)
-install (FILES evthr/evthr.h DESTINATION include)
-install (FILES oniguruma/onigposix.h DESTINATION include)
 
 find_library (LIB_DL dl)
 set (SYS_LIBS ${LIB_DL})
