--- origsrc/bufferpool-0.0.4/configure.ac	2009-03-11 00:24:21.000000000 +0900
+++ src/bufferpool-0.0.4/configure.ac	2011-06-14 22:13:41.284926800 +0900
@@ -33,7 +33,7 @@ CPPFLAGS="${CPPFLAGS} -D_POSIX_C_SOURCE=
 AC_CONFIG_HEADER([config.h])
 
 dnl Check for LibTool Dynamic Loading
-AC_PROG_LIBTOOL
+LT_INIT
 LIBTOOL="$LIBTOOL --silent"
 AC_SUBST([LIBTOOL_DEPS])
 
--- origsrc/bufferpool-0.0.4/lib/Makefile.am	2009-03-11 00:24:21.000000000 +0900
+++ src/bufferpool-0.0.4/lib/Makefile.am	2011-06-14 22:13:51.659845200 +0900
@@ -6,6 +6,6 @@ libbufferpool_la_SOURCES = 	bp.c \
 				bp_init.c \
 				bp_shm.c
 
-libbufferpool_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
+libbufferpool_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -no-undefined
 
 INCLUDES = -I$(includedir) -I$(top_srcdir)
