--- origsrc/openfst-1.5.1/configure.ac	2016-02-19 08:26:42.000000000 +0900
+++ src/openfst-1.5.1/configure.ac	2016-03-13 21:35:42.727018200 +0900
@@ -3,7 +3,7 @@ AC_INIT([OpenFst], [1.5.1], [help@www.op
 AM_INIT_AUTOMAKE([foreign nostdinc -Wall -Werror])
 AM_PROG_AR
 
-CXXFLAGS="$CXXFLAGS -std=c++11"
+CXXFLAGS="$CXXFLAGS -std=gnu++11"
 
 AC_PROG_CXX
 AC_DISABLE_STATIC
--- origsrc/openfst-1.5.1/src/extensions/compact/Makefile.am	2016-01-28 02:21:01.000000000 +0900
+++ src/openfst-1.5.1/src/extensions/compact/Makefile.am	2016-03-13 21:43:08.493175500 +0900
@@ -6,49 +6,65 @@ libfst_LTLIBRARIES = compact8_acceptor-f
 lib_LTLIBRARIES = libfstcompact.la
 
 libfstcompact_la_SOURCES = compact8_acceptor-fst.cc compact8_string-fst.cc compact8_unweighted-fst.cc compact8_unweighted_acceptor-fst.cc compact8_weighted_string-fst.cc compact16_acceptor-fst.cc compact16_string-fst.cc compact16_unweighted-fst.cc compact16_unweighted_acceptor-fst.cc compact16_weighted_string-fst.cc compact64_acceptor-fst.cc compact64_string-fst.cc compact64_unweighted-fst.cc compact64_unweighted_acceptor-fst.cc compact64_weighted_string-fst.cc
-libfstcompact_la_LDFLAGS = -version-info 2:5:0
+libfstcompact_la_LDFLAGS = -no-undefined -version-info 2:5:0
+libfstcompact_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact8_acceptor_fst_la_SOURCES = compact8_acceptor-fst.cc
-compact8_acceptor_fst_la_LDFLAGS = -module
+compact8_acceptor_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+compact8_acceptor_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact8_string_fst_la_SOURCES = compact8_string-fst.cc
-compact8_string_fst_la_LDFLAGS = -module
+compact8_string_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+compact8_string_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact8_unweighted_fst_la_SOURCES = compact8_unweighted-fst.cc
-compact8_unweighted_fst_la_LDFLAGS = -module
+compact8_unweighted_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+compact8_unweighted_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact8_unweighted_acceptor_fst_la_SOURCES = compact8_unweighted_acceptor-fst.cc
-compact8_unweighted_acceptor_fst_la_LDFLAGS = -module
+compact8_unweighted_acceptor_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+compact8_unweighted_acceptor_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact8_weighted_string_fst_la_SOURCES = compact8_weighted_string-fst.cc
-compact8_weighted_string_fst_la_LDFLAGS = -module
+compact8_weighted_string_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+compact8_weighted_string_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact16_acceptor_fst_la_SOURCES = compact16_acceptor-fst.cc
-compact16_acceptor_fst_la_LDFLAGS = -module
+compact16_acceptor_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+compact16_acceptor_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact16_string_fst_la_SOURCES = compact16_string-fst.cc
-compact16_string_fst_la_LDFLAGS = -module
+compact16_string_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+compact16_string_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact16_unweighted_fst_la_SOURCES = compact16_unweighted-fst.cc
-compact16_unweighted_fst_la_LDFLAGS = -module
+compact16_unweighted_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+compact16_unweighted_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact16_unweighted_acceptor_fst_la_SOURCES = compact16_unweighted_acceptor-fst.cc
-compact16_unweighted_acceptor_fst_la_LDFLAGS = -module
+compact16_unweighted_acceptor_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+compact16_unweighted_acceptor_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact16_weighted_string_fst_la_SOURCES = compact16_weighted_string-fst.cc
-compact16_weighted_string_fst_la_LDFLAGS = -module
+compact16_weighted_string_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+compact16_weighted_string_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact64_acceptor_fst_la_SOURCES = compact64_acceptor-fst.cc
-compact64_acceptor_fst_la_LDFLAGS = -module
+compact64_acceptor_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+compact64_acceptor_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact64_string_fst_la_SOURCES = compact64_string-fst.cc
-compact64_string_fst_la_LDFLAGS = -module
+compact64_string_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+compact64_string_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact64_unweighted_fst_la_SOURCES = compact64_unweighted-fst.cc
-compact64_unweighted_fst_la_LDFLAGS = -module
+compact64_unweighted_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+compact64_unweighted_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact64_unweighted_acceptor_fst_la_SOURCES = compact64_unweighted_acceptor-fst.cc
-compact64_unweighted_acceptor_fst_la_LDFLAGS = -module
+compact64_unweighted_acceptor_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+compact64_unweighted_acceptor_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact64_weighted_string_fst_la_SOURCES = compact64_weighted_string-fst.cc
-compact64_weighted_string_fst_la_LDFLAGS = -module
+compact64_weighted_string_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+compact64_weighted_string_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
--- origsrc/openfst-1.5.1/src/extensions/compress/Makefile.am	2016-02-19 08:26:42.000000000 +0900
+++ src/openfst-1.5.1/src/extensions/compress/Makefile.am	2016-03-13 21:41:20.704152400 +0900
@@ -14,7 +14,7 @@ endif
 
 if HAVE_SCRIPT
 libfstcompressscript_la_SOURCES = compress-script.cc
-libfstcompressscript_la_LDFLAGS = -version-info 2:5:0
+libfstcompressscript_la_LDFLAGS = -no-undefined -version-info 2:5:0
 endif
 
 if HAVE_SCRIPT
--- origsrc/openfst-1.5.1/src/extensions/const/Makefile.am	2016-01-28 02:21:01.000000000 +0900
+++ src/openfst-1.5.1/src/extensions/const/Makefile.am	2016-03-13 23:27:36.134480300 +0900
@@ -6,13 +6,17 @@ libfst_LTLIBRARIES = const8-fst.la const
 lib_LTLIBRARIES = libfstconst.la
 
 libfstconst_la_SOURCES = const8-fst.cc const16-fst.cc const64-fst.cc
-libfstconst_la_LDFLAGS = -version-info 2:5:0
+libfstconst_la_LDFLAGS = -no-undefined -version-info 2:5:0
+libfstconst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 const8_fst_la_SOURCES = const8-fst.cc
-const8_fst_la_LDFLAGS = -module
+const8_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+const8_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 const16_fst_la_SOURCES = const16-fst.cc
-const16_fst_la_LDFLAGS = -module
+const16_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+const16_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 const64_fst_la_SOURCES = const64-fst.cc
-const64_fst_la_LDFLAGS = -module
+const64_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+const64_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
--- origsrc/openfst-1.5.1/src/extensions/far/Makefile.am	2016-02-19 08:26:42.000000000 +0900
+++ src/openfst-1.5.1/src/extensions/far/Makefile.am	2016-03-13 23:34:19.761233800 +0900
@@ -7,11 +7,12 @@ lib_LTLIBRARIES = libfstfar.la
 endif
 
 libfstfar_la_SOURCES = sttable.cc stlist.cc
-libfstfar_la_LDFLAGS = -version-info 2:5:0
+libfstfar_la_LDFLAGS = -no-undefined -version-info 2:5:0
 
 if HAVE_SCRIPT
 libfstfarscript_la_SOURCES = far-class.cc farscript.cc strings.cc main.cc
-libfstfarscript_la_LDFLAGS = -version-info 2:5:0
+libfstfarscript_la_LDFLAGS = -no-undefined -version-info 2:5:0
+libfstfarscript_la_LIBADD = $(top_builddir)/src/lib/libfst.la libfstfar.la $(top_builddir)/src/script/libfstscript.la
 endif
 
 if HAVE_BIN
--- origsrc/openfst-1.5.1/src/extensions/linear/Makefile.am	2016-02-19 08:26:42.000000000 +0900
+++ src/openfst-1.5.1/src/extensions/linear/Makefile.am	2016-03-13 23:30:35.643468200 +0900
@@ -13,7 +13,8 @@ endif
 
 if HAVE_SCRIPT
 libfstlinearscript_la_SOURCES = linearscript.cc
-libfstlinearscript_la_LDFLAGS = -version-info 2:5:0
+libfstlinearscript_la_LDFLAGS = -no-undefined -version-info 2:5:0
+libfstlinearscript_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 endif
 
 if HAVE_SCRIPT
@@ -27,7 +28,9 @@ endif
 libfstdir = @libfstdir@
 
 linear_tagger_fst_la_SOURCES = linear-tagger-fst.cc
-linear_tagger_fst_la_LDFLAGS = -module
+linear_tagger_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+linear_tagger_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 linear_classifier_fst_la_SOURCES = linear-classifier-fst.cc
-linear_classifier_fst_la_LDFLAGS = -module
+linear_classifier_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+linear_classifier_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
--- origsrc/openfst-1.5.1/src/extensions/lookahead/Makefile.am	2016-01-28 02:21:01.000000000 +0900
+++ src/openfst-1.5.1/src/extensions/lookahead/Makefile.am	2016-03-14 20:09:08.781352300 +0900
@@ -8,13 +8,17 @@ lib_LTLIBRARIES = libfstlookahead.la
 
 libfstlookahead_la_SOURCES = arc_lookahead-fst.cc ilabel_lookahead-fst.cc \
 olabel_lookahead-fst.cc
-libfstlookahead_la_LDFLAGS = -version-info 2:5:0
+libfstlookahead_la_LDFLAGS = -no-undefined -version-info 2:5:0
+libfstlookahead_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 arc_lookahead_fst_la_SOURCES = arc_lookahead-fst.cc
-arc_lookahead_fst_la_LDFLAGS = -module
+arc_lookahead_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+arc_lookahead_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 ilabel_lookahead_fst_la_SOURCES = ilabel_lookahead-fst.cc
-ilabel_lookahead_fst_la_LDFLAGS = -module
+ilabel_lookahead_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+ilabel_lookahead_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 olabel_lookahead_fst_la_SOURCES = olabel_lookahead-fst.cc
-olabel_lookahead_fst_la_LDFLAGS = -module
+olabel_lookahead_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+olabel_lookahead_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
--- origsrc/openfst-1.5.1/src/extensions/mpdt/Makefile.am	2016-02-19 08:26:42.000000000 +0900
+++ src/openfst-1.5.1/src/extensions/mpdt/Makefile.am	2016-03-13 21:44:05.922235700 +0900
@@ -19,5 +19,5 @@ endif
 if HAVE_SCRIPT
 lib_LTLIBRARIES = libfstmpdtscript.la
 libfstmpdtscript_la_SOURCES = mpdtscript.cc
-libfstmpdtscript_la_LDFLAGS = -version-info 2:5:0
+libfstmpdtscript_la_LDFLAGS = -no-undefined -version-info 2:5:0
 endif
--- origsrc/openfst-1.5.1/src/extensions/ngram/Makefile.am	2016-01-28 02:21:01.000000000 +0900
+++ src/openfst-1.5.1/src/extensions/ngram/Makefile.am	2016-03-14 20:14:15.321201000 +0900
@@ -6,7 +6,9 @@ libfst_LTLIBRARIES = ngram-fst.la
 lib_LTLIBRARIES = libfstngram.la
 
 ngram_fst_la_SOURCES = bitmap-index.cc ngram-fst.cc nthbit.cc
-ngram_fst_la_LDFLAGS = -module
+ngram_fst_la_LDFLAGS = -avoid-version -no-undefined -module
+ngram_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 libfstngram_la_SOURCES = bitmap-index.cc ngram-fst.cc nthbit.cc
-libfstngram_la_LDFLAGS = -version-info 2:5:0
+libfstngram_la_LDFLAGS = -no-undefined -version-info 2:5:0
+libfstngram_la_LIBADD = $(top_builddir)/src/lib/libfst.la
--- origsrc/openfst-1.5.1/src/extensions/pdt/Makefile.am	2016-02-19 08:26:42.000000000 +0900
+++ src/openfst-1.5.1/src/extensions/pdt/Makefile.am	2016-03-14 20:19:45.820765700 +0900
@@ -23,5 +23,6 @@ endif
 if HAVE_SCRIPT
 lib_LTLIBRARIES = libfstpdtscript.la
 libfstpdtscript_la_SOURCES = pdtscript.cc
-libfstpdtscript_la_LDFLAGS = -version-info 2:5:0
+libfstpdtscript_la_LDFLAGS = -no-undefined -version-info 2:5:0
+libfstpdtscript_la_LIBADD = $(top_builddir)/src/lib/libfst.la $(top_builddir)/src/script/libfstscript.la
 endif
--- origsrc/openfst-1.5.1/src/include/fst/register.h	2016-02-25 23:36:18.000000000 +0900
+++ src/openfst-1.5.1/src/include/fst/register.h	2016-03-13 21:45:58.600017700 +0900
@@ -57,7 +57,11 @@ class FstRegister
 
     ConvertToLegalCSymbol(&legal_type);
 
+#if defined(__CYGWIN__)
+    return legal_type + "-fst.dll";
+#else
     return legal_type + "-fst.so";
+#endif
   }
 };
 
--- origsrc/openfst-1.5.1/src/include/fst/script/script-impl.h	2016-02-25 23:36:20.000000000 +0900
+++ src/openfst-1.5.1/src/include/fst/script/script-impl.h	2016-03-13 21:46:32.997248200 +0900
@@ -123,7 +123,11 @@ class GenericOperationRegister
     string legal_type(key.second);  // the arc type
     ConvertToLegalCSymbol(&legal_type);
 
+#if defined(__CYGWIN__)
+    return legal_type + "-arc.dll";
+#else
     return legal_type + "-arc.so";
+#endif
   }
 };
 
--- origsrc/openfst-1.5.1/src/include/fst/script/weight-class.h	2016-02-25 23:36:20.000000000 +0900
+++ src/openfst-1.5.1/src/include/fst/script/weight-class.h	2016-03-13 21:46:53.568997100 +0900
@@ -146,7 +146,11 @@ class WeightClassRegister : public Gener
                                                    WeightClassRegister> {
  protected:
   string ConvertKeyToSoFilename(const string &key) const override {
+#if defined(__CYGWIN__)
+    return key + ".dll";
+#else
     return key + ".so";
+#endif
   }
 };
 
--- origsrc/openfst-1.5.1/src/lib/Makefile.am	2016-02-19 08:26:42.000000000 +0900
+++ src/openfst-1.5.1/src/lib/Makefile.am	2016-03-13 21:47:05.981277500 +0900
@@ -4,4 +4,4 @@ LDADD = $(DL_LIBS)
 lib_LTLIBRARIES = libfst.la
 libfst_la_SOURCES = compat.cc flags.cc fst.cc properties.cc \
 symbol-table.cc util.cc symbol-table-ops.cc mapped-file.cc
-libfst_la_LDFLAGS = -version-info 2:5:0
+libfst_la_LDFLAGS = -no-undefined -version-info 2:5:0
--- origsrc/openfst-1.5.1/src/script/Makefile.am	2016-02-19 08:26:42.000000000 +0900
+++ src/openfst-1.5.1/src/script/Makefile.am	2016-03-13 22:45:40.083222000 +0900
@@ -14,5 +14,5 @@ shortest-distance.cc shortest-path.cc st
 text-io.cc topsort.cc union.cc weight-class.cc verify.cc
 
 libfstscript_la_LIBADD = ../lib/libfst.la
-libfstscript_la_LDFLAGS = -version-info 2:5:0
+libfstscript_la_LDFLAGS = -no-undefined -version-info 2:5:0
 endif
