--- origsrc/openfst-1.3.3/src/extensions/compact/Makefile.am	2013-01-19 10:17:11.000000000 +0900
+++ src/openfst-1.3.3/src/extensions/compact/Makefile.am	2013-02-07 09:24:29.419134100 +0900
@@ -4,49 +4,65 @@ libfstdir = @libfstdir@
 libfst_LTLIBRARIES = libfstcompact.la compact8_acceptor-fst.la compact8_string-fst.la compact8_unweighted-fst.la compact8_unweighted_acceptor-fst.la compact8_weighted_string-fst.la compact16_acceptor-fst.la compact16_string-fst.la compact16_unweighted-fst.la compact16_unweighted_acceptor-fst.la compact16_weighted_string-fst.la compact64_acceptor-fst.la compact64_string-fst.la compact64_unweighted-fst.la compact64_unweighted_acceptor-fst.la compact64_weighted_string-fst.la
 
 libfstcompact_la_SOURCES = compact8_acceptor-fst.cc compact8_string-fst.cc compact8_unweighted-fst.cc compact8_unweighted_acceptor-fst.cc compact8_weighted_string-fst.cc compact16_acceptor-fst.cc compact16_string-fst.cc compact16_unweighted-fst.cc compact16_unweighted_acceptor-fst.cc compact16_weighted_string-fst.cc compact64_acceptor-fst.cc compact64_string-fst.cc compact64_unweighted-fst.cc compact64_unweighted_acceptor-fst.cc compact64_weighted_string-fst.cc
-libfstcompact_la_LDFLAGS = -version-info 1:0:0
+libfstcompact_la_LDFLAGS = -version-info 1:0:0 -no-undefined
+libfstcompact_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact8_acceptor_fst_la_SOURCES = compact8_acceptor-fst.cc
-compact8_acceptor_fst_la_LDFLAGS = -module
+compact8_acceptor_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+compact8_acceptor_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact8_string_fst_la_SOURCES = compact8_string-fst.cc
-compact8_string_fst_la_LDFLAGS = -module
+compact8_string_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+compact8_string_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact8_unweighted_fst_la_SOURCES = compact8_unweighted-fst.cc
-compact8_unweighted_fst_la_LDFLAGS = -module
+compact8_unweighted_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+compact8_unweighted_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact8_unweighted_acceptor_fst_la_SOURCES = compact8_unweighted_acceptor-fst.cc
-compact8_unweighted_acceptor_fst_la_LDFLAGS = -module
+compact8_unweighted_acceptor_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+compact8_unweighted_acceptor_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact8_weighted_string_fst_la_SOURCES = compact8_weighted_string-fst.cc
-compact8_weighted_string_fst_la_LDFLAGS = -module
+compact8_weighted_string_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+compact8_weighted_string_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact16_acceptor_fst_la_SOURCES = compact16_acceptor-fst.cc
-compact16_acceptor_fst_la_LDFLAGS = -module
+compact16_acceptor_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+compact16_acceptor_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact16_string_fst_la_SOURCES = compact16_string-fst.cc
-compact16_string_fst_la_LDFLAGS = -module
+compact16_string_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+compact16_string_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact16_unweighted_fst_la_SOURCES = compact16_unweighted-fst.cc
-compact16_unweighted_fst_la_LDFLAGS = -module
+compact16_unweighted_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+compact16_unweighted_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact16_unweighted_acceptor_fst_la_SOURCES = compact16_unweighted_acceptor-fst.cc
-compact16_unweighted_acceptor_fst_la_LDFLAGS = -module
+compact16_unweighted_acceptor_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+compact16_unweighted_acceptor_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact16_weighted_string_fst_la_SOURCES = compact16_weighted_string-fst.cc
-compact16_weighted_string_fst_la_LDFLAGS = -module
+compact16_weighted_string_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+compact16_weighted_string_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact64_acceptor_fst_la_SOURCES = compact64_acceptor-fst.cc
-compact64_acceptor_fst_la_LDFLAGS = -module
+compact64_acceptor_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+compact64_acceptor_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact64_string_fst_la_SOURCES = compact64_string-fst.cc
-compact64_string_fst_la_LDFLAGS = -module
+compact64_string_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+compact64_string_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact64_unweighted_fst_la_SOURCES = compact64_unweighted-fst.cc
-compact64_unweighted_fst_la_LDFLAGS = -module
+compact64_unweighted_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+compact64_unweighted_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact64_unweighted_acceptor_fst_la_SOURCES = compact64_unweighted_acceptor-fst.cc
-compact64_unweighted_acceptor_fst_la_LDFLAGS = -module
+compact64_unweighted_acceptor_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+compact64_unweighted_acceptor_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 compact64_weighted_string_fst_la_SOURCES = compact64_weighted_string-fst.cc
-compact64_weighted_string_fst_la_LDFLAGS = -module
+compact64_weighted_string_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+compact64_weighted_string_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
--- origsrc/openfst-1.3.3/src/extensions/const/Makefile.am	2013-01-19 10:17:11.000000000 +0900
+++ src/openfst-1.3.3/src/extensions/const/Makefile.am	2013-02-07 09:29:39.172063800 +0900
@@ -4,13 +4,17 @@ libfstdir = @libfstdir@
 libfst_LTLIBRARIES = libfstconst.la const8-fst.la const16-fst.la const64-fst.la
 
 libfstconst_la_SOURCES = const8-fst.cc const16-fst.cc const64-fst.cc
-libfstconst_la_LDFLAGS = -version-info 1:0:0
+libfstconst_la_LDFLAGS = -version-info 1:0:0 -no-undefined
+libfstconst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 const8_fst_la_SOURCES = const8-fst.cc
-const8_fst_la_LDFLAGS = -module
+const8_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+const8_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 const16_fst_la_SOURCES = const16-fst.cc
-const16_fst_la_LDFLAGS = -module
+const16_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+const16_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 const64_fst_la_SOURCES = const64-fst.cc
-const64_fst_la_LDFLAGS = -module
+const64_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+const64_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
--- origsrc/openfst-1.3.3/src/extensions/far/Makefile.am	2013-01-19 10:17:11.000000000 +0900
+++ src/openfst-1.3.3/src/extensions/far/Makefile.am	2013-02-07 09:34:19.557806000 +0900
@@ -3,17 +3,18 @@ AM_CPPFLAGS = -I$(srcdir)/../../include
 libfstdir = @libfstdir@
 
 if HAVE_SCRIPT
-libfst_LTLIBRARIES = libfstfarscript.la libfstfar.la
+libfst_LTLIBRARIES = libfstfar.la libfstfarscript.la 
 else
 libfst_LTLIBRARIES = libfstfar.la
 endif
 
 libfstfar_la_SOURCES = sttable.cc stlist.cc
-libfstfar_la_LDFLAGS = -version-info 1:0:0
+libfstfar_la_LDFLAGS = -version-info 1:0:0 -no-undefined
 
 if HAVE_SCRIPT
 libfstfarscript_la_SOURCES = farscript.cc strings.cc main.cc
-libfstfarscript_la_LDFLAGS = -version-info 1:0:0
+libfstfarscript_la_LDFLAGS = -version-info 1:0:0 -no-undefined
+libfstfarscript_la_LIBADD = $(top_builddir)/src/lib/libfst.la ./libfstfar.la
 endif
 
 if HAVE_BIN
--- origsrc/openfst-1.3.3/src/extensions/lookahead/Makefile.am	2011-12-31 00:28:58.000000000 +0900
+++ src/openfst-1.3.3/src/extensions/lookahead/Makefile.am	2013-02-07 20:16:29.876031200 +0900
@@ -6,13 +6,17 @@ ilabel_lookahead-fst.la olabel_lookahead
 
 libfstlookahead_la_SOURCES = arc_lookahead-fst.cc ilabel_lookahead-fst.cc \
 olabel_lookahead-fst.cc
-libfstlookahead_la_LDFLAGS = -version-info 0:0:0
+libfstlookahead_la_LDFLAGS = -version-info 0:0:0 -no-undefined
+libfstlookahead_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 arc_lookahead_fst_la_SOURCES = arc_lookahead-fst.cc
-arc_lookahead_fst_la_LDFLAGS = -module
+arc_lookahead_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+arc_lookahead_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 ilabel_lookahead_fst_la_SOURCES = ilabel_lookahead-fst.cc
-ilabel_lookahead_fst_la_LDFLAGS = -module
+ilabel_lookahead_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+ilabel_lookahead_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 olabel_lookahead_fst_la_SOURCES = olabel_lookahead-fst.cc
-olabel_lookahead_fst_la_LDFLAGS = -module
+olabel_lookahead_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+olabel_lookahead_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
--- origsrc/openfst-1.3.3/src/extensions/ngram/Makefile.am	2013-01-19 10:17:11.000000000 +0900
+++ src/openfst-1.3.3/src/extensions/ngram/Makefile.am	2013-02-07 20:19:58.572105200 +0900
@@ -4,7 +4,9 @@ libfstdir = @libfstdir@
 libfst_LTLIBRARIES = libfstngram.la ngram-fst.la
 
 ngram_fst_la_SOURCES = bitmap-index.cc ngram-fst.cc nthbit.cc
-ngram_fst_la_LDFLAGS = -module
+ngram_fst_la_LDFLAGS = -avoid-version -module -no-undefined
+ngram_fst_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 
 libfstngram_la_SOURCES = bitmap-index.cc ngram-fst.cc nthbit.cc
-libfstngram_la_LDFLAGS = -version-info 1:0:0
+libfstngram_la_LDFLAGS = -version-info 1:0:0 -no-undefined
+libfstngram_la_LIBADD = $(top_builddir)/src/lib/libfst.la
--- origsrc/openfst-1.3.3/src/extensions/pdt/Makefile.am	2013-01-19 10:17:11.000000000 +0900
+++ src/openfst-1.3.3/src/extensions/pdt/Makefile.am	2013-02-07 20:23:30.873863000 +0900
@@ -25,5 +25,6 @@ libfstdir = @libfstdir@
 if HAVE_SCRIPT
 libfst_LTLIBRARIES = libfstpdtscript.la
 libfstpdtscript_la_SOURCES = pdtscript.cc
-libfstpdtscript_la_LDFLAGS = -version-info 1:0:0
+libfstpdtscript_la_LDFLAGS = -version-info 1:0:0 -no-undefined
+libfstpdtscript_la_LIBADD = $(top_builddir)/src/script/libfstscript.la
 endif
--- origsrc/openfst-1.3.3/src/include/fst/register.h	2013-01-24 11:36:23.000000000 +0900
+++ src/openfst-1.3.3/src/include/fst/register.h	2013-02-07 08:12:31.619479400 +0900
@@ -75,7 +75,11 @@ class FstRegister : public GenericRegist
 
     ConvertToLegalCSymbol(&legal_type);
 
+#if defined(__CYGWIN__)
+    return legal_type + "-fst.dll";
+#else
     return legal_type + "-fst.so";
+#endif
   }
 };
 
--- origsrc/openfst-1.3.3/src/include/fst/script/register.h	2013-01-24 11:36:26.000000000 +0900
+++ src/openfst-1.3.3/src/include/fst/script/register.h	2013-02-07 08:13:30.776706000 +0900
@@ -71,7 +71,11 @@ class FstClassIORegister
     string legal_type(key);
     ConvertToLegalCSymbol(&legal_type);
 
+#if defined(__CYGWIN__)
+    return legal_type + "-arc.dll";
+#else
     return legal_type + "-arc.so";
+#endif
   }
 };
 
--- origsrc/openfst-1.3.3/src/include/fst/script/script-impl.h	2013-01-24 11:36:26.000000000 +0900
+++ src/openfst-1.3.3/src/include/fst/script/script-impl.h	2013-02-07 08:14:05.964206000 +0900
@@ -140,7 +140,11 @@ class GenericOperationRegister
     string legal_type(key.second);  // the arc type
     ConvertToLegalCSymbol(&legal_type);
 
+#if defined(__CYGWIN__)
+    return legal_type + "-arc.dll";
+#else
     return legal_type + "-arc.so";
+#endif
   }
 };
 
--- origsrc/openfst-1.3.3/src/include/fst/script/weight-class.h	2013-01-24 11:36:26.000000000 +0900
+++ src/openfst-1.3.3/src/include/fst/script/weight-class.h	2013-02-07 08:14:43.307956000 +0900
@@ -193,7 +193,11 @@ class WeightClassRegister : public Gener
                                                    WeightClassRegister> {
  protected:
   virtual string ConvertKeyToSoFilename(const string &key) const {
+#if defined(__CYGWIN__)
+    return key + ".dll";
+#else
     return key + ".so";
+#endif
   }
 };
 
--- origsrc/openfst-1.3.3/src/lib/Makefile.am	2013-01-19 10:17:11.000000000 +0900
+++ src/openfst-1.3.3/src/lib/Makefile.am	2013-02-07 08:03:04.217870400 +0900
@@ -3,4 +3,4 @@ AM_CPPFLAGS = -I$(srcdir)/../include $(I
 lib_LTLIBRARIES = libfst.la
 libfst_la_SOURCES = compat.cc flags.cc fst.cc properties.cc \
 symbol-table.cc util.cc symbol-table-ops.cc mapped-file.cc
-libfst_la_LDFLAGS = -version-info 1:0:0
+libfst_la_LDFLAGS = -version-info 1:0:0 -no-undefined
--- origsrc/openfst-1.3.3/src/script/Makefile.am	2013-01-19 10:17:11.000000000 +0900
+++ src/openfst-1.3.3/src/script/Makefile.am	2013-02-07 09:02:36.752141900 +0900
@@ -11,5 +11,6 @@ replace.cc reverse.cc reweight.cc rmepsi
 shortest-distance.cc shortest-path.cc synchronize.cc text-io.cc topsort.cc \
 union.cc weight-class.cc verify.cc
 
-libfstscript_la_LDFLAGS = -version-info 1:0:0
+libfstscript_la_LDFLAGS = -version-info 1:0:0 -no-undefined
+libfstscript_la_LIBADD = $(top_builddir)/src/lib/libfst.la
 endif
