--- origsrc/libipuz-0.4.4/libipuz/meson.build	2023-08-30 23:48:35.000000000 +0900
+++ src/libipuz-0.4.4/libipuz/meson.build	2024-01-05 14:50:47.257325500 +0900
@@ -78,7 +78,7 @@ libipuz_deps = [
   dependency('json-glib-1.0'),
 ]
 
-libipuz_lib = shared_library('libipuz-' + api_version,
+libipuz_lib = shared_library('ipuz-' + api_version,
   libipuz_sources + enum_sources + libipuz_private_headers + libipuz_private_sources,
   dependencies: libipuz_deps,
   include_directories: [rootinc, libipuzinc],
