--- origsrc/libipuz-0.4.2/libipuz/meson.build	2023-03-30 13:22:51.000000000 +0900
+++ src/libipuz-0.4.2/libipuz/meson.build	2023-08-23 10:37:34.244589300 +0900
@@ -70,7 +70,7 @@ libipuz_deps = [
   dependency('json-glib-1.0'),
 ]
 
-libipuz_lib = shared_library('libipuz-' + api_version,
+libipuz_lib = shared_library('ipuz-' + api_version,
   libipuz_sources + enum_sources + libipuz_private_headers,
   dependencies: libipuz_deps,
   include_directories: [rootinc, libipuzinc],
