--- origsrc/dav1d-0.3.1/meson.build	2019-05-12 00:23:10.000000000 +0900
+++ src/dav1d-0.3.1/meson.build	2019-05-30 23:38:04.027410400 +0900
@@ -334,8 +334,8 @@ if is_asm_enabled and host_machine.cpu_f
         nasm_r = run_command(nasm, '-v')
         out = nasm_r.stdout().strip().split()
         if out[1].to_lower() == 'version'
-            if out[2].version_compare('<2.13.02')
-                error('nasm 2.13.02 or later is required, found nasm @0@'.format(out[2]))
+            if out[2].version_compare('<2.13.01')
+                error('nasm 2.13.01 or later is required, found nasm @0@'.format(out[2]))
             endif
         else
             error('unexpected nasm version string: @0@'.format(nasm_r.stdout()))
