--- origsrc/dav1d-0.9.1/meson.build	2021-07-29 05:31:27.000000000 +0900
+++ src/dav1d-0.9.1/meson.build	2021-08-24 22:25:09.711618700 +0900
@@ -403,8 +403,8 @@ if is_asm_enabled and host_machine.cpu_f
 
         out = nasm_r.stdout().strip().split()
         if out[1].to_lower() == 'version'
-            if out[2].version_compare('<2.13.02')
-                error('nasm 2.13.02 or later is required, found nasm @0@'.format(out[2]))
+            if out[2].version_compare('<2.13.01')
+                error('nasm 2.13.01 or later is required, found nasm @0@'.format(out[2]))
             elif out[2].version_compare('<2.14') and get_option('enable_avx512')
                 error('nasm 2.14 or later is required for AVX-512 asm.\n' +
                        'AVX-512 asm can be disabled with \'-Denable_avx512=false\'')
