--- origsrc/dav1d-0.5.2/meson.build	2019-12-03 02:19:06.000000000 +0900
+++ src/dav1d-0.5.2/meson.build	2019-12-24 10:14:02.606007700 +0900
@@ -364,8 +364,8 @@ if is_asm_enabled and host_machine.cpu_f
 
         out = nasm_r.stdout().strip().split()
         if out[1].to_lower() == 'version'
-            if out[2].version_compare('<2.13.02')
-                error('nasm 2.13.02 or later is required, found nasm @0@'.format(out[2]))
+            if out[2].version_compare('<2.13.01')
+                error('nasm 2.13.01 or later is required, found nasm @0@'.format(out[2]))
             endif
         else
             error('unexpected nasm version string: @0@'.format(nasm_r.stdout()))
