--- origsrc/libblip_buf-1.1.1/CMakeLists.txt	1970-01-01 09:00:00.000000000 +0900
+++ src/libblip_buf-1.1.1/CMakeLists.txt	2025-06-15 07:28:36.953497100 +0900
@@ -0,0 +1,24 @@
+cmake_minimum_required(VERSION 3.28)
+project(libblip_buf
+  VERSION 1.1.1
+  DESCRIPTION "Small waveform synthesis library"
+  HOMEPAGE_URL "https://gitlab.com/TASVideos/libblip_buf")
+
+include(GNUInstallDirs)
+include(CTest)
+
+add_library(blip_buf SHARED "blip_buf.c")
+set_target_properties(blip_buf PROPERTIES SOVERSION ${PROJECT_VERSION_MAJOR})
+
+install(TARGETS blip_buf
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+install(FILES blip_buf.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libblip_buf.pc.in ${CMAKE_BINARY_DIR}/libblip_buf.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libblip_buf.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libblip_buf-1.1.1/blip_buf.txt	2018-12-04 13:17:24.000000000 +0900
+++ src/libblip_buf-1.1.1/blip_buf.txt	2025-06-15 07:21:21.159693500 +0900
@@ -1,4 +1,4 @@
-blip_buf 1.1.0
+blip_buf 1.1.1
 --------------
 Author  : Shay Green <gblargg@gmail.com>
 Website : http://www.slack.net/~ant/
--- origsrc/libblip_buf-1.1.1/libblip_buf.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libblip_buf-1.1.1/libblip_buf.pc.in	2025-06-15 07:28:45.570720600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libblip_buf
+Description: Small waveform synthesis library
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lblip_buf
+Cflags: -I${includedir}
--- origsrc/libblip_buf-1.1.1/readme.txt	2018-12-04 13:17:24.000000000 +0900
+++ src/libblip_buf-1.1.1/readme.txt	2025-06-15 07:21:40.676283100 +0900
@@ -1,4 +1,4 @@
-blip_buf 1.1.0: Band-Limited Audio Buffer
+blip_buf 1.1.1: Band-Limited Audio Buffer
 -----------------------------------------
 Blip_buf is a small waveform synthesis library meant for use in classic
 video game sound chip emulation. It greatly simplifies sound chip
