--- origsrc/ChaiScript-6.0.0/unittests/catch.hpp	2017-02-23 07:56:04.000000000 +0900
+++ src/ChaiScript-6.0.0/unittests/catch.hpp	2018-02-15 12:59:53.190952900 +0900
@@ -2281,7 +2281,7 @@ namespace Catch {
 // #included from: catch_timer.h
 #define TWOBLUECUBES_CATCH_TIMER_H_INCLUDED
 
-#ifdef CATCH_PLATFORM_WINDOWS
+#if defined(CATCH_PLATFORM_WINDOWS) && !defined(__CYGWIN__)
 typedef unsigned long long uint64_t;
 #else
 #include <stdint.h>
@@ -10521,4 +10521,3 @@ int main (int argc, char * const argv[])
 using Catch::Detail::Approx;
 
 #endif // TWOBLUECUBES_SINGLE_INCLUDE_CATCH_HPP_INCLUDED
-
