--- origsrc/ChaiScript-5.8.6/unittests/catch.hpp	2016-10-29 02:03:31.000000000 +0900
+++ src/ChaiScript-5.8.6/unittests/catch.hpp	2016-12-30 15:03:32.293930900 +0900
@@ -2249,7 +2249,7 @@ namespace Catch {
 // #included from: catch_timer.h
 #define TWOBLUECUBES_CATCH_TIMER_H_INCLUDED
 
-#ifdef CATCH_PLATFORM_WINDOWS
+#if defined(CATCH_PLATFORM_WINDOWS) && !defined(__CYGWIN__)
 typedef unsigned long long uint64_t;
 #else
 #include <stdint.h>
@@ -10411,4 +10411,3 @@ int main (int argc, char * const argv[])
 using Catch::Detail::Approx;
 
 #endif // TWOBLUECUBES_SINGLE_INCLUDE_CATCH_HPP_INCLUDED
-
