--- origsrc/libev-4.11/Makefile.am	2012-01-25 01:32:47.000000000 +0900
+++ src/libev-4.11/Makefile.am	2012-06-08 08:35:03.476573900 +0900
@@ -14,7 +14,7 @@ include_HEADERS = ev.h ev++.h event.h
 lib_LTLIBRARIES = libev.la
 
 libev_la_SOURCES = ev.c event.c
-libev_la_LDFLAGS = -version-info $(VERSION_INFO)
+libev_la_LDFLAGS = -version-info $(VERSION_INFO) -no-undefined
 
 ev.3: ev.pod
 	pod2man -n LIBEV -r "libev-$(VERSION)" -c "libev - high performance full featured event loop" -s3 <$< >$@
