--- origsrc/cyassl-2.7.0/configure.ac	2013-06-18 04:06:38.000000000 +0900
+++ src/cyassl-2.7.0/configure.ac	2013-08-03 01:18:01.671721900 +0900
@@ -1015,7 +1015,7 @@ AC_ARG_ENABLE([crl-monitor],
 if test "$ENABLED_CRL_MONITOR" = "yes"
 then
     case $host_os in
-    *linux* | *darwin* | *freebsd*)
+    *linux* | *darwin* | *freebsd* | *cygwin*)
         AM_CFLAGS="$AM_CFLAGS -DHAVE_CRL_MONITOR" ;;
     *)
         AC_MSG_ERROR([crl monitor only allowed on linux, OS X, or freebsd]) ;;
