--- origsrc/ls-qpack-2.5.0/CMakeLists.txt	2022-10-20 06:36:06.000000000 +0900
+++ src/ls-qpack-2.5.0/CMakeLists.txt	2022-12-02 16:58:25.143609000 +0900
@@ -20,9 +20,12 @@ add_library(ls-qpack "")
 target_include_directories(ls-qpack PUBLIC .)
 target_sources(ls-qpack PRIVATE lsqpack.c)
 
-target_include_directories(ls-qpack PRIVATE deps/xxhash/)
 if(LSQPACK_XXH)
+    target_include_directories(ls-qpack PRIVATE deps/xxhash/)
     target_sources(ls-qpack PRIVATE deps/xxhash/xxhash.c)
+else()
+    target_link_libraries(ls-qpack PRIVATE xxhash)
+    set_target_properties(ls-qpack PROPERTIES SOVERSION ${SOVERSION})
 endif()
 
 if(MSVC)
@@ -98,3 +101,13 @@ endif()
 if(LSQPACK_BIN)
     add_subdirectory(bin)
 endif()
+
+install(TARGETS ls-qpack RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+install(FILES lsqpack.h lsxpack_header.h DESTINATION include/ls-qpack)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include/ls-qpack)
+configure_file(${CMAKE_SOURCE_DIR}/ls-qpack.pc.in ${CMAKE_BINARY_DIR}/ls-qpack.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/ls-qpack.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/ls-qpack-2.5.0/bin/CMakeLists.txt	2022-10-20 06:36:06.000000000 +0900
+++ src/ls-qpack-2.5.0/bin/CMakeLists.txt	2022-12-02 16:57:19.538060700 +0900
@@ -2,8 +2,13 @@ function(lsqpack_add_executable TARGET)
     add_executable(${TARGET} "")
     target_link_libraries(${TARGET} PRIVATE ls-qpack)
 
-    target_sources(${TARGET} PRIVATE ${TARGET}.c ../deps/xxhash/xxhash.c)
-    target_include_directories(${TARGET} PRIVATE ../deps/xxhash)
+    target_sources(${TARGET} PRIVATE ${TARGET}.c)
+    if(LSQPACK_XXH)
+        target_sources(${TARGET} PRIVATE ../deps/xxhash/xxhash.c)
+        target_include_directories(${TARGET} PRIVATE ../deps/xxhash)
+    else()
+        target_link_libraries(${TARGET} PRIVATE xxhash)
+    endif()
 
     if(MSVC)
         target_include_directories(${TARGET} PRIVATE ../wincompat)
--- origsrc/ls-qpack-2.5.0/ls-qpack.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/ls-qpack-2.5.0/ls-qpack.pc.in	2022-12-02 16:57:19.540022800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: ls-qpack
+Description: 
+Version: @VERSION@
+Libs: -L${libdir} -lls-qpack
+Cflags: -I${includedir}
