--- origsrc/chan-0.0.4/Makefile.am	2016-01-23 14:25:38.000000000 +0900
+++ src/chan-0.0.4/Makefile.am	2025-01-31 18:48:13.182881500 +0900
@@ -2,6 +2,7 @@ LDADD = $(LIBS)
 
 lib_LTLIBRARIES = libchan.la
 libchan_la_SOURCES = src/chan.c src/queue.c
+libchan_la_LDFLAGS = -no-undefined
 pkginclude_HEADERS = src/chan.h src/queue.h
 
 check_PROGRAMS = src/chan_test
@@ -28,3 +29,5 @@ examples_unbuffered_SOURCES = examples/u
 examples_unbuffered_LDADD = libchan.la
 examples_unbuffered_LDFLAGS = -no-install
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/chan-0.0.4/chan.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/chan-0.0.4/chan.pc.in	2025-01-31 18:48:13.182881500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: chan
+Description: Pure C implementation of Go channels
+Version: @VERSION@
+Libs: -L${libdir} -lchan
+Cflags: -I${includedir}
--- origsrc/chan-0.0.4/configure.ac	2016-01-23 14:25:38.000000000 +0900
+++ src/chan-0.0.4/configure.ac	2025-01-31 18:48:13.182881500 +0900
@@ -7,5 +7,5 @@ AC_PROG_CC
 AC_CHECK_LIB([pthread], [pthread_mutex_init], [], [AC_MSG_ERROR([pthread not found])])
 AC_CHECK_LIB([rt], [clock_gettime])
 
-AC_OUTPUT([Makefile])
+AC_OUTPUT([chan.pc Makefile])
 
