--- origsrc/libbatch/CMakeLists.txt	2017-10-13 08:59:53.631145900 +0900
+++ src/libbatch/CMakeLists.txt	2017-10-13 09:06:09.160000000 +0900
@@ -46,7 +46,7 @@ IF(WIN32 AND NOT CYGWIN)
   # different installation directories for LibBatchConfig.cmake:
   SET(INSTALL_CMAKE_CONFIG_DIR cmake)
 ELSE()
-  SET(INSTALL_CMAKE_CONFIG_DIR share/cmake/libbatch)
+  SET(INSTALL_CMAKE_CONFIG_DIR lib/cmake/libbatch)
 ENDIF()
 SET(INSTALL_INCLUDE_DIR include)
 
@@ -140,3 +140,9 @@ SET(CPACK_SOURCE_IGNORE_FILES /\\\\.;/CV
 SET(CPACK_SOURCE_PACKAGE_FILE_NAME ${PROJECT_NAME}-${${PROJECT_NAME}_VERSION})
 INCLUDE(CPack)
 
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libbatch.pc.in ${CMAKE_BINARY_DIR}/libbatch.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libbatch.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libbatch/libbatch.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libbatch/libbatch.pc.in	2017-10-13 09:06:02.607000200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libbatch
+Description: Generic batch management library
+Version: @LIBBATCH_VERSION@
+Libs: -L${libdir} -lbatch
+Cflags: -I${includedir}
--- origsrc/libbatch/misc/CMakeLists.txt	2017-10-13 08:59:48.000000000 +0900
+++ src/libbatch/misc/CMakeLists.txt	2017-10-13 08:59:55.333454700 +0900
@@ -21,4 +21,4 @@
 #
 
 INSTALL(FILES check_libbatch.m4
-        DESTINATION share/libbatch/misc)
+        DESTINATION share/aclocal)
--- origsrc/libbatch/src/CMakeLists.txt	2017-10-13 08:59:53.786756000 +0900
+++ src/libbatch/src/CMakeLists.txt	2017-10-13 08:59:55.333454700 +0900
@@ -77,7 +77,8 @@ ENDIF (WIN32)
 
 INSTALL(TARGETS batch 
         EXPORT LibBatchTargets 
-        DESTINATION lib${LIB_SUFFIX})
+        RUNTIME DESTINATION bin
+        ARCHIVE DESTINATION lib${LIB_SUFFIX})
 INSTALL(FILES ${HDR_FILES} DESTINATION include/libbatch)
 
 IF (LIBBATCH_PYTHON_WRAPPING AND PYTHONINTERP_FOUND AND PYTHONLIBS_FOUND AND SWIG_FOUND)
