--- origsrc/libbatch/misc/CMakeLists.txt	2014-03-30 22:34:20.000000000 +0900
+++ src/libbatch/misc/CMakeLists.txt	2014-03-30 22:37:50.762447700 +0900
@@ -21,4 +21,4 @@
 #  email : webmaster.salome@opencascade.com
 
 INSTALL(FILES check_libbatch.m4
-        DESTINATION share/libbatch/misc)
+        DESTINATION share/aclocal)
--- origsrc/libbatch/src/CMakeLists.txt	2014-03-30 22:34:23.887123000 +0900
+++ src/libbatch/src/CMakeLists.txt	2014-03-30 22:40:22.794155900 +0900
@@ -77,7 +77,8 @@ ENDIF (WIN32)
 
 INSTALL(TARGETS batch 
         EXPORT LibBatchTargets 
-        DESTINATION lib${LIB_SUFFIX})
+        RUNTIME DESTINATION bin
+        ARCHIVE DESTINATION lib${LIB_SUFFIX})
 INSTALL(FILES ${HDR_FILES} DESTINATION include/libbatch)
 
 IF (LIBBATCH_PYTHON_WRAPPING AND PYTHONINTERP_FOUND AND PYTHONLIBS_FOUND AND SWIG_FOUND)
