--- origsrc/raylib-5.5/raylib.pc.in	2024-11-18 21:21:10.000000000 +0900
+++ src/raylib-5.5/raylib.pc.in	2025-01-10 04:29:04.182450800 +0900
@@ -7,7 +7,7 @@ Name: raylib
 Description: A simple and easy-to-use library to enjoy videogames programming
 URL: https://github.com/raysan5/raylib
 Version: @PROJECT_VERSION@
-Libs: -L"${libdir}" -lraylib @PKG_CONFIG_LIBS_EXTRA@
+Libs: -L${libdir} -lraylib @PKG_CONFIG_LIBS_EXTRA@
 Libs.private: @PKG_CONFIG_LIBS_PRIVATE@
 Requires.private: @GLFW_PKG_DEPS@
-Cflags: -I"${includedir}"
+Cflags: -I${includedir}
