--- origsrc/raylib-5.0/raylib.pc.in	2023-11-15 19:30:50.000000000 +0900
+++ src/raylib-5.0/raylib.pc.in	2024-02-20 13:17:14.354338100 +0900
@@ -7,7 +7,7 @@ Name: raylib
 Description: A simple and easy-to-use library to enjoy videogames programming
 URL: https://github.com/raysan5/raylib
 Version: @PROJECT_VERSION@
-Libs: -L"${libdir}" -lraylib @PKG_CONFIG_LIBS_EXTRA@
+Libs: -L${libdir} -lraylib @PKG_CONFIG_LIBS_EXTRA@
 Libs.private: @PKG_CONFIG_LIBS_PRIVATE@
 Requires.private: @GLFW_PKG_DEPS@
-Cflags: -I"${includedir}"
+Cflags: -I${includedir}
