--- origsrc/raylib-2.0.0/src/CMakeLists.txt	2018-07-22 02:42:23.000000000 +0900
+++ src/raylib-2.0.0/src/CMakeLists.txt	2019-05-13 15:33:19.980328600 +0900
@@ -142,7 +142,8 @@ if(${SHARED})
     set_target_properties(${RAYLIB}_shared PROPERTIES OUTPUT_NAME ${RAYLIB})
     install(
       TARGETS ${RAYLIB}_shared
-      LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+      RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+      ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
       PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
     )
   endif()
--- origsrc/raylib-2.0.0/src/utils.c	2018-07-22 02:42:23.000000000 +0900
+++ src/raylib-2.0.0/src/utils.c	2019-05-13 15:37:31.311900000 +0900
@@ -57,11 +57,12 @@
 #include <stdarg.h>                 // Required for: va_list, va_start(), vfprintf(), va_end()
 #include <string.h>                 // Required for: strlen(), strrchr(), strcmp()
 
+#if !defined(__CYGWIN__)
 /* This should be in <stdio.h>, but Travis doesn't find it... */
 FILE *funopen(const void *cookie, int (*readfn)(void *, char *, int),
               int (*writefn)(void *, const char *, int),
               fpos_t (*seekfn)(void *, fpos_t, int), int (*closefn)(void *));
-
+#endif
 
 #if defined(PLATFORM_DESKTOP) || defined(PLATFORM_RPI)
     #define STB_IMAGE_WRITE_IMPLEMENTATION
