--- origsrc/libgeier/configure.ac	2011-02-21 09:58:06.000000000 +0900
+++ src/libgeier/configure.ac	2011-02-21 10:05:37.290110100 +0900
@@ -33,7 +33,7 @@ AC_CONFIG_SRCDIR([libgeier.pc.in])
 # checks for programs
 AC_PROG_CC
 
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 GTK_DOC_CHECK(1.0)
 
--- origsrc/libgeier/src/Makefile.am	2011-02-21 09:58:06.000000000 +0900
+++ src/libgeier/src/Makefile.am	2011-02-21 10:26:56.045170700 +0900
@@ -33,7 +33,7 @@ libgeier_la_SOURCES = \
 	dsig-rewrite-vorgang.c \
 	dsig-rewrite-datenlieferant.c
 
-libgeier_la_LDFLAGS = -version-info 0
+libgeier_la_LDFLAGS = -version-info 0 -no-undefined
 libgeier_la_LIBADD = $(top_builddir)/src/@crypto_module@/libgeier_@crypto_module@.la
 
 noinst_HEADERS = \
@@ -66,5 +66,5 @@ CLEANFILES = *~
 bin_PROGRAMS = geier
 
 geier_SOURCES = geier.c
-geier_LDADD = libgeier.la
+geier_LDADD = libgeier.la -largp
 
