--- origsrc/libgeier-0.11/configure.ac	2009-12-19 04:03:49.000000000 +0900
+++ src/libgeier-0.11/configure.ac	2010-12-19 12:39:09.347423800 +0900
@@ -33,7 +33,7 @@ AC_CONFIG_SRCDIR([libgeier.pc.in])
 # checks for programs
 AC_PROG_CC
 
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 GTK_DOC_CHECK(1.0)
 
--- origsrc/libgeier-0.11/src/Makefile.am	2009-12-19 04:03:49.000000000 +0900
+++ src/libgeier-0.11/src/Makefile.am	2010-12-19 13:20:13.821163000 +0900
@@ -33,7 +33,7 @@ libgeier_la_SOURCES = \
 	dsig-rewrite-vorgang.c \
 	dsig-rewrite-datenlieferant.c
 
-libgeier_la_LDFLAGS = -version-info 0
+libgeier_la_LDFLAGS = -version-info 0 -no-undefined
 libgeier_la_LIBADD = $(top_builddir)/src/@crypto_module@/libgeier_@crypto_module@.la
 
 noinst_HEADERS = \
@@ -66,5 +66,5 @@ CLEANFILES = *~
 bin_PROGRAMS = geier
 
 geier_SOURCES = geier.c
-geier_LDADD = libgeier.la
+geier_LDADD = libgeier.la -largp
 
--- origsrc/libgeier-0.11/tests/Makefile.am	2009-12-19 04:03:49.000000000 +0900
+++ src/libgeier-0.11/tests/Makefile.am	2010-12-19 13:44:02.771010800 +0900
@@ -1,7 +1,6 @@
 # tests run for make check
 TESTS = \
-	$(check_PROGRAMS) \
-	$(check_SCRIPTS)
+	$(check_PROGRAMS)
 
 #XFAIL_TESTS =
 
@@ -10,11 +9,11 @@ check_PROGRAMS = \
 	test_base64 \
 	test_pkcs7_decrypt
 
-check_SCRIPTS = \
-	test_dsig_digestvalue \
-	test_dsig_send_text \
-	test_send_text \
-	test_validate
+#check_SCRIPTS = \
+#	test_dsig_digestvalue \
+#	test_dsig_send_text \
+#	test_send_text \
+#	test_validate
 
 
 TESTDATADIR = $(top_srcdir)/tests/data
