--- origsrc/ogre-13.1.1/CMake/InstallResources.cmake	2021-10-15 05:29:33.000000000 +0900
+++ src/ogre-13.1.1/CMake/InstallResources.cmake	2021-11-04 20:25:34.588275000 +0900
@@ -243,6 +243,8 @@ include(CMakePackageConfigHelpers)
 if (NOT OGRE_CMAKE_DIR)
   if(WIN32 OR APPLE)
     set(OGRE_CMAKE_DIR "CMake")
+  elseif(CYGWIN)
+    set(OGRE_CMAKE_DIR "${OGRE_LIB_DIRECTORY}/cmake/OGRE")
   else()
     set(OGRE_CMAKE_DIR "${OGRE_LIB_DIRECTORY}/OGRE/cmake")
   endif()
--- origsrc/ogre-13.1.1/CMake/Utils/OgreConfigTargets.cmake	2021-10-15 05:29:33.000000000 +0900
+++ src/ogre-13.1.1/CMake/Utils/OgreConfigTargets.cmake	2021-11-04 20:26:37.033840000 +0900
@@ -259,7 +259,11 @@ function(ogre_config_plugin PLUGINNAME)
       set_target_properties(${PLUGINNAME} PROPERTIES PREFIX "")
     endif (CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   endif (OGRE_STATIC)
-  ogre_install_target(${PLUGINNAME} ${OGRE_PLUGIN_PATH} TRUE)
+  if (CYGWIN)
+    install(TARGETS ${PLUGINNAME} RUNTIME DESTINATION ${OGRE_LIB_DIRECTORY}/OGRE)
+  else()
+    ogre_install_target(${PLUGINNAME} ${OGRE_PLUGIN_PATH} TRUE)
+  endif()
 
   if (OGRE_INSTALL_PDB)
     # install debug pdb files
--- origsrc/ogre-13.1.1/Components/Python/CMakeLists.txt	2021-10-15 05:29:33.000000000 +0900
+++ src/ogre-13.1.1/Components/Python/CMakeLists.txt	2021-11-04 09:04:33.731572100 +0900
@@ -45,7 +45,7 @@ set_source_files_properties(../../OgreMa
 if(APPLE)
     set(PYTHON_LIBRARIES "-Wl,-undefined,dynamic_lookup")
 elseif(UNIX)
-    set(PYTHON_LIBRARIES "")
+    set(PYTHON_LIBRARIES "${PYTHON_LIBRARY}")
 endif()
 
 set(CMAKE_WARN_DEPRECATED OFF)
