diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/sinbad-ogre-dd30349ea667/CMake/CMakeLists.txt src/sinbad-ogre-dd30349ea667/CMake/CMakeLists.txt
--- origsrc/sinbad-ogre-dd30349ea667/CMake/CMakeLists.txt	2013-12-02 02:28:12.000000000 +0900
+++ src/sinbad-ogre-dd30349ea667/CMake/CMakeLists.txt	2015-12-28 20:35:39.782151100 +0900
@@ -16,7 +16,7 @@
 if(WIN32 OR APPLE)
 	set(OGRE_CMAKE_DIR "CMake")
 else()
-	set(OGRE_CMAKE_DIR "${OGRE_LIB_DIRECTORY}/OGRE/cmake")
+	set(OGRE_CMAKE_DIR "${OGRE_LIB_DIRECTORY}/cmake/OGRE")
 endif()
 
 set(INST_FILES
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/sinbad-ogre-dd30349ea667/CMakeLists.txt src/sinbad-ogre-dd30349ea667/CMakeLists.txt
--- origsrc/sinbad-ogre-dd30349ea667/CMakeLists.txt	2013-12-02 02:28:12.000000000 +0900
+++ src/sinbad-ogre-dd30349ea667/CMakeLists.txt	2015-12-28 19:51:40.919448800 +0900
@@ -183,7 +183,7 @@ if ((CMAKE_COMPILER_IS_GNUCXX OR CMAKE_C
   endif (OGRE_GCC_VISIBILITY)
 
   # Fix x64 issues on Linux
-  if(OGRE_PLATFORM_X64 AND NOT APPLE)
+  if(OGRE_PLATFORM_X64 AND NOT APPLE AND NOT CYGWIN)
     add_definitions(-fPIC)
   endif()
 endif ((CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX) AND NOT MINGW)
@@ -555,4 +555,3 @@ include(Packaging)
 
 # Show feature summary
 include(FeatureSummary)
-
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/sinbad-ogre-dd30349ea667/OgreMain/include/GLX/OgreTimerImp.h src/sinbad-ogre-dd30349ea667/OgreMain/include/GLX/OgreTimerImp.h
--- origsrc/sinbad-ogre-dd30349ea667/OgreMain/include/GLX/OgreTimerImp.h	2013-12-02 02:28:12.000000000 +0900
+++ src/sinbad-ogre-dd30349ea667/OgreMain/include/GLX/OgreTimerImp.h	2015-12-28 20:04:26.710366100 +0900
@@ -30,6 +30,7 @@ THE SOFTWARE.
 #define __GLXTimer_H__
 
 #include "../OgrePrerequisites.h"
+#include <sys/time.h>
 
 namespace Ogre
 {
