--- origsrc/libcnotify-20240414/Makefile.am	2024-04-14 14:29:46.000000000 +0900
+++ src/libcnotify-20240414/Makefile.am	2025-05-16 14:06:57.769600200 +0900
@@ -3,9 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
 	libcnotify \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libcnotify-20240414/configure.ac	2024-04-14 14:29:46.000000000 +0900
+++ src/libcnotify-20240414/configure.ac	2025-05-16 14:06:57.785229400 +0900
@@ -91,7 +91,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
 AC_CONFIG_FILES([libcnotify/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
