--- origsrc/libmbd-0.13.0/CMakeLists.txt	2024-03-06 23:50:20.000000000 +0900
+++ src/libmbd-0.13.0/CMakeLists.txt	2026-01-19 11:20:20.804589800 +0900
@@ -110,3 +110,10 @@ add_subdirectory(src)
 if(BUILD_TESTING)
     add_subdirectory(tests)
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libmbd.pc.in ${CMAKE_BINARY_DIR}/libmbd.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libmbd.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libmbd-0.13.0/libmbd.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libmbd-0.13.0/libmbd.pc.in	2026-01-19 11:20:20.806608700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libmbd
+Description: Many-body dispersion library
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lmbd
+Cflags: -I${includedir}/mbd
--- origsrc/libmbd-0.13.0/src/CMakeLists.txt	2026-01-19 11:20:19.614495000 +0900
+++ src/libmbd-0.13.0/src/CMakeLists.txt	2026-01-19 11:20:20.810377000 +0900
@@ -90,6 +90,7 @@ add_library(Mbd INTERFACE)
 target_link_libraries(Mbd INTERFACE mbd)
 if(CMAKE_INSTALL_LIBDIR)
     install(TARGETS mbd Mbd EXPORT MbdConfig
+        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
         LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
         ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
         PUBLIC_HEADER DESTINATION "${includedir}"
