--- origsrc/liberasurecode-1.6.0/Xorcode.pc.in	2018-09-21 23:28:15.000000000 +0900
+++ src/liberasurecode-1.6.0/Xorcode.pc.in	2018-11-20 17:29:44.517039100 +0900
@@ -11,5 +11,5 @@ Version: @XORCODE_VERSION@
 Requires:
 Conflicts:
 Libs: -L${libdir} @XORCODE_RLD_FLAGS@ -lXorcode
-Libs.private: @XORCODE_STATIC_LIBS@
+Libs.private: 
 Cflags: -I${includedir}/ -I${includedir}/xor_codes @XORCODE_CFLAGS@
--- origsrc/liberasurecode-1.6.0/erasurecode.pc.in	2018-09-21 23:28:15.000000000 +0900
+++ src/liberasurecode-1.6.0/erasurecode.pc.in	2018-11-20 17:29:57.875691200 +0900
@@ -11,5 +11,5 @@ Version: @LIBERASURECODE_VERSION@
 Requires:
 Conflicts:
 Libs: -L${libdir} -lerasurecode -ldl
-Libs.private: @ERASURECODE_STATIC_LIBS@ -lz
+Libs.private: -lz
 Cflags: -I${includedir}/ -I${includedir}/liberasurecode
--- origsrc/liberasurecode-1.6.0/erasurecode_rs_vand.pc.in	2018-09-21 23:28:15.000000000 +0900
+++ src/liberasurecode-1.6.0/erasurecode_rs_vand.pc.in	2018-11-20 17:30:06.367476000 +0900
@@ -11,5 +11,5 @@ Version: @ERASURECODE_RS_VAND_VERSION@
 Requires:
 Conflicts:
 Libs: -L${libdir} @ERASURECODE_RS_VAND_RLD_FLAGS@ -lerasurecode
-Libs.private: @ERASURECODE_RS_VAND_STATIC_LIBS@
+Libs.private: 
 Cflags: -I${includedir}/ -I${includedir}/erasurecode @ERASURECODE_RS_VAND_CFLAGS@
--- origsrc/liberasurecode-1.6.0/src/Makefile.am	2018-09-21 23:28:15.000000000 +0900
+++ src/liberasurecode-1.6.0/src/Makefile.am	2018-11-20 17:30:31.726919000 +0900
@@ -40,7 +40,7 @@ liberasurecode_la_LIBADD = \
 		-lpthread -lm -lz @GCOV_LDFLAGS@
 
 # Version format  (C - A).(A).(R) for C:R:A input
-liberasurecode_la_LDFLAGS = -rpath '$(libdir)' -version-info @LIBERASURECODE_VERSION_INFO@
+liberasurecode_la_LDFLAGS = -no-undefined -rpath '$(libdir)' -version-info @LIBERASURECODE_VERSION_INFO@
 
 MOSTLYCLEANFILES = *.gcda *.gcno *.gcov utils/chksum/*.gcda utils/chksum/*.gcno utils/chksum/*.gcov \
                    backends/null/*.gcda backends/null/*.gcno backends/null/*.gcov  \
--- origsrc/liberasurecode-1.6.0/src/backends/shss/shss.c	2018-09-21 23:28:15.000000000 +0900
+++ src/liberasurecode-1.6.0/src/backends/shss/shss.c	2018-11-20 17:33:41.316847100 +0900
@@ -31,6 +31,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <alloca.h>
 
 #include "erasurecode.h"
 #include "erasurecode_helpers.h"
@@ -64,6 +65,8 @@ struct shss_descriptor {
 #define SHSS_LIB_NAME "shss"
 #if defined(__MACOS__) || defined(__MACOSX__) || defined(__OSX__) || defined(__APPLE__)
 #define SHSS_SO_NAME "libshss.dylib"
+#elif defined(__CYGWIN__)
+#define SHSS_SO_NAME "cygshss-1.dll"
 #else
 #define SHSS_SO_NAME "libshss.so.1"
 #endif
--- origsrc/liberasurecode-1.6.0/src/builtin/null_code/Makefile.am	2018-09-21 23:28:15.000000000 +0900
+++ src/liberasurecode-1.6.0/src/builtin/null_code/Makefile.am	2018-11-20 17:30:53.260311800 +0900
@@ -5,7 +5,7 @@ libnullcode_la_SOURCES = null_code.c
 libnullcode_la_CPPFLAGS = -I$(top_srcdir)/include/null_code @GCOV_FLAGS@
 
 # Version format  (C - A).(A).(R) for C:R:A input
-libnullcode_la_LDFLAGS = @GCOV_LDFLAGS@ -rpath '$(libdir)' -version-info 1:1:0
+libnullcode_la_LDFLAGS = -no-undefined @GCOV_LDFLAGS@ -rpath '$(libdir)' -version-info 1:1:0
 
 MOSTLYCLEANFILES = *.gcda *.gcno *.gcov
 
--- origsrc/liberasurecode-1.6.0/src/builtin/rs_vand/Makefile.am	2018-09-21 23:28:15.000000000 +0900
+++ src/liberasurecode-1.6.0/src/builtin/rs_vand/Makefile.am	2018-11-20 17:31:07.684954300 +0900
@@ -5,6 +5,6 @@ liberasurecode_rs_vand_la_SOURCES = rs_g
 liberasurecode_rs_vand_la_CPPFLAGS = -I$(top_srcdir)/include/rs_vand @GCOV_FLAGS@
 
 # Version format  (C - A).(A).(R) for C:R:A input
-liberasurecode_rs_vand_la_LDFLAGS = @GCOV_LDFLAGS@ -rpath '$(libdir)' -version-info 1:1:0
+liberasurecode_rs_vand_la_LDFLAGS = -no-undefined @GCOV_LDFLAGS@ -rpath '$(libdir)' -version-info 1:1:0
 
 MOSTLYCLEANFILES = *.gcda *.gcno *.gcov
--- origsrc/liberasurecode-1.6.0/src/builtin/xor_codes/Makefile.am	2018-09-21 23:28:15.000000000 +0900
+++ src/liberasurecode-1.6.0/src/builtin/xor_codes/Makefile.am	2018-11-20 17:31:19.602654800 +0900
@@ -5,7 +5,7 @@ libXorcode_la_SOURCES = xor_code.c xor_h
 libXorcode_la_CPPFLAGS = -I$(top_srcdir)/include/xor_codes $(SIMD_FLAGS) @GCOV_FLAGS@
 
 # Version format  (C - A).(A).(R) for C:R:A input
-libXorcode_la_LDFLAGS = @GCOV_LDFLAGS@ -rpath '$(libdir)' -version-info 1:1:0
+libXorcode_la_LDFLAGS = -no-undefined @GCOV_LDFLAGS@ -rpath '$(libdir)' -version-info 1:1:0
 
 MOSTLYCLEANFILES = *.gcda *.gcno *.gcov
 
