--- origsrc/libtinkerforge-1.0.0/Makefile.am	2014-03-04 05:36:50.000000000 +0900
+++ src/libtinkerforge-1.0.0/Makefile.am	2024-11-01 14:12:18.992525700 +0900
@@ -54,7 +54,7 @@ libtinkerforge_@TINKERFORGE_MULTI_LIB_VE
 														tinkerforge/bricklet_voltage.c \
 														tinkerforge/bricklet_voltage_current.c
 
-libtinkerforge_@TINKERFORGE_MULTI_LIB_VERSION@_la_LDFLAGS = -version-info $(TINKERFORGE_SO_VERSION)
+libtinkerforge_@TINKERFORGE_MULTI_LIB_VERSION@_la_LDFLAGS = -no-undefined -version-info $(TINKERFORGE_SO_VERSION)
 
 tinkerforge_includedir = $(includedir)/tinkerforge-$(TINKERFORGE_MULTI_LIB_VERSION)
 nobase_tinkerforge_include_HEADERS = 	tinkerforge/bricklet_voltage_current.h \
--- origsrc/libtinkerforge-1.0.0/tinkerforge.pc.in	2014-03-04 05:36:50.000000000 +0900
+++ src/libtinkerforge-1.0.0/tinkerforge.pc.in	2024-11-01 14:23:19.212620700 +0900
@@ -7,5 +7,5 @@ Name: @PACKAGE_NAME@
 Description: libtinkerforge - shared library for Tinkerforge's C/C++ bindings
 Version: @PACKAGE_VERSION@
 URL: @PACKAGE_URL@
-Libs: -L${libdir} -ltinkerforge-@TINKERFORGE_API_VERSION@
-Cflags: -I${includedir}/tinkerforge-@TINKERFORGE_API_VERSION@ -I${libdir}/tinkerforge-@TINKERFORGE_API_VERSION@/include
+Libs: -L${libdir} -ltinkerforge-@TINKERFORGE_MULTI_LIB_VERSION@
+Cflags: -I${includedir}/tinkerforge-@TINKERFORGE_MULTI_LIB_VERSION@ -I${libdir}/tinkerforge-@TINKERFORGE_MULTI_LIB_VERSION@/include
