--- origsrc/libtinkerforge-1.0.0/Makefile.am	2014-03-04 05:36:50.000000000 +0900
+++ src/libtinkerforge-1.0.0/Makefile.am	2015-03-23 18:10:12.822624300 +0900
@@ -54,7 +54,7 @@ libtinkerforge_@TINKERFORGE_MULTI_LIB_VE
 														tinkerforge/bricklet_voltage.c \
 														tinkerforge/bricklet_voltage_current.c
 
-libtinkerforge_@TINKERFORGE_MULTI_LIB_VERSION@_la_LDFLAGS = -version-info $(TINKERFORGE_SO_VERSION)
+libtinkerforge_@TINKERFORGE_MULTI_LIB_VERSION@_la_LDFLAGS = -no-undefined -version-info $(TINKERFORGE_SO_VERSION)
 
 tinkerforge_includedir = $(includedir)/tinkerforge-$(TINKERFORGE_MULTI_LIB_VERSION)
 nobase_tinkerforge_include_HEADERS = 	tinkerforge/bricklet_voltage_current.h \
