--- origsrc/rnp-0.16.2/src/lib/CMakeLists.txt	2022-09-22 18:27:54.000000000 +0900
+++ src/rnp-0.16.2/src/lib/CMakeLists.txt	2022-10-01 20:32:45.119425900 +0900
@@ -340,7 +340,6 @@ set_target_properties(librnp
   PROPERTIES
     VERSION "${RNP_VERSION}"
     SOVERSION "${RNP_VERSION_MAJOR}"
-    PREFIX "lib"
     OUTPUT_NAME "rnp"
 )
 
@@ -401,7 +400,7 @@ install(TARGETS librnp
 )
 
 # install dll only for windows
-if (WIN32)
+if (WIN32 OR CYGWIN)
   install(TARGETS librnp
     RUNTIME
       DESTINATION "${CMAKE_INSTALL_BINDIR}"
