--- origsrc/libestk-4.0.0/CMakeLists.txt	2015-04-23 05:08:36.000000000 +0900
+++ src/libestk-4.0.0/CMakeLists.txt	2017-01-02 17:17:26.373227600 +0900
@@ -6,9 +6,9 @@ add_definitions("-Wall -O2 -std=gnu11")
 # Version/Naming
 # ==============
 
-set(MAJOR 3)
-set(MINOR 3)
-set(PATCH 1)
+set(MAJOR 4)
+set(MINOR 0)
+set(PATCH 0)
 set(VERSION ${MAJOR}.${MINOR}.${PATCH})
 set(ESTK estk${MAJOR})
 
@@ -22,7 +22,12 @@ set_target_properties(${ESTK}
 	PROPERTIES VERSION ${VERSION}
 	SOVERSION ${MAJOR})
 
-target_link_libraries(${ESTK} m GLEW SDL SDL_image SDL_mixer)
+include(FindPkgConfig)
+pkg_check_modules(GLEW glew)
+pkg_check_modules(SDL sdl)
+pkg_check_modules(SDL_IMAGE SDL_image)
+pkg_check_modules(SDL_MIXER SDL_mixer)
+target_link_libraries(${ESTK} m ${GLEW_LIBRARIES} ${SDL_LIBRARIES} ${SDL_IMAGE_LIBRARIES} ${SDL_MIXER_LIBRARIES})
 
 # Demos
 # =====
@@ -49,3 +54,9 @@ install(FILES
 	src/estk.h
 	DESTINATION include)
 
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libestk4.pc.in ${CMAKE_BINARY_DIR}/libestk4.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libestk4.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libestk-4.0.0/libestk4.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libestk-4.0.0/libestk4.pc.in	2017-01-02 17:25:17.888132200 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libestk
+Description: OpenGL ES Toolkit
+Version: @VERSION@
+Libs: -L${libdir} -lestk4
+Cflags: -I${includedir}
+Requires: glew sdl SDL_image SDL_mixer
