--- origsrc/libredwg/bindings/python/Makefile.am	2011-09-25 14:58:36.000000000 +0900
+++ src/libredwg/bindings/python/Makefile.am	2011-09-25 15:56:04.670375100 +0900
@@ -46,12 +46,12 @@ BUILT_SOURCES = $(srcdir)/swig_wrap_pyth
 ##    _libredwg_la_CPPFLAGS = $(SWIG_PYTHON_CPPFLAGS) -I$(top_srcdir)/src
     _libredwg_la_CPPFLAGS = -I$(top_srcdir)/src $(AX_SWIG_PYTHON_CPPFLAGS)
 
-    _libredwg_la_LDFLAGS = -module
-    _libredwg_la_LIBADD = ../../src/libredwg.la
+    _libredwg_la_LDFLAGS = -module -avoid-version -no-undefined
+    _libredwg_la_LIBADD = ../../src/libredwg.la $(PYTHON_LDFLAGS)
 
 ## not elegant, but works!
     $(srcdir)/swig_wrap_python.c : $(SWIG_SOURCES)
-	swig -python -includeall -I$(top_srcdir)/src -o $@ $<
+	swig -python -includeall -I$(top_srcdir)/src -I$(srcdir) -o $@ $<
 
 ## should work out of the box, but $(SWIG) resolves to swig -c++
 ## which we don't want, and SWIG_PYTHON_OPT resolves to null,
ファイルorigsrc/libredwg/doc/LibreDWG.infoとsrc/libredwg/doc/LibreDWG.infoは違います
--- origsrc/libredwg/src/Makefile.am	2011-09-25 14:58:36.000000000 +0900
+++ src/libredwg/src/Makefile.am	2011-09-25 15:19:09.020666300 +0900
@@ -28,7 +28,7 @@ libredwg_la_SOURCES = \
 	encode.c \
         print.c
 
-libredwg_la_LDFLAGS = \
+libredwg_la_LDFLAGS = -no-undefined \
 	-version-info 0:0:0
 
 include_HEADERS = dwg.h
