--- origsrc/libredwg-0.9.2.2458/.version	1970-01-01 09:00:00.000000000 +0900
+++ src/libredwg-0.9.2.2458/.version	2019-12-04 10:14:53.343579600 +0900
@@ -0,0 +1 @@
+0.9.2.2458
--- origsrc/libredwg-0.9.2.2458/bindings/python/Makefile.am	2019-11-08 17:30:14.000000000 +0900
+++ src/libredwg-0.9.2.2458/bindings/python/Makefile.am	2019-12-04 10:13:39.923198600 +0900
@@ -46,8 +46,8 @@ _LibreDWG_la_SOURCES = $(BUILT_SOURCES)
 ## more magic: SWIG_PYTHON_CPPFLAGS resolves to null and python includes
 ## are not passed to gcc via -I
 _LibreDWG_la_CPPFLAGS = -I$(top_srcdir)/include $(AX_SWIG_PYTHON_CPPFLAGS)
-_LibreDWG_la_LDFLAGS = -module
-_LibreDWG_la_LIBADD = ../../src/libredwg.la
+_LibreDWG_la_LDFLAGS = -avoid-version -no-undefined -module
+_LibreDWG_la_LIBADD = ../../src/libredwg.la $(PYTHON_LIBS)
 
 ## not elegant, but works
 ## With macports gcc you need -I/opt/local/lib/gcc6/gcc/x86_64-apple-darwin15/6.4.0/include
--- origsrc/libredwg-0.9.2.2458/configure.ac	2019-11-08 17:30:14.000000000 +0900
+++ src/libredwg-0.9.2.2458/configure.ac	2019-12-04 10:14:33.669658100 +0900
@@ -10,7 +10,7 @@ dnl along with this program.  If not, se
 
 AC_PREREQ([2.61])
 AC_INIT([LibreDWG],
-        m4_esyscmd([build-aux/git-version-gen .tarball-version]),
+        [0.9.2.2458],
         [libredwg@gnu.org],,
         [https://savannah.gnu.org/projects/libredwg/])
 AC_CONFIG_AUX_DIR([build-aux])
