--- origsrc/libredwg-0.6.0.1203/.version	1970-01-01 09:00:00.000000000 +0900
+++ src/libredwg-0.6.0.1203/.version	2018-08-24 18:04:25.089025200 +0900
@@ -0,0 +1 @@
+UNKNOWN
--- origsrc/libredwg-0.6.0.1203/bindings/python/Makefile.am	2018-08-13 18:16:16.000000000 +0900
+++ src/libredwg-0.6.0.1203/bindings/python/Makefile.am	2018-08-24 18:00:47.065212200 +0900
@@ -45,8 +45,8 @@ _LibreDWG_la_SOURCES = $(BUILT_SOURCES)
 ## more magic: SWIG_PYTHON_CPPFLAGS resolves to null and python includes
 ## are not passed to gcc via -I
 _LibreDWG_la_CPPFLAGS = -I$(top_srcdir)/include $(AX_SWIG_PYTHON_CPPFLAGS)
-_LibreDWG_la_LDFLAGS = -module
-_LibreDWG_la_LIBADD = ../../src/libredwg.la
+_LibreDWG_la_LDFLAGS = -avoid-version -no-undefined -module
+_LibreDWG_la_LIBADD = ../../src/libredwg.la $(PYTHON_LDFLAGS)
 
 ## not elegant, but works
 ## With macports gcc you need -I/opt/local/lib/gcc6/gcc/x86_64-apple-darwin15/6.4.0/include
