--- origsrc/libredwg/Makefile.am	2014-07-12 19:13:30.000000000 +0900
+++ src/libredwg/Makefile.am	2014-07-14 21:36:41.319345300 +0900
@@ -19,6 +19,6 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = src doc examples bindings unit-testing
+SUBDIRS = src doc examples bindings
 
 EXTRA_DIST = README README-alpha
--- origsrc/libredwg/bindings/dwg.i	2014-07-12 19:13:30.000000000 +0900
+++ src/libredwg/bindings/dwg.i	2014-07-14 21:28:11.071449400 +0900
@@ -19,10 +19,10 @@
 %module libredwg
 
 %{
-#include "../../src/dwg.h"
+#include "dwg.h"
 %}
 
 %include "carrays.i"
 %array_functions(Dwg_Object, Dwg_Object_Array);
 
-#include "../../src/dwg.h"
+#include "dwg.h"
--- origsrc/libredwg/bindings/python/Makefile.am	2014-07-12 19:13:30.000000000 +0900
+++ src/libredwg/bindings/python/Makefile.am	2014-07-14 21:40:11.875009300 +0900
@@ -46,8 +46,8 @@ BUILT_SOURCES = $(srcdir)/swig_wrap_pyth
 ##    _libredwg_la_CPPFLAGS = $(SWIG_PYTHON_CPPFLAGS) -I$(top_srcdir)/src
     _libredwg_la_CPPFLAGS = -I$(top_srcdir)/src $(AX_SWIG_PYTHON_CPPFLAGS)
 
-    _libredwg_la_LDFLAGS = -module
-    _libredwg_la_LIBADD = ../../src/libredwg.la
+    _libredwg_la_LDFLAGS = -avoid-version -no-undefined -module
+    _libredwg_la_LIBADD = ../../src/libredwg.la $(PYTHON_LDFLAGS)
 
 ## not elegant, but works!
     $(srcdir)/swig_wrap_python.c : $(SWIG_SOURCES)
--- origsrc/libredwg/examples/Makefile.am	2014-07-12 19:13:30.000000000 +0900
+++ src/libredwg/examples/Makefile.am	2014-07-14 21:38:25.173837400 +0900
@@ -25,7 +25,7 @@ dwg_ps_SOURCES = dwg_ps.c
 dwg_ps_LDADD = -lps
 endif
 
-AM_LDFLAGS = -L../src -lredwg
+AM_LDFLAGS = $(top_builddir)/src/libredwg.la
 
 test_SOURCES = test.c
 
--- origsrc/libredwg/src/Makefile.am	2014-07-12 19:13:30.000000000 +0900
+++ src/libredwg/src/Makefile.am	2014-07-14 21:30:26.508949400 +0900
@@ -29,7 +29,7 @@ libredwg_la_SOURCES = \
         print.c \
 	dwg_api.c
 libredwg_la_LDFLAGS = \
-	-version-info 0:0:0
+	-no-undefined -version-info 0:0:0
 
 include_HEADERS = dwg.h \
 		dwg_api.h
