--- origsrc/libredwg/bindings/python/Makefile.am	2024-09-02 15:28:59.000000000 +0900
+++ src/libredwg/bindings/python/Makefile.am	2024-09-02 15:35:22.039694900 +0900
@@ -46,7 +46,7 @@ _LibreDWG_la_SOURCES = $(BUILT_SOURCES)
 ## more magic: SWIG_PYTHON_CPPFLAGS resolves to null and python includes
 ## are not passed to gcc via -I
 _LibreDWG_la_CPPFLAGS = -I$(top_srcdir)/include $(AX_SWIG_PYTHON_CPPFLAGS)
-_LibreDWG_la_LDFLAGS = -module -no-undefined $(PYTHON_EXTRA_LDFLAGS)
+_LibreDWG_la_LDFLAGS = -module -no-undefined $(PYTHON_EXTRA_LDFLAGS) -avoid-version
 _LibreDWG_la_LIBADD = ../../src/libredwg.la $(PYTHON_LIBS) $(PYTHON_EXTRA_LIBS)
 
 ## not elegant, but works
--- origsrc/libredwg/configure.ac	2024-09-02 15:28:59.000000000 +0900
+++ src/libredwg/configure.ac	2024-09-02 16:43:20.855154200 +0900
@@ -214,9 +214,6 @@ darwin*|macos*)
     AC_DEFINE([_POSIX_C_SOURCE],[200809L],[Needed for strdup])
     ;;
 esac
-case "$host_os" in
-cygwin*) AC_DEFINE([__XSI_VISIBLE],[700],[Needed for cygwin strdup]) ;;
-esac
 
 AX_IS_RELEASE([git-directory])
 dnl ability to override with a .git dir, to test release features
--- origsrc/libredwg/src/common.h	2024-09-02 15:28:59.000000000 +0900
+++ src/libredwg/src/common.h	2024-09-02 16:13:53.711783200 +0900
@@ -346,7 +346,9 @@ EXPORT int strcasecmp (const char *a, co
 #if _GNUC_VERSION >= 303 && !defined(__cplusplus)
 #  undef __nonnull
 #  define __nonnull(params) __attribute__ ((__nonnull__ params))
+#if !defined(__nonnull_all)
 #  define __nonnull_all __attribute__ ((__nonnull__))
+#endif
 #  define HAVE_NONNULL
 #else
 #  ifndef __nonnull
