--- origsrc/magic_enum-0.9.7/CMakeLists.txt	2024-11-14 08:15:10.000000000 +0900
+++ src/magic_enum-0.9.7/CMakeLists.txt	2025-05-14 19:20:24.624813400 +0900
@@ -122,7 +122,7 @@ if(MAGIC_ENUM_OPT_INSTALL)
     EXPORT "${PROJECT_NAME}"
     FILE "${CMAKE_CONFIG_FILE_BASENAME}"
     NAMESPACE "${EXPORT_NAMESPACE}"
-    DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/cmake/${PROJECT_NAME}"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
   )
 
   write_basic_package_version_file(
@@ -133,7 +133,7 @@ if(MAGIC_ENUM_OPT_INSTALL)
     COMPATIBILITY AnyNewerVersion ARCH_INDEPENDENT
   )
   install(FILES "${CMAKE_CONFIG_VERSION_FILE_NAME}"
-          DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/cmake/${PROJECT_NAME}"
+          DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
   )
 
   configure_pkg_config_file(
--- origsrc/magic_enum-0.9.7/test/3rdparty/Catch2/include/catch2/catch.hpp	2024-11-14 08:15:10.000000000 +0900
+++ src/magic_enum-0.9.7/test/3rdparty/Catch2/include/catch2/catch.hpp	2025-05-14 17:09:24.509929500 +0900
@@ -191,7 +191,7 @@ namespace Catch {
 
 ////////////////////////////////////////////////////////////////////////////////
 // We know some environments not to support full POSIX signals
-#if defined(__CYGWIN__) || defined(__QNX__) || defined(__EMSCRIPTEN__) || defined(__DJGPP__)
+#if defined(__QNX__) || defined(__EMSCRIPTEN__) || defined(__DJGPP__)
     #define CATCH_INTERNAL_CONFIG_NO_POSIX_SIGNALS
 #endif
 
