--- origsrc/libraqm-0.5.0/src/Makefile.am	2018-02-24 21:06:47.000000000 +0900
+++ src/libraqm-0.5.0/src/Makefile.am	2018-03-06 11:09:10.142000000 +0900
@@ -27,11 +27,13 @@ libraqm_la_CPPFLAGS = \
     $(NULL)
 
 libraqm_la_LDFLAGS = \
+    -no-undefined \
     -version-info $(RAQM_LIBTOOL_VERSION) \
     $(NULL)
 
 libraqm_test_la_SOURCES = $(libraqm_la_SOURCES)
 libraqm_test_la_LIBADD = $(libraqm_la_LIBADD)
 libraqm_test_la_CPPFLAGS = $(libraqm_la_CPPFLAGS) -DRAQM_TESTING
+libraqm_test_la_LDFLAGS = -no-undefined
 
 -include $(top_srcdir)/git.mk
