--- origsrc/libraqm-0.3.0/src/Makefile.am	2017-08-22 02:39:25.000000000 +0900
+++ src/libraqm-0.3.0/src/Makefile.am	2017-10-07 20:09:29.603574900 +0900
@@ -26,8 +26,11 @@ libraqm_la_CPPFLAGS = \
     $(WARN_CFLAGS) \
     $(NULL)
 
+libraqm_la_LDFLAGS = -no-undefined
+
 libraqm_test_la_SOURCES = $(libraqm_la_SOURCES)
 libraqm_test_la_LIBADD = $(libraqm_la_LIBADD)
 libraqm_test_la_CPPFLAGS = $(libraqm_la_CPPFLAGS) -DRAQM_TESTING
+libraqm_test_la_LDFLAGS = -no-undefined
 
 -include $(top_srcdir)/git.mk
