--- origsrc/libodata-0.4.0/CMakeLists.txt	2018-12-05 22:25:54.000000000 +0900
+++ src/libodata-0.4.0/CMakeLists.txt	2019-01-06 08:47:07.860544200 +0900
@@ -17,7 +17,7 @@ find_library(TINYXML2_LIBRARIES NAMES ti
 if(NOT TINYXML2_LIBRARIES)
     message(SEND_ERROR "tinyxml2 library not found")
 endif (NOT TINYXML2_LIBRARIES)
-find_package(Boost REQUIRED COMPONENTS filesystem thread serialization iostreams program_options)
+find_package(Boost REQUIRED COMPONENTS filesystem thread serialization iostreams program_options system)
 find_package(BerkeleyDB REQUIRED)
 
 
@@ -65,6 +65,9 @@ target_link_libraries(${LIB_NAME} PUBLIC
 target_link_libraries(${LIB_NAME} PUBLIC ${TINYXML2_LIBRARIES})
 target_link_libraries(${LIB_NAME} PUBLIC ${Boost_LIBRARIES})
 target_link_libraries(${LIB_NAME} PUBLIC ${BERKELEY_DB_LIBRARIES})
+if(CYGWIN)
+    set_target_properties(${LIB_NAME} PROPERTIES SOVERSION 0)
+endif()
 
 #install
 include(CMakePackageConfigHelpers)
@@ -84,7 +87,7 @@ configure_package_config_file(
     INSTALL_DESTINATION ${CONFIG_INSTALL_DIR})
 install(TARGETS ${LIB_NAME} 
     EXPORT ${TARGETS_EXPORT_NAME}
-    LIBRARY DESTINATION ${LIB_INSTALL_DIR} COMPONENT ${LIB_COMPONENT_NAME}
+    RUNTIME DESTINATION bin COMPONENT ${LIB_COMPONENT_NAME}
     ARCHIVE DESTINATION ${LIB_INSTALL_DIR} COMPONENT ${LIB_COMPONENT_NAME})
 install(
     DIRECTORY include/odata DESTINATION ${INCLUDE_INSTALL_DIR} COMPONENT ${DEVEL_COMPONENT_NAME})
@@ -170,3 +173,10 @@ set(CPACK_DEB_COMPONENT_INSTALL ON)
 set(CPACK_DEBIAN_ENABLE_COMPONENT_DEPENDS ON)
 
 include(CPack)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libodata.pc.in ${CMAKE_BINARY_DIR}/libodata.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libodata.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libodata-0.4.0/libodata.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libodata-0.4.0/libodata.pc.in	2019-01-06 08:47:52.669144300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libodata
+Description: Access to products stored in Copernicus Data Hub
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lodata
+Cflags: -I${includedir}
--- origsrc/libodata-0.4.0/src/TemporaryFile.h	2018-12-05 22:25:54.000000000 +0900
+++ src/libodata-0.4.0/src/TemporaryFile.h	2019-01-06 08:39:28.485924800 +0900
@@ -8,6 +8,10 @@
 #include <unistd.h>
 #include <vector>
 
+#if defined(__CYGWIN__)
+#define off64_t off_t
+#endif
+
 namespace OData {
 
 class TemporaryFile {
