--- origsrc/uriparser-0.9.4/CMakeLists.txt	2020-06-01 00:11:30.000000000 +0900
+++ src/uriparser-0.9.4/CMakeLists.txt	2020-06-01 10:25:50.865342300 +0900
@@ -361,7 +361,7 @@ configure_package_config_file(
         cmake/uriparser-config.cmake.in
         cmake/uriparser-config.cmake
     INSTALL_DESTINATION
-        ${CMAKE_INSTALL_LIBDIR}/cmake/uriparser-${PROJECT_VERSION}/
+        ${CMAKE_INSTALL_LIBDIR}/cmake/uriparser/
 )
 write_basic_package_version_file(
     cmake/uriparser-config-version.cmake
@@ -378,13 +378,13 @@ uriparser_install(
         ${CMAKE_CURRENT_BINARY_DIR}/cmake/uriparser-config.cmake
         ${CMAKE_CURRENT_BINARY_DIR}/cmake/uriparser-config-version.cmake
     DESTINATION
-        ${CMAKE_INSTALL_LIBDIR}/cmake/uriparser-${PROJECT_VERSION}/
+        ${CMAKE_INSTALL_LIBDIR}/cmake/uriparser/
 )
 uriparser_install(
     EXPORT
         uriparser
     DESTINATION
-        ${CMAKE_INSTALL_LIBDIR}/cmake/uriparser-${PROJECT_VERSION}/
+        ${CMAKE_INSTALL_LIBDIR}/cmake/uriparser/
     NAMESPACE
         uriparser::
 )
