--- origsrc/uriparser-0.9.9/CMakeLists.txt	2025-09-04 06:11:01.000000000 +0900
+++ src/uriparser-0.9.9/CMakeLists.txt	2025-09-04 08:41:10.532393800 +0900
@@ -520,7 +520,7 @@ configure_package_config_file(
         ${CMAKE_CURRENT_SOURCE_DIR}/cmake/uriparser-config.cmake.in
         cmake/uriparser-config.cmake
     INSTALL_DESTINATION
-        ${CMAKE_INSTALL_LIBDIR}/cmake/uriparser-${PROJECT_VERSION}/
+        ${CMAKE_INSTALL_LIBDIR}/cmake/uriparser/
 )
 write_basic_package_version_file(
     cmake/uriparser-config-version.cmake
@@ -537,13 +537,13 @@ uriparser_install(
         ${CMAKE_CURRENT_BINARY_DIR}/cmake/uriparser-config.cmake
         ${CMAKE_CURRENT_BINARY_DIR}/cmake/uriparser-config-version.cmake
     DESTINATION
-        ${CMAKE_INSTALL_LIBDIR}/cmake/uriparser-${PROJECT_VERSION}/
+        ${CMAKE_INSTALL_LIBDIR}/cmake/uriparser/
 )
 uriparser_install(
     EXPORT
         uriparser
     DESTINATION
-        ${CMAKE_INSTALL_LIBDIR}/cmake/uriparser-${PROJECT_VERSION}/
+        ${CMAKE_INSTALL_LIBDIR}/cmake/uriparser/
     NAMESPACE
         uriparser::
 )
