--- origsrc/vortex-1.1.19.b2413/configure.ac	2022-08-31 20:32:51.000000000 +0900
+++ src/vortex-1.1.19.b2413/configure.ac	2024-09-24 16:24:20.245145600 +0900
@@ -58,7 +58,7 @@ AM_PROG_LIBTOOL
 compiler_options=""
 STRICT_PROTOTYPES=""
 if test "$compiler" = "gcc" ; then
-   compiler_options="-Wstrict-prototypes -Wall -Werror -g -fstack-protector-all -Wstack-protector"
+   compiler_options="-Wstrict-prototypes -Wall -g -fstack-protector-all -Wstack-protector"
    echo "Detected gcc compiler: $compiler, adding options: $compiler_options"
 fi
 AC_SUBST(compiler_options)
@@ -475,7 +475,7 @@ if test "$enable_py_vortex_support" = "y
       python_dir_temp=`eval echo ${pythondir}`
       # check python_install_dir and pythondir maches
       if test "$python_dir_check" != "ok" ; then
-          AC_MSG_ERROR([ERROR: pythondir variable is poiting to an unusable location: $python_dir_temp. Error detected: $python_dir_check. Fix your autoconf install or disable python building, for that, pass --disable-py-vortex to configure. This usually indicates a problem with automake package which provides the _AM_PATH_PYTHON_ macro.])
+          AC_MSG_ERROR([ERROR: pythondir variable is poiting to an unusable location: $python_dir_temp. Error detected: $python_dir_check. Fix your autoconf install or disable python building, for that, pass --disable-py-vortex to configure. This usually indicates a problem with automake package which provides the macro.])
       fi
       echo "Python install dir OK: $python_dir_temp"
    fi
--- origsrc/vortex-1.1.19.b2413/src/vortex.c	2022-08-31 20:32:54.000000000 +0900
+++ src/vortex-1.1.19.b2413/src/vortex.c	2024-09-24 16:23:32.737522000 +0900
@@ -51,7 +51,7 @@
  * with log support */
 #if defined(ENABLE_VORTEX_LOG)
 /* but avoid defining it on macosx because it seems to break the compiler */
-#if !defined(__APPLE__)
+#if !defined(__APPLE__) && !defined(__CYGWIN__)
 int vsnprintf(char *str, size_t size, const char *format, va_list ap);
 #endif
 #endif
