--- origsrc/draco-1.3.4/CMakeLists.txt	2018-08-18 04:01:27.000000000 +0900
+++ src/draco-1.3.4/CMakeLists.txt	2019-01-15 22:18:40.400265700 +0900
@@ -1124,7 +1124,8 @@ else ()
 
   # Add install rules for lib and executable targets.
   install(TARGETS dracodec dracoenc draco
-          DESTINATION "${CMAKE_INSTALL_PREFIX}/lib")
+          RUNTIME DESTINATION bin
+          ARCHIVE DESTINATION "${CMAKE_INSTALL_PREFIX}/lib")
   if (BUILD_UNITY_PLUGIN)
     install(TARGETS dracodec_unity
             DESTINATION "${CMAKE_INSTALL_PREFIX}/lib")
@@ -1141,7 +1142,7 @@ else ()
   set(draco_include_install_dir "include/draco")
   set(draco_lib_install_dir "lib")
   set(draco_script_install_dir
-      "${CMAKE_INSTALL_PREFIX}/${draco_lib_install_dir}/draco/cmake")
+      "${CMAKE_INSTALL_PREFIX}/${draco_lib_install_dir}/cmake/draco")
   set(draco_config_in_file "${draco_root}/cmake/DracoConfig.cmake")
   set(draco_config_out_file "${draco_build_dir}/DracoConfig.cmake")
 
