--- origsrc/libbitcoin-3.5.0/Makefile.am	2018-01-20 15:55:05.000000000 +0900
+++ src/libbitcoin-3.5.0/Makefile.am	2018-04-05 08:44:47.232000200 +0900
@@ -33,7 +33,7 @@ doc_DATA = \
 #------------------------------------------------------------------------------
 lib_LTLIBRARIES = src/libbitcoin.la
 src_libbitcoin_la_CPPFLAGS = -I${srcdir}/include ${icu} ${png} ${qrencode} ${boost_CPPFLAGS} ${pthread_CPPFLAGS} ${icu_i18n_CPPFLAGS} ${png_CPPFLAGS} ${qrencode_CPPFLAGS} ${secp256k1_CPPFLAGS}
-src_libbitcoin_la_LDFLAGS = ${boost_LDFLAGS}
+src_libbitcoin_la_LDFLAGS = -no-undefined ${boost_LDFLAGS}
 src_libbitcoin_la_LIBADD = ${boost_chrono_LIBS} ${boost_date_time_LIBS} ${boost_filesystem_LIBS} ${boost_iostreams_LIBS} ${boost_locale_LIBS} ${boost_log_LIBS} ${boost_program_options_LIBS} ${boost_regex_LIBS} ${boost_system_LIBS} ${boost_thread_LIBS} ${pthread_LIBS} ${rt_LIBS} ${icu_i18n_LIBS} ${dl_LIBS} ${png_LIBS} ${qrencode_LIBS} ${secp256k1_LIBS}
 src_libbitcoin_la_SOURCES = \
     src/error.cpp \
--- origsrc/libbitcoin-3.5.0/configure.ac	2018-01-20 15:55:05.000000000 +0900
+++ src/libbitcoin-3.5.0/configure.ac	2018-04-05 11:03:11.266000000 +0900
@@ -339,17 +339,19 @@ AS_CASE([${CC}], [*],
     [AX_CHECK_LINK_FLAG([-fstack-protector-all],
         [LDFLAGS="$LDFLAGS -fstack-protector-all"])])
 
-# Hide internal functions from external libs. Enabled in gcc only.
-#------------------------------------------------------------------------------
-AS_CASE([${CC}], [*gcc*],
-    [AX_CHECK_COMPILE_FLAG([-fvisibility-hidden],
-        [CXXFLAGS="$CXXFLAGS -fvisibility-hidden"])])
+if test "$host_os" != "cygwin"; then
+    # Hide internal functions from external libs. Enabled in gcc only.
+    #------------------------------------------------------------------------------
+    AS_CASE([${CC}], [*gcc*],
+        [AX_CHECK_COMPILE_FLAG([-fvisibility-hidden],
+            [CXXFLAGS="$CXXFLAGS -fvisibility-hidden"])])
 
-# Hide inlines from external libs. Enabled in gcc only.
-#------------------------------------------------------------------------------
-AS_CASE([${CC}], [*gcc*],
-    [AX_CHECK_COMPILE_FLAG([-fvisibility-inlines-hidden],
-        [CXXFLAGS="$CXXFLAGS -fvisibility-inlines-hidden"])])
+    # Hide inlines from external libs. Enabled in gcc only.
+    #------------------------------------------------------------------------------
+    AS_CASE([${CC}], [*gcc*],
+        [AX_CHECK_COMPILE_FLAG([-fvisibility-inlines-hidden],
+            [CXXFLAGS="$CXXFLAGS -fvisibility-inlines-hidden"])])
+fi
 
 # Limit delays and warnings. Enabled in gcc only.
 #------------------------------------------------------------------------------
@@ -357,7 +359,6 @@ AS_CASE([${CC}], [*gcc*],
     [AX_CHECK_COMPILE_FLAG([-fno-var-tracking-assignments],
         [CXXFLAGS="$CXXFLAGS -fno-var-tracking-assignments"])])
 
-
 # Process outputs into templates.
 #==============================================================================
 AC_CONFIG_FILES([Makefile libbitcoin.pc])
