--- origsrc/libbitcoin-2.11.0/Makefile.am	2016-01-24 08:25:45.000000000 +0900
+++ src/libbitcoin-2.11.0/Makefile.am	2016-04-03 12:39:13.619308600 +0900
@@ -33,7 +33,7 @@ doc_DATA = \
 #------------------------------------------------------------------------------
 lib_LTLIBRARIES = src/libbitcoin.la
 src_libbitcoin_la_CPPFLAGS = -I${srcdir}/include ${icu} ${testnet} ${boost_CPPFLAGS} ${pthread_CPPFLAGS} ${icu_i18n_CPPFLAGS} ${secp256k1_CPPFLAGS}
-src_libbitcoin_la_LDFLAGS = ${boost_LDFLAGS}
+src_libbitcoin_la_LDFLAGS = -no-undefined ${boost_LDFLAGS}
 src_libbitcoin_la_LIBADD = ${boost_chrono_LIBS} ${boost_date_time_LIBS} ${boost_filesystem_LIBS} ${boost_locale_LIBS} ${boost_program_options_LIBS} ${boost_regex_LIBS} ${boost_system_LIBS} ${boost_thread_LIBS} ${pthread_LIBS} ${rt_LIBS} ${icu_i18n_LIBS} ${dl_LIBS} ${secp256k1_LIBS}
 src_libbitcoin_la_SOURCES = \
     src/block.cpp \
@@ -334,4 +334,3 @@ target_examples = \
     examples/libbitcoin_examples
 
 examples: ${target_examples}
-
