diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/yafc-1.3.6/lib/fnmatch.c src/yafc-1.3.6/lib/fnmatch.c
--- origsrc/yafc-1.3.6/lib/fnmatch.c	2015-03-14 22:56:48.000000000 +0900
+++ src/yafc-1.3.6/lib/fnmatch.c	2015-10-04 22:28:07.192840300 +0900
@@ -176,6 +176,9 @@ gmatch (string, se, pattern, pe, flags)
       sc = n < se ? *n : '\0';
 
 #ifdef EXTENDED_GLOB
+#if !defined(FNM_EXTMATCH)
+#define FNM_EXTMATCH (0x20)
+#endif
       /* extmatch () will handle recursively calling gmatch, so we can
 	 just return what extmatch() returns. */
       if ((flags & FNM_EXTMATCH) && *p == '(' &&
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/yafc-1.3.6/src/main.c src/yafc-1.3.6/src/main.c
--- origsrc/yafc-1.3.6/src/main.c	2015-05-12 06:55:15.000000000 +0900
+++ src/yafc-1.3.6/src/main.c	2015-10-04 22:25:43.272437100 +0900
@@ -29,7 +29,7 @@
 # include <locale.h>
 #endif
 
-#include "yafcrc.h"
+#include "src/yafcrc.h"
 
 void print_syntax_and_exit(const char* argv0)
 {
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/yafc-1.3.6/src/syshdr.h src/yafc-1.3.6/src/syshdr.h
--- origsrc/yafc-1.3.6/src/syshdr.h	2015-05-12 06:47:30.000000000 +0900
+++ src/yafc-1.3.6/src/syshdr.h	2015-10-04 22:17:36.889483700 +0900
@@ -32,7 +32,7 @@
 #endif
 
 #ifdef HAVE_CONFIG_H
-# include "../config.h"
+# include "config.h"
 #endif
 
 #ifdef HAVE_SYS_TYPES_H
@@ -167,10 +167,10 @@ typedef CPPFunction rl_completion_func_t
 #ifdef HAVE_FNMATCH_GNU
 # include <fnmatch.h>
 #else
-# include "../lib/fnmatch.h"  /* our own, in lib/ */
+# include "lib/fnmatch.h"  /* our own, in lib/ */
 #endif
 
-#if defined(HAVE_GETOPT_LONG) && defined(HAVE_GETOPT_H)
+#if defined(__CYGWIN__) || (defined(HAVE_GETOPT_LONG) && defined(HAVE_GETOPT_H))
 # include <getopt.h>
 #else
 # include "../lib/getopt.h"  /* our own, in lib/ */
