--- origsrc/editorconfig-core-c/src/CMakeLists.txt	2026-01-07 07:30:49.000000000 +0900
+++ src/editorconfig-core-c/src/CMakeLists.txt	2026-01-07 07:35:29.508081100 +0900
@@ -73,7 +73,7 @@ endif()
 if(MSVC)
   add_compile_options(/W4 /WX)
 else()
-  add_compile_options(-Wall -Wextra -Wpedantic -Werror)
+  add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
 
 add_subdirectory(lib)
--- origsrc/editorconfig-core-c/src/lib/CMakeLists.txt	2026-01-07 07:30:49.000000000 +0900
+++ src/editorconfig-core-c/src/lib/CMakeLists.txt	2026-01-07 07:30:57.568440100 +0900
@@ -52,6 +52,7 @@ if (BUILD_STATICALLY_LINKED_EXE)
         EXCLUDE_FROM_ALL 1)
 endif()
 
+if (NOT BUILD_SHARED_LIBS)
 add_library(editorconfig_static STATIC ${editorconfig_LIBSRCS})
 target_include_directories(editorconfig_static
     INTERFACE $<INSTALL_INTERFACE:include>
@@ -65,6 +66,7 @@ if(WIN32)
     target_link_libraries(editorconfig_static shlwapi)
 endif()
 target_link_libraries(editorconfig_static ${PCRE2_LIBRARIES})
+endif()
 
 # EditorConfig package name for find_package() and the CMake package registry.
 # On UNIX the system registry is usually just "lib/cmake/<package>".
@@ -76,11 +78,13 @@ set(editorconfig_CONFIG_EXPORT_NAME "${c
 set(editorconfig_CONFIG_INSTALL_LIBDIR
     "${CMAKE_INSTALL_LIBDIR}/cmake/${config_package_name}")
 
+if (NOT BUILD_SHARED_LIBS)
 install(TARGETS editorconfig_static
     EXPORT ${editorconfig_CONFIG_EXPORT_NAME}
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+endif()
 
 if(NOT BUILD_STATICALLY_LINKED_EXE)
     install(TARGETS editorconfig_shared
