--- origsrc/libbf-1.0.0/CMakeLists.txt	2018-04-21 05:36:05.000000000 +0900
+++ src/libbf-1.0.0/CMakeLists.txt	2019-05-10 07:43:10.717197400 +0900
@@ -82,8 +82,18 @@ set(libbf_sources
 )
 
 add_library(libbf SHARED ${libbf_sources})
+if (CYGWIN)
+  set_target_properties(libbf PROPERTIES SOVERSION 0)
+endif ()
 set_target_properties(libbf PROPERTIES OUTPUT_NAME bf)
-install(TARGETS libbf DESTINATION lib)
+install(TARGETS libbf RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/bf DESTINATION include)
 
 add_subdirectory(test)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libbf.pc.in ${CMAKE_BINARY_DIR}/libbf.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libbf.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libbf-1.0.0/libbf.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libbf-1.0.0/libbf.pc.in	2019-05-10 07:43:10.451565900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libbf
+Description: Bloom filters for C++11
+Version: @VERSION@
+Libs: -L${libdir} -lbf
+Cflags: -I${includedir}
