--- origsrc/libgraphqlparser-0.7.0/CMakeLists.txt	2017-10-17 06:39:41.000000000 +0900
+++ src/libgraphqlparser-0.7.0/CMakeLists.txt	2017-10-21 13:46:28.038928700 +0900
@@ -51,6 +51,7 @@ ADD_LIBRARY(graphqlparser SHARED
   lexer.cpp
   lexer.h
   GraphQLParser.cpp)
+set_target_properties(graphqlparser PROPERTIES SOVERSION 0)
 
 # Enable this and remove CMAKE_CXX_FLAGS fiddle above when we are able
 # to upgrade to CMake 2.8.12. Blocker seems to be Travis CI being on
@@ -105,7 +106,8 @@ INSTALL(FILES
   syntaxdefs.h
   DESTINATION include/graphqlparser)
 INSTALL(TARGETS graphqlparser
-  LIBRARY DESTINATION lib)
+  RUNTIME DESTINATION bin
+  ARCHIVE DESTINATION lib)
 
 if (UNIX)
   # generate pkgconfig file
