--- origsrc/libgraphqlparser-0.6.0/CMakeLists.txt	2016-12-23 05:39:29.000000000 +0900
+++ src/libgraphqlparser-0.6.0/CMakeLists.txt	2017-01-11 00:43:56.023098600 +0900
@@ -49,6 +49,7 @@ ADD_LIBRARY(graphqlparser SHARED
   lexer.cpp
   lexer.h
   GraphQLParser.cpp)
+set_target_properties(graphqlparser PROPERTIES SOVERSION 0)
 
 # Enable this and remove CMAKE_CXX_FLAGS fiddle above when we are able
 # to upgrade to CMake 2.8.12. Blocker seems to be Travis CI being on
@@ -99,7 +100,8 @@ INSTALL(FILES
   syntaxdefs.h
   DESTINATION include/graphqlparser)
 INSTALL(TARGETS graphqlparser
-  LIBRARY DESTINATION lib)
+  RUNTIME DESTINATION bin
+  ARCHIVE DESTINATION lib)
 
 if (UNIX)
   # generate pkgconfig file
