--- origsrc/libunistring-0.9.3/configure.ac	2010-03-29 11:35:26.000000000 +0900
+++ src/libunistring-0.9.3/configure.ac	2010-12-28 00:00:00.390270900 +0900
@@ -61,7 +61,7 @@ LT_LANG([Windows Resource])
 dnl On mingw and Cygwin, we can activate special Makefile rules which add
 dnl version information to the shared libraries and executables.
 case "$host_os" in
-  mingw* | cygwin*) is_woe32=yes ;;
+  mingw*) is_woe32=yes ;;
   *) is_woe32=no ;;
 esac
 AM_CONDITIONAL([WOE32], [test $is_woe32 = yes])
@@ -79,7 +79,7 @@ dnl    must arrange to define the auxili
 dnl    exported variables _also_ in the static library.
 if test "$enable_shared" = yes; then
   case "$host_os" in
-    mingw* | cygwin*) is_woe32dll=yes ;;
+    mingw*) is_woe32dll=yes ;;
     *) is_woe32dll=no ;;
   esac
 else
