--- origsrc/libsrsirc-0.0.14.1/configure.ac	2024-04-28 07:47:18.000000000 +0900
+++ src/libsrsirc-0.0.14.1/configure.ac	2024-12-07 18:44:26.896737300 +0900
@@ -20,7 +20,10 @@ AC_PROG_MAKE_SET
 
 AC_HEADER_STDC
 
-AC_CHECK_HEADERS([arpa/inet.h fcntl.h limits.h netdb.h netinet/in.h stdbool.h stddef.h stdlib.h string.h strings.h sys/select.h sys/socket.h sys/time.h sys/types.h syslog.h unistd.h windows.h winsock2.h])
+AC_CHECK_HEADERS([arpa/inet.h fcntl.h limits.h netdb.h netinet/in.h stdbool.h stddef.h stdlib.h string.h strings.h sys/select.h sys/socket.h sys/time.h sys/types.h syslog.h unistd.h])
+if test "$host_os" != "cygwin"; then
+  AC_CHECK_HEADERS([windows.h winsock2.h])
+fi
 AC_ARG_WITH(ssl,
 [  --with-ssl            Build with SSL support],
 	if test x$withval = xno; then
@@ -88,7 +91,7 @@ AC_TYPE_UINT64_T
 AC_FUNC_MALLOC
 AC_FUNC_REALLOC
 AC_FUNC_STRERROR_R
-AC_CHECK_FUNCS([atexit close connect fcntl fileno getaddrinfo getopt getsockopt gettimeofday htons inet_addr inet_pton memmove memset nanosleep read select send setsockopt sigaction socket strcasecmp strchr strncasecmp strspn strstr strtol strtoul strtoull])
+AC_CHECK_FUNCS([atexit close connect fcntl fileno getaddrinfo getopt getsockopt gettimeofday htons inet_addr inet_pton iprintf memmove memset nanosleep read select send setsockopt sigaction socket strcasecmp strchr strncasecmp strspn strstr strtol strtoul strtoull])
 
 
 AX_HAVE_CTIME_R(
@@ -191,4 +194,4 @@ echo \
  Libraries: '${LIBS}'
 
 --------------------------------------------------"
-scripts/gentests.sh
+${srcdir}/scripts/gentests.sh
--- origsrc/libsrsirc-0.0.14.1/libsrsirc/Makefile.am	2024-04-28 07:47:18.000000000 +0900
+++ src/libsrsirc-0.0.14.1/libsrsirc/Makefile.am	2024-12-07 16:03:53.277377500 +0900
@@ -1,5 +1,5 @@
 lib_LTLIBRARIES = libsrsirc.la
 libsrsirc_la_SOURCES = io.c conn.c irc.c util.c px.c msg.c common.c irc_msghnd.c irc_track.c irc_getset.c bucklist.c skmap.c ucbase.c cmap.c common.h conn.h intdefs.h bucklist.h msg.h io.h cmap.h irc_msghnd.h px.h irc_track_int.h skmap.h ucbase.h
-libsrsirc_la_CPPFLAGS = -I$(top_srcdir)/include
-libsrsirc_la_LIBADD = $(top_srcdir)/platform/libsrsircbase.la $(top_srcdir)/logger/libsrsirclog.la
+libsrsirc_la_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)
+libsrsirc_la_LIBADD = $(top_builddir)/platform/libsrsircbase.la $(top_builddir)/logger/libsrsirclog.la
 libsrsirc_la_LDFLAGS = -no-undefined
--- origsrc/libsrsirc-0.0.14.1/logger/Makefile.am	2024-04-28 07:47:18.000000000 +0900
+++ src/libsrsirc-0.0.14.1/logger/Makefile.am	2024-12-07 16:04:07.916256700 +0900
@@ -1,2 +1,3 @@
 noinst_LTLIBRARIES = libsrsirclog.la
+libsrsirclog_la_CPPFLAGS = -I$(top_srcdir)
 libsrsirclog_la_SOURCES = intlog.c intlog.h
--- origsrc/libsrsirc-0.0.14.1/platform/Makefile.am	2024-04-28 07:47:18.000000000 +0900
+++ src/libsrsirc-0.0.14.1/platform/Makefile.am	2024-12-07 16:04:20.917906500 +0900
@@ -1,2 +1,3 @@
 noinst_LTLIBRARIES = libsrsircbase.la
+libsrsircbase_la_CPPFLAGS = -I$(top_srcdir)
 libsrsircbase_la_SOURCES = base_io.c base_net.c base_string.c base_misc.c base_time.c base_log.c base_log.h base_io.h base_misc.h base_net.h base_string.h base_time.h
--- origsrc/libsrsirc-0.0.14.1/src/Makefile.am	2024-04-28 07:47:18.000000000 +0900
+++ src/libsrsirc-0.0.14.1/src/Makefile.am	2024-12-07 18:38:44.245186500 +0900
@@ -2,13 +2,13 @@ bin_PROGRAMS = icat iwat
 noinst_PROGRAMS = helloworld
 
 icat_SOURCES = icat_core.c icat_init.c icat_misc.c icat_serv.c icat_user.c icat_common.h icat_core.h icat_misc.h icat_serv.h icat_user.h
-icat_CPPFLAGS = -I$(top_srcdir)/include
-icat_LDADD = $(top_srcdir)/logger/libsrsirclog.la $(top_srcdir)/libsrsirc/libsrsirc.la
+icat_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)
+icat_LDADD = $(top_builddir)/logger/libsrsirclog.la $(top_builddir)/libsrsirc/libsrsirc.la
 
 iwat_SOURCES = iwat.c
-iwat_CPPFLAGS = -I$(top_srcdir)/include
-iwat_LDADD = $(top_srcdir)/libsrsirc/libsrsirc.la
+iwat_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)
+iwat_LDADD = $(top_builddir)/libsrsirc/libsrsirc.la
 
 helloworld_SOURCES = helloworld.c
-helloworld_CPPFLAGS = -I$(top_srcdir)/include
-helloworld_LDADD = $(top_srcdir)/libsrsirc/libsrsirc.la
+helloworld_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)
+helloworld_LDADD = $(top_builddir)/libsrsirc/libsrsirc.la
--- origsrc/libsrsirc-0.0.14.1/src/iwat.c	2024-04-28 07:47:18.000000000 +0900
+++ src/libsrsirc-0.0.14.1/src/iwat.c	2024-12-07 18:46:57.792781600 +0900
@@ -67,7 +67,7 @@ static void usage(FILE *str, const char
 static void cleanup(void);
 int main(int argc, char **argv);
 
-
+#if !defined(HAVE_IPRINTF)
 static int
 iprintf(const char *fmt, ...)
 {
@@ -78,6 +78,7 @@ iprintf(const char *fmt, ...)
 	va_end(l);
 	return !irc_write(g_irc, buf) ? -1 : r;
 }
+#endif
 
 static bool
 tryconnect(void)
--- origsrc/libsrsirc-0.0.14.1/unittests/Makefile.am	2024-04-28 07:47:18.000000000 +0900
+++ src/libsrsirc-0.0.14.1/unittests/Makefile.am	2024-12-07 21:57:02.019644700 +0900
@@ -1,4 +1,4 @@
 noinst_PROGRAMS = test_bucklist
-test_bucklist_SOURCES = run_test_bucklist.c unittests_common.h
-test_bucklist_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/libsrsirc
-test_bucklist_LDADD = $(top_srcdir)/libsrsirc/libsrsirc.la
+test_bucklist_SOURCES = test_bucklist.c unittests_common.h
+test_bucklist_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/libsrsirc -I$(top_srcdir)
+test_bucklist_LDADD = $(top_builddir)/libsrsirc/libsrsirc.la
--- origsrc/libsrsirc-0.0.14.1/unittests/test_bucklist.c	2024-04-28 07:47:18.000000000 +0900
+++ src/libsrsirc-0.0.14.1/unittests/test_bucklist.c	2024-12-07 21:59:50.263411000 +0900
@@ -52,3 +52,5 @@ test_basic(void)
 
 	return NULL;
 }
+
+int main() { test_basic(); }
