--- origsrc/libqofono-0.124/plugin/plugin.pro	2024-12-30 22:15:12.000000000 +0900
+++ src/libqofono-0.124/plugin/plugin.pro	2025-08-13 09:05:20.386666400 +0900
@@ -3,6 +3,8 @@ TEMPLATE = lib
 QT += dbus
 CONFIG += plugin
 
+DEFINES += QOFONO_LIBRARY
+
 QT_VERSION=$$[QT_VERSION]
 
 QT += qml
--- origsrc/libqofono-0.124/src/src.pro	2025-08-13 08:49:23.566144400 +0900
+++ src/libqofono-0.124/src/src.pro	2025-08-13 09:11:05.733271800 +0900
@@ -320,5 +320,6 @@ CONFIG += hide_symbols create_pc create_
 
 QMAKE_PKGCONFIG_DESTDIR = pkgconfig
 QMAKE_PKGCONFIG_INCDIR = $$headers.path
+QMAKE_PKGCONFIG_LIBDIR = $$target.path
 
 INSTALLS += target headers qtconfig xmlfiles pkgconfig
--- origsrc/libqofono-0.124/src/version.pri	2024-12-30 22:15:12.000000000 +0900
+++ src/libqofono-0.124/src/version.pri	2025-08-13 09:32:59.488608500 +0900
@@ -1 +1 @@
-VERSION = 0.115
+VERSION = 0.124
