--- origsrc/libvhdi-20200810/Makefile.am	2020-08-11 00:33:06.000000000 +0900
+++ src/libvhdi-20200810/Makefile.am	2020-09-06 10:44:07.897197300 +0900
@@ -3,25 +3,11 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfguid \
 	libvhdi \
 	vhditools \
 	pyvhdi \
 	pyvhdi-python2 \
 	pyvhdi-python3 \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libvhdi-20200810/configure.ac	2020-08-11 00:33:06.000000000 +0900
+++ src/libvhdi-20200810/configure.ac	2020-09-06 10:44:09.114223500 +0900
@@ -170,19 +170,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
 AC_CONFIG_FILES([libvhdi/Makefile])
 AC_CONFIG_FILES([pyvhdi/Makefile])
 AC_CONFIG_FILES([pyvhdi-python2/Makefile])
--- origsrc/libvhdi-20200810/libvhdi/Makefile.am	2020-08-11 00:33:06.000000000 +0900
+++ src/libvhdi-20200810/libvhdi/Makefile.am	2020-09-06 10:44:09.133217800 +0900
@@ -1,5 +1,6 @@
 AM_CPPFLAGS = \
 	-DLOCALEDIR=\"$(datadir)/locale\" \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libvhdi-20200810/pyvhdi/Makefile.am	2020-08-11 00:33:06.000000000 +0900
+++ src/libvhdi-20200810/pyvhdi/Makefile.am	2020-09-06 10:44:09.365296800 +0900
@@ -1,5 +1,6 @@
 if HAVE_PYTHON
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -44,7 +45,7 @@ pyvhdi_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pyvhdi_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyvhdi_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyvhdi_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
--- origsrc/libvhdi-20200810/pyvhdi-python2/Makefile.am	2020-08-11 00:33:06.000000000 +0900
+++ src/libvhdi-20200810/pyvhdi-python2/Makefile.am	2020-09-06 10:44:09.852568300 +0900
@@ -6,6 +6,7 @@ pyexecdir = $(pyexecdir2)
 
 if HAVE_PYTHON2
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -54,7 +55,7 @@ pyvhdi_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pyvhdi_la_CPPFLAGS = $(PYTHON2_CPPFLAGS)
-pyvhdi_la_LDFLAGS  = -module -avoid-version $(PYTHON2_LDFLAGS)
+pyvhdi_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON2_LDFLAGS)
 
 $(BUILT_SOURCES):
 	/bin/cp -f $(top_srcdir)/pyvhdi/$@ $@
--- origsrc/libvhdi-20200810/pyvhdi-python3/Makefile.am	2020-08-11 00:33:06.000000000 +0900
+++ src/libvhdi-20200810/pyvhdi-python3/Makefile.am	2020-09-06 10:44:10.242340800 +0900
@@ -6,6 +6,7 @@ pyexecdir = $(pyexecdir3)
 
 if HAVE_PYTHON3
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -54,7 +55,7 @@ pyvhdi_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pyvhdi_la_CPPFLAGS = $(PYTHON3_CPPFLAGS)
-pyvhdi_la_LDFLAGS  = -module -avoid-version $(PYTHON3_LDFLAGS)
+pyvhdi_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON3_LDFLAGS)
 
 $(BUILT_SOURCES):
 	/bin/cp -f $(top_srcdir)/pyvhdi/$@ $@
--- origsrc/libvhdi-20200810/tests/Makefile.am	2020-08-11 00:33:06.000000000 +0900
+++ src/libvhdi-20200810/tests/Makefile.am	2020-09-06 10:44:10.588362800 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libvhdi-20200810/vhditools/Makefile.am	2020-08-11 00:33:06.000000000 +0900
+++ src/libvhdi-20200810/vhditools/Makefile.am	2020-09-06 10:44:10.733308300 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
