--- origsrc/libvhdi-20181227/Makefile.am	2018-12-27 23:10:35.000000000 +0900
+++ src/libvhdi-20181227/Makefile.am	2019-05-26 10:56:42.229709800 +0900
@@ -3,25 +3,11 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfguid \
 	libvhdi \
 	vhditools \
 	pyvhdi \
 	pyvhdi-python2 \
 	pyvhdi-python3 \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libvhdi-20181227/configure.ac	2018-12-27 23:10:35.000000000 +0900
+++ src/libvhdi-20181227/configure.ac	2019-05-26 10:56:32.119007800 +0900
@@ -186,19 +186,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
 AC_CONFIG_FILES([libvhdi/Makefile])
 AC_CONFIG_FILES([pyvhdi/Makefile])
 AC_CONFIG_FILES([pyvhdi-python2/Makefile])
--- origsrc/libvhdi-20181227/libvhdi/Makefile.am	2018-12-27 23:10:35.000000000 +0900
+++ src/libvhdi-20181227/libvhdi/Makefile.am	2019-05-26 10:57:00.582663700 +0900
@@ -1,5 +1,6 @@
 AM_CPPFLAGS = \
 	-DLOCALEDIR=\"$(datadir)/locale\" \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libvhdi-20181227/pyvhdi/Makefile.am	2018-12-27 23:10:35.000000000 +0900
+++ src/libvhdi-20181227/pyvhdi/Makefile.am	2019-05-26 10:57:17.365682700 +0900
@@ -1,5 +1,6 @@
 if HAVE_PYTHON
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -44,7 +45,7 @@ pyvhdi_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pyvhdi_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyvhdi_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyvhdi_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
--- origsrc/libvhdi-20181227/pyvhdi-python2/Makefile.am	2018-12-27 23:10:35.000000000 +0900
+++ src/libvhdi-20181227/pyvhdi-python2/Makefile.am	2019-05-26 10:57:31.421212700 +0900
@@ -6,6 +6,7 @@ pyexecdir = $(pyexecdir2)
 
 if HAVE_PYTHON2
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -54,7 +55,7 @@ pyvhdi_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pyvhdi_la_CPPFLAGS = $(PYTHON2_CPPFLAGS)
-pyvhdi_la_LDFLAGS  = -module -avoid-version $(PYTHON2_LDFLAGS)
+pyvhdi_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON2_LDFLAGS)
 
 $(BUILT_SOURCES):
 	/bin/cp -f $(top_srcdir)/pyvhdi/$@ $@
--- origsrc/libvhdi-20181227/pyvhdi-python3/Makefile.am	2018-12-27 23:10:35.000000000 +0900
+++ src/libvhdi-20181227/pyvhdi-python3/Makefile.am	2019-05-26 10:57:45.853320600 +0900
@@ -6,6 +6,7 @@ pyexecdir = $(pyexecdir3)
 
 if HAVE_PYTHON3
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -54,7 +55,7 @@ pyvhdi_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pyvhdi_la_CPPFLAGS = $(PYTHON3_CPPFLAGS)
-pyvhdi_la_LDFLAGS  = -module -avoid-version $(PYTHON3_LDFLAGS)
+pyvhdi_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON3_LDFLAGS)
 
 $(BUILT_SOURCES):
 	/bin/cp -f $(top_srcdir)/pyvhdi/$@ $@
--- origsrc/libvhdi-20181227/tests/Makefile.am	2018-12-27 23:10:35.000000000 +0900
+++ src/libvhdi-20181227/tests/Makefile.am	2019-05-26 10:57:54.296827900 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libvhdi-20181227/vhditools/Makefile.am	2018-12-27 23:10:35.000000000 +0900
+++ src/libvhdi-20181227/vhditools/Makefile.am	2019-05-26 10:58:03.859558500 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
