--- origsrc/libvhdi-20240509/Makefile.am	2024-05-09 14:05:33.000000000 +0900
+++ src/libvhdi-20240509/Makefile.am	2025-05-20 13:50:27.746880400 +0900
@@ -3,23 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfguid \
 	libvhdi \
 	vhditools \
 	pyvhdi \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libvhdi-20240509/configure.ac	2024-05-09 14:05:33.000000000 +0900
+++ src/libvhdi-20240509/configure.ac	2025-05-20 13:50:27.746880400 +0900
@@ -170,19 +170,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
 AC_CONFIG_FILES([libvhdi/Makefile])
 AC_CONFIG_FILES([pyvhdi/Makefile])
 AC_CONFIG_FILES([vhditools/Makefile])
--- origsrc/libvhdi-20240509/pyvhdi/Makefile.am	2024-05-09 14:05:33.000000000 +0900
+++ src/libvhdi-20240509/pyvhdi/Makefile.am	2025-05-20 13:50:45.306819200 +0900
@@ -44,7 +44,7 @@ pyvhdi_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pyvhdi_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyvhdi_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyvhdi_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
