--- origsrc/libvsbsdl-20240227/Makefile.am	2024-02-27 13:37:00.000000000 +0900
+++ src/libvsbsdl-20240227/Makefile.am	2024-03-02 13:26:53.495720300 +0900
@@ -3,22 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
 	libvsbsdl \
 	vsbsdltools \
 	pyvsbsdl \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libvsbsdl-20240227/configure.ac	2024-02-27 13:37:00.000000000 +0900
+++ src/libvsbsdl-20240227/configure.ac	2024-03-02 13:27:07.399376800 +0900
@@ -157,18 +157,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
 AC_CONFIG_FILES([libvsbsdl/Makefile])
 AC_CONFIG_FILES([pyvsbsdl/Makefile])
 AC_CONFIG_FILES([vsbsdltools/Makefile])
--- origsrc/libvsbsdl-20240227/libvsbsdl/Makefile.am	2024-02-27 13:37:00.000000000 +0900
+++ src/libvsbsdl-20240227/libvsbsdl/Makefile.am	2024-03-02 13:27:35.699656100 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libvsbsdl-20240227/pyvsbsdl/Makefile.am	2024-02-27 13:37:00.000000000 +0900
+++ src/libvsbsdl-20240227/pyvsbsdl/Makefile.am	2024-03-02 13:27:56.185730200 +0900
@@ -1,5 +1,6 @@
 if HAVE_PYTHON
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -41,7 +42,7 @@ pyvsbsdl_la_LIBADD = \
 	@LIBBFIO_LIBADD@
 
 pyvsbsdl_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyvsbsdl_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyvsbsdl_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
--- origsrc/libvsbsdl-20240227/tests/Makefile.am	2024-02-27 13:37:00.000000000 +0900
+++ src/libvsbsdl-20240227/tests/Makefile.am	2024-03-02 13:28:05.156829800 +0900
@@ -1,6 +1,7 @@
 AUTOMAKE_OPTIONS = subdir-objects
 
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libvsbsdl-20240227/vsbsdltools/Makefile.am	2024-02-27 13:37:00.000000000 +0900
+++ src/libvsbsdl-20240227/vsbsdltools/Makefile.am	2024-03-02 13:27:26.605852600 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
