--- origsrc/shmcat-1.9/configure.ac	2017-12-16 06:20:54.000000000 +0900
+++ src/shmcat-1.9/configure.ac	2022-12-28 06:34:00.867755500 +0900
@@ -106,7 +106,7 @@ AS_IF([test "x$enable_posix_shm" != "xno
      )
 
 AM_GNU_GETTEXT
-AM_GNU_GETTEXT_VERSION(0.18.3)
+dnl AM_GNU_GETTEXT_VERSION(0.18.3)
 
 dnl Define USE_NLS conditional to determine if we have to install translated man pages.
 AM_CONDITIONAL(USE_NLS, [test "x$USE_NLS" = "xyes"])
