--- origsrc/trademgen-1.00.13/config/FindBoost.cmake	2025-03-31 04:51:46.000000000 +0900
+++ src/trademgen-1.00.13/config/FindBoost.cmake	2025-11-27 17:50:19.372943700 +0900
@@ -384,9 +384,11 @@ the Boost CMake package configuration fo
 Set ``Boost_NO_BOOST_CMAKE`` to ``ON``, to disable the search for boost-cmake.
 #]=======================================================================]
 
-cmake_policy(GET CMP0167 _FindBoost_CMP0167)
-if(_FindBoost_CMP0167 STREQUAL "NEW")
-  message(FATAL_ERROR "The FindBoost module has been removed by policy CMP0167.")
+if (POLICY CMP0167)
+  cmake_policy(GET CMP0167 _FindBoost_CMP0167)
+  if(_FindBoost_CMP0167 STREQUAL "NEW")
+    message(FATAL_ERROR "The FindBoost module has been removed by policy CMP0167.")
+  endif()
 endif()
 
 if(_FindBoost_testing)
@@ -402,7 +404,9 @@ include(${CMAKE_CURRENT_LIST_DIR}/FindPa
 cmake_policy(PUSH)
 cmake_policy(SET CMP0057 NEW) # if IN_LIST
 cmake_policy(SET CMP0102 NEW) # if mark_as_advanced(non_cache_var)
-cmake_policy(SET CMP0159 NEW) # file(STRINGS) with REGEX updates CMAKE_MATCH_<n>
+if (POLICY CMP0159)
+  cmake_policy(SET CMP0159 NEW) # file(STRINGS) with REGEX updates CMAKE_MATCH_<n>
+endif()
 
 function(_boost_get_existing_target component target_var)
   set(names "${component}")
--- origsrc/trademgen-1.00.13/config/FindPython/Support.cmake	2025-03-31 04:51:46.000000000 +0900
+++ src/trademgen-1.00.13/config/FindPython/Support.cmake	2025-11-27 17:50:19.380846000 +0900
@@ -20,8 +20,10 @@ cmake_policy (SET CMP0057 NEW)
 cmake_policy (SET CMP0124 NEW)
 # registry view behavior
 cmake_policy (SET CMP0134 NEW)
-# file(STRINGS) with REGEX updates CMAKE_MATCH_<n>
-cmake_policy(SET CMP0159 NEW)
+if (POLICY CMP0159)
+  # file(STRINGS) with REGEX updates CMAKE_MATCH_<n>
+  cmake_policy(SET CMP0159 NEW)
+endif()
 
 if (NOT DEFINED _PYTHON_PREFIX)
   message (FATAL_ERROR "FindPython: INTERNAL ERROR")
--- origsrc/trademgen-1.00.13/config/project_config_embeddable.cmake	2025-03-31 04:51:46.000000000 +0900
+++ src/trademgen-1.00.13/config/project_config_embeddable.cmake	2025-11-27 17:50:19.384609000 +0900
@@ -1952,7 +1952,7 @@ macro (module_library_add_specific
     message (STATUS "${_lib_short_name} is assumed to be a Python library")
     install (TARGETS ${_lib_target}
       EXPORT ${PY_LIB_DEPENDENCY_EXPORT}
-      LIBRARY DESTINATION "${INSTALL_PY_LIB_DIR}" COMPONENT python-runtime)
+      RUNTIME DESTINATION "${INSTALL_PY_LIB_DIR}" COMPONENT python-runtime)
     
   else ("${_lib_prefix}" STREQUAL "py")
     # Install the library in the standard location
--- origsrc/trademgen-1.00.13/man/pytrademgen.doc	2025-03-31 04:51:46.000000000 +0900
+++ src/trademgen-1.00.13/man/pytrademgen.doc	2025-11-27 17:57:03.183903300 +0900
@@ -18,7 +18,7 @@
  \b -l, \b --long<br>
     Produce long formatted output (rather than the shorter one).
 
-See the output of the <tt>`pytrademgen --help'</tt> command for default options.
+See the output of the <tt>\`pytrademgen --help'</tt> command for default options.
 
 
 \section sec_see_also SEE ALSO
--- origsrc/trademgen-1.00.13/man/trademgen.doc	2025-03-31 04:51:46.000000000 +0900
+++ src/trademgen-1.00.13/man/trademgen.doc	2025-11-27 17:57:04.758820100 +0900
@@ -47,7 +47,7 @@
  \b -l, \b --log <path-to-output-log-file><br>
     Path (absolute or relative) of the output log file.
 
-See the output of the <tt>`trademgen --help'</tt> command for default options.
+See the output of the <tt>\`trademgen --help'</tt> command for default options.
 
 
 \section sec_see_also SEE ALSO
--- origsrc/trademgen-1.00.13/man/trademgen_drawBookingArrivals.doc	2025-03-31 04:51:46.000000000 +0900
+++ src/trademgen-1.00.13/man/trademgen_drawBookingArrivals.doc	2025-11-27 17:57:04.771796400 +0900
@@ -32,7 +32,7 @@
    \b trademgen_generateDemand utility.
    If blank, the default log file is 'trademgen_generateDemand.log'.<br>
 
-See the output of the <tt>`trademgen_drawBookingArrivals --help'</tt> command
+See the output of the <tt>\`trademgen_drawBookingArrivals --help'</tt> command
 for default options.
 
 
--- origsrc/trademgen-1.00.13/man/trademgen_extractBookingRequests.doc	2025-03-31 04:51:46.000000000 +0900
+++ src/trademgen-1.00.13/man/trademgen_extractBookingRequests.doc	2025-11-27 17:57:04.771796400 +0900
@@ -32,7 +32,7 @@
    \b trademgen_generateDemand utility.
    If blank, the default log file is 'trademgen_generateDemand.log'.<br>
 
-See the output of the <tt>`trademgen_extractBookingRequests --help'</tt> command
+See the output of the <tt>\`trademgen_extractBookingRequests --help'</tt> command
 for default options.
 
 
--- origsrc/trademgen-1.00.13/man/trademgen_generateDemand.doc	2025-03-31 04:51:46.000000000 +0900
+++ src/trademgen-1.00.13/man/trademgen_generateDemand.doc	2025-11-27 17:57:04.758820100 +0900
@@ -47,7 +47,7 @@
  \b -l, \b --log <path-to-output-log-file><br>
     Path (absolute or relative) of the output log file.
 
-See the output of the <tt>`trademgen_generateDemand --help'</tt> command for default options.
+See the output of the <tt>\`trademgen_generateDemand --help'</tt> command for default options.
 
 
 \section sec_see_also SEE ALSO
--- origsrc/trademgen-1.00.13/man/trademgen_with_db.doc	2025-03-31 04:51:46.000000000 +0900
+++ src/trademgen-1.00.13/man/trademgen_with_db.doc	2025-11-27 17:57:04.758820100 +0900
@@ -60,7 +60,7 @@
  \b -m, \b --dbname <database-name><br>
     SQL database name (e.g., sim_dsim).<br>
 
-See the output of the <tt>`trademgen_with_db --help'</tt> command for 
+See the output of the <tt>\`trademgen_with_db --help'</tt> command for 
 default options.
 
 
--- origsrc/trademgen-1.00.13/trademgen-config-python.cmake.in	2025-03-31 04:51:46.000000000 +0900
+++ src/trademgen-1.00.13/trademgen-config-python.cmake.in	2025-11-27 17:50:19.386629400 +0900
@@ -15,6 +15,6 @@ include ("@TRADEMGEN_CMAKE_DIR@/trademge
 
 # These are the TraDemGen IMPORTED targets, created by
 # trademgen-python-library-depends.cmake
-set (TRADEMGEN_PY_LIBRARIES pytrademgenlib)
+set (TRADEMGEN_PY_LIBRARIES pytrademgen)
 set (TRADEMGEN_PY_EXECUTABLES pytrademgen)
 
--- origsrc/trademgen-1.00.13/trademgen-config.cmake.in	2025-03-31 04:51:46.000000000 +0900
+++ src/trademgen-1.00.13/trademgen-config.cmake.in	2025-11-27 17:50:19.386629400 +0900
@@ -23,7 +23,7 @@ include ("${CMAKE_CURRENT_LIST_DIR}/@PRO
 
 # These are the TraDemGen IMPORTED targets, created by
 # trademgen-library-depends.cmake
-set (TRADEMGEN_LIBRARIES trademgenlib)
+set (TRADEMGEN_LIBRARIES trademgen)
 set (TRADEMGEN_EXECUTABLES trademgen
   trademgen_generateDemand
   trademgen_extractBookingRequests trademgen_drawBookingArrivals)
