--- origsrc/trademgen-trademgen-1.00.4/config/project_config_embeddable.cmake	2019-07-12 05:11:45.000000000 +0900
+++ src/trademgen-trademgen-1.00.4/config/project_config_embeddable.cmake	2019-07-31 19:59:54.295397200 +0900
@@ -1507,7 +1507,7 @@ endmacro (module_set_name)
 macro (add_modules)
   set (_embedded_components ${ARGV})
   set (LIB_DEPENDENCY_EXPORT ${PROJECT_NAME}-library-depends)
-  set (LIB_DEPENDENCY_EXPORT_PATH "${INSTALL_DATA_DIR}/${PROJECT_NAME}/CMake")
+  set (LIB_DEPENDENCY_EXPORT_PATH "${INSTALL_LIB_DIR}/cmake/${PROJECT_NAME}")
   #
   foreach (_embedded_comp ${_embedded_components})
     #
@@ -1792,13 +1792,14 @@ macro (module_library_add_specific
 	message (STATUS "${_lib_short_name} is assumed to be a Python library")
 	install (TARGETS ${_lib_target}
       EXPORT ${LIB_DEPENDENCY_EXPORT}
-      LIBRARY DESTINATION "${INSTALL_PY_LIB_DIR}" COMPONENT runtime)
+      RUNTIME DESTINATION "${INSTALL_PY_LIB_DIR}")
 
   else ("${_lib_prefix}" STREQUAL "py")
 	# Install the library in the standard location
 	install (TARGETS ${_lib_target}
       EXPORT ${LIB_DEPENDENCY_EXPORT}
-      LIBRARY DESTINATION "${INSTALL_LIB_DIR}" COMPONENT runtime)
+      RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
+      ARCHIVE DESTINATION "${INSTALL_LIB_DIR}")
   endif ("${_lib_prefix}" STREQUAL "py")
   
   # Register, for reporting purpose, the list of libraries to be built
@@ -1995,7 +1996,7 @@ endmacro (module_script_add)
 # can refer to it (for libraries, header files and binaries)
 macro (module_export_install)
   install (EXPORT ${LIB_DEPENDENCY_EXPORT}
-    DESTINATION "${INSTALL_DATA_DIR}/${PACKAGE}/CMake" COMPONENT devel)
+    DESTINATION "${INSTALL_LIB_DIR}/cmake/${PACKAGE}" COMPONENT devel)
 endmacro (module_export_install)
 
 
--- origsrc/trademgen-trademgen-1.00.4/trademgen-config.cmake.in	2019-07-12 05:11:45.000000000 +0900
+++ src/trademgen-trademgen-1.00.4/trademgen-config.cmake.in	2019-07-31 19:59:54.311029600 +0900
@@ -23,6 +23,6 @@ include ("@TRADEMGEN_CMAKE_DIR@/trademge
 
 # These are the TraDemGen IMPORTED targets, created by
 # trademgen-library-depends.cmake
-set (TRADEMGEN_LIBRARIES trademgenlib pytrademgenlib)
+set (TRADEMGEN_LIBRARIES trademgen)
 set (TRADEMGEN_EXECUTABLES trademgen
   trademgen_generateDemand trademgen_extractBookingRequests)
