--- origsrc/trademgen-trademgen-1.00.6/config/project_config_embeddable.cmake	2020-06-01 10:32:41.000000000 +0900
+++ src/trademgen-trademgen-1.00.6/config/project_config_embeddable.cmake	2020-06-24 08:13:09.740235800 +0900
@@ -1635,7 +1635,7 @@ macro (add_modules)
   set (_embedded_components ${ARGV})
   set (LIB_DEPENDENCY_EXPORT ${PROJECT_NAME}-library-depends)
   set (PY_LIB_DEPENDENCY_EXPORT ${PROJECT_NAME}-python-library-depends)
-  set (LIB_DEPENDENCY_EXPORT_PATH "${INSTALL_DATA_DIR}/${PROJECT_NAME}/CMake")
+  set (LIB_DEPENDENCY_EXPORT_PATH "${INSTALL_LIB_DIR}/cmake/${PROJECT_NAME}")
   #
   foreach (_embedded_comp ${_embedded_components})
     #
@@ -1943,13 +1943,14 @@ macro (module_library_add_specific
     message (STATUS "${_lib_short_name} is assumed to be a Python library")
     install (TARGETS ${_lib_target}
       EXPORT ${PY_LIB_DEPENDENCY_EXPORT}
-      LIBRARY DESTINATION "${INSTALL_PY_LIB_DIR}" COMPONENT python-runtime)
+      RUNTIME DESTINATION "${INSTALL_PY_LIB_DIR}" COMPONENT python-runtime)
     
   else ("${_lib_prefix}" STREQUAL "py")
     # Install the library in the standard location
     install (TARGETS ${_lib_target}
       EXPORT ${LIB_DEPENDENCY_EXPORT}
-      LIBRARY DESTINATION "${INSTALL_LIB_DIR}" COMPONENT runtime)
+      RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
+      ARCHIVE DESTINATION "${INSTALL_LIB_DIR}" COMPONENT runtime)
   endif ("${_lib_prefix}" STREQUAL "py")
   
   # Register, for reporting purpose, the list of libraries to be built
@@ -2165,11 +2166,11 @@ endmacro (module_script_add)
 # can refer to it (for libraries, header files and binaries)
 macro (module_export_install)
   install (EXPORT ${LIB_DEPENDENCY_EXPORT}
-    DESTINATION "${INSTALL_DATA_DIR}/${PACKAGE}/CMake" COMPONENT devel)
+    DESTINATION "${INSTALL_LIB_DIR}/cmake/${PACKAGE}" COMPONENT devel)
 
   if (NEED_PYTHON)
 	install (EXPORT ${PY_LIB_DEPENDENCY_EXPORT}
-      DESTINATION "${INSTALL_DATA_DIR}/${PACKAGE}/CMake" COMPONENT python-devel)
+      DESTINATION "${INSTALL_LIB_DIR}/cmake/${PACKAGE}" COMPONENT python-devel)
   endif (NEED_PYTHON)
 endmacro (module_export_install)
 
--- origsrc/trademgen-trademgen-1.00.6/trademgen-config.cmake.in	2020-06-01 10:32:41.000000000 +0900
+++ src/trademgen-trademgen-1.00.6/trademgen-config.cmake.in	2020-06-24 08:13:25.581581300 +0900
@@ -21,7 +21,7 @@ include ("@TRADEMGEN_CMAKE_DIR@/trademge
 
 # These are the TraDemGen IMPORTED targets, created by
 # trademgen-library-depends.cmake
-set (TRADEMGEN_LIBRARIES trademgenlib)
+set (TRADEMGEN_LIBRARIES trademgen)
 set (TRADEMGEN_EXECUTABLES trademgen
   trademgen_generateDemand
   trademgen_extractBookingRequests trademgen_drawBookingArrivals)
