--- origsrc/libcthreads-20180724/Makefile.am	2018-07-24 15:54:53.000000000 +0900
+++ src/libcthreads-20180724/Makefile.am	2019-05-15 08:18:55.723484400 +0900
@@ -3,9 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
 	libcthreads \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libcthreads-20180724/configure.ac	2018-07-24 15:54:53.000000000 +0900
+++ src/libcthreads-20180724/configure.ac	2019-05-15 08:12:19.641680100 +0900
@@ -109,7 +109,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
 AC_CONFIG_FILES([libcthreads/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
--- origsrc/libcthreads-20180724/libcthreads/Makefile.am	2018-07-24 15:54:53.000000000 +0900
+++ src/libcthreads-20180724/libcthreads/Makefile.am	2019-05-15 08:17:02.142810700 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libcthreads-20180724/tests/Makefile.am	2018-07-24 15:54:53.000000000 +0900
+++ src/libcthreads-20180724/tests/Makefile.am	2019-05-15 08:20:45.842608600 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
