--- origsrc/libimobiledevice-1.1.5/cython/Makefile.am	2013-03-08 07:05:33.000000000 +0900
+++ src/libimobiledevice-1.1.5/cython/Makefile.am	2013-12-23 20:25:06.760123600 +0900
@@ -37,7 +37,7 @@ imobiledevicedir = $(pyexecdir)
 imobiledevice_LTLIBRARIES = imobiledevice.la
 imobiledevice_la_SOURCES = imobiledevice.pyx
 imobiledevice_la_CFLAGS = -I../include -I../src $(PYTHON_CPPFLAGS) $(AM_CFLAGS) -Wno-shadow -Wno-redundant-decls -Wno-switch-default -Wno-strict-aliasing -Wno-implicit-function-declaration
-imobiledevice_la_LDFLAGS = -module -avoid-version -L$(libdir) $(PYTHON_LDFLAGS) $(AM_LDFLAGS)
+imobiledevice_la_LDFLAGS = -no-undefined -module -avoid-version -L$(libdir) $(PYTHON_LDFLAGS) $(AM_LDFLAGS)
 imobiledevice_la_LIBADD = ../src/libimobiledevice.la
 
 imobiledevice.c: imobiledevice.pyx $(PXDINCLUDES) $(PXIINCLUDES)
