--- origsrc/libimobiledevice-1.2.0/configure.ac	2015-01-29 07:11:57.000000000 +0900
+++ src/libimobiledevice-1.2.0/configure.ac	2020-01-13 14:01:15.003588200 +0900
@@ -69,7 +69,7 @@ fi
 # Check for operating system
 AC_MSG_CHECKING([whether to enable WIN32 build settings])
 case ${host_os} in
-  *mingw32*|*cygwin*)
+  *mingw32*)
     win32=true
     AC_MSG_RESULT([yes])
     AC_CHECK_TOOL([WINDRES], [windres], AC_MSG_ERROR([windres not found]))
