--- origsrc/valvula-1.0.8/configure.ac	2017-11-29 01:55:43.000000000 +0900
+++ src/valvula-1.0.8/configure.ac	2024-09-24 17:28:11.479185200 +0900
@@ -54,7 +54,7 @@ AM_PROG_LIBTOOL
 compiler_options=""
 STRICT_PROTOTYPES=""
 if test "$compiler" = "gcc" ; then
-   compiler_options="-Wstrict-prototypes -Wall -Werror -g -ansi -fstack-protector-all -Wstack-protector"
+   compiler_options="-Wstrict-prototypes -Wall -g -ansi -fstack-protector-all -Wstack-protector"
    echo "Detected gcc compiler: $compiler, adding options: $compiler_options"
 fi
 AC_SUBST(compiler_options)
--- origsrc/valvula-1.0.8/lib/valvula.c	2017-11-29 01:55:43.000000000 +0900
+++ src/valvula-1.0.8/lib/valvula.c	2024-09-24 17:19:07.378589600 +0900
@@ -52,7 +52,7 @@
  * vsprintf where the output buffer is limited) but unfortunately is
  * not available in ANSI C. This is only required when compile valvula
  * with log support */
-#if defined(ENABLE_VALVULA_LOG)
+#if defined(ENABLE_VALVULA_LOG) && !defined(__CYGWIN__)
 int vsnprintf(char *str, size_t size, const char *format, va_list ap);
 #endif
 
--- origsrc/valvula-1.0.8/plugins/mod-bwl/Makefile.am	2017-11-29 01:55:43.000000000 +0900
+++ src/valvula-1.0.8/plugins/mod-bwl/Makefile.am	2024-09-24 17:37:13.204432200 +0900
@@ -1,12 +1,12 @@
 EXTRA_DIST = mod-bwl.xml
 
-INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -DCOMPILATION_DATE=`date +%s` \
+INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -I$(top_srcdir)/lib -I$(top_srcdir)/server -DCOMPILATION_DATE=`date +%s` \
 	   -DVERSION=\"$(TURBULENCE_VERSION)\" \
 	   $(AXL_CFLAGS) $(VORTEX_CFLAGS) $(EXARG_CFLAGS)
 
 lib_LTLIBRARIES      = mod-bwl.la
 mod_bwl_la_SOURCES  = mod-bwl.c
-mod_bwl_la_LDFLAGS  = -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
+mod_bwl_la_LDFLAGS  = -avoid-version -no-undefined -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
 
 # reconfigure module installation directory
 libdir = $(prefix)/lib/valvulad/modules
--- origsrc/valvula-1.0.8/plugins/mod-lmm/Makefile.am	2017-11-29 01:55:43.000000000 +0900
+++ src/valvula-1.0.8/plugins/mod-lmm/Makefile.am	2024-09-24 17:37:13.214740700 +0900
@@ -1,12 +1,12 @@
 EXTRA_DIST = mod-lmm.xml
 
-INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -DCOMPILATION_DATE=`date +%s` \
+INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -I$(top_srcdir)/lib -I$(top_srcdir)/server -DCOMPILATION_DATE=`date +%s` \
 	   -DVERSION=\"$(TURBULENCE_VERSION)\" \
 	   $(AXL_CFLAGS) $(VORTEX_CFLAGS) $(EXARG_CFLAGS)
 
 lib_LTLIBRARIES      = mod-lmm.la
 mod_lmm_la_SOURCES  = mod-lmm.c
-mod_lmm_la_LDFLAGS  = -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
+mod_lmm_la_LDFLAGS  = -avoid-version -no-undefined -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
 
 # reconfigure module installation directory
 libdir = $(prefix)/lib/valvulad/modules
--- origsrc/valvula-1.0.8/plugins/mod-mquota/Makefile.am	2017-11-29 01:55:43.000000000 +0900
+++ src/valvula-1.0.8/plugins/mod-mquota/Makefile.am	2024-09-24 17:37:13.214740700 +0900
@@ -1,12 +1,12 @@
 EXTRA_DIST = mod-mquota.xml
 
-INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -DCOMPILATION_DATE=`date +%s` \
+INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -I$(top_srcdir)/lib -I$(top_srcdir)/server -DCOMPILATION_DATE=`date +%s` \
 	   -DVERSION=\"$(TURBULENCE_VERSION)\" \
 	   $(AXL_CFLAGS) $(VORTEX_CFLAGS) $(EXARG_CFLAGS)
 
 lib_LTLIBRARIES      = mod-mquota.la
 mod_mquota_la_SOURCES  = mod-mquota.c mod-mquota.h
-mod_mquota_la_LDFLAGS  = -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
+mod_mquota_la_LDFLAGS  = -avoid-version -no-undefined -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
 
 # reconfigure module installation directory
 libdir = $(prefix)/lib/valvulad/modules
--- origsrc/valvula-1.0.8/plugins/mod-mw/Makefile.am	2017-11-29 01:55:43.000000000 +0900
+++ src/valvula-1.0.8/plugins/mod-mw/Makefile.am	2024-09-24 17:37:13.214740700 +0900
@@ -1,12 +1,12 @@
 EXTRA_DIST = mod-mw.xml mod-mw.example.conf
 
-INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -DCOMPILATION_DATE=`date +%s` \
+INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -I$(top_srcdir)/lib -I$(top_srcdir)/server -DCOMPILATION_DATE=`date +%s` \
 	   -DVERSION=\"$(TURBULENCE_VERSION)\" \
 	   $(AXL_CFLAGS) $(VORTEX_CFLAGS) $(EXARG_CFLAGS)
 
 lib_LTLIBRARIES      = mod-mw.la
 mod_mw_la_SOURCES  = mod-mw.c
-mod_mw_la_LDFLAGS  = -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
+mod_mw_la_LDFLAGS  = -avoid-version -no-undefined -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
 
 # reconfigure module installation directory
 libdir = $(prefix)/lib/valvulad/modules
--- origsrc/valvula-1.0.8/plugins/mod-object-resolver/Makefile.am	2017-11-29 01:55:43.000000000 +0900
+++ src/valvula-1.0.8/plugins/mod-object-resolver/Makefile.am	2024-09-24 17:37:13.212431600 +0900
@@ -1,12 +1,12 @@
 EXTRA_DIST = mod-object-resolver.xml
 
-INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -DCOMPILATION_DATE=`date +%s` \
+INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -I$(top_srcdir)/lib -I$(top_srcdir)/server -DCOMPILATION_DATE=`date +%s` \
 	   -DVERSION=\"$(TURBULENCE_VERSION)\" \
 	   $(AXL_CFLAGS) $(VORTEX_CFLAGS) $(EXARG_CFLAGS)
 
 lib_LTLIBRARIES      = mod-object-resolver.la
 mod_object_resolver_la_SOURCES  = mod-object-resolver.c
-mod_object_resolver_la_LDFLAGS  = -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
+mod_object_resolver_la_LDFLAGS  = -avoid-version -no-undefined -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
 
 # reconfigure module installation directory
 libdir = $(prefix)/lib/valvulad/modules
--- origsrc/valvula-1.0.8/plugins/mod-slm/Makefile.am	2017-11-29 01:55:43.000000000 +0900
+++ src/valvula-1.0.8/plugins/mod-slm/Makefile.am	2024-09-24 17:37:13.214740700 +0900
@@ -1,12 +1,12 @@
 EXTRA_DIST = mod-slm.xml
 
-INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -DCOMPILATION_DATE=`date +%s` \
+INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -I$(top_srcdir)/lib -I$(top_srcdir)/server -DCOMPILATION_DATE=`date +%s` \
 	   -DVERSION=\"$(TURBULENCE_VERSION)\" \
 	   $(AXL_CFLAGS) $(VORTEX_CFLAGS) $(EXARG_CFLAGS)
 
 lib_LTLIBRARIES      = mod-slm.la
 mod_slm_la_SOURCES  = mod-slm.c
-mod_slm_la_LDFLAGS  = -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
+mod_slm_la_LDFLAGS  = -avoid-version -no-undefined -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
 
 # reconfigure module installation directory
 libdir = $(prefix)/lib/valvulad/modules
--- origsrc/valvula-1.0.8/plugins/mod-test/Makefile.am	2017-11-29 01:55:43.000000000 +0900
+++ src/valvula-1.0.8/plugins/mod-test/Makefile.am	2024-09-24 17:37:13.207445400 +0900
@@ -1,12 +1,12 @@
 EXTRA_DIST = mod-test.xml
 
-INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -DCOMPILATION_DATE=`date +%s` \
+INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -I$(top_srcdir)/lib -I$(top_srcdir)/server -DCOMPILATION_DATE=`date +%s` \
 	   -DVERSION=\"$(TURBULENCE_VERSION)\" \
 	   $(AXL_CFLAGS) $(VORTEX_CFLAGS) $(EXARG_CFLAGS)
 
 lib_LTLIBRARIES      = mod-test.la
 mod_test_la_SOURCES  = mod-test.c
-mod_test_la_LDFLAGS  = -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
+mod_test_la_LDFLAGS  = -avoid-version -no-undefined -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
 
 # reconfigure module installation directory
 libdir = $(prefix)/lib/valvulad/modules
--- origsrc/valvula-1.0.8/plugins/mod-ticket/Makefile.am	2017-11-29 01:55:43.000000000 +0900
+++ src/valvula-1.0.8/plugins/mod-ticket/Makefile.am	2024-09-24 17:37:13.209431600 +0900
@@ -1,12 +1,12 @@
 EXTRA_DIST = mod-ticket.xml
 
-INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -DCOMPILATION_DATE=`date +%s` \
+INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -I$(top_srcdir)/lib -I$(top_srcdir)/server -DCOMPILATION_DATE=`date +%s` \
 	   -DVERSION=\"$(TURBULENCE_VERSION)\" \
 	   $(AXL_CFLAGS) $(VORTEX_CFLAGS) $(EXARG_CFLAGS)
 
 lib_LTLIBRARIES      = mod-ticket.la
 mod_ticket_la_SOURCES  = mod-ticket.c
-mod_ticket_la_LDFLAGS  = -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
+mod_ticket_la_LDFLAGS  = -avoid-version -no-undefined -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
 
 # reconfigure module installation directory
 libdir = $(prefix)/lib/valvulad/modules
--- origsrc/valvula-1.0.8/plugins/mod-transport/Makefile.am	2017-11-29 01:55:43.000000000 +0900
+++ src/valvula-1.0.8/plugins/mod-transport/Makefile.am	2024-09-24 17:37:13.212431600 +0900
@@ -1,12 +1,12 @@
 EXTRA_DIST = mod-transport.xml
 
-INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -DCOMPILATION_DATE=`date +%s` \
+INCLUDES = -Wall -g -ansi -I ../../lib -I ../../server -I$(top_srcdir)/lib -I$(top_srcdir)/server -DCOMPILATION_DATE=`date +%s` \
 	   -DVERSION=\"$(TURBULENCE_VERSION)\" \
 	   $(AXL_CFLAGS) $(VORTEX_CFLAGS) $(EXARG_CFLAGS)
 
 lib_LTLIBRARIES      = mod-transport.la
 mod_transport_la_SOURCES  = mod-transport.c
-mod_transport_la_LDFLAGS  = -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
+mod_transport_la_LDFLAGS  = -avoid-version -no-undefined -module -ldl $(top_builddir)/lib/libvalvula.la $(top_builddir)/server/libvalvulad.la
 
 # reconfigure module installation directory
 libdir = $(prefix)/lib/valvulad/modules
--- origsrc/valvula-1.0.8/test/Makefile.am	2017-11-29 01:55:43.000000000 +0900
+++ src/valvula-1.0.8/test/Makefile.am	2024-09-24 17:43:33.225353300 +0900
@@ -45,5 +45,5 @@ INCLUDES = -I$(top_srcdir)/lib -I$(top_s
 LIBS            = $(AXL_LIBS) $(PTHREAD_LIBS) $(ADDITIONAL_LIBS) $(NOPOLL_LIBS) $(SQLITE3_LIBS)
 
 test_01_SOURCES = test_01.c
-test_01_LDADD = $(AXL_LIBS) $(VALVULA_LIBS) $(top_srcdir)/server/libvalvulad.la $(top_srcdir)/lib/libvalvula.la $(top_srcdir)/plugins/mod-mquota/mod-mquota.la
+test_01_LDADD = $(AXL_LIBS) $(VALVULA_LIBS) $(top_builddir)/server/libvalvulad.la $(top_builddir)/lib/libvalvula.la $(top_builddir)/plugins/mod-mquota/mod-mquota.la
 
