--- origsrc/ciso-1.0.2/Makefile	2015-07-08 03:05:38.000000000 +0900
+++ src/ciso-1.0.2/Makefile	2022-06-24 13:12:16.251620500 +0900
@@ -2,16 +2,14 @@ DESTDIR     =
 prefix      = /usr
 USRBINDIR   = $(DESTDIR)$(prefix)/bin
 
-CC = $(CROSS_COMPILE)gcc
-LD = $(CROSS_COMPILE)gcc
 INSTALL = install
 
 all : ciso
 ciso : ciso.o
-	gcc -o ciso ciso.o -lz
+	$(CC) $(LDFLAGS) -o ciso ciso.o -lz
 
 ciso.o : ciso.c
-	gcc -o ciso.o -c ciso.c
+	$(CC) $(CFLAGS) -o ciso.o -c ciso.c
 
 install :
 	$(INSTALL) -m 755 ciso $(USRBINDIR)/ciso
