--- origsrc/golem95-1.3.2/Makefile.am	2014-09-16 21:52:45.000000000 +0900
+++ src/golem95-1.3.2/Makefile.am	2015-05-19 16:55:55.175512000 +0900
@@ -7,10 +7,12 @@ pkgconfig_DATA = golem95.pc
 
 lib_LTLIBRARIES         = libgolem.la libavh_olo.la
 libgolem_la_SOURCES     =
+libgolem_la_LDFLAGS     = -no-undefined
 libavh_olo_la_SOURCES   =
-libavh_olo_la_LIBADD    = \
+libavh_olo_la_LIBADD    = -lgfortran \
        $(builddir)/avh_olo-3.5/libavh_olo.la
-libgolem_la_LIBADD      = \
+libavh_olo_la_LDFLAGS   = -no-undefined
+libgolem_la_LIBADD      = -lgfortran \
 	$(builddir)/src/form_factor/libgolem95_formfactor.la \
 	$(builddir)/src/integrals/four_point/libgolem95_integrals_four_point.la \
 	$(builddir)/src/integrals/three_point/libgolem95_integrals_three_point.la \
--- origsrc/golem95-1.3.2/src/kinematic/Makefile.am	2014-09-16 21:35:41.000000000 +0900
+++ src/golem95-1.3.2/src/kinematic/Makefile.am	2015-05-19 16:45:57.147370200 +0900
@@ -1,6 +1,6 @@
 noinst_LTLIBRARIES=libgolem95_kinematics.la
 
-AM_FCFLAGS=-I$(top_builddir)/src/module -I$(top_builddir)/avh_olo-3.5
+AM_FCFLAGS=-I$(top_builddir)/src/kinematic -I$(top_builddir)/src/module -I$(top_builddir)/avh_olo-3.5
 
 libgolem95_kinematics_la_SOURCES= inverse_matrice.f90 matrice_s.f90
 libgolem95_kinematics_la_FCFLAGS=$(AM_FCFLAGS)
