diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/hershey-fonts/CMakeLists.txt src/hershey-fonts/CMakeLists.txt
--- origsrc/hershey-fonts/CMakeLists.txt	2013-12-03 06:51:19.000000000 +0900
+++ src/hershey-fonts/CMakeLists.txt	2016-01-30 11:09:23.347637700 +0900
@@ -24,3 +24,11 @@ foreach(jhf ${jhf_files})
     add_test("${jhf}" ${CMAKE_BINARY_DIR}/tools/hershey-font-check "${jhf}")
 endforeach(jhf)
 
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "${hershey-fonts_VERSION_MAJOR}.${hershey-fonts_VERSION_MINOR}")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/${PROJECT_NAME}.pc.in ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/hershey-fonts/hershey-fonts.pc.in src/hershey-fonts/hershey-fonts.pc.in
--- origsrc/hershey-fonts/hershey-fonts.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/hershey-fonts/hershey-fonts.pc.in	2016-01-30 11:08:59.834483700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: hershey-fonts
+Description: Hershey vector fonts library, tools, and glyphs
+Version: @VERSION@
+Libs: -L${libdir} -lhersheyfonts
+Cflags: -I${includedir}
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/hershey-fonts/libhersheyfont/CMakeLists.txt src/hershey-fonts/libhersheyfont/CMakeLists.txt
--- origsrc/hershey-fonts/libhersheyfont/CMakeLists.txt	2013-12-03 06:51:19.000000000 +0900
+++ src/hershey-fonts/libhersheyfont/CMakeLists.txt	2016-01-30 11:05:01.618102700 +0900
@@ -12,10 +12,13 @@ add_definitions(-DHERSHEY_FONTS_DIR="${H
 
 add_library(hersheyfont			SHARED hersheyfont.c)
 set_target_properties(hersheyfont	PROPERTIES VERSION 0 SOVERSION 0)
-install(TARGETS hersheyfont		LIBRARY DESTINATION lib)
+install(TARGETS hersheyfont	RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+option(ENABLE_STATIC "enable static" OFF)
+if(ENABLE_STATIC)
 add_library(hersheyfontStatic		STATIC hersheyfont.c)
 set_target_properties(hersheyfontStatic	PROPERTIES OUTPUT_NAME hersheyfont)
 install(TARGETS hersheyfontStatic	ARCHIVE DESTINATION lib)
+endif()
 install(FILES hersheyfont.h		DESTINATION include)
 
 
@@ -46,4 +49,3 @@ foreach(manpage ${manpage_symlinks})
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${manpage}
 						DESTINATION share/man/man3)
 endforeach(manpage)
-
