--- origsrc/cubature-20101018/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/cubature-20101018/Makefile.am	2011-05-04 14:43:36.531711400 +0900
@@ -0,0 +1,6 @@
+lib_LTLIBRARIES = libcubature.la
+
+libcubature_la_SOURCES = cubature.c
+libcubature_la_LDFLAGS = -no-undefined
+
+include_HEADERS = cubature.h
--- origsrc/cubature-20101018/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/cubature-20101018/configure.ac	2011-05-04 14:45:14.395672200 +0900
@@ -0,0 +1,28 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([cubature], [20101018], [http://ab-initio.mit.edu/wiki/index.php/Cubature])
+AC_CONFIG_SRCDIR([cubature.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT([win32-dll disable-static])
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([float.h limits.h stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([pow sqrt])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
