--- origsrc/Minuit2-5.27.02/Makefile.am	2010-05-05 01:40:06.000000000 +0900
+++ src/Minuit2-5.27.02/Makefile.am	2010-07-31 10:58:26.777148400 +0900
@@ -7,7 +7,7 @@
 AUTOMAKE_OPTIONS = foreign
 
 # A trick to add options to the aclocal command.
-ACLOCAL = aclocal -I config/m4
+ACLOCAL_AMFLAGS = -I config/m4
 
 ## The subdirectories.  
 SUBDIRS =					\
--- origsrc/Minuit2-5.27.02/configure.in	2010-07-31 10:46:47.292773400 +0900
+++ src/Minuit2-5.27.02/configure.in	2010-07-31 10:57:28.902148400 +0900
@@ -20,6 +20,7 @@ CC=$CXX
 
 dnl disable static building
 dnl AM_DISABLE_STATIC
+AC_LIBTOOL_WIN32_DLL
 AM_PROG_LIBTOOL
 
 dnl Checks for libraries.
--- origsrc/Minuit2-5.27.02/src/Makefile.am	2010-07-31 10:46:47.667773400 +0900
+++ src/Minuit2-5.27.02/src/Makefile.am	2010-07-31 10:48:16.027148400 +0900
@@ -10,6 +10,7 @@ AUTOMAKE_OPTIONS = foreign
 lib_LTLIBRARIES = libMinuit2.la
 
 libMinuit2_la_LIBADD = $(GOMP_LIBS)
+libMinuit2_la_LDFLAGS = -no-undefined
 
 libMinuit2_la_SOURCES =				\
          AnalyticalGradientCalculator.cxx	\
