--- origsrc/Minuit2-5.28.00/Makefile.am	2011-01-07 16:46:50.000000000 +0900
+++ src/Minuit2-5.28.00/Makefile.am	2012-03-30 13:41:34.147506600 +0900
@@ -7,7 +7,7 @@
 AUTOMAKE_OPTIONS = foreign
 
 # A trick to add options to the aclocal command.
-ACLOCAL = aclocal -I config/m4
+ACLOCAL_AMFLAGS = -I config/m4
 
 ## The subdirectories.  
 SUBDIRS =					\
--- origsrc/Minuit2-5.28.00/config/m4/pfk_cxx_lib_path.m4	2011-01-07 16:46:50.000000000 +0900
+++ src/Minuit2-5.28.00/config/m4/pfk_cxx_lib_path.m4	2012-03-30 13:43:17.557662800 +0900
@@ -7,7 +7,7 @@ dnl `/usr/local/lib'.
 dnl
 dnl @author Paul_Kunz@slac.stanford.edu
 dnl
-AC_DEFUN(PFK_CXX_LIB_PATH,
+AC_DEFUN([PFK_CXX_LIB_PATH],
 [ AC_PATH_PROG(pfk_cxx_lib_path, $CXX, $CXX )
   AC_MSG_CHECKING(standard C++ library path)
   CXX_LIB_PATH=`dirname $pfk_cxx_lib_path | sed -e "s/bin/lib/"`
--- origsrc/Minuit2-5.28.00/src/Makefile.am	2012-03-30 13:39:29.834030000 +0900
+++ src/Minuit2-5.28.00/src/Makefile.am	2012-03-30 13:42:08.725631600 +0900
@@ -11,6 +11,8 @@ lib_LTLIBRARIES = libMinuit2.la
 
 libMinuit2_la_LIBADD = $(GOMP_LIBS)
 
+libMinuit2_la_LDFLAGS = -no-undefined
+
 libMinuit2_la_SOURCES =				\
          AnalyticalGradientCalculator.cxx	\
          BasicMinimumError.cxx			\
