--- origsrc/czmq-1.4.1/src/Makefile.am	2013-04-30 17:09:38.000000000 +0900
+++ src/czmq-1.4.1/src/Makefile.am	2013-05-26 20:14:43.964820300 +0900
@@ -45,10 +45,11 @@ czmq_selftest_LDADD = libczmq.la
 czmq_selftest_SOURCES = czmq_selftest.c
 
 if ON_MINGW
-libczmq_la_LDFLAGS = -no-undefined -avoid-version -version-info @LTVER@
+libczmq_la_LDFLAGS = -avoid-version -version-info @LTVER@
 else
 libczmq_la_LDFLAGS = -version-info @LTVER@
 endif
+libczmq_la_LDFLAGS += -no-undefined
 
 TESTS = czmq_selftest
 
