--- origsrc/sealcurses/CMakeLists.txt	2023-08-29 07:24:13.000000000 +0900
+++ src/sealcurses/CMakeLists.txt	2023-08-29 07:26:01.826944200 +0900
@@ -107,7 +107,7 @@ if (ENABLE_SHARED)
             the_Foundation::the_Foundation
             ${CURSES_LDFLAGS}
     )
-    install (TARGETS sealcurses DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    install (TARGETS sealcurses RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif ()
 
 install (FILES ${HEADERS} DESTINATION include/sealcurses)
