--- origsrc/libmkv-0.6.5.1/Makefile.am	2011-10-16 06:00:20.000000000 +0900
+++ src/libmkv-0.6.5.1/Makefile.am	2014-08-02 20:51:20.028203600 +0900
@@ -1,4 +1,2 @@
 ## Makefile.am -- Process this file with automake to produce Makefile.in
-ACLOCAL_AMFLAGS = -I m4
-
-SUBDIRS = src include
+SUBDIRS = src include test
--- origsrc/libmkv-0.6.5.1/configure.ac	2011-10-16 06:00:20.000000000 +0900
+++ src/libmkv-0.6.5.1/configure.ac	2014-08-02 20:51:20.031133300 +0900
@@ -40,5 +40,6 @@ AC_CHECK_FUNCS([memset strdup])
 
 AC_CONFIG_FILES([Makefile
                  include/Makefile
-                 src/Makefile])
+                 src/Makefile
+                 test/Makefile])
 AC_OUTPUT
--- origsrc/libmkv-0.6.5.1/src/Makefile.am	2011-10-16 06:00:20.000000000 +0900
+++ src/libmkv-0.6.5.1/src/Makefile.am	2014-08-02 20:51:20.035039600 +0900
@@ -4,3 +4,4 @@ INCLUDES	= -I$(top_builddir) -I$(top_src
 
 lib_LTLIBRARIES = libmkv.la
 libmkv_la_SOURCES = matroska.c matroska.h tracks.c chapters.c ebml.h ebml.c md5.h md5.c tags.c attachments.c
+libmkv_la_LDFLAGS = -no-undefined
--- origsrc/libmkv-0.6.5.1/test/Makefile.am	2011-10-16 06:00:20.000000000 +0900
+++ src/libmkv-0.6.5.1/test/Makefile.am	2014-08-02 20:51:20.039922400 +0900
@@ -2,6 +2,6 @@
 
 INCLUDES	= -I$(top_builddir) -I$(top_srcdir) -I$(top_srcdir)/include
 
-noinst_PROGRAMS = test
+check_PROGRAMS = test
 test_SOURCES = test.c
 test_LDADD = $(top_builddir)/src/libmkv.la
