--- origsrc/libmkv/Makefile.am	2010-11-11 00:28:29.000000000 +0900
+++ src/libmkv/Makefile.am	2010-11-11 00:28:38.766910800 +0900
@@ -1,4 +1,4 @@
 ## Makefile.am -- Process this file with automake to produce Makefile.in
-ACLOCAL_AMFLAGS = -I m4
+ACLOCAL_AMFLAGS = 
 
-SUBDIRS = src include
+SUBDIRS = src include test
--- origsrc/libmkv/configure.ac	2010-11-11 00:28:29.000000000 +0900
+++ src/libmkv/configure.ac	2010-11-11 00:33:40.650998800 +0900
@@ -13,7 +13,7 @@ AM_INIT_AUTOMAKE
 
 # Checks for programs.
 AC_PROG_CC
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_PROG_INSTALL
 
 AC_DEFINE(_GNU_SOURCE, 1, [GNU C extensions])
@@ -40,5 +40,6 @@ AC_CHECK_FUNCS([memset strdup])
 
 AC_CONFIG_FILES([Makefile
                  include/Makefile
-                 src/Makefile])
+                 src/Makefile
+                 test/Makefile])
 AC_OUTPUT
--- origsrc/libmkv/src/Makefile.am	2010-11-11 00:28:29.000000000 +0900
+++ src/libmkv/src/Makefile.am	2010-11-11 00:28:38.776925200 +0900
@@ -4,3 +4,4 @@ INCLUDES	= -I$(top_builddir) -I$(top_src
 
 lib_LTLIBRARIES = libmkv.la
 libmkv_la_SOURCES = matroska.c matroska.h tracks.c chapters.c ebml.h ebml.c md5.h md5.c tags.c attachments.c
+libmkv_la_LDFLAGS = -no-undefined
--- origsrc/libmkv/test/Makefile.am	2010-11-11 00:28:29.000000000 +0900
+++ src/libmkv/test/Makefile.am	2010-11-11 10:10:53.748430000 +0900
@@ -2,6 +2,6 @@
 
 INCLUDES	= -I$(top_builddir) -I$(top_srcdir) -I$(top_srcdir)/include
 
-noinst_PROGRAMS = test
+check_PROGRAMS = test
 test_SOURCES = test.c
 test_LDADD = $(top_builddir)/src/libmkv.la
