--- origsrc/Qt-AES-2.1.0/CMakeLists.txt	2026-03-24 07:44:07.000000000 +0900
+++ src/Qt-AES-2.1.0/CMakeLists.txt	2026-07-11 07:04:05.421045600 +0900
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
-project(QtAES VERSION 2.0.0 LANGUAGES CXX)
+project(QtAES VERSION 2.1.0 LANGUAGES CXX)
 
 set(QTAES_STANDALONE_BUILD OFF)
 if(PROJECT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
@@ -50,6 +50,10 @@ find_package(${QT_PACKAGE_NAME} REQUIRED
 
 add_library(QtAES)
 
+if (DEFINED SOVERSION)
+    set_target_properties(QtAES PROPERTIES SOVERSION ${SOVERSION})
+endif()
+
 # Per-target MOC/RCC — avoid polluting parent builds when used as a subdirectory.
 set_target_properties(QtAES PROPERTIES AUTOMOC ON)
 
@@ -245,3 +249,10 @@ if(QTAES_ENABLE_INSTALL)
             "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/QtAES.pc.in ${CMAKE_BINARY_DIR}/QtAES.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/QtAES.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/Qt-AES-2.1.0/QtAES.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/Qt-AES-2.1.0/QtAES.pc.in	2026-07-11 07:04:11.262654100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: QtAES
+Description: Native Qt AES encryption
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lQtAES
+Cflags: -I${includedir}
