diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x libtool.m4 -x ltoptions.m4 -x ltsugar.m4 -x ltversion.m4 -x lt~obsolete.m4 -x ltmain.sh -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x INSTALL -x Makefile.in origsrc/libmikey-0.8.0+r4734/include/libmikey/libmikey_config.h src/libmikey-0.8.0+r4734/include/libmikey/libmikey_config.h
--- origsrc/libmikey-0.8.0+r4734/include/libmikey/libmikey_config.h	2014-06-24 01:03:55.000000000 +0900
+++ src/libmikey-0.8.0+r4734/include/libmikey/libmikey_config.h	2016-03-24 00:02:40.112326900 +0900
@@ -32,7 +32,7 @@
 # endif
 #endif
 
-#if defined(WIN32) && defined(LIBMIKEY_EXPORTS)
+#if (defined(WIN32) && defined(LIBMIKEY_EXPORTS)) || defined(__CYGWIN__)
 # define LIBMIKEY_API __declspec(dllexport)
 #elif defined(WIN32) && defined(LIBMIKEY_IMPORTS)
 # define LIBMIKEY_API __declspec(dllimport)
