--- origsrc/snmp++-3.3.11/src/Makefile.am	2017-12-16 01:11:56.000000000 +0900
+++ src/snmp++-3.3.11/src/Makefile.am	2020-01-06 20:41:13.454523900 +0900
@@ -36,6 +36,6 @@ libsnmp___la_SOURCES =  address.cpp asn1
                         reentrant.cpp sha.cpp snmpmsg.cpp target.cpp \
                         timetick.cpp usm_v3.cpp uxsnmp.cpp v3.cpp vb.cpp
 
-libsnmp___la_LDFLAGS =	-version-info @SNMP_PP_MAJOR_VERSION@@SNMP_PP_MINOR_VERSION@:@SNMP_PP_MICRO_VERSION@:0 @LINKFLAGS@
+libsnmp___la_LDFLAGS =	-no-undefined -version-info @SNMP_PP_MAJOR_VERSION@@SNMP_PP_MINOR_VERSION@:@SNMP_PP_MICRO_VERSION@:0 @LINKFLAGS@
 
-EXTRA_DIST = Makefile.aix5 Makefile.bcc Makefile.dunix Makefile.gcc_solaris Makefile.hpux11 Makefile.linux Makefile.solaris common.mk
\ No newline at end of file
+EXTRA_DIST = Makefile.aix5 Makefile.bcc Makefile.dunix Makefile.gcc_solaris Makefile.hpux11 Makefile.linux Makefile.solaris common.mk
