--- origsrc/ceres-solver-1.13.0/CMakeLists.txt	2017-08-03 16:10:55.000000000 +0900
+++ src/ceres-solver-1.13.0/CMakeLists.txt	2017-09-15 18:55:39.707000000 +0900
@@ -909,3 +909,10 @@ if (EXPORT_BUILD_DIR)
     "${Ceres_BINARY_DIR}/CeresConfig.cmake" @ONLY)
 
 endif (EXPORT_BUILD_DIR)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/ceres-solver.pc.in ${CMAKE_BINARY_DIR}/ceres-solver.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/ceres-solver.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/ceres-solver-1.13.0/ceres-solver.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/ceres-solver-1.13.0/ceres-solver.pc.in	2017-09-15 18:55:34.954000000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: ceres-solver
+Description: Nonlinear Least Squares Minimizer
+Version: @CERES_VERSION@
+Libs: -L${libdir} -lceres
+Cflags: -I${includedir}
--- origsrc/ceres-solver-1.13.0/cmake/FindCXSparse.cmake	2017-08-03 16:10:55.000000000 +0900
+++ src/ceres-solver-1.13.0/cmake/FindCXSparse.cmake	2017-09-15 18:45:19.588619300 +0900
@@ -129,6 +129,7 @@ list(APPEND CXSPARSE_CHECK_INCLUDE_DIRS
   /usr/local/homebrew/include # Mac OS X
   /opt/local/var/macports/software # Mac OS X.
   /opt/local/include
+  /usr/include/suitesparse # cygwin
   /usr/include)
 list(APPEND CXSPARSE_CHECK_LIBRARY_DIRS
   /usr/local/lib
--- origsrc/ceres-solver-1.13.0/internal/ceres/CMakeLists.txt	2017-08-03 16:10:55.000000000 +0900
+++ src/ceres-solver-1.13.0/internal/ceres/CMakeLists.txt	2017-09-15 18:45:19.604301900 +0900
@@ -270,7 +270,7 @@ if (BUILD_TESTING AND GFLAGS)
 
   macro (CERES_TEST NAME)
     add_executable(${NAME}_test ${NAME}_test.cc)
-    target_link_libraries(${NAME}_test test_util ceres gtest)
+    target_link_libraries(${NAME}_test test_util ceres gtest ${CHOLMOD_LIBRARY})
     if (BUILD_SHARED_LIBS)
       # Define gtest-specific shared library flags for linking.
       append_target_property(${NAME}_test COMPILE_DEFINITIONS
