--- origsrc/ud-1.0.1/CMakeLists.txt	2023-07-28 22:05:07.000000000 +0900
+++ src/ud-1.0.1/CMakeLists.txt	2024-03-18 15:38:58.838986000 +0900
@@ -3,6 +3,4 @@ project(ud)
 set(TARGET ud PROPERTY C_STANDARD 18)
 set(SOURCES main.c)
 add_executable(ud ${SOURCES})
-set(CMAKE_C_COMPILER gcc)
-set(CMAKE_C_FLAGS "-O2 -pipe -fPIC -fno-plt -fstack-protector-strong -Wall -Wshadow -Wpedantic -Wno-parentheses -Wvla -Wignored-qualifiers")
 add_definitions(-D_GNU_SOURCE)
