--- origsrc/libmodi-20240507/Makefile.am	2024-05-07 13:25:08.000000000 +0900
+++ src/libmodi-20240507/Makefile.am	2025-05-20 10:09:18.012621100 +0900
@@ -3,29 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcdirectory \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfguid \
-	libfvalue \
-	libfmos \
-	libfplist \
-	libhmac \
-	libcaes \
 	libmodi \
 	moditools \
 	pymodi \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libmodi-20240507/configure.ac	2024-05-07 13:25:08.000000000 +0900
+++ src/libmodi-20240507/configure.ac	2025-05-20 10:09:18.012621100 +0900
@@ -199,25 +199,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcdirectory/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
-AC_CONFIG_FILES([libfmos/Makefile])
-AC_CONFIG_FILES([libfplist/Makefile])
-AC_CONFIG_FILES([libhmac/Makefile])
-AC_CONFIG_FILES([libcaes/Makefile])
 AC_CONFIG_FILES([libmodi/Makefile])
 AC_CONFIG_FILES([pymodi/Makefile])
 AC_CONFIG_FILES([moditools/Makefile])
--- origsrc/libmodi-20240507/pymodi/Makefile.am	2024-05-07 13:25:08.000000000 +0900
+++ src/libmodi-20240507/pymodi/Makefile.am	2025-05-20 10:09:33.834160600 +0900
@@ -40,7 +40,7 @@ pymodi_la_LIBADD = \
 	@LIBBFIO_LIBADD@
 
 pymodi_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pymodi_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pymodi_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
