diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/avlcal-avlcal-1.00.0/config/project_config_embeddable.cmake src/avlcal-avlcal-1.00.0/config/project_config_embeddable.cmake
--- origsrc/avlcal-avlcal-1.00.0/config/project_config_embeddable.cmake	2012-12-12 23:42:39.000000000 +0900
+++ src/avlcal-avlcal-1.00.0/config/project_config_embeddable.cmake	2018-04-18 21:04:39.794000000 +0900
@@ -152,8 +152,11 @@ macro (set_project_options _build_doc _e
     unset (DOC_NEWS_FILE)
   endif (NOT EXISTS ${DOC_NEWS_FILE})
   set (BASICDOC_FILES AUTHORS ${DOC_NEWS_FILE} README ${DOC_INSTALL_FILE})
+  if (CYGWIN)
+    set (BASICDOC_PATH "share/doc/${PACKAGE}")
+  else()
   set (BASICDOC_PATH "share/doc/${PACKAGE}-${PACKAGE_VERSION}")
-
+  endif()
 endmacro (set_project_options)
 
 #
@@ -1196,7 +1199,11 @@ endmacro (module_set_name)
 macro (add_modules)
   set (_embedded_components ${ARGV})
   set (LIB_DEPENDENCY_EXPORT ${PROJECT_NAME}-library-depends)
+  if (CYGWIN)
+    set (LIB_DEPENDENCY_EXPORT_PATH "lib/cmake/${PROJECT_NAME}")
+  else()
   set (LIB_DEPENDENCY_EXPORT_PATH "${INSTALL_DATA_DIR}/${PROJECT_NAME}/CMake")
+  endif()
   #
   foreach (_embedded_comp ${_embedded_components})
     #
@@ -1435,6 +1442,7 @@ macro (module_library_add_specific
   # Installation of the library
   install (TARGETS ${_lib_target}
     EXPORT ${LIB_DEPENDENCY_EXPORT}
+    RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
     LIBRARY DESTINATION "${INSTALL_LIB_DIR}" COMPONENT runtime)
 
   # Register, for reporting purpose, the list of libraries to be built
@@ -1581,8 +1589,13 @@ endmacro (module_script_add)
 # Installation of the CMake import helper, so that third party projects
 # can refer to it (for libraries, header files and binaries)
 macro (module_export_install)
+  if (CYGWIN)
+    install (EXPORT ${LIB_DEPENDENCY_EXPORT}
+      DESTINATION "lib/cmake/${PACKAGE}" COMPONENT devel)
+  else()
   install (EXPORT ${LIB_DEPENDENCY_EXPORT}
     DESTINATION "${INSTALL_DATA_DIR}/${PACKAGE}/CMake" COMPONENT devel)
+  endif()
 endmacro (module_export_install)
 
 
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/avlcal-avlcal-1.00.0/test/avlcal/AvlCalTestSuite.cpp src/avlcal-avlcal-1.00.0/test/avlcal/AvlCalTestSuite.cpp
--- origsrc/avlcal-avlcal-1.00.0/test/avlcal/AvlCalTestSuite.cpp	2012-12-12 23:42:39.000000000 +0900
+++ src/avlcal-avlcal-1.00.0/test/avlcal/AvlCalTestSuite.cpp	2018-04-18 21:01:18.616000000 +0900
@@ -37,7 +37,7 @@ struct UnitTestConfig {
   /** Constructor. */
   UnitTestConfig() {
     boost_utf::unit_test_log.set_stream (utfReportStream);
-    boost_utf::unit_test_log.set_format (boost_utf::XML);
+    boost_utf::unit_test_log.set_format (boost_utf::OF_XML);
     boost_utf::unit_test_log.set_threshold_level (boost_utf::log_test_units);
     //boost_utf::unit_test_log.set_threshold_level (boost_utf::log_successful_tests);
   }
