--- origsrc/avlcal-1.00.0/config/project_config_embeddable.cmake	2012-12-26 02:16:29.000000000 +0900
+++ src/avlcal-1.00.0/config/project_config_embeddable.cmake	2014-06-15 22:16:19.676575800 +0900
@@ -1475,6 +1475,7 @@ macro (module_library_add_specific
   # Installation of the library
   install (TARGETS ${_lib_target}
     EXPORT ${LIB_DEPENDENCY_EXPORT}
+    RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
     LIBRARY DESTINATION "${INSTALL_LIB_DIR}" COMPONENT runtime)
 
   # Register, for reporting purpose, the list of libraries to be built
