--- origsrc/avlcal-1.00.8/avlcal-config.cmake.in	2026-01-19 05:44:26.000000000 +0900
+++ src/avlcal-1.00.8/avlcal-config.cmake.in	2026-07-14 07:46:23.200992800 +0900
@@ -16,5 +16,5 @@ set (AVLCAL_BINARY_DIRS "@AVLCAL_BIN_DIR
 include ("@AVLCAL_CMAKE_DIR@/avlcal-library-depends.cmake")
 
 # These are the AvlCal IMPORTED targets, created by avlcal-library-depends.cmake
-set (AVLCAL_LIBRARIES airraclib avlcallib)
+set (AVLCAL_LIBRARIES avlcal)
 set (AVLCAL_EXECUTABLES avlcal)
