--- origsrc/ALeS-1.2/ALeS.cpp	2020-09-27 15:07:39.000000000 +0900
+++ src/ALeS-1.2/ALeS.cpp	2020-12-07 09:17:41.529644600 +0900
@@ -15,6 +15,8 @@
 #elif defined(__linux__) || defined(__linux) || defined(linux) || defined(__gnu_linux__)
 	#include <omp.h>
 	#include <sys/sysinfo.h>
+#elif defined(__CYGWIN__)
+	#include <sys/sysinfo.h>
 #endif
 
 
@@ -1323,7 +1325,7 @@ void ALeS(char** S){
 	//1.5 times the size is used for safety
 	
 	//if linux system detected
-	#if defined(__linux__) || defined(__linux) || defined(linux) || defined(__gnu_linux__)
+	#if defined(__linux__) || defined(__linux) || defined(linux) || defined(__gnu_linux__) || defined(__CYGWIN__)
         struct sysinfo si;
         sysinfo (&si);
         totalVirtualMem = (si.totalram + si.totalswap);
