--- origsrc/ALeS-v_1.5/ALeS.cpp	2021-04-29 11:51:48.000000000 +0900
+++ src/ALeS-v_1.5/ALeS.cpp	2025-05-31 23:28:24.164294600 +0900
@@ -15,6 +15,8 @@
 #elif defined(__linux__) || defined(__linux) || defined(linux) || defined(__gnu_linux__)
 	#include <omp.h>
 	#include <sys/sysinfo.h>
+#elif defined(__CYGWIN__)
+	#include <sys/sysinfo.h>
 #endif
 
 
@@ -1336,7 +1338,7 @@ void ALeS(char** S){
 	//1.5 times the size is used for safety
 	
 	//if linux system detected
-	#if defined(__linux__) || defined(__linux) || defined(linux) || defined(__gnu_linux__)
+	#if defined(__linux__) || defined(__linux) || defined(linux) || defined(__gnu_linux__) || defined(__CYGWIN__)
         struct sysinfo si;
         sysinfo (&si);
         totalVirtualMem = (si.totalram + si.totalswap);
