--- origsrc/libskeleton/Makefile.am	2015-02-18 21:48:59.000000000 +0900
+++ src/libskeleton/Makefile.am	2015-02-18 21:49:01.044545600 +0900
@@ -30,6 +30,8 @@
 
 # Process this file with automake to produce Makefile.in
 
+ACLOCAL_AMFLAGS = -I m4
+
 AUTOMAKE_OPTIONS = foreign
 
 DISTCHECK_CONFIGURE_FLAGS = --enable-gcc-werror
--- origsrc/libskeleton/src/Makefile.am	2015-02-18 21:48:59.000000000 +0900
+++ src/libskeleton/src/Makefile.am	2015-02-18 21:49:01.044545600 +0900
@@ -20,5 +20,5 @@ libskeleton_la_SOURCES =     		\
 	skeleton_vector.c		
 	
 libskeleton_la_CFLAGS = $(AM_CFLAGS) $(OGG_CFLAGS) $(CFLAGS)
-libskeleton_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@ 
+libskeleton_la_LDFLAGS = -no-undefined -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@ 
 libskeleton_la_LIBADD = @OGG_LIBS@
