--- origsrc/restclient-cpp-0.5.3/Makefile.am	2025-01-03 01:30:56.000000000 +0900
+++ src/restclient-cpp-0.5.3/Makefile.am	2025-03-31 09:16:27.571313000 +0900
@@ -1,25 +1,25 @@
 ACLOCAL_AMFLAGS=-I m4
-CPPFLAGS=-I${top_srcdir}/include
+CPPFLAGS = -I$(top_builddir)/include -I${top_srcdir}/include
 check_PROGRAMS = test-program
 pkginclude_HEADERS = include/restclient-cpp/restclient.h include/restclient-cpp/version.h include/restclient-cpp/connection.h include/restclient-cpp/helpers.h
 BUILT_SOURCES = include/restclient-cpp/version.h
 
-test_program_SOURCES = vendor/jsoncpp-0.10.5/dist/jsoncpp.cpp test/tests.cpp test/test_helpers.cc test/test_restclient.cc test/test_connection.cc
-test_program_LDADD = .libs/librestclient-cpp.a
-test_program_LDFLAGS=-Lvendor/googletest-1.14.0/lib -lgtest
-test_program_CPPFLAGS=-std=c++14 -Iinclude -Ivendor/googletest-1.14.0/googletest/include -Ivendor/jsoncpp-0.10.5/dist
+test_program_SOURCES = test/tests.cpp test/test_helpers.cc test/test_restclient.cc test/test_connection.cc
+test_program_LDADD = librestclient-cpp.la
+test_program_LDFLAGS = -lgtest -ljsoncpp
+test_program_CPPFLAGS = -Iinclude
 
 lib_LTLIBRARIES=librestclient-cpp.la
 librestclient_cpp_la_SOURCES=source/restclient.cc source/connection.cc source/helpers.cc
-librestclient_cpp_la_CXXFLAGS=-fPIC -std=c++14
-librestclient_cpp_la_LDFLAGS=-version-info 2:1:1
+librestclient_cpp_la_CXXFLAGS =
+librestclient_cpp_la_LDFLAGS = -no-undefined -version-info 2:1:1
 
 dist_doc_DATA = README.md
 
 .PHONY: test check clean-coverage-files coverage-html include/restclient-cpp/version.h lint ci docker-services clean-docker-services
 
 include/restclient-cpp/version.h:
-	m4 -I ${top_srcdir}/m4 -DM4_RESTCLIENT_VERSION=$(PACKAGE_VERSION) version.h.m4 > ${top_srcdir}/$@
+	mkdir -p include/restclient-cpp && m4 -I ${top_srcdir}/m4 -DM4_RESTCLIENT_VERSION=$(PACKAGE_VERSION) version.h.m4 > ${top_srcdir}/$@
 
 
 
@@ -60,3 +60,5 @@ coverage-html: coverage.info
 include $(srcdir)/Makefile.packaging.mk
 include $(srcdir)/Makefile.docs.mk
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/restclient-cpp-0.5.3/restclient-cpp.pc.in	2025-01-03 01:30:56.000000000 +0900
+++ src/restclient-cpp-0.5.3/restclient-cpp.pc.in	2025-03-31 09:12:47.830255200 +0900
@@ -7,6 +7,6 @@ Name: restclient-cpp
 Description: C++ client for making HTTP/REST requests
 URL: http://code.mrtazz.com/restclient-cpp/
 Version: @VERSION@
-Requires: curl
+Requires: libcurl
 Libs: -L${libdir} -lrestclient-cpp
 Cflags: -I${includedir}
