--- origsrc/iml-1.0.3/configure.ac	2012-05-02 11:50:52.609683300 +0900
+++ src/iml-1.0.3/configure.ac	2012-05-02 12:14:32.766909800 +0900
@@ -56,16 +56,7 @@ location with the options --with-gmp-inc
 respectively when running configure.
 )])
 
-AC_CHECK_HEADERS([cblas.h],
-	[cblas_headers_found=yes; break;])
-
-AS_IF([test "x$cblas_headers_found" != "xyes"],
-	[AC_MSG_ERROR([Unable to find the cblas headers check your installation])])
-
-AC_ARG_WITH(cblas-lib,
-            AC_HELP_STRING([--with-cblas-lib=<lib>],[Specify the flags for cblas.]),
-            with_cblas_lib=$withval)
-ATLAS_LIBS="$with_cblas_lib"
+PKG_CHECK_MODULES([ATLAS], [blas])
 
 AC_SUBST(ATLAS_LIBS)
 AC_SUBST(LDFLAGS)
--- origsrc/iml-1.0.3/src/Makefile.am	2008-07-28 23:53:07.000000000 +0900
+++ src/iml-1.0.3/src/Makefile.am	2012-05-02 12:12:23.558902000 +0900
@@ -32,4 +32,4 @@ libiml_la_SOURCES = basisop.c \
 libiml_la_CFLAGS = $(AM_CFLAGS)
 libiml_la_LIBADD = $(EXTERNLIB) \
 	 	   $(top_builddir)/repl/librepl.la
-libiml_la_LDFLAGS = -version-info 1:0:1
+libiml_la_LDFLAGS = -version-info 1:0:1 -no-undefined
--- origsrc/iml-1.0.3/src/nullspace.c	2008-06-16 07:26:25.000000000 +0900
+++ src/iml-1.0.3/src/nullspace.c	2012-05-02 12:08:17.965152000 +0900
@@ -253,7 +253,7 @@ nullspaceMP(const long n, const long m,
 	    for (i = 0; i < m; i++) {
 		for (j = 0; j < m; j++)
 		    mpz_init_set_ui(mp_N[i * m + j], 0);
-		mpz_init_ui(mp_N[i * m + i], 1);
+		mpz_init_set_ui(mp_N[i * m + i], 1);
 	    }
 	    *mp_N_pass = mp_N;
 	} else {		/* r>0 and s>0 */
