--- origsrc/konforka-0.0/configure.ac	2005-01-22 23:09:39.000000000 +0900
+++ src/konforka-0.0/configure.ac	2010-11-28 18:50:26.879259200 +0900
@@ -7,7 +7,7 @@ AC_PROG_INSTALL
 AC_PROG_AWK
 AC_PROG_CXX
 AC_PROG_CC
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 AC_HEADER_STDC
 
--- origsrc/konforka-0.0/lib/Makefile.am	2005-01-22 05:12:20.000000000 +0900
+++ src/konforka-0.0/lib/Makefile.am	2010-11-28 18:50:07.133744300 +0900
@@ -14,3 +14,6 @@ libkonforka_la_SOURCES = \
 	exception.cc \
 	pointer_map.cc \
 	${EXTRA_SOURCES}
+
+libkonforka_la_LDFLAGS = -no-undefined
+
