--- origsrc/konforka/Makefile.am	2015-05-17 20:00:48.000000000 +0900
+++ src/konforka/Makefile.am	2015-05-17 20:00:49.994254400 +0900
@@ -14,7 +14,7 @@ clean-local:
 endif
 
 NEWS: NEWS.xsl NEWS.xml
-	${XSLTPROC} -o $@ NEWS.xsl NEWS.xml
+	${XSLTPROC} -o $@ $(top_srcdir)/NEWS.xsl $(top_srcdir)/NEWS.xml
 
 if HAVE_DOXYGEN
 dox:
--- origsrc/konforka/lib/Makefile.am	2015-05-17 20:00:48.000000000 +0900
+++ src/konforka/lib/Makefile.am	2015-05-17 20:04:54.416255500 +0900
@@ -2,7 +2,6 @@ lib_LTLIBRARIES = libkonforka.la
 
 INCLUDES = -I${top_srcdir}/include
 AM_CXXFLAGS = ${PQXX_CFLAGS}
-LDADD = ${PQXX_LIBS}
 
 EXTRA_DIST = pqxx_pile.cc
 EXTRA_SOURCES=
@@ -15,3 +14,6 @@ libkonforka_la_SOURCES = \
 	util.cc \
 	pointer_map.cc \
 	${EXTRA_SOURCES}
+
+libkonforka_la_LDFLAGS = -no-undefined
+libkonforka_la_LIBADD = ${PQXX_LIBS}
--- origsrc/konforka/lib/exception.cc	2015-05-17 20:00:48.000000000 +0900
+++ src/konforka/lib/exception.cc	2015-05-17 20:00:49.994254400 +0900
@@ -1,6 +1,7 @@
 #include <string.h>
 #include <errno.h>
 #include "konforka/exception.h"
+#include <cstdio>
 
 namespace konforka {
 
