--- origsrc/cgi-util-2.2.1/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/cgi-util-2.2.1/Makefile.am	2015-04-20 17:20:29.574567300 +0900
@@ -0,0 +1,9 @@
+
+include_HEADERS = cgi-util.h
+
+lib_LTLIBRARIES = libcgi-util.la
+libcgi_util_la_SOURCES = cgi-util.c
+libcgi_util_la_LDFLAGS = -no-undefined -version-info 2
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/cgi-util-2.2.1/cgi-util.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/cgi-util-2.2.1/cgi-util.pc.in	2015-04-20 17:22:06.336153100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: cgi-util
+Description: C library for creating CGI programs
+Version: @VERSION@
+Libs: -L${libdir} -lcgi-utils
+Cflags: -I${includedir}
--- origsrc/cgi-util-2.2.1/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/cgi-util-2.2.1/configure.ac	2015-04-20 17:32:33.104101400 +0900
@@ -0,0 +1,30 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([cgi-util], [2.2.1])
+AC_CONFIG_SRCDIR([cgi-util.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_MAKE_SET
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h string.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([memset strchr strerror strstr])
+
+AC_CONFIG_FILES([Makefile cgi-util.pc])
+AC_OUTPUT
