--- origsrc/cptutils-1.72/configure.in	2019-08-08 05:30:12.000000000 +0900
+++ src/cptutils-1.72/configure.in	2019-09-06 21:08:56.906881500 +0900
@@ -68,7 +68,7 @@ AC_CHECK_LIB(m, pow)
 AC_CHECK_LIB(png, png_create_read_struct, [], AC_MSG_ERROR(libpng not found))
 AC_CHECK_LIB(jansson, json_pack, [], AC_MSG_WARN(libjansson not found))
 
-AC_SEARCH_LIBS(iconv_open, iconv, [], AC_MSG_ERROR(iconv not found))
+AC_SEARCH_LIBS(libiconv_open, iconv, [], AC_MSG_ERROR(iconv not found))
 
 dnl Checks for header files.
 
