--- origsrc/cptutils-1.83/configure.ac	2024-11-08 05:27:54.000000000 +0900
+++ src/cptutils-1.83/configure.ac	2025-01-16 13:44:21.699749700 +0900
@@ -96,7 +96,7 @@ dnl | required libraries
 
 AC_CHECK_LIB(m, pow)
 AC_CHECK_LIB(png, png_create_read_struct, [], AC_MSG_ERROR(libpng not found))
-AC_SEARCH_LIBS(iconv_open, iconv, [], AC_MSG_ERROR(iconv not found))
+AC_SEARCH_LIBS(iconv_open, iconv, [], AC_SEARCH_LIBS(libiconv_open, iconv, [], AC_MSG_ERROR(iconv not found)))
 
 dnl | header files
 
