--- origsrc/cptutils-1.47/configure.in	2013-07-09 06:25:12.000000000 +0900
+++ src/cptutils-1.47/configure.in	2013-07-28 17:27:54.706678900 +0900
@@ -47,7 +47,7 @@ dnl Checks for libraries.
 
 AC_CHECK_LIB(m, pow)
 AC_CHECK_LIB(png, png_create_read_struct, [], AC_MSG_ERROR(libpng not found))
-AC_SEARCH_LIBS(iconv, iconv, [], AC_MSG_ERROR(iconv not found))
+AC_SEARCH_LIBS(libiconv, iconv, [], AC_MSG_ERROR(iconv not found))
 
 dnl Checks for header files.
 
--- origsrc/cptutils-1.47/src/Common.mk.in	2013-02-28 07:08:41.000000000 +0900
+++ src/cptutils-1.47/src/Common.mk.in	2013-07-28 17:37:19.154921000 +0900
@@ -51,7 +51,7 @@ endif
 
 ifneq (no,$(XSLTP))
 
-MANXSL = /usr/share/xml/docbook/stylesheet/docbook-xsl/manpages/docbook.xsl
+MANXSL = /usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl
 
 %.1 : %.xml ../docbook
 	$(XSLTP) --xinclude $(MANXSL) $<
