--- origsrc/libpff-20231205/Makefile.am	2023-12-05 15:34:03.000000000 +0900
+++ src/libpff-20231205/Makefile.am	2025-05-18 16:14:07.654469000 +0900
@@ -3,28 +3,10 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
-	libfguid \
-	libfvalue \
-	libfwnt \
-	libfmapi \
 	libpff \
 	pfftools \
 	jpff \
 	pypff \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libpff-20231205/configure.ac	2023-12-05 15:34:03.000000000 +0900
+++ src/libpff-20231205/configure.ac	2025-05-18 16:14:07.654469000 +0900
@@ -179,23 +179,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
-AC_CONFIG_FILES([libfwnt/Makefile])
-AC_CONFIG_FILES([libfmapi/Makefile])
 AC_CONFIG_FILES([libpff/Makefile])
 AC_CONFIG_FILES([pypff/Makefile])
 AC_CONFIG_FILES([jpff/Makefile])
--- origsrc/libpff-20231205/jpff/Makefile.am	2023-12-05 15:34:03.000000000 +0900
+++ src/libpff-20231205/jpff/Makefile.am	2025-05-18 16:14:07.654469000 +0900
@@ -8,6 +8,7 @@ JAVA_FILES = \
 
 if HAVE_JAVA
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@
@@ -38,7 +39,7 @@ libjpff_la_LIBADD = \
 	../libpff/libpff.la
 
 libjpff_la_CPPFLAGS = $(JAVA_CPPFLAGS)
-libjpff_la_LDFLAGS  = -module -avoid-version $(JAVA_LDFLAGS)
+libjpff_la_LDFLAGS  = -no-undefined -module -avoid-version $(JAVA_LDFLAGS)
 
 endif
 
--- origsrc/libpff-20231205/libpff/Makefile.am	2023-12-05 15:34:03.000000000 +0900
+++ src/libpff-20231205/libpff/Makefile.am	2025-05-18 16:14:07.654469000 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libpff-20231205/libpff/libpff_name_to_id_map.c	2023-12-05 15:34:03.000000000 +0900
+++ src/libpff-20231205/libpff/libpff_name_to_id_map.c	2025-05-18 16:14:07.654469000 +0900
@@ -739,11 +739,13 @@ int libpff_name_to_id_map_entry_read(
 	}
 	else if( name_to_id_map_entry_type == 5 )
 	{
+#if !defined(__CYGWIN__)
 		if( memory_copy(
 		     internal_name_to_id_map_entry->guid,
 		     libfmapi_class_identifier_public_strings,
 		     16 ) == NULL )
 		{
+#endif
 			libcerror_error_set(
 			 error,
 			 LIBCERROR_ERROR_DOMAIN_MEMORY,
@@ -752,7 +754,9 @@ int libpff_name_to_id_map_entry_read(
 			 function );
 
 			 goto on_error;
+#if !defined(__CYGWIN__)
 		}
+#endif
 	}
 #if defined( HAVE_DEBUG_OUTPUT )
 	if( libcnotify_verbose != 0 )
--- origsrc/libpff-20231205/pfftools/Makefile.am	2023-12-05 15:34:03.000000000 +0900
+++ src/libpff-20231205/pfftools/Makefile.am	2025-05-18 16:14:07.654469000 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libpff-20231205/pfftools/export_handle.c	2023-12-05 15:34:03.000000000 +0900
+++ src/libpff-20231205/pfftools/export_handle.c	2025-05-18 16:14:07.669954000 +0900
@@ -1594,6 +1594,7 @@ int export_handle_export_item(
 	}
 	if( result == -1 )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
@@ -1603,6 +1604,7 @@ int export_handle_export_item(
 			 item_index + 1,
 			 number_of_items );
 		}
+#endif
 		if( ( error != NULL )
 		 && ( *error != NULL ) )
 		{
@@ -1654,6 +1656,7 @@ int export_handle_export_item(
 			     log_handle,
 			     error ) != 1 )
 			{
+#if defined( HAVE_DEBUG_OUTPUT )
 				if( libcnotify_verbose != 0 )
 				{
 					libcnotify_printf(
@@ -1662,6 +1665,7 @@ int export_handle_export_item(
 					 item_index + 1,
 					 number_of_items );
 				}
+#endif
 				if( ( error != NULL )
 				 && ( *error != NULL ) )
 				{
@@ -1772,6 +1776,7 @@ int export_handle_export_item(
 			 number_of_items,
 			 entry_value_string );
 
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
@@ -1782,6 +1787,7 @@ int export_handle_export_item(
 				 entry_value_string,
 				 item_type );
 			}
+#endif
 			log_handle_printf(
 			 log_handle,
 			 "Skipped export of item: %d out of: %d of type: %" PRIs_SYSTEM " (0x%02" PRIx8 ").\n",
@@ -2704,6 +2710,7 @@ int export_handle_export_sub_items(
 			 sub_item_iterator + 1,
 			 number_of_sub_items );
 
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
@@ -2711,6 +2718,7 @@ int export_handle_export_sub_items(
 				 function,
 				 sub_item_iterator + 1 );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -7115,12 +7123,14 @@ int export_handle_export_attachments(
 				     log_handle,
 				     error ) != 1 )
 				{
+#if defined( HAVE_DEBUG_OUTPUT )
 					if( libcnotify_verbose != 0 )
 					{
 						libcnotify_printf(
 						 "%s: unable to export item values.\n",
 						 function );
 					}
+#endif
 					if( ( error != NULL )
 					 && ( *error != NULL ) )
 					{
@@ -7192,6 +7202,7 @@ int export_handle_export_attachments(
 			     log_handle,
 			     error ) != 1 )
 			{
+#if defined( HAVE_DEBUG_OUTPUT )
 				if( libcnotify_verbose != 0 )
 				{
 					libcnotify_printf(
@@ -7199,6 +7210,7 @@ int export_handle_export_attachments(
 					 function,
 					 attachment_index + 1 );
 				}
+#endif
 				if( ( error != NULL )
 				 && ( *error != NULL ) )
 				{
@@ -7290,12 +7302,14 @@ int export_handle_export_attachment(
 		 export_handle->notify_stream,
 		 "Unsupported attachment type.\n" );
 
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
 			 "%s: unsupported attachment type.\n",
 			 function );
 		}
+#endif
 		log_handle_printf(
 		 log_handle,
 		 "Unsupported attachment type.\n" );
@@ -8248,12 +8262,14 @@ int export_handle_export_recipients(
 				     log_handle,
 				     error ) != 1 )
 				{
+#if defined( HAVE_DEBUG_OUTPUT )
 					if( libcnotify_verbose != 0 )
 					{
 						libcnotify_printf(
 						 "%s: unable to export item values.\n",
 						 function );
 					}
+#endif
 					if( ( error != NULL )
 					 && ( *error != NULL ) )
 					{
@@ -8710,6 +8726,7 @@ int export_handle_export_activity(
 
 	if( export_handle->export_mode == EXPORT_MODE_DEBUG )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
@@ -8717,6 +8734,7 @@ int export_handle_export_activity(
 			 function,
 			 identifier );
 		}
+#endif
 	}
 	if( export_handle_create_default_item_directory(
 	     export_handle,
@@ -8762,12 +8780,14 @@ int export_handle_export_activity(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export item values.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -8890,12 +8910,14 @@ int export_handle_export_activity(
 		     item_file,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export plain text message body.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -8944,12 +8966,14 @@ int export_handle_export_activity(
 	     log_handle,
 	     error ) != 1 )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
 			 "%s: unable to export message body.\n",
 			 function );
 		}
+#endif
 		if( ( error != NULL )
 		 && ( *error != NULL ) )
 		{
@@ -8971,12 +8995,14 @@ int export_handle_export_activity(
 	     log_handle,
 	     error ) != 1 )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
 			 "%s: unable to export recipients.\n",
 			 function );
 		}
+#endif
 		if( ( error != NULL )
 		 && ( *error != NULL ) )
 		{
@@ -9000,12 +9026,14 @@ int export_handle_export_activity(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export attachments.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -9116,6 +9144,7 @@ int export_handle_export_appointment(
 
 	if( export_handle->export_mode == EXPORT_MODE_DEBUG )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
@@ -9123,6 +9152,7 @@ int export_handle_export_appointment(
 			 function,
 			 identifier );
 		}
+#endif
 	}
 	if( export_handle_create_default_item_directory(
 	     export_handle,
@@ -9157,12 +9187,14 @@ int export_handle_export_appointment(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export item values.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -9276,12 +9308,14 @@ int export_handle_export_appointment(
 	     log_handle,
 	     error ) != 1 )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
 			 "%s: unable to export recipients.\n",
 			 function );
 		}
+#endif
 		if( ( error != NULL )
 		 && ( *error != NULL ) )
 		{
@@ -9305,12 +9339,14 @@ int export_handle_export_appointment(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export attachments.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -9449,6 +9485,7 @@ int export_handle_export_contact(
 
 	if( export_handle->export_mode == EXPORT_MODE_DEBUG )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
@@ -9456,6 +9493,7 @@ int export_handle_export_contact(
 			 function,
 			 identifier );
 		}
+#endif
 	}
 	if( export_handle_create_default_item_directory(
 	     export_handle,
@@ -9490,12 +9528,14 @@ int export_handle_export_contact(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export item values.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -9621,12 +9661,14 @@ int export_handle_export_contact(
 		     item_file,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export plain text message body.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -9675,12 +9717,14 @@ int export_handle_export_contact(
 	     log_handle,
 	     error ) != 1 )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
 			 "%s: unable to export message body.\n",
 			 function );
 		}
+#endif
 		if( ( error != NULL )
 		 && ( *error != NULL ) )
 		{
@@ -9702,12 +9746,14 @@ int export_handle_export_contact(
 	     log_handle,
 	     error ) != 1 )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
 			 "%s: unable to export recipients.\n",
 			 function );
 		}
+#endif
 		if( ( error != NULL )
 		 && ( *error != NULL ) )
 		{
@@ -9731,12 +9777,14 @@ int export_handle_export_contact(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export attachments.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -9861,6 +9909,7 @@ int export_handle_export_distribution_li
 
 	if( export_handle->export_mode == EXPORT_MODE_DEBUG )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
@@ -9868,6 +9917,7 @@ int export_handle_export_distribution_li
 			 function,
 			 identifier );
 		}
+#endif
 	}
 	if( export_handle_create_default_item_directory(
 	     export_handle,
@@ -9913,12 +9963,14 @@ int export_handle_export_distribution_li
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export item values.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -10294,12 +10346,14 @@ int export_handle_export_distribution_li
 		     item_file,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export plain text message body.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -10348,12 +10402,14 @@ int export_handle_export_distribution_li
 	     log_handle,
 	     error ) != 1 )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
 			 "%s: unable to export message body.\n",
 			 function );
 		}
+#endif
 		if( ( error != NULL )
 		 && ( *error != NULL ) )
 		{
@@ -10486,6 +10542,7 @@ int export_handle_export_document(
 
 	if( export_handle->export_mode == EXPORT_MODE_DEBUG )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
@@ -10493,6 +10550,7 @@ int export_handle_export_document(
 			 function,
 			 identifier );
 		}
+#endif
 	}
 	if( export_handle_create_default_item_directory(
 	     export_handle,
@@ -10538,12 +10596,14 @@ int export_handle_export_document(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export item values.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -10639,12 +10699,14 @@ int export_handle_export_document(
 	     log_handle,
 	     error ) != 1 )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
 			 "%s: unable to export recipients.\n",
 			 function );
 		}
+#endif
 		if( ( error != NULL )
 		 && ( *error != NULL ) )
 		{
@@ -10668,12 +10730,14 @@ int export_handle_export_document(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export attachments.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -10780,6 +10844,7 @@ int export_handle_export_email(
 
 	if( export_handle->export_mode == EXPORT_MODE_DEBUG )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
@@ -10787,6 +10852,7 @@ int export_handle_export_email(
 			 function,
 			 identifier );
 		}
+#endif
 	}
 	if( export_handle_create_default_item_directory(
 	     export_handle,
@@ -10832,12 +10898,14 @@ int export_handle_export_email(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export item values.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -10978,12 +11046,14 @@ int export_handle_export_email(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export message header.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -11005,12 +11075,14 @@ int export_handle_export_email(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export message conversation index.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -11108,12 +11180,14 @@ int export_handle_export_email(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export recipients.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -11174,12 +11248,14 @@ int export_handle_export_email(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export attachments.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -11406,12 +11482,14 @@ int export_handle_export_email_ftk(
 			     item_file,
 			     error ) != 1 )
 			{
+#if defined( HAVE_DEBUG_OUTPUT )
 				if( libcnotify_verbose != 0 )
 				{
 					libcnotify_printf(
 					 "%s: unable to export recipients.\n",
 					 function );
 				}
+#endif
 				if( ( error != NULL )
 				 && ( *error != NULL ) )
 				{
@@ -11629,6 +11707,7 @@ int export_handle_export_meeting(
 
 	if( export_handle->export_mode == EXPORT_MODE_DEBUG )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
@@ -11636,6 +11715,7 @@ int export_handle_export_meeting(
 			 function,
 			 identifier );
 		}
+#endif
 	}
 	if( export_handle_create_default_item_directory(
 	     export_handle,
@@ -11681,12 +11761,14 @@ int export_handle_export_meeting(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export item values.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -11809,12 +11891,14 @@ int export_handle_export_meeting(
 		     item_file,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export plain text message body.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -11863,12 +11947,14 @@ int export_handle_export_meeting(
 	     log_handle,
 	     error ) != 1 )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
 			 "%s: unable to export message body.\n",
 			 function );
 		}
+#endif
 		if( ( error != NULL )
 		 && ( *error != NULL ) )
 		{
@@ -11981,6 +12067,7 @@ int export_handle_export_note(
 
 	if( export_handle->export_mode == EXPORT_MODE_DEBUG )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
@@ -11988,6 +12075,7 @@ int export_handle_export_note(
 			 function,
 			 identifier );
 		}
+#endif
 	}
 	if( export_handle_create_default_item_directory(
 	     export_handle,
@@ -12033,12 +12121,14 @@ int export_handle_export_note(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export item values.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -12161,12 +12251,14 @@ int export_handle_export_note(
 		     item_file,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export plain text message body.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -12215,12 +12307,14 @@ int export_handle_export_note(
 	     log_handle,
 	     error ) != 1 )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
 			 "%s: unable to export message body.\n",
 			 function );
 		}
+#endif
 		if( ( error != NULL )
 		 && ( *error != NULL ) )
 		{
@@ -12242,12 +12336,14 @@ int export_handle_export_note(
 	     log_handle,
 	     error ) != 1 )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
 			 "%s: unable to export recipients.\n",
 			 function );
 		}
+#endif
 		if( ( error != NULL )
 		 && ( *error != NULL ) )
 		{
@@ -12273,12 +12369,14 @@ int export_handle_export_note(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export attachments.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -12390,6 +12488,7 @@ int export_handle_export_rss_feed(
 
 	if( export_handle->export_mode == EXPORT_MODE_DEBUG )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
@@ -12397,6 +12496,7 @@ int export_handle_export_rss_feed(
 			 function,
 			 identifier );
 		}
+#endif
 	}
 	if( export_handle_create_default_item_directory(
 	     export_handle,
@@ -12442,12 +12542,14 @@ int export_handle_export_rss_feed(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export item values.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -12570,12 +12672,14 @@ int export_handle_export_rss_feed(
 		     item_file,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export plain text message body.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -12624,12 +12728,14 @@ int export_handle_export_rss_feed(
 	     log_handle,
 	     error ) != 1 )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
 			 "%s: unable to export message body.\n",
 			 function );
 		}
+#endif
 		if( ( error != NULL )
 		 && ( *error != NULL ) )
 		{
@@ -12651,12 +12757,14 @@ int export_handle_export_rss_feed(
 	     log_handle,
 	     error ) != 1 )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
 			 "%s: unable to export recipients.\n",
 			 function );
 		}
+#endif
 		if( ( error != NULL )
 		 && ( *error != NULL ) )
 		{
@@ -12680,12 +12788,14 @@ int export_handle_export_rss_feed(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export attachments.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -12808,6 +12918,7 @@ int export_handle_export_task(
 
 	if( export_handle->export_mode == EXPORT_MODE_DEBUG )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
@@ -12815,6 +12926,7 @@ int export_handle_export_task(
 			 function,
 			 identifier );
 		}
+#endif
 	}
 	if( export_handle_create_default_item_directory(
 	     export_handle,
@@ -12860,12 +12972,14 @@ int export_handle_export_task(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export item values.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -12991,12 +13105,14 @@ int export_handle_export_task(
 		     item_file,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export plain text message body.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -13045,12 +13161,14 @@ int export_handle_export_task(
 	     log_handle,
 	     error ) != 1 )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
 			 "%s: unable to export message body.\n",
 			 function );
 		}
+#endif
 		if( ( error != NULL )
 		 && ( *error != NULL ) )
 		{
@@ -13072,12 +13190,14 @@ int export_handle_export_task(
 	     log_handle,
 	     error ) != 1 )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
 			 "%s: unable to export recipients.\n",
 			 function );
 		}
+#endif
 		if( ( error != NULL )
 		 && ( *error != NULL ) )
 		{
@@ -13101,12 +13221,14 @@ int export_handle_export_task(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export attachments.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -13422,6 +13544,7 @@ int export_handle_export_folder(
 
 	if( export_handle->export_mode == EXPORT_MODE_DEBUG )
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
@@ -13429,6 +13552,7 @@ int export_handle_export_folder(
 			 function,
 			 identifier );
 		}
+#endif
 	}
 	/* Create the folder directory
 	 */
@@ -13608,12 +13732,14 @@ int export_handle_export_folder(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export item values.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -13840,6 +13966,7 @@ int export_handle_export_sub_folders(
 			 sub_folder_iterator + 1,
 			 number_of_sub_folders );
 
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
@@ -13847,6 +13974,7 @@ int export_handle_export_sub_folders(
 				 function,
 				 sub_folder_iterator + 1 );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -13983,6 +14111,7 @@ int export_handle_export_sub_messages(
 			 sub_message_iterator + 1,
 			 number_of_sub_messages );
 
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
@@ -13990,6 +14119,7 @@ int export_handle_export_sub_messages(
 				 function,
 				 sub_message_iterator + 1 );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -14448,6 +14578,7 @@ int export_handle_export_orphan_items(
 				 orphan_item_iterator + 1,
 				 export_handle->number_of_orphan_items );
 
+#if defined( HAVE_DEBUG_OUTPUT )
 				if( libcnotify_verbose != 0 )
 				{
 					libcnotify_printf(
@@ -14455,6 +14586,7 @@ int export_handle_export_orphan_items(
 					 function,
 					 orphan_item_iterator + 1 );
 				}
+#endif
 				if( ( error != NULL )
 				 && ( *error != NULL ) )
 				{
@@ -14672,6 +14804,7 @@ int export_handle_export_recovered_items
 				 recovered_item_iterator + 1,
 				 export_handle->number_of_recovered_items );
 
+#if defined( HAVE_DEBUG_OUTPUT )
 				if( libcnotify_verbose != 0 )
 				{
 					libcnotify_printf(
@@ -14679,6 +14812,7 @@ int export_handle_export_recovered_items
 					 function,
 					 recovered_item_iterator + 1 );
 				}
+#endif
 				if( ( error != NULL )
 				 && ( *error != NULL ) )
 				{
--- origsrc/libpff-20231205/pypff/Makefile.am	2023-12-05 15:34:03.000000000 +0900
+++ src/libpff-20231205/pypff/Makefile.am	2025-05-18 16:14:07.669954000 +0900
@@ -1,5 +1,6 @@
 if HAVE_PYTHON
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -53,7 +54,7 @@ pypff_la_LIBADD = \
 	@LIBBFIO_LIBADD@
 
 pypff_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pypff_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pypff_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
--- origsrc/libpff-20231205/tests/Makefile.am	2023-12-05 15:34:03.000000000 +0900
+++ src/libpff-20231205/tests/Makefile.am	2025-05-18 16:14:07.669954000 +0900
@@ -1,6 +1,7 @@
 AUTOMAKE_OPTIONS = subdir-objects
 
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
