--- origsrc/simcrs-1.01.14/config/project_config_embeddable.cmake	2026-01-19 03:26:07.000000000 +0900
+++ src/simcrs-1.01.14/config/project_config_embeddable.cmake	2026-07-14 14:23:34.266640500 +0900
@@ -443,7 +443,7 @@ macro (get_git)
   message (STATUS "Requires Git without specifying any version")
 
   find_package (Git)
-  if (Git_FOUND)
+  if (Git_FOUND AND EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/.git)
     Git_WC_INFO (${CMAKE_CURRENT_SOURCE_DIR} PROJ)
     set (GIT_REVISION ${PROJ_WC_REVISION_HASH})
     message (STATUS "Current Git revision name: ${PROJ_WC_REVISION_NAME}")
--- origsrc/simcrs-1.01.14/simcrs-config.cmake.in	2026-01-19 03:26:07.000000000 +0900
+++ src/simcrs-1.01.14/simcrs-config.cmake.in	2026-07-14 14:23:34.272063900 +0900
@@ -20,5 +20,5 @@ set (SIMCRS_LIBRARY_DIRS "@PACKAGE_SIMCR
 include ("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@-library-depends.cmake")
 
 # These are the SimCRS IMPORTED targets, created by simcrs-library-depends.cmake
-set (SIMCRS_LIBRARIES simcrslib)
+set (SIMCRS_LIBRARIES simcrs)
 set (SIMCRS_EXECUTABLES simcrs)
