--- origsrc/simcrs-simcrs-1.01.0/config/project_config_embeddable.cmake	2013-06-09 04:48:06.000000000 +0900
+++ src/simcrs-simcrs-1.01.0/config/project_config_embeddable.cmake	2014-11-08 16:38:51.601457100 +0900
@@ -1692,7 +1692,8 @@ macro (module_library_add_specific
   # Installation of the library
   install (TARGETS ${_lib_target}
     EXPORT ${LIB_DEPENDENCY_EXPORT}
-    LIBRARY DESTINATION "${INSTALL_LIB_DIR}" COMPONENT runtime)
+    RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
+    ARCHIVE DESTINATION "${INSTALL_LIB_DIR}")
 
   # Register, for reporting purpose, the list of libraries to be built
   # and installed for that module
--- origsrc/simcrs-simcrs-1.01.0/simcrs-config.cmake.in	2013-06-09 04:48:06.000000000 +0900
+++ src/simcrs-simcrs-1.01.0/simcrs-config.cmake.in	2014-11-08 16:39:01.881730500 +0900
@@ -18,5 +18,5 @@ set (SIMCRS_LIBRARY_DIRS "@SIMCRS_LIB_DI
 include ("@SIMCRS_CMAKE_DIR@/simcrs-library-depends.cmake")
 
 # These are the SimCRS IMPORTED targets, created by simcrs-library-depends.cmake
-set (SIMCRS_LIBRARIES simcrslib)
+set (SIMCRS_LIBRARIES simcrs)
 set (SIMCRS_EXECUTABLES simcrs)
