--- origsrc/simcrs-1.01.13/config/project_config_embeddable.cmake	2025-12-25 09:02:21.000000000 +0900
+++ src/simcrs-1.01.13/config/project_config_embeddable.cmake	2025-12-29 22:29:30.342325600 +0900
@@ -439,7 +439,7 @@ macro (get_git)
   message (STATUS "Requires Git without specifying any version")
 
   find_package (Git)
-  if (Git_FOUND)
+  if (Git_FOUND AND EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/.git)
     Git_WC_INFO (${CMAKE_CURRENT_SOURCE_DIR} PROJ)
     set (GIT_REVISION ${PROJ_WC_REVISION_HASH})
     message (STATUS "Current Git revision name: ${PROJ_WC_REVISION_NAME}")
--- origsrc/simcrs-1.01.13/simcrs-config.cmake.in	2025-12-25 09:02:21.000000000 +0900
+++ src/simcrs-1.01.13/simcrs-config.cmake.in	2025-12-29 22:29:18.138250800 +0900
@@ -20,5 +20,5 @@ set (SIMCRS_LIBRARY_DIRS "@PACKAGE_SIMCR
 include ("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@-library-depends.cmake")
 
 # These are the SimCRS IMPORTED targets, created by simcrs-library-depends.cmake
-set (SIMCRS_LIBRARIES simcrslib)
+set (SIMCRS_LIBRARIES simcrs)
 set (SIMCRS_EXECUTABLES simcrs)
