--- origsrc/liblscp-1.0.1/CMakeLists.txt	2025-03-27 23:20:36.298889600 +0900
+++ src/liblscp-1.0.1/CMakeLists.txt	2025-04-15 09:25:40.341060500 +0900
@@ -9,6 +9,7 @@ project (liblscp
 set (PROJECT_COPYRIGHT "Copyright (C) 2004-2025, rncbc aka Rui Nuno Capela. All rights reserved.")
 set (PROJECT_DOMAIN    "linuxsampler.org")
 
+if (EXISTS "${CMAME_SOURCE_DIR}/.git")
 execute_process (
   COMMAND git describe --tags --dirty --abbrev=6
   WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
@@ -31,7 +32,7 @@ if (GIT_DESCRIBE_RESULT EQUAL 0)
   endif ()
   set (PROJECT_VERSION "${GIT_VERSION}")
 endif ()
-
+endif()
 
 set (CONFIG_PREFIX "${CMAKE_INSTALL_PREFIX}")
 
--- origsrc/liblscp-1.0.1/doc/CMakeLists.txt	2025-03-27 23:20:36.301157000 +0900
+++ src/liblscp-1.0.1/doc/CMakeLists.txt	2025-04-15 09:25:40.341060500 +0900
@@ -2,7 +2,7 @@
 
 find_package (Doxygen)
 
-if (DOXYGEN_FOUND)
+if (BUILD_DOCS AND DOXYGEN_FOUND)
   set (top_srcdir "${CMAKE_CURRENT_SOURCE_DIR}/..")
   configure_file (liblscp.doxygen.in liblscp.doxygen IMMEDIATE @ONLY)
   add_custom_target (doxygen ALL
--- origsrc/liblscp-1.0.1/src/CMakeLists.txt	2025-03-27 23:20:36.308889500 +0900
+++ src/liblscp-1.0.1/src/CMakeLists.txt	2025-04-15 09:25:40.341060500 +0900
@@ -57,7 +57,6 @@ endif()
 
 set_target_properties (${PROJECT_NAME} PROPERTIES
   PUBLIC_HEADER "${PUBLIC_HEADERS}"
-  PREFIX "lib"
   OUTPUT_NAME "lscp"
   VERSION ${SHARED_VERSION_INFO}
   SOVERSION ${SHARED_VERSION_CURRENT})
