--- origsrc/visp-3.6.0/CMakeLists.txt	2023-09-22 14:39:36.000000000 +0900
+++ src/visp-3.6.0/CMakeLists.txt	2025-08-24 01:18:22.945088300 +0900
@@ -640,6 +640,7 @@ if(USE_PCL)
       set(VISP_CXX_STANDARD ${VISP_CXX_STANDARD_14}) # for vpConfig.h
     endif()
   endif()
+  find_package(VTK)
 endif()
 
 # Upgrade c++ standard to 14 for IIT force-torque SDK that needs at least c++ 14 standard
@@ -663,7 +664,7 @@ VP_OPTION(WITH_APRILTAG_BIG_FAMILY ""  "
 VP_OPTION(WITH_ATIDAQ     ""           ""    "Build atidaq-c as built-in library"  "" ON IF USE_COMEDI AND NOT WINRT)
 VP_OPTION(WITH_CLIPPER    ""           ""    "Build clipper as built-in library"   "" ON IF USE_OPENCV)
 VP_OPTION(WITH_LAPACK     ""           ""    "Build lapack as built-in library"    "" ON IF NOT USE_LAPACK)
-VP_OPTION(WITH_QBDEVICE   ""           ""    "Build qbdevice-api as built-in library" "" ON IF (VISP_CXX_STANDARD GREATER VISP_CXX_STANDARD_98) AND (NOT WINRT) AND (NOT IOS))
+VP_OPTION(WITH_QBDEVICE   ""           ""    "Build qbdevice-api as built-in library" "" ON IF (VISP_CXX_STANDARD GREATER VISP_CXX_STANDARD_98) AND (NOT WINRT) AND (NOT IOS) AND (NOT CYGWIN))
 VP_OPTION(WITH_TAKKTILE2  ""           ""    "Build Right Hand takktile2 driver as built-in library"   "" ON IF (VISP_CXX_STANDARD GREATER VISP_CXX_STANDARD_98) AND (NOT WIN32) AND (NOT WINRT) AND (NOT IOS) AND (NOT ANDROID))
 VP_OPTION(WITH_CATCH2     ""           ""    "Use catch2" "" ON IF (VISP_CXX_STANDARD GREATER VISP_CXX_STANDARD_98))
 
--- origsrc/visp-3.6.0/modules/gui/CMakeLists.txt	2023-09-22 14:39:36.000000000 +0900
+++ src/visp-3.6.0/modules/gui/CMakeLists.txt	2025-08-24 01:26:07.052368600 +0900
@@ -81,6 +81,9 @@ if(USE_PCL)
   vp_find_pcl(PCL_LIBRARIES PCL_DEPS_INCLUDE_DIRS PCL_DEPS_LIBRARIES)
   list(APPEND opt_incs ${PCL_DEPS_INCLUDE_DIRS})
   list(APPEND opt_libs ${PCL_DEPS_LIBRARIES})
+  if (VTK_FOUND)
+    list(APPEND opt_libs ${VTK_LIBRARIES})
+  endif()
 endif()
 
 vp_add_module(gui visp_core)
