--- origsrc/doctrina-0.1.12/src/meson.build	2019-03-20 07:01:53.000000000 +0900
+++ src/doctrina-0.1.12/src/meson.build	2025-12-12 04:48:41.864466400 +0900
@@ -13,6 +13,7 @@ doctrina_lib = library('doctrina',
                        vala_header: 'doctrina.h',
                        vala_vapi: doctrina_name_api + '.vapi',
                        dependencies: doctrina_deps,
+                       soversion: api_version,
                        install: true,
                        install_dir: [true, true, true])
 
