--- origsrc/kokkos-kernels-3.7.01/CMakeLists.txt	2022-12-07 08:08:36.000000000 +0900
+++ src/kokkos-kernels-3.7.01/CMakeLists.txt	2023-01-24 21:13:59.987020300 +0900
@@ -230,3 +230,12 @@ ELSE()
     ADD_SUBDIRECTORY(docs)
   ENDIF ()
 ENDIF()
+
+if (CYGWIN)
+  set(prefix ${CMAKE_INSTALL_PREFIX})
+  set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+  set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+  set(includedir ${CMAKE_INSTALL_INCLUDEDIR})
+  configure_file(${CMAKE_SOURCE_DIR}/kokkos-kernels.pc.in ${CMAKE_BINARY_DIR}/kokkos-kernels.pc @ONLY)
+  install(FILES ${CMAKE_BINARY_DIR}/kokkos-kernels.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+endif()
--- origsrc/kokkos-kernels-3.7.01/cmake/kokkoskernels_tribits.cmake	2022-12-07 08:08:36.000000000 +0900
+++ src/kokkos-kernels-3.7.01/cmake/kokkoskernels_tribits.cmake	2023-01-24 21:13:59.990060800 +0900
@@ -93,6 +93,10 @@ ADD_LIBRARY(
 )
 ADD_LIBRARY(Kokkos::${LIBRARY_NAME} ALIAS ${LIBRARY_NAME})
 
+if (CYGWIN)
+  set_target_properties(${LIBRARY_NAME} PROPERTIES SOVERSION "${KokkosKernels_VERSION_MAJOR}.${KokkosKernels_VERSION_MINOR}")
+endif()
+
 INSTALL(
   TARGETS ${LIBRARY_NAME}
   EXPORT ${PROJECT_NAME}
--- origsrc/kokkos-kernels-3.7.01/kokkos-kernels.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/kokkos-kernels-3.7.01/kokkos-kernels.pc.in	2023-01-24 21:13:59.990060800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: kokkos-kernels
+Description: C++ Performance Portability Programming EcoSystem: Math Kernels
+Version: @KokkosKernels_VERSION@
+Libs: -L${libdir} -lkokkoskernels
+Cflags: -I${includedir}
--- origsrc/kokkos-kernels-3.7.01/src/batched/dense/impl/KokkosBatched_Gemm_Serial_Impl.hpp	2022-12-07 08:08:36.000000000 +0900
+++ src/kokkos-kernels-3.7.01/src/batched/dense/impl/KokkosBatched_Gemm_Serial_Impl.hpp	2023-01-24 21:13:59.990060800 +0900
@@ -45,6 +45,13 @@
 #include "KokkosBatched_Util.hpp"
 #include "KokkosBatched_Gemm_Serial_Internal.hpp"
 
+#if defined(_B)
+#undef _B
+#endif
+#if defined(_C)
+#undef _C
+#endif
+
 namespace KokkosBatched {
 /********************* BEGIN functor-level routines *********************/
 ///
--- origsrc/kokkos-kernels-3.7.01/src/batched/sparse/KokkosBatched_Krylov_Solvers.hpp	2022-12-07 08:08:36.000000000 +0900
+++ src/kokkos-kernels-3.7.01/src/batched/sparse/KokkosBatched_Krylov_Solvers.hpp	2023-01-25 10:52:28.802132400 +0900
@@ -45,6 +45,13 @@
 #ifndef __KOKKOSBATCHED_KRYLOV_SOLVERS_HPP__
 #define __KOKKOSBATCHED_KRYLOV_SOLVERS_HPP__
 
+#if defined(_B)
+#undef _B
+#endif
+#if defined(_X)
+#undef _X
+#endif
+
 namespace KokkosBatched {
 
 struct SerialGMRES {
--- origsrc/kokkos-kernels-3.7.01/test_common/KokkosKernels_TestUtils.hpp	2022-12-07 08:08:36.000000000 +0900
+++ src/kokkos-kernels-3.7.01/test_common/KokkosKernels_TestUtils.hpp	2023-01-24 21:14:00.003137800 +0900
@@ -54,6 +54,13 @@
 // Make this include-able from all subdirectories
 #include "../tpls/gtest/gtest/gtest.h"  //for EXPECT_**
 
+#if defined(_B)
+#undef _B
+#endif
+#if defined(_C)
+#undef _C
+#endif
+
 namespace Test {
 template <class ViewType,
           bool strided = std::is_same<typename ViewType::array_layout,
--- origsrc/kokkos-kernels-3.7.01/unit_test/batched/dense/Test_Batched_SerialAxpy.hpp	2022-12-07 08:08:36.000000000 +0900
+++ src/kokkos-kernels-3.7.01/unit_test/batched/dense/Test_Batched_SerialAxpy.hpp	2023-01-24 21:14:00.012022500 +0900
@@ -8,6 +8,10 @@
 
 #include "KokkosKernels_TestUtils.hpp"
 
+#if defined(_X)
+#undef _X
+#endif
+
 using namespace KokkosBatched;
 
 namespace Test {
--- origsrc/kokkos-kernels-3.7.01/unit_test/batched/sparse/Test_Batched_SerialSpmv.hpp	2022-12-07 08:08:36.000000000 +0900
+++ src/kokkos-kernels-3.7.01/unit_test/batched/sparse/Test_Batched_SerialSpmv.hpp	2023-01-24 21:14:00.031095700 +0900
@@ -12,6 +12,10 @@
 #include "KokkosKernels_TestUtils.hpp"
 #include "Test_Batched_SparseUtils.hpp"
 
+#if defined(_X)
+#undef _X
+#endif
+
 using namespace KokkosBatched;
 
 namespace Test {
