--- origsrc/scrm-1.7.4/Makefile.am	2020-03-08 20:27:58.000000000 +0900
+++ src/scrm-1.7.4/Makefile.am	2020-06-20 22:45:54.781783600 +0900
@@ -3,7 +3,7 @@ bin_PROGRAMS = scrm
 man_MANS = doc/scrm.1
 
 TESTS = unit_tests algorithm_tests
-check_PROGRAMS = unit_tests algorithm_tests scrm_dbg scrm_asan scrm_prof
+check_PROGRAMS = unit_tests algorithm_tests scrm_dbg # scrm_asan scrm_prof
 PROG = SCRM
 
 dist-hook:
@@ -58,15 +58,15 @@ alg_test_src = tests/cppunit/test_runner
 
 scrm_SOURCES = $(scrm_src) src/scrm.cc
 scrm_dbg_SOURCES = $(scrm_src) $(debug_src) src/scrm.cc
-scrm_prof_SOURCES = $(scrm_src) src/scrm.cc
-scrm_asan_SOURCES = $(scrm_src) src/scrm.cc
+# scrm_prof_SOURCES = $(scrm_src) src/scrm.cc
+# scrm_asan_SOURCES = $(scrm_src) src/scrm.cc
 unit_tests_SOURCES = $(scrm_src) $(debug_src) $(unit_test_src)
 algorithm_tests_SOURCES = $(scrm_src) $(alg_test_src)
 
 scrm_CXXFLAGS= -DNDEBUG @OPT_CXXFLAGS@
 scrm_dbg_CXXFLAGS= -g
-scrm_prof_CXXFLAGS= -pg -DNDEBUG
-scrm_asan_CXXFLAGS= -g -DNDEBUG -fsanitize=undefined,address -fno-sanitize-recover
+# scrm_prof_CXXFLAGS= -pg -DNDEBUG
+# scrm_asan_CXXFLAGS= -g -DNDEBUG -fsanitize=undefined,address -fno-sanitize-recover
 unit_tests_CXXFLAGS = -g -DUNITTEST -DNDEBUG @TEST_CXXFLAGS@ 
 algorithm_tests_CXXFLAGS = -g -DNDEBUG
 unit_tests_LDADD= -L/opt/local/lib -lcppunit -ldl #link the cppunit unittest library in mac, cppunit was installed via macports
