--- origsrc/h3-3.4.4/CMakeLists.txt	2019-06-01 02:41:47.000000000 +0900
+++ src/h3-3.4.4/CMakeLists.txt	2019-06-01 10:04:56.018969300 +0900
@@ -584,3 +584,10 @@ install(
     NAMESPACE "${namespace}"
     DESTINATION "${config_install_dir}"
 )
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/h3.pc.in ${CMAKE_BINARY_DIR}/h3.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/h3.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/h3-3.4.4/h3.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/h3-3.4.4/h3.pc.in	2019-06-01 10:04:56.038967900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: h3
+Description: Hexagonal hierarchical geospatial indexing system
+Version: @H3_VERSION@
+Libs: -L${libdir} -lh3
+Cflags: -I${includedir}
