--- origsrc/h3-4.2.0/CMakeLists.txt	2024-12-05 13:58:03.000000000 +0900
+++ src/h3-4.2.0/CMakeLists.txt	2024-12-24 08:43:23.241668200 +0900
@@ -725,3 +725,10 @@ set(CPACK_DEBIAN_LIBH3-DEV_PACKAGE_SECTI
 set(CPACK_DEBIAN_H3_PACKAGE_SECTION "science")
 
 include(CPack)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/h3.pc.in ${CMAKE_BINARY_DIR}/h3.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/h3.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/h3-4.2.0/h3.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/h3-4.2.0/h3.pc.in	2024-12-24 08:43:23.241668200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: h3
+Description: Hexagonal hierarchical geospatial indexing system
+Version: @H3_VERSION@
+Libs: -L${libdir} -lh3
+Cflags: -I${includedir}
