--- origsrc/whowatch-1.8.5/configure.ac	2015-03-11 18:11:10.903057300 +0900
+++ src/whowatch-1.8.5/configure.ac	2015-03-11 18:11:34.765362000 +0900
@@ -10,8 +10,6 @@ AC_PROG_CC
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
 
-LDFLAGS="$LDFLAGS -rdynamic"
-
 dnl Checks for libraries.
 dnl Replace `main' with a function in -lkvm:
 AC_CHECK_LIB(kvm, kvm_openfiles)
--- origsrc/whowatch-1.8.5/src/proc_plugin.c	2013-10-07 21:11:29.000000000 +0900
+++ src/whowatch-1.8.5/src/proc_plugin.c	2015-03-11 18:13:43.500713600 +0900
@@ -382,8 +382,11 @@ FOUND:
 	sscanf(buf+5, "%ld", &boot_time);
        	return boot_time;
 }		
-
+#if defined(__CYGWIN__)
+#include <sys/param.h>
+#else
 #include <asm/param.h>	// for HZ
+#endif
 
 static void proc_starttime(int pid, char *name)
 {
