--- origsrc/envv-1.7/envv.c	2011-07-21 02:59:06.000000000 +0900
+++ src/envv-1.7/envv.c	2024-10-30 14:41:34.712784300 +0900
@@ -328,7 +328,7 @@ void DoSetenv(const char *var, const cha
       PrintEscaped(val, 0);
       if (!local)
 	printf("; export %s", var);
-      printf(TrailingSemi);
+      printf("%s", TrailingSemi);
       break;
 
     case CSH_LIKE:
@@ -338,7 +338,7 @@ void DoSetenv(const char *var, const cha
 	printf("set %s=", var);
 
       PrintEscaped(val, 0);
-      printf(TrailingSemi);
+      printf("%s", TrailingSemi);
       break;
 
     default:
@@ -586,7 +586,7 @@ void PathManip(const char *var, const ch
 
    switch(shell) {
     case SH_LIKE: printf("; export %s%s", var, TrailingSemi); break;
-    case CSH_LIKE: printf(TrailingSemi); break;
+    case CSH_LIKE: printf("%s", TrailingSemi); break;
    }
    free(path);
    return;
@@ -604,12 +604,12 @@ void DoChoose(const char *val1, const ch
    switch(shell) {
       case SH_LIKE:
          PrintEscaped(val1, 0);
-	 printf(TrailingSemi);
+	 printf("%s", TrailingSemi);
 	 break;
 
      case CSH_LIKE:
 	 PrintEscaped(val2, 0);
-	 printf(TrailingSemi);
+	 printf("%s", TrailingSemi);
 	 break;
 
      default:
