--- origsrc/rudecgi-5.1.0/Makefile.am	2008-01-12 00:08:26.000000000 +0900
+++ src/rudecgi-5.1.0/Makefile.am	2012-06-17 18:42:51.095409400 +0900
@@ -87,7 +87,7 @@ librudecgi_la_SOURCES = 	src/AbstractImp
 	src/UnknownParser.cpp \
 	src/UnknownParser.h
 
-librudecgi_la_LDFLAGS = -version-info 5:1:0
+librudecgi_la_LDFLAGS = -version-info 5:1:0 -no-undefined
 
 pkginclude_HEADERS = src/cgi.h
 pkgincludedir = $(includedir)/rude
