--- origsrc/rudecgi-5.1.0/Makefile.am	2008-01-12 00:08:26.000000000 +0900
+++ src/rudecgi-5.1.0/Makefile.am	2010-12-29 22:47:25.186401200 +0900
@@ -87,7 +87,7 @@ librudecgi_la_SOURCES = 	src/AbstractImp
 	src/UnknownParser.cpp \
 	src/UnknownParser.h
 
-librudecgi_la_LDFLAGS = -version-info 5:1:0
+librudecgi_la_LDFLAGS = -version-info 5:1:0 -no-undefined
 
 pkginclude_HEADERS = src/cgi.h
 pkgincludedir = $(includedir)/rude
--- origsrc/rudecgi-5.1.0/configure.ac	2008-01-12 00:08:09.000000000 +0900
+++ src/rudecgi-5.1.0/configure.ac	2010-12-29 22:47:08.552482800 +0900
@@ -29,7 +29,7 @@ AM_INIT_AUTOMAKE(rudecgi, 5.1.0)
 # Checks for programs.
 AC_PROG_CXX
 AC_PROG_CC
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 # Checks for libraries.
 
