--- origsrc/libpuzzle-0.11/Makefile.am	2009-03-24 23:01:25.000000000 +0900
+++ src/libpuzzle-0.11/Makefile.am	2015-05-21 23:43:53.924829900 +0900
@@ -8,3 +8,6 @@ SUBDIRS = \
 	src \
 	man \
 	php
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libpuzzle-0.11/configure.ac	2009-03-24 23:53:16.000000000 +0900
+++ src/libpuzzle-0.11/configure.ac	2015-05-21 23:43:53.829672300 +0900
@@ -47,7 +47,7 @@ AC_CHECK_FUNCS([strtoul])
 
 AC_SUBST([MAINT])
 
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([libpuzzle.pc Makefile
                  man/Makefile
                  src/Makefile
                  src/pics/Makefile
--- origsrc/libpuzzle-0.11/libpuzzle.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libpuzzle-0.11/libpuzzle.pc.in	2015-05-21 23:43:53.829672300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libpuzzle
+Description: Library to quickly find visually similar images
+Version: @VERSION@
+Libs: -L${libdir} -llibpuzzle
+Cflags: -I${includedir}
--- origsrc/libpuzzle-0.11/src/Makefile.am	2009-03-24 23:01:25.000000000 +0900
+++ src/libpuzzle-0.11/src/Makefile.am	2015-05-21 23:41:54.928169000 +0900
@@ -1,7 +1,7 @@
 lib_LTLIBRARIES = \
 	libpuzzle.la
 
-libpuzzle_la_LDFLAGS = -version-info 1:0
+libpuzzle_la_LDFLAGS = -no-undefined -version-info 1:0
 
 libpuzzle_la_SOURCES = \
 	puzzle.c \
--- origsrc/libpuzzle-0.11/src/puzzle-diff.c	2009-03-24 23:01:25.000000000 +0900
+++ src/libpuzzle-0.11/src/puzzle-diff.c	2015-05-21 23:41:54.928169000 +0900
@@ -35,8 +35,6 @@ void usage(void)
 int parse_opts(Opts * const opts, PuzzleContext * context,
                int argc, char * const *argv) {
     int opt;
-    extern char *optarg;
-    extern int optind;
 
     opts->fix_for_texts = 1;
     opts->exit = 0;
