--- origsrc/libpuzzle-0.11/src/Makefile.am	2009-03-24 23:01:25.000000000 +0900
+++ src/libpuzzle-0.11/src/Makefile.am	2014-05-29 13:56:03.300770700 +0900
@@ -1,7 +1,7 @@
 lib_LTLIBRARIES = \
 	libpuzzle.la
 
-libpuzzle_la_LDFLAGS = -version-info 1:0
+libpuzzle_la_LDFLAGS = -version-info 1:0 -no-undefined
 
 libpuzzle_la_SOURCES = \
 	puzzle.c \
--- origsrc/libpuzzle-0.11/src/puzzle-diff.c	2009-03-24 23:01:25.000000000 +0900
+++ src/libpuzzle-0.11/src/puzzle-diff.c	2014-05-29 13:59:07.441424600 +0900
@@ -35,8 +35,6 @@ void usage(void)
 int parse_opts(Opts * const opts, PuzzleContext * context,
                int argc, char * const *argv) {
     int opt;
-    extern char *optarg;
-    extern int optind;
 
     opts->fix_for_texts = 1;
     opts->exit = 0;
