--- origsrc/blasr_libcpp-5.3.4/meson.build	2020-02-24 20:17:58.000000000 +0900
+++ src/blasr_libcpp-5.3.4/meson.build	2020-12-15 13:34:43.655271200 +0900
@@ -57,6 +57,12 @@ libblasr_zlib_dep = dependency('zlib', r
 # htslib
 libblasr_htslib_dep = dependency('htslib', required : true, version : '>=1.4', fallback : ['htslib', 'htslib_dep'])
 
+# hdf5
+libblasr_hdf5_dep = cpp.find_library('hdf5', required: true)
+
+# hdf5cpp
+libblasr_hdf5cpp_dep = cpp.find_library('hdf5_cpp', required: true)
+
 # clock_gettime on old glibc systems
 libblasr_rt_dep = cpp.find_library('rt', required : false)
 
@@ -65,6 +71,8 @@ libblasr_deps = [
   libblasr_pbbam_dep,
   libblasr_zlib_dep,
   libblasr_htslib_dep,
+  libblasr_hdf5_dep,
+  libblasr_hdf5cpp_dep,
   libblasr_rt_dep]
 
 ##########
