--- origsrc/pdfcube-0.0.3/configure.ac	2008-12-19 05:06:30.000000000 +0900
+++ src/pdfcube-0.0.3/configure.ac	2010-12-30 12:48:14.159814400 +0900
@@ -19,7 +19,7 @@ dnl ------------------------------------
 
 AC_PROG_CC
 AC_PROG_CXX
-AM_PROG_LIBTOOL
+LT_INIT
 AM_SANITY_CHECK
 
 dnl -----------------------------------------------
--- origsrc/pdfcube-0.0.3/src/Makefile.am	2008-12-19 05:06:10.000000000 +0900
+++ src/pdfcube-0.0.3/src/Makefile.am	2010-12-30 12:47:41.272524800 +0900
@@ -4,7 +4,7 @@ pdfcube_SOURCES = pdfcube.cc
 pdfcube_CXXFLAGS = -Wall -Weffc++ 
 
 LDADD = $(glib_LIBS) $(gtk_LIBS) $(gtkglext_LIBS) $(poppler_LIBS) $(poppler_glib_LIBS) -lglut 
-INCLUDES = $(glib_CFLAGS) $(gtk_CFLAGS) $(gtkglext_CFLAGS) $(poppler_CFLAGS) $(poppler_glib_CFLAGS) $(BOOST_LDFLAGS)
+INCLUDES = $(glib_CFLAGS) $(gtk_CFLAGS) $(gtkglext_CFLAGS) $(poppler_CFLAGS) $(poppler_glib_CFLAGS) $(BOOST_CPPFLAGS)
 
 AM_CXXFLAGS = $(BOOST_CPPFLAGS) 
 AM_LDFLAGS=$(BOOST_LDFLAGS) $(BOOST_PROGRAM_OPTIONS_LIB)
