--- origsrc/fastahack-1.0.0/LargeFileSupport.h	2019-09-04 15:16:27.000000000 +0900
+++ src/fastahack-1.0.0/LargeFileSupport.h	2022-06-10 19:16:52.194873400 +0900
@@ -5,7 +5,7 @@
 #define ftell64(a)     _ftelli64(a)
 #define fseek64(a,b,c) _fseeki64(a,b,c)
 typedef __int64 off_type;
-#elif defined(__APPLE__) || defined(__FreeBSD__)
+#elif defined(__APPLE__) || defined(__FreeBSD__) || defined(__CYGWIN__)
 #define ftell64(a)     ftello(a)
 #define fseek64(a,b,c) fseeko(a,b,c)
 typedef off_t off_type;
