--- origsrc/spdylay-1.0.0/tests/malloc_wrapper.c	2013-06-09 23:14:16.000000000 +0900
+++ src/spdylay-1.0.0/tests/malloc_wrapper.c	2013-06-27 07:52:18.240861100 +0900
@@ -36,7 +36,12 @@ static void* (*real_malloc)(size_t) = NU
 
 static void init(void)
 {
+#if defined(__CYGWIN__)
+  void* dl = dlopen("/usr/bin/cygwin1.dll", RTLD_LOCAL);
+  real_malloc = dlsym(dl, "malloc");
+#else
   real_malloc = dlsym(RTLD_NEXT, "malloc");
+#endif
 }
 
 void* malloc(size_t size)
