--- origsrc/sobby-0.4.8/Makefile.am	2012-02-19 23:45:53.000000000 +0900
+++ src/sobby-0.4.8/Makefile.am	2015-02-26 16:24:59.268597400 +0900
@@ -1,6 +1,6 @@
 bin_PROGRAMS    = sobby
 sobby_LDADD     = $(sobby_LIBS)
-sobby_CPPFLAGS  = $(sobby_CFLAGS) -I../inc/
+sobby_CPPFLAGS  = $(sobby_CFLAGS) -I$(top_builddir)/inc -I$(top_srcdir)/inc
 
 if WITH_ZEROCONF
 if WITH_AVAHI
--- origsrc/sobby-0.4.8/inc/features.hpp.in	1970-01-01 09:00:00.000000000 +0900
+++ src/sobby-0.4.8/inc/features.hpp.in	2015-02-26 16:25:39.034222400 +0900
@@ -0,0 +1,31 @@
+/* inc/features.hpp.in.  Generated from configure.ac by autoheader.  */
+
+/* Name of package */
+#undef PACKAGE
+
+/* Define to the address where bug reports for this package should be sent. */
+#undef PACKAGE_BUGREPORT
+
+/* Define to the full name of this package. */
+#undef PACKAGE_NAME
+
+/* Define to the full name and version of this package. */
+#undef PACKAGE_STRING
+
+/* Define to the one symbol short name of this package. */
+#undef PACKAGE_TARNAME
+
+/* Define to the home page for this package. */
+#undef PACKAGE_URL
+
+/* Define to the version of this package. */
+#undef PACKAGE_VERSION
+
+/* Version number of package */
+#undef VERSION
+
+/* Define whether Avahi is used as Zeroconf backend */
+#undef WITH_AVAHI
+
+/* Define whether to use Zeroconf */
+#undef WITH_ZEROCONF
