--- origsrc/vid.stab-1.1.1/CMakeLists.txt	2022-05-31 00:49:01.000000000 +0900
+++ src/vid.stab-1.1.1/CMakeLists.txt	2025-12-13 08:59:58.142748800 +0900
@@ -8,9 +8,9 @@ include (GNUInstallDirs)
 find_package(OpenMP)
 
 set(MAJOR_VERSION 1)
-set(MINOR_VERSION 2)
-set(PATCH_VERSION 0)
-set(VIDSTAB_VERSION ${MAJOR_VERSION}.${MINOR_VERSION}${PATCH_VERSION})
+set(MINOR_VERSION 1)
+set(PATCH_VERSION 1)
+set(VIDSTAB_VERSION ${MAJOR_VERSION}.${MINOR_VERSION}.${PATCH_VERSION})
 
 # Default to release builds if no explicit build type specified.
 if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
@@ -95,3 +95,6 @@ INSTALL(TARGETS vidstab
 
 include(create_pkgconfig_file)
 create_pkgconfig_file(vidstab "Vid.Stab, a library for stabilizing video clips")
+
+add_subdirectory(tests)
+
