--- origsrc/fs_mark-3.3/Makefile	2011-09-02 10:15:44.639328300 +0900
+++ src/fs_mark-3.3/Makefile	2011-09-06 00:13:43.237316900 +0900
@@ -6,8 +6,8 @@
 # Modifications include using fsync after wrting to flush to disk and changes to check return
 # values from syscalls.
 #
-DIR1= /test/dir1
-DIR2= /test/dir2
+DIR1= /tmp/dir1
+DIR2= /tmp/dir2
 
 COBJS= fs_mark.o lib_timing.o
 CFLAGS+= -O2 -Wall -D_FILE_OFFSET_BITS=64
--- origsrc/fs_mark-3.3/fs_mark.c	2008-09-09 11:23:54.000000000 +0900
+++ src/fs_mark-3.3/fs_mark.c	2011-09-02 10:17:34.907886700 +0900
@@ -42,9 +42,11 @@ char *fs_mark_version = "3.3";
 #include <ctype.h>
 #include <time.h>
 
+#if !defined(__CYGWIN__)
 #include <linux/types.h>
 #include <linux/limits.h>
 #include <linux/unistd.h>
+#endif
 
 #include "fs_mark.h"
 
--- origsrc/fs_mark-3.3/lib_timing.c	2005-07-12 03:49:09.000000000 +0900
+++ src/fs_mark-3.3/lib_timing.c	2011-09-02 10:17:06.777437100 +0900
@@ -24,7 +24,9 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <string.h>
+#if !defined(__CYGWIN__)
 #include <linux/types.h>
+#endif
 
 #define	nz(x)	((x) == 0 ? 1 : (x))
 
