--- origsrc/libmsip/Makefile.am	2012-03-21 21:22:38.000000000 +0900
+++ src/libmsip/Makefile.am	2012-03-21 21:23:25.106423600 +0900
@@ -120,7 +120,7 @@ msip_src =           source/SipDialogCon
                      source/headers/SipHeaderWarning.cxx
 
 # maintainer rules
-ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
+ACLOCAL_AMFLAGS = -I m4
 MAINTAINERCLEANFILES = \
 		$(srcdir)/Makefile.in \
 		$(srcdir)/configure \
--- origsrc/libmsip/configure.ac	2012-03-21 21:22:37.000000000 +0900
+++ src/libmsip/configure.ac	2012-03-21 21:23:11.684548600 +0900
@@ -20,8 +20,6 @@ AC_PROG_LIBTOOL
 AM_MINISIP_ENABLE_DEBUG
 AM_MINISIP_ENABLE_TEST_SUITE
 
-CPPFLAGS="${CPPFLAGS} -std=c++0x"
-
 dnl debug flag
 AC_ARG_ENABLE(dropemul,
 		AS_HELP_STRING([--enable-dropemul],
