--- origsrc/simlfs-1.00.8/simlfs-config.cmake.in	2026-01-19 05:48:05.000000000 +0900
+++ src/simlfs-1.00.8/simlfs-config.cmake.in	2026-07-14 11:38:42.311602000 +0900
@@ -1,20 +1,20 @@
 #
-# Config file for the RMOL package. It defines the following variables:
-#  RMOL_INCLUDE_DIRS - include directories for RMOL
-#  RMOL_LIBRARY_DIRS - library directories for RMOL (normally not used!)
-#  RMOL_BINARY_DIRS  - binary directories for RMOL (normally not used!)
-#  RMOL_LIBRARIES    - libraries to link against
-#  RMOL_EXECUTABLES  - the RMOL binaries/executables
+# Config file for the SIMLFS package. It defines the following variables:
+#  SIMLFS_INCLUDE_DIRS - include directories for SIMLFS
+#  SIMLFS_LIBRARY_DIRS - library directories for SIMLFS (normally not used!)
+#  SIMLFS_BINARY_DIRS  - binary directories for SIMLFS (normally not used!)
+#  SIMLFS_LIBRARIES    - libraries to link against
+#  SIMLFS_EXECUTABLES  - the SIMLFS binaries/executables
 
-# Tell the user project where to find RMOL headers and libraries
-set (RMOL_INCLUDE_DIRS "@RMOL_INCLUDE_DIRS@")
-set (RMOL_LIBRARY_DIRS "@RMOL_LIB_DIR@")
-set (RMOL_BINARY_DIRS "@RMOL_BIN_DIR@")
+# Tell the user project where to find SIMLFS headers and libraries
+set (SIMLFS_INCLUDE_DIRS "@SIMLFS_INCLUDE_DIRS@")
+set (SIMLFS_LIBRARY_DIRS "@SIMLFS_LIB_DIR@")
+set (SIMLFS_BINARY_DIRS "@SIMLFS_BIN_DIR@")
 
-# Library dependencies for RMOL (contains definitions for the RMOL IMPORTED
+# Library dependencies for SIMLFS (contains definitions for the SIMLFS IMPORTED
 # targets)
-include ("@RMOL_CMAKE_DIR@/rmol-library-depends.cmake")
+include ("@SIMLFS_CMAKE_DIR@/simlfs-library-depends.cmake")
 
-# These are the RMOL IMPORTED targets, created by rmol-library-depends.cmake
-set (RMOL_LIBRARIES airraclib rmollib)
-set (RMOL_EXECUTABLES rmol)
+# These are the SIMLFS IMPORTED targets, created by simlfs-library-depends.cmake
+set (SIMLFS_LIBRARIES simlfs)
+set (SIMLFS_EXECUTABLES simlfs)
