--- origsrc/simlfs-1.00.7/config/project_config_embeddable.cmake	2025-01-28 02:05:41.000000000 +0900
+++ src/simlfs-1.00.7/config/project_config_embeddable.cmake	2025-06-12 21:07:19.442349900 +0900
@@ -1630,7 +1630,7 @@ macro (add_modules)
   set (_embedded_components ${ARGV})
   set (LIB_DEPENDENCY_EXPORT ${PROJECT_NAME}-library-depends)
   set (PY_LIB_DEPENDENCY_EXPORT ${PROJECT_NAME}-python-library-depends)
-  set (LIB_DEPENDENCY_EXPORT_PATH "${INSTALL_DATA_DIR}/${PROJECT_NAME}/CMake")
+  set (LIB_DEPENDENCY_EXPORT_PATH "lib/cmake/${PROJECT_NAME}")
   #
   foreach (_embedded_comp ${_embedded_components})
     #
@@ -2160,7 +2160,7 @@ endmacro (module_script_add)
 # can refer to it (for libraries, header files and binaries)
 macro (module_export_install)
   install (EXPORT ${LIB_DEPENDENCY_EXPORT}
-    DESTINATION "${INSTALL_DATA_DIR}/${PACKAGE}/CMake" COMPONENT devel)
+    DESTINATION "lib/cmake/${PACKAGE}" COMPONENT devel)
 
   if (NEED_PYTHON)
 	install (EXPORT ${PY_LIB_DEPENDENCY_EXPORT}
--- origsrc/simlfs-1.00.7/simlfs-config.cmake.in	2025-01-28 02:05:41.000000000 +0900
+++ src/simlfs-1.00.7/simlfs-config.cmake.in	2025-06-12 20:59:43.473680800 +0900
@@ -1,20 +1,20 @@
 #
-# Config file for the RMOL package. It defines the following variables:
-#  RMOL_INCLUDE_DIRS - include directories for RMOL
-#  RMOL_LIBRARY_DIRS - library directories for RMOL (normally not used!)
-#  RMOL_BINARY_DIRS  - binary directories for RMOL (normally not used!)
-#  RMOL_LIBRARIES    - libraries to link against
-#  RMOL_EXECUTABLES  - the RMOL binaries/executables
+# Config file for the SIMLFS package. It defines the following variables:
+#  SIMLFS_INCLUDE_DIRS - include directories for SIMLFS
+#  SIMLFS_LIBRARY_DIRS - library directories for SIMLFS (normally not used!)
+#  SIMLFS_BINARY_DIRS  - binary directories for SIMLFS (normally not used!)
+#  SIMLFS_LIBRARIES    - libraries to link against
+#  SIMLFS_EXECUTABLES  - the SIMLFS binaries/executables
 
-# Tell the user project where to find RMOL headers and libraries
-set (RMOL_INCLUDE_DIRS "@RMOL_INCLUDE_DIRS@")
-set (RMOL_LIBRARY_DIRS "@RMOL_LIB_DIR@")
-set (RMOL_BINARY_DIRS "@RMOL_BIN_DIR@")
+# Tell the user project where to find SIMLFS headers and libraries
+set (SIMLFS_INCLUDE_DIRS "@SIMLFS_INCLUDE_DIRS@")
+set (SIMLFS_LIBRARY_DIRS "@SIMLFS_LIB_DIR@")
+set (SIMLFS_BINARY_DIRS "@SIMLFS_BIN_DIR@")
 
-# Library dependencies for RMOL (contains definitions for the RMOL IMPORTED
+# Library dependencies for SIMLFS (contains definitions for the SIMLFS IMPORTED
 # targets)
-include ("@RMOL_CMAKE_DIR@/rmol-library-depends.cmake")
+include ("@SIMLFS_CMAKE_DIR@/simlfs-library-depends.cmake")
 
-# These are the RMOL IMPORTED targets, created by rmol-library-depends.cmake
-set (RMOL_LIBRARIES airraclib rmollib)
-set (RMOL_EXECUTABLES rmol)
+# These are the SIMLFS IMPORTED targets, created by simlfs-library-depends.cmake
+set (SIMLFS_LIBRARIES simlfs)
+set (SIMLFS_EXECUTABLES simlfs)
