--- origsrc/tacopie-3.2.0/CMakeLists.txt	2017-11-14 16:45:16.000000000 +0900
+++ src/tacopie-3.2.0/CMakeLists.txt	2018-12-01 12:02:12.575471800 +0900
@@ -153,6 +153,9 @@ IF (SELECT_TIMEOUT)
   set_target_properties(${PROJECT} PROPERTIES COMPILE_DEFINITIONS "__TACOPIE_TIMEOUT=${SELECT_TIMEOUT}")
 ENDIF(SELECT_TIMEOUT)
 
+IF (CYGWIN)
+  set_target_properties(${PROJECT} PROPERTIES SOVERSION 0)
+ENDIF()
 
 ###
 # install
@@ -161,8 +164,12 @@ ENDIF(SELECT_TIMEOUT)
 install(DIRECTORY DESTINATION ${CMAKE_BINARY_DIR}/lib/)
 install(DIRECTORY DESTINATION ${CMAKE_BINARY_DIR}/bin/)
 # install tacopie
-install (DIRECTORY ${CMAKE_BINARY_DIR}/lib/ DESTINATION lib USE_SOURCE_PERMISSIONS)
-install (DIRECTORY ${CMAKE_BINARY_DIR}/bin/ DESTINATION bin USE_SOURCE_PERMISSIONS)
+IF (CYGWIN)
+  install (TARGETS ${PROJECT} RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+ELSE()
+  install (DIRECTORY ${CMAKE_BINARY_DIR}/lib/ DESTINATION lib USE_SOURCE_PERMISSIONS)
+  install (DIRECTORY ${CMAKE_BINARY_DIR}/bin/ DESTINATION bin USE_SOURCE_PERMISSIONS)
+ENDIF()
 install (DIRECTORY ${TACOPIE_INCLUDES}/ DESTINATION include USE_SOURCE_PERMISSIONS)
 
 
@@ -183,3 +190,10 @@ IF (BUILD_TESTS)
                       GIT_REPOSITORY "https://github.com/google/googletest.git"
                       CMAKE_ARGS "-DCMAKE_INSTALL_PREFIX=${PROJECT_SOURCE_DIR}/deps")
 ENDIF(BUILD_TESTS)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/tacopie.pc.in ${CMAKE_BINARY_DIR}/tacopie.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/tacopie.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/tacopie-3.2.0/tacopie.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/tacopie-3.2.0/tacopie.pc.in	2018-12-01 12:02:12.542972700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: tacopie
+Description: C++ TCP Library
+Version: @VERSION@
+Libs: -L${libdir} -ltacopie
+Cflags: -I${includedir}
