--- origsrc/fntsample-release-5.3/CMakeLists.txt	2018-12-26 16:13:37.000000000 +0900
+++ src/fntsample-release-5.3/CMakeLists.txt	2020-07-07 14:33:49.468580200 +0900
@@ -88,4 +88,6 @@ install(FILES "${PROJECT_BINARY_DIR}/fnt
               "${PROJECT_BINARY_DIR}/pdfoutline.1"
         DESTINATION "${CMAKE_INSTALL_MANDIR}/man1")
 
-add_subdirectory(po)
+if (NOT CYGWIN) # ninja: error: dependency cycle: po/uk.gmo -> po/CMakeFiles/uk.gmo -> po/uk.gmo
+  add_subdirectory(po)
+endif()
