--- origsrc/givaro-3.7.1/configure.ac	2012-07-11 16:58:59.000000000 +0900
+++ src/givaro-3.7.1/configure.ac	2012-10-09 20:20:09.144575100 +0900
@@ -151,20 +151,6 @@ AC_C_BIGENDIAN(
   [AC_DEFINE(HAVE_LITTLE_ENDIAN, 1, [Define that architecture uses little endian storage])],
   [])
 
-echo "Creating symbolic link for compilation"
-if test ! -L src/kernel/system/givaro  ; then ln -s . src/kernel/system/givaro  ; fi
-if test ! -L src/kernel/bstruct/givaro ; then ln -s . src/kernel/bstruct/givaro ; fi
-if test ! -L src/kernel/integer/givaro ; then ln -s . src/kernel/integer/givaro ; fi
-if test ! -L src/kernel/memory/givaro  ; then ln -s . src/kernel/memory/givaro  ; fi
-if test ! -L src/kernel/rational/givaro; then ln -s . src/kernel/rational/givaro; fi
-if test ! -L src/kernel/zpz/givaro     ; then ln -s . src/kernel/zpz/givaro     ; fi
-if test ! -L src/kernel/gmp++/gmp++    ; then ln -s . src/kernel/gmp++/gmp++    ; fi
-if test ! -L src/library/matrix/givaro ; then ln -s . src/library/matrix/givaro ; fi
-if test ! -L src/library/poly1/givaro  ; then ln -s . src/library/poly1/givaro  ; fi
-if test ! -L src/library/tools/givaro  ; then ln -s . src/library/tools/givaro  ; fi
-if test ! -L src/library/vector/givaro ; then ln -s . src/library/vector/givaro ; fi
-
-
 # Create some useful data types of fixed, known lengths
 
 # case 1 in
@@ -277,3 +263,16 @@ tests/Makefile
 docs/Makefile
 ])
 AC_OUTPUT
+
+echo "Creating symbolic link for compilation"
+if test ! -L src/kernel/system/givaro  ; then ln -s $srcdir/src/kernel/system   src/kernel/system/givaro  ; fi
+if test ! -L src/kernel/bstruct/givaro ; then ln -s $srcdir/src/kernel/bstruct  src/kernel/bstruct/givaro ; fi
+if test ! -L src/kernel/integer/givaro ; then ln -s $srcdir/src/kernel/integer  src/kernel/integer/givaro ; fi
+if test ! -L src/kernel/memory/givaro  ; then ln -s $srcdir/src/kernel/memory   src/kernel/memory/givaro  ; fi
+if test ! -L src/kernel/rational/givaro; then ln -s $srcdir/src/kernel/rational src/kernel/rational/givaro; fi
+if test ! -L src/kernel/zpz/givaro     ; then ln -s $srcdir/src/kernel/zpz      src/kernel/zpz/givaro     ; fi
+if test ! -L src/kernel/gmp++/gmp++    ; then ln -s $srcdir/src/kernel/gmp++    src/kernel/gmp++/gmp++    ; fi
+if test ! -L src/library/matrix/givaro ; then ln -s $srcdir/src/library/matrix  src/library/matrix/givaro ; fi
+if test ! -L src/library/poly1/givaro  ; then ln -s $srcdir/src/library/poly1   src/library/poly1/givaro  ; fi
+if test ! -L src/library/tools/givaro  ; then ln -s $srcdir/src/library/tools   src/library/tools/givaro  ; fi
+if test ! -L src/library/vector/givaro ; then ln -s $srcdir/src/library/vector  src/library/vector/givaro ; fi
--- origsrc/givaro-3.7.1/src/Makefile.am	2012-07-11 16:59:26.000000000 +0900
+++ src/givaro-3.7.1/src/Makefile.am	2012-10-09 20:49:24.570465800 +0900
@@ -20,5 +20,5 @@ libgivaro_la_LIBADD= \
 	kernel/gmp++/libgmppp.la kernel/bstruct/libgivbstruct.la kernel/integer/libgivinteger.la kernel/memory/libgivmemory.la kernel/rational/libgivrational.la kernel/system/libgivsystem.la kernel/zpz/libgivzpz.la library/tools/libgivtools.la library/poly1/libgivpoly1.la
 
 # soname of libgivaro
-libgivaro_la_LDFLAGS = -version-info 5:1:5
+libgivaro_la_LDFLAGS = -version-info 5:1:5 -no-undefined
 libgivaro_la_LDFLAGS+=$(LDFLAGS)
--- origsrc/givaro-3.7.1/src/kernel/bstruct/Makefile.am	2012-05-07 21:41:22.000000000 +0900
+++ src/givaro-3.7.1/src/kernel/bstruct/Makefile.am	2012-10-09 21:07:12.689606500 +0900
@@ -7,7 +7,7 @@
 AM_CPPFLAGS=-I$(top_srcdir)
 
 AM_CXXFLAGS = @DEFAULT_CFLAGS@
-AM_CPPFLAGS+= -I$(top_srcdir)/src/kernel/system -I$(top_srcdir)/src/kernel/memory
+AM_CPPFLAGS+= -I$(top_builddir)/src/kernel/system -I$(top_builddir)/src/kernel/memory
 
 pkginclude_HEADERS=     \
 	givarray0.h		    \
--- origsrc/givaro-3.7.1/src/kernel/gmp++/Makefile.am	2011-10-03 21:33:36.000000000 +0900
+++ src/givaro-3.7.1/src/kernel/gmp++/Makefile.am	2012-10-09 20:59:12.882965800 +0900
@@ -9,7 +9,7 @@ AM_CPPFLAGS=-I$(top_srcdir) ${GMP_VERSIO
 pkgincludesubdir=$(includedir)/gmp++
 
 AM_CXXFLAGS=@DEFAULT_CFLAGS@
-AM_CPPFLAGS+= -I$(top_srcdir)/src/kernel/memory -I$(top_srcdir)/src/kernel/system
+AM_CPPFLAGS+= -I$(top_builddir)/src/kernel/memory -I$(top_builddir)/src/kernel/system
 
 
 noinst_LTLIBRARIES=libgmppp.la
--- origsrc/givaro-3.7.1/src/kernel/integer/Makefile.am	2012-05-07 21:41:22.000000000 +0900
+++ src/givaro-3.7.1/src/kernel/integer/Makefile.am	2012-10-09 21:06:17.204254900 +0900
@@ -7,7 +7,7 @@
 AM_CPPFLAGS=-I$(top_srcdir)  ${GMP_VERSION}
 
 AM_CXXFLAGS = @DEFAULT_CFLAGS@
-AM_CPPFLAGS+= -I$(top_srcdir)/src/kernel/memory  -I$(top_srcdir)/src/kernel/system -I$(top_srcdir)/src/kernel
+AM_CPPFLAGS+= -I$(top_builddir)/src/kernel/memory  -I$(top_builddir)/src/kernel/system -I$(top_srcdir)/src/kernel
 
 pkginclude_HEADERS=         \
 	givinteger.h		    \
--- origsrc/givaro-3.7.1/src/kernel/memory/Makefile.am	2012-05-07 21:41:22.000000000 +0900
+++ src/givaro-3.7.1/src/kernel/memory/Makefile.am	2012-10-09 20:57:30.006012700 +0900
@@ -7,7 +7,7 @@
 AM_CPPFLAGS=-I$(top_srcdir)
 
 AM_CXXFLAGS=@DEFAULT_CFLAGS@
-AM_CPPFLAGS+= -I$(top_srcdir)/src/kernel/system
+AM_CPPFLAGS+= -I$(top_builddir)/src/kernel/system
 
 pkginclude_HEADERS= \
 	givaromm.h     	\
--- origsrc/givaro-3.7.1/src/kernel/rational/Makefile.am	2012-05-07 21:41:22.000000000 +0900
+++ src/givaro-3.7.1/src/kernel/rational/Makefile.am	2012-10-09 21:08:12.502106500 +0900
@@ -7,7 +7,7 @@
 AM_CPPFLAGS=-I$(top_srcdir)  ${GMP_VERSION}
 
 AM_CXXFLAGS = @DEFAULT_CFLAGS@
-AM_CPPFLAGS+= -I$(top_srcdir)/src/kernel/integer -I$(top_srcdir)/src/kernel -I$(top_srcdir)/src/kernel/system -I$(top_srcdir)/src/kernel/memory
+AM_CPPFLAGS+= -I$(top_builddir)/src/kernel/integer -I$(top_srcdir)/src/kernel -I$(top_builddir)/src/kernel/system -I$(top_builddir)/src/kernel/memory
 
 pkginclude_HEADERS= \
 	givrational.h	\
--- origsrc/givaro-3.7.1/src/kernel/system/Makefile.am	2012-05-21 22:36:10.000000000 +0900
+++ src/givaro-3.7.1/src/kernel/system/Makefile.am	2012-10-09 20:52:08.121247100 +0900
@@ -7,7 +7,7 @@
 AM_CPPFLAGS=-I$(top_srcdir)
 
 AM_CXXFLAGS=@DEFAULT_CFLAGS@
-AM_CPPFLAGS+= -I$(top_srcdir)/src/kernel/memory
+AM_CPPFLAGS+= -I$(top_builddir)/src/kernel/memory
 
 pkginclude_HEADERS= \
 	givbasictype.h	\
--- origsrc/givaro-3.7.1/src/kernel/zpz/Makefile.am	2012-05-07 21:41:22.000000000 +0900
+++ src/givaro-3.7.1/src/kernel/zpz/Makefile.am	2012-10-09 21:09:29.097809600 +0900
@@ -7,7 +7,7 @@
 AM_CPPFLAGS=-I$(top_srcdir)  ${GMP_VERSION}
 
 AM_CXXFLAGS = @DEFAULT_CFLAGS@
-AM_CPPFLAGS+= -I$(top_srcdir)/src/kernel/integer -I$(top_srcdir)/src/kernel -I$(top_srcdir)/src/kernel/system -I$(top_srcdir)/src/kernel/memory -I$(top_srcdir)/src/kernel/bstruct
+AM_CPPFLAGS+= -I$(top_builddir)/src/kernel/integer -I$(top_srcdir)/src/kernel -I$(top_builddir)/src/kernel/system -I$(top_builddir)/src/kernel/memory -I$(top_builddir)/src/kernel/bstruct
 
 pkginclude_HEADERS=     \
 	StaticElement.h		\
--- origsrc/givaro-3.7.1/tests/Makefile.am	2011-09-13 17:32:41.000000000 +0900
+++ src/givaro-3.7.1/tests/Makefile.am	2012-10-09 21:11:03.508791300 +0900
@@ -9,10 +9,10 @@ check: $(BASE_TESTS)
 
 AM_CPPFLAGS=-I$(top_srcdir)
 AM_CXXFLAGS = @TESTS_CFLAGS@
-AM_CPPFLAGS += $(OPTFLAGS) $(GMP_CFLAGS) -I$(top_srcdir)/src/kernel/system -I$(top_srcdir)/src/kernel/memory -I$(top_srcdir)/src/kernel/zpz -I$(top_srcdir)/src/kernel/integer -I$(top_srcdir)/src/kernel -I$(top_srcdir)/src/library/poly1 -I$(top_srcdir)/src/kernel/bstruct -I$(top_srcdir)/src/library/tools
+AM_CPPFLAGS += $(OPTFLAGS) $(GMP_CFLAGS) -I$(top_builddir)/src/kernel/system -I$(top_builddir)/src/kernel/memory -I$(top_builddir)/src/kernel/zpz -I$(top_builddir)/src/kernel/integer -I$(top_srcdir)/src/kernel -I$(top_builddir)/src/library/poly1 -I$(top_builddir)/src/kernel/bstruct -I$(top_builddir)/src/library/tools
 
-LDADD = $(OPTLINKS) -L${top_srcdir}/src -L${top_srcdir}/src/.libs -lgivaro $(GMP_LIBS)
-AM_LDFLAGS=-static $(LDFLAGS)
+LDADD = $(OPTLINKS) $(top_builddir)/src/libgivaro.la $(GMP_LIBS)
+AM_LDFLAGS=$(LDFLAGS)
 
 BASIC_TESTS =          \
 		test-random    \
