--- origsrc/givaro-4.1.1/configure.ac	2019-06-07 22:23:45.000000000 +0900
+++ src/givaro-4.1.1/configure.ac	2019-06-29 08:11:47.521013300 +0900
@@ -189,7 +189,7 @@ for link in \
 	dir="$(dirname "$link")"
 	AS_MKDIR_P([$dir])
 	AS_IF([ test ! -L "$link" ],
-	  [ln -s "../../../$srcdir/$dir" "$link"])
+	  [ln -s "$srcdir/$dir" "$link"])
 done
 
 AC_DEFINE_UNQUOTED(INT8,  $GIVARO_INT8, Canonical 8-bit data type)
