--- origsrc/lightning-2.1.3/lib/Makefile.am	2019-09-18 00:51:45.000000000 +0900
+++ src/lightning-2.1.3/lib/Makefile.am	2019-10-04 13:14:50.781850600 +0900
@@ -16,7 +16,7 @@
 
 AM_CFLAGS = -I$(top_srcdir)/include -D_GNU_SOURCE $(LIGHTNING_CFLAGS)
 liblightning_LTLIBRARIES = liblightning.la
-liblightning_la_LDFLAGS = -version-info 1:0:0
+liblightning_la_LDFLAGS = -no-undefined -version-info 1:0:0
 
 if get_jit_size
 JIT_SIZE_PATH = "$(top_builddir)/jit_$(cpu)-sz.c"
