--- origsrc/lightning-2.2.0/check/lightning.c	2022-11-09 01:00:32.000000000 +0900
+++ src/lightning-2.2.0/check/lightning.c	2022-12-02 15:07:25.691544700 +0900
@@ -4195,16 +4195,6 @@ main(int argc, char *argv[])
     int			 opt_short;
     char		 cmdline[8192];
 
-#if defined(__CYGWIN__)
-    /* Cause a compile warning about redefinition without dllimport
-     * attribute, *but* cause correct linkage if liblightning.a is
-     * linked to binutils (that happens to have an internal
-     * getopt* implementation and an apparently conflicting
-     * optind global variable) */
-    extern int		 optind;
-    optind = 1;
-#endif
-
     progname = argv[0];
 
     init_jit(progname);
--- origsrc/lightning-2.2.0/lib/Makefile.am	2022-11-11 22:49:28.000000000 +0900
+++ src/lightning-2.2.0/lib/Makefile.am	2022-12-02 14:58:38.555855100 +0900
@@ -17,7 +17,7 @@
 AM_CFLAGS = -I$(top_builddir)/include -I$(top_srcdir)/include	\
 	-D_GNU_SOURCE $(LIGHTNING_CFLAGS)
 liblightning_LTLIBRARIES = liblightning.la
-liblightning_la_LDFLAGS = -version-info 2:0:0
+liblightning_la_LDFLAGS = -no-undefined -version-info 2:0:0
 
 AM_CPPFLAGS =
 if get_jit_size
