--- origsrc/mbt-3.6/src/Makefile.am	2020-04-15 17:37:45.000000000 +0900
+++ src/mbt-3.6/src/Makefile.am	2021-01-29 10:24:08.712562500 +0900
@@ -22,7 +22,7 @@ mbtg_SOURCES = Mbtg.cxx
 convert_SOURCES = convert.cxx
 
 lib_LTLIBRARIES = libmbt.la
-libmbt_la_LDFLAGS= -version-info 1:0:0
+libmbt_la_LDFLAGS= -no-undefined -version-info 1:0:0
 
 libmbt_la_SOURCES = MbtAPI.cxx Pattern.cxx TagLex.cxx Sentence.cxx \
 	RunTagger.cxx GenerateTagger.cxx Tagger.cxx
