--- origsrc/kona-Win64-20211225/Makefile	2021-12-17 03:43:18.000000000 +0900
+++ src/kona-Win64-20211225/Makefile	2022-07-01 22:57:42.944976900 +0900
@@ -83,7 +83,7 @@ OBJS= src/0.o src/bswap.o src/c.o src/ge
 PRODFLAGS = -O3
 endif
 
-ifeq (cygwin_nt-6.3,$(OS))
+ifeq (cygwin_nt,$(findstring cygwin_nt,$(OS)))
 OBJS= src/0.o src/bswap.o src/c.o src/getline.o src/mt.o src/p.o src/r.o \
       src/k.o src/kc.o src/kx.o src/kg.o src/km.o src/kn.o src/ko.o src/ks.o \
       src/v.o src/va.o src/vc.o src/vd.o src/vf.o src/vg.o src/vq.o
