--- origsrc/rw-0.10/Makefile.am	2020-09-14 15:48:42.000000000 +0900
+++ src/rw-0.10/Makefile.am	2026-01-31 14:43:37.774971800 +0900
@@ -17,3 +17,5 @@ endif
 
 dist_doc_DATA = README example_graphs/Hängematte example_graphs/grid example_graphs/igrid
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/rw-0.10/configure.ac	2025-12-17 04:50:30.000000000 +0900
+++ src/rw-0.10/configure.ac	2026-01-31 14:43:37.776984200 +0900
@@ -24,7 +24,7 @@ if test "x$executable" = "xtrue"; then
   PKG_CHECK_MODULES(IGRAPH, igraph >= 0.6)
 fi
 
-AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([rw.pc Makefile])
 
 AC_OUTPUT
 
--- origsrc/rw-0.10/rw.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/rw-0.10/rw.pc.in	2026-01-31 14:43:37.780607700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: rw
+Description: Compute rank-width decompositions of graphs
+Version: @VERSION@
+Libs: -L${libdir} -lrw
+Cflags: -I${includedir}
