--- origsrc/NGT-1.8.3/CMakeLists.txt	2019-11-18 16:08:18.000000000 +0900
+++ src/NGT-1.8.3/CMakeLists.txt	2019-12-05 14:55:42.003955100 +0900
@@ -67,3 +67,10 @@ if(${UNIX})
 	add_subdirectory("${PROJECT_SOURCE_DIR}/bin")
 	add_subdirectory("${PROJECT_SOURCE_DIR}/samples")
 endif( ${UNIX} )
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/ngt.pc.in ${CMAKE_BINARY_DIR}/ngt.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/ngt.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/NGT-1.8.3/lib/NGT/CMakeLists.txt	2019-11-18 16:08:18.000000000 +0900
+++ src/NGT-1.8.3/lib/NGT/CMakeLists.txt	2019-12-05 14:47:13.475067000 +0900
@@ -7,15 +7,19 @@ if( ${UNIX} )
 	file(GLOB HEADER_FILES *.h *.hpp)
 	file(GLOB NGTQ_HEADER_FILES NGTQ/*.h NGTQ/*.hpp)
 
-	add_library(ngtstatic STATIC ${NGT_SOURCES})
-	set_target_properties(ngtstatic PROPERTIES OUTPUT_NAME ngt) 
-	set_target_properties(ngtstatic PROPERTIES COMPILE_FLAGS "-fPIC")
-	target_link_libraries(ngtstatic)
+	if (NOT BUILD_SHARED_LIBS)
+		add_library(ngtstatic STATIC ${NGT_SOURCES})
+		set_target_properties(ngtstatic PROPERTIES OUTPUT_NAME ngt) 
+		set_target_properties(ngtstatic PROPERTIES COMPILE_FLAGS "-fPIC")
+		target_link_libraries(ngtstatic)
+	endif()
 
 	add_library(ngt SHARED ${NGT_SOURCES})
 	set_target_properties(ngt PROPERTIES VERSION ${ngt_VERSION}) 
 	set_target_properties(ngt PROPERTIES SOVERSION ${ngt_SOVERSION}) 
-	add_dependencies(ngt ngtstatic) 
+	if (NOT BUILD_SHARED_LIBS)
+		add_dependencies(ngt ngtstatic) 
+	endif()
 	if(${APPLE})
 		target_link_libraries(ngt gomp)
 	else(${APPLE})
@@ -24,14 +28,22 @@ if( ${UNIX} )
 
 	add_custom_command(OUTPUT command DEPENDS ${NGT_SOURCES} WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR} COMMAND sh ${PROJECT_SOURCE_DIR}/utils/mk_version_defs_h.sh ${PROJECT_SOURCE_DIR} version_defs.h)
 	add_custom_target(versiondef DEPENDS command)
-	add_dependencies(ngtstatic versiondef)
+	if (NOT BUILD_SHARED_LIBS)
+		add_dependencies(ngtstatic versiondef)
+	endif()
 
 	install(TARGETS
 		ngt
-		ngtstatic
 		RUNTIME DESTINATION bin
 		LIBRARY DESTINATION lib
 		ARCHIVE DESTINATION lib)
+	if (NOT BUILD_SHARED_LIBS)	
+		install(TARGETS
+			ngtstatic
+			RUNTIME DESTINATION bin
+			LIBRARY DESTINATION lib
+			ARCHIVE DESTINATION lib)
+	endif()
 	install(FILES ${HEADER_FILES} ${CMAKE_CURRENT_BINARY_DIR}/defines.h  ${CMAKE_CURRENT_BINARY_DIR}/version_defs.h DESTINATION include/NGT)
 	install(FILES ${NGTQ_HEADER_FILES} DESTINATION include/NGT/NGTQ)
 endif()
--- origsrc/NGT-1.8.3/ngt.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/NGT-1.8.3/ngt.pc.in	2019-12-05 14:58:52.589305600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: ngt
+Description: Neighborhood Graph and Tree for Indexing High-dimensional Data
+Version: @ngt_VERSION@
+Libs: -L${libdir} -lngt
+Cflags: -I${includedir}
