--- origsrc/zfp-0.5.5/fortran/CMakeLists.txt	2019-05-06 09:11:11.000000000 +0900
+++ src/zfp-0.5.5/fortran/CMakeLists.txt	2019-10-12 10:50:50.803738700 +0900
@@ -15,3 +15,9 @@ set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran
 
 add_library(zFORp zfp.f)
 target_link_libraries(zFORp PRIVATE zfp)
+
+if(CYGWIN)
+  set_target_properties(zFORp PROPERTIES SOVERSION 0)
+  install(TARGETS zFORp RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+  install(FILES ${CMAKE_BINARY_DIR}/modules/zforp_module.mod DESTINATION include)
+endif()
