--- origsrc/ortp-5.3.96/CMakeLists.txt	2024-07-18 17:21:48.000000000 +0900
+++ src/ortp-5.3.96/CMakeLists.txt	2024-12-02 08:59:47.736284900 +0900
@@ -22,7 +22,7 @@
 
 cmake_minimum_required(VERSION 3.22)
 
-project(Ortp VERSION 5.3.0)
+project(Ortp VERSION 5.3.96)
 
 
 set(ORTP_MAJOR_VERSION ${PROJECT_VERSION_MAJOR})
@@ -212,7 +212,7 @@ if (ENABLE_PACKAGE_SOURCE)
 endif()
 
 include(CMakePackageConfigHelpers)
-set(CMAKE_MODULES_INSTALL_DIR "${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}/cmake")
+option(CMAKE_MODULES_INSTALL_DIR "${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}/cmake")
 configure_package_config_file("cmake/${PROJECT_NAME}Config.cmake.in" "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
 	INSTALL_DESTINATION "${CMAKE_MODULES_INSTALL_DIR}"
 	NO_SET_AND_CHECK_MACRO
--- origsrc/ortp-5.3.96/src/rtpsession_inet.c	2024-07-18 17:21:48.000000000 +0900
+++ src/ortp-5.3.96/src/rtpsession_inet.c	2024-12-02 08:41:30.976938700 +0900
@@ -542,9 +542,10 @@ int rtp_session_set_pktinfo(RtpSession *
 
 #ifdef IPV6_RECVPKTINFO
 		optname = IPV6_RECVPKTINFO;
-#else
+#elif defined(IPV6_RECVDSTADDR)
 		optname = IPV6_RECVDSTADDR;
 #endif
+#if defined(IPV6_RECVPKTINFO) || defined(IPV6_RECVDSTADDR)
 		retval = setsockopt(session->rtp.gs.socket, IPPROTO_IPV6, optname, optval, optlen);
 		if (retval < 0) {
 			ortp_warning("Fail to set IPv6 packet info on RTP socket: %s.", getSocketError());
@@ -553,8 +554,8 @@ int rtp_session_set_pktinfo(RtpSession *
 		if (retval < 0) {
 			ortp_warning("Fail to set IPv6 packet info on RTCP socket: %s.", getSocketError());
 		}
+#endif
 	}
-
 	return retval;
 }
 
@@ -1228,6 +1229,7 @@ static int rtp_sendmsg(ortp_socket_t soc
 		}
 	}
 #endif
+#if !defined(__CYGWIN__)
 #ifdef IP_PKTINFO
 	if (m->recv_addr.family == AF_INET || useV4 == TRUE) { // Add IPV4 to the message control
 		struct in_pktinfo *pktinfo;
@@ -1241,6 +1243,7 @@ static int rtp_sendmsg(ortp_socket_t soc
 		cmsg = CMSG_NXTHDR(&msg, cmsg);
 	}
 #endif
+#endif
 
 #ifdef IPV6_RECVDSTADDR
 	if (m->recv_addr.family == AF_INET6 && !IN6_IS_ADDR_UNSPECIFIED(&m->recv_addr.addr.ipi6_addr) &&
