--- origsrc/ortp-1.0.2/src/rtpsession_inet.c	2017-07-20 20:35:30.000000000 +0900
+++ src/ortp-1.0.2/src/rtpsession_inet.c	2018-08-05 14:59:40.128674700 +0900
@@ -463,6 +463,7 @@ int rtp_session_set_pktinfo(RtpSession *
 		ortp_warning ("Fail to set IPv4 packet info on RTCP socket: %s.", getSocketError());
 	}
 
+#if !defined(__CYGWIN__)
 	if (session->rtp.gs.sockfamily != AF_INET) {
 #if defined(_WIN32) || defined(_WIN32_WCE)
 		memset(optval, activate, sizeof(optval));
@@ -482,7 +483,7 @@ int rtp_session_set_pktinfo(RtpSession *
 			ortp_warning("Fail to set IPv6 packet info on RTCP socket: %s.", getSocketError());
 		}
 	}
-
+#endif
 	return retval;
 }
 
