--- origsrc/ortp-4.3.0/CMakeLists.txt	2019-11-21 00:35:54.000000000 +0900
+++ src/ortp-4.3.0/CMakeLists.txt	2020-02-18 11:15:16.117317700 +0900
@@ -223,9 +223,11 @@ install(FILES
 	"${CMAKE_CURRENT_BINARY_DIR}/${EXPORT_TARGET_NAME}ConfigVersion.cmake"
 	DESTINATION ${CONFIG_PACKAGE_LOCATION}
 )
-install(FILES 	"${CMAKE_CURRENT_SOURCE_DIR}/README.md" 
+if(NOT CYGWIN)
+	install(FILES 	"${CMAKE_CURRENT_SOURCE_DIR}/README.md" 
 	"${CMAKE_CURRENT_SOURCE_DIR}/CHANGELOG.md" 
 	"${CMAKE_CURRENT_SOURCE_DIR}/LICENSE.txt" 
 	"${CMAKE_CURRENT_SOURCE_DIR}/AUTHORS.md" 
 	DESTINATION "${CMAKE_INSTALL_DATADIR}/doc/ortp-${ORTP_VERSION}"
-)
+	)
+endif()
--- origsrc/ortp-4.3.0/src/rtpsession_inet.c	2019-11-21 00:35:54.000000000 +0900
+++ src/ortp-4.3.0/src/rtpsession_inet.c	2020-02-18 11:12:07.109469600 +0900
@@ -478,6 +478,7 @@ int rtp_session_set_pktinfo(RtpSession *
 		ortp_warning ("Fail to set IPv4 packet info on RTCP socket: %s.", getSocketError());
 	}
 
+#if !defined(__CYGWIN__)
 	if (session->rtp.gs.sockfamily != AF_INET) {
 #if defined(_WIN32) || defined(_WIN32_WCE)
 		memset(optval, activate, sizeof(optval));
@@ -497,6 +498,7 @@ int rtp_session_set_pktinfo(RtpSession *
 			ortp_warning("Fail to set IPv6 packet info on RTCP socket: %s.", getSocketError());
 		}
 	}
+#endif
 
 	return retval;
 }
