--- origsrc/ortp-4.3.3/CMakeLists.txt	2020-07-27 18:41:40.000000000 +0900
+++ src/ortp-4.3.3/CMakeLists.txt	2020-08-21 00:36:56.203868800 +0900
@@ -21,7 +21,7 @@
 ############################################################################
 
 cmake_minimum_required(VERSION 3.1)
-project(oRTP VERSION 4.3.0)
+project(oRTP VERSION 4.3.3)
 
 
 set(ORTP_MAJOR_VERSION ${PROJECT_VERSION_MAJOR})
@@ -231,5 +231,5 @@ install(FILES 	"${CMAKE_CURRENT_SOURCE_D
 	"${CMAKE_CURRENT_SOURCE_DIR}/CHANGELOG.md" 
 	"${CMAKE_CURRENT_SOURCE_DIR}/LICENSE.txt" 
 	"${CMAKE_CURRENT_SOURCE_DIR}/AUTHORS.md" 
-	DESTINATION "${CMAKE_INSTALL_DATADIR}/doc/ortp-${ORTP_DOC_VERSION}"
+	DESTINATION "${CMAKE_INSTALL_DATADIR}/doc/ortp"
 )
--- origsrc/ortp-4.3.3/src/rtpsession_inet.c	2020-07-27 18:41:40.000000000 +0900
+++ src/ortp-4.3.3/src/rtpsession_inet.c	2020-08-21 00:36:16.464930100 +0900
@@ -483,6 +483,7 @@ int rtp_session_set_pktinfo(RtpSession *
 		memset(optval, activate, sizeof(optval));
 #endif
 
+#if !defined(__CYGWIN__)
 #ifdef IPV6_RECVPKTINFO
 		optname = IPV6_RECVPKTINFO;
 #else
@@ -496,6 +497,7 @@ int rtp_session_set_pktinfo(RtpSession *
 		if (retval < 0) {
 			ortp_warning("Fail to set IPv6 packet info on RTCP socket: %s.", getSocketError());
 		}
+#endif
 	}
 
 	return retval;
