--- origsrc/ortp-0.24.1/ortp.doxygen	2015-03-31 23:26:50.000000000 +0900
+++ src/ortp-0.24.1/ortp.doxygen	2015-04-17 21:55:18.004851600 +0900
@@ -72,13 +72,13 @@ WARN_LOGFILE           =
 #---------------------------------------------------------------------------
 # configuration options related to the input files
 #---------------------------------------------------------------------------
-INPUT                  = ./src ./include/ortp
+INPUT                  = /cygdrive/c/Users/fd0/src/ortp/ortp-0.24.1-1bl1.x86_64/src/ortp-0.24.1/src /cygdrive/c/Users/fd0/src/ortp/ortp-0.24.1-1bl1.x86_64/src/ortp-0.24.1/include/ortp
 FILE_PATTERNS          = *.c *.h
 RECURSIVE              = NO
 EXCLUDE                = 
 EXCLUDE_SYMLINKS       = NO
 EXCLUDE_PATTERNS       = 
-EXAMPLE_PATH           = ./src/tests
+EXAMPLE_PATH           = /cygdrive/c/Users/fd0/src/ortp/ortp-0.24.1-1bl1.x86_64/src/ortp-0.24.1/src/tests
 EXAMPLE_PATTERNS       = *.c
 EXAMPLE_RECURSIVE      = NO
 IMAGE_PATH             = 
