--- origsrc/clustal-omega-1.2.1/src/Makefile.am	2012-06-27 23:48:35.000000000 +0900
+++ src/clustal-omega-1.2.1/src/Makefile.am	2015-05-03 22:34:21.523742100 +0900
@@ -13,11 +13,11 @@ AM_CFLAGS = @AM_CFLAGS@ @OPENMP_CFLAGS@
 lib_LTLIBRARIES = libclustalo.la
 
 libclustalo_la_SOURCES = clustal-omega.c
-libclustalo_la_LIBADD = clustal/libclustal.la \
+libclustalo_la_LIBADD = -lstdc++ clustal/libclustal.la \
 	hhalign/libhhalign.la \
 	squid/libsquid.la	\
 	kmpp/libkmpp.la 
-libclustalo_la_LDFLAGS = -static
+libclustalo_la_LDFLAGS = -no-undefined
 library_includedir=$(includedir)/clustalo/
 library_include_HEADERS = clustal-omega-config.h clustal-omega.h
 
--- origsrc/clustal-omega-1.2.1/src/clustal/Makefile.am	2011-04-12 23:35:01.000000000 +0900
+++ src/clustal-omega-1.2.1/src/clustal/Makefile.am	2015-05-03 22:32:59.615676200 +0900
@@ -1,4 +1,5 @@
 AM_CFLAGS = @OPENMP_CFLAGS@ @AM_CFLAGS@
+AM_CPPFLAGS = -I$(top_srcdir)/src
 AM_LDFLAGS = @AM_LDFLAGS@
 
 noinst_LTLIBRARIES = libclustal.la
@@ -35,4 +36,3 @@ library_include_HEADERS = \
 	tree.h \
 	util.h \
 	weights.h
-
