--- origsrc/moab-5.2.1/itaps/imesh/Makefile.am	2020-08-22 13:11:37.000000000 +0900
+++ src/moab-5.2.1/itaps/imesh/Makefile.am	2020-10-13 04:43:47.422218500 +0900
@@ -78,6 +78,8 @@ libiMesh_la_LIBADD = $(top_builddir)/src
 
 libiMesh_la_LDFLAGS = -version-info @LIBTOOL_VERSION@
 
+libiMesh_la_LDFLAGS = -no-undefined
+
 # Automake doesn't seem to have a directory defined for
 # platform-dependent data (or include) files. So put
 # in $(libdir).  Define a $(cfgdir) to get around automake's
--- origsrc/moab-5.2.1/src/Makefile.am	2020-08-22 13:11:37.000000000 +0900
+++ src/moab-5.2.1/src/Makefile.am	2020-10-13 04:43:47.433215900 +0900
@@ -411,6 +411,8 @@ if HAVE_ZOLTAN
   libMOAB_la_LIBADD += $(ZOLTAN_LIBS) $(ZOLTAN_LIB_FLAGS)
 endif
 
+libMOAB_la_LDFLAGS = -no-undefined
+
 if HAVE_CGM
   AM_CPPFLAGS += @CGM_CPPFLAGS@
 endif
