--- origsrc/moab-5.1.0/itaps/imesh/Makefile.am	2019-01-06 19:50:20.000000000 +0900
+++ src/moab-5.1.0/itaps/imesh/Makefile.am	2019-10-25 23:19:36.781844600 +0900
@@ -76,6 +76,8 @@ libiMesh_la_includedir = $(includedir)
 
 libiMesh_la_LIBADD = $(top_builddir)/src/libMOAB.la
 
+libiMesh_la_LDFLAGS = -no-undefined
+
 # Automake doesn't seem to have a directory defined for
 # platform-dependent data (or include) files. So put 
 # in $(libdir).  Define a $(cfgdir) to get around automake's
--- origsrc/moab-5.1.0/src/Makefile.am	2019-01-06 19:50:20.000000000 +0900
+++ src/moab-5.1.0/src/Makefile.am	2019-10-25 23:16:03.933635700 +0900
@@ -405,6 +405,8 @@ if HAVE_ZOLTAN
   libMOAB_la_LIBADD += $(ZOLTAN_LIBS) $(ZOLTAN_LIB_FLAGS)
 endif
 
+libMOAB_la_LDFLAGS = -no-undefined
+
 if HAVE_CGM
   AM_CPPFLAGS += @CGM_CPPFLAGS@ 
 endif
