--- origsrc/libbitcoin-network-3.6.0/Makefile.am	2019-08-29 22:54:08.000000000 +0900
+++ src/libbitcoin-network-3.6.0/Makefile.am	2019-09-22 21:00:30.576064200 +0900
@@ -59,6 +59,7 @@ src_libbitcoin_network_la_SOURCES = \
     src/sessions/session_manual.cpp \
     src/sessions/session_outbound.cpp \
     src/sessions/session_seed.cpp
+src_libbitcoin_network_la_LDFLAGS = -no-undefined
 
 # local: test/libbitcoin-network-test
 #------------------------------------------------------------------------------
