--- origsrc/rcli-2.0.1/CMakeLists.txt	2025-02-22 18:38:45.000000000 +0900
+++ src/rcli-2.0.1/CMakeLists.txt	2025-03-29 12:23:12.550895600 +0900
@@ -86,3 +86,10 @@ endif()
 set(CPACK_PACKAGE_VENDOR "RickaPrincy <rckprincy@gmail.com>")
 set(CPACK_PACKAGE_DESCRIPTION_SUMMARY Starter of styled cli for c++)
 include(CPack)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/rcli.pc.in ${CMAKE_BINARY_DIR}/rcli.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/rcli.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/rcli-2.0.1/PKGBUILD	2025-02-22 18:38:45.000000000 +0900
+++ src/rcli-2.0.1/PKGBUILD	2025-03-29 12:10:06.634506000 +0900
@@ -9,7 +9,7 @@ url="https://github.com/RickaPrincy/rcli
 license=('MIT')
 source=("rcli@$pkgver.tar.gz::https://github.com/RickaPrincy/rcli/releases/download/v2.0.1/rcli@$pkgver.tar.gz")
 
-sha256sums=("4397884e1a6c5b2d4b947e1b53b985c2d3115713807fe104c92d71690043dd0e")
+sha256sums=("SHASUM")
 
 package() {
     echo "Installing to \"$pkgdir/usr/\""
--- origsrc/rcli-2.0.1/rcli.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/rcli-2.0.1/rcli.pc.in	2025-03-29 12:23:27.456025500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: rcli
+Description: Starter of styled cli for c++
+Version: @RCLI_LIB_VERSION@
+Libs: -L${libdir} -lrcli
+Cflags: -I${includedir}
