--- origsrc/libg19draw-1.0.0/Makefile.am	2010-04-11 12:16:34.000000000 +0900
+++ src/libg19draw-1.0.0/Makefile.am	2013-10-04 08:42:06.321790700 +0900
@@ -4,5 +4,5 @@ INCLUDES = $(DEPS_CFLAGS) $(FreeType2_CF
 LLIBDIR = @LIBDIR@ 
 lib_LTLIBRARIES = libg19draw.la
 libg19draw_la_SOURCES = libg19draw.h libg19draw.c
-libg19draw_la_LDFLAGS = -version-info 1:0:0
+libg19draw_la_LDFLAGS = -no-undefined -version-info 1:0:0
 include_HEADERS = libg19draw.h
