--- origsrc/libOPNMIDI-1.2.0/CMakeLists.txt	2018-04-25 05:23:54.000000000 +0900
+++ src/libOPNMIDI-1.2.0/CMakeLists.txt	2018-06-14 08:22:32.981716700 +0900
@@ -162,6 +162,9 @@ if(libOPNMIDI_SHARED)
     target_include_directories(OPNMIDI_shared PUBLIC ${libOPNMIDI_SOURCE_DIR}/include)
     set_legacy_standard(OPNMIDI_shared)
     list(APPEND libOPNMIDI_INSTALLS OPNMIDI_shared)
+    if(CYGWIN)
+        set_target_properties(OPNMIDI_shared PROPERTIES SOVERSION 0)
+    endif()
 endif()
 
 if(NOT libOPNMIDI_STATIC AND NOT libOPNMIDI_SHARED)
@@ -278,3 +281,10 @@ message("libOPNMIDI_STATIC        = ${li
 message("libOPNMIDI_SHARED        = ${libOPNMIDI_SHARED}")
 message("WITH_UNIT_TESTS          = ${WITH_UNIT_TESTS}")
 
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libOPNMIDI.pc.in ${CMAKE_BINARY_DIR}/libOPNMIDI.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libOPNMIDI.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libOPNMIDI-1.2.0/libOPNMIDI.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libOPNMIDI-1.2.0/libOPNMIDI.pc.in	2018-06-14 08:22:32.934837800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libOPNMIDI
+Description: MIDI play library with OPN2 (YM2612) emulator
+Version: @VERSION@
+Libs: -L${libdir} -lOPNMIDI
+Cflags: -I${includedir}
