--- origsrc/opensc-0.12.1/etc/opensc.conf.in	2011-05-18 02:07:00.000000000 +0900
+++ src/opensc-0.12.1/etc/opensc.conf.in	2011-05-31 10:37:46.509208900 +0900
@@ -30,7 +30,7 @@ app default {
 
 	# CT-API module configuration.
 	reader_driver ctapi {
-		# module /usr/local/towitoko/lib/libtowitoko.so {
+		# module /usr/lib/ctapi/libtowitoko.so {
 			# CT-API ports:
 			# 0..3		COM1..4
 			# 4		Printer
