--- origsrc/multicharge-0.5.0/meson.build	2025-11-10 21:57:52.000000000 +0900
+++ src/multicharge-0.5.0/meson.build	2026-01-04 20:48:32.531187100 +0900
@@ -74,7 +74,7 @@ if install
     install_dir: get_option('datadir')/'licenses'/meson.project_name()
   )
 
-  module_id = meson.project_name() / fc_id + '-' + fc.version()
+  module_id = meson.project_name() / 'modules'
   meson.add_install_script(
     find_program(files('config'/'install-mod.py')),
     get_option('includedir') / module_id,
