--- origsrc/tomlplusplus-3.4.0/include/toml++/impl/preprocessor.hpp	2023-10-13 22:39:42.000000000 +0900
+++ src/tomlplusplus-3.4.0/include/toml++/impl/preprocessor.hpp	2026-01-02 12:38:25.664537500 +0900
@@ -241,7 +241,7 @@
 //# OS
 //#=====================================================================================================================
 
-#if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__) || defined(__CYGWIN__)
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__)
 #define TOML_WINDOWS 1
 #else
 #define TOML_WINDOWS 0
