--- origsrc/libtwitcurl/CMakeLists.txt	2012-08-26 15:27:37.000000000 +0900
+++ src/libtwitcurl/CMakeLists.txt	2015-01-10 11:25:26.846976600 +0900
@@ -1,4 +1,10 @@
 set(twitSrcs base64.cpp HMAC_SHA1.cpp oauthlib.cpp SHA1.cpp urlencode.cpp twitcurl.cpp)
+FIND_PACKAGE(CURL)
 FIND_PACKAGE(PkgConfig)
 include_directories (${PKGS_INCLUDE_DIRS}) 
-add_library(twitcurl STATIC ${twitSrcs})
+add_library(twitcurl SHARED ${twitSrcs})
+SET_TARGET_PROPERTIES(twitcurl PROPERTIES SOVERSION 0)
+target_link_libraries(twitcurl ${CURL_LIBRARY})
+install(TARGETS twitcurl RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+INSTALL(FILES twitcurl.h DESTINATION include/libtwitcurl)
+INSTALL(FILES oauthlib.h DESTINATION include/libtwitcurl)
