--- origsrc/liboauth-0.9.4/configure.ac	2011-01-19 01:14:58.000000000 +0900
+++ src/liboauth-0.9.4/configure.ac	2011-02-13 12:51:30.706828700 +0900
@@ -38,10 +38,8 @@ AC_PROG_INSTALL
 AC_PROG_CC
 AC_PROG_MAKE_SET
 AC_PROG_LN_S
-AC_PROG_LIBTOOL
-AM_PROG_LIBTOOL
 AM_PROG_CC_C_O
-AC_LIBTOOL_WIN32_DLL
+LT_INIT(win32-dll)
 AC_CONFIG_MACRO_DIR([m4])
 
 AC_HEADER_STDC
@@ -77,7 +75,7 @@ AC_SUBST(EXESUF)
 
 dnl *** misc complier/linker flags ***
 LIBOAUTH_CFLAGS="-Wall"
-LIBOAUTH_LDFLAGS="${LIBOAUTH_CFLAGS} -export-symbols-regex '^oauth_.*'"
+LIBOAUTH_LDFLAGS="${LIBOAUTH_CFLAGS} -no-undefined -export-symbols-regex '^oauth_.*'"
 #LIBOAUTH_CFLAGS="${LIBOAUTH_CFLAGS} -g -posix -std=c99 -pedantic"
 
 AC_MSG_CHECKING([if -Wl,--as-needed works])
