--- origsrc/libshveu-1.7.2/configure.ac	2011-11-01 23:53:38.000000000 +0900
+++ src/libshveu-1.7.2/configure.ac	2024-11-05 03:19:52.168025800 +0900
@@ -230,6 +230,8 @@ else
 AC_SUBST(SHLIB_VERSION_ARG)
 AC_SUBST(SHARED_VERSION_INFO)
 
+AM_CONDITIONAL(IS_LINUX, [test "$host_os" = "linux"])
+
 AC_OUTPUT([
 Makefile
 doc/Makefile
--- origsrc/libshveu-1.7.2/doc/Makefile.am	2011-11-01 23:53:38.000000000 +0900
+++ src/libshveu-1.7.2/doc/Makefile.am	2024-11-05 03:25:47.074238700 +0900
@@ -1,8 +1,12 @@
 docdir=$(prefix)/share/doc/@PACKAGE@
 
+if IS_LINUX
 man_MANS = shveu-convert.1
 
 EXTRA_DIST = $(man_MANS) Doxyfile.in
+else
+EXTRA_DIST = shveu-convert.1  Doxyfile.in
+endif
 
 # ensure doxygen-build.stamp is included,  or else the documentation will
 # not be built or distributed
--- origsrc/libshveu-1.7.2/src/Makefile.am	2011-11-01 23:53:38.000000000 +0900
+++ src/libshveu-1.7.2/src/Makefile.am	2024-11-05 03:19:52.168025800 +0900
@@ -1 +1,5 @@
-SUBDIRS = libshveu tools
+SUBDIRS = libshveu
+
+if IS_LINUX
+SUBDIRS += tools
+endif
--- origsrc/libshveu-1.7.2/src/libshveu/Makefile.am	2011-11-01 23:53:38.000000000 +0900
+++ src/libshveu-1.7.2/src/libshveu/Makefile.am	2024-11-05 03:19:52.168025800 +0900
@@ -17,5 +17,5 @@ libshveu_la_SOURCES = \
 	veu.c
 
 libshveu_la_CFLAGS = $(UIOMUX_CFLAGS)
-libshveu_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
+libshveu_la_LDFLAGS = -no-undefined -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
 libshveu_la_LIBADD = $(UIOMUX_LIBS)
