--- origsrc/libmail-0.3/Makefile.am	2009-06-08 00:11:04.000000000 +0900
+++ src/libmail-0.3/Makefile.am	2025-01-17 08:42:57.777307900 +0900
@@ -1,3 +1,5 @@
-ACLOCAL_AMFLAGS="-I m4"
 SUBDIRS = man libmail 
 EXTRA_DIST = TODO 
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libmail-0.3/configure.ac	2009-07-20 06:19:20.000000000 +0900
+++ src/libmail-0.3/configure.ac	2025-01-17 08:42:57.792934900 +0900
@@ -141,4 +141,5 @@ AC_TYPE_SIZE_T
 AC_FUNC_MALLOC
 AC_CHECK_FUNCS([memset socket strchr strdup strrchr strstr])
 AC_CONFIG_FILES([libmail/Makefile Makefile man/Makefile])
+AC_CONFIG_FILES([libmail.pc])
 AC_OUTPUT
--- origsrc/libmail-0.3/libmail/Makefile.am	2009-07-17 03:23:45.000000000 +0900
+++ src/libmail-0.3/libmail/Makefile.am	2025-01-17 08:34:57.772174900 +0900
@@ -1,10 +1,12 @@
 lib_LTLIBRARIES = libmail.la
 libmailincludedir = $(includedir)/libmail
-libmail_la_LDFLAGS = -version-info 2:1:0
+libmail_la_LDFLAGS = -no-undefined -version-info 2:1:0
 libmail_la_SOURCES = imap4.c mbox.c network.c pop3.c maildir.c error.c commons.c libmail.c
 libmailinclude_HEADERS = imap4.h libmail.h mbox.h pop3.h maildir.h error.h
 noinst_HEADERS = network.h commons.h libmail_intl.h
 
+INCLUDES = -I$(top_srcdir)
+
 if LIBMAIL_USE_SASL
 libmail_la_SOURCES += sasl.c
 noinst_HEADERS += sasl.h
--- origsrc/libmail-0.3/libmail/network.c	2009-06-09 01:10:46.000000000 +0900
+++ src/libmail-0.3/libmail/network.c	2025-01-17 08:34:57.772174900 +0900
@@ -36,7 +36,9 @@ along with libmail.  If not, see <http:/
 
 static void init_criteria(struct addrinfo *hints, short ai_family) {
 	memset(hints, 0, sizeof(struct addrinfo));
+#if defined(AI_IDN)
 	hints->ai_flags = AI_IDN;
+#endif
 	if (ai_family == LIBMAIL_IPV4)
 		hints->ai_family = AF_INET;
 	else
--- origsrc/libmail-0.3/libmail/tls.c	2009-06-14 23:23:20.000000000 +0900
+++ src/libmail-0.3/libmail/tls.c	2025-01-17 08:34:57.772174900 +0900
@@ -42,8 +42,6 @@ int initialize_tls(libmail_remote_server
 		goto error;
 	if ((ret = gnutls_set_default_priority(server->tls_session)) != 0)
 		goto error;
-	if ((ret = gnutls_certificate_type_set_priority(server->tls_session, cert)) != 0)
-		goto error;
 	if ((ret = gnutls_credentials_set(server->tls_session, GNUTLS_CRD_CERTIFICATE, cred)) != 0)
 		goto error; 
 	gnutls_transport_set_ptr (server->tls_session, (gnutls_transport_ptr_t) server->sfd);
--- origsrc/libmail-0.3/libmail.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libmail-0.3/libmail.pc.in	2025-01-17 08:42:57.746041900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libmail
+Description: Mail handling library
+Version: @VERSION@
+Libs: -L${libdir} -lmail
+Cflags: -I${includedir}
