--- origsrc/lambdacommon-1.6.4/CMakeLists.txt	2018-06-24 22:20:44.000000000 +0900
+++ src/lambdacommon-1.6.4/CMakeLists.txt	2018-06-27 18:25:07.702192500 +0900
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.8)
+cmake_minimum_required(VERSION 3.6.2)
 project(λcommon)
 
 option(LAMBDACOMMON_BUILD_STATIC "Build static libraries" OFF)
@@ -7,7 +7,9 @@ option(LAMBDACOMMON_BUILD_TESTS "Build t
 
 include(GenerateExportHeader)
 
-set(CMAKE_CXX_STANDARD 17)
+if (NOT CYGWIN)
+    set(CMAKE_CXX_STANDARD 17)
+endif()
 
 if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Ofast -march=native -mtune=native -ffast-math -D__extern_always_inline=\"extern __always_inline\"")
@@ -44,6 +46,9 @@ if (LAMBDACOMMON_BUILD_STATIC)
 endif ()
 # Build the shared library
 add_library(lambdacommon SHARED ${HEADERS_FILES} ${SOURCES_FILES})
+if (CYGWIN)
+    set_target_properties(lambdacommon PROPERTIES SOVERSION 0)
+endif()
 # Generate the export header and include it.
 GENERATE_EXPORT_HEADER(lambdacommon
         BASE_NAME lambdacommon
@@ -93,3 +98,10 @@ endif ()
 if (LAMBDACOMMON_BUILD_TESTS)
     add_subdirectory(tests)
 endif ()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/lambdacommon.pc.in ${CMAKE_BINARY_DIR}/lambdacommon.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/lambdacommon.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/lambdacommon-1.6.4/lambdacommon.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/lambdacommon-1.6.4/lambdacommon.pc.in	2018-06-27 18:25:07.655564300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: lambdacommon
+Description: Library written in C++ with common features
+Version: @VERSION@
+Libs: -L${libdir} -llambdacommon
+Cflags: -I${includedir}
