--- origsrc/profanity-0.8.1/configure.ac	2020-02-08 05:47:41.000000000 +0900
+++ src/profanity-0.8.1/configure.ac	2020-10-21 13:26:35.289856300 +0900
@@ -99,9 +99,6 @@ else
 fi
 
 # c
-if test "x$PLATFORM" = xcygwin; then
-    AM_CONDITIONAL([BUILD_C_API], [false])
-else
     LT_INIT
     if test "x$enable_plugins" = xno; then
         AM_CONDITIONAL([BUILD_C_API], [false])
@@ -116,7 +113,6 @@ else
     else
         AM_CONDITIONAL([BUILD_C_API], [false])
     fi
-fi
 
 # threading
 ACX_PTHREAD([], [AC_MSG_ERROR([pthread is required])])
--- origsrc/profanity-0.8.1/src/profanity.c	2020-02-08 05:47:41.000000000 +0900
+++ src/profanity-0.8.1/src/profanity.c	2020-10-21 16:00:56.905602900 +0900
@@ -93,6 +93,8 @@ static void _connect_default(const char
 static gboolean cont = TRUE;
 static gboolean force_quit = FALSE;
 
+pthread_mutex_t lock;
+
 void
 prof_run(char *log_level, char *account_name, char *config_file)
 {
--- origsrc/profanity-0.8.1/src/profanity.h	2020-02-08 05:47:41.000000000 +0900
+++ src/profanity-0.8.1/src/profanity.h	2020-10-21 16:00:35.981899900 +0900
@@ -43,6 +43,6 @@
 void prof_run(char *log_level, char *account_name, char * config_file);
 void prof_set_quit(void);
 
-pthread_mutex_t lock;
+extern pthread_mutex_t lock;
 
 #endif
--- origsrc/profanity-0.8.1/src/tools/http_upload.c	2020-02-08 05:47:41.000000000 +0900
+++ src/profanity-0.8.1/src/tools/http_upload.c	2020-10-21 16:00:17.773394900 +0900
@@ -65,6 +65,7 @@ struct curl_data_t {
     size_t size;
 };
 
+GSList *upload_processes;
 
 static int
 _xferinfo(void *userdata, curl_off_t dltotal, curl_off_t dlnow, curl_off_t ultotal, curl_off_t ulnow)
--- origsrc/profanity-0.8.1/src/tools/http_upload.h	2020-02-08 05:47:41.000000000 +0900
+++ src/profanity-0.8.1/src/tools/http_upload.h	2020-10-21 16:00:21.400406300 +0900
@@ -57,7 +57,7 @@ typedef struct http_upload_t {
     int cancel;
 } HTTPUpload;
 
-GSList *upload_processes;
+extern GSList *upload_processes;
 
 void* http_file_put(void *userdata);
 
--- origsrc/profanity-0.8.1/src/xmpp/omemo.c	2020-02-08 05:47:41.000000000 +0900
+++ src/profanity-0.8.1/src/xmpp/omemo.c	2020-10-21 15:48:50.660213600 +0900
@@ -33,6 +33,9 @@
  *
  */
 
+#include <stdlib.h>
+#include <string.h>
+
 #include <glib.h>
 
 #include "log.h"
