--- origsrc/profanity-0.4.6/Makefile.am	2015-02-26 09:51:03.000000000 +0900
+++ src/profanity-0.4.6/Makefile.am	2015-04-18 00:56:35.351967100 +0900
@@ -104,7 +104,7 @@ otr3_sources = \
 otr4_sources = \
 	src/otr/otrlib.h src/otr/otrlibv4.c src/otr/otr.h src/otr/otr.c
 
-themes_sources = themes/*
+themes_sources = $(top_srcdir)/themes/*
 
 script_sources = bootstrap.sh configure-debug install-all.sh
 
@@ -138,6 +138,8 @@ man_MANS = $(man_sources)
 
 EXTRA_DIST = $(man_sources) $(themes_sources) $(script_sources) profrc.example LICENSE.txt
 
+AM_CPPFLAGS = -I$(top_srcdir)/src $(glib_CFLAGS)
+
 if INCLUDE_GIT_VERSION
 EXTRA_DIST += .git/HEAD .git/index
 
