--- origsrc/profanity-0.5.1/Makefile.am	2017-01-29 06:09:53.000000000 +0900
+++ src/profanity-0.5.1/Makefile.am	2017-12-08 23:25:33.311000200 +0900
@@ -169,9 +169,12 @@ otr3_sources = \
 otr4_sources = \
 	src/otr/otrlib.h src/otr/otrlibv4.c src/otr/otr.h src/otr/otr.c
 
+AM_CPPFLAGS = -I$(top_srcdir)/src $(glib_CFLAGS) $(GTK_CFLAGS)
+
 if BUILD_PYTHON_API
 core_sources += $(python_sources)
 unittest_sources += $(python_sources)
+AM_CPPFLAGS += $(PYTHON_CPPFLAGS)
 endif
 
 if BUILD_C_API
@@ -182,9 +185,9 @@ endif
 otr_unittest_sources = \
 	tests/unittests/otr/stub_otr.c
 
-themes_sources = themes/*
+themes_sources = $(top_srcdir)/themes/*
 
-icons_sources = icons/*
+icons_sources = $(top_srcdir)/icons/*
 
 script_sources = bootstrap.sh configure-debug
 
