--- origsrc/blend2d/CMakeLists.txt	2025-12-31 20:48:40.000000000 +0900
+++ src/blend2d/CMakeLists.txt	2025-12-31 22:45:43.814067400 +0900
@@ -581,6 +581,10 @@ function(blend2d_add_target target targe
     add_library(${target} ${target_type} ${ARG_SOURCES})
   endif()
 
+  if (DEFINED SOVERSION)
+    set_target_properties(${target} PROPERTIES SOVERSION ${SOVERSION})
+  endif()
+
   set_target_properties(${target} PROPERTIES DEFINE_SYMBOL "" CXX_VISIBILITY_PRESET hidden)
   target_include_directories(${target} BEFORE PRIVATE ${ARG_INCLUDE_DIRS})
   target_compile_options(${target} PRIVATE ${ARG_CFLAGS} ${BLEND2D_SANITIZE_CFLAGS} $<IF:$<CONFIG:Debug>,${ARG_CFLAGS_DBG},${ARG_CFLAGS_REL}>)
@@ -1267,3 +1271,10 @@ else()
 endif()
 
 message(STATUS "[blend2d] == Configuring done ==")
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/blend2d.pc.in ${CMAKE_BINARY_DIR}/blend2d.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/blend2d.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/blend2d/blend2d/core/api-build_p.h	2025-12-31 20:48:40.000000000 +0900
+++ src/blend2d/blend2d/core/api-build_p.h	2025-12-31 20:51:04.701868400 +0900
@@ -152,7 +152,8 @@ extern "C" void __tsan_release(void *add
       defined(__DragonFly__) || \
       defined(__FreeBSD__  ) || \
       defined(__NetBSD__   ) || \
-      defined(__OpenBSD__  )
+      defined(__OpenBSD__  ) || \
+      defined(__CYGWIN__)
     #define BL_FILE64_API(NAME) NAME
   #else
     #define BL_FILE64_API(NAME) NAME##64
--- origsrc/blend2d/blend2d.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/blend2d/blend2d.pc.in	2025-12-31 22:21:23.192156800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: blend2d
+Description: 2D Vector Graphics Engine Powered by a JIT Compiler
+Version: @VERSION@
+Libs: -L${libdir} -lblend2d
+Cflags: -I${includedir}
