--- origsrc/cjose-0.6.2.3/test/Makefile.am	2024-04-24 16:07:39.000000000 +0900
+++ src/cjose-0.6.2.3/test/Makefile.am	2024-05-04 03:13:43.818353000 +0900
@@ -7,7 +7,9 @@ TESTS = check_cjose
 
 check_PROGRAMS = $(TESTS)
 check_cjose_CFLAGS = @CHECK_CFLAGS@ -I$(top_builddir)/include \
-                     -I$(top_builddir)/src -I$(top_srcdir)/src
+                     -I$(top_builddir)/src -I$(top_srcdir)/src \
+                     -I$(top_builddir)/include -I$(top_srcdir)/include \
+                     -I$(top_builddir)/include/cjose
 check_cjose_LDADD = $(top_builddir)/src/libcjose.la @CHECK_LIBS@
 check_cjose_SOURCES = check_cjose.c \
                       check_version.c \
