diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x libtool.m4 -x ltoptions.m4 -x ltsugar.m4 -x ltversion.m4 -x lt~obsolete.m4 -x ltmain.sh -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x Makefile.in origsrc/libmcrypto-0.8.0+r4734/Makefile.am src/libmcrypto-0.8.0+r4734/Makefile.am
--- origsrc/libmcrypto-0.8.0+r4734/Makefile.am	2014-06-24 01:03:52.000000000 +0900
+++ src/libmcrypto-0.8.0+r4734/Makefile.am	2016-03-23 23:33:12.951308700 +0900
@@ -20,13 +20,10 @@ BUILD_SRC =
 libmcrypto_la_SOURCES = $(BUILD_SRC)
 libmcrypto_la_CPPFLAGS = $(MINISIP_CFLAGS)
 libmcrypto_la_LDFLAGS = $(MINISIP_LIBRARY_LDFLAGS) 
-libmcrypto_la_LIBADD = \
+libmcrypto_la_LIBADD = -lstdc++ -lgcrypt \
 	source/libmcrypto_core.la \
 	source/libmcrypto_imp.la \
 	$(MINISIP_LIBS)
-libmcrypto_la_LINK = $(LIBTOOL) --tag=CXX --mode=link $(CXX)	\
-	$(AM_CXXFLAGS) $(CXXFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
-
 
 if OS_WIN
 res_obj = win32/$(PACKAGE)-res.o
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x libtool.m4 -x ltoptions.m4 -x ltsugar.m4 -x ltversion.m4 -x lt~obsolete.m4 -x ltmain.sh -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x Makefile.in origsrc/libmcrypto-0.8.0+r4734/source/gnutls/cert.cxx src/libmcrypto-0.8.0+r4734/source/gnutls/cert.cxx
--- origsrc/libmcrypto-0.8.0+r4734/source/gnutls/cert.cxx	2014-06-24 01:03:52.000000000 +0900
+++ src/libmcrypto-0.8.0+r4734/source/gnutls/cert.cxx	2016-03-23 23:24:15.923638000 +0900
@@ -34,6 +34,7 @@
 
 #include<iostream>
 #include<stdio.h>
+#include <unistd.h>
 
 using namespace std;
 
@@ -624,7 +625,7 @@ void GtlsCertificate::getDer( unsigned c
 	int ret;
 
 	ret = gnutls_x509_crt_export( cert, GNUTLS_X509_FMT_DER,
-			output, length );
+			output, (size_t*)length );
 	if( ret == GNUTLS_E_SHORT_MEMORY_BUFFER ){
 		throw CertificateException(
 			"Given buffer is to short" );
