--- origsrc/binaryen-version_122/CMakeLists.txt	2025-02-04 06:54:02.000000000 +0900
+++ src/binaryen-version_122/CMakeLists.txt	2025-03-12 06:50:36.117810000 +0900
@@ -447,6 +447,9 @@ if(BUILD_STATIC_LIB)
 else()
   message(STATUS "Building libbinaryen as shared library.")
   add_library(binaryen SHARED)
+  if (DEFINED SOVERSION)
+    set_target_properties(binaryen PROPERTIES SOVERSION "${SOVERSION}")
+  endif()
 endif()
 target_link_libraries(binaryen Threads::Threads)
 if(BUILD_LLVM_DWARF)
@@ -585,3 +588,10 @@ if(EMSCRIPTEN)
 endif()
 
 configure_file(scripts/binaryen-lit.in ${CMAKE_BINARY_DIR}/bin/binaryen-lit @ONLY)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/binaryen.pc.in ${CMAKE_BINARY_DIR}/binaryen.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/binaryen.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/binaryen-version_122/binaryen.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/binaryen-version_122/binaryen.pc.in	2025-03-12 06:51:08.954224000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: binaryen
+Description: Compiler infrastructure and toolchain library for WebAssembly
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lbinaryen
+Cflags: -I${includedir}
