--- origsrc/binaryen-version_97/CMakeLists.txt	2020-09-18 00:57:50.000000000 +0900
+++ src/binaryen-version_97/CMakeLists.txt	2020-09-23 20:04:44.349584700 +0900
@@ -304,6 +304,9 @@ if(BUILD_STATIC_LIB)
 else()
   message(STATUS "Building libbinaryen as shared library.")
   add_library(binaryen SHARED ${binaryen_SOURCES} ${binaryen_objs})
+  if(CYGWIN)
+    set_target_properties(binaryen PROPERTIES SOVERSION 0)
+  endif()
 endif()
 install(TARGETS binaryen
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
@@ -387,3 +390,10 @@ if(EMSCRIPTEN)
   set_property(TARGET binaryen_js PROPERTY CXX_STANDARD_REQUIRED ON)
   install(TARGETS binaryen_js DESTINATION ${CMAKE_INSTALL_BINDIR})
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/binaryen.pc.in ${CMAKE_BINARY_DIR}/binaryen.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/binaryen.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/binaryen-version_97/binaryen.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/binaryen-version_97/binaryen.pc.in	2020-09-23 20:04:44.261583300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: binaryen
+Description: Compiler infrastructure and toolchain library for WebAssembly
+Version: @VERSION@
+Libs: -L${libdir} -lbinaryen
+Cflags: -I${includedir}
