--- origsrc/libaxl-0.7.3/configure.ac	2022-08-24 21:15:04.000000000 +0900
+++ src/libaxl-0.7.3/configure.ac	2024-09-24 15:34:00.281861000 +0900
@@ -119,7 +119,7 @@ if test "$enable_py_axl_support" = "yes"
    python_dir_temp=`eval echo ${pythondir}`
    # check python_install_dir and pythondir maches
    if test "$python_dir_check" != "ok" ; then
-      AC_MSG_ERROR([ERROR: pythondir variable is poiting to an unusable location: $python_dir_temp. Error detected: $python_dir_check. Fix your autoconf install or disable python building, for that, pass --disable-py-axl to configure. This usually indicates a problem with automake package which provides the _AM_PATH_PYTHON_ macro.])
+      AC_MSG_ERROR([ERROR: pythondir variable is poiting to an unusable location: $python_dir_temp. Error detected: $python_dir_check. Fix your autoconf install or disable python building, for that, pass --disable-py-axl to configure. This usually indicates a problem with automake package which provides the macro.])
    fi
    echo "Python install dir OK: $python_dir_temp"
 fi
--- origsrc/libaxl-0.7.3/knife/Makefile.am	2022-08-24 21:15:04.000000000 +0900
+++ src/libaxl-0.7.3/knife/Makefile.am	2024-09-24 15:38:21.391204500 +0900
@@ -12,7 +12,7 @@ NS = -DAXL_NS_SUPPORT
 NS_LIB = $(top_builddir)/ns/libaxl-ns.la
 endif
 
-INCLUDES = -I$(top_srcdir)/src/ -I$(top_srcdir)/ns $(compiler_options) $(LOG) $(NS) -DVERSION=\"$(AXL_VERSION)\" -D__COMPILING_AXL_KNIFE__ 	-D__axl_disable_broken_bool_def__  $(EXARG_FLAGS)
+INCLUDES = -I$(top_builddir)/src -I$(top_srcdir)/src/ -I$(top_srcdir)/ns $(compiler_options) $(LOG) $(NS) -DVERSION=\"$(AXL_VERSION)\" -D__COMPILING_AXL_KNIFE__ 	-D__axl_disable_broken_bool_def__  $(EXARG_FLAGS)
 
 # replace with bin_PROGRAMS to check performance
 bin_PROGRAMS = axl-knife
@@ -25,4 +25,4 @@ axl_knife_LDADD   = $(top_builddir)/src/
 # two files
 exarg.h exarg.c: 
 	test -e exarg.h || cp $(top_srcdir)/../libexarg/src/exarg.h .
-	test -e exarg.c || cp $(top_srcdir)/../libexarg/src/exarg.c .
\ No newline at end of file
+	test -e exarg.c || cp $(top_srcdir)/../libexarg/src/exarg.c .
--- origsrc/libaxl-0.7.3/ns/Makefile.am	2022-08-24 21:15:04.000000000 +0900
+++ src/libaxl-0.7.3/ns/Makefile.am	2024-09-24 15:37:50.842260500 +0900
@@ -4,7 +4,7 @@ if ENABLE_AXL_LOG
 LOG = -DSHOW_DEBUG_LOG
 endif
 
-INCLUDES = $(compiler_options) -g -I$(top_srcdir)/src -I$(top_srcdir) $(LIBRARIES_CFLAGS) -DVERSION=\""$(AXL_VERSION)"\" \
+INCLUDES = $(compiler_options) -g -I$(top_builddir)/src -I$(top_srcdir)/src -I$(top_srcdir) $(LIBRARIES_CFLAGS) -DVERSION=\""$(AXL_VERSION)"\" \
 	-DPACKAGE_DTD_DIR=\""$(datadir)"\" -DPACKAGE_TOP_DIR=\""$(top_srcdir)"\"  \
 	-D__axl_disable_broken_bool_def__ \
 	-DVERSION=\"$(AXL_VERSION)\" $(LOG)
--- origsrc/libaxl-0.7.3/src/Makefile.am	2022-08-24 21:15:04.000000000 +0900
+++ src/libaxl-0.7.3/src/Makefile.am	2024-09-24 15:36:40.062006300 +0900
@@ -4,7 +4,7 @@ if ENABLE_AXL_LOG
 LOG = -DSHOW_DEBUG_LOG
 endif
 
-INCLUDES = $(compiler_options) -g -I$(top_srcdir) $(LIBRARIES_CFLAGS) -DVERSION=\""$(AXL_VERSION)"\" \
+INCLUDES = $(compiler_options) -g -I$(builddir) -I$(top_srcdir) $(LIBRARIES_CFLAGS) -DVERSION=\""$(AXL_VERSION)"\" \
 	-DPACKAGE_DTD_DIR=\""$(datadir)"\" -DPACKAGE_TOP_DIR=\""$(top_srcdir)"\"  \
 	-D__axl_disable_broken_bool_def__ \
 	-DVERSION=\"$(AXL_VERSION)\" $(LOG)
--- origsrc/libaxl-0.7.3/test/Makefile.am	2022-08-24 21:15:04.000000000 +0900
+++ src/libaxl-0.7.3/test/Makefile.am	2024-09-24 15:42:34.599519700 +0900
@@ -53,7 +53,7 @@ BABEL = -DAXL_BABEL_SUPPORT
 BABEL_LIB = $(top_builddir)/babel/libaxl-babel.la
 endif
 
-INCLUDES = -DTEST_DIR=$(top_srcdir)/test -I$(top_srcdir)/src/ -I$(top_srcdir)/ns -I$(top_srcdir)/babel $(compiler_options) $(LOG) $(NS) $(BABEL) -DVERSION=\""$(AXL_VERSION)"\" -D__axl_disable_broken_bool_def__ 
+INCLUDES = -DTEST_DIR=$(top_srcdir)/test -I$(top_builddir)/src -I$(top_srcdir)/src/ -I$(top_srcdir)/ns -I$(top_srcdir)/babel $(compiler_options) $(LOG) $(NS) $(BABEL) -DVERSION=\""$(AXL_VERSION)"\" -D__axl_disable_broken_bool_def__ 
 
 # replace with bin_PROGRAMS to check performance
 noinst_PROGRAMS = test_01 test_01a test_02a test_03a
