--- origsrc/ltsv4c/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/ltsv4c/Makefile.am	2013-07-28 10:36:02.285500000 +0900
@@ -0,0 +1,16 @@
+
+check_PROGRAMS = test
+
+test_LDADD = libltsv4c.la
+test_SOURCES = test.c
+
+
+include_HEADERS = ltsv4c.h
+
+
+lib_LTLIBRARIES = libltsv4c.la
+
+libltsv4c_la_CFLAGS = -O3 -Wall -Wextra -std=c89 -pedantic-errors
+libltsv4c_la_LDFLAGS = -no-undefined
+libltsv4c_la_SOURCES = ltsv4c.c
+
--- origsrc/ltsv4c/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/ltsv4c/configure.ac	2013-07-28 10:30:15.358693300 +0900
@@ -0,0 +1,27 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([ltsv4c], [0+git20130208])
+AC_CONFIG_SRCDIR([ltsv4c.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stddef.h stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
