--- origsrc/BLAKE3-1.5.5/c/libblake3.pc.in	2024-11-27 11:16:20.000000000 +0900
+++ src/BLAKE3-1.5.5/c/libblake3.pc.in	2024-12-13 05:02:27.527173200 +0900
@@ -1,12 +1,12 @@
-prefix="@CMAKE_INSTALL_PREFIX@"
-exec_prefix="${prefix}"
-libdir="@PKG_CONFIG_INSTALL_LIBDIR@"
-includedir="@PKG_CONFIG_INSTALL_INCLUDEDIR@"
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=@PKG_CONFIG_INSTALL_LIBDIR@
+includedir=@PKG_CONFIG_INSTALL_INCLUDEDIR@
 
 Name: @PROJECT_NAME@
 Description: @PROJECT_DESCRIPTION@
 Version: @PROJECT_VERSION@
 
 Requires:
-Libs: -L"${libdir}" -lblake3
-Cflags: -I"${includedir}" @BLAKE3_PKGCONFIG_CFLAGS@
+Libs: -L${libdir} -lblake3
+Cflags: -I${includedir} @BLAKE3_PKGCONFIG_CFLAGS@
