--- origsrc/BLAKE3-1.5.4/c/libblake3.pc.in	2024-08-20 02:41:29.000000000 +0900
+++ src/BLAKE3-1.5.4/c/libblake3.pc.in	2024-10-23 00:47:13.147167600 +0900
@@ -1,12 +1,12 @@
-prefix="@CMAKE_INSTALL_PREFIX@"
-exec_prefix="${prefix}"
-libdir="${prefix}/@CMAKE_INSTALL_LIBDIR@"
-includedir="${prefix}/@CMAKE_INSTALL_INCLUDEDIR@"
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
 
 Name: @PROJECT_NAME@
 Description: @PROJECT_DESCRIPTION@
 Version: @PROJECT_VERSION@
 
 Requires:
-Libs: -L"${libdir}" -lblake3
-Cflags: -I"${includedir}" @BLAKE3_PKGCONFIG_CFLAGS@
+Libs: -L${libdir} -lblake3
+Cflags: -I${includedir} @BLAKE3_PKGCONFIG_CFLAGS@
