diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x '*.c' -x '*.h' -x '*.pc' origsrc/libcmime-0.1.16/src/CMakeLists.txt src/libcmime-0.1.16/src/CMakeLists.txt
--- origsrc/libcmime-0.1.16/src/CMakeLists.txt	2013-10-15 18:08:38.000000000 +0900
+++ src/libcmime-0.1.16/src/CMakeLists.txt	2015-08-07 22:18:31.540521800 +0900
@@ -44,7 +44,8 @@ set_property(TARGET cmime PROPERTY VERSI
 set_property(TARGET cmime PROPERTY SOVERSION ${CMIME_SO_VERSION})
 
 install(TARGETS cmime
-	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
 	PUBLIC_HEADER DESTINATION include
 	PRIVATE_HEADER DESTINATION include/cmime
 )
