--- origsrc/tinyframe/src/Makefile.am	2026-02-15 09:18:02.000000000 +0900
+++ src/tinyframe/src/Makefile.am	2026-02-15 09:18:24.254462800 +0900
@@ -10,7 +10,7 @@ lib_LTLIBRARIES = libtinyframe.la
 libtinyframe_la_SOURCES = tinyframe.c
 nobase_include_HEADERS = tinyframe/tinyframe.h
 nobase_nodist_include_HEADERS = tinyframe/version.h
-libtinyframe_la_LDFLAGS = -version-info $(TINYFRAME_LIBRARY_VERSION)
+libtinyframe_la_LDFLAGS = -no-undefined -version-info $(TINYFRAME_LIBRARY_VERSION)
 EXTRA_DIST += tinyframe/version.h.in
 
 if ENABLE_GCOV
--- origsrc/tinyframe/src/test/Makefile.am	2026-02-15 09:18:02.000000000 +0900
+++ src/tinyframe/src/test/Makefile.am	2026-02-15 09:22:20.540882600 +0900
@@ -10,16 +10,16 @@ EXTRA_DIST = $(TESTS)
 
 test1_SOURCES = test1.c
 test1_LDADD = ../libtinyframe.la
-test1_LDFLAGS = -static
+test1_LDFLAGS = 
 EXTRA_DIST += test.fstrm
 
 test2_SOURCES = test2.c
 test2_LDADD = ../libtinyframe.la
-test2_LDFLAGS = -static
+test2_LDFLAGS = 
 
 test3_SOURCES = test3.c
 test3_LDADD = ../libtinyframe.la
-test3_LDFLAGS = -static
+test3_LDFLAGS = 
 
 if ENABLE_GCOV
 gcov-local:
