--- origsrc/gsignond-plugin-lastfm-0.2.0/src/meson.build	2018-05-06 08:25:38.000000000 +0900
+++ src/gsignond-plugin-lastfm-0.2.0/src/meson.build	2025-12-12 05:18:55.754925400 +0900
@@ -8,6 +8,7 @@ lastfm_plugin = shared_module(meson.proj
     lastfm_files,
     dependencies : dependencies,
     install: true,
-    install_dir : gplugins_dir
+    install_dir : gplugins_dir,
+    name_prefix: ''
 )
 
