--- origsrc/isa-l-2.30.0/Makefile.am	2020-11-07 10:08:16.000000000 +0900
+++ src/isa-l-2.30.0/Makefile.am	2021-10-25 18:30:43.184787100 +0900
@@ -83,7 +83,7 @@ libisal_la_SOURCES += ${lsrc_base_aliase
 endif
 
 nobase_include_HEADERS = isa-l.h
-libisal_la_LDFLAGS = $(AM_LDFLAGS) \
+libisal_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) \
 	-version-info $(LIBISAL_CURRENT):$(LIBISAL_REVISION):$(LIBISAL_AGE)
 libisal_la_LIBADD = ${noinst_LTLIBRARIES}
 
