--- origsrc/isa-l-2.32.0/Makefile.am	2026-03-04 22:56:50.000000000 +0900
+++ src/isa-l-2.32.0/Makefile.am	2026-06-10 04:48:06.127995400 +0900
@@ -85,7 +85,7 @@ libisal_la_SOURCES += ${lsrc_base_aliase
 endif
 
 nobase_include_HEADERS = isa-l.h
-libisal_la_LDFLAGS = $(AM_LDFLAGS) \
+libisal_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) \
 	-version-info $(LIBISAL_CURRENT):$(LIBISAL_REVISION):$(LIBISAL_AGE)
 libisal_la_CFLAGS = $(AM_CFLAGS) -DISAL_DEPRECATED_INTERNAL
 libisal_la_LIBADD = ${noinst_LTLIBRARIES}
--- origsrc/isa-l-2.32.0/programs/Makefile.am	2026-03-04 22:56:50.000000000 +0900
+++ src/isa-l-2.32.0/programs/Makefile.am	2026-06-10 04:48:06.132996500 +0900
@@ -34,5 +34,5 @@ programs_igzip_LDADD = libisal.la
 dist_man_MANS = programs/igzip.1
 other_src    += programs/igzip.1.h2m
 
-programs/igzip.1: % : %.h2m programs/igzip_cli.c programs/igzip
+programs/igzip.1: % : %.h2m programs/igzip_cli.c programs/igzip$(EXEEXT)
 	-help2man -o $@ -i $< -N ./programs/igzip
