--- origsrc/isa-l-2.31.1/Makefile.am	2025-01-09 00:33:59.000000000 +0900
+++ src/isa-l-2.31.1/Makefile.am	2025-01-18 07:37:44.673941800 +0900
@@ -88,7 +88,7 @@ libisal_la_SOURCES += ${lsrc_base_aliase
 endif
 
 nobase_include_HEADERS = isa-l.h
-libisal_la_LDFLAGS = $(AM_LDFLAGS) \
+libisal_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) \
 	-version-info $(LIBISAL_CURRENT):$(LIBISAL_REVISION):$(LIBISAL_AGE)
 libisal_la_LIBADD = ${noinst_LTLIBRARIES}
 
--- origsrc/isa-l-2.31.1/programs/Makefile.am	2025-01-09 00:33:59.000000000 +0900
+++ src/isa-l-2.31.1/programs/Makefile.am	2025-01-18 07:41:46.565844200 +0900
@@ -34,5 +34,5 @@ programs_igzip_LDADD = libisal.la
 dist_man_MANS = programs/igzip.1
 other_src    += programs/igzip.1.h2m
 
-programs/igzip.1: % : %.h2m programs/igzip_cli.c programs/igzip
+programs/igzip.1: % : %.h2m programs/igzip_cli.c programs/igzip$(EXEEXT)
 	-help2man -o $@ -i $< -N ./programs/igzip
