--- origsrc/isa-l-2.31.0/Makefile.am	2024-01-19 03:27:24.000000000 +0900
+++ src/isa-l-2.31.0/Makefile.am	2024-03-20 14:30:31.184695000 +0900
@@ -88,7 +88,7 @@ libisal_la_SOURCES += ${lsrc_base_aliase
 endif
 
 nobase_include_HEADERS = isa-l.h
-libisal_la_LDFLAGS = $(AM_LDFLAGS) \
+libisal_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) \
 	-version-info $(LIBISAL_CURRENT):$(LIBISAL_REVISION):$(LIBISAL_AGE)
 libisal_la_LIBADD = ${noinst_LTLIBRARIES}
 
