--- origsrc/ButterflyPACK-3.2.0/MPI_DUMMY/CMakeLists.txt	2024-09-20 03:47:45.000000000 +0900
+++ src/ButterflyPACK-3.2.0/MPI_DUMMY/CMakeLists.txt	2025-01-16 08:37:29.123266800 +0900
@@ -17,11 +17,13 @@ target_include_directories(mpi_dummy PUB
 include(GNUInstallDirs)
 
 install(TARGETS mpi_dummy EXPORT MYBPACKTargets
-  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
   PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 )
 
+set_target_properties(mpi_dummy PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${VERSION_MAJOR})
+
 install(FILES ${headers} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 
--- origsrc/ButterflyPACK-3.2.0/SRC/BPACK_defs.f90	2024-09-20 03:47:45.000000000 +0900
+++ src/ButterflyPACK-3.2.0/SRC/BPACK_defs.f90	2025-01-16 08:37:43.520371000 +0900
@@ -60,7 +60,7 @@ module BPACK_DEFS
 
     !>**** the version numbers are automatically replaced with those defined in CMakeList.txt
     integer, parameter:: BPACK_MAJOR_VERSION = 3
-    integer, parameter:: BPACK_MINOR_VERSION = 1
+    integer, parameter:: BPACK_MINOR_VERSION = 2
     integer, parameter:: BPACK_PATCH_VERSION = 0
 
     !>**** common parameters
--- origsrc/ButterflyPACK-3.2.0/SRC/CMakeLists.txt	2024-09-20 03:47:45.000000000 +0900
+++ src/ButterflyPACK-3.2.0/SRC/CMakeLists.txt	2025-01-16 08:37:44.629113100 +0900
@@ -60,7 +60,7 @@ target_include_directories(butterflypack
 include(GNUInstallDirs)
 
 install(TARGETS butterflypack EXPORT MYBPACKTargets
-  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
   PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 )
@@ -70,4 +70,4 @@ install(FILES ${headers} DESTINATION ${C
 
 #SET(CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_INSTALL_INCLUDEDIR})
 install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
-        FILES_MATCHING PATTERN "*.mod")
\ No newline at end of file
+        FILES_MATCHING PATTERN "*.mod")
