--- origsrc/unqlite-1.1.9/CMakeLists.txt	2019-05-22 22:35:14.000000000 +0900
+++ src/unqlite-1.1.9/CMakeLists.txt	2024-10-24 14:39:13.327678200 +0900
@@ -17,8 +17,19 @@ SET(SOURCES_UNQLITE
   unqlite.c
 )
 
-SET(UNQLITE_STATIC_LIB unqlite)
-ADD_LIBRARY(${UNQLITE_STATIC_LIB} STATIC ${HEADERS_UNQLITE} ${SOURCES_UNQLITE})
+SET(UNQLITE_LIB unqlite)
+ADD_LIBRARY(${UNQLITE_LIB} ${HEADERS_UNQLITE} ${SOURCES_UNQLITE})
 
-INSTALL(TARGETS ${UNQLITE_STATIC_LIB} COMPONENT devel ARCHIVE DESTINATION lib)
+if (DEFINED SOVERSION)
+  set_target_properties(${UNQLITE_LIB} PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
+INSTALL(TARGETS ${UNQLITE_LIB} COMPONENT devel RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 INSTALL(FILES ${HEADERS_UNQLITE} COMPONENT devel DESTINATION include)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/unqlite.pc.in ${CMAKE_BINARY_DIR}/unqlite.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/unqlite.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/unqlite-1.1.9/unqlite.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/unqlite-1.1.9/unqlite.pc.in	2024-10-24 14:38:19.759737900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: unqlite
+Description: Embeddable NoSQL Database Engine
+Version: @VERSION@
+Libs: -L${libdir} -lunqlite
+Cflags: -I${includedir}
