--- origsrc/libcamsrcjpegenc/src/Makefile.am	2012-03-01 20:09:14.000000000 +0900
+++ src/libcamsrcjpegenc/src/Makefile.am	2012-03-01 20:10:31.244366500 +0900
@@ -14,6 +14,8 @@ libcamsrcjpegenc_la_CFLAGS = -I$(srcdir)
 libcamsrcjpegenc_la_LIBADD = $(GLIB_LIBS) \
                              $(GMODULE_LIBS)
 
+libcamsrcjpegenc_la_LDFLAGS = -no-undefined
+
 if USE_DLOG
 libcamsrcjpegenc_la_CFLAGS += $(DLOG_CFLAGS) -DUSE_DLOG
 
