--- origsrc/rcssserver-14.0.3/configure.ac	2010-04-23 12:14:55.000000000 +0900
+++ src/rcssserver-14.0.3/configure.ac	2010-12-03 02:30:12.457365900 +0900
@@ -155,10 +155,9 @@ AX_PROG_BISON([],
 ##################################################
 
 AC_ENABLE_SHARED
-AC_DISABLE_STATIC
 #LT_INIT([shared disable-static])
 #LT_LANG([C++])
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 #AC_ENABLE_SHARED
 #AC_DISABLE_STATIC
 AC_SUBST(LIBTOOL_DEPS)
--- origsrc/rcssserver-14.0.3/rcssbase/conf/Makefile.am	2009-12-21 14:47:17.000000000 +0900
+++ src/rcssserver-14.0.3/rcssbase/conf/Makefile.am	2010-12-03 09:59:43.173084800 +0900
@@ -7,7 +7,7 @@ librcssconfparser_la_SOURCES = \
 	statushandler.cpp \
 	streamstatushandler.cpp
 
-librcssconfparser_la_LDFLAGS = -version-info 3:0:0
+librcssconfparser_la_LDFLAGS = -version-info 3:0:0 -no-undefined
 #    1. Start with version information of `0:0:0' for each libtool library.
 #
 #    2. Update the version information only immediately before a public
@@ -27,7 +27,7 @@ librcssconfparser_la_LDFLAGS = -version-
 #    6. If any interfaces have been removed since the last public release,
 #       then set AGE to 0.
 
-#librcssconfparser_la_LIBADD = $(BOOST_FILESYSTEM_LIB)
+librcssconfparser_la_LIBADD = $(BOOST_SYSTEM_LIB) $(BOOST_FILESYSTEM_LIB)
 
 rcssbase_confdir = $(includedir)/rcssbase/conf
 
--- origsrc/rcssserver-14.0.3/rcssbase/gzip/Makefile.am	2009-12-21 14:47:17.000000000 +0900
+++ src/rcssserver-14.0.3/rcssbase/gzip/Makefile.am	2010-12-03 02:31:04.762577100 +0900
@@ -7,7 +7,7 @@ librcssgz_la_SOURCES= \
 	gzstream.hpp \
 	gzstream.cpp
 
-librcssgz_la_LDFLAGS = -version-info 1:0:0
+librcssgz_la_LDFLAGS = -version-info 1:0:0 -no-undefined
 #    1. Start with version information of `0:0:0' for each libtool library.
 #
 #    2. Update the version information only immediately before a public
--- origsrc/rcssserver-14.0.3/rcssbase/net/Makefile.am	2009-12-21 14:47:17.000000000 +0900
+++ src/rcssserver-14.0.3/rcssbase/net/Makefile.am	2010-12-03 02:31:15.998733900 +0900
@@ -8,7 +8,7 @@ librcssnet_la_SOURCES = \
 	tcpsocket.cpp
 
 #librcssnet_la_LDFLAGS = -version-info 1:1:0 -L../error
-librcssnet_la_LDFLAGS = -version-info 1:1:0
+librcssnet_la_LDFLAGS = -version-info 1:1:0 -no-undefined
 #    1. Start with version information of `0:0:0' for each libtool library.
 #
 #    2. Update the version information only immediately before a public
--- origsrc/rcssserver-14.0.3/src/Makefile.am	2009-12-21 14:47:12.000000000 +0900
+++ src/rcssserver-14.0.3/src/Makefile.am	2010-12-03 10:12:54.781361600 +0900
@@ -25,7 +25,7 @@ librcssclangparser_la_SOURCES = \
 	rule.cpp \
 	rcssexceptions.cpp
 
-librcssclangparser_la_LDFLAGS = -version-info 3:0:1
+librcssclangparser_la_LDFLAGS = -version-info 3:0:1 -no-undefined
 #    Changed from 2.0.0 to 3.0.1 at 9.3.5 for addition of buildCondList
 #
 #    1. Start with version information of `0:0:0' for each libtool library.
@@ -47,6 +47,8 @@ librcssclangparser_la_LDFLAGS = -version
 #    6. If any interfaces have been removed since the last public release,
 #       then set AGE to 0.
 
+librcssclangparser_la_LIBADD = $(BOOST_SYSTEM_LIB)
+
 pkginclude_HEADERS = \
 	clangparser.h \
 	coach_lang_tok.h \
--- origsrc/rcssserver-14.0.3/src/referee.h	2009-12-21 14:47:12.000000000 +0900
+++ src/rcssserver-14.0.3/src/referee.h	2010-12-03 10:25:06.763900800 +0900
@@ -32,6 +32,8 @@
 #include <set>
 #include <vector>
 
+#include <ctime>
+
 class Stadium;
 class Player;
 class Team;
--- origsrc/rcssserver-14.0.3/src/stdoutsaver.h	2009-12-21 14:47:12.000000000 +0900
+++ src/rcssserver-14.0.3/src/stdoutsaver.h	2010-12-03 10:29:42.840880000 +0900
@@ -24,6 +24,8 @@
 
 #include "resultsaver.hpp"
 
+#include <ctime>
+
 class STDOutSaver
     : public ResultSaver {
 public:
