--- origsrc/rcssserver-rcssserver-17.0.1/rcss/clang/CMakeLists.txt	2022-04-03 20:01:13.000000000 +0900
+++ src/rcssserver-rcssserver-17.0.1/rcss/clang/CMakeLists.txt	2022-06-12 15:48:25.257743200 +0900
@@ -50,7 +50,7 @@ add_library(RCSS::CLangParser ALIAS RCSS
 
 target_link_libraries(RCSSCLangParser
   PUBLIC
-    Boost::boost
+    Boost::system
   )
 
 target_compile_definitions(RCSSCLangParser
@@ -100,7 +100,7 @@ set_property(TARGET RCSSCLangParser PROP
 
 install(TARGETS RCSSCLangParser
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     COMPONENT Libraries
     PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
   PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/rcss/clang
--- origsrc/rcssserver-rcssserver-17.0.1/rcss/conf/CMakeLists.txt	2022-04-03 20:01:13.000000000 +0900
+++ src/rcssserver-rcssserver-17.0.1/rcss/conf/CMakeLists.txt	2022-06-12 15:32:48.460317900 +0900
@@ -10,7 +10,7 @@ add_library(RCSS::ConfParser ALIAS RCSSC
 
 target_link_libraries(RCSSConfParser
   PUBLIC
-    Boost::boost
+    Boost::filesystem
 )
 
 target_compile_definitions(RCSSConfParser
--- origsrc/rcssserver-rcssserver-17.0.1/src/CMakeLists.txt	2022-04-03 20:01:13.000000000 +0900
+++ src/rcssserver-rcssserver-17.0.1/src/CMakeLists.txt	2022-06-12 16:03:55.696981200 +0900
@@ -134,7 +134,7 @@ set(exec_prefix ${CMAKE_INSTALL_PREFIX})
 set(libdir ${CMAKE_INSTALL_FULL_LIBDIR})
 configure_file(rcsoccersim.in rcsoccersim @ONLY)
 
-install(TARGETS RCSSServer
+install(TARGETS RCSSServer RCSSClient
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     COMPONENT Libraries
