--- origsrc/rcssserver-15.0.1/rcssbase/conf/Makefile.am	2011-06-04 16:42:23.000000000 +0900
+++ src/rcssserver-15.0.1/rcssbase/conf/Makefile.am	2011-11-20 19:37:02.828124000 +0900
@@ -7,7 +7,7 @@ librcssconfparser_la_SOURCES = \
 	statushandler.cpp \
 	streamstatushandler.cpp
 
-librcssconfparser_la_LDFLAGS = -version-info 3:0:0
+librcssconfparser_la_LDFLAGS = -version-info 3:0:0 -no-undefined
 #    1. Start with version information of `0:0:0' for each libtool library.
 #
 #    2. Update the version information only immediately before a public
@@ -27,7 +27,7 @@ librcssconfparser_la_LDFLAGS = -version-
 #    6. If any interfaces have been removed since the last public release,
 #       then set AGE to 0.
 
-#librcssconfparser_la_LIBADD = $(BOOST_FILESYSTEM_LIB)
+librcssconfparser_la_LIBADD = $(BOOST_SYSTEM_LIB) $(BOOST_FILESYSTEM_LIB)
 
 rcssbase_confdir = $(includedir)/rcssbase/conf
 
--- origsrc/rcssserver-15.0.1/rcssbase/gzip/Makefile.am	2011-06-04 16:42:23.000000000 +0900
+++ src/rcssserver-15.0.1/rcssbase/gzip/Makefile.am	2011-11-20 19:37:26.761456800 +0900
@@ -7,7 +7,7 @@ librcssgz_la_SOURCES= \
 	gzstream.hpp \
 	gzstream.cpp
 
-librcssgz_la_LDFLAGS = -version-info 1:0:0
+librcssgz_la_LDFLAGS = -version-info 1:0:0 -no-undefined
 #    1. Start with version information of `0:0:0' for each libtool library.
 #
 #    2. Update the version information only immediately before a public
--- origsrc/rcssserver-15.0.1/rcssbase/net/Makefile.am	2011-06-04 16:42:23.000000000 +0900
+++ src/rcssserver-15.0.1/rcssbase/net/Makefile.am	2011-11-20 19:37:55.435341500 +0900
@@ -9,7 +9,7 @@ librcssnet_la_SOURCES = \
 	tcpsocket.cpp
 
 #librcssnet_la_LDFLAGS = -version-info 1:1:0 -L../error
-librcssnet_la_LDFLAGS = -version-info 1:1:0
+librcssnet_la_LDFLAGS = -version-info 1:1:0 -no-undefined
 #    1. Start with version information of `0:0:0' for each libtool library.
 #
 #    2. Update the version information only immediately before a public
@@ -30,7 +30,7 @@ librcssnet_la_LDFLAGS = -version-info 1:
 #       then set AGE to 0.
 
 #librcssnet_la_LIBADD = @NET_LIBS@ -lrcsserror
-librcssnet_la_LIBADD =
+librcssnet_la_LIBADD = $(BOOST_SYSTEM_LIB)
 
 rcssbase_netdir = $(includedir)/rcssbase/net
 
--- origsrc/rcssserver-15.0.1/src/Makefile.am	2011-06-04 16:42:23.000000000 +0900
+++ src/rcssserver-15.0.1/src/Makefile.am	2011-11-20 19:38:17.966443400 +0900
@@ -25,7 +25,7 @@ librcssclangparser_la_SOURCES = \
 	rule.cpp \
 	rcssexceptions.cpp
 
-librcssclangparser_la_LDFLAGS = -version-info 3:0:1
+librcssclangparser_la_LDFLAGS = -version-info 3:0:1 -no-undefined
 #    Changed from 2.0.0 to 3.0.1 at 9.3.5 for addition of buildCondList
 #
 #    1. Start with version information of `0:0:0' for each libtool library.
@@ -47,6 +47,8 @@ librcssclangparser_la_LDFLAGS = -version
 #    6. If any interfaces have been removed since the last public release,
 #       then set AGE to 0.
 
+librcssclangparser_la_LIBADD = $(BOOST_SYSTEM_LIB)
+
 pkginclude_HEADERS = \
 	clangparser.h \
 	coach_lang_tok.h \
