--- origsrc/semver.c-0.2.0/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/semver.c-0.2.0/Makefile.am	2017-06-24 12:46:33.521000000 +0900
@@ -0,0 +1,16 @@
+
+check_PROGRAMS = semver_test semver_unit
+semver_test_SOURCES = semver_test.c
+semver_test_LDADD = libsemver.la
+semver_unit_SOURCES = semver_unit.c
+semver_unit_LDADD = libsemver.la
+TESTS = $(check_PROGRAMS)
+
+include_HEADERS = semver.h
+
+lib_LTLIBRARIES = libsemver.la
+libsemver_la_SOURCES = semver.c
+libsemver_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/semver.c-0.2.0/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/semver.c-0.2.0/configure.ac	2017-06-24 12:43:56.627000000 +0900
@@ -0,0 +1,29 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([semver], [0.2.0])
+AC_CONFIG_SRCDIR([semver.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([memmove memset strchr strtol])
+
+AC_CONFIG_FILES([semver.pc Makefile])
+AC_OUTPUT
--- origsrc/semver.c-0.2.0/semver.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/semver.c-0.2.0/semver.pc.in	2017-06-24 12:43:53.893679300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: semver
+Description: Semantic version parser and serializer written in ANSI C
+Version: @VERSION@
+Libs: -L${libdir} -lsemver
+Cflags: -I${includedir}
