--- origsrc/libcloudproviders-0.3.6/meson.build	2024-03-20 18:49:45.000000000 +0900
+++ src/libcloudproviders-0.3.6/meson.build	2025-12-14 22:31:26.311778200 +0900
@@ -42,7 +42,7 @@ glib = dependency ('glib-2.0', version:
 gio = dependency ('gio-2.0', version: glib_ver)
 
 host_os = host_machine.system().to_lower()
-if host_os.contains('mingw') or host_os.contains('cygwin') or host_os.contains('windows')
+if host_os.contains('mingw') or host_os.contains('windows')
     gio_platform = dependency ('gio-windows-2.0', version: glib_ver)
 else
     gio_platform = dependency ('gio-unix-2.0', version: glib_ver)
