--- origsrc/libcloudproviders-0.3.2/meson.build	2023-08-09 23:36:01.000000000 +0900
+++ src/libcloudproviders-0.3.2/meson.build	2023-08-23 09:55:43.759392200 +0900
@@ -38,7 +38,7 @@ glib = dependency ('glib-2.0', version:
 gio = dependency ('gio-2.0', version: glib_ver)
 
 host_os = host_machine.system().to_lower()
-if host_os.contains('mingw') or host_os.contains('cygwin') or host_os.contains('windows')
+if host_os.contains('mingw') or host_os.contains('windows')
     gio_platform = dependency ('gio-windows-2.0', version: glib_ver)
 else
     gio_platform = dependency ('gio-unix-2.0', version: glib_ver)
