--- origsrc/libquantum-1.1.0/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libquantum-1.1.0/Makefile.am	2011-06-18 14:43:55.441747400 +0900
@@ -0,0 +1,25 @@
+
+AM_CFLAGS = -Wall
+
+lib_LTLIBRARIES = libquantum.la
+libquantum_la_SOURCES = \
+	complex.c measure.c matrix.c gates.c oaddn.c omuln.c expn.c qft.c \
+	classic.c qureg.c decoherence.c qec.c version.c objcode.c density.c \
+	error.c qtime.c lapack.c
+libquantum_la_LDFLAGS = -version-info 7:0:0 -no-undefined
+
+include_HEADERS = quantum.h
+
+noinst_PROGRAMS = shor grover
+shor_SOURCES = shor.c
+shor_LDADD = libquantum.la
+grover_SOURCES = grover.c
+grover_LDADD = libquantum.la
+
+bin_PROGRAMS = quobprint quobdump
+quobprint_SOURCES = quobprint.c
+quobprint_LDADD = libquantum.la
+
+quobdump_SOURCES = quobdump.c
+quobdump_LDADD = libquantum.la
+
--- origsrc/libquantum-1.1.0/configure.in	2008-09-07 20:00:41.000000000 +0900
+++ src/libquantum-1.1.0/configure.in	2011-06-18 14:33:27.588939400 +0900
@@ -23,6 +23,7 @@
 AC_INIT([libquantum], [1.1.0], [libquantum@libquantum.de])
 AC_CONFIG_SRCDIR([classic.c])
 AC_CONFIG_HEADER([config.h])
+AM_INIT_AUTOMAKE
 
 # The language of our choice
 AC_LANG([C])
@@ -30,7 +31,7 @@ AC_LANG([C])
 # Checks for programs.
 AC_PROG_CC
 AC_PROG_INSTALL
-AC_PROG_LIBTOOL
+LT_INIT
 
 # Checks for libraries.
 AC_CHECK_LIB([m], [sqrt])
@@ -122,14 +123,6 @@ AC_ARG_ENABLE(profiling,
 
 # Disable LAPACK check
 
-
-# Enable -Wall for gcc
-if test $CC = "gcc"
-then
-	AC_DEFINE(HAVE_GCC)
-	CFLAGS="$CFLAGS -Wall"
-fi
-
 # Write the output files
 AC_CONFIG_FILES([Makefile quantum.h types.h])
 AC_OUTPUT
