--- origsrc/gupnp-tools-gupnp-tools-0.10.3/subprojects/gssdp-1.2.wrap	2022-05-20 17:13:50.000000000 +0900
+++ src/gupnp-tools-gupnp-tools-0.10.3/subprojects/gssdp-1.2.wrap	2025-12-14 10:26:27.658900000 +0900
@@ -3,6 +3,6 @@ url = https://gitlab.gnome.org/GNOME/gss
 revision = gssdp-1.4
 depth = 1
  
-[provides]
-dependency_name = gssdp-1.2
+[provide]
+dependency_names = gssdp-1.2
 
--- origsrc/gupnp-tools-gupnp-tools-0.10.3/subprojects/gupnp-1.2.wrap	2022-05-20 17:13:50.000000000 +0900
+++ src/gupnp-tools-gupnp-tools-0.10.3/subprojects/gupnp-1.2.wrap	2025-12-14 10:26:27.658900000 +0900
@@ -3,6 +3,6 @@ url = https://gitlab.gnome.org/GNOME/gup
 revision = gupnp-1.4
 depth = 1
  
-[provides]
-dependency_name = gupnp-1.2
+[provide]
+dependency_names = gupnp-1.2
 
--- origsrc/gupnp-tools-gupnp-tools-0.10.3/subprojects/gupnp-av-1.0.wrap	2022-05-20 17:13:50.000000000 +0900
+++ src/gupnp-tools-gupnp-tools-0.10.3/subprojects/gupnp-av-1.0.wrap	2025-12-14 10:26:35.566558400 +0900
@@ -3,6 +3,6 @@ url = https://gitlab.gnome.org/GNOME/gup
 revision = master
 depth = 1
  
-[provides]
-dependency_name = gupnp-av-1.0
+[provide]
+dependency_names = gupnp-av-1.0
 
