--- origsrc/o2-1.0.1/CMakeLists.txt	2016-11-01 22:38:08.000000000 +0900
+++ src/o2-1.0.1/CMakeLists.txt	2017-12-03 00:39:01.062000000 +0900
@@ -41,7 +41,7 @@ endif()
 
 option(o2_BUILD_EXAMPLES "Build examples" OFF)
 
-if(WIN32)
+if(WIN32 OR CYGWIN)
   add_definitions(-DO2_DLL_EXPORT)
 endif()
 
@@ -50,3 +50,10 @@ add_subdirectory(src)
 if(o2_BUILD_EXAMPLES)
   add_subdirectory(examples)
 endif(o2_BUILD_EXAMPLES)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/o2.pc.in ${CMAKE_BINARY_DIR}/o2.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/o2.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/o2-1.0.1/examples/CMakeLists.txt	2016-11-01 22:38:08.000000000 +0900
+++ src/o2-1.0.1/examples/CMakeLists.txt	2017-12-03 00:32:56.219000200 +0900
@@ -8,6 +8,6 @@ if(o2_WITH_TWITTER)
     add_subdirectory(twitterdemo)
 endif(o2_WITH_TWITTER)
 
-if(o2_WITH_SPOTIFY)
-    add_subdirectory(spotifydemo)
-endif(o2_WITH_SPOTIFY)
+# if(o2_WITH_SPOTIFY)
+#    add_subdirectory(spotifydemo)
+# endif(o2_WITH_SPOTIFY)
--- origsrc/o2-1.0.1/o2.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/o2-1.0.1/o2.pc.in	2017-12-03 00:39:09.465000000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: o2
+Description: OAuth 2.0 for Qt
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lo2
+Cflags: -I${includedir}
--- origsrc/o2-1.0.1/src/o2hubic.cpp	2016-11-01 22:38:08.000000000 +0900
+++ src/o2-1.0.1/src/o2hubic.cpp	2017-12-03 00:36:21.323999800 +0900
@@ -1,5 +1,5 @@
 #include "o2hubic.h"
-#include "o2globals.h"
+#include "o0globals.h"
 #include "o2replyserver.h"
 #include <QHostAddress>
 
--- origsrc/o2-1.0.1/src/o2surveymonkey.cpp	2016-11-01 22:38:08.000000000 +0900
+++ src/o2-1.0.1/src/o2surveymonkey.cpp	2017-12-03 00:36:18.974000000 +0900
@@ -7,7 +7,7 @@
 #endif
 
 #include "o2surveymonkey.h"
-#include "o2globals.h"
+#include "o0globals.h"
 
 static const char *SMEndpoint = "https://api.surveymonkey.net/oauth/authorize";
 static const char *SMTokenUrl = "https://api.surveymonkey.net/oauth/token";
