--- origsrc/ccrtp-2.0.9/configure.ac	2014-04-14 22:40:20.000000000 +0900
+++ src/ccrtp-2.0.9/configure.ac	2014-07-25 09:28:30.551339600 +0900
@@ -77,7 +77,7 @@ AC_SUBST(GNULIBS)
 # OST_CC_ENDIAN - now in Common C++
 # Are we using the GNU compiler?
 if test $GCC = yes ; then
-    WARN_FLAGS="-fno-strict-aliasing -Wall -ansi -pedantic"
+    WARN_FLAGS="-fno-strict-aliasing -Wall -pedantic"
 else
     WARN_FLAGS=""
 fi
