--- origsrc/ccrtp-2.1.2/src/Makefile.am	2015-01-11 23:43:52.000000000 +0900
+++ src/ccrtp-2.1.2/src/Makefile.am	2025-12-08 16:04:56.290413700 +0900
@@ -44,7 +44,7 @@ libccrtp_la_SOURCES = rtppkt.cpp rtcppkt
     outqueue.cpp queue.cpp control.cpp members.cpp socket.cpp duplex.cpp pool.cpp \
     CryptoContext.cpp CryptoContextCtrl.cpp $(srtp_src_g) $(srtp_src_o) $(skein_srcs)
 
-libccrtp_la_LDFLAGS = $(RELEASE)
+libccrtp_la_LDFLAGS = -no-undefined $(RELEASE)
 libccrtp_la_LIBADD = @GNULIBS@
 
 noinst_HEADERS = private.h
