--- origsrc/ccrtp-2.1.2/configure.ac	2015-03-29 23:23:15.000000000 +0900
+++ src/ccrtp-2.1.2/configure.ac	2017-10-11 15:36:24.708000000 +0900
@@ -86,7 +86,10 @@ AC_SUBST(GNULIBS)
 # OST_CC_ENDIAN - now in Common C++
 # Are we using the GNU compiler?
 if test $GCC = yes ; then
-    WARN_FLAGS="-fno-strict-aliasing -Wall -ansi -pedantic"
+    WARN_FLAGS="-fno-strict-aliasing -Wall -pedantic"
+    if test "$host_os" != "cygwin"; then
+        WARN_FLAGS="${WARN_FLAGS} -ansi"
+    fi
 else
     WARN_FLAGS=""
 fi
