--- origsrc/argtable2-12/Makefile.am	2010-02-05 08:32:16.000000000 +0900
+++ src/argtable2-12/Makefile.am	2010-11-13 17:29:34.660677800 +0900
@@ -1,5 +1,5 @@
 SUBDIRS = src tests doc
-ACLOCAL_AMFLAGS=-I m4
+#ACLOCAL_AMFLAGS=-I m4
 
 pcfiledir=$(libdir)/pkgconfig
 pcfile_DATA=argtable2.pc
--- origsrc/argtable2-12/configure.ac	2010-02-06 17:26:46.000000000 +0900
+++ src/argtable2-12/configure.ac	2010-11-13 17:30:27.107885800 +0900
@@ -12,7 +12,7 @@ AC_CONFIG_MACRO_DIR([m4])
     
 # Checks for programs.
 AC_PROG_CC
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_PROG_MAKE_SET
 
 # Checks for libraries.
--- origsrc/argtable2-12/src/Makefile.am	2010-02-06 16:44:52.000000000 +0900
+++ src/argtable2-12/src/Makefile.am	2010-11-13 17:29:34.670692200 +0900
@@ -17,7 +17,7 @@ libargtable2_la_SOURCES = argtable2.c ar
 # argtable-2.10 is 1:5:1 (minor tweaks to configuration and documentation, src code is identical to argtable-2.8)
 # argtable-2.11 is 1:6:1 (minor tweaks to makefiles, src code is identical to argtable-2.8)
 # argtable-2.12 is 1:7:1 (added support for parsing hex,octal,binary integer formats)
-libargtable2_la_LDFLAGS = -version-info 1:7:1
+libargtable2_la_LDFLAGS = -version-info 1:7:1 -no-undefined
 
 include_HEADERS = argtable2.h
 
