--- origsrc/libfacerec-0.04/CMakeLists.txt	2012-06-17 22:11:04.000000000 +0900
+++ src/libfacerec-0.04/CMakeLists.txt	2013-12-10 23:00:39.500209500 +0900
@@ -49,4 +49,12 @@ endif()
 # compile a sample executable
 #---------------------------------
 add_executable(fisherfaces src/main.cpp)
-target_link_libraries(fisherfaces ${the_library} opencv_core opencv_imgproc opencv_highgui)
\ No newline at end of file
+target_link_libraries(fisherfaces ${the_library} opencv_core opencv_imgproc opencv_highgui)
+
+#---------------------------------
+# install header & library
+#---------------------------------
+install(TARGETS fisherfaces DESTINATION bin)
+install(FILES ${lib_hdrs} DESTINATION include/libfacerec)
+set_target_properties(${the_library} PROPERTIES SOVERSION 0.04)
+install(TARGETS ${the_library} RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
