--- origsrc/libsmb2-4.0.0/configure.ac	2022-01-17 08:32:30.000000000 +0900
+++ src/libsmb2-4.0.0/configure.ac	2022-05-26 12:28:49.932898500 +0900
@@ -31,7 +31,7 @@ AC_ARG_WITH([libkrb5],
                              NTLMSSP module for authentication instead.])])
 
 AS_IF([test "x$with_libkrb5" != "xno"], [
-    MAYBE_LIBKRB5="-lgssapi_krb5"
+    MAYBE_LIBKRB5="-lgssapi_krb5 -lkrb5"
     AC_DEFINE([HAVE_LIBKRB5], [1], [Whether we use gssapi_krb5 or not])
     AC_MSG_NOTICE([Build with gssapi_krb5 support])
     dnl  Check for gssapi/gssapi.h
@@ -65,7 +65,7 @@ case $host in
     AC_CHECK_LIB([socket], [main], , [AC_MSG_ERROR([Can not find required library])])
     AC_CHECK_LIB([nsl],    [main], , [AC_MSG_ERROR([Can not find required library])])
     ;;
-  *mingw32* | *cygwin* | *wince* | *mingwce*)
+  *mingw32* | *wince* | *mingwce*)
     LIBSOCKET='-lws2_32'
     SYS=mingw32
     ;;
--- origsrc/libsmb2-4.0.0/examples/Makefile.am	2022-01-17 08:32:30.000000000 +0900
+++ src/libsmb2-4.0.0/examples/Makefile.am	2022-05-26 12:29:10.554146200 +0900
@@ -1,4 +1,4 @@
-noinst_PROGRAMS = smb2-cat-async smb2-cat-sync \
+bin_PROGRAMS = smb2-cat-async smb2-cat-sync \
 	smb2-dcerpc-coder-test \
 	smb2-ftruncate-sync \
 	smb2-ls-async smb2-ls-sync \
--- origsrc/libsmb2-4.0.0/include/libsmb2-private.h	2022-01-17 08:32:30.000000000 +0900
+++ src/libsmb2-4.0.0/include/libsmb2-private.h	2022-05-26 12:33:51.344599200 +0900
@@ -23,7 +23,7 @@
 #include "config.h"
 #endif
 
-#if defined(PS2_EE_PLATFORM) || defined(PS3_PPU_PLATFORM) || defined(ESP_PLATFORM) || defined(__APPLE__)
+#if defined(PS2_EE_PLATFORM) || defined(PS3_PPU_PLATFORM) || defined(ESP_PLATFORM) || defined(__APPLE__) || defined(__CYGWIN__)
 /* We need this for time_t */
 #include <time.h>
 #endif
--- origsrc/libsmb2-4.0.0/lib/sha224-256.c	2022-01-17 08:32:30.000000000 +0900
+++ src/libsmb2-4.0.0/lib/sha224-256.c	2022-05-26 12:36:30.591435100 +0900
@@ -338,7 +338,7 @@ SHA256FinalBits (SHA256Context * context
  *   sha Error Code.
  */
 int
-SHA256Result (SHA256Context * context, uint8_t Message_Digest[])
+SHA256Result (SHA256Context * context, uint8_t Message_Digest[SHA256HashSize])
 {
   return SHA224_256ResultN (context, Message_Digest, SHA256HashSize);
 }
