--- origsrc/libsmb2-libsmb2-6.2/configure.ac	2024-12-23 12:38:55.000000000 +0900
+++ src/libsmb2-libsmb2-6.2/configure.ac	2025-03-10 17:33:51.217661000 +0900
@@ -1,4 +1,4 @@
-AC_INIT([libsmb2],[6.1.0],[ronniesahlberg@gmail.com])
+AC_INIT([libsmb2],[6.2.0],[ronniesahlberg@gmail.com])
 
 AC_PREREQ([2.71])
 AC_CONFIG_HEADERS([config.h])
@@ -31,7 +31,7 @@ AC_ARG_WITH([libkrb5],
                              NTLMSSP module for authentication instead.])])
 
 AS_IF([test "x$with_libkrb5" != "xno"], [
-    MAYBE_LIBKRB5="-lgssapi_krb5"
+    MAYBE_LIBKRB5="-lgssapi_krb5 -lkrb5"
     AC_DEFINE([HAVE_LIBKRB5], [1], [Whether we use gssapi_krb5 or not])
     AC_MSG_NOTICE([Build with gssapi_krb5 support])
     dnl  Check for gssapi/gssapi.h
@@ -77,7 +77,7 @@ case $host in
     AC_CHECK_LIB([socket], [main], , [AC_MSG_ERROR([Can not find required library])])
     AC_CHECK_LIB([nsl],    [main], , [AC_MSG_ERROR([Can not find required library])])
     ;;
-  *mingw32* | *cygwin* | *wince* | *mingwce*)
+  *mingw32* | *wince* | *mingwce*)
     LIBSOCKET='-lws2_32'
     SYS=mingw32
     ;;
--- origsrc/libsmb2-libsmb2-6.2/examples/Makefile.am	2024-12-23 12:38:55.000000000 +0900
+++ src/libsmb2-libsmb2-6.2/examples/Makefile.am	2025-03-10 17:34:14.319831500 +0900
@@ -1,4 +1,4 @@
-noinst_PROGRAMS = smb2-cat-async smb2-cat-sync \
+bin_PROGRAMS = smb2-cat-async smb2-cat-sync \
 	smb2-ftruncate-sync \
 	smb2-ls-async \
 	smb2-put-async \
