--- origsrc/libsmb2-3.0.0/configure.ac	2019-06-10 06:38:20.000000000 +0900
+++ src/libsmb2-3.0.0/configure.ac	2020-01-02 17:29:03.446961100 +0900
@@ -74,7 +74,7 @@ AC_ARG_WITH([libkrb5],
     AS_HELP_STRING([--without-libkrb5], [Do not link with libkrb5 and use builtin NTLMSSP module for authentication instead.]))
 
 AS_IF([test "x$with_libkrb5" != "xno"],
-      [MAYBE_LIBKRB5="-lgssapi_krb5"
+      [MAYBE_LIBKRB5="-lgssapi_krb5 -lkrb5"
        AC_DEFINE(HAVE_LIBKRB5,1,[Whether we use gssapi_krb5 or not])
        AC_MSG_NOTICE(Build with gssapi_krb5 support)
        # check for gssapi/gssapi.h
@@ -109,7 +109,7 @@ case $host in
     AC_CHECK_LIB([socket], [main], , [AC_MSG_ERROR([Can not find required library])])
     AC_CHECK_LIB([nsl],    [main], , [AC_MSG_ERROR([Can not find required library])])
     ;;
-  *mingw32* | *cygwin* | *wince* | *mingwce*)
+  *mingw32* | *wince* | *mingwce*)
     LIBSOCKET='-lws2_32'
     SYS=mingw32
     ;;
--- origsrc/libsmb2-3.0.0/examples/Makefile.am	2019-06-10 06:38:20.000000000 +0900
+++ src/libsmb2-3.0.0/examples/Makefile.am	2020-01-02 17:36:47.087852200 +0900
@@ -1,4 +1,4 @@
-noinst_PROGRAMS = smb2-cat-async smb2-cat-sync \
+bin_PROGRAMS = smb2-cat-async smb2-cat-sync \
 	smb2-ftruncate-sync \
 	smb2-ls-async smb2-ls-sync \
 	smb2-put-async \
