--- origsrc/tinyproxy-1.11.0/configure.ac	2021-04-16 21:47:21.000000000 +0900
+++ src/tinyproxy-1.11.0/configure.ac	2021-06-19 10:55:00.923233400 +0900
@@ -170,12 +170,14 @@ if test x"$debug_enabled" != x"yes" ; th
     CFLAGS="-DNDEBUG $CFLAGS"
 fi
 
-AS_ECHO_N(["checking to see if linker understands -z,defs... "])
-LDFLAGS_OLD="-Wl $LDFLAGS"
-LDFLAGS="-Wl,-z,defs $LDFLAGS"
-AC_LINK_IFELSE([AC_LANG_PROGRAM()],
-    AS_ECHO("yes"),
-    AS_ECHO("no"); LDFLAGS="$LDFLAGS_OLD")
+if test "$host_os" != "cygwin"; then
+    AS_ECHO_N(["checking to see if linker understands -z,defs... "])
+    LDFLAGS_OLD="-Wl $LDFLAGS"
+    LDFLAGS="-Wl,-z,defs $LDFLAGS"
+    AC_LINK_IFELSE([AC_LANG_PROGRAM()],
+        AS_ECHO("yes"),
+        AS_ECHO("no"); LDFLAGS="$LDFLAGS_OLD")
+fi
 
 dnl
 dnl Make sure we can actually handle the "--with-*" and "--enable-*" stuff.
--- origsrc/tinyproxy-1.11.0/src/conf.c	2021-04-16 21:47:21.000000000 +0900
+++ src/tinyproxy-1.11.0/src/conf.c	2021-06-19 10:58:42.443841100 +0900
@@ -25,6 +25,7 @@
 
 #include "common.h"
 #include <regex.h>
+#include <limits.h>
 #include "conf.h"
 
 #include "acl.h"
