--- origsrc/tinyproxy-1.8.4/configure.ac	2014-12-13 09:57:23.000000000 +0900
+++ src/tinyproxy-1.8.4/configure.ac	2017-09-06 01:21:57.460000000 +0900
@@ -224,7 +224,9 @@ if test x"$debug_enabled" != x"yes" ; th
     CFLAGS="-DNDEBUG $CFLAGS"
 fi
 
-LDFLAGS="-Wl,-z,defs $LDFLAGS"
+if test "$host_os" != "cygwin"; then
+    LDFLAGS="-Wl,-z,defs $LDFLAGS"
+fi
 
 dnl
 dnl Make sure we can actually handle the "--with-*" and "--enable-*" stuff.
