--- origsrc/tinyproxy-1.10.0/configure.ac	2018-09-01 11:06:59.000000000 +0900
+++ src/tinyproxy-1.10.0/configure.ac	2019-01-23 08:50:17.455524600 +0900
@@ -170,12 +170,14 @@ if test x"$debug_enabled" != x"yes" ; th
     CFLAGS="-DNDEBUG $CFLAGS"
 fi
 
-AS_ECHO_N(["checking to see if linker understands -z,defs... "])
-LDFLAGS_OLD="-Wl $LDFLAGS"
-LDFLAGS="-Wl,-z,defs $LDFLAGS"
-AC_LINK_IFELSE([AC_LANG_PROGRAM()],
-    AS_ECHO("yes"),
-    AS_ECHO("no"); LDFLAGS="$LDFLAGS_OLD")
+if test "$host_os" != "cygwin"; then
+    AS_ECHO_N(["checking to see if linker understands -z,defs... "])
+    LDFLAGS_OLD="-Wl $LDFLAGS"
+    LDFLAGS="-Wl,-z,defs $LDFLAGS"
+    AC_LINK_IFELSE([AC_LANG_PROGRAM()],
+        AS_ECHO("yes"),
+        AS_ECHO("no"); LDFLAGS="$LDFLAGS_OLD")
+fi
 
 dnl
 dnl Make sure we can actually handle the "--with-*" and "--enable-*" stuff.
