--- origsrc/rgxg-0.1.1/configure.ac	2016-08-03 05:56:44.000000000 +0900
+++ src/rgxg-0.1.1/configure.ac	2017-12-11 13:50:55.835000000 +0900
@@ -1,9 +1,8 @@
 AC_PREREQ([2.67])
 
-m4_include([m4/version.m4])
 m4_include([m4/lib_version.m4])
 
-AC_INIT([rgxg], [_PACKAGE_VERSION], [https://github.com/rgxg/rgxg/issues], [rgxg], [rgxg.github.io])
+AC_INIT([rgxg], [0.1.1], [https://github.com/rgxg/rgxg/issues], [rgxg], [rgxg.github.io])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_AUX_DIR(build-aux)
 AC_CONFIG_SRCDIR([src/main.c])
