--- origsrc/libxspf-1.2.1/Makefile.am	2021-05-29 19:49:56.000000000 +0900
+++ src/libxspf-1.2.1/Makefile.am	2021-09-08 08:22:13.666031000 +0900
@@ -24,7 +24,7 @@ pkgconfig_DATA = xspf.pc
 
 libxspf_la_CPPFLAGS = $(AM_CPPFLAGS) $(URIPARSER_CFLAGS) $(EXPAT_CFLAGS)
 libxspf_la_LIBADD = $(URIPARSER_LIBS) $(EXPAT_LIBS)
-libxspf_la_LDFLAGS = -version-info 4:3:0
+libxspf_la_LDFLAGS = -no-undefined -version-info 4:3:0
 
 libxspf_la_SOURCES = \
 	src/XspfChunkCallback.cpp \
