--- origsrc/libxspf-1.2.0/Makefile.am	2009-03-07 08:45:48.000000000 +0900
+++ src/libxspf-1.2.0/Makefile.am	2011-02-13 22:05:52.383028600 +0900
@@ -41,7 +41,7 @@ endif
 
 libxspf_la_CFLAGS = @URIPARSER_CFLAGS@
 
-libxspf_la_LDFLAGS = @URIPARSER_LIBS@ -version-info 4:1:0
+libxspf_la_LDFLAGS = @URIPARSER_LIBS@ -version-info 4:1:0 -no-undefined
 
 libxspf_la_SOURCES = \
 	src/XspfChunkCallback.cpp \
--- origsrc/libxspf-1.2.0/configure.ac	2009-03-07 09:01:20.000000000 +0900
+++ src/libxspf-1.2.0/configure.ac	2011-02-13 22:04:26.579649400 +0900
@@ -4,7 +4,7 @@ AC_INIT(src/XspfReader.cpp)
 AC_CONFIG_AUX_DIR([build-aux])
 AM_INIT_AUTOMAKE(libxspf, 1.2.0)
 
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_PROG_INSTALL
 AC_PROG_CC
 AC_PROG_CXX
--- origsrc/libxspf-1.2.0/test/OfficialSuite.cpp	2009-02-22 05:13:25.000000000 +0900
+++ src/libxspf-1.2.0/test/OfficialSuite.cpp	2011-02-13 22:28:38.680951700 +0900
@@ -20,7 +20,7 @@
  *
  * Sebastian Pipping, sping@xiph.org
  */
-
+#include <cstdio>
 #include "OfficialSuite.h"
 #include <xspf/XspfReaderCallback.h>
 using namespace Xspf;
--- origsrc/libxspf-1.2.0/test/ProjectOpus/ProjectOpusSuite.cpp	2009-02-22 05:13:25.000000000 +0900
+++ src/libxspf-1.2.0/test/ProjectOpus/ProjectOpusSuite.cpp	2011-02-13 22:28:22.748041300 +0900
@@ -20,7 +20,7 @@
  *
  * Sebastian Pipping, sping@xiph.org
  */
-
+#include <cstdio>
 #include "ProjectOpusSuite.h"
 #include <xspf/XspfReader.h>
 #include <xspf/XspfTrack.h>
