--- origsrc/tblis-1.2.0/Makefile.am	2019-08-17 00:45:31.000000000 +0900
+++ src/tblis-1.2.0/Makefile.am	2021-03-15 23:41:28.392825700 +0900
@@ -205,7 +205,8 @@ stl_extinclude_HEADERS = \
     src/external/stl_ext/include/zip.hpp
     
 noinst_LTLIBRARIES =
-lib_libtblis_la_LIBADD = src/external/tci/lib/libtci.la
+lib_libtblis_la_LIBADD = -ltci -latomic
+lib_libtblis_la_LDFLAGS = -no-undefined
 
 if ENABLE_REFERENCE
 noinst_LTLIBRARIES += lib/libreference.la
@@ -514,9 +515,8 @@ bin_dpd_bench_SOURCES = test/dpd_bench.c
 
 VPATH += $(srcdir)
 
-SUBDIRS = src/external/tci
 ACLOCAL_AMFLAGS = -I m4
-AM_CPPFLAGS = -I$(srcdir) -I$(srcdir)/src -I. -Isrc -Isrc/util -I$(srcdir)/src/external/tci -Isrc/external/tci/tci
+AM_CPPFLAGS = -I$(srcdir) -I$(srcdir)/src -I. -Isrc -Isrc/util
 AM_LDFLAGS = -pthread
 bin_test_LDADD = lib/libtblis.la
 bin_skx_bench_LDADD = lib/libtblis.la $(BLAS_LIBS)
@@ -525,3 +525,6 @@ bin_batched_bench_LDADD = lib/libtblis.l
 bin_dpd_bench_LDADD = lib/libtblis.la $(BLAS_LIBS)
 
 DISTCLEANFILES = src/tblis_config.h
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/tblis-1.2.0/configure.ac	2019-08-17 00:45:31.000000000 +0900
+++ src/tblis-1.2.0/configure.ac	2021-03-15 23:41:28.340834200 +0900
@@ -446,7 +446,7 @@ LT_INIT
 
 mkdir -p bin
 
-AC_CONFIG_FILES([Makefile 
+AC_CONFIG_FILES([tblis.pc Makefile 
                  src/configs/foreach_type.h
                  src/configs/foreach_config.h
                  src/configs/foreach_config_and_type.h
--- origsrc/tblis-1.2.0/tblis.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/tblis-1.2.0/tblis.pc.in	2021-03-16 18:10:37.901049100 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: tblis
+Description: Library and framework for performing tensor operations
+Version: @VERSION@
+Libs: -L${libdir} -ltblis
+Cflags: -I${includedir}
+Requires: hwloc tci
