--- origsrc/libtextwrap-0.1/Makefile.am	2003-08-24 17:13:41.000000000 +0900
+++ src/libtextwrap-0.1/Makefile.am	2023-10-06 18:37:08.685624200 +0900
@@ -1,11 +1,14 @@
 lib_LTLIBRARIES = libtextwrap.la
 
 libtextwrap_la_SOURCES = textwrap.c
-libtextwrap_la_LDFLAGS = -version-info 1:0:0
+libtextwrap_la_LDFLAGS = -no-undefined -version-info 1:0:0
 
 include_HEADERS = textwrap.h
 man_MANS = textwrap.3 dotextwrap.1
 
 bin_PROGRAMS = dotextwrap
 dotextwrap_SOURCES = dotextwrap.c
-dotextwrap_LDADD = -ltextwrap
+dotextwrap_LDADD = libtextwrap.la
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libtextwrap-0.1/configure.in	2003-08-17 00:46:11.000000000 +0900
+++ src/libtextwrap-0.1/configure.in	2023-10-06 18:37:51.375638100 +0900
@@ -2,11 +2,11 @@
 # Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.57)
-AC_INIT(libtextwrap, 0.01, kubota@debian.org)
+AC_INIT(libtextwrap, 0.1, kubota@debian.org)
 AC_CONFIG_AUX_DIR(config)
 #AC_CONFIG_SRCDIR(textwrap.c)
 AM_CONFIG_HEADER(config.h)
-AM_INIT_AUTOMAKE(libtexrwrap, 0.01)
+AM_INIT_AUTOMAKE([foreign])
 
 # Checks for programs.
 AC_PROG_CC
@@ -29,4 +29,5 @@ AC_FUNC_REALLOC
 AC_CHECK_FUNCS(mblen nl_langinfo setlocale strstr wcwidth)
 
 AC_CONFIG_FILES(Makefile)
+AC_CONFIG_FILES([libtextwrap.pc])
 AC_OUTPUT
--- origsrc/libtextwrap-0.1/libtextwrap.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libtextwrap-0.1/libtextwrap.pc.in	2023-10-06 18:37:08.638745300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libtextwrap
+Description: Text Wrapping Library with I18N
+Version: @VERSION@
+Libs: -L${libdir} -ltextwrap
+Cflags: -I${includedir}
