--- origsrc/libfsapfs-20240429/Makefile.am	2024-04-29 13:08:24.000000000 +0900
+++ src/libfsapfs-20240429/Makefile.am	2025-05-18 19:28:38.523452400 +0900
@@ -3,27 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
-	libfguid \
-	libfmos \
-	libhmac \
-	libcaes \
 	libfsapfs \
 	pyfsapfs \
 	fsapfstools \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libfsapfs-20240429/configure.ac	2024-04-29 13:08:24.000000000 +0900
+++ src/libfsapfs-20240429/configure.ac	2025-05-18 19:28:38.539112100 +0900
@@ -186,23 +186,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfmos/Makefile])
-AC_CONFIG_FILES([libhmac/Makefile])
-AC_CONFIG_FILES([libcaes/Makefile])
 AC_CONFIG_FILES([libfsapfs/Makefile])
 AC_CONFIG_FILES([pyfsapfs/Makefile])
 AC_CONFIG_FILES([fsapfstools/Makefile])
--- origsrc/libfsapfs-20240429/pyfsapfs/Makefile.am	2024-04-29 13:08:24.000000000 +0900
+++ src/libfsapfs-20240429/pyfsapfs/Makefile.am	2025-05-18 19:28:53.887469800 +0900
@@ -50,7 +50,7 @@ pyfsapfs_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pyfsapfs_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyfsapfs_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyfsapfs_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
