--- origsrc/lightstep-tracer-cpp-0.10.0/CMakeLists.txt	2019-06-22 05:27:23.000000000 +0900
+++ src/lightstep-tracer-cpp-0.10.0/CMakeLists.txt	2019-06-25 13:41:24.347332000 +0900
@@ -255,7 +255,8 @@ if (BUILD_SHARED_LIBS)
   target_link_libraries(lightstep_tracer ${LIGHTSTEP_LINK_LIBRARIES})
   set_target_properties(lightstep_tracer PROPERTIES SOVERSION ${LIGHTSTEP_VERSION_MAJOR})
   install(TARGETS lightstep_tracer
-          LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+          RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+          ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
 
 if (BUILD_STATIC_LIBS)  
@@ -277,3 +278,10 @@ if (BUILD_TESTING AND BUILD_SHARED_LIBS)
   add_subdirectory(test/cmake)
   add_subdirectory(example)
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/lightstep-tracer-cpp.pc.in ${CMAKE_BINARY_DIR}/lightstep-tracer-cpp.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/lightstep-tracer-cpp.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/lightstep-tracer-cpp-0.10.0/lightstep-tracer-cpp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/lightstep-tracer-cpp-0.10.0/lightstep-tracer-cpp.pc.in	2019-06-25 13:46:24.094027900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: lightstep-tracer-cpp
+Description: LightStep distributed tracing library for C++
+Version: @LIGHTSTEP_VERSION_STRING@
+Libs: -L${libdir} -llightstep_tracer
+Cflags: -I${includedir}
