--- origsrc/snapd-glib-1.64/snapd-qt/Snapd/client.h	2023-07-12 07:38:40.000000000 +0900
+++ src/snapd-glib-1.64/snapd-qt/Snapd/client.h	2023-08-27 00:35:26.001412700 +0900
@@ -28,6 +28,14 @@
 #include <Snapd/SystemInformation>
 #include <Snapd/UserInformation>
 
+#if defined(stdout)
+#undef stdout
+#endif
+
+#if defined(stderr)
+#undef stderr
+#endif
+
 class QSnapdConnectRequestPrivate;
 class Q_DECL_EXPORT QSnapdConnectRequest : public QSnapdRequest
 {
