--- origsrc/snapd-glib-1.45/snapd-glib/Makefile.am	2018-11-21 13:27:56.000000000 +0900
+++ src/snapd-glib-1.45/snapd-glib/Makefile.am	2019-01-19 09:54:13.075393400 +0900
@@ -131,7 +131,7 @@ libsnapd_glib_la_CPPFLAGS = \
 	-I$(top_builddir) \
 	$(SNAPD_DEBUG_CFLAGS)
 libsnapd_glib_la_CFLAGS = $(SNAPD_CFLAGS) $(WARN_CFLAGS)
-libsnapd_glib_la_LDFLAGS = -version-info 1:0:0 -export-symbols-regex \^snapd_.*
+libsnapd_glib_la_LDFLAGS = -no-undefined -version-info 1:0:0 -export-symbols-regex \^snapd_.*
 libsnapd_glib_la_LIBADD = $(SNAPD_LIBS)
 libsnapd_glib_la_SOURCES = $(source_private_c) $(source_c) $(source_private_h) $(source_h)
 nodist_libsnapd_glib_la_SOURCES = $(BUILT_SOURCES)
--- origsrc/snapd-glib-1.45/snapd-qt/Makefile.am	2018-11-21 13:27:56.000000000 +0900
+++ src/snapd-glib-1.45/snapd-qt/Makefile.am	2019-01-19 10:00:03.077540300 +0900
@@ -35,7 +35,7 @@ client-private_moc.cpp: client-private.h
 	$(AM_V_GEN) $(MOC5) $< -o $@
 
 libsnapd_qt_la_CPPFLAGS = -fPIC -DQT_NO_SIGNALS_SLOTS_KEYWORDS --std=c++11 $(SNAPD_QT_CFLAGS) $(SNAPD_CFLAGS) $(WARN_CFLAGS) -I$(top_srcdir)
-libsnapd_qt_la_LDFLAGS = -version-info 1:0:0
+libsnapd_qt_la_LDFLAGS = -no-undefined -version-info 1:0:0
 libsnapd_qt_la_LIBADD = $(top_builddir)/snapd-glib/libsnapd-glib.la $(SNAPD_QT_LIBS)
 libsnapd_qt_la_SOURCES = $(source_cpp) stream-wrapper.cpp $(source_h) $(private_h)
 nodist_libsnapd_qt_la_SOURCES = $(source_moc)
--- origsrc/snapd-glib-1.45/snapd-qt/Snapd/client.h	2018-11-21 13:27:56.000000000 +0900
+++ src/snapd-glib-1.45/snapd-qt/Snapd/client.h	2019-01-19 09:57:02.885617300 +0900
@@ -25,6 +25,14 @@
 #include <Snapd/SystemInformation>
 #include <Snapd/UserInformation>
 
+#if defined(stdout)
+#undef stdout
+#endif
+
+#if defined(stderr)
+#undef stderr
+#endif
+
 class QSnapdConnectRequestPrivate;
 class Q_DECL_EXPORT QSnapdConnectRequest : public QSnapdRequest
 {
