--- origsrc/libstrophe-0.10.0/Makefile.am	2020-09-16 06:27:34.000000000 +0900
+++ src/libstrophe-0.10.0/Makefile.am	2020-09-26 19:35:04.235950000 +0900
@@ -157,32 +157,32 @@ tests_check_parser_SOURCES = tests/check
 tests_check_parser_CFLAGS = $(PARSER_CFLAGS) $(STROPHE_FLAGS) \
 	-I$(top_srcdir)/src
 tests_check_parser_LDADD = $(STROPHE_LIBS)
-tests_check_parser_LDFLAGS = -static
+tests_check_parser_LDFLAGS = 
 
 tests_test_ctx_SOURCES = tests/test_ctx.c
 tests_test_ctx_CFLAGS = $(STROPHE_FLAGS) -I$(top_srcdir)/src
 tests_test_ctx_LDADD = $(STROPHE_LIBS)
-tests_test_ctx_LDFLAGS = -static
+tests_test_ctx_LDFLAGS = 
 
 tests_test_base64_SOURCES = tests/test_base64.c tests/test.c tests/test.h
 tests_test_base64_CFLAGS = $(STROPHE_FLAGS) -I$(top_srcdir)/src
 tests_test_base64_LDADD = $(STROPHE_LIBS)
-tests_test_base64_LDFLAGS = -static
+tests_test_base64_LDFLAGS = 
 
 tests_test_hash_SOURCES = tests/test_hash.c
 tests_test_hash_CFLAGS = $(STROPHE_FLAGS) -I$(top_srcdir)/src
 tests_test_hash_LDADD = $(STROPHE_LIBS)
-tests_test_hash_LDFLAGS = -static
+tests_test_hash_LDFLAGS = 
 
 tests_test_jid_SOURCES = tests/test_jid.c
 tests_test_jid_CFLAGS = $(STROPHE_FLAGS) -I$(top_srcdir)/src
 tests_test_jid_LDADD = $(STROPHE_LIBS)
-tests_test_jid_LDFLAGS = -static
+tests_test_jid_LDFLAGS = 
 
 tests_test_resolver_SOURCES = tests/test_resolver.c tests/test.h
 tests_test_resolver_CFLAGS = $(STROPHE_FLAGS) -I$(top_srcdir)/src
 tests_test_resolver_LDADD = $(STROPHE_LIBS)
-tests_test_resolver_LDFLAGS = -static
+tests_test_resolver_LDFLAGS = 
 
 tests_test_rand_SOURCES = tests/test_rand.c tests/test.c src/sha1.c
 tests_test_rand_CFLAGS = $(STROPHE_FLAGS) -I$(top_srcdir)/src
@@ -209,12 +209,12 @@ tests_test_snprintf_CFLAGS = -I$(top_src
 tests_test_string_SOURCES = tests/test_string.c tests/test.h
 tests_test_string_CFLAGS = $(STROPHE_FLAGS) -I$(top_srcdir)/src
 tests_test_string_LDADD = $(STROPHE_LIBS)
-tests_test_string_LDFLAGS = -static
+tests_test_string_LDFLAGS = 
 
 tests_test_stanza_SOURCES = tests/test_stanza.c
 tests_test_stanza_CFLAGS = $(STROPHE_FLAGS)
 tests_test_stanza_LDADD = $(STROPHE_LIBS)
-tests_test_stanza_LDFLAGS = -static
+tests_test_stanza_LDFLAGS = 
 
 format:
 	@echo "   * run clang-format on all sources"
