--- origsrc/libstrophe-0.9.1/Makefile.am	2016-09-16 07:10:25.000000000 +0900
+++ src/libstrophe-0.9.1/Makefile.am	2016-12-02 11:43:45.267871300 +0900
@@ -75,36 +75,36 @@ examples_uuid_LDADD = $(STROPHE_LIBS)
 
 
 ## Tests
-TESTS = tests/check_parser tests/test_sha1 tests/test_md5 tests/test_rand \
+TESTS = tests/test_sha1 tests/test_md5 tests/test_rand \
 	tests/test_scram tests/test_ctx tests/test_base64 tests/test_jid \
-	tests/test_snprintf tests/test_string tests/test_resolver
+	tests/test_snprintf tests/test_string
 check_PROGRAMS = $(TESTS)
 
 tests_check_parser_SOURCES = tests/check_parser.c tests/test.h
 tests_check_parser_CFLAGS = $(PARSER_CFLAGS) $(STROPHE_FLAGS) \
 	-I$(top_srcdir)/src
 tests_check_parser_LDADD = $(STROPHE_LIBS)
-tests_check_parser_LDFLAGS = -static
+tests_check_parser_LDFLAGS =
 
 tests_test_ctx_SOURCES = tests/test_ctx.c
 tests_test_ctx_CFLAGS = $(STROPHE_FLAGS) -I$(top_srcdir)/src
 tests_test_ctx_LDADD = $(STROPHE_LIBS)
-tests_test_ctx_LDFLAGS = -static
+tests_test_ctx_LDFLAGS =
 
 tests_test_base64_SOURCES = tests/test_base64.c tests/test.c tests/test.h
 tests_test_base64_CFLAGS = $(STROPHE_FLAGS) -I$(top_srcdir)/src
 tests_test_base64_LDADD = $(STROPHE_LIBS)
-tests_test_base64_LDFLAGS = -static
+tests_test_base64_LDFLAGS =
 
 tests_test_jid_SOURCES = tests/test_jid.c
 tests_test_jid_CFLAGS = $(STROPHE_FLAGS) -I$(top_srcdir)/src
 tests_test_jid_LDADD = $(STROPHE_LIBS)
-tests_test_jid_LDFLAGS = -static
+tests_test_jid_LDFLAGS =
 
 tests_test_resolver_SOURCES = tests/test_resolver.c tests/test.h
 tests_test_resolver_CFLAGS = $(STROPHE_FLAGS) -I$(top_srcdir)/src
 tests_test_resolver_LDADD = $(STROPHE_LIBS)
-tests_test_resolver_LDFLAGS = -static
+tests_test_resolver_LDFLAGS =
 
 tests_test_rand_SOURCES = tests/test_rand.c tests/test.c src/sha1.c
 tests_test_rand_CFLAGS = $(STROPHE_FLAGS) -I$(top_srcdir)/src
@@ -124,4 +124,4 @@ tests_test_snprintf_CFLAGS = -I$(top_src
 tests_test_string_SOURCES = tests/test_string.c tests/test.h
 tests_test_string_CFLAGS = $(STROPHE_FLAGS) -I$(top_srcdir)/src
 tests_test_string_LDADD = $(STROPHE_LIBS)
-tests_test_string_LDFLAGS = -static
+tests_test_string_LDFLAGS =
