--- origsrc/libstrophe-0.8.6/Makefile.am	2014-08-09 01:49:46.000000000 +0900
+++ src/libstrophe-0.8.6/Makefile.am	2014-09-18 16:02:22.034919000 +0900
@@ -14,7 +14,7 @@ STROPHE_LIBS = libstrophe.la
 lib_LTLIBRARIES = libstrophe.la
 
 libstrophe_la_CFLAGS=$(STROPHE_FLAGS) $(PARSER_CFLAGS)
-libstrophe_la_LDFLAGS=$(SSL_LIBS) $(PARSER_LIBS)
+libstrophe_la_LDFLAGS=$(SSL_LIBS) $(PARSER_LIBS) -no-undefined
 libstrophe_la_SOURCES = src/auth.c src/conn.c src/ctx.c \
 	src/event.c src/handler.c src/hash.c \
 	src/jid.c src/md5.c src/sasl.c src/sha1.c \
