--- origsrc/whereami-1.0/whereami.c	2010-04-18 02:56:39.000000000 +0900
+++ src/whereami-1.0/whereami.c	2015-05-16 21:49:03.667068000 +0900
@@ -28,6 +28,7 @@
 #include <net/if.h>
 #include <inttypes.h>
 #include <sys/ioctl.h>
+#include <sys/socket.h>
 
 #define CWDLEN       4096
 #define VERSION     "1.0"
@@ -44,7 +45,6 @@ struct option lopt[] = \
     { NULL, 0, NULL, 0 }
 };
 
-extern int opterr;
 static char *prog = NULL;
 
 char *getname (void);                   /* get host name */
