--- origsrc/lwes-1.1.2/configure.ac	2018-02-13 07:57:21.000000000 +0900
+++ src/lwes-1.1.2/configure.ac	2020-02-06 19:14:25.816231600 +0900
@@ -130,7 +130,7 @@ AC_CHECK_FUNCS(gettimeofday socket strer
 
 dnl Checks for libraries.
 dnl Don't know if I need this, but it won't compile if flex is used without it
-AC_CHECK_LIB(fl,main)
+# AC_CHECK_LIB(fl,main)
 flex --nounput -V > /dev/null 2>&1
 if test $? -eq 0 ; then
   FLEX_EXTRA_ARGS="--nounput"
--- origsrc/lwes-1.1.2/src/Makefile.am	2018-02-13 07:57:21.000000000 +0900
+++ src/lwes-1.1.2/src/Makefile.am	2020-02-06 18:22:18.014054500 +0900
@@ -76,7 +76,7 @@ lib_LTLIBRARIES =							\
 
 liblwes_la_SOURCES = ${mysourcefiles}
 liblwes_la_LIBADD =
-liblwes_la_LDFLAGS = -version-info @MAJOR_VERSION@:@MINOR_VERSION@:0 @GCOV_LTFLAGS@
+liblwes_la_LDFLAGS = -no-undefined -version-info @MAJOR_VERSION@:@MINOR_VERSION@:0 @GCOV_LTFLAGS@
 
 bin_PROGRAMS = \
   lwes-event-printing-listener \
--- origsrc/lwes-1.1.2/tests/Makefile.am	2018-02-13 07:57:21.000000000 +0900
+++ src/lwes-1.1.2/tests/Makefile.am	2020-02-06 19:07:04.943804100 +0900
@@ -42,35 +42,19 @@ testtimefuncs_SOURCES = testtimefuncs.c
 testtimefuncs_LDADD =
 
 testhashtable_SOURCES = testhashtable.c
-testhashtable_LDADD = ../src/lwes_types.o
+testhashtable_LDADD = ../src/liblwes.la
 
 testeventtypedb_SOURCES = testeventtypedb.c
-testeventtypedb_LDADD = ../src/lwes_types.o \
-                        ../src/lwes_hash.o \
-                        ../src/lwes_esf_parser.o \
-                        ../src/lwes_esf_parser_y.o
+testeventtypedb_LDADD = ../src/liblwes.la
 
 testevent_SOURCES = testevent.c
-testevent_LDADD = ../src/lwes_types.o \
-                  ../src/lwes_hash.o \
-                  ../src/lwes_marshall_functions.o \
-                  ../src/lwes_esf_parser.o \
-                  ../src/lwes_esf_parser_y.o \
-                  ../src/lwes_event_type_db.o
+testevent_LDADD = ../src/liblwes.la
 
 testnetfuncs_SOURCES = testnetfuncs.c
-testnetfuncs_LDADD = ../src/lwes_types.o
+testnetfuncs_LDADD = ../src/liblwes.la
 
 testemitandlisten_SOURCES = testemitandlisten.c
-testemitandlisten_LDADD = ../src/lwes_types.o \
-                          ../src/lwes_event.o \
-                          ../src/lwes_hash.o \
-                          ../src/lwes_marshall_functions.o \
-                          ../src/lwes_esf_parser.o \
-                          ../src/lwes_esf_parser_y.o \
-                          ../src/lwes_event_type_db.o \
-                          ../src/lwes_net_functions.o \
-                          ../src/lwes_time_functions.o
+testemitandlisten_LDADD = ../src/liblwes.la
 
 testlwes_event_printing_listener_SOURCES = \
   testlwes-event-printing-listener.c
@@ -95,7 +79,7 @@ testlwes_calculate_max_event_size_LDADD
 # END: Variables to change
 # past here, hopefully, there is no need to edit anything
 
-INCLUDES = -I../src ${myincludes}
+INCLUDES = -I../src -I$(top_srcdir)/src ${myincludes}
 
 check_PROGRAMS = $(mytests)
 
