--- origsrc/SQUIRREL3/Makefile.am	2014-12-28 16:26:49.159101500 +0900
+++ src/SQUIRREL3/Makefile.am	2014-12-28 16:33:57.507734300 +0900
@@ -55,7 +55,7 @@ libsqstdlib_la_SOURCES = sqstdlib/sqstda
 	sqstdlib/sqstdstream.cpp sqstdlib/sqstdstream.h \
 	sqstdlib/sqstdstring.cpp \
 	sqstdlib/sqstdsystem.cpp
-libsqstdlib_la_LDFLAGS = -release 3.0.4
+libsqstdlib_la_LDFLAGS = -no-undefined -release 3.0.6
 libsqstdlib_la_LIBADD = libsquirrel.la
 
 libsquirrel_la_SOURCES = squirrel/sqapi.cpp \
@@ -78,7 +78,7 @@ libsquirrel_la_SOURCES = squirrel/sqapi.
 	squirrel/squserdata.h \
 	squirrel/squtils.h \
 	squirrel/sqvm.cpp squirrel/sqvm.h
-libsquirrel_la_LDFLAGS = -release 3.0.4
+libsquirrel_la_LDFLAGS = -no-undefined -release 3.0.6
 
 sq_sq_SOURCES = sq/sq.c
 sq_sq_LDADD = libsqstdlib.la libsquirrel.la
--- origsrc/SQUIRREL3/configure.ac	2014-12-28 16:26:49.143476500 +0900
+++ src/SQUIRREL3/configure.ac	2014-12-28 16:28:30.626875000 +0900
@@ -1,4 +1,4 @@
-AC_INIT([squirrel],[3.0.4],[http://squirrel-lang.org/])
+AC_INIT([squirrel],[3.0.6],[http://squirrel-lang.org/])
 
 # recommended by libtoolize:
 AC_CONFIG_MACRO_DIR([m4])
