--- origsrc/SQUIRREL2/configure.ac	2010-12-19 16:45:14.690353700 +0900
+++ src/SQUIRREL2/configure.ac	2010-12-19 16:44:07.000000000 +0900
@@ -12,7 +12,7 @@ AC_PROG_CXX
 AM_PROG_CC_STDC
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
-LT_INIT(win32-dll)
+AC_PROG_LIBTOOL
 
 # Produce output
 AC_CONFIG_FILES([Makefile
--- origsrc/SQUIRREL2/sqstdlib/Makefile.am	2010-12-19 17:51:20.355610000 +0900
+++ src/SQUIRREL2/sqstdlib/Makefile.am	2010-12-19 17:37:01.300349200 +0900
@@ -3,7 +3,9 @@ AM_CXXFLAGS = -fno-rtti
 
 lib_LTLIBRARIES = libsqstdlib.la
 
-libsqstdlib_la_LDFLAGS = -release $(VERSION)
+libsqstdlib_la_LDFLAGS = -release $(VERSION) -no-undefined
+
+libsqstdlib_la_LIBADD = $(top_builddir)/squirrel/libsquirrel.la
 
 libsqstdlib_la_SOURCES = \
 			sqstdaux.cpp \
--- origsrc/SQUIRREL2/squirrel/Makefile.am	2010-12-19 17:51:35.602534000 +0900
+++ src/SQUIRREL2/squirrel/Makefile.am	2010-12-19 17:14:17.203874800 +0900
@@ -3,7 +3,7 @@ AM_CXXFLAGS = -fno-rtti -Wall
 
 lib_LTLIBRARIES = libsquirrel.la
 
-libsquirrel_la_LDFLAGS = -release $(VERSION)
+libsquirrel_la_LDFLAGS = -release $(VERSION) -no-undefined
 
 libsquirrel_la_SOURCES = \
 		sqapi.cpp \
