--- origsrc/SQUIRREL2/sqstdlib/Makefile.am	2012-04-07 13:45:05.981798000 +0900
+++ src/SQUIRREL2/sqstdlib/Makefile.am	2012-04-07 13:50:56.897813600 +0900
@@ -3,7 +3,7 @@ AM_CXXFLAGS = -fno-rtti
 
 lib_LTLIBRARIES = libsqstdlib.la
 
-libsqstdlib_la_LDFLAGS = -release $(VERSION)
+libsqstdlib_la_LDFLAGS = -release $(VERSION) -no-undefined
 
 libsqstdlib_la_SOURCES = \
 			sqstdaux.cpp \
@@ -17,4 +17,6 @@ libsqstdlib_la_SOURCES = \
 			sqstdstring.cpp \
 			sqstdsystem.cpp
 
+libsqstdlib_la_LIBADD = $(top_builddir)/squirrel/libsquirrel.la
+
 INCLUDES = -I$(top_srcdir)/include -I.
--- origsrc/SQUIRREL2/squirrel/Makefile.am	2012-04-07 13:45:05.981798000 +0900
+++ src/SQUIRREL2/squirrel/Makefile.am	2012-04-07 13:45:41.091173000 +0900
@@ -3,7 +3,7 @@ AM_CXXFLAGS = -fno-rtti -Wall
 
 lib_LTLIBRARIES = libsquirrel.la
 
-libsquirrel_la_LDFLAGS = -release $(VERSION)
+libsquirrel_la_LDFLAGS = -release $(VERSION) -no-undefined
 
 libsquirrel_la_SOURCES = \
 		sqapi.cpp \
