--- origsrc/libanim/configure.ac	2024-11-29 15:11:51.000000000 +0900
+++ src/libanim/configure.ac	2024-11-29 15:20:28.703642800 +0900
@@ -9,7 +9,8 @@ AC_CONFIG_HEADER([config.h])
 
 # Checks for programs.
 AC_PROG_CC
-AC_PROG_LIBTOOL
+AM_PROG_AR
+LT_INIT
 
 # Checks for libraries.
 PKG_CHECK_MODULES(DEPS, glib-2.0)
--- origsrc/libanim/src/Makefile.am	2024-11-29 15:11:51.000000000 +0900
+++ src/libanim/src/Makefile.am	2024-11-29 15:19:43.279327300 +0900
@@ -6,13 +6,14 @@ include_HEADERS = anim.h
 TESTS = test
 
 example_SOURCES = example.c
-example_CFLAGS  = -std=c99 -Wall -Werror $(DEPS_CFLAGS) -g
-example_LDADD   = $(DEPS_LIBS) -lanim -lm
+example_CFLAGS  = $(DEPS_CFLAGS)
+example_LDADD   = $(DEPS_LIBS) libanim.la -lm
 
 libanim_la_SOURCES = anim.c
-libanim_la_CFLAGS  = -ansi -Wall -Werror $(DEPS_CFLAGS) -g
+libanim_la_CFLAGS  = -ansi $(DEPS_CFLAGS)
 libanim_la_LIBADD  = $(DEPS_LIBS)
+libanim_la_LDFLAGS = -no-undefined
 
 test_SOURCES = test.c
-test_CFLAGS  = -ansi -Wall -Werror $(DEPS_CFLAGS) -g
-test_LDADD   = $(DEPS_LIBS) -lanim -lm
+test_CFLAGS  = -ansi $(DEPS_CFLAGS)
+test_LDADD   = $(DEPS_LIBS) libanim.la -lm
