--- origsrc/libsrs_alt-1.0/Makefile.am	2006-06-15 21:26:43.000000000 +0900
+++ src/libsrs_alt-1.0/Makefile.am	2014-05-15 00:08:56.610512700 +0900
@@ -1,13 +1,11 @@
 SUBDIRS = lib src .
 EXTRA_DIST = MTAs
 
-LDADD = lib/libcompat.a
-
 bin_PROGRAMS = srs
 noinst_PROGRAMS = test
 
 test_SOURCES = test.c
-test_LDFLAGS = src/.libs/libsrs_alt.a
+test_LDFLAGS = src/libsrs_alt.la
 
 srs_SOURCES = srs.c
-srs_LDFLAGS = src/.libs/libsrs_alt.a
+srs_LDFLAGS = src/libsrs_alt.la
--- origsrc/libsrs_alt-1.0/configure.ac	2006-11-21 07:20:59.000000000 +0900
+++ src/libsrs_alt-1.0/configure.ac	2014-05-15 00:08:56.610512700 +0900
@@ -15,8 +15,7 @@ AC_PROG_CPP
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
-AC_PROG_LIBTOOL
-AC_PROG_RANLIB
+LT_INIT
 
 # Checks for header files.
 AC_HEADER_STDC
--- origsrc/libsrs_alt-1.0/lib/Makefile.am	2005-07-09 03:58:04.000000000 +0900
+++ src/libsrs_alt-1.0/lib/Makefile.am	2014-05-15 00:08:56.626137700 +0900
@@ -1,10 +1,5 @@
 EXTRA_DIST = hmac.h sha1.h
 
-noinst_LIBRARIES = libcompat.a
-libcompat_a_SOURCES = hmac.c sha1.c
-libcompat_a_CFLAGS = $(AM_CFLAGS)
-libcompat_a_LIBADD = $(LIBOBJS) $(ALLOCA)
-
 noinst_LTLIBRARIES = libcompat.la
 libcompat_la_SOURCES = hmac.c sha1.c
 libcompat_la_CFLAGS = $(AM_CFLAGS)
--- origsrc/libsrs_alt-1.0/src/Makefile.am	2005-07-09 03:38:33.000000000 +0900
+++ src/libsrs_alt-1.0/src/Makefile.am	2014-05-15 00:08:56.626137700 +0900
@@ -2,15 +2,8 @@ AM_CFLAGS =
 EXTRA_DIST = address.h hash.h srs_alt.h timestamp.h
 include_HEADERS = srs_alt.h
 
-LDADD = ../lib/libcompat.a
-
-lib_LIBRARIES = libsrs_alt.a
-libsrs_alt_a_SOURCES = address.c hash.c srs_alt.c timestamp.c
-libsrs_alt_a_CFLAGS = $(AM_CFLAGS)
-libsrs_alt_a_LIBADD = ../lib/libcompat.a
-
 lib_LTLIBRARIES = libsrs_alt.la
-libsrs_alt_la_LDFLAGS = -version-info 1:0
+libsrs_alt_la_LDFLAGS = -version-info 1:0 -no-undefined
 libsrs_alt_la_SOURCES = address.c hash.c srs_alt.c timestamp.c
 libsrs_alt_la_CFLAGS = $(AM_CFLAGS)
-libsrs_alt_la_LIBADD = ../lib/libcompat.la
\ No newline at end of file
+libsrs_alt_la_LIBADD = ../lib/libcompat.la
--- origsrc/libsrs_alt-1.0/test.c	2005-03-31 20:53:29.000000000 +0900
+++ src/libsrs_alt-1.0/test.c	2014-05-15 00:13:33.572169000 +0900
@@ -1,5 +1,6 @@
 #include <string.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <time.h>
 #include <sys/timeb.h>
 
