--- origsrc/gopt-10.0/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/gopt-10.0/Makefile.am	2024-05-10 17:16:35.344602700 +0900
@@ -0,0 +1,12 @@
+noinst_PROGRAMS = gopt-usage
+gopt_usage_SOURCES = gopt-usage.c
+gopt_usage_LDADD = libgopt.la
+
+include_HEADERS = gopt.h
+
+lib_LTLIBRARIES = libgopt.la
+libgopt_la_SOURCES = gopt.c gopt-errors.c
+libgopt_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/gopt-10.0/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/gopt-10.0/configure.ac	2024-05-10 17:16:06.199297700 +0900
@@ -0,0 +1,27 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.72])
+AC_INIT([gopt], [10.0])
+AC_CONFIG_SRCDIR([gopt-errors.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([strings.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_CHECK_FUNCS([strchr strcspn strncasecmp])
+
+AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([gopt.pc])
+AC_OUTPUT
--- origsrc/gopt-10.0/gopt-usage.c	2017-01-03 21:00:55.000000000 +0900
+++ src/gopt-10.0/gopt-usage.c	2024-05-10 17:15:04.366746100 +0900
@@ -7,6 +7,8 @@
 #include <string.h>
 #include "gopt.h"
 
+#include "config.h"
+
 int main (int argc, char **argv)
 {
   struct option options[5];
@@ -44,7 +46,7 @@ int main (int argc, char **argv)
 
   if (options[1].count)
   {
-    fprintf (stdout, "version 1.0\n");
+    fprintf (stdout, "version %s\n", VERSION);
     exit (EXIT_SUCCESS);
   }
 
--- origsrc/gopt-10.0/gopt.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/gopt-10.0/gopt.pc.in	2024-05-10 17:16:06.152417900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: gopt
+Description: Functions for parsing POSIX and GNU style command line arguments
+Version: @VERSION@
+Libs: -L${libdir} -lgopt
+Cflags: -I${includedir}
