--- origsrc/libfcache-20240414/Makefile.am	2024-04-14 17:14:30.000000000 +0900
+++ src/libfcache-20240414/Makefile.am	2025-05-17 03:12:04.438743200 +0900
@@ -3,11 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
 	libfcache \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libfcache-20240414/configure.ac	2024-04-14 17:14:30.000000000 +0900
+++ src/libfcache-20240414/configure.ac	2025-05-17 03:12:04.438743200 +0900
@@ -97,9 +97,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
 AC_CONFIG_FILES([libfcache/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
