--- origsrc/squeakr-0.7/include/gqf_cpp.h	2019-08-13 13:20:33.000000000 +0900
+++ src/squeakr-0.7/include/gqf_cpp.h	2020-07-24 19:41:24.820358200 +0900
@@ -29,6 +29,10 @@
 #define NUM_Q_BITS 16
 #define SEED 2038074761
 
+#if defined(FREAD)
+#undef FREAD
+#endif
+
 enum readmode {
 	MMAP,
 	FREAD
--- origsrc/squeakr-0.7/src/gqf/partitioned_counter.c	2019-08-13 13:20:33.000000000 +0900
+++ src/squeakr-0.7/src/gqf/partitioned_counter.c	2020-07-24 19:38:44.157809600 +0900
@@ -13,8 +13,12 @@
 #include <unistd.h>
 #include <sched.h>
 #include <sys/sysinfo.h>
+#if defined(__linux__)
 #include <linux/unistd.h>
+#endif
+#if !defined(__CYGWIN__)
 #include <sys/syscall.h>
+#endif
 #include <errno.h>
 
 #include "gqf/partitioned_counter.h"
