--- origsrc/MemeAssembly-1.6/compiler/compiler.c	2023-05-07 21:30:00.000000000 +0900
+++ src/MemeAssembly-1.6/compiler/compiler.c	2023-08-18 17:15:12.489239700 +0900
@@ -431,13 +431,13 @@ void compile(struct compileState compile
     } else {
         char* commandPrefix;
         if(compileState.outputMode == objectFile) {
-            #ifndef LINUX
+            #if !defined(LINUX) || defined(__CYGWIN__)
             commandPrefix = "gcc -w -O -c -x assembler - -o";
             #else
             commandPrefix = "gcc -z execstack -w -O -c -x assembler - -o";
             #endif
         } else {
-            #ifndef LINUX
+            #if !defined(LINUX) || defined(__CYGWIN__)
             commandPrefix = "gcc -w -O -x assembler - -o";
             #else
             commandPrefix = "gcc -z execstack -w -O -no-pie -x assembler - -o"; //-no-pie is only defined because for some reason, the generated stabs info does not work when a PIE object is generated
