--- origsrc/prime_server-0.6.7/Makefile.am	2020-08-04 23:52:06.000000000 +0900
+++ src/prime_server-0.6.7/Makefile.am	2020-09-15 10:46:00.806325400 +0900
@@ -54,6 +54,7 @@ libprime_server_la_SOURCES = \
 	src/http_protocol.cpp
 libprime_server_la_CPPFLAGS = $(DEPS_CFLAGS)
 libprime_server_la_LIBADD = $(DEPS_LIBS)
+libprime_server_la_LDFLAGS = -no-undefined
 
 #distributed executables
 bin_PROGRAMS = \
