--- origsrc/libmaitretarot-0.1.98/src/Makefile.am	2003-02-08 08:03:06.000000000 +0900
+++ src/libmaitretarot-0.1.98/src/Makefile.am	2014-07-24 07:59:51.874771800 +0900
@@ -11,4 +11,4 @@ libmaitretarot_la_SOURCES= \
 	config_utils.c config_utils.h \
 	libmt_xmlconfig.c
 
-libmaitretarot_la_LDFLAGS=${GLIB_LIBS} -version-info ${LIBMT_MAJOR}:${LIBMT_MICRO}:${LIBMT_MINOR}
+libmaitretarot_la_LDFLAGS=${GLIB_LIBS} -no-undefined -version-info ${LIBMT_MAJOR}:${LIBMT_MICRO}:${LIBMT_MINOR}
