--- origsrc/libsearpc-3.2.0/configure.ac	2020-03-20 17:01:27.000000000 +0900
+++ src/libsearpc-3.2.0/configure.ac	2024-10-09 22:39:03.429249200 +0900
@@ -2,7 +2,7 @@
 # Process this file with autoconf to produce a configure script.
 
 AC_PREREQ([2.61])
-AC_INIT([libsearpc], [3.1.0], [info@seafile.com])
+AC_INIT([libsearpc], [3.2.0], [info@seafile.com])
 AC_CONFIG_SRCDIR([lib/searpc-server.c])
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_MACRO_DIR([m4])
--- origsrc/libsearpc-3.2.0/libsearpc.pc.in	2020-03-20 17:01:27.000000000 +0900
+++ src/libsearpc-3.2.0/libsearpc.pc.in	2024-10-09 22:39:03.429249200 +0900
@@ -1,4 +1,4 @@
-prefix=(DESTDIR)@prefix@
+prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
