--- origsrc/EasySoap++-0.8.0/EasySoap++.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/EasySoap++-0.8.0/EasySoap++.pc.in	2015-03-18 21:44:46.145336000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: EasySoap++
+Description: C++ library for SOAP
+Version: @VERSION@
+Libs: -L${libdir} -leasysoap
+Cflags: -I${includedir}
--- origsrc/EasySoap++-0.8.0/Makefile.am	2003-07-25 18:05:52.000000000 +0900
+++ src/EasySoap++-0.8.0/Makefile.am	2015-03-18 21:45:07.582836000 +0900
@@ -17,3 +17,6 @@ google: src
 
 AM_CFLAGS=-Wall
 AM_CXXFLAGS=-Wall
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = EasySoap++.pc
--- origsrc/EasySoap++-0.8.0/configure.in	2006-11-10 10:45:42.000000000 +0900
+++ src/EasySoap++-0.8.0/configure.in	2015-03-18 21:45:39.672679700 +0900
@@ -3,10 +3,10 @@ dnl Process this file with autoconf to p
 dnl
 dnl
 
-AC_INIT(Makefile.am)
+AC_INIT([EasySoap++], [0.8.0])
 AC_CONFIG_AUX_DIR(conftools)
 
-AM_INIT_AUTOMAKE(EasySoap++, 0.8.0)
+AM_INIT_AUTOMAKE
 AM_CONFIG_HEADER(include/easysoap/es_config.h)
 
 dnl
@@ -17,8 +17,6 @@ sinclude(conftools/ac_cxx_template_keywo
 sinclude(conftools/ac_cxx_partial_specialization.m4)
 sinclude(conftools/peti_with_dmalloc.m4)
 
-AC_CYGWIN
-
 dnl Checks for programs.
 AC_PROG_CXX
 AC_PROG_AWK
@@ -43,35 +41,13 @@ AC_CHECK_LIB(m, isnan)
 AC_CHECK_LIB(stdc++, main)
 AC_CHECK_LIB(nsl, gethostbyname)
 AC_CHECK_LIB(socket, send)
-AC_CHECK_LIB(wsock32, WSAStartup)
-AC_CHECK_LIB(crypto, ERR_put_error)
-AC_CHECK_LIB(ssl, SSL_connect)
-AC_CHECK_LIB(expat, XML_Parse,,
-	AC_MSG_ERROR("Could not find the XML parsing library 'expat'.  Download and install from http://expat.sourceforge.net/"))
-AC_CHECK_HEADER(expat.h,,
-	AC_MSG_ERROR("Could not find expat.h"))
-
-dnl Check for zlib
-AC_ARG_WITH(zlib,[
-AC_HELP_STRING([--with-zlib=PATH], [search for zlib in PATH])
-AC_HELP_STRING([--without-zlib], [disable use of zlib])],[
-  case "$withval" in
-  no)
-    AC_MSG_WARN([zlib disabled]) ;;
-  *)
-    if test -d "$withval"; then
-      CPPFLAGS="$CPPFLAGS -I$withval/include"
-      LDFLAGS="$LDFLAGS -L$withval/lib"
-    fi
-
-    AC_CHECK_HEADER(zlib.h,[
-      AC_CHECK_LIB(z, gzread,[
-        AC_DEFINE(HAVE_LIBZ, 1, [If zlib is available])
-      ],),
-    ])
-    ;;
-  esac
-])
+if test "$host_os" != "cygwin"; then
+    AC_CHECK_LIB(wsock32, WSAStartup)
+fi
+
+PKG_CHECK_MODULES([OPENSSL], [openssl])
+PKG_CHECK_MODULES([EXPAT], [expat])
+PKG_CHECK_MODULES([ZLIB], [zlib])
 
 dnl Checks for header files.
 AC_HEADER_DIRENT
@@ -150,5 +126,6 @@ AC_OUTPUT( \
 	validator/Makefile \
 	tests/Makefile \
 	google/Makefile \
+	EasySoap++.pc \
 )
 
--- origsrc/EasySoap++-0.8.0/src/Makefile.am	2003-06-17 01:11:00.000000000 +0900
+++ src/EasySoap++-0.8.0/src/Makefile.am	2015-03-18 21:41:52.220531300 +0900
@@ -1,5 +1,4 @@
 lib_LTLIBRARIES = libeasysoap.la
-lib_LIBRARIES = libeasysoap.a
 THESOURCES = \
 	digcalc.c \
 	md5c.c \
@@ -35,7 +34,6 @@ THESOURCES = \
 	XMLParser.cpp 
 
 libeasysoap_la_SOURCES = $(THESOURCES)
-libeasysoap_a_SOURCES = $(THESOURCES)
 noinst_HEADERS = \
 	digcalc.h \
 	global.h \
@@ -53,14 +51,11 @@ noinst_HEADERS = \
 INCLUDES=-I@top_srcdir@/include 
 
 libeasysoap_la_LDFLAGS = -no-undefined 
-
+libeasysoap_la_LIBADD = $(EXPAT_LIBS)
 
 AM_CFLAGS=-Wall -pedantic -I/usr/kerberos/include
 AM_CXXFLAGS=-Wall -pedantic -I/usr/kerberos/include
 
-libeasysoap_a_CXXFLAGS = $(AM_CXXFLAGS)
-libeasysoap_a_CFLAGS = $(AM_CFLAGS)
-
 EXTRA_DIST = \
 	digtest.c \
 	mddriver.c \
--- origsrc/EasySoap++-0.8.0/tests/Makefile.am	2002-10-29 22:35:00.000000000 +0900
+++ src/EasySoap++-0.8.0/tests/Makefile.am	2015-03-18 21:43:13.527171900 +0900
@@ -1,4 +1,4 @@
-noinst_PROGRAMS = interopserver interopclient SSLClient SSLServer
+noinst_PROGRAMS = interopserver interopclient SSLServer
 
 interopserver_SOURCES = \
 	interopstruct.h \
