--- origsrc/OpenJPH-0.21.3/tests/CMakeLists.txt	2025-05-29 12:04:12.000000000 +0900
+++ src/OpenJPH-0.21.3/tests/CMakeLists.txt	2025-06-20 12:33:59.996144400 +0900
@@ -1,6 +1,9 @@
+include(FetchContent)
+
+find_Package(GTest)
+if (NOT GTest_FOUND)
 
 # Fetch googletest
-include(FetchContent)
 FetchContent_Declare(
   googletest
   URL https://github.com/google/googletest/archive/refs/tags/v1.14.0.tar.gz
@@ -12,6 +15,8 @@ set(BUILD_GMOCK OFF CACHE BOOL "" FORCE)
 option(INSTALL_GTEST "Enable installation of googletest." OFF)
 FetchContent_MakeAvailable(googletest)
 
+endif()
+
 # Fetch test files
 FetchContent_Declare(
   jp2k_test_codestreams
@@ -40,8 +45,12 @@ target_link_libraries(
   GTest::gtest_main
 )
 
+if (GTest_FOUND)
+add_test(Tests test_executables)
+else()
 include(GoogleTest)
 gtest_add_tests(TARGET test_executables)
+endif()
 
 if (MSVC)
   add_custom_command(TARGET test_executables POST_BUILD
