--- origsrc/libgwords/src/meson.build	2025-08-14 20:45:28.000000000 +0900
+++ src/libgwords/src/meson.build	2025-08-14 20:49:15.077391900 +0900
@@ -97,7 +97,6 @@ libgwords = shared_library(
               sources: libgwords_sources,
             soversion: gw_major_version,
               version: gw_major_version,
-          install_dir: gw_libdir,
         install_rpath: gw_libdir_abs,
   include_directories: gw_include_dirs,
          dependencies: deps,
