--- origsrc/cppglob-1.1.0/CMakeLists.txt	2018-12-03 09:59:51.000000000 +0900
+++ src/cppglob-1.1.0/CMakeLists.txt	2020-11-07 04:59:05.126675400 +0900
@@ -122,9 +122,15 @@ install(
   DESTINATION include
   COMPONENT headers)
 
-install(
-  EXPORT cppglob-config
-  DESTINATION share/cmake/cppglob)
+if (CYGWIN)
+  install(
+    EXPORT cppglob-config
+    DESTINATION lib/cmake/cppglob)
+else()
+  install(
+    EXPORT cppglob-config
+    DESTINATION share/cmake/cppglob)
+endif()
 
 add_custom_target(uninstall
   COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/cmake_uninstall.cmake
--- origsrc/cppglob-1.1.0/src/CMakeLists.txt	2018-12-03 09:59:51.000000000 +0900
+++ src/cppglob-1.1.0/src/CMakeLists.txt	2020-11-07 04:57:57.363422400 +0900
@@ -17,7 +17,8 @@ if (BUILD_SHARED)
   install(
     TARGETS cppglob
     EXPORT cppglob-config
-    DESTINATION lib
+    RUNTIME DESTINATION bin
+    ARCHIVE DESTINATION lib
     COMPONENT libs)
 
 endif()
