--- origsrc/safestr-1.0.3/Makefile.am	2005-01-31 02:55:39.000000000 +0900
+++ src/safestr-1.0.3/Makefile.am	2011-10-21 09:04:31.801285600 +0900
@@ -13,7 +13,7 @@ DIST_SUBDIRS			= xxl-1.0.1
 endif
 
 lib_LTLIBRARIES			= libsafestr.la
-libsafestr_la_LDFLAGS		= -version-info $(SAFESTR_VERSION_CURRENT):$(SAFESTR_VERSION_REVISION):$(SAFESTR_VERSION_AGE)
+libsafestr_la_LDFLAGS		= -version-info $(SAFESTR_VERSION_CURRENT):$(SAFESTR_VERSION_REVISION):$(SAFESTR_VERSION_AGE) -no-undefined
 dist_libsafestr_la_SOURCES	= safestr.c			\
 				  isafestr.h			\
 				  safefmt.c			\
@@ -26,7 +26,7 @@ nodist_include_HEADERS		= safestr.h			\
 check_PROGRAMS			= safetest
 dist_safetest_SOURCES		= safetest.c
 nodist_safetest_SOURCES		= safestr.h
-safetest_LDADD			= -L. -lsafestr
+safetest_LDADD			= $(top_builddir)/libsafestr.la
 TESTS				= safetest
 
 distclean-local:
--- origsrc/safestr-1.0.3/configure.ac	2005-01-31 03:04:16.000000000 +0900
+++ src/safestr-1.0.3/configure.ac	2011-10-21 09:07:45.525911200 +0900
@@ -1,10 +1,10 @@
 AC_PREREQ(2.57)
 AC_COPYRIGHT([(c) Copyright 2003-2005 Matt Messier and John Viega])
 AC_INIT([safestr], [1.0.3], [safestr@zork.org])
+AC_CONFIG_AUX_DIR(.)
 AM_INIT_AUTOMAKE([1.7 foreign])
 AM_CONFIG_HEADER([config.h safestr.h])
 AC_CONFIG_SRCDIR([safestr.h.in])
-AC_CONFIG_AUX_DIR(.)
 AC_PREFIX_DEFAULT([/usr/local])
 
 dnl When you change this version number here, make sure that you also change
@@ -126,5 +126,4 @@ if test "${ac_build_xxl}" = "yes"; then
 fi
 
 AC_CONFIG_FILES([Makefile], [chmod -w Makefile])
-AC_CONFIG_SUBDIRS([xxl-1.0.1])
 AC_OUTPUT
