--- origsrc/libisasl-1.0.0/Makefile.am	2012-02-15 04:33:32.000000000 +0900
+++ src/libisasl-1.0.0/Makefile.am	2025-01-19 19:16:07.965888800 +0900
@@ -25,3 +25,5 @@ pkginclude_HEADERS = \
 libsasl_la_SOURCES=src/client.c
 libsasl_la_LDFLAGS=-version-info $(LIBISASL_API_CURRENT):$(LIBISASL_API_REVISION):$(LIBISASL_API_AGE) -no-undefined
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = libsasl.pc
--- origsrc/libisasl-1.0.0/configure.ac	2012-02-15 04:33:32.000000000 +0900
+++ src/libisasl-1.0.0/configure.ac	2025-01-19 19:16:07.981283700 +0900
@@ -14,8 +14,7 @@
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
 AC_PREREQ(2.60)
-m4_include([m4/version.m4])
-AC_INIT(libisasl, VERSION_NUMBER, support@couchbase.com)
+AC_INIT(libisasl, [1.0.0], support@couchbase.com)
 AC_CONFIG_SRCDIR([src/client.c])
 AC_CONFIG_AUX_DIR(config)
 AC_USE_SYSTEM_EXTENSIONS
@@ -50,5 +49,5 @@ AC_SUBST(LIBISASL_API_AGE)
 
 dnl ----------------------------------------------------------------------------
 
-AC_CONFIG_FILES(Makefile)
+AC_CONFIG_FILES(libsasl.pc Makefile)
 AC_OUTPUT
--- origsrc/libisasl-1.0.0/libsasl.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libisasl-1.0.0/libsasl.pc.in	2025-01-19 19:16:07.981283700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libsasl
+Description: Small library providing SASL authentication
+Version: @VERSION@
+Libs: -L${libdir} -lsasl
+Cflags: -I${includedir}
