--- origsrc/astrochem-0.9/Makefile.am	2021-10-22 05:34:59.000000000 +0900
+++ src/astrochem-0.9/Makefile.am	2024-12-25 13:10:49.194000600 +0900
@@ -22,3 +22,6 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = src doc tests python networks
 
 EXTRA_DIST = AUTHORS.md CONTRIBUTING.md COPYING.md INSTALL.md README.md
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/astrochem-0.9/astrochem.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/astrochem-0.9/astrochem.pc.in	2024-12-25 13:10:49.162750700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: astrochem
+Description: Code to compute the abundances of chemical species in the interstellar medium
+Version: @VERSION@
+Libs: -L${libdir} -lastrochem
+Cflags: -I${includedir}
--- origsrc/astrochem-0.9/configure.ac	2021-10-22 05:34:59.000000000 +0900
+++ src/astrochem-0.9/configure.ac	2024-12-25 13:10:49.211002300 +0900
@@ -130,6 +130,7 @@ AC_CONFIG_FILES([Makefile
 		 tests/benchmark_openmp.py
 		 tests/Makefile])
 
+AC_CONFIG_FILES([astrochem.pc])
 AC_OUTPUT
 
 # Print configuration summary
--- origsrc/astrochem-0.9/python/libpyastrochem/Makefile.am	2021-10-22 05:34:59.000000000 +0900
+++ src/astrochem-0.9/python/libpyastrochem/Makefile.am	2024-12-25 13:24:33.272663900 +0900
@@ -29,7 +29,7 @@ pkgpyexec_LTLIBRARIES = libpyastrochem.l
 libpyastrochem_la_SOURCES = libpyastrochem.pyx
 libpyastrochem_la_CFLAGS = $(PYTHON_CPPFLAGS) $(CYTHON_CFLAGS)
 libpyastrochem_la_LIBADD = $(top_builddir)/src/libastrochem.la
-libpyastrochem_la_LDFLAGS = $(CYTHON_LDFLAGS) $(PYTHON_LDFLAGS)  -version-info 0:0:0 -module
+libpyastrochem_la_LDFLAGS = -no-undefined $(CYTHON_LDFLAGS) $(PYTHON_LDFLAGS) -avoid-version -module
 
 endif
 
--- origsrc/astrochem-0.9/src/Makefile.am	2021-10-22 05:34:59.000000000 +0900
+++ src/astrochem-0.9/src/Makefile.am	2024-12-25 12:57:52.744054900 +0900
@@ -24,13 +24,13 @@ AM_CFLAGS += ${OPENMP_CFLAGS} -Wall
 
 lib_LTLIBRARIES = libastrochem.la
 libastrochem_la_SOURCES = network.c rates.c solve.c input.c
-libastrochem_la_LDFLAGS = -version-info 0:0:0 -lsundials_nvecserial -lsundials_cvode -lm
+libastrochem_la_LDFLAGS = -no-undefined -version-info 0:0:0 -lsundials_nvecserial -lsundials_cvode -lm
 include_HEADERS = libastrochem.h
 noinst_HEADERS = astrochem.h input.h libastrochem.h network.h rates.h	\
 	solve.h
 
 bin_PROGRAMS = astrochem
-bin_SCRIPTS = chmconvert converttolegacy
+bin_SCRIPTS = chmconvert$(EXEEXT) converttolegacy$(EXEEXT)
 
 astrochem_SOURCES = astrochem.c 
 astrochem_CFLAGS = ${HDF5_CFLAGS} ${AM_CFLAGS}
--- origsrc/astrochem-0.9/src/chmconvert.py.in	2021-10-22 05:34:59.000000000 +0900
+++ src/astrochem-0.9/src/chmconvert.py.in	2024-12-25 13:25:07.463681200 +0900
@@ -1,4 +1,4 @@
-#!/usr/bin/env @PYTHON@
+#!@PYTHON@
 #
 #  chmconvert - Convert chemical networks into .chm format
 #
--- origsrc/astrochem-0.9/src/converttolegacy.py.in	2021-10-22 05:34:59.000000000 +0900
+++ src/astrochem-0.9/src/converttolegacy.py.in	2024-12-25 13:25:13.244376400 +0900
@@ -1,4 +1,4 @@
-#!/usr/bin/env @PYTHON@
+#!@PYTHON@
 #
 #  converttolegacy - Plot the abundances computed by Astrochem
 #
--- origsrc/astrochem-0.9/tests/Makefile.am	2021-10-22 05:34:59.000000000 +0900
+++ src/astrochem-0.9/tests/Makefile.am	2024-12-25 13:00:41.560228400 +0900
@@ -23,7 +23,7 @@ TESTS = input_test source_test source_dy
 
 check_PROGRAMS = input_test source_test source_dyn_test network_test rate_test abundances_test solver_test
 
-check_SCRIPTS = python_lib_test python_tools_test astrochem_test
+check_SCRIPTS = python_lib_test$(EXEEXT) python_tools_test$(EXEEXT) astrochem_test$(EXEEXT)
 
 input_test_SOURCES = input_test.c
 input_test_LDADD = $(top_builddir)/src/input.o $(top_builddir)/src/libastrochem.la
@@ -50,7 +50,7 @@ astrochem_test$(EXEEXT): astrochem_test.
 	cp $? $@
 	chmod a+x $@
 
-python_lib_test$(EXEXT): python_lib_test.py
+python_lib_test$(EXEEXT): python_lib_test.py
 	echo $(top_builddir)/libtool --mode=execute -dlopen $(top_builddir)/src/libastrochem.la @PYTHON@ python_lib_test.py > $@
 	chmod a+x $@
 
