--- origsrc/diamond-0.7.9/src/Makefile	2015-04-06 01:44:16.000000000 +0900
+++ src/diamond-0.7.9/src/Makefile	2015-07-23 20:52:31.186171500 +0900
@@ -2,7 +2,7 @@ WARN=-Wall -Wno-uninitialized
 C=gcc
 CXX=g++
 CFLAGS=-O3 -DNDEBUG
-CXXFLAGS=-O3 -DNDEBUG -Iboost/include $(WARN)
+CXXFLAGS=-O3 -DNDEBUG -I/usr/include/boost $(WARN)
 
 ifneq ($(wildcard boost/.),)
 	LIBS=boost/lib/libboost_thread.a boost/lib/libboost_system.a boost/lib/libboost_timer.a boost/lib/libboost_chrono.a boost/lib/libboost_iostreams.a boost/lib/libboost_program_options.a
