diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/diamond-0.7.11/src/Makefile src/diamond-0.7.11/src/Makefile
--- origsrc/diamond-0.7.11/src/Makefile	2016-01-24 19:03:47.000000000 +0900
+++ src/diamond-0.7.11/src/Makefile	2016-03-05 20:26:38.223965900 +0900
@@ -2,7 +2,7 @@ WARN=-Wall -Wno-uninitialized
 C=gcc
 CXX=g++
 CFLAGS=-O3 -DNDEBUG
-CXXFLAGS=-O3 -DNDEBUG -Iboost/include $(WARN)
+CXXFLAGS=-O3 -DNDEBUG -I/usr/include/boost $(WARN)
 
 ifneq ($(wildcard boost/.),)
 	LIBS=boost/lib/libboost_thread.a boost/lib/libboost_system.a boost/lib/libboost_timer.a boost/lib/libboost_chrono.a boost/lib/libboost_iostreams.a boost/lib/libboost_program_options.a
