--- origsrc/diamond-2.1.17/CMakeLists.txt	2025-12-21 21:04:49.000000000 +0900
+++ src/diamond-2.1.17/CMakeLists.txt	2026-01-05 06:17:50.566451900 +0900
@@ -476,7 +476,7 @@ target_link_libraries(diamond SQLite::SQ
 
 if(WITH_ZSTD)
   find_path(${ZSTD_INCLUDE_DIR} NAMES zstd.h PATHS ${ZSTD_INCLUDE_DIR})
-  find_library(ZSTD_LIBRARY NAMES libzstd.a libzstd_static)
+  find_library(ZSTD_LIBRARY NAMES libzstd.dll.a libzstd.a libzstd_static)
   target_include_directories(diamond PRIVATE "${ZSTD_INCLUDE_DIR}")
   target_link_libraries(diamond ${ZSTD_LIBRARY})
   add_definitions(-DWITH_ZSTD)
