--- origsrc/diamond-2.1.18/CMakeLists.txt	2026-01-07 06:18:17.000000000 +0900
+++ src/diamond-2.1.18/CMakeLists.txt	2026-01-11 12:29:42.312641100 +0900
@@ -476,7 +476,7 @@ target_link_libraries(diamond ${SQLite3_
 
 if(WITH_ZSTD)
   find_path(${ZSTD_INCLUDE_DIR} NAMES zstd.h PATHS ${ZSTD_INCLUDE_DIR})
-  find_library(ZSTD_LIBRARY NAMES libzstd.a libzstd_static)
+  find_library(ZSTD_LIBRARY NAMES libzstd.dll.a libzstd.a libzstd_static)
   target_include_directories(diamond PRIVATE "${ZSTD_INCLUDE_DIR}")
   target_link_libraries(diamond ${ZSTD_LIBRARY})
   add_definitions(-DWITH_ZSTD)
