--- origsrc/isocline-1.0.9/CMakeLists.txt	2022-01-16 10:37:06.000000000 +0900
+++ src/isocline-1.0.9/CMakeLists.txt	2022-08-21 04:08:08.093183900 +0900
@@ -142,7 +142,7 @@ message(STATUS "")
 # Static library (libisocline.a) and samples (example)
 # -----------------------------------------------------------------------------
 
-add_library(isocline STATIC ${ic_sources})
+add_library(isocline ${ic_sources})
 set_property(TARGET isocline PROPERTY POSITION_INDEPENDENT_CODE ON)
 target_compile_options(isocline PRIVATE ${ic_cflags})
 target_compile_definitions(isocline PRIVATE ${ic_cdefs})
@@ -150,6 +150,12 @@ target_include_directories(isocline PUBL
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
     $<INSTALL_INTERFACE:${ic_install_dir}/include>
 )
+if (DEFINED SOVERSION)
+  set_target_properties(isocline PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
+install(FILES include/isocline.h DESTINATION include)
+install(TARGETS isocline RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 
 add_executable(example test/example.c)
 target_compile_options(example PRIVATE ${ic_cflags})
@@ -160,3 +166,10 @@ add_executable(test_colors test/test_col
 target_compile_options(test_colors PRIVATE ${ic_cflags})
 target_include_directories(test_colors PRIVATE include)
 target_link_libraries(test_colors PRIVATE isocline)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/isocline.pc.in ${CMAKE_BINARY_DIR}/isocline.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/isocline.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/isocline-1.0.9/isocline.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/isocline-1.0.9/isocline.pc.in	2022-08-21 04:07:42.018899100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: isocline
+Description: Portable GNU readline alternative
+Version: @VERSION@
+Libs: -L${libdir} -lisocline
+Cflags: -I${includedir}
