--- origsrc/libplist-2.7.0/configure.ac	2025-05-14 17:23:37.000000000 +0900
+++ src/libplist-2.7.0/configure.ac	2025-06-24 15:13:58.168122500 +0900
@@ -62,7 +62,7 @@ AC_C_BIGENDIAN([AC_DEFINE([__BIG_ENDIAN_
 # Check for operating system
 AC_MSG_CHECKING([for platform-specific build settings])
 case ${host_os} in
-  *mingw32*|*cygwin*)
+  *mingw32*)
     AC_MSG_RESULT([${host_os}])
     win32=true
     ;;
--- origsrc/libplist-2.7.0/cython/Makefile.am	2025-05-14 17:23:37.000000000 +0900
+++ src/libplist-2.7.0/cython/Makefile.am	2025-06-24 15:13:58.183750200 +0900
@@ -42,9 +42,7 @@ plist_la_CFLAGS = \
 plist_la_LDFLAGS = -module -avoid-version $(PYTHON_LIBS) $(AM_LDFLAGS) -shared -export-dynamic
 plist_la_LIBADD = $(top_builddir)/src/libplist-2.0.la
 
-if WIN32
 plist_la_LDFLAGS += -no-undefined
-endif
 
 plist.c: plist.pyx $(PXDINCLUDES) $(PXIINCLUDES)
 
