--- origsrc/simdutf-7.4.0/benchmarks/base64/CMakeLists.txt	2025-08-24 09:19:36.000000000 +0900
+++ src/simdutf-7.4.0/benchmarks/base64/CMakeLists.txt	2025-09-15 21:21:57.570174200 +0900
@@ -1,10 +1,14 @@
-include(../../cmake/CPM.cmake)
-CPMAddPackage(
-  NAME base64
-  GITHUB_REPOSITORY aklomp/base64
-  VERSION 0.5.2
-  OPTIONS "BASE64_BUILD_CLI Off"
-)
+find_package(pkgconfig REQUIRED)
+pkg_check_modules(BASE64 base64)
+if (NOT BASE64_FOUND)
+  include(../../cmake/CPM.cmake)
+  CPMAddPackage(
+    NAME base64
+    GITHUB_REPOSITORY aklomp/base64
+    VERSION 0.5.2
+    OPTIONS "BASE64_BUILD_CLI Off"
+  )
+endif()
 
 # Check if the compiler supports C++20
 include(CheckCXXCompilerFlag)
