--- origsrc/simdutf-8.0.0/benchmarks/base64/CMakeLists.txt	2026-01-13 17:03:21.000000000 +0900
+++ src/simdutf-8.0.0/benchmarks/base64/CMakeLists.txt	2026-02-02 09:28:00.012086300 +0900
@@ -1,10 +1,14 @@
-include(../../cmake/CPM.cmake)
-CPMAddPackage(
-  NAME base64
-  GITHUB_REPOSITORY aklomp/base64
-  VERSION 0.5.2
-  OPTIONS "BASE64_BUILD_CLI Off"
-)
+find_package(pkgconfig REQUIRED)
+pkg_check_modules(BASE64 base64)
+if (NOT BASE64_FOUND)
+  include(../../cmake/CPM.cmake)
+  CPMAddPackage(
+    NAME base64
+    GITHUB_REPOSITORY aklomp/base64
+    VERSION 0.5.2
+    OPTIONS "BASE64_BUILD_CLI Off"
+  )
+endif()
 
 # Check if the compiler supports C++20
 include(CheckCXXCompilerFlag)
