diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/verateam-vera-69f28c5a5052/src/lua.cmake src/verateam-vera-69f28c5a5052/src/lua.cmake
--- origsrc/verateam-vera-69f28c5a5052/src/lua.cmake	2015-01-21 16:22:38.000000000 +0900
+++ src/verateam-vera-69f28c5a5052/src/lua.cmake	2018-04-24 09:48:07.430000000 +0900
@@ -4,9 +4,9 @@ mark_as_advanced(VERA_USE_SYSTEM_LUA)
 
 if(VERA_USE_SYSTEM_LUA)
   set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR})
-  find_package(Lua51)
+  find_package(Lua)
   find_package(Luabind)
-  if(NOT LUA51_FOUND)
+  if(NOT LUA_FOUND)
     message(FATAL_ERROR "Could NOT find Lua. Turn VERA_USE_SYSTEM_LUA to OFF to build it with vera++.")
   endif()
   if(NOT LUABIND_FOUND)
