--- origsrc/libfabric-1.13.1/Makefile.am	2021-08-25 09:38:33.000000000 +0900
+++ src/libfabric-1.13.1/Makefile.am	2021-09-04 16:52:12.527867700 +0900
@@ -104,6 +104,11 @@ common_srcs += include/linux/osd.h
 common_srcs += include/unix/osd.h
 endif
 
+if CYGWIN
+common_srcs += src/unix/osd.c
+common_srcs += include/unix/osd.h
+endif
+
 # ensure dl-built providers link back to libfabric
 linkback = src/libfabric.la
 
@@ -181,7 +186,7 @@ src_libfabric_la_SOURCES =			\
 	$(common_srcs)
 
 src_libfabric_la_CPPFLAGS = $(AM_CPPFLAGS)
-src_libfabric_la_LDFLAGS =
+src_libfabric_la_LDFLAGS = -no-undefined
 src_libfabric_la_LIBADD =
 src_libfabric_la_DEPENDENCIES = libfabric.map
 
--- origsrc/libfabric-1.13.1/configure.ac	2021-08-25 09:38:33.000000000 +0900
+++ src/libfabric-1.13.1/configure.ac	2021-09-04 16:52:12.532868000 +0900
@@ -33,6 +33,9 @@ case $host_os in
 *freebsd*)
 	freebsd=1
 	;;
+*cygwin*)
+    cygwin=1
+    ;;
 *)
 	AC_MSG_ERROR([libfabric only builds on Linux, OS X, and FreeBSD])
 	;;
@@ -41,6 +44,7 @@ esac
 AM_CONDITIONAL([MACOS], [test "x$macos" = "x1"])
 AM_CONDITIONAL([LINUX], [test "x$linux" = "x1"])
 AM_CONDITIONAL([FREEBSD], [test "x$freebsd" = "x1"])
+AM_CONDITIONAL([CYGWIN], [test "x$cygwin" = "x1"])
 
 base_c_warn_flags="-Wall -Wundef -Wpointer-arith"
 debug_c_warn_flags="-Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-missing-field-initializers"
--- origsrc/libfabric-1.13.1/include/ofi_net.h	2021-08-25 09:38:33.000000000 +0900
+++ src/libfabric-1.13.1/include/ofi_net.h	2021-09-04 16:52:12.537875400 +0900
@@ -43,6 +43,7 @@
 #include <sys/param.h>
 #include <netinet/in.h>
 #include <ifaddrs.h>
+#include <byteswap.h>
 
 #include <ofi_osd.h>
 #include <ofi_list.h>
--- origsrc/libfabric-1.13.1/include/ofi_osd.h	2021-08-25 09:38:33.000000000 +0900
+++ src/libfabric-1.13.1/include/ofi_osd.h	2021-09-04 16:52:12.543880300 +0900
@@ -64,6 +64,8 @@ typedef long double long_double;
 #include <unix/osd.h>
 #elif defined _WIN32
 #include <windows/osd.h>
+#elif defined(__CYGWIN__)
+#include <unix/osd.h>
 #else
 #include <linux/osd.h>
 #include <unix/osd.h>
--- origsrc/libfabric-1.13.1/include/unix/osd.h	2021-08-25 09:38:33.000000000 +0900
+++ src/libfabric-1.13.1/include/unix/osd.h	2021-09-04 16:52:12.549865700 +0900
@@ -282,4 +282,62 @@ ofi_cpuid(unsigned func, unsigned subfun
 
 #endif /* defined(__x86_64__) || defined(__amd64__) */
 
+#if defined(__CYGWIN__)
+static inline ssize_t ofi_read_socket(SOCKET fd, void *buf, size_t count)
+{
+	return read(fd, buf, count);
+}
+
+static inline ssize_t ofi_write_socket(SOCKET fd, const void *buf, size_t count)
+{
+	return write(fd, buf, count);
+}
+
+static inline ssize_t ofi_recv_socket(SOCKET fd, void *buf, size_t count,
+				      int flags)
+{
+	return recv(fd, buf, count, flags);
+}
+
+static inline ssize_t ofi_recvfrom_socket(SOCKET fd, void *buf, size_t count, int flags,
+					  struct sockaddr *from, socklen_t *fromlen)
+{
+	return recvfrom(fd, buf, count, flags, from, fromlen);
+}
+
+static inline ssize_t ofi_send_socket(SOCKET fd, const void *buf, size_t count,
+				      int flags)
+{
+	return send(fd, buf, count, flags);
+}
+
+static inline ssize_t ofi_sendto_socket(SOCKET fd, const void *buf, size_t count, int flags,
+					const struct sockaddr *to, socklen_t tolen)
+{
+	return sendto(fd, buf, count, flags, to, tolen);
+}
+
+static inline ssize_t ofi_writev_socket(SOCKET fd, struct iovec *iov, size_t iov_cnt)
+{
+	return writev(fd, iov, iov_cnt);
+}
+
+static inline ssize_t ofi_readv_socket(SOCKET fd, struct iovec *iov, int iov_cnt)
+{
+	return readv(fd, iov, iov_cnt);
+}
+
+static inline ssize_t
+ofi_sendmsg_tcp(SOCKET fd, const struct msghdr *msg, int flags)
+{
+	return sendmsg(fd, msg, flags);
+}
+
+static inline ssize_t
+ofi_recvmsg_tcp(SOCKET fd, struct msghdr *msg, int flags)
+{
+	return recvmsg(fd, msg, flags);
+}
+#endif
+
 #endif /* _FI_UNIX_OSD_H_ */
--- origsrc/libfabric-1.13.1/prov/sockets/include/sock_util.h	2021-08-25 09:38:33.000000000 +0900
+++ src/libfabric-1.13.1/prov/sockets/include/sock_util.h	2021-09-04 16:52:12.552864200 +0900
@@ -78,7 +78,7 @@ static inline void *sock_mremap(void *ol
 {
 #ifdef __APPLE__
 	return (void *) -1;
-#elif defined __FreeBSD__
+#elif defined __FreeBSD__ || defined(__CYGWIN__)
 	return (void *) -1;
 #else
 	return mremap(old_address, old_size, new_size, 0);
--- origsrc/libfabric-1.13.1/src/unix/osd.c	2021-08-25 09:38:33.000000000 +0900
+++ src/libfabric-1.13.1/src/unix/osd.c	2021-09-04 16:52:12.556865200 +0900
@@ -58,7 +58,7 @@
 
 #ifdef __FreeBSD__
 typedef cpuset_t ofi_cpu_set_t;
-#elif defined __linux__
+#elif defined __linux__ || defined(__CYGWIN__)
 typedef cpu_set_t ofi_cpu_set_t;
 #endif
 
@@ -338,3 +338,35 @@ void ofi_pollfds_do_del(struct ofi_pollf
 	while (pfds->nfds && pfds->fds[pfds->nfds - 1].fd == INVALID_SOCKET)
 		pfds->nfds--;
 }
+
+#if defined(__CYGWIN__)
+ssize_t ofi_get_hugepage_size(void)
+{
+	return -FI_ENOSYS;
+}
+
+int ofi_alloc_hugepage_buf(void **memptr, size_t size)
+{
+	return -FI_ENOSYS;
+}
+
+int ofi_free_hugepage_buf(void *memptr, size_t size)
+{
+	return -FI_ENOSYS;
+}
+
+size_t ofi_ifaddr_get_speed(struct ifaddrs *ifa)
+{
+	return 0;
+}
+
+size_t ofi_process_vm_writev(pid_t pid,
+			 const struct iovec *local_iov,
+			 unsigned long liovcnt,
+			 const struct iovec *remote_iov,
+			 unsigned long riovcnt,
+			 unsigned long flags)
+{
+	return -FI_ENOSYS;
+}
+#endif
