--- origsrc/crunch-2.5/Makefile.in	2002-04-23 06:26:27.000000000 +0900
+++ src/crunch-2.5/Makefile.in	2025-08-17 10:15:37.285466100 +0900
@@ -11,14 +11,14 @@ INSTALL_PROGRAM=@INSTALL_PROGRAM@
 INSTALL_DATA=	@INSTALL_DATA@
 CC=		@CC@
 CFLAGS=		@CFLAGS@
-CPPFLAGS=	@CPPFLAGS@
+CPPFLAGS=	-I . @CPPFLAGS@
 LDFLAGS=	@LDFLAGS@
 #}}}
 
 all:		crunch
 
-crunch:		crunch.o getopt.o getopt1.o
-		$(CC) -o $@ $(LDFLAGS) crunch.o getopt.o getopt1.o $(LIBS)
+crunch:		crunch.o
+		$(CC) -o $@ $(LDFLAGS) crunch.o $(LIBS)
 
 .c.o:
 		$(CC) -c $(CPPFLAGS) $(CFLAGS) $<
--- origsrc/crunch-2.5/configure.in	2002-05-14 05:22:39.000000000 +0900
+++ src/crunch-2.5/configure.in	2025-08-17 10:15:37.298234800 +0900
@@ -25,7 +25,7 @@ AC_PROG_INSTALL
 AC_CHECK_FUNCS(strerror)
 AC_SYS_LARGEFILE
 
-AC_DEFINE_UNQUOTED(VERSION,"$VERSION")
+AC_DEFINE_UNQUOTED(VERSION,"$VERSION",[version])
 AC_SUBST(VERSION)
 AC_SUBST(UPDATED)
 AC_OUTPUT(Makefile crunch.1 crunch.5)
--- origsrc/crunch-2.5/crunch.c	2002-05-01 22:20:02.000000000 +0900
+++ src/crunch-2.5/crunch.c	2025-08-17 10:15:37.300245100 +0900
@@ -15,7 +15,7 @@
 #include <unistd.h>
 
 #include "config.h"
-#include "getopt.h"
+#include <getopt.h>
 /*}}}*/
 
 static const char *cmd;
