diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/crunch-2.5/Makefile.in src/crunch-2.5/Makefile.in
--- origsrc/crunch-2.5/Makefile.in	2002-04-23 06:26:27.000000000 +0900
+++ src/crunch-2.5/Makefile.in	2015-08-30 23:12:56.018711800 +0900
@@ -11,14 +11,14 @@ INSTALL_PROGRAM=@INSTALL_PROGRAM@
 INSTALL_DATA=	@INSTALL_DATA@
 CC=		@CC@
 CFLAGS=		@CFLAGS@
-CPPFLAGS=	@CPPFLAGS@
+CPPFLAGS=	-I . @CPPFLAGS@
 LDFLAGS=	@LDFLAGS@
 #}}}
 
 all:		crunch
 
-crunch:		crunch.o getopt.o getopt1.o
-		$(CC) -o $@ $(LDFLAGS) crunch.o getopt.o getopt1.o $(LIBS)
+crunch:		crunch.o
+		$(CC) -o $@ $(LDFLAGS) crunch.o $(LIBS)
 
 .c.o:
 		$(CC) -c $(CPPFLAGS) $(CFLAGS) $<
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/crunch-2.5/configure.in src/crunch-2.5/configure.in
--- origsrc/crunch-2.5/configure.in	2002-05-14 05:22:39.000000000 +0900
+++ src/crunch-2.5/configure.in	2015-08-30 23:11:15.544272400 +0900
@@ -25,7 +25,7 @@ AC_PROG_INSTALL
 AC_CHECK_FUNCS(strerror)
 AC_SYS_LARGEFILE
 
-AC_DEFINE_UNQUOTED(VERSION,"$VERSION")
+AC_DEFINE_UNQUOTED(VERSION,"$VERSION",[version])
 AC_SUBST(VERSION)
 AC_SUBST(UPDATED)
 AC_OUTPUT(Makefile crunch.1 crunch.5)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/crunch-2.5/crunch.c src/crunch-2.5/crunch.c
--- origsrc/crunch-2.5/crunch.c	2002-05-01 22:20:02.000000000 +0900
+++ src/crunch-2.5/crunch.c	2015-08-30 23:14:13.893480500 +0900
@@ -15,7 +15,7 @@
 #include <unistd.h>
 
 #include "config.h"
-#include "getopt.h"
+#include <getopt.h>
 /*}}}*/
 
 static const char *cmd;
