--- origsrc/libmbus/Makefile.am	2018-06-30 05:23:33.000000000 +0900
+++ src/libmbus/Makefile.am	2018-06-30 07:51:02.066167600 +0900
@@ -8,7 +8,7 @@ pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libmbus.pc
 
 
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 dist_docdir = $(DESTDIR)$(docdir)
 doc_DATA = README.md \
            COPYING \
--- origsrc/libmbus/mbus/Makefile.am	2018-06-30 05:23:33.000000000 +0900
+++ src/libmbus/mbus/Makefile.am	2018-06-30 07:41:26.811817900 +0900
@@ -17,4 +17,4 @@ include_HEADERS = mbus.h mbus-protocol.h
 
 lib_LTLIBRARIES	   = libmbus.la
 libmbus_la_SOURCES = mbus.c mbus-protocol.c mbus-tcp.c mbus-serial.c mbus-protocol-aux.c
-
+libmbus_la_LDFLAGS = -no-undefined
