--- origsrc/libfcrypto-20240414/Makefile.am	2024-04-14 19:06:28.000000000 +0900
+++ src/libfcrypto-20240414/Makefile.am	2025-05-17 08:32:36.192178400 +0900
@@ -3,10 +3,8 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
 	libfcrypto \
 	pyfcrypto \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libfcrypto-20240414/configure.ac	2024-04-14 19:06:28.000000000 +0900
+++ src/libfcrypto-20240414/configure.ac	2025-05-17 08:32:36.192178400 +0900
@@ -109,7 +109,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
 AC_CONFIG_FILES([libfcrypto/Makefile])
 AC_CONFIG_FILES([pyfcrypto/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
--- origsrc/libfcrypto-20240414/pyfcrypto/Makefile.am	2024-04-14 19:06:28.000000000 +0900
+++ src/libfcrypto-20240414/pyfcrypto/Makefile.am	2025-05-17 08:32:36.192178400 +0900
@@ -26,7 +26,7 @@ pyfcrypto_la_LIBADD = \
 	../libfcrypto/libfcrypto.la
 
 pyfcrypto_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyfcrypto_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyfcrypto_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
