--- origsrc/libgcroots-0.3.2/Makefile.am	2020-08-23 08:03:25.000000000 +0900
+++ src/libgcroots-0.3.2/Makefile.am	2020-09-08 19:04:10.473707500 +0900
@@ -128,7 +128,7 @@ libgcroots_la_SOURCES = mark.c gcroots.c
 
 libgcroots_la_LIBADD = @addobjs@ $(THREADDLLIBS) $(UNWINDLIBS)
 libgcroots_la_DEPENDENCIES = @addobjs@
-libgcroots_la_LDFLAGS = -version-info $(libgcroots_version)
+libgcroots_la_LDFLAGS = -no-undefined -version-info $(libgcroots_version)
 
 EXTRA_libgcroots_la_SOURCES = alpha_mach_dep.S \
     mips_sgi_mach_dep.s mips_ultrix_mach_dep.s \
--- origsrc/libgcroots-0.3.2/include/private/gcconfig.h	2020-08-23 08:03:25.000000000 +0900
+++ src/libgcroots-0.3.2/include/private/gcconfig.h	2020-09-08 19:09:50.301895800 +0900
@@ -414,11 +414,16 @@
 #   endif
 #   define mach_type_known
 # endif
-# if defined(__CYGWIN32__) || defined(__CYGWIN__)
+# if defined(__CYGWIN32__)
 #   define I386
 #   define CYGWIN32
 #   define mach_type_known
 # endif
+# if defined(__CYGWIN__) && !defined(__CYGWIN32__)
+#   define x86_64
+#   define CYGWIN
+#   define mach_type_known
+# endif
 # if defined(__MINGW32__)
 #   define I386
 #   define MSWIN32
