--- origsrc/libgcroots-0.2.2/Makefile.am	2007-07-18 16:53:03.000000000 +0900
+++ src/libgcroots-0.2.2/Makefile.am	2010-09-26 12:45:20.516700500 +0900
@@ -15,6 +15,8 @@
 # Modified by: Petter Urkedal <petter.urkedal@nordita.dk> (2005-04)
 # Modified by: YAMAMOTO Kengo <yamaken AT bp.iij4u.or.jp> (2006-12, libgcroots)
 
+ACLOCAL_AMFLAGS = -I m4
+
 ## Process this file with automake to produce Makefile.in.
 
 include libtool-version.mk
@@ -122,7 +124,7 @@ libgcroots_la_SOURCES = mark.c gcroots.c
 
 libgcroots_la_LIBADD = @addobjs@ $(THREADDLLIBS) $(UNWINDLIBS)
 libgcroots_la_DEPENDENCIES = @addobjs@
-libgcroots_la_LDFLAGS = -version-info $(libgcroots_version)
+libgcroots_la_LDFLAGS = -version-info $(libgcroots_version) -no-undefined
 
 EXTRA_libgcroots_la_SOURCES = alpha_mach_dep.S \
     mips_sgi_mach_dep.s mips_ultrix_mach_dep.s \
--- origsrc/libgcroots-0.2.2/configure.ac	2008-02-20 11:30:04.000000000 +0900
+++ src/libgcroots-0.2.2/configure.ac	2010-09-26 12:48:46.907325500 +0900
@@ -457,7 +457,7 @@ addobjs="$addobjs $machdep"
 AC_SUBST(addobjs)
 AC_SUBST(addlibs)
 
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 #
 # Check for AViiON Machines running DGUX
