--- origsrc/chafa-1.0.1/chafa/Makefile.am	2018-12-19 04:47:50.000000000 +0900
+++ src/chafa-1.0.1/chafa/Makefile.am	2019-03-29 08:56:16.119516100 +0900
@@ -4,7 +4,7 @@ lib_LTLIBRARIES = libchafa.la
 noinst_LTLIBRARIES =
 
 libchafa_la_CFLAGS = $(LIBCHAFA_CFLAGS) $(GLIB_CFLAGS)
-libchafa_la_LDFLAGS = $(LIBCHAFA_LDFLAGS) -version-info 1:1:1
+libchafa_la_LDFLAGS = $(LIBCHAFA_LDFLAGS) -no-undefined -version-info 1:1:1
 libchafa_la_LIBADD = $(GLIB_LIBS)
 
 libchafa_la_SOURCES = \
