diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/luabind/CMakeLists.txt src/luabind/CMakeLists.txt
--- origsrc/luabind/CMakeLists.txt	2018-04-22 11:20:33.000000000 +0900
+++ src/luabind/CMakeLists.txt	2018-04-23 22:59:25.141000000 +0900
@@ -24,7 +24,7 @@ if(NOT Boost_FOUND)
 endif()
 
 if(NOT LUA_FOUND AND NOT LUA51_FOUND)
-	find_package(Lua51 REQUIRED)
+	find_package(Lua REQUIRED)
 	set(LUA_INCLUDE_DIRS "${LUA_INCLUDE_DIR}")
 endif()
 
@@ -52,3 +52,10 @@ if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMA
 	endif()
 	add_subdirectory(doc)
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/luabind.pc.in ${CMAKE_BINARY_DIR}/luabind.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/luabind.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/luabind/luabind.pc.in src/luabind/luabind.pc.in
--- origsrc/luabind/luabind.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/luabind/luabind.pc.in	2018-04-23 22:59:01.203000000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: luabind
+Description: Library that helps you create bindings between C++ and Lua
+Version: @CPACK_PACKAGE_VERSION@
+Libs: -L${libdir} -lluabind
+Cflags: -I${includedir}
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/luabind/src/CMakeLists.txt src/luabind/src/CMakeLists.txt
--- origsrc/luabind/src/CMakeLists.txt	2018-04-22 11:20:33.000000000 +0900
+++ src/luabind/src/CMakeLists.txt	2018-04-22 11:50:50.846999800 +0900
@@ -132,6 +132,10 @@ set_property(TARGET
 	PROJECT_LABEL
 	"LuaBind Library")
 
+if(CYGWIN)
+	set_target_properties(luabind PROPERTIES SOVERSION 0)
+endif()
+
 set(ALLHEADERS ${LUABIND_API} ${LUABIND_DETAIL_API} PARENT_SCOPE)
 set(APIHEADERS ${LUABIND_API} PARENT_SCOPE)
 
