--- origsrc/e57tools-1.0.1/E57Support.cmake	2021-07-29 02:45:32.000000000 +0900
+++ src/e57tools-1.0.1/E57Support.cmake	2025-03-24 23:51:31.952674800 +0900
@@ -25,13 +25,13 @@ if( ${OPTION_USE_LIBE57FORMAT} )
 	set( LIBE57FORMAT_LIB_DIR "${LIBE57FORMAT_INSTALL_DIR}/lib" CACHE INTERNAL "" )
 	
 	find_library( LIBE57FORMAT_LIBRARY_RELEASE
-					NAMES E57Format.lib libE57Format.a E57Format.so libE57Format.so
+					NAMES E57Format.lib libE57Format.a E57Format.so libE57Format.so libE57Format.dll.a
 					PATHS "${LIBE57FORMAT_LIB_DIR}"
 					NO_DEFAULT_PATH
     )
 
 	find_library( LIBE57FORMAT_LIBRARY_DEBUG
-					NAMES E57Format-d.lib libE57Format-d.a E57Format-d.so libE57Format.so
+					NAMES E57Format-d.lib libE57Format-d.a E57Format-d.so libE57Format.so libE57Format.dll.a
 					PATHS "${LIBE57FORMAT_LIB_DIR}"
 					NO_DEFAULT_PATH
 	)
--- origsrc/e57tools-1.0.1/src/e57unpack.cpp	2021-07-29 02:45:32.000000000 +0900
+++ src/e57tools-1.0.1/src/e57unpack.cpp	2025-03-25 00:07:21.808946700 +0900
@@ -10,6 +10,7 @@
 #define BUILD_VERSION STRING_HELPER1(SVN_VERSION)
 
 #include <E57Format.h>
+#include <E57Version.h>
 using e57::Node;
 using e57::ImageFile;
 using e57::StructureNode;
