diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/CMake/CMakeLists.txt src/glotzer-hoomd-blue-a3a2cfbe55b5/CMake/CMakeLists.txt
--- origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/CMake/CMakeLists.txt	2016-02-09 04:28:32.000000000 +0900
+++ src/glotzer-hoomd-blue-a3a2cfbe55b5/CMake/CMakeLists.txt	2016-03-05 10:46:06.297849600 +0900
@@ -22,6 +22,6 @@ install(DIRECTORY ${CMAKE_CURRENT_SOURCE
                   ${CMAKE_CURRENT_SOURCE_DIR}/thrust
                   ${CMAKE_CURRENT_SOURCE_DIR}/hoomd
                   ${CMAKE_CURRENT_SOURCE_DIR}/git
-        DESTINATION ${DATA_INSTALL_DIR}/CMake
+        DESTINATION lib/cmake/hoomd
         PATTERN ".svn" EXCLUDE
         )
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/CMakeLists.txt src/glotzer-hoomd-blue-a3a2cfbe55b5/CMakeLists.txt
--- origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/CMakeLists.txt	2016-02-09 04:28:32.000000000 +0900
+++ src/glotzer-hoomd-blue-a3a2cfbe55b5/CMakeLists.txt	2016-03-05 10:46:06.297849600 +0900
@@ -78,3 +78,10 @@ if (BUILD_TESTING)
     add_dependencies(check test_all)
     add_subdirectory (test)
 endif (BUILD_TESTING)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/hoomd-blue.pc.in ${CMAKE_BINARY_DIR}/hoomd-blue.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/hoomd-blue.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/hoomd-blue.pc.in src/glotzer-hoomd-blue-a3a2cfbe55b5/hoomd-blue.pc.in
--- origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/hoomd-blue.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/glotzer-hoomd-blue-a3a2cfbe55b5/hoomd-blue.pc.in	2016-03-05 10:46:06.297849600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: hoomd-blue
+Description: General-purpose particle simulation toolkit
+Version: @HOOMD_VERSION@
+Libs: -L${libdir} -lhoomd
+Cflags: -I${includedir}/hoomd
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/CMakeLists.txt src/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/CMakeLists.txt
--- origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/CMakeLists.txt	2016-02-09 04:28:32.000000000 +0900
+++ src/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/CMakeLists.txt	2016-03-05 10:46:06.297849600 +0900
@@ -63,7 +63,11 @@ if (NOT ENABLE_STATIC)
     endif(ENABLE_MPI)
 
     # tweak the properties of the output to make a functional python module
+    if (CYGWIN)
+        set_target_properties(libhoomd PROPERTIES OUTPUT_NAME "hoomd" SOVERSION 0)
+    else()
     set_target_properties(libhoomd PROPERTIES PREFIX "" OUTPUT_NAME "hoomd")
+    endif()
 
     # .dylib is not regognized as a python module by python on Mac OS X
     if(APPLE)
@@ -90,7 +94,7 @@ set(_hoomd_so_install_dir ${HOOMD_PYTHON
 
 # install the library
 install(TARGETS libhoomd
-        LIBRARY DESTINATION ${_hoomd_so_install_dir}
+        RUNTIME DESTINATION ${BIN_INSTALL_DIR}
         ARCHIVE DESTINATION ${LIB_BASE_INSTALL_DIR}
         )
 
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/computes/ConstraintSphere.cc src/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/computes/ConstraintSphere.cc
--- origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/computes/ConstraintSphere.cc	2016-02-09 04:28:32.000000000 +0900
+++ src/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/computes/ConstraintSphere.cc	2016-03-05 10:46:06.297849600 +0900
@@ -49,7 +49,6 @@ ADVISED OF THE POSSIBILITY OF SUCH DAMAG
 
 // Maintainer: joaander
 
-
 #include "ConstraintSphere.h"
 #include "EvaluatorConstraint.h"
 #include "EvaluatorConstraintSphere.h"
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/computes/EvaluatorConstraint.h src/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/computes/EvaluatorConstraint.h
--- origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/computes/EvaluatorConstraint.h	2016-02-09 04:28:32.000000000 +0900
+++ src/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/computes/EvaluatorConstraint.h	2016-03-05 10:46:06.313318300 +0900
@@ -66,6 +66,13 @@ ADVISED OF THE POSSIBILITY OF SUCH DAMAG
 #define DEVICE
 #endif
 
+#if defined(_X)
+#undef _X
+#endif
+#if defined(_P)
+#undef _P
+#endif
+
 //! Class for evaluating constraint forces
 /*! <b>General Overview</b>
     EvaluatorConstraint is a low level computation class for use on the CPU and GPU. It provides basic functionality
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/extern/cudacpu_host_defines.h src/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/extern/cudacpu_host_defines.h
--- origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/extern/cudacpu_host_defines.h	2016-02-09 04:28:32.000000000 +0900
+++ src/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/extern/cudacpu_host_defines.h	2016-03-05 10:46:06.313318300 +0900
@@ -76,7 +76,9 @@
         __thread
 #define __import__
 #define __export__
+#if !defined(__cdecl)
 #define __cdecl
+#endif
 #define __annotate__(a) \
         __attribute__((a))
 #define __location__(a) \
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/utils/PathUtils.cc src/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/utils/PathUtils.cc
--- origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/utils/PathUtils.cc	2016-02-09 04:28:32.000000000 +0900
+++ src/glotzer-hoomd-blue-a3a2cfbe55b5/libhoomd/utils/PathUtils.cc	2016-03-05 10:46:06.313318300 +0900
@@ -101,7 +101,7 @@ std::string getExePath()
     result = std::string(realbuf);
     free(realbuf);
 
-    #elif __linux__
+    #elif defined(__linux__) || defined(__CYGWIN__)
 
     char buf[path_max];
     memset(buf, 0, path_max);
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/test/unit/CMakeLists.txt src/glotzer-hoomd-blue-a3a2cfbe55b5/test/unit/CMakeLists.txt
--- origsrc/glotzer-hoomd-blue-a3a2cfbe55b5/test/unit/CMakeLists.txt	2016-02-09 04:28:32.000000000 +0900
+++ src/glotzer-hoomd-blue-a3a2cfbe55b5/test/unit/CMakeLists.txt	2016-03-05 10:46:06.313318300 +0900
@@ -21,6 +21,10 @@ if("${Boost_UNIT_TEST_FRAMEWORK_LIBRARY}
 add_definitions(-DBOOST_TEST_DYN_LINK)
 endif("${Boost_UNIT_TEST_FRAMEWORK_LIBRARY}" MATCHES "\\.so$")
 
+if("${Boost_UNIT_TEST_FRAMEWORK_LIBRARY}" MATCHES "\\.dll.a$")
+add_definitions(-DBOOST_TEST_DYN_LINK)
+endif("${Boost_UNIT_TEST_FRAMEWORK_LIBRARY}" MATCHES "\\.dll.a$")
+
 ####################################
 ## Setup valgrind options
 if (UNIX AND NOT APPLE)
