--- origsrc/re2-2021-09-01/re2/testing/re2_test.cc	2021-08-31 18:13:56.000000000 +0900
+++ src/re2-2021-09-01/re2/testing/re2_test.cc	2021-09-12 23:13:37.157127800 +0900
@@ -13,7 +13,7 @@
 #include <string>
 #include <utility>
 #include <vector>
-#if !defined(_MSC_VER) && !defined(__CYGWIN__) && !defined(__MINGW32__)
+#if !defined(_MSC_VER) && !defined(__MINGW32__)
 #include <sys/mman.h>
 #include <unistd.h>  /* for sysconf */
 #endif
