--- origsrc/re2-2023-03-01/re2/testing/re2_test.cc	2023-02-21 03:50:47.000000000 +0900
+++ src/re2-2023-03-01/re2/testing/re2_test.cc	2023-03-12 20:54:44.464458500 +0900
@@ -13,7 +13,7 @@
 #include <string>
 #include <utility>
 #include <vector>
-#if !defined(_MSC_VER) && !defined(__CYGWIN__) && !defined(__MINGW32__)
+#if !defined(_MSC_VER) && !defined(__MINGW32__)
 #include <sys/mman.h>
 #include <unistd.h>  /* for sysconf */
 #endif
