--- origsrc/re2-2019-09-01/CMakeLists.txt	2019-08-27 13:53:41.000000000 +0900
+++ src/re2-2019-09-01/CMakeLists.txt	2019-09-06 22:52:45.376635400 +0900
@@ -156,3 +156,14 @@ install(TARGETS re2 EXPORT re2Config
         INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 install(EXPORT re2Config
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/re2 NAMESPACE re2::)
+
+if (CYGWIN)
+  set_target_properties(re2 PROPERTIES SOVERSION 0)
+endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/re2.pc ${CMAKE_BINARY_DIR}/re2.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/re2.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/re2-2019-09-01/re2/testing/re2_test.cc	2019-08-27 13:53:41.000000000 +0900
+++ src/re2-2019-09-01/re2/testing/re2_test.cc	2019-09-06 22:21:00.937026800 +0900
@@ -12,7 +12,7 @@
 #include <map>
 #include <string>
 #include <utility>
-#if !defined(_MSC_VER) && !defined(__CYGWIN__) && !defined(__MINGW32__)
+#if !defined(_MSC_VER) && !defined(__MINGW32__)
 #include <sys/mman.h>
 #include <unistd.h>  /* for sysconf */
 #endif
