--- origsrc/re2/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/re2/Makefile.am	2011-11-15 00:21:06.080256300 +0900
@@ -0,0 +1,127 @@
+
+AM_CPPFLAGS = -I$(top_srcdir)
+AM_CXXFLAGS = -Wall
+
+lib_LTLIBRARIES = libre2.la
+libre2_la_SOURCES = \
+        util/arena.cc \
+        util/hash.cc \
+        util/rune.cc \
+        util/stringpiece.cc \
+        util/stringprintf.cc \
+        util/strutil.cc \
+        util/valgrind.cc \
+        re2/bitstate.cc \
+        re2/compile.cc \
+        re2/dfa.cc \
+        re2/filtered_re2.cc \
+        re2/mimics_pcre.cc \
+        re2/nfa.cc \
+        re2/onepass.cc \
+        re2/parse.cc \
+        re2/perl_groups.cc \
+        re2/prefilter.cc \
+        re2/prefilter_tree.cc \
+        re2/prog.cc \
+        re2/re2.cc \
+        re2/regexp.cc \
+        re2/set.cc \
+        re2/simplify.cc \
+        re2/tostring.cc \
+        re2/unicode_casefold.cc \
+        re2/unicode_groups.cc
+
+libre2_la_LDFLAGS = -no-undefined --version-script=$(top_srcdir)/libre2.symbols
+libre2_la_LIBADD = $(PCRE_LIBS)
+
+re2_includedir = $(includedir)/re2
+re2_include_HEADERS = \
+        re2/filtered_re2.h \
+        re2/re2.h \
+        re2/set.h \
+        re2/stringpiece.h \
+        re2/variadic_function.h
+
+noinst_LTLIBRARIES = libtest.la
+libtest_la_SOURCES = \
+        util/test.cc \
+        util/pcre.cc \
+        util/random.cc \
+        util/rune.cc \
+        util/stringpiece.cc \
+        util/stringprintf.cc \
+        util/thread.cc \
+        re2/testing/backtrack.cc \
+        re2/testing/dump.cc \
+        re2/testing/exhaustive_tester.cc \
+        re2/testing/null_walker.cc \
+        re2/testing/regexp_generator.cc \
+        re2/testing/string_generator.cc \
+        re2/testing/tester.cc
+
+TESTS = \
+        re2/testing/charclass_test \
+        re2/testing/compile_test \
+        re2/testing/filtered_re2_test \
+        re2/testing/mimics_pcre_test \
+        re2/testing/parse_test \
+        re2/testing/possible_match_test \
+        re2/testing/re2_test \
+        re2/testing/re2_arg_test \
+        re2/testing/regexp_test \
+        re2/testing/required_prefix_test \
+        re2/testing/search_test \
+        re2/testing/set_test \
+        re2/testing/simplify_test \
+        re2/testing/string_generator_test \
+        re2/testing/dfa_test \
+        re2/testing/exhaustive1_test \
+        re2/testing/exhaustive2_test \
+        re2/testing/exhaustive3_test \
+        re2/testing/exhaustive_test \
+        re2/testing/random_test
+
+re2_testing_charclass_test_SOURCES =               re2/testing/charclass_test.cc
+re2_testing_compile_test_SOURCES =                 re2/testing/compile_test.cc
+re2_testing_filtered_re2_test_SOURCES =            re2/testing/filtered_re2_test.cc
+re2_testing_mimics_pcre_test_SOURCES =             re2/testing/mimics_pcre_test.cc
+re2_testing_parse_test_SOURCES =                   re2/testing/parse_test.cc
+re2_testing_possible_match_test_SOURCES =          re2/testing/possible_match_test.cc
+re2_testing_re2_test_SOURCES =                     re2/testing/re2_test.cc
+re2_testing_re2_arg_test_SOURCES =                 re2/testing/re2_arg_test.cc
+re2_testing_regexp_test_SOURCES =                  re2/testing/regexp_test.cc
+re2_testing_required_prefix_test_SOURCES =         re2/testing/required_prefix_test.cc
+re2_testing_search_test_SOURCES =                  re2/testing/search_test.cc
+re2_testing_set_test_SOURCES =                     re2/testing/set_test.cc
+re2_testing_simplify_test_SOURCES =                re2/testing/simplify_test.cc
+re2_testing_string_generator_test_SOURCES =        re2/testing/string_generator_test.cc
+re2_testing_dfa_test_SOURCES =                     re2/testing/dfa_test.cc
+re2_testing_exhaustive1_test_SOURCES =             re2/testing/exhaustive1_test.cc
+re2_testing_exhaustive2_test_SOURCES =             re2/testing/exhaustive2_test.cc
+re2_testing_exhaustive3_test_SOURCES =             re2/testing/exhaustive3_test.cc
+re2_testing_exhaustive_test_SOURCES =              re2/testing/exhaustive_test.cc
+re2_testing_random_test_SOURCES =                  re2/testing/random_test.cc
+
+re2_testing_charclass_test_LDADD = libtest.la libre2.la
+re2_testing_compile_test_LDADD = libtest.la libre2.la
+re2_testing_filtered_re2_test_LDADD = libtest.la libre2.la
+re2_testing_mimics_pcre_test_LDADD = libtest.la libre2.la
+re2_testing_parse_test_LDADD = libtest.la libre2.la
+re2_testing_possible_match_test_LDADD = libtest.la libre2.la
+re2_testing_re2_test_LDADD = libtest.la libre2.la
+re2_testing_re2_arg_test_LDADD = libtest.la libre2.la
+re2_testing_regexp_test_LDADD = libtest.la libre2.la
+re2_testing_required_prefix_test_LDADD = libtest.la libre2.la
+re2_testing_search_test_LDADD = libtest.la libre2.la
+re2_testing_set_test_LDADD = libtest.la libre2.la
+re2_testing_simplify_test_LDADD = libtest.la libre2.la
+re2_testing_string_generator_test_LDADD = libtest.la libre2.la
+re2_testing_dfa_test_LDADD = libtest.la libre2.la
+re2_testing_exhaustive1_test_LDADD = libtest.la libre2.la
+re2_testing_exhaustive2_test_LDADD = libtest.la libre2.la
+re2_testing_exhaustive3_test_LDADD = libtest.la libre2.la
+re2_testing_exhaustive_test_LDADD = libtest.la libre2.la
+re2_testing_random_test_LDADD = libtest.la libre2.la
+
+check_PROGRAMS = $(TESTS)
+
--- origsrc/re2/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/re2/configure.ac	2011-11-15 00:36:31.581061100 +0900
@@ -0,0 +1,47 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([re2], [0+hg20111030], [re2-dev@googlegroups.com])
+AC_CONFIG_SRCDIR([testinstall.cc])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_LN_S
+AC_PROG_MAKE_SET
+LT_INIT([disable-static])
+
+# Checks for libraries.
+PKG_CHECK_MODULES([PCRE], [libpcre])
+
+# Checks for header files.
+AC_CHECK_HEADERS([stddef.h stdint.h stdlib.h string.h sys/time.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_HEADER_STDBOOL
+AC_C_INLINE
+AC_TYPE_INT16_T
+AC_TYPE_INT32_T
+AC_TYPE_INT64_T
+AC_TYPE_INT8_T
+AC_C_RESTRICT
+AC_TYPE_SIZE_T
+AC_TYPE_UINT16_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+AC_TYPE_UINT8_T
+AC_CHECK_TYPES([ptrdiff_t])
+
+# Checks for library functions.
+AC_FUNC_ERROR_AT_LINE
+AC_FUNC_MALLOC
+AC_FUNC_MMAP
+AC_FUNC_STRTOD
+AC_CHECK_FUNCS([gettimeofday memchr memmove memset munmap setlocale strchr strstr strtol strtoul strtoull])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
