--- origsrc/re2-2019-07-01/CMakeLists.txt	2019-06-14 14:35:05.000000000 +0900
+++ src/re2-2019-07-01/CMakeLists.txt	2019-07-02 21:56:26.770663000 +0900
@@ -150,3 +150,7 @@ set(RE2_HEADERS
 install(FILES ${RE2_HEADERS} DESTINATION include/re2)
 install(TARGETS re2 EXPORT re2Config ARCHIVE DESTINATION lib LIBRARY DESTINATION lib RUNTIME DESTINATION bin INCLUDES DESTINATION include)
 install(EXPORT re2Config DESTINATION lib/cmake/re2 NAMESPACE re2::)
+
+if (CYGWIN)
+  set_target_properties(re2 PROPERTIES SOVERSION 0)
+endif()
--- origsrc/re2-2019-07-01/re2/testing/re2_test.cc	2019-06-14 14:35:05.000000000 +0900
+++ src/re2-2019-07-01/re2/testing/re2_test.cc	2019-07-02 21:56:26.786289200 +0900
@@ -12,7 +12,7 @@
 #include <map>
 #include <string>
 #include <utility>
-#if !defined(_MSC_VER) && !defined(__CYGWIN__) && !defined(__MINGW32__)
+#if !defined(_MSC_VER) && !defined(__MINGW32__)
 #include <sys/mman.h>
 #include <unistd.h>  /* for sysconf */
 #endif
