--- origsrc/biobambam2-biobambam2_experimental_2_0_185/configure.ac	2022-12-11 21:21:04.000000000 +0900
+++ src/biobambam2-biobambam2_experimental_2_0_185/configure.ac	2025-01-25 14:27:01.353336900 +0900
@@ -340,7 +340,7 @@ if test "${with_xerces_c}" != "no" ; the
 
 	if test $HAVE_XERCES_C -eq 1 ; then
 		BIOBAMBAM_HAVE_XERCES_C="#define BIOBAMBAM_HAVE_XERCES_C"
-		BLASTNXMLTOBAM="blastnxmltobam"
+		BLASTNXMLTOBAM="blastnxmltobam${EXEEXT}"
 	fi
 fi
 
@@ -412,7 +412,7 @@ AC_ARG_ENABLE(install_uncommon,
         AS_HELP_STRING([--enable-install-uncommon],[enable installation of some uncommon programs (default no)]),
         [install_uncommon=${enableval}],[install_uncommon=no])
 
-UNCOMMON="bamfilter bamfilterbyname bamfixmatecoordinates bamfixmatecoordinatesnamesorted bamtoname bamdisthist fastabgzfextract bamheap bamfrontback bamrandomtag bamheap2 bamheap3 bamtagconversion fastqtobampar bambisect vcffilterinfo vcfpatchcontigprepend vcfconcat vcfsort filtergtf bamconsensus vcfreplacecontigs vcffiltersamples bamexploderg bamexondepth bamheadercat bammarkduplicatesoptdist vcfdiff bamsimpledepth bamdepthmerge bamcountflags vcffilterfilterflags vcfreplacecontigsmap fastaselectreg bamdifference bamaddne bamadapterhistogram bamsalvage"
+UNCOMMON="bamfilter${EXEEXT} bamfilterbyname${EXEEXT} bamfixmatecoordinates${EXEEXT} bamfixmatecoordinatesnamesorted${EXEEXT} bamtoname${EXEEXT} bamdisthist${EXEEXT} fastabgzfextract${EXEEXT} bamheap${EXEEXT} bamfrontback${EXEEXT} bamrandomtag${EXEEXT} bamheap2${EXEEXT} bamheap3${EXEEXT} bamtagconversion${EXEEXT} fastqtobampar${EXEEXT} bambisect${EXEEXT} vcffilterinfo${EXEEXT} vcfpatchcontigprepend${EXEEXT} vcfconcat${EXEEXT} vcfsort${EXEEXT} filtergtf${EXEEXT} bamconsensus${EXEEXT} vcfreplacecontigs${EXEEXT} vcffiltersamples${EXEEXT} bamexploderg${EXEEXT} bamexondepth${EXEEXT} bamheadercat${EXEEXT} bammarkduplicatesoptdist${EXEEXT} vcfdiff${EXEEXT} bamsimpledepth${EXEEXT} bamdepthmerge${EXEEXT} bamcountflags${EXEEXT} vcffilterfilterflags${EXEEXT} vcfreplacecontigsmap${EXEEXT} fastaselectreg${EXEEXT} bamdifference${EXEEXT} bamaddne${EXEEXT} bamadapterhistogram${EXEEXT} bamsalvage${EXEEXT}"
 UNCOMMONINSTALLED=
 UNCOMMONUNINSTALLED=
 if test "${install_uncommon}" = "yes" ; then
