--- origsrc/libformfactor-v0.3.1/CMakeLists.txt	2023-08-03 01:06:12.000000000 +0900
+++ src/libformfactor-v0.3.1/CMakeLists.txt	2023-09-09 15:18:28.345894400 +0900
@@ -91,13 +91,13 @@ write_basic_package_version_file(
 install(FILES
     "${PROJECT_BINARY_DIR}/formfactorConfig.cmake"
     "${PROJECT_BINARY_DIR}/formfactorConfigVersion.cmake"
-    DESTINATION cmake)
+    DESTINATION lib/cmake/libformfactor)
 
 ## Export targets.
 
 install(EXPORT formfactorTargets
   FILE formfactorTargets.cmake
-  DESTINATION cmake)
+  DESTINATION lib/cmake/libformfactor)
 
 export(EXPORT formfactorTargets
   FILE "${CMAKE_CURRENT_BINARY_DIR}/formfactorTargets.cmake")
--- origsrc/libformfactor-v0.3.1/ff/CMakeLists.txt	2023-08-03 01:06:12.000000000 +0900
+++ src/libformfactor-v0.3.1/ff/CMakeLists.txt	2023-09-09 14:46:14.592399900 +0900
@@ -25,12 +25,15 @@ set_target_properties(
     ${lib} PROPERTIES
     OUTPUT_NAME ${lib}
     VERSION ${PROJECT_VERSION})
+if (DEFINED SOVERSION)
+    set_target_properties(${lib} PROPERTIES SOVERSION "${SOVERSION}")
+endif()
 
 install(
     TARGETS ${lib}
     EXPORT formfactorTargets
     LIBRARY DESTINATION lib
-    RUNTIME DESTINATION lib
+    RUNTIME DESTINATION bin
     ARCHIVE DESTINATION lib)
 
 install(
