--- origsrc/mongo-c-driver-1.20.1/src/libmongoc/src/mongoc/mongoc-client.c	2022-01-06 03:47:00.000000000 +0900
+++ src/mongo-c-driver-1.20.1/src/libmongoc/src/mongoc/mongoc-client.c	2022-01-16 23:28:37.582652500 +0900
@@ -432,6 +432,10 @@ _mongoc_get_rr_search (const char *servi
                        size_t initial_buffer_size,
                        bson_error_t *error)
 {
+#if defined(__CYGWIN__)
+   return false;
+#else
+
 #ifdef MONGOC_HAVE_RES_NSEARCH
    struct __res_state state = {0};
 #endif
@@ -566,6 +570,9 @@ done:
    res_nclose (&state);
 #endif
    RETURN (dns_success && callback_success);
+
+#endif
+
 }
 #endif
 
