--- origsrc/mongo-c-driver-1.17.3/src/libmongoc/src/mongoc/mongoc-client.c	2020-12-02 07:25:25.000000000 +0900
+++ src/mongo-c-driver-1.17.3/src/libmongoc/src/mongoc/mongoc-client.c	2020-12-05 16:25:15.326142800 +0900
@@ -414,6 +414,10 @@ _mongoc_get_rr_search (const char *servi
                        size_t initial_buffer_size,
                        bson_error_t *error)
 {
+#if defined(__CYGWIN__)
+   return false;
+#else
+
 #ifdef MONGOC_HAVE_RES_NSEARCH
    struct __res_state state = {0};
 #endif
@@ -551,6 +555,9 @@ done:
    res_nclose (&state);
 #endif
    RETURN (dns_success && callback_success);
+
+#endif
+
 }
 #endif
 
