--- origsrc/mosquitto-2.0.22/src/mosquitto_broker_internal.h	2025-07-12 06:06:10.000000000 +0900
+++ src/mosquitto-2.0.22/src/mosquitto_broker_internal.h	2026-01-26 10:05:49.657141800 +0900
@@ -842,7 +842,7 @@ void handle_sighup(int signal);
 /* ============================================================
  * Window service and signal related functions
  * ============================================================ */
-#if defined(WIN32) || defined(__CYGWIN__)
+#if defined(WIN32)
 void service_install(void);
 void service_uninstall(void);
 void service_run(void);
