--- origsrc/mosquitto-1.1/lib/CMakeLists.txt	2012-12-19 20:42:34.000000000 +0900
+++ src/mosquitto-1.1/lib/CMakeLists.txt	2012-12-23 13:43:25.472843800 +0900
@@ -49,7 +49,7 @@ set_target_properties(libmosquitto PROPE
 	SOVERSION 1
 )
 
-install(TARGETS libmosquitto RUNTIME DESTINATION ${BINDIR} LIBRARY DESTINATION ${LIBDIR})
+install(TARGETS libmosquitto RUNTIME DESTINATION ${BINDIR} ARCHIVE DESTINATION ${LIBDIR})
 install(FILES mosquitto.h DESTINATION ${INCLUDEDIR})
 
 if (UNIX)
--- origsrc/mosquitto-1.1/lib/cpp/CMakeLists.txt	2012-12-19 20:42:34.000000000 +0900
+++ src/mosquitto-1.1/lib/cpp/CMakeLists.txt	2012-12-23 13:43:25.488468800 +0900
@@ -10,7 +10,7 @@ set_target_properties(mosquittopp PROPER
 	VERSION ${VERSION}
 	SOVERSION 1
 )
-install(TARGETS mosquittopp RUNTIME DESTINATION ${BINDIR} LIBRARY DESTINATION ${LIBDIR})
+install(TARGETS mosquittopp RUNTIME DESTINATION ${BINDIR} ARCHIVE DESTINATION ${LIBDIR})
 install(FILES mosquittopp.h DESTINATION ${INCLUDEDIR})
 
 if (UNIX)
