--- origsrc/mosquitto-1.4.10/CMakeLists.txt	2016-08-17 20:05:14.000000000 +0900
+++ src/mosquitto-1.4.10/CMakeLists.txt	2016-12-02 12:05:44.245613300 +0900
@@ -42,7 +42,7 @@ else (WIN32)
 	else ("${CMAKE_INSTALL_PREFIX}" STREQUAL /usr)
 		set (SYSCONFDIR etc/mosquitto)
 	endif ("${CMAKE_INSTALL_PREFIX}" STREQUAL /usr)
-	
+
 	set (LIBDIR lib${LIB_SUFFIX})
 	set (CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${LIBDIR}")
 	set (INCLUDEDIR include)
@@ -93,3 +93,12 @@ add_subdirectory(man)
 # ========================================
 
 install(FILES mosquitto.conf aclfile.example pskfile.example pwfile.example DESTINATION "${SYSCONFDIR}")
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/mosquitto.pc.in ${CMAKE_BINARY_DIR}/mosquitto.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/mosquitto.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/mosquittopp.pc.in ${CMAKE_BINARY_DIR}/mosquittopp.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/mosquittopp.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/mosquitto-1.4.10/lib/CMakeLists.txt	2016-08-17 20:05:14.000000000 +0900
+++ src/mosquitto-1.4.10/lib/CMakeLists.txt	2016-12-02 12:08:17.036196700 +0900
@@ -70,7 +70,7 @@ set_target_properties(libmosquitto PROPE
 	SOVERSION 1
 )
 
-install(TARGETS libmosquitto RUNTIME DESTINATION "${BINDIR}" LIBRARY DESTINATION "${LIBDIR}")
+install(TARGETS libmosquitto RUNTIME DESTINATION "${BINDIR}" ARCHIVE DESTINATION "${LIBDIR}")
 install(FILES mosquitto.h DESTINATION "${INCLUDEDIR}")
 
 if (UNIX)
--- origsrc/mosquitto-1.4.10/lib/cpp/CMakeLists.txt	2016-08-17 20:05:14.000000000 +0900
+++ src/mosquitto-1.4.10/lib/cpp/CMakeLists.txt	2016-12-02 12:08:24.915610400 +0900
@@ -10,7 +10,7 @@ set_target_properties(mosquittopp PROPER
 	VERSION ${VERSION}
 	SOVERSION 1
 )
-install(TARGETS mosquittopp RUNTIME DESTINATION "${BINDIR}" LIBRARY DESTINATION "${LIBDIR}")
+install(TARGETS mosquittopp RUNTIME DESTINATION "${BINDIR}" ARCHIVE DESTINATION "${LIBDIR}")
 install(FILES mosquittopp.h DESTINATION "${INCLUDEDIR}")
 
 if (UNIX)
--- origsrc/mosquitto-1.4.10/mosquitto.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/mosquitto-1.4.10/mosquitto.pc.in	2016-12-02 12:04:21.467231600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mosquitto
+Description: Open Source MQTT v3.1 Broker
+Version: @VERSION@
+Libs: -L${libdir} -lmosquitto
+Cflags: -I${includedir}
--- origsrc/mosquitto-1.4.10/mosquittopp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/mosquitto-1.4.10/mosquittopp.pc.in	2016-12-02 12:06:35.457764700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mosquittopp
+Description: Open Source MQTT v3.1 Broker (c++)
+Version: @VERSION@
+Libs: -L${libdir} -lmosquitto -lmosquittopp
+Cflags: -I${includedir}
