diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/mosquitto-1.4.4/lib/CMakeLists.txt src/mosquitto-1.4.4/lib/CMakeLists.txt
--- origsrc/mosquitto-1.4.4/lib/CMakeLists.txt	2015-09-17 06:54:44.000000000 +0900
+++ src/mosquitto-1.4.4/lib/CMakeLists.txt	2015-10-18 23:15:47.145661800 +0900
@@ -70,9 +70,19 @@ set_target_properties(libmosquitto PROPE
 	SOVERSION 1
 )
 
-install(TARGETS libmosquitto RUNTIME DESTINATION ${BINDIR} LIBRARY DESTINATION ${LIBDIR})
+install(TARGETS libmosquitto RUNTIME DESTINATION ${BINDIR} ARCHIVE DESTINATION ${LIBDIR})
 install(FILES mosquitto.h DESTINATION ${INCLUDEDIR})
 
 if (UNIX)
 	install(CODE "EXEC_PROGRAM(/sbin/ldconfig)")
 endif (UNIX)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "1.4.4")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/mosquitto.pc.in ${CMAKE_BINARY_DIR}/mosquitto.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/mosquitto.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/mosquittopp.pc.in ${CMAKE_BINARY_DIR}/mosquittopp.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/mosquittopp.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/mosquitto-1.4.4/lib/cpp/CMakeLists.txt src/mosquitto-1.4.4/lib/cpp/CMakeLists.txt
--- origsrc/mosquitto-1.4.4/lib/cpp/CMakeLists.txt	2015-09-17 06:54:44.000000000 +0900
+++ src/mosquitto-1.4.4/lib/cpp/CMakeLists.txt	2015-10-18 23:15:47.145661800 +0900
@@ -10,7 +10,7 @@ set_target_properties(mosquittopp PROPER
 	VERSION ${VERSION}
 	SOVERSION 1
 )
-install(TARGETS mosquittopp RUNTIME DESTINATION ${BINDIR} LIBRARY DESTINATION ${LIBDIR})
+install(TARGETS mosquittopp RUNTIME DESTINATION ${BINDIR} ARCHIVE DESTINATION ${LIBDIR})
 install(FILES mosquittopp.h DESTINATION ${INCLUDEDIR})
 
 if (UNIX)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/mosquitto-1.4.4/mosquitto.pc.in src/mosquitto-1.4.4/mosquitto.pc.in
--- origsrc/mosquitto-1.4.4/mosquitto.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/mosquitto-1.4.4/mosquitto.pc.in	2015-10-18 23:15:47.145661800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mosquitto
+Description: Open Source MQTT v3.1 Broker
+Version: @VERSION@
+Libs: -L${libdir} -lmosquitto
+Cflags: -I${includedir}
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/mosquitto-1.4.4/mosquittopp.pc.in src/mosquitto-1.4.4/mosquittopp.pc.in
--- origsrc/mosquitto-1.4.4/mosquittopp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/mosquitto-1.4.4/mosquittopp.pc.in	2015-10-18 23:15:47.145661800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mosquitto
+Description: Open Source MQTT v3.1 Broker (c++)
+Version: @VERSION@
+Libs: -L${libdir} -lmosquittopp
+Cflags: -I${includedir}
