--- origsrc/mosquitto-2.0.16/src/mosquitto_broker_internal.h	2023-08-16 22:03:29.000000000 +0900
+++ src/mosquitto-2.0.16/src/mosquitto_broker_internal.h	2023-08-21 13:19:41.638839300 +0900
@@ -838,7 +838,7 @@ void handle_sighup(int signal);
 /* ============================================================
  * Window service and signal related functions
  * ============================================================ */
-#if defined(WIN32) || defined(__CYGWIN__)
+#if defined(WIN32)
 void service_install(void);
 void service_uninstall(void);
 void service_run(void);
