--- origsrc/mosquitto-1.6.8/CMakeLists.txt	2019-11-29 02:15:13.000000000 +0900
+++ src/mosquitto-1.6.8/CMakeLists.txt	2019-12-23 17:17:05.659125500 +0900
@@ -103,7 +103,7 @@ endif (DOCUMENTATION)
 # Install config file
 # ========================================
 
-install(FILES mosquitto.conf aclfile.example pskfile.example pwfile.example DESTINATION "${CMAKE_INSTALL_SYSCONFDIR}/mosquitto")
+install(FILES mosquitto.conf aclfile.example pskfile.example pwfile.example DESTINATION "/etc/mosquitto")
 
 
 # ========================================
--- origsrc/mosquitto-1.6.8/lib/CMakeLists.txt	2019-11-29 02:15:13.000000000 +0900
+++ src/mosquitto-1.6.8/lib/CMakeLists.txt	2019-12-23 17:17:42.322927000 +0900
@@ -87,7 +87,7 @@ set_target_properties(libmosquitto PROPE
 	SOVERSION 1
 )
 
-install(TARGETS libmosquitto RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+install(TARGETS libmosquitto RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 
 if (WITH_STATIC_LIBRARIES)
 	add_library(libmosquitto_static STATIC ${C_SRC})
--- origsrc/mosquitto-1.6.8/lib/cpp/CMakeLists.txt	2019-11-29 02:15:13.000000000 +0900
+++ src/mosquitto-1.6.8/lib/cpp/CMakeLists.txt	2019-12-23 17:17:55.924579500 +0900
@@ -13,7 +13,7 @@ set_target_properties(mosquittopp PROPER
 	VERSION ${VERSION}
 	SOVERSION 1
 )
-install(TARGETS mosquittopp RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+install(TARGETS mosquittopp RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 
 if (WITH_STATIC_LIBRARIES)
 	add_library(mosquittopp_static STATIC
--- origsrc/mosquitto-1.6.8/src/mosquitto_broker_internal.h	2019-11-29 02:15:13.000000000 +0900
+++ src/mosquitto-1.6.8/src/mosquitto_broker_internal.h	2019-12-23 17:18:30.030999300 +0900
@@ -719,7 +719,7 @@ void session_expiry__send_all(struct mos
 /* ============================================================
  * Window service and signal related functions
  * ============================================================ */
-#if defined(WIN32) || defined(__CYGWIN__)
+#if defined(WIN32)
 void service_install(void);
 void service_uninstall(void);
 void service_run(void);
