--- origsrc/mosquitto-1.2.2/lib/CMakeLists.txt	2013-10-22 05:24:36.000000000 +0900
+++ src/mosquitto-1.2.2/lib/CMakeLists.txt	2013-10-29 21:01:03.708510400 +0900
@@ -57,7 +57,7 @@ set_target_properties(libmosquitto PROPE
 	SOVERSION 1
 )
 
-install(TARGETS libmosquitto RUNTIME DESTINATION ${BINDIR} LIBRARY DESTINATION ${LIBDIR})
+install(TARGETS libmosquitto RUNTIME DESTINATION ${BINDIR} ARCHIVE DESTINATION ${LIBDIR})
 install(FILES mosquitto.h DESTINATION ${INCLUDEDIR})
 
 if (UNIX)
--- origsrc/mosquitto-1.2.2/lib/cpp/CMakeLists.txt	2013-10-22 05:24:36.000000000 +0900
+++ src/mosquitto-1.2.2/lib/cpp/CMakeLists.txt	2013-10-29 21:01:03.708510400 +0900
@@ -10,7 +10,7 @@ set_target_properties(mosquittopp PROPER
 	VERSION ${VERSION}
 	SOVERSION 1
 )
-install(TARGETS mosquittopp RUNTIME DESTINATION ${BINDIR} LIBRARY DESTINATION ${LIBDIR})
+install(TARGETS mosquittopp RUNTIME DESTINATION ${BINDIR} ARCHIVE DESTINATION ${LIBDIR})
 install(FILES mosquittopp.h DESTINATION ${INCLUDEDIR})
 
 if (UNIX)
--- origsrc/mosquitto-1.2.2/src/CMakeLists.txt	2013-10-22 05:24:36.000000000 +0900
+++ src/mosquitto-1.2.2/src/CMakeLists.txt	2013-10-29 21:01:03.708510400 +0900
@@ -73,7 +73,7 @@ add_definitions (-DWITH_BROKER)
 
 add_executable(mosquitto ${MOSQ_SRCS})
 
-set (MOSQ_LIBS ${OPENSSL_LIBRARIES})
+set (MOSQ_LIBS ${MOSQ_LIBS} ${OPENSSL_LIBRARIES})
 
 if (UNIX)
 	if (APPLE)
