--- origsrc/mosquitto-2.0.21/src/mosquitto_broker_internal.h	2025-03-07 01:23:49.000000000 +0900
+++ src/mosquitto-2.0.21/src/mosquitto_broker_internal.h	2025-03-18 23:45:37.977505300 +0900
@@ -842,7 +842,7 @@ void handle_sighup(int signal);
 /* ============================================================
  * Window service and signal related functions
  * ============================================================ */
-#if defined(WIN32) || defined(__CYGWIN__)
+#if defined(WIN32)
 void service_install(void);
 void service_uninstall(void);
 void service_run(void);
