--- origsrc/mosquitto-1.5.3/lib/CMakeLists.txt	2018-09-26 00:27:43.000000000 +0900
+++ src/mosquitto-1.5.3/lib/CMakeLists.txt	2018-10-27 09:53:03.906531500 +0900
@@ -108,7 +108,7 @@ set_target_properties(libmosquitto PROPE
 	SOVERSION 1
 )
 
-install(TARGETS libmosquitto RUNTIME DESTINATION "${BINDIR}" LIBRARY DESTINATION "${LIBDIR}")
+install(TARGETS libmosquitto RUNTIME DESTINATION "${BINDIR}" ARCHIVE DESTINATION "${LIBDIR}")
 
 if (${WITH_STATIC_LIBRARIES} STREQUAL ON)
 	add_library(libmosquitto_static STATIC ${C_SRC})
--- origsrc/mosquitto-1.5.3/lib/cpp/CMakeLists.txt	2018-09-26 00:27:43.000000000 +0900
+++ src/mosquitto-1.5.3/lib/cpp/CMakeLists.txt	2018-10-27 09:53:03.915537500 +0900
@@ -13,7 +13,7 @@ set_target_properties(mosquittopp PROPER
 	VERSION ${VERSION}
 	SOVERSION 1
 )
-install(TARGETS mosquittopp RUNTIME DESTINATION "${BINDIR}" LIBRARY DESTINATION "${LIBDIR}")
+install(TARGETS mosquittopp RUNTIME DESTINATION "${BINDIR}" ARCHIVE DESTINATION "${LIBDIR}")
 
 if (${WITH_STATIC_LIBRARIES} STREQUAL ON)
 	add_library(mosquittopp_static STATIC
--- origsrc/mosquitto-1.5.3/src/mosquitto_broker_internal.h	2018-09-26 00:27:43.000000000 +0900
+++ src/mosquitto-1.5.3/src/mosquitto_broker_internal.h	2018-10-27 09:53:03.927650400 +0900
@@ -626,7 +626,7 @@ int mosquitto_psk_key_get_default(struct
 /* ============================================================
  * Window service and signal related functions
  * ============================================================ */
-#if defined(WIN32) || defined(__CYGWIN__)
+#if defined(WIN32)
 void service_install(void);
 void service_uninstall(void);
 void service_run(void);
