--- origsrc/liboauth2-2.0.0/Makefile.am	2024-08-22 20:19:08.000000000 +0900
+++ src/liboauth2-2.0.0/Makefile.am	2024-10-29 09:45:39.189762300 +0900
@@ -6,7 +6,7 @@ EXTRA_DIST = autogen.sh ChangeLog README
 # generic
 #
 
-AM_CPPFLAGS = -Wall -Werror -Wno-error=deprecated-declarations -I${srcdir}/include -I${srcdir}/src @JANSSON_CFLAGS@ @OPENSSL_CFLAGS@
+AM_CPPFLAGS = -Wall -Wno-error=deprecated-declarations -I${builddir}/include -I${srcdir}/include -I${srcdir}/src @JANSSON_CFLAGS@ @OPENSSL_CFLAGS@
 AM_CPPFLAGS += $(CODE_COVERAGE_CPPFLAGS) $(CODE_COVERAGE_CFLAGS)
 AM_LDFLAGS = --coverage
 
@@ -45,6 +45,7 @@ liboauth2_la_pkgconfig_DATA = liboauth2.
 
 liboauth2_la_CFLAGS = @CURL_CFLAGS@ @CJOSE_CFLAGS@ @PCRE2_CFLAGS@
 liboauth2_la_LIBADD = @CURL_LIBS@ @CJOSE_LIBS@ @PCRE2_LIBS@
+liboauth2_la_LDFLAGS = -no-undefined
 
 liboauth2_la_SOURCES = \
 	src/version.c \
@@ -121,7 +122,8 @@ liboauth2_apache_la_pkgconfig_DATA = lib
 
 liboauth2_apache_la_CPPFLAGS = $(AM_CPPFLAGS) -DHAVE_APACHE
 liboauth2_apache_la_CFLAGS = @APACHE_CFLAGS@ @PCRE2_CFLAGS@
-liboauth2_apache_la_LIBADD = liboauth2.la @APR_LIBS@ @JANSSON_LIBS@
+liboauth2_apache_la_LIBADD = liboauth2.la @APR_LIBS@ @JANSSON_LIBS@ -lhttpd
+liboauth2_apache_la_LDFLAGS = -no-undefined
 
 includesub_HEADERS += \
 	include/oauth2/apache.h
