--- origsrc/libcotp-1.2.6/CMakeLists.txt	2022-05-18 23:27:42.000000000 +0900
+++ src/libcotp-1.2.6/CMakeLists.txt	2022-05-23 09:56:42.427075500 +0900
@@ -41,6 +41,7 @@ set(COTP_INC_DIR "${CMAKE_INSTALL_INCLUD
 
 install(
   TARGETS cotp
+  RUNTIME DESTINATION bin
   ARCHIVE DESTINATION ${COTP_LIB_DIR}
   LIBRARY DESTINATION ${COTP_LIB_DIR}
   COMPONENT library
--- origsrc/libcotp-1.2.6/cotp.pc.in	2022-05-18 23:27:42.000000000 +0900
+++ src/libcotp-1.2.6/cotp.pc.in	2022-05-23 09:57:17.267129100 +0900
@@ -7,5 +7,6 @@ Name: libcotp
 Description: C library that generates TOTP and HOTP
 Version: @BUILD_VERSION@
 URL: URL: https://github.com/paolostivanin/libcotp
-Libs: -L${libdir} -lcotp
+Libs: -L${libdir} -lcotp -lgcrypt
 Cflags: -I${includedir}
+Requires: baseencode
