--- origsrc/libcotp-2.0.2/CMakeLists.txt	2023-10-26 15:10:04.000000000 +0900
+++ src/libcotp-2.0.2/CMakeLists.txt	2023-11-21 11:58:59.201989700 +0900
@@ -32,7 +32,9 @@ add_compile_options(-Wall -Wextra -O3 -W
         -fdiagnostics-color=always -Wstrict-prototypes -Wunreachable-code -Wchar-subscripts -Wwrite-strings -Wpointer-arith -Wbad-function-cast
         -Wcast-align -Werror=format-security -Werror=implicit-function-declaration -Wno-sign-compare -Wno-format-nonliteral -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3)
 
-add_link_options(-Wl,--no-add-needed -Wl,--as-needed -Wl,-z,relro,-z,now)
+if (NOT CYGWIN)
+  add_link_options(-Wl,--no-add-needed -Wl,--as-needed -Wl,-z,relro,-z,now)
+endif()
 
 add_library(cotp ${SOURCE_FILES})
 
@@ -57,6 +59,7 @@ set(COTP_INC_DIR "${CMAKE_INSTALL_INCLUD
 
 install(
   TARGETS cotp
+  RUNTIME DESTINATION bin
   ARCHIVE DESTINATION ${COTP_LIB_DIR}
   LIBRARY DESTINATION ${COTP_LIB_DIR}
   COMPONENT library
--- origsrc/libcotp-2.0.2/cotp.pc.in	2023-10-26 15:10:04.000000000 +0900
+++ src/libcotp-2.0.2/cotp.pc.in	2023-11-21 11:58:59.201989700 +0900
@@ -7,5 +7,5 @@ Name: libcotp
 Description: C library that generates TOTP and HOTP
 Version: @CMAKE_PROJECT_VERSION@
 URL: URL: https://github.com/paolostivanin/libcotp
-Libs: -L${libdir} -lcotp
+Libs: -L${libdir} -lcotp -lgcrypt
 Cflags: -I${includedir}
