--- origsrc/libcotp-1.2.4/CMakeLists.txt	2021-12-29 17:35:40.000000000 +0900
+++ src/libcotp-1.2.4/CMakeLists.txt	2022-01-09 22:05:02.022200100 +0900
@@ -39,6 +39,7 @@ set(LIB_INSTALL_DIR "${CMAKE_INSTALL_LIB
 
 install(
   TARGETS cotp
+  RUNTIME DESTINATION bin
   ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
   LIBRARY DESTINATION ${LIB_INSTALL_DIR}
   COMPONENT library
--- origsrc/libcotp-1.2.4/cotp.pc.in	2021-12-29 17:35:40.000000000 +0900
+++ src/libcotp-1.2.4/cotp.pc.in	2022-01-09 22:05:02.077192700 +0900
@@ -6,5 +6,6 @@ includedir=${prefix}/@CMAKE_INSTALL_INCL
 Name: libcotp
 Description: C library that generates TOTP and HOTP
 Version: @BUILD_VERSION@
-Libs: -L${libdir} -lcotp
+Libs: -L${libdir} -lcotp -lgcrypt
 Cflags: -I${includedir}
+Requires: baseencode
