--- origsrc/ijs-0.35/Makefile.am	2018-02-28 02:31:37.000000000 +0900
+++ src/ijs-0.35/Makefile.am	2026-01-12 20:10:44.918104900 +0900
@@ -59,7 +59,7 @@ else
 libijs_la_LIBADD = ijs_exec_unix.lo $(IJS_DEPLIBS)
 endif
 # Uncommment to build an unversioned library (version in soname)
-libijs_la_LDFLAGS = -release $(IJS_VERSION) -rpath $(libdir)
+libijs_la_LDFLAGS = -no-undefined -release $(IJS_VERSION) -rpath $(libdir)
 # Uncomment to build a versioned library
 #libijs_la_LDFLAGS = \
 # -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
--- origsrc/ijs-0.35/configure.ac	2018-02-28 02:31:37.000000000 +0900
+++ src/ijs-0.35/configure.ac	2026-01-12 20:10:44.936213200 +0900
@@ -137,7 +137,7 @@ AC_SUBST(PS2PDF)
 dnl conditional building
 
 case ${host_os} in
-  cygwin | mingw32 | pw32 | interix3 | interix | uwin) libijs_sysdeps="windows";;
+  mingw32 | pw32 | interix3 | interix | uwin) libijs_sysdeps="windows";;
   *) libijs_sysdeps="unix";;
 esac
 AM_CONDITIONAL(SYSDEPS_WINDOWS, test x$libijs_sysdeps = xwindows)
