--- origsrc/libhandoff-0.1/Makefile.am	2007-06-08 21:13:47.000000000 +0900
+++ src/libhandoff-0.1/Makefile.am	2015-02-16 20:01:40.619477100 +0900
@@ -3,7 +3,7 @@ lib_LTLIBRARIES = libhandoff.la
 AM_CPPFLAGS = @DEPS_CFLAGS@ -DPACKAGE_LOCALE_DIR=\"@datadir@/locale/@PACKAGE_NAME@\"
 
 libhandoff_la_LIBADD = @DEPS_LIBS@
-libhandoff_la_LDFLAGS = -version-info 0:0:0
+libhandoff_la_LDFLAGS = -no-undefined -version-info 0:0:0
 
 libhandoff_la_SOURCES = handoff.c handoff.h
 
--- origsrc/libhandoff-0.1/configure.ac	2007-06-08 21:13:05.000000000 +0900
+++ src/libhandoff-0.1/configure.ac	2015-02-16 20:04:22.308528100 +0900
@@ -10,7 +10,7 @@ AM_INIT_AUTOMAKE([foreign dist-bzip2])
 AC_PROG_CC
 AC_PROG_LIBTOOL
 
-PKG_CHECK_MODULES(DEPS, glib-2.0)
+PKG_CHECK_MODULES([DEPS], [gobject-2.0])
 
 AC_SUBST(DEPS_CFLAGS)
 AC_SUBST(DEPS_LIBS)
