--- origsrc/streamvbyte-1.0.0/CMakeLists.txt	2023-05-20 12:10:46.000000000 +0900
+++ src/streamvbyte-1.0.0/CMakeLists.txt	2024-10-06 15:47:44.170624500 +0900
@@ -136,7 +136,7 @@ install(
         ${PROJECT_SOURCE_DIR}/include/streamvbyte_zigzag.h
     DESTINATION include
 )
-install(TARGETS streamvbyte DESTINATION lib)
+install(TARGETS streamvbyte RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 
 option(STREAMVBYTE_SANITIZE_UNDEFINED "Sanitize undefined behavior" OFF)
 if(STREAMVBYTE_SANITIZE_UNDEFINED)
@@ -184,3 +184,10 @@ if(STREAMVBYTE_ENABLE_TESTS)
     add_test(NAME unit COMMAND unit)
     add_custom_target(check COMMAND ctest --output-on-failure DEPENDS unit)
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/streamvbyte.pc.in ${CMAKE_BINARY_DIR}/streamvbyte.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/streamvbyte.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/streamvbyte-1.0.0/streamvbyte.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/streamvbyte-1.0.0/streamvbyte.pc.in	2024-10-06 15:47:32.439125600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: streamvbyte
+Description: Fast integer compression in C using the StreamVByte codec
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lstreamvbyte
+Cflags: -I${includedir}
