--- origsrc/maeparser-1.2.1/CMakeLists.txt	2019-09-21 06:14:38.000000000 +0900
+++ src/maeparser-1.2.1/CMakeLists.txt	2019-09-28 16:01:43.685138200 +0900
@@ -42,7 +42,7 @@ install(TARGETS maeparser
 
 INSTALL(EXPORT maeparser-targets
     FILE ${PROJECT_NAME}-config.cmake
-    DESTINATION lib/cmake)
+    DESTINATION lib/cmake/maeparser)
 
 target_compile_definitions(maeparser PRIVATE IN_MAEPARSER)
 
@@ -58,3 +58,9 @@ add_subdirectory(test)
 install(FILES Buffer.hpp MaeBlock.hpp  MaeParserConfig.hpp  MaeParser.hpp  Reader.hpp MaeConstants.hpp
         DESTINATION include/maeparser)
 
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/maeparser.pc.in ${CMAKE_BINARY_DIR}/maeparser.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/maeparser.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/maeparser-1.2.1/maeparser.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/maeparser-1.2.1/maeparser.pc.in	2019-09-28 16:01:04.291691300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: maeparser
+Description: Maestro file parser
+Version: @VERSION@
+Libs: -L${libdir} -lmaeparser
+Cflags: -I${includedir}
