--- origsrc/lunar-date-2.4.2/configure.ac	2015-11-20 14:26:13.000000000 +0900
+++ src/lunar-date-2.4.2/configure.ac	2017-10-12 08:11:48.375538500 +0900
@@ -80,10 +80,14 @@ dnl check for win32 build
 dnl ================================================================
 AC_MSG_CHECKING([for some Win32 platform])
 case "$host" in
-  *-*-mingw*|*-*-cygwin*)
+  *-*-mingw*)
     platform_win32=yes
     os_win32=yes
     ;;
+  *-*-cygwin*)
+    platform_win32=yes
+    os_win32=no
+    ;;
   *)
     platform_win32=no
     os_win32=no
--- origsrc/lunar-date-2.4.2/lunar-date/Makefile.am	2015-11-20 14:26:13.000000000 +0900
+++ src/lunar-date-2.4.2/lunar-date/Makefile.am	2017-10-12 08:27:40.504000000 +0900
@@ -63,7 +63,7 @@ libtool_opts =							\
 
 source_h = 	\
 	$(srcdir)/lunar-date.h		\
-	$(srcdir)/lunar-version.h
+	$(abs_top_builddir)/lunar-date/lunar-version.h
 
 source_c =	\
 	$(srcdir)/lunar-date.c		\
