--- origsrc/lunar-date-2.4.0/lunar-date/Makefile.am	2011-05-07 20:45:08.000000000 +0900
+++ src/lunar-date-2.4.0/lunar-date/Makefile.am	2015-03-17 14:51:51.581722600 +0900
@@ -30,7 +30,7 @@ liblunar_date_2_0_includedir = $(include
 liblunar_date_2_0_include_HEADERS = $(source_h)
 liblunar_date_2_0_la_SOURCES =	$(source_c) $(BUILT_SOURCES)
 liblunar_date_2_0_la_LIBADD = 	$(LUNAR_DATE_LIBS)
-liblunar_date_2_0_la_LDFLAGS = \
+liblunar_date_2_0_la_LDFLAGS = -no-undefined \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
 	-export-dynamic $(export_symbols)
 
