--- origsrc/smack/CMakeLists.txt	2017-07-06 23:17:30.000000000 +0900
+++ src/smack/CMakeLists.txt	2017-07-06 23:23:16.987000000 +0900
@@ -31,3 +31,11 @@ add_custom_target(dist
     COMMAND git archive --prefix=${ARCHIVE_NAME}-${smack_VERSION}/ HEAD
     | bzip2 > ${CMAKE_BINARY_DIR}/${ARCHIVE_NAME}-${smack_VERSION}.tar.bz2
     WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "0.5.0")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/${PROJECT_NAME}.pc.in ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/smack/smack.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/smack/smack.pc.in	2017-07-06 23:23:45.115000000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: smack
+Description: Low-level IO storage which packs data into sorted (zlib/bzip2/snappy compressed) blobs
+Version: @VERSION@
+Libs: -L${libdir} -lsmack
+Cflags: -I${includedir}
--- origsrc/smack/src/CMakeLists.txt	2017-07-06 23:17:30.000000000 +0900
+++ src/smack/src/CMakeLists.txt	2017-07-06 23:17:31.440842300 +0900
@@ -7,4 +7,4 @@ IF (NOT CMAKE_INSTALL_LIBDIR)
 SET(CMAKE_INSTALL_LIBDIR lib CACHE PATH "Output directory for libraries")
 ENDIF (NOT CMAKE_INSTALL_LIBDIR)
 
-install(TARGETS smack LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS smack RUNTIME DESTINATION bin ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
