--- origsrc/bliss-0.72/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/bliss-0.72/Makefile.am	2012-03-27 20:50:01.519663200 +0900
@@ -0,0 +1,20 @@
+
+bin_PROGRAMS = bliss
+bliss_CPPFLAGS = -DBLISS_USE_GMP
+bliss_LDADD = libbliss.la
+bliss_SOURCES = bliss.cc
+
+lib_LTLIBRARIES = libbliss.la
+
+libbliss_la_CPPFLAGS = -DBLISS_USE_GMP
+libbliss_la_LDFLAGS = -no-undefined
+libbliss_la_LIBADD = -lgmp
+libbliss_la_SOURCES = defs.cc graph.cc partition.cc orbit.cc \
+uintseqhash.cc heap.cc timer.cc utils.cc bliss_C.cc
+
+man_MANS = bliss.1
+
+pkginclude_HEADERS = \
+bignum.hh  bliss_C.h  defs.hh       graph.hh  heap.hh         kqueue.hh \
+kstack.hh  orbit.hh   partition.hh  timer.hh  uintseqhash.hh  utils.hh
+
--- origsrc/bliss-0.72/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/bliss-0.72/configure.ac	2012-03-27 20:46:20.535288200 +0900
@@ -0,0 +1,31 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([bliss], [0.72])
+AC_CONFIG_SRCDIR([bignum.hh])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+LT_INIT([disable-static])
+
+# Checks for libraries.
+# FIXME: Replace `main' with a function in `-lgmp':
+AC_CHECK_LIB([gmp], [main])
+
+# Checks for header files.
+AC_CHECK_HEADERS([limits.h stdlib.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_HEADER_STDBOOL
+AC_C_INLINE
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
