--- origsrc/bliss-0.77/CMakeLists.txt	2026-06-19 19:32:18.572839500 +0900
+++ src/bliss-0.77/CMakeLists.txt	2026-06-19 20:24:49.095019800 +0900
@@ -54,13 +54,13 @@ set_property(TARGET bliss PROPERTY POSIT
 if(USE_GMP)
   target_link_libraries(bliss ${GMP_LIBRARIES})
 endif(USE_GMP)
-
-# Add the static library
-add_library(bliss_static STATIC ${BLISS_SOURCE_FILES})
+if (DEFINED SOVERSION)
+  set_target_properties(bliss PROPERTIES SOVERSION ${SOVERSION})
+endif()
 
 # Add the executable
 add_executable(bliss-executable src/bliss.cc)
-target_link_libraries(bliss-executable bliss_static)
+target_link_libraries(bliss-executable bliss)
 if(USE_GMP)
   target_link_libraries(bliss-executable ${GMP_LIBRARIES})
 endif(USE_GMP)
@@ -89,3 +89,10 @@ set(
 install(TARGETS bliss-executable RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 install(TARGETS bliss LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(FILES ${BLISS_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/bliss)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/b/libbliss-cxx.pc.in ${CMAKE_BINARY_DIR}/libbliss-cxx.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libbliss-cxx.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/bliss-0.77/b/libbliss-cxx.pc.in	2026-06-19 19:32:18.078164400 +0900
+++ src/bliss-0.77/b/libbliss-cxx.pc.in	2026-06-19 20:25:03.025342700 +0900
@@ -8,5 +8,5 @@ Description: library to compute graph au
 Requires: gmp
 Version: @PACKAGE_VERSION@
 URL: @PACKAGE_URL@
-Cflags: -D BLISS_USE_GMP -I${includedir}
+Cflags: -DBLISS_USE_GMP -I${includedir}
 Libs: -L${libdir} -lbliss
