--- origsrc/libticables2-1.3.5/configure.ac	2016-03-23 20:38:40.000000000 +0900
+++ src/libticables2-1.3.5/configure.ac	2025-01-19 16:57:41.664856000 +0900
@@ -91,7 +91,7 @@ if [test "x$cab_cv_use_libusb" != xno];
       ])
   else
     case $host_os in
-    *mingw*|*cygwin*)
+    *mingw*)
       # use builtin libusb_dyn.c to load libusb at runtime
       HAVE_LIBUSB=yes
       LIBUSB_CFLAGS='-I$(top_srcdir)/src/win32'
@@ -174,7 +174,7 @@ CFLAGS="$CFLAGS $ARCH"
 # What flag to depends on gcc version: gcc3 uses "-mms-bitfields", while
 # gcc2 uses "-fnative-struct".
 case $host_os in
-*mingw*|*cygwin*)
+*mingw*)
   if test x$GCC = xyes; then
     msnative_struct=''
     AC_MSG_CHECKING([how to get MSVC-compatible struct packing])
--- origsrc/libticables2-1.3.5/src/detect.c	2013-04-13 22:26:51.000000000 +0900
+++ src/libticables2-1.3.5/src/detect.c	2025-01-19 16:57:41.664856000 +0900
@@ -32,7 +32,7 @@
 # include <config.h>
 #endif
 
-#if defined(__LINUX__)
+#if defined(__LINUX__) || defined(__CYGWIN__)
 #include "linux/detect.c"
 #elif defined(__MACOSX__)
 #include "macos/detect.c"
--- origsrc/libticables2-1.3.5/src/ioports.c	2013-04-13 22:26:51.000000000 +0900
+++ src/libticables2-1.3.5/src/ioports.c	2025-01-19 16:57:41.664856000 +0900
@@ -23,7 +23,7 @@
 # include <config.h>
 #endif
 
-#if defined(__LINUX__)
+#if defined(__LINUX__) || defined(__CYGWIN__)
 #include "linux/ioports.c"
 #elif defined(__BSD__)
 #include "linux/ioports.c"
--- origsrc/libticables2-1.3.5/src/link_blk.c	2015-04-09 01:06:16.000000000 +0900
+++ src/libticables2-1.3.5/src/link_blk.c	2025-01-19 16:57:41.664856000 +0900
@@ -27,7 +27,7 @@
 
 #ifndef NO_CABLE_BLK
 
-#if defined(__LINUX__) && defined(HAVE_TERMIOS_H)
+#if (defined(__LINUX__) || defined(__CYGWIN__))
 #include "linux/link_ser.c"
 
 #elif defined(__BSD__)
--- origsrc/libticables2-1.3.5/src/link_gry.c	2015-04-09 01:06:16.000000000 +0900
+++ src/libticables2-1.3.5/src/link_gry.c	2025-01-19 16:57:41.664856000 +0900
@@ -27,7 +27,7 @@
 
 #ifndef NO_CABLE_GRY
 
-#if defined(__LINUX__)
+#if defined(__LINUX__) || defined(__CYGWIN__)
 #include "linux/link_gry.c"
 
 #elif defined(__BSD__)
--- origsrc/libticables2-1.3.5/src/link_tcpc.c	2016-04-04 14:59:39.000000000 +0900
+++ src/libticables2-1.3.5/src/link_tcpc.c	2025-01-19 16:57:41.664856000 +0900
@@ -26,7 +26,7 @@
 
 #ifndef NO_CABLE_TCPC
 
-#if defined(__LINUX__)
+#if defined(__LINUX__) || defined(__CYGWIN__)
 #include "linux/link_tcpc.c"
 
 #elif defined(__BSD__)
--- origsrc/libticables2-1.3.5/src/link_tcps.c	2016-04-04 14:59:39.000000000 +0900
+++ src/libticables2-1.3.5/src/link_tcps.c	2025-01-19 16:57:41.686013100 +0900
@@ -26,7 +26,7 @@
 
 #ifndef NO_CABLE_TCPS
 
-#if defined(__LINUX__)
+#if defined(__LINUX__) || defined(__CYGWIN__)
 #include "linux/link_tcps.c"
 
 #elif defined(__BSD__)
--- origsrc/libticables2-1.3.5/src/link_tie.c	2013-04-13 22:26:51.000000000 +0900
+++ src/libticables2-1.3.5/src/link_tie.c	2025-01-19 16:57:41.686013100 +0900
@@ -27,7 +27,7 @@
 
 #ifndef NO_CABLE_TIE
 
-#if defined(__LINUX__)
+#if defined(__LINUX__) || defined(__CYGWIN__)
 #include "linux/link_tie.c"
 
 #elif defined(__BSD__)
--- origsrc/libticables2-1.3.5/src/link_vti.c	2013-04-13 22:26:51.000000000 +0900
+++ src/libticables2-1.3.5/src/link_vti.c	2025-01-19 16:57:41.686013100 +0900
@@ -27,7 +27,7 @@
 
 #ifndef NO_CABLE_VTI
 
-#if defined(__LINUX__)
+#if defined(__LINUX__) || defined(__CYGWIN__)
 #include "linux/link_vti.c"
 
 #elif defined(__BSD__)
