--- origsrc/resiprocate-resiprocate-1.12.0/repro/Makefile.am	2020-05-09 20:55:09.000000000 +0900
+++ src/resiprocate-resiprocate-1.12.0/repro/Makefile.am	2022-06-07 22:49:45.298461700 +0900
@@ -45,11 +45,12 @@ repro_LDADD += @LIBSSL_LIBADD@ @LIBPTHRE
 
 lib_LTLIBRARIES = librepro.la
 
+librepro_la_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/contrib/cajun/include
 if BUILD_REPRO_DSO_PLUGINS
 librepro_la_CXXFLAGS = -DREPRO_DSO_PLUGIN_DIR_DEFAULT='"@reproplugindir@"'
 endif
-librepro_la_LDFLAGS = @LIBTOOL_VERSION_RELEASE@ -export-dynamic
-librepro_la_LIBADD = -ldb_cxx
+librepro_la_LDFLAGS = -no-undefined @LIBTOOL_VERSION_RELEASE@ -export-dynamic
+librepro_la_LIBADD = -ldb -ldb_cxx
 librepro_la_LIBADD += ../resip/dum/libdum.la
 librepro_la_LIBADD += ../resip/stack/libresip.la
 librepro_la_LIBADD += ../rutil/librutil.la
--- origsrc/resiprocate-resiprocate-1.12.0/resip/dum/Makefile.am	2020-05-09 20:55:09.000000000 +0900
+++ src/resiprocate-resiprocate-1.12.0/resip/dum/Makefile.am	2022-06-07 13:42:45.900925400 +0900
@@ -15,7 +15,7 @@ libdum_la_LIBADD = ../stack/libresip.la
 libdum_la_LIBADD += ../../rutil/librutil.la
 libdum_la_LIBADD += @LIBSSL_LIBADD@ @LIBRADIUS_LIBADD@
 libdum_la_LIBADD += @LIBSTL_LIBADD@
-libdum_la_LDFLAGS = @LIBTOOL_VERSION_RELEASE@ -export-dynamic
+libdum_la_LDFLAGS = -no-undefined @LIBTOOL_VERSION_RELEASE@ -export-dynamic
 
 libdum_la_SOURCES =   \
 	AppDialog.cxx \
--- origsrc/resiprocate-resiprocate-1.12.0/resip/stack/Makefile.am	2020-05-09 20:55:09.000000000 +0900
+++ src/resiprocate-resiprocate-1.12.0/resip/stack/Makefile.am	2022-06-07 13:11:01.678958600 +0900
@@ -41,7 +41,7 @@ lib_LTLIBRARIES = libresip.la
 libresip_la_LIBADD = ../../rutil/librutil.la
 libresip_la_LIBADD += @LIBSSL_LIBADD@
 libresip_la_LIBADD += @LIBSTL_LIBADD@
-libresip_la_LDFLAGS = @LIBTOOL_VERSION_RELEASE@ -export-dynamic
+libresip_la_LDFLAGS = -no-undefined @LIBTOOL_VERSION_RELEASE@ -export-dynamic
 
 libresip_la_SOURCES = \
 	Auth.cxx \
--- origsrc/resiprocate-resiprocate-1.12.0/rutil/Condition.cxx	2020-05-09 20:55:09.000000000 +0900
+++ src/resiprocate-resiprocate-1.12.0/rutil/Condition.cxx	2022-06-07 11:34:54.481987300 +0900
@@ -5,7 +5,9 @@
 #  include <pthread.h>
 #  include <errno.h>
 #  include <sys/time.h>
-#  include <sys/syscall.h>
+#  if !defined(__CYGWIN__)
+#    include <sys/syscall.h>
+#  endif
 #  include <unistd.h>
 #endif
 
--- origsrc/resiprocate-resiprocate-1.12.0/rutil/Makefile.am	2020-05-09 20:55:09.000000000 +0900
+++ src/resiprocate-resiprocate-1.12.0/rutil/Makefile.am	2022-06-07 12:51:30.250987200 +0900
@@ -30,7 +30,7 @@ librutil_la_LIBADD += @LIBSTL_LIBADD@
 librutil_la_LIBADD += @LIBLOG_LIBADD@
 #librutil_la_LIBADD += -lrt
 librutil_la_LIBADD += @LIBPTHREAD_LIBADD@
-librutil_la_LDFLAGS = @LIBTOOL_VERSION_RELEASE@ -export-dynamic
+librutil_la_LDFLAGS = -no-undefined @LIBTOOL_VERSION_RELEASE@ -export-dynamic
 
 librutil_la_SOURCES = \
 	AbstractFifo.cxx \
--- origsrc/resiprocate-resiprocate-1.12.0/rutil/Timer.cxx	2020-05-09 20:55:09.000000000 +0900
+++ src/resiprocate-resiprocate-1.12.0/rutil/Timer.cxx	2022-06-07 11:39:50.375089400 +0900
@@ -9,7 +9,9 @@
 #  include <winbase.h>
 #else
 #  include <sys/time.h>
-#  include <sys/syscall.h>
+#  if !defined(__CYGWIN__)
+#    include <sys/syscall.h>
+#  endif
 #  include <unistd.h>
 #endif
 
--- origsrc/resiprocate-resiprocate-1.12.0/rutil/dns/AresDns.cxx	2020-05-09 20:55:09.000000000 +0900
+++ src/resiprocate-resiprocate-1.12.0/rutil/dns/AresDns.cxx	2022-06-07 11:41:49.121770900 +0900
@@ -22,10 +22,8 @@
 #include "rutil/FdPoll.hxx"
 
 #if !defined(WIN32)
-#if !defined(__CYGWIN__)
 #include <arpa/nameser.h>
 #endif
-#endif
 
 using namespace resip;
 
--- origsrc/resiprocate-resiprocate-1.12.0/rutil/dns/DnsCnameRecord.cxx	2020-05-09 20:55:09.000000000 +0900
+++ src/resiprocate-resiprocate-1.12.0/rutil/dns/DnsCnameRecord.cxx	2022-06-07 11:43:19.648632800 +0900
@@ -6,14 +6,12 @@
 
 #include "AresCompat.hxx"
 
-#ifndef __CYGWIN__
 #ifndef RRFIXEDSZ
 #define RRFIXEDSZ 10
 #endif
 #ifndef NS_RRFIXEDSZ
 #define NS_RRFIXEDSZ 10
 #endif
-#endif
 
 #include "rutil/Data.hxx"
 #include "rutil/BaseException.hxx"
--- origsrc/resiprocate-resiprocate-1.12.0/rutil/dns/DnsHostRecord.cxx	2020-05-09 20:55:09.000000000 +0900
+++ src/resiprocate-resiprocate-1.12.0/rutil/dns/DnsHostRecord.cxx	2022-06-07 11:44:04.037948400 +0900
@@ -6,14 +6,12 @@
 
 #include "AresCompat.hxx"
 
-#ifndef __CYGWIN__
 #ifndef RRFIXEDSZ
 #define RRFIXEDSZ 10
 #endif
 #ifndef NS_RRFIXEDSZ
 #define NS_RRFIXEDSZ 10
 #endif
-#endif
 
 #include "rutil/Data.hxx"
 #include "rutil/DnsUtil.hxx"
--- origsrc/resiprocate-resiprocate-1.12.0/rutil/dns/DnsNaptrRecord.cxx	2020-05-09 20:55:09.000000000 +0900
+++ src/resiprocate-resiprocate-1.12.0/rutil/dns/DnsNaptrRecord.cxx	2022-06-07 11:48:29.319744300 +0900
@@ -6,14 +6,12 @@
 
 #include "AresCompat.hxx"
 
-#ifndef __CYGWIN__
 #ifndef RRFIXEDSZ
 #define RRFIXEDSZ 10
 #endif
 #ifndef NS_RRFIXEDSZ
 #define NS_RRFIXEDSZ 10
 #endif
-#endif
 
 #include "rutil/Data.hxx"
 #include "rutil/Logger.hxx"
--- origsrc/resiprocate-resiprocate-1.12.0/rutil/dns/DnsSrvRecord.cxx	2020-05-09 20:55:09.000000000 +0900
+++ src/resiprocate-resiprocate-1.12.0/rutil/dns/DnsSrvRecord.cxx	2022-06-07 11:49:14.031918000 +0900
@@ -6,14 +6,12 @@
 
 #include "AresCompat.hxx"
 
-#ifndef __CYGWIN__
 #ifndef RRFIXEDSZ
 #define RRFIXEDSZ 10
 #endif
 #ifndef NS_RRFIXEDSZ
 #define NS_RRFIXEDSZ 10
 #endif
-#endif
 
 #include "rutil/Data.hxx"
 #include "rutil/BaseException.hxx"
--- origsrc/resiprocate-resiprocate-1.12.0/rutil/dns/DnsStub.cxx	2020-05-09 20:55:09.000000000 +0900
+++ src/resiprocate-resiprocate-1.12.0/rutil/dns/DnsStub.cxx	2022-06-07 11:49:47.621360900 +0900
@@ -13,10 +13,8 @@
 #include "AresCompat.hxx"
 
 #ifndef WIN32
-#ifndef __CYGWIN__
 #include <arpa/nameser.h>
 #endif
-#endif
 
 #include "rutil/FdPoll.hxx"
 #include "rutil/Logger.hxx"
--- origsrc/resiprocate-resiprocate-1.12.0/rutil/dns/RRCache.cxx	2020-05-09 20:55:09.000000000 +0900
+++ src/resiprocate-resiprocate-1.12.0/rutil/dns/RRCache.cxx	2022-06-07 12:45:05.410631000 +0900
@@ -8,11 +8,9 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <arpa/inet.h>
-#ifndef __CYGWIN__
-#  include <netinet/in.h>
-#  include <arpa/nameser.h>
-#  include <resolv.h>
-#endif
+#include <netinet/in.h>
+#include <arpa/nameser.h>
+#include <resolv.h>
 #include <netdb.h>
 #include <netinet/in.h>
 #else
--- origsrc/resiprocate-resiprocate-1.12.0/rutil/dns/RRList.cxx	2020-05-09 20:55:09.000000000 +0900
+++ src/resiprocate-resiprocate-1.12.0/rutil/dns/RRList.cxx	2022-06-07 12:48:38.425509800 +0900
@@ -11,10 +11,8 @@
 #endif
 
 #ifndef WIN32
-#ifndef __CYGWIN__
 #include <arpa/nameser.h>
 #endif
-#endif
 
 #include "rutil/Logger.hxx"
 #include "rutil/compat.hxx"
--- origsrc/resiprocate-resiprocate-1.12.0/rutil/dns/RROverlay.cxx	2020-05-09 20:55:09.000000000 +0900
+++ src/resiprocate-resiprocate-1.12.0/rutil/dns/RROverlay.cxx	2022-06-07 12:50:09.441243400 +0900
@@ -4,14 +4,12 @@
 
 #include "AresCompat.hxx"
 
-#ifndef __CYGWIN__
 #ifndef RRFIXEDSZ
 #define RRFIXEDSZ 10
 #endif
 #ifndef NS_RRFIXEDSZ
 #define NS_RRFIXEDSZ 10
 #endif
-#endif
 
 #include <cstdlib>
 #include <vector>
