--- origsrc/libzstd-seek-1.2.0/CMakeLists.txt	2022-08-12 22:25:51.000000000 +0900
+++ src/libzstd-seek-1.2.0/CMakeLists.txt	2022-12-02 14:39:43.342273600 +0900
@@ -6,4 +6,18 @@ set(CMAKE_C_STANDARD 99)
 add_library(zstd-seek zstd-seek.c zstd-seek.h)
 target_link_libraries(zstd-seek zstd m)
 
+if (DEFINED SOVERSION)
+    set_target_properties(zstd-seek PROPERTIES SOVERSION ${SOVERSION})
+endif()
+
 add_subdirectory(examples)
+
+install(TARGETS zstd-seek RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+install(FILES zstd-seek.h DESTINATION include)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libzstd-seek.pc.in ${CMAKE_BINARY_DIR}/libzstd-seek.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libzstd-seek.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libzstd-seek-1.2.0/libzstd-seek.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libzstd-seek-1.2.0/libzstd-seek.pc.in	2022-12-02 14:39:43.347412400 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libzstd-seek
+Description: Library to read zstd files at arbitraty positions
+Version: @VERSION@
+Libs: -L${libdir} -lzstd-seek
+Cflags: -I${includedir}
+Requires: libzstd
