--- origsrc/skstream-0.3.9/configure.ac	2012-11-07 08:24:01.000000000 +0900
+++ src/skstream-0.3.9/configure.ac	2014-03-16 13:51:42.011970000 +0900
@@ -81,6 +81,9 @@ SKSTREAM_SOCKET_TYPE=int;
 SKSTREAM_SOCKET_HEADER="#include <unistd.h>"
 AC_TRY_COMPILE([#include <ws2tcpip.h>],
 [
+#if defined(__CYGWIN__)
+#error "cygwin have no SOCKET"
+#endif
     SOCKET foo;
     return foo; /* Avoids warning */
 ],
@@ -143,6 +146,9 @@ SKSTREAM_WINDOWS_HEADER_FLAG=0
 
 AC_TRY_COMPILE([#include <winsock2.h>],
 [
+#if defined(__CYGWIN__)
+#error "cygwin have no winsock"
+#endif
     timeval foo;
     return &foo == 0; /* Avoids warning */
 ],
@@ -274,6 +280,9 @@ dnl Test for closesocket
 AC_MSG_CHECKING([for closesocket])
 AC_TRY_COMPILE([#include <ws2tcpip.h>],
 [
+#if defined(__CYGWIN__)
+#error "cygwin have no closesocket"
+#endif
     closesocket(0);
 ],
 [
--- origsrc/skstream-0.3.9/skstream/Makefile.am	2012-07-12 23:51:20.000000000 +0900
+++ src/skstream-0.3.9/skstream/Makefile.am	2014-03-16 13:51:42.011970000 +0900
@@ -1,7 +1,7 @@
 AM_CPPFLAGS = -I$(top_srcdir) -I$(top_builddir)
 
 lib_LTLIBRARIES = libskstream-0.3.la
-libskstream_0_3_la_LDFLAGS = -version-info @SKSTREAM_VERSION_INFO@
+libskstream_0_3_la_LDFLAGS = -version-info @SKSTREAM_VERSION_INFO@ -no-undefined
 
 libskstream_0_3_la_SOURCES = sksocket.cpp skstream.cpp skserver.cpp \
                              skaddress.cpp skpoll.cpp
