--- origsrc/skstream-0.3.7/configure.ac	2010-08-03 02:53:43.000000000 +0900
+++ src/skstream-0.3.7/configure.ac	2010-10-31 12:59:56.612528500 +0900
@@ -29,9 +29,7 @@ AC_SUBST(SKSTREAM_VERSION_INFO)
 AC_SUBST(SKSTREAM_VERSION_NO)
 
 dnl Test for GNU libtool
-
-AC_DISABLE_STATIC
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 dnl Test for C++ compiler
 
@@ -97,6 +95,9 @@ AC_CACHE_VAL(ac_cv_socket_type,
 [
     AC_TRY_COMPILE([#include <ws2tcpip.h>],
     [
+#if defined(__CYGWIN__)
+#error "cygwin have no SOCKET"
+#endif
         SOCKET foo;
         return foo; /* Avoids warning */
     ],
@@ -168,6 +169,9 @@ AC_CACHE_VAL(ac_cv_timeval_header,
 [
     AC_TRY_COMPILE([#include <winsock2.h>],
     [
+#if defined(__CYGWIN__)
+#error "cygwin have no winsock"
+#endif
         timeval foo;
         return &foo == 0; /* Avoids warning */
     ],
@@ -358,6 +362,9 @@ dnl Test for closesocket
 AC_MSG_CHECKING([for closesocket])
 AC_TRY_COMPILE([#include <ws2tcpip.h>],
 [
+#if defined(__CYGWIN__)
+#error "cygwin have no closesocket"
+#endif
     closesocket(0);
 ],
 [
--- origsrc/skstream-0.3.7/skstream/Makefile.am	2010-07-09 06:21:15.000000000 +0900
+++ src/skstream-0.3.7/skstream/Makefile.am	2010-10-31 13:00:29.846903500 +0900
@@ -1,7 +1,7 @@
 INCLUDES = -I$(top_srcdir)
 
 lib_LTLIBRARIES = libskstream-0.3.la
-libskstream_0_3_la_LDFLAGS = -version-info @SKSTREAM_VERSION_INFO@
+libskstream_0_3_la_LDFLAGS = -version-info @SKSTREAM_VERSION_INFO@ -no-undefined
 
 libskstream_0_3_la_SOURCES = sksocket.cpp skstream.cpp skserver.cpp skpoll.cpp
 
--- origsrc/skstream-0.3.7/test/childskstreamtest.h	2010-07-09 06:21:15.000000000 +0900
+++ src/skstream-0.3.7/test/childskstreamtest.h	2010-10-31 13:33:57.003153500 +0900
@@ -257,9 +257,11 @@ class rawskstreamtest : public CppUnit::
             raw_socket_stream skstream;
             CPPUNIT_ASSERT(skstream);
 
+#if !defined(__CYGWIN__)
             CPPUNIT_ASSERT_MESSAGE(
                     "Raw only works on GNU/Linux and you must be root", 
                     skstream.is_open());
+#endif
         }
 
         void setUp()
