--- origsrc/skstream-0.3.8/configure.ac	2011-02-14 05:11:14.000000000 +0900
+++ src/skstream-0.3.8/configure.ac	2011-05-12 00:02:34.380453900 +0900
@@ -94,6 +94,9 @@ SKSTREAM_SOCKET_TYPE=int;
 SKSTREAM_SOCKET_HEADER="#include <unistd.h>"
 AC_TRY_COMPILE([#include <ws2tcpip.h>],
 [
+#if defined(__CYGWIN__)
+#error "cygwin have no SOCKET"
+#endif
     SOCKET foo;
     return foo; /* Avoids warning */
 ],
@@ -154,6 +157,9 @@ SKSTREAM_BEOS_HEADER_FLAG=0
 SKSTREAM_WINDOWS_HEADER_FLAG=0
 AC_TRY_COMPILE([#include <winsock2.h>],
 [
+#if defined(__CYGWIN__)
+#error "cygwin have no winsock"
+#endif
     timeval foo;
     return &foo == 0; /* Avoids warning */
 ],
@@ -340,6 +346,9 @@ dnl Test for closesocket
 AC_MSG_CHECKING([for closesocket])
 AC_TRY_COMPILE([#include <ws2tcpip.h>],
 [
+#if defined(__CYGWIN__)
+#error "cygwin have no closesocket"
+#endif
     closesocket(0);
 ],
 [
--- origsrc/skstream-0.3.8/skstream/Makefile.am	2011-02-14 05:02:11.000000000 +0900
+++ src/skstream-0.3.8/skstream/Makefile.am	2011-05-12 00:02:50.243263500 +0900
@@ -1,7 +1,7 @@
 AM_CPPFLAGS = -I$(top_srcdir) -I$(top_builddir)
 
 lib_LTLIBRARIES = libskstream-0.3.la
-libskstream_0_3_la_LDFLAGS = -version-info @SKSTREAM_VERSION_INFO@
+libskstream_0_3_la_LDFLAGS = -version-info @SKSTREAM_VERSION_INFO@ -no-undefined
 
 libskstream_0_3_la_SOURCES = sksocket.cpp skstream.cpp skserver.cpp skpoll.cpp
 
