--- origsrc/librcsc-rc2024/CMakeLists.txt	2024-07-13 09:43:45.000000000 +0900
+++ src/librcsc-rc2024/CMakeLists.txt	2024-09-02 14:24:06.646004900 +0900
@@ -16,7 +16,6 @@ set(CMAKE_CXX_STANDARD_REQUIRED True)
 if(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE "Release" CACHE STRING "" FORCE)
 endif()
-set(CMAKE_CXX_FLAGS "-W -Wall")
 
 # install destination
 if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
@@ -36,6 +35,7 @@ check_include_file_cxx("netdb.h" HAVE_NE
 check_include_file_cxx("sys/socket.h" HAVE_SYS_SOCKET_H)
 check_include_file_cxx("sys/time.h" HAVE_SYS_TIME_H)
 check_include_file_cxx("unistd.h" HAVE_UNISTD_H)
+check_include_file_cxx("netinet/in.h" HAVE_NETINET_IN_H)
 
 # check funcs
 include(CheckCXXSymbolExists)
--- origsrc/librcsc-rc2024/librcsc-config.in	2024-07-13 09:43:45.000000000 +0900
+++ src/librcsc-rc2024/librcsc-config.in	2024-09-02 14:52:34.960141300 +0900
@@ -37,7 +37,7 @@ if test ${libdir} != /usr/lib ; then
   libsL=-L${libdir}
 fi
 
-libsl="-lrcsc_agent -lrcsc_clang -lrcsc_ann -lrcsc_net -lrcsc_color -lrcsc_time -lrcsc_param -lrcsc_rcg -lrcsc_gz -lrcsc_geom -lz -lm"
+libsl="-lrcsc -lz -lm"
 
 while test $# -gt 0; do
   case "$1" in
--- origsrc/librcsc-rc2024/rcsc/CMakeLists.txt	2024-07-13 09:43:45.000000000 +0900
+++ src/librcsc-rc2024/rcsc/CMakeLists.txt	2024-09-02 14:40:16.801165500 +0900
@@ -50,6 +50,8 @@ target_include_directories(rcsc
 #  $<INSTALL_INTERFACE:include>
   )
 
+target_link_libraries(rcsc ${Boost_SYSTEM_LIBRARY} ${ZLIB_LIBRARY})
+
 set_target_properties(rcsc PROPERTIES
   VERSION ${LIBRCSC_BUILDVERSION}
   SOVERSION ${LIBRCSC_SOVERSION}
--- origsrc/librcsc-rc2024/rcsc/net/host_address.h	2024-07-13 09:43:45.000000000 +0900
+++ src/librcsc-rc2024/rcsc/net/host_address.h	2024-09-02 13:27:34.212360300 +0900
@@ -36,7 +36,7 @@
 #include <string>
 #include <cstdint>
 
-struct sockaddr_in;
+#include <netinet/in.h>
 
 namespace rcsc {
 
