--- origsrc/librcsc-4.1.0/rcsc/Makefile.am	2011-05-11 18:19:18.000000000 +0900
+++ src/librcsc-4.1.0/rcsc/Makefile.am	2014-09-18 21:36:05.709177000 +0900
@@ -21,7 +21,14 @@ SUBDIRS = \
 lib_LTLIBRARIES = librcsc_agent.la
 
 librcsc_agent_la_SOURCES =
-librcsc_agent_la_LIBADD = \
+librcsc_agent_la_LIBADD = -lstdc++ \
+	ann/librcsc_ann.la \
+	geom/librcsc_geom.la \
+	gz/librcsc_gz.la \
+	net/librcsc_net.la \
+	param/librcsc_param.la \
+	rcg/librcsc_rcg.la \
+	time/librcsc_time.la \
 	util/librcsc_util.la \
 	formation/librcsc_formation.la \
 	common/librcsc_common.la \
@@ -33,7 +40,7 @@ librcsc_agent_la_LIBADD = \
 
 ##	action/librcsc_action_obsolete.la
 
-librcsc_agent_la_LDFLAGS = -version-info 7:0:0
+librcsc_agent_la_LDFLAGS = -no-undefined -version-info 7:0:0
 #libXXXX_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 #    1. Start with version information of `0:0:0' for each libtool library.
 #
--- origsrc/librcsc-4.1.0/rcsc/ann/Makefile.am	2011-04-01 18:58:18.000000000 +0900
+++ src/librcsc-4.1.0/rcsc/ann/Makefile.am	2014-09-18 21:07:57.790382800 +0900
@@ -14,7 +14,7 @@ librcsc_anninclude_HEADERS = \
 	ngnet.h \
 	rbf.h
 
-librcsc_ann_la_LDFLAGS = -version-info 1:1:0
+librcsc_ann_la_LDFLAGS = -no-undefined -version-info 1:1:0
 #libXXXX_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 #    1. Start with version information of `0:0:0' for each libtool library.
 #
--- origsrc/librcsc-4.1.0/rcsc/geom/Makefile.am	2011-05-11 18:19:36.000000000 +0900
+++ src/librcsc-4.1.0/rcsc/geom/Makefile.am	2014-09-18 21:08:15.993507800 +0900
@@ -61,7 +61,7 @@ librcsc_geom_la_LIBADD = \
 #	tetgen/librcsc_geom_tetgen.la
 
 
-librcsc_geom_la_LDFLAGS = -version-info 7:0:0
+librcsc_geom_la_LDFLAGS = -no-undefined -version-info 7:0:0
 #libXXXX_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 #    1. Start with version information of `0:0:0' for each libtool library.
 #
--- origsrc/librcsc-4.1.0/rcsc/gz/Makefile.am	2011-04-01 18:58:12.000000000 +0900
+++ src/librcsc-4.1.0/rcsc/gz/Makefile.am	2014-09-18 21:08:29.915382800 +0900
@@ -15,7 +15,7 @@ librcsc_gzinclude_HEADERS = \
 	gzfstream.h \
 	gzfilterstream.h
 
-librcsc_gz_la_LDFLAGS = -version-info 0:2:0
+librcsc_gz_la_LDFLAGS = -no-undefined -version-info 0:2:0
 ##libXXXX_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 ##    1. Start with version information of `0:0:0' for each libtool library.
 ##
--- origsrc/librcsc-4.1.0/rcsc/net/Makefile.am	2011-04-01 18:58:12.000000000 +0900
+++ src/librcsc-4.1.0/rcsc/net/Makefile.am	2014-09-18 21:08:55.149757800 +0900
@@ -15,7 +15,7 @@ librcsc_netinclude_HEADERS = \
 	udp_socket.h \
 	tcp_socket.h
 
-librcsc_net_la_LDFLAGS = -version-info 0:1:0
+librcsc_net_la_LDFLAGS = -no-undefined -version-info 0:1:0
 #libXXXX_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 #    1. Start with version information of `0:0:0' for each libtool library.
 #
--- origsrc/librcsc-4.1.0/rcsc/param/Makefile.am	2011-04-01 18:58:17.000000000 +0900
+++ src/librcsc-4.1.0/rcsc/param/Makefile.am	2014-09-18 21:09:10.337257800 +0900
@@ -18,7 +18,7 @@ librcsc_paraminclude_HEADERS = \
 	param_parser.h \
 	rcss_param_parser.h
 
-librcsc_param_la_LDFLAGS = -version-info 3:0:0
+librcsc_param_la_LDFLAGS = -no-undefined -version-info 3:0:0
 #libXXXX_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 #    1. Start with version information of `0:0:0' for each libtool library.
 #
--- origsrc/librcsc-4.1.0/rcsc/rcg/Makefile.am	2011-04-01 18:58:18.000000000 +0900
+++ src/librcsc-4.1.0/rcsc/rcg/Makefile.am	2014-09-18 21:09:30.102882800 +0900
@@ -40,7 +40,7 @@ librcsc_rcginclude_HEADERS = \
 	types.h \
 	util.h
 
-librcsc_rcg_la_LDFLAGS = -version-info 5:0:0
+librcsc_rcg_la_LDFLAGS = -no-undefined -version-info 5:0:0
 #libXXXX_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 #		 1. Start with version information of `0:0:0' for each libtool library.
 #
--- origsrc/librcsc-4.1.0/rcsc/time/Makefile.am	2011-04-01 18:58:17.000000000 +0900
+++ src/librcsc-4.1.0/rcsc/time/Makefile.am	2014-09-18 21:09:44.946632800 +0900
@@ -10,7 +10,7 @@ librcsc_timeincludedir = $(includedir)/r
 librcsc_timeinclude_HEADERS = \
 	timer.h
 
-librcsc_time_la_LDFLAGS = -version-info 0:0:0
+librcsc_time_la_LDFLAGS = -no-undefined -version-info 0:0:0
 #libXXXX_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 #    1. Start with version information of `0:0:0' for each libtool library.
 #
