--- origsrc/tci/Makefile.am	2021-03-15 18:15:57.000000000 +0900
+++ src/tci/Makefile.am	2021-03-15 18:46:25.633103900 +0900
@@ -13,6 +13,7 @@ else
 lib_libtci_la_SOURCES += tci/communicator.c \
 						 tci/task_set.c
 endif
+lib_libtci_la_LDFLAGS = -no-undefined
 
 nobase_include_HEADERS = tci.h \
 						 tci/tci_config.h \
@@ -35,3 +36,6 @@ VPATH += $(srcdir)
 
 AM_CPPFLAGS = -I$(srcdir) -I$(srcdir)/tci -I. -Itci
 ACLOCAL_AMFLAGS = -I m4
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = tci.pc
--- origsrc/tci/configure.ac	2021-03-15 18:15:57.000000000 +0900
+++ src/tci/configure.ac	2021-03-15 18:45:48.154153200 +0900
@@ -453,5 +453,5 @@ if test x$USE_TBB_THREADS = x1 || \
     AX_CXX_COMPILE_STDCXX_11([noext])
 fi
 
-AC_CONFIG_FILES([Makefile tci/tci_config.h])
+AC_CONFIG_FILES([tci.pc Makefile tci/tci_config.h])
 AC_OUTPUT
--- origsrc/tci/tci.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/tci/tci.pc.in	2021-03-15 18:37:18.477186100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: tci
+Description: Thread Control Interface
+Version: @VERSION@
+Libs: -L${libdir} -ltci
+Cflags: -I${includedir}
