--- origsrc/rpc2-2.10/rpc2-src/Makefile.am	2010-03-23 04:18:17.000000000 +0900
+++ src/rpc2-2.10/rpc2-src/Makefile.am	2012-03-30 14:24:02.138071200 +0900
@@ -15,11 +15,11 @@ librpc2_la_SOURCES = rpc2a.c rpc2b.c sl.
     lists.c ct.c debug.c conn.c cbuf.c ffs.c host.c multi1.c multi2.c \
     multi3.c mkcall.c stats.c errors.c cbuf.h trace.h dllist.c rpc2_addrinfo.c \
     compat.c lua.c delay.c
-librpc2_la_LDFLAGS = $(LIBTOOL_LDFLAGS)
+librpc2_la_LDFLAGS = $(LIBTOOL_LDFLAGS) -no-undefined
 librpc2_la_LIBADD = $(top_builddir)/secure/libsecure.la $(LUA_LIBS) $(LWP_LIBS)
 
 libse_la_SOURCES = sftp1.c sftp2.c sftp3.c sftp4.c sftp5.c sftp6.c
-libse_la_LDFLAGS = $(LIBTOOL_LDFLAGS)
+libse_la_LDFLAGS = $(LIBTOOL_LDFLAGS) -no-undefined
 libse_la_LIBADD = librpc2.la $(LWP_LIBS)
 
 MAINTAINERCLEANFILES = Makefile.in
