--- origsrc/libhtp-0.5.22/htp/Makefile.am	2016-09-07 01:15:03.000000000 +0900
+++ src/libhtp-0.5.22/htp/Makefile.am	2016-10-23 12:09:40.160949000 +0900
@@ -22,8 +22,9 @@ AM_CFLAGS = -I$(top_srcdir) -I$(top_buil
 
 noinst_LTLIBRARIES = libhtp-c.la
 libhtp_c_la_SOURCES = $(h_sources) $(h_sources_private) $(c_sources)
+libhtp_c_la_LIBADD = $(LTLIBICONV)
 
 lib_LTLIBRARIES = libhtp.la
 libhtp_la_SOURCES =
 libhtp_la_LIBADD = libhtp-c.la
-libhtp_la_LDFLAGS = -version-info $(GENERIC_LIBRARY_VERSION) -release $(GENERIC_RELEASE)
+libhtp_la_LDFLAGS = -no-undefined -version-info $(GENERIC_LIBRARY_VERSION) -release $(GENERIC_RELEASE)
