--- origsrc/libhtp-0.5.29/htp/Makefile.am	2018-12-21 02:55:58.000000000 +0900
+++ src/libhtp-0.5.29/htp/Makefile.am	2019-01-24 11:33:30.520436300 +0900
@@ -25,5 +25,5 @@ libhtp_c_la_SOURCES = $(h_sources) $(h_s
 
 lib_LTLIBRARIES = libhtp.la
 libhtp_la_SOURCES =
-libhtp_la_LIBADD = libhtp-c.la
-libhtp_la_LDFLAGS = -version-info $(GENERIC_LIBRARY_VERSION)
+libhtp_la_LIBADD = libhtp-c.la $(LTLIBICONV)
+libhtp_la_LDFLAGS = -no-undefined -version-info $(GENERIC_LIBRARY_VERSION)
