--- origsrc/libhtp/configure.ac	2011-05-14 17:52:03.000000000 +0900
+++ src/libhtp/configure.ac	2011-05-14 17:54:07.868690000 +0900
@@ -15,7 +15,7 @@ GENERIC_LIBRARY_NAME=htp
 
 GENERIC_MAJOR_VERSION=0
 GENERIC_MINOR_VERSION=2
-GENERIC_MICRO_VERSION=X
+GENERIC_MICRO_VERSION=5
 
 # API version (often = GENERIC_MAJOR_VERSION.GENERIC_MINOR_VERSION)
 GENERIC_API_VERSION=1.0
@@ -67,7 +67,7 @@ dnl Checks for programs.
 dnl -----------------------------------------------
 
 AC_PROG_CC
-AM_PROG_LIBTOOL
+LT_INIT
 AM_SANITY_CHECK
 
 
--- origsrc/libhtp/htp/Makefile.am	2011-05-14 17:52:03.000000000 +0900
+++ src/libhtp/htp/Makefile.am	2011-05-14 17:53:33.018578000 +0900
@@ -10,5 +10,5 @@ AM_CFLAGS = -D_GNU_SOURCE -g -O2 -Wall -
 
 lib_LTLIBRARIES= libhtp.la
 libhtp_la_SOURCES= $(h_sources) $(c_sources)
-libhtp_la_LDFLAGS= -version-info $(GENERIC_LIBRARY_VERSION) -release $(GENERIC_RELEASE)
+libhtp_la_LDFLAGS= -version-info $(GENERIC_LIBRARY_VERSION) -release $(GENERIC_RELEASE) -no-undefined
 
--- origsrc/libhtp/test/Makefile.am	2011-05-14 17:52:04.000000000 +0900
+++ src/libhtp/test/Makefile.am	2011-05-14 18:05:36.208474000 +0900
@@ -2,7 +2,7 @@
 check_PROGRAMS = main
 main_SOURCES = main.c test.c test.h  test-tcpick.c
 
-LDADD = ../htp/.libs/libhtp.a -lz
+LDADD = $(top_builddir)/htp/libhtp.la -lz
 AM_CFLAGS = -g -O2
 
 
