--- origsrc/randomname-0.1.0/meson.build	2023-02-22 06:12:11.000000000 +0900
+++ src/randomname-0.1.0/meson.build	2025-04-04 10:14:20.538487600 +0900
@@ -11,7 +11,7 @@ subdir('src')
 
 prefix = get_option('prefix')
 datadir = join_paths(prefix, get_option('datadir'))
-docdir = join_paths(datadir, 'doc', package + '-' + version)
+docdir = join_paths(datadir, 'doc', package)
 
 docs = [
   'AUTHORS',
--- origsrc/randomname-0.1.0/src/meson.build	2023-02-22 06:12:11.000000000 +0900
+++ src/randomname-0.1.0/src/meson.build	2025-04-04 09:33:01.252948800 +0900
@@ -1,4 +1,4 @@
-librandomname = library('librandomname', 'librandomname.c', install: true, version: version, soversion: soversion, name_prefix: '')
+librandomname = library('randomname', 'librandomname.c', install: true, version: version, soversion: soversion)
 executable('randomname', 'randomname.c', link_with: librandomname, install: true)
 
 install_headers('librandomname.h')
