--- origsrc/libtess2-1.0.2/meson.build	2024-11-29 09:26:34.597643300 +0900
+++ src/libtess2-1.0.2/meson.build	2024-11-29 09:28:12.607806900 +0900
@@ -1,9 +1,9 @@
 project('libtess2', 'c',
-version : '0.12.0',
+version : '1.0.2',
 default_options : [
   'buildtype=release',
   'optimization=3',
-  'default_library=static',
+  'default_library=shared',
   'prefer_static=true',
   'b_ndebug=if-release',
   'c_args=-pipe',
@@ -31,7 +31,8 @@ sources = files(
 lib = library('tess2',
   sources,
   include_directories: inc,
-  install: true)
+  install: true,
+  soversion: 1)
 
 pkg = import('pkgconfig')
 pkg.generate(lib)
