--- origsrc/LASzip-3.4.4/src/lasmessage.cpp	2024-04-17 15:32:41.000000000 +0900
+++ src/LASzip-3.4.4/src/lasmessage.cpp	2024-04-26 19:27:38.175387000 +0900
@@ -139,8 +139,8 @@ void las_default_message_handler(LAS_MES
 	if (!prefix.empty())
 	{
 		format_message(message, (unsigned)prefix.size());
-		fprintf(stderr, prefix.c_str());
-		fprintf(stderr, message.c_str());
+		fprintf(stderr, "%s", prefix.c_str());
+		fprintf(stderr, "%s", message.c_str());
 	}
 	else
 	{
