--- origsrc/LASzip-3.2.2/cmake/unix_compiler_options.cmake	2018-03-27 21:31:13.000000000 +0900
+++ src/LASzip-3.2.2/cmake/unix_compiler_options.cmake	2018-08-25 13:49:21.235751900 +0900
@@ -2,10 +2,12 @@ set(LASZIP_COMMON_CXX_FLAGS "-Wextra -Wa
 )
 
 if (${CMAKE_CXX_COMPILER_ID} MATCHES "GNU")
-    if (${CMAKE_CXX_COMPILER_VERSION} VERSION_LESS 4.7)
-       set(CXX_STANDARD "-std=c++0x")
-    else()
-       set(CXX_STANDARD "-std=c++11")
+    if (NOT CYGWIN)
+        if (${CMAKE_CXX_COMPILER_VERSION} VERSION_LESS 4.7)
+            set(CXX_STANDARD "-std=c++0x")
+        else()
+            set(CXX_STANDARD "-std=c++11")
+        endif()
     endif()
     if (${CMAKE_CXX_COMPILER_VERSION} VERSION_GREATER 4.6)
         set(LASZIP_NO_AS_NEEDED_START "-Wl,--no-as-needed")
