--- origsrc/laszip-src-2.2.0/Makefile.am	2013-08-05 23:20:31.000000000 +0900
+++ src/laszip-src-2.2.0/Makefile.am	2013-10-11 12:22:55.520455700 +0900
@@ -3,6 +3,6 @@ SUBDIRS = src . include/laszip tools
 lib_LTLIBRARIES = liblaszip.la
 
 liblaszip_la_SOURCES =
-liblaszip_la_LIBADD = src/liblibrary.la 
+liblaszip_la_LIBADD = src/liblibrary.la -lstdc++
 
-liblaszip_la_LDFLAGS = -version-info 6:0:0
+liblaszip_la_LDFLAGS = -no-undefined -version-info 6:0:0
--- origsrc/laszip-src-2.2.0/configure.ac	2013-08-05 23:19:56.000000000 +0900
+++ src/laszip-src-2.2.0/configure.ac	2013-10-11 12:19:53.209908900 +0900
@@ -22,7 +22,7 @@ dnl ####################################
 m4_define([debug_default],[no])
 
 CFLAGS="-Wall -Wno-long-long -pedantic $CFLAGS"
-CXXFLAGS="-Wall -Wno-long-long -pedantic -std=c++98 $CXXFLAGS"
+CXXFLAGS="-Wall -Wno-long-long -pedantic -std=gnu++98 $CXXFLAGS"
 
 dnl #########################################################################
 dnl Checks for programs.
--- origsrc/laszip-src-2.2.0/src/Makefile.am	2013-08-06 02:09:37.000000000 +0900
+++ src/laszip-src-2.2.0/src/Makefile.am	2013-10-11 12:16:47.083932300 +0900
@@ -1,6 +1,6 @@
 SUBDIRS = . 
 
-AM_CPPFLAGS = -I. -I../include -I../include/laszip
+AM_CPPFLAGS = -I. -I../include -I$(top_srcdir)/include/laszip
 noinst_LTLIBRARIES = liblibrary.la 
 
 liblibrary_la_SOURCES =				arithmeticmodel.cpp \
--- origsrc/laszip-src-2.2.0/tools/Makefile.am	2013-08-05 22:44:28.000000000 +0900
+++ src/laszip-src-2.2.0/tools/Makefile.am	2013-10-11 12:24:04.896432300 +0900
@@ -1,5 +1,5 @@
 
-AM_CPPFLAGS = -I../include -I../include/laszip
+AM_CPPFLAGS = -I../include -I$(top_srcdir)/include/laszip
 LDADD = ../liblaszip.la 
 
 laszippertest_SOURCES = laszippertest.cpp
