--- origsrc/forge-1.0.4/CMakeModules/FindOpenGL.cmake	2019-03-02 20:06:13.000000000 +0900
+++ src/forge-1.0.4/CMakeModules/FindOpenGL.cmake	2019-05-26 18:23:34.926127500 +0900
@@ -55,9 +55,9 @@ if (CYGWIN)
   find_path(OPENGL_INCLUDE_DIR GL/gl.h )
   list(APPEND _OpenGL_REQUIRED_VARS OPENGL_INCLUDE_DIR)
 
-  find_library(OPENGL_gl_LIBRARY opengl32 )
+  find_library(OPENGL_gl_LIBRARY GL )
 
-  find_library(OPENGL_glu_LIBRARY glu32 )
+  find_library(OPENGL_glu_LIBRARY GLU )
 
 elseif (WIN32)
 
--- origsrc/forge-1.0.4/src/backend/opengl/CMakeLists.txt	2019-03-02 20:06:13.000000000 +0900
+++ src/forge-1.0.4/src/backend/opengl/CMakeLists.txt	2019-05-26 18:17:36.668611900 +0900
@@ -99,7 +99,8 @@ target_link_libraries(${BackendTarget}
 if (FG_ENABLE_HUNTER)
     target_link_libraries(${BackendTarget} PRIVATE freetype::freetype)
 else ()
-    target_link_libraries(${BackendTarget} PRIVATE Freetype::Freetype)
+    include_directories(${FREETYPE_INCLUDE_DIR_ft2build})
+    target_link_libraries(${BackendTarget} PRIVATE freetype)
 endif ()
 
 if(UNIX)
