--- origsrc/rcssmonitor-rcssmonitor-19.0.1/CMakeLists.txt	2024-04-27 15:49:18.000000000 +0900
+++ src/rcssmonitor-rcssmonitor-19.0.1/CMakeLists.txt	2025-03-16 08:17:30.198437000 +0900
@@ -43,6 +43,9 @@ if(ZLIB_FOUND)
   set(HAVE_LIBZ TRUE)
 endif()
 
+# check simdjson
+find_package(pkgconfig)
+pkg_check_modules(SIMDJSON simdjson)
 
 # generate config.h
 add_definitions(-DHAVE_CONFIG_H)
@@ -54,3 +57,10 @@ configure_file(
 #
 add_subdirectory(rcss)
 add_subdirectory(src)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/rcssmonitor.pc.in ${CMAKE_BINARY_DIR}/rcssmonitor.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/rcssmonitor.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/rcssmonitor-rcssmonitor-19.0.1/rcss/rcg/CMakeLists.txt	2024-04-27 15:49:18.000000000 +0900
+++ src/rcssmonitor-rcssmonitor-19.0.1/rcss/rcg/CMakeLists.txt	2025-03-16 08:17:30.198437000 +0900
@@ -1,6 +1,9 @@
+if (NOT SIMDJSON_FOUND)
+  set(SIMDJSON_SOURCE "simdjson/simdjson.cpp")
+endif()
 
 add_library(rcssrcg SHARED
-  simdjson/simdjson.cpp
+  ${SIMDJSON_SOURCE}
   handler.cpp
   parser.cpp
   parser_v1.cpp
@@ -12,6 +15,10 @@ add_library(rcssrcg SHARED
   types.cpp
   )
 
+if (SIMDJSON_FOUND)
+  target_link_libraries(rcssrcg ${SIMDJSON_LIBRARIES})
+endif()
+
 target_include_directories(rcssrcg
   PUBLIC
 #  ${Boost_INCLUDE_DIRS}
--- origsrc/rcssmonitor-rcssmonitor-19.0.1/rcssmonitor.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/rcssmonitor-rcssmonitor-19.0.1/rcssmonitor.pc.in	2025-03-16 08:17:30.207988400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: rcssmonitor
+Description: The RoboCup Soccer Simulator Monitor
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lrcssrcg
+Cflags: -I${includedir}
