--- origsrc/rcssmonitor-rcssmonitor-18.0.0/CMakeLists.txt	2023-03-08 15:11:27.000000000 +0900
+++ src/rcssmonitor-rcssmonitor-18.0.0/CMakeLists.txt	2023-11-08 15:56:43.874823300 +0900
@@ -54,3 +54,10 @@ configure_file(
 #
 add_subdirectory(rcss)
 add_subdirectory(src)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/rcssmonitor.pc.in ${CMAKE_BINARY_DIR}/rcssmonitor.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/rcssmonitor.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/rcssmonitor-rcssmonitor-18.0.0/rcssmonitor.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/rcssmonitor-rcssmonitor-18.0.0/rcssmonitor.pc.in	2023-11-08 15:56:22.592192900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: rcssmonitor
+Description: The RoboCup Soccer Simulator Monitor
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lrcssrcg
+Cflags: -I${includedir}
