--- origsrc/movit-1.7.1/Makefile.in	2023-08-02 05:15:00.000000000 +0900
+++ src/movit-1.7.1/Makefile.in	2024-10-16 11:56:24.503755700 +0900
@@ -27,7 +27,7 @@ CXXFLAGS += -DHAVE_BENCHMARK
 endif
 LDFLAGS=@LDFLAGS@
 LDLIBS=@epoxy_LIBS@ @FFTW3_LIBS@ -lpthread
-TEST_LDLIBS=@epoxy_LIBS@ @SDL2_LIBS@ @benchmark_LIBS@ -lpthread
+TEST_LDLIBS=@epoxy_LIBS@ @SDL2_LIBS@ @benchmark_LIBS@ -lpthread @gtest_LIBS@
 DEMO_LDLIBS=@SDL2_image_LIBS@ -lrt -lpthread @libpng_LIBS@ @FFTW3_LIBS@
 SHELL=@SHELL@
 LIBTOOL=@LIBTOOL@ --tag=CXX
@@ -100,12 +100,7 @@ endif
 
 # Google Test and other test library functions.
 OWN_TEST_OBJS = gtest_sdl_main.o test_util.o
-TEST_OBJS = gtest-all.o $(OWN_TEST_OBJS)
-
-gtest-all.o: $(GTEST_DIR)/src/gtest-all.cc
-	$(CXX) -MMD $(CPPFLAGS) -I$(GTEST_DIR) $(CXXFLAGS) -c $< -o $@
-gtest_sdl_main.o: gtest_sdl_main.cpp
-	$(CXX) -MMD $(CPPFLAGS) -I$(GTEST_DIR) $(CXXFLAGS) -c $< -o $@
+TEST_OBJS = $(OWN_TEST_OBJS)
 
 # Unit tests.
 $(TESTS): %: %.o $(TEST_OBJS) libmovit.la
@@ -120,7 +115,7 @@ demo: libmovit.la $(DEMO_OBJS)
 
 # The library itself.
 libmovit.la: $(LIB_OBJS:.o=.lo)
-	$(LIBTOOL) --mode=link $(CXX) $(LDFLAGS) -rpath $(libdir) -version-info $(movit_ltversion) -o $@ $^ $(LDLIBS)
+	$(LIBTOOL) --mode=link $(CXX) $(LDFLAGS) -rpath $(libdir) -version-info $(movit_ltversion) -o $@ $^ $(LDLIBS) -no-undefined
 
 %.lo: %.cpp
 	$(LIBTOOL) --mode=compile $(CXX) -MMD -MP $(CPPFLAGS) $(CXXFLAGS) -o $@ -c $<
--- origsrc/movit-1.7.1/configure.ac	2024-10-16 11:56:23.003760200 +0900
+++ src/movit-1.7.1/configure.ac	2024-10-16 11:56:24.519381500 +0900
@@ -24,6 +24,7 @@ PKG_CHECK_MODULES([libpng], [libpng], []
 
 # This is only needed for microbenchmarks, so optional.
 PKG_CHECK_MODULES([benchmark], [benchmark], [with_benchmark=yes], [with_benchmark=no; AC_MSG_WARN([Google microbenchmark framework not found, microbenchmarks will not be built])])
+PKG_CHECK_MODULES([gtest], [gtest])
 
 AC_SUBST([with_demo_app])
 AC_SUBST([with_benchmark])
--- origsrc/movit-1.7.1/movit.pc.in	2023-08-02 05:15:00.000000000 +0900
+++ src/movit-1.7.1/movit.pc.in	2024-10-16 11:56:24.519381500 +0900
@@ -10,7 +10,7 @@ shaderdir=@datadir@/movit
 
 Name: movit
 Description: Movit is a library for high-quality, high-performance video filters.
-Version: git
+Version: @PACKAGE_VERSION@
 Requires: epoxy eigen3
 Requires.private: fftw3
 Conflicts:
