--- origsrc/test-drive-0.4.0/meson.build	2021-11-14 23:51:34.000000000 +0900
+++ src/test-drive-0.4.0/meson.build	2022-05-12 14:49:06.895227200 +0900
@@ -56,18 +56,22 @@ testdrive_lic = files(
 )
 
 if install
-  # Distribute the license files in share/licenses/<name>
-  install_data(
-    testdrive_lic,
-    install_dir: get_option('datadir')/'licenses'/meson.project_name()
-  )
-
-  module_id = meson.project_name() / fc_id + '-' + fc.version()
-  meson.add_install_script(
-    'config'/'install-mod.py',
-    get_option('includedir') / module_id,
-  )
+  if host_machine.system() != 'cygwin'
+    # Distribute the license files in share/licenses/<name>
+    install_data(
+      testdrive_lic,
+      install_dir: get_option('datadir')/'licenses'/meson.project_name()
+    )
 
+    module_id = meson.project_name() / fc_id + '-' + fc.version()
+    meson.add_install_script(
+      'config'/'install-mod.py',
+      get_option('includedir') / module_id,
+    )
+  else
+    module_id = null
+    meson.add_install_script('config'/'install-mod.py')
+  endif
   pkg = import('pkgconfig')
   pkg.generate(
     testdrive_lib,
