--- origsrc/test-drive-0.5.0/meson.build	2024-08-30 17:59:20.000000000 +0900
+++ src/test-drive-0.5.0/meson.build	2025-09-09 14:48:59.036425300 +0900
@@ -56,18 +56,22 @@ testdrive_lic = files(
 )
 
 if install
-  # Distribute the license files in share/licenses/<name>
-  install_data(
-    testdrive_lic,
-    install_dir: get_option('datadir')/'licenses'/meson.project_name()
-  )
-
-  module_id = meson.project_name() / fc_id + '-' + fc.version()
-  meson.add_install_script(
-    'config'/'install-mod.py',
-    get_option('includedir') / module_id,
-  )
+  if host_machine.system() != 'cygwin'
+    # Distribute the license files in share/licenses/<name>
+    install_data(
+      testdrive_lic,
+      install_dir: get_option('datadir')/'licenses'/meson.project_name()
+    )
 
+    module_id = meson.project_name() / fc_id + '-' + fc.version()
+    meson.add_install_script(
+      'config'/'install-mod.py',
+      get_option('includedir') / module_id,
+    )
+  else
+    module_id = ''
+    meson.add_install_script('config'/'install-mod.py')
+  endif
   pkg = import('pkgconfig')
   pkg.generate(
     testdrive_lib,
