--- origsrc/flare-engine/CMakeLists.txt	2012-10-02 21:11:52.000000000 +0900
+++ src/flare-engine/CMakeLists.txt	2012-10-02 22:40:19.189485300 +0900
@@ -2,7 +2,7 @@ Project (Flare)
 cmake_minimum_required (VERSION 2.6)
 
 Set (PACKAGE "FLARE")
-Set (VERSION "0.15")
+Set (VERSION "0.16")
 
 
 # Default definitions
@@ -24,7 +24,7 @@ elseif(CMAKE_BUILD_TYPE STREQUAL "Debug"
   set(CMAKE_MODULE_LINKER_FLAGS_DEBUG "-pg")
 endif()
 
-set(BINDIR  "games"             CACHE STRING "Directory from CMAKE_INSTALL_PREFIX where game executable will be installed.")
+set(BINDIR  "bin"               CACHE STRING "Directory from CMAKE_INSTALL_PREFIX where game executable will be installed.")
 set(DATADIR "share/games/flare" CACHE STRING "Directory from CMAKE_INSTALL_PREFIX where game data files will be installed.")
 
 add_definitions(-DDATA_INSTALL_DIR="${CMAKE_INSTALL_PREFIX}/${DATADIR}")
@@ -161,8 +161,8 @@ configure_file("${CMAKE_CURRENT_SOURCE_D
 
 
 # installing to the proper places
-install(PROGRAMS
-  ${CMAKE_CURRENT_BINARY_DIR}/flare
+install(TARGETS
+  flare
   DESTINATION ${CMAKE_INSTALL_PREFIX}/${BINDIR})
 install(DIRECTORY
   "${CMAKE_CURRENT_SOURCE_DIR}/mods"
@@ -170,7 +170,7 @@ install(DIRECTORY
 install(FILES
   "${CMAKE_CURRENT_BINARY_DIR}/flare.desktop"
   DESTINATION ${CMAKE_INSTALL_PREFIX}/share/applications)
-install(FILES
-  "${CMAKE_CURRENT_SOURCE_DIR}/art_src/logo/flare_logo.svg"
-  DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/scalable/apps
-  RENAME flare.svg)
+#install(FILES
+#  "${CMAKE_CURRENT_SOURCE_DIR}/art_src/logo/flare_logo.svg"
+#  DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/scalable/apps
+#  RENAME flare.svg)
