diff --strip-trailing-cr -uprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.log -x config.status -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/flare-engine/CMakeLists.txt src/flare-engine/CMakeLists.txt
--- origsrc/flare-engine/CMakeLists.txt	2013-04-21 08:35:42.000000000 +0900
+++ src/flare-engine/CMakeLists.txt	2013-04-21 08:38:24.758374700 +0900
@@ -2,7 +2,7 @@ Project (Flare)
 cmake_minimum_required (VERSION 2.6)
 
 Set (PACKAGE "FLARE")
-Set (VERSION "0.17")
+Set (VERSION "0.18")
 
 
 # Default definitions
@@ -38,7 +38,7 @@ else()
   set(CMAKE_MODULE_LINKER_FLAGS "")
 endif()
 
-set(BINDIR  "games"             CACHE STRING "Directory from CMAKE_INSTALL_PREFIX where game executable will be installed.")
+set(BINDIR  "bin"             CACHE STRING "Directory from CMAKE_INSTALL_PREFIX where game executable will be installed.")
 set(DATADIR "share/games/flare" CACHE STRING "Directory from CMAKE_INSTALL_PREFIX where game data files will be installed.")
 
 If(NOT IS_ABSOLUTE "${DATADIR}")
@@ -193,8 +193,8 @@ configure_file("${CMAKE_CURRENT_SOURCE_D
 
 
 # installing to the proper places
-install(PROGRAMS
-  ${CMAKE_CURRENT_BINARY_DIR}/flare
+install(TARGETS
+  flare
   DESTINATION ${BINDIR})
 install(DIRECTORY
   "${CMAKE_CURRENT_SOURCE_DIR}/mods"
