diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libosl/CMakeLists.txt src/libosl/CMakeLists.txt
--- origsrc/libosl/CMakeLists.txt	2015-10-04 01:20:19.000000000 +0900
+++ src/libosl/CMakeLists.txt	2016-03-05 21:05:03.503689400 +0900
@@ -8,7 +8,11 @@ if (UNIX)
   set (CMAKE_CXX_FLAGS "-Wall")
   set (CMAKE_CXX_FLAGS_DEBUG "-O0 -g -DDEBUG")
   set (CMAKE_CXX_FLAGS_RELEASE "-O3 -DNDEBUG")
+  if (CYGWIN)
+    add_definitions(-march=native -fpermissive)
+  else()
   add_definitions(-march=native -DBOOST_FILESYSTEM_VERSION=2) 
+  endif()
 endif()
 
 if(MINGW)
@@ -51,11 +55,24 @@ list(REMOVE_ITEM OSL_SRCS ${EXCLUDE_SRCS
 #message (STATUS "Headers: ${OSL_HEADERS}")
 #message (STATUS "Sources: ${OSL_SRCS}")
 #set (SampleAleriUDF_ALL_SRCS ${SampleAleriUDF_HEADERS} ${SampleAleriUDF_SRCS})
-add_library (osl-static STATIC ${OSL_SRCS})
-SET_TARGET_PROPERTIES(osl-static PROPERTIES OUTPUT_NAME "osl")
-SET_TARGET_PROPERTIES(osl-static PROPERTIES PREFIX "lib")
+option(ENABLE_STATIC "Enable static" OFF)
+if (ENABLE_STATIC)
+  add_library (osl-static STATIC ${OSL_SRCS})
+  SET_TARGET_PROPERTIES(osl-static PROPERTIES OUTPUT_NAME "osl")
+  SET_TARGET_PROPERTIES(osl-static PROPERTIES PREFIX "lib")
+endif()
 if (UNIX)
   add_library (osl SHARED ${OSL_SRCS})
-  target_link_libraries (osl ${CMAKE_THREAD_LIBS_INIT})
+  set_target_properties(osl PROPERTIES SOVERSION 0)
+  target_link_libraries (osl ${CMAKE_THREAD_LIBS_INIT} iconv boost_date_time boost_filesystem boost_iostreams boost_serialization boost_system boost_thread)
+  install(TARGETS osl RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 endif()
+install(DIRECTORY include/osl DESTINATION include)
 
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "0.6.0")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libosl.pc.in ${CMAKE_BINARY_DIR}/libosl.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libosl.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libosl/libosl.pc.in src/libosl/libosl.pc.in
--- origsrc/libosl/libosl.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libosl/libosl.pc.in	2016-03-05 21:05:03.503689400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libosl
+Description: Library for Shogi playing programs
+Version: @VERSION@
+Libs: -L${libdir} -losl
+Cflags: -I${includedir}
