--- origsrc/libosl/CMakeLists.txt	2017-10-06 01:19:30.717200700 +0900
+++ src/libosl/CMakeLists.txt	2017-10-06 09:54:11.355000000 +0900
@@ -62,6 +62,10 @@ elseif (UNIX)
   #add_compile_options(-march=native)
 endif()
 
+if (CYGWIN)
+  add_compile_options(-fpermissive)
+endif()
+
 if (APPLE)
   add_definitions(-DOSL_HOME="/Applications/gpsfish.app/Contents/Resources")
 else()
@@ -106,18 +110,27 @@ list(REMOVE_ITEM OSL_SRCS ${EXCLUDE_SRCS
 #message (STATUS "Sources: ${OSL_SRCS}")
 #set (SampleAleriUDF_ALL_SRCS ${SampleAleriUDF_HEADERS} ${SampleAleriUDF_SRCS})
 
-add_library (osl-static STATIC ${OSL_SRCS})
-SET_TARGET_PROPERTIES(osl-static PROPERTIES OUTPUT_NAME "osl")
-SET_TARGET_PROPERTIES(osl-static PROPERTIES PREFIX "lib")
-if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+option(ENABLE_STATIC "Enable static" OFF)
+if (ENABLE_STATIC)
+  add_library (osl-static STATIC ${OSL_SRCS})
+  SET_TARGET_PROPERTIES(osl-static PROPERTIES OUTPUT_NAME "osl")
+  SET_TARGET_PROPERTIES(osl-static PROPERTIES PREFIX "lib")
+endif()
+if (${CMAKE_SYSTEM_NAME} MATCHES "Linux" OR CYGWIN)
   add_library (osl SHARED ${OSL_SRCS})
-  target_link_libraries (osl ${CMAKE_THREAD_LIBS_INIT})
+  target_link_libraries (osl ${CMAKE_THREAD_LIBS_INIT} boost_date_time boost_filesystem boost_iostreams boost_serialization boost_system iconv)
   set_target_properties(osl PROPERTIES
                             VERSION 1.0.0
                             SOVERSION 1)
 endif()
 
-install(TARGETS osl osl-static
+if(ENABLE_STATIC)
+install(TARGETS osl-static
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+endif()
+install(TARGETS osl
         RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
@@ -133,4 +146,10 @@ install(DIRECTORY "full/osl"
         DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
         FILES_MATCHING PATTERN "*.h" PATTERN "*.tcc")
 
-
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "0.8.0")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libosl.pc.in ${CMAKE_BINARY_DIR}/libosl.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libosl.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libosl/libosl.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libosl/libosl.pc.in	2017-10-06 09:54:03.652417300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libosl
+Description: Library for Shogi playing programs
+Version: @VERSION@
+Libs: -L${libdir} -losl
+Cflags: -I${includedir}
