--- origsrc/gsnmp-0.3.0/gsnmp.m4	2010-02-19 17:35:29.000000000 +0900
+++ src/gsnmp-0.3.0/gsnmp.m4	2011-08-25 08:30:47.087180500 +0900
@@ -1,7 +1,7 @@
 dnl AM_PATH_GSNMP()
 dnl 
 
-AC_DEFUN(AM_PATH_GSNMP,
+AC_DEFUN([AM_PATH_GSNMP],
 [
   PKG_CHECK_MODULES(GSNMP, glib-2.0 >= 2.0.1)
 ])
--- origsrc/gsnmp-0.3.0/src/Makefile.am	2010-02-19 17:35:29.000000000 +0900
+++ src/gsnmp-0.3.0/src/Makefile.am	2011-08-25 08:30:26.351617300 +0900
@@ -24,3 +24,5 @@ gsnmpinclude_HEADERS	= ber.h		\
 			  gsnmp.h
 
 # libgsnmp_la_LDFLAGS	= -version-info @VERSION_LIBTOOL@
+libgsnmp_la_LDFLAGS	= -no-undefined
+libgsnmp_la_LIBADD	= @GLIB_LIBS@ @GNET_LIBS@
