--- origsrc/gsnmp-0.3.0/configure.ac	2010-02-19 17:35:29.000000000 +0900
+++ src/gsnmp-0.3.0/configure.ac	2011-06-04 01:29:27.607980700 +0900
@@ -19,7 +19,7 @@ dnl Checks for programs.
 AC_PROG_INSTALL
 AC_PROG_CC
 
-AC_PROG_LIBTOOL
+LT_INIT
 
 dnl Checks for libraries.
 
--- origsrc/gsnmp-0.3.0/src/Makefile.am	2010-02-19 17:35:29.000000000 +0900
+++ src/gsnmp-0.3.0/src/Makefile.am	2011-06-04 01:40:27.807300700 +0900
@@ -23,4 +23,5 @@ gsnmpinclude_HEADERS	= ber.h		\
 			  utils.h	\
 			  gsnmp.h
 
-# libgsnmp_la_LDFLAGS	= -version-info @VERSION_LIBTOOL@
+libgsnmp_la_LDFLAGS	= -no-undefined
+libgsnmp_la_LIBADD	= @GLIB_LIBS@ @GNET_LIBS@
