--- origsrc/libgeohash/CMakeLists.txt	1970-01-01 09:00:00.000000000 +0900
+++ src/libgeohash/CMakeLists.txt	2025-06-06 21:17:34.504990700 +0900
@@ -0,0 +1,28 @@
+cmake_minimum_required(VERSION 3.28)
+project(geohash-simplegeo
+  VERSION 0.0.0
+  DESCRIPTION "Pure C implementation of the Geohash algorithm"
+  HOMEPAGE_URL "https://github.com/simplegeo/libgeohash")
+
+include(GNUInstallDirs)
+include(CTest)
+
+add_library(geohash-simplegeo SHARED geohash.c)
+set_target_properties(geohash-simplegeo PROPERTIES SOVERSION 0)
+
+install(TARGETS geohash-simplegeo
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+add_executable(geohash-simplegeo-test geohash_test.c)
+target_link_libraries(geohash-simplegeo-test geohash-simplegeo)
+add_test(NAME geohash-simplegeo-test COMMAND geohash-simplegeo-test)
+
+install(FILES geohash.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} RENAME geohash-simplegeo.h)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/geohash-simplegeo.pc.in ${CMAKE_BINARY_DIR}/geohash-simplegeo.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/geohash-simplegeo.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libgeohash/geohash-simplegeo.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libgeohash/geohash-simplegeo.pc.in	2025-06-06 21:17:28.623542900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: geohash-simplegeo
+Description: Pure C implementation of the Geohash algorithm
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lgeohash-simplegeo
+Cflags: -I${includedir}
