--- origsrc/lcm-1.0.0/configure.in	2013-06-08 13:20:10.000000000 +0900
+++ src/lcm-1.0.0/configure.in	2013-12-22 01:04:59.438883700 +0900
@@ -8,7 +8,7 @@ AC_CONFIG_MACRO_DIR([m4])
 AC_PROG_CC
 PKG_CHECK_MODULES(GLIB,[glib-2.0 >= 2.0.0 gthread-2.0 >= 2.0.0],,[AC_MSG_ERROR([LCM requires GLib >= 2.0])])
 AC_PATH_PROGS(JAR, [jar])
-AC_PATH_PROGS(JAVAC, [javac])
+AC_PATH_PROGS(JAVAC, [javac ecj])
 
 dnl enable silent build if possible (automake >= 1.11)
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
--- origsrc/lcm-1.0.0/lcm/Makefile.am	2012-11-29 14:15:44.000000000 +0900
+++ src/lcm-1.0.0/lcm/Makefile.am	2013-12-22 00:46:21.722086900 +0900
@@ -2,7 +2,7 @@ INCLUDES = $(GLIB_CFLAGS)
 
 lib_LTLIBRARIES = liblcm.la
 
-liblcm_la_LDFLAGS = -version-info @LIBLCM_VERSION_INFO@
+liblcm_la_LDFLAGS = -no-undefined -version-info @LIBLCM_VERSION_INFO@
 liblcm_la_SOURCES = \
 	dbg.h \
 	lcm.c \
--- origsrc/lcm-1.0.0/lcm-lua/Makefile.am	2013-02-25 13:55:37.000000000 +0900
+++ src/lcm-1.0.0/lcm-lua/Makefile.am	2013-12-22 00:52:41.811930600 +0900
@@ -13,7 +13,7 @@ lcm_la_SOURCES = \
     lua_ref_helper.h \
     lua_ref_helper.c
 
-lcm_la_LDFLAGS = -module -avoid-version 
+lcm_la_LDFLAGS = -no-undefined -module -avoid-version
 lcm_la_LIBADD = ../lcm/liblcm.la $(LUA_LIB)
 lcm_la_CFLAGS = -I$(top_srcdir) $(LUA_INCLUDE)
 
--- origsrc/lcm-1.0.0/lcm-python/Makefile.am	2010-03-26 06:39:24.000000000 +0900
+++ src/lcm-1.0.0/lcm-python/Makefile.am	2013-12-22 00:52:02.518961900 +0900
@@ -13,11 +13,9 @@ _lcm_la_SOURCES = \
 
 # linker flags
 _lcm_la_LDFLAGS = -module -avoid-version
-#if PLATFORM_WIN32
-#_lcm_la_LDFLAGS += -no-undefined
-#endif
+_lcm_la_LDFLAGS += -no-undefined
 
-_lcm_la_LIBADD = ../lcm/liblcm.la
+_lcm_la_LIBADD = ../lcm/liblcm.la -lpython${PYTHON_VERSION}
 _lcm_la_CFLAGS = -I$(top_srcdir)
 
 EXTRA_DIST = setup.py
