--- origsrc/gumbo-libxml/Makefile.am	2023-09-05 10:34:42.000000000 +0900
+++ src/gumbo-libxml/Makefile.am	2023-09-05 10:36:12.408555100 +0900
@@ -1,7 +1,7 @@
 lib_LTLIBRARIES = libgumbo_xml.la
 libgumbo_xml_la_CFLAGS = -Wall
 libgumbo_xml_la_CPPFLAGS = -I"$(srcdir)" $(GUMBO_CFLAGS) $(XML_CFLAGS)
-libgumbo_xml_la_LDFLAGS = -version-info 1:0:0 -no-undefine
+libgumbo_xml_la_LDFLAGS = -version-info 1:0:0 -no-undefined
 libgumbo_xml_la_LIBADD = $(GUMBO_LIBS) $(XML_LIBS)
 libgumbo_xml_la_SOURCES = gumbo_libxml.c gumbo_libxml.h
 include_HEADERS = gumbo_libxml.h
--- origsrc/gumbo-libxml/gumbo_libxml.pc.in	2023-09-05 10:34:42.000000000 +0900
+++ src/gumbo-libxml/gumbo_libxml.pc.in	2023-09-05 11:02:23.376386600 +0900
@@ -5,7 +5,7 @@ includedir=@includedir@
 
 Name: Gumbo libxml bindings
 Description: A libxml2-compatible API for the Gumbo HTML parser.
-Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lgumbo_libxml
+Version: @VERSION@
+Libs: -L${libdir} -lgumbo_xml
 Libs.private: @LIBS@
 Cflags: -I${includedir}
