--- origsrc/libnpupnp-6.1.3/meson.build	2024-05-21 16:33:44.000000000 +0900
+++ src/libnpupnp-6.1.3/meson.build	2024-07-21 15:52:35.342031900 +0900
@@ -134,11 +134,10 @@ auto.set10('UPNP_HAVE_WEBSERVER', get_op
 ufile = configure_file(output: 'upnpconfig.h', configuration: auto)
 cfile = configure_file(output: 'autoconfig.h', configuration: auto)
 
-libnpupnp = library(
-  'libnpupnp',
+libnpupnp = shared_library(
+  'npupnp',
   npupnp_sources,
   gnu_symbol_visibility: 'hidden',
-  name_prefix: '',
   version: npupnp_soversion + npupnp_soversion_minor,
   soversion: host_machine.system() == 'windows' ? '' : npupnp_soversion,
   include_directories: libnpupnp_incdir,
