--- origsrc/libnpupnp-6.1.0/configure.ac	2024-01-08 18:03:02.000000000 +0900
+++ src/libnpupnp-6.1.0/configure.ac	2024-02-15 15:16:38.099975100 +0900
@@ -25,7 +25,7 @@ case "${host_os}" in
     linux*|uclinux*) build_linux=yes;;
     darwin*) build_mac=yes;;
     *freebsd*) build_freebsd=yes;;
-    *mingw*) build_mingw=yes;;
+    *cygwin*|*mingw*) build_mingw=yes;;
     *) AC_MSG_WARN(["OS $host_os is unknown"]);;
 esac
 # Pass the conditionals to automake
