--- origsrc/libnpupnp-4.1.4/configure.ac	2021-04-19 19:04:43.000000000 +0900
+++ src/libnpupnp-4.1.4/configure.ac	2021-06-16 09:49:01.815987500 +0900
@@ -28,7 +28,7 @@ case "${host_os}" in
     linux*|uclinux*) build_linux=yes;;
     darwin*) build_mac=yes;;
     *freebsd*) build_freebsd=yes;;
-    *mingw*) build_mingw=yes;;
+    *cygwin*|*mingw*) build_mingw=yes;;
     *) AC_MSG_WARN(["OS $host_os is unknown"]);;
 esac
 # Pass the conditionals to automake
--- origsrc/libnpupnp-4.1.4/src/threadutil/ThreadPool.cpp	2021-02-04 02:07:11.000000000 +0900
+++ src/libnpupnp-4.1.4/src/threadutil/ThreadPool.cpp	2021-06-16 09:49:59.410205100 +0900
@@ -255,7 +255,7 @@ static int SetPolicyType(ThreadPoolAttr:
  */
 static int SetPriority(ThreadPool::ThreadPriority priority)
 {
-#if defined(_POSIX_PRIORITY_SCHEDULING) && _POSIX_PRIORITY_SCHEDULING > 0
+#if !defined(__CYGWIN__) && defined(_POSIX_PRIORITY_SCHEDULING) && _POSIX_PRIORITY_SCHEDULING > 0
     int retVal = 0;
     int currentPolicy;
     int minPriority = 0;
--- origsrc/libnpupnp-4.1.4/src/utils/netif.cpp	2021-04-16 16:28:41.000000000 +0900
+++ src/libnpupnp-4.1.4/src/utils/netif.cpp	2021-06-16 09:51:02.212422600 +0900
@@ -46,7 +46,7 @@
 #include <ifaddrs.h>
 #ifdef __linux__
 #include <netpacket/packet.h>
-#else
+#elif !defined(__CYGWIN__)
 #include <net/if_dl.h>
 #endif
 
@@ -565,7 +565,7 @@ Interfaces::Internal::Internal()
             ifit->m->sethwaddr(reinterpret_cast<const char*>(sll->sll_addr), sll->sll_halen);
         }
         break;
-#else
+#elif !defined(__CYGWIN__)
         case AF_LINK:
         {
             auto sdl = reinterpret_cast<struct sockaddr_dl*>(ifa->ifa_addr);
