--- origsrc/tpm2-pkcs11-1.0.1/Makefile.am	2020-01-07 05:25:11.000000000 +0900
+++ src/tpm2-pkcs11-1.0.1/Makefile.am	2020-01-16 10:17:46.726890500 +0900
@@ -46,8 +46,9 @@ libtpm2_pkcs11 = src/libtpm2_pkcs11.la
 pkgconfig_DATA += lib/tpm2-pkcs11.pc
 EXTRA_DIST += lib/tpm2-pkcs11.map
 
+src_libtpm2_pkcs11_la_LDFLAGS = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_libtpm2_pkcs11_la_LDFLAGS = -Wl,--version-script=$(srcdir)/lib/tpm2-pkcs11.map
+src_libtpm2_pkcs11_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tpm2-pkcs11.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_libtpm2_pkcs11_la_LIBADD = $(AM_LDFLAGS)
 src_libtpm2_pkcs11_la_SOURCES = $(LIB_PKCS11_SRC) $(LIB_PKCS11_INTERNAL_LIB_SRC)
--- origsrc/tpm2-pkcs11-1.0.1/configure.ac	2020-01-07 05:25:11.000000000 +0900
+++ src/tpm2-pkcs11-1.0.1/configure.ac	2020-01-16 09:55:46.031663300 +0900
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: BSD-2-Clause
 
 AC_INIT([tpm2-pkcs11],
-  [m4_esyscmd_s([git describe --tags --always --dirty])],
+  [1.0.1],
   [https://github.com/tpm2-software/tpm2-pkcs11/issues],
   [],
   [https://github.com/tpm2-software/tpm2-pkcs11])
--- origsrc/tpm2-pkcs11-1.0.1/misc/p11-kit/tpm2_pkcs11.module	2020-01-07 05:25:11.000000000 +0900
+++ src/tpm2-pkcs11-1.0.1/misc/p11-kit/tpm2_pkcs11.module	2020-01-16 10:20:11.308089200 +0900
@@ -1,3 +1,3 @@
-module: libtpm2_pkcs11.so
+module: cygtpm2_pkcs11-0.dll
 
 critical: no
--- origsrc/tpm2-pkcs11-1.0.1/src/lib/db.c	2020-01-07 05:25:11.000000000 +0900
+++ src/tpm2-pkcs11-1.0.1/src/lib/db.c	2020-01-16 10:07:39.007602400 +0900
@@ -7,7 +7,9 @@
 #include <stdio.h>
 #include <string.h>
 
+#if defined(__linux__)
 #include <linux/limits.h>
+#endif
 
 #include <sys/stat.h>
 #include <sys/types.h>
