--- origsrc/tpm2-pkcs11-1.2.0/Makefile.am	2020-03-30 23:45:55.000000000 +0900
+++ src/tpm2-pkcs11-1.2.0/Makefile.am	2020-04-09 15:52:00.302854600 +0900
@@ -50,8 +50,9 @@ libtpm2_pkcs11 = src/libtpm2_pkcs11.la
 pkgconfig_DATA += lib/tpm2-pkcs11.pc
 EXTRA_DIST += lib/tpm2-pkcs11.map
 
+src_libtpm2_pkcs11_la_LDFLAGS = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_libtpm2_pkcs11_la_LDFLAGS = -Wl,--version-script=$(srcdir)/lib/tpm2-pkcs11.map
+src_libtpm2_pkcs11_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tpm2-pkcs11.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_libtpm2_pkcs11_la_LIBADD = $(AM_LDFLAGS)
 src_libtpm2_pkcs11_la_SOURCES = $(LIB_PKCS11_SRC) $(LIB_PKCS11_INTERNAL_LIB_SRC)
--- origsrc/tpm2-pkcs11-1.2.0/configure.ac	2020-03-30 23:45:55.000000000 +0900
+++ src/tpm2-pkcs11-1.2.0/configure.ac	2020-04-09 15:52:00.318486200 +0900
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: BSD-2-Clause
 
 AC_INIT([tpm2-pkcs11],
-  [m4_esyscmd_s([git describe --tags --always --dirty])],
+  [1.0.1],
   [https://github.com/tpm2-software/tpm2-pkcs11/issues],
   [],
   [https://github.com/tpm2-software/tpm2-pkcs11])
--- origsrc/tpm2-pkcs11-1.2.0/misc/p11-kit/tpm2_pkcs11.module	2020-03-30 23:45:55.000000000 +0900
+++ src/tpm2-pkcs11-1.2.0/misc/p11-kit/tpm2_pkcs11.module	2020-04-09 15:52:00.365377800 +0900
@@ -1,3 +1,3 @@
-module: libtpm2_pkcs11.so
+module: cygtpm2_pkcs11-0.dll
 
 critical: no
--- origsrc/tpm2-pkcs11-1.2.0/src/lib/db.c	2020-03-30 23:45:55.000000000 +0900
+++ src/tpm2-pkcs11-1.2.0/src/lib/db.c	2020-04-09 15:52:00.396644900 +0900
@@ -11,7 +11,9 @@
 #include <libgen.h>
 #include <unistd.h>
 
+#if defined(__linux__)
 #include <linux/limits.h>
+#endif
 
 #include <sys/file.h>
 #include <sys/stat.h>
