--- origsrc/tpm2-pkcs11-1.9.1/Makefile.am	2025-09-13 04:31:09.306150300 +0900
+++ src/tpm2-pkcs11-1.9.1/Makefile.am	2025-09-13 04:31:52.615348300 +0900
@@ -1,5 +1,5 @@
 # SPDX-License-Identifier: BSD-2-Clause
-src_libtpm2_pkcs11_la_LDFLAGS    =  -version-number 1 -version-info 1:9:0
+src_libtpm2_pkcs11_la_LDFLAGS    = -no-undefined -version-number 1 -version-info 1:9:0
 INCLUDE_DIRS    = -I$(srcdir)/src -I$(top_srcdir)/src/lib
 ACLOCAL_AMFLAGS = -I m4 --install
 AM_CFLAGS       = $(INCLUDE_DIRS) $(EXTRA_CFLAGS) $(CODE_COVERAGE_CFLAGS) \
--- origsrc/tpm2-pkcs11-1.9.1/bootstrap	2024-09-24 09:54:23.000000000 +0900
+++ src/tpm2-pkcs11-1.9.1/bootstrap	2025-09-13 04:32:11.187487700 +0900
@@ -4,7 +4,7 @@ set -e
 
 # Generate a VERSION file that is included in the dist tarball to avoid needed git
 # when calling autoreconf in a release tarball.
-git describe --tags --always --dirty > VERSION
+test -f VERSION || git describe --tags --always --dirty > VERSION
 
 # generate list of source files for use in Makefile.am
 # if you add new source files, you must run ./bootstrap again
--- origsrc/tpm2-pkcs11-1.9.1/misc/p11-kit/tpm2_pkcs11.module	2024-09-24 09:54:23.000000000 +0900
+++ src/tpm2-pkcs11-1.9.1/misc/p11-kit/tpm2_pkcs11.module	2025-09-13 04:32:34.648131500 +0900
@@ -1,3 +1,3 @@
-module: libtpm2_pkcs11.so
+module: cygtpm2_pkcs11-1.dll
 
 critical: no
--- origsrc/tpm2-pkcs11-1.9.1/src/lib/db.c	2024-09-24 09:54:23.000000000 +0900
+++ src/tpm2-pkcs11-1.9.1/src/lib/db.c	2025-09-13 04:33:03.169116500 +0900
@@ -12,7 +12,9 @@
 #include <libgen.h>
 #include <unistd.h>
 
+#if defined(__linux__)
 #include <linux/limits.h>
+#endif
 
 #include <sys/file.h>
 #include <sys/stat.h>
