--- origsrc/libfwevt-20240504/Makefile.am	2024-05-04 13:35:42.000000000 +0900
+++ src/libfwevt-20240504/Makefile.am	2025-05-17 22:51:12.702843900 +0900
@@ -3,17 +3,8 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libcnotify \
-	libuna \
-	libfdatetime \
-	libfguid \
-	libfwnt \
 	libfwevt \
 	pyfwevt \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libfwevt-20240504/configure.ac	2024-05-04 13:35:42.000000000 +0900
+++ src/libfwevt-20240504/configure.ac	2025-05-17 22:51:12.702843900 +0900
@@ -136,14 +136,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfwnt/Makefile])
 AC_CONFIG_FILES([libfwevt/Makefile])
 AC_CONFIG_FILES([pyfwevt/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
--- origsrc/libfwevt-20240504/pyfwevt/Makefile.am	2024-05-04 13:35:42.000000000 +0900
+++ src/libfwevt-20240504/pyfwevt/Makefile.am	2025-05-17 22:51:12.702843900 +0900
@@ -48,7 +48,7 @@ pyfwevt_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pyfwevt_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyfwevt_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyfwevt_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
