--- origsrc/libfwevt-20200709/Makefile.am	2020-07-09 17:38:12.000000000 +0900
+++ src/libfwevt-20200709/Makefile.am	2020-09-03 10:43:58.428160200 +0900
@@ -3,16 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libcnotify \
-	libuna \
-	libfdatetime \
-	libfguid \
-	libfvalue \
 	libfwevt \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libfwevt-20200709/configure.ac	2020-07-09 17:38:12.000000000 +0900
+++ src/libfwevt-20200709/configure.ac	2020-09-03 10:43:58.438178500 +0900
@@ -139,14 +139,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
 AC_CONFIG_FILES([libfwevt/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
--- origsrc/libfwevt-20200709/libfwevt/Makefile.am	2020-07-09 17:38:12.000000000 +0900
+++ src/libfwevt-20200709/libfwevt/Makefile.am	2020-09-03 10:43:58.449180500 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libfwevt-20200709/libfwevt/libfwevt_xml_tag.h	2020-07-09 17:38:12.000000000 +0900
+++ src/libfwevt-20200709/libfwevt/libfwevt_xml_tag.h	2020-09-03 10:43:58.465161300 +0900
@@ -88,6 +88,7 @@ int libfwevt_xml_tag_set_name_data(
      int encoding,
      libcerror_error_t **error );
 
+LIBFWEVT_EXTERN \
 int libfwevt_xml_tag_get_value(
      libfwevt_xml_tag_t *xml_tag,
      libfvalue_value_t **value,
--- origsrc/libfwevt-20200709/tests/Makefile.am	2020-07-09 17:38:12.000000000 +0900
+++ src/libfwevt-20200709/tests/Makefile.am	2020-09-03 10:43:58.522184300 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
