diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*.whl' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libclsp-0.1.3/CMakeLists.txt src/libclsp-0.1.3/CMakeLists.txt
--- origsrc/libclsp-0.1.3/CMakeLists.txt	2020-01-16 05:40:52.000000000 +0900
+++ src/libclsp-0.1.3/CMakeLists.txt	2020-01-22 09:10:21.768226000 +0900
@@ -59,7 +59,7 @@ configure_file(libclsp.pc.in
 )
 install(FILES
 	${CMAKE_BINARY_DIR}/libclsp.pc
-	DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/pkgconfig
+	DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
 )
 
 # Libraries
