--- origsrc/libpipeline-1.2.2/lib/Makefile.am	2012-09-18 06:53:59.000000000 +0900
+++ src/libpipeline-1.2.2/lib/Makefile.am	2012-11-04 14:44:15.361463400 +0900
@@ -37,7 +37,7 @@ libpipeline_la_LIBADD = ../gnulib/lib/li
 
 libpipeline_la_LDFLAGS = \
 	-export-symbols-regex '^(pipecmd|pipeline)_' \
-	-version-info 3:2:2
+	-version-info 3:2:2 -no-undefined
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libpipeline.pc
--- origsrc/libpipeline-1.2.2/lib/pipeline.c	2012-03-05 02:43:07.000000000 +0900
+++ src/libpipeline-1.2.2/lib/pipeline.c	2012-11-04 14:56:30.263460800 +0900
@@ -706,7 +706,11 @@ void pipecmd_exec (pipecmd *cmd)
 			else
 				unsetenv (cmd->env[i].name);
 		} else
+#if defined(__CYGWIN__)
+                	environ = NULL;
+#else
 			clearenv ();
+#endif
 	}
 
 	switch (cmd->tag) {
