--- origsrc/libpipeline-1.2.5/lib/Makefile.am	2013-12-03 23:12:09.000000000 +0900
+++ src/libpipeline-1.2.5/lib/Makefile.am	2013-12-22 00:04:31.766292400 +0900
@@ -37,7 +37,7 @@ include_HEADERS = pipeline.h
 
 libpipeline_la_LIBADD = ../gnulib/lib/libgnu.la $(LTLIBOBJS)
 
-libpipeline_la_LDFLAGS = \
+libpipeline_la_LDFLAGS = -no-undefined \
 	-export-symbols-regex '^(pipecmd|pipeline)_' \
 	-version-info 3:5:2
 
--- origsrc/libpipeline-1.2.5/lib/pipeline.c	2013-12-01 02:06:28.000000000 +0900
+++ src/libpipeline-1.2.5/lib/pipeline.c	2013-12-22 00:04:31.766292400 +0900
@@ -89,6 +89,10 @@ int clearenv (void)
 }
 #endif
 
+#if defined(__CYGWIN__)
+#define clearenv() (environ = NULL)
+#endif
+
 /* ---------------------------------------------------------------------- */
 
 /* Functions to build individual commands. */
