--- origsrc/libpipeline-1.2.4/lib/Makefile.am	2013-06-06 20:39:41.000000000 +0900
+++ src/libpipeline-1.2.4/lib/Makefile.am	2013-10-30 22:16:46.338581000 +0900
@@ -37,7 +37,7 @@ include_HEADERS = pipeline.h
 
 libpipeline_la_LIBADD = ../gnulib/lib/libgnu.la $(LTLIBOBJS)
 
-libpipeline_la_LDFLAGS = \
+libpipeline_la_LDFLAGS = -no-undefined \
 	-export-symbols-regex '^(pipecmd|pipeline)_' \
 	-version-info 3:4:2
 
--- origsrc/libpipeline-1.2.4/lib/pipeline.c	2013-06-06 04:28:18.000000000 +0900
+++ src/libpipeline-1.2.4/lib/pipeline.c	2013-10-30 22:16:46.348466400 +0900
@@ -89,6 +89,10 @@ int clearenv (void)
 }
 #endif
 
+#if defined(__CYGWIN__)
+#define clearenv() (environ = NULL)
+#endif
+
 /* ---------------------------------------------------------------------- */
 
 /* Functions to build individual commands. */
