--- origsrc/libpipeline-1.3.0/lib/Makefile.am	2014-03-26 20:12:03.000000000 +0900
+++ src/libpipeline-1.3.0/lib/Makefile.am	2014-03-30 18:24:14.010938500 +0900
@@ -37,7 +37,7 @@ include_HEADERS = pipeline.h
 
 libpipeline_la_LIBADD = ../gnulib/lib/libgnu.la $(LTLIBOBJS)
 
-libpipeline_la_LDFLAGS = \
+libpipeline_la_LDFLAGS = -no-undefined \
 	-export-symbols-regex '^(pipecmd|pipeline)_' \
 	-version-info 4:0:3
 
--- origsrc/libpipeline-1.3.0/lib/pipeline.c	2014-03-26 20:07:37.000000000 +0900
+++ src/libpipeline-1.3.0/lib/pipeline.c	2014-03-30 18:24:14.026563500 +0900
@@ -89,6 +89,10 @@ int clearenv (void)
 }
 #endif
 
+#if defined(__CYGWIN__)
+#define clearenv() (environ = NULL)
+#endif
+
 /* ---------------------------------------------------------------------- */
 
 /* Functions to build individual commands. */
