--- origsrc/libwav/CMakeLists.txt	2019-04-30 12:21:17.000000000 +0900
+++ src/libwav/CMakeLists.txt	2019-04-30 12:24:50.432428000 +0900
@@ -29,6 +29,10 @@ add_library(${PROJECT_NAME} ${sources})
 
 target_include_directories(${PROJECT_NAME} PUBLIC ${includes})
 
+if (CYGWIN)
+  set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION 0)
+endif()
+
 install(
   TARGETS ${PROJECT_NAME} ${PROJECT_NAME}
   RUNTIME DESTINATION bin
@@ -52,3 +56,10 @@ if(WIN32)
     diff-wav
   )
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libwav.pc.in ${CMAKE_BINARY_DIR}/libwav.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libwav.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libwav/libwav.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libwav/libwav.pc.in	2019-04-30 12:25:11.912145100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libwav
+Description: Simple C library for reading/writing PCM wave (.wav) files
+Version: @WAV_VERSION@
+Libs: -L${libdir} -lwav
+Cflags: -I${includedir}
