--- origsrc/libneurosim-1.2.0/libpyneurosim/Makefile.am	2020-11-12 01:11:31.000000000 +0900
+++ src/libneurosim-1.2.0/libpyneurosim/Makefile.am	2021-03-02 12:14:04.974497100 +0900
@@ -13,8 +13,8 @@ libpy3neurosim_la_SOURCES = \
 libpy3neurosim_la_HEADERS = \
   pyneurosim.h
 
-libpy3neurosim_la_CXXFLAGS = -I@PYTHON_INCLUDE@ -I$(srcdir)/../neurosim/conngen
-libpy3neurosim_la_LDFLAGS = $(top_builddir)/neurosim/libneurosim.la
+libpy3neurosim_la_CXXFLAGS = -I@PYTHON_INCLUDE@ -I$(top_srcdir)
+libpy3neurosim_la_LDFLAGS = -no-undefined $(top_builddir)/neurosim/libneurosim.la
 libpy3neurosim_ladir = $(includedir)/neurosim
 else
 lib_LTLIBRARIES += libpy2neurosim.la libpyneurosim.la
@@ -25,8 +25,8 @@ libpy2neurosim_la_SOURCES = \
 libpy2neurosim_la_HEADERS = \
   pyneurosim.h
 
-libpy2neurosim_la_CXXFLAGS = -I@PYTHON_INCLUDE@ -I$(srcdir)/../neurosim/conngen
-libpy2neurosim_la_LDFLAGS = $(top_builddir)/neurosim/libneurosim.la
+libpy2neurosim_la_CXXFLAGS = -I@PYTHON_INCLUDE@ -I$(top_srcdir)
+libpy2neurosim_la_LDFLAGS = -no-undefined $(top_builddir)/neurosim/libneurosim.la
 libpy2neurosim_ladir = $(includedir)/neurosim
 # generate versionless version for backwards compatibility
 libpyneurosim_la_SOURCES = \
@@ -38,7 +38,7 @@ libpyneurosim_la_HEADERS = \
 
 libpyneurosim_la_CXXFLAGS = -I@PYTHON_INCLUDE@ -I$(top_srcdir)
 
-libpyneurosim_la_LDFLAGS = $(top_builddir)/neurosim/libneurosim.la
+libpyneurosim_la_LDFLAGS = -no-undefined $(top_builddir)/neurosim/libneurosim.la
 
 libpyneurosim_ladir = $(includedir)/neurosim
 endif
--- origsrc/libneurosim-1.2.0/neurosim/Makefile.am	2020-11-12 01:11:31.000000000 +0900
+++ src/libneurosim-1.2.0/neurosim/Makefile.am	2021-03-02 12:12:00.515730200 +0900
@@ -2,7 +2,7 @@
 
 AUTOMAKE_OPTIONS = subdir-objects
 
-SUBDIRS = . examples
+SUBDIRS = . # examples
 
 lib_LTLIBRARIES = libneurosim.la
 
@@ -24,3 +24,5 @@ libneurosim_la_LIBADD= @LIBLTDL@ @LIBADD
 libneurosim_ladir = $(includedir)/neurosim
 
 AM_CPPFLAGS= -I$(top_srcdir) @INCLTDL@ @MPI_INCLUDE@
+
+libneurosim_la_LDFLAGS = -no-undefined
