--- origsrc/urdfdom-3.0.0/CMakeLists.txt	2021-10-07 01:45:49.000000000 +0900
+++ src/urdfdom-3.0.0/CMakeLists.txt	2021-10-12 17:39:52.910071000 +0900
@@ -64,6 +64,8 @@ enable_testing()
 
 if(WIN32 AND NOT CYGWIN)
   set(CMAKE_CONFIG_INSTALL_DIR CMake)
+elseif(CYGWIN)
+  set(CMAKE_CONFIG_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
 else()
   set(CMAKE_CONFIG_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/${PROJECT_NAME}/cmake)
 endif()
