--- origsrc/urdfdom-1.0.4/CMakeLists.txt	2020-01-18 03:30:42.000000000 +0900
+++ src/urdfdom-1.0.4/CMakeLists.txt	2020-07-25 11:16:08.231456800 +0900
@@ -41,7 +41,7 @@ if (NOT MSVC)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 endif()
 
-find_package(console_bridge 0.3 REQUIRED)
+find_package(console_bridge 1.0 REQUIRED)
 include_directories(SYSTEM ${console_bridge_INCLUDE_DIRS})
 link_directories(${console_bridge_LIBRARY_DIRS})
 
@@ -62,6 +62,8 @@ add_subdirectory(urdf_parser)
 
 if(WIN32 AND NOT CYGWIN)
   set(CMAKE_CONFIG_INSTALL_DIR CMake)
+elseif(CYGWIN)
+  set(CMAKE_CONFIG_INSTALL_DIR ${CMAKE_INSTALL_FULL_LIBDIR}/cmake/${PROJECT_NAME}/)
 else()
     set(CMAKE_CONFIG_INSTALL_DIR ${CMAKE_INSTALL_FULL_LIBDIR}/${PROJECT_NAME}/cmake/)
 endif()
--- origsrc/urdfdom-1.0.4/urdf_parser/CMakeLists.txt	2020-01-18 03:30:42.000000000 +0900
+++ src/urdfdom-1.0.4/urdf_parser/CMakeLists.txt	2020-07-25 11:26:09.226683300 +0900
@@ -33,10 +33,10 @@ if(BUILD_TESTING)
   add_subdirectory(test)
 endif()
 
-INSTALL(TARGETS urdfdom_model DESTINATION ${CMAKE_INSTALL_LIBDIR})
-INSTALL(TARGETS urdfdom_world DESTINATION ${CMAKE_INSTALL_LIBDIR})
+INSTALL(TARGETS urdfdom_model RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+INSTALL(TARGETS urdfdom_world RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 INSTALL(TARGETS check_urdf urdf_to_graphiz urdf_mem_test
   DESTINATION ${CMAKE_INSTALL_BINDIR})
-INSTALL(TARGETS urdfdom_sensor DESTINATION ${CMAKE_INSTALL_LIBDIR})
-INSTALL(TARGETS urdfdom_model_state DESTINATION ${CMAKE_INSTALL_LIBDIR})
+INSTALL(TARGETS urdfdom_sensor RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+INSTALL(TARGETS urdfdom_model_state RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 INSTALL(DIRECTORY include/urdf_parser DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
--- origsrc/urdfdom-1.0.4/urdf_parser/test/CMakeLists.txt	2020-01-18 03:30:42.000000000 +0900
+++ src/urdfdom-1.0.4/urdf_parser/test/CMakeLists.txt	2020-07-25 11:22:02.429886900 +0900
@@ -1,17 +1,20 @@
-include_directories (
-  ${CMAKE_CURRENT_SOURCE_DIR}/gtest/include
-  ${CMAKE_CURRENT_SOURCE_DIR}/gtest
-  ${CMAKE_CURRENT_SOURCE_DIR}
-)
+option(USE_SYSTEM_GTEST "Use system gtest" ON)
+if (NOT USE_SYSTEM_GTEST)
+  include_directories (
+    ${CMAKE_CURRENT_SOURCE_DIR}/gtest/include
+    ${CMAKE_CURRENT_SOURCE_DIR}/gtest
+    ${CMAKE_CURRENT_SOURCE_DIR}
+  )
 
-# Build gtest
-add_library(gtest STATIC gtest/src/gtest-all.cc)
-add_library(gtest_main STATIC gtest/src/gtest_main.cc)
-target_link_libraries(gtest_main gtest)
+  # Build gtest
+  add_library(gtest STATIC gtest/src/gtest-all.cc)
+  add_library(gtest_main STATIC gtest/src/gtest_main.cc)
+  target_link_libraries(gtest_main gtest)
 
-execute_process(COMMAND cmake -E remove_directory ${CMAKE_BINARY_DIR}/test_results)
-execute_process(COMMAND cmake -E make_directory ${CMAKE_BINARY_DIR}/test_results)
-include_directories(${GTEST_INCLUDE_DIRS})
+  execute_process(COMMAND cmake -E remove_directory ${CMAKE_BINARY_DIR}/test_results)
+  execute_process(COMMAND cmake -E make_directory ${CMAKE_BINARY_DIR}/test_results)
+  include_directories(${GTEST_INCLUDE_DIRS})
+endif()
 
 # unit test to fix geometry problems
 set(tests
