--- origsrc/libperseus-sdr-0.8.2/Makefile.am	2020-03-16 06:39:53.000000000 +0900
+++ src/libperseus-sdr-0.8.2/Makefile.am	2021-04-19 18:20:42.038415600 +0900
@@ -41,6 +41,7 @@ lib_LTLIBRARIES = libperseus-sdr.la
 libperseus_sdr_la_CFLAGS = $(AM_CFLAGS)
 libperseus_sdr_la_SOURCES =  perseus-sdr.c perseuserr.c perseusfx2.c perseus-in.c \
                              perseusfx2.h perseus-in.h perseus-sdr.h fpga_data.h
+libperseus_sdr_la_LDFLAGS = -no-undefined
 
 nodist_libperseus_sdr_la_SOURCES = fpga_data.c
 
@@ -69,11 +70,11 @@ dist_pkgdata_DATA = $(udev_DATA)
 # sudo udevadm control --reload-rules
 #
 udevrule_DATA = $(top_srcdir)/95-perseus.rules
-udevruledir = $(bindir)
+udevruledir = /etc/udev/rules.d
 EXTRA_DIST = $(top_srcdir)/95-perseus.rules README.Windows $(top_srcdir)/build-aux/git-version-gen
 
 install-data-hook:
-	-cp $(bindir)/95-perseus.rules /etc/udev/rules.d/ ; groupadd perseususb ; echo $(INST_USER) ;usermod -a -G perseususb $(INST_USER) ; udevadm control --reload-rules
+#	-cp $(bindir)/95-perseus.rules /etc/udev/rules.d/ ; groupadd perseususb ; echo $(INST_USER) ;usermod -a -G perseususb $(INST_USER) ; udevadm control --reload-rules
 
 uninstall-hook:
 	-rm /etc/udev/rules.d/95-perseus.rules ; groupdel perseususb ; udevadm control --reload-rules
--- origsrc/libperseus-sdr-0.8.2/examples/Makefile.am	2020-03-16 06:39:53.000000000 +0900
+++ src/libperseus-sdr-0.8.2/examples/Makefile.am	2021-04-19 18:21:09.502089300 +0900
@@ -1,6 +1,8 @@
 
 ACLOCAL_AMFLAGS = -I build-aux
 
+AM_CPPFLAGS = -I$(top_srcdir)
+
 $(top_srcdir)/.version:
 	echo $(VERSION) > $@-t && mv $@-t $@
 dist-hook:
