--- origsrc/fpzip-1.3.0/CMakeLists.txt	2019-12-21 10:48:18.000000000 +0900
+++ src/fpzip-1.3.0/CMakeLists.txt	2020-11-23 22:49:27.643361300 +0900
@@ -205,3 +205,10 @@ install(FILES
 # Install the export set for use with the install-tree
 install(EXPORT fpzip-targets NAMESPACE fpzip::
   DESTINATION "${CMAKE_INSTALL_CMAKEDIR}")
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/fpzip.pc.in ${CMAKE_BINARY_DIR}/fpzip.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/fpzip.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/fpzip-1.3.0/fpzip.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/fpzip-1.3.0/fpzip.pc.in	2020-11-23 22:49:45.945718300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: fpzip
+Description: Compressed Large Multidimensional Floating-Point Arrays
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lfpzip
+Cflags: -I${includedir}
