--- origsrc/sha2-1.0.1/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/sha2-1.0.1/Makefile.am	2014-12-21 16:58:39.638698200 +0900
@@ -0,0 +1,13 @@
+bin_PROGRAMS = sha2 sha2speed
+sha2_SOURCES = sha2prog.c
+sha2_LDADD = libsha2.la
+sha2speed_SOURCES = sha2speed.c
+sha2speed_LDADD = libsha2.la
+
+include_HEADERS = sha2.h
+
+lib_LTLIBRARIES = libsha2.la
+libsha2_la_SOURCES = sha2.c
+libsha2_la_LDFLAGS = -no-undefined
+
+TESTS = sha2test.pl
--- origsrc/sha2-1.0.1/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/sha2-1.0.1/configure.ac	2014-12-21 16:58:06.327174800 +0900
@@ -0,0 +1,29 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([sha2], [1.0.1])
+AC_CONFIG_SRCDIR([sha2.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([inttypes.h stdlib.h string.h sys/time.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+AC_TYPE_UINT8_T
+
+# Checks for library functions.
+AC_CHECK_FUNCS([gettimeofday memset])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
