--- origsrc/ugrep-indexer-0.9.6/src/Makefile.am	2024-03-01 05:29:51.000000000 +0900
+++ src/ugrep-indexer-0.9.6/src/Makefile.am	2024-03-22 10:52:04.314292000 +0900
@@ -1,4 +1,4 @@
 bin_PROGRAMS   = ugrep-indexer
-ugrep_indexer_CPPFLAGS = $(EXTRA_CFLAGS) -DPLATFORM=\"$(PLATFORM)\"
+ugrep_indexer_CPPFLAGS = $(EXTRA_CFLAGS) -DPLATFORM=\"$(PLATFORM)\" -I$(top_srcdir)/lzma/C
 ugrep_indexer_SOURCES  = ugrep-indexer.cpp glob.hpp glob.cpp input.h input.cpp utf8.h zopen.h zopen.c zstream.hpp
 ugrep_indexer_LDADD    = $(PTHREAD_LIBS)
