--- origsrc/ugrep-indexer-0.9.5/src/Makefile.am	2024-01-01 10:15:07.000000000 +0900
+++ src/ugrep-indexer-0.9.5/src/Makefile.am	2024-01-15 10:17:55.015057400 +0900
@@ -1,4 +1,4 @@
 bin_PROGRAMS   = ugrep-indexer
-ugrep_indexer_CPPFLAGS = -DPLATFORM=\"$(PLATFORM)\"
+ugrep_indexer_CPPFLAGS = -DPLATFORM=\"$(PLATFORM)\" -I$(top_srcdir)/lzma/C
 ugrep_indexer_SOURCES  = ugrep-indexer.cpp glob.hpp glob.cpp input.h input.cpp utf8.h zopen.h zopen.c zstream.hpp
 ugrep_indexer_LDADD    = $(PTHREAD_LIBS)
