--- origsrc/ugrep-indexer-1.0.0/src/Makefile.am	2024-04-19 02:32:26.000000000 +0900
+++ src/ugrep-indexer-1.0.0/src/Makefile.am	2024-06-07 17:36:12.373789700 +0900
@@ -1,4 +1,4 @@
 bin_PROGRAMS   = ugrep-indexer
-ugrep_indexer_CPPFLAGS = $(EXTRA_CFLAGS) -DPLATFORM=\"$(PLATFORM)\"
+ugrep_indexer_CPPFLAGS = $(EXTRA_CFLAGS) -DPLATFORM=\"$(PLATFORM)\" -I$(top_srcdir)/lzma/C
 ugrep_indexer_SOURCES  = ugrep-indexer.cpp glob.hpp glob.cpp reflex/input.h input.cpp reflex/utf8.h zopen.h zopen.c zstream.hpp zthread.hpp
 ugrep_indexer_LDADD    = $(PTHREAD_LIBS)
