--- origsrc/maloc/Makefile.am	2010-08-10 04:01:25.000000000 +0900
+++ src/maloc/Makefile.am	2015-03-24 13:51:59.015664100 +0900
@@ -37,6 +37,9 @@ SUFFIXES = .y .l .H .h .m .C .cc .c .f .
 always_built_SUBDIRS = src doc
 SUBDIRS = $(always_built_SUBDIRS) .
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = maloc.pc
+
 dox:
 	cd ${top_srcdir}/doc/doxygen; make
 
--- origsrc/maloc/configure.ac	2010-12-03 05:22:07.000000000 +0900
+++ src/maloc/configure.ac	2015-03-24 13:51:34.109414100 +0900
@@ -820,4 +820,5 @@ AC_OUTPUT([
 
     doc/Makefile
     doc/doxygen/Makefile
+    maloc.pc
 ])
--- origsrc/maloc/maloc.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/maloc/maloc.pc.in	2015-03-24 13:51:41.406289100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: maloc
+Description: Minimal Abstraction Layer for Object-oriented C
+Version: @VERSION@
+Libs: -L${libdir} -lmaloc
+Cflags: -I${includedir}
--- origsrc/maloc/src/aaa_lib/Makefile.am	2015-03-24 13:44:44.235390600 +0900
+++ src/maloc/src/aaa_lib/Makefile.am	2015-03-24 13:46:12.837929700 +0900
@@ -55,9 +55,9 @@ lib_LTLIBRARIES = libmaloc.la ${LIBEFENC
 libmaloc_la_SOURCES =
 libmaloc_la_LIBADD  = ${BASE_LIBS} ${VSYS_LIBS} ${VSH_LIBS} ${PSH_LIBS}
 # libmaloc_la_LDFLAGS = -version-info ${FETK_VERSION}
-libmaloc_la_LDFLAGS = @rl_lib@ @math_lib@ -version-info ${FETK_VERSION}
+libmaloc_la_LDFLAGS = -no-undefined @rl_lib@ @math_lib@ -version-info ${FETK_VERSION}
 
 libefence_la_SOURCES =
 libefence_la_LIBADD  = ${EFENCE_LIBS}
-libefence_la_LDFLAGS = -version-info ${FETK_VERSION}
+libefence_la_LDFLAGS = -no-undefined -version-info ${FETK_VERSION}
 
