--- origsrc/maloc/configure.ac	2010-12-03 05:22:07.000000000 +0900
+++ src/maloc/configure.ac	2011-04-02 12:17:24.225771700 +0900
@@ -166,8 +166,7 @@ dnl AM_PROG_LEX
 dnl AC_PROG_YACC
 
 dnl # Libtool stuff
-AC_DISABLE_STATIC
-AM_PROG_LIBTOOL
+LT_INIT([win32-dll disable-static])
 dnl AM_PROG_LIBTOOL makes AC_PROG_RANLIB unnecessary
 dnl striplib="";
 dnl old_striplib="";
--- origsrc/maloc/src/aaa_lib/Makefile.am	2011-04-02 12:16:42.275450100 +0900
+++ src/maloc/src/aaa_lib/Makefile.am	2011-04-02 12:18:19.715562100 +0900
@@ -55,9 +55,9 @@ lib_LTLIBRARIES = libmaloc.la ${LIBEFENC
 libmaloc_la_SOURCES =
 libmaloc_la_LIBADD  = ${BASE_LIBS} ${VSYS_LIBS} ${VSH_LIBS} ${PSH_LIBS}
 # libmaloc_la_LDFLAGS = -version-info ${FETK_VERSION}
-libmaloc_la_LDFLAGS = @rl_lib@ @math_lib@ -version-info ${FETK_VERSION}
+libmaloc_la_LDFLAGS = @rl_lib@ @math_lib@ -version-info ${FETK_VERSION} -no-undefined
 
 libefence_la_SOURCES =
 libefence_la_LIBADD  = ${EFENCE_LIBS}
-libefence_la_LDFLAGS = -version-info ${FETK_VERSION}
+libefence_la_LDFLAGS = -version-info ${FETK_VERSION} -no-undefined
 
