--- origsrc/libticonv-1.1.1/build/mingw/Makefile.am	2010-03-03 22:10:23.000000000 +0900
+++ src/libticonv-1.1.1/build/mingw/Makefile.am	2011-03-17 17:28:48.556471700 +0900
@@ -10,6 +10,6 @@ endif
 if OS_WIN32
 all: $(BASE)-rc.o
 
-$(BASE)-rc.o : $(BASE).rc
-	$(RC) $(BASE).rc $@
-endif
\ No newline at end of file
+$(BASE)-rc.o : $(srcdir)/$(BASE).rc
+	$(RC) $< $@
+endif
--- origsrc/libticonv-1.1.1/configure.ac	2010-03-10 22:45:12.000000000 +0900
+++ src/libticonv-1.1.1/configure.ac	2011-03-17 17:31:36.327714900 +0900
@@ -19,6 +19,9 @@ dnl    reset 'revision' and reset 'age'.
 LT_CURRENT=4
 LT_REVISION=4
 LT_AGE=0
+AC_SUBST(LT_CURRENT)
+AC_SUBST(LT_REVISION)
+AC_SUBST(LT_AGE)
 LT_LIBVERSION="$LT_CURRENT:$LT_REVISION:$LT_AGE"
 AC_SUBST(LT_LIBVERSION, $LT_CURRENT:$LT_REVISION:$LT_AGE)
 
@@ -41,8 +44,7 @@ AC_CONFIG_FILES([
 
 # Setup libtool.
 AC_DISABLE_STATIC
-AC_LIBTOOL_WIN32_DLL
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 # Checks for programs.
 AC_PROG_CC
@@ -91,7 +93,7 @@ CFLAGS="$CFLAGS $ARCH"
 # What flag to depends on gcc version: gcc3 uses "-mms-bitfields", while
 # gcc2 uses "-fnative-struct".
 case $host_os in
-*mingw*|*cygwin*)
+*mingw*)
   if test x$GCC = xyes; then
     msnative_struct=''
     AC_MSG_CHECKING([how to get MSVC-compatible struct packing])
--- origsrc/libticonv-1.1.1/tests/Makefile.am	2010-03-03 22:10:23.000000000 +0900
+++ src/libticonv-1.1.1/tests/Makefile.am	2011-03-17 17:41:54.086007700 +0900
@@ -2,6 +2,7 @@
 
 AM_CPPFLAGS = -I$(top_srcdir)/src @GLIB_CFLAGS@
 
-noinst_PROGRAMS = test_ticonv
+TESTS = test_ticonv
+check_PROGRAMS = test_ticonv
 
 test_ticonv_LDADD = $(top_builddir)/src/libticonv.la @GLIB_LIBS@
