--- origsrc/libticonv-1.1.4/configure.ac	2013-03-29 21:42:48.000000000 +0900
+++ src/libticonv-1.1.4/configure.ac	2015-05-19 23:39:30.484896200 +0900
@@ -92,7 +92,7 @@ CFLAGS="$CFLAGS $ARCH"
 # What flag to depends on gcc version: gcc3 uses "-mms-bitfields", while
 # gcc2 uses "-fnative-struct".
 case $host_os in
-*mingw*|*cygwin*)
+*mingw*)
   if test x$GCC = xyes; then
     msnative_struct=''
     AC_MSG_CHECKING([how to get MSVC-compatible struct packing])
--- origsrc/libticonv-1.1.4/tests/Makefile.am	2013-03-29 21:43:13.000000000 +0900
+++ src/libticonv-1.1.4/tests/Makefile.am	2015-05-19 23:39:30.500210200 +0900
@@ -7,4 +7,4 @@ noinst_PROGRAMS = test_ticonv torture_ti
 test_ticonv_LDADD = $(top_builddir)/src/libticonv.la @GLIB_LIBS@
 torture_ticonv_LDADD = $(top_builddir)/src/libticonv.la @GLIB_LIBS@
 
-TESTS = torture_ticonv
+TESTS = test_ticonv torture_ticonv
