--- origsrc/embedded_innodb-1.0.6.6750/Makefile.am	2010-02-22 21:57:21.000000000 +0900
+++ src/embedded_innodb-1.0.6.6750/Makefile.am	2010-12-04 14:21:05.255206400 +0900
@@ -27,13 +27,13 @@ embedded_configdir	= $(includedir)/$(INS
 
 IB_API_VERSION		= $(IB_API_VERSION_CURRENT):$(IB_API_VERSION_REVISION):$(IB_API_VERSION_AGE)
 
-libinnodb_la_LDFLAGS	= -version-info $(IB_API_VERSION)
+libinnodb_la_LDFLAGS	= -version-info $(IB_API_VERSION) -no-undefined
 
 # Note: Requires pkg-config to be installed
 #pkgconfigdir		= $(libdir)/pkgconfig
 #pkgconfig_DATA		= embedded-1.0.pc
 
-INCLUDES		= -I$(top_srcdir)/include -I$(top_builddir)/include
+INCLUDES		= -I$(top_builddir)/include -I$(top_srcdir)/include
 
 noinst_HEADERS 		=						\
 			include/btr0btr.h include/btr0btr.ic		\
--- origsrc/embedded_innodb-1.0.6.6750/configure.in	2010-02-22 21:57:21.000000000 +0900
+++ src/embedded_innodb-1.0.6.6750/configure.in	2010-12-04 11:55:47.639891200 +0900
@@ -24,7 +24,7 @@ AC_CONFIG_AUX_DIR(aconf)
 AC_CANONICAL_SYSTEM
 
 AC_PROG_CC
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 AC_SUBST(NM)
 
--- origsrc/embedded_innodb-1.0.6.6750/tests/Makefile.am	2010-02-22 21:57:00.000000000 +0900
+++ src/embedded_innodb-1.0.6.6750/tests/Makefile.am	2010-12-04 15:40:16.687532800 +0900
@@ -107,6 +107,6 @@ ib_logger_SOURCES	= test0aux.c ib_logger
 ib_update_SOURCES	= test0aux.c ib_update.c
 ib_zip_SOURCES		= test0aux.c ib_zip.c
 
-INCLUDES		= -I@top_srcdir@/include -I@top_srcdir@/tests
+INCLUDES		= -I@top_srcdir@ -I@top_srcdir@/include -I@top_srcdir@/tests
 
-LDADD			= @top_srcdir@/libinnodb.la -lpthread  -lm
+LDADD			= @top_builddir@/libinnodb.la -lpthread  -lm
