--- origsrc/cpp-rotor-0.23/CMakeLists.txt	2022-04-24 02:20:04.000000000 +0900
+++ src/cpp-rotor-0.23/CMakeLists.txt	2022-05-11 06:49:06.685116400 +0900
@@ -103,6 +103,10 @@ set_target_properties(rotor PROPERTIES
     CXX_STANDARD_REQUIRED YES
     CXX_EXTENSIONS NO
 )
+if (CYGWIN)
+    set_target_properties(rotor PROPERTIES SOVERSION "${CMAKE_PROJECT_VERSION_MAJOR}")
+endif()
+
 add_library(rotor::core ALIAS rotor)
 list(APPEND ROTOR_TARGETS_TO_INSTALL rotor)
 list(APPEND ROTOR_HEADERS_TO_INSTALL
@@ -164,7 +168,9 @@ if (BUILD_BOOST_ASIO)
     target_compile_definitions(rotor_asio PRIVATE ${ROTOR_ASIO_PRIVATE_FLAGS})
     target_link_libraries(rotor_asio PUBLIC rotor Threads::Threads)
     add_library(rotor::asio ALIAS rotor_asio)
-
+    if (CYGWIN)
+        set_target_properties(rotor_asio PROPERTIES SOVERSION "${CMAKE_PROJECT_VERSION_MAJOR}")
+    endif()
     list(APPEND ROTOR_TARGETS_TO_INSTALL rotor_asio)
     list(APPEND ROTOR_HEADERS_TO_INSTALL
         include/rotor/asio.hpp
@@ -190,6 +196,9 @@ if (BUILD_WX)
     )
     target_link_libraries(rotor_wx PUBLIC rotor ${wxWidgets_LIBRARIES})
     add_library(rotor::wx ALIAS rotor_wx)
+    if (CYGWIN)
+        set_target_properties(rotor_wx PROPERTIES SOVERSION "${CMAKE_PROJECT_VERSION_MAJOR}")
+    endif()
     list(APPEND ROTOR_TARGETS_TO_INSTALL rotor_wx)
     list(APPEND ROTOR_HEADERS_TO_INSTALL
         include/rotor/wx.hpp
@@ -214,6 +223,9 @@ if (BUILD_EV)
     target_include_directories(rotor_ev PUBLIC ${LIBEV_INCLUDE_DIRS})
     target_link_libraries(rotor_ev PUBLIC rotor ${LIBEV_LIBRARY})
     add_library(rotor::ev ALIAS rotor_ev)
+    if (CYGWIN)
+        set_target_properties(rotor_ev PROPERTIES SOVERSION "${CMAKE_PROJECT_VERSION_MAJOR}")
+    endif()
     list(APPEND ROTOR_TARGETS_TO_INSTALL rotor_ev)
     list(APPEND ROTOR_HEADERS_TO_INSTALL
         include/rotor/ev.hpp
@@ -237,6 +249,9 @@ if (BUILD_THREAD)
     )
     target_link_libraries(rotor_thread PUBLIC rotor Threads::Threads)
     add_library(rotor::thread ALIAS rotor_thread)
+    if (CYGWIN)
+        set_target_properties(rotor_thread PROPERTIES SOVERSION "${CMAKE_PROJECT_VERSION_MAJOR}")
+    endif()
     list(APPEND ROTOR_TARGETS_TO_INSTALL rotor_thread)
     list(APPEND ROTOR_HEADERS_TO_INSTALL
         include/rotor/thread.hpp
--- origsrc/cpp-rotor-0.23/examples/boost-asio/ping-pong-timer.cpp	2022-04-24 02:20:04.000000000 +0900
+++ src/cpp-rotor-0.23/examples/boost-asio/ping-pong-timer.cpp	2022-05-11 07:22:51.123200000 +0900
@@ -18,6 +18,7 @@
 #include <utility>
 #include <optional>
 #include <unordered_map>
+#include <thread>
 
 namespace asio = boost::asio;
 namespace pt = boost::posix_time;
