--- origsrc/libtreadstone-releases-0.2.0/Makefile.am	2015-05-27 00:15:42.000000000 +0900
+++ src/libtreadstone-releases-0.2.0/Makefile.am	2025-08-29 13:37:46.175241700 +0900
@@ -51,7 +51,7 @@ libtreadstone_la_SOURCES =
 libtreadstone_la_SOURCES += treadstone.cc
 libtreadstone_la_SOURCES += treadstone-validate.cc
 libtreadstone_la_LIBADD = $(E_LIBS)
-libtreadstone_la_LDFLAGS = -version-info 1:0:0
+libtreadstone_la_LDFLAGS = -no-undefined -version-info 1:0:0
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA =
