--- origsrc/libcdr-0.0.10/configure.ac	2013-01-24 17:56:57.000000000 +0900
+++ src/libcdr-0.0.10/configure.ac	2013-02-05 07:56:02.022996800 +0900
@@ -127,7 +127,7 @@ AC_SUBST(CMX2XHTML_WIN32_RESOURCE)
 
 AC_MSG_CHECKING([for Win32 platform in general])
 AS_CASE([$host],
-	[*-*-mingw*|*-*-cygwin*], [platform_win32=yes],
+	[*-*-mingw*], [platform_win32=yes],
 	[platform_win32=no]
 )
 AC_MSG_RESULT([$platform_win32])
