--- origsrc/SZ3-3.2.1/tools/H5Z-SZ3/CMakeLists.txt	2024-10-04 04:15:50.000000000 +0900
+++ src/SZ3-3.2.1/tools/H5Z-SZ3/CMakeLists.txt	2025-06-10 22:57:28.434709200 +0900
@@ -13,6 +13,10 @@ add_library(hdf5sz3
         src/H5Z_SZ3.cpp
 )
 
+if (DEFINED SOVERSION)
+        set_target_properties(hdf5sz3 PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 target_link_libraries(hdf5sz3
         PUBLIC SZ3 HDF5::HDF5
 )
--- origsrc/SZ3-3.2.1/tools/sz3c/CMakeLists.txt	2024-10-04 04:15:50.000000000 +0900
+++ src/SZ3-3.2.1/tools/sz3c/CMakeLists.txt	2025-06-10 22:40:18.657138400 +0900
@@ -3,6 +3,10 @@ add_library(
         src/sz3c.cpp
 )
 
+if (DEFINED SOVERSION)
+    set_target_properties(SZ3c PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 target_link_libraries(
         SZ3c
         PUBLIC SZ3
