--- origsrc/ktoblzcheck-1.45/configure.ac	2014-02-17 05:23:07.000000000 +0900
+++ src/ktoblzcheck-1.45/configure.ac	2014-04-01 13:29:31.557677500 +0900
@@ -64,7 +64,7 @@ AC_ARG_WITH(wget,
 AC_C_LONG_LONG
 
 # Check headers
-AC_CHECK_HEADERS(dirent.h windows.h)
+AC_CHECK_HEADERS(dirent.h)
 
 # Disable static libraries by default. Modern distros want the shared
 # ones.
@@ -88,6 +88,7 @@ AM_BINRELOC
 case "${target_os}" in
     mingw*)
 	AC_DEFINE(OS_WIN32, 1, [Defined if Windows is used])
+	AC_CHECK_HEADERS(windows.h)
 	;;
     *)
 	# Nothing to do
