--- origsrc/libiscsi-1.7.0/test-tool/iscsi-test.c	2012-11-05 08:41:01.000000000 +0900
+++ src/libiscsi-1.7.0/test-tool/iscsi-test.c	2012-12-23 20:28:40.886826700 +0900
@@ -18,7 +18,9 @@
 */
 
 #define _GNU_SOURCE
+#if !defined(__CYGWIN__)
 #include <sys/syscall.h>
+#endif
 #include <dlfcn.h>
 #include <sys/types.h>
 #include <stdio.h>
@@ -396,6 +398,9 @@ int main(int argc, const char *argv[])
 		POPT_TABLEEND
 	};
 
+#if !defined(RTLD_NEXT)
+#define RTLD_NEXT RTLD_DEFAULT
+#endif
 	real_iscsi_queue_pdu = dlsym(RTLD_NEXT, "iscsi_queue_pdu");
 
 	pc = poptGetContext(argv[0], argc, argv, popt_options, POPT_CONTEXT_POSIXMEHARDER);
