--- origsrc/cgreen-1.6.3/CMakeLists.txt	2023-09-27 02:25:00.000000000 +0900
+++ src/cgreen-1.6.3/CMakeLists.txt	2023-12-13 11:42:46.593862800 +0900
@@ -186,3 +186,10 @@ install(FILES
         "${CONFIG_INSTALL_DIR}" )
 
 #### End cgreen package configuration steps. ####
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/cgreen.pc.in ${CMAKE_BINARY_DIR}/cgreen.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/cgreen.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/cgreen-1.6.3/cgreen.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/cgreen-1.6.3/cgreen.pc.in	2023-12-13 11:42:46.593862800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: cgreen
+Description: Modern Unit Test and Mocking Framework for C and C++
+Version: @LIBRARY_VERSION@
+Libs: -L${libdir} -lcgreen
+Cflags: -I${includedir}
