--- origsrc/tree-sitter-css-0.23.1/CMakeLists.txt	2024-11-11 14:20:16.000000000 +0900
+++ src/tree-sitter-css-0.23.1/CMakeLists.txt	2024-11-12 19:45:30.506106200 +0900
@@ -17,12 +17,14 @@ endif()
 
 find_program(TREE_SITTER_CLI tree-sitter DOC "Tree-sitter CLI")
 
+if (NOT TREE_SITTER_CLI STREQUAL "TREE_SITTER_CLI-NOTFOUND")
 add_custom_command(OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/src/parser.c"
                    DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/src/grammar.json"
                    COMMAND "${TREE_SITTER_CLI}" generate src/grammar.json
                             --abi=${TREE_SITTER_ABI_VERSION}
                    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
                    COMMENT "Generating parser.c")
+endif()
 
 add_library(tree-sitter-css src/parser.c)
 if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/src/scanner.c)
@@ -41,15 +43,15 @@ set_target_properties(tree-sitter-css
                       SOVERSION "${TREE_SITTER_ABI_VERSION}.${PROJECT_VERSION_MAJOR}"
                       DEFINE_SYMBOL "")
 
+include(GNUInstallDirs)
+
 configure_file(bindings/c/tree-sitter-css.pc.in
                "${CMAKE_CURRENT_BINARY_DIR}/tree-sitter-css.pc" @ONLY)
 
-include(GNUInstallDirs)
-
 install(FILES bindings/c/tree-sitter-css.h
         DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/tree_sitter")
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/tree-sitter-css.pc"
-        DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/pkgconfig")
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 install(TARGETS tree-sitter-css
         LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 
