--- origsrc/tree-sitter-css-0.25.0/CMakeLists.txt	2025-09-28 20:34:26.000000000 +0900
+++ src/tree-sitter-css-0.25.0/CMakeLists.txt	2026-02-05 22:07:06.428850300 +0900
@@ -19,6 +19,7 @@ include(GNUInstallDirs)
 
 find_program(TREE_SITTER_CLI tree-sitter DOC "Tree-sitter CLI")
 
+if (NOT TREE_SITTER_CLI STREQUAL "TREE_SITTER_CLI-NOTFOUND")
 add_custom_command(OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/src/grammar.json"
                    DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/grammar.js"
                    COMMAND "${TREE_SITTER_CLI}" generate grammar.js
@@ -32,6 +33,7 @@ add_custom_command(OUTPUT "${CMAKE_CURRE
                             --emit=parser --abi=${TREE_SITTER_ABI_VERSION}
                    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
                    COMMENT "Generating parser.c")
+endif()
 
 add_library(tree-sitter-css src/parser.c)
 if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/src/scanner.c)
