--- origsrc/eatmemory-1.1.0/Makefile	2026-05-15 01:35:08.000000000 +0000
+++ src/eatmemory-1.1.0/Makefile	2026-08-11 07:42:44.873709200 +0000
@@ -18,7 +18,8 @@ VERSION ?= $(shell (git describe --tags
 # because $(CFLAGS) appears on the link command line below.
 EXTRA_CFLAGS ?=
 
-CFLAGS := -Wall -Wextra -std=c99 -O2 -g $(EXTRA_CFLAGS) -DVERSION='"$(VERSION)"'
+CFLAGS ?=
+CFLAGS += $(EXTRA_CFLAGS) -DVERSION='"$(VERSION)"'
 LDFLAGS :=
 
 # Prefer compiler target macros so cross-compilation selects the target
@@ -69,12 +70,12 @@ ifndef SYS_SRC
         SYS_BACKEND := Windows
     else
         SYS_SRC := $(SRC_DIR)/eatmemory.unknown.c
-        SYS_BACKEND := Unkown OS
+        SYS_BACKEND := Unknown OS
     endif
 endif
 SYS_BACKEND ?= $(patsubst eatmemory.%.c,%,$(notdir $(SYS_SRC)))
 ifeq ($(SYS_BACKEND),unknown)
-    SYS_BACKEND := Unkown OS
+    SYS_BACKEND := Unknown OS
 endif
 CFLAGS += -DSYSMEM_BACKEND='"$(SYS_BACKEND)"'
 
