--- origsrc/librope/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/librope/Makefile.am	2013-12-11 00:46:47.842066900 +0900
@@ -0,0 +1,16 @@
+
+AM_CFLAGS = -std=c99
+
+check_PROGRAMS = librope-test
+librope_test_CPPFLAGS = -I$(top_srcdir)/test
+librope_test_SOURCES = test/benchmark.c test/slowstring.c test/tests.c
+librope_test_LDADD = librope.la
+
+include_HEADERS = rope.h
+
+lib_LTLIBRARIES = librope.la
+librope_la_SOURCES = rope.c
+librope_la_LDFLAGS = -no-undefined
+
+TESTS = $(check_PROGRAMS)
+
--- origsrc/librope/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/librope/configure.ac	2013-12-11 00:46:17.592066900 +0900
@@ -0,0 +1,32 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([librope], [0+git20130723])
+AC_CONFIG_SRCDIR([rope.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stddef.h stdint.h stdlib.h string.h sys/time.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+AC_C_INLINE
+AC_TYPE_SIZE_T
+AC_TYPE_UINT16_T
+AC_TYPE_UINT8_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([gettimeofday memmove])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
