--- origsrc/librope-1.0.0/CMakeLists.txt	1970-01-01 09:00:00.000000000 +0900
+++ src/librope-1.0.0/CMakeLists.txt	2025-11-17 20:42:46.527719500 +0900
@@ -0,0 +1,23 @@
+cmake_minimum_required(VERSION 3.28)
+project(librope
+  VERSION 1.0.0
+  DESCRIPTION "UTF-8 rope library for C"
+  HOMEPAGE_URL "https://github.com/josephg/librope")
+
+include(GNUInstallDirs)
+
+add_library(rope SHARED rope.c)
+set_target_properties(rope PROPERTIES SOVERSION ${PROJECT_VERSION_MAJOR})
+
+install(TARGETS rope
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+install(FILES rope.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/librope.pc.in ${CMAKE_BINARY_DIR}/librope.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/librope.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/librope-1.0.0/librope.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/librope-1.0.0/librope.pc.in	2025-11-17 20:42:58.250603000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: librope
+Description: UTF-8 rope library for C
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lrope
+Cflags: -I${includedir}
--- origsrc/librope-1.0.0/rope.c	2014-12-13 09:46:22.000000000 +0900
+++ src/librope-1.0.0/rope.c	2025-11-17 20:35:26.061971200 +0900
@@ -3,6 +3,8 @@
 #include <stdlib.h>
 #include <string.h>
 
+#include <sys/types.h>
+
 // Needed for VC++, which always compiles in C++ mode and doesn't have stdbool.
 #ifndef __cplusplus
 #include <stdbool.h>
