--- origsrc/bcftools-1.19/Makefile	2023-12-12 23:02:47.000000000 +0900
+++ src/bcftools-1.19/Makefile	2024-01-19 20:21:36.595160000 +0900
@@ -45,7 +45,7 @@ OBJS     = main.o vcfindex.o tabix.o \
            ccall.o em.o prob1.o kmin.o str_finder.o gff.o
 PLUGIN_OBJS = vcfplugin.o
 
-prefix      = /usr/local
+prefix      = /usr
 exec_prefix = $(prefix)
 bindir      = $(exec_prefix)/bin
 libdir      = $(exec_prefix)/lib
--- origsrc/bcftools-1.19/misc/guess-ploidy.py	2023-12-12 23:02:47.000000000 +0900
+++ src/bcftools-1.19/misc/guess-ploidy.py	2024-01-19 20:21:36.595160000 +0900
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Plot the output of "bcftools +guess-ploidy -v"
 #
