--- origsrc/bcftools-1.16/Makefile	2022-08-18 19:56:00.000000000 +0900
+++ src/bcftools-1.16/Makefile	2023-01-16 14:39:32.244971000 +0900
@@ -45,7 +45,7 @@ OBJS     = main.o vcfindex.o tabix.o \
            ccall.o em.o prob1.o kmin.o str_finder.o
 PLUGIN_OBJS = vcfplugin.o
 
-prefix      = /usr/local
+prefix      = /usr
 exec_prefix = $(prefix)
 bindir      = $(exec_prefix)/bin
 libdir      = $(exec_prefix)/lib
--- origsrc/bcftools-1.16/misc/guess-ploidy.py	2022-08-18 19:56:00.000000000 +0900
+++ src/bcftools-1.16/misc/guess-ploidy.py	2023-01-16 15:34:02.863766400 +0900
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Plot the output of "bcftools +guess-ploidy -v"
 #
