--- origsrc/bcftools-1.11/Makefile	2020-09-22 21:16:14.000000000 +0900
+++ src/bcftools-1.11/Makefile	2020-10-01 10:03:01.039593600 +0900
@@ -45,7 +45,7 @@ OBJS     = main.o vcfindex.o tabix.o \
            ccall.o em.o prob1.o kmin.o # the original samtools calling
 PLUGIN_OBJS = vcfplugin.o
 
-prefix      = /usr/local
+prefix      = /usr
 exec_prefix = $(prefix)
 bindir      = $(exec_prefix)/bin
 libdir      = $(exec_prefix)/lib
