--- origsrc/bcftools-1.22/Makefile	2025-05-30 17:18:34.000000000 +0900
+++ src/bcftools-1.22/Makefile	2025-06-19 14:59:41.646985400 +0900
@@ -47,7 +47,7 @@ OBJS     = main.o vcfindex.o tabix.o \
            mpileup2/mpileup.o
 PLUGIN_OBJS = vcfplugin.o
 
-prefix      = /usr/local
+prefix      = /usr
 exec_prefix = $(prefix)
 bindir      = $(exec_prefix)/bin
 libdir      = $(exec_prefix)/lib
--- origsrc/bcftools-1.22/misc/guess-ploidy.py	2025-05-30 17:18:34.000000000 +0900
+++ src/bcftools-1.22/misc/guess-ploidy.py	2025-06-19 14:59:41.646985400 +0900
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Plot the output of "bcftools +guess-ploidy -v"
 #
