--- origsrc/libcircle-0.2.0-rc.1/Makefile.am	2014-02-14 01:35:37.000000000 +0900
+++ src/libcircle-0.2.0-rc.1/Makefile.am	2014-03-19 07:37:52.066666600 +0900
@@ -8,7 +8,7 @@ ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 lib_LTLIBRARIES = libcircle.la
 
 libcircle_la_LIBADD = $(MPI_CLDFLAGS)
-libcircle_la_LDFLAGS = -version-info $(LIBCIRCLE_SO_VERSION)
+libcircle_la_LDFLAGS = -no-undefined -version-info $(LIBCIRCLE_SO_VERSION)
 libcircle_la_SOURCES = libcircle/lib.c    \
                                                 libcircle/queue.c  \
                                                 libcircle/worker.c \
--- origsrc/libcircle-0.2.0-rc.1/tests/Makefile.am	2014-02-14 01:35:37.000000000 +0900
+++ src/libcircle-0.2.0-rc.1/tests/Makefile.am	2014-03-19 07:40:36.098893200 +0900
@@ -2,7 +2,7 @@ TESTS = check_queue #check_checkpoint
 check_PROGRAMS = check_queue #check_checkpoint
 
 check_queue_SOURCES = check_queue.c $(top_srcdir)/libcircle.h
-check_queue_CFLAGS = -I$(top_srcdir)/libcircle/ @CHECK_CFLAGS@
+check_queue_CFLAGS = -I$(top_srcdir)/libcircle/ -I$(top_srcdir) @CHECK_CFLAGS@
 check_queue_LDADD = $(top_builddir)/libcircle.la @CHECK_LIBS@
 
 #check_checkpoint_SOURCES = check_checkpoint.c $(top_srcdir)/libcircle.h
