--- origsrc/libcircle-0.3/Makefile.am	2020-01-08 05:20:20.000000000 +0900
+++ src/libcircle-0.3/Makefile.am	2020-11-25 09:14:37.000019500 +0900
@@ -13,7 +13,7 @@ ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 lib_LTLIBRARIES = libcircle.la
 
 libcircle_la_LIBADD = $(MPI_CLDFLAGS)
-libcircle_la_LDFLAGS = -version-info $(LIBCIRCLE_SO_VERSION)
+libcircle_la_LDFLAGS = -no-undefined -version-info $(LIBCIRCLE_SO_VERSION)
 libcircle_la_SOURCES = \
         libcircle/lib.c \
         libcircle/queue.c \
--- origsrc/libcircle-0.3/m4/lx_find_mpi.m4	2020-01-08 05:20:20.000000000 +0900
+++ src/libcircle-0.3/m4/lx_find_mpi.m4	2020-11-25 09:24:56.532611400 +0900
@@ -126,6 +126,7 @@ AC_DEFUN([LX_QUERY_MPI_COMPILER],
      lx_mpi_compile_line=`$$1 -showme:compile 2>/dev/null`
      if [[ "$?" -eq 0 ]]; then
          echo yes
+         lx_mpi_compile_line="-I/usr/include" # dummy
          lx_mpi_link_line=`$$1 -showme:link 2>/dev/null`
      else
          echo no
