--- origsrc/colibri-core-2.5.9/src/Makefile.am	2023-07-03 18:52:48.000000000 +0900
+++ src/colibri-core-2.5.9/src/Makefile.am	2023-07-27 13:57:03.379864300 +0900
@@ -1,6 +1,6 @@
 AM_CXXFLAGS = -Wall -O3 -std=c++11 -Wno-vla -fPIC
 AM_CPPFLAGS = -I@top_srcdir@/include
-AM_LDFLAGS = -static -fPIC
+AM_LDFLAGS = -fPIC
 
 bin_PROGRAMS = colibri-classdecode colibri-classencode colibri-test colibri-benchmarks colibri-patternmodeller colibri-comparemodels colibri-extractngrams
 
@@ -8,6 +8,7 @@ LDADD = libcolibricore.la
 lib_LTLIBRARIES = libcolibricore.la
 
 libcolibricore_la_SOURCES = SpookyV2.cpp common.cpp algorithms.cpp classdecoder.cpp classencoder.cpp pattern.cpp patternmodel.cpp
+libcolibricore_la_LDFLAGS = -no-undefined
 
 colibri_classdecode_SOURCES = classdecode.cpp
 colibri_classencode_SOURCES = classencode.cpp
