--- origsrc/libhttpserver-0.17.5/Makefile.am	2019-01-28 15:02:39.000000000 +0900
+++ src/libhttpserver-0.17.5/Makefile.am	2019-02-23 14:57:29.119998200 +0900
@@ -40,7 +40,7 @@ DISTCLEANFILES = DIST_REVISION
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libhttpserver.pc
 
-cmakemoduledir = $(datadir)/cmake/Modules
+cmakemoduledir = $(libdir)/cmake/$(PACKAGE)
 cmakemodule_DATA = cmakemodule/FindLibHttpServer.cmake
 
 include $(top_srcdir)/aminclude.am
--- origsrc/libhttpserver-0.17.5/configure.ac	2019-01-28 15:02:39.000000000 +0900
+++ src/libhttpserver-0.17.5/configure.ac	2019-02-23 14:57:29.135558800 +0900
@@ -54,7 +54,7 @@ AC_ARG_ENABLE([same-directory-build],
 AC_MSG_RESULT([$samedirectory])
 
 if test x"$samedirectory" = x"no"; then
-    if test "`cd $srcdir; /bin/pwd`" = "`/bin/pwd`"; then 
+    if test "`cd $srcdir; /bin/pwd`" = "`/bin/pwd`"; then
         AC_MSG_ERROR("you must configure in a separate build directory")
     fi
 fi
@@ -66,10 +66,6 @@ case "$host" in
         REGEX_LIBS="-lregex -lpthread -no-undefined"
         native_srcdir=$(cd $srcdir; pwd -W)
         ;;
-    *-cygwin*)
-        NETWORK_HEADER="winsock2.h"
-        REGEX_LIBS="-lregex -lpthread -no-undefined"
-        ;;
     *)
         NETWORK_HEADER="arpa/inet.h"
         REGEX_LIBS=""
@@ -352,7 +348,7 @@ AC_OUTPUT(
       examples/Makefile
 )
 
-AC_MSG_NOTICE([Configuration Summary:  
+AC_MSG_NOTICE([Configuration Summary:
   Operating System:  ${host_os}
   Target directory:  ${prefix}
   License         :  LGPL only
--- origsrc/libhttpserver-0.17.5/src/webserver.cpp	2019-01-28 15:02:39.000000000 +0900
+++ src/libhttpserver-0.17.5/src/webserver.cpp	2019-02-23 15:05:01.921272600 +0900
@@ -30,7 +30,7 @@
 #include <sys/time.h>
 #include <stdexcept>
 
-#if defined(__MINGW32__) || defined(__CYGWIN32__)
+#if defined(__MINGW32__)
 #include <winsock2.h>
 #define _WINDOWS
 #else
@@ -41,6 +41,7 @@
 #include <fcntl.h>
 #include <algorithm>
 
+#include <sys/select.h>
 #include <microhttpd.h>
 
 #include "gettext.h"
