--- origsrc/libhttpserver-0.18.1/Makefile.am	2020-06-07 02:24:46.000000000 +0900
+++ src/libhttpserver-0.18.1/Makefile.am	2020-08-24 11:11:47.481523900 +0900
@@ -46,7 +46,7 @@ DISTCLEANFILES = DIST_REVISION
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libhttpserver.pc
 
-cmakemoduledir = $(datadir)/cmake/Modules
+cmakemoduledir = $(libdir)/cmake/$(PACKAGE)
 cmakemodule_DATA = cmakemodule/FindLibHttpServer.cmake
 
 include $(top_srcdir)/aminclude.am
