--- origsrc/falco-1.3.2/configure.ac	2026-06-10 05:49:47.000000000 +0900
+++ src/falco-1.3.2/configure.ac	2026-06-26 12:45:04.897548100 +0900
@@ -25,12 +25,6 @@ AC_CONFIG_MACRO_DIR([m4])
 AC_LANG(C++)
 AC_PROG_CXX
 
-# Set CXXFLAGS only if the user hasn't set them
-AS_IF([test "x$CXXFLAGS_set" != "xset"], [
-  CXXFLAGS="-O3 -DNDEBUG"
-])
-AC_MSG_NOTICE([CXXFLAGS is $CXXFLAGS])
-
 AC_DEFINE_UNQUOTED([PROGRAM_PATH], ["$srcdir"],
   [The PROGRAM_PATH variable])
 
--- origsrc/falco-1.3.2/src/Module.cpp	2026-06-10 05:49:47.000000000 +0900
+++ src/falco-1.3.2/src/Module.cpp	2026-06-26 12:49:17.649478500 +0900
@@ -319,7 +319,7 @@ Module::Module(const std::string &module
   placeholder.erase(
     std::remove_if(std::begin(placeholder), std::end(placeholder),
                    [](const auto c) { return std::isspace(c); }),
-    std::cend(placeholder));
+    std::end(placeholder));
 
   // lowercases it
   std::transform(std::cbegin(placeholder), std::cend(placeholder),
