--- origsrc/gupnp-dlna-0.9.3/libgupnp-dlna/metadata-backends/gstreamer/gstreamer.am	2013-01-12 00:40:53.000000000 +0900
+++ src/gupnp-dlna-0.9.3/libgupnp-dlna/metadata-backends/gstreamer/gstreamer.am	2013-01-13 14:58:39.038278600 +0900
@@ -16,8 +16,8 @@ metadata_LTLIBRARIES += \
 	$(gstreamer_metadata_backend)
 
 metadata_backends_gstreamer_libgstreamer_la_LDFLAGS = \
-	$(convenience_ldflags) \
-	-shared -fPIC -module -avoid-version -Wl,-z,defs
+	$(convenience_ldflags) -no-undefined \
+	-shared -fPIC -module -avoid-version
 metadata_backends_gstreamer_libgstreamer_la_SOURCES = \
 	metadata-backends/gstreamer/gupnp-dlna-gst-audio-information.c \
 	metadata-backends/gstreamer/gupnp-dlna-gst-container-information.c \
