--- origsrc/svrcore-4.0.4/configure.ac	2007-03-13 23:33:00.000000000 +0900
+++ src/svrcore-4.0.4/configure.ac	2010-10-16 16:04:50.369037800 +0900
@@ -9,7 +9,7 @@ AC_CONFIG_SRCDIR([src/svrcore.h])
 AC_CONFIG_HEADER([config.h])
 
 AC_PROG_CC
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 # Checks for header files.
 AC_HEADER_STDC
--- origsrc/svrcore-4.0.4/src/Makefile.am	2006-12-15 02:07:32.000000000 +0900
+++ src/svrcore-4.0.4/src/Makefile.am	2010-10-16 16:04:34.212787800 +0900
@@ -17,5 +17,5 @@ libsvrcore_la_SOURCES =	alt.c \
 			std.c \
 			user.c
 
-libsvrcore_la_LDFLAGS = -version-info 0:0:0 @NSS_LIBS@ @NSPR_LIBS@
+libsvrcore_la_LDFLAGS = -version-info 0:0:0 @NSS_LIBS@ @NSPR_LIBS@ -no-undefined
 libsvrcore_la_CFLAGS = @NSS_CFLAGS@ @NSPR_CFLAGS@
