--- origsrc/svrcore-4.0.4/src/Makefile.am	2006-12-15 02:07:32.000000000 +0900
+++ src/svrcore-4.0.4/src/Makefile.am	2015-03-10 22:52:35.933242900 +0900
@@ -17,5 +17,5 @@ libsvrcore_la_SOURCES =	alt.c \
 			std.c \
 			user.c
 
-libsvrcore_la_LDFLAGS = -version-info 0:0:0 @NSS_LIBS@ @NSPR_LIBS@
+libsvrcore_la_LDFLAGS = -no-undefined -version-info 0:0:0 @NSS_LIBS@ @NSPR_LIBS@
 libsvrcore_la_CFLAGS = @NSS_CFLAGS@ @NSPR_CFLAGS@
