--- origsrc/yapet-1.1/libyacurs/tests/Makefile.am	2018-02-10 20:56:08.000000000 +0900
+++ src/yapet-1.1/libyacurs/tests/Makefile.am	2018-04-13 23:48:44.115245600 +0900
@@ -1,4 +1,4 @@
 # $Id$
 
-SUBDIRS = preloadlib basic ui-interactive ui-non-interactive
+SUBDIRS = basic ui-interactive ui-non-interactive
 
--- origsrc/yapet-1.1/tests/Makefile.am	2018-02-10 20:56:06.000000000 +0900
+++ src/yapet-1.1/tests/Makefile.am	2018-04-13 23:48:21.661000000 +0900
@@ -3,8 +3,6 @@
 # Tests may fail if not run serial
 AUTOMAKE_OPTIONS = serial-tests
 
-SUBDIRS = preload
-
 # For silent rules
 cpy_verbose = $(cpy_verbose_$(V))
 cpy_verbose_ = $(cpy_verbose_$(AM_DEFAULT_VERBOSITY))
--- origsrc/yapet-1.1/yapet/Makefile.am	2018-02-10 20:56:06.000000000 +0900
+++ src/yapet-1.1/yapet/Makefile.am	2018-04-13 23:46:36.693000000 +0900
@@ -16,8 +16,6 @@ ui_sources = ../ui/help.cc ../ui/mainwin
 ../ui/yapetunlockdialog.cc					\
 ../ui/yapetlockscreen.cc ../ui/pwgendialog.cc
 
-yapet_LDFLAGS = -static
-
 yapet_SOURCES = $(pwgen_sources) $(glue_sources) $(ui_sources)		\
 main.cc cfgfile.cc consts.cc cfg.cc globals.cc cfgfile.h consts.h	\
 cfg.h globals.h
