--- origsrc/liblcf-0.7.0/Makefile.am	2021-10-30 01:42:43.000000000 +0900
+++ src/liblcf-0.7.0/Makefile.am	2021-11-17 11:37:23.789636800 +0900
@@ -31,17 +31,18 @@ liblcf_la_CXXFLAGS = \
 	-fno-math-errno \
 	$(AM_CXXFLAGS) \
 	$(EXPAT_CFLAGS) \
-	$(ICU_CFLAGS)
+	$(ICU_CFLAGS) \
+	$(INIH_CFLAGS)
 liblcf_la_LIBADD = \
 	$(EXPAT_LIBS) \
-	$(ICU_LIBS)
+	$(ICU_LIBS) \
+	$(INIH_LIBS)
 liblcf_la_LDFLAGS = \
 	$(AM_LDFLAGS) \
 	-no-undefined
 liblcf_la_SOURCES = \
 	src/dbarray.cpp \
 	src/encoder.cpp \
-	src/ini.cpp \
 	src/inireader.cpp \
 	src/ldb_equipment.cpp \
 	src/ldb_eventcommand.cpp \
@@ -206,7 +207,6 @@ lcfinclude_HEADERS = \
 	src/lcf/encoder.h \
 	src/lcf/enum_tags.h \
 	src/lcf/flag_set.h \
-	src/lcf/ini.h \
 	src/lcf/inireader.h \
 	src/lcf/reader_lcf.h \
 	src/lcf/reader_util.h \
--- origsrc/liblcf-0.7.0/builds/liblcf.pc.in	2021-10-30 01:42:43.000000000 +0900
+++ src/liblcf-0.7.0/builds/liblcf.pc.in	2021-11-17 11:36:36.622087500 +0900
@@ -7,6 +7,6 @@ Name: @PACKAGE_TARNAME@
 Description: Library to handle RPG Maker 2000 and 2003 game data
 Version: @PACKAGE_VERSION@
 URL: https://easyrpg.org/
-Requires.private: @AX_PACKAGE_REQUIRES_PRIVATE@
+Requires.private: @AX_PACKAGE_REQUIRES_PRIVATE@ inih
 Libs: -L${libdir} -llcf
 Cflags: -I${includedir}
--- origsrc/liblcf-0.7.0/configure.ac	2021-10-30 01:42:43.000000000 +0900
+++ src/liblcf-0.7.0/configure.ac	2021-11-17 11:34:16.726985600 +0900
@@ -49,6 +49,9 @@ AS_IF([test "x$enable_update_mimedb" !=
 ])
 AM_CONDITIONAL(ENABLE_UPDATE_MIMEDB,[test x$enable_update_mimedb != xno])
 
+# Dependencies
+PKG_CHECK_MODULES([INIH], [inih])
+
 # Checks for header files.
 AC_CHECK_HEADERS([stdint.h],[],[AC_MSG_ERROR([cannot find stdint.h, bailing out])])
 
--- origsrc/liblcf-0.7.0/src/inireader.cpp	2021-10-30 01:42:43.000000000 +0900
+++ src/liblcf-0.7.0/src/inireader.cpp	2021-11-17 11:35:19.728772200 +0900
@@ -34,7 +34,7 @@
 #include <cstdlib>
 #include <cstring>
 #include <istream>
-#include "lcf/ini.h"
+#include <ini.h>
 #include "lcf/inireader.h"
 
 namespace lcf {
