--- origsrc/liblcf-0.8/Makefile.am	2023-04-30 00:40:07.000000000 +0900
+++ src/liblcf-0.8/Makefile.am	2023-05-19 12:21:02.783510200 +0900
@@ -35,10 +35,12 @@ liblcf_la_CXXFLAGS = \
 	-fno-math-errno \
 	$(AM_CXXFLAGS) \
 	$(EXPAT_CFLAGS) \
-	$(ICU_CFLAGS)
+	$(ICU_CFLAGS) \
+	$(INIH_CFLAGS)
 liblcf_la_LIBADD = \
 	$(EXPAT_LIBS) \
-	$(ICU_LIBS)
+	$(ICU_LIBS) \
+	$(INIH_LIBS)
 liblcf_la_LDFLAGS = \
 	$(AM_LDFLAGS) \
 	-no-undefined
@@ -46,7 +48,6 @@ liblcf_la_SOURCES = \
 	src/dbarray.cpp \
 	src/dbstring_struct.cpp \
 	src/encoder.cpp \
-	src/ini.cpp \
 	src/inireader.cpp \
 	src/ldb_equipment.cpp \
 	src/ldb_eventcommand.cpp \
@@ -217,7 +218,6 @@ lcfinclude_HEADERS = \
 	src/lcf/encoder.h \
 	src/lcf/enum_tags.h \
 	src/lcf/flag_set.h \
-	src/lcf/ini.h \
 	src/lcf/inireader.h \
 	src/lcf/reader_lcf.h \
 	src/lcf/reader_util.h \
--- origsrc/liblcf-0.8/configure.ac	2023-04-30 00:40:07.000000000 +0900
+++ src/liblcf-0.8/configure.ac	2023-05-19 12:20:51.740085300 +0900
@@ -49,6 +49,9 @@ AS_IF([test "x$enable_update_mimedb" !=
 ])
 AM_CONDITIONAL(ENABLE_UPDATE_MIMEDB,[test x$enable_update_mimedb != xno])
 
+# Dependencies
+PKG_CHECK_MODULES([INIH], [inih])
+
 # Checks for header files.
 AC_CHECK_HEADERS([stdint.h],[],[AC_MSG_ERROR([cannot find stdint.h, bailing out])])
 
--- origsrc/liblcf-0.8/src/inireader.cpp	2023-04-30 00:40:07.000000000 +0900
+++ src/liblcf-0.8/src/inireader.cpp	2023-05-19 12:21:18.986683400 +0900
@@ -34,7 +34,7 @@
 #include <cstdlib>
 #include <cstring>
 #include <istream>
-#include "lcf/ini.h"
+#include <ini.h>
 #include "lcf/inireader.h"
 
 namespace lcf {
