--- origsrc/libvdwxc-0.4.0/src/Makefile.am	2019-03-12 22:11:42.000000000 +0900
+++ src/libvdwxc-0.4.0/src/Makefile.am	2025-08-21 00:43:01.830937400 +0900
@@ -9,6 +9,7 @@ libvdwxc_la_SOURCES = \
   vdw_kernel.c \
   vdw_core.c \
   vdw_q0.c
+libvdwxc_la_LDFLAGS = -no-undefined
 
 if DO_BUILD_MPI
 libvdwxc_la_SOURCES += vdw_mpi.c
@@ -19,6 +20,7 @@ endif
 
 libvdwxcfort_la_SOURCES = vdw_fortran.c
 libvdwxcfort_la_LIBADD = libvdwxc.la
+libvdwxcfort_la_LDFLAGS = -no-undefined
 
 # Main headers
 include_HEADERS = \
