--- origsrc/ifstat-1.1/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/ifstat-1.1/Makefile.am	2011-05-04 12:24:38.654804300 +0900
@@ -0,0 +1,15 @@
+lib_LTLIBRARIES = libifstat.la
+
+libifstat_la_SOURCES = drivers.c data.c snmp.c
+libifstat_la_LDFLAGS = -no-undefined
+
+bin_PROGRAMS = ifstat
+ifstat_SOURCES = ifstat.c
+ifstat_LDADD = libifstat.la
+
+ifstat_includedir = $(includedir)/ifstat
+ifstat_include_HEADERS = ifstat.h snmp.h
+
+man_MANS = ifstat.1
+
+bin_SCRIPTS = libifstat-config
--- origsrc/ifstat-1.1/configure.in	2003-11-22 10:27:51.000000000 +0900
+++ src/ifstat-1.1/configure.in	2011-05-04 12:29:41.350058700 +0900
@@ -2,6 +2,7 @@ dnl configure.in for ifstat
 dnl Process this file with autoconf to create configure
 dnl $Id: configure.in,v 1.51 2003/11/22 01:27:51 gael Exp $
 AC_INIT(ifstat.c)
+AM_INIT_AUTOMAKE(ifstat, 1.1)
 
 AC_ARG_ENABLE(debug,   [  --enable-debug          setup flags (gcc) for debugging (default=no)],
 CFLAGS="$CFLAGS -O2 -Wall -g"
@@ -20,7 +21,7 @@ AC_CONFIG_HEADER(config.h)
 dnl Programs
 AC_PROG_INSTALL
 AC_PROG_CC
-AC_PROG_RANLIB
+LT_INIT([win32-dll disable-static])
 
 AC_HEADER_STDC
 AC_CHECK_HEADERS(sys/time.h)
@@ -488,4 +489,4 @@ AC_SUBST(CPPFLAGS)
 AC_SUBST(LDFLAGS)
 AC_SUBST(LIBTARGET)
 
-AC_OUTPUT(Makefile)
+AC_OUTPUT(Makefile libifstat-config)
