--- origsrc/subunit-1.4.2/Makefile.am	2022-11-17 08:48:50.000000000 +0900
+++ src/subunit-1.4.2/Makefile.am	2024-08-11 17:04:46.585109300 +0900
@@ -99,10 +99,15 @@ libsubunit_la_SOURCES = \
 	c/lib/child.c \
 	c/include/subunit/child.h
 
+libsubunit_la_LDFLAGS = -no-undefined
+
 libcppunit_subunit_la_SOURCES = \
 	c++/SubunitTestProgressListener.cpp \
 	c++/SubunitTestProgressListener.h
 
+libcppunit_subunit_la_LDFLAGS = -no-undefined
+libcppunit_subunit_la_LIBADD = $(CPPUNIT_LIBS) libsubunit.la
+
 tests_LDADD = @CHECK_LIBS@ $(top_builddir)/libsubunit.la
 c_tests_test_child_CFLAGS = -I$(top_srcdir)/c/include $(SUBUNIT_CFLAGS) @CHECK_CFLAGS@
 c_tests_test_child_LDADD = $(tests_LDADD)
