--- origsrc/libconflate/Makefile.am	2010-12-24 00:41:23.000000000 +0900
+++ src/libconflate/Makefile.am	2010-12-24 00:43:37.448568600 +0900
@@ -37,6 +37,7 @@ libconflate_la_SOURCES = adhoc_commands.
                          xmpp.c
 
 libconflate_la_LIBADD = $(LIBCONFLATE_LINKFLAGS) $(LTLIBCURL)
+libconflate_la_LDFLAGS = -no-undefined
 
 MOSTLYCLEANFILES = *.gcov *.gcno *.gcda *.tcov
 
@@ -45,4 +46,4 @@ docs:
 	(cat Doxyfile ; echo "PROJECT_NUMBER=`git describe`") | doxygen -
 
 test: check
-	@for f in *.gcda; do gcov $$f; done | egrep -v ':creating'
\ No newline at end of file
+	@for f in *.gcda; do gcov $$f; done | egrep -v ':creating'
--- origsrc/libconflate/configure.ac	2010-12-24 00:41:23.000000000 +0900
+++ src/libconflate/configure.ac	2010-12-24 00:53:36.209228300 +0900
@@ -5,9 +5,9 @@ AC_CONFIG_SRCDIR([conflate.h])
 AC_CONFIG_AUX_DIR(config)
 AM_CONFIG_HEADER([config.h])
 AC_CONFIG_MACRO_DIR([m4])
-AC_DISABLE_STATIC
 ac_cv_func_malloc_0_nonnull=yes
 ac_cv_func_realloc_0_nonnull=yes
+LT_INIT(win32-dll)
 PANDORA_CANONICAL_TARGET(less-warnings, force-gcc42)
 PANDORA_REQUIRE_PTHREAD
 PANDORA_HAVE_LIBCURL
