--- origsrc/libconflate/Makefile.am	2012-02-16 09:03:14.000000000 +0900
+++ src/libconflate/Makefile.am	2012-02-16 09:04:46.490345700 +0900
@@ -38,6 +38,8 @@ libconflate_la_SOURCES = adhoc_commands.
 
 libconflate_la_LIBADD = $(LIBCONFLATE_LINKFLAGS) $(LTLIBCURL)
 
+libconflate_la_LDFLAGS = -no-undefined
+
 MOSTLYCLEANFILES = *.gcov *.gcno *.gcda *.tcov
 
 .PHONY: docs
@@ -45,4 +47,4 @@ docs:
 	(cat Doxyfile ; echo "PROJECT_NUMBER=`git describe`") | doxygen -
 
 test: check
-	@for f in *.gcda; do gcov $$f; done | egrep -v ':creating'
\ No newline at end of file
+	@for f in *.gcda; do gcov $$f; done | egrep -v ':creating'
--- origsrc/libconflate/configure.ac	2012-02-16 09:03:14.000000000 +0900
+++ src/libconflate/configure.ac	2012-02-16 09:05:23.316082900 +0900
@@ -9,6 +9,7 @@ AC_DISABLE_STATIC
 ac_cv_func_malloc_0_nonnull=yes
 ac_cv_func_realloc_0_nonnull=yes
 PANDORA_CANONICAL_TARGET(less-warnings, force-gcc42)
+PANDORA_LIBTOOL
 PANDORA_REQUIRE_PTHREAD
 PANDORA_HAVE_LIBCURL
 PANDORA_HAVE_LIBSQLITE3
--- origsrc/libconflate/m4/version.m4	1970-01-01 09:00:00.000000000 +0900
+++ src/libconflate/m4/version.m4	2012-02-16 09:07:37.054480200 +0900
@@ -0,0 +1 @@
+m4_define([VERSION_NUMBER], [1.7.1])
