--- origsrc/yomm2-1.1.2/CMakeLists.txt	2020-12-27 05:38:24.000000000 +0900
+++ src/yomm2-1.1.2/CMakeLists.txt	2021-06-14 14:52:31.062523000 +0900
@@ -6,12 +6,16 @@
 cmake_minimum_required(VERSION 3.5)
 cmake_policy(SET CMP0057 NEW)
 
-project(YOMM2 VERSION 1.0)
+project(YOMM2 VERSION 1.1.2)
 
 # Find Boost dependency
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
-include(find_or_download_package)
-find_or_download_package(Boost INSTALL_WITH_YOMM)
+if (CYGWIN)
+  find_package(boost)
+else()
+  include(find_or_download_package)
+  find_or_download_package(Boost INSTALL_WITH_YOMM)
+endif()
 message(STATUS "Using Boost libraries from ${Boost_INCLUDE_DIRS}")
 
 if(CMAKE_COMPILER_IS_GNUCXX OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
@@ -77,3 +81,10 @@ install(FILES
 )
 
 install(DIRECTORY include/yorel DESTINATION include)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/yomm2.pc.in ${CMAKE_BINARY_DIR}/yomm2.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/yomm2.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/yomm2-1.1.2/examples/CMakeLists.txt	2020-12-27 05:38:24.000000000 +0900
+++ src/yomm2-1.1.2/examples/CMakeLists.txt	2021-06-14 13:02:18.734105100 +0900
@@ -30,7 +30,7 @@ add_test(NAME asteroids COMMAND asteroid
 add_subdirectory(containers)
 add_test(NAME containers COMMAND containers)
 
-if(NOT MSVC)
+if(NOT (MSVC OR CYGWIN))
   set(CMAKE_SKIP_BUILD_RPATH TRUE)
   add_executable(dl_main dl_main.cpp)
   add_library(dl_shared SHARED dl_shared.cpp)
--- origsrc/yomm2-1.1.2/src/CMakeLists.txt	2020-12-27 05:38:24.000000000 +0900
+++ src/yomm2-1.1.2/src/CMakeLists.txt	2021-06-14 13:01:36.177728800 +0900
@@ -3,7 +3,12 @@
 # See accompanying file LICENSE_1_0.txt
 # or copy at http://www.boost.org/LICENSE_1_0.txt)
 
-add_library(yomm2 STATIC yomm2.cpp)
+if (CYGWIN)
+  add_library(yomm2 SHARED yomm2.cpp)
+  set_target_properties(yomm2 PROPERTIES SOVERSION 0)
+else()
+  add_library(yomm2 STATIC yomm2.cpp)
+endif()
 
 target_include_directories(yomm2 PUBLIC
   $<BUILD_INTERFACE:${YOMM2_SOURCE_DIR}/include>
@@ -14,7 +19,7 @@ target_link_libraries(yomm2 PUBLIC Boost
 
 install(TARGETS yomm2
   EXPORT YOMM2Targets
-  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin
   ARCHIVE DESTINATION lib
 )
 
--- origsrc/yomm2-1.1.2/yomm2.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/yomm2-1.1.2/yomm2.pc.in	2021-06-14 14:52:53.375453500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: yomm2
+Description: Fast, orthogonal, open multi-methods
+Version: @VERSION@
+Libs: -L${libdir} -lyomm2
+Cflags: -I${includedir}
