--- origsrc/orocos_kinematics_dynamics-1.4.0/orocos_kdl/CMakeLists.txt	2018-03-22 05:48:25.000000000 +0900
+++ src/orocos_kinematics_dynamics-1.4.0/orocos_kdl/CMakeLists.txt	2020-07-25 12:24:04.933787000 +0900
@@ -110,8 +110,14 @@ CONFIGURE_FILE(KDLConfig.cmake.in
 CONFIGURE_FILE(KDLConfigVersion.cmake.in
   ${PROJECT_BINARY_DIR}/orocos_kdl-config-version.cmake @ONLY)
 
-INSTALL(FILES ${PROJECT_BINARY_DIR}/orocos_kdl-config.cmake DESTINATION share/orocos_kdl/cmake)
-INSTALL(FILES ${PROJECT_BINARY_DIR}/orocos_kdl-config-version.cmake DESTINATION share/orocos_kdl/cmake)
-INSTALL(EXPORT OrocosKDLTargets DESTINATION share/orocos_kdl/cmake)
+if (CYGWIN)
+  INSTALL(FILES ${PROJECT_BINARY_DIR}/orocos_kdl-config.cmake DESTINATION lib/cmake/orocos_kdl)
+  INSTALL(FILES ${PROJECT_BINARY_DIR}/orocos_kdl-config-version.cmake DESTINATION lib/cmake/orocos_kdl)
+  INSTALL(EXPORT OrocosKDLTargets DESTINATION lib/cmake/orocos_kdl)
+else()
+  INSTALL(FILES ${PROJECT_BINARY_DIR}/orocos_kdl-config.cmake DESTINATION share/orocos_kdl/cmake)
+  INSTALL(FILES ${PROJECT_BINARY_DIR}/orocos_kdl-config-version.cmake DESTINATION share/orocos_kdl/cmake)
+  INSTALL(EXPORT OrocosKDLTargets DESTINATION share/orocos_kdl/cmake)
+endif()
 
 INSTALL(FILES package.xml DESTINATION share/orocos_kdl)
--- origsrc/orocos_kinematics_dynamics-1.4.0/orocos_kdl/models/CMakeLists.txt	2018-03-22 05:48:25.000000000 +0900
+++ src/orocos_kinematics_dynamics-1.4.0/orocos_kdl/models/CMakeLists.txt	2020-07-25 12:21:18.062960700 +0900
@@ -17,6 +17,7 @@ IF(BUILD_MODELS)
 
   INSTALL( TARGETS orocos-kdl-models
       EXPORT OrocosKDLTargets
+      RUNTIME DESTINATION bin
       ARCHIVE DESTINATION lib${LIB_SUFFIX}
       LIBRARY DESTINATION lib${LIB_SUFFIX}
       PUBLIC_HEADER DESTINATION include/kdl
--- origsrc/orocos_kinematics_dynamics-1.4.0/orocos_kdl/src/CMakeLists.txt	2018-03-22 05:48:25.000000000 +0900
+++ src/orocos_kinematics_dynamics-1.4.0/orocos_kdl/src/CMakeLists.txt	2020-07-25 12:21:50.038217100 +0900
@@ -131,6 +131,7 @@ TARGET_LINK_LIBRARIES(orocos-kdl ${Boost
 
 INSTALL(TARGETS orocos-kdl
   EXPORT OrocosKDLTargets
+  RUNTIME DESTINATION bin
   ARCHIVE DESTINATION lib${LIB_SUFFIX}
   LIBRARY DESTINATION lib${LIB_SUFFIX}
   PUBLIC_HEADER DESTINATION include/kdl
--- origsrc/orocos_kinematics_dynamics-1.4.0/orocos_kdl/src/chainiksolverpos_lma.hpp	2018-03-22 05:48:25.000000000 +0900
+++ src/orocos_kinematics_dynamics-1.4.0/orocos_kdl/src/chainiksolverpos_lma.hpp	2020-07-25 12:03:10.605932500 +0900
@@ -35,6 +35,10 @@
 #include "chain.hpp"
 #include <Eigen/Dense>
 
+#if defined(_L)
+#undef _L
+#endif
+
 namespace KDL
 {
 
