--- origsrc/bzip3-1.1.4/Makefile.am	2022-06-20 02:18:56.000000000 +0900
+++ src/bzip3-1.1.4/Makefile.am	2022-06-23 18:50:12.982716600 +0900
@@ -1,7 +1,7 @@
 ACLOCAL_AMFLAGS = -I build-aux
 
 AM_CFLAGS = -I$(top_srcdir)/include
-AM_LDFLAGS = -avoid-version
+AM_LDFLAGS = -no-undefined
 
 EXTRA_DIST = LICENSE PORTING.md README.md build-aux/git-version-gen
 
@@ -72,7 +72,7 @@ CLEANFILES += LICENSE2
 BZIP3 := bzip3$(EXEEXT)
 
 roundtrip: $(BZIP3)
-	rm -f LICENSE2
-	./$(BZIP3) -e -b 6 LICENSE
-	./$(BZIP3) -d LICENSE.bz3 LICENSE2
-	cmp LICENSE LICENSE2
+	rm -f $(builddir)/LICENSE2
+	./$(BZIP3) -e -b 6 $(srcdir)/LICENSE $(builddir)/LICENSE.bz3
+	./$(BZIP3) -d $(builddir)/LICENSE.bz3 $(builddir)/LICENSE2
+	cmp $(srcdir)/LICENSE $(builddir)/LICENSE2
--- origsrc/bzip3-1.1.4/bzip3.pc.in	2022-06-20 02:18:56.000000000 +0900
+++ src/bzip3-1.1.4/bzip3.pc.in	2022-06-23 18:56:18.307507200 +0900
@@ -7,5 +7,5 @@ includedir=@includedir@
 Name: @PACKAGE@
 Description: A better and stronger spiritual successor to BZip2
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lbz3
+Libs: -L${libdir} -lbzip3
 Cflags: -I${includedir}
