--- origsrc/bzip3-1.1.5/Makefile.am	2022-09-04 19:55:16.000000000 +0900
+++ src/bzip3-1.1.5/Makefile.am	2022-09-26 15:43:56.015114800 +0900
@@ -1,7 +1,7 @@
 ACLOCAL_AMFLAGS = -I build-aux
 
 AM_CFLAGS = -I$(top_srcdir)/include
-AM_LDFLAGS = -avoid-version
+AM_LDFLAGS = -no-undefined
 
 EXTRA_DIST = LICENSE PORTING.md README.md build-aux/git-version-gen
 
@@ -17,8 +17,8 @@ libbzip3_la_SOURCES = src/libbz3.c
 
 bin_PROGRAMS = bzip3
 bzip3_CFLAGS = $(AM_CFLAGS)
-bzip3_LDDADD = libbzip3.la
-bzip3_SOURCES = src/main.c $(libbzip3_la_SOURCES)
+bzip3_LDADD = libbzip3.la
+bzip3_SOURCES = src/main.c
 
 dist_man_MANS = bzip3.1 bz3cat.1 bz3more.1 bz3less.1 bz3grep.1
 
@@ -68,7 +68,7 @@ CLEANFILES += LICENSE2
 BZIP3 := bzip3$(EXEEXT)
 
 roundtrip: $(BZIP3)
-	rm -f LICENSE2
-	./$(BZIP3) -feb 6 LICENSE
-	./$(BZIP3) -d LICENSE.bz3 LICENSE2
-	cmp LICENSE LICENSE2
+	rm -f $(builddir)/LICENSE2
+	./$(BZIP3) -feb 6 $(srcdir)/LICENSE $(builddir)/LICENSE.bz3
+	./$(BZIP3) -d $(builddir)/LICENSE.bz3 $(builddir)/LICENSE2
+	cmp $(srcdir)/LICENSE $(builddir)/LICENSE2
--- origsrc/bzip3-1.1.5/bzip3.pc.in	2022-09-04 19:55:16.000000000 +0900
+++ src/bzip3-1.1.5/bzip3.pc.in	2022-09-26 15:39:54.458502500 +0900
@@ -7,5 +7,5 @@ includedir=@includedir@
 Name: @PACKAGE@
 Description: A better and stronger spiritual successor to BZip2
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lbz3
+Libs: -L${libdir} -lbzip3
 Cflags: -I${includedir}
