--- origsrc/bzip3-1.2.1/Makefile.am	2022-11-09 21:19:18.000000000 +0900
+++ src/bzip3-1.2.1/Makefile.am	2022-12-12 11:33:43.092670100 +0900
@@ -13,7 +13,7 @@ noinst_HEADERS = include/common.h \
 
 lib_LTLIBRARIES = libbzip3.la
 libbzip3_la_SOURCES = src/libbz3.c
-libbzip3_la_LDFLAGS = -version-info 0:0:0
+libbzip3_la_LDFLAGS = -no-undefined -version-info 0:0:0
 
 bin_PROGRAMS = bzip3
 bzip3_CFLAGS = $(AM_CFLAGS)
@@ -72,7 +72,7 @@ CLEANFILES += LICENSE2
 BZIP3 := bzip3$(EXEEXT)
 
 roundtrip: $(BZIP3)
-	rm -f LICENSE2
-	./$(BZIP3) -v -feb 6 LICENSE
-	./$(BZIP3) -v -d LICENSE.bz3 LICENSE2
-	cmp LICENSE LICENSE2
+	rm -f $(builddir)/LICENSE2
+	./$(BZIP3) -v -feb6 $(srcdir)/LICENSE $(builddir)/LICENSE.bz3
+	./$(BZIP3) -v -d $(builddir)/LICENSE.bz3 $(builddir)/LICENSE2
+	cmp $(srcdir)/LICENSE $(builddir)/LICENSE2
