--- origsrc/jsonxx/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/jsonxx/Makefile.am	2019-12-24 07:51:08.065758100 +0900
@@ -0,0 +1,14 @@
+
+check_PROGRAMS = jsonxx_test
+jsonxx_test_SOURCES = jsonxx_test.cc
+jsonxx_test_LDADD = libjsonxx.la
+TESTS = $(check_PROGRAMS)
+
+include_HEADERS = jsonxx.h
+
+lib_LTLIBRARIES = libjsonxx.la
+libjsonxx_la_SOURCES = jsonxx.cc
+libjsonxx_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/jsonxx/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/jsonxx/configure.ac	2019-12-24 07:51:08.065758100 +0900
@@ -0,0 +1,27 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([jsonxx], [0])
+AC_CONFIG_SRCDIR([jsonxx.cc])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+AC_C_INLINE
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+
+AC_CONFIG_FILES([jsonxx.pc Makefile])
+AC_OUTPUT
--- origsrc/jsonxx/jsonxx.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/jsonxx/jsonxx.pc.in	2019-12-24 07:51:28.847424600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: jsonxx
+Description: JSON parser in C++
+Version: @VERSION@
+Libs: -L${libdir} -ljsonxx
+Cflags: -I${includedir}
