--- origsrc/vsearch-2.14.2/configure.ac	2020-01-28 23:10:36.000000000 +0900
+++ src/vsearch-2.14.2/configure.ac	2020-03-03 09:44:08.892739200 +0900
@@ -92,8 +92,11 @@ case $target in
      powerpc64*) target_ppc="yes" ;;
 esac
 
-AC_CHECK_HEADERS([windows.h], [AM_CONDITIONAL(TARGET_WIN, true)], [AM_CONDITIONAL(TARGET_WIN, false)])
-
+if test "$host_os" = "cygwin"; then
+  AM_CONDITIONAL(TARGET_WIN, false)
+else
+  AC_CHECK_HEADERS([windows.h], [AM_CONDITIONAL(TARGET_WIN, true)], [AM_CONDITIONAL(TARGET_WIN, false)])
+fi
 AM_CONDITIONAL(HAVE_BZLIB, test "x${have_bzip2}" = "xyes")
 AM_CONDITIONAL(HAVE_ZLIB, test "x${have_zlib}" = "xyes")
 AM_CONDITIONAL(HAVE_PTHREADS, test "x${have_pthreads}" = "xyes")
