--- origsrc/vsearch-2.21.1/configure.ac	2022-01-19 02:26:39.000000000 +0900
+++ src/vsearch-2.21.1/configure.ac	2022-06-19 23:20:57.681558300 +0900
@@ -76,7 +76,11 @@ case $target in
      powerpc64*) target_ppc="yes" ;;
 esac
 
-AC_CHECK_HEADERS([windows.h], [AM_CONDITIONAL(TARGET_WIN, true)], [AM_CONDITIONAL(TARGET_WIN, false)])
+if test "$host_os" = "cygwin"; then
+  AM_CONDITIONAL(TARGET_WIN, false)
+else
+  AC_CHECK_HEADERS([windows.h], [AM_CONDITIONAL(TARGET_WIN, true)], [AM_CONDITIONAL(TARGET_WIN, false)])
+fi
 
 AM_CONDITIONAL(HAVE_PS2PDF, test "x${have_ps2pdf}" = "xyes")
 AM_CONDITIONAL(HAVE_MAN_HTML, test "x${have_man_html}" = "xyes")
--- origsrc/vsearch-2.21.1/src/dynlibs.cc	2022-01-19 02:26:39.000000000 +0900
+++ src/vsearch-2.21.1/src/dynlibs.cc	2022-06-19 23:20:57.685561300 +0900
@@ -67,6 +67,8 @@ HMODULE gz_lib;
 # else
 #  ifdef __APPLE__
 const char gz_libname[] = "libz.dylib";
+#  elif defined(__CYGWIN__)
+const char gz_libname[] = "cygz.dll";
 #  else
 const char gz_libname[] = "libz.so";
 #  endif
@@ -86,6 +88,8 @@ HMODULE bz2_lib;
 # else
 #  ifdef __APPLE__
 const char bz2_libname[] = "libbz2.dylib";
+#  elif defined(__CYGWIN__)
+const char bz2_libname[] = "cygbz2-1.dll";
 #  else
 const char bz2_libname[] = "libbz2.so";
 #  endif
