--- origsrc/vsearch-2.30.0/configure.ac	2025-02-28 02:39:06.000000000 +0900
+++ src/vsearch-2.30.0/configure.ac	2025-09-05 23:38:52.480130800 +0900
@@ -93,7 +93,11 @@ case $target in
      x86_64*) target_x86_64="yes" ;;
 esac
 
-AC_CHECK_HEADERS([windows.h], [AM_CONDITIONAL(TARGET_WIN, true)], [AM_CONDITIONAL(TARGET_WIN, false)])
+if test "$host_os" = "cygwin"; then
+  AM_CONDITIONAL(TARGET_WIN, false)
+else
+  AC_CHECK_HEADERS([windows.h], [AM_CONDITIONAL(TARGET_WIN, true)], [AM_CONDITIONAL(TARGET_WIN, false)])
+fi
 
 AM_CONDITIONAL(HAVE_PS2PDF, test "x${have_ps2pdf}" = "xyes")
 AM_CONDITIONAL(HAVE_MAN_HTML, test "x${have_man_html}" = "xyes")
--- origsrc/vsearch-2.30.0/src/dynlibs.cc	2025-02-28 02:39:06.000000000 +0900
+++ src/vsearch-2.30.0/src/dynlibs.cc	2025-09-05 23:38:52.483836900 +0900
@@ -70,6 +70,8 @@ HMODULE gz_lib;
 # else
 #  ifdef __APPLE__
 const char gz_libname[] = "libz.dylib";
+#  elif defined(__CYGWIN__)
+const char gz_libname[] = "cygz.dll";
 #  else
 const char gz_libname[] = "libz.so.1";
 #  endif
@@ -89,6 +91,8 @@ HMODULE bz2_lib;
 # else
 #  ifdef __APPLE__
 const char bz2_libname[] = "libbz2.dylib";
+#  elif defined(__CYGWIN__)
+const char bz2_libname[] = "cygbz2-1.dll";
 #  else
 const char bz2_libname[] = "libbz2.so.1";
 #  endif
