--- origsrc/mdds_0.11.2/Makefile.in	2014-12-19 05:05:56.174026000 +0900
+++ src/mdds_0.11.2/Makefile.in	2015-01-03 15:08:54.789479000 +0900
@@ -246,7 +246,7 @@ install: $(HEADERS)
 	install -m 644 @top_srcdir@/$(INCDIR)/mdds/compat/*.hpp $(DESTDIR)@includedir@/mdds/compat 
 	install -m 644 @top_srcdir@/$(INCDIR)/mdds/hash_container/*.hpp $(DESTDIR)@includedir@/mdds/hash_container 
 	install -m 644 $(MISCDIR)/mdds.pc $(DESTDIR)@datarootdir@/pkgconfig 
-	install -m 644 @top_srcdir@/AUTHORS @top_srcdir@/COPYING @top_srcdir@/NEWS @top_srcdir@/README @top_srcdir@/VERSION $(DESTDIR)@docdir@ 
+	install -m 644 @top_srcdir@/AUTHORS @top_srcdir@/COPYING @top_srcdir@/NEWS @top_srcdir@/README @top_builddir@/VERSION $(DESTDIR)@docdir@ 
 
 check: $(ALL_TESTS)
 
