--- origsrc/mdds_0.9.0/Makefile.in	2013-06-23 09:47:59.920825000 +0900
+++ src/mdds_0.9.0/Makefile.in	2013-10-05 10:36:46.581854900 +0900
@@ -220,7 +220,7 @@ install: $(HEADERS)
 	install -m 644 -t $(DESTDIR)@includedir@/mdds/compat @top_srcdir@/$(INCDIR)/mdds/compat/*.hpp
 	install -m 644 -t $(DESTDIR)@includedir@/mdds/hash_container @top_srcdir@/$(INCDIR)/mdds/hash_container/*.hpp
 	install -m 644 -t $(DESTDIR)@datarootdir@/pkgconfig $(MISCDIR)/mdds.pc
-	install -m 644 -t $(DESTDIR)@docdir@ @top_srcdir@/AUTHORS @top_srcdir@/COPYING @top_srcdir@/NEWS @top_srcdir@/README @top_srcdir@/VERSION
+	install -m 644 -t $(DESTDIR)@docdir@ @top_srcdir@/AUTHORS @top_srcdir@/COPYING @top_srcdir@/NEWS @top_srcdir@/README @top_builddir@/VERSION
 
 check: $(ALL_TESTS)
 
