--- origsrc/mdds_0.12.0/Makefile.in	2015-02-06 11:32:45.000000000 +0900
+++ src/mdds_0.12.0/Makefile.in	2015-02-11 14:44:21.945972100 +0900
@@ -246,7 +246,7 @@ install: $(HEADERS)
 	install -m 644 @top_srcdir@/$(INCDIR)/mdds/compat/*.hpp $(DESTDIR)@includedir@/mdds/compat 
 	install -m 644 @top_srcdir@/$(INCDIR)/mdds/hash_container/*.hpp $(DESTDIR)@includedir@/mdds/hash_container 
 	install -m 644 $(MISCDIR)/mdds.pc $(DESTDIR)@datarootdir@/pkgconfig 
-	install -m 644 @top_srcdir@/AUTHORS @top_srcdir@/COPYING @top_srcdir@/NEWS @top_srcdir@/README @top_srcdir@/VERSION $(DESTDIR)@docdir@ 
+	install -m 644 @top_srcdir@/AUTHORS @top_srcdir@/COPYING @top_srcdir@/NEWS @top_srcdir@/README @top_builddir@/VERSION $(DESTDIR)@docdir@ 
 
 check: $(ALL_TESTS)
 
