--- origsrc/mdds_0.9.1/Makefile.in	2013-10-22 04:53:37.556374500 +0900
+++ src/mdds_0.9.1/Makefile.in	2013-11-17 13:06:27.419336000 +0900
@@ -220,7 +220,7 @@ install: $(HEADERS)
 	install -m 644 -t $(DESTDIR)@includedir@/mdds/compat @top_srcdir@/$(INCDIR)/mdds/compat/*.hpp
 	install -m 644 -t $(DESTDIR)@includedir@/mdds/hash_container @top_srcdir@/$(INCDIR)/mdds/hash_container/*.hpp
 	install -m 644 -t $(DESTDIR)@datarootdir@/pkgconfig $(MISCDIR)/mdds.pc
-	install -m 644 -t $(DESTDIR)@docdir@ @top_srcdir@/AUTHORS @top_srcdir@/COPYING @top_srcdir@/NEWS @top_srcdir@/README @top_srcdir@/VERSION
+	install -m 644 -t $(DESTDIR)@docdir@ @top_srcdir@/AUTHORS @top_srcdir@/COPYING @top_srcdir@/NEWS @top_srcdir@/README @top_builddir@/VERSION
 
 check: $(ALL_TESTS)
 
