--- origsrc/libjxl/CMakeLists.txt	2025-03-08 15:34:14.000000000 +0900
+++ src/libjxl/CMakeLists.txt	2025-03-08 15:36:26.731676300 +0900
@@ -496,7 +496,7 @@ if(JPEGXL_ENABLE_MANPAGES)
         add_custom_command(
           OUTPUT "${PAGE}.1"
           COMMAND "${ASCIIDOC_PY}"
-          ARGS ${ASCIIDOC}
+          ARGS ${ASCIIDOC} -L
             --format manpage --destination-dir="${CMAKE_CURRENT_BINARY_DIR}"
             "${CMAKE_CURRENT_SOURCE_DIR}/doc/man/${PAGE}.txt"
           MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/doc/man/${PAGE}.txt")
