--- origsrc/carbon-c-relay-3.5/configure.ac	2019-02-22 17:17:13.000000000 +0900
+++ src/carbon-c-relay-3.5/configure.ac	2019-04-05 07:01:41.233263900 +0900
@@ -108,8 +108,9 @@ AC_SUBST([DEF_EXTENSIONS], [-D_GNU_SOURC
 found_rtld_next=yes],
 [AC_MSG_RESULT(no)])
 fi
-AC_DEFINE([HAVE_RTLD_NEXT], [$found_rtld_next], [Define if you have RTLD_NEXT])
-
+if test "$found_rtld_next" = "yes" ; then
+	AC_DEFINE([HAVE_RTLD_NEXT], [1], [Define if you have RTLD_NEXT])
+fi
 AC_ARG_WITH([gzip], [support gzip compression for sending/receiving],
 			[], [with_gzip=check])
 LIBGZIP=
