--- origsrc/uvg266/CMakeLists.txt	2023-03-09 17:00:15.000000000 +0900
+++ src/uvg266/CMakeLists.txt	2023-03-09 18:23:05.423362400 +0900
@@ -131,6 +131,9 @@ if(BUILD_SHARED_LIBS)
   list( APPEND CMAKE_INSTALL_RPATH "${UVG266_INSTALL_LIBDIR}" "./" "../lib" )
   set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
   add_library(uvg266 SHARED ${LIB_SOURCES})
+  if (DEFINED SOVERSION)
+    set_target_properties(uvg266 PROPERTIES SOVERSION "${SOVERSION}")
+  endif()
 else()
   add_library(uvg266 STATIC ${LIB_SOURCES})
   if(MSVC) # Fix a linking problem with visual studio when the library is the same name as the binary
--- origsrc/uvg266/src/uvg266.h	2023-03-09 17:00:15.000000000 +0900
+++ src/uvg266/src/uvg266.h	2023-03-09 18:06:27.962589100 +0900
@@ -49,7 +49,7 @@ extern "C" {
   #if !defined(PIC)
     // Building static uvg266 library.
     #define UVG_PUBLIC
-  #elif defined(_WIN32) || defined(__CYGWIN__)
+  #elif defined(_WIN32)
     // Building uvg266 DLL on Windows.
     #define UVG_PUBLIC __declspec(dllexport)
   #elif defined(__GNUC__)
@@ -62,7 +62,7 @@ extern "C" {
   #if defined(UVG_STATIC_LIB)
     // Using static uvg266 library.
     #define UVG_PUBLIC
-  #elif defined(_WIN32) || defined(__CYGWIN__)
+  #elif defined(_WIN32)
     // Using uvg266 DLL on Windows.
     #define UVG_PUBLIC __declspec(dllimport)
   #else
