diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/tiger/CMakeLists.txt src/tiger/CMakeLists.txt
--- origsrc/tiger/CMakeLists.txt	2013-05-31 19:32:06.000000000 +0900
+++ src/tiger/CMakeLists.txt	2015-05-28 20:41:32.063363300 +0900
@@ -17,9 +17,9 @@ ENDIF( NOT CMAKE_INSTALL_PREFIX )
 
 IF( NOT LIB_INSTALL_DIR )
   SET( LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib" )
-  IF (CMAKE_SIZEOF_VOID_P MATCHES "8")
+  IF (CMAKE_SIZEOF_VOID_P MATCHES "8" AND NOT CYGWIN)
     SET( LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib64" )
-  ENDIF (CMAKE_SIZEOF_VOID_P MATCHES "8")
+  ENDIF (CMAKE_SIZEOF_VOID_P MATCHES "8" AND NOT CYGWIN)
 ENDIF( NOT LIB_INSTALL_DIR )
 
 IF( NOT INCLUDE_INSTALL_DIR )
@@ -117,6 +117,7 @@ SET_TARGET_PROPERTIES(
 INSTALL(
 	TARGETS ${PROJECTNAME}
 	EXPORT ${PROJECTNAME_UC}LibraryDepends
+	RUNTIME DESTINATION bin
 	LIBRARY DESTINATION "${LIB_INSTALL_DIR}"
 	ARCHIVE DESTINATION "${LIB_INSTALL_DIR}"
 	PUBLIC_HEADER DESTINATION "${INCLUDE_INSTALL_DIR}/${PROJECTNAME}"
