--- origsrc/libacars-2.2.0/libacars/asn1/INTEGER.c	2023-08-22 06:18:46.000000000 +0900
+++ src/libacars-2.2.0/libacars/asn1/INTEGER.c	2024-01-04 05:52:15.105538800 +0900
@@ -6,6 +6,7 @@
 #include "asn_internal.h"
 #include "INTEGER.h"
 #include "asn_codecs_prim.h"	/* Encoder and decoder of a primitive type */
+#include <alloca.h>
 #include <errno.h>
 
 /*
--- origsrc/libacars-2.2.0/libacars/asn1/NativeEnumerated.c	2023-08-22 06:18:46.000000000 +0900
+++ src/libacars-2.2.0/libacars/asn1/NativeEnumerated.c	2024-01-04 05:52:35.614386600 +0900
@@ -11,6 +11,7 @@
  */
 #include "asn_internal.h"
 #include "NativeEnumerated.h"
+#include <alloca.h>
 
 /*
  * NativeEnumerated basic type description.
--- origsrc/libacars-2.2.0/libacars/asn1/der_encoder.c	2023-08-22 06:18:46.000000000 +0900
+++ src/libacars-2.2.0/libacars/asn1/der_encoder.c	2024-01-04 05:52:50.041975700 +0900
@@ -3,6 +3,7 @@
  * Redistribution and modifications are permitted subject to BSD license.
  */
 #include "asn_internal.h"
+#include <alloca.h>
 #include <errno.h>
 
 static ssize_t der_write_TL(ber_tlv_tag_t tag, ber_tlv_len_t len,
