--- origsrc/yazproxy-1.3.10/Makefile.am	2015-12-17 18:50:06.000000000 +0900
+++ src/yazproxy-1.3.10/Makefile.am	2019-06-26 07:25:23.197230300 +0900
@@ -13,3 +13,5 @@ dist-hook:
 	mkdir $(distdir)/win
 	cp $(srcdir)/win/makefile $(distdir)/win/
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/yazproxy-1.3.10/configure.ac	2015-12-17 18:50:06.000000000 +0900
+++ src/yazproxy-1.3.10/configure.ac	2019-06-26 07:25:17.620769500 +0900
@@ -29,7 +29,7 @@ AC_SUBST(YAZPROXY_SRC_ROOT)
 AC_SUBST(YAZPROXY_BUILD_ROOT)
 YAZPROXY_SRC_ROOT=`cd ${srcdir}; pwd`
 YAZPROXY_BUILD_ROOT=`pwd`
-AC_OUTPUT([
+AC_OUTPUT([yazproxy.pc
 	Makefile
 	src/Makefile
 	include/Makefile
--- origsrc/yazproxy-1.3.10/src/Makefile.am	2015-12-17 18:50:06.000000000 +0900
+++ src/yazproxy-1.3.10/src/Makefile.am	2019-06-26 07:15:11.921732900 +0900
@@ -1,7 +1,7 @@
 AM_CXXFLAGS = -I$(srcdir)/../include $(YAZPPINC) $(USEMARCONINC)
 
 lib_LTLIBRARIES = libyazproxy.la
-libyazproxy_la_LDFLAGS=-version-info 2:0:0
+libyazproxy_la_LDFLAGS = -no-undefined -version-info 2:0:0
 
 libyazproxy_la_SOURCES= yaz-proxy.cpp yaz-proxy-config.cpp yaz-bw.cpp \
  proxyp.h yaz-usemarcon.cpp charset-converter.cpp msg-thread.cpp msg-thread.h \
@@ -24,10 +24,11 @@ LDADD=libyazproxy.la $(USEMARCONLALIB) $
 
 # Modules
 mod_proxy_sample_la_SOURCES = mod_sample.cpp
-mod_proxy_sample_la_LDFLAGS = -rpath $(pkglibdir) -module -avoid-version
+mod_proxy_sample_la_LDFLAGS = -no-undefined -rpath $(pkglibdir) -module -avoid-version
 
 mod_proxy_helsinki_la_SOURCES = mod_helsinki.cpp
-mod_proxy_helsinki_la_LDFLAGS = -rpath $(pkglibdir) -module -avoid-version
+mod_proxy_helsinki_la_LDFLAGS = -no-undefined -rpath $(pkglibdir) -module -avoid-version
+mod_proxy_helsinki_la_LIBADD = $(YAZPPLIB)
 
 pkglib_LTLIBRARIES = mod_proxy_sample.la mod_proxy_helsinki.la
 
--- origsrc/yazproxy-1.3.10/yazproxy.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/yazproxy-1.3.10/yazproxy.pc.in	2019-06-26 07:24:39.103672400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: yazproxy
+Description: YAZ Proxy application
+Version: @VERSION@
+Libs: -L${libdir} -lyazproxy
+Cflags: -I${includedir}
