--- origsrc/libqpbo-1.4.1/CMakeLists.txt	2021-04-15 00:38:54.000000000 +0900
+++ src/libqpbo-1.4.1/CMakeLists.txt	2024-12-08 01:33:48.091486000 +0900
@@ -6,20 +6,22 @@ include(GNUInstallDirs)
 
 set(QPBO_VERSION_MAJOR 1)
 set(QPBO_VERSION_MINOR 4)
-set(QPBO_VERSION_PATCH 0)
+set(QPBO_VERSION_PATCH 1)
 set(QPBO_VERSION_STRING "${QPBO_VERSION_MAJOR}.${QPBO_VERSION_MINOR}.${QPBO_VERSION_PATCH}")
 
 set(SRC_FILES qpbo.cpp qpbo_extra.cpp qpbo_maxflow.cpp qpbo_postprocessing.cpp)
 
 add_library(qpbo_shared SHARED ${SRC_FILES})
-add_library(qpbo_static STATIC ${SRC_FILES})
 set_target_properties(
-	qpbo_shared qpbo_static
+	qpbo_shared
 	PROPERTIES
 		OUTPUT_NAME qpbo
 		VERSION "${QPBO_VERSION_STRING}")
+if (DEFINED SOVERSION)
+	set_target_properties(qpbo_shared PROPERTIES SOVERSION "${SOVERSION}")
+endif()
 
-install(TARGETS qpbo_shared qpbo_static DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+install(TARGETS qpbo_shared RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 install(FILES block.h                   DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/qpbo")
 install(FILES qpbo.h                    DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/qpbo")
 
