--- origsrc/protobuf-2.4.1/configure.ac	2011-05-01 02:22:48.000000000 +0900
+++ src/protobuf-2.4.1/configure.ac	2011-05-28 14:45:08.739336300 +0900
@@ -71,7 +71,7 @@ ACX_CHECK_SUNCC
 
 # Have to do libtool after SUNCC, other wise it "helpfully" adds Crun Cstd
 # to the link
-AC_PROG_LIBTOOL
+LT_INIT
 
 # Checks for header files.
 AC_HEADER_STDC
--- origsrc/protobuf-2.4.1/src/Makefile.am	2011-05-28 14:44:43.879573500 +0900
+++ src/protobuf-2.4.1/src/Makefile.am	2011-05-28 15:45:01.419338300 +0900
@@ -338,8 +338,6 @@ nodist_protobuf_lite_test_SOURCES = $(pr
 # Test plugin binary.
 test_plugin_LDADD = $(PTHREAD_LIBS) libprotobuf.la libprotoc.la \
                     -lgtest
-test_plugin_CPPFLAGS = -I$(top_srcdir)/gtest/include         \
-                       -I$(top_builddir)/gtest/include
 test_plugin_SOURCES =                                          \
   google/protobuf/compiler/mock_code_generator.cc              \
   google/protobuf/testing/file.cc                              \
