--- origsrc/tpm2-tss-2.3.3/Makefile.am	2020-02-19 07:24:32.000000000 +0900
+++ src/tpm2-tss-2.3.3/Makefile.am	2020-03-04 11:33:31.956526600 +0900
@@ -245,8 +245,9 @@ lib_LTLIBRARIES += $(libtss2_mu)
 pkgconfig_DATA += lib/tss2-mu.pc
 EXTRA_DIST += lib/tss2-mu.map lib/tss2-mu.def src/tss2-mu/tss2-mu.vcxproj
 
+src_tss2_mu_libtss2_mu_la_LDFLAGS = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_mu_libtss2_mu_la_LDFLAGS = -Wl,--version-script=$(srcdir)/lib/tss2-mu.map
+src_tss2_mu_libtss2_mu_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-mu.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_mu_libtss2_mu_la_LIBADD = $(libutil)
 src_tss2_mu_libtss2_mu_la_SOURCES = $(TSS2_MU_SRC)
@@ -260,8 +261,9 @@ pkgconfig_DATA += lib/tss2-tctildr.pc
 EXTRA_DIST += lib/tss2-tctildr.map
 
 src_tss2_tcti_libtss2_tctildr_la_CFLAGS = $(AM_CFLAGS)
+src_tss2_tcti_libtss2_tctildr_la_LDFLAGS = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_tcti_libtss2_tctildr_la_LDFLAGS = \
+src_tss2_tcti_libtss2_tctildr_la_LDFLAGS += \
     -Wl,--version-script=$(srcdir)/lib/tss2-tctildr.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tctildr_la_LIBADD = $(libutil)
@@ -284,8 +286,9 @@ lib_LTLIBRARIES += $(libtss2_tcti_device
 pkgconfig_DATA += lib/tss2-tcti-device.pc
 EXTRA_DIST += lib/tss2-tcti-device.map
 
+src_tss2_tcti_libtss2_tcti_device_la_LDFLAGS  = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_tcti_libtss2_tcti_device_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/lib/tss2-tcti-device.map
+src_tss2_tcti_libtss2_tcti_device_la_LDFLAGS  += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-device.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_device_la_LIBADD   = $(libtss2_mu) $(libutil)
 src_tss2_tcti_libtss2_tcti_device_la_SOURCES  = \
@@ -304,8 +307,9 @@ EXTRA_DIST += lib/tss2-tcti-mssim.map \
               src/tss2-tcti/tss2-tcti-mssim.vcxproj \
               src/tss2-tcti/tss2-tcti-mssim.vcxproj.filters
 
+src_tss2_tcti_libtss2_tcti_mssim_la_LDFLAGS  = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_tcti_libtss2_tcti_mssim_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/lib/tss2-tcti-mssim.map
+src_tss2_tcti_libtss2_tcti_mssim_la_LDFLAGS  += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-mssim.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_mssim_la_LIBADD   = $(libtss2_mu) $(libutil)
 src_tss2_tcti_libtss2_tcti_mssim_la_SOURCES  = \
@@ -320,7 +324,7 @@ lib_LTLIBRARIES += $(libtss2_sys)
 pkgconfig_DATA += lib/tss2-sys.pc
 
 src_tss2_sys_libtss2_sys_la_CFLAGS = $(AM_CFLAGS) -I$(srcdir)/src/tss2-sys
-src_tss2_sys_libtss2_sys_la_LDFLAGS = $(AM_LDFLAGS) $(LIBSOCKET_LDFLAGS)
+src_tss2_sys_libtss2_sys_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) $(LIBSOCKET_LDFLAGS)
 
 if HAVE_LD_VERSION_SCRIPT
 src_tss2_sys_libtss2_sys_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-sys.map
@@ -348,7 +352,7 @@ src_tss2_esys_libtss2_esys_la_CFLAGS  =
     $(TSS2_ESYS_CFLAGS_CRYPTO)
 src_tss2_esys_libtss2_esys_la_LIBADD  = $(libtss2_sys) $(libtss2_mu) $(libutil)
 
-src_tss2_esys_libtss2_esys_la_LDFLAGS = $(AM_LDFLAGS) $(LIBSOCKET_LDFLAGS) \
+src_tss2_esys_libtss2_esys_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) $(LIBSOCKET_LDFLAGS) \
                                         $(TSS2_ESYS_LDFLAGS_CRYPTO)
 if HAVE_LD_VERSION_SCRIPT
 src_tss2_esys_libtss2_esys_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-esys.map
@@ -376,8 +380,9 @@ lib_LTLIBRARIES += $(libtss2_rc)
 pkgconfig_DATA += lib/tss2-rc.pc
 EXTRA_DIST += lib/tss2-rc.map lib/tss2-rc.def
 
+src_tss2_rc_libtss2_rc_la_LDFLAGS = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_rc_libtss2_rc_la_LDFLAGS = -Wl,--version-script=$(srcdir)/lib/tss2-rc.map
+src_tss2_rc_libtss2_rc_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-rc.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_rc_libtss2_rc_la_SOURCES = $(TSS2_RC_SRC)
 
