--- origsrc/tpm2-tss-4.0.1/Makefile.am	2023-01-24 03:36:52.000000000 +0900
+++ src/tpm2-tss-4.0.1/Makefile.am	2024-04-17 19:03:20.287024700 +0900
@@ -257,7 +257,7 @@ pkgconfig_DATA += lib/tss2-mu.pc
 EXTRA_DIST += lib/tss2-mu.map lib/tss2-mu.def src/tss2-mu/tss2-mu.vcxproj
 
 # Bump sys soname revision due to change in TPMU_CAPABILITY ABI
-src_tss2_mu_libtss2_mu_la_LDFLAGS = -version-info 0:1:0
+src_tss2_mu_libtss2_mu_la_LDFLAGS = -no-undefined -version-info 0:1:0
 
 if HAVE_LD_VERSION_SCRIPT
 src_tss2_mu_libtss2_mu_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-mu.map
@@ -275,8 +275,9 @@ pkgconfig_DATA += lib/tss2-tctildr.pc
 EXTRA_DIST += lib/tss2-tctildr.map
 
 src_tss2_tcti_libtss2_tctildr_la_CFLAGS = $(AM_CFLAGS)
+src_tss2_tcti_libtss2_tctildr_la_LDFLAGS = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_tcti_libtss2_tctildr_la_LDFLAGS = \
+src_tss2_tcti_libtss2_tctildr_la_LDFLAGS += \
     -Wl,--version-script=$(srcdir)/lib/tss2-tctildr.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tctildr_la_LIBADD = $(libutil)
@@ -298,8 +299,9 @@ tss2_HEADERS += $(srcdir)/include/tss2/t
 lib_LTLIBRARIES += $(libtss2_tcti_device)
 pkgconfig_DATA += lib/tss2-tcti-device.pc
 
+src_tss2_tcti_libtss2_tcti_device_la_LDFLAGS  = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_tcti_libtss2_tcti_device_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/lib/tss2-tcti-device.map
+src_tss2_tcti_libtss2_tcti_device_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-device.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_device_la_LIBADD   = $(libtss2_mu) $(libutil)
 src_tss2_tcti_libtss2_tcti_device_la_SOURCES  = \
@@ -315,8 +317,9 @@ tss2_HEADERS += $(srcdir)/include/tss2/t
 lib_LTLIBRARIES += $(libtss2_tcti_swtpm)
 pkgconfig_DATA += lib/tss2-tcti-swtpm.pc
 
+src_tss2_tcti_libtss2_tcti_swtpm_la_LDFLAGS  = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_tcti_libtss2_tcti_swtpm_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/lib/tss2-tcti-swtpm.map
+src_tss2_tcti_libtss2_tcti_swtpm_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-swtpm.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_swtpm_la_LIBADD   = $(libtss2_mu) $(libutil)
 src_tss2_tcti_libtss2_tcti_swtpm_la_SOURCES  = \
@@ -333,8 +336,9 @@ tss2_HEADERS += $(srcdir)/include/tss2/t
 lib_LTLIBRARIES += $(libtss2_tcti_mssim)
 pkgconfig_DATA += lib/tss2-tcti-mssim.pc
 
+src_tss2_tcti_libtss2_tcti_mssim_la_LDFLAGS  = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_tcti_libtss2_tcti_mssim_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/lib/tss2-tcti-mssim.map
+src_tss2_tcti_libtss2_tcti_mssim_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-mssim.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_mssim_la_LIBADD   = $(libtss2_mu) $(libutil)
 src_tss2_tcti_libtss2_tcti_mssim_la_SOURCES  = \
@@ -352,8 +356,9 @@ tss2_HEADERS += $(srcdir)/include/tss2/t
 lib_LTLIBRARIES += $(libtss2_tcti_pcap)
 pkgconfig_DATA += lib/tss2-tcti-pcap.pc
 
+src_tss2_tcti_libtss2_tcti_pcap_la_LDFLAGS  = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_tcti_libtss2_tcti_pcap_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/lib/tss2-tcti-pcap.map
+src_tss2_tcti_libtss2_tcti_pcap_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-pcap.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_pcap_la_LIBADD   = $(libtss2_tctildr) $(libtss2_mu) $(libutil)
 src_tss2_tcti_libtss2_tcti_pcap_la_SOURCES  = \
@@ -370,8 +375,9 @@ tss2_HEADERS += $(srcdir)/include/tss2/t
 lib_LTLIBRARIES += $(libtss2_tcti_libtpms)
 pkgconfig_DATA += lib/tss2-tcti-libtpms.pc
 
+src_tss2_tcti_libtss2_tcti_libtpms_la_LDFLAGS  = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_tcti_libtss2_tcti_libtpms_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/lib/tss2-tcti-libtpms.map
+src_tss2_tcti_libtss2_tcti_libtpms_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-libtpms.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_libtpms_la_LIBADD   = $(libtss2_mu) $(libutil)
 src_tss2_tcti_libtss2_tcti_libtpms_la_SOURCES  = \
@@ -388,9 +394,10 @@ tss2_HEADERS += $(srcdir)/include/tss2/t
 lib_LTLIBRARIES += $(libtss2_tcti_cmd)
 pkgconfig_DATA += lib/tss2-tcti-cmd.pc
 
+src_tss2_tcti_libtss2_tcti_cmd_la_LDFLAGS  = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
 if !UNIT
-src_tss2_tcti_libtss2_tcti_cmd_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/lib/tss2-tcti-cmd.map
+src_tss2_tcti_libtss2_tcti_cmd_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-cmd.map
 endif # UNIT
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_cmd_la_LIBADD   = $(libutil) $(libtss2_mu)
@@ -411,9 +418,10 @@ pkgconfig_DATA += lib/tss2-tcti-spi-help
 EXTRA_DIST += lib/tss2-tcti-spi-helper.map \
               lib/tss2-tcti-spi-helper.def
 
+src_tss2_tcti_libtss2_tcti_spi_helper_la_LDFLAGS  = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
 if !UNIT
-src_tss2_tcti_libtss2_tcti_spi_helper_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/lib/tss2-tcti-spi-helper.map
+src_tss2_tcti_libtss2_tcti_spi_helper_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-spi-helper.map
 endif # UNIT
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_spi_helper_la_LIBADD   = $(libutil) $(libtss2_mu)
@@ -430,7 +438,7 @@ lib_LTLIBRARIES += $(libtss2_sys)
 pkgconfig_DATA += lib/tss2-sys.pc
 
 src_tss2_sys_libtss2_sys_la_CFLAGS = $(AM_CFLAGS) -I$(srcdir)/src/tss2-sys
-src_tss2_sys_libtss2_sys_la_LDFLAGS = $(AM_LDFLAGS) $(LIBSOCKET_LDFLAGS)
+src_tss2_sys_libtss2_sys_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) $(LIBSOCKET_LDFLAGS)
 
 if HAVE_LD_VERSION_SCRIPT
 src_tss2_sys_libtss2_sys_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-sys.map
@@ -464,7 +472,7 @@ src_tss2_esys_libtss2_esys_la_CFLAGS  =
     $(TSS2_ESYS_CFLAGS_CRYPTO)
 src_tss2_esys_libtss2_esys_la_LIBADD  = $(libtss2_sys) $(libtss2_mu) $(libutil)
 
-src_tss2_esys_libtss2_esys_la_LDFLAGS = $(AM_LDFLAGS) $(LIBSOCKET_LDFLAGS) \
+src_tss2_esys_libtss2_esys_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) $(LIBSOCKET_LDFLAGS) \
                                         $(TSS2_ESYS_LDFLAGS_CRYPTO)
 if HAVE_LD_VERSION_SCRIPT
 src_tss2_esys_libtss2_esys_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-esys.map
@@ -495,8 +503,9 @@ lib_LTLIBRARIES += $(libtss2_rc)
 pkgconfig_DATA += lib/tss2-rc.pc
 EXTRA_DIST += lib/tss2-rc.map lib/tss2-rc.def
 
+src_tss2_rc_libtss2_rc_la_LDFLAGS = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_rc_libtss2_rc_la_LDFLAGS = -Wl,--version-script=$(srcdir)/lib/tss2-rc.map
+src_tss2_rc_libtss2_rc_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-rc.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_rc_libtss2_rc_la_SOURCES = $(TSS2_RC_SRC)
 
@@ -511,8 +520,9 @@ src_tss2_policy_libtss2_policy_la_CFLAGS
     $(JSONC_CFLAGS)
 src_tss2_policy_libtss2_policy_la_LIBADD  = $(libtss2_esys) $(libtss2_mu) $(libutil) $(JSONC_LIBS) $(UUID_LIBS)
 
+src_tss2_policy_libtss2_policy_la_LDFLAGS = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_policy_libtss2_policy_la_LDFLAGS = -Wl,--version-script=$(srcdir)/lib/tss2-policy.map
+src_tss2_policy_libtss2_policy_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-policy.map
 endif # HAVE_LD_VERSION_SCRIPT
 
 src_tss2_policy_libtss2_policy_la_SOURCES = $(TSS2_POLICY_SRC) \
@@ -671,7 +681,7 @@ src_tss2_fapi_libtss2_fapi_la_LIBADD  =
 
 src_tss2_fapi_libtss2_fapi_la_SOURCES = $(TSS2_FAPI_SRC)
 src_tss2_fapi_libtss2_fapi_la_CFLAGS  = $(AM_CFLAGS) -I$(srcdir)/src/tss2-fapi $(CRYPTO_CFLAGS) $(JSONC_CFLAGS) $(CURL_CFLAGS)
-src_tss2_fapi_libtss2_fapi_la_LDFLAGS = $(AM_LDFLAGS) $(CRYPTO_LIBS) $(JSONC_LIBS) $(CURL_LIBS) $(UUID_LIBS)
+src_tss2_fapi_libtss2_fapi_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) $(CRYPTO_LIBS) $(JSONC_LIBS) $(CURL_LIBS) $(UUID_LIBS)
 if HAVE_LD_VERSION_SCRIPT
 src_tss2_fapi_libtss2_fapi_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-fapi.map
 endif # HAVE_LD_VERSION_SCRIPT
--- origsrc/tpm2-tss-4.0.1/bootstrap	2023-01-24 03:36:52.000000000 +0900
+++ src/tpm2-tss-4.0.1/bootstrap	2024-04-16 16:43:38.130357300 +0900
@@ -1,7 +1,7 @@
 #!/bin/sh
 set -e
 
-git describe --tags --always --dirty > VERSION
+test -f VERSION || git describe --tags --always --dirty > VERSION
 
 # generate list of source files for use in Makefile.am
 # if you add new source files, you must run ./bootstrap again
--- origsrc/tpm2-tss-4.0.1/configure.ac	2023-01-24 03:36:52.000000000 +0900
+++ src/tpm2-tss-4.0.1/configure.ac	2024-04-16 16:48:07.136106500 +0900
@@ -44,6 +44,10 @@ case "${host_os}" in
         HOSTOS='BSD'
         LIBSOCKET_LDFLAGS=""
         ;;
+    *cygwin*)
+        HOSTOS='Cygwin'
+        LIBSOCKET_LDFLAGS=""
+        ;;
     *)
         #Assume linux
         HOSTOS='Linux'
