--- origsrc/tpm2-tss-4.1.3/Makefile.am	2024-05-21 21:43:43.000000000 +0900
+++ src/tpm2-tss-4.1.3/Makefile.am	2025-09-12 14:51:09.656500800 +0900
@@ -270,7 +270,7 @@ pkgconfig_DATA += lib/tss2-mu.pc
 EXTRA_DIST += lib/tss2-mu.map lib/tss2-mu.def src/tss2-mu/tss2-mu.vcxproj
 
 # Bump sys soname revision due to change in TPMU_CAPABILITY ABI
-src_tss2_mu_libtss2_mu_la_LDFLAGS = -version-info 0:1:0
+src_tss2_mu_libtss2_mu_la_LDFLAGS = -no-undefined -version-info 0:1:0
 
 if HAVE_LD_VERSION_SCRIPT
 src_tss2_mu_libtss2_mu_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-mu.map
@@ -288,8 +288,9 @@ pkgconfig_DATA += lib/tss2-tctildr.pc
 EXTRA_DIST += lib/tss2-tctildr.map
 
 src_tss2_tcti_libtss2_tctildr_la_CFLAGS = $(AM_CFLAGS)
+src_tss2_tcti_libtss2_tctildr_la_LDFLAGS = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_tcti_libtss2_tctildr_la_LDFLAGS = \
+src_tss2_tcti_libtss2_tctildr_la_LDFLAGS += \
     -Wl,--version-script=$(srcdir)/lib/tss2-tctildr.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tctildr_la_LIBADD = $(libutil)
@@ -311,8 +312,9 @@ tss2_HEADERS += $(srcdir)/include/tss2/t
 lib_LTLIBRARIES += $(libtss2_tcti_device)
 pkgconfig_DATA += lib/tss2-tcti-device.pc
 
+src_tss2_tcti_libtss2_tcti_device_la_LDFLAGS  = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_tcti_libtss2_tcti_device_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/lib/tss2-tcti-device.map
+src_tss2_tcti_libtss2_tcti_device_la_LDFLAGS  += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-device.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_device_la_LIBADD   = $(libtss2_mu) $(libutil) $(libutilio)
 src_tss2_tcti_libtss2_tcti_device_la_SOURCES  = \
@@ -328,8 +330,9 @@ tss2_HEADERS += $(srcdir)/include/tss2/t
 lib_LTLIBRARIES += $(libtss2_tcti_swtpm)
 pkgconfig_DATA += lib/tss2-tcti-swtpm.pc
 
+src_tss2_tcti_libtss2_tcti_swtpm_la_LDFLAGS  = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_tcti_libtss2_tcti_swtpm_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/lib/tss2-tcti-swtpm.map
+src_tss2_tcti_libtss2_tcti_swtpm_la_LDFLAGS  += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-swtpm.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_swtpm_la_LIBADD   = $(libtss2_mu) $(libutil) $(libutilio)
 src_tss2_tcti_libtss2_tcti_swtpm_la_SOURCES  = \
@@ -346,8 +349,9 @@ tss2_HEADERS += $(srcdir)/include/tss2/t
 lib_LTLIBRARIES += $(libtss2_tcti_mssim)
 pkgconfig_DATA += lib/tss2-tcti-mssim.pc
 
+src_tss2_tcti_libtss2_tcti_mssim_la_LDFLAGS  = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_tcti_libtss2_tcti_mssim_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/lib/tss2-tcti-mssim.map
+src_tss2_tcti_libtss2_tcti_mssim_la_LDFLAGS  += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-mssim.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_mssim_la_LIBADD   = $(libtss2_mu) $(libutil) $(libutilio)
 src_tss2_tcti_libtss2_tcti_mssim_la_SOURCES  = \
@@ -365,8 +369,9 @@ tss2_HEADERS += $(srcdir)/include/tss2/t
 lib_LTLIBRARIES += $(libtss2_tcti_pcap)
 pkgconfig_DATA += lib/tss2-tcti-pcap.pc
 
+src_tss2_tcti_libtss2_tcti_pcap_la_LDFLAGS  = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_tcti_libtss2_tcti_pcap_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/lib/tss2-tcti-pcap.map
+src_tss2_tcti_libtss2_tcti_pcap_la_LDFLAGS  += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-pcap.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_pcap_la_LIBADD   = $(libtss2_tctildr) $(libtss2_mu) $(libutil) $(libutilio)
 src_tss2_tcti_libtss2_tcti_pcap_la_SOURCES  = \
@@ -383,8 +388,9 @@ tss2_HEADERS += $(srcdir)/include/tss2/t
 lib_LTLIBRARIES += $(libtss2_tcti_libtpms)
 pkgconfig_DATA += lib/tss2-tcti-libtpms.pc
 
+src_tss2_tcti_libtss2_tcti_libtpms_la_LDFLAGS  = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_tcti_libtss2_tcti_libtpms_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/lib/tss2-tcti-libtpms.map
+src_tss2_tcti_libtss2_tcti_libtpms_la_LDFLAGS  += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-libtpms.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_libtpms_la_LIBADD   = $(libtss2_mu) $(libutil)
 src_tss2_tcti_libtss2_tcti_libtpms_la_SOURCES  = \
@@ -401,9 +407,10 @@ tss2_HEADERS += $(srcdir)/include/tss2/t
 lib_LTLIBRARIES += $(libtss2_tcti_cmd)
 pkgconfig_DATA += lib/tss2-tcti-cmd.pc
 
+src_tss2_tcti_libtss2_tcti_cmd_la_LDFLAGS  = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
 if !UNIT
-src_tss2_tcti_libtss2_tcti_cmd_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/lib/tss2-tcti-cmd.map
+src_tss2_tcti_libtss2_tcti_cmd_la_LDFLAGS  += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-cmd.map
 endif # UNIT
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_cmd_la_LIBADD   = $(libutil) $(libtss2_mu)
@@ -422,9 +429,10 @@ tss2_HEADERS += $(srcdir)/include/tss2/t
 lib_LTLIBRARIES += $(libtss2_tcti_spi_helper)
 pkgconfig_DATA += lib/tss2-tcti-spi-helper.pc
 
+src_tss2_tcti_libtss2_tcti_spi_helper_la_LDFLAGS  = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
 if !UNIT
-src_tss2_tcti_libtss2_tcti_spi_helper_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/lib/tss2-tcti-spi-helper.map
+src_tss2_tcti_libtss2_tcti_spi_helper_la_LDFLAGS  += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-spi-helper.map
 endif # UNIT
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_spi_helper_la_LIBADD   = $(libutil) $(libtss2_mu)
@@ -443,7 +451,7 @@ tss2_HEADERS += $(srcdir)/include/tss2/t
 lib_LTLIBRARIES += $(libtss2_tcti_spi_ltt2go)
 pkgconfig_DATA += lib/tss2-tcti-spi-ltt2go.pc
 
-src_tss2_tcti_libtss2_tcti_spi_ltt2go_la_LDFLAGS  = -lusb-1.0
+src_tss2_tcti_libtss2_tcti_spi_ltt2go_la_LDFLAGS  = -no-undefined -lusb-1.0
 
 if HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_spi_ltt2go_la_LDFLAGS  += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-spi-ltt2go.map
@@ -464,7 +472,7 @@ tss2_HEADERS += $(srcdir)/include/tss2/t
 lib_LTLIBRARIES += $(libtss2_tcti_spidev)
 pkgconfig_DATA += lib/tss2-tcti-spidev.pc
 
-src_tss2_tcti_libtss2_tcti_spidev_la_LDFLAGS  =
+src_tss2_tcti_libtss2_tcti_spidev_la_LDFLAGS  = -no-undefined
 
 if HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_spidev_la_LDFLAGS  += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-spidev.map
@@ -485,7 +493,7 @@ lib_LTLIBRARIES += $(libtss2_tcti_spi_ft
 pkgconfig_DATA += lib/tss2-tcti-spi-ftdi.pc
 
 src_tss2_tcti_libtss2_tcti_spi_ftdi_la_CFLAGS  = $(AM_CFLAGS) $(LIBFTDI_CFLAGS) -Wno-deprecated-declarations
-src_tss2_tcti_libtss2_tcti_spi_ftdi_la_LDFLAGS  = $(LIBFTDI_LIBS)
+src_tss2_tcti_libtss2_tcti_spi_ftdi_la_LDFLAGS  = -no-undefined $(LIBFTDI_LIBS)
 if HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_spi_ftdi_la_LDFLAGS  += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-spi-ftdi.map
 endif # HAVE_LD_VERSION_SCRIPT
@@ -509,9 +517,10 @@ tss2_HEADERS += $(srcdir)/include/tss2/t
 lib_LTLIBRARIES += $(libtss2_tcti_i2c_helper)
 pkgconfig_DATA += lib/tss2-tcti-i2c-helper.pc
 
+src_tss2_tcti_libtss2_tcti_i2c_helper_la_LDFLAGS  = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
 if !UNIT
-src_tss2_tcti_libtss2_tcti_i2c_helper_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/lib/tss2-tcti-i2c-helper.map
+src_tss2_tcti_libtss2_tcti_i2c_helper_la_LDFLAGS  += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-i2c-helper.map
 endif # UNIT
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_i2c_helper_la_LIBADD   = $(libutil) $(libtss2_mu)
@@ -531,7 +540,7 @@ lib_LTLIBRARIES += $(libtss2_tcti_i2c_ft
 pkgconfig_DATA += lib/tss2-tcti-i2c-ftdi.pc
 
 src_tss2_tcti_libtss2_tcti_i2c_ftdi_la_CFLAGS  = $(AM_CFLAGS) $(LIBFTDI_CFLAGS) -Wno-deprecated-declarations
-src_tss2_tcti_libtss2_tcti_i2c_ftdi_la_LDFLAGS  = $(LIBFTDI_LIBS)
+src_tss2_tcti_libtss2_tcti_i2c_ftdi_la_LDFLAGS  = -no-undefined $(LIBFTDI_LIBS)
 if HAVE_LD_VERSION_SCRIPT
 src_tss2_tcti_libtss2_tcti_i2c_ftdi_la_LDFLAGS  += -Wl,--version-script=$(srcdir)/lib/tss2-tcti-i2c-ftdi.map
 endif # HAVE_LD_VERSION_SCRIPT
@@ -555,7 +564,7 @@ lib_LTLIBRARIES += $(libtss2_sys)
 pkgconfig_DATA += lib/tss2-sys.pc
 
 src_tss2_sys_libtss2_sys_la_CFLAGS = $(AM_CFLAGS) -I$(srcdir)/src/tss2-sys
-src_tss2_sys_libtss2_sys_la_LDFLAGS = $(AM_LDFLAGS) $(LIBSOCKET_LDFLAGS)
+src_tss2_sys_libtss2_sys_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) $(LIBSOCKET_LDFLAGS)
 
 if HAVE_LD_VERSION_SCRIPT
 src_tss2_sys_libtss2_sys_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-sys.map
@@ -589,7 +598,7 @@ src_tss2_esys_libtss2_esys_la_CFLAGS  =
     $(TSS2_ESYS_CFLAGS_CRYPTO)
 src_tss2_esys_libtss2_esys_la_LIBADD  = $(libtss2_sys) $(libtss2_mu) $(libutil)
 
-src_tss2_esys_libtss2_esys_la_LDFLAGS = $(AM_LDFLAGS) $(LIBSOCKET_LDFLAGS) \
+src_tss2_esys_libtss2_esys_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) $(LIBSOCKET_LDFLAGS) \
                                         $(TSS2_ESYS_LDFLAGS_CRYPTO)
 if HAVE_LD_VERSION_SCRIPT
 src_tss2_esys_libtss2_esys_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-esys.map
@@ -620,8 +629,9 @@ lib_LTLIBRARIES += $(libtss2_rc)
 pkgconfig_DATA += lib/tss2-rc.pc
 EXTRA_DIST += lib/tss2-rc.map lib/tss2-rc.def
 
+src_tss2_rc_libtss2_rc_la_LDFLAGS = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_rc_libtss2_rc_la_LDFLAGS = -Wl,--version-script=$(srcdir)/lib/tss2-rc.map
+src_tss2_rc_libtss2_rc_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-rc.map
 endif # HAVE_LD_VERSION_SCRIPT
 src_tss2_rc_libtss2_rc_la_SOURCES = $(TSS2_RC_SRC)
 
@@ -637,8 +647,9 @@ src_tss2_policy_libtss2_policy_la_CFLAGS
 src_tss2_policy_libtss2_policy_la_LIBADD  = $(libtss2_esys) $(libtss2_mu) $(libutil) $(JSONC_LIBS) $(UUID_LIBS) \
 	$(CRYPTO_LIBS)
 
+src_tss2_policy_libtss2_policy_la_LDFLAGS = -no-undefined
 if HAVE_LD_VERSION_SCRIPT
-src_tss2_policy_libtss2_policy_la_LDFLAGS = -Wl,--version-script=$(srcdir)/lib/tss2-policy.map
+src_tss2_policy_libtss2_policy_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-policy.map
 endif # HAVE_LD_VERSION_SCRIPT
 
 src_tss2_policy_libtss2_policy_la_SOURCES = $(TSS2_POLICY_SRC) \
@@ -806,7 +817,7 @@ src_tss2_fapi_libtss2_fapi_la_LIBADD  =
 src_tss2_fapi_libtss2_fapi_la_SOURCES = $(TSS2_FAPI_SRC)
 src_tss2_fapi_libtss2_fapi_la_CFLAGS  = $(AM_CFLAGS) -I$(srcdir)/src/tss2-fapi $(CRYPTO_CFLAGS) $(JSONC_CFLAGS) \
                                         $(UUID_CFLAGS)
-src_tss2_fapi_libtss2_fapi_la_LDFLAGS = $(AM_LDFLAGS) $(CRYPTO_LIBS) $(JSONC_LIBS) $(CURL_LIBS) $(UUID_LIBS)
+src_tss2_fapi_libtss2_fapi_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) $(CRYPTO_LIBS) $(JSONC_LIBS) $(CURL_LIBS) $(UUID_LIBS)
 if HAVE_LD_VERSION_SCRIPT
 src_tss2_fapi_libtss2_fapi_la_LDFLAGS += -Wl,--version-script=$(srcdir)/lib/tss2-fapi.map
 endif # HAVE_LD_VERSION_SCRIPT
--- origsrc/tpm2-tss-4.1.3/bootstrap	2024-05-21 21:43:43.000000000 +0900
+++ src/tpm2-tss-4.1.3/bootstrap	2025-09-12 14:51:39.537210200 +0900
@@ -1,7 +1,7 @@
 #!/bin/sh
 set -e
 
-git describe --tags --always --dirty > VERSION
+test -f VERSION || git describe --tags --always --dirty > VERSION
 
 # generate list of source files for use in Makefile.am
 # if you add new source files, you must run ./bootstrap again
--- origsrc/tpm2-tss-4.1.3/configure.ac	2024-05-21 21:43:43.000000000 +0900
+++ src/tpm2-tss-4.1.3/configure.ac	2025-09-12 14:52:00.543255800 +0900
@@ -45,6 +45,10 @@ case "${host_os}" in
         HOSTOS='BSD'
         LIBSOCKET_LDFLAGS=""
         ;;
+    *cygwin*)
+        HOSTOS='Cygwin'
+        LIBSOCKET_LDFLAGS=""
+        ;;
     *)
         #Assume linux
         HOSTOS='Linux'
