--- origsrc/curlpp-0.7.3/examples/Makefile.am	2009-12-06 08:33:38.000000000 +0900
+++ src/curlpp-0.7.3/examples/Makefile.am	2014-06-11 22:06:19.125375000 +0900
@@ -83,9 +83,9 @@ example23_SOURCES = example23.cpp
 
 example24_SOURCES = example24.cpp
 
-AM_LDFLAGS = -L../src/curlpp/ -lcurlpp -static 
+AM_LDFLAGS = ../src/curlpp/libcurlpp.la
 
-INCLUDES = -I$(top_builddir)/include
+INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include
 
 
 
--- origsrc/curlpp-0.7.3/examples/example21.cpp	2009-12-06 08:33:38.000000000 +0900
+++ src/curlpp-0.7.3/examples/example21.cpp	2014-06-11 22:04:53.672250000 +0900
@@ -38,23 +38,6 @@
 #include <curlpp/Easy.hpp>
 #include <curlpp/Options.hpp>
 #include <curlpp/Exception.hpp>
- 
-/*
-   anonymous namespace to prevent name clash in case other examples using the same global entities
-   would be compiled in the same project
-*/
-namespace
-{
-
-char *data = NULL;
-
-size_t readData(char *buffer, size_t size, size_t nitems)
-{
-  strncpy(buffer, data, size * nitems);
-  return size * nitems;
-}
-
-} // namespace
 
 int main(int argc, char *argv[])
 {
--- origsrc/curlpp-0.7.3/src/curlpp/Makefile.am	2009-12-06 08:33:38.000000000 +0900
+++ src/curlpp-0.7.3/src/curlpp/Makefile.am	2014-06-10 22:37:19.053442000 +0900
@@ -6,7 +6,7 @@ if MAINTENER_CODE
 maintener_source = 
 endif
 
-INCLUDES = -I$(top_builddir)/include/curlpp -I$(top_builddir)/include
+INCLUDES = -I$(top_builddir)/include/curlpp -I$(top_builddir)/include -I$(top_srcdir)/include
 
 sources = \
 	cURLpp.cpp  \
--- origsrc/curlpp-0.7.3/src/curlpp/internal/Makefile.am	2009-12-06 08:33:38.000000000 +0900
+++ src/curlpp-0.7.3/src/curlpp/internal/Makefile.am	2014-06-10 22:37:19.053442000 +0900
@@ -4,7 +4,7 @@ if MAINTENER_CODE
 maintener_source = 
 endif
 
-INCLUDES = -I$(top_builddir)/include/curlpp -I$(top_builddir)/include/curlpp/internal -I$(top_builddir)/include
+INCLUDES = -I$(top_builddir)/include/curlpp -I$(top_builddir)/include/curlpp/internal -I$(top_builddir)/include -I$(top_srcdir)/include
 
 sources = \
 	CurlHandle.cpp \
--- origsrc/curlpp-0.7.3/src/utilspp/Makefile.am	2009-12-06 08:33:38.000000000 +0900
+++ src/curlpp-0.7.3/src/utilspp/Makefile.am	2014-06-10 22:37:19.053442000 +0900
@@ -1,9 +1,9 @@
 lib_LTLIBRARIES = libutilspp.la
 
-INCLUDES = -I$(top_builddir)/include
+INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include
 
 libutilspp_la_SOURCES = \
 	LifetimeLibrary.cpp \
 	PrivateMembers.cpp
 
-
+libutilspp_la_LDFLAGS = -no-undefined
