--- origsrc/curlpp-0.7.3/examples/Makefile.am	2009-12-06 08:33:38.000000000 +0900
+++ src/curlpp-0.7.3/examples/Makefile.am	2012-02-05 16:52:45.741217200 +0900
@@ -83,9 +83,9 @@ example23_SOURCES = example23.cpp
 
 example24_SOURCES = example24.cpp
 
-AM_LDFLAGS = -L../src/curlpp/ -lcurlpp -static 
+AM_LDFLAGS = -L../src/curlpp/ -lcurlpp
 
-INCLUDES = -I$(top_builddir)/include
+INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include
 
 
 
--- origsrc/curlpp-0.7.3/src/curlpp/Makefile.am	2009-12-06 08:33:38.000000000 +0900
+++ src/curlpp-0.7.3/src/curlpp/Makefile.am	2012-02-05 16:39:11.375940700 +0900
@@ -6,7 +6,7 @@ if MAINTENER_CODE
 maintener_source = 
 endif
 
-INCLUDES = -I$(top_builddir)/include/curlpp -I$(top_builddir)/include
+INCLUDES = -I$(top_builddir)/include/curlpp -I$(top_builddir)/include -I$(top_srcdir)/include
 
 sources = \
 	cURLpp.cpp  \
--- origsrc/curlpp-0.7.3/src/curlpp/internal/Makefile.am	2009-12-06 08:33:38.000000000 +0900
+++ src/curlpp-0.7.3/src/curlpp/internal/Makefile.am	2012-02-05 16:49:43.951959800 +0900
@@ -4,7 +4,7 @@ if MAINTENER_CODE
 maintener_source = 
 endif
 
-INCLUDES = -I$(top_builddir)/include/curlpp -I$(top_builddir)/include/curlpp/internal -I$(top_builddir)/include
+INCLUDES = -I$(top_builddir)/include/curlpp -I$(top_builddir)/include/curlpp/internal -I$(top_builddir)/include -I$(top_srcdir)/include
 
 sources = \
 	CurlHandle.cpp \
--- origsrc/curlpp-0.7.3/src/utilspp/Makefile.am	2009-12-06 08:33:38.000000000 +0900
+++ src/curlpp-0.7.3/src/utilspp/Makefile.am	2012-02-05 17:02:07.552921400 +0900
@@ -1,9 +1,9 @@
 lib_LTLIBRARIES = libutilspp.la
 
-INCLUDES = -I$(top_builddir)/include
+INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include
 
 libutilspp_la_SOURCES = \
 	LifetimeLibrary.cpp \
 	PrivateMembers.cpp
 
-
+libutilspp_la_LDFLAGS = -no-undefined
