--- origsrc/openvr-1.23.7/CMakeLists.txt	2022-08-20 05:12:52.000000000 +0900
+++ src/openvr-1.23.7/CMakeLists.txt	2022-10-11 12:58:01.831228700 +0900
@@ -46,7 +46,7 @@ if(WIN32)
     add_definitions(-DOPENVR_BUILD_STATIC)
   endif()
 elseif(UNIX AND NOT APPLE)
-  if(CMAKE_SYSTEM_NAME MATCHES ".*Linux")
+  if(CMAKE_SYSTEM_NAME MATCHES ".*Linux" OR CYGWIN)
     set(PLATFORM_NAME "linux")
     add_definitions(-DLINUX -DPOSIX)
     if(PROCESSOR_ARCH MATCHES "64")
@@ -75,7 +75,11 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMA
 # Enable some properties.
 if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang")
 	# Enable c++11 and hide symbols which shouldn't be visible
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -fPIC -fvisibility=hidden")
+  if(CYGWIN)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -fvisibility=hidden")
+  else()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -fPIC -fvisibility=hidden")
+  endif()
 
 	# Set custom libc++ usage here
 	if(CMAKE_C_COMPILER_ID MATCHES "Clang" AND USE_LIBCXX)
--- origsrc/openvr-1.23.7/headers/openvr_capi.h	2022-08-20 05:12:52.000000000 +0900
+++ src/openvr-1.23.7/headers/openvr_capi.h	2022-10-11 12:56:36.435482900 +0900
@@ -24,8 +24,8 @@
 #endif
 
 // OPENVR API export macro
-#if defined( _WIN32 ) && !defined( _X360 )
-	#if defined( OPENVR_API_EXPORTS )
+#if (defined( _WIN32 ) || defined( __CYGWIN__ )) && !defined( _X360 )
+	#if defined( OPENVR_API_EXPORTS ) || defined( __CYGWIN__ )
 	#define S_API EXTERN_C __declspec( dllexport )
 	#elif defined( OPENVR_API_NODLL )
 	#define S_API EXTERN_C
--- origsrc/openvr-1.23.7/src/CMakeLists.txt	2022-08-20 05:12:52.000000000 +0900
+++ src/openvr-1.23.7/src/CMakeLists.txt	2022-10-11 13:10:54.185548600 +0900
@@ -68,6 +68,9 @@ source_group("VRCommon" FILES
 # Build the library.
 if(BUILD_SHARED)
 	add_library(${LIBNAME} SHARED ${SOURCE_FILES})
+	if(CYGWIN)
+		set_target_properties(${LIBNAME} PROPERTIES SOVERSION 1.0.0)
+	endif()
 elseif(BUILD_FRAMEWORK)
 	set( LIBNAME "OpenVR" )
 	add_library( ${LIBNAME} 
@@ -101,16 +104,15 @@ endif()
 target_link_libraries(${LIBNAME} ${EXTRA_LIBS} ${CMAKE_DL_LIBS})
 target_include_directories(${LIBNAME} PUBLIC ${OPENVR_HEADER_DIR})
 
-install(TARGETS ${LIBNAME} DESTINATION lib)
+install(TARGETS ${LIBNAME} RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 install(FILES ${PUBLIC_HEADER_FILES} DESTINATION include/openvr)
 
 # Generate a .pc file for linux environments
-if(PLATFORM_NAME MATCHES "linux")
-    set(INSTALL_PKGCONFIG_DIR "${CMAKE_INSTALL_PREFIX}/share/pkgconfig" CACHE PATH "Installation directory for pkgconfig (.pc) files")
+if(PLATFORM_NAME MATCHES "linux" OR CYGWIN)
     CONFIGURE_FILE("openvr.pc.in" "openvr.pc" @ONLY)
 
     set(OPENVR_PC ${CMAKE_CURRENT_BINARY_DIR}/openvr.pc)
     if(NOT SKIP_INSTALL_FILES AND NOT SKIP_INSTALL_ALL )
-        install(FILES ${OPENVR_PC} DESTINATION "${INSTALL_PKGCONFIG_DIR}")
+        install(FILES ${OPENVR_PC} DESTINATION lib/pkgconfig)
     endif()
 endif()
--- origsrc/openvr-1.23.7/src/openvr_api_public.cpp	2022-08-20 05:12:52.000000000 +0900
+++ src/openvr-1.23.7/src/openvr_api_public.cpp	2022-10-11 12:56:36.449417400 +0900
@@ -2,12 +2,12 @@
 #define VR_API_EXPORT 1
 #include "openvr.h"
 #include "ivrclientcore.h"
-#include <vrcore/pathtools_public.h>
-#include <vrcore/sharedlibtools_public.h>
-#include <vrcore/envvartools_public.h>
+#include <vrcommon/pathtools_public.h>
+#include <vrcommon/sharedlibtools_public.h>
+#include <vrcommon/envvartools_public.h>
 #include "hmderrors_public.h"
-#include <vrcore/strtools_public.h>
-#include <vrcore/vrpathregistry_public.h>
+#include <vrcommon/strtools_public.h>
+#include <vrcommon/vrpathregistry_public.h>
 #include <mutex>
 
 using vr::EVRInitError;
--- origsrc/openvr-1.23.7/src/vrcommon/dirtools_public.cpp	2022-08-20 05:12:52.000000000 +0900
+++ src/openvr-1.23.7/src/vrcommon/dirtools_public.cpp	2022-10-11 12:56:36.466423100 +0900
@@ -1,7 +1,7 @@
 //========= Copyright Valve Corporation ============//
-#include <vrcore/dirtools_public.h>
-#include <vrcore/strtools_public.h>
-#include <vrcore/pathtools_public.h>
+#include <vrcommon/dirtools_public.h>
+#include <vrcommon/strtools_public.h>
+#include <vrcommon/pathtools_public.h>
 
 #include <errno.h>
 #include <string.h>
--- origsrc/openvr-1.23.7/src/vrcommon/envvartools_public.cpp	2022-08-20 05:12:52.000000000 +0900
+++ src/openvr-1.23.7/src/vrcommon/envvartools_public.cpp	2022-10-11 12:56:36.472438400 +0900
@@ -1,6 +1,6 @@
 //========= Copyright Valve Corporation ============//
-#include <vrcore/envvartools_public.h>
-#include <vrcore/strtools_public.h>
+#include <vrcommon/envvartools_public.h>
+#include <vrcommon/strtools_public.h>
 #include <stdlib.h>
 #include <string>
 #include <cctype>
--- origsrc/openvr-1.23.7/src/vrcommon/hmderrors_public.cpp	2022-08-20 05:12:52.000000000 +0900
+++ src/openvr-1.23.7/src/vrcommon/hmderrors_public.cpp	2022-10-11 13:07:46.871290900 +0900
@@ -180,7 +180,7 @@ const char *GetIDForVRInitError( vr::EVR
 		RETURN_ENUM_AS_STRING( VRInitError_Init_PrismNeedsNewDrivers );
 		RETURN_ENUM_AS_STRING( VRInitError_Init_PrismStartupTimedOut );
 		RETURN_ENUM_AS_STRING( VRInitError_Init_CouldNotStartPrism );
-		RETURN_ENUM_AS_STRING( VRInitError_Init_CreateDriverDirectDeviceFailed );
+		//RETURN_ENUM_AS_STRING( VRInitError_Init_CreateDriverDirectDeviceFailed );
 		RETURN_ENUM_AS_STRING( VRInitError_Init_PrismExitedUnexpectedly );
 
 		RETURN_ENUM_AS_STRING( VRInitError_Driver_Failed );
--- origsrc/openvr-1.23.7/src/vrcommon/pathtools_public.cpp	2022-08-20 05:12:52.000000000 +0900
+++ src/openvr-1.23.7/src/vrcommon/pathtools_public.cpp	2022-10-11 12:56:36.479422500 +0900
@@ -1,6 +1,6 @@
 //========= Copyright Valve Corporation ============//
-#include <vrcore/strtools_public.h>
-#include <vrcore/pathtools_public.h>
+#include <vrcommon/strtools_public.h>
+#include <vrcommon/pathtools_public.h>
 
 #if defined( _WIN32)
 #include <windows.h>
--- origsrc/openvr-1.23.7/src/vrcommon/pathtools_public.h	2022-08-20 05:12:52.000000000 +0900
+++ src/openvr-1.23.7/src/vrcommon/pathtools_public.h	2022-10-11 12:56:36.484417300 +0900
@@ -123,10 +123,12 @@ std::string Path_SanitizeFilename( const
 #endif
 
 //-----------------------------------------------------------------------------
-#if defined(WIN32)
+#if defined(WIN32) || defined(__CYGWIN__)
 #define DYNAMIC_LIB_EXT	".dll"
 #define PROGRAM_EXT ".exe"
-#ifdef _WIN64
+#if defined(__CYGWIN__)
+#define PLATSUBDIR  "cygwin"
+#elif defined(_WIN64)
 #define PLATSUBDIR	"win64"
 #else
 #define PLATSUBDIR	"win32"
--- origsrc/openvr-1.23.7/src/vrcommon/sharedlibtools_public.cpp	2022-08-20 05:12:52.000000000 +0900
+++ src/openvr-1.23.7/src/vrcommon/sharedlibtools_public.cpp	2022-10-11 12:56:36.494437800 +0900
@@ -1,5 +1,5 @@
 //========= Copyright Valve Corporation ============//
-#include <vrcore/sharedlibtools_public.h>
+#include <vrcommon/sharedlibtools_public.h>
 #include <string.h>
 
 #if defined(_WIN32)
--- origsrc/openvr-1.23.7/src/vrcommon/strtools_public.cpp	2022-08-20 05:12:52.000000000 +0900
+++ src/openvr-1.23.7/src/vrcommon/strtools_public.cpp	2022-10-11 12:56:36.555432400 +0900
@@ -1,5 +1,5 @@
 //========= Copyright Valve Corporation ============//
-#include <vrcore/strtools_public.h>
+#include <vrcommon/strtools_public.h>
 #include <string.h>
 #include <stdio.h>
 #include <stdlib.h>
@@ -9,7 +9,7 @@
 #include <functional>
 #include <locale>
 #include <codecvt>
-#include <vrcore/assert.h>
+#include <stdarg.h>
 
 #if defined( _WIN32 )
 #include <windows.h>
@@ -128,7 +128,7 @@ std::string Format( const char *pchForma
 	// Something went fairly wrong
 	if ( unSize < 0 )
 	{
-		AssertMsg( false, "Format string parse failure" );
+		//AssertMsg( false, "Format string parse failure" );
 		return "";
 	}
 
@@ -149,7 +149,7 @@ std::string Format( const char *pchForma
 	// Double check, just in case
 	if ( unSize < 0 )
 	{
-		AssertMsg( false, "Format string parse failure" );
+		//AssertMsg( false, "Format string parse failure" );
 		return "";
 	}
 
--- origsrc/openvr-1.23.7/src/vrcommon/vrpathregistry_public.cpp	2022-08-20 05:12:52.000000000 +0900
+++ src/openvr-1.23.7/src/vrcommon/vrpathregistry_public.cpp	2022-10-11 12:56:36.562416600 +0900
@@ -1,11 +1,11 @@
 //========= Copyright Valve Corporation ============//
 
-#include <vrcore/vrpathregistry_public.h>
+#include <vrcommon/vrpathregistry_public.h>
 #include <json/json.h>
-#include <vrcore/pathtools_public.h>
-#include <vrcore/envvartools_public.h>
-#include <vrcore/strtools_public.h>
-#include <vrcore/dirtools_public.h>
+#include <vrcommon/pathtools_public.h>
+#include <vrcommon/envvartools_public.h>
+#include <vrcommon/strtools_public.h>
+#include <vrcommon/dirtools_public.h>
 
 #if defined( WIN32 )
 #include <windows.h>
