--- origsrc/libyahoo2-1.0.1/acinclude.m4	2010-07-27 10:09:05.000000000 +0900
+++ src/libyahoo2-1.0.1/acinclude.m4	2014-12-20 20:14:25.696023000 +0900
@@ -5,7 +5,7 @@ dnl AM_PATH_GLIB_2_0([MINIMUM-VERSION, [
 dnl Test for GLIB, and define GLIB_CFLAGS and GLIB_LIBS, if gmodule, gobject or 
 dnl gthread is specified in MODULES, pass to pkg-config
 dnl
-AC_DEFUN(AM_PATH_GLIB_2_0,
+AC_DEFUN([AM_PATH_GLIB_2_0],
 [dnl 
 dnl Get the cflags and libraries from pkg-config
 dnl
@@ -217,7 +217,7 @@ dnl AM_PATH_GLIB([MINIMUM-VERSION, [ACTI
 dnl Test for GLIB, and define GLIB_CFLAGS and GLIB_LIBS, if "gmodule" or 
 dnl gthread is specified in MODULES, pass to glib-config
 dnl
-AC_DEFUN(AM_PATH_GLIB,
+AC_DEFUN([AM_PATH_GLIB],
 [dnl 
 dnl Get the cflags and libraries from the glib-config script
 dnl
--- origsrc/libyahoo2-1.0.1/src/Makefile.am	2010-07-27 10:09:05.000000000 +0900
+++ src/libyahoo2-1.0.1/src/Makefile.am	2014-12-20 20:12:28.189187100 +0900
@@ -12,7 +12,7 @@ LIBS = @LIBS@ $(GLIB_LIBS)
 libyahoo2_la_SOURCES = libyahoo2.c crypt.c md5.c sha1.c \
 	yahoo_httplib.c yahoo_util.c yahoo_list.c yahoo_fn.c
 
-libyahoo2_la_LDFLAGS = -module -version-info $(YAHOO2_VERSION)
+libyahoo2_la_LDFLAGS = -no-undefined -version-info $(YAHOO2_VERSION)
 
 pkginclude_HEADERS = yahoo2.h yahoo2_callbacks.h yahoo2_types.h \
 	yahoo_httplib.h yahoo_list.h yahoo_fn.h
