--- origsrc/libyahoo2-1.0.1/acinclude.m4	2010-07-27 10:09:05.000000000 +0900
+++ src/libyahoo2-1.0.1/acinclude.m4	2024-10-24 13:52:44.366470900 +0900
@@ -5,7 +5,7 @@ dnl AM_PATH_GLIB_2_0([MINIMUM-VERSION, [
 dnl Test for GLIB, and define GLIB_CFLAGS and GLIB_LIBS, if gmodule, gobject or 
 dnl gthread is specified in MODULES, pass to pkg-config
 dnl
-AC_DEFUN(AM_PATH_GLIB_2_0,
+AC_DEFUN([AM_PATH_GLIB_2_0],
 [dnl 
 dnl Get the cflags and libraries from pkg-config
 dnl
@@ -217,7 +217,7 @@ dnl AM_PATH_GLIB([MINIMUM-VERSION, [ACTI
 dnl Test for GLIB, and define GLIB_CFLAGS and GLIB_LIBS, if "gmodule" or 
 dnl gthread is specified in MODULES, pass to glib-config
 dnl
-AC_DEFUN(AM_PATH_GLIB,
+AC_DEFUN([AM_PATH_GLIB],
 [dnl 
 dnl Get the cflags and libraries from the glib-config script
 dnl
--- origsrc/libyahoo2-1.0.1/libyahoo2.pc.in	2010-07-27 10:09:05.000000000 +0900
+++ src/libyahoo2-1.0.1/libyahoo2.pc.in	2024-10-24 14:18:00.318200200 +0900
@@ -5,7 +5,7 @@ includedir=@includedir@
 
 Name: @PACKAGE@
 Description: Library for the Yahoo! Messenger Protocol
-Version: @VERSION@-@RELEASE@
+Version: @VERSION@
 Requires: @YAHOOPKGREQ@
 Libs: -L${libdir} -lyahoo2
 Cflags: -I${includedir}/@PACKAGE@
--- origsrc/libyahoo2-1.0.1/src/Makefile.am	2010-07-27 10:09:05.000000000 +0900
+++ src/libyahoo2-1.0.1/src/Makefile.am	2024-10-24 13:52:44.366470900 +0900
@@ -12,7 +12,7 @@ LIBS = @LIBS@ $(GLIB_LIBS)
 libyahoo2_la_SOURCES = libyahoo2.c crypt.c md5.c sha1.c \
 	yahoo_httplib.c yahoo_util.c yahoo_list.c yahoo_fn.c
 
-libyahoo2_la_LDFLAGS = -module -version-info $(YAHOO2_VERSION)
+libyahoo2_la_LDFLAGS = -no-undefined -version-info $(YAHOO2_VERSION)
 
 pkginclude_HEADERS = yahoo2.h yahoo2_callbacks.h yahoo2_types.h \
 	yahoo_httplib.h yahoo_list.h yahoo_fn.h
--- origsrc/libyahoo2-1.0.1/src/autoresponder.c	2010-07-27 10:09:05.000000000 +0900
+++ src/libyahoo2-1.0.1/src/autoresponder.c	2024-10-24 13:58:57.412922300 +0900
@@ -708,7 +708,7 @@ static void ext_yahoo_contact_added(int
 	}
 	strcat(buff, "Do you want to allow this [Y/N]?");
 
-/*	print_message((buff));
+/*	print_message(("%s", buff));
 	scanf("%c", &choice);
 	if(choice != 'y' && choice != 'Y')
 		yahoo_reject_buddy(id, who, "Thanks, but no thanks.");
@@ -745,7 +745,7 @@ static void ext_yahoo_mail_notify(int id
 	}
 
 	if(buff[0])
-		print_message((buff));
+		print_message(("%s", buff));
 }
 
 static void ext_yahoo_got_webcam_image(int id, const char *who,
@@ -970,7 +970,7 @@ static void ext_yahoo_login_response(int
 	}
 
 	ylad->status = YAHOO_STATUS_OFFLINE;
-	print_message((buff));
+	print_message(("%s", buff));
 	yahoo_logout();
 	/* poll_loop=0; */
 }
@@ -1155,7 +1155,7 @@ void yahoo_callback(struct conn_handler
 				"Yahoo read error: Server closed socket");
 
 		if(buff[0])
-			print_message((buff));
+			print_message(("%s", buff));
 	}
 }
 
--- origsrc/libyahoo2-1.0.1/src/sample_client.c	2010-07-27 10:09:05.000000000 +0900
+++ src/libyahoo2-1.0.1/src/sample_client.c	2024-10-24 13:58:05.187576500 +0900
@@ -713,7 +713,7 @@ static void ext_yahoo_mail_notify(int id
 	}
 
 	if(buff[0])
-		print_message((buff));
+		print_message(("%s", buff));
 }
 
 static void ext_yahoo_got_webcam_image(int id, const char *who,
@@ -922,7 +922,7 @@ static void ext_yahoo_login_response(int
 	}
 
 	ylad->status = YAHOO_STATUS_OFFLINE;
-	print_message((buff));
+	print_message(("%s", buff));
 	yahoo_logout();
 	poll_loop=0;
 }
@@ -1126,7 +1126,7 @@ void yahoo_callback(struct conn_handler
 				"Yahoo read error: Server closed socket");
 
 		if(buff[0])
-			print_message((buff));
+			print_message(("%s", buff));
 	}
 }
 
