--- origsrc/libfilezilla-0.5.3/configure.ac	2016-06-20 17:09:42.000000000 +0900
+++ src/libfilezilla-0.5.3/configure.ac	2016-06-26 18:18:45.797376100 +0900
@@ -50,7 +50,7 @@ AC_CHECK_FUNCS([clock_gettime])
 AC_CHECK_DECLS([pthread_condattr_setclock], [], [], [[#include <pthread.h>]])
 
 # Check if we're on Windows
-if echo $host_os | grep 'cygwin\|mingw\|^msys$' > /dev/null 2>&1; then
+if echo $host_os | grep 'mingw\|^msys$' > /dev/null 2>&1; then
   windows=1
 else
   windows=0
