--- origsrc/libfilezilla-0.9.0/configure.ac	2016-11-29 01:33:58.000000000 +0900
+++ src/libfilezilla-0.9.0/configure.ac	2016-12-01 22:24:26.448954100 +0900
@@ -67,7 +67,7 @@ CHECK_CLOCK_GETTIME
 AC_CHECK_DECLS([pthread_condattr_setclock], [], [], [[#include <pthread.h>]])
 
 # Check if we're on Windows
-if echo $host_os | grep 'cygwin\|mingw\|^msys$' > /dev/null 2>&1; then
+if echo $host_os | grep 'mingw\|^msys$' > /dev/null 2>&1; then
   windows=1
 else
   windows=0
@@ -182,8 +182,7 @@ fi
 # CppUnit tests
 # -------------
 
-AM_PATH_CPPUNIT([1.10.2], [has_cppunit="yes"])
-
+PKG_CHECK_MODULES([CPPUNIT], [cppunit >= 1.10.2], [has_cppunit="yes"])
 
 # Doxygen
 # -------
