--- origsrc/anerley/anerley/Makefile.am	2011-05-07 08:24:43.000000000 +0900
+++ src/anerley/anerley/Makefile.am	2011-05-07 09:20:47.801939100 +0900
@@ -60,4 +60,4 @@ libanerley_la_CFLAGS = -I$(top_srcdir) \
 		       -D_GNU_SOURCE
 libanerley_la_LIBADD = $(GLIB_LIBS) $(TELEPATHY_LIBS) \
 		       $(MX_LIBS) $(EBOOK_LIBS)
-
+libanerley_la_LDFLAGS = -no-undefined
--- origsrc/anerley/configure.ac	2011-05-07 08:24:43.000000000 +0900
+++ src/anerley/configure.ac	2011-05-07 08:26:50.275179800 +0900
@@ -21,7 +21,7 @@ GLIB_DEFINE_LOCALEDIR(LOCALEDIR)
 AC_PROG_CC
 AC_PROG_CC_STDC
 AC_PROG_INSTALL
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_ISC_POSIX
 AC_HEADER_STDC
 AM_PROG_CC_C_O
