--- origsrc/libspatialite-4.2.0/configure.ac	2014-07-25 19:36:10.000000000 +0900
+++ src/libspatialite-4.2.0/configure.ac	2015-03-30 22:50:54.127954400 +0900
@@ -364,4 +364,6 @@ AM_CONDITIONAL([MACOSX], [test "$target_
 # Checking for Android
 AM_CONDITIONAL([ANDROID], [test "$target_alias" = "android"])
 
+AM_CONDITIONAL([CYGWIN], [test "$host_os" = "cygwin"])
+
 AC_OUTPUT
--- origsrc/libspatialite-4.2.0/examples/Makefile.am	2014-07-25 19:36:10.000000000 +0900
+++ src/libspatialite-4.2.0/examples/Makefile.am	2015-03-30 22:46:51.040499800 +0900
@@ -1,7 +1,7 @@
 noinst_PROGRAMS = demo1 demo2 demo3 demo4 demo5
 
 AM_CFLAGS = -I@srcdir@/../src/headers 
-AM_LDFLAGS = -L../src -lspatialite -lm $(GCOV_FLAGS)
+AM_LDFLAGS = $(top_builddir)/src/libspatialite.la -lm $(GCOV_FLAGS)
 
 LDADD = @LIBXML2_LIBS@
 
--- origsrc/libspatialite-4.2.0/src/Makefile.am	2014-07-25 19:36:10.000000000 +0900
+++ src/libspatialite-4.2.0/src/Makefile.am	2015-03-30 22:52:14.571830800 +0900
@@ -42,6 +42,9 @@ libspatialite_la_LDFLAGS = -version-info
 libspatialite_la_LIBADD += -ldl
 else
 libspatialite_la_LDFLAGS = -version-info 7:0:0
+if CYGWIN
+libspatialite_la_LDFLAGS += -no-undefined
+endif
 libspatialite_la_LIBADD += -lpthread -ldl
 endif
 endif
@@ -73,9 +76,13 @@ if ANDROID
 mod_spatialite_la_LDFLAGS = -module -version-info 7:0:0
 mod_spatialite_la_LIBADD += -ldl
 else
+if CYGWIN
+mod_spatialite_la_LDFLAGS = -module -avoid-version -no-undefined
+else
 mod_spatialite_la_LDFLAGS = -module -version-info 7:0:0
 mod_spatialite_la_LIBADD += -lpthread -ldl
 endif
 endif
+endif
 
 MOSTLYCLEANFILES = *.gcna *.gcno *.gcda
--- origsrc/libspatialite-4.2.0/src/headers/spatialite.h	2014-07-25 19:36:10.000000000 +0900
+++ src/libspatialite-4.2.0/src/headers/spatialite.h	2015-03-30 23:14:12.088977400 +0900
@@ -48,7 +48,7 @@ the terms of any one of the MPL, the GPL
  Main SpatiaLite header file
  */
 #ifndef DOXYGEN_SHOULD_SKIP_THIS
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__CYGWIN__)
 #ifdef DLL_EXPORT
 #define SPATIALITE_DECLARE __declspec(dllexport)
 #else
--- origsrc/libspatialite-4.2.0/test/Makefile.am	2014-07-25 19:36:10.000000000 +0900
+++ src/libspatialite-4.2.0/test/Makefile.am	2015-03-30 22:48:56.017596700 +0900
@@ -85,7 +85,7 @@ check_PROGRAMS += \
 endif
 
 AM_CFLAGS = -I@srcdir@/../src/headers -I@srcdir@
-AM_LDFLAGS = -L../src -lspatialite -lm $(GCOV_FLAGS)
+AM_LDFLAGS = $(top_builddir)/src/libspatialite.la -lm $(GCOV_FLAGS)
 
 TESTS = $(check_PROGRAMS)
 
