--- origsrc/libspatialite-4.3.0a/examples/Makefile.am	2015-07-01 15:59:16.000000000 +0900
+++ src/libspatialite-4.3.0a/examples/Makefile.am	2016-04-21 08:22:29.239264100 +0900
@@ -1,7 +1,7 @@
 noinst_PROGRAMS = demo1 demo2 demo3 demo4 demo5
 
 AM_CFLAGS = -I@srcdir@/../src/headers 
-AM_LDFLAGS = -L../src -lspatialite -lm $(GCOV_FLAGS)
+AM_LDFLAGS = $(top_builddir)/src/libspatialite.la -lm $(GCOV_FLAGS)
 
 LDADD = @LIBXML2_LIBS@
 
--- origsrc/libspatialite-4.3.0a/src/Makefile.am	2015-07-01 15:59:16.000000000 +0900
+++ src/libspatialite-4.3.0a/src/Makefile.am	2016-04-21 08:22:29.239264100 +0900
@@ -45,7 +45,7 @@ if ANDROID
 libspatialite_la_LDFLAGS = -version-info 8:0:1
 libspatialite_la_LIBADD += -ldl -lm
 else
-libspatialite_la_LDFLAGS = -version-info 8:0:1
+libspatialite_la_LDFLAGS = -no-undefined -version-info 8:0:1
 libspatialite_la_LIBADD += -lpthread -ldl -lm
 endif
 endif
@@ -79,7 +79,7 @@ if ANDROID
 mod_spatialite_la_LDFLAGS = -module -version-info 8:0:1
 mod_spatialite_la_LIBADD += -ldl -lm
 else
-mod_spatialite_la_LDFLAGS = -module -version-info 8:0:1
+mod_spatialite_la_LDFLAGS = -no-undefined -module -avoid-version
 mod_spatialite_la_LIBADD += -lpthread -ldl -lm
 endif
 endif
--- origsrc/libspatialite-4.3.0a/src/headers/spatialite.h	2015-07-01 15:59:16.000000000 +0900
+++ src/libspatialite-4.3.0a/src/headers/spatialite.h	2016-04-21 08:22:29.239264100 +0900
@@ -48,7 +48,7 @@ the terms of any one of the MPL, the GPL
  Main SpatiaLite header file
  */
 #ifndef DOXYGEN_SHOULD_SKIP_THIS
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__CYGWIN__)
 #ifdef DLL_EXPORT
 #define SPATIALITE_DECLARE __declspec(dllexport)
 #else
--- origsrc/libspatialite-4.3.0a/test/Makefile.am	2015-07-01 15:59:16.000000000 +0900
+++ src/libspatialite-4.3.0a/test/Makefile.am	2016-04-21 08:22:29.239264100 +0900
@@ -91,9 +91,9 @@ endif
 AM_CFLAGS = -I@srcdir@/../src/headers -I@srcdir@
 
 if MINGW
-AM_LDFLAGS = -L../src -lspatialite -lm -lxml2 $(GCOV_FLAGS)
+AM_LDFLAGS = $(top_builddir)/src/libspatialite.la -lm -lxml2 $(GCOV_FLAGS)
 else 
-AM_LDFLAGS = -L../src -lpthread -lspatialite -lm -lxml2 $(GCOV_FLAGS)
+AM_LDFLAGS = $(top_builddir)/src/libspatialite.la -lpthread -lm -lxml2 $(GCOV_FLAGS)
 endif
 
 TESTS = $(check_PROGRAMS)
