--- origsrc/polarssl-0.13.1/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/polarssl-0.13.1/Makefile.am	2010-06-19 15:34:24.226496700 +0900
@@ -0,0 +1,2 @@
+SUBDIRS = include library programs
+AM_CFLAGS = -D_FILE_OFFSET_BITS=64 -Wall -Wdeclaration-after-statement
--- origsrc/polarssl-0.13.1/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/polarssl-0.13.1/configure.ac	2010-06-19 15:04:54.202233700 +0900
@@ -0,0 +1,52 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.65])
+AC_INIT([FULL-PACKAGE-NAME], [VERSION], [BUG-REPORT-ADDRESS])
+AC_CONFIG_SRCDIR([config.h.in])
+AC_CONFIG_HEADERS([config.h])
+
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_AWK
+AC_PROG_CC
+AC_PROG_CPP
+AC_PROG_INSTALL
+AC_PROG_LN_S
+AC_PROG_MAKE_SET
+
+AC_LIBTOOL_WIN32_DLL
+AC_PROG_LIBTOOL
+
+# Checks for libraries.
+# FIXME: Replace `main' with a function in `-lpolarssl':
+AC_CHECK_LIB([polarssl], [main])
+
+# Checks for header files.
+AC_CHECK_HEADERS([arpa/inet.h fcntl.h float.h inttypes.h netdb.h netinet/in.h stdlib.h string.h sys/socket.h sys/time.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_C_INLINE
+AC_TYPE_OFF_T
+AC_TYPE_SIZE_T
+AC_TYPE_UINT32_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([alarm gethostbyname gettimeofday memset select socket strchr strstr])
+
+AC_CONFIG_FILES([Makefile
+                 include/Makefile
+                 include/polarssl/Makefile
+                 library/Makefile
+                 programs/Makefile
+                 programs/aes/Makefile
+                 programs/hash/Makefile
+                 programs/pkey/Makefile
+                 programs/ssl/Makefile
+                 programs/test/Makefile
+                 programs/x509/Makefile])
+AC_OUTPUT
--- origsrc/polarssl-0.13.1/include/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/polarssl-0.13.1/include/Makefile.am	2010-06-19 14:53:03.483483700 +0900
@@ -0,0 +1 @@
+SUBDIRS = polarssl
--- origsrc/polarssl-0.13.1/include/polarssl/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/polarssl-0.13.1/include/polarssl/Makefile.am	2010-06-19 14:54:03.874108700 +0900
@@ -0,0 +1,6 @@
+polarssl_includedir = $(includedir)/polarssl
+polarssl_include_HEADERS = \
+aes.h     bn_mul.h    debug.h   md2.h  openssl.h  sha2.h    x509.h \
+arc4.h    camellia.h  des.h     md4.h  padlock.h  sha4.h    xtea.h \
+base64.h  certs.h     dhm.h     md5.h  rsa.h      ssl.h \
+bignum.h  config.h    havege.h  net.h  sha1.h     timing.h
--- origsrc/polarssl-0.13.1/library/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/polarssl-0.13.1/library/Makefile.am	2010-06-19 14:55:06.592858700 +0900
@@ -0,0 +1,8 @@
+lib_LTLIBRARIES=libpolarssl.la
+libpolarssl_la_SOURCES = \
+aes.c     camellia.c  dhm.c     md5.c      sha1.c     ssl_srv.c    xtea.c \
+arc4.c    certs.c     havege.c  net.c      sha2.c     ssl_tls.c \
+base64.c  debug.c     md2.c     padlock.c  sha4.c     timing.c \
+bignum.c  des.c       md4.c     rsa.c      ssl_cli.c  x509parse.c
+libpolarssl_la_LDFLAGS = -no-undefined
+INCLUDES = -I$(top_srcdir)/include
--- origsrc/polarssl-0.13.1/programs/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/polarssl-0.13.1/programs/Makefile.am	2010-06-19 12:54:20.327233700 +0900
@@ -0,0 +1 @@
+SUBDIRS = aes hash pkey ssl test x509
--- origsrc/polarssl-0.13.1/programs/aes/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/polarssl-0.13.1/programs/aes/Makefile.am	2010-06-19 13:57:05.717858700 +0900
@@ -0,0 +1,4 @@
+bin_PROGRAMS = polarssl_aescrypt2
+polarssl_aescrypt2_SOURCES = aescrypt2.c
+polarssl_aescrypt2_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+INCLUDES = -I$(top_srcdir)/include
--- origsrc/polarssl-0.13.1/programs/hash/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/polarssl-0.13.1/programs/hash/Makefile.am	2010-06-19 13:59:31.217858700 +0900
@@ -0,0 +1,10 @@
+bin_PROGRAMS = polarssl_hello polarssl_md5sum polarssl_sha1sum polarssl_sha2sum
+polarssl_hello_SOURCES = hello.c
+polarssl_hello_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+polarssl_md5sum_SOURCES = md5sum.c
+polarssl_md5sum_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+polarssl_sha1sum_SOURCES = sha1sum.c
+polarssl_sha1sum_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+polarssl_sha2sum_SOURCES = sha2sum.c
+polarssl_sha2sum_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+INCLUDES = -I$(top_srcdir)/include
--- origsrc/polarssl-0.13.1/programs/pkey/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/polarssl-0.13.1/programs/pkey/Makefile.am	2010-06-19 13:59:40.577233700 +0900
@@ -0,0 +1,17 @@
+bin_PROGRAMS = polarssl_dh_client polarssl_dh_genprime polarssl_dh_server \
+polarssl_mpi_demo polarssl_rsa_genkey polarssl_rsa_sign polarssl_rsa_verify
+polarssl_dh_client_SOURCES = dh_client.c
+polarssl_dh_client_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+polarssl_dh_genprime_SOURCES = dh_genprime.c
+polarssl_dh_genprime_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+polarssl_dh_server_SOURCES = dh_server.c
+polarssl_dh_server_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+polarssl_mpi_demo_SOURCES = mpi_demo.c
+polarssl_mpi_demo_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+polarssl_rsa_genkey_SOURCES = rsa_genkey.c
+polarssl_rsa_genkey_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+polarssl_rsa_sign_SOURCES = rsa_sign.c
+polarssl_rsa_sign_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+polarssl_rsa_verify_SOURCES = rsa_verify.c
+polarssl_rsa_verify_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+INCLUDES = -I$(top_srcdir)/include
--- origsrc/polarssl-0.13.1/programs/ssl/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/polarssl-0.13.1/programs/ssl/Makefile.am	2010-06-19 13:59:50.139733700 +0900
@@ -0,0 +1,8 @@
+bin_PROGRAMS = polarssl_ssl_client1 polarssl_ssl_client2 polarssl_ssl_server
+polarssl_ssl_client1_SOURCES = ssl_client1.c
+polarssl_ssl_client1_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+polarssl_ssl_client2_SOURCES = ssl_client2.c
+polarssl_ssl_client2_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+polarssl_ssl_server_SOURCES = ssl_server.c
+polarssl_ssl_server_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+INCLUDES = -I$(top_srcdir)/include
--- origsrc/polarssl-0.13.1/programs/test/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/polarssl-0.13.1/programs/test/Makefile.am	2010-06-19 14:54:41.624108700 +0900
@@ -0,0 +1,11 @@
+TESTS = selftest benchmark #ssl_test ssl_cert_test
+check_PROGRAMS = selftest benchmark ssl_test ssl_cert_test
+selftest_SOURCES = selftest.c
+selftest_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+benchmark_SOURCES = benchmark.c
+benchmark_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+ssl_test_SOURCES = ssl_test.c
+ssl_test_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+ssl_cert_test_SOURCES = ssl_cert_test.c
+ssl_cert_test_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+INCLUDES = -I$(top_srcdir)/include
--- origsrc/polarssl-0.13.1/programs/x509/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/polarssl-0.13.1/programs/x509/Makefile.am	2010-06-19 14:00:09.233483700 +0900
@@ -0,0 +1,4 @@
+bin_PROGRAMS = polarssl_cert_app
+polarssl_cert_app_SOURCES = cert_app.c
+polarssl_cert_app_LDFLAGS = $(top_builddir)/library/libpolarssl.la
+INCLUDES = -I$(top_srcdir)/include
