--- origsrc/polarssl-1.2.8/library/CMakeLists.txt	2013-06-21 22:11:10.000000000 +0900
+++ src/polarssl-1.2.8/library/CMakeLists.txt	2013-09-09 22:18:09.126408400 +0900
@@ -66,5 +66,6 @@ endif(NOT USE_SHARED_POLARSSL_LIBRARY)
 target_link_libraries(polarssl ${libs})
 
 install(TARGETS polarssl
-        DESTINATION ${LIB_INSTALL_DIR}
+        RUNTIME DESTINATION bin
+        ARCHIVE DESTINATION lib
         PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
