--- origsrc/polarssl-1.1.3/library/CMakeLists.txt	2012-04-30 05:30:16.000000000 +0900
+++ src/polarssl-1.1.3/library/CMakeLists.txt	2012-05-14 19:38:04.860708300 +0900
@@ -52,5 +52,6 @@ set_target_properties(polarssl PROPERTIE
 endif(NOT USE_SHARED_POLARSSL_LIBRARY)
 
 install(TARGETS polarssl
-        DESTINATION ${LIB_INSTALL_DIR}
+        RUNTIME DESTINATION bin
+        ARCHIVE DESTINATION lib
         PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
--- origsrc/polarssl-1.1.3/programs/aes/CMakeLists.txt	2011-12-12 01:35:09.000000000 +0900
+++ src/polarssl-1.1.3/programs/aes/CMakeLists.txt	2012-05-14 19:38:04.876409400 +0900
@@ -1,9 +1,9 @@
-add_executable(aescrypt2 aescrypt2.c)
-target_link_libraries(aescrypt2 polarssl)
+add_executable(polarssl_aescrypt2 aescrypt2.c)
+target_link_libraries(polarssl_aescrypt2 polarssl)
 
-add_executable(crypt_and_hash crypt_and_hash.c)
-target_link_libraries(crypt_and_hash polarssl)
+add_executable(polarssl_crypt_and_hash crypt_and_hash.c)
+target_link_libraries(polarssl_crypt_and_hash polarssl)
 
-install(TARGETS aescrypt2 crypt_and_hash
+install(TARGETS polarssl_aescrypt2 polarssl_crypt_and_hash
         DESTINATION "bin"
         PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
--- origsrc/polarssl-1.1.3/programs/hash/CMakeLists.txt	2011-12-12 01:35:09.000000000 +0900
+++ src/polarssl-1.1.3/programs/hash/CMakeLists.txt	2012-05-14 19:38:04.876409400 +0900
@@ -1,18 +1,18 @@
-add_executable(hello hello.c)
-target_link_libraries(hello polarssl)
+add_executable(polarssl_hello hello.c)
+target_link_libraries(polarssl_hello polarssl)
 
-add_executable(generic_sum generic_sum.c)
-target_link_libraries(generic_sum polarssl)
+add_executable(polarssl_generic_sum generic_sum.c)
+target_link_libraries(polarssl_generic_sum polarssl)
 
-add_executable(md5sum md5sum.c)
-target_link_libraries(md5sum polarssl)
+add_executable(polarssl_md5sum md5sum.c)
+target_link_libraries(polarssl_md5sum polarssl)
 
-add_executable(sha1sum sha1sum.c)
-target_link_libraries(sha1sum polarssl)
+add_executable(polarssl_sha1sum sha1sum.c)
+target_link_libraries(polarssl_sha1sum polarssl)
 
-add_executable(sha2sum sha2sum.c)
-target_link_libraries(sha2sum polarssl)
+add_executable(polarssl_sha2sum sha2sum.c)
+target_link_libraries(polarssl_sha2sum polarssl)
 
-install(TARGETS hello md5sum sha1sum sha2sum generic_sum
+install(TARGETS polarssl_hello polarssl_md5sum polarssl_sha1sum polarssl_sha2sum polarssl_generic_sum
         DESTINATION "bin"
         PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
--- origsrc/polarssl-1.1.3/programs/pkey/CMakeLists.txt	2011-12-12 01:35:09.000000000 +0900
+++ src/polarssl-1.1.3/programs/pkey/CMakeLists.txt	2012-05-14 19:38:04.892110500 +0900
@@ -1,39 +1,39 @@
-add_executable(dh_client dh_client.c)
-target_link_libraries(dh_client polarssl)
+add_executable(polarssl_dh_client dh_client.c)
+target_link_libraries(polarssl_dh_client polarssl)
 
-add_executable(dh_genprime dh_genprime.c)
-target_link_libraries(dh_genprime polarssl)
+add_executable(polarssl_dh_genprime dh_genprime.c)
+target_link_libraries(polarssl_dh_genprime polarssl)
 
-add_executable(dh_server dh_server.c)
-target_link_libraries(dh_server polarssl)
+add_executable(polarssl_dh_server dh_server.c)
+target_link_libraries(polarssl_dh_server polarssl)
 
-add_executable(key_app key_app.c)
-target_link_libraries(key_app polarssl)
+add_executable(polarssl_key_app key_app.c)
+target_link_libraries(polarssl_key_app polarssl)
 
-add_executable(mpi_demo mpi_demo.c)
-target_link_libraries(mpi_demo polarssl)
+add_executable(polarssl_mpi_demo mpi_demo.c)
+target_link_libraries(polarssl_mpi_demo polarssl)
 
-add_executable(rsa_genkey rsa_genkey.c)
-target_link_libraries(rsa_genkey polarssl)
+add_executable(polarssl_rsa_genkey rsa_genkey.c)
+target_link_libraries(polarssl_rsa_genkey polarssl)
 
-add_executable(rsa_sign rsa_sign.c)
-target_link_libraries(rsa_sign polarssl)
+add_executable(polarssl_rsa_sign rsa_sign.c)
+target_link_libraries(polarssl_rsa_sign polarssl)
 
-add_executable(rsa_verify rsa_verify.c)
-target_link_libraries(rsa_verify polarssl)
+add_executable(polarssl_rsa_verify rsa_verify.c)
+target_link_libraries(polarssl_rsa_verify polarssl)
 
-add_executable(rsa_sign_pss rsa_sign_pss.c)
-target_link_libraries(rsa_sign_pss polarssl)
+add_executable(polarssl_rsa_sign_pss rsa_sign_pss.c)
+target_link_libraries(polarssl_rsa_sign_pss polarssl)
 
-add_executable(rsa_verify_pss rsa_verify_pss.c)
-target_link_libraries(rsa_verify_pss polarssl)
+add_executable(polarssl_rsa_verify_pss rsa_verify_pss.c)
+target_link_libraries(polarssl_rsa_verify_pss polarssl)
 
-add_executable(rsa_encrypt rsa_encrypt.c)
-target_link_libraries(rsa_encrypt polarssl)
+add_executable(polarssl_rsa_encrypt rsa_encrypt.c)
+target_link_libraries(polarssl_rsa_encrypt polarssl)
 
-add_executable(rsa_decrypt rsa_decrypt.c)
-target_link_libraries(rsa_decrypt polarssl)
+add_executable(polarssl_rsa_decrypt rsa_decrypt.c)
+target_link_libraries(polarssl_rsa_decrypt polarssl)
 
-install(TARGETS dh_client dh_genprime dh_server key_app mpi_demo rsa_genkey rsa_sign rsa_verify rsa_encrypt rsa_decrypt
+install(TARGETS polarssl_dh_client polarssl_dh_genprime polarssl_dh_server polarssl_key_app polarssl_mpi_demo polarssl_rsa_genkey polarssl_rsa_sign polarssl_rsa_verify polarssl_rsa_encrypt polarssl_rsa_decrypt
         DESTINATION "bin"
         PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
--- origsrc/polarssl-1.1.3/programs/random/CMakeLists.txt	2011-12-12 01:35:09.000000000 +0900
+++ src/polarssl-1.1.3/programs/random/CMakeLists.txt	2012-05-14 19:38:04.892110500 +0900
@@ -1,12 +1,12 @@
-add_executable(gen_random_havege gen_random_havege.c)
-target_link_libraries(gen_random_havege polarssl)
+add_executable(polarssl_gen_random_havege gen_random_havege.c)
+target_link_libraries(polarssl_gen_random_havege polarssl)
 
-add_executable(gen_random_ctr_drbg gen_random_ctr_drbg.c)
-target_link_libraries(gen_random_ctr_drbg polarssl)
+add_executable(polarssl_gen_random_ctr_drbg gen_random_ctr_drbg.c)
+target_link_libraries(polarssl_gen_random_ctr_drbg polarssl)
 
-add_executable(gen_entropy gen_entropy.c)
-target_link_libraries(gen_entropy polarssl)
+add_executable(polarssl_gen_entropy gen_entropy.c)
+target_link_libraries(polarssl_gen_entropy polarssl)
 
-install(TARGETS gen_random_havege gen_random_ctr_drbg gen_entropy
+install(TARGETS polarssl_gen_random_havege polarssl_gen_random_ctr_drbg polarssl_gen_entropy
         DESTINATION "bin"
         PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
--- origsrc/polarssl-1.1.3/programs/ssl/CMakeLists.txt	2011-12-12 01:35:09.000000000 +0900
+++ src/polarssl-1.1.3/programs/ssl/CMakeLists.txt	2012-05-14 19:38:04.892110500 +0900
@@ -6,21 +6,21 @@ if(USE_PKCS11_HELPER_LIBRARY)
     set(libs ${libs} pkcs11-helper)
 endif(USE_PKCS11_HELPER_LIBRARY)
 
-add_executable(ssl_client1 ssl_client1.c)
-target_link_libraries(ssl_client1 ${libs})
+add_executable(polarssl_ssl_client1 ssl_client1.c)
+target_link_libraries(polarssl_ssl_client1 ${libs})
 
-add_executable(ssl_client2 ssl_client2.c)
-target_link_libraries(ssl_client2 ${libs})
+add_executable(polarssl_ssl_client2 ssl_client2.c)
+target_link_libraries(polarssl_ssl_client2 ${libs})
 
-add_executable(ssl_server ssl_server.c)
-target_link_libraries(ssl_server ${libs})
+add_executable(polarssl_ssl_server ssl_server.c)
+target_link_libraries(polarssl_ssl_server ${libs})
 
-add_executable(ssl_fork_server ssl_fork_server.c)
-target_link_libraries(ssl_fork_server ${libs})
+add_executable(polarssl_ssl_fork_server ssl_fork_server.c)
+target_link_libraries(polarssl_ssl_fork_server ${libs})
 
-add_executable(ssl_mail_client ssl_mail_client.c)
-target_link_libraries(ssl_mail_client ${libs})
+add_executable(polarssl_ssl_mail_client ssl_mail_client.c)
+target_link_libraries(polarssl_ssl_mail_client ${libs})
 
-install(TARGETS ssl_client1 ssl_client2 ssl_server ssl_fork_server ssl_mail_client
+install(TARGETS polarssl_ssl_client1 polarssl_ssl_client2 polarssl_ssl_server polarssl_ssl_fork_server polarssl_ssl_mail_client
         DESTINATION "bin"
         PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
--- origsrc/polarssl-1.1.3/programs/test/CMakeLists.txt	2011-12-12 01:35:09.000000000 +0900
+++ src/polarssl-1.1.3/programs/test/CMakeLists.txt	2012-05-14 19:38:04.892110500 +0900
@@ -6,18 +6,18 @@ if(USE_PKCS11_HELPER_LIBRARY)
     set(libs ${libs} pkcs11-helper)
 endif(USE_PKCS11_HELPER_LIBRARY)
 
-add_executable(selftest selftest.c)
-target_link_libraries(selftest ${libs})
+add_executable(polarssl_selftest selftest.c)
+target_link_libraries(polarssl_selftest ${libs})
 
-add_executable(benchmark benchmark.c)
-target_link_libraries(benchmark ${libs})
+add_executable(polarssl_benchmark benchmark.c)
+target_link_libraries(polarssl_benchmark ${libs})
 
-add_executable(ssl_test ssl_test.c)
-target_link_libraries(ssl_test ${libs})
+add_executable(polarssl_ssl_test ssl_test.c)
+target_link_libraries(polarssl_ssl_test ${libs})
 
-add_executable(ssl_cert_test ssl_cert_test.c)
-target_link_libraries(ssl_cert_test ${libs})
+add_executable(polarssl_ssl_cert_test ssl_cert_test.c)
+target_link_libraries(polarssl_ssl_cert_test ${libs})
 
-install(TARGETS selftest benchmark ssl_test ssl_cert_test
+install(TARGETS polarssl_selftest polarssl_benchmark polarssl_ssl_test polarssl_ssl_cert_test
         DESTINATION "bin"
         PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
--- origsrc/polarssl-1.1.3/programs/x509/CMakeLists.txt	2011-12-12 01:35:09.000000000 +0900
+++ src/polarssl-1.1.3/programs/x509/CMakeLists.txt	2012-05-14 19:38:04.907811600 +0900
@@ -6,12 +6,12 @@ if(USE_PKCS11_HELPER_LIBRARY)
     set(libs ${libs} pkcs11-helper)
 endif(USE_PKCS11_HELPER_LIBRARY)
 
-add_executable(cert_app cert_app.c)
-target_link_libraries(cert_app ${libs})
+add_executable(polarssl_cert_app cert_app.c)
+target_link_libraries(polarssl_cert_app ${libs})
 
-add_executable(crl_app crl_app.c)
-target_link_libraries(crl_app ${libs})
+add_executable(polarssl_crl_app crl_app.c)
+target_link_libraries(polarssl_crl_app ${libs})
 
-install(TARGETS cert_app crl_app
+install(TARGETS polarssl_cert_app polarssl_crl_app
         DESTINATION "bin"
         PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
