--- origsrc/jacui-0.2.2/Makefile.am	2011-05-15 02:40:45.000000000 +0900
+++ src/jacui-0.2.2/Makefile.am	2012-12-04 15:16:25.804325300 +0900
@@ -11,7 +11,8 @@ AM_CXXFLAGS =
 
 AUTOMAKE_OPTIONS = subdir-objects
 
-include_HEADERS = \
+jacui_includedir = $(includedir)/jacui
+jacui_include_HEADERS = \
 	src/jacui/canvas.hpp \
 	src/jacui/cursor.hpp \
 	src/jacui/cursors.hpp \
@@ -41,7 +42,7 @@ libjacui_sdl1_2_la_SOURCES = \
 	src/sdl1.2/types.cpp \
 	src/sdl1.2/window.cpp
 
-libjacui_sdl1_2_la_LDFLAGS = -version-info $(SO_VERSION)
+libjacui_sdl1_2_la_LDFLAGS = -version-info $(SO_VERSION) -no-undefined
 libjacui_sdl1_2_la_CPPFLAGS = -I$(top_srcdir)/src $(SDL_CFLAGS)
 libjacui_sdl1_2_la_LIBADD = $(SDL_LIBS)
 
