--- origsrc/liblnk-20240423/Makefile.am	2024-04-23 13:34:36.000000000 +0900
+++ src/liblnk-20240423/Makefile.am	2025-05-18 14:28:39.761048300 +0900
@@ -3,25 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfdatetime \
-	libfguid \
-	libfole \
-	libfwps \
-	libfwsi \
 	liblnk \
 	lnktools \
 	pylnk \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/liblnk-20240423/configure.ac	2024-04-23 13:34:36.000000000 +0900
+++ src/liblnk-20240423/configure.ac	2025-05-18 14:28:39.761048300 +0900
@@ -166,21 +166,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfole/Makefile])
-AC_CONFIG_FILES([libfwps/Makefile])
-AC_CONFIG_FILES([libfwsi/Makefile])
 AC_CONFIG_FILES([liblnk/Makefile])
 AC_CONFIG_FILES([pylnk/Makefile])
 AC_CONFIG_FILES([lnktools/Makefile])
--- origsrc/liblnk-20240423/pylnk/Makefile.am	2024-04-23 13:34:36.000000000 +0900
+++ src/liblnk-20240423/pylnk/Makefile.am	2025-05-18 14:28:54.897105800 +0900
@@ -54,7 +54,7 @@ pylnk_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pylnk_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pylnk_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pylnk_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
