--- origsrc/siscone-3.0.4/siscone/Makefile.am	2018-10-01 16:13:20.000000000 +0900
+++ src/siscone-3.0.4/siscone/Makefile.am	2019-05-06 12:22:08.084480600 +0900
@@ -8,6 +8,7 @@ libsiscone_la_SOURCES = ranlux.cpp refer
    momentum.cpp hash.cpp quadtree.cpp vicinity.cpp\
    protocones.cpp split_merge.cpp siscone.cpp\
    siscone_error.cpp area.cpp
+libsiscone_la_LDFLAGS = -no-undefined
 
 EXTRA_DIST = makefile.static
 
--- origsrc/siscone-3.0.4/siscone/spherical/Makefile.am	2018-10-01 16:13:20.000000000 +0900
+++ src/siscone-3.0.4/siscone/spherical/Makefile.am	2019-05-06 12:22:24.130777400 +0900
@@ -11,6 +11,7 @@ libsiscone_spherical_la_CXXFLAGS = $(AM_
 libsiscone_spherical_la_SOURCES = geom_2d.cpp momentum.cpp hash.cpp\
    vicinity.cpp protocones.cpp split_merge.cpp siscone.cpp
 libsiscone_spherical_la_LIBADD = ../libsiscone.la
+libsiscone_spherical_la_LDFLAGS = -no-undefined
 
 EXTRA_DIST = makefile.static
 
