--- origsrc/siscone-3.0.6/siscone/Makefile.am	2023-06-20 21:57:07.000000000 +0900
+++ src/siscone-3.0.6/siscone/Makefile.am	2024-09-28 08:53:33.862091400 +0900
@@ -8,6 +8,7 @@ libsiscone_la_SOURCES = ranlux.cpp refer
    momentum.cpp hash.cpp quadtree.cpp vicinity.cpp\
    protocones.cpp split_merge.cpp siscone.cpp\
    siscone_error.cpp area.cpp
+libsiscone_la_LDFLAGS = -no-undefined
 
 EXTRA_DIST = makefile.static
 
--- origsrc/siscone-3.0.6/siscone/spherical/Makefile.am	2023-06-20 21:57:07.000000000 +0900
+++ src/siscone-3.0.6/siscone/spherical/Makefile.am	2024-09-28 08:53:33.862091400 +0900
@@ -11,6 +11,7 @@ libsiscone_spherical_la_CXXFLAGS = $(AM_
 libsiscone_spherical_la_SOURCES = geom_2d.cpp momentum.cpp hash.cpp\
    vicinity.cpp protocones.cpp split_merge.cpp siscone.cpp
 libsiscone_spherical_la_LIBADD = ../libsiscone.la
+libsiscone_spherical_la_LDFLAGS = -no-undefined
 
 EXTRA_DIST = makefile.static
 
