--- origsrc/siscone-2.0.6/siscone/Makefile.am	2013-02-05 07:44:20.000000000 +0900
+++ src/siscone-2.0.6/siscone/Makefile.am	2014-03-09 11:15:30.559563400 +0900
@@ -7,6 +7,7 @@ libsiscone_la_SOURCES = ranlux.cpp refer
    momentum.cpp hash.cpp quadtree.cpp vicinity.cpp\
    protocones.cpp split_merge.cpp siscone.cpp\
    siscone_error.cpp area.cpp
+libsiscone_la_LDFLAGS = -no-undefined
 
 EXTRA_DIST = makefile.static
 
--- origsrc/siscone-2.0.6/siscone/spherical/Makefile.am	2013-02-05 02:12:21.000000000 +0900
+++ src/siscone-2.0.6/siscone/spherical/Makefile.am	2014-03-09 11:15:30.559563400 +0900
@@ -10,6 +10,8 @@ lib_LTLIBRARIES = libsiscone_spherical.l
 libsiscone_spherical_la_CXXFLAGS = $(AM_CXXFLAGS) -I$(srcdir)/../.. -I../..
 libsiscone_spherical_la_SOURCES = geom_2d.cpp momentum.cpp hash.cpp\
    vicinity.cpp protocones.cpp split_merge.cpp siscone.cpp
+libsiscone_spherical_la_LDFLAGS = -no-undefined
+libsiscone_spherical_la_LIBADD = ../libsiscone.la
 
 EXTRA_DIST = makefile.static
 
