--- origsrc/AMQP-CPP-4.3.10/CMakeLists.txt	2020-11-23 17:59:55.000000000 +0900
+++ src/AMQP-CPP-4.3.10/CMakeLists.txt	2020-12-08 17:50:07.522018200 +0900
@@ -16,7 +16,7 @@ cmake_minimum_required(VERSION 3.2 FATAL
 project(amqpcpp)
 set (VERSION_MAJOR 4)
 set (VERSION_MINOR 3)
-set (VERSION_PATCH 9)
+set (VERSION_PATCH 10)
 set (SO_VERSION ${VERSION_MAJOR}.${VERSION_MINOR})
 
 # build options
@@ -97,7 +97,7 @@ if(AMQP-CPP_BUILD_SHARED)
     install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME}Config
             ARCHIVE DESTINATION lib
             LIBRARY DESTINATION lib
-            RUNTIME DESTINATION lib
+            RUNTIME DESTINATION bin
     )
 else()
     # copy static lib
@@ -111,7 +111,7 @@ install(DIRECTORY include/amqpcpp/ DESTI
         FILES_MATCHING PATTERN "*.h")
 install(FILES include/amqpcpp.h DESTINATION include)
 
-install(EXPORT ${PROJECT_NAME}Config DESTINATION cmake)
+install(EXPORT ${PROJECT_NAME}Config DESTINATION lib/cmake/amqpcpp)
 export(TARGETS ${PROJECT_NAME} FILE ${PROJECT_NAME}Config.cmake)
 
 set(DEST_DIR "${CMAKE_INSTALL_PREFIX}")
