--- origsrc/AMQP-CPP-4.3.15/CMakeLists.txt	2021-09-24 18:11:57.000000000 +0900
+++ src/AMQP-CPP-4.3.15/CMakeLists.txt	2021-10-24 20:14:16.897106800 +0900
@@ -97,7 +97,7 @@ if(AMQP-CPP_BUILD_SHARED)
     install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME}Config
             ARCHIVE DESTINATION lib
             LIBRARY DESTINATION lib
-            RUNTIME DESTINATION lib
+            RUNTIME DESTINATION bin
     )
 else()
     # copy static lib
@@ -111,7 +111,7 @@ install(DIRECTORY include/amqpcpp/ DESTI
         FILES_MATCHING PATTERN "*.h")
 install(FILES include/amqpcpp.h DESTINATION include)
 
-install(EXPORT ${PROJECT_NAME}Config DESTINATION cmake)
+install(EXPORT ${PROJECT_NAME}Config DESTINATION lib/cmake/amqpcpp)
 export(TARGETS ${PROJECT_NAME} FILE ${PROJECT_NAME}Config.cmake)
 
 set(DEST_DIR "${CMAKE_INSTALL_PREFIX}")
--- origsrc/AMQP-CPP-4.3.15/src/linux_tcp/sslerrorprinter.h	2021-09-24 18:11:57.000000000 +0900
+++ src/AMQP-CPP-4.3.15/src/linux_tcp/sslerrorprinter.h	2021-10-24 20:14:16.904063000 +0900
@@ -20,6 +20,7 @@
 #include "openssl.h"
 #include <string>
 #include <memory>
+#include <string>
 
 /**
  *  Begin namespace
