--- origsrc/AMQP-CPP-4.1.4/CMakeLists.txt	2019-03-08 01:19:48.000000000 +0900
+++ src/AMQP-CPP-4.1.4/CMakeLists.txt	2019-05-28 09:09:07.820286200 +0900
@@ -97,7 +97,7 @@ if(AMQP-CPP_BUILD_SHARED)
     install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME}Config
             ARCHIVE DESTINATION lib
             LIBRARY DESTINATION lib
-            RUNTIME DESTINATION lib
+            RUNTIME DESTINATION bin
     )
 else()
     # copy static lib
@@ -111,7 +111,7 @@ install(DIRECTORY include/amqpcpp/ DESTI
         FILES_MATCHING PATTERN "*.h")
 install(FILES include/amqpcpp.h DESTINATION include)
 
-install(EXPORT ${PROJECT_NAME}Config DESTINATION cmake)
+install(EXPORT ${PROJECT_NAME}Config DESTINATION lib/cmake/amqpcpp)
 export(TARGETS ${PROJECT_NAME} FILE ${PROJECT_NAME}Config.cmake)
 
 set(DEST_DIR "${CMAKE_INSTALL_PREFIX}")
