--- origsrc/libais-0.15/Makefile.am	2015-06-17 03:29:20.000000000 +0900
+++ src/libais-0.15/Makefile.am	2025-01-01 10:42:14.680956800 +0900
@@ -1,3 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4
 
 SUBDIRS = src
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libais-0.15/configure.ac	2015-06-17 03:29:20.000000000 +0900
+++ src/libais-0.15/configure.ac	2025-01-01 10:42:14.696591700 +0900
@@ -41,6 +41,7 @@ LT_INIT
 AC_SUBST([LIBTOOL_DEPS])
 LT_LIB_M
 
+AC_CONFIG_FILES([libais.pc])
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([src/Makefile])
 AC_CONFIG_FILES([src/libais/Makefile])
--- origsrc/libais-0.15/libais.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libais-0.15/libais.pc.in	2025-01-01 10:42:14.696591700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libais
+Description: Library for decoding maritime Automatic Identification System messages
+Version: @VERSION@
+Libs: -L${libdir} -lais
+Cflags: -I${includedir}
--- origsrc/libais-0.15/src/libais/Makefile.am	2015-06-17 03:29:20.000000000 +0900
+++ src/libais-0.15/src/libais/Makefile.am	2025-01-01 10:42:14.696591700 +0900
@@ -2,7 +2,7 @@ lib_LTLIBRARIES = libais.la
 
 include_HEADERS = ais.h
 
-libais_la_LDFLAGS = -version-info 0:0:0
+libais_la_LDFLAGS = -no-undefined -version-info 0:0:0
 
 libais_la_SOURCES = \
 	$(libais_a_HEADERS) \
@@ -36,4 +36,3 @@ libais_la_SOURCES = \
 	ais25.cpp \
 	ais26.cpp \
 	ais27.cpp
-
