--- origsrc/trlan-201009/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/trlan-201009/Makefile.am	2011-11-23 19:07:56.545283100 +0900
@@ -0,0 +1,6 @@
+lib_LTLIBRARIES = libtrlan.la
+libtrlan_la_SOURCES = \
+    SRC/dsort2.f SRC/trl_info.f90 SRC/dstqrb.f SRC/trlcore.f90 SRC/trlan.f90 \
+    SRC/restart.f90 SRC/trlaux.f90 SRC/trlan77.f90 SRC/trl_comm_none.f90
+libtrlan_la_LIBADD = @LAPACK_LIBS@
+libtrlan_la_LDFLAGS = -no-undefined
--- origsrc/trlan-201009/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/trlan-201009/configure.ac	2011-11-23 19:06:33.283854300 +0900
@@ -0,0 +1,30 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([trlan], [201009])
+AC_CONFIG_SRCDIR([examples/psp/dmem.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_LN_S
+AC_PROG_MAKE_SET
+AC_PROG_F77([gfortran])
+AC_PROG_FC([gfortran])
+LT_INIT([disable-static])
+
+# Checks for libraries.
+PKG_CHECK_MODULES([LAPACK], [lapack])
+
+# Checks for header files.
+AC_CHECK_HEADERS([malloc.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
