--- origsrc/naev/Makefile.am	2012-03-31 19:54:31.000000000 +0900
+++ src/naev/Makefile.am	2012-03-31 21:22:05.658135100 +0900
@@ -37,7 +37,7 @@ $(NAEV):
 $(PACK):
 	$(MAKE) -C utils/pack
 
-VERSION: src/config.h
+$(top_srcdir)/VERSION: src/config.h
 	echo $(VERSION) > $@
 
 if HAVE_UTILS
@@ -46,17 +46,27 @@ utils: $(PACK) $(MKSPR)
 $(MKSPR):
 	$(MAKE) -C utils/mkspr
 
-$(DATA_ARCHIVE): $(PACK) VERSION
+$(DATA_ARCHIVE): $(PACK) $(top_srcdir)/VERSION
 	@echo " PACK   $(DATA_ARCHIVE)"
+#	cd "$(top_srcdir)" && \
+#	  (find AUTHORS VERSION naev-confupdate.sh -print0; \
+#	     find ai      -type f -name '*.lua' -print0; \
+#	     find gfx     -type f -name '*.png' -print0; \
+#	     find snd     -type f \
+#	          \( -name '*.ogg' -o -name '*.wav' -o -name '*.lua' \) -print0; \
+#	     find scripts -type f -name '*.lua' -print0; \
+#	     find dat     -type f -print0) | xargs -0 \
+#	$(abs_top_builddir)/$(PACK) "$(abs_top_builddir)/$(DATA_ARCHIVE)";
 	cd "$(top_srcdir)" && \
-	  (find AUTHORS VERSION naev-confupdate.sh -print0; \
-	     find ai      -type f -name '*.lua' -print0; \
-	     find gfx     -type f -name '*.png' -print0; \
+	  (find AUTHORS VERSION naev-confupdate.sh; \
+	     find ai      -type f -name '*.lua'; \
+	     find gfx     -type f -name '*.png'; \
 	     find snd     -type f \
-	          \( -name '*.ogg' -o -name '*.wav' -o -name '*.lua' \) -print0; \
-	     find scripts -type f -name '*.lua' -print0; \
-	     find dat     -type f -print0) | xargs -0 \
-	$(abs_top_builddir)/$(PACK) "$(abs_top_builddir)/$(DATA_ARCHIVE)";
+	          \( -name '*.ogg' -o -name '*.wav' -o -name '*.lua' \); \
+	     find scripts -type f -name '*.lua'; \
+	     find dat     -type f) > $(abs_top_builddir)/ndata.lst && \
+		$(abs_top_builddir)/$(PACK) "$(abs_top_builddir)/$(DATA_ARCHIVE)" \
+		`cat $(abs_top_builddir)/ndata.lst`
 else
 utils:
 
