--- origsrc/secp256k1/Makefile.am	2022-04-29 02:03:31.000000000 +0900
+++ src/secp256k1/Makefile.am	2022-04-29 02:04:54.478177000 +0900
@@ -123,7 +123,7 @@ if !ENABLE_COVERAGE
 tests_CPPFLAGS += -DVERIFY
 endif
 tests_LDADD = $(SECP_LIBS) $(SECP_TEST_LIBS) $(COMMON_LIB) $(PRECOMPUTED_LIB)
-tests_LDFLAGS = -static
+tests_LDFLAGS =
 TESTS += tests
 endif
 
@@ -136,7 +136,7 @@ exhaustive_tests_CPPFLAGS += -DVERIFY
 endif
 # Note: do not include $(PRECOMPUTED_LIB) in exhaustive_tests (it uses runtime-generated tables).
 exhaustive_tests_LDADD = $(SECP_LIBS) $(COMMON_LIB)
-exhaustive_tests_LDFLAGS = -static
+exhaustive_tests_LDFLAGS =
 TESTS += exhaustive_tests
 endif
 
@@ -145,7 +145,7 @@ noinst_PROGRAMS += ecdsa_example
 ecdsa_example_SOURCES = examples/ecdsa.c
 ecdsa_example_CPPFLAGS = -I$(top_srcdir)/include
 ecdsa_example_LDADD = libsecp256k1.la
-ecdsa_example_LDFLAGS = -static
+ecdsa_example_LDFLAGS =
 if BUILD_WINDOWS
 ecdsa_example_LDFLAGS += -lbcrypt
 endif
@@ -155,7 +155,7 @@ noinst_PROGRAMS += ecdh_example
 ecdh_example_SOURCES = examples/ecdh.c
 ecdh_example_CPPFLAGS = -I$(top_srcdir)/include
 ecdh_example_LDADD = libsecp256k1.la
-ecdh_example_LDFLAGS = -static
+ecdh_example_LDFLAGS =
 if BUILD_WINDOWS
 ecdh_example_LDFLAGS += -lbcrypt
 endif
@@ -166,7 +166,7 @@ noinst_PROGRAMS += schnorr_example
 schnorr_example_SOURCES = examples/schnorr.c
 schnorr_example_CPPFLAGS = -I$(top_srcdir)/include
 schnorr_example_LDADD = libsecp256k1.la
-schnorr_example_LDFLAGS = -static
+schnorr_example_LDFLAGS =
 if BUILD_WINDOWS
 schnorr_example_LDFLAGS += -lbcrypt
 endif
