--- origsrc/secp256k1/Makefile.am	2021-03-11 16:56:52.000000000 +0900
+++ src/secp256k1/Makefile.am	2021-03-11 17:15:39.156167000 +0900
@@ -70,6 +70,7 @@ endif
 libsecp256k1_la_SOURCES = src/secp256k1.c
 libsecp256k1_la_CPPFLAGS = -DSECP256K1_BUILD -I$(top_srcdir)/include -I$(top_srcdir)/src $(SECP_INCLUDES)
 libsecp256k1_la_LIBADD = $(SECP_LIBS) $(COMMON_LIB)
+libsecp256k1_la_LDFLAGS = -no-undefined
 
 if VALGRIND_ENABLED
 libsecp256k1_la_CPPFLAGS += -DVALGRIND
@@ -107,7 +108,7 @@ if !ENABLE_COVERAGE
 tests_CPPFLAGS += -DVERIFY
 endif
 tests_LDADD = $(SECP_LIBS) $(SECP_TEST_LIBS) $(COMMON_LIB)
-tests_LDFLAGS = -static
+tests_LDFLAGS =
 TESTS += tests
 endif
 
@@ -119,7 +120,7 @@ if !ENABLE_COVERAGE
 exhaustive_tests_CPPFLAGS += -DVERIFY
 endif
 exhaustive_tests_LDADD = $(SECP_LIBS) $(COMMON_LIB)
-exhaustive_tests_LDFLAGS = -static
+exhaustive_tests_LDFLAGS =
 TESTS += exhaustive_tests
 endif
 
--- origsrc/secp256k1/src/modules/ecdh/main_impl.h	2021-03-11 16:56:52.000000000 +0900
+++ src/secp256k1/src/modules/ecdh/main_impl.h	2021-03-11 17:14:30.436276800 +0900
@@ -8,7 +8,6 @@
 #define SECP256K1_MODULE_ECDH_MAIN_H
 
 #include "include/secp256k1_ecdh.h"
-#include "ecmult_const_impl.h"
 
 static int ecdh_hash_function_sha256(unsigned char *output, const unsigned char *x32, const unsigned char *y32, void *data) {
     unsigned char version = (y32[31] & 0x01) | 0x02;
