--- origsrc/mtbl-tags-v1.2.1/Makefile.am	2018-05-26 07:49:08.000000000 +0900
+++ src/mtbl-tags-v1.2.1/Makefile.am	2018-06-18 04:39:12.609948400 +0900
@@ -57,6 +57,7 @@ mtbl_libmtbl_la_SOURCES = \
 
 mtbl_libmtbl_la_LIBADD = -lsnappy -lz $(liblz4_LIBS) $(libzstd_LIBS)
 mtbl_libmtbl_la_LDFLAGS = $(AM_LDFLAGS) \
+	-no-undefined \
 	-version-info $(LIBMTBL_VERSION_INFO)
 if HAVE_LD_VERSION_SCRIPT
 mtbl_libmtbl_la_LDFLAGS += \
