--- origsrc/mtbl-tags-v1.1.1/Makefile.am	2017-08-03 04:06:41.000000000 +0900
+++ src/mtbl-tags-v1.1.1/Makefile.am	2017-09-28 19:14:29.386000000 +0900
@@ -57,6 +57,7 @@ mtbl_libmtbl_la_SOURCES = \
 
 mtbl_libmtbl_la_LIBADD = -lsnappy -lz $(liblz4_LIBS) $(libzstd_LIBS)
 mtbl_libmtbl_la_LDFLAGS = $(AM_LDFLAGS) \
+	-no-undefined \
 	-version-info $(LIBMTBL_VERSION_INFO)
 if HAVE_LD_VERSION_SCRIPT
 mtbl_libmtbl_la_LDFLAGS += \
