--- origsrc/mp3cat-0.5/Makefile	2014-02-26 06:03:01.000000000 +0900
+++ src/mp3cat-0.5/Makefile	2019-06-07 20:52:12.465505000 +0900
@@ -4,7 +4,7 @@ INSTALL_GROUP?=$(shell id -g)
 INSTALL_COMMAND?=install -o $(INSTALL_USER) -g $(INSTALL_GROUP) -m 0555
 all: mp3cat
 mp3cat: *.c *.h
-	cc -o mp3cat *.c
+	cc -o mp3cat *.c $(CFLAGS)
 install: install_bin install_scripts
 install_bin: mp3cat
 	mkdir -p $(PREFIX)/bin
