--- origsrc/mp3cat-0.5/Makefile	2014-02-26 06:03:01.000000000 +0900
+++ src/mp3cat-0.5/Makefile	2024-11-08 09:36:47.742863500 +0900
@@ -1,10 +1,11 @@
+CC?=gcc
 PREFIX?=/usr/local
 INSTALL_USER?=$(shell id -u)
 INSTALL_GROUP?=$(shell id -g)
 INSTALL_COMMAND?=install -o $(INSTALL_USER) -g $(INSTALL_GROUP) -m 0555
 all: mp3cat
 mp3cat: *.c *.h
-	cc -o mp3cat *.c
+	$(CC) -o mp3cat *.c $(CPPFLAGS) $(CFLAGS) $(LDFLAGS)
 install: install_bin install_scripts
 install_bin: mp3cat
 	mkdir -p $(PREFIX)/bin
