--- origsrc/mp3cat-0.4/Makefile	2003-05-05 07:02:23.000000000 +0900
+++ src/mp3cat-0.4/Makefile	2015-05-06 11:32:08.702763300 +0900
@@ -1,6 +1,6 @@
 all: mp3cat
 mp3cat: *.c *.h
-	cc -o mp3cat *.c
+	cc -o mp3cat *.c -O3
 install: mp3cat mp3log mp3log-conf mp3stream-conf
 	install -o root -g wheel mp3cat /usr/local/bin/mp3cat
 	install -o root -g wheel mp3log /usr/local/bin/mp3log
--- origsrc/mp3cat-0.4/write_buf.c	2003-05-05 07:02:23.000000000 +0900
+++ src/mp3cat-0.4/write_buf.c	2015-05-06 11:32:42.797830600 +0900
@@ -9,6 +9,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <errno.h>
+#include <string.h>
 
 #ifndef O_NOFOLLOW
 #define O_NOFOLLOW 0
