--- origsrc/grok/src/bin/jp2/CMakeLists.txt	2021-12-28 17:17:38.000000000 +0900
+++ src/grok/src/bin/jp2/CMakeLists.txt	2021-12-28 17:18:18.644003300 +0900
@@ -87,8 +87,8 @@ add_definitions(-DSPDLOG_COMPILED_LIB)
 foreach(exe grk_decompress grk_compress grk_dump)
   add_executable(${exe} ${exe}.cpp ${common_SRCS})
   target_compile_options(${exe} PRIVATE ${GROK_COMPILE_OPTIONS})
-  if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
-	target_link_options(${exe}  PUBLIC "LINKER:-z,now")
+  if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND NOT CYGWIN)
+  	target_link_options(${exe}  PUBLIC "LINKER:-z,now")
   endif()
 
   target_link_libraries(${exe} 
--- origsrc/grok/src/lib/jp2/CMakeLists.txt	2021-12-28 17:17:38.000000000 +0900
+++ src/grok/src/lib/jp2/CMakeLists.txt	2021-12-28 17:18:18.649014800 +0900
@@ -257,7 +257,7 @@ else()
 	  set(INSTALL_LIBS ${GROK_LIBRARY_NAME})
   endif()
 endif()
-if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND NOT CYGWIN)
 	target_link_options(${GROK_LIBRARY_NAME} PUBLIC "LINKER:-z,now")
 endif() 
 add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/highway EXCLUDE_FROM_ALL)
