--- origsrc/grok-7.3.0/src/bin/jp2/CMakeLists.txt	2020-06-06 08:54:25.000000000 +0900
+++ src/grok-7.3.0/src/bin/jp2/CMakeLists.txt	2020-07-07 15:14:00.951040000 +0900
@@ -82,8 +82,8 @@ add_definitions(-DSPDLOG_COMPILED_LIB)
 foreach(exe grk_decompress grk_compress grk_dump)
   add_executable(${exe} ${exe}.cpp ${common_SRCS})
   target_compile_options(${exe} PRIVATE ${GROK_COMPILE_OPTIONS})
-  if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
-	target_link_options(${exe}  PUBLIC "LINKER:-z,now")
+  if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND NOT CYGWIN)
+	  target_link_options(${exe}  PUBLIC "LINKER:-z,now")
   endif()  
   target_link_libraries(${exe} ${GROK_LIBRARY_NAME}
     ${PNG_LIBNAME} ${TIFF_LIBNAME} ${LCMS_LIBNAME}  
--- origsrc/grok-7.3.0/src/lib/jp2/CMakeLists.txt	2020-06-06 08:54:25.000000000 +0900
+++ src/grok-7.3.0/src/lib/jp2/CMakeLists.txt	2020-07-07 15:07:53.385335900 +0900
@@ -228,7 +228,7 @@ else()
 	  set(INSTALL_LIBS ${GROK_LIBRARY_NAME})
   endif()
 endif()
-if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND NOT CYGWIN)
 	target_link_options(${GROK_LIBRARY_NAME} PUBLIC "LINKER:-z,now")
 endif() 
 if(UNIX)
