--- origsrc/libMXF-1.1/mxf/CMakeLists.txt	2023-01-26 21:39:54.000000000 +0900
+++ src/libMXF-1.1/mxf/CMakeLists.txt	2025-02-11 00:33:06.947987700 +0900
@@ -122,8 +122,8 @@ include("${PROJECT_SOURCE_DIR}/cmake/sou
 set_source_filename(MXF "${CMAKE_CURRENT_LIST_DIR}" "libMXF")
 
 install(TARGETS MXF
+    RUNTIME DESTINATION bin
     ARCHIVE DESTINATION lib
-    LIBRARY DESTINATION lib
 )
 
 install(FILES ${MXF_headers} DESTINATION include/mxf)
--- origsrc/libMXF-1.1/tools/MXFDump/MXFDump.cpp	2023-01-26 21:39:54.000000000 +0900
+++ src/libMXF-1.1/tools/MXFDump/MXFDump.cpp	2025-02-11 00:36:12.533388800 +0900
@@ -90,7 +90,7 @@
 #elif defined(__GNUC__) && defined(__i386__) && defined(__FreeBSD__)
 #define MXF_COMPILER_GCC_INTEL_FREEBSD
 #define MXF_OS_UNIX
-#elif defined(__GNUC__) && defined(__i386__) && defined(__CYGWIN__)
+#elif defined(__GNUC__) && defined(__CYGWIN__)
 #define MXF_COMPILER_GCC_INTEL_CYGWIN
 #define MXF_OS_UNIX
 #elif defined(__GNUC__) && defined(__sparc__) && defined(__sun__)
