--- origsrc/libdftw-0.0.2/Makefile.am	2013-01-12 06:23:05.000000000 +0900
+++ src/libdftw-0.0.2/Makefile.am	2015-03-07 16:43:52.778837900 +0900
@@ -7,7 +7,7 @@ ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
 lib_LTLIBRARIES = libdftw.la
 
-libdftw_la_LDFLAGS = -version-info $(LIBDFTW_SO_VERSION)
+libdftw_la_LDFLAGS = -no-undefined -version-info $(LIBDFTW_SO_VERSION)
 libdftw_la_SOURCES = libdftw/dftw.c
 libdftw_la_LIBADD = $(libcircle_LIBS)
 
--- origsrc/libdftw-0.0.2/libdftw/dftw.c	2013-01-12 06:23:05.000000000 +0900
+++ src/libdftw-0.0.2/libdftw/dftw.c	2015-03-07 16:46:45.821806600 +0900
@@ -1,3 +1,5 @@
+
+#include <stdio.h>
 #include <libcircle.h>
 
 #include <dirent.h>
--- origsrc/libdftw-0.0.2/tests/Makefile.am	2013-01-12 06:23:05.000000000 +0900
+++ src/libdftw-0.0.2/tests/Makefile.am	2015-03-07 16:47:05.937041000 +0900
@@ -3,7 +3,7 @@ include $(top_srcdir)/common.mk
 noinst_PROGRAMS = dftw_test
 
 dftw_test_SOURCES = dftw_test.c
-dftw_test_LDADD = $(top_srcdir)/libdftw.la
+dftw_test_LDADD = $(top_builddir)/libdftw.la
 dftw_test_CPPFLAGS = \
     $(libdftw_CFLAGS) \
     -I$(top_srcdir)/
