--- origsrc/sha1collisiondetection-stable-v1.0.3/Makefile	2017-03-28 01:11:23.000000000 +0900
+++ src/sha1collisiondetection-stable-v1.0.3/Makefile	2025-05-05 11:06:44.584846400 +0900
@@ -34,7 +34,7 @@ endif
 
 
 CFLAGS=-O2 -Wall -Werror -Wextra -pedantic -std=c90 -Ilib
-LDFLAGS=
+LDFLAGS=-no-undefined
 
 LT_CC:=$(LIBTOOL) --tag=CC --mode=compile $(CC)
 LT_CC_DEP:=$(CC)
@@ -83,11 +83,11 @@ all: library tools
 
 .PHONY: install
 install: all
-	$(LT_INSTALL) -d $(LIBDIR) $(BINDIR) $(INCLUDEDIR)
-	$(LT_INSTALL) bin/libsha1detectcoll.$(LIB_EXT) $(LIBDIR)/libsha1detectcoll.$(LIB_EXT)
-	$(LT_INSTALL) lib/sha1.h $(INCLUDEDIR)/sha1.h
-	$(LT_INSTALL) bin/sha1dcsum $(BINDIR)/sha1dcsum
-	$(LT_INSTALL) bin/sha1dcsum_partialcoll $(BINDIR)/sha1dcsum_partialcoll
+	$(LT_INSTALL) -d $(DESTDIR)$(LIBDIR) $(DESTDIR)$(BINDIR) $(DESTDIR)$(INCLUDEDIR)
+	$(LT_INSTALL) bin/libsha1detectcoll.$(LIB_EXT) $(DESTDIR)$(LIBDIR)/libsha1detectcoll.$(LIB_EXT)
+	$(LT_INSTALL) lib/sha1.h $(DESTDIR)$(INCLUDEDIR)/sha1.h
+	$(LT_INSTALL) bin/sha1dcsum $(DESTDIR)$(BINDIR)/sha1dcsum
+	$(LT_INSTALL) bin/sha1dcsum_partialcoll $(DESTDIR)$(BINDIR)/sha1dcsum_partialcoll
 
 .PHONY: uninstall
 uninstall:
--- origsrc/sha1collisiondetection-stable-v1.0.3/sha1collisiondetection.pc	1970-01-01 09:00:00.000000000 +0900
+++ src/sha1collisiondetection-stable-v1.0.3/sha1collisiondetection.pc	2025-05-05 11:36:19.312126500 +0900
@@ -0,0 +1,10 @@
+prefix=/usr
+exec_prefix=/usr
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: sha1collisiondetection
+Description: Library to detect SHA-1 collisions in files
+Version: 1.0.3
+Libs: -L${libdir} -lsha1detectcoll
+Cflags: -I${includedir}
