--- origsrc/libmpq-libmpq-0.4.2/libmpq/Makefile.am	2008-05-16 23:27:06.000000000 +0900
+++ src/libmpq-libmpq-0.4.2/libmpq/Makefile.am	2021-03-09 17:32:27.607367400 +0900
@@ -12,7 +12,7 @@ libmpq_includedir		= $(includedir)/libmp
 libmpq_include_HEADERS		= mpq.h
 
 libmpq_la_SOURCES		= $(GENERAL_SRCS)
-libmpq_la_LDFLAGS		= -release $(PACKAGE_VERSION)
+libmpq_la_LDFLAGS		= -no-undefined -release $(PACKAGE_VERSION)
 
 GENERAL_SRCS =			\
 	common.c		\
--- origsrc/libmpq-libmpq-0.4.2/libmpq.pc.in	2008-05-16 23:27:06.000000000 +0900
+++ src/libmpq-libmpq-0.4.2/libmpq.pc.in	2021-03-09 17:40:47.844193600 +0900
@@ -6,5 +6,5 @@ includedir=@includedir@
 Name: libmpq
 Description: GPL version of the libmpq library
 Version: @VERSION@
-Libs: -L${libdir} -lmpq -lz
+Libs: -L${libdir} -lmpq -lbz2 -lz
 Cflags: -I${includedir}/libmpq @LFS_CFLAGS@
