--- origsrc/libconfig-1.4.7/configure.ac	2011-03-11 17:20:03.000000000 +0900
+++ src/libconfig-1.4.7/configure.ac	2011-05-14 14:07:38.560202100 +0900
@@ -113,7 +113,7 @@ AM_CONDITIONAL(BUILDEXAMPLES, test x$doe
 dnl Check for MinGW. Workaround for libtool's DLL_EXPORT stupidity.
 
 case "$target" in
-	*-*-cygwin* | *-*-mingw*)
+	*-*-mingw*)
 		gnuwin=yes;;
 esac
 
--- origsrc/libconfig-1.4.7/tests/Makefile.am	2010-10-29 14:47:09.000000000 +0900
+++ src/libconfig-1.4.7/tests/Makefile.am	2011-05-14 14:21:46.959630900 +0900
@@ -3,7 +3,7 @@ bin_PROGRAMS = libconfig_tests
 
 libconfig_tests_SOURCES = tests.c
 
-libconfig_tests_CPPFLAGS = -I../tinytest -I../lib
+libconfig_tests_CPPFLAGS = -I$(top_srcdir)/tinytest -I../lib
 
 libconfig_tests_LDADD = -L../tinytest -ltinytest \
 	-L../lib/.libs -lconfig
