--- origsrc/libconfig-1.4.8/configure.ac	2011-08-04 12:37:07.000000000 +0900
+++ src/libconfig-1.4.8/configure.ac	2012-04-26 07:51:36.100478300 +0900
@@ -112,7 +112,7 @@ AM_CONDITIONAL(BUILDEXAMPLES, test x$doe
 dnl Check for MinGW. Workaround for libtool's DLL_EXPORT stupidity.
 
 case "$target" in
-	*-*-cygwin* | *-*-mingw*)
+	*-*-mingw*)
 		gnuwin=yes;;
 esac
 
--- origsrc/libconfig-1.4.8/tests/Makefile.am	2011-08-04 12:36:33.000000000 +0900
+++ src/libconfig-1.4.8/tests/Makefile.am	2012-04-26 07:51:36.100478300 +0900
@@ -3,7 +3,7 @@ noinst_PROGRAMS = libconfig_tests
 
 libconfig_tests_SOURCES = tests.c
 
-libconfig_tests_CPPFLAGS = -I../tinytest -I../lib
+libconfig_tests_CPPFLAGS = -I$(top_srcdir)/tinytest -I../lib
 
 libconfig_tests_LDADD = -L../tinytest -ltinytest \
 	-L../lib/.libs -lconfig
