--- origsrc/pdjson/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/pdjson/Makefile.am	2024-09-11 13:09:39.329350600 +0900
@@ -0,0 +1,18 @@
+check_PROGRAMS = pdjson-pretty pdjson-tests pdjson-stream
+pdjson_pretty_LDADD = libpdjson.la
+pdjson_pretty_SOURCES = tests/pretty.c
+pdjson_tests_LDADD = libpdjson.la
+pdjson_tests_SOURCES = tests/tests.c
+pdjson_stream_LDADD = libpdjson.la
+pdjson_stream_SOURCES = tests/stream.c
+
+include_HEADERS = pdjson.h
+
+lib_LTLIBRARIES = libpdjson.la
+libpdjson_la_SOURCES = pdjson.c
+libpdjson_la_LDFLAGS = -no-undefined
+
+TESTS = pdjson-tests
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/pdjson/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/pdjson/configure.ac	2024-09-11 13:09:39.329350600 +0900
@@ -0,0 +1,30 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.71])
+AC_INIT([pdjson], [0])
+AC_CONFIG_SRCDIR([pdjson.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_FUNC_STRTOD
+AC_CHECK_FUNCS([strchr strerror])
+
+AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([pdjson.pc])
+AC_OUTPUT
--- origsrc/pdjson/pdjson.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/pdjson/pdjson.pc.in	2024-09-11 13:09:39.329350600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: pdjson
+Description: Public Domain JSON Parser for C
+Version: @VERSION@
+Libs: -L${libdir} -lpdjson
+Cflags: -I${includedir}
