--- origsrc/dbus-cxx-0.12.0/CMakeLists.txt	2020-08-05 10:01:16.000000000 +0900
+++ src/dbus-cxx-0.12.0/CMakeLists.txt	2020-11-09 11:48:53.992286000 +0900
@@ -152,7 +152,7 @@ include_directories( ${DBUS_CXX_INCLUDE_
 add_library( dbus-cxx SHARED ${DBUS_CXX_SOURCES} )
 #add_library( dbus-cxx-static STATIC ${DBUS_CXX_SOURCES} )
 set_target_properties( dbus-cxx PROPERTIES VERSION 1.0.0 SOVERSION 1 )
-target_link_libraries( dbus-cxx ${dbus_LIBS} ${sigc_LIBS} -lrt -pthread )
+target_link_libraries( dbus-cxx ${dbus_LIBRARIES} ${sigc_LIBRARIES} -lrt -pthread )
 target_compile_options( dbus-cxx PUBLIC -pthread )
 
 #
@@ -165,10 +165,11 @@ if( ENABLE_GLIBMM )
     set_target_properties( dbus-cxx-glibmm PROPERTIES VERSION 1.0.0 SOVERSION 1 )
     target_include_directories( dbus-cxx-glibmm PRIVATE ${glibmm-2.4_INCLUDE_DIRS} )
     add_dependencies( dbus-cxx-glibmm dbus-cxx )
+    target_link_libraries( dbus-cxx-glibmm dbus-cxx ${glibmm-2.4_LIBRARIES} )
 
     set( install_targets ${install_targets} dbus-cxx-glibmm )
     install( TARGETS ${install_targets}
-        LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
         ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" )
     install( FILES dbus-cxx-glibmm.h DESTINATION include/dbus-cxx-${DBUS_CXX_MAJMIN_VERSION} )
     install( FILES dbus-cxx-glibmm/dispatcher.h DESTINATION include/dbus-cxx-${DBUS_CXX_MAJMIN_VERSION}/dbus-cxx-glibmm/ )
@@ -222,7 +223,7 @@ endforeach( filename )
 #
 set( install_targets ${install_targets} dbus-cxx )
 install( TARGETS ${install_targets}
-    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
     ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" )
 install( FILES ${DBUS_CXX_HEADERS} DESTINATION include/dbus-cxx-${DBUS_CXX_MAJMIN_VERSION}/dbus-cxx )
 install( FILES ${DBUS_CXX_HEADERS1} DESTINATION include/dbus-cxx-${DBUS_CXX_MAJMIN_VERSION} )
--- origsrc/dbus-cxx-0.12.0/tools/CMakeLists.txt	2020-08-05 10:01:16.000000000 +0900
+++ src/dbus-cxx-0.12.0/tools/CMakeLists.txt	2020-11-09 11:41:02.557934500 +0900
@@ -9,7 +9,7 @@ if( TOOLS_BUNDLED_CPPGENERATE )
     set( cppgenerate_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/libcppgenerate )
     set( cppgenerate_LDFLAGS ${BINARY_DIR}/libcppgenerate.a )
 else()
-    pkg_check_modules( cppgenerate REQUIRED cppgenerate-static )
+    pkg_check_modules( cppgenerate REQUIRED cppgenerate )
 endif( TOOLS_BUNDLED_CPPGENERATE )
 
 pkg_check_modules( popt REQUIRED popt )
