--- origsrc/dbus-cxx-0.11.0/CMakeLists.txt	2019-06-07 09:38:51.000000000 +0900
+++ src/dbus-cxx-0.11.0/CMakeLists.txt	2019-10-13 06:48:59.470598700 +0900
@@ -148,7 +148,7 @@ include_directories( ${DBUS_CXX_INCLUDE_
 add_library( dbus-cxx SHARED ${DBUS_CXX_SOURCES} )
 #add_library( dbus-cxx-static STATIC ${DBUS_CXX_SOURCES} )
 set_target_properties( dbus-cxx PROPERTIES VERSION 1.0.0 SOVERSION 1 )
-target_link_libraries( dbus-cxx ${dbus_LIBS} ${sigc_LIBS} -lrt )
+target_link_libraries( dbus-cxx ${dbus_LIBRARIES} ${sigc_LIBRARIES} -lrt )
 
 #
 # If GLIBMM is used, add in the appropriate packages
@@ -160,10 +160,11 @@ if( ENABLE_GLIBMM )
     set_target_properties( dbus-cxx-glibmm PROPERTIES VERSION 1.0.0 SOVERSION 1 )
     target_include_directories( dbus-cxx-glibmm PRIVATE ${glibmm-2.4_INCLUDE_DIRS} )
     add_dependencies( dbus-cxx-glibmm dbus-cxx )
+    target_link_libraries( dbus-cxx-glibmm dbus-cxx ${glibmm-2.4_LIBRARIES} )
 
     set( install_targets ${install_targets} dbus-cxx-glibmm )
     install( TARGETS ${install_targets}
-        LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
         ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" )
     install( FILES dbus-cxx-glibmm.h DESTINATION include/dbus-cxx-${DBUS_CXX_MAJMIN_VERSION} )
     install( FILES dbus-cxx-glibmm/dispatcher.h DESTINATION include/dbus-cxx-${DBUS_CXX_MAJMIN_VERSION}/dbus-cxx-glibmm/ )
@@ -217,7 +218,7 @@ endforeach( filename )
 #
 set( install_targets ${install_targets} dbus-cxx )
 install( TARGETS ${install_targets}
-    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
     ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" )
 install( FILES ${DBUS_CXX_HEADERS} DESTINATION include/dbus-cxx-${DBUS_CXX_MAJMIN_VERSION}/dbus-cxx )
 install( FILES ${DBUS_CXX_HEADERS1} DESTINATION include/dbus-cxx-${DBUS_CXX_MAJMIN_VERSION} )
--- origsrc/dbus-cxx-0.11.0/tools/CMakeLists.txt	2019-06-07 09:38:51.000000000 +0900
+++ src/dbus-cxx-0.11.0/tools/CMakeLists.txt	2019-10-13 06:24:53.378071100 +0900
@@ -9,7 +9,7 @@ if( TOOLS_BUNDLED_CPPGENERATE )
     set( cppgenerate_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/libcppgenerate )
     set( cppgenerate_LDFLAGS ${BINARY_DIR}/libcppgenerate.a )
 else()
-    pkg_check_modules( cppgenerate REQUIRED cppgenerate-static )
+    pkg_check_modules( cppgenerate REQUIRED cppgenerate )
 endif( TOOLS_BUNDLED_CPPGENERATE )
 
 pkg_check_modules( popt REQUIRED popt )
