--- origsrc/dbus-cxx/dbus-cxx/Makefile.am	2017-05-30 10:49:27.000000000 +0900
+++ src/dbus-cxx/dbus-cxx/Makefile.am	2018-02-28 14:30:38.681678600 +0900
@@ -56,7 +56,7 @@ libdbus_cxx_la_SOURCES = callmessage.cpp
 	utility.cpp watch.cpp
 
 
-libdbus_cxx_la_LDFLAGS = -version-info @LIBTOOL_SO_VERSION@
+libdbus_cxx_la_LDFLAGS = -no-undefined -version-info @LIBTOOL_SO_VERSION@
 
 libdbus_cxx_la_LIBADD = $(PROJECT_LIBS)
 
--- origsrc/dbus-cxx/dbus-cxx/signature.h	2017-05-14 11:01:35.000000000 +0900
+++ src/dbus-cxx/dbus-cxx/signature.h	2018-02-28 15:09:19.287000000 +0900
@@ -92,7 +92,7 @@ namespace DBus
   };
 
   inline std::string signature( uint8_t& )     { return DBUS_TYPE_BYTE_AS_STRING; }
-  inline std::string signature( bool& )        { return DBUS_TYPE_BOOLEAN_AS_STRING; }
+  inline std::string signature( const bool& )        { return DBUS_TYPE_BOOLEAN_AS_STRING; }
   inline std::string signature( int16_t& )     { return DBUS_TYPE_INT16_AS_STRING; }
   inline std::string signature( uint16_t& )    { return DBUS_TYPE_UINT16_AS_STRING; }
   inline std::string signature( int32_t& )     { return DBUS_TYPE_INT32_AS_STRING; }
--- origsrc/dbus-cxx/dbus-cxx/timeout.cpp	2017-01-06 11:33:58.000000000 +0900
+++ src/dbus-cxx/dbus-cxx/timeout.cpp	2018-02-28 14:19:49.070000000 +0900
@@ -106,7 +106,7 @@ namespace DBus
       nsec = (intv % 1000) * 1000000;
       struct itimerspec its = { {sec, nsec}, {sec, nsec} };
 
-      timer_settime( &m_timer_id, 0, &its, NULL );
+      timer_settime( m_timer_id, 0, &its, NULL );
 
       m_is_armed = true;
     }
--- origsrc/dbus-cxx/dbus-cxx-glibmm/Makefile.am	2017-05-30 10:49:27.000000000 +0900
+++ src/dbus-cxx/dbus-cxx-glibmm/Makefile.am	2018-02-28 14:31:58.365000000 +0900
@@ -28,7 +28,7 @@ library_includedir = $(includedir)/dbus-
 libdbus_cxx_glibmm_la_LIBADD = $(top_builddir)/dbus-cxx/libdbus-cxx.la \
 	$(DBUS_CXX_GLIBMM_LIBS) $(PROJECT_LIBS)
 
-libdbus_cxx_glibmm_la_LDFLAGS = -version-info @LIBTOOL_SO_VERSION@
+libdbus_cxx_glibmm_la_LDFLAGS = -no-undefined -version-info @LIBTOOL_SO_VERSION@
 library_include_HEADERS = dispatcher.h
 libdbus_cxx_glibmm_la_SOURCES = dispatcher.cpp
 
