--- origsrc/dbus-cxx-0.7.0/dbus-cxx/Makefile.am	2010-03-11 06:18:50.000000000 +0900
+++ src/dbus-cxx-0.7.0/dbus-cxx/Makefile.am	2015-04-05 00:30:50.507325900 +0900
@@ -55,7 +55,7 @@ libdbus_cxx_la_SOURCES = callmessage.cpp
 	utility.cpp watch.cpp
 
 
-libdbus_cxx_la_LDFLAGS = -version-info @LIBTOOL_SO_VERSION@
+libdbus_cxx_la_LDFLAGS = -no-undefined -version-info @LIBTOOL_SO_VERSION@
 
 libdbus_cxx_la_LIBADD = $(PROJECT_LIBS)
 
--- origsrc/dbus-cxx-0.7.0/dbus-cxx/pointer.h	2009-09-29 07:00:12.000000000 +0900
+++ src/dbus-cxx-0.7.0/dbus-cxx/pointer.h	2015-04-05 00:30:50.507325900 +0900
@@ -23,7 +23,7 @@
   // Headers for smart pointers
   #ifdef DBUS_CXX_USE_BOOST_SMART_POINTER
     #include <boost/shared_ptr.hpp>
-    //#include <boost/weak_ptr.hpp>
+    #include <boost/weak_ptr.hpp>
     #define DBusCxxPointer boost::shared_ptr
     #define DBusCxxWeakPointer boost::weak_ptr
     #define dbus_cxx_static_pointer_cast  boost::static_pointer_cast
--- origsrc/dbus-cxx-0.7.0/dbus-cxx/timeout.cpp	2010-01-23 01:31:29.000000000 +0900
+++ src/dbus-cxx-0.7.0/dbus-cxx/timeout.cpp	2015-04-05 00:30:50.507325900 +0900
@@ -105,7 +105,7 @@ namespace DBus
       nsec = (intv % 1000) * 1000000;
       struct itimerspec its = { {sec, nsec}, {sec, nsec} };
 
-      timer_settime( &m_timer_id, 0, &its, NULL );
+      timer_settime( m_timer_id, 0, &its, NULL );
 
       m_is_armed = true;
     }
--- origsrc/dbus-cxx-0.7.0/dbus-cxx-glibmm/Makefile.am	2009-08-24 23:09:23.000000000 +0900
+++ src/dbus-cxx-0.7.0/dbus-cxx-glibmm/Makefile.am	2015-04-05 00:30:50.507325900 +0900
@@ -28,7 +28,7 @@ library_includedir = $(includedir)/dbus-
 libdbus_cxx_glibmm_la_LIBADD = $(top_builddir)/dbus-cxx/libdbus-cxx.la \
 	$(DBUS_CXX_GLIBMM_LIBS) $(PROJECT_LIBS)
 
-libdbus_cxx_glibmm_la_LDFLAGS = -version-info @LIBTOOL_SO_VERSION@
+libdbus_cxx_glibmm_la_LDFLAGS = -no-undefined -version-info @LIBTOOL_SO_VERSION@
 library_include_HEADERS = dispatcher.h
 libdbus_cxx_glibmm_la_SOURCES = dispatcher.cpp
 
--- origsrc/dbus-cxx-0.7.0/unit-tests/Makefile.am	2009-06-02 00:58:58.000000000 +0900
+++ src/dbus-cxx-0.7.0/unit-tests/Makefile.am	2015-04-05 00:43:01.108958200 +0900
@@ -22,11 +22,12 @@ METASOURCES = AUTO
 
 if DBUS_CXX_UNIT_TESTS
 
-noinst_PROGRAMS = dbus-cxx-unit-tests
+check_PROGRAMS = dbus-cxx-unit-tests
 dbus_cxx_unit_tests_LDADD = $(top_builddir)/dbus-cxx/libdbus-cxx.la \
 	$(PROJECT_LIBS) $(UNIT_TEST_LIBS)
 dbus_cxx_unit_tests_SOURCES = callmessagetests.cpp dbus_cxx_unit_tests.cpp \
 	messageiteratortests.cpp pathclasstests.cpp
 noinst_HEADERS = callmessagetests.h messageiteratortests.h pathclasstests.h
+TESTS = $(check_PROGRAMS)
 
 endif
