--- origsrc/mrtdreader-0.1.6/src/CMakeLists.txt	2016-02-10 04:11:56.000000000 +0900
+++ src/mrtdreader-0.1.6/src/CMakeLists.txt	2025-02-21 14:19:00.977410700 +0900
@@ -10,9 +10,9 @@ set_target_properties(mrtd PROPERTIES VE
 
 install(TARGETS mrtdreader DESTINATION bin)
 if(${CMAKE_INSTALL_PREFIX} STREQUAL "/usr")
- install(TARGETS mrtd DESTINATION lib/${CMAKE_LIBRARY_ARCHITECTURE})
+ install(TARGETS mrtd RUNTIME DESTINATION bin ARCHIVE DESTINATION lib/${CMAKE_LIBRARY_ARCHITECTURE})
 else()
- install(TARGETS mrtd DESTINATION lib)
+ install(TARGETS mrtd RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 endif()
 
 install(FILES mrtd.h DESTINATION include/mrtd)
