--- origsrc/gearmand-0.20/configure.ac	2011-04-08 11:36:49.000000000 +0900
+++ src/gearmand-0.20/configure.ac	2011-05-16 00:16:13.607393100 +0900
@@ -53,7 +53,7 @@ AC_PROG_CPP
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
-AC_PROG_RANLIB
+LT_INIT
 
 # Checks for header files.
 AC_HEADER_STDC
--- origsrc/gearmand-0.20/libgearman/include.am	2011-04-09 00:42:14.000000000 +0900
+++ src/gearmand-0.20/libgearman/include.am	2011-05-16 00:17:57.915454400 +0900
@@ -77,7 +77,7 @@ libgearman_libgearman_la_CXXFLAGS= \
 libgearman_libgearman_la_LDFLAGS= \
 				  -fno-exceptions \
 				  -fno-rtti \
-				  $(AM_LDFLAGS) \
+				  $(AM_LDFLAGS) -no-undefined \
 				  -version-info $(GEARMAN_LIBRARY_VERSION)
 
 libgearman_libgearman_la_LIBADD= \
