--- origsrc/libnsfdb-20170128/Makefile.am	2017-01-28 19:12:42.000000000 +0900
+++ src/libnsfdb-20170128/Makefile.am	2019-05-21 07:41:13.650305000 +0900
@@ -3,23 +3,8 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
-	libfvalue \
 	libnsfdb \
 	nsfdbtools \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libnsfdb-20170128/configure.ac	2017-01-28 19:12:42.000000000 +0900
+++ src/libnsfdb-20170128/configure.ac	2019-05-21 07:38:23.000709000 +0900
@@ -148,20 +148,6 @@ dnl Have configure make the Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
 AC_CONFIG_FILES([libnsfdb/Makefile])
 AC_CONFIG_FILES([nsfdbtools/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
--- origsrc/libnsfdb-20170128/libnsfdb/Makefile.am	2017-01-28 19:12:42.000000000 +0900
+++ src/libnsfdb-20170128/libnsfdb/Makefile.am	2019-05-21 07:41:33.841499200 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libnsfdb-20170128/libnsfdb/libnsfdb_bucket_list.c	2017-01-28 19:12:42.000000000 +0900
+++ src/libnsfdb-20170128/libnsfdb/libnsfdb_bucket_list.c	2019-05-21 08:03:56.910826700 +0900
@@ -21,6 +21,7 @@
 
 #include <common.h>
 #include <types.h>
+#include <libfcache.h>
 
 #include "libnsfdb_bucket.h"
 #include "libnsfdb_libbfio.h"
--- origsrc/libnsfdb-20170128/nsfdbtools/Makefile.am	2017-01-28 19:12:42.000000000 +0900
+++ src/libnsfdb-20170128/nsfdbtools/Makefile.am	2019-05-21 07:41:45.199192900 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libnsfdb-20170128/tests/Makefile.am	2017-01-28 19:12:42.000000000 +0900
+++ src/libnsfdb-20170128/tests/Makefile.am	2019-05-21 07:41:55.201393000 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
