--- origsrc/libnsfdb-20160324/Makefile.am	2016-03-24 13:51:40.000000000 +0900
+++ src/libnsfdb-20160324/Makefile.am	2016-04-07 20:01:02.079487000 +0900
@@ -3,25 +3,8 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
-	libfvalue \
 	libnsfdb \
-	libcsystem \
 	nsfdbtools \
-	po \
 	manuals \
 	tests \
 	msvscpp
@@ -122,4 +105,3 @@ splint:
 	(cd $(srcdir)/nsfdbtools && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
-
--- origsrc/libnsfdb-20160324/configure.ac	2016-03-24 13:51:40.000000000 +0900
+++ src/libnsfdb-20160324/configure.ac	2016-04-07 20:00:52.510495000 +0900
@@ -187,23 +187,7 @@ dnl Have configure make the Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
 AC_CONFIG_FILES([libnsfdb/Makefile])
-AC_CONFIG_FILES([libcsystem/Makefile])
 AC_CONFIG_FILES([nsfdbtools/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
@@ -255,4 +239,3 @@ Features:
    Verbose output:                             $ac_cv_enable_verbose_output
    Debug output:                               $ac_cv_enable_debug_output
 ]);
-
--- origsrc/libnsfdb-20160324/libnsfdb/Makefile.am	2016-03-24 13:51:40.000000000 +0900
+++ src/libnsfdb-20160324/libnsfdb/Makefile.am	2016-04-07 20:01:17.211990900 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -99,4 +100,3 @@ distclean: clean
 splint:
 	@echo "Running splint on libnsfdb ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(libnsfdb_la_SOURCES)
-
--- origsrc/libnsfdb-20160324/libnsfdb/libnsfdb_bucket_list.c	2016-03-24 13:51:40.000000000 +0900
+++ src/libnsfdb-20160324/libnsfdb/libnsfdb_bucket_list.c	2016-04-07 22:33:51.781086800 +0900
@@ -25,6 +25,7 @@
 #include "libnsfdb_bucket.h"
 #include "libnsfdb_libbfio.h"
 #include "libnsfdb_libcerror.h"
+#include "libnsfdb_libfcache.h"
 #include "libnsfdb_libfdata.h"
 
 /* Retrieves a specific summary or non summary bucket
@@ -73,4 +74,3 @@ int libnsfdb_bucket_list_get_bucket_by_i
 	}
 	return( 1 );
 }
-
--- origsrc/libnsfdb-20160324/nsfdbtools/Makefile.am	2016-03-24 13:51:40.000000000 +0900
+++ src/libnsfdb-20160324/nsfdbtools/Makefile.am	2016-04-07 20:01:26.362049800 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -82,4 +83,3 @@ splint:
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(nsfdbexport_SOURCES)
 	@echo "Running splint on nsfdbinfo ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(nsfdbinfo_SOURCES)
-
--- origsrc/libnsfdb-20160324/tests/Makefile.am	2016-03-24 13:51:40.000000000 +0900
+++ src/libnsfdb-20160324/tests/Makefile.am	2016-04-07 20:01:33.577716700 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -52,4 +53,3 @@ MAINTAINERCLEANFILES = \
 
 distclean: clean
 	/bin/rm -f Makefile
-
