--- origsrc/libpar2-0.4/Makefile.am	2013-07-18 19:02:58.000000000 +0900
+++ src/libpar2-0.4/Makefile.am	2015-03-31 20:01:28.625828700 +0900
@@ -63,11 +63,14 @@ EXTRA_DIST = PORTING ROADMAP debian/chan
 #TESTS = pretest test1 test2 test3 test4 test5 test6 posttest
 
 library_includedir=$(includedir)/libpar2
-library_include_HEADERS = *.h
+library_include_HEADERS = $(top_srcdir)/*.h
 
 
 
 libpar2_configdir = $(libdir)/libpar2/include
 libpar2_config_DATA = config.h
 
-libpar2_la_LDFLAGS = -version-info 1:0:0
+libpar2_la_LDFLAGS = -no-undefined -version-info 1:0:0
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = libpar2.pc
--- origsrc/libpar2-0.4/configure.ac	2013-07-13 00:34:04.000000000 +0900
+++ src/libpar2-0.4/configure.ac	2015-03-31 19:58:15.369431300 +0900
@@ -21,14 +21,14 @@ dnl
 dnl Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.58a)
-AC_INIT([libpar2], [0.3.1], [flesueur@users.sourceforge.net])
+AC_INIT([libpar2], [0.4], [flesueur@users.sourceforge.net])
 AC_CONFIG_SRCDIR([par2cmdline.cpp])
 
 AC_CANONICAL_HOST
 
 AM_CONFIG_HEADER(config.h)
 AC_CONFIG_MACRO_DIR([m4])
-AM_INIT_AUTOMAKE(libpar2,0.3.1)
+AM_INIT_AUTOMAKE
 
 
 dnl Checks for programs.
@@ -67,6 +67,5 @@ AC_CHECK_FUNCS([strchr] [memcpy])
 AC_CHECK_FUNCS([getopt] [getopt_long])
 
 
-AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([Makefile libpar2.pc])
 AC_OUTPUT
-
--- origsrc/libpar2-0.4/libpar2.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libpar2-0.4/libpar2.pc.in	2015-03-31 19:58:09.696223700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libpar2
+Description: Library for performing comman tasks related to PAR recovery sets
+Version: @VERSION@
+Libs: -L${libdir} -lpar2
+Cflags: -I${includedir}
