--- origsrc/libpar2-0.2/Makefile.am	2006-02-27 06:40:08.000000000 +0900
+++ src/libpar2-0.2/Makefile.am	2011-01-03 15:20:46.353475500 +0900
@@ -68,4 +68,4 @@ library_include_HEADERS = *.h
 libpar2_configdir = $(libdir)/libpar2/include
 libpar2_config_DATA = config.h
 
-libpar2_la_LDFLAGS = -version-info 0:1:0
\ No newline at end of file
+libpar2_la_LDFLAGS = -version-info 0:1:0 -no-undefined
--- origsrc/libpar2-0.2/configure.ac	2006-02-04 01:07:23.000000000 +0900
+++ src/libpar2-0.2/configure.ac	2011-01-03 15:20:33.334755500 +0900
@@ -36,8 +36,7 @@ AC_LANG_CPLUSPLUS
 AC_PROG_INSTALL
 
 dnl Checks for libraries.
-AC_PROG_LIBTOOL
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_CHECK_LIB(stdc++, main,,AC_MSG_ERROR(You need libstdc++))
 AC_CHECK_LIB(sigc-2.0, main,,AC_MSG_ERROR(You need libsigc++))
 
