--- origsrc/rudewebplatform/Makefile.am	2024-11-07 13:11:01.000000000 +0900
+++ src/rudewebplatform/Makefile.am	2024-11-07 13:11:03.245904200 +0900
@@ -41,7 +41,7 @@ librudewebplatform_la_SOURCES =	src/Comm
 	src/webplatform.h
 
 
-librudewebplatform_la_LDFLAGS = -version-info 3:2:1
+librudewebplatform_la_LDFLAGS = -no-undefined -version-info 3:2:1
 librudewebplatform_la_LIBADD = -lrudesocket -lrudeconfig -lrudecgi -lrudedatabase -lrudesmtp -lrudesession
 
 pkginclude_HEADERS = src/Command.h src/View.h src/CommandFactory.h src/ViewFactory.h src/Decorator.h src/webplatform.h
@@ -51,3 +51,6 @@ pkgincludedir = $(includedir)/rude
 AUTOMAKE_OPTIONS = gnits dist-bzip2 dist-shar dist-zip
 
 
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/rudewebplatform/configure.ac	2024-11-07 13:11:01.000000000 +0900
+++ src/rudewebplatform/configure.ac	2024-11-07 13:11:03.245904200 +0900
@@ -52,6 +52,6 @@ AC_TYPE_SIZE_T
 
 # Checks for library functions.
 AC_CHECK_FUNCS([strstr])
-AC_OUTPUT(Makefile )
+AC_OUTPUT(rudewebplatform.pc Makefile)
 
 
--- origsrc/rudewebplatform/rudewebplatform.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/rudewebplatform/rudewebplatform.pc.in	2024-11-07 13:11:03.245904200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: rudewebplatform
+Description: C++ Web Framework used by rudeserver
+Version: @VERSION@
+Libs: -L${libdir} -lrudewebplatform
+Cflags: -I${includedir}
