--- origsrc/aatm-0.6/src/Makefile.am	2016-02-27 05:20:50.000000000 +0900
+++ src/aatm-0.6/src/Makefile.am	2017-06-29 15:02:55.426665100 +0900
@@ -24,7 +24,7 @@ libaatm_la_CPPFLAGS = -Wall
 
 libaatm_la_LIBADD = 
 
-libaatm_la_LDFLAGS = -version-info $(AATM_LT_VERSION)
+libaatm_la_LDFLAGS = -no-undefined -version-info $(AATM_LT_VERSION)
 
 
 include_HEADERS = ATMAngle.h ATMEnumerations.h ATMError.h		\
--- origsrc/aatm-0.6/test/Makefile.am	2016-02-27 05:20:49.000000000 +0900
+++ src/aatm-0.6/test/Makefile.am	2017-06-29 15:02:55.426665100 +0900
@@ -25,7 +25,7 @@ dispersive_SOURCES = dispersive.cpp $(GE
 absorption_SOURCES = absorption.cpp $(GENFILES)
 
 LDADD=../src/libaatm.la
-INCLUDES=-I../src
+INCLUDES=-I$(top_srcdir)/src
 
 EXTRA_DIST=data/SMA_17JUL2006_skydip1.dat
 
