--- origsrc/libnabo-1.0.4/CMakeLists.txt	2013-09-03 21:02:16.000000000 +0900
+++ src/libnabo-1.0.4/CMakeLists.txt	2014-03-19 21:24:23.380712500 +0900
@@ -23,6 +23,7 @@ include(UseDoxygen)
 
 # Compilation
 add_definitions(-Wall)
+set(CMAKE_CXX_FLAGS "-mcmodel=large")
 
 # Boost required for any and cstdint on MSVC < 2010
 find_package(Boost COMPONENTS REQUIRED)
@@ -89,7 +90,7 @@ set(NABO_SRC
 set(SHARED_LIBS "false" CACHE BOOL "To build shared (true) or static (false) library")
 if (SHARED_LIBS)
 	add_library(${LIB_NAME} SHARED ${NABO_SRC})
-	install(TARGETS ${LIB_NAME} LIBRARY DESTINATION lib)
+	install(TARGETS ${LIB_NAME} RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 else(SHARED_LIBS)
 	add_library(${LIB_NAME} ${NABO_SRC})
 	install(TARGETS ${LIB_NAME} ARCHIVE DESTINATION lib)
