--- origsrc/ccaudio2-2.2.0/src/Makefile.am	2015-03-29 00:16:52.000000000 +0900
+++ src/ccaudio2-2.2.0/src/Makefile.am	2025-01-15 13:10:11.990787600 +0900
@@ -15,7 +15,7 @@ AM_CXXFLAGS = -I$(top_srcdir)/inc @CCAUD
 
 lib_LTLIBRARIES = libccaudio2.la
 
-libccaudio2_la_LDFLAGS = @CCAUDIO_LIBS@ $(RELEASE)
+libccaudio2_la_LDFLAGS = -no-undefined @CCAUDIO_LIBS@ $(RELEASE)
 libccaudio2_la_SOURCES = audiobase.cpp audiofile.cpp audiobuffer.cpp \
 	codec.cpp detect.cpp dialers.cpp fileio.cpp friends.cpp \
 	mapper.cpp oss.cpp osx.cpp resample.cpp stream.cpp w32.cpp \
--- origsrc/ccaudio2-2.2.0/src/oss.cpp	2015-11-03 17:30:29.000000000 +0900
+++ src/ccaudio2-2.2.0/src/oss.cpp	2025-01-15 13:05:04.851230700 +0900
@@ -27,7 +27,7 @@ int _oss_ccaudio_dummy = 0;
 #ifdef  HAVE_SYS_SOUNDCARD_H
 
 #include <sys/ioctl.h>
-#include <sys/fcntl.h>
+#include <fcntl.h>
 #include <unistd.h>
 #include <sys/soundcard.h>
 #ifdef  HAVE_ENDIAN_H
