--- origsrc/ophidia-terminal-1.8.0/src/Makefile.am	2023-08-01 01:14:21.000000000 +0900
+++ src/ophidia-terminal-1.8.0/src/Makefile.am	2024-09-24 11:36:41.302721100 +0900
@@ -24,7 +24,7 @@ LIBRARY += liboph_term_client.la
 if INTERFACE_TYPE_IS_SSL
 IFLAG += -DINTERFACE_TYPE_IS_SSL
 liboph_term_client_la_SOURCES = client/$(INTERFACE_TYPE)/oph_term_client.c client/stdsoap2.c client/soapClient.c client/soapC.c
-liboph_term_client_la_CFLAGS= -prefer-pic -I. -Iclient -Iclient/$(INTERFACE_TYPE) -Ienv -Iworkflow ${lib_CFLAGS} $(OPT) -DWITH_DOM $(LIBXML_INCLUDE) $(LIBCURL_INCLUDE) -DWITH_OPENSSL -DHAVE_OPENSSL_SSL_H $(LIBSSL_INCLUDE) $(IFLAG)
+liboph_term_client_la_CFLAGS= -prefer-pic -I. -Iclient -Iclient/$(INTERFACE_TYPE) -Ienv -Iworkflow -I$(top_srcdir)/src/client -I$(top_srcdir)/src/workflow ${lib_CFLAGS} $(OPT) -DWITH_DOM $(LIBXML_INCLUDE) $(LIBCURL_INCLUDE) -DWITH_OPENSSL -DHAVE_OPENSSL_SSL_H $(LIBSSL_INCLUDE) $(IFLAG)
 liboph_term_client_la_LDFLAGS = -static
 liboph_term_client_la_LIBADD = $(LIBXML_LIB) $(LIBCURL_LIB) -lpthread $(LIBSSL_LIB) -lcrypto
 endif
@@ -94,7 +94,7 @@ bindir=${prefix}/bin
 oph_term_DEPENDENCIES= liboph_term_client.la liboph_term_env.la liboph_term_help.la liboph_term_viewer.la liboph_term_workflow.la
 oph_term_SOURCES= oph_term.c
 oph_term_CFLAGS= $(OPT) -I. -Iclient -Ienv -Ihelp -Iviewer -Iworkflow ${lib_CFLAGS} $(LIBZ_INCLUDE) $(LIBXML_INCLUDE) $(FLAG) -DPACKAGE_VERSION=\"@PACKAGE_VERSION@\"
-oph_term_LDADD= -L. -loph_term_client -loph_term_env -loph_term_help -loph_term_viewer -loph_term_workflow -lreadline -lhistory $(LIBZ_LIB) -lncurses -lm -lpthread
+oph_term_LDADD= liboph_term_client.la liboph_term_env.la liboph_term_help.la liboph_term_viewer.la liboph_term_workflow.la -lreadline -lhistory $(LIBZ_LIB) -lncurses -lm -lpthread
 
 distclean-local:
 	cd client && rm -f *~ #*.nsmap *.xsd *.xml soap* *.wsdl
