--- origsrc/ponder-3.0.0/CMakeLists.txt	2018-09-30 01:38:46.000000000 +0900
+++ src/ponder-3.0.0/CMakeLists.txt	2025-06-15 14:09:24.780952000 +0900
@@ -261,6 +261,10 @@ if (WIN32 AND CMAKE_COMPILER_IS_GNUCXX)
     endif()
 endif()
 
+if (DEFINED SOVERSION)
+    set_target_properties(ponder PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 # add the test subdirectory, but do not build it by default
 if(BUILD_TEST)
     enable_testing()
@@ -316,12 +320,12 @@ install(FILES ${CMAKE_BINARY_DIR}/ponder
 
 install(EXPORT PonderTargets
     NAMESPACE ponder::
-    DESTINATION lib/ponder/cmake
+    DESTINATION lib/cmake/ponder
     COMPONENT devel
 )
 
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/PonderConfig.cmake ${CMAKE_CURRENT_BINARY_DIR}/PonderConfigVersion.cmake
-    DESTINATION lib/ponder/cmake
+    DESTINATION lib/cmake/ponder
     COMPONENT devel
 )
 
