--- origsrc/ponder-3.0.0/CMakeLists.txt	2018-09-30 01:38:46.000000000 +0900
+++ src/ponder-3.0.0/CMakeLists.txt	2019-01-22 08:18:57.707105000 +0900
@@ -44,7 +44,6 @@ set(PONDER_VERSION_STR "${PONDER_VERSION
 
 project(ponder
         VERSION ${PONDER_VERSION}
-        DESCRIPTION "C++ reflection library"
         LANGUAGES C CXX)
 message("Project version: ${PONDER_VERSION_STR}")
 
@@ -244,6 +243,9 @@ target_include_directories(ponder PUBLIC
 # define the export macro
 if(BUILD_SHARED_LIBS)
     set_target_properties(ponder PROPERTIES DEFINE_SYMBOL PONDER_EXPORTS)
+    if (CYGWIN)
+        set_target_properties(ponder PROPERTIES SOVERSION 0)
+    endif()
 else()
     add_definitions(-DPONDER_STATIC)
 endif()
@@ -316,23 +318,27 @@ install(FILES ${CMAKE_BINARY_DIR}/ponder
 
 install(EXPORT PonderTargets
     NAMESPACE ponder::
-    DESTINATION lib/ponder/cmake
+    DESTINATION lib/cmake/ponder
     COMPONENT devel
 )
 
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/PonderConfig.cmake ${CMAKE_CURRENT_BINARY_DIR}/PonderConfigVersion.cmake
-    DESTINATION lib/ponder/cmake
+    DESTINATION lib/cmake/ponder
     COMPONENT devel
 )
 
-install(FILES README.md COPYING.txt
-    DESTINATION ${INSTALL_MISC_DIR}
-)
-
-install(DIRECTORY cmake
-    DESTINATION ${INSTALL_MISC_DIR}
-    COMPONENT utils
-)
+if (CYGWIN)
+    install(FILES cmake/Modules/FindPonder.cmake DESTINATION lib/cmake/ponder)
+else()
+    install(FILES README.md COPYING.txt
+        DESTINATION ${INSTALL_MISC_DIR}
+    )
+
+    install(DIRECTORY cmake
+        DESTINATION ${INSTALL_MISC_DIR}
+        COMPONENT utils
+    )
+endif()
 
 ###############################
 # packaging
