--- origsrc/tamu_anova-0.2/Makefile.am	2015-05-17 12:31:28.336384200 +0900
+++ src/tamu_anova-0.2/Makefile.am	2015-05-17 13:21:31.694921100 +0900
@@ -2,6 +2,7 @@ lib_LTLIBRARIES = libtamuanova.la
 pkginclude_HEADERS = tamu_anova.h
 libtamuanova_la_SOURCES = anova_1.c anova_2.c
 libtamuanova_la_LIBADD = $(GSL_LIBS)
+libtamuanova_la_LDFLAGS = -no-undefined
 
 TESTS = $(check_PROGRAMS)
 check_PROGRAMS = test
@@ -10,7 +11,7 @@ test_SOURCES = test.c
 
 CLEANFILES = test.dat
 
-info_TEXINFOS = tamu_anova.texi 
-tamu_anova_TEXINFOS = fdl.texi mathinclude.texi
-
 EXTRA_DIST = autogen.sh
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/tamu_anova-0.2/configure.ac	2015-05-17 12:31:28.336384200 +0900
+++ src/tamu_anova-0.2/configure.ac	2015-05-17 13:21:16.963093700 +0900
@@ -29,5 +29,5 @@ AC_CHECK_HEADERS([stdlib.h])
 AC_C_CONST
 AC_C_INLINE
 
-AC_CONFIG_FILES(Makefile)
+AC_CONFIG_FILES(Makefile tamu_anova.pc)
 AC_OUTPUT
--- origsrc/tamu_anova-0.2/tamu_anova.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/tamu_anova-0.2/tamu_anova.pc.in	2015-05-17 12:35:44.567498300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: tamu_anova
+Description: ANOVA Extensions to the GNU Scientific Library
+Version: @VERSION@
+Libs: -L${libdir} -ltamuanova
+Cflags: -I${includedir}
