--- origsrc/cgcode/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/cgcode/Makefile.am	2012-02-12 14:48:31.052297500 +0900
@@ -0,0 +1,9 @@
+SUBDIRS = src
+
+bin_PROGRAMS = goos good
+
+goos_SOURCES = mains.f
+goos_LDADD = src/libcgcode.la $(BLAS_LIBS)
+
+good_SOURCES = maind.f
+good_LDADD = src/libcgcode.la $(BLAS_LIBS)
--- origsrc/cgcode/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/cgcode/configure.ac	2012-02-12 14:48:49.298534300 +0900
@@ -0,0 +1,24 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([cgcode], [1.0])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_MAKE_SET
+AC_PROG_F77
+LT_INIT([disable-static])
+
+# Checks for libraries.
+PKG_CHECK_MODULES([BLAS], [blas])
+
+# Checks for header files.
+
+# Checks for typedefs, structures, and compiler characteristics.
+
+# Checks for library functions.
+
+AC_CONFIG_FILES([Makefile
+                 src/Makefile])
+AC_OUTPUT
--- origsrc/cgcode/src/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/cgcode/src/Makefile.am	2012-02-12 14:56:09.211132700 +0900
@@ -0,0 +1,11 @@
+lib_LTLIBRARIES = libcgcode.la
+
+libcgcode_la_SOURCES = \
+d1mach.f  dcbfix.f  dcg.f     dcgchk.f  dcgdrv.f  dcgne.f   dcgnr.f   dckchb.f \
+dcr.f     dcrind.f  ddpchb.f  depsln.f  donest.f  dpcg.f    dpcgca.f  dpcgne.f \
+dpcgnr.f  dppcg.f   dratqr.f  mdstop.f  msstop.f  r1mach.f  scbfix.f  scg.f    \
+scgchk.f  scgdrv.f  scgne.f   scgnr.f   sckchb.f  scr.f     scrind.f  sdpchb.f \
+sonest.f  spcg.f    spcgca.f  spcgne.f  spcgnr.f  sppcg.f   sratqr.f
+libcgcode_la_LIBADD = $(BLAS_LIBS)
+libcgcode_la_LDFLAGS = -no-undefined
+
