--- origsrc/janet-1.18.1/src/core/util.c	2021-10-17 05:05:48.000000000 +0900
+++ src/janet-1.18.1/src/core/util.c	2021-10-27 22:49:30.377496000 +0900
@@ -850,7 +850,7 @@ int janet_cryptorand(uint8_t *out, size_
     }
     RETRY_EINTR(rc, close(randfd));
     return 0;
-#elif defined(JANET_BSD) || defined(MAC_OS_X_VERSION_10_7)
+#elif defined(JANET_BSD) || defined(MAC_OS_X_VERSION_10_7) || defined(__CYGWIN__)
     arc4random_buf(out, n);
     return 0;
 #else
