--- origsrc/janet-1.16.0/src/core/util.c	2021-05-31 02:15:56.000000000 +0900
+++ src/janet-1.16.0/src/core/util.c	2021-06-09 08:58:47.396565800 +0900
@@ -744,7 +744,7 @@ int janet_cryptorand(uint8_t *out, size_
     }
     RETRY_EINTR(rc, close(randfd));
     return 0;
-#elif defined(JANET_BSD) || defined(MAC_OS_X_VERSION_10_7)
+#elif defined(JANET_BSD) || defined(MAC_OS_X_VERSION_10_7) || defined(__CYGWIN__)
     arc4random_buf(out, n);
     return 0;
 #else
