--- origsrc/janet-1.23.0/src/core/util.c	2022-06-21 01:23:21.000000000 +0900
+++ src/janet-1.23.0/src/core/util.c	2022-07-01 17:18:55.712355800 +0900
@@ -874,7 +874,7 @@ int janet_cryptorand(uint8_t *out, size_
     }
     RETRY_EINTR(rc, close(randfd));
     return 0;
-#elif defined(JANET_BSD) || defined(MAC_OS_X_VERSION_10_7)
+#elif defined(JANET_BSD) || defined(MAC_OS_X_VERSION_10_7) || defined(__CYGWIN__)
     arc4random_buf(out, n);
     return 0;
 #else
