--- origsrc/janet-1.25.1/src/core/util.c	2022-10-30 01:58:29.000000000 +0900
+++ src/janet-1.25.1/src/core/util.c	2023-01-03 21:57:36.474460500 +0900
@@ -859,7 +859,7 @@ int janet_cryptorand(uint8_t *out, size_
         }
     }
     return 0;
-#elif defined(JANET_LINUX) || ( defined(JANET_APPLE) && !defined(MAC_OS_X_VERSION_10_7) )
+#elif defined(JANET_LINUX) || ( defined(JANET_APPLE) && !defined(MAC_OS_X_VERSION_10_7) ) || defined(JANET_CYGWIN)
     /* We should be able to call getrandom on linux, but it doesn't seem
        to be uniformly supported on linux distros.
        On Mac, arc4random_buf wasn't available on until 10.7.
--- origsrc/janet-1.25.1/src/include/janet.h	2022-10-30 01:58:29.000000000 +0900
+++ src/janet-1.25.1/src/include/janet.h	2023-01-03 21:56:14.228939700 +0900
@@ -67,6 +67,10 @@ extern "C" {
 #define JANET_LINUX 1
 #endif
 
+#if defined(__CYGWIN__)
+#define JANET_CYGWIN 1
+#endif
+
 /* Check Unix */
 #if defined(_AIX) \
     || defined(__APPLE__) /* Darwin */ \
