--- origsrc/janet-1.12.2/src/core/util.c	2020-09-21 02:03:59.000000000 +0900
+++ src/janet-1.12.2/src/core/util.c	2020-09-25 22:41:47.146780900 +0900
@@ -676,7 +676,7 @@ int janet_cryptorand(uint8_t *out, size_
     }
     RETRY_EINTR(rc, close(randfd));
     return 0;
-#elif defined(JANET_BSD) || defined(MAC_OS_X_VERSION_10_7)
+#elif defined(JANET_BSD) || defined(MAC_OS_X_VERSION_10_7) || defined(__CYGWIN__)
     arc4random_buf(out, n);
     return 0;
 #else
