--- origsrc/janet-1.13.1/src/core/util.c	2020-12-14 02:59:52.000000000 +0900
+++ src/janet-1.13.1/src/core/util.c	2020-12-24 13:41:53.727467300 +0900
@@ -677,7 +677,7 @@ int janet_cryptorand(uint8_t *out, size_
     }
     RETRY_EINTR(rc, close(randfd));
     return 0;
-#elif defined(JANET_BSD) || defined(MAC_OS_X_VERSION_10_7)
+#elif defined(JANET_BSD) || defined(MAC_OS_X_VERSION_10_7) || defined(__CYGWIN__)
     arc4random_buf(out, n);
     return 0;
 #else
