--- origsrc/oaml-1.3.4/CMakeLists.txt	2018-07-25 22:07:28.000000000 +0900
+++ src/oaml-1.3.4/CMakeLists.txt	2019-01-03 04:57:55.260435900 +0900
@@ -82,6 +82,8 @@ else()
 	message("Build oamlTest executable: No  (Enable by param -DENABLE_TEST=ON)")
 endif()
 
+include(FindPkgConfig)
+pkg_check_modules(TINYXML2 tinyxml2)
 
 ##
 # Set CXX_FLAGS depending on compiler
@@ -227,9 +229,12 @@ set(SOURCES
 	src/oamlStudioApi.cpp
 	src/oamlTrack.cpp
 	src/oamlUtil.cpp
-	src/tinyxml2.cpp
 	src/wav.cpp)
 
+if (NOT TINYXML2_FOUND)
+	list(APPEND SOURCES src/tinyxml2.cpp)
+endif()
+
 if (ENABLE_OGG)
 	list(APPEND SOURCES src/ogg.cpp)
 endif()
@@ -245,6 +250,9 @@ endif()
 if (ENABLE_SHARED)
 	add_library(oaml_shared SHARED ${SOURCES} src/oamlC.cpp)
 	target_link_libraries(oaml_shared ${OAML_LIBS})
+	if (TINYXML2_FOUND)
+		target_link_libraries(oaml_shared ${TINYXML2_LIBRARIES})
+	endif()
 	set_target_properties(oaml_shared
 		PROPERTIES
 			OUTPUT_NAME oaml
@@ -256,6 +264,9 @@ endif()
 if (ENABLE_TEST)
 	add_executable(oamlTest ${SOURCES} src/test.cpp)
 	target_link_libraries(oamlTest ${OAML_LIBS})
+	if (TINYXML2_FOUND)
+		target_link_libraries(oamlTest ${TINYXML2_LIBRARIES})
+	endif()
 endif()
 
 ##
@@ -282,7 +293,7 @@ if (ENABLE_STATIC)
 endif()
 
 if (ENABLE_SHARED)
-	install(TARGETS oaml_shared DESTINATION lib${LIB_SUFFIX})
+	install(TARGETS oaml_shared RUNTIME DESTINATION bin ARCHIVE DESTINATION lib${LIB_SUFFIX})
 endif()
 
 configure_file("oaml.cmake.in" "oaml.cmake" @ONLY)
