--- origsrc/liboop-1.0/Makefile.am	2003-10-27 01:39:10.000000000 +0900
+++ src/liboop-1.0/Makefile.am	2012-03-27 14:33:10.037698500 +0900
@@ -14,35 +14,35 @@ pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = liboop.pc liboop-glib2.pc
 
 # versions updated as of 1.0
-liboop_la_LDFLAGS = -version-info 4:1:0 # version:revision:age
+liboop_la_LDFLAGS = -version-info 4:1:0 -no-undefined # version:revision:age
 liboop_la_SOURCES = sys.c select.c signal.c alloc.c read.c read-fd.c read-mem.c
 
-liboop_adns_la_LDFLAGS = -version-info 3:0:0 -rpath '$(libdir)'
-liboop_adns_la_LIBADD = $(ADNS_LIBS)
+liboop_adns_la_LDFLAGS = -version-info 3:0:0 -no-undefined -rpath '$(libdir)'
+liboop_adns_la_LIBADD = $(ADNS_LIBS) liboop.la
 liboop_adns_la_SOURCES = adns.c
 
-liboop_glib_la_LDFLAGS = -version-info 1:0:1 -rpath '$(libdir)'
-liboop_glib_la_LIBADD = $(GLIB_LIBS)
+liboop_glib_la_LDFLAGS = -version-info 1:0:1 -no-undefined -rpath '$(libdir)'
+liboop_glib_la_LIBADD = $(GLIB_LIBS) liboop.la
 liboop_glib_la_CFLAGS = $(GLIB_INCLUDES)
 liboop_glib_la_SOURCES = glib.c
 
-liboop_glib2_la_LDFLAGS = -version-info 0:0:0 -rpath '$(libdir)'
-liboop_glib2_la_LIBADD = $(GLIB2_LIBS)
+liboop_glib2_la_LDFLAGS = -version-info 0:0:0 -no-undefined -rpath '$(libdir)'
+liboop_glib2_la_LIBADD = $(GLIB2_LIBS) liboop.la
 liboop_glib2_la_CFLAGS = $(GLIB2_CFLAGS)
 liboop_glib2_la_SOURCES = glib.c
 
-liboop_tcl_la_LDFLAGS = -version-info 0:0:0 -rpath '$(libdir)'
-liboop_tcl_la_LIBADD = $(TCL_LIBS)
+liboop_tcl_la_LDFLAGS = -version-info 0:0:0 -no-undefined -rpath '$(libdir)'
+liboop_tcl_la_LIBADD = $(TCL_LIBS) liboop.la
 liboop_tcl_la_CFLAGS = $(TCL_INCLUDES)
 liboop_tcl_la_SOURCES = tcl.c
 
-liboop_www_la_LDFLAGS = -version-info 0:0:0 -rpath '$(libdir)'
-liboop_www_la_LIBADD = $(WWW_LIBS)
+liboop_www_la_LDFLAGS = -version-info 0:0:0 -no-undefined -rpath '$(libdir)'
+liboop_www_la_LIBADD = $(WWW_LIBS) liboop.la
 liboop_www_la_CFLAGS = $(WWW_INCLUDES)
 liboop_www_la_SOURCES = www.c
 
-liboop_rl_la_LDFLAGS = -version-info 0:0:0 -rpath '$(libdir)'
-liboop_rl_la_LIBADD = $(READLINE_LIBS)
+liboop_rl_la_LDFLAGS = -version-info 0:0:0 -no-undefined -rpath '$(libdir)'
+liboop_rl_la_LIBADD = $(READLINE_LIBS) liboop.la
 liboop_rl_la_SOURCES = readline.c
 
 include_HEADERS = oop.h oop-adns.h oop-glib.h oop-tcl.h oop-www.h oop-rl.h oop-read.h
--- origsrc/liboop-1.0/configure.ac	2003-10-27 01:39:21.000000000 +0900
+++ src/liboop-1.0/configure.ac	2012-03-27 14:36:34.897073500 +0900
@@ -110,7 +110,7 @@ if test -z "$no_wacky_libs" ; then
 fi
 
 test yes = "$GCC" && 
-CFLAGS="-Wall -Wno-comment -Wmissing-prototypes -Wstrict-prototypes -Wpointer-arith -Wwrite-strings $CFLAGS"' $(EXTRA_CFLAGS)'
+CFLAGS="-Wall -Wno-comment -Wmissing-prototypes -Wstrict-prototypes -Wpointer-arith -Wwrite-strings $CFLAGS"
 
 AC_SUBST(PROG_LDCONFIG)
 AC_SUBST(GLIB_INCLUDES)
