--- origsrc/dbcsr-2.5.0/CMakeLists.txt	2022-12-28 06:23:37.000000000 +0900
+++ src/dbcsr-2.5.0/CMakeLists.txt	2023-01-17 21:57:56.953010100 +0900
@@ -329,3 +329,9 @@ endif ()
 add_subdirectory(docs)
 
 include(CustomTargets)
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/dbcsr.pc.in ${CMAKE_BINARY_DIR}/dbcsr.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/dbcsr.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/dbcsr-2.5.0/dbcsr.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/dbcsr-2.5.0/dbcsr.pc.in	2023-01-17 21:57:56.960103500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: dbcsr
+Description: Distributed Block Compressed Sparse Row matrix library
+Version: @dbcsr_VERSION@
+Libs: -L${libdir} -ldbcsr
+Cflags: -I${includedir}
--- origsrc/dbcsr-2.5.0/src/CMakeLists.txt	2022-12-28 06:23:37.000000000 +0900
+++ src/dbcsr-2.5.0/src/CMakeLists.txt	2023-01-17 21:57:56.963043400 +0900
@@ -288,6 +288,7 @@ set(config_namespace "DBCSR::")
 install(
   TARGETS dbcsr
   EXPORT DBCSRTargets
+  RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
   LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
   ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/dbcsr_api.mod"
@@ -300,6 +301,7 @@ if (WITH_C_API)
     TARGETS dbcsr_c
     EXPORT DBCSRTargets
     COMPONENT C
+    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
     LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
     ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
   install(
