--- origsrc/dbcsr-2.8.0/CMakeLists.txt	2024-12-11 21:01:26.000000000 +0900
+++ src/dbcsr-2.8.0/CMakeLists.txt	2024-12-13 13:20:52.910585300 +0900
@@ -358,3 +358,10 @@ include(CustomTargets)
 
 # Disable LTO
 set(CMAKE_INTERPROCEDURAL_OPTIMIZATION FALSE FORCE)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/dbcsr.pc.in ${CMAKE_BINARY_DIR}/dbcsr.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/dbcsr.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/dbcsr-2.8.0/dbcsr.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/dbcsr-2.8.0/dbcsr.pc.in	2024-12-13 13:21:24.353718200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: dbcsr
+Description: Distributed Block Compressed Sparse Row matrix library
+Version: @dbcsr_VERSION@
+Libs: -L${libdir} -ldbcsr
+Cflags: -I${includedir}
--- origsrc/dbcsr-2.8.0/src/CMakeLists.txt	2024-12-11 21:01:26.000000000 +0900
+++ src/dbcsr-2.8.0/src/CMakeLists.txt	2024-12-13 12:32:17.373297500 +0900
@@ -324,7 +324,7 @@ set(config_namespace "DBCSR::")
 install(
   TARGETS dbcsr
   EXPORT DBCSRTargets
-  LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+  RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
   ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 # See https://gitlab.kitware.com/cmake/cmake/-/issues/19608
 # CMAKE_INSTALL_Fortran_MODULES is not an "official" cmake variable yet, but
@@ -342,7 +342,7 @@ if (WITH_C_API)
     TARGETS dbcsr_c
     EXPORT DBCSRTargets
     COMPONENT C
-    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
     ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
   install(
     FILES "${CMAKE_CURRENT_BINARY_DIR}/dbcsr.h"
