--- origsrc/isl-0.12/Makefile.am	2013-06-23 21:13:54.000000000 +0900
+++ src/isl-0.12/Makefile.am	2013-06-27 23:33:52.923050500 +0900
@@ -155,7 +155,7 @@ EXTRA_libisl_la_SOURCES = \
 	isl_piplib.c
 libisl_la_LIBADD = @PIPLIB_LIBS@ @GMP_LIBS@
 libisl_la_LDFLAGS = -version-info @versioninfo@ \
-	@PIPLIB_LDFLAGS@ @GMP_LDFLAGS@
+	@PIPLIB_LDFLAGS@ @GMP_LDFLAGS@ -no-undefined
 libisl_la_CPPFLAGS = $(INCLUDES) @PIPLIB_CPPFLAGS@ @GMP_CPPFLAGS@
 
 isl_test_CPPFLAGS = $(INCLUDES) @GMP_CPPFLAGS@
