--- origsrc/yetris-2.3.0/Makefile	2014-10-11 01:01:34.000000000 +0900
+++ src/yetris-2.3.0/Makefile	2023-11-25 00:49:01.858832800 +0900
@@ -68,8 +68,8 @@ INCLUDESDIR = -I"src/" -I"deps/"
 LIBSDIR     =
 
 # All source files
-CFILES   = $(shell find src -type f -name '*.c')
-CXXFILES = $(shell find src -type f -name '*.cpp')
+CFILES   = $(shell find -L src -type f -name '*.c')
+CXXFILES = $(shell find -L src -type f -name '*.cpp')
 OBJECTS  = $(CFILES:.c=.o) \
            $(CXXFILES:.cpp=.o)
 
--- origsrc/yetris-2.3.0/src/Engine/Graphics/Ncurses.cpp	2014-10-11 01:01:34.000000000 +0900
+++ src/yetris-2.3.0/src/Engine/Graphics/Ncurses.cpp	2023-11-25 00:50:39.071078400 +0900
@@ -16,7 +16,7 @@ bool Ncurses::init()
 	//
 	// VIM uses 25ms, so should you.
 	// Source: http://en.chys.info/2009/09/esdelay-ncurses/
-	ESCDELAY = 25;
+	set_escdelay(25);
 
 	refresh();   // Refresh the layout (prints whats in the layout bu
 	return true;
