--- origsrc/irrtoolset-5.0.1/src/Makefile.am	2012-07-31 21:10:40.000000000 +0900
+++ src/irrtoolset-5.0.1/src/Makefile.am	2013-10-06 10:11:48.499800200 +0900
@@ -28,7 +28,7 @@
 SUBDIRS= 	dataset irr normalform re2dfa rpsl irrutil . \
 		peval rpslcheck rtconfig
 
-INCLUDES=	-I@top_srcdir@/
+AM_CPPFLAGS=	-I$(top_srcdir)/src
 
 noinst_LTLIBRARIES = libirrtoolset.la
 libirrtoolset_la_SOURCES =
--- origsrc/irrtoolset-5.0.1/src/dataset/Makefile.am	2012-07-31 21:10:39.000000000 +0900
+++ src/irrtoolset-5.0.1/src/dataset/Makefile.am	2013-10-06 10:13:58.367964200 +0900
@@ -22,6 +22,7 @@
 #  Author(s): Nick Hilliard <nick@foobar.org>
 
 noinst_LTLIBRARIES = libdataset.la
+libdataset_la_CPPFLAGS = -I$(top_srcdir)/src
 libdataset_la_SOURCES = \
 	CacheKey.CacheValue.AVLMap.cc CacheKey.CacheValue.Map.cc \
 	NormalTermPtr.DLList.cc PrefixRange.Plex.cc PrefixRange.XPlex.cc \
--- origsrc/irrtoolset-5.0.1/src/irr/Makefile.am	2012-07-31 21:10:40.000000000 +0900
+++ src/irrtoolset-5.0.1/src/irr/Makefile.am	2013-10-06 10:16:20.846479900 +0900
@@ -22,6 +22,7 @@
 #  Author(s): Nick Hilliard <nick@foobar.org>
 
 noinst_LTLIBRARIES = libirr.la
+libirr_la_CPPFLAGS = -I$(top_srcdir)/src
 libirr_la_SOURCES = autnum.cc birdwhoisc.cc cache_value.cc irr.cc rawhoisc.cc route.cc \
                    autnum.hh birdwhoisc.hh cache.hh cache_value.hh classes.hh irr.hh rawhoisc.hh route.hh
 
--- origsrc/irrtoolset-5.0.1/src/normalform/Makefile.am	2012-07-31 21:10:40.000000000 +0900
+++ src/irrtoolset-5.0.1/src/normalform/Makefile.am	2013-10-06 10:17:38.220503300 +0900
@@ -22,6 +22,7 @@
 #  Author(s): Nick Hilliard <nick@foobar.org>
 
 noinst_LTLIBRARIES = libnormalform.la
+libnormalform_la_CPPFLAGS = -I$(top_srcdir)/src
 libnormalform_la_SOURCES = \
 	Filter.cc FilterOfASPath.cc FilterOfCommunity.cc FilterOfPrefix.cc \
 	FilterOfSymbol.cc IPv6RadixSet.cc NE.cc NT.cc RadixSet.cc \
--- origsrc/irrtoolset-5.0.1/src/peval/Makefile.am	2012-07-31 21:10:39.000000000 +0900
+++ src/irrtoolset-5.0.1/src/peval/Makefile.am	2013-10-06 10:24:10.626753300 +0900
@@ -22,6 +22,7 @@
 #  Author(s): Nick Hilliard <nick@foobar.org>
 
 bin_PROGRAMS = peval
+peval_CPPFLAGS = -I$(top_srcdir)/src
 peval_SOURCES = peval.cc
 
 dist_man1_MANS = peval.1
--- origsrc/irrtoolset-5.0.1/src/re2dfa/Makefile.am	2012-07-31 21:10:39.000000000 +0900
+++ src/irrtoolset-5.0.1/src/re2dfa/Makefile.am	2013-10-06 10:19:30.003706400 +0900
@@ -22,6 +22,7 @@
 #  Author(s): Nick Hilliard <nick@foobar.org>
 
 noinst_LTLIBRARIES = libre2dfa.la
+libre2dfa_la_CPPFLAGS = -I$(top_srcdir)/src
 libre2dfa_la_SOURCES = re2dfa.c regexp_nf.cc re2dfa.h regexp_nf.hh
 
 CLEANFILES = *~ *.bak
--- origsrc/irrtoolset-5.0.1/src/rpsl/Makefile.am	2012-07-31 21:10:39.000000000 +0900
+++ src/irrtoolset-5.0.1/src/rpsl/Makefile.am	2013-10-06 10:23:14.193159500 +0900
@@ -22,6 +22,7 @@
 #  Author(s): Nick Hilliard <nick@foobar.org>
 
 noinst_LTLIBRARIES = librpsl.la
+librpsl_la_CPPFLAGS = -I$(top_srcdir)/src
 librpsl_la_SOURCES = \
 	Buffer.cc afi.cc object.cc prefix.cc rangelist.cc regexp.cc \
 	rpsl_attr.cc rpsl_filter.cc rpsl_item.cc rpsl_policy.cc rptype.cc \
@@ -41,10 +42,10 @@ schema_rpsl.cc: rpsl.y.cc
 
 # FIXME: dependency on lex / yacc sources
 object_log.l.cc:
-	$(LEX) object_log.l
+	$(LEX) $(srcdir)/object_log.l
 
 rpsl.l.cc: rpsl.y.cc
-	$(LEX) rpsl.l
+	$(LEX) $(srcdir)/rpsl.l
         
 rpsl.y.cc:
-	$(YACC) -d rpsl.y
+	$(YACC) -d $(srcdir)/rpsl.y
--- origsrc/irrtoolset-5.0.1/src/rpslcheck/Makefile.am	2012-07-31 21:10:40.000000000 +0900
+++ src/irrtoolset-5.0.1/src/rpslcheck/Makefile.am	2013-10-06 10:26:31.174677400 +0900
@@ -22,6 +22,7 @@
 #  Author(s): Nick Hilliard <nick@foobar.org>
 
 bin_PROGRAMS = rpslcheck
+rpslcheck_CPPFLAGS = -I$(top_srcdir)/src
 rpslcheck_SOURCES = rpslcheck.cc
 
 dist_man1_MANS = rpslcheck.1
--- origsrc/irrtoolset-5.0.1/src/rtconfig/Makefile.am	2012-07-31 21:10:40.000000000 +0900
+++ src/irrtoolset-5.0.1/src/rtconfig/Makefile.am	2013-10-06 10:28:14.218399100 +0900
@@ -22,7 +22,7 @@
 #  Author(s): Nick Hilliard <nick@foobar.org>
 
 bin_PROGRAMS = rtconfig
-
+rtconfig_CPPFLAGS = -I$(top_srcdir)/src
 rtconfig_SOURCES = rtconfig.cc f_cisco.cc f_junos.cc \
                    rtconfig.hh f_cisco.hh f_junos.hh
 
@@ -36,8 +36,8 @@ CLEANFILES = *~ *.bak
 LDADD = ../libirrtoolset.la @EXTRA_LIBS@
 
 command.l.cc: command.y.cc
-	$(LEX) command.l
+	$(LEX) $(srcdir)/command.l
 
 command.y.cc:
-	$(YACC) -d command.y
+	$(YACC) -d $(srcdir)/command.y
 
