--- origsrc/libcerf-1.7/lib/CMakeLists.txt	2018-09-26 23:48:18.000000000 +0900
+++ src/libcerf-1.7/lib/CMakeLists.txt	2018-11-22 11:38:09.124442200 +0900
@@ -29,10 +29,11 @@ endif()
 target_link_libraries(${library_name} m)
 
 install(
-    TARGETS ${library_name} LIBRARY
-    DESTINATION ${destination}/lib
+    TARGETS ${library_name}
+    RUNTIME DESTINATION bin
+    ARCHIVE DESTINATION lib
     COMPONENT Libraries)
 install(
     FILES ${inc_files}
-    DESTINATION ${destination}/include
+    DESTINATION include
     COMPONENT Headers)
