--- origsrc/libpreludedb-1.0.1p1/plugins/format/classic/Makefile.am	2012-03-19 22:41:40.000000000 +0900
+++ src/libpreludedb-1.0.1p1/plugins/format/classic/Makefile.am	2012-12-17 07:17:44.282047400 +0900
@@ -3,7 +3,7 @@ SUBDIRS=include
 AM_CPPFLAGS=@PCFLAGS@ -I$(top_srcdir)/src/include -I$(srcdir)/include -I$(top_srcdir)/libmissing -I$(top_builddir)/libmissing @LIBPRELUDE_CFLAGS@
 
 classic_la_LIBADD  = $(top_builddir)/src/libpreludedb.la @LIBPRELUDE_LIBS@
-classic_la_LDFLAGS = -module -avoid-version @LIBPRELUDE_LDFLAGS@
+classic_la_LDFLAGS = -no-undefined -module -avoid-version @LIBPRELUDE_LDFLAGS@
 classic_la_SOURCES = classic.c classic-delete.c classic-get.c classic-insert.c classic-path-resolve.c classic-sql-join.c classic-sql-select.c
 classic_LTLIBRARIES = classic.la
 classicdir = $(format_plugin_dir)
--- origsrc/libpreludedb-1.0.1p1/plugins/sql/mysql/Makefile.am	2012-03-19 22:41:40.000000000 +0900
+++ src/libpreludedb-1.0.1p1/plugins/sql/mysql/Makefile.am	2012-12-17 07:18:22.883609900 +0900
@@ -1,5 +1,5 @@
 AM_CPPFLAGS=@PCFLAGS@ -I$(top_srcdir)/src/include -I$(top_srcdir)/libmissing -I$(top_builddir)/libmissing @LIBPRELUDE_CFLAGS@ @MYSQL_CFLAGS@
-mysql_la_LDFLAGS = -module -avoid-version @LIBPRELUDE_LDFLAGS@
+mysql_la_LDFLAGS = -no-undefined -module -avoid-version @LIBPRELUDE_LDFLAGS@
 
 if HAVE_MYSQL
 
--- origsrc/libpreludedb-1.0.1p1/plugins/sql/pgsql/Makefile.am	2012-03-19 22:41:40.000000000 +0900
+++ src/libpreludedb-1.0.1p1/plugins/sql/pgsql/Makefile.am	2012-12-17 07:18:45.805484900 +0900
@@ -1,5 +1,5 @@
 AM_CPPFLAGS=@PCFLAGS@ -I$(top_srcdir)/src/include -I$(top_srcdir)/libmissing -I$(top_builddir)/libmissing @LIBPRELUDE_CFLAGS@ @POSTGRESQL_CFLAGS@
-pgsql_la_LDFLAGS = -module -avoid-version @LIBPRELUDE_LDFLAGS@ 
+pgsql_la_LDFLAGS = -no-undefined -module -avoid-version @LIBPRELUDE_LDFLAGS@ 
 
 if HAVE_PGSQL
 
--- origsrc/libpreludedb-1.0.1p1/plugins/sql/sqlite3/Makefile.am	2012-03-19 22:41:40.000000000 +0900
+++ src/libpreludedb-1.0.1p1/plugins/sql/sqlite3/Makefile.am	2012-12-17 07:19:06.313297400 +0900
@@ -1,5 +1,5 @@
 AM_CPPFLAGS=@PCFLAGS@ -I$(top_srcdir)/src/include -I$(top_srcdir)/libmissing -I$(top_builddir)/libmissing @LIBPRELUDE_CFLAGS@ @SQLITE3_CFLAGS@
-sqlite3_la_LDFLAGS = -module -avoid-version @LIBPRELUDE_LDFLAGS@
+sqlite3_la_LDFLAGS = -no-undefined -module -avoid-version @LIBPRELUDE_LDFLAGS@
 
 if HAVE_SQLITE3
 
