--- origsrc/asyncplusplus-1.0/CMakeLists.txt	2018-05-14 23:49:48.000000000 +0900
+++ src/asyncplusplus-1.0/CMakeLists.txt	2018-11-22 19:16:31.471568600 +0900
@@ -118,7 +118,7 @@ configure_package_config_file("${CMAKE_C
 )
 
 install(FILES "${PROJECT_BINARY_DIR}/Async++Config.cmake"
-	DESTINATION cmake
+	DESTINATION lib/cmake/async++
 )
 
 # Install the library and produce a CMake export script
@@ -130,13 +130,16 @@ install(TARGETS Async++
 	FRAMEWORK DESTINATION Frameworks
 )
 export(EXPORT Async++)
-install(EXPORT Async++ DESTINATION cmake)
+install(EXPORT Async++ DESTINATION lib/cmake/async++)
 if (APPLE AND BUILD_FRAMEWORK)
 	set_target_properties(Async++ PROPERTIES OUTPUT_NAME Async++ FRAMEWORK ON)
 	set_source_files_properties(${ASYNCXX_INCLUDE} PROPERTIES MACOSX_PACKAGE_LOCATION Headers/async++)
 	set_source_files_properties(${PROJECT_SOURCE_DIR}/include/async++.h PROPERTIES MACOSX_PACKAGE_LOCATION Headers)
 else()
 	set_target_properties(Async++ PROPERTIES OUTPUT_NAME async++)
+	if (CYGWIN)
+		set_target_properties(Async++ PROPERTIES SOVERSION 0)
+	endif()
 	target_include_directories(Async++ INTERFACE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include> $<INSTALL_INTERFACE:include>)
 	install(FILES ${PROJECT_SOURCE_DIR}/include/async++.h DESTINATION include)
 	install(FILES ${ASYNCXX_INCLUDE} DESTINATION include/async++)
@@ -146,3 +149,10 @@ SET(CPACK_GENERATOR "DEB")
 SET(CPACK_DEBIAN_PACKAGE_MAINTAINER "none") #required
 
 INCLUDE(CPack)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/async++.pc.in ${CMAKE_BINARY_DIR}/async++.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/async++.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/asyncplusplus-1.0/async++.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/asyncplusplus-1.0/async++.pc.in	2018-11-22 19:16:47.666195400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: asyncplusplus
+Description: Async++ concurrency framework for C++11
+Version: @VERSION@
+Libs: -L${libdir} -lasync++
+Cflags: -I${includedir}
