--- origsrc/tpm2-abrmd-2.3.2/Makefile.am	2020-05-19 04:03:23.000000000 +0900
+++ src/tpm2-abrmd-2.3.2/Makefile.am	2020-08-11 15:23:41.344445300 +0900
@@ -278,7 +278,7 @@ test_integration_libtest_la_SOURCES = \
 
 src_libtss2_tcti_tabrmd_la_LIBADD = $(GIO_LIBS) $(GLIB_LIBS) $(TSS2_MU_LIBS) \
     $(libutil) $(TSS2_TCTILDR_LIBS)
-src_libtss2_tcti_tabrmd_la_LDFLAGS = -fPIC -Wl,--no-undefined -Wl,-z,nodelete -Wl,--version-script=$(srcdir)/src/tcti-tabrmd.map
+src_libtss2_tcti_tabrmd_la_LDFLAGS = -fPIC -no-undefined -Wl,--version-script=$(srcdir)/src/tcti-tabrmd.map
 src_libtss2_tcti_tabrmd_la_SOURCES = src/tcti-tabrmd.c src/tcti-tabrmd-priv.h $(srcdir)/src/tcti-tabrmd.map
 
 src_tpm2_abrmd_LDADD   = $(GIO_LIBS) $(GLIB_LIBS) $(PTHREAD_LIBS) \
--- origsrc/tpm2-abrmd-2.3.2/src/tcti-tabrmd.c	2020-05-19 04:03:23.000000000 +0900
+++ src/tpm2-abrmd-2.3.2/src/tcti-tabrmd.c	2020-08-11 15:23:41.575489400 +0900
@@ -123,7 +123,7 @@ gerror_code_to_tcti_rc (int error_number
     }
 }
 
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD__) || defined(__CYGWIN__)
 #ifndef POLLRDHUP
 #define POLLRDHUP 0x0
 #endif
