--- origsrc/tpm2-abrmd-3.0.0/Makefile.am	2022-12-06 00:30:45.000000000 +0900
+++ src/tpm2-abrmd-3.0.0/Makefile.am	2024-04-17 20:01:29.012826500 +0900
@@ -282,7 +282,7 @@ src_libtss2_tcti_tabrmd_la_LIBADD = $(GI
 if !ENABLE_ASAN
 UNDEFINED_SYMS := -Wl,--no-undefined
 endif
-src_libtss2_tcti_tabrmd_la_LDFLAGS = -fPIC $(UNDEFINED_SYMS) -Wl,-z,nodelete -Wl,--version-script=$(srcdir)/src/tcti-tabrmd.map
+src_libtss2_tcti_tabrmd_la_LDFLAGS = -no-undefined -fPIC $(UNDEFINED_SYMS) -Wl,--version-script=$(srcdir)/src/tcti-tabrmd.map
 src_libtss2_tcti_tabrmd_la_SOURCES = src/tcti-tabrmd.c src/tcti-tabrmd-priv.h $(srcdir)/src/tcti-tabrmd.map
 
 src_tpm2_abrmd_LDADD   = $(GIO_LIBS) $(GLIB_LIBS) $(PTHREAD_LIBS) \
--- origsrc/tpm2-abrmd-3.0.0/src/tcti-tabrmd.c	2022-12-06 00:30:45.000000000 +0900
+++ src/tpm2-abrmd-3.0.0/src/tcti-tabrmd.c	2024-04-17 19:59:46.719259200 +0900
@@ -123,11 +123,9 @@ gerror_code_to_tcti_rc (int error_number
     }
 }
 
-#if defined(__FreeBSD__)
 #ifndef POLLRDHUP
 #define POLLRDHUP 0x0
 #endif
-#endif
 /*
  * This is a thin wrapper around a call to poll. It packages up the provided
  * file descriptor and timeout and polls on that same FD for data or a hangup.
