--- origsrc/librtlsdr-0.5.3/src/Makefile.am	2014-02-07 10:04:24.000000000 +0900
+++ src/librtlsdr-0.5.3/src/Makefile.am	2014-05-25 23:23:36.460187400 +0900
@@ -10,7 +10,7 @@ AM_CFLAGS = ${CFLAGS} -fPIC ${SYMBOL_VIS
 lib_LTLIBRARIES = librtlsdr.la
 
 librtlsdr_la_SOURCES = librtlsdr.c tuner_e4k.c tuner_fc0012.c tuner_fc0013.c tuner_fc2580.c tuner_r82xx.c
-librtlsdr_la_LDFLAGS = -version-info $(LIBVERSION)
+librtlsdr_la_LDFLAGS = -no-undefined -version-info $(LIBVERSION)
 
 bin_PROGRAMS         = rtl_sdr rtl_tcp rtl_test rtl_fm rtl_eeprom rtl_adsb rtl_power
 
