--- origsrc/libisofs-1.0.2/Makefile.am	2010-09-27 23:54:39.000000000 +0900
+++ src/libisofs-1.0.2/Makefile.am	2011-03-09 00:17:20.533443700 +0900
@@ -12,7 +12,7 @@ lib_LTLIBRARIES = libisofs/libisofs.la
 # Build libraries
 
 libisofs_libisofs_la_LDFLAGS = \
-	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
+	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -no-undefined
 
 # Eventually enabling system adapters for ACL and EA.
 # ts A90409: Eventually enabling use of zlib.
@@ -22,6 +22,8 @@ libisofs_libisofs_la_CFLAGS = $(LIBACL_D
 
 # ts A90114 : added aaip_0_2.*
 
+INCLUDES = -I$(top_builddir)/libisofs
+
 libisofs_libisofs_la_SOURCES = \
 	libisofs/builder.h \
 	libisofs/builder.c \
@@ -108,7 +110,7 @@ noinst_PROGRAMS = \
 
 # ts A90807
 # Consolidated demo code for having less linker mesages with a make run.
-demo_demo_CPPFLAGS = -Ilibisofs
+demo_demo_CPPFLAGS = -I$(top_srcdir)/libisofs
 demo_demo_LDADD = $(libisofs_libisofs_la_OBJECTS) $(libisofs_libisofs_la_LIBADD)
 demo_demo_SOURCES = demo/demo.c
 
--- origsrc/libisofs-1.0.2/configure.ac	2011-02-23 20:16:16.000000000 +0900
+++ src/libisofs-1.0.2/configure.ac	2011-03-08 23:16:42.461356400 +0900
@@ -89,7 +89,7 @@ dnl To abort configuration if iconv() st
 LIBBURNIA_ASSERT_ICONV
 
 
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_SUBST(LIBTOOL_DEPS)
 LIBTOOL="$LIBTOOL --silent"
 
