--- origsrc/libisofs-1.5.6/Makefile.am	2020-10-28 07:03:12.000000000 +0900
+++ src/libisofs-1.5.6/Makefile.am	2026-01-19 00:34:54.764143700 +0900
@@ -13,13 +13,13 @@ lib_LTLIBRARIES = libisofs/libisofs.la
 ACLOCAL_AMFLAGS = -I ./
 
 # Enable this if the source includes generated files like version.h
-# AM_CPPFLAGS = -I $(top_builddir)/libisofs
+AM_CPPFLAGS = -I $(top_builddir)/libisofs
 
 ## ========================================================================= ##
 
 # Build libraries
 
-libisofs_libisofs_la_LDFLAGS = \
+libisofs_libisofs_la_LDFLAGS = -no-undefined \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LIBLDFLAGS)
 
 # Eventually enabling system adapters for ACL and EA.
--- origsrc/libisofs-1.5.6/demo/demo.c	2016-07-22 21:40:46.000000000 +0900
+++ src/libisofs-1.5.6/demo/demo.c	2026-01-19 00:34:54.767743700 +0900
@@ -243,8 +243,10 @@ need_abs_path:;
 
 
 static const char * const optstring = "JRIL:b:hV:";
+#if !defined(__CYGWIN__)
 extern char *optarg;
 extern int optind;
+#endif
 
 void iso_usage(char **argv)
 {
