--- origsrc/rgf-3.12.0/FastRGF/CMakeLists.txt	2022-01-08 06:24:57.000000000 +0900
+++ src/rgf-3.12.0/FastRGF/CMakeLists.txt	2024-10-30 03:04:44.838306200 +0900
@@ -9,7 +9,7 @@ project (FastRGF)
 # whether to use OpenMP (default is ON)
 option(OPENMP "Use OpenMP for multithreading" ON)
 
-set(CMAKE_CXX_FLAGS "-O3 -std=c++11")
+#set(CMAKE_CXX_FLAGS "-O3 -std=c++11")
 #set(CMAKE_CXX_FLAGS "-g -std=c++11 -Wall")
 
 
--- origsrc/rgf-3.12.0/FastRGF/src/exe/CMakeLists.txt	2022-01-08 06:24:57.000000000 +0900
+++ src/rgf-3.12.0/FastRGF/src/exe/CMakeLists.txt	2024-10-30 03:41:18.907916800 +0900
@@ -21,9 +21,9 @@ target_link_libraries(discretized_traine
 add_executable(forest_train forest_train.cpp) 
 target_link_libraries(forest_train forest base)
 
-install(TARGETS forest_train DESTINATION ${CMAKE_SOURCE_DIR}/bin)
+install(TARGETS forest_train DESTINATION bin)
 
 add_executable(forest_predict forest_predict.cpp) 
 target_link_libraries(forest_predict forest base)
 
-install(TARGETS forest_predict DESTINATION ${CMAKE_SOURCE_DIR}/bin)
+install(TARGETS forest_predict DESTINATION bin)
--- origsrc/rgf-3.12.0/RGF/CMakeLists.txt	2022-01-08 06:24:57.000000000 +0900
+++ src/rgf-3.12.0/RGF/CMakeLists.txt	2024-10-30 03:42:26.103345600 +0900
@@ -1,6 +1,6 @@
 project(rgf)
-set(CMAKE_INSTALL_PREFIX ${CMAKE_CURRENT_SOURCE_DIR})
-set(EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_SOURCE_DIR}/bin)
+# set(CMAKE_INSTALL_PREFIX ${CMAKE_CURRENT_SOURCE_DIR})
+# set(EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_SOURCE_DIR}/bin)
 if(MINGW)
     set(CMAKE_EXE_LINKER_FLAGS ${CMAKE_EXE_LINKER_FLAGS} -static-libstdc++)
 endif()
