--- origsrc/hypre-2.24.0/src/CMakeLists.txt	2022-02-11 10:12:32.000000000 +0900
+++ src/hypre-2.24.0/src/CMakeLists.txt	2022-04-22 16:33:13.526957400 +0900
@@ -348,6 +348,10 @@ endif ()
 # Create the HYPRE library object
 add_library(${PROJECT_NAME})
 
+if (CYGWIN)
+  set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 # Headers and sources
 set(HYPRE_HEADERS "")
 
@@ -578,3 +582,12 @@ export(EXPORT HYPRETargets
 add_library(HYPRE::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
 
 export(PACKAGE ${PROJECT_NAME})
+
+if (CYGWIN)
+  set(prefix ${CMAKE_INSTALL_PREFIX})
+  set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+  set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+  set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+  configure_file(${CMAKE_SOURCE_DIR}/hypre.pc.in ${CMAKE_BINARY_DIR}/hypre.pc @ONLY)
+  install(FILES ${CMAKE_BINARY_DIR}/hypre.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+endif()
--- origsrc/hypre-2.24.0/src/hypre.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/hypre-2.24.0/src/hypre.pc.in	2022-04-22 16:32:16.551895900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: hypre
+Description: Parallel solvers for sparse linear systems featuring multigrid methods
+Version: @HYPRE_VERSION@
+Libs: -L${libdir} -lhypre
+Cflags: -I${includedir}
