--- origsrc/linbox-1.3.2/interfaces/driver/Makefile.am	2012-06-07 17:30:26.000000000 +0900
+++ src/linbox-1.3.2/interfaces/driver/Makefile.am	2012-10-08 21:32:27.846392500 +0900
@@ -46,7 +46,7 @@ liblbdriver_la_SOURCES= lb-element.C 	\
 # \
 			#  lb-solve.C
 
-liblbdriver_la_LDFLAGS=$(GIVARO_LIBS) $(GMP_LIBS) $(NTL_LIBS) $(BLAS_LIBS) $(LDFLAGS) $(top_srcdir)/linbox/liblinbox.la -Wl,-zmuldefs
+liblbdriver_la_LDFLAGS=$(GIVARO_LIBS) $(GMP_LIBS) $(NTL_LIBS) $(BLAS_LIBS) $(LDFLAGS) $(top_srcdir)/linbox/liblinbox.la -Wl,-zmuldefs -no-undefined
 
 
 
--- origsrc/linbox-1.3.2/interfaces/maple/Makefile.am	2012-06-07 17:30:26.000000000 +0900
+++ src/linbox-1.3.2/interfaces/maple/Makefile.am	2012-10-08 21:32:45.605008900 +0900
@@ -30,7 +30,7 @@ lib_LTLIBRARIES=liblbmaple.la
 
 liblbmaple_la_SOURCES= lb-maple.C
 
-liblbmaple_la_LDFLAGS=$(GIVARO_LIBS) $(GMP_LIBS) $(NTL_LIBS) $(BLAS_LIBS) $(MAPLE_LIBS) $(top_srcdir)/interfaces/driver/liblbdriver.la -Wl,-zmuldefs $(LDFLAGS)
+liblbmaple_la_LDFLAGS=$(GIVARO_LIBS) $(GMP_LIBS) $(NTL_LIBS) $(BLAS_LIBS) $(MAPLE_LIBS) $(top_srcdir)/interfaces/driver/liblbdriver.la -Wl,-zmuldefs $(LDFLAGS) -no-undefined
 
 pkginclude_HEADERS =lb-maple.h lb-maple-utilities.h
 
--- origsrc/linbox-1.3.2/interfaces/sage/Makefile.am	2012-06-07 17:30:25.000000000 +0900
+++ src/linbox-1.3.2/interfaces/sage/Makefile.am	2012-10-08 21:33:07.133009300 +0900
@@ -35,5 +35,5 @@ pkginclude_HEADERS = linbox-sage.h
 #gentoo's linbox-1.1.6-fix-undefined-symbols.patch
 liblinboxsage_la_LIBADD = $(top_builddir)/linbox/liblinbox.la
 
-liblinboxsage_la_LDFLAGS =  $(GIVARO_LIBS) $(GMP_LIBS) $(NTL_LIBS) $(BLAS_LIBS) $(MAPLE_LIBS) $(LDFLAGS) -version-info 0:0:0 #-Wl,-zmuldefs
+liblinboxsage_la_LDFLAGS =  $(GIVARO_LIBS) $(GMP_LIBS) $(NTL_LIBS) $(BLAS_LIBS) $(MAPLE_LIBS) $(LDFLAGS) -version-info 0:0:0 -no-undefined
 endif
--- origsrc/linbox-1.3.2/linbox/Makefile.am	2012-06-07 17:30:31.000000000 +0900
+++ src/linbox-1.3.2/linbox/Makefile.am	2012-10-08 21:34:00.976409000 +0900
@@ -65,4 +65,7 @@ liblinbox_la_SOURCES = dummy.C
 liblinbox_la_LIBADD =           \
     util/libutil.la             \
     randiter/libranditer.la     \
-    algorithms/libalgorithms.la
+    algorithms/libalgorithms.la	\
+    $(BLAS_LIBS)
+
+liblinbox_la_LDFLAGS = -no-undefined
--- origsrc/linbox-1.3.2/linbox/algorithms/Makefile.am	2012-06-07 23:24:14.000000000 +0900
+++ src/linbox-1.3.2/linbox/algorithms/Makefile.am	2012-10-08 21:34:32.921261500 +0900
@@ -28,6 +28,7 @@ noinst_LTLIBRARIES=libalgorithms.la
 libalgorithms_la_SOURCES= diophantine-solver.C
 
 #  AM_CPPFLAGS= $(CBLAS_FLAG) $(GMP_CFLAGS) $(NTL_CFLAGS) $(GIVARO_CFLAGS) $(LIDIA_CFLAGS)
+AM_CPPFLAGS = -I$(top_srcdir)
 
 AM_CXXFLAGS = @DEFAULT_CFLAGS@ -DDISABLE_COMMENTATOR $(CBLAS_FLAG) $(FFLAS_FFPACK_CFLAGS) $(GMP_CFLAGS) $(NTL_CFLAGS) $(ATLAS_CFLAGS) $(GIVARO_CFLAGS) $(LIDIA_CFLAGS) $(FPLLL_CFLAGS)
 
--- origsrc/linbox-1.3.2/linbox/randiter/Makefile.am	2012-06-07 17:30:31.000000000 +0900
+++ src/linbox-1.3.2/linbox/randiter/Makefile.am	2012-10-08 21:35:51.056781500 +0900
@@ -19,7 +19,7 @@
 
 
 
-AM_CPPFLAGS=-I$(top_srcdir)/linbox
+AM_CPPFLAGS=-I$(top_srcdir)/linbox -I$(top_srcdir)
 pkgincludesubdir=$(pkgincludedir)/randiter
 
 AM_CXXFLAGS = @DEFAULT_CFLAGS@ -DDISABLE_COMMENTATOR $(GMP_CFLAGS) $(NTL_CFLAGS) $(ATLAS_CFLAGS) $(GIVARO_CFLAGS)  $(FFLAS_FFPACK_CFLAGS)
--- origsrc/linbox-1.3.2/tests/Makefile.am	2012-06-08 17:27:18.000000000 +0900
+++ src/linbox-1.3.2/tests/Makefile.am	2012-10-08 21:36:31.919878200 +0900
@@ -22,7 +22,7 @@
 
 AM_CPPFLAGS=-I$(top_srcdir) -I$(top_srcdir)/linbox
 # AM_LDFLAGS = -llapack -lcblas -latlas
-AM_LDFLAGS = $(top_srcdir)/linbox/.libs/liblinbox.a
+AM_LDFLAGS = $(top_builddir)/linbox/liblinbox.la
 #  SAFER_FLAGS = -g -Wall -Wextra -Wno-unused-parameter
 # bds: no need to optimize for unit/regression tests, and opt adds a lot to compilation time
 #AM_CXXFLAGS = @TESTS_CFLAGS@ -DLinBoxSrcOnly
