--- origsrc/classias-1.1/configure.in	2009-12-27 23:20:03.000000000 +0900
+++ src/classias-1.1/configure.in	2015-05-15 18:53:57.500866200 +0900
@@ -144,8 +144,7 @@ CPPFLAGS="${INCLUDES} ${CPPFLAGS}"
 AC_CHECK_LIB(m, rand)
 AC_CHECK_LIB(pthread, pthread_create)
 
-AC_CHECK_HEADERS(lbfgs.h)
-AC_CHECK_LIB(lbfgs, lbfgs)
+PKG_CHECK_MODULES([LIBLBFGS], [liblbfgs])
 
 AC_CHECK_HEADERS(tr1/unordered_map)
 AC_CHECK_HEADERS(boost/unordered_map.hpp)
--- origsrc/classias-1.1/frontend/tag/Makefile.am	2009-11-13 12:36:22.000000000 +0900
+++ src/classias-1.1/frontend/tag/Makefile.am	2015-05-15 18:58:39.894743200 +0900
@@ -18,6 +18,5 @@ classias_tag_SOURCES = \
 	main.cpp
 
 AM_CXXFLAGS = @CXXFLAGS@
-INCLUDES = @INCLUDES@ -I../contrib -I../include
+INCLUDES = @INCLUDES@ -I$(top_srcdir)/frontend/contrib -I$(top_srcdir)/frontend/include
 AM_LDFLAGS = @LDFLAGS@
-
--- origsrc/classias-1.1/frontend/train/Makefile.am	2009-09-26 00:15:28.000000000 +0900
+++ src/classias-1.1/frontend/train/Makefile.am	2015-05-15 18:58:07.440449700 +0900
@@ -20,7 +20,8 @@ classias_train_SOURCES = \
 	candidate.cpp \
 	main.cpp
 
-AM_CXXFLAGS = @CXXFLAGS@
-INCLUDES = @INCLUDES@ -I../contrib -I../include
-AM_LDFLAGS = @LDFLAGS@
+classias_train_LDADD = $(LIBLBFGS_LIBS)
 
+AM_CXXFLAGS = @CXXFLAGS@ -fpermissive
+INCLUDES = @INCLUDES@ -I$(top_srcdir)/frontend/contrib -I$(top_srcdir)/frontend/include
+AM_LDFLAGS = @LDFLAGS@
