--- origsrc/apertium-3.6.3/Makefile.am	2020-07-01 23:13:02.000000000 +0900
+++ src/apertium-3.6.3/Makefile.am	2020-07-23 18:33:53.366363900 +0900
@@ -19,7 +19,7 @@ EXTRA_DIST=autogen.sh README-MODES apert
 
 install-data-local:
 	mkdir -p $(DESTDIR)$(modesdir)
-	$(INSTALL_DATA) README-MODES  $(DESTDIR)$(modesdir)/README
+	$(INSTALL_DATA) $(top_srcdir)/README-MODES  $(DESTDIR)$(modesdir)/README
 
 # just a synonym:
 test: check
--- origsrc/apertium-3.6.3/apertium/Makefile.am	2020-07-01 23:13:02.000000000 +0900
+++ src/apertium-3.6.3/apertium/Makefile.am	2020-07-23 18:45:03.181145200 +0900
@@ -184,7 +184,7 @@ GENERATEDSCRIPTS = apertium-gen-deformat
 
 lib_LTLIBRARIES = libapertium3.la
 libapertium3_la_SOURCES = $(h_sources) $(cc_sources)
-libapertium3_la_LDFLAGS = -version-info $(GENERIC_LIBRARY_VERSION) -release $(GENERIC_RELEASE)
+libapertium3_la_LDFLAGS = -no-undefined -version-info $(GENERIC_LIBRARY_VERSION) -release $(GENERIC_RELEASE)
 
 bin_PROGRAMS = apertium-cleanstream \
 	       apertium-deshtml \
@@ -324,7 +324,7 @@ apertium_adapt_docx_LDADD = $(APERTIUM_L
 if WINDOWS
 INCLUDES = -I$(top_srcdir)/utf8 -I$(top_srcdir)/apertium/win32 -I$(top_srcdir) $(APERTIUM_CFLAGS) $(ICU_CFLAGS)
 else
-AM_CPPFLAGS = -I$(top_srcdir) $(APERTIUM_CFLAGS) $(ICU_CFLAGS)
+AM_CPPFLAGS = -I$(top_builddir) -I$(top_srcdir) $(APERTIUM_CFLAGS) $(ICU_CFLAGS)
 endif
 CLEANFILES = *~ apertium_destxt.cc apertium_retxt.cc apertium_deshtml.cc \
              apertium_rehtml.cc apertium_desrtf.cc apertium_rertf.cc \
@@ -340,183 +340,183 @@ CLEANFILES = *~ apertium_destxt.cc apert
              apertium_prelatex.cc apertium_postlatex.cc \
              $(GENERATEDSCRIPTS)
 
-apertium_destxt.cc: txt-format.xml Makefile.am deformat.xsl
-	$(XSLTPROC) deformat.xsl txt-format.xml >$@tmp
+apertium_destxt.cc: $(srcdir)/txt-format.xml Makefile.am $(srcdir)/deformat.xsl
+	$(XSLTPROC) $(srcdir)/deformat.xsl $(srcdir)/txt-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_desxpresstag.cc: xpresstag-format.xml Makefile.am deformat.xsl
-	$(XSLTPROC) deformat.xsl xpresstag-format.xml >$@tmp
+apertium_desxpresstag.cc: $(srcdir)/xpresstag-format.xml Makefile.am $(srcdir)/deformat.xsl
+	$(XSLTPROC) $(srcdir)/deformat.xsl $(srcdir)/xpresstag-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_rexpresstag.cc: xpresstag-format.xml Makefile.am reformat.xsl
-	$(XSLTPROC) reformat.xsl xpresstag-format.xml >$@tmp
+apertium_rexpresstag.cc: $(srcdir)/xpresstag-format.xml Makefile.am $(srcdir)/reformat.xsl
+	$(XSLTPROC) $(srcdir)/reformat.xsl $(srcdir)/xpresstag-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_retxt.cc: txt-format.xml Makefile.am reformat.xsl
-	$(XSLTPROC) reformat.xsl txt-format.xml >$@tmp
+apertium_retxt.cc: $(srcdir)/txt-format.xml Makefile.am $(srcdir)/reformat.xsl
+	$(XSLTPROC) $(srcdir)/reformat.xsl $(srcdir)/txt-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_deshtml.cc: html-format.xml Makefile.am deformat.xsl
-	$(XSLTPROC) deformat.xsl html-format.xml >$@tmp
+apertium_deshtml.cc: $(srcdir)/html-format.xml Makefile.am $(srcdir)/deformat.xsl
+	$(XSLTPROC) $(srcdir)/deformat.xsl $(srcdir)/html-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_deshtml_alt.cc: html-alt-format.xml Makefile.am deformat.xsl
-	$(XSLTPROC) deformat.xsl $< >$@tmp
+apertium_deshtml_alt.cc: $(srcdir)/html-alt-format.xml Makefile.am $(srcdir)/deformat.xsl
+	$(XSLTPROC) $(srcdir)/deformat.xsl $< >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_rehtml.cc: html-format.xml Makefile.am reformat.xsl
-	$(XSLTPROC) reformat.xsl html-format.xml >$@tmp
+apertium_rehtml.cc: $(srcdir)/html-format.xml Makefile.am $(srcdir)/reformat.xsl
+	$(XSLTPROC) $(srcdir)/reformat.xsl $(srcdir)/html-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_rehtml_noent.cc: html-noent-format.xml Makefile.am reformat.xsl
-	$(XSLTPROC) reformat.xsl html-noent-format.xml >$@tmp
+apertium_rehtml_noent.cc: $(srcdir)/html-noent-format.xml Makefile.am $(srcdir)/reformat.xsl
+	$(XSLTPROC) $(srcdir)/reformat.xsl $(srcdir)/html-noent-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_desodt.cc: odt-format.xml Makefile.am deformat.xsl
-	$(XSLTPROC) deformat.xsl odt-format.xml >$@tmp
+apertium_desodt.cc: $(srcdir)/odt-format.xml Makefile.am $(srcdir)/deformat.xsl
+	$(XSLTPROC) $(srcdir)/deformat.xsl $(srcdir)/odt-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_reodt.cc: odt-format.xml Makefile.am reformat.xsl
-	$(XSLTPROC) reformat.xsl odt-format.xml >$@tmp
+apertium_reodt.cc: $(srcdir)/odt-format.xml Makefile.am $(srcdir)/reformat.xsl
+	$(XSLTPROC) $(srcdir)/reformat.xsl $(srcdir)/odt-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_desrtf.cc: rtf-format.xml Makefile.am deformat.xsl
-	$(XSLTPROC) deformat.xsl rtf-format.xml >$@tmp
+apertium_desrtf.cc: $(srcdir)/rtf-format.xml Makefile.am $(srcdir)/deformat.xsl
+	$(XSLTPROC) $(srcdir)/deformat.xsl $(srcdir)/rtf-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_rertf.cc: rtf-format.xml Makefile.am reformat.xsl
-	$(XSLTPROC) reformat.xsl rtf-format.xml >$@tmp
+apertium_rertf.cc: $(srcdir)/rtf-format.xml Makefile.am $(srcdir)/reformat.xsl
+	$(XSLTPROC) $(srcdir)/reformat.xsl $(srcdir)/rtf-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_deswxml.cc: wxml-format.xml Makefile.am deformat.xsl
-	$(XSLTPROC) deformat.xsl wxml-format.xml >$@tmp
+apertium_deswxml.cc: $(srcdir)/wxml-format.xml Makefile.am $(srcdir)/deformat.xsl
+	$(XSLTPROC) $(srcdir)/deformat.xsl $(srcdir)/wxml-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_rewxml.cc: wxml-format.xml Makefile.am reformat.xsl
-	$(XSLTPROC) reformat.xsl wxml-format.xml >$@tmp
+apertium_rewxml.cc: $(srcdir)/wxml-format.xml Makefile.am $(srcdir)/reformat.xsl
+	$(XSLTPROC) $(srcdir)/reformat.xsl $(srcdir)/wxml-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_deslatex.cc: latex-format.xml Makefile.am deformat.xsl
-	$(XSLTPROC) deformat.xsl latex-format.xml >$@tmp
+apertium_deslatex.cc: $(srcdir)/latex-format.xml Makefile.am $(srcdir)/deformat.xsl
+	$(XSLTPROC) $(srcdir)/deformat.xsl $(srcdir)/latex-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_relatex.cc: latex-format.xml Makefile.am reformat.xsl
-	$(XSLTPROC) reformat.xsl latex-format.xml >$@tmp
+apertium_relatex.cc: $(srcdir)/latex-format.xml Makefile.am $(srcdir)/reformat.xsl
+	$(XSLTPROC) $(srcdir)/reformat.xsl $(srcdir)/latex-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
 
 
-apertium_desxlsx.cc: xlsx-format.xml Makefile.am deformat.xsl
-	$(XSLTPROC) deformat.xsl xlsx-format.xml >$@tmp
+apertium_desxlsx.cc: $(srcdir)/xlsx-format.xml Makefile.am $(srcdir)/deformat.xsl
+	$(XSLTPROC) $(srcdir)/deformat.xsl $(srcdir)/xlsx-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_rexlsx.cc: xlsx-format.xml Makefile.am reformat.xsl
-	$(XSLTPROC) reformat.xsl xlsx-format.xml >$@tmp
+apertium_rexlsx.cc: $(srcdir)/xlsx-format.xml Makefile.am $(srcdir)/reformat.xsl
+	$(XSLTPROC) $(srcdir)/reformat.xsl $(srcdir)/xlsx-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_despptx.cc: pptx-format.xml Makefile.am deformat.xsl
-	$(XSLTPROC) deformat.xsl pptx-format.xml >$@tmp
+apertium_despptx.cc: $(srcdir)/pptx-format.xml Makefile.am $(srcdir)/deformat.xsl
+	$(XSLTPROC) $(srcdir)/deformat.xsl $(srcdir)/pptx-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_repptx.cc: pptx-format.xml Makefile.am reformat.xsl
-	$(XSLTPROC) reformat.xsl pptx-format.xml >$@tmp
+apertium_repptx.cc: $(srcdir)/pptx-format.xml Makefile.am $(srcdir)/reformat.xsl
+	$(XSLTPROC) $(srcdir)/reformat.xsl $(srcdir)/pptx-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_desmediawiki.cc: mediawiki-format.xml Makefile.am deformat.xsl
-	$(XSLTPROC) deformat.xsl mediawiki-format.xml >$@tmp
+apertium_desmediawiki.cc: $(srcdir)/mediawiki-format.xml Makefile.am $(srcdir)/deformat.xsl
+	$(XSLTPROC) $(srcdir)/deformat.xsl $(srcdir)/mediawiki-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_remediawiki.cc: mediawiki-format.xml Makefile.am reformat.xsl
-	$(XSLTPROC) reformat.xsl mediawiki-format.xml >$@tmp
+apertium_remediawiki.cc: $(srcdir)/mediawiki-format.xml Makefile.am $(srcdir)/reformat.xsl
+	$(XSLTPROC) $(srcdir)/reformat.xsl $(srcdir)/mediawiki-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_prelatex.cc: apertium-prelatex.l
-	$(FLEX) -Cfer -o$@ apertium-prelatex.l
+apertium_prelatex.cc: $(srcdir)/apertium-prelatex.l
+	$(FLEX) -Cfer -o$@ $(srcdir)/apertium-prelatex.l
 
-apertium_postlatex.cc: apertium-postlatex.l
-	$(FLEX) -Cfer -o$@ apertium-postlatex.l
+apertium_postlatex.cc: $(srcdir)/apertium-postlatex.l
+	$(FLEX) -Cfer -o$@ $(srcdir)/apertium-postlatex.l
 
-apertium_postlatex_raw.cc: apertium-postlatex-raw.l
-	$(FLEX) -Cfer -o$@ apertium-postlatex-raw.l
+apertium_postlatex_raw.cc: $(srcdir)/apertium-postlatex-raw.l
+	$(FLEX) -Cfer -o$@ $(srcdir)/apertium-postlatex-raw.l
 
-apertium-validate-tagger: Makefile.am validate-header.sh
+apertium-validate-tagger: Makefile.am $(srcdir)/validate-header.sh
 	@echo "Creating apertium-validate-tagger script"
 	@echo "#!$(BASH)" > $@
-	@cat validate-header.sh >> $@
+	@cat $(srcdir)/validate-header.sh >> $@
 	@echo "$(XMLLINT) --dtdvalid \"$(apertiumdir)\"/tagger.dtd --noout \"\$$FILE1\"" >>$@
 	@chmod a+x $@
 
-apertium-validate-transfer: Makefile.am validate-header.sh
+apertium-validate-transfer: Makefile.am $(srcdir)/validate-header.sh
 	@echo "Creating apertium-validate-transfer script"
 	@echo "#!$(BASH)" > $@
-	@cat validate-header.sh >> $@
+	@cat $(srcdir)/validate-header.sh >> $@
 	@echo "$(XMLLINT) --dtdvalid \"$(apertiumdir)\"/transfer.dtd --noout \"\$$FILE1\"" >>$@
 	@chmod a+x $@
 
-apertium-validate-interchunk: Makefile.am validate-header.sh
+apertium-validate-interchunk: Makefile.am $(srcdir)/validate-header.sh
 	@echo "Creating apertium-validate-interchunk script"
 	@echo "#!$(BASH)" > $@
-	@cat validate-header.sh >> $@
+	@cat $(srcdir)/validate-header.sh >> $@
 	@echo "$(XMLLINT) --dtdvalid \"$(apertiumdir)\"/interchunk.dtd --noout \"\$$FILE1\"" >>$@
 	@chmod a+x $@
 
-apertium-validate-postchunk: Makefile.am validate-header.sh
+apertium-validate-postchunk: Makefile.am $(srcdir)/validate-header.sh
 	@echo "Creating apertium-validate-postchunk script"
 	@echo "#!$(BASH)" > $@
-	@cat validate-header.sh >> $@
+	@cat $(srcdir)/validate-header.sh >> $@
 	@echo "$(XMLLINT) --dtdvalid \"$(apertiumdir)\"/postchunk.dtd --noout \"\$$FILE1\"" >>$@
 	@chmod a+x $@
 
-apertium-validate-acx: Makefile.am validate-header.sh
+apertium-validate-acx: Makefile.am $(srcdir)/validate-header.sh
 	@echo "Creating apertium-validate-acx script"
 	@echo "#!$(BASH)" > $@
-	@cat validate-header.sh >> $@
+	@cat $(srcdir)/validate-header.sh >> $@
 	@echo "$(XMLLINT) --relaxng \"$(prefix)\"/share/lttoolbox/acx.rng --schema \"$(prefix)\"/share/lttoolbox/acx.xsd --noout \"\$$FILE1\"" >>$@
 	@chmod a+x $@
 
-apertium-validate-modes: Makefile.am validate-header.sh
+apertium-validate-modes: Makefile.am $(srcdir)/validate-header.sh
 	@echo "Creating apertium-validate-modes script"
 	@echo "#!$(BASH)" > $@
-	@cat validate-header.sh >> $@
+	@cat $(srcdir)/validate-header.sh >> $@
 	@echo "$(XMLLINT) --dtdvalid \"$(apertiumdir)\"/modes.dtd --noout \"\$$FILE1\"" >>$@
 	@chmod a+x $@
 
 
-apertium-validate-dictionary: Makefile.am validate-header.sh
+apertium-validate-dictionary: Makefile.am $(srcdir)/validate-header.sh
 	@echo "Creating apertium-validate-dictionary script"
 	@echo "#!$(BASH)" > $@
-	@cat validate-header.sh >> $@
+	@cat $(srcdir)/validate-header.sh >> $@
 	@echo "# xsd is a non-final command, so just treated as a warning when compiling:" >> $@
 	@echo "$(XMLLINT) --schema \"$(prefix)\"/share/lttoolbox/dix.xsd --noout \"\$$FILE1\" 2>&1 | grep -ve ' validates$$' -e ' fails to validate$$'" >> $@
 	@echo "$(XMLLINT) --dtdvalid \"$(prefix)\"/share/lttoolbox/dix.dtd --noout \"\$$FILE1\"" >> $@
 	@chmod a+x $@
 
-apertium-gen-deformat: Makefile.am deformat-header.sh
+apertium-gen-deformat: Makefile.am $(srcdir)/deformat-header.sh
 	@echo "Creating apertium-gen-deformat script"
 	@echo "#!$(BASH)" > $@
-	@cat deformat-header.sh >> $@
+	@cat $(srcdir)/deformat-header.sh >> $@
 	@echo "$(XMLLINT) --dtdvalid \"$(apertiumdir)\"/format.dtd --noout \"\$$FILE1\" && \\" >> $@
 	@if [ `basename $(XSLTPROC)` == xsltproc ]; \
 	  then echo "$(XSLTPROC) --stringparam mode \$$MODE \"$(apertiumdir)\"/deformat.xsl \"\$$FILE1\" >/tmp/\$$\$$.deformat.l && \\"; \
@@ -527,10 +527,10 @@ apertium-gen-deformat: Makefile.am defor
 	@echo "rm /tmp/\$$\$$.deformat.l /tmp/\$$\$$.lex.cc" >> $@
 	@chmod a+x $@
 
-apertium-gen-reformat: Makefile.am gen-header.sh
+apertium-gen-reformat: Makefile.am $(srcdir)/gen-header.sh
 	@echo "Creating apertium-gen-reformat script"
 	@echo "#!$(BASH)" > $@
-	@cat gen-header.sh >> $@
+	@cat $(srcdir)/gen-header.sh >> $@
 	@echo "$(XMLLINT) --dtdvalid \"$(apertiumdir)\"/format.dtd --noout \"\$$FILE1\" && \\" >> $@
 	@echo "$(XSLTPROC) \"$(apertiumdir)\"/reformat.xsl \"\$$FILE1\" >/tmp/\$$\$$.reformat.l && \\" >> $@
 	@echo "$(FLEX) \$$FLEXOPTS -o/tmp/\$$\$$.lex.cc /tmp/\$$\$$.reformat.l && \\" >> $@
@@ -538,7 +538,7 @@ apertium-gen-reformat: Makefile.am gen-h
 	@echo "rm /tmp/\$$\$$.reformat.l /tmp/\$$\$$.lex.cc" >> $@
 	@chmod a+x $@
 
-apertium-gen-modes: apertium-gen-modes.in Makefile.am
+apertium-gen-modes: $(srcdir)/apertium-gen-modes.in Makefile.am
 	@echo "#!$(BASH)" > $@
 	@echo "APERTIUMDIR=$(apertiumdir)" >> $@
 	@echo "XMLLINT=$(XMLLINT)" >> $@
@@ -546,25 +546,25 @@ apertium-gen-modes: apertium-gen-modes.i
 	@cat $< >> $@
 	@chmod a+x $@
 
-apertium-utils-fixlatex: Makefile.am utils-fixlatex-header.sh
+apertium-utils-fixlatex: Makefile.am $(srcdir)/utils-fixlatex-header.sh
 	@echo "Creating apertium-utils-fixlatex script"
 	@echo "#!$(BASH)" > $@
-	@cat utils-fixlatex-header.sh >> $@
+	@cat $(srcdir)/utils-fixlatex-header.sh >> $@
 	@chmod a+x $@
 
-apertium: Makefile.am apertium-header.sh
+apertium: Makefile.am $(srcdir)/apertium-header.sh
 	@echo 'Creating apertium script'
 	@echo '#!$(BASH)' > $@
 	@echo 'APERTIUM_PATH="$${APERTIUM_PATH:-$(prefix)/bin}"' >>$@
 	@echo 'DEFAULT_DIRECTORY="$(prefix)/share/apertium"' >>$@
-	@cat apertium-header.sh >>$@
+	@cat $(srcdir)/apertium-header.sh >>$@
 	@chmod a+x $@
 
-apertium-unformat: Makefile.am apertium-unformat-header.sh
+apertium-unformat: Makefile.am $(srcdir)/apertium-unformat-header.sh
 	@echo "Creating apertium-unformat script"
 	@echo "#!$(BASH)" > $@
 	@echo "APERTIUM_PATH=\"$(prefix)/bin\"" >>$@
-	@cat apertium-unformat-header.sh >>$@
+	@cat $(srcdir)/apertium-unformat-header.sh >>$@
 	@chmod a+x $@
 
 man_MANS=apertium.1 apertium-deshtml.1 apertium-desrtf.1 apertium-destxt.1 \
--- origsrc/apertium-3.6.3/configure.ac	2020-07-01 23:13:02.000000000 +0900
+++ src/apertium-3.6.3/configure.ac	2020-07-23 18:45:44.269994900 +0900
@@ -161,7 +161,7 @@ AC_CHECK_DECLS([fread_unlocked, fwrite_u
 AC_CHECK_FUNCS([setlocale strdup getopt snprintf mbtowc])
 AC_REPLACE_FUNCS(getopt_long)
 
-AM_CONDITIONAL([WINDOWS], [test x$version_type = xwindows])
+AM_CONDITIONAL([WINDOWS], [test x$version_type = xwindows -a x$host_os != xcygwin])
 #AS_IF([test x$version_type = xwindows], [AC_DEFINE(HAVE_GETOPT_LONG,0)], [])
 
 # Checks for highest supported C++ standard
--- origsrc/apertium-3.6.3/python/Makefile.am	2020-07-01 23:13:02.000000000 +0900
+++ src/apertium-3.6.3/python/Makefile.am	2020-07-23 19:07:57.834246700 +0900
@@ -2,7 +2,7 @@ SWIG_INTERFACE = apertium_core.i
 
 BUILT_SOURCES = %_wrap.cpp
 
-%_wrap.cpp: $(SWIG_INTERFACE) setup.py
+%_wrap.cpp: $(srcdir)/$(SWIG_INTERFACE) setup.py
 	$(PYTHON) setup.py build
 
 install-exec-local:
--- origsrc/apertium-3.6.3/python/setup.py.in	2020-07-01 23:13:02.000000000 +0900
+++ src/apertium-3.6.3/python/setup.py.in	2020-07-23 19:29:15.877977500 +0900
@@ -18,7 +18,7 @@ class CustomBuild(build):
 
 
 def get_sources():
-    sources = ['apertium_core.i']
+    sources = ['../python/apertium_core.i']
     cc_sources = [
                   # interchunk.cc postchunk.cc transfer.cc
                   'apertium_re.cc', 'interchunk.cc', 'interchunk_word.cc', 'postchunk.cc', 'string_utils.cc', 'transfer.cc',
@@ -45,13 +45,13 @@ def get_sources():
 def get_include_dirs():
     # Remove '-I' from Flags, as python add '-I' on its own
     dirs = '@APERTIUM_CFLAGS@'.replace('-I', '').split()
-    return dirs + ['..']
+    return dirs + ['..', '../apertium', '@top_srcdir@']
 
 
 apertium_core_module = Extension(
     name='_apertium_core',
     sources=get_sources(),
-    swig_opts=['-c++', '-I..', '-I@top_srcdir@/apertium', '-Wall']+'@APERTIUM_CFLAGS@'.split(),
+    swig_opts=['-c++', '-I@top_builddir@', '-I@top_srcdir@', '-I@top_srcdir@/apertium', '-Wall']+'@APERTIUM_CFLAGS@'.split(),
     include_dirs=get_include_dirs(),
     library_dirs=['/usr/include/libxml2', '/usr/local/lib'],
     extra_compile_args='@CPPFLAGS@'.split()+'@CXXFLAGS@'.split(),
