--- origsrc/apertium-3.9.12/Makefile.am	2025-03-26 18:26:26.000000000 +0900
+++ src/apertium-3.9.12/Makefile.am	2025-05-25 02:14:39.711533900 +0900
@@ -19,7 +19,7 @@ EXTRA_DIST=autogen.sh README-MODES apert
 
 install-data-local:
 	mkdir -p $(DESTDIR)$(modesdir)
-	$(INSTALL_DATA) README-MODES  $(DESTDIR)$(modesdir)/README
+	$(INSTALL_DATA) $(top_srcdir)/README-MODES  $(DESTDIR)$(modesdir)/README
 
 # just a synonym:
 test: check
--- origsrc/apertium-3.9.12/apertium/Makefile.am	2025-03-26 18:26:26.000000000 +0900
+++ src/apertium-3.9.12/apertium/Makefile.am	2025-05-25 02:21:53.621427600 +0900
@@ -160,7 +160,7 @@ GENERATEDSCRIPTS = apertium-gen-deformat
 
 lib_LTLIBRARIES = libapertium.la
 libapertium_la_SOURCES = $(h_sources) $(cc_sources)
-libapertium_la_LDFLAGS = -version-info $(VERSION_ABI)
+libapertium_la_LDFLAGS = -no-undefined -version-info $(VERSION_ABI)
 
 bin_PROGRAMS = apertium-cleanstream \
 	       apertium-deshtml \
@@ -227,7 +227,7 @@ apertium_DATA = deformat.xsl reformat.xs
 				tagger.dtd interchunk.dtd format.dtd  transfer.dtd postchunk.dtd modes.dtd crx.dtd \
 				tagger.rnc interchunk.rnc format.rnc  transfer.rnc postchunk.rnc modes.rnc
 
-LDADD = -lapertium $(lib_LTLIBRARIES)
+LDADD = $(lib_LTLIBRARIES)
 
 apertium_cleanstream_SOURCES = apertium_cleanstream.cc
 apertium_pretransfer_SOURCES = apertium_pretransfer.cc
@@ -294,9 +294,9 @@ apertium_adapt_docx_SOURCES = adapt_docx
 apertium_gen_modes_SOURCES = gen_modes.cc
 
 if WINDOWS
-AM_CPPFLAGS = -I$(top_srcdir)/apertium/win32 -I$(top_srcdir) $(APERTIUM_CFLAGS) $(ICU_CFLAGS)
+AM_CPPFLAGS = -I$(top_builddir) -I$(top_srcdir)/apertium/win32 -I$(top_srcdir) $(APERTIUM_CFLAGS) $(ICU_CFLAGS)
 else
-AM_CPPFLAGS = -I$(top_srcdir) $(APERTIUM_CFLAGS) $(ICU_CFLAGS)
+AM_CPPFLAGS = -I$(top_builddir) -I$(top_srcdir) $(APERTIUM_CFLAGS) $(ICU_CFLAGS)
 endif
 CLEANFILES = *~ apertium_destxt.cc apertium_retxt.cc apertium_deshtml.cc \
              apertium_rehtml.cc apertium_desrtf.cc apertium_rertf.cc \
@@ -314,200 +314,200 @@ CLEANFILES = *~ apertium_destxt.cc apert
              apertium_prelatex.cc apertium_postlatex.cc \
              $(GENERATEDSCRIPTS)
 
-apertium_destxt.cc: txt-format.xml Makefile.am deformat.xsl
-	xsltproc deformat.xsl txt-format.xml >$@tmp
+apertium_destxt.cc: $(srcdir)/txt-format.xml Makefile.am $(srcdir)/deformat.xsl
+	xsltproc $(srcdir)/deformat.xsl $(srcdir)/txt-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_desxpresstag.cc: xpresstag-format.xml Makefile.am deformat.xsl
-	xsltproc deformat.xsl xpresstag-format.xml >$@tmp
+apertium_desxpresstag.cc: $(srcdir)/xpresstag-format.xml Makefile.am $(srcdir)/deformat.xsl
+	xsltproc $(srcdir)/deformat.xsl $(srcdir)/xpresstag-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_rexpresstag.cc: xpresstag-format.xml Makefile.am reformat.xsl
-	xsltproc reformat.xsl xpresstag-format.xml >$@tmp
+apertium_rexpresstag.cc: $(srcdir)/xpresstag-format.xml Makefile.am $(srcdir)/reformat.xsl
+	xsltproc $(srcdir)/reformat.xsl $(srcdir)/xpresstag-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_retxt.cc: txt-format.xml Makefile.am reformat.xsl
-	xsltproc reformat.xsl txt-format.xml >$@tmp
+apertium_retxt.cc: $(srcdir)/txt-format.xml Makefile.am $(srcdir)/reformat.xsl
+	xsltproc $(srcdir)/reformat.xsl $(srcdir)/txt-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_deshtml.cc: html-format.xml Makefile.am deformat.xsl
-	xsltproc deformat.xsl html-format.xml >$@tmp
+apertium_deshtml.cc: $(srcdir)/html-format.xml Makefile.am $(srcdir)/deformat.xsl
+	xsltproc $(srcdir)/deformat.xsl $(srcdir)/html-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_deshtml_alt.cc: html-alt-format.xml Makefile.am deformat.xsl
-	xsltproc deformat.xsl $< >$@tmp
+apertium_deshtml_alt.cc: $(srcdir)/html-alt-format.xml Makefile.am $(srcdir)/deformat.xsl
+	xsltproc $(srcdir)/deformat.xsl $< >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_deshtml_noent.cc: html-noent-format.xml Makefile.am deformat.xsl
-	xsltproc deformat.xsl $< >$@tmp
+apertium_deshtml_noent.cc: $(srcdir)/html-noent-format.xml Makefile.am $(srcdir)/deformat.xsl
+	xsltproc $(srcdir)/deformat.xsl $< >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_rehtml.cc: html-format.xml Makefile.am reformat.xsl
-	xsltproc reformat.xsl html-format.xml >$@tmp
+apertium_rehtml.cc: $(srcdir)/html-format.xml Makefile.am $(srcdir)/reformat.xsl
+	xsltproc $(srcdir)/reformat.xsl $(srcdir)/html-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_rehtml_noent.cc: html-noent-format.xml Makefile.am reformat.xsl
-	xsltproc reformat.xsl html-noent-format.xml >$@tmp
+apertium_rehtml_noent.cc: $(srcdir)/html-noent-format.xml Makefile.am $(srcdir)/reformat.xsl
+	xsltproc $(srcdir)/reformat.xsl $(srcdir)/html-noent-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_rehtml_alt.cc: html-alt-format.xml Makefile.am reformat.xsl
-	xsltproc reformat.xsl html-alt-format.xml >$@tmp
+apertium_rehtml_alt.cc: $(srcdir)/html-alt-format.xml Makefile.am $(srcdir)/reformat.xsl
+	xsltproc $(srcdir)/reformat.xsl $(srcdir)/html-alt-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_desodt.cc: odt-format.xml Makefile.am deformat.xsl
-	xsltproc deformat.xsl odt-format.xml >$@tmp
+apertium_desodt.cc: $(srcdir)/odt-format.xml Makefile.am $(srcdir)/deformat.xsl
+	xsltproc $(srcdir)/deformat.xsl $(srcdir)/odt-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_reodt.cc: odt-format.xml Makefile.am reformat.xsl
-	xsltproc reformat.xsl odt-format.xml >$@tmp
+apertium_reodt.cc: $(srcdir)/odt-format.xml Makefile.am $(srcdir)/reformat.xsl
+	xsltproc $(srcdir)/reformat.xsl $(srcdir)/odt-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_desrtf.cc: rtf-format.xml Makefile.am deformat.xsl
-	xsltproc deformat.xsl rtf-format.xml >$@tmp
+apertium_desrtf.cc: $(srcdir)/rtf-format.xml Makefile.am $(srcdir)/deformat.xsl
+	xsltproc $(srcdir)/deformat.xsl $(srcdir)/rtf-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_rertf.cc: rtf-format.xml Makefile.am reformat.xsl
-	xsltproc reformat.xsl rtf-format.xml >$@tmp
+apertium_rertf.cc: $(srcdir)/rtf-format.xml Makefile.am $(srcdir)/reformat.xsl
+	xsltproc $(srcdir)/reformat.xsl $(srcdir)/rtf-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_deswxml.cc: wxml-format.xml Makefile.am deformat.xsl
-	xsltproc deformat.xsl wxml-format.xml >$@tmp
+apertium_deswxml.cc: $(srcdir)/wxml-format.xml Makefile.am $(srcdir)/deformat.xsl
+	xsltproc $(srcdir)/deformat.xsl $(srcdir)/wxml-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_rewxml.cc: wxml-format.xml Makefile.am reformat.xsl
-	xsltproc reformat.xsl wxml-format.xml >$@tmp
+apertium_rewxml.cc: $(srcdir)/wxml-format.xml Makefile.am $(srcdir)/reformat.xsl
+	xsltproc $(srcdir)/reformat.xsl $(srcdir)/wxml-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_deslatex.cc: latex-format.xml Makefile.am deformat.xsl
-	xsltproc deformat.xsl latex-format.xml >$@tmp
+apertium_deslatex.cc: $(srcdir)/latex-format.xml Makefile.am $(srcdir)/deformat.xsl
+	xsltproc $(srcdir)/deformat.xsl $(srcdir)/latex-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_relatex.cc: latex-format.xml Makefile.am reformat.xsl
-	xsltproc reformat.xsl latex-format.xml >$@tmp
+apertium_relatex.cc: $(srcdir)/latex-format.xml Makefile.am $(srcdir)/reformat.xsl
+	xsltproc $(srcdir)/reformat.xsl $(srcdir)/latex-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
 
 
-apertium_desxlsx.cc: xlsx-format.xml Makefile.am deformat.xsl
-	xsltproc deformat.xsl xlsx-format.xml >$@tmp
+apertium_desxlsx.cc: $(srcdir)/xlsx-format.xml Makefile.am $(srcdir)/deformat.xsl
+	xsltproc $(srcdir)/deformat.xsl $(srcdir)/xlsx-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_rexlsx.cc: xlsx-format.xml Makefile.am reformat.xsl
-	xsltproc reformat.xsl xlsx-format.xml >$@tmp
+apertium_rexlsx.cc: $(srcdir)/xlsx-format.xml Makefile.am $(srcdir)/reformat.xsl
+	xsltproc $(srcdir)/reformat.xsl $(srcdir)/xlsx-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_despptx.cc: pptx-format.xml Makefile.am deformat.xsl
-	xsltproc deformat.xsl pptx-format.xml >$@tmp
+apertium_despptx.cc: $(srcdir)/pptx-format.xml Makefile.am $(srcdir)/deformat.xsl
+	xsltproc $(srcdir)/deformat.xsl $(srcdir)/pptx-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_repptx.cc: pptx-format.xml Makefile.am reformat.xsl
-	xsltproc reformat.xsl pptx-format.xml >$@tmp
+apertium_repptx.cc: $(srcdir)/pptx-format.xml Makefile.am $(srcdir)/reformat.xsl
+	xsltproc $(srcdir)/reformat.xsl $(srcdir)/pptx-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_desmediawiki.cc: mediawiki-format.xml Makefile.am deformat.xsl
-	xsltproc deformat.xsl mediawiki-format.xml >$@tmp
+apertium_desmediawiki.cc: $(srcdir)/mediawiki-format.xml Makefile.am $(srcdir)/deformat.xsl
+	xsltproc $(srcdir)/deformat.xsl $(srcdir)/mediawiki-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_remediawiki.cc: mediawiki-format.xml Makefile.am reformat.xsl
-	xsltproc reformat.xsl mediawiki-format.xml >$@tmp
+apertium_remediawiki.cc: $(srcdir)/mediawiki-format.xml Makefile.am $(srcdir)/reformat.xsl
+	xsltproc $(srcdir)/reformat.xsl $(srcdir)/mediawiki-format.xml >$@tmp
 	$(FLEX) -Cfer -o$@ $@tmp
 	rm $@tmp
 
-apertium_prelatex.cc: apertium-prelatex.l
-	$(FLEX) -Cfer -o$@ apertium-prelatex.l
+apertium_prelatex.cc: $(srcdir)/apertium-prelatex.l
+	$(FLEX) -Cfer -o$@ $(srcdir)/apertium-prelatex.l
 
-apertium_postlatex.cc: apertium-postlatex.l
-	$(FLEX) -Cfer -o$@ apertium-postlatex.l
+apertium_postlatex.cc: $(srcdir)/apertium-postlatex.l
+	$(FLEX) -Cfer -o$@ $(srcdir)/apertium-postlatex.l
 
-apertium_postlatex_raw.cc: apertium-postlatex-raw.l
-	$(FLEX) -Cfer -o$@ apertium-postlatex-raw.l
+apertium_postlatex_raw.cc: $(srcdir)/apertium-postlatex-raw.l
+	$(FLEX) -Cfer -o$@ $(srcdir)/apertium-postlatex-raw.l
 
-apertium-validate-tagger: Makefile.am validate-header.sh
+apertium-validate-tagger: Makefile.am $(srcdir)/validate-header.sh
 	@echo "Creating apertium-validate-tagger script"
 	@echo "#!/usr/bin/env bash" > $@
-	@cat validate-header.sh >> $@
+	@cat $(srcdir)/validate-header.sh >> $@
 	@echo "xmllint --dtdvalid \"$(apertiumdir)\"/tagger.dtd --noout \"\$$FILE1\"" >>$@
 	@chmod a+x $@
 
-apertium-validate-transfer: Makefile.am validate-header.sh
+apertium-validate-transfer: Makefile.am $(srcdir)/validate-header.sh
 	@echo "Creating apertium-validate-transfer script"
 	@echo "#!/usr/bin/env bash" > $@
-	@cat validate-header.sh >> $@
+	@cat $(srcdir)/validate-header.sh >> $@
 	@echo "xmllint --dtdvalid \"$(apertiumdir)\"/transfer.dtd --noout \"\$$FILE1\"" >>$@
 	@chmod a+x $@
 
-apertium-validate-interchunk: Makefile.am validate-header.sh
+apertium-validate-interchunk: Makefile.am $(srcdir)/validate-header.sh
 	@echo "Creating apertium-validate-interchunk script"
 	@echo "#!/usr/bin/env bash" > $@
-	@cat validate-header.sh >> $@
+	@cat $(srcdir)/validate-header.sh >> $@
 	@echo "xmllint --dtdvalid \"$(apertiumdir)\"/interchunk.dtd --noout \"\$$FILE1\"" >>$@
 	@chmod a+x $@
 
-apertium-validate-postchunk: Makefile.am validate-header.sh
+apertium-validate-postchunk: Makefile.am $(srcdir)/validate-header.sh
 	@echo "Creating apertium-validate-postchunk script"
 	@echo "#!/usr/bin/env bash" > $@
-	@cat validate-header.sh >> $@
+	@cat $(srcdir)/validate-header.sh >> $@
 	@echo "xmllint --dtdvalid \"$(apertiumdir)\"/postchunk.dtd --noout \"\$$FILE1\"" >>$@
 	@chmod a+x $@
 
-apertium-validate-acx: Makefile.am validate-header.sh
+apertium-validate-acx: Makefile.am $(srcdir)/validate-header.sh
 	@echo "Creating apertium-validate-acx script"
 	@echo "#!/usr/bin/env bash" > $@
-	@cat validate-header.sh >> $@
+	@cat $(srcdir)/validate-header.sh >> $@
 	@echo "xmllint --relaxng \"$(prefix)\"/share/lttoolbox/acx.rng --schema \"$(prefix)\"/share/lttoolbox/acx.xsd --noout \"\$$FILE1\"" >>$@
 	@chmod a+x $@
 
-apertium-validate-crx: Makefile.am validate-header.sh
+apertium-validate-crx: Makefile.am $(srcdir)/validate-header.sh
 	@echo "Creating apertium-validate-crx script"
 	@echo "#!/usr/bin/env bash" > $@
-	@cat validate-header.sh >> $@
+	@cat $(srcdir)/validate-header.sh >> $@
 	@echo "xmllint --dtdvalid \"$(apertiumdir)\"/crx.dtd --noout \"\$$FILE1\"" >>$@
 	@chmod a+x $@
 
-apertium-validate-modes: Makefile.am validate-header.sh
+apertium-validate-modes: Makefile.am $(srcdir)/validate-header.sh
 	@echo "Creating apertium-validate-modes script"
 	@echo "#!/usr/bin/env bash" > $@
-	@cat validate-header.sh >> $@
+	@cat $(srcdir)/validate-header.sh >> $@
 	@echo "xmllint --dtdvalid \"$(apertiumdir)\"/modes.dtd --noout \"\$$FILE1\"" >>$@
 	@chmod a+x $@
 
 
-apertium-validate-dictionary: Makefile.am validate-header.sh
+apertium-validate-dictionary: Makefile.am $(srcdir)/validate-header.sh
 	@echo "Creating apertium-validate-dictionary script"
 	@echo "#!/usr/bin/env bash" > $@
-	@cat validate-header.sh >> $@
+	@cat $(srcdir)/validate-header.sh >> $@
 	@echo "# xsd is a non-final command, so just treated as a warning when compiling:" >> $@
 	@echo "xmllint --schema \"$(prefix)\"/share/lttoolbox/dix.xsd --noout \"\$$FILE1\" 2>&1 | grep -ve ' validates$$' -e ' fails to validate$$'" >> $@
 	@echo "xmllint --dtdvalid \"$(prefix)\"/share/lttoolbox/dix.dtd --noout \"\$$FILE1\"" >> $@
 	@chmod a+x $@
 
-apertium-gen-deformat: Makefile.am deformat-header.sh
+apertium-gen-deformat: Makefile.am $(srcdir)/deformat-header.sh
 	@echo "Creating apertium-gen-deformat script"
 	@echo "#!/usr/bin/env bash" > $@
-	@cat deformat-header.sh >> $@
+	@cat $(srcdir)/deformat-header.sh >> $@
 	@echo "xmllint --dtdvalid \"$(apertiumdir)\"/format.dtd --noout \"\$$FILE1\" && \\" >> $@
 	@if [ `basename xsltproc` == xsltproc ]; \
 	  then echo "xsltproc --stringparam mode \$$MODE \"$(apertiumdir)\"/deformat.xsl \"\$$FILE1\" >/tmp/\$$\$$.deformat.l && \\"; \
@@ -518,10 +518,10 @@ apertium-gen-deformat: Makefile.am defor
 	@echo "rm /tmp/\$$\$$.deformat.l /tmp/\$$\$$.lex.cc" >> $@
 	@chmod a+x $@
 
-apertium-gen-reformat: Makefile.am gen-header.sh
+apertium-gen-reformat: Makefile.am $(srcdir)/gen-header.sh
 	@echo "Creating apertium-gen-reformat script"
 	@echo "#!/usr/bin/env bash" > $@
-	@cat gen-header.sh >> $@
+	@cat $(srcdir)/gen-header.sh >> $@
 	@echo "xmllint --dtdvalid \"$(apertiumdir)\"/format.dtd --noout \"\$$FILE1\" && \\" >> $@
 	@echo "xsltproc \"$(apertiumdir)\"/reformat.xsl \"\$$FILE1\" >/tmp/\$$\$$.reformat.l && \\" >> $@
 	@echo "$(FLEX) \$$FLEXOPTS -o/tmp/\$$\$$.lex.cc /tmp/\$$\$$.reformat.l && \\" >> $@
@@ -529,26 +529,26 @@ apertium-gen-reformat: Makefile.am gen-h
 	@echo "rm /tmp/\$$\$$.reformat.l /tmp/\$$\$$.lex.cc" >> $@
 	@chmod a+x $@
 
-apertium-utils-fixlatex: Makefile.am utils-fixlatex-header.sh
+apertium-utils-fixlatex: Makefile.am $(srcdir)/utils-fixlatex-header.sh
 	@echo "Creating apertium-utils-fixlatex script"
 	@echo "#!/usr/bin/env bash" > $@
-	@cat utils-fixlatex-header.sh >> $@
+	@cat $(srcdir)/utils-fixlatex-header.sh >> $@
 	@chmod a+x $@
 
-apertium: Makefile.am apertium-header.sh
+apertium: Makefile.am $(srcdir)/apertium-header.sh
 	@echo 'Creating apertium script'
 	@echo '#!/usr/bin/env bash' > $@
 	@echo 'APERTIUM_PATH="$${APERTIUM_PATH:-$(prefix)/bin}"' >>$@
 	@echo 'APERTIUM_DATADIR="$${APERTIUM_DATADIR:-$(prefix)/share/apertium}"' >>$@
 	@echo 'apertium_version="Apertium $(VERSION)"' >>$@
-	@cat apertium-header.sh >>$@
+	@cat $(srcdir)/apertium-header.sh >>$@
 	@chmod a+x $@
 
-apertium-unformat: Makefile.am apertium-unformat-header.sh
+apertium-unformat: Makefile.am $(srcdir)/apertium-unformat-header.sh
 	@echo "Creating apertium-unformat script"
 	@echo "#!/usr/bin/env bash" > $@
 	@echo "APERTIUM_PATH=\"$(prefix)/bin\"" >>$@
-	@cat apertium-unformat-header.sh >>$@
+	@cat $(srcdir)/apertium-unformat-header.sh >>$@
 	@chmod a+x $@
 
 man_MANS=apertium.1 apertium-deshtml.1 apertium-desrtf.1 apertium-destxt.1 \
--- origsrc/apertium-3.9.12/configure.ac	2025-03-26 18:26:26.000000000 +0900
+++ src/apertium-3.9.12/configure.ac	2025-05-25 02:22:34.955397900 +0900
@@ -78,7 +78,9 @@ AC_C_BIGENDIAN
 AC_LANG(C++)
 AC_HEADER_STDC
 AC_CHECK_HEADERS([stdlib.h string.h unistd.h stddef.h filesystem string_view])
-AC_CHECK_LIB([stdc++fs], [_ZNSt12experimental10filesystem2v112current_pathEv])
+if test "$host_os" != "cygwin"; then
+  AC_CHECK_LIB([stdc++fs], [_ZNSt12experimental10filesystem2v112current_pathEv])
+fi
 AC_CHECK_HEADER([utf8cpp/utf8.h], [CPPFLAGS="-I/usr/include/utf8cpp/ $CPPFLAGS"], [
   AC_CHECK_HEADER([utf8.h], [], [AC_MSG_ERROR([You don't have utfcpp installed.])])
 ])
@@ -87,7 +89,7 @@ AC_CHECK_DECLS([fread_unlocked, fwrite_u
 AC_CHECK_FUNCS([setlocale strdup getopt snprintf])
 AC_REPLACE_FUNCS(getopt_long)
 
-AM_CONDITIONAL([WINDOWS], [test x$version_type = xwindows])
+AM_CONDITIONAL([WINDOWS], [test x$version_type = xwindows -a x$host_os != xcygwin])
 #AS_IF([test x$version_type = xwindows], [AC_DEFINE(HAVE_GETOPT_LONG,0)], [])
 
 # Checks for highest supported C++ standard
--- origsrc/apertium-3.9.12/python/Makefile.am	2025-03-26 18:26:26.000000000 +0900
+++ src/apertium-3.9.12/python/Makefile.am	2025-05-25 02:22:55.924983200 +0900
@@ -2,7 +2,7 @@ SWIG_INTERFACE = apertium_core.i
 
 BUILT_SOURCES = %_wrap.cpp
 
-%_wrap.cpp: $(SWIG_INTERFACE) setup.py
+%_wrap.cpp: $(srcdir)/$(SWIG_INTERFACE) setup.py
 	$(PYTHON) setup.py build
 
 install-exec-local:
--- origsrc/apertium-3.9.12/python/setup.py.in	2025-03-26 18:26:26.000000000 +0900
+++ src/apertium-3.9.12/python/setup.py.in	2025-05-25 02:24:24.353049600 +0900
@@ -7,7 +7,7 @@ from distutils.core import Extension, se
 from sys import platform
 
 compile_args = '@CPPFLAGS@'.split() + '@CXXFLAGS@'.split() + '@LTTOOLBOX_CFLAGS@'.split() + '@ICU_CFLAGS@'.split()
-link_args = []
+link_args = '@LIBS@'.split()
 if platform == 'darwin':
     compile_args += ['-stdlib=libc++', '-mmacosx-version-min=10.7']
     link_args.append('-mmacosx-version-min=10.7')
@@ -15,13 +15,13 @@ if platform == 'darwin':
 apertium_core_module = Extension(
     name='_apertium_core',
     language='c++',
-    sources=['apertium_core.i'],
-    swig_opts=['-c++', '-I..', '-I@top_srcdir@/apertium', '-Wall'],
-    include_dirs=['@top_srcdir@', '@top_srcdir@/apertium'] + '@LIBXML2_CFLAGS@'.replace('-I', '').split(),
-    library_dirs=['@top_srcdir@/apertium/.libs'],
-    libraries = ['apertium'],
+    sources=['@srcdir@/apertium_core.i'],
+    swig_opts=['-c++', '-I@top_srcdir@', '-I@top_srcdir@/apertium', '-Wall'],
+    include_dirs=['@top_builddir@/apertium', '@top_srcdir@', '@top_srcdir@/apertium'] + '@LIBXML2_CFLAGS@'.replace('-I', '').split() + '@CPPFLAGS@'.replace('-I', '').replace('-D_GNU_SOURCE', '').split(),
+    library_dirs=[],
+    libraries = [],
     extra_compile_args=compile_args,
-    extra_link_args=link_args,
+    extra_link_args=link_args + ['-L@top_builddir@/apertium/.libs', '-lapertium'],
 )
 
 setup(
--- origsrc/apertium-3.9.12/tests/tagger/Makefile.am	2025-03-26 18:26:26.000000000 +0900
+++ src/apertium-3.9.12/tests/tagger/Makefile.am	2025-05-25 02:24:39.655306500 +0900
@@ -14,4 +14,4 @@ apertiumlib = $(prefix)/lib
 apertiumsysconf = $(prefix)/etc/apertium
 
 test_find_similar_ambiguity_class_SOURCES = test_find_similar_ambiguity_classes.cc
-test_find_similar_ambiguity_class_LDADD = -L$(top_srcdir)/$(PACKAGE_NAME)/.libs/ -l$(PACKAGE_NAME)
+test_find_similar_ambiguity_class_LDADD = -L$(top_builddir)/$(PACKAGE_NAME)/.libs/ -l$(PACKAGE_NAME)
