--- origsrc/c-toxcore-0.2.21/auto_tests/network_test.c	2025-05-15 21:08:17.000000000 +0900
+++ src/c-toxcore-0.2.21/auto_tests/network_test.c	2026-02-05 18:14:41.878658400 +0900
@@ -1,4 +1,7 @@
 #include <string.h>
+#if defined(__CYGWIN__)
+#include <errno.h>
+#endif
 
 #include "../toxcore/network.h"
 #include "check_compat.h"
@@ -14,7 +17,8 @@ static void test_addr_resolv_localhost(v
      * normally this should happen automatically
      * cygwin doesn't do it for every network related function though
      * e.g. not for getaddrinfo... */
-    net_socket(0, 0, 0);
+    Family f = {0};
+    net_socket(0, f, 0, 0);
     errno = 0;
 #endif
 
--- origsrc/c-toxcore-0.2.21/configure.ac	2025-05-15 21:08:17.000000000 +0900
+++ src/c-toxcore-0.2.21/configure.ac	2026-02-05 18:28:17.778427500 +0900
@@ -225,6 +225,9 @@ case $host_os in
         WIN32="yes"
         EXTRA_LT_LDFLAGS="$EXTRA_LT_LDFLAGS -no-undefined"
     ;;
+    *cygwin*)
+        EXTRA_LT_LDFLAGS="$EXTRA_LT_LDFLAGS -no-undefined"
+    ;;
     *solaris*)
         LIBS="$LIBS -lssp -lsocket -lnsl"
     ;;
@@ -394,6 +397,15 @@ else
     AC_MSG_WARN([pkg-config was not found on your system, will search for libraries manually])
 fi
 
+PKG_CHECK_MODULES([CMP], [cmp],
+    [
+        CMP_FOUND="yes"
+    ],
+    [
+        CMP_FOUND="no"
+    ])
+AM_CONDITIONAL(CMP_FOUND, test "x$CMP_FOUND" = "xyes")
+
 if (test "x$BUILD_DHT_BOOTSTRAP_DAEMON" = "xyes") && \
    (test "x$LIBCONFIG_FOUND" = "xno"); then
     AC_CHECK_HEADER(libconfig.h,
--- origsrc/c-toxcore-0.2.21/toxcore/Makefile.inc	2025-05-15 21:08:17.000000000 +0900
+++ src/c-toxcore-0.2.21/toxcore/Makefile.inc	2026-02-05 18:28:59.607765700 +0900
@@ -6,9 +6,7 @@ libtoxcore_la_include_HEADERS = \
 
 libtoxcore_la_includedir = $(includedir)/tox
 
-libtoxcore_la_SOURCES = ../third_party/cmp/cmp.c \
-                        ../third_party/cmp/cmp.h \
-                        ../toxcore/events/conference_connected.c \
+libtoxcore_la_SOURCES = ../toxcore/events/conference_connected.c \
                         ../toxcore/events/conference_invite.c \
                         ../toxcore/events/conference_message.c \
                         ../toxcore/events/conference_peer_list_changed.c \
@@ -172,6 +170,15 @@ libtoxcore_la_LIBADD =  $(LIBSODIUM_LIBS
                         $(MSGPACK_LIBS) \
                         $(PTHREAD_LIBS)
 
+if CMP_FOUND
+libtoxcore_la_CFLAGS += $(CMP_CFLAGS)
+libtoxcore_la_LIBADD += $(CMP_LIBS)
+else
+libtoxcore_la_CFLAGS += -I$(top_srcdir)/third_party/cmp
+libtoxcore_la_SOURCES += ../third_party/cmp/cmp.c \
+                         ../third_party/cmp/cmp.h
+endif
+
 if SET_SO_VERSION
 
 EXTRA_libtoxcore_la_DEPENDENCIES = ../so.version
--- origsrc/c-toxcore-0.2.21/toxcore/bin_pack.c	2025-05-15 21:08:17.000000000 +0900
+++ src/c-toxcore-0.2.21/toxcore/bin_pack.c	2026-02-05 18:29:34.736820200 +0900
@@ -7,7 +7,7 @@
 #include <assert.h>
 #include <string.h>
 
-#include "../third_party/cmp/cmp.h"
+#include <cmp.h>
 #include "attributes.h"
 #include "ccompat.h"
 #include "logger.h"
--- origsrc/c-toxcore-0.2.21/toxcore/bin_unpack.c	2025-05-15 21:08:17.000000000 +0900
+++ src/c-toxcore-0.2.21/toxcore/bin_unpack.c	2026-02-05 18:29:41.750057500 +0900
@@ -7,7 +7,7 @@
 #include <assert.h>
 #include <string.h>
 
-#include "../third_party/cmp/cmp.h"
+#include <cmp.h>
 #include "attributes.h"
 #include "ccompat.h"
 #include "mem.h"
