--- origsrc/toxcore/toxav/Makefile.inc	2015-05-21 14:00:33.000000000 +0900
+++ src/toxcore/toxav/Makefile.inc	2015-05-21 14:05:44.689695200 +0900
@@ -23,7 +23,8 @@ libtoxav_la_CFLAGS = -I../toxcore \
                      $(AV_CFLAGS) \
                      $(PTHREAD_CFLAGS)
 
-libtoxav_la_LDFLAGS = $(TOXAV_LT_LDFLAGS) \
+libtoxav_la_LDFLAGS = -no-undefined \
+                      $(TOXAV_LT_LDFLAGS) \
                       $(LIBSODIUM_LDFLAGS) \
                       $(NACL_LDFLAGS) \
                       $(EXTRA_LT_LDFLAGS) \
--- origsrc/toxcore/toxcore/Makefile.inc	2015-05-21 14:00:34.000000000 +0900
+++ src/toxcore/toxcore/Makefile.inc	2015-05-21 14:06:05.481073000 +0900
@@ -58,7 +58,8 @@ libtoxcore_la_CFLAGS =  -I$(top_srcdir)
                         $(NACL_CFLAGS) \
                         $(PTHREAD_CFLAGS)
 
-libtoxcore_la_LDFLAGS = $(TOXCORE_LT_LDFLAGS) \
+libtoxcore_la_LDFLAGS = -no-undefined \
+                        $(TOXCORE_LT_LDFLAGS) \
                         $(EXTRA_LT_LDFLAGS) \
                         $(LIBSODIUM_LDFLAGS) \
                         $(NACL_LDFLAGS) \
--- origsrc/toxcore/toxdns/Makefile.inc	2015-05-21 14:00:34.000000000 +0900
+++ src/toxcore/toxdns/Makefile.inc	2015-05-21 14:06:22.012461100 +0900
@@ -14,7 +14,8 @@ libtoxdns_la_CFLAGS =   -I$(top_srcdir)
                         $(NACL_CFLAGS) \
                         $(PTHREAD_CFLAGS)
 
-libtoxdns_la_LDFLAGS =  $(TOXCORE_LT_LDFLAGS) \
+libtoxdns_la_LDFLAGS =  -no-undefined \
+                        $(TOXCORE_LT_LDFLAGS) \
                         $(EXTRA_LT_LDFLAGS) \
                         $(LIBSODIUM_LDFLAGS) \
                         $(NACL_LDFLAGS) \
--- origsrc/toxcore/toxencryptsave/Makefile.inc	2015-05-21 14:00:34.000000000 +0900
+++ src/toxcore/toxencryptsave/Makefile.inc	2015-05-21 14:06:38.619092700 +0900
@@ -30,7 +30,8 @@ libtoxencryptsave_la_CFLAGS =   -I$(top_
                         $(NACL_CFLAGS) \
                         $(PTHREAD_CFLAGS)
 
-libtoxencryptsave_la_LDFLAGS =  $(TOXCORE_LT_LDFLAGS) \
+libtoxencryptsave_la_LDFLAGS =  -no-undefined \
+                        $(TOXCORE_LT_LDFLAGS) \
                         $(EXTRA_LT_LDFLAGS) \
                         $(LIBSODIUM_LDFLAGS) \
                         $(NACL_LDFLAGS) \
