--- origsrc/nghttp2-0.7.4/configure.ac	2015-02-15 13:09:25.000000000 +0900
+++ src/nghttp2-0.7.4/configure.ac	2015-02-19 20:39:25.361359400 +0900
@@ -161,7 +161,7 @@ else
   AC_DEFINE([_U_], , [Hint to the compiler that a function parameters is not used])
 fi
 
-AX_CXX_COMPILE_STDCXX_11([noext], [optional])
+AX_CXX_COMPILE_STDCXX_11([ext], [optional])
 
 AC_LANG_PUSH(C++)
 
@@ -261,6 +261,8 @@ case "$host" in
     # android does not need -pthread, but needs followng 3 libs for C++
     APPLDFLAGS="$APPLDFLAGS -lstdc++ -latomic -lsupc++"
     ;;
+  *cygwin*)
+    ;;
   *)
     PTHREAD_LDFLAGS="-pthread"
     APPLDFLAGS="$APPLDFLAGS $PTHREAD_LDFLAGS"
@@ -311,19 +313,7 @@ AM_CONDITIONAL([HAVE_CUNIT], [ test "x${
 
 # libev (for src)
 # libev does not have pkg-config file.  Check it in an old way.
-LIBS_OLD=$LIBS
-# android requires -lm for floor
-AC_CHECK_LIB([ev], [ev_time], [have_libev=yes], [have_libev=no], [-lm])
-if test "x${have_libev}" = "xyes"; then
-  AC_CHECK_HEADER([ev.h], [have_libev=yes], [have_libev=no])
-  if test "x${have_libev}" = "xyes"; then
-    LIBEV_LIBS=-lev
-    LIBEV_CFLAGS=
-    AC_SUBST([LIBEV_LIBS])
-    AC_SUBST([LIBEV_CFLAGS])
-  fi
-fi
-LIBS=$LIBS_OLD
+PKG_CHECK_MODULES([LIBEV], [libev])
 
 # openssl (for src)
 PKG_CHECK_MODULES([OPENSSL], [openssl >= 1.0.1],
--- origsrc/nghttp2-0.7.4/tests/Makefile.am	2015-02-15 13:09:25.000000000 +0900
+++ src/nghttp2-0.7.4/tests/Makefile.am	2015-02-19 20:32:44.662140600 +0900
@@ -50,7 +50,6 @@ HFILES = nghttp2_pq_test.h nghttp2_map_t
 main_SOURCES = $(HFILES) $(OBJECTS)
 
 main_LDADD = ${top_builddir}/lib/libnghttp2.la @CUNIT_LIBS@ @TESTLDADD@
-main_LDFLAGS = -static
 
 if ENABLE_FAILMALLOC
 failmalloc_SOURCES = failmalloc.c failmalloc_test.c failmalloc_test.h \
