--- origsrc/nghttp2-0.5.1/configure.ac	2014-07-23 00:48:33.000000000 +0900
+++ src/nghttp2-0.5.1/configure.ac	2014-07-25 20:14:35.308666800 +0900
@@ -135,7 +135,7 @@ else
   AC_SUBST([CYTHON])
 fi
 
-AX_CXX_COMPILE_STDCXX_11([noext], [optional])
+AX_CXX_COMPILE_STDCXX_11([ext], [optional])
 
 AC_LANG_PUSH(C++)
 
@@ -204,6 +204,8 @@ case "$host" in
     # android does not need -pthread, but needs followng 2 libs for C++
     SRC_LIBS="$SRC_LIBS -lstdc++ -lsupc++"
     ;;
+  *cygwin*)
+    ;;
   *)
     SRC_LIBS="$SRC_LIBS -pthread"
     ;;
--- origsrc/nghttp2-0.5.1/tests/Makefile.am	2014-07-23 00:48:33.000000000 +0900
+++ src/nghttp2-0.5.1/tests/Makefile.am	2014-07-25 09:50:47.277293500 +0900
@@ -50,7 +50,7 @@ HFILES = nghttp2_pq_test.h nghttp2_map_t
 main_SOURCES = $(HFILES) $(OBJECTS)
 
 main_LDADD = ${top_builddir}/lib/libnghttp2.la
-main_LDFLAGS = -static @CUNIT_LIBS@ @TESTS_LIBS@
+main_LDFLAGS = @CUNIT_LIBS@ @TESTS_LIBS@
 
 if ENABLE_FAILMALLOC
 failmalloc_SOURCES = failmalloc.c failmalloc_test.c failmalloc_test.h \
