--- origsrc/nghttp2/configure.ac	2013-09-29 16:36:33.000000000 +0900
+++ src/nghttp2/configure.ac	2013-09-29 16:59:57.199409900 +0900
@@ -78,7 +78,7 @@ AM_PROG_CC_C_O
 PKG_PROG_PKG_CONFIG([0.20])
 AM_PATH_PYTHON([2.6],, [:])
 
-AX_CXX_COMPILE_STDCXX_11([noext], [optional])
+AX_CXX_COMPILE_STDCXX_11([ext], [optional])
 
 # Check that std::chrono::steady_clock is available. In particular,
 # gcc 4.6 does not have one, but has monotonic_clock which is the old
@@ -277,7 +277,7 @@ case "${host}" in
 esac
 
 if test "x$maintainer_mode" != "xno"; then
-    CFLAGS="$CFLAGS -Wall -Wextra -Werror"
+    CFLAGS="$CFLAGS -Wall -Wextra"
     CFLAGS="$CFLAGS -Wmissing-prototypes -Wstrict-prototypes"
     CFLAGS="$CFLAGS -Wmissing-declarations -Wpointer-arith"
     CFLAGS="$CFLAGS -Wdeclaration-after-statement"
--- origsrc/nghttp2/src/Makefile.am	2013-09-29 16:36:33.000000000 +0900
+++ src/nghttp2/src/Makefile.am	2013-09-29 16:52:16.353857800 +0900
@@ -105,7 +105,6 @@ nghttpx_unittest_SOURCES = shrpx-unittes
 	util_test.cc util_test.h
 nghttpx_unittest_CPPFLAGS = ${AM_CPPFLAGS}\
 	 -DNGHTTP2_TESTS_DIR=\"$(top_srcdir)/tests\"
-nghttpx_unittest_LDFLAGS = -static
 nghttpx_unittest_LDADD = libnghttpx.a ${LDADD} ${AM_LDFLAGS} \
 	@CUNIT_LIBS@ @TESTS_LIBS@
 
--- origsrc/nghttp2/tests/Makefile.am	2013-09-29 16:36:33.000000000 +0900
+++ src/nghttp2/tests/Makefile.am	2013-09-29 16:57:44.491553100 +0900
@@ -47,7 +47,7 @@ HFILES = nghttp2_pq_test.h nghttp2_map_t
 main_SOURCES = $(HFILES) $(OBJECTS)
 
 main_LDADD = ${top_builddir}/lib/libnghttp2.la
-main_LDFLAGS = -static @CUNIT_LIBS@ @TESTS_LIBS@
+main_LDFLAGS = @CUNIT_LIBS@ @TESTS_LIBS@
 
 # failmalloc_SOURCES = failmalloc.c failmalloc_test.c failmalloc_test.h \
 # 	malloc_wrapper.c malloc_wrapper.h \
