--- origsrc/libtommath-1.1.0/makefile.shared	2019-01-29 04:32:32.000000000 +0900
+++ src/libtommath-1.1.0/makefile.shared	2019-02-23 20:58:39.190072900 +0900
@@ -59,7 +59,7 @@ objs: $(OBJECTS)
 LOBJECTS = $(OBJECTS:.o=.lo)
 
 $(LIBNAME):  $(OBJECTS)
-	$(LIBTOOL) --mode=link --tag=CC $(CC) $(LDFLAGS) $(LOBJECTS) -o $(LIBNAME) -rpath $(LIBPATH) -version-info $(VERSION_SO) $(LIBTOOLFLAGS)
+	$(LIBTOOL) --mode=link --tag=CC $(CC) $(LDFLAGS) $(LOBJECTS) -o $(LIBNAME) -rpath $(LIBPATH) -version-info $(VERSION_SO) $(LIBTOOLFLAGS) -no-undefined
 
 install: $(LIBNAME)
 	install -d $(DESTDIR)$(LIBPATH)
