--- origsrc/expatmm-1.0.3/expatmm.pc.in	2015-08-10 05:09:58.000000000 +0900
+++ src/expatmm-1.0.3/expatmm.pc.in	2016-03-16 23:24:01.813684900 +0900
@@ -5,17 +5,16 @@
 # IntelliTree ExpatMM - Expat C++ Implementation
 # Copyright (C) 2006, 2007, 2008, 2009 IntelliTree Solutions llc
 # Written by Coleman Kane <ckane@intellitree.com>
-# 
+#
 # The autoconf template for the pkg-config description.
 #
 prefix=@prefix@
 exec_prefix=${prefix}
 libdir=${exec_prefix}/lib
-includedir=${prefix}/include/expatmm-1.0
+includedir=${prefix}/include/expatmm-1.0/expatmm
 
 Name: expatmm
 Description: ExpatMM C++ Wrapper for Expat
-Version: 1.0.1
-Libs: -L${exec_prefix}/lib -lexpatmm
-Cflags: -I${includedir}
+Version: 1.0.3
+Libs: -lexpatmm
 URL: https://github.com/ckane/expatmm
--- origsrc/expatmm-1.0.3/src/Makefile.am	2015-08-10 05:09:58.000000000 +0900
+++ src/expatmm-1.0.3/src/Makefile.am	2016-03-16 23:22:22.307334000 +0900
@@ -12,3 +12,5 @@ libexpatmm_la_include_HEADERS= $(top_src
                                $(top_srcdir)/include/ExpatXMLFileParser.h \
                                $(top_srcdir)/include/expatmm-libdef.h \
                                $(top_builddir)/include/expatmm.h
+
+libexpatmm_la_LDFLAGS = -no-undefined
