--- origsrc/roscpp_core-0.7.2/roscpp_serialization/CMakeLists.txt	2020-05-29 13:45:12.000000000 +0900
+++ src/roscpp_core-0.7.2/roscpp_serialization/CMakeLists.txt	2020-07-17 14:55:57.202149600 +0900
@@ -12,6 +12,10 @@ link_directories(${catkin_LIBRARY_DIRS})
 
 add_library(roscpp_serialization src/serialization.cpp)
 
+if (CYGWIN)
+  set_target_properties(roscpp_serialization PROPERTIES SOVERSION 0)
+endif()
+
 install(TARGETS roscpp_serialization
   ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
   LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
