--- origsrc/libtddmac/configure.ac	2024-11-07 13:22:06.000000000 +0900
+++ src/libtddmac/configure.ac	2024-11-07 13:22:07.243594800 +0900
@@ -1,10 +1,10 @@
 #                                               -*- Autoconf -*-
 # Process this file with autoconf to produce a configure script.
 
-AC_INIT
+AC_INIT([libtddmac], [1.0], [dhobsong@igel.co.jp])
 
 AC_PREREQ([2.65])
-AM_INIT_AUTOMAKE([tddmac], [1.0], [dhobsong@igel.co.jp])
+AM_INIT_AUTOMAKE
 
 AC_CONFIG_SRCDIR([src/libtddmac/tddmac.c])
 AC_CONFIG_HEADERS([config.h])
@@ -62,7 +62,6 @@ AC_OUTPUT([
 Makefile
 src/Makefile
 src/libtddmac/Makefile
-src/tools/Makefile
 include/Makefile
 include/tddmac/Makefile
 tddmac.pc
--- origsrc/libtddmac/src/Makefile.am	2024-11-07 13:22:06.000000000 +0900
+++ src/libtddmac/src/Makefile.am	2024-11-07 13:22:07.243594800 +0900
@@ -1 +1 @@
-SUBDIRS = libtddmac tools
+SUBDIRS = libtddmac
--- origsrc/libtddmac/src/libtddmac/Makefile.am	2024-11-07 13:22:06.000000000 +0900
+++ src/libtddmac/src/libtddmac/Makefile.am	2024-11-07 13:22:07.243594800 +0900
@@ -6,6 +6,6 @@ lib_LTLIBRARIES = libtddmac.la
 libtddmac_la_SOURCES = \
 	tddmac.c
 
-libtddmac_la_CFLAGS = $(UIOMUX_CFLAGS) -Wall -Werror
-libtddmac_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
+libtddmac_la_CFLAGS = $(UIOMUX_CFLAGS) -Wall
+libtddmac_la_LDFLAGS = -no-undefined -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
 libtddmac_la_LIBADD = $(UIOMUX_LIBS)
--- origsrc/libtddmac/tddmac.pc.in	2024-11-07 13:22:06.000000000 +0900
+++ src/libtddmac/tddmac.pc.in	2024-11-07 13:25:18.353529100 +0900
@@ -7,7 +7,7 @@ Name: 2ddmac
 Description: A library for controlling SH-Mobile 2DDMAC
 Requires:
 Version: @VERSION@
-Libs: -L${libdir} -l2ddmac
+Libs: -L${libdir} -ltddmac
 Libs.private:
 Cflags: -I${includedir}
 
