--- origsrc/libtddmac/configure.ac	2015-04-26 17:59:57.000000000 +0900
+++ src/libtddmac/configure.ac	2015-04-26 18:06:21.604548400 +0900
@@ -1,10 +1,10 @@
 #                                               -*- Autoconf -*-
 # Process this file with autoconf to produce a configure script.
 
-AC_INIT
+AC_INIT([libtddmac], [1.0], [dhobsong@igel.co.jp])
 
 AC_PREREQ([2.65])
-AM_INIT_AUTOMAKE([tddmac], [1.0], [dhobsong@igel.co.jp])
+AM_INIT_AUTOMAKE
 
 AC_CONFIG_SRCDIR([src/libtddmac/tddmac.c])
 AC_CONFIG_HEADERS([config.h])
@@ -62,7 +62,6 @@ AC_OUTPUT([
 Makefile
 src/Makefile
 src/libtddmac/Makefile
-src/tools/Makefile
 include/Makefile
 include/tddmac/Makefile
 tddmac.pc
--- origsrc/libtddmac/src/Makefile.am	2015-04-26 17:59:57.000000000 +0900
+++ src/libtddmac/src/Makefile.am	2015-04-26 18:07:12.964117500 +0900
@@ -1 +1 @@
-SUBDIRS = libtddmac tools
+SUBDIRS = libtddmac
--- origsrc/libtddmac/src/libtddmac/Makefile.am	2015-04-26 17:59:57.000000000 +0900
+++ src/libtddmac/src/libtddmac/Makefile.am	2015-04-26 18:09:55.024479700 +0900
@@ -6,6 +6,6 @@ lib_LTLIBRARIES = libtddmac.la
 libtddmac_la_SOURCES = \
 	tddmac.c
 
-libtddmac_la_CFLAGS = $(UIOMUX_CFLAGS) -Wall -Werror
-libtddmac_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
+libtddmac_la_CFLAGS = $(UIOMUX_CFLAGS) -Wall
+libtddmac_la_LDFLAGS = -no-undefined -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
 libtddmac_la_LIBADD = $(UIOMUX_LIBS)
