--- origsrc/lapackpp-2.5.4/include/arch.h	2010-11-11 05:29:09.000000000 +0900
+++ src/lapackpp-2.5.4/include/arch.h	2015-03-23 00:39:08.915672400 +0900
@@ -53,7 +53,7 @@ extern "C" {
 
 /* Needed for windows DLLs */
 #ifndef DLLIMPORT
-#  if defined( __declspec ) | defined ( _MSC_VER )
+#  if !defined(__CYGWIN__) && (defined( __declspec ) | defined ( _MSC_VER ))
 /*     _MSC_VER checks for Microsoft Visual C++. */
 /*      Microsoft Visual C++ 7.1  _MSC_VER = 1310 */
 /*      Microsoft Visual C++ 7.0  _MSC_VER = 1300 */
