--- origsrc/lapackpp-2025.05.28/CMakeLists.txt	2025-05-28 21:26:20.000000000 +0900
+++ src/lapackpp-2025.05.28/CMakeLists.txt	2025-06-11 03:06:48.500612000 +0900
@@ -696,6 +696,10 @@ set_target_properties(
     SOVERSION "${soversion}"
 )
 
+if (DEFINED SOVERSION)
+    set_target_properties(lapackpp PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.15)
     # Conditionally add -Wall. See CMake tutorial.
     set( gcc_like_cxx "$<COMPILE_LANG_AND_ID:CXX,ARMClang,AppleClang,Clang,GNU>" )
@@ -795,6 +799,8 @@ endif()
 include( GNUInstallDirs )
 if (WIN32)
     set( install_configdir "lapackpp" )
+elseif(CYGWIN)
+    set( install_configdir "${CMAKE_INSTALL_LIBDIR}/cmake/lapackpp" )
 else()
     set( install_configdir "${CMAKE_INSTALL_LIBDIR}/cmake/lapackpp" )
 endif()
@@ -850,3 +856,10 @@ install(
           "${CMAKE_CURRENT_BINARY_DIR}/lapackppConfigVersion.cmake"
     DESTINATION "${install_configdir}"
 )
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/lapackpp.pc.in ${CMAKE_BINARY_DIR}/lapackpp.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/lapackpp.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/lapackpp-2025.05.28/lapackpp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/lapackpp-2025.05.28/lapackpp.pc.in	2025-06-11 03:06:48.516260200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: lapackpp
+Description: C++ API for the Linear Algebra PACKage
+Version: @VERSION@
+Libs: -L${libdir} -llapackpp
+Cflags: -I${includedir}
