--- origsrc/QuantLib-QuantLib-v1.29/ql/Makefile.am	2023-01-16 00:57:44.000000000 +0900
+++ src/QuantLib-QuantLib-v1.29/ql/Makefile.am	2023-02-14 17:36:05.452800000 +0900
@@ -105,7 +105,7 @@ EXTRA_DIST =
 endif
 
 lib_LTLIBRARIES = libQuantLib.la
-libQuantLib_la_LDFLAGS = -version-info 0:0:0
+libQuantLib_la_LDFLAGS = -no-undefined -version-info 0:0:0
 
 libQuantLib_la_LIBADD = \
     cashflows/libCashFlows.la \
