--- origsrc/QuantLib-QuantLib-v1.18/ql/Makefile.am	2020-03-19 04:25:38.000000000 +0900
+++ src/QuantLib-QuantLib-v1.18/ql/Makefile.am	2020-03-27 10:16:45.991024500 +0900
@@ -104,7 +104,7 @@ EXTRA_DIST =
 endif
 
 lib_LTLIBRARIES = libQuantLib.la
-libQuantLib_la_LDFLAGS = -version-info 0:0:0
+libQuantLib_la_LDFLAGS = -no-undefined -version-info 0:0:0
 
 libQuantLib_la_LIBADD = \
     cashflows/libCashFlows.la \
