--- origsrc/QuantLib-1.37/ql/Makefile.am	2025-01-20 17:46:24.000000000 +0900
+++ src/QuantLib-1.37/ql/Makefile.am	2025-03-28 23:19:41.836303200 +0900
@@ -107,7 +107,7 @@ EXTRA_DIST =
 endif
 
 lib_LTLIBRARIES = libQuantLib.la
-libQuantLib_la_LDFLAGS = -version-info 0:0:0
+libQuantLib_la_LDFLAGS = -no-undefined -version-info 0:0:0
 
 libQuantLib_la_LIBADD = \
     cashflows/libCashFlows.la \
--- origsrc/QuantLib-1.37/ql/cashflow.cpp	2025-01-20 17:46:24.000000000 +0900
+++ src/QuantLib-1.37/ql/cashflow.cpp	2025-03-28 23:19:41.836303200 +0900
@@ -42,7 +42,7 @@ namespace QuantLib {
             // specified in the settings (if any)
             ext::optional<bool> includeToday =
                 Settings::instance().includeTodaysCashFlows();
-            if (includeToday.has_value())
+            if (includeToday.is_initialized())
                 includeRefDate = includeToday;
         }
         return Event::hasOccurred(refDate, includeRefDate);
