--- origsrc/QuantLib-1.3/ql/Makefile.am	2013-06-12 20:30:29.000000000 +0900
+++ src/QuantLib-1.3/ql/Makefile.am	2013-09-16 13:07:40.839314400 +0900
@@ -76,7 +76,7 @@ libQuantLib_la_SOURCES = \
 	timegrid.cpp
 
 lib_LTLIBRARIES = libQuantLib.la
-libQuantLib_la_LDFLAGS = -version-info 0:0:0
+libQuantLib_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 
 libQuantLib_la_LIBADD = \
     cashflows/libCashFlows.la \
--- origsrc/QuantLib-1.3/ql/config.hpp	2013-06-14 23:34:08.000000000 +0900
+++ src/QuantLib-1.3/ql/config.hpp	2013-09-16 13:12:09.026814400 +0900
@@ -64,14 +64,14 @@
 
 /* Define this if tracing messages should allowed (whether they are actually
    emitted will depend on run-time settings.) */
-#define QL_ENABLE_TRACING 1
+/* #undef QL_ENABLE_TRACING */
 
 /* Define this if error messages should include current function information.
    */
-#define QL_ERROR_FUNCTIONS 1
+/* #undef QL_ERROR_FUNCTIONS */
 
 /* Define this if error messages should include file and line information. */
-#define QL_ERROR_LINES 1
+/* #undef QL_ERROR_LINES */
 
 /* Define this if extra safety checks should be performed. This can degrade
    performance. */
