--- origsrc/QuantLib-1.9/ql/Makefile.am	2016-09-07 18:50:39.000000000 +0900
+++ src/QuantLib-1.9/ql/Makefile.am	2016-12-28 21:07:09.239437000 +0900
@@ -78,7 +78,7 @@ libQuantLib_la_SOURCES = \
 	timegrid.cpp
 
 lib_LTLIBRARIES = libQuantLib.la
-libQuantLib_la_LDFLAGS = -version-info 0:0:0
+libQuantLib_la_LDFLAGS = -no-undefined -version-info 0:0:0
 
 libQuantLib_la_LIBADD = \
     cashflows/libCashFlows.la \
