--- origsrc/QuantLib-1.1/configure.ac	2010-04-19 18:41:13.000000000 +0900
+++ src/QuantLib-1.1/configure.ac	2011-07-24 13:07:43.662388400 +0900
@@ -57,7 +57,7 @@ fi
 AC_PROG_CC
 AC_PROG_CXX
 AC_PROG_CPP
-AC_PROG_LIBTOOL
+LT_INIT
 
 AM_CONDITIONAL(ENABLE_STATIC, test "${enable_static}" == "yes")
 
--- origsrc/QuantLib-1.1/ql/Makefile.am	2010-04-19 18:41:13.000000000 +0900
+++ src/QuantLib-1.1/ql/Makefile.am	2011-07-24 13:08:20.184905200 +0900
@@ -76,7 +76,7 @@ libQuantLib_la_SOURCES = \
 	timegrid.cpp
 
 lib_LTLIBRARIES = libQuantLib.la
-libQuantLib_la_LDFLAGS = -version-info 0:0:0
+libQuantLib_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 
 libQuantLib_la_LIBADD = \
     cashflows/libCashFlows.la \
--- origsrc/QuantLib-1.1/test-suite/Makefile.am	2010-10-19 17:38:53.000000000 +0900
+++ src/QuantLib-1.1/test-suite/Makefile.am	2011-07-24 20:45:09.881705000 +0900
@@ -172,12 +172,12 @@ noinst_PROGRAMS = quantlib-benchmark
 endif
 
 quantlib_test_suite_SOURCES = ${QL_TESTS}
-quantlib_test_suite_LDADD = libUnitMain.la ${top_builddir}/ql/libQuantLib.la
-quantlib_test_suite_LDFLAGS = -l${BOOST_UNIT_TEST_LIB}
+quantlib_test_suite_LDADD = libUnitMain.la ${top_builddir}/ql/libQuantLib.la -l${BOOST_UNIT_TEST_LIB}
+quantlib_test_suite_LDFLAGS = 
 
 quantlib_benchmark_SOURCES = ${QL_BENCHMARKS}
-quantlib_benchmark_LDADD = libUnitMain.la ${top_builddir}/ql/libQuantLib.la
-quantlib_benchmark_LDFLAGS = -l${BOOST_UNIT_TEST_LIB}
+quantlib_benchmark_LDADD = libUnitMain.la ${top_builddir}/ql/libQuantLib.la -l${BOOST_UNIT_TEST_LIB}
+quantlib_benchmark_LDFLAGS = 
 
 TESTS = quantlib-test-suite$(EXEEXT)
 TESTS_ENVIRONMENT = BOOST_TEST_LOG_LEVEL=message
