--- origsrc/GDCM-3.0.5/CMake/FindCharLS.cmake	2020-02-14 18:57:18.000000000 +0900
+++ src/GDCM-3.0.5/CMake/FindCharLS.cmake	2020-02-22 09:29:20.662711700 +0900
@@ -12,7 +12,7 @@ find_path(CHARLS_INCLUDE_DIR CharLS/char
 )
 
 find_library(CHARLS_LIBRARY
-  NAMES CharLS
+  NAMES charls
   PATHS /usr/lib /usr/local/lib
   )
 
--- origsrc/GDCM-3.0.5/Source/Common/CMakeLists.txt	2020-02-14 18:57:18.000000000 +0900
+++ src/GDCM-3.0.5/Source/Common/CMakeLists.txt	2020-02-21 21:59:42.181976600 +0900
@@ -18,7 +18,9 @@ mark_as_advanced(
 
 
 CHECK_INCLUDE_FILE_CONCAT("sys/time.h"     GDCM_HAVE_SYS_TIME_H)
-CHECK_INCLUDE_FILE_CONCAT("winsock.h"       GDCM_HAVE_WINSOCK_H)
+if (NOT CYGWIN)
+  CHECK_INCLUDE_FILE_CONCAT("winsock.h"       GDCM_HAVE_WINSOCK_H)
+endif()
 CHECK_INCLUDE_FILE_CONCAT("byteswap.h"       GDCM_HAVE_BYTESWAP_FILE_H)
 # see bug #324
 if(GDCM_HAVE_BYTESWAP_FILE_H)
@@ -27,7 +29,9 @@ if(GDCM_HAVE_BYTESWAP_FILE_H)
     "\#include <byteswap.h>\nint main() { return bswap_32( 42 ); }"
     GDCM_HAVE_BYTESWAP_H)
 endif()
-CHECK_INCLUDE_FILE("rpc.h"       GDCM_HAVE_RPC_H)
+if (NOT CYGWIN)
+  CHECK_INCLUDE_FILE("rpc.h"       GDCM_HAVE_RPC_H)
+endif()
 CHECK_INCLUDE_FILE("langinfo.h"       GDCM_HAVE_LANGINFO_H)
 
 include(CheckFunctionExists)
--- origsrc/GDCM-3.0.5/Source/MediaStorageAndFileFormat/CMakeLists.txt	2020-02-14 18:57:18.000000000 +0900
+++ src/GDCM-3.0.5/Source/MediaStorageAndFileFormat/CMakeLists.txt	2020-02-22 09:29:31.841806900 +0900
@@ -216,6 +216,9 @@ endif()
 if(GDCM_USE_SYSTEM_JSON)
   target_link_libraries(gdcmMSFF LINK_PRIVATE ${JSON_LIBRARIES})
 endif()
+if(GDCM_USE_SYSTEM_CHARLS)
+  target_link_libraries(gdcmMSFF LINK_PRIVATE ${CHARLS_LIBRARY})
+endif()
 # handling of static lib within shared is a mess:
 #target_link_libraries(gdcmMSFF gdcmrle)
 set_target_properties(gdcmMSFF PROPERTIES ${GDCM_LIBRARY_PROPERTIES})
--- origsrc/GDCM-3.0.5/Source/MediaStorageAndFileFormat/gdcmFileStreamer.cxx	2020-02-14 18:57:18.000000000 +0900
+++ src/GDCM-3.0.5/Source/MediaStorageAndFileFormat/gdcmFileStreamer.cxx	2020-02-21 17:36:55.706120000 +0900
@@ -32,7 +32,7 @@
 #include <io.h>
 typedef int64_t off64_t;
 #else
-#if defined(__APPLE__) || defined(__FreeBSD__)
+#if defined(__APPLE__) || defined(__FreeBSD__) || defined(__CYGWIN__)
 #  define off64_t off_t
 #endif
 #include <unistd.h> // ftruncate
--- origsrc/GDCM-3.0.5/Source/MediaStorageAndFileFormat/gdcmUIDGenerator.cxx	2020-02-14 18:57:18.000000000 +0900
+++ src/GDCM-3.0.5/Source/MediaStorageAndFileFormat/gdcmUIDGenerator.cxx	2020-02-21 21:49:53.901961300 +0900
@@ -19,7 +19,7 @@
 #include <cstring>
 
 // FIXME...
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 #define HAVE_UUIDCREATE
 #else
 #define HAVE_UUID_GENERATE
--- origsrc/GDCM-3.0.5/Source/MediaStorageAndFileFormat/gdcmUUIDGenerator.cxx	2020-02-14 18:57:18.000000000 +0900
+++ src/GDCM-3.0.5/Source/MediaStorageAndFileFormat/gdcmUUIDGenerator.cxx	2020-02-21 22:23:34.586267600 +0900
@@ -18,7 +18,7 @@
 #include <cstring> // memcpy
 
 // FIXME...
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 #define HAVE_UUIDCREATE
 #else
 #define HAVE_UUID_GENERATE
