--- origsrc/dbh-4.6.2/configure.ac	2011-04-01 14:19:31.000000000 +0900
+++ src/dbh-4.6.2/configure.ac	2011-08-12 10:01:43.904144400 +0900
@@ -110,7 +110,7 @@ AC_PROG_LIBTOOL
 AC_PROG_MAKE_SET
 
 AC_CHECK_PROG(cv_dpkg, dpkg, yes, no)
-AM_CONDITIONAL(WANT_DEBIAN, [test "$cv_dpkg"=yes])
+AM_CONDITIONAL(WANT_DEBIAN, [test "$cv_dpkg"=yes && `uname -s`!="*CYGWIN*" ])
 
 AC_CHECK_PROG(cv_pkgconfig, pkg-config, yes, no)
 GTK_DOC_CHECK([1.15],[--flavour no-tmpl])
--- origsrc/dbh-4.6.2/dbh2.pc.in	2011-04-01 13:55:15.000000000 +0900
+++ src/dbh-4.6.2/dbh2.pc.in	2011-08-12 08:50:34.972778200 +0900
@@ -7,5 +7,5 @@ Name: dbh
 Description: Diskbased Hashtables
 Requires:
 Version: @DBH_VERSION@
-Libs: -Wl,-R${libdir} -L${libdir}/dbh2 -ldbh2 -lm -lpthread
+Libs: -L${libdir}/dbh2 -ldbh2 -lm -lpthread
 Cflags: -I${includedir}
--- origsrc/dbh-4.6.2/src/Makefile.am	2011-04-01 14:18:41.000000000 +0900
+++ src/dbh-4.6.2/src/Makefile.am	2011-08-12 09:43:49.927094100 +0900
@@ -15,5 +15,6 @@ dbhincludedir = $(includedir)/dbh
 dbhinclude_HEADERS = dbh.h 
 
 #libdbh2_la_LDFLAGS = -release @DBH_RELEASE@  -version-info @LIBDBH_VERINFO@ -export-dynamic
+libdbh2_la_LDFLAGS = -no-undefined -version-info @LIBDBH_VERINFO@
 
 #	-release @DBH_VERSION_MAJOR@.@DBH_VERSION_MINOR@	
