--- origsrc/sombok-2.3.1beta1/Makefile.am	2012-07-12 12:04:15.000000000 +0900
+++ src/sombok-2.3.1beta1/Makefile.am	2013-09-17 21:07:20.119579900 +0900
@@ -6,7 +6,7 @@ if MALLOC_DEBUG
 libsombok_la_SOURCES += src/mymalloc.c
 endif
 libsombok_la_CFLAGS = -I. -Iinclude @LIBTHAI_CFLAGS@
-libsombok_la_LDFLAGS = -version-info 4:6:1 @LIBTHAI_LIBS@
+libsombok_la_LDFLAGS = -no-undefined -version-info 4:6:1 @LIBTHAI_LIBS@
 
 include_HEADERS = include/sombok_constants.h
 nodist_include_HEADERS = include/sombok.h
--- origsrc/sombok-2.3.1beta1/configure.ac	2012-03-18 17:39:00.000000000 +0900
+++ src/sombok-2.3.1beta1/configure.ac	2013-09-17 21:07:42.303085100 +0900
@@ -120,7 +120,7 @@ AC_ARG_WITH(unicode-version,
   AC_HELP_STRING(--with-unicode-version=VERSION,
   [version of Unicode Standard @<:@default=current version@:>@]),
   UNICODE_VERSION=$withval,UNICODE_VERSION=$DEFAULT_UNICODE_VERSION)
-AC_CHECK_FILE([lib/$UNICODE_VERSION.c],
+AC_CHECK_FILE([${srcdir}/lib/$UNICODE_VERSION.c],
   AC_MSG_RESULT($UNICODE_VERSION),
   AC_MSG_ERROR(Unknown Unicode version $UNICODE_VERSION.))
 AC_SUBST(UNICODE_VERSION)
