--- origsrc/sombok-2.3.0/Makefile.am	2012-07-11 17:00:01.000000000 +0900
+++ src/sombok-2.3.0/Makefile.am	2012-11-05 07:00:19.318840900 +0900
@@ -5,8 +5,8 @@ libsombok_la_SOURCES = lib/break.c lib/c
 if MALLOC_DEBUG
 libsombok_la_SOURCES += src/mymalloc.c
 endif
-libsombok_la_CFLAGS = -I. -Iinclude @LIBTHAI_CFLAGS@
-libsombok_la_LDFLAGS = -version-info 4:6:1 @LIBTHAI_LIBS@
+libsombok_la_CFLAGS = -I. -Iinclude -I$(top_srcdir)/include @LIBTHAI_CFLAGS@
+libsombok_la_LDFLAGS = -version-info 4:6:1 @LIBTHAI_LIBS@ -no-undefined
 
 include_HEADERS = include/sombok_constants.h
 nodist_include_HEADERS = include/sombok.h
--- origsrc/sombok-2.3.0/configure.ac	2012-10-01 16:56:42.000000000 +0900
+++ src/sombok-2.3.0/configure.ac	2012-11-05 06:59:44.268063000 +0900
@@ -120,7 +120,7 @@ AC_ARG_WITH(unicode-version,
   AC_HELP_STRING(--with-unicode-version=VERSION,
   [version of Unicode Standard @<:@default=current version@:>@]),
   UNICODE_VERSION=$withval,UNICODE_VERSION=$DEFAULT_UNICODE_VERSION)
-AC_CHECK_FILE([lib/$UNICODE_VERSION.c],
+AC_CHECK_FILE([${srcdir}/lib/$UNICODE_VERSION.c],
   AC_MSG_RESULT($UNICODE_VERSION),
   AC_MSG_ERROR(Unknown Unicode version $UNICODE_VERSION.))
 AC_SUBST(UNICODE_VERSION)
