--- origsrc/sombok/Makefile.am	2025-01-03 22:27:43.421402100 +0900
+++ src/sombok/Makefile.am	2025-01-03 22:28:29.512274400 +0900
@@ -5,8 +5,8 @@ libsombok_la_SOURCES = lib/break.c lib/c
 if MALLOC_DEBUG
 libsombok_la_SOURCES += src/mymalloc.c
 endif
-libsombok_la_CFLAGS = -I. -Iinclude @LIBTHAI_CFLAGS@
-libsombok_la_LDFLAGS = -version-info 4:7:1 @LIBTHAI_LIBS@
+libsombok_la_CFLAGS = -I. -Iinclude -I$(srcdir)/include @LIBTHAI_CFLAGS@
+libsombok_la_LDFLAGS = -no-undefined -version-info 4:7:1 @LIBTHAI_LIBS@
 
 include_HEADERS = include/sombok_constants.h
 nodist_include_HEADERS = include/sombok.h
--- origsrc/sombok/configure.ac	2025-01-03 22:27:43.028859500 +0900
+++ src/sombok/configure.ac	2025-01-03 22:37:59.059997900 +0900
@@ -120,7 +120,7 @@ AC_ARG_WITH(unicode-version,
   AC_HELP_STRING(--with-unicode-version=VERSION,
   [version of Unicode Standard @<:@default=current version@:>@]),
   UNICODE_VERSION=$withval,UNICODE_VERSION=$DEFAULT_UNICODE_VERSION)
-AS_IF([test -f "lib/$UNICODE_VERSION.c"],
+AS_IF([test -f "${srcdir}/lib/$UNICODE_VERSION.c"],
   AC_MSG_RESULT($UNICODE_VERSION),
   AC_MSG_ERROR(Unknown Unicode version $UNICODE_VERSION.))
 AC_SUBST(UNICODE_VERSION)
