--- origsrc/sombok-sombok-2.4.0/Makefile.am	2015-06-21 10:57:51.000000000 +0900
+++ src/sombok-sombok-2.4.0/Makefile.am	2016-12-08 18:28:44.738132400 +0900
@@ -5,8 +5,8 @@ libsombok_la_SOURCES = lib/break.c lib/c
 if MALLOC_DEBUG
 libsombok_la_SOURCES += src/mymalloc.c
 endif
-libsombok_la_CFLAGS = -I. -Iinclude @LIBTHAI_CFLAGS@
-libsombok_la_LDFLAGS = -version-info 4:7:1 @LIBTHAI_LIBS@
+libsombok_la_CFLAGS = -I. -Iinclude -I$(srcdir)/include @LIBTHAI_CFLAGS@
+libsombok_la_LDFLAGS = -no-undefined -version-info 4:7:1 @LIBTHAI_LIBS@
 
 include_HEADERS = include/sombok_constants.h
 nodist_include_HEADERS = include/sombok.h
@@ -37,7 +37,7 @@ $(html_DATA): stamp-html
 stamp-html: doc/doxyfile $(include_HEADERS) $(nodist_include_HEADERS) \
 	    $(libsombok_la_SOURCES)
 	$(DOXYGEN) doc/doxyfile
-	touch stamp-html	
+	touch stamp-html
 endif
 
 EXTRA_DIST = sombok.spec debian/* \
@@ -47,4 +47,3 @@ EXTRA_DIST = sombok.spec debian/* \
 	data/*.custom data/Rules-*.txt data/LBCLASSES \
 	ARTISTIC ChangeLog.REL1 INSTALL.ja_JP README.ja_JP \
 	UNICODE VERSION
-
--- origsrc/sombok-sombok-2.4.0/configure.ac	2015-06-21 10:57:51.000000000 +0900
+++ src/sombok-sombok-2.4.0/configure.ac	2016-12-08 18:12:39.005167100 +0900
@@ -120,7 +120,7 @@ AC_ARG_WITH(unicode-version,
   AC_HELP_STRING(--with-unicode-version=VERSION,
   [version of Unicode Standard @<:@default=current version@:>@]),
   UNICODE_VERSION=$withval,UNICODE_VERSION=$DEFAULT_UNICODE_VERSION)
-AC_CHECK_FILE([lib/$UNICODE_VERSION.c],
+AC_CHECK_FILE([${srcdir}/lib/$UNICODE_VERSION.c],
   AC_MSG_RESULT($UNICODE_VERSION),
   AC_MSG_ERROR(Unknown Unicode version $UNICODE_VERSION.))
 AC_SUBST(UNICODE_VERSION)
