--- origsrc/libvirt-1.2.12/tests/Makefile.am	2015-01-23 20:46:24.000000000 +0900
+++ src/libvirt-1.2.12/tests/Makefile.am	2015-02-11 20:48:23.763265100 +0900
@@ -546,7 +546,7 @@ qemuxml2argvtest_LDADD = $(qemu_LDADDS)
 qemuxml2argvmock_la_SOURCES = \
 	qemuxml2argvmock.c
 qemuxml2argvmock_la_CFLAGS = $(AM_CFLAGS)
-qemuxml2argvmock_la_LDFLAGS = -module -avoid-version \
+qemuxml2argvmock_la_LDFLAGS = -no-undefined -module -avoid-version \
 	-rpath /evil/libtool/hack/to/force/shared/lib/creation
 
 qemuxml2xmltest_SOURCES = \
@@ -697,7 +697,7 @@ if WITH_BHYVE
 bhyvexml2argvmock_la_SOURCES = \
 	bhyvexml2argvmock.c
 bhyvexml2argvmock_la_CFLAGS = $(AM_CFLAGS)
-bhyvexml2argvmock_la_LDFLAGS = -module -avoid-version \
+bhyvexml2argvmock_la_LDFLAGS = -no-undefined -module -avoid-version \
 			-rpath /evil/libtool/hack/to/force/shared/lib/creation
 
 bhyve_LDADDS = ../src/libvirt_driver_bhyve_impl.la
@@ -896,7 +896,7 @@ virnetserverclienttest_LDADD = $(LDADDS)
 virnetserverclientmock_la_SOURCES = \
 	virnetserverclientmock.c
 virnetserverclientmock_la_CFLAGS = $(AM_CFLAGS)
-virnetserverclientmock_la_LDFLAGS = -module -avoid-version \
+virnetserverclientmock_la_LDFLAGS = -no-undefined -module -avoid-version \
         -rpath /evil/libtool/hack/to/force/shared/lib/creation
 
 if WITH_GNUTLS
@@ -975,7 +975,7 @@ virportallocatortest_LDADD = $(LDADDS)
 libvirportallocatormock_la_SOURCES = \
 	virportallocatortest.c
 libvirportallocatormock_la_CFLAGS = $(AM_CFLAGS) -DMOCK_HELPER=1
-libvirportallocatormock_la_LDFLAGS = -module -avoid-version \
+libvirportallocatormock_la_LDFLAGS = -no-undefined -module -avoid-version \
         -rpath /evil/libtool/hack/to/force/shared/lib/creation
 
 vircgrouptest_SOURCES = \
@@ -985,7 +985,7 @@ vircgrouptest_LDADD = $(LDADDS)
 vircgroupmock_la_SOURCES = \
 	vircgroupmock.c
 vircgroupmock_la_CFLAGS = $(AM_CFLAGS)
-vircgroupmock_la_LDFLAGS = -module -avoid-version \
+vircgroupmock_la_LDFLAGS = -no-undefined -module -avoid-version \
         -rpath /evil/libtool/hack/to/force/shared/lib/creation
 
 vircryptotest_SOURCES = \
@@ -1005,7 +1005,7 @@ virpcimock_la_SOURCES = \
 virpcimock_la_CFLAGS = $(AM_CFLAGS)
 virpcimock_la_LIBADD = $(GNULIB_LIBS) \
 					   ../src/libvirt.la
-virpcimock_la_LDFLAGS = -module -avoid-version \
+virpcimock_la_LDFLAGS = -no-undefined -module -avoid-version \
         -rpath /evil/libtool/hack/to/force/shared/lib/creation
 
 if WITH_LINUX
@@ -1019,13 +1019,13 @@ virnetdevbandwidthtest_LDADD = $(LDADDS)
 
 virusbmock_la_SOURCES = virusbmock.c
 virusbmock_la_CFLAGS = $(AM_CFLAGS)
-virusbmock_la_LDFLAGS = -module -avoid-version \
+virusbmock_la_LDFLAGS = -no-undefined -module -avoid-version \
         -rpath /evil/libtool/hack/to/force/shared/lib/creation
 
 virnetdevbandwidthmock_la_SOURCES = \
 	virnetdevbandwidthmock.c
 virnetdevbandwidthmock_la_CFLAGS = $(AM_CFLAGS)
-virnetdevbandwidthmock_la_LDFLAGS = -module -avoid-version \
+virnetdevbandwidthmock_la_LDFLAGS = -no-undefined -module -avoid-version \
         -rpath /evil/libtool/hack/to/force/shared/lib/creation
 
 else ! WITH_LINUX
@@ -1042,7 +1042,7 @@ virdbustest_LDADD = $(LDADDS) $(DBUS_LIB
 virmockdbus_la_SOURCES = \
 	virmockdbus.c
 virmockdbus_la_CFLAGS = $(AM_CFLAGS) $(DBUS_CFLAGS)
-virmockdbus_la_LDFLAGS = -module -avoid-version \
+virmockdbus_la_LDFLAGS = -no-undefined -module -avoid-version \
         -rpath /evil/libtool/hack/to/force/shared/lib/creation
 
 virpolkittest_SOURCES = \
@@ -1089,7 +1089,7 @@ endif ! WITH_TESTS
 
 libsecurityselinuxhelper_la_SOURCES = \
 	securityselinuxhelper.c
-libsecurityselinuxhelper_la_LDFLAGS = -module -avoid-version \
+libsecurityselinuxhelper_la_LDFLAGS = -no-undefined -module -avoid-version \
         -rpath /evil/libtool/hack/to/force/shared/lib/creation
 
 securityselinuxtest_SOURCES = \
@@ -1162,7 +1162,7 @@ endif WITH_LIBVIRTD
 
 libshunload_la_SOURCES = shunloadhelper.c
 libshunload_la_LIBADD = ../src/libvirt.la
-libshunload_la_LDFLAGS = -module -avoid-version \
+libshunload_la_LDFLAGS = -no-undefined -module -avoid-version \
 	-rpath /evil/libtool/hack/to/force/shared/lib/creation
 
 shunloadtest_SOURCES = \
