--- origsrc/libvirt-1.2.16/src/Makefile.am	2015-05-27 10:26:03.000000000 +0900
+++ src/libvirt-1.2.16/src/Makefile.am	2015-06-18 21:18:28.353848400 +0900
@@ -34,7 +34,7 @@ INCLUDES =	-I../gnulib/lib					\
 		-DIN_LIBVIRT					\
 		-Dabs_topbuilddir="\"$(abs_topbuilddir)\""	\
 		-Dabs_topsrcdir="\"$(abs_topsrcdir)\""		\
-		$(GETTEXT_CPPFLAGS)
+		$(GETTEXT_CPPFLAGS) -I/usr/include/tirpc
 
 AM_CFLAGS =	$(LIBXML_CFLAGS)				\
 		$(WARN_CFLAGS)					\
--- origsrc/libvirt-1.2.16/tests/Makefile.am	2015-04-29 14:24:16.000000000 +0900
+++ src/libvirt-1.2.16/tests/Makefile.am	2015-06-18 21:18:28.369473400 +0900
@@ -47,7 +47,7 @@ AM_CFLAGS = \
 AM_LDFLAGS = \
 	-export-dynamic
 
-MOCKLIBS_LDFLAGS = -module -avoid-version \
+MOCKLIBS_LDFLAGS = -no-undefined -module -avoid-version \
 	-rpath /evil/libtool/hack/to/force/shared/lib/creation \
 	$(MINGW_EXTRA_LDFLAGS)
 
