--- origsrc/libvirt-1.2.9/src/util/virprocess.c	2014-09-16 19:55:43.000000000 +0900
+++ src/libvirt-1.2.9/src/util/virprocess.c	2014-10-24 23:41:51.670249700 +0900
@@ -83,7 +83,7 @@ VIR_LOG_INIT("util.process");
 #endif
 
 #ifndef HAVE_SETNS
-# if defined(__NR_setns) && !defined(WIN32)
+# if defined(__NR_setns) && !defined(WIN32) && !defined(__CYGWIN__)
 #  include <sys/syscall.h>
 
 static inline int setns(int fd, int nstype)
