--- origsrc/fv-src-1.03/Makefile	2012-05-07 10:48:43.000000000 +0900
+++ src/fv-src-1.03/Makefile	2012-08-14 17:38:01.387770800 +0900
@@ -46,14 +46,14 @@ OPT=\
 -Wall -g
 endif
 
-ifeq ("$(shell uname -o)", "Cygwin")
-CXX=g++-3
-CXXFLAGS=\
-$(OPT) -mno-cygwin -mwindows -Inocygwin/inc -DFREEGLUT_STATIC
-LDFLAGS=\
--Lnocygwin/lib -lpng15 -lfreeglut_static -lz -lbz2 -llzma -lpthreadGC2 \
--lglu32 -lopengl32 -lwinmm
-else
+#ifeq ("$(shell uname -o)", "Cygwin")
+#CXX=g++-3
+#CXXFLAGS=\
+#$(OPT) -mno-cygwin -mwindows -Inocygwin/inc -DFREEGLUT_STATIC
+#LDFLAGS=\
+#-Lnocygwin/lib -lpng15 -lfreeglut_static -lz -lbz2 -llzma -lpthreadGC2 \
+#-lglu32 -lopengl32 -lwinmm
+#else
 ifeq ("$(shell uname -o)", "Darwin")
 CXXFLAGS=\
 $(OPT) -I/opt/local/include
@@ -66,19 +66,19 @@ CXXXFLAGS=\
 $(OPT)
 LDFLAGS=\
 -lpng -lz -lbz2 -llzma -lpthread \
--lfreeglut -lGLU -lGL -L/usr/X11R6/lib -lXi -lXmu -lX11 -lXext -lSM \
+-lglut -lGLU -lGL -L/usr/X11R6/lib -lXi -lXmu -lX11 -lXext -lSM \
 -lm 
 endif
-endif
+#endif
 
-ifeq ("$(shell uname -o)", "Cygwin")
-$(TARGET): $(OBJS)
-	windres -i resource.rc -o resource.o
-	$(CXX) $(CXXFLAGS) -o $@ $(OBJS) resource.o $(LDFLAGS)
-else
+#ifeq ("$(shell uname -o)", "Cygwin")
+#$(TARGET): $(OBJS)
+#	windres -i resource.rc -o resource.o
+#	$(CXX) $(CXXFLAGS) -o $@ $(OBJS) resource.o $(LDFLAGS)
+#else
 $(TARGET): $(OBJS)
 	$(CXX) $(CXXFLAGS) -o $@ $(OBJS) $(LDFLAGS)
-endif
+#endif
 
 allclean: clean
 	rm -f $(TARGET) $(TARGET).exe
