--- origsrc/c++-gtk-utils-2.2.18/Makefile.am	2017-10-03 01:36:54.000000000 +0900
+++ src/c++-gtk-utils-2.2.18/Makefile.am	2019-06-21 11:31:58.084731300 +0900
@@ -3,7 +3,7 @@
 # for glib's testing framework
 include $(top_srcdir)/Makefile.decl
 
-SUBDIRS = c++-gtk-utils po docs @COND_TESTS@
+SUBDIRS = c++-gtk-utils po @COND_TESTS@
 DIST_SUBDIRS = c++-gtk-utils po docs tests
 
 .PHONY: docs
@@ -13,8 +13,8 @@ docs:
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = c++-gtk-utils-@CGU_GTK_SUFFIX@-2.2.pc
 
-pkgdocdir = $(docdir)/2.2
-pkgdoc_DATA = README BUGS COPYING NEWS PORTING-TO-2.2
+# pkgdocdir = $(docdir)/2.2
+# pkgdoc_DATA = README BUGS COPYING NEWS PORTING-TO-2.2
 
 EXTRA_DIST +=	m4/ChangeLog m4/*.m4 config.rpath compile BUGS		\
 		configure-gtk2 configure-gtk3 configure-gtk4		\
--- origsrc/c++-gtk-utils-2.2.18/acinclude.m4	2019-05-27 03:17:58.000000000 +0900
+++ src/c++-gtk-utils-2.2.18/acinclude.m4	2019-06-21 11:31:58.100356400 +0900
@@ -105,14 +105,14 @@ AC_DEFUN([AC_CHECK_COMPILERS],
     ])
   fi
 
-  AC_CHECK_COMPILER_FLAG(std=c++11,
+  AC_CHECK_COMPILER_FLAG(std=gnu++11,
     [
-      CXXFLAGS="$CXXFLAGS -std=c++11"
+      CXXFLAGS="$CXXFLAGS -std=gnu++11"
     ],
     [
-    AC_CHECK_COMPILER_FLAG(std=c++0x,
+    AC_CHECK_COMPILER_FLAG(std=gnu++0x,
       [
-        CXXFLAGS="$CXXFLAGS -std=c++0x"
+        CXXFLAGS="$CXXFLAGS -std=gnu++0x"
     ])
   ])
 
--- origsrc/c++-gtk-utils-2.2.18/c++-gtk-utils/Makefile.am	2016-10-30 18:44:06.000000000 +0900
+++ src/c++-gtk-utils-2.2.18/c++-gtk-utils/Makefile.am	2019-06-21 11:31:58.272245800 +0900
@@ -36,13 +36,13 @@ pkginclude_HEADERS =  fdstream.h fdstrea
 
 AM_CXXFLAGS = -D_XOPEN_SOURCE=600 @PTHREAD_FLAG_CXX@ @GTK_UNIX_PRINT_CFLAGS@	\
 	      @GTK_CFLAGS@ @GIO_CFLAGS@ @DBUS_GLIB_CFLAGS@ @GTHREAD_CFLAGS@	\
-	      @GLIB_CFLAGS@ -I..
+	      @GLIB_CFLAGS@ -I.. -I$(top_srcdir)
 
 libcxx_gtk_utils_@CGU_GTK_SUFFIX@_2_2_la_LIBADD = @PTHREAD_FLAG_CXX@ @GTK_UNIX_PRINT_LIBS@ \
 						  @GTK_LIBS@ @GIO_LIBS@ @DBUS_GLIB_LIBS@   \
 						  @GTHREAD_LIBS@ @GLIB_LIBS@ @LTLIBINTL@
 
-libcxx_gtk_utils_@CGU_GTK_SUFFIX@_2_2_la_LDFLAGS = -version-info @CXX_GTK_UTILS_SO_VERSION@
+libcxx_gtk_utils_@CGU_GTK_SUFFIX@_2_2_la_LDFLAGS = -no-undefined -version-info @CXX_GTK_UTILS_SO_VERSION@
 
 EXTRA_DIST +=	LGPL.TXT doxygen.main doxygen.main.in cgu_config.h.in \
 		prog-presenter/make_bindings.sh prog-presenter/prog-presenter.xml
--- origsrc/c++-gtk-utils-2.2.18/docs/Makefile.am	2018-10-22 23:01:58.000000000 +0900
+++ src/c++-gtk-utils-2.2.18/docs/Makefile.am	2019-06-21 11:31:58.303493400 +0900
@@ -9,7 +9,7 @@ docs:
 
 install-data-hook:
 	install -d $(DESTDIR)$(htmldir)/2.2/html
-	install -m a+r,u+w html/* $(DESTDIR)$(htmldir)/2.2/html/
+	install -m a+r,u+w $(srcdir)/html/* $(DESTDIR)$(htmldir)/2.2/html/
 
 uninstall-hook:
 	for files in `ls html`; do rm -f $(DESTDIR)$(htmldir)/2.2/html/$$files; done
--- origsrc/c++-gtk-utils-2.2.18/tests/Makefile.am	2016-11-02 21:21:16.000000000 +0900
+++ src/c++-gtk-utils-2.2.18/tests/Makefile.am	2019-06-21 11:31:58.319165700 +0900
@@ -11,7 +11,7 @@ libcxx_gtk_utils = $(top_builddir)/c++-g
 
 AM_CXXFLAGS =	-I.. @PTHREAD_FLAG_CXX@ @GTK_UNIX_PRINT_CFLAGS@ \
 		@GTK_CFLAGS@ @GIO_CFLAGS@ @GTHREAD_CFLAGS@	\
-		@GLIB_CFLAGS@
+		@GLIB_CFLAGS@ -I$(top_srcdir)
 
 LDADD =		$(libcxx_gtk_utils) @PTHREAD_FLAG_CXX@ @GTK_UNIX_PRINT_LIBS@	\
 		@GTK_LIBS@ @GIO_LIBS@ @DBUS_GLIB_LIBS@ @GTHREAD_LIBS@		\
