diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x vbindiff.1 origsrc/vbindiff-3.0_beta5/Makefile.am src/vbindiff-3.0_beta5/Makefile.am
--- origsrc/vbindiff-3.0_beta5/Makefile.am	2017-09-11 11:12:09.000000000 +0900
+++ src/vbindiff-3.0_beta5/Makefile.am	2018-05-10 12:52:36.222445800 +0900
@@ -53,7 +53,7 @@ AM_CPPFLAGS = -I$(srcdir)/curses
 AM_CFLAGS = -Wall -D_FILE_OFFSET_BITS=64
 AM_CXXFLAGS = $(AM_CFLAGS)
 
-GENFILE = perl tools/genfile.pl
+GENFILE = perl $(top_srcdir)/tools/genfile.pl
 
 README : tools/ReadMe.tt configure.ac
 	$(GENFILE) $<
