--- origsrc/libdbus-c++-0.9.0/src/Makefile.am	2011-11-29 06:20:54.000000000 +0900
+++ src/libdbus-c++-0.9.0/src/Makefile.am	2013-01-01 19:02:34.604522500 +0900
@@ -31,12 +31,16 @@ libdbus_c___1_la_CXXFLAGS = \
 libdbus_c___1_la_LIBADD = \
 	$(dbus_LIBS)
 
+libdbus_c___1_la_LDFLAGS = \
+	-no-undefined
+
 AM_CPPFLAGS = \
 	$(dbus_CFLAGS) \
 	$(glib_CFLAGS) \
 	$(ecore_CFLAGS)
 
 SUBDIRS = \
+	. \
 	integration
 
 HEADER_DIR  = $(top_srcdir)/include/dbus-c++
--- origsrc/libdbus-c++-0.9.0/src/integration/ecore/Makefile.am	2011-11-29 06:22:50.000000000 +0900
+++ src/libdbus-c++-0.9.0/src/integration/ecore/Makefile.am	2013-01-01 19:01:45.429717800 +0900
@@ -12,7 +12,11 @@ libdbus_c___ecore_1_la_CXXFLAGS = \
 
 libdbus_c___ecore_1_la_LIBADD = \
 	$(dbus_LIBS) \
-	$(ecore_LIBS)
+	$(ecore_LIBS) \
+	$(top_builddir)/src/libdbus-c++-1.la
+
+libdbus_c___ecore_1_la_LDFLAGS = \
+	-no-undefined
 
 AM_CPPFLAGS = \
 	$(dbus_CFLAGS) \
--- origsrc/libdbus-c++-0.9.0/src/integration/glib/Makefile.am	2011-11-29 06:22:22.000000000 +0900
+++ src/libdbus-c++-0.9.0/src/integration/glib/Makefile.am	2013-01-01 19:07:03.827178800 +0900
@@ -12,7 +12,11 @@ libdbus_c___glib_1_la_CXXFLAGS = \
 
 libdbus_c___glib_1_la_LIBADD = \
 	$(dbus_LIBS) \
-	$(glib_LIBS)
+	$(glib_LIBS) \
+	$(top_builddir)/src/libdbus-c++-1.la
+
+libdbus_c___glib_1_la_LDFLAGS = \
+	-no-undefined
 
 AM_CPPFLAGS = \
 	$(dbus_CFLAGS) \
--- origsrc/libdbus-c++-0.9.0/src/pipe.cpp	2011-11-30 07:16:31.000000000 +0900
+++ src/libdbus-c++-0.9.0/src/pipe.cpp	2013-01-01 19:06:08.391631900 +0900
@@ -25,13 +25,14 @@
 #include <config.h>
 #endif
 
+#include <unistd.h>
+
 /* Project */
 #include <dbus-c++/pipe.h>
 #include <dbus-c++/util.h>
 #include <dbus-c++/error.h>
 
 /* STD */
-#include <unistd.h>
 #include <sys/poll.h>
 #include <fcntl.h>
 #include <errno.h>
