--- origsrc/mpir-mpir-3.0.0/Makefile.am	2017-03-01 20:09:48.000000000 +0900
+++ src/mpir-mpir-3.0.0/Makefile.am	2025-01-31 10:09:54.379053000 +0900
@@ -270,3 +270,6 @@ BUILT_SOURCES += mpn/dummy1.c
 #
 dist-hook:
 	-find $(distdir) \( -name CVS -type d \) -o -name "*~" -o -name ".#*" -o -name ".svn" | xargs rm -rf
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = mpir.pc mpirxx.pc
--- origsrc/mpir-mpir-3.0.0/configure.ac	2017-03-01 20:09:48.000000000 +0900
+++ src/mpir-mpir-3.0.0/configure.ac	2025-01-31 10:09:54.379053000 +0900
@@ -3038,6 +3038,7 @@ if test $enable_gmpcompat = yes; then
     AC_CONFIG_FILES(gmpxx.h:mpirxx.h)
   fi
 fi
+AC_CONFIG_FILES(mpir.pc mpirxx.pc)
 
 AC_OUTPUT
 
--- origsrc/mpir-mpir-3.0.0/mpir.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/mpir-mpir-3.0.0/mpir.pc.in	2025-01-31 10:09:54.379053000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mpir
+Description: Multiple Precision Integers and Rationals
+Version: @VERSION@
+Libs: -L${libdir} -lmpir
+Cflags: -I${includedir}
--- origsrc/mpir-mpir-3.0.0/mpirxx.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/mpir-mpir-3.0.0/mpirxx.pc.in	2025-01-31 10:09:54.379053000 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mpirxx
+Description: Multiple Precision Integers and Rationals (C++)
+Version: @VERSION@
+Libs: -L${libdir} -lmpirxx
+Cflags: -I${includedir}
+Requires: mpir
