--- origsrc/sysrepo-2.2.105/CMakeLists.txt	2023-08-16 17:57:36.000000000 +0900
+++ src/sysrepo-2.2.105/CMakeLists.txt	2023-08-28 23:23:52.943006100 +0900
@@ -441,7 +441,7 @@ configure_file("${PROJECT_SOURCE_DIR}/sr
 # installation
 install(DIRECTORY "${PROJECT_SOURCE_DIR}/modules/" DESTINATION ${YANG_MODULE_DIR} FILES_MATCHING PATTERN "*.yang")
 
-install(TARGETS sysrepo DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS sysrepo RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(FILES ${PROJECT_SOURCE_DIR}/src/sysrepo.h ${PROJECT_SOURCE_DIR}/src/sysrepo_types.h
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 install(FILES ${PROJECT_SOURCE_DIR}/src/plugins_datastore.h ${PROJECT_BINARY_DIR}/version.h
--- origsrc/sysrepo-2.2.105/src/common.c	2023-08-16 17:57:36.000000000 +0900
+++ src/sysrepo-2.2.105/src/common.c	2023-08-28 23:23:52.943006100 +0900
@@ -56,6 +56,10 @@
 #include "shm_sub.h"
 #include "sysrepo.h"
 
+#if defined(__CYGWIN__)
+#define pthread_mutex_consistent(mutex) (EINVAL)
+#endif
+
 /**
  * @brief Internal DS plugin array.
  */
@@ -2203,11 +2207,13 @@ _sr_mutex_init(pthread_mutex_t *lock, in
             return err_info;
         }
 
+#if defined(PTHREAD_MUTEX_ROBUST)
         if (robust && (ret = pthread_mutexattr_setrobust(&attr, PTHREAD_MUTEX_ROBUST))) {
             pthread_mutexattr_destroy(&attr);
             sr_errinfo_new(&err_info, SR_ERR_SYS, "Changing pthread attr failed (%s).", strerror(ret));
             return err_info;
         }
+#endif
 
         if ((ret = pthread_mutex_init(lock, &attr))) {
             pthread_mutexattr_destroy(&attr);
