--- origsrc/yaml-cpp-yaml-cpp-0.7.0/CMakeLists.txt	2021-07-11 00:53:22.000000000 +0900
+++ src/yaml-cpp-yaml-cpp-0.7.0/CMakeLists.txt	2021-09-15 18:26:34.993195900 +0900
@@ -147,14 +147,25 @@ if (YAML_CPP_INSTALL)
 	install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 		FILES_MATCHING PATTERN "*.h")
-  install(EXPORT yaml-cpp-targets
-    DESTINATION "${CMAKE_INSTALL_DATADIR}/cmake/yaml-cpp")
-	install(FILES
-		"${PROJECT_BINARY_DIR}/yaml-cpp-config.cmake"
-		"${PROJECT_BINARY_DIR}/yaml-cpp-config-version.cmake"
-    DESTINATION "${CMAKE_INSTALL_DATADIR}/cmake/yaml-cpp")
-  install(FILES "${PROJECT_BINARY_DIR}/yaml-cpp.pc"
-    DESTINATION ${CMAKE_INSTALL_DATADIR}/pkgconfig)
+  if (CYGWIN)
+    install(EXPORT yaml-cpp-targets
+      DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/yaml-cpp")
+    install(FILES
+      "${PROJECT_BINARY_DIR}/yaml-cpp-config.cmake"
+      "${PROJECT_BINARY_DIR}/yaml-cpp-config-version.cmake"
+      DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/yaml-cpp")
+    install(FILES "${PROJECT_BINARY_DIR}/yaml-cpp.pc"
+      DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+  else()
+    install(EXPORT yaml-cpp-targets
+      DESTINATION "${CMAKE_INSTALL_DATADIR}/cmake/yaml-cpp")
+    install(FILES
+      "${PROJECT_BINARY_DIR}/yaml-cpp-config.cmake"
+      "${PROJECT_BINARY_DIR}/yaml-cpp-config-version.cmake"
+      DESTINATION "${CMAKE_INSTALL_DATADIR}/cmake/yaml-cpp")
+    install(FILES "${PROJECT_BINARY_DIR}/yaml-cpp.pc"
+      DESTINATION ${CMAKE_INSTALL_DATADIR}/pkgconfig)
+  endif()
 endif()
 
 if(YAML_CPP_BUILD_TESTS)
