--- origsrc/yaml-cpp-0.8.0/test/CMakeLists.txt	2023-08-11 04:27:56.000000000 +0900
+++ src/yaml-cpp-0.8.0/test/CMakeLists.txt	2025-09-10 09:41:14.700308000 +0900
@@ -4,11 +4,14 @@ set(gtest_force_shared_crt ON CACHE BOOL
 set(BUILD_MOCK ON CACHE BOOL "" FORCE)
 set(CMAKE_POLICY_DEFAULT_CMP0048 NEW)
 
-add_subdirectory(
-  "${CMAKE_CURRENT_SOURCE_DIR}/gtest-1.11.0"
-  "${CMAKE_CURRENT_BINARY_DIR}/prefix")
+find_package(GTest)
+if (NOT GTEST_FOUND)
+  add_subdirectory(
+    "${CMAKE_CURRENT_SOURCE_DIR}/gtest-1.11.0"
+    "${CMAKE_CURRENT_BINARY_DIR}/prefix")
 
-include_directories(SYSTEM "${CMAKE_CURRENT_SOURCE_DIR}/gtest-1.11.0/googletest/include")
+  include_directories(SYSTEM "${CMAKE_CURRENT_SOURCE_DIR}/gtest-1.11.0/googletest/include")
+endif()
 
 set(test-new-api-pattern "new-api/*.cpp")
 set(test-source-pattern "*.cpp" "integration/*.cpp" "node/*.cpp")
@@ -38,6 +41,7 @@ target_link_libraries(yaml-cpp-tests
   PRIVATE
     Threads::Threads
     yaml-cpp
+    ${GTEST_LIBRARIES}
     gmock)
 
 set_property(TARGET yaml-cpp-tests PROPERTY CXX_STANDARD_REQUIRED ON)
