--- origsrc/urdfdom_headers-1.0.0/CMakeLists.txt	2016-07-21 02:47:51.000000000 +0900
+++ src/urdfdom_headers-1.0.0/CMakeLists.txt	2017-12-22 13:46:23.136000000 +0900
@@ -25,6 +25,8 @@ add_subdirectory(urdf_exception)
 
 if(WIN32 AND NOT CYGWIN)
   set(CMAKE_CONFIG_INSTALL_DIR CMake)
+elseif(CYGWIN)
+  set(CMAKE_CONFIG_INSTALL_DIR ${CMAKE_INSTALL_FULL_LIBDIR}/cmake/${PROJECT_NAME}/)
 else()
   set(CMAKE_CONFIG_INSTALL_DIR ${CMAKE_INSTALL_FULL_LIBDIR}/${PROJECT_NAME}/cmake/)
 endif()
