diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/urdfdom_headers-0.4.0/CMakeLists.txt src/urdfdom_headers-0.4.0/CMakeLists.txt
--- origsrc/urdfdom_headers-0.4.0/CMakeLists.txt	2015-12-10 03:36:05.000000000 +0900
+++ src/urdfdom_headers-0.4.0/CMakeLists.txt	2016-02-06 13:42:58.809510200 +0900
@@ -24,7 +24,7 @@ add_subdirectory(urdf_exception)
 if(WIN32 AND NOT CYGWIN)
   set(CMAKE_CONFIG_INSTALL_DIR CMake)
 else()
-  set(CMAKE_CONFIG_INSTALL_DIR share/${PROJECT_NAME}/cmake/)
+  set(CMAKE_CONFIG_INSTALL_DIR lib/cmake/${PROJECT_NAME})
 endif()
 
 set(PACKAGE_NAME ${PROJECT_NAME})
