--- origsrc/urdfdom_headers-2.0.0/CMakeLists.txt	2025-07-02 02:51:04.000000000 +0900
+++ src/urdfdom_headers-2.0.0/CMakeLists.txt	2025-09-03 18:47:28.333757100 +0900
@@ -30,6 +30,8 @@ install(DIRECTORY include/ DESTINATION $
 
 if(WIN32 AND NOT CYGWIN)
   set(CMAKE_CONFIG_INSTALL_DIR CMake)
+elseif(CYGWIN)
+  set(CMAKE_CONFIG_INSTALL_DIR ${CMAKE_INSTALL_FULL_LIBDIR}/cmake/${PROJECT_NAME}/)
 else()
   set(CMAKE_CONFIG_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/${PROJECT_NAME}/cmake)
 endif()
