--- origsrc/libgrp-1.0.3/CMakeLists.txt	2014-06-02 14:50:16.000000000 +0900
+++ src/libgrp-1.0.3/CMakeLists.txt	2024-05-10 16:46:18.967636400 +0900
@@ -220,7 +220,7 @@ include(cmake/InstallPkgConfigFile.cmake
 install_pkg_config_file(libgrp
 	DESCRIPTION "A library to convert Blizzard GRP images to common image formats"
     CFLAGS
-    LIBS ${Magick++_LIBRARIES}
+    LIBS -lgrp ${Magick++_LIBRARIES}
     REQUIRES
     VERSION ${LIBGRP_MAJOR_VERSION}.${LIBGRP_MINOR_VERSION}.${LIBGRP_PATCH_VERSION})
 
@@ -235,7 +235,7 @@ add_custom_target(uninstall
 
 install(DIRECTORY Source/ DESTINATION include/libgrp
           FILES_MATCHING PATTERN "*.hpp")
-install(TARGETS grp LIBRARY DESTINATION lib ARCHIVE DESTINATION lib FRAMEWORK DESTINATION /Library/Frameworks)
+install(TARGETS grp RUNTIME DESTINATION bin ARCHIVE DESTINATION lib FRAMEWORK DESTINATION /Library/Frameworks)
 
 #Tell the developer the current build options
 message("\n\nlibgrp Compile Options\n Unit Tests: ${UNITTESTS} [-DUNITESTS=(off-on)]")
@@ -243,4 +243,4 @@ message(" Verbose Level: ${VERBOSE} [-DV
 if(APPLE)
 message(" Compile Framework: ${MAKEFRAMEWORK} [-DMAKEFRAMEWORK=(off-on)")
 endif()
-message(" Sample Code: ${SAMPLECODE} [-DSAMPLECODE=(off-on)\n")
\ No newline at end of file
+message(" Sample Code: ${SAMPLECODE} [-DSAMPLECODE=(off-on)\n")
