--- origsrc/win32pe-0.1.0/CMakeLists.txt	2017-12-03 04:27:56.000000000 +0900
+++ src/win32pe-0.1.0/CMakeLists.txt	2026-03-29 08:33:04.639064900 +0900
@@ -17,3 +17,10 @@ if(BUILD_TESTS)
     enable_testing()
     add_subdirectory(tests)
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/win32pe.pc.in ${CMAKE_BINARY_DIR}/win32pe.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/win32pe.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/win32pe-0.1.0/win32pe.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/win32pe-0.1.0/win32pe.pc.in	2026-03-29 08:33:14.826709600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: win32pe
+Description: C++ library for working with Win32 PE files
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lwin32pe
+Cflags: -I${includedir}
