--- origsrc/cmigemo-default-src/config.mk	2011-02-27 19:45:40.000000000 +0900
+++ src/cmigemo-default-src/config.mk	2011-03-20 11:49:06.164343800 +0900
@@ -13,14 +13,14 @@ outdir = ./build/
 ##############################################################################
 # インストールディレクトリの設定
 #
-prefix		= /usr/local
+prefix		= $(DESTDIR)/usr
 bindir		= $(prefix)/bin
 libdir		= $(prefix)/lib
 incdir		= $(prefix)/include
 # 警告: $(dictdir)と$(docdir)はアンインストール実行時にディレクトリごと消去
 # されます。
 dictdir		= $(prefix)/share/migemo
-docdir		= $(prefix)/doc/migemo
+docdir		= $(prefix)/share/doc/migemo
 
 ##############################################################################
 # コマンド設定
@@ -38,8 +38,8 @@ GUNZIP		= gzip -d
 FILTER_CP932	= qkc -q -u -s
 FILTER_EUCJP	= qkc -q -u -e
 FILTER_UTF8	= iconv -t utf-8 -f cp932
-#FILTER_CP932	= nkf -s
-#FILTER_EUCJP	= nkf -e
+FILTER_CP932	= nkf -s
+FILTER_EUCJP	= nkf -e
 INSTALL		= /usr/bin/install -c
 #INSTALL	= /usr/ucb/install -c
 INSTALL_PROGRAM	= $(INSTALL) -m 755
