--- origsrc/cpluff-0.1.3/test/plugins-source/callbackcounter/Makefile.am	2007-03-24 12:25:20.000000000 +0900
+++ src/cpluff-0.1.3/test/plugins-source/callbackcounter/Makefile.am	2012-01-20 22:50:46.772326600 +0900
@@ -14,4 +14,4 @@ plugin_LTLIBRARIES = libruntime.la
 plugin_DATA = plugin.xml
 
 libruntime_la_SOURCES = callbackcounter.c callbackcounter.h
-libruntime_la_LDFLAGS = -module -avoid-version
+libruntime_la_LDFLAGS = -module -avoid-version -no-undefined
--- origsrc/cpluff-0.1.3/test/plugins-source/symprovider/Makefile.am	2007-03-24 12:25:30.000000000 +0900
+++ src/cpluff-0.1.3/test/plugins-source/symprovider/Makefile.am	2012-01-20 22:50:56.390233800 +0900
@@ -14,4 +14,4 @@ plugin_LTLIBRARIES = libruntime.la
 plugin_DATA = plugin.xml
 
 libruntime_la_SOURCES = symprovider.c
-libruntime_la_LDFLAGS = -module -avoid-version
+libruntime_la_LDFLAGS = -module -avoid-version -no-undefined
--- origsrc/cpluff-0.1.3/test/plugins-source/symuser/Makefile.am	2007-03-24 12:25:41.000000000 +0900
+++ src/cpluff-0.1.3/test/plugins-source/symuser/Makefile.am	2012-01-20 22:51:04.727265800 +0900
@@ -14,4 +14,4 @@ plugin_LTLIBRARIES = libruntime.la
 plugin_DATA = plugin.xml
 
 libruntime_la_SOURCES = symuser.c
-libruntime_la_LDFLAGS = -module -avoid-version
+libruntime_la_LDFLAGS = -module -avoid-version -no-undefined
