--- origsrc/cpluff-0.1.4/Makefile.am	2016-01-01 23:44:49.000000000 +0900
+++ src/cpluff-0.1.4/Makefile.am	2018-03-27 14:04:56.825893100 +0900
@@ -63,3 +63,6 @@ clean-local:
 	test ! -f examples/Makefile || (cd examples && $(MAKE) $(AM_MAKEFLAGS) clean)
 
 .PHONY: doc examples examples-install examples-clean
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/cpluff-0.1.4/configure.ac	2016-01-02 06:06:44.000000000 +0900
+++ src/cpluff-0.1.4/configure.ac	2018-03-27 14:04:56.841515900 +0900
@@ -280,7 +280,7 @@ AC_SUBST(CPLUFF_LOADER)
 
 # Output Makefiles
 # ----------------
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([cpluff.pc Makefile
 libcpluff/Makefile
 libcpluff/cpluffdef.h
 libcpluff/docsrc/Makefile
--- origsrc/cpluff-0.1.4/cpluff.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/cpluff-0.1.4/cpluff.pc.in	2018-03-27 14:04:56.794893500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: cpluff
+Description: Plug-in framework for C
+Version: @VERSION@
+Libs: -L${libdir} -lcpluff
+Cflags: -I${includedir}
--- origsrc/cpluff-0.1.4/test/plugins-source/callbackcounter/Makefile.am	2016-01-01 22:31:20.000000000 +0900
+++ src/cpluff-0.1.4/test/plugins-source/callbackcounter/Makefile.am	2018-03-27 13:46:50.060729900 +0900
@@ -14,4 +14,4 @@ plugin_LTLIBRARIES = libruntime.la
 plugin_DATA = plugin.xml
 
 libruntime_la_SOURCES = callbackcounter.c callbackcounter.h
-libruntime_la_LDFLAGS = -module -avoid-version
+libruntime_la_LDFLAGS = -no-undefined -module -avoid-version
--- origsrc/cpluff-0.1.4/test/plugins-source/symprovider/Makefile.am	2016-01-01 22:31:20.000000000 +0900
+++ src/cpluff-0.1.4/test/plugins-source/symprovider/Makefile.am	2018-03-27 13:46:50.077180500 +0900
@@ -14,4 +14,4 @@ plugin_LTLIBRARIES = libruntime.la
 plugin_DATA = plugin.xml
 
 libruntime_la_SOURCES = symprovider.c
-libruntime_la_LDFLAGS = -module -avoid-version
+libruntime_la_LDFLAGS = -no-undefined -module -avoid-version
--- origsrc/cpluff-0.1.4/test/plugins-source/symuser/Makefile.am	2016-01-01 22:31:20.000000000 +0900
+++ src/cpluff-0.1.4/test/plugins-source/symuser/Makefile.am	2018-03-27 13:46:50.091937600 +0900
@@ -14,4 +14,4 @@ plugin_LTLIBRARIES = libruntime.la
 plugin_DATA = plugin.xml
 
 libruntime_la_SOURCES = symuser.c
-libruntime_la_LDFLAGS = -module -avoid-version
+libruntime_la_LDFLAGS = -no-undefined -module -avoid-version
