--- origsrc/gwydump-2.1/Makefile	2011-02-04 09:44:43.000000000 +0900
+++ src/gwydump-2.1/Makefile	2024-12-09 08:13:45.173680600 +0900
@@ -17,7 +17,7 @@ GLIB = glib-2.0
 
 DEPS_CFLAGS := $(shell $(PKGCONFIG) $(GLIB) --cflags)
 DEPS_LDFLAGS := $(shell $(PKGCONFIG) $(GLIB) --libs)
-EXTRA_CFLAGS = -Wall -O2 -static -DVERSION=\"$(VERSION)\" -DPACKAGE=\"$(PACKAGE)\"
+EXTRA_CFLAGS = -Wall -O2 -DVERSION=\"$(VERSION)\" -DPACKAGE=\"$(PACKAGE)\"
 EXTRA_LDFLAGS =
 OBJECTS = $(SOURCES:.c=.o)
 
