--- origsrc/openqube-0.1.0/CMakeLists.txt	2012-01-04 23:40:07.000000000 +0900
+++ src/openqube-0.1.0/CMakeLists.txt	2015-04-09 21:10:56.607530600 +0900
@@ -35,3 +35,11 @@ configure_file("${OpenQube_SOURCE_DIR}/c
   "${OpenQube_BINARY_DIR}/OpenQubeConfigVersion.cmake" @ONLY)
 install(FILES "${OpenQube_BINARY_DIR}/OpenQubeConfigVersion.cmake"
   DESTINATION "${LIB_INSTALL_DIR}/cmake/openqube")
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "${OpenQube_VERSION}")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/openqube.pc.in ${CMAKE_BINARY_DIR}/openqube.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/openqube.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/openqube-0.1.0/openqube.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/openqube-0.1.0/openqube.pc.in	2015-04-09 21:10:00.446360800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: openqube
+Description: Library for reading quantum chemistry data files
+Version: @VERSION@
+Libs: -L${libdir} -lOpenQube
+Cflags: -I${includedir}
--- origsrc/openqube-0.1.0/src/CMakeLists.txt	2012-01-04 23:40:07.000000000 +0900
+++ src/openqube-0.1.0/src/CMakeLists.txt	2015-04-09 20:58:01.804448100 +0900
@@ -48,7 +48,8 @@ target_link_libraries(OpenQube ${QT_QTCO
 # Install the library
 install(TARGETS OpenQube
   EXPORT "OpenQubeTargets"
-  DESTINATION "${LIB_INSTALL_DIR}")
+  RUNTIME DESTINATION bin
+  ARCHIVE DESTINATION "${LIB_INSTALL_DIR}")
 install(EXPORT OpenQubeTargets DESTINATION "${LIB_INSTALL_DIR}/cmake/openqube")
 # Install the headers for the public API
 install(FILES ${openqube_HDRS} DESTINATION "${INC_INSTALL_DIR}/openqube")
