--- origsrc/zecwallet-0.8.1/zec-qt-wallet.pro	2019-12-04 04:12:19.000000000 +0900
+++ src/zecwallet-0.8.1/zec-qt-wallet.pro	2019-12-13 11:05:52.739929200 +0900
@@ -133,19 +133,21 @@ QMAKE_INFO_PLIST = res/Info.plist
 win32: RC_ICONS = res/icon.ico
 ICON = res/logo.icns
 
-libsodium.target = $$PWD/res/libsodium.a
-libsodium.commands = res/libsodium/buildlibsodium.sh
+# libsodium.target = $$PWD/res/libsodium.a
+# libsodium.commands = res/libsodium/buildlibsodium.sh
 
-QMAKE_EXTRA_TARGETS += libsodium
-QMAKE_CLEAN += res/libsodium.a
+# QMAKE_EXTRA_TARGETS += libsodium
+# QMAKE_CLEAN += res/libsodium.a
 
 # Default rules for deployment.
 qnx: target.path = /tmp/$${TARGET}/bin
+cygwin: target.path = /usr/bin
 else: unix:!android: target.path = /opt/$${TARGET}/bin
 !isEmpty(target.path): INSTALLS += target
 
 win32:CONFIG(release, debug|release): LIBS += -L$$PWD/res/ -llibsodium
 else:win32:CONFIG(debug, debug|release): LIBS += -L$$PWD/res/ -llibsodiumd
+else:cygwin: LIBS += -lsodium
 else:unix: LIBS += -L$$PWD/res/ -lsodium
 
 INCLUDEPATH += $$PWD/res
@@ -155,6 +157,7 @@ win32-g++:CONFIG(release, debug|release)
 else:win32-g++:CONFIG(debug, debug|release): PRE_TARGETDEPS += $$PWD/res/liblibsodium.a
 else:win32:!win32-g++:CONFIG(release, debug|release): PRE_TARGETDEPS += $$PWD/res/libsodium.lib
 else:win32:!win32-g++:CONFIG(debug, debug|release): PRE_TARGETDEPS += $$PWD/res/libsodiumd.lib
+else:cygwin: PRE_TARGETDEPS +=
 else:unix: PRE_TARGETDEPS += $$PWD/res/libsodium.a
 
 DISTFILES +=
