--- origsrc/zecwallet-0.7.8/zec-qt-wallet.pro	2019-09-06 01:59:07.000000000 +0900
+++ src/zecwallet-0.7.8/zec-qt-wallet.pro	2019-09-25 23:19:02.172778600 +0900
@@ -25,6 +25,7 @@ DEFINES += \
     QT_DEPRECATED_WARNINGS
 
 INCLUDEPATH  += src/3rdparty/
+INCLUDEPATH += src
 
 RESOURCES     = application.qrc
 
@@ -128,19 +129,21 @@ QMAKE_INFO_PLIST = res/Info.plist
 win32: RC_ICONS = res/icon.ico
 ICON = res/logo.icns
 
-libsodium.target = $$PWD/res/libsodium.a
-libsodium.commands = res/libsodium/buildlibsodium.sh
+# libsodium.target = $$PWD/res/libsodium.a
+# libsodium.commands = res/libsodium/buildlibsodium.sh
 
-QMAKE_EXTRA_TARGETS += libsodium
-QMAKE_CLEAN += res/libsodium.a
+# QMAKE_EXTRA_TARGETS += libsodium
+# QMAKE_CLEAN += res/libsodium.a
 
 # Default rules for deployment.
 qnx: target.path = /tmp/$${TARGET}/bin
+cygwin: target.path = /usr/bin
 else: unix:!android: target.path = /opt/$${TARGET}/bin
 !isEmpty(target.path): INSTALLS += target
 
 win32:CONFIG(release, debug|release): LIBS += -L$$PWD/res/ -llibsodium
 else:win32:CONFIG(debug, debug|release): LIBS += -L$$PWD/res/ -llibsodiumd
+else:cygwin: LIBS += -lsodium
 else:unix: LIBS += -L$$PWD/res/ -lsodium
 
 INCLUDEPATH += $$PWD/res
@@ -150,4 +153,5 @@ win32-g++:CONFIG(release, debug|release)
 else:win32-g++:CONFIG(debug, debug|release): PRE_TARGETDEPS += $$PWD/res/liblibsodium.a
 else:win32:!win32-g++:CONFIG(release, debug|release): PRE_TARGETDEPS += $$PWD/res/libsodium.lib
 else:win32:!win32-g++:CONFIG(debug, debug|release): PRE_TARGETDEPS += $$PWD/res/libsodiumd.lib
+else:cygwin: PRE_TARGETDEPS +=
 else:unix: PRE_TARGETDEPS += $$PWD/res/libsodium.a
