--- origsrc/libnxml-0.18.5/src/Makefile.am	2024-02-21 22:39:18.000000000 +0900
+++ src/libnxml-0.18.5/src/Makefile.am	2024-08-30 17:51:36.602897500 +0900
@@ -14,7 +14,7 @@ libnxml_la_SOURCES = \
 	nxml_tools.c \
 	nxml_namespace.c
 
-libnxml_la_LDFLAGS = -version-info @LIBNXML_VERSION_INFO@
+libnxml_la_LDFLAGS = -no-undefined -version-info @LIBNXML_VERSION_INFO@
 
 EXTRA_DIST = nxml.h.in nxml.h nxml_internal.h
 
