--- origsrc/wtdbg2-2.5/Makefile	2019-09-13 16:40:02.000000000 +0900
+++ src/wtdbg2-2.5/Makefile	2020-10-01 11:07:33.693413600 +0900
@@ -14,6 +14,8 @@ else
 CFLAGS=-g3 -W -Wall -Wno-unused-but-set-variable -O4 -DVERSION="$(VERSION)" -DRELEASE="$(RELEASE)" -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -mpopcnt -msse4.2
 endif
 
+CFLAGS+=-D__HEADER_NO_EXECINFO
+
 GLIBS=-lm -lrt -lpthread -lz
 GENERIC_SRC=mem_share.h chararray.h sort.h list.h pgzf.h sort.h list.h dna.h thread.h filereader.h filewriter.h bitvec.h bit2vec.h bitsvec.h hashset.h
 
