--- origsrc/scrypt-1.1.6/Makefile.in	2010-01-17 05:48:23.000000000 +0900
+++ src/scrypt-1.1.6/Makefile.in	2013-06-29 11:02:38.546952700 +0900
@@ -183,7 +183,7 @@ scrypt_SOURCES = main.c				\
 			lib/crypto/sha256.c		\
 			lib/crypto/crypto_scrypt-@SCRYPTVER@.c
 
-scrypt_CFLAGS = -I . -I lib/util -I lib/scryptenc -I lib/crypto
+scrypt_CFLAGS = -I . -I $(top_srcdir)/lib/util -I $(top_srcdir)/lib/scryptenc -I $(top_srcdir)/lib/crypto
 scrypt_LDADD = -lcrypto
 scrypt_man_MANS = scrypt.1
 all: config.h
