--- origsrc/vvdec-1.5.0/source/Lib/vvdec/CMakeLists.txt	2022-04-19 16:53:16.000000000 +0900
+++ src/vvdec-1.5.0/source/Lib/vvdec/CMakeLists.txt	2022-06-20 07:54:15.130738500 +0900
@@ -126,5 +126,6 @@ target_link_libraries( ${LIB_NAME} ${INT
 # set the folder where to place the projects
 set_target_properties( ${LIB_NAME} PROPERTIES FOLDER lib )
 
-
-
+if (DEFINED SOVERSION)
+  set_target_properties(${LIB_NAME} PROPERTIES SOVERSION "${SOVERSION}")
+endif()
