--- origsrc/vvdec-1.2.0/source/Lib/vvdec/CMakeLists.txt	2021-09-21 23:58:58.000000000 +0900
+++ src/vvdec-1.2.0/source/Lib/vvdec/CMakeLists.txt	2021-10-02 10:36:52.541283800 +0900
@@ -114,5 +114,7 @@ target_link_libraries( ${LIB_NAME} ${INT
 # set the folder where to place the projects
 set_target_properties( ${LIB_NAME} PROPERTIES FOLDER lib )
 
-
+if (CYGWIN)
+  set_target_properties(${LIB_NAME} PROPERTIES SOVERSION ${SOVERSION})
+endif()
 
