--- origsrc/vvdec-3.0.0/source/Lib/CommonLib/InterpolationFilter.cpp	2024-10-25 17:48:32.000000000 +0900
+++ src/vvdec-3.0.0/source/Lib/CommonLib/InterpolationFilter.cpp	2025-06-22 16:52:40.317547200 +0900
@@ -56,6 +56,8 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #include "ChromaFormat.h"
 
+#include <alloca.h>
+
 namespace vvdec
 {
 
--- origsrc/vvdec-3.0.0/source/Lib/CommonLib/x86/InterpolationFilterX86.h	2024-10-25 17:48:32.000000000 +0900
+++ src/vvdec-3.0.0/source/Lib/CommonLib/x86/InterpolationFilterX86.h	2025-06-22 16:53:02.465702400 +0900
@@ -55,6 +55,8 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #include "../Unit.h"
 
+#include <alloca.h>
+
 #ifdef TARGET_SIMD_X86
 
 namespace vvdec
