--- origsrc/googletest-release-1.11.0/googlemock/CMakeLists.txt	2021-06-12 02:42:26.000000000 +0900
+++ src/googletest-release-1.11.0/googlemock/CMakeLists.txt	2021-06-29 13:07:35.877311600 +0900
@@ -116,6 +116,9 @@ if (DEFINED CMAKE_VERSION AND NOT "${CMA
     "$<BUILD_INTERFACE:${gmock_build_include_dirs}>"
     "$<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${CMAKE_INSTALL_INCLUDEDIR}>")
 endif()
+if (CYGWIN)
+  set_target_properties(gmock gmock_main PROPERTIES SOVERSION 0)
+endif()
 
 ########################################################################
 #
--- origsrc/googletest-release-1.11.0/googletest/CMakeLists.txt	2021-06-12 02:42:26.000000000 +0900
+++ src/googletest-release-1.11.0/googletest/CMakeLists.txt	2021-06-29 13:07:35.898309200 +0900
@@ -144,6 +144,9 @@ if (DEFINED CMAKE_VERSION AND NOT "${CMA
     "$<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${CMAKE_INSTALL_INCLUDEDIR}>")
 endif()
 target_link_libraries(gtest_main PUBLIC gtest)
+if (CYGWIN)
+  set_target_properties(gtest gtest_main PROPERTIES SOVERSION 0)
+endif()
 
 ########################################################################
 #
--- origsrc/googletest-release-1.11.0/googletest/cmake/internal_utils.cmake	2021-06-12 02:42:26.000000000 +0900
+++ src/googletest-release-1.11.0/googletest/cmake/internal_utils.cmake	2021-06-29 13:07:35.901300700 +0900
@@ -90,7 +90,10 @@ macro(config_compiler_and_linker)
     set(cxx_strict_flags "-W -Wpointer-arith -Wreturn-type -Wcast-qual -Wwrite-strings -Wswitch -Wunused-parameter -Wcast-align -Wchar-subscripts -Winline -Wredundant-decls")
     set(cxx_no_rtti_flags "-fno-rtti")
   elseif (CMAKE_COMPILER_IS_GNUCXX)
-    set(cxx_base_flags "-Wall -Wshadow -Werror")
+    set(cxx_base_flags "-Wall -Wshadow")
+    if(NOT CYGWIN)
+      set(cxx_base_flags "${cxx_base_flags} -Werror")
+    endif()
     if(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0.0)
       set(cxx_base_flags "${cxx_base_flags} -Wno-error=dangling-else")
     endif()
