--- origsrc/namefix-1.0.1/makefile	2023-11-10 06:52:50.000000000 +0900
+++ src/namefix-1.0.1/makefile	2023-12-29 23:46:37.238650700 +0900
@@ -1,5 +1,5 @@
 CXX = g++
-CXXFLAGS = -std=c++17 -Wall -Wpedantic -Wextra -licuuc -licui18n
+CXXFLAGS = $(CFLAGS) -Wall -Wpedantic -Wextra -licuuc -licui18n
 SOURCES = $(wildcard src/*.cpp)
 NAME = namefix
 
--- origsrc/namefix-1.0.1/src/main.hpp	2023-11-10 06:52:50.000000000 +0900
+++ src/namefix-1.0.1/src/main.hpp	2023-12-29 23:49:02.710836800 +0900
@@ -31,10 +31,10 @@
 #include <unicode/normalizer2.h>
 
 // defines
-#define NAMEFIX_VERSION "1.0.0"
+#define NAMEFIX_VERSION "1.0.1"
 #define NAMEFIX_VERSION_MAJOR 1
 #define NAMEFIX_VERSION_MINOR 0
-#define NAMEFIX_VERSION_PATCH 0
+#define NAMEFIX_VERSION_PATCH 1
 
 #define DEFAULT_MAX_CHARACTERS 64
 #define DEFAULT_NON_ASCII_REPLACEMENT "?"
