--- origsrc/libregf-20160107/Makefile.am	2016-01-07 15:34:55.000000000 +0900
+++ src/libregf-20160107/Makefile.am	2016-04-07 23:42:44.199804500 +0900
@@ -3,32 +3,11 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
-	libfguid \
-	libfwnt \
 	libregf \
-	libfole \
-	libfwps \
-	libfwsi \
-	libcsystem \
 	regftools \
 	pyregf \
 	pyregf2 \
 	pyregf3 \
-	po \
 	manuals \
 	tests \
 	msvscpp
@@ -137,4 +116,3 @@ splint:
 	(cd $(srcdir)/pyregf && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
-
--- origsrc/libregf-20160107/configure.ac	2016-01-07 15:34:55.000000000 +0900
+++ src/libregf-20160107/configure.ac	2016-04-07 23:42:30.127060100 +0900
@@ -211,27 +211,7 @@ dnl Have configure make the Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfwnt/Makefile])
 AC_CONFIG_FILES([libregf/Makefile])
-AC_CONFIG_FILES([libfole/Makefile])
-AC_CONFIG_FILES([libfwps/Makefile])
-AC_CONFIG_FILES([libfwsi/Makefile])
-AC_CONFIG_FILES([libcsystem/Makefile])
 AC_CONFIG_FILES([regftools/Makefile])
 AC_CONFIG_FILES([pyregf/Makefile])
 AC_CONFIG_FILES([pyregf2/Makefile])
@@ -294,4 +274,3 @@ Features:
    Verbose output:                            $ac_cv_enable_verbose_output
    Debug output:                              $ac_cv_enable_debug_output
 ]);
-
--- origsrc/libregf-20160107/libregf/Makefile.am	2016-01-07 15:34:55.000000000 +0900
+++ src/libregf-20160107/libregf/Makefile.am	2016-04-07 23:42:54.306085400 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -94,4 +95,3 @@ distclean: clean
 splint:
 	@echo "Running splint on libregf ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(libregf_la_SOURCES)
-
--- origsrc/libregf-20160107/regftools/Makefile.am	2016-01-07 15:34:55.000000000 +0900
+++ src/libregf-20160107/regftools/Makefile.am	2016-04-07 23:43:02.937634900 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -163,4 +164,3 @@ splint:
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(regfmount_SOURCES)
 	@echo "Running splint on regfreport ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(regfreport_SOURCES)
-
--- origsrc/libregf-20160107/tests/Makefile.am	2016-01-07 15:34:55.000000000 +0900
+++ src/libregf-20160107/tests/Makefile.am	2016-04-07 23:43:10.590160000 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -81,4 +82,3 @@ MAINTAINERCLEANFILES = \
 
 distclean: clean
 	/bin/rm -f Makefile
-
