--- origsrc/libregf-20200805/Makefile.am	2020-08-05 21:03:57.000000000 +0900
+++ src/libregf-20200805/Makefile.am	2020-09-04 10:53:42.985793700 +0900
@@ -3,26 +3,11 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
-	libfwnt \
 	libregf \
 	regftools \
 	pyregf \
 	pyregf-python2 \
 	pyregf-python3 \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libregf-20200805/configure.ac	2020-08-05 21:03:57.000000000 +0900
+++ src/libregf-20200805/configure.ac	2020-09-04 10:53:42.999804000 +0900
@@ -173,20 +173,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfwnt/Makefile])
 AC_CONFIG_FILES([libregf/Makefile])
 AC_CONFIG_FILES([pyregf/Makefile])
 AC_CONFIG_FILES([pyregf-python2/Makefile])
--- origsrc/libregf-20200805/libregf/Makefile.am	2020-08-05 21:03:57.000000000 +0900
+++ src/libregf-20200805/libregf/Makefile.am	2020-09-04 10:53:43.029797900 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libregf-20200805/pyregf/Makefile.am	2020-08-05 21:03:57.000000000 +0900
+++ src/libregf-20200805/pyregf/Makefile.am	2020-09-04 10:53:43.069794800 +0900
@@ -1,5 +1,6 @@
 if HAVE_PYTHON
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -47,7 +48,7 @@ pyregf_la_LIBADD = \
 	@LIBCDATA_LIBADD@
 
 pyregf_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyregf_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyregf_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
--- origsrc/libregf-20200805/pyregf-python2/Makefile.am	2020-08-05 21:03:57.000000000 +0900
+++ src/libregf-20200805/pyregf-python2/Makefile.am	2020-09-04 10:53:43.098792100 +0900
@@ -6,6 +6,7 @@ pyexecdir = $(pyexecdir2)
 
 if HAVE_PYTHON2
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -57,7 +58,7 @@ pyregf_la_LIBADD = \
 	@LIBCDATA_LIBADD@
 
 pyregf_la_CPPFLAGS = $(PYTHON2_CPPFLAGS)
-pyregf_la_LDFLAGS  = -module -avoid-version $(PYTHON2_LDFLAGS)
+pyregf_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON2_LDFLAGS)
 
 $(BUILT_SOURCES):
 	/bin/cp -f $(top_srcdir)/pyregf/$@ $@
--- origsrc/libregf-20200805/pyregf-python3/Makefile.am	2020-08-05 21:03:57.000000000 +0900
+++ src/libregf-20200805/pyregf-python3/Makefile.am	2020-09-04 10:53:43.213786700 +0900
@@ -6,6 +6,7 @@ pyexecdir = $(pyexecdir3)
 
 if HAVE_PYTHON3
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -57,7 +58,7 @@ pyregf_la_LIBADD = \
 	@LIBCDATA_LIBADD@
 
 pyregf_la_CPPFLAGS = $(PYTHON3_CPPFLAGS)
-pyregf_la_LDFLAGS  = -module -avoid-version $(PYTHON3_LDFLAGS)
+pyregf_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON3_LDFLAGS)
 
 $(BUILT_SOURCES):
 	/bin/cp -f $(top_srcdir)/pyregf/$@ $@
--- origsrc/libregf-20200805/regftools/Makefile.am	2020-08-05 21:03:57.000000000 +0900
+++ src/libregf-20200805/regftools/Makefile.am	2020-09-04 10:53:43.268806500 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libregf-20200805/tests/Makefile.am	2020-08-05 21:03:57.000000000 +0900
+++ src/libregf-20200805/tests/Makefile.am	2020-09-04 10:53:43.378780800 +0900
@@ -1,6 +1,7 @@
 AUTOMAKE_OPTIONS = subdir-objects
 
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
