--- origsrc/libregf-20240421/Makefile.am	2024-04-21 16:23:00.000000000 +0900
+++ src/libregf-20240421/Makefile.am	2025-05-18 12:30:02.244024800 +0900
@@ -3,24 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
-	libfwnt \
 	libregf \
 	regftools \
 	pyregf \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libregf-20240421/configure.ac	2024-04-21 16:23:00.000000000 +0900
+++ src/libregf-20240421/configure.ac	2025-05-18 12:30:02.244024800 +0900
@@ -173,20 +173,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfwnt/Makefile])
 AC_CONFIG_FILES([libregf/Makefile])
 AC_CONFIG_FILES([pyregf/Makefile])
 AC_CONFIG_FILES([regftools/Makefile])
--- origsrc/libregf-20240421/pyregf/Makefile.am	2024-04-21 16:23:00.000000000 +0900
+++ src/libregf-20240421/pyregf/Makefile.am	2025-05-18 12:30:18.964935100 +0900
@@ -47,7 +47,7 @@ pyregf_la_LIBADD = \
 	@LIBCDATA_LIBADD@
 
 pyregf_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyregf_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyregf_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
