--- origsrc/libharu-2.2.1/configure.in	2010-11-22 18:59:56.000000000 +0900
+++ src/libharu-2.2.1/configure.in	2010-12-02 10:20:24.971978200 +0900
@@ -103,7 +103,7 @@ AM_MAINTAINER_MODE
 dnl Check for programs
 AC_PROG_CC
 AC_PROG_LD
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_PROG_INSTALL
 AC_PROG_EGREP
 
@@ -206,106 +206,8 @@ AC_ARG_WITH(libdir,
 )
 dnl }}}
 
-dnl {{{ --with-zlib
-AC_ARG_WITH(zlib,
-  [AS_HELP_STRING([--with-zlib],[specify Zlib install prefix])
-  ],
-  [ ],
-  [with_zlib=yes]
-)
-
-if test "x$with_zlib" = "xno"; then
-  AC_DEFINE([HAVE_NOZLIB], [], [zlib is not available])
-else
-  AC_MSG_CHECKING([Zlib install prefix])
-
-  if test "x$with_zlib" = "xyes"; then
-    for i in `echo "$STANDARD_PREFIXES"`; do
-      if test -f "$i/include/zlib.h"; then
-        ZLIB_DIR="$i"
-        break;
-      fi
-    done
-  else
-    if test -f "$with_zlib/include/zlib.h"; then
-      ZLIB_DIR="$with_zlib"
-      break;
-    else
-      AC_MSG_ERROR([Can't find Zlib headers under $with_zlib directory]);
-    fi
-  fi
-
-  if test "x$ZLIB_DIR" = "x"; then
-    AC_MSG_ERROR([Unable to locate Zlib headers, please use --with-zlib=<DIR>]);
-  fi
-
-  AC_MSG_RESULT([$ZLIB_DIR])
-  LDFLAGS="$LDFLAGS -L$ZLIB_DIR/$LIBDIR"
-  CFLAGS="$CFLAGS -I$ZLIB_DIR/include"
-  LIBS="$LIBS -lz"
-
-  AC_CHECK_LIB([z], [deflate], [], [
-    AC_MSG_ERROR([deflate() is missing, check config.log for more details])
-  ])
-
-  HAVE_ZLIB=yes
-fi
-dnl }}}
-
-dnl {{{ --with-png
-AC_ARG_WITH(png,
-  [AS_HELP_STRING([--with-png],[specify libpng install prefix])
-  ],
-  [ ],
-  [with_png=yes]
-)
-
-if test "x$with_png" = "xno"; then
-  AC_DEFINE([HAVE_NOPNGLIB], [], [libpng is not available])
-else
-
-  if test "x$HAVE_ZLIB" != "xyes"; then
-    AC_MSG_ERROR([PNG support requires Zlib, but it's not enabled. Either enable Zlib or disable PNG support.]);
-  fi
-
-  AC_MSG_CHECKING([libpng install prefix])
-
-  if test "x$with_png" = "xyes"; then
-    for i in `echo "$STANDARD_PREFIXES"`; do
-      if test -f "$i/include/png.h"; then
-        LIBPNG_DIR="$i"
-        break;
-      fi
-    done
-  else
-    if test -f "$with_png/include/png.h"; then
-      LIBPNG_DIR="$with_png"
-      break;
-    else
-      AC_MSG_ERROR([Can't find libpng headers under $with_png directory]);
-    fi
-  fi
-
-  if test "x$LIBPNG_DIR" = "x"; then
-    AC_MSG_ERROR([Unable to locate libpng headers, please use --with-png=<DIR>]);
-  fi
-
-  AC_MSG_RESULT([$LIBPNG_DIR])
-  LDFLAGS="$LDFLAGS -L$LIBPNG_DIR/$LIBDIR"
-  CFLAGS="$CFLAGS -I$LIBPNG_DIR/include"
-  LIBS="$LIBS -lpng"
-
-  AC_CHECK_LIB([png], [png_init_io], [], [
-    AC_MSG_ERROR([png_init_io() is missing, check config.log for more details])
-  ])
-fi
-dnl }}}
-
-case "$host_alias" in
-  *cygwin*[)]
-    CFLAGS="$CFLAGS -mno-cygwin -DHPDF_DLL_MAKE_CDECL"
-    ;;
-esac
+PKG_CHECK_MODULES(ZLIB, zlib)
+PKG_CHECK_MODULES(LIBPNG, libpng14)
 
 if test "$GCC" = yes ; then
   AC_MSG_CHECKING([whether GCC supports -fexceptions])
--- origsrc/libharu-2.2.1/src/Makefile.am	2010-11-22 18:57:27.000000000 +0900
+++ src/libharu-2.2.1/src/Makefile.am	2010-12-02 10:21:57.620349500 +0900
@@ -16,6 +16,6 @@ libhpdf_la_SOURCES =	hpdf_annotation.c h
 						hpdf_page_label.c hpdf_page_operator.c hpdf_pages.c hpdf_real.c \
 						hpdf_streams.c hpdf_string.c hpdf_u3d.c hpdf_utils.c hpdf_xref.c hpdf_pdfa.c 
 
-INCLUDES = -I$(top_srcdir)/include 
-libhpdf_la_LIBADD = @LTLIBOBJS@
-libhpdf_la_LDFLAGS = -release @VERSION@
+INCLUDES = -I$(top_srcdir)/include @ZLIB_CFLAGS@ @LIBPNG_CFLAGS@
+libhpdf_la_LIBADD = @LTLIBOBJS@ @ZLIB_LIBS@ @LIBPNG_LIBS@
+libhpdf_la_LDFLAGS = -release @VERSION@ -no-undefined
