--- origsrc/libharu-2.4.5/CMakeLists.txt	2025-03-11 23:13:06.000000000 +0900
+++ src/libharu-2.4.5/CMakeLists.txt	2025-04-15 03:57:27.866008700 +0900
@@ -135,10 +135,21 @@ set(
 install(FILES ${haru_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 # install various files
-install(FILES README.md CHANGES INSTALL DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/libharu)
-install(DIRECTORY bindings DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/libharu)
+if (INSTALL_DOCS)
+  install(FILES README.md CHANGES INSTALL DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/libharu)
+endif()
+if (INSTALL_BINDINGS)
+  install(DIRECTORY bindings DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/libharu)
+endif()
 
 # =======================================================================
 # print out some information
 # =======================================================================
 summary()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libharu.pc.in ${CMAKE_BINARY_DIR}/libharu.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libharu.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libharu-2.4.5/demo/CMakeLists.txt	2025-03-11 23:13:06.000000000 +0900
+++ src/libharu-2.4.5/demo/CMakeLists.txt	2025-04-15 03:55:37.517311000 +0900
@@ -84,16 +84,17 @@ set_target_properties(grid_sheet PROPERT
 # =======================================================================
 # install demo source, demo executables and other files
 # =======================================================================
+option(CMAKE_INSTALL_DEMODIR "demo")
 install(
   TARGETS ${demos_NAMES} ${demos_with_grid_NAMES} grid_sheet
-  DESTINATION demo
+  DESTINATION ${CMAKE_INSTALL_DEMODIR}
 )
 foreach(demo ${demos_NAMES} ${demos_with_grid_NAMES} grid_sheet)
-  install(FILES ${demo}.c DESTINATION demo)
+  install(FILES ${demo}.c DESTINATION ${CMAKE_INSTALL_DEMODIR})
 endforeach()
 
 # install demo files
 install(
   DIRECTORY images mbtext pdf_a pngsuite rawimage ttfont type1
-  DESTINATION demo
+  DESTINATION ${CMAKE_INSTALL_DEMODIR}
 )
--- origsrc/libharu-2.4.5/libharu.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libharu-2.4.5/libharu.pc.in	2025-04-15 03:58:58.450369300 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libharu
+Description: Free PDF library
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lhpdf
+Cflags: -I${includedir}
+Requires: libpng zlib
