--- origsrc/libharu-2.4.0/CMakeLists.txt	2022-08-07 05:17:12.000000000 +0900
+++ src/libharu-2.4.0/CMakeLists.txt	2022-08-11 22:23:21.510731000 +0900
@@ -132,10 +132,21 @@ set(
 install(FILES ${haru_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 # install various files
-install(FILES README.md CHANGES INSTALL DESTINATION .)
-install(DIRECTORY bindings DESTINATION .)
+if (NOT CYGWIN)
+  install(FILES README.md CHANGES INSTALL DESTINATION .)
+endif()
+if (INSTALL_BINDINGS)
+  install(DIRECTORY bindings DESTINATION .)
+endif()
 
 # =======================================================================
 # print out some information
 # =======================================================================
 summary()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libharu.pc.in ${CMAKE_BINARY_DIR}/libharu.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libharu.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libharu-2.4.0/demo/CMakeLists.txt	2022-08-07 05:17:12.000000000 +0900
+++ src/libharu-2.4.0/demo/CMakeLists.txt	2022-08-11 22:17:05.593062500 +0900
@@ -82,16 +82,17 @@ set_target_properties(grid_sheet PROPERT
 # =======================================================================
 # install demo source, demo executables and other files
 # =======================================================================
+option(CMAKE_INSTALL_DEMODIR "demo")
 install(
   TARGETS ${demos_NAMES} ${demos_with_grid_NAMES} grid_sheet
-  DESTINATION demo
+  DESTINATION ${CMAKE_INSTALL_DEMODIR}
 )
 foreach(demo ${demos_NAMES} ${demos_with_grid_NAMES} grid_sheet)
-  install(FILES ${demo}.c DESTINATION demo)
+  install(FILES ${demo}.c DESTINATION ${CMAKE_INSTALL_DEMODIR})
 endforeach(demo)
 
 # install demo files
 install(
   DIRECTORY images mbtext pngsuite rawimage ttfont type1
-  DESTINATION demo
+  DESTINATION ${CMAKE_INSTALL_DEMODIR}
 )
--- origsrc/libharu-2.4.0/libharu.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libharu-2.4.0/libharu.pc.in	2022-08-11 22:31:51.987636400 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libharu
+Description: Free PDF library
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lharu
+Cflags: -I${includedir}
+Requires: libpng zlib
--- origsrc/libharu-2.4.0/src/CMakeLists.txt	2022-08-07 05:17:12.000000000 +0900
+++ src/libharu-2.4.0/src/CMakeLists.txt	2022-08-11 22:20:14.298707500 +0900
@@ -87,8 +87,12 @@ if(UNIX AND NOT APPLE)
     target_link_libraries (haru ${M_LIB})
 endif()
 
+if (DEFINED SOVERSION)
+	set_target_properties(haru PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 install(
-   TARGETS
+   TARGETS haru
    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
