--- origsrc/libharu-2.4.3/CMakeLists.txt	2022-10-14 19:04:02.000000000 +0900
+++ src/libharu-2.4.3/CMakeLists.txt	2022-11-20 17:28:33.793894700 +0900
@@ -139,10 +139,21 @@ set(
 install(FILES ${haru_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 # install various files
-install(FILES README.md CHANGES INSTALL DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/libharu)
-install(DIRECTORY bindings DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/libharu)
+if (INSTALL_DOCS)
+  install(FILES README.md CHANGES INSTALL DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/libharu)
+endif()
+if (INSTALL_BINDINGS)
+  install(DIRECTORY bindings DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/libharu)
+endif()
 
 # =======================================================================
 # print out some information
 # =======================================================================
 summary()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libharu.pc.in ${CMAKE_BINARY_DIR}/libharu.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libharu.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libharu-2.4.3/demo/CMakeLists.txt	2022-10-14 19:04:02.000000000 +0900
+++ src/libharu-2.4.3/demo/CMakeLists.txt	2022-11-20 00:11:12.551951400 +0900
@@ -83,16 +83,17 @@ set_target_properties(grid_sheet PROPERT
 # =======================================================================
 # install demo source, demo executables and other files
 # =======================================================================
+option(CMAKE_INSTALL_DEMODIR "demo")
 install(
   TARGETS ${demos_NAMES} ${demos_with_grid_NAMES} grid_sheet
-  DESTINATION demo
+  DESTINATION ${CMAKE_INSTALL_DEMODIR}
 )
 foreach(demo ${demos_NAMES} ${demos_with_grid_NAMES} grid_sheet)
-  install(FILES ${demo}.c DESTINATION demo)
+  install(FILES ${demo}.c DESTINATION ${CMAKE_INSTALL_DEMODIR})
 endforeach()
 
 # install demo files
 install(
   DIRECTORY images mbtext pngsuite rawimage ttfont type1
-  DESTINATION demo
+  DESTINATION ${CMAKE_INSTALL_DEMODIR}
 )
--- origsrc/libharu-2.4.3/libharu.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libharu-2.4.3/libharu.pc.in	2022-11-20 17:14:43.986341800 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libharu
+Description: Free PDF library
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lharu
+Cflags: -I${includedir}
+Requires: libpng zlib
--- origsrc/libharu-2.4.3/src/CMakeLists.txt	2022-10-14 19:04:02.000000000 +0900
+++ src/libharu-2.4.3/src/CMakeLists.txt	2022-11-20 00:13:14.227428300 +0900
@@ -87,6 +87,10 @@ if(UNIX AND NOT APPLE)
     target_link_libraries (hpdf ${M_LIB})
 endif()
 
+if (DEFINED SOVERSION)
+	set_target_properties(hpdf PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 install(
    TARGETS hpdf
    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
