--- origsrc/libharu-2.4.6/CMakeLists.txt	2026-03-27 04:06:59.000000000 +0900
+++ src/libharu-2.4.6/CMakeLists.txt	2026-05-13 10:37:28.893145900 +0900
@@ -141,3 +141,10 @@ install(DIRECTORY bindings DESTINATION $
 # print out some information
 # =======================================================================
 summary()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libharu.pc.in ${CMAKE_BINARY_DIR}/libharu.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libharu.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libharu-2.4.6/libharu.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libharu-2.4.6/libharu.pc.in	2026-05-13 10:40:11.409778700 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libharu
+Description: Free PDF library
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lhpdf
+Cflags: -I${includedir}
+Requires: libpng zlib
\ No newline at end of file
