--- origsrc/gosu-0.14.5/Gosu/Platform.hpp	2019-01-09 06:58:14.000000000 +0900
+++ src/gosu-0.14.5/Gosu/Platform.hpp	2019-03-22 11:22:20.152023300 +0900
@@ -50,7 +50,7 @@ namespace Gosu
 # define GOSU_IS_WIN
 #else
 # define GOSU_IS_UNIX
-# if defined(__linux) || defined(__FreeBSD__)
+# if defined(__linux) || defined(__FreeBSD__) || defined(__CYGWIN__)
 #  define GOSU_IS_X
 # else
 #  define GOSU_IS_MAC
--- origsrc/gosu-0.14.5/cmake/Compilation.cmake	2019-01-09 06:58:14.000000000 +0900
+++ src/gosu-0.14.5/cmake/Compilation.cmake	2019-03-22 11:22:20.152023300 +0900
@@ -16,10 +16,10 @@ if(WIN32)
     set(GOSU_LIBRARY_INSTALL_DIRS "Gosu/lib")
     set(GOSU_INCLUDE_INSTALL_DIRS "Gosu/include")
 else()
-    set(CONFIG_FILE_DIR "share/Gosu/cmake")
+    set(CONFIG_FILE_DIR "lib/cmake/Gosu")
     set(GOSU_LIBRARY_INSTALL_DIRS "lib")
     set(GOSU_INCLUDE_INSTALL_DIRS "include/Gosu")
-    set(INSTALL_PKGCONFIG_DIR "share/pkgconfig")
+    set(INSTALL_PKGCONFIG_DIR "lib/pkgconfig")
 endif()
 
 
--- origsrc/gosu-0.14.5/cmake/GosuConfig.cmake.in	2019-01-09 06:58:14.000000000 +0900
+++ src/gosu-0.14.5/cmake/GosuConfig.cmake.in	2019-03-22 11:22:20.167692100 +0900
@@ -79,6 +79,10 @@ if(APPLE)
         ${AUDIO_TOOLBOX_LIBRARY}
         ${ICONV_LIBRARY}
     )
+
+    if (CYGWIN)
+        set(GOSU_DEPENDENCIES ${GOSU_DEPENDENCIES} iconv)
+    endif ()    
 endif()
 
 if(NOT COMPILING_GOSU)
