--- origsrc/gosu-0.15.2/CMakeLists.txt	2020-06-02 15:31:51.000000000 +0900
+++ src/gosu-0.15.2/CMakeLists.txt	2020-11-04 16:51:05.788630900 +0900
@@ -25,8 +25,13 @@ if(APPLE)
 endif()
 
 # Set up install paths e.g. for GosuConfig.cmake
-set(CONFIG_FILE_DIR "share/Gosu/cmake")
-set(INSTALL_PKGCONFIG_DIR "share/pkgconfig")
+if (CYGWIN)
+        set(CONFIG_FILE_DIR "lib/cmake/Gosu")
+        set(INSTALL_PKGCONFIG_DIR "lib/pkgconfig")
+else()
+        set(CONFIG_FILE_DIR "share/Gosu/cmake")
+        set(INSTALL_PKGCONFIG_DIR "share/pkgconfig")
+endif()
 
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/cmake/GosuConfig.cmake
         DESTINATION ${CONFIG_FILE_DIR}
--- origsrc/gosu-0.15.2/Gosu/Platform.hpp	2020-06-02 15:31:51.000000000 +0900
+++ src/gosu-0.15.2/Gosu/Platform.hpp	2020-11-04 16:43:10.064866600 +0900
@@ -50,7 +50,7 @@ namespace Gosu
 # define GOSU_IS_WIN
 #else
 # define GOSU_IS_UNIX
-# if defined(__linux) || defined(__FreeBSD__)
+# if defined(__linux) || defined(__FreeBSD__) || defined(__CYGWIN__)
 #  define GOSU_IS_X
 # else
 #  define GOSU_IS_MAC
