--- origsrc/zimwriterfs-1.3.8/meson.build	2020-02-10 22:33:58.000000000 +0900
+++ src/zimwriterfs-1.3.8/meson.build	2020-02-16 21:48:50.638426200 +0900
@@ -1,7 +1,7 @@
 project('zimwriterfs', ['c', 'cpp'],
   version : '1.3.8',
   license : 'GPL-3.0-or-later',
-  default_options : ['c_std=c11', 'cpp_std=c++11', 'werror=true'])
+  default_options : ['c_std=c11', 'cpp_std=c++11'])
 
 add_global_arguments('-DVERSION="@0@"'.format(meson.project_version()), language : 'cpp')
 
