--- origsrc/oneTBB-2022.0.0/cmake/compilers/GNU.cmake	2024-11-01 02:23:11.000000000 +0900
+++ src/oneTBB-2022.0.0/cmake/compilers/GNU.cmake	2025-03-22 11:04:49.263953300 +0900
@@ -106,7 +106,7 @@ set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMO
 set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} -Wformat -Wformat-security -Werror=format-security
     -fstack-protector-strong )
 # -z switch is not supported on MacOS and MinGW
-if (NOT APPLE AND NOT MINGW)
+if (NOT APPLE AND NOT MINGW AND NOT CYGWIN)
     set(TBB_LIB_LINK_FLAGS ${TBB_LIB_LINK_FLAGS} -Wl,-z,relro,-z,now,-z,noexecstack)
 endif()
 if (NOT CMAKE_CXX_FLAGS MATCHES "_FORTIFY_SOURCE")
--- origsrc/oneTBB-2022.0.0/include/oneapi/tbb/detail/_config.h	2024-11-01 02:23:11.000000000 +0900
+++ src/oneTBB-2022.0.0/include/oneapi/tbb/detail/_config.h	2025-03-22 11:04:49.263953300 +0900
@@ -201,7 +201,7 @@
 
 /** __TBB_WEAK_SYMBOLS_PRESENT denotes that the system supports the weak symbol mechanism **/
 #ifndef __TBB_WEAK_SYMBOLS_PRESENT
-    #define __TBB_WEAK_SYMBOLS_PRESENT ( !__EMSCRIPTEN__ && !_WIN32 && !__APPLE__ && !__sun && (__TBB_GCC_VERSION >= 40000 || __INTEL_COMPILER ) )
+    #define __TBB_WEAK_SYMBOLS_PRESENT ( !__EMSCRIPTEN__ && !_WIN32 && !__CYGWIN__ && !__APPLE__ && !__sun && (__TBB_GCC_VERSION >= 40000 || __INTEL_COMPILER ) )
 #endif
 
 /** Presence of compiler features **/
--- origsrc/oneTBB-2022.0.0/test/common/common_arena_constraints.h	2024-11-01 02:23:11.000000000 +0900
+++ src/oneTBB-2022.0.0/test/common/common_arena_constraints.h	2025-03-22 11:04:49.263953300 +0900
@@ -86,7 +86,7 @@ int get_processors_group_count() { retur
 #define __HWLOC_TOPOLOGY_FLAG_RESTRICT_TO_CPUBINDING_PRESENT (HWLOC_API_VERSION >= 0x20500)
 // At this moment the hybrid CPUs HWLOC interfaces returns unexpected results on some Windows machines
 // in the 32-bit arch mode.
-#define __HWLOC_HYBRID_CPUS_INTERFACES_VALID (!_WIN32 || _WIN64)
+#define __HWLOC_HYBRID_CPUS_INTERFACES_VALID (!_WIN32)
 
 #define __HYBRID_CPUS_TESTING __HWLOC_HYBRID_CPUS_INTERFACES_PRESENT && __HWLOC_HYBRID_CPUS_INTERFACES_VALID
 #define __HWLOC_CPUBIND_PRESENT (!__APPLE__)
--- origsrc/oneTBB-2022.0.0/test/tbb/test_eh_thread.cpp	2024-11-01 02:23:11.000000000 +0900
+++ src/oneTBB-2022.0.0/test/tbb/test_eh_thread.cpp	2025-03-22 11:04:49.263953300 +0900
@@ -34,7 +34,7 @@
 // On Windows there is no real thread number limit beside available memory.
 // Therefore, the test for thread limit is unreasonable.
 // TODO: enable limitThreads with sanitizer under docker
-#if TBB_USE_EXCEPTIONS && !_WIN32 && !__ANDROID__
+#if TBB_USE_EXCEPTIONS && !_WIN32 && !__ANDROID__ && !__CYGWIN__
 
 #include <limits.h>
 #include <sys/types.h>
