--- origsrc/oneTBB-2021.7.0/include/oneapi/tbb/detail/_config.h	2022-10-07 23:34:12.000000000 +0900
+++ src/oneTBB-2021.7.0/include/oneapi/tbb/detail/_config.h	2022-12-21 22:43:19.194711700 +0900
@@ -195,7 +195,7 @@
 
 /** __TBB_WEAK_SYMBOLS_PRESENT denotes that the system supports the weak symbol mechanism **/
 #ifndef __TBB_WEAK_SYMBOLS_PRESENT
-    #define __TBB_WEAK_SYMBOLS_PRESENT ( !_WIN32 && !__APPLE__ && !__sun && (__TBB_GCC_VERSION >= 40000 || __INTEL_COMPILER ) )
+    #define __TBB_WEAK_SYMBOLS_PRESENT ( !_WIN32 && !__CYGWIN__ && !__APPLE__ && !__sun && (__TBB_GCC_VERSION >= 40000 || __INTEL_COMPILER ) )
 #endif
 
 /** Presence of compiler features **/
--- origsrc/oneTBB-2021.7.0/test/common/common_arena_constraints.h	2022-10-07 23:34:12.000000000 +0900
+++ src/oneTBB-2021.7.0/test/common/common_arena_constraints.h	2022-12-22 08:21:30.999676400 +0900
@@ -88,7 +88,7 @@ int get_processors_group_count() { retur
 #define __HWLOC_TOPOLOGY_FLAG_RESTRICT_TO_CPUBINDING_PRESENT (HWLOC_API_VERSION >= 0x20500)
 // At this moment the hybrid CPUs HWLOC interfaces returns unexpected results on some Windows machines
 // in the 32-bit arch mode.
-#define __HWLOC_HYBRID_CPUS_INTERFACES_VALID (!_WIN32 || _WIN64)
+#define __HWLOC_HYBRID_CPUS_INTERFACES_VALID (!_WIN32)
 
 #define __HYBRID_CPUS_TESTING __HWLOC_HYBRID_CPUS_INTERFACES_PRESENT && __HWLOC_HYBRID_CPUS_INTERFACES_VALID
 
--- origsrc/oneTBB-2021.7.0/test/tbb/test_eh_thread.cpp	2022-10-07 23:34:12.000000000 +0900
+++ src/oneTBB-2021.7.0/test/tbb/test_eh_thread.cpp	2022-12-22 08:06:03.623268000 +0900
@@ -34,7 +34,7 @@
 // On Windows there is no real thread number limit beside available memory.
 // Therefore, the test for thread limit is unreasonable.
 // TODO: enable limitThreads with sanitizer under docker
-#if TBB_USE_EXCEPTIONS && !_WIN32 && !__ANDROID__
+#if TBB_USE_EXCEPTIONS && !_WIN32 && !__ANDROID__ && !defined(__CYGWIN__)
 
 #include <limits.h>
 #include <sys/types.h>
