--- origsrc/wslay-1.1.1/tests/Makefile.am	2020-06-20 16:00:03.000000000 +0900
+++ src/wslay-1.1.1/tests/Makefile.am	2025-12-19 17:11:39.047984800 +0900
@@ -36,8 +36,7 @@ HFILES = wslay_session_test.h wslay_fram
 main_SOURCES = $(HFILES) $(OBJECTS)
 
 main_LDADD = ${top_builddir}/lib/libwslay.la -lcunit
-main_LDFLAGS = -static
-AM_CFLAGS = -Wall -g -O2 -I${top_srcdir}/lib -I${top_srcdir}/lib/includes\
+AM_CFLAGS = -Wall -g -O2 -I${top_srcdir}/lib -I${top_builddir}/lib/includes -I${top_srcdir}/lib/includes\
 	@DEFS@
 # DEFS += -D_ISOC99_SOURCE -D_GNU_SOURCE
 TESTS = main
