--- origsrc/hiredis-1.1.0/CMakeLists.txt	2022-11-15 13:34:30.000000000 +0900
+++ src/hiredis-1.1.0/CMakeLists.txt	2023-01-21 11:49:38.059829500 +0900
@@ -45,19 +45,24 @@ IF(WIN32)
 ENDIF()
 
 ADD_LIBRARY(hiredis SHARED ${hiredis_sources})
-ADD_LIBRARY(hiredis_static STATIC ${hiredis_sources})
+if (BUILD_STATIC_LIBS)
+    ADD_LIBRARY(hiredis_static STATIC ${hiredis_sources})
+endif()
 ADD_LIBRARY(hiredis::hiredis ALIAS hiredis)
-ADD_LIBRARY(hiredis::hiredis_static ALIAS hiredis_static)
+if (BUILD_STATIC_LIBS)
+    ADD_LIBRARY(hiredis::hiredis_static ALIAS hiredis_static)
+endif()
 
-IF(NOT MSVC)
+IF(BUILD_STATIC_LIBS)
     SET_TARGET_PROPERTIES(hiredis_static
         PROPERTIES OUTPUT_NAME hiredis)
 ENDIF()
 
 SET_TARGET_PROPERTIES(hiredis
     PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS TRUE
-    VERSION "${HIREDIS_SONAME}")
-IF(MSVC)
+    VERSION "${HIREDIS_SONAME}"
+    SOVERSION "${HIREDIS_SONAME}")
+IF(BUILD_STATIC_LIBS)
     SET_TARGET_PROPERTIES(hiredis_static
         PROPERTIES COMPILE_FLAGS /Z7)
 ENDIF()
@@ -73,7 +78,9 @@ ELSEIF(CMAKE_SYSTEM_NAME MATCHES "SunOS"
 ENDIF()
 
 TARGET_INCLUDE_DIRECTORIES(hiredis PUBLIC $<INSTALL_INTERFACE:include> $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
-TARGET_INCLUDE_DIRECTORIES(hiredis_static PUBLIC $<INSTALL_INTERFACE:include> $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
+if (BUILD_STATIC_LIBS)
+    TARGET_INCLUDE_DIRECTORIES(hiredis_static PUBLIC $<INSTALL_INTERFACE:include> $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
+endif()
 
 CONFIGURE_FILE(hiredis.pc.in hiredis.pc @ONLY)
 
@@ -103,11 +110,19 @@ set(CPACK_RPM_PACKAGE_AUTOREQPROV ON)
 
 include(CPack)
 
-INSTALL(TARGETS hiredis hiredis_static
-    EXPORT hiredis-targets
-    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+if (BUILD_STATIC_LIBS)
+    INSTALL(TARGETS hiredis hiredis_static
+        EXPORT hiredis-targets
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+else()
+    INSTALL(TARGETS hiredis
+        EXPORT hiredis-targets
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+endif()
 
 if (MSVC)
     INSTALL(FILES $<TARGET_PDB_FILE:hiredis>
@@ -163,12 +178,12 @@ IF(ENABLE_SSL)
         ssl.c)
     ADD_LIBRARY(hiredis_ssl SHARED
             ${hiredis_ssl_sources})
-    ADD_LIBRARY(hiredis_ssl_static STATIC
+    if (BUILD_STATIC_LIBS)
+        ADD_LIBRARY(hiredis_ssl_static STATIC
             ${hiredis_ssl_sources})
-    IF(NOT MSVC)
         SET_TARGET_PROPERTIES(hiredis_ssl_static
             PROPERTIES OUTPUT_NAME hiredis_ssl)
-    ENDIF()
+    endif()
 
     IF (APPLE)
         SET_PROPERTY(TARGET hiredis_ssl PROPERTY LINK_FLAGS "-Wl,-undefined -Wl,dynamic_lookup")
@@ -177,27 +192,38 @@ IF(ENABLE_SSL)
     SET_TARGET_PROPERTIES(hiredis_ssl
         PROPERTIES
         WINDOWS_EXPORT_ALL_SYMBOLS TRUE
-        VERSION "${HIREDIS_SONAME}")
-    IF(MSVC)
+        VERSION "${HIREDIS_SONAME}"
+        SOVERSION "${HIREDIS_SONAME}")
+    if (BUILD_STATIC_LIBS)
         SET_TARGET_PROPERTIES(hiredis_ssl_static
             PROPERTIES COMPILE_FLAGS /Z7)
-    ENDIF()
+    endif()
 
     TARGET_INCLUDE_DIRECTORIES(hiredis_ssl PRIVATE "${OPENSSL_INCLUDE_DIR}")
-    TARGET_INCLUDE_DIRECTORIES(hiredis_ssl_static PRIVATE "${OPENSSL_INCLUDE_DIR}")
+    if (BUILD_STATIC_LIBS)
+        TARGET_INCLUDE_DIRECTORIES(hiredis_ssl_static PRIVATE "${OPENSSL_INCLUDE_DIR}")
+    endif()
 
     TARGET_LINK_LIBRARIES(hiredis_ssl PRIVATE ${OPENSSL_LIBRARIES})
-    IF (WIN32 OR MINGW)
-        TARGET_LINK_LIBRARIES(hiredis_ssl PRIVATE hiredis)
+    TARGET_LINK_LIBRARIES(hiredis_ssl PRIVATE hiredis)
+    if (BUILD_STATIC_LIBS)
         TARGET_LINK_LIBRARIES(hiredis_ssl_static PUBLIC hiredis_static)
-    ENDIF()
+    endif()
     CONFIGURE_FILE(hiredis_ssl.pc.in hiredis_ssl.pc @ONLY)
 
-    INSTALL(TARGETS hiredis_ssl hiredis_ssl_static
-        EXPORT hiredis_ssl-targets
-        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    if (BUILD_STATIC_LIBS)
+        INSTALL(TARGETS hiredis_ssl hiredis_ssl_static
+            EXPORT hiredis_ssl-targets
+            RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+            LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    else()
+        INSTALL(TARGETS hiredis_ssl
+            EXPORT hiredis_ssl-targets
+            RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+            LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    endif()
 
     if (MSVC)
         INSTALL(FILES $<TARGET_PDB_FILE:hiredis_ssl>
