--- origsrc/hiredis-1.3.0/CMakeLists.txt	2025-04-23 20:10:29.000000000 +0900
+++ src/hiredis-1.3.0/CMakeLists.txt	2025-06-08 01:00:14.191305100 +0900
@@ -177,7 +177,7 @@ IF(ENABLE_SSL)
             PROPERTIES COMPILE_FLAGS /Z7)
     ENDIF()
     TARGET_LINK_LIBRARIES(hiredis_ssl PRIVATE OpenSSL::SSL)
-    IF(WIN32)
+    IF(CYGWIN OR WIN32)
         TARGET_LINK_LIBRARIES(hiredis_ssl PRIVATE hiredis)
     ENDIF()
     CONFIGURE_FILE(hiredis_ssl.pc.in hiredis_ssl.pc @ONLY)
