--- origsrc/hiredis-1.4.1/CMakeLists.txt	2026-08-06 14:45:20.000000000 +0000
+++ src/hiredis-1.4.1/CMakeLists.txt	2026-08-15 06:45:35.963896300 +0000
@@ -188,7 +188,7 @@ IF(ENABLE_SSL)
             PROPERTIES COMPILE_FLAGS /Z7)
     ENDIF()
     TARGET_LINK_LIBRARIES(hiredis_ssl PRIVATE OpenSSL::SSL)
-    IF(WIN32)
+    IF(CYGWIN OR WIN32)
         TARGET_LINK_LIBRARIES(hiredis_ssl PRIVATE hiredis)
     ENDIF()
     CONFIGURE_FILE(hiredis_ssl.pc.in hiredis_ssl.pc @ONLY)
--- origsrc/hiredis-1.4.1/hiredis.h	2026-08-06 14:45:20.000000000 +0000
+++ src/hiredis-1.4.1/hiredis.h	2026-08-15 06:45:35.963896300 +0000
@@ -47,8 +47,8 @@ typedef intptr_t ssize_t;
 
 #define HIREDIS_MAJOR 1
 #define HIREDIS_MINOR 4
-#define HIREDIS_PATCH 0
-#define HIREDIS_SONAME 1.4.0
+#define HIREDIS_PATCH 1
+#define HIREDIS_SONAME 1.4.1
 
 /* Connection type can be blocking or non-blocking and is set in the
  * least significant bit of the flags field in redisContext. */
