--- origsrc/liblist/configure.ac	2024-11-25 16:17:55.000000000 +0900
+++ src/liblist/configure.ac	2024-11-25 16:40:50.677917200 +0900
@@ -12,7 +12,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
-AC_INIT([liblist], [2.4], [https://ohnopublishing.net/bugzilla/enter_bug.cgi?product=liblist], [liblist], [http://ohnopub.net/liblist])
+AC_INIT([liblist], [2.4], [https://github.com/JackieXie168/liblist], [liblist], [https://github.com/JackieXie168/liblist])
 AC_CONFIG_SRCDIR([list.c])
 
 AM_INIT_AUTOMAKE([dist-bzip2 parallel-tests subdir-objects])
--- origsrc/liblist/liblist.pc.in	2024-11-25 16:17:55.000000000 +0900
+++ src/liblist/liblist.pc.in	2024-11-25 17:43:37.803335100 +0900
@@ -7,5 +7,5 @@ Name: @PACKAGE_NAME@
 Description: A generic list package for C
 Version: @VERSION@
 URL: @PACKAGE_BUGREPORT@
-Libs: '-L${libdir}' -llist
-Cflags: '-I${includedir}'
+Libs: -L${libdir} -llist
+Cflags: -I${includedir}
