--- origsrc/xxtea-c/CMakeLists.txt	2019-06-10 23:16:23.000000000 +0900
+++ src/xxtea-c/CMakeLists.txt	2019-06-10 23:17:31.683940700 +0900
@@ -3,17 +3,34 @@ cmake_minimum_required(VERSION 2.8.12)
 set(CMAKE_MACOSX_RPATH 1)
 set(LIBXXTEA_SRC xxtea.c)
 add_library(xxtea SHARED ${LIBXXTEA_SRC})
-add_library(xxtea_static STATIC ${LIBXXTEA_SRC})
+if(NOT BUILD_SHARED_LIBS)
+        add_library(xxtea_static STATIC ${LIBXXTEA_SRC})
+endif()
 set_target_properties(xxtea PROPERTIES VERSION 1.0 SOVERSION 1)
-set_target_properties(xxtea_static PROPERTIES OUTPUT_NAME "xxtea")
+if(NOT BUILD_SHARED_LIBS)
+        set_target_properties(xxtea_static PROPERTIES OUTPUT_NAME "xxtea")
+endif()
 set_target_properties(xxtea PROPERTIES CLEAN_DIRECT_OUTPUT 1)
-set_target_properties(xxtea_static PROPERTIES CLEAN_DIRECT_OUTPUT 1)
+if(NOT BUILD_SHARED_LIBS)
+        set_target_properties(xxtea_static PROPERTIES CLEAN_DIRECT_OUTPUT 1)
+endif()
 set_target_properties(xxtea PROPERTIES INSTALL_RPATH "@loader_path/../lib")
-set_target_properties(xxtea_static PROPERTIES INSTALL_RPATH "@loader_path/../lib")
+if(NOT BUILD_SHARED_LIBS)
+        set_target_properties(xxtea_static PROPERTIES INSTALL_RPATH "@loader_path/../lib")
+endif()
 install(TARGETS xxtea
-        LIBRARY DESTINATION lib
-        ARCHIVE DESTINATION lib)
-install(TARGETS xxtea_static
-        LIBRARY DESTINATION lib
+        RUNTIME DESTINATION bin
         ARCHIVE DESTINATION lib)
+if(NOT BUILD_SHARED_LIBS)
+        install(TARGETS xxtea_static
+                LIBRARY DESTINATION lib
+                ARCHIVE DESTINATION lib)
+endif()
 install(FILES xxtea.h DESTINATION include)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/${PROJECT_NAME}.pc.in ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/xxtea-c/xxtea.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/xxtea-c/xxtea.pc.in	2019-06-10 23:16:26.659112100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: xxtea-c
+Description: XXTEA encryption algorithm library for C
+Version: @VERSION@
+Libs: -L${libdir} -lxxtea
+Cflags: -I${includedir}
