--- origsrc/libibtk-upstream-0.0.14/Makefile.am	2024-12-17 15:28:41.052435600 +0900
+++ src/libibtk-upstream-0.0.14/Makefile.am	2024-12-17 16:14:46.223800500 +0900
@@ -48,3 +48,6 @@ dist-slack: all install
 ## End of Makefile.am
 ##
 ###############################################################################
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = libibtk.pc
--- origsrc/libibtk-upstream-0.0.14/configure.in	2024-12-17 15:28:41.052435600 +0900
+++ src/libibtk-upstream-0.0.14/configure.in	2024-12-17 16:11:47.993744600 +0900
@@ -8,7 +8,7 @@ dnl #  libibtk is builded as static by d
 dnl ###########################################################################
 dnl # next ?
 dnl ###########################################################################
-dnl  Copyright (C) 1999 Steaphan Greene, Øyvind Aabling, Octavian Purdila,
+dnl  Copyright (C) 1999 Steaphan Greene, ï¿½yvind Aabling, Octavian Purdila,
 dnl        Vladimir Dergachev and Christian Lupien.
 dnl  
 dnl This program is free software; you can redistribute it and/or modify
@@ -92,7 +92,6 @@ AC_SUBST(VERSION)
 
 
 AM_INIT_AUTOMAKE($PACKAGE,$VERSION)
-AM_MAINTAINER_MODE        dnl comment this for release version
 AM_CONFIG_HEADER(config.h)
 
 
@@ -184,6 +183,7 @@ m4/Makefile
 src/Makefile
 src/version.h
 ])
+AC_OUTPUT([libibtk.pc])
 
 dnl
 dnl Change attribute.
--- origsrc/libibtk-upstream-0.0.14/libibtk.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libibtk-upstream-0.0.14/libibtk.pc.in	2024-12-17 16:08:14.477022800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libibtk
+Description: Insomnia's Basic ToolKit
+Version: @VERSION@
+Libs: -L${libdir} -libtk
+Cflags: -I${includedir}
--- origsrc/libibtk-upstream-0.0.14/src/Makefile.am	2024-12-17 15:28:41.052435600 +0900
+++ src/libibtk-upstream-0.0.14/src/Makefile.am	2024-12-17 15:31:30.928696700 +0900
@@ -38,7 +38,7 @@ libibtk_la_SOURCES = \
 	itextbox.cpp	\
 	iversion.cpp	\
 	iwindow.cpp
-libibtk_la_LDFLAGS = -version-info $(IBTK_MAJOR_VERSION):$(IBTK_SUB_VERSION):$(IBTK_MINOR_VERSION)
+libibtk_la_LDFLAGS = -no-undefined -version-info $(IBTK_MAJOR_VERSION):$(IBTK_SUB_VERSION):$(IBTK_MINOR_VERSION)
 
 ##
 ## IBTK header files (must be installed)
