--- origsrc/apache-log4cxx-1.6.1/src/main/include/log4cxx/helpers/asyncbuffer.h	2026-01-05 10:53:17.000000000 +0900
+++ src/apache-log4cxx-1.6.1/src/main/include/log4cxx/helpers/asyncbuffer.h	2026-01-24 11:05:20.789288700 +0900
@@ -204,7 +204,7 @@ public: // Modifiers
 	{
 		auto store = FmtArgStore();
 		( store.push_back(std::forward<Args>(args)), ...);
-		initializeForFmt(std::move(fmt_str), std::move(store));
+		initializeForFmt(StringViewType(fmt_str), std::move(store));
 	}
 
 #if LOG4CXX_WCHAR_T_API || LOG4CXX_LOGCHAR_IS_WCHAR
@@ -215,7 +215,7 @@ public: // Modifiers
 	{
 		auto store = WideFmtArgStore();
 		( store.push_back(std::forward<Args>(args)), ...);
-		initializeForFmt(std::move(fmt_str), std::move(store));
+		initializeForFmt(WideStringViewType(fmt_str), std::move(store));
 	}
 #endif // LOG4CXX_WCHAR_T_API || LOG4CXX_LOGCHAR_IS_WCHAR
 #endif // LOG4CXX_ASYNC_BUFFER_SUPPORTS_FMT
