--- origsrc/libmp4tag-2.0.0/CMakeLists.txt	2024-12-13 06:12:42.000000000 +0900
+++ src/libmp4tag-2.0.0/CMakeLists.txt	2025-01-10 20:34:27.089189600 +0900
@@ -116,13 +116,10 @@ if (LIBMP4TAG_FORTIFY STREQUAL T)
   add_compile_options (-fstack-protector-all)
   add_compile_options (-fstack-protector-strong)
   add_compile_options (-fstack-protector-all)
-  add_compile_options (-D_FORTIFY_SOURCE=2)
 else()
   if (CMAKE_C_COMPILER_ID STREQUAL "Clang")
     add_compile_options (-Wno-macro-redefined)
   endif()
-  add_compile_options (-U_FORTIFY_SOURCE)
-  add_compile_options (-D_FORTIFY_SOURCE=0)
 endif()
 
 #### system specific compile options
@@ -142,7 +139,7 @@ if (NOT WIN32)
     # universal binary
     set (CMAKE_OSX_ARCHITECTURES "arm64;x86_64")
   endif()
-else()
+elseif(NOT CYGWIN)
   add_link_options (-static-libgcc)
   add_link_options (-static-libstdc++)
 endif()
@@ -156,12 +153,16 @@ include (CheckStructHasMember)
 check_include_file (arpa/inet.h _hdr_arpa_inet)
 check_include_file (endian.h _hdr_endian)
 check_include_file (sys/resource.h _sys_resource)
-check_include_file (windows.h _hdr_windows)
-check_include_file (winsock2.h _hdr_winsock2)
+if (NOT CYGWIN)
+  check_include_file (windows.h _hdr_windows)
+  check_include_file (winsock2.h _hdr_winsock2)
+endif()
 
-check_function_exists (GetFileTime _lib_GetFileTime)
-check_function_exists (Sleep _lib_Sleep)
-check_function_exists (CommandLineToArgvW _lib_CommandLinetoArgvW)
+if (NOT CYGWIN)
+  check_function_exists (GetFileTime _lib_GetFileTime)
+  check_function_exists (Sleep _lib_Sleep)
+  check_function_exists (CommandLineToArgvW _lib_CommandLinetoArgvW)
+endif()
 check_function_exists (_wrename _lib__wrename)
 check_function_exists (_wstat64 _lib__wstat64)
 check_function_exists (_wunlink _lib__wunlink)
@@ -197,7 +198,7 @@ if (WIN32)
 endif()
 set_target_properties (${LIBMP4TAG_LIBNAME} PROPERTIES
   VERSION ${CMAKE_PROJECT_VERSION}
-  SOVERSION ${CMAKE_PROJECT_VERSION}
+  SOVERSION ${SOVERSION}
 )
 
 #### executables
