--- origsrc/rospack-2.6.2/CMakeLists.txt	2020-05-14 03:14:22.000000000 +0900
+++ src/rospack-2.6.2/CMakeLists.txt	2020-07-17 18:36:32.579436800 +0900
@@ -31,6 +31,10 @@ add_library(rospack
 )
 target_link_libraries(rospack ${TinyXML2_LIBRARIES} ${Boost_LIBRARIES} ${PYTHON_LIBRARIES})
 
+if(CYGWIN)
+  set_target_properties(rospack PROPERTIES SOVERSION 0)
+endif()
+
 add_executable(rospackexe src/rospack_main.cpp)
 # Set the name, and make it a "global" executable
 set_target_properties(rospackexe PROPERTIES
