--- origsrc/libmapidb-20150105/Makefile.am	2015-01-05 15:44:23.000000000 +0900
+++ src/libmapidb-20150105/Makefile.am	2016-04-05 19:05:36.682686100 +0900
@@ -3,11 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libcnotify \
 	libmapidb \
-	po \
 	manuals \
 	tests \
 	msvscpp
@@ -81,4 +77,3 @@ splint:
 	(cd $(srcdir)/libmapidb && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
-
--- origsrc/libmapidb-20150105/configure.ac	2015-01-05 15:44:23.000000000 +0900
+++ src/libmapidb-20150105/configure.ac	2016-04-05 19:05:29.422935700 +0900
@@ -123,9 +123,6 @@ dnl Have configure make the Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
 AC_CONFIG_FILES([libmapidb/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
@@ -160,4 +157,3 @@ Building:
 Features:
    Wide character type support: $ac_cv_enable_wide_character_type
 ]);
-
--- origsrc/libmapidb-20150105/libmapidb/Makefile.am	2015-01-05 15:44:23.000000000 +0900
+++ src/libmapidb-20150105/libmapidb/Makefile.am	2016-04-05 19:05:47.913592300 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -44,4 +45,3 @@ distclean: clean
 splint:
 	@echo "Running splint on libmapidb ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(libmapidb_la_SOURCES)
-
--- origsrc/libmapidb-20150105/tests/Makefile.am	2015-01-05 15:44:23.000000000 +0900
+++ src/libmapidb-20150105/tests/Makefile.am	2016-04-05 19:05:59.053659000 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -20,4 +21,3 @@ MAINTAINERCLEANFILES = \
 
 distclean: clean
 	/bin/rm -f Makefile
-
