--- origsrc/glbinding-3.0.2/CMakeLists.txt	2018-07-11 19:51:35.000000000 +0900
+++ src/glbinding-3.0.2/CMakeLists.txt	2019-01-06 13:29:31.443730900 +0900
@@ -138,7 +138,11 @@ endif()
 if(UNIX AND SYSTEM_DIR_INSTALL)
     # Install into the system (/usr/bin or /usr/local/bin)
     set(INSTALL_ROOT      "share/${project}")       # /usr/[local]/share/<project>
-    set(INSTALL_CMAKE     "share/${project}/cmake") # /usr/[local]/share/<project>/cmake
+    if(CYGWIN)
+        set(INSTALL_CMAKE     "lib/cmake/${project}") # /usr/[local]/lib/cmake/<project>
+    else()
+        set(INSTALL_CMAKE     "share/${project}/cmake") # /usr/[local]/share/<project>/cmake
+    endif()
     set(INSTALL_EXAMPLES  "share/${project}")       # /usr/[local]/share/<project>
     set(INSTALL_DATA      "share/${project}")       # /usr/[local]/share/<project>
     set(INSTALL_BIN       "bin")                    # /usr/[local]/bin
@@ -192,6 +196,9 @@ add_subdirectory(deploy)
 # 
 # Deployment (global project files)
 # 
+if(CYGWIN)
+install(FILES ${META_PROJECT_NAME}-config.cmake DESTINATION ${INSTALL_CMAKE} COMPONENT dev)
+else()
 
 # Install version file
 install(FILES "${PROJECT_BINARY_DIR}/VERSION" DESTINATION ${INSTALL_ROOT} COMPONENT runtime)
@@ -208,3 +215,12 @@ install(FILES README.md DESTINATION ${IN
 if (TARGET cubescape OR TARGET cubescape-log OR TARGET cubescape-qt)
     install(DIRECTORY ${PROJECT_SOURCE_DIR}/data DESTINATION ${INSTALL_DATA} COMPONENT examples_data)
 endif()
+
+endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/glbinding.pc.in ${CMAKE_BINARY_DIR}/glbinding.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/glbinding.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/glbinding-3.0.2/glbinding-config.cmake	2018-07-11 19:51:35.000000000 +0900
+++ src/glbinding-3.0.2/glbinding-config.cmake	2019-01-06 13:29:31.459993400 +0900
@@ -26,7 +26,7 @@ macro(find_modules PREFIX)
         if(TARGET ${module_name})
             set(MODULE_FOUND TRUE)
         else()
-            find_module("${CMAKE_CURRENT_LIST_DIR}/${PREFIX}/${module_name}/${module_name}-export.cmake")
+            find_module("${CMAKE_CURRENT_LIST_DIR}/${module_name}-export.cmake")
         endif()
     endforeach(module_name)
 endmacro()
--- origsrc/glbinding-3.0.2/glbinding.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/glbinding-3.0.2/glbinding.pc.in	2019-01-06 13:29:31.466817100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: glbinding
+Description: C++ binding for the OpenGL API, generated using the gl.xml specification
+Version: @META_VERSION@
+Libs: -L${libdir} -lglbinding
+Cflags: -I${includedir}
--- origsrc/glbinding-3.0.2/source/glbinding/CMakeLists.txt	2018-07-11 19:51:35.000000000 +0900
+++ src/glbinding-3.0.2/source/glbinding/CMakeLists.txt	2019-01-06 13:29:31.473678000 +0900
@@ -356,8 +356,16 @@ install(DIRECTORY
 )
 
 # CMake config
+if(CYGWIN)
+install(EXPORT ${target}-export
+    NAMESPACE   ${META_PROJECT_NAME}::
+    DESTINATION ${INSTALL_CMAKE}
+    COMPONENT   dev
+)
+else()
 install(EXPORT ${target}-export
     NAMESPACE   ${META_PROJECT_NAME}::
     DESTINATION ${INSTALL_CMAKE}/${target}
     COMPONENT   dev
 )
+endif()
\ No newline at end of file
--- origsrc/glbinding-3.0.2/source/tests/CMakeLists.txt	2018-07-11 19:51:35.000000000 +0900
+++ src/glbinding-3.0.2/source/tests/CMakeLists.txt	2019-01-06 13:29:31.490793200 +0900
@@ -52,7 +52,9 @@ set(gtest_disable_pthreads      OFF CACH
 set(gtest_force_shared_crt      ON  CACHE BOOL "")
 set(gtest_hide_internal_symbols OFF CACHE BOOL "")
 
-add_subdirectory(googletest/googlemock)
+if(NOT CYGWIN)
+    add_subdirectory(googletest/googlemock)
+endif()
 
 # Create interface library to link against gmock
 add_library(gmock-dev INTERFACE)
--- origsrc/glbinding-3.0.2/source/tests/glbinding-test/CMakeLists.txt	2018-07-11 19:51:35.000000000 +0900
+++ src/glbinding-3.0.2/source/tests/glbinding-test/CMakeLists.txt	2019-01-06 13:29:31.505221100 +0900
@@ -85,6 +85,7 @@ target_link_libraries(${target}
     ${META_PROJECT_NAME}::glbinding
     ${META_PROJECT_NAME}::glbinding-aux
     gmock-dev
+    gtest
 )
 
 
