--- origsrc/libdrm-2.4.92/etnaviv/etnaviv_device.c	2018-05-10 07:26:03.000000000 +0900
+++ src/libdrm-2.4.92/etnaviv/etnaviv_device.c	2018-07-06 12:12:06.181223300 +0900
@@ -25,8 +25,10 @@
  */
 
 #include <stdlib.h>
+#if !defined(__CYGWIN__)
 #include <linux/stddef.h>
 #include <linux/types.h>
+#endif
 #include <errno.h>
 #include <sys/mman.h>
 #include <fcntl.h>
--- origsrc/libdrm-2.4.92/exynos/exynos_drm.c	2018-05-10 07:26:03.000000000 +0900
+++ src/libdrm-2.4.92/exynos/exynos_drm.c	2018-07-06 12:12:06.196823100 +0900
@@ -31,7 +31,9 @@
 #include <unistd.h>
 
 #include <sys/mman.h>
+#if !defined(__CYGWIN__)
 #include <linux/stddef.h>
+#endif
 
 #include <xf86drm.h>
 
--- origsrc/libdrm-2.4.92/exynos/exynos_fimg2d.c	2018-05-10 07:26:03.000000000 +0900
+++ src/libdrm-2.4.92/exynos/exynos_fimg2d.c	2018-07-06 12:12:06.212364800 +0900
@@ -30,7 +30,9 @@
 #include <assert.h>
 
 #include <sys/mman.h>
+#if !defined(__CYGWIN__)
 #include <linux/stddef.h>
+#endif
 
 #include <xf86drm.h>
 
--- origsrc/libdrm-2.4.92/freedreno/kgsl/kgsl_bo.c	2018-05-10 07:26:03.000000000 +0900
+++ src/libdrm-2.4.92/freedreno/kgsl/kgsl_bo.c	2018-07-06 12:12:06.212364800 +0900
@@ -28,7 +28,9 @@
 
 #include "kgsl_priv.h"
 
+#if !defined(__CYGWIN__)
 #include <linux/fb.h>
+#endif
 
 static int set_memtype(struct fd_device *dev, uint32_t handle, uint32_t flags)
 {
--- origsrc/libdrm-2.4.92/include/drm/drm.h	2018-05-10 07:26:03.000000000 +0900
+++ src/libdrm-2.4.92/include/drm/drm.h	2018-07-06 12:12:06.227992000 +0900
@@ -44,7 +44,11 @@ typedef unsigned int drm_handle_t;
 
 #else /* One of the BSDs */
 
+#if defined(__CYGWIN__)
+#include <sys/ioctl.h>
+#else
 #include <sys/ioccom.h>
+#endif
 #include <sys/types.h>
 typedef int8_t   __s8;
 typedef uint8_t  __u8;
--- origsrc/libdrm-2.4.92/omap/omap_drm.c	2018-05-10 07:26:03.000000000 +0900
+++ src/libdrm-2.4.92/omap/omap_drm.c	2018-07-06 12:12:06.259241500 +0900
@@ -27,8 +27,10 @@
  */
 
 #include <stdlib.h>
+#if !defined(__CYGWIN__)
 #include <linux/stddef.h>
 #include <linux/types.h>
+#endif
 #include <errno.h>
 #include <sys/mman.h>
 #include <fcntl.h>
--- origsrc/libdrm-2.4.92/tests/exynos/exynos_fimg2d_test.c	2018-05-10 07:26:03.000000000 +0900
+++ src/libdrm-2.4.92/tests/exynos/exynos_fimg2d_test.c	2018-07-06 12:12:06.290492000 +0900
@@ -31,7 +31,9 @@
 #include <unistd.h>
 
 #include <sys/mman.h>
+#if !defined(__CYGWIN__)
 #include <linux/stddef.h>
+#endif
 
 #include <xf86drm.h>
 #include <xf86drmMode.h>
--- origsrc/libdrm-2.4.92/tests/modetest/Makefile.am	2018-05-10 07:26:03.000000000 +0900
+++ src/libdrm-2.4.92/tests/modetest/Makefile.am	2018-07-06 12:12:06.290492000 +0900
@@ -19,6 +19,6 @@ endif
 modetest_SOURCES = $(MODETEST_FILES)
 
 modetest_LDADD = \
-	$(top_builddir)/libdrm.la \
 	$(top_builddir)/tests/util/libutil.la \
+	$(top_builddir)/libdrm.la \
 	$(CAIRO_LIBS)
--- origsrc/libdrm-2.4.92/tests/proptest/Makefile.am	2018-05-10 07:26:03.000000000 +0900
+++ src/libdrm-2.4.92/tests/proptest/Makefile.am	2018-07-06 12:12:06.321740100 +0900
@@ -17,5 +17,5 @@ endif
 proptest_SOURCES = $(PROPTEST_FILES)
 
 proptest_LDADD = \
-	$(top_builddir)/libdrm.la \
-	$(top_builddir)/tests/util/libutil.la
+	$(top_builddir)/tests/util/libutil.la \
+	$(top_builddir)/libdrm.la
--- origsrc/libdrm-2.4.92/tests/util/Makefile.am	2018-05-10 07:26:03.000000000 +0900
+++ src/libdrm-2.4.92/tests/util/Makefile.am	2018-07-06 12:12:06.337364000 +0900
@@ -10,4 +10,6 @@ libutil_la_CPPFLAGS = \
 libutil_la_CFLAGS = \
 	$(CAIRO_CFLAGS)
 
+libutil_la_LDFLAGS = $(top_builddir)/libdrm.la
+
 libutil_la_SOURCES = $(UTIL_FILES)
--- origsrc/libdrm-2.4.92/tests/vbltest/Makefile.am	2018-05-10 07:26:03.000000000 +0900
+++ src/libdrm-2.4.92/tests/vbltest/Makefile.am	2018-07-06 12:12:06.384248200 +0900
@@ -15,5 +15,5 @@ endif
 vbltest_SOURCES = \
 	vbltest.c
 vbltest_LDADD = \
-	$(top_builddir)/libdrm.la \
-	$(top_builddir)/tests/util/libutil.la
+	$(top_builddir)/tests/util/libutil.la \
+	$(top_builddir)/libdrm.la
--- origsrc/libdrm-2.4.92/xf86drm.h	2018-05-10 07:26:03.000000000 +0900
+++ src/libdrm-2.4.92/xf86drm.h	2018-07-06 12:12:06.399869300 +0900
@@ -47,7 +47,7 @@ extern "C" {
 #define DRM_MAX_MINOR   16
 #endif
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__CYGWIN__)
 
 #define DRM_IOCTL_NR(n)		_IOC_NR(n)
 #define DRM_IOC_VOID		_IOC_NONE
