--- origsrc/libllcp/libnfc-llcp/Makefile.am	2013-10-14 17:42:35.000000000 +0900
+++ src/libllcp/libnfc-llcp/Makefile.am	2013-10-14 17:47:58.872749700 +0900
@@ -29,6 +29,9 @@ if WITH_DEBUG
 libnfc_llcp_la_SOURCES += llcp_log.c
 endif
 
+libnfc_llcp_la_LDFLAGS = -no-undefined
+libnfc_llcp_la_LIBADD = $(LIBNFC_LIBS)
+
 EXTRA_DIST = \
 	     llcp_log.h \
 	     llcp_parameters.h \
--- origsrc/libllcp/libnfc-llcp/llc_link.h	2013-10-14 17:42:35.000000000 +0900
+++ src/libllcp/libnfc-llcp/llc_link.h	2013-10-14 17:49:09.325874700 +0900
@@ -24,6 +24,7 @@
 
 #include <mqueue.h>
 #include <stdint.h>
+#include <sys/time.h>
 
 #include "llcp_pdu.h"
 #include "llcp.h"
