--- origsrc/csnake-v1.8.3/makefile	2025-03-06 22:18:36.000000000 +0900
+++ src/csnake-v1.8.3/makefile	2025-04-09 20:56:27.656800000 +0900
@@ -9,7 +9,7 @@ sourcefiles = entrypoint.c csnake.c
 defined_macros = -D _COMPILE_VERSION=$(_COMPILE_VERSION)
 VERSION ?= "null"
 
-EXTRA_CFLAGS ?= `pkg-config --libs ncurses` -fdiagnostics-color=always -Wall -Wextra -Wl,-z,relro,-z,now
+EXTRA_CFLAGS ?= `pkg-config --libs ncurses` -largp
 
 CC ?= gcc
 
