--- origsrc/sparkey-sparkey-1.0.0/Makefile.am	2016-08-17 19:27:49.000000000 +0900
+++ src/sparkey-sparkey-1.0.0/Makefile.am	2025-09-14 04:46:13.455244800 +0900
@@ -1,6 +1,9 @@
 SUBDIRS = src
 dist_doc_DATA = README.md
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = sparkey.pc
+
 if HAVE_DOXYGEN
 
 doxyfile.stamp:
--- origsrc/sparkey-sparkey-1.0.0/configure.ac	2016-08-17 19:27:49.000000000 +0900
+++ src/sparkey-sparkey-1.0.0/configure.ac	2025-09-14 04:46:13.457255000 +0900
@@ -1,4 +1,4 @@
-AC_INIT([libsparkey], [0.1.0], [krka@spotify.com], [sparkey])
+AC_INIT([libsparkey], [1.0.0], [krka@spotify.com], [sparkey])
 AM_INIT_AUTOMAKE([-Werror foreign])
 LT_INIT
 AC_PROG_CC_C99
@@ -28,5 +28,6 @@ AM_COND_IF([NOT_APPLE], [
 AC_CONFIG_FILES([
  Makefile
  src/Makefile
+ sparkey.pc
 ])
 AC_OUTPUT
--- origsrc/sparkey-sparkey-1.0.0/sparkey.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/sparkey-sparkey-1.0.0/sparkey.pc.in	2025-09-14 04:46:13.459268900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: sparkey
+Description: Simple constant key/value storage library, for read-heavy systems with infrequent large bulk inserts
+Version: @VERSION@
+Libs: -L${libdir} -lsparkey
+Cflags: -I${includedir}
--- origsrc/sparkey-sparkey-1.0.0/src/Makefile.am	2016-08-17 19:27:49.000000000 +0900
+++ src/sparkey-sparkey-1.0.0/src/Makefile.am	2025-09-14 04:52:05.083018600 +0900
@@ -1,5 +1,3 @@
-CFLAGS = -O2 -Wall -Wextra -Wfloat-equal -Wshadow -Wpointer-arith -Werror -pedantic
-
 lib_LTLIBRARIES = libsparkey.la
 libsparkey_la_SOURCES = endiantools.h hashheader.h logheader.h \
 MurmurHash3.h buf.c hashalgorithms.c hashiter.c hashwriter.c \
@@ -7,12 +5,13 @@ logreader.c returncodes.c util.c buf.h h
 sparkey.h util.h endiantools.c \
 hashheader.c hashreader.c logheader.c logwriter.c MurmurHash3.c \
 sparkey-internal.h
+libsparkey_la_LDFLAGS = -no-undefined
 
 pkginclude_HEADERS = sparkey.h
 
-bin_PROGRAMS = sparkey bench
+bin_PROGRAMS = sparkey sparkey-bench
 sparkey_SOURCES = main.c
-bench_SOURCES = bench.c
+sparkey_bench_SOURCES = bench.c
 LDADD = libsparkey.la
 
 check_PROGRAMS = testvlq testhash testutil testsystem
--- origsrc/sparkey-sparkey-1.0.0/src/endiantools.c	2016-08-17 19:27:49.000000000 +0900
+++ src/sparkey-sparkey-1.0.0/src/endiantools.c	2025-09-14 04:46:13.461343100 +0900
@@ -13,7 +13,7 @@
 * License for the specific language governing permissions and limitations under
 * the License.
 */
-#if defined(__linux)
+#if defined(__linux) || defined(__CYGWIN__)
 #include <byteswap.h>
 #elif defined(__APPLE__)
 #include <libkern/OSByteOrder.h>
