--- origsrc/tweetnacl-20140427/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/tweetnacl-20140427/Makefile.am	2024-10-06 00:21:19.688088200 +0900
@@ -0,0 +1,8 @@
+include_HEADERS = tweetnacl.h
+
+lib_LTLIBRARIES = libtweetnacl.la
+libtweetnacl_la_SOURCES = tweetnacl.c
+libtweetnacl_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/tweetnacl-20140427/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/tweetnacl-20140427/configure.ac	2024-10-06 00:21:19.688088200 +0900
@@ -0,0 +1,24 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.72])
+AC_INIT([tweetnacl], [20140427])
+AC_CONFIG_SRCDIR([tweetnacl.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+
+# Checks for typedefs, structures, and compiler characteristics.
+
+# Checks for library functions.
+
+AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([tweetnacl.pc])
+AC_OUTPUT
--- origsrc/tweetnacl-20140427/tweetnacl.c	2024-10-06 00:13:58.345073100 +0900
+++ src/tweetnacl-20140427/tweetnacl.c	2024-10-06 00:13:58.891946300 +0900
@@ -7,7 +7,11 @@ typedef unsigned long u32;
 typedef unsigned long long u64;
 typedef long long i64;
 typedef i64 gf[16];
-extern void randombytes(u8 *,u64);
+#include <stdlib.h>
+sv randombytes(u8 * buf ,u64 size)
+{
+  arc4random_buf(buf, size);
+}
 
 static const u8
   _0[16],
--- origsrc/tweetnacl-20140427/tweetnacl.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/tweetnacl-20140427/tweetnacl.pc.in	2024-10-06 00:21:19.672453700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: tweetnacl
+Description: The world's first auditable high-security cryptographic library
+Version: @VERSION@
+Libs: -L${libdir} -ltweetnacl
+Cflags: -I${includedir}
