--- origsrc/cpmio-1.1.1/check/Makefile.am	2001-07-07 22:15:30.000000000 +0900
+++ src/cpmio-1.1.1/check/Makefile.am	2011-09-25 17:05:46.149919600 +0900
@@ -1,8 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
 INCLUDES=-I$(top_srcdir)/include
-LDADD=../lib/libcpmio.a
 
 check_PROGRAMS = testio
 testio_SOURCES = testio.c
-
+testio_LDADD = ../lib/libcpmio.la @NCURSES_LIBS@
--- origsrc/cpmio-1.1.1/configure.in	2003-03-29 05:04:25.000000000 +0900
+++ src/cpmio-1.1.1/configure.in	2011-09-25 16:59:15.588284400 +0900
@@ -1,9 +1,9 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(lib/cpmio.c)
+AC_INIT([cpmio], [1.1.1])
 AC_CONFIG_AUX_DIR(config)
 VERSION=1.1.1
 UPDATED='March 28, 2001'
-AM_INIT_AUTOMAKE(cpmio, $VERSION)
+AM_INIT_AUTOMAKE
 AM_CONFIG_HEADER(config.h)
 
 AC_ARG_WITH(readline,   
@@ -15,11 +15,10 @@ AC_PROG_CC
 AC_PROG_CPP
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
-AC_PROG_RANLIB
+LT_INIT([disable-static])
 
 dnl Checks for libraries.
-AC_CHECK_LIB(ncurses, printw, LIBS="-lncurses $LIBS")
-AC_CHECK_LIB(curses,  printw, LIBS="-lcurses  $LIBS")
+PKG_CHECK_MODULES(NCURSES, ncurses)
 if test x"$with_readline" = x"yes"; then
   CFLAGS="$CFLAGS -DUSE_READLINE"
   AC_CHECK_LIB(readline, readline)
@@ -28,7 +27,6 @@ fi
 dnl Checks for header files.
 AC_HEADER_STDC
 AC_CHECK_HEADERS(sys/time.h unistd.h)
-AC_CHECK_HEADERS(ncurses.h curses.h, break)
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_HEADER_TIME
--- origsrc/cpmio-1.1.1/lib/Makefile.am	2001-07-07 22:14:31.000000000 +0900
+++ src/cpmio-1.1.1/lib/Makefile.am	2011-09-25 17:13:42.992292800 +0900
@@ -1,7 +1,9 @@
 ## Process this file with automake to produce Makefile.in
 
-INCLUDES=-I$(top_srcdir)/include
+INCLUDES=-I$(top_srcdir)/include @NCURSES_CFLAGS@
 
-lib_LIBRARIES=libcpmio.a
-libcpmio_a_SOURCES = conbdos.c  cpmio_i.h  termansi.c  termcore.h \
+lib_LTLIBRARIES=libcpmio.la
+libcpmio_la_SOURCES = conbdos.c  cpmio_i.h  termansi.c  termcore.h \
                      cpmio.c    iosig.c    termcore.c  termvt52.c
+libcpmio_la_LDFLAGS = -no-undefined
+libcpmio_la_LIBADD = @NCURSES_LIBS@
--- origsrc/cpmio-1.1.1/lib/cpmio_i.h	2001-07-07 22:31:10.000000000 +0900
+++ src/cpmio-1.1.1/lib/cpmio_i.h	2011-09-25 16:57:11.827108400 +0900
@@ -23,13 +23,9 @@
 
 #include "config.h"
 
-#ifdef HAVE_NCURSES_H
 #include <ncurses.h>
-#endif
 
-#ifdef HAVE_CURSES_H
 #include <curses.h>
-#endif
 
 #include <termios.h>
 #include <signal.h>
