--- origsrc/cpmio-1.1.1/Makefile.am	2001-07-07 22:16:30.000000000 +0900
+++ src/cpmio-1.1.1/Makefile.am	2025-08-06 07:13:08.629763700 +0900
@@ -4,3 +4,5 @@ SUBDIRS= . include lib check
 
 EXTRA_DIST= $(srcdir)/config/* 
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/cpmio-1.1.1/configure.in	2025-08-05 19:29:46.453979700 +0900
+++ src/cpmio-1.1.1/configure.in	2025-08-06 07:13:28.321954800 +0900
@@ -37,4 +37,4 @@ dnl Checks for library functions.
 AC_TYPE_SIGNAL
 AC_CHECK_FUNCS(select)
 
-AC_OUTPUT(Makefile lib/Makefile include/Makefile check/Makefile)
+AC_OUTPUT(cpmio.pc Makefile lib/Makefile include/Makefile check/Makefile)
--- origsrc/cpmio-1.1.1/cpmio.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/cpmio-1.1.1/cpmio.pc.in	2025-08-06 07:12:35.197944100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: cpmio
+Description: Library intended to support the console access functions of an emulated CP/M computer
+Version: @VERSION@
+Libs: -L${libdir} -lcpmio
+Cflags: -I${includedir}
--- origsrc/cpmio-1.1.1/lib/Makefile.am	2025-08-05 19:29:46.444883800 +0900
+++ src/cpmio-1.1.1/lib/Makefile.am	2025-08-05 19:31:47.897460800 +0900
@@ -5,3 +5,5 @@ INCLUDES=-I$(top_srcdir)/include
 lib_LTLIBRARIES=libcpmio.la
 libcpmio_la_SOURCES = conbdos.c  cpmio_i.h  termansi.c  termcore.h \
                      cpmio.c    iosig.c    termcore.c  termvt52.c
+
+libcpmio_la_LDFLAGS = -no-undefined
