--- origsrc/fluent-logger-c-1.0.4/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/fluent-logger-c-1.0.4/Makefile.am	2014-03-09 16:51:57.874157900 +0900
@@ -0,0 +1,7 @@
+
+include_HEADERS = fluentlogger.h
+
+lib_LTLIBRARIES = libfluent-logger-c.la
+libfluent_logger_c_la_SOURCES = fluentlogger.c
+libfluent_logger_c_la_LDFLAGS = -no-undefined
+
--- origsrc/fluent-logger-c-1.0.4/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/fluent-logger-c-1.0.4/configure.ac	2014-03-09 16:52:48.562385800 +0900
@@ -0,0 +1,25 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([fluent-logger-c], [1.0.4])
+AC_CONFIG_SRCDIR([fluentlogger.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for header files.
+AC_CHECK_HEADERS([netdb.h stdlib.h string.h sys/socket.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([memset socket])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
