diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libjson++-0.5.0/CMakeLists.txt src/libjson++-0.5.0/CMakeLists.txt
--- origsrc/libjson++-0.5.0/CMakeLists.txt	2013-10-04 22:58:27.000000000 +0900
+++ src/libjson++-0.5.0/CMakeLists.txt	2015-11-03 09:25:38.467720300 +0900
@@ -20,3 +20,10 @@ include_directories("${PROJECT_SOURCE_DI
 add_subdirectory(include)
 add_subdirectory(src)
 
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "${libjson++_VERSION_MAJOR}.${libjson++_VERSION_MINOR}.${libjson++_VERSION_PATCH}")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/${PROJECT_NAME}.pc.in ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libjson++-0.5.0/libjson++.pc.in src/libjson++-0.5.0/libjson++.pc.in
--- origsrc/libjson++-0.5.0/libjson++.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libjson++-0.5.0/libjson++.pc.in	2015-11-03 09:25:45.106916600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libjson++
+Description: Fast and intuitive C++ interface for handling JSON structures
+Version: @VERSION@
+Libs: -L${libdir} -ljson++
+Cflags: -I${includedir}
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libjson++-0.5.0/src/CMakeLists.txt src/libjson++-0.5.0/src/CMakeLists.txt
--- origsrc/libjson++-0.5.0/src/CMakeLists.txt	2013-10-04 22:58:27.000000000 +0900
+++ src/libjson++-0.5.0/src/CMakeLists.txt	2015-11-03 09:23:47.669449900 +0900
@@ -9,16 +9,14 @@ set(json_SOURCES
 		value
    )
 
-add_library(jsonstatic STATIC ${json_SOURCES})
 add_library(jsonshared SHARED ${json_SOURCES})
 
-set_target_properties (jsonstatic PROPERTIES OUTPUT_NAME json++)
 set_target_properties (jsonshared PROPERTIES OUTPUT_NAME json++
 		VERSION ${libjson++_VERSION_MAJOR}.${libjson++_VERSION_MINOR}.${libjson++_VERSION_PATCH}
 		SOVERSION ${libjson++_VERSION_MAJOR}.${libjson++_VERSION_MINOR}
 )
 
 install(TARGETS
-		jsonstatic
 		jsonshared
-		DESTINATION lib)
+		RUNTIME DESTINATION bin
+		ARCHIVE DESTINATION lib)
