--- origsrc/signon-oauth2-0.19/common-project-config.pri	2012-11-09 17:12:21.000000000 +0900
+++ src/signon-oauth2-0.19/common-project-config.pri	2015-04-05 16:58:02.146256100 +0900
@@ -48,11 +48,7 @@ exists( meego-release ) {
     ARCH = $$system(uname -m)
 }
 
-contains( ARCH, x86_64 ) {
-    INSTALL_LIBDIR = $${INSTALL_PREFIX}/lib64
-} else {
     INSTALL_LIBDIR = $${INSTALL_PREFIX}/lib
-}
 
 # default library directory can be overriden by defining LIBDIR when
 # running qmake
