--- origsrc/stratagus-2.2.5.5/configure.in	2010-11-21 12:45:52.000000000 +0900
+++ src/stratagus-2.2.5.5/configure.in	2011-03-21 12:26:16.754432400 +0900
@@ -14,7 +14,7 @@ PATH=$prefix/bin:$PATH
 LDFLAGS="$LDFLAGS -L$prefix/lib -L$prefix/bin"
 
 OBJDIR=obj
-if test "`uname -a | grep -is cygwin`" -o "`uname -a | grep -is mingw`"; then
+if test "`uname -a | grep -is mingw`"; then
 	ARCH=win32
 	OBJDIR=winobj
 	CPPFLAGS="$CPPFLAGS -DUSE_WIN32"
