--- origsrc/nanopb-0.4.9/CMakeLists.txt	2024-09-19 19:56:19.000000000 +0900
+++ src/nanopb-0.4.9/CMakeLists.txt	2024-09-23 14:40:32.823934500 +0900
@@ -185,3 +185,10 @@ if(nanopb_BUILD_RUNTIME)
     install(FILES pb.h pb_common.h pb_encode.h pb_decode.h
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/nanopb)
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/nanopb.pc.in ${CMAKE_BINARY_DIR}/nanopb.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/nanopb.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/nanopb-0.4.9/nanopb.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/nanopb-0.4.9/nanopb.pc.in	2024-09-23 14:40:32.823934500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: nanopb
+Description: Protocol Buffers with small code size
+Version: @nanopb_VERSION@
+Libs: -L${libdir} -lprotobuf-nanopb
+Cflags: -I${includedir}
