--- origsrc/sais-lite-LCP-0.0.1/CMakeLists.txt	1970-01-01 09:00:00.000000000 +0900
+++ src/sais-lite-LCP-0.0.1/CMakeLists.txt	2025-03-10 16:28:47.474663000 +0900
@@ -0,0 +1,24 @@
+cmake_minimum_required(VERSION 3.28)
+project(sais-lite-LCP
+  VERSION 0.0.1
+  DESCRIPTION "Induced Suffix Array with LCP construction"
+  HOMEPAGE_URL "https://github.com/elventear/sais-lite-lcp")
+
+include(GNUInstallDirs)
+include(CTest)
+
+add_library(sais-lite-lcp SHARED sais.c)
+set_target_properties(sais-lite-lcp PROPERTIES SOVERSION 0)
+
+install(TARGETS sais-lite-lcp
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+install(FILES sais.h sais.hxx DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/sais-lite-lcp.pc.in ${CMAKE_BINARY_DIR}/sais-lite-lcp.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/sais-lite-lcp.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/sais-lite-LCP-0.0.1/sais-lite-lcp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/sais-lite-LCP-0.0.1/sais-lite-lcp.pc.in	2025-03-10 16:28:30.339196600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: sais-lite-LCP
+Description: Induced Suffix Array with LCP construction
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lsais-lite-lcp
+Cflags: -I${includedir}
