--- origsrc/switch-tools-1.13.1/Makefile.am	2024-10-24 00:24:50.000000000 +0900
+++ src/switch-tools-1.13.1/Makefile.am	2026-02-04 19:57:34.330956100 +0900
@@ -9,11 +9,11 @@ elf2nro_SOURCES	=	src/elf2nro.c src/elf6
 
 elf2nso_SOURCES	=	src/elf2nso.c src/sha256.c src/sha256.h src/elf64.h src/elf_common.h
 
-elf2kip_SOURCES	=	src/elf2kip.c src/cJSON.c src/cJSON.h src/blz.c src/blz.h src/elf64.h src/elf_common.h
+elf2kip_SOURCES	=	src/elf2kip.c src/blz.c src/blz.h src/elf64.h src/elf_common.h
 
 nacptool_SOURCES	=	src/nacptool.c
 
-npdmtool_SOURCES	=	src/npdmtool.c src/cJSON.c src/cJSON.h
+npdmtool_SOURCES	=	src/npdmtool.c
 
 nxlink_SOURCES = src/nxlink.c
 
@@ -26,5 +26,8 @@ elf2nso_LDADD = @LZ4_LIBS@
 elf2nro_CPPFLAGS = @LZ4_CFLAGS@
 elf2nro_LDADD = @LZ4_LIBS@
 
+elf2kip_LDADD = $(CJSON_LIBS)
+
+npdmtool_LDADD = $(CJSON_LIBS)
 
 EXTRA_DIST = autogen.sh
--- origsrc/switch-tools-1.13.1/configure.ac	2024-10-24 00:24:50.000000000 +0900
+++ src/switch-tools-1.13.1/configure.ac	2026-02-04 19:57:34.330956100 +0900
@@ -26,6 +26,8 @@ PKG_CHECK_MODULES([ZLIB], zlib, [
   AC_DEFINE([HAVE_LIBZ], [1], [Define if using zlib.])
 ])
 
+PKG_CHECK_MODULES([CJSON], [libcjson])
+
 NET_LIBS=""
 
 case "$host" in
