--- origsrc/assimp-4.0.1/CMakeLists.txt	2017-07-29 05:33:41.000000000 +0900
+++ src/assimp-4.0.1/CMakeLists.txt	2017-11-27 11:31:26.832000000 +0900
@@ -184,7 +184,7 @@ ENDIF( UNIX )
 
 
 # Grouped compiler settings
-IF ((CMAKE_C_COMPILER_ID MATCHES "GNU") AND NOT CMAKE_COMPILER_IS_MINGW)
+IF ((CMAKE_C_COMPILER_ID MATCHES "GNU") AND NOT CMAKE_COMPILER_IS_MINGW AND NOT CYGWIN)
   # hide all not-exported symbols
   SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -fvisibility=hidden -fPIC -Wall -std=c++0x")
   SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
@@ -244,7 +244,7 @@ ENDIF()
 # cmake configuration files
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/assimp-config.cmake.in"         "${CMAKE_CURRENT_BINARY_DIR}/assimp-config.cmake" @ONLY IMMEDIATE)
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/assimp-config-version.cmake.in" "${CMAKE_CURRENT_BINARY_DIR}/assimp-config-version.cmake" @ONLY IMMEDIATE)
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/assimp-config.cmake"             "${CMAKE_CURRENT_BINARY_DIR}/assimp-config-version.cmake" DESTINATION "${ASSIMP_LIB_INSTALL_DIR}/cmake/assimp-${ASSIMP_VERSION_MAJOR}.${ASSIMP_VERSION_MINOR}" COMPONENT ${LIBASSIMP-DEV_COMPONENT})
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/assimp-config.cmake"             "${CMAKE_CURRENT_BINARY_DIR}/assimp-config-version.cmake" DESTINATION "${ASSIMP_LIB_INSTALL_DIR}/cmake/assimp" COMPONENT ${LIBASSIMP-DEV_COMPONENT})
 
 FIND_PACKAGE( DirectX )
 
--- origsrc/assimp-4.0.1/tools/assimp_cmd/CMakeLists.txt	2017-07-29 05:33:41.000000000 +0900
+++ src/assimp-4.0.1/tools/assimp_cmd/CMakeLists.txt	2017-11-27 11:32:27.458000000 +0900
@@ -41,6 +41,7 @@ cmake_minimum_required( VERSION 2.6 )
 INCLUDE_DIRECTORIES(
   ${Assimp_SOURCE_DIR}/include
   ${Assimp_SOURCE_DIR}/code
+  ${Assimp_BINARY_DIR}
 )
 
 LINK_DIRECTORIES( ${Assimp_BINARY_DIR} ${Assimp_BINARY_DIR}/lib )
--- origsrc/assimp-4.0.1/tools/assimp_cmd/assimp_cmd.rc	2017-07-29 05:33:41.000000000 +0900
+++ src/assimp-4.0.1/tools/assimp_cmd/assimp_cmd.rc	2017-11-27 12:00:59.110198700 +0900
@@ -1,7 +1,7 @@
 // Microsoft Visual C++ generated resource script.
 //
 #include "resource.h"
-#include "../../revision.h"
+#include "revision.h"
 
 #define APSTUDIO_READONLY_SYMBOLS
 /////////////////////////////////////////////////////////////////////////////
