--- origsrc/yubico-piv-tool-yubico-piv-tool-2.1.1/lib/internal.h	2020-07-20 18:37:51.000000000 +0900
+++ src/yubico-piv-tool-yubico-piv-tool-2.1.1/lib/internal.h	2020-08-07 11:56:18.660973200 +0900
@@ -247,7 +247,7 @@ void yc_log_event(uint32_t id, yc_log_le
 #elif defined(BSD)
 #include <strings.h>
 #define yc_memzero explicit_bzero
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__CYGWIN__)
 #include <openssl/crypto.h>
 #define yc_memzero OPENSSL_cleanse
 #else
