--- origsrc/xmlindent-0.2.17/Makefile	2004-09-07 20:36:01.000000000 +0900
+++ src/xmlindent-0.2.17/Makefile	2013-10-27 20:36:03.896806700 +0900
@@ -1,11 +1,11 @@
-all: xmlindent
+all: xmlindent.exe
 
 PREFIX=/usr/local
 BIN_INSTALL_DIR=$(PREFIX)/bin
 MAN_INSTALL_DIR=$(PREFIX)/share/man/man1
 
-xmlindent: indent.c buffer.c main.c
-	gcc -Wall -g error.c indent.c buffer.c main.c -o xmlindent -lfl
+xmlindent.exe: indent.c buffer.c main.c
+	$(CC) -Wall $(CFLAGS) error.c indent.c buffer.c main.c -o $@ -lfl
 
 indent.c: lex.yy.c
 
