--- origsrc/libccp4-8.0.0/Makefile.am	2022-05-24 01:11:07.000000000 +0900
+++ src/libccp4-8.0.0/Makefile.am	2024-10-12 01:03:09.815174200 +0900
@@ -65,12 +65,13 @@ fortran_libccp4f_la_LDFLAGS = -no-undefi
 # They are included in CCP4 and may be needed for running programs
 # compiled with libccp4 (see http://www.ccp4.ac.uk/dist/html/ccp4.html)
 
-dist_ccp4data_DATA = ccp4-config.cmake
-dist_data_DATA = data/atomsf_neutron.lib data/atomsf_electron.lib \
+libcmakedir = $(libdir)/cmake/libccp4
+libcmake_DATA = ccp4-config.cmake
+pkgdata_DATA = data/atomsf_neutron.lib data/atomsf_electron.lib \
 		     data/atomsf.lib data/syminfo.lib data/symop.lib
 pkgconfig_DATA = ccp4c.pc
 if FORTRAN_LIB
-  dist_data_DATA += data/font84.ascii
+  pkgdata_DATA += data/font84.ascii
   pkgconfig_DATA += ccp4f.pc
 endif
 
--- origsrc/libccp4-8.0.0/ccp4f.pc.in	2022-05-24 01:11:07.000000000 +0900
+++ src/libccp4-8.0.0/ccp4f.pc.in	2024-10-12 01:10:23.143189200 +0900
@@ -6,6 +6,6 @@ includedir=@includedir@
 Name: libccp4f
 Description: CCP4 core FORTRAN libraries
 Version: @VERSION@
-Requires.private: libccp4c mmdb2
+Requires.private: ccp4c mmdb2
 Libs: -L${libdir} -lccp4f
 Cflags: -I${includedir}
--- origsrc/libccp4-8.0.0/configure.ac	2022-05-24 01:11:07.000000000 +0900
+++ src/libccp4-8.0.0/configure.ac	2024-10-12 01:03:53.986999000 +0900
@@ -38,7 +38,7 @@ if test "x$enable_fortran" != "xno"; the
   # but I'm afraid of changing this and unix.m4.
   cc_basename=`$ECHO "X$compiler_F77" | $Xsed -e 's%.*/%%' -e "s%^$host_alias-%%"`
   case $host_os in
-    linux*|mingw* )
+    linux*|mingw*|cygwin* )
       case $cc_basename,$host_cpu in
         ifort*)
           M4_ARG=${M4_ARG:-"-D_ifc=8"}
