--- origsrc/libccp4-6.5.1/Makefile.am	2015-03-25 23:16:29.000000000 +0900
+++ src/libccp4-6.5.1/Makefile.am	2018-07-07 00:15:15.856583600 +0900
@@ -65,12 +65,13 @@ fortran_libccp4f_la_LDFLAGS = -no-undefi
 # They are included in CCP4 and may be needed for running programs
 # compiled with libccp4 (see http://www.ccp4.ac.uk/dist/html/ccp4.html)
 
-dist_ccp4data_DATA = ccp4-config.cmake
-dist_data_DATA = data/atomsf_neutron.lib data/atomsf_electron.lib \
+libcmakedir = $(libdir)/cmake/libccp4
+libcmake_DATA = ccp4-config.cmake
+pkgdata_DATA = data/atomsf_neutron.lib data/atomsf_electron.lib \
 		     data/atomsf.lib data/syminfo.lib data/symop.lib
 pkgconfig_DATA = libccp4c.pc
 if FORTRAN_LIB
-  dist_data_DATA += data/font84.ascii
+  pkgdata_DATA += data/font84.ascii
   pkgconfig_DATA += libccp4f.pc
 endif
 
--- origsrc/libccp4-6.5.1/configure.ac	2015-10-05 08:10:10.000000000 +0900
+++ src/libccp4-6.5.1/configure.ac	2018-07-07 00:15:15.887832300 +0900
@@ -23,7 +23,7 @@ AC_ARG_ENABLE(fortran, AS_HELP_STRING([-
 AM_CONDITIONAL([FORTRAN_LIB], [ test "x$enable_fortran" != xno ])
 
 if test "x$enable_fortran" != "xno"; then
-  # sorting_main.f (mix of F77 and F90) cannot be compiled with g77 
+  # sorting_main.f (mix of F77 and F90) cannot be compiled with g77
   AC_PROG_F77([_AC_F90_FC _AC_F95_FC _AC_F77_FC])
   # AC_PROG_F77 doesn't fail if compiler is not found, test it explicitely
   AC_LANG_PUSH(Fortran 77)
@@ -38,7 +38,7 @@ if test "x$enable_fortran" != "xno"; the
   # but I'm afraid of changing this and unix.m4.
   cc_basename=`$ECHO "X$compiler_F77" | $Xsed -e 's%.*/%%' -e "s%^$host_alias-%%"`
   case $host_os in
-    linux*|mingw* )
+    linux*|mingw*|cygwin* )
       case $cc_basename,$host_cpu in
         ifort*)
           M4_ARG=${M4_ARG:-"-D_ifc=8"}
@@ -119,4 +119,3 @@ AC_DEFINE_UNQUOTED([PACKAGE_ROOT],["$PAC
 AC_SUBST(M4_ARG)
 
 AC_OUTPUT(Makefile libccp4c.pc libccp4f.pc)
-
--- origsrc/libccp4-6.5.1/libccp4f.pc.in	2013-06-02 01:37:01.000000000 +0900
+++ src/libccp4-6.5.1/libccp4f.pc.in	2018-07-07 00:15:15.887832300 +0900
@@ -6,6 +6,6 @@ includedir=@includedir@
 Name: libccp4f
 Description: CCP4 core FORTRAN libraries
 Version: @VERSION@
-Requires.private: libccp4c mmdb
+Requires.private: libccp4c mmdb2
 Libs: -L${libdir} -lccp4f
 Cflags: -I${includedir}
