--- origsrc/libccp4-6.4.6/Makefile.am	2014-09-02 00:27:29.000000000 +0900
+++ src/libccp4-6.4.6/Makefile.am	2015-02-26 12:36:24.300949800 +0900
@@ -65,12 +65,13 @@ fortran_libccp4f_la_LDFLAGS = -no-undefi
 # They are included in CCP4 and may be needed for running programs
 # compiled with libccp4 (see http://www.ccp4.ac.uk/dist/html/ccp4.html)
 
-dist_ccp4data_DATA = ccp4-config.cmake
-dist_data_DATA = data/atomsf_neutron.lib \
+libcmakedir = $(libdir)/cmake/libccp4
+libcmake_DATA = ccp4-config.cmake
+pkgdata_DATA = data/atomsf_neutron.lib \
 		     data/atomsf.lib data/syminfo.lib data/symop.lib
 pkgconfig_DATA = libccp4c.pc
 if FORTRAN_LIB
-  dist_data_DATA += data/font84.ascii
+  pkgdata_DATA += data/font84.ascii
   pkgconfig_DATA += libccp4f.pc
 endif
 
--- origsrc/libccp4-6.4.6/configure.ac	2014-09-11 21:54:14.000000000 +0900
+++ src/libccp4-6.4.6/configure.ac	2015-02-26 12:28:02.048996700 +0900
@@ -38,7 +38,7 @@ if test "x$enable_fortran" != "xno"; the
   # but I'm afraid of changing this and unix.m4.
   cc_basename=`$ECHO "X$compiler_F77" | $Xsed -e 's%.*/%%' -e "s%^$host_alias-%%"`
   case $host_os in
-    linux*|mingw* )
+    linux*|mingw*|cygwin* )
       case $cc_basename,$host_cpu in
         ifort*)
           M4_ARG=${M4_ARG:-"-D_ifc=8"}
