--- origsrc/libmodelfile/configure.ac	2010-11-28 02:15:08.000000000 +0900
+++ src/libmodelfile/configure.ac	2010-11-28 02:15:38.516945900 +0900
@@ -22,8 +22,7 @@ AC_SUBST(LIBMF_INTERFACE_VERSION)
 
 dnl Test for GNU libtool
 
-AC_DISABLE_STATIC
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 dnl Test for C compiler
 
--- origsrc/libmodelfile/libcolladadae/Makefile.am	2010-11-28 02:15:08.000000000 +0900
+++ src/libmodelfile/libcolladadae/Makefile.am	2010-11-28 02:16:06.186733100 +0900
@@ -4,7 +4,7 @@ INCLUDES = -I$(top_srcdir)
 
 lib_LTLIBRARIES = libcolladadae-0.2.la
 
-libcolladadae_0_2_la_LDFLAGS = $(LIBMF_INTERFACE_VERSION)
+libcolladadae_0_2_la_LDFLAGS = $(LIBMF_INTERFACE_VERSION) -no-undefined
 
 libcolladadae_0_2_la_SOURCES = parser.c
 
--- origsrc/libmodelfile/libmd3/Makefile.am	2010-11-28 02:15:08.000000000 +0900
+++ src/libmodelfile/libmd3/Makefile.am	2010-11-28 02:16:20.397166700 +0900
@@ -4,7 +4,7 @@ INCLUDES = -I$(top_srcdir)
 
 lib_LTLIBRARIES = libmd3-0.2.la
 
-libmd3_0_2_la_LDFLAGS = $(LIBMF_INTERFACE_VERSION)
+libmd3_0_2_la_LDFLAGS = $(LIBMF_INTERFACE_VERSION) -no-undefined
 
 libmd3_0_2_la_SOURCES = convert.c loader.c endian.c endian.h
 
