--- origsrc/libmodelfile/libcolladadae/Makefile.am	2012-04-12 16:23:35.000000000 +0900
+++ src/libmodelfile/libcolladadae/Makefile.am	2012-04-12 16:23:38.469999500 +0900
@@ -4,7 +4,7 @@ INCLUDES = -I$(top_srcdir)
 
 lib_LTLIBRARIES = libcolladadae-0.2.la
 
-libcolladadae_0_2_la_LDFLAGS = $(LIBMF_INTERFACE_VERSION)
+libcolladadae_0_2_la_LDFLAGS = $(LIBMF_INTERFACE_VERSION) -no-undefined
 
 libcolladadae_0_2_la_SOURCES = parser.c
 
--- origsrc/libmodelfile/libmd3/Makefile.am	2012-04-12 16:23:35.000000000 +0900
+++ src/libmodelfile/libmd3/Makefile.am	2012-04-12 16:23:38.469999500 +0900
@@ -4,7 +4,7 @@ INCLUDES = -I$(top_srcdir)
 
 lib_LTLIBRARIES = libmd3-0.2.la
 
-libmd3_0_2_la_LDFLAGS = $(LIBMF_INTERFACE_VERSION)
+libmd3_0_2_la_LDFLAGS = $(LIBMF_INTERFACE_VERSION) -no-undefined
 
 libmd3_0_2_la_SOURCES = convert.c loader.c endian.c endian.h
 
