diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libwebsockets-1.6.1/CMakeLists.txt src/libwebsockets-1.6.1/CMakeLists.txt
--- origsrc/libwebsockets-1.6.1/CMakeLists.txt	2016-01-22 16:55:13.000000000 +0900
+++ src/libwebsockets-1.6.1/CMakeLists.txt	2016-01-25 20:10:28.603223500 +0900
@@ -485,11 +485,11 @@ if (CMAKE_COMPILER_IS_GNUCC OR CMAKE_COM
     if (LWS_HAVE_VISIBILITY)
                 set(VISIBILITY_FLAG -fvisibility=hidden)
     endif()
-    if (UNIX)
+    if (UNIX AND NOT CYGWIN)
                set(CMAKE_C_FLAGS "-Wall -Werror ${VISIBILITY_FLAG} ${CMAKE_C_FLAGS}" )
-    else(UNIX)
+    else(UNIX AND NOT CYGWIN)
                set(CMAKE_C_FLAGS "-Wall ${VISIBILITY_FLAG} ${CMAKE_C_FLAGS}" )
-    endif(UNIX)
+    endif(UNIX AND NOT CYGWIN)
 endif ()
 
 if (CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
@@ -661,8 +661,8 @@ endif(LWS_WITH_SSL)
 
 if (LWS_WITH_LIBEV)
 	if (NOT LIBEV_FOUND)
-		find_path(LIBEV_INCLUDE_DIRS NAMES ev.h)
-		find_library(LIBEV_LIBRARIES NAMES ev)
+		pkg_check_modules(LIBEV REQUIRED libev)
+		include_directories("${LIBEV_INCLUDEDIR}")
 		if(LIBEV_INCLUDE_DIRS AND LIBEV_LIBRARIES)
 			set(LIBEV_FOUND 1)
 		endif()
