--- origsrc/libwebsockets-4.0.20/CMakeLists.txt	2020-07-09 22:11:46.000000000 +0900
+++ src/libwebsockets-4.0.20/CMakeLists.txt	2020-07-18 15:03:58.801847400 +0900
@@ -2440,7 +2440,7 @@ if ((LWS_ROLE_H1 OR LWS_ROLE_H2) AND NOT
 
 	if (UNIX AND LWS_WITH_PLUGINS)
 		set(CMAKE_C_FLAGS "-fPIC ${CMAKE_C_FLAGS}")
-		if(NOT((${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD") OR (${CMAKE_SYSTEM_NAME} MATCHES "QNX")))
+		if(NOT((${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD") OR (${CMAKE_SYSTEM_NAME} MATCHES "QNX") OR (CYGWIN)))
 			target_link_libraries(websockets dl)
 		endif()
 	endif()
--- origsrc/libwebsockets-4.0.20/lib/plat/unix/unix-sockets.c	2020-07-09 22:11:46.000000000 +0900
+++ src/libwebsockets-4.0.20/lib/plat/unix/unix-sockets.c	2020-07-18 15:03:58.833123100 +0900
@@ -29,7 +29,7 @@
 
 #include <sys/ioctl.h>
 
-#if !defined(LWS_DETECTED_PLAT_IOS)
+#if !defined(LWS_DETECTED_PLAT_IOS) && !defined(__CYGWIN__)
 #include <net/route.h>
 #endif
 
