--- origsrc/libwebsockets-3.0.0/CMakeLists.txt	2018-05-07 14:38:53.000000000 +0900
+++ src/libwebsockets-3.0.0/CMakeLists.txt	2018-09-22 14:42:38.407991700 +0900
@@ -1603,7 +1603,7 @@ if ((LWS_ROLE_H1 OR LWS_ROLE_H2) AND NOT
 
 	if (UNIX AND LWS_WITH_PLUGINS)
 		set(CMAKE_C_FLAGS "-fPIC ${CMAKE_C_FLAGS}")
-		if(NOT((${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD") OR (${CMAKE_SYSTEM_NAME} MATCHES "QNX")))
+		if(NOT((${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD") OR (${CMAKE_SYSTEM_NAME} MATCHES "QNX") OR (CYGWIN)))
 			target_link_libraries(websockets dl)
 		endif()
 	endif()
