--- origsrc/libwebsockets-2.2.2/CMakeLists.txt	2017-07-19 02:05:49.000000000 +0900
+++ src/libwebsockets-2.2.2/CMakeLists.txt	2017-07-22 21:45:10.376078000 +0900
@@ -699,7 +699,7 @@ if (CMAKE_COMPILER_IS_GNUCC OR CMAKE_COM
     if (LWS_HAVE_VISIBILITY)
                 set(VISIBILITY_FLAG -fvisibility=hidden)
     endif()
-    if (UNIX OR LWS_WITH_ESP8266)
+    if ((UNIX OR LWS_WITH_ESP8266) AND NOT CYGWIN)
                set(CMAKE_C_FLAGS "-Wall -Werror ${VISIBILITY_FLAG} ${CMAKE_C_FLAGS}" )
     else()
                set(CMAKE_C_FLAGS "-Wall ${VISIBILITY_FLAG} ${CMAKE_C_FLAGS}" )
