--- origsrc/libwebsockets/CMakeLists.txt	2013-12-01 17:52:13.000000000 +0900
+++ src/libwebsockets/CMakeLists.txt	2013-12-01 18:02:56.252973400 +0900
@@ -310,7 +310,7 @@ endif(UNIX)
 
 
 if (CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
-	set( CMAKE_C_FLAGS "-Wall -Werror -O4 -fvisibility=hidden ${CMAKE_C_FLAGS}" )
+	set( CMAKE_C_FLAGS "-Wall -O4 -fvisibility=hidden ${CMAKE_C_FLAGS}" )
 endif ()
 
 source_group("Headers Private"  FILES ${HDR_PRIVATE})
@@ -320,10 +320,12 @@ source_group("Sources"          FILES ${
 #
 # Create the lib.
 #
+if (NOT CYGWIN)
 add_library(websockets STATIC
 			${HDR_PRIVATE}
 			${HDR_PUBLIC}
 			${SOURCES})
+endif (NOT CYGWIN)
 add_library(websockets_shared SHARED
 			${HDR_PRIVATE}
 			${HDR_PUBLIC}
@@ -355,11 +357,17 @@ set_target_properties(websockets_shared
 # Set the so version of the lib.
 # Equivalent to LDFLAGS=-version-info x:x:x
 if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
-	foreach(lib websockets websockets_shared)
-		set_target_properties(${lib} 
+	if (CYGWIN)
+		set_target_properties(websockets_shared
 			PROPERTIES
 			SOVERSION ${SOVERSION})
-	endforeach()
+	else (CYGWIN)
+		foreach(lib websockets websockets_shared)
+			set_target_properties(${lib} 
+				PROPERTIES
+				SOVERSION ${SOVERSION})
+		endforeach()
+	endif (CYGWIN)
 endif()
 
 set(LIB_LIST)
@@ -408,9 +416,13 @@ if (NOT WITHOUT_EXTENSIONS)
 	endif()
 
 	# Make sure ZLib is compiled before the libs.
-	foreach (lib websockets websockets_shared)
-		add_dependencies(${lib} ZLIB)
-	endforeach()
+	if (CYGWIN)
+		add_dependencies(websockets_shared ZLIB)
+	else (CYGWIN)
+		foreach (lib websockets websockets_shared)
+			add_dependencies(${lib} ZLIB)
+		endforeach()
+	endif (CYGWIN)
 
 	message("ZLib include dirs: ${ZLIB_INCLUDE_DIRS}")
 	message("ZLib libraries: ${ZLIB_LIBRARIES}")
@@ -462,9 +474,13 @@ if (UNIX)
 endif()
 
 # Setup the linking for all libs.
+if (CYGWIN)
+	target_link_libraries(websockets_shared ${LIB_LIST})
+else (CYGWIN)
 foreach (lib websockets websockets_shared)
 	target_link_libraries(${lib} ${LIB_LIST})
 endforeach()
+endif (CYGWIN)
 
 #
 # Test applications
@@ -727,11 +743,19 @@ install(FILES ${HDR_PUBLIC}
 set(CPACK_COMPONENT_HEADERS_DISPLAY_NAME "Header files")
 
 # Install libs.
+if (CYGWIN)
+install(TARGETS websockets_shared
+		LIBRARY DESTINATION lib${LIB_SUFFIX}
+		ARCHIVE DESTINATION lib${LIB_SUFFIX}
+		RUNTIME DESTINATION bin
+		COMPONENT libraries)
+else (CYGWIN)
 install(TARGETS websockets websockets_shared
 		LIBRARY DESTINATION lib${LIB_SUFFIX}
 		ARCHIVE DESTINATION lib${LIB_SUFFIX}
 		RUNTIME DESTINATION bin
 		COMPONENT libraries)
+endif (CYGWIN)
 set(CPACK_COMPONENT_LIBRARIES_DISPLAY_NAME "Libraries")
 
 # Install test apps.
