--- origsrc/libwebsockets-3.1.0/CMakeLists.txt	2018-11-23 12:30:56.000000000 +0900
+++ src/libwebsockets-3.1.0/CMakeLists.txt	2019-02-24 09:11:45.932900900 +0900
@@ -1820,7 +1820,7 @@ if ((LWS_ROLE_H1 OR LWS_ROLE_H2) AND NOT
 
 	if (UNIX AND LWS_WITH_PLUGINS)
 		set(CMAKE_C_FLAGS "-fPIC ${CMAKE_C_FLAGS}")
-		if(NOT((${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD") OR (${CMAKE_SYSTEM_NAME} MATCHES "QNX")))
+		if(NOT((${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD") OR (${CMAKE_SYSTEM_NAME} MATCHES "QNX") OR (CYGWIN)))
 			target_link_libraries(websockets dl)
 		endif()
 	endif()
