--- origsrc/libwebsockets-4.0.13/CMakeLists.txt	2020-05-25 21:22:51.000000000 +0900
+++ src/libwebsockets-4.0.13/CMakeLists.txt	2020-06-01 11:02:47.596401500 +0900
@@ -2440,7 +2440,7 @@ if ((LWS_ROLE_H1 OR LWS_ROLE_H2) AND NOT
 
 	if (UNIX AND LWS_WITH_PLUGINS)
 		set(CMAKE_C_FLAGS "-fPIC ${CMAKE_C_FLAGS}")
-		if(NOT((${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD") OR (${CMAKE_SYSTEM_NAME} MATCHES "QNX")))
+		if(NOT((${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD") OR (${CMAKE_SYSTEM_NAME} MATCHES "QNX") OR (CYGWIN)))
 			target_link_libraries(websockets dl)
 		endif()
 	endif()
--- origsrc/libwebsockets-4.0.13/lib/plat/unix/unix-sockets.c	2020-05-25 21:22:51.000000000 +0900
+++ src/libwebsockets-4.0.13/lib/plat/unix/unix-sockets.c	2020-06-01 11:12:46.108669200 +0900
@@ -29,7 +29,7 @@
 
 #include <sys/ioctl.h>
 
-#if !defined(LWS_DETECTED_PLAT_IOS)
+#if !defined(LWS_DETECTED_PLAT_IOS) && !defined(__CYGWIN__)
 #include <net/route.h>
 #endif
 
