--- origsrc/libwebsockets/lib/Makefile.am	2012-03-28 18:22:12.000000000 +0900
+++ src/libwebsockets/lib/Makefile.am	2012-03-28 18:31:21.443037800 +0900
@@ -19,15 +19,15 @@ else
 dist_libwebsockets_la_SOURCES += md5.c sha-1.c
 endif
 
-libwebsockets_la_CFLAGS:=-rdynamic -fPIC -Wall -Werror -std=gnu99 -pedantic -c \
+libwebsockets_la_CFLAGS:=-Wall -std=gnu99 -pedantic -c \
 	-DDATADIR=\"@datadir@\" -DLWS_OPENSSL_CLIENT_CERTS=\"@clientcertdir@\"
-libwebsockets_la_LDFLAGS=-lz -version-info 0:3
+libwebsockets_la_LDFLAGS=-lz -version-info 0:3 -no-undefined
 
 all-local:
-	 ../scripts/kernel-doc -html \
-                libwebsockets.c \
-		parsers.c \
-		client-handshake.c \
-                libwebsockets.h \
+	 $(top_srcdir)/scripts/kernel-doc -html \
+                $(srcdir)/libwebsockets.c \
+		$(srcdir)/parsers.c \
+		$(srcdir)/client-handshake.c \
+                $(srcdir)/libwebsockets.h \
 			> ../libwebsockets-api-doc.html
 
--- origsrc/libwebsockets/test-server/Makefile.am	2012-03-28 18:22:12.000000000 +0900
+++ src/libwebsockets/test-server/Makefile.am	2012-03-28 18:33:15.155928400 +0900
@@ -39,6 +39,6 @@ clean-local:
 
 install-data-local:libwebsockets-test-server.key.pem libwebsockets-test-server.pem
 	mkdir -p $(DESTDIR)$(datadir)/libwebsockets-test-server
-	cp -a test.html favicon.ico libwebsockets-test-server.key.pem libwebsockets-test-server.pem \
+	cp -a $(srcdir)/test.html $(srcdir)/favicon.ico libwebsockets-test-server.key.pem libwebsockets-test-server.pem \
 		$(DESTDIR)$(datadir)/libwebsockets-test-server
 
