--- origsrc/libwebsockets-3.2.0/CMakeLists.txt	2019-08-14 17:22:50.000000000 +0900
+++ src/libwebsockets-3.2.0/CMakeLists.txt	2019-10-04 10:35:34.527544700 +0900
@@ -2109,7 +2109,7 @@ if ((LWS_ROLE_H1 OR LWS_ROLE_H2) AND NOT
 
 	if (UNIX AND LWS_WITH_PLUGINS)
 		set(CMAKE_C_FLAGS "-fPIC ${CMAKE_C_FLAGS}")
-		if(NOT((${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD") OR (${CMAKE_SYSTEM_NAME} MATCHES "QNX")))
+		if(NOT((${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD") OR (${CMAKE_SYSTEM_NAME} MATCHES "QNX") OR (CYGWIN)))
 			target_link_libraries(websockets dl)
 		endif()
 	endif()
