--- origsrc/platform-p8-platform-2.1.0.1/CMakeLists.txt	2016-10-26 19:51:27.000000000 +0900
+++ src/platform-p8-platform-2.1.0.1/CMakeLists.txt	2017-09-30 19:54:02.224000000 +0900
@@ -41,7 +41,7 @@ target_link_libraries(p8-platform ${p8-p
 set_target_properties(p8-platform PROPERTIES VERSION ${p8-platform_VERSION_MAJOR}.${p8-platform_VERSION_MINOR}.${p8-platform_VERSION_PATCH}
                                               SOVERSION ${p8-platform_VERSION_MAJOR})
 
-install(TARGETS p8-platform DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS p8-platform RUNTIME DESTINATION bin ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(FILES src/os.h DESTINATION include/p8-platform)
 IF(WIN32)
   INSTALL(FILES src/windows/dlfcn-win32.h
@@ -81,5 +81,5 @@ ENDIF(NOT WIN32)
 configure_file (p8-platform-config.cmake.in
                 p8-platform-config.cmake @ONLY)
 install(FILES ${CMAKE_BINARY_DIR}/p8-platform-config.cmake
-        DESTINATION ${CMAKE_INSTALL_LIBDIR}/p8-platform)
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/p8-platform)
 
