--- origsrc/gcontainer-1.99.3/configure.ac	2008-08-11 04:34:39.000000000 +0900
+++ src/gcontainer-1.99.3/configure.ac	2010-10-16 18:39:58.690175800 +0900
@@ -18,8 +18,7 @@ AC_CANONICAL_HOST
 ##################################################
 
 AC_PROG_CC
-AC_DISABLE_SHARED
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 
 ##################################################
--- origsrc/gcontainer-1.99.3/gcontainer/Makefile.am	2008-08-11 04:34:39.000000000 +0900
+++ src/gcontainer-1.99.3/gcontainer/Makefile.am	2010-10-16 18:38:43.065175800 +0900
@@ -11,7 +11,7 @@ libgcontainer_include_DATA =	gbin.h \
 				gcontainerable.h
 
 lib_LTLIBRARIES = 		libgcontainer.la
-libgcontainer_la_LDFLAGS =	-release @PACKAGE_VERSION@
+libgcontainer_la_LDFLAGS =	-release @PACKAGE_VERSION@ -no-undefined
 libgcontainer_la_SOURCES =	gbin.c \
 				gbin.h \
 				gbinprivate.h \
@@ -27,3 +27,4 @@ libgcontainer_la_SOURCES =	gbin.c \
 				gcontainerable.h \
 				gcontainerintl.h \
 				gobjectmissings.h
+libgcontainer_la_LIBADD =	@GOBJECT_LIBS@
