--- origsrc/gcontainer-1.99.3/Makefile.am	2008-08-11 04:33:28.000000000 +0900
+++ src/gcontainer-1.99.3/Makefile.am	2025-12-21 16:08:50.677449700 +0900
@@ -4,11 +4,5 @@ SUBDIRS =			gcontainer \
 				test \
 				docs
 
-dist_pkgdata_DATA =		NEWS \
-				README \
-				AUTHORS \
-				ChangeLog \
-				COPYING \
-				TODO
 pkgconfigdir =			$(libdir)/pkgconfig
 dist_pkgconfig_DATA =		gcontainer.pc
--- origsrc/gcontainer-1.99.3/gcontainer/Makefile.am	2008-08-11 04:33:28.000000000 +0900
+++ src/gcontainer-1.99.3/gcontainer/Makefile.am	2025-12-21 16:08:50.681620100 +0900
@@ -11,7 +11,7 @@ libgcontainer_include_DATA =	gbin.h \
 				gcontainerable.h
 
 lib_LTLIBRARIES = 		libgcontainer.la
-libgcontainer_la_LDFLAGS =	-release @PACKAGE_VERSION@
+libgcontainer_la_LDFLAGS =	-no-undefined -release @PACKAGE_VERSION@
 libgcontainer_la_SOURCES =	gbin.c \
 				gbin.h \
 				gbinprivate.h \
@@ -27,3 +27,4 @@ libgcontainer_la_SOURCES =	gbin.c \
 				gcontainerable.h \
 				gcontainerintl.h \
 				gobjectmissings.h
+libgcontainer_la_LIBADD =	@GOBJECT_LIBS@
--- origsrc/gcontainer-1.99.3/gcontainer.pc	2008-08-19 10:45:32.000000000 +0900
+++ src/gcontainer-1.99.3/gcontainer.pc	2025-12-21 16:11:34.960867300 +0900
@@ -1,5 +1,5 @@
-prefix=/usr/local
-exec_prefix=${prefix}
+prefix=/usr
+exec_prefix=/usr
 
 libdir=${exec_prefix}/lib
 datarootdir=${prefix}/share
