--- origsrc/rremove-1.0.5/Makefile.am	2012-02-19 09:57:02.000000000 +0900
+++ src/rremove-1.0.5/Makefile.am	2012-05-01 10:19:51.056256300 +0900
@@ -1,6 +1,6 @@
 lib_LTLIBRARIES = librremove.la
      librremove_la_SOURCES = rremove.c
-     librremove_la_LDFLAGS = -version-info 0:3:0
+     librremove_la_LDFLAGS = -version-info 0:3:0 -no-undefined
      librremove_la_CFLAGS = -Wall -Wextra -pedantic -std=c99
 rremove_includedir = $(includedir)
 nobase_rremove_include_HEADERS = rremove.h
