--- origsrc/libtwin/Makefile.am	2024-11-14 07:06:58.000000000 +0900
+++ src/libtwin/Makefile.am	2024-11-14 09:05:17.270032200 +0900
@@ -21,7 +21,7 @@ bin_PROGRAMS =
 lib_LTLIBRARIES = libtwin/libtwin.la
 pkginclude_HEADERS = libtwin/twin.h
 
-libtwin_libtwin_la_LDFLAGS = -version-info @LIB_VERSION@
+libtwin_libtwin_la_LDFLAGS = -no-undefined -version-info @LIB_VERSION@
 libtwin_libtwin_la_SOURCES = \
 	libtwin/twin_box.c \
 	libtwin/twin_button.c \
@@ -55,6 +55,8 @@ libtwin_libtwin_la_SOURCES = \
 	libtwin/twin_work.c \
 	libtwin/twinint.h
 
+libtwin_libtwin_la_LIBADD = @TWIN_DEP_LDFLAGS@
+
 if TWIN_X11
 pkginclude_HEADERS += libtwin/twin_x11.h
 libtwin_libtwin_la_SOURCES += libtwin/twin_x11.c
--- origsrc/libtwin/libtwin/twin_draw.c	2024-11-14 07:06:58.000000000 +0900
+++ src/libtwin/libtwin/twin_draw.c	2024-11-14 07:11:52.291136200 +0900
@@ -19,6 +19,8 @@
  * Boston, MA 02111-1307, USA.
  */
 
+#include <endian.h>
+
 #include "twinint.h"
 
 /* op, src, dst */
