--- origsrc/libtwin-0.0.3/Makefile.am	2007-06-27 15:57:22.000000000 +0900
+++ src/libtwin-0.0.3/Makefile.am	2012-04-13 17:35:00.624991300 +0900
@@ -6,7 +6,7 @@ bin_PROGRAMS =
 lib_LTLIBRARIES = libtwin/libtwin.la
 pkginclude_HEADERS = libtwin/twin.h
 
-libtwin_libtwin_la_LDFLAGS = -version-info @LIB_VERSION@
+libtwin_libtwin_la_LDFLAGS = -version-info @LIB_VERSION@ -no-undefined
 libtwin_libtwin_la_SOURCES = \
 	libtwin/twin_box.c \
 	libtwin/twin_button.c \
@@ -40,6 +40,7 @@ libtwin_libtwin_la_SOURCES = \
 	libtwin/twin_work.c \
 	libtwin/twinint.h \
 	twin_def.h
+libtwin_libtwin_la_LIBADD = @TWIN_DEP_LDFLAGS@
 
 if TWIN_X11
 pkginclude_HEADERS += libtwin/twin_x11.h
--- origsrc/libtwin-0.0.3/configure.ac	2007-12-20 23:55:27.000000000 +0900
+++ src/libtwin-0.0.3/configure.ac	2012-04-13 17:35:27.921866300 +0900
@@ -113,7 +113,7 @@ AC_ARG_ENABLE(png,
 
 if test "x$twin_png" = "xyes"
 then
-	PKG_CHECK_MODULES(PNG, libpng12, twin_png="yes", twin_png="no")
+	PKG_CHECK_MODULES(PNG, libpng, twin_png="yes", twin_png="no")
 fi
 
 AC_SUBST(PNG_CFLAGS)
--- origsrc/libtwin-0.0.3/libtwin/twin_png.c	2007-04-03 13:36:31.000000000 +0900
+++ src/libtwin-0.0.3/libtwin/twin_png.c	2012-04-13 17:35:00.624991300 +0900
@@ -29,6 +29,13 @@
 #include <fcntl.h>
 #include <png.h>
 
+#if !defined(int_p_NULL)
+#define int_p_NULL (int *)NULL
+#endif
+#if !defined(png_infopp_NULL)
+#define png_infopp_NULL (png_infopp)NULL
+#endif
+
 #include "twin_png.h"
 
 #if 0
@@ -114,7 +121,12 @@ twin_pixmap_t *twin_png_to_pixmap(const
 	if (ctype == PNG_COLOR_TYPE_PALETTE)
 		png_set_palette_to_rgb(png);
 	if (ctype == PNG_COLOR_TYPE_GRAY && depth < 8)
+#if PNG_LIBPNG_VER < 10400
 		png_set_gray_1_2_4_to_8(png);
+#else
+		png_set_expand_gray_1_2_4_to_8(png);
+#endif
+
 	if (png_get_valid(png, info, PNG_INFO_tRNS))
 		png_set_tRNS_to_alpha(png);
 
