--- origsrc/libg19-1.1.1/Makefile.am	2010-04-11 12:25:09.000000000 +0900
+++ src/libg19-1.1.1/Makefile.am	2013-10-04 07:50:55.056165700 +0900
@@ -1,8 +1,8 @@
 LIBS = $(DEPS_LIBS) $(libg19_LIBS)
-INCLUDES = $(DEPS_CFLAGS) $(libg19_CFLAGS)
+AM_CPPFLAGS = $(DEPS_CFLAGS) $(libg19_CFLAGS)
 
 LLIBDIR = @LIBDIR@ 
 lib_LTLIBRARIES = libg19.la
 libg19_la_SOURCES = libg19.h libg19.c
-libg19_la_LDFLAGS = -version-info 1:1:0
+libg19_la_LDFLAGS = -no-undefined -version-info 1:1:0
 include_HEADERS = libg19.h
