--- origsrc/libtifiles2-1.1.6/configure.ac	2013-03-29 21:49:37.000000000 +0900
+++ src/libtifiles2-1.1.6/configure.ac	2013-09-29 19:20:48.577269100 +0900
@@ -102,7 +102,7 @@ CFLAGS="$CFLAGS $ARCH"
 # What flag to depends on gcc version: gcc3 uses "-mms-bitfields", while
 # gcc2 uses "-fnative-struct".
 case $host_os in
-*mingw*|*cygwin*)
+*mingw*)
   if test x$GCC = xyes; then
     msnative_struct=''
     AC_MSG_CHECKING([how to get MSVC-compatible struct packing])
--- origsrc/libtifiles2-1.1.6/src/Makefile.am	2013-03-29 21:49:37.000000000 +0900
+++ src/libtifiles2-1.1.6/src/Makefile.am	2013-09-29 19:25:28.032347300 +0900
@@ -7,7 +7,7 @@ lib_LTLIBRARIES = libtifiles2.la
 libtifilesincludedir = $(includedir)/tilp2
 libtifilesinclude_HEADERS = \
 	tifiles.h export2.h stdints2.h \
-	files8x.h files9x.h types*.h 
+	files8x.h files9x.h $(srcdir)/types*.h 
 
 # build instructions
 libtifiles2_la_CPPFLAGS = -I$(top_srcdir)/intl \
