diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libcapn-1.1.0/CMakeLists.txt src/libcapn-1.1.0/CMakeLists.txt
--- origsrc/libcapn-1.1.0/CMakeLists.txt	2015-11-01 12:31:16.000000000 +0900
+++ src/libcapn-1.1.0/CMakeLists.txt	2016-01-10 09:09:41.052245500 +0900
@@ -69,20 +69,6 @@ ENDIF(CMAKE_SIZEOF_VOID_P EQUAL 8)
 
 INCLUDE(ExternalProject)
 
-ExternalProject_Add(
-    jansson
-    SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/jansson"
-    #GIT_REPOSITORY git://github.com/akheron/jansson.git
-    BINARY_DIR "${CMAKE_CURRENT_SOURCE_DIR}/jansson"
-    INSTALL_DIR "${CMAKE_CURRENT_SOURCE_DIR}/jansson"
-    CMAKE_COMMAND cmake .
-    BUILD_COMMAND ""
-    INSTALL_COMMAND ""
-)
-
-INCLUDE_DIRECTORIES ("${CMAKE_CURRENT_SOURCE_DIR}/jansson")
-INCLUDE_DIRECTORIES ("${CMAKE_CURRENT_SOURCE_DIR}/jansson/include")
-
 CONFIGURE_FILE("${CMAKE_CURRENT_SOURCE_DIR}/src/platform.h.cmake" "${PROJECT_BINARY_DIR}/src/platform.h")
 CONFIGURE_FILE("${CMAKE_CURRENT_SOURCE_DIR}/src/version.h.cmake" "${PROJECT_BINARY_DIR}/src/version.h")
 
@@ -105,12 +91,12 @@ IF(UNIX)
 	SET(CAPN_INSTALL_PATH_INCLUDES "${CMAKE_INSTALL_PREFIX}/include/${CAPN_LIB_NAME}"  CACHE PATH "")
     ENDIF()    
 
-    IF(NOT DEFINED CAPN_INSTALL_PATH_LIB)
+    IF(NOT DEFINED CAPN_INSTALL_PATH_BIN)
 	SET(CAPN_INSTALL_PATH_BIN "${CMAKE_INSTALL_PREFIX}/bin"  CACHE PATH "")
     ENDIF()
 
     IF(NOT DEFINED CAPN_INSTALL_PATH_PKGCONFIG)
-	SET(CAPN_INSTALL_PATH_PKGCONFIG "${CMAKE_INSTALL_PREFIX}/share/pkgconfig"  CACHE PATH "")
+	SET(CAPN_INSTALL_PATH_PKGCONFIG "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig"  CACHE PATH "")
     ENDIF()
 
     IF(NOT DEFINED CAPN_INSTALL_PATH_SYSCONFIG)
@@ -146,8 +132,6 @@ IF(UNIX)
 
 ENDIF(UNIX)
 
-FILE (GLOB JANSSON_SOURCE_FILES jansson/src/*.c)
-
 SET(CAPN_SOURCE_FILES
     src/apn.c
     src/apn_strings.c
@@ -160,7 +144,7 @@ SET(CAPN_PUBLIC_HEADER_FILES
     ${PROJECT_BINARY_DIR}/src/version.h 
 )
 
-IF(DEFINED WIN32)
+IF(WIN32)
     SET(CAPN_INSTALL_DIR ${PROJECT_BINARY_DIR}/capn-bin)
     SET(CAPN_INSTALL_PATH_LIB ${CAPN_INSTALL_DIR}/lib)
     SET(CAPN_INSTALL_PATH_INCLUDES ${CAPN_INSTALL_DIR}/include)
@@ -199,7 +183,7 @@ IF(DEFINED WIN32)
           ${CMAKE_CURRENT_SOURCE_DIR}/LICENSE
           DESTINATION ${CAPN_INSTALL_DIR})
     	ADD_CUSTOM_TARGET(zip COMMAND ${PROJECT_BINARY_DIR}/win/createpkg.bat)
-ENDIF(DEFINED WIN32)
+ENDIF(WIN32)
 
 IF(BUILD_SHARED_LIBS)
     ADD_LIBRARY(${CAPN_LIB_NAME} SHARED ${CAPN_SOURCE_FILES})
@@ -209,15 +193,15 @@ ENDIF(BUILD_SHARED_LIBS)
 
 TARGET_LINK_LIBRARIES(${CAPN_LIB_NAME} ${OPENSSL_LIBRARIES})
 
-IF(DEFINED WIN32)
+IF(WIN32)
 TARGET_LINK_LIBRARIES(${CAPN_LIB_NAME} Ws2_32.lib)
-ENDIF(DEFINED WIN32)
+ENDIF(WIN32)
 
 SET_TARGET_PROPERTIES(${CAPN_LIB_NAME} PROPERTIES
     VERSION ${CAPN_VERSION} SOVERSION ${CAPN_VERSION_MAJOR}
     CLEAN_DIRECT_OUTPUT 1	# for cmake 2.6.x and lower
 )
-ADD_DEPENDENCIES(${CAPN_LIB_NAME} jansson)
+TARGET_LINK_LIBRARIES(${CAPN_LIB_NAME} jansson)
 
 INSTALL(TARGETS ${CAPN_LIB_NAME}
          RUNTIME DESTINATION ${CAPN_INSTALL_PATH_BIN}
@@ -232,4 +216,3 @@ MESSAGE(STATUS "Cmake version: ${CMAKE_V
 MESSAGE(STATUS "System: ${CMAKE_SYSTEM}")
 MESSAGE(STATUS "Processor:  ${CMAKE_HOST_SYSTEM_PROCESSOR}")
 MESSAGE(STATUS "Using CFLAGS: ${CMAKE_C_FLAGS}")
-
