--- origsrc/lexbor-2.6.0/CMakeLists.txt	2025-11-11 03:37:57.000000000 +0900
+++ src/lexbor-2.6.0/CMakeLists.txt	2026-01-15 13:17:44.722866800 +0900
@@ -377,3 +377,10 @@ endif()
 set(CPACK_PACKAGE_FILE_NAME "${CPACK_PACKAGE_NAME}-${CPACK_PACKAGE_VERSION}-${CPACK_PACKAGE_RELEASE}.${CPACK_RPM_PACKAGE_ARCHITECTURE}")
 
 include(CPack)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/lexbor.pc.in ${CMAKE_BINARY_DIR}/lexbor.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/lexbor.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/lexbor-2.6.0/lexbor.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/lexbor-2.6.0/lexbor.pc.in	2026-01-15 13:17:44.724889500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: lexbor
+Description: HTML Renderer library
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -llexbor
+Cflags: -I${includedir}
