--- origsrc/lexbor-2.3.0/CMakeLists.txt	2023-08-18 01:27:34.000000000 +0900
+++ src/lexbor-2.3.0/CMakeLists.txt	2024-10-22 15:27:25.926287200 +0900
@@ -342,3 +342,10 @@ endif()
 set(CPACK_PACKAGE_FILE_NAME "${CPACK_PACKAGE_NAME}-${CPACK_PACKAGE_VERSION}-${CPACK_PACKAGE_RELEASE}.${CPACK_RPM_PACKAGE_ARCHITECTURE}")
 
 include(CPack)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/lexbor.pc.in ${CMAKE_BINARY_DIR}/lexbor.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/lexbor.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/lexbor-2.3.0/lexbor.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/lexbor-2.3.0/lexbor.pc.in	2024-10-22 15:28:02.346693300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: lexbor
+Description: HTML Renderer library
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -llexbor
+Cflags: -I${includedir}
