--- origsrc/velvet-1.2.10/Makefile	2025-12-26 15:23:12.955979900 +0900
+++ src/velvet-1.2.10/Makefile	2025-12-26 15:25:06.661478700 +0900
@@ -1,5 +1,5 @@
 CC = gcc
-CFLAGS += -Wall
+#CFLAGS += -Wall
 DEBUG = -g
 LIBS = -lm
 OPT = -O3
@@ -9,11 +9,13 @@ DEF = -D MAXKMERLENGTH=$(MAXKMERLENGTH)
 PDFLATEX_VERSION := $(shell pdflatex --version 2> /dev/null)
 
 # Mac OS users: uncomment the following lines
-CFLAGS = -Wall -m64
+# CFLAGS = -Wall -m64
 
 # Sparc/Solaris users: uncomment the following line
 # CFLAGS = -Wall -m64
 
+CFLAGS ?=
+
 ifdef BIGASSEMBLY
 override DEF := $(DEF) -D BIGASSEMBLY
 endif
