--- origsrc/compile-time-regular-expressions-3.11.0/CMakeLists.txt	2026-05-08 17:47:27.000000000 +0900
+++ src/compile-time-regular-expressions-3.11.0/CMakeLists.txt	2026-07-06 21:51:06.570336700 +0900
@@ -19,7 +19,7 @@ endif()
 # When updating to a newer version of CMake, see if we can use the following
 project(ctre
   HOMEPAGE_URL "https://compile-time.re"
-  VERSION 3.9.0
+  VERSION 3.11.0
   LANGUAGES CXX)
 set(PROJECT_DESCRIPTION "Fast compile-time regular expressions with support for matching/searching/capturing during compile-time or runtime.")
 
