--- origsrc/libSavitar-4.6.1/CMakeLists.txt	2020-04-16 21:54:12.000000000 +0900
+++ src/libSavitar-4.6.1/CMakeLists.txt	2020-05-29 11:28:42.267841900 +0900
@@ -15,7 +15,10 @@ if(BUILD_TESTS)
     find_package(Threads QUIET)
 endif()
 
-add_subdirectory(pugixml)
+find_package(PUGIXML)
+if(NOT PUGIXML_FOUND)
+    add_subdirectory(pugixml)
+endif()
 
 if(BUILD_PYTHON)
     list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
