--- origsrc/libSavitar-5.0.0/CMakeLists.txt	2022-04-14 21:59:23.000000000 +0900
+++ src/libSavitar-5.0.0/CMakeLists.txt	2022-07-15 14:07:25.539466800 +0900
@@ -13,7 +13,10 @@ if(BUILD_TESTS)
     find_package(GTest REQUIRED)
 endif()
 
-add_subdirectory(pugixml)
+find_package(PUGIXML)
+if(NOT PUGIXML_FOUND)
+    add_subdirectory(pugixml)
+endif()
 
 set(savitar_SRCS
     src/Namespace.cpp
@@ -184,3 +187,10 @@ install(FILES
     ${CMAKE_BINARY_DIR}/SavitarConfigVersion.cmake
     DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/Savitar
 )
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libSavitar.pc.in ${CMAKE_BINARY_DIR}/libSavitar.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libSavitar.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libSavitar-5.0.0/libSavitar.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libSavitar-5.0.0/libSavitar.pc.in	2022-07-15 14:08:53.184911300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libSavitar
+Description: C++ implementation of 3mf loading with SIP python bindings
+Version: @SAVITAR_VERSION@
+Libs: -L${libdir} -lSavitar
+Cflags: -I${includedir}
