--- origsrc/sslscan-1.8.2/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/sslscan-1.8.2/Makefile.am	2010-12-19 15:31:46.643706400 +0900
@@ -0,0 +1,7 @@
+bin_PROGRAMS = sslscan
+
+sslscan_SOURCES = sslscan.c
+sslscan_CFLAGS = @OPENSSL_CFLAGS@
+sslscan_LDADD = @OPENSSL_LIBS@
+
+man_MANS = sslscan.1
--- origsrc/sslscan-1.8.2/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/sslscan-1.8.2/configure.ac	2010-12-19 15:56:59.856612900 +0900
@@ -0,0 +1,29 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.67])
+AC_INIT([sslscan], [1.8.2], [enquiries@titania.co.uk])
+AC_CONFIG_SRCDIR([sslscan.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE([foreign])
+
+# Checks for programs.
+AC_PROG_CC
+AM_PROG_CC_C_O
+LT_INIT(win32-dll)
+
+# Checks for libraries.
+PKG_CHECK_MODULES([OPENSSL], [openssl])
+
+# Checks for header files.
+AC_CHECK_HEADERS([netdb.h string.h sys/socket.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_HEADER_STDBOOL
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([gethostbyname memset socket])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
