--- origsrc/sslscan-2.0.0/Makefile	2020-07-23 03:00:42.000000000 +0900
+++ src/sslscan-2.0.0/Makefile	2020-09-19 01:28:49.139589500 +0900
@@ -49,9 +49,11 @@ CFLAGS  += -D_FORTIFY_SOURCE=2 -fstack-p
 # Don't enable some hardening flags on OS X because it uses an old version of Clang
 ifneq ($(OS), Darwin)
 ifneq ($(OS), SunOS)
+ifneq ($(findstring CYGWIN,$(OS)),CYGWIN)
 	LDFLAGS += -pie -z relro -z now
 endif
 endif
+endif
 
 # Force C11 mode to fix the build on very old version of GCC
 CFLAGS += -std=gnu11
