--- origsrc/sslscan-2.0.15/Makefile	2022-07-04 00:42:37.000000000 +0900
+++ src/sslscan-2.0.15/Makefile	2022-08-01 11:18:19.409760900 +0900
@@ -50,9 +50,11 @@ CFLAGS  += -D_FORTIFY_SOURCE=2 -fstack-p
 # Don't enable some hardening flags on OS X because it uses an old version of Clang
 ifneq ($(OS), Darwin)
 ifneq ($(OS), SunOS)
+ifneq ($(findstring CYGWIN,$(OS)),CYGWIN)
 	LDFLAGS += -pie -z relro -z now
 endif
 endif
+endif
 
 # Force C11 mode to fix the build on very old version of GCC
 CFLAGS += -std=gnu11
