--- origsrc/ecm-git-7.0.7/Makefile.am	2026-05-19 08:07:10.000000000 +0000
+++ src/ecm-git-7.0.7/Makefile.am	2026-08-11 10:03:18.557684900 +0000
@@ -196,3 +196,6 @@ if MAKE_MANPAGE
 ecm.1: $(srcdir)/ecm.xml
 	xsltproc -o ecm.1 $(XSLDIR)/manpages/docbook.xsl $(srcdir)/ecm.xml
 endif
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/ecm-git-7.0.7/configure.ac	2026-05-19 08:07:10.000000000 +0000
+++ src/ecm-git-7.0.7/configure.ac	2026-08-11 10:03:18.559700700 +0000
@@ -727,6 +727,7 @@ endif
 AC_SUBST([COV_FRAG])
 AM_SUBST_NOTMAKE([COV_FRAG])
 
+AC_CONFIG_FILES([ecm.pc])
 AC_OUTPUT
 
 AC_MSG_NOTICE([Configuration:])
--- origsrc/ecm-git-7.0.7/ecm.pc.in	1970-01-01 00:00:00.000000000 +0000
+++ src/ecm-git-7.0.7/ecm.pc.in	2026-08-11 10:03:18.560716300 +0000
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: ecm
+Description: Factor integers using the Elliptic Curve Method
+Version: @VERSION@
+Libs: -L${libdir} -lecm
+Cflags: -I${includedir}
