--- origsrc/xmlwrapp-0.6.2/configure.ac	2009-12-20 20:37:00.000000000 +0900
+++ src/xmlwrapp-0.6.2/configure.ac	2010-11-26 00:32:38.281936900 +0900
@@ -42,7 +42,6 @@ AC_CONFIG_SRCDIR([xmlwrapp.pc.in])
 AC_CONFIG_AUX_DIR([admin])
 
 AM_INIT_AUTOMAKE
-AM_MAINTAINER_MODE
 m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
 
 dnl remember, "build" is where we compile, "host" is where the resulting
@@ -68,7 +67,7 @@ dnl === Program checks ===
 
 AC_PROG_CXX
 AC_PROG_LD
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 
 dnl === Library checks ===
--- origsrc/xmlwrapp-0.6.2/src/Makefile.am	2009-11-26 19:11:48.000000000 +0900
+++ src/xmlwrapp-0.6.2/src/Makefile.am	2010-11-26 00:33:07.443869700 +0900
@@ -9,7 +9,7 @@ endif
 
 libxmlwrapp_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBXML_CFLAGS)
 libxmlwrapp_la_LIBADD = $(LIBXML_LIBS)
-libxmlwrapp_la_LDFLAGS = -version-info 5:1:0
+libxmlwrapp_la_LDFLAGS = -version-info 5:1:0 -no-undefined
 
 libxmlwrapp_la_SOURCES = \
 		libxml/ait_impl.cxx \
@@ -36,7 +36,7 @@ if WITH_XSLT
 
 libxsltwrapp_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBEXSLT_CFLAGS) $(LIBXSLT_CFLAGS)
 libxsltwrapp_la_LIBADD = libxmlwrapp.la $(LIBEXSLT_LIBS) $(LIBXSLT_LIBS)
-libxsltwrapp_la_LDFLAGS = -version-info 3:0:0
+libxsltwrapp_la_LDFLAGS = -version-info 3:0:0 -no-undefined
 
 libxsltwrapp_la_SOURCES = \
 		libxslt/init.cxx \
--- origsrc/xmlwrapp-0.6.2/tests/Makefile.am	2009-11-26 19:11:48.000000000 +0900
+++ src/xmlwrapp-0.6.2/tests/Makefile.am	2010-11-26 01:32:17.158108100 +0900
@@ -3,8 +3,8 @@ TESTS = test
 
 AM_CPPFLAGS = -I$(top_srcdir)/include
 LIBS = $(top_builddir)/src/libxmlwrapp.la \
-	   -lboost_unit_test_framework \
-	   -lboost_iostreams
+	   -lboost_unit_test_framework-mt \
+	   -lboost_iostreams-mt
 
 noinst_PROGRAMS = test
 
