--- origsrc/yara-3.11.0/Makefile.am	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/Makefile.am	2019-10-12 09:29:32.943048700 +0900
@@ -23,31 +23,31 @@ ACLOCAL_AMFLAGS=-I m4
 bin_PROGRAMS = yara yarac
 
 yara_SOURCES = args.c args.h common.h threading.c threading.h yara.c
-yara_LDADD = libyara/.libs/libyara.a #-Llibyara/.libs -lyara
+yara_LDADD = libyara/libyara.la
 
 yarac_SOURCES = args.c args.h common.h yarac.c
-yarac_LDADD = -Llibyara/.libs -lyara
+yarac_LDADD = libyara/libyara.la
 
 test_alignment_SOURCES = tests/test-alignment.c
 test_atoms_SOURCES = tests/test-atoms.c tests/util.c libyara/atoms.c
-test_atoms_LDADD = libyara/.libs/libyara.a
+test_atoms_LDADD = libyara/libyara.la
 test_rules_SOURCES = tests/test-rules.c tests/util.c
-test_rules_LDADD = libyara/.libs/libyara.a
+test_rules_LDADD = libyara/libyara.la
 test_pe_SOURCES = tests/test-pe.c tests/util.c
-test_pe_LDADD = libyara/.libs/libyara.a
+test_pe_LDADD = libyara/libyara.la
 test_elf_SOURCES = tests/test-elf.c tests/util.c
-test_elf_LDADD = libyara/.libs/libyara.a
+test_elf_LDADD = libyara/libyara.la
 test_version_SOURCES = tests/test-version.c
-test_api_LDADD = libyara/.libs/libyara.a
+test_api_LDADD = libyara/libyara.la
 test_api_SOURCES = tests/test-api.c tests/util.c
 test_bitmask_SOURCES = tests/test-bitmask.c
-test_bitmask_LDADD = libyara/.libs/libyara.a
+test_bitmask_LDADD = libyara/libyara.la
 test_math_SOURCES = tests/test-math.c tests/util.c
-test_math_LDADD = libyara/.libs/libyara.a
+test_math_LDADD = libyara/libyara.la
 test_stack_SOURCES = tests/test-stack.c
-test_stack_LDADD = libyara/.libs/libyara.a
+test_stack_LDADD = libyara/libyara.la
 test_re_split_SOURCES = tests/test-re-split.c
-test_re_split_LDADD = libyara/.libs/libyara.a
+test_re_split_LDADD = libyara/libyara.la
 
 TESTS = $(check_PROGRAMS)
 TESTS_ENVIRONMENT = TOP_SRCDIR=$(top_srcdir)
@@ -70,26 +70,26 @@ if POSIX
 if !ADDRESS_SANITIZER
 check_PROGRAMS+=test-exception
 test_exception_SOURCES = tests/test-exception.c tests/util.c
-test_exception_LDADD = libyara/.libs/libyara.a
+test_exception_LDADD = libyara/libyara.la
 endif
 endif
 
 if MACHO_MODULE
 check_PROGRAMS+=test-macho
 test_macho_SOURCES = tests/test-macho.c tests/util.c
-test_macho_LDADD = libyara/.libs/libyara.a
+test_macho_LDADD = libyara/libyara.la
 endif
 
 if DEX_MODULE
 check_PROGRAMS+=test-dex
 test_dex_SOURCES = tests/test-dex.c tests/util.c
-test_dex_LDADD = libyara/.libs/libyara.a
+test_dex_LDADD = libyara/libyara.la
 endif
 
 if DOTNET_MODULE
 check_PROGRAMS+=test-dotnet
 test_dotnet_SOURCES = tests/test-dotnet.c tests/util.c
-test_dotnet_LDADD = libyara/.libs/libyara.a
+test_dotnet_LDADD = libyara/libyara.la
 endif
 
 # man pages
--- origsrc/yara-3.11.0/libyara/Makefile.am	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/libyara/Makefile.am	2019-10-12 09:30:40.361099100 +0900
@@ -187,5 +187,7 @@ if USE_NO_PROC
 libyara_la_SOURCES += proc/none.c
 endif
 
+libyara_la_LDFLAGS += -no-undefined
+
 pkgconfigdir = $(libdir)/pkgconfig
-nodist_pkgconfig_DATA = yara.pc
+pkgconfig_DATA = yara.pc
--- origsrc/yara-3.11.0/libyara/exception.h	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/libyara/exception.h	2019-10-12 09:30:57.198959400 +0900
@@ -32,7 +32,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 
 #include <assert.h>
 
-#if _WIN32 || __CYGWIN__
+#if _WIN32
 
 #include <windows.h>
 
--- origsrc/yara-3.11.0/libyara/filemap.c	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/libyara/filemap.c	2019-10-12 09:31:46.387137600 +0900
@@ -29,7 +29,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 
 #include <fcntl.h>
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 #include <windows.h>
 #else
 #include <sys/stat.h>
@@ -90,7 +90,7 @@ YR_API int yr_filemap_map(
 //       ERROR_COULD_NOT_MAP_FILE
 //
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 
 YR_API int yr_filemap_map_fd(
     YR_FILE_DESCRIPTOR file,
@@ -258,7 +258,7 @@ YR_API int yr_filemap_map_fd(
 //       ERROR_COULD_NOT_MAP_FILE
 //
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 
 YR_API int yr_filemap_map_ex(
     const char* file_path,
--- origsrc/yara-3.11.0/libyara/include/yara/error.h	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/libyara/include/yara/error.h	2019-10-12 09:32:23.244353000 +0900
@@ -32,7 +32,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 
 #include <string.h>
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 #include <windows.h>
 #endif
 
--- origsrc/yara-3.11.0/libyara/include/yara/filemap.h	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/libyara/include/yara/filemap.h	2019-10-12 09:32:41.965800500 +0900
@@ -32,7 +32,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 
 #include <sys/types.h>
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 #include <windows.h>
 #define YR_FILE_DESCRIPTOR    HANDLE
 #else
@@ -50,7 +50,7 @@ typedef struct _YR_MAPPED_FILE
   YR_FILE_DESCRIPTOR  file;
   size_t              size;
   const uint8_t*      data;
-  #if defined(_WIN32) || defined(__CYGWIN__)
+  #if defined(_WIN32)
   HANDLE              mapping;
   #endif
 
--- origsrc/yara-3.11.0/libyara/include/yara/limits.h	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/libyara/include/yara/limits.h	2019-10-12 09:32:56.918903600 +0900
@@ -30,7 +30,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 #ifndef YR_LIMITS_H
 #define YR_LIMITS_H
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 #include <windows.h>
 #endif
 
--- origsrc/yara-3.11.0/libyara/include/yara/pe.h	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/libyara/include/yara/pe.h	2019-10-12 09:33:16.643845900 +0900
@@ -35,7 +35,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 
 #pragma pack(push, 1)
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 #include <windows.h>
 
 // These definitions are not present in older Windows headers.
--- origsrc/yara-3.11.0/libyara/include/yara/strutils.h	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/libyara/include/yara/strutils.h	2019-10-12 09:33:30.916954200 +0900
@@ -37,7 +37,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 #include <yara/integers.h>
 
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 
 #if !defined(PRIu64)
 #define PRIu64 "I64u"
--- origsrc/yara-3.11.0/libyara/include/yara/threading.h	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/libyara/include/yara/threading.h	2019-10-12 09:33:44.222401800 +0900
@@ -32,7 +32,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 
 
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 
 #include <windows.h>
 
--- origsrc/yara-3.11.0/libyara/lexer.l	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/libyara/lexer.l	2019-10-12 09:34:31.735540600 +0900
@@ -51,7 +51,7 @@ with noyywrap then we can remove this pr
 #include <string.h>
 #include <setjmp.h>
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 #include <windows.h>
 #else
 #include <sys/stat.h>
@@ -127,7 +127,7 @@ static bool is_absolute_path(
   if (path == NULL)
     return false;
 
-  #if defined(_WIN32) || defined(__CYGWIN__)
+  #if defined(_WIN32)
   return strlen(path) > 2 &&
       path[1] == ':' && (path[2] == '/' || path[2] == '\\');
   #else
@@ -905,7 +905,7 @@ int yr_lex_parse_rules_fd(
   size_t file_size;
   void* buffer;
 
-  #if defined(_WIN32) || defined(__CYGWIN__)
+  #if defined(_WIN32)
   DWORD bytes_read;
   #endif
 
@@ -914,7 +914,7 @@ int yr_lex_parse_rules_fd(
   if (setjmp(compiler->error_recovery) != 0)
     return compiler->errors;
 
-  #if defined(_WIN32) || defined(__CYGWIN__)
+  #if defined(_WIN32)
   file_size = (size_t) GetFileSize(rules_fd, NULL);
   #else
   struct stat fs;
@@ -936,7 +936,7 @@ int yr_lex_parse_rules_fd(
     return compiler->errors;
   }
 
-  #if defined(_WIN32) || defined(__CYGWIN__)
+  #if defined(_WIN32)
   if (!ReadFile(rules_fd, buffer, file_size, &bytes_read, NULL))
   #else
   if (read(rules_fd, buffer, file_size) != file_size)
--- origsrc/yara-3.11.0/libyara/libyara.c	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/libyara/libyara.c	2019-10-12 09:34:51.136284700 +0900
@@ -44,7 +44,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 
 #include "crypto.h"
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 #if !defined(_MSC_VER) || (defined(_MSC_VER) && (_MSC_VER < 1900))
 #define snprintf _snprintf
 #endif
--- origsrc/yara-3.11.0/libyara/mem.c	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/libyara/mem.c	2019-10-12 09:35:10.041699300 +0900
@@ -30,7 +30,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 #include <yara/mem.h>
 #include <yara/error.h>
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 
 #include <windows.h>
 #include <string.h>
--- origsrc/yara-3.11.0/libyara/modules/cuckoo.c	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/libyara/modules/cuckoo.c	2019-10-12 09:35:25.240256300 +0900
@@ -34,7 +34,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 #include <yara/re.h>
 #include <yara/modules.h>
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 #define strcasecmp _stricmp
 #endif
 
--- origsrc/yara-3.11.0/libyara/threading.c	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/libyara/threading.c	2019-10-12 09:35:38.834748900 +0900
@@ -30,7 +30,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 #include <yara/error.h>
 #include <yara/threading.h>
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 
 YR_THREAD_ID yr_current_thread_id(void)
 {
--- origsrc/yara-3.11.0/tests/test-api.c	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/tests/test-api.c	2019-10-12 09:36:17.684129900 +0900
@@ -3,7 +3,7 @@
 #include <unistd.h>
 #include "util.h"
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 #include <fileapi.h>
 #else
 #include <unistd.h>
@@ -105,7 +105,7 @@ void test_file_descriptor()
   YR_COMPILER* compiler = NULL;
   YR_RULES* rules = NULL;
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
   HANDLE fd = CreateFile("tests/data/baz.yar", GENERIC_READ, 0, NULL, OPEN_EXISTING, 0, NULL);
   if (fd == INVALID_HANDLE_VALUE)
   {
@@ -135,7 +135,7 @@ void test_file_descriptor()
     exit(EXIT_FAILURE);
   }
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
   CloseHandle(fd);
 #else
   close(fd);
--- origsrc/yara-3.11.0/threading.c	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/threading.c	2019-10-12 09:37:24.284487900 +0900
@@ -29,7 +29,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 
 #include <fcntl.h>
 
-#if !defined(_WIN32) && !defined(__CYGWIN__)
+#if !defined(_WIN32)
 #include <errno.h>
 #include <stdio.h>
 #include <unistd.h>
@@ -45,7 +45,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 int mutex_init(
     MUTEX* mutex)
 {
-  #if defined(_WIN32) || defined(__CYGWIN__)
+  #if defined(_WIN32)
     InitializeCriticalSection(mutex);
     return 0;
   #else
@@ -56,7 +56,7 @@ int mutex_init(
 void mutex_destroy(
     MUTEX* mutex)
 {
-  #if defined(_WIN32) || defined(__CYGWIN__)
+  #if defined(_WIN32)
   DeleteCriticalSection(mutex);
   #else
   pthread_mutex_destroy(mutex);
@@ -67,7 +67,7 @@ void mutex_destroy(
 void mutex_lock(
     MUTEX* mutex)
 {
-  #if defined(_WIN32) || defined(__CYGWIN__)
+  #if defined(_WIN32)
   EnterCriticalSection(mutex);
   #else
   pthread_mutex_lock(mutex);
@@ -78,7 +78,7 @@ void mutex_lock(
 void mutex_unlock(
     MUTEX* mutex)
 {
-  #if defined(_WIN32) || defined(__CYGWIN__)
+  #if defined(_WIN32)
   LeaveCriticalSection(mutex);
   #else
   pthread_mutex_unlock(mutex);
@@ -90,7 +90,7 @@ int semaphore_init(
     SEMAPHORE* semaphore,
     int value)
 {
-  #if defined(_WIN32) || defined(__CYGWIN__)
+  #if defined(_WIN32)
   *semaphore = CreateSemaphore(NULL, value, 65535, NULL);
   if (*semaphore == NULL)
     return GetLastError();
@@ -127,7 +127,7 @@ int semaphore_init(
 void semaphore_destroy(
     SEMAPHORE* semaphore)
 {
-  #if defined(_WIN32) || defined(__CYGWIN__)
+  #if defined(_WIN32)
   CloseHandle(*semaphore);
   #elif defined(__FreeBSD__)
   sem_close(*semaphore);
@@ -141,7 +141,7 @@ void semaphore_destroy(
 void semaphore_wait(
     SEMAPHORE* semaphore)
 {
-  #if defined(_WIN32) || defined(__CYGWIN__)
+  #if defined(_WIN32)
   WaitForSingleObject(*semaphore, INFINITE);
   #else
   sem_wait(*semaphore);
@@ -152,7 +152,7 @@ void semaphore_wait(
 void semaphore_release(
     SEMAPHORE* semaphore)
 {
-  #if defined(_WIN32) || defined(__CYGWIN__)
+  #if defined(_WIN32)
   ReleaseSemaphore(*semaphore, 1, NULL);
   #else
   sem_post(*semaphore);
@@ -165,7 +165,7 @@ int create_thread(
     THREAD_START_ROUTINE start_routine,
     void* param)
 {
-  #if defined(_WIN32) || defined(__CYGWIN__)
+  #if defined(_WIN32)
   *thread = CreateThread(NULL, 0, start_routine, param, 0, NULL);
   if (*thread == NULL)
     return GetLastError();
@@ -180,7 +180,7 @@ int create_thread(
 void thread_join(
     THREAD* thread)
 {
-  #if defined(_WIN32) || defined(__CYGWIN__)
+  #if defined(_WIN32)
   WaitForSingleObject(*thread, INFINITE);
   #else
   pthread_join(*thread, NULL);
--- origsrc/yara-3.11.0/threading.h	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/threading.h	2019-10-12 09:37:41.596725500 +0900
@@ -31,7 +31,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 #ifndef THREADING_H
 #define THREADING_H
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 #include <windows.h>
 #else
 #include <sys/stat.h>
@@ -39,7 +39,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 #include <semaphore.h>
 #endif
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 
 typedef HANDLE SEMAPHORE;
 typedef CRITICAL_SECTION MUTEX;
--- origsrc/yara-3.11.0/yara.c	2019-10-10 20:10:50.000000000 +0900
+++ src/yara-3.11.0/yara.c	2019-10-12 09:38:08.873383200 +0900
@@ -27,7 +27,7 @@ ANY THEORY OF LIABILITY, WHETHER IN CONT
 SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 
-#if !defined(_WIN32) && !defined(__CYGWIN__)
+#if !defined(_WIN32)
 
 #include <sys/stat.h>
 #include <dirent.h>
@@ -335,7 +335,7 @@ static char* file_queue_get()
 }
 
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 
 static bool is_directory(
     const char* path)
@@ -874,7 +874,7 @@ static int callback(
 }
 
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 static DWORD WINAPI scanning_thread(LPVOID param)
 #else
 static void* scanning_thread(void* param)
