--- origsrc/vanitygen-0.21/Makefile	2012-08-22 11:33:26.000000000 +0900
+++ src/vanitygen-0.21/Makefile	2026-06-21 06:47:00.610891700 +0900
@@ -1,5 +1,5 @@
 LIBS=-lpcre -lcrypto -lm -lpthread
-CFLAGS=-ggdb -O3 -Wall
+CFLAGS?=-ggdb -O3 -Wall
 OBJS=vanitygen.o oclvanitygen.o oclvanityminer.o oclengine.o keyconv.o pattern.o util.o
 PROGS=vanitygen keyconv oclvanitygen oclvanityminer
 
@@ -10,6 +10,8 @@ else
 OPENCL_LIBS=-lOpenCL
 endif
 
+CFLAGS+=-I/usr/lib/openssl-1.0/include
+LIBS+=-L/usr/lib/openssl-1.0/lib
 
 most: vanitygen keyconv
 
