--- origsrc/antioch-0.4.0/Makefile.am	2016-04-30 00:42:33.000000000 +0900
+++ src/antioch-0.4.0/Makefile.am	2026-07-24 22:06:29.399712100 +0900
@@ -95,12 +95,12 @@ MAINTAINERCLEANFILES += m4/lt~obsolete.m
 # Since we don't distribute the lic_utils, check and make sure it's there.
 # This way, we won't run this on distributed tarballs, only on the repos clones
 if ANTIOCH_LICENSESTAMPEXISTS
-BUILT_SOURCES += .license.stamp
-.license.stamp: $(top_srcdir)/LICENSE
+BUILT_SOURCES += license.stamp
+license.stamp: $(top_srcdir)/LICENSE
 	@$(top_srcdir)/src/common/lic_utils/update_license.pl $(top_srcdir)/LICENSE $(ANTIOCH_STAMPED_FILES)
 	echo 'updated source license headers' >$@
 
-CLEANFILES = .license.stamp
+CLEANFILES = license.stamp
 endif
 
 # -------------------------------------------
--- origsrc/antioch-0.4.0/src/Makefile.am	2016-04-30 00:42:33.000000000 +0900
+++ src/antioch-0.4.0/src/Makefile.am	2026-07-24 20:33:32.054111800 +0900
@@ -7,6 +7,8 @@ bin_PROGRAMS    = antioch_version
 
 lib_LTLIBRARIES = libantioch.la
 
+libantioch_la_LDFLAGS = -no-undefined
+
 #----------------------------------------------
 # List of source files to build dynamic library
 #----------------------------------------------
