--- origsrc/antioch-0.4.0/src/Makefile.am	2016-04-30 00:42:33.000000000 +0900
+++ src/antioch-0.4.0/src/Makefile.am	2021-03-10 18:36:57.507892100 +0900
@@ -7,6 +7,8 @@ bin_PROGRAMS    = antioch_version
 
 lib_LTLIBRARIES = libantioch.la
 
+libantioch_la_LDFLAGS = -no-undefined
+
 #----------------------------------------------
 # List of source files to build dynamic library
 #----------------------------------------------
