--- origsrc/Zipios-2.2.5.0/CMakeLists.txt	2019-08-17 09:13:23.000000000 +0900
+++ src/Zipios-2.2.5.0/CMakeLists.txt	2019-10-12 13:25:15.227048400 +0900
@@ -39,7 +39,7 @@ endif()
 
 set( ZIPIOS_VERSION_MAJOR 2 )
 set( ZIPIOS_VERSION_MINOR 2 )
-set( ZIPIOS_VERSION_PATCH 0 )
+set( ZIPIOS_VERSION_PATCH 5 )
 set( ZIPIOS_VERSION_BUILD 0 )
 
 set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake" ${CMAKE_MODULE_PATH})
@@ -205,3 +205,10 @@ include(CPack)
 # End:
 
 # vim: ts=4 sw=4 et nocindent
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/zipios.pc.in ${CMAKE_BINARY_DIR}/zipios.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/zipios.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/Zipios-2.2.5.0/cmake/CMakeLists.txt	2019-08-17 09:13:23.000000000 +0900
+++ src/Zipios-2.2.5.0/cmake/CMakeLists.txt	2019-10-12 13:17:57.337054700 +0900
@@ -19,6 +19,6 @@
 #
 project(zipios2_cmake)
 
-install( FILES ZipIosConfig.cmake DESTINATION share/cmake/ZipIos )
+install( FILES ZipIosConfig.cmake DESTINATION lib/cmake/ZipIos )
 
 # vim: ts=4 sw=4 et nocindent
--- origsrc/Zipios-2.2.5.0/zipios.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/Zipios-2.2.5.0/zipios.pc.in	2019-10-12 13:17:57.368310000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: zipios
+Description: C++ library for reading and writing Zip files
+Version: @CPACK_PACKAGE_VERSION@
+Libs: -L${libdir} -lzipios
+Cflags: -I${includedir}
