--- origsrc/zipios-2.1.1/CMakeLists.txt	2015-09-15 09:05:36.000000000 +0900
+++ src/zipios-2.1.1/CMakeLists.txt	2016-04-22 10:05:47.173964700 +0900
@@ -113,7 +113,7 @@ if(NOT CMAKE_VERSION VERSION_GREATER "2.
 else()
     include(GNUInstallDirs)
 endif()
-set(CMAKE_INSTALL_CMAKEMODULESDIR ${CMAKE_INSTALL_DATADIR}/cmake-2.8/Modules)
+set(CMAKE_INSTALL_CMAKEMODULESDIR ${CMAKE_INSTALL_LIBDIR}/cmake/zipios)
 
 set(BIN_INSTALL_DIR             ${CMAKE_INSTALL_BINDIR}             CACHE PATH "Location to install binaries relative to the install prefix.")
 set(INCLUDE_INSTALL_DIR         ${CMAKE_INSTALL_INCLUDEDIR}         CACHE PATH "Location to install headers relative to the install prefix.")
@@ -196,3 +196,10 @@ include(CPack)
 # End:
 
 # vim: ts=4 sw=4 et nocindent
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/zipios.pc.in ${CMAKE_BINARY_DIR}/zipios.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/zipios.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/zipios-2.1.1/zipios.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/zipios-2.1.1/zipios.pc.in	2016-04-22 10:06:09.364004400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: zipios
+Description: C++ library for reading and writing Zip files
+Version: @CPACK_PACKAGE_VERSION@
+Libs: -L${libdir} -lzipios
+Cflags: -I${includedir}
