--- origsrc/asm6809-2.17/tests/Makefile.am	2024-11-29 20:39:27.000000000 +0900
+++ src/asm6809-2.17/tests/Makefile.am	2025-06-30 05:04:38.086961100 +0900
@@ -22,6 +22,6 @@ EXTRA_DIST = \
 	pseudo-phase.s pseudo-phase.cmp \
 	pseudo-section.s pseudo-section.cmp
 
-AM_TESTS_ENVIRONMENT =
+AM_TESTS_ENVIRONMENT = srcdir=$(srcdir)
 
 TESTS = test-isa6809.sh test-isa6309.sh test-pseudo.sh
--- origsrc/asm6809-2.17/tests/test-isa6309.sh	2023-07-01 00:20:51.000000000 +0900
+++ src/asm6809-2.17/tests/test-isa6309.sh	2025-06-30 05:04:38.086961100 +0900
@@ -4,8 +4,8 @@ fail=0
 tests="isa6309-immediate isa6309-direct isa6309-indexed isa6309-extended isa6309-inherent"
 
 for t in ${tests}; do
-	../src/asm6809${EXEEXT} -3 -S -l ${t}.lis -o ${t}.out ${t}.s
-	cmp ${t}.out ${t}.cmp || fail=1
+	../src/asm6809${EXEEXT} -3 -S -l ${t}.lis -o ${t}.out ${srcdir}/${t}.s
+	cmp ${t}.out ${srcdir}/${t}.cmp || fail=1
 done
 
 exit $fail
--- origsrc/asm6809-2.17/tests/test-isa6809.sh	2023-07-01 00:20:51.000000000 +0900
+++ src/asm6809-2.17/tests/test-isa6809.sh	2025-06-30 05:04:38.086961100 +0900
@@ -4,8 +4,8 @@ fail=0
 tests="isa6809-direct isa6809-extended isa6809-immediate isa6809-indexed isa6809-inherent isa6809-relative"
 
 for t in ${tests}; do
-	../src/asm6809${EXEEXT} -S -l ${t}.lis -o ${t}.out ${t}.s
-	cmp ${t}.out ${t}.cmp || fail=1
+	../src/asm6809${EXEEXT} -S -l ${t}.lis -o ${t}.out ${srcdir}/${t}.s
+	cmp ${t}.out ${srcdir}/${t}.cmp || fail=1
 done
 
 exit $fail
--- origsrc/asm6809-2.17/tests/test-pseudo.sh	2024-11-29 20:39:27.000000000 +0900
+++ src/asm6809-2.17/tests/test-pseudo.sh	2025-06-30 05:04:38.086961100 +0900
@@ -4,8 +4,8 @@ fail=0
 tests="pseudo-cond pseudo-org-put-setdp pseudo-phase pseudo-section"
 
 for t in ${tests}; do
-	../src/asm6809${EXEEXT} -S -l ${t}.lis -o ${t}.out ${t}.s
-	cmp ${t}.out ${t}.cmp || fail=1
+	../src/asm6809${EXEEXT} -S -l ${t}.lis -o ${t}.out ${srcdir}/${t}.s
+	cmp ${t}.out ${srcdir}/${t}.cmp || fail=1
 done
 
 exit $fail
