--- origsrc/asm6809-2.1/tests/Makefile.am	2014-05-04 16:05:18.000000000 +0900
+++ src/asm6809-2.1/tests/Makefile.am	2014-07-05 11:33:57.951871400 +0900
@@ -28,6 +28,6 @@ EXTRA_DIST = \
 	isa6809-relative.cmp \
 	isa6809-relative.s
 
-AM_TESTS_ENVIRONMENT =
+AM_TESTS_ENVIRONMENT = srcdir=$(srcdir)
 
 TESTS = test-isa6809.sh test-isa6309.sh
--- origsrc/asm6809-2.1/tests/test-isa6309.sh	2014-05-04 16:05:18.000000000 +0900
+++ src/asm6809-2.1/tests/test-isa6309.sh	2014-07-05 11:34:24.936246400 +0900
@@ -4,8 +4,8 @@ fail=0
 tests="isa6309-immediate isa6309-direct isa6309-indexed isa6309-extended isa6309-inherent"
 
 for t in ${tests}; do
-	../src/asm6809${EXEEXT} -3 -B -l ${t}.lis -o ${t}.out ${t}.s
-	cmp ${t}.out ${t}.cmp || fail=1
+	../src/asm6809${EXEEXT} -3 -B -l ${t}.lis -o ${t}.out ${srcdir}/${t}.s
+	cmp ${t}.out ${srcdir}/${t}.cmp || fail=1
 done
 
 exit $fail
--- origsrc/asm6809-2.1/tests/test-isa6809.sh	2014-05-04 16:05:18.000000000 +0900
+++ src/asm6809-2.1/tests/test-isa6809.sh	2014-07-05 11:34:28.186246400 +0900
@@ -4,8 +4,8 @@ fail=0
 tests="isa6809-direct isa6809-extended isa6809-immediate isa6809-indexed isa6809-inherent isa6809-relative"
 
 for t in ${tests}; do
-	../src/asm6809${EXEEXT} -B -l ${t}.lis -o ${t}.out ${t}.s
-	cmp ${t}.out ${t}.cmp || fail=1
+	../src/asm6809${EXEEXT} -B -l ${t}.lis -o ${t}.out ${srcdir}/${t}.s
+	cmp ${t}.out ${srcdir}/${t}.cmp || fail=1
 done
 
 exit $fail
