--- origsrc/kissfft-131.1.0/CMakeLists.txt	2021-02-10 10:29:17.000000000 +0900
+++ src/kissfft-131.1.0/CMakeLists.txt	2025-01-21 16:06:57.461293800 +0900
@@ -105,7 +105,7 @@ endif()
 # Add GNUInstallDirs for GNU infrastructure before target)include_directories
 #
 
-if(CMAKE_SYSTEM_NAME MATCHES "^(Linux|kFreeBSD|GNU)$" AND NOT CMAKE_CROSSCOMPILING)
+if((CMAKE_SYSTEM_NAME MATCHES "^(Linux|kFreeBSD|GNU)$" OR CYGWIN) AND NOT CMAKE_CROSSCOMPILING)
     include(GNUInstallDirs)
 endif()
 
