--- origsrc/kissfft-131.2.0/Makefile	2025-09-12 14:55:11.000000000 +0900
+++ src/kissfft-131.2.0/Makefile	2026-01-14 10:36:43.787683200 +0900
@@ -18,7 +18,7 @@
 #
 
 export KFVER_MAJOR = 131
-export KFVER_MINOR = 1
+export KFVER_MINOR = 2
 export KFVER_PATCH = 0
 
 #
--- origsrc/kissfft-131.2.0/test/testcpp.cc	2025-09-12 14:55:11.000000000 +0900
+++ src/kissfft-131.2.0/test/testcpp.cc	2026-01-14 10:21:53.102616300 +0900
@@ -43,9 +43,13 @@ void dotest(int nfft)
     long double totalpower=0;
     long double difpower=0;
 
+#if defined(M_PIl)
+    const long double pi = M_PIl;
+#else
     // Create long double constant for pi because M_PIl is not defined by
     // all toolchains.
     const long double pi = std::acosl(-1);
+#endif
 
     for (int k0=0;k0<nfft;++k0) {
         complex<long double> acc = 0;
