--- origsrc/libcaes-20150514/tests/caes_test_crypt_cbc.c	2015-01-04 20:30:17.000000000 +0900
+++ src/libcaes-20150514/tests/caes_test_crypt_cbc.c	2016-04-05 08:36:44.024427700 +0900
@@ -19,7 +19,7 @@
  * along with this software.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include <common.h>
+#include <memory.h>
 
 #if defined( HAVE_STDLIB_H ) || defined( WINAPI )
 #include <stdlib.h>
--- origsrc/libcaes-20150514/tests/caes_test_crypt_ccm.c	2015-01-04 20:30:17.000000000 +0900
+++ src/libcaes-20150514/tests/caes_test_crypt_ccm.c	2016-04-05 08:36:44.024427700 +0900
@@ -19,7 +19,7 @@
  * along with this software.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include <common.h>
+#include <memory.h>
 
 #if defined( HAVE_STDLIB_H ) || defined( WINAPI )
 #include <stdlib.h>
--- origsrc/libcaes-20150514/tests/caes_test_crypt_cfb.c	2015-01-04 20:30:17.000000000 +0900
+++ src/libcaes-20150514/tests/caes_test_crypt_cfb.c	2016-04-05 08:36:44.024427700 +0900
@@ -19,7 +19,7 @@
  * along with this software.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include <common.h>
+#include <memory.h>
 
 #if defined( HAVE_STDLIB_H ) || defined( WINAPI )
 #include <stdlib.h>
--- origsrc/libcaes-20150514/tests/caes_test_crypt_ecb.c	2015-04-03 22:40:42.000000000 +0900
+++ src/libcaes-20150514/tests/caes_test_crypt_ecb.c	2016-04-05 08:36:44.038267500 +0900
@@ -19,7 +19,7 @@
  * along with this software.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include <common.h>
+#include <memory.h>
 
 #if defined( HAVE_STDLIB_H ) || defined( WINAPI )
 #include <stdlib.h>
--- origsrc/libcaes-20150514/tests/caes_test_crypt_xts.c	2015-01-04 20:30:17.000000000 +0900
+++ src/libcaes-20150514/tests/caes_test_crypt_xts.c	2016-04-05 08:36:44.052107300 +0900
@@ -19,7 +19,7 @@
  * along with this software.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include <common.h>
+#include <memory.h>
 
 #if defined( HAVE_STDLIB_H ) || defined( WINAPI )
 #include <stdlib.h>
