--- origsrc/libcaes-20190102/Makefile.am	2019-01-03 00:15:33.000000000 +0900
+++ src/libcaes-20190102/Makefile.am	2019-05-16 11:43:29.079256400 +0900
@@ -3,12 +3,10 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
 	libcaes \
 	pycaes \
 	pycaes-python2 \
 	pycaes-python3 \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libcaes-20190102/configure.ac	2019-01-03 00:15:33.000000000 +0900
+++ src/libcaes-20190102/configure.ac	2019-05-16 11:43:20.111568400 +0900
@@ -128,7 +128,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
 AC_CONFIG_FILES([libcaes/Makefile])
 AC_CONFIG_FILES([pycaes/Makefile])
 AC_CONFIG_FILES([pycaes-python2/Makefile])
--- origsrc/libcaes-20190102/libcaes/Makefile.am	2019-01-03 00:15:33.000000000 +0900
+++ src/libcaes-20190102/libcaes/Makefile.am	2019-05-16 11:43:46.357683800 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libcaes-20190102/pycaes/Makefile.am	2019-01-03 00:15:33.000000000 +0900
+++ src/libcaes-20190102/pycaes/Makefile.am	2019-05-16 11:44:04.983584900 +0900
@@ -1,5 +1,6 @@
 if HAVE_PYTHON
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -24,7 +25,7 @@ pycaes_la_LIBADD = \
 	../libcaes/libcaes.la
 
 pycaes_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pycaes_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pycaes_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
--- origsrc/libcaes-20190102/pycaes-python2/Makefile.am	2019-01-03 00:15:33.000000000 +0900
+++ src/libcaes-20190102/pycaes-python2/Makefile.am	2019-05-16 11:44:19.788366900 +0900
@@ -6,6 +6,7 @@ pyexecdir = $(pyexecdir2)
 
 if HAVE_PYTHON2
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -34,7 +35,7 @@ pycaes_la_LIBADD = \
 	../libcaes/libcaes.la
 
 pycaes_la_CPPFLAGS = $(PYTHON2_CPPFLAGS)
-pycaes_la_LDFLAGS  = -module -avoid-version $(PYTHON2_LDFLAGS)
+pycaes_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON2_LDFLAGS)
 
 $(BUILT_SOURCES):
 	/bin/cp -f $(top_srcdir)/pycaes/$@ $@
--- origsrc/libcaes-20190102/pycaes-python3/Makefile.am	2019-01-03 00:15:33.000000000 +0900
+++ src/libcaes-20190102/pycaes-python3/Makefile.am	2019-05-16 11:44:36.203204300 +0900
@@ -6,6 +6,7 @@ pyexecdir = $(pyexecdir3)
 
 if HAVE_PYTHON3
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -34,7 +35,7 @@ pycaes_la_LIBADD = \
 	../libcaes/libcaes.la
 
 pycaes_la_CPPFLAGS = $(PYTHON3_CPPFLAGS)
-pycaes_la_LDFLAGS  = -module -avoid-version $(PYTHON3_LDFLAGS)
+pycaes_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON3_LDFLAGS)
 
 $(BUILT_SOURCES):
 	/bin/cp -f $(top_srcdir)/pycaes/$@ $@
--- origsrc/libcaes-20190102/tests/Makefile.am	2019-01-03 00:15:33.000000000 +0900
+++ src/libcaes-20190102/tests/Makefile.am	2019-05-16 11:44:47.674928800 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
