--- origsrc/libavif-0.4.0/CMakeLists.txt	2019-10-03 06:13:32.000000000 +0900
+++ src/libavif-0.4.0/CMakeLists.txt	2019-10-12 23:35:57.065136800 +0900
@@ -78,6 +78,7 @@ set(AVIF_SRCS
     src/utils.c
     src/write.c
 )
+install(FILES include/avif/avif.h DESTINATION include)
 
 set(AVIF_CODEC_LIBARIES)
 if(AVIF_CODEC_AOM)
@@ -141,10 +142,15 @@ if(NOT AVIF_CODEC_AOM AND NOT AVIF_CODEC
     message(FATAL_ERROR "libavif: No decoding library is enabled, bailing out.")
 endif()
 
-add_library(avif STATIC ${AVIF_SRCS})
+add_library(avif SHARED ${AVIF_SRCS})
 target_link_libraries(avif ${AVIF_CODEC_LIBARIES})
 target_include_directories(avif SYSTEM INTERFACE include)
 
+if(CYGWIN)
+    set_target_properties(avif PROPERTIES SOVERSION 0)
+endif()
+install(TARGETS avif RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+
 set(AVIF_MATH_LIBRARY)
 if(NOT WIN32)
     set(AVIF_MATH_LIBRARY m)
@@ -248,3 +254,10 @@ if(WIN32)
     avif_set_folder_safe(avif "ext/avif")
     avif_set_folder_safe(dist "ext/avif")
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libavif.pc.in ${CMAKE_BINARY_DIR}/libavif.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libavif.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libavif-0.4.0/libavif.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libavif-0.4.0/libavif.pc.in	2019-10-12 23:35:57.080763200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libavif
+Description: Library for encoding and decoding .avif files
+Version: @VERSION@
+Libs: -L${libdir} -lavif
+Cflags: -I${includedir}
--- origsrc/libavif-0.4.0/src/codec_aom.c	2019-10-03 06:13:32.000000000 +0900
+++ src/libavif-0.4.0/src/codec_aom.c	2019-10-12 23:35:57.096385000 +0900
@@ -130,7 +130,9 @@ static avifBool aomCodecGetNextImage(avi
                 break;
             case AOM_IMG_FMT_YV12:
             case AOM_IMG_FMT_AOMYV12:
+#if !defined(__CYGWIN__)
             case AOM_IMG_FMT_YV1216:
+#endif
                 yuvFormat = AVIF_PIXEL_FORMAT_YV12;
                 break;
             case AOM_IMG_FMT_NONE:
@@ -339,7 +341,9 @@ static avifBool encodeOBU(avifImage * im
         aom_codec_control(&aomEncoder, AV1E_SET_LOSSLESS, 1);
     }
     if (encoder->maxThreads > 1) {
+#if !defined(__CYGWIN__)
         aom_codec_control(&aomEncoder, AV1E_SET_ROW_MT, 1);
+#endif
     }
     if (encoder->tileRowsLog2 != 0) {
         int tileRowsLog2 = AVIF_CLAMP(encoder->tileRowsLog2, 0, 6);
