--- origsrc/libdatovka-0.6.2/src/Makefile.am	2024-02-20 23:48:25.000000000 +0900
+++ src/libdatovka-0.6.2/src/Makefile.am	2024-03-17 12:28:23.162354900 +0900
@@ -69,5 +69,5 @@ endif
 libdatovka_la_CPPFLAGS = @XML_CPPFLAGS@ @LIBCURL_CPPFLAGS@ \
 	-DLOCALEDIR=\"$(localedir)\" $(AM_CPPFLAGS)
 libdatovka_la_CFLAGS = @LIBGCRYPT_CFLAGS@ @GPGME_CFLAGS@
-libdatovka_la_LDFLAGS = -version-info 7:2:1 @XML_LIBS@ @LIBCURL@ @LIBGCRYPT_LIBS@ \
+libdatovka_la_LDFLAGS = -no-undefined -version-info 7:2:1 @XML_LIBS@ @LIBCURL@ @LIBGCRYPT_LIBS@ \
 	@GPGME_LIBS@ @EXPAT_LIBS@ @LTLIBINTL@
