--- origsrc/liblcfg-0.2.2/src/Makefile.am	2012-04-05 21:21:14.000000000 +0900
+++ src/liblcfg-0.2.2/src/Makefile.am	2012-06-22 08:45:00.057718200 +0900
@@ -2,7 +2,7 @@
 
 AUTOMAKE_OPTIONS = foreign
 
-AM_CFLAGS = -pipe -D _GNU_SOURCE -I../include -Werror -Wall -g --std=c99
+AM_CFLAGS = -pipe -D _GNU_SOURCE -I../include -I$(top_srcdir)/include -Wall -g --std=c99
 
 lib_LTLIBRARIES = liblcfg.la
 
--- origsrc/liblcfg-0.2.2/tools/Makefile.am	2012-04-05 21:21:14.000000000 +0900
+++ src/liblcfg-0.2.2/tools/Makefile.am	2012-06-22 08:50:55.437601500 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = foreign
 
-AM_CPPFLAGS = -I../include -I ../.. -Werror -Wall -g
+AM_CPPFLAGS = -I../include -I ../.. -I$(top_srcdir)/include -Wall -g
 AM_LDFLAGS = -llcfg -L../src 
 
 bin_PROGRAMS = lcfg
