--- origsrc/libcitygml/CMakeLists.txt	2011-02-26 01:10:17.000000000 +0900
+++ src/libcitygml/CMakeLists.txt	2014-12-04 08:39:38.665593700 +0900
@@ -140,4 +140,4 @@ CONFIGURE_FILE(
   "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
   IMMEDIATE @ONLY)
 ADD_CUSTOM_TARGET(uninstall
-  "${CMAKE_COMMAND}" -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
\ No newline at end of file
+  "${CMAKE_COMMAND}" -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
--- origsrc/libcitygml/CMakeModules/ModuleInstall.cmake	2010-06-02 20:42:13.000000000 +0900
+++ src/libcitygml/CMakeModules/ModuleInstall.cmake	2014-12-04 08:40:10.371648400 +0900
@@ -4,7 +4,7 @@
 # ${LIB_NAME}
 # ${LIB_PUBLIC_HEADERS}
 
-SET(INSTALL_INCDIR include)
+SET(INSTALL_INCDIR include/libcitygml)
 SET(INSTALL_BINDIR bin)
 IF(WIN32)
     SET(INSTALL_LIBDIR bin)
--- origsrc/libcitygml/src/CMakeLists.txt	2012-02-07 18:38:06.000000000 +0900
+++ src/libcitygml/src/CMakeLists.txt	2014-12-04 08:40:35.796453100 +0900
@@ -77,6 +77,7 @@ SET( LIB_PUBLIC_HEADERS
 ADD_LIBRARY( ${LIB_NAME} ${LIBCITYGML_USER_DEFINED_DYNAMIC_OR_STATIC} ${LIB_SRCS} ${LIB_PUBLIC_HEADERS} )
 
 TARGET_LINK_LIBRARIES( ${LIB_NAME} ${XERCESC_LIBRARIES} ${LIBXML2_LIBRARIES} ${OPENGL_LIBRARIES} ${GDAL_LIBRARY} )
+SET_TARGET_PROPERTIES( ${LIB_NAME} PROPERTIES SOVERSION 0)
 
 # IF( MSVC_IDE )
 	# SET_TARGET_PROPERTIES( ${LIB_NAME} PROPERTIES PREFIX "../" )
