--- origsrc/libcitygml-2.5.0/CMakeLists.txt	2023-02-14 19:52:40.000000000 +0900
+++ src/libcitygml-2.5.0/CMakeLists.txt	2023-04-03 13:31:56.713975700 +0900
@@ -3,8 +3,8 @@ CMAKE_MINIMUM_REQUIRED(VERSION 3.0.0 FAT
 PROJECT ( libcitygml )
 
 set(META_VERSION_MAJOR       "2")
-set(META_VERSION_MINOR       "4")
-set(META_VERSION_PATCH       "1")
+set(META_VERSION_MINOR       "5")
+set(META_VERSION_PATCH       "0")
 
 set(META_VERSION             "${META_VERSION_MAJOR}.${META_VERSION_MINOR}.${META_VERSION_PATCH}")
 
