--- origsrc/libcitygml-2.5.1/CMakeLists.txt	2023-05-11 16:51:21.000000000 +0900
+++ src/libcitygml-2.5.1/CMakeLists.txt	2023-05-18 15:26:43.008606600 +0900
@@ -3,7 +3,7 @@ CMAKE_MINIMUM_REQUIRED(VERSION 3.0.0 FAT
 PROJECT ( libcitygml )
 
 set(META_VERSION_MAJOR       "2")
-set(META_VERSION_MINOR       "4")
+set(META_VERSION_MINOR       "5")
 set(META_VERSION_PATCH       "1")
 
 set(META_VERSION             "${META_VERSION_MAJOR}.${META_VERSION_MINOR}.${META_VERSION_PATCH}")
