--- origsrc/libcitygml-2.1.0/CMakeLists.txt	2018-12-02 19:06:40.000000000 +0900
+++ src/libcitygml-2.1.0/CMakeLists.txt	2019-02-15 12:21:37.884978400 +0900
@@ -60,8 +60,10 @@ if(UNIX)
         MESSAGE(STATUS "APPLE")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -fPIC")
     else(APPLE)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -fPIC")
-
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+        if(NOT CYGWIN)
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
+        endif()
         MESSAGE(STATUS "Setting UNIX compile flags: ${CMAKE_CXX_FLAGS}")
     endif(APPLE)
 endif(UNIX)
--- origsrc/libcitygml-2.1.0/sources/src/parser/geocoordinatetransformer.cpp	2018-12-02 19:06:40.000000000 +0900
+++ src/libcitygml-2.1.0/sources/src/parser/geocoordinatetransformer.cpp	2019-02-15 12:21:37.896406800 +0900
@@ -11,7 +11,10 @@
 #include <citygml/envelope.h>
 
 #ifdef USE_GDAL
-#include "ogrsf_frmts.h"
+#if defined(__CYGWIN__)
+#include <xlocale.h>
+#endif
+#include <ogrsf_frmts.h>
 
 #include <stdexcept>
 
