--- origsrc/libcitygml-2.0.8/CMakeLists.txt	2017-12-19 23:49:17.000000000 +0900
+++ src/libcitygml-2.0.8/CMakeLists.txt	2018-02-26 10:45:32.643813900 +0900
@@ -60,8 +60,10 @@ if(UNIX)
         MESSAGE(STATUS "APPLE")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -fPIC")
     else(APPLE)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -fPIC")
-
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+        if(NOT CYGWIN)
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
+        endif()
         MESSAGE(STATUS "Setting UNIX compile flags: ${CMAKE_CXX_FLAGS}")
     endif(APPLE)
 endif(UNIX)
--- origsrc/libcitygml-2.0.8/sources/src/parser/geocoordinatetransformer.cpp	2017-12-19 23:49:17.000000000 +0900
+++ src/libcitygml-2.0.8/sources/src/parser/geocoordinatetransformer.cpp	2018-02-26 10:45:32.643813900 +0900
@@ -11,7 +11,10 @@
 #include <citygml/envelope.h>
 
 #ifdef USE_GDAL
-#include "ogrsf_frmts.h"
+#if defined(__CYGWIN__)
+#include <xlocale.h>
+#endif
+#include <ogrsf_frmts.h>
 
 #include <stdexcept>
 
