--- origsrc/libcitygml-2.4.2/CMakeLists.txt	2022-03-14 17:05:25.000000000 +0900
+++ src/libcitygml-2.4.2/CMakeLists.txt	2022-05-17 15:38:27.084587600 +0900
@@ -4,7 +4,7 @@ PROJECT ( libcitygml )
 
 set(META_VERSION_MAJOR       "2")
 set(META_VERSION_MINOR       "4")
-set(META_VERSION_PATCH       "1")
+set(META_VERSION_PATCH       "2")
 
 set(META_VERSION             "${META_VERSION_MAJOR}.${META_VERSION_MINOR}.${META_VERSION_PATCH}")
 
