--- origsrc/geometry-central/CMakeLists.txt	2025-11-13 09:47:28.000000000 +0900
+++ src/geometry-central/CMakeLists.txt	2025-11-13 10:58:43.607952800 +0900
@@ -63,3 +63,10 @@ install(
   FILES_MATCHING
   PATTERN "*.h"
   PATTERN "*.ipp")
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/geometry-central.pc.in ${CMAKE_BINARY_DIR}/geometry-central.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/geometry-central.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/geometry-central/geometry-central.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/geometry-central/geometry-central.pc.in	2025-11-13 10:58:34.715025200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: geometry-central
+Description: Applied 3D geometry in C++, with a focus on surface meshes
+Version: @VERSION@
+Libs: -L${libdir} -lgeometry-central
+Cflags: -I${includedir}
--- origsrc/geometry-central/src/CMakeLists.txt	2025-11-13 09:47:29.000000000 +0900
+++ src/geometry-central/src/CMakeLists.txt	2025-11-13 09:49:24.738225300 +0900
@@ -199,3 +199,10 @@ endif()
 if(MSVC AND BUILD_SHARED_LIBS)
   set_target_properties(geometry-central PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
 endif()
+
+if (DEFINED SOVERSION)
+  set_target_properties(geometry-central PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
+install(TARGETS geometry-central RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+install(DIRECTORY ${INCLUDE_ROOT} DESTINATION include/geometrycentral)
