--- origsrc/libuna-20150927/Makefile.am	2015-09-28 00:29:55.000000000 +0900
+++ src/libuna-20150927/Makefile.am	2016-04-04 22:48:57.985526800 +0900
@@ -3,16 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
 	libuna \
-	libcdatetime \
-	libclocale \
-	libcnotify \
-	libcfile \
-	libcsystem \
-	unatools \
-	po \
 	documents \
 	manuals \
 	tests \
@@ -92,4 +83,3 @@ splint:
 	(cd $(srcdir)/unatools && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
-
--- origsrc/libuna-20150927/configure.ac	2015-09-28 00:29:55.000000000 +0900
+++ src/libuna-20150927/configure.ac	2016-04-04 22:48:57.985526800 +0900
@@ -154,14 +154,7 @@ dnl Have configure make the Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
 AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcdatetime/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcsystem/Makefile])
 AC_CONFIG_FILES([unatools/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
@@ -202,4 +195,3 @@ Features:
    Wide character type support:              $ac_cv_enable_wide_character_type
    unatools are build as static executables: $ac_cv_enable_static_executables
 ]);
-
--- origsrc/libuna-20150927/libuna/Makefile.am	2015-09-28 00:29:55.000000000 +0900
+++ src/libuna-20150927/libuna/Makefile.am	2016-04-04 22:48:57.985526800 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -81,4 +82,3 @@ distclean: clean
 splint:
 	@echo "Running splint on libuna ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(libuna_la_SOURCES)
-
--- origsrc/libuna-20150927/tests/Makefile.am	2015-09-28 00:29:55.000000000 +0900
+++ src/libuna-20150927/tests/Makefile.am	2016-04-04 22:48:57.985526800 +0900
@@ -1,4 +1,6 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
+	-I$(top_srcdir)/common \
 	-I$(top_srcdir)/include \
 	@LIBCSTRING_CPPFLAGS@ \
 	@LIBCERROR_CPPFLAGS@ \
@@ -264,4 +266,3 @@ MAINTAINERCLEANFILES = \
 
 distclean: clean
 	/bin/rm -f Makefile
-
--- origsrc/libuna-20150927/unatools/Makefile.am	2015-09-28 00:29:55.000000000 +0900
+++ src/libuna-20150927/unatools/Makefile.am	2016-04-04 22:48:57.985526800 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -81,4 +82,3 @@ distclean: clean
 splint:
 	@echo "Running splint on unaexport ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(unaexport_SOURCES)
-
