--- origsrc/leafpak/Makefile	2000-11-22 20:53:10.000000000 +0900
+++ src/leafpak/Makefile	2026-03-30 13:07:02.938675200 +0900
@@ -1,16 +1,16 @@
 DEST=/usr/local/bin
-CFLAGS=-Wall
+CFLAGS?=-Wall
 
 all: leafpak leafpak2
 
 leafpak.o: leafpak.c leafpak.h
-	$(CC) -c -Wall -g leafpak.c
+	$(CC) -c $(CFLAGS) leafpak.c
 
 main.o: main.c leafpak.h
-	$(CC) -c -Wall -g main.c
+	$(CC) -c $(CFLAGS) main.c
 
 leafpak: main.o leafpak.o
-	$(CC) -o leafpak main.o leafpak.o
+	$(CC) $(CFLAGS) -o leafpak main.o leafpak.o
 
 leafpak2: leafpak2.c
 	$(CC) $(CFLAGS) -o leafpak2 leafpak2.c
--- origsrc/leafpak/main.c	2000-02-05 23:47:38.000000000 +0900
+++ src/leafpak/main.c	2026-03-30 13:05:06.789355000 +0900
@@ -13,6 +13,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 
 #define MAIN
