--- origsrc/gamer/src/aaa_lib/Makefile.am	2016-12-29 00:27:31.853925600 +0900
+++ src/gamer/src/aaa_lib/Makefile.am	2016-12-29 11:05:44.834574400 +0900
@@ -39,13 +39,12 @@ lib_LTLIBRARIES = libgamer.la ${LIBTRIAN
 
 libgamer_la_SOURCES = 
 libgamer_la_LIBADD  = ${BASE_LIBS} ${BIOM_LIBS} -lm -lstdc++ ../tetgen/libtetgen.la
-libgamer_la_LDFLAGS = -version-info ${FETK_VERSION}
+libgamer_la_LDFLAGS = -no-undefined -version-info ${FETK_VERSION}
 
 libtriangle_la_SOURCES = 
 libtriangle_la_LIBADD  = ${TRIANGLE_LIBS} -lm
-libtriangle_la_LDFLAGS = -version-info ${FETK_VERSION}
+libtriangle_la_LDFLAGS = -no-undefined -version-info ${FETK_VERSION}
 
 libtetgen_la_SOURCES = 
 libtetgen_la_LIBADD  = ${TETGEN_LIBS} -lm -lstdc++
-libtetgen_la_LDFLAGS = -version-info ${FETK_VERSION}
-
+libtetgen_la_LDFLAGS = -no-undefined -version-info ${FETK_VERSION}
--- origsrc/gamer/swig/Makefile.am	2010-12-10 08:35:47.000000000 +0900
+++ src/gamer/swig/Makefile.am	2016-12-29 00:29:28.011405700 +0900
@@ -27,7 +27,7 @@ pkgpython_PYTHON       = cgamer.py ${top
 pkgpyexec_LTLIBRARIES  = _cgamer.la
 _cgamer_la_SOURCES     = cgamer_wrap.cc ${SWIG_SOURCES}
 _cgamer_la_CPPFLAGS    = ${SWIG_PYTHON_CPPFLAGS} -I. -I${FETK_PYTHON_INCLUDE} -I${FETK_NUMPY_INCLUDE}
-_cgamer_la_LDFLAGS     = -module
+_cgamer_la_LDFLAGS     = -avoid-version -no-undefined -module
 _cgamer_la_LIBADD      = @fetk_lib@/libgamer.la @fetk_lib@/libmaloc.la
 
 cgamer_wrap.cc : ${top_srcdir}/src/gamer.i
@@ -35,4 +35,3 @@ cgamer_wrap.cc : ${top_srcdir}/src/gamer
 
 clean-local :
 	rm -f cgamer.py cgamer_wrap.cc
-
