--- origsrc/voacapl-0.6.7/Makefile.am	2015-04-24 04:42:59.000000000 +0900
+++ src/voacapl-0.6.7/Makefile.am	2017-12-24 00:45:36.632872700 +0900
@@ -17,36 +17,36 @@ dist_doc_DATA = README.md\
 		AUTHORS\
 		logo.png
 
-nobase_dist_doc_DATA = voacapl/itshfbc/news/*
+nobase_dist_doc_DATA = $(top_srcdir)/voacapl/itshfbc/news/*
 
 dist_man_MANS = man/voacapl.1
 
 dist_bin_SCRIPTS = makeitshfbc
 
 nobase_dist_data_DATA = \
-	voacapl/itshfbc/antennas/default/ccir.* \
-	voacapl/itshfbc/antennas/default/const17.voa\
-	voacapl/itshfbc/antennas/default/isotrope\
-	voacapl/itshfbc/antennas/default/swwhip.voa\
-	voacapl/itshfbc/antennas/samples/sample.*\
-	voacapl/itshfbc/areadata/default/default.*\
-	voacapl/itshfbc/area_inv/default/default.*\
-	voacapl/itshfbc/coeffs/*.asc\
-	voacapl/itshfbc/database/cirafp.911\
-	voacapl/itshfbc/database/colors.dat\
-	voacapl/itshfbc/database/colors.win\
-	voacapl/itshfbc/database/north_pole.txt\
-	voacapl/itshfbc/database/version.w32\
-	voacapl/itshfbc/database/version.win\
-	voacapl/itshfbc/database/voacap.def\
-	voacapl/itshfbc/geocity/*.geo\
-	voacapl/itshfbc/geonatio/*.geo\
-	voacapl/itshfbc/geostate/*.geo \
-	voacapl/itshfbc/run/*.dat\
-	voacapl/itshfbc/run/voaareax.da1\
-	voacapl/itshfbc/run/voacapx.out\
-	voacapl/itshfbc/run/voacapw.asc\
-	voacapl/itshfbc/run/voacap.cir
+	$(top_srcdir)/voacapl/itshfbc/antennas/default/ccir.* \
+	$(top_srcdir)/voacapl/itshfbc/antennas/default/const17.voa\
+	$(top_srcdir)/voacapl/itshfbc/antennas/default/isotrope\
+	$(top_srcdir)/voacapl/itshfbc/antennas/default/swwhip.voa\
+	$(top_srcdir)/voacapl/itshfbc/antennas/samples/sample.*\
+	$(top_srcdir)/voacapl/itshfbc/areadata/default/default.*\
+	$(top_srcdir)/voacapl/itshfbc/area_inv/default/default.*\
+	$(top_srcdir)/voacapl/itshfbc/coeffs/*.asc\
+	$(top_srcdir)/voacapl/itshfbc/database/cirafp.911\
+	$(top_srcdir)/voacapl/itshfbc/database/colors.dat\
+	$(top_srcdir)/voacapl/itshfbc/database/colors.win\
+	$(top_srcdir)/voacapl/itshfbc/database/north_pole.txt\
+	$(top_srcdir)/voacapl/itshfbc/database/version.w32\
+	$(top_srcdir)/voacapl/itshfbc/database/version.win\
+	$(top_srcdir)/voacapl/itshfbc/database/voacap.def\
+	$(top_srcdir)/voacapl/itshfbc/geocity/*.geo\
+	$(top_srcdir)/voacapl/itshfbc/geonatio/*.geo\
+	$(top_srcdir)/voacapl/itshfbc/geostate/*.geo \
+	$(top_srcdir)/voacapl/itshfbc/run/*.dat\
+	$(top_srcdir)/voacapl/itshfbc/run/voaareax.da1\
+	$(top_srcdir)/voacapl/itshfbc/run/voacapx.out\
+	$(top_srcdir)/voacapl/itshfbc/run/voacapw.asc\
+	$(top_srcdir)/voacapl/itshfbc/run/voacap.cir
 
 install-exec-hook:
 	sed "s|__PREFIX__|@prefix@|" $(DESTDIR)/$(bindir)/makeitshfbc > makeitshfbc.tmp
--- origsrc/voacapl-0.6.7/configure.ac	2015-05-09 10:10:18.000000000 +0900
+++ src/voacapl-0.6.7/configure.ac	2017-12-24 01:02:22.115000000 +0900
@@ -1,9 +1,9 @@
 AC_INIT([voacapl], [0.6.7], [jwatson@neomailbox.ch])
 m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
-AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+AM_INIT_AUTOMAKE([-Wall foreign])
 AC_PROG_F77([gfortran])
 AC_PROG_LN_S
-AC_PROG_RANLIB
+LT_INIT
 AC_CONFIG_FILES([
  	Makefile
 	src/modules/Makefile
--- origsrc/voacapl-0.6.7/src/hfmufesw/Makefile.am	2015-04-24 04:42:59.000000000 +0900
+++ src/voacapl-0.6.7/src/hfmufesw/Makefile.am	2017-12-24 00:52:24.069000000 +0900
@@ -1,6 +1,6 @@
-lib_LIBRARIES = libhfmufesw.a
-AM_FFLAGS = -w -fno-backslash -ffixed-line-length-none -fno-sign-zero -I ../modules/
-libhfmufesw_a_SOURCES =  agauss.f\
+lib_LTLIBRARIES = libhfmufesw.la
+AM_FFLAGS = -w -fno-backslash -ffixed-line-length-none -fno-sign-zero -I ../modules/
+libhfmufesw_la_SOURCES =  agauss.f\
 	cang.f\
 	cmpinv.f\
 	coll.f\
@@ -16,4 +16,4 @@ libhfmufesw_a_SOURCES =  agauss.f\
 	sim.f \
 	sqmult.f \
 	zmut.f
-
+libhfmufesw_la_LDFLAGS = -no-undefined
--- origsrc/voacapl-0.6.7/src/modules/Makefile.am	2015-04-24 04:42:59.000000000 +0900
+++ src/voacapl-0.6.7/src/modules/Makefile.am	2017-12-24 01:01:24.928000000 +0900
@@ -1,3 +1,3 @@
-bin_PROGRAMS = mods
+# bin_PROGRAMS = mods
 AM_FFLAGS = -w -c
 mods_SOURCES =  error_codes.f 
--- origsrc/voacapl-0.6.7/src/voa_lib/Makefile.am	2015-04-24 04:42:59.000000000 +0900
+++ src/voacapl-0.6.7/src/voa_lib/Makefile.am	2017-12-24 00:54:21.631000000 +0900
@@ -1,9 +1,9 @@
-lib_LIBRARIES = libvoa.a
-AM_FFLAGS = -w -fno-backslash -ffixed-line-length-none -fno-sign-zero
-libvoa_a_SOURCES =  dazel1.f\
+lib_LTLIBRARIES = libvoa.la
+AM_FFLAGS = -w -fno-backslash -ffixed-line-length-none -fno-sign-zero
+libvoa_la_SOURCES =  dazel1.f\
 	deg2chr.f\
-	doscolr.f\
-	icoldos.f\
+	doscolr.f\
+	icoldos.f\
 	col_dos.f\
 	wincolr.f\
 	chardeg.f\
@@ -12,5 +12,6 @@ libvoa_a_SOURCES =  dazel1.f\
 	rblankc.f\
 	set_run.f\
 	suffix.f\
-	pwrcut.f
-
+	pwrcut.f
+libvoa_la_LDFLAGS = -no-undefined
+
--- origsrc/voacapl-0.6.7/src/voacapw/Makefile.am	2015-04-24 04:42:59.000000000 +0900
+++ src/voacapl-0.6.7/src/voacapw/Makefile.am	2017-12-24 00:46:56.351000000 +0900
@@ -1,6 +1,6 @@
 bin_PROGRAMS = voacapl
 AM_FFLAGS = -w -fno-backslash -ffixed-line-length-none -fno-sign-zero -I ../modules/
-voacapl_LDADD = ../hfmufesw/libhfmufesw.a ../voa_lib/libvoa.a  ../wp10dwin/libwp10dwin.a
+voacapl_LDADD = ../hfmufesw/libhfmufesw.la ../voa_lib/libvoa.la  ../wp10dwin/libwp10dwin.la
 voacapl_SOURCES = ficepac.hdr \
  	fice_ssn.hdr \
 	voa_defs_mod.f \
--- origsrc/voacapl-0.6.7/src/wp10dwin/Makefile.am	2015-04-24 04:42:59.000000000 +0900
+++ src/voacapl-0.6.7/src/wp10dwin/Makefile.am	2017-12-24 00:57:36.210000000 +0900
@@ -1,6 +1,6 @@
-lib_LIBRARIES = libwp10dwin.a
+lib_LTLIBRARIES = libwp10dwin.la
 AM_FFLAGS = -w -fno-backslash -ffixed-line-length-none -fno-sign-zero -I ../modules/
-libwp10dwin_a_SOURCES =  antcal.f\
+libwp10dwin_la_SOURCES =  antcal.f\
 	azmax.f\
 	bessel.f\
 	antinit2.f\
@@ -30,4 +30,5 @@ libwp10dwin_a_SOURCES =  antcal.f\
 	dbltrap.f\
 	f2.f\
 	invcon.f
-
+libwp10dwin_la_LDFLAGS = -no-undefined
+libwp10dwin_la_LIBADD = ../voa_lib/libvoa.la
