--- origsrc/voacapl-0.6.5/Makefile.am	2014-09-30 13:40:21.000000000 +0900
+++ src/voacapl-0.6.5/Makefile.am	2015-04-09 21:44:33.545881200 +0900
@@ -17,36 +17,36 @@ dist_doc_DATA = README.md\
 		AUTHORS\
 		logo.png
 
-nobase_dist_doc_DATA = voacapl/itshfbc/news/*
+nobase_dist_doc_DATA = $(top_srcdir)/voacapl/itshfbc/news/*
 
 dist_man_MANS = man/voacapl.1
 
 dist_bin_SCRIPTS = makeitshfbc
 
 nobase_dist_data_DATA = \
-	voacapl/itshfbc/antennas/default/ccir.* \
-	voacapl/itshfbc/antennas/default/const17.voa\
-	voacapl/itshfbc/antennas/default/isotrope\
-	voacapl/itshfbc/antennas/default/swwhip.voa\
-	voacapl/itshfbc/antennas/samples/sample.*\
-	voacapl/itshfbc/areadata/default/default.*\
-	voacapl/itshfbc/area_inv/default/default.*\
-	voacapl/itshfbc/coeffs/*.asc\
-	voacapl/itshfbc/database/cirafp.911\
-	voacapl/itshfbc/database/colors.dat\
-	voacapl/itshfbc/database/colors.win\
-	voacapl/itshfbc/database/north_pole.txt\
-	voacapl/itshfbc/database/version.w32\
-	voacapl/itshfbc/database/version.win\
-	voacapl/itshfbc/database/voacap.def\
-	voacapl/itshfbc/geocity/*.geo\
-	voacapl/itshfbc/geonatio/*.geo\
-	voacapl/itshfbc/geostate/*.geo \
-	voacapl/itshfbc/run/*.dat\
-	voacapl/itshfbc/run/voaareax.da1\
-	voacapl/itshfbc/run/voacapx.out\
-	voacapl/itshfbc/run/voacapw.asc\
-	voacapl/itshfbc/run/voacap.cir
+	$(top_srcdir)/voacapl/itshfbc/antennas/default/ccir.* \
+	$(top_srcdir)/voacapl/itshfbc/antennas/default/const17.voa\
+	$(top_srcdir)/voacapl/itshfbc/antennas/default/isotrope\
+	$(top_srcdir)/voacapl/itshfbc/antennas/default/swwhip.voa\
+	$(top_srcdir)/voacapl/itshfbc/antennas/samples/sample.*\
+	$(top_srcdir)/voacapl/itshfbc/areadata/default/default.*\
+	$(top_srcdir)/voacapl/itshfbc/area_inv/default/default.*\
+	$(top_srcdir)/voacapl/itshfbc/coeffs/*.asc\
+	$(top_srcdir)/voacapl/itshfbc/database/cirafp.911\
+	$(top_srcdir)/voacapl/itshfbc/database/colors.dat\
+	$(top_srcdir)/voacapl/itshfbc/database/colors.win\
+	$(top_srcdir)/voacapl/itshfbc/database/north_pole.txt\
+	$(top_srcdir)/voacapl/itshfbc/database/version.w32\
+	$(top_srcdir)/voacapl/itshfbc/database/version.win\
+	$(top_srcdir)/voacapl/itshfbc/database/voacap.def\
+	$(top_srcdir)/voacapl/itshfbc/geocity/*.geo\
+	$(top_srcdir)/voacapl/itshfbc/geonatio/*.geo\
+	$(top_srcdir)/voacapl/itshfbc/geostate/*.geo \
+	$(top_srcdir)/voacapl/itshfbc/run/*.dat\
+	$(top_srcdir)/voacapl/itshfbc/run/voaareax.da1\
+	$(top_srcdir)/voacapl/itshfbc/run/voacapx.out\
+	$(top_srcdir)/voacapl/itshfbc/run/voacapw.asc\
+	$(top_srcdir)/voacapl/itshfbc/run/voacap.cir
 
 install-exec-hook:
 	sed "s|__PREFIX__|@prefix@|" $(DESTDIR)/$(bindir)/makeitshfbc > makeitshfbc.tmp
@@ -67,5 +67,3 @@ install-data-hook:
 	echo "***************************************************" &&\
 	echo "" &&\
 	echo "" 
-
-
--- origsrc/voacapl-0.6.5/src/hfmufesw/Makefile.am	2009-12-04 17:45:51.000000000 +0900
+++ src/voacapl-0.6.5/src/hfmufesw/Makefile.am	2015-04-09 21:47:33.527467200 +0900
@@ -1,4 +1,4 @@
-lib_LIBRARIES = libhfmufesw.a
+noinst_LIBRARIES = libhfmufesw.a
 AM_FFLAGS = -w -fno-backslash -ffixed-line-length-none -fno-sign-zero -I ../modules/
 libhfmufesw_a_SOURCES =  agauss.f\
 	cang.f\
@@ -16,4 +16,3 @@ libhfmufesw_a_SOURCES =  agauss.f\
 	sim.f \
 	sqmult.f \
 	zmut.f
-
--- origsrc/voacapl-0.6.5/src/voa_lib/Makefile.am	2009-12-04 17:45:51.000000000 +0900
+++ src/voacapl-0.6.5/src/voa_lib/Makefile.am	2015-04-09 21:47:45.311635400 +0900
@@ -1,4 +1,4 @@
-lib_LIBRARIES = libvoa.a
+noinst_LIBRARIES = libvoa.a
 AM_FFLAGS = -w -fno-backslash -ffixed-line-length-none -fno-sign-zero
 libvoa_a_SOURCES =  dazel1.f\
 	deg2chr.f\
@@ -13,4 +13,3 @@ libvoa_a_SOURCES =  dazel1.f\
 	set_run.f\
 	suffix.f\
 	pwrcut.f
-
--- origsrc/voacapl-0.6.5/src/wp10dwin/Makefile.am	2013-09-24 03:31:57.000000000 +0900
+++ src/voacapl-0.6.5/src/wp10dwin/Makefile.am	2015-04-09 21:47:56.438707200 +0900
@@ -1,4 +1,4 @@
-lib_LIBRARIES = libwp10dwin.a
+noinst_LIBRARIES = libwp10dwin.a
 AM_FFLAGS = -w -fno-backslash -ffixed-line-length-none -fno-sign-zero -I ../modules/
 libwp10dwin_a_SOURCES =  antcal.f\
 	azmax.f\
@@ -30,4 +30,3 @@ libwp10dwin_a_SOURCES =  antcal.f\
 	dbltrap.f\
 	f2.f\
 	invcon.f
-
