--- origsrc/libmediaart-0.3.0/configure.ac	2014-03-08 04:05:10.000000000 +0900
+++ src/libmediaart-0.3.0/configure.ac	2014-03-10 23:21:54.387728800 +0900
@@ -104,7 +104,6 @@ AC_PROG_CPP
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
-AC_PROG_RANLIB
 AC_PROG_MKDIR_P
 AC_PROG_YACC
 
@@ -120,7 +119,6 @@ AC_PROG_CPP
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
-AC_PROG_RANLIB
 
 # Checks for libraries.
 
--- origsrc/libmediaart-0.3.0/libmediaart/Makefile.am	2014-02-15 03:20:59.000000000 +0900
+++ src/libmediaart-0.3.0/libmediaart/Makefile.am	2014-03-10 23:21:06.538639100 +0900
@@ -45,7 +45,7 @@ libmediaart_@LIBMEDIAART_API_VERSION@_la
 endif
 endif
 
-libmediaart_@LIBMEDIAART_API_VERSION@_la_LDFLAGS = \
+libmediaart_@LIBMEDIAART_API_VERSION@_la_LDFLAGS = -no-undefined \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 
 libmediaart_@LIBMEDIAART_API_VERSION@_la_LIBADD = \
