--- origsrc/zvbi-0.2.33/configure.in	2008-09-05 03:14:50.000000000 +0900
+++ src/zvbi-0.2.33/configure.in	2011-04-03 17:40:10.393561200 +0900
@@ -107,7 +107,7 @@ AC_C_BIGENDIAN(
   AC_DEFINE(Z_BYTE_ORDER, 4321, [Byte order]),
   AC_DEFINE(Z_BYTE_ORDER, 1234, [Byte order]))
 
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 test -e site_def.h || cat <<EOF >site_def.h
 /* Site specific definitions */
--- origsrc/zvbi-0.2.33/src/Makefile.am	2008-02-19 09:35:14.000000000 +0900
+++ src/zvbi-0.2.33/src/Makefile.am	2011-04-03 18:11:48.570181900 +0900
@@ -110,7 +110,7 @@ libzvbi_la_LIBADD = \
 	$(LIBS) \
 	$(UNICODE_LIBS)
 
-libzvbi_la_LDFLAGS = -version-info $(LIBZVBI_SO_VERSION)
+libzvbi_la_LDFLAGS = -version-info $(LIBZVBI_SO_VERSION) -no-undefined
 
 libzvbi_chains_la_SOURCES = \
 	chains.c \
@@ -119,13 +119,9 @@ libzvbi_chains_la_SOURCES = \
 	proxy-client.c \
 	proxy-msg.c
 
-if MAINTAINER_MODE
-
-hammgen: hammgen.c
-	$(CC) -o hammgen $(srcdir)/hammgen.c
+libzvbi_chains_la_LDFLAGS = -no-undefined
 
-hamm-tables.h: hammgen$(EXEEXT) Makefile.am
-	$(top_builddir)/src/hammgen$(EXEEXT) > $@
+if MAINTAINER_MODE
 
 version.h: ../configure.in
 	echo -e "\n#define VBI_VERSION_MAJOR" `echo @VERSION@ \
