--- origsrc/zvbi-0.2.35/src/Makefile.am	2013-07-10 18:14:47.000000000 +0900
+++ src/zvbi-0.2.35/src/Makefile.am	2013-09-08 00:31:52.498067600 +0900
@@ -115,7 +115,7 @@ libzvbi_la_LIBADD = \
 	$(LIBS) \
 	$(UNICODE_LIBS)
 
-libzvbi_la_LDFLAGS = -version-info $(LIBZVBI_SO_VERSION)
+libzvbi_la_LDFLAGS = -version-info $(LIBZVBI_SO_VERSION) -no-undefined
 
 libzvbi_chains_la_SOURCES = \
 	chains.c \
@@ -124,13 +124,9 @@ libzvbi_chains_la_SOURCES = \
 	proxy-client.c \
 	proxy-msg.c
 
-if MAINTAINER_MODE
-
-hammgen: hammgen.c
-	$(CC) -o hammgen $(srcdir)/hammgen.c
+libzvbi_chains_la_LDFLAGS = -no-undefined
 
-hamm-tables.h: hammgen$(EXEEXT) Makefile.am
-	$(top_builddir)/src/hammgen$(EXEEXT) > $@
+if MAINTAINER_MODE
 
 version.h: ../configure.in
 	echo -e "\n#define VBI_VERSION_MAJOR" `echo @VERSION@ \
