--- origsrc/zvbi-0.2.41/src/Makefile.am	2023-02-13 22:23:51.000000000 +0900
+++ src/zvbi-0.2.41/src/Makefile.am	2023-04-23 10:16:19.562598500 +0900
@@ -115,7 +115,7 @@ libzvbi_la_LIBADD = \
 	$(LIBS) \
 	$(UNICODE_LIBS)
 
-libzvbi_la_LDFLAGS = -version-info $(LIBZVBI_SO_VERSION)
+libzvbi_la_LDFLAGS = -no-undefined -version-info $(LIBZVBI_SO_VERSION)
 
 libzvbi_chains_la_SOURCES = \
 	chains.c \
@@ -123,10 +123,11 @@ libzvbi_chains_la_SOURCES = \
 	misc.c \
 	proxy-client.c \
 	proxy-msg.c
+libzvbi_chains_la_LDFLAGS = -no-undefined
 
 if MAINTAINER_MODE
 
-hammgen: hammgen.c
+hammgen$(EXEEXT): hammgen.c
 	$(CC) -o hammgen $(srcdir)/hammgen.c
 
 hamm-tables.h: hammgen$(EXEEXT) Makefile.am
