--- origsrc/libgit2-0.20.0/CMakeLists.txt	2013-11-20 20:53:33.000000000 +0900
+++ src/libgit2-0.20.0/CMakeLists.txt	2013-12-01 15:05:46.796992200 +0900
@@ -288,7 +288,7 @@ ELSE ()
 		ADD_DEFINITIONS(-D__USE_MINGW_ANSI_STDIO=1)
 
 	ELSEIF (BUILD_SHARED_LIBS)
-		SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fvisibility=hidden -fPIC")
+		SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fvisibility=hidden")
 	ENDIF ()
 	IF (APPLE) # Apple deprecated OpenSSL
 		SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-deprecated-declarations")
--- origsrc/libgit2-0.20.0/src/util.c	2013-11-20 20:53:33.000000000 +0900
+++ src/libgit2-0.20.0/src/util.c	2013-12-01 15:04:22.442500000 +0900
@@ -726,7 +726,7 @@ static int GIT_STDLIB_CALL git__qsort_r_
 void git__qsort_r(
 	void *els, size_t nel, size_t elsize, git__sort_r_cmp cmp, void *payload)
 {
-#if defined(__MINGW32__) || defined(AMIGA) || \
+#if defined(__MINGW32__) || defined(AMIGA) || defined(__CYGWIN__) || \
 	defined(__OpenBSD__) || defined(__NetBSD__) || \
 	defined(__gnu_hurd__) || defined(__ANDROID_API__) || \
 	(__GLIBC__ == 2 && __GLIBC_MINOR__ < 8)
