--- origsrc/fIcy-1.0.21/Makefile	2017-07-07 03:36:50.000000000 +0900
+++ src/fIcy-1.0.21/Makefile	2025-12-08 18:25:07.589551800 +0900
@@ -3,7 +3,7 @@
 
 # Flags
 CWARN += -Wall -Wextra -Wno-unused-parameter
-CXXFLAGS += -std=c++03 $(CWARN)
+CXXFLAGS += -std=gnu++03 $(CWARN)
 CPPFLAGS += -MD -D_FILE_OFFSET_BITS=64
 
 # Paths
--- origsrc/fIcy-1.0.21/rewrite.cc	2013-12-23 22:25:51.000000000 +0900
+++ src/fIcy-1.0.21/rewrite.cc	2025-12-08 18:25:07.593587500 +0900
@@ -28,7 +28,7 @@ using std::runtime_error;
 #endif
 
 // streams
-#if !defined(BSD) && !defined(__linux__)
+#if !defined(BSD) && !defined(__linux__) && !defined(__CYGWIN__)
 #define HAS_STREAMS
 #include <sys/ioctl.h>
 #include <stropts.h>
@@ -42,6 +42,9 @@ struct winsize;
 pid_t
 forkpty(int* amaster, char*, const termios*, const winsize*)
 {
+#if defined(__CYGWIN__)
+  return -1;
+#else
   pid_t pid;
   int masterFd;
 
@@ -98,6 +101,7 @@ forkpty(int* amaster, char*, const termi
   }
 
   return pid;
+#endif
 }
 
 
