diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/biblesync/CMakeLists.txt src/biblesync/CMakeLists.txt
--- origsrc/biblesync/CMakeLists.txt	2016-01-29 19:02:10.000000000 +0900
+++ src/biblesync/CMakeLists.txt	2016-01-29 19:06:16.817540400 +0900
@@ -42,7 +42,7 @@ ELSE(WIN32)
     # Headers and linkage from libuuid
     FIND_PACKAGE(UUID REQUIRED)
     INCLUDE_DIRECTORIES("${UUID_INCLUDE_DIRS}")
-    TARGET_LINK_LIBRARIES(biblesync "${UUID_LIBRARIES}")
+    TARGET_LINK_LIBRARIES(biblesync "${UUID_LIBRARIES}" "intl")
 ENDIF(WIN32)
 
 # Allow build systems to specify non-standard install locations
@@ -107,10 +107,15 @@ CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DI
 
 # Install both the library and its headers
 INSTALL(TARGETS biblesync
-    DESTINATION "${LIBDIR}")
+    RUNTIME DESTINATION bin
+    ARCHIVE DESTINATION "${LIBDIR}")
 INSTALL(FILES ${biblesync_headers}
     DESTINATION "${INCLUDEDIR}")
 INSTALL(FILES "${CMAKE_CURRENT_BINARY_DIR}/biblesync.pc"
     DESTINATION "${LIBDIR}/pkgconfig")
 INSTALL(FILES "${CMAKE_CURRENT_SOURCE_DIR}/man/biblesync.7"
     DESTINATION "${MANDIR}")
+
+IF(CYGWIN)
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fpermissive")
+ENDIF()
