--- origsrc/biblesync-2.1.0/CMakeLists.txt	2020-05-25 02:57:02.000000000 +0900
+++ src/biblesync-2.1.0/CMakeLists.txt	2020-06-05 14:44:35.850894200 +0900
@@ -42,7 +42,7 @@ ELSE(WIN32)
     # Headers and linkage from libuuid
     FIND_PACKAGE(UUID REQUIRED)
     INCLUDE_DIRECTORIES("${UUID_INCLUDE_DIRS}")
-    TARGET_LINK_LIBRARIES(biblesync "${UUID_LIBRARIES}")
+    TARGET_LINK_LIBRARIES(biblesync "${UUID_LIBRARIES}" "intl")
 ENDIF(WIN32)
 
 # Allow build systems to specify non-standard install locations
@@ -107,7 +107,8 @@ CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DI
 
 # Install both the library and its headers
 INSTALL(TARGETS biblesync
-    DESTINATION "${LIBDIR}")
+    RUNTIME DESTINATION bin
+    ARCHIVE DESTINATION "${LIBDIR}")
 INSTALL(FILES ${biblesync_headers}
     DESTINATION "${INCLUDEDIR}")
 INSTALL(FILES "${CMAKE_CURRENT_BINARY_DIR}/biblesync.pc"
