--- origsrc/biblesync-2.1.0/CMakeLists.txt	2026-03-30 16:41:24.258539000 +0900
+++ src/biblesync-2.1.0/CMakeLists.txt	2026-03-30 16:56:12.412124500 +0900
@@ -107,7 +107,8 @@ CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DI
 
 # Install both the library and its headers
 INSTALL(TARGETS biblesync
-    DESTINATION "${LIBDIR}")
+    RUNTIME DESTINATION bin
+    ARCHIVE DESTINATION "${LIBDIR}")
 INSTALL(FILES ${biblesync_headers}
     DESTINATION "${INCLUDEDIR}")
 INSTALL(FILES "${CMAKE_CURRENT_BINARY_DIR}/biblesync.pc"
