--- origsrc/biblesync-2.2.0/CMakeLists.txt	2026-07-23 00:55:38.000000000 +0900
+++ src/biblesync-2.2.0/CMakeLists.txt	2026-07-31 12:21:32.920188500 +0900
@@ -107,7 +107,8 @@ CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DI
 
 # Install both the library and its headers
 INSTALL(TARGETS biblesync
-    DESTINATION "${LIBDIR}")
+    RUNTIME DESTINATION bin
+    ARCHIVE DESTINATION "${LIBDIR}")
 INSTALL(FILES ${biblesync_headers}
     DESTINATION "${INCLUDEDIR}")
 INSTALL(FILES "${CMAKE_CURRENT_BINARY_DIR}/biblesync.pc"
