--- origsrc/libsqon-1.2.0a/Makefile.am	2015-07-27 21:05:58.000000000 +0900
+++ src/libsqon-1.2.0a/Makefile.am	2025-08-12 06:25:18.089828200 +0900
@@ -7,7 +7,7 @@ include_HEADERS = sqon.h
 lib_LTLIBRARIES = libsqon.la
 libsqon_la_SOURCES = sqon.c result.c
 
-libsqon_la_LDFLAGS = -version-info 3:0:2 `mysql_config --libs`
+libsqon_la_LDFLAGS = -no-undefined -version-info 3:0:2 `mysql_config --libs`
 
 libsqon_la_CFLAGS = -Wall -Wextra -Wunreachable-code -ftrapv -std=c11
 
--- origsrc/libsqon-1.2.0a/result.h	2015-07-27 21:05:58.000000000 +0900
+++ src/libsqon-1.2.0a/result.h	2025-08-12 06:50:38.209924200 +0900
@@ -19,7 +19,7 @@
 #define DELWINK_SQON_SQLQUERY_H
 
 #include <mysql/mysql.h>
-#include <postgresql/libpq-fe.h>
+#include <libpq-fe.h>
 #include <stdint.h>
 
 union res
--- origsrc/libsqon-1.2.0a/sqon.c	2015-07-27 21:05:58.000000000 +0900
+++ src/libsqon-1.2.0a/sqon.c	2025-08-12 06:50:31.619061400 +0900
@@ -17,7 +17,7 @@
 
 #include <jansson.h>
 #include <mysql/mysql.h>
-#include <postgresql/libpq-fe.h>
+#include <libpq-fe.h>
 #include <string.h>
 
 #include "sqon.h"
