--- origsrc/gtest-1.5.0/Makefile.am	2010-04-16 07:02:08.000000000 +0900
+++ src/gtest-1.5.0/Makefile.am	2011-02-14 00:46:35.277277400 +0900
@@ -180,6 +180,7 @@ endif
 lib_LTLIBRARIES = lib/libgtest.la lib/libgtest_main.la
 
 lib_libgtest_la_SOURCES = src/gtest-all.cc
+lib_libgtest_la_LDFLAGS = -no-undefined
 
 pkginclude_HEADERS = include/gtest/gtest.h \
                      include/gtest/gtest-death-test.h \
@@ -206,6 +207,7 @@ pkginclude_internal_HEADERS = \
 
 lib_libgtest_main_la_SOURCES = src/gtest_main.cc
 lib_libgtest_main_la_LIBADD = lib/libgtest.la
+lib_libgtest_main_la_LDFLAGS = -no-undefined
 
 # Bulid rules for samples and tests. Automake's naming for some of
 # these variables isn't terribly obvious, so this is a brief
--- origsrc/gtest-1.5.0/configure.ac	2010-04-16 07:02:09.000000000 +0900
+++ src/gtest-1.5.0/configure.ac	2011-02-14 00:45:47.036953000 +0900
@@ -26,7 +26,7 @@ AM_INIT_AUTOMAKE([1.9 dist-bzip2 dist-zi
 AC_PROG_CC
 AC_PROG_CXX
 AC_LANG([C++])
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 # TODO(chandlerc@google.com): Currently we aren't running the Python tests
 # against the interpreter detected by AM_PATH_PYTHON, and so we condition
