--- origsrc/gtest-1.5.0/configure.ac	2010-04-16 07:02:09.000000000 +0900
+++ src/gtest-1.5.0/configure.ac	2010-05-23 13:59:23.472775000 +0900
@@ -26,6 +26,7 @@ AM_INIT_AUTOMAKE([1.9 dist-bzip2 dist-zi
 AC_PROG_CC
 AC_PROG_CXX
 AC_LANG([C++])
+AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
 
 # TODO(chandlerc@google.com): Currently we aren't running the Python tests
--- origsrc/gtest-1.5.0/Makefile.am	2010-04-16 07:02:08.000000000 +0900
+++ src/gtest-1.5.0/Makefile.am	2010-05-23 13:59:23.457397500 +0900
@@ -181,6 +181,8 @@ lib_LTLIBRARIES = lib/libgtest.la lib/li
 
 lib_libgtest_la_SOURCES = src/gtest-all.cc
 
+lib_libgtest_la_LDFLAGS = -no-undefined
+
 pkginclude_HEADERS = include/gtest/gtest.h \
                      include/gtest/gtest-death-test.h \
                      include/gtest/gtest-message.h \
@@ -206,6 +208,7 @@ pkginclude_internal_HEADERS = \
 
 lib_libgtest_main_la_SOURCES = src/gtest_main.cc
 lib_libgtest_main_la_LIBADD = lib/libgtest.la
+lib_libgtest_main_la_LDFLAGS = -no-undefined
 
 # Bulid rules for samples and tests. Automake's naming for some of
 # these variables isn't terribly obvious, so this is a brief
@@ -224,6 +227,8 @@ samples_libsamples_la_SOURCES = samples/
                                 samples/sample4.cc \
                                 samples/sample4.h
 
+samples_libsamples_la_LDFLAGS = -no-undefined
+
 TESTS=
 TESTS_ENVIRONMENT = GTEST_SOURCE_DIR="$(srcdir)/test" \
                     GTEST_BUILD_DIR="$(top_builddir)/test"
