--- origsrc/libaribcaption-1.0.0/CMakeLists.txt	2022-12-17 23:29:03.000000000 +0900
+++ src/libaribcaption-1.0.0/CMakeLists.txt	2023-09-18 08:48:01.749262200 +0900
@@ -265,6 +265,10 @@ set_target_properties(aribcaption
         CXX_VISIBILITY_PRESET hidden
 )
 
+if (DEFINED SOVERSION)
+    set_target_properties(aribcaption PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 # Enable /utf-8 for MSVC
 target_compile_options(aribcaption
     PRIVATE
