--- origsrc/qpid-proton-0.37.0/c/src/proactor/libuv.c	2022-03-16 01:38:34.000000000 +0900
+++ src/qpid-proton-0.37.0/c/src/proactor/libuv.c	2022-10-12 18:11:47.850280500 +0900
@@ -135,6 +135,10 @@ static void work_init(work_t* w, pn_proa
 
 /* ================ IO ================ */
 
+#if defined(__addr_t_defined)
+#define addr_t libuv_addr_t
+#endif
+
 /* A resolvable address */
 typedef struct addr_t {
   char addr_buf[PN_MAX_ADDR];
