--- origsrc/sqlsmith-1.2.1/Makefile.am	2018-05-04 03:21:03.000000000 +0900
+++ src/sqlsmith-1.2.1/Makefile.am	2019-07-12 11:10:36.352665400 +0900
@@ -34,6 +34,7 @@ gitrev.h: $(HEADERS) $(SOURCES)
 	-if git describe --dirty --tags --always > /dev/null ; then \
 	echo "#define GITREV \"$$(git describe --dirty --tags --always)\"" > $@ ;\
 	fi
+	[ -x $@ ] || echo "#define GITREV \"${VERSION}\"" > $@
 
 filterdump:
 	psql -c 'copy (select error from known) to stdout' smith|sort -u > known.txt
