--- origsrc/aaflib-0.1/CMakeLists.txt	2019-10-09 03:18:52.000000000 +0900
+++ src/aaflib-0.1/CMakeLists.txt	2025-08-29 17:28:29.119673600 +0900
@@ -11,36 +11,26 @@ SET(AAF_HEADER aa.h aa_aaf.h aa_exceptio
 SET(AAF_SOURCE aa_rounding.cpp aa_interval.cpp aa_aaftrigo.cpp aa_aafapprox.cpp aa_aafarithm.cpp aa_aafcommon.cpp aa_exceptions.cpp aa_aafspecial.cpp)
 
 add_library(
-    aaf.lin64
+    aaf
     SHARED 
     ${AAF_SOURCE}
 )
-target_compile_definitions(aaf.lin64 PRIVATE LIN64 USE_AAF_EXTENSIONS)
-target_compile_options(aaf.lin64 PRIVATE -Wall -Weffc++ -Wno-deprecated -O3)
+target_compile_definitions(aaf PRIVATE LIN64 USE_AAF_EXTENSIONS)
+target_compile_options(aaf PRIVATE -Wall -Weffc++ -Wno-deprecated -O3)
 set_target_properties(
-    aaf.lin64 PROPERTIES
+    aaf PROPERTIES
     SOVERSION 0
     VERSION 0.1.0
     )
 
-add_library(
-    aaf.lin64.static
-    STATIC 
-    ${AAF_SOURCE}
-)
-set_target_properties(
-    aaf.lin64.static PROPERTIES
-    SOVERSION 0
-    VERSION 0.1.0
-    )
-
-target_compile_definitions(aaf.lin64.static PRIVATE LIN64 USE_AAF_EXTENSIONS)
-target_compile_options(aaf.lin64.static PRIVATE -Wall -Weffc++ -Wno-deprecated -O3)
-set_target_properties(aaf.lin64.static PROPERTIES OUTPUT_NAME aaf.lin64)
-
 include(GNUInstallDirs)
 set(AAF_INSTALL_INCLUDEDIR ${CMAKE_INSTALL_INCLUDEDIR}/aaf)
 install(FILES ${AAF_HEADER} DESTINATION ${AAF_INSTALL_INCLUDEDIR} COMPONENT devel)
-install(TARGETS aaf.lin64 DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT core)
-install(TARGETS aaf.lin64.static DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT core)
+install(TARGETS aaf RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT core)
 
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/aaflib.pc.in ${CMAKE_BINARY_DIR}/aaflib.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/aaflib.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/aaflib-0.1/aaflib.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/aaflib-0.1/aaflib.pc.in	2025-08-29 17:30:40.524664700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: aaflib
+Description: Implementation of the mathematical construct of affine arithmetics in C++
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -laaf
+Cflags: -I${includedir}
