--- origsrc/tnylpo-1.1.5/makefile	2023-07-23 23:20:22.000000000 +0900
+++ src/tnylpo-1.1.5/makefile	2023-08-29 11:43:26.129229900 +0900
@@ -57,6 +57,9 @@ CFLAGS+=-m64
 LIBS=-L $(NCURSESROOT)/lib/64 -R $(NCURSESROOT)/lib/64
 endif
 LIBS+=-lncursesw -lrt
+else
+CFLAGS+=`pkg-config --cflags ncurses`
+LIBS+=`pkg-config --libs ncurses`
 endif
 OBJS=main.o readconf.o util.o screen.o cpu.o os.o chario.o
 CONVERT_OBJS=tnylpo-convert.o readconf.o util.o
