--- origsrc/iat/src/Makefile.am	2012-02-11 20:41:45.000000000 +0900
+++ src/iat/src/Makefile.am	2012-02-11 20:50:07.099871900 +0900
@@ -14,7 +14,7 @@ libiat_la_SOURCES =   libiat/util.c \
 		libiat/mkcore.c 
 
 
-libiat_la_LDFLAGS = -version-info 0:0:0
+libiat_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 
 
 
--- origsrc/iat/src/libiat/define.h	2012-02-11 20:41:45.000000000 +0900
+++ src/iat/src/libiat/define.h	2012-02-11 20:50:18.587072200 +0900
@@ -45,7 +45,7 @@
 #include <stdio.h>
 
 #ifdef HAVE_CONFIG_H
-#include "../../config.h"
+#include "../config.h"
 #endif
 
 /*
