--- origsrc/quill-2.5.1/quill/CMakeLists.txt	2022-11-26 06:39:21.000000000 +0900
+++ src/quill-2.5.1/quill/CMakeLists.txt	2022-12-27 23:24:58.448049400 +0900
@@ -205,8 +205,13 @@ if (QUILL_MASTER_PROJECT OR QUILL_ENABLE
     set(QUILL_INC_DIR ${CMAKE_INSTALL_INCLUDEDIR} CACHE STRING
             "Installation directory for include files, relative to ${CMAKE_INSTALL_PREFIX}.")
 
-    set(QUILL_PKGCONFIG_DIR ${CMAKE_INSTALL_PREFIX}/pkgconfig CACHE PATH
+    if (CYGWIN)
+        set(QUILL_PKGCONFIG_DIR ${CMAKE_INSTALL_LIBDIR}/pkgconfig CACHE PATH
+            "Installation directory for pkgconfig (.pc) files, relative to ${CMAKE_INSTALL_LIBDIR}.")
+    else()
+        set(QUILL_PKGCONFIG_DIR ${CMAKE_INSTALL_PREFIX}/pkgconfig CACHE PATH
             "Installation directory for pkgconfig (.pc) files, relative to ${CMAKE_INSTALL_PREFIX}.")
+    endif()
 
     # Generate the version, config and target files into the build directory.
     # Generate version_config
