--- origsrc/quill-1.6.2/quill/CMakeLists.txt	2021-03-29 00:43:11.000000000 +0900
+++ src/quill-1.6.2/quill/CMakeLists.txt	2021-05-08 05:41:37.781183500 +0900
@@ -222,8 +222,13 @@ if (QUILL_MASTER_PROJECT OR QUILL_ENABLE
     set(QUILL_INC_DIR ${CMAKE_INSTALL_INCLUDEDIR} CACHE STRING
             "Installation directory for include files, relative to ${CMAKE_INSTALL_PREFIX}.")
 
-    set(QUILL_PKGCONFIG_DIR ${CMAKE_INSTALL_PREFIX}/pkgconfig CACHE PATH
+    if (CYGWIN)
+        set(QUILL_PKGCONFIG_DIR ${CMAKE_INSTALL_LIBDIR}/pkgconfig CACHE PATH
+            "Installation directory for pkgconfig (.pc) files, relative to ${CMAKE_INSTALL_LIBDIR}.")
+    else()
+        set(QUILL_PKGCONFIG_DIR ${CMAKE_INSTALL_PREFIX}/pkgconfig CACHE PATH
             "Installation directory for pkgconfig (.pc) files, relative to ${CMAKE_INSTALL_PREFIX}.")
+    endif()
 
     # Generate the version, config and target files into the build directory.
     # Generate version_config
