--- origsrc/quill-2.7.0/quill/CMakeLists.txt	2023-02-02 07:46:08.000000000 +0900
+++ src/quill-2.7.0/quill/CMakeLists.txt	2023-02-15 08:19:29.030506100 +0900
@@ -222,8 +222,13 @@ if (QUILL_MASTER_PROJECT OR QUILL_ENABLE
     set(QUILL_INC_DIR ${CMAKE_INSTALL_INCLUDEDIR} CACHE STRING
             "Installation directory for include files, relative to ${CMAKE_INSTALL_PREFIX}.")
 
-    set(QUILL_PKGCONFIG_DIR ${CMAKE_INSTALL_PREFIX}/pkgconfig CACHE PATH
+    if (CYGWIN)
+        set(QUILL_PKGCONFIG_DIR ${CMAKE_INSTALL_LIBDIR}/pkgconfig CACHE PATH
+            "Installation directory for pkgconfig (.pc) files, relative to ${CMAKE_INSTALL_LIBDIR}.")
+    else()
+        set(QUILL_PKGCONFIG_DIR ${CMAKE_INSTALL_PREFIX}/pkgconfig CACHE PATH
             "Installation directory for pkgconfig (.pc) files, relative to ${CMAKE_INSTALL_PREFIX}.")
+    endif()
 
     # Generate the version, config and target files into the build directory.
     # Generate version_config
