--- origsrc/quill-3.5.0/quill/CMakeLists.txt	2023-11-28 01:47:31.000000000 +0900
+++ src/quill-3.5.0/quill/CMakeLists.txt	2023-12-04 08:34:40.437398100 +0900
@@ -224,8 +224,13 @@ if (QUILL_MASTER_PROJECT OR QUILL_ENABLE
     set(QUILL_INC_DIR ${CMAKE_INSTALL_INCLUDEDIR} CACHE STRING
             "Installation directory for include files, relative to ${CMAKE_INSTALL_PREFIX}.")
 
-    set(QUILL_PKGCONFIG_DIR ${CMAKE_INSTALL_PREFIX}/pkgconfig CACHE PATH
+    if (CYGWIN)
+        set(QUILL_PKGCONFIG_DIR ${CMAKE_INSTALL_LIBDIR}/pkgconfig CACHE PATH
+            "Installation directory for pkgconfig (.pc) files, relative to ${CMAKE_INSTALL_LIBDIR}.")
+    else()
+        set(QUILL_PKGCONFIG_DIR ${CMAKE_INSTALL_PREFIX}/pkgconfig CACHE PATH
             "Installation directory for pkgconfig (.pc) files, relative to ${CMAKE_INSTALL_PREFIX}.")
+    endif()
 
     # Generate the version, config and target files into the build directory.
     # Generate version_config
