--- origsrc/quill-8.2.0/CMakeLists.txt	2025-02-22 23:17:41.000000000 +0900
+++ src/quill-8.2.0/CMakeLists.txt	2025-03-28 18:08:59.864927800 +0900
@@ -323,7 +323,7 @@ if (QUILL_MASTER_PROJECT OR QUILL_ENABLE
     set(QUILL_INC_DIR ${CMAKE_INSTALL_INCLUDEDIR} CACHE STRING
             "Installation directory for include files, relative to ${CMAKE_INSTALL_PREFIX}.")
 
-    set(QUILL_PKGCONFIG_DIR ${CMAKE_INSTALL_PREFIX}/pkgconfig CACHE PATH
+    set(QUILL_PKGCONFIG_DIR ${CMAKE_INSTALL_LIBDIR}/pkgconfig CACHE PATH
             "Installation directory for pkgconfig (.pc) files, relative to ${CMAKE_INSTALL_PREFIX}.")
 
     if (WIN32)
--- origsrc/quill-8.2.0/cmake/QuillUtils.cmake	2025-02-22 23:17:41.000000000 +0900
+++ src/quill-8.2.0/cmake/QuillUtils.cmake	2025-03-28 18:09:18.840519900 +0900
@@ -35,7 +35,7 @@ function(set_common_compile_options targ
     target_compile_options(${target_name} ${COMPILE_OPTIONS_VISIBILITY}
             # General warnings for Clang, AppleClang, and GNU
             $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>:
-            -Wall -Wextra -pedantic -Werror -Wredundant-decls -Wfloat-equal>
+            -Wall -Wextra -pedantic -Wredundant-decls -Wfloat-equal>
 
             # Clang specific options
             $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>>:
