--- origsrc/libevws/test/Makefile.am	2024-10-10 11:18:37.000000000 +0900
+++ src/libevws/test/Makefile.am	2024-10-10 11:18:38.444872000 +0900
@@ -31,5 +31,5 @@ evws_util_test_SOURCES = evws_util_test.
 	$(top_builddir)/src/evws_util.c \
 	$(top_builddir)/src/http_parser.h \
 	$(top_builddir)/src/http_parser.c \
-evws_util_test_LDFLAGS = -static
-evws_util_test_CFLAGS = -I$(top_builddir)/src
+evws_util_test_LDFLAGS =
+evws_util_test_CFLAGS = -I$(top_builddir)/src -I$(top_srcdir)/src
