--- origsrc/scrub-2.5.1/Makefile.am	2011-08-30 02:21:39.000000000 +0900
+++ src/scrub-2.5.1/Makefile.am	2012-06-16 12:23:04.478975400 +0900
@@ -11,3 +11,5 @@ rpm: dist rpmdirs
 	rpmbuild -ta --define "_topdir $(realpath $(top_builddir)/rpm)" $(DIST_ARCHIVES)
 
 EXTRA_DIST = DISCLAIMER META scrub.spec
+
+ACLOCAL_AMFLAGS = -I config
--- origsrc/scrub-2.5.1/src/hwrand.c	2012-04-17 07:57:23.000000000 +0900
+++ src/scrub-2.5.1/src/hwrand.c	2012-06-16 12:44:19.303194200 +0900
@@ -25,6 +25,7 @@
 #if HAVE_CONFIG_H
 #include "config.h"
 #endif
+#include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <errno.h>
