--- origsrc/scrub-2.6.1/Makefile.am	2014-08-27 03:15:12.000000000 +0900
+++ src/scrub-2.6.1/Makefile.am	2025-12-14 11:52:17.106940500 +0900
@@ -5,3 +5,5 @@ SUBDIRS += libscrub
 endif
 
 EXTRA_DIST = DISCLAIMER META scrub.spec
+
+ACLOCAL_AMFLAGS = -I config
--- origsrc/scrub-2.6.1/src/getsize.c	2014-08-27 03:15:12.000000000 +0900
+++ src/scrub-2.6.1/src/getsize.c	2025-12-14 11:48:15.825576200 +0900
@@ -231,6 +231,33 @@ error:
     return -1;
 }
 
+#elif defined(__CYGWIN__)
+/* Cygwin - uses POSIX lseek */
+int
+getsize(char *path, off_t *sizep)
+{
+    int fd = -1;
+    off_t size;
+
+    if ((fd = open(path, O_RDONLY)) < 0)
+        goto error;
+    
+    /* Seek to end to get size */
+    size = lseek(fd, 0, SEEK_END);
+    if (size < 0)
+        goto error;
+    
+    if (close(fd) < 0)
+        goto error;
+    
+    *sizep = size;
+    return 0;
+error:
+    if (fd != -1)
+        (void)close(fd);
+    return -1;
+}
+
 #else
 /* Unimplemented!  Scrub will tell user to use -s.
  */
