--- origsrc/scrub-2.6.1/Makefile.am	2014-08-27 03:15:12.000000000 +0900
+++ src/scrub-2.6.1/Makefile.am	2018-04-13 22:01:42.984000000 +0900
@@ -1,7 +1,9 @@
-SUBDIRS = src man test
+SUBDIRS = src man
 
 if LIBSCRUB
 SUBDIRS += libscrub
 endif
 
 EXTRA_DIST = DISCLAIMER META scrub.spec
+
+ACLOCAL_AMFLAGS = -I config
--- origsrc/scrub-2.6.1/src/getsize.c	2014-08-27 03:15:12.000000000 +0900
+++ src/scrub-2.6.1/src/getsize.c	2018-04-13 21:54:04.731000000 +0900
@@ -85,6 +85,31 @@ error:
     return -1;
 }
 
+#elif defined(__CYGWIN__)
+
+#include <sys/ioctl.h>
+#include <cygwin/fs.h>
+
+int
+getsize(char *path, off_t *sizep)
+{
+    int fd = -1;
+    unsigned long long numbytes;
+
+    if ((fd = open(path, O_RDONLY)) < 0)
+        goto error;
+    if (ioctl(fd, BLKGETSIZE64, &numbytes) < 0)
+        goto error;
+    if (close(fd) < 0)
+        goto error;
+    *sizep = (off_t)numbytes;
+    return 0;
+error:
+    if (fd != -1)
+        (void)close(fd);
+    return -1;       
+}
+
 #elif defined(__FreeBSD__)
 /* scrub-1.7 tested freebsd 5.3-RELEASE-p5 */
 #include <sys/ioctl.h>
