--- origsrc/libpwstor-0.1/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libpwstor-0.1/Makefile.am	2013-12-15 12:30:45.702028300 +0900
@@ -0,0 +1,17 @@
+
+bin_PROGRAMS = pwtool
+pwtool_SOURCES = pwtool.c
+pwtool_LDADD = libpwstor.la
+
+include_HEADERS = pwstor.h
+
+lib_LTLIBRARIES = libpwstor.la
+libpwstor_la_SOURCES = base64.c passwords.c utils.c
+libpwstor_la_LDFLAGS = -no-undefined
+
+man_MANS = man1/pwtool.1 \
+	man3/pws_base64_decode.3  man3/pws_base64_encode.3 \
+	man3/pws_memnuke.3        man3/pws_passwords_converttomulti.3 \
+	man3/pws_rand_get.3       man3/pwscheck.3 \
+	man3/pwsencoder.3         man3/pwstor.3
+
--- origsrc/libpwstor-0.1/configure.ac	2008-10-24 07:59:52.000000000 +0900
+++ src/libpwstor-0.1/configure.ac	2013-12-15 12:25:13.181549100 +0900
@@ -11,6 +11,8 @@ AC_CONFIG_HEADER(localconfig.h)
 
 AC_PREFIX_DEFAULT(/usr/local)
 
+AM_INIT_AUTOMAKE
+
 OLD_CFLAGS="$CFLAGS"
 
 AC_PROG_CC
@@ -18,6 +20,7 @@ AC_LANG(C)
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
 AC_PATH_PROG(TEST, [test], [test])
+LT_INIT
 
 CFLAGS="$OLD_CFLAGS"
 
@@ -43,6 +46,13 @@ AC_C_CONST
 AC_C_VOLATILE
 AC_C_BIGENDIAN
 AC_CHECK_HEADERS([sys/endian.h machine/endian.h])
+AC_CHECK_HEADERS([fcntl.h limits.h stdint.h stdlib.h string.h sys/param.h unistd.h])
+AC_TYPE_SIZE_T
+AC_TYPE_UINT16_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT8_T
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([memset strerror])
 
 AC_HEADER_STDC
 AC_CHECK_HEADER([stdint.h])
@@ -155,12 +165,12 @@ else
 fi
 
 if test "$cf_enable_hcalloc" = "yes"; then
-  AC_DEFINE([PWS_HCALLOC_ENABLE], [1])
+  AC_DEFINE([PWS_HCALLOC_ENABLE], [1], [enable hcalloc enable])
 fi
 
 if test "$cf_debug" = "1"; then
   DFLAGS="-Wall -Wno-long-long -Wno-unused-function -Wno-unused-variable -Wno-unused-value -ggdb"
-  AC_DEFINE([PWS_DEBUG], [1])
+  AC_DEFINE([PWS_DEBUG], [1], [enable debug])
 else
   DFLAGS=""
 fi
--- origsrc/libpwstor-0.1/pwtool.c	2008-10-24 07:59:52.000000000 +0900
+++ src/libpwstor-0.1/pwtool.c	2013-12-15 12:30:17.639528300 +0900
@@ -40,8 +40,6 @@
 #define OPT_MULTI		0x002
 
 
-extern char *optarg;
-extern int optind;
 uint16_t Options;
 
 static char *pwtool_prompt(uint32_t l);
