--- origsrc/dieharder-3.31.1/Makefile.am	2011-10-14 22:41:37.000000000 +0900
+++ src/dieharder-3.31.1/Makefile.am	2015-04-30 10:45:51.212536900 +0900
@@ -294,3 +294,8 @@ rdhprep:
 rdh:	$(RDHPKG)
 	R CMD INSTALL $(RDHPKG)
 
+
+doc_DATA = dieharder.html
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/dieharder-3.31.1/configure.ac	2011-10-14 22:41:37.000000000 +0900
+++ src/dieharder-3.31.1/configure.ac	2015-04-30 10:46:20.799366700 +0900
@@ -162,7 +162,7 @@ AC_HEADER_TIME
 # Files that configure needs to process with this so that they are
 # all happy and macro-replaced.
 #==================================================================
-AC_CONFIG_FILES([
+AC_CONFIG_FILES([dieharder.pc
  dieharder_version.h 
  dieharder.spec
  dieharder.html
--- origsrc/dieharder-3.31.1/dieharder.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/dieharder-3.31.1/dieharder.pc.in	2015-04-30 10:45:03.634526400 +0900
@@ -0,0 +1,5 @@
+Name: dieharder
+Description: Random Number Test Suite
+Version: @VERSION@
+Libs: -L@libdir@ -ldieharder
+Cflags: -I@includedir@
--- origsrc/dieharder-3.31.1/libdieharder/Makefile.am	2011-10-14 22:41:37.000000000 +0900
+++ src/dieharder-3.31.1/libdieharder/Makefile.am	2015-04-30 10:21:45.705408600 +0900
@@ -21,8 +21,8 @@ man3_MANS = libdieharder.3
 # Library definitions
 lib_LTLIBRARIES = libdieharder.la
 libdieharder_la_LIBADD = -lgsl -lgslcblas -lm
-libdieharder_la_LDFLAGS = -version-number @DIEHARDER_LT_VERSION@
-libdieharder_la_CFLAGS = -std=c99 -Wall -pedantic
+libdieharder_la_LDFLAGS = -no-undefined -version-number @DIEHARDER_LT_VERSION@
+libdieharder_la_CFLAGS = -std=gnu99 -Wall -pedantic
 
 # Sources
 # The shell script is the easy way to do this, by far.  But it may not
