--- origsrc/libschwa-0.4.0/configure.ac	2014-09-05 10:00:14.000000000 +0900
+++ src/libschwa-0.4.0/configure.ac	2025-08-09 21:05:15.945368800 +0900
@@ -195,7 +195,7 @@ PKG_CHECK_MODULES([ZMQLIB], [libzmq >= 3
 AM_CONDITIONAL([HAVE_LIBZMQ],  [test "$have_libzmq" = "yes" && test "$libschwa_cv_enable_libzmq" = "yes"])
 
 dnl Construct our base set of CPPFLAGS, depending on the compiler.
-LIBSCHWA_BASE_CPPFLAGS='-Wall -Wextra -Werror -pedantic -Wformat-security -Wpointer-arith -Wformat-nonliteral -Winit-self -Wfloat-equal -ffast-math -fstack-protector'
+LIBSCHWA_BASE_CPPFLAGS='-Wall -Wextra -pedantic -Wformat-security -Wpointer-arith -Wformat-nonliteral -Winit-self -Wfloat-equal -ffast-math -fstack-protector'
 if test "x$GXX" = xyes; then
   dnl Add compiler specific flags.
   if test "x$CLANG" = xyes; then
@@ -204,6 +204,9 @@ if test "x$GXX" = xyes; then
     LIBSCHWA_BASE_CPPFLAGS="$LIBSCHWA_BASE_CPPFLAGS -Wdouble-promotion"
   fi
 fi
+if test "$host_os" != "cygwin"; then
+  LIBSCHWA_BASE_CPPFLAGS="$LIBSCHWA_BASE_CPPFLAGS -Werror"
+fi
 AC_SUBST([LIBSCHWA_BASE_CPPFLAGS])
 
 dnl Configure autoconf inputs and outputs.
--- origsrc/libschwa-0.4.0/src/apps/ccg-pprint/Makefile.am	2014-09-05 10:00:14.000000000 +0900
+++ src/libschwa-0.4.0/src/apps/ccg-pprint/Makefile.am	2025-08-09 21:05:15.951431400 +0900
@@ -1,4 +1,4 @@
 bin_PROGRAMS = ccg-pprint
-ccg_pprint_CPPFLAGS = $(SIMD_FLAGS) -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
+ccg_pprint_CPPFLAGS = $(SIMD_FLAGS) -I../../lib -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
 ccg_pprint_LDADD = ../../lib/libschwa.la
 ccg_pprint_SOURCES = main.cc
--- origsrc/libschwa-0.4.0/src/apps/dr/Makefile.am	2014-09-05 10:00:14.000000000 +0900
+++ src/libschwa-0.4.0/src/apps/dr/Makefile.am	2025-08-09 21:05:15.955459100 +0900
@@ -1,4 +1,4 @@
 bin_PROGRAMS = dr
-dr_CPPFLAGS = $(SIMD_FLAGS) -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
+dr_CPPFLAGS = $(SIMD_FLAGS) -I../../lib -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
 dr_LDADD = ../../lib/libschwa.la
 dr_SOURCES = main.cc
--- origsrc/libschwa-0.4.0/src/apps/dr-count/Makefile.am	2014-09-05 10:00:14.000000000 +0900
+++ src/libschwa-0.4.0/src/apps/dr-count/Makefile.am	2025-08-09 21:05:15.973316600 +0900
@@ -1,4 +1,4 @@
 bin_PROGRAMS = dr-count
-dr_count_CPPFLAGS = $(SIMD_FLAGS) -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
+dr_count_CPPFLAGS = $(SIMD_FLAGS) -I../../lib -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
 dr_count_LDADD = ../../lib/libschwa.la
 dr_count_SOURCES = main.cc processor.cc processor.h
--- origsrc/libschwa-0.4.0/src/apps/dr-dist/Makefile.am	2014-09-05 10:00:14.000000000 +0900
+++ src/libschwa-0.4.0/src/apps/dr-dist/Makefile.am	2025-08-09 21:05:15.973316600 +0900
@@ -1,4 +1,4 @@
 bin_PROGRAMS = dr-dist
-dr_dist_CPPFLAGS = $(SIMD_FLAGS) -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS) $(ZMQLIB_CFLAGS)
+dr_dist_CPPFLAGS = $(SIMD_FLAGS) -I../../lib -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS) $(ZMQLIB_CFLAGS)
 dr_dist_LDADD = ../../lib/libschwa.la $(ZMQLIB_LIBS)
 dr_dist_SOURCES = main.cc
--- origsrc/libschwa-0.4.0/src/apps/dr-grep/Makefile.am	2014-09-05 10:00:14.000000000 +0900
+++ src/libschwa-0.4.0/src/apps/dr-grep/Makefile.am	2025-08-09 21:05:15.977363500 +0900
@@ -1,4 +1,4 @@
 bin_PROGRAMS = dr-grep
-dr_grep_CPPFLAGS = $(SIMD_FLAGS) -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
+dr_grep_CPPFLAGS = $(SIMD_FLAGS) -I../../lib -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
 dr_grep_LDADD = ../../lib/libschwa.la
 dr_grep_SOURCES = main.cc
--- origsrc/libschwa-0.4.0/src/apps/dr-head/Makefile.am	2014-09-05 10:00:14.000000000 +0900
+++ src/libschwa-0.4.0/src/apps/dr-head/Makefile.am	2025-08-09 21:05:15.981400300 +0900
@@ -1,4 +1,4 @@
 bin_PROGRAMS = dr-head
-dr_head_CPPFLAGS = $(SIMD_FLAGS) -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
+dr_head_CPPFLAGS = $(SIMD_FLAGS) -I../../lib -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
 dr_head_LDADD = ../../lib/libschwa.la
 dr_head_SOURCES = main.cc
--- origsrc/libschwa-0.4.0/src/apps/dr-less/Makefile.am	2014-09-05 10:00:14.000000000 +0900
+++ src/libschwa-0.4.0/src/apps/dr-less/Makefile.am	2025-08-09 21:05:15.983413400 +0900
@@ -1,4 +1,4 @@
 bin_PROGRAMS = dr-less
-dr_less_CPPFLAGS = $(SIMD_FLAGS) -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
+dr_less_CPPFLAGS = $(SIMD_FLAGS) -I../../lib -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
 dr_less_LDADD = ../../lib/libschwa.la
 dr_less_SOURCES = main.cc processor.cc processor.h
--- origsrc/libschwa-0.4.0/src/apps/dr-offsets/Makefile.am	2014-09-05 10:00:14.000000000 +0900
+++ src/libschwa-0.4.0/src/apps/dr-offsets/Makefile.am	2025-08-09 21:05:15.995259700 +0900
@@ -1,4 +1,4 @@
 bin_PROGRAMS = dr-offsets
-dr_offsets_CPPFLAGS = $(SIMD_FLAGS) -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
+dr_offsets_CPPFLAGS = $(SIMD_FLAGS) -I../../lib -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
 dr_offsets_LDADD = ../../lib/libschwa.la
 dr_offsets_SOURCES = main.cc
--- origsrc/libschwa-0.4.0/src/apps/dr-sample/Makefile.am	2014-09-05 10:00:14.000000000 +0900
+++ src/libschwa-0.4.0/src/apps/dr-sample/Makefile.am	2025-08-09 21:05:15.999327600 +0900
@@ -1,4 +1,4 @@
 bin_PROGRAMS = dr-sample
-dr_sample_CPPFLAGS = $(SIMD_FLAGS) -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
+dr_sample_CPPFLAGS = $(SIMD_FLAGS) -I../../lib -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
 dr_sample_LDADD = ../../lib/libschwa.la
 dr_sample_SOURCES = main.cc
--- origsrc/libschwa-0.4.0/src/apps/dr-tail/Makefile.am	2014-09-05 10:00:14.000000000 +0900
+++ src/libschwa-0.4.0/src/apps/dr-tail/Makefile.am	2025-08-09 21:05:16.001346400 +0900
@@ -1,4 +1,4 @@
 bin_PROGRAMS = dr-tail
-dr_tail_CPPFLAGS = $(SIMD_FLAGS) -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
+dr_tail_CPPFLAGS = $(SIMD_FLAGS) -I../../lib -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
 dr_tail_LDADD = ../../lib/libschwa.la
 dr_tail_SOURCES = main.cc
--- origsrc/libschwa-0.4.0/src/apps/dr-worker-example/Makefile.am	2014-09-05 10:00:14.000000000 +0900
+++ src/libschwa-0.4.0/src/apps/dr-worker-example/Makefile.am	2025-08-09 21:05:16.003357500 +0900
@@ -1,4 +1,4 @@
 noinst_PROGRAMS = dr-worker-example
-dr_worker_example_CPPFLAGS = $(SIMD_FLAGS) -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS) $(ZMQLIB_CFLAGS)
+dr_worker_example_CPPFLAGS = $(SIMD_FLAGS) -I../../lib -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS) $(ZMQLIB_CFLAGS)
 dr_worker_example_LDADD = ../../lib/libschwa.la $(ZMQLIB_LIBS)
 dr_worker_example_SOURCES = main.cc
--- origsrc/libschwa-0.4.0/src/apps/schwa-tokenizer/Makefile.am	2014-09-05 10:00:14.000000000 +0900
+++ src/libschwa-0.4.0/src/apps/schwa-tokenizer/Makefile.am	2025-08-09 21:05:16.007122700 +0900
@@ -1,4 +1,4 @@
 bin_PROGRAMS = schwa-tokenizer
-schwa_tokenizer_CPPFLAGS = $(SIMD_FLAGS) -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
+schwa_tokenizer_CPPFLAGS = $(SIMD_FLAGS) -I../../lib -I$(srcdir)/../../lib $(LIBSCHWA_BASE_CPPFLAGS)
 schwa_tokenizer_LDADD = ../../lib/libschwa.la
 schwa_tokenizer_SOURCES = main.cc
--- origsrc/libschwa-0.4.0/src/lib/schwa/io/array_reader.cc	2014-09-05 10:00:14.000000000 +0900
+++ src/libschwa-0.4.0/src/lib/schwa/io/array_reader.cc	2025-08-09 21:05:16.009130000 +0900
@@ -3,7 +3,7 @@
 
 #include <algorithm>
 #include <cstring>
-
+#include <stdio.h>
 
 namespace schwa {
 namespace io {
--- origsrc/libschwa-0.4.0/src/lib/schwa/io/paths.cc	2014-09-05 10:00:14.000000000 +0900
+++ src/libschwa-0.4.0/src/lib/schwa/io/paths.cc	2025-08-09 21:05:16.009130000 +0900
@@ -37,7 +37,7 @@ abspath_to_argv0(void) {
   }
   return pathbuf;
 }
-#elif defined(__linux)
+#elif defined(__linux) || defined(__CYGWIN__)
 std::string
 abspath_to_argv0(void) {
   char buf[PATH_MAX];
