--- origsrc/qd-2.3.14/fortran/Makefile.am	2012-03-14 08:35:00.000000000 +0900
+++ src/qd-2.3.14/fortran/Makefile.am	2013-10-14 12:24:58.276882900 +0900
@@ -14,9 +14,10 @@ QDMOD=qdmodule.$(module_ext)
 DDMOD=ddmodule.$(module_ext)
 endif
 
-lib_LTLIBRARIES = libqdmod.la libqd_f_main.la
+lib_LTLIBRARIES = libqdmod.la
 libqdmod_la_SOURCES = ddext.f ddmod.f qdext.f qdmod.f f_dd.cpp f_qd.cpp
-libqd_f_main_la_SOURCES = main.cpp
+libqdmod_la_LDFLAGS = -no-undefined
+libqdmod_la_LIBADD = -lgfortran $(top_builddir)/src/libqd.la
 ddmod.lo: $(DDEXT) ddext.lo
 qdmod.lo: ddmod.lo $(DDMOD) qdext.lo
 $(QDMOD): qdmod.lo $(DDMOD)
--- origsrc/qd-2.3.14/src/Makefile.am	2012-03-14 08:35:00.000000000 +0900
+++ src/qd-2.3.14/src/Makefile.am	2013-10-14 12:10:04.272976600 +0900
@@ -4,6 +4,7 @@ SRC = c_dd.cpp c_qd.cpp dd_real.cpp dd_c
 lib_LTLIBRARIES = libqd.la
 
 libqd_la_SOURCES = $(SRC)
+libqd_la_LDFLAGS = -no-undefined
 
 AM_CPPFLAGS = -I$(top_builddir) -I$(top_builddir)/include -I$(top_srcdir)/include
 
