--- origsrc/qd-2.3.24/fortran/Makefile.am	2023-11-03 23:39:34.000000000 +0900
+++ src/qd-2.3.24/fortran/Makefile.am	2023-11-07 05:59:09.882665400 +0900
@@ -14,11 +14,10 @@ QDMOD=qdmodule.$(module_ext)
 DDMOD=ddmodule.$(module_ext)
 endif
 
-lib_LTLIBRARIES = libqdmod.la libqd_f_main.la
+lib_LTLIBRARIES = libqdmod.la
 libqdmod_la_SOURCES = ddext.f ddmod.f qdext.f qdmod.f f_dd.cpp f_qd.cpp
 libqdmod_la_LIBADD = $(top_builddir)/src/libqd.la $(FCLIBS)
-libqd_f_main_la_SOURCES = main.cpp
-libqd_f_main_la_LIBADD = $(top_builddir)/src/libqd.la libqdmod.la $(FCLIBS)
+libqdmod_la_LDFLAGS = -no-undefined
 ddmod.lo: $(DDEXT) ddext.lo
 qdmod.lo: ddmod.lo $(DDMOD) qdext.lo
 $(QDMOD): qdmod.lo $(DDMOD)
--- origsrc/qd-2.3.24/src/Makefile.am	2023-11-03 23:39:34.000000000 +0900
+++ src/qd-2.3.24/src/Makefile.am	2023-11-07 05:59:09.898293500 +0900
@@ -4,6 +4,7 @@ SRC = c_dd.cpp c_qd.cpp dd_real.cpp dd_c
 lib_LTLIBRARIES = libqd.la
 
 libqd_la_SOURCES = $(SRC)
+libqd_la_LDFLAGS = -no-undefined
 
 AM_CPPFLAGS = -I$(top_builddir) -I$(top_builddir)/include -I$(top_srcdir)/include
 
--- origsrc/qd-2.3.24/tests/Makefile.am	2023-11-03 23:39:34.000000000 +0900
+++ src/qd-2.3.24/tests/Makefile.am	2023-11-07 05:59:09.898293500 +0900
@@ -16,7 +16,6 @@ AM_FFLAGS = $(AM_FCFLAGS)
 f_test_SOURCES = f_test.f
 f_test_LINK=$(CXXLINK)
 f_test_LDADD = $(top_builddir)/fortran/libqdmod.la \
-               $(top_builddir)/fortran/libqd_f_main.la \
                $(LDADD) $(top_builddir)/src/libqd.la $(FCLIBS)
 endif
 
--- origsrc/qd-2.3.24/tests/f_test.f	2023-11-03 23:39:34.000000000 +0900
+++ src/qd-2.3.24/tests/f_test.f	2023-11-07 05:59:09.898293500 +0900
@@ -28,3 +28,6 @@ subroutine f_main
   call f_fpu_fix_end (old_cw)
 end
 
+program main
+  call f_main
+end
