--- origsrc/mppp-0.17/CMakeLists.txt	2019-09-14 03:38:10.000000000 +0900
+++ src/mppp-0.17/CMakeLists.txt	2019-10-06 16:26:15.915672400 +0900
@@ -333,3 +333,10 @@ endif()
 if(MPPP_BUILD_BENCHMARKS)
     add_subdirectory(benchmark)
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/mppp.pc.in ${CMAKE_BINARY_DIR}/mppp.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/mppp.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/mppp-0.17/mppp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/mppp-0.17/mppp.pc.in	2019-10-06 16:25:14.220881100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mppp
+Description: Multiprecision for modern C++
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lmp++
+Cflags: -I${includedir}
