--- origsrc/liblxi-1.19/src/meson.build	2023-05-03 08:58:38.000000000 +0900
+++ src/liblxi-1.19/src/meson.build	2023-05-19 22:26:07.845035600 +0900
@@ -25,7 +25,7 @@ tirpc_incdir = []
 
 if host_machine.system() not in tirpc_excluded_systems
   tirpc_dep = dependency('libtirpc', required: true)
-  tirpc_incpath = join_paths(tirpc_dep.get_variable(pkgconfig: 'includedir'), 'tirpc')
+  tirpc_incpath = tirpc_dep.get_variable(pkgconfig: 'includedir')
   tirpc_incdir = include_directories(tirpc_incpath)
   liblxi_link_args += ['-Wl,-init,init']
 elif not meson.get_compiler('c').has_header('rpc/rpc.h')
@@ -43,7 +43,7 @@ liblxi_c_args = [
 liblxi = shared_library(
   'lxi',
   liblxi_sources,
-  dependencies: liblxi_deps,
+  dependencies: [liblxi_deps, tirpc_dep],
   install: true,
   c_args: liblxi_c_args,
   include_directories: tirpc_incdir,
