--- origsrc/gltools-193/include/GLBatch.h	2011-12-09 08:36:59.000000000 +0900
+++ src/gltools-193/include/GLBatch.h	2015-04-11 02:54:27.719348500 +0900
@@ -62,6 +62,9 @@ ANY WAY OUT OF THE USE OF THIS SOFTWARE,
 #include <glew.h>
 #endif
 
+#if defined(__CYGWIN__)
+#include <glew.h>
+#endif
 
 #include <math3d.h>
 #include <GLBatchBase.h>
--- origsrc/gltools-193/include/GLShaderManager.h	2011-12-09 08:36:59.000000000 +0900
+++ src/gltools-193/include/GLShaderManager.h	2015-04-11 02:53:44.427405100 +0900
@@ -63,6 +63,10 @@ ANY WAY OUT OF THE USE OF THIS SOFTWARE,
 #include <glew.h>
 #endif
 
+#if defined(__CYGWIN__)
+#include <glew.h>
+#endif
+
 //#include <vector>
 //using namespace std;
 
--- origsrc/gltools-193/src/GLTools.cpp	2011-12-09 08:36:59.000000000 +0900
+++ src/gltools-193/src/GLTools.cpp	2015-04-11 02:55:21.923410700 +0900
@@ -40,7 +40,7 @@ ANY WAY OUT OF THE USE OF THIS SOFTWARE,
 #include <assert.h>
 #include <stdarg.h>
 
-#ifdef linux
+#if defined(linux) || defined(__CYGWIN__) 
 #include <cstdlib> 
 #endif
 #ifdef __APPLE__
--- origsrc/gltools-193/src/Makefile.am	2011-12-09 08:36:59.000000000 +0900
+++ src/gltools-193/src/Makefile.am	2015-04-11 02:50:39.970103400 +0900
@@ -19,7 +19,9 @@ libgltools_la_CFLAGS = \
 	$(GL_CFLAGS) \
 	$(GLU_CFLAGS)
 
-libgltools_la_LDFLAGS = \
+libgltools_la_LDFLAGS = -no-undefined
+
+libgltools_la_LIBADD = \
 	$(GL_LIBS) \
 	$(GLU_LIBS)
 
