--- origsrc/gltools-193/include/GLBatch.h	2011-12-09 08:36:59.000000000 +0900
+++ src/gltools-193/include/GLBatch.h	2024-11-05 04:51:58.342420700 +0900
@@ -62,6 +62,9 @@ ANY WAY OUT OF THE USE OF THIS SOFTWARE,
 #include <glew.h>
 #endif
 
+#if defined(__CYGWIN__)
+#include <glew.h>
+#endif
 
 #include <math3d.h>
 #include <GLBatchBase.h>
--- origsrc/gltools-193/include/GLShaderManager.h	2011-12-09 08:36:59.000000000 +0900
+++ src/gltools-193/include/GLShaderManager.h	2024-11-05 04:51:58.342420700 +0900
@@ -63,6 +63,10 @@ ANY WAY OUT OF THE USE OF THIS SOFTWARE,
 #include <glew.h>
 #endif
 
+#if defined(__CYGWIN__)
+#include <glew.h>
+#endif
+
 //#include <vector>
 //using namespace std;
 
--- origsrc/gltools-193/src/GLTools.cpp	2011-12-09 08:36:59.000000000 +0900
+++ src/gltools-193/src/GLTools.cpp	2024-11-05 04:55:13.195536600 +0900
@@ -40,7 +40,7 @@ ANY WAY OUT OF THE USE OF THIS SOFTWARE,
 #include <assert.h>
 #include <stdarg.h>
 
-#ifdef linux
+#if defined(linux) || defined(__CYGWIN__) 
 #include <cstdlib> 
 #endif
 #ifdef __APPLE__
@@ -1057,7 +1057,7 @@ GLbyte* gltReadBMPBits(const char *szFil
 		{
 		free(pBitmapInfo);
 		fclose(pFile);
-		return false;
+		return NULL;
 		}
 
 	// Save the size and dimensions of the bitmap
@@ -1069,7 +1069,7 @@ GLbyte* gltReadBMPBits(const char *szFil
 	if(pBitmapInfo->header.bits != 24)
 		{
 		free(pBitmapInfo);
-		return false;
+		return NULL;
 		}
 
 	if(lBitSize == 0)
@@ -1252,7 +1252,7 @@ GLuint gltLoadShaderTripletWithAttribute
     if(testVal == GL_FALSE) {
         char infoLog[1024];
         glGetProgramInfoLog(hReturn, 1024, NULL, infoLog);
-        fprintf(stderr, infoLog);
+        fprintf(stderr, "%s", infoLog);
         goto failed;
     }
 
--- origsrc/gltools-193/src/Makefile.am	2011-12-09 08:36:59.000000000 +0900
+++ src/gltools-193/src/Makefile.am	2024-11-05 04:51:58.342420700 +0900
@@ -19,7 +19,9 @@ libgltools_la_CFLAGS = \
 	$(GL_CFLAGS) \
 	$(GLU_CFLAGS)
 
-libgltools_la_LDFLAGS = \
+libgltools_la_LDFLAGS = -no-undefined
+
+libgltools_la_LIBADD = \
 	$(GL_LIBS) \
 	$(GLU_LIBS)
 
