--- origsrc/Gikou-2.0.2/Makefile	2017-05-15 00:16:11.000000000 +0900
+++ src/Gikou-2.0.2/Makefile	2021-03-09 23:39:46.065031000 +0900
@@ -3,7 +3,8 @@
 #
 CXX       = g++
 CXXFLAGS  = -std=c++11 -Wall -Wextra -Wcast-qual -fno-exceptions -fno-rtti \
-            -pedantic -Wno-long-long -msse4.2 -D__STDC_CONSTANT_MACROS -fopenmp
+            -pedantic -Wno-long-long -msse4.2 -D__STDC_CONSTANT_MACROS -fopenmp \
+			-D_GNU_SOURCE
 INCLUDES  =
 LIBRARIES = -lpthread
 
