--- origsrc/nopoll-0.4.9/VERSION	2024-10-22 17:05:41.000000000 +0900
+++ src/nopoll-0.4.9/VERSION	2026-01-29 08:59:18.118881800 +0900
@@ -1 +1 @@
-0.4.8.b456
+0.4.9
--- origsrc/nopoll-0.4.9/src/nopoll_conn.c	2024-10-22 17:05:41.000000000 +0900
+++ src/nopoll-0.4.9/src/nopoll_conn.c	2026-01-28 09:54:57.600045000 +0900
@@ -208,7 +208,7 @@ nopoll_bool                 nopoll_conn_
 	/* no local variable here please */
 
 	if ((NULL != options) && (NULL != options->_interface)) {
-#if defined(NOPOLL_OS_WIN32) || defined(NOPOLL_OS_WIN64)
+#if defined(NOPOLL_OS_WIN32) || defined(NOPOLL_OS_WIN64) || defined(__CYGWIN__)
 		/* Windows still not supported: send us a patch! */ 
 		return nopoll_false;
 #elif defined(__APPLE__) || defined(__FreeBSD__)
