--- origsrc/rudeconfig-5.0.5/Makefile.am	2007-01-20 11:18:58.000000000 +0900
+++ src/rudeconfig-5.0.5/Makefile.am	2010-12-29 22:12:27.543282600 +0900
@@ -54,7 +54,7 @@ librudeconfig_la_SOURCES = src/AbstractD
 	src/RealOrganiser.h \
 	src/Writer.h
 
-librudeconfig_la_LDFLAGS = -version-info 5:1:2
+librudeconfig_la_LDFLAGS = -version-info 5:1:2 -no-undefined
 
 pkginclude_HEADERS = src/config.h
 pkgincludedir = $(includedir)/rude
--- origsrc/rudeconfig-5.0.5/configure.ac	2007-02-02 13:01:19.000000000 +0900
+++ src/rudeconfig-5.0.5/configure.ac	2010-12-29 22:12:05.531631400 +0900
@@ -28,7 +28,7 @@ AM_INIT_AUTOMAKE(rudeconfig, 5.0.5)
 # Checks for programs.
 AC_PROG_CXX
 AC_PROG_CC
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 # Checks for libraries.
 
