--- origsrc/rudeconfig-5.0.5/Makefile.am	2007-01-20 11:18:58.000000000 +0900
+++ src/rudeconfig-5.0.5/Makefile.am	2015-04-25 07:36:41.830739300 +0900
@@ -54,7 +54,7 @@ librudeconfig_la_SOURCES = src/AbstractD
 	src/RealOrganiser.h \
 	src/Writer.h
 
-librudeconfig_la_LDFLAGS = -version-info 5:1:2
+librudeconfig_la_LDFLAGS = -no-undefined -version-info 5:1:2
 
 pkginclude_HEADERS = src/config.h
 pkgincludedir = $(includedir)/rude
@@ -65,4 +65,5 @@ EXTRA_DIST = $(man_MANS)
 
 AUTOMAKE_OPTIONS = gnits dist-bzip2 dist-shar dist-zip
 
-
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = rudeconfig.pc
--- origsrc/rudeconfig-5.0.5/configure.ac	2007-02-02 13:01:19.000000000 +0900
+++ src/rudeconfig-5.0.5/configure.ac	2015-04-25 07:36:32.942191900 +0900
@@ -42,6 +42,4 @@ AC_TYPE_SIZE_T
 
 # Checks for library functions.
 AC_CHECK_FUNCS([strrchr])
-AC_OUTPUT(Makefile )
-
-
+AC_OUTPUT(rudeconfig.pc Makefile )
--- origsrc/rudeconfig-5.0.5/rudeconfig.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/rudeconfig-5.0.5/rudeconfig.pc.in	2015-04-25 07:36:53.635316000 +0900
@@ -0,0 +1,5 @@
+Name: rudeconfig
+Description: Library (C++ API) for reading and writing configuration/.ini files
+Version: @VERSION@
+Libs: -L@libdir@ -lrudeconfig
+Cflags: -I@includedir@
