--- origsrc/wolfssh-1.4.20-stable/Makefile.am	2025-02-21 03:57:45.000000000 +0900
+++ src/wolfssh-1.4.20-stable/Makefile.am	2025-09-06 07:15:15.850697000 +0900
@@ -98,4 +98,5 @@ merge-clean:
 	@find ./ | $(GREP) \.BASE | xargs rm -f
 	@find ./ | $(GREP) \~$$ | xargs rm -f
 
-
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/wolfssh-1.4.20-stable/apps/wolfsshd/auth.c	2025-02-21 03:57:45.000000000 +0900
+++ src/wolfssh-1.4.20-stable/apps/wolfsshd/auth.c	2025-09-06 07:15:15.855717600 +0900
@@ -74,7 +74,7 @@
 #include <errno.h>
 #endif
 
-#if !defined(_WIN32) && !(defined(__OSX__) || defined(__APPLE__))
+#if !defined(_WIN32) && !(defined(__OSX__) || defined(__APPLE__)) && !defined(__CYGWIN__)
 #include <shadow.h>
 #define HAVE_SHADOW
 #endif
--- origsrc/wolfssh-1.4.20-stable/configure.ac	2025-02-21 03:57:45.000000000 +0900
+++ src/wolfssh-1.4.20-stable/configure.ac	2025-09-06 07:15:15.857743500 +0900
@@ -288,7 +288,7 @@ AC_SUBST([AM_CFLAGS])
 AC_SUBST([AM_LDFLAGS])
 
 # FINAL
-AC_CONFIG_FILES([Makefile wolfssh/version.h])
+AC_CONFIG_FILES([wolfssh.pc Makefile wolfssh/version.h])
 
 AX_CREATE_GENERIC_CONFIG
 
--- origsrc/wolfssh-1.4.20-stable/wolfssh.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/wolfssh-1.4.20-stable/wolfssh.pc.in	2025-09-06 07:15:15.861776400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: wolfssh
+Description: wolfSSL's Embeddable SSH Server
+Version: @VERSION@
+Libs: -L${libdir} -lwolfssh @LIBS@
+Cflags: -I${includedir}
