--- origsrc/wolfssh-1.4.7-stable/Makefile.am	2021-07-24 01:40:05.000000000 +0900
+++ src/wolfssh-1.4.7-stable/Makefile.am	2021-09-15 11:18:05.958488900 +0900
@@ -97,4 +97,5 @@ merge-clean:
 	@find ./ | $(GREP) \.BASE | xargs rm -f
 	@find ./ | $(GREP) \~$$ | xargs rm -f
 
-
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/wolfssh-1.4.7-stable/configure.ac	2021-07-24 01:40:05.000000000 +0900
+++ src/wolfssh-1.4.7-stable/configure.ac	2021-09-15 11:18:05.965487600 +0900
@@ -183,7 +183,7 @@ AC_SUBST([AM_CFLAGS])
 AC_SUBST([AM_LDFLAGS])
 
 # FINAL
-AC_CONFIG_FILES([Makefile wolfssh/version.h])
+AC_CONFIG_FILES([wolfssh.pc Makefile wolfssh/version.h])
 
 AX_CREATE_GENERIC_CONFIG
 
File origsrc/wolfssh-1.4.7-stable/wolfssh/version.h is a regular file while file src/wolfssh-1.4.7-stable/wolfssh/version.h is a symbolic link
--- origsrc/wolfssh-1.4.7-stable/wolfssh.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/wolfssh-1.4.7-stable/wolfssh.pc.in	2021-09-15 11:18:05.978934300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: wolfssh
+Description: wolfSSL's Embeddable SSH Server
+Version: @VERSION@
+Libs: -L${libdir} -lwolfssh
+Cflags: -I${includedir}
