--- origsrc/date-3.0.1/CMakeLists.txt	2021-04-17 10:26:33.000000000 +0900
+++ src/date-3.0.1/CMakeLists.txt	2022-05-30 17:04:38.656362800 +0900
@@ -7,7 +7,7 @@
      include( FetchContent )
      FetchContent_Declare( date_src
        GIT_REPOSITORY https://github.com/HowardHinnant/date.git
-       GIT_TAG        v3.0.0  # adjust tag/branch/commit as needed
+       GIT_TAG        v3.0.1  # adjust tag/branch/commit as needed
      )
      FetchContent_MakeAvailable(date_src)
      ...
@@ -17,7 +17,7 @@
 
 cmake_minimum_required( VERSION 3.7 )
 
-project( date VERSION 3.0.0 )
+project( date VERSION 3.0.1 )
 set(ABI_VERSION 3) # used as SOVERSION, increment when ABI changes
 
 include( GNUInstallDirs )
@@ -275,3 +275,12 @@ if( ENABLE_DATE_TESTING )
         endif( )
     endforeach( )
 endif( )
+
+if (CYGWIN)
+  set(prefix ${CMAKE_INSTALL_PREFIX})
+  set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+  set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+  set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+  configure_file(${CMAKE_SOURCE_DIR}/libdate-tz.pc.in ${CMAKE_BINARY_DIR}/libdate-tz.pc @ONLY)
+  install(FILES ${CMAKE_BINARY_DIR}/libdate-tz.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+endif()
--- origsrc/date-3.0.1/libdate-tz.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/date-3.0.1/libdate-tz.pc.in	2022-05-30 17:04:54.399810000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libdate-tz
+Description: C++ library for date and time operations based on <chrono>
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -ldate-tz
+Cflags: -I${includedir}
