--- origsrc/libibcommon-1.2.0/Makefile.am	2008-07-27 00:38:35.000000000 +0900
+++ src/libibcommon-1.2.0/Makefile.am	2010-12-19 18:59:21.125658000 +0900
@@ -15,7 +15,7 @@ endif
 
 libibcommon_la_SOURCES = src/stack.c src/sysfs.c src/util.c src/time.c src/hash.c
 libibcommon_la_LDFLAGS = -version-info $(ibcommon_api_version) \
-    -export-dynamic $(libibcommon_version_script)
+    -export-dynamic $(libibcommon_version_script) -no-undefined
 libibcommon_la_DEPENDENCIES = $(srcdir)/src/libibcommon.map
 
 libibcommonincludedir = $(includedir)/infiniband
--- origsrc/libibcommon-1.2.0/configure.in	2008-12-14 02:44:36.000000000 +0900
+++ src/libibcommon-1.2.0/configure.in	2010-12-19 18:58:54.973739100 +0900
@@ -23,7 +23,7 @@ AC_PROG_CPP
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 dnl Checks for header files.
 AC_HEADER_STDC
