--- origsrc/libibcommon-1.2.0/Makefile.am	2008-07-27 00:38:35.000000000 +0900
+++ src/libibcommon-1.2.0/Makefile.am	2014-07-21 11:46:48.380309800 +0900
@@ -15,7 +15,7 @@ endif
 
 libibcommon_la_SOURCES = src/stack.c src/sysfs.c src/util.c src/time.c src/hash.c
 libibcommon_la_LDFLAGS = -version-info $(ibcommon_api_version) \
-    -export-dynamic $(libibcommon_version_script)
+    -export-dynamic $(libibcommon_version_script) -no-undefined
 libibcommon_la_DEPENDENCIES = $(srcdir)/src/libibcommon.map
 
 libibcommonincludedir = $(includedir)/infiniband
