--- origsrc/libzim-9.3.0/meson.build	2025-04-17 18:15:37.000000000 +0900
+++ src/libzim-9.3.0/meson.build	2025-04-23 15:05:29.890999000 +0900
@@ -78,8 +78,10 @@ endif
 
 if xapian_dep.found()
     icu_dep = dependency('icu-i18n', static:static_linkage)
+    icu_uc_dep = dependency('icu-uc', static:static_linkage)
 else
     icu_dep = dependency('icu-i18n', required:false, static:static_linkage)
+    icu_uc_dep = dependency('icu-uc', required:false, static:static_linkage)
 endif
 
 gtest_dep = dependency('gtest', version: '>=1.10.0', main:true, fallback:['gtest', 'gtest_main_dep'], required:false)
--- origsrc/libzim-9.3.0/src/meson.build	2025-04-17 18:15:37.000000000 +0900
+++ src/libzim-9.3.0/src/meson.build	2025-04-23 15:05:36.443606700 +0900
@@ -65,7 +65,7 @@ endif
 if xapian_dep.found()
     sources += xapian_sources
     sources += lib_resources
-    deps += [xapian_dep, icu_dep]
+    deps += [xapian_dep, icu_dep, icu_uc_dep]
 endif
 
 libzim = library('zim',
