--- origsrc/libzim-8.1.0/src/file_reader.cpp	2022-12-01 02:08:43.000000000 +0900
+++ src/libzim-8.1.0/src/file_reader.cpp	2022-12-02 10:36:36.288826300 +0900
@@ -134,7 +134,7 @@ class MMapException : std::exception {};
 char*
 mmapReadOnly(int fd, offset_type offset, size_type size)
 {
-#if defined(__APPLE__) || defined(__OpenBSD__)
+#if defined(__APPLE__) || defined(__OpenBSD__) || defined(__CYGWIN__)
   const auto MAP_FLAGS = MAP_PRIVATE;
 #elif defined(__FreeBSD__)
   const auto MAP_FLAGS = MAP_PRIVATE|MAP_PREFAULT_READ;
--- origsrc/libzim-8.1.0/src/fs_unix.cpp	2022-12-01 02:08:43.000000000 +0900
+++ src/libzim-8.1.0/src/fs_unix.cpp	2022-12-02 10:36:36.308865100 +0900
@@ -37,7 +37,7 @@ namespace unix {
 
 zsize_t FD::readAt(char* dest, zsize_t size, offset_t offset) const
 {
-#if defined(__APPLE__) || defined(__OpenBSD__) || defined(__FreeBSD__)
+#if defined(__APPLE__) || defined(__OpenBSD__) || defined(__FreeBSD__) || defined(__CYGWIN__)
 # define PREAD pread
 #else
 # define PREAD pread64
