--- origsrc/libzim-8.0.1/src/file_reader.cpp	2022-09-07 22:43:40.000000000 +0900
+++ src/libzim-8.0.1/src/file_reader.cpp	2022-10-29 19:53:23.233202400 +0900
@@ -134,7 +134,7 @@ class MMapException : std::exception {};
 char*
 mmapReadOnly(int fd, offset_type offset, size_type size)
 {
-#if defined(__APPLE__) || defined(__OpenBSD__)
+#if defined(__APPLE__) || defined(__OpenBSD__) || defined(__CYGWIN__)
   const auto MAP_FLAGS = MAP_PRIVATE;
 #elif defined(__FreeBSD__)
   const auto MAP_FLAGS = MAP_PRIVATE|MAP_PREFAULT_READ;
--- origsrc/libzim-8.0.1/src/fs_unix.cpp	2022-09-07 22:43:40.000000000 +0900
+++ src/libzim-8.0.1/src/fs_unix.cpp	2022-10-29 20:12:31.624870500 +0900
@@ -37,7 +37,7 @@ namespace unix {
 
 zsize_t FD::readAt(char* dest, zsize_t size, offset_t offset) const
 {
-#if defined(__APPLE__) || defined(__OpenBSD__) || defined(__FreeBSD__)
+#if defined(__APPLE__) || defined(__OpenBSD__) || defined(__FreeBSD__) || defined(__CYGWIN__)
 # define PREAD pread
 #else
 # define PREAD pread64
