--- origsrc/libzim-6.3.0/src/debug.h	2020-11-17 21:00:47.000000000 +0900
+++ src/libzim-6.3.0/src/debug.h	2020-12-01 14:29:08.034674000 +0900
@@ -28,7 +28,7 @@
 # define ASSERT(left, operator, right) (void(0))
 #else
 
-#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__)
+#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__) && !defined(__CYGWIN__)
 #include <execinfo.h>
 #endif
 
@@ -40,7 +40,7 @@ void _on_assert_fail(const char* vara, c
       vara << "[" << a << "] " << op << " " << varb << "[" << b << "]";
   std::cerr << ss.str() << std::endl;
 
-#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__)
+#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__) && !defined(__CYGWIN__)
   void *callstack[64];
   size_t size;
   size = backtrace(callstack, 64);
--- origsrc/libzim-6.3.0/src/file_reader.cpp	2020-11-17 21:00:47.000000000 +0900
+++ src/libzim-6.3.0/src/file_reader.cpp	2020-12-01 14:31:22.088008300 +0900
@@ -134,7 +134,7 @@ class MMapException : std::exception {};
 char*
 mmapReadOnly(int fd, offset_type offset, size_type size)
 {
-#if defined(__APPLE__) || defined(__OpenBSD__)
+#if defined(__APPLE__) || defined(__OpenBSD__) || defined(__CYGWIN__)
   const auto MAP_FLAGS = MAP_PRIVATE;
 #elif defined(__FreeBSD__)
   const auto MAP_FLAGS = MAP_PRIVATE|MAP_PREFAULT_READ;
--- origsrc/libzim-6.3.0/src/fs_unix.cpp	2020-11-17 21:00:47.000000000 +0900
+++ src/libzim-6.3.0/src/fs_unix.cpp	2020-12-01 14:32:35.539010800 +0900
@@ -34,7 +34,7 @@ namespace unix {
 
 zsize_t FD::readAt(char* dest, zsize_t size, offset_t offset) const
 {
-#if defined(__APPLE__) || defined(__OpenBSD__) || defined(__FreeBSD__)
+#if defined(__APPLE__) || defined(__OpenBSD__) || defined(__FreeBSD__) || defined(__CYGWIN__)
 # define PREAD pread
 #else
 # define PREAD pread64
