--- origsrc/libzim-7.2.2/src/debug.h	2022-05-18 05:48:02.000000000 +0900
+++ src/libzim-7.2.2/src/debug.h	2022-07-28 15:44:08.188519400 +0900
@@ -29,7 +29,7 @@
 # define ASSERT(left, operator, right) (void(0))
 #else
 
-#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__) && !defined(__EMSCRIPTEN__)
+#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__) && !defined(__EMSCRIPTEN__) && !defined(__CYGWIN__)
 #include <execinfo.h>
 #endif
 
--- origsrc/libzim-7.2.2/src/file_reader.cpp	2022-05-18 05:48:02.000000000 +0900
+++ src/libzim-7.2.2/src/file_reader.cpp	2022-07-28 15:44:08.198308700 +0900
@@ -134,7 +134,7 @@ class MMapException : std::exception {};
 char*
 mmapReadOnly(int fd, offset_type offset, size_type size)
 {
-#if defined(__APPLE__) || defined(__OpenBSD__)
+#if defined(__APPLE__) || defined(__OpenBSD__) || defined(__CYGWIN__)
   const auto MAP_FLAGS = MAP_PRIVATE;
 #elif defined(__FreeBSD__)
   const auto MAP_FLAGS = MAP_PRIVATE|MAP_PREFAULT_READ;
--- origsrc/libzim-7.2.2/src/fs_unix.cpp	2022-05-18 05:48:02.000000000 +0900
+++ src/libzim-7.2.2/src/fs_unix.cpp	2022-07-28 15:44:08.204282100 +0900
@@ -37,7 +37,7 @@ namespace unix {
 
 zsize_t FD::readAt(char* dest, zsize_t size, offset_t offset) const
 {
-#if defined(__APPLE__) || defined(__OpenBSD__) || defined(__FreeBSD__)
+#if defined(__APPLE__) || defined(__OpenBSD__) || defined(__FreeBSD__) || defined(__CYGWIN__)
 # define PREAD pread
 #else
 # define PREAD pread64
