--- origsrc/libipathverbs-1.2.1/Makefile.am	2014-02-04 23:23:58.000000000 +0900
+++ src/libipathverbs-1.2.1/Makefile.am	2014-07-21 14:47:36.664144600 +0900
@@ -43,14 +43,14 @@ if HAVE_IBV_DEVICE_LIBRARY_EXTENSION
     lib_LTLIBRARIES = src/libipathverbs.la
     src_libipathverbs_la_SOURCES = $(IPATHVERBS_SOURCES)
     src_libipathverbs_la_LDFLAGS = -avoid-version -release @IBV_DEVICE_LIBRARY_EXTENSION@ \
-        $(ipathverbs_version_script)
+        $(ipathverbs_version_script) -no-undefined
     ipathverbsconfdir = $(sysconfdir)/libibverbs.d
     ipathverbsconf_DATA = ipath.driver
 else
     ipathverbslibdir = $(libdir)/infiniband
     ipathverbslib_LTLIBRARIES = src/ipathverbs.la
     src_ipathverbs_la_SOURCES = $(IPATHVERBS_SOURCES)
-    src_ipathverbs_la_LDFLAGS = -avoid-version -module $(ipathverbs_version_script)
+    src_ipathverbs_la_LDFLAGS = -no-undefined -avoid-version -module $(ipathverbs_version_script)
 endif
 
 EXTRA_DIST = src/ipathverbs.h \
