--- origsrc/libipathverbs-1.3/Makefile.am	2014-07-24 05:52:55.000000000 +0900
+++ src/libipathverbs-1.3/Makefile.am	2024-12-17 12:18:36.849420500 +0900
@@ -42,7 +42,7 @@ IPATHVERBS_SOURCES = src/ipathverbs.c sr
 if HAVE_IBV_DEVICE_LIBRARY_EXTENSION
     lib_LTLIBRARIES = src/libipathverbs.la
     src_libipathverbs_la_SOURCES = $(IPATHVERBS_SOURCES)
-    src_libipathverbs_la_LDFLAGS = -avoid-version -release @IBV_DEVICE_LIBRARY_EXTENSION@ \
+    src_libipathverbs_la_LDFLAGS = -no-undefined -avoid-version -release @IBV_DEVICE_LIBRARY_EXTENSION@ \
         $(ipathverbs_version_script)
     ipathverbsconfdir = $(sysconfdir)/libibverbs.d
     ipathverbsconf_DATA = ipath.driver
@@ -50,7 +50,7 @@ else
     ipathverbslibdir = $(libdir)/infiniband
     ipathverbslib_LTLIBRARIES = src/ipathverbs.la
     src_ipathverbs_la_SOURCES = $(IPATHVERBS_SOURCES)
-    src_ipathverbs_la_LDFLAGS = -avoid-version -module $(ipathverbs_version_script)
+    src_ipathverbs_la_LDFLAGS = -no-undefined -avoid-version -module $(ipathverbs_version_script)
 endif
 ipathverbsmodprobedir = $(sysconfdir)/modprobe.d
 ipathverbsmodprobe_DATA = truescale.conf
