--- origsrc/cpputest-4.0/src/CppUTest/CMakeLists.txt	2020-05-27 19:59:41.000000000 +0900
+++ src/cpputest-4.0/src/CppUTest/CMakeLists.txt	2026-05-25 15:45:13.449221200 +0900
@@ -68,11 +68,16 @@ set(CppUTest_headers
 set_target_properties(CppUTest PROPERTIES
     PUBLIC_HEADER "${CppUTest_headers}")
 
+if (DEFINED SOVERSION)
+    set_target_properties(CppUTest PROPERTIES SOVERSION ${SOVERSION})
+endif()
+
 if (WIN32)
     target_link_libraries(CppUTest winmm.lib)
 endif (WIN32)
 install(TARGETS CppUTest
     EXPORT CppUTestTargets
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/CppUTest"
--- origsrc/cpputest-4.0/src/CppUTestExt/CMakeLists.txt	2026-05-25 13:05:40.221630300 +0900
+++ src/cpputest-4.0/src/CppUTestExt/CMakeLists.txt	2026-05-25 15:45:07.140868600 +0900
@@ -41,7 +41,7 @@ set(CppUTestExt_headers
 )
 
 add_library(CppUTestExt ${CppUTestExt_src} ${CppUTestExt_headers})
-target_link_libraries(CppUTestExt ${CPPUNIT_EXTERNAL_LIBRARIES})
+target_link_libraries(CppUTestExt CppUTest ${CPPUNIT_EXTERNAL_LIBRARIES})
 
 #[[Arrange for the include directory to be added to the include paths of any CMake target depending on CppUTestExt.]]
 target_include_directories(CppUTestExt
@@ -52,8 +52,14 @@ target_include_directories(CppUTestExt
 
 set_target_properties(CppUTestExt PROPERTIES
     PUBLIC_HEADER "${CppUTestExt_headers}")
+
+if (DEFINED SOVERSION)
+    set_target_properties(CppUTestExt PROPERTIES SOVERSION ${SOVERSION})
+endif() 
+
 install(TARGETS CppUTestExt
     EXPORT CppUTestTargets
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/CppUTestExt"
