--- origsrc/libscrypt-1.20/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libscrypt-1.20/Makefile.am	2015-04-17 13:35:29.628652200 +0900
@@ -0,0 +1,21 @@
+
+AM_CPPFLAGS = -D_FORTIFY_SOURCE=2
+AM_CFLAGS = -O2 -Wall -g -fstack-protector
+
+check_PROGRAMS = reference
+TESTS = reference
+reference_SOURCES = main.c
+reference_LDADD = libscrypt.la
+
+include_HEADERS = libscrypt.h
+
+lib_LTLIBRARIES = libscrypt.la
+libscrypt_la_SOURCES = \
+	crypto_scrypt-nosse.c sha256.c crypto_scrypt-hexconvert.c \
+	crypto-mcf.c b64.c crypto-scrypt-saltgen.c crypto_scrypt-check.c \
+	crypto_scrypt-hash.c slowequals.c
+libscrypt_la_LDFLAGS = -no-undefined \
+	-Wl,--version-script=$(top_srcdir)/libscrypt.version
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = libscrypt.pc
--- origsrc/libscrypt-1.20/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/libscrypt-1.20/configure.ac	2015-04-17 13:30:22.225853300 +0900
@@ -0,0 +1,35 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([libscrypt], [1.20])
+AC_CONFIG_SRCDIR([crypto-mcf.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_LN_S
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([float.h stdint.h stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_C_INLINE
+AC_TYPE_SIZE_T
+AC_TYPE_UINT16_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+AC_TYPE_UINT8_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_MMAP
+AC_CHECK_FUNCS([memset modf munmap strtoul])
+
+AC_CONFIG_FILES([Makefile libscrypt.pc])
+AC_OUTPUT
--- origsrc/libscrypt-1.20/libscrypt.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libscrypt-1.20/libscrypt.pc.in	2015-04-17 13:30:16.029370300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libscrypt
+Description: Replacement for bcrypt
+Version: @VERSION@
+Libs: -L${libdir} -lscrypt
+Cflags: -I${includedir}
--- origsrc/libscrypt-1.20/libscrypt.version	2014-12-01 19:39:02.000000000 +0900
+++ src/libscrypt-1.20/libscrypt.version	2015-04-17 13:40:31.213787100 +0900
@@ -4,5 +4,7 @@ libscrypt_hash;
 libscrypt_mcf; 
 libscrypt_salt_gen; 
 libscrypt_scrypt;
+libscrypt_hexconvert;
+libscrypt_b64_encode;
 	local: *;
 };
