--- origsrc/Chipmunk2D-Chipmunk-6.2.1/Demo/CMakeLists.txt	2013-10-16 11:16:23.000000000 +0900
+++ src/Chipmunk2D-Chipmunk-6.2.1/Demo/CMakeLists.txt	2014-11-08 00:10:31.598962100 +0900
@@ -1,13 +1,5 @@
 find_package(OpenGL REQUIRED)
-
-# Quick and dirty package searching.
-# Will probably only work on Unixes.
-# TODO need to fix this up eventually.
-FIND_LIBRARY(GLFW_LIBRARIES glfw /usr/lib)
-FIND_LIBRARY(GLEW_LIBRARIES GLEW /usr/lib)
-FIND_PATH(GLFW_INCLUDE_DIR GL/glfw.h /usr/include/GL)
-FIND_PATH(GLEW_INCLUDE_DIR GL/glew.h /usr/include/GL)
-
+find_package(glfw REQUIRED)
 
 set(chipmunk_demos_include_dirs
   ${chipmunk_SOURCE_DIR}/include/chipmunk
@@ -17,7 +9,7 @@ set(chipmunk_demos_include_dirs
 )
 
 set(chipmunk_demos_libraries
-  chipmunk_static
+  chipmunk
   ${GLFW_LIBRARIES}
   ${GLEW_LIBRARIES}
   ${OPENGL_LIBRARIES}
--- origsrc/Chipmunk2D-Chipmunk-6.2.1/src/CMakeLists.txt	2013-10-16 11:16:23.000000000 +0900
+++ src/Chipmunk2D-Chipmunk-6.2.1/src/CMakeLists.txt	2014-11-07 11:41:13.299976500 +0900
@@ -16,13 +16,13 @@ if(BUILD_SHARED)
   # set the lib's version number
   # But avoid on Android because symlinks to version numbered .so's don't work with Android's Java-side loadLibrary.
   if(NOT ANDROID)
-	  set_target_properties(chipmunk PROPERTIES VERSION 6.2.1)
+	  set_target_properties(chipmunk PROPERTIES VERSION 6.2.1 SOVERSION 6.2.1)
   endif(NOT ANDROID)
   if(ANDROID)
 	  # need to explicitly link to the math library because the CMake/Android toolchains may not do it automatically
 	  target_link_libraries(chipmunk m)
   endif(ANDROID)
-  install(TARGETS chipmunk RUNTIME DESTINATION lib LIBRARY DESTINATION lib)
+  install(TARGETS chipmunk RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 endif(BUILD_SHARED)
 
 if(BUILD_STATIC)
