--- origsrc/Chipmunk2D-Chipmunk-7.0.2/CMakeLists.txt	2017-08-29 00:45:32.000000000 +0900
+++ src/Chipmunk2D-Chipmunk-7.0.2/CMakeLists.txt	2018-03-27 11:14:14.040000000 +0900
@@ -70,3 +70,10 @@ add_subdirectory(src)
 if(BUILD_DEMOS)
   add_subdirectory(demo)
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/chipmunk.pc.in ${CMAKE_BINARY_DIR}/chipmunk.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/chipmunk.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/Chipmunk2D-Chipmunk-7.0.2/chipmunk.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/Chipmunk2D-Chipmunk-7.0.2/chipmunk.pc.in	2018-03-27 11:14:04.611000000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: chipmunk
+Description: Fast and lightweight 2D game physics library
+Version: @VERSION@
+Libs: -L${libdir} -lchipmunk
+Cflags: -I${includedir}/chipmunk
--- origsrc/Chipmunk2D-Chipmunk-7.0.2/demo/CMakeLists.txt	2017-08-29 00:45:32.000000000 +0900
+++ src/Chipmunk2D-Chipmunk-7.0.2/demo/CMakeLists.txt	2018-03-27 11:10:15.392000000 +0900
@@ -7,7 +7,9 @@ if(APPLE)
 	set(GLFW_INCLUDE_DIR ../xcode/libglfw/include)
 	set(chipmunk_demos_library_dirs ../xcode/libglfw/lib ../xcode/libGLEW/lib)
 	set(GLEW_LIBRARIES GLEW)
-	set(GLFW_LIBRARIES glfw)
+  set(GLFW_LIBRARIES glfw)
+elseif(CYGWIN)
+  find_package(glfw REQUIRED)
 else(APPLE)
 	FIND_PATH(GLFW_INCLUDE_DIR GL/glfw.h /usr/include/GL)
 	FIND_PATH(GLEW_INCLUDE_DIR GL/glew.h /usr/include/GL)
@@ -23,7 +25,7 @@ set(chipmunk_demos_include_dirs
 )
 
 set(chipmunk_demos_libraries
-  chipmunk_static
+  chipmunk
   ${GLFW_LIBRARIES}
   ${GLEW_LIBRARIES}
   ${OPENGL_LIBRARIES}
--- origsrc/Chipmunk2D-Chipmunk-7.0.2/src/cpHastySpace.c	2017-08-29 00:45:32.000000000 +0900
+++ src/Chipmunk2D-Chipmunk-7.0.2/src/cpHastySpace.c	2018-03-27 11:11:21.688999800 +0900
@@ -8,7 +8,9 @@
 
 //#include <sys/param.h >
 #ifndef _WIN32
+#if !defined(__CYGWIN__)
 #include <sys/sysctl.h>
+#endif
 #include <pthread.h>
 #else
 #ifndef WIN32_LEAN_AND_MEAN
