--- origsrc/Chipmunk2D/src/CMakeLists.txt	2013-07-07 22:38:19.000000000 +0900
+++ src/Chipmunk2D/src/CMakeLists.txt	2013-07-07 22:39:31.125363500 +0900
@@ -14,8 +14,8 @@ if(BUILD_SHARED)
     set_target_properties(chipmunk PROPERTIES LINKER_LANGUAGE CXX)
   endif(MSVC)
   # set the lib's version number
-  set_target_properties(chipmunk PROPERTIES VERSION 6.1.5)
-  install(TARGETS chipmunk RUNTIME DESTINATION lib LIBRARY DESTINATION lib)
+  set_target_properties(chipmunk PROPERTIES VERSION 6.1.5 SOVERSION 6.1.5)
+  install(TARGETS chipmunk RUNTIME DESTINATION bin ARCHIVE DESTINATION lib LIBRARY DESTINATION lib)
 endif(BUILD_SHARED)
 
 if(BUILD_STATIC)
