--- origsrc/gxml-0.20.4/gxml/meson.build	2024-04-17 02:27:12.000000000 +0900
+++ src/gxml-0.20.4/gxml/meson.build	2025-12-11 09:53:57.324351100 +0900
@@ -129,8 +129,8 @@ LT_AGE='0'
 LT_VERSION='2'
 libgxml = library(VERSIONED_PROJECT_NAME,
 	valasources+sources+configvapi,
-	version : LT_VERSION,
-	soversion : LT_VERSION+'.'+LT_AGE+'.'+LT_REVISION,
+	soversion : LT_VERSION,
+	version : LT_VERSION+'.'+LT_AGE+'.'+LT_REVISION,
 	darwin_versions : [LT_CURRENT.to_int() + 1, '@0@.@1@'.format(LT_CURRENT.to_int() + 1, LT_REVISION)],
 	vala_header : PROJECT_NAME+'.h',
 	vala_vapi : VAPI_NAME,
--- origsrc/gxml-0.20.4/meson.build	2024-04-17 02:27:12.000000000 +0900
+++ src/gxml-0.20.4/meson.build	2025-12-11 09:41:40.584272100 +0900
@@ -10,7 +10,7 @@ VERSIONED_CAMEL_CASE_NAME = CAMEL_CASE_N
 libgxml_deps = [
 	dependency('libxml-2.0', version: '>=2.9.13'),
 	dependency('gio-2.0', version: '>=2.72'),
-	dependency('gee-0.8', version:'>=0.20.5')
+	dependency('gee-0.8', version:'>=0.20.1')
 	]
 
 VALADEPS = 'gee-0.8\ngio-2.0\nlibxml-2.0'
