--- origsrc/uacme-1.7.6/Makefile.am	2024-12-30 01:27:26.000000000 +0900
+++ src/uacme-1.7.6/Makefile.am	2025-09-02 21:31:44.810062800 +0900
@@ -80,6 +80,7 @@ ualpn.1: ualpn.1.txt $(top_srcdir)/.vers
 	    -a runstatedir="${runstatedir}" $<
 
 docs/ualpn.html: ualpn.1.txt $(top_srcdir)/.version
+	$(MKDIR_P) docs
 	$(AM_V_GEN)$(ASCIIDOC) -d manpage -b html5 -o $@ \
 	    -a revision=$(VERSION) \
 	    -a sysconfdir="${sysconfdir}" \
@@ -93,6 +94,7 @@ uacme.1: uacme.1.txt $(top_srcdir)/.vers
 	    -a runstatedir="${runstatedir}" $<
 
 docs/uacme.html: uacme.1.txt $(top_srcdir)/.version
+	$(MKDIR_P) docs
 	$(AM_V_GEN)$(ASCIIDOC) -d manpage -b html5 -o $@ \
 	    -a revision=$(VERSION) \
 	    -a sysconfdir="${sysconfdir}" \
