--- origsrc/uacme-1.7/Makefile.am	2021-01-18 00:58:32.000000000 +0900
+++ src/uacme-1.7/Makefile.am	2021-05-16 00:30:01.006265300 +0900
@@ -80,6 +80,7 @@ ualpn.1: ualpn.1.txt $(top_srcdir)/.vers
 	    -a runstatedir="${runstatedir}" $<
 
 docs/ualpn.html: ualpn.1.txt $(top_srcdir)/.version
+	$(MKDIR_P) docs
 	$(AM_V_GEN)$(ASCIIDOC) -d manpage -b html5 -o $@ \
 	    -a revision=$(VERSION) \
 	    -a sysconfdir="${sysconfdir}" \
@@ -93,6 +94,7 @@ uacme.1: uacme.1.txt $(top_srcdir)/.vers
 	    -a runstatedir="${runstatedir}" $<
 
 docs/uacme.html: uacme.1.txt $(top_srcdir)/.version
+	$(MKDIR_P) docs
 	$(AM_V_GEN)$(ASCIIDOC) -d manpage -b html5 -o $@ \
 	    -a revision=$(VERSION) \
 	    -a sysconfdir="${sysconfdir}" \
--- origsrc/uacme-1.7/configure.ac	2021-01-18 00:58:32.000000000 +0900
+++ src/uacme-1.7/configure.ac	2021-05-16 00:30:01.015262200 +0900
@@ -37,11 +37,11 @@ if test "x$ac_cv_prog_cc_c99" = "xno"; t
     AC_MSG_ERROR([Could not find a C99 compatible compiler])
 fi
 
-AX_CHECK_COMPILE_FLAG([$CFLAGS -Wall], [WCFLAGS="-Wall"])
-AX_CHECK_COMPILE_FLAG([$CFLAGS -Wextra], [WCFLAGS="$WCFLAGS -Wextra"])
-AX_CHECK_COMPILE_FLAG([$CFLAGS -pedantic], [WCFLAGS="$WCFLAGS -pedantic"])
-AX_CHECK_COMPILE_FLAG([$CFLAGS -fno-strict-aliasing],
-                      [WCFLAGS="$WCFLAGS -fno-strict-aliasing"])
+#AX_CHECK_COMPILE_FLAG([$CFLAGS -Wall], [WCFLAGS="-Wall"])
+#AX_CHECK_COMPILE_FLAG([$CFLAGS -Wextra], [WCFLAGS="$WCFLAGS -Wextra"])
+#AX_CHECK_COMPILE_FLAG([$CFLAGS -pedantic], [WCFLAGS="$WCFLAGS -pedantic"])
+#AX_CHECK_COMPILE_FLAG([$CFLAGS -fno-strict-aliasing],
+#                      [WCFLAGS="$WCFLAGS -fno-strict-aliasing"])
 AC_SUBST([WCFLAGS])
 
 AC_SYS_LARGEFILE
