--- origsrc/uacme-1.1.2/Makefile.am	2020-03-13 02:28:11.000000000 +0900
+++ src/uacme-1.1.2/Makefile.am	2020-04-09 17:45:12.426382100 +0900
@@ -79,6 +79,7 @@ ualpn.1: ualpn.1.txt $(top_srcdir)/.vers
 	    -a runstatedir="${runstatedir}" $<
 
 docs/ualpn.html: ualpn.1.txt $(top_srcdir)/.version
+	$(MKDIR_P) docs
 	$(AM_V_GEN)$(ASCIIDOC) -d manpage -b html5 -o $@ \
 	    -a revision=$(VERSION) \
 	    -a sysconfdir="${sysconfdir}" \
@@ -92,6 +93,7 @@ uacme.1: uacme.1.txt $(top_srcdir)/.vers
 	    -a runstatedir="${runstatedir}" $<
 
 docs/uacme.html: uacme.1.txt $(top_srcdir)/.version
+	$(MKDIR_P) docs
 	$(AM_V_GEN)$(ASCIIDOC) -d manpage -b html5 -o $@ \
 	    -a revision=$(VERSION) \
 	    -a sysconfdir="${sysconfdir}" \
