--- origsrc/libcbor-0.5.0/src/CMakeLists.txt	2017-02-06 08:27:34.000000000 +0900
+++ src/libcbor-0.5.0/src/CMakeLists.txt	2019-10-13 12:15:05.681540300 +0900
@@ -11,16 +11,15 @@ if (NOT DEFINED CMAKE_MACOSX_RPATH)
 	set(CMAKE_MACOSX_RPATH 0)
 endif()
 
-add_library(cbor ${SOURCES})
-add_library(cbor_shared SHARED ${SOURCES})
+add_library(cbor SHARED ${SOURCES})
 
-set_target_properties(cbor_shared PROPERTIES OUTPUT_NAME cbor VERSION "0.0.0" SOVERSION 0)
+set_target_properties(cbor PROPERTIES OUTPUT_NAME cbor VERSION "0.0.0" SOVERSION 0)
 
 configure_file(libcbor.pc.in libcbor.pc @ONLY)
 
 #http://www.cmake.org/Wiki/CMake:Install_Commands
 
-install(TARGETS cbor  cbor_shared
+install(TARGETS cbor
 	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
 	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
 	RUNTIME DESTINATION bin)
