--- origsrc/bcmatroska2-5.4.117/CMakeLists.txt	2025-04-14 18:07:41.000000000 +0900
+++ src/bcmatroska2-5.4.117/CMakeLists.txt	2026-05-28 15:52:12.802774600 +0900
@@ -23,7 +23,7 @@
 cmake_minimum_required(VERSION 3.22)
 
 project(BCMatroska2
-	VERSION 0.23
+	VERSION 5.4.117
 	LANGUAGES C
 )
 
@@ -74,7 +74,7 @@ install(FILES ${PROJECT_BINARY_DIR}/bcma
 add_subdirectory(build)
 
 include(CMakePackageConfigHelpers)
-set(CMAKE_MODULES_INSTALL_DIR "${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}/cmake")
+option(CMAKE_MODULES_INSTALL_DIR "${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}/cmake")
 configure_package_config_file("${PROJECT_NAME}Config.cmake.in" "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
 	INSTALL_DESTINATION "${CMAKE_MODULES_INSTALL_DIR}"
 	NO_SET_AND_CHECK_MACRO
@@ -93,3 +93,10 @@ install(EXPORT ${PROJECT_NAME}Targets
 	FILE "${PROJECT_NAME}Targets.cmake"
 	DESTINATION ${CMAKE_MODULES_INSTALL_DIR}
 )
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_INCLUDEDIR})
+configure_file(${CMAKE_SOURCE_DIR}/bcmatroska2.pc.in ${CMAKE_BINARY_DIR}/bcmatroska2.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/bcmatroska2.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/bcmatroska2-5.4.117/bcmatroska2.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/bcmatroska2-5.4.117/bcmatroska2.pc.in	2026-05-28 15:51:43.535646800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: bcmatroska2
+Description: Matroska library for mediastreamer2
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lbcmatroska2
+Cflags: -I${includedir}
--- origsrc/bcmatroska2-5.4.117/corec/corec/portab.h	2025-04-14 18:07:41.000000000 +0900
+++ src/bcmatroska2-5.4.117/corec/corec/portab.h	2026-05-28 13:06:22.728663800 +0900
@@ -341,7 +341,7 @@ typedef unsigned char uint_fast8_t;
 typedef signed long long int_fast64_t;
 typedef unsigned long long uint_fast64_t;
 
-#elif !defined(__GLIBC__) && !defined(__MINGW32__) && !defined(TARGET_PS2SDK) && !defined(TARGET_IPHONE) && !defined(TARGET_ANDROID) && !defined(__FreeBSD__) && !defined(__QNXNTO__)
+#elif !defined(__GLIBC__) && !defined(__MINGW32__) && !defined(TARGET_PS2SDK) && !defined(TARGET_IPHONE) && !defined(TARGET_ANDROID) && !defined(__FreeBSD__) && !defined(__QNXNTO__) && !defined(__CYGWIN__)
 
 #include <inttypes.h>
 
