--- origsrc/utimer-0.4/src/tests/maintests.c	2009-06-01 23:31:26.000000000 +0900
+++ src/utimer-0.4/src/tests/maintests.c	2025-01-16 18:47:12.243457800 +0900
@@ -32,6 +32,11 @@
 #include <glib-object.h>
 #include "../utimer.h"
 
+GMainLoop         *loop;
+gboolean          paused;
+struct termios    savedttystate;
+Config            ut_config;
+
 #define TEST_DURATION_MAX_OFFSET_MSECONDS 100
 
 /**
--- origsrc/utimer-0.4/src/utimer.c	2009-06-01 23:23:45.000000000 +0900
+++ src/utimer-0.4/src/utimer.c	2025-01-16 18:45:10.310990700 +0900
@@ -23,6 +23,11 @@
 
 #include "utimer.h"
 
+GMainLoop         *loop;
+gboolean          paused;
+struct termios    savedttystate;
+Config            ut_config;
+
 //~ static    char            **remaining_args;
 static ut_timer        *ttimer;
 static gchar           *timer_info, *countdown_info;
--- origsrc/utimer-0.4/src/utimer.h	2009-12-18 12:44:46.000000000 +0900
+++ src/utimer-0.4/src/utimer.h	2025-01-16 18:45:24.488790200 +0900
@@ -66,9 +66,9 @@
 #define TIMER_PRINT_RATE_MSEC  79
 #define TIMER_CHECK_RATE_MSEC  500
 
-GMainLoop         *loop;
-gboolean          paused;
-struct termios    savedttystate;
-Config            ut_config;
+extern GMainLoop         *loop;
+extern gboolean          paused;
+extern struct termios    savedttystate;
+extern Config            ut_config;
 
 #endif /* UTIMER_H */
