--- origsrc/disfetch-1.21/disfetch	2021-03-20 22:12:13.000000000 +0900
+++ src/disfetch-1.21/disfetch	2021-03-21 12:28:02.384137600 +0900
@@ -406,6 +406,20 @@ zorin*)
   LOGO[7]="     $a\\\\\\\\ZZZZZZZZZZZ//$r   "
   PACKAGES=$(dpkg -l | grep -c "^ii")
   ;;
+cygwin*)
+  motif="\x1B[93m$l" # bright yellow
+  a="\x1B[90m"       # bright black
+  b="\x1B[33m$l"     # yellow
+  LOGO[0]="        ${a}XXXX$r      "
+  LOGO[1]="       ${a}X$r${l}^$r${a}XX$r${l}^$r${a}X$r     "
+  LOGO[2]="       ${a}X$r${b}<XX>$r${a}X$r     "
+  LOGO[3]="     ${a}XX${r}X${a}XXXX${r}X${a}XX$r   "
+  LOGO[4]="    ${a}XX${r}XXXXXXXX${a}XX$r  "
+  LOGO[5]="   ${a}XX${r}XXXXXXXXXX${a}XX$r "
+  LOGO[6]="  ${b}I$r${a}XXX${r}XXXXXXXX${a}XXX$r${b}I$r"
+  LOGO[7]="  ${b}IL>$r${a}XX${r}XXXXXX$r${a}XX$r${b}<JI$r"
+  PACKAGES=$(cygcheck -cd | wc -l)
+  ;;
 *)
   motif="\x1B[93m$l" # bright yellow
   a="\x1B[90m"       # bright black
@@ -424,7 +438,7 @@ esac
 ###########################################
 # adding only existing layers to info     #
 ###########################################
-                         INFO+=("$motif$(whoami)$r @ $motif$(cat /etc/hostname)$r")
+                         INFO+=("$motif$(whoami)$r @ $motif$(hostname)$r")
 [[ $OS ]]             && INFO+=("${motif}OS:$r        $OS")
 [[ $KERNEL ]]         && INFO+=("${motif}KERNEL:$r    $KERNEL")
 [[ $ARCH ]]           && INFO+=("${motif}ARCH:$r      $ARCH")
