--- origsrc/tree-sitter-c-sharp-0.23.0/CMakeLists.txt	2024-11-09 22:21:17.080238200 +0900
+++ src/tree-sitter-c-sharp-0.23.0/CMakeLists.txt	2024-11-09 22:40:39.621150000 +0900
@@ -17,15 +17,17 @@ endif()
 
 find_program(TREE_SITTER_CLI tree-sitter DOC "Tree-sitter CLI")
 
+if (NOT TREE_SITTER_CLI STREQUAL "TREE_SITTER_CLI-NOTFOUND")
 add_custom_command(OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/src/parser.c"
                    DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/src/grammar.json"
                    COMMAND "${TREE_SITTER_CLI}" generate src/grammar.json
                             --abi=${TREE_SITTER_ABI_VERSION}
                    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
                    COMMENT "Generating parser.c")
+endif()
 
 add_library(tree-sitter-c-sharp src/parser.c)
-if(EXISTS src/scanner.c)
+if(EXISTS ${PROJECT_SOURCE_DIR}/src/scanner.c)
   target_sources(tree-sitter-c-sharp PRIVATE src/scanner.c)
 endif()
 target_include_directories(tree-sitter-c-sharp PRIVATE src)
@@ -41,15 +43,16 @@ set_target_properties(tree-sitter-c-shar
                       SOVERSION "${TREE_SITTER_ABI_VERSION}.${PROJECT_VERSION_MAJOR}"
                       DEFINE_SYMBOL "")
 
-configure_file(bindings/c/tree-sitter-c-sharp.pc.in
+configure_file(bindings/c/tree-sitter-c_sharp.pc.in
                "${CMAKE_CURRENT_BINARY_DIR}/tree-sitter-c-sharp.pc" @ONLY)
 
 include(GNUInstallDirs)
 
-install(FILES bindings/c/tree-sitter-c-sharp.h
-        DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/tree_sitter")
+install(FILES bindings/c/tree-sitter-c_sharp.h
+        DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/tree_sitter"
+        RENAME tree-sitter-c-sharp.h)
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/tree-sitter-c-sharp.pc"
-        DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/pkgconfig")
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 install(TARGETS tree-sitter-c-sharp
         LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 
--- origsrc/tree-sitter-c-sharp-0.23.0/bindings/c/tree-sitter-c_sharp.pc.in	2024-09-01 18:59:14.000000000 +0900
+++ src/tree-sitter-c-sharp-0.23.0/bindings/c/tree-sitter-c_sharp.pc.in	2024-11-09 22:23:54.685765400 +0900
@@ -1,11 +1,11 @@
-prefix=@PREFIX@
-libdir=@LIBDIR@
-includedir=@INCLUDEDIR@
+prefix=@CMAKE_INSTALL_PREFIX@
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
 
 Name: tree-sitter-c_sharp
 Description: C# grammar for tree-sitter
 URL: @URL@
-Version: @VERSION@
+Version: @PROJECT_VERSION@
 Requires: @REQUIRES@
-Libs: -L${libdir} @ADDITIONAL_LIBS@ -ltree-sitter-c_sharp
+Libs: -L${libdir} @ADDITIONAL_LIBS@ -ltree-sitter-c-sharp
 Cflags: -I${includedir}
