--- origsrc/recutils-1.8/Makefile.am	2018-03-17 19:11:31.000000000 +0900
+++ src/recutils-1.8/Makefile.am	2019-01-21 13:31:32.672862700 +0900
@@ -28,3 +28,6 @@ perf:
 .PHONY: perf
 
 # End of Makefile.am
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/recutils-1.8/configure.ac	2019-01-03 18:26:36.000000000 +0900
+++ src/recutils-1.8/configure.ac	2019-01-21 13:32:24.168014100 +0900
@@ -214,7 +214,7 @@ AC_ARG_ENABLE([coverage],
 AM_CONDITIONAL([USE_COVERAGE], [test "x$use_gcov" = "xyes"])
 
 dnl Generate output files
-AC_CONFIG_FILES(Makefile
+AC_CONFIG_FILES(recutils.pc Makefile
                 lib/Makefile
                 libcsv/Makefile
                 src/Makefile
--- origsrc/recutils-1.8/recutils.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/recutils-1.8/recutils.pc.in	2019-01-21 13:32:48.861151900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: recutils
+Description: Tools and libraries to access recfiles
+Version: @VERSION@
+Libs: -L${libdir} -lrec
+Cflags: -I${includedir}
--- origsrc/recutils-1.8/src/Makefile.am	2019-01-03 17:45:53.000000000 +0900
+++ src/recutils-1.8/src/Makefile.am	2019-01-21 13:07:50.208683100 +0900
@@ -62,7 +62,7 @@ endif
 AM_LDFLAGS = -no-undefined
 AM_CPPFLAGS = -I$(top_builddir)/lib -I$(top_srcdir)/lib \
               -DLOCALEDIR=\"$(localedir)\"
-librec_la_LDFLAGS = -version-info 1:0:0
+librec_la_LDFLAGS = -no-undefined -version-info 1:0:0
 librec_la_LIBADD = $(top_builddir)/lib/librecutils.la \
                    $(LIB_CLOCK_GETTIME)
 
