--- origsrc/recutils-1.7/src/Makefile.am	2014-02-25 05:55:06.000000000 +0900
+++ src/recutils-1.7/src/Makefile.am	2014-05-24 14:28:59.954785600 +0900
@@ -61,7 +61,7 @@ endif
 
 AM_LDFLAGS = -no-undefined
 AM_CPPFLAGS = -I$(top_srcdir)/lib -DLOCALEDIR=\"$(localedir)\"
-librec_la_LDFLAGS = -version-info 1:0:0
+librec_la_LDFLAGS = $(AM_LDFLAGS) -version-info 1:0:0
 librec_la_LIBADD = $(top_builddir)/lib/librecutils.la \
                    $(LIB_CLOCK_GETTIME)
 
--- origsrc/recutils-1.7/src/rec-db.c	2014-03-18 07:08:36.000000000 +0900
+++ src/recutils-1.7/src/rec-db.c	2014-05-24 13:55:16.697192900 +0900
@@ -1497,7 +1497,9 @@ rec_db_add_random_indexes (size_t **inde
 
   size_t i;
   char random_state[128];
+#if defined(HAVE_STRUCT_RANDOM_DATA)
   struct random_data random_data;
+#endif
 
   *index = malloc (sizeof(size_t) * ((num + 1) * 2));
   if (*index == NULL)
@@ -1512,14 +1514,20 @@ rec_db_add_random_indexes (size_t **inde
     }
 
   /* Insert the random indexes.  */
-
+#if defined(HAVE_STRUCT_RANDOM_DATA)
   memset (&random_data, 0, sizeof (random_data));
   initstate_r (time(NULL), (char *) &random_state, 128, &random_data);
+#else
+  initstate (time(NULL), (char *) &random_state, 128);
+#endif
   for (i = 0; i < (num * 2); i = i + 2)
     {
       size_t random_value = 0;
-      
+#if defined(HAVE_STRUCT_RANDOM_DATA)
       random_r (&random_data, (int32_t *) &random_value); /* Can't fail.  */
+#else
+      random_value = random ();
+#endif
       random_value = random_value % limit;
 
       /* Avoid having repeated random indexes.  */
