--- origsrc/libbfio-20240414/Makefile.am	2024-04-14 17:29:21.000000000 +0900
+++ src/libbfio-20240414/Makefile.am	2025-05-18 10:51:34.576982800 +0900
@@ -3,17 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
 	libbfio \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libbfio-20240414/configure.ac	2024-04-14 17:29:21.000000000 +0900
+++ src/libbfio-20240414/configure.ac	2025-05-18 10:51:34.576982800 +0900
@@ -124,15 +124,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
 AC_CONFIG_FILES([libbfio/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
