--- origsrc/xmrig-6.4.0/CMakeLists.txt	2020-10-19 05:21:04.000000000 +0900
+++ src/xmrig-6.4.0/CMakeLists.txt	2020-10-29 07:22:19.595419900 +0900
@@ -206,7 +206,7 @@ if (WITH_DEBUG_LOG)
 endif()
 
 add_executable(${CMAKE_PROJECT_NAME} ${HEADERS} ${SOURCES} ${SOURCES_OS} ${SOURCES_CPUID} ${HEADERS_CRYPTO} ${SOURCES_CRYPTO} ${SOURCES_SYSLOG} ${TLS_SOURCES} ${XMRIG_ASM_SOURCES})
-target_link_libraries(${CMAKE_PROJECT_NAME} ${XMRIG_ASM_LIBRARY} ${OPENSSL_LIBRARIES} ${UV_LIBRARIES} ${EXTRA_LIBS} ${CPUID_LIB} ${ARGON2_LIBRARY} ${ETHASH_LIBRARY})
+target_link_libraries(${CMAKE_PROJECT_NAME} ${XMRIG_ASM_LIBRARY} ${ARGON2_LIBRARY} ${OPENSSL_LIBRARIES} ${UV_LIBRARIES} ${EXTRA_LIBS} ${CPUID_LIB} ${ETHASH_LIBRARY})
 
 if (WIN32)
     add_custom_command(TARGET ${CMAKE_PROJECT_NAME} POST_BUILD
--- origsrc/xmrig-6.4.0/src/crypto/common/VirtualMemory_unix.cpp	2020-10-19 05:21:04.000000000 +0900
+++ src/xmrig-6.4.0/src/crypto/common/VirtualMemory_unix.cpp	2020-10-29 07:36:52.165852100 +0900
@@ -47,6 +47,14 @@
 #endif
 
 
+#if !defined(MAP_POPULATE)
+#define MAP_POPULATE (0)
+#endif
+#if !defined(MAP_HUGETLB)
+#define MAP_HUGETLB (0)
+#endif
+
+
 bool xmrig::VirtualMemory::isHugepagesAvailable()
 {
     return true;
