--- origsrc/htscodecs-1.1.1/Makefile.am	2021-07-07 01:35:24.000000000 +0900
+++ src/htscodecs-1.1.1/Makefile.am	2021-09-03 20:00:14.752606100 +0900
@@ -43,6 +43,10 @@ nobase_include_HEADERS = \
 	htscodecs/pack.h \
 	htscodecs/rle.h \
 	htscodecs/varint.h \
+	htscodecs/varint2.h \
 	htscodecs/htscodecs.h
 
 EXTRA_DIST = README.md LICENSE.md NEWS
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/htscodecs-1.1.1/configure.ac	2021-07-07 01:35:24.000000000 +0900
+++ src/htscodecs-1.1.1/configure.ac	2021-09-03 20:00:14.757612100 +0900
@@ -114,5 +114,5 @@ AC_CHECK_HEADER(zlib.h)
 dnl Checks for library functions.
 AC_SEARCH_LIBS([pthread_join], [pthread])
 
-AC_CONFIG_FILES([Makefile htscodecs/Makefile tests/Makefile])
+AC_CONFIG_FILES([htscodecs.pc Makefile htscodecs/Makefile tests/Makefile])
 AC_OUTPUT
--- origsrc/htscodecs-1.1.1/htscodecs/Makefile.am	2021-07-07 01:35:24.000000000 +0900
+++ src/htscodecs-1.1.1/htscodecs/Makefile.am	2021-09-03 17:59:23.683558400 +0900
@@ -57,7 +57,7 @@ libhtscodecs_la_SOURCES = \
 	htscodecs_endian.h \
 	utils.h
 
-libhtscodecs_la_LDFLAGS = -version-info @VERS_CURRENT@:@VERS_REVISION@:@VERS_AGE@ 
+libhtscodecs_la_LDFLAGS = -no-undefined -version-info @VERS_CURRENT@:@VERS_REVISION@:@VERS_AGE@ 
 libhtscodecs_la_LIBADD = @LIBZ@ -lm
 
 version.h: force
--- origsrc/htscodecs-1.1.1/htscodecs.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/htscodecs-1.1.1/htscodecs.pc.in	2021-09-03 20:00:14.707593200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: htscodecs
+Description: Custom compression for CRAM and others
+Version: @VERSION@
+Libs: -L${libdir} -lhtscodecs
+Cflags: -I${includedir}
--- origsrc/htscodecs-1.1.1/tests/arith_dynamic_test.c	2021-07-07 01:35:24.000000000 +0900
+++ src/htscodecs-1.1.1/tests/arith_dynamic_test.c	2021-09-03 17:59:58.802475700 +0900
@@ -41,6 +41,7 @@
 #include <sys/time.h>
 #include <limits.h>
 #include <fcntl.h>
+#include <getopt.h>
 
 #include "htscodecs/arith_dynamic.h"
 
@@ -92,9 +93,6 @@ int main(int argc, char **argv) {
         _setmode(_fileno(stdout), _O_BINARY);
 #endif
 
-    extern char *optarg;
-    extern int optind;
-
     while ((opt = getopt(argc, argv, "o:dtr")) != -1) {
 	switch (opt) {
 	case 'o': {
--- origsrc/htscodecs-1.1.1/tests/fqzcomp_qual_test.c	2021-07-07 01:35:24.000000000 +0900
+++ src/htscodecs-1.1.1/tests/fqzcomp_qual_test.c	2021-09-03 18:05:16.926197200 +0900
@@ -41,6 +41,7 @@
 #include <fcntl.h>
 #include <ctype.h>
 #include <limits.h>
+#include <getopt.h>
 
 #include "htscodecs/fqzcomp_qual.h"
 #include "htscodecs/varint.h"
@@ -327,8 +328,6 @@ int main(int argc, char **argv) {
         _setmode(_fileno(stdout), _O_BINARY);
 #endif
 
-    extern char *optarg;
-    extern int optind;
     int opt;
 
     while ((opt = getopt(argc, argv, "ds:s:b:rx:")) != -1) {
--- origsrc/htscodecs-1.1.1/tests/rANS_static4x16pr_test.c	2021-07-07 01:35:24.000000000 +0900
+++ src/htscodecs-1.1.1/tests/rANS_static4x16pr_test.c	2021-09-03 18:00:21.487050400 +0900
@@ -40,6 +40,7 @@
 #include <string.h>
 #include <fcntl.h>
 #include <sys/time.h>
+#include <getopt.h>
 
 #include "htscodecs/rANS_static4x16.h"
 
@@ -91,9 +92,6 @@ int main(int argc, char **argv) {
         _setmode(_fileno(stdout), _O_BINARY);
 #endif
 
-    extern char *optarg;
-    extern int optind;
-
     while ((opt = getopt(argc, argv, "o:dtr")) != -1) {
 	switch (opt) {
 	case 'o': {
--- origsrc/htscodecs-1.1.1/tests/rANS_static_test.c	2021-07-07 01:35:24.000000000 +0900
+++ src/htscodecs-1.1.1/tests/rANS_static_test.c	2021-09-03 18:00:46.208681700 +0900
@@ -40,6 +40,7 @@
 #include <string.h>
 #include <fcntl.h>
 #include <sys/time.h>
+#include <getopt.h>
 
 #include "htscodecs/rANS_static.h"
 
@@ -104,9 +105,6 @@ int main(int argc, char **argv) {
         _setmode(_fileno(stdout), _O_BINARY);
 #endif
 
-    extern char *optarg;
-    extern int optind;
-
     while ((opt = getopt(argc, argv, "o:dtr")) != -1) {
 	switch (opt) {
 	case 'o':
