--- origsrc/fmt-10.1.1/include/fmt/core.h	2023-08-27 00:20:41.000000000 +0900
+++ src/fmt-10.1.1/include/fmt/core.h	2023-12-02 12:35:55.873427300 +0900
@@ -18,7 +18,7 @@
 #include <type_traits>
 
 // The fmt library version in the form major * 10000 + minor * 100 + patch.
-#define FMT_VERSION 100100
+#define FMT_VERSION 100101
 
 #if defined(__clang__) && !defined(__ibmxl__)
 #  define FMT_CLANG_VERSION (__clang_major__ * 100 + __clang_minor__)
