--- origsrc/finalcut-0.8.0/examples/Makefile.am	2021-10-31 19:35:18.000000000 +0900
+++ src/finalcut-0.8.0/examples/Makefile.am	2022-05-02 17:26:50.348590800 +0900
@@ -7,7 +7,7 @@ LIBS = -lfinal
 if ! CPPUNIT_TEST
 
 AM_LDFLAGS = -L$(top_builddir)/final/.libs
-AM_CPPFLAGS = -I$(top_srcdir)/final -Wall -Werror -std=c++11
+AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/final -Wall -Werror -std=c++11
 
 noinst_PROGRAMS = \
 	7segment \
--- origsrc/finalcut-0.8.0/final/Makefile.am	2021-10-31 19:35:18.000000000 +0900
+++ src/finalcut-0.8.0/final/Makefile.am	2022-05-02 16:59:50.412989700 +0900
@@ -4,7 +4,7 @@
 
 LIBS = @FINAL_LIBS@
 
-AM_CPPFLAGS = -I$(top_srcdir)/final -Wall -Werror -DCOMPILE_FINAL_CUT -std=c++11
+AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/final -Wall -Werror -DCOMPILE_FINAL_CUT -std=c++11
 AUTOMAKE_OPTIONS = subdir-objects
 
 SUBDIRS = . font
--- origsrc/finalcut-0.8.0/src/Makefile.am	2021-10-31 19:35:18.000000000 +0900
+++ src/finalcut-0.8.0/src/Makefile.am	2022-05-02 16:59:50.412989700 +0900
@@ -4,7 +4,7 @@
 
 LIBS = @FINAL_LIBS@
 
-AM_CPPFLAGS = -I$(top_srcdir)/final -Wall -Werror -DCOMPILE_FINAL_CUT -std=c++11
+AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/final -Wall -Werror -DCOMPILE_FINAL_CUT -std=c++11
 AUTOMAKE_OPTIONS = subdir-objects
 
 SUBDIRS = . font
