--- origsrc/finalcut-0.9.1/examples/Makefile.am	2024-07-27 22:21:14.000000000 +0900
+++ src/finalcut-0.9.1/examples/Makefile.am	2024-09-11 08:50:31.805720700 +0900
@@ -7,7 +7,7 @@ LIBS = -lfinal
 if ! CPPUNIT_TEST
 
 AM_LDFLAGS = -L$(top_builddir)/final/.libs
-AM_CPPFLAGS = -I$(top_srcdir)/final -Wall -Werror -std=c++14
+AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/final -Wall -Werror -std=c++14
 
 noinst_PROGRAMS = \
 	7segment \
--- origsrc/finalcut-0.9.1/final/Makefile.am	2024-07-27 22:21:14.000000000 +0900
+++ src/finalcut-0.9.1/final/Makefile.am	2024-09-11 08:50:31.805720700 +0900
@@ -4,7 +4,7 @@
 
 LIBS = @FINAL_LIBS@
 
-AM_CPPFLAGS = -I$(top_srcdir)/final -Wall -Werror -DCOMPILE_FINAL_CUT -std=c++14
+AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/final -Wall -Werror -DCOMPILE_FINAL_CUT -std=c++14
 AUTOMAKE_OPTIONS = subdir-objects
 
 SUBDIRS = . font
--- origsrc/finalcut-0.9.1/final/output/tty/ftermios.cpp	2024-07-27 22:21:14.000000000 +0900
+++ src/finalcut-0.9.1/final/output/tty/ftermios.cpp	2024-09-11 08:50:31.805720700 +0900
@@ -26,12 +26,12 @@
   #define _BSD_SOURCE
 #endif
 
-#include <termios.h>
-
 #if defined(__CYGWIN__)
   #define __STRICT_ANSI__  // need for bits/c++config.h
 #endif
 
+#include <termios.h>
+
 #include <system_error>
 #include <unordered_map>
 
