--- origsrc/libdop-1.2.0/CMakeLists.txt	2017-12-26 05:15:43.000000000 +0900
+++ src/libdop-1.2.0/CMakeLists.txt	2018-08-19 20:02:06.548562400 +0900
@@ -28,6 +28,10 @@ add_library(dop SHARED
     src/dmath.c
     src/dstring.c)
 
+if (CYGWIN)
+    set_target_properties(dop PROPERTIES SOVERSION 0)
+endif()
+
 add_executable(hexdump_test
     tests/hexdump-test.c)
 
@@ -111,13 +115,13 @@ target_link_libraries(dstring_test_strch
     dop)
 
 # Install
-install(TARGETS dop LIBRARY DESTINATION lib)
+install(TARGETS dop RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 install_files("/include/dop" .h
     ${CMAKE_SOURCE_DIR}/src/dop/utils.h
     ${CMAKE_SOURCE_DIR}/src/dop/memory.h
     ${CMAKE_SOURCE_DIR}/src/dop/filesystem.h
     ${CMAKE_SOURCE_DIR}/src/dop/version.h)
-install_files("/share/pkgconfig" .pc
+install_files("/lib/pkgconfig" .pc
     ${CMAKE_SOURCE_DIR}/pkgconfig/dop.pc)
 
 # Tests
