--- origsrc/aws-c-event-stream-0.1.5/CMakeLists.txt	2019-11-21 11:04:55.000000000 +0900
+++ src/aws-c-event-stream-0.1.5/CMakeLists.txt	2020-06-05 11:33:20.526076100 +0900
@@ -32,6 +32,9 @@ set(AWS_MODULE_DIR "/${CMAKE_INSTALL_LIB
 string(REPLACE ";" "${AWS_MODULE_DIR};" AWS_MODULE_PATH "${CMAKE_PREFIX_PATH}${AWS_MODULE_DIR}")
 # Append that generated list to the module search path
 list(APPEND CMAKE_MODULE_PATH ${AWS_MODULE_PATH})
+if (CYGWIN)
+    list(APPEND CMAKE_MODULE_PATH "${AWS_MODULE_PATH}/aws-c-common")
+endif()
 
 include(AwsCFlags)
 include(AwsSharedLibSetup)
@@ -81,7 +84,11 @@ target_include_directories(${CMAKE_PROJE
 
 
 set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES VERSION 1.0.0)
-set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES SOVERSION 0unstable)
+if(CYGWIN)
+    set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES SOVERSION 0)
+else()
+    set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES SOVERSION 0unstable)
+endif()
 
 find_package(aws-c-common REQUIRED)
 find_package(aws-checksums REQUIRED)
@@ -101,7 +108,7 @@ else()
 endif()
 
 install(EXPORT "${CMAKE_PROJECT_NAME}-targets"
-    DESTINATION "${LIBRARY_DIRECTORY}/${CMAKE_PROJECT_NAME}/cmake/${TARGET_DIR}/"
+    DESTINATION "${LIBRARY_DIRECTORY}/cmake/${CMAKE_PROJECT_NAME}"
     NAMESPACE AWS::
     COMPONENT Development)
 
@@ -110,7 +117,7 @@ configure_file("cmake/${CMAKE_PROJECT_NA
     @ONLY)
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_PROJECT_NAME}-config.cmake"
-    DESTINATION "${LIBRARY_DIRECTORY}/${CMAKE_PROJECT_NAME}/cmake/"
+    DESTINATION "${LIBRARY_DIRECTORY}/cmake/${CMAKE_PROJECT_NAME}"
     COMPONENT Development)
 
 
@@ -123,3 +130,10 @@ if(NOT MSVC)
 endif()
 
 add_subdirectory(bin)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/aws-c-event-stream.pc.in ${CMAKE_BINARY_DIR}/aws-c-event-stream.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/aws-c-event-stream.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/aws-c-event-stream-0.1.5/aws-c-event-stream.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/aws-c-event-stream-0.1.5/aws-c-event-stream.pc.in	2020-06-05 11:33:21.038375600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: aws-c-event-stream
+Description: C99 implementation of the vnd.amazon.eventstream content-type
+Version: @VERSION@
+Libs: -L${libdir} -laws-c-event-stream
+Cflags: -I${includedir}
--- origsrc/aws-c-event-stream-0.1.5/cmake/aws-c-event-stream-config.cmake	2019-11-21 11:04:55.000000000 +0900
+++ src/aws-c-event-stream-0.1.5/cmake/aws-c-event-stream-config.cmake	2020-06-05 11:33:21.053731000 +0900
@@ -2,9 +2,4 @@ include(CMakeFindDependencyMacro)
 find_dependency(aws-c-common)
 find_dependency(aws-checksums)
 
-if (BUILD_SHARED_LIBS)
-    include(${CMAKE_CURRENT_LIST_DIR}/shared/@CMAKE_PROJECT_NAME@-targets.cmake)
-else()
-    include(${CMAKE_CURRENT_LIST_DIR}/static/@CMAKE_PROJECT_NAME@-targets.cmake)
-endif()
-
+include(${CMAKE_CURRENT_LIST_DIR}/@CMAKE_PROJECT_NAME@-targets.cmake)
