--- origsrc/lcalc-2.2.1/configure.ac	2026-04-07 12:37:04.000000000 +0900
+++ src/lcalc-2.2.1/configure.ac	2026-07-28 15:32:11.970704300 +0900
@@ -11,7 +11,7 @@ PACKAGE_DESCRIPTION="Command-line utilit
 AC_SUBST(PACKAGE_DESCRIPTION)
 
 dnl Store build artifacts in this directory to keep the house clean.
-AC_CONFIG_AUX_DIR([build-aux])
+dnl AC_CONFIG_AUX_DIR([build-aux])
 AM_INIT_AUTOMAKE([-Wall foreign no-dist-gzip dist-xz tar-ustar])
 AM_MAINTAINER_MODE([enable])
 AC_CONFIG_MACRO_DIR([m4])
