--- origsrc/libfplist-20200803/Makefile.am	2020-08-03 15:49:02.000000000 +0900
+++ src/libfplist-20200803/Makefile.am	2020-09-03 00:55:11.731566500 +0900
@@ -3,15 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libcnotify \
-	libuna \
-	libfguid \
-	libfvalue \
 	libfplist \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libfplist-20200803/configure.ac	2020-08-03 15:49:02.000000000 +0900
+++ src/libfplist-20200803/configure.ac	2020-09-03 00:55:11.744541100 +0900
@@ -117,13 +117,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
 AC_CONFIG_FILES([libfplist/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
--- origsrc/libfplist-20200803/libfplist/Makefile.am	2020-08-03 15:49:02.000000000 +0900
+++ src/libfplist-20200803/libfplist/Makefile.am	2020-09-03 00:55:11.756548100 +0900
@@ -2,6 +2,7 @@ AM_LFLAGS = -Cf
 AM_YFLAGS = -d -v -l -p xml_scanner_
 
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libfplist-20200803/tests/Makefile.am	2020-08-03 15:49:02.000000000 +0900
+++ src/libfplist-20200803/tests/Makefile.am	2020-09-03 00:55:11.764541100 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
