--- origsrc/libfplist-20190101/Makefile.am	2019-01-01 19:00:38.000000000 +0900
+++ src/libfplist-20190101/Makefile.am	2019-05-16 23:25:16.001746000 +0900
@@ -3,15 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libcnotify \
-	libuna \
-	libfguid \
-	libfvalue \
 	libfplist \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libfplist-20190101/configure.ac	2019-01-01 19:00:38.000000000 +0900
+++ src/libfplist-20190101/configure.ac	2019-05-16 23:25:04.764759400 +0900
@@ -136,13 +136,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
 AC_CONFIG_FILES([libfplist/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
--- origsrc/libfplist-20190101/libfplist/Makefile.am	2019-01-01 19:00:38.000000000 +0900
+++ src/libfplist-20190101/libfplist/Makefile.am	2019-05-16 23:25:32.811161300 +0900
@@ -2,6 +2,7 @@ AM_LFLAGS = -Cf
 AM_YFLAGS = -d -v -l -p xml_scanner_
 
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libfplist-20190101/tests/Makefile.am	2019-01-01 19:00:38.000000000 +0900
+++ src/libfplist-20190101/tests/Makefile.am	2019-05-16 23:25:49.380572600 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
