--- origsrc/libuiomux-1.2.0/Makefile.am	2010-06-28 13:18:06.000000000 +0900
+++ src/libuiomux-1.2.0/Makefile.am	2014-12-20 19:42:12.049826400 +0900
@@ -10,3 +10,5 @@ pkgconfig_DATA = uiomux.pc
 
 # Extra files to distribute in the source tarball.
 EXTRA_DIST = uiomux.pc.in uiomux-uninstalled.pc.in
+
+ACLOCAL_AMFLAGS = -I m4
--- origsrc/libuiomux-1.2.0/src/libuiomux/Makefile.am	2010-06-28 13:18:06.000000000 +0900
+++ src/libuiomux-1.2.0/src/libuiomux/Makefile.am	2014-12-20 19:42:12.049826400 +0900
@@ -18,5 +18,5 @@ libuiomux_la_SOURCES = \
 	uio.c \
 	uiomux.c
 
-libuiomux_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
+libuiomux_la_LDFLAGS = -no-undefined -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
 libuiomux_la_LIBADD = $(PTHREAD_LIBS) $(RT_LIBS)
