--- origsrc/libuiomux-1.6.2/src/libuiomux/Makefile.am	2011-08-26 22:31:41.000000000 +0900
+++ src/libuiomux-1.6.2/src/libuiomux/Makefile.am	2015-04-09 00:57:10.207774100 +0900
@@ -17,5 +17,5 @@ libuiomux_la_SOURCES = \
 	uio.c \
 	uiomux.c
 
-libuiomux_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
+libuiomux_la_LDFLAGS = -no-undefined -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
 libuiomux_la_LIBADD = $(PTHREAD_LIBS) $(RT_LIBS)
