--- origsrc/libuiomux/Makefile.am	2012-04-26 11:40:25.000000000 +0900
+++ src/libuiomux/Makefile.am	2012-04-26 11:43:28.985175200 +0900
@@ -10,3 +10,5 @@ pkgconfig_DATA = uiomux.pc
 
 # Extra files to distribute in the source tarball.
 EXTRA_DIST = uiomux.pc.in uiomux-uninstalled.pc.in
+
+ACLOCAL_AMFLAGS = -I m4
--- origsrc/libuiomux/src/libuiomux/Makefile.am	2012-04-26 11:40:25.000000000 +0900
+++ src/libuiomux/src/libuiomux/Makefile.am	2012-04-26 11:40:55.988104900 +0900
@@ -18,5 +18,5 @@ libuiomux_la_SOURCES = \
 	uio.c \
 	uiomux.c
 
-libuiomux_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
+libuiomux_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@ -no-undefined
 libuiomux_la_LIBADD = $(PTHREAD_LIBS) $(RT_LIBS)
