--- origsrc/MicroJSON-0.3.2/CMakeLists.txt	2013-08-05 16:41:31.000000000 +0900
+++ src/MicroJSON-0.3.2/CMakeLists.txt	2014-12-08 09:19:43.242179400 +0900
@@ -4,11 +4,11 @@ project (MicroJSON)
 
 
 add_library (SharedTarget SHARED Node.cpp Exception.cpp )
-
+target_link_libraries(SharedTarget UTF8Strings-1)
 
 set_target_properties(SharedTarget PROPERTIES OUTPUT_NAME MicroJSON-0)
 
-INSTALL(TARGETS SharedTarget DESTINATION lib${LIB_SUFFIX})
+INSTALL(TARGETS SharedTarget RUNTIME DESTINATION bin ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
 INSTALL(FILES Node.h Exception.h DESTINATION include/MicroJSON-0)
 
