--- origsrc/MicroJSON-0.2.0/CMakeLists.txt	2013-02-14 16:34:45.000000000 +0900
+++ src/MicroJSON-0.2.0/CMakeLists.txt	2013-02-24 21:50:59.522631900 +0900
@@ -4,14 +4,21 @@ project (MicroJSON)
 
 
 add_library (SharedTarget SHARED Node.cpp Exception.cpp )
+target_link_libraries(SharedTarget UTF8Strings-1)
+if (NOT CYGWIN)
 add_library (StaticTarget STATIC Node.cpp Exception.cpp )
+endif ()
 
 
 set_target_properties(SharedTarget PROPERTIES OUTPUT_NAME MicroJSON-0)
+if (NOT CYGWIN)
 set_target_properties(StaticTarget PROPERTIES OUTPUT_NAME MicroJSON-0)
+endif ()
 
-INSTALL(TARGETS SharedTarget DESTINATION lib)
+INSTALL(TARGETS SharedTarget RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+if (NOT CYGWIN)
 INSTALL(TARGETS StaticTarget DESTINATION lib)
+endif ()
 
 INSTALL(FILES Node.h Exception.h DESTINATION include/MicroJSON-0)
 
