--- origsrc/abseil-cpp-20240722.0/CMake/AbseilDll.cmake	2024-08-02 03:05:11.000000000 +0900
+++ src/abseil-cpp-20240722.0/CMake/AbseilDll.cmake	2024-09-23 11:58:09.334912400 +0900
@@ -835,9 +835,15 @@ function(absl_make_dll)
       ${_dll_libs}
       ${ABSL_DEFAULT_LINKOPTS}
   )
+  if (CYGWIN)
+    string(REPLACE "_dll" "" _dll_output_name ${_dll})
+  else()
+    set(_dll_output_name ${_dll})
+  endif()
   set_target_properties(${_dll} PROPERTIES
     LINKER_LANGUAGE "CXX"
     SOVERSION ${ABSL_SOVERSION}
+    OUTPUT_NAME ${_dll_output_name}
   )
   target_include_directories(
     ${_dll}
@@ -866,19 +872,19 @@ function(absl_make_dll)
   endforeach()
   string(REPLACE ";" " " PC_LINKOPTS "${ABSL_CC_LIB_LINKOPTS}")
 
-  FILE(GENERATE OUTPUT "${CMAKE_BINARY_DIR}/lib/pkgconfig/${_dll}.pc" CONTENT "\
+  FILE(GENERATE OUTPUT "${CMAKE_BINARY_DIR}/lib/pkgconfig/${_dll_output_name}.pc" CONTENT "\
 prefix=${CMAKE_INSTALL_PREFIX}\n\
 exec_prefix=\${prefix}\n\
 libdir=${CMAKE_INSTALL_FULL_LIBDIR}\n\
 includedir=${CMAKE_INSTALL_FULL_INCLUDEDIR}\n\
 \n\
-Name: ${_dll}\n\
+Name: ${_dll_output_name}\n\
 Description: Abseil DLL library\n\
 URL: https://abseil.io/\n\
 Version: ${absl_VERSION}\n\
-Libs: -L\${libdir} $<$<NOT:$<BOOL:${ABSL_CC_LIB_IS_INTERFACE}>>:-l${_dll}> ${PC_LINKOPTS}\n\
+Libs: -L\${libdir} $<$<NOT:$<BOOL:${ABSL_CC_LIB_IS_INTERFACE}>>:-l${_dll_output_name}> ${PC_LINKOPTS}\n\
 Cflags: -I\${includedir}${PC_CFLAGS}\n")
-  INSTALL(FILES "${CMAKE_BINARY_DIR}/lib/pkgconfig/${_dll}.pc"
+  INSTALL(FILES "${CMAKE_BINARY_DIR}/lib/pkgconfig/${_dll_output_name}.pc"
     DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 
   target_compile_definitions(
--- origsrc/abseil-cpp-20240722.0/CMake/AbseilHelpers.cmake	2024-08-02 03:05:11.000000000 +0900
+++ src/abseil-cpp-20240722.0/CMake/AbseilHelpers.cmake	2024-09-23 12:33:55.418394600 +0900
@@ -158,9 +158,14 @@ function(absl_cc_library)
     else()
       set(PC_VERSION "head")
     endif()
-    if(NOT _build_type STREQUAL "dll")
+    if(NOT CYGWIN AND NOT _build_type STREQUAL "dll")
       set(LNK_LIB "${LNK_LIB} -labsl_${_NAME}")
     endif()
+    if (CYGWIN)
+      set(_abseil_dll_name "abseil")
+    else()
+      set(_abseil_dll_name "abseil_dll")
+    endif()
     foreach(dep ${ABSL_CC_LIB_DEPS})
       if(${dep} MATCHES "^absl::(.*)")
         # for DLL builds many libs are not created, but add
@@ -168,14 +173,14 @@ function(absl_cc_library)
         if(_build_type STREQUAL "dll")
           # hide this MATCHES in an if-clause so it doesn't overwrite
           # the CMAKE_MATCH_1 from (${dep} MATCHES "^absl::(.*)")
-          if(NOT PC_DEPS MATCHES "abseil_dll")
+          if(NOT PC_DEPS MATCHES ${_abseil_dll_name})
             # Join deps with commas.
             if(PC_DEPS)
               set(PC_DEPS "${PC_DEPS},")
             endif()
             # don't duplicate dll-dep if it exists already
-            set(PC_DEPS "${PC_DEPS} abseil_dll = ${PC_VERSION}")
-            set(LNK_LIB "${LNK_LIB} -labseil_dll")
+            set(PC_DEPS "${PC_DEPS} ${_abseil_dll_name} = ${PC_VERSION}")
+            set(LNK_LIB "${LNK_LIB} -l${abseil_dll_name}")
           endif()
         else()
           # Join deps with commas.
@@ -207,7 +212,11 @@ function(absl_cc_library)
         set(PC_CFLAGS "${PC_CFLAGS} ${cflag}")
       endif()
     endforeach()
-    string(REPLACE ";" " " PC_LINKOPTS "${ABSL_CC_LIB_LINKOPTS}")
+    if (CYGWIN AND (_build_type STREQUAL "dll"))
+      set(PC_LINKOPTS "-l${_abseil_dll_name}")
+    else()
+      string(REPLACE ";" " " PC_LINKOPTS "${ABSL_CC_LIB_LINKOPTS}")
+    endif()
     FILE(GENERATE OUTPUT "${CMAKE_BINARY_DIR}/lib/pkgconfig/absl_${_NAME}.pc" CONTENT "\
 prefix=${CMAKE_INSTALL_PREFIX}\n\
 exec_prefix=\${prefix}\n\
--- origsrc/abseil-cpp-20240722.0/absl/base/attributes.h	2024-08-02 03:05:11.000000000 +0900
+++ src/abseil-cpp-20240722.0/absl/base/attributes.h	2024-09-23 06:55:48.329461000 +0900
@@ -139,7 +139,7 @@
 #if (ABSL_HAVE_ATTRIBUTE(weak) ||                                         \
      (defined(__GNUC__) && !defined(__clang__))) &&                       \
     (!defined(_WIN32) || (defined(__clang__) && __clang_major__ >= 9)) && \
-    !defined(__MINGW32__)
+    !defined(__MINGW32__) && !defined(__CYGWIN__)
 #undef ABSL_ATTRIBUTE_WEAK
 #define ABSL_ATTRIBUTE_WEAK __attribute__((weak))
 #define ABSL_HAVE_ATTRIBUTE_WEAK 1
--- origsrc/abseil-cpp-20240722.0/absl/base/config.h	2024-08-02 03:05:11.000000000 +0900
+++ src/abseil-cpp-20240722.0/absl/base/config.h	2024-09-23 01:19:09.618373900 +0900
@@ -380,7 +380,7 @@ static_assert(ABSL_INTERNAL_INLINE_NAMES
     defined(__asmjs__) || defined(__EMSCRIPTEN__) || defined(__Fuchsia__) || \
     defined(__sun) || defined(__myriad2__) || defined(__HAIKU__) ||          \
     defined(__OpenBSD__) || defined(__NetBSD__) || defined(__QNX__) ||       \
-    defined(__VXWORKS__) || defined(__hexagon__)
+    defined(__VXWORKS__) || defined(__hexagon__) || defined(__CYGWIN__)
 #define ABSL_HAVE_MMAP 1
 #endif
 
--- origsrc/abseil-cpp-20240722.0/absl/base/internal/sysinfo.cc	2024-08-02 03:05:11.000000000 +0900
+++ src/abseil-cpp-20240722.0/absl/base/internal/sysinfo.cc	2024-09-23 01:07:02.342782900 +0900
@@ -461,6 +461,12 @@ pid_t GetTID() {
   return reinterpret_cast<pid_t>(thread);
 }
 
+#elif defined(__CYGWIN__)
+
+pid_t GetTID() {
+  return reinterpret_cast<pid_t>(pthread_self());
+}
+
 #else
 
 // Fallback implementation of `GetTID` using `pthread_self`.
--- origsrc/abseil-cpp-20240722.0/absl/base/policy_checks.h	2024-08-02 03:05:11.000000000 +0900
+++ src/abseil-cpp-20240722.0/absl/base/policy_checks.h	2024-09-23 01:00:30.989682700 +0900
@@ -33,14 +33,6 @@
 #endif
 
 // -----------------------------------------------------------------------------
-// Operating System Check
-// -----------------------------------------------------------------------------
-
-#if defined(__CYGWIN__)
-#error "Cygwin is not supported."
-#endif
-
-// -----------------------------------------------------------------------------
 // Toolchain Check
 // -----------------------------------------------------------------------------
 
