--- origsrc/povray-3.7.0.10/vfe/unix/syspovconfig.h	2021-07-08 18:56:01.000000000 +0900
+++ src/povray-3.7.0.10/vfe/unix/syspovconfig.h	2024-09-16 13:50:33.626855900 +0900
@@ -155,8 +155,8 @@ const int NULL=0;
     #error Sun/Oracle Solaris detected, but not explicitly supported yet; proceed at your own risk.
 #elif defined(__CYGWIN__)
     // Cygwin detected.
-    // Not officially supported yet; comment-out the following line to try with default POSIX settings.
-    #error Cygwin detected, but not explicitly supported yet; proceed at your own risk.
+    #include <unistd.h>
+    #define lseek64(handle,offset,whence) lseek(handle,offset,whence)
 #elif defined(__unix__)
     // Some Unix other than the above detected.
     #include <sys/param.h>
