--- origsrc/povray-3.7.0.8/vfe/unix/syspovconfig.h	2018-05-27 18:54:06.000000000 +0900
+++ src/povray-3.7.0.8/vfe/unix/syspovconfig.h	2018-11-30 22:29:46.934950700 +0900
@@ -156,7 +156,9 @@ const int NULL=0;
 #elif defined(__CYGWIN__)
     // Cygwin detected.
     // Not officially supported yet; comment-out the following line to try with default POSIX settings.
-    #error Cygwin detected, but not explicitly supported yet; proceed at your own risk.
+    // #error Cygwin detected, but not explicitly supported yet; proceed at your own risk.
+    #include <unistd.h>
+    #define lseek64(handle,offset,whence) lseek(handle,offset,whence)
 #elif defined(__unix__)
     // Some Unix other than the above detected.
     #include <sys/param.h>
