--- origsrc/libibcm-1.0.5/Makefile.am	2007-04-28 02:54:41.000000000 +0900
+++ src/libibcm-1.0.5/Makefile.am	2013-02-24 17:07:13.913173700 +0900
@@ -14,7 +14,7 @@ endif
 
 src_libibcm_la_SOURCES = src/cm.c
 src_libibcm_la_LDFLAGS = -version-info 1 -export-dynamic \
-			 $(libibcm_version_script)
+			 $(libibcm_version_script) -no-undefined
 src_libibcm_la_DEPENDENCIES = $(srcdir)/src/libibcm.map
 
 # Sample program requires use if librdmacm.
--- origsrc/libibcm-1.0.5/include/infiniband/cm_abi.h	2007-04-28 02:54:41.000000000 +0900
+++ src/libibcm-1.0.5/include/infiniband/cm_abi.h	2013-02-24 17:42:44.333921000 +0900
@@ -36,7 +36,13 @@
 #ifndef CM_ABI_H
 #define CM_ABI_H
 
+#if defined(__CYGWIN__)
+#include <asm/types.h>
+typedef __int64_t __s64;
+typedef __uint64_t __u64;
+#else
 #include <linux/types.h>
+#endif
 #include <infiniband/sa.h>
 #include <infiniband/marshall.h>
 
