--- origsrc/libticalcs2-1.1.8/configure.ac	2013-03-29 21:45:46.000000000 +0900
+++ src/libticalcs2-1.1.8/configure.ac	2013-09-29 19:40:02.796995700 +0900
@@ -108,7 +108,7 @@ CFLAGS="$CFLAGS $ARCH"
 # What flag to depends on gcc version: gcc3 uses "-mms-bitfields", while
 # gcc2 uses "-fnative-struct".
 case $host_os in
-*mingw*|*cygwin*)
+*mingw*)
   if test x$GCC = xyes; then
     msnative_struct=''
     AC_MSG_CHECKING([how to get MSVC-compatible struct packing])
