--- origsrc/coinfinder-1.0.1/CMakeLists.txt	2019-12-09 22:26:54.000000000 +0900
+++ src/coinfinder-1.0.1/CMakeLists.txt	2020-01-05 20:08:19.634990800 +0900
@@ -5,7 +5,7 @@ set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
-set(Boost_USE_STATIC_LIBS ON)
+set(Boost_USE_STATIC_LIBS OFF)
 
 set(SOURCE_FILES coinfind-code/main.cpp coinfind-code/id_lookup.cpp coinfind-code/id_lookup.h coinfind-code/dataset.cpp coinfind-code/dataset.h coinfind-code/coincidence.cpp coinfind-code/coincidence.h coinfind-code/connectivity.cpp coinfind-code/connectivity.h coinfind-code/binomial_test.cpp coinfind-code/binomial_test.h coinfind-code/significance.cpp coinfind-code/significance.h coinfind-code/constants.h coinfind-code/parameters.cpp coinfind-code/parameters.h coinfind-code/elements.cpp coinfind-code/elements.h coinfind-code/test_cases.cpp coinfind-code/test_cases.h coinfind-code/phylomax.py coinfind-code/common_ancestor.py coinfind-code/bugfix.h coinfind-code/bugfix.cpp coinfind-code/lineage.cpp coinfind-code/lineage.h coinfind-code/gexf.cpp coinfind-code/gexf.h coinfind-code/network.cpp coinfind-code/network.h)
 add_executable(coinfinder ${SOURCE_FILES})
