--- origsrc/aravis-0.5.10/Makefile.am	2017-05-31 21:10:50.000000000 +0900
+++ src/aravis-0.5.10/Makefile.am	2017-11-27 12:26:33.561334600 +0900
@@ -22,14 +22,6 @@ endif
 
 SUBDIRS += tests po docs
 
-aravisdocdir = ${docdir}/aravis-@ARAVIS_API_VERSION@
-aravisdoc_DATA = 	\
-	README.md	\
-	COPYING		\
-	ChangeLog 	\
-	AUTHORS		\
-	NEWS
-
 EXTRA_DIST += m4/introspection.m4 README.md
 
 # Copy all the spec files. Of cource, only one is actually used.
--- origsrc/aravis-0.5.10/gst/Makefile.am	2016-09-16 23:04:31.000000000 +0900
+++ src/aravis-0.5.10/gst/Makefile.am	2017-11-27 12:26:33.561334600 +0900
@@ -16,7 +16,7 @@ libgstaravis_@ARAVIS_API_VERSION@_la_CFL
 libgstaravis_@ARAVIS_API_VERSION@_la_LIBADD =		\
 	$(ARAVIS_GST_PLUGIN_LIBS) 			\
 	$(top_builddir)/src/libaravis-@ARAVIS_API_VERSION@.la
-libgstaravis_@ARAVIS_API_VERSION@_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS)
+libgstaravis_@ARAVIS_API_VERSION@_la_LDFLAGS = -no-undefined $(GST_PLUGIN_LDFLAGS)
 libgstaravis_@ARAVIS_API_VERSION@_la_LIBTOOLFLAGS = --tag=disable-static
 
 libgstaravis_@ARAVIS_API_VERSION@_ladir = $(includedir)/aravis-@ARAVIS_API_VERSION@
--- origsrc/aravis-0.5.10/src/Makefile.am	2017-09-17 22:07:35.000000000 +0900
+++ src/aravis-0.5.10/src/Makefile.am	2017-11-27 12:26:33.561334600 +0900
@@ -24,6 +24,7 @@ EXTRA_DIST += arvenumtypes.h.template \
 
 libaravis_@ARAVIS_API_VERSION@_la_LIBADD = 	\
 	$(ARAVIS_LIBS) $(CODE_COVERAGE_LIBS)
+libaravis_@ARAVIS_API_VERSION@_la_LDFLAGS = -no-undefined
 
 ARAVIS_SRCS =					\
 	arvenums.c				\
--- origsrc/aravis-0.5.10/src/arvgvdevice.c	2017-09-17 22:07:35.000000000 +0900
+++ src/aravis-0.5.10/src/arvgvdevice.c	2017-11-27 12:26:33.577786800 +0900
@@ -36,7 +36,7 @@
 #include <arvenumtypes.h>
 #include <string.h>
 #include <stdlib.h>
-#ifndef __APPLE__
+#if defined(__linux__)
 #include <linux/ip.h>
 #endif
 #include <netinet/udp.h>
@@ -45,7 +45,7 @@ static GObjectClass *parent_class = NULL
 
 /* Shared data (main thread - heartbeat) */
 
-#ifdef __APPLE__
+#if !defined(__linux__)
 struct iphdr
   {
 #if __BYTE_ORDER == __LITTLE_ENDIAN
--- origsrc/aravis-0.5.10/src/arvrealtime.c	2017-09-17 22:07:35.000000000 +0900
+++ src/aravis-0.5.10/src/arvrealtime.c	2017-11-27 12:26:33.577786800 +0900
@@ -35,7 +35,9 @@
 #include <sys/time.h>
 #include <sys/types.h>
 #include <sys/resource.h>
+#if !defined(__CYGWIN__)
 #include <sys/syscall.h>
+#endif
 
 #define RTKIT_SERVICE_NAME "org.freedesktop.RealtimeKit1"
 #define RTKIT_OBJECT_PATH "/org/freedesktop/RealtimeKit1"
@@ -219,7 +221,11 @@ arv_rtkit_make_high_priority (GDBusConne
 #endif
 
 static pid_t _gettid(void) {
-        return (pid_t) syscall(SYS_gettid);
+#if defined(__CYGWIN__)
+		return getpid(); 
+#else
+		return (pid_t) syscall(SYS_gettid);
+#endif
 }
 
 /**
