--- origsrc/aravis-0.8.22/src/arvnetwork.c	2022-07-13 23:17:07.000000000 +0900
+++ src/aravis-0.8.22/src/arvnetwork.c	2022-07-23 08:24:43.529559900 +0900
@@ -365,6 +365,7 @@ arv_enumerate_network_interfaces (void)
 			a->addr = arv_memdup (ifap_iter->ifa_addr, sizeof(struct sockaddr));
 			if (ifap_iter->ifa_netmask)
 				a->netmask = arv_memdup (ifap_iter->ifa_netmask, sizeof(struct sockaddr));
+#if !defined(__CYGWIN__)
 #if (defined(__APPLE__) && defined(__MACH__)) || defined(BSD)
 			if (ifap_iter->ifa_broadaddr &&
 			    ifap_iter->ifa_broadaddr->sa_len != 0) {
@@ -381,7 +382,7 @@ arv_enumerate_network_interfaces (void)
 			if (ifap_iter->ifa_ifu.ifu_broadaddr)
 				a->broadaddr = arv_memdup(ifap_iter->ifa_ifu.ifu_broadaddr, sizeof(struct sockaddr));
 #endif
-
+#endif
 			if (ifap_iter->ifa_name)
 				a->name = g_strdup(ifap_iter->ifa_name);
 
--- origsrc/aravis-0.8.22/src/arvrealtime.c	2022-07-13 23:17:07.000000000 +0900
+++ src/aravis-0.8.22/src/arvrealtime.c	2022-07-23 08:24:43.539174300 +0900
@@ -41,7 +41,7 @@
 #include <sys/types.h>
 #endif
 
-#if !defined(__APPLE__) && !defined(G_OS_WIN32) && !defined(BSD)
+#if !defined(__APPLE__) && !defined(G_OS_WIN32) && !defined(BSD) && !defined(__CYGWIN__)
 
 #define RTKIT_SERVICE_NAME "org.freedesktop.RealtimeKit1"
 #define RTKIT_OBJECT_PATH "/org/freedesktop/RealtimeKit1"
