--- origsrc/aravis-0.8.10/src/arvnetwork.c	2021-05-12 15:52:35.204482300 +0900
+++ src/aravis-0.8.10/src/arvnetwork.c	2021-07-03 18:21:12.629056800 +0900
@@ -362,7 +362,7 @@ arv_enumerate_network_interfaces (void)
 			a->addr = arv_memdup (ifap_iter->ifa_addr, sizeof(struct sockaddr));
 			if (ifap_iter->ifa_netmask)
 				a->netmask = arv_memdup (ifap_iter->ifa_netmask, sizeof(struct sockaddr));
-#if !defined(__APPLE__)
+#if !defined(__APPLE__) && !defined(__CYGWIN__)
 			if (ifap_iter->ifa_ifu.ifu_broadaddr)
 				a->broadaddr = arv_memdup(ifap_iter->ifa_ifu.ifu_broadaddr, sizeof(struct sockaddr));
 #endif
--- origsrc/aravis-0.8.10/src/arvnetworkprivate.h	2021-05-12 15:52:35.204482300 +0900
+++ src/aravis-0.8.10/src/arvnetworkprivate.h	2021-07-03 18:18:50.869560800 +0900
@@ -36,7 +36,7 @@
 #include <gio/gnetworking.h>
 #include <gio/gio.h>
 
-#if !defined(__APPLE__) && !defined(G_OS_WIN32)
+#if !defined(__APPLE__) && !defined(G_OS_WIN32) && !defined(__CYGWIN__)
 #include <linux/ip.h>
 #endif
 
@@ -57,7 +57,7 @@ struct udphdr {
 };
 #endif
 
-#if defined(__APPLE__) || defined(G_OS_WIN32)
+#if defined(__APPLE__) || defined(G_OS_WIN32) || defined(__CYGWIN__)
 struct iphdr
   {
 #if __BYTE_ORDER == __LITTLE_ENDIAN
--- origsrc/aravis-0.8.10/src/arvrealtime.c	2021-05-12 15:52:35.205482200 +0900
+++ src/aravis-0.8.10/src/arvrealtime.c	2021-07-03 18:23:48.320449900 +0900
@@ -216,7 +216,7 @@ arv_rtkit_make_high_priority (GDBusConne
 #define RLIMIT_RTTIME 15
 #endif
 
-#if !defined(__APPLE__) && !defined(G_OS_WIN32)
+#if !defined(__APPLE__) && !defined(G_OS_WIN32) && !defined(__CYGWIN__)
 
 #include <sys/resource.h>
 #include <sys/syscall.h>
