--- origsrc/libchaos-1.0/CMakeLists.txt	2016-12-29 10:26:13.000000000 +0900
+++ src/libchaos-1.0/CMakeLists.txt	2017-01-11 01:40:23.014647100 +0900
@@ -77,10 +77,13 @@ ADD_COMPILE_OPTIONS(
 	-Wall -Wextra
 	-Wstrict-aliasing
 	-Wunreachable-code
-	-O3 -flto
+	-O3
 	-mtune=native
 	-march=native
 	)
+IF(NOT CYGWIN)
+	ADD_COMPILE_OPTIONS(-flto)
+ENDIF()
 
 # == COMPONENTS ================================================================
 
@@ -145,6 +148,7 @@ FILE(COPY ${CMAKE_CURRENT_SOURCE_DIR}/te
 
 FILE(GLOB_RECURSE FILES src/*.cc)
 ADD_LIBRARY(chaos ${FILES})
+SET_TARGET_PROPERTIES(chaos PROPERTIES SOVERSION 0)
 
 # Define headers for library. PUBLIC headers are used for compiling
 # the library, and will be added to client build paths.
@@ -193,8 +197,11 @@ IF(LIBCHAOS_ENABLE_TESTING) ############
 		"Enable testing of the testing library.")
 
 	# External package for testing.
+	OPTION(USE_SYSTEM_GTEST "use system gtest" ON)
+	IF(NOT USE_SYSTEM_GTEST)
 	ADD_SUBDIRECTORY(deps/googletest EXCLUDE_FROM_ALL)
 	INCLUDE_DIRECTORIES(${GTEST_SOURCE_DIR}/include ${GTEST_SOURCE_DIR})
+	ENDIF()
 
 	# All files containing tests should be listed here.
 	FILE(GLOB FILES tests/t_*.cc)
@@ -216,8 +223,11 @@ IF(LIBCHAOS_ENABLE_TESTING) ############
 		"Enable testing of the benchmarking library.")
 
 	# External package for benchmarking.
+	OPTION(USE_SYSTEM_BENCHMARK "use system benchmark" ON)
+	IF(NOT USE_SYSTEM_BENCHMARK)
 	ADD_SUBDIRECTORY(deps/benchmark EXCLUDE_FROM_ALL)
 	INCLUDE_DIRECTORIES(${BENCH_SOURCE_DIR}/include ${BENCH_SOURCE_DIR})
+	ENDIF()
 
 	# All files containing tests should be listed here.
 	FILE(GLOB FILES tests/b_*.cc)
@@ -231,3 +241,11 @@ IF(LIBCHAOS_ENABLE_TESTING) ############
 	ADD_TEST(bchaos bchaos)
 
 ENDIF() ############################################################# TESTING ##
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "1.0")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/${PROJECT_NAME}.pc.in ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libchaos-1.0/libchaos.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libchaos-1.0/libchaos.pc.in	2017-01-11 01:40:37.917867400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libchaos
+Description: Advanced library for randomization, hashing and statistical analysis
+Version: @VERSION@
+Libs: -L${libdir} -lchaos
+Cflags: -I${includedir}
