--- origsrc/librelp-1.0.0/configure.ac	2009-12-10 19:42:00.000000000 +0900
+++ src/librelp-1.0.0/configure.ac	2010-05-04 20:45:53.311129600 +0900
@@ -14,6 +14,7 @@ if test "$GCC" = "yes"
 then CFLAGS="$CFLAGS -W -Wall -Wformat-security -Wshadow -Wcast-align -Wpointer-arith -Wmissing-format-attribute -g"
 fi
 
+AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
 
 # Checks for libraries.
--- origsrc/librelp-1.0.0/src/Makefile.am	2009-12-01 20:55:08.000000000 +0900
+++ src/librelp-1.0.0/src/Makefile.am	2010-05-04 20:45:53.311129600 +0900
@@ -35,6 +35,6 @@ librelp_la_SOURCES = \
 	cmdif.h
 librelp_la_CPPFLAGS = $(PTHREADS_CFLAGS)
 librelp_la_LIBADD = $(rt_libs)
-librelp_la_LDFLAGS = -version-info 0:0:0
+librelp_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 
 include_HEADERS = librelp.h
