--- origsrc/librelp-1.0.0/src/Makefile.am	2009-12-01 20:55:08.000000000 +0900
+++ src/librelp-1.0.0/src/Makefile.am	2012-03-31 12:39:20.655182700 +0900
@@ -35,6 +35,6 @@ librelp_la_SOURCES = \
 	cmdif.h
 librelp_la_CPPFLAGS = $(PTHREADS_CFLAGS)
 librelp_la_LIBADD = $(rt_libs)
-librelp_la_LDFLAGS = -version-info 0:0:0
+librelp_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 
 include_HEADERS = librelp.h
