--- origsrc/librelp-1.2.17/src/Makefile.am	2018-07-20 18:58:36.000000000 +0900
+++ src/librelp-1.2.17/src/Makefile.am	2018-08-05 09:06:05.748327900 +0900
@@ -48,6 +48,6 @@ librelp_la_CPPFLAGS = $(AM_CLFAGS) $(PTH
 librelp_la_LIBADD = $(rt_libs) $(GNUTLS_LIBS) $(OPENSSL_LIBS)
 # info on version-info:
 # http://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
-librelp_la_LDFLAGS = -version-info 4:0:4 -export-symbols-regex '^relp'
+librelp_la_LDFLAGS = -no-undefined -version-info 4:0:4 -export-symbols-regex '^relp'
 
 include_HEADERS = librelp.h
--- origsrc/librelp-1.2.17/tests/receive.pid	1970-01-01 09:00:00.000000000 +0900
+++ src/librelp-1.2.17/tests/receive.pid	2018-08-05 09:09:32.578675100 +0900
@@ -0,0 +1 @@
+5304
\ No newline at end of file
