--- origsrc/humanzip-0.5/humanunzip.cpp	2007-05-02 14:22:48.000000000 +0900
+++ src/humanzip-0.5/humanunzip.cpp	2015-04-20 00:19:45.242639100 +0900
@@ -25,6 +25,10 @@ using namespace std;
 #include <sys/stat.h>
 #include "humanzip.h"
 
+#include <cstdlib>
+#include <getopt.h>
+#include <unistd.h>
+
 static int keeporig = 0; // keep the input files?
 static int tostdout = 0; // write to stdout?
 static int overwrite = 0; // overwrite existing files?
--- origsrc/humanzip-0.5/humanzip.cpp	2007-05-02 14:22:48.000000000 +0900
+++ src/humanzip-0.5/humanzip.cpp	2015-04-20 00:19:49.011578900 +0900
@@ -27,6 +27,11 @@ using namespace std;
 #include <sys/stat.h>
 #include "humanzip.h"
 
+#include <algorithm>
+#include <cstdlib>
+#include <getopt.h>
+#include <unistd.h>
+
 #define TABLESIZE 55
 const string utftable[TABLESIZE] = { 
 
