--- origsrc/libMems/configure.ac	2014-09-17 18:55:53.000000000 +0900
+++ src/libMems/configure.ac	2025-08-25 16:33:46.245085100 +0900
@@ -1,6 +1,6 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_PREREQ([2.59])
-AC_INIT(libMems/Match.h)
+AC_INIT([libMems], [1.6.0])
 AC_CONFIG_AUX_DIR(config)
 AC_CONFIG_MACRO_DIR([m4])
 
--- origsrc/libMems/libMems/AbstractGappedAlignment.h	2008-02-21 15:31:12.000000000 +0900
+++ src/libMems/libMems/AbstractGappedAlignment.h	2025-08-25 16:33:46.249726200 +0900
@@ -13,8 +13,8 @@
 #include "config.h"
 #endif
 
-#include "libMems/AbstractMatch.h"
-#include "libGenome/gnFilter.h"
+#include "AbstractMatch.h"
+#include <libGenome/gnFilter.h>
 
 namespace mems {
 
@@ -39,8 +39,8 @@ public:
 	void SetAlignmentLength(gnSeqI len){ align_length = len; }
 
 protected:
-	// for use by derived classes in order to swap contents
-	void swap( AbstractGappedAlignment* other );	
+	// for use by derived classes in order to swap contents
+	void swap( AbstractGappedAlignment* other );	
 private:
 	std::vector< gnSeqI > length;
 	gnSeqI align_length;
@@ -61,12 +61,12 @@ AbstractGappedAlignment<AbstractMatchImp
 }
 
 template<class AbstractMatchImpl>
-void AbstractGappedAlignment<AbstractMatchImpl>::swap( AbstractGappedAlignment* other )
-{
-	std::swap( length, other->length );
-	std::swap( align_length, other->align_length );
-	AbstractMatchImpl::swap( other );
-}
+void AbstractGappedAlignment<AbstractMatchImpl>::swap( AbstractGappedAlignment* other )
+{
+	std::swap( length, other->length );
+	std::swap( align_length, other->align_length );
+	AbstractMatchImpl::swap( other );
+}
 
 template<class AbstractMatchImpl>
 gnSeqI AbstractGappedAlignment<AbstractMatchImpl>::Length( uint seqI ) const 
--- origsrc/libMems/libMems/AbstractMatch.h	2008-01-07 17:20:09.000000000 +0900
+++ src/libMems/libMems/AbstractMatch.h	2025-08-25 16:33:46.253768000 +0900
@@ -13,14 +13,15 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnClone.h"
+#include <libGenome/gnClone.h>
 #include <vector>
 #include <algorithm>
+#include <boost/detail/workaround.hpp>
 #include <boost/type_traits/remove_pointer.hpp>
 #include <boost/type_traits/add_pointer.hpp>
 #include <boost/dynamic_bitset.hpp>
-#include <libMems/SlotAllocator.h>
-#include <libMems/configuration.h>
+#include "SlotAllocator.h"
+#include "configuration.h"
 
 namespace mems {
 
@@ -40,7 +41,7 @@ static const gnSeqI NO_MATCH = 0;
 typedef boost::dynamic_bitset<> bitset_t;
 
 #ifdef _USE_SLOT_ALLOCATOR
-#include "libMems/SlotAllocator.h"
+#include "SlotAllocator.h"
 #elif defined(_USE_BOOST_MATCH_ALLOCATOR)
 #include <boost/pool/pool_alloc.hpp>
 #endif
--- origsrc/libMems/libMems/Aligner.cpp	2008-04-06 21:39:50.000000000 +0900
+++ src/libMems/libMems/Aligner.cpp	2025-08-25 16:33:46.255788400 +0900
@@ -6,13 +6,13 @@
  * **************
  ******************************************************************************/
 
-#include "libMems/Aligner.h"
-#include "libMems/Islands.h"
-#include "libMems/DNAFileSML.h"
-#include "libMems/MuscleInterface.h"	// it's the default gapped aligner
-#include "libGenome/gnRAWSource.h"
-#include "libMems/DistanceMatrix.h"
-#include "libMems/Files.h"
+#include "Aligner.h"
+#include "Islands.h"
+#include "DNAFileSML.h"
+#include "MuscleInterface.h"	// it's the default gapped aligner
+#include <libGenome/gnRAWSource.h>
+#include "DistanceMatrix.h"
+#include "Files.h"
 
 #include <map>
 #include <fstream>	// for debugging
@@ -939,7 +939,7 @@ public:
 	MatchLeftEndComparator( unsigned seq = 0 ){
 		m_seq = seq;
 	}
-	MatchLeftEndComparator( MatchLeftEndComparator& msc ){
+	MatchLeftEndComparator( const MatchLeftEndComparator& msc ){
 		m_seq = msc.m_seq;
 	}
 	// TODO??  make this do a wraparound comparison if all is equal?
--- origsrc/libMems/libMems/Aligner.h	2008-03-15 16:08:53.000000000 +0900
+++ src/libMems/libMems/Aligner.h	2025-08-25 16:33:46.258811400 +0900
@@ -13,20 +13,20 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnSequence.h"
-#include "libMems/DNAMemorySML.h"
-#include "libMems/GappedAligner.h"
-#include "libMems/MatchList.h"
-#include "libMems/Interval.h"
-#include "libMems/IntervalList.h"
-#include "libMems/MemHash.h"
-#include "libMems/MaskedMemHash.h"
+#include <libGenome/gnSequence.h>
+#include "DNAMemorySML.h"
+#include "GappedAligner.h"
+#include "MatchList.h"
+#include "Interval.h"
+#include "IntervalList.h"
+#include "MemHash.h"
+#include "MaskedMemHash.h"
 #include <map>
-#include "libMems/NumericMatrix.h"
-#include "libMems/GreedyBreakpointElimination.h"
+#include "NumericMatrix.h"
+#include "GreedyBreakpointElimination.h"
 #include <list>
-#include "libMems/LCB.h"
-#include "libMUSCLE/threadstorage.h"
+#include "LCB.h"
+#include <libMUSCLE/threadstorage.h>
 
 namespace mems {
 
@@ -49,7 +49,7 @@ public:
 	LabeledMemComparator( uint seq ){
 		m_seq = seq;
 	}
-	LabeledMemComparator( LabeledMemComparator& lmc ){
+	LabeledMemComparator( const LabeledMemComparator& lmc ){
 		m_seq = lmc.m_seq;
 	}
 	boolean operator()(const LabeledMem& a, const LabeledMem& b) const{
--- origsrc/libMems/libMems/Backbone.cpp	2012-06-08 01:37:17.000000000 +0900
+++ src/libMems/libMems/Backbone.cpp	2025-08-25 16:33:46.260826700 +0900
@@ -10,10 +10,10 @@
 #include "config.h"
 #endif
 
-#include "libMems/ProgressiveAligner.h"
-#include "libMems/Backbone.h"
-#include "libMems/Islands.h"
-#include "libMems/CompactGappedAlignment.h"
+#include "ProgressiveAligner.h"
+#include "Backbone.h"
+#include "Islands.h"
+#include "CompactGappedAlignment.h"
 
 #include <boost/graph/graph_traits.hpp>
 #include <boost/graph/adjacency_list.hpp>
--- origsrc/libMems/libMems/Backbone.h	2010-06-04 04:25:43.000000000 +0900
+++ src/libMems/libMems/Backbone.h	2025-08-25 16:33:46.264857900 +0900
@@ -13,15 +13,15 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnSequence.h"
-#include "libMems/SubstitutionMatrix.h"
-#include "libMems/IntervalList.h"
-#include "libMems/NumericMatrix.h"
-#include "libMems/MatchList.h"
-#include "libMems/GappedAlignment.h"
-#include "libMems/CompactGappedAlignment.h"
-#include "libMems/Aligner.h"
-#include "libMems/Islands.h"
+#include <libGenome/gnSequence.h>
+#include "SubstitutionMatrix.h"
+#include "IntervalList.h"
+#include "NumericMatrix.h"
+#include "MatchList.h"
+#include "GappedAlignment.h"
+#include "CompactGappedAlignment.h"
+#include "Aligner.h"
+#include "Islands.h"
 #include <boost/multi_array.hpp>
 
 #include <sstream>
--- origsrc/libMems/libMems/ClustalInterface.cpp	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/ClustalInterface.cpp	2025-08-25 16:33:46.268885400 +0900
@@ -10,14 +10,14 @@
 #include "config.h"
 #endif
 
-#include "libMems/ClustalInterface.h"
+#include "ClustalInterface.h"
 #include <sstream>
-#include "libGenome/gnFilter.h"
+#include <libGenome/gnFilter.h>
 
 #include <fstream>
 
 extern "C" {
-#include "libClustalW/clustalw.h"
+#include <libClustalW/clustalw.h>
 
 extern sint max_names;
 extern Boolean usemenu, dnaflag, explicit_dnaflag;
--- origsrc/libMems/libMems/ClustalInterface.h	2008-03-15 16:08:53.000000000 +0900
+++ src/libMems/libMems/ClustalInterface.h	2025-08-25 16:33:46.278578500 +0900
@@ -13,13 +13,13 @@
 #include "config.h"
 #endif
 
-#include "libMems/NumericMatrix.h"
-#include "libGenome/gnFilter.h"
-#include "libGenome/gnSequence.h"
-#include "libMems/GappedAlignment.h"
-#include "libMems/GappedAligner.h"
+#include "NumericMatrix.h"
+#include <libGenome/gnFilter.h>
+#include <libGenome/gnSequence.h>
+#include "GappedAlignment.h"
+#include "GappedAligner.h"
 
-// attempt to auto-link the ClustalW library on windows
+// attempt to auto-link the ClustalW library on windows
 #if defined(WIN64)&&defined(NDEBUG)&&!defined(FASTDEBUG)&&defined(_OPENMP)
 #pragma comment(lib, "ClustalW64omp.lib")
 #endif
--- origsrc/libMems/libMems/CompactGappedAlignment.h	2014-09-17 18:57:33.000000000 +0900
+++ src/libMems/libMems/CompactGappedAlignment.h	2025-08-25 16:33:46.278578500 +0900
@@ -13,13 +13,13 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnDebug.h"
-#include "libGenome/gnFilter.h"
-#include "libGenome/gnSequence.h"
-#include "libMems/SparseAbstractMatch.h"
-#include "libMems/HybridAbstractMatch.h"
-#include "libMems/AbstractGappedAlignment.h"
-#include "libMems/UngappedLocalAlignment.h"
+#include <libGenome/gnDebug.h>
+#include <libGenome/gnFilter.h>
+#include <libGenome/gnSequence.h>
+#include "SparseAbstractMatch.h"
+#include "HybridAbstractMatch.h"
+#include "AbstractGappedAlignment.h"
+#include "UngappedLocalAlignment.h"
 
 #include <algorithm>
 
--- origsrc/libMems/libMems/DNAFileSML.cpp	2008-03-15 16:08:53.000000000 +0900
+++ src/libMems/libMems/DNAFileSML.cpp	2025-08-25 16:33:46.287651400 +0900
@@ -10,9 +10,9 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnSequence.h"
-#include "libGenome/gnFilter.h"
-#include "libMems/DNAFileSML.h"
+#include <libGenome/gnSequence.h>
+#include <libGenome/gnFilter.h>
+#include "DNAFileSML.h"
 
 using namespace std;
 using namespace genome;
--- origsrc/libMems/libMems/DNAFileSML.h	2008-03-15 16:08:53.000000000 +0900
+++ src/libMems/libMems/DNAFileSML.h	2025-08-25 16:33:46.287651400 +0900
@@ -13,7 +13,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/FileSML.h"
+#include "FileSML.h"
 
 namespace mems {
 
--- origsrc/libMems/libMems/DNAMemorySML.cpp	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/DNAMemorySML.cpp	2025-08-25 16:33:46.287651400 +0900
@@ -10,8 +10,8 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnSequence.h"
-#include "libMems/DNAMemorySML.h"
+#include <libGenome/gnSequence.h>
+#include "DNAMemorySML.h"
 
 using namespace std;
 using namespace genome;
--- origsrc/libMems/libMems/DNAMemorySML.h	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/DNAMemorySML.h	2025-08-25 16:33:46.287651400 +0900
@@ -13,8 +13,8 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnSequence.h"
-#include "libMems/MemorySML.h"
+#include <libGenome/gnSequence.h>
+#include "MemorySML.h"
 
 namespace mems {
 
--- origsrc/libMems/libMems/DenseAbstractMatch.h	2008-02-20 17:17:46.000000000 +0900
+++ src/libMems/libMems/DenseAbstractMatch.h	2025-08-25 16:33:46.287651400 +0900
@@ -13,8 +13,8 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnClone.h"
-#include "libMems/AbstractMatch.h"
+#include <libGenome/gnClone.h>
+#include "AbstractMatch.h"
 #include <limits>
 
 namespace mems {
--- origsrc/libMems/libMems/DistanceMatrix.h	2007-09-11 16:58:42.000000000 +0900
+++ src/libMems/libMems/DistanceMatrix.h	2025-08-25 16:33:46.298497300 +0900
@@ -12,12 +12,12 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnSequence.h"
-#include "libMems/SubstitutionMatrix.h"
-#include "libMems/IntervalList.h"
-#include "libMems/MatchList.h"
-#include "libMems/GappedAlignment.h"
-#include "libMems/CompactGappedAlignment.h"
+#include <libGenome/gnSequence.h>
+#include "SubstitutionMatrix.h"
+#include "IntervalList.h"
+#include "MatchList.h"
+#include "GappedAlignment.h"
+#include "CompactGappedAlignment.h"
 
 
 namespace mems {
--- origsrc/libMems/libMems/FileSML.cpp	2008-02-21 18:59:51.000000000 +0900
+++ src/libMems/libMems/FileSML.cpp	2025-08-25 16:33:46.302522100 +0900
@@ -7,14 +7,14 @@
  ******************************************************************************/
 
 
-#include "libMems/FileSML.h"
+#include "FileSML.h"
 // for CreateTempFileName():
-#include "libMems/Aligner.h"
-#include "libGenome/gnFilter.h"
-#include "libGenome/gnRAWSource.h"
+#include "Aligner.h"
+#include <libGenome/gnFilter.h>
+#include <libGenome/gnRAWSource.h>
 #include <algorithm>
 #include <cmath>
-#include "boost/filesystem/operations.hpp"
+#include <boost/filesystem/operations.hpp>
 
 using namespace std;
 using namespace genome;
@@ -173,7 +173,7 @@ void FileSML::SetID(const sarID_t d){
 
 
 extern "C" {
-#include "libMems/dmSML/dmsort.h"
+#include "dmSML/dmsort.h"
 }
 
 char** FileSML::tmp_paths = NULL;
--- origsrc/libMems/libMems/FileSML.h	2008-01-09 13:01:15.000000000 +0900
+++ src/libMems/libMems/FileSML.h	2025-08-25 16:33:46.322450000 +0900
@@ -17,8 +17,8 @@
 #pragma warning(disable : 4996)
 #pragma warning(pop)
 
-#include "libGenome/gnSequence.h"
-#include "libMems/SortedMerList.h"
+#include <libGenome/gnSequence.h>
+#include "SortedMerList.h"
 #include <boost/iostreams/device/mapped_file.hpp>
 #include <fstream>
 #include <vector>
--- origsrc/libMems/libMems/Files.h	2014-12-21 15:57:42.000000000 +0900
+++ src/libMems/libMems/Files.h	2025-08-25 16:33:46.324486000 +0900
@@ -20,9 +20,9 @@
 #include "unistd.h"
 #endif
 
-#include "boost/filesystem/operations.hpp"
-#include "boost/filesystem/exception.hpp"
-#include "boost/algorithm/string.hpp"
+#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/exception.hpp>
+#include <boost/algorithm/string.hpp>
 #include <string>
 #include <sstream>
 #include <iostream>
--- origsrc/libMems/libMems/GappedAligner.h	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/GappedAligner.h	2025-08-25 16:33:46.328561300 +0900
@@ -13,9 +13,9 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnSequence.h"
-#include "libMems/GappedAlignment.h"
-#include "libMems/Match.h"
+#include <libGenome/gnSequence.h>
+#include "GappedAlignment.h"
+#include "Match.h"
 
 namespace mems {
 
--- origsrc/libMems/libMems/GappedAlignment.cpp	2007-12-14 13:27:53.000000000 +0900
+++ src/libMems/libMems/GappedAlignment.cpp	2025-08-25 16:33:46.330320100 +0900
@@ -10,9 +10,9 @@
 #include "config.h"
 #endif
 
-#include "libMems/GappedAlignment.h"
+#include "GappedAlignment.h"
 #include <sstream>
-#include "libGenome/gnFilter.h"
+#include <libGenome/gnFilter.h>
 
 #include <fstream>
 
--- origsrc/libMems/libMems/GappedAlignment.h	2008-02-21 18:59:03.000000000 +0900
+++ src/libMems/libMems/GappedAlignment.h	2025-08-25 16:33:46.336164000 +0900
@@ -13,11 +13,11 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnFilter.h"
-#include "libGenome/gnSequence.h"
-#include "libMems/SparseAbstractMatch.h"
-#include "libMems/AbstractGappedAlignment.h"
-#include "libMems/Memory.h"
+#include <libGenome/gnFilter.h>
+#include <libGenome/gnSequence.h>
+#include "SparseAbstractMatch.h"
+#include "AbstractGappedAlignment.h"
+#include "Memory.h"
 #include <iostream>
 
 namespace mems {
@@ -66,14 +66,14 @@ public:
 
 	virtual bool IsGap( uint seq, gnSeqI col ) const;
 
-	void swap( GappedAlignment& other ){ swap(&other); }
-
-protected:
-	// for use by derived classes in order to swap contents
-	void swap( GappedAlignment* other ){
-		std::swap( align_matrix, other->align_matrix );
-		AbstractGappedAlignment< SparseAbstractMatch<> >::swap( other );
-	}
+	void swap( GappedAlignment& other ){ swap(&other); }
+
+protected:
+	// for use by derived classes in order to swap contents
+	void swap( GappedAlignment* other ){
+		std::swap( align_matrix, other->align_matrix );
+		AbstractGappedAlignment< SparseAbstractMatch<> >::swap( other );
+	}
 
 	std::vector< std::string > align_matrix;
 
@@ -270,13 +270,13 @@ bool GappedAlignment::IsGap( uint seq, g
 }
 
 
-namespace std {
-template<> inline
-void swap( mems::GappedAlignment& a, mems::GappedAlignment& b )
-{
-	a.swap(b);
-}
-}
+namespace std {
+template<> inline
+void swap( mems::GappedAlignment& a, mems::GappedAlignment& b )
+{
+	a.swap(b);
+}
+}
 
 
 #endif // __GappedAlignment_h__
--- origsrc/libMems/libMems/GreedyBreakpointElimination.cpp	2008-03-14 14:03:23.000000000 +0900
+++ src/libMems/libMems/GreedyBreakpointElimination.cpp	2025-08-25 16:33:46.336164000 +0900
@@ -3,19 +3,19 @@
 #endif
 
 
-#include "libMems/GreedyBreakpointElimination.h"
-#include "libMems/ProgressiveAligner.h"
-#include "libMems/Aligner.h"
-#include "libMems/Islands.h"
-#include "libMems/DNAFileSML.h"
-#include "libMems/MuscleInterface.h"	// it's the default gapped aligner
-#include "libGenome/gnRAWSource.h"
-#include "libMems/gnAlignedSequences.h"
-#include "libMems/CompactGappedAlignment.h"
-#include "libMems/MatchProjectionAdapter.h"
-#include "libMems/PairwiseMatchFinder.h"
-#include "libMems/TreeUtilities.h"
-#include "libMems/PairwiseMatchAdapter.h"
+#include "GreedyBreakpointElimination.h"
+#include "ProgressiveAligner.h"
+#include "Aligner.h"
+#include "Islands.h"
+#include "DNAFileSML.h"
+#include "MuscleInterface.h"	// it's the default gapped aligner
+#include <libGenome/gnRAWSource.h>
+#include "gnAlignedSequences.h"
+#include "CompactGappedAlignment.h"
+#include "MatchProjectionAdapter.h"
+#include "PairwiseMatchFinder.h"
+#include "TreeUtilities.h"
+#include "PairwiseMatchAdapter.h"
 
 #include <boost/dynamic_bitset.hpp>
 #include <boost/tuple/tuple.hpp>
@@ -34,7 +34,7 @@ using namespace genome;
 namespace mems {
 // working in mems
 
-bool penalize_repeats = false;
+bool penalize_repeats = false;
 
 void printProgress( uint prev_prog, uint cur_prog, ostream& os )
 {
@@ -322,7 +322,7 @@ void undoLcbRemoval( uint seq_count, Lcb
 
 EvenFasterSumOfPairsBreakpointScorer::EvenFasterSumOfPairsBreakpointScorer( 
 	double breakpoint_penalty,
-	double minimum_breakpoint_penalty,
+	double minimum_breakpoint_penalty,
 	boost::multi_array<double,2> bp_weight_matrix, 
 	boost::multi_array<double,2> conservation_weight_matrix,
 	vector< TrackingMatch* > tracking_match,
@@ -337,7 +337,7 @@ EvenFasterSumOfPairsBreakpointScorer::Ev
 	size_t seqJ_end
 	) :
   bp_penalty( breakpoint_penalty ),
-  min_breakpoint_penalty( minimum_breakpoint_penalty ),
+  min_breakpoint_penalty( minimum_breakpoint_penalty ),
   bp_weights( bp_weight_matrix ), 
   conservation_weights( conservation_weight_matrix ),
   tracking_matches( tracking_match ),
--- origsrc/libMems/libMems/GreedyBreakpointElimination.h	2008-01-09 13:02:19.000000000 +0900
+++ src/libMems/libMems/GreedyBreakpointElimination.h	2025-08-25 16:34:11.399528100 +0900
@@ -5,14 +5,14 @@
 #ifndef __GreedyBreakpointElimination_h__
 #define __GreedyBreakpointElimination_h__
 
-#include <libMems/AbstractMatch.h>
+#include "AbstractMatch.h"
 #include <iostream>
 #include <boost/multi_array.hpp>
-#include <libMems/PhyloTree.h>
-#include <libMems/SubstitutionMatrix.h>
-#include <libMems/SeedOccurrenceList.h>
-#include <libMems/IntervalList.h>
-#include <libMems/LCB.h>
+#include "PhyloTree.h"
+#include "SubstitutionMatrix.h"
+#include "SeedOccurrenceList.h"
+#include "IntervalList.h"
+#include "LCB.h"
 #include <stack>
 
 namespace mems {
--- origsrc/libMems/libMems/HybridAbstractMatch.h	2008-05-04 09:00:55.000000000 +0900
+++ src/libMems/libMems/HybridAbstractMatch.h	2025-08-25 16:33:46.348475300 +0900
@@ -13,9 +13,9 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnClone.h"
-#include "libGenome/gnDefs.h"
-#include "libMems/AbstractMatch.h"
+#include <libGenome/gnClone.h>
+#include <libGenome/gnDefs.h>
+#include "AbstractMatch.h"
 #include <vector>
 #include <limits>
 #include <cstring>
@@ -93,29 +93,29 @@ protected:
 
 	uint SeqToIndex( uint seqI ) const;
 
-	// for use by derived classes in order to swap contents
-	void swap( HybridAbstractMatch* other );
+	// for use by derived classes in order to swap contents
+	void swap( HybridAbstractMatch* other );
 };
 
 
 template< unsigned FIXED_SEQ_COUNT, class gnSeqIAlloc, class uintAlloc >
-void HybridAbstractMatch< FIXED_SEQ_COUNT, gnSeqIAlloc, uintAlloc >::swap( HybridAbstractMatch* other )
-{
-	std::swap( m_seq_count, other->m_seq_count );
-
-	uint tmp_ids[FIXED_SEQ_COUNT];
-	for( int i = 0; i < FIXED_SEQ_COUNT; i++ ) tmp_ids[i] = other->fixed_seq_ids[i];
-	for( int i = 0; i < FIXED_SEQ_COUNT; i++ ) other->fixed_seq_ids[i] = fixed_seq_ids[i];
-	for( int i = 0; i < FIXED_SEQ_COUNT; i++ ) fixed_seq_ids[i] = tmp_ids[i];
-
-	int64 tmp_starts[FIXED_SEQ_COUNT];
-	for( int i = 0; i < FIXED_SEQ_COUNT; i++ ) tmp_starts[i] = other->fixed_starts[i];
-	for( int i = 0; i < FIXED_SEQ_COUNT; i++ ) other->fixed_starts[i] = fixed_starts[i];
-	for( int i = 0; i < FIXED_SEQ_COUNT; i++ ) fixed_starts[i] = tmp_starts[i];
-
-	std::swap( seq_ids, other->seq_ids );
-	std::swap( starts, other->starts );
-}
+void HybridAbstractMatch< FIXED_SEQ_COUNT, gnSeqIAlloc, uintAlloc >::swap( HybridAbstractMatch* other )
+{
+	std::swap( m_seq_count, other->m_seq_count );
+
+	uint tmp_ids[FIXED_SEQ_COUNT];
+	for( int i = 0; i < FIXED_SEQ_COUNT; i++ ) tmp_ids[i] = other->fixed_seq_ids[i];
+	for( int i = 0; i < FIXED_SEQ_COUNT; i++ ) other->fixed_seq_ids[i] = fixed_seq_ids[i];
+	for( int i = 0; i < FIXED_SEQ_COUNT; i++ ) fixed_seq_ids[i] = tmp_ids[i];
+
+	int64 tmp_starts[FIXED_SEQ_COUNT];
+	for( int i = 0; i < FIXED_SEQ_COUNT; i++ ) tmp_starts[i] = other->fixed_starts[i];
+	for( int i = 0; i < FIXED_SEQ_COUNT; i++ ) other->fixed_starts[i] = fixed_starts[i];
+	for( int i = 0; i < FIXED_SEQ_COUNT; i++ ) fixed_starts[i] = tmp_starts[i];
+
+	std::swap( seq_ids, other->seq_ids );
+	std::swap( starts, other->starts );
+}
 
 template< unsigned FIXED_SEQ_COUNT, class gnSeqIAlloc, class uintAlloc >
 uint HybridAbstractMatch< FIXED_SEQ_COUNT, gnSeqIAlloc, uintAlloc >::FirstStart() const
--- origsrc/libMems/libMems/Interval.cpp	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/Interval.cpp	2025-08-25 16:33:46.352510500 +0900
@@ -10,10 +10,10 @@
 #include "config.h"
 #endif
 
-#include "libMems/Interval.h"
-#include "libMems/gnAlignedSequences.h"
-#include "libMems/GappedAlignment.h"
-#include "libMems/Match.h"
+#include "Interval.h"
+#include "gnAlignedSequences.h"
+#include "GappedAlignment.h"
+#include "Match.h"
 #include <list>
 #include <iterator>
 
--- origsrc/libMems/libMems/Interval.h	2010-04-14 09:06:07.000000000 +0900
+++ src/libMems/libMems/Interval.h	2025-08-25 16:33:46.360654700 +0900
@@ -1,33 +1,33 @@
-/*******************************************************************************
- * $Id: GenericInterval.h,v 1.4 2004/03/01 02:40:08 darling Exp $
- * This file is copyright 2002-2007 Aaron Darling and authors listed in the AUTHORS file.
- * This file is licensed under the GPL.
- * Please see the file called COPYING for licensing details.
- * **************
- ******************************************************************************/
-
-#ifndef __Interval_h__
-#define __Interval_h__
-
-#ifdef HAVE_CONFIG_H
-#include "config.h"
-#endif
-
-#include "libGenome/gnClone.h"
-#include "libGenome/gnDebug.h"
-#include "libMems/SparseAbstractMatch.h"
-#include "libMems/gnAlignedSequences.h"
-#include "libMems/AbstractGappedAlignment.h"
-#include "libMems/Match.h"
-#include "libMems/GappedAlignment.h"
-#include <iostream>
-#include <vector>
-#include "libMems/twister.h"
-
-//#include "boost/pool/object_pool.hpp"
-
-namespace mems {
-
+/*******************************************************************************
+ * $Id: GenericInterval.h,v 1.4 2004/03/01 02:40:08 darling Exp $
+ * This file is copyright 2002-2007 Aaron Darling and authors listed in the AUTHORS file.
+ * This file is licensed under the GPL.
+ * Please see the file called COPYING for licensing details.
+ * **************
+ ******************************************************************************/
+
+#ifndef __Interval_h__
+#define __Interval_h__
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <libGenome/gnClone.h>
+#include <libGenome/gnDebug.h>
+#include "SparseAbstractMatch.h"
+#include "gnAlignedSequences.h"
+#include "AbstractGappedAlignment.h"
+#include "Match.h"
+#include "GappedAlignment.h"
+#include <iostream>
+#include <vector>
+#include "twister.h"
+
+//#include <boost/pool/object_pool.hpp>
+
+namespace mems {
+
 // adapter function to allow inserts on reverse iterators
 template< typename ListType, typename RanIt, typename Ty >
 void insert( ListType& the_list, std::reverse_iterator<RanIt>& riter, Ty& val )
@@ -40,211 +40,211 @@ void insert( ListType& the_list, const t
 {
 	the_list.insert( iter, val );
 }
-
-
-template< class GappedBaseImpl = AbstractGappedAlignment< SparseAbstractMatch<> > >
-class GenericInterval : public GappedBaseImpl
-{
-public:
-	GenericInterval(){};
-
-//	GenericInterval( uint seq_count, gnSeqI aln_length) : GappedBaseImpl( seq_count, aln_length ){};
-
-	/** construct from a MatchList or a vector of pointers to AbstractMatches */
-	template<typename BidIt>
-	GenericInterval( BidIt it_begin, const BidIt& it_end ) : GappedBaseImpl( (*it_begin)->SeqCount(), 0 )
-	{
-		std::vector<gnSeqI> pos((*it_begin)->SeqCount(), NO_MATCH);
-		for( ; it_begin != it_end; ++it_begin )
-			this->matches.push_back( (*it_begin)->Copy() );
-		CalculateOffset();
-		addUnalignedRegions();
-		CalculateAlignmentLength();
-		ValidateMatches();
-	}
-
-	GenericInterval( const GenericInterval& iv );
-	~GenericInterval();
-	GenericInterval& operator=( const GenericInterval& iv );
-	
-	GenericInterval* Clone() const;
+
+
+template< class GappedBaseImpl = AbstractGappedAlignment< SparseAbstractMatch<> > >
+class GenericInterval : public GappedBaseImpl
+{
+public:
+	GenericInterval(){};
+
+//	GenericInterval( uint seq_count, gnSeqI aln_length) : GappedBaseImpl( seq_count, aln_length ){};
+
+	/** construct from a MatchList or a vector of pointers to AbstractMatches */
+	template<typename BidIt>
+	GenericInterval( BidIt it_begin, const BidIt& it_end ) : GappedBaseImpl( (*it_begin)->SeqCount(), 0 )
+	{
+		std::vector<gnSeqI> pos((*it_begin)->SeqCount(), NO_MATCH);
+		for( ; it_begin != it_end; ++it_begin )
+			this->matches.push_back( (*it_begin)->Copy() );
+		CalculateOffset();
+		addUnalignedRegions();
+		CalculateAlignmentLength();
+		ValidateMatches();
+	}
+
+	GenericInterval( const GenericInterval& iv );
+	~GenericInterval();
+	GenericInterval& operator=( const GenericInterval& iv );
+	
+	GenericInterval* Clone() const;
 	GenericInterval* Copy() const;
 	virtual void Free();
-	
-	/** Set the matches in this interval *without* making a copy.  The GenericInterval takes ownership of matches */
-	template< class MatchVector >
-	void SetMatches( MatchVector& matches )
-	{
-		// Set the SeqCount and other bits
-		Match m( matches[0]->SeqCount() );
-		std::vector<AbstractMatch*> tmp(1, &m);
-		*this = GenericInterval( tmp.begin(), tmp.end() );
-
-		// then delete the allocated dummy match
-		for( std::size_t mI = 0; mI < this->matches.size(); mI++ )
-			this->matches[mI]->Free();
-		
-		// now set the matches and update the interval data
-		this->matches.resize(matches.size());
-		std::copy(matches.begin(), matches.end(), this->matches.begin());
-//		this->matches.insert( this->matches.end(), matches.begin(), matches.end() );
-		CalculateOffset();
- 	    addUnalignedRegions();
-		CalculateAlignmentLength();
-		ValidateMatches();
-
-		// finally, clear the user supplied matches to indicate that we own the memory
-		matches.clear();
-	}
-
+	
+	/** Set the matches in this interval *without* making a copy.  The GenericInterval takes ownership of matches */
+	template< class MatchVector >
+	void SetMatches( MatchVector& matches )
+	{
+		// Set the SeqCount and other bits
+		Match m( matches[0]->SeqCount() );
+		std::vector<AbstractMatch*> tmp(1, &m);
+		*this = GenericInterval( tmp.begin(), tmp.end() );
+
+		// then delete the allocated dummy match
+		for( std::size_t mI = 0; mI < this->matches.size(); mI++ )
+			this->matches[mI]->Free();
+		
+		// now set the matches and update the interval data
+		this->matches.resize(matches.size());
+		std::copy(matches.begin(), matches.end(), this->matches.begin());
+//		this->matches.insert( this->matches.end(), matches.begin(), matches.end() );
+		CalculateOffset();
+ 	    addUnalignedRegions();
+		CalculateAlignmentLength();
+		ValidateMatches();
+
+		// finally, clear the user supplied matches to indicate that we own the memory
+		matches.clear();
+	}
+
 	/** Set the matches in this interval *without* cloberring the interval.*/
 	template< class MatchVector >
-	void SetMatchesTemp( MatchVector& matches )
-	{
-		// Set the SeqCount and other bits
-		Match m( matches[0]->SeqCount() );
-		std::vector<AbstractMatch*> tmp(1, &m);
-		*this = GenericInterval( tmp.begin(), tmp.end() );
-
-		// then delete the allocated dummy match
-		for( std::size_t mI = 0; mI < this->matches.size(); mI++ )
-			this->matches[mI]->Free();
-		
-		// now set the matches and update the interval data
-		this->matches.resize(matches.size());
-		std::copy(matches.begin(), matches.end(), this->matches.begin());
-		CalculateOffset();
-		CalculateAlignmentLength();
-		ValidateMatches();
-
-		// finally, clear the user supplied matches to indicate that we own the memory
-		matches.clear();
-	}
-	/**
-	 * Writes this GenericInterval to the specified output stream (e.g. cout).
-	 */
-	template<typename BaseImpl> friend std::ostream& operator<<(std::ostream& os, const GenericInterval<BaseImpl>& iv); //write to source.
-
-	/**
-	 * Reads a GenericInterval from the specified input stream (e.g. cin).
-	 */
-	template<typename BaseImpl> friend std::istream& operator>>(std::istream& is, const GenericInterval<BaseImpl>& iv); //read from source
-
-	// Inherited methods from AbstractMatch:
-	void Invert();
-	void CropStart(gnSeqI crop_amount);
-	void CropEnd(gnSeqI crop_amount);
+	void SetMatchesTemp( MatchVector& matches )
+	{
+		// Set the SeqCount and other bits
+		Match m( matches[0]->SeqCount() );
+		std::vector<AbstractMatch*> tmp(1, &m);
+		*this = GenericInterval( tmp.begin(), tmp.end() );
+
+		// then delete the allocated dummy match
+		for( std::size_t mI = 0; mI < this->matches.size(); mI++ )
+			this->matches[mI]->Free();
+		
+		// now set the matches and update the interval data
+		this->matches.resize(matches.size());
+		std::copy(matches.begin(), matches.end(), this->matches.begin());
+		CalculateOffset();
+		CalculateAlignmentLength();
+		ValidateMatches();
+
+		// finally, clear the user supplied matches to indicate that we own the memory
+		matches.clear();
+	}
+	/**
+	 * Writes this GenericInterval to the specified output stream (e.g. cout).
+	 */
+	template<typename BaseImpl> friend std::ostream& operator<<(std::ostream& os, const GenericInterval<BaseImpl>& iv); //write to source.
+
+	/**
+	 * Reads a GenericInterval from the specified input stream (e.g. cin).
+	 */
+	template<typename BaseImpl> friend std::istream& operator>>(std::istream& is, const GenericInterval<BaseImpl>& iv); //read from source
+
+	// Inherited methods from AbstractMatch:
+	void Invert();
+	void CropStart(gnSeqI crop_amount);
+	void CropEnd(gnSeqI crop_amount);
 	void MoveStart(int64 move_amount);
 	void MoveEnd(int64 move_amount);
-
-	virtual void CalculateOffset();
-
-	void add( AbstractMatch* am ){ matches.push_back( am->Copy() ); }
-	
-	/** 
-	 * Get a gnAlignedSequences object
-	 * TODO: get rid of this
-	 */
-	virtual void GetAlignedSequences( gnAlignedSequences& gnas, const std::vector< genome::gnSequence* >& seq_table ) const;
-
+
+	virtual void CalculateOffset();
+
+	void add( AbstractMatch* am ){ matches.push_back( am->Copy() ); }
+	
+	/** 
+	 * Get a gnAlignedSequences object
+	 * TODO: get rid of this
+	 */
+	virtual void GetAlignedSequences( gnAlignedSequences& gnas, const std::vector< genome::gnSequence* >& seq_table ) const;
+
 	void GetAlignment( std::vector< bitset_t >& align_matrix ) const;
-
-	void CropLeft( gnSeqI amount, uint seqI );
-	void CropRight( gnSeqI amount, uint seqI );
-
-	void SetAlignment( const std::vector< std::string >& seq_align );
-
-	// TODO: get rid of code that uses this hack...
-	const std::vector<AbstractMatch*>& GetMatches() const{ return matches; }
-	void StealMatches( std::vector<AbstractMatch*>& matches );
-
+
+	void CropLeft( gnSeqI amount, uint seqI );
+	void CropRight( gnSeqI amount, uint seqI );
+
+	void SetAlignment( const std::vector< std::string >& seq_align );
+
+	// TODO: get rid of code that uses this hack...
+	const std::vector<AbstractMatch*>& GetMatches() const{ return matches; }
+	void StealMatches( std::vector<AbstractMatch*>& matches );
+
 	/** marbles the gaps so that no sequence has more than "size" contiguous gaps */
 	void Marble( gnSeqI size );
 
-	void GetColumn( gnSeqI col, std::vector<gnSeqI>& pos, std::vector<bool>& column ) const;
-	
+	void GetColumn( gnSeqI col, std::vector<gnSeqI>& pos, std::vector<bool>& column ) const;
+	
 	bool IsGap( uint seq, gnSeqI col ) const;
-
-	/** self test code */
-	void ValidateMatches() const;
-
-	void swap( GenericInterval& other ){ swap(&other); }
-
-protected:
-	// for use by derived classes in order to swap contents
-	void swap( GenericInterval* other ){
-		std::swap( matches, other->matches );
-		GappedBaseImpl::swap( other );
-	}
-	std::vector< AbstractMatch* > matches;
-private:
-	void addUnalignedRegions();
-	void FindMatchPos( uint seqI, gnSeqI pos, size_t& matchI, gnSeqI& match_pos );
-	void GetColumnAndMatch( gnSeqI col, std::vector<gnSeqI>& pos, std::vector<bool>& column, size_t& matchI, gnSeqI& match_col ) const;
-	void CalculateAlignmentLength();
-};
-
-typedef GenericInterval<> Interval;
-
-
-template<class GappedBaseImpl>
+
+	/** self test code */
+	void ValidateMatches() const;
+
+	void swap( GenericInterval& other ){ swap(&other); }
+
+protected:
+	// for use by derived classes in order to swap contents
+	void swap( GenericInterval* other ){
+		std::swap( matches, other->matches );
+		GappedBaseImpl::swap( other );
+	}
+	std::vector< AbstractMatch* > matches;
+private:
+	void addUnalignedRegions();
+	void FindMatchPos( uint seqI, gnSeqI pos, size_t& matchI, gnSeqI& match_pos );
+	void GetColumnAndMatch( gnSeqI col, std::vector<gnSeqI>& pos, std::vector<bool>& column, size_t& matchI, gnSeqI& match_col ) const;
+	void CalculateAlignmentLength();
+};
+
+typedef GenericInterval<> Interval;
+
+
+template<class GappedBaseImpl>
 GenericInterval<GappedBaseImpl>* GenericInterval<GappedBaseImpl>::Copy() const
 {
 	return m_allocateAndCopy( *this );
 }
-template<class GappedBaseImpl>
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::Free()
 {
 	m_free(this);
 }
-
-template<class GappedBaseImpl>
-GenericInterval<GappedBaseImpl>::~GenericInterval()
-{
-	for( std::size_t mI = 0; mI < matches.size(); mI++ )
-		matches[mI]->Free();
-}
-
-template<class GappedBaseImpl>
-void GenericInterval<GappedBaseImpl>::StealMatches( std::vector<AbstractMatch*>& matches ){
-	matches = this->matches;
-	this->matches.clear();
-	for( uint seqI = 0; seqI < this->SeqCount(); seqI++ )
-	{
-		this->SetLeftEnd( seqI, NO_MATCH );
-		this->SetLength( 0, seqI );
-	}
-	this->SetAlignmentLength(0);
-}
-
-template<class GappedBaseImpl>
-GenericInterval<GappedBaseImpl>::GenericInterval( const GenericInterval<GappedBaseImpl>& iv )
-{
-	*this = iv;
-}
-
-template<class GappedBaseImpl>
-GenericInterval<GappedBaseImpl>& GenericInterval<GappedBaseImpl>::operator=( const GenericInterval& iv )
-{
-	GappedBaseImpl::operator=( iv );
-	for( std::size_t mI = 0; mI < matches.size(); mI++ )
-		matches[mI]->Free();
-	matches.clear();
-	for( std::size_t mI = 0; mI < iv.matches.size(); mI++ )
-		matches.push_back( iv.matches[mI]->Copy() );
-	return *this;
-}
-
-template<class GappedBaseImpl>
-GenericInterval<GappedBaseImpl>* GenericInterval<GappedBaseImpl>::Clone() const 
-{
-	return new GenericInterval( *this );
-}
-
-
-static bool debug_interval = false;
-
-template<class GappedBaseImpl>
+
+template<class GappedBaseImpl>
+GenericInterval<GappedBaseImpl>::~GenericInterval()
+{
+	for( std::size_t mI = 0; mI < matches.size(); mI++ )
+		matches[mI]->Free();
+}
+
+template<class GappedBaseImpl>
+void GenericInterval<GappedBaseImpl>::StealMatches( std::vector<AbstractMatch*>& matches ){
+	matches = this->matches;
+	this->matches.clear();
+	for( uint seqI = 0; seqI < this->SeqCount(); seqI++ )
+	{
+		this->SetLeftEnd( seqI, NO_MATCH );
+		this->SetLength( 0, seqI );
+	}
+	this->SetAlignmentLength(0);
+}
+
+template<class GappedBaseImpl>
+GenericInterval<GappedBaseImpl>::GenericInterval( const GenericInterval<GappedBaseImpl>& iv )
+{
+	*this = iv;
+}
+
+template<class GappedBaseImpl>
+GenericInterval<GappedBaseImpl>& GenericInterval<GappedBaseImpl>::operator=( const GenericInterval& iv )
+{
+	GappedBaseImpl::operator=( iv );
+	for( std::size_t mI = 0; mI < matches.size(); mI++ )
+		matches[mI]->Free();
+	matches.clear();
+	for( std::size_t mI = 0; mI < iv.matches.size(); mI++ )
+		matches.push_back( iv.matches[mI]->Copy() );
+	return *this;
+}
+
+template<class GappedBaseImpl>
+GenericInterval<GappedBaseImpl>* GenericInterval<GappedBaseImpl>::Clone() const 
+{
+	return new GenericInterval( *this );
+}
+
+
+static bool debug_interval = false;
+
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::ValidateMatches() const
 {
 	if( !debug_interval )
@@ -298,7 +298,7 @@ void GenericInterval<GappedBaseImpl>::Va
 	}
 }
 
-template<class GappedBaseImpl>
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::GetColumnAndMatch( gnSeqI col, std::vector<gnSeqI>& pos, std::vector<bool>& column, size_t& matchI, gnSeqI& match_col ) const
 {
 	// bail when the appropriate match is found
@@ -346,7 +346,7 @@ void GenericInterval<GappedBaseImpl>::Ge
 	}
 }
 
-template<typename ListType, typename Iter>
+template<typename ListType, typename Iter>
 void AddGapMatches( ListType& the_list, const Iter& first, const Iter& last, 
 				   uint seqI, int64 left_end, int64 right_end, 
 				   AbstractMatch::orientation seq_orient, uint seq_count )
@@ -408,7 +408,7 @@ void AddGapMatches( ListType& the_list,
 }
 
 // The best steaks are well marbled
-template<class GappedBaseImpl>
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::Marble( gnSeqI size )
 {
 	if( this->SeqCount() > 2 )
@@ -501,7 +501,7 @@ void GenericInterval<GappedBaseImpl>::Ma
 }
 
 
-template<class GappedBaseImpl>
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::CropStart(gnSeqI crop_amount)
 {
 	if( crop_amount > this->AlignmentLength() )
@@ -526,8 +526,8 @@ void GenericInterval<GappedBaseImpl>::Cr
 	this->CalculateOffset();
 	this->ValidateMatches();
 }
-
-template<class GappedBaseImpl>
+
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::CropEnd(gnSeqI crop_amount)
 {
 	if( crop_amount > this->AlignmentLength() )
@@ -552,9 +552,9 @@ void GenericInterval<GappedBaseImpl>::Cr
 
 	this->CalculateOffset();
 	this->ValidateMatches();
-}
-
-template<class GappedBaseImpl>
+}
+
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::GetAlignment( std::vector< bitset_t >& align_matrix ) const
 {
 	gnSeqI cur_col = 0;
@@ -578,8 +578,8 @@ void GenericInterval<GappedBaseImpl>::Ge
 		cur_col += matches[matchI]->AlignmentLength();
 	}
 }
-
-template<class GappedBaseImpl>
+
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::CropLeft( gnSeqI amount, uint seqI )
 {
 	if( amount > this->Length(seqI) )
@@ -630,7 +630,7 @@ void GenericInterval<GappedBaseImpl>::Cr
 	}
 }
 
-template<class GappedBaseImpl>
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::CropRight( gnSeqI amount, uint seqI )
 {
 	if( amount > this->Length(seqI) )
@@ -682,8 +682,8 @@ void GenericInterval<GappedBaseImpl>::Cr
 		genome::breakHere();
 	}
 }
-
-template<class GappedBaseImpl>
+
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::MoveStart(int64 move_amount)
 {
 	GappedBaseImpl::MoveStart(move_amount);
@@ -691,18 +691,18 @@ void GenericInterval<GappedBaseImpl>::Mo
 		matches[mI]->MoveStart(move_amount);
 }
 
-template<class GappedBaseImpl>
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::MoveEnd(int64 move_amount)
 {
 	GappedBaseImpl::MoveEnd(move_amount);
 	for( size_t mI = 0; mI < matches.size(); mI++ )
 		matches[mI]->MoveEnd(move_amount);
 }
-
-
-template< class MatchVector >
-void FindBoundaries( const MatchVector& matches, std::vector<gnSeqI>& left_ends, std::vector<gnSeqI>& lengths, std::vector<bool>& orientations )
-{
+
+
+template< class MatchVector >
+void FindBoundaries( const MatchVector& matches, std::vector<gnSeqI>& left_ends, std::vector<gnSeqI>& lengths, std::vector<bool>& orientations )
+{
 	uint seqI;
 	boolean zero_exists = false;
 	uint seq_count = matches.front()->SeqCount();
@@ -779,10 +779,10 @@ void FindBoundaries( const MatchVector&
 		if( !zero_exists )
 			break;
 	}
-}
-
+}
+
 
-template<class GappedBaseImpl>
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::addUnalignedRegions()
 {
 	std::list<AbstractMatch*> new_matches(matches.begin(), matches.end());
@@ -800,7 +800,7 @@ void GenericInterval<GappedBaseImpl>::ad
 	matches.insert(matches.end(), new_matches.begin(), new_matches.end());
 }
 
-template<class GappedBaseImpl>
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::Invert(){
 	GappedBaseImpl::Invert();
 	for( uint matchI = 0; matchI < matches.size(); ++matchI )
@@ -809,7 +809,7 @@ void GenericInterval<GappedBaseImpl>::In
 	std::reverse( matches.begin(), matches.end() );
 }
 
-template<class GappedBaseImpl>
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::GetColumn( gnSeqI col, std::vector<gnSeqI>& pos, std::vector<bool>& column ) const
 {
 	size_t matchI;
@@ -817,7 +817,7 @@ void GenericInterval<GappedBaseImpl>::Ge
 	this->GetColumnAndMatch( col, pos, column, matchI, match_col );
 }
 
-template<class GappedBaseImpl>
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::FindMatchPos( uint seqI, gnSeqI pos, size_t& matchI, gnSeqI& match_pos )
 {
 	match_pos = pos;
@@ -847,35 +847,35 @@ void GenericInterval<GappedBaseImpl>::Fi
 		matchI--;
 }
 
-template<class GappedBaseImpl>
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::CalculateOffset(){
 	std::vector<gnSeqI> left_end( this->SeqCount(), NO_MATCH );
 	std::vector<gnSeqI> length( this->SeqCount(), 0 );
 	std::vector<bool> orientation;
 	if( this->matches.size() > 0 )
-		FindBoundaries( this->matches, left_end, length, orientation );
-	for( uint seqI = 0; seqI < this->SeqCount(); seqI++ )
-	{
-		if( left_end[seqI] != 0 )
-		{
-			this->SetLeftEnd(seqI, left_end[seqI]);
-			this->SetLength(length[seqI], seqI);
-			if( orientation[seqI] )
-				this->SetOrientation(seqI, AbstractMatch::forward);
-			else
-				this->SetOrientation(seqI, AbstractMatch::reverse);
-		}else if( this->LeftEnd(seqI) != NO_MATCH )
-		{
-			this->SetLength(0, seqI);
-			this->SetLeftEnd(seqI, NO_MATCH);
-		}
-
-	}
+		FindBoundaries( this->matches, left_end, length, orientation );
+	for( uint seqI = 0; seqI < this->SeqCount(); seqI++ )
+	{
+		if( left_end[seqI] != 0 )
+		{
+			this->SetLeftEnd(seqI, left_end[seqI]);
+			this->SetLength(length[seqI], seqI);
+			if( orientation[seqI] )
+				this->SetOrientation(seqI, AbstractMatch::forward);
+			else
+				this->SetOrientation(seqI, AbstractMatch::reverse);
+		}else if( this->LeftEnd(seqI) != NO_MATCH )
+		{
+			this->SetLength(0, seqI);
+			this->SetLeftEnd(seqI, NO_MATCH);
+		}
+
+	}
 
 	this->CalculateAlignmentLength();
 }
 
-template<class GappedBaseImpl>
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::SetAlignment( const std::vector< std::string >& seq_align )
 {
 	GappedAlignment* ga = new GappedAlignment(seq_align.size(), seq_align[0].size());
@@ -893,7 +893,7 @@ void GenericInterval<GappedBaseImpl>::Se
 /**
  * Writes this GenericInterval to the specified output stream (e.g. cout).
  */
-template<class GappedBaseImpl>
+template<class GappedBaseImpl>
 std::ostream& operator<<(std::ostream& os, const GenericInterval<GappedBaseImpl>& cr){
 	try{
 	for( uint matchI = 0; matchI < cr.matches.size(); ++matchI ){
@@ -912,7 +912,7 @@ std::ostream& operator<<(std::ostream& o
 	return os;
 }
 
-template<class GappedBaseImpl>
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::CalculateAlignmentLength()
 {
 	gnSeqI aln_len = 0;
@@ -922,7 +922,7 @@ void GenericInterval<GappedBaseImpl>::Ca
 	this->SetAlignmentLength(aln_len);
 }
 
-template<class GappedBaseImpl>
+template<class GappedBaseImpl>
 void GenericInterval<GappedBaseImpl>::GetAlignedSequences( gnAlignedSequences& gnas, const std::vector< genome::gnSequence* >& seq_table ) const 
 {
 	gnas.names.clear();
@@ -935,8 +935,8 @@ void GenericInterval<GappedBaseImpl>::Ge
 	}
 	mems::GetAlignment( *this, seq_table, gnas.sequences );
 }
-
-template<class GappedBaseImpl>
+
+template<class GappedBaseImpl>
 bool GenericInterval<GappedBaseImpl>::IsGap( uint seq, gnSeqI col ) const
 {
 	std::vector<gnSeqI> pos;
@@ -944,15 +944,15 @@ bool GenericInterval<GappedBaseImpl>::Is
 	GetColumn(col, pos, column);
 	return column[seq];
 }
-
-}
-
-namespace std {
-template<> inline
-void swap( mems::Interval& a, mems::Interval& b )
-{
-	a.swap(b);
-}
-}
-
-#endif	// __Interval_h__
+
+}
+
+namespace std {
+template<> inline
+void swap( mems::Interval& a, mems::Interval& b )
+{
+	a.swap(b);
+}
+}
+
+#endif	// __Interval_h__
--- origsrc/libMems/libMems/IntervalList.cpp	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/IntervalList.cpp	2025-08-25 16:33:46.364433500 +0900
@@ -10,10 +10,10 @@
 #include "config.h"
 #endif
 
-#include "libMems/IntervalList.h"
-#include "libMems/DNAFileSML.h"
-#include "libMems/MemHash.h"
-#include "libMems/GappedAlignment.h"
+#include "IntervalList.h"
+#include "DNAFileSML.h"
+#include "MemHash.h"
+#include "GappedAlignment.h"
 
 #include <map>
 #include <sstream>
--- origsrc/libMems/libMems/IntervalList.h	2014-09-17 18:57:33.000000000 +0900
+++ src/libMems/libMems/IntervalList.h	2025-08-25 16:33:46.368488800 +0900
@@ -17,16 +17,16 @@
 #include <list>
 #include <sstream>
 
-#include "libMems/SortedMerList.h"
-#include "libGenome/gnSequence.h"
-#include "libMems/Interval.h"
-#include "libMems/MemHash.h"
-#include "libMems/CompactGappedAlignment.h"
-#include "libGenome/gnSourceFactory.h"
-#include "libGenome/gnFASSource.h"
-#include "libGenome/gnSEQSource.h"
-#include "libGenome/gnGBKSource.h"
-#include "libGenome/gnRAWSource.h"
+#include "SortedMerList.h"
+#include <libGenome/gnSequence.h>
+#include "Interval.h"
+#include "MemHash.h"
+#include "CompactGappedAlignment.h"
+#include <libGenome/gnSourceFactory.h>
+#include <libGenome/gnFASSource.h>
+#include <libGenome/gnSEQSource.h>
+#include <libGenome/gnGBKSource.h>
+#include <libGenome/gnRAWSource.h>
 
 namespace mems {
 
--- origsrc/libMems/libMems/Islands.cpp	2007-10-18 05:12:17.000000000 +0900
+++ src/libMems/libMems/Islands.cpp	2025-08-25 16:33:46.370506100 +0900
@@ -10,9 +10,9 @@
 #include "config.h"
 #endif
 
-#include "libMems/Islands.h"
-#include "libMems/Aligner.h"
-#include "libMems/GappedAlignment.h"
+#include "Islands.h"
+#include "Aligner.h"
+#include "GappedAlignment.h"
 
 using namespace std;
 using namespace genome;
--- origsrc/libMems/libMems/Islands.h	2008-04-06 21:41:57.000000000 +0900
+++ src/libMems/libMems/Islands.h	2025-08-25 16:33:46.370506100 +0900
@@ -13,17 +13,17 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnSequence.h"
-#include "libMems/SubstitutionMatrix.h"
-#include "libMems/IntervalList.h"
-#include "libMems/NumericMatrix.h"
-#include "libMems/MatchList.h"
-#include "libMems/GappedAlignment.h"
-#include "libMems/CompactGappedAlignment.h"
-#include "libMems/Aligner.h"
+#include <libGenome/gnSequence.h>
+#include "SubstitutionMatrix.h"
+#include "IntervalList.h"
+#include "NumericMatrix.h"
+#include "MatchList.h"
+#include "GappedAlignment.h"
+#include "CompactGappedAlignment.h"
+#include "Aligner.h"
 #include <boost/multi_array.hpp>
-#include "libMems/HomologyHMM/homology.h"
-#include "libMems/Scoring.h"
+#include "HomologyHMM/homology.h"
+#include "Scoring.h"
 
 namespace mems {
 
--- origsrc/libMems/libMems/Makefile.am	2014-09-17 18:55:53.000000000 +0900
+++ src/libMems/libMems/Makefile.am	2025-08-25 16:33:46.372529800 +0900
@@ -62,7 +62,8 @@ libmems_include_HEADERS = $(LIBMEMS_H)
 lib_LTLIBRARIES = libMems-1.6.la
 libMems_1_6_la_SOURCES = $(LIBMEMS_SRC) $(HOMOLOGYHMM_SRC) $(DMSML_SRC)
 
-libMems_1_6_la_LDFLAGS= -version-info $(GENERIC_LIBRARY_VERSION)  
+libMems_1_6_la_LDFLAGS= -no-undefined -version-info $(GENERIC_LIBRARY_VERSION)  
+libMems_1_6_la_LIBADD = $(DEPS_LIBS) $(BOOST_FILESYSTEM_LIBS) $(BOOST_IOSTREAMS_LIBS) $(BOOST_SYSTEM_LIBS)
 
 homologyhmm_includedir=$(includedir)/$(GENERIC_LIBRARY_NAME)-$(GENERIC_API_VERSION)/$(GENERIC_LIBRARY_NAME)/HomologyHMM
 homologyhmm_include_HEADERS = $(HOMOLOGYHMM_H)
--- origsrc/libMems/libMems/MaskedMemHash.cpp	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/MaskedMemHash.cpp	2025-08-25 16:33:46.374540100 +0900
@@ -10,7 +10,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/MaskedMemHash.h"
+#include "MaskedMemHash.h"
 #include <list>
 
 using namespace std;
--- origsrc/libMems/libMems/MaskedMemHash.h	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/MaskedMemHash.h	2025-08-25 16:33:46.378299500 +0900
@@ -13,7 +13,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/MemHash.h"
+#include "MemHash.h"
 
 namespace mems {
 
--- origsrc/libMems/libMems/Match.h	2008-01-07 17:20:09.000000000 +0900
+++ src/libMems/libMems/Match.h	2025-08-25 16:33:46.380310500 +0900
@@ -13,13 +13,13 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnClone.h"
+#include <libGenome/gnClone.h>
 #include <iostream>
 #include <set>
-#include "libMems/UngappedLocalAlignment.h"
-#include "libMems/SparseAbstractMatch.h"
-#include "libMems/DenseAbstractMatch.h"
-#include "libMems/HybridAbstractMatch.h"
+#include "UngappedLocalAlignment.h"
+#include "SparseAbstractMatch.h"
+#include "DenseAbstractMatch.h"
+#include "HybridAbstractMatch.h"
 
 namespace mems {
 
--- origsrc/libMems/libMems/MatchFinder.cpp	2008-05-30 01:27:13.000000000 +0900
+++ src/libMems/libMems/MatchFinder.cpp	2025-08-25 16:33:46.384342800 +0900
@@ -10,7 +10,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/MatchFinder.h"
+#include "MatchFinder.h"
 
 using namespace std;
 using namespace genome;
--- origsrc/libMems/libMems/MatchFinder.h	2010-04-14 03:10:19.000000000 +0900
+++ src/libMems/libMems/MatchFinder.h	2025-08-25 16:33:46.386368900 +0900
@@ -13,9 +13,9 @@
 #include "config.h"
 #endif
 
-#include "libMems/SortedMerList.h"
-#include "libMems/Match.h"
-#include "libMems/MatchList.h"
+#include "SortedMerList.h"
+#include "Match.h"
+#include "MatchList.h"
 #include <list>
 #include <iostream>
 #include <boost/pool/pool_alloc.hpp>
--- origsrc/libMems/libMems/MatchHashEntry.cpp	2007-12-13 11:48:47.000000000 +0900
+++ src/libMems/libMems/MatchHashEntry.cpp	2025-08-25 16:33:46.388389300 +0900
@@ -10,9 +10,9 @@
 #include "config.h"
 #endif
 
-#include "libMems/MatchHashEntry.h"
-#include "libGenome/gnException.h"
-#include "libGenome/gnDebug.h"
+#include "MatchHashEntry.h"
+#include <libGenome/gnException.h>
+#include <libGenome/gnDebug.h>
 
 using namespace std;
 using namespace genome;
--- origsrc/libMems/libMems/MatchHashEntry.h	2008-01-07 17:20:09.000000000 +0900
+++ src/libMems/libMems/MatchHashEntry.h	2025-08-25 16:33:46.396989600 +0900
@@ -13,10 +13,10 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnClone.h"
+#include <libGenome/gnClone.h>
 #include <iostream>
 #include <set>
-#include "libMems/Match.h"
+#include "Match.h"
 
 namespace mems {
 
--- origsrc/libMems/libMems/MatchList.cpp	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/MatchList.cpp	2025-08-25 16:33:46.396989600 +0900
@@ -10,10 +10,10 @@
 #include "config.h"
 #endif
 
-#include "libMems/MatchList.h"
-#include "libMems/DNAFileSML.h"
-#include "libMems/DNAMemorySML.h"
-#include "libMems/MemHash.h"
+#include "MatchList.h"
+#include "DNAFileSML.h"
+#include "DNAMemorySML.h"
+#include "MemHash.h"
 #include <map>
 #include <sstream>
 #include <ctime>
--- origsrc/libMems/libMems/MatchList.h	2008-05-16 01:04:20.000000000 +0900
+++ src/libMems/libMems/MatchList.h	2025-08-25 16:33:46.413828600 +0900
@@ -15,14 +15,14 @@
 
 #include <iostream>
 #include <list>
-#include "libMems/SortedMerList.h"
-#include "libMems/DNAFileSML.h"
-#include "libMems/DNAMemorySML.h"
-#include "libGenome/gnSequence.h"
-#include "libMems/Match.h"
-#include "libMems/gnRAWSequence.h"
-#include "libGenome/gnRAWSource.h"
-#include "libMems/Files.h"
+#include "SortedMerList.h"
+#include "DNAFileSML.h"
+#include "DNAMemorySML.h"
+#include <libGenome/gnSequence.h>
+#include "Match.h"
+#include "gnRAWSequence.h"
+#include <libGenome/gnRAWSource.h>
+#include "Files.h"
 #include <sstream>
 #include <map>
 #include <ctime>
--- origsrc/libMems/libMems/MatchProjectionAdapter.h	2008-01-07 17:20:09.000000000 +0900
+++ src/libMems/libMems/MatchProjectionAdapter.h	2025-08-25 16:33:46.415840600 +0900
@@ -9,7 +9,7 @@
 #ifndef __MatchProjectionAdapter_h__
 #define __MatchProjectionAdapter_h__
 
-#include "libMems/AbstractMatch.h"
+#include "AbstractMatch.h"
 #include <vector>
 
 #ifdef HAVE_CONFIG_H
--- origsrc/libMems/libMems/Matrix.h	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/Matrix.h	2025-08-25 16:33:46.419889500 +0900
@@ -13,7 +13,7 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnSetup.h"
+#include <libGenome/gnSetup.h>
 #include <iostream>
 #include <sstream>
 #include <vector>
@@ -25,7 +25,7 @@ class Matrix
 public:
 	Matrix();
 	Matrix(unsigned nrows, unsigned ncols);
-	// Throws a BadSize object if either size is zero
+	//ï¿½Throws a BadSize object if either size is zero
 	class BadSize : public std::range_error{
 	public:
 		BadSize() : std::range_error( "Bad matrix size" ){}
--- origsrc/libMems/libMems/MemHash.cpp	2010-08-19 08:36:58.000000000 +0900
+++ src/libMems/libMems/MemHash.cpp	2025-08-25 16:33:46.421925000 +0900
@@ -10,8 +10,8 @@
 #include "config.h"
 #endif
 
-#include "libMems/MemHash.h"
-#include "libGenome/gnFilter.h"
+#include "MemHash.h"
+#include <libGenome/gnFilter.h>
 #include <list>
 #include <map>
 #include <sstream>
--- origsrc/libMems/libMems/MemHash.h	2008-02-21 13:31:11.000000000 +0900
+++ src/libMems/libMems/MemHash.h	2025-08-25 16:33:46.428008800 +0900
@@ -17,13 +17,13 @@
 #include <map>
 #include <iostream>
 
-#include "libMems/MatchFinder.h"
-#include "libMems/Match.h"
-#include "libGenome/gnException.h"
-#include "libMems/MatchList.h"
-#include "libMems/MatchHashEntry.h"
-#include "libMems/SlotAllocator.h"
-#include "boost/pool/object_pool.hpp"
+#include "MatchFinder.h"
+#include "Match.h"
+#include <libGenome/gnException.h>
+#include "MatchList.h"
+#include "MatchHashEntry.h"
+#include "SlotAllocator.h"
+#include <boost/pool/object_pool.hpp>
 
 namespace mems {
 
--- origsrc/libMems/libMems/MemorySML.cpp	2008-03-15 16:08:53.000000000 +0900
+++ src/libMems/libMems/MemorySML.cpp	2025-08-25 16:33:46.431801200 +0900
@@ -10,7 +10,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/MemorySML.h"
+#include "MemorySML.h"
 #include <algorithm>
 
 using namespace std;
--- origsrc/libMems/libMems/MemorySML.h	2008-03-15 16:08:53.000000000 +0900
+++ src/libMems/libMems/MemorySML.h	2025-08-25 16:33:46.431801200 +0900
@@ -13,8 +13,8 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnSequence.h"
-#include "libMems/SortedMerList.h"
+#include <libGenome/gnSequence.h>
+#include "SortedMerList.h"
 
 namespace mems {
 
--- origsrc/libMems/libMems/MuscleInterface.cpp	2008-05-27 02:22:37.000000000 +0900
+++ src/libMems/libMems/MuscleInterface.cpp	2025-08-25 16:33:46.437848000 +0900
@@ -10,25 +10,25 @@
 #include "config.h"
 #endif
 
-#include "libMems/MuscleInterface.h"
+#include "MuscleInterface.h"
 
-#include "libGenome/gnFilter.h"
-#include "libGenome/gnFASSource.h"
-#include "libGenome/gnStringTools.h"
-#include "libMUSCLE/muscle.h"
-#include "libMUSCLE/params.h"
-#include "libMUSCLE/msa.h"
-#include "libMUSCLE/seq.h"
-#include "libMUSCLE/seqvect.h"
-#include "libMUSCLE/tree.h"
-#include "libMUSCLE/clust.h"
-#include "libMUSCLE/profile.h"
-#include "libMUSCLE/distfunc.h"
-#include "libMUSCLE/clustsetdf.h"
-#include "libMUSCLE/textfile.h"
-#include "libMUSCLE/types.h"
-#include "boost/algorithm/string/erase.hpp"
-#include "boost/algorithm/string/case_conv.hpp"
+#include <libGenome/gnFilter.h>
+#include <libGenome/gnFASSource.h>
+#include <libGenome/gnStringTools.h>
+#include <libMUSCLE/muscle.h>
+#include <libMUSCLE/params.h>
+#include <libMUSCLE/msa.h>
+#include <libMUSCLE/seq.h>
+#include <libMUSCLE/seqvect.h>
+#include <libMUSCLE/tree.h>
+#include <libMUSCLE/clust.h>
+#include <libMUSCLE/profile.h>
+#include <libMUSCLE/distfunc.h>
+#include <libMUSCLE/clustsetdf.h>
+#include <libMUSCLE/textfile.h>
+#include <libMUSCLE/types.h>
+#include <boost/algorithm/string/erase.hpp>
+#include <boost/algorithm/string/case_conv.hpp>
 
 #include <sstream>
 #include <fstream>
--- origsrc/libMems/libMems/MuscleInterface.h	2008-05-27 01:42:42.000000000 +0900
+++ src/libMems/libMems/MuscleInterface.h	2025-08-25 16:33:46.439859100 +0900
@@ -13,13 +13,13 @@
 #include "config.h"
 #endif
 
-#include "libMems/NumericMatrix.h"
-#include "libGenome/gnFilter.h"
-#include "libGenome/gnSequence.h"
-#include "libMems/GappedAlignment.h"
-#include "libMems/GappedAligner.h"
+#include "NumericMatrix.h"
+#include <libGenome/gnFilter.h>
+#include <libGenome/gnSequence.h>
+#include "GappedAlignment.h"
+#include "GappedAligner.h"
 
-// attempt to auto-link the MUSCLE library on windows
+// attempt to auto-link the MUSCLE library on windows
 #if defined(WIN64)&&defined(NDEBUG)&&!defined(FASTDEBUG)&&defined(_OPENMP)
 #pragma comment(lib, "MUSCLE64omp.lib")
 #endif
@@ -103,14 +103,14 @@ public:
 	 * ga2 must have equal sequence count and contain disjoint sets of sequences, e.g. for any given
 	 * seqI, if ga1.LeftEnd(seqI) != NO_MATCH, then ga2.LeftEnd(seqI) == NO_MATCH 
 	 */
-	bool ProfileAlign( const GappedAlignment& ga1, const GappedAlignment& ga2, GappedAlignment& aln, bool anchored = true );
+	bool ProfileAlign( const GappedAlignment& ga1, const GappedAlignment& ga2, GappedAlignment& aln, bool anchored = true );
 	boolean CallMuscle( std::vector< std::string >& aln_matrix, const std::vector< std::string >& seq_table );
 	boolean CallMuscleFast( std::vector< std::string >& aln_matrix, const std::vector< std::string >& seq_table, int gap_open = 0, int gap_extend = 0);
-	bool RefineFast( GappedAlignment& ga, size_t windowsize = 0 );
-	bool ProfileAlignFast( const GappedAlignment& ga1, const GappedAlignment& ga2, GappedAlignment& aln, bool anchored = true );
-
-	void CreateTree( const NumericMatrix<double>& distances, const std::string& tree_filename  );
-
+	bool RefineFast( GappedAlignment& ga, size_t windowsize = 0 );
+	bool ProfileAlignFast( const GappedAlignment& ga1, const GappedAlignment& ga2, GappedAlignment& aln, bool anchored = true );
+
+	void CreateTree( const NumericMatrix<double>& distances, const std::string& tree_filename  );
+
 protected:
 	std::string muscle_path;
 	std::string muscle_arguments;
@@ -140,7 +140,7 @@ private:
 };
 
 
-void stripGapColumns( std::vector< std::string >& aln );
+void stripGapColumns( std::vector< std::string >& aln );
 
 
 }
--- origsrc/libMems/libMems/NumericMatrix.h	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/NumericMatrix.h	2025-08-25 16:33:46.455725800 +0900
@@ -13,7 +13,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/Matrix.h"
+#include "Matrix.h"
 
 template<class T>  // See section on templates for more
 class NumericMatrix : public Matrix<T>
--- origsrc/libMems/libMems/PairwiseMatchAdapter.h	2008-01-07 17:20:09.000000000 +0900
+++ src/libMems/libMems/PairwiseMatchAdapter.h	2025-08-25 16:33:46.457738700 +0900
@@ -1,8 +1,8 @@
 #ifndef __PairwiseMatchAdapter_h__
 #define __PairwiseMatchAdapter_h__
 
-#include "libMems/AbstractMatch.h"
-#include "libMems/ProgressiveAligner.h"
+#include "AbstractMatch.h"
+#include "ProgressiveAligner.h"
 #include <vector>
 
 namespace mems {
--- origsrc/libMems/libMems/PairwiseMatchFinder.cpp	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/PairwiseMatchFinder.cpp	2025-08-25 16:33:46.459755300 +0900
@@ -10,7 +10,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/PairwiseMatchFinder.h"
+#include "PairwiseMatchFinder.h"
 #include <list>
 
 using namespace std;
--- origsrc/libMems/libMems/PairwiseMatchFinder.h	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/PairwiseMatchFinder.h	2025-08-25 16:33:46.461512200 +0900
@@ -13,9 +13,9 @@
 #include "config.h"
 #endif
 
-#include "libMems/MemHash.h"
+#include "MemHash.h"
 
-namespace mems {
+namespace mems {
 
 /**
  * Finds all pairwise matches with unique seeds among a group of sequences
--- origsrc/libMems/libMems/ParallelMemHash.cpp	2008-01-14 10:47:21.000000000 +0900
+++ src/libMems/libMems/ParallelMemHash.cpp	2025-08-25 16:33:46.465559800 +0900
@@ -10,7 +10,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/ParallelMemHash.h"
+#include "ParallelMemHash.h"
 #include <vector>
 
 #ifdef _OPENMP
--- origsrc/libMems/libMems/ParallelMemHash.h	2008-01-13 13:21:03.000000000 +0900
+++ src/libMems/libMems/ParallelMemHash.h	2025-08-25 16:33:46.467575700 +0900
@@ -15,9 +15,9 @@
 
 #ifdef _OPENMP
 
-#include "libMUSCLE/threadstorage.h"
+#include <libMUSCLE/threadstorage.h>
 #include <omp.h>
-#include "libMems/MemHash.h"
+#include "MemHash.h"
 
 namespace mems {
 
--- origsrc/libMems/libMems/ProgressiveAligner.cpp	2012-06-08 01:37:17.000000000 +0900
+++ src/libMems/libMems/ProgressiveAligner.cpp	2025-08-25 16:33:46.473640900 +0900
@@ -11,19 +11,19 @@
 #endif
 
 
-#include "libMems/ProgressiveAligner.h"
-#include "libMems/GreedyBreakpointElimination.h"
-#include "libMems/Aligner.h"
-#include "libMems/Islands.h"
-#include "libMems/DNAFileSML.h"
-#include "libMems/MuscleInterface.h"	// it's the default gapped aligner
-#include "libMems/gnAlignedSequences.h"
-#include "libMems/CompactGappedAlignment.h"
-#include "libMems/MatchProjectionAdapter.h"
-#include "libMems/PairwiseMatchFinder.h"
-#include "libMems/TreeUtilities.h"
-#include "libMems/PairwiseMatchAdapter.h"
-#include "libMems/DistanceMatrix.h"
+#include "ProgressiveAligner.h"
+#include "GreedyBreakpointElimination.h"
+#include "Aligner.h"
+#include "Islands.h"
+#include "DNAFileSML.h"
+#include "MuscleInterface.h"	// it's the default gapped aligner
+#include "gnAlignedSequences.h"
+#include "CompactGappedAlignment.h"
+#include "MatchProjectionAdapter.h"
+#include "PairwiseMatchFinder.h"
+#include "TreeUtilities.h"
+#include "PairwiseMatchAdapter.h"
+#include "DistanceMatrix.h"
 
 #include <boost/dynamic_bitset.hpp>
 #include <boost/tuple/tuple.hpp>
@@ -1599,7 +1599,7 @@ void ProgressiveAligner::constructLcbTra
 		c.SetLeftEnd(child_1, ancestral_matches[mI]->LeftEnd(1));
 		c.SetOrientation(child_1, ancestral_matches[mI]->Orientation(1));
 		c.SetLength(ancestral_matches[mI]->Length(1), child_1);
-		cga_list.push_back(make_tuple(c.Copy(), &bs[mI], ancestral_matches[mI]));
+		cga_list.push_back(boost::make_tuple(c.Copy(), &bs[mI], ancestral_matches[mI]));
 	}
 
 	stack<node_id_t> node_stack;
--- origsrc/libMems/libMems/ProgressiveAligner.h	2008-03-12 15:34:33.000000000 +0900
+++ src/libMems/libMems/ProgressiveAligner.h	2025-08-25 16:33:46.477436000 +0900
@@ -13,17 +13,17 @@
 #include "config.h"
 #endif
 
-#include "libMems/SuperInterval.h"
-#include "libMems/Aligner.h"
-#include "libMems/PhyloTree.h"
-#include "libMems/GreedyBreakpointElimination.h"
-#include "libMems/CompactGappedAlignment.h"
-#include "libMems/Islands.h"
+#include "SuperInterval.h"
+#include "Aligner.h"
+#include "PhyloTree.h"
+#include "GreedyBreakpointElimination.h"
+#include "CompactGappedAlignment.h"
+#include "Islands.h"
 #include <boost/type_traits/remove_pointer.hpp>
 #include <boost/multi_array.hpp>
-#include "libMems/SeedOccurrenceList.h"
-#include "libMems/SubstitutionMatrix.h"
-#include "libMems/MatchProjectionAdapter.h"
+#include "SeedOccurrenceList.h"
+#include "SubstitutionMatrix.h"
+#include "MatchProjectionAdapter.h"
 
 namespace mems
 {
@@ -36,7 +36,7 @@ extern bool debug_aligner;
 class AlignmentTreeNode : public TreeNode
 {
 public:
-	AlignmentTreeNode() : TreeNode(), refined(false) {};
+	AlignmentTreeNode() : TreeNode(), refined(false) {};
 	std::vector< SuperInterval > ordering;	/**< A total ordering on alignments of sequence contained by leafs below this node */
 	std::vector< boolean > parents_aligned;		/**< have parents been aligned? */
 	std::vector< boolean > children_aligned;	/**< have children been aligned? */
@@ -76,7 +76,8 @@ public:
 	void setOutputGuideTreeFileName( std::string& fname ){ this->output_guide_tree_fname = fname; }
 	/** set the max length (in columns) of alignments passed to MUSCLE */
 	void SetMaxGappedAlignmentLength( size_t len );
-	/** set whether a cache database should be used to speed up recursive anchor search */ 
+	/** set whether a cache database should be used to speed up recursive anchor search */
+ 
 	void SetUseCacheDb( bool cbd ){ this->using_cache_db = cbd; }
 
 	/** Set whether iterative refinement using MUSCLE should be performed (true/false) */
@@ -396,8 +397,8 @@ void EliminateOverlaps_v2( MatchVector&
 template <class MatchVector>
 void EliminateOverlaps_v2( MatchVector& ml, bool eliminate_both = false )
 {
-	if( ml.size() < 2 )
-		return;	// can't eliminate overlaps between fewer than 2 matches
+	if( ml.size() < 2 )
+		return;	// can't eliminate overlaps between fewer than 2 matches
 	uint seq_count = ml[0]->SeqCount();
 	std::vector< uint > seq_ids( seq_count );
 	for( uint i = 0; i < seq_count; ++i )
@@ -630,8 +631,8 @@ void applyBreakpoints( std::vector< gnSe
 
 }
 
-//namespace std {
-//	void swap( PhyloTree<mems::AlignmentTreeNode>& a, PhyloTree<mems::AlignmentTreeNode>& b);
-//}
+//namespace std {
+//	void swap( PhyloTree<mems::AlignmentTreeNode>& a, PhyloTree<mems::AlignmentTreeNode>& b);
+//}
 
 #endif // _ProgressiveAligner_h_
--- origsrc/libMems/libMems/RepeatHash.cpp	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/RepeatHash.cpp	2025-08-25 16:33:46.479450500 +0900
@@ -10,7 +10,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/RepeatHash.h"
+#include "RepeatHash.h"
 #include <list>
 
 using namespace std;
--- origsrc/libMems/libMems/RepeatHash.h	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/RepeatHash.h	2025-08-25 16:33:46.515007900 +0900
@@ -13,7 +13,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/MemHash.h"
+#include "MemHash.h"
 
 namespace mems {
 
--- origsrc/libMems/libMems/RepeatMatch.cpp	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/RepeatMatch.cpp	2025-08-25 16:33:46.534725900 +0900
@@ -10,9 +10,9 @@
 #include "config.h"
 #endif
 
-#include "libMems/RepeatMatch.h"
-#include "libGenome/gnException.h"
-#include "libGenome/gnDebug.h"
+#include "RepeatMatch.h"
+#include <libGenome/gnException.h>
+#include <libGenome/gnDebug.h>
 
 namespace mems {
 
--- origsrc/libMems/libMems/RepeatMatch.h	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/RepeatMatch.h	2025-08-25 16:33:46.538758300 +0900
@@ -13,11 +13,11 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnClone.h"
+#include <libGenome/gnClone.h>
 #include <iostream>
 #include <vector>
 #include <set>
-#include "libMems/MatchHashEntry.h"
+#include "MatchHashEntry.h"
 
 namespace mems {
 
--- origsrc/libMems/libMems/RepeatMatchList.cpp	2007-12-02 15:27:57.000000000 +0900
+++ src/libMems/libMems/RepeatMatchList.cpp	2025-08-25 16:33:46.540770300 +0900
@@ -10,10 +10,10 @@
 #include "config.h"
 #endif
 
-#include "libMems/RepeatMatchList.h"
-#include "libMems/DNAFileSML.h"
-#include "libMems/DNAMemorySML.h"
-#include "libMems/MemHash.h"
+#include "RepeatMatchList.h"
+#include "DNAFileSML.h"
+#include "DNAMemorySML.h"
+#include "MemHash.h"
 #include <map>
 #include <sstream>
 #include <ctime>
--- origsrc/libMems/libMems/RepeatMatchList.h	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/RepeatMatchList.h	2025-08-25 16:33:46.542781800 +0900
@@ -15,10 +15,10 @@
 
 #include <iostream>
 #include <list>
-#include "libMems/SortedMerList.h"
-#include "libGenome/gnSequence.h"
-#include "libMems/Match.h"
-#include "libMems/MatchList.h"
+#include "SortedMerList.h"
+#include <libGenome/gnSequence.h>
+#include "Match.h"
+#include "MatchList.h"
 //#include <valarray>
 
 namespace mems {
--- origsrc/libMems/libMems/Scoring.h	2007-07-30 10:58:41.000000000 +0900
+++ src/libMems/libMems/Scoring.h	2025-08-25 16:33:46.542781800 +0900
@@ -12,7 +12,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/SubstitutionMatrix.h"
+#include "SubstitutionMatrix.h"
 #include <string>
 #include <vector>
 
--- origsrc/libMems/libMems/SeedMasks.h	2008-03-22 17:21:05.000000000 +0900
+++ src/libMems/libMems/SeedMasks.h	2025-08-25 16:33:46.544794100 +0900
@@ -18,7 +18,7 @@
 #else
 #include <math.h>
 #endif
-#include "libGenome/gnDefs.h"
+#include <libGenome/gnDefs.h>
 
 /* Seed patterns taken from: AE Darling, T Treangen, L Zhang, C Kuiken, X Messeguer, NT Perna
  * "Procrastination leads to efficient match filtration for local multiple alignment" 
--- origsrc/libMems/libMems/SeedOccurrenceList.h	2015-02-13 14:38:25.000000000 +0900
+++ src/libMems/libMems/SeedOccurrenceList.h	2025-08-25 16:33:46.548558500 +0900
@@ -2,11 +2,11 @@
 #define __SeedOccurrenceList_h__
 
 #include <vector>
-#include "libMems/SortedMerList.h"
+#include "SortedMerList.h"
 #include <boost/iostreams/device/mapped_file.hpp>
 #include <boost/filesystem.hpp>
 #include <fstream>
-#include "libMems/Files.h"
+#include "Files.h"
 
 namespace mems
 {
--- origsrc/libMems/libMems/SlotAllocator.h	2009-09-03 09:29:15.000000000 +0900
+++ src/libMems/libMems/SlotAllocator.h	2025-08-25 16:33:46.548558500 +0900
@@ -17,7 +17,7 @@
 #include <list>
 #include <stdexcept>
 #include <iostream>
-#include "libMUSCLE/threadstorage.h"
+#include <libMUSCLE/threadstorage.h>
 
 
 namespace mems {
--- origsrc/libMems/libMems/SortedMerList.cpp	2009-11-04 08:27:17.000000000 +0900
+++ src/libMems/libMems/SortedMerList.cpp	2025-08-25 16:33:46.550568000 +0900
@@ -10,7 +10,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/SortedMerList.h"
+#include "SortedMerList.h"
 
 using namespace std;
 using namespace genome;
--- origsrc/libMems/libMems/SortedMerList.h	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/SortedMerList.h	2025-08-25 16:33:46.552606800 +0900
@@ -13,15 +13,15 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnDefs.h"
-#include "libGenome/gnClone.h"
-#include "libGenome/gnDebug.h"
-#include "libGenome/gnSequence.h"
-#include "libGenome/gnException.h"
-#include "stdlib.h"
+#include <libGenome/gnDefs.h>
+#include <libGenome/gnClone.h>
+#include <libGenome/gnDebug.h>
+#include <libGenome/gnSequence.h>
+#include <libGenome/gnException.h>
+#include <stdlib.h>
 #include <string>
 #include <vector>
-#include "libMems/SeedMasks.h"
+#include "SeedMasks.h"
 
 namespace mems {
 
--- origsrc/libMems/libMems/SparseAbstractMatch.h	2008-02-21 15:31:12.000000000 +0900
+++ src/libMems/libMems/SparseAbstractMatch.h	2025-08-25 16:33:46.554628900 +0900
@@ -13,9 +13,9 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnClone.h"
-#include "libGenome/gnDefs.h"
-#include "libMems/AbstractMatch.h"
+#include <libGenome/gnClone.h>
+#include <libGenome/gnDefs.h>
+#include "AbstractMatch.h"
 #include <vector>
 #include <limits>
 
@@ -67,8 +67,8 @@ protected:
 	bitset_t orient;	// bitset_t has its own allocator
 	uint SeqToIndex( uint seqI ) const;
 
-	// for use by derived classes in order to swap contents
-	void swap( SparseAbstractMatch* other );	
+	// for use by derived classes in order to swap contents
+	void swap( SparseAbstractMatch* other );	
 };
 
 
@@ -78,13 +78,13 @@ m_seq_count(seq_count)
 {}
 
 template< class gnSeqIAlloc, class uintAlloc >
-void SparseAbstractMatch< gnSeqIAlloc, uintAlloc >::swap( SparseAbstractMatch* other )
-{
-	std::swap(seq_ids, other->seq_ids);
-	std::swap(m_seq_count, other->m_seq_count);
-	std::swap(leftend, other->leftend);
-	std::swap(orient, other->orient);
-}
+void SparseAbstractMatch< gnSeqIAlloc, uintAlloc >::swap( SparseAbstractMatch* other )
+{
+	std::swap(seq_ids, other->seq_ids);
+	std::swap(m_seq_count, other->m_seq_count);
+	std::swap(leftend, other->leftend);
+	std::swap(orient, other->orient);
+}
 
 template< class gnSeqIAlloc, class uintAlloc >
 uint SparseAbstractMatch< gnSeqIAlloc, uintAlloc >::FirstStart() const
--- origsrc/libMems/libMems/SubstitutionMatrix.h	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/SubstitutionMatrix.h	2025-08-25 16:33:46.556641900 +0900
@@ -13,7 +13,7 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnSequence.h"
+#include <libGenome/gnSequence.h>
 #include <iostream>
 #include <sstream>
 
--- origsrc/libMems/libMems/SuperInterval.cpp	2007-07-09 14:41:05.000000000 +0900
+++ src/libMems/libMems/SuperInterval.cpp	2025-08-25 16:33:46.560681100 +0900
@@ -1,9 +1,9 @@
-#ifdef HAVE_CONFIG_H
-#include "config.h"
-#endif
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 
-#include "libMems/SuperInterval.h"
+#include "SuperInterval.h"
 
 using namespace std;
 using namespace genome;
--- origsrc/libMems/libMems/SuperInterval.h	2008-02-21 15:31:12.000000000 +0900
+++ src/libMems/libMems/SuperInterval.h	2025-08-25 16:33:46.562527500 +0900
@@ -1,11 +1,11 @@
-#ifdef HAVE_CONFIG_H
-#include "config.h"
-#endif
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #ifndef __SuperInterval_h__
 #define __SuperInterval_h__
 
-#include "libMems/Interval.h"
+#include "Interval.h"
 
 namespace mems {
 
@@ -70,12 +70,12 @@ protected:
 
 } // namespace mems
 
-namespace std {
+namespace std {
 template<> inline
 void swap( mems::SuperInterval& a, mems::SuperInterval& b )
 {
 	a.swap(b);
 }
-}
+}
 
 #endif //__SuperInterval_h__
--- origsrc/libMems/libMems/UngappedLocalAlignment.h	2008-01-07 17:20:09.000000000 +0900
+++ src/libMems/libMems/UngappedLocalAlignment.h	2025-08-25 16:33:46.566556700 +0900
@@ -13,9 +13,9 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnClone.h"
-#include "libGenome/gnException.h"
-#include "libMems/AbstractMatch.h"
+#include <libGenome/gnClone.h>
+#include <libGenome/gnException.h>
+#include "AbstractMatch.h"
 
 namespace mems {
 
--- origsrc/libMems/libMems/dmSML/aPOSIXaio.c	2005-02-27 07:06:53.000000000 +0900
+++ src/libMems/libMems/dmSML/aPOSIXaio.c	2025-08-25 16:33:46.582421600 +0900
@@ -2,10 +2,10 @@
 #include "config.h"
 #endif
 
-#include "libMems/dmSML/aPOSIXaio.h"
+#include "aPOSIXaio.h"
 #ifdef USE_POSIX_AIO
 
-#include "libMems/dmSML/asyncio.h"
+#include "asyncio.h"
 #include <fcntl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
--- origsrc/libMems/libMems/dmSML/aPOSIXaio.h	2005-02-27 07:06:53.000000000 +0900
+++ src/libMems/libMems/dmSML/aPOSIXaio.h	2025-08-25 16:33:46.586486400 +0900
@@ -5,7 +5,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/dmSML/asyncio.h"
+#include "asyncio.h"
 
 int OpenPAIO( aFILE * file, const char *path, int mode );
 int ClosePAIO( aFILE * file );
--- origsrc/libMems/libMems/dmSML/alibc.c	2005-02-27 07:06:53.000000000 +0900
+++ src/libMems/libMems/dmSML/alibc.c	2025-08-25 16:33:46.588508000 +0900
@@ -2,8 +2,8 @@
 #include "config.h"
 #endif
 
-#include "libMems/dmSML/asyncio.h"
-#include "libMems/dmSML/alibc.h"
+#include "asyncio.h"
+#include "alibc.h"
 
 #if defined USE_LIBC
 
--- origsrc/libMems/libMems/dmSML/alinuxaio.c	2005-02-27 07:06:53.000000000 +0900
+++ src/libMems/libMems/dmSML/alinuxaio.c	2025-08-25 16:33:46.592553100 +0900
@@ -2,12 +2,12 @@
 #include "config.h"
 #endif
 
-#include "libMems/dmSML/alinuxaio.h"
+#include "alinuxaio.h"
 #ifdef USE_LINUX_AIO
 
 #include <libaio.h>
 
-#include "libMems/dmSML/asyncio.h"
+#include "asyncio.h"
 #include <fcntl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
--- origsrc/libMems/libMems/dmSML/alinuxaio.h	2005-02-27 07:06:53.000000000 +0900
+++ src/libMems/libMems/dmSML/alinuxaio.h	2025-08-25 16:33:46.594311100 +0900
@@ -5,7 +5,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/dmSML/asyncio.h"
+#include "asyncio.h"
 
 
 int OpenLinux( aFILE * file, const char *path, int mode );
--- origsrc/libMems/libMems/dmSML/asyncio.c	2005-02-27 07:06:53.000000000 +0900
+++ src/libMems/libMems/dmSML/asyncio.c	2025-08-25 16:33:46.594311100 +0900
@@ -2,15 +2,15 @@
 #include "config.h"
 #endif
 
-#include "libMems/dmSML/asyncio.h"
+#include "asyncio.h"
 
-#include "libMems/dmSML/alibc.h"
-#include "libMems/dmSML/awin32aio.h"
-#include "libMems/dmSML/aPOSIXaio.h"
+#include "alibc.h"
+#include "awin32aio.h"
+#include "aPOSIXaio.h"
 
-#include "libMems/dmSML/util.h"
+#include "util.h"
 
-#include "libMems/dmSML/buffer.h"
+#include "buffer.h"
 #include <string.h>
 
 #if defined(USE_POSIX_AIO)||defined(USE_LINUX_AIO)
--- origsrc/libMems/libMems/dmSML/awin32aio.c	2005-02-27 07:06:53.000000000 +0900
+++ src/libMems/libMems/dmSML/awin32aio.c	2025-08-25 16:33:46.600482700 +0900
@@ -2,8 +2,8 @@
 #include "config.h"
 #endif
 
-#include "libMems/dmSML/awin32aio.h"
-#include "libMems/dmSML/util.h"
+#include "awin32aio.h"
+#include "util.h"
 #ifdef USE_WIN32
 
 
--- origsrc/libMems/libMems/dmSML/awin32aio.h	2005-02-27 07:06:53.000000000 +0900
+++ src/libMems/libMems/dmSML/awin32aio.h	2025-08-25 16:33:46.604504200 +0900
@@ -5,7 +5,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/dmSML/asyncio.h"
+#include "asyncio.h"
 
 int OpenWIN32( aFILE * file, const char *path, int mode );
 int CloseWIN32( aFILE * file );
--- origsrc/libMems/libMems/dmSML/buffer.c	2005-02-27 07:06:53.000000000 +0900
+++ src/libMems/libMems/dmSML/buffer.c	2025-08-25 16:33:46.606521400 +0900
@@ -4,7 +4,7 @@
 
 #include <time.h>
 #include <stddef.h>
-#include "libMems/dmSML/buffer.h"
+#include "buffer.h"
 #include <string.h>
 
 // portably fills an int with reasonably random bits.
--- origsrc/libMems/libMems/dmSML/buffer.h	2005-02-27 07:06:53.000000000 +0900
+++ src/libMems/libMems/dmSML/buffer.h	2025-08-25 16:33:46.610285700 +0900
@@ -5,7 +5,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/dmSML/asyncio.h"
+#include "asyncio.h"
 
 // forward decl for the benefit of iodevice_t
 // (can't be avoided)
--- origsrc/libMems/libMems/dmSML/dmsort.c	2005-02-27 07:06:53.000000000 +0900
+++ src/libMems/libMems/dmSML/dmsort.c	2025-08-25 16:33:46.616350700 +0900
@@ -5,13 +5,13 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
-#include "libMems/dmSML/util.h"
-#include "libMems/dmSML/timing.h"
-#include "libMems/dmSML/asyncio.h"
-#include "libMems/dmSML/buffer.h"
-#include "libMems/dmSML/sorting.h"
-#include "libMems/dmSML/sml.h"
-#include "libMems/dmSML/dmsort.h"
+#include "util.h"
+#include "timing.h"
+#include "asyncio.h"
+#include "buffer.h"
+#include "sorting.h"
+#include "sml.h"
+#include "dmsort.h"
 
 // define this if you're using the ASCII sortgen data.
 // don't define if you're using random data (dmsortgen)
@@ -825,7 +825,7 @@ int InitdmSML( long working_mb, long buf
 		if ( fgets(buf, sizeof(buf), fp) )
 		{
 			sscanf(buf, "MemTotal: %ld kB", &memTotal);
-			fprintf( stderr, buf );
+			fprintf( stderr, "%s", buf );
 		}
 		fclose(fp);
 		// allocate about 6/10 of physical memory
--- origsrc/libMems/libMems/dmSML/dmsort.h	2008-02-21 18:59:51.000000000 +0900
+++ src/libMems/libMems/dmSML/dmsort.h	2025-08-25 16:33:46.626156900 +0900
@@ -8,12 +8,12 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
-#include "libMems/dmSML/util.h"
-#include "libMems/dmSML/timing.h"
-#include "libMems/dmSML/asyncio.h"
-#include "libMems/dmSML/buffer.h"
-#include "libMems/dmSML/sorting.h"
-#include "libMems/dmSML/sml.h"
+#include "util.h"
+#include "timing.h"
+#include "asyncio.h"
+#include "buffer.h"
+#include "sorting.h"
+#include "sml.h"
 
 // define this if you're using the ASCII sortgen data.
 // don't define if you're using random data (dmsortgen)
--- origsrc/libMems/libMems/dmSML/sml.c	2005-02-27 07:06:53.000000000 +0900
+++ src/libMems/libMems/dmSML/sml.c	2025-08-25 16:33:46.645780700 +0900
@@ -2,8 +2,8 @@
 #include "config.h"
 #endif
 
-#include "libMems/dmSML/sml.h"
-#include "libMems/SeedMasks.h"
+#include "sml.h"
+#include "SeedMasks.h"
 
 
 SMLHeader_t InitSML( aFILE* file, uint64 file_size, uint64 seed ){
--- origsrc/libMems/libMems/dmSML/sml.h	2006-01-06 09:57:50.000000000 +0900
+++ src/libMems/libMems/dmSML/sml.h	2025-08-25 16:33:46.703222800 +0900
@@ -5,9 +5,9 @@
 #include "config.h"
 #endif
 
-#include "libMems/dmSML/asyncio.h"
+#include "asyncio.h"
 #include <string.h>
-#include "libGenome/gnDefs.h"
+#include <libGenome/gnDefs.h>
 
 #ifndef UINT8_MAX
 
--- origsrc/libMems/libMems/dmSML/sorting.c	2005-02-27 07:06:53.000000000 +0900
+++ src/libMems/libMems/dmSML/sorting.c	2025-08-25 16:33:46.705236300 +0900
@@ -2,7 +2,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/dmSML/sorting.h"
+#include "sorting.h"
 #include "math.h"
 #include <string.h>
 
--- origsrc/libMems/libMems/dmSML/sorting.h	2005-02-27 07:06:53.000000000 +0900
+++ src/libMems/libMems/dmSML/sorting.h	2025-08-25 16:33:46.707246700 +0900
@@ -5,7 +5,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/dmSML/buffer.h"
+#include "buffer.h"
 
 
 // define this if you want to use the qsort only version
--- origsrc/libMems/libMems/dmSML/timing.c	2005-02-27 07:06:53.000000000 +0900
+++ src/libMems/libMems/dmSML/timing.c	2025-08-25 16:33:46.711011200 +0900
@@ -12,8 +12,8 @@
 #include <stdio.h>
 
 
-#include "libMems/dmSML/util.h"
-#include "libMems/dmSML/timing.h"
+#include "util.h"
+#include "timing.h"
 
 
 struct dmtimer_s {
--- origsrc/libMems/libMems/dmSML/util.c	2005-02-27 07:06:53.000000000 +0900
+++ src/libMems/libMems/dmSML/util.c	2025-08-25 16:33:46.715044300 +0900
@@ -3,7 +3,7 @@
 #endif
 
 #include <stdio.h>
-#include "libMems/dmSML/util.h"
+#include "util.h"
 
 #define FMT_BUFFER_SIZE     (32)
 #define FMT_MAX_STRING      (1024)
--- origsrc/libMems/libMems/gnAlignedSequences.cpp	2007-07-29 14:54:27.000000000 +0900
+++ src/libMems/libMems/gnAlignedSequences.cpp	2025-08-25 16:33:46.719081900 +0900
@@ -10,7 +10,7 @@
 #include "config.h"
 #endif
 
-#include "libMems/gnAlignedSequences.h"
+#include "gnAlignedSequences.h"
 #include <sstream>
 
 using namespace std;
--- origsrc/libMems/libMems/gnAlignedSequences.h	2008-03-15 16:08:53.000000000 +0900
+++ src/libMems/libMems/gnAlignedSequences.h	2025-08-25 16:33:46.721100200 +0900
@@ -29,13 +29,13 @@
 #include "config.h"
 #endif
 
-#include "libGenome/gnSequence.h"
-#include "libGenome/gnFilter.h"
+#include <libGenome/gnSequence.h>
+#include <libGenome/gnFilter.h>
 #include <list>
 #include <fstream>
 #include <vector>
 
-namespace mems {
+namespace mems {
 
 // the number of characters in each row of an alignment file
 const int MEGA_ALIGN_COLUMNS = 60;
--- origsrc/libMems/libMems/gnRAWSequence.h	2008-01-10 09:30:00.000000000 +0900
+++ src/libMems/libMems/gnRAWSequence.h	2025-08-25 16:33:46.723121600 +0900
@@ -18,12 +18,12 @@
 #ifndef _gnRAWSequence_h_
 #define _gnRAWSequence_h_
 
-#include "libGenome/gnDefs.h"
+#include <libGenome/gnDefs.h>
 
 #include <string>
 #include <iostream>
 #include <list>
-#include "libGenome/gnSequence.h"
+#include <libGenome/gnSequence.h>
 
 namespace genome {
 
