--- origsrc/gibson-1.1.0/CMakeLists.txt	2015-02-09 23:53:12.000000000 +0900
+++ src/gibson-1.1.0/CMakeLists.txt	2015-03-30 18:19:20.356829400 +0900
@@ -5,6 +5,7 @@ project(gibson)
 OPTION( WITH_DEBUG "enable debug module" OFF )
 
 # cmake needed modules
+include_directories("${PROJECT_BINARY_DIR}/src")
 include_directories("${PROJECT_SOURCE_DIR}/src")
 include(CheckIncludeFiles)
 include(CheckLibraryExists)
@@ -93,6 +94,9 @@ endif ( HAVE_JEMALLOC EQUAL 1 )
 
 install( TARGETS ${PROJECT} DESTINATION ${PREFIX}/bin )
 install( FILES debian/etc/${PROJECT}/${PROJECT}.conf DESTINATION /etc/${PROJECT}/ )
+if ( CYGWIN )
+	install( FILES debian/etc/init.d/${PROJECT} DESTINATION /etc/rc.d/init.d/)
+else ()
 install( FILES debian/etc/init.d/${PROJECT} DESTINATION /etc/init.d/
 		 PERMISSIONS
 		 OWNER_READ
@@ -103,6 +107,7 @@ install( FILES debian/etc/init.d/${PROJE
 		 WORLD_READ
 		 WORLD_EXECUTE
 )
+endif()
 
 ## tests
 
