--- origsrc/foremost-1.5.7/Makefile	2010-03-02 11:49:30.000000000 +0900
+++ src/foremost-1.5.7/Makefile	2010-12-19 18:13:55.222159700 +0900
@@ -24,9 +24,9 @@ MAN_PAGES = $(NAME).8.gz
 RAW_FLAGS += -DVERSION=\"$(VERSION)\"
 
 # Where we get installed
-BIN = /usr/local/bin
+BIN = /usr/bin
 MAN = /usr/share/man/man8
-CONF= /usr/local/etc
+CONF= /etc
 # Setup for compiling and cross-compiling for Windows
 # The CR_ prefix refers to cross compiling from OSX to Windows
 CR_CC = $(CR_BASE)/gcc
--- origsrc/foremost-1.5.7/config.c	2010-03-02 11:49:30.000000000 +0900
+++ src/foremost-1.5.7/config.c	2010-12-19 18:10:10.589153300 +0900
@@ -287,6 +287,8 @@ int load_config_file(f_state *s)
     * So lets try the /usr/local/etc*/
 #ifdef __WIN32
 		set_config_file(s, "/Program Files/foremost/foremost.conf");
+#elif defined(__CYGWIN__)
+		set_config_file(s, "/etc/foremost.conf");
 #else
 		set_config_file(s, "/usr/local/etc/foremost.conf");
 #endif
