--- origsrc/annabell/src/Makefile.am	2019-07-12 10:29:54.000000000 +0900
+++ src/annabell/src/Makefile.am	2019-07-12 10:45:24.736936600 +0900
@@ -1,6 +1,6 @@
 annabellincludedir = ${includedir}/annabell
 
-annabellinclude_HEADERS = \
+noinst_HEADERS = \
 commands/CommandConstants.h commands/CommandUtils.h AnnabellParams.h AnnabellFlags.h \
 commands/Command.h commands/EmptyCommand.h commands/CommentCommand.h commands/ContinueContextCommand.h commands/RetrievePhraseCommand.h \
 commands/ExtractWordGroupCommand.h commands/MacroCommand.h commands/InputPhraseCommand.h commands/FileCommand.h \
--- origsrc/annabell/src/annabell_main.cc	2019-07-12 10:29:54.000000000 +0900
+++ src/annabell/src/annabell_main.cc	2019-07-12 10:41:51.679445000 +0900
@@ -116,7 +116,7 @@ int Interface(Annabell *annabell, Monito
 	while (lineRead && (commandResult != 2)) {
 		cout << "Enter command: ";
 
-		lineRead = getline(cin, input_line);
+		lineRead = (bool) getline(cin, input_line);
 
 		if (lineRead) {
 			Command* c = CommandFactory::newCommand(input_line);
