--- origsrc/libklel-1.2.0/Makefile.am	2019-08-01 00:57:24.000000000 +0900
+++ src/libklel-1.2.0/Makefile.am	2025-01-24 14:24:49.114693900 +0900
@@ -16,3 +16,6 @@ dist-sign: dist
 	@gpg --default-key "KL-EL Project (Code Signing Key)" --detach-sign $(distdir).tar.gz
 
 include $(top_srcdir)/utils/common.am
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libklel-1.2.0/configure.ac	2019-08-01 00:57:24.000000000 +0900
+++ src/libklel-1.2.0/configure.ac	2025-01-24 14:24:49.114693900 +0900
@@ -218,6 +218,7 @@ dnl ####################################
 
 AC_CONFIG_FILES(
   [
+    libklel.pc
     Makefile
     doc/Makefile
     doc/klel-expr/Makefile
--- origsrc/libklel-1.2.0/libklel.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libklel-1.2.0/libklel.pc.in	2025-01-24 14:24:49.114693900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libklel
+Description: KoreLogic Expression Language Library
+Version: @VERSION@
+Libs: -L${libdir} -lklel
+Cflags: -I${includedir}
--- origsrc/libklel-1.2.0/src/libklel/Makefile.am	2019-08-01 00:57:24.000000000 +0900
+++ src/libklel-1.2.0/src/libklel/Makefile.am	2025-01-24 14:24:49.114693900 +0900
@@ -5,6 +5,6 @@ EXTRA_DIST = all-includes.h api.h app-in
 
 libklel_la_SOURCES = api.c interpreter.c inner-string.c library.c parser.c steel.c tokenizer.c typecheck.c
 
-libklel_la_LDFLAGS = -version-info @libklel_version@
+libklel_la_LDFLAGS = -no-undefined -version-info @libklel_version@
 
 include $(top_srcdir)/utils/common.am
