--- origsrc/qmckl-1.0.0/Makefile.am	2024-04-10 23:38:10.000000000 +0900
+++ src/qmckl-1.0.0/Makefile.am	2025-12-22 20:16:01.064288400 +0900
@@ -60,7 +60,7 @@ AM_CPPFLAGS += -DQMCKL_TEST_DIR=\"$(QMCK
 
 lib_LTLIBRARIES = src/libqmckl.la
 src_libqmckl_la_SOURCES = $(qmckl_h) $(qmckl_f) $(C_FILES) $(F_FILES) $(H_PRIVATE_FUNC_FILES) $(H_PRIVATE_TYPE_FILES)
-src_libqmckl_la_LDFLAGS = $(LDFLAGS)
+src_libqmckl_la_LDFLAGS = -no-undefined $(LDFLAGS)
 
 CLEANFILES+=$(qmckl_fo) $(test_qmckl_o) $(FH_TYPE_FILES) $(FH_FUNC_FILES)
 
--- origsrc/qmckl-1.0.0/tools/build_makefile.py	2024-04-10 23:38:10.000000000 +0900
+++ src/qmckl-1.0.0/tools/build_makefile.py	2025-12-22 19:59:36.510267400 +0900
@@ -10,7 +10,7 @@ import subprocess
 def main():
     wd = os.getcwd()
     try:
-        srcdir = os.environ["srcdir"]
+        srcdir = os.environ["_srcdir"]
         os.chdir(srcdir)
     except KeyError:
         pass
