--- origsrc/userspace-rcu-0.7.3/Makefile.am	2012-05-30 22:55:36.000000000 +0900
+++ src/userspace-rcu-0.7.3/Makefile.am	2012-06-16 13:00:41.080512000 +0900
@@ -4,7 +4,7 @@ INCLUDES = -I$(top_builddir)/urcu
 
 #Add the -version-info directly here since we are only building
 # library that use the version-info
-AM_LDFLAGS=-lpthread -version-info $(URCU_LIBRARY_VERSION)
+AM_LDFLAGS=-lpthread -version-info $(URCU_LIBRARY_VERSION) -no-undefined
 AM_CFLAGS=-Wall
 
 SUBDIRS = . doc tests
