--- origsrc/CLBlast-1.5.0/CMakeLists.txt	2018-12-05 04:46:02.000000000 +0900
+++ src/CLBlast-1.5.0/CMakeLists.txt	2018-12-31 11:55:00.394123600 +0900
@@ -323,6 +323,9 @@ endforeach()
 # Creates and links the library
 if(BUILD_SHARED_LIBS)
   add_library(clblast SHARED ${SOURCES} ${HEADERS})
+  if(CYGWIN)
+    set_target_properties(clblast PROPERTIES SOVERSION 0)
+  endif()
 else(BUILD_SHARED_LIBS)
   add_library(clblast STATIC ${SOURCES} ${HEADERS})
 endif()
@@ -347,7 +350,7 @@ endif()
 # Installs the library
 include(GNUInstallDirs)
 
-install(TARGETS clblast EXPORT CLBlast DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS clblast EXPORT CLBlast RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 install(FILES include/clblast_half.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 if(OPENCL)
   install(FILES include/clblast.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
