--- origsrc/CLBlast-1.0.1/CMakeLists.txt	2017-08-09 03:35:49.000000000 +0900
+++ src/CLBlast-1.0.1/CMakeLists.txt	2017-08-13 17:32:57.902000000 +0900
@@ -226,7 +226,8 @@ if(MSVC)
 endif()
 
 # Installs the library
-install(TARGETS clblast EXPORT CLBlast DESTINATION lib)
+install(TARGETS clblast EXPORT CLBlast RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+set_target_properties(clblast PROPERTIES SOVERSION 0)
 install(FILES include/clblast.h DESTINATION include)
 install(FILES include/clblast_c.h DESTINATION include)
 install(FILES include/clblast_half.h DESTINATION include)
