--- origsrc/CLBlast-1.5.3/CMakeLists.txt	2022-09-28 05:45:49.000000000 +0900
+++ src/CLBlast-1.5.3/CMakeLists.txt	2022-12-13 11:51:05.012927500 +0900
@@ -347,7 +347,7 @@ endif()
 # Installs the library
 include(GNUInstallDirs)
 
-install(TARGETS clblast EXPORT CLBlast DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS clblast EXPORT CLBlast RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(FILES include/clblast_half.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 if(OPENCL)
   install(FILES include/clblast.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
