--- origsrc/dgc-0.98/app/Makefile.am	2009-11-16 23:44:54.000000000 +0900
+++ src/dgc-0.98/app/Makefile.am	2013-07-28 23:20:17.769870300 +0900
@@ -7,12 +7,12 @@
 # AUTOMAKE_OPTIONS = foreign
 
 AM_CPPFLAGS = \
-  -I../util\
-  -I../cube\
-  -I../syl\
-  -I../gnet\
-  -I../app\
-  -I../encoding
+  -I$(top_srcdir)/util\
+  -I$(top_srcdir)/cube\
+  -I$(top_srcdir)/syl\
+  -I$(top_srcdir)/gnet\
+  -I$(top_srcdir)/app\
+  -I$(top_srcdir)/encoding
 
 AM_LDFLAGS = -lm
 
--- origsrc/dgc-0.98/cube/Makefile.am	2009-11-16 23:44:55.000000000 +0900
+++ src/dgc-0.98/cube/Makefile.am	2013-07-28 22:53:27.137057800 +0900
@@ -68,11 +68,11 @@ xbmvhdl.c\
 xbmvcg.c\
 xbmdot.c
 
-libdgccube_la_CPPFLAGS = -I../util $(XML_CFLAGS)
+libdgccube_la_CPPFLAGS = -I$(top_srcdir)/util $(XML_CFLAGS)
 
 libdgccube_la_LIBADD = \
 ../util/libdgcutil.la \
 $(XML_LIBS)
 
-
+libdgccube_la_LDFLAGS = -no-undefined
 
--- origsrc/dgc-0.98/encoding/Makefile.am	2009-11-16 23:44:55.000000000 +0900
+++ src/dgc-0.98/encoding/Makefile.am	2013-07-28 23:16:31.849948400 +0900
@@ -22,10 +22,11 @@ encode_func.h\
 fsmenc.c\
 fsmenc.h
 
-libdgcencode_la_CPPFLAGS = -I../util -I../cube
+libdgcencode_la_CPPFLAGS = -I$(top_srcdir)/util -I$(top_srcdir)/cube
 
 
 libdgcencode_la_LIBADD = \
 ../util/libdgcutil.la\
 ../cube/libdgccube.la
 
+libdgcencode_la_LDFLAGS = -no-undefined
--- origsrc/dgc-0.98/gnet/Makefile.am	2009-11-16 23:44:55.000000000 +0900
+++ src/dgc-0.98/gnet/Makefile.am	2013-07-28 23:18:22.275729700 +0900
@@ -50,7 +50,7 @@ synth.c\
 dgd_opt.h\
 dgd.c
 
-libdgcgnet_la_CPPFLAGS = -I../util -I../cube -I../encoding -I../syl -I../gnet
+libdgcgnet_la_CPPFLAGS = -I$(top_srcdir)/util -I$(top_srcdir)/cube -I$(top_srcdir)/encoding -I$(top_srcdir)/syl -I$(top_srcdir)/gnet
 
 libdgcgnet_la_LIBADD =\
 ../util/libdgcutil.la\
@@ -58,4 +58,4 @@ libdgcgnet_la_LIBADD =\
 ../encoding/libdgcencode.la\
 ../syl/libdgcsyl.la
 
-
+libdgcgnet_la_LDFLAGS = -no-undefined
--- origsrc/dgc-0.98/syl/Makefile.am	2009-11-16 23:44:55.000000000 +0900
+++ src/dgc-0.98/syl/Makefile.am	2013-07-28 23:17:10.351901500 +0900
@@ -4,8 +4,9 @@ libdgcsyl_la_SOURCES = \
 sc_bexpr.h sc_data.h sc_io.h sc_phr.h sc_sld.h sc_util.h sc_cdef.h sc_sdef.h sc_strl.h \
 sc_bexpr.c sc_data.c sc_io.c sc_phr.c sc_sld.c sc_util.c sc_cdef.c sc_sdef.c sc_strl.c
 
-libdgcsyl_la_CPPFLAGS = -I../util
+libdgcsyl_la_CPPFLAGS = -I$(top_srcdir)/util
 
 libdgcsyl_la_LIBADD = \
 ../util/libdgcutil.la
 
+libdgcsyl_la_LDFLAGS = -no-undefined
--- origsrc/dgc-0.98/util/Makefile.am	2009-11-16 23:44:56.000000000 +0900
+++ src/dgc-0.98/util/Makefile.am	2013-07-28 21:40:26.837404200 +0900
@@ -40,3 +40,4 @@ libdgcutil_la_SOURCES = \
 	port.c \
 	port.h
 
+libdgcutil_la_LDFLAGS = -no-undefined
