--- origsrc/stumpless-2.1.0/CMakeLists.txt	2023-09-28 21:03:33.825589200 +0900
+++ src/stumpless-2.1.0/CMakeLists.txt	2023-09-28 21:15:53.453521900 +0900
@@ -855,6 +855,9 @@ endif()
 add_library(stumpless ${STUMPLESS_SOURCES})
 set_target_properties(stumpless PROPERTIES VERSION ${PROJECT_VERSION})
 set_target_properties(stumpless PROPERTIES PUBLIC_HEADER include/stumpless.h)
+if(DEFINED SOVERSION)
+  set_target_properties(stumpless PROPERTIES SOVERSION "${SOVERSION}")
+endif()
 if(MINGW)
   target_compile_options(stumpless PRIVATE -D__USE_MINGW_ANSI_STDIO)
   set_target_properties(stumpless PROPERTIES PREFIX "")
