--- origsrc/libnsl-1.2.0/src/Makefile.am	2017-11-11 17:15:40.000000000 +0900
+++ src/libnsl-1.2.0/src/Makefile.am	2019-01-02 15:13:59.461679100 +0900
@@ -27,8 +27,8 @@ noinst_HEADERS = internal.h libc-lock.h
 lib_LTLIBRARIES = libnsl.la
 
 libnsl_la_LDFLAGS = -version-info 2:0:0 \
-	-Wl,--no-undefined \
-	-Wl,--version-script=$(srcdir)/libnsl.map
+	-no-undefined \
+	-version-script=$(srcdir)/libnsl.map
 libnsl_la_LIBADD = @TIRPC_LIBS@ @LTLIBINTL@
 
 libnsl_la_SOURCES = yp_xdr.c do_ypcall.c ypprot_err.c yp_master.c \
--- origsrc/libnsl-1.2.0/src/nisplus/nis_call.c	2017-11-11 17:15:40.000000000 +0900
+++ src/libnsl-1.2.0/src/nisplus/nis_call.c	2019-01-02 15:11:15.133149700 +0900
@@ -37,6 +37,10 @@
 #include "nis_intern.h"
 #include "nis_xdr.h"
 
+#if defined(__CYGWIN__)
+#define stat64 stat
+#endif
+
 static const struct timeval RPCTIMEOUT = {10, 0};
 static const struct timeval UDPTIMEOUT = {5, 0};
 
