--- origsrc/crfsuite-0.12/configure.in	2011-08-09 23:17:49.000000000 +0900
+++ src/crfsuite-0.12/configure.in	2015-05-15 18:39:21.826874200 +0900
@@ -126,12 +126,7 @@ AC_CHECK_FUNCS(strdup strerror strtol st
 dnl Check for math library
 AC_CHECK_LIB(m, rand)
 
-AC_ARG_WITH(
-	liblbfgs,
-	[AS_HELP_STRING([--with-liblbfgs=DIR],[liblbfgs directory])],
-	[INCLUDES="${INCLUDES} -I${withval}/include"; LDFLAGS="${LDFLAGS} -L${withval}/lib"]
-)
-AC_CHECK_LIB(lbfgs, lbfgs)
+PKG_CHECK_MODULES([LIBLBFGS], [liblbfgs])
 
 dnl ------------------------------------------------------------------
 dnl Export variables
--- origsrc/crfsuite-0.12/lib/cqdb/Makefile.am	2011-07-04 23:14:50.000000000 +0900
+++ src/crfsuite-0.12/lib/cqdb/Makefile.am	2015-05-15 18:41:27.871558400 +0900
@@ -15,7 +15,7 @@ libcqdb_la_LDFLAGS = \
 	-no-undefined \
 	-release @VERSION@
 
-libcqdb_la_CFLAGS = -I./include
+libcqdb_la_CFLAGS = -I$(srcdir)/include
 
 AM_CFLAGS = @CFLAGS@
 INCLUDES = @INCLUDES@
--- origsrc/crfsuite-0.12/lib/crf/Makefile.am	2011-07-04 23:15:04.000000000 +0900
+++ src/crfsuite-0.12/lib/crf/Makefile.am	2015-05-15 18:44:04.330289200 +0900
@@ -33,13 +33,13 @@ libcrfsuite_la_SOURCES = \
 	src/crfsuite_train.c \
 	src/crfsuite.c
 
-libcrfsuite_la_CFLAGS = -I$(top_builddir)/lib/cqdb/include
+libcrfsuite_la_CFLAGS = -I$(top_srcdir)/lib/cqdb/include
 
 libcrfsuite_la_LDFLAGS = \
 	-no-undefined \
 	-release @VERSION@
 
-libcrfsuite_la_LIBADD = \
+libcrfsuite_la_LIBADD = $(LIBLBFGS_LIBS) \
 	$(top_builddir)/lib/cqdb/libcqdb.la
 
 AM_CFLAGS = @CFLAGS@
