diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/fruit-1.0.2/CMakeLists.txt src/fruit-1.0.2/CMakeLists.txt
--- origsrc/fruit-1.0.2/CMakeLists.txt	2015-03-01 04:34:13.000000000 +0900
+++ src/fruit-1.0.2/CMakeLists.txt	2016-03-22 18:34:43.381893000 +0900
@@ -40,6 +40,13 @@ endif()
 
 add_subdirectory(doc EXCLUDE_FROM_ALL)
 
-install(DIRECTORY include
-  DESTINATION ${INSTALL_INCLUDE_DIR}/fruit
+install(DIRECTORY include/fruit
+  DESTINATION ${INSTALL_INCLUDE_DIR}
   FILES_MATCHING PATTERN "*.h")
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/fruit.pc.in ${CMAKE_BINARY_DIR}/fruit.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/fruit.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/fruit-1.0.2/fruit.pc.in src/fruit-1.0.2/fruit.pc.in
--- origsrc/fruit-1.0.2/fruit.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/fruit-1.0.2/fruit.pc.in	2016-03-22 18:34:59.427143600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: fruit
+Description: Dependency injection framework for C++
+Version: @FRUIT_VERSION@
+Libs: -L${libdir} -lfruit
+Cflags: -I${includedir}/fruit
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/fruit-1.0.2/src/CMakeLists.txt src/fruit-1.0.2/src/CMakeLists.txt
--- origsrc/fruit-1.0.2/src/CMakeLists.txt	2015-03-01 04:34:13.000000000 +0900
+++ src/fruit-1.0.2/src/CMakeLists.txt	2016-03-22 18:27:08.871940600 +0900
@@ -9,4 +9,7 @@ semistatic_map.cpp
 )
 
 install(TARGETS fruit
-   LIBRARY DESTINATION ${INSTALL_LIBRARY_DIR})
+   RUNTIME DESTINATION bin
+   ARCHIVE DESTINATION ${INSTALL_LIBRARY_DIR})
+
+set_target_properties(fruit PROPERTIES SOVERSION 0)
