--- origsrc/jthread-1.3.1/src/CMakeLists.txt	2011-08-29 03:34:12.000000000 +0900
+++ src/jthread-1.3.1/src/CMakeLists.txt	2012-01-29 13:30:19.207938300 +0900
@@ -22,6 +22,7 @@ if (NOT UNIX)
 	set(JTHREAD_COMPILE_STATIC ON CACHE BOOL "Flag indicating if a static library should be built, or a dynamic one")
 endif (NOT UNIX)
 
+if (NOT CYGWIN)
 if (UNIX OR JTHREAD_COMPILE_STATIC)
 	set(JTHREAD_INSTALLTARGETS jthread-static)
 	add_library(jthread-static STATIC ${SOURCES} ${HEADERS})
@@ -29,9 +30,11 @@ if (UNIX OR JTHREAD_COMPILE_STATIC)
 	set_target_properties(jthread-static PROPERTIES CLEAN_DIRECT_OUTPUT 1)
 	target_link_libraries(jthread-static ${CMAKE_THREAD_LIBS_INIT})
 endif(UNIX OR JTHREAD_COMPILE_STATIC)
+endif (NOT CYGWIN)
 if (UNIX OR NOT JTHREAD_COMPILE_STATIC)
 	add_library(jthread-shared SHARED ${SOURCES} ${HEADERS})
 	set_target_properties(jthread-shared PROPERTIES VERSION ${VERSION})
+	set_target_properties(jthread-shared PROPERTIES SOVERSION ${VERSION})
 	set_target_properties(jthread-shared PROPERTIES OUTPUT_NAME jthread)
 	set_target_properties(jthread-shared PROPERTIES CLEAN_DIRECT_OUTPUT 1)
 	set(JTHREAD_INSTALLTARGETS ${JTHREAD_INSTALLTARGETS} jthread-shared)
@@ -42,7 +45,13 @@ include_directories(${PROJECT_SOURCE_DIR
 include_directories(${PROJECT_BINARY_DIR}/src)
 
 install(FILES ${HEADERS} DESTINATION include/jthread)
+if (CYGWIN)
+install(TARGETS ${JTHREAD_INSTALLTARGETS}
+	RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION ${LIBRARY_INSTALL_DIR})
+else (CYGWIN)
 install(TARGETS ${JTHREAD_INSTALLTARGETS} DESTINATION ${LIBRARY_INSTALL_DIR})
+endif (CYGWIN)
 
 if (UNIX)
 	get_target_property(JTHREAD_LOCAL_LIBRARY_NAME jthread-shared LOCATION)
@@ -87,7 +96,7 @@ install(FILES
 
 if (UNIX)
 	foreach(ARG ${JTHREAD_LIBS})
-		set(JTHREAD_LIBS_PKGCONFIG "${JTHREAD_LIBS_PKGCONFIG} ${ARG}")
+		set(JTHREAD_LIBS_PKGCONFIG "-ljthread ${ARG}")
 	endforeach()
 
 	configure_file(${PROJECT_SOURCE_DIR}/pkgconfig/jthread.pc.in ${PROJECT_BINARY_DIR}/pkgconfig/jthread.pc)
