--- origsrc/mongoose-6.9/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/mongoose-6.9/Makefile.am	2017-09-21 00:01:00.117159300 +0900
@@ -0,0 +1,9 @@
+
+pkginclude_HEADERS = mongoose.h
+
+lib_LTLIBRARIES = libmongoose.la
+libmongoose_la_SOURCES = mongoose.c
+libmongoose_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/mongoose-6.9/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/mongoose-6.9/configure.ac	2017-09-21 00:07:31.241000000 +0900
@@ -0,0 +1,54 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([mongoose], [6.9])
+AC_CONFIG_SRCDIR([config.h.in])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_AWK
+AC_PROG_CC
+AC_PROG_CPP
+AC_PROG_INSTALL
+AC_PROG_LN_S
+AC_PROG_MAKE_SET
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([arpa/inet.h fcntl.h inttypes.h limits.h malloc.h memory.h netdb.h netinet/in.h stddef.h stdint.h stdlib.h string.h sys/file.h sys/ioctl.h sys/mount.h sys/param.h sys/socket.h sys/statvfs.h sys/time.h unistd.h utime.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+AC_TYPE_UID_T
+AC_C_INLINE
+AC_TYPE_INT16_T
+AC_TYPE_INT32_T
+AC_TYPE_INT64_T
+AC_TYPE_INT8_T
+AC_TYPE_MODE_T
+AC_TYPE_OFF_T
+AC_TYPE_PID_T
+AC_TYPE_SIZE_T
+AC_TYPE_SSIZE_T
+AC_CHECK_MEMBERS([struct stat.st_blksize])
+AC_TYPE_UINT16_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+AC_TYPE_UINT8_T
+
+# Checks for library functions.
+AC_FUNC_FORK
+AC_FUNC_FSEEKO
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_FUNC_STRERROR_R
+AC_FUNC_STRTOD
+AC_CHECK_FUNCS([clock_gettime dup2 fdatasync gethostbyname getpagesize gettimeofday inet_ntoa localtime_r memchr memmove memset mkdir rmdir select socket strchr strdup strerror strrchr strstr strtoul strtoull utime])
+
+AC_CONFIG_FILES([Makefile mongoose.pc])
+AC_OUTPUT
--- origsrc/mongoose-6.9/mongoose.h	2017-09-13 20:57:33.000000000 +0900
+++ src/mongoose-6.9/mongoose.h	2017-09-21 00:01:00.132797800 +0900
@@ -669,7 +669,9 @@ typedef struct stat cs_stat_t;
 #define to64(x) strtoll(x, NULL, 10)
 #define INT64_FMT PRId64
 #define INT64_X_FMT PRIx64
+#if !defined(__cdecl)
 #define __cdecl
+#endif
 
 #define fileno(x) -1
 
--- origsrc/mongoose-6.9/mongoose.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/mongoose-6.9/mongoose.pc.in	2017-09-21 00:01:00.132797800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mongoose
+Description: Embedded web server for C/C++
+Version: @VERSION@
+Libs: -L${libdir} -lmongoose
+Cflags: -I${includedir}/mongoose
