--- origsrc/uptimed-0.4.0/libuptimed/Makefile.am	2015-04-07 20:57:26.000000000 +0900
+++ src/uptimed-0.4.0/libuptimed/Makefile.am	2017-12-22 00:11:57.706000000 +0900
@@ -1,6 +1,6 @@
 lib_LTLIBRARIES = libuptimed.la
 libuptimed_la_SOURCES = misc.c milestone.c urec.c
-libuptimed_la_LDFLAGS = -version-info 0:0:0
+libuptimed_la_LDFLAGS = -no-undefined -version-info 0:0:0
 
 noinst_HEADERS = milestone.h misc.h urec.h
 
--- origsrc/uptimed-0.4.0/libuptimed/urec.c	2015-04-07 20:57:26.000000000 +0900
+++ src/uptimed-0.4.0/libuptimed/urec.c	2017-12-22 00:13:40.030000000 +0900
@@ -21,6 +21,7 @@ uptimed - Copyright (c) 1998-2004 Rob Ka
 #include "urec.h"
 
 Urec *urec_list = NULL;
+Urec *u_current = NULL;
 static Urec *urec_last = NULL;
 
 Urec *add_urec(time_t utime, time_t btime, char *sys) {
