--- origsrc/uptimed-0.3.16/configure.ac	2009-01-02 08:46:00.000000000 +0900
+++ src/uptimed-0.3.16/configure.ac	2010-07-17 21:36:03.798115800 +0900
@@ -3,6 +3,7 @@ dnl configure.in for uptimed
 AC_INIT([uptimed], 0.3.16)
 AM_CONFIG_HEADER(config.h)
 AC_PROG_CC
+AC_LIBTOOL_WIN32_DLL
 AM_PROG_LIBTOOL
 AM_INIT_AUTOMAKE
 
--- origsrc/uptimed-0.3.16/libuptimed/Makefile.am	2009-01-02 08:46:00.000000000 +0900
+++ src/uptimed-0.3.16/libuptimed/Makefile.am	2010-07-17 21:35:31.798115800 +0900
@@ -1,6 +1,6 @@
 lib_LTLIBRARIES = libuptimed.la
 libuptimed_la_SOURCES = misc.c milestone.c urec.c
-libuptimed_la_LDFLAGS = -version-info 0:0:0
+libuptimed_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 
 noinst_HEADERS = milestone.h misc.h urec.h
 
--- origsrc/uptimed-0.3.16/libuptimed/urec.c	2009-01-02 08:46:00.000000000 +0900
+++ src/uptimed-0.3.16/libuptimed/urec.c	2010-07-17 22:13:46.041128800 +0900
@@ -21,6 +21,7 @@ uptimed - Copyright (c) 1998-2004 Rob Ka
 #include "urec.h"
 
 Urec *urec_list = NULL;
+Urec *u_current = NULL;
 static Urec *urec_last = NULL;
 
 Urec *add_urec(time_t utime, time_t btime, char *sys) {
