--- origsrc/piplib/Makefile.am	2011-01-02 00:09:26.000000000 +0900
+++ src/piplib/Makefile.am	2011-01-02 00:11:24.089277300 +0900
@@ -40,13 +40,13 @@ pkginclude_HEADERS = \
 
 libpiplib32_la_SOURCES = $(LIBRARY_FILES)
 libpiplib32_la_CPPFLAGS = @DFLAGS32@ -I$(srcdir)/include
-libpiplib32_la_LDFLAGS = -rpath $(libdir) -version-info @versioninfo@
+libpiplib32_la_LDFLAGS = -rpath $(libdir) -version-info @versioninfo@ -no-undefined
 libpiplib64_la_SOURCES = $(LIBRARY_FILES)
 libpiplib64_la_CPPFLAGS = @DFLAGS64@ -I$(srcdir)/include
-libpiplib64_la_LDFLAGS = -rpath $(libdir) -version-info @versioninfo@
+libpiplib64_la_LDFLAGS = -rpath $(libdir) -version-info @versioninfo@ -no-undefined
 libpiplibMP_la_SOURCES = $(LIBRARY_FILES)
 libpiplibMP_la_CPPFLAGS = @MP_DFLAGS@ -I$(srcdir)/include
-libpiplibMP_la_LDFLAGS = @EXTRA_LIBS@ -rpath $(libdir) -version-info @versioninfo@
+libpiplibMP_la_LDFLAGS = @EXTRA_LIBS@ -rpath $(libdir) -version-info @versioninfo@ -no-undefined
 
 pip32_SOURCES = source/maind.c
 pip32_CPPFLAGS = @DFLAGS32@ -I$(srcdir)/include
--- origsrc/piplib/configure.in	2011-01-02 00:09:26.000000000 +0900
+++ src/piplib/configure.in	2011-01-02 00:10:56.319346100 +0900
@@ -48,7 +48,7 @@ AC_CHECK_PROG(CD, cd)
 dnl Configure needs an empty install.sh file with this, i HATE that...
 AC_PROG_INSTALL
 AC_CANONICAL_HOST
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_CHECK_PROG(TEXI2DVI, texi2dvi, texi2dvi, [])
 AM_CONDITIONAL(HAVE_TEXI2DVI, test -n "$TEXI2DVI")
 
