--- origsrc/voro++-0.4.6/CMakeLists.txt	2024-11-24 21:46:04.670436600 +0900
+++ src/voro++-0.4.6/CMakeLists.txt	2024-11-24 22:07:29.750599000 +0900
@@ -64,7 +64,7 @@ add_library(voro++ ${VORO_SOURCES})
 set_target_properties(voro++ PROPERTIES 
   LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/src"
   SOVERSION ${SOVERSION})
-install(TARGETS voro++ LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS voro++ RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 add_executable(cmd_line src/cmd_line.cc)
 target_link_libraries(cmd_line voro++)
@@ -96,3 +96,9 @@ if (DOXYGEN_FOUND)
     COMMENT "Build doxygen documentation")
 endif (DOXYGEN_FOUND)
 
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/voro++.pc.in ${CMAKE_BINARY_DIR}/voro++.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/voro++.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/voro++-0.4.6/voro++.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/voro++-0.4.6/voro++.pc.in	2024-11-24 22:09:34.012121500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: voro++
+Description: Library for 3D computations of the Voronoi tessellation
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lvoro++
+Cflags: -I${includedir}
