--- origsrc/libfame-0.9.1/Makefile.am	2003-01-06 03:38:11.000000000 +0900
+++ src/libfame-0.9.1/Makefile.am	2026-05-21 08:06:07.975562100 +0900
@@ -36,3 +36,5 @@ rpm: $(PACKAGE)-$(VERSION).tar.gz
 	cp $(PACKAGE)-$(VERSION).tar.gz /usr/src/redhat/SOURCES
 	rpm -ba libfame.spec
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libfame-0.9.1/configure.ac	2026-05-21 07:55:05.291681100 +0900
+++ src/libfame-0.9.1/configure.ac	2026-05-21 08:07:08.665757700 +0900
@@ -1,6 +1,6 @@
 
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(README)
+AC_INIT([libfame], [0.9.1])
 
 dnl Set various version strings - taken gratefully from the GTk sources
 #
@@ -42,11 +42,10 @@ AC_SUBST(LT_REVISION)
 AC_SUBST(LT_AGE)
 
 dnl Setup for automake
-AM_INIT_AUTOMAKE(libfame, $VERSION)
+AM_INIT_AUTOMAKE
 
 dnl Check for tools
-AC_LIBTOOL_WIN32_DLL
-AM_PROG_LIBTOOL
+LT_INIT
 AC_PROG_MAKE_SET
 AC_PROG_CC
 AC_C_INLINE
@@ -218,6 +217,7 @@ fi
 dnl Finally create all the generated files
 
 AC_OUTPUT([
+libfame.pc
 libfame-config
 libfame.spec
 libfame.m4
--- origsrc/libfame-0.9.1/libfame.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libfame-0.9.1/libfame.pc.in	2026-05-21 08:04:16.973952800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libfame
+Description: Fast Assembly Mpeg Encoding library
+Version: @VERSION@
+Libs: -L${libdir} -lfame
+Cflags: -I${includedir}
--- origsrc/libfame-0.9.1/src/Makefile.am	2002-05-01 01:25:20.000000000 +0900
+++ src/libfame-0.9.1/src/Makefile.am	2026-05-21 07:57:33.848989000 +0900
@@ -56,7 +56,7 @@ libfame_la_include_HEADERS =			\
 	fame_motion_pmvfast.h			\
 	fame_malloc.h
 
-libfame_la_LDFLAGS = \
+libfame_la_LDFLAGS = -no-undefined \
         -release $(LT_RELEASE) \
         -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 
