--- origsrc/xmlsec1-1.2.18/src/Makefile.am	2011-05-12 08:02:01.000000000 +0900
+++ src/xmlsec1-1.2.18/src/Makefile.am	2011-05-17 23:19:04.998623000 +0900
@@ -64,7 +64,7 @@ libxmlsec1_la_LIBADD = \
 	$(XMLSEC_DL_LIBS) \
 	$(NULL)
 
-libxmlsec1_la_LDFLAGS = \
+libxmlsec1_la_LDFLAGS = -no-undefined \
 	@XMLSEC_EXTRA_LDFLAGS@ \
 	-version-info @XMLSEC_VERSION_INFO@ \
 	$(NULL)
--- origsrc/xmlsec1-1.2.18/src/gcrypt/Makefile.am	2011-05-12 08:02:01.000000000 +0900
+++ src/xmlsec1-1.2.18/src/gcrypt/Makefile.am	2011-05-17 23:19:21.272023000 +0900
@@ -49,7 +49,7 @@ libxmlsec1_gcrypt_la_LIBADD = \
 libxmlsec1_gcrypt_la_DEPENDENCIES = \
 	$(NULL)
 
-libxmlsec1_gcrypt_la_LDFLAGS = \
+libxmlsec1_gcrypt_la_LDFLAGS = -no-undefined \
 	@XMLSEC_CRYPTO_EXTRA_LDFLAGS@ \
 	-version-info @XMLSEC_VERSION_INFO@ \
 	$(NULL)
--- origsrc/xmlsec1-1.2.18/src/gnutls/Makefile.am	2011-05-12 08:02:01.000000000 +0900
+++ src/xmlsec1-1.2.18/src/gnutls/Makefile.am	2011-05-17 23:19:37.675610200 +0900
@@ -52,7 +52,7 @@ libxmlsec1_gnutls_la_LIBADD = \
 libxmlsec1_gnutls_la_DEPENDENCIES = \
 	$(NULL)
 
-libxmlsec1_gnutls_la_LDFLAGS = \
+libxmlsec1_gnutls_la_LDFLAGS = -no-undefined \
 	@XMLSEC_CRYPTO_EXTRA_LDFLAGS@ \
 	-version-info @XMLSEC_VERSION_INFO@ \
 	$(NULL)
--- origsrc/xmlsec1-1.2.18/src/mscrypto/Makefile.am	2011-05-12 08:02:01.000000000 +0900
+++ src/xmlsec1-1.2.18/src/mscrypto/Makefile.am	2011-05-17 23:20:01.810314200 +0900
@@ -56,7 +56,7 @@ libxmlsec1_mscrypto_la_DEPENDENCIES = \
 	mingw-crypt32.def \
 	$(NULL)
 
-libxmlsec1_mscrypto_la_LDFLAGS = \
+libxmlsec1_mscrypto_la_LDFLAGS = -no-undefined \
 	@XMLSEC_CRYPTO_EXTRA_LDFLAGS@ \
 	-version-info @XMLSEC_VERSION_INFO@ \
 	$(NULL)
--- origsrc/xmlsec1-1.2.18/src/nss/Makefile.am	2011-05-12 08:02:01.000000000 +0900
+++ src/xmlsec1-1.2.18/src/nss/Makefile.am	2011-05-17 23:20:20.957847000 +0900
@@ -51,7 +51,7 @@ libxmlsec1_nss_la_LIBADD = \
 libxmlsec1_nss_la_DEPENDENCIES = \
 	$(NULL)
 
-libxmlsec1_nss_la_LDFLAGS = \
+libxmlsec1_nss_la_LDFLAGS = -no-undefined \
 	@XMLSEC_CRYPTO_EXTRA_LDFLAGS@ \
 	-version-info @XMLSEC_VERSION_INFO@ \
 	$(NULL)
--- origsrc/xmlsec1-1.2.18/src/openssl/Makefile.am	2011-05-12 08:02:01.000000000 +0900
+++ src/xmlsec1-1.2.18/src/openssl/Makefile.am	2011-05-17 23:20:33.365688600 +0900
@@ -50,7 +50,7 @@ libxmlsec1_openssl_la_LIBADD = \
 libxmlsec1_openssl_la_DEPENDENCIES = \
 	$(NULL)
 
-libxmlsec1_openssl_la_LDFLAGS = \
+libxmlsec1_openssl_la_LDFLAGS = -no-undefined \
 	@XMLSEC_CRYPTO_EXTRA_LDFLAGS@ \
 	-version-info @XMLSEC_VERSION_INFO@ \
 	$(NULL)
--- origsrc/xmlsec1-1.2.18/src/skeleton/Makefile.am	2011-05-12 08:02:01.000000000 +0900
+++ src/xmlsec1-1.2.18/src/skeleton/Makefile.am	2011-05-17 23:20:49.188440600 +0900
@@ -39,7 +39,7 @@ libxmlsec1_skeleton_la_LIBADD = \
 libxmlsec1_skeleton_la_DEPENDENCIES = \
 	$(NULL)
 
-libxmlsec1_skeleton_la_LDFLAGS = \
+libxmlsec1_skeleton_la_LDFLAGS = -no-undefined \
 	@XMLSEC_CRYPTO_EXTRA_LDFLAGS@ \
 	-version-info @XMLSEC_VERSION_INFO@ \
 	$(NULL)
