--- origsrc/pocketsphinx-5.0.4/gst/CMakeLists.txt	2025-01-10 23:46:14.000000000 +0900
+++ src/pocketsphinx-5.0.4/gst/CMakeLists.txt	2025-03-27 09:42:46.917278800 +0900
@@ -18,4 +18,4 @@ target_include_directories(
   gstpocketsphinx PUBLIC ${GSTREAMER_INCLUDE_DIRS} ${GOBJECT_INCLUDE_DIRS}
   )
 message("Installing GStreamer plugin to ${CMAKE_INSTALL_FULL_LIBDIR}/gstreamer-1.0")
-install(TARGETS gstpocketsphinx LIBRARY DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}/gstreamer-1.0)
+install(TARGETS gstpocketsphinx RUNTIME DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}/gstreamer-1.0)
