--- origsrc/pocketsphinx-0.8/src/gst-plugin/Makefile.am	2012-09-29 07:17:02.000000000 +0900
+++ src/pocketsphinx-0.8/src/gst-plugin/Makefile.am	2014-06-01 21:03:59.582209300 +0900
@@ -13,7 +13,7 @@ libgstpocketsphinx_la_SOURCES = gstpocke
 libgstpocketsphinx_la_CFLAGS = $(GST_CFLAGS) -DMODELDIR=\"$(pkgdatadir)/model\"
 libgstpocketsphinx_la_LIBADD = $(GST_LIBS) -lgstaudio-$(GST_MAJORMINOR) \
 	$(top_builddir)/src/libpocketsphinx/libpocketsphinx.la -lsphinxbase
-libgstpocketsphinx_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS)
+libgstpocketsphinx_la_LDFLAGS = -no-undefined $(GST_PLUGIN_LDFLAGS)
 
 noinst_HEADERS = gstpocketsphinx.h gstvader.h $(my_headers)
 
--- origsrc/pocketsphinx-0.8/src/libpocketsphinx/Makefile.am	2012-12-13 03:51:36.000000000 +0900
+++ src/pocketsphinx-0.8/src/libpocketsphinx/Makefile.am	2014-06-01 21:04:14.023615600 +0900
@@ -2,7 +2,7 @@ ARCH   = `uname -m | sed s/i.86/i386/`_`
 
 lib_LTLIBRARIES = libpocketsphinx.la
 
-libpocketsphinx_la_LDFLAGS = -version-info 2:0:1 -lm
+libpocketsphinx_la_LDFLAGS = -no-undefined -version-info 2:0:1 -lm
 
 libpocketsphinx_la_SOURCES =			\
 	acmod.c					\
