--- origsrc/lrslib-042c/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/lrslib-042c/Makefile.am	2012-03-04 16:19:39.378425800 +0900
@@ -0,0 +1,61 @@
+bin_PROGRAMS = \
+	lrs lrs-2gnash lrs-buffer lrs-float2rat lrs-fourier lrs-nash \
+	lrs-rat2float lrs-redund lrs-setupnash lrs-setupnash2
+
+lrs_CPPFLAGS = -DGMP
+lrs_LDADD = liblrsgmp.la
+lrs_SOURCES = lrs.c
+
+lrs_2gnash_SOURCES = 2gnash.c
+
+lrs_buffer_SOURCES = buffer.c
+
+lrs_float2rat_SOURCES = float2rat.c
+
+lrs_fourier_CPPFLAGS = -DGMP
+lrs_fourier_LDADD = liblrsgmp.la
+lrs_fourier_SOURCES = fourier.c
+
+lrs_nash_CPPFLAGS = -DGMP
+lrs_nash_LDADD = liblrsgmp.la
+lrs_nash_SOURCES = nash.c
+
+lrs_rat2float_CPPFLAGS = -DLRSMP
+lrs_rat2float_LDADD = liblrsmp.la
+lrs_rat2float_SOURCES = rat2float.c
+
+lrs_redund_CPPFLAGS = -DGMP
+lrs_redund_LDADD = liblrsgmp.la
+lrs_redund_SOURCES = redund.c
+
+lrs_setupnash_CPPFLAGS = -DGMP
+lrs_setupnash_LDADD = liblrsgmp.la
+lrs_setupnash_SOURCES = setupnash.c
+
+lrs_setupnash2_CPPFLAGS = -DGMP
+lrs_setupnash2_LDADD = liblrsgmp.la
+lrs_setupnash2_SOURCES = setupnash2.c
+
+
+lrslib_includedir = $(includedir)/lrslib
+lrslib_include_HEADERS = lrsgmp.h lrslib.h lrslong.h
+
+
+lib_LTLIBRARIES = liblrsgmp.la liblrslong.la liblrsmp.la
+
+liblrsgmp_la_CPPFLAGS = -DGMP
+liblrsgmp_la_LDFLAGS = -no-undefined -version-info 4
+liblrsgmp_la_LIBADD = -lgmp
+liblrsgmp_la_SOURCES = lrsgmp.c lrslib.c
+
+liblrslong_la_CPPFLAGS = -DLONG
+liblrslong_la_LDFLAGS = -no-undefined -version-info 4
+liblrslong_la_SOURCES = lrslib.c lrslong.c
+
+liblrsmp_la_CPPFLAGS = -DLRSMP
+liblrsmp_la_LDFLAGS = -no-undefined -version-info 4
+liblrsmp_la_SOURCES = lrslib.c lrsmp.c
+
+
+exampledir = $(datadir)/lrslib/examples
+example_DATA = chdemo.c lpdemo.c vedemo.c
--- origsrc/lrslib-042c/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/lrslib-042c/configure.ac	2012-03-04 16:10:31.019925000 +0900
@@ -0,0 +1,30 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([lrslib], [4.2c])
+AC_CONFIG_SRCDIR([2gnash.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE([foreign])
+
+# Checks for programs.
+AC_PROG_CC
+AM_PROG_CC_C_O
+
+LT_INIT([disable-static])
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([fcntl.h float.h stdlib.h string.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_PID_T
+
+# Checks for library functions.
+AC_FUNC_FORK
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([alarm dup2 strchr])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
