--- origsrc/bti-034/Makefile.am	2014-01-16 01:48:37.000000000 +0900
+++ src/bti-034/Makefile.am	2015-03-07 21:21:54.303577600 +0900
@@ -27,11 +27,13 @@ dist_man_MANS = \
 	bti.1 \
 	bti-shrink-urls.1
 
+completiondir = /etc/bash_completion.d
+completion_DATA = bti-bashcompletion
+
 EXTRA_DIST = \
 	bti.xml \
 	bti-shrink-urls.xml \
 	bti.example \
-	bti-bashcompletion \
 	RELEASE-NOTES \
 	bti-shrink-urls
 
