--- origsrc/libdbusmenu-12.10.2/tests/glib-server-nomenu.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/glib-server-nomenu.c	2014-11-03 09:12:37.088474200 +0900
@@ -27,8 +27,6 @@ with this program.  If not, see <http://
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	DbusmenuServer * server = dbusmenu_server_new("/org/test");
 	DbusmenuMenuitem * menuitem = dbusmenu_menuitem_new();
 	dbusmenu_menuitem_property_set(menuitem, "test", "test");
--- origsrc/libdbusmenu-12.10.2/tests/test-glib-events-client.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-glib-events-client.c	2014-11-03 09:11:05.541599200 +0900
@@ -114,8 +114,6 @@ timer_func (gpointer data)
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	DbusmenuClient * client = dbusmenu_client_new("org.dbusmenu.test", "/org/test");
 	g_signal_connect(G_OBJECT(client), DBUSMENU_CLIENT_SIGNAL_LAYOUT_UPDATED, G_CALLBACK(layout_updated), NULL);
 	g_signal_connect(G_OBJECT(client), DBUSMENU_CLIENT_SIGNAL_EVENT_RESULT, G_CALLBACK(event_status), GINT_TO_POINTER(USER_VALUE));
--- origsrc/libdbusmenu-12.10.2/tests/test-glib-events-nogroup-client.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-glib-events-nogroup-client.c	2014-11-03 09:11:16.432224200 +0900
@@ -118,8 +118,6 @@ timer_func (gpointer data)
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	DbusmenuClient * client = dbusmenu_client_new("org.dbusmenu.test", "/org/test");
 	g_signal_connect(G_OBJECT(client), DBUSMENU_CLIENT_SIGNAL_LAYOUT_UPDATED, G_CALLBACK(layout_updated), NULL);
 	g_signal_connect(G_OBJECT(client), DBUSMENU_CLIENT_SIGNAL_EVENT_RESULT, G_CALLBACK(event_status), GINT_TO_POINTER(USER_VALUE));
--- origsrc/libdbusmenu-12.10.2/tests/test-glib-events-server.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-glib-events-server.c	2014-11-03 09:11:23.916599200 +0900
@@ -68,8 +68,6 @@ name_lost (GDBusConnection * connection,
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	g_bus_own_name(G_BUS_TYPE_SESSION,
 	               "org.dbusmenu.test",
 	               G_BUS_NAME_OWNER_FLAGS_NONE,
--- origsrc/libdbusmenu-12.10.2/tests/test-glib-layout-client.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-glib-layout-client.c	2014-11-03 09:11:38.010349200 +0900
@@ -114,8 +114,6 @@ timer_func (gpointer data)
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	DbusmenuClient * client = dbusmenu_client_new("org.dbusmenu.test", "/org/test");
 	g_signal_connect(G_OBJECT(client), DBUSMENU_CLIENT_SIGNAL_LAYOUT_UPDATED, G_CALLBACK(layout_updated), NULL);
 
--- origsrc/libdbusmenu-12.10.2/tests/test-glib-layout-server.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-glib-layout-server.c	2014-11-03 09:11:44.760349200 +0900
@@ -90,8 +90,6 @@ name_lost (GDBusConnection * connection,
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	g_bus_own_name(G_BUS_TYPE_SESSION,
 	               "org.dbusmenu.test",
 	               G_BUS_NAME_OWNER_FLAGS_NONE,
--- origsrc/libdbusmenu-12.10.2/tests/test-glib-objects.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-glib-objects.c	2014-11-03 09:11:53.291599200 +0900
@@ -324,7 +324,6 @@ test_glib_objects_suite (void)
 gint
 main (gint argc, gchar * argv[])
 {
-	g_type_init();
 	g_test_init(&argc, &argv, NULL);
 
 	/* Test suites */
--- origsrc/libdbusmenu-12.10.2/tests/test-glib-properties-client.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-glib-properties-client.c	2014-11-03 09:12:01.307224200 +0900
@@ -152,8 +152,6 @@ layout_verify_timer (gpointer data)
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	/* Make sure the server starts up and all that */
 	g_usleep(500000);
 
--- origsrc/libdbusmenu-12.10.2/tests/test-glib-properties-server.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-glib-properties-server.c	2014-11-03 09:12:07.291599200 +0900
@@ -86,8 +86,6 @@ timer_func (gpointer data)
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	server = dbusmenu_server_new("/org/test");
 
 	timer_func(NULL);
--- origsrc/libdbusmenu-12.10.2/tests/test-glib-proxy-client.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-glib-proxy-client.c	2014-11-03 09:14:24.822849200 +0900
@@ -159,8 +159,6 @@ layout_verify_timer (gpointer data)
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	DbusmenuClient * client = dbusmenu_client_new("test.proxy.first_proxy", "/org/test");
 	g_signal_connect(G_OBJECT(client), DBUSMENU_CLIENT_SIGNAL_LAYOUT_UPDATED, G_CALLBACK(layout_updated), NULL);
 
--- origsrc/libdbusmenu-12.10.2/tests/test-glib-proxy-proxy.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-glib-proxy-proxy.c	2014-11-03 09:15:20.104099200 +0900
@@ -49,8 +49,6 @@ name_lost (GDBusConnection * connection,
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	if (argc != 3) {
 		g_error ("Need two params");
 		return 1;
--- origsrc/libdbusmenu-12.10.2/tests/test-glib-proxy-server.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-glib-proxy-server.c	2014-11-03 09:14:36.322849200 +0900
@@ -120,8 +120,6 @@ name_lost (GDBusConnection * connection,
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	g_bus_own_name(G_BUS_TYPE_SESSION,
 	               "test.proxy.server",
 	               G_BUS_NAME_OWNER_FLAGS_NONE,
--- origsrc/libdbusmenu-12.10.2/tests/test-glib-simple-items.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-glib-simple-items.c	2014-11-03 09:15:42.604099200 +0900
@@ -32,8 +32,6 @@ quititall (gpointer data)
 int
 main (int argc, char ** argv)
 {
-    g_type_init();
-
     DbusmenuServer * server = dbusmenu_server_new("/test/object");
     root_menuitem = dbusmenu_menuitem_new();
     dbusmenu_server_set_root(server, root_menuitem);
--- origsrc/libdbusmenu-12.10.2/tests/test-glib-submenu-client.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-glib-submenu-client.c	2014-11-03 09:14:46.088474200 +0900
@@ -97,8 +97,6 @@ timer_func (gpointer data)
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	DbusmenuClient * client = dbusmenu_client_new("org.dbusmenu.test", "/org/test");
 	g_signal_connect(G_OBJECT(client), DBUSMENU_CLIENT_SIGNAL_LAYOUT_UPDATED, G_CALLBACK(layout_updated), NULL);
 
--- origsrc/libdbusmenu-12.10.2/tests/test-glib-submenu-server.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-glib-submenu-server.c	2014-11-03 09:14:53.838474200 +0900
@@ -90,8 +90,6 @@ name_lost (GDBusConnection * connection,
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	g_bus_own_name(G_BUS_TYPE_SESSION,
 	               "org.dbusmenu.test",
 	               G_BUS_NAME_OWNER_FLAGS_NONE,
--- origsrc/libdbusmenu-12.10.2/tests/test-gtk-label-server.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-gtk-label-server.c	2014-11-03 09:16:00.161716400 +0900
@@ -71,8 +71,6 @@ name_lost (GDBusConnection * connection,
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	JsonParser * parser = json_parser_new();
 	GError * error = NULL;
 	if (!json_parser_load_from_file(parser, argv[1], &error)) {
--- origsrc/libdbusmenu-12.10.2/tests/test-gtk-reorder-server.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-gtk-reorder-server.c	2014-11-03 09:16:41.865818000 +0900
@@ -99,8 +99,6 @@ name_lost (GDBusConnection * connection,
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	g_bus_own_name(G_BUS_TYPE_SESSION,
 	               "glib.label.test",
 	               G_BUS_NAME_OWNER_FLAGS_NONE,
--- origsrc/libdbusmenu-12.10.2/tests/test-gtk-shortcut-server.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-gtk-shortcut-server.c	2014-11-03 09:16:34.068943000 +0900
@@ -83,8 +83,6 @@ name_lost (GDBusConnection * connection,
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	g_bus_own_name(G_BUS_TYPE_SESSION,
 	               "glib.label.test",
 	               G_BUS_NAME_OWNER_FLAGS_NONE,
--- origsrc/libdbusmenu-12.10.2/tests/test-gtk-submenu-server.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tests/test-gtk-submenu-server.c	2014-11-03 09:16:50.178318000 +0900
@@ -92,8 +92,6 @@ name_lost (GDBusConnection * connection,
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
-
 	g_bus_own_name(G_BUS_TYPE_SESSION,
 	               "glib.label.test",
 	               G_BUS_NAME_OWNER_FLAGS_NONE,
--- origsrc/libdbusmenu-12.10.2/tools/dbusmenu-dumper.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tools/dbusmenu-dumper.c	2014-11-03 09:09:32.269916200 +0900
@@ -388,7 +388,6 @@ static GOptionEntry general_options[] =
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
 	GError * error = NULL;
 	GOptionContext * context;
 
--- origsrc/libdbusmenu-12.10.2/tools/testapp/main.c	2012-10-04 00:54:22.000000000 +0900
+++ src/libdbusmenu-12.10.2/tools/testapp/main.c	2014-11-03 09:09:10.302971000 +0900
@@ -135,8 +135,6 @@ name_lost (GDBusConnection * connection,
 
 int main (int argc, char ** argv)
 {
-	g_type_init();
-
 	if (argc != 2) {
 		g_warning(USAGE);
 		return 1;
