--- origsrc/superlu_dist-7.0.0/FORTRAN/CMakeLists.txt	2021-05-13 04:29:07.000000000 +0900
+++ src/superlu_dist-7.0.0/FORTRAN/CMakeLists.txt	2021-09-06 14:45:21.435355000 +0900
@@ -3,9 +3,9 @@ include_directories(${SuperLU_DIST_SOURC
 include_directories(${SuperLU_DIST_BINARY_DIR}/FORTRAN)
 
 set(headers
-    ${CMAKE_INSTALL_PREFIX}/FORTRAN/superlu_mod.mod
-    ${CMAKE_INSTALL_PREFIX}/FORTRAN/superlupara_mod.mod
-    ${CMAKE_INSTALL_PREFIX}/FORTRAN/superlu_dist_config.fh
+    ${CMAKE_CURRENT_BINARY_DIR}/superlu_mod.mod
+    ${CMAKE_CURRENT_BINARY_DIR}/superlupara_mod.mod
+    ${CMAKE_CURRENT_BINARY_DIR}/superlu_dist_config.fh
 #    ${CMAKE_CURRENT_BINARY_DIR}/fortran_config.h
     )
 
@@ -24,6 +24,12 @@ set(targets superlu_dist_fortran)
 # depends on FPP defs
 add_dependencies(superlu_dist_fortran config_f)
 
+target_link_libraries(superlu_dist_fortran superlu_dist)
+
+if (CYGWIN)
+  set_target_properties(superlu_dist_fortran PROPERTIES SOVERSION ${VERSION_MAJOR})
+endif()
+
 install(TARGETS superlu_dist_fortran
 # DESTINATION ${CMAKE_INSTALL_LIBDIR}
     RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
