--- origsrc/dynamips-0.2.24/cmake/dependencies.cmake	2026-04-18 20:11:19.000000000 +0900
+++ src/dynamips-0.2.24/cmake/dependencies.cmake	2026-05-10 14:25:19.924527900 +0900
@@ -39,9 +39,6 @@ set ( DYNAMIPS_FLAGS -Wall -O2 -fomit-fr
 set ( DYNAMIPS_DEFINITIONS )
 set ( DYNAMIPS_INCLUDES )
 set ( DYNAMIPS_LIBRARIES ${CMAKE_DL_LIBS} )
-if ( CYGWIN )
-   list ( APPEND DYNAMIPS_FLAGS -static -static-libgcc )
-endif ( CYGWIN )
 macro ( set_cmake_required )
    set ( CMAKE_REQUIRED_FLAGS       ${DYNAMIPS_FLAGS} )
    set ( CMAKE_REQUIRED_DEFINITIONS ${DYNAMIPS_DEFINITIONS} )
