--- origsrc/dynamips-0.2.21/cmake/dependencies.cmake	2019-07-12 18:41:16.000000000 +0900
+++ src/dynamips-0.2.21/cmake/dependencies.cmake	2019-08-09 10:44:40.009835100 +0900
@@ -39,9 +39,10 @@ set ( DYNAMIPS_FLAGS -Wall -O2 -fomit-fr
 set ( DYNAMIPS_DEFINITIONS )
 set ( DYNAMIPS_INCLUDES )
 set ( DYNAMIPS_LIBRARIES ${CMAKE_DL_LIBS} )
-if ( CYGWIN )
-   list ( APPEND DYNAMIPS_FLAGS -static -static-libgcc )
-endif ( CYGWIN )
+if (CYGWIN)
+  list ( APPEND DYNAMIPS_LIBRARIES iconv )
+  list ( APPEND DYNAMIPS_LIBRARIES intl )
+endif()
 macro ( set_cmake_required )
    set ( CMAKE_REQUIRED_FLAGS       ${DYNAMIPS_FLAGS} )
    set ( CMAKE_REQUIRED_DEFINITIONS ${DYNAMIPS_DEFINITIONS} )
