--- origsrc/libskk-0.0.12/libskk/Makefile.am	2012-02-27 15:28:16.000000000 +0900
+++ src/libskk-0.0.12/libskk/Makefile.am	2012-04-11 23:12:27.701654900 +0900
@@ -44,7 +44,7 @@ libskk_la_CFLAGS =				\
 	-DPKGDATADIR=\"$(pkgdatadir)\"		\
 	$(NULL)
 libskk_la_LIBADD = $(LIBSKK_LIBS)
-libskk_la_LDFLAGS =							\
+libskk_la_LDFLAGS = -no-undefined					\
 	$(CODE_COVERAGE_LDFLAGS)					\
 	-version-info "$(LT_CURRENT)":"$(LT_REVISION)":"$(LT_AGE)"	\
 	-export-dynamic							\
