--- origsrc/rfc2822/Makefile.am	2010-12-26 14:08:53.000000000 +0900
+++ src/rfc2822/Makefile.am	2010-12-26 14:40:01.917503700 +0900
@@ -2,9 +2,11 @@
 # Build librfc2822
 #
 
+SUBDIRS = test
+
 lib_LTLIBRARIES = librfc2822.la
 
-librfc2822_la_LDFLAGS = -version-info 2:0:0
+librfc2822_la_LDFLAGS = -version-info 2:0:0 -no-undefined
 
 librfc2822_la_SOURCES =		\
   src/addr-spec.cpp		\
--- origsrc/rfc2822/configure.ac	2010-12-26 14:08:53.000000000 +0900
+++ src/rfc2822/configure.ac	2010-12-26 14:40:48.552360700 +0900
@@ -12,8 +12,8 @@ CXXFLAGS=${CXXFLAGS}
 dnl Checks for programs.
 AC_PROG_CXX
 AC_LANG([C++])
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 dnl Write results.
-AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([Makefile test/Makefile])
 AC_OUTPUT
--- origsrc/rfc2822/test/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/rfc2822/test/Makefile.am	2010-12-26 15:01:17.275783300 +0900
@@ -0,0 +1,12 @@
+
+INCLUDES = -I$(top_srcdir)
+
+check_PROGRAMS = address date linkage
+
+address_SOURCES = address.cpp
+address_LDADD = $(top_builddir)/librfc2822.la -lboost_unit_test_framework-mt
+date_SOURCES = date.cpp
+date_LDADD = $(top_builddir)/librfc2822.la -lboost_unit_test_framework-mt
+linkage_SOURCES = linkage.cpp linkage1.cpp linkage2.cpp
+linkage_LDADD = $(top_builddir)/librfc2822.la -lboost_unit_test_framework-mt
+
