--- origsrc/libsvg2fps/Makefile.am	2025-01-10 13:48:40.000000000 +0900
+++ src/libsvg2fps/Makefile.am	2025-01-10 13:52:27.158860400 +0900
@@ -21,7 +21,7 @@ pkginclude_HEADERS = svg2fps.h svg2fps_a
 
 lib_LTLIBRARIES = libsvg2fps.la
 libsvg2fps_la_SOURCES = svg2fps.c svg2fps_error.c
-libsvg2fps_la_LDFLAGS = -release $(engine) -version-info $(VERSION_INFO)
+libsvg2fps_la_LDFLAGS = -no-undefined -version-info $(VERSION_INFO)
 
 if GTK
 libsvg2fps_la_SOURCES += svg2fps_animation_gtk.c
@@ -31,3 +31,5 @@ endif
 
 dist_doc_DATA = README.md LICENSE AUTHORS
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libsvg2fps/configure.ac	2025-01-10 13:48:40.000000000 +0900
+++ src/libsvg2fps/configure.ac	2025-01-10 13:48:41.859236800 +0900
@@ -26,6 +26,5 @@ AC_ARG_VAR([engine], [implementation use
 AS_CASE([$engine], [engine=gtk AC_SUBST([VERSION_INFO], [0:0:0]) PKG_CHECK_MODULES([libwebkitgtk], [webkitgtk-3.0])])
 AM_CONDITIONAL([GTK], [test "$engine" = gtk])
 
-AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([libsvg2fps.pc Makefile])
 AC_OUTPUT
-
--- origsrc/libsvg2fps/libsvg2fps.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libsvg2fps/libsvg2fps.pc.in	2025-01-10 13:48:41.859236800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libsvg2fps
+Description: Render SVG animation as series of PNG images
+Version: @VERSION@
+Libs: -L${libdir} -lsvg2fps
+Cflags: -I${includedir}
