--- origsrc/wolfTPM-2.4.0/Makefile.am	2022-05-08 06:28:20.000000000 +0900
+++ src/wolfTPM-2.4.0/Makefile.am	2022-06-20 20:15:38.872115800 +0900
@@ -106,3 +106,6 @@ merge-clean:
 	@find ./ | $(GREP) \.OTHER | xargs rm -f
 	@find ./ | $(GREP) \.BASE | xargs rm -f
 	@find ./ | $(GREP) \~$$ | xargs rm -f
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/wolfTPM-2.4.0/configure.ac	2022-05-08 06:28:20.000000000 +0900
+++ src/wolfTPM-2.4.0/configure.ac	2022-06-20 20:16:34.356212000 +0900
@@ -6,7 +6,6 @@ AC_COPYRIGHT([Copyright (C) 2014-2021 wo
 AC_INIT([wolftpm],[2.4.0],[https://github.com/wolfssl/wolfTPM/issues],[wolftpm],[http://www.wolfssl.com])
 
 AC_PREREQ([2.63])
-AC_CONFIG_AUX_DIR([build-aux])
 
 # The following sets CFLAGS to empty if unset on command line.
 : ${CFLAGS=""}
@@ -63,6 +62,7 @@ AC_CHECK_SIZEOF([long], 4)
 # Check headers/libs
 AC_CHECK_FUNCS([gethostbyname getaddrinfo gettimeofday inet_ntoa memset socket])
 AC_CHECK_LIB([network],[socket])
+AM_ICONV
 
 # DEBUG
 DEBUG_CFLAGS="-g -O0 -DDEBUG_WOLFTPM"
@@ -418,7 +418,7 @@ AC_SUBST([AM_CFLAGS])
 AC_SUBST([AM_LDFLAGS])
 
 # FINAL
-AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([wolftpm.pc Makefile])
 AC_CONFIG_FILES([wolftpm/version.h])
 AC_CONFIG_FILES([wolftpm/options.h])
 
--- origsrc/wolfTPM-2.4.0/docs/include.am	2022-05-08 06:28:20.000000000 +0900
+++ src/wolfTPM-2.4.0/docs/include.am	2022-06-20 20:27:39.484804300 +0900
@@ -5,7 +5,7 @@
 dist_doc_DATA+= docs/README.md
 dist_doc_DATA+= docs/SWTPM.md
 dist_doc_DATA+= docs/WindowTBS.md
-dist_doc_DATA+= docs/Doxyfile
+# dist_doc_DATA+= docs/Doxyfile
 
 
 dox:
--- origsrc/wolfTPM-2.4.0/wolftpm.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/wolfTPM-2.4.0/wolftpm.pc.in	2022-06-20 20:16:54.859685900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: wolfTPM
+Description: Portable TPM 2.0 project designed for embedded use
+Version: @VERSION@
+Libs: -L${libdir} -lwolftpm
+Cflags: -I${includedir}
