--- origsrc/wolfTPM-3.10.0/Makefile.am	2025-12-05 07:29:41.000000000 +0900
+++ src/wolfTPM-3.10.0/Makefile.am	2026-02-08 23:37:25.097695400 +0900
@@ -127,3 +127,6 @@ cppcheck:
 		--suppress=invalidPrintfArgType_sint \
 		--error-exitcode=89 --std=c89 \
 		-I wolftpm src/ hal/ examples
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/wolfTPM-3.10.0/configure.ac	2025-12-05 07:29:41.000000000 +0900
+++ src/wolfTPM-3.10.0/configure.ac	2026-02-08 23:37:25.118570700 +0900
@@ -6,7 +6,6 @@ AC_COPYRIGHT([Copyright (C) 2014-2025 wo
 AC_INIT([wolftpm],[3.10.0],[https://github.com/wolfssl/wolfTPM/issues],[wolftpm],[http://www.wolfssl.com])
 
 AC_PREREQ([2.63])
-AC_CONFIG_AUX_DIR([build-aux])
 
 # The following sets CFLAGS to empty if unset on command line.
 : ${CFLAGS=""}
@@ -62,6 +61,7 @@ AC_CHECK_SIZEOF([long], 4)
 AC_CHECK_HEADERS([netdb.h])
 AC_CHECK_FUNCS([gethostbyname getaddrinfo gettimeofday inet_ntoa memset socket getpid])
 AC_CHECK_LIB([network],[socket])
+AM_ICONV
 
 # Thread local storage
 AX_TLS([thread_ls_on=yes],[thread_ls_on=no])
@@ -508,6 +508,7 @@ AC_CONFIG_FILES([wolftpm/options.h])
 AX_CREATE_GENERIC_CONFIG
 AX_AM_JOBSERVER([yes])
 
+AC_CONFIG_FILES([wolftpm.pc])
 AC_OUTPUT
 
 # force make clean
--- origsrc/wolfTPM-3.10.0/docs/include.am	2025-12-05 07:29:41.000000000 +0900
+++ src/wolfTPM-3.10.0/docs/include.am	2026-02-08 23:13:50.933734000 +0900
@@ -5,7 +5,7 @@
 dist_doc_DATA+= docs/README.md
 dist_doc_DATA+= docs/SWTPM.md
 dist_doc_DATA+= docs/WindowTBS.md
-dist_doc_DATA+= docs/Doxyfile
+# dist_doc_DATA+= docs/Doxyfile
 
 
 dox:
--- origsrc/wolfTPM-3.10.0/wolftpm.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/wolfTPM-3.10.0/wolftpm.pc.in	2026-02-08 23:37:25.047610500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: wolftpm
+Description: Portable TPM 2.0 project designed for embedded use
+Version: @VERSION@
+Libs: -L${libdir} -lwolftpm
+Cflags: -I${includedir}
