--- origsrc/ltl-2.0.19/configure.ac	2015-03-10 01:22:54.000000000 +0900
+++ src/ltl-2.0.19/configure.ac	2019-06-13 11:48:17.748784800 +0900
@@ -88,7 +88,7 @@ AC_SYS_LARGEFILE_SENSITIVE
 AC_SYS_LARGEFILE
 
 AC_PROG_CXXCPP
-AC_PROG_RANLIB
+LT_INIT
 
 dnl Checks for libraries.
 
--- origsrc/ltl-2.0.19/ltl/Makefile.am	2013-12-18 05:45:33.000000000 +0900
+++ src/ltl-2.0.19/ltl/Makefile.am	2019-06-13 11:54:47.719747700 +0900
@@ -1,6 +1,6 @@
-lib_LIBRARIES = libltl.a 
+lib_LTLIBRARIES = libltl.la 
 
-libltl_a_SOURCES = 				\
+libltl_la_SOURCES = 				\
 	io/fits.cpp			 	\
 	io/fits_card.cpp			\
 	io/fits_const.cpp			\
@@ -17,6 +17,7 @@ libltl_a_SOURCES = 				\
 	util/region.cpp				\
 	util/utdate.cpp				\
 	util/gnuplot.cpp
+libltl_la_LDFLAGS = -no-undefined
 
-AM_CPPFLAGS = -I../ 
+AM_CPPFLAGS = -I../ -I$(top_srcdir)
 
--- origsrc/ltl-2.0.19/test/Makefile.am	2014-09-04 05:12:59.000000000 +0900
+++ src/ltl-2.0.19/test/Makefile.am	2019-06-13 12:06:44.912541500 +0900
@@ -157,15 +157,15 @@ testcomplex_SOURCES = testcomplex.cpp
 testcomplex_LDFLAGS = $(LIBS)
 endif
 
-testfitsio_LDADD = -L../ltl -lltl
-testfitsextensionio_LDADD = -L../ltl -lltl
-testascio_LDADD  = -L../ltl -lltl
-testutildate_LDADD  = -L../ltl -lltl
-testutilregion_LDADD  = -L../ltl -lltl
-testconfreader_LDADD  = -L../ltl -lltl
+testfitsio_LDADD = $(top_builddir)/ltl/libltl.la
+testfitsextensionio_LDADD = $(top_builddir)/ltl/libltl.la
+testascio_LDADD  = $(top_builddir)/ltl/libltl.la
+testutildate_LDADD  = $(top_builddir)/ltl/libltl.la
+testutilregion_LDADD  = $(top_builddir)/ltl/libltl.la
+testconfreader_LDADD  = $(top_builddir)/ltl/libltl.la
 
 
-AM_CPPFLAGS = -I../
+AM_CPPFLAGS = -I../ -I$(top_srcdir) -I$(top_srcdir)/ltl
 CLEANFILES = test-64.fits test-32.fits test32.fits test16.fits test8.fits \
 	     testextensionio.fits test_inmem_float.fits \
              testc.fits test-reg.fits test.map test.dat ltl-test.conf \
