--- origsrc/libtinynotify-0.2.1/Makefile.am	2012-01-10 05:34:25.000000000 +0900
+++ src/libtinynotify-0.2.1/Makefile.am	2012-01-15 20:20:35.057252700 +0900
@@ -16,7 +16,7 @@ subinclude_HEADERS = \
 	lib/notification.h \
 	lib/event.h
 
-libtinynotify_la_LDFLAGS = -version-info 2:1:2
+libtinynotify_la_LDFLAGS = -version-info 2:1:2 -no-undefined
 libtinynotify_la_CPPFLAGS = $(DBUS_CFLAGS)
 libtinynotify_la_LIBADD = $(DBUS_LIBS)
 libtinynotify_la_SOURCES = \
