--- origsrc/librfm5-5.3.16.3/Build/lib/Makefile.am	2015-04-21 00:55:25.000000000 +0900
+++ src/librfm5-5.3.16.3/Build/lib/Makefile.am	2017-10-07 23:16:56.386594600 +0900
@@ -10,7 +10,7 @@ AUTOMAKE_OPTIONS = subdir-objects
 AM_CPPFLAGS = \
         -I$(top_srcdir) \
         -I$(includedir) \
-        -I../../primary  \
+        -I$(top_srcdir)/primary  \
 	@DBH_CFLAGS@	\
 	@LIBXML_CFLAGS@	\
 	@GLIB_CFLAGS@	\
@@ -25,11 +25,9 @@ AM_CPPFLAGS = \
 
 common_ldflags = @DBH_LIBS@ @TUBO_LIBS@ @GMODULE_LIBS@ @GTHREAD_LIBS@ @CAIRO_LIBS@	 @GLIB_LIBS@	 @GTK_LIBS@	 @LIBXML_LIBS@	@LIBZIP_LIBS@
 
-if WINDOZE
 no_undefined = -no-undefined
 # List symbols here...
 #export_symbols = -export-symbols 
-endif
 
 ## rfm library
 lib_LTLIBRARIES += librfm.la
@@ -43,7 +41,7 @@ lib_LTLIBRARIES += librodent.la
 librodent_la_CFLAGS =  -I../../rodent
 librodent_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
     -export-dynamic $(no_undefined) $(export_symbols) $(bsd_compat)
-librodent_la_LIBADD = $(common_ldflags) 
+librodent_la_LIBADD = $(common_ldflags) librfm.la
 
 # sources...
 librfm_la_SOURCES = \
--- origsrc/librfm5-5.3.16.3/Build/modules/Makefile.am	2014-11-13 23:38:54.000000000 +0900
+++ src/librfm5-5.3.16.3/Build/modules/Makefile.am	2017-10-07 23:16:56.386594600 +0900
@@ -21,11 +21,11 @@ common_ldflags = @DBH_LIBS@ @TUBO_LIBS@
 
 
 
-module_libs = $(rfmlibs) -avoid-version -module	
+module_libs = -no-undefined $(rfmlibs) -avoid-version -module $(top_builddir)/Build/lib/librfm.la $(top_builddir)/Build/lib/librodent.la
 AM_CPPFLAGS =  \
         -I$(top_srcdir) \
         -I$(includedir) \
-	-I../../primary\
+	-I$(top_srcdir)/primary\
 	@DBH_CFLAGS@	\
 	@LIBXML_CFLAGS@	\
 	@GLIB_CFLAGS@	\
--- origsrc/librfm5-5.3.16.3/Build/share/Makefile.am	2014-01-16 00:33:09.000000000 +0900
+++ src/librfm5-5.3.16.3/Build/share/Makefile.am	2017-10-07 23:16:56.402042700 +0900
@@ -5,9 +5,7 @@ DIST_SUBDIRS =  rfm mime
 
 if INSTALL_IMAGES
 images_DATA=images/juchi-2007.jpg images/roa153b.jpg
-imagesdir=$(datadir)/images
+imagesdir=$(datadir)/librfm/images
 endif
 
 EXTRA_DIST =  $(images_DATA) 
-
-
--- origsrc/librfm5-5.3.16.3/configure.ac	2015-06-04 00:26:18.000000000 +0900
+++ src/librfm5-5.3.16.3/configure.ac	2017-10-07 23:16:56.402042700 +0900
@@ -371,7 +371,9 @@ AC_CHECK_HEADERS([sys/time.h sys/select.
 AC_CHECK_HEADERS([mntent.h])
 
 AC_CHECK_HEADERS([fstab.h sys/param.h sys/ucred.h sys/mount.h])
+if test "$host_os" != "cygwin"; then
 AC_CHECK_HEADERS([windows.h])
+fi
 AM_CONDITIONAL(WINDOZE, [test "$ac_cv_header_windows_h" = yes])
 
 AC_SUBST(LARGEFILE) LARGEFILE=yes
@@ -514,5 +516,3 @@ if test x"$I_PREFIX" != x ; then
     AC_MSG_WARN([BIG FAT WARNING Package is already installed in $I_PREFIX. Removal of installation in $I_PREFIX is recommended before proceeding to install package in $ac_default_prefix.])
   fi
 fi
-
-
--- origsrc/librfm5-5.3.16.3/include/rfm.h	2015-06-04 03:12:22.000000000 +0900
+++ src/librfm5-5.3.16.3/include/rfm.h	2017-10-07 23:20:17.324354400 +0900
@@ -10,7 +10,7 @@
 #include "tubo.h"
 
 // These defines are set by configuration script when librfm is installed.
-#define RFM_MODULE_DIR "/tmp/test/lib/rfm/rmodules"
+#define RFM_MODULE_DIR "/usr/lib/rfm/rmodules"
 
 
 # define SHOW_IT(y) if(rfm_get_widget(y)&&GTK_IS_WIDGET(rfm_get_widget(y))) gtk_widget_show_all(rfm_get_widget(y))
