--- origsrc/hfstospell-0.5.1/configure.ac	2018-10-15 20:23:24.000000000 +0900
+++ src/hfstospell-0.5.1/configure.ac	2019-01-08 08:07:24.024852000 +0900
@@ -148,6 +148,7 @@ AC_CHECK_FUNCS([strndup error])
 # Checks for system services
 
 # Require highest supported C++ standard
+if test "$host_os" != "cygwin"; then
 AC_LANG(C++)
 AX_CHECK_COMPILE_FLAG([-std=c++20], [CXXFLAGS="$CXXFLAGS -std=c++20"], [
  AX_CHECK_COMPILE_FLAG([-std=c++2a], [CXXFLAGS="$CXXFLAGS -std=c++2a"], [
@@ -162,6 +163,7 @@ AX_CHECK_COMPILE_FLAG([-std=c++20], [CXX
   ])
  ])
 ])
+fi
 
 # config files
 AC_CONFIG_FILES([Makefile hfstospell.pc])
