--- origsrc/discount-3.0.1.3/cmake/CMakeLists.txt	2026-06-10 04:44:21.000000000 +0900
+++ src/discount-3.0.1.3/cmake/CMakeLists.txt	2026-08-06 11:23:03.068370900 +0900
@@ -168,6 +168,10 @@ target_include_directories(libmarkdown
         $<BUILD_INTERFACE:${_ROOT}>
 )
 
+if (DEFINED SOVERSION)
+    set_target_properties(libmarkdown PROPERTIES SOVERSION ${SOVERSION})
+endif()
+
 if(NOT ${PROJECT_NAME}_ONLY_LIBRARY)
     add_library(common OBJECT
         "${_ROOT}/pgm_options.c"
