--- origsrc/discount-3.0.1.2/cmake/CMakeLists.txt	2025-09-09 01:51:18.000000000 +0900
+++ src/discount-3.0.1.2/cmake/CMakeLists.txt	2026-01-05 06:59:47.271823800 +0900
@@ -168,6 +168,10 @@ target_include_directories(libmarkdown
         $<BUILD_INTERFACE:${_ROOT}>
 )
 
+if (DEFINED SOVERSION)
+    set_target_properties(libmarkdown PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 if(NOT ${PROJECT_NAME}_ONLY_LIBRARY)
     add_library(common OBJECT
         "${_ROOT}/pgm_options.c"
