diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/flyby/CMakeLists.txt src/flyby/CMakeLists.txt
--- origsrc/flyby/CMakeLists.txt	2018-04-25 23:41:10.000000000 +0900
+++ src/flyby/CMakeLists.txt	2018-04-25 23:47:51.835000000 +0900
@@ -35,7 +35,8 @@ link_directories(${PREDICT_LIBRARY_DIRS}
 add_executable(flyby src/ui.c src/hamlib.c src/main.c src/string_array.c src/xdg_basedirs.c src/xdg_basedir_extras.c src/tle_db.c src/transponder_db.c src/qth_config.c src/filtered_menu.c src/transponder_editor.c src/multitrack.c src/locator.c src/option_help.c src/singletrack.c src/prediction_schedules.c src/hamlib_status.c src/field_helpers.c src/track_astronomical_bodies.c)
 install(TARGETS flyby RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 
-target_link_libraries(flyby m ncurses menu form ${PREDICT_LIBRARIES})
+pkg_check_modules(NCURSES ncurses menu form REQUIRED)
+target_link_libraries(flyby m ${NCURSES_LIBRARIES} ${PREDICT_LIBRARIES})
 
 #transponder database utility
 set(TRANSPONDER_UTILITY_NAME "flyby-transponder-dbutil") #name of transponder utility executable
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/flyby/tests/CMakeLists.txt src/flyby/tests/CMakeLists.txt
--- origsrc/flyby/tests/CMakeLists.txt	2018-04-25 23:41:10.000000000 +0900
+++ src/flyby/tests/CMakeLists.txt	2018-04-25 23:50:09.834000200 +0900
@@ -1,7 +1,8 @@
 add_executable(dummy-t dummy-t.c)
 add_test(NAME dummy COMMAND dummy-t)
 
-find_package(CMocka REQUIRED)
+pkg_check_modules(CMOCKA cmocka REQUIRED)
+set(CMOCKA_LIBRARY ${CMOCKA_LIBRARIES})
 
 #xdg basedir tests
 add_executable(xdg-dirs-t xdg-dirs-t.c ${CMAKE_SOURCE_DIR}/src/xdg_basedirs.c ${CMAKE_SOURCE_DIR}/src/string_array.c)
