--- origsrc/mp3check-0.8.7/Makefile	2012-06-05 07:33:07.000000000 +0900
+++ src/mp3check-0.8.7/Makefile	2015-05-06 11:42:19.625067400 +0900
@@ -4,7 +4,7 @@
 # --- config ----------------------------------------------------------------
 WARN = -Wall -W -g
 #OPT = -O2
-OPT =
+OPT = -O3
 CPPFLAGS = $(ADDITIONAL_CPPFLAGS)
 CXXFLAGS = $(WARN) $(OPT)
 CXX = g++
@@ -66,4 +66,3 @@ ifeq ($(findstring $(MAKECMDGOALS),svncl
 -include $(DEP)
 endif
 endif
-		
