--- origsrc/g15daemon-1.9.5.3/Makefile.am	2008-01-24 21:51:09.000000000 +0900
+++ src/g15daemon-1.9.5.3/Makefile.am	2012-12-24 15:18:43.824546900 +0900
@@ -1,6 +1,6 @@
 LLIBDIR=@LIBDIR@
 
-SUBDIRS =  libg15daemon_client g15daemon plugins
+SUBDIRS =  libg15daemon_client g15daemon
 INCLUDES =  -I$(top_srcdir)/libg15daemon_client -I$(top_srcdir)/g15daemon
 
 EXTRA_DIST = debian contrib Documentation lang-bindings patches rpm README.usage FAQ LICENSE images README.Linux README.FreeBSD README.Solaris
--- origsrc/g15daemon-1.9.5.3/g15daemon/Makefile.am	2008-01-24 21:51:07.000000000 +0900
+++ src/g15daemon-1.9.5.3/g15daemon/Makefile.am	2012-12-24 13:31:52.966739700 +0900
@@ -8,6 +8,6 @@ g15daemon_SOURCES = utility_funcs.c g15d
 g15daemon_LDADD = -ldl
 g15daemon_LDFLAGS = -rdynamic
 g15daemontest_SOURCES = lcdclient_test.c
-INCLUDES = -I$(top_builddir)/libg15daemon_client/
+INCLUDES = -I$(top_builddir)/libg15daemon_client/ -I$(top_srcdir)/libg15daemon_client/
 g15daemontest_LDADD = $(top_builddir)/libg15daemon_client/libg15daemon_client.la
 include_HEADERS = g15daemon.h
--- origsrc/g15daemon-1.9.5.3/libg15daemon_client/Makefile.am	2008-01-24 21:51:09.000000000 +0900
+++ src/g15daemon-1.9.5.3/libg15daemon_client/Makefile.am	2012-12-24 13:04:59.795432700 +0900
@@ -1,6 +1,6 @@
 
 METASOURCES = AUTO
 lib_LTLIBRARIES = libg15daemon_client.la
-libg15daemon_client_la_LDFLAGS = -version-info 1:2:0
+libg15daemon_client_la_LDFLAGS = -version-info 1:2:0 -no-undefined
 libg15daemon_client_la_SOURCES = g15daemon_client.h g15daemon_net.c
-include_HEADERS= g15daemon_client.h
\ No newline at end of file
+include_HEADERS= g15daemon_client.h
--- origsrc/g15daemon-1.9.5.3/plugins/Makefile.am	2008-01-24 21:51:07.000000000 +0900
+++ src/g15daemon-1.9.5.3/plugins/Makefile.am	2012-12-24 15:14:48.797203200 +0900
@@ -5,16 +5,16 @@ AM_CFLAGS = -Wall
 
 if UINPUT_INTERFACE_PLUGIN
 g15plugin_uinput_la_SOURCES = g15_plugin_uinput.c
-g15plugin_uinput_la_LDFLAGS = -avoid-version -module 
+g15plugin_uinput_la_LDFLAGS = -avoid-version -module -no-undefined
 input_la = g15plugin_uinput.la
 endif
 
 lib_LTLIBRARIES = ${input_la} g15plugin_tcpserver.la g15plugin_clock.la
-INCLUDES = -I$(top_builddir)/libg15daemon_client/ -I$(top_builddir)/g15daemon
+INCLUDES = -I$(top_builddir)/libg15daemon_client/ -I$(top_builddir)/g15daemon -I$(top_srcdir)/g15daemon
 
 g15plugin_tcpserver_la_SOURCES = g15_plugin_net.c
-g15plugin_tcpserver_la_LDFLAGS = -avoid-version -module 
+g15plugin_tcpserver_la_LDFLAGS = -avoid-version -module -no-undefined
 
 g15plugin_clock_la_SOURCES = g15_plugin_clock.c
-g15plugin_clock_la_LDFLAGS = -avoid-version -module 
+g15plugin_clock_la_LDFLAGS = -avoid-version -module -no-undefined
 
