--- origsrc/libbraiding-1.3.1/configure.ac	2024-11-07 03:51:29.000000000 +0900
+++ src/libbraiding-1.3.1/configure.ac	2025-01-05 14:23:21.285706300 +0900
@@ -8,4 +8,3 @@ AC_PROG_CXX
 LT_INIT
 AC_CONFIG_FILES([Makefile lib/Makefile libbraiding.pc])
 AC_OUTPUT
-s
--- origsrc/libbraiding-1.3.1/lib/Makefile.am	2024-11-07 03:51:29.000000000 +0900
+++ src/libbraiding-1.3.1/lib/Makefile.am	2025-01-05 14:17:33.679230400 +0900
@@ -1,4 +1,4 @@
 lib_LTLIBRARIES = libbraiding.la
 libbraiding_la_SOURCES = cbraid.cpp braiding.cpp braiding.h cbraid.h cbraid_implementation.h  cbraid_interface.h
-libbraiding_la_LDFLAGS = -lm
+libbraiding_la_LDFLAGS = -no-undefined -lm
 include_HEADERS = braiding.h cbraid.h cbraid_implementation.h  cbraid_interface.h
\ No newline at end of file
