--- origsrc/unibilium-1.1.2/Makefile	2015-02-22 10:20:48.000000000 +0900
+++ src/unibilium-1.1.2/Makefile	2015-04-10 13:03:22.233139900 +0900
@@ -59,7 +59,7 @@ uniutil.lo: uniutil.c unibilium.h
 	$(LIBTOOL) --mode=compile --tag=CC $(CC) -I. -DTERMINFO_DIRS='$(TERMINFO_DIRS)' -Wall -std=c99 $(CFLAGS) $(CFLAGS_DEBUG) -o $@ -c $<
 
 $(LIBRARY): $(OBJECTS)
-	$(LIBTOOL) --mode=link --tag=CC $(CC) -rpath '$(LIBDIR)' -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -o $@ $^
+	$(LIBTOOL) --mode=link --tag=CC $(CC) -rpath '$(LIBDIR)' -no-undefined -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -o $@ $^
 
 tools/%: $(LIBRARY) tools/%.lo
 	$(LIBTOOL) --mode=link --tag=CC $(CC) -o $@ $^
