--- origsrc/wolfssl-4.4.0-stable/configure.ac	2020-04-23 02:43:44.000000000 +0900
+++ src/wolfssl-4.4.0-stable/configure.ac	2020-08-08 09:11:32.502009600 +0900
@@ -2848,7 +2848,7 @@ AC_ARG_ENABLE([crl-monitor],
 if test "$ENABLED_CRL_MONITOR" = "yes"
 then
     case $host_os in
-    *linux* | *darwin* | *freebsd*)
+    *linux* | *darwin* | *freebsd* | *cygwin*)
         AM_CFLAGS="$AM_CFLAGS -DHAVE_CRL_MONITOR" ;;
     *)
         if test "x$ENABLED_DISTRO" = "xyes" ; then
