--- origsrc/wolfssl-4.6.0-stable/configure.ac	2020-12-23 09:58:25.000000000 +0900
+++ src/wolfssl-4.6.0-stable/configure.ac	2021-01-11 17:59:20.782946300 +0900
@@ -3218,7 +3218,7 @@ AC_ARG_ENABLE([crl-monitor],
 if test "$ENABLED_CRL_MONITOR" = "yes"
 then
     case $host_os in
-    *linux* | *darwin* | *freebsd*)
+    *linux* | *darwin* | *freebsd* | *cygwin*)
         if test "x$ENABLED_SINGLETHREADED" = "xno"; then
             AM_CFLAGS="$AM_CFLAGS -DHAVE_CRL_MONITOR"
         else
