--- origsrc/htslib/Makefile	2026-08-11 07:17:56.000000000 +0000
+++ src/htslib/Makefile	2026-08-11 07:18:03.644496300 +0000
@@ -27,7 +27,7 @@ AR     = ar
 RANLIB = ranlib
 
 # Default libraries to link if configure is not used
-htslib_default_libs = -lz -lm -lbz2 -llzma -lcurl
+htslib_default_libs = -lz -lm -lbz2 -llzma -lcurl -lhtscodecs
 
 CPPFLAGS =
 # TODO: make the 64-bit support for VCF optional via configure, for now add -DVCF_ALLOW_INT64
@@ -148,7 +148,7 @@ HTS_BUILD_AVX512 =
 HTS_BUILD_SSE4 =
 
 include htslib_vars.mk
-include htscodecs.mk
+# include htscodecs.mk
 
 # If not using GNU make, you need to copy the version number from version.sh
 # into here.
@@ -566,16 +566,16 @@ htscodecs/htscodecs/rANS_static32x16pr_a
 htscodecs/htscodecs/rANS_static32x16pr_sse4.o htscodecs/htscodecs/rANS_static32x16pr_sse4.pico: TARGET_CFLAGS = $(HTS_CFLAGS_SSE4)
 
 annot-tsv: annot-tsv.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ annot-tsv.o libhts.a $(LIBS) -lpthread
+	$(CC) $(LDFLAGS) -o $@ annot-tsv.o $(LIBS) -lpthread -L. -lhts
 
 bgzip: bgzip.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ bgzip.o libhts.a $(LIBS) -lpthread
+	$(CC) $(LDFLAGS) -o $@ bgzip.o $(LIBS) -lpthread -L. -lhts
 
 htsfile: htsfile.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ htsfile.o libhts.a $(LIBS) -lpthread
+	$(CC) $(LDFLAGS) -o $@ htsfile.o $(LIBS) -lpthread -L. -lhts
 
 tabix: tabix.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ tabix.o libhts.a $(LIBS) -lpthread
+	$(CC) $(LDFLAGS) -o $@ tabix.o $(LIBS) -lpthread -L. -lhts
 
 ref_cache/ref-cache: $(REF_CACHE_OBJS)
 	$(CC) $(LDFLAGS) $(REF_CACHE_EXTRA_LD_FLAGS) -o $@ $(REF_CACHE_OBJS) -lcurl
