--- origsrc/htslib-1.17/Makefile	2023-02-21 21:24:29.000000000 +0900
+++ src/htslib-1.17/Makefile	2023-07-24 22:06:24.151707600 +0900
@@ -497,13 +497,13 @@ htscodecs/htscodecs/rANS_static32x16pr_a
 htscodecs/htscodecs/rANS_static32x16pr_sse4.o htscodecs/htscodecs/rANS_static32x16pr_sse4.pico: TARGET_CFLAGS = $(HTS_CFLAGS_SSE4)
 
 bgzip: bgzip.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ bgzip.o libhts.a $(LIBS) -lpthread
+	$(CC) $(LDFLAGS) -o $@ bgzip.o $(LIBS) -lpthread -L. -lhts
 
 htsfile: htsfile.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ htsfile.o libhts.a $(LIBS) -lpthread
+	$(CC) $(LDFLAGS) -o $@ htsfile.o $(LIBS) -lpthread -L. -lhts
 
 tabix: tabix.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ tabix.o libhts.a $(LIBS) -lpthread
+	$(CC) $(LDFLAGS) -o $@ tabix.o $(LIBS) -lpthread -L. -lhts
 
 bgzip.o: bgzip.c config.h $(htslib_bgzf_h) $(htslib_hts_h) $(htslib_hfile_h)
 htsfile.o: htsfile.c config.h $(htslib_hfile_h) $(htslib_hts_h) $(htslib_sam_h) $(htslib_vcf_h)
--- origsrc/htslib-1.17/cram/cram_external.c	2023-01-20 19:05:41.000000000 +0900
+++ src/htslib-1.17/cram/cram_external.c	2023-07-24 22:08:28.153015200 +0900
@@ -43,7 +43,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 #include <stdint.h>
 
 #if defined(HAVE_EXTERNAL_LIBHTSCODECS)
-#include <htscodecs/rANS_static4x16.h.h>
+#include <htscodecs/rANS_static4x16.h>
 #else
 #include "../htscodecs/htscodecs/rANS_static4x16.h"
 #endif
