--- origsrc/htslib-1.12/Makefile	2021-03-17 18:14:15.000000000 +0900
+++ src/htslib-1.12/Makefile	2021-05-09 17:07:01.852864800 +0900
@@ -405,14 +405,14 @@ htscodecs/htscodecs/rle.o htscodecs/htsc
 htscodecs/htscodecs/tokenise_name3.o htscodecs/htscodecs/tokenise_name3.pico: htscodecs/htscodecs/tokenise_name3.c config.h $(htscodecs_pooled_alloc_h) $(htscodecs_arith_dynamic_h) $(htscodecs_rANS_static4x16_h) $(htscodecs_tokenise_name3_h) $(htscodecs_varint_h)
 
 
-bgzip: bgzip.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ bgzip.o libhts.a $(LIBS) -lpthread
+bgzip: bgzip.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ bgzip.o libhts.dll.a $(LIBS) -lpthread
 
-htsfile: htsfile.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ htsfile.o libhts.a $(LIBS) -lpthread
+htsfile: htsfile.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ htsfile.o libhts.dll.a $(LIBS) -lpthread
 
-tabix: tabix.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ tabix.o libhts.a $(LIBS) -lpthread
+tabix: tabix.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ tabix.o libhts.dll.a $(LIBS) -lpthread
 
 bgzip.o: bgzip.c config.h $(htslib_bgzf_h) $(htslib_hts_h)
 htsfile.o: htsfile.c config.h $(htslib_hfile_h) $(htslib_hts_h) $(htslib_sam_h) $(htslib_vcf_h)
@@ -498,67 +498,67 @@ test/hts_endian: test/hts_endian.o
 	$(CC) $(LDFLAGS) -o $@ test/hts_endian.o $(LIBS)
 
 test/fuzz/hts_open_fuzzer: test/fuzz/hts_open_fuzzer.o
-	$(CC) $(LDFLAGS) -o $@ test/fuzz/hts_open_fuzzer.o libhts.a $(LIBS) -lpthread
+	$(CC) $(LDFLAGS) -o $@ test/fuzz/hts_open_fuzzer.o libhts.dll.a $(LIBS) -lpthread
 
-test/fieldarith: test/fieldarith.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/fieldarith.o libhts.a $(LIBS) -lpthread
+test/fieldarith: test/fieldarith.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/fieldarith.o libhts.dll.a $(LIBS) -lpthread
 
-test/hfile: test/hfile.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/hfile.o libhts.a $(LIBS) -lpthread
+test/hfile: test/hfile.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/hfile.o libhts.dll.a $(LIBS) -lpthread
 
-test/pileup: test/pileup.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/pileup.o libhts.a $(LIBS) -lpthread
+test/pileup: test/pileup.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/pileup.o libhts.dll.a $(LIBS) -lpthread
 
 test/plugins-dlhts: test/plugins-dlhts.o
 	$(CC) $(LDFLAGS) -o $@ test/plugins-dlhts.o $(LIBS)
 
-test/sam: test/sam.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/sam.o libhts.a $(LIBS) -lpthread
+test/sam: test/sam.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/sam.o libhts.dll.a $(LIBS) -lpthread
 
-test/test_bgzf: test/test_bgzf.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/test_bgzf.o libhts.a -lz $(LIBS) -lpthread
+test/test_bgzf: test/test_bgzf.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/test_bgzf.o libhts.dll.a -lz $(LIBS) -lpthread
 
-test/test_expr: test/test_expr.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/test_expr.o libhts.a -lz $(LIBS) -lpthread
+test/test_expr: test/test_expr.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/test_expr.o libhts.dll.a -lz $(LIBS) -lpthread
 
-test/test_kfunc: test/test_kfunc.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/test_kfunc.o libhts.a -lz $(LIBS) -lpthread
+test/test_kfunc: test/test_kfunc.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/test_kfunc.o libhts.dll.a -lz $(LIBS) -lpthread
 
-test/test_kstring: test/test_kstring.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/test_kstring.o libhts.a -lz $(LIBS) -lpthread
+test/test_kstring: test/test_kstring.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/test_kstring.o libhts.dll.a -lz $(LIBS) -lpthread
 
-test/test_realn: test/test_realn.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/test_realn.o libhts.a $(LIBS) -lpthread
+test/test_realn: test/test_realn.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/test_realn.o libhts.dll.a $(LIBS) -lpthread
 
-test/test-regidx: test/test-regidx.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/test-regidx.o libhts.a $(LIBS) -lpthread
+test/test-regidx: test/test-regidx.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/test-regidx.o libhts.dll.a $(LIBS) -lpthread
 
-test/test-parse-reg: test/test-parse-reg.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/test-parse-reg.o libhts.a $(LIBS) -lpthread
+test/test-parse-reg: test/test-parse-reg.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/test-parse-reg.o libhts.dll.a $(LIBS) -lpthread
 
-test/test_str2int: test/test_str2int.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/test_str2int.o libhts.a $(LIBS) -lpthread
+test/test_str2int: test/test_str2int.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/test_str2int.o libhts.dll.a $(LIBS) -lpthread
 
-test/test_view: test/test_view.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/test_view.o libhts.a $(LIBS) -lpthread
+test/test_view: test/test_view.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/test_view.o libhts.dll.a $(LIBS) -lpthread
 
-test/test_index: test/test_index.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/test_index.o libhts.a $(LIBS) -lpthread
+test/test_index: test/test_index.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/test_index.o libhts.dll.a $(LIBS) -lpthread
 
-test/test-vcf-api: test/test-vcf-api.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/test-vcf-api.o libhts.a $(LIBS) -lpthread
+test/test-vcf-api: test/test-vcf-api.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/test-vcf-api.o libhts.dll.a $(LIBS) -lpthread
 
-test/test-vcf-sweep: test/test-vcf-sweep.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/test-vcf-sweep.o libhts.a $(LIBS) -lpthread
+test/test-vcf-sweep: test/test-vcf-sweep.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/test-vcf-sweep.o libhts.dll.a $(LIBS) -lpthread
 
-test/test-bcf-sr: test/test-bcf-sr.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/test-bcf-sr.o libhts.a -lz $(LIBS) -lpthread
+test/test-bcf-sr: test/test-bcf-sr.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/test-bcf-sr.o libhts.dll.a -lz $(LIBS) -lpthread
 
-test/test-bcf-translate: test/test-bcf-translate.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/test-bcf-translate.o libhts.a -lz $(LIBS) -lpthread
+test/test-bcf-translate: test/test-bcf-translate.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/test-bcf-translate.o libhts.dll.a -lz $(LIBS) -lpthread
 
-test/test_introspection: test/test_introspection.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/test_introspection.o libhts.a $(LIBS) -lpthread
+test/test_introspection: test/test_introspection.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/test_introspection.o libhts.dll.a $(LIBS) -lpthread
 
 # Extra tests for bundled htscodecs
 test_htscodecs_rans4x8: htscodecs/tests/rans4x8
@@ -634,26 +634,26 @@ test/test-bcf-translate.o: test/test-bcf
 test/test_introspection.o: test/test_introspection.c config.h $(htslib_hts_h) $(htslib_hfile_h)
 
 
-test/thrash_threads1: test/thrash_threads1.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/thrash_threads1.o libhts.a -lz $(LIBS) -lpthread
+test/thrash_threads1: test/thrash_threads1.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/thrash_threads1.o libhts.dll.a -lz $(LIBS) -lpthread
 
-test/thrash_threads2: test/thrash_threads2.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/thrash_threads2.o libhts.a -lz $(LIBS) -lpthread
+test/thrash_threads2: test/thrash_threads2.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/thrash_threads2.o libhts.dll.a -lz $(LIBS) -lpthread
 
-test/thrash_threads3: test/thrash_threads3.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/thrash_threads3.o libhts.a -lz $(LIBS) -lpthread
+test/thrash_threads3: test/thrash_threads3.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/thrash_threads3.o libhts.dll.a -lz $(LIBS) -lpthread
 
-test/thrash_threads4: test/thrash_threads4.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/thrash_threads4.o libhts.a -lz $(LIBS) -lpthread
+test/thrash_threads4: test/thrash_threads4.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/thrash_threads4.o libhts.dll.a -lz $(LIBS) -lpthread
 
-test/thrash_threads5: test/thrash_threads5.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/thrash_threads5.o libhts.a -lz $(LIBS) -lpthread
+test/thrash_threads5: test/thrash_threads5.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/thrash_threads5.o libhts.dll.a -lz $(LIBS) -lpthread
 
-test/thrash_threads6: test/thrash_threads6.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/thrash_threads6.o libhts.a -lz $(LIBS) -lpthread
+test/thrash_threads6: test/thrash_threads6.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/thrash_threads6.o libhts.dll.a -lz $(LIBS) -lpthread
 
-test/thrash_threads7: test/thrash_threads7.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ test/thrash_threads7.o libhts.a -lz $(LIBS) -lpthread
+test/thrash_threads7: test/thrash_threads7.o libhts.dll.a
+	$(CC) $(LDFLAGS) -o $@ test/thrash_threads7.o libhts.dll.a -lz $(LIBS) -lpthread
 
 test_thrash: $(BUILT_THRASH_PROGRAMS)
 
--- origsrc/htslib-1.12/textutils.c	2020-07-17 20:01:59.000000000 +0900
+++ src/htslib-1.12/textutils.c	2021-05-09 17:05:16.829130900 +0900
@@ -449,6 +449,7 @@ char *stringify_argv(int argc, char *arg
 
 /* Utility function for printing possibly malicious text data
  */
+HTSLIB_EXPORT
 const char *
 hts_strprint(char *buf, size_t buflen, char quote, const char *s, size_t len)
 {
