--- origsrc/htslib-1.14/Makefile	2021-10-22 20:14:47.000000000 +0900
+++ src/htslib-1.14/Makefile	2021-10-24 21:55:57.111277300 +0900
@@ -410,13 +410,13 @@ htscodecs/htscodecs/tokenise_name3.o hts
 
 
 bgzip: bgzip.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ bgzip.o libhts.a $(LIBS) -lpthread
+	$(CC) $(LDFLAGS) -o $@ bgzip.o $(LIBS) -lpthread -L. -lhts
 
 htsfile: htsfile.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ htsfile.o libhts.a $(LIBS) -lpthread
+	$(CC) $(LDFLAGS) -o $@ htsfile.o $(LIBS) -lpthread -L. -lhts
 
 tabix: tabix.o libhts.a
-	$(CC) $(LDFLAGS) -o $@ tabix.o libhts.a $(LIBS) -lpthread
+	$(CC) $(LDFLAGS) -o $@ tabix.o $(LIBS) -lpthread -L. -lhts
 
 bgzip.o: bgzip.c config.h $(htslib_bgzf_h) $(htslib_hts_h)
 htsfile.o: htsfile.c config.h $(htslib_hfile_h) $(htslib_hts_h) $(htslib_sam_h) $(htslib_vcf_h)
