--- origsrc/lunar-calendar-2.4.1/configure.ac	2011-05-07 22:29:29.000000000 +0900
+++ src/lunar-calendar-2.4.1/configure.ac	2011-06-05 21:12:52.307991400 +0900
@@ -41,7 +41,7 @@ AC_ISC_POSIX
 AC_HEADER_STDC
 
 AC_PROG_INSTALL
-AC_PROG_LIBTOOL
+LT_INIT
 
 AM_PATH_PYTHON
 
--- origsrc/lunar-calendar-2.4.1/lunar-calendar/Makefile.am	2011-05-07 22:16:43.000000000 +0900
+++ src/lunar-calendar-2.4.1/lunar-calendar/Makefile.am	2011-06-05 21:13:48.328545000 +0900
@@ -19,12 +19,12 @@ liblunar_calendar_2_0_la_SOURCES = $(sou
 liblunar_calendar_2_0_la_LIBADD = $(LUNAR_CALENDAR_LIBS)
 liblunar_calendar_2_0_la_LDFLAGS = \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
-	-export-dynamic $(export_symbols)
+	-export-dynamic $(export_symbols) -no-undefined
 
 liblunar_calendar_preload_2_0_includedir = $(includedir)/liblunar-2.0/lunar-calendar
 liblunar_calendar_preload_2_0_include_HEADERS = $(source_h)
 liblunar_calendar_preload_2_0_la_SOURCES = lunar-calendar-preload.c
-liblunar_calendar_preload_2_0_la_LDFLAGS =  -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) 
+liblunar_calendar_preload_2_0_la_LDFLAGS =  -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -no-undefined
 liblunar_calendar_preload_2_0_la_LIBADD = 	\
 	$(LUNAR_CALENDAR_LIBS)	\
 	liblunar-calendar-2.0.la		\
