--- origsrc/knot-3.4.6/src/knot/server/server.c	2025-04-10 13:47:25.000000000 +0900
+++ src/knot-3.4.6/src/knot/server/server.c	2025-04-13 15:14:08.764794800 +0900
@@ -203,7 +203,9 @@ static bool enable_pktinfo(int sock, int
 		break;
 	case AF_INET6:
 		level = IPPROTO_IPV6;
+#if defined(IPV6_RECVPKTINFO)
 		option = IPV6_RECVPKTINFO; /* Multiplatform */
+#endif
 		break;
 	default:
 		assert(0);
--- origsrc/knot-3.4.6/src/knot/server/udp-handler.c	2025-04-10 13:47:25.000000000 +0900
+++ src/knot-3.4.6/src/knot/server/udp-handler.c	2025-04-13 15:14:08.764794800 +0900
@@ -117,7 +117,7 @@ static void cmsg_handle_ecn(int **p_ecn,
 static void cmsg_handle_pktinfo(sockaddr_t *local, const iface_t *iface,
                                 struct cmsghdr *cmsg)
 {
-#if defined(IP_PKTINFO)
+#if defined(IP_PKTINFO) && !defined(__CYGWIN__)
 	if (cmsg->cmsg_level == IPPROTO_IP && cmsg->cmsg_type == IP_PKTINFO) {
 		struct in_pktinfo *info = (struct in_pktinfo *)CMSG_DATA(cmsg);
 		info->ipi_spec_dst = info->ipi_addr;
--- origsrc/knot-3.4.6/tests/Makefile.am	2025-04-10 13:47:25.000000000 +0900
+++ src/knot-3.4.6/tests/Makefile.am	2025-04-13 15:14:08.764794800 +0900
@@ -230,7 +230,7 @@ knot/test_semantic_check:
 	@chmod +x $(top_builddir)/tests/$@
 endif HAVE_LIBUTILS
 
-libzscanner/test_zscanner: libzscanner/zscanner-tool
+libzscanner/test_zscanner: libzscanner/zscanner-tool$(EXEEXT)
 	@$(edit) < $(top_srcdir)/tests/$@.in > $(top_builddir)/tests/$@
 	@chmod +x $(top_builddir)/tests/$@
 
