--- origsrc/libfdata-20240415/Makefile.am	2024-04-15 13:35:53.000000000 +0900
+++ src/libfdata-20240415/Makefile.am	2025-05-17 09:19:04.840986800 +0900
@@ -3,13 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libcnotify \
-	libfcache \
 	libfdata \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libfdata-20240415/configure.ac	2024-04-15 13:35:53.000000000 +0900
+++ src/libfdata-20240415/configure.ac	2025-05-17 09:19:04.840986800 +0900
@@ -109,11 +109,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
 AC_CONFIG_FILES([libfdata/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
--- origsrc/libfdata-20240415/libfdata/libfdata_list_element.h	2024-04-15 13:35:53.000000000 +0900
+++ src/libfdata-20240415/libfdata/libfdata_list_element.h	2025-05-17 09:19:04.840986800 +0900
@@ -64,16 +64,19 @@ struct libfdata_internal_list_element
 	uint8_t flags;
 };
 
+LIBFDATA_EXTERN \
 int libfdata_list_element_initialize(
      libfdata_list_element_t **element,
      libfdata_list_t *list,
      int element_index,
      libcerror_error_t **error );
 
+LIBFDATA_EXTERN \
 int libfdata_list_element_free(
      libfdata_list_element_t **element,
      libcerror_error_t **error );
 
+LIBFDATA_EXTERN \
 int libfdata_list_element_clone(
      libfdata_list_element_t **destination_element,
      libfdata_list_element_t *source_element,
@@ -85,11 +88,13 @@ int libfdata_list_element_get_element_in
      int *element_index,
      libcerror_error_t **error );
 
+LIBFDATA_EXTERN \
 int libfdata_list_element_set_element_index(
      libfdata_list_element_t *element,
      int element_index,
      libcerror_error_t **error );
 
+LIBFDATA_EXTERN \
 int libfdata_list_element_get_timestamp(
      libfdata_list_element_t *element,
      int64_t *timestamp,
@@ -106,6 +111,7 @@ int libfdata_list_element_get_data_range
      uint32_t *flags,
      libcerror_error_t **error );
 
+LIBFDATA_EXTERN \
 int libfdata_list_element_set_data_range(
      libfdata_list_element_t *element,
      int file_index,
