--- origsrc/libfdata-20190811/Makefile.am	2019-08-11 17:54:33.000000000 +0900
+++ src/libfdata-20190811/Makefile.am	2019-10-01 11:37:11.765722300 +0900
@@ -3,13 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libcnotify \
-	libfcache \
 	libfdata \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libfdata-20190811/configure.ac	2019-08-11 17:54:33.000000000 +0900
+++ src/libfdata-20190811/configure.ac	2019-10-01 11:37:11.781346500 +0900
@@ -130,11 +130,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
 AC_CONFIG_FILES([libfdata/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
--- origsrc/libfdata-20190811/libfdata/Makefile.am	2019-08-11 17:54:33.000000000 +0900
+++ src/libfdata-20190811/libfdata/Makefile.am	2019-10-01 11:37:11.812600300 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libfdata-20190811/libfdata/libfdata_list_element.h	2019-08-11 17:54:33.000000000 +0900
+++ src/libfdata-20190811/libfdata/libfdata_list_element.h	2019-10-01 11:37:11.828231900 +0900
@@ -64,31 +64,37 @@ struct libfdata_internal_list_element
 	uint8_t flags;
 };
 
+LIBFDATA_EXTERN \
 int libfdata_list_element_initialize(
      libfdata_list_element_t **element,
      libfdata_list_t *list,
      int element_index,
      libcerror_error_t **error );
 
+LIBFDATA_EXTERN \
 int libfdata_list_element_free(
      libfdata_list_element_t **element,
      libcerror_error_t **error );
 
+LIBFDATA_EXTERN \
 int libfdata_list_element_clone(
      libfdata_list_element_t **destination_element,
      libfdata_list_element_t *source_element,
      libcerror_error_t **error );
 
+LIBFDATA_EXTERN \
 int libfdata_list_element_get_element_index(
      libfdata_list_element_t *element,
      int *element_index,
      libcerror_error_t **error );
 
+LIBFDATA_EXTERN \
 int libfdata_list_element_set_element_index(
      libfdata_list_element_t *element,
      int element_index,
      libcerror_error_t **error );
 
+LIBFDATA_EXTERN \
 int libfdata_list_element_get_timestamp(
      libfdata_list_element_t *element,
      int64_t *timestamp,
@@ -96,6 +102,7 @@ int libfdata_list_element_get_timestamp(
 
 /* Data range functions
  */
+LIBFDATA_EXTERN \
 int libfdata_list_element_get_data_range(
      libfdata_list_element_t *element,
      int *file_index,
@@ -104,6 +111,7 @@ int libfdata_list_element_get_data_range
      uint32_t *flags,
      libcerror_error_t **error );
 
+LIBFDATA_EXTERN \
 int libfdata_list_element_set_data_range(
      libfdata_list_element_t *element,
      int file_index,
--- origsrc/libfdata-20190811/tests/Makefile.am	2019-08-11 17:54:33.000000000 +0900
+++ src/libfdata-20190811/tests/Makefile.am	2019-10-01 11:37:11.859501100 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
