--- origsrc/grt-0.2.4/GRT/Util/GRTVersionInfo.h	2016-12-05 08:58:01.000000000 +0900
+++ src/grt-0.2.4/GRT/Util/GRTVersionInfo.h	2026-05-21 12:51:53.648853100 +0900
@@ -36,7 +36,7 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN
     #define __GRT_OSX_BUILD__
 #endif
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__CYGWIN__)
     #define __GRT_LINUX_BUILD__
 #endif
 
--- origsrc/grt-0.2.4/build/CMakeLists.txt	2016-12-05 08:58:01.000000000 +0900
+++ src/grt-0.2.4/build/CMakeLists.txt	2026-05-21 18:02:58.959357000 +0900
@@ -54,7 +54,9 @@ else()
 if(UNIX) 
     message(STATUS "Building " ${LIB_BUILD_TYPE_INFO}  " library for UNIX/Linux")
     configure_file(grt.pc.in ${CMAKE_CURRENT_BINARY_DIR}/grt.pc @ONLY)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread -std=c++11")
+    if (NOT CYGWIN)
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread -std=c++11")
+    endif()
     set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Wall")
 endif()
 
@@ -129,6 +131,10 @@ add_library(${GRT_LIB_NAME}
             ${GRT_HEADERS}
 )
 
+if (DEFINED SOVERSION)
+    set_target_properties(${GRT_LIB_NAME} PROPERTIES SOVERSION ${SOVERSION})
+endif()
+
 #Add the options for what should be installed (when the user runs: sudo make install)
 if(NOT EXCLUDE_FROM_INSTALL)
 	install(
@@ -190,8 +196,10 @@ if( BUILD_TESTS )
         #Include the main GRT include directory
         include_directories( "${GRT_SRC_DIR}" )
 
-        #Include the gtest parent directory
-        include_directories( "${GRT_THIRD_PARTY_DIR}/gtest/googletest/include" )
+        if (BUILD_GTEST)
+            #Include the gtest parent directory
+            include_directories( "${GRT_THIRD_PARTY_DIR}/gtest/googletest/include" )
+        endif()
 
         #Link against the main GRT library 
         target_link_libraries(${unit_test_name} ${GRT_LIB_NAME} gtest)
