--- origsrc/curlcpp-2.0/CMakeLists.txt	2022-10-10 02:30:01.000000000 +0900
+++ src/curlcpp-2.0/CMakeLists.txt	2022-10-17 14:45:48.837059100 +0900
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 3.10...3.
 
 # Setting up project
 project(curlcpp
-	VERSION 1.5
+	VERSION 2.0
 	DESCRIPTION "An object oriented C++ wrapper for CURL (libcurl)"
 	HOMEPAGE_URL "https://github.com/JosephP91/curlcpp/"
 	LANGUAGES CXX)
--- origsrc/curlcpp-2.0/src/CMakeLists.txt	2022-10-10 02:30:01.000000000 +0900
+++ src/curlcpp-2.0/src/CMakeLists.txt	2022-10-17 14:53:48.308043500 +0900
@@ -83,6 +83,10 @@ if(IPO_SUPPORTED AND CMAKE_BUILD_TYPE ST
     set_target_properties(curlcpp PROPERTIES INTERPROCEDURAL_OPTIMIZATION TRUE)
 endif()
 
+if(DEFINED SOVERSION)
+    set_target_properties(curlcpp PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 set_target_properties (curlcpp PROPERTIES PUBLIC_HEADER "${CURLCPP_HEADER_LIST}")
 
 include(GNUInstallDirs)
@@ -90,4 +94,4 @@ install(TARGETS curlcpp
     EXPORT ${PROJECT_NAME}Targets
     PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}/
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
--- origsrc/curlcpp-2.0/test/cookie.cpp	2022-10-10 02:30:01.000000000 +0900
+++ src/curlcpp-2.0/test/cookie.cpp	2022-10-17 14:49:14.644459100 +0900
@@ -1,10 +1,10 @@
 #include <iostream>
 
-#include "curlcpp/curl_easy.h"
-#include "curlcpp/curl_exception.h"
-#include "curlcpp/curl_cookie.h"
-#include "curlcpp/curl_ios.h"
-#include "curlcpp/cookie_datetime.h"
+#include "curl_easy.h"
+#include "curl_exception.h"
+#include "curl_cookie.h"
+#include "curl_ios.h"
+#include "cookie_datetime.h"
 
 using std::ostringstream;
 
--- origsrc/curlcpp-2.0/test/custom_request.cpp	2022-10-10 02:30:01.000000000 +0900
+++ src/curlcpp-2.0/test/custom_request.cpp	2022-10-17 14:49:45.325101500 +0900
@@ -1,6 +1,6 @@
-#include "curlcpp/curl_easy.h"
-#include "curlcpp/curl_form.h"
-#include "curlcpp/curl_header.h"
+#include "curl_easy.h"
+#include "curl_form.h"
+#include "curl_header.h"
 
 using curl::curl_easy;
 using curl::curl_easy_exception;
--- origsrc/curlcpp-2.0/test/easy.cpp	2022-10-10 02:30:01.000000000 +0900
+++ src/curlcpp-2.0/test/easy.cpp	2022-10-17 14:50:52.470241700 +0900
@@ -1,4 +1,4 @@
-#include "curlcpp/curl_easy.h"
+#include "curl_easy.h"
 
 using curl::curl_easy;
 using curl::curl_easy_exception;
--- origsrc/curlcpp-2.0/test/easy_info.cpp	2022-10-10 02:30:01.000000000 +0900
+++ src/curlcpp-2.0/test/easy_info.cpp	2022-10-17 14:52:48.419754400 +0900
@@ -1,6 +1,6 @@
-#include "curlcpp/curl_easy.h"
-#include "curlcpp/curl_ios.h"
-#include "curlcpp/curl_exception.h"
+#include "curl_easy.h"
+#include "curl_ios.h"
+#include "curl_exception.h"
 
 using std::ostringstream;
 
--- origsrc/curlcpp-2.0/test/header.cpp	2022-10-10 02:30:01.000000000 +0900
+++ src/curlcpp-2.0/test/header.cpp	2022-10-17 14:53:01.666121200 +0900
@@ -1,6 +1,6 @@
-#include "curlcpp/curl_easy.h"
-#include "curlcpp/curl_form.h"
-#include "curlcpp/curl_header.h"
+#include "curl_easy.h"
+#include "curl_form.h"
+#include "curl_header.h"
 
 using curl::curl_easy;
 using curl::curl_easy_exception;
--- origsrc/curlcpp-2.0/test/multi.cpp	2022-10-10 02:30:01.000000000 +0900
+++ src/curlcpp-2.0/test/multi.cpp	2022-10-17 14:53:13.615354300 +0900
@@ -1,9 +1,9 @@
 #include <iostream>
 #include <ostream>
 
-#include "curlcpp/curl_easy.h"
-#include "curlcpp/curl_multi.h"
-#include "curlcpp/curl_ios.h"
+#include "curl_easy.h"
+#include "curl_multi.h"
+#include "curl_ios.h"
 
 using curl::curl_easy;
 using curl::curl_multi;
--- origsrc/curlcpp-2.0/test/output_variable.cpp	2022-10-10 02:30:01.000000000 +0900
+++ src/curlcpp-2.0/test/output_variable.cpp	2022-10-17 14:53:24.758603500 +0900
@@ -1,10 +1,10 @@
 #include <iostream>
 #include <ostream>
 
-#include "curlcpp/curl_easy.h"
-#include "curlcpp/curl_form.h"
-#include "curlcpp/curl_ios.h"
-#include "curlcpp/curl_exception.h"
+#include "curl_easy.h"
+#include "curl_form.h"
+#include "curl_ios.h"
+#include "curl_exception.h"
 
 using std::cout;
 using std::endl;
--- origsrc/curlcpp-2.0/test/recv_header.cpp	2022-10-10 02:30:01.000000000 +0900
+++ src/curlcpp-2.0/test/recv_header.cpp	2022-10-17 14:53:36.322724700 +0900
@@ -1,5 +1,5 @@
-#include "curlcpp/curl_easy.h"
-#include "curlcpp/curl_form.h"
+#include "curl_easy.h"
+#include "curl_form.h"
 
 using curl::curl_easy;
 using curl::curl_ios;
