--- origsrc/curlcpp-3.1/src/CMakeLists.txt	2023-04-18 04:48:59.000000000 +0900
+++ src/curlcpp-3.1/src/CMakeLists.txt	2023-07-27 23:39:18.800354700 +0900
@@ -104,6 +104,10 @@ if(IPO_SUPPORTED AND CMAKE_BUILD_TYPE ST
     set_target_properties(curlcpp PROPERTIES INTERPROCEDURAL_OPTIMIZATION TRUE)
 endif()
 
+if(DEFINED SOVERSION)
+    set_target_properties(curlcpp PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 set_target_properties (curlcpp PROPERTIES PUBLIC_HEADER "${CURLCPP_HEADER_LIST}")
 
 include(GNUInstallDirs)
@@ -111,4 +115,4 @@ install(TARGETS curlcpp
     EXPORT ${PROJECT_NAME}Targets
     PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}/
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
--- origsrc/curlcpp-3.1/test/cookie.cpp	2023-04-18 04:48:59.000000000 +0900
+++ src/curlcpp-3.1/test/cookie.cpp	2023-07-27 23:39:18.800354700 +0900
@@ -1,10 +1,10 @@
 #include <iostream>
 
-#include "curlcpp/curl_easy.h"
-#include "curlcpp/curl_exception.h"
-#include "curlcpp/curl_cookie.h"
-#include "curlcpp/curl_ios.h"
-#include "curlcpp/cookie_datetime.h"
+#include "curl_easy.h"
+#include "curl_exception.h"
+#include "curl_cookie.h"
+#include "curl_ios.h"
+#include "cookie_datetime.h"
 
 using std::ostringstream;
 
--- origsrc/curlcpp-3.1/test/custom_request.cpp	2023-04-18 04:48:59.000000000 +0900
+++ src/curlcpp-3.1/test/custom_request.cpp	2023-07-27 23:39:18.800354700 +0900
@@ -1,6 +1,6 @@
-#include "curlcpp/curl_easy.h"
-#include "curlcpp/curl_form.h"
-#include "curlcpp/curl_header.h"
+#include "curl_easy.h"
+#include "curl_form.h"
+#include "curl_header.h"
 
 using curl::curl_easy;
 using curl::curl_easy_exception;
--- origsrc/curlcpp-3.1/test/easy.cpp	2023-04-18 04:48:59.000000000 +0900
+++ src/curlcpp-3.1/test/easy.cpp	2023-07-27 23:39:18.800354700 +0900
@@ -1,4 +1,4 @@
-#include "curlcpp/curl_easy.h"
+#include "curl_easy.h"
 
 using curl::curl_easy;
 using curl::curl_easy_exception;
--- origsrc/curlcpp-3.1/test/easy_info.cpp	2023-04-18 04:48:59.000000000 +0900
+++ src/curlcpp-3.1/test/easy_info.cpp	2023-07-27 23:39:18.800354700 +0900
@@ -1,6 +1,6 @@
-#include "curlcpp/curl_easy.h"
-#include "curlcpp/curl_ios.h"
-#include "curlcpp/curl_exception.h"
+#include "curl_easy.h"
+#include "curl_ios.h"
+#include "curl_exception.h"
 
 using std::ostringstream;
 
--- origsrc/curlcpp-3.1/test/header.cpp	2023-04-18 04:48:59.000000000 +0900
+++ src/curlcpp-3.1/test/header.cpp	2023-07-27 23:39:18.800354700 +0900
@@ -1,6 +1,6 @@
-#include "curlcpp/curl_easy.h"
-#include "curlcpp/curl_form.h"
-#include "curlcpp/curl_header.h"
+#include "curl_easy.h"
+#include "curl_form.h"
+#include "curl_header.h"
 
 using curl::curl_easy;
 using curl::curl_easy_exception;
--- origsrc/curlcpp-3.1/test/multi.cpp	2023-04-18 04:48:59.000000000 +0900
+++ src/curlcpp-3.1/test/multi.cpp	2023-07-27 23:39:18.800354700 +0900
@@ -1,9 +1,9 @@
 #include <iostream>
 #include <ostream>
 
-#include "curlcpp/curl_easy.h"
-#include "curlcpp/curl_multi.h"
-#include "curlcpp/curl_ios.h"
+#include "curl_easy.h"
+#include "curl_multi.h"
+#include "curl_ios.h"
 
 using curl::curl_easy;
 using curl::curl_multi;
--- origsrc/curlcpp-3.1/test/output_variable.cpp	2023-04-18 04:48:59.000000000 +0900
+++ src/curlcpp-3.1/test/output_variable.cpp	2023-07-27 23:39:18.800354700 +0900
@@ -1,10 +1,10 @@
 #include <iostream>
 #include <ostream>
 
-#include "curlcpp/curl_easy.h"
-#include "curlcpp/curl_form.h"
-#include "curlcpp/curl_ios.h"
-#include "curlcpp/curl_exception.h"
+#include "curl_easy.h"
+#include "curl_form.h"
+#include "curl_ios.h"
+#include "curl_exception.h"
 
 using std::cout;
 using std::endl;
--- origsrc/curlcpp-3.1/test/recv_header.cpp	2023-04-18 04:48:59.000000000 +0900
+++ src/curlcpp-3.1/test/recv_header.cpp	2023-07-27 23:39:18.800354700 +0900
@@ -1,5 +1,5 @@
-#include "curlcpp/curl_easy.h"
-#include "curlcpp/curl_form.h"
+#include "curl_easy.h"
+#include "curl_form.h"
 
 using curl::curl_easy;
 using curl::curl_ios;
