--- origsrc/lzfse-lzfse-1.0/CMakeLists.txt	2017-03-09 07:48:59.000000000 +0900
+++ src/lzfse-lzfse-1.0/CMakeLists.txt	2019-06-27 10:30:23.223163400 +0900
@@ -91,6 +91,10 @@ if(BUILD_SHARED_LIBS)
   set_property(TARGET lzfse APPEND PROPERTY INTERFACE_COMPILE_DEFINITIONS LZFSE_DLL)
 endif()
 
+if(CYGWIN)
+  set_target_properties(lzfse PROPERTIES SOVERSION 0)
+endif()
+
 # Installation
 if(NOT LZFSE_BUNDLE_MODE)
   include(GNUInstallDirs)
@@ -135,3 +139,10 @@ if(NOT LZFSE_DISABLE_TESTS)
         -P ${CMAKE_CURRENT_SOURCE_DIR}/tests/round-trip.cmake)
   endforeach()
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/lzfse.pc.in ${CMAKE_BINARY_DIR}/lzfse.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/lzfse.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/lzfse-lzfse-1.0/lzfse.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/lzfse-lzfse-1.0/lzfse.pc.in	2019-06-27 10:30:23.079737300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: lzfse
+Description: Compression library and command line tool
+Version: @VERSION@
+Libs: -L${libdir} -llzfse
+Cflags: -I${includedir}
