--- origsrc/apt-cacher-ng-0.8.2/include/sockio.h	2015-04-29 18:32:43.000000000 +0900
+++ src/apt-cacher-ng-0.8.2/include/sockio.h	2015-05-02 23:32:19.524257800 +0900
@@ -14,6 +14,7 @@
 #include <netdb.h>
 #include <unistd.h>
 #include <cstddef>
+#include <sys/select.h>
 
 using namespace std;
 
--- origsrc/apt-cacher-ng-0.8.2/source/acbuf.cc	2015-04-29 18:32:43.000000000 +0900
+++ src/apt-cacher-ng-0.8.2/source/acbuf.cc	2015-05-02 23:32:24.895596700 +0900
@@ -7,6 +7,7 @@
 #include "acbuf.h"
 #include "fileio.h"
 #include <unistd.h>
+#include <sys/select.h>
 
 bool acbuf::setsize(unsigned int c) {
 	if(m_nCapacity==c)
