--- origsrc/quvi-0.2.10/configure.ac	2010-12-10 16:06:38.000000000 +0900
+++ src/quvi-0.2.10/configure.ac	2010-12-12 00:32:43.999645900 +0900
@@ -1,7 +1,7 @@
 #                                               -*- Autoconf -*-
 # Process this file with autoconf to produce a configure script.
 
-AC_PREREQ([2.68])
+AC_PREREQ([2.67])
 
 m4_include([m4/version.m4])
 
@@ -32,18 +32,23 @@ AC_DEFINE_UNQUOTED([CANONICAL_HOST], "$h
 
 AM_INIT_AUTOMAKE([-Wall -Werror foreign dist-bzip2])
 
-LT_INIT()
+LT_INIT(win32-dll)
 LT_PREREQ([2.2])
 
 # Host specific.
 NO_UNDEFINED=
 case "$host" in
-    *-*-mingw32* | *-*-cygwin*)
+    *-*-mingw32*)
         AC_DEFINE([HOST_W32], [1], [...])
         host_w32=yes
         if [ test x"$enable_shared" != "xno" ]; then
             NO_UNDEFINED="-no-undefined"
         fi
+    ;;
+    *-*-cygwin*)
+        if [ test x"$enable_shared" != "xno" ]; then
+            NO_UNDEFINED="-no-undefined"
+        fi
     ;;
 esac
 EXTRA_LIBQUVI_LDFLAGS="$NO_UNDEFINED"
