--- origsrc/resid-0.16/Makefile.am	2005-01-16 18:54:10.000000000 +0900
+++ src/resid-0.16/Makefile.am	2015-04-25 14:37:45.078296200 +0900
@@ -20,6 +20,7 @@ endif
 CLEANFILES = libresid.la libresidc.la
 
 libresid_la_SOURCES = $(resid_sources)
+libresid_la_LDFLAGS = -no-undefined
 libresidc_la_SOURCES = $(resid_sources)
 
 BUILT_SOURCES = $(noinst_DATA:.dat=.cc)
@@ -32,5 +33,8 @@ EXTRA_DIST = $(noinst_DATA) $(noinst_SCR
 
 SUFFIXES = .dat
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
+
 .dat.cc:
 	$(PERL) $(srcdir)/samp2src.pl $* $< $(srcdir)/$@
--- origsrc/resid-0.16/configure.in	2005-01-16 18:54:10.000000000 +0900
+++ src/resid-0.16/configure.in	2015-04-25 14:38:14.418921800 +0900
@@ -1,8 +1,8 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(sid.h)
+AC_INIT([resid], [0.16])
 
 dnl Use Automake
-AM_INIT_AUTOMAKE(resid, 0.16)
+AM_INIT_AUTOMAKE
 LTVERSION=5:0:0
 
 dnl Use C++ for tests.
@@ -100,4 +100,4 @@ AC_ARG_ENABLE(resid-install,
 esac],[install=false])
 AM_CONDITIONAL(INSTALL_RESID, test x$install = xtrue)
 
-AC_OUTPUT(Makefile siddefs.h)
+AC_OUTPUT(resid.pc Makefile siddefs.h)
--- origsrc/resid-0.16/resid.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/resid-0.16/resid.pc.in	2015-04-25 14:37:31.622858500 +0900
@@ -0,0 +1,5 @@
+Name: resid
+Description: C++ library to emulate the C64 SID chip
+Version: @VERSION@
+Libs: -L@libdir@ -lresid
+Cflags: -I@includedir@
