--- origsrc/cclive/configure.ac	2010-10-31 12:18:49.000000000 +0900
+++ src/cclive/configure.ac	2010-10-31 12:21:24.425028500 +0900
@@ -1,7 +1,7 @@
 #                                               -*- Autoconf -*-
 # Process this file with autoconf to produce a configure script.
 
-AC_PREREQ([2.68])
+AC_PREREQ([2.67])
 
 m4_include([m4/version.m4])
 AC_INIT([cclive], [_CCLIVE_VERSION], [legatvs@gmail.com])
