--- origsrc/mfem-4.2/CMakeLists.txt	2020-10-31 07:30:58.000000000 +0900
+++ src/mfem-4.2/CMakeLists.txt	2020-11-24 05:19:41.336819400 +0900
@@ -575,7 +575,8 @@ set(CMAKE_INSTALL_DEFAULT_COMPONENT_NAME
 # Install the library
 install(TARGETS ${PROJECT_NAME}
   EXPORT ${PROJECT_NAME_UC}Targets
-  DESTINATION ${INSTALL_LIB_DIR})
+  RUNTIME DESTINATION bin
+  ARCHIVE DESTINATION ${INSTALL_LIB_DIR})
 
 # Install the master headers
 foreach(Header mfem.hpp mfem-performance.hpp)
@@ -675,3 +676,10 @@ install(EXPORT ${PROJECT_NAME_UC}Targets
 #-------------------------------------------------------------------------------
 
 mfem_export_mk_files()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/mfem.pc.in ${CMAKE_BINARY_DIR}/mfem.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/mfem.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/mfem-4.2/fem/coefficient.hpp	2020-10-31 07:30:58.000000000 +0900
+++ src/mfem-4.2/fem/coefficient.hpp	2020-11-23 23:19:41.675893900 +0900
@@ -19,6 +19,10 @@
 #include "intrules.hpp"
 #include "eltrans.hpp"
 
+#if defined(_B)
+#undef _B
+#endif
+
 namespace mfem
 {
 
--- origsrc/mfem-4.2/mfem.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/mfem-4.2/mfem.pc.in	2020-11-24 05:20:39.357292100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mfem
+Description: Lightweight, general, scalable C++ library for finite element methods
+Version: @mfem_VERSION@
+Libs: -L${libdir} -lmfem
+Cflags: -I${includedir}
