--- origsrc/libheinz-v2.0.0/CMakeLists.txt	2023-06-14 21:38:11.000000000 +0900
+++ src/libheinz-v2.0.0/CMakeLists.txt	2023-09-09 10:38:46.703264700 +0900
@@ -66,4 +66,4 @@ write_basic_package_version_file(
 install(FILES
     "${PROJECT_BINARY_DIR}/LibHeinzConfig.cmake"
     "${PROJECT_BINARY_DIR}/LibHeinzConfigVersion.cmake"
-    DESTINATION cmake COMPONENT dev)
+    DESTINATION lib/cmake/libheinz COMPONENT dev)
