--- origsrc/xmlstarlet-1.0.5/Makefile.am	2011-02-12 06:21:22.000000000 +0900
+++ src/xmlstarlet-1.0.5/Makefile.am	2011-02-13 21:29:11.236962000 +0900
@@ -52,10 +52,12 @@ clean-local:
 include examples/tests.mk
 
 # building executable
-bin_PROGRAMS = xml
+bin_PROGRAMS = xmlstarlet
+
+xmlstarlet_CFLAGS = @LIBXML_CFLAGS@ @LIBXSLT_CFLAGS@ @LIBEXSLT_CFLAGS@
 
 include src/sources.mk
-xml_SOURCES += version.h
+xmlstarlet_SOURCES += version.h
 
 # doc
 man_MANS = doc/xmlstarlet.1
--- origsrc/xmlstarlet-1.0.5/configure.ac	2011-02-12 06:42:41.000000000 +0900
+++ src/xmlstarlet-1.0.5/configure.ac	2011-02-13 20:58:13.703347900 +0900
@@ -7,7 +7,6 @@ AC_REVISION(AC_PACKAGE_VERSION)
 
 AM_INIT_AUTOMAKE([1.11 foreign -Wall subdir-objects no-define std-options color-tests])
 AM_SILENT_RULES
-AM_MAINTAINER_MODE
 
 [LIBXML_REQUIRED_VERSION=2.6.23]
 [LIBXSLT_REQUIRED_VERSION=1.1.9]
@@ -34,33 +33,9 @@ AM_PROG_CC_C_O
 AC_ARG_PROGRAM          dnl Transforming Program Names When Installing
 AC_PROG_SED
 
-
-XSTAR_LIB_CHECK([LIBXML], [xml2-config])
-
-AS_IF([test "x$LIBXML_SRCDIR" != x],
-      [LIBXML_INCDIR="$LIBXML_SRCDIR/include"])
-[: ${LIBXML_INCDIR="$LIBXML_PREFIX/include/libxml2"}]
-
-AS_IF([test "x$STATIC_LIBS" != xno],
-      [LIBXML_CPPFLAGS="-I$LIBXML_INCDIR -DLIBXML_STATIC"],
-      [LIBXML_CPPFLAGS="-I$LIBXML_INCDIR"])
-
-AS_IF([test "x$STATIC_LIBS" != xno],
-      [LIBXML_LIBS="$LIBXML_LIBDIR/libxml2.a"],
-      [LIBXML_LIBS="-lxml2"])
-[LIBXML_LDFLAGS="-L$LIBXML_LIBDIR"]
-
-
-XSTAR_LIB_CHECK([LIBXSLT], [xslt-config])
-
-AS_IF([test "x$LIBXSLT_SRCDIR" != x],
-      [LIBXSLT_INCDIR="$LIBXSLT_SRCDIR"
-       LIBXSLT_LDFLAGS="-L$LIBXSLT_SRCDIR/libexslt/.libs -L$LIBXSLT_SRCDIR/libxslt/.libs"
-       AS_IF([test "x$STATIC_LIBS" != xno],
-             [LIBXSLT_LIBS="$LIBXSLT_SRCDIR/libexslt/.libs/libexslt.a $LIBXSLT_SRCDIR/libxslt/.libs/libxslt.a"])],
-      [LIBXSLT_LDFLAGS="-L$LIBXSLT_LIBDIR"
-      AS_IF([test "x$STATIC_LIBS" != xno],
-             [LIBXSLT_LIBS="$LIBXSLT_LIBDIR/libexslt.a $LIBXSLT_LIBDIR/libxslt.a"])])
+PKG_CHECK_MODULES([LIBXML], [libxml-2.0])
+PKG_CHECK_MODULES([LIBXSLT], [libxslt])
+PKG_CHECK_MODULES([LIBEXSLT], [libexslt])
 
 AS_IF([test "x$STATIC_LIBS" = xno],
       [LIBXSLT_LIBS="-lexslt -lxslt"])
--- origsrc/xmlstarlet-1.0.5/examples/tests.mk	2011-02-11 13:11:39.000000000 +0900
+++ src/xmlstarlet-1.0.5/examples/tests.mk	2011-02-13 21:44:10.820499600 +0900
@@ -1,4 +1,4 @@
-TESTS_ENVIRONMENT = srcdir=$(srcdir) $(srcdir)/tests/runTest.sh
+TESTS_ENVIRONMENT = srcdir=$(builddir) $(builddir)/tests/runTest.sh
 
 TESTS = examples/bigxml-dtd\
 examples/bigxml-embed-ref\
--- origsrc/xmlstarlet-1.0.5/examples/xmlstarlet	2011-02-12 06:21:22.000000000 +0900
+++ src/xmlstarlet-1.0.5/examples/xmlstarlet	2011-02-13 20:59:05.927858800 +0900
@@ -1,6 +1,6 @@
 #!/bin/sh
 exepath=`dirname $0`
-xml="${srcdir-$exepath/..}"/xml
+xml="${srcdir-$exepath/..}"/xmlstarlet
 
 if [ x"$OSTYPE" = xmsys ] ; then
     # MSYS does unix -> windows path conversion, eg: /x -> C:\Mingw\msys\1.0\x
--- origsrc/xmlstarlet-1.0.5/src/sources.mk	2011-02-12 01:41:32.000000000 +0900
+++ src/xmlstarlet-1.0.5/src/sources.mk	2011-02-13 21:00:09.248249400 +0900
@@ -1,4 +1,4 @@
-xml_SOURCES =\
+xmlstarlet_SOURCES =\
 src/binsert.c\
 src/binsert.h\
 src/escape.h\
