--- origsrc/xmlstarlet-1.0.6/Makefile.am	2011-03-08 09:48:17.000000000 +0900
+++ src/xmlstarlet-1.0.6/Makefile.am	2011-04-24 15:37:02.528238700 +0900
@@ -58,10 +58,12 @@ src/xml.o : version.h
 include examples/tests.mk
 
 # building executable
-bin_PROGRAMS = xml
+bin_PROGRAMS = xmlstarlet
+
+xmlstarlet_CFLAGS = @LIBXML_CFLAGS@ @LIBXSLT_CFLAGS@ @LIBEXSLT_CFLAGS@
 
 include src/sources.mk
-xml_SOURCES += version.h
+xmlstarlet_SOURCES += version.h
 
 # doc
 man_MANS = doc/xmlstarlet.1
--- origsrc/xmlstarlet-1.0.6/configure.ac	2011-03-14 04:23:31.000000000 +0900
+++ src/xmlstarlet-1.0.6/configure.ac	2011-04-24 15:27:32.503515600 +0900
@@ -34,33 +34,9 @@ AM_PROG_CC_C_O
 AC_ARG_PROGRAM          dnl Transforming Program Names When Installing
 AC_PROG_SED
 
-
-XSTAR_LIB_CHECK([LIBXML], [xml2-config])
-
-AS_IF([test "x$LIBXML_SRCDIR" != x],
-      [LIBXML_INCDIR="$LIBXML_SRCDIR/include"])
-[: ${LIBXML_INCDIR="$LIBXML_PREFIX/include/libxml2"}]
-
-AS_IF([test "x$STATIC_LIBS" != xno],
-      [LIBXML_CPPFLAGS="-I$LIBXML_INCDIR -DLIBXML_STATIC"],
-      [LIBXML_CPPFLAGS="-I$LIBXML_INCDIR"])
-
-AS_IF([test "x$STATIC_LIBS" != xno],
-      [LIBXML_LIBS="$LIBXML_LIBDIR/libxml2.a"],
-      [LIBXML_LIBS="-lxml2"])
-[LIBXML_LDFLAGS="-L$LIBXML_LIBDIR"]
-
-
-XSTAR_LIB_CHECK([LIBXSLT], [xslt-config])
-
-AS_IF([test "x$LIBXSLT_SRCDIR" != x],
-      [LIBXSLT_INCDIR="$LIBXSLT_SRCDIR"
-       LIBXSLT_LDFLAGS="-L$LIBXSLT_SRCDIR/libexslt/.libs -L$LIBXSLT_SRCDIR/libxslt/.libs"
-       AS_IF([test "x$STATIC_LIBS" != xno],
-             [LIBXSLT_LIBS="$LIBXSLT_SRCDIR/libexslt/.libs/libexslt.a $LIBXSLT_SRCDIR/libxslt/.libs/libxslt.a"])],
-      [LIBXSLT_LDFLAGS="-L$LIBXSLT_LIBDIR"
-      AS_IF([test "x$STATIC_LIBS" != xno],
-             [LIBXSLT_LIBS="$LIBXSLT_LIBDIR/libexslt.a $LIBXSLT_LIBDIR/libxslt.a"])])
+PKG_CHECK_MODULES([LIBXML], [libxml-2.0])
+PKG_CHECK_MODULES([LIBXSLT], [libxslt])
+PKG_CHECK_MODULES([LIBEXSLT], [libexslt])
 
 AS_IF([test "x$STATIC_LIBS" = xno],
       [LIBXSLT_LIBS="-lexslt -lxslt"])
--- origsrc/xmlstarlet-1.0.6/examples/xmlstarlet	2011-03-08 10:16:28.000000000 +0900
+++ src/xmlstarlet-1.0.6/examples/xmlstarlet	2011-04-24 15:29:04.702418500 +0900
@@ -1,6 +1,6 @@
 #!/bin/sh
 exepath=`dirname $0`
-xml="${abs_builddir-$exepath/..}"/xml
+xml="${abs_builddir-$exepath/..}"/xmlstarlet
 
 if [ x"$OSTYPE" = xmsys ] ; then
     # MSYS does unix -> windows path conversion, eg: /x -> C:\Mingw\msys\1.0\x
--- origsrc/xmlstarlet-1.0.6/src/sources.mk	2011-03-07 04:52:01.000000000 +0900
+++ src/xmlstarlet-1.0.6/src/sources.mk	2011-04-24 15:29:19.844267500 +0900
@@ -1,4 +1,4 @@
-xml_SOURCES =\
+xmlstarlet_SOURCES =\
 src/binsert.c\
 src/binsert.h\
 src/escape.h\
