--- origsrc/capnproto-1.2.0/c++/src/kj/exception.c++	2025-06-16 01:02:22.000000000 +0900
+++ src/capnproto-1.2.0/c++/src/kj/exception.c++	2025-07-27 16:27:05.657045800 +0900
@@ -254,7 +254,7 @@ ArrayPtr<void* const> getStackTrace(Arra
 #endif
 }
 
-#if (__GNUC__ && !_WIN32) || __clang__
+#if (__GNUC__ && !_WIN32 && !defined(__CYGWIN__)) || __clang__
 // Allow dependents to override the implementation of stack symbolication by making it a weak
 // symbol. We prefer weak symbols over some sort of callback registration mechanism becasue this
 // allows an alternate symbolication library to be easily linked into tests without changing the
