--- origsrc/capnproto-1.3.0/c++/src/kj/exception.c++	2025-12-17 08:54:34.000000000 +0900
+++ src/capnproto-1.3.0/c++/src/kj/exception.c++	2026-01-01 14:24:13.592149500 +0900
@@ -259,7 +259,7 @@ ArrayPtr<void* const> getStackTrace(Arra
 #endif
 }
 
-#if (__GNUC__ && !_WIN32) || __clang__
+#if (__GNUC__ && !_WIN32 && !defined(__CYGWIN__)) || __clang__
 // Allow dependents to override the implementation of stack symbolication by making it a weak
 // symbol. We prefer weak symbols over some sort of callback registration mechanism becasue this
 // allows an alternate symbolication library to be easily linked into tests without changing the
