--- origsrc/dmg2img-1.6.5/Makefile	2010-03-25 04:38:08.000000000 +0900
+++ src/dmg2img-1.6.5/Makefile	2015-05-08 23:20:38.793263300 +0900
@@ -1,5 +1,5 @@
 #CC = gcc
-CFLAGS = -O2 -Wall
+CFLAGS = -O3 -Wall
 BIN_DIR = ${DESTDIR}/usr/bin
 
 all: dmg2img vfdecrypt 
--- origsrc/dmg2img-1.6.5/vfdecrypt.c	2010-03-25 02:52:45.000000000 +0900
+++ src/dmg2img-1.6.5/vfdecrypt.c	2015-05-08 23:21:43.380574700 +0900
@@ -331,8 +331,6 @@ int main(int argc, char *argv[]) {
   int hdr_version, c, optError = 0;
   char inFile[512], outFile[512], passphrase[512], cmd[640];
   int iflag = 0, oflag = 0, pflag = 0, kflag = 0, verbose = 0;
-  extern char *optarg;
-  extern int optind, optopt;
   
   memset(inFile, 0, 512);
   memset(outFile, 0, 512);
