--- origsrc/exodus-6.09/exodus/CMakeLists.txt	2014-10-16 05:20:05.000000000 +0900
+++ src/exodus-6.09/exodus/CMakeLists.txt	2015-01-09 21:16:25.529331600 +0900
@@ -1,9 +1,9 @@
 cmake_minimum_required(VERSION 2.6)
 PROJECT(ExodusII C CXX Fortran)
 
-SET(EXODUSII_VERSION_MAJOR "5")
-SET(EXODUSII_VERSION_MINOR "14")
-SET(EXODUSII_VERSION_PATCH "0")
+SET(EXODUSII_VERSION_MAJOR "6")
+SET(EXODUSII_VERSION_MINOR "0")
+SET(EXODUSII_VERSION_PATCH "9")
 SET(EXODUSII_VERSION "${EXODUSII_VERSION_MAJOR}.${EXODUSII_VERSION_MINOR}")
 SET(EXODUSII_VERSION_FULL "${EXODUSII_VERSION}.${EXODUSII_VERSION_PATCH}")
 
@@ -22,7 +22,7 @@ find_path( NETCDF_INCLUDE_DIR netcdf.h
    $ENV{NETCDF_DIR}/inc
    $ENV{NETCDF_DIR}/include
    $ENV{NETCDF_DIR}/libsrc
-)	
+)
 
 find_library( NETCDF_LIBRARY
    NAMES netcdf
@@ -76,6 +76,10 @@ if (PYTHON)
 
   configure_file(${EXODUS_PY_IN} ${EXODUS_PY} @ONLY)
 
-  INSTALL(FILES ${EXODUS_PY} DESTINATION ${PYTHON_INSTALL})
+  INSTALL(FILES ${EXODUS_PY} DESTINATION /usr/lib/python/site-packages)
 
 endif()
+
+if (CYGWIN)
+  set(EXODUS_LIBRARY_PROPERTIES SOVERSION 0)
+endif()
\ No newline at end of file
--- origsrc/exodus-6.09/exodus/cbind/CMakeLists.txt	2014-10-16 05:20:05.000000000 +0900
+++ src/exodus-6.09/exodus/cbind/CMakeLists.txt	2015-01-09 20:45:56.680020800 +0900
@@ -293,7 +293,7 @@ ENDIF(EXODUS_LIBRARY_PROPERTIES)
 
 IF(NOT EXODUS_INSTALL_NO_LIBRARIES)
   INSTALL(TARGETS exodus
-    RUNTIME DESTINATION lib COMPONENT Runtime
+    RUNTIME DESTINATION bin COMPONENT Runtime
     LIBRARY DESTINATION lib COMPONENT Runtime
     ARCHIVE DESTINATION lib COMPONENT Development)
   INSTALL(FILES ${cbind_HEADERS} DESTINATION include COMPONENT Development)
--- origsrc/exodus-6.09/exodus/forbind/CMakeLists.txt	2014-10-16 05:20:05.000000000 +0900
+++ src/exodus-6.09/exodus/forbind/CMakeLists.txt	2015-01-09 20:46:20.740567700 +0900
@@ -1,4 +1,4 @@
-# Set Mangling by hand 
+# Set Mangling by hand
 # Assume _name mangling see src/exo_jack.c code for how ADDC_ will be used in
 # Fortran mangling
 add_definitions (-DADDC_)
@@ -9,7 +9,7 @@ find_path( NETCDF_INCLUDE_DIR netcdf.h
    $ENV{NETCDF_DIR}/inc
    $ENV{NETCDF_DIR}/include
    $ENV{NETCDF_DIR}/libsrc
-)	
+)
 
 find_library( NETCDF_LIBRARY
    NAMES netcdf
@@ -61,7 +61,7 @@ ENDIF(EXODUS_LIBRARY_PROPERTIES)
 
 IF(NOT EXODUS_INSTALL_NO_LIBRARIES)
   INSTALL(TARGETS exoIIv2for
-    RUNTIME DESTINATION lib COMPONENT Runtime
+    RUNTIME DESTINATION bin COMPONENT Runtime
     LIBRARY DESTINATION lib COMPONENT Runtime
     ARCHIVE DESTINATION lib COMPONENT Development)
   INSTALL(FILES ${forbind_HEADERS} DESTINATION include COMPONENT Development)
