diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x config-h.in origsrc/uisp-20050207/Makefile.am src/uisp-20050207/Makefile.am
--- origsrc/uisp-20050207/Makefile.am	2004-05-12 01:47:19.000000000 +0900
+++ src/uisp-20050207/Makefile.am	2016-03-19 14:39:09.816371800 +0900
@@ -49,15 +49,5 @@ install_aux_files    = AUTHORS COPYING I
 
 DOC_INST_DIR         = $(DESTDIR)$(datadir)/doc/uisp-$(VERSION)
 
-install-data-local:
-	$(mkinstalldirs) $(DOC_INST_DIR)
-	for file in $(install_aux_files) ; do \
-	  echo " $(INSTALL_DATA) $$file $(DOC_INST_DIR)/$$file"; \
-	  $(INSTALL_DATA) $(srcdir)/$$file $(DOC_INST_DIR)/$$file; \
-	done
-
-uninstall-local:
-	rm -rf $(DOC_INST_DIR)
-
 dist-hook: uisp.spec
 	cp uisp.spec $(distdir)/uisp.spec
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x config-h.in origsrc/uisp-20050207/src/cygwinp.C src/uisp-20050207/src/cygwinp.C
--- origsrc/uisp-20050207/src/cygwinp.C	2003-10-07 08:01:34.000000000 +0900
+++ src/uisp-20050207/src/cygwinp.C	2016-03-19 14:37:00.213389900 +0900
@@ -101,16 +101,4 @@ bool cygwinp_delay_usec(long t)
     }
 }
 
-
-int cfmakeraw(struct termios *termios_p)
-{
-    termios_p->c_iflag &=
-	~(IGNBRK|BRKINT|PARMRK|ISTRIP |INLCR|IGNCR|ICRNL|IXON);
-    termios_p->c_oflag &= ~OPOST;
-    termios_p->c_lflag &= ~(ECHO|ECHONL|ICANON|ISIG|IEXTEN);
-    termios_p->c_cflag &= ~(CSIZE|PARENB);
-    termios_p->c_cflag |= CS8;
-    return 0;
-}
-
 #endif
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x config-h.in origsrc/uisp-20050207/src/cygwinp.h src/uisp-20050207/src/cygwinp.h
--- origsrc/uisp-20050207/src/cygwinp.h	2002-05-26 02:59:46.000000000 +0900
+++ src/uisp-20050207/src/cygwinp.h	2016-03-19 14:37:03.150914100 +0900
@@ -29,5 +29,4 @@
 unsigned char inb(unsigned short port);
 void outb(unsigned char value, unsigned short port);
 int ioperm(unsigned short port, int num, int enable);
-int cfmakeraw(struct termios *termios_p);
 bool cygwinp_delay_usec(long t);
