--- origsrc/uisp-20050207/src/cygwinp.C	2003-10-07 08:01:34.000000000 +0900
+++ src/uisp-20050207/src/cygwinp.C	2025-08-20 13:31:40.273430400 +0900
@@ -101,16 +101,4 @@ bool cygwinp_delay_usec(long t)
     }
 }
 
-
-int cfmakeraw(struct termios *termios_p)
-{
-    termios_p->c_iflag &=
-	~(IGNBRK|BRKINT|PARMRK|ISTRIP |INLCR|IGNCR|ICRNL|IXON);
-    termios_p->c_oflag &= ~OPOST;
-    termios_p->c_lflag &= ~(ECHO|ECHONL|ICANON|ISIG|IEXTEN);
-    termios_p->c_cflag &= ~(CSIZE|PARENB);
-    termios_p->c_cflag |= CS8;
-    return 0;
-}
-
 #endif
--- origsrc/uisp-20050207/src/cygwinp.h	2002-05-26 02:59:46.000000000 +0900
+++ src/uisp-20050207/src/cygwinp.h	2025-08-20 13:31:40.275436500 +0900
@@ -29,5 +29,4 @@
 unsigned char inb(unsigned short port);
 void outb(unsigned char value, unsigned short port);
 int ioperm(unsigned short port, int num, int enable);
-int cfmakeraw(struct termios *termios_p);
 bool cygwinp_delay_usec(long t);
