--- origsrc/libfsntfs-20240501/Makefile.am	2024-05-01 17:05:06.000000000 +0900
+++ src/libfsntfs-20240501/Makefile.am	2025-05-18 23:45:10.494269700 +0900
@@ -3,27 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
-	libfguid \
-	libfwnt \
 	libfsntfs \
-	libfusn \
-	libhmac \
 	fsntfstools \
 	pyfsntfs \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libfsntfs-20240501/configure.ac	2024-05-01 17:05:06.000000000 +0900
+++ src/libfsntfs-20240501/configure.ac	2025-05-18 23:45:10.494269700 +0900
@@ -182,25 +182,8 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfwnt/Makefile])
 AC_CONFIG_FILES([libfsntfs/Makefile])
 AC_CONFIG_FILES([pyfsntfs/Makefile])
-AC_CONFIG_FILES([libfusn/Makefile])
-AC_CONFIG_FILES([libhmac/Makefile])
 AC_CONFIG_FILES([fsntfstools/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
--- origsrc/libfsntfs-20240501/pyfsntfs/Makefile.am	2024-05-01 17:05:06.000000000 +0900
+++ src/libfsntfs-20240501/pyfsntfs/Makefile.am	2025-05-18 23:45:10.494269700 +0900
@@ -65,7 +65,7 @@ pyfsntfs_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pyfsntfs_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyfsntfs_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyfsntfs_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
--- origsrc/libfsntfs-20240501/tests/Makefile.am	2024-05-01 17:05:06.000000000 +0900
+++ src/libfsntfs-20240501/tests/Makefile.am	2025-05-18 23:45:10.494269700 +0900
@@ -3,6 +3,7 @@ AUTOMAKE_OPTIONS = subdir-objects
 AM_CPPFLAGS = \
 	-I../include -I$(top_srcdir)/include \
 	-I../common -I$(top_srcdir)/common \
+	-I$(top_srcdir)/libfsntfs \
 	@LIBCERROR_CPPFLAGS@ \
 	@LIBCTHREADS_CPPFLAGS@ \
 	@LIBCDATA_CPPFLAGS@ \
