--- origsrc/re2c-2.2/Makefile.lib.am	2021-08-01 18:05:50.000000000 +0900
+++ src/re2c-2.2/Makefile.lib.am	2022-01-26 09:13:40.488865200 +0900
@@ -3,7 +3,7 @@
 lib_LTLIBRARIES += libre2c.la
 
 libre2c_la_CXXFLAGS = $(AM_CXXFLAGS)
-libre2c_la_LDFLAGS = -module -no-undefined
+libre2c_la_LDFLAGS = -no-undefined
 
 libre2c_la_HDR = \
 	lib/lex.h \
--- origsrc/re2c-2.2/benchmarks/submatch_dfa_aot/Makefile.am	2021-08-01 18:05:50.000000000 +0900
+++ src/re2c-2.2/benchmarks/submatch_dfa_aot/Makefile.am	2022-01-26 09:09:19.842968000 +0900
@@ -178,7 +178,7 @@ $(BIN_CLANG): $(BIN_DIR)/%-clang: $(GEN_
 
 $(DAT): $(DAT_DIR)/%/big: $(DAT_DIR)/%/small
 	$(AM_V_at)mkdir -p $(@D) && cp $(srcdir)/$(@D)/small $(@D)
-	$(AM_V_GEN)( cd $(@D) && ../../$(srcdir)/data/gen.py )
+	$(AM_V_GEN)( cd $(@D) && $(srcdir)/data/gen.py )
 
 $(KLEENEX):
 	$(AM_V_at)mkdir -p $(@D) && cp $(srcdir)/engines/kleenex/getkleenex.sh $(@D)
