--- origsrc/FLIF-0.4/src/CMakeLists.txt	2021-11-21 22:31:43.000000000 +0900
+++ src/FLIF-0.4/src/CMakeLists.txt	2024-10-22 19:15:50.642759000 +0900
@@ -243,18 +243,10 @@ if(GDK_FOUND AND (BUILD_SHARED_LIBS OR B
     target_link_libraries(pixbufloader-flif ${GDK_LIBRARIES})
     set_target_properties(pixbufloader-flif PROPERTIES OUTPUT_NAME pixbufloader-flif)
 
-    add_library(pixbufloader-flif-static STATIC flif-pixbuf-loader.c)
-    target_include_directories(pixbufloader-flif-static PRIVATE ${FLIF_SRC_DIR}/library ${GDK_INCLUDE_DIRS})
-    target_compile_options(pixbufloader-flif-static PRIVATE ${GDK_CFLAGS})
-    target_compile_definitions(pixbufloader-flif-static PRIVATE ${DEFINITIONS_FOR_ALL_TARGETS})
-    target_link_libraries(pixbufloader-flif-static ${GDK_LIBRARIES})
-    set_target_properties(pixbufloader-flif-static PROPERTIES OUTPUT_NAME pixbufloader-flif)
     if (BUILD_SHARED_LIBS)
        target_link_libraries(pixbufloader-flif flif_lib)
-       target_link_libraries(pixbufloader-flif-static flif_lib) 
     else()
        target_link_libraries(pixbufloader-flif flif_lib_static)
-       target_link_libraries(pixbufloader-flif-static flif_lib_static)
     endif(BUILD_SHARED_LIBS)
 #We have to be sure that the "-dont-define-prefix", otherwise, on mingw-w64, a complete path
 #is given with pkg-config and that's not desirable for some packaging systems.
@@ -266,10 +258,8 @@ if(GDK_FOUND AND (BUILD_SHARED_LIBS OR B
            message(FATAL_ERROR "Error: ${RES_VAR}")
     endif(("${RES_VAR}" STREQUAL "") OR ("${RES_VAR}" STREQUAL "0"))
     if ("${GDK_RES_ERROR}" STREQUAL "")
-    install(TARGETS pixbufloader-flif pixbufloader-flif-static 
-      RUNTIME DESTINATION ${GDK_MODULES_DIR}
-      LIBRARY DESTINATION ${GDK_MODULES_DIR}
-      ARCHIVE DESTINATION ${GDK_MODULES_DIR}) 
+    install(TARGETS pixbufloader-flif
+      RUNTIME DESTINATION ${GDK_MODULES_DIR})
     else()
       message(FATAL_ERROR "Error: ${GDK_RES_ERROR}")
     endif("${GDK_RES_ERROR}" STREQUAL "")
