--- origsrc/meep-1.2.1/libctl/Makefile.am	2014-03-29 06:24:02.000000000 +0900
+++ src/meep-1.2.1/libctl/Makefile.am	2015-01-04 10:01:28.234604400 +0900
@@ -23,7 +23,7 @@ meep_wrap.o: meep_wrap.cxx $(srcdir)/mee
 if WITH_LIBCTL
 if MAINTAINER_MODE
 meep_wrap.cxx: meep.i meep_op_renames.i meep_enum_renames.i meep_renames.i ctl-io.i meep-ctl-swig.hpp meep_swig_bug_workaround.i $(LIBHDRS)
-	swig -I$(top_srcdir)/src -c++ -guile -o $@ meep.i
+	swig -I$(top_srcdir)/src -c++ -guile -o $@ $(srcdir)/meep.i
 	patch -p0 $@ < $(srcdir)/meep_wrap.patch
 else
 meep_wrap.cxx:
--- origsrc/meep-1.2.1/src/Makefile.am	2014-03-29 02:26:58.000000000 +0900
+++ src/meep-1.2.1/src/Makefile.am	2015-01-04 10:01:28.234604400 +0900
@@ -19,7 +19,7 @@ stress.cpp structure.cpp susceptibility.
 mpb.cpp update_pols.cpp vec.cpp step_generic.cpp $(HDRS)		\
 $(BUILT_SOURCES)
 
-libmeep@MEEP_SUFFIX@_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@
+libmeep@MEEP_SUFFIX@_la_LDFLAGS = -no-undefined -version-info @SHARED_VERSION_INFO@
 
 PRELUDE = "/* This file was automatically generated --- DO NOT EDIT */"
 
