--- origsrc/meep-1.5.0/libmeepgeom/Makefile.am	2018-06-08 10:44:48.000000000 +0900
+++ src/meep-1.5.0/libmeepgeom/Makefile.am	2018-06-16 18:08:40.651628600 +0900
@@ -11,8 +11,8 @@ AM_CPPFLAGS = -I$(top_srcdir)/src -DDATA
 libmeepgeom_la_SOURCES = \
  meepgeom.cpp GDSIIgeom.cpp meepgeom.hpp material_data.hpp
 
-libmeepgeom_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@
-libmeepgeom_la_LIBADD = $(LIBMEEP)
+libmeepgeom_la_LDFLAGS = -no-undefined -version-info @SHARED_VERSION_INFO@
+libmeepgeom_la_LIBADD = $(MEEPLIBS)
 
 check_PROGRAMS      = pw-source-ll ring-ll cyl-ellipsoid-ll absorber-1d-ll array-slice-ll user-defined-material dft-fields
 
--- origsrc/meep-1.5.0/libpympb/Makefile.am	2018-06-08 10:44:48.000000000 +0900
+++ src/meep-1.5.0/libpympb/Makefile.am	2018-06-16 20:30:27.624587300 +0900
@@ -10,6 +10,6 @@ lib_LTLIBRARIES = libpympb.la
 pkginclude_HEADERS = pympb.hpp
 
 libpympb_la_SOURCES = pympb.cpp pympb.hpp
-libpympb_la_LIBADD = $(LIBMEEP)
-libpympb_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@
+libpympb_la_LIBADD = $(LIBMEEP) $(LIBCTL_LIBS) ../libmeepgeom/libmeepgeom.la
+libpympb_la_LDFLAGS = -no-undefined -version-info @SHARED_VERSION_INFO@
 libpympb_la_CPPFLAGS = $(AM_CPPFLAGS)
--- origsrc/meep-1.5.0/python/Makefile.am	2018-06-08 10:44:48.000000000 +0900
+++ src/meep-1.5.0/python/Makefile.am	2018-06-16 20:48:38.639556900 +0900
@@ -22,17 +22,18 @@ LIBMEEP = $(top_builddir)/src/libmeep.la
 
 AM_CPPFLAGS = -I$(top_srcdir)/src                   \
               -I$(top_srcdir)/libmeepgeom           \
+              -I$(top_srcdir)/libpympb              \
               $(PYMPBINCLUDE)                       \
               -I$(top_builddir) # for config.h
 
 _meep_la_SOURCES = meep-python.cpp
-_meep_la_LIBADD = $(LIBMEEP) $(LIBMEEPGEOM) $(PYTHON_LIBS) @LIBCTLGEOM_LIBS@
-_meep_la_LDFLAGS = -module -version-info @SHARED_VERSION_INFO@
+_meep_la_LIBADD = $(LIBMEEP) $(LIBMEEPGEOM) $(PYTHON_LIBS) @LIBCTLGEOM_LIBS@ -lpython$(PYTHON_VERSION)
+_meep_la_LDFLAGS = -no-undefined -module -avoid-version # -version-info @SHARED_VERSION_INFO@
 _meep_la_CPPFLAGS = $(PYTHON_INCLUDES) $(AM_CPPFLAGS)
 
 _mpb_la_SOURCES = mpb-python.cpp
-_mpb_la_LIBADD = $(LIBMEEPGEOM) $(PYTHON_LIBS) $(LIBPYMPB) @LIBCTLGEOM_LIBS@
-_mpb_la_LDFLAGS = -module -version-info @SHARED_VERSION_INFO@
+_mpb_la_LIBADD = $(LIBMEEPGEOM) $(PYTHON_LIBS) $(LIBPYMPB) @LIBCTLGEOM_LIBS@ -lpython$(PYTHON_VERSION)
+_mpb_la_LDFLAGS = -no-undefined -module -avoid-version # -version-info @SHARED_VERSION_INFO@
 _mpb_la_CPPFLAGS = $(PYTHON_INCLUDES) $(AM_CPPFLAGS)
 
 # material_dispersion.py test must be excluded from test suite for MPI build
@@ -133,13 +134,11 @@ PY_PKG_FILES =               \
     $(INIT_PY)               \
     $(srcdir)/geom.py        \
     $(srcdir)/simulation.py  \
-    $(srcdir)/source.py      \
-    .libs/_meep.so
+    $(srcdir)/source.py
 
 meep: _meep.la _mpb.la __init__.py
 	mkdir -p meep/mpb
 	cp $(PY_PKG_FILES) meep
-	cp .libs/_mpb.so meep/mpb
 	cp $(MPB_PY) meep/mpb/__init__.py
 	cp $(srcdir)/solver.py $(srcdir)/mpb_data.py meep/mpb
 
--- origsrc/meep-1.5.0/src/Makefile.am	2018-06-08 10:44:48.000000000 +0900
+++ src/meep-1.5.0/src/Makefile.am	2018-06-16 17:59:25.798702200 +0900
@@ -19,7 +19,7 @@ sources.cpp step.cpp step_db.cpp stress.
 susceptibility.cpp time.cpp update_eh.cpp mpb.cpp update_pols.cpp 	\
 vec.cpp step_generic.cpp $(HDRS) $(BUILT_SOURCES)
 
-libmeep_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@
+libmeep_la_LDFLAGS = -no-undefined -version-info @SHARED_VERSION_INFO@
 
 PRELUDE = "/* This file was automatically generated --- DO NOT EDIT */"
 
@@ -27,7 +27,7 @@ noinst_PROGRAMS = sphere_quad
 sphere_quad_SOURCES = sphere-quad.cpp
 SPHERE_QUAD = $(top_builddir)/src/sphere_quad
 sphere-quad.h:
-	make sphere_quad
+	make sphere_quad$(EXEEXT)
 	(echo $(PRELUDE); echo; $(SPHERE_QUAD)) > $@
 
 step_generic_stride1.cpp: step_generic.cpp
