--- origsrc/meep-1.2/libctl/Makefile.am	2012-07-21 02:44:40.000000000 +0900
+++ src/meep-1.2/libctl/Makefile.am	2012-09-26 21:06:59.337533400 +0900
@@ -23,7 +23,7 @@ meep_wrap.o: meep_wrap.cxx $(srcdir)/mee
 if WITH_LIBCTL
 if MAINTAINER_MODE
 meep_wrap.cxx: meep.i meep_op_renames.i meep_enum_renames.i meep_renames.i ctl-io.i meep-ctl-swig.hpp meep_swig_bug_workaround.i $(LIBHDRS)
-	swig -I$(top_srcdir)/src -c++ -guile -o $@ meep.i
+	swig -I$(top_srcdir)/src -c++ -guile -o $@ $(srcdir)/meep.i
 	patch -p0 $@ < $(srcdir)/meep_wrap.patch
 else
 meep_wrap.cxx:
--- origsrc/meep-1.2/src/Makefile.am	2012-06-27 04:20:19.000000000 +0900
+++ src/meep-1.2/src/Makefile.am	2012-09-26 21:07:12.493783400 +0900
@@ -19,7 +19,7 @@ stress.cpp structure.cpp susceptibility.
 mpb.cpp update_pols.cpp vec.cpp step_generic.cpp $(HDRS)		\
 $(BUILT_SOURCES)
 
-libmeep@MEEP_SUFFIX@_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@
+libmeep@MEEP_SUFFIX@_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ -no-undefined
 
 PRELUDE = "/* This file was automatically generated --- DO NOT EDIT */"
 
--- origsrc/meep-1.2/src/bicgstab.cpp	2012-07-20 08:17:24.000000000 +0900
+++ src/meep-1.2/src/bicgstab.cpp	2012-09-26 21:07:56.353158400 +0900
@@ -18,6 +18,8 @@
 #include <math.h>
 #include <string.h>
 
+#include <cstdio>
+
 #include "meep/mympi.hpp"
 #include "bicgstab.hpp"
 
