--- origsrc/libUnihan-0.5.3-Source/CMakeLists.txt	2008-10-20 16:13:07.000000000 +0900
+++ src/libUnihan-0.5.3-Source/CMakeLists.txt	2011-09-12 01:07:13.246904400 +0900
@@ -103,7 +103,6 @@ ADD_CUSTOM_TARGET(${DB_NAME}
     DEPENDS ${DB_NAME}.${DB_VER}-${DB_PATCH_LEVEL})
 
 
-
 SET(MAIN_DOCS AUTHORS README ChangeLog NEWS COPYING COPYING.LESSER )
 
 INSTALL(FILES ${MAIN_DOCS} 
@@ -154,15 +153,15 @@ INCLUDE(CPack)
 ###==================================================================
 # Document generation
 #
-FIND_PACKAGE(Doxygen)
-IF (EXISTS ${DOXYGEN})
-  ADD_CUSTOM_TARGET(doxygen    
-    COMMAND "${DOXYGEN}" "Doxyfile"
-    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}")
-  INSTALL(DIRECTORY doc/html
-    DESTINATION "${docdir}/${PROJECT_NAME}-${PRJ_VER}"
-  )
-ENDIF (EXISTS ${DOXYGEN})
+#FIND_PACKAGE(Doxygen)
+#IF (EXISTS ${DOXYGEN})
+#  ADD_CUSTOM_TARGET(doxygen    
+#    COMMAND "${DOXYGEN}" "Doxyfile"
+#    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}")
+#  INSTALL(DIRECTORY doc/html
+#    DESTINATION "${docdir}/${PROJECT_NAME}-${PRJ_VER}"
+#  )
+#ENDIF (EXISTS ${DOXYGEN})
 
 ### For Uninstall
 CONFIGURE_FILE(
--- origsrc/libUnihan-0.5.3-Source/src/CMakeLists.txt	2008-10-20 16:13:07.000000000 +0900
+++ src/libUnihan-0.5.3-Source/src/CMakeLists.txt	2011-09-12 01:12:02.326693200 +0900
@@ -50,6 +50,7 @@ ENDIF(NOT DEFINED LIB_INSTALL_DIR)
 INSTALL(TARGETS unihan_converter unihan_query Unihan
      RUNTIME DESTINATION bin
      LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+     ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
 )
 
 INSTALL(FILES ${DOCUMENTED_HEADERS}
