--- origsrc/libcomplearn-1.1.7/configure.ac	2010-11-25 06:26:29.000000000 +0900
+++ src/libcomplearn-1.1.7/configure.ac	2025-08-09 07:28:42.222039700 +0900
@@ -114,7 +114,7 @@ else
   GOMPFLAGS=""
 fi
 
-if (uname) | grep -i darwin 2>/dev/null 1>/dev/null ; then
+if (uname) | grep -i -e darwin -e cygwin 2>/dev/null 1>/dev/null ; then
   SONAMEOPTS=""
 else
   SONAMEOPTS="-Wl,-soname -Wl,libcomplearn${GOMPSTR}-1.0${shrext_cmds}.0"
--- origsrc/libcomplearn-1.1.7/src/Makefile.am	2010-01-11 10:14:37.000000000 +0900
+++ src/libcomplearn-1.1.7/src/Makefile.am	2025-08-09 07:28:42.229929500 +0900
@@ -2,10 +2,10 @@
 bin_PROGRAMS = ncd anycompress anydecompress complearn-config
 
 if USING_GOMP
-EXELDFLAGS = -lcomplearn-gomp @PRIVATELIBS@
+EXELDFLAGS = libcomplearn-gomp.la @PRIVATELIBS@
 lib_LTLIBRARIES   = libcomplearn-gomp.la
 else
-EXELDFLAGS = -lcomplearn @PRIVATELIBS@
+EXELDFLAGS = libcomplearn.la @PRIVATELIBS@
 lib_LTLIBRARIES   = libcomplearn.la
 endif
 
@@ -119,11 +119,12 @@ complearn/util.h
 
 if USING_GOMP
 libcomplearn_gomp_la_SOURCES = $(GOBCS) $(BASESOURCEC)
-libcomplearn_gomp_la_LDFLAGS= -version-info @libcomplearn_version@ @SONAMEOPTS@ @PRIVATELIBS@
+libcomplearn_gomp_la_LDFLAGS= -version-info @libcomplearn_version@ @SONAMEOPTS@ @PRIVATELIBS@ -no-undefined
 libcomplearn_gomp_la_CPPFLAGS = $(GLOBCPPFLAGS)
+libcomplearn_gomp_la_LIBADD = -lgomp
 else
 libcomplearn_la_SOURCES = $(GOBCS) $(BASESOURCEC)
-libcomplearn_la_LDFLAGS= -version-info @libcomplearn_version@ @SONAMEOPTS@ @PRIVATELIBS@
+libcomplearn_la_LDFLAGS= -version-info @libcomplearn_version@ @SONAMEOPTS@ @PRIVATELIBS@ -no-undefined
 libcomplearn_la_CPPFLAGS = $(GLOBCPPFLAGS)
 endif
 
--- origsrc/libcomplearn-1.1.7/src/anycompress.c	2010-01-11 09:03:47.000000000 +0900
+++ src/libcomplearn-1.1.7/src/anycompress.c	2025-08-09 07:28:42.230939100 +0900
@@ -29,7 +29,6 @@ SUCH DAMAGE.
 #include <stdio.h>
 #include <stdlib.h>
 #include <glib.h>
-#include <gmodule.h>
 #include <getopt.h>
 #include <locale.h>
 #include <libintl.h>
--- origsrc/libcomplearn-1.1.7/src/anydecompress.c	2010-01-11 09:03:47.000000000 +0900
+++ src/libcomplearn-1.1.7/src/anydecompress.c	2025-08-09 07:28:42.232948900 +0900
@@ -29,7 +29,7 @@ SUCH DAMAGE.
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <glib/garray.h>
+#include <glib.h>
 #include <getopt.h>
 #include <locale.h>
 #include <libintl.h>
--- origsrc/libcomplearn-1.1.7/src/complearn-config.c.in	2010-01-11 09:03:47.000000000 +0900
+++ src/libcomplearn-1.1.7/src/complearn-config.c.in	2025-08-09 07:28:42.232948900 +0900
@@ -12,6 +12,12 @@
 
 #define _(O) gettext(O)
 
+#if OPENMP_ENABLED
+#define LIBCOMPLEARN "complearn-gomp"
+#else
+#define LIBCOMPLEARN "complearn"
+#endif
+
 static void print_help_and_exit(void)
 {
   printf(_("Usage: complearn-config [options]\n"));
@@ -75,10 +81,10 @@ int main(int argc, char **argv)
         print_help_and_exit();
         break;
       case 's':
-        printf("%s -L%s/lib -lcomplearn\n", "@PRIVATELIBS@", fixup("@exec_prefix@"));
+        printf("%s -L%s/lib -l%s\n", "@PRIVATELIBS@", fixup("@exec_prefix@"), LIBCOMPLEARN);
         break;
       case 'l':
-        printf("-L%s/lib -lcomplearn\n", fixup("@exec_prefix@"));
+        printf("-L%s/lib -l%s\n", fixup("@exec_prefix@"), LIBCOMPLEARN);
         break;
       case 'c':
         printf("-I%s/include/complearn-1.0 %s\n", "@prefix@", "@PRIVATECFLAGS@");
--- origsrc/libcomplearn-1.1.7/tests/Makefile.am	2010-01-11 09:03:47.000000000 +0900
+++ src/libcomplearn-1.1.7/tests/Makefile.am	2025-08-09 07:28:42.232948900 +0900
@@ -2,8 +2,14 @@
 if FOUND_CHECK
 TESTS = check_complearn
 
+if USING_GOMP
+COMPLEARN_LIBS = -lcomplearn-gomp
+else
+COMPLEARN_LIBS = -lcomplearn
+endif
+
 check_PROGRAMS = check_complearn
 check_complearn_SOURCES = check_complearn.c check_ncd.c check_environment.c check_util.c check_cloutput.c suite.h
 check_complearn_CFLAGS = @CHECK_CFLAGS@ -I../src
-check_complearn_LDADD = @CHECK_LIBS@ -lcomplearn -L../src
+check_complearn_LDADD = @CHECK_LIBS@ $(COMPLEARN_LIBS) -L../src
 endif
--- origsrc/libcomplearn-1.1.7/tests/check_ncd.c	2010-01-11 09:03:47.000000000 +0900
+++ src/libcomplearn-1.1.7/tests/check_ncd.c	2025-08-09 15:40:33.216575900 +0900
@@ -23,6 +23,7 @@ START_TEST (test_complearn_ncd)
 END_TEST
 
 START_TEST (test_complearn_ncd_literal)
+{
   complearn_ncd_set_literal_mode ();
   complearn_ncd_set_zero_diagonal(complearn_ncd_top(), FALSE);
   complearn_ncd_add_argument ("hello");
@@ -45,6 +46,7 @@ complearn_ncd_get_argument(1));
   fail_if (lm->mat->size1 != 1 || lm->mat->size2 != 1,
     "Error, got matrix of %dx%d expected 1x1\n", lm->mat->size1,lm->mat->size2);
   fail_if (gsl_matrix_get(lm->mat, 0, 0) != 0.0);
+}
 END_TEST
 
 Suite *
--- origsrc/libcomplearn-1.1.7/tests/check_util.c	2010-01-11 09:03:47.000000000 +0900
+++ src/libcomplearn-1.1.7/tests/check_util.c	2025-08-09 15:41:13.862367700 +0900
@@ -5,6 +5,7 @@
 #include "../src/complearn/complearn.h"
 
 START_TEST (test_complearn_util_dir1)
+{
   char *tnam;
   int rv;
   tnam = complearn_make_temp_dir_name();
@@ -20,9 +21,11 @@ START_TEST (test_complearn_util_dir1)
   fail_unless (rv == 0);
   rv = complearn_remove_directory_recursively(tnam,0);
   fail_if (rv == 0);
+}
 END_TEST
 
 START_TEST (test_complearn_util_dir2)
+{
   char *tnam, *c1, *c2, *t2;
   int rv;
   tnam = complearn_make_temp_dir_name();
@@ -35,6 +38,7 @@ START_TEST (test_complearn_util_dir2)
   fail_unless(strcmp(tnam, c2) == 0);
   t2 = complearn_make_temp_dir_name();
   fail_if(strcmp(tnam, t2) == 0);
+}
 END_TEST
 
 Suite *
