--- origsrc/reprepro-5.4.8/database.c	2026-01-13 03:37:22.000000000 +0900
+++ src/reprepro-5.4.8/database.c	2026-01-31 23:07:55.565851400 +0900
@@ -312,11 +312,7 @@ static retvalue database_opentable(const
 		}
 	}
 
-#if DB_VERSION_MAJOR == 5 || DB_VERSION_MAJOR == 6
-#define DB_OPEN(database, filename, name, type, flags) \
-	database->open(database, NULL, filename, name, type, flags, 0664)
-#else
-#if DB_VERSION_MAJOR == 4
+#if DB_VERSION_MAJOR > 4
 #define DB_OPEN(database, filename, name, type, flags) \
 	database->open(database, NULL, filename, name, type, flags, 0664)
 #else
@@ -327,7 +323,6 @@ static retvalue database_opentable(const
 #error Unexpected DB_VERSION_MAJOR!
 #endif
 #endif
-#endif
 	dbret = DB_OPEN(table, filename, subtable, types[type], flags);
 	if (dbret == ENOENT && !ISSET(flags, DB_CREATE)) {
 		(void)table->close(table, 0);
--- origsrc/reprepro-5.4.8/uncompression.h	2026-01-13 03:37:22.000000000 +0900
+++ src/reprepro-5.4.8/uncompression.h	2026-01-31 23:08:10.020385200 +0900
@@ -1,6 +1,8 @@
 #ifndef REPREPRO_UNCOMPRESS_H
 #define REPREPRO_UNCOMPRESS_H
 
+#include <unistd.h>
+
 /* "", ".gz", ... */
 extern const char * const uncompression_suffix[c_COUNT];
 extern /*@null@*/ char *extern_uncompressors[c_COUNT];
