--- origsrc/avro-c-1.10.0/examples/CMakeLists.txt	2020-06-09 20:02:56.000000000 +0900
+++ src/avro-c-1.10.0/examples/CMakeLists.txt	2020-07-01 10:31:17.285076500 +0900
@@ -18,7 +18,7 @@
 #
 
 add_executable(quickstop quickstop.c)
-target_link_libraries(quickstop avro-static)
+target_link_libraries(quickstop avro-shared)
 
 if (WIN32)
     set(exec_name ${CMAKE_CURRENT_BINARY_DIR}/Debug/quickstop.exe)
--- origsrc/avro-c-1.10.0/src/CMakeLists.txt	2020-06-09 20:02:56.000000000 +0900
+++ src/avro-c-1.10.0/src/CMakeLists.txt	2020-07-01 10:32:57.740409400 +0900
@@ -84,9 +84,11 @@ source_group(Avro FILES ${AVRO_SRC})
 
 string(REPLACE ":" "." LIBAVRO_DOT_VERSION ${LIBAVRO_VERSION})
 
-add_library(avro-static STATIC ${AVRO_SRC})
-target_link_libraries(avro-static ${JANSSON_LIBRARIES} ${CODEC_LIBRARIES} ${THREADS_LIBRARIES})
-set_target_properties(avro-static PROPERTIES OUTPUT_NAME avro)
+if (WIN32 OR NOT BUILD_SHARED_LIBS)
+        add_library(avro-static STATIC ${AVRO_SRC})
+        target_link_libraries(avro-static ${JANSSON_LIBRARIES} ${CODEC_LIBRARIES} ${THREADS_LIBRARIES})
+        set_target_properties(avro-static PROPERTIES OUTPUT_NAME avro)
+endif()
 
 if (NOT WIN32)
 # TODO: Create Windows DLLs. See https://www.cmake.org/Wiki/BuildingWinDLL
@@ -108,14 +110,15 @@ install(DIRECTORY
 
 include(GNUInstallDirs)
 
-if (WIN32)
+if (WIN32 OR NOT BUILD_SHARED_LIBS)
 install(TARGETS avro-static
 	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
        )
-else(WIN32)
-install(TARGETS avro-static avro-shared
+endif()
+if (NOT WIN32)
+install(TARGETS avro-shared
 	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
 	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
@@ -131,20 +134,20 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 
 add_executable(avrocat avrocat.c)
-target_link_libraries(avrocat avro-static)
+target_link_libraries(avrocat avro-shared)
 install(TARGETS avrocat RUNTIME DESTINATION bin)
 
 add_executable(avroappend avroappend.c)
-target_link_libraries(avroappend avro-static)
+target_link_libraries(avroappend avro-shared)
 install(TARGETS avroappend RUNTIME DESTINATION bin)
 
 if (NOT WIN32)
 #TODO: Port getopt() to Windows to compile avropipe.c and avromod.c
 add_executable(avropipe avropipe.c)
-target_link_libraries(avropipe avro-static)
+target_link_libraries(avropipe avro-shared)
 install(TARGETS avropipe RUNTIME DESTINATION bin)
 
 add_executable(avromod avromod.c)
-target_link_libraries(avromod avro-static)
+target_link_libraries(avromod avro-shared)
 install(TARGETS avromod RUNTIME DESTINATION bin)
 endif(NOT WIN32)
--- origsrc/avro-c-1.10.0/tests/CMakeLists.txt	2020-06-09 20:02:56.000000000 +0900
+++ src/avro-c-1.10.0/tests/CMakeLists.txt	2020-07-01 10:31:39.291142700 +0900
@@ -23,7 +23,7 @@ macro(add_avro_executable name)
         set(source "${name}.c")
     endif (NOT source)
     add_executable(${name} ${source})
-    target_link_libraries(${name} avro-static)
+    target_link_libraries(${name} avro-shared)
 endmacro(add_avro_executable)
 
 macro(add_avro_test name)
