--- origsrc/avro-c-1.7.3/CMakeLists.txt	2012-12-04 03:11:45.000000000 +0900
+++ src/avro-c-1.7.3/CMakeLists.txt	2012-12-31 23:39:10.353007800 +0900
@@ -121,7 +121,18 @@ if(UNIX AND THREADSAFE AND CMAKE_COMPILE
 endif(UNIX AND THREADSAFE AND CMAKE_COMPILER_IS_GNUCC)
 
 include_directories(${AvroC_SOURCE_DIR}/src)
-include_directories(${AvroC_SOURCE_DIR}/jansson/src)
+find_package(PkgConfig)
+if(CYGWIN)
+    pkg_check_modules(JANSSON jansson)
+    if(JANSSON_FOUND)
+        include_directories(${JANSSON_INCLUDE_DIRS})
+        link_directories(${JANSSON_LIBRARY_DIRS})
+    else(JANSSON_FOUND)
+        include_directories(${AvroC_SOURCE_DIR}/jansson/src)
+    endif(JANSSON_FOUND)
+else(CYGWIN)
+    include_directories(${AvroC_SOURCE_DIR}/jansson/src)
+endif(CYGWIN)
 
 
 # Enable codecs
@@ -136,7 +147,6 @@ else (ZLIB_FOUND)
     message("Disabled deflate codec. zlib not found.")
 endif (ZLIB_FOUND)
 
-find_package(PkgConfig)
 pkg_check_modules(LZMA liblzma)
 if (LZMA_FOUND)
     set(LZMA_PKG liblzma)
--- origsrc/avro-c-1.7.3/examples/CMakeLists.txt	2012-12-04 03:11:45.000000000 +0900
+++ src/avro-c-1.7.3/examples/CMakeLists.txt	2012-12-31 23:43:11.290507800 +0900
@@ -18,7 +18,11 @@
 #
 
 add_executable(quickstop quickstop.c)
-target_link_libraries(quickstop avro-static)
+if(CYGWIN)
+    target_link_libraries(quickstop avro-shared)
+else(CYGWIN)
+    target_link_libraries(quickstop avro-static)
+endif(CYGWIN)
 
 if (WIN32)
     set(exec_name ${CMAKE_CURRENT_BINARY_DIR}/Debug/quickstop.exe)
--- origsrc/avro-c-1.7.3/src/CMakeLists.txt	2012-12-04 03:11:45.000000000 +0900
+++ src/avro-c-1.7.3/src/CMakeLists.txt	2012-12-31 23:42:01.338359300 +0900
@@ -101,14 +101,21 @@ source_group(Jansson FILES ${JANSSON_SRC
 
 string(REPLACE ":" "." LIBAVRO_DOT_VERSION ${LIBAVRO_VERSION})
 
+if (NOT CYGWIN)
 add_library(avro-static STATIC ${AVRO_SRC} ${JANSSON_SRC})
 target_link_libraries(avro-static ${CODEC_LIBRARIES} ${THREADS_LIBRARIES})
 set_target_properties(avro-static PROPERTIES OUTPUT_NAME avro)
+endif(NOT CYGWIN)
 
 if (NOT WIN32)
 # TODO: Create Windows DLLs. See http://www.cmake.org/Wiki/BuildingWinDLL
-add_library(avro-shared SHARED ${AVRO_SRC} ${JANSSON_SRC})
-target_link_libraries(avro-shared ${CODEC_LIBRARIES} ${THREADS_LIBRARIES})
+if(JANSSON_FOUND)
+    add_library(avro-shared SHARED ${AVRO_SRC})
+    target_link_libraries(avro-shared ${CODEC_LIBRARIES} ${THREADS_LIBRARIES} ${JANSSON_LIBRARIES})
+else(JANSSON_FOUND)
+    add_library(avro-shared SHARED ${AVRO_SRC} ${JANSSON_SRC})
+    target_link_libraries(avro-shared ${CODEC_LIBRARIES} ${THREADS_LIBRARIES})
+endif(JANSSON_FOUND)
 set_target_properties(avro-shared PROPERTIES
         OUTPUT_NAME avro
         SOVERSION ${LIBAVRO_DOT_VERSION})
@@ -128,6 +135,12 @@ install(TARGETS avro-static
         LIBRARY DESTINATION lib
         ARCHIVE DESTINATION lib
        )
+elseif (CYGWIN)
+install(TARGETS avro-shared
+        RUNTIME DESTINATION bin
+        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION lib
+       )
 else(WIN32)
 install(TARGETS avro-static avro-shared
         RUNTIME DESTINATION bin
@@ -145,20 +158,36 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR
         DESTINATION lib/pkgconfig)
 
 add_executable(avrocat avrocat.c)
-target_link_libraries(avrocat avro-static)
+if(CYGWIN)
+    target_link_libraries(avrocat avro-shared)
+else(CYGWIN)
+    target_link_libraries(avrocat avro-static)
+endif(CYGWIN)
 install(TARGETS avrocat RUNTIME DESTINATION bin)
 
 add_executable(avroappend avroappend.c)
-target_link_libraries(avroappend avro-static)
+if(CYGWIN)
+    target_link_libraries(avroappend avro-shared)
+else(CYGWIN)
+    target_link_libraries(avroappend avro-static)
+endif(CYGWIN)
 install(TARGETS avroappend RUNTIME DESTINATION bin)
 
 if (NOT WIN32)
 #TODO: Port getopt() to Windows to compile avropipe.c and avromod.c
 add_executable(avropipe avropipe.c)
-target_link_libraries(avropipe avro-static)
+if(CYGWIN)
+    target_link_libraries(avropipe avro-shared)
+else(CYGWIN)
+    target_link_libraries(avropipe avro-static)
+endif(CYGWIN)
 install(TARGETS avropipe RUNTIME DESTINATION bin)
 
 add_executable(avromod avromod.c)
-target_link_libraries(avromod avro-static)
+if(CYGWIN)
+    target_link_libraries(avromod avro-shared)
+else(CYGWIN)
+    target_link_libraries(avromod avro-static)
+endif(CYGWIN)
 install(TARGETS avromod RUNTIME DESTINATION bin)
 endif(NOT WIN32)
--- origsrc/avro-c-1.7.3/tests/CMakeLists.txt	2012-12-04 03:11:45.000000000 +0900
+++ src/avro-c-1.7.3/tests/CMakeLists.txt	2012-12-31 23:43:56.418437500 +0900
@@ -23,7 +23,11 @@ macro(add_avro_executable name)
         set(source "${name}.c")
     endif (NOT source)
     add_executable(${name} ${source})
-    target_link_libraries(${name} avro-static)
+    if(CYGWIN)
+        target_link_libraries(${name} avro-shared)
+    else(CYGWIN)
+        target_link_libraries(${name} avro-static)
+    endif(CYGWIN)
 endmacro(add_avro_executable)
 
 macro(add_avro_test name)
