--- origsrc/avro-c-1.12.0/CMakeLists.txt	2024-07-26 04:23:22.000000000 +0900
+++ src/avro-c-1.12.0/CMakeLists.txt	2024-08-19 06:29:20.472633400 +0900
@@ -151,7 +151,9 @@ endif (ZLIB_FOUND)
 
 find_package(Snappy)
 if (SNAPPY_FOUND AND ZLIB_FOUND)  # Snappy borrows crc32 from zlib
+    if (NOT CYGWIN) # official cygwin package of snappy does not contain snappy.pc
     set(SNAPPY_PKG snappy)
+    endif()
     add_definitions(-DSNAPPY_CODEC)
     include_directories(${SNAPPY_INCLUDE_DIRS})
     message("Enabled snappy codec")
--- origsrc/avro-c-1.12.0/examples/CMakeLists.txt	2024-07-26 04:23:22.000000000 +0900
+++ src/avro-c-1.12.0/examples/CMakeLists.txt	2024-08-19 06:24:08.158887500 +0900
@@ -18,7 +18,7 @@
 #
 
 add_executable(quickstop quickstop.c)
-target_link_libraries(quickstop avro-static)
+target_link_libraries(quickstop avro-shared)
 
 if (WIN32)
     set(exec_name ${CMAKE_CURRENT_BINARY_DIR}/Debug/quickstop.exe)
--- origsrc/avro-c-1.12.0/src/CMakeLists.txt	2024-07-26 04:23:22.000000000 +0900
+++ src/avro-c-1.12.0/src/CMakeLists.txt	2024-08-19 06:23:30.382903500 +0900
@@ -84,10 +84,6 @@ source_group(Avro FILES ${AVRO_SRC})
 
 string(REPLACE ":" "." LIBAVRO_DOT_VERSION ${LIBAVRO_VERSION})
 
-add_library(avro-static STATIC ${AVRO_SRC})
-target_link_libraries(avro-static ${JANSSON_LIBRARIES} ${CODEC_LIBRARIES} ${THREADS_LIBRARIES})
-set_target_properties(avro-static PROPERTIES OUTPUT_NAME avro)
-
 if (NOT WIN32)
 # TODO: Create Windows DLLs. See https://www.cmake.org/Wiki/BuildingWinDLL
 add_library(avro-shared SHARED ${AVRO_SRC})
@@ -108,19 +104,11 @@ install(DIRECTORY
 
 include(GNUInstallDirs)
 
-if (WIN32)
-install(TARGETS avro-static
-	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-       )
-else(WIN32)
-install(TARGETS avro-static avro-shared
+install(TARGETS avro-shared
 	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
 	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
        )
-endif(WIN32)
 
 # Install pkg-config file
 
@@ -131,20 +119,20 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 
 add_executable(avrocat avrocat.c)
-target_link_libraries(avrocat avro-static)
+target_link_libraries(avrocat avro-shared)
 install(TARGETS avrocat RUNTIME DESTINATION bin)
 
 add_executable(avroappend avroappend.c)
-target_link_libraries(avroappend avro-static)
+target_link_libraries(avroappend avro-shared)
 install(TARGETS avroappend RUNTIME DESTINATION bin)
 
 if (NOT WIN32)
 #TODO: Port getopt() to Windows to compile avropipe.c and avromod.c
 add_executable(avropipe avropipe.c)
-target_link_libraries(avropipe avro-static)
+target_link_libraries(avropipe avro-shared)
 install(TARGETS avropipe RUNTIME DESTINATION bin)
 
 add_executable(avromod avromod.c)
-target_link_libraries(avromod avro-static)
+target_link_libraries(avromod avro-shared)
 install(TARGETS avromod RUNTIME DESTINATION bin)
 endif(NOT WIN32)
--- origsrc/avro-c-1.12.0/tests/CMakeLists.txt	2024-07-26 04:23:22.000000000 +0900
+++ src/avro-c-1.12.0/tests/CMakeLists.txt	2024-08-19 06:23:45.882872800 +0900
@@ -23,7 +23,7 @@ macro(add_avro_executable name)
         set(source "${name}.c")
     endif (NOT source)
     add_executable(${name} ${source})
-    target_link_libraries(${name} avro-static)
+    target_link_libraries(${name} avro-shared)
 endmacro(add_avro_executable)
 
 macro(add_avro_test name)
