--- origsrc/libfprint-0.7.0/examples/Makefile.am	2015-09-01 13:52:30.000000000 +0900
+++ src/libfprint-0.7.0/examples/Makefile.am	2017-10-05 01:03:33.029000000 +0900
@@ -1,4 +1,4 @@
-AM_CFLAGS = -I$(top_srcdir)
+AM_CPPFLAGS = -I$(top_srcdir)
 noinst_PROGRAMS = verify_live enroll verify img_capture cpp-test
 
 verify_live_SOURCES = verify_live.c
--- origsrc/libfprint-0.7.0/libfprint/Makefile.am	2017-01-06 02:18:55.000000000 +0900
+++ src/libfprint-0.7.0/libfprint/Makefile.am	2017-10-05 00:53:00.841000000 +0900
@@ -96,7 +96,7 @@ NBIS_SRC = \
 	nbis/mindtct/util.c
 
 libfprint_la_CFLAGS = -fvisibility=hidden -I$(srcdir)/nbis/include $(LIBUSB_CFLAGS) $(GLIB_CFLAGS) $(CRYPTO_CFLAGS) $(AM_CFLAGS)
-libfprint_la_LDFLAGS = -version-info @lt_major@:@lt_revision@:@lt_age@
+libfprint_la_LDFLAGS = -no-undefined -version-info @lt_major@:@lt_revision@:@lt_age@
 libfprint_la_LIBADD = -lm $(LIBUSB_LIBS) $(GLIB_LIBS) $(CRYPTO_LIBS)
 
 fprint_list_udev_rules_SOURCES = fprint-list-udev-rules.c
--- origsrc/libfprint-0.7.0/libfprint/assembling.c	2017-01-06 02:18:55.000000000 +0900
+++ src/libfprint-0.7.0/libfprint/assembling.c	2017-10-05 00:59:37.044000000 +0900
@@ -342,8 +342,6 @@ static void interpolate_lines(struct fpi
 	}
 }
 
-static int min(int a, int b) {return (a < b) ? a : b; }
-
 /* Rescale image to account for variable swiping speed */
 struct fp_img *fpi_assemble_lines(struct fpi_line_asmbl_ctx *ctx,
 				  GSList *lines, size_t lines_len)
--- origsrc/libfprint-0.7.0/libfprint/drivers/vfs101.c	2017-01-06 02:18:55.000000000 +0900
+++ src/libfprint-0.7.0/libfprint/drivers/vfs101.c	2017-10-05 01:01:01.029000000 +0900
@@ -136,6 +136,10 @@ struct vfs101_dev
 	int height;
 };
 
+#if defined(__CYGWIN__)
+#define byte(p,v) byte_(p,v)
+#endif
+
 /* Return byte at specified position */
 static inline unsigned char byte(int position, int value)
 {
--- origsrc/libfprint-0.7.0/libfprint/nbis/include/bz_array.h	2013-01-12 22:18:37.000000000 +0900
+++ src/libfprint-0.7.0/libfprint/nbis/include/bz_array.h	2017-10-05 00:58:34.514069500 +0900
@@ -37,6 +37,9 @@ identified are necessarily the best avai
 #define SCOLPT_SIZE 20000
 #define FCOLPT_SIZE 20000
 
+#if defined(SC_SIZE)
+#undef SC_SIZE
+#endif
 #define SC_SIZE 20000
 
 
--- origsrc/libfprint-0.7.0/libfprint/nbis/include/lfs.h	2015-12-17 14:44:42.000000000 +0900
+++ src/libfprint-0.7.0/libfprint/nbis/include/lfs.h	2017-10-05 00:57:23.639226600 +0900
@@ -68,8 +68,12 @@ identified are necessarily the best avai
 /*        MACRO DEFINITIONS                                              */
 /*************************************************************************/
 
+#if !defined(max)
 #define max(a, b)   ((a) > (b) ? (a) : (b))
+#endif
+#if !defined(min)
 #define min(a, b)   ((a) < (b) ? (a) : (b))
+#endif
 #define sround(x) ((int) (((x)<0) ? (x)-0.5 : (x)+0.5))
 #define trunc_dbl_precision(x, scale) ((double) (((x)<0.0) \
                  ? ((int)(((x)*(scale))-0.5))/(scale) \
@@ -647,6 +651,10 @@ typedef struct lfsparms{
 #define NOT_FOUND            FALSE
 #endif
 
+#if defined(IGNORE)
+#undef IGNORE
+#endif
+
 #define HOOK_FOUND               1
 #define LOOP_FOUND               1
 #define IGNORE                   2
