--- origsrc/libfprint/libfprint/Makefile.am	2013-01-07 21:30:31.000000000 +0900
+++ src/libfprint/libfprint/Makefile.am	2013-01-07 21:31:32.523318000 +0900
@@ -83,7 +83,7 @@ NBIS_SRC = \
 	nbis/mindtct/util.c
 
 libfprint_la_CFLAGS = -fvisibility=hidden -I$(srcdir)/nbis/include $(LIBUSB_CFLAGS) $(GLIB_CFLAGS) $(CRYPTO_CFLAGS) $(AM_CFLAGS)
-libfprint_la_LDFLAGS = -version-info @lt_major@:@lt_revision@:@lt_age@
+libfprint_la_LDFLAGS = -version-info @lt_major@:@lt_revision@:@lt_age@ -no-undefined
 libfprint_la_LIBADD = -lm $(LIBUSB_LIBS) $(GLIB_LIBS) $(CRYPTO_LIBS)
 
 fprint_list_udev_rules_SOURCES = fprint-list-udev-rules.c
--- origsrc/libfprint/libfprint/drivers/vfs101.c	2013-01-07 21:30:32.000000000 +0900
+++ src/libfprint/libfprint/drivers/vfs101.c	2013-01-07 21:37:14.910036800 +0900
@@ -139,6 +139,10 @@ struct vfs101_dev
 	int height;
 };
 
+#if defined(__CYGWIN__)
+#define byte(p,v) byte_(p,v)
+#endif
+
 /* Return byte at specified position */
 static inline unsigned char byte(int position, int value)
 {
--- origsrc/libfprint/libfprint/poll.c	2013-01-07 21:30:32.000000000 +0900
+++ src/libfprint/libfprint/poll.c	2013-01-07 21:35:43.612185200 +0900
@@ -108,6 +108,12 @@ struct fpi_timeout *fpi_timeout_add(unsi
 	timeout = g_malloc(sizeof(*timeout));
 	timeout->callback = callback;
 	timeout->data = data;
+#if !defined(TIMESPEC_TO_TIMEVAL)
+#define TIMESPEC_TO_TIMEVAL(tv, ts) do {                                \
+        (tv)->tv_sec = (ts)->tv_sec;                                    \
+        (tv)->tv_usec = (suseconds_t)(ts)->tv_nsec / 1000;              \
+} while (0)
+#endif
 	TIMESPEC_TO_TIMEVAL(&timeout->expiry, &ts);
 
 	/* calculate timeout expiry by adding delay to current monotonic clock */
