--- origsrc/libfprint-v1.94.6/libfprint/meson.build	2023-08-17 12:08:04.000000000 +0900
+++ src/libfprint-v1.94.6/libfprint/meson.build	2024-08-31 13:35:56.852899700 +0900
@@ -16,8 +16,10 @@ libfprint_private_sources = [
     'fpi-print.c',
     'fpi-ssm.c',
     'fpi-usb-transfer.c',
-    'fpi-spi-transfer.c',
 ]
+if host_machine.system() == 'linux'
+    libfprint_private_sources += ['fpi-spi-transfer.c']
+endif
 
 libfprint_public_headers = [
     'fp-context.h',
@@ -41,9 +43,11 @@ libfprint_private_headers = [
     'fpi-minutiae.h',
     'fpi-print.h',
     'fpi-usb-transfer.h',
-    'fpi-spi-transfer.h',
     'fpi-ssm.h',
 ]
+if host_machine.system() == 'linux'
+    libfprint_private_headers += ['fpi-spi-transfer.h']
+endif
 
 nbis_sources = [
     'nbis/bozorth3/bozorth3.c',
--- origsrc/libfprint-v1.94.6/meson.build	2023-08-17 12:08:04.000000000 +0900
+++ src/libfprint-v1.94.6/meson.build	2024-08-31 13:27:44.773587000 +0900
@@ -131,10 +131,10 @@ default_drivers = [
     'goodixmoc',
     'nb1010',
     'fpcmoc',
-
-    # SPI
-    'elanspi',
 ]
+if host_machine.system() == 'linux'
+    default_drivers += ['elanspi']
+endif
 
 # FIXME: All the drivers should be fixed by adjusting the byte order.
 # See https://gitlab.freedesktop.org/libfprint/libfprint/-/issues/236
