--- origsrc/libfprint/libfprint/Makefile.am	2012-03-28 13:31:54.000000000 +0900
+++ src/libfprint/libfprint/Makefile.am	2012-03-28 13:34:05.301151600 +0900
@@ -71,7 +71,7 @@ NBIS_SRC = \
 	nbis/mindtct/util.c
 
 libfprint_la_CFLAGS = -fvisibility=hidden -I$(srcdir)/nbis/include $(LIBUSB_CFLAGS) $(GLIB_CFLAGS) $(CRYPTO_CFLAGS) $(AM_CFLAGS)
-libfprint_la_LDFLAGS = -version-info @lt_major@:@lt_revision@:@lt_age@
+libfprint_la_LDFLAGS = -version-info @lt_major@:@lt_revision@:@lt_age@ -no-undefined
 libfprint_la_LIBADD = -lm $(LIBUSB_LIBS) $(GLIB_LIBS) $(CRYPTO_LIBS)
 
 fprint_list_udev_rules_SOURCES = fprint-list-udev-rules.c
--- origsrc/libfprint/libfprint/drivers/vfs101.c	2012-03-28 13:31:54.000000000 +0900
+++ src/libfprint/libfprint/drivers/vfs101.c	2012-03-28 13:47:40.727909500 +0900
@@ -25,6 +25,10 @@
 
 #include <fp_internal.h>
 
+#if defined(__CYGWIN__)
+#define byte(p,v) byte0(p,v)
+#endif
+
 /* Input-Output usb endpoint */
 #define EP_IN(n)	(n | LIBUSB_ENDPOINT_IN)
 #define EP_OUT(n)	(n | LIBUSB_ENDPOINT_OUT)
--- origsrc/libfprint/libfprint/poll.c	2012-03-28 13:31:54.000000000 +0900
+++ src/libfprint/libfprint/poll.c	2012-03-28 13:42:38.913456300 +0900
@@ -29,6 +29,13 @@
 
 #include "fp_internal.h"
 
+#if !defined(TIMESPEC_TO_TIMEVAL)
+#define TIMESPEC_TO_TIMEVAL(tv, ts) do {                                \
+        (tv)->tv_sec = (ts)->tv_sec;                                    \
+        (tv)->tv_usec = (suseconds_t)(ts)->tv_nsec / 1000;              \
+} while (0)
+#endif
+
 /**
  * @defgroup poll Polling and timing operations
  * These functions are only applicable to users of libfprint's asynchronous
