--- origsrc/libfprint-0.5.1/libfprint/Makefile.am	2013-06-26 21:14:49.000000000 +0900
+++ src/libfprint-0.5.1/libfprint/Makefile.am	2013-09-16 12:00:47.327746700 +0900
@@ -83,7 +83,7 @@ NBIS_SRC = \
 	nbis/mindtct/util.c
 
 libfprint_la_CFLAGS = -fvisibility=hidden -I$(srcdir)/nbis/include $(LIBUSB_CFLAGS) $(GLIB_CFLAGS) $(CRYPTO_CFLAGS) $(AM_CFLAGS)
-libfprint_la_LDFLAGS = -version-info @lt_major@:@lt_revision@:@lt_age@
+libfprint_la_LDFLAGS = -version-info @lt_major@:@lt_revision@:@lt_age@ -no-undefined
 libfprint_la_LIBADD = -lm $(LIBUSB_LIBS) $(GLIB_LIBS) $(CRYPTO_LIBS)
 
 fprint_list_udev_rules_SOURCES = fprint-list-udev-rules.c
--- origsrc/libfprint-0.5.1/libfprint/drivers/vfs101.c	2013-06-26 21:13:00.000000000 +0900
+++ src/libfprint-0.5.1/libfprint/drivers/vfs101.c	2013-09-16 12:00:47.343371700 +0900
@@ -139,6 +139,10 @@ struct vfs101_dev
 	int height;
 };
 
+#if defined(__CYGWIN__)
+#define byte(p,v) byte_(p,v)
+#endif
+
 /* Return byte at specified position */
 static inline unsigned char byte(int position, int value)
 {
--- origsrc/libfprint-0.5.1/libfprint/nbis/include/bz_array.h	2013-06-26 21:13:00.000000000 +0900
+++ src/libfprint-0.5.1/libfprint/nbis/include/bz_array.h	2013-09-16 12:08:05.460408100 +0900
@@ -37,6 +37,9 @@ identified are necessarily the best avai
 #define SCOLPT_SIZE 20000
 #define FCOLPT_SIZE 20000
 
+#if defined(SC_SIZE)
+#undef SC_SIZE
+#endif
 #define SC_SIZE 20000
 
 
--- origsrc/libfprint-0.5.1/libfprint/nbis/include/lfs.h	2013-06-26 21:13:00.000000000 +0900
+++ src/libfprint-0.5.1/libfprint/nbis/include/lfs.h	2013-09-16 12:07:20.411580000 +0900
@@ -68,6 +68,13 @@ identified are necessarily the best avai
 /*        MACRO DEFINITIONS                                              */
 /*************************************************************************/
 
+#if defined(max)
+#undef max
+#endif
+#if defined(min)
+#undef min
+#endif
+
 #define max(a, b)   ((a) > (b) ? (a) : (b))
 #define min(a, b)   ((a) < (b) ? (a) : (b))
 #define sround(x) ((int) (((x)<0) ? (x)-0.5 : (x)+0.5))
@@ -644,6 +651,9 @@ typedef struct lfsparms{
 
 #define HOOK_FOUND               1
 #define LOOP_FOUND               1
+#if defined(IGNORE)
+#undef IGNORE
+#endif
 #define IGNORE                   2
 #define LIST_FULL                3
 #define INCOMPLETE               3
--- origsrc/libfprint-0.5.1/libfprint/poll.c	2013-06-26 21:13:00.000000000 +0900
+++ src/libfprint-0.5.1/libfprint/poll.c	2013-09-16 12:00:47.358996700 +0900
@@ -108,6 +108,12 @@ struct fpi_timeout *fpi_timeout_add(unsi
 	timeout = g_malloc(sizeof(*timeout));
 	timeout->callback = callback;
 	timeout->data = data;
+#if !defined(TIMESPEC_TO_TIMEVAL)
+#define TIMESPEC_TO_TIMEVAL(tv, ts) do {                                \
+        (tv)->tv_sec = (ts)->tv_sec;                                    \
+        (tv)->tv_usec = (suseconds_t)(ts)->tv_nsec / 1000;              \
+} while (0)
+#endif
 	TIMESPEC_TO_TIMEVAL(&timeout->expiry, &ts);
 
 	/* calculate timeout expiry by adding delay to current monotonic clock */
