--- origsrc/libfprint-v1.94.100/libfprint/meson.build	2026-07-25 22:36:25.000000000 +0000
+++ src/libfprint-v1.94.100/libfprint/meson.build	2026-08-25 22:00:55.285114400 +0000
@@ -308,7 +308,6 @@ udev_hwdb = executable('fprint-list-udev
     dependencies: libfprint_private_dep,
     link_with: libfprint_drivers,
     install: false,
-    link_args: ['-Wl,--disable-new-dtags'],
 )
 
 udev_hwdb_generator = custom_target('udev-hwdb',
@@ -324,7 +323,6 @@ metainfo = executable('fprint-list-metai
     dependencies: libfprint_private_dep,
     link_with: libfprint_drivers,
     install: false,
-    link_args: ['-Wl,--disable-new-dtags'],
 )
 
 metainfo_generator = custom_target('metainfo',
@@ -342,7 +340,6 @@ if install_udev_rules
         dependencies: libfprint_private_dep,
         link_with: libfprint_drivers,
         install: false,
-        link_args: ['-Wl,--disable-new-dtags'],
     )
 
     custom_target('udev-rules',
