--- origsrc/libsml-0.1.1/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libsml-0.1.1/Makefile.am	2016-04-11 08:25:56.721979900 +0900
@@ -0,0 +1,65 @@
+
+pkginclude_HEADERS = \
+sml/include/sml/sml_attention_response.h \
+sml/include/sml/sml_boolean.h \
+sml/include/sml/sml_close_request.h \
+sml/include/sml/sml_close_response.h \
+sml/include/sml/sml_crc16.h \
+sml/include/sml/sml_file.h \
+sml/include/sml/sml_get_list_request.h \
+sml/include/sml/sml_get_list_response.h \
+sml/include/sml/sml_get_proc_parameter_request.h \
+sml/include/sml/sml_get_proc_parameter_response.h \
+sml/include/sml/sml_get_profile_list_request.h \
+sml/include/sml/sml_get_profile_list_response.h \
+sml/include/sml/sml_get_profile_pack_request.h \
+sml/include/sml/sml_get_profile_pack_response.h \
+sml/include/sml/sml_list.h \
+sml/include/sml/sml_message.h \
+sml/include/sml/sml_number.h \
+sml/include/sml/sml_octet_string.h \
+sml/include/sml/sml_open_request.h \
+sml/include/sml/sml_open_response.h \
+sml/include/sml/sml_set_proc_parameter_request.h \
+sml/include/sml/sml_shared.h \
+sml/include/sml/sml_status.h \
+sml/include/sml/sml_time.h \
+sml/include/sml/sml_transport.h \
+sml/include/sml/sml_tree.h \
+sml/include/sml/sml_value.h
+
+lib_LTLIBRARIES = libsml.la
+libsml_la_SOURCES = \
+sml/src/sml_attention_response.c \
+sml/src/sml_boolean.c \
+sml/src/sml_close_request.c \
+sml/src/sml_close_response.c \
+sml/src/sml_crc16.c \
+sml/src/sml_file.c \
+sml/src/sml_get_list_request.c \
+sml/src/sml_get_list_response.c \
+sml/src/sml_get_proc_parameter_request.c \
+sml/src/sml_get_proc_parameter_response.c \
+sml/src/sml_get_profile_list_request.c \
+sml/src/sml_get_profile_list_response.c \
+sml/src/sml_get_profile_pack_request.c \
+sml/src/sml_get_profile_pack_response.c \
+sml/src/sml_list.c \
+sml/src/sml_message.c \
+sml/src/sml_number.c \
+sml/src/sml_octet_string.c \
+sml/src/sml_open_request.c \
+sml/src/sml_open_response.c \
+sml/src/sml_set_proc_parameter_request.c \
+sml/src/sml_shared.c \
+sml/src/sml_status.c \
+sml/src/sml_time.c \
+sml/src/sml_transport.c \
+sml/src/sml_tree.c \
+sml/src/sml_value.c
+libsml_la_CPPFLAGS = -I$(top_srcdir)/sml/include
+libsml_la_LIBADD = $(UUID_LIBS)
+libsml_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = sml.pc
--- origsrc/libsml-0.1.1/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/libsml-0.1.1/configure.ac	2016-04-11 08:25:56.721979900 +0900
@@ -0,0 +1,40 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([libsml], [0.1.1])
+AC_CONFIG_SRCDIR([examples/sml_server.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+# FIXME: Replace `main' with a function in `-luuid':
+PKG_CHECK_MODULES([UUID], [uuid])
+
+# Checks for header files.
+AC_CHECK_HEADERS([fcntl.h stdint.h stdlib.h string.h sys/ioctl.h sys/time.h termios.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+AC_C_INLINE
+AC_TYPE_INT16_T
+AC_TYPE_INT32_T
+AC_TYPE_INT64_T
+AC_TYPE_INT8_T
+AC_TYPE_SIZE_T
+AC_TYPE_UINT16_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+AC_TYPE_UINT8_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([memset select strerror strstr])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
