--- origsrc/msgpack-c-cpp-7.0.0/example/boost/CMakeLists.txt	2024-11-02 11:41:41.000000000 +0900
+++ src/msgpack-c-cpp-7.0.0/example/boost/CMakeLists.txt	2025-03-19 14:03:31.220308900 +0900
@@ -23,6 +23,7 @@ FOREACH (source_file ${exec_PROGRAMS})
     TARGET_LINK_LIBRARIES (${source_file_we} PRIVATE
         msgpack-cxx
         Threads::Threads
+        boost_system
     )
     IF (ZLIB_FOUND)
         TARGET_LINK_LIBRARIES (${source_file_we} PRIVATE ZLIB::ZLIB)
