--- origsrc/libdebian-installer/configure.ac	2012-10-20 21:31:09.000000000 +0900
+++ src/libdebian-installer/configure.ac	2013-01-13 16:27:08.324109300 +0900
@@ -1,4 +1,4 @@
-AC_INIT([libdebian-installer],m4_esyscmd(dpkg-parsechangelog | perl -ne 'print $1 if m/^Version: (.*)$/;'))
+AC_INIT([libdebian-installer],[0.85])
 AM_INIT_AUTOMAKE([])
 AM_MAINTAINER_MODE
 
--- origsrc/libdebian-installer/src/Makefile.am	2012-10-20 21:31:10.000000000 +0900
+++ src/libdebian-installer/src/Makefile.am	2013-01-13 16:26:43.183484300 +0900
@@ -35,12 +35,12 @@ libdebian_installer_extra_la_LIBADD = \
 libdebian_installer_la_LDFLAGS = \
 	-version-number $(LIBRARY_VERSION_LIBTOOL) \
 	-Wl,--version-script=$(srcdir)/libdebian-installer.ver \
-	-Wl,--no-undefined-version
+	-Wl,--no-undefined-version -no-undefined
 
 libdebian_installer_extra_la_LDFLAGS = \
 	-version-number $(LIBRARY_VERSION_LIBTOOL) \
 	-Wl,--version-script=$(srcdir)/libdebian-installer-extra.ver \
-	-Wl,--no-undefined-version
+	-Wl,--no-undefined-version -no-undefined
 
 libdebian_installer_la_DEPENDENCIES = \
 	$(srcdir)/libdebian-installer.ver \
--- origsrc/libdebian-installer/src/log.c	2012-10-20 21:31:10.000000000 +0900
+++ src/libdebian-installer/src/log.c	2013-01-13 16:26:43.183484300 +0900
@@ -212,7 +212,9 @@ void di_log (di_log_level_flags log_leve
   va_end (args);
 }
 
+#if !defined(__CYGWIN__)
 __asm__ (".symver di_log,di_log_real_4_0@LIBDI_4.0");
+#endif
 
 void di_vlog (di_log_level_flags log_level, const char *format, va_list args)
 {
--- origsrc/libdebian-installer/src/package_parser.c	2012-10-20 21:31:10.000000000 +0900
+++ src/libdebian-installer/src/package_parser.c	2013-01-13 16:26:43.199109300 +0900
@@ -431,7 +431,9 @@ void di_package_parser_read_name_real_4_
   p->key.size = value->size;
 }
 
+#if !defined(__CYGWIN__)
 __asm__ (".symver di_package_parser_read_name_real_4_0,di_package_parser_read_name@LIBDI_4.0");
+#endif
 
 void di_package_parser_read_priority (
   void **data,
--- origsrc/libdebian-installer/src/system/packages.c	2012-10-20 21:31:10.000000000 +0900
+++ src/libdebian-installer/src/system/packages.c	2013-01-13 16:26:43.199109300 +0900
@@ -257,5 +257,7 @@ void di_system_packages_resolve_dependen
   di_system_packages_resolve_dependencies_mark_anna (packages, "unknown", uts.release);
 }
 
+#if !defined(__CYGWIN__)
 __asm__ (".symver di_system_packages_resolve_dependencies_mark_kernel_real_4_2_unstable,di_system_packages_resolve_dependencies_mark_kernel@LIBDI_4.2_UNSTABLE");
+#endif
 
--- origsrc/libdebian-installer/src/system/prebaseconfig.c	2012-10-20 21:31:10.000000000 +0900
+++ src/libdebian-installer/src/system/prebaseconfig.c	2013-01-13 16:26:43.214734300 +0900
@@ -53,5 +53,7 @@ int di_system_prebaseconfig_append (cons
 
 int di_prebaseconfig_append (const char *udeb, const char *fmt, ...) __attribute__ ((alias("di_system_prebaseconfig_append")));
 
+#if !defined(__CYGWIN__)
 __asm__ (".symver di_system_prebaseconfig_append,di_system_prebaseconfig_append@LIBDI_4.0");
 __asm__ (".symver di_prebaseconfig_append,di_prebaseconfig_append@LIBDI_4.0");
+#endif
