--- origsrc/benoic-2.1.0/Makefile	2018-03-07 14:01:25.000000000 +0900
+++ src/benoic-2.1.0/Makefile	2026-06-15 06:19:41.236377100 +0900
@@ -24,7 +24,8 @@
 # Environment variables
 PREFIX=/usr/local
 CC=gcc
-CFLAGS=-c -Wall -D_REENTRANT -I$(PREFIX)/include $(ADDITIONALFLAGS)
+CFLAGS?=
+CFLAGS+=-c -Wall -D_REENTRANT -I$(PREFIX)/include $(ADDITIONALFLAGS)
 LIBS=-L$(PREFIX)/lib -lc -ldl -lpthread -ljansson -lulfius -lhoel -lyder -lorcania
 MODULES_LOCATION=device-modules
 
