--- origsrc/csmith-2.1.0/runtime/Makefile.am	2011-11-23 02:57:09.000000000 +0900
+++ src/csmith-2.1.0/runtime/Makefile.am	2012-03-30 23:13:59.039345900 +0900
@@ -41,6 +41,7 @@ libcsmith_a_SOURCES = \
 lib_LTLIBRARIES = libcsmith.la
 libcsmith_la_SOURCES = \
 	$(libcsmith_a_SOURCES)
+libcsmith_la_LDFLAGS = -no-undefined
 
 GIT_HASH := $(shell (cd "$(top_srcdir)" && git show -s --format=%h . 2>/dev/null) || echo exported)
 GIT_FLAG = -DGIT_VERSION=\"$(GIT_HASH)\"
