--- origsrc/csmith-csmith-2.3.0/Makefile.am	2017-06-22 05:50:24.000000000 +0900
+++ src/csmith-csmith-2.3.0/Makefile.am	2025-12-13 18:35:55.704846800 +0900
@@ -44,3 +44,6 @@ EXTRA_DIST = \
 ###############################################################################
 
 ## End of file.
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/csmith-csmith-2.3.0/configure.ac	2017-06-22 05:50:24.000000000 +0900
+++ src/csmith-csmith-2.3.0/configure.ac	2025-12-13 18:36:11.477672500 +0900
@@ -89,6 +89,7 @@ fi
 AC_CHECK_PROGS([M4], [m4 gm4], [m4])
 
 AC_OUTPUT(
+  csmith.pc
   Makefile
   doc/Makefile
   runtime/Makefile
--- origsrc/csmith-csmith-2.3.0/csmith.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/csmith-csmith-2.3.0/csmith.pc.in	2025-12-13 18:35:55.633486000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: csmith
+Description: Random C program generator
+Version: @VERSION@
+Libs: -L${libdir} -lcsmith
+Cflags: -I${includedir}
--- origsrc/csmith-csmith-2.3.0/runtime/Makefile.am	2025-12-13 18:05:34.442288800 +0900
+++ src/csmith-csmith-2.3.0/runtime/Makefile.am	2025-12-13 18:06:37.594084000 +0900
@@ -31,7 +31,6 @@
 
 .DELETE_ON_ERROR:
 
-lib_LIBRARIES = libcsmith.a
 libcsmith_a_SOURCES = \
 	volatile_runtime.c
 
@@ -44,6 +43,8 @@ lib_LTLIBRARIES = libcsmith.la
 libcsmith_la_SOURCES = \
 	$(libcsmith_a_SOURCES)
 
+libcsmith_la_LDFLAGS = -no-undefined
+
 libcsmith_a_CPPFLAGS =
 libcsmith_la_CPPFLAGS = \
 	$(libcsmith_a_CPPFLAGS)
