--- origsrc/log4shib-2.0.0/src/Makefile.am	2018-07-07 03:39:40.000000000 +0900
+++ src/log4shib-2.0.0/src/Makefile.am	2018-08-26 13:33:50.959644100 +0900
@@ -48,5 +48,5 @@ liblog4shib_la_SOURCES = \
 	PortabilityImpl.cpp \
 	AbortAppender.cpp
 
-liblog4shib_la_LDFLAGS = -version-info @LT_VERSION@
+liblog4shib_la_LDFLAGS = -no-undefined -version-info @LT_VERSION@
 liblog4shib_la_LIBADD = $(PTHREAD_LIBS)
