--- origsrc/nghttp3/configure.ac	2025-04-25 07:29:00.000000000 +0900
+++ src/nghttp3/configure.ac	2025-04-25 07:29:08.980646000 +0900
@@ -101,6 +101,7 @@ PKG_PROG_PKG_CONFIG([0.20])
 AX_CXX_COMPILE_STDCXX([17], [], [optional])
 
 # Checks for libraries.
+PKG_CHECK_MODULES([SFPARSE], [sfparse])
 
 enable_examples=yes
 if test "x${lib_only}" = "xyes"; then
--- origsrc/nghttp3/lib/Makefile.am	2025-04-25 07:29:00.000000000 +0900
+++ src/nghttp3/lib/Makefile.am	2025-04-25 07:29:08.980646000 +0900
@@ -62,8 +62,7 @@ OBJECTS = \
 	nghttp3_balloc.c \
 	nghttp3_opl.c \
 	nghttp3_objalloc.c \
-	nghttp3_unreachable.c \
-	sfparse/sfparse.c
+	nghttp3_unreachable.c
 HFILES = \
 	nghttp3_rcbuf.h \
 	nghttp3_mem.h \
@@ -91,9 +90,9 @@ HFILES = \
 	nghttp3_opl.h \
 	nghttp3_objalloc.h \
 	nghttp3_unreachable.h \
-	sfparse/sfparse.h \
 	nghttp3_macro.h
 
 libnghttp3_la_SOURCES = $(HFILES) $(OBJECTS)
+libnghttp3_la_LIBADD  = $(SFPARSE_LIBS)
 libnghttp3_la_LDFLAGS = -no-undefined \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
--- origsrc/nghttp3/lib/nghttp3_http.c	2025-04-25 07:29:00.000000000 +0900
+++ src/nghttp3/lib/nghttp3_http.c	2025-04-25 07:29:08.980646000 +0900
@@ -36,7 +36,7 @@
 #include "nghttp3_macro.h"
 #include "nghttp3_conv.h"
 #include "nghttp3_unreachable.h"
-#include "sfparse/sfparse.h"
+#include <sfparse.h>
 
 static uint8_t downcase(uint8_t c) {
   return 'A' <= c && c <= 'Z' ? (uint8_t)(c - 'A' + 'a') : c;
--- origsrc/nghttp3/tests/Makefile.am	2025-04-25 07:29:00.000000000 +0900
+++ src/nghttp3/tests/Makefile.am	2025-04-25 07:29:08.980646000 +0900
@@ -50,9 +50,8 @@ main_SOURCES = $(HFILES) $(OBJECTS)
 
 # With static lib disabled and symbol hiding enabled, we have to link object
 # files directly because the tests use symbols not included in public API.
-main_LDADD = ${top_builddir}/lib/.libs/*.o \
-	${top_builddir}/lib/sfparse/.libs/*.o
-main_LDFLAGS = -static
+main_LDADD = ${top_builddir}/lib/libnghttp3.la $(SFPARSE_LIBS)
+main_LDFLAGS =
 
 AM_CFLAGS = $(WARNCFLAGS) \
 	-I${top_srcdir}/lib \
