--- origsrc/c-toxcore-0.1.9/configure.ac	2017-06-07 05:50:57.000000000 +0900
+++ src/c-toxcore-0.1.9/configure.ac	2017-06-25 14:42:33.579000000 +0900
@@ -289,6 +289,9 @@ case $host_os in
         WIN32="yes"
         EXTRA_LT_LDFLAGS="$EXTRA_LT_LDFLAGS -no-undefined"
     ;;
+    *cygwin*)
+        EXTRA_LT_LDFLAGS="$EXTRA_LT_LDFLAGS -no-undefined"
+    ;;
     *solaris*)
         LIBS="$LIBS -lssp -lsocket -lnsl"
     ;;
