--- origsrc/flameshot-0.9.0/src/CMakeLists.txt	2021-02-28 05:36:12.000000000 +0900
+++ src/flameshot-0.9.0/src/CMakeLists.txt	2021-05-31 19:40:15.645697900 +0900
@@ -112,10 +112,16 @@ target_sources(
         ${CMAKE_CURRENT_SOURCE_DIR}/../external/Qt-Color-Widgets/src/color_wheel.cpp
         ${CMAKE_CURRENT_SOURCE_DIR}/../external/Qt-Color-Widgets/include/color_wheel.hpp
         ${CMAKE_CURRENT_SOURCE_DIR}/../data/graphics.qrc
-        ${CMAKE_CURRENT_SOURCE_DIR}/../data/flameshot.rc # windows binary icon resource file
         ${QM_FILES}
         main.cpp)
 
+if (WIN32)
+  target_sources(
+    flameshot
+    PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../data/flameshot.rc # windows binary icon resource file
+  )
+endif()
+
 target_include_directories(
         flameshot
         PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../>
