--- origsrc/flameshot-0.8.5/src/CMakeLists.txt	2020-10-15 08:27:04.000000000 +0900
+++ src/flameshot-0.8.5/src/CMakeLists.txt	2020-10-18 21:57:30.900098700 +0900
@@ -67,9 +67,14 @@ target_sources(
           ${CMAKE_CURRENT_SOURCE_DIR}/../external/Qt-Color-Widgets/src/color_wheel.cpp
           ${CMAKE_CURRENT_SOURCE_DIR}/../external/Qt-Color-Widgets/include/color_wheel.hpp
           ${CMAKE_CURRENT_SOURCE_DIR}/../data/graphics.qrc
-          ${CMAKE_CURRENT_SOURCE_DIR}/../data/flameshot.rc # windows binary icon resource file
           ${QM_FILES}
           main.cpp)
+if (WIN32)
+  target_sources(
+    flameshot
+    PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../data/flameshot.rc # windows binary icon resource file
+  )
+endif()
 
 target_include_directories(
   flameshot
