--- origsrc/libpll-1.0.11/MPI/Makefile.am	2014-10-16 20:55:44.000000000 +0900
+++ src/libpll-1.0.11/MPI/Makefile.am	2017-08-25 00:48:30.269000000 +0900
@@ -12,7 +12,8 @@ lib_LTLIBRARIES += ../src/libpll-sse3-mp
 lib_LIBRARIES += ../src/libpll-sse3-mpi.a
 ___src_libpll_sse3_mpi_la_SOURCES = $(ALL_SOURCES)
 ___src_libpll_sse3_mpi_la_CFLAGS = -D_FINE_GRAIN_MPI -D__SSE3 -msse3 -c -O2 -fomit-frame-pointer -funroll-loops -Wall -Wredundant-decls  -Wreturn-type  -Wswitch-default -Wimplicit  -Wimplicit-function-declaration  -Wimplicit-int -Wimport -Wunused-label -Wno-int-to-pointer-cast -Wbad-function-cast  -Wmissing-declarations -Wmissing-prototypes  -Wnested-externs  -Wold-style-definition -Wstrict-prototypes -Wpointer-sign -Wextra -Wredundant-decls -Wunused -Wunused-function -Wunused-parameter -Wunused-value  -Wunused-variable -Wformat  -Wformat-nonliteral -Wparentheses -Wsequence-point -Wuninitialized -Wundef 
-___src_libpll_sse3_mpi_la_LDFLAGS = -version-info @LIBPLL_MAJOR@:@LIBPLL_MINOR@:@LIBPLL_REV@
+___src_libpll_sse3_mpi_la_LDFLAGS = -no-undefined -version-info @LIBPLL_MAJOR@:@LIBPLL_MINOR@:@LIBPLL_REV@
+___src_libpll_sse3_mpi_la_LIBADD = -lmpi
 ___src_libpll_sse3_mpi_a_SOURCES = $(___src_libpll_sse3_mpi_la_SOURCES)
 ___src_libpll_sse3_mpi_a_CFLAGS = $(___src_libpll_sse3_mpi_la_CFLAGS)
 endif
@@ -22,7 +23,8 @@ lib_LTLIBRARIES += ../src/libpll-avx-mpi
 lib_LIBRARIES += ../src/libpll-avx-mpi.a
 ___src_libpll_avx_mpi_la_SOURCES = $(AVX_SOURCES)
 ___src_libpll_avx_mpi_la_CFLAGS = -D_FINE_GRAIN_MPI -D__SSE3 -D__AVX -mavx -c -O2 -fomit-frame-pointer -funroll-loops -Wall -Wredundant-decls  -Wreturn-type  -Wswitch-default -Wimplicit  -Wimplicit-function-declaration  -Wimplicit-int -Wimport -Wunused-label -Wno-int-to-pointer-cast -Wbad-function-cast  -Wmissing-declarations -Wmissing-prototypes  -Wnested-externs  -Wold-style-definition -Wstrict-prototypes -Wpointer-sign -Wextra -Wredundant-decls -Wunused -Wunused-function -Wunused-parameter -Wunused-value  -Wunused-variable -Wformat  -Wformat-nonliteral -Wparentheses -Wsequence-point -Wuninitialized -Wundef 
-___src_libpll_avx_mpi_la_LDFLAGS = -version-info @LIBPLL_MAJOR@:@LIBPLL_MINOR@:@LIBPLL_REV@
+___src_libpll_avx_mpi_la_LDFLAGS = -no-undefined -version-info @LIBPLL_MAJOR@:@LIBPLL_MINOR@:@LIBPLL_REV@
+___src_libpll_avx_mpi_la_LIBADD = -lmpi
 ___src_libpll_avx_mpi_a_SOURCES = $(___src_libpll_avx_mpi_la_SOURCES)
 ___src_libpll_avx_mpi_a_CFLAGS = $(___src_libpll_avx_mpi_la_CFLAGS)
 endif
--- origsrc/libpll-1.0.11/src/Makefile.am	2015-02-25 00:10:45.000000000 +0900
+++ src/libpll-1.0.11/src/Makefile.am	2017-08-24 20:15:25.593000000 +0900
@@ -4,7 +4,7 @@ lib_LTLIBRARIES =
 lib_LIBRARIES = 
 libpll_generic_la_SOURCES = hash.c stack.c ssort.c queue.c utils.c randomTree.c optimizeModel.c trash.c searchAlgo.c topologies.c fastDNAparsimony.c treeIO.c models.c evaluatePartialGenericSpecial.c evaluateGenericSpecial.c newviewGenericSpecial.c makenewzGenericSpecial.c bipartitionList.c restartHashTable.c recom.c lexer.c alignment.c newick.c parsePartition.c parsimony.c
 libpll_generic_la_CFLAGS = -c -D_GNU_SOURCE -O2 -fomit-frame-pointer -funroll-loops -Wall -Wredundant-decls  -Wreturn-type  -Wswitch-default -Wimplicit  -Wimplicit-function-declaration  -Wimplicit-int -Wimport -Wunused-label -Wno-int-to-pointer-cast -Wbad-function-cast  -Wmissing-declarations -Wmissing-prototypes  -Wnested-externs  -Wold-style-definition -Wstrict-prototypes -Wpointer-sign -Wextra -Wredundant-decls -Wunused -Wunused-function -Wunused-parameter -Wunused-value  -Wunused-variable -Wformat  -Wformat-nonliteral -Wparentheses -Wsequence-point -Wuninitialized -Wundef 
-libpll_generic_la_LDFLAGS = -version-info @LIBPLL_MAJOR@:@LIBPLL_MINOR@:@LIBPLL_REV@
+libpll_generic_la_LDFLAGS = -no-undefined -version-info @LIBPLL_MAJOR@:@LIBPLL_MINOR@:@LIBPLL_REV@
 libpll_generic_a_SOURCES = $(libpll_generic_la_SOURCES)
 libpll_generic_a_CFLAGS = $(libpll_generic_la_CFLAGS)
 
@@ -12,7 +12,7 @@ if BUILD_SSE3
 lib_LTLIBRARIES += libpll-sse3.la
 libpll_sse3_la_SOURCES = $(libpll_generic_la_SOURCES)
 libpll_sse3_la_CFLAGS = -D__SSE3 -msse3 $(libpll_generic_la_CFLAGS)
-libpll_sse3_la_LDFLAGS = -version-info @LIBPLL_MAJOR@:@LIBPLL_MINOR@:@LIBPLL_REV@
+libpll_sse3_la_LDFLAGS = -no-undefined -version-info @LIBPLL_MAJOR@:@LIBPLL_MINOR@:@LIBPLL_REV@
 lib_LIBRARIES += libpll-sse3.a
 libpll_sse3_a_SOURCES = $(libpll_sse3_la_SOURCES)
 libpll_sse3_a_CFLAGS = $(libpll_sse3_la_CFLAGS)
@@ -23,7 +23,7 @@ if BUILD_AVX
 lib_LTLIBRARIES += libpll-avx.la
 libpll_avx_la_SOURCES = avxLikelihood.c $(libpll_generic_la_SOURCES)
 libpll_avx_la_CFLAGS = -D__SSE3 -D__AVX -mavx -msse3 $(libpll_generic_la_CFLAGS)
-libpll_avx_la_LDFLAGS = -version-info @LIBPLL_MAJOR@:@LIBPLL_MINOR@:@LIBPLL_REV@
+libpll_avx_la_LDFLAGS = -no-undefined -version-info @LIBPLL_MAJOR@:@LIBPLL_MINOR@:@LIBPLL_REV@
 lib_LIBRARIES += libpll-avx.a
 libpll_avx_a_SOURCES = $(libpll_avx_la_SOURCES)
 libpll_avx_a_CFLAGS = $(libpll_avx_la_CFLAGS)
@@ -33,7 +33,7 @@ if BUILD_SSE3_PTHREADS
 lib_LTLIBRARIES += libpll-sse3-pthreads.la
 libpll_sse3_pthreads_la_SOURCES = genericParallelization.c $(libpll_generic_la_SOURCES)
 libpll_sse3_pthreads_la_CFLAGS = -D_USE_PTHREADS -D__SSE3 -msse3 $(libpll_generic_la_CFLAGS)
-libpll_sse3_pthreads_la_LDFLAGS = -version-info @LIBPLL_MAJOR@:@LIBPLL_MINOR@:@LIBPLL_REV@
+libpll_sse3_pthreads_la_LDFLAGS = -no-undefined -version-info @LIBPLL_MAJOR@:@LIBPLL_MINOR@:@LIBPLL_REV@
 lib_LIBRARIES += libpll-sse3-pthreads.a
 libpll_sse3_pthreads_a_SOURCES = $(libpll_sse3_pthreads_la_SOURCES)
 libpll_sse3_pthreads_a_CFLAGS = $(libpll_sse3_pthreads_la_CFLAGS)
@@ -43,7 +43,7 @@ if BUILD_AVX_PTHREADS
 lib_LTLIBRARIES += libpll-avx-pthreads.la
 libpll_avx_pthreads_la_SOURCES = avxLikelihood.c genericParallelization.c $(libpll_generic_la_SOURCES)
 libpll_avx_pthreads_la_CFLAGS = -D_USE_PTHREADS -D__AVX -mavx -D__SSE3 -msse3 $(libpll_generic_la_CFLAGS)
-libpll_avx_pthreads_la_LDFLAGS = -version-info @LIBPLL_MAJOR@:@LIBPLL_MINOR@:@LIBPLL_REV@
+libpll_avx_pthreads_la_LDFLAGS = -no-undefined -version-info @LIBPLL_MAJOR@:@LIBPLL_MINOR@:@LIBPLL_REV@
 lib_LIBRARIES += libpll-avx-pthreads.a
 libpll_avx_pthreads_a_SOURCES = $(libpll_avx_pthreads_la_SOURCES)
 libpll_avx_pthreads_a_CFLAGS = $(libpll_avx_pthreads_la_CFLAGS)
--- origsrc/libpll-1.0.11/src/newviewGenericSpecial.c	2015-01-21 20:21:23.000000000 +0900
+++ src/libpll-1.0.11/src/newviewGenericSpecial.c	2017-08-25 00:28:29.456000000 +0900
@@ -6407,6 +6407,9 @@ static void newviewGTRCAT( int tipCase,
     @return
       Returns the value of the bit vector (\b 1 if set, \b 0 if not)
 */
+#if defined(__CYGWIN__)
+extern
+#endif
 #ifndef __clang__
 inline 
 #endif
@@ -6426,6 +6429,9 @@ boolean isGap(unsigned int *x, int pos)
     @return
       Returns the value of the bit vector (\b 1 if set, \b 0 if not)
 */
+#if defined(__CYGWIN__)
+extern
+#endif
 #ifndef __clang__
 inline 
 #endif
