--- origsrc/libe131-1.4.0/Makefile.am	2025-06-15 18:03:52.527799100 +0900
+++ src/libe131-1.4.0/Makefile.am	2025-06-15 18:11:12.355280800 +0900
@@ -5,8 +5,11 @@ dist_doc_DATA = LICENSE README.md
 
 lib_LTLIBRARIES = libe131.la
 libe131_la_SOURCES = src/e131.c
-libe131_la_LDFLAGS = -export-symbols-regex '^[eE]131_[^_]'
+libe131_la_LDFLAGS = -no-undefined -export-symbols-regex '^[eE]131_[^_]'
 include_HEADERS = src/e131.h
 
 # see: https://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
 libe131_la_LDFLAGS += -version-info 5:0:4
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libe131-1.4.0/configure.ac	2018-05-14 22:30:34.000000000 +0900
+++ src/libe131-1.4.0/configure.ac	2025-06-15 18:11:12.355280800 +0900
@@ -37,4 +37,5 @@ AC_TYPE_UINT8_T
 AC_CHECK_FUNCS([gethostbyname memset socket])
 
 AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([libe131.pc])
 AC_OUTPUT
--- origsrc/libe131-1.4.0/libe131.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libe131-1.4.0/libe131.pc.in	2025-06-15 18:11:12.323594900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libe131
+Description: Lightweight C/C++ library for the E1.31 (sACN) protocol
+Version: @VERSION@
+Libs: -L${libdir} -le131
+Cflags: -I${includedir}
--- origsrc/libe131-1.4.0/src/e131.c	2018-05-14 22:30:34.000000000 +0900
+++ src/libe131-1.4.0/src/e131.c	2025-06-15 18:08:44.406271900 +0900
@@ -102,6 +102,10 @@ int e131_dest_str(char *str, const e131_
 
 /* Join a socket file descriptor to an E1.31 multicast group using a universe */
 int e131_multicast_join(int sockfd, const uint16_t universe) {
+#if defined(__CYGWIN__)
+  errno = ENOPROTOOPT;
+  return -1;
+#else
   if (universe < 1 || universe > 63999) {
     errno = EINVAL;
     return -1;
@@ -111,6 +115,7 @@ int e131_multicast_join(int sockfd, cons
   mreq.imr_address.s_addr = htonl(INADDR_ANY);
   mreq.imr_ifindex = 0;
   return setsockopt(sockfd, IPPROTO_IP, IP_ADD_MEMBERSHIP, &mreq, sizeof mreq);
+#endif
 }
 
 /* Initialize an E1.31 packet using a universe and a number of slots */
