--- origsrc/libplctag-2.4.0/CMakeLists.txt	2021-09-19 00:08:14.000000000 +0900
+++ src/libplctag-2.4.0/CMakeLists.txt	2021-10-06 22:09:26.378206000 +0900
@@ -593,7 +593,9 @@ if(ANDROID_BUILD)
 endif(ANDROID_BUILD)
 
 # for installation
-if(UNIX)
+if(CYGWIN)
+    install(TARGETS plctag_dyn RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+elseif(UNIX)
 	install(TARGETS plctag_dyn DESTINATION lib${LIB_SUFFIX})
 	install(TARGETS plctag_static DESTINATION lib${LIB_SUFFIX})
 elseif(WIN32)
