--- origsrc/libplctag-2.1.21/CMakeLists.txt	2020-11-23 01:57:08.000000000 +0900
+++ src/libplctag-2.1.21/CMakeLists.txt	2020-11-25 22:06:55.752615400 +0900
@@ -305,14 +305,14 @@ if(BASE_LINK_FLAGS)
     set_target_properties(plctag_dyn PROPERTIES LINK_FLAGS "${BASE_LINK_FLAGS}")
 endif()
 
-if(UNIX)
+if(UNIX AND NOT CYGWIN)
 	# static library
 	add_library(plctag_static STATIC ${libplctag_SRCS} )
     set_target_properties(plctag_static PROPERTIES LINK_FLAGS "${BASE_LINK_FLAGS}")
 	set_target_properties(plctag_static PROPERTIES VERSION "${libplctag_VERSION_MAJOR}.${libplctag_VERSION_MINOR}" OUTPUT_NAME "plctag")
 
 	set(tool_lib "plctag_dyn")
-elseif(WIN32)
+elseif(WIN32 OR CYGWIN)
     # skipping static lib on Windows
 
 	set(tool_lib "plctag_dyn")
@@ -463,11 +463,11 @@ else()
 endif(ANDROID_BUILD)
 
 # for installation
-if(UNIX)
+if(UNIX AND NOT CYGWIN)
 	install(TARGETS plctag_dyn DESTINATION lib${LIB_SUFFIX})
 	install(TARGETS plctag_static DESTINATION lib${LIB_SUFFIX})
-elseif(WIN32)
-	install(TARGETS plctag_dyn DESTINATION lib${LIB_SUFFIX})
+elseif(WIN32 OR CYGWIN)
+	install(TARGETS plctag_dyn RUNTIME DESTINATION bin ARCHIVE DESTINATION lib${LIB_SUFFIX})
 endif()
 
 if(ANDROID_BUILD)
