--- origsrc/libplctag-2.0.20/CMakeLists.txt	2019-07-21 07:05:12.000000000 +0900
+++ src/libplctag-2.0.20/CMakeLists.txt	2019-08-11 13:49:45.745861200 +0900
@@ -167,6 +167,9 @@ ENDFOREACH()
 # shared library
 add_library(plctag SHARED ${libplctag_SRCS} )
 
+if(CYGWIN)
+    set_target_properties(plctag PROPERTIES SOVERSION 0)
+endif()
 
 #if(THREADS_HAVE_PTHREAD_ARG)
 #  set_property(TARGET plctag PROPERTY COMPILE_OPTIONS "-pthread")
@@ -253,12 +256,17 @@ endif()
 set_source_files_properties("${example_SRC_PATH}/simple_cpp.cpp" PROPERTIES COMPILE_FLAGS ${BASE_CXX_FLAGS})
 add_executable (simple_cpp "${example_SRC_PATH}/simple_cpp.cpp" "${example_SRC_PATH}/${example_PROG_UTIL}" "${example_SRC_PATH}/utils.h" )
 target_link_libraries (simple_cpp ${example_LIBRARIES} )
-
+if(CYGWIN)
+    install(TARGETS simple_cpp RUNTIME DESTINATION share/libplctag/example)
+endif()
 
 foreach ( example ${example_PROGRAMS} )
     set_source_files_properties("${example_SRC_PATH}/${example}.c" PROPERTIES COMPILE_FLAGS ${BASE_C_FLAGS})
     add_executable( ${example} "${example_SRC_PATH}/${example}.c" "${example_SRC_PATH}/${example_PROG_UTIL}" "${example_SRC_PATH}/utils.h" )
     target_link_libraries(${example} ${example_LIBRARIES} )
+    if(CYGWIN)
+        install(TARGETS ${example} RUNTIME DESTINATION share/libplctag/example)
+    endif()
 endforeach(example)
 
 
@@ -266,6 +274,6 @@ endforeach(example)
 CONFIGURE_FILE("${CMAKE_CURRENT_SOURCE_DIR}/libplctag.pc.in" "${CMAKE_CURRENT_BINARY_DIR}/libplctag.pc" @ONLY)
 
 # for installation
-install(TARGETS plctag DESTINATION lib${LIB_SUFFIX})
+install(TARGETS plctag RUNTIME DESTINATION bin ARCHIVE DESTINATION lib${LIB_SUFFIX})
 install(FILES "${lib_SRC_PATH}/libplctag.h" DESTINATION include)
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libplctag.pc" DESTINATION "lib${LIB_SUFFIX}/pkgconfig")
