--- origsrc/mydumper-0.9.3/cmake/modules/FindMySQL.cmake	2017-09-07 23:53:02.000000000 +0900
+++ src/mydumper-0.9.3/cmake/modules/FindMySQL.cmake	2017-11-24 13:21:36.184000000 +0900
@@ -84,7 +84,9 @@ find_path(MYSQL_INCLUDE_DIR mysql.h
 )
 
 set(TMP_MYSQL_LIBRARIES "")
-set(CMAKE_FIND_LIBRARY_SUFFIXES .so .a .lib .so.1)
+if(NOT CYGWIN)
+    set(CMAKE_FIND_LIBRARY_SUFFIXES .so .a .lib .so.1)
+endif()
 foreach(MY_LIB ${MYSQL_ADD_LIBRARIES})
     find_library("MYSQL_LIBRARIES_${MY_LIB}" NAMES ${MY_LIB}
         HINTS
