--- origsrc/mydumper-0.9.5/cmake/modules/FindMySQL.cmake	2018-05-04 23:53:10.000000000 +0900
+++ src/mydumper-0.9.5/cmake/modules/FindMySQL.cmake	2018-11-25 13:18:20.256197600 +0900
@@ -84,7 +84,9 @@ find_path(MYSQL_INCLUDE_DIR mysql.h
 )
 
 set(TMP_MYSQL_LIBRARIES "")
-set(CMAKE_FIND_LIBRARY_SUFFIXES .so .a .lib .so.1)
+if(NOT CYGWIN)
+    set(CMAKE_FIND_LIBRARY_SUFFIXES .so .a .lib .so.1)
+endif()
 foreach(MY_LIB ${MYSQL_ADD_LIBRARIES})
     find_library("MYSQL_LIBRARIES_${MY_LIB}" NAMES ${MY_LIB}
         HINTS
