--- origsrc/keybinder-0.2.2/Makefile.am	2010-05-31 04:51:53.000000000 +0900
+++ src/keybinder-0.2.2/Makefile.am	2012-04-27 18:43:55.268006600 +0900
@@ -25,3 +25,4 @@ DISTCLEANFILES = \
 	intltool-update \
 	gnome-doc-utils.make
 
+ACLOCAL_AMFLAGS = -I m4
--- origsrc/keybinder-0.2.2/configure.ac	2010-05-31 04:51:53.000000000 +0900
+++ src/keybinder-0.2.2/configure.ac	2012-04-27 18:53:15.545350300 +0900
@@ -165,6 +165,7 @@ if test "x$enable_lua" != "xno"; then
     LUA_VERSION=[5.1]
     AC_SUBST(LUA_VERSION)
     AC_SUBST(LUA_INCLUDE)
+    PKG_CHECK_MODULES([LUA], [lua])
     have_lua="yes"
   else
     have_lua="no"
--- origsrc/keybinder-0.2.2/libkeybinder/Makefile.am	2010-05-28 06:37:33.000000000 +0900
+++ src/keybinder-0.2.2/libkeybinder/Makefile.am	2012-04-27 18:42:26.549622800 +0900
@@ -9,7 +9,7 @@ LIBKEYBINDER_VERSION = $(LT_CURRENT):$(L
 keybinderdir = $(libdir)
 keybinder_LTLIBRARIES = libkeybinder.la
 libkeybinder_la_SOURCES = bind.c
-libkeybinder_la_LDFLAGS = $(GTK_LIBS)  -version-info $(LIBKEYBINDER_VERSION)
+libkeybinder_la_LDFLAGS = $(GTK_LIBS)  -version-info $(LIBKEYBINDER_VERSION) -no-undefined
 include_HEADERS = keybinder.h
 
 pkgconfigdir = $(libdir)/pkgconfig
--- origsrc/keybinder-0.2.2/lua-keybinder/Makefile.am	2010-04-30 01:33:28.000000000 +0900
+++ src/keybinder-0.2.2/lua-keybinder/Makefile.am	2012-04-27 18:54:06.310975300 +0900
@@ -1,9 +1,9 @@
-INCLUDES = $(LUA_INCLUDE) $(GTK_CFLAGS) -I../libkeybinder
+INCLUDES = $(LUA_CFLAGS) $(GTK_CFLAGS) -I$(top_srcdir)/libkeybinder
 
 AM_CFLAGS = -Wall
 
 luakeybinderdir = $(libdir)/lua/$(LUA_VERSION)
 luakeybinder_LTLIBRARIES = keybinder.la
-keybinder_la_LDFLAGS = -module -avoid-version
-keybinder_la_LIBADD = ../libkeybinder/libkeybinder.la $(LUA_LIB)
+keybinder_la_LDFLAGS = -module -avoid-version -no-undefined
+keybinder_la_LIBADD = ../libkeybinder/libkeybinder.la $(LUA_LIBS)
 keybinder_la_SOURCES = lkeybinder.c
--- origsrc/keybinder-0.2.2/python-keybinder/Makefile.am	2010-04-25 00:09:47.000000000 +0900
+++ src/keybinder-0.2.2/python-keybinder/Makefile.am	2012-04-27 18:42:51.363254000 +0900
@@ -5,7 +5,7 @@ AM_CFLAGS = -Wall
 _keybinderdir = $(pyexecdir)/keybinder
 _keybinder_LTLIBRARIES = _keybinder.la
 _keybinder_la_CPPFLAGS = $(X_CFLAGS) $(PYEXTRAFLAGS)
-_keybinder_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_keybinder $(X_LDFLAGS)
+_keybinder_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_keybinder $(X_LDFLAGS) -no-undefined
 _keybinder_la_LIBADD = ../libkeybinder/libkeybinder.la $(PYGTK_LIBS)
 _keybinder_la_SOURCES = _keybindermodule.c
 
