--- origsrc/harp-0.6.0/util.c	2015-07-25 02:35:12.000000000 +0900
+++ src/harp-0.6.0/util.c	2016-12-20 13:58:19.524591500 +0900
@@ -15,6 +15,10 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#if defined(__CYGWIN__)
+#define _GNU_SOURCE
+#endif
+
 #include "util.h"
 #include "defs.h"
 #include "dbact.h"
@@ -652,4 +656,3 @@ void db_safe(char *str, const char *data
 	}
 	*str=0;
 }
-
