--- origsrc/gf256/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/gf256/Makefile.am	2024-10-11 23:38:11.101343200 +0900
@@ -0,0 +1,10 @@
+
+include_HEADERS = gf256.h
+
+lib_LTLIBRARIES = libgf256.la
+libgf256_la_CXXFLAGS = -msse4.1
+libgf256_la_SOURCES = gf256.cpp
+libgf256_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/gf256/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/gf256/configure.ac	2024-10-11 23:38:11.101343200 +0900
@@ -0,0 +1,33 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([gf256], [2])
+AC_CONFIG_SRCDIR([gf256.cpp])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([fcntl.h stdint.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+AC_C_INLINE
+AC_C_RESTRICT
+AC_TYPE_UINT16_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+AC_TYPE_UINT8_T
+
+# Checks for library functions.
+AC_CHECK_FUNCS([memset])
+
+AC_CONFIG_FILES([gf256.pc Makefile])
+AC_OUTPUT
--- origsrc/gf256/gf256.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/gf256/gf256.pc.in	2024-10-11 23:38:11.101343200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: gf256
+Description: Fast 8-bit Galois Field Math in C
+Version: @VERSION@
+Libs: -L${libdir} -lgf256
+Cflags: -I${includedir}
