--- origsrc/tippecanoe-1.30.0/Makefile	2018-06-20 07:10:28.000000000 +0900
+++ src/tippecanoe-1.30.0/Makefile	2018-07-09 15:55:28.386878000 +0900
@@ -7,7 +7,7 @@ SHELL = /bin/bash
 CC := $(CC)
 CXX := $(CXX)
 CFLAGS := $(CFLAGS)
-CXXFLAGS := $(CXXFLAGS) -std=c++11
+CXXFLAGS := $(CXXFLAGS)
 LDFLAGS := $(LDFLAGS)
 WARNING_FLAGS := -Wall -Wshadow -Wsign-compare -Wextra -Wunreachable-code -Wuninitialized -Wshadow
 RELEASE_FLAGS := -O3 -DNDEBUG
