--- origsrc/ghostz-1.0.2/Makefile	2018-08-08 13:45:55.000000000 +0900
+++ src/ghostz-1.0.2/Makefile	2025-12-25 09:48:56.857997000 +0900
@@ -1,6 +1,6 @@
 LAST_CC=g++
-CC=gcc
-CXX=g++
+CC?=gcc
+CXX?=g++
 COMMON_FLAGS=-Wall -Wextra -pedantic -fopenmp -O3 -g
 CC_FLAGS=${COMMON_FLAGS} 
 LDLIBS	=-lm -fopenmp
@@ -9,6 +9,7 @@ ifeq ($(PROFILE), Yes)
 CC_FLAGS=${COMMON_FLAGS} -g
 endif
 
+CC_FLAGS+= ${CFLAGS}
 
 C_SRC=./ext/karlin/src/karlin.c 
 
