--- origsrc/cabocha-0.68/configure.in	2014-03-02 11:26:01.000000000 +0900
+++ src/cabocha-0.68/configure.in	2014-03-30 17:20:54.772404200 +0900
@@ -34,7 +34,10 @@ AC_C_CONST
 AC_C_BIGENDIAN
 AC_CHECK_HEADERS(string.h stdlib.h unistd.h fcntl.h \
                  sys/stat.h sys/mman.h sys/times.h sys/types.h dirent.h \
-                 ctype.h sys/types.h io.h windows.h pthread.h)
+                 ctype.h sys/types.h io.h pthread.h)
+if test "$host_os" != "cygwin"; then
+   AC_CHECK_HEADERS(windows.h)
+fi
 AC_TYPE_OFF_T
 AC_TYPE_SIZE_T
 AC_CHECK_SIZEOF(char)
@@ -132,7 +135,7 @@ AC_SUBST(MECAB_CFLAGS)
 dnl
 dnl check C++ features
 dnl
-AC_DEFUN(CHECK_CXX_STL, [
+AC_DEFUN([CHECK_CXX_STL], [
   AC_MSG_CHECKING(if ${CXX-c++} supports stl <$1> (required))
   AC_TRY_COMPILE(
 	[
@@ -323,8 +326,8 @@ AC_SUBST(CABOCHA_DEFAULT_RC)
 AC_SUBST(datarootdir)
 
 CABOCHA_DEFAULT_RC="$sysconfdir/cabocharc"
-CABOCHA_TXTMODEL_LIST=`(cd model; echo *.txt)`
-CABOCHA_MODEL_LIST=`(cd model; echo *.txt | sed s/.txt/.model/g)`
+CABOCHA_TXTMODEL_LIST=`(cd ${srcdir}/model; echo *.txt)`
+CABOCHA_MODEL_LIST=`(cd ${srcdir}/model; echo *.txt | sed s/.txt/.model/g)`
 AC_SUBST(CABOCHA_TXTMODEL_LIST)
 AC_SUBST(CABOCHA_MODEL_LIST)
 
--- origsrc/cabocha-0.68/model/Makefile.am	2012-09-30 01:43:41.000000000 +0900
+++ src/cabocha-0.68/model/Makefile.am	2014-03-30 17:21:17.494083900 +0900
@@ -6,4 +6,4 @@ SUFFIXES = .txt .model
 CHARSET = @CHARSET@
 
 @CABOCHA_MODEL_LIST@:
-	../src/cabocha-model-index -f UTF8 -t $(CHARSET) $*.txt $@
+	../src/cabocha-model-index -f UTF8 -t $(CHARSET) $(srcdir)/$*.txt $@
--- origsrc/cabocha-0.68/src/winmain.h	2012-09-30 01:42:42.000000000 +0900
+++ src/cabocha-0.68/src/winmain.h	2014-03-30 17:21:55.226505800 +0900
@@ -3,7 +3,7 @@
 //  $Id: libcabocha.cpp 50 2009-05-03 08:25:36Z taku-ku $;
 //
 //  Copyright(C) 2001-2008 Taku Kudo <taku@chasen.org>
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 
 #include <windows.h>
 #include <string>
