--- origsrc/airrac-1.00.14/airrac-config.cmake.in	2025-12-25 07:45:49.000000000 +0900
+++ src/airrac-1.00.14/airrac-config.cmake.in	2025-12-29 18:47:08.918295400 +0900
@@ -20,5 +20,5 @@ set (AIRRAC_LIBRARY_DIRS "@PACKAGE_AIRRA
 include ("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@-library-depends.cmake")
 
 # These are the AirRAC IMPORTED targets, created by airrac-library-depends.cmake
-set (AIRRAC_LIBRARIES airraclib)
+set (AIRRAC_LIBRARIES airrac)
 set (AIRRAC_EXECUTABLES airrac)
--- origsrc/airrac-1.00.14/config/project_config_embeddable.cmake	2025-12-25 07:45:49.000000000 +0900
+++ src/airrac-1.00.14/config/project_config_embeddable.cmake	2025-12-29 18:47:33.530828100 +0900
@@ -439,7 +439,7 @@ macro (get_git)
   message (STATUS "Requires Git without specifying any version")
 
   find_package (Git)
-  if (Git_FOUND)
+  if (Git_FOUND AND EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/.git)
     Git_WC_INFO (${CMAKE_CURRENT_SOURCE_DIR} PROJ)
     set (GIT_REVISION ${PROJ_WC_REVISION_HASH})
     message (STATUS "Current Git revision name: ${PROJ_WC_REVISION_NAME}")
