--- origsrc/airrac-airrac-1.00.2/airrac-config.cmake.in	2019-01-18 06:15:44.000000000 +0900
+++ src/airrac-airrac-1.00.2/airrac-config.cmake.in	2019-05-26 20:02:52.118110000 +0900
@@ -18,5 +18,5 @@ set (AIRRAC_LIBRARY_DIRS "@AIRRAC_LIB_DI
 include ("@AIRRAC_CMAKE_DIR@/airrac-library-depends.cmake")
 
 # These are the AirRAC IMPORTED targets, created by airrac-library-depends.cmake
-set (AIRRAC_LIBRARIES airraclib)
+set (AIRRAC_LIBRARIES airrac)
 set (AIRRAC_EXECUTABLES airrac)
--- origsrc/airrac-airrac-1.00.2/config/project_config_embeddable.cmake	2019-01-18 06:15:44.000000000 +0900
+++ src/airrac-airrac-1.00.2/config/project_config_embeddable.cmake	2019-05-26 20:04:07.134700200 +0900
@@ -1507,7 +1507,7 @@ endmacro (module_set_name)
 macro (add_modules)
   set (_embedded_components ${ARGV})
   set (LIB_DEPENDENCY_EXPORT ${PROJECT_NAME}-library-depends)
-  set (LIB_DEPENDENCY_EXPORT_PATH "${INSTALL_DATA_DIR}/${PROJECT_NAME}/CMake")
+  set (LIB_DEPENDENCY_EXPORT_PATH "${INSTALL_LIB_DIR}/cmake/${PROJECT_NAME}")
   #
   foreach (_embedded_comp ${_embedded_components})
     #
@@ -1798,7 +1798,8 @@ macro (module_library_add_specific
 	# Install the library in the standard location
 	install (TARGETS ${_lib_target}
       EXPORT ${LIB_DEPENDENCY_EXPORT}
-      LIBRARY DESTINATION "${INSTALL_LIB_DIR}" COMPONENT runtime)
+      RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
+      ARCHIVE DESTINATION "${INSTALL_LIB_DIR}")
   endif ("${_lib_prefix}" STREQUAL "py")
   
   # Register, for reporting purpose, the list of libraries to be built
@@ -1995,7 +1996,7 @@ endmacro (module_script_add)
 # can refer to it (for libraries, header files and binaries)
 macro (module_export_install)
   install (EXPORT ${LIB_DEPENDENCY_EXPORT}
-    DESTINATION "${INSTALL_DATA_DIR}/${PACKAGE}/CMake" COMPONENT devel)
+    DESTINATION "${INSTALL_LIB_DIR}/cmake/${PACKAGE}" COMPONENT devel)
 endmacro (module_export_install)
 
 
