diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/sbw-core-2.11.1/CMakeLists.txt src/sbw-core-2.11.1/CMakeLists.txt
--- origsrc/sbw-core-2.11.1/CMakeLists.txt	2014-12-31 21:08:47.000000000 +0900
+++ src/sbw-core-2.11.1/CMakeLists.txt	2015-11-08 22:16:33.233665300 +0900
@@ -317,3 +317,10 @@ else()
 message(STATUS "  Build SBW static library        = no")
 endif()
 message(STATUS "")
\ No newline at end of file
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/sbw-core.pc.in ${CMAKE_BINARY_DIR}/sbw-core.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/sbw-core.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/sbw-core-2.11.1/SBWBroker/CMakeLists.txt src/sbw-core-2.11.1/SBWBroker/CMakeLists.txt
--- origsrc/sbw-core-2.11.1/SBWBroker/CMakeLists.txt	2014-12-31 22:58:21.000000000 +0900
+++ src/sbw-core-2.11.1/SBWBroker/CMakeLists.txt	2015-11-08 22:14:35.149691300 +0900
@@ -54,7 +54,7 @@ if (MSVC OR USING_INTEL)
 endif(MSVC OR USING_INTEL)
 
 add_executable (${BROKER_NAME} ${EXECUTABLE_FLAGS} ${SBW_SOURCES} )
-target_link_libraries(${BROKER_NAME} ${SBW_LIBS} SBW-static)
+target_link_libraries(${BROKER_NAME} ${SBW_LIBS} SBW)
 
 INSTALL(TARGETS ${BROKER_NAME}
 	RUNTIME DESTINATION bin
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/sbw-core-2.11.1/sbw-core.pc.in src/sbw-core-2.11.1/sbw-core.pc.in
--- origsrc/sbw-core-2.11.1/sbw-core.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/sbw-core-2.11.1/sbw-core.pc.in	2015-11-08 22:17:08.130543800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: sbw-core
+Description: C++ Broker library
+Version: @PACKAGE_VERSION@
+Libs: -L${libdir} -lSBW
+Cflags: -I${includedir}
