--- origsrc/rlog-1.4/configure.ac	2008-06-15 14:48:32.000000000 +0900
+++ src/rlog-1.4/configure.ac	2010-12-06 10:42:57.473934400 +0900
@@ -49,13 +49,13 @@ AC_ENABLE_SHARED(yes)
 
 dnl set the following to yes, if you want to create static
 dnl libtool-libraries, too.
-AC_ENABLE_STATIC(no)
+#AC_ENABLE_STATIC(no)
 
 dnl checks for programs.
 dnl first check for c/c++ compilers
 AC_LANG([C++])
 AC_PROG_CXX
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 dnl CXXFLAGS="$NOOPT_CXXFLAGS" dnl __kdevelop[noopt]__
 dnl CFLAGS="$NOOPT_CFLAGS" dnl __kdevelop[noopt]__
--- origsrc/rlog-1.4/docs/Makefile.am	2005-06-02 21:25:55.000000000 +0900
+++ src/rlog-1.4/docs/Makefile.am	2010-12-06 11:44:14.300438300 +0900
@@ -32,7 +32,7 @@ all-local: html/index.html
 
 install-data-local: all-local
 	-echo installing to ${DESTDIR}${pkgdocdir}/html
-	/bin/sh ../mkinstalldirs ${DESTDIR}${pkgdocdir}/html
-	install html/* ${DESTDIR}${pkgdocdir}/html
+	/bin/sh $(top_srcdir)/mkinstalldirs ${DESTDIR}${pkgdocdir}/html
+	install $(srcdir)/html/* ${DESTDIR}${pkgdocdir}/html
 
 
--- origsrc/rlog-1.4/rlog/Makefile.am	2007-12-19 18:18:51.000000000 +0900
+++ src/rlog-1.4/rlog/Makefile.am	2010-12-06 11:20:56.182092500 +0900
@@ -18,7 +18,7 @@ librlog_la_SOURCES = \
     StdioNode.cpp \
     SyslogNode.cpp
 
-librlog_la_LDFLAGS = -version-info $(LIBRLOG_VERSION) $(USER_LDFLAGS)
+librlog_la_LDFLAGS = -version-info $(LIBRLOG_VERSION) $(USER_LDFLAGS) -no-undefined
 
 librlogincludedir=$(includedir)/rlog
 librloginclude_HEADERS = \
@@ -40,7 +40,8 @@ librloginclude_HEADERS = \
     SyslogNode.h
 
 
-noinst_PROGRAMS = test testlog
+TESTS = test testlog
+check_PROGRAMS = test testlog
 
 test_SOURCES = test.cpp
 test_LDADD = librlog.la 
