--- origsrc/Atlas-C++-0.6.2/Atlas/Codecs/Makefile.am	2009-10-26 08:18:36.000000000 +0900
+++ src/Atlas-C++-0.6.2/Atlas/Codecs/Makefile.am	2010-12-30 18:34:51.933483600 +0900
@@ -3,7 +3,7 @@ INCLUDES = -I$(top_srcdir)
 
 lib_LTLIBRARIES = libAtlasCodecs-0.6.la
 
-libAtlasCodecs_0_6_la_LDFLAGS = $(INTERFACE_VERSION)
+libAtlasCodecs_0_6_la_LDFLAGS = $(INTERFACE_VERSION) -no-undefined
 
 libAtlasCodecs_0_6_la_LIBADD = $(top_builddir)/Atlas/libAtlas-0.6.la
 
--- origsrc/Atlas-C++-0.6.2/Atlas/Filters/Makefile.am	2009-10-26 08:18:36.000000000 +0900
+++ src/Atlas-C++-0.6.2/Atlas/Filters/Makefile.am	2010-12-30 18:34:51.933483600 +0900
@@ -3,7 +3,7 @@ INCLUDES = -I$(top_srcdir) -I$(top_build
 
 lib_LTLIBRARIES = libAtlasFilters-0.6.la
 
-libAtlasFilters_0_6_la_LDFLAGS = $(INTERFACE_VERSION)
+libAtlasFilters_0_6_la_LDFLAGS = $(INTERFACE_VERSION) -no-undefined
 
 libAtlasFilters_0_6_la_LIBADD = $(top_builddir)/Atlas/libAtlas-0.6.la \
 	$(OTHER_LIBS)
--- origsrc/Atlas-C++-0.6.2/Atlas/Funky/Makefile.am	2009-10-26 08:18:36.000000000 +0900
+++ src/Atlas-C++-0.6.2/Atlas/Funky/Makefile.am	2010-12-30 18:34:51.944280200 +0900
@@ -3,7 +3,7 @@ INCLUDES = -I$(top_srcdir)
 
 lib_LTLIBRARIES = libAtlasFunky-0.6.la
 
-libAtlasFunky_0_6_la_LDFLAGS = $(INTERFACE_VERSION)
+libAtlasFunky_0_6_la_LDFLAGS = $(INTERFACE_VERSION) -no-undefined
 
 libAtlasFunky_0_6_la_SOURCES = \
         FEncoder.cpp
--- origsrc/Atlas-C++-0.6.2/Atlas/Makefile.am	2009-10-26 08:18:36.000000000 +0900
+++ src/Atlas-C++-0.6.2/Atlas/Makefile.am	2010-12-30 18:34:51.944280200 +0900
@@ -4,7 +4,7 @@ INCLUDES = -I$(top_srcdir)
 
 lib_LTLIBRARIES = libAtlas-0.6.la
 
-libAtlas_0_6_la_LDFLAGS = $(INTERFACE_VERSION)
+libAtlas_0_6_la_LDFLAGS = $(INTERFACE_VERSION) -no-undefined
 
 libAtlas_0_6_la_SOURCES = \
 	Bridge.cpp Codec.cpp Negotiate.cpp EncoderBase.cpp Exception.cpp \
--- origsrc/Atlas-C++-0.6.2/Atlas/Message/Makefile.am	2009-10-26 08:18:36.000000000 +0900
+++ src/Atlas-C++-0.6.2/Atlas/Message/Makefile.am	2010-12-30 18:34:51.944280200 +0900
@@ -3,7 +3,7 @@ INCLUDES = -I$(top_srcdir)
 
 lib_LTLIBRARIES = libAtlasMessage-0.6.la
 
-libAtlasMessage_0_6_la_LDFLAGS = $(INTERFACE_VERSION)
+libAtlasMessage_0_6_la_LDFLAGS = $(INTERFACE_VERSION) -no-undefined
 
 libAtlasMessage_0_6_la_LIBADD = $(top_builddir)/Atlas/libAtlas-0.6.la
 
--- origsrc/Atlas-C++-0.6.2/Atlas/Net/Makefile.am	2009-10-26 08:18:36.000000000 +0900
+++ src/Atlas-C++-0.6.2/Atlas/Net/Makefile.am	2010-12-30 18:34:51.955076800 +0900
@@ -3,7 +3,7 @@ INCLUDES = -I$(top_srcdir)
 
 lib_LTLIBRARIES = libAtlasNet-0.6.la
 
-libAtlasNet_0_6_la_LDFLAGS = $(INTERFACE_VERSION)
+libAtlasNet_0_6_la_LDFLAGS = $(INTERFACE_VERSION) -no-undefined
 
 libAtlasNet_0_6_la_LIBADD = $(top_builddir)/Atlas/Codecs/libAtlasCodecs-0.6.la \
     $(top_builddir)/Atlas/libAtlas-0.6.la
--- origsrc/Atlas-C++-0.6.2/Atlas/Objects/Makefile.am	2009-10-26 08:18:36.000000000 +0900
+++ src/Atlas-C++-0.6.2/Atlas/Objects/Makefile.am	2010-12-30 18:34:51.955076800 +0900
@@ -3,7 +3,7 @@ INCLUDES = -I$(top_srcdir)
 
 lib_LTLIBRARIES = libAtlasObjects-0.6.la
 
-libAtlasObjects_0_6_la_LDFLAGS = $(INTERFACE_VERSION)
+libAtlasObjects_0_6_la_LDFLAGS = $(INTERFACE_VERSION) -no-undefined
 
 libAtlasObjects_0_6_la_LIBADD = $(top_builddir)/Atlas/libAtlas-0.6.la \
     $(top_builddir)/Atlas/Message/libAtlasMessage-0.6.la \
--- origsrc/Atlas-C++-0.6.2/Doxyfile	2009-10-26 08:18:36.000000000 +0900
+++ src/Atlas-C++-0.6.2/Doxyfile	2010-12-30 18:34:51.965873400 +0900
@@ -176,7 +176,7 @@ TAB_SIZE             = 8
 # directories like "/usr/src/myproject". Separate the files or directories 
 # with spaces.
 
-INPUT                = .
+INPUT                = ../src/Atlas-C++-0.6.2/
 
 # If the value of the INPUT tag contains directories, you can use the 
 # FILE_PATTERNS tag to specify one or more wildcard pattern (like *.cpp 
@@ -256,7 +256,7 @@ HTML_HEADER          =
 # each generated HTML page. If it is left blank doxygen will generate a 
 # standard footer.
 
-HTML_FOOTER          = footer.html
+HTML_FOOTER          = ../src/Atlas-C++-0.6.2/footer.html
 
 # The HTML_STYLESHEET tag can be used to specify a user defined cascading
 # style sheet that is used by each HTML page. It can be used to 
--- origsrc/Atlas-C++-0.6.2/Makefile.am	2009-10-26 08:18:36.000000000 +0900
+++ src/Atlas-C++-0.6.2/Makefile.am	2010-12-30 18:34:51.965873400 +0900
@@ -8,5 +8,5 @@ EXTRA_DIST = HACKING ROADMAP Doxyfile fo
 
 docs:
 	@echo "running doxygen..."
-	@doxygen Doxyfile
+	@doxygen $(top_srcdir)/Doxyfile
 	@echo "documentation is in doc/."
--- origsrc/Atlas-C++-0.6.2/configure.ac	2009-11-14 00:23:38.000000000 +0900
+++ src/Atlas-C++-0.6.2/configure.ac	2010-12-30 18:35:38.176062200 +0900
@@ -22,8 +22,7 @@ AC_SUBST(INTERFACE_VERSION)
 
 dnl Test for GNU libtool
 
-AC_DISABLE_STATIC
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 dnl Test for C++ compiler
 
