--- origsrc/cmake-modules-webos-versions-1.0.0-rc6/CMakeLists.txt	2013-10-04 08:05:59.000000000 +0900
+++ src/cmake-modules-webos-versions-1.0.0-rc6/CMakeLists.txt	2018-09-02 20:01:51.896040000 +0900
@@ -35,8 +35,12 @@ webos_component(${VERSION_ARGS})
 
 # Install the modules directly into the /Modules directory of the currently running cmake instance.
 # This avoids users having to add a path to CMAKE_MODULES_PATH before they can include webOS/webOS
-install(DIRECTORY webOS DESTINATION ${CMAKE_ROOT}/Modules FILES_MATCHING PATTERN "*" PATTERN ".*" EXCLUDE)
-install(FILES REFERENCE.md DESTINATION ${CMAKE_ROOT}/Modules/webOS)
+if(CYGWIN)
+    install(DIRECTORY webOS DESTINATION /usr/lib/cmake FILES_MATCHING PATTERN "*" PATTERN ".*" EXCLUDE)
+else()
+    install(DIRECTORY webOS DESTINATION ${CMAKE_ROOT}/Modules FILES_MATCHING PATTERN "*" PATTERN ".*" EXCLUDE)
+    install(FILES REFERENCE.md DESTINATION ${CMAKE_ROOT}/Modules/webOS)
+endif()
 
 #
 # Tests - uncomment the appropriate lines to test the modules.
