--- origsrc/ophidia-primitives-1.8.0/src/Makefile.am	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/Makefile.am	2024-09-24 10:48:56.055253600 +0900
@@ -39,479 +39,479 @@ endif
 lib_LTLIBRARIES = liboph_debug.la
 liboph_debug_la_SOURCES = oph_debug.c oph_debug.h
 liboph_debug_la_CFLAGS = -prefer-pic $(CFLAGS)
-liboph_debug_la_LDFLAGS = -avoid-version -static $(LIBS)
+liboph_debug_la_LDFLAGS = -no-undefined -avoid-version $(LIBS)
 
 lib_LTLIBRARIES += liboph_core.la
 liboph_core_la_SOURCES = oph_core.c oph_core.h
 liboph_core_la_CFLAGS = -prefer-pic $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_core_la_LDFLAGS = -avoid-version -static $(LIBS)
+liboph_core_la_LDFLAGS = -no-undefined -avoid-version $(LIBS)
 
 lib_LTLIBRARIES += liboph_core_subset.la
 liboph_core_subset_la_SOURCES = oph_core_subset.c oph_core_subset.h
 liboph_core_subset_la_CFLAGS = -prefer-pic $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_core_subset_la_LDFLAGS = -avoid-version -static $(LIBS)
+liboph_core_subset_la_LDFLAGS = -no-undefined -avoid-version $(LIBS)
 
 # Primitives operating on IDs
 
 lib_LTLIBRARIES += liboph_id_to_index.la
 liboph_id_to_index_la_SOURCES = oph_id_to_index.c oph_id_to_index.h
 liboph_id_to_index_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_id_to_index_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug
+liboph_id_to_index_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug
 
 lib_LTLIBRARIES += liboph_id3.la
 liboph_id3_la_SOURCES = oph_id3.c oph_id3.h
 liboph_id3_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_id3_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug
+liboph_id3_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug
 
 lib_LTLIBRARIES += liboph_id_of_subset.la
 liboph_id_of_subset_la_SOURCES = oph_id_of_subset.c oph_id_of_subset.h
 liboph_id_of_subset_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_id_of_subset_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core -loph_core_subset
+liboph_id_of_subset_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core -loph_core_subset
 
 # Primitives operating on measures
 
 lib_LTLIBRARIES += liboph_aggregate_stats.la
 liboph_aggregate_stats_la_SOURCES = oph_aggregate_stats.c oph_aggregate_stats.h
 liboph_aggregate_stats_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_aggregate_stats_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_aggregate_stats_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_aggregate_stats_partial.la
 liboph_aggregate_stats_partial_la_SOURCES = oph_aggregate_stats_partial.c oph_aggregate_stats_partial.h
 liboph_aggregate_stats_partial_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_aggregate_stats_partial_la_LDFLAGS = -avoid-version -shared $(LIBS) -lm -L. -loph_debug -loph_core
+liboph_aggregate_stats_partial_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -lm -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_aggregate_stats_final.la
 liboph_aggregate_stats_final_la_SOURCES = oph_aggregate_stats_final.c oph_aggregate_stats_final.h
 liboph_aggregate_stats_final_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_aggregate_stats_final_la_LDFLAGS = -avoid-version -shared $(LIBS) -lm -L. -loph_debug -loph_core
+liboph_aggregate_stats_final_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -lm -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_append.la
 liboph_append_la_SOURCES = oph_append.c oph_append.h
 liboph_append_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_append_la_LDFLAGS = -avoid-version -shared $(LIBS) -lm -L. -loph_debug -loph_core
+liboph_append_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -lm -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_compress.la
 liboph_compress_la_SOURCES = oph_compress.c oph_compress.h
 liboph_compress_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_compress_la_LDFLAGS = -avoid-version -shared $(LIBS) -lm -lz -L. -loph_debug -loph_core
+liboph_compress_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -lm -lz -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_uncompress.la
 liboph_uncompress_la_SOURCES = oph_uncompress.c oph_uncompress.h
 liboph_uncompress_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_uncompress_la_LDFLAGS = -avoid-version -shared $(LIBS) -lz -L. -loph_debug -loph_core
+liboph_uncompress_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -lz -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_aggregate_operator.la
 liboph_aggregate_operator_la_SOURCES = oph_aggregate_operator.c oph_aggregate_operator.h
 liboph_aggregate_operator_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_aggregate_operator_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_aggregate_operator_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_normalize.la
 liboph_normalize_la_SOURCES = oph_normalize.c oph_normalize.h
 liboph_normalize_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_normalize_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_normalize_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_reduce.la
 liboph_reduce_la_SOURCES = oph_reduce.c oph_reduce.h
 liboph_reduce_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_reduce_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_reduce_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_reduce2.la
 liboph_reduce2_la_SOURCES = oph_reduce2.c oph_reduce2.h
 liboph_reduce2_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_reduce2_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_reduce2_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_reduce3.la
 liboph_reduce3_la_SOURCES = oph_reduce3.c oph_reduce3.h
 liboph_reduce3_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_reduce3_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_reduce3_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_dump.la
 liboph_dump_la_SOURCES = oph_dump.c oph_dump.h
 liboph_dump_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_dump_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_dump_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_convert_d.la
 liboph_convert_d_la_SOURCES = oph_convert_d.c oph_convert_d.h
 liboph_convert_d_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_convert_d_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_convert_d_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_convert_l.la
 liboph_convert_l_la_SOURCES = oph_convert_l.c oph_convert_l.h
 liboph_convert_l_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_convert_l_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_convert_l_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_count_array.la
 liboph_count_array_la_SOURCES = oph_count_array.c oph_count_array.h
 liboph_count_array_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_count_array_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_count_array_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_get_subarray.la
 liboph_get_subarray_la_SOURCES = oph_get_subarray.c oph_get_subarray.h
 liboph_get_subarray_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_get_subarray_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_get_subarray_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_get_subarray2.la
 liboph_get_subarray2_la_SOURCES = oph_get_subarray2.c oph_get_subarray2.h
 liboph_get_subarray2_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_get_subarray2_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core -loph_core_subset
+liboph_get_subarray2_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core -loph_core_subset
 
 lib_LTLIBRARIES += liboph_get_subarray3.la
 liboph_get_subarray3_la_SOURCES = oph_get_subarray3.c oph_get_subarray3.h
 liboph_get_subarray3_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_get_subarray3_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core -loph_core_subset
+liboph_get_subarray3_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core -loph_core_subset
 
 lib_LTLIBRARIES += liboph_sum_scalar.la
 liboph_sum_scalar_la_SOURCES = oph_sum_scalar.c oph_sum_scalar.h
 liboph_sum_scalar_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_sum_scalar_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_sum_scalar_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_mul_scalar.la
 liboph_mul_scalar_la_SOURCES = oph_mul_scalar.c oph_mul_scalar.h
 liboph_mul_scalar_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_mul_scalar_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_mul_scalar_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_sum_scalar2.la
 liboph_sum_scalar2_la_SOURCES = oph_sum_scalar2.c oph_sum_scalar2.h
 liboph_sum_scalar2_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_sum_scalar2_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_sum_scalar2_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_mul_scalar2.la
 liboph_mul_scalar2_la_SOURCES = oph_mul_scalar2.c oph_mul_scalar2.h
 liboph_mul_scalar2_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_mul_scalar2_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_mul_scalar2_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_affine.la
 liboph_affine_la_SOURCES = oph_affine.c oph_affine.h oph_debug.c oph_debug.h oph_core.c oph_core.h
 liboph_affine_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_affine_la_LDFLAGS = -avoid-version -shared $(LIBS)
+liboph_affine_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS)
 
 lib_LTLIBRARIES += liboph_get_index_array.la
 liboph_get_index_array_la_SOURCES = oph_get_index_array.c oph_get_index_array.h
 liboph_get_index_array_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_get_index_array_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_get_index_array_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_to_bin.la
 liboph_to_bin_la_SOURCES = oph_to_bin.c oph_to_bin.h
 liboph_to_bin_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_to_bin_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_to_bin_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_compare.la
 liboph_compare_la_SOURCES = oph_compare.c oph_compare.h
 liboph_compare_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_compare_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug
+liboph_compare_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug
 
 lib_LTLIBRARIES += liboph_extract.la
 liboph_extract_la_SOURCES = oph_extract.c oph_extract.h
 liboph_extract_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_extract_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_extract_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_find.la
 liboph_find_la_SOURCES = oph_find.c oph_find.h
 liboph_find_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_find_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_find_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_interlace.la
 liboph_interlace_la_SOURCES = oph_interlace.c oph_interlace.h
 liboph_interlace_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_interlace_la_LDFLAGS = -avoid-version -shared $(LIBS) -lm -L. -loph_debug -loph_core
+liboph_interlace_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -lm -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_interlace2.la
 liboph_interlace2_la_SOURCES = oph_interlace2.c oph_interlace2.h
 liboph_interlace2_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_interlace2_la_LDFLAGS = -avoid-version -shared $(LIBS) -lm -L. -loph_debug -loph_core
+liboph_interlace2_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -lm -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_math.la
 liboph_math_la_SOURCES = oph_math.c oph_math.h oph_core_math.c oph_core_math.h
 liboph_math_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_math_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_math_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_padding.la
 liboph_padding_la_SOURCES = oph_padding.c oph_padding.h
 liboph_padding_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_padding_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_padding_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_permute.la
 liboph_permute_la_SOURCES = oph_permute.c oph_permute.h
 liboph_permute_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_permute_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_permute_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_reverse.la
 liboph_reverse_la_SOURCES = oph_reverse.c oph_reverse.h
 liboph_reverse_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_reverse_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_reverse_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_roll_up.la
 liboph_roll_up_la_SOURCES = oph_roll_up.c oph_roll_up.h
 liboph_roll_up_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_roll_up_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_roll_up_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_rotate.la
 liboph_rotate_la_SOURCES = oph_rotate.c oph_rotate.h
 liboph_rotate_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_rotate_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_rotate_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_sequence.la
 liboph_sequence_la_SOURCES = oph_sequence.c oph_sequence.h
 liboph_sequence_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_sequence_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_sequence_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_size_array.la
 liboph_size_array_la_SOURCES = oph_size_array.c oph_size_array.h
 liboph_size_array_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_size_array_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_size_array_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_shift.la
 liboph_shift_la_SOURCES = oph_shift.c oph_shift.h
 liboph_shift_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_shift_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_shift_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_value_to_bin.la
 liboph_value_to_bin_la_SOURCES = oph_value_to_bin.c oph_value_to_bin.h
 liboph_value_to_bin_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_value_to_bin_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_value_to_bin_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_operation_array.la
 liboph_operation_array_la_SOURCES = oph_operation_array.c oph_operation_array.h
 liboph_operation_array_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_operation_array_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_operation_array_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_sum_array.la
 liboph_sum_array_la_SOURCES = oph_sum_array.c oph_sum_array.h
 liboph_sum_array_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_sum_array_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_sum_array_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_mul_array.la
 liboph_mul_array_la_SOURCES = oph_mul_array.c oph_mul_array.h
 liboph_mul_array_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_mul_array_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_mul_array_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_mask_array.la
 liboph_mask_array_la_SOURCES = oph_mask_array.c oph_mask_array.h
 liboph_mask_array_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_mask_array_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_mask_array_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_sub_array.la
 liboph_sub_array_la_SOURCES = oph_sub_array.c oph_sub_array.h
 liboph_sub_array_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_sub_array_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_sub_array_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_div_array.la
 liboph_div_array_la_SOURCES = oph_div_array.c oph_div_array.h
 liboph_div_array_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_div_array_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_div_array_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_abs_array.la
 liboph_abs_array_la_SOURCES = oph_abs_array.c oph_abs_array.h
 liboph_abs_array_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_abs_array_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_abs_array_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_arg_array.la
 liboph_arg_array_la_SOURCES = oph_arg_array.c oph_arg_array.h
 liboph_arg_array_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_arg_array_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_arg_array_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_max_array.la
 liboph_max_array_la_SOURCES = oph_max_array.c oph_max_array.h
 liboph_max_array_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_max_array_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_max_array_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_min_array.la
 liboph_min_array_la_SOURCES = oph_min_array.c oph_min_array.h
 liboph_min_array_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_min_array_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_min_array_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_arg_max_array.la
 liboph_arg_max_array_la_SOURCES = oph_arg_max_array.c oph_arg_max_array.h
 liboph_arg_max_array_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_arg_max_array_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_arg_max_array_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_arg_min_array.la
 liboph_arg_min_array_la_SOURCES = oph_arg_min_array.c oph_arg_min_array.h
 liboph_arg_min_array_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_arg_min_array_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_arg_min_array_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_operator.la
 liboph_operator_la_SOURCES = oph_operator.c oph_operator.h
 liboph_operator_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_operator_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_operator_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_operator_array.la
 liboph_operator_array_la_SOURCES = oph_operator_array.c oph_operator_array.h
 liboph_operator_array_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_operator_array_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_operator_array_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_cast.la
 liboph_cast_la_SOURCES = oph_cast.c oph_cast.h
 liboph_cast_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_cast_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_cast_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_concat.la
 liboph_concat_la_SOURCES = oph_concat.c oph_concat.h
 liboph_concat_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_concat_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_concat_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_concat2.la
 liboph_concat2_la_SOURCES = oph_concat2.c oph_concat2.h
 liboph_concat2_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_concat2_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_concat2_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_moving_avg.la
 liboph_moving_avg_la_SOURCES = oph_moving_avg.c oph_moving_avg.h
 liboph_moving_avg_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_moving_avg_la_LDFLAGS = -avoid-version -shared $(LIBS) -lm -L. -loph_debug -loph_core
+liboph_moving_avg_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -lm -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_accumulate.la
 liboph_accumulate_la_SOURCES = oph_accumulate.c oph_accumulate.h
 liboph_accumulate_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_accumulate_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_accumulate_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_deaccumulate.la
 liboph_deaccumulate_la_SOURCES = oph_deaccumulate.c oph_deaccumulate.h
 liboph_deaccumulate_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_deaccumulate_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_deaccumulate_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_replace.la
 liboph_replace_la_SOURCES = oph_replace.c oph_replace.h
 liboph_replace_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_replace_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_replace_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_ccluster_kcluster.la
 liboph_ccluster_kcluster_la_SOURCES = oph_ccluster_kcluster.c oph_ccluster_kcluster.h cluster.c cluster.h
 liboph_ccluster_kcluster_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_ccluster_kcluster_la_LDFLAGS = -avoid-version -shared $(LIBS) -lm -L. -loph_debug -loph_core
+liboph_ccluster_kcluster_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -lm -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_quantize.la
 liboph_quantize_la_SOURCES = oph_quantize.c oph_quantize.h
 liboph_quantize_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_quantize_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_quantize_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_expand.la
 liboph_expand_la_SOURCES = oph_expand.c oph_expand.h
 liboph_expand_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_expand_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_expand_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_extend.la
 liboph_extend_la_SOURCES = oph_extend.c oph_extend.h
 liboph_extend_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS)
-liboph_extend_la_LDFLAGS = -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
+liboph_extend_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) -L. -loph_debug -loph_core
 
 if MATHEVAL_SUPPORT
 lib_LTLIBRARIES += liboph_predicate.la
 liboph_predicate_la_SOURCES = oph_predicate.c oph_predicate.h oph_core_matheval.c oph_core_matheval.h
 liboph_predicate_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(MATHEVAL_CFLAGS)
-liboph_predicate_la_LDFLAGS = -avoid-version -shared $(LIBS) $(MATHEVAL_LIBS) -lpthread -L. -loph_debug -loph_core
+liboph_predicate_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(MATHEVAL_LIBS) -lpthread -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_predicate2.la
 liboph_predicate2_la_SOURCES = oph_predicate2.c oph_predicate2.h oph_core_matheval.c oph_core_matheval.h
 liboph_predicate2_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(MATHEVAL_CFLAGS)
-liboph_predicate2_la_LDFLAGS = -avoid-version -shared $(LIBS) $(MATHEVAL_LIBS) -lpthread -L. -loph_debug -loph_core
+liboph_predicate2_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(MATHEVAL_LIBS) -lpthread -L. -loph_debug -loph_core
 endif
 
 if GSL_SUPPORT
 lib_LTLIBRARIES += liboph_gsl_boxplot.la
 liboph_gsl_boxplot_la_SOURCES = oph_gsl_boxplot.c oph_gsl_boxplot.h oph_core_gsl.h
 liboph_gsl_boxplot_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_boxplot_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_boxplot_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_sd.la
 liboph_gsl_sd_la_SOURCES = oph_gsl_sd.c oph_gsl_sd.h oph_core_gsl.h
 liboph_gsl_sd_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_sd_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_sd_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_complex_get_real.la
 liboph_gsl_complex_get_real_la_SOURCES = oph_gsl_complex_get_real.c oph_gsl_complex_get_real.h oph_core_gsl.h
 liboph_gsl_complex_get_real_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_complex_get_real_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_complex_get_real_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_complex_get_imag.la
 liboph_gsl_complex_get_imag_la_SOURCES = oph_gsl_complex_get_imag.c oph_gsl_complex_get_imag.h oph_core_gsl.h
 liboph_gsl_complex_get_imag_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_complex_get_imag_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_complex_get_imag_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_complex_get_arg.la
 liboph_gsl_complex_get_arg_la_SOURCES = oph_gsl_complex_get_arg.c oph_gsl_complex_get_arg.h oph_core_gsl.h
 liboph_gsl_complex_get_arg_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_complex_get_arg_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_complex_get_arg_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_complex_get_abs.la
 liboph_gsl_complex_get_abs_la_SOURCES = oph_gsl_complex_get_abs.c oph_gsl_complex_get_abs.h oph_core_gsl.h
 liboph_gsl_complex_get_abs_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_complex_get_abs_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_complex_get_abs_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_complex_to_polar.la
 liboph_gsl_complex_to_polar_la_SOURCES = oph_gsl_complex_to_polar.c oph_gsl_complex_to_polar.h oph_core_gsl.h
 liboph_gsl_complex_to_polar_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_complex_to_polar_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_complex_to_polar_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_complex_to_rect.la
 liboph_gsl_complex_to_rect_la_SOURCES = oph_gsl_complex_to_rect.c oph_gsl_complex_to_rect.h oph_core_gsl.h
 liboph_gsl_complex_to_rect_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_complex_to_rect_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_complex_to_rect_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_ifft.la
 liboph_gsl_ifft_la_SOURCES = oph_gsl_ifft.c oph_gsl_ifft.h oph_core_gsl.h
 liboph_gsl_ifft_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_ifft_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_ifft_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_idwt.la
 liboph_gsl_idwt_la_SOURCES = oph_gsl_idwt.c oph_gsl_idwt.h oph_core_gsl.h
 liboph_gsl_idwt_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_idwt_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_idwt_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_fft.la
 liboph_gsl_fft_la_SOURCES = oph_gsl_fft.c oph_gsl_fft.h oph_core_gsl.h
 liboph_gsl_fft_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_fft_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_fft_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_filter.la
 liboph_filter_la_SOURCES = oph_filter.c oph_filter.h oph_core_gsl.h
 liboph_filter_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_filter_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LIBS) -L. -loph_debug -loph_core
+liboph_filter_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LIBS) -L. -loph_debug -loph_core
 
 
 lib_LTLIBRARIES += liboph_gsl_dwt.la
 liboph_gsl_dwt_la_SOURCES = oph_gsl_dwt.c oph_gsl_dwt.h oph_core_gsl.h
 liboph_gsl_dwt_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_dwt_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_dwt_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_histogram.la
 liboph_gsl_histogram_la_SOURCES = oph_gsl_histogram.c oph_cast.h oph_core_gsl.h
 liboph_gsl_histogram_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_histogram_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_histogram_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_quantile.la
 liboph_gsl_quantile_la_SOURCES = oph_gsl_quantile.c oph_gsl_quantile.h oph_core_gsl.h
 liboph_gsl_quantile_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_quantile_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_quantile_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_stats.la
 liboph_gsl_stats_la_SOURCES = oph_gsl_stats.c oph_gsl_stats.h oph_core_gsl.h
 liboph_gsl_stats_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_stats_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_stats_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_sort.la
 liboph_gsl_sort_la_SOURCES = oph_gsl_sort.c oph_gsl_sort.h oph_core_gsl.h
 liboph_gsl_sort_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_sort_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_sort_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_spline.la
 liboph_gsl_spline_la_SOURCES = oph_gsl_spline.c oph_gsl_spline.h oph_core_gsl.h
 liboph_gsl_spline_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_spline_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_spline_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_fit_linear.la
 liboph_gsl_fit_linear_la_SOURCES = oph_gsl_fit_linear.c oph_gsl_fit_linear.h oph_core_gsl.h
 liboph_gsl_fit_linear_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_fit_linear_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_fit_linear_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_fit_linear_coeff.la
 liboph_gsl_fit_linear_coeff_la_SOURCES = oph_gsl_fit_linear_coeff.c oph_gsl_fit_linear_coeff.h oph_core_gsl.h
 liboph_gsl_fit_linear_coeff_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_fit_linear_coeff_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_fit_linear_coeff_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 
 lib_LTLIBRARIES += liboph_gsl_correlation.la
 liboph_gsl_correlation_la_SOURCES = oph_gsl_correlation.c oph_gsl_correlation.h oph_core_gsl.h
 liboph_gsl_correlation_la_CFLAGS = $(MYSQL_CFLAGS) $(CFLAGS) $(GSL_CFLAG)
-liboph_gsl_correlation_la_LDFLAGS = -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
+liboph_gsl_correlation_la_LDFLAGS = -no-undefined -avoid-version -shared $(LIBS) $(GSL_LDFLAG) -L. -loph_debug -loph_core
 endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_abs_array.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_abs_array.h	2024-09-24 10:59:15.809813700 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_accumulate.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_accumulate.h	2024-09-24 10:59:15.808770800 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_affine.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_affine.h	2024-09-24 10:59:15.807766900 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_aggregate_operator.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_aggregate_operator.h	2024-09-24 10:59:15.803783100 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_aggregate_stats.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_aggregate_stats.h	2024-09-24 10:59:15.801502500 +0900
@@ -31,7 +31,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_aggregate_stats_final.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_aggregate_stats_final.h	2024-09-24 10:59:15.795704700 +0900
@@ -31,7 +31,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_aggregate_stats_partial.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_aggregate_stats_partial.h	2024-09-24 10:59:15.794714100 +0900
@@ -31,7 +31,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_append.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_append.h	2024-09-24 10:59:15.784370300 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_arg_array.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_arg_array.h	2024-09-24 10:59:15.784370300 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_arg_max_array.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_arg_max_array.h	2024-09-24 10:59:15.784370300 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_arg_min_array.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_arg_min_array.h	2024-09-24 10:59:15.784370300 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_cast.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_cast.h	2024-09-24 10:59:15.782344800 +0900
@@ -30,7 +30,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_ccluster_kcluster.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_ccluster_kcluster.h	2024-09-24 10:59:15.781346700 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_compare.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_compare.h	2024-09-24 10:59:15.781346700 +0900
@@ -24,7 +24,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_compress.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_compress.h	2024-09-24 10:59:15.781346700 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_concat.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_concat.h	2024-09-24 10:59:15.781346700 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_concat2.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_concat2.h	2024-09-24 10:59:15.813769300 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_convert_d.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_convert_d.h	2024-09-24 10:59:15.795704700 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_convert_l.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_convert_l.h	2024-09-24 10:59:15.781346700 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_count_array.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_count_array.h	2024-09-24 10:59:15.781346700 +0900
@@ -28,7 +28,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_deaccumulate.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_deaccumulate.h	2024-09-24 10:59:15.781346700 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_div_array.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_div_array.h	2024-09-24 10:59:15.781346700 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_dump.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_dump.h	2024-09-24 10:59:15.781346700 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_expand.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_expand.h	2024-09-24 10:59:15.781346700 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_extend.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_extend.h	2024-09-24 10:59:15.781346700 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_extract.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_extract.h	2024-09-24 10:59:15.781346700 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_filter.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_filter.h	2024-09-24 10:59:15.781346700 +0900
@@ -30,7 +30,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_find.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_find.h	2024-09-24 10:59:15.781346700 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_get_index_array.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_get_index_array.h	2024-09-24 10:59:15.780358500 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_get_subarray.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_get_subarray.h	2024-09-24 10:59:15.782344800 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_get_subarray2.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_get_subarray2.h	2024-09-24 10:59:15.780358500 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_get_subarray3.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_get_subarray3.h	2024-09-24 10:59:15.780358500 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_boxplot.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_boxplot.h	2024-09-24 10:59:15.773393200 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_complex_get_abs.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_complex_get_abs.h	2024-09-24 10:59:15.773393200 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_complex_get_arg.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_complex_get_arg.h	2024-09-24 10:59:15.773393200 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_complex_get_imag.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_complex_get_imag.h	2024-09-24 10:59:15.773393200 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_complex_get_real.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_complex_get_real.h	2024-09-24 10:59:15.770351600 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_complex_to_polar.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_complex_to_polar.h	2024-09-24 10:59:15.773393200 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_complex_to_rect.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_complex_to_rect.h	2024-09-24 10:59:15.770035700 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_correlation.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_correlation.h	2024-09-24 10:59:15.768700000 +0900
@@ -30,7 +30,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_dwt.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_dwt.h	2024-09-24 10:59:15.768700000 +0900
@@ -30,7 +30,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_fft.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_fft.h	2024-09-24 10:59:15.768700000 +0900
@@ -30,7 +30,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_fit_linear.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_fit_linear.h	2024-09-24 10:59:15.768700000 +0900
@@ -28,7 +28,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_fit_linear_coeff.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_fit_linear_coeff.h	2024-09-24 10:59:15.766704900 +0900
@@ -28,7 +28,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_histogram.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_histogram.h	2024-09-24 10:59:15.762791000 +0900
@@ -30,7 +30,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_idwt.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_idwt.h	2024-09-24 10:59:15.762791000 +0900
@@ -30,7 +30,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_ifft.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_ifft.h	2024-09-24 10:59:15.762791000 +0900
@@ -30,7 +30,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_quantile.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_quantile.h	2024-09-24 10:59:15.756760300 +0900
@@ -30,7 +30,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_sd.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_sd.h	2024-09-24 10:59:15.756760300 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_sort.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_sort.h	2024-09-24 10:59:15.755724300 +0900
@@ -30,7 +30,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_spline.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_spline.h	2024-09-24 10:59:15.755724300 +0900
@@ -28,7 +28,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_gsl_stats.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_gsl_stats.h	2024-09-24 10:59:15.755724300 +0900
@@ -30,7 +30,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_id3.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_id3.h	2024-09-24 10:59:44.320626300 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_id_of_subset.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_id_of_subset.h	2024-09-24 10:59:54.981829000 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_id_to_index.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_id_to_index.h	2024-09-24 10:59:50.149240200 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_interlace.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_interlace.h	2024-09-24 10:59:15.753037200 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_interlace2.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_interlace2.h	2024-09-24 10:59:15.752020500 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_mask_array.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_mask_array.h	2024-09-24 10:59:15.752020500 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_math.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_math.h	2024-09-24 10:59:15.809813700 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_max_array.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_max_array.h	2024-09-24 10:59:15.749010000 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_min_array.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_min_array.h	2024-09-24 10:59:15.755724300 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_moving_avg.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_moving_avg.h	2024-09-24 10:59:15.748004500 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_mul_array.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_mul_array.h	2024-09-24 10:59:15.748004500 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_mul_scalar.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_mul_scalar.h	2024-09-24 10:59:15.748004500 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_mul_scalar2.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_mul_scalar2.h	2024-09-24 10:59:15.810768800 +0900
@@ -31,7 +31,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_normalize.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_normalize.h	2024-09-24 10:59:15.748004500 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_operation_array.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_operation_array.h	2024-09-24 10:59:15.746000200 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_operator.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_operator.h	2024-09-24 10:59:15.744028900 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_operator_array.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_operator_array.h	2024-09-24 10:59:15.744028900 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_padding.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_padding.h	2024-09-24 10:59:15.744028900 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_permute.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_permute.h	2024-09-24 10:59:15.744028900 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_predicate.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_predicate.h	2024-09-24 10:59:15.744028900 +0900
@@ -26,7 +26,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_predicate2.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_predicate2.h	2024-09-24 10:59:15.742010600 +0900
@@ -26,7 +26,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_quantize.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_quantize.h	2024-09-24 10:59:15.740005400 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_reduce.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_reduce.h	2024-09-24 10:59:15.739009200 +0900
@@ -31,7 +31,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_reduce2.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_reduce2.h	2024-09-24 10:59:15.738321600 +0900
@@ -31,7 +31,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_reduce3.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_reduce3.h	2024-09-24 10:59:15.737332300 +0900
@@ -31,7 +31,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_replace.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_replace.h	2024-09-24 10:59:15.737332300 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_reverse.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_reverse.h	2024-09-24 10:59:15.722883800 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_roll_up.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_roll_up.h	2024-09-24 10:59:15.817479200 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_rotate.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_rotate.h	2024-09-24 10:59:15.817479200 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_sequence.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_sequence.h	2024-09-24 10:59:15.817479200 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_shift.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_shift.h	2024-09-24 10:59:15.817479200 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_size_array.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_size_array.h	2024-09-24 10:59:15.817479200 +0900
@@ -28,7 +28,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_sub_array.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_sub_array.h	2024-09-24 10:59:15.817479200 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_sum_array.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_sum_array.h	2024-09-24 10:59:15.817479200 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_sum_array_r.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_sum_array_r.h	2024-09-24 10:59:15.817479200 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_sum_scalar.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_sum_scalar.h	2024-09-24 10:59:15.817479200 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_sum_scalar2.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_sum_scalar2.h	2024-09-24 10:59:15.817479200 +0900
@@ -31,7 +31,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_to_bin.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_to_bin.h	2024-09-24 10:59:15.810768800 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_uncompress.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_uncompress.h	2024-09-24 10:59:15.810768800 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
--- origsrc/ophidia-primitives-1.8.0/src/oph_value_to_bin.h	2023-08-01 02:17:29.000000000 +0900
+++ src/ophidia-primitives-1.8.0/src/oph_value_to_bin.h	2024-09-24 10:59:15.815778500 +0900
@@ -27,7 +27,7 @@
 
 /* MySQL headers  */
 #include <mysql.h>		// It contains UDF-related symbols and data structures
-#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
+#if MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002 && MYSQL_VERSION_ID < 80100
 typedef bool my_bool;
 #endif
 
