--- origsrc/libxcb-glib-0.99.2/src/libxcb-glib.mk	2012-10-13 08:31:22.000000000 +0900
+++ src/libxcb-glib-0.99.2/src/libxcb-glib.mk	2013-12-22 16:54:42.751855100 +0900
@@ -16,7 +16,7 @@ libxcb_glib_la_CFLAGS = \
 	$(XCB_CFLAGS) \
 	$(GLIB_CFLAGS)
 
-libxcb_glib_la_LDFLAGS = \
+libxcb_glib_la_LDFLAGS = -no-undefined \
 	$(AM_LDFLAGS) \
 	-version-info $(LIBXCB_GLIB_CURRENT):$(LIBXCB_GLIB_REVISION):$(LIBXCB_GLIB_AGE) \
 	-export-dynamic
