--- origsrc/json-schema-validator-2.4.0/CMakeLists.txt	2025-11-05 04:58:21.000000000 +0900
+++ src/json-schema-validator-2.4.0/CMakeLists.txt	2026-01-13 13:34:07.362628000 +0900
@@ -218,3 +218,9 @@ if (NOT PROJECT_IS_TOP_LEVEL)
         set(nlohmann_json_schema_validator_VERSION_TWEAK ${nlohmann_json_schema_validator_VERSION_TWEAK} PARENT_SCOPE)
     endif ()
 endif ()
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/json-schema-validator.pc.in ${CMAKE_BINARY_DIR}/json-schema-validator.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/json-schema-validator.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/json-schema-validator-2.4.0/json-schema-validator.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/json-schema-validator-2.4.0/json-schema-validator.pc.in	2026-01-13 13:35:01.633053200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: json-schema-validator
+Description: JSON schema validator for JSON for Modern C++
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lnlohmann_json_schema_validator
+Cflags: -I${includedir}
