--- origsrc/json-schema-validator-2.3.0/CMakeLists.txt	2023-11-27 18:20:26.000000000 +0900
+++ src/json-schema-validator-2.3.0/CMakeLists.txt	2024-09-12 14:45:50.486432700 +0900
@@ -215,3 +215,10 @@ if (NOT PROJECT_IS_TOP_LEVEL)
         set(nlohmann_json_schema_validator_VERSION_TWEAK ${nlohmann_json_schema_validator_VERSION_TWEAK} PARENT_SCOPE)
     endif ()
 endif ()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/json-schema-validator.pc.in ${CMAKE_BINARY_DIR}/json-schema-validator.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/json-schema-validator.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/json-schema-validator-2.3.0/json-schema-validator.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/json-schema-validator-2.3.0/json-schema-validator.pc.in	2024-09-12 14:46:17.675398200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: json-schema-validator
+Description: JSON schema validator for JSON for Modern C++
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lnlohmann_json_schema_validator
+Cflags: -I${includedir}
--- origsrc/json-schema-validator-2.3.0/src/CMakeLists.txt	2023-11-27 18:20:26.000000000 +0900
+++ src/json-schema-validator-2.3.0/src/CMakeLists.txt	2024-09-12 13:43:03.141152000 +0900
@@ -69,6 +69,7 @@ if (JSON_VALIDATOR_INSTALL)
                 LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT nlohmann_json_schema_validator_Runtime
                 NAMELINK_COMPONENT nlohmann_json_schema_validator_Development
                 ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT nlohmann_json_schema_validator_Development
-                PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/nlohmann COMPONENT nlohmann_json_schema_validator_Development)
+                PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/nlohmann COMPONENT nlohmann_json_schema_validator_Development
+		RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT nlohmann_json_schema_validator_Runtime)
     endif ()
 endif ()
