--- origsrc/xa-2.4.0/Makefile	2023-11-02 14:41:40.000000000 +0900
+++ src/xa-2.4.0/Makefile	2024-01-16 10:36:37.072687200 +0900
@@ -1,13 +1,13 @@
 # Unix gcc or DOS go32 cross-compiling gcc
 #
 VERS = 2.4.0
-CC = gcc
-LD = gcc
+CC ?= gcc
+LD ?= gcc
 # for testing. not to be used; build failures in misc/.
 #CFLAGS = -O2 -W -Wall -pedantic -ansi
 #CFLAGS = -O2 -g
-CFLAGS = -O2 
-LDFLAGS = -lc
+CFLAGS ?= -O2 
+LDFLAGS ?= -lc
 
 # for DOS?
 # CC = gcc-go32
@@ -19,18 +19,18 @@ LDFLAGS = -lc
 #CFLAGS =
 #LD = ld
 
-DESTDIR = /usr/local
+DESTDIR =
 
-BINDIR = $(DESTDIR)/bin
-MANDIR = $(DESTDIR)/share/man/man1
-DOCDIR = $(DESTDIR)/share/doc
+BINDIR = $(DESTDIR)/usr/bin
+MANDIR = $(DESTDIR)/usr/share/man/man1
+DOCDIR = $(DESTDIR)/usr/share/doc
 
 MKDIR = mkdir -p
 INSTALL = install
 
 TESTS=ALL
 
-all: killxa xa uncpk
+all: xa uncpk
 
 killxa:
 	rm -f xa
