--- origsrc/xa-2.3.13/Makefile	2022-03-13 10:46:55.000000000 +0900
+++ src/xa-2.3.13/Makefile	2022-06-20 21:39:09.287614800 +0900
@@ -1,11 +1,11 @@
 # Unix gcc or DOS go32 cross-compiling gcc
 #
-CC = gcc
-LD = gcc
+CC ?= gcc
+LD ?= gcc
 # for testing. not to be used; build failures in misc/.
 #CFLAGS = -O2 -W -Wall -pedantic -ansi
-CFLAGS = -O2
-LDFLAGS = -lc
+CFLAGS ?= -O2
+LDFLAGS ?= -lc
 
 # for DOS?
 # CC = gcc-go32
@@ -17,11 +17,11 @@ LDFLAGS = -lc
 #CFLAGS =
 #LD = ld
 
-DESTDIR = /usr/local
+DESTDIR =
 
-BINDIR = $(DESTDIR)/bin
-MANDIR = $(DESTDIR)/share/man/man1
-DOCDIR = $(DESTDIR)/share/doc
+BINDIR = $(DESTDIR)/usr/bin
+MANDIR = $(DESTDIR)/usr/share/man/man1
+DOCDIR = $(DESTDIR)/usr/share/doc
 
 MKDIR = mkdir -p
 INSTALL = install
