--- origsrc/bento4-1.6.0.639/CMakeLists.txt	2021-07-22 13:17:29.000000000 +0900
+++ src/bento4-1.6.0.639/CMakeLists.txt	2022-09-26 13:02:42.919536600 +0900
@@ -16,6 +16,9 @@ set(SOURCE_CORE ${SOURCE_ROOT}/Core)
 set(SOURCE_CRYPTO ${SOURCE_ROOT}/Crypto)
 set(SOURCE_METADATA ${SOURCE_ROOT}/MetaData)
 set(SOURCE_SYSTEM ${SOURCE_ROOT}/System)
+if (CYGWIN)
+  set(SOURCE_CAPI ${SOURCE_ROOT}/CApi)
+endif()
 
 # Compiler warning and optimization flags
 if(CMAKE_C_COMPILER_ID MATCHES "GNU")
@@ -41,6 +44,22 @@ file(GLOB AP4_SOURCES
   ${SOURCE_METADATA}/*.cpp
   ${SOURCE_SYSTEM}/StdC/*.cpp
 )
+if (CYGWIN)
+  file(GLOB AP4_SOURCES_CAPI ${SOURCE_CAPI}/*.cpp)
+  set(AP4_SOURCES ${AP4_SOURCES} ${AP4_SOURCES_CAPI})
+
+  file(GLOB AP4_HEADERS_CODES ${SOURCE_CODECS}/*.h)
+  file(GLOB AP4_HEADERS_CORE ${SOURCE_CORE}/*.h)
+  file(GLOB AP4_HEADERS_CRYPTO ${SOURCE_CRYPTO}/*.h)
+  file(GLOB AP4_HEADERS_METADATA ${SOURCE_METADATA}/*.h)
+  file(GLOB AP4_HEADERS_CAPI ${SOURCE_CAPI}/*.h)
+
+  install(FILES ${AP4_HEADERS_CODES} DESTINATION include/bento4/Codes)
+  install(FILES ${AP4_HEADERS_CORE} DESTINATION include/bento4/Core)
+  install(FILES ${AP4_HEADERS_CRYPTO} DESTINATION include/bento4/Crypto)
+  install(FILES ${AP4_HEADERS_METADATA} DESTINATION include/bento4/Metadata)
+  install(FILES ${AP4_HEADERS_CAPI} DESTINATION include/bento4/Capi)
+endif()
 
 # Platform specifics
 if(WIN32)
@@ -56,9 +75,18 @@ set(AP4_INCLUDE_DIRS
   ${SOURCE_CRYPTO}
   ${SOURCE_METADATA}
 )
+if (CYGWIN)
+  set(AP4_INCLUDE_DIRS ${AP4_INCLUDE_DIRS} ${SOURCE_CAPI})
+endif()
 
-add_library(ap4 STATIC ${AP4_SOURCES})
+add_library(ap4 SHARED ${AP4_SOURCES})
 target_include_directories(ap4 PUBLIC ${AP4_INCLUDE_DIRS})
+if (DEFINED SOVERSION)
+  set_target_properties(ap4 PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+if (CYGWIN)
+  install(TARGETS ap4 RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+endif()
 
 # Use the statically linked C runtime library
 if(MSVC)
@@ -74,10 +102,19 @@ foreach(app ${BENTO4_APPS})
   string(TOLOWER ${app} binary_name)
   add_executable(${binary_name} ${SOURCE_ROOT}/Apps/${app}/${app}.cpp)
   target_link_libraries(${binary_name} ap4)
-
+  if (CYGWIN)
+    install(TARGETS ${binary_name} RUNTIME DESTINATION bin)
+  endif()
   if(MSVC)
     set_property(TARGET ${binary_name} PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
     target_compile_definitions(${binary_name} PRIVATE -D_CONSOLE)
   endif()
 endforeach()
 endif(BUILD_APPS)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/bento4.pc.in ${CMAKE_BINARY_DIR}/bento4.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/bento4.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/bento4-1.6.0.639/bento4.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/bento4-1.6.0.639/bento4.pc.in	2022-09-26 12:57:43.626730300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: bento4
+Description: Full-featured MP4 format and MPEG DASH library and tools
+Version: @VERSION@
+Libs: -L${libdir} -lap4
+Cflags: -I${includedir}/bento4
