--- origsrc/d52v341/Makefile	2025-03-16 18:12:50.745848500 +0900
+++ src/d52v341/Makefile	2025-03-16 18:13:50.996323100 +0900
@@ -5,7 +5,7 @@
 OBJDIR = ./obj
 
 CC = gcc
-CFLAGS = -Wall -O2
+CFLAGS ?= -Wall -O2
 LIBS =
 D52OBJS = $(OBJDIR)/d52.o $(OBJDIR)/common.o $(OBJDIR)/d52pass1.o $(OBJDIR)/d52pass2.o $(OBJDIR)/d52table.o $(OBJDIR)/analyze52.o $(OBJDIR)/analyze.o
 D48OBJS = $(OBJDIR)/d48.o $(OBJDIR)/common.o $(OBJDIR)/d48pass.o $(OBJDIR)/d48table.o
