--- origsrc/libdrycli/Makefile.am	2025-04-30 10:35:04.000000000 +0900
+++ src/libdrycli/Makefile.am	2025-04-30 11:18:51.618420500 +0900
@@ -1,3 +1,6 @@
 SUBDIRS         = lib src test #doc
 EXTRA_DIST      = AUTHORS CREDITS README UNLICENSE VERSION
 ACLOCAL_AMFLAGS = -I etc/aclocal
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libdrycli/configure.ac	2025-04-30 10:35:04.000000000 +0900
+++ src/libdrycli/configure.ac	2025-04-30 11:18:51.634055400 +0900
@@ -99,4 +99,5 @@ AC_SUBST([PACKAGE_VERSION_MINOR], ["VERS
 AC_SUBST([PACKAGE_VERSION_PATCH], ["VERSION_PATCH"])
 AC_CONFIG_FILES([src/dry/cli/version.h])
 AH_BOTTOM([#include "libcompat.h"])
+AC_CONFIG_FILES([libdrycli.pc])
 AC_OUTPUT
--- origsrc/libdrycli/libdrycli.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libdrycli/libdrycli.pc.in	2025-04-30 11:19:52.212602200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libdrycli
+Description: Command-line interface (CLI) library for C++14
+Version: @PACKAGE_VERSION_MAJOR@.@PACKAGE_VERSION_MINOR@.@PACKAGE_VERSION_PATCH@
+Libs: -L${libdir} -ldrycli
+Cflags: -I${includedir}
--- origsrc/libdrycli/src/dry/cli/Makefile.am	2025-04-30 10:35:04.000000000 +0900
+++ src/libdrycli/src/dry/cli/Makefile.am	2025-04-30 11:07:15.182044600 +0900
@@ -1,5 +1,8 @@
 lib_LTLIBRARIES = libdrycli.la
 
+libdrycli_la_CPPFLAGS = -I$(builddir) -I$(top_builddir)/src -I$(top_srcdir)/lib
+libdrycli_la_LDFLAGS = -no-undefined
+
 libdrycli_la_SOURCES =    \
   environment.cc          \
   extern.cc               \
