diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libzonotope/CMakeLists.txt src/libzonotope/CMakeLists.txt
--- origsrc/libzonotope/CMakeLists.txt	2016-01-12 01:39:00.000000000 +0900
+++ src/libzonotope/CMakeLists.txt	2016-01-12 01:44:38.532988900 +0900
@@ -47,3 +47,11 @@ add_custom_target(uninstall
 # Header installation target
 #
 install(FILES ${LIBZONOTOPE_HPP} DESTINATION ${LIBZONOTOPE_INCLUDE_DIR})
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "0")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libzonotope.pc.in ${CMAKE_BINARY_DIR}/libzonotope.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libzonotope.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libzonotope/lib/CMakeLists.txt src/libzonotope/lib/CMakeLists.txt
--- origsrc/libzonotope/lib/CMakeLists.txt	2016-01-12 01:39:00.000000000 +0900
+++ src/libzonotope/lib/CMakeLists.txt	2016-01-12 01:42:34.894966300 +0900
@@ -11,4 +11,6 @@ target_link_libraries(zonotope_c ${GMP_L
 file(GLOB ZONOTOPE_C_H ../src/zonotope_c.h )
 
 install(FILES ${ZONOTOPE_C_H} DESTINATION ${LIBZONOTOPE_INCLUDE_DIR})
-install(TARGETS zonotope_c DESTINATION ${LIBZONOTOPE_LIBRARY_DIR})
+install(TARGETS zonotope_c RUNTIME DESTINATION bin ARCHIVE DESTINATION ${LIBZONOTOPE_LIBRARY_DIR})
+
+set_target_properties(zonotope_c PROPERTIES SOVERSION 0)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libzonotope/libzonotope.pc.in src/libzonotope/libzonotope.pc.in
--- origsrc/libzonotope/libzonotope.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libzonotope/libzonotope.pc.in	2016-01-12 01:45:05.405186500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libzonotope
+Description: Collection of scripts and algorithms for zonotope manipulations
+Version: @VERSION@
+Libs: -L${libdir} -lzonotope_c
+Cflags: -I${includedir}/zonotope
