--- origsrc/cmp-20/CMakeLists.txt	2022-03-30 01:59:14.000000000 +0900
+++ src/cmp-20/CMakeLists.txt	2025-09-11 20:49:43.426386800 +0900
@@ -1,4 +1,35 @@
-# Edit following two lines to set component requirements (see docs)
-set(COMPONENT_SRCS "cmp.c" )
-set(COMPONENT_ADD_INCLUDEDIRS ".")
-register_component()
+project(cmp)
+
+add_library(cmp SHARED cmp.c)
+target_include_directories(cmp PRIVATE ${CMAKE_SOURCE_DIR})
+if (DEFINED SOVERSION)
+    set_target_properties(cmp PROPERTIES SOVERSION ${SOVERSION})
+endif()
+
+add_executable(example1 examples/example1.c)
+target_include_directories(example1 PRIVATE ${CMAKE_SOURCE_DIR})
+target_link_libraries(example1 cmp)
+
+add_executable(example2 examples/example2.c)
+target_include_directories(example2 PRIVATE ${CMAKE_SOURCE_DIR})
+target_link_libraries(example2 cmp)
+
+enable_testing()
+find_package(pkgconfig)
+pkg_check_modules(CMOCKA cmocka)
+add_executable(cmpunittest test/test.c test/tests.c test/buf.c test/utils.c)
+target_include_directories(cmpunittest PRIVATE ${CMAKE_SOURCE_DIR})
+target_link_libraries(cmpunittest cmp ${CMOCKA_LIBRARIES})
+add_test(NAME cmpunitest COMMAND cmpunittest)
+
+install(FILES cmp.h DESTINATION include)
+install(TARGETS cmp RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+
+if (DEFINED VERSION)
+    set(prefix ${CMAKE_INSTALL_PREFIX})
+    set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+    set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+    set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+    configure_file(${CMAKE_SOURCE_DIR}/cmp.pc.in ${CMAKE_BINARY_DIR}/cmp.pc @ONLY)
+    install(FILES ${CMAKE_BINARY_DIR}/cmp.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+endif()
--- origsrc/cmp-20/cmp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/cmp-20/cmp.pc.in	2025-09-11 20:49:11.481839300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: cmp
+Description: C implementation of the MessagePack serialization format
+Version: @VERSION@
+Libs: -L${libdir} -lcmp
+Cflags: -I${includedir}
