--- origsrc/warp/CMakeLists.txt	2013-11-02 22:06:52.000000000 +0900
+++ src/warp/CMakeLists.txt	2013-11-02 22:15:06.356408200 +0900
@@ -22,32 +22,37 @@ link_directories(
 )
 
 add_executable(warp_ann_zparser ann_zparser.cpp)
-target_link_libraries(warp_ann_zparser
+target_link_libraries(warp_ann_zparser ssp
 	${Boost_LIBRARIES}
 )
+install(TARGETS warp_ann_zparser DESTINATION bin)
 
 add_executable(warp_ann_learn ann_learn.cpp)
-target_link_libraries(warp_ann_learn
+target_link_libraries(warp_ann_learn ssp
 	${Boost_LIBRARIES}
 	${FANN_LIBRARIES}
 )
+install(TARGETS warp_ann_learn DESTINATION bin)
 
 add_executable(warp_ann_check ann_check.cpp)
-target_link_libraries(warp_ann_check
+target_link_libraries(warp_ann_check ssp
 	${Boost_LIBRARIES}
 	${FANN_LIBRARIES}
 )
+install(TARGETS warp_ann_check DESTINATION bin)
 
 add_executable(warp_trie_test trie.cpp)
-target_link_libraries(warp_trie_test
+target_link_libraries(warp_trie_test ssp
 	${Boost_LIBRARIES}
 )
+install(TARGETS warp_trie_test DESTINATION bin)
 
 add_executable(warp_zpack pack.cpp)
-target_link_libraries(warp_zpack
+target_link_libraries(warp_zpack ssp
 	${Boost_LIBRARIES}
 	${MSGPACK_LIBRARIES}
 )
+install(TARGETS warp_zpack DESTINATION bin)
 
 FILE(GLOB headers
 	"${CMAKE_CURRENT_SOURCE_DIR}/*.hpp"
--- origsrc/warp/ann_learn.cpp	2013-11-02 22:06:52.000000000 +0900
+++ src/warp/ann_learn.cpp	2013-11-02 22:09:02.403283200 +0900
@@ -14,6 +14,8 @@
  * limitations under the License.
  */
 
+#include <iostream>
+
 #include <fann.h>
 
 #include <fstream>
--- origsrc/warp/lex.hpp	2013-11-02 22:06:52.000000000 +0900
+++ src/warp/lex.hpp	2013-11-02 22:12:04.356408200 +0900
@@ -17,6 +17,11 @@
 #ifndef __IOREMAP_WARP_LEX_HPP
 #define __IOREMAP_WARP_LEX_HPP
 
+#if defined(__CYGWIN__)
+#include <strings.h>
+#define ffsll(i) ffs(i)
+#endif
+
 #include <boost/locale.hpp>
 
 #include "pack.hpp"
