--- origsrc/vcftools_0.1.11/Makefile	2013-06-13 23:40:56.000000000 +0900
+++ src/vcftools_0.1.11/Makefile	2013-12-08 13:50:05.860364100 +0900
@@ -17,7 +17,7 @@ ifndef PREFIX
     export PREFIX = $(dir $(realpath $(lastword $(MAKEFILE_LIST))))
 endif
 export BINDIR = ${PREFIX}/bin
-export MODDIR = ${PREFIX}/lib/perl5/site_perl
+export MODDIR = ${PREFIX}/lib/perl5/vendor_perl
 
 DIRS = cpp perl
 install:
--- origsrc/vcftools_0.1.11/cpp/Makefile	2013-06-13 23:40:54.000000000 +0900
+++ src/vcftools_0.1.11/cpp/Makefile	2013-12-08 13:44:26.765797600 +0900
@@ -12,9 +12,9 @@ ifndef VCFTOOLS_PCA
 	VCFTOOLS_PCA = 0
 endif
 # Compiler flags
-CFLAGS = -O2 -m64
+CFLAGS = -O3
 #CFLAGS = -Wall -O2 -pg -m64
-CPPFLAGS = -O2 -D_FILE_OFFSET_BITS=64 
+CPPFLAGS = -O3 -D_FILE_OFFSET_BITS=64 
 #CPPFLAGS = -O2 -Wall -pg -D_FILE_OFFSET_BITS=64
 # Included libraries (zlib)
 LIB = -lz 
