--- origsrc/arb-2.21.0/CMakeLists.txt	2021-09-25 18:53:01.000000000 +0900
+++ src/arb-2.21.0/CMakeLists.txt	2021-10-06 08:13:07.048875000 +0900
@@ -148,7 +148,11 @@ foreach (FOLDER ${FOLDERS})
     set(HEADERS ${HEADERS} ${FOLDER}.h)
 endforeach ()
 
-install(FILES ${HEADERS} DESTINATION include)
+if (CYGWIN)
+    install(FILES ${HEADERS} DESTINATION include/arb)
+else()
+    install(FILES ${HEADERS} DESTINATION include)
+endif()
 
 if (BUILD_TESTING)
     enable_testing()
@@ -163,3 +167,11 @@ if (BUILD_TESTING)
     endforeach ()
 endif ()
 
+if (CYGWIN)
+    set(prefix ${CMAKE_INSTALL_PREFIX})
+    set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+    set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+    set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+    configure_file(${CMAKE_SOURCE_DIR}/arb.pc.in ${CMAKE_BINARY_DIR}/arb.pc @ONLY)
+    install(FILES ${CMAKE_BINARY_DIR}/arb.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+endif()
--- origsrc/arb-2.21.0/arb.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/arb-2.21.0/arb.pc.in	2021-10-06 08:11:00.354291900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: arb
+Description: C library for arbitrary-precision interval arithmetic
+Version: @ARB_VERSION@
+Libs: -L${libdir} -larb -lflint -lgmp -lmpfr
+Cflags: -I${includedir}
--- origsrc/arb-2.21.0/configure	2021-09-25 18:53:01.000000000 +0900
+++ src/arb-2.21.0/configure	2021-10-06 08:39:27.320853900 +0900
@@ -10,7 +10,7 @@
 # arb     => soname
 # 2.7.0   => 0.0.0
 ARB_MAJOR=2
-ARB_MINOR=12
+ARB_MINOR=21
 ARB_PATCH=0
 
 PREFIX="/usr/local"
