--- origsrc/arb-2.18.1/CMakeLists.txt	2020-06-25 17:49:16.000000000 +0900
+++ src/arb-2.18.1/CMakeLists.txt	2020-10-14 18:22:31.764727800 +0900
@@ -161,3 +161,10 @@ if (BUILD_TESTING)
     endforeach ()
 endif ()
 
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/arb.pc.in ${CMAKE_BINARY_DIR}/arb.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/arb.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/arb-2.18.1/arb.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/arb-2.18.1/arb.pc.in	2020-10-14 18:31:14.908903600 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: arb
+Description: C library for arbitrary-precision interval arithmetic
+Version: @ARB_VERSION@
+Libs: -L${libdir} -larb -lflint
+Cflags: -I${includedir}/arb
+Requires: gmp mpfr
--- origsrc/arb-2.18.1/configure	2020-06-25 17:49:16.000000000 +0900
+++ src/arb-2.18.1/configure	2020-10-14 18:25:36.448737400 +0900
@@ -10,7 +10,7 @@
 # arb     => soname
 # 2.7.0   => 0.0.0
 ARB_MAJOR=2
-ARB_MINOR=9
+ARB_MINOR=18
 ARB_PATCH=1
 
 PREFIX="/usr/local"
