--- origsrc/replxx-release-0.0.3/CMakeLists.txt	2020-12-04 05:27:21.000000000 +0900
+++ src/replxx-release-0.0.3/CMakeLists.txt	2021-05-09 09:21:21.888247700 +0900
@@ -111,6 +111,9 @@ else()
 endif()
 target_link_libraries(replxx PUBLIC Threads::Threads)
 set_target_properties(replxx PROPERTIES VERSION ${PROJECT_VERSION})
+if (CYGWIN)
+	set_target_properties(replxx PROPERTIES SOVERSION 0)
+endif()
 
 set_property(TARGET replxx PROPERTY DEBUG_POSTFIX -d)
 set_property(TARGET replxx PROPERTY RELWITHDEBINFO_POSTFIX -rd)
@@ -124,7 +127,7 @@ generate_export_header(replxx)
 configure_package_config_file(
 	"${PROJECT_SOURCE_DIR}/replxx-config.cmake.in"
 	"${PROJECT_BINARY_DIR}/replxx-config.cmake"
-	INSTALL_DESTINATION ${CMAKE_INSTALL_DATADIR}/cmake/replxx
+	INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/replxx
 	NO_CHECK_REQUIRED_COMPONENTS_MACRO
 	NO_SET_AND_CHECK_MACRO
 )
@@ -144,7 +147,7 @@ install(
 install(
 	EXPORT replxx-targets
 	NAMESPACE replxx::
-	DESTINATION ${CMAKE_INSTALL_DATADIR}/cmake/replxx
+	DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/replxx
 )
 
 install(
@@ -152,7 +155,7 @@ install(
 		"${PROJECT_BINARY_DIR}/replxx-config-version.cmake"
 		"${PROJECT_BINARY_DIR}/replxx-config.cmake"
 	DESTINATION
-		${CMAKE_INSTALL_DATADIR}/cmake/replxx
+		${CMAKE_INSTALL_LIBDIR}/cmake/replxx
 )
 
 # headers
@@ -218,3 +221,9 @@ set(CPACK_STRIP_FILES "ON")
 
 set(CPACK_DEBIAN_PACKAGE_SECTION "utilities")
 
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/replxx.pc.in ${CMAKE_BINARY_DIR}/replxx.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/replxx.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/replxx-release-0.0.3/replxx.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/replxx-release-0.0.3/replxx.pc.in	2021-05-09 08:46:16.765847300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: replxx
+Description: Readline and libedit replacement
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lreplxx
+Cflags: -I${includedir}
