--- origsrc/cc1541-4.2/Makefile	2024-02-28 18:14:14.000000000 +0900
+++ src/cc1541-4.2/Makefile	2024-04-04 10:20:22.311685500 +0900
@@ -32,7 +32,7 @@ cc1541.1.txt: cc1541.1.txt.in
 	sed -e "s/@@VERSION@@/$(VERSION)/g" < cc1541.1.txt.in > cc1541.1.txt
 
 cc1541.1: cc1541.1.txt
-	a2x -d manpage -f manpage cc1541.1.txt
+	a2x -d manpage -f manpage -L cc1541.1.txt
 
 man: cc1541.1
 
--- origsrc/cc1541-4.2/cc1541.1.txt.in	2024-02-28 18:14:14.000000000 +0900
+++ src/cc1541-4.2/cc1541.1.txt.in	2024-04-04 10:19:21.650404000 +0900
@@ -17,9 +17,6 @@ control over the layout of the data on t
 supports writing the file format for the Transwarp loader by 
 Krill.
 
-Find a more extensive online documentation here:
-https://acoustic-velocity.com/cc1541
-
 == Synopsis
 
 *cc1541* [_options_] image.[_d64|d71|d81_]
