--- origsrc/diffball-1.0.1/include/Makefile.am	2011-06-13 19:11:21.000000000 +0900
+++ src/diffball-1.0.1/include/Makefile.am	2012-05-15 18:14:25.577096600 +0900
@@ -1,3 +1,3 @@
 include_HEADERS = cfile.h
 diffballincdir = $(includedir)/diffball
-diffballinc_HEADERS = diffball/*.h $(top_builddir)/config.h
+diffballinc_HEADERS = $(srcdir)/diffball/*.h $(top_builddir)/config.h
--- origsrc/diffball-1.0.1/libcfile/Makefile.am	2011-07-24 16:01:01.000000000 +0900
+++ src/diffball-1.0.1/libcfile/Makefile.am	2012-05-15 18:20:11.780221600 +0900
@@ -1,5 +1,5 @@
 lib_LTLIBRARIES = libcfile.la
-AM_CFLAGS = -I$(top_builddir)/include/
+AM_CPPFLAGS = -I$(top_builddir)/include/ -I$(top_srcdir)/include
 libcfile_la_SOURCES = cfile.c
 libcfile_la_CFLAGS = \
 	$(AM_CFLAGS) \
@@ -7,4 +7,5 @@ libcfile_la_CFLAGS = \
 libcfile_la_LIBADD = \
 	$(AM_LIBADD) \
 	$(LIBLZMA_LIBS)
+libcfile_la_LDFLAGS = -no-undefined
 noinst_HEADERS = defs.h
--- origsrc/diffball-1.0.1/libdiffball/Makefile.am	2011-06-13 19:11:21.000000000 +0900
+++ src/diffball-1.0.1/libdiffball/Makefile.am	2012-05-15 18:23:29.601510600 +0900
@@ -6,8 +6,8 @@ FORMATS = bit-functions.c gdiff.c switch
 DCB_FILES  = dcbuffer.c dcb-cfh-funcs.c command_list.c
 DIFF_FILES = diff-algs.c adler32.c primes.c hash.c $(DCB_FILES) errors.c apply-patch.c api.c
 
-libdiffball_la_LDFLAGS = -lm
-libdiffball_la_CFLAGS = $(AM_CFLAGS) -I$(top_builddir)/include
+libdiffball_la_LDFLAGS = -lm -no-undefined
+libdiffball_la_CFLAGS = $(AM_CFLAGS) -I$(top_builddir)/include -I$(top_srcdir) -I$(top_srcdir)/include
 lib_LTLIBRARIES = libdiffball.la
 libdiffball_la_LIBADD = $(top_builddir)/libcfile/libcfile.la
 libdiffball_la_SOURCES = $(FORMATS) $(DIFF_FILES)
