--- origsrc/libiff/src/iffjoin/Makefile.am	2019-04-23 08:51:52.000000000 +0900
+++ src/libiff/src/iffjoin/Makefile.am	2019-04-23 09:00:21.671227700 +0900
@@ -1,7 +1,7 @@
 iffjoin.1: main.c
 	$(HELP2MAN) --output=$@ --no-info --name 'Joins an arbitrary number of IFF files into a single concatenation IFF file' --libtool ./iffjoin
 
-AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H)
+AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H) -I$(top_srcdir)/src/libiff
 
 bin_PROGRAMS = iffjoin
 noinst_HEADERS = join.h
--- origsrc/libiff/src/iffpp/Makefile.am	2019-04-23 08:51:52.000000000 +0900
+++ src/libiff/src/iffpp/Makefile.am	2019-04-23 09:47:57.872055700 +0900
@@ -1,7 +1,7 @@
 iffpp.1: main.c
 	$(HELP2MAN) --output=$@ --no-info --name 'IFF pretty printer' --libtool ./iffpp
 
-AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H)
+AM_CPPFLAGS = -DHAVE_GETOPT_H=$(HAVE_GETOPT_H) -I$(top_srcdir)/src/libiff
 
 bin_PROGRAMS = iffpp
 noinst_HEADERS = pp.h
--- origsrc/libiff/src/libiff/Makefile.am	2019-04-23 08:51:52.000000000 +0900
+++ src/libiff/src/libiff/Makefile.am	2019-04-23 08:52:34.859098000 +0900
@@ -1,3 +1,4 @@
 lib_LTLIBRARIES = libiff.la
 pkginclude_HEADERS = io.h id.h extension.h chunk.h group.h cat.h form.h list.h prop.h rawchunk.h util.h error.h iff.h ifftypes.h
 libiff_la_SOURCES = io.c id.c extension.c chunk.c group.c cat.c form.c list.c prop.c rawchunk.c util.c error.c iff.c
+libiff_la_LDFLAGS = -no-undefined
--- origsrc/libiff/tests/Makefile.am	2019-04-23 08:51:52.000000000 +0900
+++ src/libiff/tests/Makefile.am	2019-04-23 09:50:16.093329200 +0900
@@ -121,3 +121,5 @@ EXTRA_DIST = invalidcat-contentstype.sh
     invalidid1.TEST invalidid2.TEST invalidlist-contentstype.TEST invalidlist-raw.TEST invalidlist-size.TEST invalidprop-size.TEST invalidprop.TEST \
     lookupproperty-nested.TEST lookupproperty-override.TEST pp-text.TEST validcat-wildcard.TEST validlist-wildcard.TEST \
     join.HELO join.BYE invalidlist-negsize.sh invalidlist-negsize.TEST
+
+AM_CPPFLAGS = -I$(top_srcdir)/src/libiff
