--- origsrc/libxtract-0.6.3/configure.in	2010-05-11 05:59:23.000000000 +0900
+++ src/libxtract-0.6.3/configure.in	2011-08-25 07:49:05.042064600 +0900
@@ -26,7 +26,7 @@ AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
 AC_ENABLE_STATIC(no)
 AC_ENABLE_SHARED(yes)
 AC_PROG_LIBTOOL
-AC_CHECK_HEADERS([math.h, stdlib.h, stdio.h])
+AC_CHECK_HEADERS([math.h stdlib.h stdio.h])
 AC_CHECK_PROG([DOXYGEN], [doxygen], [doc], [])
 AC_SUBST(DOXYGEN)
 
--- origsrc/libxtract-0.6.3/src/Makefile.am	2010-02-07 03:49:27.000000000 +0900
+++ src/libxtract-0.6.3/src/Makefile.am	2011-08-25 08:03:26.606769900 +0900
@@ -9,9 +9,11 @@ else
 FFT_DEFINE = 
 endif
 
+INCLUDES = -I$(top_srcdir)
+
 lib_LTLIBRARIES = libxtract.la
 libxtract_la_CFLAGS = $(FFT_DEFINE)
 libxtract_la_SOURCES = $(SOURCES)
-libxtract_la_LDFLAGS = -export-dynamic
+libxtract_la_LDFLAGS = -export-dynamic -no-undefined
 
 EXTRA_DIST = xtract_globals_private.h xtract_macros_private.h xtract_window_private.h
