--- origsrc/treelite-2.1.0/CMakeLists.txt	2021-09-21 06:40:18.000000000 +0900
+++ src/treelite-2.1.0/CMakeLists.txt	2021-10-04 19:53:06.988920900 +0900
@@ -67,6 +67,9 @@ endif()
 
 foreach(lib ${TREELITE_TARGETS})
   set_output_directory(${lib} ${PROJECT_BINARY_DIR})
+  if (CYGWIN)
+    set_target_properties(${lib} PROPERTIES SOVERSION ${SOVERSION})
+  endif()
 endforeach()
 
 # Export install targets
--- origsrc/treelite-2.1.0/cmake/ExternalLibs.cmake	2021-09-21 06:40:18.000000000 +0900
+++ src/treelite-2.1.0/cmake/ExternalLibs.cmake	2021-10-04 19:32:21.305970600 +0900
@@ -39,7 +39,7 @@ add_library(RapidJSON::rapidjson ALIAS r
 
 # Google C++ tests
 if(BUILD_CPP_TEST)
-  find_package(GTest 1.11.0 CONFIG)
+  find_package(GTest 1.11.0)
   if(NOT GTEST_FOUND)
     message(STATUS "Did not find Google Test in the system root. Fetching Google Test now...")
     FetchContent_Declare(
