--- origsrc/yxml/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/yxml/Makefile.am	2024-09-20 10:46:11.655481000 +0900
@@ -0,0 +1,13 @@
+
+check_PROGRAMS = test/test
+test_test_SOURCES = test/test.c
+test_test_LDADD = libyxml.la
+
+include_HEADERS = yxml.h
+
+lib_LTLIBRARIES = libyxml.la
+libyxml_la_SOURCES = yxml.c
+libyxml_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/yxml/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/yxml/configure.ac	2024-09-20 10:40:41.875795900 +0900
@@ -0,0 +1,30 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.71])
+AC_INIT([yxml], [0])
+AC_CONFIG_SRCDIR([yxml.c.in])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_MAKE_SET
+LT_INIT
+
+# Checks for header files.
+AC_CHECK_HEADERS([fcntl.h inttypes.h stdint.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_C_INLINE
+AC_TYPE_SIZE_T
+AC_TYPE_SSIZE_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([memset])
+AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([yxml.pc])
+AC_OUTPUT
--- origsrc/yxml/yxml.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/yxml/yxml.pc.in	2024-09-20 10:40:41.875795900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: yxml
+Description: Small, fast and mostly-correct XML parser
+Version: @VERSION@
+Libs: -L${libdir} -lyxml
+Cflags: -I${includedir}
