--- origsrc/sz81-2.1.7/Makefile	2011-02-13 07:28:52.000000000 +0900
+++ src/sz81-2.1.7/Makefile	2025-08-14 10:16:28.717984900 +0900
@@ -37,7 +37,7 @@ CFLAGS?=-O3
 CFLAGS+=-Wall `$(SDL_CONFIG) --cflags` -DVERSION=\"$(VERSION)\" \
 	-DPACKAGE_DATA_DIR=\"$(PACKAGE_DATA_DIR)\" $(SOUNDDEF) -DSZ81 
 LINK=$(CC)
-LDFLAGS=
+LDFLAGS?=
 LIBS=`$(SDL_CONFIG) --libs` 
 
 # You won't need to alter anything below
