--- origsrc/libsufr-0.6.2/CMake/CompilerFlags_Fortran.cmake	2013-08-11 02:30:12.000000000 +0900
+++ src/libsufr-0.6.2/CMake/CompilerFlags_Fortran.cmake	2014-09-13 19:46:25.760189000 +0900
@@ -70,9 +70,9 @@ if( Fortran_COMPILER_NAME MATCHES "gfort
     set( WARN_FLAGS "${WARN_FLAGS} -Werror" )
   endif( STOP_ON_WARNING )
   
-  if( WANT_LIBRARY )
+  if( WANT_LIBRARY AND NOT CYGWIN )
     set( LIB_FLAGS "-fPIC -g" )
-  endif( WANT_LIBRARY )
+  endif( WANT_LIBRARY AND NOT CYGWIN )
   
   
   # Package-specific flags:
--- origsrc/libsufr-0.6.2/CMakeLists.txt	2013-11-21 22:15:12.000000000 +0900
+++ src/libsufr-0.6.2/CMakeLists.txt	2014-09-13 19:50:32.041439000 +0900
@@ -89,6 +89,7 @@ if( CREATE_SHAREDLIB )
     set_target_properties( libSUFR_shared PROPERTIES OUTPUT_NAME "SUFR" )
   endif( COMPILER_SPECIFIC_LIBS )
   set_target_properties( libSUFR_shared PROPERTIES Fortran_MODULE_DIRECTORY ${MODULE_DIRECTORY} )
+  set_target_properties( libSUFR_shared PROPERTIES SOVERSION 0 )
 endif( CREATE_SHAREDLIB )
 
 # Use add_dependencies below in order to pretend the static lib depends on the shared.  This
