--- origsrc/mio/mio/Makefile.am	2015-02-08 15:05:02.000000000 +0900
+++ src/mio/mio/Makefile.am	2015-02-08 15:05:20.411095000 +0900
@@ -6,7 +6,7 @@ if ENABLE_GLIB
 libmio_la_CFLAGS   = @GLIB_CFLAGS@
 libmio_la_LIBADD   = @GLIB_LIBS@
 endif
-libmio_la_LDFLAGS  = -version-info @MIO_LTVERSION@
+libmio_la_LDFLAGS  = -no-undefined -version-info @MIO_LTVERSION@
 
 EXTRA_DIST = mio-file.c \
              mio-memory.c
