--- origsrc/mio/Makefile.am	2010-10-09 21:36:23.000000000 +0900
+++ src/mio/Makefile.am	2010-10-09 21:43:46.373456500 +0900
@@ -1,6 +1,6 @@
 SUBDIRS = mio tests docs
 
-ACLOCAL_AMFLAGS = -I build/m4
+ACLOCAL_AMFLAGS = 
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = mio.pc
--- origsrc/mio/configure.ac	2010-10-09 21:36:23.000000000 +0900
+++ src/mio/configure.ac	2010-10-09 21:37:10.717206500 +0900
@@ -27,7 +27,7 @@ AC_SUBST([MIO_LTVERSION])
 
 # Checks for programs.
 LT_PREREQ([2.2.0])
-LT_INIT
+LT_INIT(win32-dll)
 AC_PROG_CC
 
 # check for gtk-doc
--- origsrc/mio/mio/Makefile.am	2010-10-09 21:36:23.000000000 +0900
+++ src/mio/mio/Makefile.am	2010-10-09 21:37:24.764081500 +0900
@@ -4,7 +4,7 @@ libmio_la_SOURCES  = mio.c
 libmio_la_CPPFLAGS = -DG_LOG_DOMAIN=\"MIO\" -DMIO_DEBUG
 libmio_la_CFLAGS   = @GLIB_CFLAGS@
 libmio_la_LIBADD   = @GLIB_LIBS@
-libmio_la_LDFLAGS  = -version-info @MIO_LTVERSION@
+libmio_la_LDFLAGS  = -version-info @MIO_LTVERSION@ -no-undefined
 
 mio_includedir = $(includedir)/mio
 mio_include_HEADERS = mio.h
--- origsrc/mio/tests/Makefile.am	2010-10-09 21:36:23.000000000 +0900
+++ src/mio/tests/Makefile.am	2010-10-09 22:16:20.310956500 +0900
@@ -2,6 +2,6 @@ noinst_PROGRAMS = test
 
 test_SOURCES  = main.c
 test_CPPFLAGS = 
-test_CFLAGS   = -I.. @GLIB_CFLAGS@
+test_CFLAGS   = -I$(top_srcdir) @GLIB_CFLAGS@
 test_LDFLAGS  = 
 test_LDADD    = @GLIB_LIBS@ ../mio/libmio.la
