--- origsrc/libLAS-1.8.1/CMakeLists.txt	2016-08-22 22:07:43.000000000 +0900
+++ src/libLAS-1.8.1/CMakeLists.txt	2016-12-31 02:12:31.396726800 +0900
@@ -154,7 +154,7 @@ else()
   if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
 
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${LIBLAS_COMMON_CXX_FLAGS}")
-    if (CMAKE_COMPILER_IS_GNUCXX)
+    if (CMAKE_COMPILER_IS_GNUCXX AND NOT CYGWIN)
       set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++98 -ansi")
     endif()
 
--- origsrc/libLAS-1.8.1/cmake/CMakeLists.txt	2016-08-22 22:06:34.000000000 +0900
+++ src/libLAS-1.8.1/cmake/CMakeLists.txt	2016-12-31 02:29:02.653095600 +0900
@@ -6,7 +6,7 @@
 # ${CMAKE_INSTALL_PREFIX}/${INSTALL_CMAKE_DIR} and ${PROJECT_ROOT_DIR}
 # is the relative path to the root from there.
 if (NOT WIN32)
-  set(INSTALL_CMAKE_DIR "share/cmake/${PROJECT_NAME}")
+  set(INSTALL_CMAKE_DIR "lib/cmake/${PROJECT_NAME}")
   set (PROJECT_ROOT_DIR "../../..")
 else ()
   set(INSTALL_CMAKE_DIR "cmake")
--- origsrc/libLAS-1.8.1/cmake/modules/FindLASzip.cmake	2016-08-22 22:06:34.000000000 +0900
+++ src/libLAS-1.8.1/cmake/modules/FindLASzip.cmake	2016-12-31 02:13:28.219064800 +0900
@@ -62,7 +62,7 @@ ENDIF()
 IF(LASZIP_INCLUDE_DIR)
   SET(LASZIP_VERSION 0)
 
-  SET(LASZIP_VERSION_H "${LASZIP_INCLUDE_DIR}/laszip/laszip.hpp")
+  SET(LASZIP_VERSION_H "${LASZIP_INCLUDE_DIR}/laszip.hpp")
   FILE(READ ${LASZIP_VERSION_H} LASZIP_VERSION_H_CONTENTS)
 
   IF (DEFINED LASZIP_VERSION_H_CONTENTS)
--- origsrc/libLAS-1.8.1/src/detail/reader/zipreader.cpp	2016-08-22 22:06:34.000000000 +0900
+++ src/libLAS-1.8.1/src/detail/reader/zipreader.cpp	2016-12-31 02:21:48.727581400 +0900
@@ -46,7 +46,7 @@
 #include <liblas/detail/private_utility.hpp>
 #include <liblas/detail/zippoint.hpp>
 // laszip
-#include <laszip/lasunzipper.hpp>
+#include <lasunzipper.hpp>
 // boost
 #include <boost/cstdint.hpp>
 // std
--- origsrc/libLAS-1.8.1/src/detail/writer/zipwriter.cpp	2016-08-22 22:06:34.000000000 +0900
+++ src/libLAS-1.8.1/src/detail/writer/zipwriter.cpp	2016-12-31 02:22:50.108961500 +0900
@@ -48,8 +48,8 @@
 #include <liblas/detail/private_utility.hpp>
 #include <liblas/detail/zippoint.hpp>
 // laszip
-#include <laszip/laszipper.hpp>
-#include <laszip/laszip.hpp>
+#include <laszipper.hpp>
+#include <laszip.hpp>
 // std
 #include <vector>
 #include <fstream>
--- origsrc/libLAS-1.8.1/src/detail/zippoint.cpp	2016-08-22 22:06:34.000000000 +0900
+++ src/libLAS-1.8.1/src/detail/zippoint.cpp	2016-12-31 02:20:54.926280800 +0900
@@ -45,8 +45,8 @@
 #include <liblas/detail/private_utility.hpp>
 #include <liblas/detail/zippoint.hpp>
 // laszip
-#include <laszip/laszip.hpp>
-#include <laszip/laszipper.hpp>
+#include <laszip.hpp>
+#include <laszipper.hpp>
 // std
 #include <vector>
 #include <fstream>
--- origsrc/libLAS-1.8.1/src/header.cpp	2016-08-22 22:06:34.000000000 +0900
+++ src/libLAS-1.8.1/src/header.cpp	2016-12-31 02:18:39.373477200 +0900
@@ -48,7 +48,7 @@
 
 #ifdef HAVE_LASZIP
 #include <liblas/detail/zippoint.hpp>
-#include <laszip/laszip.hpp>
+#include <laszip.hpp>
 #endif
 
 // boost
--- origsrc/libLAS-1.8.1/src/version.cpp	2016-08-22 22:06:34.000000000 +0900
+++ src/libLAS-1.8.1/src/version.cpp	2016-12-31 02:20:11.603788600 +0900
@@ -52,7 +52,7 @@
 #endif
 
 #ifdef HAVE_LASZIP
-#include <laszip/laszip.hpp>
+#include <laszip.hpp>
 #endif
 
 #include <liblas/spatialreference.hpp>
@@ -131,4 +131,3 @@ std::string GetVersion() {
 }
 
 } // namespace liblas
-
