--- origsrc/libLAS-1.8.0/CMakeLists.txt	2014-08-02 08:01:30.000000000 +0900
+++ src/libLAS-1.8.0/CMakeLists.txt	2014-09-28 21:25:55.837170200 +0900
@@ -155,7 +155,7 @@ else()
 
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${LIBLAS_COMMON_CXX_FLAGS}")
     if (CMAKE_COMPILER_IS_GNUCXX)
-      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++98 -ansi")
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++98")
     endif()
 
   elseif("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" OR "${CMAKE_CXX_COMPILER}" MATCHES "clang")
--- origsrc/libLAS-1.8.0/cmake/modules/FindLASzip.cmake	2014-08-02 07:48:16.000000000 +0900
+++ src/libLAS-1.8.0/cmake/modules/FindLASzip.cmake	2014-09-28 19:27:20.221510000 +0900
@@ -62,7 +62,7 @@ ENDIF()
 IF(LASZIP_INCLUDE_DIR)
   SET(LASZIP_VERSION 0)
 
-  SET(LASZIP_VERSION_H "${LASZIP_INCLUDE_DIR}/laszip/laszip.hpp")
+  SET(LASZIP_VERSION_H "${LASZIP_INCLUDE_DIR}/laszip.hpp")
   FILE(READ ${LASZIP_VERSION_H} LASZIP_VERSION_H_CONTENTS)
 
   IF (DEFINED LASZIP_VERSION_H_CONTENTS)
--- origsrc/libLAS-1.8.0/src/detail/reader/zipreader.cpp	2014-08-02 07:48:16.000000000 +0900
+++ src/libLAS-1.8.0/src/detail/reader/zipreader.cpp	2014-09-28 19:57:09.050202300 +0900
@@ -46,7 +46,7 @@
 #include <liblas/detail/private_utility.hpp>
 #include <liblas/detail/zippoint.hpp>
 // laszip
-#include <laszip/lasunzipper.hpp>
+#include <lasunzipper.hpp>
 // boost
 #include <boost/cstdint.hpp>
 // std
--- origsrc/libLAS-1.8.0/src/detail/writer/zipwriter.cpp	2014-08-02 07:48:16.000000000 +0900
+++ src/libLAS-1.8.0/src/detail/writer/zipwriter.cpp	2014-09-28 19:58:38.264069500 +0900
@@ -48,8 +48,8 @@
 #include <liblas/detail/private_utility.hpp>
 #include <liblas/detail/zippoint.hpp>
 // laszip
-#include <laszip/laszipper.hpp>
-#include <laszip/laszip.hpp>
+#include <laszipper.hpp>
+#include <laszip.hpp>
 // std
 #include <vector>
 #include <fstream>
--- origsrc/libLAS-1.8.0/src/detail/zippoint.cpp	2014-08-02 07:48:16.000000000 +0900
+++ src/libLAS-1.8.0/src/detail/zippoint.cpp	2014-09-28 19:50:01.899029200 +0900
@@ -45,8 +45,8 @@
 #include <liblas/detail/private_utility.hpp>
 #include <liblas/detail/zippoint.hpp>
 // laszip
-#include <laszip/laszip.hpp>
-#include <laszip/laszipper.hpp>
+#include <laszip.hpp>
+#include <laszipper.hpp>
 // std
 #include <vector>
 #include <fstream>
--- origsrc/libLAS-1.8.0/src/header.cpp	2014-08-02 07:48:16.000000000 +0900
+++ src/libLAS-1.8.0/src/header.cpp	2014-09-28 19:33:54.044752100 +0900
@@ -48,7 +48,7 @@
 
 #ifdef HAVE_LASZIP
 #include <liblas/detail/zippoint.hpp>
-#include <laszip/laszip.hpp>
+#include <laszip.hpp>
 #endif
 
 // boost
--- origsrc/libLAS-1.8.0/src/version.cpp	2014-08-02 07:48:16.000000000 +0900
+++ src/libLAS-1.8.0/src/version.cpp	2014-09-28 19:49:29.124615100 +0900
@@ -52,7 +52,7 @@
 #endif
 
 #ifdef HAVE_LASZIP
-#include <laszip/laszip.hpp>
+#include <laszip.hpp>
 #endif
 
 #include <liblas/spatialreference.hpp>
