--- origsrc/roscpp_core-0.7.2/rostime/CMakeLists.txt	2020-05-29 13:45:12.000000000 +0900
+++ src/roscpp_core-0.7.2/rostime/CMakeLists.txt	2020-07-17 14:18:11.014134800 +0900
@@ -22,6 +22,10 @@ if(NOT APPLE)
   target_link_libraries(rostime ${RT_LIBRARY})
 endif()
 
+if (CYGWIN)
+  set_target_properties(rostime PROPERTIES SOVERSION 0)
+endif()
+
 install(TARGETS rostime
   ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
   LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
