--- origsrc/unbound-1.6.0/Makefile.in	2016-12-07 00:29:35.000000000 +0900
+++ src/unbound-1.6.0/Makefile.in	2017-01-12 00:34:45.227780900 +0900
@@ -15,7 +15,7 @@ datarootdir=@datarootdir@
 datadir=@datadir@
 includedir=@includedir@
 doxygen=@doxygen@
-libtool=@libtool@
+libtool=libtool
 staticexe=@staticexe@
 EXEEXT=@EXEEXT@
 configfile=@ub_conf_file@
--- origsrc/unbound-1.6.0/configure.ac	2016-12-07 21:58:47.000000000 +0900
+++ src/unbound-1.6.0/configure.ac	2017-01-11 23:35:43.929907700 +0900
@@ -300,7 +300,7 @@ ACX_YYLEX_DESTROY
 AC_PROG_YACC
 AC_CHECK_PROG(doxygen, doxygen, doxygen)
 AC_CHECK_TOOL(STRIP, strip)
-ACX_LIBTOOL_C_ONLY
+LT_INIT
 
 # Checks for header files.
 AC_CHECK_HEADERS([stdarg.h stdbool.h netinet/in.h netinet/tcp.h sys/param.h sys/socket.h sys/un.h sys/uio.h sys/resource.h arpa/inet.h syslog.h netdb.h sys/wait.h pwd.h glob.h grp.h login_cap.h winsock2.h ws2tcpip.h endian.h],,, [AC_INCLUDES_DEFAULT])
