--- origsrc/vanessa_socket-0.0.12/libvanessa_socket/Makefile.am	2010-06-22 16:21:37.000000000 +0900
+++ src/vanessa_socket-0.0.12/libvanessa_socket/Makefile.am	2014-03-09 14:58:08.459026200 +0900
@@ -36,6 +36,6 @@ vanessa_socket_pipe.c \
 vanessa_socket_server.c \
 unused.h
 
-libvanessa_socket_la_LDFLAGS    = -version-info 3:0:1
+libvanessa_socket_la_LDFLAGS    = -version-info 3:0:1 -no-undefined
 
 libvanessa_socket_la_LIBADD = @extra_libs@ -lvanessa_logger
