--- origsrc/vanessa_socket-0.0.12/configure.ac	2010-06-22 16:22:35.000000000 +0900
+++ src/vanessa_socket-0.0.12/configure.ac	2011-05-05 17:57:18.088565400 +0900
@@ -19,13 +19,10 @@ dnl Inc., 59 Temple Place, Suite 330, Bo
 AC_INIT(libvanessa_socket/vanessa_socket.h)
 AM_CONFIG_HEADER(config.h)
 
-AC_CONFIG_MACRO_DIR([libltdl/m4])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 LT_INIT
-LTDL_INIT
 AM_INIT_AUTOMAKE(vanessa_socket, 0.0.12)
 
-AC_PROG_LIBTOOL
 AC_PROG_CC
 AC_PROG_INSTALL
 AC_PROG_LN_S
--- origsrc/vanessa_socket-0.0.12/libvanessa_socket/Makefile.am	2010-06-22 16:21:37.000000000 +0900
+++ src/vanessa_socket-0.0.12/libvanessa_socket/Makefile.am	2011-05-05 17:56:47.334343000 +0900
@@ -36,6 +36,6 @@ vanessa_socket_pipe.c \
 vanessa_socket_server.c \
 unused.h
 
-libvanessa_socket_la_LDFLAGS    = -version-info 3:0:1
+libvanessa_socket_la_LDFLAGS    = -version-info 3:0:1 -no-undefined
 
 libvanessa_socket_la_LIBADD = @extra_libs@ -lvanessa_logger
