--- origsrc/vanessa_socket-0.0.13/libvanessa_socket/Makefile.am	2014-09-11 10:23:01.000000000 +0900
+++ src/vanessa_socket-0.0.13/libvanessa_socket/Makefile.am	2016-11-27 16:51:22.371468000 +0900
@@ -36,6 +36,6 @@ vanessa_socket_pipe.c \
 vanessa_socket_server.c \
 unused.h
 
-libvanessa_socket_la_LDFLAGS    = -version-info 3:0:1
+libvanessa_socket_la_LDFLAGS    = -version-info 3:0:1 -no-undefined
 
 libvanessa_socket_la_LIBADD = @extra_libs@ -lvanessa_logger
