--- origsrc/vanessa_socket-0.0.13/libvanessa_socket/Makefile.am	2014-09-11 10:23:01.000000000 +0900
+++ src/vanessa_socket-0.0.13/libvanessa_socket/Makefile.am	2025-01-03 21:16:14.068114300 +0900
@@ -36,6 +36,6 @@ vanessa_socket_pipe.c \
 vanessa_socket_server.c \
 unused.h
 
-libvanessa_socket_la_LDFLAGS    = -version-info 3:0:1
+libvanessa_socket_la_LDFLAGS    = -version-info 3:0:1 -no-undefined
 
 libvanessa_socket_la_LIBADD = @extra_libs@ -lvanessa_logger
--- origsrc/vanessa_socket-0.0.13/vanessa-socket.pc.in	2014-09-11 10:23:01.000000000 +0900
+++ src/vanessa_socket-0.0.13/vanessa-socket.pc.in	2025-01-03 21:27:52.213106000 +0900
@@ -6,5 +6,5 @@ includedir=@includedir@
 Name: Vanessa Socket
 Description: Library to simplify TCP socket operations
 Version: @VERSION@
-Libs: -L${libdir} -lvanessa-logger0
+Libs: -L${libdir} -lvanessa_socket
 Cflags: -I${includedir}
