--- origsrc/log4g-0.1.2/log4g/Makefile.am	2010-11-05 09:40:47.000000000 +0900
+++ src/log4g-0.1.2/log4g/Makefile.am	2024-11-05 05:01:31.047996400 +0900
@@ -55,6 +55,7 @@ liblog4g_@log4g_series_major@_@log4g_ser
     -DLOG4G_MODULEDIR=\"$(moduledir)\" \
     $(liblog4g_@log4g_series_major@_@log4g_series_minor@_la_INCLUDES)
 liblog4g_@log4g_series_major@_@log4g_series_minor@_la_LDFLAGS = \
+	-no-undefined \
     -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
     $(GLIB_LIBS) \
     $(GMODULE_LIBS) \
--- origsrc/log4g-0.1.2/modules/appenders/Makefile.am	2010-11-05 09:40:47.000000000 +0900
+++ src/log4g-0.1.2/modules/appenders/Makefile.am	2024-11-05 05:35:55.632596600 +0900
@@ -1,9 +1,9 @@
 #                                        -*- Automake -*-
 # Process this file with automake to produce Makefile.in.
 
-module_LTLIBRARIES = \
-    liblog4g-appenders.la
-liblog4g_appenders_la_SOURCES = \
+pkglib_LTLIBRARIES = \
+    log4g-appenders.la
+log4g_appenders_la_SOURCES = \
     appender/async-appender.h \
     appender/console-appender.h \
     appender/file-appender.h \
@@ -23,11 +23,12 @@ liblog4g_appenders_la_SOURCES = \
     rolling-file-appender.c \
     syslog-appender.c \
     writer-appender.c
-liblog4g_appenders_la_CFLAGS = \
+log4g_appenders_la_CFLAGS = \
     -I$(srcdir) -I$(top_srcdir) \
     $(GLIB_CFLAGS) \
     $(GOBJECT_CFLAGS)
-liblog4g_appenders_la_LDFLAGS = \
+log4g_appenders_la_LDFLAGS = \
+    -module -no-undefined \
     -avoid-version -shared \
     $(top_builddir)/log4g/liblog4g-$(log4g_series).la \
     $(GLIB_LIBS) \
--- origsrc/log4g-0.1.2/modules/couchdb/Makefile.am	2010-11-05 09:40:47.000000000 +0900
+++ src/log4g-0.1.2/modules/couchdb/Makefile.am	2024-11-05 05:36:11.084707100 +0900
@@ -1,20 +1,21 @@
 #                                        -*- Automake -*-
 # Process this file with automake to produce Makefile.in.
 
-module_LTLIBRARIES = \
-    liblog4g-couchdb.la
-liblog4g_couchdb_la_SOURCES = \
+pkglib_LTLIBRARIES = \
+    log4g-couchdb.la
+log4g_couchdb_la_SOURCES = \
     appender/couchdb-appender.h \
     couchdb-appender.c \
     couchdb-layout.c \
     layout/couchdb-layout.h \
     module.c
-liblog4g_couchdb_la_CFLAGS = \
+log4g_couchdb_la_CFLAGS = \
     -I$(srcdir) -I$(top_srcdir) \
     $(COUCHDB_GLIB_CFLAGS) \
     $(GLIB_CFLAGS) \
     $(GOBJECT_CFLAGS)
-liblog4g_couchdb_la_LDFLAGS = \
+log4g_couchdb_la_LDFLAGS = \
+    -module -no-undefined \
     -avoid-version -shared \
     $(top_builddir)/log4g/liblog4g-$(log4g_series).la \
     $(COUCHDB_GLIB_LIBS) \
--- origsrc/log4g-0.1.2/modules/filters/Makefile.am	2010-11-05 09:40:47.000000000 +0900
+++ src/log4g-0.1.2/modules/filters/Makefile.am	2024-11-05 05:36:24.833265000 +0900
@@ -1,9 +1,9 @@
 #                                        -*- Automake -*-
 # Process this file with automake to produce Makefile.in.
 
-module_LTLIBRARIES = \
-    liblog4g-filters.la
-liblog4g_filters_la_SOURCES = \
+pkglib_LTLIBRARIES = \
+    log4g-filters.la
+log4g_filters_la_SOURCES = \
     deny-all-filter.c \
     filter/deny-all-filter.h \
     filter/level-range-filter.h \
@@ -13,11 +13,12 @@ liblog4g_filters_la_SOURCES = \
     level-match-filter.c \
     module.c \
     string-match-filter.c
-liblog4g_filters_la_CFLAGS = \
+log4g_filters_la_CFLAGS = \
     -I$(srcdir) -I$(top_srcdir) \
     $(GLIB_CFLAGS) \
     $(GOBJECT_CFLAGS)
-liblog4g_filters_la_LDFLAGS = \
+log4g_filters_la_LDFLAGS = \
+    -module -no-undefined \
     -avoid-version -shared \
     $(top_builddir)/log4g/liblog4g-$(log4g_series).la \
     $(GLIB_LIBS) \
--- origsrc/log4g-0.1.2/modules/layouts/Makefile.am	2010-11-05 09:40:47.000000000 +0900
+++ src/log4g-0.1.2/modules/layouts/Makefile.am	2024-11-05 05:36:41.912358100 +0900
@@ -1,9 +1,9 @@
 #                                        -*- Automake -*-
 # Process this file with automake to produce Makefile.in.
 
-module_LTLIBRARIES = \
-    liblog4g-layouts.la
-liblog4g_layouts_la_SOURCES = \
+pkglib_LTLIBRARIES = \
+    log4g-layouts.la
+log4g_layouts_la_SOURCES = \
     date-layout.c \
     html-layout.c \
     layout/date-layout.h \
@@ -21,11 +21,12 @@ liblog4g_layouts_la_SOURCES = \
     simple-layout.c \
     ttcc-layout.c \
     xml-layout.c
-liblog4g_layouts_la_CFLAGS = \
+log4g_layouts_la_CFLAGS = \
     -I$(srcdir) -I$(top_srcdir) \
     $(GLIB_CFLAGS) \
     $(GOBJECT_CFLAGS)
-liblog4g_layouts_la_LDFLAGS = \
+log4g_layouts_la_LDFLAGS = \
+    -module -no-undefined \
     -avoid-version -shared \
     $(top_builddir)/log4g/liblog4g-$(log4g_series).la \
     $(GLIB_LIBS) \
