--- origsrc/libleaconfig/Makefile.am	2024-12-07 22:41:08.000000000 +0900
+++ src/libleaconfig/Makefile.am	2024-12-07 22:41:09.563351100 +0900
@@ -1,2 +1,5 @@
 ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = include libleaconfig src
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libleaconfig/configure.ac	2024-12-07 22:41:08.000000000 +0900
+++ src/libleaconfig/configure.ac	2024-12-07 22:41:09.563351100 +0900
@@ -33,7 +33,7 @@ AC_FUNC_STRTOD
 AC_CHECK_FUNCS([memmove memset strchr strstr strtol])
 
 
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([libleaconfig.pc Makefile
 		 include/Makefile
 		 include/libleaconfig/Makefile
 		 libleaconfig/Makefile
--- origsrc/libleaconfig/libleaconfig.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libleaconfig/libleaconfig.pc.in	2024-12-07 22:41:09.563351100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libleaconfig
+Description: Library that provides basic configuration capabilities
+Version: @VERSION@
+Libs: -L${libdir} -lleaconfig
+Cflags: -I${includedir}
