--- origsrc/cmark-gfm-0.29.0.gfm.9/CMakeLists.txt	2023-02-01 01:07:31.000000000 +0900
+++ src/cmark-gfm-0.29.0.gfm.9/CMakeLists.txt	2023-03-24 21:55:09.633608500 +0900
@@ -4,7 +4,7 @@ project(cmark-gfm)
 set(PROJECT_VERSION_MAJOR 0)
 set(PROJECT_VERSION_MINOR 29)
 set(PROJECT_VERSION_PATCH 0)
-set(PROJECT_VERSION_GFM 6)
+set(PROJECT_VERSION_GFM 9)
 set(PROJECT_VERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.gfm.${PROJECT_VERSION_GFM})
 
 include("FindAsan.cmake")
--- origsrc/cmark-gfm-0.29.0.gfm.9/extensions/CMakeLists.txt	2023-02-01 01:07:31.000000000 +0900
+++ src/cmark-gfm-0.29.0.gfm.9/extensions/CMakeLists.txt	2023-03-24 21:49:51.032936300 +0900
@@ -79,7 +79,7 @@ if (CMARK_SHARED OR CMARK_STATIC)
   DESTINATION include
   )
 
-  install(EXPORT cmark-gfm-extensions DESTINATION lib${LIB_SUFFIX}/cmake-gfm-extensions)
+  install(EXPORT cmark-gfm-extensions DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/cmake-gfm-extensions)
 endif()
 
 # Feature tests
--- origsrc/cmark-gfm-0.29.0.gfm.9/src/CMakeLists.txt	2023-02-01 01:07:31.000000000 +0900
+++ src/cmark-gfm-0.29.0.gfm.9/src/CMakeLists.txt	2023-03-24 21:49:51.038949600 +0900
@@ -175,7 +175,7 @@ if(CMARK_SHARED OR CMARK_STATIC)
     DESTINATION include
     )
 
-  install(EXPORT cmark-gfm DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake)
+  install(EXPORT cmark-gfm DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/cmake-gfm)
 endif()
 
 # Feature tests
--- origsrc/cmark-gfm-0.29.0.gfm.9/test/cmark.py	2023-02-01 01:07:31.000000000 +0900
+++ src/cmark-gfm-0.29.0.gfm.9/test/cmark.py	2023-03-24 21:49:51.045009300 +0900
@@ -5,6 +5,7 @@ from ctypes import CDLL, c_char_p, c_siz
 from subprocess import *
 import platform
 import os
+import re
 
 def pipe_through_prog(prog, text):
     p1 = Popen(prog.split(), stdout=PIPE, stdin=PIPE, stderr=PIPE)
@@ -88,6 +89,8 @@ class CMark:
                 libnames = [ ["lib", ".dylib" ] ]
             elif sysname == 'Windows':
                 libnames = [ ["", ".dll"], ["lib", ".dll"] ]
+            elif re.match("^CYGWIN_NT-", sysname) :
+                libnames = [ ["cyg", ".dll"] ]
             else:
                 libnames = [ ["lib", ".so"] ]
             if not library_dir:
