--- origsrc/cddlib-094g/lib-src/Makefile.am	2012-03-23 17:43:56.000000000 +0900
+++ src/cddlib-094g/lib-src/Makefile.am	2014-07-20 15:15:15.831531200 +0900
@@ -16,4 +16,4 @@ cddtypes.h \
 setoper.h
 
 AM_CPPFLAGS = -UGMPRATIONAL
-AM_LDFLAGS = -version-info 0:0:0
+AM_LDFLAGS = -no-undefined -version-info 0:0:0
--- origsrc/cddlib-094g/lib-src-gmp/Makefile.am	2012-03-23 17:44:20.000000000 +0900
+++ src/cddlib-094g/lib-src-gmp/Makefile.am	2014-07-20 15:15:15.831531200 +0900
@@ -25,4 +25,4 @@ cddtypes_f.h \
 setoper.h
 
 AM_CPPFLAGS = -I$(gmpincludedir) -DGMPRATIONAL
-AM_LDFLAGS = -version-info 0:0:0
+AM_LDFLAGS = -no-undefined -version-info 0:0:0
--- origsrc/cddlib-094g/src/Makefile.am	2014-07-20 15:15:13.878406200 +0900
+++ src/cddlib-094g/src/Makefile.am	2014-07-20 15:15:15.847156200 +0900
@@ -32,5 +32,5 @@ cdd_both_reps_SOURCES          = cdd_bot
 # cddmathlink_SOURCES                = cddmathlink.c cddmlio.h cddmlio.c
 
 LDADD        = ../lib-src/libcdd.la
-INCLUDES     = -I../lib-src
+INCLUDES     = -I$(top_srcdir)/lib-src
 AM_CPPFLAGS  = -UGMPRATIONAL
--- origsrc/cddlib-094g/src-gmp/Makefile.am	2014-07-20 15:15:13.878406200 +0900
+++ src/cddlib-094g/src-gmp/Makefile.am	2014-07-20 15:15:15.847156200 +0900
@@ -33,6 +33,6 @@ cdd_both_reps_gmp_SOURCES          = cdd
 
 LDADD        = ../lib-src-gmp/libcddgmp.la
 AM_LDFLAGS   = -L$(gmplibdir)
-INCLUDES     = -I../lib-src-gmp
+INCLUDES     = -I$(top_srcdir)/lib-src-gmp
 INCLUDES    += -I$(gmpincludedir)
 AM_CPPFLAGS  = -DGMPRATIONAL
