--- origsrc/gemmi-0.7.4/CMakeLists.txt	2025-11-23 20:59:15.000000000 +0900
+++ src/gemmi-0.7.4/CMakeLists.txt	2026-01-08 12:48:18.297064300 +0900
@@ -266,6 +266,10 @@ else()
   endif()
 endif()
 
+if (DEFINED SOVERSION)
+  set_target_properties(gemmi_cpp PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 
 ### programs from prog/ ###
 
@@ -622,3 +626,10 @@ if (USE_PYTHON)
             DESTINATION "${Python_SITEARCH}/gemmi" COMPONENT py)
   endif()
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/gemmi.pc.in ${CMAKE_BINARY_DIR}/gemmi.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/gemmi.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/gemmi-0.7.4/gemmi.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/gemmi-0.7.4/gemmi.pc.in	2026-01-08 12:48:18.301100000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: gemmi
+Description: Macromolecular crystallography library and utilities
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lgemmi_cpp
+Cflags: -I${includedir}
--- origsrc/gemmi-0.7.4/include/gemmi/fail.hpp	2025-11-23 20:59:15.000000000 +0900
+++ src/gemmi-0.7.4/include/gemmi/fail.hpp	2026-01-08 12:48:18.303110600 +0900
@@ -39,7 +39,7 @@
 #  define GEMMI_UNLIKELY(x) (x)
 #endif
 
-#if defined(_WIN32)
+#if defined(_WIN32) || defined(__CYGWIN__)
 # if defined(GEMMI_SHARED)
 #  if defined(GEMMI_BUILD)
 #   define GEMMI_DLL __declspec(dllexport)
