--- origsrc/gemmi-0.7.0/CMakeLists.txt	2024-11-30 22:58:57.000000000 +0900
+++ src/gemmi-0.7.0/CMakeLists.txt	2024-12-25 13:53:42.070242300 +0900
@@ -265,6 +265,10 @@ else()
   endif()
 endif()
 
+if (DEFINED SOVERSION)
+  set_target_properties(gemmi_cpp PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 
 ### programs from prog/ ###
 
@@ -621,3 +625,10 @@ if (USE_PYTHON)
             DESTINATION "${Python_SITEARCH}/gemmi" COMPONENT py)
   endif()
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/gemmi.pc.in ${CMAKE_BINARY_DIR}/gemmi.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/gemmi.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/gemmi-0.7.0/gemmi.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/gemmi-0.7.0/gemmi.pc.in	2024-12-25 13:53:42.070242300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: gemmi
+Description: Macromolecular crystallography library and utilities
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lgemmi_cpp
+Cflags: -I${includedir}
