--- origsrc/fast_float-8.2.2/CMakeLists.txt	2026-01-07 11:06:44.908719500 +0900
+++ src/fast_float-8.2.2/CMakeLists.txt	2026-01-07 11:06:46.376359900 +0900
@@ -70,7 +70,7 @@ if(FASTFLOAT_INSTALL)
 
   set(FASTFLOAT_VERSION_CONFIG "${CMAKE_CURRENT_BINARY_DIR}/module/FastFloatConfigVersion.cmake")
   set(FASTFLOAT_PROJECT_CONFIG "${CMAKE_CURRENT_BINARY_DIR}/module/FastFloatConfig.cmake")
-  set(FASTFLOAT_CONFIG_INSTALL_DIR "${CMAKE_INSTALL_DATAROOTDIR}/cmake/FastFloat")
+  set(FASTFLOAT_CONFIG_INSTALL_DIR "${CMAKE_INSTALL_DATAROOTDIR}/cmake/FastFloat" CACHE PATH "Install directory for FastFloat CMake package config files")
 
   if(${CMAKE_VERSION} VERSION_LESS "3.14")
     write_basic_package_version_file("${FASTFLOAT_VERSION_CONFIG}" VERSION ${PROJECT_VERSION} COMPATIBILITY SameMajorVersion)
