--- origsrc/fast_float-5.3.0/CMakeLists.txt	2023-11-14 00:44:24.000000000 +0900
+++ src/fast_float-5.3.0/CMakeLists.txt	2023-12-08 09:49:20.993631000 +0900
@@ -52,7 +52,7 @@ if(FASTFLOAT_INSTALL)
 
   set(FASTFLOAT_VERSION_CONFIG "${CMAKE_CURRENT_BINARY_DIR}/module/FastFloatConfigVersion.cmake")
   set(FASTFLOAT_PROJECT_CONFIG "${CMAKE_CURRENT_BINARY_DIR}/module/FastFloatConfig.cmake")
-  set(FASTFLOAT_CONFIG_INSTALL_DIR "${CMAKE_INSTALL_DATAROOTDIR}/cmake/FastFloat")
+  option(FASTFLOAT_CONFIG_INSTALL_DIR "${CMAKE_INSTALL_DATAROOTDIR}/cmake/FastFloat")
 
   if(${CMAKE_VERSION} VERSION_LESS "3.14")
     write_basic_package_version_file("${FASTFLOAT_VERSION_CONFIG}" VERSION ${PROJECT_VERSION} COMPATIBILITY SameMajorVersion)
