--- origsrc/fast_float-8.1.0/CMakeLists.txt	2025-09-19 00:38:45.000000000 +0900
+++ src/fast_float-8.1.0/CMakeLists.txt	2025-11-07 00:05:06.189812000 +0900
@@ -70,7 +70,7 @@ if(FASTFLOAT_INSTALL)
 
   set(FASTFLOAT_VERSION_CONFIG "${CMAKE_CURRENT_BINARY_DIR}/module/FastFloatConfigVersion.cmake")
   set(FASTFLOAT_PROJECT_CONFIG "${CMAKE_CURRENT_BINARY_DIR}/module/FastFloatConfig.cmake")
-  set(FASTFLOAT_CONFIG_INSTALL_DIR "${CMAKE_INSTALL_DATAROOTDIR}/cmake/FastFloat")
+  option(FASTFLOAT_CONFIG_INSTALL_DIR "${CMAKE_INSTALL_DATAROOTDIR}/cmake/FastFloat")
 
   if(${CMAKE_VERSION} VERSION_LESS "3.14")
     write_basic_package_version_file("${FASTFLOAT_VERSION_CONFIG}" VERSION ${PROJECT_VERSION} COMPATIBILITY SameMajorVersion)
