--- origsrc/libcommuni-3.7.0/tests/benchmarks/benchmarks.pro	2021-12-19 07:16:35.000000000 +0900
+++ src/libcommuni-3.7.0/tests/benchmarks/benchmarks.pro	2022-05-17 16:40:14.411562500 +0900
@@ -9,4 +9,4 @@ SUBDIRS += irctextformat
 
 # - windows has problems with symbols
 # - mac with private headers (frameworks)
-!win32:!mac:SUBDIRS += ircmessagedecoder
+!win32:!mac:!cygwin:SUBDIRS += ircmessagedecoder
