--- origsrc/libechonest-1.2.1/CMakeLists.txt	2012-04-18 21:39:02.516388400 +0900
+++ src/libechonest-1.2.1/CMakeLists.txt	2012-04-18 21:42:29.677521200 +0900
@@ -26,7 +26,7 @@ set( ECHONEST_LIB_VERSION_SONAME "${ECHO
 if (CMAKE_COMPILER_IS_GNUCXX)
     ADD_DEFINITIONS( -Wall -Wundef -Wcast-align -Wchar-subscripts -Wpointer-arith
        -Wwrite-strings -Wpacked -Wformat-security -Wmissing-format-attribute
-       -Wold-style-cast -Woverloaded-virtual -Wnon-virtual-dtor -Werror )
+       -Wold-style-cast -Woverloaded-virtual -Wnon-virtual-dtor )
     if ( NOT WIN32 )
         add_definitions( -fvisibility=hidden )
     endif()
--- origsrc/libechonest-1.2.1/src/CMakeLists.txt	2011-11-21 11:47:05.000000000 +0900
+++ src/libechonest-1.2.1/src/CMakeLists.txt	2012-04-18 21:41:53.598419600 +0900
@@ -40,9 +40,11 @@ set( LIBECHONEST_H
 
 QT4_WRAP_CPP( ${LIBECHONEST_H} )
 
+if (NOT CYGWIN)
 if (CMAKE_COMPILER_IS_GNUCXX)
     add_definitions( -Werror )
 endif (CMAKE_COMPILER_IS_GNUCXX)
+endif (NOT CYGWIN)
 
 SET( OS_SPECIFIC_LINK_LIBRARIES "" )
 IF( APPLE OR MINGW )
