diff --strip-trailing-cr -uprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.log -x config.status -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libechonest-2.0.3/CMakeLists.txt src/libechonest-2.0.3/CMakeLists.txt
--- origsrc/libechonest-2.0.3/CMakeLists.txt	2013-02-17 00:17:33.000000000 +0900
+++ src/libechonest-2.0.3/CMakeLists.txt	2013-03-09 22:44:43.393607400 +0900
@@ -32,7 +32,7 @@ if (CMAKE_COMPILER_IS_GNUCXX)
     endif()
 
     # disable -Werror on ARM since Qt containers cause a lot of alignment warnings
-    if ( NOT CMAKE_SYSTEM_PROCESSOR MATCHES "arm" )
+    if ( NOT CMAKE_SYSTEM_PROCESSOR MATCHES "arm" AND NOT CYGWIN )
         add_definitions( -Werror )
     endif()
 
