--- origsrc/s2n-tls/CMakeLists.txt	2023-08-24 18:05:06.000000000 +0900
+++ src/s2n-tls/CMakeLists.txt	2023-08-24 19:25:57.521516400 +0900
@@ -160,8 +160,11 @@ set(CMAKE_C_FLAGS_DEBUGOPT "")
 
 target_compile_options(${PROJECT_NAME} PRIVATE -pedantic -std=gnu99 -Wall -Wimplicit -Wunused -Wcomment -Wchar-subscripts
         -Wuninitialized -Wshadow -Wcast-align -Wwrite-strings -Wno-deprecated-declarations -Wno-unknown-pragmas -Wformat-security
-        -Wno-missing-braces -Wsign-compare -Wno-strict-prototypes -Wa,--noexecstack
+        -Wno-missing-braces -Wsign-compare -Wno-strict-prototypes
 )
+if (NOT CYGWIN)
+    target_compile_options(${PROJECT_NAME} PRIVATE -Wa,--noexecstack)
+endif()
 
 if (UNSAFE_TREAT_WARNINGS_AS_ERRORS)
     target_compile_options(${PROJECT_NAME} PRIVATE -Werror )
@@ -181,7 +184,7 @@ if(S2N_LTO)
     endif()
 endif()
 
-if(NOT APPLE)
+if(NOT APPLE AND NOT CYGWIN)
     set(CMAKE_SHARED_LINKER_FLAGS -Wl,-z,noexecstack,-z,relro,-z,now)
 endif()
 
@@ -664,14 +667,21 @@ else()
 endif()
 
 install(EXPORT "${PROJECT_NAME}-targets"
-        DESTINATION "${CMAKE_INSTALL_LIBDIR}/${PROJECT_NAME}/cmake/${TARGET_DIR}"
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
         NAMESPACE AWS::
         COMPONENT Development)
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake"
-        DESTINATION "${CMAKE_INSTALL_LIBDIR}/${PROJECT_NAME}/cmake/"
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
         COMPONENT Development)
 
 install(FILES "cmake/modules/Findcrypto.cmake"
-        DESTINATION "${CMAKE_INSTALL_LIBDIR}/${PROJECT_NAME}/cmake/modules/"
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
         COMPONENT Development)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/s2n.pc.in ${CMAKE_BINARY_DIR}/s2n.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/s2n.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/s2n-tls/cmake/modules/Findcrypto.cmake	2023-08-24 18:05:06.000000000 +0900
+++ src/s2n-tls/cmake/modules/Findcrypto.cmake	2023-08-24 18:07:14.433902800 +0900
@@ -40,7 +40,7 @@ else()
     )
 
     find_library(crypto_SHARED_LIBRARY
-        NAMES libcrypto.so libcrypto.dylib
+        NAMES libcrypto.dll.a libcrypto.so libcrypto.dylib
         HINTS
         "${CMAKE_PREFIX_PATH}"
         "${CMAKE_INSTALL_PREFIX}"
--- origsrc/s2n-tls/cmake/s2n-config.cmake	2023-08-24 18:05:06.000000000 +0900
+++ src/s2n-tls/cmake/s2n-config.cmake	2023-08-24 18:08:12.452549900 +0900
@@ -8,19 +8,4 @@ endif()
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/modules")
 find_dependency(crypto)
 
-# Allow static or shared lib to be used.
-# If both are installed, choose based on BUILD_SHARED_LIBS.
-if (BUILD_SHARED_LIBS)
-    if (EXISTS "${CMAKE_CURRENT_LIST_DIR}/shared")
-        include(${CMAKE_CURRENT_LIST_DIR}/shared/@PROJECT_NAME@-targets.cmake)
-    else()
-        include(${CMAKE_CURRENT_LIST_DIR}/static/@PROJECT_NAME@-targets.cmake)
-    endif()
-else()
-    if (EXISTS "${CMAKE_CURRENT_LIST_DIR}/static")
-        include(${CMAKE_CURRENT_LIST_DIR}/static/@PROJECT_NAME@-targets.cmake)
-    else()
-        include(${CMAKE_CURRENT_LIST_DIR}/shared/@PROJECT_NAME@-targets.cmake)
-    endif()
-endif()
-
+include(${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@-targets.cmake)
--- origsrc/s2n-tls/s2n.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/s2n-tls/s2n.pc.in	2023-08-24 19:25:29.061693400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: s2n
+Description: Implementation of the TLS/SSL protocols
+Version: @VERSION@
+Libs: -L${libdir} -ls2n
+Cflags: -I${includedir}
