--- origsrc/libkpass-6/Makefile.am	2013-08-24 03:06:59.000000000 +0900
+++ src/libkpass-6/Makefile.am	2014-08-02 19:27:45.539656300 +0900
@@ -2,7 +2,8 @@ TESTS=test1
 check_PROGRAMS = test1 test2 test3
 lib_LTLIBRARIES = libkpass.la
 libkpass_la_SOURCES = src/kpass.c
-libkpass_la_LDFLAGS = -version-info 1:0:0
+libkpass_la_LDFLAGS = -no-undefined -version-info 1:0:0
+libkpass_la_LIBADD = $(LTLIBINTL)
 include_HEADERS = src/kpass.h
 test1_SOURCES  = src/test1.c
 test1_LDFLAGS = $(LTLIBINTL)
