--- origsrc/urjtag-2021.03/bindings/python/setup.py	2021-03-22 04:59:30.000000000 +0900
+++ src/urjtag-2021.03/bindings/python/setup.py	2025-12-05 13:30:00.277524900 +0900
@@ -9,9 +9,9 @@ setup(name="urjtag",
       version="2021.03",
       description="urJtag Python Bindings",
       ext_modules=[
-        Extension("urjtag", ["./chain.c", "./register.c"],
+        Extension("urjtag", ["/tmp/urjtag/urjtag-2021.03-1bl2.x86_64/src/urjtag-2021.03/bindings/python/chain.c", "/tmp/urjtag/urjtag-2021.03-1bl2.x86_64/src/urjtag-2021.03/bindings/python/register.c"],
                   define_macros=[('HAVE_CONFIG_H', None)],
-                  include_dirs=['../..', '../../include', '../..'],
+                  include_dirs=['../../include', '/tmp/urjtag/urjtag-2021.03-1bl2.x86_64/src/urjtag-2021.03', '/tmp/urjtag/urjtag-2021.03-1bl2.x86_64/src/urjtag-2021.03/include/urjtag', '/tmp/urjtag/urjtag-2021.03-1bl2.x86_64/src/urjtag-2021.03/include', '../..'],
                   library_dirs=['../../src/.libs'],
                   libraries=libraries)
          ])
--- origsrc/urjtag-2021.03/bindings/python/setup.py.in	2018-03-24 21:27:41.000000000 +0900
+++ src/urjtag-2021.03/bindings/python/setup.py.in	2025-12-05 13:28:02.212763800 +0900
@@ -11,7 +11,7 @@ setup(name="urjtag",
       ext_modules=[
         Extension("urjtag", ["@srcdir@/chain.c", "@srcdir@/register.c"],
                   define_macros=[('HAVE_CONFIG_H', None)],
-                  include_dirs=['@top_srcdir@', '@top_srcdir@/include', '@top_builddir@'],
+                  include_dirs=['@top_builddir@/include', '@top_srcdir@', '@top_srcdir@/include/urjtag', '@top_srcdir@/include', '@top_builddir@'],
                   library_dirs=['@top_builddir@/src/.libs'],
                   libraries=libraries)
          ])
