--- origsrc/pinelog/Makefile.am	2024-12-10 01:39:47.000000000 +0900
+++ src/pinelog/Makefile.am	2024-12-10 02:42:13.072666200 +0900
@@ -12,12 +12,14 @@ EXTRA_DIST = \
 	README.md \
 	src/pinelog.h
 
-noinst_LTLIBRARIES = libpinelog.la
+lib_LTLIBRARIES = libpinelog.la
+
+include_HEADERS = src/pinelog.h
 
 # pinelog logging library
 libpinelog_la_SOURCES = src/pinelog.c
 libpinelog_la_CFLAGS = @PINELOG_CFLAGS@ $(WARN_CFLAGS) -I $(top_builddir)
-libpinelog_la_LDFLAGS = $(WARN_LDFLAGS)
+libpinelog_la_LDFLAGS = -no-undefined $(WARN_LDFLAGS)
 
 test_SRCFILES = test/test_pinelog.c $(libpinelog_la_SOURCES)
 bench_SRCFILES = test/bench_pinelog.c $(libpinelog_la_SOURCES)
@@ -229,3 +231,5 @@ str_bench_nots_nolvl_notr_CFLAGS = $(WAR
 	-DPINELOG_SHOW_DATE=0 -DPINELOG_SHOW_LEVEL=0 -DPINELOG_SHOW_BACKTRACE=0
 str_bench_nots_nolvl_notr_LDFLAGS = $(WARN_LDFLAGS)
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/pinelog/configure.ac	2024-12-10 01:39:47.000000000 +0900
+++ src/pinelog/configure.ac	2024-12-10 02:41:56.142750700 +0900
@@ -20,4 +20,5 @@ AX_COMPILER_FLAGS
 AC_SUBST([PINELOG_CFLAGS])
 
 AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([pinelog.pc])
 AC_OUTPUT
--- origsrc/pinelog/pinelog.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/pinelog/pinelog.pc.in	2024-12-10 02:41:56.111487800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: pinelog
+Description: Lightweight logging API for C programs
+Version: @VERSION@
+Libs: -L${libdir} -lpinelog
+Cflags: -I${includedir}
