--- origsrc/scws-1.2.3/Makefile.am	2016-02-24 11:40:03.000000000 +0900
+++ src/scws-1.2.3/Makefile.am	2020-10-13 19:21:10.101706200 +0900
@@ -28,3 +28,5 @@ dist-web: dist-bzip2
 	cd etc && tar czf ../web/down/rules.tgz *.ini
 	make sync-web
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/scws-1.2.3/cli/Makefile.am	2016-02-24 11:40:03.000000000 +0900
+++ src/scws-1.2.3/cli/Makefile.am	2020-10-13 19:18:16.739279500 +0900
@@ -3,9 +3,9 @@
 
 LIBTOOL = @LIBTOOL@ $(QUIET:y=--quiet)
 
-libscwsdir = $(top_srcdir)/libscws
+libscwsdir = $(top_builddir)/libscws
 
-INCLUDES = -I$(top_srcdir) -I$(libscwsdir)
+INCLUDES =  -I$(libscwsdir) -I$(top_srcdir)/libscws -I$(top_srcdir)
 
 bin_PROGRAMS = scws scws-gen-dict
 
--- origsrc/scws-1.2.3/configure.ac	2016-02-24 11:40:03.000000000 +0900
+++ src/scws-1.2.3/configure.ac	2020-10-13 19:21:06.202731200 +0900
@@ -94,5 +94,5 @@ AC_FUNC_MMAP
 AC_FUNC_MEMCMP
 AC_CHECK_FUNCS([flock gettimeofday malloc memset munmap pow realpath strcasecmp strchr strdup strrchr strndup strtok_r])
 
-AC_CONFIG_FILES([Makefile cli/Makefile etc/Makefile libscws/Makefile libscws/version.h])
+AC_CONFIG_FILES([scws.pc Makefile cli/Makefile etc/Makefile libscws/Makefile libscws/version.h])
 AC_OUTPUT
--- origsrc/scws-1.2.3/scws.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/scws-1.2.3/scws.pc.in	2020-10-13 19:21:06.059565700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: scws
+Description: Simple Chinese Word Segmentation
+Version: @VERSION@
+Libs: -L${libdir} -lscws
+Cflags: -I${includedir}
