--- origsrc/gcstr-4.0.0/CMakeLists.txt	2021-06-01 06:38:51.000000000 +0900
+++ src/gcstr-4.0.0/CMakeLists.txt	2021-06-22 08:24:57.945206900 +0900
@@ -31,10 +31,16 @@ include_directories(${GC_INCLUDE_DIRS})
 target_link_libraries(${PROJECT_NAME} ${GC_LIBRARIES})
 
 install(TARGETS ${PROJECT_NAME}
-  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin
+  ARCHIVE DESTINATION lib
   PUBLIC_HEADER DESTINATION include)
 
-set(INSTALL_CMAKE_DIR ${CMAKE_INSTALL_PREFIX}/share/cmake/${PROJECT_NAME})
+if (CYGWIN)
+  set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION 0)
+  set(INSTALL_CMAKE_DIR ${CMAKE_INSTALL_PREFIX}/lib/cmake/${PROJECT_NAME})
+else()
+  set(INSTALL_CMAKE_DIR ${CMAKE_INSTALL_PREFIX}/share/cmake/${PROJECT_NAME})
+endif()
 
 set(INSTALL_PC_DIR ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
 
