--- origsrc/libcacard-2.6.1/Makefile.am	2018-08-23 00:10:11.000000000 +0900
+++ src/libcacard-2.6.1/Makefile.am	2019-01-23 16:28:04.180239000 +0900
@@ -63,7 +63,7 @@ pkgconfig_DATA = libcacard.pc
 
 include $(srcdir)/build-aux/glib-tap.mk
 
-noinst_PROGRAMS += vscclient
+bin_PROGRAMS = vscclient
 vscclient_SOURCES = src/vscclient.c
 vscclient_LDADD = libcacard.la $(GTHREAD_LIBS) $(PCSC_LIBS)
 vscclient_CFLAGS = $(AM_CPPFLAGS) $(GTHREAD_CFLAGS) $(PCSC_CFLAGS)
