--- origsrc/regtools-0.6.0/CMakeLists.txt	2020-04-18 02:07:45.000000000 +0900
+++ src/regtools-0.6.0/CMakeLists.txt	2020-12-06 21:41:37.536906300 +0900
@@ -11,8 +11,8 @@ include(TestHelper)
 
 #versioning stuff
 set (regtools_VERSION_MAJOR 0)
-set (regtools_VERSION_MINOR 5)
-set (regtools_VERSION_PATCH 2)
+set (regtools_VERSION_MINOR 6)
+set (regtools_VERSION_PATCH 0)
 
 configure_file (
     "${PROJECT_SOURCE_DIR}/src/version.h.in"
