--- origsrc/libclocale-20220107/Makefile.am	2022-01-07 13:45:36.000000000 +0900
+++ src/libclocale-20220107/Makefile.am	2022-03-05 13:20:20.944759600 +0900
@@ -3,9 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
 	libclocale \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libclocale-20220107/configure.ac	2022-01-07 13:45:36.000000000 +0900
+++ src/libclocale-20220107/configure.ac	2022-03-05 13:20:20.948734400 +0900
@@ -94,7 +94,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
 AC_CONFIG_FILES([libclocale/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
--- origsrc/libclocale-20220107/libclocale/Makefile.am	2022-01-07 13:45:36.000000000 +0900
+++ src/libclocale-20220107/libclocale/Makefile.am	2022-03-05 13:20:20.964245800 +0900
@@ -1,5 +1,6 @@
 AM_CPPFLAGS = \
 	-DLOCALEDIR=\"$(datadir)/locale\" \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libclocale-20220107/libclocale/libclocale_codepage.h	2022-01-07 13:45:36.000000000 +0900
+++ src/libclocale-20220107/libclocale/libclocale_codepage.h	2022-03-05 13:20:20.971874300 +0900
@@ -32,12 +32,7 @@
 extern "C" {
 #endif
 
-#if defined( __CYGWIN__ ) && ( __GNUC__ < 10 )
-int libclocale_codepage;
-#else
-LIBCLOCALE_EXTERN_VARIABLE \
-int libclocale_codepage;
-#endif
+extern int libclocale_codepage;
 
 LIBCLOCALE_EXTERN \
 int libclocale_codepage_get(
--- origsrc/libclocale-20220107/tests/Makefile.am	2022-01-07 13:45:36.000000000 +0900
+++ src/libclocale-20220107/tests/Makefile.am	2022-03-05 13:20:20.974820400 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
