--- origsrc/libclocale-20220107/Makefile.am	2022-01-07 13:45:36.000000000 +0900
+++ src/libclocale-20220107/Makefile.am	2022-03-08 09:44:49.418027800 +0900
@@ -3,9 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
 	libclocale \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libclocale-20220107/configure.ac	2022-01-07 13:45:36.000000000 +0900
+++ src/libclocale-20220107/configure.ac	2022-03-08 09:44:49.423043400 +0900
@@ -94,7 +94,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
 AC_CONFIG_FILES([libclocale/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
--- origsrc/libclocale-20220107/include/libclocale/extern.h	2022-01-07 13:45:36.000000000 +0900
+++ src/libclocale-20220107/include/libclocale/extern.h	2022-03-08 11:07:27.165459300 +0900
@@ -30,7 +30,7 @@
  * Otherwise use default extern statement
  */
 #if defined( LIBCLOCALE_DLL_EXPORT )
-#define LIBCLOCALE_EXTERN __declspec(dllexport)
+#define LIBCLOCALE_EXTERN extern __declspec(dllexport)
 
 #elif defined( LIBCLOCALE_DLL_IMPORT )
 #define LIBCLOCALE_EXTERN extern __declspec(dllimport)
--- origsrc/libclocale-20220107/libclocale/Makefile.am	2022-01-07 13:45:36.000000000 +0900
+++ src/libclocale-20220107/libclocale/Makefile.am	2022-03-08 09:44:49.428064900 +0900
@@ -1,5 +1,6 @@
 AM_CPPFLAGS = \
 	-DLOCALEDIR=\"$(datadir)/locale\" \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libclocale-20220107/libclocale/libclocale_codepage.h	2022-01-07 13:45:36.000000000 +0900
+++ src/libclocale-20220107/libclocale/libclocale_codepage.h	2022-03-08 10:35:25.988691000 +0900
@@ -32,12 +32,8 @@
 extern "C" {
 #endif
 
-#if defined( __CYGWIN__ ) && ( __GNUC__ < 10 )
-int libclocale_codepage;
-#else
 LIBCLOCALE_EXTERN_VARIABLE \
 int libclocale_codepage;
-#endif
 
 LIBCLOCALE_EXTERN \
 int libclocale_codepage_get(
--- origsrc/libclocale-20220107/libclocale/libclocale_extern.h	2022-01-07 13:45:36.000000000 +0900
+++ src/libclocale-20220107/libclocale/libclocale_extern.h	2022-03-08 10:48:52.456412700 +0900
@@ -30,7 +30,7 @@
 
 #include <libclocale/extern.h>
 
-#if defined( __CYGWIN__ ) || defined( __MINGW32__ )
+#if defined( __MINGW32__ )
 #define LIBCLOCALE_EXTERN_VARIABLE	extern
 #else
 #define LIBCLOCALE_EXTERN_VARIABLE	LIBCLOCALE_EXTERN
--- origsrc/libclocale-20220107/tests/Makefile.am	2022-01-07 13:45:36.000000000 +0900
+++ src/libclocale-20220107/tests/Makefile.am	2022-03-08 09:44:49.445071900 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
