--- origsrc/libsass-3.3.6/include/sass/version.h	2016-04-23 15:43:23.000000000 +0900
+++ src/libsass-3.3.6/include/sass/version.h	2016-04-24 16:25:28.330913300 +0900
@@ -2,11 +2,11 @@
 #define SASS_VERSION_H
 
 #ifndef LIBSASS_VERSION
-#define LIBSASS_VERSION "[NA]"
+#define LIBSASS_VERSION "na"
 #endif
 
 #ifndef LIBSASS_LANGUAGE_VERSION
-#define LIBSASS_LANGUAGE_VERSION "[NA]"
+#define LIBSASS_LANGUAGE_VERSION "3.4"
 #endif
 
 #endif
