--- origsrc/aws-c-compression-0.3.1/CMakeLists.txt	2025-01-29 08:10:38.000000000 +0900
+++ src/aws-c-compression-0.3.1/CMakeLists.txt	2025-04-08 17:05:08.240183400 +0900
@@ -50,6 +50,10 @@ aws_add_sanitizers(${PROJECT_NAME})
 # We are not ABI stable yet
 set_target_properties(${PROJECT_NAME} PROPERTIES VERSION 1.0.0)
 
+if (DEFINED SOVERSION)
+    set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 target_include_directories(${PROJECT_NAME} PUBLIC
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
         $<INSTALL_INTERFACE:include>)
@@ -70,7 +74,7 @@ else()
 endif()
 
 install(EXPORT "${PROJECT_NAME}-targets"
-        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}/${TARGET_DIR}"
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
         NAMESPACE AWS::
         COMPONENT Development)
 
@@ -91,3 +95,10 @@ include(CTest)
 if (BUILD_TESTING)
     add_subdirectory(tests)
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/aws-c-compression.pc.in ${CMAKE_BINARY_DIR}/aws-c-compression.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/aws-c-compression.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/aws-c-compression-0.3.1/aws-c-compression.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/aws-c-compression-0.3.1/aws-c-compression.pc.in	2025-04-08 17:05:15.049916000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: aws-c-compression
+Description: C99 implementation of huffman encoding/decoding
+Version: @VERSION@
+Libs: -L${libdir} -laws-c-compression
+Cflags: -I${includedir}
--- origsrc/aws-c-compression-0.3.1/cmake/aws-c-compression-config.cmake	2025-01-29 08:10:38.000000000 +0900
+++ src/aws-c-compression-0.3.1/cmake/aws-c-compression-config.cmake	2025-04-08 16:58:35.445785700 +0900
@@ -2,21 +2,4 @@ include(CMakeFindDependencyMacro)
 
 find_dependency(aws-c-common)
 
-macro(aws_load_targets type)
-    include(${CMAKE_CURRENT_LIST_DIR}/${type}/@PROJECT_NAME@-targets.cmake)
-endmacro()
-
-# try to load the lib follow BUILD_SHARED_LIBS. Fall back if not exist.
-if (BUILD_SHARED_LIBS)
-    if (EXISTS "${CMAKE_CURRENT_LIST_DIR}/shared")
-        aws_load_targets(shared)
-    else()
-        aws_load_targets(static)
-    endif()
-else()
-    if (EXISTS "${CMAKE_CURRENT_LIST_DIR}/static")
-        aws_load_targets(static)
-    else()
-        aws_load_targets(shared)
-    endif()
-endif()
+include(${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@-targets.cmake)
