--- origsrc/unittest-cpp-1.6.1/UnitTest++/Makefile.am	2016-05-05 12:48:32.000000000 +0900
+++ src/unittest-cpp-1.6.1/UnitTest++/Makefile.am	2016-06-27 20:54:30.321353400 +0900
@@ -12,6 +12,4 @@ nobase_pkginclude_HEADERS += UnitTest++/
 UnitTest___libUnitTest___la_SOURCES += UnitTest++/Posix/SignalTranslator.cpp UnitTest++/Posix/TimeHelpers.cpp
 endif
 
-UnitTest___libUnitTest___la_LDFLAGS = -version-number @LIBUNITTEST_SO_VERSION@
-
-
+UnitTest___libUnitTest___la_LDFLAGS = -no-undefined -version-number @LIBUNITTEST_SO_VERSION@
--- origsrc/unittest-cpp-1.6.1/UnitTest++/Posix/SignalTranslator.cpp	2016-05-05 12:48:32.000000000 +0900
+++ src/unittest-cpp-1.6.1/UnitTest++/Posix/SignalTranslator.cpp	2016-06-27 21:00:32.469538200 +0900
@@ -1,5 +1,10 @@
 #include "SignalTranslator.h"
 
+extern "C" {
+    int sigaction(int, const struct sigaction *, struct sigaction *);
+    int sigemptyset(sigset_t *);
+}
+
 namespace UnitTest {
 
    sigjmp_buf* SignalTranslator::s_jumpTarget = 0;
--- origsrc/unittest-cpp-1.6.1/configure.ac	2016-05-05 12:48:32.000000000 +0900
+++ src/unittest-cpp-1.6.1/configure.ac	2016-06-27 20:55:45.874044500 +0900
@@ -3,7 +3,7 @@
 
 AC_PREREQ([2.69])
 AC_INIT([UnitTest++],
-        m4_esyscmd_s([git describe --tags | cut -c2-]),
+        [1.6.1],
         [pjohnmeyer@gmail.com],
         [unittest-cpp])
 
