--- origsrc/unittest-cpp-2.0.0/UnitTest++/Makefile.am	2017-01-14 06:55:01.000000000 +0900
+++ src/unittest-cpp-2.0.0/UnitTest++/Makefile.am	2025-01-25 18:36:49.435480500 +0900
@@ -12,6 +12,6 @@ nobase_pkginclude_HEADERS += UnitTest++/
 UnitTest___libUnitTest___la_SOURCES += UnitTest++/Posix/SignalTranslator.cpp UnitTest++/Posix/TimeHelpers.cpp
 endif
 
-UnitTest___libUnitTest___la_LDFLAGS = -version-number @LIBUNITTEST_SO_VERSION@
+UnitTest___libUnitTest___la_LDFLAGS = -no-undefined -version-number @LIBUNITTEST_SO_VERSION@
 
 
