--- origsrc/tqsllib-2.2/Makefile.am	2005-03-03 21:12:46.000000000 +0900
+++ src/tqsllib-2.2/Makefile.am	2015-04-08 15:11:52.036289800 +0900
@@ -8,10 +8,10 @@ datadir = @datadir@/tqsl
 docdir = @datadir@/doc
 
 DLLVER=@DLLVER@
-INCLUDES = -I$(OPENSSL)/include -I$(EXPAT)/include -I$(ZLIB)/include
+INCLUDES = @LIBCRYPTO_CFLAGS@ @ZLIB_CFLAGS@
 
 AM_CXXFLAGS = -g3 -O -Wall $(LOTW_SERVER) $(PLATFORM) -DCONFDIR=\"$(datadir)\"
-AM_LDFLAGS = @ldflags@ -L$(OPENSSL)/lib -L$(EXPAT)/lib -L$(ZLIB)/lib
+AM_LDFLAGS = @ldflags@ @LIBCRYPTO_LIBS@ @ZLIB_LIBS@
 
 EXTRA_LIBRARIES = libtqsllibd.a
 lib_LIBRARIES = @add_libraries@
@@ -20,7 +20,7 @@ libtqsllib_la_SOURCES = tqsllib.cpp open
 	tqslconvert.cpp cabrillo.cpp location.h tqsllib-doc.h xml.h \
 	openssl_cert.h tqslexc.h
 include_HEADERS = tqsllib.h tqslerrno.h cabrillo.h adif.h tqslconvert.h
-libtqsllib_la_LDFLAGS = -version-info @SHLIBVER@:@TQSLLIB_VERSION_MINOR@:0
+libtqsllib_la_LDFLAGS = -version-info @SHLIBVER@:@TQSLLIB_VERSION_MINOR@:0 -no-undefined
 
 EXTRA_PROGRAMS = gen_crq load_cert station_loc converter
 noinst_PROGRAMS = $(make_extra)
@@ -57,6 +57,9 @@ EXTRA_DIST = config.xml LICENSE tqsllib.
 
 MOSTLYCLEANFILES = tqsllib$(DLLVER).dll tqsllib.def tqsllib.exports
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = tqsllib.pc
+
 tqsllib-resources.o: tqsllib.rc
 	windres -i tqsllib.rc -o tqsllib-resources.o $(RESFLAGS)
 
@@ -76,4 +79,3 @@ rsync-exclude:
 install-exec-local:
 	$(mkinstalldirs) $(DESTDIR)$(bindir)
 	test "$(build_os)" != "mingw32" || $(INSTALL_PROGRAM) tqsllib$(DLLVER).dll $(DESTDIR)$(bindir)/tqsllib$(DLLVER).dll
-
--- origsrc/tqsllib-2.2/configure.in	2010-09-30 00:09:55.000000000 +0900
+++ src/tqsllib-2.2/configure.in	2015-04-08 14:54:32.918288700 +0900
@@ -55,14 +55,6 @@ AC_ARG_ENABLE(docs,
 AC_MSG_RESULT($docok)
 AM_CONDITIONAL(NODOC, test x$docok = xno)
 
-checkssldir(){
-  if test -z "$OPENSSL"; then
-    if test -e "$1/lib/libcrypto.a" || test -e "$1/lib/libcrypto$dyn_lib_suf"; then
-      OPENSSL="$1"
-    fi
-  fi
-}
-
 checkexpatdir(){
   if test -z "$EXPAT"; then
     if test -e "$1/lib/libexpat.a" || test -e "$1/lib/libexpat$dyn_lib_suf"; then
@@ -71,100 +63,15 @@ checkexpatdir(){
   fi
 }
 
-checkzlibdir(){
-  if test -z "$ZLIB"; then
-    if test -e "$1/lib/libz.a" || test -e "$1/lib/libz$dyn_lib_suf"; then
-      ZLIB="$1"
-    fi
-  fi
-}
-
 dnl Check for OpenSSL
-AC_MSG_CHECKING([for OpenSSL library (libcrypto.a)])
-AC_ARG_WITH(openssl,
-  [AC_HELP_STRING([--with-openssl=DIR], [location of OpenSSL files])],
-  [ checkssldir "$withval" ],
-  [ checkssldir "$HOME/openssl"; checkssldir "/usr"; checkssldir "/usr/local";
-    checkssldir "/usr/local/ssl"; checkssldir "../../openssl" ]
-)
-if test -z "$OPENSSL"; then
-  AC_MSG_RESULT(Not found)
-  echo
-  echo "OpenSSL not found"
-  echo "Use --with-openssl option to fix this problem"
-  echo
-  exit 1
-fi
-AC_MSG_RESULT($OPENSSL)
-AC_SUBST(OPENSSL)
-
-dnl Check for an openssl header file
-AC_MSG_CHECKING([for OpenSSL headers in $OPENSSL/include])
-if test ! -r "$OPENSSL/include/openssl/opensslv.h"; then
-  AC_MSG_RESULT(Not found)
-  echo
-  echo OpenSSL headers appear to be mising
-  echo
-  exit 1
-fi
-AC_MSG_RESULT(yes)
-
-dnl Check for expat
-AC_MSG_CHECKING([for expat library (libexpat.a)])
-AC_ARG_WITH(expat,
-  [AC_HELP_STRING([--with-expat=DIR], [location of expat library files])],
-  [ checkexpatdir "$withval" ],
-  [ checkexpatdir "$HOME/expat"; checkexpatdir "/usr"; checkexpatdir "/usr/local";
-    checkexpatdir "../../expat" ]
-)
-if test -z "$EXPAT"; then
-  AC_MSG_RESULT(Not found)
-  echo
-  echo "expat not found"
-  echo "Use --with-expat option to fix this problem"
-  echo
-  exit 1
-fi
-AC_MSG_RESULT($EXPAT)
-AC_SUBST(EXPAT)
+PKG_CHECK_MODULES(LIBCRYPTO, libcrypto)
 
 dnl Check for zlib
-AC_MSG_CHECKING([for zlib library (libz.a)])
-AC_ARG_WITH(zlib,
-  [AC_HELP_STRING([--with-zlib=DIR], [location of zlib library files])],
-  [ checkzlibdir "$withval" ],
-  [ checkzlibdir "$HOME/zlib"; checkzlibdir "/usr"; checkzlibdir "/usr/local";
-    checkzlibdir "../../zlib" ]
-)
-if test -z "$ZLIB"; then
-  AC_MSG_RESULT(Not found)
-  echo
-  echo "zlib not found"
-  echo "Use --with-zlib option to fix this problem"
-  echo
-  exit 1
-fi
-AC_MSG_RESULT($ZLIB)
-AC_SUBST(ZLIB)
-
-dnl Check for an expat header file
-AC_MSG_CHECKING([for expat headers in $EXPAT/include])
-if test ! -r "$EXPAT/include/expat.h"; then
-  AC_MSG_RESULT(Not found)
-  echo
-  echo expat headers appear to be mising
-  echo
-  exit 1
-fi
-AC_MSG_RESULT(yes)
-
-export LIBRARY_PATH="$OPENSSL/lib:$EXPAT/lib:$ZLIB/lib"
+PKG_CHECK_MODULES(ZLIB, zlib)
 
 dnl Checks for libraries.
 AC_CHECK_LIB(gdi32, main)
-AC_CHECK_LIB(crypto, MD5_Init)
 AC_CHECK_LIB(expat, XML_ParserCreate)
-AC_CHECK_LIB(z, gzopen)
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
@@ -196,4 +103,4 @@ else
 fi
 AC_SUBST(DLLVER)
 
-AC_OUTPUT(Makefile tqsllib.rc tqsllib.spec)
+AC_OUTPUT(Makefile tqsllib.pc tqsllib.rc tqsllib.spec)
--- origsrc/tqsllib-2.2/openssl_cert.cpp	2010-09-30 00:09:56.000000000 +0900
+++ src/tqsllib-2.2/openssl_cert.cpp	2015-04-08 14:40:40.590020800 +0900
@@ -1583,6 +1583,10 @@ tqsl_exportPKCS12File(tQSL_Cert cert, co
 
 	/* Form into PKCS12 data */
 	p12 = PKCS12_init(NID_pkcs7_data);
+#if defined(__CYGWIN__)
+#undef CHECKED_PTR_OF
+#define CHECKED_PTR_OF(type, p) ((STACK_OF(OPENSSL_BLOCK)*) (1 ? p : (type*)0))
+#endif
 #if (OPENSSL_VERSION_NUMBER & 0xfffff000) >= 0x00908000
 	ASN1_seq_pack_PKCS7(safes, (int(*)(PKCS7*, unsigned char**))i2d_PKCS7, &(p12)->authsafes->d.data->data,
 #else
--- origsrc/tqsllib-2.2/tqsllib.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/tqsllib-2.2/tqsllib.pc.in	2015-04-08 14:53:31.623403900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: tqsllib
+Description: TrustedQSL library
+Version: @VERSION@
+Libs: -L${libdir} -ltqsllib
+Cflags: -I${includedir}
