--- origsrc/gfatools-0.5/Makefile	2021-03-22 06:00:28.000000000 +0900
+++ src/gfatools-0.5/Makefile	2024-10-11 23:22:07.022888200 +0900
@@ -1,6 +1,6 @@
 CC=			gcc
-CFLAGS=		-g -Wall -Wc++-compat -std=c99 -O2
-CPPFLAGS=
+CFLAGS?=		-g -Wall -Wc++-compat -std=c99 -O2
+CPPFLAGS?=
 INCLUDES=	-I.
 OBJS=		kalloc.o gfa-base.o gfa-io.o gfa-aug.o gfa-sub.o gfa-asm.o gfa-util.o \
 			gfa-bbl.o gfa-sql.o gfa-gt.o
