--- origsrc/souffle-2.3/src/CMakeLists.txt	2022-05-14 22:40:57.000000000 +0900
+++ src/souffle-2.3/src/CMakeLists.txt	2022-06-15 13:50:32.052512700 +0900
@@ -214,7 +214,10 @@ set_target_properties(libsouffle PROPERT
 set_target_properties(compiled PROPERTIES CXX_EXTENSIONS OFF)
 
 if (NOT MSVC)
-  target_compile_options(libsouffle PUBLIC -Wall -Wextra -Werror -fwrapv)
+  target_compile_options(libsouffle PUBLIC -Wall -Wextra -fwrapv)
+  if (NOT CYGWIN)
+    target_compile_options(libsouffle PUBLIC -Werror)
+  endif()
 else ()
   target_compile_options(libsouffle PUBLIC /W3 /WX)
 endif ()
