--- origsrc/souffle-2.4/src/CMakeLists.txt	2023-03-05 22:48:25.000000000 +0900
+++ src/souffle-2.4/src/CMakeLists.txt	2023-08-27 23:25:26.417253200 +0900
@@ -229,7 +229,10 @@ set_target_properties(compiled PROPERTIE
 set_target_properties(compiled PROPERTIES POSITION_INDEPENDENT_CODE ON)
 
 if (NOT MSVC)
-  target_compile_options(libsouffle PUBLIC -Wall -Wextra -Werror -fwrapv)
+  target_compile_options(libsouffle PUBLIC -Wall -Wextra -fwrapv)
+  if (NOT CYGWIN)
+    target_compile_options(libsouffle PUBLIC -Werror)
+  endif()
 else ()
   target_compile_options(libsouffle PUBLIC /W3 /WX)
 endif ()
