--- origsrc/libsbsms-2.0.2/src/Makefile.am	2014-06-12 06:20:40.000000000 +0900
+++ src/libsbsms-2.0.2/src/Makefile.am	2015-01-03 01:45:14.554726800 +0900
@@ -1,4 +1,4 @@
-AM_CXXFLAGS = -I../include @SBSMS_CFLAGS@
+AM_CXXFLAGS = -I$(top_srcdir)/include @SBSMS_CFLAGS@
 
 libsbsmsincludedir = $(includedir)
 libsbsmsinclude_HEADERS = ../include/sbsms.h
@@ -31,3 +31,4 @@ libsbsms_la_SOURCES = \
 	subband.h \
 	track.h \
 	$(NULL)
+libsbsms_la_LDFLAGS = -no-undefined
