--- origsrc/libltcsmpte-0.4.3/configure.ac	2010-10-21 00:29:07.000000000 +0900
+++ src/libltcsmpte-0.4.3/configure.ac	2011-06-18 16:14:13.152423000 +0900
@@ -45,7 +45,7 @@ AC_CONFIG_MACRO_DIR([m4])
 
 dnl *** misc complier/linker flags ***
 LIBSMPTE_CFLAGS="-Wall"
-LIBSMPTE_LDFLAGS="-export-symbols-regex '^(FR_|SMPTE).*'"
+LIBSMPTE_LDFLAGS="-export-symbols-regex '^(FR_|SMPTE).*' -no-undefined"
 dnl LIBSMPTE_CFLAGS="${LIBSMPTE_CFLAGS} -g -posix -std=c99 -pedantic"
 
 AC_MSG_CHECKING([if -Wl,--as-needed works])
--- origsrc/libltcsmpte-0.4.3/tests/Makefile.am	2010-10-20 19:40:42.000000000 +0900
+++ src/libltcsmpte-0.4.3/tests/Makefile.am	2011-06-18 16:33:53.525052500 +0900
@@ -24,7 +24,7 @@ check: $(noinst_PROGRAMS)
 	 ./encodetst$(EXEEXT) ${TMPFILE}
 	 @echo "-----------------------------------------------------------------"
 	 @echo "reading test file"
-	 ./decodetst$(EXEEXT) ../testdata/smpte.raw
+	 ./decodetst$(EXEEXT) $(top_srcdir)/testdata/smpte.raw
 	 @echo "-----------------------------------------------------------------"
 	 @echo "reading generated file"
 	 ./decodetst$(EXEEXT) ${TMPFILE}
