--- origsrc/artifastring-release-2.4/Makefile.am	2013-05-31 15:21:51.000000000 +0900
+++ src/artifastring-release-2.4/Makefile.am	2015-05-04 04:05:33.163200600 +0900
@@ -7,7 +7,7 @@ else
 if IS_DARWIN
 AM_CXXFLAGS = -I$(srcdir)/otherlibs/ -fPIC -O3 -mtune=native -DNDEBUG
 else
-AM_CXXFLAGS = -I$(srcdir)/otherlibs/ -fPIC -O3 -march=native -DNDEBUG
+AM_CXXFLAGS = -O3 -march=native -DNDEBUG
 endif
 MORE_WARNINGS_CXXFLAGS =
 endif
@@ -71,21 +71,24 @@ artifastring_h = artifastring/artifastri
 	artifastring/artifastring_instrument.h
 monowav_h = artifastring/monowav.h
 
-lib_LIBRARIES = libartifastring.a libmonowav.a
-libartifastring_a_SOURCES = $(artifastring_h) \
+lib_LTLIBRARIES = libartifastring.la libmonowav.la
+libartifastring_la_SOURCES = $(artifastring_h) \
 	artifastring/actions_file.cpp \
 	artifastring/midi_pos.cpp \
 	artifastring/fft_convolution.cpp \
 	artifastring/artifastring_string.cpp \
 	artifastring/artifastring_instrument.cpp
-libartifastring_a_CXXFLAGS = $(AM_CXXFLAGS) \
+libartifastring_la_CXXFLAGS = $(AM_CXXFLAGS) \
 	$(MORE_WARNINGS_CXXFLAGS) \
 	$(EIGEN3_CFLAGS)
+libartifastring_la_LDFLAGS = -no-undefined
+libartifastring_la_LIBADD = $(FFTW3F_LIBS)
 artifastring_includedir=$(includedir)/artifastring/
-libmonowav_a_SOURCES = $(monowav_h) \
+libmonowav_la_SOURCES = $(monowav_h) \
 	artifastring/monowav.cpp 
-libmonowav_a_CXXFLAGS = $(AM_CXXFLAGS) $(MORE_WARNINGS_CXXFLAGS) \
+libmonowav_la_CXXFLAGS = $(AM_CXXFLAGS) $(MORE_WARNINGS_CXXFLAGS) \
 	$(EIGEN3_CFLAGS)
+libmonowav_la_LDFLAGS = -no-undefined
 monowav_includedir=$(includedir)/artifastring/
 nobase_include_HEADERS = $(artifastring_h) $(monowav_h)
 
@@ -101,7 +104,7 @@ actions2wav_SOURCES = src/actions2wav.cp
 actions2wav_CXXFLAGS = $(AM_CXXFLAGS) $(ARTIFASTRING_INCLUDE) \
 	$(MORE_WARNINGS_CXXFLAGS) \
 	$(EIGEN3_CFLAGS)
-actions2wav_LDADD = $(lib_LIBRARIES) \
+actions2wav_LDADD = $(lib_LTLIBRARIES) \
 	$(EIGEN3_LIBS)
 
 
@@ -120,8 +123,9 @@ EXTRA_DIST += \
 
 if HAS_SWIG
 # unified definitions
-SWIGCPP = $(PYTHON_CPPFLAGS) $(AX_SWIG_PYTHON_CPPFLAGS) $(ARTIFASTRING_INCLUDE)
-SWIGLD = $(PYTHON_LDFLAGS) -module
+SWIGCPP = $(PYTHON_CPPFLAGS) $(AX_SWIG_PYTHON_CPPFLAGS) $(ARTIFASTRING_INCLUDE) \
+	-I$(PYTHON_SITE_PKG)/numpy/core/include
+SWIGLD = $(PYTHON_LDFLAGS) -module -avoid-version -no-undefined
 swig/%_wrap.cxx swig/%.py: $(srcdir)/swig/%.i
 	mkdir -p swig
 	$(SWIG) $(AX_SWIG_PYTHON_OPT) $(ARTIFASTRING_INCLUDE) -builtin -o $@ $<
@@ -153,6 +157,7 @@ nodist__artifastring_instrument_la_SOURC
 	artifastring/artifastring_instrument.cpp
 _artifastring_instrument_la_CPPFLAGS = $(SWIGCPP) $(EIGEN3_CFLAGS)
 _artifastring_instrument_la_LDFLAGS = $(SWIGLD) $(EIGEN3_LIBS)
+_artifastring_instrument_la_LIBADD = $(FFTW3F_LIBS)
 
 nodist__actions_file_la_SOURCES = swig/actions_file_wrap.cxx \
 	artifastring/actions_file.cpp
@@ -202,6 +207,3 @@ bin_SCRIPTS += blender/actions2images.py
 endif
 EXTRA_DIST += $(srcdir)/blender/actions2images.in.py
 EXTRA_DIST += $(srcdir)/blender/artifastring_movie.py
-
-
-
--- origsrc/artifastring-release-2.4/configure.ac	2013-05-31 15:21:51.000000000 +0900
+++ src/artifastring-release-2.4/configure.ac	2015-05-04 03:51:00.836330200 +0900
@@ -10,9 +10,8 @@ AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_FILES([Makefile artifastring.pc monowav.pc doc/doxyfile])
 
 AC_PREREQ([2.59])
-AM_INIT_AUTOMAKE([1.10 no-define foreign -Wall -Werror])
-AM_PROG_AR
-
+AM_INIT_AUTOMAKE([1.10 no-define foreign -Wall subdir-objects])
+LT_INIT
 
 ### libraries
 AC_PROG_CXX
@@ -21,9 +20,8 @@ AC_HEADER_STDC
 
 #AC_LANG([C++])
 
-AC_CHECK_LIB(stdc++, main,, AC_MSG_ERROR(requires libstc++))
-AC_CHECK_LIB(m, main,, AC_MSG_ERROR(requires libm (math)))
-AC_CHECK_LIB(fftw3f, main,, AC_MSG_ERROR(requires libfftw3))
+PKG_CHECK_MODULES([EIGEN3], [eigen3])
+PKG_CHECK_MODULES([FFTW3F], [fftw3f])
 
 AC_ARG_ENABLE([debug],
   [AS_HELP_STRING(--enable-debug,[Compile with debug support])],
