--- origsrc/szl-1.0/src/Makefile.am	2010-08-16 07:04:17.000000000 +0900
+++ src/szl-1.0/src/Makefile.am	2013-09-14 15:07:12.634175900 +0900
@@ -250,12 +250,13 @@ libengine_la_SOURCES = \
 
 ##### Libraries to install
 
-libszl_la_LIBADD = \
+libszl_la_LIBADD = -lstdc++ \
   libengine.la \
   libfmt.la \
   libutilities.la \
   libvalues.la
 libszl_la_SOURCES =
+libszl_la_LDFLAGS = -no-undefined
 
 libszlintrinsics_la_SOURCES = \
   intrinsics/additionalinputintrinsic.cc \
@@ -265,6 +266,12 @@ libszlintrinsics_la_SOURCES = \
   intrinsics/miscintrinsic.cc \
   intrinsics/sortintrinsic.cc \
   intrinsics/ziplib.cc
+libszlintrinsics_la_LDFLAGS = -no-undefined
+libszlintrinsics_la_LIBADD = \
+  libengine.la \
+  libfmt.la \
+  libutilities.la \
+  libvalues.la
 
 libszlemitters_la_SOURCES = \
   emitters/szlbootstrapsum.cc \
@@ -307,8 +314,12 @@ libszlemitters_la_SOURCES = \
   emitters/weighted-reservoir-sampler-impl.cc \
   emitters/weighted-reservoir-sampler-impl.h \
   emitters/weighted-reservoir-sampler.h
-
-
+libszlemitters_la_LDFLAGS = -no-undefined
+libszlemitters_la_LIBADD = \
+  libengine.la \
+  libfmt.la \
+  libutilities.la \
+  libvalues.la
 
 ##### The szl program
 
--- origsrc/szl-1.0/src/emitters/tests/szlquantile_unittest.cc	2010-08-16 07:04:16.000000000 +0900
+++ src/szl-1.0/src/emitters/tests/szlquantile_unittest.cc	2013-09-14 14:52:12.061910300 +0900
@@ -233,7 +233,7 @@ bool SzlQuantileTest::CheckCorrectness(i
   // within the tolerable range. The ideal index
   // we are expecting is calculated as next_rank.
   for (int i = 1; i < size - 1; ++i) {
-    int64 next_rank = num_inserts * i / (size - 1);
+    int next_rank = num_inserts * i / (size - 1);
     dec.Init(rvec[i].data(), rvec[i].size());
     dec.GetString(&vstr);
     CHECK_EQ(sscanf(vstr.c_str(), "xx-%d", &index), 1);
--- origsrc/szl-1.0/src/engine/code.cc	2010-08-16 07:04:14.000000000 +0900
+++ src/szl-1.0/src/engine/code.cc	2013-09-14 14:55:59.414449300 +0900
@@ -18,6 +18,7 @@
 #include <string>
 #include <errno.h>
 #include <sys/mman.h>
+#include <unistd.h>
 
 #include "engine/globals.h"
 #include "public/logging.h"
--- origsrc/szl-1.0/src/engine/scanner.h	2010-08-16 07:04:13.000000000 +0900
+++ src/szl-1.0/src/engine/scanner.h	2013-09-14 14:52:12.061910300 +0900
@@ -14,6 +14,7 @@
 // ------------------------------------------------------------------------
 
 #include <string>
+#include <cstdio>
 
 namespace sawzall {
 
--- origsrc/szl-1.0/src/engine/symboltable.cc	2010-08-16 07:04:14.000000000 +0900
+++ src/szl-1.0/src/engine/symboltable.cc	2013-09-14 14:52:12.077535300 +0900
@@ -44,7 +44,7 @@ namespace sawzall {
 // ------------------------------------------------------------------------------
 // Implementation of SymbolTable
 
-Proc::Proc* SymbolTable::init_proc_ = NULL;
+Proc* SymbolTable::init_proc_ = NULL;
 
 List<TableType*>* SymbolTable::table_types_ = NULL;
 TableType* SymbolTable::collection_type_ = NULL;
--- origsrc/szl-1.0/src/public/porting.h	2010-08-16 07:04:17.000000000 +0900
+++ src/szl-1.0/src/public/porting.h	2013-09-14 14:52:12.077535300 +0900
@@ -48,7 +48,9 @@ typedef uint64_t uint64;
 
 typedef signed char schar;
 
+#if !defined(__CYGWIN__)
 typedef unsigned long uintptr_t;
+#endif
 typedef long sword_t;
 typedef unsigned long uword_t;
 
--- origsrc/szl-1.0/src/utilities/port_ieee.h	2010-08-16 07:04:16.000000000 +0900
+++ src/szl-1.0/src/utilities/port_ieee.h	2013-09-14 14:52:12.077535300 +0900
@@ -14,7 +14,7 @@
 // ------------------------------------------------------------------------
 
 #include <config.h>
-#if defined(OS_CYGWIN)
+#if defined(__CYGWIN__)
 #include <ieeefp.h>
 #elif defined(OS_MACOSX)
 #include <machine/endian.h>
--- origsrc/szl-1.0/src/utilities/random_base.cc	2010-08-16 07:04:15.000000000 +0900
+++ src/szl-1.0/src/utilities/random_base.cc	2013-09-14 15:03:59.236715000 +0900
@@ -18,6 +18,7 @@
 #include <string>
 #include <memory.h>
 #include <assert.h>
+#include <unistd.h>
 
 #include "public/porting.h"
 #include "public/logging.h"
@@ -194,7 +195,7 @@ uint64 RandomBase::UnbiasedUniform64(uin
 }
 
 float RandomBase::RandFloat() {
-#ifdef OS_CYGWIN
+#ifdef __CYGWIN__
   __ieee_float_shape_type v;
   v.number.sign = 0;
   v.number.fraction0 = Rand8();   // Lower 7 bits
@@ -215,7 +216,7 @@ float RandomBase::RandFloat() {
 }
 
 double RandomBase::RandDouble() {
-#ifdef OS_CYGWIN
+#ifdef __CYGWIN__
   __ieee_double_shape_type v;
   v.number.sign = 0;
   v.number.fraction0 = Rand32();  // Lower 20 bits
