--- origsrc/libuvc-0.0.5/CMakeLists.txt	2014-07-20 08:29:01.000000000 +0900
+++ src/libuvc-0.0.5/CMakeLists.txt	2014-10-25 10:53:42.773817700 +0900
@@ -23,6 +23,7 @@ find_path(LIBUSB_INCLUDE_DIR libusb-1.0/
 	PATHS /opt/local/include)
 
 # Try to find JPEG using a module or pkg-config. If that doesn't work, search for the header.
+if (NOT CYGWIN)
 find_package(jpeg QUIET)
 if(NOT JPEG_FOUND)
   find_path(JPEG_INCLUDE_DIR jpeglib.h)
@@ -31,6 +32,7 @@ if(NOT JPEG_FOUND)
     set(JPEG_LIBRARIES -ljpeg)
   endif()
 endif()
+endif()
 
 SET(CMAKE_C_FLAGS_DEBUG "-g -DUVC_DEBUGGING")
 
@@ -68,6 +70,7 @@ configure_file(include/libuvc/libuvc_con
   ${PROJECT_BINARY_DIR}/include/libuvc/libuvc_config.h @ONLY)
 
 set_target_properties(uvc PROPERTIES
+  SOVERSION 0
   PUBLIC_HEADER "include/libuvc/libuvc.h;${libuvc_BINARY_DIR}/include/libuvc/libuvc_config.h" )
 
 if(JPEG_FOUND)
@@ -82,6 +85,7 @@ target_link_libraries(uvc ${LIBUSB_LIBRA
 
 install(TARGETS uvc
   EXPORT libuvcTargets
+  RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/bin"
   LIBRARY DESTINATION "${CMAKE_INSTALL_PREFIX}/lib"
   ARCHIVE DESTINATION "${CMAKE_INSTALL_PREFIX}/lib"
   PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_PREFIX}/include/libuvc"
