--- origsrc/libuvc-0.0.7/CMakeLists.txt	2023-04-01 10:38:41.000000000 +0900
+++ src/libuvc-0.0.7/CMakeLists.txt	2023-08-16 11:16:04.402524800 +0900
@@ -43,7 +43,7 @@ find_package(LibUSB)
 
 # JpegPkg name to differ from shipped with CMake
 find_package(JpegPkg QUIET)
-if(JPEG_FOUND)
+if(ENABLE_JPEG AND JPEG_FOUND)
   message(STATUS "Building libuvc with JPEG support.")
   set(LIBUVC_HAS_JPEG TRUE)
   list(APPEND SOURCES src/frame-mjpeg.c)
--- origsrc/libuvc-0.0.7/include/libuvc/libuvc_internal.h	2023-04-01 10:38:41.000000000 +0900
+++ src/libuvc-0.0.7/include/libuvc/libuvc_internal.h	2023-08-16 11:14:52.384307100 +0900
@@ -72,6 +72,9 @@
 /* http://stackoverflow.com/questions/19452971/array-size-macro-that-rejects-pointers */
 #define IS_INDEXABLE(arg) (sizeof(arg[0]))
 #define IS_ARRAY(arg) (IS_INDEXABLE(arg) && (((void *) &arg) == ((void *) arg)))
+#if defined(ARRAYSIZE)
+#undef ARRAYSIZE
+#endif
 #define ARRAYSIZE(arr) (sizeof(arr) / (IS_ARRAY(arr) ? sizeof(arr[0]) : 0))
 
 /** Video interface subclass code (A.2) */
