--- origsrc/yyjson-0.5.1/CMakeLists.txt	2022-06-17 22:12:30.000000000 +0900
+++ src/yyjson-0.5.1/CMakeLists.txt	2022-06-21 23:30:06.493848900 +0900
@@ -60,7 +60,7 @@ endif()
 add_library(yyjson src/yyjson.h src/yyjson.c)
 target_include_directories(yyjson PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>)
 set_target_properties(yyjson PROPERTIES VERSION ${PROJECT_VERSION})
-
+set_target_properties(yyjson PROPERTIES SOVERSION ${PROJECT_VERSION_MAJOR})
 
 
 # ------------------------------------------------------------------------------
@@ -438,3 +438,10 @@ if(YYJSON_BUILD_DOC)
         message(WARNING "Doxygen not found")
     endif()
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/yyjson.pc.in ${CMAKE_BINARY_DIR}/yyjson.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/yyjson.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/yyjson-0.5.1/yyjson.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/yyjson-0.5.1/yyjson.pc.in	2022-06-21 23:30:24.981566400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: yyjson
+Description: Fastest JSON library in C
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lyyjson
+Cflags: -I${includedir}
