--- origsrc/wdns-tags-v0.9.1/Makefile.am	2017-01-04 03:49:33.000000000 +0900
+++ src/wdns-tags-v0.9.1/Makefile.am	2017-12-26 15:01:00.084000200 +0900
@@ -14,7 +14,7 @@ AM_CPPFLAGS = \
 	-include $(top_builddir)/config.h \
 	-I$(top_srcdir)/wdns
 AM_CFLAGS = $(my_CFLAGS)
-AM_LDFLAGS =
+AM_LDFLAGS = -no-undefined
 
 LIBWDNS_CURRENT=1
 LIBWDNS_REVISION=0
@@ -28,13 +28,13 @@ EXTRA_DIST += wdns/gen_rrclass_to_str
 EXTRA_DIST += wdns/gen_rrtype_to_str
 
 wdns/rcode_to_str.c: wdns/wdns.h wdns/gen_rcode_to_str
-	$(AM_V_GEN)wdns/gen_rcode_to_str wdns/wdns.h wdns/rcode_to_str.c
+	$(AM_V_GEN)$(top_srcdir)/wdns/gen_rcode_to_str $(top_srcdir)/wdns/wdns.h wdns/rcode_to_str.c
 
 wdns/rrclass_to_str.c: wdns/wdns.h wdns/gen_rrclass_to_str
-	$(AM_V_GEN)wdns/gen_rrclass_to_str wdns/wdns.h wdns/rrclass_to_str.c
+	$(AM_V_GEN)$(top_srcdir)/wdns/gen_rrclass_to_str $(top_srcdir)/wdns/wdns.h wdns/rrclass_to_str.c
 
 wdns/rrtype_to_str.c: wdns/wdns.h wdns/gen_rrtype_to_str
-	$(AM_V_GEN)wdns/gen_rrtype_to_str wdns/wdns.h wdns/rrtype_to_str.c
+	$(AM_V_GEN)$(top_srcdir)/wdns/gen_rrtype_to_str $(top_srcdir)/wdns/wdns.h wdns/rrtype_to_str.c
 
 wdns_libwdns_la_CPPFLAGS = \
 	-include $(top_builddir)/wdns/wdns-private.h
@@ -113,42 +113,42 @@ pkgconfig_DATA = wdns/libwdns.pc
 EXTRA_DIST += wdns/libwdns.pc.in
 CLEANFILES += wdns/libwdns.pc
 
-noinst_PROGRAMS += examples/wdns-dump-file
+bin_PROGRAMS += examples/wdns-dump-file
 examples_wdns_dump_file_LDADD = wdns/libwdns.la
 examples_wdns_dump_file_SOURCES = \
 	examples/private.h \
 	examples/wdns-dump-file.c
 
 if LIBPCAP
-noinst_PROGRAMS += examples/wdns-dump-pcap
+bin_PROGRAMS += examples/wdns-dump-pcap
 examples_wdns_dump_pcap_LDADD = wdns/libwdns.la -lpcap
 examples_wdns_dump_pcap_SOURCES = \
 	examples/private.h \
 	examples/wdns-dump-pcap.c
 endif
 
-noinst_PROGRAMS += examples/wdns-test-deserialize-rrset
+bin_PROGRAMS += examples/wdns-test-deserialize-rrset
 examples_wdns_test_deserialize_rrset_LDADD = wdns/libwdns.la
 examples_wdns_test_deserialize_rrset_SOURCES = \
 	examples/private.h \
 	examples/wdns-hex-driver.c \
 	examples/wdns-test-deserialize-rrset.c
 
-noinst_PROGRAMS += examples/wdns-test-downcase-rrset
+bin_PROGRAMS += examples/wdns-test-downcase-rrset
 examples_wdns_test_downcase_rrset_LDADD = wdns/libwdns.la
 examples_wdns_test_downcase_rrset_SOURCES = \
 	examples/private.h \
 	examples/wdns-hex-driver.c \
 	examples/wdns-test-downcase-rrset.c
 
-noinst_PROGRAMS += examples/wdns-test-print-message
+bin_PROGRAMS += examples/wdns-test-print-message
 examples_wdns_test_print_message_LDADD = wdns/libwdns.la
 examples_wdns_test_print_message_SOURCES = \
 	examples/private.h \
 	examples/wdns-hex-driver.c \
 	examples/wdns-test-print-message.c
 
-noinst_PROGRAMS += examples/wdns-test-serialize-rrset
+bin_PROGRAMS += examples/wdns-test-serialize-rrset
 examples_wdns_test_serialize_rrset_LDADD = wdns/libwdns.la
 examples_wdns_test_serialize_rrset_SOURCES = \
 	examples/private.h \
