--- origsrc/wdns-tags-v0.9.0/Makefile.am	2016-04-19 08:43:21.000000000 +0900
+++ src/wdns-tags-v0.9.0/Makefile.am	2016-04-23 14:12:20.326091700 +0900
@@ -12,7 +12,7 @@ AM_CPPFLAGS = \
 	-include $(top_builddir)/config.h \
 	-I$(top_srcdir)/wdns
 AM_CFLAGS = $(my_CFLAGS)
-AM_LDFLAGS =
+AM_LDFLAGS = -no-undefined
 
 LIBWDNS_CURRENT=1
 LIBWDNS_REVISION=0
@@ -26,13 +26,13 @@ EXTRA_DIST += wdns/gen_rrclass_to_str
 EXTRA_DIST += wdns/gen_rrtype_to_str
 
 wdns/rcode_to_str.c: wdns/wdns.h wdns/gen_rcode_to_str
-	$(AM_V_GEN)wdns/gen_rcode_to_str wdns/wdns.h wdns/rcode_to_str.c
+	$(AM_V_GEN)$(top_srcdir)/wdns/gen_rcode_to_str $(top_srcdir)/wdns/wdns.h wdns/rcode_to_str.c
 
 wdns/rrclass_to_str.c: wdns/wdns.h wdns/gen_rrclass_to_str
-	$(AM_V_GEN)wdns/gen_rrclass_to_str wdns/wdns.h wdns/rrclass_to_str.c
+	$(AM_V_GEN)$(top_srcdir)/wdns/gen_rrclass_to_str $(top_srcdir)/wdns/wdns.h wdns/rrclass_to_str.c
 
 wdns/rrtype_to_str.c: wdns/wdns.h wdns/gen_rrtype_to_str
-	$(AM_V_GEN)wdns/gen_rrtype_to_str wdns/wdns.h wdns/rrtype_to_str.c
+	$(AM_V_GEN)$(top_srcdir)/wdns/gen_rrtype_to_str $(top_srcdir)/wdns/wdns.h wdns/rrtype_to_str.c
 
 wdns_libwdns_la_CPPFLAGS = \
 	-include $(top_builddir)/wdns/wdns-private.h
