--- origsrc/doctest-2.4.11/scripts/cmake/common.cmake	2023-03-15 21:35:44.000000000 +0900
+++ src/doctest-2.4.11/scripts/cmake/common.cmake	2023-03-22 22:17:00.307373100 +0900
@@ -79,7 +79,9 @@ macro(add_compiler_flags)
 endmacro()
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-    add_compiler_flags(-Werror)
+    if (NOT CYGWIN)
+        add_compiler_flags(-Werror)
+    endif()
     add_compiler_flags(-fstrict-aliasing)
 
     # The following options are not valid when clang-cl is used.
