--- origsrc/doctest-2.4.12/scripts/cmake/common.cmake	2025-04-28 21:14:46.000000000 +0900
+++ src/doctest-2.4.12/scripts/cmake/common.cmake	2025-09-01 01:10:07.814824300 +0900
@@ -82,7 +82,9 @@ macro(add_compiler_flags)
 endmacro()
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-    add_compiler_flags(-Werror)
+    if (NOT CYGWIN)
+        add_compiler_flags(-Werror)
+    endif()
     add_compiler_flags(-fstrict-aliasing)
 
     # The following options are not valid when clang-cl is used.
