--- origsrc/doctest-2.4.8/scripts/cmake/common.cmake	2022-01-11 02:39:35.000000000 +0900
+++ src/doctest-2.4.8/scripts/cmake/common.cmake	2022-01-28 23:33:42.162938600 +0900
@@ -79,7 +79,9 @@ macro(add_compiler_flags)
 endmacro()
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-    add_compiler_flags(-Werror)
+    if (NOT CYGWIN)
+        add_compiler_flags(-Werror)
+    endif()
     add_compiler_flags(-fstrict-aliasing)
 
     # The following options are not valid when clang-cl is used.
