--- origsrc/imageworsener-0.9.9/Makefile.am	2012-03-30 08:07:24.000000000 +0900
+++ src/imageworsener-0.9.9/Makefile.am	2012-04-27 18:08:40.855493400 +0900
@@ -14,7 +14,7 @@ libimageworsener_la_SOURCES=src/imagew-m
  src/imagew-webp.c \
  src/imagew-util.c
 libimageworsener_la_LIBADD=-lm
-libimageworsener_la_LDFLAGS=-release 0.9.9
+libimageworsener_la_LDFLAGS=-release 0.9.9 -no-undefined
 bin_PROGRAMS=imagew
 imagew_SOURCES=src/imagew-cmd.c
 imagew_LDADD=libimageworsener.la
