--- origsrc/imageworsener-1.1.0/Makefile.am	2013-02-03 01:05:37.000000000 +0900
+++ src/imageworsener-1.1.0/Makefile.am	2013-05-25 12:00:07.828456100 +0900
@@ -14,7 +14,7 @@ libimageworsener_la_SOURCES=src/imagew-m
  src/imagew-webp.c \
  src/imagew-util.c
 libimageworsener_la_LIBADD=-lm
-libimageworsener_la_LDFLAGS=-version-info 1:0:0
+libimageworsener_la_LDFLAGS=-version-info 1:0:0 -no-undefined
 bin_PROGRAMS=imagew
 imagew_SOURCES=src/imagew-cmd.c
 imagew_LDADD=libimageworsener.la
