--- origsrc/libsylph-1.1.0/Makefile.am	2006-11-20 16:31:43.000000000 +0900
+++ src/libsylph-1.1.0/Makefile.am	2015-01-04 21:52:12.989985200 +0900
@@ -15,4 +15,4 @@ release: distdir
 	BZIP2=$(BZIP2_ENV) $(TAR) --bzip2 -chof $(distdir).tar.bz2 $(distdir)
 	-rm -rf $(distdir)
 
-ACLOCAL_AMFLAGS = -I m4
+ACLOCAL_AMFLAGS = -I ac
--- origsrc/libsylph-1.1.0/libsylph/Makefile.am	2006-10-04 11:14:10.000000000 +0900
+++ src/libsylph-1.1.0/libsylph/Makefile.am	2015-01-04 21:52:13.005610200 +0900
@@ -89,10 +89,10 @@ libsylphinclude_HEADERS = \
 	xml.h
 
 if OS_WIN32
-no_undefined = -no-undefined
 export_symbols = -export-symbols libsylph.def
 endif
 
+no_undefined = -no-undefined
 libsylph_la_LDFLAGS = \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
 	-export-dynamic $(no_undefined) $(export_symbols)
--- origsrc/libsylph-1.1.0/libsylph/defs.h	2006-09-22 13:10:24.000000000 +0900
+++ src/libsylph-1.1.0/libsylph/defs.h	2015-01-04 21:57:40.376704000 +0900
@@ -24,7 +24,7 @@
 #  include "config.h"
 #endif
 
-#include <glibconfig.h>
+#include <glib.h>
 
 #ifdef G_OS_WIN32
 #  include <glib/gwin32.h>
