--- origsrc/libsylph-1.1.0/Makefile.am	2006-11-20 16:31:43.000000000 +0900
+++ src/libsylph-1.1.0/Makefile.am	2025-08-20 16:08:53.406090400 +0900
@@ -15,4 +15,7 @@ release: distdir
 	BZIP2=$(BZIP2_ENV) $(TAR) --bzip2 -chof $(distdir).tar.bz2 $(distdir)
 	-rm -rf $(distdir)
 
-ACLOCAL_AMFLAGS = -I m4
+ACLOCAL_AMFLAGS = -I ac
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libsylph-1.1.0/configure.in	2025-08-20 15:59:37.306220200 +0900
+++ src/libsylph-1.1.0/configure.in	2025-08-20 16:09:28.247868200 +0900
@@ -156,6 +156,7 @@ if test "$ac_cv_enable_oniguruma" = yes;
 		CFLAGS="$CFLAGS `pkg-config --cflags oniguruma`"
 		LIBS="$LIBS `pkg-config --libs oniguruma`"
 		AC_DEFINE(USE_ONIGURUMA, 1, Define if you want to use Oniguruma.)
+		AC_CHECK_HEADERS([onigposix.h])
 	else
 		AC_MSG_RESULT(no)
 		ac_cv_enable_oniguruma=no
@@ -271,6 +272,7 @@ doc/Makefile
 examples/Makefile
 libsylph/Makefile
 po/Makefile.in
+libsylph.pc
 ])
 
 dnl Output the configuration summary
--- origsrc/libsylph-1.1.0/libsylph/Makefile.am	2006-10-04 11:14:10.000000000 +0900
+++ src/libsylph-1.1.0/libsylph/Makefile.am	2025-08-20 16:02:04.361309800 +0900
@@ -89,10 +89,11 @@ libsylphinclude_HEADERS = \
 	xml.h
 
 if OS_WIN32
-no_undefined = -no-undefined
 export_symbols = -export-symbols libsylph.def
 endif
 
+no_undefined = -no-undefined
+
 libsylph_la_LDFLAGS = \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
 	-export-dynamic $(no_undefined) $(export_symbols)
--- origsrc/libsylph-1.1.0/libsylph/filter.c	2006-08-03 15:28:53.000000000 +0900
+++ src/libsylph-1.1.0/libsylph/filter.c	2025-08-20 16:02:45.189408500 +0900
@@ -29,7 +29,7 @@
 #include <strings.h>
 #include <stdlib.h>
 #include <sys/types.h>
-#if USE_ONIGURUMA
+#if USE_ONIGURUMA && defined(HAVE_ONIGPOSIX_H)
 #  include <onigposix.h>
 #elif HAVE_REGEX_H
 #  include <regex.h>
@@ -282,7 +282,7 @@ static gboolean strmatch_regex(const gch
 	gint ret = 0;
 	regex_t preg;
 
-#if USE_ONIGURUMA
+#if USE_ONIGURUMA && defined(HAVE_ONIGPOSIX_H)
 	reg_set_encoding(REG_POSIX_ENCODING_UTF8);
 #endif
 	ret = regcomp(&preg, needle, REG_ICASE|REG_EXTENDED);
--- origsrc/libsylph-1.1.0/libsylph.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libsylph-1.1.0/libsylph.pc.in	2025-08-20 16:08:53.361166700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libsylph
+Description: Sylpheed's core features independent from the UI of Sylpheed
+Version: @VERSION@
+Libs: -L${libdir} -lsylph
+Cflags: -I${includedir}
