--- origsrc/libdeltacloud-0.8/Makefile.am	2011-04-20 03:23:01.000000000 +0900
+++ src/libdeltacloud-0.8/Makefile.am	2011-05-14 12:15:43.129578700 +0900
@@ -3,9 +3,9 @@
 EXTRA_DIST = libdeltacloud.spec build-aux m4
 SUBDIRS = include/libdeltacloud
 
-AM_CPPFLAGS = $(LIBXML_CFLAGS) $(LIBCURL_CFLAGS) -Wall -Iinclude/libdeltacloud -fno-strict-aliasing -Werror
-libdeltacloud_la_LDFLAGS = $(LIBXML_LIBS) $(LIBCURL_LIBS) -lpthread \
-	$(VERSION_SCRIPT_FLAGS)libdeltacloud.syms -version-info 6:0:0
+AM_CPPFLAGS = $(LIBXML_CFLAGS) $(LIBCURL_CFLAGS) -Wall -I$(top_srcdir)/include/libdeltacloud -fno-strict-aliasing -Werror
+libdeltacloud_la_LDFLAGS = $(LIBXML_LIBS) $(LIBCURL_LIBS) -lpthread -no-undefined \
+	$(VERSION_SCRIPT_FLAGS)$(top_srcdir)/libdeltacloud.syms -version-info 6:0:0
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libdeltacloud.pc
