--- origsrc/libp11-libp11-0.4.13/src/Makefile.am	2024-12-14 03:13:57.000000000 +0900
+++ src/libp11-libp11-0.4.13/src/Makefile.am	2025-01-07 08:59:14.247119600 +0900
@@ -22,7 +22,7 @@ libp11_la_LIBADD += libp11.lo
 else
 dist_noinst_DATA = libp11.rc
 endif
-libp11_la_LDFLAGS = $(AM_LDFLAGS) \
+libp11_la_LDFLAGS = $(AM_LDFLAGS) -no-undefined \
 	-version-info @LIBP11_LT_CURRENT@:@LIBP11_LT_REVISION@:@LIBP11_LT_AGE@
 
 if HAVE_LD_VERSION_SCRIPT
@@ -46,7 +46,7 @@ pkcs11_la_LIBADD = $(libp11_la_OBJECTS)
 
 # We intentionally not version symbols in this module because no
 # application links with it. It is dynamically opened.
-pkcs11_la_LDFLAGS = $(AM_LDFLAGS) -module -shared -shrext $(SHARED_EXT) \
+pkcs11_la_LDFLAGS = $(AM_LDFLAGS) -no-undefined -module -shared -shrext $(SHARED_EXT) \
 	-avoid-version -export-symbols "$(srcdir)/pkcs11.exports"
 
 # Create a static version of the engine as well to allow applications
