--- origsrc/semiprof/CMakeLists.txt	2026-03-26 13:19:44.000000000 +0900
+++ src/semiprof/CMakeLists.txt	2026-03-26 13:25:55.589747800 +0900
@@ -22,6 +22,7 @@ if (NOT TARGET semiprof)
      )
     target_compile_definitions(semiprof PUBLIC SEMIPROF) # Turn on profiler
     set(INSTALLED_TARGETS_LIST "semiprof")
+    set_target_properties(semiprof PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR})
 endif()
 
 if(SEMIPROF_WITH_EXAMPLES)
@@ -34,7 +35,7 @@ if(SEMIPROF_WITH_INSTALL AND INSTALLED_T
 
     install(TARGETS ${INSTALLED_TARGETS_LIST}
             EXPORT semiprof_targets 
-            LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+            RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
             ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
             INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
 
@@ -55,3 +56,9 @@ if(SEMIPROF_WITH_INSTALL AND INSTALLED_T
             DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/semiprof")
 endif()
 
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/semiprof.pc.in ${CMAKE_BINARY_DIR}/semiprof.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/semiprof.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/semiprof/semiprof.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/semiprof/semiprof.pc.in	2026-03-26 14:05:56.225217900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: semiprof
+Description: Simple thread safe annotation based C++ profiler
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lsemiprof
+Cflags: -I${includedir}
