--- origsrc/libslave/CMakeLists.txt	2015-02-18 22:07:28.000000000 +0900
+++ src/libslave/CMakeLists.txt	2015-02-18 22:14:58.618764400 +0900
@@ -6,7 +6,7 @@ SET (SLAVE_VERSION 0)
 
 SET (CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
-ADD_DEFINITIONS (-O3 -finline-functions -Wno-inline -Wall -pthread -fPIC)
+ADD_DEFINITIONS (-O3 -finline-functions -Wno-inline -Wall)
 
 OPTION (ENABLE_STATIC "Build static libslave" ON)
 OPTION (ENABLE_TEST "Test binary" ON)
@@ -113,7 +113,7 @@ ENDIF (ENABLE_STATIC)
 ADD_LIBRARY (slave SHARED ${SOURCES})
 
 SET_TARGET_PROPERTIES (slave PROPERTIES
-	VERSION ${SLAVE_VERSION})
+	VERSION ${SLAVE_VERSION} SOVERSION ${SLAVE_VERSION})
 
 TARGET_LINK_LIBRARIES (slave
 	${MYSQL_CLIENT_LIBS})
@@ -134,6 +134,15 @@ INSTALL (FILES ${HEADERS}
 	DESTINATION "include/libslave")
 
 INSTALL (TARGETS slave
+	RUNTIME DESTINATION bin
 	LIBRARY DESTINATION lib COMPONENT runtime
 	ARCHIVE DESTINATION lib COMPONENT development
 	PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION 0)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libslave.pc.in ${CMAKE_BINARY_DIR}/libslave.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libslave.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libslave/Slave.cpp	2015-02-18 22:07:28.000000000 +0900
+++ src/libslave/Slave.cpp	2015-02-18 22:07:29.710561200 +0900
@@ -736,7 +736,9 @@ void Slave::check_master_binlog_format()
 static bool checkAlterQuery(const std::string& str)
 {
     //RegularExpression regexp("^\\s*ALTER\\s+TABLE)", RegularExpression::RE_CASELESS);
-
+#if defined(_B)
+#undef _B
+#endif
     enum {
         SP0, _A0, _L0, _T0, _E0, _R, SP1, _T1, _A1, _B, _L1, _E1 
     } state = SP0;
--- origsrc/libslave/collate.cpp	2015-02-18 22:07:28.000000000 +0900
+++ src/libslave/collate.cpp	2015-02-18 22:07:29.710561200 +0900
@@ -6,6 +6,7 @@
 #include <vector>
 #include "nanomysql.h"
 #include "collate.h"
+#include <cstdlib>
 
 using namespace slave;
 
--- origsrc/libslave/libslave.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libslave/libslave.pc.in	2015-02-18 22:09:47.299428400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libslave
+Description: Standalone mysql replication client library
+Version: @VERSION@
+Libs: -L${libdir} -lslave
+Cflags: -I${includedir}
