--- origsrc/libslave/CMakeLists.txt	2013-02-02 21:45:18.000000000 +0900
+++ src/libslave/CMakeLists.txt	2013-02-02 21:52:02.469653400 +0900
@@ -6,7 +6,7 @@ SET (SLAVE_VERSION 0)
 
 SET (CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
-ADD_DEFINITIONS (-O3 -finline-functions -Wno-inline -Wall -pthread -fPIC)
+ADD_DEFINITIONS (-O3 -finline-functions -Wno-inline -Wall)
 
 OPTION (ENABLE_STATIC "Build static libslave" ON)
 OPTION (ENABLE_TEST "Test binary" ON)
@@ -98,7 +98,7 @@ ENDIF (ENABLE_STATIC)
 ADD_LIBRARY (slave SHARED ${SOURCES})
 
 SET_TARGET_PROPERTIES (slave PROPERTIES
-	VERSION ${SLAVE_VERSION})
+	VERSION ${SLAVE_VERSION} SOVERSION ${SLAVE_VERSION})
 
 TARGET_LINK_LIBRARIES (slave
 	${MYSQL_CLIENT_LIBS})
@@ -119,6 +119,7 @@ INSTALL (FILES ${HEADERS}
 	DESTINATION "include/libslave")
 
 INSTALL (TARGETS slave
+	RUNTIME DESTINATION bin
 	LIBRARY DESTINATION lib COMPONENT runtime
 	ARCHIVE DESTINATION lib COMPONENT development
 	PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
--- origsrc/libslave/Slave.cpp	2013-02-02 21:45:18.000000000 +0900
+++ src/libslave/Slave.cpp	2013-02-02 21:48:11.891528400 +0900
@@ -720,7 +720,9 @@ void Slave::check_master_binlog_format()
 static bool checkAlterQuery(const std::string& str)
 {
     //RegularExpression regexp("^\\s*ALTER\\s+TABLE)", RegularExpression::RE_CASELESS);
-
+#if defined(_B)
+#undef _B
+#endif
     enum {
         SP0, _A0, _L0, _T0, _E0, _R, SP1, _T1, _A1, _B, _L1, _E1 
     } state = SP0;
--- origsrc/libslave/collate.cpp	2013-02-02 21:45:18.000000000 +0900
+++ src/libslave/collate.cpp	2013-02-02 21:49:04.750903400 +0900
@@ -6,6 +6,7 @@
 #include <vector>
 #include "nanomysql.h"
 #include "collate.h"
+#include <cstdlib>
 
 using namespace slave;
 
