--- origsrc/LHAPDF-6.5.4/Makefile.am	2022-06-09 18:41:38.000000000 +0900
+++ src/LHAPDF-6.5.4/Makefile.am	2024-10-25 06:40:57.584165100 +0900
@@ -1,6 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = src include wrappers tests bin examples doc
+SUBDIRS = src include wrappers tests bin # examples doc
 #DIST_SUBDIRS = $(SUBDIRS) examples doc
 
 dist_noinst_SCRIPTS = mkindex plotpdfs testpdfs testlhapdf
--- origsrc/LHAPDF-6.5.4/configure.ac	2023-03-11 02:35:19.000000000 +0900
+++ src/LHAPDF-6.5.4/configure.ac	2024-10-25 06:40:57.584165100 +0900
@@ -163,11 +163,11 @@ AC_SUBST(AM_CPPFLAGS)
 AC_SUBST(AM_CXXFLAGS)
 AC_SUBST(AM_LDFLAGS)
 
+PKG_CHECK_MODULES([YAML_CPP], [yaml-cpp])
 
 ## Output
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([src/Makefile])
-AC_CONFIG_FILES([src/yamlcpp/Makefile])
 AC_CONFIG_FILES([include/Makefile include/LHAPDF/Makefile])
 AC_CONFIG_FILES([bin/Makefile])
 AC_CONFIG_FILES([bin/lhapdf-config])
--- origsrc/LHAPDF-6.5.4/lhapdf.pc.in	2022-06-09 18:41:38.000000000 +0900
+++ src/LHAPDF-6.5.4/lhapdf.pc.in	2024-10-25 06:48:46.404652600 +0900
@@ -10,3 +10,4 @@ URL: https://lhapdf.hepforge.org
 Version: @VERSION@
 Libs: -L@libdir@ -lLHAPDF
 Cflags: -I@includedir@
+Requires: yaml-cpp
--- origsrc/LHAPDF-6.5.4/src/Info.cc	2023-03-11 02:35:19.000000000 +0900
+++ src/LHAPDF-6.5.4/src/Info.cc	2024-10-25 06:40:57.584165100 +0900
@@ -7,7 +7,7 @@
 #include "LHAPDF/PDFIndex.h"
 #include "LHAPDF/FileIO.h"
 
-#include "yaml-cpp/yaml.h"
+#include <yaml-cpp/yaml.h>
 #ifdef YAML_NAMESPACE
 #define YAML YAML_NAMESPACE
 #endif
--- origsrc/LHAPDF-6.5.4/src/Makefile.am	2022-06-09 18:41:38.000000000 +0900
+++ src/LHAPDF-6.5.4/src/Makefile.am	2024-10-25 06:40:57.584165100 +0900
@@ -1,10 +1,10 @@
-SUBDIRS = yamlcpp
+SUBDIRS = 
 
 lib_LTLIBRARIES = libLHAPDF.la
 noinst_LTLIBRARIES = libLHAPDFInfo.la libLHAPDFPaths.la
 
 AM_CPPFLAGS += -I$(top_srcdir)/include -I$(top_builddir)/include 
-AM_LDFLAGS += -L$(top_builddir)/src -avoid-version
+AM_LDFLAGS += -no-undefined -L$(top_builddir)/src
 if USING_MPI
   AM_CPPFLAGS += -DHAVE_MPI=1
 endif
@@ -19,8 +19,8 @@ libLHAPDF_la_SOURCES = \
   Config.cc Factories.cc PDFIndex.cc Utils.cc FileIO.cc
 
 libLHAPDFInfo_la_SOURCES = Info.cc
-libLHAPDFInfo_la_CPPFLAGS = -I$(srcdir)/yamlcpp -DYAML_NAMESPACE=LHAPDF_YAML $(AM_CPPFLAGS)
-libLHAPDFInfo_la_LIBADD = $(builddir)/yamlcpp/liblhapdf-yaml-cpp.la
+libLHAPDFInfo_la_CPPFLAGS = $(YAML_CPP_CFLAGS) $(AM_CPPFLAGS)
+libLHAPDFInfo_la_LIBADD = $(YAML_CPP_LIBS)
 
 libLHAPDFPaths_la_SOURCES = Paths.cc
 libLHAPDFPaths_la_CPPFLAGS = $(AM_CPPFLAGS) -DLHAPDF_INSTALL_PREFIX=\"$(prefix)\" -DLHAPDF_DATA_PREFIX=\"$(datadir)\"
