--- origsrc/LHAPDF-6.0.4/src/Makefile.am	2013-09-30 22:41:08.000000000 +0900
+++ src/LHAPDF-6.0.4/src/Makefile.am	2013-12-13 23:24:25.206869900 +0900
@@ -2,7 +2,7 @@ lib_LTLIBRARIES = libLHAPDF.la
 
 AM_CPPFLAGS += -I$(top_srcdir)/include -I$(prefix)/include $(BOOST_CPPFLAGS) $(YAMLCPP_CPPFLAGS)
 AM_CPPFLAGS += -DLHAPDF_INSTALL_PREFIX=\"$(prefix)\" -DLHAPDF_DATA_PREFIX=\"$(datadir)\" -DYAMLCPP_API=$(YAMLCPP_MAJOR_VERSION)
-AM_LDFLAGS += -L$(top_builddir)/src -L$(prefix)/lib $(BOOST_LDFLAGS) $(YAMLCPP_LDFLAGS) -avoid-version
+AM_LDFLAGS += -L$(top_builddir)/src -L$(prefix)/lib $(BOOST_LDFLAGS) $(YAMLCPP_LDFLAGS) -no-undefined
 
 libLHAPDF_la_SOURCES = \
   PDF.cc GridPDF.cc Paths.cc \
--- origsrc/LHAPDF-6.0.4/src/Paths.cc	2013-09-30 22:41:08.000000000 +0900
+++ src/LHAPDF-6.0.4/src/Paths.cc	2013-12-14 15:04:21.154368400 +0900
@@ -25,7 +25,9 @@ namespace LHAPDF {
 
 
   void setPaths(const std::string& pathstr) {
+#if !defined(__CYGWIN__)
     setenv("LHAPDF_DATA_PATH", pathstr.c_str(), 1);
+#endif
   }
 
 
