--- origsrc/lhapdf-5.8.7/configure.ac	2012-03-08 18:50:40.000000000 +0900
+++ src/lhapdf-5.8.7/configure.ac	2012-05-02 23:11:48.185085600 +0900
@@ -12,7 +12,7 @@ m4_ifdef([AM_SILENT_RULES], [AM_SILENT_R
 
 
 ## Library version flags (at least update the version comment with each new release version)
-VERSIONFLAGS="-version-info 1:6:5" ## for 5.8.6
+VERSIONFLAGS="-version-info 5:6:1" ## for 5.8.6
 AC_SUBST(VERSIONFLAGS)
 AC_DEFINE_UNQUOTED(LHAPDF_VERSION, "$PACKAGE_VERSION", "LHAPDF version string")
 
--- origsrc/lhapdf-5.8.7/lib/Makefile.am	2008-10-21 20:07:39.000000000 +0900
+++ src/lhapdf-5.8.7/lib/Makefile.am	2012-05-02 23:34:28.879421600 +0900
@@ -1,20 +1,16 @@
 ## Make main LHAPDF library
 lib_LTLIBRARIES = libLHAPDF.la
 libLHAPDF_la_SOURCES = 
-libLHAPDF_la_LDFLAGS = $(VERSIONFLAGS)
+libLHAPDF_la_LDFLAGS = $(VERSIONFLAGS) -no-undefined
 libLHAPDF_la_LIBADD = $(FCLIBS) $(top_builddir)/src/libLHAPDFCore.la
 if ENABLE_CCWRAP
 libLHAPDF_la_LIBADD += $(top_builddir)/ccwrap/libLHAPDFCCWrap.la
 endif
-## Hack to get linking done by CXX
-libLHAPDF_la_LINK = $(LIBTOOL) --tag=CXX --mode=link $(CXX) $(AM_CXXFLAGS) $(CXXFLAGS) \
-                    $(AM_LDFLAGS) $(LDFLAGS) -o $@
-
 
 ## Make deprecated compatibility C++ library
 if ENABLE_OLDCCWRAP
 lib_LTLIBRARIES += libLHAPDFWrap.la
 libLHAPDFWrap_la_SOURCES = 
-libLHAPDFWrap_la_LDFLAGS = -version-info 1:0:1
+libLHAPDFWrap_la_LDFLAGS = -version-info 1:0:1 -no-undefined
 libLHAPDFWrap_la_LIBADD = $(FCLIBS) libLHAPDF.la $(top_builddir)/ccwrap/libLHAPDFOldWrap.la
 endif
