--- origsrc/LHAPDF-6.5.6/src/Makefile.am	2025-03-13 19:39:25.000000000 +0900
+++ src/LHAPDF-6.5.6/src/Makefile.am	2026-05-12 22:12:34.983348500 +0900
@@ -7,7 +7,7 @@ lib_LTLIBRARIES = libLHAPDF.la
 noinst_LTLIBRARIES = libLHAPDFInfo.la libLHAPDFPaths.la
 
 AM_CPPFLAGS += -I$(top_srcdir)/include -I$(top_builddir)/include
-AM_LDFLAGS += -L$(top_builddir)/src -avoid-version
+AM_LDFLAGS += -L$(top_builddir)/src -no-undefined
 if USING_MPI
   AM_CPPFLAGS += -DHAVE_MPI=1
 endif
