--- origsrc/LHAPDF-6.1.0/include/LHAPDF/Utils.h	2014-04-15 00:00:58.000000000 +0900
+++ src/LHAPDF-6.1.0/include/LHAPDF/Utils.h	2014-05-08 15:42:46.410004300 +0900
@@ -29,10 +29,12 @@
 
 // Cosmetic wrapper for BOOST_FOREACH macro (until we can use C++11 range-based for loops)
 #ifndef foreach
+#if !defined(__CYGWIN__)
 namespace boost {
   // Suggested work-around for https://svn.boost.org/trac/boost/ticket/6131
   namespace BOOST_FOREACH = foreach;
 }
+#endif
 #define foreach BOOST_FOREACH
 #endif
 
--- origsrc/LHAPDF-6.1.0/lhapdf.pc	2014-04-26 00:02:17.000000000 +0900
+++ src/LHAPDF-6.1.0/lhapdf.pc	2014-05-08 15:21:11.683633600 +0900
@@ -1,13 +1,13 @@
-prefix=/home/andy/heplocal
-exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+prefix=/usr
+exec_prefix=/usr
+libdir=/usr/lib
 includedir=${prefix}/include
-datarootdir=${prefix}/share
+datarootdir=/usr/share
 
 Name: lhapdf
 Description: LHAPDF parton density library
 URL: https://lhapdf.hepforge.org
 Version: 6.1.0
-Libs: -L${exec_prefix}/lib -lLHAPDF
+Libs: -L/usr/lib -lLHAPDF
 Cflags: -I${prefix}/include
 #-I
--- origsrc/LHAPDF-6.1.0/src/GridPDF.cc	2014-04-17 21:40:51.000000000 +0900
+++ src/LHAPDF-6.1.0/src/GridPDF.cc	2014-05-08 15:45:57.624848100 +0900
@@ -72,7 +72,13 @@ namespace LHAPDF {
 
     private:
       void _get(double& x) { x = std::strtod(_next, &_new_next); }
-      void _get(float& x) { x = std::strtof(_next, &_new_next); }
+      void _get(float& x) {
+#if defined(__CYGWIN__)
+        x = std::strtod(_next, &_new_next);
+#else
+        x = std::strtof(_next, &_new_next);
+#endif
+      }
       void _get(int& i) { i = std::strtol(_next, &_new_next, 10); } // force base 10!
 
       char *_next, *_new_next;
--- origsrc/LHAPDF-6.1.0/src/Makefile.am	2014-03-24 03:54:46.000000000 +0900
+++ src/LHAPDF-6.1.0/src/Makefile.am	2014-05-08 15:18:00.700887300 +0900
@@ -4,7 +4,7 @@ lib_LTLIBRARIES = libLHAPDF.la
 noinst_LTLIBRARIES = libLHAPDFInfo.la libLHAPDFPaths.la
 
 AM_CPPFLAGS += -I$(top_srcdir)/include -I$(prefix)/include $(BOOST_CPPFLAGS)
-AM_LDFLAGS += -L$(top_builddir)/src -L$(prefix)/lib -avoid-version
+AM_LDFLAGS += -L$(top_builddir)/src -L$(prefix)/lib -no-undefined
 
 libLHAPDF_la_SOURCES = \
   PDF.cc PDFSet.cc GridPDF.cc PDFInfo.cc \
