--- origsrc/LHAPDF-6.1.5/lhapdf.pc	2014-12-23 02:09:01.000000000 +0900
+++ src/LHAPDF-6.1.5/lhapdf.pc	2015-01-10 10:02:12.411007800 +0900
@@ -1,13 +1,13 @@
-prefix=/home/andy/heplocal
-exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+prefix=/usr
+exec_prefix=/usr
+libdir=/usr/lib
 includedir=${prefix}/include
-datarootdir=${prefix}/share
+datarootdir=/usr/share
 
 Name: lhapdf
 Description: LHAPDF parton density library
 URL: https://lhapdf.hepforge.org
 Version: 6.1.5
-Libs: -L${exec_prefix}/lib -lLHAPDF
+Libs: -L/usr/lib -lLHAPDF
 Cflags: -I${prefix}/include
 #-I
--- origsrc/LHAPDF-6.1.5/src/GridPDF.cc	2014-12-24 06:54:48.000000000 +0900
+++ src/LHAPDF-6.1.5/src/GridPDF.cc	2015-01-10 10:06:00.989132800 +0900
@@ -13,6 +13,10 @@
 #include <stdexcept>
 #include <cstring>
 
+#if defined(__CYGWIN__)
+#define strtof strtod
+#endif
+
 using namespace std;
 
 namespace LHAPDF {
--- origsrc/LHAPDF-6.1.5/src/Makefile.am	2014-10-14 19:22:20.000000000 +0900
+++ src/LHAPDF-6.1.5/src/Makefile.am	2015-01-10 09:58:30.207882800 +0900
@@ -4,7 +4,7 @@ lib_LTLIBRARIES = libLHAPDF.la
 noinst_LTLIBRARIES = libLHAPDFInfo.la libLHAPDFPaths.la
 
 AM_CPPFLAGS += -I$(top_srcdir)/include -I$(prefix)/include $(BOOST_CPPFLAGS)
-AM_LDFLAGS += -L$(top_builddir)/src -L$(prefix)/lib -avoid-version
+AM_LDFLAGS += -no-undefined -L$(top_builddir)/src -L$(prefix)/lib
 
 libLHAPDF_la_SOURCES = \
   PDF.cc PDFSet.cc GridPDF.cc PDFInfo.cc \
