--- origsrc/HepMC3-3.2.3/CMakeLists.txt	2020-11-08 15:06:33.000000000 +0000
+++ src/HepMC3-3.2.3/CMakeLists.txt	2021-05-31 00:33:06.350164100 +0000
@@ -254,7 +254,7 @@ if (UNIX)
   target_link_libraries(HepMC3 PRIVATE dl)
 endif()
 #---Install build products---------------------------------------------------
-install(TARGETS HepMC3 DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT libs)
+install(TARGETS HepMC3 RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT libs)
 
 if (HEPMC3_BUILD_STATIC_LIBS)
   add_library(HepMC3_static STATIC  ${hepmc3core_sources})
@@ -286,8 +286,8 @@ install(DIRECTORY include/ DESTINATION $
 
 include(CMakePackageConfigHelpers)
 
-set(INTERFACES_INSTALL_DIR ${CMAKE_INSTALL_DATADIR}/HepMC3/interfaces)
-set(CONFIG_INSTALL_DIR ${CMAKE_INSTALL_DATADIR}/HepMC3/cmake)
+set(INTERFACES_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/HepMC3/interfaces)
+set(CONFIG_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/cmake/HepMC3)
 
 configure_package_config_file(cmake/Templates/HepMC3Config.cmake.in
   ${CMAKE_CURRENT_BINARY_DIR}/outputs/share/HepMC3/cmake/HepMC3Config.cmake
--- origsrc/HepMC3-3.2.3/search/CMakeLists.txt	2020-11-08 15:06:33.000000000 +0000
+++ src/HepMC3-3.2.3/search/CMakeLists.txt	2021-05-31 00:33:06.350164100 +0000
@@ -24,7 +24,7 @@ else()
                                                RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/outputs/${CMAKE_INSTALL_LIBDIR}/$<0:>)
 endif()
 # installs 
-install(TARGETS HepMC3search DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT searchlibs)
+install(TARGETS HepMC3search RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT searchlibs)
 if (HEPMC3_BUILD_STATIC_LIBS)
   add_library(HepMC3search_static STATIC  ${hepmc3search_sources})
   target_include_directories(HepMC3search_static PUBLIC ${PROJECT_SOURCE_DIR}/include)
