--- origsrc/twemcache-2.6.3/configure.ac	2017-09-30 08:51:36.000000000 +0900
+++ src/twemcache-2.6.3/configure.ac	2017-12-21 20:39:04.150000000 +0900
@@ -35,6 +35,7 @@ case $host_os in
     darwin*)  OS_DARWIN=yes
               AC_DEFINE([OS_DARWIN], [1], [Define to 1 if OS has Darwin kernel])
               ;;
+    cygwin*)  ;;
     *)        AC_MSG_ERROR([Your platform is not currently supported]) ;;
 esac
 AM_CONDITIONAL([RDYNAMIC], [test x$OS_LINUX = xyes])
