--- origsrc/zlog-1.2.17/CMakeLists.txt	2023-12-04 21:07:04.000000000 +0900
+++ src/zlog-1.2.17/CMakeLists.txt	2024-06-23 08:35:43.444666300 +0900
@@ -14,7 +14,7 @@ set(CMAKE_MODULE_PATH ${zlog_SOURCE_DIR}
 #=====================================
 SET(CPACK_PACKAGE_VERSION_MAJOR "1")
 SET(CPACK_PACKAGE_VERSION_MINOR "2")
-SET(CPACK_PACKAGE_VERSION_PATCH "12")
+SET(CPACK_PACKAGE_VERSION_PATCH "17")
 SET(CPACK_RPM_PACKAGE_RELEASE 1)    #release version.
 SET(CPACK_PACKAGE_VERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}.${CPACK_PACKAGE_VERSION_PATCH}")
 SET(zlog_ver ${CPACK_PACKAGE_VERSION})
@@ -76,3 +76,9 @@ if(UNIT_TEST)
 endif()
 #========================================================
 
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/zlog.pc.in ${CMAKE_BINARY_DIR}/zlog.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/zlog.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/zlog-1.2.17/src/CMakeLists.txt	2023-12-04 21:07:04.000000000 +0900
+++ src/zlog-1.2.17/src/CMakeLists.txt	2024-06-23 08:28:31.253105000 +0900
@@ -26,23 +26,6 @@ endif()
 
 set_target_properties(zlog PROPERTIES VERSION ${zlog_ver} SOVERSION ${zlog_so_ver})
 
-add_library(zlog_s
-    STATIC
-    ${SRCS}
-    )
-target_link_libraries(zlog_s
-    ${CMAKE_THREAD_PREFER_PTHREAD}
-    )
-
-if (WIN32)
-    target_link_libraries(zlog_s
-        ${UNIXEM_LIBRARY}
-        Ws2_32
-        )
-endif()
-
-set_target_properties(zlog_s PROPERTIES OUTPUT_NAME zlog)
-
 #================================================================
 # generate zlog-chk-conf
 #================================================================
@@ -53,7 +36,7 @@ target_link_libraries(zlog-chk-conf zlog
 # install
 #================================================================
 install(TARGETS
-    zlog zlog_s zlog-chk-conf
+    zlog zlog-chk-conf
     COMPONENT zlog
     ARCHIVE DESTINATION lib
     LIBRARY DESTINATION lib
--- origsrc/zlog-1.2.17/src/event.c	2023-12-04 21:07:04.000000000 +0900
+++ src/zlog-1.2.17/src/event.c	2024-06-23 08:29:48.307480100 +0900
@@ -20,7 +20,9 @@
 #include <sys/time.h>
 
 #include <sys/types.h>
+#if !defined(__CYGWIN__)
 #include <sys/syscall.h>
+#endif
 
 #include "zc_defs.h"
 #include "event.h"
--- origsrc/zlog-1.2.17/src/spec.c	2023-12-04 21:07:04.000000000 +0900
+++ src/zlog-1.2.17/src/spec.c	2024-06-23 08:34:43.734662200 +0900
@@ -259,10 +259,13 @@ static int zlog_spec_write_tid_long(zlog
 
 static int zlog_spec_write_ktid(zlog_spec_t * a_spec, zlog_thread_t * a_thread, zlog_buf_t * a_buf)
 {
-
+#if defined(__LINUX__) || defined(__APPLE__)
 	/* don't need to get ktid again, as tmap_new_thread fetched it already */
 	/* and fork not change tid */
 	return zlog_buf_append(a_buf, a_thread->event->ktid_str, a_thread->event->ktid_str_len);
+#else
+	return 0;
+#endif
 }
 
 static int zlog_spec_write_level_lowercase(zlog_spec_t * a_spec, zlog_thread_t * a_thread, zlog_buf_t * a_buf)
--- origsrc/zlog-1.2.17/test/CMakeLists.txt	2023-12-04 21:07:04.000000000 +0900
+++ src/zlog-1.2.17/test/CMakeLists.txt	2024-06-23 08:29:23.149578400 +0900
@@ -49,7 +49,7 @@ add_test(test_bitmap "${CMAKE_RUNTIME_OU
 
 file(GLOB CONF_FILES . *.conf)
 file(COPY
-    ${CONF_FILES} hello_output
+    ${CONF_FILES} # hello_output
     DESTINATION ${CMAKE_CURRENT_BINARY_DIR}
     )
 
--- origsrc/zlog-1.2.17/zlog.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/zlog-1.2.17/zlog.pc.in	2024-06-23 08:36:23.297068100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: zlog
+Description: Reliable, high-performance, thread safe, flexsible, clear-model, pure C logging library
+Version: @zlog_ver@
+Libs: -L${libdir} -lzlog
+Cflags: -I${includedir}
