--- origsrc/zlog-1.2.18/CMakeLists.txt	2024-04-11 15:16:08.000000000 +0900
+++ src/zlog-1.2.18/CMakeLists.txt	2024-09-09 08:39:29.828695300 +0900
@@ -24,7 +24,7 @@ set(CMAKE_MODULE_PATH ${zlog_SOURCE_DIR}
 
 SET(CPACK_PACKAGE_VERSION_MAJOR "1")
 SET(CPACK_PACKAGE_VERSION_MINOR "2")
-SET(CPACK_PACKAGE_VERSION_PATCH "12")
+SET(CPACK_PACKAGE_VERSION_PATCH "18")
 SET(CPACK_RPM_PACKAGE_RELEASE 1) # release version.
 SET(CPACK_PACKAGE_VERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}.${CPACK_PACKAGE_VERSION_PATCH}")
 SET(ZLOG_VERSION ${CPACK_PACKAGE_VERSION})
@@ -64,3 +64,10 @@ if (UNIT_TEST)
     enable_testing()
     add_subdirectory(test)
 endif ()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/zlog.pc.in ${CMAKE_BINARY_DIR}/zlog.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/zlog.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/zlog-1.2.18/src/CMakeLists.txt	2024-04-11 15:16:08.000000000 +0900
+++ src/zlog-1.2.18/src/CMakeLists.txt	2024-09-09 08:35:54.286276800 +0900
@@ -23,28 +23,11 @@ endif ()
 
 set_target_properties(zlog PROPERTIES VERSION ${ZLOG_VERSION} SOVERSION ${ZLOG_SO_VERSION})
 
-add_library(zlog_s
-        STATIC
-        ${SRCS}
-)
-target_link_libraries(zlog_s
-        ${CMAKE_THREAD_PREFER_PTHREAD}
-)
-
-if (WIN32)
-    target_link_libraries(zlog_s
-            ${UNIXEM_LIBRARY}
-            Ws2_32
-    )
-endif ()
-
-set_target_properties(zlog_s PROPERTIES OUTPUT_NAME zlog)
-
 add_executable(zlog-chk-conf zlog-chk-conf.c)
 target_link_libraries(zlog-chk-conf zlog)
 
 install(TARGETS
-        zlog zlog_s zlog-chk-conf
+        zlog zlog-chk-conf
         COMPONENT zlog
         ARCHIVE DESTINATION lib
         LIBRARY DESTINATION lib
--- origsrc/zlog-1.2.18/src/event.c	2024-04-11 15:16:08.000000000 +0900
+++ src/zlog-1.2.18/src/event.c	2024-09-09 08:36:20.878597300 +0900
@@ -27,7 +27,9 @@
 #include <sys/time.h>
 
 #include <sys/types.h>
+#if !defined(__CYGWIN__)
 #include <sys/syscall.h>
+#endif
 
 #include "zc_defs.h"
 #include "event.h"
--- origsrc/zlog-1.2.18/src/spec.c	2024-04-11 15:16:08.000000000 +0900
+++ src/zlog-1.2.18/src/spec.c	2024-09-09 08:38:24.084355100 +0900
@@ -266,10 +266,13 @@ static int zlog_spec_write_tid_long(zlog
 
 static int zlog_spec_write_ktid(zlog_spec_t * a_spec, zlog_thread_t * a_thread, zlog_buf_t * a_buf)
 {
-
+#if defined(__LINUX__) || defined(__APPLE__)
 	/* don't need to get ktid again, as tmap_new_thread fetched it already */
 	/* and fork not change tid */
 	return zlog_buf_append(a_buf, a_thread->event->ktid_str, a_thread->event->ktid_str_len);
+#else
+	return 0;
+#endif
 }
 
 static int zlog_spec_write_level_lowercase(zlog_spec_t * a_spec, zlog_thread_t * a_thread, zlog_buf_t * a_buf)
--- origsrc/zlog-1.2.18/zlog.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/zlog-1.2.18/zlog.pc.in	2024-09-09 08:42:16.145681900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: zlog
+Description: Reliable, high-performance, thread safe, flexsible, clear-model, pure C logging library
+Version: @ZLOG_VERSION@
+Libs: -L${libdir} -lzlog
+Cflags: -I${includedir}
