--- origsrc/ZRTPCPP-4.7.0/CMakeLists.txt	2020-02-02 03:04:00.000000000 +0900
+++ src/ZRTPCPP-4.7.0/CMakeLists.txt	2020-08-23 16:43:05.053032100 +0900
@@ -105,7 +105,7 @@ if (NOT LIB_SUFFIX)
     set(LIBDIRNAME "lib")
     # this caused problems in debian where it has to always be lib....
     if (NOT EXISTS /etc/debian_version)
-        if ( "${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86_64" )
+        if ( "${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86_64" AND NOT CYGWIN )
             set(LIBDIRNAME "lib64")
         endif()
     endif()
--- origsrc/ZRTPCPP-4.7.0/clients/ccrtp/CMakeLists.txt	2020-02-02 03:04:00.000000000 +0900
+++ src/ZRTPCPP-4.7.0/clients/ccrtp/CMakeLists.txt	2020-08-23 16:39:31.079207700 +0900
@@ -118,7 +118,7 @@ install(FILES ${CMAKE_SOURCE_DIR}/common
 
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/lib${zrtplibName}.pc DESTINATION ${LIBDIRNAME}/pkgconfig)
 
-install(TARGETS ${zrtplibName} DESTINATION ${LIBDIRNAME})
+install(TARGETS ${zrtplibName} RUNTIME DESTINATION bin ARCHIVE DESTINATION ${LIBDIRNAME})
 
 if(${PROJECT_NAME} STREQUAL ${CMAKE_PROJECT_NAME})
 
--- origsrc/ZRTPCPP-4.7.0/cryptcommon/ZrtpRandom.h	2020-02-02 03:04:00.000000000 +0900
+++ src/ZRTPCPP-4.7.0/cryptcommon/ZrtpRandom.h	2020-08-23 16:17:44.677782800 +0900
@@ -30,6 +30,7 @@
 #endif
 #include <stdint.h>
 #include <sys/types.h>
+#include <stdbool.h>
 
 #ifdef __cplusplus
 class ZrtpRandom {
--- origsrc/ZRTPCPP-4.7.0/zrtp/ZRtp.cpp	2020-02-02 03:04:00.000000000 +0900
+++ src/ZRTPCPP-4.7.0/zrtp/ZRtp.cpp	2020-08-23 16:09:45.973264600 +0900
@@ -18,6 +18,7 @@
  * Authors: Werner Dittmann <Werner.Dittmann@t-online.de>
  */
 #include <sstream>
+#include <sys/types.h>
 
 #include <crypto/zrtpDH.h>
 #include <crypto/hmac256.h>
