--- origsrc/libzrtpcpp-2.1.2/CMakeLists.txt	2012-02-25 00:33:01.000000000 +0900
+++ src/libzrtpcpp-2.1.2/CMakeLists.txt	2012-03-20 17:49:29.119243500 +0900
@@ -130,7 +130,7 @@ include_directories(BEFORE ${CMAKE_CURRE
 add_definitions(-g -O2 -fno-strict-aliasing)
 if(CMAKE_COMPILER_IS_GNUCXX)
   add_definitions(-Wno-long-long -Wno-char-subscripts)
-  add_definitions(-Wall -ansi -pedantic)
+  add_definitions(-Wall -pedantic)
   add_definitions(-DNEW_STDCPP)
 endif()
 
--- origsrc/libzrtpcpp-2.1.2/src/CMakeLists.txt	2012-02-25 00:33:01.000000000 +0900
+++ src/libzrtpcpp-2.1.2/src/CMakeLists.txt	2012-03-20 17:50:41.166118500 +0900
@@ -79,5 +79,5 @@ target_link_libraries(zrtpcpp ${LIBS})
 add_dependencies(zrtpcpp ccrtp)
 add_subdirectory(libzrtpcpp)
 
-install(TARGETS zrtpcpp DESTINATION ${LIBDIRNAME})
-
+install(TARGETS zrtpcpp RUNTIME DESTINATION bin
+                        ARCHIVE DESTINATION lib)
