--- origsrc/getxbook-1.2/config.mk	2015-07-29 00:52:40.000000000 +0900
+++ src/getxbook-1.2/config.mk	2024-09-21 18:43:13.413460700 +0900
@@ -8,8 +8,8 @@ MANPREFIX = $(PREFIX)/share/man
 
 LIBS = -lssl -lcrypto
 
-CFLAGS = -std=c99 -pedantic -Wall -Wextra -Werror -g -D_POSIX_C_SOURCE=200112L \
-         -DVERSION=\"$(VERSION)\"
+CFLAGS = -std=c99 -pedantic -Wall -Wextra -g -D_POSIX_C_SOURCE=200112L \
+         -DVERSION=\"$(VERSION)\" $(CYGCFLAGS)
 
 W32TCLKIT = tclkit-8.5.9-win32.upx.exe
 
