--- origsrc/bosphorus-bosphorus-3.0/CMakeLists.txt	2020-03-11 05:02:40.000000000 +0900
+++ src/bosphorus-bosphorus-3.0/CMakeLists.txt	2020-09-12 11:50:56.922734900 +0900
@@ -411,3 +411,10 @@ install(FILES
 install(EXPORT ${BOSPHORUS_EXPORT_NAME}
     DESTINATION "${BOSPHORUS_INSTALL_CMAKE_DIR}"
 )
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/bosphorus.pc.in ${CMAKE_BINARY_DIR}/bosphorus.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/bosphorus.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/bosphorus-bosphorus-3.0/bosphorus.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/bosphorus-bosphorus-3.0/bosphorus.pc.in	2020-09-12 11:58:49.426584100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: bosphorus
+Description: ANF simplifier and solver, and ANF-to-CNF converter
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lbosphorus
+Cflags: -I${includedir}
