--- origsrc/gnunet-0.19.2/pkgconfig/Makefile.am	2022-05-24 23:46:51.000000000 +0900
+++ src/gnunet-0.19.2/pkgconfig/Makefile.am	2023-01-19 18:29:48.080731600 +0900
@@ -10,7 +10,6 @@ pcfiles = \
        gnunetdatastore.pc \
        gnunetdht.pc \
        gnunetdns.pc \
-       gnunetenv.pc \
        gnunetfragmentation.pc \
        gnunetfs.pc \
        gnunetgns.pc \
--- origsrc/gnunet-0.19.2/pkgconfig/gnunetnat.pc.in	2022-05-24 23:46:51.000000000 +0900
+++ src/gnunet-0.19.2/pkgconfig/gnunetnat.pc.in	2023-01-19 18:29:48.084675800 +0900
@@ -8,5 +8,5 @@ Description: library for NAT traversal
 URL: https://gnunet.org
 Version: @VERSION@
 Requires:
-Libs: -L${libdir} -lgnunetnat
+Libs: -L${libdir} -lgnunetnatnew
 Cflags: -I${includedir}
--- origsrc/gnunet-0.19.2/src/dhtu/plugin_dhtu_ip.c	2022-12-05 13:12:02.000000000 +0900
+++ src/gnunet-0.19.2/src/dhtu/plugin_dhtu_ip.c	2023-01-19 18:29:48.088685000 +0900
@@ -1061,7 +1061,7 @@ libgnunet_plugin_dhtu_ip_init (void *cls
   case AF_INET6:
     {
       int on = 1;
-
+#if defined(IPV6_RECVPKTINFO)
       if (0 !=
           setsockopt (sock,
                       IPPROTO_IPV6,
@@ -1072,6 +1072,7 @@ libgnunet_plugin_dhtu_ip_init (void *cls
         GNUNET_log_strerror (GNUNET_ERROR_TYPE_ERROR,
                              "setsockopt");
       }
+#endif
     }
     {
       struct sockaddr_in6 sa = {
--- origsrc/gnunet-0.19.2/src/util/gnunet-qr.c	2022-12-05 13:12:02.000000000 +0900
+++ src/gnunet-0.19.2/src/util/gnunet-qr.c	2023-01-19 18:29:48.092793800 +0900
@@ -474,7 +474,7 @@ run_png_reader (void)
   zbar_image_scanner_set_config (scanner,0, ZBAR_CFG_ENABLE, 1);
 
   zbar_image_t *zimage = zbar_image_create ();
-  zbar_image_set_format (zimage, zbar_fourcc ('Y', '8', '0', '0'));
+  zbar_image_set_format (zimage, ((unsigned long)('Y') | ((unsigned long)('8') << 8) | ((unsigned long)('0') << 16) | ((unsigned long)('0') << 24)));
   zbar_image_set_size (zimage, width, height);
   zbar_image_set_data (zimage, buffer, width * height, &zbar_image_free_data);
 
