--- origsrc/gnunet-0.15.3/src/dhtu/plugin_dhtu_ip.c	2021-08-02 17:11:06.000000000 +0900
+++ src/gnunet-0.15.3/src/dhtu/plugin_dhtu_ip.c	2021-10-06 13:27:49.276722600 +0900
@@ -807,6 +807,7 @@ read_cb (void *cls)
         else
           GNUNET_break (0);
       }
+#if defined(IPV6_RECVPKTINFO)
       if ( (cmsg->cmsg_level == IPPROTO_IPV6) &&
            (cmsg->cmsg_type == IPV6_RECVPKTINFO) )
       {
@@ -834,6 +835,7 @@ read_cb (void *cls)
         else
           GNUNET_break (0);
       }
+#endif
     }
     dst = find_target (plugin,
                        &sa,
@@ -974,6 +976,7 @@ libgnunet_plugin_dhtu_ip_init (void *cls
     {
       int on = 1;
 
+#if defined(IPV6_RECVPKTINFO)
       if (0 !=
           setsockopt (sock,
                       IPPROTO_IPV6,
@@ -984,6 +987,7 @@ libgnunet_plugin_dhtu_ip_init (void *cls
         GNUNET_log_strerror (GNUNET_ERROR_TYPE_ERROR,
                              "setsockopt");
       }
+#endif
     }
     {
       struct sockaddr_in6 sa = {
