--- origsrc/tabixpp-1.1.2/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/tabixpp-1.1.2/Makefile.am	2023-09-27 20:56:01.685675000 +0900
@@ -0,0 +1,18 @@
+
+AM_CPPFLAGS = -D_FILE_OFFSET_BITS=64 -D_USE_KNETFILE
+
+bin_PROGRAMS = tabix++
+tabix___CPPFLAGS = $(AM_CPPFLAGS)
+tabix___SOURCES = main.cpp
+tabix___LDADD = libtabix.la
+
+include_HEADERS = tabix.hpp
+
+lib_LTLIBRARIES = libtabix.la
+libtabix_la_SOURCES = tabix.cpp
+libtabix_la_CPPFLAGS = $(AM_CPPFLAGS) $(HTSLIB_CFLAGS)
+libtabix_la_LDFLAGS = -no-undefined -version-info 1
+libtabix_la_LIBADD = $(HTSLIB_LIBS)
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/tabixpp-1.1.2/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/tabixpp-1.1.2/configure.ac	2023-09-27 20:43:53.156321400 +0900
@@ -0,0 +1,29 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.71])
+AC_INIT([tabixpp], [1.1.2])
+AC_CONFIG_SRCDIR([main.cpp])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+AC_PROG_INSTALL
+LT_INIT
+
+# Checks for libraries.
+PKG_CHECK_MODULES([HTSLIB], [htslib])
+
+# Checks for header files.
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+AC_TYPE_OFF_T
+
+# Checks for library functions.
+
+AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([tabixpp.pc])
+AC_OUTPUT
--- origsrc/tabixpp-1.1.2/tabixpp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/tabixpp-1.1.2/tabixpp.pc.in	2023-09-27 20:43:53.125074300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: tabixpp
+Description: C++ wrapper to tabix indexer
+Version: @VERSION@
+Libs: -L${libdir} -ltabixpp
+Cflags: -I${includedir}
