--- origsrc/sipcalc-1.1.6/configure.ac	2013-01-15 03:42:54.000000000 +0900
+++ src/sipcalc-1.1.6/configure.ac	2025-01-15 17:42:12.572464300 +0900
@@ -1,6 +1,6 @@
-AC_INIT(include/sub.h)
+AC_INIT([sipcalc], [1.1.6])
 AM_CONFIG_HEADER(config.h)
-AM_INIT_AUTOMAKE(sipcalc,1.1.6)
+AM_INIT_AUTOMAKE
 AC_PROG_CC
 AC_PROG_INSTALL
 
@@ -71,7 +71,7 @@ AC_TRY_COMPILE(
 AC_MSG_RESULT(no)))
 
 
-AC_DEFUN(sii_GETHOSTBYNAME,
+AC_DEFUN([sii_GETHOSTBYNAME],
 AC_MSG_CHECKING(for gethostbyname in netdb.h)
 AC_TRY_COMPILE(
 [
@@ -85,7 +85,7 @@ AC_TRY_COMPILE(
 AC_MSG_RESULT(no)))
 
 
-AC_DEFUN(sii_GETHOSTBYNAME2,
+AC_DEFUN([sii_GETHOSTBYNAME2],
 AC_MSG_CHECKING(for gethostbyname2 in netdb.h)
 AC_TRY_COMPILE(
 [
@@ -101,7 +101,7 @@ AC_MSG_RESULT(no)))
 
 dnl Check if struct sockaddr has member sa_len
 
-AC_DEFUN(sii_SA_LEN,
+AC_DEFUN([sii_SA_LEN],
 AC_MSG_CHECKING(if struct sockaddr has sa_len field)
 AC_TRY_COMPILE(
 [
@@ -119,7 +119,7 @@ AC_MSG_RESULT(no)))
 dnl Check if getopt_long is available.
 
 dnl Solaris has socket(2) in libsocket.
-AC_DEFUN(sii_LIBSOCKET,
+AC_DEFUN([sii_LIBSOCKET],
 [
   AC_CHECK_LIB(socket, socket,
   [
@@ -131,7 +131,7 @@ AC_DEFUN(sii_LIBSOCKET,
 
 dnl Check for uint[8,16,32]_t
 
-AC_DEFUN(sii_UINT8_T,
+AC_DEFUN([sii_UINT8_T],
 AC_MSG_CHECKING(if type uint8_t exists)
 AC_TRY_COMPILE(
 [
@@ -144,7 +144,7 @@ AC_TRY_COMPILE(
 ],
 AC_MSG_RESULT(no)))
 
-AC_DEFUN(sii_U_INT8_T,
+AC_DEFUN([sii_U_INT8_T],
 AC_MSG_CHECKING(if type u_int8_t exists)
 AC_TRY_COMPILE(
 [
@@ -159,7 +159,7 @@ AC_MSG_RESULT(no)))
 
 
 
-AC_DEFUN(sii_UINT16_T,
+AC_DEFUN([sii_UINT16_T],
 AC_MSG_CHECKING(if type uint16_t exists)
 AC_TRY_COMPILE(
 [
@@ -172,7 +172,7 @@ AC_TRY_COMPILE(
 ],
 AC_MSG_RESULT(no)))
 
-AC_DEFUN(sii_U_INT16_T,
+AC_DEFUN([sii_U_INT16_T],
 AC_MSG_CHECKING(if type u_int16_t exists)
 AC_TRY_COMPILE(
 [
@@ -187,7 +187,7 @@ AC_MSG_RESULT(no)))
 
 
 
-AC_DEFUN(sii_UINT32_T,
+AC_DEFUN([sii_UINT32_T],
 AC_MSG_CHECKING(if type uint32_t exists)
 AC_TRY_COMPILE(
 [
@@ -200,7 +200,7 @@ AC_TRY_COMPILE(
 ],
 AC_MSG_RESULT(no)))
 
-AC_DEFUN(sii_U_INT32_T,
+AC_DEFUN([sii_U_INT32_T],
 AC_MSG_CHECKING(if type u_int32_t exists)
 AC_TRY_COMPILE(
 [
@@ -231,4 +231,3 @@ AC_OUTPUT(Makefile			\
 	  include/Makefile		\
 	  src/Makefile			\
 	 )
-
--- origsrc/sipcalc-1.1.6/src/Makefile.am	2013-01-08 07:04:43.000000000 +0900
+++ src/sipcalc-1.1.6/src/Makefile.am	2025-01-15 17:42:12.572464300 +0900
@@ -1,9 +1,3 @@
 INCLUDES = -I$(top_srcdir)/include
 bin_PROGRAMS = sipcalc
 sipcalc_SOURCES = sub-func.c interface.c sub-output.c sub.c
-
-sub-func.o: sub-func.c ../include/sub.h ../include/sub-o.h
-interface.o: interface.c ../include/sub.h
-sub-output.o: sub-output.c ../include/sub.h
-sub.o: sub.c ../include/sub.h ../include/sub-o.h
-
--- origsrc/sipcalc-1.1.6/src/sub.c	2013-01-15 03:20:22.000000000 +0900
+++ src/sipcalc-1.1.6/src/sub.c	2025-01-15 17:42:12.572464300 +0900
@@ -51,8 +51,10 @@ static const char rcsid[] = "$Id: sub.c,
 #include "sub.h"
 #include "sub-o.h"
 
+#if !defined(__CYGWIN__)
 extern char *optarg;
 extern int optind, opterr, optopt;
+#endif
 int resolve;
 
 int
