--- origsrc/libxo-1.7.5/libxo/libxo.c	2024-04-19 07:56:33.000000000 +0900
+++ src/libxo-1.7.5/libxo/libxo.c	2024-09-14 08:49:38.670749800 +0900
@@ -43,6 +43,10 @@
 #include <wctype.h>
 #include <getopt.h>
 
+#if defined(__CYGWIN__)
+#include <rpc/types.h>
+#endif
+
 #include "xo_config.h"
 
 #ifdef HAVE_LANGINFO_H
--- origsrc/libxo-1.7.5/libxo/xo_syslog.c	2024-04-19 07:56:33.000000000 +0900
+++ src/libxo-1.7.5/libxo/xo_syslog.c	2024-09-14 08:46:14.219514900 +0900
@@ -148,6 +148,10 @@ static xo_syslog_close_t xo_syslog_close
 
 static char xo_syslog_enterprise_id[12];
 
+#if !defined(_PATH_CONSOLE)
+#define _PATH_CONSOLE "/dev/console"
+#endif
+
 /*
  * Record an enterprise ID, which functions as a namespace for syslog
  * messages.  The value is pre-formatted into a string.  This allows
