--- origsrc/stdair-1.00.1/config/project_config_embeddable.cmake	2012-12-23 10:29:49.000000000 +0900
+++ src/stdair-1.00.1/config/project_config_embeddable.cmake	2014-11-08 12:41:12.520539000 +0900
@@ -525,7 +525,7 @@ macro (get_boost)
   set (Boost_USE_STATIC_RUNTIME OFF)
   set (BOOST_REQUIRED_COMPONENTS
     regex program_options date_time iostreams serialization filesystem 
-    unit_test_framework python)
+    unit_test_framework python system)
 
   # The first check is for the required components.
   find_package (Boost COMPONENTS ${BOOST_REQUIRED_COMPONENTS})
@@ -543,7 +543,8 @@ macro (get_boost)
     list (APPEND PROJ_DEP_LIBS_FOR_LIB
       ${Boost_REGEX_LIBRARY} ${Boost_IOSTREAMS_LIBRARY} 
 	  ${Boost_SERIALIZATION_LIBRARY} ${Boost_FILESYSTEM_LIBRARY}
-	  ${Boost_DATE_TIME_LIBRARY} ${Boost_PYTHON_LIBRARY})
+	  ${Boost_DATE_TIME_LIBRARY} ${Boost_PYTHON_LIBRARY}
+	  ${Boost_SYSTEM_LIBRARY})
     list (APPEND PROJ_DEP_LIBS_FOR_BIN
 	  ${Boost_REGEX_LIBRARY} ${Boost_PROGRAM_OPTIONS_LIBRARY})
     list (APPEND PROJ_DEP_LIBS_FOR_TST ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY})
@@ -1443,7 +1444,8 @@ macro (module_library_add_specific
   # Installation of the library
   install (TARGETS ${_lib_target}
     EXPORT ${LIB_DEPENDENCY_EXPORT}
-    LIBRARY DESTINATION "${INSTALL_LIB_DIR}" COMPONENT runtime)
+    RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
+    ARCHIVE DESTINATION "${INSTALL_LIB_DIR}")
 
   # Register, for reporting purpose, the list of libraries to be built
   # and installed for that module
