--- origsrc/liblo-0.32/src/Makefile.am	2023-07-05 04:54:07.000000000 +0900
+++ src/liblo-0.32/src/Makefile.am	2024-03-19 15:56:43.493511300 +0900
@@ -19,12 +19,12 @@ endif
 lib_LTLIBRARIES = liblo.la
 
 if WINDOWS_DLL
-lt_windows = -no-undefined -export-symbols liblo.def
+lt_windows = -export-symbols liblo.def
 endif
 
 liblo_la_CFLAGS = -Wall -I$(top_srcdir)
 liblo_la_SOURCES = $(SOURCE_FILES)
-liblo_la_LDFLAGS  = $(lt_windows) -export-dynamic -version-info @LO_SO_VERSION@
+liblo_la_LDFLAGS  = $(lt_windows) -no-undefined -export-dynamic -version-info @LO_SO_VERSION@
 
 # Test programs depend on server threads, so skip them if threads are
 # disabled.
