--- origsrc/linenoise/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/linenoise/Makefile.am	2014-06-29 10:30:12.053048900 +0900
@@ -0,0 +1,11 @@
+
+check_PROGRAMS = example
+example_SOURCES = example.c
+example_LDADD = liblinenoise.la
+
+include_HEADERS = linenoise.h
+
+lib_LTLIBRARIES = liblinenoise.la
+liblinenoise_la_SOURCES = linenoise.c
+liblinenoise_la_LDFLAGS = -no-undefined
+
--- origsrc/linenoise/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/linenoise/configure.ac	2014-06-29 10:33:10.907541100 +0900
@@ -0,0 +1,28 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([linenoise], [0+git20140419])
+AC_CONFIG_SRCDIR([example.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h string.h sys/ioctl.h termios.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([atexit memmove memset strcasecmp strchr strdup])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
