--- origsrc/enkiTS-1.11/CMakeLists.txt	2022-03-04 00:34:03.000000000 +0900
+++ src/enkiTS-1.11/CMakeLists.txt	2022-05-02 08:39:23.184993100 +0900
@@ -60,7 +60,12 @@ if( UNIX )
 endif()
 
 if( ENKITS_INSTALL )
-    install(TARGETS enkiTS DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/enkiTS")
+    if (CYGWIN)
+        install(TARGETS enkiTS RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+        set_target_properties(enkiTS PROPERTIES SOVERSION "${SOVERSION}")
+    else()
+        install(TARGETS enkiTS DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/enkiTS")
+    endif()
     install(FILES ${ENKITS_HEADERS} DESTINATION "${CMAKE_INSTALL_PREFIX}/include/enkiTS")
 endif()
 
@@ -143,3 +148,12 @@ if( ENKITS_BUILD_C_INTERFACE )
 endif()    
 
 endif()
+
+if (CYGWIN)
+    set(prefix ${CMAKE_INSTALL_PREFIX})
+    set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+    set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+    set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+    configure_file(${CMAKE_SOURCE_DIR}/enkiTS.pc.in ${CMAKE_BINARY_DIR}/enkiTS.pc @ONLY)
+    install(FILES ${CMAKE_BINARY_DIR}/enkiTS.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+endif()
--- origsrc/enkiTS-1.11/enkiTS.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/enkiTS-1.11/enkiTS.pc.in	2022-05-02 08:37:04.400767200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: enkiTS
+Description: C and C++ Task Scheduler for creating parallel programs
+Version: @VERSION@
+Libs: -L${libdir} -lenkiTS
+Cflags: -I${includedir}
