--- origsrc/c-blosc2-2.0.3/blosc/CMakeLists.txt	2021-08-31 19:03:21.000000000 +0900
+++ src/c-blosc2-2.0.3/blosc/CMakeLists.txt	2021-09-20 16:49:16.732126600 +0900
@@ -291,7 +291,7 @@ if(BLOSC_INSTALL)
         install(TARGETS blosc2_shared
                 LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
                 ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-                RUNTIME DESTINATION bin COMPONENT LIB)
+                RUNTIME DESTINATION bin)
     endif()
     if(BUILD_STATIC)
         install(TARGETS blosc2_static
--- origsrc/c-blosc2-2.0.3/blosc/config.h	1970-01-01 09:00:00.000000000 +0900
+++ src/c-blosc2-2.0.3/blosc/config.h	2021-09-20 16:14:13.234991300 +0900
@@ -0,0 +1,11 @@
+#ifndef _CONFIGURATION_HEADER_GUARD_H_
+#define _CONFIGURATION_HEADER_GUARD_H_
+
+#define HAVE_ZLIB TRUE
+/* #undef HAVE_ZLIB_NG */
+#define HAVE_ZSTD TRUE
+/* #undef HAVE_IPP */
+/* #undef BLOSC_DLL_EXPORT */
+#define HAVE_PLUGINS TRUE
+
+#endif
