diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*.whl' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x config.h origsrc/c-blosc2-2.7.1/bench/b2nd/CMakeLists.txt src/c-blosc2-2.7.1/bench/b2nd/CMakeLists.txt
--- origsrc/c-blosc2-2.7.1/bench/b2nd/CMakeLists.txt	2023-02-22 21:12:49.000000000 +0900
+++ src/c-blosc2-2.7.1/bench/b2nd/CMakeLists.txt	2023-03-23 20:05:34.131595400 +0900
@@ -12,5 +12,6 @@ foreach (source ${SOURCES})
     get_filename_component(target_name ${source} NAME_WE)
     set(target b2nd_${target_name})
     add_executable(${target} ${target_name}.c)
-    target_link_libraries(${target} blosc2_static ${LIBS})
+    target_link_libraries(${target} blosc2_shared ${LIBS})
+    target_include_directories(${target} PUBLIC "${CMAKE_SOURCE_DIR}/blosc")
 endforeach (source)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*.whl' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x config.h origsrc/c-blosc2-2.7.1/examples/b2nd/CMakeLists.txt src/c-blosc2-2.7.1/examples/b2nd/CMakeLists.txt
--- origsrc/c-blosc2-2.7.1/examples/b2nd/CMakeLists.txt	2023-02-22 21:12:49.000000000 +0900
+++ src/c-blosc2-2.7.1/examples/b2nd/CMakeLists.txt	2023-03-23 20:07:12.609564600 +0900
@@ -17,7 +17,7 @@ foreach (source ${SOURCES})
     get_filename_component(target_name ${source} NAME_WE)
     set(target b2nd_${target_name})
     add_executable(${target} ${target_name}.c)
-    target_link_libraries(${target} blosc2_static ${LIBS})
+    target_link_libraries(${target} blosc2_shared ${LIBS})
     if (${target_name} STREQUAL example_print_meta)
         add_test(NAME ${target} COMMAND ${target}
                 "${PROJECT_BINARY_DIR}/examples/b2nd/example_big_float_frame.b2nd")
