--- origsrc/jansson-1.2.1/configure.ac	2010-04-03 19:49:56.000000000 +0900
+++ src/jansson-1.2.1/configure.ac	2010-05-23 19:39:39.151768500 +0900
@@ -8,6 +8,7 @@ AC_CONFIG_HEADERS([config.h])
 
 # Checks for programs.
 AC_PROG_CC
+AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
 
 # Checks for libraries.
--- origsrc/jansson-1.2.1/Makefile.am	2010-01-20 04:29:06.000000000 +0900
+++ src/jansson-1.2.1/Makefile.am	2010-05-23 20:09:20.526768500 +0900
@@ -1,5 +1,5 @@
 EXTRA_DIST = CHANGES LICENSE README.rst
-SUBDIRS = doc src test
+SUBDIRS = src test
 
 check-local: html
 
--- origsrc/jansson-1.2.1/README	1970-01-01 09:00:00.000000000 +0900
+++ src/jansson-1.2.1/README	2010-05-23 20:33:05.729893500 +0900
@@ -0,0 +1,11 @@
+Jansson  is a C library for encoding, decoding and manipulating JSON data.
+Its main features and design principles are:
+
+    * Simple and intuitive API and data model
+    * Comprehensive documentation
+    * No dependencies on other libraries
+    * Full Unicode support (UTF-8)
+    * Extensive test suite
+
+Jansson is licensed under the MIT license; see LICENSE in the source
+distribution for details.
--- origsrc/jansson-1.2.1/src/Makefile.am	2010-04-03 19:49:56.000000000 +0900
+++ src/jansson-1.2.1/src/Makefile.am	2010-05-23 19:51:00.792393500 +0900
@@ -15,6 +15,7 @@ libjansson_la_SOURCES = \
 	value.c
 libjansson_la_LDFLAGS = \
 	-export-symbols-regex '^json_' \
-	-version-info 2:1:2
+	-version-info 2:1:2 \
+	-no-undefined
 
-AM_CFLAGS = -Wall -Wextra -Werror
+AM_CFLAGS = -Wall -Wextra #-Werror
