--- origsrc/jansson-1.3/Makefile.am	2010-06-11 02:49:16.000000000 +0900
+++ src/jansson-1.3/Makefile.am	2010-07-30 00:29:00.913814100 +0900
@@ -1,5 +1,5 @@
 EXTRA_DIST = CHANGES LICENSE README.rst
-SUBDIRS = doc src test
+SUBDIRS = src test
 
 check-local: html
 
--- origsrc/jansson-1.3/README	1970-01-01 09:00:00.000000000 +0900
+++ src/jansson-1.3/README	2010-07-30 00:30:17.752994100 +0900
@@ -0,0 +1,11 @@
+Jansson  is a C library for encoding, decoding and manipulating JSON data.
+Its main features and design principles are:
+
+    * Simple and intuitive API and data model
+    * Comprehensive documentation
+    * No dependencies on other libraries
+    * Full Unicode support (UTF-8)
+    * Extensive test suite
+
+Jansson is licensed under the MIT license; see LICENSE in the source
+distribution for details.
--- origsrc/jansson-1.3/configure.ac	2010-06-13 04:45:49.000000000 +0900
+++ src/jansson-1.3/configure.ac	2010-07-30 00:29:28.831119100 +0900
@@ -9,6 +9,7 @@ AC_CONFIG_HEADERS([config.h])
 # Checks for programs.
 AC_PROG_CC
 AC_PROG_CXX
+AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
 
 # Checks for libraries.
--- origsrc/jansson-1.3/src/Makefile.am	2010-06-13 04:45:49.000000000 +0900
+++ src/jansson-1.3/src/Makefile.am	2010-07-30 00:46:17.440494100 +0900
@@ -15,6 +15,6 @@ libjansson_la_SOURCES = \
 	value.c
 libjansson_la_LDFLAGS = \
 	-export-symbols-regex '^json_' \
-	-version-info 3:0:3
+	-version-info 3:0:3 -no-undefined
 
-AM_CFLAGS = -Wall -Wextra -Werror
+AM_CFLAGS = -Wall -Wextra #-Werror
