--- origsrc/jansson-2.7/Makefile.am	2014-10-02 13:59:26.000000000 +0900
+++ src/jansson-2.7/Makefile.am	2014-10-23 23:51:36.039409900 +0900
@@ -1,5 +1,5 @@
 EXTRA_DIST = CHANGES LICENSE README.rst CMakeLists.txt cmake android
-SUBDIRS = doc src test
+SUBDIRS = src test
 
 # "make distcheck" builds the dvi target, so use it to check that the
 # documentation is built correctly.
--- origsrc/jansson-2.7/test/bin/Makefile.am	2014-10-02 13:59:26.000000000 +0900
+++ src/jansson-2.7/test/bin/Makefile.am	2014-10-23 23:50:50.239605200 +0900
@@ -1,5 +1,4 @@
 check_PROGRAMS = json_process
 
 AM_CPPFLAGS = -I$(top_builddir)/src -I$(top_srcdir)/src
-LDFLAGS = -static  # for speed and Valgrind
 LDADD = $(top_builddir)/src/libjansson.la
--- origsrc/jansson-2.7/test/suites/api/Makefile.am	2014-10-02 13:59:26.000000000 +0900
+++ src/jansson-2.7/test/suites/api/Makefile.am	2014-10-23 23:51:04.014019300 +0900
@@ -30,5 +30,4 @@ test_simple_SOURCES = test_simple.c util
 test_unpack_SOURCES = test_unpack.c util.h
 
 AM_CPPFLAGS = -I$(top_builddir)/src -I$(top_srcdir)/src
-LDFLAGS = -static  # for speed and Valgrind
 LDADD = $(top_builddir)/src/libjansson.la
