--- origsrc/libhdate-1.6.02/bindings/python/Makefile.am	2026-05-23 10:10:59.921943600 +0900
+++ src/libhdate-1.6.02/bindings/python/Makefile.am	2026-05-23 10:40:03.176413800 +0900
@@ -3,23 +3,14 @@ EXTRA_DIST = hdate.i
 
 if WITH_PYTHON
 
-DEPS= $(top_builddir)/src/libhdate.la
-
-CLEANFILES = *.o *.cxx *.so *wrap* *.py
-
-all-am: _hdate.so
-	
-
-_hdate.so: hdate_wrap.cxx
-	g++ -fpic -c hdate_wrap.cxx $(CPPFLAGS) $(CFLAGS) $(PY_CFLAGS) -I$(top_srcdir)/src
-	g++ -shared hdate_wrap.o ../../src/.libs/libhdate.so -o _hdate.so $(LDFLAGS)
-
 hdate_wrap.cxx: hdate.i
-	swig -python -c++ hdate.i
+	$(SWIG) -python -c++ $(srcdir)/hdate.i
 
-install-data-local: _hdate.so
-	$(mkinstalldirs) $(DESTDIR)$(PY_MODULES_PATH)
-	$(INSTALL_DATA) _hdate.so $(DESTDIR)$(PY_MODULES_PATH)/_hdate.so
-	$(INSTALL_DATA) hdate.py $(DESTDIR)$(PY_MODULES_PATH)/hdate.py
+python_PYTHON = $(builddir)/hdate.py
+pyexec_LTLIBRARIES = _hdate.la
+_hdate_la_SOURCES = $(builddir)/hdate_wrap.cxx
+_hdate_la_CPPFLAGS = -I$(top_srcdir)/src $(PYTHON_CPPFLAGS)
+_hdate_la_LDFLAGS = -avoid-version -no-undefined -module $(PYTHON_LDFLAGS)
+_hdate_la_LIBADD = $(top_builddir)/src/libhdate.la $(PYTHON_LIBS)
 
 endif
--- origsrc/libhdate-1.6.02/configure.in	2026-05-23 10:10:59.937406100 +0900
+++ src/libhdate-1.6.02/configure.in	2026-05-23 10:21:12.217856300 +0900
@@ -14,20 +14,18 @@ VERSION_INFO=$LIBHDATE_MAJOR_VERSION:$LI
 AC_SUBST(VERSION_INFO)
 
 AM_INIT_AUTOMAKE($PACKAGE, $VERSION)
-AM_MAINTAINER_MODE
 
 dnl Checks for programs.
 AC_ISC_POSIX
 AC_GNU_SOURCE
 AC_USE_SYSTEM_EXTENSIONS
 AC_PROG_CC
+AC_PROG_CXX
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
 AC_STDC_HEADERS
-AM_PROG_LIBTOOL
-
-AM_MAINTAINER_MODE
+LT_INIT
 
 AC_C_CONST
 AC_CONFIG_MACRO_DIR([m4])
@@ -83,44 +81,16 @@ AM_CONDITIONAL(WITH_FPC, [test "$build_g
 
 dnl =======================================================================================
 
-AC_CHECK_PROG(have_python, "python", yes, no)
-
-build_python=$have_python
+AX_PKG_SWIG
 
 AC_ARG_ENABLE([python], 
   [AS_HELP_STRING([--disable-python],
   [do not build python binding])],
-  [build_python=no], [build_python=$have_python])
+  [build_python=no], [build_python=yes])
 
-AC_ARG_WITH(python-sitelib-dir, 
-  [AS_HELP_STRING([--with-python-sitelib-dir=PATH],
-  [path to python site lib])])
-
-if test "$have_python" = "yes"; then
-	  PY_PREFIX=`python -c 'import sys ; print sys.prefix'`
-	  PY_VERSION=`python -c 'import sys ; print sys.version[[0:3]]'`
-	  AC_MSG_CHECKING(for $PY_PREFIX/include/python$PY_VERSION/Python.h)
-	  if test -f $PY_PREFIX/include/python$PY_VERSION/Python.h; then
-		  AC_MSG_RESULT(yes)
-		  PY_CFLAGS="-I$PY_PREFIX/include/python$PY_VERSION"
-		  
-		  if test "$with_python_sitelib_dir" = "" ; then
-		    PY_MODULES_PATH="$PY_PREFIX/lib/python$PY_VERSION/dist-packages"
-		  else
-		    PY_MODULES_PATH=$with_python_sitelib_dir
-		  fi
-		  
-		  AC_SUBST([PY_VERSION])
-		  AC_SUBST([PY_CFLAGS])
-		  AC_SUBST([PY_MODULES_PATH])
-		  if test "$have_swig" = "no"; then
-			build_python=no
-		  fi
-	  else
-		  AC_MSG_RESULT(no)
-		  build_python=no
-	  fi
-fi
+AM_PATH_PYTHON([3])
+AX_PYTHON_DEVEL
+AX_SWIG_PYTHON
 
 AM_CONDITIONAL(WITH_PYTHON, test "$build_python" = "yes")
 
--- origsrc/libhdate-1.6.02/src/Makefile.am	2013-02-22 06:27:08.000000000 +0900
+++ src/libhdate-1.6.02/src/Makefile.am	2026-05-23 10:11:26.924430300 +0900
@@ -12,7 +12,7 @@ libhdate_la_SOURCES = \
 	hdate.h\
 	support.h
 
-libhdate_la_LDFLAGS = $(libhdate_LDFLAGS) -version-info $(VERSION_INFO)
+libhdate_la_LDFLAGS = -no-undefined $(libhdate_LDFLAGS) -version-info $(VERSION_INFO)
 
 libhdate_la_LIBADD = 
 
