--- origsrc/libBoxedText/acinclude.m4	2019-07-20 22:47:51.000000000 +0900
+++ src/libBoxedText/acinclude.m4	2019-07-20 22:48:39.328191900 +0900
@@ -6,7 +6,7 @@ dnl Define HAVE_STRUCT_UTIMBUF if `struc
 dnl usually in <utime.h>.
 dnl Some systems have utime.h but don't declare the struct anywhere.
 
-AC_DEFUN(jm_CHECK_TYPE_STRUCT_UTIMBUF,
+AC_DEFUN([jm_CHECK_TYPE_STRUCT_UTIMBUF],
 [
   AC_CHECK_HEADERS(utime.h)
   AC_REQUIRE([AC_HEADER_TIME])
--- origsrc/libBoxedText/configure.ac	2019-07-20 22:47:51.000000000 +0900
+++ src/libBoxedText/configure.ac	2019-07-20 22:56:17.772222500 +0900
@@ -225,7 +225,7 @@ AC_PROG_MAKE_SET
 AC_PROG_AWK
 AC_PROG_CPP
 AC_PROG_LN_S
-AC_PROG_RANLIB
+LT_INIT
 AC_PROG_EGREP
 
 AC_REVISION([m4_esyscmd_s([./config/git-version-gen --timestamp --force-update configure.ac])])
--- origsrc/libBoxedText/doc/Makefile.am	2019-07-20 22:47:51.000000000 +0900
+++ src/libBoxedText/doc/Makefile.am	2019-07-20 23:05:00.214984000 +0900
@@ -78,26 +78,26 @@
 #                                                                             #
 ###############################################################################
 
-include doxygen.am
+# include doxygen.am
 
 # ensure the distribution of the doxygen configuration file
 #EXTRA_DIST = $(DX_CONFIG)
 
-if DX_COND_doc
+# if DX_COND_doc
 
-if DX_COND_man
+# if DX_COND_man
 
-directory = $(top_builddir)/doc/docs/man/man3/
+# directory = $(top_builddir)/doc/docs/man/man3/
 
-man3_MANS = $(directory)/DoccY.3 $(directory)/DoccY_BoxedText.3 $(directory)/boxedtext.h.3
+# man3_MANS = $(directory)/DoccY.3 $(directory)/DoccY_BoxedText.3 $(directory)/boxedtext.h.3
 
-$(man3_MANS): @DX_DOCDIR@/@PACKAGE@.tag
+# $(man3_MANS): @DX_DOCDIR@/@PACKAGE@.tag
 
-endif
+# endif
 
-all-local: doxygen-run
+# all-local: doxygen-run
 
-endif
+# endif
 
 MOSTLYCLEANFILES=*~ $(DX_CLEANFILES)
 CLEANFILES=*~
--- origsrc/libBoxedText/src/Makefile.am	2019-07-20 22:47:51.000000000 +0900
+++ src/libBoxedText/src/Makefile.am	2019-07-20 23:15:31.764206900 +0900
@@ -93,14 +93,14 @@ libBoxedText_la_SOURCES = boxedtext.cpp
 
 libBoxedText_la_configdir = @libdir@/@PACKAGE@-@VERSION@/include
 libBoxedText_la_config_DATA = @abs_top_builddir@/config/config.h
-libBoxedText_la_LDFLAGS = -version-info "@LIB_VERSION@" -release "@VERSION@" $(AM_LDFLAGS)
+libBoxedText_la_LDFLAGS = -no-undefined -version-info "@LIB_VERSION@" -release "@VERSION@" $(AM_LDFLAGS)
 libBoxedText_la_CXXFLAGS = -DPACKAGE_DATADIR=\"$(pkgdatadir)\"
 
 
 noinst_PROGRAMS = test-bt
 test_bt_SOURCES = test-bt.cpp
 test_bt_LDADD=libBoxedText.la
-test_bt_LDFLAGS = -lBoxedText $(AM_LDFLAGS)
+test_bt_LDFLAGS = libBoxedText.la $(AM_LDFLAGS)
 
 install-data-hook:
 	$(LN_S) -f "@PACKAGE@-@VERSION@" $(DESTDIR)@includedir@/@PACKAGE@
