--- origsrc/libHX-4.0.1/src/socket.c	2021-10-03 08:05:06.000000000 +0900
+++ src/libHX-4.0.1/src/socket.c	2021-10-10 21:52:55.220541400 +0900
@@ -30,18 +30,22 @@ static int try_sk_from_env(int fd, const
 	int ret = getsockopt(fd, SOL_SOCKET, SO_ACCEPTCONN, &value, &optlen);
 	if (ret < 0 || value == 0)
 		return -1;
+#if defined(SO_DOMAIN)
 	optlen = sizeof(value);
 	ret = getsockopt(fd, SOL_SOCKET, SO_DOMAIN, &value, &optlen);
 	if (ret < 0 || value != ai->ai_family)
 		return -1;
+#endif
 	optlen = sizeof(value);
 	ret = getsockopt(fd, SOL_SOCKET, SO_TYPE, &value, &optlen);
 	if (ret < 0 || value != ai->ai_socktype)
 		return -1;
+#if defined(SO_PROTOCOL)
 	optlen = sizeof(value);
 	ret = getsockopt(fd, SOL_SOCKET, SO_PROTOCOL, &value, &optlen);
 	if (ret < 0 || value != ai->ai_protocol)
 		return -1;
+#endif
 	struct sockaddr_storage addr;
 	memset(&addr, 0, sizeof(addr));
 	optlen = sizeof(addr);
@@ -54,6 +58,7 @@ static int try_sk_from_env(int fd, const
 		return -1;
 	if (intf == nullptr)
 		return fd;
+#if defined(SO_BINDTODEVICE)
 	char ifname[32];
 	optlen = sizeof(ifname);
 	ret = getsockopt(fd, SOL_SOCKET, SO_BINDTODEVICE, ifname, &optlen);
@@ -65,6 +70,7 @@ static int try_sk_from_env(int fd, const
 		ifname[sizeof(ifname)-1] = '\0';
 	if (strcmp(intf, ifname) != 0)
 		return -1;
+#endif
 	return fd;
 }
 
--- origsrc/libHX-4.0.1/src/tc-shconfig.c	2021-10-03 08:05:06.000000000 +0900
+++ src/libHX-4.0.1/src/tc-shconfig.c	2021-10-10 21:43:47.532285400 +0900
@@ -56,8 +56,8 @@ int main(int argc, const char **argv)
 		fprintf(stderr, "HX_init: %s\n", strerror(-ret));
 		return EXIT_FAILURE;
 	}
-	t_shconfig((argc >= 2) ? argv[1] : "tc-shconf.c");
-	t_shconfig2((argc >= 2) ? argv[1] : "tc-shconf.c");
+	t_shconfig((argc >= 2) ? argv[1] : "tc-shconfig.c");
+	t_shconfig2((argc >= 2) ? argv[1] : "tc-shconfig.c");
 	HX_exit();
 	return EXIT_SUCCESS;
 }
