--- origsrc/libHX-3.21/src/tc-shconfig.c	2014-06-02 10:17:37.000000000 +0900
+++ src/libHX-3.21/src/tc-shconfig.c	2014-06-27 20:51:49.143967300 +0900
@@ -56,8 +56,8 @@ int main(int argc, const char **argv)
 		fprintf(stderr, "HX_init: %s\n", strerror(-ret));
 		return EXIT_FAILURE;
 	}
-	t_shconfig((argc >= 2) ? argv[1] : "tc-shconf.c");
-	t_shconfig2((argc >= 2) ? argv[1] : "tc-shconf.c");
+	t_shconfig((argc >= 2) ? argv[1] : "tc-shconfig.c");
+	t_shconfig2((argc >= 2) ? argv[1] : "tc-shconfig.c");
 	HX_exit();
 	return EXIT_SUCCESS;
 }
