--- origsrc/libHX-3.10.1/src/Makefile.am	2011-04-07 01:57:07.000000000 +0900
+++ src/libHX-3.10.1/src/Makefile.am	2011-08-15 09:08:52.996865300 +0900
@@ -9,7 +9,7 @@ libHX_la_SOURCES = deque.c dl.c format.c
                    mc.c misc.c opt.c \
                    rand.c string.c time.c ${mingw32_extra} ${proc_sources}
 libHX_la_LIBADD  = ${libdl_LIBS} ${libpthread_LIBS} ${librt_LIBS}
-libHX_la_LDFLAGS = -version-info 27:0:0 \
+libHX_la_LDFLAGS = -version-info 27:0:0 -no-undefined \
                    -Wl,--version-script=${srcdir}/libHX.map
 
 if MINGW32
