diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libebur128-1.0.3/CMakeLists.txt src/libebur128-1.0.3/CMakeLists.txt
--- origsrc/libebur128-1.0.3/CMakeLists.txt	2015-09-08 02:04:22.000000000 +0900
+++ src/libebur128-1.0.3/CMakeLists.txt	2015-10-04 23:42:46.219712200 +0900
@@ -40,3 +40,11 @@ if(ENABLE_TESTS)
 else()
   message(STATUS "not building tests, set ENABLE_TESTS to ON to enable")
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "1.0.3")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/${PROJECT_NAME}.pc.in ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libebur128-1.0.3/ebur128/CMakeLists.txt src/libebur128-1.0.3/ebur128/CMakeLists.txt
--- origsrc/libebur128-1.0.3/ebur128/CMakeLists.txt	2015-09-08 02:04:22.000000000 +0900
+++ src/libebur128-1.0.3/ebur128/CMakeLists.txt	2015-10-04 23:35:41.520739400 +0900
@@ -71,5 +71,5 @@ install(FILES ebur128.h DESTINATION ${CM
 if(BUILD_STATIC_LIBS)
   install(TARGETS ebur128 ebur128_static DESTINATION ${CMAKE_INSTALL_LIBDIR})
 else()
-  install(TARGETS ebur128 DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  install(TARGETS ebur128 RUNTIME DESTINATION bin ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libebur128-1.0.3/libebur128.pc.in src/libebur128-1.0.3/libebur128.pc.in
--- origsrc/libebur128-1.0.3/libebur128.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libebur128-1.0.3/libebur128.pc.in	2015-10-04 23:42:53.326180400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libebur128
+Description: Library implementing the EBU R128 loudness standard
+Version: @VERSION@
+Libs: -L${libdir} -lebur128
+Cflags: -I${includedir}
