--- origsrc/belcard-4.3.1/CMakeLists.txt	2019-12-11 19:39:40.000000000 +0900
+++ src/belcard-4.3.1/CMakeLists.txt	2020-02-28 09:46:11.041674500 +0900
@@ -148,3 +148,6 @@ install(FILES
 if (ENABLE_PACKAGE_SOURCE)
 	add_subdirectory(build)
 endif()
+
+configure_file(${CMAKE_SOURCE_DIR}/belcard.pc.in ${CMAKE_BINARY_DIR}/belcard.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/belcard.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/belcard-4.3.1/src/CMakeLists.txt	2019-12-11 19:39:40.000000000 +0900
+++ src/belcard-4.3.1/src/CMakeLists.txt	2020-02-28 09:46:27.549185500 +0900
@@ -70,7 +70,7 @@ if(ENABLE_SHARED)
 			PUBLIC_HEADER "${BELCARD_HEADER_FILES}"
 		)
 	endif()
-	set_target_properties(belcard PROPERTIES VERSION ${BELCARD_SO_VERSION})
+	set_target_properties(belcard PROPERTIES SOVERSION ${BELCARD_SO_VERSION})
 	target_include_directories(belcard INTERFACE
 	  $<INSTALL_INTERFACE:include>
 	)
