--- origsrc/tagdust-2.33/src/nuc_code.c	2025-12-23 15:46:53.006935900 +0900
+++ src/tagdust-2.33/src/nuc_code.c	2025-12-23 15:47:11.742110700 +0900
@@ -39,6 +39,10 @@
 #include "nuc_code.h"
 
 
+unsigned int nuc_code[256];
+unsigned int rev_nuc_code[5];
+
+
 /** \fn void init_nuc_code()
  \brief Initializes nucleotide conversion arrays. 
  
--- origsrc/tagdust-2.33/src/nuc_code.h	2025-12-23 15:46:55.275700800 +0900
+++ src/tagdust-2.33/src/nuc_code.h	2025-12-23 15:47:11.742110700 +0900
@@ -27,20 +27,21 @@
  \bug No known bugs.
  */
 
+#ifndef TAGDUST2_NUC_CODE_H
+#define TAGDUST2_NUC_CODE_H
+
 /**
  * @brief Converts nucleotides in ASCII to 0-5. 
- *
- *
  */
-unsigned int nuc_code[256];
+extern unsigned int nuc_code[256];
 
 /**
  * @brief Converts 0-5 nucleotides into printable ASCII to.
- *
- *
  */
-unsigned int rev_nuc_code[5];
+extern unsigned int rev_nuc_code[5];
 
 int init_nuc_code(void);
 
+#endif
+
 
