--- origsrc/v7-1.0/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/v7-1.0/Makefile.am	2015-05-14 17:27:41.475959200 +0900
@@ -0,0 +1,15 @@
+
+TESTS = unit_test
+check_PROGRAMS = unit_test
+unit_test_CPPFLAGS = -DV7_EXPOSE_PRIVATE -DV7_UNIT_TEST
+unit_test_SOURCES = tests/unit_test.c
+unit_test_LDADD = libv7.la
+
+pkginclude_HEADERS = v7.h
+
+lib_LTLIBRARIES = libv7.la
+libv7_la_SOURCES = v7.c
+libv7_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/v7-1.0/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/v7-1.0/configure.ac	2015-05-14 17:25:30.193051300 +0900
@@ -0,0 +1,35 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([v7], [1.0])
+AC_CONFIG_SRCDIR([v7.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([fcntl.h float.h inttypes.h limits.h locale.h stddef.h stdint.h stdlib.h string.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_INT32_T
+AC_TYPE_INT64_T
+AC_TYPE_SIZE_T
+AC_TYPE_UINT16_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+AC_TYPE_UINT8_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_FUNC_STRTOD
+AC_CHECK_FUNCS([floor memmove memset setlocale strchr strerror strstr strtol strtoul tzset])
+
+AC_CONFIG_FILES([Makefile v7.pc])
+AC_OUTPUT
--- origsrc/v7-1.0/v7.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/v7-1.0/v7.pc.in	2015-05-14 17:25:27.944024700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: v7
+Description: Embedded JavaScript engine for C/C++
+Version: @VERSION@
+Libs: -L${libdir} -lv7
+Cflags: -I${includedir}/v7
