--- origsrc/v7-2.0/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/v7-2.0/Makefile.am	2016-03-13 09:08:14.193727600 +0900
@@ -0,0 +1,14 @@
+
+check_PROGRAMS = unit_test
+unit_test_CPPFLAGS = -DV7_EXPOSE_PRIVATE -DV7_UNIT_TEST
+unit_test_SOURCES = tests/unit_test.c
+TESTS = $(check_PROGRAMS)
+
+pkginclude_HEADERS = v7.h
+
+lib_LTLIBRARIES = libv7.la
+libv7_la_SOURCES = v7.c
+libv7_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/v7-2.0/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/v7-2.0/configure.ac	2016-03-13 09:00:48.407983300 +0900
@@ -0,0 +1,35 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([v7], [2.0])
+AC_CONFIG_SRCDIR([v7.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([fcntl.h float.h inttypes.h limits.h locale.h stddef.h stdint.h stdlib.h string.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_INT32_T
+AC_TYPE_INT64_T
+AC_TYPE_SIZE_T
+AC_TYPE_UINT16_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+AC_TYPE_UINT8_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_FUNC_STRTOD
+AC_CHECK_FUNCS([floor memmove memset setlocale strchr strerror strstr strtol strtoul tzset])
+
+AC_CONFIG_FILES([Makefile v7.pc])
+AC_OUTPUT
--- origsrc/v7-2.0/v7.c	2016-03-05 01:16:39.000000000 +0900
+++ src/v7-2.0/v7.c	2016-03-13 09:06:20.195304600 +0900
@@ -115,6 +115,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <sys/stat.h>
+#include <sys/time.h>
 
 #define random() rand()
 #ifdef _MSC_VER
@@ -279,7 +280,9 @@ typedef struct stat cs_stat_t;
 #define to64(x) strtoll(x, NULL, 10)
 #define INT64_FMT PRId64
 #define INT64_X_FMT PRIx64
+#if !defined(__cdecl)
 #define __cdecl
+#endif
 
 #ifndef va_copy
 #ifdef __va_copy
@@ -5657,7 +5660,7 @@ static Rune __space2[] = {
     0x0020, 0x0020, /* space */
     0x00a0, 0x00a0, /*   */
     0x2000, 0x200b, /*   - ​ */
-    0x2028, 0x2029, /*   -   */
+    0x2028, 0x2029, /*  -  */
     0x3000, 0x3000, /* 　 */
     0xfeff, 0xfeff, /* ﻿ */
 };
--- origsrc/v7-2.0/v7.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/v7-2.0/v7.pc.in	2016-03-13 09:00:48.423297300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: v7
+Description: Embedded JavaScript engine for C/C++
+Version: @VERSION@
+Libs: -L${libdir} -lv7
+Cflags: -I${includedir}/v7
