--- origsrc/Log4Qt-1.5.0/CMakeLists.txt	2018-06-26 03:30:27.000000000 +0900
+++ src/Log4Qt-1.5.0/CMakeLists.txt	2019-10-04 13:38:50.840095500 +0900
@@ -163,3 +163,10 @@ add_custom_target(uninstall
   "${CMAKE_COMMAND}" -P
   "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
 
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/log4qt.pc.in ${CMAKE_BINARY_DIR}/log4qt.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/log4qt.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/Log4Qt-1.5.0/log4qt.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/Log4Qt-1.5.0/log4qt.pc.in	2019-10-04 13:38:50.840095500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: log4qt
+Description: Logging for the Qt cross-platform application framework
+Version: @Log4Qt_VERSION@
+Libs: -L${libdir} -llog4qt
+Cflags: -I${includedir}/log4qt
