--- origsrc/Log4Qt-1.5.1/CMakeLists.txt	2020-02-15 05:49:06.000000000 +0900
+++ src/Log4Qt-1.5.1/CMakeLists.txt	2020-11-26 17:57:34.562417100 +0900
@@ -287,3 +287,10 @@ endif()
 
 # This must always be last!
 include( CPack )
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/log4qt.pc.in ${CMAKE_BINARY_DIR}/log4qt.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/log4qt.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/Log4Qt-1.5.1/log4qt.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/Log4Qt-1.5.1/log4qt.pc.in	2020-11-26 17:57:26.084138300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: log4qt
+Description: Logging for the Qt cross-platform application framework
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -llog4qt
+Cflags: -I${includedir}
