--- origsrc/libexword-rel1.0.1/configure.ac	2011-05-31 05:49:04.000000000 +0900
+++ src/libexword-rel1.0.1/configure.ac	2014-12-12 19:56:47.882085900 +0900
@@ -9,6 +9,7 @@ AC_CONFIG_HEADERS([config.h])
 
 # Checks for programs.
 #AC_PROG_CC
+AM_PROG_AR
 AC_PROG_LIBTOOL
 AM_PROG_CC_C_O
 
@@ -16,6 +17,9 @@ AC_PATH_PROGS([DOX_PROG], [doxygen], mis
 AM_CONDITIONAL([DOX_ENABLED], [test "x$DOX_PROG" != "xmissing"])
 
 case $host in
+  *-*-cygwin*)
+	EXTRA_LDFLAGS="-no-undefined"
+    ;;
   *-*-mingw32*)
 	EXTRA_LIBS="-lws2_32 -lshlwapi"
 	EXTRA_LDFLAGS="-no-undefined"
--- origsrc/libexword-rel1.0.1/docs/Makefile.am	2011-05-31 05:49:04.000000000 +0900
+++ src/libexword-rel1.0.1/docs/Makefile.am	2014-12-12 19:59:49.310796900 +0900
@@ -5,7 +5,7 @@ endif
 
 Doxygen:
 	@(echo "Creating Documentation"; \
-	  $(DOX_PROG) doxy.cfg)
+	  $(DOX_PROG) $(srcdir)/doxy.cfg)
 
 clean-local:
 	-rm -rf html
