--- origsrc/libnscd-2.0.2/src/Makefile.am	2006-11-30 01:30:03.000000000 +0900
+++ src/libnscd-2.0.2/src/Makefile.am	2012-03-24 22:12:20.702360800 +0900
@@ -12,8 +12,8 @@ XMLS = nscd_flush_cache.3.xml
 
 include_HEADERS = libnscd.h
 
-libnscd_la_LDFLAGS = -version-info 1:2:0 \
-	-Wl,--version-script=$(srcdir)/libnscd.map
+libnscd_la_LDFLAGS = -version-info 1:2:0 -no-undefined \
+	-Wl,--version-script=$(srcdir)/libnscd.map 
 
 lib_LTLIBRARIES = libnscd.la
 
--- origsrc/libnscd-2.0.2/src/libnscd.c	2006-11-30 01:27:17.000000000 +0900
+++ src/libnscd-2.0.2/src/libnscd.c	2012-03-24 22:12:20.702360800 +0900
@@ -95,6 +95,16 @@ nscd_open_socket (void)
   return sock;
 }
 
+#if !defined(TEMP_FAILURE_RETRY)
+#define TEMP_FAILURE_RETRY(expression) \
+    ({ \
+        long int _result; \
+        do _result = (long int) (expression); \
+        while (_result == -1L && errno == EINTR); \
+        _result; \
+    })
+#endif // !defined(TEMP_FAILURE_RETRY)
+
 int
 nscd_flush_cache (const char *service)
 {
