--- origsrc/libasyncns-0.8/configure.ac	2009-10-15 10:42:51.000000000 +0900
+++ src/libasyncns-0.8/configure.ac	2010-10-24 14:33:53.891703900 +0900
@@ -48,7 +48,7 @@ AC_USE_SYSTEM_EXTENSIONS
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 CC_CHECK_CFLAGS_APPEND([-pipe -Wall -W -Wextra -pedantic -Wformat -Wold-style-definition -Wdeclaration-after-statement -Wfloat-equal -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wredundant-decls -Wmissing-noreturn -Wshadow -Wendif-labels -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wwrite-strings -Winline -Wno-unused-parameter -Wstrict-aliasing])
 
--- origsrc/libasyncns-0.8/libasyncns/Makefile.am	2009-10-15 07:43:45.000000000 +0900
+++ src/libasyncns-0.8/libasyncns/Makefile.am	2010-10-24 14:57:38.392682100 +0900
@@ -21,7 +21,7 @@ AM_CFLAGS=-D__EXTENSIONS__ $(PTHREAD_CFL
 lib_LTLIBRARIES=libasyncns.la
 libasyncns_la_CC=$(PTHREAD_CC)
 libasyncns_la_SOURCES=asyncns.c asyncns.h
-libasyncns_la_LDFLAGS=-version-info $(LIBASYNCNS_VERSION_INFO) $(LDFLAGS_NOUNDEFINED)
+libasyncns_la_LDFLAGS=-version-info $(LIBASYNCNS_VERSION_INFO) $(LDFLAGS_NOUNDEFINED) -no-undefined
 libasyncns_la_LIBADD=$(PTHREAD_LIBS)
 
 check_PROGRAMS=asyncns-test
