--- origsrc/checkasm-v1.0.1/src/meson.build	2026-01-02 06:06:12.000000000 +0900
+++ src/checkasm-v1.0.1/src/meson.build	2026-01-22 15:15:06.735313000 +0900
@@ -226,6 +226,7 @@ checkasm_library = library(meson.project
   include_directories: checkasm_inc_dirs,
   dependencies: checkasm_dependencies,
   install: true,
+  soversion: meson.project_version().split('.')[0],
 )
 
 checkasm_dep = declare_dependency(
