--- origsrc/elk-3.99.8/configure.ac	2011-06-18 20:07:02.000000000 +0900
+++ src/elk-3.99.8/configure.ac	2024-10-08 13:24:45.187040100 +0900
@@ -55,6 +55,8 @@ case "${target_os}" in
     ;;
   *cygwin*)
     SYSTEM="cygwin"
+    ELK_CFLAGS="${ELK_CFLAGS} -DELK_BUILD_DLL"
+    ELK_LDFLAGS="${ELK_LDFLAGS} -no-undefined"
     ;;
   *darwin*)
     AC_DEFINE(SYS_DARWIN, 1, Define if the system is Darwin)
@@ -620,4 +622,3 @@ Xaw support: ${ac_cv_my_have_xaw}
 Motif support: ${ac_cv_my_have_motif}
 build documentation: ${ac_cv_my_have_groff}
 EOF
-
--- origsrc/elk-3.99.8/src/libelk.c	2011-06-18 19:24:06.000000000 +0900
+++ src/elk-3.99.8/src/libelk.c	2024-10-08 13:24:45.187040100 +0900
@@ -213,7 +213,7 @@ void Elk_Init (int ac, char **av, int in
 
     Argc = ac; Argv = av;
     First_Arg = 1;
-#ifdef CAN_DUMP
+#if defined(CAN_DUMP) && !defined(__CYGWIN__)
     if (Was_Dumped) {
         /* Check if beginning of stack has moved by a large amount.
          * This is the case, for instance, on a Sun-4m when the
