--- origsrc/ulfius-2.6.3/test/framework.c	2019-07-31 08:31:43.000000000 +0900
+++ src/ulfius-2.6.3/test/framework.c	2019-08-29 19:45:57.198557400 +0900
@@ -24,6 +24,10 @@
 #include <check.h>
 #include <ulfius.h>
 
+#if !defined(SO_REUSEPORT)
+#define SO_REUSEPORT (0)
+#endif
+
 #define CERT_KEY "-----BEGIN PRIVATE KEY-----\
 MIIEvwIBADANBgkqhkiG9w0BAQEFAASCBKkwggSlAgEAAoIBAQDr90HrswgEmln/\
 rXeNqYq0boIvas5wu27hmeHDdGGKtkCWIWGAo9GUy45xqsI4mDl3bOWS+pmb/3yi\
