--- origsrc/ulfius-2.6.5/CMakeLists.txt	2020-01-03 22:33:31.000000000 +0900
+++ src/ulfius-2.6.5/CMakeLists.txt	2020-01-05 01:25:31.867746000 +0900
@@ -254,7 +254,7 @@ if (WITH_YDER)
         set(Yder_FIND_QUIETLY ON) # force to find Yder quietly
         include(FindYder)
         find_package(Yder ${YDER_VERSION_REQUIRED} QUIET) # try to find Yder
-        if (NOT YDER_FOUND)
+        if (NOT PC_YDER_FOUND)
             include(DownloadProject)
             download_project(PROJ yder # ... otherwise, download archive
                               URL "https://github.com/babelouest/yder/archive/v${YDER_VERSION_REQUIRED}.tar.gz"
--- origsrc/ulfius-2.6.5/test/framework.c	2020-01-03 22:33:31.000000000 +0900
+++ src/ulfius-2.6.5/test/framework.c	2020-01-05 01:14:09.177750900 +0900
@@ -24,6 +24,10 @@
 #include <check.h>
 #include <ulfius.h>
 
+#if !defined(SO_REUSEPORT)
+#define SO_REUSEPORT (0)
+#endif
+
 #define CERT_KEY "-----BEGIN PRIVATE KEY-----\
 MIIEvwIBADANBgkqhkiG9w0BAQEFAASCBKkwggSlAgEAAoIBAQDr90HrswgEmln/\
 rXeNqYq0boIvas5wu27hmeHDdGGKtkCWIWGAo9GUy45xqsI4mDl3bOWS+pmb/3yi\
