--- origsrc/librouteros-1.1.2/src/Makefile.am	2009-12-25 19:20:05.000000000 +0900
+++ src/librouteros-1.1.2/src/Makefile.am	2015-01-05 22:29:28.680061800 +0900
@@ -1,7 +1,7 @@
 AUTOMAKE_OPTIONS = foreign no-dependencies
 
 if COMPILER_IS_GCC
-AM_CFLAGS = -Wall -Werror
+AM_CFLAGS = -Wall
 if BUILD_WITH_DEBUG
 AM_CFLAGS += -Wextra -g -O0
 endif
@@ -11,7 +11,7 @@ include_HEADERS = routeros_api.h routero
 lib_LTLIBRARIES = librouteros.la
 
 librouteros_la_LDFLAGS = -version-info @LIBROUTEROS_CURRENT@:@LIBROUTEROS_REVISION@:@LIBROUTEROS_AGE@
-librouteros_la_LDFLAGS += -export-symbols-regex "^ros_"
+librouteros_la_LDFLAGS += -no-undefined -export-symbols-regex "^ros_"
 librouteros_la_LIBADD = -lgcrypt
 if BUILD_WITH_LIBSOCKET
 librouteros_la_LIBADD += -lsocket
