--- origsrc/ColPack-1.0.9/BipartiteGraphPartialColoring/BipartiteGraphPartialColoring.cpp	2012-03-21 13:05:35.000000000 +0900
+++ src/ColPack-1.0.9/BipartiteGraphPartialColoring/BipartiteGraphPartialColoring.cpp	2014-08-15 08:28:04.154571800 +0900
@@ -389,7 +389,6 @@ namespace ColPack
 							//Algo 4 - Line 18: add [v] to R ; cont <- false; break
 #pragma omp critical
 							vi_verticesNeedNewColor.push_back(v);
-#pragma omp end critical
 							cont = false;
 							break;
 						}
@@ -576,7 +575,6 @@ namespace ColPack
 							//Algo 4 - Line 18: add [v] to R ; cont <- false; break
 #pragma omp critical
 							vi_verticesNeedNewColor.push_back(v);
-#pragma omp end critical
 							cont = false;
 							break;
 						}
--- origsrc/ColPack-1.0.9/Makefile.am	2012-03-21 13:37:34.000000000 +0900
+++ src/ColPack-1.0.9/Makefile.am	2014-08-15 08:19:47.472893300 +0900
@@ -17,8 +17,7 @@ endif
 LDADD = libColPack.la
 AM_DEFAULT_SOURCE_EXT = .cpp
 
-noinst_bin_PROGRAMS = ColPack
-noinst_bindir = $(abs_top_builddir)/progs
+bin_PROGRAMS = ColPack
 lib_LTLIBRARIES = libColPack.la
 pkginclude_HEADERS = \
 			Utilities/CoutLock.h Utilities/command_line_parameter_processor.h  \
@@ -72,12 +71,17 @@ libColPack_la_SOURCES = \
 			Recovery/JacobianRecovery2D.cpp \
 			Recovery/HessianRecovery.cpp 
 
+libColPack_la_LDFLAGS = -no-undefined
+if ENABLE_OPENMP
+libColPack_la_LIBADD = -lgomp
+endif
+
 ColPack_SOURCES = \
 			Main/Main.cpp 
 
 
 if EXAMPLES
-examplesdir = ${prefix}/examples
+examplesdir = ${datadir}/ColPack/examples
 basic_examplesdir = ${examplesdir}/Basic
 basic_examples_PROGRAMS = \
 			SampleDrivers/Basic/color_bipartite_graph_using_BipartiteGraphBicoloringInterface \
