--- origsrc/librm-2.2.3/plugins/meson.build	2021-05-01 05:01:43.000000000 +0900
+++ src/librm-2.2.3/plugins/meson.build	2021-07-08 13:47:53.775606000 +0900
@@ -1,7 +1,9 @@
 schema_dir = join_paths(get_option('datadir'), 'glib-2.0', 'schemas')
 
 subdir('areacodes_global')
+if host_machine.system().contains('cygwin') == false
 subdir('capi')
+endif
 subdir('fritzbox')
 subdir('fritzfon')
 
--- origsrc/librm-2.2.3/rm/rmplugins.c	2021-05-01 05:01:43.000000000 +0900
+++ src/librm-2.2.3/rm/rmplugins.c	2021-07-08 13:47:53.787651900 +0900
@@ -78,7 +78,11 @@ static gint rm_plugins_load_plugin(char
 	suffix = G_MODULE_SUFFIX;
 #endif
 
+#if defined(__CYGWIN__)
+	lib_name = g_strdup_printf("%s/cyg%s.%s", g_path_get_dirname(name), module_name, suffix);
+#else
 	lib_name = g_strdup_printf("%s/lib%s.%s", g_path_get_dirname(name), module_name, suffix);
+#endif
 	//g_debug("%s(): lib_name: %s", __FUNCTION__, lib_name);
 	module = g_module_open(lib_name, G_MODULE_BIND_LAZY);
 	if (!module) {
