--- origsrc/hfst-3.16.2/libhfst/hfst_c.pc.in	2025-01-18 11:46:40.000000000 +0900
+++ src/hfst-3.16.2/libhfst/hfst_c.pc.in	2025-03-12 23:20:38.051767800 +0900
@@ -5,6 +5,6 @@ includedir=@includedir@
 
 Name: hfst_c
 Description: C bindings for HFST library
-Version: 0.0.0
+Version: @LIBHFST_VERSION@
 Libs: -L${libdir} -l@LIBHFST_NAME@ -lhfst_c
 Cflags: -I${includedir}/hfst/
--- origsrc/hfst-3.16.2/libhfst/src/Makefile.am	2025-01-18 11:46:40.000000000 +0900
+++ src/hfst-3.16.2/libhfst/src/Makefile.am	2025-03-12 13:25:07.231545400 +0900
@@ -14,7 +14,7 @@ AUTOMAKE_OPTIONS=subdir-objects
 lib_LTLIBRARIES = libhfst.la
 
 AM_CXXFLAGS=-Wno-deprecated -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare -g
-AM_CPPFLAGS = -I${top_srcdir}/libhfst/src
+AM_CPPFLAGS = -I$(builddir)/implementations -I${top_srcdir}/libhfst/src
 
 # HFST bridge specific stuff
 HFST_SRCS=HfstApply.cc HfstInputStream.cc HfstTransducer.cc HfstOutputStream.cc\
--- origsrc/hfst-3.16.2/libhfst/src/implementations/Makefile.am	2025-01-18 11:46:40.000000000 +0900
+++ src/hfst-3.16.2/libhfst/src/implementations/Makefile.am	2025-03-12 12:42:20.835296000 +0900
@@ -27,7 +27,7 @@ IMPLEMENTATION_SRCS=ConvertTransducerFor
 
 AM_CXXFLAGS=-Wno-deprecated -g
 
-AM_CPPFLAGS = -I${top_srcdir}/libhfst/src -I${top_srcdir}/back-ends ${ICU_CPPFLAGS}
+AM_CPPFLAGS = -I$(top_srcdir) -I${top_srcdir}/libhfst/src -I${top_srcdir}/back-ends ${ICU_CPPFLAGS}
 
 # the bridges
 if WANT_SFST
--- origsrc/hfst-3.16.2/man/Makefile.am	2025-01-18 11:46:40.000000000 +0900
+++ src/hfst-3.16.2/man/Makefile.am	2025-03-12 15:52:45.759573900 +0900
@@ -58,8 +58,7 @@ endif
 
 aliases= hfst-determinise.1 hfst-expand.1 hfst-intersect.1 hfst-minimise.1 \
 	 hfst-minus.1 hfst-sfstpl2fst.1 hfst-summarise.1 hfst-union.1 hfst-proc.1 \
-	 hfst-optimised-lookup.1 \
-	 hfst-proc2.1 hfst-tokenise.1
+	 hfst-optimised-lookup.1
 
 dist_man1_MANS=$(hfst_tools) $(hfst_xfst) $(hfst_apertium_proc) \
 	$(hfst_tag_and_reweight) $(hfst_train_tagger) $(hfst_train_tagger_subprograms) $(hfst_twolc) \
--- origsrc/hfst-3.16.2/test/libhfst/Makefile.am	2025-01-18 11:46:40.000000000 +0900
+++ src/hfst-3.16.2/test/libhfst/Makefile.am	2025-03-12 16:35:42.286428200 +0900
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 LDADD = ../../libhfst/src/libhfst.la
-AM_CPPFLAGS = -I$(top_srcdir)/libhfst/src/ -I. -g
+AM_CPPFLAGS = -I$(top_builddir)/libhfst/src/implementations -I$(top_srcdir)/libhfst/src/ -I. -g
 
 AM_CPPFLAGS += -I${top_srcdir}/back-ends
 
--- origsrc/hfst-3.16.2/tools/src/Makefile.am	2025-01-18 11:46:40.000000000 +0900
+++ src/hfst-3.16.2/tools/src/Makefile.am	2025-03-12 14:03:50.621124600 +0900
@@ -18,7 +18,7 @@
 SUBDIRS=hfst-proc hfst-twolc hfst-tagger parsers
 AUTOMAKE_OPTIONS=std-options subdir-objects
 LDADD = $(top_builddir)/libhfst/src/libhfst.la $(ICU_LIBS)
-AM_CPPFLAGS = -I${top_srcdir}/libhfst/src -I${top_srcdir}/libhfst/src/parsers -I${top_srcdir}/tools/src/parsers -Wno-sign-compare ${ICU_CPPFLAGS}
+AM_CPPFLAGS = -I$(top_builddir)/libhfst/src/implementations -I${top_srcdir}/libhfst/src -I${top_srcdir}/libhfst/src/parsers -I${top_srcdir}/tools/src/parsers -Wno-sign-compare ${ICU_CPPFLAGS}
 AM_CXXFLAGS = -Wno-deprecated -Wno-sign-compare
 
 # sort alphabetically:
--- origsrc/hfst-3.16.2/tools/src/hfst-tagger/src/Makefile.am	2025-03-12 12:34:17.330364200 +0900
+++ src/hfst-3.16.2/tools/src/hfst-tagger/src/Makefile.am	2025-03-12 13:57:28.427468300 +0900
@@ -14,7 +14,7 @@ EXTRA_DIST=hfst-train-tagger.sh hfst-tra
 
 BUILD_MODEL_SRC=$(srcdir)/build_model_src
 USE_MODEL_SRC=$(srcdir)/use_model_src
-AM_CPPFLAGS=-I${top_srcdir}/libhfst/src -Wno-sign-compare -Wno-unused-parameter
+AM_CPPFLAGS = -I$(top_srcdir)/tools/src -I$(top_builddir)/libhfst/src/implementations -I${top_srcdir}/libhfst/src -Wno-sign-compare -Wno-unused-parameter
 LDADD = $(top_builddir)/libhfst/src/libhfst.la
 
 if WANT_TRAIN_TAGGER
--- origsrc/hfst-3.16.2/tools/src/hfst-tagger/src/build_model_src/tagger_aux.py	2025-01-18 11:46:40.000000000 +0900
+++ src/hfst-3.16.2/tools/src/hfst-tagger/src/build_model_src/tagger_aux.py	2025-03-12 16:48:41.458729200 +0900
@@ -212,7 +212,7 @@ def print_conditional_penalties(pair_cou
         for count in object_counter.itervalues():
             max_object_count = max(max_object_count, count)
 
-        print "PENALTY_WEIGHT=" + str(get_penalty(1, max_object_count + 1))
+        print("PENALTY_WEIGHT=" + str(get_penalty(1, max_object_count + 1)))
 
     for pair, penalty in penalty_map.iteritems():
         if type(pair[0]) == type(u"") or type(pair[0]) == type(""):
@@ -234,13 +234,13 @@ def print_conditional_penalties(pair_cou
                 s = string.join([p[0] + appended_suffix,
                                  p[1],
                                  str(penalty)],"\t")
-                print s.encode("utf-8")
+                print(s.encode("utf-8"))
             else:
                 
                 s = string.join([p[1] + appended_suffix,
                                  p[0],
                                  str(penalty)],"\t")
-                print s.encode("utf-8")
+                print(s.encode("utf-8"))
         else:
             for entry in pair[0]:
                 e = list(entry)
@@ -256,9 +256,9 @@ def print_conditional_penalties(pair_cou
                     pass
                 
                 s = e[0] + "\t" + e[1] + "\t"
-                print s.encode("utf-8"),
+                print(s.encode("utf-8")),
 
-            print penalty
+            print(penalty)
 
 def print_penalties(object_counter, total_count, appended_suffix):
 
@@ -274,4 +274,4 @@ def print_penalties(object_counter, tota
 
         s = o + appended_suffix + "T\t" + str(penalty)
 
-        print s.encode("utf-8")
+        print(s.encode("utf-8"))
--- origsrc/hfst-3.16.2/tools/src/hfst-twolc/src/Makefile.am	2025-01-18 11:46:40.000000000 +0900
+++ src/hfst-3.16.2/tools/src/hfst-twolc/src/Makefile.am	2025-03-12 13:54:01.427238600 +0900
@@ -8,7 +8,7 @@ else
   bin_SCRIPTS=
 endif
 
-AM_CPPFLAGS = -I${top_srcdir}/libhfst/src -I${top_srcdir}/libhfst/src/parsers
+AM_CPPFLAGS = -I$(top_builddir)/libhfst/src/implementations -I${top_srcdir}/libhfst/src -I${top_srcdir}/libhfst/src/parsers
 LDADD = $(top_builddir)/libhfst/src/libhfst.la $(ICU_LIBS)
 
 # the preprocessor scripts do not support options per se, so we whitelist
--- origsrc/hfst-3.16.2/tools/src/parsers/test/test.sh	2025-01-18 11:46:40.000000000 +0900
+++ src/hfst-3.16.2/tools/src/parsers/test/test.sh	2025-03-12 15:54:12.557715300 +0900
@@ -263,15 +263,15 @@ do
     do
     for testfile in test_overlap test_sublanguage # the function to be tested
     do
-        if ! (${LS} $testfile$testcase.xfst > /dev/null 2> /dev/null); then
+        if ! (${LS} ${srcdir}/$testfile$testcase.xfst > /dev/null 2> /dev/null); then
         echo "skipping missing test for "$testfile$testcase"..."
         continue
         fi
-        if ! (${CAT} $testfile$testcase.xfst | ../hfst-xfst --pipe-mode=input -s -f $format | ${TR} -d '\r' > tmp); then
+        if ! (${CAT} ${srcdir}/$testfile$testcase.xfst | ../hfst-xfst --pipe-mode=input -s -f $format | ${TR} -d '\r' > tmp); then
         echo "ERROR: in compiling "$testfile$testcase.xfst
         exit 1;
         fi
-        if ! ($DIFF tmp "test"$testcase.output); then
+        if ! ($DIFF tmp ${srcdir}/"test"$testcase.output); then
         echo "ERROR: in testing "$testfile$testcase.xfst
         exit 1;
         fi
@@ -280,7 +280,7 @@ do
 
     for file in quit-on-fail.xfst assert.xfst
     do
-        if (${CAT} $file | ../hfst-xfst --pipe-mode=input -s -f $format > tmp 2> /dev/null); then
+        if (${CAT} ${srcdir}/$file | ../hfst-xfst --pipe-mode=input -s -f $format > tmp 2> /dev/null); then
             echo "ERROR: in compiling "$file
             exit 1;
         fi
