--- origsrc/hfst-3.15.5/back-ends/foma/Makefile.am	2021-05-26 20:35:18.000000000 +0900
+++ src/hfst-3.15.5/back-ends/foma/Makefile.am	2021-06-29 09:38:29.460028700 +0900
@@ -13,7 +13,7 @@ noinst_LTLIBRARIES = libfoma.la
 endif
 
 lex.yy.c: regex.l regex.h
-	$(LEX) -8 regex.l
+	$(LEX) -8 $(srcdir)/regex.l
 
 lex.lexc.c: lexc.l
 	$(LEX) -8 --prefix=lexc $<
--- origsrc/hfst-3.15.5/libhfst/src/Makefile.am	2021-05-26 20:35:18.000000000 +0900
+++ src/hfst-3.15.5/libhfst/src/Makefile.am	2021-06-29 09:38:29.464048400 +0900
@@ -14,7 +14,7 @@ AUTOMAKE_OPTIONS=subdir-objects
 lib_LTLIBRARIES = libhfst.la
 
 AM_CXXFLAGS=-Wno-deprecated -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare -g
-AM_CPPFLAGS = -I${top_srcdir}/libhfst/src
+AM_CPPFLAGS = -I${top_srcdir}/libhfst/src -I$(builddir)/implementations
 
 # HFST bridge specific stuff
 HFST_SRCS=HfstApply.cc HfstInputStream.cc HfstTransducer.cc HfstOutputStream.cc\
--- origsrc/hfst-3.15.5/libhfst/src/implementations/Makefile.am	2021-05-26 20:35:18.000000000 +0900
+++ src/hfst-3.15.5/libhfst/src/implementations/Makefile.am	2021-06-29 09:38:29.468056900 +0900
@@ -28,7 +28,8 @@ IMPLEMENTATION_SRCS=ConvertTransducerFor
 AM_CXXFLAGS=-Wno-deprecated -g
 
 AM_CPPFLAGS = -I${top_srcdir}/libhfst/src -I${top_srcdir}/back-ends/foma \
-		-I${top_srcdir}/back-ends ${ICU_CPPFLAGS}
+		-I${top_srcdir}/back-ends ${ICU_CPPFLAGS} -I$(top_srcdir) \
+		-I$(top_builddir)/back-ends/openfst/src/lib
 
 if WANT_MINGW
   AM_CPPFLAGS += -I${top_srcdir}/back-ends/openfstwin/src/include \
--- origsrc/hfst-3.15.5/man/Makefile.am	2021-05-26 20:35:18.000000000 +0900
+++ src/hfst-3.15.5/man/Makefile.am	2021-06-29 09:38:29.473050100 +0900
@@ -72,8 +72,8 @@ endif
 
 aliases= hfst-determinise.1 hfst-expand.1 hfst-intersect.1 hfst-minimise.1 \
 	 hfst-minus.1 hfst-sfstpl2fst.1 hfst-summarise.1 hfst-union.1 hfst-proc.1 \
-	 hfst-optimised-lookup.1 \
-	 hfst-proc2.1 hfst-tokenise.1
+	 hfst-optimised-lookup.1 
+#	 hfst-proc2.1 hfst-tokenise.1
 
 dist_man1_MANS=$(hfst_tools) $(hfst_xfst) $(hfst_apertium_proc) $(hfst_foma) $(hfst_foma_wrapper) \
 	$(hfst_tag_and_reweight) $(hfst_train_tagger) $(hfst_train_tagger_subprograms) $(hfst_twolc) \
--- origsrc/hfst-3.15.5/tools/src/Makefile.am	2021-05-26 20:35:18.000000000 +0900
+++ src/hfst-3.15.5/tools/src/Makefile.am	2021-06-29 09:38:29.477057200 +0900
@@ -18,7 +18,8 @@
 SUBDIRS=hfst-proc hfst-twolc hfst-tagger parsers
 AUTOMAKE_OPTIONS=std-options subdir-objects
 LDADD = $(top_builddir)/libhfst/src/libhfst.la $(ICU_LIBS)
-AM_CPPFLAGS = -I${top_srcdir}/libhfst/src -I${top_srcdir}/libhfst/src/parsers -I${top_srcdir}/tools/src/parsers -Wno-sign-compare ${GLIB_CPPFLAGS} ${ICU_CPPFLAGS}
+AM_CPPFLAGS = -I${top_srcdir}/libhfst/src -I${top_srcdir}/libhfst/src/parsers -I${top_srcdir}/tools/src/parsers -Wno-sign-compare ${GLIB_CPPFLAGS} ${ICU_CPPFLAGS} \
+	-I${top_srcdir}/tools/src/parsers -Wno-sign-compare -I$(top_builddir)/libhfst/src/implementations
 AM_CXXFLAGS = -Wno-deprecated -Wno-sign-compare
 
 # sort alphabetically:
--- origsrc/hfst-3.15.5/tools/src/hfst-tagger/src/Makefile.am	2021-05-26 20:35:18.000000000 +0900
+++ src/hfst-3.15.5/tools/src/hfst-tagger/src/Makefile.am	2021-06-29 10:52:06.455740200 +0900
@@ -14,7 +14,8 @@ EXTRA_DIST=hfst-train-tagger.sh hfst-tra
 
 BUILD_MODEL_SRC=$(srcdir)/build_model_src
 USE_MODEL_SRC=$(srcdir)/use_model_src
-AM_CPPFLAGS=-I${top_srcdir}/libhfst/src -Wno-sign-compare -Wno-unused-parameter
+AM_CPPFLAGS=-I${top_srcdir}/libhfst/src -Wno-sign-compare -Wno-unused-parameter \
+	 -I$(top_srcdir)/tools/src -I$(top_builddir)/libhfst/src/implementations
 LDADD = $(top_builddir)/libhfst/src/libhfst.la
 
 if WANT_TRAIN_TAGGER
@@ -90,52 +91,52 @@ check_PROGRAMS=FstBuilder LexicalModelBu
 
 FstBuilder_SOURCES=$(BUILD_MODEL_SRC)/FstBuilder.cc
 FstBuilder_LDADD=$(BUILD_MODEL_SRC)/string_handling.o $(BUILD_MODEL_SRC)/FstBuilder.o \
-                 $(top_srcdir)/libhfst/src/libhfst.la
+                 $(top_builddir)/libhfst/src/libhfst.la
 FstBuilder_CXXFLAGS=-DMAIN_TEST -Wno-deprecated
 
 LexicalModelBuilder_SOURCES=$(BUILD_MODEL_SRC)/LexicalModelBuilder.cc
 LexicalModelBuilder_LDADD=$(BUILD_MODEL_SRC)/string_handling.o $(BUILD_MODEL_SRC)/FstBuilder.o $(BUILD_MODEL_SRC)/ModelBuilder.o \
 			  $(BUILD_MODEL_SRC)/WeightedStringVectorCollection.o \
 	                  $(BUILD_MODEL_SRC)/LexicalModelBuilder.o \
-                          $(top_srcdir)/libhfst/src/libhfst.la
+                          $(top_builddir)/libhfst/src/libhfst.la
 LexicalModelBuilder_CXXFLAGS=-DMAIN_TEST -Wno-deprecated
 
 SequenceModelBuilder_SOURCES=$(BUILD_MODEL_SRC)/SequenceModelBuilder.cc
 SequenceModelBuilder_LDADD=$(BUILD_MODEL_SRC)/string_handling.o $(BUILD_MODEL_SRC)/FstBuilder.o $(BUILD_MODEL_SRC)/ModelBuilder.o \
 			  $(BUILD_MODEL_SRC)/WeightedStringVectorCollection.o \
 	                  $(BUILD_MODEL_SRC)/SequenceModelBuilder.o \
-                          $(top_srcdir)/libhfst/src/libhfst.la
+                          $(top_builddir)/libhfst/src/libhfst.la
 SequenceModelBuilder_CXXFLAGS=-DMAIN_TEST -Wno-deprecated
 
 ModelBuilder_SOURCES=$(BUILD_MODEL_SRC)/ModelBuilder.cc
 ModelBuilder_LDADD=$(BUILD_MODEL_SRC)/string_handling.o $(BUILD_MODEL_SRC)/FstBuilder.o $(BUILD_MODEL_SRC)/ModelBuilder.o \
 		   $(BUILD_MODEL_SRC)/WeightedStringVectorCollection.o \
-                   $(top_srcdir)/libhfst/src/libhfst.la
+                   $(top_builddir)/libhfst/src/libhfst.la
 ModelBuilder_CXXFLAGS=-DMAIN_TEST -Wno-deprecated
 
 WeightedStringVectorCollection_SOURCES=\
 	$(BUILD_MODEL_SRC)/WeightedStringVectorCollection.cc
 WeightedStringVectorCollection_LDADD=$(BUILD_MODEL_SRC)/string_handling.o \
 	$(BUILD_MODEL_SRC)/WeightedStringVectorCollection.o \
-	$(top_srcdir)/libhfst/src/libhfst.la
+	$(top_builddir)/libhfst/src/libhfst.la
 WeightedStringVectorCollection_CXXFLAGS=-DMAIN_TEST -Wno-deprecated
 
 TaggerBuilder_SOURCES=$(BUILD_MODEL_SRC)/TaggerBuilder.cc
 TaggerBuilder_LDADD=$(BUILD_MODEL_SRC)/string_handling.o $(BUILD_MODEL_SRC)/TaggerBuilder.o \
 	$(BUILD_MODEL_SRC)/LexicalModelBuilder.o $(BUILD_MODEL_SRC)/SequenceModelBuilder.o \
 	$(BUILD_MODEL_SRC)/WeightedStringVectorCollection.o $(BUILD_MODEL_SRC)/ModelBuilder.o $(BUILD_MODEL_SRC)/FstBuilder.o \
-	$(top_srcdir)/libhfst/src/libhfst.la
+	$(top_builddir)/libhfst/src/libhfst.la
 TaggerBuilder_CXXFLAGS=-DMAIN_TEST -Wno-deprecated
 
 string_handling_SOURCES=$(BUILD_MODEL_SRC)/string_handling.cc
 
-string_handling_LDADD=$(BUILD_MODEL_SRC)/string_handling.o $(top_srcdir)/libhfst/src/libhfst.la \
+string_handling_LDADD=$(BUILD_MODEL_SRC)/string_handling.o $(top_builddir)/libhfst/src/libhfst.la \
 		      unit_test_aux.o
 string_handling_CXXFLAGS=-DMAIN_TEST -Wno-deprecated
 
 AcyclicAutomaton_SOURCES=$(USE_MODEL_SRC)/AcyclicAutomaton.cc
 
-AcyclicAutomaton_LDADD=$(USE_MODEL_SRC)/AcyclicAutomaton.o $(top_srcdir)/libhfst/src/libhfst.la\
+AcyclicAutomaton_LDADD=$(USE_MODEL_SRC)/AcyclicAutomaton.o $(top_builddir)/libhfst/src/libhfst.la\
 		       $(USE_MODEL_SRC)/SequenceModelComponent.o unit_test_aux.o
 AcyclicAutomaton_CXXFLAGS=-DMAIN_TEST -Wno-deprecated
 
@@ -143,19 +144,19 @@ DelayedSequenceModelComponent_SOURCES= \
                        $(USE_MODEL_SRC)/DelayedSequenceModelComponent.cc
 
 DelayedSequenceModelComponent_LDADD=$(USE_MODEL_SRC)/DelayedSequenceModelComponent.o \
-                       $(top_srcdir)/libhfst/src/libhfst.la \
+                       $(top_builddir)/libhfst/src/libhfst.la \
 		       $(USE_MODEL_SRC)/SequenceModelComponent.o unit_test_aux.o
 DelayedSequenceModelComponent_CXXFLAGS=-DMAIN_TEST -Wno-deprecated
 
 NewLexicalModel_SOURCES=$(USE_MODEL_SRC)/NewLexicalModel.cc
 
-NewLexicalModel_LDADD=$(USE_MODEL_SRC)/NewLexicalModel.o $(top_srcdir)/libhfst/src/libhfst.la \
+NewLexicalModel_LDADD=$(USE_MODEL_SRC)/NewLexicalModel.o $(top_builddir)/libhfst/src/libhfst.la \
 		      unit_test_aux.o
 NewLexicalModel_CXXFLAGS=-DMAIN_TEST -Wno-deprecated
 
 SentenceTagger_SOURCES=$(USE_MODEL_SRC)/SentenceTagger.cc
 
-SentenceTagger_LDADD=$(USE_MODEL_SRC)/SentenceTagger.o $(top_srcdir)/libhfst/src/libhfst.la \
+SentenceTagger_LDADD=$(USE_MODEL_SRC)/SentenceTagger.o $(top_builddir)/libhfst/src/libhfst.la \
 		     $(USE_MODEL_SRC)/DelayedSequenceModelComponent.o $(USE_MODEL_SRC)/SequenceModelComponent.o \
 		     $(USE_MODEL_SRC)/SentenceTransducer.o $(USE_MODEL_SRC)/NewLexicalModel.o \
                      $(USE_MODEL_SRC)/SequenceModelComponentPair.o unit_test_aux.o \
@@ -166,28 +167,28 @@ SentenceTagger_CXXFLAGS=-DMAIN_TEST -Wno
 SentenceTransducer_SOURCES=$(USE_MODEL_SRC)/SentenceTransducer.cc
 
 SentenceTransducer_LDADD=$(USE_MODEL_SRC)/SentenceTransducer.o \
-                       $(top_srcdir)/libhfst/src/libhfst.la \
+                       $(top_builddir)/libhfst/src/libhfst.la \
 		       $(USE_MODEL_SRC)/SequenceModelComponent.o unit_test_aux.o
 SentenceTransducer_CXXFLAGS=-DMAIN_TEST -Wno-deprecated
 
 SequenceModelComponent_SOURCES=$(USE_MODEL_SRC)/SequenceModelComponent.cc\
 			       unit_test_aux.cc
 SequenceModelComponent_LDADD=$(USE_MODEL_SRC)/SequenceModelComponent.o \
-                             $(top_srcdir)/libhfst/src/libhfst.la
+                             $(top_builddir)/libhfst/src/libhfst.la
 SequenceModelComponent_CXXFLAGS=-DMAIN_TEST -Wno-deprecated
 
 SequenceModelComponentPair_SOURCES= \
                                 $(USE_MODEL_SRC)/SequenceModelComponentPair.cc
 
 SequenceModelComponentPair_LDADD=$(USE_MODEL_SRC)/SequenceModelComponentPair.o \
-                                 $(top_srcdir)/libhfst/src/libhfst.la \
+                                 $(top_builddir)/libhfst/src/libhfst.la \
                                  $(USE_MODEL_SRC)/DelayedSequenceModelComponent.o \
 				 $(USE_MODEL_SRC)/SequenceModelComponent.o unit_test_aux.o
 SequenceModelComponentPair_CXXFLAGS=-DMAIN_TEST -Wno-deprecated
 
 SequenceTagger_SOURCES=$(USE_MODEL_SRC)/SequenceTagger.cc
 
-SequenceTagger_LDADD=$(USE_MODEL_SRC)/SequenceTagger.o $(top_srcdir)/libhfst/src/libhfst.la\
+SequenceTagger_LDADD=$(USE_MODEL_SRC)/SequenceTagger.o $(top_builddir)/libhfst/src/libhfst.la\
 		     $(USE_MODEL_SRC)/SequenceModelComponent.o $(USE_MODEL_SRC)/SentenceTransducer.o \
                      unit_test_aux.o $(USE_MODEL_SRC)/AcyclicAutomaton.o \
                      $(USE_MODEL_SRC)/SequenceModelComponentPair.o \
--- origsrc/hfst-3.15.5/tools/src/hfst-twolc/src/Makefile.am	2021-05-26 20:35:18.000000000 +0900
+++ src/hfst-3.15.5/tools/src/hfst-twolc/src/Makefile.am	2021-06-29 09:38:29.484034200 +0900
@@ -8,7 +8,8 @@ else
   bin_SCRIPTS=
 endif
 
-AM_CPPFLAGS = -I${top_srcdir}/libhfst/src -I${top_srcdir}/libhfst/src/parsers
+AM_CPPFLAGS = -I${top_srcdir}/libhfst/src -I${top_srcdir}/libhfst/src/parsers \
+	-I$(top_builddir)/libhfst/src/implementations
 LDADD = $(top_builddir)/libhfst/src/libhfst.la $(ICU_LIBS)
 
 # the preprocessor scripts do not support options per se, so we whitelist
