--- origsrc/libfsext-20240501/Makefile.am	2024-05-01 13:11:32.000000000 +0900
+++ src/libfsext-20240501/Makefile.am	2025-05-18 20:17:55.673373400 +0900
@@ -3,25 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
-	libfguid \
 	libfsext \
-	libhmac \
 	fsexttools \
 	pyfsext \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libfsext-20240501/configure.ac	2024-05-01 13:11:32.000000000 +0900
+++ src/libfsext-20240501/configure.ac	2025-05-18 20:17:55.673373400 +0900
@@ -176,23 +176,8 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
 AC_CONFIG_FILES([libfsext/Makefile])
 AC_CONFIG_FILES([pyfsext/Makefile])
-AC_CONFIG_FILES([libhmac/Makefile])
 AC_CONFIG_FILES([fsexttools/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
--- origsrc/libfsext-20240501/pyfsext/Makefile.am	2024-05-01 13:11:32.000000000 +0900
+++ src/libfsext-20240501/pyfsext/Makefile.am	2025-05-18 20:17:55.673373400 +0900
@@ -43,7 +43,7 @@ pyfsext_la_LIBADD = \
 	@LIBBFIO_LIBADD@
 
 pyfsext_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyfsext_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyfsext_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
