--- origsrc/foma/foma/CMakeLists.txt	2024-12-23 22:05:32.000000000 +0900
+++ src/foma/foma/CMakeLists.txt	2024-12-23 23:02:00.069722400 +0900
@@ -120,10 +120,6 @@ set(SOURCES
 	${BISON_Bregex_OUTPUTS}
 	)
 
-add_library(foma-static STATIC ${SOURCES})
-target_link_libraries(foma-static PUBLIC ${ZLIB_LIBS})
-set_target_properties(foma-static PROPERTIES ARCHIVE_OUTPUT_NAME foma)
-
 add_library(foma-shared SHARED ${SOURCES})
 target_link_libraries(foma-shared PRIVATE ${ZLIB_LIBS})
 set_target_properties(foma-shared PROPERTIES
@@ -134,22 +130,22 @@ if(NOT MSVC)
 endif()
 
 add_executable(foma-bin foma.c stack.c iface.c ${FLEX_Finterface_OUTPUTS})
-target_link_libraries(foma-bin PRIVATE foma-static ${READLINE_LIBRARIES} ${GETOPT_LIB})
+target_link_libraries(foma-bin PRIVATE foma-shared ${READLINE_LIBRARIES} ${GETOPT_LIB} ${ZLIB_LIBS})
 set_target_properties(foma-bin PROPERTIES RUNTIME_OUTPUT_NAME foma)
 
 if(MSYS OR NOT WIN32)
 	add_executable(flookup flookup.c)
-	target_link_libraries(flookup PRIVATE foma-static ${GETOPT_LIB})
+	target_link_libraries(flookup PRIVATE foma-shared ${GETOPT_LIB})
 	install(TARGETS flookup RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 endif()
 
 add_executable(cgflookup cgflookup.c)
-target_link_libraries(cgflookup PRIVATE foma-static ${GETOPT_LIB})
+target_link_libraries(cgflookup PRIVATE foma-shared ${GETOPT_LIB})
 
 configure_file(libfoma.pc.in libfoma.pc @ONLY)
 
 # Install
-install(TARGETS foma-static foma-shared ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(TARGETS foma-shared ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 install(FILES fomalib.h fomalibconf.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 install(TARGETS foma-bin cgflookup RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libfoma.pc" DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig/")
