--- origsrc/gupnp-ui-0.1.1/configure.ac	2008-09-28 03:41:49.000000000 +0900
+++ src/gupnp-ui-0.1.1/configure.ac	2018-04-12 00:22:31.349000000 +0900
@@ -10,7 +10,7 @@ AC_PROG_CC
 AC_STDC_HEADERS
 AC_PROG_LIBTOOL
 
-PKG_CHECK_MODULES(LIBGUPNP_UI, gupnp-1.0 >= 0.3 gtk+-2.0)
+PKG_CHECK_MODULES(LIBGUPNP_UI, gssdp-1.0 gupnp-1.0 >= 0.3 gtk+-2.0)
 
 PKG_CHECK_MODULES(GTHREAD, gthread-2.0)
 
--- origsrc/gupnp-ui-0.1.1/libgupnp-ui/Makefile.am	2007-08-15 16:51:40.000000000 +0900
+++ src/gupnp-ui-0.1.1/libgupnp-ui/Makefile.am	2018-04-12 00:12:35.567968900 +0900
@@ -12,6 +12,7 @@ libgupnp_ui_1_0_la_SOURCES = gupnp-ui-de
 			     gupnp-ui-device-view.c
 
 libgupnp_ui_1_0_la_LIBADD = $(LIBGUPNP_UI_LIBS)
+libgupnp_ui_1_0_la_LDFLAGS = -no-undefined
 
 CLEANFILES = $(BUILT_SOURCES)
 DISTCLEANFILES = $(BUILT_SOURCES)
