diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/g3log-1.3.2/Build.cmake src/g3log-1.3.2/Build.cmake
--- origsrc/g3log-1.3.2/Build.cmake	2018-03-31 11:50:35.000000000 +0900
+++ src/g3log-1.3.2/Build.cmake	2018-05-10 19:21:14.122365000 +0900
@@ -55,6 +55,8 @@ ELSEIF(${CMAKE_CXX_COMPILER_ID} STREQUAL
 
        # deal with ERROR level conflicts with windows.h
        ADD_DEFINITIONS (-DNOGDI)
+   ELSEIF (CYGWIN)
+       set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -rdynamic -Wunused -pthread -lrt -D_GLIBCXX_USE_NANOSLEEP -D_GLIBCXX_USE_SCHED_YIELD")
    ELSE()
        set(PLATFORM_LINK_LIBRIES rt)
        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -rdynamic -Wunused -std=c++11 -pthread -lrt -D_GLIBCXX_USE_NANOSLEEP -D_GLIBCXX_USE_SCHED_YIELD")
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/g3log-1.3.2/CMakeLists.txt src/g3log-1.3.2/CMakeLists.txt
--- origsrc/g3log-1.3.2/CMakeLists.txt	2018-03-31 11:50:35.000000000 +0900
+++ src/g3log-1.3.2/CMakeLists.txt	2018-05-10 19:56:31.036165300 +0900
@@ -179,7 +179,7 @@ IF(${CMAKE_SYSTEM_NAME} MATCHES "Linux"
    INCLUDE (${g3log_SOURCE_DIR}/CPackLists.txt)
 ENDIF()
 
-IF(MINGW)
+IF(MINGW OR CYGWIN)
    # ==========================================================================
    #
    #   usage: make; make install
@@ -201,7 +201,9 @@ IF (NOT MSVC)
    )
 ENDIF()
 
-
-
-
-
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/g3log.pc.in ${CMAKE_BINARY_DIR}/g3log.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/g3log.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/g3log-1.3.2/CPackLists.txt src/g3log-1.3.2/CPackLists.txt
--- origsrc/g3log-1.3.2/CPackLists.txt	2018-03-31 11:50:35.000000000 +0900
+++ src/g3log-1.3.2/CPackLists.txt	2018-05-10 19:53:04.505074800 +0900
@@ -41,6 +41,7 @@ SET(CMAKE_INSTALL_HEADERDIR ${CMAKE_INST
 
 IF(NOT MINGW)
    INSTALL( TARGETS g3logger
+               RUNTIME DESTINATION ${CPACK_PACKAGING_INSTALL_PREFIX}/bin
                ARCHIVE DESTINATION ${CPACK_PACKAGING_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}
                LIBRARY DESTINATION ${CPACK_PACKAGING_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}
                COMPONENT libraries)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/g3log-1.3.2/g3log.pc.in src/g3log-1.3.2/g3log.pc.in
--- origsrc/g3log-1.3.2/g3log.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/g3log-1.3.2/g3log.pc.in	2018-05-10 19:56:40.850055200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: g3log
+Description: Asynchronous, 'crash safe', logger that is easy to use
+Version: @VERSION@
+Libs: -L${libdir} -lg3logger
+Cflags: -I${includedir}
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/g3log-1.3.2/src/crashhandler_unix.cpp src/g3log-1.3.2/src/crashhandler_unix.cpp
--- origsrc/g3log-1.3.2/src/crashhandler_unix.cpp	2018-03-31 11:50:35.000000000 +0900
+++ src/g3log-1.3.2/src/crashhandler_unix.cpp	2018-05-10 19:25:17.862598600 +0900
@@ -19,7 +19,9 @@
 #include <csignal>
 #include <cstring>
 #include <unistd.h>
+#if !defined(__CYGWIN__)
 #include <execinfo.h>
+#endif
 #include <cxxabi.h>
 #include <cstdlib>
 #include <sstream>
@@ -156,9 +158,12 @@ namespace g3 {
 
          const size_t max_dump_size = 50;
          void* dump[max_dump_size];
-         size_t size = backtrace(dump, max_dump_size);
-         char** messages = backtrace_symbols(dump, static_cast<int>(size)); // overwrite sigaction with caller's address
-
+         size_t size = 0;
+         char** messages = NULL;
+#if !defined(__CYGWIN__)
+         size = backtrace(dump, max_dump_size);
+         messages = backtrace_symbols(dump, static_cast<int>(size)); // overwrite sigaction with caller's address
+#endif
          // dump stack: skip first frame, since that is here
          std::ostringstream oss;
          for (size_t idx = 1; idx < size && messages != nullptr; ++idx) {
@@ -199,7 +204,7 @@ namespace g3 {
                oss << "\tstack dump [" << idx << "]  " << messages[idx] << std::endl;
             }
          } // END: for(size_t idx = 1; idx < size && messages != nullptr; ++idx)
-         free(messages);
+         if (messages) free(messages);
          return oss.str();
       }
 
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/g3log-1.3.2/test_unit/Test.cmake src/g3log-1.3.2/test_unit/Test.cmake
--- origsrc/g3log-1.3.2/test_unit/Test.cmake	2018-03-31 11:50:35.000000000 +0900
+++ src/g3log-1.3.2/test_unit/Test.cmake	2018-05-10 19:43:05.179013700 +0900
@@ -31,11 +31,17 @@
    IF (ADD_G3LOG_UNIT_TEST)
       set(DIR_UNIT_TEST ${g3log_SOURCE_DIR}/test_unit)
       message( STATUS "-DADD_G3LOG_UNIT_TEST=ON" )  
+      option(USE_SYSTEM_GTEST "use system gtest" ON)
+      if (USE_SYSTEM_GTEST)
+        set(GTEST_LIBRARIES gtest)
+      else()
       set(GTEST_DIR ${g3log_SOURCE_DIR}/3rdParty/gtest/gtest-1.7.0)
       set(GTEST_INCLUDE_DIRECTORIES ${GTEST_DIR}/include ${GTEST_DIR} ${GTEST_DIR}/src)
       include_directories(${GTEST_INCLUDE_DIRECTORIES})
       add_library(gtest_170_lib ${GTEST_DIR}/src/gtest-all.cc)
       set_target_properties(gtest_170_lib  PROPERTIES COMPILE_DEFINITIONS "GTEST_HAS_RTTI=0")
+        set(GTEST_LIBRARIES gtest_170_lib)
+      endif()
       enable_testing(true)
 
      # obs see this: http://stackoverflow.com/questions/9589192/how-do-i-change-the-number-of-template-arguments-supported-by-msvcs-stdtupl
@@ -63,7 +69,7 @@
         IF( NOT(MSVC))
            set_target_properties(${test} PROPERTIES COMPILE_FLAGS "-isystem -pthread ")
         ENDIF( NOT(MSVC)) 
-        target_link_libraries(${test} g3logger gtest_170_lib)
+        target_link_libraries(${test} g3logger ${GTEST_LIBRARIES} )
 		add_test( ${test} ${test} )
       ENDFOREACH(test)
    
@@ -77,7 +83,7 @@
        add_executable(test_dynamic_loaded_shared_lib ${g3log_SOURCE_DIR}/test_main/test_main.cpp ${DIR_UNIT_TEST}/test_linux_dynamic_loaded_sharedlib.cpp)
        set_target_properties(test_dynamic_loaded_shared_lib PROPERTIES COMPILE_DEFINITIONS "GTEST_HAS_TR1_TUPLE=0")
        set_target_properties(test_dynamic_loaded_shared_lib PROPERTIES COMPILE_DEFINITIONS "GTEST_HAS_RTTI=0")
-       target_link_libraries(test_dynamic_loaded_shared_lib  ${G3LOG_LIBRARY} -ldl  gtest_170_lib )
+       target_link_libraries(test_dynamic_loaded_shared_lib  ${G3LOG_LIBRARY} -ldl ${GTEST_LIBRARIES} )
     ENDIF()
 ELSE() 
   message( STATUS "-DADD_G3LOG_UNIT_TEST=OFF" ) 
