--- origsrc/ssdeep-2.12/Makefile.am	2014-10-24 23:44:44.000000000 +0900
+++ src/ssdeep-2.12/Makefile.am	2015-04-22 00:16:55.728978400 +0900
@@ -2,7 +2,6 @@
 bin_PROGRAMS=ssdeep
 
 ssdeep_LDADD=libfuzzy.la
-ssdeep_LDFLAGS=-static
 
 ACLOCAL_AMFLAGS = -I m4
 
@@ -10,7 +9,7 @@ lib_LTLIBRARIES=libfuzzy.la
 libfuzzy_la_SOURCES=fuzzy.c edit_dist.c find-file-size.c
 libfuzzy_la_LDFLAGS=-no-undefined -version-info 2:0:0
 
-include_HEADERS=fuzzy.h edit_dist.h
+pkginclude_HEADERS=fuzzy.h edit_dist.h
 
 man_MANS=ssdeep.1
 
@@ -72,3 +71,5 @@ nice:
 preflight:
 	@grep RBF $(DISTFILES)
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/ssdeep-2.12/configure.ac	2014-10-24 22:28:12.000000000 +0900
+++ src/ssdeep-2.12/configure.ac	2015-04-22 00:16:15.467107500 +0900
@@ -1,7 +1,7 @@
 AC_INIT([SSDEEP],[2.12],[research@jessekornblum.com])
 AM_INIT_AUTOMAKE
 
-AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([Makefile ssdeep.pc])
 AM_CONFIG_HEADER([config.h])
 
 AC_CANONICAL_HOST
@@ -53,5 +53,3 @@ AC_CHECK_HEADERS([sys/mount.h],[],[],
 AC_FUNC_FSEEKO
 
 AC_OUTPUT
-
-
--- origsrc/ssdeep-2.12/ssdeep.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/ssdeep-2.12/ssdeep.pc.in	2015-04-22 00:15:51.538379200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: ssdeep
+Description: Program for computing context triggered piecewise hashes
+Version: @VERSION@
+Libs: -L${libdir} -lfuzzy
+Cflags: -I${includedir}/@PACKAGE@
