--- origsrc/sqlcheck-1.0.5/CMakeLists.txt	2017-08-16 04:13:40.000000000 +0900
+++ src/sqlcheck-1.0.5/CMakeLists.txt	2017-08-19 23:15:17.892999800 +0900
@@ -35,8 +35,11 @@ endif ()
 
 # ---[ C++11 Flags
 include(CheckCXXCompilerFlag)
+CHECK_CXX_COMPILER_FLAG("-std=gnu++11" COMPILER_SUPPORTS_GNUXX11)
 CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
-if(COMPILER_SUPPORTS_CXX11)
+if(COMPILER_SUPPORTS_GNUXX11)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11")
+elseif(COMPILER_SUPPORTS_CXX11)
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 else()
     message(STATUS "The compiler ${CMAKE_CXX_COMPILER} has no C++11 support. Please use a different C++ compiler.")
@@ -47,6 +50,9 @@ set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_F
 
 # -- [ GTest
 enable_testing()
+if(CYGWIN)
+    find_package(GTest REQUIRED)
+else()
 set(GOOGLETEST_ROOT external/googletest/googletest CACHE STRING "Google Test source root")
 include_directories(
     ${PROJECT_SOURCE_DIR}/${GOOGLETEST_ROOT}
@@ -61,6 +67,7 @@ foreach(_source ${GOOGLETEST_SOURCES})
 endforeach()
 
 add_library(googletest ${GOOGLETEST_SOURCES})
+endif()
 
 # -- [ Threads
 
@@ -68,7 +75,11 @@ find_package(Threads REQUIRED)
 
 # ---[ Flags
 if(UNIX OR APPLE)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -Wall -Wextra -Werror -lpthread")
+  if(CYGWIN)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror -lpthread")
+  else()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -Wall -Wextra -Werror -lpthread")
+  endif()
 endif()
 
 # ---[ Packaging
--- origsrc/sqlcheck-1.0.5/src/CMakeLists.txt	2017-08-16 04:13:40.000000000 +0900
+++ src/sqlcheck-1.0.5/src/CMakeLists.txt	2017-08-19 23:15:57.581000000 +0900
@@ -21,4 +21,4 @@ ${CMAKE_THREAD_LIBS_INIT}
 )
 
 # Add installation target
-install (TARGETS sqlcheck sqlcheck_library DESTINATION bin)
+install (TARGETS sqlcheck DESTINATION bin)
--- origsrc/sqlcheck-1.0.5/test/CMakeLists.txt	2017-08-16 04:13:40.000000000 +0900
+++ src/sqlcheck-1.0.5/test/CMakeLists.txt	2017-08-19 23:14:01.126000000 +0900
@@ -5,11 +5,15 @@
 # Make sure the compiler can find include files for our sqlcheck library
 include_directories (${CMAKE_SOURCE_DIR}/src/include)
 
+if(NOT CYGWIN)
+    set(GTEST_LIBRARY googletest)
+endif()
+
 # ---[ ARGS TEST
 add_executable(args_test args_test.cpp)
 target_link_libraries(args_test sqlcheck_library 
 ${GTEST_BOTH_LIBRARIES} 
-googletest
+${GTEST_LIBRARY}
 ${GLOG_LIBRARIES} 
 ${CMAKE_THREAD_LIBS_INIT}
 )
@@ -20,7 +24,7 @@ add_test(NAME ArgsTest COMMAND args_test
 add_executable(basic_test basic_test.cpp)
 target_link_libraries(basic_test sqlcheck_library 
 ${GTEST_BOTH_LIBRARIES} 
-googletest
+${GTEST_LIBRARY}
 ${GLOG_LIBRARIES} 
 ${CMAKE_THREAD_LIBS_INIT}
 )
