--- origsrc/libjunos-libjunos-0.0.0/Makefile.am	2012-09-09 20:48:31.000000000 +0900
+++ src/libjunos-libjunos-0.0.0/Makefile.am	2025-03-02 10:21:05.474831800 +0900
@@ -9,3 +9,5 @@ version: FORCE
 	@# As a side-effect, this updates version.
 	@echo Building $(PACKAGE_NAME) version $$( cd .. && ./version-gen.sh )
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = libjunos.pc
--- origsrc/libjunos-libjunos-0.0.0/configure.ac	2012-09-09 20:48:31.000000000 +0900
+++ src/libjunos-libjunos-0.0.0/configure.ac	2025-03-02 10:21:05.474831800 +0900
@@ -146,7 +146,7 @@ AC_ARG_ENABLE([strict-checks],
 		[enable_strict_checks="yes"])
 
 STRICT_CFLAGS=""
-for flag in -Wall -Werror; do
+for flag in -Wall; do
 	AC_MSG_CHECKING([whether $CC accepts $flag])
 
 	if test_cc_flags $flag; then
@@ -225,7 +225,7 @@ for flag in $LIBSSH2_IDIRS $LIBXML2_IDIR
 	fi
 done
 
-AC_CONFIG_FILES([Makefile doc/Makefile src/Makefile])
+AC_CONFIG_FILES([libjunos.pc Makefile doc/Makefile src/Makefile])
 AC_OUTPUT
 
 BUILD_DATE="`date --utc '+%F %T'` (UTC)"
--- origsrc/libjunos-libjunos-0.0.0/libjunos.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libjunos-libjunos-0.0.0/libjunos.pc.in	2025-03-02 10:21:05.474831800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libjunos
+Description: JUNOScript client library
+Version: @VERSION@
+Libs: -L${libdir} -ljunos
+Cflags: -I${includedir}
--- origsrc/libjunos-libjunos-0.0.0/src/Makefile.am	2012-09-09 20:48:31.000000000 +0900
+++ src/libjunos-libjunos-0.0.0/src/Makefile.am	2025-03-02 10:21:05.474831800 +0900
@@ -1,4 +1,5 @@
 AM_CFLAGS = @STRICT_CFLAGS@
+AM_LDFLAGS = -no-undefined
 
 include_HEADERS = junos.h libjunos_features.h
 lib_LTLIBRARIES = libjunos.la
@@ -22,14 +23,14 @@ junosc_SOURCES = junosc.c junos.h
 junosc_CFLAGS = $(AM_CFLAGS) -DBUILD_DATE="\"$$( date --utc '+%F %T' ) (UTC)\""
 junosc_LDADD = libjunos.la
 
-libjunos_features.h: libjunos_features.h.in ../version
-	source ../version; sed \
+libjunos_features.h: $(srcdir)/libjunos_features.h.in $(top_srcdir)/version
+	source $(top_srcdir)/version; sed \
 	    -e "s/@LIBJUNOS_VERSION_MAJOR@/$$VERSION_MAJOR/g" \
 	    -e "s/@LIBJUNOS_VERSION_MINOR@/$$VERSION_MINOR/g" \
 	    -e "s/@LIBJUNOS_VERSION_PATCH@/$$VERSION_PATCH/g" \
 	    -e "s/@LIBJUNOS_VERSION_EXTRA@/$$VERSION_EXTRA/g" \
 	    -e "s/@LIBJUNOS_VERSION_STRING@/$$VERSION_STRING/g" \
-	    libjunos_features.h.in > libjunos_features.h
+	    $(srcdir)/libjunos_features.h.in > libjunos_features.h
 
 .PHONY: FORCE
 
