--- origsrc/gamerzillagobj-release-0.1.2/Makefile	2023-09-14 10:19:18.000000000 +0900
+++ src/gamerzillagobj-release-0.1.2/Makefile	2024-03-23 09:40:13.724969400 +0900
@@ -35,7 +35,7 @@ TYPELIB_FILE=$(NAMESPACE)-$(NSVERSION).t
 all: libgamerzillagobj.la $(TYPELIB_FILE)
 
 libgamerzillagobj.la: $(OBJECTS)	
-	libtool link $(CC) $(LIBS) -rpath $(LIBDIR) $(OBJECTS) -o $@ 
+	libtool link $(CC) $(LIBS) -rpath $(LIBDIR) $(OBJECTS) -no-undefined -o $@ 
 
 $(TYPELIB_FILE): $(GIR_FILE)
 	g-ir-compiler $(GIR_FILE) --output=$(TYPELIB_FILE)
--- origsrc/gamerzillagobj-release-0.1.2/gamerzillagobj.c	2023-09-14 10:19:18.000000000 +0900
+++ src/gamerzillagobj-release-0.1.2/gamerzillagobj.c	2024-03-23 09:39:24.300537800 +0900
@@ -6,6 +6,8 @@
 
 #include <stdio.h>
 #include <pthread.h>
+#include <ctype.h>
+#include <errno.h>
 #include <gamerzilla.h>
 #include "gamerzillagobj.h"
 
