--- origsrc/mboxgrep-0.7.9/src/info.c	2003-04-14 04:15:58.000000000 +0900
+++ src/mboxgrep-0.7.9/src/info.c	2021-07-27 11:38:54.940410000 +0900
@@ -31,7 +31,7 @@ print_wrap (char *str, int len, int *n)
      /* {{{  */
 {
   *n += len;
-  fprintf (stdout, str);
+  fprintf (stdout, "%s", str);
   if (*n >= 50)
     {
       fprintf (stdout, "\n");
--- origsrc/mboxgrep-0.7.9/src/main.c	2003-08-25 04:23:50.000000000 +0900
+++ src/mboxgrep-0.7.9/src/main.c	2021-07-27 11:41:28.510854300 +0900
@@ -58,6 +58,8 @@ int count = 0;
 void *tmpp;
 checksum_t *cs;
 
+option_t config;
+
 int
 main (int argc, char **argv)
      /* {{{  */
--- origsrc/mboxgrep-0.7.9/src/mboxgrep.h	2003-04-07 06:01:49.000000000 +0900
+++ src/mboxgrep-0.7.9/src/mboxgrep.h	2021-07-27 11:39:44.414723200 +0900
@@ -129,6 +129,6 @@ typedef struct
 }
 option_t;
 
-option_t config;
+extern option_t config;
 
 #endif /* MBOXGREP_H */
