--- origsrc/s2tc-1.0/Makefile.am	2013-11-04 21:14:46.000000000 +0900
+++ src/s2tc-1.0/Makefile.am	2015-03-23 01:01:29.437281800 +0900
@@ -26,7 +26,7 @@ endif
 if ENABLE_LIB
 lib_LTLIBRARIES = libtxc_dxtn.la
 libtxc_dxtn_la_SOURCES = s2tc_algorithm.cpp s2tc_libtxc_dxtn.cpp s2tc_common.h s2tc_algorithm.h txc_dxtn.h s2tc_license.h
-libtxc_dxtn_la_LDFLAGS = -versioninfo 0:0:0
+libtxc_dxtn_la_LDFLAGS = -no-undefined -version-info 0:0:0
 libtxc_dxtn_la_LIBADD = -lm
 libtxc_dxtn_la_CFLAGS = -fvisibility=hidden -Wold-style-definition -Wstrict-prototypes -Wsign-compare -Wdeclaration-after-statement
 library_includedir = $(includedir)
