--- origsrc/STRUMPACK-6.3.1/CMakeLists.txt	2022-03-04 04:00:49.000000000 +0900
+++ src/STRUMPACK-6.3.1/CMakeLists.txt	2022-06-14 12:22:05.712350400 +0900
@@ -507,6 +507,10 @@ endif()
 # with clang
 set_property(TARGET strumpack PROPERTY POSITION_INDEPENDENT_CODE ON)
 
+if (DEFINED SOVERSION)
+  set_target_properties(strumpack PROPERTIES SOVERSION ${SOVERSION})
+endif()
+
 add_subdirectory(src)
 
 target_sources(strumpack
@@ -598,6 +602,7 @@ if(ButterflyPACK_FOUND)
   ## for some reason just adding the MPI::MPI_Fortran target only
   ## links with the f08 module, not with the regular fortran MPI
   ## library
+  target_include_directories(strumpack PRIVATE /usr/include/ButterflyPACK)
 endif()
 
 if(slate_FOUND)
@@ -671,7 +676,7 @@ install(FILES
   DESTINATION include)
 
 install(TARGETS strumpack EXPORT STRUMPACKTargets
-  LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+  RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 
 
 # export targets and all
@@ -699,3 +704,10 @@ install(FILES "${CMAKE_BINARY_DIR}/strum
 install(DIRECTORY cmake/Modules/
   DESTINATION ${ConfigPackageLocation}
   FILES_MATCHING PATTERN "*.cmake")
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/strumpack.pc.in ${CMAKE_BINARY_DIR}/strumpack.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/strumpack.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/STRUMPACK-6.3.1/src/HSS/HSSBasisID.hpp	2022-03-04 04:00:49.000000000 +0900
+++ src/STRUMPACK-6.3.1/src/HSS/HSSBasisID.hpp	2022-06-14 12:20:07.771425400 +0900
@@ -33,6 +33,10 @@
 
 #include "dense/DenseMatrix.hpp"
 
+#if defined(_P)
+#undef _P
+#endif
+
 namespace strumpack {
   namespace HSS {
 
--- origsrc/STRUMPACK-6.3.1/src/HSS/HSSBasisIDMPI.hpp	2022-03-04 04:00:49.000000000 +0900
+++ src/STRUMPACK-6.3.1/src/HSS/HSSBasisIDMPI.hpp	2022-06-14 12:20:07.774404100 +0900
@@ -32,6 +32,10 @@
 
 #include "dense/DistributedMatrix.hpp"
 
+#if defined(_P)
+#undef _P
+#endif
+
 namespace strumpack {
   namespace HSS {
 
--- origsrc/STRUMPACK-6.3.1/src/HSS/HSSExtra.hpp	2022-03-04 04:00:49.000000000 +0900
+++ src/STRUMPACK-6.3.1/src/HSS/HSSExtra.hpp	2022-06-14 12:20:07.783398900 +0900
@@ -30,6 +30,10 @@
 
 #include "dense/DenseMatrix.hpp"
 
+#if defined(_L)
+#undef _L
+#endif
+
 namespace strumpack {
   namespace HSS {
 
--- origsrc/STRUMPACK-6.3.1/src/HSS/HSSExtraMPI.hpp	2022-03-04 04:00:49.000000000 +0900
+++ src/STRUMPACK-6.3.1/src/HSS/HSSExtraMPI.hpp	2022-06-14 12:20:07.794363100 +0900
@@ -31,6 +31,10 @@
 #include "HSSExtra.hpp"
 #include "dense/DistributedMatrix.hpp"
 
+#if defined(_U)
+#undef _U
+#endif
+
 namespace strumpack {
   namespace HSS {
 
--- origsrc/STRUMPACK-6.3.1/src/sparse/EliminationTreeMPI.hpp	2022-03-04 04:00:49.000000000 +0900
+++ src/STRUMPACK-6.3.1/src/sparse/EliminationTreeMPI.hpp	2022-06-14 12:20:07.801367600 +0900
@@ -32,6 +32,10 @@
 #include "EliminationTree.hpp"
 #include "misc/MPIWrapper.hpp"
 
+#if defined(_P)
+#undef _P
+#endif
+
 namespace strumpack {
 
   // forward declarations
--- origsrc/STRUMPACK-6.3.1/src/sparse/EliminationTreeMPIDist.hpp	2022-03-04 04:00:49.000000000 +0900
+++ src/STRUMPACK-6.3.1/src/sparse/EliminationTreeMPIDist.hpp	2022-06-14 12:20:07.831549200 +0900
@@ -35,6 +35,10 @@
 #include "PropMapSparseMatrix.hpp"
 #include "dense/DistributedMatrix.hpp"
 
+#if defined(_P)
+#undef _P
+#endif
+
 namespace strumpack {
 
   // forward declarations
--- origsrc/STRUMPACK-6.3.1/strumpack.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/STRUMPACK-6.3.1/strumpack.pc.in	2022-06-14 12:20:07.768344000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: STRUMPACK
+Description: Structured Matrix Package
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lstrumpack
+Cflags: -I${includedir}/STRUMPACK
