--- origsrc/STRUMPACK-6.1.0/CMakeLists.txt	2021-10-27 15:34:20.000000000 +0900
+++ src/STRUMPACK-6.1.0/CMakeLists.txt	2021-11-08 13:57:16.748620600 +0900
@@ -461,6 +461,10 @@ endif()
 # with clang
 set_property(TARGET strumpack PROPERTY POSITION_INDEPENDENT_CODE ON)
 
+if (CYGWIN)
+  set_target_properties(strumpack PROPERTIES SOVERSION ${SOVERSION})
+endif()
+
 add_subdirectory(src)
 
 target_sources(strumpack
@@ -550,6 +554,7 @@ if(ButterflyPACK_FOUND)
   ## for some reason just adding the MPI::MPI_Fortran target only
   ## links with the f08 module, not with the regular fortran MPI
   ## library
+  target_include_directories(strumpack PRIVATE /usr/include/ButterflyPACK)
 endif()
 
 if(slate_FOUND)
@@ -623,7 +628,7 @@ install(FILES
   DESTINATION include)
 
 install(TARGETS strumpack EXPORT STRUMPACKTargets
-  LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+  RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 
 
 # export targets and all
@@ -651,3 +656,10 @@ install(FILES "${CMAKE_BINARY_DIR}/strum
 install(DIRECTORY cmake/Modules/
   DESTINATION ${ConfigPackageLocation}
   FILES_MATCHING PATTERN "*.cmake")
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/STRUMPACK.pc.in ${CMAKE_BINARY_DIR}/STRUMPACK.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/STRUMPACK.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/STRUMPACK-6.1.0/STRUMPACK.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/STRUMPACK-6.1.0/STRUMPACK.pc.in	2021-11-08 13:57:16.753600400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: STRUMPACK
+Description: Structured Matrix Package
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lstrumpack
+Cflags: -I${includedir}/STRUMPACK
--- origsrc/STRUMPACK-6.1.0/src/HSS/HSSBasisID.hpp	2021-10-27 15:34:20.000000000 +0900
+++ src/STRUMPACK-6.1.0/src/HSS/HSSBasisID.hpp	2021-11-08 13:57:16.759644900 +0900
@@ -33,6 +33,10 @@
 
 #include "dense/DenseMatrix.hpp"
 
+#if defined(_P)
+#undef _P
+#endif
+
 namespace strumpack {
   namespace HSS {
 
--- origsrc/STRUMPACK-6.1.0/src/HSS/HSSBasisIDMPI.hpp	2021-10-27 15:34:20.000000000 +0900
+++ src/STRUMPACK-6.1.0/src/HSS/HSSBasisIDMPI.hpp	2021-11-08 13:57:16.766629200 +0900
@@ -32,6 +32,10 @@
 
 #include "dense/DistributedMatrix.hpp"
 
+#if defined(_P)
+#undef _P
+#endif
+
 namespace strumpack {
   namespace HSS {
 
--- origsrc/STRUMPACK-6.1.0/src/HSS/HSSExtra.hpp	2021-10-27 15:34:20.000000000 +0900
+++ src/STRUMPACK-6.1.0/src/HSS/HSSExtra.hpp	2021-11-08 13:57:16.771840700 +0900
@@ -30,6 +30,10 @@
 
 #include "dense/DenseMatrix.hpp"
 
+#if defined(_L)
+#undef _L
+#endif
+
 namespace strumpack {
   namespace HSS {
 
--- origsrc/STRUMPACK-6.1.0/src/HSS/HSSExtraMPI.hpp	2021-10-27 15:34:20.000000000 +0900
+++ src/STRUMPACK-6.1.0/src/HSS/HSSExtraMPI.hpp	2021-11-08 13:57:16.777628500 +0900
@@ -31,6 +31,10 @@
 #include "HSSExtra.hpp"
 #include "dense/DistributedMatrix.hpp"
 
+#if defined(_U)
+#undef _U
+#endif
+
 namespace strumpack {
   namespace HSS {
 
--- origsrc/STRUMPACK-6.1.0/src/sparse/EliminationTreeMPI.hpp	2021-10-27 15:34:20.000000000 +0900
+++ src/STRUMPACK-6.1.0/src/sparse/EliminationTreeMPI.hpp	2021-11-08 13:57:16.790670700 +0900
@@ -32,6 +32,10 @@
 #include "EliminationTree.hpp"
 #include "misc/MPIWrapper.hpp"
 
+#if defined(_P)
+#undef _P
+#endif
+
 namespace strumpack {
 
   // forward declarations
--- origsrc/STRUMPACK-6.1.0/src/sparse/EliminationTreeMPIDist.hpp	2021-10-27 15:34:20.000000000 +0900
+++ src/STRUMPACK-6.1.0/src/sparse/EliminationTreeMPIDist.hpp	2021-11-08 13:57:16.802664600 +0900
@@ -35,6 +35,10 @@
 #include "PropMapSparseMatrix.hpp"
 #include "dense/DistributedMatrix.hpp"
 
+#if defined(_P)
+#undef _P
+#endif
+
 namespace strumpack {
 
   // forward declarations
