diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x version.hpp origsrc/steins-gate-new-0.5.2/CMakeLists.txt src/steins-gate-new-0.5.2/CMakeLists.txt
--- origsrc/steins-gate-new-0.5.2/CMakeLists.txt	2018-05-02 07:14:40.000000000 +0900
+++ src/steins-gate-new-0.5.2/CMakeLists.txt	2018-05-21 08:31:52.460210500 +0900
@@ -21,8 +21,10 @@ message(STATUS "Configuring steins-gate
 # project name and language
 project (steins-gate CXX)
 
-set(CMAKE_BUILD_TYPE Debug)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -O2 -ljpeg")
+if(NOT CYGWIN)
+    set(CMAKE_BUILD_TYPE Debug)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -O2 -ljpeg")
+endif()
 
 # include version number in header
 configure_file(${PROJECT_SOURCE_DIR}/src/version.hpp.in
