--- origsrc/steins-gate-0.5.3/CMakeLists.txt	2018-05-24 21:15:57.000000000 +0900
+++ src/steins-gate-0.5.3/CMakeLists.txt	2019-01-22 17:43:37.003808800 +0900
@@ -21,8 +21,10 @@ message(STATUS "Configuring steins-gate
 # project name and language
 project (steins-gate CXX)
 
-set(CMAKE_BUILD_TYPE Debug)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -O2 -ljpeg")
+if(NOT CYGWIN)
+    set(CMAKE_BUILD_TYPE Debug)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -O2 -ljpeg")
+endif()
 
 # include version number in header
 configure_file(${PROJECT_SOURCE_DIR}/src/version.hpp.in
