--- origsrc/dftd4-4.0.2/meson.build	2026-02-06 16:37:36.000000000 +0900
+++ src/dftd4-4.0.2/meson.build	2026-03-17 05:05:39.003498000 +0900
@@ -83,7 +83,7 @@ if install
     )
   endif
 
-  module_id = meson.project_name() / fc_id + '-' + fc.version()
+  module_id = meson.project_name() / 'modules'
   meson.add_install_script(
     find_program(files('config'/'install-mod.py')),
     get_option('includedir') / module_id,
