--- origsrc/dftd4-3.7.0/CMakeLists.txt	2024-08-29 22:40:10.000000000 +0900
+++ src/dftd4-3.7.0/CMakeLists.txt	2025-01-16 15:12:14.563574300 +0900
@@ -119,6 +119,7 @@ install(
   "${PROJECT_NAME}-lib"
   EXPORT
   "${PROJECT_NAME}-targets"
+  RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
   LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
   ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
 )
--- origsrc/dftd4-3.7.0/config/CMakeLists.txt	2024-08-29 22:40:10.000000000 +0900
+++ src/dftd4-3.7.0/config/CMakeLists.txt	2025-01-16 15:12:14.563574300 +0900
@@ -27,12 +27,16 @@ if(NOT DEFINED "${PROJECT_NAME}-dependen
   )
 endif()
 
-set(
-  "${PROJECT_NAME}-module-dir"
-  "${PROJECT_NAME}/${CMAKE_Fortran_COMPILER_ID}-${CMAKE_Fortran_COMPILER_VERSION}"
-  CACHE STRING
-  "Subdirectory to install generated module files to"
-)
+if (CYGWIN)
+  set("${PROJECT_NAME}-module-dir" "${PROJECT_NAME}")
+else()
+  set(
+    "${PROJECT_NAME}-module-dir"
+    "${PROJECT_NAME}/${CMAKE_Fortran_COMPILER_ID}-${CMAKE_Fortran_COMPILER_VERSION}"
+    CACHE STRING
+    "Subdirectory to install generated module files to"
+  )
+endif()
 set(
   module-dir
   "${${PROJECT_NAME}-module-dir}"
