--- origsrc/libcdada-0.6.1/Makefile.am	2024-07-30 06:44:31.000000000 +0900
+++ src/libcdada-0.6.1/Makefile.am	2024-09-12 23:25:03.525842400 +0900
@@ -25,3 +25,6 @@ uninstall-hook:
 maintainer-clean-local:
 	rm -rf $(top_srcdir)/m4
 	rm -rf $(top_srcdir)/build-aux
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libcdada-0.6.1/configure.ac	2024-07-30 06:44:31.000000000 +0900
+++ src/libcdada-0.6.1/configure.ac	2024-09-12 23:25:03.525842400 +0900
@@ -1,5 +1,5 @@
 m4_include([VERSION]) #Force reconf on VERSION change
-AC_INIT(LIBCDADA, m4_esyscmd_s(cat VERSION), marcdevel@gmail.com, libcdada, http://github.com/msune/libcdada)
+AC_INIT(LIBCDADA, m4_esyscmd_s([cat VERSION | sed -e 's/^v//']), marcdevel@gmail.com, libcdada, http://github.com/msune/libcdada)
 
 AC_CONFIG_AUX_DIR([build-aux])
 AC_CONFIG_MACRO_DIR([m4])
@@ -51,7 +51,7 @@ m4_include([config_m4/tests.m4])
 m4_include([config_m4/examples.m4])
 
 # Output files
-AC_CONFIG_FILES([
+AC_CONFIG_FILES([libcdada.pc 
 	Makefile
 	include/Makefile
 	examples/Makefile
--- origsrc/libcdada-0.6.1/examples/Makefile.am	2024-07-30 06:44:31.000000000 +0900
+++ src/libcdada-0.6.1/examples/Makefile.am	2024-09-12 23:25:03.525842400 +0900
@@ -11,32 +11,32 @@ noinst_PROGRAMS = bbitmap      \
 
 list_u32_SOURCES = list_u32.c
 list_u32_CPPFLAGS = -I$(top_builddir)/include/ -I$(top_srcdir)/include/
-list_u32_LDADD = ../src/.libs/libcdada.a -lstdc++
+list_u32_LDADD = ../src/libcdada.la -lstdc++
 
 map_struct_SOURCES = map_struct.c
 map_struct_CPPFLAGS = -I$(top_builddir)/include/ -I$(top_srcdir)/include/
-map_struct_LDADD = ../src/.libs/libcdada.a -lstdc++
+map_struct_LDADD = ../src/libcdada.la -lstdc++
 
 custom_gen_SOURCES = custom-gen/my_program_gen.c custom-gen/cdada.cc
 custom_gen_CPPFLAGS = -I$(top_builddir)/include/ -I$(top_srcdir)/include/
-custom_gen_LDADD = ../src/.libs/libcdada.a -lstdc++
+custom_gen_LDADD = ../src/libcdada.la -lstdc++
 
 custom_macro_SOURCES = custom-macro/my_program.c custom-macro/cdada.cc
 custom_macro_CPPFLAGS = -I$(top_builddir)/include/ -I$(top_srcdir)/include/
-custom_macro_LDADD = ../src/.libs/libcdada.a -lstdc++
+custom_macro_LDADD = ../src/libcdada.la -lstdc++
 
 str_SOURCES = str.c
 str_CPPFLAGS = -I$(top_builddir)/include/ -I$(top_srcdir)/include/
-str_LDADD = ../src/.libs/libcdada.a -lstdc++
+str_LDADD = ../src/libcdada.la -lstdc++
 
 bbitmap_SOURCES = bbitmap.c
 bbitmap_CPPFLAGS = -I$(top_builddir)/include/ -I$(top_srcdir)/include/
-bbitmap_LDADD = ../src/.libs/libcdada.a -lstdc++
+bbitmap_LDADD = ../src/libcdada.la -lstdc++
 
 queue_u32_SOURCES = queue_u32.c
 queue_u32_CPPFLAGS = -I$(top_builddir)/include/ -I$(top_srcdir)/include/
-queue_u32_LDADD = ../src/.libs/libcdada.a -lstdc++
+queue_u32_LDADD = ../src/libcdada.la -lstdc++
 
 stack_u32_SOURCES = stack_u32.c
 stack_u32_CPPFLAGS = -I$(top_builddir)/include/ -I$(top_srcdir)/include/
-stack_u32_LDADD = ../src/.libs/libcdada.a -lstdc++
+stack_u32_LDADD = ../src/libcdada.la -lstdc++
--- origsrc/libcdada-0.6.1/libcdada.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libcdada-0.6.1/libcdada.pc.in	2024-09-12 23:25:03.525842400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libcdada
+Description: Basic data structures in C
+Version: @VERSION@
+Libs: -L${libdir} -lcdada
+Cflags: -I${includedir}
--- origsrc/libcdada-0.6.1/src/Makefile.am	2024-07-30 06:44:31.000000000 +0900
+++ src/libcdada-0.6.1/src/Makefile.am	2024-09-12 23:25:03.525842400 +0900
@@ -12,3 +12,4 @@ libcdada_la_SOURCES = bbitmap.cc \
                       utils.c
 libcdada_la_CPPFLAGS = -I$(top_builddir)/include/ -I$(top_srcdir)/include/ -D__CDADA_INTERNAL_INCLUDE
 libcdada_la_LIBADD =
+libcdada_la_LDFLAGS = -no-undefined
--- origsrc/libcdada-0.6.1/test/Makefile.am	2024-07-30 06:44:31.000000000 +0900
+++ src/libcdada-0.6.1/test/Makefile.am	2024-09-12 23:25:03.541473900 +0900
@@ -63,7 +63,7 @@ str_test_LDADD= -lpthread ../src/libcdad
 benchmark_SOURCES= benchmark.cc
 benchmark_CXXFLAGS= -O3
 benchmark_CPPFLAGS= -I$(top_builddir)/include/ -I$(top_srcdir)/include/
-benchmark_LDADD= -lpthread ../src/.libs/libcdada.a
+benchmark_LDADD= -lpthread ../src/libcdada.la
 
 #Valgrind
 if VALGRIND_ENABLED
