--- origsrc/qdldl-0.1.9/CMakeLists.txt	2025-08-29 08:13:26.000000000 +0900
+++ src/qdldl-0.1.9/CMakeLists.txt	2025-09-18 14:15:37.359254800 +0900
@@ -205,6 +205,10 @@ message( STATUS "Shared library build is
 if( QDLDL_BUILD_SHARED_LIB )
     add_library (qdldl SHARED ${qdldl_src} ${qdldl_headers})
 
+    if (DEFINED SOVERSION)
+      set_target_properties(qdldl PROPERTIES SOVERSION "${SOVERSION}")
+    endif()
+
     # Declare that we are building the shared library to get proper symbol exports.
     # Shared library consumers should also define QDLDL_SHARED_LIB to get the library
     # exports properly, so we do it for them in the CMake interface by defining it as
@@ -304,3 +308,10 @@ if( QDLDL_UNITTESTS )
     enable_testing()
     add_test(NAME tester COMMAND $<TARGET_FILE:qdldl_tester>)
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/qdldl.pc.in ${CMAKE_BINARY_DIR}/qdldl.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/qdldl.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/qdldl-0.1.9/qdldl.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/qdldl-0.1.9/qdldl.pc.in	2025-09-18 14:15:37.359254800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: qdldl
+Description: Free LDL factorisation routine
+Version: @QDLDL_VERSION@
+Libs: -L${libdir} -lqdldl
+Cflags: -I${includedir}
