--- origsrc/qdldl-0.1.5/CMakeLists.txt	2020-07-30 05:59:52.000000000 +0900
+++ src/qdldl-0.1.5/CMakeLists.txt	2021-11-24 07:40:48.473273200 +0900
@@ -101,25 +101,26 @@ target_include_directories(qdldlobject P
 
 include(GNUInstallDirs)
 
-# Static library
-add_library (qdldlstatic STATIC ${qdldl_src} ${qdldl_headers})
-# Give same name to static library output
-set_target_properties(qdldlstatic PROPERTIES OUTPUT_NAME qdldl)
+if (NOT BUILD_SHARED_LIBS)
+  # Static library
+  add_library (qdldlstatic STATIC ${qdldl_src} ${qdldl_headers})
+  # Give same name to static library output
+  set_target_properties(qdldlstatic PROPERTIES OUTPUT_NAME qdldl)
 
-# Declare include directories for the cmake exported target
-target_include_directories(qdldlstatic
+  # Declare include directories for the cmake exported target
+  target_include_directories(qdldlstatic
                            PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
                                   "$<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${CMAKE_INSTALL_INCLUDEDIR}/qdldl>")
 
-# Install Static Library
-# ----------------------------------------------
+  # Install Static Library
+  # ----------------------------------------------
 
-install(TARGETS qdldlstatic
+  install(TARGETS qdldlstatic
         EXPORT  ${PROJECT_NAME}
         ARCHIVE       DESTINATION "${CMAKE_INSTALL_LIBDIR}"
         LIBRARY       DESTINATION "${CMAKE_INSTALL_LIBDIR}"
         RUNTIME       DESTINATION "${CMAKE_INSTALL_BINDIR}")
-
+endif()
 
 # Install Headers
 # ----------------------------------------------
@@ -131,6 +132,10 @@ install(FILES ${qdldl_headers} DESTINATI
 # Create qdldl shared library
 add_library (qdldl SHARED ${qdldl_src} ${qdldl_headers})
 
+if (CYGWIN)
+  set_target_properties(qdldl PROPERTIES SOVERSION ${SOVERSION})
+endif()
+
 # Declare include directories for the cmake exported target
 target_include_directories(qdldl
 	PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
@@ -143,9 +148,9 @@ install(TARGETS qdldl
 	ARCHIVE       DESTINATION "${CMAKE_INSTALL_LIBDIR}"
 	RUNTIME       DESTINATION "${CMAKE_INSTALL_BINDIR}")
 
-# Create demo executable (linked to static library)
+# Create demo executable (linked to shared library)
 add_executable (qdldl_example ${PROJECT_SOURCE_DIR}/examples/example.c)
-target_link_libraries (qdldl_example qdldlstatic)
+target_link_libraries (qdldl_example qdldl)
 
 
 # Create CMake packages for the build directory
@@ -202,10 +207,19 @@ if (UNITTESTS)
     add_executable(qdldl_tester
                 ${PROJECT_SOURCE_DIR}/tests/qdldl_tester.c ${PROJECT_SOURCE_DIR}/tests/minunit.h
                 ${test_headers})
-    target_link_libraries (qdldl_tester qdldlstatic)
+    target_link_libraries (qdldl_tester qdldl)
 
     # Add testing
     include(CTest)
     enable_testing()
     add_test(NAME tester COMMAND $<TARGET_FILE:qdldl_tester>)
 endif()
+
+if (CYGWIN)
+  set(prefix ${CMAKE_INSTALL_PREFIX})
+  set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+  set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+  set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+  configure_file(${CMAKE_SOURCE_DIR}/qdldl.pc.in ${CMAKE_BINARY_DIR}/qdldl.pc @ONLY)
+  install(FILES ${CMAKE_BINARY_DIR}/qdldl.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+endif()
--- origsrc/qdldl-0.1.5/qdldl.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/qdldl-0.1.5/qdldl.pc.in	2021-11-24 07:35:58.397426600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: qdldl
+Description: Free LDL factorisation routine
+Version: @VERSION@
+Libs: -L${libdir} -lqdldl
+Cflags: -I${includedir}
