--- origsrc/libvc-gdm70x/src/Makefile.am	2013-10-13 00:19:17.000000000 +0900
+++ src/libvc-gdm70x/src/Makefile.am	2013-10-13 00:19:38.481216700 +0900
@@ -3,7 +3,7 @@ lib_LTLIBRARIES = libvc-gdm70x.la
 libvc_gdm70x_la_SOURCES = libvc-gdm70x.c
 include_HEADERS = vc-gdm70x.h
 
-libvc_gdm70x_la_LDFLAGS = -version-info 2:0:2
+libvc_gdm70x_la_LDFLAGS = -no-undefined -version-info 2:0:2
 
 bin_PROGRAMS = vc-gdm70x
 vc_gdm70x_SOURCES = vc-gdm70x.c
