--- origsrc/absco/configure.ac	2023-09-08 16:37:59.000000000 +0900
+++ src/absco/configure.ac	2023-09-08 16:59:54.287930700 +0900
@@ -1,4 +1,4 @@
-AC_INIT([absco], [1:0], [http://apps.jcns.fz-juelich.de/absco])
+AC_INIT([absco], [1.0.0], [http://apps.jcns.fz-juelich.de/absco])
 
 # standard configuration for shared libraries
 AC_CONFIG_AUX_DIR([build-aux])
@@ -19,8 +19,8 @@ AC_CONFIG_SRCDIR([lib/absco.c])
 # dependency checks
 AC_CHECK_LIB([m], [cos], ,
 		  [AC_MSG_ERROR(libm not found or corrupted)])
-AC_CHECK_LIB([gslcblas], [cblas_dgemm], ,
-			 [AC_MSG_ERROR(libgslcblas not found or out of sync)])
+AC_CHECK_LIB([blas], [cblas_dgemm], ,
+			 [AC_MSG_ERROR(libblas not found or out of sync)])
 AC_CHECK_LIB([gsl], [gsl_blas_dgemm], ,
 		    [AC_MSG_ERROR(libgsl not found or out of sync)])
 
--- origsrc/absco/lib/Makefile.am	2023-09-08 16:37:59.000000000 +0900
+++ src/absco/lib/Makefile.am	2023-09-08 17:00:17.209722200 +0900
@@ -2,4 +2,4 @@ AM_CFLAGS = -std=c99 -D_GNU_SOURCE
 lib_LTLIBRARIES = libabsco.la
 libabsco_la_SOURCES = absco.c absco.h
 include_HEADERS = absco.h
-libabsco_la_LDFLAGS = -version-info $(VERSION)
+libabsco_la_LDFLAGS = -no-undefined -version-info 1:0
