--- origsrc/goaccess-1.5/Makefile.am	2021-05-25 22:32:30.000000000 +0900
+++ src/goaccess-1.5/Makefile.am	2021-06-07 22:01:24.408113200 +0900
@@ -46,6 +46,7 @@ CLEANFILES =                          \
 # Tpls
 src/tpls.h: bin2c$(EXEEXT) $(srcdir)/resources/tpls.html
 if HAS_SEDTR
+	mkdir -p resources
 	cat $(srcdir)/resources/tpls.html | sed "s/^[[:space:]]*//" | sed "/^$$/d" | tr -d "\r\n" > resources/tpls.html.tmp
 	./bin2c resources/tpls.html.tmp src/tpls.h tpls
 else
