--- origsrc/goaccess-1.4.6/Makefile.am	2021-02-23 00:29:43.000000000 +0900
+++ src/goaccess-1.4.6/Makefile.am	2021-05-20 15:00:42.000000000 +0900
@@ -46,6 +46,7 @@ CLEANFILES =                          \
 # Tpls
 src/tpls.h: bin2c$(EXEEXT) $(srcdir)/resources/tpls.html
 if HAS_SEDTR
+	mkdir -p resources
 	cat $(srcdir)/resources/tpls.html | sed "s/^[[:space:]]*//" | sed "/^$$/d" | tr -d "\r\n" > resources/tpls.html.tmp
 	./bin2c resources/tpls.html.tmp src/tpls.h tpls
 else
