--- origsrc/goaccess-1.6/src/parser.c	2022-06-01 01:48:57.000000000 +0900
+++ src/goaccess-1.6/src/parser.c	2022-06-29 10:44:01.999542300 +0900
@@ -1345,7 +1345,7 @@ find_xff_host (GLogItem * logitem, char
   /* if the log format current char is not within the braces special chars, then
    * we assume the range of IPs are within hard delimiters */
   if (!strchr (skips, **p) && strchr (*str, **p)) {
-    strcpy (pch, (char[2]) { (char) **p, '\0' });
+    strcpy (pch, ((char[2]) { (char) **p, '\0' }));
     if (!(extract = parse_string (&(*str), pch, 1)))
       goto clean;
 
