--- origsrc/goaccess-1.1.1/Makefile.am	2016-09-30 11:28:08.000000000 +0900
+++ src/goaccess-1.1.1/Makefile.am	2016-11-27 10:27:09.329350600 +0900
@@ -37,28 +37,28 @@ CLEANFILES =          \
 
 # Tpls
 tpls.h: bin2c$(EXEEXT) resources/tpls.html
-	./bin2c resources/tpls.html src/tpls.h tpls
+	./bin2c $(srcdir)/resources/tpls.html src/tpls.h tpls
 # Bootstrap
 bootstrapcss.h: bin2c$(EXEEXT) resources/css/bootstrap.min.css
-	./bin2c resources/css/bootstrap.min.css src/bootstrapcss.h bootstrap_css
+	./bin2c $(srcdir)/resources/css/bootstrap.min.css src/bootstrapcss.h bootstrap_css
 # Font Awesome
 facss.h: bin2c$(EXEEXT) resources/css/fa.min.css
-	./bin2c resources/css/fa.min.css src/facss.h fa_css
+	./bin2c $(srcdir)/resources/css/fa.min.css src/facss.h fa_css
 # App.css
 appcss.h: bin2c$(EXEEXT) resources/css/app.css
-	./bin2c resources/css/app.css src/appcss.h app_css
+	./bin2c $(srcdir)/resources/css/app.css src/appcss.h app_css
 # D3.js
 d3js.h: bin2c$(EXEEXT) resources/js/d3.v3.min.js
-	./bin2c resources/js/d3.v3.min.js src/d3js.h d3_js
+	./bin2c $(srcdir)/resources/js/d3.v3.min.js src/d3js.h d3_js
 # Hogan.js
 hoganjs.h: bin2c$(EXEEXT) resources/js/hogan.min.js
-	./bin2c resources/js/hogan.min.js src/hoganjs.h hogan_js
+	./bin2c $(srcdir)/resources/js/hogan.min.js src/hoganjs.h hogan_js
 # Charts.js
 chartsjs.h: bin2c$(EXEEXT) resources/js/charts.js
-	./bin2c resources/js/charts.js src/chartsjs.h charts_js
+	./bin2c $(srcdir)/resources/js/charts.js src/chartsjs.h charts_js
 # App.js
 appjs.h: bin2c$(EXEEXT) resources/js/app.js
-	./bin2c resources/js/app.js src/appjs.h app_js
+	./bin2c $(srcdir)/resources/js/app.js src/appjs.h app_js
 
 confdir = $(sysconfdir)
 dist_conf_DATA = config/goaccess.conf
