--- origsrc/goaccess-0.8.5/Makefile.am	2014-09-06 22:35:11.000000000 +0900
+++ src/goaccess-0.8.5/Makefile.am	2014-09-19 22:20:22.972380300 +0900
@@ -62,8 +62,6 @@ else
 AM_CFLAGS = -O2 -DSYSCONFDIR=\"$(sysconfdir)\"
 endif
 
-AM_LDFLAGS = -rdynamic
-
 AM_CFLAGS += @GLIB2_CFLAGS@
 AM_CFLAGS += -Wno-long-long -Wall -W -Wnested-externs -Wformat=2
 AM_CFLAGS += -Wmissing-prototypes -Wstrict-prototypes -Wmissing-declarations
--- origsrc/goaccess-0.8.5/configure.ac	2014-09-14 21:51:43.000000000 +0900
+++ src/goaccess-0.8.5/configure.ac	2014-09-19 22:18:56.072966200 +0900
@@ -16,7 +16,6 @@ AC_PROG_CC
 
 # pthread
 AC_CHECK_LIB([pthread], [pthread_create], [], [AC_MSG_ERROR([pthread is missing])])
-CFLAGS="$CFLAGS -pthread"
 
 # DEBUG
 AC_ARG_ENABLE(debug, [  --enable-debug   Create a debug build. Default is disabled],
