--- origsrc/goaccess-1.4/Makefile.am	2020-05-17 21:34:51.000000000 +0900
+++ src/goaccess-1.4/Makefile.am	2020-07-09 09:44:47.491189000 +0900
@@ -46,6 +46,7 @@ CLEANFILES =                          \
 # Tpls
 src/tpls.h: bin2c$(EXEEXT) $(srcdir)/resources/tpls.html
 if HAS_SEDTR
+	mkdir -p resources
 	cat $(srcdir)/resources/tpls.html | sed "s/^[[:space:]]*//" | sed "/^$$/d" | tr -d "\r\n" > resources/tpls.html.tmp
 	./bin2c resources/tpls.html.tmp src/tpls.h tpls
 else
--- origsrc/goaccess-1.4/TODO	2020-05-17 21:34:51.000000000 +0900
+++ src/goaccess-1.4/TODO	2020-07-09 10:01:50.408636100 +0900
@@ -4,7 +4,7 @@
 For a more comprehensive list of to-do items, please refer to the GitHub site.
 https://github.com/allinurl/goaccess/issues
 
-or visit http://goaccess.io/faq#todo
+or visit https://goaccess.io/faq#todo
 
 If you are interested in working on any of the items listed in there, email
 goaccess@prosoftcorp.com or better, open a new issue:
--- origsrc/goaccess-1.4/src/commons.c	2020-05-17 21:34:51.000000000 +0900
+++ src/goaccess-1.4/src/commons.c	2020-07-09 10:00:56.519181700 +0900
@@ -151,7 +151,7 @@ display_default_config_file (void) {
 void
 display_version (void) {
   fprintf (stdout, "GoAccess - %s.\n", GO_VERSION);
-  fprintf (stdout, "%s: http://goaccess.io\n", INFO_MORE_INFO);
+  fprintf (stdout, "%s: https://goaccess.io\n", INFO_MORE_INFO);
   fprintf (stdout, "Copyright (C) 2009-2020 by Gerardo Orellana\n");
   fprintf (stdout, "\nBuild configure arguments:\n");
 #ifdef DEBUG
--- origsrc/goaccess-1.4/src/commons.h	2020-05-17 21:34:51.000000000 +0900
+++ src/goaccess-1.4/src/commons.h	2020-07-09 10:00:44.535296600 +0900
@@ -44,7 +44,7 @@
 #endif
 #define GO_UNUSED __attribute__((unused))
 #define GO_VERSION 		"1.4"
-#define GO_WEBSITE 		"http://goaccess.io/"
+#define GO_WEBSITE 		"https://goaccess.io/"
 extern struct tm *now_tm;
 
 /* common char array buffer size */
--- origsrc/goaccess-1.4/src/options.c	2020-05-18 01:15:14.000000000 +0900
+++ src/goaccess-1.4/src/options.c	2020-07-09 10:01:18.052938600 +0900
@@ -276,7 +276,7 @@ cmd_help (void)
   "\n"
 
   "%s `man goaccess`.\n\n"
-  "%s: http://goaccess.io\n"
+  "%s: https://goaccess.io\n"
   "GoAccess Copyright (C) 2009-2017 by Gerardo Orellana"
   "\n\n"
   , INFO_HELP_EXAMPLES, INFO_MORE_INFO
--- origsrc/goaccess-1.4/src/tpl.c	2020-05-17 21:34:51.000000000 +0900
+++ src/goaccess-1.4/src/tpl.c	2020-07-09 09:49:52.011576200 +0900
@@ -57,7 +57,7 @@ typedef unsigned __int64 uint64_t;
 #define S_ISREG(mode)  (((mode) & S_IFMT) == S_IFREG)
 #endif
 
-#if ( defined __CYGWIN__ || defined __MINGW32__ || defined _WIN32 )
+#if ( defined __MINGW32__ || defined _WIN32 )
 #include "win/mman.h"   /* mmap */
 #else
 #include <sys/mman.h>   /* mmap */
--- origsrc/goaccess-1.4/src/ui.c	2020-05-17 21:34:51.000000000 +0900
+++ src/goaccess-1.4/src/ui.c	2020-07-09 10:01:33.140980100 +0900
@@ -1757,7 +1757,7 @@ load_sort_win (WINDOW * main_win, GModul
 /* Help menu data (F1/h). */
 static const char *help_main[] = {
   "Copyright (C) 2009-2017 by Gerardo Orellana",
-  "http://goaccess.io - <hello@goaccess.io>",
+  "https://goaccess.io - <hello@goaccess.io>",
   "Released under the MIT License.",
   "",
   "See `man` page for more details",
