--- origsrc/libgeohash36/CMakeLists.txt	1970-01-01 09:00:00.000000000 +0900
+++ src/libgeohash36/CMakeLists.txt	2025-06-05 20:34:32.213926400 +0900
@@ -0,0 +1,27 @@
+cmake_minimum_required(VERSION 3.28)
+project(libgeohash36
+  VERSION 0.0.0
+  DESCRIPTION "C implementation of Geohash-36"
+  HOMEPAGE_URL "https://github.com/rafamdr/libgeohash36")
+
+include(GNUInstallDirs)
+include(CTest)
+
+add_library(geohash36 SHARED geohash36.c)
+set_target_properties(geohash36 PROPERTIES SOVERSION 0)
+
+install(TARGETS geohash36
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+add_executable(gethash36_test geohash36_test.c)
+target_link_libraries(gethash36_test geohash36)
+add_test(gethash36_test gethash36_test)
+
+install(FILES geohash36.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libgeohash36.pc.in ${CMAKE_BINARY_DIR}/libgeohash36.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libgeohash36.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libgeohash36/geohash36.c	2025-06-05 18:52:43.000000000 +0900
+++ src/libgeohash36/geohash36.c	2025-06-05 20:33:13.114112000 +0900
@@ -61,12 +61,12 @@ Constants
 ***********************************************************************************************************************/
 const char base36[GEOHASH_MATRIX_SIDE][GEOHASH_MATRIX_SIDE] =
 {
-   '2', '3', '4', '5', '6', '7',
-   '8', '9', 'b', 'B', 'C', 'd',
-   'D', 'F', 'g', 'G', 'h', 'H',
-   'j', 'J', 'K', 'l', 'L', 'M',
-   'n', 'N', 'P', 'q', 'Q', 'r',
-   'R', 't', 'T', 'V', 'W', 'X'
+   {'2', '3', '4', '5', '6', '7'},
+   {'8', '9', 'b', 'B', 'C', 'd'},
+   {'D', 'F', 'g', 'G', 'h', 'H'},
+   {'j', 'J', 'K', 'l', 'L', 'M'},
+   {'n', 'N', 'P', 'q', 'Q', 'r'},
+   {'R', 't', 'T', 'V', 'W', 'X'}
 };
 //----------------------------------------------------------------------------------------------------------------------
 
--- origsrc/libgeohash36/libgeohash36.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libgeohash36/libgeohash36.pc.in	2025-06-05 20:34:49.807601600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libgeohash36
+Description: C implementation of Geohash-36
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lgeohash36
+Cflags: -I${includedir}
