--- origsrc/libkkc-0.2.7/Makefile.am	2013-05-11 05:18:47.000000000 +0900
+++ src/libkkc-0.2.7/Makefile.am	2013-09-07 18:32:57.252808300 +0900
@@ -14,7 +14,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-SUBDIRS = marisa-glib libkkc data tools tests po
+SUBDIRS = marisa-glib libkkc data tools po
 DISTCHECK_CONFIGURE_FLAGS = --enable-docs
 ACLOCAL_AMFLAGS = -I m4
 
--- origsrc/libkkc-0.2.7/libkkc/Makefile.am	2013-06-29 16:02:46.000000000 +0900
+++ src/libkkc-0.2.7/libkkc/Makefile.am	2013-09-07 18:32:57.252808300 +0900
@@ -48,8 +48,8 @@ libkkc_la_CFLAGS =				\
 	-DLOCALEDIR=\"$(datadir)/locale\"	\
 	-I $(top_srcdir)			\
 	$(NULL)
-libkkc_la_LIBADD = $(LIBKKC_LIBS) $(top_builddir)/marisa-glib/libmarisa-glib.la
-libkkc_la_LDFLAGS =							\
+libkkc_la_LIBADD = $(LIBKKC_LIBS) $(top_builddir)/marisa-glib/libmarisa-glib.la -lstdc++
+libkkc_la_LDFLAGS = -no-undefined					\
 	$(CODE_COVERAGE_LDFLAGS)					\
 	-version-info "$(LT_CURRENT)":"$(LT_REVISION)":"$(LT_AGE)"	\
 	-export-dynamic							\
--- origsrc/libkkc-0.2.7/marisa-glib/Makefile.am	2013-04-27 09:39:34.000000000 +0900
+++ src/libkkc-0.2.7/marisa-glib/Makefile.am	2013-09-07 18:32:57.276245800 +0900
@@ -37,7 +37,7 @@ INTROSPECTION_SCANNER_ARGS = --add-inclu
 INTROSPECTION_COMPILER_ARGS = --includedir=$(srcdir)
 
 Marisa.gir: libmarisa-glib.la
-Marisa_gir_SCANNERFLAGS = --pkg-export=marisa-glib --pkg=marisa --namespace=Marisa $(MARISA_GLIB_STATIC_DEPENDENCIES)
+Marisa_gir_SCANNERFLAGS = --pkg-export=marisa-glib --pkg=marisa --namespace=Marisa
 Marisa_gir_INCLUDES = GLib-2.0 GObject-2.0 Gio-2.0
 Marisa_gir_CFLAGS = $(libmarisa_glib_la_CFLAGS)
 Marisa_gir_LIBS = libmarisa-glib.la
