--- origsrc/libhomfly-1.02r6/Makefile.am	2018-11-08 22:08:16.000000000 +0900
+++ src/libhomfly-1.02r6/Makefile.am	2025-05-15 15:32:06.313448400 +0900
@@ -4,3 +4,6 @@ dist-hook:
 	cp .travis.yml $(distdir)
 	cp LICENSE $(distdir)
 	cp README.md $(distdir)
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libhomfly-1.02r6/configure.ac	2018-11-08 22:08:16.000000000 +0900
+++ src/libhomfly-1.02r6/configure.ac	2025-05-15 15:42:57.769874200 +0900
@@ -1,4 +1,4 @@
-AC_INIT([libhomfly], [1.02r5], [mmarco@unizar.es])
+AC_INIT([libhomfly], [1.02r6], [mmarco@unizar.es])
 AC_CONFIG_AUX_DIR([build-aux])
 AC_CONFIG_MACRO_DIR([m4])
 AM_INIT_AUTOMAKE([foreign -Wall])
@@ -15,6 +15,6 @@ case $host_os in
         fi
     ;;
 esac
-AC_CONFIG_FILES([Makefile lib/Makefile test/Makefile])
+AC_CONFIG_FILES([libhomfly.pc Makefile lib/Makefile test/Makefile])
 AC_OUTPUT
 
--- origsrc/libhomfly-1.02r6/libhomfly.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libhomfly-1.02r6/libhomfly.pc.in	2025-05-15 15:32:06.313448400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libhomfly
+Description: Library to compute the homfly polynomial of a link
+Version: @VERSION@
+Libs: -L${libdir} -lhomfly
+Cflags: -I${includedir}
