--- origsrc/isomd5sum/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/isomd5sum/Makefile.am	2012-03-18 13:23:34.337145900 +0900
@@ -0,0 +1,24 @@
+
+bin_PROGRAMS = checkisomd5 implantisomd5
+
+checkisomd5_SOURCES = checkisomd5.c
+checkisomd5_LDADD = libcheckisomd5.la -lpopt
+
+implantisomd5_SOURCES = implantisomd5.c
+implantisomd5_LDADD = libimplantisomd5.la -lpopt
+
+
+lib_LTLIBRARIES = libcheckisomd5.la libimplantisomd5.la
+
+libcheckisomd5_la_SOURCES = libcheckisomd5.c md5.c
+libcheckisomd5_la_CPPFLAGS = -D_GNU_SOURCE=1 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE=1 -D_LARGEFILE64_SOURCE=1
+libcheckisomd5_la_LDFLAGS = -no-undefined
+
+libimplantisomd5_la_SOURCES = libimplantisomd5.c md5.c
+libimplantisomd5_la_CPPFLAGS = -D_GNU_SOURCE=1 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE=1 -D_LARGEFILE64_SOURCE=1
+libimplantisomd5_la_LDFLAGS = -no-undefined
+
+include_HEADERS = libcheckisomd5.h libimplantisomd5.h
+
+man_MANS = checkisomd5.1 implantisomd5.1
+
--- origsrc/isomd5sum/checkisomd5.c	2012-03-18 13:23:32.000000000 +0900
+++ src/isomd5sum/checkisomd5.c	2012-03-18 13:23:34.337145900 +0900
@@ -23,6 +23,7 @@
 #include <string.h>
 #include <popt.h>
 #include <termios.h>
+#include <sys/time.h>
 
 #include "md5.h"
 #include "libcheckisomd5.h"
--- origsrc/isomd5sum/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/isomd5sum/configure.ac	2012-03-18 13:24:14.602770900 +0900
@@ -0,0 +1,22 @@
+AC_PREREQ([2.68])
+AC_INIT([isomd5sum], [1.0.9], [anaconda-devel-list@redhat.com])
+AC_CONFIG_SRCDIR([checkisomd5.c])
+AC_CONFIG_HEADERS([config.h])
+
+AM_INIT_AUTOMAKE
+LT_INIT([disable-static])
+
+AC_PROG_CC
+AC_PROG_INSTALL
+
+AC_CHECK_LIB([popt], [poptGetContext], , AC_MSG_ERROR(Test for libpopt failed.))
+
+AC_CHECK_HEADERS([fcntl.h stdlib.h string.h termios.h unistd.h])
+
+AC_TYPE_OFF_T
+
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([memset select strtol])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
