--- origsrc/libident-0.32/Makefile.am	2004-01-19 02:28:37.000000000 +0900
+++ src/libident-0.32/Makefile.am	2014-05-18 16:29:36.517398200 +0900
@@ -5,6 +5,7 @@ man3_MANS = ident.3
 lib_LTLIBRARIES = libident.la
 libident_la_SOURCES = ident.c id_open.c id_close.c id_query.c id_parse.c \
 		     ident.h support.c
+libident_la_LDFLAGS = -no-undefined
 include_HEADERS = ident.h
 
 ACLOCAL_AMFLAGS = -I .
--- origsrc/libident-0.32/identtestd-xinetd	2003-08-11 17:07:42.000000000 +0900
+++ src/libident-0.32/identtestd-xinetd	2014-05-18 16:29:36.517398200 +0900
@@ -9,7 +9,7 @@ service identtestd
 	port		= 9999
 	protocol	= tcp
 	user		= nobody
-	server		= /usr/local/sbin/in.identtestd
+	server		= /usr/sbin/in.identtestd
 	wait		= no
 	disable		= yes
 }                                                                               
--- origsrc/libident-0.32/sockaddr.m4	2003-11-23 19:29:54.000000000 +0900
+++ src/libident-0.32/sockaddr.m4	2014-05-18 16:31:13.735171700 +0900
@@ -10,7 +10,7 @@
 # even the implied warranty of MERCHANTABILITY or FITNESS FOR A
 # PARTICULAR PURPOSE.
 
-AC_DEFUN(RDC_STRUCT_SOCKADDR_LEN,
+AC_DEFUN([RDC_STRUCT_SOCKADDR_LEN],
 [AH_TEMPLATE(HAVE_SA_LEN, [Define to 1 if `struct sockaddr' has a `sa_len' member.])
 AC_CACHE_CHECK([if struct sockaddr has a sa_len member],
 rdc_cv_struct_sockaddr_len,
@@ -27,7 +27,7 @@ if test $rdc_cv_struct_sockaddr_len = ye
 fi
 ]) 
 
-AC_DEFUN(RDC_STRUCT_SOCKADDR_STORAGE,
+AC_DEFUN([RDC_STRUCT_SOCKADDR_STORAGE],
 [AH_TEMPLATE(sockaddr_storage, [Define to `sockaddr' if <sys/socket.h> does not define.])
 AH_TEMPLATE(ss_family, [Define to `sa_family' if <sys/socket.h> does not define.])
 AC_CACHE_CHECK([for struct sockaddr_storage in sys/socket.h],
@@ -44,7 +44,7 @@ if test $rdc_cv_struct_sockaddr_storage
 fi 
 ])
 
-AC_DEFUN(RDC_TYPE_SOCKLEN_T,
+AC_DEFUN([RDC_TYPE_SOCKLEN_T],
 [AH_TEMPLATE(socklen_t, [Define to `int' if <sys/socket.h> does not define.])
 AC_CACHE_CHECK([for socklen_t in sys/socket.h],
 rdc_cv_type_socklen_t,
@@ -59,11 +59,11 @@ if test $rdc_cv_type_socklen_t = no; the
 fi
 ])
 
-AC_DEFUN(RDC_FUNC_SOCKET,
+AC_DEFUN([RDC_FUNC_SOCKET],
 [AC_SEARCH_LIBS(socket, [socket], $1, $2)
 ])
 
-AC_DEFUN(RDC_FUNC_GETADDRINFO,
+AC_DEFUN([RDC_FUNC_GETADDRINFO],
 [AC_CHECK_FUNCS([getaddrinfo getnameinfo gai_strerror], gai_support=yes,
 gai_support=no)
 AC_LIBSOURCES([getaddrinfo.h, getaddrinfo.c])dnl
