--- origsrc/libwebp/configure.ac	2011-02-20 15:55:19.000000000 +0900
+++ src/libwebp/configure.ac	2011-03-26 16:58:51.932467000 +0900
@@ -1,6 +1,6 @@
 AC_INIT([webpdecode], [0.1])
 AM_INIT_AUTOMAKE
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AM_PROG_CC_C_O
 
 dnl === check for PNG support ===
--- origsrc/libwebp/src/Makefile.am	2011-02-20 15:55:19.000000000 +0900
+++ src/libwebp/src/Makefile.am	2011-03-26 16:59:31.249001400 +0900
@@ -6,7 +6,7 @@ lib_LTLIBRARIES = libwebp.la
 libwebp_la_SOURCES =
 libwebp_la_LIBADD = dec/libwebpdecode.la \
                     enc/libwebpencode.la
-libwebp_la_LDFLAGS = -version-info 0:0:0
+libwebp_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libwebpinclude_HEADERS = webp/types.h webp/decode.h webp/decode_vp8.h \
                          webp/encode.h
 libwebpincludedir = $(includedir)/webp
--- origsrc/libwebp/src/dec/Makefile.am	2011-02-20 15:55:19.000000000 +0900
+++ src/libwebp/src/dec/Makefile.am	2011-03-26 17:14:28.420598700 +0900
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -I$(top_builddir)/src
+AM_CPPFLAGS = -I$(top_builddir)/src -I$(top_srcdir)/src
 
 libwebpdecode_la_SOURCES = bits.h vp8i.h yuv.h bits.c dsp.c frame.c \
                           quant.c tree.c vp8.c webp.c yuv.c
--- origsrc/libwebp/src/enc/Makefile.am	2011-02-20 15:55:19.000000000 +0900
+++ src/libwebp/src/enc/Makefile.am	2011-03-26 17:17:21.040730500 +0900
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -I$(top_builddir)/src
+AM_CPPFLAGS = -I$(top_builddir)/src -I$(top_srcdir)/src
 
 libwebpencode_la_SOURCES = analysis.c bit_writer.c bit_writer.h \
                           config.c cost.c cost.h dsp.c filter.c \
