--- origsrc/libnpa-0.1.1/CMakeLists.txt	2013-12-20 21:52:10.000000000 +0900
+++ src/libnpa-0.1.1/CMakeLists.txt	2013-12-22 21:16:59.541461900 +0900
@@ -14,4 +14,10 @@ ${CMAKE_SOURCE_DIR}/src/*
 )
 
 add_library(npa SHARED ${LIBNPA_SRC})
+target_link_libraries(npa ${Boost_LOCALE_LIBRARY})
 set_target_properties(npa PROPERTIES LINKER_LANGUAGE CXX)
+set_target_properties(npa PROPERTIES SOVERSION 0.1.1)
+
+file(GLOB HEADERS "include/*.hpp")
+install(FILES ${HEADERS} DESTINATION include/libnpa)
+install(TARGETS npa RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
