--- origsrc/libnpa-0.5.3/CMakeLists.txt	2018-05-24 21:13:47.000000000 +0900
+++ src/libnpa-0.5.3/CMakeLists.txt	2019-01-18 14:21:09.228046500 +0900
@@ -25,8 +25,10 @@ message(STATUS "Configuring libnpa versi
 # project name and language
 project (libnpa CXX)
 
-set(CMAKE_BUILD_TYPE Debug)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -O2")
+if(NOT CYGWIN)
+    set(CMAKE_BUILD_TYPE Debug)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -O2")
+endif()
 
 # include version number in header
 configure_file(${PROJECT_SOURCE_DIR}/include/npaversion.hpp.in
@@ -79,14 +81,24 @@ add_library(npa SHARED
     src/nsbconstants.cpp
 )
 target_link_libraries(npa ${Boost_LIBRARIES} ${ZLIB_LIBRARIES})
+if(CYGWIN)
+    set_target_properties(npa PROPERTIES SOVERSION 0)
+endif()
 
 # install headers and library
 install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/
     DESTINATION include/libnpa
     FILES_MATCHING PATTERN "*.hpp")
-install(TARGETS npa DESTINATION lib)
+install(TARGETS npa RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 
 # create packages
 set(CPACK_GENERATOR "TBZ2")
 set(CPACK_PACKAGE_VERSION ${LIBNPA_VERSION})
 include(CPack)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libnpa.pc.in ${CMAKE_BINARY_DIR}/libnpa.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libnpa.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libnpa-0.5.3/libnpa.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libnpa-0.5.3/libnpa.pc.in	2019-01-18 14:26:34.290336700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libnpa
+Description: Free library for manipulating Nitroplus game files
+Version: @LIBNPA_VERSION@
+Libs: -L${libdir} -lnpa
+Cflags: -I${includedir}
