--- origsrc/OpenHMD-0.3.0/examples/opengl/Makefile.am	2019-07-12 19:27:11.000000000 +0900
+++ src/OpenHMD-0.3.0/examples/opengl/Makefile.am	2019-08-15 21:59:52.690352500 +0900
@@ -1,5 +1,5 @@
 bin_PROGRAMS = openglexample
-AM_CPPFLAGS = -Wall -Werror -I$(top_srcdir)/include -DOHMD_STATIC $(sdl2_CFLAGS) $(GLEW_CFLAGS)
+AM_CPPFLAGS = -Wall -Werror -I$(top_srcdir)/include $(sdl2_CFLAGS) $(GLEW_CFLAGS)
 openglexample_SOURCES = gl.c main.c
 openglexample_LDADD = $(top_builddir)/src/libopenhmd.la -lm
-openglexample_LDFLAGS = -static-libtool-libs $(sdl2_LIBS) $(GLEW_LIBS) $(GL_LIBS)
+openglexample_LDFLAGS = $(sdl2_LIBS) $(GLEW_LIBS) $(GL_LIBS)
--- origsrc/OpenHMD-0.3.0/examples/simple/Makefile.am	2019-07-12 19:27:11.000000000 +0900
+++ src/OpenHMD-0.3.0/examples/simple/Makefile.am	2019-08-15 22:00:06.722458100 +0900
@@ -1,5 +1,5 @@
 bin_PROGRAMS = simple
-AM_CPPFLAGS = -Wall -I$(top_srcdir)/include -DOHMD_STATIC
+AM_CPPFLAGS = -Wall -I$(top_srcdir)/include
 simple_SOURCES = simple.c
 simple_LDADD = $(top_builddir)/src/libopenhmd.la -lm
-simple_LDFLAGS = -static-libtool-libs
+simple_LDFLAGS =
--- origsrc/OpenHMD-0.3.0/tests/unittests/Makefile.am	2019-07-12 19:27:11.000000000 +0900
+++ src/OpenHMD-0.3.0/tests/unittests/Makefile.am	2019-08-15 22:00:26.689223500 +0900
@@ -1,5 +1,5 @@
 bin_PROGRAMS = unittests
-AM_CPPFLAGS = -Wall -Werror -I$(top_srcdir)/include -I$(top_srcdir)/src -DOHMD_STATIC
+AM_CPPFLAGS = -Wall -Werror -I$(top_srcdir)/include -I$(top_srcdir)/src
 unittests_SOURCES = main.c quat.c vec.c highlevel.c
 unittests_LDADD = $(top_builddir)/src/libopenhmd.la -lm
-unittests_LDFLAGS = -static-libtool-libs
+unittests_LDFLAGS =
