--- origsrc/openhmd-0.1.0/examples/opengl/Makefile.am	2014-03-17 06:55:44.000000000 +0900
+++ src/openhmd-0.1.0/examples/opengl/Makefile.am	2015-03-11 21:43:33.773371200 +0900
@@ -1,5 +1,5 @@
 bin_PROGRAMS = openglexample
 AM_CPPFLAGS = -Wall -Werror -I$(top_srcdir)/include -DOHMD_STATIC $(sdl_CFLAGS) $(GLEW_CFLAGS)
 openglexample_SOURCES = gl.c main.c
-openglexample_LDADD = $(top_srcdir)/src/libopenhmd.la -lm 
-openglexample_LDFLAGS = -static-libtool-libs $(sdl_LIBS) $(GLEW_LIBS) $(GL_LIBS)
+openglexample_LDADD = $(top_builddir)/src/libopenhmd.la -lm 
+openglexample_LDFLAGS = $(sdl_LIBS) $(GLEW_LIBS) $(GL_LIBS)
--- origsrc/openhmd-0.1.0/examples/simple/Makefile.am	2014-03-17 06:55:44.000000000 +0900
+++ src/openhmd-0.1.0/examples/simple/Makefile.am	2015-03-11 21:43:17.484291000 +0900
@@ -1,5 +1,4 @@
 bin_PROGRAMS = simple
 AM_CPPFLAGS = -Wall -I$(top_srcdir)/include -DOHMD_STATIC
 simple_SOURCES = simple.c
-simple_LDADD = $(top_srcdir)/src/libopenhmd.la -lm
-simple_LDFLAGS = -static-libtool-libs
+simple_LDADD = $(top_builddir)/src/libopenhmd.la -lm
--- origsrc/openhmd-0.1.0/tests/unittests/Makefile.am	2014-03-17 06:55:44.000000000 +0900
+++ src/openhmd-0.1.0/tests/unittests/Makefile.am	2015-03-11 21:42:51.493718100 +0900
@@ -1,5 +1,4 @@
 bin_PROGRAMS = unittests
 AM_CPPFLAGS = -Wall -Werror -I$(top_srcdir)/include -I$(top_srcdir)/src -DOHMD_STATIC
 unittests_SOURCES = main.c quat.c vec.c
-unittests_LDADD = $(top_srcdir)/src/libopenhmd.la -lm
-unittests_LDFLAGS = -static-libtool-libs
+unittests_LDADD = $(top_builddir)/src/libopenhmd.la -lm
