--- origsrc/OpenHMD-0.2.0/examples/opengl/Makefile.am	2016-09-26 07:20:18.000000000 +0900
+++ src/OpenHMD-0.2.0/examples/opengl/Makefile.am	2017-11-30 08:53:55.881000000 +0900
@@ -1,5 +1,5 @@
 bin_PROGRAMS = openglexample
-AM_CPPFLAGS = -Wall -Werror -I$(top_srcdir)/include -DOHMD_STATIC $(sdl_CFLAGS) $(GLEW_CFLAGS)
+AM_CPPFLAGS = -Wall -Werror -I$(top_srcdir)/include $(sdl_CFLAGS) $(GLEW_CFLAGS)
 openglexample_SOURCES = gl.c main.c
 openglexample_LDADD = $(top_builddir)/src/libopenhmd.la -lm
-openglexample_LDFLAGS = -static-libtool-libs $(sdl_LIBS) $(GLEW_LIBS) $(GL_LIBS)
+openglexample_LDFLAGS = $(sdl_LIBS) $(GLEW_LIBS) $(GL_LIBS)
--- origsrc/OpenHMD-0.2.0/examples/simple/Makefile.am	2016-09-26 07:20:18.000000000 +0900
+++ src/OpenHMD-0.2.0/examples/simple/Makefile.am	2017-11-30 08:53:46.990000000 +0900
@@ -1,5 +1,5 @@
 bin_PROGRAMS = simple
-AM_CPPFLAGS = -Wall -I$(top_srcdir)/include -DOHMD_STATIC
+AM_CPPFLAGS = -Wall -I$(top_srcdir)/include
 simple_SOURCES = simple.c
 simple_LDADD = $(top_builddir)/src/libopenhmd.la -lm
-simple_LDFLAGS = -static-libtool-libs
+simple_LDFLAGS = 
--- origsrc/OpenHMD-0.2.0/tests/unittests/Makefile.am	2016-09-26 07:20:18.000000000 +0900
+++ src/OpenHMD-0.2.0/tests/unittests/Makefile.am	2017-11-30 08:54:12.202000000 +0900
@@ -1,5 +1,5 @@
 bin_PROGRAMS = unittests
-AM_CPPFLAGS = -Wall -Werror -I$(top_srcdir)/include -I$(top_srcdir)/src -DOHMD_STATIC
+AM_CPPFLAGS = -Wall -Werror -I$(top_srcdir)/include -I$(top_srcdir)/src
 unittests_SOURCES = main.c quat.c vec.c highlevel.c
 unittests_LDADD = $(top_builddir)/src/libopenhmd.la -lm
-unittests_LDFLAGS = -static-libtool-libs
+unittests_LDFLAGS =
