--- origsrc/libwhirlpool-1.1/Makefile.am	2021-08-28 02:11:56.000000000 +0900
+++ src/libwhirlpool-1.1/Makefile.am	2021-09-08 00:32:24.620662400 +0900
@@ -4,3 +4,5 @@ SUBDIRS = src
 
 EXTRA_DIST = config.h.in src/nessie.h patch/*
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libwhirlpool-1.1/configure.ac	2021-08-28 02:11:56.000000000 +0900
+++ src/libwhirlpool-1.1/configure.ac	2021-09-08 00:32:24.624665700 +0900
@@ -28,6 +28,6 @@ AC_CHECK_FUNCS([memmove memset strcasecm
 AM_INIT_AUTOMAKE
 LT_INIT
 
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([libwhirlpool.pc Makefile
                  src/Makefile])
 AC_OUTPUT
--- origsrc/libwhirlpool-1.1/libwhirlpool.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libwhirlpool-1.1/libwhirlpool.pc.in	2021-09-08 00:32:24.626667000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libwhirlpool
+Description: Whirlpool cryptographic hash function library
+Version: @VERSION@
+Libs: -L${libdir} -lwhirlpool
+Cflags: -I${includedir}
