--- origsrc/wfmath-0.3.10/configure.ac	2009-12-29 10:06:29.000000000 +0900
+++ src/wfmath-0.3.10/configure.ac	2010-12-30 19:23:26.896007400 +0900
@@ -22,8 +22,7 @@ AC_SUBST(WFMATH_VERSION_INFO)
 
 dnl Checks for programs.
 
-AC_DISABLE_STATIC
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_PROG_CXX
 
 AC_ARG_ENABLE(debug,
--- origsrc/wfmath-0.3.10/doc/Doxyfile.in	2009-01-07 09:43:51.000000000 +0900
+++ src/wfmath-0.3.10/doc/Doxyfile.in	2010-12-30 19:43:18.869981800 +0900
@@ -316,7 +316,7 @@ WARN_LOGFILE           =
 # directories like "/usr/src/myproject". Separate the files or directories 
 # with spaces.
 
-INPUT                  = ../wfmath shape.h
+INPUT                  = ../../src/wfmath-0.3.10/wfmath ../../src/wfmath-0.3.10/doc/shape.h
 
 # If the value of the INPUT tag contains directories, you can use the 
 # FILE_PATTERNS tag to specify one or more wildcard pattern (like *.cpp 
--- origsrc/wfmath-0.3.10/wfmath/Makefile.am	2009-01-13 00:52:54.000000000 +0900
+++ src/wfmath-0.3.10/wfmath/Makefile.am	2010-12-30 19:24:01.645975400 +0900
@@ -24,7 +24,7 @@ libwfmath_0_3_la_SOURCES = basis.h const
 	probability.cpp timestamp.cpp randgen.cpp rotmatrix.cpp vector.cpp \
 	point.cpp quaternion.cpp polygon.cpp intersect.cpp \
 	axisbox.cpp polygon_intersect.cpp
-libwfmath_0_3_la_LDFLAGS = -lm -version-info @WFMATH_VERSION_INFO@
+libwfmath_0_3_la_LDFLAGS = -lm -version-info @WFMATH_VERSION_INFO@ -no-undefined
 
 check_PROGRAMS = vector_test rotmatrix_test point_test \
 	quaternion_test shape_test polygon_test probability_test \
