--- origsrc/wfmath-1.0.2/doc/Doxyfile.in	2011-04-20 06:32:41.000000000 +0900
+++ src/wfmath-1.0.2/doc/Doxyfile.in	2014-03-16 13:33:57.717190700 +0900
@@ -574,7 +574,7 @@ WARN_LOGFILE           =
 # directories like "/usr/src/myproject". Separate the files or directories
 # with spaces.
 
-INPUT                  = ../wfmath \
+INPUT                  = @top_srcdir@/wfmath \
                          shape.h
 
 # This tag can be used to specify the character encoding of the source files
--- origsrc/wfmath-1.0.2/wfmath/Makefile.am	2013-04-10 05:30:09.000000000 +0900
+++ src/wfmath-1.0.2/wfmath/Makefile.am	2014-03-16 13:33:57.717190700 +0900
@@ -28,7 +28,7 @@ libwfmath_1_0_la_SOURCES = basis.h const
 	point.cpp quaternion.cpp polygon.cpp intersect.cpp \
 	axisbox.cpp polygon_intersect.cpp ball.cpp rotbox.cpp segment.cpp \
 	line.cpp
-libwfmath_1_0_la_LDFLAGS = -lm -version-info @WFMATH_VERSION_INFO@
+libwfmath_1_0_la_LDFLAGS = -lm -version-info @WFMATH_VERSION_INFO@ -no-undefined
 
 check_PROGRAMS = vector_test rotmatrix_test point_test \
 	quaternion_test shape_test polygon_test probability_test \
