--- origsrc/cre2-0.3.6/Makefile.am	2019-04-11 14:14:34.000000000 +0900
+++ src/cre2-0.3.6/Makefile.am	2025-12-07 12:44:53.078559800 +0900
@@ -23,15 +23,15 @@ include_HEADERS		= src/cre2.h
 lib_LTLIBRARIES		= libcre2.la
 libcre2_la_CFLAGS	= $(RE2_CFLAGS)
 libcre2_la_CXXFLAGS	= $(RE2_CFLAGS)
-libcre2_la_LDFLAGS	= -version-info $(cre2_CURRENT):$(cre2_REVISION):$(cre2_AGE) $(RE2_LIBS)
+libcre2_la_LDFLAGS	= -no-undefined -version-info $(cre2_CURRENT):$(cre2_REVISION):$(cre2_AGE) $(RE2_LIBS)
 libcre2_la_SOURCES	= src/cre2.cpp
 
 ## --------------------------------------------------------------------
 
 AM_MAKEINFOFLAGS	= --no-split
 
-info_TEXINFOS		= doc/cre2.texi
-doc_cre2_TEXINFOS	= doc/fdl-1.3.texi
+# info_TEXINFOS		= doc/cre2.texi
+# doc_cre2_TEXINFOS	= doc/fdl-1.3.texi
 
 ## --------------------------------------------------------------------
 
