--- origsrc/synfig-0.63.04/configure.ac	2012-04-02 16:16:34.493806800 +0900
+++ src/synfig-0.63.04/configure.ac	2012-04-02 22:25:13.365642600 +0900
@@ -160,11 +160,11 @@ AC_ARG_WITH(magickpp,[
 ])
 if test $with_magickpp = "yes" ; then {
   AC_CHECK_HEADER([Magick++.h],[
-  	MAGICKPP_LIBS="-lMagick++"
+  	MAGICKPP_LIBS=`Magick++-config --libs`
   	AC_DEFINE(HAVE_MAGICKPP,[], [ Define if Magick++ library is available ] )
   	AC_SUBST(MAGICKPP_LIBS)
   	AM_CONDITIONAL(HAVE_LIBMAGICKPP,true)
-	AC_CHECK_LIB(Magick,OptimizeImageTransparency,[
+	AC_CHECK_LIB(MagickCore,OptimizeImageTransparency,[
 		AC_DEFINE(HAVE_MAGICK_OPTIMIZE,[], [ Define if Magick++ defines OptimizeImageTransparency() ] )
 		AM_CONDITIONAL(HAVE_MAGICK_OPTIMIZE,true)
 	],[
@@ -549,7 +549,7 @@ AC_CHECK_HEADERS(signal.h termios.h sys/
 
 
 case "$host" in
-	*mingw* | *cygwin*)
+	*mingw*)
 		AM_CONDITIONAL(WIN32_PKG, true)
 		AM_CONDITIONAL(MACOSX_PKG, false)
 	;;
--- origsrc/synfig-0.63.04/src/synfig/color.h	2012-02-09 04:31:48.000000000 +0900
+++ src/synfig-0.63.04/src/synfig/color.h	2012-04-02 17:22:12.784822500 +0900
@@ -65,6 +65,10 @@ inline bool isnan(float x) { return x !=
 #define SYNFIG_ISNAN_FIX 1
 #endif
 
+#if defined(__CYGWIN__)
+#define isnan __builtin_isnan
+#endif
+
 namespace synfig {
 
 #ifdef USE_HALF_TYPE
--- origsrc/synfig-0.63.04/src/synfig/module.cpp	2012-02-09 04:31:48.000000000 +0900
+++ src/synfig-0.63.04/src/synfig/module.cpp	2012-04-02 22:49:11.517835300 +0900
@@ -127,6 +127,7 @@ synfig::Module::Register(const String &m
 
 	module=lt_dlopenext((string("lib")+module_name).c_str());
 	if(!module)module=lt_dlopenext(module_name.c_str());
+	if(!module)module=lt_dlopenext((string("cyg")+module_name).c_str());
 
 	if(!module)
 	{
--- origsrc/synfig-0.63.04/src/synfig/vector.h	2012-02-09 04:31:48.000000000 +0900
+++ src/synfig-0.63.04/src/synfig/vector.h	2012-04-02 16:27:48.714510000 +0900
@@ -54,6 +54,9 @@ inline bool isnan(float x) { return x !=
 #define SYNFIG_ISNAN_FIX 1
 #endif
 
+#if defined(__CYGWIN__)
+#define isnan __builtin_isnan
+#endif
 
 /* === T Y P E D E F S ===================================================== */
 
