--- origsrc/rationL-0.1.0/Makefile.am	2021-04-06 09:26:18.000000000 +0900
+++ src/rationL-0.1.0/Makefile.am	2022-12-04 16:13:24.965344000 +0900
@@ -1,6 +1,6 @@
 SUBDIRS = include . test docs
 EXTRA_DIST = docs test
-AM_CPPFLAGS = -I$(top_srcdir)/src/  -include $(top_srcdir)/include/config.h
+AM_CPPFLAGS = -I$(top_srcdir)/src/  -include $(top_builddir)/include/config.h
 AM_LDFLAGS = -shared -O3
 man3_MANS =	\
 	docs/man/regex_compile.man \
@@ -41,7 +41,7 @@ header_files = \
 	src/automaton/thompson.h
 
 librationl_la_SOURCES = $(source_files) $(header_files)
-
+librationl_la_LDFLAGS = -no-undefined
 
 if COVERAGE
 
@@ -75,3 +75,6 @@ coverage:
 	@echo "Generated coverage report succesfully"
 
 endif
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = rationl.pc
--- origsrc/rationL-0.1.0/configure.ac	2021-04-06 09:26:18.000000000 +0900
+++ src/rationL-0.1.0/configure.ac	2022-12-04 17:31:02.504132400 +0900
@@ -1,4 +1,4 @@
-AC_INIT([rationL], [v0.1.0], [simon.scatton@epita.fr])
+AC_INIT([rationL], [0.1.0], [simon.scatton@epita.fr])
 
 AC_CONFIG_SRCDIR([src/])
 
@@ -64,4 +64,5 @@ AH_TEMPLATE([DEBUG],
 AC_CONFIG_HEADERS([include/config.h])
 AC_CONFIG_FILES([Makefile include/Makefile test/Makefile docs/Makefile ])
 # Output the files
+AC_CONFIG_FILES([rationl.pc])
 AC_OUTPUT
--- origsrc/rationL-0.1.0/rationl.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/rationL-0.1.0/rationl.pc.in	2022-12-04 16:12:58.639473800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: rationL
+Description: Expressive regular expression library
+Version: @VERSION@
+Libs: -L${libdir} -lrationl
+Cflags: -I${includedir}
