--- origsrc/SampleICC-1.6.6/Contrib/ICC_utils/Makefile.am	2010-05-07 13:55:41.000000000 +0900
+++ src/SampleICC-1.6.6/Contrib/ICC_utils/Makefile.am	2012-04-14 17:57:41.232952100 +0900
@@ -9,7 +9,8 @@ libICC_utils_la_SOURCES = Stubs.cpp \
 	ICC_tool_exception.cpp \
 	Vetters.cpp
 
-libICC_utils_la_LDFLAGS = -version-info @LIBTOOL_VERSION@
+libICC_utils_la_LDFLAGS = -version-info @LIBTOOL_VERSION@ -no-undefined
+libICC_utils_la_LIBADD = $(top_builddir)/IccProfLib/libSampleICC.la
 
 libICC_utils_includedir = $(includedir)/SampleICC
 
@@ -22,6 +23,3 @@ libICC_utils_include_HEADERS = Stubs.h \
 
 INCLUDES = -I$(top_builddir) -I$(top_srcdir)/IccProfLib \
 	-I$(top_srcdir)/Contrib/ICC_utils
-
-LDADD = -L$(top_builddir)/IccProfLib -lSampleICC
-
--- origsrc/SampleICC-1.6.6/IccProfLib/Makefile.am	2010-08-24 18:53:09.000000000 +0900
+++ src/SampleICC-1.6.6/IccProfLib/Makefile.am	2012-04-14 17:30:44.487834900 +0900
@@ -23,7 +23,7 @@ libSampleICC_la_SOURCES = \
 	IccUtil.cpp \
 	md5.cpp
 
-libSampleICC_la_LDFLAGS = -version-info @LIBTOOL_VERSION@
+libSampleICC_la_LDFLAGS = -version-info @LIBTOOL_VERSION@ -no-undefined
 
 libSampleICCincludedir = $(includedir)/SampleICC
 
--- origsrc/SampleICC-1.6.6/Tools/CmdLine/Makefile.am	2011-09-01 18:24:14.000000000 +0900
+++ src/SampleICC-1.6.6/Tools/CmdLine/Makefile.am	2012-04-14 17:33:58.825725500 +0900
@@ -4,7 +4,6 @@ SUBDIRS = IccApplyNamedCmm \
 	@SICC_ICC_APPLY_PROFILES@ \
 	IccDumpProfile \
 	IccGamutMapGirdle \
-	IccGetBPCInfo \
 	IccRoundTrip \
 	IccProfLibTest \
 	IccStripUnknownTags \
@@ -13,7 +12,6 @@ SUBDIRS = IccApplyNamedCmm \
 DIST_SUBDIRS = IccApplyNamedCmm \
 	IccApplyProfiles \
 	IccDumpProfile \
-	IccGetBPCInfo \
 	IccGamutMapGirdle \
 	IccRoundTrip \
 	IccProfLibTest \
