--- origsrc/SampleICC-1.6.8/Contrib/ICC_utils/Makefile.am	2014-03-06 00:17:21.000000000 +0900
+++ src/SampleICC-1.6.8/Contrib/ICC_utils/Makefile.am	2014-06-28 11:00:03.931417500 +0900
@@ -9,7 +9,8 @@ libICC_utils_la_SOURCES = Stubs.cpp \
 	ICC_tool_exception.cpp \
 	Vetters.cpp
 
-libICC_utils_la_LDFLAGS = -version-info @LIBTOOL_VERSION@
+libICC_utils_la_LDFLAGS = -no-undefined -version-info @LIBTOOL_VERSION@
+libICC_utils_la_LIBADD = $(top_builddir)/IccProfLib/libSampleICC.la
 
 libICC_utils_includedir = $(includedir)/SampleICC
 
@@ -23,5 +24,3 @@ libICC_utils_include_HEADERS = Stubs.h \
 INCLUDES = -I$(top_builddir) -I$(top_srcdir)/IccProfLib \
 	-I$(top_srcdir)/Contrib/ICC_utils
 
-LDADD = -L$(top_builddir)/IccProfLib -lSampleICC
-
--- origsrc/SampleICC-1.6.8/IccProfLib/Makefile.am	2014-03-06 00:17:21.000000000 +0900
+++ src/SampleICC-1.6.8/IccProfLib/Makefile.am	2014-06-28 11:00:29.477315900 +0900
@@ -23,7 +23,7 @@ libSampleICC_la_SOURCES = \
 	IccUtil.cpp \
 	md5.cpp
 
-libSampleICC_la_LDFLAGS = -version-info @LIBTOOL_VERSION@
+libSampleICC_la_LDFLAGS = -no-undefined -version-info @LIBTOOL_VERSION@
 
 libSampleICCincludedir = $(includedir)/SampleICC
 
