--- origsrc/lux-0.9.1/Makefile.am	2010-02-03 15:31:07.000000000 +0900
+++ src/lux-0.9.1/Makefile.am	2014-11-08 19:38:02.628493200 +0900
@@ -5,11 +5,11 @@ CLEANFILES = lux/luxconfig.h
 EXTRA_DIR = example
 EXTRA_DIST = update_protobuf.sh
 
-lux/luxconfig.h: $(top_srcdir)/config.h
-	cp $(top_srcdir)/config.h $(top_srcdir)/lux/luxconfig.h
+lux/luxconfig.h: $(top_builddir)/config.h
+	cp $(top_builddir)/config.h $(top_srcdir)/lux/luxconfig.h
 
 update_protobuf:
-	sh update_protobuf.sh
+	( cd $(top_srcdir); sh update_protobuf.sh )
 
 dist-hook:
 	for subdir in $(EXTRA_DIR); do \
--- origsrc/lux-0.9.1/configure.ac	2010-02-03 16:06:32.000000000 +0900
+++ src/lux-0.9.1/configure.ac	2014-11-08 19:38:02.644118200 +0900
@@ -54,28 +54,30 @@ AC_CHECK_LIB(luxio, main, [],[echo "libl
 LUXIO_CFLAGS=""
 LUXIO_LIBS="-lluxio"
 
-AC_CHECK_LIB(mecab, mecab_new, [],[echo "libmecab is not found"; exit 1])
-MECAB_CFLAGS=""
-MECAB_LIBS="-lmecab -lstdc++"
+if ! mecab-config --version &> /dev/null ; then
+	AC_MSG_ERROR([Mecab is required.])
+else
+	MECAB_CFLAGS="`mecab-config --cflags`"
+	MECAB_LIBS="`mecab-config --libs`"
+fi
 #AC_CHECK_LIB(qdbm, dpopen, [],[echo "libqdbm is not found"; exit 1])
 #QDBM_CFLAGS=""
 #QDBM_LIBS="-lqdbm"
 
-AC_CHECK_LIB(protobuf, main, [],[echo "libprotobuf is not found"; exit 1])
-PROTOBUF_CFLAGS=""
-PROTOBUF_LIBS="-lprotobuf"
+PKG_CHECK_MODULES([PROTOBUF], [protobuf])
 
 # zlib
 AC_ARG_ENABLE(zlib,
   AC_HELP_STRING([--enable-zlib], [use zlib for data compression.]))
 if test "$enable_zlib" = "yes"
 then
-  AC_CHECK_LIB(z, main)
-  AC_DEFINE(USE_ZLIB, [1], [using zlib])
-  ZLIB_LIBS="-lz"
+  PKG_CHECK_MODULES([ZLIB], [zlib], [AC_DEFINE([USE_ZLIB], [1], [using zlib])])
 fi
 
-LUX_LIBS="$LUXIO_LIBS $MECAB_LIBS $PROTOBUF_LIBS $QDBM_LIBS $ZLIB_LIBS"
+BOOST_REQUIRE
+BOOST_SYSTEM
+
+LUX_LIBS="$LUXIO_LIBS $MECAB_LIBS $PROTOBUF_LIBS $QDBM_LIBS $ZLIB_LIBS $BOOST_SYSTEM_LIBS"
 AC_SUBST(LUX_LIBS)
 
 AC_OUTPUT(Makefile lux/Makefile)
--- origsrc/lux-0.9.1/lux/Makefile.am	2010-02-03 15:35:57.000000000 +0900
+++ src/lux-0.9.1/lux/Makefile.am	2014-11-08 19:38:02.644118200 +0900
@@ -1,5 +1,5 @@
 INCLUDES = -I. -I./document -I./lexer -I./query -I./storage \
-           -I./lang -I./index -I./api
+           -I./lang -I./index -I./api -I$(top_srcdir)
 
 lib_LTLIBRARIES = liblux.la
 
@@ -23,6 +23,9 @@ liblux_la_SOURCES = lexer/lexer.h lexer/
 					storage/luxio_data_unit.h storage/luxio_data_unit.cpp \
 					util.h scoped_ptr.h
 
+liblux_la_LDFLAGS = -no-undefined
+liblux_la_LIBADD = $(LUX_LIBS)
+
 liblux_includedir = $(includedir)/lux
 liblux_include_HEADERS = index.h search.h lux.h types.h luxconfig.h
 
--- origsrc/lux-0.9.1/lux/util.h	2009-02-17 13:49:12.000000000 +0900
+++ src/lux-0.9.1/lux/util.h	2014-11-08 19:38:02.644118200 +0900
@@ -13,6 +13,8 @@
 #include <iostream>
 #include <unistd.h>
 
+#include <cstdio>
+
 #ifdef USE_ZLIB
 #include <zlib.h>
 #endif
