--- origsrc/coin/CMakeLists.txt	2025-08-21 20:13:54.000000000 +0900
+++ src/coin/CMakeLists.txt	2025-08-21 20:14:58.894209900 +0900
@@ -636,6 +636,9 @@ check_include_file(time.h HAVE_TIME_H)
 check_include_files("stdlib.h;stdarg.h;string.h;float.h" STDC_HEADERS)
 
 check_include_file(windows.h HAVE_WINDOWS_H)
+if (CYGWIN)
+  set(HAVE_WINDOWS_H FALSE)
+endif()
 if(HAVE_WINDOWS_H)
   check_include_file(direct.h HAVE_DIRECT_H)
   check_include_files("windows.h;tlhelp32.h" HAVE_TLHELP32_H)
--- origsrc/coin/src/CMakeLists.txt	2025-08-21 20:13:56.000000000 +0900
+++ src/coin/src/CMakeLists.txt	2025-08-21 20:14:58.906142300 +0900
@@ -389,7 +389,7 @@ if(COIN_BUILD_MAC_FRAMEWORK)
   set(COIN_CONFIG_PACKAGE_LOCATION ${COIN_MAC_FRAMEWORK_NAME}.framework/Resources/CMake)
   string(TOLOWER ${COIN_MAC_FRAMEWORK_NAME} COIN_CONFIG_PACKAGE_BASENAME)
 else()
-  set(COIN_CONFIG_PACKAGE_LOCATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}-${PROJECT_VERSION})
+  set(COIN_CONFIG_PACKAGE_LOCATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
   string(TOLOWER ${PROJECT_NAME} COIN_CONFIG_PACKAGE_BASENAME)
   if(UNIX OR MINGW)
     configure_file("${PROJECT_SOURCE_DIR}/${PROJECT_NAME}.pc.cmake.in" "${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc" @ONLY)
--- origsrc/coin/src/io/SoInput.cpp	2025-08-21 20:13:57.000000000 +0900
+++ src/coin/src/io/SoInput.cpp	2025-08-21 20:14:58.909174600 +0900
@@ -1418,7 +1418,7 @@ SoInput::read(double & d)
   return TRUE;
 }
 
-#ifdef __CYGWIN__
+#if 0
 #include <boost/static_assert.hpp>
 
 SbBool
--- origsrc/coin/src/io/SoOutput.cpp	2025-08-21 20:13:57.000000000 +0900
+++ src/coin/src/io/SoOutput.cpp	2025-08-21 20:14:58.911184000 +0900
@@ -1002,7 +1002,7 @@ SoOutput::write(const double d)
   }
 }
 
-#ifdef __CYGWIN__
+#if 0
 #include <boost/static_assert.hpp>
 
 void
