--- origsrc/libbluray-0.2.2/src/Makefile.am	2012-03-16 01:02:31.000000000 +0900
+++ src/libbluray-0.2.2/src/Makefile.am	2012-03-31 22:19:49.762543800 +0900
@@ -88,7 +88,7 @@ libbluray_la_SOURCES=libbluray/bluray.h
 	util/bits.h \
 	util/logging.h
 libbluray_la_HEADERS= libbluray/bluray.h file/filesystem.h util/log_control.h libbluray/keys.h libbluray/decoders/overlay.h libbluray/bdnav/meta_data.h libbluray/bdnav/clpi_data.h libbluray/bluray-version.h
-libbluray_la_LDFLAGS= -version-info $(LIB_VERSION_INFO) $(DLOPEN_LDFLAGS)
+libbluray_la_LDFLAGS= -version-info $(LIB_VERSION_INFO) $(DLOPEN_LDFLAGS) -no-undefined
 libbluray_la_LIBADD= $(LIBXML2_LIBS)
 
 if USING_BDJAVA
