--- origsrc/libbluray-0.2.1/src/Makefile.am	2011-11-30 10:13:03.000000000 +0900
+++ src/libbluray-0.2.1/src/Makefile.am	2012-01-28 11:47:15.338191100 +0900
@@ -86,7 +86,7 @@ libbluray_la_SOURCES=libbluray/bluray.h
 	util/bits.h \
 	util/logging.h
 libbluray_la_HEADERS= libbluray/bluray.h file/filesystem.h util/log_control.h libbluray/keys.h libbluray/decoders/overlay.h libbluray/bdnav/meta_data.h libbluray/bdnav/clpi_data.h libbluray/bluray-version.h
-libbluray_la_LDFLAGS= -version-info $(LIB_VERSION_INFO) $(DLOPEN_LDFLAGS) $(LIBXML2_LIBS)
+libbluray_la_LDFLAGS= -version-info $(LIB_VERSION_INFO) $(DLOPEN_LDFLAGS) $(LIBXML2_LIBS) -no-undefined
 
 if USING_BDJAVA
 libbluray_la_SOURCES+=$(SOURCES_bdj)
