--- origsrc/unicorn-1.0.2-rc1/CMakeLists.txt	2019-10-08 11:44:50.000000000 +0900
+++ src/unicorn-1.0.2-rc1/CMakeLists.txt	2019-12-30 20:45:59.354303000 +0900
@@ -875,7 +875,7 @@ if(NOT MSVC)
     file(GLOB UNICORN_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/include/unicorn/*.h)
     install(TARGETS unicorn
         RUNTIME DESTINATION bin
-        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     )
     install(FILES ${UNICORN_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/unicorn)
     file(WRITE ${CMAKE_BINARY_DIR}/unicorn.pc "Name: unicorn\n\
