--- origsrc/unicorn-2.1.4/bindings/python/setup.py	2025-09-10 01:38:25.000000000 +0900
+++ src/unicorn-2.1.4/bindings/python/setup.py	2025-09-11 07:48:00.072715700 +0900
@@ -24,7 +24,10 @@ BUILD_DIR = os.path.join(UC_DIR, 'build_
 if sys.platform == 'darwin':
     LIBRARY_FILE = "libunicorn.2.dylib"
     STATIC_LIBRARY_FILE = "libunicorn.a"
-elif sys.platform in ('win32', 'cygwin'):
+elif sys.platform == 'cygwin':
+    LIBRARY_FILE = "cygunicorn-2.dll"
+    STATIC_LIBRARY_FILE = "libunicorn.a"
+elif sys.platform in ('win32'):
     LIBRARY_FILE = "unicorn.dll"
     STATIC_LIBRARY_FILE = "unicorn.lib"
 else:
--- origsrc/unicorn-2.1.4/qemu/configure	2025-09-10 01:38:25.000000000 +0900
+++ src/unicorn-2.1.4/qemu/configure	2025-09-11 07:48:00.087960500 +0900
@@ -416,6 +416,8 @@ elif check_define __NetBSD__; then
   targetos='NetBSD'
 elif check_define __APPLE__; then
   targetos='Darwin'
+elif check_define __CYGWIN__; then
+  targetos='Cygwin'
 else
   # This is a fatal error, but don't report it yet, because we
   # might be going to just print the --help text, or it might
