--- origsrc/stringencoders-v3.10.3/Makefile.am	2007-06-27 22:16:38.000000000 +0900
+++ src/stringencoders-v3.10.3/Makefile.am	2012-12-24 08:58:22.058262500 +0900
@@ -1,4 +1,6 @@
 
+AM_CPPFLAGS = -I$(top_srcdir)/src -I$(top_srcdir)/test
+
 SOURCES = \
 	src/modp_b2.c src/modp_b2.h \
 	src/modp_b16.c src/modp_b16.h modp_b16_data.h \
@@ -13,6 +15,8 @@ SOURCES = \
 lib_LTLIBRARIES = libmodpbase64.la
 libmodpbase64_la_SOURCES = ${SOURCES}
 
+libmodpbase64_la_LDFLAGS = -no-undefined
+
 libmodpbase64_la_DEPENDENCIES = \
 	modp_b2_data.h modp_b2_gen \
 	modp_b16_data.h modp_b16_gen \
@@ -47,7 +51,8 @@ src/modp_b16.c: src/modp_b16.h modp_b16_
 src/modp_b64.c: src/modp_b64.h modp_b64_data.h
 
 src/modp_b64w.c: src/modp_b64.c src/modp_b64w.h modp_b64w_data.h
-	perl -p -i -e 's/b64/b64w/g' < src/modp_b64.c > src/modp_b64w.c
+	mkdir -p src
+	perl -p -i -e 's/b64/b64w/g' < $(top_srcdir)/src/modp_b64.c > src/modp_b64w.c
 
 src/modp_b85.c: src/modp_b85.h modp_b85_data.h
 
@@ -57,31 +62,31 @@ src/modp_bjavascript.c: src/modp_bjavasc
 
 src/modp_ascii.c: src/modp_ascii.h modp_ascii_data.h
 
-modp_b2_data.h: modp_b2_gen
+modp_b2_data.h: modp_b2_gen$(EXEEXT)
 	./modp_b2_gen > modp_b2_data.h
 
-modp_b16_data.h: modp_b16_gen
+modp_b16_data.h: modp_b16_gen$(EXEEXT)
 	./modp_b16_gen > modp_b16_data.h
 
-modp_b64_data.h: modp_b64_gen
+modp_b64_data.h: modp_b64_gen$(EXEEXT)
 	./modp_b64_gen > modp_b64_data.h
 
-modp_ascii_data.h: modp_ascii_gen
+modp_ascii_data.h: modp_ascii_gen$(EXEEXT)
 	./modp_ascii_gen > modp_ascii_data.h
 
 #
 # Recall B64WCHARS by default is "-_."
 #
-modp_b64w_data.h: modp_b64_gen
+modp_b64w_data.h: modp_b64_gen$(EXEEXT)
 	./modp_b64_gen $(B64WCHARS) > modp_b64w_data.h
 
-modp_b85_data.h: modp_b85_gen
+modp_b85_data.h: modp_b85_gen$(EXEEXT)
 	./modp_b85_gen > modp_b85_data.h
 
-modp_burl_data.h: modp_burl_gen
+modp_burl_data.h: modp_burl_gen$(EXEEXT)
 	./modp_burl_gen > modp_burl_data.h 
 
-modp_bjavascript_data.h: modp_bjavascript_gen
+modp_bjavascript_data.h: modp_bjavascript_gen$(EXEEXT)
 	./modp_bjavascript_gen > modp_bjavascript_data.h 
 
 noinst_PROGRAMS = \
@@ -122,7 +127,8 @@ modp_b64_test_SOURCES = test/modp_b64_te
 modp_b64_test_LDADD = libmodpbase64.la
 
 test/modp_b64w_test.c: test/modp_b64_test.c
-	perl -p -i -e 's/b64/b64w/g' < test/modp_b64_test.c > test/modp_b64w_test.c
+	mkdir -p test
+	perl -p -i -e 's/b64/b64w/g' < $(top_srcdir)/test/modp_b64_test.c > test/modp_b64w_test.c
 
 modp_b64w_test_SOURCES = test/modp_b64w_test.c
 modp_b64w_test_LDADD = libmodpbase64.la
--- origsrc/stringencoders-v3.10.3/configure.ac	2010-04-06 06:37:39.000000000 +0900
+++ src/stringencoders-v3.10.3/configure.ac	2012-12-24 08:59:46.382735100 +0900
@@ -18,7 +18,7 @@ B64WCHARS="-_."
 AC_ARG_WITH([b64wchars], AC_HELP_STRING([--with-b64wchars=3CHARS],[change b64 web extra chars, default is '-_.']),
           [B64WCHARS=$withval], [])
 
-EXTRACFLAGS="-Isrc -Wall -Werror"
+EXTRACFLAGS="-Isrc -Wall"
 
 if test "x$enable_gcov" = "xyes";
 then
--- origsrc/stringencoders-v3.10.3/src/modp_numtoa.c	2010-03-20 00:09:46.000000000 +0900
+++ src/stringencoders-v3.10.3/src/modp_numtoa.c	2012-12-24 08:56:43.777012500 +0900
@@ -19,7 +19,7 @@
  * Powers of 10
  * 10^0 to 10^9
  */
-static const double pow10[] = {1, 10, 100, 1000, 10000, 100000, 1000000,
+static const double pow10_[] = {1, 10, 100, 1000, 10000, 100000, 1000000,
                                10000000, 100000000, 1000000000};
 
 static void strreverse(char* begin, char* end)
@@ -112,14 +112,14 @@ void modp_dtoa(double value, char* str,
 
 
     int whole = (int) value;
-    double tmp = (value - whole) * pow10[prec];
+    double tmp = (value - whole) * pow10_[prec];
     uint32_t frac = (uint32_t)(tmp);
     diff = tmp - frac;
 
     if (diff > 0.5) {
         ++frac;
         /* handle rollover, e.g.  case 0.99 with prec 1 is 1.0  */
-        if (frac >= pow10[prec]) {
+        if (frac >= pow10_[prec]) {
             frac = 0;
             ++whole;
         }
@@ -214,14 +214,14 @@ void modp_dtoa2(double value, char* str,
 
 
     int whole = (int) value;
-    double tmp = (value - whole) * pow10[prec];
+    double tmp = (value - whole) * pow10_[prec];
     uint32_t frac = (uint32_t)(tmp);
     diff = tmp - frac;
 
     if (diff > 0.5) {
         ++frac;
         /* handle rollover, e.g.  case 0.99 with prec 1 is 1.0  */
-        if (frac >= pow10[prec]) {
+        if (frac >= pow10_[prec]) {
             frac = 0;
             ++whole;
         }
