--- origsrc/libfreefare-0.4.0/libfreefare/Makefile.am	2013-01-21 01:26:17.000000000 +0900
+++ src/libfreefare-0.4.0/libfreefare/Makefile.am	2015-02-21 21:16:09.219226600 +0900
@@ -16,7 +16,8 @@ libfreefare_la_SOURCES = freefare.c \
 			 mad.c \
 			 mifare_application.c \
 			 tlv.c
-libfreefare_la_LIBADD =
+libfreefare_la_LIBADD = $(LIBNFC_LIBS)
+libfreefare_la_LDFLAGS = -no-undefined
 
 if WITH_DEBUG
   AM_CFLAGS += -DWITH_DEBUG
