--- origsrc/spglib-1.12.2/Makefile.am	2019-02-06 13:07:34.000000000 +0900
+++ src/spglib-1.12.2/Makefile.am	2019-05-05 13:18:07.639818100 +0900
@@ -9,3 +9,6 @@ EXTRA_DIST += \
 
 clean-local:
 	-rm -rf src/spglib_test.dSYM
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/spglib-1.12.2/configure.ac	2019-02-06 13:07:34.000000000 +0900
+++ src/spglib-1.12.2/configure.ac	2019-05-05 13:18:34.621059400 +0900
@@ -2,9 +2,8 @@
 # Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.61)
-#AC_INIT([spglib], 1.9.1, [atz.togo@gmail.com])
 AC_INIT([spglib],
-        m4_esyscmd_s(grep VERSION src/version.h|awk '{printf("%d.", $3)}'|sed s/.$//),
+        [1.12.2],
         [atz.togo@gmail.com])
 AM_INIT_AUTOMAKE
 AC_CONFIG_SRCDIR([src/spglib.c])
@@ -30,6 +29,6 @@ AC_C_CONST
 # Checks for library functions.
 AC_FUNC_MALLOC
 
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([spglib.pc Makefile
                  src/Makefile])
 AC_OUTPUT
--- origsrc/spglib-1.12.2/spglib.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/spglib-1.12.2/spglib.pc.in	2019-05-05 13:18:07.655476300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: spglib
+Description: C library for finding and handling crystal symmetries
+Version: @VERSION@
+Libs: -L${libdir} -lsymspg
+Cflags: -I${includedir}
--- origsrc/spglib-1.12.2/src/Makefile.am	2019-02-06 13:07:34.000000000 +0900
+++ src/spglib-1.12.2/src/Makefile.am	2019-05-05 13:18:07.749198600 +0900
@@ -49,7 +49,7 @@ version.h
 
 libsymspg_la_SOURCES = $(spglib_c) $(spglib_h)
 pkginclude_HEADERS = $(spglib_h)
-# libsymspg_la_LDFLAGS = -version-info 0:1:0
+libsymspg_la_LDFLAGS = -no-undefined -version-info 0:1:0
 libsymspg_la_LIBADD = -lm
 
 check_PROGRAMS = spglib_test
