--- origsrc/spglib-1.16.0/Makefile.am	2020-07-31 00:32:44.000000000 +0900
+++ src/spglib-1.16.0/Makefile.am	2020-08-13 16:00:47.941271700 +0900
@@ -9,3 +9,6 @@ EXTRA_DIST += \
 
 clean-local:
 	-rm -rf src/spglib_test.dSYM
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/spglib-1.16.0/configure.ac	2020-07-31 00:32:44.000000000 +0900
+++ src/spglib-1.16.0/configure.ac	2020-08-13 16:00:47.944274800 +0900
@@ -30,6 +30,6 @@ AC_C_CONST
 # Checks for library functions.
 AC_FUNC_MALLOC
 
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([spglib.pc Makefile
                  src/Makefile])
 AC_OUTPUT
--- origsrc/spglib-1.16.0/spglib.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/spglib-1.16.0/spglib.pc.in	2020-08-13 16:01:20.778279300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: spglib
+Description: C library for finding and handling crystal symmetries
+Version: @VERSION@
+Libs: -L${libdir} -lsymspg
+Cflags: -I${includedir}
--- origsrc/spglib-1.16.0/src/Makefile.am	2020-07-31 00:32:44.000000000 +0900
+++ src/spglib-1.16.0/src/Makefile.am	2020-08-13 15:50:18.957625800 +0900
@@ -49,7 +49,7 @@ version.h
 
 libsymspg_la_SOURCES = $(spglib_c) $(spglib_h)
 pkginclude_HEADERS = $(spglib_h)
-# libsymspg_la_LDFLAGS = -version-info 0:1:0
+libsymspg_la_LDFLAGS = -no-undefined -version-info 0:1:0
 libsymspg_la_LIBADD = -lm
 
 check_PROGRAMS = spglib_test
