--- origsrc/hopm-1.1.10/src/libopm/src/opm_gettime.c	2021-01-30 23:01:40.000000000 +0900
+++ src/hopm-1.1.10/src/libopm/src/opm_gettime.c	2021-02-05 16:51:11.062416900 +0900
@@ -28,7 +28,7 @@ opm_gettime(void)
 
 #ifdef CLOCK_MONOTONIC_RAW
   if (clock_gettime(CLOCK_MONOTONIC_RAW, &ts) == 0)
-#elif CLOCK_MONOTONIC
+#elif defined(CLOCK_MONOTONIC)
   if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0)
 #else
   if (clock_gettime(CLOCK_REALTIME, &ts) == 0)
