diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/handystats-1.11.3/CMakeLists.txt src/handystats-1.11.3/CMakeLists.txt
--- origsrc/handystats-1.11.3/CMakeLists.txt	2015-06-15 20:06:37.000000000 +0900
+++ src/handystats-1.11.3/CMakeLists.txt	2015-09-15 20:38:44.782851900 +0900
@@ -8,7 +8,7 @@ SET (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2
 
 GET_PROPERTY (LIB64 GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS)
 
-IF (LIB64)
+IF (LIB64 AND NOT CYGWIN)
 	SET (LIBSUFFIX 64)
 ELSE ()
 	SET (LIBSUFFIX "")
@@ -27,9 +27,9 @@ FILE (GLOB_RECURSE handy_src RELATIVE ${
 ADD_LIBRARY (${PROJECT_NAME} SHARED ${handy_src})
 
 TARGET_LINK_LIBRARIES (${PROJECT_NAME} rt)
+SET_TARGET_PROPERTIES(${PROJECT_NAME} PROPERTIES SOVERSION 0)
 
-
-INSTALL (TARGETS ${PROJECT_NAME} LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/${LIBDIR})
+INSTALL (TARGETS ${PROJECT_NAME} RUNTIME DESTINATION bin ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/${LIBDIR})
 INSTALL (DIRECTORY ${PROJECT_SOURCE_DIR}/include/${PROJECT_NAME} DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
 
 ADD_SUBDIRECTORY (benchmarks EXCLUDE_FROM_ALL)
@@ -37,3 +37,10 @@ ADD_SUBDIRECTORY (benchmarks EXCLUDE_FRO
 ENABLE_TESTING ()
 ADD_SUBDIRECTORY (tests)
 
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "1.1.3")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/${PROJECT_NAME}.pc.in ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/handystats-1.11.3/handystats.pc.in src/handystats-1.11.3/handystats.pc.in
--- origsrc/handystats-1.11.3/handystats.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/handystats-1.11.3/handystats.pc.in	2015-09-15 20:38:09.118985400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: handystats
+Description: C++ library for collecting user-defined in-process runtime statistics with low overhead
+Version: @VERSION@
+Libs: -L${libdir} -lhandystats
+Cflags: -I${includedir}
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/handystats-1.11.3/include/handystats/rapidjson/writer.h src/handystats-1.11.3/include/handystats/rapidjson/writer.h
--- origsrc/handystats-1.11.3/include/handystats/rapidjson/writer.h	2015-06-15 20:06:37.000000000 +0900
+++ src/handystats-1.11.3/include/handystats/rapidjson/writer.h	2015-09-15 20:31:39.296358600 +0900
@@ -164,7 +164,7 @@ protected:
 
 	//! \todo Optimization with custom double-to-string converter.
 	void WriteDouble(double d) {
-		if (isnan(d)) {
+		if (std::isnan(d)) {
 			d = 0;
 		}
 		char buffer[100];
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/handystats-1.11.3/src/core.cpp src/handystats-1.11.3/src/core.cpp
--- origsrc/handystats-1.11.3/src/core.cpp	2015-06-15 20:06:37.000000000 +0900
+++ src/handystats-1.11.3/src/core.cpp	2015-09-15 20:33:18.750302300 +0900
@@ -1,9 +1,12 @@
 // Copyright (c) 2014 Yandex LLC. All rights reserved.
 
+#include <cstdio>
 #include <chrono>
 #include <algorithm>
 #include <thread>
+#if !defined(__CYGWIN__)
 #include <sys/prctl.h>
+#endif
 #include <handystats/atomic.hpp>
 
 #include <handystats/chrono.hpp>
@@ -49,9 +52,9 @@ static void run_processor() {
 	memset(thread_name, 0, sizeof(thread_name));
 
 	sprintf(thread_name, "handystats");
-
+#if !defined(__CYGWIN__)
 	prctl(PR_SET_NAME, thread_name);
-
+#endif
 	while (is_enabled()) {
 		if (!message_queue::empty()) {
 			process_message_queue();
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/handystats-1.11.3/src/statistics.cpp src/handystats-1.11.3/src/statistics.cpp
--- origsrc/handystats-1.11.3/src/statistics.cpp	2015-06-15 20:06:37.000000000 +0900
+++ src/handystats-1.11.3/src/statistics.cpp	2015-09-15 20:35:48.536705200 +0900
@@ -11,6 +11,10 @@
 
 #include <handystats/statistics.hpp>
 
+#if defined(__CYGWIN__)
+#include <boost/lexical_cast.hpp>
+#endif
+
 // a x^2 + b x + c == 0
 // z -- root in [0, 1]
 static long double
@@ -24,7 +28,11 @@ find_z(const long double& a, const long
 			return roots[index];
 		}
 	}
+#if defined(__CYGWIN__)
+	throw std::logic_error("find_z: not found (" + boost::lexical_cast<std::string>(a) + ", " + boost::lexical_cast<std::string>(b) + ", " + boost::lexical_cast<std::string>(c) + ")");
+#else
 	throw std::logic_error("find_z: not found (" + std::to_string(a) + ", " + std::to_string(b) + ", " + std::to_string(c) + ")");
+#endif
 }
 
 namespace handystats {
