--- origsrc/libvalkey-0.5.0/CMakeLists.txt	2026-05-13 11:10:49.000000000 +0000
+++ src/libvalkey-0.5.0/CMakeLists.txt	2026-08-11 02:06:35.733841300 +0000
@@ -32,7 +32,7 @@ SET(CMAKE_DEBUG_POSTFIX d)
 
 # Set target-common flags
 if(NOT WIN32)
-  add_compile_options(-Werror -Wall -Wextra -pedantic)
+  add_compile_options(-Wall -Wextra -pedantic)
   add_compile_options(-Wstrict-prototypes -Wwrite-strings -Wno-missing-field-initializers)
 else()
   add_definitions(-D_CRT_SECURE_NO_WARNINGS -DWIN32_LEAN_AND_MEAN)
