--- origsrc/libvalkey-0.2.1/CMakeLists.txt	2025-08-20 23:30:56.000000000 +0900
+++ src/libvalkey-0.2.1/CMakeLists.txt	2026-01-22 17:18:46.239065600 +0900
@@ -30,7 +30,7 @@ SET(CMAKE_DEBUG_POSTFIX d)
 
 # Set target-common flags
 if(NOT WIN32)
-  add_compile_options(-Werror -Wall -Wextra -pedantic)
+  add_compile_options(-Wall -Wextra -pedantic)
   add_compile_options(-Wstrict-prototypes -Wwrite-strings -Wno-missing-field-initializers)
 else()
   add_definitions(-D_CRT_SECURE_NO_WARNINGS -DWIN32_LEAN_AND_MEAN)
