--- origsrc/libvalkey-0.4.0/CMakeLists.txt	2026-02-23 20:51:21.000000000 +0900
+++ src/libvalkey-0.4.0/CMakeLists.txt	2026-03-24 11:29:41.445271800 +0900
@@ -32,7 +32,7 @@ SET(CMAKE_DEBUG_POSTFIX d)
 
 # Set target-common flags
 if(NOT WIN32)
-  add_compile_options(-Werror -Wall -Wextra -pedantic)
+  add_compile_options(-Wall -Wextra -pedantic)
   add_compile_options(-Wstrict-prototypes -Wwrite-strings -Wno-missing-field-initializers)
 else()
   add_definitions(-D_CRT_SECURE_NO_WARNINGS -DWIN32_LEAN_AND_MEAN)
