--- origsrc/libgridxc-2.1.0/src/Makefile.am	2024-05-03 18:49:50.000000000 +0900
+++ src/libgridxc-2.1.0/src/Makefile.am	2024-09-01 10:06:12.775747800 +0900
@@ -103,7 +103,7 @@ gxc_f03_int_mods = \
 lib_LTLIBRARIES = libgridxc@PREC_SUFFIX@@ARCH_SUFFIX@.la
 
 libgridxc@PREC_SUFFIX@@ARCH_SUFFIX@_la_SOURCES = $(gxc_srcs)
-libgridxc@PREC_SUFFIX@@ARCH_SUFFIX@_la_LDFLAGS = -version-info 0:0:0
+libgridxc@PREC_SUFFIX@@ARCH_SUFFIX@_la_LDFLAGS = -no-undefined -version-info 0:0:0
 libgridxc@PREC_SUFFIX@@ARCH_SUFFIX@_la_LIBADD =
 if GRIDXC_USES_LIBXC
 libgridxc@PREC_SUFFIX@@ARCH_SUFFIX@_la_LIBADD += @LIBS_LIBXC@
