--- origsrc/opentracing-cpp-1.5.1/CMakeLists.txt	2019-01-17 05:28:05.000000000 +0900
+++ src/opentracing-cpp-1.5.1/CMakeLists.txt	2019-06-25 11:09:12.858838800 +0900
@@ -154,7 +154,7 @@ if (BUILD_SHARED_LIBS)
                                              SOVERSION ${OPENTRACING_VERSION_MAJOR})
   target_compile_definitions(opentracing PRIVATE OPENTRACING_EXPORTS)
   install(TARGETS opentracing EXPORT OpenTracingTargets
-          RUNTIME DESTINATION ${LIB_INSTALL_DIR}
+          RUNTIME DESTINATION bin
           LIBRARY DESTINATION ${LIB_INSTALL_DIR}
           ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
   if (CLANG_TIDY_EXE)
--- origsrc/opentracing-cpp-1.5.1/mocktracer/CMakeLists.txt	2019-01-17 05:28:05.000000000 +0900
+++ src/opentracing-cpp-1.5.1/mocktracer/CMakeLists.txt	2019-06-25 11:21:07.569244300 +0900
@@ -19,7 +19,7 @@ if (BUILD_SHARED_LIBS)
   target_link_libraries(opentracing_mocktracer PUBLIC opentracing)
   target_compile_definitions(opentracing_mocktracer PRIVATE OPENTRACING_MOCK_TRACER_EXPORTS)
   install(TARGETS opentracing_mocktracer EXPORT OpenTracingTargets
-          LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+          RUNTIME DESTINATION bin
           ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
 
   
