--- origsrc/opentracing-cpp-1.6.0/CMakeLists.txt	2019-09-19 06:59:25.000000000 +0900
+++ src/opentracing-cpp-1.6.0/CMakeLists.txt	2025-09-17 18:29:16.386114100 +0900
@@ -96,7 +96,7 @@ option(BUILD_MOCKTRACER "Build mocktrace
 option(BUILD_DYNAMIC_LOADING "Build with dynamic loading support" ON)
 
 if (BUILD_DYNAMIC_LOADING)
-  if (NOT WIN32)
+  if (NOT WIN32 AND NOT CYGWIN)
     if (NOT SUPPORTS_WEAK_SYMBOLS OR NOT UNIX)
       message(WARNING "Building without dynamic loading support.")
       set(BUILD_DYNAMIC_LOADING OFF)
@@ -165,7 +165,7 @@ if (BUILD_SHARED_LIBS)
   target_compile_definitions(opentracing PRIVATE OPENTRACING_EXPORTS)
   install(TARGETS opentracing EXPORT OpenTracingTargets
           COMPONENT DIST
-          RUNTIME DESTINATION ${LIB_INSTALL_DIR}
+          RUNTIME DESTINATION bin
           LIBRARY DESTINATION ${LIB_INSTALL_DIR}
           ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
           )
--- origsrc/opentracing-cpp-1.6.0/mocktracer/CMakeLists.txt	2019-09-19 06:59:25.000000000 +0900
+++ src/opentracing-cpp-1.6.0/mocktracer/CMakeLists.txt	2025-09-17 18:29:16.397069600 +0900
@@ -21,6 +21,7 @@ if (BUILD_SHARED_LIBS)
   install(TARGETS opentracing_mocktracer
           COMPONENT DIST
           EXPORT OpenTracingTargets
+          RUNTIME DESTINATION bin
           LIBRARY DESTINATION ${LIB_INSTALL_DIR}
           ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
 
