--- origsrc/opentracing-cpp-1.6.0/CMakeLists.txt	2019-09-19 06:59:25.000000000 +0900
+++ src/opentracing-cpp-1.6.0/CMakeLists.txt	2019-12-04 23:31:41.071077200 +0900
@@ -165,7 +165,7 @@ if (BUILD_SHARED_LIBS)
   target_compile_definitions(opentracing PRIVATE OPENTRACING_EXPORTS)
   install(TARGETS opentracing EXPORT OpenTracingTargets
           COMPONENT DIST
-          RUNTIME DESTINATION ${LIB_INSTALL_DIR}
+          RUNTIME DESTINATION bin
           LIBRARY DESTINATION ${LIB_INSTALL_DIR}
           ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
           )
--- origsrc/opentracing-cpp-1.6.0/mocktracer/CMakeLists.txt	2019-09-19 06:59:25.000000000 +0900
+++ src/opentracing-cpp-1.6.0/mocktracer/CMakeLists.txt	2019-12-04 23:32:17.683865100 +0900
@@ -21,6 +21,7 @@ if (BUILD_SHARED_LIBS)
   install(TARGETS opentracing_mocktracer
           COMPONENT DIST
           EXPORT OpenTracingTargets
+          RUNTIME DESTINATION bin
           LIBRARY DESTINATION ${LIB_INSTALL_DIR}
           ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
 
