--- origsrc/simdcomp-0.1.1/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/simdcomp-0.1.1/Makefile.am	2024-10-07 06:52:35.787904200 +0900
@@ -0,0 +1,27 @@
+AM_CPPFLAGS = -I$(top_srcdir)/include
+AM_CFLAGS = -march=native
+
+check_PROGRAMS = bitpackingbenchmark benchmark unit unit_chars 
+bitpackingbenchmark_LDADD = libsimdcomp.la
+bitpackingbenchmark_SOURCES = benchmarks/bitpackingbenchmark.c
+benchmark_LDADD = libsimdcomp.la
+benchmark_SOURCES = benchmarks/benchmark.c
+unit_LDADD = libsimdcomp.la
+unit_SOURCES = tests/unit.c
+unit_chars_LDADD = libsimdcomp.la
+unit_chars_SOURCES = tests/unit_chars.c
+TESTS = $(check_PROGRAMS)
+
+pkginclude_HEADERS = include/simdbitpacking.h include/simdcomputil.h \
+include/simdintegratedbitpacking.h include/simdcomp.h include/simdfor.h \
+include/avxbitpacking.h include/avx512bitpacking.h include/portability.h
+
+lib_LTLIBRARIES = libsimdcomp.la
+libsimdcomp_la_LDFLAGS = -no-undefined
+libsimdcomp_la_SOURCES = \
+src/avx512bitpacking.c  src/avxbitpacking.c     src/simdbitpacking.c \
+src/simdcomputil.c      src/simdfor.c           src/simdintegratedbitpacking.c \
+src/simdpackedsearch.c  src/simdpackedselect.c
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/simdcomp-0.1.1/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/simdcomp-0.1.1/configure.ac	2024-10-07 06:51:26.783811000 +0900
@@ -0,0 +1,33 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([simdcomp], [0.1.1])
+AC_CONFIG_SRCDIR([example.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_LN_S
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdint.h stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_INT8_T
+AC_TYPE_SIZE_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+AC_TYPE_UINT8_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([memmove memset])
+
+AC_CONFIG_FILES([simdcomp.pc Makefile])
+AC_OUTPUT
--- origsrc/simdcomp-0.1.1/simdcomp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/simdcomp-0.1.1/simdcomp.pc.in	2024-10-07 06:51:26.783811000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: simdcomp
+Description: C library for compressing lists of integers using binary packing
+Version: @VERSION@
+Libs: -L${libdir} -lsimdcomp
+Cflags: -I${includedir}/simdcomp
