--- origsrc/euler-1.61.0/docs/Makefile.am	2005-07-09 12:31:49.000000000 +0900
+++ src/euler-1.61.0/docs/Makefile.am	2012-08-05 10:29:58.255630600 +0900
@@ -3,7 +3,7 @@ SUBDIRS = french german images reference
 docsdir = $(prefix)/share/doc/euler
 
 docs_DATA = \
-	*.html\
-	*.css
+	$(srcdir)/*.html\
+	$(srcdir)/*.css
 
 EXTRA_DIST = $(docs_DATA)
--- origsrc/euler-1.61.0/docs/french/Makefile.am	2005-07-09 12:32:12.000000000 +0900
+++ src/euler-1.61.0/docs/french/Makefile.am	2012-08-05 10:27:25.549575900 +0900
@@ -1,6 +1,6 @@
 frenchdocdir = $(prefix)/share/doc/euler/french
 
 frenchdoc_DATA = \
-	*.html
+	$(srcdir)/*.html
 
 EXTRA_DIST = $(frenchdoc_DATA)
--- origsrc/euler-1.61.0/docs/german/Makefile.am	2005-07-09 12:32:27.000000000 +0900
+++ src/euler-1.61.0/docs/german/Makefile.am	2012-08-05 10:28:33.076919700 +0900
@@ -3,6 +3,6 @@ SUBDIRS = images
 germandocdir = $(prefix)/share/doc/euler/german
 
 germandoc_DATA = \
-	*.html
+	$(srcdir)/*.html
 
 EXTRA_DIST = $(germandoc_DATA)
--- origsrc/euler-1.61.0/docs/german/images/Makefile.am	2005-07-09 12:32:45.000000000 +0900
+++ src/euler-1.61.0/docs/german/images/Makefile.am	2012-08-05 10:28:07.247818100 +0900
@@ -1,8 +1,8 @@
 germanimagesdir = $(prefix)/share/doc/euler/german/images
 
 germanimages_DATA = \
-	*.gif\
-	*.php\
-	*.css
+	$(srcdir)/*.gif\
+	$(srcdir)/*.php\
+	$(srcdir)/*.css
 
 EXTRA_DIST = $(germanimages_DATA)
--- origsrc/euler-1.61.0/docs/images/Makefile.am	2005-07-09 12:33:01.000000000 +0900
+++ src/euler-1.61.0/docs/images/Makefile.am	2012-08-05 10:28:58.875747800 +0900
@@ -1,7 +1,7 @@
 imagesdir = $(prefix)/share/doc/euler/images
 
 images_DATA = \
-	*.gif\
-	*.php
+	$(srcdir)/*.gif\
+	$(srcdir)/*.php
 
 EXTRA_DIST = $(images_DATA)
--- origsrc/euler-1.61.0/docs/reference/Makefile.am	2005-07-09 12:33:26.000000000 +0900
+++ src/euler-1.61.0/docs/reference/Makefile.am	2012-08-05 10:29:32.256607200 +0900
@@ -1,8 +1,8 @@
 referencedir = $(prefix)/share/doc/euler/reference
 
 reference_DATA = \
-	*.gif\
-	*.html\
-	*.css
+	$(srcdir)/*.gif\
+	$(srcdir)/*.html\
+	$(srcdir)/*.css
 
 EXTRA_DIST = $(reference_DATA)
--- origsrc/euler-1.61.0/progs/Makefile.am	2005-07-09 11:58:16.000000000 +0900
+++ src/euler-1.61.0/progs/Makefile.am	2012-08-05 10:30:50.087661900 +0900
@@ -3,8 +3,8 @@ SUBDIRS = user
 progsdir = $(prefix)/share/euler/progs
 
 progs_DATA = \
-	*.e\
-	*.en\
-	*.dat
+	$(srcdir)/*.e\
+	$(srcdir)/*.en\
+	$(srcdir)/*.dat
 
 EXTRA_DIST = $(progs_DATA)
--- origsrc/euler-1.61.0/progs/user/Makefile.am	2005-07-09 11:55:20.000000000 +0900
+++ src/euler-1.61.0/progs/user/Makefile.am	2012-08-05 10:30:24.805435300 +0900
@@ -1,8 +1,8 @@
 userdir = $(prefix)/share/euler/progs/user
 
 user_DATA = \
-	*.e\
-	*.en\
-	*.txt
+	$(srcdir)/*.e\
+	$(srcdir)/*.en\
+	$(srcdir)/*.txt
 
 EXTRA_DIST = $(user_DATA)
