--- origsrc/sevmgr-1.00.12/config/FindBoost.cmake	2025-03-31 04:33:23.000000000 +0900
+++ src/sevmgr-1.00.12/config/FindBoost.cmake	2025-06-12 14:54:44.913979400 +0900
@@ -384,9 +384,11 @@ the Boost CMake package configuration fo
 Set ``Boost_NO_BOOST_CMAKE`` to ``ON``, to disable the search for boost-cmake.
 #]=======================================================================]
 
-cmake_policy(GET CMP0167 _FindBoost_CMP0167)
-if(_FindBoost_CMP0167 STREQUAL "NEW")
-  message(FATAL_ERROR "The FindBoost module has been removed by policy CMP0167.")
+if (POLICY CMP0167)
+  cmake_policy(GET CMP0167 _FindBoost_CMP0167)
+  if(_FindBoost_CMP0167 STREQUAL "NEW")
+    message(FATAL_ERROR "The FindBoost module has been removed by policy CMP0167.")
+  endif()
 endif()
 
 if(_FindBoost_testing)
@@ -402,7 +404,9 @@ include(${CMAKE_CURRENT_LIST_DIR}/FindPa
 cmake_policy(PUSH)
 cmake_policy(SET CMP0057 NEW) # if IN_LIST
 cmake_policy(SET CMP0102 NEW) # if mark_as_advanced(non_cache_var)
-cmake_policy(SET CMP0159 NEW) # file(STRINGS) with REGEX updates CMAKE_MATCH_<n>
+if (POLICY CMP0159)
+  cmake_policy(SET CMP0159 NEW) # file(STRINGS) with REGEX updates CMAKE_MATCH_<n>
+endif()
 
 function(_boost_get_existing_target component target_var)
   set(names "${component}")
--- origsrc/sevmgr-1.00.12/config/FindPython/Support.cmake	2025-03-31 04:33:23.000000000 +0900
+++ src/sevmgr-1.00.12/config/FindPython/Support.cmake	2025-06-12 14:55:03.613847100 +0900
@@ -20,8 +20,10 @@ cmake_policy (SET CMP0057 NEW)
 cmake_policy (SET CMP0124 NEW)
 # registry view behavior
 cmake_policy (SET CMP0134 NEW)
-# file(STRINGS) with REGEX updates CMAKE_MATCH_<n>
-cmake_policy(SET CMP0159 NEW)
+if (POLICY CMP0159)
+  # file(STRINGS) with REGEX updates CMAKE_MATCH_<n>
+  cmake_policy(SET CMP0159 NEW)
+endif()
 
 if (NOT DEFINED _PYTHON_PREFIX)
   message (FATAL_ERROR "FindPython: INTERNAL ERROR")
--- origsrc/sevmgr-1.00.12/sevmgr-config-python.cmake.in	2025-03-31 04:33:23.000000000 +0900
+++ src/sevmgr-1.00.12/sevmgr-config-python.cmake.in	2025-06-12 14:43:36.310747600 +0900
@@ -13,7 +13,7 @@ include ("@SEVMGR_CMAKE_DIR@/sevmgr-libr
 include ("@SEVMGR_CMAKE_DIR@/sevmgr-python-library-depends.cmake")
 
 # These are the SEvMgr IMPORTED targets, created by sevmgr-library-depends.cmake
-#set (SEVMGR_PY_LIBRARIES pysevmgrlib)
+#set (SEVMGR_PY_LIBRARIES pysevmgr)
 #set (SEVMGR_PY_EXECUTABLES pysevmgr)
 set (SEVMGR_PY_LIBRARIES)
 set (SEVMGR_PY_EXECUTABLES)
--- origsrc/sevmgr-1.00.12/sevmgr-config.cmake.in	2025-03-31 04:33:23.000000000 +0900
+++ src/sevmgr-1.00.12/sevmgr-config.cmake.in	2025-06-12 14:43:26.836396800 +0900
@@ -22,5 +22,5 @@ set (SEVMGR_LIBEXEC_DIR "@PACKAGE_SEVMGR
 include ("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@-library-depends.cmake")
 
 # These are the SEvMgr IMPORTED targets, created by sevmgr-library-depends.cmake
-set (SEVMGR_LIBRARIES sevmgrlib)
+set (SEVMGR_LIBRARIES sevmgr)
 set (SEVMGR_EXECUTABLES sevmgr sevmgr_demo)
