--- origsrc/sevmgr-1.00.2/config/project_config_embeddable.cmake	2015-06-14 21:30:00.000000000 +0900
+++ src/sevmgr-1.00.2/config/project_config_embeddable.cmake	2017-06-29 09:23:12.258391200 +0900
@@ -1466,7 +1466,7 @@ endmacro (module_set_name)
 macro (add_modules)
   set (_embedded_components ${ARGV})
   set (LIB_DEPENDENCY_EXPORT ${PROJECT_NAME}-library-depends)
-  set (LIB_DEPENDENCY_EXPORT_PATH "${INSTALL_DATA_DIR}/${PROJECT_NAME}/CMake")
+  set (LIB_DEPENDENCY_EXPORT_PATH "${INSTALL_LIB_DIR}/cmake/${PROJECT_NAME}")
   #
   foreach (_embedded_comp ${_embedded_components})
     #
@@ -1745,7 +1745,8 @@ macro (module_library_add_specific
 	# Install the library in the standard location
 	install (TARGETS ${_lib_target}
       EXPORT ${LIB_DEPENDENCY_EXPORT}
-      LIBRARY DESTINATION "${INSTALL_LIB_DIR}" COMPONENT runtime)
+      RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
+      ARCHIVE DESTINATION "${INSTALL_LIB_DIR}")
   endif ("${_lib_prefix}" STREQUAL "py")
   
   # Register, for reporting purpose, the list of libraries to be built
@@ -1942,7 +1943,7 @@ endmacro (module_script_add)
 # can refer to it (for libraries, header files and binaries)
 macro (module_export_install)
   install (EXPORT ${LIB_DEPENDENCY_EXPORT}
-    DESTINATION "${INSTALL_DATA_DIR}/${PACKAGE}/CMake" COMPONENT devel)
+  DESTINATION "${INSTALL_LIB_DIR}/cmake/${PACKAGE}" COMPONENT devel)
 endmacro (module_export_install)
 
 
--- origsrc/sevmgr-1.00.2/test/sevmgr/EventQueueManagementTestSuite.cpp	2015-06-14 20:58:09.000000000 +0900
+++ src/sevmgr-1.00.2/test/sevmgr/EventQueueManagementTestSuite.cpp	2017-06-29 09:31:17.555000000 +0900
@@ -43,7 +43,7 @@ struct UnitTestConfig {
   /** Constructor. */
   UnitTestConfig() {
     boost_utf::unit_test_log.set_stream (utfReportStream);
-    boost_utf::unit_test_log.set_format (boost_utf::XML);
+    boost_utf::unit_test_log.set_format (boost_utf::OF_XML);
     boost_utf::unit_test_log.set_threshold_level (boost_utf::log_test_units);
     //boost_utf::unit_test_log.set_threshold_level (boost_utf::log_successful_tests);
   }
