--- origsrc/libaec-v1.0.4/Makefile.am	2019-02-11 17:57:11.000000000 +0900
+++ src/libaec-v1.0.4/Makefile.am	2020-02-01 16:45:05.981397200 +0900
@@ -13,3 +13,6 @@ update-sampledata:
 	  cd $(sampledata)/AllOptions && mv -f test_P512n22.dat test_p512n22.dat
 
 .PHONY: update-sampledata
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libaec-v1.0.4/configure.ac	2019-02-11 17:57:11.000000000 +0900
+++ src/libaec-v1.0.4/configure.ac	2020-02-01 16:44:59.685788100 +0900
@@ -30,5 +30,5 @@ AC_CHECK_DECLS(__builtin_clzll)
 
 AM_EXTRA_RECURSIVE_TARGETS([bench benc bdec])
 
-AC_CONFIG_FILES([Makefile src/Makefile tests/Makefile])
+AC_CONFIG_FILES([libaec.pc Makefile src/Makefile tests/Makefile])
 AC_OUTPUT
--- origsrc/libaec-v1.0.4/libaec.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libaec-v1.0.4/libaec.pc.in	2020-02-01 16:45:12.897574300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libaec
+Description: Adaptive Entropy Coding library
+Version: @VERSION@
+Libs: -L${libdir} -laec
+Cflags: -I${includedir}
--- origsrc/libaec-v1.0.4/src/Makefile.am	2019-02-11 17:57:11.000000000 +0900
+++ src/libaec-v1.0.4/src/Makefile.am	2020-02-01 16:38:13.907618300 +0900
@@ -1,15 +1,11 @@
 AM_CFLAGS = $(CFLAG_VISIBILITY)
 AM_CPPFLAGS = -DBUILDING_LIBAEC
-lib_LTLIBRARIES = libaec.la libsz.la
+lib_LTLIBRARIES = libaec.la
 libaec_la_SOURCES = encode.c encode_accessors.c decode.c \
 encode.h encode_accessors.h decode.h
 libaec_la_LDFLAGS = -version-info 0:10:0 -no-undefined
 
-libsz_la_SOURCES = sz_compat.c
-libsz_la_LIBADD = libaec.la
-libsz_la_LDFLAGS = -version-info 2:1:0 -no-undefined
-
-include_HEADERS = libaec.h szlib.h
+include_HEADERS = libaec.h
 
 bin_PROGRAMS = aec
 noinst_PROGRAMS = utime
