--- origsrc/libdvbcsa-1.1.0/configure.ac	2011-04-03 04:33:02.000000000 +0900
+++ src/libdvbcsa-1.1.0/configure.ac	2011-05-11 00:21:13.386747400 +0900
@@ -19,8 +19,8 @@ AM_INIT_AUTOMAKE(libdvbcsa, 1.1.0)
 AM_CONFIG_HEADER(config.h)
 
 AC_PROG_CC
-
-AM_PROG_LIBTOOL 
+AM_PROG_CC_C_O
+LT_INIT(win32-dll)
 
 AC_LANG(C)
 
--- origsrc/libdvbcsa-1.1.0/src/Makefile.am	2011-02-11 01:10:02.000000000 +0900
+++ src/libdvbcsa-1.1.0/src/Makefile.am	2011-05-11 00:16:17.865663400 +0900
@@ -21,5 +21,5 @@ if TRANSPOSE_32
 libdvbcsa_la_SOURCES += dvbcsa_bs_transpose32.c
 endif
 
-libdvbcsa_la_LDFLAGS = -version-info 1:1:0 $(libtool_flags)
+libdvbcsa_la_LDFLAGS = -version-info 1:1:0 $(libtool_flags) -no-undefined
 
