--- origsrc/libmarquise-2.1.2/Makefile.am	2014-12-08 11:43:24.000000000 +0900
+++ src/libmarquise-2.1.2/Makefile.am	2024-11-21 11:48:54.794377800 +0900
@@ -4,3 +4,6 @@ docdir = ${datadir}/doc/${PACKAGE}
 dist_doc_DATA = README
 
 test: check
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libmarquise-2.1.2/configure.ac	2014-12-08 11:43:24.000000000 +0900
+++ src/libmarquise-2.1.2/configure.ac	2024-11-21 11:48:54.794377800 +0900
@@ -20,7 +20,7 @@ AC_CHECK_FUNCS([strerror])
 
 AC_CONFIG_HEADERS([config.h])
 AC_CHECK_HEADERS([stdlib.h string.h syslog.h])
-AC_CONFIG_FILES([
+AC_CONFIG_FILES([libmarquise.pc
  Makefile
  src/Makefile
  doc/man/Makefile
--- origsrc/libmarquise-2.1.2/libmarquise.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libmarquise-2.1.2/libmarquise.pc.in	2024-11-21 11:48:54.794377800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libmarquise
+Description: Library for queueing and sending datapoints to Vaultaire
+Version: @VERSION@
+Libs: -L${libdir} -lmarquise
+Cflags: -I${includedir}
--- origsrc/libmarquise-2.1.2/src/Makefile.am	2014-12-08 11:43:24.000000000 +0900
+++ src/libmarquise-2.1.2/src/Makefile.am	2024-11-21 11:48:54.794377800 +0900
@@ -2,7 +2,7 @@ AM_CFLAGS  = -Wall $(GLIB_2_CFLAGS)
 AM_LDFLAGS = $(GLIB_2_LIBS) 
 
 lib_LTLIBRARIES = libmarquise.la
-libmarquise_la_LDFLAGS = $(AM_LDFLAGS) -version-info 2:0:0
+libmarquise_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) -version-info 2:0:0
 libmarquise_la_SOURCES = marquise.c siphash24.c
 include_HEADERS = marquise.h
 dist_noinst_HEADERS = siphash24.h
