--- origsrc/libmarquise-2.0.6/src/Makefile.am	2014-10-01 13:27:35.000000000 +0900
+++ src/libmarquise-2.0.6/src/Makefile.am	2014-10-25 22:30:39.025960600 +0900
@@ -2,7 +2,7 @@ AM_CFLAGS  = -Wall $(GLIB_2_CFLAGS)
 AM_LDFLAGS = $(GLIB_2_LIBS) 
 
 lib_LTLIBRARIES = libmarquise.la
-libmarquise_la_LDFLAGS = $(AM_LDFLAGS) -version-info 2:0:0
+libmarquise_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) -version-info 2:0:0
 libmarquise_la_SOURCES = marquise.c siphash24.c
 include_HEADERS = marquise.h
 dist_noinst_HEADERS = siphash24.h
