--- origsrc/sitecing/lib/Makefile.am	2015-06-15 22:36:49.000000000 +0900
+++ src/sitecing/lib/Makefile.am	2015-06-15 22:36:51.864403000 +0900
@@ -19,4 +19,5 @@ libsitecing_la_SOURCES = \
 	scoreboard.cc \
 	process_manager.cc
 libsitecing_la_LDFLAGS = \
-	-version-info 2:0:0
+	-version-info 2:0:0 -no-undefined
+libsitecing_la_LIBADD = ${DOTCONF_LIBS} ${PCREPP_LIBS} ${KONFORKA_LIBS} ${KINGATE_LIBS}
--- origsrc/sitecing/lib/sitecing_util.cc	2015-06-15 22:36:49.000000000 +0900
+++ src/sitecing/lib/sitecing_util.cc	2015-06-15 22:36:51.864403000 +0900
@@ -11,6 +11,7 @@
  #include <iostream>
  #include <fstream>
  #include <cassert>
+ #include <cstdlib>
  #include "sitecing/sitecing_util.h"
 #endif
 
--- origsrc/sitecing/lib/util.cc	2015-06-15 22:36:49.000000000 +0900
+++ src/sitecing/lib/util.cc	2015-06-15 22:36:51.881079800 +0900
@@ -2,6 +2,7 @@
  #include "pch.h"
 #else
  #include <cassert>
+ #include <cstring>
  #include "sitecing/util.h"
 #endif
 
--- origsrc/sitecing/src/Makefile.am	2015-06-15 22:36:49.000000000 +0900
+++ src/sitecing/src/Makefile.am	2015-06-17 21:10:31.052797100 +0900
@@ -15,12 +15,12 @@ LIBS += ${top_builddir}/lib/libsitecing.
 sitecing_fastcgi_SOURCES = sitecing-fastcgi.cc \
     COPYING.cc
 sitecing_fastcgi_LDADD = ${KINGATE_FCGI_LIBS}
-sitecing_fastcgi_LDFLAGS = -rdynamic
+sitecing_fastcgi_LDFLAGS =
 sitecing_fastcgi_DEPENDENCIES = ${top_builddir}/lib/libsitecing.la
 
 sitecing_plaincgi_SOURCES = sitecing-plaincgi.cc \
     COPYING.cc
-sitecing_plaincgi_LDFLAGS = -rdynamic
+sitecing_plaincgi_LDFLAGS =
 sitecing_plaincgi_LDADD = ${KINGATE_PLAINCGI_LIBS}
 sitecing_plaincgi_DEPENDENCIES = ${top_builddir}/lib/libsitecing.la
 
--- origsrc/sitecing/src/sitecing-build.cc	2015-06-15 22:36:49.000000000 +0900
+++ src/sitecing/src/sitecing-build.cc	2015-06-15 22:36:51.881079800 +0900
@@ -5,6 +5,7 @@
 #include <memory>
 #include <fstream>
 #include <cassert>
+#include <cstring>
 #include <set>
 using namespace std;
 #include <konforka/util.h>
--- origsrc/sitecing/src/sitecing-fastcgi.cc	2015-06-15 22:36:49.000000000 +0900
+++ src/sitecing/src/sitecing-fastcgi.cc	2015-06-15 22:36:51.881079800 +0900
@@ -9,6 +9,7 @@
 #include <iostream>
 #include <memory>
 #include <typeinfo>
+#include <cstring>
 using namespace std;
 #include "kingate/fastcgi.h"
 #include "kingate/cgi_gateway.h"
--- origsrc/sitecing/src/sitecing-plaincgi.cc	2015-06-15 22:36:49.000000000 +0900
+++ src/sitecing/src/sitecing-plaincgi.cc	2015-06-15 22:36:51.881079800 +0900
@@ -9,6 +9,7 @@
 #include <iostream>
 #include <memory>
 #include <typeinfo>
+#include <cstring>
 using namespace std;
 #include "kingate/plaincgi.h"
 #include "kingate/cgi_gateway.h"
