--- origsrc/croissant/CMakeLists.txt	2018-06-07 18:26:48.000000000 +0900
+++ src/croissant/CMakeLists.txt	2018-06-07 18:30:47.907823500 +0900
@@ -16,15 +16,17 @@ enable_testing()
 #-----------------------------------------------------------------------
 # Retrieve the current version number
 
-execute_process(COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/version.sh
-    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-    RESULT_VARIABLE VERSION_RESULT
-    OUTPUT_VARIABLE VERSION
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-if(VERSION_RESULT)
-    message(FATAL_ERROR
-            "Cannot determine version number: " ${VERSION_RESULT})
-endif(VERSION_RESULT)
+if(NOT DEFINED VERSION)
+    execute_process(COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/version.sh
+        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+        RESULT_VARIABLE VERSION_RESULT
+        OUTPUT_VARIABLE VERSION
+        OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if(VERSION_RESULT)
+        message(FATAL_ERROR
+                "Cannot determine version number: " ${VERSION_RESULT})
+    endif(VERSION_RESULT)
+endif()
 # This causes an annoying extra prompt in ccmake.
 # message("Current version: " ${VERSION})
 
@@ -87,13 +89,15 @@ endif(NOT CMAKE_BUILD_TYPE)
 set(CMAKE_INSTALL_LIBDIR lib CACHE STRING
     "The base name of the installation directory for libraries")
 
-if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
-    add_definitions(-Wall -Werror)
-elseif(CMAKE_C_COMPILER_ID STREQUAL "Clang")
-    add_definitions(-Wall -Werror)
-elseif(CMAKE_C_COMPILER_ID STREQUAL "Intel")
-    add_definitions(-Wall -Werror)
-endif(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+if(NOT CYGWIN)
+    if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+        add_definitions(-Wall -Werror)
+    elseif(CMAKE_C_COMPILER_ID STREQUAL "Clang")
+        add_definitions(-Wall -Werror)
+    elseif(CMAKE_C_COMPILER_ID STREQUAL "Intel")
+        add_definitions(-Wall -Werror)
+    endif(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+endif()
 
 #-----------------------------------------------------------------------
 # Include our subdirectories
--- origsrc/croissant/src/CMakeLists.txt	2018-06-07 18:26:49.000000000 +0900
+++ src/croissant/src/CMakeLists.txt	2018-06-07 18:33:28.243294000 +0900
@@ -62,7 +62,7 @@ target_link_libraries(libcroissant
     ${CLOG_LIBRARIES}
 )
 
-install(TARGETS libcroissant DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS libcroissant RUNTIME DESTINATION bin ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 #-----------------------------------------------------------------------
 # Generate the pkg-config file
--- origsrc/croissant/src/croissant/test.c	2018-06-07 18:26:49.000000000 +0900
+++ src/croissant/src/croissant/test.c	2018-06-07 18:30:56.657756500 +0900
@@ -8705,11 +8705,7 @@ case 654:
     /* A hack to suppress some unused variable warnings */
     (void) crs_command_en_main;
 
-    if (CORK_UNLIKELY(cs < 
-#line 8710 "croissant/test.c"
-655
-#line 221 "croissant/test.c.rl"
-)) {
+    if (CORK_UNLIKELY(cs < 655)) {
         crs_parse_error("Invalid command");
         goto error;
     }
