--- origsrc/libcpuid-0.8.0/libcpuid/Makefile.am	2025-05-04 19:39:32.000000000 +0900
+++ src/libcpuid-0.8.0/libcpuid/Makefile.am	2025-06-23 23:53:10.098970600 +0900
@@ -46,7 +46,7 @@ noinst_HEADERS = 		\
 EXTRA_DIST += libcpuid.sym libcpuid_vc71.vcproj libcpuid_vc10.vcxproj libcpuid_vc10.vcxproj.filters
 
 if HAVE_DOXYGEN
-directory = $(top_srcdir)/libcpuid/docs/man/man3
+directory = $(top_builddir)/libcpuid/docs/man/man3
 
 dist_man_MANS = $(directory)/cpu_id_t.3		\
 	$(directory)/cpu_list_t.3		\
