--- origsrc/libcpuid-0.6.3/libcpuid/Makefile.am	2023-04-02 18:50:02.000000000 +0900
+++ src/libcpuid-0.6.3/libcpuid/Makefile.am	2023-04-03 14:25:55.479423700 +0900
@@ -41,7 +41,7 @@ noinst_HEADERS = 		\
 EXTRA_DIST += libcpuid.sym libcpuid_vc71.vcproj libcpuid_vc10.vcxproj libcpuid_vc10.vcxproj.filters
 
 if HAVE_DOXYGEN
-directory = $(top_srcdir)/libcpuid/docs/man/man3
+directory = $(top_builddir)/libcpuid/docs/man/man3
 
 dist_man_MANS = $(directory)/cpu_id_t.3		\
 	$(directory)/cpu_list_t.3		\
--- origsrc/libcpuid-0.6.3/libcpuid/libcpuid_internal.h	2023-04-02 18:50:02.000000000 +0900
+++ src/libcpuid-0.6.3/libcpuid/libcpuid_internal.h	2023-04-03 14:32:49.480063200 +0900
@@ -103,6 +103,16 @@ struct internal_cache_instances_t {
 	struct internal_cache_id_t htable[NUM_CACHE_TYPES][CACHES_HTABLE_SIZE];
 };
 
+#if defined(_N)
+#undef _N
+#endif
+#if defined(_P)
+#undef _P
+#endif
+#if defined(_X)
+#undef _X
+#endif
+
 #define LBIT(x) (((long long) 1) << x)
 
 enum _common_bits_t {
