--- origsrc/libcpuid-0.7.1/libcpuid/Makefile.am	2024-12-01 00:00:13.000000000 +0900
+++ src/libcpuid-0.7.1/libcpuid/Makefile.am	2025-01-05 14:55:12.536541700 +0900
@@ -50,7 +50,7 @@ noinst_HEADERS = 		\
 EXTRA_DIST += libcpuid.sym libcpuid_vc71.vcproj libcpuid_vc10.vcxproj libcpuid_vc10.vcxproj.filters
 
 if HAVE_DOXYGEN
-directory = $(top_srcdir)/libcpuid/docs/man/man3
+directory = $(top_builddir)/libcpuid/docs/man/man3
 
 dist_man_MANS = $(directory)/cpu_id_t.3		\
 	$(directory)/cpu_list_t.3		\
--- origsrc/libcpuid-0.7.1/libcpuid/libcpuid_internal.h	2024-12-01 00:00:13.000000000 +0900
+++ src/libcpuid-0.7.1/libcpuid/libcpuid_internal.h	2025-01-05 14:55:12.536541700 +0900
@@ -122,6 +122,25 @@ struct internal_type_info_array_t {
 	struct internal_type_info_t* data;
 };
 
+#if defined(_C)
+#undef _C
+#endif
+#if defined(_N)
+#undef _N
+#endif
+#if defined(_P)
+#undef _P
+#endif
+#if defined(_S)
+#undef _S
+#endif
+#if defined(_U)
+#undef _U
+#endif
+#if defined(_X)
+#undef _X
+#endif
+
 #define LBIT(x) (((long long) 1) << x)
 
 enum _common_bits_t {
