--- origsrc/libcpuid-0.8.1/libcpuid/Makefile.am	2025-08-20 02:19:24.000000000 +0900
+++ src/libcpuid-0.8.1/libcpuid/Makefile.am	2025-09-21 04:32:41.242850300 +0900
@@ -46,7 +46,7 @@ noinst_HEADERS = 		\
 EXTRA_DIST += libcpuid.sym libcpuid_vc71.vcproj libcpuid_vc10.vcxproj libcpuid_vc10.vcxproj.filters
 
 if HAVE_DOXYGEN
-directory = $(top_srcdir)/libcpuid/docs/man/man3
+directory = $(top_builddir)/libcpuid/docs/man/man3
 
 dist_man_MANS = $(directory)/cpu_id_t.3		\
 	$(directory)/cpu_list_t.3		\
