--- origsrc/cpp-pinyin-1.0.2/CMakeLists.txt	2025-03-14 18:15:54.000000000 +0900
+++ src/cpp-pinyin-1.0.2/CMakeLists.txt	2026-05-30 09:13:14.830852300 +0900
@@ -56,6 +56,10 @@ target_compile_definitions(${PROJECT_NAM
 target_sources(${PROJECT_NAME} PRIVATE ${_src})
 add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
 
+if (DEFINED SOVERSION)
+    set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION ${SOVERSION})
+endif ()
+
 if (CPP_PINYIN_BUILD_TESTS)
     add_subdirectory(tests)
 
--- origsrc/cpp-pinyin-1.0.2/cpp-pinyin.pc.in	2025-03-14 18:15:54.000000000 +0900
+++ src/cpp-pinyin-1.0.2/cpp-pinyin.pc.in	2026-05-30 09:26:18.339101700 +0900
@@ -1,4 +1,4 @@
-prefix=${pcfiledir}/../..
+prefix=@CMAKE_INSTALL_PREFIX@
 
 # cpp-pinyin pkg-config file
 
