--- origsrc/ncnn-20231027/CMakeLists.txt	2023-10-27 12:50:52.000000000 +0900
+++ src/ncnn-20231027/CMakeLists.txt	2023-12-29 18:18:44.932942200 +0900
@@ -637,3 +637,10 @@ endif()
 if(NCNN_PYTHON)
     add_subdirectory(python)
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/ncnn.pc.in ${CMAKE_BINARY_DIR}/ncnn.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/ncnn.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/ncnn-20231027/ncnn.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/ncnn-20231027/ncnn.pc.in	2023-12-29 18:18:44.932942200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: ncnn
+Description: High-performance neural network inference framework optimized for the mobile platform
+Version: @VERSION@
+Libs: -L${libdir} -lncnn
+Cflags: -I${includedir}
