--- origsrc/ncnn-20210525/CMakeLists.txt	2021-05-26 00:39:26.000000000 +0900
+++ src/ncnn-20210525/CMakeLists.txt	2021-06-11 16:31:47.280545800 +0900
@@ -273,3 +273,10 @@ endif()
 if(NCNN_PYTHON)
     add_subdirectory(python)
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/ncnn.pc.in ${CMAKE_BINARY_DIR}/ncnn.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/ncnn.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/ncnn-20210525/ncnn.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/ncnn-20210525/ncnn.pc.in	2021-06-11 16:31:47.283545100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: ncnn
+Description: High-performance neural network inference framework optimized for the mobile platform
+Version: @VERSION@
+Libs: -L${libdir} -lncnn
+Cflags: -I${includedir}
