--- origsrc/ncnn-20260113/CMakeLists.txt	2026-01-13 11:57:43.000000000 +0900
+++ src/ncnn-20260113/CMakeLists.txt	2026-01-26 18:29:00.812983600 +0900
@@ -890,3 +890,10 @@ endif()
 if(NCNN_PYTHON)
     add_subdirectory(python)
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/ncnn.pc.in ${CMAKE_BINARY_DIR}/ncnn.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/ncnn.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/ncnn-20260113/ncnn.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/ncnn-20260113/ncnn.pc.in	2026-01-26 18:29:00.818789100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: ncnn
+Description: High-performance neural network inference framework optimized for the mobile platform
+Version: @VERSION@
+Libs: -L${libdir} -lncnn
+Cflags: -I${includedir}
