--- origsrc/cronutils-1.2/Makefile	2012-05-21 13:22:48.000000000 +0900
+++ src/cronutils-1.2/Makefile	2012-12-01 18:31:05.403170100 +0900
@@ -20,12 +20,12 @@ runlock: runlock.c subprocess.c tempdir.
 
 runstat: runstat.c subprocess.c tempdir.c
 
-CFLAGS=-Wall -Werror -Wextra -D_XOPEN_SOURCE=500 -g -O0 -ansi -pedantic-errors -Wwrite-strings -Wcast-align -Wcast-qual -Winit-self -Wformat=2 -Wuninitialized -Wmissing-declarations -Wpointer-arith -Wstrict-aliasing -fstrict-aliasing
+CFLAGS=-Wall -Werror -Wextra -D_XOPEN_SOURCE=500 -g -O0 -pedantic-errors -Wwrite-strings -Wcast-align -Wcast-qual -Winit-self -Wformat=2 -Wuninitialized -Wmissing-declarations -Wpointer-arith -Wstrict-aliasing -fstrict-aliasing
 LDLIBS=-lrt
 
 SOURCES = runalarm.c runlock.c runstat.c subprocess.c subprocess.h tempdir.c tempdir.h Makefile runalarm.1 runlock.1 runstat.1 version
 
-prefix = usr/local
+prefix = usr
 BINDIR = $(prefix)/bin
 MANDIR = $(prefix)/share/man/man1
 VERSION = $(shell cat version)
