--- origsrc/cronutils-version-1.9/Makefile	2015-12-30 13:21:21.000000000 +0900
+++ src/cronutils-version-1.9/Makefile	2018-03-27 15:52:50.855744900 +0900
@@ -20,12 +20,12 @@ runlock: runlock.c subprocess.c tempdir.
 
 runstat: runstat.c subprocess.c tempdir.c
 
-CFLAGS+=-Wall -Werror -Wextra -D_XOPEN_SOURCE=500 -g -ansi -pedantic-errors -Wwrite-strings -Wcast-align -Wcast-qual -Winit-self -Wformat=2 -Wuninitialized -Wmissing-declarations -Wpointer-arith -Wstrict-aliasing -fstrict-aliasing
+CFLAGS+=-Wall -Werror -Wextra -D_XOPEN_SOURCE=500 -g -pedantic-errors -Wwrite-strings -Wcast-align -Wcast-qual -Winit-self -Wformat=2 -Wuninitialized -Wmissing-declarations -Wpointer-arith -Wstrict-aliasing -fstrict-aliasing
 LDLIBS+=-lrt
 
 SOURCES = runalarm.c runlock.c runstat.c subprocess.c subprocess.h tempdir.c tempdir.h Makefile runalarm.1 runlock.1 runstat.1 version examples cronutils.spec runcron regtest.sh tests
 
-prefix = usr/local
+prefix = usr
 BINDIR = $(prefix)/bin
 MANDIR = $(prefix)/share/man/man1
 VERSION = $(shell cat version)
