--- origsrc/lasso-2.3.5/bindings/java/Makefile.am	2011-01-05 22:57:32.000000000 +0900
+++ src/lasso-2.3.5/bindings/java/Makefile.am	2011-04-03 20:04:29.134478600 +0900
@@ -55,7 +55,7 @@ com_entrouvert_lasso_LassoJNI.h: com/ent
 libjnilasso_la_CFLAGS = -fno-strict-aliasing $(LASSO_CORE_CFLAGS) -I$(top_srcdir) -I$(top_builddir) $(AM_CFLAGS) @JAVA_INCLUDE@
 libjnilasso_la_CFLAGS += -Wno-unused-parameter
 
-libjnilasso_la_LDFLAGS = -export-dynamic -prefer-pic -module -avoid-version
+libjnilasso_la_LDFLAGS = -export-dynamic -prefer-pic -module -avoid-version -no-undefined
 libjnilasso_la_LIBADD = $(top_builddir)/lasso/liblasso.la $(LASSO_LIBS)
 nodist_libjnilasso_la_SOURCES = com_entrouvert_lasso_LassoJNI.c 
 
--- origsrc/lasso-2.3.5/bindings/php5/Makefile.am	2011-01-05 22:57:32.000000000 +0900
+++ src/lasso-2.3.5/bindings/php5/Makefile.am	2011-04-03 20:04:53.369326600 +0900
@@ -18,7 +18,7 @@ php_config_DATA = lasso.ini
 
 lasso_la_CFLAGS = -fno-strict-aliasing $(LASSO_CORE_CFLAGS) -I$(top_srcdir) -I$(top_builddir) $(PHP5_INCLUDES) $(AM_CFLAGS)
 lasso_la_CFLAGS += -Wno-unused-parameter
-lasso_la_LDFLAGS = -export-dynamic -prefer-pic -module -avoid-version
+lasso_la_LDFLAGS = -export-dynamic -prefer-pic -module -avoid-version -no-undefined
 lasso_la_LIBADD = $(top_builddir)/lasso/liblasso.la $(LASSO_LIBS) $(PHP5_LDFLAGS)
 nodist_lasso_la_SOURCES = _lasso.c
 
--- origsrc/lasso-2.3.5/lasso/Makefile.am	2011-01-05 22:57:32.000000000 +0900
+++ src/lasso-2.3.5/lasso/Makefile.am	2011-04-03 22:33:28.065187000 +0900
@@ -81,8 +81,7 @@ liblasso_la_LIBADD = \
 # Just make damn sure the ABI stays the same between
 # upgrades.
 # Obviously if you don't, fixing the situation up again can be painful. 
-liblasso_la_LDFLAGS = -no-undefined -version-info @LASSO_VERSION_INFO@  \
-	-export-symbols symbols.sym
+liblasso_la_LDFLAGS = -no-undefined -version-info @LASSO_VERSION_INFO@
 endif
 
 $(srcdir)/errors.c: $(srcdir)/errors.h $(srcdir)/build_strerror.py
--- origsrc/lasso-2.3.5/lasso/id-wsf/data_service.c	2011-01-05 22:57:32.000000000 +0900
+++ src/lasso-2.3.5/lasso/id-wsf/data_service.c	2011-04-03 21:51:53.568969000 +0900
@@ -502,7 +502,6 @@ lasso_data_service_get_answers_by_item_i
 	int rc = 0;
 
 	lasso_bad_param(DATA_SERVICE, service);
-	lasso_null_param(select);
 	wsf_profile = &service->parent;
 	lasso_extract_node_or_fail(query_response, wsf_profile->request, DST_QUERY_RESPONSE,
 			LASSO_PROFILE_ERROR_MISSING_REQUEST);
--- origsrc/lasso-2.3.5/tests/Makefile.am	2011-01-05 22:57:32.000000000 +0900
+++ src/lasso-2.3.5/tests/Makefile.am	2011-04-03 20:43:47.343964500 +0900
@@ -26,7 +26,7 @@ tests_LDADD = \
 	$(top_builddir)/lasso/liblasso.la \
 	$(LASSO_LIBS) \
 	$(CHECK_LIBS)
-tests_LDFLAGS = -rpath `cd $(top_builddir)/lasso/.libs/; pwd` -static
+tests_LDFLAGS = -rpath `cd $(top_builddir)/lasso/.libs/; pwd`
 
 tests2_SOURCES = tests2.c
 tests2_LDADD = \
