--- origsrc/lasso-2.8.2/bindings/python/Makefile.am	2022-04-27 21:54:06.114460000 +0900
+++ src/lasso-2.8.2/bindings/python/Makefile.am	2023-12-03 10:39:27.930097600 +0900
@@ -15,7 +15,7 @@ python_LTLIBRARIES = _lasso.la
 
 nodist__lasso_la_SOURCES = _lasso.c
 _lasso_la_CFLAGS = -fno-strict-aliasing $(LASSO_CORE_CFLAGS) $(PY_CFLAGS) $(AM_CFLAGS)
-_lasso_la_LIBADD = $(top_builddir)/lasso/liblasso.la $(LASSO_LIBS)
+_lasso_la_LIBADD = $(top_builddir)/lasso/liblasso.la $(LASSO_LIBS) -lpython$(PYTHON_VERSION)
 if DARWIN
 _lasso_la_LDFLAGS = -no-undefined -module -avoid-version -Wl,-F. -Wl,-F. -bundle -framework Python 
 else
--- origsrc/lasso-2.8.2/tests/Makefile.am	2023-01-26 02:10:30.770769200 +0900
+++ src/lasso-2.8.2/tests/Makefile.am	2023-12-03 10:39:27.930097600 +0900
@@ -23,19 +23,16 @@ tests_LDADD = \
 	$(top_builddir)/lasso/liblasso.la \
 	$(LASSO_LIBS) \
 	$(CHECK_LIBS)
-tests_LDFLAGS = -rpath `cd $(top_builddir)/lasso/.libs/; pwd` -static
 
 tests2_SOURCES = tests2.c
 tests2_LDADD = \
 	$(top_builddir)/lasso/liblasso.la \
 	$(LASSO_LIBS) \
 	$(CHECK_LIBS)
-tests2_LDFLAGS = -rpath `cd $(top_builddir)/lasso/.libs/; pwd`
 tests2_CFLAGS = $(AM_CFLAGS) -DSRCDIR=\"$(srcdir)\"
 
 perfs_SOURCES = perfs.c
 perfs_LDADD = $(top_builddir)/lasso/liblasso.la $(LASSO_LIBS)
-perfs_LDFLAGS = -rpath `cd $(top_builddir)/lasso/.libs/; pwd`
 
 .PHONY: valgrind
 
