--- origsrc/lasem-0.3.1/src/Makefile.am	2011-10-10 01:44:25.000000000 +0900
+++ src/lasem-0.3.1/src/Makefile.am	2012-04-01 18:29:26.211108000 +0900
@@ -5,7 +5,7 @@ include $(top_srcdir)/Makefile.decl
   INTROSPECTION_SCANNER_ARGS = --add-include-path=$(srcdir)
   INTROSPECTION_COMPILER_ARGS = --includedir=$(srcdir)
 
-INCLUDES = \
+INCLUDES = -I$(srcdir) \
 	-DPACKAGE_LOCALE_DIR=\""$(prefix)/$(DATADIRNAME)/locale"\" \
 	-DPACKAGE_SRC_DIR=\""$(srcdir)"\" \
 	-DPACKAGE_DATA_DIR=\""$(datadir)"\" $(LASEM_CFLAGS) \
@@ -241,7 +241,7 @@ liblasem_@LASEM_API_VERSION@_la_SOURCES
 liblasem_@LASEM_API_VERSION@_la_HEADERS = $(LASEM_DOM_HDRS)
 liblasem_@LASEM_API_VERSION@_la_HEADERS += lsmdomenumtypes.h
 
-liblasem_@LASEM_API_VERSION@_la_LDFLAGS = -version-info $(LASEM_LIBTOOL_VERSION)
+liblasem_@LASEM_API_VERSION@_la_LDFLAGS = -version-info $(LASEM_LIBTOOL_VERSION) -no-undefined
 
 lsmdomenumtypes.h: lsmdomenumtypes.h.template $(LASEM_DOM_HDRS) $(GLIB_MKENUMS)
 	$(AM_V_GEN) (cd $(srcdir) && $(GLIB_MKENUMS) --template lsmdomenumtypes.h.template $(LASEM_DOM_HDRS)) > $@
