--- origsrc/libnemesi-0.7.0/Makefile.am	2011-02-15 22:51:50.000000000 +0900
+++ src/libnemesi-0.7.0/Makefile.am	2015-05-08 14:25:33.989790700 +0900
@@ -122,6 +122,6 @@ endif
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = pkgconfig/libnemesi.pc
 
-libnemesi_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
+libnemesi_la_LDFLAGS = -no-undefined -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 
 EXTRA_DIST = Doxyfile CodingStyle
--- origsrc/libnemesi-0.7.0/m4/lscube.m4	2011-02-15 22:51:50.000000000 +0900
+++ src/libnemesi-0.7.0/m4/lscube.m4	2015-05-08 14:26:16.187841700 +0900
@@ -86,7 +86,7 @@ AC_DEFUN([LSC_TESTS], [
        PKG_CHECK_MODULES([GTESTER], [glib-2.0 >= 2.20], [], [have_tests=no])
        AC_CHECK_PROGS([GAWK], [gawk])
        AS_IF([test "x$GAWK" = "x"], [have_tests=no])
-       AC_CHECK_PROGS([EXUBERANT_CTAGS], [exuberant-ctags])
+       AC_CHECK_PROGS([EXUBERANT_CTAGS], [ctags])
        AS_IF([test "x$EXUBERANT_CTAGS" = "x"], [have_tests=no])
       ])
     AC_SUBST([GAWK])
--- origsrc/libnemesi-0.7.0/src/rtp/rtp_thread.c	2011-02-15 22:51:50.000000000 +0900
+++ src/libnemesi-0.7.0/src/rtp/rtp_thread.c	2015-05-08 14:29:08.926349900 +0900
@@ -28,6 +28,7 @@
 #include "config.h"
 
 #include <unistd.h>
+#include <sys/select.h>
 
 #include "nemesi/rtp.h"
 #include "nemesi/comm.h"
@@ -186,6 +187,8 @@ static void *rtp(void *args)
     }
 
     pthread_cleanup_pop(1);
+
+    return NULL;
 }
 
 /**
