--- origsrc/rgbds-0.8.0/CMakeLists.txt	2024-06-29 03:53:17.000000000 +0900
+++ src/rgbds-0.8.0/CMakeLists.txt	2024-09-07 14:07:50.924680100 +0900
@@ -75,15 +75,15 @@ endif()
 # the git revision is used but uses the fallback in an archive
 
 find_program(GIT git)
-if(GIT)
+if(GIT AND EXISTS "${PROJECT_SOURCE_DIR}/.git")
   execute_process(COMMAND ${GIT} --git-dir=.git describe --tags --dirty --always
                   WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
                   OUTPUT_VARIABLE GIT_REV OUTPUT_STRIP_TRAILING_WHITESPACE
                   ERROR_QUIET)
   message(STATUS "RGBDS version: ${GIT_REV}")
-else(GIT)
+else(GIT AND EXISTS "${PROJECT_SOURCE_DIR}/.git")
   message(STATUS "Cannot determine RGBDS version (Git not installed), falling back")
-endif(GIT)
+endif(GIT AND EXISTS "${PROJECT_SOURCE_DIR}/.git")
 
 find_package(PkgConfig)
 if(MSVC OR NOT PKG_CONFIG_FOUND)
--- origsrc/rgbds-0.8.0/test/CMakeLists.txt	2024-06-29 03:53:17.000000000 +0900
+++ src/rgbds-0.8.0/test/CMakeLists.txt	2024-09-07 14:09:00.101955200 +0900
@@ -5,12 +5,12 @@ add_executable(rgbgfx_test gfx/rgbgfx_te
 add_executable(unmangle link/unmangle.cpp)
 
 install(TARGETS randtilegen rgbgfx_test
-        DESTINATION ${rgbds_SOURCE_DIR}/test/gfx
+        DESTINATION share/rgbds
         COMPONENT "Test support programs"
         EXCLUDE_FROM_ALL
         )
 install(TARGETS unmangle
-        DESTINATION ${rgbds_SOURCE_DIR}/test/link
+        DESTINATION share/rgbds
         COMPONENT "Test support programs"
         EXCLUDE_FROM_ALL
         )
--- origsrc/rgbds-0.8.0/test/gfx/rgbgfx_test.cpp	2024-06-29 03:53:17.000000000 +0900
+++ src/rgbds-0.8.0/test/gfx/rgbgfx_test.cpp	2024-09-07 14:10:07.142567500 +0900
@@ -307,6 +307,7 @@ static char *execProg(char const *name,
 		return strerror(err);
 	}
 
+#if defined(__GLIBC__)
 	siginfo_t info;
 	if (waitid(P_PID, pid, &info, WEXITED) != 0) {
 		fatal("Error waiting for %s: %s", name, strerror(errno));
@@ -327,6 +328,10 @@ static char *execProg(char const *name,
 		    formatArgv()
 		);
 	}
+#else
+	int wstatus;
+	waitpid(pid, &wstatus, 0);
+#endif
 
 #else // defined(_MSC_VER) || defined(__MINGW32__)
 
