--- origsrc/c-rbtree-3/src/meson.build	2017-08-14 03:27:01.000000000 +0900
+++ src/c-rbtree-3/src/meson.build	2017-09-10 12:21:57.215000000 +0900
@@ -59,8 +59,10 @@ test('Basic API Behavior', test_basic)
 test_map = executable('test-map', ['test-map.c'], link_with: libcrbtree_private)
 test('Generic Map', test_map)
 
-test_parallel = executable('test-parallel', ['test-parallel.c'], link_with: libcrbtree_private)
-test('Lockless Parallel Readers', test_parallel)
+if build_machine.system() != 'cygwin'
+        test_parallel = executable('test-parallel', ['test-parallel.c'], link_with: libcrbtree_private)
+        test('Lockless Parallel Readers', test_parallel)
+endif
 
 test_posix = executable('test-posix', ['test-posix.c'], link_with: libcrbtree_private)
 test('Posix tsearch(3p) Comparison', test_posix)
