--- origsrc/libvsapm-20240226/Makefile.am	2024-02-26 15:28:54.000000000 +0900
+++ src/libvsapm-20240226/Makefile.am	2024-03-02 13:03:30.001438800 +0900
@@ -3,22 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
 	libvsapm \
 	vsapmtools \
 	pyvsapm \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libvsapm-20240226/configure.ac	2024-02-26 15:28:54.000000000 +0900
+++ src/libvsapm-20240226/configure.ac	2024-03-02 13:03:43.954571100 +0900
@@ -157,18 +157,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
 AC_CONFIG_FILES([libvsapm/Makefile])
 AC_CONFIG_FILES([pyvsapm/Makefile])
 AC_CONFIG_FILES([vsapmtools/Makefile])
--- origsrc/libvsapm-20240226/libvsapm/Makefile.am	2024-02-26 15:28:54.000000000 +0900
+++ src/libvsapm-20240226/libvsapm/Makefile.am	2024-03-02 13:04:11.133062600 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libvsapm-20240226/pyvsapm/Makefile.am	2024-02-26 15:28:54.000000000 +0900
+++ src/libvsapm-20240226/pyvsapm/Makefile.am	2024-03-02 13:04:28.915630100 +0900
@@ -1,5 +1,6 @@
 if HAVE_PYTHON
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -41,7 +42,7 @@ pyvsapm_la_LIBADD = \
 	@LIBBFIO_LIBADD@
 
 pyvsapm_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyvsapm_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyvsapm_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
--- origsrc/libvsapm-20240226/tests/Makefile.am	2024-02-26 15:28:54.000000000 +0900
+++ src/libvsapm-20240226/tests/Makefile.am	2024-03-02 13:04:39.619123800 +0900
@@ -1,6 +1,7 @@
 AUTOMAKE_OPTIONS = subdir-objects
 
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libvsapm-20240226/vsapmtools/Makefile.am	2024-02-26 15:28:54.000000000 +0900
+++ src/libvsapm-20240226/vsapmtools/Makefile.am	2024-03-02 13:04:00.630715000 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
