--- origsrc/template-glib-3.36.3/meson.build	2025-01-08 23:33:25.000000000 +0900
+++ src/template-glib-3.36.3/meson.build	2025-01-09 20:07:04.542993300 +0900
@@ -94,7 +94,6 @@ test_c_args = [
   ['-Werror=format-security', '-Werror=format=2' ],
   '-Werror=empty-body',
   '-Werror=implicit-function-declaration',
-  '-Werror=incompatible-pointer-types',
   '-Werror=pointer-arith',
   '-Werror=init-self',
   '-Werror=int-conversion',
@@ -107,6 +106,9 @@ test_c_args = [
   '-Werror=strict-prototypes',
   '-Werror=undef',
 ]
+if host_machine.system() != 'cygwin'
+  test_c_args += '-Werror=incompatible-pointer-types'
+endif
 if get_option('buildtype') != 'plain'
   test_c_args += '-fstack-protector-strong'
 endif
