--- origsrc/CQRlib-1.1.3/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/CQRlib-1.1.3/Makefile.am	2014-06-27 23:12:35.866126500 +0900
@@ -0,0 +1,13 @@
+
+check_PROGRAMS = CQRlibTest CPPQRTest
+CQRlibTest_SOURCES = CQRlibTest.c
+CQRlibTest_LDADD = libCQRlib.la
+CPPQRTest_SOURCES = CPPQRTest.cpp
+CPPQRTest_LDADD = libCQRlib.la
+
+include_HEADERS = cqrlib.h
+
+lib_LTLIBRARIES = libCQRlib.la
+libCQRlib_la_SOURCES = cqrlib.c
+libCQRlib_la_LDFLAGS = -no-undefined -version-info 3:0:1
+
--- origsrc/CQRlib-1.1.3/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/CQRlib-1.1.3/configure.ac	2014-06-27 23:13:40.216712400 +0900
@@ -0,0 +1,30 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([CQRlib], [1.1.3])
+AC_CONFIG_SRCDIR([CPPQRTest.cpp])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_MAKE_SET
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([float.h malloc.h stdlib.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+AC_C_INLINE
+
+# Checks for library functions.
+AC_CHECK_FUNCS([pow sqrt])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
