diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/OpenCoarrays-2.0.0/CMakeLists.txt src/OpenCoarrays-2.0.0/CMakeLists.txt
--- origsrc/OpenCoarrays-2.0.0/CMakeLists.txt	2018-03-09 07:13:40.000000000 +0900
+++ src/OpenCoarrays-2.0.0/CMakeLists.txt	2018-05-11 18:32:39.290656100 +0900
@@ -475,7 +475,7 @@ function(caf_compile_executable target m
   endforeach()
   add_custom_command(OUTPUT "${target}"
     COMMAND "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}/caf" ${includes} ${localDefs} ${config_Fortran_flags} -o "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${target}" "${CMAKE_CURRENT_SOURCE_DIR}/${main_depend}" ${ARGN}
-    DEPENDS "${main_depend}" ${ARGN} caf_mpi_static
+    DEPENDS "${main_depend}" ${ARGN} caf_mpi
     VERBATIM
     )
   add_custom_target("build_${target}" ALL
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/OpenCoarrays-2.0.0/src/mpi/CMakeLists.txt src/OpenCoarrays-2.0.0/src/mpi/CMakeLists.txt
--- origsrc/OpenCoarrays-2.0.0/src/mpi/CMakeLists.txt	2018-03-09 07:13:40.000000000 +0900
+++ src/OpenCoarrays-2.0.0/src/mpi/CMakeLists.txt	2018-05-11 18:40:33.279582800 +0900
@@ -151,10 +151,16 @@ set_target_properties(opencoarrays_mod
   Fortran_MODULE_DIRECTORY "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_INCLUDEDIR}/${mod_dir_tail}"
   POSITION_INDEPENDENT_CODE TRUE)
 
+option(ENABLE_STATIC "enable static" OFF)
+
 add_library(caf_mpi SHARED mpi_caf.c ../common/caf_auxiliary.c $<TARGET_OBJECTS:opencoarrays_mod>)
-add_library(caf_mpi_static STATIC  mpi_caf.c ../common/caf_auxiliary.c $<TARGET_OBJECTS:opencoarrays_mod>)
+if (ENABLE_STATIC)
+  add_library(caf_mpi_static STATIC  mpi_caf.c ../common/caf_auxiliary.c $<TARGET_OBJECTS:opencoarrays_mod>)
+endif()
 target_link_libraries(caf_mpi PRIVATE ${MPI_C_LIBRARIES} ${MPI_Fortran_LIBRARIES})
-target_link_libraries(caf_mpi_static PRIVATE ${MPI_C_LIBRARIES} ${MPI_Fortran_LIBRARIES})
+if (ENABLE_STATIC)
+  target_link_libraries(caf_mpi_static PRIVATE ${MPI_C_LIBRARIES} ${MPI_Fortran_LIBRARIES})
+endif()
 
 set(CAF_SO_VERSION 0)
 if(gfortran_compiler)
@@ -184,25 +190,31 @@ endif()
 install(DIRECTORY "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_INCLUDEDIR}/" DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
   FILES_MATCHING PATTERN "*.mod")
 
-set_target_properties( caf_mpi_static
+if (ENABLE_STATIC)
+  set_target_properties( caf_mpi_static
   PROPERTIES
   SOVERSION ${CAF_SO_VERSION}
-#  VERSION ${PROJECT_VERSION}
-)
+  #  VERSION ${PROJECT_VERSION}
+  )
+endif()
 
 if (gfortran_compiler)
   target_compile_options(caf_mpi INTERFACE -fcoarray=lib)
+  if (ENABLE_STATIC)
   target_compile_options(caf_mpi_static INTERFACE -fcoarray=lib)
+  endif()
 endif()
 
 install(TARGETS caf_mpi EXPORT OpenCoarraysTargets
-  DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-  LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+  RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+  ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
 )
-install(TARGETS caf_mpi_static EXPORT OpenCoarraysTargets
+if (ENABLE_STATIC)
+  install(TARGETS caf_mpi_static EXPORT OpenCoarraysTargets
   ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
   LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
   )
+endif()
 
 ##############################################
 # Configure `caf` and `cafrun` wrapper scripts
@@ -233,10 +245,16 @@ foreach( fcflag IN LISTS MPI_Fortran_COM
   set(CAF_MPI_Fortran_COMPILE_FLAGS "${CAF_MPI_Fortran_COMPILE_FLAGS} ${fcflag}" )
 endforeach()
 string(STRIP "${CAF_MPI_Fortran_COMPILE_FLAGS}" CAF_MPI_Fortran_COMPILE_FLAGS)
-set_target_properties(caf_mpi_static
+if (ENABLE_STATIC)
+  set_target_properties(caf_mpi_static
   PROPERTIES OUTPUT_NAME caf_mpi)
-get_target_property(libcaf_static caf_mpi_static OUTPUT_NAME)
-set(CAF_LIBS "${CMAKE_INSTALL_LIBDIR}/${CMAKE_STATIC_LIBRARY_PREFIX}${libcaf_static}${CMAKE_STATIC_LIBRARY_SUFFIX}")
+  get_target_property(libcaf_static caf_mpi_static OUTPUT_NAME)
+endif()
+if (CYGWIN AND NOT ENABLE_STATIC)
+  set(CAF_LIBS "${CMAKE_INSTALL_LIBDIR}/libcaf_mpi.dll.a")
+else()
+  set(CAF_LIBS "${CMAKE_INSTALL_LIBDIR}/${CMAKE_STATIC_LIBRARY_PREFIX}${libcaf_static}${CMAKE_STATIC_LIBRARY_SUFFIX}")
+endif()
 
 configure_file("${CMAKE_SOURCE_DIR}/src/extensions/caf.in" "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}/caf"
   @ONLY)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/OpenCoarrays-2.0.0/src/tests/integration/pde_solvers/coarrayBurgers/CMakeLists.txt src/OpenCoarrays-2.0.0/src/tests/integration/pde_solvers/coarrayBurgers/CMakeLists.txt
--- origsrc/OpenCoarrays-2.0.0/src/tests/integration/pde_solvers/coarrayBurgers/CMakeLists.txt	2018-03-09 07:13:40.000000000 +0900
+++ src/OpenCoarrays-2.0.0/src/tests/integration/pde_solvers/coarrayBurgers/CMakeLists.txt	2018-05-11 18:31:00.655217000 +0900
@@ -27,5 +27,5 @@ add_executable(coarray_burgers_pde
   ${library_directory}/object_interface.F90
   ${library_directory}/co_object_interface.F90
 )
-add_dependencies(coarray_burgers_pde caf_mpi_static)
+add_dependencies(coarray_burgers_pde caf_mpi)
 target_include_directories(coarray_burgers_pde PRIVATE ${config_directory})
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/OpenCoarrays-2.0.0/src/tests/integration/pde_solvers/coarrayHeatSimplified/CMakeLists.txt src/OpenCoarrays-2.0.0/src/tests/integration/pde_solvers/coarrayHeatSimplified/CMakeLists.txt
--- origsrc/OpenCoarrays-2.0.0/src/tests/integration/pde_solvers/coarrayHeatSimplified/CMakeLists.txt	2018-03-09 07:13:40.000000000 +0900
+++ src/OpenCoarrays-2.0.0/src/tests/integration/pde_solvers/coarrayHeatSimplified/CMakeLists.txt	2018-05-11 18:31:21.774780500 +0900
@@ -1,11 +1,11 @@
 set(CMAKE_Fortran_COMPILER "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}/caf")
 add_library(local_field OBJECT local_field.f90)
 add_library(global_field OBJECT global_field.f90)
-add_dependencies(local_field caf_mpi_static)
-add_dependencies(global_field local_field caf_mpi_static)
+add_dependencies(local_field caf_mpi)
+add_dependencies(global_field local_field caf_mpi)
 add_executable(co_heat
   main.f90
   $<TARGET_OBJECTS:local_field>
   $<TARGET_OBJECTS:global_field>
 )
-add_dependencies(co_heat caf_mpi_static)
+add_dependencies(co_heat caf_mpi)
