--- origsrc/c-blosc-1.3.2/bench/plot-speeds.py	2014-01-17 17:21:41.000000000 +0900
+++ src/c-blosc-1.3.2/bench/plot-speeds.py	2014-02-11 23:19:19.902000000 +0900
@@ -1,3 +1,4 @@
+#!/usr/bin/python
 """Script for plotting the results of the 'suite' benchmark.
 Invoke without parameters for usage hints.
 
--- origsrc/c-blosc-1.3.2/blosc/CMakeLists.txt	2014-01-17 17:21:41.000000000 +0900
+++ src/c-blosc-1.3.2/blosc/CMakeLists.txt	2014-02-11 23:20:10.614890600 +0900
@@ -98,7 +98,7 @@ endif(BUILD_STATIC)
 
 # install
 install(FILES blosc.h DESTINATION include COMPONENT DEV)
-install(TARGETS blosc_shared DESTINATION ${lib_dir} COMPONENT LIB)
+install(TARGETS blosc_shared RUNTIME DESTINATION bin ARCHIVE DESTINATION ${lib_dir} COMPONENT LIB)
 if(BUILD_STATIC)
     install(TARGETS blosc_static DESTINATION ${lib_dir} COMPONENT DEV)
 endif(BUILD_STATIC)
--- origsrc/c-blosc-1.3.2/hdf5/CMakeLists.txt	2014-02-11 23:18:17.875928100 +0900
+++ src/c-blosc-1.3.2/hdf5/CMakeLists.txt	2014-02-11 23:36:34.524070300 +0900
@@ -7,10 +7,15 @@ include_directories("${PROJECT_SOURCE_DI
 find_package(HDF5 REQUIRED)
 include_directories(HDF5_INCLIDE_DIRS)
 
+set(version_string ${BLOSC_VERSION_MAJOR}.${BLOSC_VERSION_MINOR}.${BLOSC_VERSION_PATCH})
 
 # targets
 add_library(blosc_filter_shared SHARED ${SOURCES})
 set_target_properties(blosc_filter_shared PROPERTIES OUTPUT_NAME blosc_filter)
+set_target_properties(blosc_filter_shared PROPERTIES
+        VERSION ${version_string}
+        SOVERSION ${version_string}
+)
 target_link_libraries(blosc_filter_shared blosc_shared ${HDF5_LIBRARIES})
 
 if(BUILD_STATIC)
@@ -23,7 +28,7 @@ endif(BUILD_STATIC)
 
 # install
 install(FILES blosc_filter.h DESTINATION include COMPONENT HDF5_FILTER_DEV)
-install(TARGETS blosc_filter_shared DESTINATION lib${LIB_SUFFIX} COMPONENT HDF5_FILTER)
+install(TARGETS blosc_filter_shared RUNTIME DESTINATION bin ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT HDF5_FILTER)
 if(BUILD_STATIC)
     install(
         TARGETS blosc_filter_static DESTINATION lib${LIB_SUFFIX} COMPONENT HDF5_FILTER_DEV)
