--- origsrc/lastpass-cli-1.3.4/process.c	2022-12-09 20:04:16.000000000 +0900
+++ src/lastpass-cli-1.3.4/process.c	2023-01-09 19:31:20.436808200 +0900
@@ -41,6 +41,9 @@
 #include <sys/resource.h>
 #include <limits.h>
 
+int ARGC;
+char **ARGV;
+
 #if defined(__linux__)
 #include <sys/prctl.h>
 #define USE_PRCTL
--- origsrc/lastpass-cli-1.3.4/process.h	2022-12-09 20:04:16.000000000 +0900
+++ src/lastpass-cli-1.3.4/process.h	2023-01-09 19:30:57.857056100 +0900
@@ -4,8 +4,8 @@
 #include <stdbool.h>
 #include <sys/types.h>
 
-int ARGC;
-char **ARGV;
+extern int ARGC;
+extern char **ARGV;
 
 void process_set_name(const char *name);
 void process_disable_ptrace(void);
