--- origsrc/csv2json/process.c	2011-12-16 01:24:33.000000000 +0900
+++ src/csv2json/process.c	2015-02-18 22:25:49.654897200 +0900
@@ -13,6 +13,7 @@ int process(FILE *file, char *sep) {
     char *p = buf;
     list *l = NULL;
 
+    printf("[");
     while((p = fgets(buf, BUFLEN, file)) != NULL) {
         if(first) {
             first = 0;
@@ -29,6 +30,7 @@ int process(FILE *file, char *sep) {
             count++;
         }
     }
+    printf("]");
 
     if(l != NULL) {
         freeList(l);
@@ -51,7 +53,7 @@ static int printRec(char *rec, char *sep
     while((q = (char *)strtok(p, sep)) != NULL) {
         p = NULL;
         stripNewline(q);
-        printf("%s: \"%s\"", getList(l), q);
+        printf("\"%s\": \"%s\"", getList(l), q);
         count++;
 
         if(count < collength)
--- origsrc/csv2json/test.csv	1970-01-01 09:00:00.000000000 +0900
+++ src/csv2json/test.csv	2015-02-18 22:25:49.654897200 +0900
@@ -0,0 +1,4 @@
+id,name,size
+1,red,3.14
+2,green,1.5
+3,blue,42.195
--- origsrc/csv2json/test.json	1970-01-01 09:00:00.000000000 +0900
+++ src/csv2json/test.json	2015-02-18 22:25:49.654897200 +0900
@@ -0,0 +1 @@
+[{"id": "1","name": "red","size": "3.14"},{"id": "2","name": "green","size": "1.5"},{"id": "3","name": "blue","size": "42.195"}]
\ No newline at end of file
