--- origsrc/pire-release-0.0.6/pire/Makefile.am	2020-05-15 00:40:25.000000000 +0900
+++ src/pire-release-0.0.6/pire/Makefile.am	2020-06-17 00:24:19.023772400 +0900
@@ -70,6 +70,8 @@ libpire_la_SOURCES += \
 	extra/glyphs.h
 endif
 
+libpire_la_LDFLAGS = -no-undefined
+
 pire_hdrdir = $(includedir)/pire
 pire_hdr_HEADERS = \
 	approx_matching.h \
--- origsrc/pire-release-0.0.6/tests/Makefile.am	2020-05-15 00:40:25.000000000 +0900
+++ src/pire-release-0.0.6/tests/Makefile.am	2020-06-17 00:24:38.405029100 +0900
@@ -12,6 +12,8 @@ libpire_unit_la_SOURCES = \
 	stub/cppunit.h
 libpire_unit_la_CXXFLAGS = -I$(top_srcdir)/pire
 
+libpire_unit_la_LDFLAGS = -no-undefined
+
 check_PROGRAMS = pire_test
 
 pire_test_SOURCES = \
