--- origsrc/varconf-1.0.1/tests/Makefile.am	2012-11-20 21:56:17.000000000 +0900
+++ src/varconf-1.0.1/tests/Makefile.am	2013-09-25 14:39:11.796547100 +0900
@@ -15,7 +15,7 @@ LDADD = $(top_builddir)/varconf/libvarco
 
 conftest_SOURCES = conftest.cpp
 
-symbol-check:
+symbol-check$(EXEEXT):
 	@set -e; \
 	newsymbols=$$(mktemp); \
 	test -f ${srcdir}/symbols.${host_cpu} && \
--- origsrc/varconf-1.0.1/varconf/Makefile.am	2012-02-12 22:25:53.000000000 +0900
+++ src/varconf-1.0.1/varconf/Makefile.am	2013-09-25 14:26:57.394203300 +0900
@@ -4,7 +4,7 @@ INCLUDES = -I$(top_srcdir)
 
 lib_LTLIBRARIES = libvarconf-1.0.la
 
-libvarconf_1_0_la_LDFLAGS = -version-info @VARCONF_VERSION_INFO@ $(SIGC_LIBS)
+libvarconf_1_0_la_LDFLAGS = -version-info @VARCONF_VERSION_INFO@ $(SIGC_LIBS) -no-undefined
 
 libvarconf_1_0_la_SOURCES = \
     config.cpp parse_error.cpp variable.cpp \
