--- origsrc/libfusn-20240416/Makefile.am	2024-04-16 12:50:34.000000000 +0900
+++ src/libfusn-20240416/Makefile.am	2025-05-17 18:03:26.387573100 +0900
@@ -3,12 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcnotify \
-	libuna \
-	libfdatetime \
 	libfusn \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libfusn-20240416/configure.ac	2024-04-16 12:50:34.000000000 +0900
+++ src/libfusn-20240416/configure.ac	2025-05-17 18:03:26.387573100 +0900
@@ -108,10 +108,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
 AC_CONFIG_FILES([libfusn/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
