--- origsrc/libfusn-20150902/Makefile.am	2015-01-04 22:36:23.000000000 +0900
+++ src/libfusn-20150902/Makefile.am	2016-04-05 10:27:23.711748800 +0900
@@ -3,13 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libcnotify \
-	libuna \
-	libfdatetime \
 	libfusn \
-	po \
 	manuals \
 	tests \
 	msvscpp
@@ -87,4 +81,3 @@ splint:
 	(cd $(srcdir)/libfusn && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
-
--- origsrc/libfusn-20150902/configure.ac	2015-09-03 02:20:17.000000000 +0900
+++ src/libfusn-20150902/configure.ac	2016-04-05 10:27:58.766899700 +0900
@@ -136,11 +136,6 @@ dnl Have configure make the Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
 AC_CONFIG_FILES([libfusn/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
@@ -177,4 +172,3 @@ Features:
    Verbose output:       $ac_cv_enable_verbose_output
    Debug output:         $ac_cv_enable_debug_output
 ]);
-
--- origsrc/libfusn-20150902/libfusn/Makefile.am	2014-10-05 04:02:42.000000000 +0900
+++ src/libfusn-20150902/libfusn/Makefile.am	2016-04-05 10:27:55.716566000 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -51,4 +52,3 @@ distclean: clean
 splint:
 	@echo "Running splint on libfusn ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(libfusn_la_SOURCES)
-
--- origsrc/libfusn-20150902/tests/Makefile.am	2014-11-01 20:38:41.000000000 +0900
+++ src/libfusn-20150902/tests/Makefile.am	2016-04-05 10:28:13.787629700 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -18,4 +19,3 @@ MAINTAINERCLEANFILES = \
 
 distclean: clean
 	/bin/rm -f Makefile
-
