--- origsrc/lwp-2.6/src/Makefile.am	2008-08-09 04:35:28.000000000 +0900
+++ src/lwp-2.6/src/Makefile.am	2012-03-30 22:23:22.690555900 +0900
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
 lib_LTLIBRARIES = liblwp.la
-noinst_PROGRAMS = cswitch testlwp testlwp-static tdb rw
+noinst_PROGRAMS = cswitch testlwp tdb rw
 
 AM_CPPFLAGS = -I$(top_srcdir)/include
 LDADD = liblwp.la
--- origsrc/lwp-2.6/src-pt/Makefile.am	2007-12-12 02:29:05.000000000 +0900
+++ src/lwp-2.6/src-pt/Makefile.am	2012-03-30 22:25:17.083134000 +0900
@@ -4,7 +4,7 @@ VPATH=$(srcdir):$(top_srcdir)/src
 
 if build_pthread
 BUILD_LIB = liblwp_pt.la
-BUILD_BIN = cswitch_pt testlwp_pt testlwp-static_pt tdb_pt rw_pt
+BUILD_BIN = cswitch_pt testlwp_pt tdb_pt rw_pt
 endif
 
 lib_LTLIBRARIES = $(BUILD_LIB)
@@ -14,7 +14,7 @@ AM_CPPFLAGS = -I$(top_srcdir)/include -D
 liblwp_pt_la_SOURCES = iomgr_pt.c lock_pt.c lwp_pt.c preempt_pt.c \
     lwp.private_pt.h dllist.h
 liblwp_pt_la_LIBADD = fasttime_pt.lo timer_pt.lo
-liblwp_pt_la_LDFLAGS = -version-info $(LIBTOOL_VERSION) $(LIBPTHREAD)
+liblwp_pt_la_LDFLAGS = $(LIBTOOL_LDFLAGS) $(LIBPTHREAD)
 
 noinst_PROGRAMS = $(BUILD_BIN)
 cswitch_pt_SOURCES =
--- origsrc/lwp-2.6/src-pt/lwp_pt.c	2005-03-04 03:53:51.000000000 +0900
+++ src/lwp-2.6/src-pt/lwp_pt.c	2012-03-30 22:21:22.540165300 +0900
@@ -295,7 +295,7 @@ static void *lwp_newprocess(void *arg)
 }
 
 int LWP_CreateProcess(void (*ep)(void *), int stacksize, int priority,
-		      void *parm, char *name, PROCESS *ret)
+		      void *parm, const char *name, PROCESS *ret)
 {
     PROCESS             pid;
     struct lwp_forkinfo newproc;
@@ -420,7 +420,7 @@ int LWP_QWait()
     return LWP_SUCCESS;
 }
 
-int LWP_INTERNALSIGNAL(void *event, int yield)
+int LWP_INTERNALSIGNAL(const void *event, int yield)
 {
     struct list_head *ptr;
     PROCESS           this, pid;
@@ -460,7 +460,7 @@ int LWP_INTERNALSIGNAL(void *event, int
  * cleanup handler if we get cancelled while waiting on the condition
  * variable. (cleanup needs to lock the run_mutex to removing us from the
  * list of threads, but we're already sort of `joined') */
-int LWP_MwaitProcess (int wcount, char *evlist[])
+int LWP_MwaitProcess (int wcount, const void *evlist[])
 {
     PROCESS pid;
     int     entries, i;
@@ -496,12 +496,12 @@ int LWP_MwaitProcess (int wcount, char *
     return LWP_SUCCESS;
 }
 
-int LWP_WaitProcess (void *event)
+int LWP_WaitProcess (const void *event)
 {
     void *evlist[2];
 
     evlist[0] = event; evlist[1] = NULL;
-    return LWP_MwaitProcess(1, (char**)evlist);
+    return LWP_MwaitProcess(1, (const void**)evlist);
 }
 
 int LWP_NewRock (int Tag, char *Value)
