--- origsrc/qthreads-1.20/src/fastcontext/asm.S	2024-05-03 02:30:53.000000000 +0900
+++ src/qthreads-1.20/src/fastcontext/asm.S	2024-09-06 19:10:17.345536800 +0900
@@ -50,7 +50,7 @@
 # else
 #  error What kind of a Mac is this?
 # endif
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__CYGWIN__)
 # if (QTHREAD_ASSEMBLY_ARCH == QTHREAD_ARM)
 #  define NEEDARMCONTEXT 1
 #  define SET qt_setmctxt
--- origsrc/qthreads-1.20/test/argparsing.h	2024-05-03 02:30:53.000000000 +0900
+++ src/qthreads-1.20/test/argparsing.h	2024-09-06 19:10:17.345536800 +0900
@@ -91,7 +91,7 @@ static ARGP_Atomic(int) verbose;
 #if defined(SILENT_ARGPARSING)
 # warning Silencing iprintf() output.
 # define iprintf(...)
-#elif defined(__tile__) || defined(__CYGWIN32__)
+#elif defined(__tile__) || defined(__CYGWIN__)
 # define iprintf printf
 #else
 static void iprintf(const char *restrict format,
