--- origsrc/qthreads-1.22/CMakeLists.txt	2025-03-04 08:31:59.000000000 +0900
+++ src/qthreads-1.22/CMakeLists.txt	2025-05-12 15:19:55.708913300 +0900
@@ -28,3 +28,9 @@ if(CMAKE_PROJECT_NAME STREQUAL PROJECT_N
   add_subdirectory(test)
 endif()
 
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/qthread.pc.in ${CMAKE_BINARY_DIR}/qthread.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/qthread.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/qthreads-1.22/qthread.pc.in	2025-03-04 08:31:59.000000000 +0900
+++ src/qthreads-1.22/qthread.pc.in	2025-05-12 15:20:03.769107200 +0900
@@ -6,7 +6,7 @@ includedir=@includedir@
 Name: qthreads
 Description: Lightweight locality-aware user-level threading runtime.
 URL: https://github.com/sandialabs/qthreads
-Version: @VERSION@
+Version: @PROJECT_VERSION@
 Cflags: -I${includedir}
 Libs: -L${libdir} -lqthread
 Libs.private: @LIBS@
--- origsrc/qthreads-1.22/src/CMakeLists.txt	2025-03-04 08:31:59.000000000 +0900
+++ src/qthreads-1.22/src/CMakeLists.txt	2025-05-12 15:08:16.138175700 +0900
@@ -64,6 +64,10 @@ set(QTHREADS_SOURCES
 
 add_library(qthread ${QTHREADS_SOURCES})
 
+if (DEFINED SOVERSION)
+  set_target_properties(qthread PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 if ("${QTHREADS_CONTEXT_SWAP_IMPL}" STREQUAL "fastcontext")
   target_sources(qthread PRIVATE fastcontext/asm.S fastcontext/context.c)
 elseif ("${QTHREADS_CONTEXT_SWAP_IMPL}" STREQUAL "system")
--- origsrc/qthreads-1.22/src/fastcontext/asm.S	2025-03-04 08:31:59.000000000 +0900
+++ src/qthreads-1.22/src/fastcontext/asm.S	2025-05-12 14:48:50.701043300 +0900
@@ -45,7 +45,7 @@
 # else
 #  error What kind of a Mac is this?
 # endif
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__CYGWIN__)
 # if (QTHREAD_ASSEMBLY_ARCH == QTHREAD_ARM)
 #  define NEEDARMCONTEXT 1
 #  define SET qt_setmctxt
--- origsrc/qthreads-1.22/src/io.c	2025-03-04 08:31:59.000000000 +0900
+++ src/qthreads-1.22/src/io.c	2025-05-12 14:41:39.250950200 +0900
@@ -4,8 +4,10 @@
 #include <stdio.h>    /* for fprintf() */
 #include <stdlib.h>   /* for abort() */
 #include <sys/time.h> /* for gettimeofday() */
+#if defined(__linux__)
 /* - syscall(2) */
 #include <sys/syscall.h>
+#endif
 #include <unistd.h>
 /* - accept(2) */
 #include <sys/socket.h>
--- origsrc/qthreads-1.22/src/syscalls/accept.c	2025-03-04 08:31:59.000000000 +0900
+++ src/qthreads-1.22/src/syscalls/accept.c	2025-05-12 14:43:17.098652400 +0900
@@ -1,7 +1,9 @@
 #include <stdint.h>
 
 /* System Headers */
+#if defined(__linux__)
 #include <sys/syscall.h> /* for SYS_accept and others */
+#endif
 #include <unistd.h>
 
 /* Public Headers */
--- origsrc/qthreads-1.22/src/syscalls/connect.c	2025-03-04 08:31:59.000000000 +0900
+++ src/qthreads-1.22/src/syscalls/connect.c	2025-05-12 14:43:23.737831600 +0900
@@ -1,7 +1,9 @@
 #include <stdint.h>
 
 /* System Headers */
+#if defined(__linux__)
 #include <sys/syscall.h> /* for SYS_accept and others */
+#endif
 #include <unistd.h>
 
 /* Public Headers */
--- origsrc/qthreads-1.22/src/syscalls/poll.c	2025-03-04 08:31:59.000000000 +0900
+++ src/qthreads-1.22/src/syscalls/poll.c	2025-05-12 14:42:24.715125700 +0900
@@ -1,6 +1,8 @@
 #include <stdint.h>
 
+#if defined(__linux__)
 #include <sys/syscall.h> /* for SYS_accept and others */
+#endif
 #include <unistd.h>
 
 /* Public Headers */
--- origsrc/qthreads-1.22/src/syscalls/pread.c	2025-03-04 08:31:59.000000000 +0900
+++ src/qthreads-1.22/src/syscalls/pread.c	2025-05-12 14:43:30.969401000 +0900
@@ -3,7 +3,9 @@
 /* System Headers */
 #include <sys/types.h>
 
+#if defined(__linux__)
 #include <sys/syscall.h> /* for SYS_accept and others */
+#endif
 #include <unistd.h>
 
 /* Public Headers */
--- origsrc/qthreads-1.22/src/syscalls/pwrite.c	2025-03-04 08:31:59.000000000 +0900
+++ src/qthreads-1.22/src/syscalls/pwrite.c	2025-05-12 14:43:36.345594500 +0900
@@ -3,7 +3,9 @@
 /* System Headers */
 #include <sys/types.h>
 
+#if defined(__linux__)
 #include <sys/syscall.h> /* for SYS_accept and others */
+#endif
 #include <unistd.h>
 
 /* Public Headers */
--- origsrc/qthreads-1.22/src/syscalls/read.c	2025-03-04 08:31:59.000000000 +0900
+++ src/qthreads-1.22/src/syscalls/read.c	2025-05-12 14:42:41.053643800 +0900
@@ -3,7 +3,9 @@
 /* System Headers */
 #include <sys/types.h>
 
+#if defined(__linux__)
 #include <sys/syscall.h> /* for SYS_accept and others */
+#endif
 #include <unistd.h>
 
 /* Public Headers */
--- origsrc/qthreads-1.22/src/syscalls/select.c	2025-03-04 08:31:59.000000000 +0900
+++ src/qthreads-1.22/src/syscalls/select.c	2025-05-12 14:43:45.710326800 +0900
@@ -3,7 +3,9 @@
 /* System Headers */
 #include <sys/select.h>
 
+#if defined(__linux__)
 #include <sys/syscall.h> /* for SYS_accept and others */
+#endif
 #include <unistd.h>
 
 /* Public Headers */
--- origsrc/qthreads-1.22/src/syscalls/system.c	2025-03-04 08:31:59.000000000 +0900
+++ src/qthreads-1.22/src/syscalls/system.c	2025-05-12 14:43:58.563799300 +0900
@@ -1,7 +1,9 @@
 #include <stdint.h>
 
 /* System Headers */
+#if defined(__linux__)
 #include <sys/syscall.h> /* for SYS_accept and others */
+#endif
 
 /* Public Headers */
 #include "qthread/qt_syscalls.h"
--- origsrc/qthreads-1.22/src/syscalls/user_defined.c	2025-03-04 08:31:59.000000000 +0900
+++ src/qthreads-1.22/src/syscalls/user_defined.c	2025-05-12 14:44:07.331757200 +0900
@@ -5,7 +5,9 @@
 #include <sys/uio.h>
 #include <unistd.h>
 
+#if defined(__linux__)
 #include <sys/syscall.h> /* for SYS_accept and others */
+#endif
 
 /* Internal Headers */
 #include "qt_asserts.h"
--- origsrc/qthreads-1.22/src/syscalls/wait4.c	2025-03-04 08:31:59.000000000 +0900
+++ src/qthreads-1.22/src/syscalls/wait4.c	2025-05-12 14:44:14.485408300 +0900
@@ -1,7 +1,9 @@
 #include <stdint.h>
 
 /* System Headers */
+#if defined(__linux__)
 #include <sys/syscall.h> /* for SYS_accept and others */
+#endif
 #include <unistd.h>
 
 /* Public Headers */
--- origsrc/qthreads-1.22/src/syscalls/write.c	2025-03-04 08:31:59.000000000 +0900
+++ src/qthreads-1.22/src/syscalls/write.c	2025-05-12 14:44:20.723771600 +0900
@@ -3,7 +3,9 @@
 /* System Headers */
 #include <sys/types.h>
 
+#if defined(__linux__)
 #include <sys/syscall.h> /* for SYS_accept and others */
+#endif
 #include <unistd.h>
 
 /* Public Headers */
--- origsrc/qthreads-1.22/test/argparsing.h	2025-03-04 08:31:59.000000000 +0900
+++ src/qthreads-1.22/test/argparsing.h	2025-05-12 14:47:58.099130900 +0900
@@ -113,6 +113,7 @@ static ARGP_Atomic(int) verbose;
 #warning Silencing iprintf() output.
 #define iprintf(...)
 #else
+#if !defined(__CYGWIN__)
 inline static void iprintf(char const *restrict format, ...) {
   if (atomic_load_explicit(&verbose, memory_order_relaxed)) {
     va_list ap;
@@ -123,6 +124,7 @@ inline static void iprintf(char const *r
     va_end(ap);
   }
 }
+#endif
 
 #endif // if defined(__CYGWIN32__)
 
