--- origsrc/rvm-1.17/configure.in	2010-03-23 03:00:37.000000000 +0900
+++ src/rvm-1.17/configure.in	2010-05-13 22:19:17.587526000 +0900
@@ -36,6 +36,7 @@ AC_PROG_CC
 AC_PROG_CPP
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
+AC_LIBTOOL_WIN32_DLL
 AM_PROG_LIBTOOL
 
 dnl pkg-config stuff
--- origsrc/rvm-1.17/rds/Makefile.am	2010-03-23 03:00:37.000000000 +0900
+++ src/rvm-1.17/rds/Makefile.am	2010-05-13 22:20:46.853151000 +0900
@@ -33,7 +33,7 @@ librds_la_LIBADD = $(top_builddir)/seg/l
 
 librdslwp_la_CPPFLAGS = $(AM_CPPFLAGS) -DRVM_USELWP $(LWP_CFLAGS)
 librdslwp_la_SOURCES = $(librds_sources)
-librdslwp_la_LDFLAGS = $(LIBTOOL_LDFLAGS)
+librdslwp_la_LDFLAGS = $(LIBTOOL_LDFLAGS) -no-undefined
 librdslwp_la_LIBADD = $(top_builddir)/seg/libseglwp.la \
 	$(top_builddir)/rvm/librvmlwp.la $(LWP_LIBS)
 
--- origsrc/rvm-1.17/rvm/Makefile.am	2010-03-23 03:00:37.000000000 +0900
+++ src/rvm-1.17/rvm/Makefile.am	2010-05-13 22:21:02.431276000 +0900
@@ -32,7 +32,7 @@ librvm_la_LDFLAGS = $(LIBTOOL_LDFLAGS)
 
 librvmlwp_la_CPPFLAGS = $(AM_CPPFLAGS) -DRVM_USELWP $(LWP_CFLAGS)
 librvmlwp_la_SOURCES = $(librvm_sources)
-librvmlwp_la_LDFLAGS = $(LIBTOOL_LDFLAGS)
+librvmlwp_la_LDFLAGS = $(LIBTOOL_LDFLAGS) -no-undefined
 librvmlwp_la_LIBADD = $(LWP_LIBS)
 
 librvmpt_la_CPPFLAGS = $(AM_CPPFLAGS) -DRVM_USEPT $(PTHREAD_CFLAGS)
--- origsrc/rvm-1.17/seg/Makefile.am	2010-03-23 03:00:38.000000000 +0900
+++ src/rvm-1.17/seg/Makefile.am	2010-05-13 22:21:21.884401000 +0900
@@ -27,7 +27,7 @@ libseg_la_LIBADD = $(top_builddir)/rvm/l
 
 libseglwp_la_CPPFLAGS = $(AM_CPPFLAGS) -DUSE_RVMLWP $(LWP_CFLAGS)
 libseglwp_la_SOURCES = $(libseg_sources)
-libseglwp_la_LDFLAGS = $(LIBTOOL_LDFLAGS)
+libseglwp_la_LDFLAGS = $(LIBTOOL_LDFLAGS) -no-undefined
 libseglwp_la_LIBADD = $(top_builddir)/rvm/librvmlwp.la $(LWP_LIBS)
 
 libsegpt_la_CPPFLAGS = $(AM_CPPFLAGS) -DUSE_RVMPT $(PTHREAD_CFLAGS)
