--- origsrc/hivex-1.3.8/lib/Makefile.am	2013-07-25 00:34:06.000000000 +0900
+++ src/hivex-1.3.8/lib/Makefile.am	2013-08-31 22:13:50.785390400 +0900
@@ -39,7 +39,7 @@ libhivex_la_SOURCES = \
 	write.c
 
 libhivex_la_LIBADD =  ../gnulib/lib/libgnu.la $(LTLIBOBJS)
-libhivex_la_LDFLAGS = \
+libhivex_la_LDFLAGS = -no-undefined \
 	-version-info 0:0:0 \
 	$(VERSION_SCRIPT_FLAGS)$(srcdir)/hivex.syms \
 	$(LTLIBICONV) \
--- origsrc/hivex-1.3.8/python/Makefile.am	2013-07-25 23:09:00.000000000 +0900
+++ src/hivex-1.3.8/python/Makefile.am	2013-08-31 22:14:37.551015400 +0900
@@ -35,8 +35,8 @@ python_LTLIBRARIES = libhivexmod.la
 libhivexmod_la_SOURCES = hivex-py.c
 libhivexmod_la_CFLAGS = -Wall $(PYTHON_CFLAGS) \
 		        -I$(top_srcdir)/lib -I$(top_builddir)/lib
-libhivexmod_la_LIBADD = $(top_builddir)/lib/libhivex.la
-libhivexmod_la_LDFLAGS = -avoid-version -shared -module -shrext $(PYTHON_EXT_SUFFIX)
+libhivexmod_la_LIBADD = $(top_builddir)/lib/libhivex.la -lpython$(PYTHON_VERSION)
+libhivexmod_la_LDFLAGS = -no-undefined -avoid-version -shared -module -shrext $(PYTHON_EXT_SUFFIX)
 
 TESTS_ENVIRONMENT = ../run
 
