--- origsrc/hivex-1.3.13/lib/Makefile.am	2013-09-10 20:04:12.000000000 +0900
+++ src/hivex-1.3.13/lib/Makefile.am	2016-03-13 14:42:12.868081700 +0900
@@ -39,7 +39,7 @@ libhivex_la_SOURCES = \
 	write.c
 
 libhivex_la_LIBADD =  ../gnulib/lib/libgnu.la $(LTLIBOBJS)
-libhivex_la_LDFLAGS = \
+libhivex_la_LDFLAGS = -no-undefined \
 	-version-info 0:0:0 \
 	$(VERSION_SCRIPT_FLAGS)$(srcdir)/hivex.syms \
 	$(LTLIBICONV) \
--- origsrc/hivex-1.3.13/python/Makefile.am	2014-09-25 00:44:00.000000000 +0900
+++ src/hivex-1.3.13/python/Makefile.am	2016-03-13 14:42:12.868081700 +0900
@@ -39,8 +39,8 @@ python_LTLIBRARIES = libhivexmod.la
 libhivexmod_la_SOURCES = hivex-py.c
 libhivexmod_la_CFLAGS = -Wall $(PYTHON_CFLAGS) \
 		        -I$(top_srcdir)/lib -I$(top_builddir)/lib
-libhivexmod_la_LIBADD = $(top_builddir)/lib/libhivex.la
-libhivexmod_la_LDFLAGS = -avoid-version -shared -module -shrext $(PYTHON_EXT_SUFFIX)
+libhivexmod_la_LIBADD = $(top_builddir)/lib/libhivex.la -lpython$(PYTHON_VERSION)
+libhivexmod_la_LDFLAGS = -no-undefined -avoid-version -shared -module -shrext $(PYTHON_EXT_SUFFIX)
 
 TESTS_ENVIRONMENT = ../run
 
