--- origsrc/libclastfm-0.5/src/Makefile.am	2012-03-03 07:56:57.000000000 +0900
+++ src/libclastfm-0.5/src/Makefile.am	2012-06-07 23:02:02.382595100 +0900
@@ -1,6 +1,6 @@
 noinst_PROGRAMS = test example
 
-AM_LDFLAGS =  -Wl,--as-needed -static
+AM_LDFLAGS =  -Wl,--as-needed
 AM_CPPFLAGS = -g -Wall
 LDADD = libclastfm.la
 
@@ -17,7 +17,7 @@ libclastfm_la_SOURCES = \
 
 # Libtool doesnt respect AM_LDFLAGS that much
 libclastfm_la_CPPFLAGS = -g -Wall $(CURL_CFLAGS) 
-libclastfm_la_LDFLAGS = -Wl,--as-needed $(CURL_LIBS) 
+libclastfm_la_LDFLAGS = -Wl,--as-needed -no-undefined $(CURL_LIBS) 
 
 # Public header file
 include_HEADERS = clastfm.h
