--- origsrc/uniutils-2.28/Makefile.am	2023-08-21 03:45:44.000000000 +0900
+++ src/uniutils-2.28/Makefile.am	2023-11-12 15:52:01.679190500 +0900
@@ -9,11 +9,10 @@ unireverse_SOURCES = unireverse.c
 unifuzz_SOURCES = unifuzz.c putu8.c unirange.c
 unisurrogate_SOURCES = unisurrogate.c
 noinst_HEADERS = nelson.h  offsettbl.h  unicode.h  uniftypes.h  unirange.h  utf8error.h
-LOCALEDIR=/usr/local/share/locale
-AM_CFLAGS = -DLOCALEDIR=\"${LOCALEDIR}\" -fPIE
+AM_CFLAGS = -DLOCALEDIR=\"${localedir}\" -fPIE
 AUTOMAKE_OPTIONS = dist-zip dist-bzip2
 EXTRA_DIST = genunames.awk TestData utf8lookup CREDITS
-
+LIBS += -lintl
 
 
 
--- origsrc/uniutils-2.28/uniname.c	2020-11-20 17:28:53.000000000 +0900
+++ src/uniutils-2.28/uniname.c	2023-11-12 15:47:54.218059600 +0900
@@ -42,6 +42,8 @@
 #include "unirange.h"
 #include "offsettbl.h"
 #include "nelson.h"
+#include <ctype.h>
+#include <getopt.h>
 
 #define MSGSIZE 128
 
@@ -283,9 +285,6 @@ int main (int ac, char *av[])
 
   extern UTF32 Get_UTF32_From_UTF8 (int,int *,unsigned char **);
   extern UTF32 Get_UTF32(int,int *,unsigned char **);
-  extern int optind;
-  extern int opterr;
-  extern int optopt;
   extern int GetRange(UTF32);
   extern struct cr Range_Table [];
   extern int GetNelson(UTF32,unsigned int *);
--- origsrc/uniutils-2.28/unisurrogate.c	2020-11-20 17:28:53.000000000 +0900
+++ src/uniutils-2.28/unisurrogate.c	2023-11-12 15:49:29.941391600 +0900
@@ -2,6 +2,7 @@
 #include <stdio.h>
 #include <errno.h>
 #include "config.h"
+#include <sys/types.h>
 
 char compdate[]="Compiled " __DATE__ " " __TIME__ ;
 char pgname[]="unisurrogate";
