--- origsrc/libcouchbase-2.7.6/CMakeLists.txt	2017-07-12 17:19:31.000000000 +0900
+++ src/libcouchbase-2.7.6/CMakeLists.txt	2017-07-20 23:22:37.604595800 +0900
@@ -326,8 +326,6 @@ ELSE()
     SET(LCB_SNAPPY_LINK "")
 ENDIF()
 
-SET_TARGET_PROPERTIES(couchbase PROPERTIES PREFIX "lib")
-SET_TARGET_PROPERTIES(couchbase PROPERTIES IMPORT_PREFIX "lib")
 SET(LCB_LINK_DEPS ${lcb_plat_libs} ${lcb_ssl_libs} ${LCB_SNAPPY_LINK} ${LCB_HDR_HISTOGRAM_LINK})
 IF(CMAKE_SYSTEM_NAME STREQUAL "Linux")
     SET(LCB_LINK_DEPS ${LCB_LINK_DEPS} rt)
@@ -379,7 +377,6 @@ ADD_SUBDIRECTORY(plugins/io/select)
 ADD_SUBDIRECTORY(plugins/io/iocp)
 INSTALL(TARGETS couchbase
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 INSTALL(DIRECTORY include/libcouchbase ${LCB_GENSRCDIR}/libcouchbase
--- origsrc/libcouchbase-2.7.6/example/subdoc/subdoc-multi.cc	2017-07-12 17:19:31.000000000 +0900
+++ src/libcouchbase-2.7.6/example/subdoc/subdoc-multi.cc	2017-07-20 23:22:37.604595800 +0900
@@ -4,6 +4,7 @@
 #include <libcouchbase/couchbase.h>
 #include <libcouchbase/api3.h>
 #include <assert.h>
+#include <stdlib.h>
 #include <string.h>
 #include <string>
 #include <vector>
--- origsrc/libcouchbase-2.7.6/plugins/io/libev/CMakeLists.txt	2017-07-12 17:19:31.000000000 +0900
+++ src/libcouchbase-2.7.6/plugins/io/libev/CMakeLists.txt	2017-07-20 23:22:37.604595800 +0900
@@ -12,8 +12,6 @@ ENDIF()
 ADD_LIBRARY(couchbase_libev SHARED plugin-libev.c)
 SET_TARGET_PROPERTIES(couchbase_libev
     PROPERTIES
-    PREFIX "lib"
-    IMPORT_PREFIX "lib"
     COMPILE_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_CORE_CFLAGS} -std=gnu99")
 TARGET_LINK_LIBRARIES(couchbase_libev ${LIBEV_LIBRARIES})
 INCLUDE_DIRECTORIES(AFTER ${LIBEV_INCLUDE_DIR})
@@ -26,4 +24,5 @@ ENDIF()
 
 INSTALL(TARGETS
     couchbase_libev
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
--- origsrc/libcouchbase-2.7.6/plugins/io/libevent/CMakeLists.txt	2017-07-12 17:19:31.000000000 +0900
+++ src/libcouchbase-2.7.6/plugins/io/libevent/CMakeLists.txt	2017-07-20 23:22:37.620223800 +0900
@@ -22,7 +22,9 @@ IF(LCB_EMBED_PLUGIN_LIBEVENT)
     SET(LCB_LINK_SPEC "${LCB_LINKS_SPEC} ${LIBEVENT_LIBRARIES}")
 ELSE()
     ADD_LIBRARY(couchbase_libevent SHARED plugin-libevent.c)
-    INSTALL(TARGETS couchbase_libevent LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    INSTALL(TARGETS couchbase_libevent
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
     TARGET_LINK_LIBRARIES(couchbase_libevent ${LIBEVENT_LIBRARIES})
 ENDIF()
 
--- origsrc/libcouchbase-2.7.6/plugins/io/libuv/CMakeLists.txt	2017-07-12 17:19:31.000000000 +0900
+++ src/libcouchbase-2.7.6/plugins/io/libuv/CMakeLists.txt	2017-07-21 00:06:36.453000000 +0900
@@ -29,8 +29,6 @@ ADD_LIBRARY(couchbase_libuv SHARED plugi
 #ENDIF()
 SET_TARGET_PROPERTIES(couchbase_libuv
     PROPERTIES
-    PREFIX "lib"
-    IMPORT_PREFIX "lib"
     COMPILE_FLAGS "${CMAKE_C_FLAGS} ${LCB_CORE_FLAGS} ${_lcbuv_cflags}")
 
 TARGET_LINK_LIBRARIES(couchbase_libuv ${LIBUV_LIBRARIES} couchbase)
--- origsrc/libcouchbase-2.7.6/src/dns-srv.cc	2017-07-12 17:19:31.000000000 +0900
+++ src/libcouchbase-2.7.6/src/dns-srv.cc	2017-07-20 23:32:19.854000000 +0900
@@ -9,7 +9,9 @@
 #include <string>
 
 #ifdef HAVE_ARPA_NAMESER_H
+#if !defined(__CYGWIN__)
 #include <arpa/nameser.h>
+#endif
 #if __NAMESER < 19991006
 #undef HAVE_RES_SEARCH
 #endif /* __NAMESER < NNN */
--- origsrc/libcouchbase-2.7.6/src/iofactory.c	2017-07-12 17:19:31.000000000 +0900
+++ src/libcouchbase-2.7.6/src/iofactory.c	2017-07-20 23:22:37.620223800 +0900
@@ -68,6 +68,8 @@ typedef struct {
 #else
     #define PLUGIN_SO(NAME) "libcouchbase_"NAME".dll."
 #endif /* _DEBUG */
+#elif defined(__CYGWIN__)
+#define PLUGIN_SO(NAME) "cygcouchbase_"NAME".dll"
 #else
 #define PLUGIN_SO(NAME) "libcouchbase_"NAME".so"
 #endif
