--- origsrc/libcouchbase-2.4.7/CMakeLists.txt	2015-02-16 11:43:00.000000000 +0900
+++ src/libcouchbase-2.4.7/CMakeLists.txt	2015-03-17 23:42:54.412464700 +0900
@@ -240,7 +240,10 @@ SET(LCB_CORE_OBJS
     ${lcb_plat_objs}
     ${lcb_ssl_objs})
 
-ADD_LIBRARY(couchbaseS STATIC ${LCB_CORE_OBJS})
+SET(ENABLE_STATIC "enable static" OFF)
+IF(ENABLE_STATIC)
+    ADD_LIBRARY(couchbaseS STATIC ${LCB_CORE_OBJS})
+ENDIF()
 ADD_LIBRARY(couchbase ${_lcb_linkspec} ${LCB_CORE_OBJS} ${PROJECT_BINARY_DIR}/dllversion.rc)
 
 # For DTrace implementations which need to gain access to all the *.o files first
@@ -251,7 +254,9 @@ ADD_LIBRARY(couchbase ${_lcb_linkspec} $
 IF(LCB_DTRACE_INSTRO)
     SET(_lcb_linkhook "${PROJECT_SOURCE_DIR}/cmake/dtrace-instr-link.pl ${LCB_DTRACE_SRC}")
     SET_TARGET_PROPERTIES(couchbase PROPERTIES RULE_LAUNCH_LINK ${_lcb_linkhook})
-    SET_TARGET_PROPERTIES(couchbaseS PROPERTIES RULE_LAUNCH_LINK ${_lcb_linkhook})
+    IF(ENABLE_STATIC)
+        SET_TARGET_PROPERTIES(couchbaseS PROPERTIES RULE_LAUNCH_LINK ${_lcb_linkhook})
+    ENDIF()
 ENDIF()
 
 # This is done to be ABI-compatible with Autotools. On OSX, autotools generates
@@ -296,15 +301,15 @@ ELSE()
     SET(LCB_SNAPPY_LINK "")
 ENDIF()
 
-SET_TARGET_PROPERTIES(couchbase PROPERTIES PREFIX "lib")
-SET_TARGET_PROPERTIES(couchbase PROPERTIES IMPORT_PREFIX "lib")
 SET(LCB_LINK_DEPS ${lcb_plat_libs} ${lcb_ssl_libs} ${LCB_SNAPPY_LINK})
 IF(CMAKE_SYSTEM_NAME STREQUAL "Linux")
     SET(LCB_LINK_DEPS ${LCB_LINK_DEPS} rt)
 ENDIF()
 
 TARGET_LINK_LIBRARIES(couchbase ${LCB_LINK_DEPS})
-TARGET_LINK_LIBRARIES(couchbaseS ${LCB_LINK_DEPS})
+IF(ENABLE_STATIC)
+    TARGET_LINK_LIBRARIES(couchbaseS ${LCB_LINK_DEPS})
+ENDIF()
 
 ENABLE_TESTING()
 IF(NOT LCB_NO_TESTS)
--- origsrc/libcouchbase-2.4.7/plugins/io/libev/CMakeLists.txt	2015-02-16 11:43:00.000000000 +0900
+++ src/libcouchbase-2.4.7/plugins/io/libev/CMakeLists.txt	2015-03-17 23:42:24.610706900 +0900
@@ -12,8 +12,6 @@ ENDIF()
 ADD_LIBRARY(couchbase_libev SHARED plugin-libev.c)
 SET_TARGET_PROPERTIES(couchbase_libev
     PROPERTIES
-    PREFIX "lib"
-    IMPORT_PREFIX "lib"
     COMPILE_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_CORE_CFLAGS} -std=gnu99")
 TARGET_LINK_LIBRARIES(couchbase_libev ${LIBEV_LIBRARIES})
 INCLUDE_DIRECTORIES(AFTER ${LIBEV_INCLUDE_DIR})
@@ -26,4 +24,5 @@ ENDIF()
 
 INSTALL(TARGETS
     couchbase_libev
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
--- origsrc/libcouchbase-2.4.7/plugins/io/libevent/CMakeLists.txt	2015-02-16 11:43:00.000000000 +0900
+++ src/libcouchbase-2.4.7/plugins/io/libevent/CMakeLists.txt	2015-03-17 23:42:33.479847500 +0900
@@ -20,10 +20,9 @@ ENDIF()
 ADD_LIBRARY(couchbase_libevent SHARED plugin-libevent.c)
 SET_TARGET_PROPERTIES(couchbase_libevent
     PROPERTIES
-    PREFIX "lib"
-    IMPORT_PREFIX "lib"
     COMPILE_FLAGS "${CMAKE_C_FLAGS} ${LCB_CORE_CFLAGS}")
 TARGET_LINK_LIBRARIES(couchbase_libevent ${LIBEVENT_LIBRARIES})
 INSTALL(TARGETS
     couchbase_libevent
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
