--- origsrc/libcouchbase-2.4.3/Makefile.am	2014-10-22 04:29:47.000000000 +0900
+++ src/libcouchbase-2.4.3/Makefile.am	2014-10-23 22:33:04.025738000 +0900
@@ -169,7 +169,7 @@ endif
 if HAVE_LIBEVENT
 lib_LTLIBRARIES += libcouchbase_libevent.la
 libcouchbase_libevent_la_SOURCES = plugins/io/libevent/plugin-libevent.c
-libcouchbase_libevent_la_LDFLAGS=-avoid-version $(AM_LDFLAGS) $(AM_PROFILE_SOLDFLAGS)
+libcouchbase_libevent_la_LDFLAGS=-avoid-version $(AM_LDFLAGS) $(AM_PROFILE_SOLDFLAGS) -no-undefined
 libcouchbase_libevent_la_LIBADD=-levent
 libcouchbase_libevent_la_CPPFLAGS=$(AM_CPPFLAGS) $(CPPFLAGS) -DLIBCOUCHBASE_INTERNAL=1
 if BUILD_EXAMPLES
@@ -182,7 +182,7 @@ endif
 if HAVE_LIBEV
 lib_LTLIBRARIES += libcouchbase_libev.la
 libcouchbase_libev_la_SOURCES = plugins/io/libev/plugin-libev.c
-libcouchbase_libev_la_LDFLAGS=-avoid-version $(AM_LDFLAGS) $(AM_PROFILE_SOLDFLAGS)
+libcouchbase_libev_la_LDFLAGS=-avoid-version $(AM_LDFLAGS) $(AM_PROFILE_SOLDFLAGS) -no-undefined
 libcouchbase_libev_la_LIBADD=-lev
 libcouchbase_libev_la_CPPFLAGS=$(AM_CPPFLAGS) $(CPPFLAGS) -DLIBCOUCHBASE_INTERNAL=1
 endif
@@ -300,9 +300,9 @@ tests_check_all_CPPFLAGS = $(AM_CPPFLAGS
 tests_check_all_LDADD = libcouchbase.la libcliopts.la
 
 tools_cbc_SOURCES =	tools/cbc.cc tools/cbc-handlers.h
-tools_cbc_LDADD = libcouchbase.la liblcbtools.la libcliopts.la
+tools_cbc_LDADD = liblcbtools.la libcliopts.la libcouchbase.la
 tools_cbc_pillowfight_SOURCES = tools/cbc-pillowfight.cc
-tools_cbc_pillowfight_LDADD = libcouchbase.la liblcbtools.la libcliopts.la liblcbutils.la -lpthread
+tools_cbc_pillowfight_LDADD = liblcbtools.la libcliopts.la liblcbutils.la -lpthread libcouchbase.la
 
 
 TESTS_ENVIRONMENT = export outdir="$(top_builddir)/tests";
--- origsrc/libcouchbase-2.4.3/configure.ac	2014-10-22 04:29:47.000000000 +0900
+++ src/libcouchbase-2.4.3/configure.ac	2014-10-23 22:12:31.675523400 +0900
@@ -194,10 +194,12 @@ AC_ARG_WITH([couchbasemock],
 
 AC_CHECK_HEADERS_ONCE([mach/mach_time.h sys/socket.h sys/time.h
                        netinet/in.h inttypes.h netdb.h unistd.h
-                       ws2tcpip.h winsock2.h event.h stdint.h
+                       event.h stdint.h
                        sys/uio.h sys/types.h fcntl.h dlfcn.h
                        ev.h libev/ev.h sys/sdt.h limits.h stdarg.h])
-
+if test "$host_os" != "cygwin"; then
+    AC_CHECK_HEADERS([ws2tcpip.h winsock2.h])
+fi
 AS_IF([test "x$ac_cv_header_stdint_h" != "xyes"],
       [AC_MSG_ERROR(Failed to locate stdint.h)])
 
@@ -215,6 +217,7 @@ AS_IF([test "x$ac_cv_enable_plugins" = "
       [AC_DEFINE([BUILD_PLUGINS], [1], [Build plugins])])
 
 dnl Check that we can use winsock2 plugin
+if test "$host_os" != "cygwin"; then
 AC_CACHE_CHECK([for winsock2], [ac_cv_have_winsock2],
   [ saved_libs="$LIBS"
     LIBS=-lws2_32
@@ -230,7 +233,8 @@ AC_CACHE_CHECK([for winsock2], [ac_cv_ha
       ])
     LIBS="$saved_libs"
   ])
-AM_CONDITIONAL(HAVE_WINSOCK2, [test "$ac_cv_have_winsock2" = "yes"])
+fi
+AM_CONDITIONAL(HAVE_WINSOCK2, [test x"$ac_cv_have_winsock2" = "xyes"])
 
 dnl Check that we've got a usable libev3
 AC_CACHE_CHECK([for libev3], [ac_cv_have_libev3],
