--- origsrc/libbios/Makefile.am	2014-07-06 15:24:20.000000000 +0900
+++ src/libbios/Makefile.am	2014-07-06 15:25:04.463325600 +0900
@@ -55,6 +55,7 @@ libbios_la_SOURCES = \
     bios/stringUtil.c
 
 libbios_la_LIBADD = -lm -lgsl
+libbios_la_LDFLAGS = -no-undefined
 
 nobase_dist_include_HEADERS = \
 	bios/args.h \
--- origsrc/libbios/bios/types.h	2014-07-06 15:24:20.000000000 +0900
+++ src/libbios/bios/types.h	2014-07-06 15:28:14.589302100 +0900
@@ -10,7 +10,9 @@
 /**
  * Get offset of a member
  */
+#if !defined(__CYGWIN__)
 #define offsetof(TYPE, MEMBER) ((size_t) &((TYPE *)0)->MEMBER)
+#endif
 
 /**
  * container_of - cast a member of a structure out to the containing structure
