--- origsrc/libbios/Makefile.am	2024-11-19 08:29:44.000000000 +0900
+++ src/libbios/Makefile.am	2024-11-19 08:29:45.580894900 +0900
@@ -55,6 +55,7 @@ libbios_la_SOURCES = \
     bios/stringUtil.c
 
 libbios_la_LIBADD = -lm -lgsl
+libbios_la_LDFLAGS = -no-undefined
 
 nobase_dist_include_HEADERS = \
 	bios/args.h \
--- origsrc/libbios/bios/types.h	2024-11-19 08:29:44.000000000 +0900
+++ src/libbios/bios/types.h	2024-11-19 08:30:04.041466000 +0900
@@ -10,7 +10,9 @@
 /**
  * Get offset of a member
  */
+#if !defined(offsetof)
 #define offsetof(TYPE, MEMBER) ((size_t) &((TYPE *)0)->MEMBER)
+#endif
 
 /**
  * container_of - cast a member of a structure out to the containing structure
--- origsrc/libbios/bios.pc.in	2024-11-19 08:29:44.000000000 +0900
+++ src/libbios/bios.pc.in	2024-11-19 08:36:43.242955300 +0900
@@ -5,7 +5,7 @@ includedir=@includedir@
 
 Name: @PACKAGE_NAME@
 Description: A library of data structures and algorithms for bioinformatics
-Version: @PACKAGE_VERSION@
+Version: @VERSION@
 URL: @PACKAGE_URL@
 Requries.private: gsl
 Cflags: -I${includedir}
