--- origsrc/e16-1.0.23/lib/Makefile.am	2020-05-06 14:52:54.000000000 +0900
+++ src/e16-1.0.23/lib/Makefile.am	2021-05-07 13:36:44.039379500 +0900
@@ -14,7 +14,7 @@ LIB_HACK = libhack.la
 libhack_la_SOURCES  = e16_hack.c
 
 libhack_la_LIBADD   = $(DLOPEN_LIBS)
-libhack_la_LDFLAGS  = -module -avoid-version
+libhack_la_LDFLAGS  = -no-undefined -module -avoid-version
 
 endif
 
@@ -25,7 +25,7 @@ LIB_TRIP = libtrip.la
 libtrip_la_SOURCES  = e16_trip.c
 
 libtrip_la_LIBADD   = $(DLOPEN_LIBS)
-libtrip_la_LDFLAGS  = -module -avoid-version
+libtrip_la_LDFLAGS  = -no-undefined -module -avoid-version
 
 endif
 
--- origsrc/e16-1.0.23/src/xwin.h	2021-04-02 15:03:26.000000000 +0900
+++ src/e16-1.0.23/src/xwin.h	2021-05-07 13:30:21.635509900 +0900
@@ -232,6 +232,10 @@ int                 EQueryPointer(Win wi
 
 unsigned int        EAllocColor(EX_Colormap cmap, unsigned int argb);
 
+#if defined(_B)
+#undef _B
+#endif
+
 #define _A(x)   (((x) >> 24) & 0xff)
 #define _R(x)   (((x) >> 16) & 0xff)
 #define _G(x)   (((x) >>  8) & 0xff)
