--- origsrc/cdo-1.9.4/Makefile.am	2018-05-09 17:49:58.000000000 +0900
+++ src/cdo-1.9.4/Makefile.am	2018-07-13 11:03:45.142930500 +0900
@@ -1,7 +1,7 @@
 # Process this file with automake to produce Makefile.in
 SUBDIRS = libcdi src/lib/ncl src/lib/yac src contrib test/data test
 if MAINTAINER_MODE
-SUBDIRS += test/bandit_tests
+# SUBDIRS += test/bandit_tests
 endif
 #
 ACLOCAL_AMFLAGS = -I m4 -I libcdi/m4
--- origsrc/cdo-1.9.4/libcdi/app/cdi.c	2018-04-03 16:29:04.000000000 +0900
+++ src/cdo-1.9.4/libcdi/app/cdi.c	2018-07-13 11:03:45.142930500 +0900
@@ -56,9 +56,10 @@ void cdiDefTableID(int tableID);
 
 int getopt(int argc, char *const argv[], const char *optstring);
 
+#if !defined(__CYGWIN__)
 extern char *optarg;
 extern int optind, opterr, optopt;
-
+#endif
 
 static char *Progname;
 
--- origsrc/cdo-1.9.4/libcdi/m4/acx_options.m4	2018-04-25 20:33:54.000000000 +0900
+++ src/cdo-1.9.4/libcdi/m4/acx_options.m4	2018-07-13 11:03:45.158675000 +0900
@@ -35,7 +35,7 @@ AC_ARG_WITH([szlib],
             [AS_CASE(["$with_szlib"],
                      [no],[AC_MSG_CHECKING([for szlib library])
                            AC_MSG_RESULT([suppressed])],
-                     [yes],[AC_CHECK_HEADERS(szlib.h)
+                     [yes],[AC_CHECK_HEADERS(szip/szlib.h)
                             AC_SEARCH_LIBS([SZ_BufftoBuffCompress],
                                            [sz],
                                            [AC_DEFINE([HAVE_LIBSZ],[1],[Define to 1 for SZIP support])],
--- origsrc/cdo-1.9.4/libcdi/src/Makefile.am	2018-01-08 16:55:44.000000000 +0900
+++ src/cdo-1.9.4/libcdi/src/Makefile.am	2018-07-13 11:15:47.468538900 +0900
@@ -7,6 +7,7 @@ noinst_LTLIBRARIES = libcdiresunpack.la
 if ENABLE_CDI_LIB
   lib_LTLIBRARIES = libcdi.la
   include_HEADERS = cdi.h cdi.inc
+  libcdi_la_LDFLAGS = -no-undefined
 else
   noinst_LTLIBRARIES += libcdi.la
 endif
@@ -169,6 +170,7 @@ if USE_MPI
 if ENABLE_CDI_LIB
 lib_LTLIBRARIES += libcdipio.la
 include_HEADERS += cdipio.h cdipio.inc
+libcdipio_la_LDFLAGS = -no-undefined
 else
 noinst_LTLIBRARIES += libcdipio.la
 endif
@@ -224,7 +226,7 @@ endif
 
 libcdipio_la_LIBADD    = libcdi.la $(PPM_CORE_LIBS) $(YAXT_LIBS) $(MPI_C_LIB)
 
-libcdi_la_LIBADD    = $(UUID_C_LIB)
+libcdi_la_LIBADD    = $(UUID_C_LIB) -lhdf5
 #
 #cdilib.c:
 #	$(top_srcdir)/src/make_cdilib $(top_srcdir)/src
--- origsrc/cdo-1.9.4/libcdi/src/cgribexlib.c	2018-03-12 16:52:49.000000000 +0900
+++ src/cdo-1.9.4/libcdi/src/cgribexlib.c	2018-07-13 11:03:45.189804400 +0900
@@ -5772,7 +5772,7 @@ void gribRepair1(int nrec, long recsize,
 #if defined(__cplusplus)
 extern "C" {
 #endif
-#include <szlib.h>
+#include <szip/szlib.h>
 #ifdef  __cplusplus
 }
 #endif
--- origsrc/cdo-1.9.4/m4/acx_options.m4	2018-05-09 17:49:58.000000000 +0900
+++ src/cdo-1.9.4/m4/acx_options.m4	2018-07-13 11:03:45.205429500 +0900
@@ -40,7 +40,7 @@ AC_ARG_WITH([szlib],
             [AS_CASE(["$with_szlib"],
                      [no],[AC_MSG_CHECKING([for szlib library])
                            AC_MSG_RESULT([suppressed])],
-                     [yes],[AC_CHECK_HEADERS(szlib.h)
+                     [yes],[AC_CHECK_HEADERS(szip/szlib.h)
                             AC_SEARCH_LIBS([SZ_BufftoBuffCompress],
                                            [sz],
                                            [AC_DEFINE([HAVE_LIBSZ],[1],[Define to 1 for SZIP support])],
