--- origsrc/cdo-1.9.8/Makefile.am	2019-10-18 17:22:52.000000000 +0900
+++ src/cdo-1.9.8/Makefile.am	2020-01-11 17:53:43.683259900 +0900
@@ -1,7 +1,7 @@
 # Process this file with automake to produce Makefile.in
 SUBDIRS = libcdi src/lib/ncl src/lib/yac  src/lib/gradsdes src contrib test/data test
 if MAINTAINER_MODE
-SUBDIRS += test/bandit_tests test/executeOnly
+SUBDIRS += # test/bandit_tests test/executeOnly
 endif
 #
 ACLOCAL_AMFLAGS = -I m4 -I libcdi/m4
--- origsrc/cdo-1.9.8/libcdi/app/cdi.c	2019-10-04 15:05:48.000000000 +0900
+++ src/cdo-1.9.8/libcdi/app/cdi.c	2020-01-11 17:14:27.399457900 +0900
@@ -54,11 +54,7 @@ void cdiDefTableID(int tableID);
 }
 #endif
 
-int getopt(int argc, char *const argv[], const char *optstring);
-
-extern char *optarg;
-extern int optind, opterr, optopt;
-
+#include <getopt.h>
 
 static char *Progname;
 
--- origsrc/cdo-1.9.8/libcdi/src/Makefile.am	2019-08-29 15:10:55.000000000 +0900
+++ src/cdo-1.9.8/libcdi/src/Makefile.am	2020-01-11 17:15:52.707125600 +0900
@@ -10,12 +10,15 @@ lib_LTLIBRARIES =
 if ENABLE_CDI_LIB
   lib_LTLIBRARIES += libcdi.la
   include_HEADERS += cdi.h
+  libcdi_la_LDFLAGS = -no-undefined
+  libcdi_la_LIBADD = -lhdf5
 # add cdi.inc only if cfortran interface is required
 if USE_FC
   include_HEADERS += cdi.inc
 endif
 if CREATE_ISOC
   lib_LTLIBRARIES += libcdi_f2003.la
+  libcdi_f2003_la_LDFLAGS = -no-undefined
 endif
 else
   noinst_LTLIBRARIES += libcdi.la
@@ -197,6 +200,7 @@ if USE_MPI
 if ENABLE_CDI_LIB
 lib_LTLIBRARIES += libcdipio.la
 include_HEADERS += cdipio.h cdipio.inc
+libcdipio_la_LDFLAGS = -no-undefined
 else
 noinst_LTLIBRARIES += libcdipio.la
 endif
--- origsrc/cdo-1.9.8/libcdi/src/cgribexlib.c	2019-10-29 21:53:10.000000000 +0900
+++ src/cdo-1.9.8/libcdi/src/cgribexlib.c	2020-01-11 17:16:23.522980300 +0900
@@ -5744,7 +5744,7 @@ void gribRepair1(int nrec, long recsize,
 #if defined(__cplusplus)
 extern "C" {
 #endif
-#include <szlib.h>
+#include <szip/szlib.h>
 #ifdef  __cplusplus
 }
 #endif
--- origsrc/cdo-1.9.8/m4/acx_options.m4	2019-08-06 16:05:11.000000000 +0900
+++ src/cdo-1.9.8/m4/acx_options.m4	2020-01-11 17:16:54.604940000 +0900
@@ -40,7 +40,7 @@ AC_ARG_WITH([szlib],
             [AS_CASE(["$with_szlib"],
                      [no],[AC_MSG_CHECKING([for szlib library])
                            AC_MSG_RESULT([suppressed])],
-                     [yes],[AC_CHECK_HEADERS(szlib.h)
+                     [yes],[AC_CHECK_HEADERS(szip/szlib.h)
                             AC_SEARCH_LIBS([SZ_BufftoBuffCompress],
                                            [sz],
                                            [AC_DEFINE([HAVE_LIBSZ],[1],[Define to 1 for SZIP support])],
--- origsrc/cdo-1.9.8/src/cdo.cc	2019-10-22 21:23:15.000000000 +0900
+++ src/cdo-1.9.8/src/cdo.cc	2020-01-11 17:17:38.044342200 +0900
@@ -157,7 +157,7 @@ cdo_stackframe()
 static int
 cdo_feenableexcept(int excepts)
 {
-#if defined HAVE_FEENABLEEXCEPT
+#if defined HAVE_FEENABLEEXCEPT && !defined(__CYGWIN__)
   int feenableexcept(int);
   int old_excepts = feenableexcept(excepts);
   return old_excepts;
