--- origsrc/cdo-1.6.1/libcdi/src/Makefile.am	2013-05-28 23:23:40.000000000 +0900
+++ src/cdo-1.6.1/libcdi/src/Makefile.am	2013-08-03 15:07:10.423035600 +0900
@@ -5,6 +5,7 @@ EXTRA_DIST = cdilib.c mo_cdi.f90
 if ENABLE_CDI_LIB
   lib_LTLIBRARIES = libcdi.la
   include_HEADERS = cdi.h cdi.inc
+  libcdi_la_LDFLAGS = -no-undefined
 else
   noinst_LTLIBRARIES = libcdi.la
 endif
--- origsrc/cdo-1.6.1/libcdi/src/cgribexlib.c	2013-06-27 22:40:57.000000000 +0900
+++ src/cdo-1.6.1/libcdi/src/cgribexlib.c	2013-08-03 15:07:10.423035600 +0900
@@ -9902,7 +9902,7 @@ extern "C" {
 #if defined (HAVE_LIBAEC)
 #  include <libaec.h>
 #else
-#  include <szlib.h>
+#  include <szip/szlib.h>
 #endif
 #if defined (__cplusplus)
 }
