diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libcint-2.5/CMakeLists.txt src/libcint-2.5/CMakeLists.txt
--- origsrc/libcint-2.5/CMakeLists.txt	2015-01-05 13:19:54.000000000 +0900
+++ src/libcint-2.5/CMakeLists.txt	2015-08-18 18:11:49.732639900 +0900
@@ -44,11 +44,12 @@ else()
 
   set_target_properties(cint PROPERTIES
     VERSION ${cint_VERSION}
+    SOVERSION 0
     LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR})
 
   target_link_libraries(cint ${BLAS_LIBRARIES})
 
-  install(TARGETS cint DESTINATION lib)
+  install(TARGETS cint RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
   install(FILES ${PROJECT_SOURCE_DIR}/include/cint.h DESTINATION include)
 
   if (${ENABLE_EXAMPLE})
@@ -65,3 +66,10 @@ else()
   endif()
 
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libcint.pc.in ${CMAKE_BINARY_DIR}/libcint.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libcint.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libcint-2.5/libcint.pc.in src/libcint-2.5/libcint.pc.in
--- origsrc/libcint-2.5/libcint.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libcint-2.5/libcint.pc.in	2015-08-18 18:10:56.703799100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libcint
+Description: General GTO integrals for quantum chemistry
+Version: @cint_VERSION@
+Libs: -L${libdir} -lcint
+Cflags: -I${includedir}
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libcint-2.5/src/rys_roots.c src/libcint-2.5/src/rys_roots.c
--- origsrc/libcint-2.5/src/rys_roots.c	2015-01-05 13:19:54.000000000 +0900
+++ src/libcint-2.5/src/rys_roots.c	2015-08-18 18:09:49.100974400 +0900
@@ -19,6 +19,10 @@
 //#define MXROOTS1  (MXROOTS+1)
 #define MXROOTS1  MXROOTS
 
+#if defined(__CYGWIN__)
+#define expl(x) exp(x)
+#endif
+
 // FIXME: R_dsmit causes big errors for nroots > 13
 void CINTrys_roots(FINT nroots, double x, double *u, double *w)
 {
@@ -2005,4 +2009,3 @@ L70:
     }
     return;
 }
-
