--- origsrc/libcint-5.3.0/CMakeLists.txt	2023-03-29 08:59:31.000000000 +0900
+++ src/libcint-5.3.0/CMakeLists.txt	2023-04-03 13:00:25.207805300 +0900
@@ -177,8 +177,10 @@ set(CintHeaders
   ${PROJECT_SOURCE_DIR}/include/cint_funcs.h
   ${PROJECT_BINARY_DIR}/include/cint.h)
 
-install(TARGETS cint DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}" COMPONENT "lib")
-install(FILES ${CintHeaders} DESTINATION ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR} COMPONENT "dev")
+  install(TARGETS cint
+    RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}"
+    ARCHIVE DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}" COMPONENT "lib")
+  install(FILES ${CintHeaders} DESTINATION ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR} COMPONENT "dev")
 
 
 if(ENABLE_EXAMPLE)
@@ -240,3 +242,10 @@ endif()
 set(CPACK_COMPONENTS_GROUPING "IGNORE")
 
 include(CPack)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libcint.pc.in ${CMAKE_BINARY_DIR}/libcint.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libcint.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libcint-5.3.0/examples/time_c2h6.c	2023-03-29 08:59:31.000000000 +0900
+++ src/libcint-5.3.0/examples/time_c2h6.c	2023-04-03 13:00:25.213772400 +0900
@@ -716,9 +716,15 @@ void run_all(int *atm, int natm, int *ba
 
         printf("\tcint2e_sph without optimizer: total num ERI = %.2e\n", tot);
         pct = 0; count = 0;
+#if defined(stdout)
+#pragma omp parallel default(none) \
+        shared(atm, natm, bas, nbas, env, ishls, jshls, non_opt, time0, pct, count) \
+        private(di, dj, dk, dl, i, j, k, l, ij, kl, kl_max, shls, buf, time1)
+#else
 #pragma omp parallel default(none) \
         shared(atm, natm, bas, nbas, env, ishls, jshls, non_opt, time0, pct, count, stdout) \
         private(di, dj, dk, dl, i, j, k, l, ij, kl, kl_max, shls, buf, time1)
+#endif
 #pragma omp for nowait schedule(dynamic, 2)
         for (ij = 0; ij < nbas*(nbas+1)/2; ij++) {
                 i = ishls[ij];
@@ -757,9 +763,15 @@ void run_all(int *atm, int natm, int *ba
         printf("\tcint2e_sph with optimizer: total num ERI = %.2e\n", tot);
         pct = 0; count = 0;
 
+#if defined(stdout)
+#pragma omp parallel default(none) \
+        shared(atm, natm, bas, nbas, env, ishls, jshls, opt_for_cint2e, time0, pct, count) \
+        private(di, dj, dk, dl, i, j, k, l, ij, kl, kl_max, shls, buf, time1)
+#else
 #pragma omp parallel default(none) \
         shared(atm, natm, bas, nbas, env, ishls, jshls, opt_for_cint2e, time0, pct, count, stdout) \
         private(di, dj, dk, dl, i, j, k, l, ij, kl, kl_max, shls, buf, time1)
+#endif
 #pragma omp for nowait schedule(dynamic, 2)
         for (ij = 0; ij < nbas*(nbas+1)/2; ij++) {
                 i = ishls[ij];
@@ -799,9 +811,16 @@ void run_all(int *atm, int natm, int *ba
         tot = (double)ncgto*ncgto*ncgto*ncgto/2*3;
         printf("\tGradients with optimizer: total num ERI = %.2e\n", tot);
         pct = 0; count = 0;
+
+#if defined(stdout)
+#pragma omp parallel default(none) \
+        shared(atm, natm, bas, nbas, env, ishls, jshls, opt_for_ip1, time0, pct, count) \
+        private(di, dj, dk, dl, i, j, k, l, ij, kl, shls, buf, time1)
+#else
 #pragma omp parallel default(none) \
         shared(atm, natm, bas, nbas, env, ishls, jshls, opt_for_ip1, time0, pct, count, stdout) \
         private(di, dj, dk, dl, i, j, k, l, ij, kl, shls, buf, time1)
+#endif
 #pragma omp for nowait schedule(dynamic, 2)
         for (ij = 0; ij < nbas*nbas; ij++) {
                 i = ij / nbas;
--- origsrc/libcint-5.3.0/examples/time_c60.c	2023-03-29 08:59:31.000000000 +0900
+++ src/libcint-5.3.0/examples/time_c60.c	2023-04-03 13:00:25.219887900 +0900
@@ -193,9 +193,15 @@ void run_all(int *atm, int natm, int *ba
         cint2e_sph_optimizer(&opt, atm, natm, bas, nbas, env);
 
         pct = 0; count = 0;
+#if defined(stdout)
+#pragma omp parallel default(none) \
+        shared(atm, natm, bas, nbas, env, ishls, jshls, opt, time0, pct, count) \
+        private(di, dj, dk, dl, i, j, k, l, ij, kl, kl_max, shls, buf, time1)
+#else
 #pragma omp parallel default(none) \
         shared(atm, natm, bas, nbas, env, ishls, jshls, opt, time0, pct, count, stdout) \
         private(di, dj, dk, dl, i, j, k, l, ij, kl, kl_max, shls, buf, time1)
+#endif
 #pragma omp for nowait schedule(dynamic, 2)
         for (ij = 0; ij < nbas*(nbas+1)/2; ij++) {
                 i = ishls[ij];
--- origsrc/libcint-5.3.0/libcint.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libcint-5.3.0/libcint.pc.in	2023-04-03 13:00:25.226058200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libcint
+Description: General GTO integrals for quantum chemistry
+Version: @cint_VERSION@
+Libs: -L${libdir} -lcint
+Cflags: -I${includedir}
--- origsrc/libcint-5.3.0/testsuite/test_3c2e.py	2023-03-29 08:59:31.000000000 +0900
+++ src/libcint-5.3.0/testsuite/test_3c2e.py	2023-04-03 13:00:25.285834400 +0900
@@ -13,8 +13,10 @@ import os
 import ctypes
 import numpy
 
-_cint = numpy.ctypeslib.load_library('libcint', os.path.abspath(os.path.join(__file__, '../../build')))
-
+if sys.platform == 'cygwin':
+    _cint = numpy.ctypeslib.load_library('cygcint-5.dll', '.')
+else:
+    _cint = numpy.ctypeslib.load_library('libcint', os.path.abspath(os.path.join(__file__, '../../build')))
 
 PTR_LIGHT_SPEED    = 0
 PTR_COMMON_ORIG    = 1
--- origsrc/libcint-5.3.0/testsuite/test_cint.py	2023-03-29 08:59:31.000000000 +0900
+++ src/libcint-5.3.0/testsuite/test_cint.py	2023-04-03 13:00:25.289841000 +0900
@@ -13,8 +13,10 @@ import os
 import ctypes
 import numpy
 
-_cint = numpy.ctypeslib.load_library('libcint', os.path.abspath(os.path.join(__file__, '../../build')))
-
+if sys.platform == 'cygwin':
+    _cint = numpy.ctypeslib.load_library('cygcint-5.dll', '.')
+else:
+    _cint = numpy.ctypeslib.load_library('libcint', os.path.abspath(os.path.join(__file__, '../../build')))
 
 PTR_EXPCUTOFF      = 0
 PTR_COMMON_ORIG    = 1
