--- origsrc/libcint-3.0.20/CMakeLists.txt	2020-05-14 15:46:35.000000000 +0900
+++ src/libcint-3.0.20/CMakeLists.txt	2020-06-10 10:22:28.733249500 +0900
@@ -111,7 +111,9 @@ if(QUADMATH_FOUND)
 endif()
 target_link_libraries(cint "-lm")
 
-install(TARGETS cint DESTINATION ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
+install(TARGETS cint 
+  RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
 set(CintHeaders
   ${PROJECT_SOURCE_DIR}/include/cint_funcs.h
   ${PROJECT_BINARY_DIR}/include/cint.h)
@@ -139,3 +141,10 @@ if(BUILD_SHARED_LIBS AND ENABLE_TEST)
       COMMAND ${PYTHON_EXECUTABLE} ${PROJECT_SOURCE_DIR}/testsuite/test_3c2e.py ${RUN_QUICK_TEST})
   endif()
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libcint.pc.in ${CMAKE_BINARY_DIR}/libcint.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libcint.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libcint-3.0.20/examples/time_c2h6.c	2020-05-14 15:46:35.000000000 +0900
+++ src/libcint-3.0.20/examples/time_c2h6.c	2020-06-10 10:22:28.748871000 +0900
@@ -716,9 +716,15 @@ void run_all(int *atm, int natm, int *ba
 
         printf("\tcint2e_sph without optimizer: total num ERI = %.2e\n", tot);
         pct = 0; count = 0;
+#if defined(__CYGWIN__)
+#pragma omp parallel default(none) \
+        shared(atm, natm, bas, nbas, env, ishls, jshls, non_opt, time0, pct, count) \
+        private(di, dj, dk, dl, i, j, k, l, ij, kl, kl_max, shls, buf, time1)
+#else
 #pragma omp parallel default(none) \
         shared(atm, natm, bas, nbas, env, ishls, jshls, non_opt, time0, pct, count, stdout) \
         private(di, dj, dk, dl, i, j, k, l, ij, kl, kl_max, shls, buf, time1)
+#endif
 #pragma omp for nowait schedule(dynamic, 2)
         for (ij = 0; ij < nbas*(nbas+1)/2; ij++) {
                 i = ishls[ij];
@@ -757,9 +763,15 @@ void run_all(int *atm, int natm, int *ba
         printf("\tcint2e_sph with optimizer: total num ERI = %.2e\n", tot);
         pct = 0; count = 0;
 
+#if defined(__CYGWIN__)
+#pragma omp parallel default(none) \
+        shared(atm, natm, bas, nbas, env, ishls, jshls, opt_for_cint2e, time0, pct, count) \
+        private(di, dj, dk, dl, i, j, k, l, ij, kl, kl_max, shls, buf, time1)
+#else
 #pragma omp parallel default(none) \
         shared(atm, natm, bas, nbas, env, ishls, jshls, opt_for_cint2e, time0, pct, count, stdout) \
         private(di, dj, dk, dl, i, j, k, l, ij, kl, kl_max, shls, buf, time1)
+#endif
 #pragma omp for nowait schedule(dynamic, 2)
         for (ij = 0; ij < nbas*(nbas+1)/2; ij++) {
                 i = ishls[ij];
@@ -799,9 +811,15 @@ void run_all(int *atm, int natm, int *ba
         tot = (double)ncgto*ncgto*ncgto*ncgto/2*3;
         printf("\tGradients with optimizer: total num ERI = %.2e\n", tot);
         pct = 0; count = 0;
+#if defined(__CYGWIN__)
+#pragma omp parallel default(none) \
+        shared(atm, natm, bas, nbas, env, ishls, jshls, opt_for_ip1, time0, pct, count) \
+        private(di, dj, dk, dl, i, j, k, l, ij, kl, shls, buf, time1)
+#else
 #pragma omp parallel default(none) \
         shared(atm, natm, bas, nbas, env, ishls, jshls, opt_for_ip1, time0, pct, count, stdout) \
         private(di, dj, dk, dl, i, j, k, l, ij, kl, shls, buf, time1)
+#endif
 #pragma omp for nowait schedule(dynamic, 2)
         for (ij = 0; ij < nbas*nbas; ij++) {
                 i = ij / nbas;
--- origsrc/libcint-3.0.20/examples/time_c60.c	2020-05-14 15:46:35.000000000 +0900
+++ src/libcint-3.0.20/examples/time_c60.c	2020-06-10 10:22:28.842788700 +0900
@@ -193,9 +193,15 @@ void run_all(int *atm, int natm, int *ba
         cint2e_sph_optimizer(&opt, atm, natm, bas, nbas, env);
 
         pct = 0; count = 0;
+#if defined(__CYGWIN__)
+#pragma omp parallel default(none) \
+        shared(atm, natm, bas, nbas, env, ishls, jshls, opt, time0, pct, count) \
+        private(di, dj, dk, dl, i, j, k, l, ij, kl, kl_max, shls, buf, time1)
+#else
 #pragma omp parallel default(none) \
         shared(atm, natm, bas, nbas, env, ishls, jshls, opt, time0, pct, count, stdout) \
         private(di, dj, dk, dl, i, j, k, l, ij, kl, kl_max, shls, buf, time1)
+#endif
 #pragma omp for nowait schedule(dynamic, 2)
         for (ij = 0; ij < nbas*(nbas+1)/2; ij++) {
                 i = ishls[ij];
--- origsrc/libcint-3.0.20/libcint.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libcint-3.0.20/libcint.pc.in	2020-06-10 10:22:29.502893000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libcint
+Description: General GTO integrals for quantum chemistry
+Version: @cint_VERSION@
+Libs: -L${libdir} -lcint
+Cflags: -I${includedir}
