--- origsrc/vttest-20110705/aclocal.m4	2010-08-27 07:43:09.000000000 +0900
+++ src/vttest-20110705/aclocal.m4	2011-08-18 15:32:45.453398700 +0900
@@ -168,7 +168,7 @@ else
 		system_name="`(hostname) 2>/dev/null`"
 	fi
 fi
-test -n "$system_name" && AC_DEFINE_UNQUOTED(SYSTEM_NAME,"$system_name")
+test -n "$system_name" && AC_DEFINE_UNQUOTED([SYSTEM_NAME], "$system_name", [SYSTEM_NAME])
 AC_CACHE_VAL(cf_cv_system_name,[cf_cv_system_name="$system_name"])
 
 test -z "$system_name" && system_name="$cf_cv_system_name"
@@ -236,7 +236,7 @@ ioctl (0, FIONREAD, &l1);
 	[cf_cv_use_fionread=no])
 ])
 AC_MSG_RESULT($cf_cv_use_fionread)
-test $cf_cv_use_fionread = yes && AC_DEFINE(USE_FIONREAD)
+test $cf_cv_use_fionread = yes && AC_DEFINE([USE_FIONREAD], [], [USE_FIONREAD])
 ])dnl
 dnl ---------------------------------------------------------------------------
 dnl CF_GCC_ATTRIBUTES version: 13 updated: 2009/08/11 20:19:56
@@ -603,7 +603,7 @@ else
 	rm -f conftest CONFTEST
 fi
 ])
-test "$cf_cv_mixedcase" = yes && AC_DEFINE(MIXEDCASE_FILENAMES)
+test "$cf_cv_mixedcase" = yes && AC_DEFINE([MIXEDCASE_FILENAMES], [], [MIXEDCASE_FILENAMES])
 ])dnl
 dnl ---------------------------------------------------------------------------
 dnl CF_MSG_LOG version: 4 updated: 2007/07/29 09:55:12
@@ -728,7 +728,7 @@ this did not work
 	)])
 ])
 AC_MSG_RESULT($cf_cv_posix_vdisable)
-test $cf_cv_posix_vdisable = yes && AC_DEFINE(HAVE_POSIX_VDISABLE)
+test $cf_cv_posix_vdisable = yes && AC_DEFINE([HAVE_POSIX_VDISABLE], [], [HAVE_POSIX_VDISABLE])
 ])dnl
 dnl ---------------------------------------------------------------------------
 dnl CF_PROG_CC_U_D version: 1 updated: 2005/07/14 16:59:30
@@ -778,7 +778,7 @@ AC_OBJEXT
 
 PROG_EXT="$EXEEXT"
 AC_SUBST(PROG_EXT)
-test -n "$PROG_EXT" && AC_DEFINE_UNQUOTED(PROG_EXT,"$PROG_EXT")
+test -n "$PROG_EXT" && AC_DEFINE_UNQUOTED([PROG_EXT], "$PROG_EXT", [PROG_EXT])
 ])dnl
 dnl ---------------------------------------------------------------------------
 dnl CF_REMOVE_DEFINE version: 3 updated: 2010/01/09 11:05:50
