--- origsrc/munge-munge-0.5.17/src/libmunge/Makefile.am	2025-11-12 17:00:57.000000000 +0900
+++ src/munge-munge-0.5.17/src/libmunge/Makefile.am	2026-01-26 13:30:51.664429200 +0900
@@ -50,7 +50,7 @@ libmunge_la_CPPFLAGS = \
 	-I$(top_srcdir)/src/libcommon \
 	# End of libmunge_la_CPPFLAGS
 
-libmunge_la_LDFLAGS = \
+libmunge_la_LDFLAGS = -no-undefined \
 	-export-symbols-regex "^munge_.*" \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
 	# End of libmunge_la_LDFLAGS
