--- origsrc/libg15render-3.0.4/Makefile.am	2021-06-18 07:14:14.000000000 +0900
+++ src/libg15render-3.0.4/Makefile.am	2024-01-02 02:27:52.851240800 +0900
@@ -3,13 +3,15 @@ AM_CFLAGS=@CFLAGS@
 lib_LTLIBRARIES = libg15render.la
 libg15render_la_SOURCES = libg15render.h text.c pixel.c screen.c font_6x4.c font_7x5.c font_8x8.c
 libg15render_la_LIBADD = @FTLIB@
-libg15render_la_LDFLAGS = -version-info 3:0
+libg15render_la_LDFLAGS = -no-undefined -version-info 3:0
 include_HEADERS= libg15render.h
 EXTRA_DIST = debian libg15render.3
 man3_MANS = $(top_srcdir)/libg15render.3
-docdir = $(prefix)/share/doc/$(PACKAGE)-$(VERSION)
 doc_DATA = AUTHORS ChangeLog COPYING NEWS README
 ACLOCAL_AMFLAGS = -I m4
 
 dist-hook:
 	rm -rf `find $(distdir)/debian -name .svn`
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libg15render-3.0.4/configure.ac	2021-06-18 07:14:14.000000000 +0900
+++ src/libg15render-3.0.4/configure.ac	2024-01-02 02:27:52.866865000 +0900
@@ -48,4 +48,5 @@ AC_CHECK_FUNCS([memset])
 
 AC_SUBST(FTLIB)
 AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([libg15render.pc])
 AC_OUTPUT
--- origsrc/libg15render-3.0.4/libg15render.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libg15render-3.0.4/libg15render.pc.in	2024-01-02 02:30:17.607860700 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libg15render
+Description: Small library for display text and graphics on a Logitech G15 keyboard
+Version: @VERSION@
+Libs: -L${libdir} -lg15render
+Cflags: -I${includedir}
+Requires: freetype2 libg15
