--- origsrc/md4sum-0.02.03/Makefile.Linux	2007-04-12 06:15:46.000000000 +0900
+++ src/md4sum-0.02.03/Makefile.Linux	2024-10-31 05:31:27.544292300 +0900
@@ -3,8 +3,8 @@
 #
 
 CC=gcc
-CFLAGS=-Wall -Wstrict-prototypes -O2 -fomit-frame-pointer -pipe -I$(HDDIR) -DOS_TYPE_Linux -DOS_TYPE_STR=\"Linux\" -Dprg_name=\"$(PRG)\" -DMANUAL_NUMBER=$(MAN_NO)
-LDFLAGS=-Wl,-warn-common -s -L$(LIBDIR)
+CFLAGS=-Wall -Wstrict-prototypes -O2 -fomit-frame-pointer -pipe -I$(HDDIR) -DOS_TYPE_Linux -DOS_TYPE_STR=\"Linux\" -Dprg_name=\"$(PRG)\" -DMANUAL_NUMBER=$(MAN_NO) ${MY_CPPFLAGS} ${MY_CFLAGS}
+LDFLAGS=-Wl,-warn-common -s -L$(LIBDIR) ${MY_LDFLAGS}
 
 LDLIBS = 
 HDDIR = ./
