--- origsrc/libvc/Makefile.am	2019-12-22 19:51:55.000000000 +0900
+++ src/libvc/Makefile.am	2019-12-22 19:53:19.142826600 +0900
@@ -4,3 +4,6 @@
 SUBDIRS = src doc test
 AUTOMAKE_OPTIONS = dist-bzip2
 EXTRA_DIST = depcomp libvc.spec
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libvc/configure.ac	2019-12-22 19:51:55.000000000 +0900
+++ src/libvc/configure.ac	2019-12-22 19:53:19.158455100 +0900
@@ -33,5 +33,5 @@ AC_FUNC_MALLOC
 AC_CHECK_FUNCS([strdup strcasecmp])
 
 ### Output files
-AC_CONFIG_FILES([Makefile src/Makefile doc/Makefile test/Makefile])
+AC_CONFIG_FILES([libvc.pc Makefile src/Makefile doc/Makefile test/Makefile])
 AC_OUTPUT
--- origsrc/libvc/libvc.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libvc/libvc.pc.in	2019-12-22 19:53:19.095958100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libvc
+Description: A vCard library
+Version: @VERSION@
+Libs: -L${libdir} -lvc
+Cflags: -I${includedir}
--- origsrc/libvc/src/Makefile.am	2019-12-22 19:51:55.000000000 +0900
+++ src/libvc/src/Makefile.am	2019-12-22 19:52:23.130977200 +0900
@@ -7,4 +7,4 @@ AM_LFLAGS = -i
 include_HEADERS = vc.h
 lib_LTLIBRARIES = libvc.la
 libvc_la_SOURCES = vc.c vc_parse.y vc_scan.l
-libvc_la_LDFLAGS=-version-info $(LIBVC_ABI_VERSION)
+libvc_la_LDFLAGS = -no-undefined -version-info $(LIBVC_ABI_VERSION)
