--- origsrc/delly-2.5.1/Makefile	2026-07-23 21:30:25.000000000 +0900
+++ src/delly-2.5.1/Makefile	2026-08-04 14:26:17.206992500 +0900
@@ -20,7 +20,7 @@ bindir ?= $(exec_prefix)/bin
 # Flags
 CXX ?= g++
 CXXFLAGS += -std=c++17 -isystem ${HTSLIBINCDIR} -pedantic -W -Wall -Wno-unknown-pragmas -D__STDC_LIMIT_MACROS -fno-strict-aliasing -fpermissive -pthread
-LDFLAGS += -L${HTSLIBLIBDIR} -lboost_iostreams -lboost_filesystem -lboost_program_options -lboost_date_time -pthread
+LDFLAGS += -L${HTSLIBLIBDIR} -lboost_iostreams -lboost_filesystem -lboost_program_options -lboost_date_time -lboost_system -pthread
 
 # Boost location
 ifneq (${BOOSTINCDIR},)
