--- origsrc/delly-1.7.2/Makefile	2025-11-06 20:49:58.000000000 +0900
+++ src/delly-1.7.2/Makefile	2026-01-04 16:29:13.304921200 +0900
@@ -14,7 +14,7 @@ bindir ?= $(exec_prefix)/bin
 # Flags
 CXX ?= g++
 CXXFLAGS += -std=c++17 -isystem ${EBROOTHTSLIB} -pedantic -W -Wall -Wno-unknown-pragmas -D__STDC_LIMIT_MACROS -fno-strict-aliasing -fpermissive -pthread
-LDFLAGS += -L${EBROOTHTSLIB} -lboost_iostreams -lboost_filesystem -lboost_program_options -lboost_date_time -pthread
+LDFLAGS += -L${EBROOTHTSLIB} -lboost_iostreams -lboost_filesystem -lboost_program_options -lboost_date_time -lboost_system -pthread
 
 # Flags for static compile
 ifeq (${STATIC}, 1)
