--- origsrc/libultrahdr-1.4.0/CMakeLists.txt	2025-01-11 04:31:48.000000000 +0900
+++ src/libultrahdr-1.4.0/CMakeLists.txt	2026-02-06 15:45:34.305627400 +0900
@@ -38,6 +38,7 @@ elseif(${CMAKE_SYSTEM_NAME} MATCHES "Ems
 elseif(${CMAKE_SYSTEM_NAME} MATCHES "Android")
 elseif(WIN32)
 elseif(APPLE)
+elseif(CYGWIN)
 else()
   message(FATAL_ERROR "Platform ${CMAKE_SYSTEM_NAME} not recognized")
 endif()
@@ -766,7 +767,7 @@ endif()
 combine_static_libs(${UHDR_CORE_LIB_NAME} ${UHDR_TARGET_NAME})
 
 # Build static library as well
-if(BUILD_SHARED_LIBS)
+if(NOT BUILD_SHARED_LIBS)
   set(UHDR_TARGET_NAME_STATIC uhdr-static)
   add_library(${UHDR_TARGET_NAME_STATIC} STATIC)
   add_dependencies(${UHDR_TARGET_NAME_STATIC} ${UHDR_CORE_LIB_NAME})
