--- origsrc/freedroidrpg-0.14.1/Makefile.am	2010-11-27 07:47:26.000000000 +0900
+++ src/freedroidrpg-0.14.1/Makefile.am	2010-12-02 01:42:25.979832300 +0900
@@ -14,7 +14,7 @@ SUBDIRS = src gluem $(extradirs)
 
 freedroid_data = map graphics sound dialogs
 
-EXTRA_DIST = m4 $(freedroid_data) HELP_WANTED INSTALL.generic lua
+EXTRA_DIST = m4 $(freedroid_data) HELP_WANTED INSTALL.generic
 
 dist-hook:
 	make -C lua clean
@@ -26,8 +26,8 @@ dist-hook:
 install-data-local:
 	@echo "Installing the data-files ..."
 	$(mkinstalldirs) $(DESTDIR)$(pkgdatadir)
-	find $(srcdir)/$(freedroid_data) -type d ! -name "*.blend" -and ! -path "*CVS*" -and ! -path "*.svn*" -and ! -name "Makefile*" -and ! -name "*~" -and ! -name ".#*" -exec echo $(DESTDIR)$(pkgdatadir)/{} \; | xargs $(mkinstalldirs)
-	find $(srcdir)/$(freedroid_data) -type f ! -name "*.blend" -and ! -path "*CVS*" -and ! -path "*.svn*" -and ! -name "Makefile*" -and ! -name "*~" -and ! -name ".#*" -exec $(INSTALL_DATA) {} $(DESTDIR)$(pkgdatadir)/{} \;
+	(cd $(top_srcdir) && find $(freedroid_data) -type d ! -name "*.blend" -and ! -path "*CVS*" -and ! -path "*.svn*" -and ! -name "Makefile*" -and ! -name "*~" -and ! -name ".#*" -exec echo $(DESTDIR)$(pkgdatadir)/{} \; | xargs $(mkinstalldirs))
+	(cd $(top_srcdir) && find $(freedroid_data) -type f ! -name "*.blend" -and ! -path "*CVS*" -and ! -path "*.svn*" -and ! -name "Makefile*" -and ! -name "*~" -and ! -name ".#*" -exec $(INSTALL_DATA) {} $(DESTDIR)$(pkgdatadir)/{} \;)
 	@echo "..done."
 
 uninstall-local:
--- origsrc/freedroidrpg-0.14.1/configure.ac	2010-11-27 07:47:26.000000000 +0900
+++ src/freedroidrpg-0.14.1/configure.ac	2010-12-02 01:05:50.317615200 +0900
@@ -30,7 +30,7 @@ fi
 CFLAGS="$CFLAGS -Wall"
 
 case "$target" in
-	*-*-cygwin* | *-*-mingw32*)
+	*-*-mingw32*)
 	CFLAGS="$CFLAGS  -DFD_DATADIR='\".\"'"
 	SYS_GL_LIB=opengl32
 	win32=true
@@ -187,6 +187,10 @@ It is recommended to use openGL.
    fi
 fi
 
+PKG_CHECK_MODULES(LUA, lua)
+AC_SUBST(LUA_CFLAGS)
+AC_SUBST(LUA_LIBS)
+
 # Checks for header files.
 AC_CHECK_HEADERS([fcntl.h stdlib.h string.h sys/ioctl.h sys/time.h time.h \
 			  unistd.h dirent.h sys/soundcard.h soundcard.h execinfo.h signal.h locale.h stdint.h libintl.h libgen.h langinfo.h])
--- origsrc/freedroidrpg-0.14.1/src/Makefile.am	2010-11-27 07:41:13.000000000 +0900
+++ src/freedroidrpg-0.14.1/src/Makefile.am	2010-12-02 01:08:16.527875700 +0900
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-freedroidRPG_LDADD = ../lua/liblua.a
+freedroidRPG_LDADD = @LUA_LIBS@
 
 if WIN32
 freedroidRPG_LDADD += ../win32/freedroidRPG.coff
@@ -56,8 +56,3 @@ freedroidRPG_SOURCES = menu.c shop.c ite
 savestruct.c savestruct.h: struct.h gen_savestruct.py
 	python gen_savestruct.py struct.h savestruct
 
-clean-local:
-	cd ../lua; make clean
-
-../lua/liblua.a: 
-	cd ../lua; make CC=${CC} AR="${AR} rcu" RANLIB="${RANLIB}" liblua.a
--- origsrc/freedroidrpg-0.14.1/src/init.c	2010-11-27 07:41:13.000000000 +0900
+++ src/freedroidrpg-0.14.1/src/init.c	2010-12-02 01:05:50.333194600 +0900
@@ -37,7 +37,7 @@
 #include "global.h"
 #include "proto.h"
 
-#ifdef __OpenBSD__
+#if defined(__OpenBSD__) || defined(__CYGWIN__)
 #include "ieeefp.h"
 #else
 #include "fenv.h"
--- origsrc/freedroidrpg-0.14.1/src/main.c	2010-11-27 07:41:13.000000000 +0900
+++ src/freedroidrpg-0.14.1/src/main.c	2010-12-02 01:05:50.340984300 +0900
@@ -37,7 +37,7 @@
 #include "proto.h"
 #include "vars.h"
 
-#ifdef __OpenBSD__
+#if defined(__OpenBSD__) || defined(__CYGWIN__)
 #include "ieeefp.h"
 #else
 #include "fenv.h"
