--- origsrc/freedroidrpg-0.15/Makefile.am	2011-12-10 18:32:37.000000000 +0900
+++ src/freedroidrpg-0.15/Makefile.am	2012-01-11 00:45:50.662644500 +0900
@@ -31,8 +31,8 @@ dist-hook:
 install-data-local:
 	@echo "Installing the data-files ..."
 	$(mkinstalldirs) $(DESTDIR)$(pkgdatadir)
-	find $(srcdir)/$(freedroid_data) -type d ! -path "*.svn*" -and ! -name "Makefile*" -and ! -name "*~" -and ! -name ".#*" -exec echo $(DESTDIR)$(pkgdatadir)/{} \; | xargs $(mkinstalldirs)
-	find $(srcdir)/$(freedroid_data) -type f ! -path "*.svn*" -and ! -name "Makefile*" -and ! -name "*~" -and ! -name ".#*" -exec $(INSTALL_DATA) {} $(DESTDIR)$(pkgdatadir)/{} \;
+	(cd $(top_srcdir) && find $(freedroid_data) -type d ! -name "*.blend" -and ! -path "*CVS*" -and ! -path "*.svn*" -and ! -name "Makefile*" -and ! -name "*~" -and ! -name ".#*" -exec echo $(DESTDIR)$(pkgdatadir)/{} \; | xargs $(mkinstalldirs))
+	(cd $(top_srcdir) && find $(freedroid_data) -type f ! -name "*.blend" -and ! -path "*CVS*" -and ! -path "*.svn*" -and ! -name "Makefile*" -and ! -name "*~" -and ! -name ".#*" -exec $(INSTALL_DATA) {} $(DESTDIR)$(pkgdatadir)/{} \;)
 	@echo "..done."
 
 uninstall-local:
--- origsrc/freedroidrpg-0.15/configure.ac	2011-12-23 16:53:11.000000000 +0900
+++ src/freedroidrpg-0.15/configure.ac	2012-01-11 00:46:08.398146900 +0900
@@ -30,7 +30,7 @@ fi
 CFLAGS="$CFLAGS -Wall -Wno-format-zero-length"
 
 case "$target" in
-	*-*-cygwin* | *-*-mingw32*)
+	*-*-mingw32*)
 	CFLAGS="$CFLAGS  -DFD_DATADIR='\".\"'"
 	SYS_GL_LIB=opengl32
 	win32=true
--- origsrc/freedroidrpg-0.15/graphics/Makefile.am	2011-12-10 18:32:35.000000000 +0900
+++ src/freedroidrpg-0.15/graphics/Makefile.am	2012-01-11 01:14:05.070242900 +0900
@@ -1,11 +1,11 @@
 ## Process this file with automake to produce Makefile.in
 noinst_DATA = obstacles/atlas.txt obstacles/shadow_atlas.txt floor_tiles/atlas.txt
 
-obstacles/atlas.txt: obstacles/iso_*.png obstacles/iso_*.offset
-	cd obstacles; ./../../gluem/make_atlas obstacles_atlas 2048 2048 atlas.txt iso_*.png
+obstacles/atlas.txt: $(srcdir)/obstacles/iso_*.png $(srcdir)/obstacles/iso_*.offset
+	mkdir -p obstacles; cd obstacles; ./../../gluem/make_atlas obstacles_atlas 2048 2048 atlas.txt $(srcdir)/obstacles/iso_*.png
 
-obstacles/shadow_atlas.txt: obstacles/shadow_*.png obstacles/shadow_*.offset
-	cd obstacles; ./../../gluem/make_atlas obstacles_shadows_atlas 2048 1024 shadow_atlas.txt shadow_*.png
+obstacles/shadow_atlas.txt: $(srcdir)/obstacles/shadow_*.png $(srcdir)/obstacles/shadow_*.offset
+	mkdir -p obstacles; cd obstacles; ./../../gluem/make_atlas obstacles_shadows_atlas 2048 1024 shadow_atlas.txt $(srcdir)/obstacles/shadow_*.png
 
-floor_tiles/atlas.txt: floor_tiles/iso_*.png floor_tiles/iso_*.offset
-	cd floor_tiles; ./../../gluem/make_atlas floor_tiles_atlas 2048 1024 atlas.txt iso_*.png
+floor_tiles/atlas.txt: $(srcdir)/floor_tiles/iso_*.png $(srcdir)/floor_tiles/iso_*.offset
+	mkdir -p floor_tiles; cd floor_tiles; ./../../gluem/make_atlas floor_tiles_atlas 2048 1024 atlas.txt $(srcdir)/floor_tiles/iso_*.png
--- origsrc/freedroidrpg-0.15/src/Makefile.am	2011-12-10 18:24:37.000000000 +0900
+++ src/freedroidrpg-0.15/src/Makefile.am	2012-01-11 00:47:59.194529300 +0900
@@ -3,7 +3,7 @@
 if BUILD_LUA
 freedroidRPG_LDADD = ../lua/liblua.a
 else
-freedroidRPG_LDADD = 
+freedroidRPG_LDADD = @LUA_LIBS@
 endif
 
 if WIN32
