--- origsrc/thrax-1.3.9/Makefile.am	2024-01-26 03:19:45.000000000 +0900
+++ src/thrax-1.3.9/Makefile.am	2024-10-21 08:04:21.403724000 +0900
@@ -2,3 +2,6 @@ SUBDIRS = src
 ACLOCAL_AMFLAGS = -I m4
 
 EXTRA_DIST = AUTHORS CONTRIBUTING LICENSE NEWS README.md WORKSPACE.bazel BUILD.bazel .bazelrc
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/thrax-1.3.9/configure.ac	2024-01-26 09:33:33.000000000 +0900
+++ src/thrax-1.3.9/configure.ac	2024-10-21 08:04:21.403724000 +0900
@@ -62,4 +62,5 @@ AC_CHECK_HEADER([fst/extensions/mpdt/mpd
 AC_CHECK_LIB([dl], dlopen, [DL_LIBS=-ldl])
 AC_SUBST([DL_LIBS])
 
+AC_CONFIG_FILES([thrax.pc])
 AC_OUTPUT
--- origsrc/thrax-1.3.9/src/lib/Makefile.am	2024-01-26 07:44:54.000000000 +0900
+++ src/thrax-1.3.9/src/lib/Makefile.am	2024-10-21 07:56:36.988654900 +0900
@@ -30,4 +30,5 @@ libthrax_la_SOURCES = ast/collection-nod
                       walker/stringfile.cc walker/stringfst.cc \
                       walker/symbols.cc walker/symboltable.cc walker/union.cc \
                       walker/walker.cc
-libthrax_la_LDFLAGS = -version-info 139:0:0
+libthrax_la_LDFLAGS = -no-undefined -version-info 139:0:0
+libthrax_la_LIBADD = -lfst -lfstfar -lfstscript
--- origsrc/thrax-1.3.9/thrax.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/thrax-1.3.9/thrax.pc.in	2024-10-21 08:04:21.388103400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: thrax
+Description: OpenGrm Thrax Grammar Development Tools
+Version: @VERSION@
+Libs: -L${libdir} -lthrax
+Cflags: -I${includedir}
