--- origsrc/libesedb-20240420/Makefile.am	2024-04-20 22:57:28.000000000 +0900
+++ src/libesedb-20240420/Makefile.am	2025-05-18 11:30:32.671045700 +0900
@@ -3,28 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
-	libfguid \
-	libfvalue \
-	libfwnt \
 	libesedb \
-	libfmapi \
-	libmapidb \
 	esedbtools \
 	pyesedb \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libesedb-20240420/configure.ac	2024-04-20 22:57:28.000000000 +0900
+++ src/libesedb-20240420/configure.ac	2025-05-18 11:30:32.671045700 +0900
@@ -182,26 +182,8 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
-AC_CONFIG_FILES([libfwnt/Makefile])
 AC_CONFIG_FILES([libesedb/Makefile])
 AC_CONFIG_FILES([pyesedb/Makefile])
-AC_CONFIG_FILES([libfmapi/Makefile])
-AC_CONFIG_FILES([libmapidb/Makefile])
 AC_CONFIG_FILES([esedbtools/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
--- origsrc/libesedb-20240420/esedbtools/export_handle.c	2024-04-20 22:57:28.000000000 +0900
+++ src/libesedb-20240420/esedbtools/export_handle.c	2025-05-18 11:30:32.681570500 +0900
@@ -2673,6 +2673,7 @@ int export_handle_export_record_value(
 				{
 					if( value_string_size == 0 )
 					{
+#if defined( HAVE_DEBUG_OUTPUT )
 						if( libcnotify_verbose != 0 )
 						{
 							libcnotify_printf(
@@ -2681,6 +2682,7 @@ int export_handle_export_record_value(
 							 record_value_entry,
 							 column_identifier );
 						}
+#endif
 					}
 					else
 					{
@@ -3577,6 +3579,7 @@ int export_handle_export_basic_record_va
 			{
 				if( value_string_size == 0 )
 				{
+#if defined( HAVE_DEBUG_OUTPUT )
 					if( libcnotify_verbose != 0 )
 					{
 						libcnotify_printf(
@@ -3585,6 +3588,7 @@ int export_handle_export_basic_record_va
 						 record_value_entry,
 						 column_identifier );
 					}
+#endif
 				}
 				else
 				{
@@ -3844,6 +3848,7 @@ int export_handle_export_long_record_val
 		 "Unable to retrieve long value of record entry: %d.\n",
 		 record_value_entry );
 
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
@@ -3851,6 +3856,7 @@ int export_handle_export_long_record_val
 			 function,
 			 record_value_entry );
 		}
+#endif
 		libcerror_error_set(
 		 error,
 		 LIBCERROR_ERROR_DOMAIN_RUNTIME,
@@ -3926,6 +3932,7 @@ int export_handle_export_long_record_val
 			{
 				if( value_string_size == 0 )
 				{
+#if defined( HAVE_DEBUG_OUTPUT )
 					if( libcnotify_verbose != 0 )
 					{
 						libcnotify_printf(
@@ -3934,6 +3941,7 @@ int export_handle_export_long_record_val
 						 record_value_entry,
 						 column_identifier );
 					}
+#endif
 				}
 				else
 				{
--- origsrc/libesedb-20240420/esedbtools/windows_search.c	2024-04-20 22:57:28.000000000 +0900
+++ src/libesedb-20240420/esedbtools/windows_search.c	2025-05-18 11:30:32.681570500 +0900
@@ -244,7 +244,7 @@ int windows_search_export_compressed_str
 	}
 #endif
 */
-#if defined( HAVE_DEBUG_OUTPUT ) && defined( HAVE_EXTRA_DEBUG_OUTPUT )
+#if defined( HAVE_DEBUG_OUTPUT )
 	if( libcnotify_verbose != 0 )
 	{
 		libcnotify_printf(
@@ -338,7 +338,7 @@ int windows_search_export_compressed_str
 		}
 		else
 		{
-#if defined( HAVE_DEBUG_OUTPUT ) && defined( HAVE_EXTRA_DEBUG_OUTPUT )
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
@@ -428,7 +428,7 @@ int windows_search_export_compressed_str
 
 				return( -1 );
 			}
-#if defined( HAVE_DEBUG_OUTPUT ) && defined( HAVE_EXTRA_DEBUG_OUTPUT )
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
@@ -630,19 +630,19 @@ int windows_search_export_compressed_str
 	}
 	else
 	{
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			libcnotify_printf(
 			 "unsupported compression type: 0x%02" PRIx8 "\n",
 			 compression_type );
 
-#if defined( HAVE_DEBUG_OUTPUT )
 			libcnotify_print_data(
 			 value_data,
 			 value_data_size,
 			 0 );
-#endif
 		}
+#endif
 		memory_free(
 		 decoded_value_data );
 	}
@@ -1523,6 +1523,7 @@ int windows_search_export_record_value_c
 			 "Unable to retrieve long value of record entry: %d.\n",
 			 record_value_entry );
 
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
@@ -1530,6 +1531,7 @@ int windows_search_export_record_value_c
 				 function,
 				 record_value_entry );
 			}
+#endif
 			libcerror_error_set(
 			 error,
 			 LIBCERROR_ERROR_DOMAIN_RUNTIME,
--- origsrc/libesedb-20240420/libesedb/libesedb_page.c	2024-04-20 22:57:28.000000000 +0900
+++ src/libesedb-20240420/libesedb/libesedb_page.c	2025-05-18 11:30:32.681570500 +0900
@@ -985,7 +985,7 @@ int libesedb_page_read_file_io_handle(
 			 calculated_xor32_checksum );
 
 			goto on_error;
-#else
+#elif defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
@@ -1009,7 +1009,7 @@ int libesedb_page_read_file_io_handle(
 			 calculated_ecc32_checksum );
 
 			goto on_error;
-#else
+#elif defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
--- origsrc/libesedb-20240420/pyesedb/Makefile.am	2024-04-20 22:57:28.000000000 +0900
+++ src/libesedb-20240420/pyesedb/Makefile.am	2025-05-18 11:30:43.590328500 +0900
@@ -53,7 +53,7 @@ pyesedb_la_LIBADD = \
 	@LIBBFIO_LIBADD@
 
 pyesedb_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyesedb_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyesedb_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
--- origsrc/libesedb-20240420/tests/Makefile.am	2024-04-20 22:57:28.000000000 +0900
+++ src/libesedb-20240420/tests/Makefile.am	2025-05-18 11:42:03.813959000 +0900
@@ -3,6 +3,7 @@ AUTOMAKE_OPTIONS = subdir-objects
 AM_CPPFLAGS = \
 	-I../include -I$(top_srcdir)/include \
 	-I../common -I$(top_srcdir)/common \
+	-I$(top_srcdir)/libesedb \
 	@LIBCERROR_CPPFLAGS@ \
 	@LIBCDATA_CPPFLAGS@ \
 	@LIBCLOCALE_CPPFLAGS@ \
