--- origsrc/libesedb-20120102/esedbtools/Makefile.am	2011-09-04 17:42:03.000000000 +0900
+++ src/libesedb-20120102/esedbtools/Makefile.am	2012-11-11 00:19:23.195856700 +0900
@@ -1,9 +1,16 @@
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libbfio \
 	-I$(top_srcdir)/libcstring \
 	-I$(top_srcdir)/liberror \
+	-I$(top_srcdir)/libfdatetime \
+	-I$(top_srcdir)/libfguid \
+	-I$(top_srcdir)/libfmapi \
+	-I$(top_srcdir)/libfvalue \
+	-I$(top_srcdir)/libfwnt \
 	-I$(top_srcdir)/libsystem \
+	-I$(top_srcdir)/libuna \
 	@LIBUNA_CPPFLAGS@ \
 	@LIBBFIO_CPPFLAGS@ \
 	@LIBFDATETIME_CPPFLAGS@ \
--- origsrc/libesedb-20120102/libbfio/Makefile.am	2012-01-03 00:08:04.000000000 +0900
+++ src/libesedb-20120102/libbfio/Makefile.am	2012-11-10 23:56:31.588434900 +0900
@@ -4,6 +4,7 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/common \
 	-I$(top_srcdir)/libcstring \
 	-I$(top_srcdir)/liberror \
+	-I$(top_srcdir)/libuna \
 	@LIBUNA_CPPFLAGS@ \
 	@MEMWATCH_CPPFLAGS@
 
--- origsrc/libesedb-20120102/libesedb/Makefile.am	2011-11-28 01:45:40.000000000 +0900
+++ src/libesedb-20120102/libesedb/Makefile.am	2012-11-11 00:09:36.694880200 +0900
@@ -2,9 +2,15 @@ AM_CPPFLAGS = \
 	-DLOCALEDIR=\"$(datadir)/locale\" \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libbfio \
 	-I$(top_srcdir)/libcstring \
 	-I$(top_srcdir)/liberror \
+	-I$(top_srcdir)/libfcache \
+	-I$(top_srcdir)/libfdata \
+	-I$(top_srcdir)/libfguid \
+	-I$(top_srcdir)/libfvalue \
 	-I$(top_srcdir)/libnotify \
+	-I$(top_srcdir)/libuna \
 	@LIBUNA_CPPFLAGS@ \
 	@LIBBFIO_CPPFLAGS@ \
 	@LIBFCACHE_CPPFLAGS@ \
--- origsrc/libesedb-20120102/libfdata/Makefile.am	2011-11-28 01:42:32.000000000 +0900
+++ src/libesedb-20120102/libfdata/Makefile.am	2012-11-10 23:59:54.256403600 +0900
@@ -2,8 +2,10 @@ if HAVE_LOCAL_LIBFDATA
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
+	-I$(top_srcdir)/libbfio \
 	-I$(top_srcdir)/libcstring \
 	-I$(top_srcdir)/liberror \
+	-I$(top_srcdir)/libfcache \
 	-I$(top_srcdir)/libnotify \
 	@LIBBFIO_CPPFLAGS@ \
 	@LIBFCACHE_CPPFLAGS@ \
--- origsrc/libesedb-20120102/libfmapi/Makefile.am	2011-07-12 04:13:12.000000000 +0900
+++ src/libesedb-20120102/libfmapi/Makefile.am	2012-11-11 00:13:39.735895800 +0900
@@ -4,7 +4,11 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/common \
 	-I$(top_srcdir)/libcstring \
 	-I$(top_srcdir)/liberror \
+	-I$(top_srcdir)/libfdatetime \
+	-I$(top_srcdir)/libfguid \
+	-I$(top_srcdir)/libfwnt \
 	-I$(top_srcdir)/libnotify \
+	-I$(top_srcdir)/libuna \
 	@LIBUNA_CPPFLAGS@ \
 	@LIBFDATETIME_CPPFLAGS@ \
 	@LIBFGUID_CPPFLAGS@ \
--- origsrc/libesedb-20120102/libfvalue/Makefile.am	2011-07-12 04:15:19.000000000 +0900
+++ src/libesedb-20120102/libfvalue/Makefile.am	2012-11-11 00:03:39.865778600 +0900
@@ -4,7 +4,10 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/common \
 	-I$(top_srcdir)/libcstring \
 	-I$(top_srcdir)/liberror \
+	-I$(top_srcdir)/libfdatetime \
+	-I$(top_srcdir)/libfguid \
 	-I$(top_srcdir)/libnotify \
+	-I$(top_srcdir)/libuna \
 	@LIBUNA_CPPFLAGS@ \
 	@LIBFDATETIME_CPPFLAGS@ \
 	@LIBFGUID_CPPFLAGS@ \
--- origsrc/libesedb-20120102/libsystem/Makefile.am	2012-01-03 00:07:03.000000000 +0900
+++ src/libesedb-20120102/libsystem/Makefile.am	2012-11-11 00:14:36.178278600 +0900
@@ -4,6 +4,7 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/common \
 	-I$(top_srcdir)/libcstring \
 	-I$(top_srcdir)/liberror \
+	-I$(top_srcdir)/libuna \
 	@LIBUNA_CPPFLAGS@ \
 	@MEMWATCH_CPPFLAGS@
 
