--- origsrc/flatzebra-0.1.7/Makefile.am	2015-11-18 10:36:43.000000000 +0900
+++ src/flatzebra-0.1.7/Makefile.am	2020-01-01 21:23:34.932855500 +0900
@@ -5,14 +5,6 @@ SUBDIRS = src
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = $(PACKAGE)-$(API).pc
 
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
-doc_DATA = \
-	AUTHORS \
-	COPYING \
-	NEWS \
-	README \
-	THANKS
-
 RPMBUILD = rpmbuild
 rpm: dist
 	$(RPMBUILD) -ta $(distdir).tar.gz
