--- origsrc/drg2sbg-2.2/configure.ac	2012-09-20 12:24:34.000000000 +0900
+++ src/drg2sbg-2.2/configure.ac	2015-04-20 11:53:42.222598200 +0900
@@ -2,7 +2,7 @@
 # Process this file with autoconf to produce a configure script.
 
 AC_INIT([drg2sbg],
-        m4_esyscmd([./GIT-VERSION-GEN]),
+        [2.2],
         [manuel.arguelles@gmail.com])
 
 AC_CONFIG_MACRO_DIR([m4]) 
@@ -27,4 +27,3 @@ AC_CHECK_FUNCS([memset strerror])
 AC_CONFIG_FILES([Makefile src/Makefile doc/Makefile])
 
 AC_OUTPUT
-
