--- origsrc/cg3-1.5.1/python/setup.py.in	2025-03-22 00:56:10.000000000 +0900
+++ src/cg3-1.5.1/python/setup.py.in	2025-05-27 15:50:56.062893400 +0900
@@ -18,7 +18,7 @@ constraint_grammar_module = Extension(
     swig_opts = ['-c++', '-I/usr/include', '-I${CMAKE_SOURCE_DIR}', '-I${CMAKE_SOURCE_DIR}/src', '-I${CMAKE_SOURCE_DIR}/include', '-Wall'],
     include_dirs=['${CMAKE_SOURCE_DIR}', '${CMAKE_SOURCE_DIR}/src', '${CMAKE_SOURCE_DIR}/include', '${CMAKE_SOURCE_DIR}/include/posix'],
     library_dirs=['${CMAKE_BINARY_DIR}/src'],
-    libraries=['cg3', 'icuuc', 'icuio', 'icui18n'],
+    libraries=['cg3', 'icuuc', 'icuio', 'icui18n', 'cg3'],
     extra_compile_args=compile_args,
     extra_link_args=link_args,
 )
