--- origsrc/wrk-4.1.0/Makefile	2018-01-21 15:00:58.000000000 +0900
+++ src/wrk-4.1.0/Makefile	2020-08-08 08:58:21.804972300 +0900
@@ -24,7 +24,7 @@ VER  ?= $(shell git describe --tags --al
 
 ODIR := obj
 OBJ  := $(patsubst %.c,$(ODIR)/%.o,$(SRC)) $(ODIR)/bytecode.o $(ODIR)/version.o
-LIBS := -lluajit-5.1 $(LIBS)
+LIBS := -lluajit-5.1 -llua-5.3 $(LIBS)
 
 DEPS    :=
 CFLAGS  += -I$(ODIR)/include
--- origsrc/wrk-4.1.0/src/script.c	2018-01-21 15:00:58.000000000 +0900
+++ src/wrk-4.1.0/src/script.c	2020-08-08 08:50:08.691748200 +0900
@@ -2,6 +2,7 @@
 
 #include <stdlib.h>
 #include <string.h>
+#include <netdb.h>
 #include "script.h"
 #include "http_parser.h"
 #include "zmalloc.h"
--- origsrc/wrk-4.1.0/src/wrk.c	2018-01-21 15:00:58.000000000 +0900
+++ src/wrk-4.1.0/src/wrk.c	2020-08-08 08:49:51.261956900 +0900
@@ -1,5 +1,7 @@
 // Copyright (C) 2012 - Will Glozer.  All rights reserved.
 
+#include <signal.h>
+
 #include "wrk.h"
 #include "script.h"
 #include "main.h"
