diff --strip-trailing-cr -uprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.log -x config.status -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/wrk/src/stats.c src/wrk/src/stats.c
--- origsrc/wrk/src/stats.c	2013-03-09 23:21:53.000000000 +0900
+++ src/wrk/src/stats.c	2013-03-09 23:23:22.351454900 +0900
@@ -54,9 +54,9 @@ long double stats_stdev(stats *stats, lo
     long double sum = 0.0;
     if (stats->limit < 2) return 0.0;
     for (uint64_t i = 0; i < stats->limit; i++) {
-        sum += powl(stats->data[i] - mean, 2);
+        sum += pow(stats->data[i] - mean, 2);
     }
-    return sqrtl(sum / (stats->limit - 1));
+    return sqrt(sum / (stats->limit - 1));
 }
 
 long double stats_within_stdev(stats *stats, long double mean, long double stdev, uint64_t n) {
