--- origsrc/wrk/src/stats.c	2013-07-14 19:13:44.000000000 +0900
+++ src/wrk/src/stats.c	2013-07-14 19:15:36.129925800 +0900
@@ -60,9 +60,9 @@ long double stats_stdev(stats *stats, lo
     long double sum = 0.0;
     if (stats->limit < 2) return 0.0;
     for (uint64_t i = 0; i < stats->limit; i++) {
-        sum += powl(stats->data[i] - mean, 2);
+        sum += pow(stats->data[i] - mean, 2);
     }
-    return sqrtl(sum / (stats->limit - 1));
+    return sqrt(sum / (stats->limit - 1));
 }
 
 long double stats_within_stdev(stats *stats, long double mean, long double stdev, uint64_t n) {
--- origsrc/wrk/src/wrk.h	2013-07-14 19:13:44.000000000 +0900
+++ src/wrk/src/wrk.h	2013-07-14 19:34:10.824328200 +0900
@@ -15,7 +15,7 @@
 
 #define VERSION  "2.2.2"
 #define RECVBUF  8192
-#define SAMPLES  100000000
+#define SAMPLES  10000
 
 #define SOCKET_TIMEOUT_MS   2000
 #define CALIBRATE_DELAY_MS  500
