--- origsrc/wrk/src/stats.c	2012-12-31 18:20:48.000000000 +0900
+++ src/wrk/src/stats.c	2012-12-31 22:17:18.654226600 +0900
@@ -54,9 +54,9 @@ long double stats_stdev(stats *stats, lo
     long double sum = 0.0;
     if (stats->limit < 2) return 0.0;
     for (int i = 0; i < stats->limit; i++) {
-        sum += powl(stats->data[i] - mean, 2);
+        sum += pow(stats->data[i] - mean, 2);
     }
-    return sqrtl(sum / (stats->limit - 1));
+    return sqrt(sum / (stats->limit - 1));
 }
 
 long double stats_within_stdev(stats *stats, long double mean, long double stdev, uint64_t n) {
