--- origsrc/suil-0.10.22/src/suil_config.h	2025-01-19 22:59:34.000000000 +0900
+++ src/suil-0.10.22/src/suil_config.h	2025-04-04 18:16:54.344055000 +0900
@@ -68,6 +68,8 @@
 #ifndef SUIL_MODULE_PREFIX
 #  ifdef _WIN32
 #    define SUIL_MODULE_PREFIX ""
+#elif defined(__CYGWIN__)
+#    define SUIL_MODULE_PREFIX "cyg"
 #  else
 #    define SUIL_MODULE_PREFIX "lib"
 #  endif
@@ -77,7 +79,7 @@
 #ifndef SUIL_MODULE_EXT
 #  if defined(__APPLE__)
 #    define SUIL_MODULE_EXT ".dylib"
-#  elif defined(_WIN32)
+#  elif defined(_WIN32) || defined(__CYGWIN__)
 #    define SUIL_MODULE_EXT ".dll"
 #  else
 #    define SUIL_MODULE_EXT ".so"
