--- origsrc/armadillo-4.400.2/CMakeLists.txt	2014-09-08 14:03:22.000000000 +0900
+++ src/armadillo-4.400.2/CMakeLists.txt	2014-09-18 14:29:16.632102400 +0900
@@ -224,14 +224,14 @@ if(APPLE)
 endif()
 
 
-# find_package(HDF5)
-# message(STATUS "HDF5_FOUND     = ${HDF5_FOUND}")
-# 
-# if(HDF5_FOUND)
-#   set(ARMA_USE_HDF5 true)
-#   set(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} ${HDF5_INCLUDE_DIRS})
-#   set(ARMA_LIBS ${ARMA_LIBS} ${HDF5_LIBRARIES})
-# endif()
+find_package(HDF5)
+message(STATUS "HDF5_FOUND     = ${HDF5_FOUND}")
+
+if(HDF5_FOUND)
+  set(ARMA_USE_HDF5 true)
+  set(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} ${HDF5_INCLUDE_DIRS})
+  set(ARMA_LIBS ${ARMA_LIBS} ${HDF5_LIBRARIES})
+endif()
 
 
 include(ARMA_FindARPACK)
