--- origsrc/armadillo-4.450.4/CMakeLists.txt	2014-10-12 20:40:24.000000000 +0900
+++ src/armadillo-4.450.4/CMakeLists.txt	2014-10-24 21:30:53.925612900 +0900
@@ -228,14 +228,14 @@ endif()
 ##       need to specify the hdf5 include folder in include/armadillo_bits/config.hpp and include/armadillo_bits/config.hpp.cmake
 ##       and then include the hdf5 header file within the main armadillo header in a similar manner to including the ATLAS header
 
-# find_package(HDF5)
-# message(STATUS "HDF5_FOUND     = ${HDF5_FOUND}")
-# 
-# if(HDF5_FOUND)
-#   set(ARMA_USE_HDF5_ALT true)
-#   set(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} ${HDF5_INCLUDE_DIRS})
-#   set(ARMA_LIBS ${ARMA_LIBS} ${HDF5_LIBRARIES})
-# endif()
+find_package(HDF5)
+message(STATUS "HDF5_FOUND     = ${HDF5_FOUND}")
+
+if(HDF5_FOUND)
+  set(ARMA_USE_HDF5_ALT true)
+  set(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} ${HDF5_INCLUDE_DIRS})
+  set(ARMA_LIBS ${ARMA_LIBS} ${HDF5_LIBRARIES})
+endif()
 
 
 include(ARMA_FindARPACK)
