--- origsrc/armadillo-3.2.4/CMakeLists.txt	2012-07-11 17:08:37.000000000 +0900
+++ src/armadillo-3.2.4/CMakeLists.txt	2012-08-11 20:20:18.171198500 +0900
@@ -345,8 +345,9 @@ DESTINATION ${INSTALL_INCLUDE_DIR}/armad
 )
 
 install(TARGETS armadillo 
-  DESTINATION ${INSTALL_LIB_DIR}
-  EXPORT ArmadilloLibraryDepends)
+  EXPORT ArmadilloLibraryDepends
+  RUNTIME DESTINATION /usr/bin
+  ARCHIVE DESTINATION ${INSTALL_LIB_DIR})
 
 # Export the package for use from the build-tree
 # (this registers the build-tree with a global CMake-registry)
