--- origsrc/armadillo-8.100.1/CMakeLists.txt	2016-06-17 01:17:01.000000000 +0900
+++ src/armadillo-8.100.1/CMakeLists.txt	2017-09-28 15:23:48.306000000 +0900
@@ -464,7 +464,7 @@ configure_file(${PROJECT_SOURCE_DIR}/cma
 
 # Install the export set for use with the install-tree
 install(EXPORT ArmadilloLibraryDepends DESTINATION
-  "${CMAKE_INSTALL_DATADIR}/Armadillo/CMake"
+  "${CMAKE_INSTALL_FULL_LIBDIR}/cmake/armadillo"
   COMPONENT dev)
 
 
@@ -473,7 +473,7 @@ install(EXPORT ArmadilloLibraryDepends D
 # and install it
 set(ARMADILLO_INCLUDE_DIRS "${CMAKE_INSTALL_FULL_INCLUDEDIR}")
 set(ARMADILLO_LIB_DIR      "${CMAKE_INSTALL_FULL_LIBDIR}")
-set(ARMADILLO_CMAKE_DIR    "${CMAKE_INSTALL_FULL_DATADIR}/Armadillo/CMake")
+set(ARMADILLO_CMAKE_DIR    "${CMAKE_INSTALL_FULL_LIBDIR}/cmake/armadillo")
 
 
 message(STATUS "Generating '${PROJECT_BINARY_DIR}/InstallFiles/ArmadilloConfig.cmake'")
