--- origsrc/armadillo-8.600.1/CMakeLists.txt	2016-06-17 01:18:05.000000000 +0900
+++ src/armadillo-8.600.1/CMakeLists.txt	2018-08-12 09:48:53.159394900 +0900
@@ -364,7 +364,7 @@ message(STATUS "")
 
 
 if(DEFINED CMAKE_CXX_COMPILER_ID AND DEFINED CMAKE_CXX_COMPILER_VERSION)
-  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND NOT ${CMAKE_CXX_COMPILER_VERSION} VERSION_LESS 4.8.3)
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND NOT ${CMAKE_CXX_COMPILER_VERSION} VERSION_LESS 4.8.3 AND NOT CYGWIN)
     set(ARMA_USE_EXTERN_CXX11_RNG true)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
     message(STATUS "Detected gcc 4.8.3 or later. Added '-std=c++11' to compiler flags")
@@ -466,7 +466,7 @@ configure_file(${PROJECT_SOURCE_DIR}/cma
 
 # Install the export set for use with the install-tree
 install(EXPORT ArmadilloLibraryDepends DESTINATION
-  "${CMAKE_INSTALL_DATADIR}/Armadillo/CMake"
+  "${CMAKE_INSTALL_FULL_LIBDIR}/cmake/armadillo"
   COMPONENT dev)
 
 
@@ -475,7 +475,7 @@ install(EXPORT ArmadilloLibraryDepends D
 # and install it
 set(ARMADILLO_INCLUDE_DIRS "${CMAKE_INSTALL_FULL_INCLUDEDIR}")
 set(ARMADILLO_LIB_DIR      "${CMAKE_INSTALL_FULL_LIBDIR}")
-set(ARMADILLO_CMAKE_DIR    "${CMAKE_INSTALL_FULL_DATADIR}/Armadillo/CMake")
+set(ARMADILLO_CMAKE_DIR    "${CMAKE_INSTALL_FULL_LIBDIR}/cmake/armadillo")
 
 
 message(STATUS "Generating '${PROJECT_BINARY_DIR}/InstallFiles/ArmadilloConfig.cmake'")
@@ -503,3 +503,11 @@ message(STATUS "Generating '${PROJECT_BI
 configure_file(${PROJECT_BINARY_DIR}/tmp/misc/armadillo.pc.in "${PROJECT_BINARY_DIR}/tmp/misc/armadillo.pc" @ONLY)
 
 install(FILES "${PROJECT_BINARY_DIR}/tmp/misc/armadillo.pc" DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+
+# test
+
+enable_testing()
+file(GLOB test_sources tests/*.cpp)
+add_executable(main ${test_sources})
+target_link_libraries(main armadillo)
+add_test(NAME main COMMAND $<TARGET_FILE:main>)
