diff --strip-trailing-cr -uprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.log -x config.status -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/armadillo-3.810.2/CMakeLists.txt src/armadillo-3.810.2/CMakeLists.txt
--- origsrc/armadillo-3.810.2/CMakeLists.txt	2013-04-26 15:48:11.000000000 +0900
+++ src/armadillo-3.810.2/CMakeLists.txt	2013-04-28 09:56:25.956710900 +0900
@@ -368,8 +368,9 @@ DESTINATION ${INSTALL_INCLUDE_DIR}/armad
 )
 
 install(TARGETS armadillo 
-  DESTINATION ${INSTALL_LIB_DIR}
-  EXPORT ArmadilloLibraryDepends)
+  EXPORT ArmadilloLibraryDepends
+  RUNTIME DESTINATION /usr/bin
+  ARCHIVE DESTINATION ${INSTALL_LIB_DIR})
 
 # Export the package for use from the build-tree
 # (this registers the build-tree with a global CMake-registry)
