--- origsrc/armadillo-3.6.1/CMakeLists.txt	2012-12-14 21:38:18.000000000 +0900
+++ src/armadillo-3.6.1/CMakeLists.txt	2012-12-31 14:18:20.127695300 +0900
@@ -352,8 +352,9 @@ DESTINATION ${INSTALL_INCLUDE_DIR}/armad
 )
 
 install(TARGETS armadillo 
-  DESTINATION ${INSTALL_LIB_DIR}
-  EXPORT ArmadilloLibraryDepends)
+  EXPORT ArmadilloLibraryDepends
+  RUNTIME DESTINATION /usr/bin
+  ARCHIVE DESTINATION ${INSTALL_LIB_DIR})
 
 # Export the package for use from the build-tree
 # (this registers the build-tree with a global CMake-registry)
