--- origsrc/armadillo-3.4.3/CMakeLists.txt	2012-10-03 10:42:50.000000000 +0900
+++ src/armadillo-3.4.3/CMakeLists.txt	2012-10-21 16:43:21.660981700 +0900
@@ -344,8 +344,9 @@ DESTINATION ${INSTALL_INCLUDE_DIR}/armad
 )
 
 install(TARGETS armadillo 
-  DESTINATION ${INSTALL_LIB_DIR}
-  EXPORT ArmadilloLibraryDepends)
+  EXPORT ArmadilloLibraryDepends
+  RUNTIME DESTINATION /usr/bin
+  ARCHIVE DESTINATION ${INSTALL_LIB_DIR})
 
 # Export the package for use from the build-tree
 # (this registers the build-tree with a global CMake-registry)
