--- origsrc/armadillo-14.4.3/CMakeLists.txt	2016-06-17 02:14:43.000000000 +0900
+++ src/armadillo-14.4.3/CMakeLists.txt	2025-05-25 19:50:28.304570500 +0900
@@ -671,7 +671,7 @@ configure_file(${PROJECT_SOURCE_DIR}/cma
 
 # Install the export set for use with the install-tree
 install(EXPORT ArmadilloLibraryDepends DESTINATION
-  "${CMAKE_INSTALL_DATADIR}/Armadillo/CMake"
+  "${CMAKE_INSTALL_LIBDIR}/cmake/Armadillo"
   COMPONENT dev)
 
 
@@ -680,7 +680,7 @@ install(EXPORT ArmadilloLibraryDepends D
 # and install it
 set(ARMADILLO_INCLUDE_DIRS "${CMAKE_INSTALL_FULL_INCLUDEDIR}")
 set(ARMADILLO_LIB_DIR      "${CMAKE_INSTALL_FULL_LIBDIR}")
-set(ARMADILLO_CMAKE_DIR    "${CMAKE_INSTALL_FULL_DATADIR}/Armadillo/CMake")
+set(ARMADILLO_CMAKE_DIR    "${CMAKE_INSTALL_FULL_LIBDIR}/cmake/Armadillo")
 
 
 message(STATUS "Generating '${PROJECT_BINARY_DIR}/InstallFiles/ArmadilloConfig.cmake'")
