--- origsrc/armadillo-4.300.3/CMakeLists.txt	2014-05-15 16:51:14.000000000 +0900
+++ src/armadillo-4.300.3/CMakeLists.txt	2014-05-16 21:04:14.018238600 +0900
@@ -384,8 +384,9 @@ PATTERN "*orig" EXCLUDE
 )
 
 install(TARGETS armadillo 
-  DESTINATION ${INSTALL_LIB_DIR}
-  EXPORT ArmadilloLibraryDepends)
+  EXPORT ArmadilloLibraryDepends
+  RUNTIME DESTINATION /usr/bin
+  ARCHIVE DESTINATION ${INSTALL_LIB_DIR})
 
 # Export the package for use from the build-tree
 # (this registers the build-tree with a global CMake-registry)
