--- origsrc/bctoolbox-4.4.0/CMakeLists.txt	2020-06-16 02:24:39.000000000 +0900
+++ src/bctoolbox-4.4.0/CMakeLists.txt	2020-10-18 13:52:16.284111300 +0900
@@ -151,6 +151,7 @@ set(EXPORT_TARGETS_NAME "bctoolbox")
 
 check_library_exists("rt" "clock_gettime" "" HAVE_LIBRT)
 check_library_exists("dl" "dladdr" "" HAVE_LIBDL)
+check_library_exists("iconv" "libiconv" "" HAVE_LIBICONV)
 check_include_file("execinfo.h" HAVE_EXECINFO)
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/config.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config.h)
@@ -167,7 +168,11 @@ if(ENABLE_PACKAGE_SOURCE)
 endif()
 
 include(CMakePackageConfigHelpers)
-set(CONFIG_PACKAGE_LOCATION "${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}/cmake")
+if(CYGWIN)
+	set(CONFIG_PACKAGE_LOCATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+else()
+	set(CONFIG_PACKAGE_LOCATION "${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}/cmake")
+endif()
 
 write_basic_package_version_file(
 	"${CMAKE_CURRENT_BINARY_DIR}/${EXPORT_TARGETS_NAME}ConfigVersion.cmake"
--- origsrc/bctoolbox-4.4.0/src/CMakeLists.txt	2020-06-16 02:24:39.000000000 +0900
+++ src/bctoolbox-4.4.0/src/CMakeLists.txt	2020-10-18 13:33:35.548200300 +0900
@@ -160,6 +160,9 @@ if(ENABLE_SHARED)
 	if(HAVE_LIBDL)
 		target_link_libraries(bctoolbox PRIVATE dl)
 	endif()
+	if(HAVE_LIBICONV)
+		target_link_libraries(bctoolbox PRIVATE iconv)
+	endif()
 	if(ANDROID)
 		target_link_libraries(bctoolbox PRIVATE log)
 	endif()
