--- origsrc/libpqstego-0.0.1/configure.ac	2009-01-11 04:18:18.000000000 +0900
+++ src/libpqstego-0.0.1/configure.ac	2010-12-29 19:05:55.671327700 +0900
@@ -5,9 +5,6 @@ AC_PREREQ(2.61)
 AC_INIT([libpqstego],[0.0.1],[ckuka@madkooky.de])
 AM_INIT_AUTOMAKE([gnu 1.6])
 
-AM_MAINTAINER_MODE
-
-
 # Library versioning
 LIB_CURRENT=0
 LIB_REVISION=1
@@ -24,9 +21,7 @@ AC_CANONICAL_HOST
 # ====================================
 
 AC_PROG_CC
-AC_LIBTOOL_WIN32_DLL
-AC_LIBTOOL_DLOPEN
-AM_PROG_LIBTOOL
+LT_INIT(dlopen,win32-dll)
 
 AC_CHECK_HEADERS([dlfcn.h],
     [AC_SEARCH_LIBS([dlopen],[dl],
