--- origsrc/xraylib-xraylib-3.0.0/fortran/Makefile.am	2014-02-13 19:12:48.000000000 +0900
+++ src/xraylib-xraylib-3.0.0/fortran/Makefile.am	2014-09-13 20:09:23.977472700 +0900
@@ -18,7 +18,7 @@ lib_LTLIBRARIES=libxrlf03.la
 libxrlf03_la_SOURCES = xraylib_wrap.F90
 libxrlf03_la_LIBADD = ../src/libxrl.la
 libxrlf03_la_FCFLAGS = $(ARCHFLAGS)
-libxrlf03_la_LDFLAGS=-version-info @LIB_CURRENT@:@LIB_REVISION@:@LIB_AGE@
+libxrlf03_la_LDFLAGS=-no-undefined -version-info @LIB_CURRENT@:@LIB_REVISION@:@LIB_AGE@
 nodist_xraylibinclude_HEADERS = xraylib.mod
 
 xraylib.mod: libxrlf03.la
--- origsrc/xraylib-xraylib-3.0.0/idl/Makefile.am	2014-02-13 19:12:48.000000000 +0900
+++ src/xraylib-xraylib-3.0.0/idl/Makefile.am	2014-09-13 20:09:38.879669600 +0900
@@ -26,7 +26,7 @@ plugin_LTLIBRARIES = libxrlidl.la
 libxrlidl_la_CFLAGS = -I@RSIIDL_INCDIR@
 libxrlidl_la_LIBADD = ../src/libxrl.la @IDL_LIBADD@
 libxrlidl_la_SOURCES = xraylib_idl.c
-libxrlidl_la_LDFLAGS = -avoid-version -module -shared -export-dynamic
+libxrlidl_la_LDFLAGS = -no-undefined -avoid-version -module -shared -export-dynamic
 
 
 
--- origsrc/xraylib-xraylib-3.0.0/java/Makefile.am	2014-02-13 19:12:48.000000000 +0900
+++ src/xraylib-xraylib-3.0.0/java/Makefile.am	2014-09-13 20:10:14.050926000 +0900
@@ -23,12 +23,12 @@ endif
 libxraylib_la_CFLAGS = -I${top_srcdir}/include $(JAVACFLAGS) $(JAVAINC)
 libxraylib_la_LIBADD = ../src/libxrl.la
 nodist_libxraylib_la_SOURCES = xraylib_wrap.c
-libxraylib_la_LDFLAGS = -avoid-version -module -shared -export-dynamic -shrext $(JAVASO)
+libxraylib_la_LDFLAGS = -no-undefined -avoid-version -module -shared -export-dynamic -shrext $(JAVASO)
 
 xraylib_la_CFLAGS = -I${top_srcdir}/include $(JAVACFLAGS) $(JAVAINC)
 xraylib_la_LIBADD = ../src/libxrl.la
 nodist_xraylib_la_SOURCES = xraylib_wrap.c
-xraylib_la_LDFLAGS = -avoid-version -module -shared -export-dynamic -shrext $(JAVASO)
+xraylib_la_LDFLAGS = -no-undefined -avoid-version -module -shared -export-dynamic -shrext $(JAVASO)
 
 
 
--- origsrc/xraylib-xraylib-3.0.0/lua/Makefile.am	2014-02-13 19:12:48.000000000 +0900
+++ src/xraylib-xraylib-3.0.0/lua/Makefile.am	2014-09-13 20:24:38.322758700 +0900
@@ -13,9 +13,9 @@ if ENABLE_LUA
 
 plugindir = $(libdir)/lua/$(LUA_VERSION)
 plugin_LTLIBRARIES = xraylib.la
-xraylib_la_LIBADD = ../src/libxrl.la
+xraylib_la_LIBADD = ../src/libxrl.la $(LUA_LIB)
 nodist_xraylib_la_SOURCES = xraylib_wrap.c
-xraylib_la_LDFLAGS = -avoid-version -module -shared -export-dynamic
+xraylib_la_LDFLAGS = -no-undefined -avoid-version -module -shared -export-dynamic
 xraylib_la_CFLAGS = $(LUA_INCLUDE) -I${top_builddir} -I${top_srcdir}/include
 
 
--- origsrc/xraylib-xraylib-3.0.0/perl/Makefile.am	2014-02-13 19:12:48.000000000 +0900
+++ src/xraylib-xraylib-3.0.0/perl/Makefile.am	2014-09-13 20:23:51.093266500 +0900
@@ -14,9 +14,9 @@ if ENABLE_PERL
 
 plugindir = $(PERL_EXT_LIB)/auto/xraylib
 plugin_LTLIBRARIES = xraylib.la
-xraylib_la_LIBADD = ../src/libxrl.la
+xraylib_la_LIBADD = ../src/libxrl.la -L$(PERL_EXT_INC) -lperl
 nodist_xraylib_la_SOURCES = xraylib_wrap.c
-xraylib_la_LDFLAGS = -avoid-version -module -shared -export-dynamic -shrext $(PERL_EXT_DLEXT)
+xraylib_la_LDFLAGS = -no-undefined -avoid-version -module -shared -export-dynamic -shrext $(PERL_EXT_DLEXT)
 xraylib_la_CFLAGS = $(PERL_EXT_CPPFLAGS) -I$(PERL_EXT_INC) -I${top_builddir} -I${top_srcdir}/include
 
 perldir = $(PERL_EXT_LIB)
--- origsrc/xraylib-xraylib-3.0.0/php/Makefile.am	2014-02-13 19:12:48.000000000 +0900
+++ src/xraylib-xraylib-3.0.0/php/Makefile.am	2014-09-13 20:10:57.037254500 +0900
@@ -16,7 +16,7 @@ plugin_LTLIBRARIES = xraylib.la
 xraylib_la_CFLAGS = $(PHP_CFLAGS) -I${top_builddir} -I${top_srcdir}/include
 xraylib_la_LIBADD = ../src/libxrl.la
 nodist_xraylib_la_SOURCES = xraylib_wrap.c
-xraylib_la_LDFLAGS = -avoid-version -module -shared -export-dynamic $(PHP_LDFLAGS)
+xraylib_la_LDFLAGS = -no-undefined -avoid-version -module -shared -export-dynamic $(PHP_LDFLAGS)
 nodist_plugin_SCRIPTS = xraylib.php
 
 
--- origsrc/xraylib-xraylib-3.0.0/python/Makefile.am	2014-02-13 19:12:48.000000000 +0900
+++ src/xraylib-xraylib-3.0.0/python/Makefile.am	2014-09-13 20:18:06.309086800 +0900
@@ -19,9 +19,9 @@ if ENABLE_PYTHON
 
 pyexec_LTLIBRARIES = _xraylib.la
 _xraylib_la_CFLAGS = $(PYTHON_CFLAGS) -I$(top_srcdir)/include $(PYTHON_CPPFLAGS) 
-_xraylib_la_LIBADD = ../src/libxrl.la
+_xraylib_la_LIBADD = ../src/libxrl.la $(PYTHON_LDFLAGS)
 nodist__xraylib_la_SOURCES = xraylib_wrap.c
-_xraylib_la_LDFLAGS = -avoid-version -module -shared -export-dynamic
+_xraylib_la_LDFLAGS = -no-undefined -avoid-version -module -shared -export-dynamic
 
 dist_bin_SCRIPTS = xraylib
 nodist_python_PYTHON = xraylib.py xrayhelp.py xraymessages.py
--- origsrc/xraylib-xraylib-3.0.0/ruby/Makefile.am	2014-02-13 19:12:48.000000000 +0900
+++ src/xraylib-xraylib-3.0.0/ruby/Makefile.am	2014-09-13 20:25:48.981938400 +0900
@@ -14,9 +14,9 @@ if ENABLE_RUBY
 plugindir = $(RUBYDIR)
 plugin_LTLIBRARIES = xraylib.la
 xraylib_la_CFLAGS = $(RUBY_EXT_CPPFLAGS) -DSWIGRUBY_TRICK -I${top_builddir} -I${top_srcdir}/include $(RUBY_EXT_INC)
-xraylib_la_LIBADD = ../src/libxrl.la
+xraylib_la_LIBADD = ../src/libxrl.la $(RUBY_EXT_LDFLAGS)
 nodist_xraylib_la_SOURCES = xraylib_wrap.c
-xraylib_la_LDFLAGS = -avoid-version -module -shared -export-dynamic -shrext $(RUBY_EXT_DLEXT)
+xraylib_la_LDFLAGS = -no-undefined -avoid-version -module -shared -export-dynamic -shrext $(RUBY_EXT_DLEXT)
 
 
 xraylib_wrap.c: ${top_srcdir}/src/xraylib.i ${top_srcdir}/include/xraylib*h
--- origsrc/xraylib-xraylib-3.0.0/src/Makefile.am	2014-02-13 19:12:48.000000000 +0900
+++ src/xraylib-xraylib-3.0.0/src/Makefile.am	2014-09-13 20:11:36.867536100 +0900
@@ -75,7 +75,7 @@ libxrl_la_CFLAGS = $(ARCHFLAGS)
 
 nodist_libxrl_la_SOURCES = xrayglob_inline.c
 
-libxrl_la_LDFLAGS=-version-info @LIB_CURRENT@:@LIB_REVISION@:@LIB_AGE@ $(LDFLAGS_LIBXRL)
+libxrl_la_LDFLAGS=-no-undefined -version-info @LIB_CURRENT@:@LIB_REVISION@:@LIB_AGE@ $(LDFLAGS_LIBXRL)
 
 EXTRA_DIST = xraylib.i
 
