--- origsrc/xraylib-xraylib-3.3.0/configure.ac	2017-09-18 23:43:27.000000000 +0900
+++ src/xraylib-xraylib-3.3.0/configure.ac	2017-10-21 18:00:40.207000000 +0900
@@ -133,6 +133,11 @@ else
 			LDFLAGS_IDL="-no-undefined"
 			LDFLAGS_PYTHON="-no-undefined -Wl,-subsystem,windows"
 			;;
+		*cygwin*)
+			LDFLAGS_FORTRAN="-no-undefined"
+			LDFLAGS_IDL="-no-undefined"
+			LDFLAGS_PYTHON="-no-undefined"
+			;;
 	esac
 fi
 AM_CONDITIONAL([OS_WINDOWS_64],[test x$OS_WINDOWS_64 = x1])
--- origsrc/xraylib-xraylib-3.3.0/lua/Makefile.am	2017-09-18 23:43:27.000000000 +0900
+++ src/xraylib-xraylib-3.3.0/lua/Makefile.am	2017-10-21 18:13:10.566000000 +0900
@@ -13,9 +13,9 @@ if ENABLE_LUA
 
 plugindir = $(libdir)/lua/$(LUA_VERSION)
 plugin_LTLIBRARIES = xraylib.la
-xraylib_la_LIBADD = ../src/libxrl.la
+xraylib_la_LIBADD = ../src/libxrl.la $(LUA_LIB)
 nodist_xraylib_la_SOURCES = xraylib_wrap.c
-xraylib_la_LDFLAGS = -avoid-version -module -shared -export-dynamic
+xraylib_la_LDFLAGS = -no-undefined -avoid-version -module -shared -export-dynamic
 xraylib_la_CFLAGS = $(LUA_INCLUDE) -I${top_builddir} -I${top_srcdir}/include
 
 
--- origsrc/xraylib-xraylib-3.3.0/perl/Makefile.am	2017-09-18 23:43:27.000000000 +0900
+++ src/xraylib-xraylib-3.3.0/perl/Makefile.am	2017-10-21 18:11:48.578000000 +0900
@@ -14,9 +14,9 @@ if ENABLE_PERL
 
 plugindir = $(PERL_EXT_LIB)/auto/xraylib
 plugin_LTLIBRARIES = xraylib.la
-xraylib_la_LIBADD = ../src/libxrl.la
+xraylib_la_LIBADD = ../src/libxrl.la -L$(PERL_EXT_INC) -lperl
 nodist_xraylib_la_SOURCES = xraylib_wrap.c
-xraylib_la_LDFLAGS = -avoid-version -module -shared -export-dynamic -shrext $(PERL_EXT_DLEXT)
+xraylib_la_LDFLAGS = -no-undefined -avoid-version -module -shared -export-dynamic -shrext $(PERL_EXT_DLEXT)
 xraylib_la_CFLAGS = $(PERL_EXT_CPPFLAGS) -I$(PERL_EXT_INC) -I${top_builddir} -I${top_srcdir}/include
 
 perldir = $(PERL_EXT_LIB)
--- origsrc/xraylib-xraylib-3.3.0/python/Makefile.am	2017-09-18 23:43:27.000000000 +0900
+++ src/xraylib-xraylib-3.3.0/python/Makefile.am	2017-10-21 18:08:04.030999800 +0900
@@ -21,7 +21,7 @@ pyexec_LTLIBRARIES += _xraylib.la
 _xraylib_la_CFLAGS = $(PYTHON_CFLAGS) -I$(top_srcdir)/include $(PYTHON_CPPFLAGS) $(NUMPY_HEADERS)
 _xraylib_la_LIBADD = ../src/libxrl.la
 nodist__xraylib_la_SOURCES = xraylib_wrap.c
-_xraylib_la_LDFLAGS = $(LDFLAGS_PYTHON) -avoid-version -module -shared -export-dynamic -shrext $(PYTHON_EXT)
+_xraylib_la_LDFLAGS = $(LDFLAGS_PYTHON) -avoid-version -module -shared -export-dynamic -shrext $(PYTHON_EXT) $(PYTHON_LDFLAGS)
 if OS_WINDOWS
 _xraylib_la_LDFLAGS += $(PYTHON_LDFLAGS)
 #_xraylib_la_CPPFLAGS = -mdll
@@ -36,7 +36,7 @@ pyexec_LTLIBRARIES += xraylib_np.la
 xraylib_np_la_CFLAGS = $(OPENMP_CFLAGS) $(PYTHON_CFLAGS) -I$(top_srcdir)/include $(PYTHON_CPPFLAGS) $(NUMPY_HEADERS)
 xraylib_np_la_LIBADD = ../src/libxrl.la
 nodist_xraylib_np_la_SOURCES = xraylib_np.c
-xraylib_np_la_LDFLAGS = $(LDFLAGS_PYTHON) -avoid-version -module -shared -export-dynamic $(OPENMP_CFLAGS) -shrext $(PYTHON_EXT)
+xraylib_np_la_LDFLAGS = $(LDFLAGS_PYTHON) -avoid-version -module -shared -export-dynamic $(OPENMP_CFLAGS) -shrext $(PYTHON_EXT) $(PYTHON_LDFLAGS)
 if OS_WINDOWS
 xraylib_np_la_LDFLAGS += $(PYTHON_LDFLAGS)
 #xraylib_np_la_CPPFLAGS = -mdll
--- origsrc/xraylib-xraylib-3.3.0/ruby/Makefile.am	2017-09-18 23:43:27.000000000 +0900
+++ src/xraylib-xraylib-3.3.0/ruby/Makefile.am	2017-10-21 18:14:22.858000000 +0900
@@ -14,9 +14,9 @@ if ENABLE_RUBY
 plugindir = $(RUBYDIR)
 plugin_LTLIBRARIES = xraylib.la
 xraylib_la_CFLAGS = $(RUBY_EXT_CPPFLAGS) -DSWIGRUBY_TRICK -I${top_builddir} -I${top_srcdir}/include $(RUBY_EXT_INC)
-xraylib_la_LIBADD = ../src/libxrl.la
+xraylib_la_LIBADD = ../src/libxrl.la $(RUBY_EXT_LDFLAGS)
 nodist_xraylib_la_SOURCES = xraylib_wrap.c
-xraylib_la_LDFLAGS = -avoid-version -module -shared -export-dynamic -shrext $(RUBY_EXT_DLEXT)
+xraylib_la_LDFLAGS = -no-undefined -avoid-version -module -shared -export-dynamic -shrext $(RUBY_EXT_DLEXT)
 
 
 xraylib_wrap.c: ${top_srcdir}/src/xraylib.i ${top_srcdir}/include/xraylib*h
--- origsrc/xraylib-xraylib-3.3.0/src/Makefile.am	2017-09-18 23:43:27.000000000 +0900
+++ src/xraylib-xraylib-3.3.0/src/Makefile.am	2017-10-21 18:03:43.488991400 +0900
@@ -86,7 +86,7 @@ libxrl_la_CFLAGS = $(ARCHFLAGS)
 
 nodist_libxrl_la_SOURCES = xrayglob_inline.c
 
-libxrl_la_LDFLAGS=-version-info @LIB_CURRENT@:@LIB_REVISION@:@LIB_AGE@ $(LDFLAGS_LIBXRL)
+libxrl_la_LDFLAGS=-no-undefined -version-info @LIB_CURRENT@:@LIB_REVISION@:@LIB_AGE@ $(LDFLAGS_LIBXRL)
 
 EXTRA_DIST = xraylib.i
 
