--- origsrc/tempestremap-2.2.0/Makefile.am	2023-06-15 02:34:02.000000000 +0900
+++ src/tempestremap-2.2.0/Makefile.am	2026-01-25 20:17:53.019927500 +0900
@@ -92,6 +92,7 @@ libTempestRemap_la_SOURCES = \
 	src/node_multimap_3d.h
 
 libTempestRemap_la_LIBADD = ${NETCDF_LIBS} ${LAPACK_LIBS} ${BLAS_LIBS} ${LIBS}
+libTempestRemap_la_LDFLAGS = -no-undefined
 
 # Load system-specific defaults
 AM_CPPFLAGS = -I$(srcdir)/src -I$(builddir)/src ${NETCDF_CPPFLAGS}
@@ -188,3 +189,6 @@ EXTRA_DIST = $(doc_DATA) $(GMAKE_FILES)
 						 src/netcdf-cxx-4.2.COPYRIGHT src/netcdf-cxx-4.2.README src/netcdf-cxx-4.2.VERSION
 
 CLEANFILES = configs.sed
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = tempestremap.pc
--- origsrc/tempestremap-2.2.0/configure.ac	2023-06-15 02:34:02.000000000 +0900
+++ src/tempestremap-2.2.0/configure.ac	2026-01-25 20:17:25.647001900 +0900
@@ -3,7 +3,7 @@
 # Usage: autoreconf -fi
 #
 AC_PREREQ([2.69])
-AC_INIT([TempestRemap],[2.1.2],[paullrich@ucdavis.edu])
+AC_INIT([TempestRemap],[2.2.0],[paullrich@ucdavis.edu])
 AC_CONFIG_AUX_DIR([config])
 AC_CONFIG_MACRO_DIR([config])
 #AC_CONFIG_SRCDIR([test/optimtest.m])
@@ -59,5 +59,6 @@ AC_CHECK_FUNCS([memset pow sqrt])
 
 AC_SUBST(DISTCHECK_CONFIGURE_FLAGS)
 AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([tempestremap.pc])
 AC_OUTPUT
 
--- origsrc/tempestremap-2.2.0/tempestremap.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/tempestremap-2.2.0/tempestremap.pc.in	2026-01-25 20:16:33.870087700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: tempestremap
+Description: Remapping software for climate applications
+Version: @VERSION@
+Libs: -L${libdir} -lTempestRemap
+Cflags: -I${includedir}
