--- origsrc/davix-R_0_8_10/CMakeLists.txt	2025-02-13 18:20:50.000000000 +0900
+++ src/davix-R_0_8_10/CMakeLists.txt	2025-04-10 08:40:21.639342000 +0900
@@ -13,21 +13,37 @@ find_package(Python COMPONENTS Interpret
 # Regenerate include/davix/features.hpp and version.cmake at _build_ time
 #-------------------------------------------------------------------------------
 add_custom_target(GenerateVersionInfo ALL DEPENDS Version)
-add_custom_command(
-  OUTPUT Version
-  COMMAND ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/genversion.py --template include/davix/features.hpp.in --out include/davix/features.hpp
-  COMMAND ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/genversion.py --template version.cmake.in --out version.cmake
-  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-)
+if (DEFINED CUSTOM_VERSION)
+  add_custom_command(
+    OUTPUT Version
+    COMMAND ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/genversion.py --template include/davix/features.hpp.in --out include/davix/features.hpp --custom-version ${CUSTOM_VERSION}
+    COMMAND ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/genversion.py --template version.cmake.in --out version.cmake --custom-version ${CUSTOM_VERSION}
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  )
+else()
+  add_custom_command(
+    OUTPUT Version
+    COMMAND ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/genversion.py --template include/davix/features.hpp.in --out include/davix/features.hpp
+    COMMAND ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/genversion.py --template version.cmake.in --out version.cmake
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  )
+endif()
 
 #-------------------------------------------------------------------------------
 # Regenerate version.cmake at _configure_ time. Important since the above
 # only regenerates it at compile time.
 #-------------------------------------------------------------------------------
-execute_process(
-  COMMAND ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/genversion.py --template version.cmake.in --out version.cmake
-  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-)
+if (DEFINED CUSTOM_VERSION)
+  execute_process(
+    COMMAND ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/genversion.py --template version.cmake.in --out version.cmake --custom-version ${CUSTOM_VERSION}
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  )
+else()
+  execute_process(
+    COMMAND ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/genversion.py --template version.cmake.in --out version.cmake
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  )
+endif()
 
 #-------------------------------------------------------------------------------
 # Include version.cmake. Whenever the version changes, ie you commit or tag,
--- origsrc/davix-R_0_8_10/src/CMakeLists.txt	2025-02-13 18:20:50.000000000 +0900
+++ src/davix-R_0_8_10/src/CMakeLists.txt	2025-04-10 08:40:56.185858800 +0900
@@ -126,7 +126,7 @@ set_target_properties(libdavix PROPERTIE
                             CLEAN_DIRECT_OUTPUT 1
                             OUTPUT_NAME ${OUTPUT_NAME_DAVIX})
 
-install(TARGETS libdavix DESTINATION ${LIB_INSTALL_DIR} )
+install(TARGETS libdavix RUNTIME DESTINATION ${BIN_INSTALL_DIR} ARCHIVE DESTINATION ${LIB_INSTALL_DIR} )
 
 add_PkgConfigFile_for_Library("davix.pc" libdavix
                                HEADER_DIRS "davix"
--- origsrc/davix-R_0_8_10/test/functional/CMakeLists.txt	2025-02-13 18:20:50.000000000 +0900
+++ src/davix-R_0_8_10/test/functional/CMakeLists.txt	2025-04-10 08:41:36.918356400 +0900
@@ -78,7 +78,7 @@ add_library(test_davix_functional_lib ST
 #target_link_libraries(test_mv libdavix test_davix_functional_lib)
 
 add_executable(davix-tester ../TestcaseHandler.cpp davix-tester.cpp)
-target_link_libraries(davix-tester libdavix test_davix_functional_lib)
+target_link_libraries(davix-tester test_davix_functional_lib libdavix)
 
 install(TARGETS davix-tester
   DESTINATION ${BIN_INSTALL_DIR}/)
