--- origsrc/libs3-2.0/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libs3-2.0/Makefile.am	2012-01-28 21:16:08.200382400 +0900
@@ -0,0 +1 @@
+SUBDIRS = src inc
--- origsrc/libs3-2.0/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/libs3-2.0/configure.ac	2012-01-28 21:11:06.092299200 +0900
@@ -0,0 +1,45 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([libs3], [2.0], [http://libs3.ischo.com.s3.amazonaws.com/index.html])
+AC_CONFIG_SRCDIR([inc/error_parser.h])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+LIBS3_VER_MAJOR=2
+AC_SUBST([LIBS3_VER_MAJOR])
+LIBS3_VER_MINOR=0
+AC_SUBST([LIBS3_VER_MINOR])
+LIBS3_VER=2.0
+AC_SUBST([LIBS3_VER])
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_LN_S
+LT_INIT([disable-static])
+
+# Checks for libraries.
+PKG_CHECK_MODULES([CURL], [libcurl])
+PKG_CHECK_MODULES([XML2], [libxml-2.0])
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdint.h stdlib.h string.h strings.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_HEADER_STDBOOL
+AC_TYPE_INT64_T
+AC_TYPE_SIZE_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_MKTIME
+AC_CHECK_FUNCS([memset putenv select setenv uname])
+
+AC_CONFIG_FILES([Makefile
+                 inc/Makefile
+                 src/Makefile])
+AC_OUTPUT
--- origsrc/libs3-2.0/inc/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libs3-2.0/inc/Makefile.am	2012-01-28 20:56:54.096870300 +0900
@@ -0,0 +1 @@
+include_HEADERS = libs3.h
--- origsrc/libs3-2.0/src/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libs3-2.0/src/Makefile.am	2012-01-28 21:08:09.053983900 +0900
@@ -0,0 +1,27 @@
+
+AM_CFLAGS = -Wall -Wshadow -Wextra -I$(top_srcdir)/inc \
+          $(CURL_CFLAGS) $(XML2_CFLAGS) \
+          -DLIBS3_VER_MAJOR=\"$(LIBS3_VER_MAJOR)\" \
+          -DLIBS3_VER_MINOR=\"$(LIBS3_VER_MINOR)\" \
+          -DLIBS3_VER=\"$(LIBS3_VER)\" \
+          -D__STRICT_ANSI__ \
+          -D_ISOC99_SOURCE \
+          -D_POSIX_C_SOURCE=200112L
+
+bin_PROGRAMS = s3
+s3_SOURCES = s3.c
+s3_LDADD = libs3.la
+
+check_PROGRAMS = testsimplexml
+testsimplexml_SOURCES = testsimplexml.c
+testsimplexml_LDADD = libs3.la
+
+lib_LTLIBRARIES = libs3.la
+libs3_la_SOURCES = acl.c bucket.c error_parser.c general.c \
+                 object.c request.c request_context.c \
+                 response_headers_handler.c service_access_logging.c \
+                 service.c simplexml.c util.c
+
+libs3_la_LDFLAGS = -version-info $(LIBS3_VER_MAJOR):$(LIBS3_VER_MINOR) -no-undefined
+libs3_la_LIBADD = $(CURL_LIBS) $(XML2_LIBS)
+
