--- origsrc/vnstat-1.14/Makefile	2015-01-06 07:19:37.000000000 +0900
+++ src/vnstat-1.14/Makefile	2015-05-03 08:56:44.793972200 +0900
@@ -35,12 +35,6 @@ debug:
 install:
 	@echo "Installing vnStat..."
 
-# check that system is really Linux
-	@if [ `uname` != "Linux" ]; \
-	then echo "This isn't a Linux system. Maybe 'make bsdinstall' is what you need?"; \
-	false; \
-	fi
-
 # check that there's something to install
 	@if [ ! -f "src/vnstat" ] || [ ! -f "src/vnstatd" ]; \
 	then echo "Nothing to install, run make first."; \
