--- origsrc/libcaldav-0.6.2/configure.ac	2010-09-12 07:45:58.000000000 +0900
+++ src/libcaldav-0.6.2/configure.ac	2010-11-13 20:45:03.641975900 +0900
@@ -52,7 +52,7 @@ AC_PROG_LN_S
 AC_PROG_MAKE_SET
 AC_HEADER_STDC
 AC_PROG_CPP
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_PROG_INSTALL
 
 # Checks for libraries.
--- origsrc/libcaldav-0.6.2/src/Makefile.am	2010-09-11 09:26:33.000000000 +0900
+++ src/libcaldav-0.6.2/src/Makefile.am	2010-11-13 20:47:12.076655900 +0900
@@ -9,7 +9,7 @@ endif
 
 if DYNAMIC_LINK
 lib_LTLIBRARIES = libcaldav.la
-libcaldav_la_LDFLAGS = -version-info @LIBVERSION@
+libcaldav_la_LDFLAGS = -version-info @LIBVERSION@ -no-undefined
 endif
 
 libcaldav_la_SOURCES = \
