--- origsrc/libcaldav-0.6.2/configure.ac	2010-09-12 07:45:58.000000000 +0900
+++ src/libcaldav-0.6.2/configure.ac	2012-03-31 16:39:02.549046400 +0900
@@ -30,8 +30,8 @@ AM_MAINTAINER_MODE
 
 if test $USE_MAINTAINER_MODE = yes; then
 	CFLAGS="-g -Wall -D_REENTRANT"
-	STATIC=1
-	DYNAMIC=0
+	STATIC=0
+	DYNAMIC=1
 	BUILD_TEST=true
 else
 	#CFLAGS="-g -Wall"
--- origsrc/libcaldav-0.6.2/src/Makefile.am	2010-09-11 09:26:33.000000000 +0900
+++ src/libcaldav-0.6.2/src/Makefile.am	2012-03-31 16:29:55.995335500 +0900
@@ -9,7 +9,7 @@ endif
 
 if DYNAMIC_LINK
 lib_LTLIBRARIES = libcaldav.la
-libcaldav_la_LDFLAGS = -version-info @LIBVERSION@
+libcaldav_la_LDFLAGS = -version-info @LIBVERSION@ -no-undefined
 endif
 
 libcaldav_la_SOURCES = \
