--- origsrc/rsgain-3.5.2/CMakeLists.txt	2024-08-07 09:31:48.000000000 +0900
+++ src/rsgain-3.5.2/CMakeLists.txt	2024-09-19 09:12:11.187780200 +0900
@@ -63,7 +63,7 @@ endif ()
 
 # Embed Git information
 find_package(Git QUIET)
-if (Git_FOUND)
+if (Git_FOUND AND EXISTS "${CMAKE_SOURCE_DIR}/.git")
   execute_process(COMMAND "${GIT_EXECUTABLE}" describe --long --tags
     WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}"
     OUTPUT_VARIABLE GIT_OUTPUT
