--- origsrc/rsgain-3.7/CMakeLists.txt	2026-03-17 22:27:38.000000000 +0900
+++ src/rsgain-3.7/CMakeLists.txt	2026-05-30 14:05:52.491364200 +0900
@@ -63,7 +63,7 @@ endif ()
 
 # Embed Git information
 find_package(Git QUIET)
-if (Git_FOUND)
+if (Git_FOUND AND EXISTS "${CMAKE_SOURCE_DIR}/.git")
   execute_process(COMMAND "${GIT_EXECUTABLE}" describe --long --tags
     WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}"
     OUTPUT_VARIABLE GIT_OUTPUT
