--- origsrc/libmsv-0.1/Makefile.am	2013-02-02 12:23:28.000000000 +0900
+++ src/libmsv-0.1/Makefile.am	2015-04-11 21:12:38.749621900 +0900
@@ -18,7 +18,7 @@ endif
 
 lib_LTLIBRARIES = libmsv.la
 libmsv_la_SOURCES = msv.c msv/msv.h
-libmsv_la_LDFLAGS = -version-info 1:0:0 $(VLDFLAGS)
+libmsv_la_LDFLAGS = -no-undefined -version-info 1:0:0 $(VLDFLAGS)
 
 nobase_include_HEADERS = msv/msv.h
 
@@ -27,3 +27,6 @@ checkmsva_SOURCES = checkmsva.c msv/msv.
 checkmsva_LDADD = libmsv.la
 
 TESTS = checkmsva
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = libmsv.pc
--- origsrc/libmsv-0.1/configure.ac	2013-02-02 12:22:55.000000000 +0900
+++ src/libmsv-0.1/configure.ac	2015-04-11 21:12:12.152865000 +0900
@@ -18,5 +18,5 @@ AC_CHECK_LIB(jansson, json_loads, [], [A
 
 RRA_LD_VERSION_SCRIPT
 
-AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([Makefile libmsv.pc])
 AC_OUTPUT
--- origsrc/libmsv-0.1/libmsv.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libmsv-0.1/libmsv.pc.in	2015-04-11 21:12:04.955275200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libmsv
+Description: Monkeysphere validation library
+Version: @VERSION@
+Libs: -L${libdir} -lmsv
+Cflags: -I${includedir}
