--- origsrc/libmsv-1.1.1/Makefile.am	2014-03-15 01:44:51.000000000 +0900
+++ src/libmsv-1.1.1/Makefile.am	2025-12-08 14:27:37.200609000 +0900
@@ -18,7 +18,7 @@ endif
 
 lib_LTLIBRARIES = libmsv.la
 libmsv_la_SOURCES = msv.c msv/msv.h
-libmsv_la_LDFLAGS = -version-info 1:1:0 $(VLDFLAGS)
+libmsv_la_LDFLAGS = -no-undefined -version-info 1:1:0 $(VLDFLAGS)
 
 nobase_include_HEADERS = msv/msv.h
 
@@ -27,3 +27,6 @@ checkmsva_SOURCES = checkmsva.c msv/msv.
 checkmsva_LDADD = libmsv.la
 
 TESTS = checkmsva
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libmsv-1.1.1/configure.ac	2018-02-04 05:12:17.000000000 +0900
+++ src/libmsv-1.1.1/configure.ac	2025-12-08 14:27:37.207421000 +0900
@@ -20,5 +20,5 @@ RRA_LD_VERSION_SCRIPT
 
 AX_GCC_VAR_ATTRIBUTE(deprecated)
 
-AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([libmsv.pc Makefile])
 AC_OUTPUT
--- origsrc/libmsv-1.1.1/libmsv.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libmsv-1.1.1/libmsv.pc.in	2025-12-08 14:27:37.209447700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libmsv
+Description: Monkeysphere validation library
+Version: @VERSION@
+Libs: -L${libdir} -lmsv
+Cflags: -I${includedir}
