--- origsrc/libgtextutils-0.7/src/gtextutils/Makefile.am	2014-01-06 05:09:32.000000000 +0900
+++ src/libgtextutils-0.7/src/gtextutils/Makefile.am	2014-07-20 19:06:02.876355600 +0900
@@ -11,7 +11,7 @@
 
 lib_LTLIBRARIES = libgtextutils.la
 
-libgtextutils_la_LDFLAGS = -version-info @LIBGTU_CURRENT@:@LIBGTU_REVISION@:@LIBGTU_AGE@ -release @VERSION@
+libgtextutils_la_LDFLAGS = -no-undefined -version-info @LIBGTU_CURRENT@:@LIBGTU_REVISION@:@LIBGTU_AGE@ -release @VERSION@
 
 libgtextutils_la_SOURCES = stream_wrapper.cpp stream_wrapper.h \
 		          text_line_reader.cpp text_line_reader.h \
--- origsrc/libgtextutils-0.7/tests/test_in_out_buf.cpp	2014-01-06 05:09:32.000000000 +0900
+++ src/libgtextutils-0.7/tests/test_in_out_buf.cpp	2014-07-20 19:06:58.016980600 +0900
@@ -72,6 +72,8 @@ int fork_pipe_command ( const char* prog
 
 	//Should never get here...
 	err(1,"execlp(%s) failed",program);
+
+	return fd;
 }
 
 void test_compress_output()
