--- origsrc/libva-utils-2.5.0/test/Makefile.am	2019-07-08 18:44:36.000000000 +0900
+++ src/libva-utils-2.5.0/test/Makefile.am	2019-08-11 18:34:35.193260800 +0900
@@ -11,7 +11,7 @@ AM_CPPFLAGS =							\
 	-DGTEST_HAS_TR1_TUPLE=1					\
 	-DPTHREADS						\
 	-DVA_DRIVERS_PATH="\"$(LIBVA_DRIVERS_PATH)\""		\
-	-std=c++11						\
+	-std=gnu++11						\
 	$(NULL)
 
 AM_LDFLAGS =							\
--- origsrc/libva-utils-2.5.0/vendor/intel/sfcsample/VDecAccelVA.cpp	2019-07-08 18:44:36.000000000 +0900
+++ src/libva-utils-2.5.0/vendor/intel/sfcsample/VDecAccelVA.cpp	2019-08-11 18:34:35.287038800 +0900
@@ -32,7 +32,9 @@
 #include <fcntl.h>
 #include <sys/mman.h>
 #include <sys/ioctl.h>
+#if defined(__linux__)
 #include <linux/fb.h>
+#endif
 #include <stdio.h>
 #include <stdlib.h>
 #include <iostream>
