--- origsrc/libADLMIDI-1.4.0.1/CMakeLists.txt	2018-10-02 00:59:51.000000000 +0900
+++ src/libADLMIDI-1.4.0.1/CMakeLists.txt	2019-04-23 11:53:14.596172500 +0900
@@ -296,6 +296,9 @@ endif()
 if(libADLMIDI_SHARED)
     add_library(ADLMIDI_shared SHARED ${libADLMIDI_SOURCES})
     set_target_properties(ADLMIDI_shared PROPERTIES OUTPUT_NAME ADLMIDI)
+    if (CYGWIN)
+        set_target_properties(ADLMIDI_shared PROPERTIES SOVERSION 0)
+    endif()
     target_include_directories(ADLMIDI_shared PUBLIC ${libADLMIDI_SOURCE_DIR}/include)
     set_legacy_standard(ADLMIDI_shared)
     set_visibility_hidden(ADLMIDI_shared)
@@ -588,3 +591,10 @@ message("WITH_VLC_PLUGIN          = ${WI
 message("WITH_OLD_UTILS           = ${WITH_OLD_UTILS}")
 message("WITH_XMI2MID             = ${WITH_XMI2MID}")
 message("EXAMPLE_SDL2_AUDIO       = ${EXAMPLE_SDL2_AUDIO}")
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libADLMIDI.pc.in ${CMAKE_BINARY_DIR}/libADLMIDI.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libADLMIDI.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libADLMIDI-1.4.0.1/libADLMIDI.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libADLMIDI-1.4.0.1/libADLMIDI.pc.in	2019-04-23 11:53:14.511865100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libADLMIDI
+Description: MIDI play library with OPL3 (YMF262) emulator
+Version: @VERSION@
+Libs: -L${libdir} -lADLMIDI
+Cflags: -I${includedir}
