--- origsrc/libde265-0.1/dec265/Makefile.am	2013-08-27 01:35:37.000000000 +0900
+++ src/libde265-0.1/dec265/Makefile.am	2013-09-29 15:43:35.936865600 +0900
@@ -1,7 +1,7 @@
 
 bin_PROGRAMS = dec265
 
-INCLUDES = -I../libde265
+INCLUDES = -I$(top_srcdir)/libde265
 
 dec265_DEPENDENCIES = ../libde265/libde265.la
 dec265_CXXFLAGS = $(VIDEOGFX_CFLAGS)
--- origsrc/libde265-0.1/libde265/Makefile.am	2013-08-27 01:35:37.000000000 +0900
+++ src/libde265-0.1/libde265/Makefile.am	2013-09-29 15:42:44.903662500 +0900
@@ -4,7 +4,9 @@ lib_LTLIBRARIES = libde265.la
 libde265_ladir = \
         $(includedir)/libde265
 
-libde265_la_LDFLAGS = -version-info $(LIBDE265_CURRENT):$(LIBDE265_REVISION):$(LIBDE265_AGE)
+libde265_la_CPPFLAGS = -I$(top_srcdir)
+
+libde265_la_LDFLAGS = -no-undefined -version-info $(LIBDE265_CURRENT):$(LIBDE265_REVISION):$(LIBDE265_AGE)
 
 libde265_la_SOURCES = \
   bitstream.c \
--- origsrc/libde265-0.1/sherlock265/Makefile.am	2013-08-27 01:35:37.000000000 +0900
+++ src/libde265-0.1/sherlock265/Makefile.am	2013-09-29 15:44:45.009131300 +0900
@@ -1,7 +1,7 @@
 
 bin_PROGRAMS = sherlock265
 
-INCLUDES = -I../libde265
+INCLUDES = -I$(top_srcdir)/libde265 -I$(top_srcdir)
 
 sherlock265_DEPENDENCIES = ../libde265/libde265.la
 sherlock265_CXXFLAGS = $(QT_CFLAGS) $(VIDEOGFX_CFLAGS)
