--- origsrc/aespipe-v2.4e/Makefile.in	2011-02-23 23:14:39.000000000 +0900
+++ src/aespipe-v2.4e/Makefile.in	2019-04-02 11:27:18.914588300 +0900
@@ -121,7 +121,7 @@ test-part2:
 	cmp test-file3 test-file2
 test-part3:
 	$(MD5SUM) test-file1 >test-file2
-	echo "$(MD)  test-file1" >test-file5
+	echo "$(MD) *test-file1" >test-file5
 test-part4:
 	$(MD5) test-file1 >test-file2
 	echo "MD5 (test-file1) = $(MD)" >test-file5
