--- origsrc/aespipe-v2.4g/Makefile.in	2023-10-24 01:37:05.000000000 +0900
+++ src/aespipe-v2.4g/Makefile.in	2023-12-11 16:34:11.436031900 +0900
@@ -121,7 +121,7 @@ test-part2:
 	cmp test-file3 test-file2
 test-part3:
 	$(MD5SUM) test-file1 >test-file2
-	echo "$(MD)  test-file1" >test-file5
+	echo "$(MD) *test-file1" >test-file5
 test-part4:
 	$(MD5) test-file1 >test-file2
 	echo "MD5 (test-file1) = $(MD)" >test-file5
