--- origsrc/nkf-2.1.3/Makefile	2013-11-22 14:17:36.000000000 +0900
+++ src/nkf-2.1.3/Makefile	2015-04-20 13:54:48.192619600 +0900
@@ -1,6 +1,5 @@
 CC = cc
-CFLAGS = -g -O2 -Wall -pedantic
-# CFLAGS = -O3
+CFLAGS = -O3
 SHAR = shar 
 # SHAR = shar -T
 PERL = perl
@@ -13,7 +12,7 @@ PYTHON3 = python
 
 .PHONY: clean install test tar shar
 
-nkf : nkf.o utf8tbl.o
+nkf.exe : nkf.o utf8tbl.o
 	$(CC) $(CFLAGS) -o nkf nkf.o utf8tbl.o
 
 nkf.o : nkf.c nkf.h utf8tbl.h config.h
@@ -26,7 +25,7 @@ clean:
 	-$(RM) nkf.o nkf nkf.exe nkf.in nkf.out nkf-$(VERSION) *~ *.bad utf8tbl.o
 	cd NKF.mod; if [ -f Makefile ]; then make clean; fi
 
-test:	nkf
+test:	nkf.exe
 	$(PERL) test.pl
 
 perl:
