--- origsrc/aribb24-v1.0.4/Makefile.am	2019-09-17 00:57:19.000000000 +0900
+++ src/aribb24-v1.0.4/Makefile.am	2023-01-13 16:39:24.362189300 +0900
@@ -8,6 +8,7 @@ libaribb24_la_SOURCES = src/aribb24.c sr
 	src/decoder_macro.h
 libaribb24_la_LIBADD = $(PNG_LIBS)
 libaribb24_la_CFLAGS = -Wall -fvisibility=hidden $(PNG_CFLAGS)
+libaribb24_la_LDFLAGS = -no-undefined
 
 pkginclude_HEADERS = src/aribb24/decoder.h src/aribb24/parser.h	\
 	src/aribb24/bits.h src/aribb24/aribb24.h
