--- origsrc/aribb24-0.0.1/Makefile.am	2014-08-10 16:43:16.000000000 +0900
+++ src/aribb24-0.0.1/Makefile.am	2015-03-26 18:42:50.454712700 +0900
@@ -1,2 +1,5 @@
 SUBDIRS = src
 ACLOCAL_AMFLAGS = -I m4
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = aribb24.pc
--- origsrc/aribb24-0.0.1/aribb24.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/aribb24-0.0.1/aribb24.pc.in	2015-03-26 18:45:19.799952000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: aribb24
+Description: library for ARIB STD-B24, decoding JIS 8 bit characters and parsing MPEG-TS stream
+Version: @VERSION@
+Libs: -L${libdir} -laribb24
+Cflags: -I${includedir}
--- origsrc/aribb24-0.0.1/configure.ac	2014-08-10 16:43:16.000000000 +0900
+++ src/aribb24-0.0.1/configure.ac	2015-03-26 18:42:27.477386500 +0900
@@ -20,6 +20,7 @@ AC_PROG_MAKE_SET
 AC_PROG_LIBTOOL
 
 # Checks for libraries.
+PKG_CHECK_MODULES([LIBPNG], [libpng])
 
 # Checks for header files.
 AC_CHECK_HEADERS([inttypes.h stddef.h stdint.h stdlib.h string.h])
@@ -39,6 +40,6 @@ AC_TYPE_UINT8_T
 # Checks for library functions.
 AC_CHECK_FUNCS([memset mkdir sqrt strchr strdup strstr strtoul])
 
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([aribb24.pc Makefile
                  src/Makefile])
 AC_OUTPUT
--- origsrc/aribb24-0.0.1/src/Makefile.am	2014-08-10 16:43:16.000000000 +0900
+++ src/aribb24-0.0.1/src/Makefile.am	2015-03-26 18:44:43.591646900 +0900
@@ -1,5 +1,5 @@
 lib_LTLIBRARIES = libaribb24.la
 libaribb24_la_SOURCES = decoder.c parser.c md5.c
-libaribb24_la_LIBADD = -lpng
+libaribb24_la_LIBADD = $(LIBPNG_LIBS)
+libaribb24_la_LDFLAGS = -no-undefined
 pkginclude_HEADERS = aribb24/decoder.h aribb24/parser.h aribb24/bits.h
-
