--- origsrc/aribb24-1.0.3/Makefile.am	2014-08-19 00:29:48.000000000 +0900
+++ src/aribb24-1.0.3/Makefile.am	2017-12-27 00:26:37.185000000 +0900
@@ -8,6 +8,7 @@ libaribb24_la_SOURCES = src/aribb24.c sr
 	src/decoder_macro.h
 libaribb24_la_LIBADD = $(PNG_LIBS)
 libaribb24_la_CFLAGS = -Wall -fvisibility=hidden
+libaribb24_la_LDFLAGS = -no-undefined
 
 pkginclude_HEADERS = src/aribb24/decoder.h src/aribb24/parser.h	\
 	src/aribb24/bits.h src/aribb24/aribb24.h
