--- origsrc/libpgm-5.1.118-1~dfsg/openpgm/pgm/Makefile.am	2011-09-28 02:59:08.000000000 +0900
+++ src/libpgm-5.1.118-1~dfsg/openpgm/pgm/Makefile.am	2012-04-26 16:57:00.225342400 +0900
@@ -94,7 +94,7 @@ libpgm_noinst_la_CFLAGS = \
 lib_LTLIBRARIES   = libpgm.la
 libpgm_la_SOURCES =
 libpgm_la_CFLAGS  = $(libpgm_noinst_la_CFLAGS)
-libpgm_la_LDFLAGS = -release @RELEASE_INFO@ -version-info @VERSION_INFO@
+libpgm_la_LDFLAGS = -release @RELEASE_INFO@ -version-info @VERSION_INFO@ -no-undefined
 libpgm_la_LIBADD  = libpgm_noinst.la
 
 pkgconfigdir = $(libdir)/pkgconfig
--- origsrc/libpgm-5.1.118-1~dfsg/openpgm/pgm/getifaddrs.c	2011-09-28 02:59:08.000000000 +0900
+++ src/libpgm-5.1.118-1~dfsg/openpgm/pgm/getifaddrs.c	2012-04-26 16:57:42.319092400 +0900
@@ -33,7 +33,7 @@
 #endif
 #include <impl/i18n.h>
 #include <impl/framework.h>
-
+#include <sys/ioctl.h>
 
 //#define GETIFADDRS_DEBUG
 
--- origsrc/libpgm-5.1.118-1~dfsg/openpgm/pgm/time.c	2011-09-28 02:59:08.000000000 +0900
+++ src/libpgm-5.1.118-1~dfsg/openpgm/pgm/time.c	2012-04-26 17:04:05.230225200 +0900
@@ -31,6 +31,10 @@
 #include <impl/i18n.h>
 #include <impl/framework.h>
 
+#if defined(HAVE_GETTIMEOFDAY)
+#define CONFIG_HAVE_GETTIMEOFDAY
+#endif
+
 //#define TIME_DEBUG
 
 
