--- origsrc/ggml/CMakeLists.txt	2024-06-23 02:33:59.000000000 +0900
+++ src/ggml/CMakeLists.txt	2024-06-23 02:34:08.044573500 +0900
@@ -221,4 +221,4 @@ configure_file(${CMAKE_CURRENT_SOURCE_DI
                ${CMAKE_CURRENT_BINARY_DIR}/ggml.pc
                @ONLY)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/ggml.pc
-        DESTINATION share/pkgconfig)
+        DESTINATION lib/pkgconfig)
--- origsrc/ggml/ggml.pc.in	2024-06-23 02:34:00.000000000 +0900
+++ src/ggml/ggml.pc.in	2024-06-23 02:34:08.060202600 +0900
@@ -5,6 +5,6 @@ libdir=${prefix}/lib
 
 Name: ggml
 Description: The GGML Tensor Library for Machine Learning
-Version: 0.0.0
+Version: @PROJECT_VERSION@
 Cflags: -I${includedir}/ggml
 Libs: -L${libdir} -lggml
--- origsrc/ggml/src/CMakeLists.txt	2024-06-23 02:34:00.000000000 +0900
+++ src/ggml/src/CMakeLists.txt	2024-06-23 02:34:08.060202600 +0900
@@ -612,6 +612,10 @@ if (BUILD_SHARED_LIBS)
     if (GGML_METAL)
         set_target_properties(${TARGET} PROPERTIES RESOURCE "${CMAKE_CURRENT_SOURCE_DIR}/ggml-metal.metal")
     endif()
+
+    if (DEFINED SOVERSION)
+        set_target_properties(${TARGET} PROPERTIES SOVERSION "${SOVERSION}")
+    endif()
 endif()
 
 target_compile_definitions(${TARGET} PUBLIC
