--- origsrc/ggml-0.17.0/CMakeLists.txt	2026-07-27 04:59:07.861155600 +0900
+++ src/ggml-0.17.0/CMakeLists.txt	2026-07-27 05:03:10.513531400 +0900
@@ -11,7 +11,7 @@ set(GGML_VERSION_BASE "${GGML_VERSION_MA
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/")
 
 find_program(GIT_EXE NAMES git git.exe NO_CMAKE_FIND_ROOT_PATH)
-if(GIT_EXE)
+if(GIT_EXE AND EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.git")
     # Get current git commit hash
     execute_process(COMMAND ${GIT_EXE} rev-parse --short HEAD
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
