--- origsrc/libreplot/demo/CMakeLists.txt	2025-08-20 21:37:36.000000000 +0900
+++ src/libreplot/demo/CMakeLists.txt	2025-08-20 22:35:41.098921100 +0900
@@ -2,6 +2,8 @@ OPTION(ENABLE_QT5 "Enable Examples that
 OPTION(ENABLE_GLUT "Enable Examples that use GLUT" ON)
 OPTION(ENABLE_GLEW "Enable Examples that use GLEW" ON)
 
+find_package(pkgconfig)
+
 IF (ENABLE_GLUT AND ENABLE_GLEW)
 ADD_SUBDIRECTORY(glut-area-test)
 ADD_SUBDIRECTORY(glut-cartesian-test)
--- origsrc/libreplot/demo/glut-area-test/CMakeLists.txt	2025-08-20 21:37:36.000000000 +0900
+++ src/libreplot/demo/glut-area-test/CMakeLists.txt	2025-08-20 22:32:56.464730300 +0900
@@ -11,7 +11,12 @@ FIND_PACKAGE(GLUT REQUIRED)
 INCLUDE_DIRECTORIES(${GLUT_INCLUDE_DIR})
 
 #GLEW
-FIND_PACKAGE(GLEW REQUIRED)
+if (CYGWIN)
+    find_package(pkgconfig)
+    pkg_check_modules(GLEW REQUIRED glew)
+else()
+    FIND_PACKAGE(GLEW REQUIRED)
+endif()
 INCLUDE_DIRECTORIES(${GLEW_INCLUDE_DIRS})
 
 #OpenGL
--- origsrc/libreplot/demo/glut-cartesian-hover-line-test/CMakeLists.txt	2025-08-20 21:37:36.000000000 +0900
+++ src/libreplot/demo/glut-cartesian-hover-line-test/CMakeLists.txt	2025-08-20 22:36:48.977547500 +0900
@@ -11,7 +11,12 @@ FIND_PACKAGE(GLUT REQUIRED)
 INCLUDE_DIRECTORIES(${GLUT_INCLUDE_DIR})
 
 #GLEW
-FIND_PACKAGE(GLEW REQUIRED)
+if (CYGWIN)
+    find_package(pkgconfig)
+    pkg_check_modules(GLEW REQUIRED glew)
+else()
+    FIND_PACKAGE(GLEW REQUIRED)
+endif()
 INCLUDE_DIRECTORIES(${GLEW_INCLUDE_DIRS})
 
 #OpenGL
--- origsrc/libreplot/demo/glut-cartesian-test/CMakeLists.txt	2025-08-20 21:37:36.000000000 +0900
+++ src/libreplot/demo/glut-cartesian-test/CMakeLists.txt	2025-08-20 22:36:37.406366800 +0900
@@ -11,7 +11,12 @@ FIND_PACKAGE(GLUT REQUIRED)
 INCLUDE_DIRECTORIES(${GLUT_INCLUDE_DIR})
 
 #GLEW
-FIND_PACKAGE(GLEW REQUIRED)
+if (CYGWIN)
+    find_package(pkgconfig)
+    pkg_check_modules(GLEW REQUIRED glew)
+else()
+    FIND_PACKAGE(GLEW REQUIRED)
+endif()
 INCLUDE_DIRECTORIES(${GLEW_INCLUDE_DIRS})
 
 #OpenGL
--- origsrc/libreplot/libreplot/libreplot.pc.in	2025-08-20 21:37:36.000000000 +0900
+++ src/libreplot/libreplot/libreplot.pc.in	2025-08-20 23:18:13.955087000 +0900
@@ -7,7 +7,7 @@ Name: ${PROJECT_NAME}
 Description: ${PROJECT_DESC}
 URL: ${PROJECT_URL}
 Requires: ${deps}
-Requires.private: ${libs}
+Requires.private:
 Version: ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}
 Libs: -L${libdir} -lreplot
 Libs.private: ${libs}
