--- origsrc/libsocketcpp-1.0.7/Makefile.am	2001-02-16 16:06:50.000000000 +0900
+++ src/libsocketcpp-1.0.7/Makefile.am	2013-02-10 13:44:32.767324400 +0900
@@ -9,7 +9,7 @@ MAINTAINERCLEANFILES = \
 	aux/config.guess \
 	aux/config.sub \
 	aux/ltconfig \
-	aux/ltmain.sh \ 
+	aux/ltmain.sh \
 	aux/install-sh \
 	aux/mkinstalldirs \
 	aux/missing 
--- origsrc/libsocketcpp-1.0.7/src/Makefile.am	2001-02-16 15:20:39.000000000 +0900
+++ src/libsocketcpp-1.0.7/src/Makefile.am	2013-02-10 13:46:11.471426000 +0900
@@ -13,8 +13,12 @@ tcpservertest_SOURCES = \
 libsocketcpp_la_SOURCES = \
         tcpsocket.cpp
 
+libsocketcpp_la_LDFLAGS = -no-undefined
+
 pkincludedir = ${includedir}/libsocketcpp
 pkginclude_HEADERS = \
         tcpsocket.h
 
+MAINTAINERCLEANFILES =
+
 include $(top_srcdir)/Makefile.extra
--- origsrc/libsocketcpp-1.0.7/src/tcpservertest.cpp	2002-02-18 11:19:47.000000000 +0900
+++ src/libsocketcpp-1.0.7/src/tcpservertest.cpp	2013-02-10 13:50:32.596426000 +0900
@@ -22,7 +22,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-#include <iostream.h>
+#include <iostream>
 #include <unistd.h>
 #include <sys/wait.h>
 #include <sys/types.h>
--- origsrc/libsocketcpp-1.0.7/src/tcpsocket.cpp	2002-02-18 11:19:47.000000000 +0900
+++ src/libsocketcpp-1.0.7/src/tcpsocket.cpp	2013-02-10 13:50:17.533926000 +0900
@@ -212,7 +212,7 @@ bool TCPSocket::Listen() {
 }
 
 bool TCPSocket::Accept() {
-   size_t sin_size = sizeof(struct sockaddr_in);
+   socklen_t sin_size = sizeof(struct sockaddr_in);
    //sockaddr_in RemoteAddress;
    if((newfd = accept(sockfd, (struct sockaddr *)&RemoteAddress, &sin_size)) == -1) {
       perror("accept");
