--- origsrc/lib3ds-1.3.0/Makefile.am	2007-06-20 18:05:29.000000000 +0900
+++ src/lib3ds-1.3.0/Makefile.am	2010-07-10 10:40:09.093643100 +0900
@@ -34,6 +34,7 @@ man_MANS = \
 
 EXTRA_DIST = \
   lib3ds.m4 \
+  lib3ds.pc \
   lib3ds-config.1 \
   AUTHORS \
   COPYING \
@@ -51,6 +52,9 @@ EXTRA_DIST = \
 m4datadir = $(datadir)/aclocal
 m4data_DATA = lib3ds.m4
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = lib3ds.pc
+
 .PHONY: release snapshot
 
 release:
--- origsrc/lib3ds-1.3.0/configure.in	2007-06-21 02:04:07.000000000 +0900
+++ src/lib3ds-1.3.0/configure.in	2010-07-10 10:32:04.749893100 +0900
@@ -14,6 +14,7 @@ AM_CONFIG_HEADER(config.h)
 AM_INIT_AUTOMAKE(lib3ds, $LIB3DS_VERSION)
 
 AC_PROG_CC
+AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
 
 AC_MSG_CHECKING([for debugging mode])
@@ -39,4 +40,5 @@ AC_OUTPUT([ \
   tools/3dsdump.1 \
   lib3ds.spec \
   3ds-utils.spec \
+  lib3ds.pc \
 ],[chmod a+x lib3ds-config])
--- origsrc/lib3ds-1.3.0/lib3ds/Makefile.am	2007-06-15 18:53:20.000000000 +0900
+++ src/lib3ds-1.3.0/lib3ds/Makefile.am	2010-07-10 10:32:42.546768100 +0900
@@ -28,7 +28,7 @@ lib_LTLIBRARIES = lib3ds.la
 
 lib3ds_la_LDFLAGS = \
   -version-info $(LIB3DS_MINOR_VERSION):$(LIB3DS_MICRO_VERSION):0 \
-  -release $(LIB3DS_MAJOR_VERSION)
+  -release $(LIB3DS_MAJOR_VERSION) -no-undefined
 
 lib3ds_la_LIBADD = -lm
 
