--- origsrc/pcmsolver-1.2.1/cmake/custom/pcmsolver.cmake	2018-05-03 00:03:08.000000000 +0900
+++ src/pcmsolver-1.2.1/cmake/custom/pcmsolver.cmake	2019-01-09 07:02:47.801727500 +0900
@@ -20,7 +20,7 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PRO
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_BINDIR})
 # Hardcode to share, rather than use CMAKE_INSTALL_DATAROOTDIR as the latter
 # might resolve to a place not recognized by CMake
-set(CMAKECONFIG_INSTALL_DIR "share/cmake/${PROJECT_NAME}")
+set(CMAKECONFIG_INSTALL_DIR "lib/cmake/${PROJECT_NAME}")
 
 if(NOT DEFINED PYMOD_INSTALL_LIBDIR)
   message(STATUS "Setting (unspecified) option PYMOD_INSTALL_LIBDIR: python")
--- origsrc/pcmsolver-1.2.1/src/CMakeLists.txt	2018-05-03 00:03:08.000000000 +0900
+++ src/pcmsolver-1.2.1/src/CMakeLists.txt	2019-01-09 06:54:41.979056300 +0900
@@ -75,8 +75,8 @@ if(NOT STATIC_LIBRARY_ONLY)
     EXPORT
       "PCMSolverTargets-shared"
     RUNTIME DESTINATION
-      ${CMAKE_INSTALL_LIBDIR}
-    LIBRARY DESTINATION
+      ${CMAKE_INSTALL_BINDIR}
+    ARCHIVE DESTINATION
       ${CMAKE_INSTALL_LIBDIR}
     )
   add_dependencies(pcm-shared generate-config-hpp)
