--- origsrc/compact-language-detector-0.1/Makefile.am	2012-05-23 19:33:00.000000000 +0900
+++ src/compact-language-detector-0.1/Makefile.am	2013-02-10 12:04:03.966007800 +0900
@@ -12,8 +12,8 @@ pkgconfig_DATA = cld.pc
 check_PROGRAMS = basic_test
 
 basic_test_SOURCES = tests/basic_test.cc
-basic_test_CXXFLAGS = -Wall -fPIC -Isrc/ -I../src -O2 -DCLD_WINDOWS
-basic_test_LDFLAGS = -lcld -Lsrc/ -lstdc++
+basic_test_CXXFLAGS = -Wall -fPIC -Isrc/ -I$(top_srcdir)/src -O2 -DCLD_WINDOWS
+basic_test_LDFLAGS = $(top_builddir)/src/libcld.la -lstdc++
 
 TESTS = basic_test
 
--- origsrc/compact-language-detector-0.1/src/Makefile.am	2012-05-21 21:13:38.000000000 +0900
+++ src/compact-language-detector-0.1/src/Makefile.am	2013-02-10 11:58:28.137882800 +0900
@@ -25,7 +25,7 @@ libcld_la_SOURCES = encodings/compact_la
         languages/internal/languages.cc
 
 libcld_la_CXXFLAGS = -Wall -fPIC -I. -O2 -DCLD_WINDOWS
-libcld_la_LDFLAGS = -version-info $(CLD_LT_VERSION)
+libcld_la_LDFLAGS = -no-undefined -version-info $(CLD_LT_VERSION)
 # libcld_la_LDADD =
 #libcld_includedir = $(includedir)/cld-$(CLD_API_VERSION)/
 
--- origsrc/compact-language-detector-0.1/src/base/build_config.h	2012-05-21 21:13:38.000000000 +0900
+++ src/compact-language-detector-0.1/src/base/build_config.h	2013-02-10 12:02:06.072453100 +0900
@@ -41,6 +41,9 @@
 #elif defined(__sun)
 #define OS_SOLARIS 1
 #define TOOLKIT_GTK
+#elif defined(__CYGWIN__)
+#define OS_LINUX 1
+#define TOOLKIT_GTK
 #else
 #error Please add support for your platform in build/build_config.h
 #endif
