diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x _generated_vm_opcodes.c -x compiler -x '*.inc' -x '*.h' origsrc/saffire/src/components/modules/CMakeLists.txt src/saffire/src/components/modules/CMakeLists.txt
--- origsrc/saffire/src/components/modules/CMakeLists.txt	2018-04-18 00:41:29.000000000 +0900
+++ src/saffire/src/components/modules/CMakeLists.txt	2018-04-18 10:05:45.624000000 +0900
@@ -23,7 +23,11 @@ add_library(modules STATIC ${sources})
 
 add_dependencies(modules generate_interfaces generate_exceptions)
 
-include(${CMAKE_SOURCE_DIR}/lib/CPUID.cmake)
+include(FindPkgConfig)
+pkg_check_modules(LIBCPUID libcpuid)
+if (NOT LIBCPUID_FOUND)
+    include(${CMAKE_SOURCE_DIR}/lib/CPUID.cmake)
+endif()
 
 target_link_libraries(
     modules
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x _generated_vm_opcodes.c -x compiler -x '*.inc' -x '*.h' origsrc/saffire/src/components/objects/numerical.c src/saffire/src/components/objects/numerical.c
--- origsrc/saffire/src/components/objects/numerical.c	2018-04-18 00:41:29.000000000 +0900
+++ src/saffire/src/components/objects/numerical.c	2018-04-18 10:33:52.627000000 +0900
@@ -25,9 +25,8 @@
  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 #include <stdio.h>
-#include <string.h>
-#include <stdlib.h>
 #include <stdlib.h>
+#include <string.h>
 #include <saffire/debug.h>
 #include <saffire/objects/object.h>
 #include <saffire/objects/objects.h>
@@ -45,8 +44,7 @@ t_numerical_object **numerical_cache;
 
 
 
-
-static char *itoa(long value, char *result, int radix) {
+static char *itoa_(long value, char *result, int radix) {
     if (radix < 2 || radix > 36) {
         *result = '\0';
         return result;
@@ -75,7 +73,6 @@ static char *itoa(long value, char *resu
 
 
 
-
 /* ======================================================================
  *   Object methods
  * ======================================================================
@@ -133,7 +130,7 @@ SAFFIRE_METHOD(numerical, bin) {
     // Make sure we have room for all binary digits
     char tmp[8 * sizeof(self->data.value) + 1];
 
-    itoa(self->data.value, tmp, 2);
+    itoa_(self->data.value, tmp, 2);
     RETURN_STRING_FROM_CHAR(tmp);
 }
 
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x _generated_vm_opcodes.c -x compiler -x '*.inc' -x '*.h' origsrc/saffire/src/components/vm/CMakeLists.txt src/saffire/src/components/vm/CMakeLists.txt
--- origsrc/saffire/src/components/vm/CMakeLists.txt	2018-04-18 00:41:29.000000000 +0900
+++ src/saffire/src/components/vm/CMakeLists.txt	2018-04-18 10:26:39.094000200 +0900
@@ -8,10 +8,12 @@ add_custom_target(
         ${CMAKE_SOURCE_DIR}/include/saffire/vm/_generated_vm_opcodes.h
     COMMENT "Generating vm opcodes source files")
 
-add_custom_command(
+if (NOT CYGWIN)
+    add_custom_command(
     OUTPUT
         ${CMAKE_CURRENT_SOURCE_DIR}/_generated_vm_opcodes.c
     COMMAND generate_vm_opcodes)
+endif()
 
 set_source_files_properties(_generated_vm_opcodes.c PROPERTIES GENERATED TRUE)
 
@@ -23,7 +25,7 @@ set(sources
     context.c
     thread.c
     import.c
-    _generated_vm_opcodes.c)
+    ${CMAKE_CURRENT_SOURCE_DIR}/_generated_vm_opcodes.c)
 
 add_library(vm STATIC ${sources})
 
