--- origsrc/babeltrace-1.2.4/bindings/python/Makefile.am	2014-11-05 04:16:00.000000000 +0900
+++ src/babeltrace-1.2.4/bindings/python/Makefile.am	2015-03-07 21:10:28.966683500 +0900
@@ -13,7 +13,7 @@ MAINTAINERCLEANFILES = babeltrace_wrap.c
 
 nodist__babeltrace_la_SOURCES = babeltrace_wrap.c
 _babeltrace_la_SOURCES = python-complements.h python-complements.c
-_babeltrace_la_LDFLAGS = -module
+_babeltrace_la_LDFLAGS = -avoid-version -module -no-undefined
 
 _babeltrace_la_CFLAGS = $(GLIB_CFLAGS) $(AM_CFLAGS)
 
--- origsrc/babeltrace-1.2.4/formats/bt-dummy/Makefile.am	2014-09-24 01:10:01.000000000 +0900
+++ src/babeltrace-1.2.4/formats/bt-dummy/Makefile.am	2015-03-07 21:10:49.552772600 +0900
@@ -6,7 +6,7 @@ libbabeltrace_dummy_la_SOURCES = \
 	bt-dummy.c
 
 # Request that the linker keeps all static libraries objects.
-libbabeltrace_dummy_la_LDFLAGS = \
+libbabeltrace_dummy_la_LDFLAGS = -no-undefined \
 	-Wl,--no-as-needed -version-info $(BABELTRACE_LIBRARY_VERSION)
 
 libbabeltrace_dummy_la_LIBADD = \
--- origsrc/babeltrace-1.2.4/formats/ctf/Makefile.am	2014-11-05 04:16:00.000000000 +0900
+++ src/babeltrace-1.2.4/formats/ctf/Makefile.am	2015-03-07 21:11:05.835897400 +0900
@@ -12,7 +12,7 @@ libbabeltrace_ctf_la_SOURCES = \
 	events-private.h
 
 # Request that the linker keeps all static libraries objects.
-libbabeltrace_ctf_la_LDFLAGS = \
+libbabeltrace_ctf_la_LDFLAGS = -no-undefined \
 	-Wl,--no-as-needed -version-info $(BABELTRACE_LIBRARY_VERSION)
 
 libbabeltrace_ctf_la_LIBADD = \
--- origsrc/babeltrace-1.2.4/formats/ctf-metadata/Makefile.am	2014-09-24 01:10:01.000000000 +0900
+++ src/babeltrace-1.2.4/formats/ctf-metadata/Makefile.am	2015-03-07 21:12:01.651873100 +0900
@@ -6,7 +6,7 @@ libbabeltrace_ctf_metadata_la_SOURCES =
 	ctf-metadata.c
 
 # Request that the linker keeps all static libraries objects.
-libbabeltrace_ctf_metadata_la_LDFLAGS = \
+libbabeltrace_ctf_metadata_la_LDFLAGS = -no-undefined \
 	-Wl,--no-as-needed -version-info $(BABELTRACE_LIBRARY_VERSION)
 
 libbabeltrace_ctf_metadata_la_LIBADD = \
--- origsrc/babeltrace-1.2.4/formats/ctf-text/Makefile.am	2014-09-24 01:10:01.000000000 +0900
+++ src/babeltrace-1.2.4/formats/ctf-text/Makefile.am	2015-03-07 21:12:13.737091600 +0900
@@ -7,7 +7,7 @@ lib_LTLIBRARIES = libbabeltrace-ctf-text
 libbabeltrace_ctf_text_la_SOURCES = \
 	ctf-text.c
 
-libbabeltrace_ctf_text_la_LDFLAGS = \
+libbabeltrace_ctf_text_la_LDFLAGS = -no-undefined \
 	-Wl,--no-as-needed -version-info $(BABELTRACE_LIBRARY_VERSION) \
 	types/libctf-text-types.la
 
--- origsrc/babeltrace-1.2.4/formats/lttng-live/Makefile.am	2014-09-24 01:10:01.000000000 +0900
+++ src/babeltrace-1.2.4/formats/lttng-live/Makefile.am	2015-03-07 21:18:38.878415900 +0900
@@ -10,8 +10,9 @@ libbabeltrace_lttng_live_la_SOURCES = \
 	lttng-live-plugin.c lttng-live-comm.c
 
 # Request that the linker keeps all static libraries objects.
-libbabeltrace_lttng_live_la_LDFLAGS = \
+libbabeltrace_lttng_live_la_LDFLAGS = -no-undefined \
 	-Wl,--no-as-needed -version-info $(BABELTRACE_LIBRARY_VERSION)
 
 libbabeltrace_lttng_live_la_LIBADD = \
-	$(top_builddir)/lib/libbabeltrace.la
+	$(top_builddir)/lib/libbabeltrace.la \
+	$(top_builddir)/formats/ctf/libbabeltrace-ctf.la
--- origsrc/babeltrace-1.2.4/lib/Makefile.am	2014-09-24 01:10:01.000000000 +0900
+++ src/babeltrace-1.2.4/lib/Makefile.am	2015-03-07 21:12:35.593937300 +0900
@@ -11,7 +11,7 @@ libbabeltrace_la_SOURCES = babeltrace.c
 			   trace-collection.c \
 			   registry.c
 
-libbabeltrace_la_LDFLAGS = -version-info $(BABELTRACE_LIBRARY_VERSION)
+libbabeltrace_la_LDFLAGS = -no-undefined -version-info $(BABELTRACE_LIBRARY_VERSION)
 
 libbabeltrace_la_LIBADD = \
 	prio_heap/libprio_heap.la \
--- origsrc/babeltrace-1.2.4/tests/Makefile.am	2014-09-24 01:10:01.000000000 +0900
+++ src/babeltrace-1.2.4/tests/Makefile.am	2015-03-07 21:19:53.904783100 +0900
@@ -21,4 +21,4 @@ clean-local:
 	fi
 
 check-am:
-	./run.sh tests
+	./run.sh $(srcdir)/tests
