--- origsrc/babeltrace-1.5.11/bindings/python/setup.py.in	2022-11-03 00:09:19.000000000 +0900
+++ src/babeltrace-1.5.11/bindings/python/setup.py.in	2026-03-26 19:46:53.093449300 +0900
@@ -40,7 +40,7 @@ following command to your .bashrc/.zshrc
 def main():
     babeltrace_ext = Extension('babeltrace/_babeltrace',
                         sources=['babeltrace/babeltrace.i','babeltrace/python-complements.c'],
-                        libraries=['babeltrace', 'babeltrace-ctf'],)
+                        libraries=['babeltrace', 'babeltrace-ctf', 'glib-2.0'],)
 
     dist = setup(name='babeltrace',
             version='@PACKAGE_VERSION@',
