--- origsrc/babeltrace-1.5.7/bindings/python/setup.py.in	2019-06-26 02:53:23.000000000 +0900
+++ src/babeltrace-1.5.7/bindings/python/setup.py.in	2019-07-17 14:54:21.266174600 +0900
@@ -37,7 +37,7 @@ following command to your .bashrc/.zshrc
 def main():
     babeltrace_ext = Extension('babeltrace/_babeltrace',
                         sources=['babeltrace/babeltrace.i','babeltrace/python-complements.c'],
-                        libraries=['babeltrace', 'babeltrace-ctf'],)
+                        libraries=['babeltrace', 'babeltrace-ctf', 'glib-2.0'],)
 
     dist = setup(name='babeltrace',
             version='@PACKAGE_VERSION@',
