--- origsrc/ADOL-C-2.5.2/ADOL-C/include/adolc/adolc_settings.h	2013-12-02 22:30:22.000000000 +0900
+++ src/ADOL-C-2.5.2/ADOL-C/include/adolc/adolc_settings.h	2015-03-22 13:25:55.230968700 +0900
@@ -1,7 +1,7 @@
 /*----------------------------------------------------------------------------
  ADOL-C -- Automatic Differentiation by Overloading in C++
  File:     adolc_settings.h
- Revision: $Id: adolc_settings.h 439 2013-12-02 13:30:22Z kulshres $
+ Revision: $Id: adolc_settings.h.in 439 2013-12-02 13:30:22Z kulshres $
  Contents: ADOL-C Settings:
            These parameters might affect the performance of the ADOL-C
            system; and are set by the configure script during the build.
--- origsrc/ADOL-C-2.5.2/configure.ac	2015-03-22 12:10:09.616710900 +0900
+++ src/ADOL-C-2.5.2/configure.ac	2015-03-22 13:23:41.100109300 +0900
@@ -76,11 +76,7 @@ AC_PROG_LIBTOOL
 MPI_CONF_PRE
 
 AC_CHECK_SIZEOF([void *])
-if test "x$ac_cv_sizeof_void_p" = "x8" ; then
-   _lib=lib64
-else
-   _lib=lib
-fi
+_lib=lib
 if test "x$libdir_set_on_cmd" = "xno" ; then
    libdir='${exec_prefix}'/"${_lib}"
    AC_SUBST(_lib)
