--- origsrc/glfw-3.1.1/src/CMakeLists.txt	2015-03-20 00:21:52.000000000 +0900
+++ src/glfw-3.1.1/src/CMakeLists.txt	2015-04-04 21:42:48.285884700 +0900
@@ -91,6 +91,6 @@ if (BUILD_SHARED_LIBS)
 endif()
 
 if (GLFW_INSTALL)
-    install(TARGETS glfw EXPORT glfwTargets DESTINATION lib${LIB_SUFFIX})
+    install(TARGETS glfw EXPORT glfwTargets RUNTIME DESTINATION bin ARCHIVE DESTINATION lib${LIB_SUFFIX})
 endif()
 
