--- origsrc/jcal/sources/libjalali/Makefile.am	2012-02-06 00:09:11.000000000 +0900
+++ src/jcal/sources/libjalali/Makefile.am	2012-02-06 00:12:28.300298600 +0900
@@ -14,7 +14,7 @@ libjalali_la_SOURCES = jalali.c jtime.c
 #      compatible (have not removed a function from the API, for
 #      example...)
 
-libjalali_la_LDFLAGS = -version-info 0:2:0
+libjalali_la_LDFLAGS = -version-info 0:2:0 -no-undefined
 includedir= $(prefix)/include/jalali
 include_HEADERS = jalali.h jtime.h jconfig.h
 INCLUDES        =  -I. -I@includedir@
