--- origsrc/libappindicator-12.10.0/Makefile.am	2012-07-12 02:28:34.415113800 +0900
+++ src/libappindicator-12.10.0/Makefile.am	2015-01-05 01:13:59.654941200 +0900
@@ -3,8 +3,11 @@ ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 SUBDIRS = \
       src		\
 	  bindings	\
-	  example	\
-	  docs
+	  example
+
+if ENABLE_GTK_DOC
+SUBDIRS += docs
+endif
 
 if WANT_TESTS
 SUBDIRS += tests
--- origsrc/libappindicator-12.10.0/bindings/python/Makefile.am	2012-07-12 02:28:34.415113800 +0900
+++ src/libappindicator-12.10.0/bindings/python/Makefile.am	2015-01-05 00:29:39.795447400 +0900
@@ -19,7 +19,7 @@ pkgappindicator_PYTHON = __init__.py
 appindicatordir = $(pkgpyexecdir)/appindicator
 appindicator_LTLIBRARIES = _appindicator.la
 
-_appindicator_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_appindicator
+_appindicator_la_LDFLAGS = -no-undefined -module -avoid-version -export-symbols-regex init_appindicator
 _appindicator_la_LIBADD = $(APPINDICATOR_PYTHON_LIBS) -L$(top_builddir)/src/.libs -lappindicator
 _appindicator_la_SOURCES = appindicatormodule.c
 nodist__appindicator_la_SOURCES = appindicator.c
--- origsrc/libappindicator-12.10.0/tests/Makefile.am	2012-07-12 02:28:34.415113800 +0900
+++ src/libappindicator-12.10.0/tests/Makefile.am	2015-01-05 00:58:49.425330200 +0900
@@ -32,7 +32,7 @@ test_libappindicator_SOURCES = \
 test_libappindicator_CFLAGS = \
 	$(TESTDEPS_CFLAGS) $(LIBRARY_CFLAGS) \
 	-DSRCDIR="\"$(srcdir)\"" \
-	-Wall -Werror \
+	-Wall \
 	-I$(top_srcdir)/src
 
 test_libappindicator_LDADD = \
@@ -49,7 +49,7 @@ test_libappindicator_dbus_client_SOURCES
 
 test_libappindicator_dbus_client_CFLAGS = \
 	$(TESTDEPS_CFLAGS) $(LIBRARY_CFLAGS) \
-	-Wall -Werror \
+	-Wall \
 	-I$(top_srcdir)/src
 
 test_libappindicator_dbus_client_LDADD = \
@@ -66,7 +66,7 @@ test_libappindicator_dbus_server_SOURCES
 
 test_libappindicator_dbus_server_CFLAGS = \
 	$(TESTDEPS_CFLAGS) $(LIBRARY_CFLAGS) \
-	-Wall -Werror \
+	-Wall \
 	-I$(top_srcdir)/src
 
 test_libappindicator_dbus_server_LDADD = \
@@ -83,7 +83,7 @@ test_libappindicator_status_client_SOURC
 
 test_libappindicator_status_client_CFLAGS = \
 	$(TESTDEPS_CFLAGS) $(LIBRARY_CFLAGS) \
-	-Wall -Werror \
+	-Wall \
 	-I$(top_srcdir)/src
 
 test_libappindicator_status_client_LDADD = \
@@ -100,7 +100,7 @@ test_libappindicator_status_server_SOURC
 
 test_libappindicator_status_server_CFLAGS = \
 	$(TESTDEPS_CFLAGS) $(LIBRARY_CFLAGS) \
-	-Wall -Werror \
+	-Wall \
 	-I$(top_srcdir)/src
 
 test_libappindicator_status_server_LDADD = \
@@ -116,7 +116,7 @@ test_libappindicator_fallback_watcher_SO
 
 test_libappindicator_fallback_watcher_CFLAGS = \
 	$(TESTDEPS_CFLAGS) $(LIBRARY_CFLAGS) \
-	-Wall -Werror \
+	-Wall \
 	-I$(top_srcdir)/src
 
 test_libappindicator_fallback_watcher_LDADD = \
@@ -128,7 +128,7 @@ test_libappindicator_fallback_item_SOURC
 
 test_libappindicator_fallback_item_CFLAGS = \
 	$(TESTDEPS_CFLAGS) $(LIBRARY_CFLAGS) \
-	-Wall -Werror \
+	-Wall \
 	-I$(top_srcdir)/src
 
 test_libappindicator_fallback_item_LDADD = \
@@ -194,7 +194,7 @@ test_simple_app_SOURCES = \
 
 test_simple_app_CFLAGS = \
 	$(TESTDEPS_CFLAGS) $(LIBRARY_CFLAGS) \
-	-Wall -Werror \
+	-Wall \
 	-I$(top_srcdir)/src
 
 test_simple_app_LDADD = \
