--- origsrc/clipper-6.4.2/cpp/CMakeLists.txt	2015-04-23 09:04:20.000000000 +0900
+++ src/clipper-6.4.2/cpp/CMakeLists.txt	2020-08-08 20:40:18.733464700 +0900
@@ -5,7 +5,7 @@ SET(CMAKE_BUILD_TYPE "Release" CACHE STR
 # The header name clipper.hpp is too generic, so install in a subdirectory
 SET(CMAKE_INSTALL_INCDIR "${CMAKE_INSTALL_PREFIX}/include/polyclipping")
 SET(CMAKE_INSTALL_LIBDIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
-SET(CMAKE_INSTALL_PKGCONFIGDIR "${CMAKE_INSTALL_PREFIX}/share/pkgconfig")
+SET(CMAKE_INSTALL_PKGCONFIGDIR "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig")
 SET(PCFILE "${CMAKE_CURRENT_BINARY_DIR}/polyclipping.pc")
 
 SET(BUILD_SHARED_LIBS ON CACHE BOOL
@@ -15,7 +15,13 @@ ADD_LIBRARY(polyclipping clipper.cpp)
 CONFIGURE_FILE (polyclipping.pc.cmakein "${PCFILE}" @ONLY)
 
 INSTALL (FILES clipper.hpp DESTINATION "${CMAKE_INSTALL_INCDIR}")
-INSTALL (TARGETS polyclipping LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+INSTALL (TARGETS polyclipping
+    EXPORT polyclipping
+    RUNTIME DESTINATION bin
+    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 INSTALL (FILES "${PCFILE}" DESTINATION "${CMAKE_INSTALL_PKGCONFIGDIR}")
+INSTALL (EXPORT polyclipping
+    FILE clipper-config.cmake
+    DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/cmake/clipper")
 
 SET_TARGET_PROPERTIES(polyclipping PROPERTIES VERSION 22.0.0 SOVERSION 22 )
