--- origsrc/libhpack/CMakeLists.txt	2025-12-08 13:31:57.000000000 +0900
+++ src/libhpack/CMakeLists.txt	2025-12-08 13:32:27.401664600 +0900
@@ -20,6 +20,9 @@ math(EXPR hpack_SOVERSION_PATCH "${hpack
 set(hpack_VERSION ${hpack_SOVERSION_MAJOR}.${hpack_SOVERSION_MINOR}.${hpack_SOVERSION_PATCH})
 set(hpack_SOVERSION ${hpack_SOVERSION_MAJOR})
 
+set(chula_VERSION ${hpack_VERSION})
+set(chula_SOVERSION ${hpack_SOVERSION})
+
 # Standard replacements
 set(prefix      ${CMAKE_INSTALL_PREFIX})
 set(exec_prefix ${CMAKE_INSTALL_PREFIX})
--- origsrc/libhpack/libchula/CMakeLists.txt	2025-12-08 13:31:57.000000000 +0900
+++ src/libhpack/libchula/CMakeLists.txt	2025-12-08 13:56:52.983909700 +0900
@@ -17,7 +17,7 @@ list(REMOVE_ITEM chula_HDRS_wo chula-ret
 
 add_library (
     ${LIB_NAME}
-    STATIC
+    SHARED
     ${chula_SRCS}
 )
 
@@ -31,9 +31,9 @@ set_target_properties (
 
 # Auto-ret header
 add_custom_command (
-    OUTPUT  ${CMAKE_CURRENT_SOURCE_DIR}/chula-ret.h
+    OUTPUT  ${CMAKE_CURRENT_BINARY_DIR}/chula-ret.h
     COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/../tools/auto-ret.py
-    ARGS    --path ${CMAKE_CURRENT_SOURCE_DIR} --output ${CMAKE_CURRENT_SOURCE_DIR}/chula-ret.h --suffix _RET
+    ARGS    --path ${CMAKE_CURRENT_SOURCE_DIR} --output ${CMAKE_CURRENT_BINARY_DIR}/chula-ret.h --suffix _RET
     DEPENDS ${chula_HDRS_wo}
     COMMENT "Generating the chula-ret.h file..."
 )
@@ -41,6 +41,7 @@ add_custom_command (
 # Installation
 install (
   TARGETS ${LIB_NAME}
+  RUNTIME DESTINATION bin
   ARCHIVE DESTINATION lib COMPONENT Development
   LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
 )
--- origsrc/libhpack/libchula/test/OOM/CMakeLists.txt	2025-12-08 13:31:57.000000000 +0900
+++ src/libhpack/libchula/test/OOM/CMakeLists.txt	2025-12-08 13:32:27.414372700 +0900
@@ -7,7 +7,7 @@ add_dependencies (oom_libchula chula chu
 include_directories (${CMAKE_SOURCE_DIR})
 target_link_libraries(oom_libchula chula chula-qa)
 
-if (UNIX AND NOT APPLE)
+if (UNIX AND NOT APPLE AND NOT CYGWIN)
     set_target_properties (
         oom_libchula
         PROPERTIES
--- origsrc/libhpack/libchula/test/util_test.c	2025-12-08 13:31:57.000000000 +0900
+++ src/libhpack/libchula/test/util_test.c	2025-12-08 13:32:27.417396700 +0900
@@ -51,7 +51,7 @@
 # define LOG_INFO 6
 #endif
 
-START_TEST (_strerror_r)
+START_TEST (_strerror_r_)
 {
     char *s;
     char  tmp[256];
@@ -549,7 +549,7 @@ START_TEST (_localtime)
 }
 END_TEST
 
-START_TEST (_timezone)
+START_TEST (_timezone_)
 {
     const int  _12h1s = (12*60*60) + 1;
     long      *tz     = NULL;
@@ -927,7 +927,7 @@ util_tests (void)
 {
     Suite *s1 = suite_create("Utility Functions");
 
-    check_add (s1, _strerror_r);
+    check_add (s1, _strerror_r_);
     check_add (s1, min_str);
     check_add (s1, max_str);
     check_add (s1, _atoi);
--- origsrc/libhpack/libchula-qa/CMakeLists.txt	2025-12-08 13:31:57.000000000 +0900
+++ src/libhpack/libchula-qa/CMakeLists.txt	2025-12-08 13:52:04.836279800 +0900
@@ -8,7 +8,7 @@ file(GLOB chula_qa_HDRS *.h)
 
 add_library (
     ${LIB_NAME}
-    STATIC
+    SHARED
     ${chula_qa_SRCS}
 )
 
@@ -20,7 +20,7 @@ set_target_properties (
     COMPILE_FLAGS -DCHULA_QA_COMPILATION
 )
 
-add_dependencies(${LIB_NAME} libchula)
+add_dependencies(${LIB_NAME} chula)
 
 if (UNIX AND NOT APPLE)
     set_target_properties (
@@ -37,6 +37,7 @@ target_link_libraries (${LIB_NAME} ${CHE
 # Installation
 install (
   TARGETS ${LIB_NAME}
+  RUNTIME DESTINATION bin
   ARCHIVE DESTINATION lib COMPONENT Development
   LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
 )
--- origsrc/libhpack/libchula-qa/mem_mgr.c	2025-12-08 13:31:57.000000000 +0900
+++ src/libhpack/libchula-qa/mem_mgr.c	2025-12-08 13:32:27.425467900 +0900
@@ -133,7 +133,7 @@ chula_mem_mgr_thaw (chula_mem_mgr_t *mgr
 /* Guts
  */
 
-#ifdef LINUX
+#if defined(LINUX) || defined(__CYGWIN__)
 # define FUNC_MALLOC(name)  static void *_ ## name ## _malloc  (size_t size)
 # define FUNC_REALLOC(name) static void *_ ## name ## _realloc (void *ptr, size_t size)
 # define FUNC_FREE(name)    static void  _ ## name ## _free    (void *ptr)
--- origsrc/libhpack/libhpack/CMakeLists.txt	2025-12-08 13:31:57.000000000 +0900
+++ src/libhpack/libhpack/CMakeLists.txt	2025-12-08 13:32:27.427483700 +0900
@@ -11,10 +11,12 @@ list(REMOVE_ITEM hpack_HDRS_wo hpack-ret
 
 add_library (
     ${LIB_NAME}
-    STATIC
+    SHARED
     ${hpack_SRCS}
 )
 
+target_link_libraries(${LIB_NAME} chula)
+
 set_target_properties (
     ${LIB_NAME}
     PROPERTIES
@@ -44,6 +46,7 @@ target_link_libraries (${LIB_NAME} ${LIB
 # Installation
 install (
   TARGETS ${LIB_NAME}
+  RUNTIME DESTINATION bin
   ARCHIVE DESTINATION lib COMPONENT Development
   LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
 )
--- origsrc/libhpack/libhpack/header_field.h	2025-12-08 13:31:57.000000000 +0900
+++ src/libhpack/libhpack/header_field.h	2025-12-08 13:59:42.842846800 +0900
@@ -80,7 +80,7 @@ typedef enum {
  * String representation of the different header field types we may encounter as
  * enumerated in [hpack_header_field_type_t](@ref hpack_header_field_type_t).
  */
-chula_buffer_t hpack_header_field_field_type_repr[4];
+extern chula_buffer_t hpack_header_field_field_type_repr[4];
 
 
 /**
@@ -100,7 +100,7 @@ typedef enum {
  * String representation of the different header field representations we may
  * encounter as enumerated in [hpack_header_field_representation_t](@ref hpack_header_field_representation_t).
  */
-chula_buffer_t hpack_header_field_representations_repr[6];
+extern chula_buffer_t hpack_header_field_representations_repr[6];
 
 /**
  * Flags representing how this field was encoded.
--- origsrc/libhpack/tools/auto-ret.py	2025-12-08 13:31:57.000000000 +0900
+++ src/libhpack/tools/auto-ret.py	2025-12-08 13:53:59.240198200 +0900
@@ -39,7 +39,7 @@ def parse_file (fullpath_h, replacement_
 		# Parse it
 		tmp = (re.findall (r'ret_t (.+?)\s*(\(.+\));', func) or [()])[0]
 		if not tmp:
-			print func
+			print(func)
 			raise SystemExit
 		func_orig = tmp[0]
 		func_name = func_orig.replace(replacement_pair[0], replacement_pair[1])
@@ -50,7 +50,7 @@ def parse_file (fullpath_h, replacement_
 		for arg in func_args:
 			if arg == '(void)':
 				break
-			args += [filter(lambda x: len(x), re.split(' |\*|\&|\)', arg))[-1]]
+			args += [list(filter(lambda x: len(x), re.split(' |\*|\&|\)', arg)))[-1]]
 		macro_args = '(%s)' %(','.join(args))
 
 		output += MACRO %(locals())
@@ -64,7 +64,7 @@ def parse_dir (path_dir, path_header, re
 		for filename in fnmatch.filter(filenames, "*.h"):
 			fp = os.path.abspath (os.path.join (root, filename))
 			if fp == path_header:
-				print "Ignoring %s" %(fp)
+				print("Ignoring %s" %(fp))
 				continue
 			header += parse_file(fp, replacement_pair, func_suffix)
 
@@ -72,7 +72,7 @@ def parse_dir (path_dir, path_header, re
 	with open(path_header, 'w+') as f:
 		f.write(header)
 
-	print '%s: %s definitions' %(path_header, header.count('#define'))
+	print('%s: %s definitions' %(path_header, header.count('#define')))
 
 
 def main():
