--- origsrc/FreeSOLID-2.1.1/FreeSOLID.pc	2015-02-22 18:25:51.181000200 +0900
+++ src/FreeSOLID-2.1.1/FreeSOLID.pc	2015-02-22 18:49:19.214203400 +0900
@@ -1,10 +1,11 @@
 prefix=/usr
-exec_prefix=/usr
 libdir=/usr/lib
-includedir=/usr/include
+includedir=${prefix}/include
 
 Name: FreeSolid
 Description: 3D collision detection C++ library
 Version: 2.1.1
-Libs: -L${libdir} -lFreeSOLID
+Requires: qhull
+Libs: -L${libdir} -lFreeSOLID -lqhull
 Cflags: -I${includedir} -I/usr/include/FreeSOLID
+
--- origsrc/FreeSOLID-2.1.1/configure.in	2015-02-22 18:25:51.306000200 +0900
+++ src/FreeSOLID-2.1.1/configure.in	2015-02-22 18:47:08.870453400 +0900
@@ -1,5 +1,5 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT
+AC_INIT([FreeSOLID], [2.1.1])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_SRCDIR([sample/sample.cpp])
 
@@ -16,7 +16,7 @@ AC_SUBST(FREESOLID_REVISION)
 AC_SUBST(FREESOLID_AGE)
 AC_SUBST(VERSION)
 
-AM_INIT_AUTOMAKE($PACKAGE, $VERSION)
+AM_INIT_AUTOMAKE([subdir-objects])
 
 AC_CONFIG_HEADERS([include/config.h])
 
--- origsrc/FreeSOLID-2.1.1/libsolid/Makefile.am	2015-02-22 18:25:51.556000200 +0900
+++ src/FreeSOLID-2.1.1/libsolid/Makefile.am	2015-02-22 18:32:26.776703400 +0900
@@ -23,7 +23,7 @@ libFreeSOLID_la_SOURCES = \
 
 libFreeSOLID_la_LDFLAGS = -no-undefined -version-info @FREESOLID_CURRENT@:@FREESOLID_REVISION@:@FREESOLID_AGE@ 
 
-AM_CPPFLAGS = @CPPFLAGS@
+AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/libmoto @CPPFLAGS@
 LIBS = @QHULL_LIBS@ -lm
 
 pkginclude_HEADERS = \
--- origsrc/FreeSOLID-2.1.1/sample/Makefile.am	2015-02-22 18:25:50.212250200 +0900
+++ src/FreeSOLID-2.1.1/sample/Makefile.am	2015-02-22 18:51:03.585297100 +0900
@@ -4,7 +4,7 @@ sample_SOURCES = sample.cpp
 x_wing_SOURCES = x_wing.cpp
 teapot_SOURCES = teapot.cpp
 
-AM_CPPFLAGS = @CPPFLAGS@
+AM_CPPFLAGS = -I$(top_srcdir)/include @CPPFLAGS@
 
 LDADD = ../libsolid/libFreeSOLID.la
 
