--- origsrc/libewf-20240506/Makefile.am	2024-05-06 13:26:24.000000000 +0900
+++ src/libewf-20240506/Makefile.am	2025-05-20 09:26:07.991669700 +0900
@@ -3,32 +3,10 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libcdatetime \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
-	libfguid \
-	libfvalue \
-	libhmac \
-	libcaes \
 	libewf \
-	libodraw \
-	libsmdev \
-	libsmraw \
 	ewftools \
 	ewf.net \
 	pyewf \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libewf-20240506/configure.ac	2024-05-06 13:26:24.000000000 +0900
+++ src/libewf-20240506/configure.ac	2025-05-20 09:25:50.369786600 +0900
@@ -230,30 +230,9 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libcdatetime/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
-AC_CONFIG_FILES([libhmac/Makefile])
-AC_CONFIG_FILES([libcaes/Makefile])
 AC_CONFIG_FILES([libewf/Makefile])
 AC_CONFIG_FILES([pyewf/Makefile])
 AC_CONFIG_FILES([ewf.net/Makefile])
-AC_CONFIG_FILES([libodraw/Makefile])
-AC_CONFIG_FILES([libsmdev/Makefile])
-AC_CONFIG_FILES([libsmraw/Makefile])
 AC_CONFIG_FILES([ewftools/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
--- origsrc/libewf-20240506/libewf/libewf_chunk_group.c	2024-05-06 13:26:24.000000000 +0900
+++ src/libewf-20240506/libewf/libewf_chunk_group.c	2025-05-20 09:36:57.326365400 +0900
@@ -474,6 +474,7 @@ int libewf_chunk_group_fill_v1(
 		}
 		if( chunk_data_size == 0 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
@@ -481,10 +482,12 @@ int libewf_chunk_group_fill_v1(
 				 function,
 				 table_entry_index );
 			}
+#endif
 			corrupted = 1;
 		}
 		if( chunk_data_size > (uint32_t) INT32_MAX )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
@@ -492,6 +495,7 @@ int libewf_chunk_group_fill_v1(
 				 function,
 				 table_entry_index );
 			}
+#endif
 			corrupted = 1;
 		}
 		if( is_compressed != 0 )
@@ -979,7 +983,7 @@ int libewf_chunk_group_fill_v2(
 		{
 		}
 */
-#if defined( HAVE_VERBOSE_OUTPUT )
+#if defined( HAVE_DEBUG_OUTPUT )
 		if( libcnotify_verbose != 0 )
 		{
 			if( ( chunk_data_flags & ~( 0x00000007UL ) ) != 0 )
@@ -1191,6 +1195,7 @@ int libewf_chunk_group_correct_v1(
 		}
 		if( chunk_data_size == 0 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
@@ -1198,10 +1203,12 @@ int libewf_chunk_group_correct_v1(
 				 function,
 				 chunk_index );
 			}
+#endif
 			corrupted = 1;
 		}
 		if( chunk_data_size > (uint32_t) INT32_MAX )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
@@ -1209,6 +1216,7 @@ int libewf_chunk_group_correct_v1(
 				 function,
 				 chunk_index );
 			}
+#endif
 			corrupted = 1;
 		}
 		range_flags = LIBEWF_RANGE_FLAG_HAS_CHECKSUM;
--- origsrc/libewf-20240506/pyewf/Makefile.am	2024-05-06 13:26:24.000000000 +0900
+++ src/libewf-20240506/pyewf/Makefile.am	2025-05-20 09:26:17.271220100 +0900
@@ -47,7 +47,7 @@ pyewf_la_LIBADD = \
 	@LIBBFIO_LIBADD@
 
 pyewf_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyewf_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyewf_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
--- origsrc/libewf-20240506/tests/Makefile.am	2024-05-06 13:26:24.000000000 +0900
+++ src/libewf-20240506/tests/Makefile.am	2025-05-20 09:50:26.776699300 +0900
@@ -3,6 +3,7 @@ AUTOMAKE_OPTIONS = subdir-objects
 AM_CPPFLAGS = \
 	-I../include -I$(top_srcdir)/include \
 	-I../common -I$(top_srcdir)/common \
+	-I$(top_srcdir)/libewf \
 	@LIBCERROR_CPPFLAGS@ \
 	@LIBCTHREADS_CPPFLAGS@ \
 	@LIBCDATA_CPPFLAGS@ \
