--- origsrc/cdson-1.0.0/meson.build	2021-12-04 05:59:54.000000000 +0900
+++ src/cdson-1.0.0/meson.build	2023-09-16 01:21:15.114470200 +0900
@@ -17,6 +17,7 @@ add_global_arguments('-D_GNU_SOURCE', la
 cdson = library('cdson', 'cdson.h',
                 'src/dump.c', 'src/sniff.c', 'src/fetch.c',
                 'src/allocation.h', 'src/unicode.h', 'src/unicode.c',
+                soversion: 1,
                 dependencies: deps,
                 install: true)
 pkg = import('pkgconfig')
