--- origsrc/fqzcomp-4.6/Makefile	2013-05-23 23:18:41.000000000 +0900
+++ src/fqzcomp-4.6/Makefile	2025-12-24 03:10:15.608884300 +0900
@@ -2,11 +2,13 @@ VERS=4.6
 
 all: fqz_comp
 
-CC = g++
+CXX ?= g++
 #CFLAGS  = -g -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
+CXXFLAGS ?=
 CFLAGS  = -O2 -g -fomit-frame-pointer -fstrict-aliasing -ffast-math -DNDEBUG -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -Wall -msse2
+CXXFLAGS += $(CFLAGS)
 
-#CC=icpc
+#CXX=icpc
 #CFLAGS  = -O3 -g -fomit-frame-pointer -fstrict-aliasing -fast -DNDEBUG -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -Wall -m64 -msse2
 
 CFLAGS += -DPTHREADS
@@ -15,10 +17,10 @@ LIBS += -lpthread
 #CFLAGS += -DORIG_MODEL
 
 .c.o:
-	$(CC) $(CFLAGS) -c $< 
+	$(CXX) $(CXXFLAGS) -c $< 
 
 fqz_comp: fqz_comp.o sfh.o
-	$(CC) $(CFLAGS) -o $@ fqz_comp.o sfh.o $(LIBS)
+	$(CXX) $(CXXFLAGS) -o $@ fqz_comp.o sfh.o $(LIBS)
 
 clean:
 	-rm *.o fqz_comp
