--- origsrc/gcab-0.7/configure.ac	2015-12-04 22:15:06.000000000 +0900
+++ src/gcab-0.7/configure.ac	2016-03-15 14:12:40.867062500 +0900
@@ -1,5 +1,5 @@
 AC_PREREQ(2.57)
-AC_INIT([gcab], m4_esyscmd([build-aux/git-version-gen .tarball-version]),
+AC_INIT([gcab], [0.7],
         [https://bugzilla.gnome.org/enter_bug.cgi?product=msitools&component=gcab])
 AC_CONFIG_SRCDIR([gcab.c])
 AC_CONFIG_AUX_DIR([build-aux])
