--- origsrc/libClustalW/libClustalW/Makefile.am	2008-02-24 19:12:47.000000000 +0900
+++ src/libClustalW/libClustalW/Makefile.am	2014-12-29 20:21:06.931197300 +0900
@@ -13,6 +13,6 @@ library_include_HEADERS = $(LIBCLUSTALW_
 
 lib_LTLIBRARIES = libClustalW-1.0.la
 libClustalW_1_0_la_SOURCES = $(LIBCLUSTALW_SRC)
-libClustalW_1_0_la_LDFLAGS= -version-info $(GENERIC_LIBRARY_VERSION) 
+libClustalW_1_0_la_LDFLAGS= -no-undefined -version-info $(GENERIC_LIBRARY_VERSION)
 
 
