--- origsrc/libClustalW/libClustalW/Makefile.am	2012-04-03 00:32:22.000000000 +0900
+++ src/libClustalW/libClustalW/Makefile.am	2012-04-03 00:32:39.414319700 +0900
@@ -13,6 +13,6 @@ library_include_HEADERS = $(LIBCLUSTALW_
 
 lib_LTLIBRARIES = libClustalW-1.0.la
 libClustalW_1_0_la_SOURCES = $(LIBCLUSTALW_SRC)
-libClustalW_1_0_la_LDFLAGS= -version-info $(GENERIC_LIBRARY_VERSION) 
+libClustalW_1_0_la_LDFLAGS= -version-info $(GENERIC_LIBRARY_VERSION) -no-undefined
 
 
