--- origsrc/libcxi/CMakeLists.txt	2024-10-14 15:07:36.000000000 +0900
+++ src/libcxi/CMakeLists.txt	2024-10-14 15:08:02.045022600 +0900
@@ -13,24 +13,34 @@ include_directories(${HDF5_INCLUDE_DIR}
 add_library(cxi SHARED src/cxi.c include/cxi.h)
 target_link_libraries(cxi ${HDF5_LIBRARIES})
 
+if (DEFINED SOVERSION)
+        set_target_properties(cxi PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 add_executable(simple EXCLUDE_FROM_ALL src/cxi.c tests/simple.c)
 target_link_libraries(simple ${HDF5_LIBRARIES})
 
 add_executable(writer EXCLUDE_FROM_ALL src/cxi.c tests/writer.c)
 target_link_libraries(writer ${HDF5_LIBRARIES})
 
-add_executable(typical_reader  src/cxi.c examples/typical_reader.c)
-target_link_libraries(typical_reader ${HDF5_LIBRARIES})
-
-add_executable(typical_writer  src/cxi.c examples/typical_writer.c)
-target_link_libraries(typical_writer ${HDF5_LIBRARIES})
+add_executable(typical_reader examples/typical_reader.c)
+target_link_libraries(typical_reader cxi ${HDF5_LIBRARIES})
 
-add_executable(minimal_reader  src/cxi.c examples/minimal_reader.c)
-target_link_libraries(minimal_reader ${HDF5_LIBRARIES})
+add_executable(minimal_reader examples/minimal_reader.c)
+target_link_libraries(minimal_reader cxi ${HDF5_LIBRARIES})
 
-add_executable(minimal_writer  src/cxi.c examples/minimal_writer.c)
-target_link_libraries(minimal_writer ${HDF5_LIBRARIES})
+add_executable(minimal_writer examples/minimal_writer.c)
+target_link_libraries(minimal_writer cxi ${HDF5_LIBRARIES})
 
+install(FILES include/cxi.h DESTINATION include)
+install(TARGETS cxi RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libcxi.pc.in ${CMAKE_BINARY_DIR}/libcxi.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libcxi.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
 
 enable_testing()
 add_custom_target(check COMMAND ${CMAKE_CTEST_COMMAND})
@@ -38,6 +48,3 @@ add_test(simple simple ${CMAKE_SOURCE_DI
 
 add_test(writer writer ${CMAKE_SOURCE_DIR}/data/dummy.cxi)
 add_dependencies(check simple writer)
-
-
-
--- origsrc/libcxi/libcxi.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libcxi/libcxi.pc.in	2024-10-14 15:07:37.778685900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libcxi
+Description: Library to help read and write CXI files
+Version: @VERSION@
+Libs: -L${libdir} -lcxi
+Cflags: -I${includedir}
