--- origsrc/open62541-1.0.1/CMakeLists.txt	2020-01-30 21:08:22.000000000 +0900
+++ src/open62541-1.0.1/CMakeLists.txt	2020-03-05 15:24:35.125422600 +0900
@@ -515,6 +515,7 @@ endif()
 file(MAKE_DIRECTORY "${PROJECT_BINARY_DIR}/src_generated")
 configure_file(include/open62541/config.h.in ${PROJECT_BINARY_DIR}/src_generated/open62541/config.h)
 configure_file(open62541.pc.in ${PROJECT_BINARY_DIR}/src_generated/open62541.pc @ONLY)
+install(FILES ${PROJECT_BINARY_DIR}/src_generated/open62541.pc DESTINATION lib/pkgconfig)
 
 if(UA_ENABLE_DISCOVERY_MULTICAST)
     set(MDNSD_LOGLEVEL 300 CACHE STRING "Level at which logs shall be reported" FORCE)
@@ -1160,7 +1161,7 @@ install(TARGETS open62541 ${EXTRATARGETS
         EXPORT open62541Targets
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
         INCLUDES DESTINATION include)
 
 if(UA_ENABLE_AMALGAMATION)
