--- origsrc/open62541-1.3/CMakeLists.txt	2022-04-05 05:33:03.000000000 +0900
+++ src/open62541-1.3/CMakeLists.txt	2022-06-05 23:31:29.354835400 +0900
@@ -41,8 +41,8 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMA
 # number from the latest release plus the "-undefined" label. Will be
 # overwritten with more detailed information if git is available.
 set(OPEN62541_VER_MAJOR 1)
-set(OPEN62541_VER_MINOR 2)
-set(OPEN62541_VER_PATCH 2)
+set(OPEN62541_VER_MINOR 3)
+set(OPEN62541_VER_PATCH 0)
 set(OPEN62541_VER_LABEL "-unknown") # like "-rc1" or "-g4538abcd" or "-g4538abcd-dirty"
 set(OPEN62541_VER_COMMIT "undefined")
 
@@ -1756,3 +1756,12 @@ set_target_properties(open62541-generato
 set_target_properties(open62541-generator-statuscode PROPERTIES FOLDER "open62541/generators")
 set_target_properties(open62541-generator-transport PROPERTIES FOLDER "open62541/generators")
 set_target_properties(open62541-generator-types PROPERTIES FOLDER "open62541/generators")
+
+if (CYGWIN)
+  set(prefix ${CMAKE_INSTALL_PREFIX})
+  set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+  set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+  set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+  configure_file(${CMAKE_SOURCE_DIR}/open62541.pc.in ${CMAKE_BINARY_DIR}/open62541.pc @ONLY)
+  install(FILES ${CMAKE_BINARY_DIR}/open62541.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+endif()
--- origsrc/open62541-1.3/open62541.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/open62541-1.3/open62541.pc.in	2022-06-05 23:32:33.659135600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: open62541
+Description: Open source implementation of OPC UA (OPC Unified Architecture) aka IEC 62541
+Version: @VERSION@
+Libs: -L${libdir} -lopen62541
+Cflags: -I${includedir}
