--- origsrc/open62541-1.4.12/CMakeLists.txt	2025-06-20 01:57:55.376369700 +0900
+++ src/open62541-1.4.12/CMakeLists.txt	2025-06-20 11:06:18.441028300 +0900
@@ -510,7 +510,7 @@ if(UA_ENABLE_ENCRYPTION_OPENSSL)
     # use the OpenSSL encryption library
     # https://cmake.org/cmake/help/v3.13/module/FindOpenSSL.html
     find_package(OpenSSL REQUIRED)
-    list(APPEND open62541_LIBRARIES "${OPENSSL_LIBRARIES}")
+    list(APPEND open62541_LIBRARIES "${_OPENSSL_LIBRARIES}")
     if(WIN32)
         # Add bestcrypt for windows systems
         list(APPEND open62541_LIBRARIES bcrypt)
--- origsrc/open62541-1.4.12/arch/eventloop_posix.c	2025-05-15 23:27:26.000000000 +0900
+++ src/open62541-1.4.12/arch/eventloop_posix.c	2025-06-20 02:05:32.902795100 +0900
@@ -553,8 +553,10 @@ UA_EventLoopPOSIX_setReusable(UA_FD sock
 #ifndef UA_ARCHITECTURE_WIN32
     int res = UA_setsockopt(sockfd, SOL_SOCKET, SO_REUSEADDR,
                             (const char*)&enableReuseVal, sizeof(enableReuseVal));
+#if defined(SO_REUSEPORT)
     res |= UA_setsockopt(sockfd, SOL_SOCKET, SO_REUSEPORT,
                             (const char*)&enableReuseVal, sizeof(enableReuseVal));
+#endif
     return (res == 0) ? UA_STATUSCODE_GOOD : UA_STATUSCODE_BADINTERNALERROR;
 #else
     int res = UA_setsockopt(sockfd, SOL_SOCKET, SO_REUSEADDR,
