diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/gde-0.09/Makefile src/gde-0.09/Makefile
--- origsrc/gde-0.09/Makefile	2011-06-24 04:18:44.000000000 +0900
+++ src/gde-0.09/Makefile	2015-09-26 23:32:37.801638400 +0900
@@ -84,10 +84,10 @@ clean:
 	-mv *~ junk
 
 tests:
-	gde testaopb.exe > testaopb.out
-	gde testaopw.exe > testaopw.out
-	gde testbopb.exe > testbopb.out
-	gde testbopw.exe > testbopw.out
+	./gde testaopb.exe > testaopb.out || :
+	./gde testaopw.exe > testaopw.out || :
+	./gde testbopb.exe > testbopb.out || :
+	./gde testbopw.exe > testbopw.out || :
 
 
 sys_init.o:sys_init.c            $(INCLUDES) Makefile
@@ -119,5 +119,3 @@ kbd_termios.o: kbd_termios.c     $(INCLU
 kbd_stdio.o: kbd_stdio.c         $(INCLUDES) Makefile
 
 main.o: main.c                   $(INCLUDES) Makefile
-
-
