--- origsrc/iniparser/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/iniparser/Makefile.am	2012-05-15 15:12:13.161396300 +0900
@@ -0,0 +1,20 @@
+
+AM_CPPFLAGS = -I$(top_srcdir)/src
+
+iniparser_includedir = $(includedir)/iniparser
+iniparser_include_HEADERS = src/iniparser.h src/dictionary.h
+
+
+lib_LTLIBRARIES = libiniparser.la
+
+libiniparser_la_SOURCES = src/iniparser.c src/dictionary.c
+libiniparser_la_LDFLAGS = -no-undefined
+
+
+check_PROGRAMS = iniexample parse
+
+iniexample_SOURCES = test/iniexample.c
+iniexample_LDADD = libiniparser.la
+parse_SOURCES = test/parse.c
+parse_LDADD = libiniparser.la
+
--- origsrc/iniparser/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/iniparser/configure.ac	2012-05-15 15:10:21.074482200 +0900
@@ -0,0 +1,24 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([iniparser], [3.1])
+AC_CONFIG_SRCDIR([src/dictionary.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h string.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([memset strchr strtol])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
