--- origsrc/libosmo-dsp-0.3/src/Makefile.am	2013-06-17 06:47:36.000000000 +0900
+++ src/libosmo-dsp-0.3/src/Makefile.am	2015-04-19 21:00:27.709162800 +0900
@@ -8,4 +8,4 @@ AM_CFLAGS = -Wall -ffast-math $(FFTW3F_C
 lib_LTLIBRARIES = libosmodsp.la
 
 libosmodsp_la_SOURCES = cfile.c cxvec.c cxvec_math.c iqbal.c
-libosmodsp_la_LDFLAGS = -version-info $(LIBVERSION) $(FFTW3F_LIBS)
+libosmodsp_la_LDFLAGS = -no-undefined -version-info $(LIBVERSION) $(FFTW3F_LIBS)
