--- origsrc/radius-engine-1.1/Makefile.am	2024-11-08 14:54:39.917046300 +0900
+++ src/radius-engine-1.1/Makefile.am	2024-11-08 14:55:46.080673500 +0900
@@ -93,7 +93,7 @@ libradius_engine_la_SOURCES = \
                              radius.c \
                              radius.h
 
-libradius_engine_la_LDFLAGS = -version-info $(LIBRADIUSENGINE_LIBRARY_VERSION)
+libradius_engine_la_LDFLAGS = -no-undefined -version-info $(LIBRADIUSENGINE_LIBRARY_VERSION)
 
 dist_noinst_HEADERS = r_platform_defs_windows.h \
                       r_platform_windows.h
