--- origsrc/libopkele/Makefile.am	2012-03-21 13:53:23.000000000 +0900
+++ src/libopkele/Makefile.am	2012-03-21 13:58:45.517379400 +0900
@@ -1,7 +1,7 @@
 SUBDIRS=include lib test
 EXTRA_DIST= NEWS NEWS.xml NEWS.xsl
 
-ACLOCAL_AMFLAGS=-I aclocal.d
+ACLOCAL_AMFLAGS=-I acinclude.d
 
 pkgconfigdir=${libdir}/pkgconfig
 pkgconfig_DATA=libopkele.pc
@@ -13,7 +13,7 @@ clean-local:
 endif
 
 NEWS: NEWS.xsl NEWS.xml
-	${XSLTPROC} -o $@ NEWS.xsl NEWS.xml
+	${XSLTPROC} -o $@ $(top_srcdir)/NEWS.xsl $(top_srcdir)/NEWS.xml
 
 if HAVE_DOXYGEN
 dox: Doxyfile
--- origsrc/libopkele/configure.ac	2012-03-21 13:53:23.000000000 +0900
+++ src/libopkele/configure.ac	2012-03-21 13:55:25.689254400 +0900
@@ -1,7 +1,7 @@
 AC_INIT([libopkele], [2.0.4], [libopkele-bugs@klever.net])
 AC_CONFIG_SRCDIR([include/opkele/opkele-config.h])
 AC_CONFIG_HEADERS([config.h include/opkele/acconfig.h])
-AC_CONFIG_MACRO_DIR([aclocal.d])
+AC_CONFIG_MACRO_DIR([acinclude.d])
 AC_CONFIG_AUX_DIR([aux.d])
 AM_INIT_AUTOMAKE([dist-bzip2])
 
--- origsrc/libopkele/lib/Makefile.am	2012-03-21 13:53:24.000000000 +0900
+++ src/libopkele/lib/Makefile.am	2012-03-21 13:53:43.904249200 +0900
@@ -30,4 +30,4 @@ libopkele_la_SOURCES = \
 		       basic_op.cc verify_op.cc \
 		       ax.cc oauth_ext.cc
 libopkele_la_LDFLAGS = \
-	-version-info 3:0:0
+	-version-info 3:0:0 -no-undefined
