--- origsrc/libopkele-2.0.4/Makefile.am	2009-08-31 20:57:35.000000000 +0900
+++ src/libopkele-2.0.4/Makefile.am	2010-11-29 02:59:21.752001800 +0900
@@ -1,4 +1,4 @@
-SUBDIRS=include lib test
+SUBDIRS=include lib #test
 EXTRA_DIST= NEWS NEWS.xml NEWS.xsl
 
 ACLOCAL_AMFLAGS=-I aclocal.d
--- origsrc/libopkele-2.0.4/configure.ac	2009-12-23 05:56:57.000000000 +0900
+++ src/libopkele-2.0.4/configure.ac	2010-11-28 18:43:17.888354800 +0900
@@ -8,7 +8,7 @@ AM_INIT_AUTOMAKE([dist-bzip2])
 AC_PROG_INSTALL
 AC_PROG_CXX
 AC_PROG_CC
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 PKG_PROG_PKG_CONFIG
 
 AC_HEADER_STDC
--- origsrc/libopkele-2.0.4/lib/Makefile.am	2009-12-23 05:47:26.000000000 +0900
+++ src/libopkele-2.0.4/lib/Makefile.am	2010-11-28 18:43:36.957662400 +0900
@@ -30,4 +30,4 @@ libopkele_la_SOURCES = \
 		       basic_op.cc verify_op.cc \
 		       ax.cc oauth_ext.cc
 libopkele_la_LDFLAGS = \
-	-version-info 3:0:0
+	-version-info 3:0:0 -no-undefined
