--- origsrc/fidlib-0.9.10/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/fidlib-0.9.10/Makefile.am	2012-03-30 12:23:46.326820900 +0900
@@ -0,0 +1,12 @@
+
+bin_PROGRAMS = firun
+firun_LDADD = libfidlib.la
+firun_SOURCES = firun.c
+
+pkginclude_HEADERS = fidlib.h
+
+lib_LTLIBRARIES = libfidlib.la
+libfidlib_la_CPPFLAGS = -DT_LINUX
+libfidlib_la_LDFLAGS = -no-undefined
+libfidlib_la_SOURCES = fidlib.c
+
--- origsrc/fidlib-0.9.10/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/fidlib-0.9.10/configure.ac	2012-03-30 12:24:29.449867800 +0900
@@ -0,0 +1,30 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([fidlib], [0.9.10])
+AC_CONFIG_SRCDIR([fidlib.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT([disable-static])
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_C_INLINE
+AC_HEADER_STDBOOL
+
+# Checks for library functions.
+AC_FUNC_ERROR_AT_LINE
+AC_FUNC_REALLOC
+AC_FUNC_STRTOD
+AC_CHECK_FUNCS([floor memmove memset pow sqrt strchr strdup strerror strstr strtol])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
