--- origsrc/libmusicxml-3.14/build/CMakeLists.txt	2018-11-19 17:13:42.000000000 +0900
+++ src/libmusicxml-3.14/build/CMakeLists.txt	2019-01-16 08:53:47.836733500 +0900
@@ -147,7 +147,7 @@ endif()
 if (NOT IOS)
 message (STATUS "Install location is ${CMAKE_INSTALL_PREFIX}")		
 install ( TARGETS ${target} 
-	RUNTIME DESTINATION lib 
+	RUNTIME DESTINATION bin 
     LIBRARY DESTINATION lib 
     ARCHIVE DESTINATION lib 
     FRAMEWORK DESTINATION lib CONFIGURATIONS Release
@@ -191,3 +191,10 @@ install (
 )
 
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libmusicxml.pc.in ${CMAKE_BINARY_DIR}/libmusicxml.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libmusicxml.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libmusicxml-3.14/build/libmusicxml.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libmusicxml-3.14/build/libmusicxml.pc.in	2019-01-16 08:54:42.177269700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libmusicxml
+Description: C/C++ library to support the MusicXML format
+Version: @VERSION@
+Libs: -L${libdir} -lmusicxml2
+Cflags: -I${includedir}
