--- origsrc/libkibi-0.1.1/kibi/Makefile.am	2011-01-07 18:51:58.000000000 +0900
+++ src/libkibi-0.1.1/kibi/Makefile.am	2025-08-21 00:04:51.130099400 +0900
@@ -3,6 +3,7 @@ lib_LTLIBRARIES = libkibi.la
 libkibi_la_SOURCES = \
     kibi.c \
     kibi.h
+libkibi_la_LDFLAGS = -no-undefined
 
 library_includedir = $(includedir)
 library_include_HEADERS = kibi.h
--- origsrc/libkibi-0.1.1/test/Makefile.am	2011-01-16 08:14:12.000000000 +0900
+++ src/libkibi-0.1.1/test/Makefile.am	2025-08-21 00:04:51.132034400 +0900
@@ -1,7 +1,7 @@
 check_PROGRAMS = test
 
 test_SOURCES = test.c
-test_CFLAGS = -I$(top_builddir)/kibi $(AM_CFLAGS)
+test_CFLAGS = -I$(top_builddir)/kibi -I$(top_srcdir)/kibi $(AM_CFLAGS)
 test_LDADD = $(top_builddir)/kibi/libkibi.la
 
 test_scripts = test-base10 test-base2 test-historic
