--- origsrc/Healpix_3.50/src/C/autotools/Makefile.am	2014-02-27 18:33:31.000000000 +0900
+++ src/Healpix_3.50/src/C/autotools/Makefile.am	2019-10-25 10:00:48.290753200 +0900
@@ -6,6 +6,7 @@ include_HEADERS = chealpix.h
 
 libchealpix_la_SOURCES = chealpix.c chealpix.h
 libchealpix_la_LIBADD = $(CFITSIO_LIBS)
+libchealpix_la_LDFLAGS = -no-undefined
 
 TESTS = test_chealpix
 check_PROGRAMS = $(TESTS)
--- origsrc/Healpix_3.50/src/C/autotools/configure.ac	2016-03-22 23:28:18.000000000 +0900
+++ src/Healpix_3.50/src/C/autotools/configure.ac	2019-10-25 10:43:15.528994100 +0900
@@ -1,4 +1,4 @@
-AC_INIT([chealpix], [3.30.0])
+AC_INIT([chealpix], [3.50.0])
 AM_INIT_AUTOMAKE([foreign -Wall -Werror])
 
 dnl
--- origsrc/Healpix_3.50/src/cxx/autotools/Makefile.am	2018-11-29 17:40:45.000000000 +0900
+++ src/Healpix_3.50/src/cxx/autotools/Makefile.am	2019-10-25 10:03:53.405812000 +0900
@@ -159,7 +159,7 @@ include_HEADERS = \
 
 libhealpix_cxx_la_SOURCES = $(src_c_utils) $(src_libpocketfft) $(src_libsharp) $(src_cxxsupport) $(src_cxxsupport_fits) $(src_healpix_cxx) $(src_healpix_cxx_fits) $(src_healpix_cxx_mod)
 libhealpix_cxx_la_LIBADD = $(CFITSIO_LIBS)
-libhealpix_cxx_la_LDFLAGS = -version-info 2:1:0
+libhealpix_cxx_la_LDFLAGS = -no-undefined -version-info 2:1:0
 
 bin_PROGRAMS = alm2map_cxx anafast_cxx calc_powspec hotspots_cxx map2tga \
   median_filter_cxx mult_alm rotalm_cxx smoothing_cxx syn_alm_cxx udgrade_cxx \
