--- origsrc/Healpix_3.60/src/C/autotools/Makefile.am	2014-02-27 18:33:31.000000000 +0900
+++ src/Healpix_3.60/src/C/autotools/Makefile.am	2020-07-10 09:51:51.672602600 +0900
@@ -6,6 +6,7 @@ include_HEADERS = chealpix.h
 
 libchealpix_la_SOURCES = chealpix.c chealpix.h
 libchealpix_la_LIBADD = $(CFITSIO_LIBS)
+libchealpix_la_LDFLAGS = -no-undefined
 
 TESTS = test_chealpix
 check_PROGRAMS = $(TESTS)
--- origsrc/Healpix_3.60/src/cxx/Makefile.am	2019-10-30 00:06:22.000000000 +0900
+++ src/Healpix_3.60/src/cxx/Makefile.am	2020-07-10 09:53:15.853339800 +0900
@@ -128,7 +128,7 @@ libhealpix_cxx_la_LIBADD = $(SHARP_LIBS)
 # - Always increase the revision value.
 # - Increase the current value whenever an interface has been added, removed or changed.
 # - Increase the age value only if the changes made to the ABI are backward compatible.
-libhealpix_cxx_la_LDFLAGS = -version-info 2:1:0
+libhealpix_cxx_la_LDFLAGS = -no-undefined -version-info 2:1:0
 
 bin_PROGRAMS = alm2map_cxx anafast_cxx calc_powspec hotspots_cxx map2tga \
   median_filter_cxx mult_alm rotalm_cxx smoothing_cxx syn_alm_cxx udgrade_cxx \
