--- origsrc/libcdata-20160108/Makefile.am	2016-01-09 02:56:49.000000000 +0900
+++ src/libcdata-20160108/Makefile.am	2016-04-04 21:56:46.035827000 +0900
@@ -3,11 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libcthreads \
 	libcdata \
-	po \
 	manuals \
 	tests \
 	msvscpp
@@ -81,4 +77,3 @@ splint:
 	(cd $(srcdir)/libcdata && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
-
--- origsrc/libcdata-20160108/configure.ac	2016-01-09 02:56:49.000000000 +0900
+++ src/libcdata-20160108/configure.ac	2016-04-04 21:56:46.052842500 +0900
@@ -128,9 +128,6 @@ dnl Have configure make the Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
 AC_CONFIG_FILES([libcdata/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
@@ -166,4 +163,3 @@ Features:
    Multi-threading support:     $ac_cv_libcthreads_multi_threading
    Wide character type support: $ac_cv_enable_wide_character_type
 ]);
-
--- origsrc/libcdata-20160108/libcdata/Makefile.am	2016-01-09 02:56:49.000000000 +0900
+++ src/libcdata-20160108/libcdata/Makefile.am	2016-04-04 21:56:46.052842500 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -51,4 +52,3 @@ distclean: clean
 splint:
 	@echo "Running splint on libcdata ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(libcdata_la_SOURCES)
-
--- origsrc/libcdata-20160108/tests/Makefile.am	2016-01-09 02:56:49.000000000 +0900
+++ src/libcdata-20160108/tests/Makefile.am	2016-04-04 21:56:46.052842500 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -104,4 +105,3 @@ MAINTAINERCLEANFILES = \
 
 distclean: clean
 	/bin/rm -f Makefile
-
