--- origsrc/flickcurl-1.26/docs/Makefile.am	2014-08-17 14:34:29.000000000 +0900
+++ src/flickcurl-1.26/docs/Makefile.am	2021-09-18 10:25:56.096326400 +0900
@@ -94,16 +94,16 @@ if MAINTAINER_MODE
 search-photos.c: $(srcdir)/../examples/search-photos.c
 	$(SED) -e '1,/^$$/d' $< > $@
 
-flickcurl.1: $(srcdir)/flickcurl.1.in $(top_builddir)/utils/mangen
+flickcurl.1: $(srcdir)/flickcurl.1.in $(top_builddir)/utils/mangen$(EXEEXT)
 	sed -ne '1,/SH COMMANDS/p' $(srcdir)/flickcurl.1.in > flickcurl.1.head && \
 	$(top_builddir)/utils/mangen manpage > flickcurl.1.body && \
 	sed -ne '/SH SEE ALSO/,$$p' $(srcdir)/flickcurl.1.in > flickcurl.1.tail && \
 	cat flickcurl.1.head flickcurl.1.body flickcurl.1.tail > flickcurl.1 && \
 	rm -f flickcurl.1.head flickcurl.1.body flickcurl.1.tail
 
-flickcurl-extras.xml: $(top_builddir)/utils/mangen
+flickcurl-extras.xml: $(top_builddir)/utils/mangen$(EXEEXT)
 	$(top_builddir)/utils/mangen extras > flickcurl-extras.xml
 
-$(top_builddir)/utils/mangen:
-	cd $(top_builddir)/utils && $(MAKE) mangen
+$(top_builddir)/utils/mangen$(EXEEXT):
+	cd $(top_builddir)/utils && $(MAKE) mangen$(EXEEXT)
 endif
--- origsrc/flickcurl-1.26/src/Makefile.am	2014-01-14 16:16:05.000000000 +0900
+++ src/flickcurl-1.26/src/Makefile.am	2021-09-18 09:43:38.595855000 +0900
@@ -97,7 +97,7 @@ AM_CFLAGS = -DMTWIST_CONFIG -I$(top_srcd
 libflickcurl_la_LIBADD = $(top_builddir)/libmtwist/libmtwist.la
 libflickcurl_la_DEPENDENCIES = $(top_builddir)/libmtwist/libmtwist.la
 
-libflickcurl_la_LDFLAGS = -version-info @LIBFLICKCURL_LIBTOOL_VERSION@
+libflickcurl_la_LDFLAGS = -no-undefined -version-info @LIBFLICKCURL_LIBTOOL_VERSION@
 
 $(top_builddir)/libmtwist/libmtwist.la:
 	cd $(top_builddir)/libmtwist && $(MAKE) libmtwist.la
