--- origsrc/flickcurl-1.22/src/Makefile.am	2011-11-12 15:21:04.000000000 +0900
+++ src/flickcurl-1.22/src/Makefile.am	2012-02-20 23:07:45.117708500 +0900
@@ -92,7 +92,7 @@ flickcurl_internal.h
 AM_CFLAGS=
 libflickcurl_la_LIBADD=
 
-libflickcurl_la_LDFLAGS = -version-info @LIBFLICKCURL_LIBTOOL_VERSION@
+libflickcurl_la_LDFLAGS = -version-info @LIBFLICKCURL_LIBTOOL_VERSION@ -no-undefined
 
 if OFFLINE
 if RAPTOR
