--- origsrc/flickcurl-1.23/src/Makefile.am	2012-08-29 10:48:37.000000000 +0900
+++ src/flickcurl-1.23/src/Makefile.am	2012-09-23 20:31:31.124859100 +0900
@@ -101,7 +101,7 @@ AM_CFLAGS = -DMTWIST_CONFIG -I$(top_srcd
 
 libflickcurl_la_LIBADD =
 
-libflickcurl_la_LDFLAGS = -version-info @LIBFLICKCURL_LIBTOOL_VERSION@
+libflickcurl_la_LDFLAGS = -version-info @LIBFLICKCURL_LIBTOOL_VERSION@ -no-undefined
 
 if OFFLINE
 if RAPTOR
