--- origsrc/flickcurl-1.19/configure.ac	2010-05-10 12:13:51.000000000 +0900
+++ src/flickcurl-1.19/configure.ac	2010-08-07 02:18:07.144844500 +0900
@@ -79,6 +79,7 @@ m4_undefine([AC_PROG_CXX])
 m4_defun([AC_PROG_CXX],[])
 m4_undefine([AC_PROG_F77])
 m4_defun([AC_PROG_F77],[])
+AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
 
 # Checks for libraries.
--- origsrc/flickcurl-1.19/src/Makefile.am	2010-07-25 01:54:48.000000000 +0900
+++ src/flickcurl-1.19/src/Makefile.am	2010-08-07 02:18:07.216485500 +0900
@@ -92,7 +92,7 @@ flickcurl_internal.h
 AM_CFLAGS=
 libflickcurl_la_LIBADD=
 
-libflickcurl_la_LDFLAGS = -version-info @LIBFLICKCURL_LIBTOOL_VERSION@
+libflickcurl_la_LDFLAGS = -no-undefined -version-info @LIBFLICKCURL_LIBTOOL_VERSION@
 
 if OFFLINE
 if RAPTOR
