--- origsrc/flickcurl-1.21/configure.ac	2011-03-27 02:48:47.000000000 +0900
+++ src/flickcurl-1.21/configure.ac	2011-05-26 00:32:55.486399200 +0900
@@ -79,7 +79,7 @@ m4_undefine([AC_PROG_CXX])
 m4_defun([AC_PROG_CXX],[])
 m4_undefine([AC_PROG_F77])
 m4_defun([AC_PROG_F77],[])
-AC_PROG_LIBTOOL
+LT_INIT
 
 # Checks for libraries.
 
--- origsrc/flickcurl-1.21/src/Makefile.am	2010-07-25 01:54:48.000000000 +0900
+++ src/flickcurl-1.21/src/Makefile.am	2011-05-26 00:33:13.593401400 +0900
@@ -92,7 +92,7 @@ flickcurl_internal.h
 AM_CFLAGS=
 libflickcurl_la_LIBADD=
 
-libflickcurl_la_LDFLAGS = -version-info @LIBFLICKCURL_LIBTOOL_VERSION@
+libflickcurl_la_LDFLAGS = -version-info @LIBFLICKCURL_LIBTOOL_VERSION@ -no-undefined
 
 if OFFLINE
 if RAPTOR
