--- origsrc/libtheo/configure.ac	2024-10-24 16:46:43.000000000 +0900
+++ src/libtheo/configure.ac	2024-10-24 16:46:44.682745700 +0900
@@ -14,7 +14,7 @@ AC_PROG_CPP
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
-AC_PROG_RANLIB
+LT_INIT
 AC_PATH_PROG(TAR, tar)
 
 # Checks for libraries.
--- origsrc/libtheo/lib/Makefile.am	2024-10-24 16:46:43.000000000 +0900
+++ src/libtheo/lib/Makefile.am	2024-10-24 16:46:44.682745700 +0900
@@ -1,3 +1,4 @@
 lib_LTLIBRARIES = libtheo.la
 libtheo_la_SOURCES = theo.c
-INCLUDES = -I$(top_srcdir)/include
+libtheo_la_CFLAGS = -I$(top_srcdir)/include
+libtheo_la_LDFLAGS = -no-undefined
