--- origsrc/libelperiodic-1.4/Makefile.am	2024-07-31 02:31:24.000000000 +0900
+++ src/libelperiodic-1.4/Makefile.am	2025-01-10 14:31:28.995914600 +0900
@@ -22,3 +22,6 @@ uninstall-local:
 	cat $(DESTDIR)$(pkgpythondir)/install_files.txt | xargs rm -rf
 	rm -rf $(DESTDIR)$(pkgpythondir)
 endif
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libelperiodic-1.4/configure.ac	2024-07-31 02:31:24.000000000 +0900
+++ src/libelperiodic-1.4/configure.ac	2025-01-10 14:31:29.000938500 +0900
@@ -1,4 +1,4 @@
-AC_INIT([libelperiodic],[m4_esyscmd_s([git rev-parse --abbrev-ref HEAD])],[sobomax@sippysoft.com])
+AC_INIT([libelperiodic],[1.4],[sobomax@sippysoft.com])
 
 AM_INIT_AUTOMAKE
 AM_MAINTAINER_MODE([disable])
@@ -8,7 +8,7 @@ AC_PROG_CC
 
 LT_INIT
 
-mydef_version=`git describe --always --dirty || date "+%Y%m%d%H%M%S"`
+mydef_version="1.4"
 AC_DEFINE_UNQUOTED([VERSION_GIT], ["$mydef_version"])
 
 AC_ARG_ENABLE(coverage,
@@ -34,4 +34,5 @@ AM_CONDITIONAL(ENABLE_LTO, [test "x${ENA
 AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != :])
 
 AC_CONFIG_FILES([Makefile src/Makefile])
+AC_CONFIG_FILES([libelperiodic.pc])
 AC_OUTPUT
--- origsrc/libelperiodic-1.4/libelperiodic.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libelperiodic-1.4/libelperiodic.pc.in	2025-01-10 14:31:28.979810400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libelperiodic
+Description: Library to run frequent periodic tasks
+Version: @VERSION@
+Libs: -L${libdir} -lelperiodic
+Cflags: -I${includedir}
--- origsrc/libelperiodic-1.4/src/Makefile.am	2024-07-31 02:31:24.000000000 +0900
+++ src/libelperiodic-1.4/src/Makefile.am	2025-01-10 14:19:37.943310300 +0900
@@ -12,8 +12,8 @@ libelperiodic_la_SOURCES = periodic.c pr
   prdic_sign_ctx.h prdic_sign_impl.h prdic_band.h src/prdic_inst.h \
   src/prdic_time.h
 libelperiodic_la_DEPENDENCIES = Symbol.map
-libelperiodic_la_CFLAGS = -g3 -O0 -fPIC ${LTO_FLAG}
-libelperiodic_la_LDFLAGS = ${LTO_FLAG} -version-info 1 -Wl,--version-script=$(srcdir)/Symbol.map -lm @RT_LIB@
+libelperiodic_la_CFLAGS = -fPIC ${LTO_FLAG}
+libelperiodic_la_LDFLAGS = -no-undefined ${LTO_FLAG} -version-info 1 -Wl,--version-script=$(srcdir)/Symbol.map -lm @RT_LIB@
 
 noinst_PROGRAMS = testskew
 testskew_SOURCES = testskew.c
