diff --strip-trailing-cr -uprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.log -x config.status -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/stdsyslog-0.03/Makefile src/stdsyslog-0.03/Makefile
--- origsrc/stdsyslog-0.03/Makefile	2013-02-08 03:31:07.000000000 +0900
+++ src/stdsyslog-0.03/Makefile	2013-04-27 12:11:39.597778600 +0900
@@ -70,7 +70,7 @@ SHAREGRP?=	${BINGRP}
 SHAREMODE?=	644
 
 # development/debugging flags, you may safely ignore them
-BDECFLAGS=	-W -Wall -std=c99 -pedantic -Wbad-function-cast -Wcast-align \
+BDECFLAGS=	-W -Wall -std=gnu99 -pedantic -Wbad-function-cast -Wcast-align \
 		-Wcast-qual -Wchar-subscripts -Winline \
 		-Wmissing-prototypes -Wnested-externs -Wpointer-arith \
 		-Wredundant-decls -Wshadow -Wstrict-prototypes -Wwrite-strings
