--- origsrc/libkeccak-1.2/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libkeccak-1.2/Makefile.am	2019-02-23 16:43:45.312982400 +0900
@@ -0,0 +1,51 @@
+.NONPOSIX:
+
+# If possible, use CONFIGFILE=optimised.mk
+CONFIGFILE = $(top_srcdir)/config.mk
+
+include $(CONFIGFILE)
+
+# The version of the library.
+LIB_MAJOR = 1
+LIB_MINOR = 2
+LIB_VERSION = $(LIB_MAJOR).$(LIB_MINOR)
+
+lib_LTLIBRARIES = libkeccak.la
+libkeccak_la_SOURCES = \
+	digest.c \
+	libkeccak_behex_lower.c \
+	libkeccak_behex_upper.c \
+	libkeccak_degeneralise_spec.c \
+	libkeccak_generalised_sum_fd.c \
+	libkeccak_hmac_copy.c \
+	libkeccak_hmac_digest.c \
+	libkeccak_hmac_fast_digest.c \
+	libkeccak_hmac_fast_update.c \
+	libkeccak_hmac_set_key.c \
+	libkeccak_hmac_unmarshal.c \
+	libkeccak_hmac_update.c \
+	libkeccak_hmac_wipe.c \
+	libkeccak_state_copy.c \
+	libkeccak_state_initialise.c \
+	libkeccak_state_marshal.c \
+	libkeccak_state_unmarshal.c \
+	libkeccak_state_unmarshal_skip.c \
+	libkeccak_state_wipe.c \
+	libkeccak_state_wipe_message.c \
+	libkeccak_state_wipe_sponge.c \
+	libkeccak_unhex.c
+libkeccak_la_LDFLAGS = -no-undefined -version-info $(LIB_MAJOR)
+
+pkginclude_HEADERS = \
+	libkeccak.h \
+	libkeccak-legacy.h \
+	common.h
+
+noinst_PROGRAMS = test benchmark
+test_SOURCES = test.c
+test_LDADD = libkeccak.la
+benchmark_SOURCES = benchmark.c
+benchmark_LDADD = libkeccak.la
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libkeccak-1.2/config.mk	2019-02-12 02:56:25.000000000 +0900
+++ src/libkeccak-1.2/config.mk	2019-02-23 16:34:29.203520000 +0900
@@ -1,6 +1,6 @@
-PREFIX = /usr/local
+PREFIX = /usr
 MANPREFIX = $(PREFIX)/share/man
 
-CFLAGS   = -std=c99 -O3
-CPPFLAGS = -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_XOPEN_SOURCE=700
-LDFLAGS  = -s
+CFLAGS   += -std=c99 -O3
+CPPFLAGS += -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_XOPEN_SOURCE=700
+LDFLAGS  += -s
--- origsrc/libkeccak-1.2/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/libkeccak-1.2/configure.ac	2019-02-23 16:43:18.854137800 +0900
@@ -0,0 +1,37 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([libkeccak], [1.2])
+AC_CONFIG_SRCDIR([benchmark.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_LN_S
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_FUNC_ALLOCA
+AC_CHECK_HEADERS([fcntl.h inttypes.h limits.h stddef.h stdint.h stdlib.h string.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_C_INLINE
+AC_TYPE_INT32_T
+AC_TYPE_INT64_T
+AC_C_RESTRICT
+AC_TYPE_SIZE_T
+AC_TYPE_SSIZE_T
+AC_CHECK_MEMBERS([struct stat.st_blksize])
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([clock_gettime memset])
+
+AC_CONFIG_FILES([libkeccak.pc Makefile])
+AC_OUTPUT
--- origsrc/libkeccak-1.2/libkeccak.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libkeccak-1.2/libkeccak.pc.in	2019-02-23 16:34:29.203520000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libkeccak
+Description: Keccak-family hashing library
+Version: @VERSION@
+Libs: -L${libdir} -lkeccak
+Cflags: -I${includedir}
