--- origsrc/libcerror-20201121/Makefile.am	2020-11-22 03:02:13.000000000 +0900
+++ src/libcerror-20201121/Makefile.am	2021-04-13 20:13:03.370817100 +0900
@@ -4,7 +4,6 @@ SUBDIRS = \
 	include \
 	common \
 	libcerror \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libcerror-20201121/libcerror/Makefile.am	2020-11-22 03:02:13.000000000 +0900
+++ src/libcerror-20201121/libcerror/Makefile.am	2021-04-13 20:13:03.373805300 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_DLL_EXPORT@
--- origsrc/libcerror-20201121/tests/Makefile.am	2020-11-22 03:02:13.000000000 +0900
+++ src/libcerror-20201121/tests/Makefile.am	2021-04-13 20:13:03.377803800 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_DLL_IMPORT@
