--- origsrc/libcerror-20191112/Makefile.am	2019-11-13 05:02:44.000000000 +0900
+++ src/libcerror-20191112/Makefile.am	2019-12-22 10:06:36.121802400 +0900
@@ -4,7 +4,6 @@ SUBDIRS = \
 	include \
 	common \
 	libcerror \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libcerror-20191112/libcerror/Makefile.am	2019-11-13 05:02:44.000000000 +0900
+++ src/libcerror-20191112/libcerror/Makefile.am	2019-12-22 10:06:36.121802400 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_DLL_EXPORT@
--- origsrc/libcerror-20191112/tests/Makefile.am	2019-11-13 05:02:44.000000000 +0900
+++ src/libcerror-20191112/tests/Makefile.am	2019-12-22 10:06:36.121802400 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_DLL_IMPORT@
