--- origsrc/libcerror-20240101/Makefile.am	2024-01-01 13:30:55.000000000 +0900
+++ src/libcerror-20240101/Makefile.am	2024-02-05 13:27:38.869102200 +0900
@@ -4,7 +4,6 @@ SUBDIRS = \
 	include \
 	common \
 	libcerror \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libcerror-20240101/libcerror/Makefile.am	2024-01-01 13:30:55.000000000 +0900
+++ src/libcerror-20240101/libcerror/Makefile.am	2024-02-05 13:27:38.869102200 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_DLL_EXPORT@
--- origsrc/libcerror-20240101/tests/Makefile.am	2024-01-01 13:30:55.000000000 +0900
+++ src/libcerror-20240101/tests/Makefile.am	2024-02-05 13:27:38.869102200 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_DLL_IMPORT@
