--- origsrc/libcerror-20181117/Makefile.am	2018-11-17 23:20:31.000000000 +0900
+++ src/libcerror-20181117/Makefile.am	2019-05-15 00:20:55.842910100 +0900
@@ -4,7 +4,6 @@ SUBDIRS = \
 	include \
 	common \
 	libcerror \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libcerror-20181117/libcerror/Makefile.am	2018-11-17 23:20:31.000000000 +0900
+++ src/libcerror-20181117/libcerror/Makefile.am	2019-05-15 00:19:55.443559500 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_DLL_EXPORT@
--- origsrc/libcerror-20181117/tests/Makefile.am	2018-11-17 23:20:31.000000000 +0900
+++ src/libcerror-20181117/tests/Makefile.am	2019-05-15 00:22:22.145996100 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_DLL_IMPORT@
