--- origsrc/liblazy/liblazy/Makefile.am	2012-06-10 20:09:27.000000000 +0900
+++ src/liblazy/liblazy/Makefile.am	2012-06-10 20:09:30.288881600 +0900
@@ -8,7 +8,7 @@ include_HEADERS = liblazy.h
 
 liblazy_la_SOURCES = liblazy_hal.c liblazy_dbus.c liblazy.c liblazy_local.h
 liblazy_la_CFLAGS = $(DBUS_CFLAGS) -Wall
-liblazy_la_LDFLAGS = -version-info 1:0:0 $(DBUS_LIBS)
+liblazy_la_LDFLAGS = -version-info 1:0:0 $(DBUS_LIBS) -no-undefined
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = lazy.pc
