--- origsrc/libxc-4.0.3-c0ba6bfbb36a211a13760b92ff3d3db7a9f9d11d/src/Makefile.am	2017-11-20 19:08:32.000000000 +0900
+++ src/libxc-4.0.3-c0ba6bfbb36a211a13760b92ff3d3db7a9f9d11d/src/Makefile.am	2017-11-29 21:27:46.032000000 +0900
@@ -97,7 +97,7 @@ libxc_la_FUNC_SOURCES = \
 	hyb_mgga_xc_kcis.c mgga_c_b88.c mgga_x_gx.c mgga_x_pbe_gx.c
 
 libxc_la_SOURCES = references.c $(libxc_la_FUNC_SOURCES)
-libxc_la_LDFLAGS = -version-info $(XC_LT_VERSION)
+libxc_la_LDFLAGS = -no-undefined -version-info $(XC_LT_VERSION)
 
 if COMPILE_FORTRAN
   lib_LTLIBRARIES += libxcf90.la
@@ -106,11 +106,8 @@ if COMPILE_FORTRAN
   nodist_libxcf90_la_SOURCES = libxc_funcs.f90 libxc.f90
 
   # libtool stuff
-  libxcf90_la_LDFLAGS = -version-info $(XC_LT_VERSION)
+  libxcf90_la_LDFLAGS = -no-undefined -version-info $(XC_LT_VERSION)
   libxcf90_la_LIBADD = libxc.la
-
-  # this is a hack to go around buggy libtool/automake versions
-  libxcf90_la_LIBTOOLFLAGS = --tag=F77
 endif
 
 if COMPILE_FORTRAN03
@@ -119,19 +116,10 @@ if COMPILE_FORTRAN03
   nodist_libxcf03_la_SOURCES = libxcf03.f90
 
   # libtool stuff
-  libxcf03_la_LDFLAGS = -version-info $(XC_LT_VERSION)
+  libxcf03_la_LDFLAGS = -no-undefined -version-info $(XC_LT_VERSION)
   libxcf03_la_LIBADD = libxc.la
-
-  # this is a hack to go around buggy libtool/automake versions
-  libxcf03_la_LIBTOOLFLAGS = --tag=F77
 endif
 
-
-# this is a hack to go around buggy libtool/automake versions
-LTFCCOMPILE = $(LIBTOOL) --mode=compile --tag=F77 $(FC) $(AM_FCFLAGS) $(FCFLAGS)
-FCLINK = $(LIBTOOL) --mode=link --tag=F77 $(FCLD) $(AM_FCFLAGS) $(FCFLAGS) \
-         $(AM_LDFLAGS) $(LDFLAGS) -o $@
-
 noinst_HEADERS = \
 	string_f.h references.h util.h work_lda.c \
 	work_gga_x.c work_gga_c.c \
--- origsrc/libxc-4.0.3-c0ba6bfbb36a211a13760b92ff3d3db7a9f9d11d/src/util.h	2017-11-20 19:08:32.000000000 +0900
+++ src/libxc-4.0.3-c0ba6bfbb36a211a13760b92ff3d3db7a9f9d11d/src/util.h	2017-11-29 21:35:36.047000000 +0900
@@ -75,9 +75,14 @@ double erf(double);
 double erfc(double);
 #endif
 
-
+#if defined(M_SQRTPI)
+#undef M_SQRTPI
 #define M_SQRTPI        1.772453850905516027298167483341145182798L
+#endif
+#if defined(M_SQRT3)
+#undef M_SQRT3
 #define M_SQRT3         1.732050807568877293527446341505872366943L
+#endif
 #define M_CBRT2         1.259921049894873164767210607278228350570L
 #define M_CBRT3         1.442249570307408382321638310780109588392L
 #define M_CBRT4         1.587401051968199474751705639272308260391L
--- origsrc/libxc-4.0.3-c0ba6bfbb36a211a13760b92ff3d3db7a9f9d11d/testsuite/Makefile.am	2017-11-20 19:08:32.000000000 +0900
+++ src/libxc-4.0.3-c0ba6bfbb36a211a13760b92ff3d3db7a9f9d11d/testsuite/Makefile.am	2017-11-29 21:28:21.626000000 +0900
@@ -21,18 +21,18 @@
 noinst_PROGRAMS = xc-get_data xc-consistency xc-regression xc-error
 dist_noinst_SCRIPTS = xc-run_testsuite xc-reference.pl
 TESTS = xc-run_testsuite
-TESTS_ENVIRONMENT = buildir=$(top_builddir)/testsuite
+TESTS_ENVIRONMENT = buildir=$(top_builddir)/testsuite srcdir=$(srcdir)
 
 xc_get_data_SOURCES = xc-get_data.c
-xc_get_data_LDADD = -L../src/ -lxc -lm
+xc_get_data_LDADD = $(top_builddir)/src/libxc.la -lm
 xc_get_data_CPPFLAGS = -I$(srcdir)/../src/ -I$(top_builddir)/src
 
 xc_consistency_SOURCES = xc-consistency.c
-xc_consistency_LDADD = -L../src/ -lxc -lm
+xc_consistency_LDADD = $(top_builddir)/src/libxc.la -lm
 xc_consistency_CPPFLAGS = -I$(srcdir)/../src/ -I$(top_builddir)/src
 
 xc_regression_SOURCES = xc-regression.c
-xc_regression_LDADD = -L../src/ -lxc -lm
+xc_regression_LDADD = $(top_builddir)/src/libxc.la -lm
 xc_regression_CPPFLAGS = -I$(srcdir)/../src/ -I$(top_builddir)/src
 
 xc_error_SOURCES = xc-error.c
