--- origsrc/libxc-2.0.0/src/Makefile.am	2012-11-28 18:03:30.000000000 +0900
+++ src/libxc-2.0.0/src/Makefile.am	2012-12-23 14:52:58.565466100 +0900
@@ -63,13 +63,7 @@ endif
 endif
 
 # libtool stuff
-libxc_la_LDFLAGS = -version-info $(XC_LT_VERSION)
-# this is a hack to go around buggy libtool/automake versions
-libxc_la_LIBTOOLFLAGS = --tag=F77
-LTFCCOMPILE = $(LIBTOOL) --mode=compile --tag=F77 $(FC) $(AM_FCFLAGS) $(FCFLAGS)
-FCLINK = $(LIBTOOL) --mode=link --tag=F77 $(FCLD) $(AM_FCFLAGS) $(FCFLAGS) \
-        $(AM_LDFLAGS) $(LDFLAGS) -o $@
-
+libxc_la_LDFLAGS = -version-info $(XC_LT_VERSION) -no-undefined
 
 dist_noinst_SCRIPTS = get_funcs.pl
 
--- origsrc/libxc-2.0.0/testsuite/Makefile.am	2012-11-28 18:03:30.000000000 +0900
+++ src/libxc-2.0.0/testsuite/Makefile.am	2012-12-23 14:53:21.737341100 +0900
@@ -22,7 +22,7 @@
 noinst_PROGRAMS = xc-get_data xc-consistency xc-regression
 dist_noinst_SCRIPTS = xc-run_testsuite xc-reference.pl
 TESTS = xc-run_testsuite
-TESTS_ENVIRONMENT = buildir=$(top_builddir)/testsuite
+TESTS_ENVIRONMENT = buildir=$(top_builddir)/testsuite srcdir=$(srcdir)
 
 xc_get_data_SOURCES = xc-get_data.c
 xc_get_data_LDADD = -L../src/ -lxc -lm
