--- origsrc/libxc-2.0.3/src/Makefile.am	2014-02-18 00:14:49.000000000 +0900
+++ src/libxc-2.0.3/src/Makefile.am	2014-03-05 23:17:13.538032400 +0900
@@ -63,13 +63,7 @@ endif
 endif
 
 # libtool stuff
-libxc_la_LDFLAGS = -version-info $(XC_LT_VERSION)
-# this is a hack to go around buggy libtool/automake versions
-libxc_la_LIBTOOLFLAGS = --tag=F77
-LTFCCOMPILE = $(LIBTOOL) --mode=compile --tag=F77 $(FC) $(AM_FCFLAGS) $(FCFLAGS)
-FCLINK = $(LIBTOOL) --mode=link --tag=F77 $(FCLD) $(AM_FCFLAGS) $(FCFLAGS) \
-        $(AM_LDFLAGS) $(LDFLAGS) -o $@
-
+libxc_la_LDFLAGS = -version-info $(XC_LT_VERSION) -no-undefined
 
 dist_noinst_SCRIPTS = get_funcs.pl
 
--- origsrc/libxc-2.0.3/src/util.h	2014-02-18 00:14:48.000000000 +0900
+++ src/libxc-2.0.3/src/util.h	2014-03-05 23:22:10.331977700 +0900
@@ -33,8 +33,16 @@ float  asinhf(float  x);
 #endif
 
 #define M_CBRT2         1.259921049894873164767210607278228350570
+
+#if defined(M_SQRTPI)
+#undef M_SQRTPI
 #define M_SQRTPI        1.772453850905516027298167483341145182798
+#endif
+
+#if defined(M_SQRT3)
+#undef M_SQRT3
 #define M_SQRT3         1.732050807568877293527446341505872366943
+#endif
 
 /* Very useful macros */
 #define min(x,y)  ((x<y) ? (x) : (y))
--- origsrc/libxc-2.0.3/testsuite/Makefile.am	2014-02-18 00:14:49.000000000 +0900
+++ src/libxc-2.0.3/testsuite/Makefile.am	2014-03-05 23:17:13.538032400 +0900
@@ -22,7 +22,7 @@
 noinst_PROGRAMS = xc-get_data xc-consistency xc-regression
 dist_noinst_SCRIPTS = xc-run_testsuite xc-reference.pl
 TESTS = xc-run_testsuite
-TESTS_ENVIRONMENT = buildir=$(top_builddir)/testsuite
+TESTS_ENVIRONMENT = buildir=$(top_builddir)/testsuite srcdir=$(srcdir)
 
 xc_get_data_SOURCES = xc-get_data.c
 xc_get_data_LDADD = -L../src/ -lxc -lm
