--- origsrc/libparsifal-1.1.0/Makefile.am	2004-02-15 23:02:42.000000000 +0900
+++ src/libparsifal-1.1.0/Makefile.am	2020-02-07 15:17:48.841950100 +0900
@@ -18,3 +18,7 @@ EXTRA_DIST = conf \
 	APIChanges \
 	BUGS
 	
+bin_SCRIPTS = parsifal-config
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libparsifal-1.1.0/configure.ac	2008-10-12 13:59:04.000000000 +0900
+++ src/libparsifal-1.1.0/configure.ac	2020-02-07 15:43:23.963812400 +0900
@@ -69,7 +69,7 @@ iconv(cd,NULL,NULL,NULL,NULL);'
   fi
 fi
 
-AC_OUTPUT( Makefile \
+AC_OUTPUT( libparsifal.pc Makefile \
 	src/Makefile \
 	include/Makefile \
 	include/libparsifal/Makefile \
--- origsrc/libparsifal-1.1.0/libparsifal.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libparsifal-1.1.0/libparsifal.pc.in	2020-02-07 15:24:12.108809700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libparsifal
+Description: Validating XML 1.0 parser written in ANSI C
+Version: @VERSION@
+Libs: -L${libdir} -lparsifal -liconv
+Cflags: -I${includedir}
