diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*.whl' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x libcaer.h origsrc/libcaer-3.3.1/CMakeLists.txt src/libcaer-3.3.1/CMakeLists.txt
--- origsrc/libcaer-3.3.1/CMakeLists.txt	2020-01-18 04:37:00.000000000 +0900
+++ src/libcaer-3.3.1/CMakeLists.txt	2020-01-24 18:38:23.472511500 +0900
@@ -52,7 +52,7 @@ INCLUDE(GNUInstallDirs)
 
 # Custom CMake base module
 SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmakemod)
-INSTALL(FILES cmakemod/caer-base.cmake DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/caer)
+INSTALL(FILES cmakemod/caer-base.cmake DESTINATION lib/cmake/libcaer)
 
 # Basic setup for cAER
 INCLUDE(caer-base)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*.whl' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x libcaer.h origsrc/libcaer-3.3.1/src/CMakeLists.txt src/libcaer-3.3.1/src/CMakeLists.txt
--- origsrc/libcaer-3.3.1/src/CMakeLists.txt	2020-01-18 04:37:00.000000000 +0900
+++ src/libcaer-3.3.1/src/CMakeLists.txt	2020-01-24 18:38:23.488136400 +0900
@@ -62,7 +62,7 @@ SET_TARGET_PROPERTIES(caer
 	VERSION ${PROJECT_VERSION}
 )
 
-IF (OS_WINDOWS)
+IF (OS_WINDOWS OR CYGWIN)
 	INSTALL(TARGETS caer EXPORT libcaer-exports RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 ELSE()
 	INSTALL(TARGETS caer EXPORT libcaer-exports DESTINATION ${CMAKE_INSTALL_LIBDIR})
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*.whl' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x libcaer.h origsrc/libcaer-3.3.1/src/edvs.c src/libcaer-3.3.1/src/edvs.c
--- origsrc/libcaer-3.3.1/src/edvs.c	2020-01-18 04:37:00.000000000 +0900
+++ src/libcaer-3.3.1/src/edvs.c	2020-01-24 18:38:23.488136400 +0900
@@ -244,7 +244,7 @@ caerDeviceHandle edvsOpen(uint16_t devic
 		return (NULL);
 	}
 
-#if defined(OS_UNIX)
+#if defined(OS_UNIX) && !defined(__CYGWIN__)
 	// Set exclusive access to serial port. Only needed on Unix (TIOCEXCL flag).
 	int serialFd = -1;
 
