--- origsrc/libcaer-3.3.12/CMakeLists.txt	2021-11-13 00:00:57.000000000 +0900
+++ src/libcaer-3.3.12/CMakeLists.txt	2022-01-09 20:06:19.105994400 +0900
@@ -9,7 +9,11 @@ SET(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR
 INCLUDE(inivation-setup)
 
 # Legacy cmake support.
-INSTALL(FILES cmakemod/caer-base.cmake DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/caer)
+if (CYGWIN)
+	INSTALL(FILES cmakemod/caer-base.cmake DESTINATION lib/cmake/${PROJECT_NAME})
+else()
+	INSTALL(FILES cmakemod/caer-base.cmake DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/caer)
+endif()
 
 # Override default settings
 SET(CMAKE_C_VISIBILITY_PRESET default)
--- origsrc/libcaer-3.3.12/src/CMakeLists.txt	2021-11-13 00:00:57.000000000 +0900
+++ src/libcaer-3.3.12/src/CMakeLists.txt	2022-01-09 19:59:24.990899800 +0900
@@ -59,7 +59,7 @@ TARGET_INCLUDE_DIRECTORIES(caer INTERFAC
 
 SET_TARGET_PROPERTIES(caer PROPERTIES SOVERSION ${PROJECT_VERSION_MAJOR} VERSION ${PROJECT_VERSION})
 
-IF(OS_WINDOWS)
+IF(OS_WINDOWS OR CYGWIN)
 	INSTALL(
 		TARGETS caer
 		EXPORT libcaer-exports
--- origsrc/libcaer-3.3.12/src/edvs.c	2021-11-13 00:00:57.000000000 +0900
+++ src/libcaer-3.3.12/src/edvs.c	2022-01-09 19:58:32.373759600 +0900
@@ -244,7 +244,7 @@ caerDeviceHandle edvsOpen(uint16_t devic
 		return (NULL);
 	}
 
-#if defined(OS_UNIX)
+#if defined(OS_UNIX) && defined(TIOCEXCL)
 	// Set exclusive access to serial port. Only needed on Unix (TIOCEXCL flag).
 	int serialFd = -1;
 
