--- origsrc/libcaer-3.3.2/CMakeLists.txt	2020-01-24 21:41:43.000000000 +0900
+++ src/libcaer-3.3.2/CMakeLists.txt	2020-01-28 14:50:01.911134700 +0900
@@ -52,7 +52,7 @@ INCLUDE(GNUInstallDirs)
 
 # Custom CMake base module
 SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmakemod)
-INSTALL(FILES cmakemod/caer-base.cmake DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/caer)
+INSTALL(FILES cmakemod/caer-base.cmake DESTINATION lib/cmake/libcaer)
 
 # Basic setup for cAER
 INCLUDE(caer-base)
--- origsrc/libcaer-3.3.2/src/CMakeLists.txt	2020-01-24 21:41:43.000000000 +0900
+++ src/libcaer-3.3.2/src/CMakeLists.txt	2020-01-28 14:50:01.911134700 +0900
@@ -62,7 +62,7 @@ SET_TARGET_PROPERTIES(caer
 	VERSION ${PROJECT_VERSION}
 )
 
-IF (OS_WINDOWS)
+IF (OS_WINDOWS OR CYGWIN)
 	INSTALL(TARGETS caer EXPORT libcaer-exports RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 ELSE()
 	INSTALL(TARGETS caer EXPORT libcaer-exports DESTINATION ${CMAKE_INSTALL_LIBDIR})
--- origsrc/libcaer-3.3.2/src/edvs.c	2020-01-24 21:41:43.000000000 +0900
+++ src/libcaer-3.3.2/src/edvs.c	2020-01-28 14:50:01.926757100 +0900
@@ -244,7 +244,7 @@ caerDeviceHandle edvsOpen(uint16_t devic
 		return (NULL);
 	}
 
-#if defined(OS_UNIX)
+#if defined(OS_UNIX) && !defined(__CYGWIN__)
 	// Set exclusive access to serial port. Only needed on Unix (TIOCEXCL flag).
 	int serialFd = -1;
 
