--- origsrc/libcaer-3.3.13/CMakeLists.txt	2022-04-26 22:11:00.000000000 +0900
+++ src/libcaer-3.3.13/CMakeLists.txt	2022-05-16 17:52:58.675961200 +0900
@@ -10,7 +10,11 @@ SET(ENABLE_ALL_WARNINGS ON CACHE BOOL "T
 INCLUDE(inivation-setup)
 
 # Legacy cmake support.
-INSTALL(FILES cmakemod/caer-base.cmake DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/caer)
+if (CYGWIN)
+	INSTALL(FILES cmakemod/caer-base.cmake DESTINATION lib/cmake/${PROJECT_NAME})
+else()
+	INSTALL(FILES cmakemod/caer-base.cmake DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/caer)
+endif()
 
 # Override default settings
 SET(CMAKE_C_VISIBILITY_PRESET default)
--- origsrc/libcaer-3.3.13/src/CMakeLists.txt	2022-04-26 22:11:00.000000000 +0900
+++ src/libcaer-3.3.13/src/CMakeLists.txt	2022-05-16 17:52:58.683956200 +0900
@@ -59,7 +59,7 @@ TARGET_INCLUDE_DIRECTORIES(caer INTERFAC
 
 SET_TARGET_PROPERTIES(caer PROPERTIES SOVERSION ${PROJECT_VERSION_MAJOR} VERSION ${PROJECT_VERSION})
 
-IF(OS_WINDOWS)
+IF(OS_WINDOWS OR CYGWIN)
 	INSTALL(
 		TARGETS caer
 		EXPORT libcaer-exports
--- origsrc/libcaer-3.3.13/src/edvs.c	2022-04-26 22:11:00.000000000 +0900
+++ src/libcaer-3.3.13/src/edvs.c	2022-05-16 17:52:58.689933600 +0900
@@ -244,7 +244,7 @@ caerDeviceHandle edvsOpen(uint16_t devic
 		return (NULL);
 	}
 
-#if defined(OS_UNIX)
+#if defined(OS_UNIX) && defined(TIOCEXCL)
 	// Set exclusive access to serial port. Only needed on Unix (TIOCEXCL flag).
 	int serialFd = -1;
 
