--- origsrc/libcaer-3.2.0/CMakeLists.txt	2019-05-22 00:40:31.000000000 +0900
+++ src/libcaer-3.2.0/CMakeLists.txt	2019-06-04 17:36:27.693856400 +0900
@@ -52,7 +52,7 @@ INCLUDE(GNUInstallDirs)
 
 # Custom CMake base module
 SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmakemod)
-INSTALL(FILES cmakemod/caer-base.cmake DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/caer)
+INSTALL(FILES cmakemod/caer-base.cmake DESTINATION lib/cmake/libcaer)
 
 # Basic setup for cAER
 INCLUDE(caer-base)
--- origsrc/libcaer-3.2.0/src/CMakeLists.txt	2019-05-22 00:40:31.000000000 +0900
+++ src/libcaer-3.2.0/src/CMakeLists.txt	2019-06-04 17:28:44.625444300 +0900
@@ -60,7 +60,7 @@ SET_TARGET_PROPERTIES(caer
 	VERSION ${PROJECT_VERSION}
 )
 
-IF (OS_WINDOWS)
+IF (OS_WINDOWS OR CYGWIN)
 	INSTALL(TARGETS caer EXPORT libcaer-exports RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 ELSE()
 	INSTALL(TARGETS caer EXPORT libcaer-exports DESTINATION ${CMAKE_INSTALL_LIBDIR})
--- origsrc/libcaer-3.2.0/src/edvs.c	2019-05-22 00:40:31.000000000 +0900
+++ src/libcaer-3.2.0/src/edvs.c	2019-06-04 17:35:03.739363100 +0900
@@ -242,7 +242,7 @@ caerDeviceHandle edvsOpen(uint16_t devic
 		return (NULL);
 	}
 
-#if defined(OS_UNIX)
+#if defined(OS_UNIX) && !defined(__CYGWIN__)
 	// Set exclusive access to serial port. Only needed on Unix (TIOCEXCL flag).
 	int serialFd = -1;
 
