--- origsrc/last-1652/makefile	2026-07-08 15:54:03.000000000 +0900
+++ src/last-1652/makefile	2026-07-28 17:01:39.173612900 +0900
@@ -1,4 +1,4 @@
-CXXFLAGS = -msse4 -O3 -std=c++11 -pthread
+CXXFLAGS ?= -msse4 -O3 -std=c++11 -pthread
 all:
 	@cd src && $(MAKE) CXXFLAGS="$(CXXFLAGS)"
 
--- origsrc/last-1652/src/makefile	2026-07-08 15:54:03.000000000 +0900
+++ src/last-1652/src/makefile	2026-07-28 17:01:39.173612900 +0900
@@ -1,4 +1,4 @@
-CXXFLAGS = -msse4 -O3 -Wall -g -std=c++11 -pthread
+CXXFLAGS ?= -msse4 -O3 -Wall -g -std=c++11 -pthread
 # -fomit-frame-pointer ?
 
 # The number of symbol types that we can keep in sequences (e.g. 20
@@ -7,7 +7,7 @@ CXXFLAGS = -msse4 -O3 -Wall -g -std=c++1
 ALPHABET_CAPACITY = 66
 CPPF = -DALPHABET_CAPACITY=$(ALPHABET_CAPACITY) -DHAS_CXX_THREADS $(CPPFLAGS)
 
-CFLAGS = -Wall -O2
+CFLAGS ?= -Wall -O2
 
 alpObj = alp/sls_alignment_evaluer.o alp/sls_pvalues.o		\
 alp/sls_alp_sim.o alp/sls_alp_regression.o alp/sls_alp_data.o	\
