--- origsrc/last-1654/makefile	2026-08-03 00:04:08.000000000 +0000
+++ src/last-1654/makefile	2026-08-09 08:17:41.386666000 +0000
@@ -1,4 +1,4 @@
-CXXFLAGS = -msse4 -O3 -std=c++11 -pthread
+CXXFLAGS ?= -msse4 -O3 -std=c++11 -pthread
 all:
 	@cd src && $(MAKE) CXXFLAGS="$(CXXFLAGS)"
 
--- origsrc/last-1654/src/makefile	2026-08-03 00:04:08.000000000 +0000
+++ src/last-1654/src/makefile	2026-08-09 08:17:41.386666000 +0000
@@ -1,4 +1,4 @@
-CXXFLAGS = -msse4 -O3 -Wall -g -std=c++11 -pthread
+CXXFLAGS ?= -msse4 -O3 -Wall -g -std=c++11 -pthread
 # -fomit-frame-pointer ?
 
 # The number of symbol types that we can keep in sequences (e.g. 20
@@ -7,7 +7,7 @@ CXXFLAGS = -msse4 -O3 -Wall -g -std=c++1
 ALPHABET_CAPACITY = 66
 CPPF = -DALPHABET_CAPACITY=$(ALPHABET_CAPACITY) -DHAS_CXX_THREADS $(CPPFLAGS)
 
-CFLAGS = -Wall -O2
+CFLAGS ?= -Wall -O2
 
 alpObj = alp/sls_alignment_evaluer.o alp/sls_pvalues.o		\
 alp/sls_alp_sim.o alp/sls_alp_regression.o alp/sls_alp_data.o	\
