--- origsrc/argtable3-3.2.2.f25c624/CMakeLists.txt	2022-10-23 11:20:16.000000000 +0900
+++ src/argtable3-3.2.2.f25c624/CMakeLists.txt	2023-03-19 10:13:36.728698300 +0900
@@ -102,3 +102,10 @@ if(ARGTABLE3_ENABLE_TESTS)
   enable_testing()
   add_subdirectory(tests)
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/argtable3.pc.in ${CMAKE_BINARY_DIR}/argtable3.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/argtable3.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/argtable3-3.2.2.f25c624/argtable3.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/argtable3-3.2.2.f25c624/argtable3.pc.in	2023-03-19 10:13:05.948895200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: argtable3
+Description: ANSI C, command-line parsing library that parses GNU-style command-line options
+Version: @ARGTABLE3_VERSION@
+Libs: -L${libdir} -largtable3
+Cflags: -I${includedir}
