--- origsrc/apache-log4cxx-0.10.0/Makefile.am	2008-04-01 07:34:52.000000000 +0900
+++ src/apache-log4cxx-0.10.0/Makefile.am	2020-07-25 17:57:33.505931000 +0900
@@ -14,6 +14,8 @@
 # limitations under the License.
 #
 
+ACLOCAL_AMFLAGS = -I .
+
 AUTOMAKE_OPTIONS = foreign
 
 SUBDIRS = src
--- origsrc/apache-log4cxx-0.10.0/src/examples/cpp/console.cpp	2008-04-01 07:34:52.000000000 +0900
+++ src/apache-log4cxx-0.10.0/src/examples/cpp/console.cpp	2020-07-25 23:05:43.458071000 +0900
@@ -21,6 +21,7 @@
 #include <log4cxx/simplelayout.h>
 #include <log4cxx/logmanager.h>
 #include <iostream>
+#include <cstring>
 #include <locale.h>
 
 using namespace log4cxx;
--- origsrc/apache-log4cxx-0.10.0/src/main/cpp/Makefile.am	2008-04-01 07:34:09.000000000 +0900
+++ src/apache-log4cxx-0.10.0/src/main/cpp/Makefile.am	2020-07-25 17:52:23.389325000 +0900
@@ -169,5 +169,5 @@ liblog4cxx_la_SOURCES = \
         zipcompressaction.cpp
 
 AM_CPPFLAGS = @CPPFLAGS_ODBC@
-liblog4cxx_la_LDFLAGS = -version-info @LT_VERSION@ @LIBS_ODBC@ -@APR_LIBS@
+liblog4cxx_la_LDFLAGS = -no-undefined -version-info @LT_VERSION@ @LIBS_ODBC@ -@APR_LIBS@
 
--- origsrc/apache-log4cxx-0.10.0/src/main/cpp/inputstreamreader.cpp	2008-04-01 07:34:09.000000000 +0900
+++ src/apache-log4cxx-0.10.0/src/main/cpp/inputstreamreader.cpp	2020-07-25 19:56:09.141531500 +0900
@@ -20,6 +20,7 @@
 #include <log4cxx/helpers/exception.h>
 #include <log4cxx/helpers/pool.h>
 #include <log4cxx/helpers/bytebuffer.h>
+#include <string.h>
 
 using namespace log4cxx;
 using namespace log4cxx::helpers;
--- origsrc/apache-log4cxx-0.10.0/src/main/cpp/socketoutputstream.cpp	2008-04-01 07:34:09.000000000 +0900
+++ src/apache-log4cxx-0.10.0/src/main/cpp/socketoutputstream.cpp	2020-07-25 22:22:52.503424500 +0900
@@ -19,6 +19,7 @@
 #include <log4cxx/helpers/socketoutputstream.h>
 #include <log4cxx/helpers/socket.h>
 #include <log4cxx/helpers/bytebuffer.h>
+#include <cstring>
 
 using namespace log4cxx;
 using namespace log4cxx::helpers;
