--- origsrc/snzip-1.0.4/Makefile.am	2016-10-02 18:50:57.000000000 +0900
+++ src/snzip-1.0.4/Makefile.am	2017-07-16 02:08:30.512000000 +0900
@@ -31,4 +31,4 @@ dist-hook:
 # Use CFLAGS_SSE4_2 only for crc32_sse4_2.c.
 # Otherwise, SSE4.2 instructions may be used elsewhere.
 crc32_sse4_2.o: crc32_sse4_2.c crc32.h
-	$(COMPILE) $(CFLAGS_SSE4_2) -c crc32_sse4_2.c
+	$(COMPILE) $(CFLAGS_SSE4_2) -c $(srcdir)/crc32_sse4_2.c
