--- origsrc/libsvm-3.12/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libsvm-3.12/Makefile.am	2012-04-26 21:29:54.035963200 +0900
@@ -0,0 +1,15 @@
+AM_CFLAGS = -Wall -Wconversion -O3 -fopenmp
+
+lib_LTLIBRARIES = libsvm.la
+libsvm_la_SOURCES = svm.cpp
+libsvm_la_LDFLAGS = -no-undefined -version-info 2:0:0
+
+bin_PROGRAMS = svm-predict svm-train svm-scale
+svm_predict_SOURCES = svm-predict.c
+svm_predict_LDADD = libsvm.la
+svm_train_SOURCES = svm-train.c
+svm_train_LDADD = libsvm.la
+svm_scale_SOURCES = svm-scale.c
+svm_scale_LDADD = libsvm.la
+
+include_HEADERS = svm.h
--- origsrc/libsvm-3.12/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/libsvm-3.12/configure.ac	2012-04-26 21:30:54.070142900 +0900
@@ -0,0 +1,30 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([libsvm], [3.12])
+AC_CONFIG_SRCDIR([svm-predict.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+AC_PROG_MAKE_SET
+LT_INIT
+
+# Checks for header files.
+AC_CHECK_HEADERS([float.h limits.h stdlib.h string.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_HEADER_STDBOOL
+AC_C_INLINE
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_FUNC_STRTOD
+AC_CHECK_FUNCS([memset sqrt strrchr strtol])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
