--- origsrc/unshield/src/CMakeLists.txt	2025-09-03 09:33:32.000000000 +0900
+++ src/unshield/src/CMakeLists.txt	2025-09-03 09:33:47.514309700 +0900
@@ -5,7 +5,7 @@ endif()
 
 add_executable(unshield unshield.c)
 target_link_libraries(unshield libunshield)
-if(WIN32)
+if(WIN32 OR CYGWIN)
     if(HAVE_ICONV)
         target_link_libraries(unshield iconv)
     endif()
@@ -14,7 +14,7 @@ endif()
 
 add_executable(unshield-deobfuscate unshield-deobfuscate.c)
 target_link_libraries(unshield-deobfuscate libunshield)
-if(WIN32)
+if(WIN32 OR CYGWIN)
     target_link_libraries(unshield-deobfuscate ${ZLIB_LIBRARIES})
 endif()
 
