--- origsrc/redis-plus-plus-1.3.15/CMakeLists.txt	2025-07-22 23:58:50.000000000 +0900
+++ src/redis-plus-plus-1.3.15/CMakeLists.txt	2026-01-31 21:51:42.526982900 +0900
@@ -298,7 +298,7 @@ install(TARGETS ${REDIS_PLUS_PLUS_TARGET
         RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
         INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
-set(REDIS_PLUS_PLUS_CMAKE_DESTINATION share/cmake/redis++)
+option(REDIS_PLUS_PLUS_CMAKE_DESTINATION "Destination to *.cmake" share/cmake/redis++)
 
 install(EXPORT redis++-targets
         FILE redis++-targets.cmake
