--- origsrc/redis-plus-plus-1.3.13/CMakeLists.txt	2024-10-09 22:55:29.000000000 +0900
+++ src/redis-plus-plus-1.3.13/CMakeLists.txt	2025-03-29 15:12:12.241716600 +0900
@@ -326,7 +326,7 @@ install(TARGETS ${REDIS_PLUS_PLUS_TARGET
         RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
         INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
-set(REDIS_PLUS_PLUS_CMAKE_DESTINATION share/cmake/redis++)
+option(REDIS_PLUS_PLUS_CMAKE_DESTINATION "Destination to *.cmake" share/cmake/redis++)
 
 install(EXPORT redis++-targets
         FILE redis++-targets.cmake
