--- origsrc/unshieldv3-0.2.1/CMakeLists.txt	2022-05-09 06:18:27.000000000 +0900
+++ src/unshieldv3-0.2.1/CMakeLists.txt	2025-03-16 16:02:35.723750500 +0900
@@ -1,6 +1,6 @@
 cmake_minimum_required (VERSION 3.5)
 
-project (unshieldv3 LANGUAGES CXX C VERSION "0.2.0")
+project (unshieldv3 LANGUAGES CXX C VERSION "0.2.1")
 
 if (NOT ${CMAKE_C_BYTE_ORDER} STREQUAL "LITTLE_ENDIAN")
 	message (FATAL_ERROR "Incompatible target. Only LITTLE_ENDIAN byte order is supported.")
