--- origsrc/unshieldv3-0.2.2/CMakeLists.txt	2025-04-25 08:57:34.000000000 +0900
+++ src/unshieldv3-0.2.2/CMakeLists.txt	2025-09-03 10:35:03.688800600 +0900
@@ -1,6 +1,6 @@
 cmake_minimum_required (VERSION 3.5)
 
-project (unshieldv3 LANGUAGES CXX C VERSION "0.2.0")
+project (unshieldv3 LANGUAGES CXX C VERSION "0.2.2")
 
 if (NOT ${CMAKE_C_BYTE_ORDER} STREQUAL "LITTLE_ENDIAN")
 	message (FATAL_ERROR "Incompatible target. Only LITTLE_ENDIAN byte order is supported.")
