--- origsrc/libklvanc/Makefile.am	2020-02-20 17:11:02.000000000 +0900
+++ src/libklvanc/Makefile.am	2020-02-20 17:24:25.806413400 +0900
@@ -7,3 +7,6 @@ docs:
 
 test:
 	cd tools && make test
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libklvanc/configure.ac	2020-02-20 17:11:02.000000000 +0900
+++ src/libklvanc/configure.ac	2020-02-20 17:24:25.806413400 +0900
@@ -57,6 +57,6 @@ AM_COND_IF(DEBUG,
     AC_DEFINE(DEBUG, 1, [Define to 0 if this is a release build]),
     AC_DEFINE(DEBUG, 0, [Define to 1 or higher if this is a debug build]))
 
-AC_CONFIG_FILES([Makefile src/Makefile tools/Makefile])
+AC_CONFIG_FILES([libklvanc.pc Makefile src/Makefile tools/Makefile])
 AC_OUTPUT
 
--- origsrc/libklvanc/libklvanc.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libklvanc/libklvanc.pc.in	2020-02-20 17:24:25.697055200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libklvanc
+Description: VANC Processing Framework
+Version: @VERSION@
+Libs: -L${libdir} -lklvanc
+Cflags: -I${includedir}
--- origsrc/libklvanc/src/Makefile.am	2020-02-20 17:11:02.000000000 +0900
+++ src/libklvanc/src/Makefile.am	2020-02-20 17:21:40.548509200 +0900
@@ -24,6 +24,8 @@ libklvanc_la_SOURCES += klbitstream_read
 libklvanc_la_CFLAGS = -Wall -DVERSION=\"$(VERSION)\" -DPROG="\"$(PACKAGE)\"" \
 	-D_FILE_OFFSET_BITS=64 -O3 -D_BSD_SOURCE -I$(top_srcdir)/include
 
+libklvanc_la_LDFLAGS = -no-undefined
+
 if DEBUG
   libklvanc_la_CFLAGS += -g
 endif
