--- origsrc/libecpint-1.0.7/CMakeLists.txt	2021-12-05 23:40:21.000000000 +0900
+++ src/libecpint-1.0.7/CMakeLists.txt	2022-05-18 15:32:08.409817600 +0900
@@ -83,3 +83,12 @@ message(STATUS "Maximum angular momentum
 message(STATUS "Maximum angular momentum unrolling : ${LIBECPINT_MAX_UNROL}")
 message(STATUS "Build tests                        : ${LIBECPINT_BUILD_TESTS}")
 message("")
+
+if (CYGWIN)
+  set(prefix ${CMAKE_INSTALL_PREFIX})
+  set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+  set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+  set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+  configure_file(${CMAKE_SOURCE_DIR}/libecpint.pc.in ${CMAKE_BINARY_DIR}/libecpint.pc @ONLY)
+  install(FILES ${CMAKE_BINARY_DIR}/libecpint.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+endif()
--- origsrc/libecpint-1.0.7/libecpint.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libecpint-1.0.7/libecpint.pc.in	2022-05-18 15:32:27.273299100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libecpint
+Description: C++ library for the efficient evaluation of integrals over effective core potentials
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lecpint
+Cflags: -I${includedir}
--- origsrc/libecpint-1.0.7/src/CMakeLists.txt	2021-12-05 23:40:21.000000000 +0900
+++ src/libecpint-1.0.7/src/CMakeLists.txt	2022-05-18 15:42:17.338238600 +0900
@@ -86,7 +86,9 @@ endif()
 #  Install
 # =========
 
-install(TARGETS ecpint EXPORT ecpintTargets DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS ecpint EXPORT ecpintTargets
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 include(CMakePackageConfigHelpers)
 # Version file
 write_basic_package_version_file(
@@ -117,5 +119,3 @@ install(
     ${CMAKE_CURRENT_BINARY_DIR}/ecpint-config-version.cmake
   DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/ecpint
 )
-
-install(TARGETS ecpint DESTINATION ${CMAKE_INSTALL_LIBDIR})
--- origsrc/libecpint-1.0.7/src/lib/bessel.cpp	2021-12-05 23:40:21.000000000 +0900
+++ src/libecpint-1.0.7/src/lib/bessel.cpp	2022-05-18 15:03:54.981367100 +0900
@@ -28,6 +28,10 @@
 #include <cassert>
 #include <iostream>
 
+#if defined(_N)
+#undef _N
+#endif
+
 namespace libecpint {
 
 	// Constructor
