--- origsrc/libecpint-1.0.2/external/Faddeeva/CMakeLists.txt	2020-10-04 03:42:59.000000000 +0900
+++ src/libecpint-1.0.2/external/Faddeeva/CMakeLists.txt	2020-12-05 13:07:10.198473800 +0900
@@ -9,4 +9,5 @@ target_include_directories(Faddeeva
 if(NOT BUILD_SHARED_LIBS)
   set_property(TARGET Faddeeva PROPERTY POSITION_INDEPENDENT_CODE ON)
 endif()
-install(TARGETS Faddeeva EXPORT ecpintTargets DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS Faddeeva EXPORT ecpintTargets RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(FILES Faddeeva.hpp DESTINATION include)
--- origsrc/libecpint-1.0.2/src/CMakeLists.txt	2020-10-04 03:42:59.000000000 +0900
+++ src/libecpint-1.0.2/src/CMakeLists.txt	2020-12-05 13:03:21.677545200 +0900
@@ -61,7 +61,6 @@ target_link_libraries(ecpint
 #  Install
 # =========
 
-install(TARGETS ecpint EXPORT ecpintTargets DESTINATION ${CMAKE_INSTALL_LIBDIR})
 include(CMakePackageConfigHelpers)
 # Version file
 write_basic_package_version_file(
@@ -92,4 +91,4 @@ install(
   DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/ecpint
 )
 
-install(TARGETS ecpint DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS ecpint RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
--- origsrc/libecpint-1.0.2/src/lib/bessel.cpp	2020-10-04 03:42:59.000000000 +0900
+++ src/libecpint-1.0.2/src/lib/bessel.cpp	2020-12-05 12:34:51.234595100 +0900
@@ -28,6 +28,10 @@
 #include <cassert>
 #include <iostream>
 
+#if defined(_N)
+#undef _N
+#endif
+
 namespace libecpint {
 
 	// Constructor
