--- origsrc/libecpint-1.0.3/external/Faddeeva/CMakeLists.txt	2020-12-12 03:57:33.000000000 +0900
+++ src/libecpint-1.0.3/external/Faddeeva/CMakeLists.txt	2020-12-13 23:51:45.104674400 +0900
@@ -9,4 +9,5 @@ target_include_directories(Faddeeva
 if(NOT BUILD_SHARED_LIBS)
   set_property(TARGET Faddeeva PROPERTY POSITION_INDEPENDENT_CODE ON)
 endif()
-install(TARGETS Faddeeva EXPORT ecpintTargets DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS Faddeeva EXPORT ecpintTargets RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(FILES Faddeeva.hpp DESTINATION include)
--- origsrc/libecpint-1.0.3/src/CMakeLists.txt	2020-12-12 03:57:33.000000000 +0900
+++ src/libecpint-1.0.3/src/CMakeLists.txt	2020-12-13 23:51:45.121627800 +0900
@@ -62,7 +62,6 @@ target_link_libraries(ecpint
 #  Install
 # =========
 
-install(TARGETS ecpint EXPORT ecpintTargets DESTINATION ${CMAKE_INSTALL_LIBDIR})
 include(CMakePackageConfigHelpers)
 # Version file
 write_basic_package_version_file(
@@ -94,4 +93,4 @@ install(
   DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/ecpint
 )
 
-install(TARGETS ecpint DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS ecpint RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
--- origsrc/libecpint-1.0.3/src/lib/bessel.cpp	2020-12-12 03:57:33.000000000 +0900
+++ src/libecpint-1.0.3/src/lib/bessel.cpp	2020-12-13 23:51:45.129633200 +0900
@@ -28,6 +28,10 @@
 #include <cassert>
 #include <iostream>
 
+#if defined(_N)
+#undef _N
+#endif
+
 namespace libecpint {
 
 	// Constructor
