--- origsrc/libphorward-1.0.0/Makefile.am	2019-11-08 02:22:11.000000000 +0900
+++ src/libphorward-1.0.0/Makefile.am	2025-12-25 14:55:49.118242200 +0900
@@ -13,5 +13,7 @@ AUTOMAKE_OPTIONS = foreign
 SUBDIRS = src tools run examples
 
 phorwarddir = $(datadir)/phorward
-dist_phorward_DATA = LICENSE README.md CHANGELOG.md doc/phorward.html doc/phorward.css doc/phorward.png doc/prism.css doc/prism.js examples/*.c
+# dist_phorward_DATA = LICENSE README.md CHANGELOG.md doc/phorward.html doc/phorward.css doc/phorward.png doc/prism.css doc/prism.js examples/*.c
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = phorward.pc
--- origsrc/libphorward-1.0.0/configure.ac	2019-11-08 02:22:11.000000000 +0900
+++ src/libphorward-1.0.0/configure.ac	2025-12-25 14:55:49.126045900 +0900
@@ -57,7 +57,7 @@ AC_TYPE_SIZE_T
 #AC_CHECK_FUNCS([iswlower iswupper towlower towupper], [],
 #	[AC_DEFINE([UNICODE],[0],[Disabling UNICODE build])])
 
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([phorward.pc Makefile
                  src/Makefile
 				 tools/Makefile
 				 run/Makefile
--- origsrc/libphorward-1.0.0/phorward.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libphorward-1.0.0/phorward.pc.in	2025-12-25 14:55:49.130105300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: phorward
+Description: C library for parsing, lexing, regular expressions and more
+Version: @VERSION@
+Libs: -L${libdir} -lphorward
+Cflags: -I${includedir}
