--- origsrc/GeographicLib-2.3/cmake/Makefile.am	2023-07-25 21:37:40.000000000 +0900
+++ src/GeographicLib-2.3/cmake/Makefile.am	2023-08-01 11:30:22.348527500 +0900
@@ -3,7 +3,7 @@
 #
 # Copyright (C) 2011-2022, Charles Karney <karney@alum.mit.edu>
 
-cmakedir=$(datadir)/cmake/GeographicLib
+cmakedir=$(libdir)/cmake/GeographicLib
 
 install:
 	$(INSTALL) -d $(DESTDIR)$(cmakedir)
--- origsrc/GeographicLib-2.3/configure.ac	2023-07-25 21:37:40.000000000 +0900
+++ src/GeographicLib-2.3/configure.ac	2023-08-01 11:59:46.165553400 +0900
@@ -21,6 +21,9 @@ AC_SUBST(GEOGRAPHICLIB_VERSION_MAJOR)
 AC_SUBST(GEOGRAPHICLIB_VERSION_MINOR)
 AC_SUBST(GEOGRAPHICLIB_VERSION_PATCH)
 
+lib_postfix=
+AC_SUBST(lib_postfix)
+
 dnl
 dnl This directive is deprecated by someone, but I prefer to avoid
 dnl running autotools if not required explicitly. The reason is
--- origsrc/GeographicLib-2.3/include/GeographicLib/DST.hpp	2023-07-25 21:37:40.000000000 +0900
+++ src/GeographicLib-2.3/include/GeographicLib/DST.hpp	2023-08-01 11:42:53.153878800 +0900
@@ -63,7 +63,7 @@ namespace GeographicLib {
   class DST {
   private:
     typedef Math::real real;
-    int _N;
+    int _NP;
     typedef kissfft<real> fft_t;
     std::shared_ptr<fft_t> _fft;
     // Implement DST-III (centerp = false) or DST-IV (centerp = true)
@@ -90,7 +90,7 @@ namespace GeographicLib {
      *
      * @return the number of points to use.
      **********************************************************************/
-    int N() const { return _N; }
+    int N() const { return _NP; }
 
     /**
      * Determine first \e N terms in the Fourier series
--- origsrc/GeographicLib-2.3/src/DST.cpp	2023-07-25 21:37:40.000000000 +0900
+++ src/GeographicLib-2.3/src/DST.cpp	2023-08-01 11:44:43.670100100 +0900
@@ -14,6 +14,11 @@
 
 namespace GeographicLib {
 
+#if defined(_N)
+#undef _N
+#define _N _NP
+#endif
+
   using namespace std;
 
   DST::DST(int N)
--- origsrc/GeographicLib-2.3/src/Makefile.am	2023-07-25 21:37:40.000000000 +0900
+++ src/GeographicLib-2.3/src/Makefile.am	2023-08-01 11:29:55.762588700 +0900
@@ -7,7 +7,7 @@ AM_CPPFLAGS = -I$(top_builddir)/include
 
 lib_LTLIBRARIES = libGeographicLib.la
 
-libGeographicLib_la_LDFLAGS = \
+libGeographicLib_la_LDFLAGS = -no-undefined \
 		-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 libGeographicLib_la_SOURCES = Accumulator.cpp \
 	AlbersEqualArea.cpp \
