--- origsrc/GeographicLib-1.50/GeographicLib.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/GeographicLib-1.50/GeographicLib.pc.in	2019-10-22 13:18:54.368762900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: GeographicLib
+Description: Geographic library
+Version: @VERSION@
+Libs: -L${libdir} -lGeographic
+Cflags: -I${includedir}
--- origsrc/GeographicLib-1.50/Makefile.am	2019-09-24 21:55:55.000000000 +0900
+++ src/GeographicLib-1.50/Makefile.am	2019-10-22 13:18:54.368762900 +0900
@@ -17,6 +17,9 @@ EXTRA_DIST = AUTHORS 00README.txt LICENS
 # PKG_INSTALLDIR in configure.ac.
 pkgconfig_DATA = cmake/geographiclib.pc
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = GeographicLib.pc
+
 dist-hook:
 	rm -rf $(distdir)/doc/html $(distdir)/doc/manpages \
 		$(distdir)/doc/GeographicLib.dox
--- origsrc/GeographicLib-1.50/cmake/Makefile.am	2019-09-24 21:55:55.000000000 +0900
+++ src/GeographicLib-1.50/cmake/Makefile.am	2019-10-22 13:18:54.400026400 +0900
@@ -3,7 +3,7 @@
 #
 # Copyright (C) 2011, Charles Karney <charles@karney.com>
 
-cmakedir=$(datadir)/cmake/GeographicLib
+cmakedir=$(libdir)/cmake/GeographicLib
 
 install:
 	$(INSTALL) -d $(DESTDIR)$(cmakedir)
--- origsrc/GeographicLib-1.50/configure.ac	2019-09-24 21:55:55.000000000 +0900
+++ src/GeographicLib-1.50/configure.ac	2019-10-22 13:18:54.400026400 +0900
@@ -98,6 +98,7 @@ dnl
 dnl Add here new file to be generated
 dnl
 AC_CONFIG_FILES([
+GeographicLib.pc
 Makefile
 src/Makefile
 include/Makefile
--- origsrc/GeographicLib-1.50/src/Makefile.am	2019-09-24 21:55:55.000000000 +0900
+++ src/GeographicLib-1.50/src/Makefile.am	2019-10-22 13:18:54.446894000 +0900
@@ -7,7 +7,7 @@ AM_CPPFLAGS = -I$(top_builddir)/include
 
 lib_LTLIBRARIES = libGeographic.la
 
-libGeographic_la_LDFLAGS = \
+libGeographic_la_LDFLAGS = -no-undefined \
 		-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 libGeographic_la_SOURCES = Accumulator.cpp \
 		AlbersEqualArea.cpp \
