--- origsrc/cpuminer-2.4.5/Makefile.am	2016-06-10 17:25:21.000000000 +0900
+++ src/cpuminer-2.4.5/Makefile.am	2017-01-03 18:02:58.288022300 +0900
@@ -33,6 +33,6 @@ minerd_SOURCES += sha2-ppc.S scrypt-ppc.
 endif
 endif
 minerd_LDFLAGS	= $(PTHREAD_FLAGS)
-minerd_LDADD	= @LIBCURL@ @JANSSON_LIBS@ @PTHREAD_LIBS@ @WS2_LIBS@
-minerd_CPPFLAGS = @LIBCURL_CPPFLAGS@
+minerd_LDADD	= @CURL_LIBS@ @JANSSON_LIBS@ @PTHREAD_LIBS@ @WS2_LIBS@
+minerd_CPPFLAGS = @CURL_CFLAGS@
 
--- origsrc/cpuminer-2.4.5/configure.ac	2016-06-10 17:25:21.000000000 +0900
+++ src/cpuminer-2.4.5/configure.ac	2017-01-03 18:02:58.290922300 +0900
@@ -53,7 +53,6 @@ case $target in
     ;;
 esac
 
-PTHREAD_FLAGS="-pthread"
 WS2_LIBS=""
 
 case $target in
@@ -120,8 +119,7 @@ else
 	JANSSON_LIBS=-ljansson
 fi
 
-LIBCURL_CHECK_CONFIG(, 7.15.2, ,
-  [AC_MSG_ERROR([Missing required libcurl >= 7.15.2])])
+PKG_CHECK_MODULES([CURL], [libcurl >= 7.15.2])
 
 AC_SUBST(JANSSON_LIBS)
 AC_SUBST(PTHREAD_FLAGS)
