--- origsrc/cpuminer-2.4/Makefile.am	2014-05-20 18:47:04.000000000 +0900
+++ src/cpuminer-2.4/Makefile.am	2014-12-20 14:41:53.432620700 +0900
@@ -30,6 +30,6 @@ minerd_SOURCES += sha2-arm.S scrypt-arm.
 endif
 endif
 minerd_LDFLAGS	= $(PTHREAD_FLAGS)
-minerd_LDADD	= @LIBCURL@ @JANSSON_LIBS@ @PTHREAD_LIBS@ @WS2_LIBS@
-minerd_CPPFLAGS = @LIBCURL_CPPFLAGS@
+minerd_LDADD	= @CURL_LIBS@ @JANSSON_LIBS@ @PTHREAD_LIBS@ @WS2_LIBS@
+minerd_CPPFLAGS = @CURL_CFLAGS@
 
--- origsrc/cpuminer-2.4/configure.ac	2014-05-20 18:47:04.000000000 +0900
+++ src/cpuminer-2.4/configure.ac	2014-12-20 14:43:36.897480600 +0900
@@ -50,7 +50,6 @@ case $target in
     ;;
 esac
 
-PTHREAD_FLAGS="-pthread"
 WS2_LIBS=""
 
 case $target in
@@ -116,8 +115,7 @@ else
 	JANSSON_LIBS=-ljansson
 fi
 
-LIBCURL_CHECK_CONFIG(, 7.15.2, ,
-  [AC_MSG_ERROR([Missing required libcurl >= 7.15.2])])
+PKG_CHECK_MODULES([CURL], [libcurl >= 7.15.2])
 
 AC_SUBST(JANSSON_LIBS)
 AC_SUBST(PTHREAD_FLAGS)
