--- origsrc/cpuminer-2.3.2/Makefile.am	2013-07-10 22:49:50.000000000 +0900
+++ src/cpuminer-2.3.2/Makefile.am	2013-10-13 18:15:46.836293300 +0900
@@ -18,6 +18,6 @@ minerd_SOURCES	= elist.h miner.h compat.
 		  sha2.c sha2-arm.S sha2-x86.S sha2-x64.S \
 		  scrypt.c scrypt-arm.S scrypt-x86.S scrypt-x64.S
 minerd_LDFLAGS	= $(PTHREAD_FLAGS)
-minerd_LDADD	= @LIBCURL@ @JANSSON_LIBS@ @PTHREAD_LIBS@ @WS2_LIBS@
-minerd_CPPFLAGS = @LIBCURL_CPPFLAGS@
+minerd_LDADD	= @CURL_LIBS@ @JANSSON_LIBS@ @PTHREAD_LIBS@ @WS2_LIBS@
+minerd_CPPFLAGS = @CURL_CFLAGS@
 
--- origsrc/cpuminer-2.3.2/configure.ac	2013-07-10 22:49:50.000000000 +0900
+++ src/cpuminer-2.3.2/configure.ac	2013-10-13 18:17:12.461293300 +0900
@@ -110,8 +110,7 @@ else
 	JANSSON_LIBS=-ljansson
 fi
 
-LIBCURL_CHECK_CONFIG(, 7.15.2, ,
-  [AC_MSG_ERROR([Missing required libcurl >= 7.15.2])])
+PKG_CHECK_MODULES([CURL], [libcurl >= 7.15.2])
 
 AC_SUBST(JANSSON_LIBS)
 AC_SUBST(PTHREAD_FLAGS)
