--- origsrc/desmume-0.9.8/configure.ac	2012-04-26 02:45:41.000000000 +0900
+++ src/desmume-0.9.8/configure.ac	2012-12-21 19:07:13.844710200 +0900
@@ -10,6 +10,7 @@ case $host in
   *mingw*) desmume_arch=windows;;
   *darwin*) desmume_arch=linux;;
   *bsd*) desmume_arch=linux;;
+  *cygwin*) desmume_arch=linux;;
 esac
 AC_SUBST(desmume_arch)
 
@@ -167,7 +168,7 @@ if test "x$glade" = "xyes" ; then
     AC_SUBST(LIBGLADE_LIBS)
 
     dnl uninstalled glade ui dir
-    AC_DEFINE_UNQUOTED(GLADEUI_UNINSTALLED_DIR,"`pwd`/src/gtk-glade/glade/",[path to glade ui dir])
+    AC_DEFINE_UNQUOTED(GLADEUI_UNINSTALLED_DIR,"${srcdir}/src/gtk-glade/glade/",[path to glade ui dir])
     AC_SUBST(GLADEUI_UNINSTALLED_DIR)
 
     PKG_CHECK_MODULES(GTKGLEXT,
--- origsrc/desmume-0.9.8/src/utils/libfat/libfat_pc.h	2012-04-26 02:45:40.000000000 +0900
+++ src/desmume-0.9.8/src/utils/libfat/libfat_pc.h	2012-12-21 19:07:13.860335200 +0900
@@ -59,9 +59,11 @@ typedef int32_t ssize_t;
 
 #endif //_MSC_VER
 
+#if !defined(__CYGWIN__)
 struct _reent {
 	intptr_t _errno;
 };
+#endif
 
 #ifdef __APPLE__
 typedef __darwin_mode_t mode_t;
