--- origsrc/libbde-20160110/Makefile.am	2016-01-11 05:07:09.000000000 +0900
+++ src/libbde-20160110/Makefile.am	2016-04-08 18:28:54.272370800 +0900
@@ -3,31 +3,11 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
-	libfguid \
-	libfvalue \
-	libhmac \
-	libcaes \
 	libbde \
-	libcsystem \
 	bdetools \
 	pybde \
 	pybde-python2 \
 	pybde-python3 \
-	po \
 	manuals \
 	tests \
 	msvscpp
@@ -141,4 +121,3 @@ splint:
 	(cd $(srcdir)/pybde && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
-
--- origsrc/libbde-20160110/bdetools/Makefile.am	2016-01-11 05:07:09.000000000 +0900
+++ src/libbde-20160110/bdetools/Makefile.am	2016-04-08 18:29:13.129243300 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -97,4 +98,3 @@ splint:
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(bdeinfo_SOURCES)
 	@echo "Running splint on bdemount ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(bdemount_SOURCES)
-
--- origsrc/libbde-20160110/configure.ac	2016-01-11 05:07:09.000000000 +0900
+++ src/libbde-20160110/configure.ac	2016-04-08 18:28:29.768562500 +0900
@@ -212,26 +212,7 @@ dnl Have configure make the Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
-AC_CONFIG_FILES([libhmac/Makefile])
-AC_CONFIG_FILES([libcaes/Makefile])
 AC_CONFIG_FILES([libbde/Makefile])
-AC_CONFIG_FILES([libcsystem/Makefile])
 AC_CONFIG_FILES([bdetools/Makefile])
 AC_CONFIG_FILES([pybde/Makefile])
 AC_CONFIG_FILES([pybde-python2/Makefile])
@@ -295,4 +276,3 @@ Features:
    Verbose output:                           $ac_cv_enable_verbose_output
    Debug output:                             $ac_cv_enable_debug_output
 ]);
-
--- origsrc/libbde-20160110/libbde/Makefile.am	2016-01-11 05:07:09.000000000 +0900
+++ src/libbde-20160110/libbde/Makefile.am	2016-04-08 18:29:29.971977200 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -109,4 +110,3 @@ distclean: clean
 splint:
 	@echo "Running splint on libbde ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(libbde_la_SOURCES)
-
--- origsrc/libbde-20160110/tests/Makefile.am	2016-01-11 05:07:09.000000000 +0900
+++ src/libbde-20160110/tests/Makefile.am	2016-04-08 18:29:42.765888800 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -100,4 +101,3 @@ MAINTAINERCLEANFILES = \
 
 distclean: clean
 	/bin/rm -f Makefile
-
