--- origsrc/libbde-20240502/Makefile.am	2024-05-02 13:08:30.000000000 +0900
+++ src/libbde-20240502/Makefile.am	2025-05-19 08:33:32.075604600 +0900
@@ -3,26 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdatetime \
-	libfguid \
-	libfvalue \
-	libhmac \
-	libcaes \
 	libbde \
 	bdetools \
 	pybde \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libbde-20240502/configure.ac	2024-05-02 13:08:30.000000000 +0900
+++ src/libbde-20240502/configure.ac	2025-05-19 08:33:14.104714600 +0900
@@ -179,22 +179,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
-AC_CONFIG_FILES([libhmac/Makefile])
-AC_CONFIG_FILES([libcaes/Makefile])
 AC_CONFIG_FILES([libbde/Makefile])
 AC_CONFIG_FILES([pybde/Makefile])
 AC_CONFIG_FILES([bdetools/Makefile])
--- origsrc/libbde-20240502/pybde/Makefile.am	2024-05-02 13:08:30.000000000 +0900
+++ src/libbde-20240502/pybde/Makefile.am	2025-05-19 08:33:43.775504100 +0900
@@ -48,7 +48,7 @@ pybde_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pybde_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pybde_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pybde_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
