--- origsrc/tree-sitter-python-0.23.6/CMakeLists.txt	2024-12-23 08:06:13.000000000 +0900
+++ src/tree-sitter-python-0.23.6/CMakeLists.txt	2025-01-05 10:49:57.821399900 +0900
@@ -17,15 +17,17 @@ endif()
 
 find_program(TREE_SITTER_CLI tree-sitter DOC "Tree-sitter CLI")
 
+if (NOT TREE_SITTER_CLI STREQUAL "TREE_SITTER_CLI-NOTFOUND")
 add_custom_command(OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/src/parser.c"
                    DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/src/grammar.json"
                    COMMAND "${TREE_SITTER_CLI}" generate src/grammar.json
                             --abi=${TREE_SITTER_ABI_VERSION}
                    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
                    COMMENT "Generating parser.c")
+endif()
 
 add_library(tree-sitter-python src/parser.c)
-if(EXISTS src/scanner.c)
+if(EXISTS ${CMAKE_SOURCE_DIR}/src/scanner.c)
   target_sources(tree-sitter-python PRIVATE src/scanner.c)
 endif()
 target_include_directories(tree-sitter-python PRIVATE src)
@@ -41,15 +43,15 @@ set_target_properties(tree-sitter-python
                       SOVERSION "${TREE_SITTER_ABI_VERSION}.${PROJECT_VERSION_MAJOR}"
                       DEFINE_SYMBOL "")
 
+include(GNUInstallDirs)
+
 configure_file(bindings/c/tree-sitter-python.pc.in
                "${CMAKE_CURRENT_BINARY_DIR}/tree-sitter-python.pc" @ONLY)
 
-include(GNUInstallDirs)
-
 install(FILES bindings/c/tree-sitter-python.h
         DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/tree_sitter")
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/tree-sitter-python.pc"
-        DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/pkgconfig")
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 install(TARGETS tree-sitter-python
         LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 
