--- origsrc/libsmacker-1.0/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libsmacker-1.0/Makefile.am	2015-01-04 19:16:27.110102400 +0900
@@ -0,0 +1,14 @@
+
+AM_CFLAGS = -pipe -Wall -Wextra -pedantic -O2 -fomit-frame-pointer -frename-registers
+
+bin_PROGRAMS = smk2avi smk2bmp
+smk2avi_SOURCES = smk2avi.c
+smk2avi_LDADD = libsmacker.la
+smk2bmp_SOURCES = driver.c
+smk2bmp_LDADD = libsmacker.la
+
+include_HEADERS = smacker.h smk_bitstream.h smk_hufftree.h smk_malloc.h
+
+lib_LTLIBRARIES = libsmacker.la
+libsmacker_la_SOURCES = smacker.c smk_bitstream.c smk_hufftree.c
+libsmacker_la_LDFLAGS = -no-undefined
--- origsrc/libsmacker-1.0/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/libsmacker-1.0/configure.ac	2015-01-04 19:18:01.493891500 +0900
@@ -0,0 +1,27 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([libsmacker], [1.0])
+AC_CONFIG_SRCDIR([driver.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([memset])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
