--- origsrc/libcli-1.9.6/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libcli-1.9.6/Makefile.am	2012-07-26 18:28:21.241316800 +0900
@@ -0,0 +1,11 @@
+
+check_PROGRAMS = clitest
+clitest_LDADD = libcli.la
+clitest_SOURCES = clitest.c
+
+include_HEADERS = libcli.h
+
+lib_LTLIBRARIES = libcli.la
+libcli_la_CFLAGS = -Wall -std=c99 -pedantic -Wformat-security -Wno-format-zero-length -Wwrite-strings -Wformat -fdiagnostics-show-option -Wextra -Wsign-compare -Wcast-align -Wno-unused-parameter -O3
+libcli_la_LDFLAGS = -no-undefined -release $(VERSION)
+libcli_la_SOURCES = libcli.c
--- origsrc/libcli-1.9.6/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/libcli-1.9.6/configure.ac	2012-07-26 18:25:31.397566800 +0900
@@ -0,0 +1,33 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([libcli], [1.9.6])
+AC_CONFIG_SRCDIR([clitest.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_LN_S
+LT_INIT
+
+# Checks for libraries.
+AC_CHECK_LIB([crypt], [main])
+
+# Checks for header files.
+AC_CHECK_HEADERS([arpa/inet.h malloc.h memory.h netinet/in.h stdlib.h string.h strings.h sys/socket.h sys/time.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+AC_TYPE_SSIZE_T
+
+# Checks for library functions.
+AC_FUNC_FORK
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([inet_ntoa memmove memset regcomp select socket strcasecmp strchr strdup strerror strncasecmp strpbrk strspn strstr])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
