--- origsrc/gcli-2.2.0/Makefile.am	2024-02-06 02:00:56.000000000 +0900
+++ src/gcli-2.2.0/Makefile.am	2024-02-20 18:49:26.809558900 +0900
@@ -20,13 +20,13 @@ pgen_SOURCES = \
 	src/pgen/lexer.l
 
 lib_LTLIBRARIES = libgcli.la
-noinst_LTLIBRARIES = libsn.la libpdjson.la
+noinst_LTLIBRARIES = libsn.la
 
 # For testing puproses I'll reenable parallel builds. If it breaks again, uncomment.
 # .NOTPARALLEL: pgen$(EXEEXT) $(builddir)/src/pgen/parser.c $(builddir)/src/pgen/parser.h $(builddir)/src/pgen/lexer.c
 src/pgen/lexer.c: src/pgen/parser.h
 
-libgcli_la_DEPENDENCIES = pgen$(EXEEXT) libpdjson.la libsn.la
+libgcli_la_DEPENDENCIES = pgen$(EXEEXT) libsn.la
 
 $(BUILT_SOURCES): pgen$(EXEEXT)
 
@@ -40,7 +40,8 @@ SUFFIXES = .t
 bin_PROGRAMS = gcli$(EXEEXT)
 gcli_LDADD = libgcli.la
 
-libgcli_la_LIBADD = libpdjson.la libsn.la
+libgcli_la_LIBADD = $(PDJSON_LIBS) libsn.la
+libgcli_la_LDFLAGS = -no-undefined
 
 dist_man_MANS = \
 	docs/gcli-api.1 \
@@ -89,10 +90,6 @@ libsn_la_SOURCES = \
 	thirdparty/sn/sn.c \
 	thirdparty/sn/sn.h
 
-libpdjson_la_SOURCES = \
-	thirdparty/pdjson/pdjson.c \
-	thirdparty/pdjson/pdjson.h
-
 TEMPLATES = \
 	templates/github/api.t \
 	templates/github/checks.t \
@@ -246,43 +243,43 @@ do_test: $(check_PROGRAMS) tests/Kyuafil
 tests_json_escape_SOURCES = \
 	tests/json-escape.c
 tests_json_escape_LDADD = \
-	libgcli.la libpdjson.la libsn.la \
+	libgcli.la libsn.la \
 	$(LIBATFC_LDFLAGS)
 
 tests_github_parse_tests_SOURCES = \
 	tests/github-parse-tests.c
 tests_github_parse_tests_LDADD = \
-	libgcli.la libpdjson.la libsn.la \
+	libgcli.la libsn.la \
 	$(LIBATFC_LDFLAGS)
 
 tests_gitlab_parse_tests_SOURCES = \
 	tests/gitlab-parse-tests.c
 tests_gitlab_parse_tests_LDADD = \
-	libgcli.la libpdjson.la libsn.la \
+	libgcli.la libsn.la \
 	$(LIBATFC_LDFLAGS)
 
 tests_gitea_parse_tests_SOURCES = \
 	tests/gitea-parse-tests.c
 tests_gitea_parse_tests_LDADD = \
-	libgcli.la libpdjson.la libsn.la \
+	libgcli.la libsn.la \
 	$(LIBATFC_LDFLAGS)
 
 tests_bugzilla_parse_tests_SOURCES = \
 	tests/bugzilla-parse-tests.c
 tests_bugzilla_parse_tests_LDADD = \
-	libgcli.la libpdjson.la libsn.la \
+	libgcli.la libsn.la \
 	$(LIBATFC_LDFLAGS)
 
 tests_base64_tests_SOURCES = \
 	tests/base64-tests.c
 tests_base64_tests_LDADD = \
-	libgcli.la libpdjson.la libsn.la \
+	libgcli.la libsn.la \
 	$(LIBATFC_LDFLAGS)
 
 tests_url_encode_SOURCES = \
 	tests/url-encode.c
 tests_url_encode_LDADD = \
-	libgcli.la libpdjson.la libsn.la \
+	libgcli.la libsn.la \
 	$(LIBATFC_LDFLAGS)
 
 tests_pretty_print_test_SOURCES = \
--- origsrc/gcli-2.2.0/configure.ac	2024-02-06 02:00:56.000000000 +0900
+++ src/gcli-2.2.0/configure.ac	2024-02-20 18:49:53.330729100 +0900
@@ -35,6 +35,8 @@ dnl Inspired by the way dpkg solves the
 m4_ifndef([PKG_PROG_PKG_CONFIG], [m4_fatal([pkgconf macros missing. do you have pkgconf installed?])])
 PKG_PROG_PKG_CONFIG
 
+PKG_CHECK_MODULES([PDJSON], [pdjson])
+
 dnl Use ccache if found
 CCACHE=""
 AC_CHECK_PROG([CCACHE], [ccache], [ccache])
--- origsrc/gcli-2.2.0/libgcli.pc.in	2024-02-06 02:00:56.000000000 +0900
+++ src/gcli-2.2.0/libgcli.pc.in	2024-02-20 18:57:17.444622400 +0900
@@ -7,7 +7,7 @@ includedir=@includedir@
 
 Name: gcli
 Description: Library to interact with various Git forges
-Version: 2.0
+Version: @VERSION@
 URL: https://herrhotzenplotz.de/gcli
 Requires: libcurl >= 8.0
 Libs: -L${libdir} -lgcli
