--- origsrc/libexception/src/Makefile.am	2014-11-02 19:24:40.000000000 +0900
+++ src/libexception/src/Makefile.am	2014-11-02 19:24:41.767970500 +0900
@@ -7,6 +7,6 @@ lib_LTLIBRARIES = libexception.la
 
 libexception_la_SOURCES = exception.c tryenv.c
 libexception_la_LIBADD = @PTHREAD_LIBS@
-libexception_la_LDFLAGS = -version-info 0:0:0
+libexception_la_LDFLAGS = -no-undefined -version-info 0:0:0
 
 # vim: ts=4 expandtab
