--- origsrc/libexception/Makefile.am	2024-09-27 05:59:58.000000000 +0900
+++ src/libexception/Makefile.am	2024-09-27 06:02:37.394484600 +0900
@@ -1 +1,4 @@
 SUBDIRS = src test
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libexception/configure.ac	2024-09-27 05:59:58.000000000 +0900
+++ src/libexception/configure.ac	2024-09-27 06:02:37.394484600 +0900
@@ -69,6 +69,7 @@ AC_CONFIG_FILES(Makefile
                 Doxyfile
                 src/Makefile
                 test/Makefile)
+AC_CONFIG_FILES([libexception.pc])
 AC_OUTPUT
 
 AC_MSG_NOTICE([$SUMMARY])
--- origsrc/libexception/libexception.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libexception/libexception.pc.in	2024-09-27 06:02:37.379969200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libexception
+Description: Exception handling library for C
+Version: @VERSION@
+Libs: -L${libdir} -lexception
+Cflags: -I${includedir}
--- origsrc/libexception/src/Makefile.am	2024-09-27 05:59:58.000000000 +0900
+++ src/libexception/src/Makefile.am	2024-09-27 05:59:59.129033800 +0900
@@ -7,6 +7,6 @@ lib_LTLIBRARIES = libexception.la
 
 libexception_la_SOURCES = exception.c tryenv.c
 libexception_la_LIBADD = @PTHREAD_LIBS@
-libexception_la_LDFLAGS = -version-info 0:0:0
+libexception_la_LDFLAGS = -no-undefined -version-info 0:0:0
 
 # vim: ts=4 expandtab
