--- origsrc/libdigidoc-2.7.0/CMakeLists.txt	2010-09-30 05:52:41.000000000 +0900
+++ src/libdigidoc-2.7.0/CMakeLists.txt	2012-12-31 17:23:48.550820600 +0900
@@ -12,6 +12,9 @@ if( WIN32 )
 elseif( APPLE )
     set( DIGIDOC_CONF_NAME "digidoc.conf" )
     set( PKCS11_MODULE "/Library/OpenSC/lib/opensc-pkcs11.so" )
+elseif( CYGWIN )
+    set( DIGIDOC_CONF_NAME "digidoc.ini" )
+    set( PKCS11_MODULE "opensc-pkcs11.dll" )
 else()
     set( DIGIDOC_CONF_NAME "digidoc.conf" )
     set( PKCS11_MODULE "opensc-pkcs11.so" )
