--- origsrc/libmaus2-libmaus2_experimental_2_0_813/configure.ac	2022-12-11 02:30:10.000000000 +0900
+++ src/libmaus2-libmaus2_experimental_2_0_813/configure.ac	2025-01-03 08:27:39.264756600 +0900
@@ -90,12 +90,16 @@ AC_CHECK_SIZEOF([unsigned long])
 LIBMAUS2_SIZEOF_UNSIGNED_LONG="#define LIBMAUS2_SIZEOF_UNSIGNED_LONG $ac_cv_sizeof_unsigned_long"
 
 AC_LANG_PUSH([C++])
+if test "$host_os" != "cygwin"; then
 AC_CHECK_HEADER(winsock2.h, [LIBMAUS2_HAVE_WINSOCK2_H="#define LIBMAUS2_HAVE_WINSOCK2_H"], [LIBMAUS2_HAVE_WINSOCK2_H=])
 AC_CHECK_HEADER(ws2tcpip.h, [LIBMAUS2_HAVE_WS2TCPIP_H="#define LIBMAUS2_HAVE_WS2TCPIP_H"], [LIBMAUS2_HAVE_WS2TCPIP_H=])
+fi
 AC_CHECK_HEADER(sys/wait.h, [LIBMAUS2_HAVE_SYS_WAIT_H="#define LIBMAUS2_HAVE_SYS_WAIT_H"], [LIBMAUS2_HAVE_SYS_WAIT_H=])
 AC_CHECK_HEADER(sys/ioctl.h, [LIBMAUS2_HAVE_SYS_IOCTL_H="#define LIBMAUS2_HAVE_SYS_IOCTL_H"], [LIBMAUS2_HAVE_SYS_IOCTL_H=])
 AC_CHECK_HEADER(execinfo.h, [LIBMAUS2_HAVE_EXECINFO_H="#define LIBMAUS2_HAVE_EXECINFO_H"], [LIBMAUS2_HAVE_EXECINFO_H=])
+if test "$host_os" != "cygwin"; then
 AC_CHECK_HEADER(windows.h, [LIBMAUS2_HAVE_WINDOWS_H="#define LIBMAUS2_HAVE_WINDOWS_H"], [LIBMAUS2_HAVE_WINDOWS_H=])
+fi
 AC_CHECK_HEADER(unistd.h, [LIBMAUS2_HAVE_UNISTD_H="#define LIBMAUS2_HAVE_UNISTD_H"], [LIBMAUS2_HAVE_UNISTD_H=])
 AC_CHECK_HEADER(linux/sysctl.h, [LIBMAUS2_HAVE_LINUX_SYSCTL_H="#define LIBMAUS2_HAVE_LINUX_SYSCTL_H"], [LIBMAUS2_HAVE_LINUX_SYSCTL_H=])
 AC_CHECK_HEADER(features.h, [LIBMAUS2_HAVE_FEATURES_H="#define LIBMAUS2_HAVE_FEATURES_H"], [LIBMAUS2_HAVE_FEATURES_H=])
@@ -2870,7 +2874,7 @@ if test "$have_unsigned_int128" = "yes"
 fi
 
 case $host_os in
-  *mingw32*)
+  *cygwin*|*mingw32*)
     LIBMAUSLDFLAGS="${LIBMAUSLDFLAGS} -no-undefined" # -avoid-version
     ;;
   *)
--- origsrc/libmaus2-libmaus2_experimental_2_0_813/src/Makefile.am	2022-12-11 02:30:10.000000000 +0900
+++ src/libmaus2-libmaus2_experimental_2_0_813/src/Makefile.am	2025-01-03 13:05:01.272284700 +0900
@@ -241,7 +241,7 @@ libmaus2_la_SOURCES = libmaus2/StaticIni
 libmaus2_la_CPPFLAGS = ${AM_CPPFLAGS} @LIBMAUSCPPFLAGS@ @SNAPPYCPPFLAGS@ @ZLIBCPPFLAGS@ @NETTLECPPFLAGS@ @IGZIPCPPFLAGS@ @GMPCPPFLAGS@ @LIBMAUS2_LZMA_CPPFLAGS@ @LIBDEFLATECPPFLAGS@ @LIBMAUS2_LIBCURL_CPPFLAGS@ @LIBMAUS2_LIBSECRECY_CPPFLAGS@ @LIBMAUS2_PARASAIL_CPPFLAGS@ @IOLIBCPPFLAGS@
 libmaus2_la_CXXFLAGS = @LIBMAUSCXXFLAGS@ ${AM_CXXFLAGS} @WARNCXXFLAGS@
 libmaus2_la_LDFLAGS = @LIBMAUSLDFLAGS@ @SNAPPYLDFLAGS@ -version-info ${LIBRARY_VERSION} @LIBMAUSSTATIC@ @IGZIPLDFLAGS@ @NETTLELDFLAGS@ @GMPLDFLAGS@ @LIBDEFLATELDFLAGS@ @IOLIBLDFLAGS@
-libmaus2_la_LIBADD = @IOLIBLIBS@ @DLLIB@ @SNAPPYLIBS@ @ZLIBLIBS@ @IGZIPLIBS@ @NETTLELIBS@ @GMPLIBS@ @LIBMAUS2_LZMA_LIBS@ @LIBMAUS2_LIBSECRECY_LIBS@ @LIBDEFLATELIBS@ @LIBMAUS2_LIBCURL_LIBS@ @LIBMAUS2_PARASAIL_LIBS@ @LIBMAUS2_WS2_32@ @LIBMAUS2_KERNEL32_LIB@ @LIBMAUS2_PSAPI_LIB@ libmaus2_parallel.la libmaus2_autoarray.la libmaus2_exception.la libmaus2_stacktrace.la libmaus2_posix.la libmaus2_demangle.la libmaus2_arch.la
+libmaus2_la_LIBADD = @IOLIBLIBS@ @DLLIB@ @SNAPPYLIBS@ @ZLIBLIBS@ @IGZIPLIBS@ @NETTLELIBS@ @GMPLIBS@ @LIBMAUS2_LZMA_LIBS@ @LIBMAUS2_LIBSECRECY_LIBS@ @LIBDEFLATELIBS@ @LIBMAUS2_LIBCURL_LIBS@ @LIBMAUS2_PARASAIL_LIBS@ @LIBMAUS2_WS2_32@ @LIBMAUS2_KERNEL32_LIB@ @LIBMAUS2_PSAPI_LIB@ libmaus2_parallel.la libmaus2_autoarray.la libmaus2_exception.la libmaus2_stacktrace.la libmaus2_posix.la libmaus2_demangle.la libmaus2_arch.la libmaus2_simd_align_128.la libmaus2_simd_align_256.la
 
 noinst_HEADERS = libmaus2/lz/lz4.h
 
--- origsrc/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/bambam/BamAlignmentDecoderBase.hpp	2022-12-11 02:30:10.000000000 +0900
+++ src/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/bambam/BamAlignmentDecoderBase.hpp	2025-01-03 08:38:57.766441400 +0900
@@ -38,6 +38,8 @@
 #include <libmaus2/bambam/PileQualityVectorElement.hpp>
 #include <libmaus2/math/IntegerInterval.hpp>
 
+#include <alloca.h>
+
 namespace libmaus2
 {
 	namespace bambam
--- origsrc/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/bambam/MdStringComputationContext.hpp	2022-12-11 02:30:10.000000000 +0900
+++ src/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/bambam/MdStringComputationContext.hpp	2025-01-03 08:41:33.220961600 +0900
@@ -23,6 +23,8 @@
 #include <libmaus2/bambam/BamAuxFilterVector.hpp>
 #include <libmaus2/bambam/CigarOperation.hpp>
 
+#include <alloca.h>
+
 namespace libmaus2
 {
 	namespace bambam
--- origsrc/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/math/IntegerInterval.hpp	2022-12-11 02:30:10.000000000 +0900
+++ src/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/math/IntegerInterval.hpp	2025-01-03 09:16:14.219900500 +0900
@@ -30,6 +30,10 @@
 #include <cstring>
 #include <libmaus2/exception/LibMausException.hpp>
 
+#if defined(_N)
+#undef _N
+#endif
+
 namespace libmaus2
 {
 	namespace math
--- origsrc/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/math/Rational.hpp	2022-12-11 02:30:10.000000000 +0900
+++ src/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/math/Rational.hpp	2025-01-03 21:08:03.725654900 +0900
@@ -24,6 +24,10 @@
 #include <cassert>
 #include <cmath>
 
+#if defined(_N)
+#undef _N
+#endif
+
 namespace libmaus2
 {
 	namespace math
--- origsrc/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/serialize/Serialize.hpp	2022-12-11 02:30:10.000000000 +0900
+++ src/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/serialize/Serialize.hpp	2025-01-03 08:37:23.053995300 +0900
@@ -29,6 +29,10 @@
 #include <sstream>
 #include <libmaus2/demangle/Demangle.hpp>
 
+#if defined(_N)
+#undef _N
+#endif
+
 namespace libmaus2
 {
 	namespace serialize
--- origsrc/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/wavelet/DynamicWaveletTree.hpp	2022-12-11 02:30:10.000000000 +0900
+++ src/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/wavelet/DynamicWaveletTree.hpp	2025-01-03 13:22:55.279714000 +0900
@@ -26,6 +26,8 @@
 
 #include <libmaus2/LibMausWindows.hpp>
 
+#include <alloca.h>
+
 namespace libmaus2
 {
 	namespace wavelet
--- origsrc/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/wavelet/ImpCompactHuffmanWaveletTree.hpp	2022-12-11 02:30:10.000000000 +0900
+++ src/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/wavelet/ImpCompactHuffmanWaveletTree.hpp	2025-01-03 08:52:18.427431300 +0900
@@ -27,6 +27,8 @@
 #include <libmaus2/math/numbits.hpp>
 #include <libmaus2/aio/InputStreamFactoryContainer.hpp>
 
+#include <alloca.h>
+
 namespace libmaus2
 {
 	namespace wavelet
--- origsrc/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/wavelet/ImpHuffmanWaveletTree.hpp	2022-12-11 02:30:10.000000000 +0900
+++ src/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/wavelet/ImpHuffmanWaveletTree.hpp	2025-01-03 15:55:11.198083200 +0900
@@ -28,6 +28,8 @@
 #include <libmaus2/math/numbits.hpp>
 #include <libmaus2/aio/InputStreamFactoryContainer.hpp>
 
+#include <alloca.h>
+
 namespace libmaus2
 {
 	namespace wavelet
--- origsrc/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/wavelet/WaveletTree.hpp	2022-12-11 02:30:10.000000000 +0900
+++ src/libmaus2-libmaus2_experimental_2_0_813/src/libmaus2/wavelet/WaveletTree.hpp	2025-01-03 13:40:25.617375300 +0900
@@ -41,6 +41,8 @@
 
 #include <iostream>
 
+#include <alloca.h>
+
 namespace libmaus2
 {
 	namespace wavelet
