--- origsrc/assimp2json-2.0/CMakeLists.txt	2013-10-07 06:06:09.000000000 +0900
+++ src/assimp2json-2.0/CMakeLists.txt	2019-07-13 08:36:30.257577500 +0900
@@ -19,13 +19,8 @@ set ( ASSIMP_BUILD_ASSIMP_TOOLS OFF CACH
 	"If the supplementary tools for Assimp are built in addition to the library."
 )
 
-add_subdirectory (assimp)
-set (EXTRA_LIBS ${EXTRA_LIBS} assimp)
-
-include_directories("assimp/include")
-
 add_executable(assimp2json assimp2json/main.cpp assimp2json/json_exporter.cpp assimp2json/cencode.c assimp2json/mesh_splitter.h  assimp2json/mesh_splitter.cpp)
-target_link_libraries (assimp2json  ${EXTRA_LIBS})
+target_link_libraries (assimp2json assimp)
 
 if( MSVC_IDE )
    add_custom_command(
--- origsrc/assimp2json-2.0/assimp2json/json_exporter.cpp	2013-10-07 06:06:09.000000000 +0900
+++ src/assimp2json-2.0/assimp2json/json_exporter.cpp	2019-07-13 08:32:43.952126200 +0900
@@ -19,7 +19,7 @@ Licensed under a 3-clause BSD license. S
 
 
 // grab scoped_ptr from assimp to avoid a dependency on boost. 
-#include <assimp/../../code/BoostWorkaround/boost/scoped_ptr.hpp>
+#include <boost/scoped_ptr.hpp>
 
 #include "mesh_splitter.h"
 
