diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/assimp2json-2.0/CMakeLists.txt src/assimp2json-2.0/CMakeLists.txt
--- origsrc/assimp2json-2.0/CMakeLists.txt	2013-10-07 06:06:09.000000000 +0900
+++ src/assimp2json-2.0/CMakeLists.txt	2015-11-14 14:04:22.595195400 +0900
@@ -19,13 +19,12 @@ set ( ASSIMP_BUILD_ASSIMP_TOOLS OFF CACH
 	"If the supplementary tools for Assimp are built in addition to the library."
 )
 
-add_subdirectory (assimp)
-set (EXTRA_LIBS ${EXTRA_LIBS} assimp)
-
 include_directories("assimp/include")
 
 add_executable(assimp2json assimp2json/main.cpp assimp2json/json_exporter.cpp assimp2json/cencode.c assimp2json/mesh_splitter.h  assimp2json/mesh_splitter.cpp)
-target_link_libraries (assimp2json  ${EXTRA_LIBS})
+target_link_libraries (assimp2json assimp)
+
+add_definitions(-fpermissive)
 
 if( MSVC_IDE )
    add_custom_command(
@@ -34,3 +33,5 @@ if( MSVC_IDE )
       COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_BINARY_DIR}/assimp/code/$(ConfigurationName)/assimp.dll ${CMAKE_BINARY_DIR}/$(ConfigurationName)/assimp.dll
    )
 endif( MSVC_IDE )
+
+install(TARGETS assimp2json DESTINATION bin)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/assimp2json-2.0/assimp2json/json_exporter.cpp src/assimp2json-2.0/assimp2json/json_exporter.cpp
--- origsrc/assimp2json-2.0/assimp2json/json_exporter.cpp	2013-10-07 06:06:09.000000000 +0900
+++ src/assimp2json-2.0/assimp2json/json_exporter.cpp	2015-11-14 13:59:04.882050100 +0900
@@ -19,7 +19,7 @@ Licensed under a 3-clause BSD license. S
 
 
 // grab scoped_ptr from assimp to avoid a dependency on boost. 
-#include <assimp/../../code/BoostWorkaround/boost/scoped_ptr.hpp>
+#include <boost/scoped_ptr.hpp>
 
 #include "mesh_splitter.h"
 
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/assimp2json-2.0/assimp2json/version.h src/assimp2json-2.0/assimp2json/version.h
--- origsrc/assimp2json-2.0/assimp2json/version.h	1970-01-01 09:00:00.000000000 +0900
+++ src/assimp2json-2.0/assimp2json/version.h	2015-11-14 13:57:51.860702400 +0900
@@ -0,0 +1,2 @@
+#define ASSIMP2JSON_VERSION_MAJOR 2
+#define ASSIMP2JSON_VERSION_MINOR 0
