--- origsrc/libupnpp-0.19.4/Makefile.am	2020-07-02 00:18:31.000000000 +0900
+++ src/libupnpp-0.19.4/Makefile.am	2020-09-17 14:34:19.293424500 +0900
@@ -116,7 +116,7 @@ EXTRA_DIST = autogen.sh
 if LINUX
 # Curiously, -no-undefined seems to do nothing?? -Wl,-zdefs works though.
 # automake 1.14.1
-libupnpp_la_LDFLAGS = -Wl,-zdefs -no-undefined -version-info $(VERSION_INFO)
+libupnpp_la_LDFLAGS = -no-undefined -version-info $(VERSION_INFO)
 endif
 
 libupnpp_la_LIBADD = $(LIBUPNPP_LIBS)
--- origsrc/libupnpp-0.19.4/configure.ac	2020-09-03 18:55:55.000000000 +0900
+++ src/libupnpp-0.19.4/configure.ac	2020-09-17 14:13:17.818501600 +0900
@@ -36,7 +36,7 @@ build_linux=no
 build_mac=no
 # Detect the target system
 case "${host_os}" in
-    linux*|uclinux*) build_linux=yes;;
+    linux*|uclinux*|cygwin*) build_linux=yes;;
     darwin*) build_mac=yes;;
     *) AC_MSG_ERROR(["OS $host_os is not supported"]);;
 esac
