--- origsrc/libupnpp-0.26.6/meson.build	2024-05-31 00:12:08.000000000 +0900
+++ src/libupnpp-0.26.6/meson.build	2024-07-21 20:46:29.180752300 +0900
@@ -26,7 +26,12 @@ elif host_machine.system() == 'sunos'
 endif
 
 deps += dependency('threads')
-deps += dependency('libnpupnp')
+npupnp_dep = dependency('libnpupnp', required : false)
+if npupnp_dep.found()
+  deps += npupnp_dep
+else
+  deps += dependency('npupnp')
+endif
 deps += dependency('libcurl')
 deps += dependency('expat')
 
@@ -109,10 +114,9 @@ cfile = configure_file(
 
 
 libupnpp = library(
-  'libupnpp',
+  'upnpp',
   libupnpp_sources,
   gnu_symbol_visibility: 'hidden',
-  name_prefix: '',
   version: libupnpp_soversion + libupnpp_soversion_minor,
   soversion: host_machine.system() == 'windows' ? '' : libupnpp_soversion,
   include_directories: libupnpp_incdir,
