--- origsrc/libupnpp-0.26.8/meson.build	2025-03-07 16:43:55.000000000 +0900
+++ src/libupnpp-0.26.8/meson.build	2025-03-09 23:09:11.272948700 +0900
@@ -27,7 +27,12 @@ elif host_machine.system() == 'sunos'
 endif
 
 deps += dependency('threads')
-deps += dependency('libnpupnp')
+npupnp_dep = dependency('libnpupnp', required : false)
+if npupnp_dep.found()
+  deps += npupnp_dep
+else
+  deps += dependency('npupnp')
+endif
 deps += dependency('libcurl')
 deps += dependency('expat')
 
@@ -110,10 +115,9 @@ cfile = configure_file(
 
 
 libupnpp = library(
-  'libupnpp',
+  'upnpp',
   libupnpp_sources,
   gnu_symbol_visibility: 'hidden',
-  name_prefix: '',
   version: libupnpp_soversion + libupnpp_soversion_minor,
   soversion: host_machine.system() == 'windows' ? '' : libupnpp_soversion,
   include_directories: libupnpp_incdir,
