--- origsrc/libupnpp-0.20.0/Makefile.am	2020-11-06 18:43:36.000000000 +0900
+++ src/libupnpp-0.20.0/Makefile.am	2020-11-26 12:57:32.860746700 +0900
@@ -143,7 +143,10 @@ EXTRA_DIST = autogen.sh
 if LINUX
 # Curiously, -no-undefined seems to do nothing?? -Wl,-zdefs works though.
 # automake 1.14.1
-libupnpp_la_LDFLAGS = -Wl,-zdefs -no-undefined -version-info $(VERSION_INFO) 
+libupnpp_la_LDFLAGS = -Wl,-zdefs -no-undefined -version-info $(VERSION_INFO)
+else
+libupnpp_la_CPPFLAGS = $(AM_CPPFLAGS) -DUPNPP_DLL_EXPORTS
+libupnpp_la_LDFLAGS = -no-undefined -version-info $(VERSION_INFO)
 endif
 
 libupnpp_la_LIBADD = $(LIBUPNPP_LIBS) libutility.a
