--- origsrc/whisper.cpp-1.5.5/CMakeLists.txt	2024-04-16 20:08:31.000000000 +0900
+++ src/whisper.cpp-1.5.5/CMakeLists.txt	2024-04-28 19:31:21.374077000 +0900
@@ -633,8 +633,15 @@ endif()
 # whisper - this is the main library of the project
 #
 
+find_package(pkgconfig)
+pkg_check_modules(GGML ggml)
+
 set(TARGET whisper)
 
+if (GGML_FOUND)
+add_library(${TARGET} whisper.h whisper.cpp)
+target_link_libraries(${TARGET} PUBLIC ${GGML_LIBRARIES})
+else()
 add_library(${TARGET}
     ggml.h
     ggml.c
@@ -652,6 +659,7 @@ add_library(${TARGET}
     whisper.h
     whisper.cpp
     )
+endif()
 
 # Set the version numbers
 set_target_properties(whisper PROPERTIES
@@ -728,14 +736,18 @@ target_compile_definitions(${TARGET} PUB
     ${WHISPER_EXTRA_FLAGS}
     )
 
+if (GGML_FOUND)
+set_target_properties(${TARGET} PROPERTIES PUBLIC_HEADER "whisper.h")
+else()
 set_target_properties(${TARGET} PROPERTIES PUBLIC_HEADER "ggml.h;whisper.h")
+endif()
 set_target_properties(${TARGET} PROPERTIES FOLDER "libs")
 
 include(GNUInstallDirs)
 
 install(TARGETS ${TARGET}
     LIBRARY  DESTINATION lib
-    ARCHIVE  DESTINATION lib/static
+    ARCHIVE  DESTINATION lib
     RUNTIME  DESTINATION bin
     RESOURCE DESTINATION bin
     PUBLIC_HEADER DESTINATION include
@@ -759,3 +771,10 @@ endif ()
 if (WHISPER_BUILD_EXAMPLES)
     add_subdirectory(examples)
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/whisper.cpp.pc.in ${CMAKE_BINARY_DIR}/whisper.cpp.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/whisper.cpp.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/whisper.cpp-1.5.5/whisper.cpp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/whisper.cpp-1.5.5/whisper.cpp.pc.in	2024-04-28 19:31:11.590455000 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: whisper.cpp
+Description: Port of OpenAI's Whisper model in C/C++
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lwhisper
+Cflags: -I${includedir}
+Requires: ggml
