--- origsrc/libspectrum-0.5.0.1/configure.in	2009-01-15 04:50:55.000000000 +0900
+++ src/libspectrum-0.5.0.1/configure.in	2010-11-28 13:04:30.479686600 +0900
@@ -40,7 +40,7 @@ fi
 AC_MSG_RESULT([$CC_FOR_BUILD])
 AC_SUBST(CC_FOR_BUILD)
 
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_PATH_PROG(PERL, perl)
 AC_SUBST(PERL)
 
--- origsrc/libspectrum-0.5.0.1/test/Makefile.am	2008-11-28 05:47:00.000000000 +0900
+++ src/libspectrum-0.5.0.1/test/Makefile.am	2010-11-28 13:37:01.899539400 +0900
@@ -21,7 +21,8 @@
 ##
 ## E-mail: philip-fuse@shadowmagic.org.uk
 
-noinst_PROGRAMS += test/test
+TESTS = test/test
+check_PROGRAMS = test/test
 
 test_test_SOURCES = \
 	test/edges.c \
