--- origsrc/libspectrum-1.0.0/configure.in	2010-12-17 06:46:49.000000000 +0900
+++ src/libspectrum-1.0.0/configure.in	2012-05-22 08:47:40.309180600 +0900
@@ -56,6 +56,9 @@ AC_C_CONST
 dnl Check for Win32 resource tool
 AC_CHECK_TOOL(WINDRES, windres, no)
 AC_SUBST(WINDRES)
+case "$host_os" in
+  cygwin*) ;;
+  *)
 if test "$WINDRES" != no; then
   WINDRES_OBJ="windres.o"
   WINDRES_LDFLAGS="-Xlinker windres.o"
@@ -65,6 +68,8 @@ if test "$WINDRES" != no; then
                  [LIBSPECTRUM_VERSION])])dnl
   AC_DEFINE([VERSION_WIN32], [translit(LIBSPECTRUM_FULL_VERSION, [.], [,])], [Define version information for win32 dll library])
 fi
+  ;;
+esac
 AC_SUBST(WINDRES_OBJ)
 AM_SUBST_NOTMAKE(WINDRES_OBJ)
 AC_SUBST(WINDRES_LDFLAGS)
@@ -178,10 +183,7 @@ if test "$withval" = no; then libaudiofi
 libaudiofile=yes)
 AC_MSG_RESULT($libaudiofile)
 if test "$libaudiofile" = yes; then
-  AM_PATH_AUDIOFILE(
-    0.2.3,
-    AC_DEFINE([HAVE_LIB_AUDIOFILE], 1, [Defined if we've got audiofile])
-  )
+  PKG_CHECK_MODULES([AUDIOFILE], [audiofile], AC_DEFINE([HAVE_LIB_AUDIOFILE], 1, [Defined if we've got audiofile]))
 fi
 
 AC_OUTPUT(
--- origsrc/libspectrum-1.0.0/test/Makefile.am	2010-12-17 06:58:51.000000000 +0900
+++ src/libspectrum-1.0.0/test/Makefile.am	2012-05-22 07:33:24.102149400 +0900
@@ -21,7 +21,8 @@
 ##
 ## E-mail: philip-fuse@shadowmagic.org.uk
 
-noinst_PROGRAMS += test/test
+TESTS = test/test
+check_PROGRAMS = test/test
 
 test_test_SOURCES = \
 	test/edges.c \
