--- origsrc/libspectrum-1.4.4/test/Makefile.am	2018-04-28 10:45:48.000000000 +0900
+++ src/libspectrum-1.4.4/test/Makefile.am	2018-08-05 10:00:34.991316100 +0900
@@ -34,7 +34,7 @@ noinst_DATA += test/complete-tzx.tzx
 noinst_HEADERS += test/test.h
 
 test/complete-tzx.tzx: test/complete-tzx.pl
-	$(AM_V_GEN)$(PERL) $< > $@.tmp && mv $@.tmp $@
+	$(AM_V_GEN)mkdir -p test && $(PERL) $< > $@.tmp && mv $@.tmp $@
 
 test_test_LDADD = libspectrum.la
 
