--- origsrc/glewlwyd-2.7.6/CMakeLists.txt	2023-11-16 04:43:26.000000000 +0900
+++ src/glewlwyd-2.7.6/CMakeLists.txt	2025-11-29 09:15:56.315566400 +0900
@@ -18,7 +18,7 @@ cmake_minimum_required(VERSION 3.5)
 project(glewlwyd C)
 
 set(CMAKE_C_STANDARD 99)
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
 
 # library info
 
@@ -418,7 +418,7 @@ if (WITH_SCHEME_WEBAUTHN)
   set_target_properties(schememodwebauthn PROPERTIES
       COMPILE_OPTIONS "-Wextra;-Wconversion"
       LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/scheme")
-  target_link_libraries(schememodwebauthn ${GLWD_LIBS} CBOR::CBOR OpenLDAP::OpenLDAP Rhonabwy::Rhonabwy)
+  target_link_libraries(schememodwebauthn ${GLWD_LIBS} CBOR::CBOR OpenLDAP::OpenLDAP Rhonabwy::Rhonabwy lber)
   list(APPEND SCHEME_MODULES schememodwebauthn)
 endif ()
 
@@ -489,7 +489,7 @@ if (WITH_MOCK OR BUILD_GLEWLWYD_TESTING)
   set_target_properties(pluginmodmock PROPERTIES
       COMPILE_OPTIONS "-Wextra;-Wconversion"
       LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/plugin")
-  target_link_libraries(pluginmodmock ${MOCK_LIBS})
+  target_link_libraries(pluginmodmock ${MOCK_LIBS} ${GLWD_LIBS})
   list(APPEND PLUGIN_MODULES pluginmodmock)
 endif ()
 
