--- origsrc/glewlwyd-2.3.3/CMakeLists.txt	2020-08-24 21:00:18.000000000 +0900
+++ src/glewlwyd-2.3.3/CMakeLists.txt	2020-09-06 14:06:13.564488300 +0900
@@ -18,7 +18,9 @@ cmake_minimum_required(VERSION 3.5)
 project(glewlwyd C)
 
 set(CMAKE_C_STANDARD 99)
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror")
+if (NOT CYGWIN)
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror")
+endif()
 
 # library info
 
@@ -431,6 +433,9 @@ if (WITH_SCHEME_WEBAUTHN)
 	find_package(Ldap REQUIRED)
 	if (LDAP_FOUND)
 		include_directories(${LDAP_INCLUDE_DIRS})
+		if (CYGWIN)
+			set(LDAP_LIBRARIES ${LDAP_LIBRARIES} lber)
+		endif()
 	endif ()
 	
 	set(LIB_SRC ${CMAKE_CURRENT_SOURCE_DIR}/src/glewlwyd-common.h ${CMAKE_CURRENT_SOURCE_DIR}/src/misc.c ${SCHEME_MODULES_SRC_PATH}/webauthn.c)
@@ -577,7 +582,7 @@ if (WITH_MOCK OR BUILD_GLEWLWYD_TESTING)
 	set_target_properties(pluginmodmock PROPERTIES
 			COMPILE_OPTIONS -Wextra
 			LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/plugin")
-	target_link_libraries(pluginmodmock ${MOCK_LIBS})
+	target_link_libraries(pluginmodmock ${MOCK_LIBS} ${GLWD_LIBS})
 	set(PLUGIN_MODULES ${PLUGIN_MODULES} pluginmodmock)
 endif ()
 
--- origsrc/glewlwyd-2.3.3/test/glewlwyd_register.c	2020-08-24 21:00:18.000000000 +0900
+++ src/glewlwyd-2.3.3/test/glewlwyd_register.c	2020-09-06 14:06:13.835424100 +0900
@@ -73,6 +73,10 @@ struct smtp_manager {
   const char * body_pattern;
 };
 
+#if !defined(SO_REUSEPORT)
+#define SO_REUSEPORT (0)
+#endif
+
 /**
  * 
  * Function that emulates a very simple SMTP server
--- origsrc/glewlwyd-2.3.3/test/glewlwyd_scheme_mail.c	2020-08-24 21:00:18.000000000 +0900
+++ src/glewlwyd-2.3.3/test/glewlwyd_scheme_mail.c	2020-09-06 14:06:13.991612600 +0900
@@ -64,6 +64,10 @@ struct smtp_manager {
   const char * body_pattern;
 };
 
+#if !defined(SO_REUSEPORT)
+#define SO_REUSEPORT (0)
+#endif
+
 /**
  * 
  * Function that emulates a very simple SMTP server
