--- origsrc/mimetic-0.9.8/mimetic/Makefile.am	2014-06-17 17:12:00.000000000 +0900
+++ src/mimetic-0.9.8/mimetic/Makefile.am	2014-07-15 21:10:34.569231800 +0900
@@ -22,7 +22,7 @@ libmimetic_la_SOURCES=body.cxx contenttr
 libmimetic_la_LIBADD =codec/libmimeticcodec.la os/libmimeticos.la rfc822/libmimeticrfc822.la 
 
 # ver: interface:revision:age
-libmimetic_la_LDFLAGS= -version-info 0:0:0
+libmimetic_la_LDFLAGS= -no-undefined -version-info 0:0:0
 
 
 config_win32.h: config_win32.h.in
--- origsrc/mimetic-0.9.8/test/Makefile.am	2014-06-17 17:12:00.000000000 +0900
+++ src/mimetic-0.9.8/test/Makefile.am	2014-07-15 21:14:37.282122400 +0900
@@ -15,9 +15,10 @@ all-local: cutee runtest
 clean-local: touch-autocutee.mk runtest-clean
 	rm -f tests.out
 
-noinst_PROGRAMS=cutee
+noinst_PROGRAMS=cutee runtest
 cutee_SOURCES=cutee.cxx cutee.h
 cutee_LDADD=
+runtest_SOURCES=runtest.cxx
 
 touch-autocutee.mk:
 	touch autocutee.mk
