--- origsrc/mimetic-0.9.6/configure.in	2009-05-13 17:18:12.000000000 +0900
+++ src/mimetic-0.9.6/configure.in	2010-11-28 19:36:34.361820700 +0900
@@ -9,7 +9,7 @@ AM_CONFIG_HEADER(mimetic/config.h)
 dnl Checks for programs.
 AC_PROG_CC
 AC_PROG_MAKE_SET
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 dnl defines HAVE_MMAP if mmapping works
 AC_FUNC_MMAP
--- origsrc/mimetic-0.9.6/mimetic/Makefile.am	2009-05-08 22:47:58.000000000 +0900
+++ src/mimetic-0.9.6/mimetic/Makefile.am	2010-11-28 20:29:08.722317300 +0900
@@ -2,7 +2,7 @@ SUBDIRS=codec os rfc822 parser
 EXTRA_DIST=config_win32.h config_win32.h.in
 lib_LTLIBRARIES = libmimetic.la
 
-INCLUDES=-I$(top_srcdir)
+INCLUDES=-I$(top_srcdir) -I$(top_builddir)
 #CPPFLAGS=-Winline
 includedir=@INCLUDEDIR@
 
@@ -22,7 +22,7 @@ libmimetic_la_SOURCES=body.cxx contenttr
 libmimetic_la_LIBADD =codec/libmimeticcodec.la os/libmimeticos.la rfc822/libmimeticrfc822.la 
 
 # ver: interface:revision:age
-libmimetic_la_LDFLAGS= -version-info 0:0:0
+libmimetic_la_LDFLAGS= -version-info 0:0:0 -no-undefined
 
 
 config_win32.h: config_win32.h.in
--- origsrc/mimetic-0.9.6/test/Makefile.am	2009-05-08 22:47:58.000000000 +0900
+++ src/mimetic-0.9.6/test/Makefile.am	2010-11-28 23:56:54.466568000 +0900
@@ -15,9 +15,10 @@ all-local: cutee runtest
 clean-local: touch-autocutee.mk runtest-clean
 	rm -f tests.out
 
-noinst_PROGRAMS=cutee
+noinst_PROGRAMS=cutee runtest
 cutee_SOURCES=cutee.cxx cutee.h
 cutee_LDADD=
+runtest_SOURCES=runtest.cxx
 
 touch-autocutee.mk:
 	touch autocutee.mk
