--- origsrc/libxdg-basedir-libxdg-basedir-1.2.3/m4/attributes.m4	2021-04-20 18:58:13.000000000 +0900
+++ src/libxdg-basedir-libxdg-basedir-1.2.3/m4/attributes.m4	2021-04-24 19:05:00.210998900 +0900
@@ -106,6 +106,9 @@ AC_DEFUN([CC_NOUNDEFINED], [
   AC_REQUIRE([AC_CANONICAL_HOST])
 
   case $host in
+     *cygwin*)
+      LDFLAGS_NOUNDEFINED=-no-undefined
+      ;;
      dnl FreeBSD (et al.) does not complete linking for shared objects when pthreads
      dnl are requested, as different implementations are present; to avoid problems
      dnl use -Wl,-z,defs only for those platform not behaving this way.
