--- origsrc/vmime/CMakeLists.txt	2023-04-06 08:15:15.000000000 +0900
+++ src/vmime/CMakeLists.txt	2023-04-06 08:15:43.788088300 +0900
@@ -237,7 +237,8 @@ IF(VMIME_BUILD_SHARED_LIBRARY)
 	INSTALL(
 		TARGETS ${VMIME_LIBRARY_NAME}
 		EXPORT ${VMIME_LIBRARY_NAME}-config
-		LIBRARY DESTINATION "${VMIME_INSTALL_LIBDIR}"
+		RUNTIME DESTINATION "${VMIME_INSTALL_BINDIR}"
+		ARCHIVE DESTINATION "${VMIME_INSTALL_LIBDIR}"
 		PUBLIC_HEADER DESTINATION "${VMIME_INSTALL_INCLUDEDIR}"
 		COMPONENT sharedlibs
 	)
@@ -310,9 +311,9 @@ IF(VMIME_BUILD_TESTS)
 		)
 
 		TARGET_LINK_LIBRARIES(
-			${VMIME_TEST_NAME}
+			${VMIME_TEST_NAME} test-runner
 			${VMIME_LIBRARY_NAME}
-			${CPPUNIT_LIBRARY} test-runner
+			${CPPUNIT_LIBRARY}
 		)
 
 		ADD_DEPENDENCIES(
--- origsrc/vmime/src/vmime/charsetConverter.hpp	2023-04-06 08:15:15.000000000 +0900
+++ src/vmime/src/vmime/charsetConverter.hpp	2023-04-06 08:15:43.795053300 +0900
@@ -66,7 +66,7 @@ public:
 
 	/** Holds information about a conversion.
 	  */
-	struct status {
+	struct VMIME_EXPORT status {
 
 		status();
 
--- origsrc/vmime/src/vmime/net/smtp/SMTPResponse.hpp	2023-04-06 08:15:16.000000000 +0900
+++ src/vmime/src/vmime/net/smtp/SMTPResponse.hpp	2023-04-06 08:15:43.816097300 +0900
@@ -73,7 +73,7 @@ public:
 	};
 
 	/** An element of a SMTP response. */
-	class responseLine {
+	class VMIME_EXPORT responseLine {
 
 	public:
 
--- origsrc/vmime/src/vmime/platforms/posix/posixHandler.cpp	2023-04-06 08:15:16.000000000 +0900
+++ src/vmime/src/vmime/platforms/posix/posixHandler.cpp	2023-04-06 08:15:43.826077900 +0900
@@ -227,6 +227,8 @@ unsigned int posixHandler::getThreadId()
 	return static_cast <unsigned int>(::syscall(SYS_gettid));
 #elif VMIME_HAVE_GETTHRID  // OpenBSD
 	return static_cast <unsigned int>(::getthrid());
+#elif defined(__CYGWIN__)
+	return getProcessId();
 #else
 	#error We have no implementation of getThreadId() for this platform!
 #endif
--- origsrc/vmime/src/vmime/utility/filteredStream.hpp	2023-04-06 08:15:16.000000000 +0900
+++ src/vmime/src/vmime/utility/filteredStream.hpp	2023-04-06 08:15:43.875087500 +0900
@@ -393,7 +393,7 @@ private:
 };
 
 
-template <>
+template <> VMIME_EXPORT
 size_t stopSequenceFilteredInputStream <1>::read(byte_t* const data, const size_t count);
 
 
