diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x config.hpp -x export-shared.hpp origsrc/vmime/CMakeLists.txt src/vmime/CMakeLists.txt
--- origsrc/vmime/CMakeLists.txt	2018-04-18 11:01:43.000000000 +0900
+++ src/vmime/CMakeLists.txt	2018-04-18 18:48:58.381000000 +0900
@@ -213,6 +213,7 @@ IF(VMIME_BUILD_SHARED_LIBRARY)
 	INSTALL(
 		TARGETS ${VMIME_LIBRARY_NAME}
 		EXPORT ${VMIME_LIBRARY_NAME}-config
+		RUNTIME DESTINATION bin COMPONENT sharedlibs
 		LIBRARY DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}" COMPONENT sharedlibs
 		ARCHIVE DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}" COMPONENT sharedlibs
 	)
@@ -237,8 +238,11 @@ INSTALL_HEADERS_WITH_DIRECTORY(VMIME_LIB
 #	COMPONENT headers
 #)
 
-install(EXPORT ${VMIME_LIBRARY_NAME}-config DESTINATION cmake)
-
+if(CYGWIN)
+	install(EXPORT ${VMIME_LIBRARY_NAME}-config DESTINATION lib/cmake)
+else()
+	install(EXPORT ${VMIME_LIBRARY_NAME}-config DESTINATION cmake)
+endif()
 
 ##############################################################################
 # Tests
@@ -284,9 +288,9 @@ IF(VMIME_BUILD_TESTS)
 		)
 
 		TARGET_LINK_LIBRARIES(
-			${VMIME_TEST_NAME}
+			${VMIME_TEST_NAME} test-runner
 			${VMIME_LIBRARY_NAME}
-			${CPPUNIT_LIBRARY} test-runner
+			${CPPUNIT_LIBRARY}
 		)
 
 		ADD_DEPENDENCIES(
@@ -308,9 +312,9 @@ IF(VMIME_BUILD_TESTS)
 	)
 
 	TARGET_LINK_LIBRARIES(
-		"run-tests"
+		"run-tests" test-runner
 		${VMIME_LIBRARY_NAME}
-		${CPPUNIT_LIBRARY} test-runner
+		${CPPUNIT_LIBRARY}
 	)
 
 	ADD_DEPENDENCIES(
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x config.hpp -x export-shared.hpp origsrc/vmime/src/vmime/charsetConverter.hpp src/vmime/src/vmime/charsetConverter.hpp
--- origsrc/vmime/src/vmime/charsetConverter.hpp	2018-04-18 11:01:44.000000000 +0900
+++ src/vmime/src/vmime/charsetConverter.hpp	2018-04-18 14:01:26.188999800 +0900
@@ -70,7 +70,7 @@ public:
 
 	/** Holds information about a conversion.
 	  */
-	struct status
+	struct VMIME_EXPORT status
 	{
 		status();
 
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x config.hpp -x export-shared.hpp origsrc/vmime/src/vmime/net/smtp/SMTPResponse.hpp src/vmime/src/vmime/net/smtp/SMTPResponse.hpp
--- origsrc/vmime/src/vmime/net/smtp/SMTPResponse.hpp	2018-04-18 11:01:44.000000000 +0900
+++ src/vmime/src/vmime/net/smtp/SMTPResponse.hpp	2018-04-18 13:59:25.206000000 +0900
@@ -71,7 +71,7 @@ public:
 	};
 
 	/** An element of a SMTP response. */
-	class responseLine
+	class VMIME_EXPORT responseLine
 	{
 	public:
 
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x config.hpp -x export-shared.hpp origsrc/vmime/src/vmime/platforms/posix/posixHandler.cpp src/vmime/src/vmime/platforms/posix/posixHandler.cpp
--- origsrc/vmime/src/vmime/platforms/posix/posixHandler.cpp	2018-04-18 11:01:44.000000000 +0900
+++ src/vmime/src/vmime/platforms/posix/posixHandler.cpp	2018-04-18 11:26:25.251000000 +0900
@@ -262,6 +262,8 @@ unsigned int posixHandler::getThreadId()
 	return static_cast <unsigned int>(::syscall(SYS_gettid));
 #elif VMIME_HAVE_GETTHRID  // OpenBSD
 	return static_cast <unsigned int>(::getthrid());
+#elif defined(__CYGWIN__)
+	return getProcessId();
 #else
 	#error We have no implementation of getThreadId() for this platform!
 #endif
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x config.hpp -x export-shared.hpp origsrc/vmime/src/vmime/utility/filteredStream.hpp src/vmime/src/vmime/utility/filteredStream.hpp
--- origsrc/vmime/src/vmime/utility/filteredStream.hpp	2018-04-18 11:01:44.000000000 +0900
+++ src/vmime/src/vmime/utility/filteredStream.hpp	2018-04-18 17:43:29.895000000 +0900
@@ -394,7 +394,7 @@ private:
 };
 
 
-template <>
+template <> VMIME_EXPORT
 size_t stopSequenceFilteredInputStream <1>::read
 	(byte_t* const data, const size_t count);
 
