--- origsrc/fflas-ffpack-1.4.1/configure.ac	2011-05-16 19:47:20.000000000 +0900
+++ src/fflas-ffpack-1.4.1/configure.ac	2011-07-31 23:57:50.483493400 +0900
@@ -214,17 +214,9 @@ echo '**********************************
 # LB_CHECK_MAPLE
 # LB_CHECK_SAGE
 #LB_CHECK_ATLAS
-FF_CHECK_BLAS(,,[
-echo ''
-echo '*******************************************************************************'
-echo ' ERROR: BLAS not found!'
-echo
-echo ' BLAS routines are required for this library to compile. Please'
-echo ' make sure BLAS are installed and specify its location with the option'
-echo ' --with-blas=<lib> when running configure.'
-echo '*******************************************************************************'
-exit 1
-])
+PKG_CHECK_MODULES(BLAS, blas)
+CBLAS_FLAG="-D__FFLAFLAS_HAVE_CBLAS"
+AC_SUBST(CBLAS_FLAG)
 # LB_CHECK_EXPAT
 
 # LB_BENCH
--- origsrc/fflas-ffpack-1.4.1/tests/Makefile.am	2011-05-16 19:47:19.000000000 +0900
+++ src/fflas-ffpack-1.4.1/tests/Makefile.am	2011-08-01 00:34:10.741734400 +0900
@@ -12,7 +12,7 @@ AM_CPPFLAGS=-I$(top_srcdir)
 AM_CXXFLAGS = @TESTS_CFLAGS@
 AM_CPPFLAGS += $(OPTFLAGS) -I$(top_srcdir)/fflas-ffpack/utils/ -I$(top_srcdir)/fflas-ffpack/fflas/  -I$(top_srcdir)/fflas-ffpack/ffpack  -I$(top_srcdir)/fflas-ffpack/field  $(CBLAS_FLAG) $(GMP_CFLAGS)  $(GIVARO_CFLAGS)
 LDADD = $(BLAS_LIBS) $(GIVARO_LIBS) $(GMP_LIBS)
-AM_LDFLAGS=-static
+AM_LDFLAGS=
 
 BASIC_TESTS =              \
 		test-charpoly          \
