--- origsrc/fflas-ffpack-1.6.0/configure.ac	2012-10-09 21:29:55.945314800 +0900
+++ src/fflas-ffpack-1.6.0/configure.ac	2012-10-09 21:43:23.254885100 +0900
@@ -240,15 +240,12 @@ echo ' Do not forget to set/export LD_LI
 echo '*******************************************************************************'
 ])
 
-PKG_CHECK_MODULES([BLAS_LIBS], [cblas])
-
-PKG_CHECK_MODULES([LAPACK_LIBS], [lapack])
+PKG_CHECK_MODULES([BLAS], [blas])
+PKG_CHECK_MODULES([LAPACK], [lapack])
+AC_CHECK_LIB([clapack], [main])
 
 BLAS_FOUND=true
 
-AC_SUBST(BLAS_LIBS)
-AC_SUBST(LAPACK_LIBS)
-
 # AM_CONDITIONAL(FFLASFFPACK_HAVE_BLAS,  test "x$BLAS_FOUND" != "xfalse")
 
 
--- origsrc/fflas-ffpack-1.6.0/tests/Makefile.am	2012-03-26 22:24:57.000000000 +0900
+++ src/fflas-ffpack-1.6.0/tests/Makefile.am	2012-10-09 21:31:05.730471000 +0900
@@ -28,7 +28,7 @@ AM_CPPFLAGS=-I$(top_srcdir)
 AM_CXXFLAGS = @TESTS_CFLAGS@
 AM_CPPFLAGS += $(OPTFLAGS) -I$(top_srcdir)/fflas-ffpack/utils/ -I$(top_srcdir)/fflas-ffpack/fflas/  -I$(top_srcdir)/fflas-ffpack/ffpack  -I$(top_srcdir)/fflas-ffpack/field  $(CBLAS_FLAG) $(GMP_CFLAGS)  $(GIVARO_CFLAGS)
 LDADD = $(BLAS_LIBS) $(GIVARO_LIBS) $(GMP_LIBS)
-AM_LDFLAGS=-static
+AM_LDFLAGS=
 
 BASIC_TESTS =              \
 		test-lqup              \
