--- origsrc/rndpassw-0.2/CMakeLists.txt	2014-03-02 21:53:23.000000000 +0900
+++ src/rndpassw-0.2/CMakeLists.txt	2015-03-01 22:59:26.526108200 +0900
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 2.8)
 project(Cian C)
 
-set(CMAKE_C_FLAGS "-fPIC -std=iso9899:1999 -Werror -Wall -Wextra -Wmissing-prototypes")
+set(CMAKE_C_FLAGS "-std=iso9899:1999 -Wall -Wextra -Wmissing-prototypes")
 set(CMAKE_C_FLAGS_DEBUG "-O0 -g3")
 set(CMAKE_C_FLAGS_RELEASE "-O2 -fno-strict-aliasing")
 
--- origsrc/rndpassw-0.2/doc/man/CMakeLists.txt	2014-03-02 21:53:00.000000000 +0900
+++ src/rndpassw-0.2/doc/man/CMakeLists.txt	2015-03-01 22:59:50.245848100 +0900
@@ -16,6 +16,6 @@ add_custom_target(manpage ALL
 
 foreach(manpage ${man1_src})
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${manpage}.gz
-            DESTINATION man/man1)
+            DESTINATION share/man/man1)
 endforeach()
 
--- origsrc/rndpassw-0.2/src/CMakeLists.txt	2014-01-08 01:59:06.000000000 +0900
+++ src/rndpassw-0.2/src/CMakeLists.txt	2015-03-01 23:01:46.371234700 +0900
@@ -5,3 +5,4 @@ list(SORT src)
 
 add_executable(rndpassw ${src})
 
+install(TARGETS rndpassw DESTINATION bin)
