--- origsrc/qmdnsengine-0.2.1/CMakeLists.txt	2025-05-01 13:27:44.000000000 +0900
+++ src/qmdnsengine-0.2.1/CMakeLists.txt	2025-09-18 14:37:18.535433700 +0900
@@ -84,3 +84,10 @@ set(CPACK_COMPONENT_EXAMPLES_DISPLAY_NAM
 set(CPACK_COMPONENT_EXAMPLES_DESCRIPTION "Sample applications using the library")
 
 include(CPack)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/qmdnsengine.pc.in ${CMAKE_BINARY_DIR}/qmdnsengine.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/qmdnsengine.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/qmdnsengine-0.2.1/qmdnsengine.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/qmdnsengine-0.2.1/qmdnsengine.pc.in	2025-09-18 14:37:24.053846700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: qmdnsengine
+Description: Multicast DNS library for Qt applications
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lqmdnsengine
+Cflags: -I${includedir}
