--- origsrc/groonga-10.1.0/configure.ac	2020-12-21 11:01:18.000000000 +0900
+++ src/groonga-10.1.0/configure.ac	2021-01-24 10:59:40.577348400 +0900
@@ -1440,7 +1440,7 @@ AC_MSG_CHECKING([whether package platfor
 AC_ARG_WITH(package-platform,
   [AS_HELP_STRING([--with-package-platform=PLATFORM],
     [install package platform related files. [default=no]
-     (supported package platforms: centos, centos6, centos7, fedora)])],
+     (supported package platforms: centos, centos6, centos7, fedora, cygwin)])],
   [package_platform="$withval"],
   [package_platform="no"])
 if test "$package_platform" = "centos"; then
@@ -1461,7 +1461,7 @@ centos_systemd_platform=no
 case "${package_platform}" in
   no)
     ;;
-  centos6)
+  centos6|cygwin)
     centos_platform=yes
     centos_init_platform=yes
     ;;
@@ -1602,7 +1602,7 @@ AC_ARG_ENABLE(groonga_httpd,
   [enable_groonga_httpd="yes"])
 if test "x$enable_groonga_httpd" != "xno"; then
   enable_groonga_httpd="yes"
-  AC_CONFIG_SUBDIRS([src/httpd])
+  # AC_CONFIG_SUBDIRS([src/httpd])
 else
   enable_groonga_httpd="no"
 fi
@@ -1682,7 +1682,7 @@ if test "x$with_onigmo" != "xno"; then
     AC_MSG_ERROR("No Onigmo found")
   fi
   if test "x$with_onigmo" = "xbundled" -o "$have_onigmo" = "no"; then
-    AC_CONFIG_SUBDIRS([vendor/onigmo])
+    # AC_CONFIG_SUBDIRS([vendor/onigmo])
     ONIGMO_CFLAGS="-I\$(top_srcdir)/vendor/onigmo-source"
     ONIGMO_LIBS="\$(top_builddir)/vendor/onigmo-source/libonigmo.la"
   fi
--- origsrc/groonga-10.1.0/test/unit/Makefile.am	2020-12-21 11:01:18.000000000 +0900
+++ src/groonga-10.1.0/test/unit/Makefile.am	2021-01-24 10:59:40.602371000 +0900
@@ -32,7 +32,7 @@ AM_CPPFLAGS =					\
 	-I$(top_srcdir)/lib			\
 	$(GROONGA_INCLUDEDIR)
 
-AM_LDFLAGS = -module -rpath $(libdir) -avoid-version
+AM_LDFLAGS = -no-undefined -module -rpath $(libdir) -avoid-version
 
 LIBS =						\
 	$(top_builddir)/lib/libgroonga.la
