--- origsrc/groonga-1.2.0/Makefile.am	2011-03-27 16:26:15.000000000 +0900
+++ src/groonga-1.2.0/Makefile.am	2011-05-27 00:44:03.440790400 +0900
@@ -1,6 +1,5 @@
 # release: update-latest-release (commit) update-changelog (commit) tag
 
-ACLOCAL_AMFLAGS = $$ACLOCAL_ARGS -I .
 AUTOMAKE_OPTIONS = 1.9.6
 SUBDIRS =					\
 	include					\
--- origsrc/groonga-1.2.0/lib/ctx.c	2011-03-27 16:26:15.000000000 +0900
+++ src/groonga-1.2.0/lib/ctx.c	2011-05-27 01:05:32.352745400 +0900
@@ -2531,6 +2531,9 @@ grn_set_segv_handler(void)
 
   sigemptyset(&action.sa_mask);
   action.sa_sigaction = segv_handler;
+#if !defined(SA_ONSTACK)
+#define SA_ONSTACK 0
+#endif
   action.sa_flags = SA_SIGINFO | SA_ONSTACK;
 
   if (sigaction(SIGSEGV, &action, NULL)) {
--- origsrc/groonga-1.2.0/src/Makefile.am	2011-01-27 20:45:31.000000000 +0900
+++ src/groonga-1.2.0/src/Makefile.am	2011-05-27 01:05:54.366651200 +0900
@@ -8,7 +8,7 @@ DEFS += $(GRN_DEFS)
 
 AM_LDFLAGS = -no-undefined
 
-DEFAULT_INCLUDES = -I$(top_builddir) -I$(srcdir) -I$(top_srcdir)/include $(GROONGA_INCLUDEDIR)
+DEFAULT_INCLUDES = -I$(top_builddir) -I$(srcdir) -I$(top_srcdir)/include $(GROONGA_INCLUDEDIR) -I$(top_srcdir)
 
 groonga_SOURCES = groonga.c
 groonga_CFLAGS = $(AM_CFLAGS) $(LIBEDIT_CFLAGS)
