--- origsrc/groonga-2.0.0/Makefile.am	2011-11-15 11:14:11.890792500 +0900
+++ src/groonga-2.0.0/Makefile.am	2012-03-23 12:59:27.031139400 +0900
@@ -2,7 +2,7 @@
 
 LOCALES = ja
 
-ACLOCAL_AMFLAGS = $$ACLOCAL_ARGS -I .
+ACLOCAL_AMFLAGS = -I .
 AUTOMAKE_OPTIONS = 1.9.6
 SUBDIRS =					\
 	include					\
--- origsrc/groonga-2.0.0/configure.ac	2012-02-28 16:02:46.782476200 +0900
+++ src/groonga-2.0.0/configure.ac	2012-03-23 14:29:42.209699200 +0900
@@ -980,7 +980,7 @@ AC_MSG_CHECKING([whether package platfor
 AC_ARG_WITH(package-platform,
   [AS_HELP_STRING([--with-package-platform=PLATFORM],
     [install package platform related files. [default=no]
-     (supported package platforms: redhat, fedora)])],
+     (supported package platforms: redhat, fedora, cygwin)])],
   [package_platform="$withval"],
   [package_platform="no"])
 AC_MSG_RESULT($package_platform)
@@ -989,6 +989,8 @@ AM_CONDITIONAL([REDHAT_PLATFORM],
                [test "${package_platform}" = "redhat"])
 AM_CONDITIONAL([FEDORA_PLATFORM],
                [test "${package_platform}" = "fedora"])
+AM_CONDITIONAL([CYGWIN_PLATFORM],
+               [test "${package_platform}" = "cygwin"])
 
 # plugins check
 relative_pluginsdir="\$(PACKAGE)/plugins"
--- origsrc/groonga-2.0.0/data/init.d/redhat/Makefile.am	2012-02-02 13:20:59.984480400 +0900
+++ src/groonga-2.0.0/data/init.d/redhat/Makefile.am	2012-03-23 14:30:20.944074200 +0900
@@ -6,5 +6,10 @@ if REDHAT_PLATFORM
 initdir = $(sysconfdir)/init.d
 dist_init_SCRIPTS = $(scripts)
 else
+if CYGWIN_PLATFORM
+initdir = $(sysconfdir)/rc.d/init.d
+dist_init_SCRIPTS = $(scripts)
+else
 EXTRA_DIST = $(scripts)
 endif
+endif
--- origsrc/groonga-2.0.0/lib/ctx.c	2012-02-28 13:31:29.898338800 +0900
+++ src/groonga-2.0.0/lib/ctx.c	2012-03-23 13:24:59.015514400 +0900
@@ -2383,6 +2383,9 @@ grn_set_segv_handler(void)
 
   sigemptyset(&action.sa_mask);
   action.sa_sigaction = segv_handler;
+#if !defined(SA_ONSTACK)
+#define SA_ONSTACK 0
+#endif
   action.sa_flags = SA_SIGINFO | SA_ONSTACK;
 
   if (sigaction(SIGSEGV, &action, NULL)) {
--- origsrc/groonga-2.0.0/lib/dat/dat.hpp	2012-02-28 16:02:46.718476200 +0900
+++ src/groonga-2.0.0/lib/dat/dat.hpp	2012-03-23 13:48:17.549694100 +0900
@@ -56,6 +56,13 @@ typedef ::uint32_t UInt32;
 typedef ::uint64_t UInt64;
 #endif  // _MSC_VER
 
+#if defined(__CYGWIN__)
+#undef UINT8_MAX
+#undef UINT16_MAX
+#undef UINT32_MAX
+#undef UINT64_MAX
+#endif
+
 #ifndef UINT8_MAX
 const UInt8 UINT8_MAX = static_cast<UInt8>(0xFFU);
 #endif  // UINT8_MAX
--- origsrc/groonga-2.0.0/src/Makefile.am	2012-02-15 17:56:28.052729400 +0900
+++ src/groonga-2.0.0/src/Makefile.am	2012-03-23 13:28:32.734264400 +0900
@@ -14,7 +14,7 @@ DEFS += $(GRN_DEFS)
 
 AM_LDFLAGS = -no-undefined
 
-DEFAULT_INCLUDES = -I$(top_builddir) -I$(srcdir) -I$(top_srcdir)/include $(GROONGA_INCLUDEDIR)
+DEFAULT_INCLUDES = -I$(top_builddir) -I$(srcdir) -I$(top_srcdir)/include $(GROONGA_INCLUDEDIR) -I$(top_srcdir)
 
 groonga_SOURCES = groonga.c
 nodist_EXTRA_groonga_SOURCES = $(NONEXISTENT_CXX_SOURCE)
