--- origsrc/groonga-1.2.3/Makefile.am	2011-06-28 21:51:56.297963600 +0900
+++ src/groonga-1.2.3/Makefile.am	2011-07-09 20:34:27.631077500 +0900
@@ -1,6 +1,5 @@
 # release: update-latest-release (commit) tag
 
-ACLOCAL_AMFLAGS = $$ACLOCAL_ARGS -I .
 AUTOMAKE_OPTIONS = 1.9.6
 SUBDIRS =					\
 	include					\
--- origsrc/groonga-1.2.3/lib/ctx.c	2011-06-28 21:51:56.485958000 +0900
+++ src/groonga-1.2.3/lib/ctx.c	2011-07-09 20:34:27.706653700 +0900
@@ -2532,6 +2532,9 @@ grn_set_segv_handler(void)
 
   sigemptyset(&action.sa_mask);
   action.sa_sigaction = segv_handler;
+#if !defined(SA_ONSTACK)
+#define SA_ONSTACK 0
+#endif
   action.sa_flags = SA_SIGINFO | SA_ONSTACK;
 
   if (sigaction(SIGSEGV, &action, NULL)) {
--- origsrc/groonga-1.2.3/src/Makefile.am	2011-01-27 20:45:31.772653600 +0900
+++ src/groonga-1.2.3/src/Makefile.am	2011-07-09 20:34:27.836212900 +0900
@@ -8,7 +8,7 @@ DEFS += $(GRN_DEFS)
 
 AM_LDFLAGS = -no-undefined
 
-DEFAULT_INCLUDES = -I$(top_builddir) -I$(srcdir) -I$(top_srcdir)/include $(GROONGA_INCLUDEDIR)
+DEFAULT_INCLUDES = -I$(top_builddir) -I$(srcdir) -I$(top_srcdir)/include $(GROONGA_INCLUDEDIR) -I$(top_srcdir)
 
 groonga_SOURCES = groonga.c
 groonga_CFLAGS = $(AM_CFLAGS) $(LIBEDIT_CFLAGS)
