--- origsrc/groonga-1.2.9/Makefile.am	2011-11-13 13:48:24.729515700 +0900
+++ src/groonga-1.2.9/Makefile.am	2012-01-15 12:14:55.128126300 +0900
@@ -2,7 +2,6 @@
 
 LOCALES = ja
 
-ACLOCAL_AMFLAGS = $$ACLOCAL_ARGS -I .
 AUTOMAKE_OPTIONS = 1.9.6
 SUBDIRS =					\
 	include					\
--- origsrc/groonga-1.2.9/lib/ctx.c	2011-12-27 21:11:05.659912000 +0900
+++ src/groonga-1.2.9/lib/ctx.c	2012-01-15 12:14:55.138140700 +0900
@@ -2560,6 +2560,9 @@ grn_set_segv_handler(void)
 
   sigemptyset(&action.sa_mask);
   action.sa_sigaction = segv_handler;
+#if !defined(SA_ONSTACK)
+#define SA_ONSTACK 0
+#endif
   action.sa_flags = SA_SIGINFO | SA_ONSTACK;
 
   if (sigaction(SIGSEGV, &action, NULL)) {
--- origsrc/groonga-1.2.9/src/Makefile.am	2011-11-13 13:34:21.710597200 +0900
+++ src/groonga-1.2.9/src/Makefile.am	2012-01-15 12:14:55.138140700 +0900
@@ -10,7 +10,7 @@ DEFS += $(GRN_DEFS)
 
 AM_LDFLAGS = -no-undefined
 
-DEFAULT_INCLUDES = -I$(top_builddir) -I$(srcdir) -I$(top_srcdir)/include $(GROONGA_INCLUDEDIR)
+DEFAULT_INCLUDES = -I$(top_builddir) -I$(srcdir) -I$(top_srcdir)/include $(GROONGA_INCLUDEDIR) -I$(top_srcdir)
 
 groonga_SOURCES = groonga.c
 nodist_EXTRA_groonga_SOURCES = $(NONEXISTENT_CXX_SOURCE)
