--- origsrc/groonga-1.2.7/Makefile.am	2011-10-25 16:52:21.234295600 +0900
+++ src/groonga-1.2.7/Makefile.am	2011-10-29 21:13:59.610673800 +0900
@@ -2,7 +2,6 @@
 
 LOCALES = ja
 
-ACLOCAL_AMFLAGS = $$ACLOCAL_ARGS -I .
 AUTOMAKE_OPTIONS = 1.9.6
 SUBDIRS =					\
 	include					\
--- origsrc/groonga-1.2.7/lib/ctx.c	2011-09-16 11:03:07.306290700 +0900
+++ src/groonga-1.2.7/lib/ctx.c	2011-10-29 21:13:59.620688200 +0900
@@ -2540,6 +2540,9 @@ grn_set_segv_handler(void)
 
   sigemptyset(&action.sa_mask);
   action.sa_sigaction = segv_handler;
+#if !defined(SA_ONSTACK)
+#define SA_ONSTACK 0
+#endif
   action.sa_flags = SA_SIGINFO | SA_ONSTACK;
 
   if (sigaction(SIGSEGV, &action, NULL)) {
--- origsrc/groonga-1.2.7/src/Makefile.am	2011-03-28 17:49:28.920513900 +0900
+++ src/groonga-1.2.7/src/Makefile.am	2011-10-29 21:13:59.620688200 +0900
@@ -8,7 +8,7 @@ DEFS += $(GRN_DEFS)
 
 AM_LDFLAGS = -no-undefined
 
-DEFAULT_INCLUDES = -I$(top_builddir) -I$(srcdir) -I$(top_srcdir)/include $(GROONGA_INCLUDEDIR)
+DEFAULT_INCLUDES = -I$(top_builddir) -I$(srcdir) -I$(top_srcdir)/include $(GROONGA_INCLUDEDIR) -I$(top_srcdir)
 
 groonga_SOURCES = groonga.c
 groonga_CFLAGS = $(AM_CFLAGS) $(LIBEDIT_CFLAGS)
