--- origsrc/parallel-netcdf-1.10.0/src/libs/Makefile.am	2018-06-17 03:07:24.000000000 +0900
+++ src/parallel-netcdf-1.10.0/src/libs/Makefile.am	2018-07-09 21:49:38.425720800 +0900
@@ -11,7 +11,7 @@ lib_LTLIBRARIES = libpnetcdf.la
 # Using libpnetcdf_la_LDFLAGS = $(ABIVERSIONFLAGS) does NOT seem to work,
 # because libpnetcdf_la_LINK is explicitly set below to deal with cross
 # language link.
-AM_LDFLAGS = $(ABIVERSIONFLAGS)
+AM_LDFLAGS = -no-undefined $(ABIVERSIONFLAGS)
 
 libpnetcdf_la_SOURCES =
 libpnetcdf_la_LIBADD  = ../dispatchers/libdispatchers.la
