--- origsrc/rlpvalue/Makefile.am	2020-01-31 02:47:17.000000000 +0900
+++ src/rlpvalue/Makefile.am	2020-01-31 02:49:08.901730000 +0900
@@ -1,9 +1,5 @@
 ACLOCAL_AMFLAGS = -I build-aux/m4
 
-SUBDIRS = univalue
-
-UNIVALUE_CFLAGS = -I$(top_srcdir)/univalue/include
-
 include_HEADERS = include/rlpvalue.h
 noinst_HEADERS = lib/rlpvalue_utffilter.h
 
@@ -32,9 +28,9 @@ src_rlp_SOURCES = \
 	src/tool.cpp \
 	src/rlp2json.cpp \
 	test/utilstrencodings.cpp
-src_rlp_LDADD = librlpvalue.la univalue/.libs/libunivalue.a $(ARGP_LIBS)
+src_rlp_LDADD = librlpvalue.la $(UNIVALUE_LIBS) $(ARGP_LIBS)
 src_rlp_CXXFLAGS = -I$(top_srcdir)/include $(UNIVALUE_CFLAGS)
-src_rlp_LDFLAGS = -static $(LIBTOOL_APP_LDFLAGS)
+src_rlp_LDFLAGS = $(LIBTOOL_APP_LDFLAGS)
 
 TEST_DATA_DIR=test/data
 
@@ -42,15 +38,15 @@ test_unitester_SOURCES = \
 	src/rlp2json.cpp \
 	test/unitester.cpp \
 	test/utilstrencodings.h test/utilstrencodings.cpp
-test_unitester_LDADD = librlpvalue.la univalue/.libs/libunivalue.a
+test_unitester_LDADD = librlpvalue.la $(UNIVALUE_LIBS)
 test_unitester_CXXFLAGS = -I$(top_srcdir)/include $(UNIVALUE_CFLAGS) \
 	-DJSON_TEST_SRC=\"$(srcdir)/$(TEST_DATA_DIR)\"
-test_unitester_LDFLAGS = -static $(LIBTOOL_APP_LDFLAGS)
+test_unitester_LDFLAGS = $(LIBTOOL_APP_LDFLAGS)
 
 test_object_SOURCES = test/object.cpp
 test_object_LDADD = librlpvalue.la
 test_object_CXXFLAGS = -I$(top_srcdir)/include
-test_object_LDFLAGS = -static $(LIBTOOL_APP_LDFLAGS)
+test_object_LDFLAGS = $(LIBTOOL_APP_LDFLAGS)
 
 TEST_FILES = \
 	test/data/example.json \
--- origsrc/rlpvalue/configure.ac	2020-01-31 02:47:17.000000000 +0900
+++ src/rlpvalue/configure.ac	2020-01-31 02:52:42.705669500 +0900
@@ -14,7 +14,7 @@ m4_define([librlpvalue_age], [m4_eval(li
 m4_define([librlpvalue_version], [librlpvalue_major_version().librlpvalue_minor_version().librlpvalue_micro_version()librlpvalue_extraversion()])
 
 
-AC_INIT([rlpvalue], [0.0.1],
+AC_INIT([rlpvalue], [librlpvalue_version],
         [http://github.com/jgarzik/rlpvalue/])
 
 dnl make the compilation flags quiet unless V=1 is used
@@ -25,7 +25,6 @@ AC_CONFIG_SRCDIR([lib/rlpvalue.cpp])
 AC_CONFIG_AUX_DIR([build-aux])
 AC_CONFIG_MACRO_DIR([build-aux/m4])
 AC_CONFIG_HEADERS([rlpvalue-config.h])
-AX_SUBDIRS_CONFIGURE([univalue], [--disable-shared])
 AM_INIT_AUTOMAKE([subdir-objects foreign])
 
 dnl Require C++11 compiler (no GNU extensions)
@@ -67,6 +66,8 @@ case $build in
   ;;
 esac
 
+PKG_CHECK_MODULES([UNIVALUE], [libunivalue])
+
 AC_CONFIG_FILES([
     Makefile
     pc/librlpvalue.pc
--- origsrc/rlpvalue/src/InfInt.h	2020-01-31 02:47:17.000000000 +0900
+++ src/rlpvalue/src/InfInt.h	2020-01-31 02:58:06.258805000 +0900
@@ -40,6 +40,7 @@
 #include <sstream>
 #include <iomanip>
 #include <climits>
+#include <limits>
 
 //#include <limits.h>
 //#include <stdlib.h>
