--- origsrc/SIROCCO2-2.1.0/Makefile.am	2021-06-30 18:57:14.000000000 +0900
+++ src/SIROCCO2-2.1.0/Makefile.am	2021-11-27 11:47:33.723155700 +0900
@@ -5,3 +5,6 @@ dist-hook:
 	cp ZVK.py $(distdir)
 	cp sage-sirocco_interface.pyx $(distdir)
 	cp LICENSE $(distdir)
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/SIROCCO2-2.1.0/configure.ac	2021-06-30 18:57:14.000000000 +0900
+++ src/SIROCCO2-2.1.0/configure.ac	2021-11-27 11:53:29.321066100 +0900
@@ -8,5 +8,5 @@ LT_INIT([shared win32-dll])
 AC_CANONICAL_HOST
 AS_CASE([$host_os], [*cygwin* | *mingw*], [SIROCCO_LDFLAGS=-no-undefined])
 AC_SUBST(SIROCCO_LDFLAGS)
-AC_CONFIG_FILES([Makefile lib/Makefile tests/Makefile])
+AC_CONFIG_FILES([libsirocco.pc Makefile lib/Makefile tests/Makefile])
 AC_OUTPUT
--- origsrc/SIROCCO2-2.1.0/libsirocco.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/SIROCCO2-2.1.0/libsirocco.pc.in	2021-11-27 11:47:33.633093300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: SIROCCO2
+Description: ROot Certified COntinuator
+Version: @VERSION@
+Libs: -L${libdir} -lSIROCCO2
+Cflags: -I${includedir}
