--- origsrc/libproccpuinfo-0.0.8/CMakeLists.txt	2009-04-04 10:10:21.000000000 +0900
+++ src/libproccpuinfo-0.0.8/CMakeLists.txt	2014-06-09 21:16:04.079414800 +0900
@@ -33,14 +33,14 @@ ENABLE_TESTING()
 FIND_PROGRAM(FLEX_EXECUTABLE NAMES flex PATHS /usr/bin DOC "Flex executable")
 IF (FLEX_EXECUTABLE)
   MESSAGE(STATUS "Found flex: ${FLEX_EXECUTABLE}")
-  ADD_CUSTOM_COMMAND(OUTPUT lex.yy.c DEPENDS scanner.l COMMAND ${FLEX_EXECUTABLE} scanner.l)
+  ADD_CUSTOM_COMMAND(OUTPUT lex.yy.c DEPENDS ${libproccpuinfo_SOURCE_DIR}/scanner.l COMMAND ${FLEX_EXECUTABLE} ${libproccpuinfo_SOURCE_DIR}/scanner.l)
 ELSE (FLEX_EXECUTABLE)
   MESSAGE (FATAL_ERROR "flex not found")
 ENDIF (FLEX_EXECUTABLE)
 
 ADD_LIBRARY(proccpuinfo SHARED libproccpuinfo.c cputype.c lex.yy.c)
 SET_TARGET_PROPERTIES(proccpuinfo PROPERTIES SOVERSION "${SOVERSION}")
-INSTALL(TARGETS proccpuinfo DESTINATION lib)
+INSTALL(TARGETS proccpuinfo RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 INSTALL(FILES proccpuinfo.h DESTINATION include)
 
 ADD_SUBDIRECTORY(examples)
