--- origsrc/cmark-0.20.0/src/CMakeLists.txt	2015-06-09 04:53:31.000000000 +0900
+++ src/cmark-0.20.0/src/CMakeLists.txt	2015-06-20 13:14:00.570682700 +0900
@@ -115,7 +115,7 @@ set(CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_NO
 include (InstallRequiredSystemLibraries)
 install(TARGETS ${PROGRAM} ${LIBRARY}
   RUNTIME DESTINATION bin
-  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib
   )
 
 install(FILES
