--- origsrc/cmark-0.28.3/src/CMakeLists.txt	2017-10-22 01:06:00.000000000 +0900
+++ src/cmark-0.28.3/src/CMakeLists.txt	2017-11-11 20:24:17.879000000 +0900
@@ -148,7 +148,7 @@ if(CMARK_SHARED OR CMARK_STATIC)
     DESTINATION include
     )
 
-  install(EXPORT cmark DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake)
+  install(EXPORT cmark DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/cmark)
 endif()
 
 # Feature tests
