diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/cmark-0.19.0/src/CMakeLists.txt src/cmark-0.19.0/src/CMakeLists.txt
--- origsrc/cmark-0.19.0/src/CMakeLists.txt	2015-04-29 13:24:05.000000000 +0900
+++ src/cmark-0.19.0/src/CMakeLists.txt	2015-05-31 09:20:06.077424400 +0900
@@ -115,7 +115,7 @@ set(CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_NO
 include (InstallRequiredSystemLibraries)
 install(TARGETS ${PROGRAM} ${LIBRARY}
   RUNTIME DESTINATION bin
-  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib
   )
 
 install(FILES
