diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/cmark-0.24.1/src/CMakeLists.txt src/cmark-0.24.1/src/CMakeLists.txt
--- origsrc/cmark-0.24.1/src/CMakeLists.txt	2016-01-18 14:53:29.000000000 +0900
+++ src/cmark-0.24.1/src/CMakeLists.txt	2016-01-25 19:52:15.065299000 +0900
@@ -1,6 +1,9 @@
 cmake_minimum_required(VERSION 2.8)
 set(LIBRARY "libcmark")
+option(ENABLE_STATIC "enable static" OFF)
+if (ENABLE_STATIC)
 set(STATICLIBRARY "libcmark_static")
+endif ()
 set(HEADERS
   cmark.h
   parser.h
@@ -89,12 +92,15 @@ elseif(CMAKE_COMPILER_IS_GNUCC OR ${CMAK
 endif ()
 
 add_library(${LIBRARY} SHARED ${LIBRARY_SOURCES})
+if (ENABLE_STATIC)
 add_library(${STATICLIBRARY} STATIC ${LIBRARY_SOURCES})
+endif ()
 # Include minor version and patch level in soname for now.
 set_target_properties(${LIBRARY} PROPERTIES
   OUTPUT_NAME "cmark"
   SOVERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}
   VERSION ${PROJECT_VERSION})
+if (ENABLE_STATIC)
 set_target_properties(${STATICLIBRARY} PROPERTIES
   COMPILE_FLAGS -DCMARK_STATIC_DEFINE
   POSITION_INDEPENDENT_CODE ON)
@@ -108,6 +114,7 @@ else()
     OUTPUT_NAME "cmark"
     VERSION ${PROJECT_VERSION})
 endif(MSVC)
+endif()
 
 set_property(TARGET ${LIBRARY}
   APPEND PROPERTY MACOSX_RPATH true)
