--- origsrc/cmark-0.27.1/src/CMakeLists.txt	2016-11-19 17:44:32.000000000 +0900
+++ src/cmark-0.27.1/src/CMakeLists.txt	2016-12-02 19:25:08.027787500 +0900
@@ -4,7 +4,10 @@ if(${CMAKE_VERSION} VERSION_GREATER "3.3
 endif()
 
 set(LIBRARY "libcmark")
+option(ENABLE_STATIC "enable static" OFF)
+if (ENABLE_STATIC)
 set(STATICLIBRARY "libcmark_static")
+endif ()
 set(HEADERS
   cmark.h
   parser.h
@@ -93,12 +96,15 @@ elseif(CMAKE_COMPILER_IS_GNUCC OR ${CMAK
 endif ()
 
 add_library(${LIBRARY} SHARED ${LIBRARY_SOURCES})
+if (ENABLE_STATIC)
 add_library(${STATICLIBRARY} STATIC ${LIBRARY_SOURCES})
+endif ()
 # Include minor version and patch level in soname for now.
 set_target_properties(${LIBRARY} PROPERTIES
   OUTPUT_NAME "cmark"
   SOVERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}
   VERSION ${PROJECT_VERSION})
+if (ENABLE_STATIC)
 set_target_properties(${STATICLIBRARY} PROPERTIES
   COMPILE_FLAGS -DCMARK_STATIC_DEFINE
   POSITION_INDEPENDENT_CODE ON)
@@ -112,6 +118,7 @@ else()
     OUTPUT_NAME "cmark"
     VERSION ${PROJECT_VERSION})
 endif(MSVC)
+endif()
 
 set_property(TARGET ${LIBRARY}
   APPEND PROPERTY MACOSX_RPATH true)
