--- origsrc/trexio-2.6.0/src/CMakeLists.txt	2025-08-02 21:28:27.000000000 +0900
+++ src/trexio-2.6.0/src/CMakeLists.txt	2025-12-22 18:45:03.960823900 +0900
@@ -58,6 +58,10 @@ set_target_properties(trexio PROPERTIES
 	PUBLIC_HEADER ${TREXIO_PUBLIC_HEADERS}
 	)
 
+if (DEFINED SOVERSION)
+    set_target_properties(trexio PROPERTIES SOVERSION ${SOVERSION})
+endif()
+
 # ============= CONFIGURE HDF5 ==============
 
 # By defaulti, TREXIO is configured with the HDF5 support.
@@ -118,6 +122,10 @@ add_library(trexio_f SHARED)
 target_sources(trexio_f PUBLIC ${TREXIO_MOD_FILE})
 target_link_libraries(trexio_f PUBLIC trexio)
 
+if (DEFINED SOVERSION)
+    set_target_properties(trexio_f PROPERTIES SOVERSION ${SOVERSION})
+endif()
+
 # ====== CODE GENERATION FOR DEVEL MODE =====
 
 add_custom_command(
@@ -163,7 +171,7 @@ endif()
 include(GNUInstallDirs)
 # Use standard GNU directories for installation of TREXIO (e.g. /usr/local/lib|include).
 # The installation prefix can be modified using -DCMAKE_INSTALL_PREFIX=<custom/path> option of cmake.
-install(TARGETS trexio
+install(TARGETS trexio trexio_f
 	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
 	PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 	)
