--- origsrc/cppzmq-4.10.0/cppzmq.pc.in	2023-06-20 19:05:35.000000000 +0900
+++ src/cppzmq-4.10.0/cppzmq.pc.in	2023-08-25 23:45:59.091768800 +0900
@@ -1,9 +1,9 @@
-prefix="@CMAKE_INSTALL_PREFIX@"
-includedir="@CMAKE_INSTALL_FULL_INCLUDEDIR@"
+prefix=@CMAKE_INSTALL_PREFIX@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
 
 Name: @PROJECT_NAME@
 Description: C++ binding for libzmq
 URL: https://github.com/zeromq/cppzmq
 Version: @PROJECT_VERSION@
 Requires: libzmq
-Cflags: -I"${includedir}" @pkg_config_defines@
+Cflags: -I${includedir} @pkg_config_defines@
