--- origsrc/cppzmq-4.3.0/CMakeLists.txt	2018-07-15 20:06:19.000000000 +0900
+++ src/cppzmq-4.3.0/CMakeLists.txt	2019-05-26 21:28:20.672757400 +0900
@@ -8,6 +8,11 @@ project(cppzmq VERSION ${DETECTED_CPPZMQ
 
 find_package(ZeroMQ QUIET)
 
+if(NOT ZeroMQ_FOUND)
+    include(FindPkgConfig)
+    pkg_check_modules(ZeroMQ libzmq)
+endif()
+
 # libzmq autotools install: fallback to pkg-config
 if(NOT ZeroMQ_FOUND)
     message(STATUS "CMake libzmq package not found, trying again with pkg-config (normal install of zeromq)")
@@ -20,7 +25,7 @@ if(NOT ZeroMQ_FOUND)
     message(FATAL_ERROR "ZeroMQ was not found, neither as a CMake package nor via pkg-config")
 endif()
 
-if (ZeroMQ_FOUND AND (NOT TARGET libzmq OR NOT TARGET libzmq-static))
+if ((NOT CYGWIN) AND ZeroMQ_FOUND AND (NOT TARGET libzmq OR NOT TARGET libzmq-static))
   message(FATAL_ERROR "ZeroMQ version not supported!")
 endif()
 
@@ -43,19 +48,18 @@ set(CPPZMQ_HEADERS
     zmq_addon.hpp
 )
 
-foreach (target cppzmq cppzmq-static)
+foreach (target cppzmq)
   add_library(${target} INTERFACE)
   target_include_directories(${target} INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
                                                  $<INSTALL_INTERFACE:include>)
 endforeach()
 
 target_link_libraries(cppzmq INTERFACE libzmq)
-target_link_libraries(cppzmq-static INTERFACE libzmq-static)
 
 include(GNUInstallDirs)
 include(CMakePackageConfigHelpers)
 
-install(TARGETS cppzmq cppzmq-static
+install(TARGETS cppzmq
         EXPORT ${PROJECT_NAME}-targets)
 
 install(FILES ${CPPZMQ_HEADERS}
--- origsrc/cppzmq-4.3.0/cppzmqConfig.cmake.in	2018-07-15 20:06:19.000000000 +0900
+++ src/cppzmq-4.3.0/cppzmqConfig.cmake.in	2019-05-26 21:29:26.124082200 +0900
@@ -12,7 +12,6 @@
 # cppzmq_FOUND - true if cppzmq found on the system
 # cppzmq_INCLUDE_DIR - the directory containing cppzmq headers
 # cppzmq_LIBRARY - the ZeroMQ library for dynamic linking
-# cppzmq_STATIC_LIBRARY - the ZeroMQ library for static linking
 
 @PACKAGE_INIT@
 
--- origsrc/cppzmq-4.3.0/libzmq-pkg-config/FindZeroMQ.cmake	2018-07-15 20:06:19.000000000 +0900
+++ src/cppzmq-4.3.0/libzmq-pkg-config/FindZeroMQ.cmake	2019-05-26 21:29:09.647787400 +0900
@@ -3,12 +3,10 @@ find_package(PkgConfig)
 pkg_check_modules(PC_LIBZMQ QUIET libzmq)
 
 set(ZeroMQ_VERSION ${PC_LIBZMQ_VERSION})
-find_library(ZeroMQ_LIBRARY NAMES libzmq.so libzmq.dylib libzmq.dll
-             PATHS ${PC_LIBZMQ_LIBDIR} ${PC_LIBZMQ_LIBRARY_DIRS})
-find_library(ZeroMQ_STATIC_LIBRARY NAMES libzmq-static.a libzmq.a libzmq.dll.a
+find_library(ZeroMQ_LIBRARY NAMES zmq libzmq.so libzmq.dylib libzmq.dll
              PATHS ${PC_LIBZMQ_LIBDIR} ${PC_LIBZMQ_LIBRARY_DIRS})
 
-if(ZeroMQ_LIBRARY AND ZeroMQ_STATIC_LIBRARY)
+if(ZeroMQ_LIBRARY)
     set(ZeroMQ_FOUND ON)
 endif()
 
@@ -20,7 +18,3 @@ endif()
 add_library(libzmq SHARED IMPORTED)
 set_property(TARGET libzmq PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${PC_LIBZMQ_INCLUDE_DIRS})
 set_property(TARGET libzmq PROPERTY IMPORTED_LOCATION ${ZeroMQ_LIBRARY})
-
-add_library(libzmq-static STATIC IMPORTED ${PC_LIBZMQ_INCLUDE_DIRS})
-set_property(TARGET libzmq-static PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${PC_LIBZMQ_INCLUDE_DIRS})
-set_property(TARGET libzmq-static PROPERTY IMPORTED_LOCATION ${ZeroMQ_STATIC_LIBRARY})
