--- origsrc/zopfli-zopfli-1.0.1/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/zopfli-zopfli-1.0.1/Makefile.am	2016-03-11 16:58:35.838081300 +0900
@@ -0,0 +1,44 @@
+
+AM_CFLAGS = -W -Wall -Wextra -ansi -pedantic -lm -O3 -std=gnu11
+AM_CXXFLAGS = -W -Wall -Wextra -ansi -pedantic -O3 -std=gnu++11
+
+zopfliincludedir = $(includedir)/zopfli
+zopfliinclude_HEADERS = \
+src/zopfli/blocksplitter.h   src/zopfli/cache.h \
+src/zopfli/deflate.h         src/zopfli/gzip_container.h \
+src/zopfli/hash.h            src/zopfli/katajainen.h \
+src/zopfli/lz77.h            src/zopfli/squeeze.h \
+src/zopfli/tree.h            src/zopfli/util.h \
+src/zopfli/zlib_container.h  src/zopfli/zopfli.h
+zopflipngincludedir = $(includedir)/zopflipng
+zopflipnginclude_HEADERS = src/zopflipng/zopflipng_lib.h
+zopflilodepngincludedir = $(includedir)/zopflipng/lodepng
+zopflilodepnginclude_HEADERS = \
+src/zopflipng/lodepng/lodepng.h src/zopflipng/lodepng/lodepng_util.h
+
+bin_PROGRAMS = zopfli zopflipng
+lib_LTLIBRARIES = libzopfli.la libzopflipng.la
+
+ZOPFLILIB_SRC = src/zopfli/blocksplitter.c src/zopfli/cache.c\
+                src/zopfli/deflate.c src/zopfli/gzip_container.c\
+                src/zopfli/hash.c src/zopfli/katajainen.c\
+                src/zopfli/lz77.c src/zopfli/squeeze.c\
+                src/zopfli/tree.c src/zopfli/util.c\
+                src/zopfli/zlib_container.c src/zopfli/zopfli_lib.c
+libzopfli_la_SOURCES = $(ZOPFLILIB_SRC)
+libzopfli_la_LDFLAGS = -no-undefined -version-info 1
+
+ZOPFLIBIN_SRC = src/zopfli/zopfli_bin.c
+zopfli_SOURCES = $(ZOPFLIBIN_SRC)
+zopfli_LDADD = libzopfli.la
+
+LODEPNG_SRC = src/zopflipng/lodepng/lodepng.cpp src/zopflipng/lodepng/lodepng_util.cpp
+
+ZOPFLIPNGLIB_SRC = src/zopflipng/zopflipng_lib.cc
+libzopflipng_la_SOURCES = $(ZOPFLIPNGLIB_SRC) $(LODEPNG_SRC)
+libzopflipng_la_LDFLAGS = -no-undefined -version-info 1
+libzopflipng_la_LIBADD = libzopfli.la
+
+zopflipng_SOURCES = $(ZOPFLIPNGBIN_SRC)
+ZOPFLIPNGBIN_SRC = src/zopflipng/zopflipng_bin.cc
+zopflipng_LDADD = libzopflipng.la
--- origsrc/zopfli-zopfli-1.0.1/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/zopfli-zopfli-1.0.1/configure.ac	2016-03-11 16:45:11.792077900 +0900
@@ -0,0 +1,28 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([zopfli], [1.0.1])
+AC_CONFIG_SRCDIR([src/zopfli/blocksplitter.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+LT_INIT
+
+# Checks for header files.
+AC_CHECK_HEADERS([fcntl.h stddef.h stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([memset])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
