--- origsrc/espresso-3.3.0/src/acconfig.hpp.in	2014-08-11 23:48:25.000000000 +0900
+++ src/espresso-3.3.0/src/acconfig.hpp.in	2015-03-08 18:30:16.217202000 +0900
@@ -119,8 +119,7 @@
 /* Define to 1 if you have the <unistd.h> header file. */
 #undef HAVE_UNISTD_H
 
-/* Define to the sub-directory in which libtool stores uninstalled libraries.
-   */
+/* Define to the sub-directory where libtool stores uninstalled libraries. */
 #undef LT_OBJDIR
 
 /* Name of package */
--- origsrc/espresso-3.3.0/src/core/Makefile.am	2014-08-11 23:45:26.000000000 +0900
+++ src/espresso-3.3.0/src/core/Makefile.am	2015-03-08 17:26:28.062671900 +0900
@@ -17,6 +17,8 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
+AM_CXXFLAGS = -fpermissive
+
 EXTRA_DIST =
 CLEANFILES =
 
--- origsrc/espresso-3.3.0/src/tcl/Makefile.am	2014-08-11 23:45:26.000000000 +0900
+++ src/espresso-3.3.0/src/tcl/Makefile.am	2015-03-08 18:19:36.141431000 +0900
@@ -155,12 +155,12 @@ endif
 noinst_PROGRAMS = Espresso
 Espresso_CPPFLAGS = -D ESPRESSO_SCRIPTS_DEFAULT=\"$(buildscriptsdir)\" $(AM_CPPFLAGS)
 Espresso_SOURCES = scriptsdir.cpp main.cpp
-Espresso_LDADD = libEspressoTcl.la ../core/libEspresso.la
+Espresso_LDADD = libEspressoTcl.la ../core/libEspresso.la -lmpi_cxx
 
 bin_PROGRAMS = Espresso.install
 Espresso_install_CPPFLAGS = -D ESPRESSO_SCRIPTS_DEFAULT=\"$(scriptsdir)\" $(AM_CPPFLAGS)
 Espresso_install_SOURCES = scriptsdir.cpp main.cpp
-Espresso_install_LDADD = libEspressoTcl.la ../core/libEspresso.la 
+Espresso_install_LDADD = libEspressoTcl.la ../core/libEspresso.la -lmpi_cxx
 
 ESPRESSO = `echo Espresso | sed '$(transform)'`$(EXEEXT)
 ESPRESSO_INSTALL = `echo Espresso.install | sed '$(transform)'`$(EXEEXT)
