--- origsrc/Hamlib-4.6.5/bindings/Makefile.am	2025-09-06 04:49:48.000000000 +0900
+++ src/Hamlib-4.6.5/bindings/Makefile.am	2026-01-10 13:05:43.002386200 +0900
@@ -209,7 +209,7 @@ nodist_Hamliblua_la_SOURCES = hamliblua_
 
 Hamliblua_la_CPPFLAGS = $(LUA_INCLUDE) -I$(top_srcdir)/include -I$(top_srcdir)/src
 Hamliblua_la_LDFLAGS = -no-undefined -module -avoid-version
-Hamliblua_la_LIBADD = $(top_builddir)/src/libhamlib.la
+Hamliblua_la_LIBADD = $(top_builddir)/src/libhamlib.la $(LUA_LIB)
 Hamliblua_ladir = $(luaexecdir)
 Hamliblua_la_LTLIBRARIES = $(luaexec_ltlib)
 
