--- origsrc/Hamlib-4.4/bindings/Makefile.am	2021-12-03 08:46:51.000000000 +0900
+++ src/Hamlib-4.4/bindings/Makefile.am	2021-12-29 10:24:06.317203500 +0900
@@ -205,7 +205,7 @@ nodist_Hamliblua_la_SOURCES = hamliblua_
 
 Hamliblua_la_CPPFLAGS = $(LUA_INCLUDE) -I$(top_srcdir)/include -I$(top_srcdir)/src
 Hamliblua_la_LDFLAGS = -no-undefined -module -avoid-version
-Hamliblua_la_LIBADD = $(top_builddir)/src/libhamlib.la
+Hamliblua_la_LIBADD = $(top_builddir)/src/libhamlib.la $(LUA_LIB)
 Hamliblua_ladir = $(luaexecdir)
 Hamliblua_la_LTLIBRARIES = $(luaexec_ltlib)
 
--- origsrc/Hamlib-4.4/src/hamlibdatetime.h	1970-01-01 09:00:00.000000000 +0900
+++ src/Hamlib-4.4/src/hamlibdatetime.h	2021-12-29 10:17:11.271226600 +0900
@@ -0,0 +1,4 @@
+/* This is a placeholder that will be overwritten by 'make' when
+ * invoked in a Git working tree.
+ */
+ #define HAMLIBDATETIME "from indeterminate source revision."
