--- origsrc/Hamlib-3.3/bindings/Makefile.am	2018-08-21 00:08:55.000000000 +0900
+++ src/Hamlib-3.3/bindings/Makefile.am	2019-01-07 06:28:18.738728500 +0900
@@ -202,7 +202,7 @@ nodist_Hamliblua_la_SOURCES = hamliblua_
 
 Hamliblua_la_CPPFLAGS = $(LUA_INCLUDE) -I$(top_srcdir)/include -I$(top_srcdir)/src
 Hamliblua_la_LDFLAGS = -no-undefined -module -avoid-version
-Hamliblua_la_LIBADD = $(top_builddir)/src/libhamlib.la
+Hamliblua_la_LIBADD = $(top_builddir)/src/libhamlib.la $(LUA_LIB)
 Hamliblua_ladir = $(luaexecdir)
 Hamliblua_la_LTLIBRARIES = $(luaexec_ltlib)
 
--- origsrc/Hamlib-3.3/configure.ac	2018-08-21 00:08:55.000000000 +0900
+++ src/Hamlib-3.3/configure.ac	2019-01-07 06:28:18.747715400 +0900
@@ -281,7 +281,7 @@ AS_CASE(["$host_os"],
         OSXLDFLAGS="-Wl,-headerpad_max_install_names"
         AC_SUBST([OSXLDFLAGS])],
 
-    [mingw* | pw32* | cygwin*], [
+    [mingw* | pw32*], [
         WINLDFLAGS="-Wl,--output-def,libhamlib.def -Wl,--add-stdcall-alias"
         AC_SUBST([WINLDFLAGS])
 
@@ -657,7 +657,7 @@ dnl Check for libdl and set DL_LIBS if f
 
 dnl Set conditional compilation for G-313.
 AS_CASE(["$host_os"],
-        [mingw* | pw32* | cygwin ], [
+        [mingw* | pw32*], [
              AM_CONDITIONAL(G313_LINUX_GNU, false)
              AM_CONDITIONAL(G313_WINDOWS, true)
         ],
