--- origsrc/honggfuzz-2.2/Makefile	2020-04-24 06:47:14.000000000 +0900
+++ src/honggfuzz-2.2/Makefile	2020-07-10 15:05:38.008356700 +0900
@@ -26,7 +26,7 @@ LD = $(CC)
 BIN := honggfuzz
 HFUZZ_CC_BIN := hfuzz_cc/hfuzz-cc
 HFUZZ_CC_SRCS := hfuzz_cc/hfuzz-cc.c
-COMMON_CFLAGS := -std=c11 -I/usr/local/include -D_GNU_SOURCE -Wall -Wextra -Werror -Wno-format-truncation -Wno-override-init -I.
+COMMON_CFLAGS := -std=c11 -I/usr/local/include -D_GNU_SOURCE -Wall -Wextra -Wno-format-truncation -Wno-override-init -I.
 COMMON_LDFLAGS := -pthread -lm
 COMMON_SRCS := $(sort $(wildcard *.c))
 CFLAGS ?= -O3 -mtune=native -funroll-loops
@@ -363,16 +363,16 @@ BIN_PATH	= $(PREFIX)/bin
 INC_PATH	= $(PREFIX)/include
 
 install: all
-	mkdir -p -m 755 $${DESTDIR}$(BIN_PATH)
-	install -m 755 honggfuzz $${DESTDIR}$(BIN_PATH)
-	install -m 755 hfuzz_cc/hfuzz-cc $${DESTDIR}$(BIN_PATH)
-	install -m 755 hfuzz_cc/hfuzz-clang $${DESTDIR}$(BIN_PATH)
-	install -m 755 hfuzz_cc/hfuzz-clang++ $${DESTDIR}$(BIN_PATH)
-	install -m 755 hfuzz_cc/hfuzz-gcc $${DESTDIR}$(BIN_PATH)
-	install -m 755 hfuzz_cc/hfuzz-g++ $${DESTDIR}$(BIN_PATH)
-	install -m 755 -t $${DESTDIR}$(INC_PATH)/libhfcommon -D includes/libhfcommon/*.h
-	install -m 755 -t $${DESTDIR}$(INC_PATH)/libhfuzz -D includes/libhfuzz/*.h
-	install -m 755 -t $${DESTDIR}$(INC_PATH)/libhnetdriver -D includes/libhfnetdriver/*.h
+	mkdir -p -m 755 ${DESTDIR}$(BIN_PATH)
+	install -m 755 honggfuzz ${DESTDIR}$(BIN_PATH)
+	install -m 755 hfuzz_cc/hfuzz-cc ${DESTDIR}$(BIN_PATH)
+	install -m 755 hfuzz_cc/hfuzz-clang ${DESTDIR}$(BIN_PATH)
+	install -m 755 hfuzz_cc/hfuzz-clang++ ${DESTDIR}$(BIN_PATH)
+	install -m 755 hfuzz_cc/hfuzz-gcc ${DESTDIR}$(BIN_PATH)
+	install -m 755 hfuzz_cc/hfuzz-g++ ${DESTDIR}$(BIN_PATH)
+	install -m 755 -t ${DESTDIR}$(INC_PATH)/libhfcommon -D includes/libhfcommon/*.h
+	install -m 755 -t ${DESTDIR}$(INC_PATH)/libhfuzz -D includes/libhfuzz/*.h
+	install -m 755 -t ${DESTDIR}$(INC_PATH)/libhnetdriver -D includes/libhfnetdriver/*.h
 
 # DO NOT DELETE
 
