--- origsrc/honggfuzz-0.8/Makefile	2016-09-15 00:01:17.000000000 +0900
+++ src/honggfuzz-0.8/Makefile	2016-12-30 22:28:55.075761600 +0900
@@ -22,7 +22,7 @@
 CC ?= gcc
 LD = $(CC)
 BIN := honggfuzz
-COMMON_CFLAGS := -D_GNU_SOURCE -Wall -Werror -Wframe-larger-than=131072
+COMMON_CFLAGS := -D_GNU_SOURCE -Wall -Wframe-larger-than=131072
 COMMON_LDFLAGS := -lm
 COMMON_SRCS := honggfuzz.c cmdline.c display.c files.c fuzz.c log.c mangle.c report.c sancov.c subproc.c util.c
 CFLAGS ?= -O3
