--- origsrc/enchive-3.5/Makefile	2019-12-02 03:23:16.000000000 +0900
+++ src/enchive-3.5/Makefile	2020-02-20 20:08:37.530316200 +0900
@@ -1,10 +1,10 @@
 .POSIX:
 .SUFFIXES:
 CC     = cc
-CFLAGS = -ansi -pedantic -Wall -Wextra -O3 -g3
-PREFIX = /usr/local
-
-sources = src/enchive.c src/chacha.c src/curve25519-donna.c src/sha256.c
+CFLAGS = -ansi -pedantic -Wall -Wextra -O3
+PREFIX = /usr
+LDLIBS = -lcurve25519-donna
+sources = src/enchive.c src/chacha.c src/sha256.c
 objects = $(sources:.c=.o)
 headers = config.h src/docs.h src/chacha.h src/sha256.h src/optparse.h
 
