--- origsrc/enchive-3.5/Makefile	2019-12-02 03:23:16.000000000 +0900
+++ src/enchive-3.5/Makefile	2026-03-29 08:18:45.562504700 +0900
@@ -1,10 +1,10 @@
 .POSIX:
 .SUFFIXES:
-CC     = cc
-CFLAGS = -ansi -pedantic -Wall -Wextra -O3 -g3
-PREFIX = /usr/local
+CC     ?= cc
+CFLAGS ?= -ansi -pedantic -Wall -Wextra -O3 -g3
+PREFIX ?= /usr/local
 
-sources = src/enchive.c src/chacha.c src/curve25519-donna.c src/sha256.c
+sources = src/enchive.c src/chacha.c src/sha256.c
 objects = $(sources:.c=.o)
 headers = config.h src/docs.h src/chacha.h src/sha256.h src/optparse.h
 
@@ -12,7 +12,6 @@ enchive: $(objects)
 	$(CC) $(LDFLAGS) -o $@ $(objects) $(LDLIBS)
 src/enchive.o: src/enchive.c config.h src/docs.h
 src/chacha.o: src/chacha.c config.h
-src/curve25519-donna.o: src/curve25519-donna.c config.h
 src/sha256.o: src/sha256.c config.h
 
 enchive-cli.c: $(sources) $(headers)
