--- origsrc/lib765-0.4.2/configure.in	2009-01-13 08:03:03.000000000 +0900
+++ src/lib765-0.4.2/configure.in	2010-10-17 11:52:46.440526600 +0900
@@ -32,7 +32,7 @@ dnl Checks for programs.
 AC_PROG_INSTALL
 AC_PROG_RANLIB
 AC_PROG_MAKE_SET
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 dnl GCC extra options
 if test "$ac_cv_prog_gcc" = "yes"; then
--- origsrc/lib765-0.4.2/lib/Makefile.am	2007-06-12 05:49:40.000000000 +0900
+++ src/lib765-0.4.2/lib/Makefile.am	2010-10-17 11:52:10.768651600 +0900
@@ -12,6 +12,6 @@ lib_LTLIBRARIES=lib765.la
 # If interfaces have been added increment the right-hand number.
 # If interfaces have been removed set the right-hand number to 0.
 #
-lib765_la_LDFLAGS = -version-info 4:2:1
+lib765_la_LDFLAGS = -version-info 4:2:1 -no-undefined
 lib765_la_SOURCES = 765drive.c 765dsk.c 765fdc.c 765ldsk.c 765i.h error.c
 
