--- origsrc/lib765-0.4.2/lib/Makefile.am	2007-06-12 05:49:40.000000000 +0900
+++ src/lib765-0.4.2/lib/Makefile.am	2014-04-06 22:52:13.205629900 +0900
@@ -12,6 +12,6 @@ lib_LTLIBRARIES=lib765.la
 # If interfaces have been added increment the right-hand number.
 # If interfaces have been removed set the right-hand number to 0.
 #
-lib765_la_LDFLAGS = -version-info 4:2:1
+lib765_la_LDFLAGS = -no-undefined -version-info 4:2:1
 lib765_la_SOURCES = 765drive.c 765dsk.c 765fdc.c 765ldsk.c 765i.h error.c
 
