--- origsrc/Normaliz-3.9.4/Makefile.am	2022-08-25 22:48:17.000000000 +0900
+++ src/Normaliz-3.9.4/Makefile.am	2022-10-12 16:21:51.509962200 +0900
@@ -62,3 +62,6 @@ dist-hook:
 	cd $(distdir)/PyNormaliz && git checkout $(PYNORMALIZ_COMMIT_SHA)
 	rm -rf $(distdir)/PyNormaliz/.git
 
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/Normaliz-3.9.4/Normaliz.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/Normaliz-3.9.4/Normaliz.pc.in	2022-10-12 16:21:51.514954500 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: Normaliz
+Description: Open source tool for computations in affine monoids, vector configurations, lattice polytopes, and rational cones
+Version: @VERSION@
+Libs: -L${libdir} -lNormaliz -lflint
+Cflags: -I${includedir}
+Requires: gmp gmpxx
--- origsrc/Normaliz-3.9.4/configure.ac	2022-08-25 22:48:17.000000000 +0900
+++ src/Normaliz-3.9.4/configure.ac	2022-10-12 16:21:51.519009400 +0900
@@ -477,7 +477,7 @@ AS_IF([test x$enable_shared = xyes],
 dnl
 dnl Output everything
 dnl
-AC_CONFIG_FILES([
+AC_CONFIG_FILES([Normaliz.pc 
     Makefile
 	example/Makefile
 	source/libnormaliz/nmz_config.h
