--- origsrc/libsparql/Makefile.am	2025-04-30 21:33:09.000000000 +0900
+++ src/libsparql/Makefile.am	2025-04-30 21:55:42.154010500 +0900
@@ -1,3 +1,6 @@
-SUBDIRS         = lib src test #doc
+SUBDIRS         = src test #doc
 EXTRA_DIST      = AUTHORS CREDITS README UNLICENSE VERSION
 ACLOCAL_AMFLAGS = -I etc/aclocal
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libsparql/configure.ac	2025-04-30 21:33:09.000000000 +0900
+++ src/libsparql/configure.ac	2025-04-30 21:55:42.169629200 +0900
@@ -110,6 +110,8 @@ AS_IF([test "x$enable_xslt" != "xno"], [
 ])
 AM_CONDITIONAL([HAVE_LIBXQILLA], [test "x$enable_xslt" != "xno" && test "x$ac_res" != "xno"])
 
+PKG_CHECK_MODULES([LIBRFC], [librfc])
+
 dnl Check for header files:
 
 dnl Check for types:
@@ -143,4 +145,5 @@ AC_SUBST([PACKAGE_VERSION_MINOR], ["VERS
 AC_SUBST([PACKAGE_VERSION_PATCH], ["VERSION_PATCH"])
 AC_CONFIG_FILES([src/sparql++/version.h])
 AH_BOTTOM([#include "libcompat.h"])
+AC_CONFIG_FILES([libsparql++.pc])
 AC_OUTPUT
--- origsrc/libsparql/libsparql++.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libsparql/libsparql++.pc.in	2025-04-30 21:56:59.665075400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libsparql++
+Description: SPARQL Protocol and RDF Query Language (SPARQL) library for C++14
+Version: @PACKAGE_VERSION_MAJOR@.@PACKAGE_VERSION_MINOR@.@PACKAGE_VERSION_PATCH@
+Libs: -L${libdir} -lsparql++
+Cflags: -I${includedir}
--- origsrc/libsparql/src/sparql++/Makefile.am	2025-04-30 21:33:09.000000000 +0900
+++ src/libsparql/src/sparql++/Makefile.am	2025-04-30 21:45:45.258163800 +0900
@@ -1,6 +1,8 @@
 lib_LTLIBRARIES = libsparql++.la
 
-libsparql___la_LIBADD = $(top_srcdir)/lib/librfcjson.la
+libsparql___la_CPPFLAGS = -I$(top_builddir)/src -I$(top_srcdir)/lib -I$(top_srcdir)/src
+libsparql___la_LIBADD = $(LIBRFC_LIBS)
+libsparql___la_LDFLAGS = -no-undefined
 
 libsparql___la_SOURCES =  \
   extern.cc               \
@@ -56,7 +58,7 @@ if HAVE_LIBYAJL
 endif
 
 if HAVE_LIBXML2
-  AM_CPPFLAGS += $(libxml2_CFLAGS)
+  libsparql___la_CPPFLAGS += $(libxml2_CFLAGS)
   libsparql___la_LIBADD += $(libxml2_LIBS)
 endif
 
--- origsrc/libsparql/test/Makefile.am	2025-04-30 21:33:09.000000000 +0900
+++ src/libsparql/test/Makefile.am	2025-04-30 21:46:04.757842300 +0900
@@ -1,8 +1,8 @@
-AM_CPPFLAGS += $(TEST_CPPFLAGS)
+AM_CPPFLAGS += $(TEST_CPPFLAGS) -I$(top_builddir)/src
 AM_CXXFLAGS += $(TEST_CXXFLAGS)
 AM_LDFLAGS  += $(TEST_LDFLAGS)
 
-LDADD = $(top_srcdir)/src/sparql++/libsparql++.la
+LDADD = $(top_builddir)/src/sparql++/libsparql++.la
 
 check_PROGRAMS = \
   check_feature  \
