--- origsrc/ginac-1.5.8/configure.ac	2010-07-06 21:39:19.000000000 +0900
+++ src/ginac-1.5.8/configure.ac	2010-11-23 12:55:19.193820100 +0900
@@ -71,7 +71,7 @@ dnl Check for the compiler and all the u
 AC_PROG_CXX
 AC_PROG_CXXCPP
 AC_PROG_INSTALL
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AM_PROG_LEX
 AC_PROG_YACC
 
--- origsrc/ginac-1.5.8/ginac/Makefile.am	2010-07-06 21:39:14.000000000 +0900
+++ src/ginac-1.5.8/ginac/Makefile.am	2010-11-23 12:56:23.801721700 +0900
@@ -55,7 +55,7 @@ polynomial/primpart_content.cpp \
 polynomial/smod_helpers.h \
 polynomial/debug.h
 
-libginac_la_LDFLAGS = -version-info $(LT_VERSION_INFO) -release $(LT_RELEASE)
+libginac_la_LDFLAGS = -version-info $(LT_VERSION_INFO) -release $(LT_RELEASE) -no-undefined
 libginac_la_LIBADD = $(DL_LIBS)
 ginacincludedir = $(includedir)/ginac
 ginacinclude_HEADERS = ginac.h add.h archive.h assertion.h basic.h class_info.h \
--- origsrc/ginac-1.5.8/ginac.pc	2010-07-06 21:44:12.000000000 +0900
+++ src/ginac-1.5.8/ginac.pc	2010-11-23 13:23:30.794075300 +0900
@@ -1,5 +1,5 @@
-prefix=/usr/local
-exec_prefix=${prefix}
+prefix=/usr
+exec_prefix=/usr
 libdir=${exec_prefix}/lib
 includedir=${prefix}/include
 
@@ -7,5 +7,5 @@ Name: GiNaC
 Description: C++ library for symbolic calculations
 Version: 1.5.8
 Requires: cln >= 1.1.6
-Libs: -L${libdir} -lginac -Wl,-rpath -Wl,${libdir}
+Libs: -L${libdir} -lginac
 Cflags: -I${includedir}
