--- origsrc/ginac-1.7.11/ginac/Makefile.am	2020-08-04 03:03:25.000000000 +0900
+++ src/ginac-1.7.11/ginac/Makefile.am	2020-08-26 16:49:07.871465700 +0900
@@ -57,7 +57,7 @@ polynomial/primpart_content.cpp \
 polynomial/smod_helpers.h \
 polynomial/debug.h
 
-libginac_la_LDFLAGS = -version-info $(LT_VERSION_INFO)
+libginac_la_LDFLAGS = -no-undefined -version-info $(LT_VERSION_INFO)
 libginac_la_CPPFLAGS = -DLIBEXECDIR='"$(libexecdir)/"'
 libginac_la_LIBADD = $(DL_LIBS)
 ginacincludedir = $(includedir)/ginac
@@ -80,4 +80,3 @@ $(srcdir)/function.cpp: function.cppy fu
 
 $(srcdir)/function.h: function.hppy function.py
 	$(PYTHON) $(srcdir)/function.py -o $@ $<
-
