--- origsrc/ginac-1.8.10/ginac/Makefile.am	2026-01-29 23:00:59.000000000 +0000
+++ src/ginac-1.8.10/ginac/Makefile.am	2026-08-11 09:50:35.086941400 +0000
@@ -58,7 +58,7 @@ polynomial/primpart_content.cpp \
 polynomial/smod_helpers.h \
 polynomial/debug.h
 
-libginac_la_LDFLAGS = -version-info $(LT_VERSION_INFO)
+libginac_la_LDFLAGS = -no-undefined -version-info $(LT_VERSION_INFO)
 libginac_la_CPPFLAGS = -DLIBEXECDIR='"$(libexecdir)/"'
 libginac_la_LIBADD = $(DL_LIBS)
 ginacincludedir = $(includedir)/ginac
@@ -81,4 +81,3 @@ $(srcdir)/function.cpp: function.cppy fu
 
 $(srcdir)/function.h: function.hppy function.py
 	$(PYTHON) $(srcdir)/function.py -o $@ $<
-
