--- origsrc/ginac-1.6.3/ginac/Makefile.am	2014-02-03 04:14:52.000000000 +0900
+++ src/ginac-1.6.3/ginac/Makefile.am	2015-02-19 08:14:52.061046900 +0900
@@ -55,7 +55,7 @@ polynomial/primpart_content.cpp \
 polynomial/smod_helpers.h \
 polynomial/debug.h
 
-libginac_la_LDFLAGS = -version-info $(LT_VERSION_INFO)
+libginac_la_LDFLAGS = -no-undefined -version-info $(LT_VERSION_INFO)
 libginac_la_LIBADD = $(DL_LIBS)
 ginacincludedir = $(includedir)/ginac
 ginacinclude_HEADERS = ginac.h add.h archive.h assertion.h basic.h class_info.h \
