--- origsrc/ginac-1.6.7/ginac/Makefile.am	2016-02-28 20:17:51.000000000 +0900
+++ src/ginac-1.6.7/ginac/Makefile.am	2016-03-17 20:32:17.821943200 +0900
@@ -55,7 +55,7 @@ polynomial/primpart_content.cpp \
 polynomial/smod_helpers.h \
 polynomial/debug.h
 
-libginac_la_LDFLAGS = -version-info $(LT_VERSION_INFO)
+libginac_la_LDFLAGS = -no-undefined -version-info $(LT_VERSION_INFO)
 libginac_la_LIBADD = $(DL_LIBS)
 ginacincludedir = $(includedir)/ginac
 ginacinclude_HEADERS = ginac.h add.h archive.h assertion.h basic.h class_info.h \
