--- origsrc/libgtask-0.1/configure.in	2004-02-05 15:14:08.000000000 +0900
+++ src/libgtask-0.1/configure.in	2010-12-31 00:07:51.886721200 +0900
@@ -21,16 +21,10 @@ AC_PROG_CPP
 AC_PROG_MAKE_SET
 AC_PROG_INSTALL
 AC_PROG_LN_S
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 AM_INIT_AUTOMAKE( [] )
 
-dnl Checks for libraries.
-dnl Replace `main' with a function in -libs:
-AC_CHECK_LIB(ibs, main)
-
-dnl Checks for header files.
-
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
 
--- origsrc/libgtask-0.1/gtask/Makefile.am	2004-02-03 14:29:54.000000000 +0900
+++ src/libgtask-0.1/gtask/Makefile.am	2010-12-31 00:22:19.207395400 +0900
@@ -19,9 +19,11 @@ libgtask_la_SOURCES = \
 
 libgtask_la_CFLAGS  = ${GLIB_CFLAGS} ${GOBJECT_CFLAGS} ${LIBXML_CFLAGS} -I../ -DG_LOG_DOMAIN=\"GTask\" -Wall -Wmissing-prototypes -g3
 
-libgtask_la_LDFLAGS = -version-info $(LIBGTASK_CURRENT_VERSION):$(LIBGTASK_REVISION_VERSION):$(LIBGTASK_COMPAT_VERSIONS)
+libgtask_la_LDFLAGS = -version-info $(LIBGTASK_CURRENT_VERSION):$(LIBGTASK_REVISION_VERSION):$(LIBGTASK_COMPAT_VERSIONS) -no-undefined
 
-libgtask_la_LIBADD  = ${GLIB_LIBS} ${GOBJECT_LIBS} ${LIBXML_LIBS} -lnsl
+libgtask_la_LIBADD  = ${GLIB_LIBS} ${GOBJECT_LIBS} ${LIBXML_LIBS}
+
+INCLUDES = -I$(top_srcdir)
 
 gtaskincludedir      = $(includedir)/gtask-$(LIBGTASK_RELEASE)/gtask/
 gtaskinclude_HEADERS = gtask.h \
