--- origsrc/vanessa_adt-0.0.9/configure.ac	2010-06-08 15:48:01.000000000 +0900
+++ src/vanessa_adt-0.0.9/configure.ac	2010-11-07 21:11:34.229123300 +0900
@@ -20,13 +20,11 @@ dnl Inc., 59 Temple Place, Suite 330, Bo
 AC_INIT(libvanessa_adt/vanessa_adt.h)
 AM_CONFIG_HEADER(config.h)
 
-AC_CONFIG_MACRO_DIR([libltdl/m4])
+#AC_CONFIG_MACRO_DIR([libltdl/m4])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
-LT_INIT
-LTDL_INIT
+LT_INIT(win32-dll)
 AM_INIT_AUTOMAKE(vanessa_adt, 0.0.9)
 
-AC_PROG_LIBTOOL
 AC_PROG_CC
 AC_PROG_INSTALL
 AC_PROG_LN_S
--- origsrc/vanessa_adt-0.0.9/libvanessa_adt/Makefile.am	2010-01-25 08:53:47.000000000 +0900
+++ src/vanessa_adt-0.0.9/libvanessa_adt/Makefile.am	2010-11-07 21:11:09.974246500 +0900
@@ -38,6 +38,6 @@ list.c \
 hash.c \
 unused.h
 
-libvanessa_adt_la_LDFLAGS    = -version-info 1:0:0
+libvanessa_adt_la_LDFLAGS    = -version-info 1:0:0 -no-undefined
 
 libvanessa_adt_la_LIBADD     = -lvanessa_logger
--- origsrc/vanessa_adt-0.0.9/test/Makefile.am	2010-06-08 11:32:44.000000000 +0900
+++ src/vanessa_adt-0.0.9/test/Makefile.am	2010-11-07 21:45:30.058566900 +0900
@@ -23,7 +23,8 @@
 #
 ######################################################################
 
-noinst_PROGRAMS = dynamic_array_test list_test hash_test config_file_test
+check_PROGRAMS = dynamic_array_test list_test hash_test config_file_test
+TESTS = $(check_PROGRAMS)
 
 dynamic_array_test_SOURCES = dynamic_array_test.c
 
