--- origsrc/vanessa_adt-0.0.9/configure.ac	2010-06-08 15:48:01.000000000 +0900
+++ src/vanessa_adt-0.0.9/configure.ac	2010-07-24 22:14:19.224414000 +0900
@@ -20,12 +20,13 @@ dnl Inc., 59 Temple Place, Suite 330, Bo
 AC_INIT(libvanessa_adt/vanessa_adt.h)
 AM_CONFIG_HEADER(config.h)
 
-AC_CONFIG_MACRO_DIR([libltdl/m4])
+#AC_CONFIG_MACRO_DIR([libltdl/m4])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
-LT_INIT
-LTDL_INIT
+#LT_INIT
+#LTDL_INIT
 AM_INIT_AUTOMAKE(vanessa_adt, 0.0.9)
 
+AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
 AC_PROG_CC
 AC_PROG_INSTALL
--- origsrc/vanessa_adt-0.0.9/libvanessa_adt/Makefile.am	2010-01-25 08:53:47.000000000 +0900
+++ src/vanessa_adt-0.0.9/libvanessa_adt/Makefile.am	2010-07-24 22:14:03.755664000 +0900
@@ -38,6 +38,6 @@ list.c \
 hash.c \
 unused.h
 
-libvanessa_adt_la_LDFLAGS    = -version-info 1:0:0
+libvanessa_adt_la_LDFLAGS    = -version-info 1:0:0 -no-undefined
 
 libvanessa_adt_la_LIBADD     = -lvanessa_logger
