--- origsrc/mathex/CMakeLists.txt	2024-11-19 10:15:16.175769800 +0900
+++ src/mathex/CMakeLists.txt	2024-11-19 10:36:29.011591100 +0900
@@ -16,5 +16,12 @@ target_link_libraries(tabletest mathex)
 target_link_libraries(userfunctest mathex) 
 target_link_libraries(inttest mathex) 
 
-install(TARGETS mathex LIBRARY DESTINATION lib${LIB_SUFFIX})
+install(TARGETS mathex RUNTIME DESTINATION bin ARCHIVE DESTINATION lib${LIB_SUFFIX})
 install(FILES mathex.h DESTINATION include)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/mathex.pc.in ${CMAKE_BINARY_DIR}/mathex.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/mathex.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/mathex/mathex.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/mathex/mathex.pc.in	2024-11-19 10:36:36.934844600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mathex
+Description: C++ library to parse/evaluate mathematical expressions
+Version: @VERSION@
+Libs: -L${libdir} -lmathex
+Cflags: -I${includedir}
