--- origsrc/mathex/CMakeLists.txt	2014-07-05 11:45:33.498746400 +0900
+++ src/mathex/CMakeLists.txt	2014-07-05 11:47:33.182340200 +0900
@@ -16,5 +16,5 @@ target_link_libraries(tabletest mathex)
 target_link_libraries(userfunctest mathex) 
 target_link_libraries(inttest mathex) 
 
-install(TARGETS mathex LIBRARY DESTINATION lib${LIB_SUFFIX})
+install(TARGETS mathex RUNTIME DESTINATION bin ARCHIVE DESTINATION lib${LIB_SUFFIX})
 install(FILES mathex.h DESTINATION include)
