--- origsrc/fcppt-5.0.0/CMakeLists.txt	2025-10-04 16:10:17.000000000 +0900
+++ src/fcppt-5.0.0/CMakeLists.txt	2025-12-13 13:37:20.238885700 +0900
@@ -127,7 +127,7 @@ function(make_fcppt_library_base targetn
   endif()
 
   if(NOT ${_VARIANT} STREQUAL INTERFACE)
-    set_target_properties(${targetname} PROPERTIES VERSION
+    set_target_properties(${targetname} PROPERTIES SOVERSION
                                                    ${FCPPT_USED_SO_VERSION})
 
     fcppt_utils_set_target_compiler_flags(${targetname})
--- origsrc/fcppt-5.0.0/cmake/modules/FcpptCMakeUtilsFunctions.cmake	2025-10-04 16:10:17.000000000 +0900
+++ src/fcppt-5.0.0/cmake/modules/FcpptCMakeUtilsFunctions.cmake	2025-12-13 13:41:00.387995500 +0900
@@ -117,8 +117,10 @@ set(FCPPT_UTILS_EXPORT_NAME "${PROJECT_N
 function(fcppt_utils_export_install_target target_name)
   install(
     TARGETS ${target_name}
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-    EXPORT "${FCPPT_UTILS_EXPORT_NAME}")
+    EXPORT "${FCPPT_UTILS_EXPORT_NAME}"
+    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    )
 endfunction()
 
 # Generates a config file for other projects to use via find_package.
--- origsrc/fcppt-5.0.0/cmake/modules/FcpptCMakeUtilsPaths.cmake	2025-10-04 16:10:17.000000000 +0900
+++ src/fcppt-5.0.0/cmake/modules/FcpptCMakeUtilsPaths.cmake	2025-12-13 14:26:28.482839500 +0900
@@ -7,7 +7,7 @@ set(FCPPT_UTILS_PROJECT_BINARY_DIR "${${
 set(FCPPT_UTILS_INSTALL_DATA_DIR "${CMAKE_INSTALL_DATAROOTDIR}/${PROJECT_NAME}")
 
 set(FCPPT_UTILS_INSTALL_CMAKEMODULES_DIR
-    "${FCPPT_UTILS_INSTALL_DATA_DIR}/cmake")
+    "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}/")
 
 set(FCPPT_UTILS_INSTALL_CMAKECONFIG_DIR_BASE "${CMAKE_INSTALL_LIBDIR}/cmake")
 
