--- origsrc/avro-cpp-1.8.0/CMakeLists.txt	2016-01-22 23:47:08.000000000 +0900
+++ src/avro-cpp-1.8.0/CMakeLists.txt	2016-04-11 15:54:14.285159300 +0900
@@ -81,22 +81,27 @@ add_library (avrocpp SHARED ${AVRO_SOURC
 set_property (TARGET avrocpp
     APPEND PROPERTY COMPILE_DEFINITIONS AVRO_DYN_LINK)
 
+option(ENABLE_STATIC "Enable static" OFF)
+if (ENABLE_STATIC)
 add_library (avrocpp_s STATIC ${AVRO_SOURCE_FILES})
 
 set_property (TARGET avrocpp avrocpp_s
     APPEND PROPERTY COMPILE_DEFINITIONS AVRO_SOURCE)
+endif (ENABLE_STATIC)
 
 set_target_properties (avrocpp PROPERTIES
-    VERSION ${AVRO_VERSION_MAJOR}.${AVRO_VERSION_MINOR})
-
-set_target_properties (avrocpp_s PROPERTIES
-    VERSION ${AVRO_VERSION_MAJOR}.${AVRO_VERSION_MINOR})
+    VERSION ${AVRO_VERSION_MAJOR}.${AVRO_VERSION_MINOR}
+    SOVERSION 0)
 
 target_link_libraries (avrocpp ${Boost_LIBRARIES})
 
 add_executable (precompile test/precompile.cc)
 
+if (ENABLE_STATIC)
 target_link_libraries (precompile avrocpp_s ${Boost_LIBRARIES})
+else (ENABLE_STATIC)
+target_link_libraries (precompile avrocpp ${Boost_LIBRARIES})
+endif (ENABLE_STATIC)
 
 macro (gen file ns)
     add_custom_command (OUTPUT ${file}.hh
@@ -121,7 +126,11 @@ gen (reuse ru)
 gen (circulardep cd)
 
 add_executable (avrogencpp impl/avrogencpp.cc)
+if (NOT ENABLE_STATIC)
+target_link_libraries (avrogencpp avrocpp ${Boost_LIBRARIES})
+else (NOT ENABLE_STATIC)
 target_link_libraries (avrogencpp avrocpp_s ${Boost_LIBRARIES})
+endif (NOT ENABLE_STATIC)
 
 enable_testing()
 
@@ -154,10 +163,17 @@ set (CPACK_PACKAGE_FILE_NAME "avrocpp-${
 
 include (CPack)
 
+if (NOT ENABLE_STATIC)
+install (TARGETS avrocpp
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib
+    RUNTIME DESTINATION bin)
+else (NOT ENABLE_STATIC)
 install (TARGETS avrocpp avrocpp_s
     LIBRARY DESTINATION lib
     ARCHIVE DESTINATION lib
     RUNTIME DESTINATION lib)
+endif (NOT ENABLE_STATIC)
 
 install (TARGETS avrogencpp RUNTIME DESTINATION bin)
 
@@ -169,3 +185,10 @@ if (NOT CMAKE_BUILD_TYPE)
       "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel."
       FORCE)
 endif (NOT CMAKE_BUILD_TYPE)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/avro-cpp.pc.in ${CMAKE_BINARY_DIR}/avro-cpp.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/avro-cpp.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/avro-cpp-1.8.0/avro-cpp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/avro-cpp-1.8.0/avro-cpp.pc.in	2016-04-11 15:51:43.243283700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: avro-cpp
+Description: Data serialization system (C++)
+Version: @AVRO_VERSION@
+Libs: -L${libdir} -lavro-cpp
+Cflags: -I${includedir}/avro
