--- origsrc/avro-cpp-1.9.1/CMakeLists.txt	2019-08-28 18:35:41.000000000 +0900
+++ src/avro-cpp-1.9.1/CMakeLists.txt	2019-09-05 18:06:44.445376200 +0900
@@ -107,22 +107,36 @@ add_library (avrocpp SHARED ${AVRO_SOURC
 set_property (TARGET avrocpp
     APPEND PROPERTY COMPILE_DEFINITIONS AVRO_DYN_LINK)
 
-add_library (avrocpp_s STATIC ${AVRO_SOURCE_FILES})
-
-set_property (TARGET avrocpp avrocpp_s
+set_property (TARGET avrocpp
     APPEND PROPERTY COMPILE_DEFINITIONS AVRO_SOURCE)
 
 set_target_properties (avrocpp PROPERTIES
     VERSION ${AVRO_VERSION_MAJOR}.${AVRO_VERSION_MINOR})
 
-set_target_properties (avrocpp_s PROPERTIES
-    VERSION ${AVRO_VERSION_MAJOR}.${AVRO_VERSION_MINOR})
+if (CYGWIN)
+    set_target_properties (avrocpp PROPERTIES SOVERSION 0)
+endif()
+
+if (NOT BUILD_SHARED_LIBS)
+    add_library (avrocpp_s STATIC ${AVRO_SOURCE_FILES})
+
+    set_property (TARGET avrocpp_s
+        APPEND PROPERTY COMPILE_DEFINITIONS AVRO_SOURCE)
+
+    set_target_properties (avrocpp_s PROPERTIES
+        VERSION ${AVRO_VERSION_MAJOR}.${AVRO_VERSION_MINOR})
+endif()
+
 
 target_link_libraries (avrocpp ${Boost_LIBRARIES} ${SNAPPY_LIBRARIES})
 
 add_executable (precompile test/precompile.cc)
 
-target_link_libraries (precompile avrocpp_s ${Boost_LIBRARIES} ${SNAPPY_LIBRARIES})
+if (BUILD_SHARED_LIBS)
+    target_link_libraries (precompile avrocpp ${Boost_LIBRARIES} ${SNAPPY_LIBRARIES})
+else()
+    target_link_libraries (precompile avrocpp_s ${Boost_LIBRARIES} ${SNAPPY_LIBRARIES})
+endif()
 
 macro (gen file ns)
     add_custom_command (OUTPUT ${file}.hh
@@ -151,7 +165,11 @@ gen (crossref cr)
 gen (primitivetypes pt)
 
 add_executable (avrogencpp impl/avrogencpp.cc)
-target_link_libraries (avrogencpp avrocpp_s ${Boost_LIBRARIES} ${SNAPPY_LIBRARIES})
+if (BUILD_SHARED_LIBS)
+    target_link_libraries (avrogencpp avrocpp ${Boost_LIBRARIES} ${SNAPPY_LIBRARIES})
+else()
+    target_link_libraries (avrogencpp avrocpp_s ${Boost_LIBRARIES} ${SNAPPY_LIBRARIES})
+endif()
 
 enable_testing()
 
@@ -186,10 +204,16 @@ set (CPACK_PACKAGE_FILE_NAME "avrocpp-${
 
 include (CPack)
 
-install (TARGETS avrocpp avrocpp_s
+install (TARGETS avrocpp
     LIBRARY DESTINATION lib
     ARCHIVE DESTINATION lib
-    RUNTIME DESTINATION lib)
+    RUNTIME DESTINATION bin)
+if (NOT BUILD_SHARED_LIBS)
+    install (TARGETS avrocpp_s
+        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION lib
+        RUNTIME DESTINATION bin)
+endif()
 
 install (TARGETS avrogencpp RUNTIME DESTINATION bin)
 
@@ -201,3 +225,10 @@ if (NOT CMAKE_BUILD_TYPE)
       "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel."
       FORCE)
 endif (NOT CMAKE_BUILD_TYPE)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/avro-cpp.pc.in ${CMAKE_BINARY_DIR}/avro-cpp.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/avro-cpp.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/avro-cpp-1.9.1/avro-cpp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/avro-cpp-1.9.1/avro-cpp.pc.in	2019-09-05 18:06:44.445376200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: avro-cpp
+Description: Data serialization system (C++)
+Version: @AVRO_VERSION@
+Libs: -L${libdir} -lavro-cpp
+Cflags: -I${includedir}/avro
