--- origsrc/libuninum-2.7/Makefile.am	2008-02-10 12:59:33.000000000 +0900
+++ src/libuninum-2.7/Makefile.am	2015-05-14 14:32:26.761990000 +0900
@@ -9,7 +9,7 @@ library_includedir=$(includedir)/uninum
 library_include_HEADERS =  nsdefs.h unicode.h uninum.h
 numconv_LDADD = libuninum.la
 LIBS+=-lm
-libuninum_la_LDFLAGS = -version-info 8:0:3
+libuninum_la_LDFLAGS = -no-undefined -version-info 8:0:3
 EXTRA_DIST = Examples GPL TestFiles uninum.i tcl_uninum_wrap.c README_TCL TclMakefile CREDITS NumberConverter.tcl README_NUMBERCONVERTER
 AUTOMAKE_OPTIONS = dist-zip dist-bzip2
 if ALLOCAOK
@@ -17,3 +17,6 @@ AM_CFLAGS = -DALLOCAOK -std=gnu99
 else
 AM_CFLAGS = -std=gnu99
 endif
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libuninum-2.7/configure.ac	2008-05-22 08:17:35.000000000 +0900
+++ src/libuninum-2.7/configure.ac	2015-05-14 14:32:07.133126400 +0900
@@ -33,5 +33,5 @@ AC_FUNC_MALLOC
 AC_CHECK_FUNCS([getopt_long localeconv setlocale strrchr wcschr])
 AC_CHECK_LIB(gmp, __gmpz_init,,[AC_MSG_ERROR([GNU MP not found, see http://swox.com/gmp/])])
 
-AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([Makefile libuninum.pc])
 AC_OUTPUT
--- origsrc/libuninum-2.7/libuninum.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libuninum-2.7/libuninum.pc.in	2015-05-14 14:31:50.470869000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libuninum
+Description: Library for converting unicode strings to numbers
+Version: @VERSION@
+Libs: -L${libdir} -luninum
+Cflags: -I${includedir}/uninum
