--- origsrc/libuninum-2.7/Makefile.am	2008-02-10 12:59:33.000000000 +0900
+++ src/libuninum-2.7/Makefile.am	2013-12-10 20:16:01.267871900 +0900
@@ -9,7 +9,7 @@ library_includedir=$(includedir)/uninum
 library_include_HEADERS =  nsdefs.h unicode.h uninum.h
 numconv_LDADD = libuninum.la
 LIBS+=-lm
-libuninum_la_LDFLAGS = -version-info 8:0:3
+libuninum_la_LDFLAGS = -no-undefined -version-info 8:0:3
 EXTRA_DIST = Examples GPL TestFiles uninum.i tcl_uninum_wrap.c README_TCL TclMakefile CREDITS NumberConverter.tcl README_NUMBERCONVERTER
 AUTOMAKE_OPTIONS = dist-zip dist-bzip2
 if ALLOCAOK
