--- origsrc/signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/lib/SignOn/SignOn.pro	2018-09-18 01:28:27.000000000 +0900
+++ src/signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/lib/SignOn/SignOn.pro	2019-01-01 16:00:56.440405000 +0900
@@ -1,7 +1,6 @@
 TEMPLATE = subdirs
 CONFIG   += ordered
 SUBDIRS = \
-    libsignon-qt-shared.pro \
-    libsignon-qt-static.pro
+    libsignon-qt-shared.pro
 
-include(doc/doc.pri)
+# include(doc/doc.pri)
--- origsrc/signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/lib/plugins/plugins.pro	2018-09-18 01:28:27.000000000 +0900
+++ src/signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/lib/plugins/plugins.pro	2019-01-01 15:59:46.491144900 +0900
@@ -3,5 +3,5 @@ CONFIG += ordered
 SUBDIRS = signon-plugins.pro signon-plugins-common
 
 TARGET=signon-plugins
-include(doc/doc.pri)
+# include(doc/doc.pri)
 
--- origsrc/signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/signon.pro	2018-09-18 01:28:27.000000000 +0900
+++ src/signond-VERSION_8.60-4d195e4dc7a47ff5cb51e36a83d4d05808c5befe/signon.pro	2019-01-01 15:44:01.899417000 +0900
@@ -7,6 +7,6 @@ tests.depends = lib src
 
 include( common-installs-config.pri )
 
-include( doc/doc.pri )
+# include( doc/doc.pri )
 
 # End of File
