--- origsrc/libburrow/Makefile.am	2017-12-01 12:29:05.000000000 +0900
+++ src/libburrow/Makefile.am	2017-12-01 12:38:31.072000000 +0900
@@ -99,8 +99,8 @@ libburrow_la_CFLAGS = \
 	${AM_CFLAGS} \
 	-DBUILDING_LIBBURROW
 
-libburrow_la_LIBADD = $(LIBDL_LIBS)
-libburrow_la_LDFLAGS = $(AM_LDFLAGS) -version-info $(LIBBURROW_VERSION)
+libburrow_la_LIBADD = $(LIBDL_LIBS) $(LIBCURL)
+libburrow_la_LDFLAGS = $(AM_LDFLAGS) -no-undefined -version-info $(LIBBURROW_VERSION)
 
 #
 # Simple libburrow application
@@ -180,3 +180,6 @@ lcov-clean: clean
 	find . -name '*.gcno' -exec rm {} \;
 	find . -name '*.gcda' -exec rm {} \;
 	find . -name 'lcov*.out' -exec rm {} \;
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = support/$(PACKAGE).pc
