--- origsrc/fastlz/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/fastlz/Makefile.am	2014-10-13 16:01:08.512541300 +0900
@@ -0,0 +1,13 @@
+
+bin_PROGRAMS = 6pack 6unpack
+6pack_SOURCES = 6pack.c
+6pack_LDADD = libfastlz.la
+6unpack_SOURCES = 6unpack.c
+6unpack_LDADD = libfastlz.la
+
+include_HEADERS = fastlz.h
+
+lib_LTLIBRARIES = libfastlz.la
+libfastlz_la_SOURCES = fastlz.c
+libfastlz_la_LDFLAGS = -no-undefined
+
--- origsrc/fastlz/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/fastlz/configure.ac	2014-10-13 16:00:24.819182000 +0900
@@ -0,0 +1,28 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([fastlz], [0.1.0+svn12])
+AC_CONFIG_SRCDIR([6pack.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_C_INLINE
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([memset])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
