--- origsrc/corona-1.0.2/configure.in	2003-09-26 22:19:55.000000000 +0900
+++ src/corona-1.0.2/configure.in	2012-02-12 13:12:45.471138200 +0900
@@ -1,13 +1,6 @@
-AC_INIT(src/corona.h)
+AC_INIT([corona], [1.0.2])
 AC_CONFIG_AUX_DIR(config)
-
-VERSION=1.0.2
-PACKAGE="corona"
-
-AC_SUBST(VERSION)
-AC_SUBST(PACKAGE)
-
-AM_INIT_AUTOMAKE($PACKAGE, $VERSION)
+AM_INIT_AUTOMAKE
 
 dnl Checks for options.
 
--- origsrc/corona-1.0.2/src/Convert.cpp	2003-02-26 13:06:53.000000000 +0900
+++ src/corona-1.0.2/src/Convert.cpp	2012-02-12 13:16:49.071865600 +0900
@@ -10,6 +10,7 @@
 #include "SimpleImage.h"
 #include "Utility.h"
 
+#include <cstring>
 
 namespace corona {
 
--- origsrc/corona-1.0.2/src/Makefile.am	2003-02-18 11:15:50.000000000 +0900
+++ src/corona-1.0.2/src/Makefile.am	2012-02-12 13:13:10.846795800 +0900
@@ -1,7 +1,3 @@
-SUBDIRS = libungif-4.1.0
-
-INCLUDES = -I $(top_srcdir)/src/libungif-4.1.0
-
 lib_LTLIBRARIES = libcorona.la
 include_HEADERS = corona.h
 
@@ -40,8 +36,7 @@ libcorona_la_SOURCES =				\
 	Types.h					\
 	Utility.h
 
-libcorona_la_LIBADD = \
-	libungif-4.1.0/libungif.la
+libcorona_la_LIBADD = -lgif
 
-libcorona_la_LDFLAGS = \
+libcorona_la_LDFLAGS = -no-undefined \
 	-release $(VERSION)
--- origsrc/corona-1.0.2/src/OpenTGA.cpp	2002-12-27 18:33:42.000000000 +0900
+++ src/corona-1.0.2/src/OpenTGA.cpp	2012-02-12 13:17:23.100796800 +0900
@@ -4,6 +4,7 @@
 #include "SimpleImage.h"
 #include "Utility.h"
 
+#include <cstring>
 
 namespace corona {
 
