--- origsrc/ding-libs-0.6.2/Makefile.am	2022-01-22 04:44:04.000000000 +0900
+++ src/ding-libs-0.6.2/Makefile.am	2023-11-20 10:50:44.841970100 +0900
@@ -69,7 +69,7 @@ libpath_utils_la_SOURCES = path_utils/pa
 libpath_utils_la_DEPENDENCIES = path_utils/libpath_utils.sym
 libpath_utils_la_LIBADD = $(LTLIBICONV) \
     $(LTLIBINTL)
-libpath_utils_la_LDFLAGS = \
+libpath_utils_la_LDFLAGS = -no-undefined \
     -version-info 1:1:0
 
 if HAVE_LD_VERSION_SCRIPT
@@ -117,7 +117,7 @@ dist_include_HEADERS += dhash/dhash.h
 
 libdhash_la_SOURCES = dhash/dhash.c
 libdhash_la_DEPENDENCIES = dhash/libdhash.sym
-libdhash_la_LDFLAGS = \
+libdhash_la_LDFLAGS = -no-undefined \
     -version-info 2:0:1
 if HAVE_LD_VERSION_SCRIPT
 libdhash_la_LDFLAGS += -Wl,--version-script=$(top_srcdir)/dhash/libdhash.sym
@@ -175,7 +175,7 @@ libcollection_la_SOURCES = \
     collection/collection_priv.h \
     trace/trace.h
 libcollection_la_DEPENDENCIES = collection/libcollection.sym
-libcollection_la_LDFLAGS = \
+libcollection_la_LDFLAGS = -no-undefined \
     -version-info 5:1:1
 if HAVE_LD_VERSION_SCRIPT
 libcollection_la_LDFLAGS += -Wl,--version-script=$(top_srcdir)/collection/libcollection.sym
@@ -215,7 +215,7 @@ libref_array_la_SOURCES = \
     refarray/ref_array.c \
     trace/trace.h
 libref_array_la_DEPENDENCIES = refarray/libref_array.sym
-libref_array_la_LDFLAGS = \
+libref_array_la_LDFLAGS = -no-undefined \
     -version-info 3:1:2
 if HAVE_LD_VERSION_SCRIPT
 libref_array_la_LDFLAGS += -Wl,--version-script=$(top_srcdir)/refarray/libref_array.sym
@@ -246,7 +246,7 @@ libbasicobjects_la_SOURCES = \
     basicobjects/simplebuffer.c \
     trace/trace.h
 libbasicobjects_la_DEPENDENCIES = basicobjects/libbasicobjects.sym
-libbasicobjects_la_LDFLAGS = \
+libbasicobjects_la_LDFLAGS = -no-undefined \
     -version-info 1:0:1
 if HAVE_LD_VERSION_SCRIPT
 libbasicobjects_la_LDFLAGS += -Wl,--version-script=$(top_srcdir)/basicobjects/libbasicobjects.sym
@@ -309,7 +309,7 @@ libini_config_la_LIBADD = \
     libbasicobjects.la \
     $(LTLIBICONV) \
     $(LTLIBINTL)
-libini_config_la_LDFLAGS = \
+libini_config_la_LDFLAGS = -no-undefined \
     -version-info 7:1:2
 if HAVE_LD_VERSION_SCRIPT
 libini_config_la_LDFLAGS += -Wl,--version-script=$(top_srcdir)/ini/libini_config.sym
