--- origsrc/kvazaar-1.2.0/tests/Makefile.am	2017-11-17 22:23:33.000000000 +0900
+++ src/kvazaar-1.2.0/tests/Makefile.am	2018-03-26 14:47:28.665999800 +0900
@@ -29,7 +29,7 @@ kvazaar_tests_SOURCES = \
 	test_strategies.c \
 	test_strategies.h
 kvazaar_tests_CFLAGS = -I$(srcdir) -I$(top_srcdir) -I$(top_srcdir)/src
-kvazaar_tests_LDFLAGS = -static $(top_builddir)/src/libkvazaar.la $(LIBS)
+kvazaar_tests_LDFLAGS = $(top_builddir)/src/libkvazaar.la $(LIBS)
 
 # This makes sure that CXXLD gets defined.
 nodist_EXTRA_kvazaar_tests_SOURCES = cpp.cpp
