--- origsrc/kvazaar-2.0.0/tests/Makefile.am	2020-04-22 07:03:36.000000000 +0900
+++ src/kvazaar-2.0.0/tests/Makefile.am	2020-07-15 13:26:52.059763100 +0900
@@ -46,7 +46,7 @@ kvazaar_tests_SOURCES = \
 	test_strategies.c \
 	test_strategies.h
 kvazaar_tests_CFLAGS = -I$(srcdir) -I$(top_srcdir) -I$(top_srcdir)/src
-kvazaar_tests_LDFLAGS = -static $(top_builddir)/src/libkvazaar.la $(LIBS)
+kvazaar_tests_LDFLAGS = $(top_builddir)/src/libkvazaar.la $(LIBS)
 
 # This makes sure that CXXLD gets defined.
 nodist_EXTRA_kvazaar_tests_SOURCES = cpp.cpp
