--- origsrc/kvazaar-2.3.0/CMakeLists.txt	2024-01-17 20:49:34.000000000 +0900
+++ src/kvazaar-2.3.0/CMakeLists.txt	2024-02-14 14:22:33.461411500 +0900
@@ -131,6 +131,7 @@ if(BUILD_SHARED_LIBS)
   list( APPEND CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib" "./" "../lib" )  
   set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
   add_library(kvazaar SHARED ${LIB_SOURCES})
+  set_target_properties(kvazaar PROPERTIES SOVERSION 7)
 else()
   add_library(kvazaar STATIC ${LIB_SOURCES})
   if(MSVC) # Fix a linking problem with visual studio when the library is the same name as the binary
@@ -231,16 +232,14 @@ source_group( "" FILES ${SOURCE_GROUP_TO
 
 # ToDo: make configurable
 
-install(FILES ${PROJECT_SOURCE_DIR}/src/kvazaar.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/share/pkgconfig)
+install(FILES ${PROJECT_SOURCE_DIR}/src/kvazaar.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
 install(TARGETS kvazaar-bin DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
-install(TARGETS kvazaar DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
+install(TARGETS kvazaar ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
 if(BUILD_SHARED_LIBS) # Just add the lib to the bin directory for now
-  if(MSVC)
-    install(TARGETS kvazaar DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
-  endif()
+  install(TARGETS kvazaar RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
 endif()
 install(FILES ${PROJECT_SOURCE_DIR}/src/kvazaar.h DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
-install(FILES ${PROJECT_SOURCE_DIR}/doc/kvazaar.1 DESTINATION ${CMAKE_INSTALL_PREFIX}/share/man)
+install(FILES ${PROJECT_SOURCE_DIR}/doc/kvazaar.1 DESTINATION ${CMAKE_INSTALL_PREFIX}/share/man/man1)
 
 IF(UNIX)
 # DIST
