--- origsrc/rg_etc1_v104/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/rg_etc1_v104/Makefile.am	2015-01-08 10:34:19.575962500 +0900
@@ -0,0 +1,6 @@
+
+include_HEADERS = rg_etc1.h
+
+lib_LTLIBRARIES = librg_etc1.la
+librg_etc1_la_SOURCES = rg_etc1.cpp
+librg_etc1_la_LDFLAGS = -no-undefined
--- origsrc/rg_etc1_v104/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/rg_etc1_v104/configure.ac	2015-01-08 10:35:02.408409400 +0900
@@ -0,0 +1,29 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([rg_etc1], [1.04])
+AC_CONFIG_SRCDIR([rg_etc1.cpp])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([memory.h stdlib.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+AC_C_INLINE
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_CHECK_FUNCS([memset])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
