--- origsrc/yazpp-1.5.2/configure.ac	2013-10-30 19:17:19.000000000 +0900
+++ src/yazpp-1.5.2/configure.ac	2013-12-09 07:19:49.249863000 +0900
@@ -2,7 +2,7 @@ dnl This file is part of yazpp
 dnl Copyright (C) 1998-2013 Index Data
 AC_PREREQ(2.60)
 AC_INIT([yazpp],
-m4_esyscmd([. ./IDMETA; echo $VERSION|tr -d '\n']),
+[1.5.2],
 [adam@indexdata.dk])
 AC_CONFIG_HEADERS([include/config.h])
 AC_CONFIG_SRCDIR(configure.ac)
@@ -40,8 +40,7 @@ AC_OUTPUT([
 	doc/common/print.dsl
 ],[
 	sed s%echo_source=yes%echo_source=no%g < yazpp-config >src/yazpp-config && chmod +x yazpp-config src/yazpp-config
-	diff doc/local.ent doc/local0.ent >/dev/null 2>/dev/null \
-		|| cp doc/local0.ent doc/local.ent
+	cp doc/local0.ent doc/local.ent
   ]
 )
 
--- origsrc/yazpp-1.5.2/src/Makefile.am	2013-10-30 19:23:25.000000000 +0900
+++ src/yazpp-1.5.2/src/Makefile.am	2013-12-09 07:20:28.515488000 +0900
@@ -8,7 +8,7 @@ TESTS = $(check_PROGRAMS)
 AM_CXXFLAGS = -I$(srcdir)/../include $(YAZINC)
 
 lib_LTLIBRARIES = libyazpp.la
-libyazpp_la_LDFLAGS=-version-info 6:0:0
+libyazpp_la_LDFLAGS = -no-undefined -version-info 6:0:0
 
 DISTCLEANFILES = yazpp-config
 
--- origsrc/yazpp-1.5.2/zoom/Makefile.am	2013-10-02 18:03:26.000000000 +0900
+++ src/yazpp-1.5.2/zoom/Makefile.am	2013-12-09 07:20:41.749863000 +0900
@@ -7,7 +7,7 @@ lib_LTLIBRARIES = libzoompp.la
 noinst_PROGRAMS = zclient
 endif
 
-libzoompp_la_LDFLAGS=-version-info 6:0:0
+libzoompp_la_LDFLAGS = -no-undefined -version-info 6:0:0
 LDADD=libzoompp.la $(YAZLALIB)
 
 libzoompp_la_LIBADD = $(YAZLALIB)
