--- origsrc/tree-sitter-lua-0.2.0/CMakeLists.txt	2024-11-12 10:09:03.849415400 +0900
+++ src/tree-sitter-lua-0.2.0/CMakeLists.txt	2024-11-12 10:12:35.113672000 +0900
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.13)
 
 project(tree-sitter-lua
-        VERSION "0.3.0"
+        VERSION "0.2.0"
         DESCRIPTION "Lua grammar for tree-sitter"
         HOMEPAGE_URL "https://github.com/tree-sitter-grammars/tree-sitter-lua"
         LANGUAGES C)
@@ -17,12 +17,14 @@ endif()
 
 find_program(TREE_SITTER_CLI tree-sitter DOC "Tree-sitter CLI")
 
+if (NOT TREE_SITTER_CLI STREQUAL "TREE_SITTER_CLI-NOTFOUND")
 add_custom_command(OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/src/parser.c"
                    DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/src/grammar.json"
                    COMMAND "${TREE_SITTER_CLI}" generate src/grammar.json
                             --abi=${TREE_SITTER_ABI_VERSION}
                    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
                    COMMENT "Generating parser.c")
+endif()
 
 add_library(tree-sitter-lua src/parser.c src/scanner.c)
 target_include_directories(tree-sitter-lua PRIVATE src)
@@ -38,15 +40,15 @@ set_target_properties(tree-sitter-lua
                       SOVERSION "${TREE_SITTER_ABI_VERSION}.${PROJECT_VERSION_MAJOR}"
                       DEFINE_SYMBOL "")
 
+include(GNUInstallDirs)
+
 configure_file(bindings/c/tree-sitter-lua.pc.in
                "${CMAKE_CURRENT_BINARY_DIR}/tree-sitter-lua.pc" @ONLY)
 
-include(GNUInstallDirs)
-
 install(FILES bindings/c/tree-sitter-lua.h
         DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/tree_sitter")
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/tree-sitter-lua.pc"
-        DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/pkgconfig")
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 install(TARGETS tree-sitter-lua
         LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 
--- origsrc/tree-sitter-lua-0.2.0/bindings/c/tree-sitter-lua.pc.in	2024-09-10 00:10:03.000000000 +0900
+++ src/tree-sitter-lua-0.2.0/bindings/c/tree-sitter-lua.pc.in	2024-11-12 10:28:33.771092000 +0900
@@ -1,11 +1,11 @@
-prefix=@PREFIX@
-libdir=@LIBDIR@
-includedir=@INCLUDEDIR@
+prefix=@CMAKE_INSTALL_PREFIX@
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
 
 Name: tree-sitter-lua
 Description: lua grammar for tree-sitter
 URL: @URL@
-Version: @VERSION@
+Version: @PROJECT_VERSION@
 Requires: @REQUIRES@
 Libs: -L${libdir} @ADDITIONAL_LIBS@ -ltree-sitter-lua
 Cflags: -I${includedir}
