--- origsrc/liblightgrep-1.4/Makefile.am	2014-09-10 02:16:18.000000000 +0900
+++ src/liblightgrep-1.4/Makefile.am	2022-12-03 15:26:18.504776100 +0900
@@ -1,6 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4
 
-AM_CPPFLAGS = -Iinclude $(BOOST_CPPFLAGS) $(ICU_CPPFLAGS) $(LG_CPPFLAGS)
+AM_CPPFLAGS = -I$(top_srcdir)/include $(BOOST_CPPFLAGS) $(ICU_CPPFLAGS) $(LG_CPPFLAGS)
 AM_CFLAGS = $(LG_CFLAGS)
 AM_CXXFLAGS = $(LG_CXXFLAGS)
 AM_LDFLAGS = $(BOOST_LDFLAGS) $(ICU_LDFLAGS) $(LG_LDFLAGS)
@@ -68,9 +68,9 @@ src_lib_liblightgrepint_la_SOURCES = \
 
 src_lib_liblightgrep_la_LIBADD = $(LG_LIB_INT) $(ICU_LIBS) $(STDCXX_LIB)
 if BUILD_DLL
-src_lib_liblightgrep_la_LDFLAGS = $(AM_LDFLAGS) $(LG_LIB_LDFLAGS) -avoid-version
+src_lib_liblightgrep_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) $(LG_LIB_LDFLAGS) -avoid-version
 else
-src_lib_liblightgrep_la_LDFLAGS = $(AM_LDFLAGS) $(LG_LIB_LDFLAGS)
+src_lib_liblightgrep_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) $(LG_LIB_LDFLAGS)
 endif
 
 src_lib_liblightgrep_ladir = $(includedir)/lightgrep
--- origsrc/liblightgrep-1.4/configure.ac	2014-09-10 02:16:18.000000000 +0900
+++ src/liblightgrep-1.4/configure.ac	2022-12-03 16:21:22.404366200 +0900
@@ -3,7 +3,7 @@
 AC_PREREQ([2.64])
 # TODO: required versions for automake, libtool?
 
-AC_INIT([liblightgrep], [1.3.0], [joel@lightboxtechnologies.com])
+AC_INIT([liblightgrep], [1.4], [joel@lightboxtechnologies.com])
 
 # Set the default C and C++ compilation flags to nothing.
 # Do this immediately after AC_INIT, as other macros might invoke
