--- origsrc/liblightgrep-1.4/Makefile.am	2014-09-10 02:16:18.000000000 +0900
+++ src/liblightgrep-1.4/Makefile.am	2018-06-24 09:53:39.447820000 +0900
@@ -1,6 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4
 
-AM_CPPFLAGS = -Iinclude $(BOOST_CPPFLAGS) $(ICU_CPPFLAGS) $(LG_CPPFLAGS)
+AM_CPPFLAGS = -I$(top_srcdir)/include $(BOOST_CPPFLAGS) $(ICU_CPPFLAGS) $(LG_CPPFLAGS)
 AM_CFLAGS = $(LG_CFLAGS)
 AM_CXXFLAGS = $(LG_CXXFLAGS)
 AM_LDFLAGS = $(BOOST_LDFLAGS) $(ICU_LDFLAGS) $(LG_LDFLAGS)
@@ -68,9 +68,9 @@ src_lib_liblightgrepint_la_SOURCES = \
 
 src_lib_liblightgrep_la_LIBADD = $(LG_LIB_INT) $(ICU_LIBS) $(STDCXX_LIB)
 if BUILD_DLL
-src_lib_liblightgrep_la_LDFLAGS = $(AM_LDFLAGS) $(LG_LIB_LDFLAGS) -avoid-version
+src_lib_liblightgrep_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) $(LG_LIB_LDFLAGS) -avoid-version
 else
-src_lib_liblightgrep_la_LDFLAGS = $(AM_LDFLAGS) $(LG_LIB_LDFLAGS)
+src_lib_liblightgrep_la_LDFLAGS = -no-undefined $(AM_LDFLAGS) $(LG_LIB_LDFLAGS)
 endif
 
 src_lib_liblightgrep_ladir = $(includedir)/lightgrep
