--- origsrc/cryptominisat-5.6.8/python/CMakeLists.txt	2019-02-20 05:49:20.000000000 +0900
+++ src/cryptominisat-5.6.8/python/CMakeLists.txt	2020-01-25 08:56:20.034025800 +0900
@@ -41,7 +41,7 @@ print(sysconfig.get_config_var('LINKFORS
 
 string(REPLACE "\n" " " PY_C_CONFIG ${PY_C_CONFIG})
 string(REPLACE "\n" " " PY_LD_CONFIG ${PY_LD_CONFIG})
-string(REPLACE "\n" " " PY_LINKFORSHARED_CONFIG ${PY_LINKFORSHARED_CONFIG})
+string(REPLACE "\n" " " PY_LINKFORSHARED_CONFIG "${PY_LINKFORSHARED_CONFIG}")
 
 message(STATUS "Python CFLAGS:  '${PY_C_CONFIG}'")
 message(STATUS "Python LDFLAGS: '${PY_LD_CONFIG}'")
