--- origsrc/libemu-0.2.0/src/Makefile.am	2008-03-25 03:22:31.000000000 +0900
+++ src/libemu-0.2.0/src/Makefile.am	2013-01-30 22:14:17.963326500 +0900
@@ -6,9 +6,9 @@ AUTOMAKE_OPTIONS = foreign
 
 
 
-AM_CFLAGS = -pipe -D _GNU_SOURCE -I../include -Werror -Wall -g
+AM_CFLAGS = -pipe -D _GNU_SOURCE -I$(top_srcdir)/include -Werror -Wall -g -I$(top_builddir)
 
-pkglib_LTLIBRARIES = libemu.la
+lib_LTLIBRARIES = libemu.la
 
 SUBDIRS = functions
 
--- origsrc/libemu-0.2.0/src/environment/win32/env_w32_dll_export_kernel32_hooks.c	2008-12-01 05:31:51.000000000 +0900
+++ src/libemu-0.2.0/src/environment/win32/env_w32_dll_export_kernel32_hooks.c	2013-01-30 22:03:19.523047900 +0900
@@ -40,7 +40,7 @@
 #include <unistd.h>
 #include <errno.h>
 
-#include "../../../config.h"
+#include "config.h"
 #include "emu/emu.h"
 #include "emu/emu_log.h"
 #include "emu/emu_memory.h"
--- origsrc/libemu-0.2.0/src/environment/win32/env_w32_dll_export_msvcrt_hooks.c	2008-07-30 10:21:42.000000000 +0900
+++ src/libemu-0.2.0/src/environment/win32/env_w32_dll_export_msvcrt_hooks.c	2013-01-30 22:03:31.163672900 +0900
@@ -32,7 +32,7 @@
 #include <sys/wait.h>
 #include <unistd.h>
 
-#include "../../../config.h"
+#include "config.h"
 #include "emu/emu.h"
 #include "emu/emu_log.h"
 #include "emu/emu_memory.h"
--- origsrc/libemu-0.2.0/src/environment/win32/env_w32_dll_export_urlmon_hooks.c	2008-07-31 19:27:04.000000000 +0900
+++ src/libemu-0.2.0/src/environment/win32/env_w32_dll_export_urlmon_hooks.c	2013-01-30 22:03:23.757422900 +0900
@@ -32,7 +32,7 @@
 #include <sys/wait.h>
 #include <unistd.h>
 
-#include "../../../config.h"
+#include "config.h"
 #include "emu/emu.h"
 #include "emu/emu_log.h"
 #include "emu/emu_memory.h"
--- origsrc/libemu-0.2.0/src/environment/win32/env_w32_dll_export_ws2_32_hooks.c	2008-07-30 10:21:26.000000000 +0900
+++ src/libemu-0.2.0/src/environment/win32/env_w32_dll_export_ws2_32_hooks.c	2013-01-30 22:03:39.882422900 +0900
@@ -39,7 +39,7 @@
 #include <sys/wait.h>
 #include <unistd.h>
 
-#include "../../../config.h"
+#include "config.h"
 #include "emu/emu.h"
 #include "emu/emu_log.h"
 #include "emu/emu_memory.h"
--- origsrc/libemu-0.2.0/testsuite/Makefile.am	2008-07-16 18:43:33.000000000 +0900
+++ src/libemu-0.2.0/testsuite/Makefile.am	2013-01-30 22:13:03.113717100 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = foreign
 
-AM_CPPFLAGS = -I../include -I ../.. -Werror -Wall -g
+AM_CPPFLAGS = -I$(top_srcdir)/include -I ../.. -Werror -Wall -g
 AM_LDFLAGS = -lemu -L../src 
 
 bin_PROGRAMS = scprofiler
@@ -29,3 +29,5 @@ scprofiler_LDADD = ../src/libemu.la
 scprofiler_SOURCES = scprofiler.c
 
 EXTRA_DIST = emunids.c
+
+TESTS = hashtest instrtest instrtree memtest testsuite
--- origsrc/libemu-0.2.0/tools/sctest/Makefile.am	2008-11-19 00:48:42.000000000 +0900
+++ src/libemu-0.2.0/tools/sctest/Makefile.am	2013-01-30 22:07:54.501563500 +0900
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = foreign
 
-AM_CPPFLAGS = -I../../include -I ../../.. -Werror -Wall -g
+AM_CPPFLAGS = -I$(top_srcdir)/include -I ../../.. -Wall -g -I$(top_builddir)
 AM_LDFLAGS = -lemu -L../../src 
 
 
--- origsrc/libemu-0.2.0/tools/sctest/dot.c	2008-09-02 19:39:03.000000000 +0900
+++ src/libemu-0.2.0/tools/sctest/dot.c	2013-01-30 22:05:40.970313500 +0900
@@ -26,7 +26,7 @@
  *******************************************************************************/
 
 
-#include "../config.h"
+#include "config.h"
 
 #define HAVE_GETOPT_H
 #ifdef HAVE_GETOPT_H
--- origsrc/libemu-0.2.0/tools/sctest/sctestmain.c	2008-12-01 05:30:06.000000000 +0900
+++ src/libemu-0.2.0/tools/sctest/sctestmain.c	2013-01-30 22:06:21.798438500 +0900
@@ -26,7 +26,7 @@
  *******************************************************************************/
 
 
-#include "../config.h"
+#include "config.h"
 
 #define HAVE_GETOPT_H
 #ifdef HAVE_GETOPT_H
--- origsrc/libemu-0.2.0/tools/sctest/userhooks.c	2008-12-01 05:59:17.000000000 +0900
+++ src/libemu-0.2.0/tools/sctest/userhooks.c	2013-01-30 22:07:14.829688500 +0900
@@ -26,7 +26,7 @@
  *******************************************************************************/
 
 
-#include "../config.h"
+#include "config.h"
 
 #define HAVE_GETOPT_H
 #ifdef HAVE_GETOPT_H
@@ -97,6 +97,8 @@
 #include <fcntl.h>
 #include <string.h>
 
+#include <ctype.h>
+
 uint32_t user_hook_ExitProcess(struct emu_env *env, struct emu_env_hook *hook, ...)
 {
 	printf("Hook me Captain Cook!\n");
