--- origsrc/rudesession-3.2.2/Makefile.am	2007-05-30 01:17:56.000000000 +0900
+++ src/rudesession-3.2.2/Makefile.am	2015-04-25 13:11:46.679529000 +0900
@@ -56,7 +56,7 @@ librudesession_la_SOURCES = 		src/Implem
 	src/IPValidator.cpp \
 	src/IPValidator.h
 
-librudesession_la_LDFLAGS = -version-info 3:2:1
+librudesession_la_LDFLAGS = -no-undefined -version-info 3:2:1
 
 #  -------------------------------------
 librudesession_la_LIBADD = -lrudeconfig -lrudedatabase
@@ -71,4 +71,5 @@ EXTRA_DIST = $(man_MANS)
 
 AUTOMAKE_OPTIONS = gnits dist-bzip2 dist-shar dist-zip
 
-
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = rudesession.pc
--- origsrc/rudesession-3.2.2/configure.ac	2007-05-30 01:18:51.000000000 +0900
+++ src/rudesession-3.2.2/configure.ac	2015-04-25 13:49:14.889004200 +0900
@@ -48,6 +48,4 @@ AC_TYPE_SIZE_T
 
 # Checks for library functions.
 #AC_CHECK_FUNCS([gethostbyname select socket strerror])
-AC_OUTPUT(Makefile )
-
-
+AC_OUTPUT(rudesession.pc Makefile )
--- origsrc/rudesession-3.2.2/rudesession.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/rudesession-3.2.2/rudesession.pc.in	2015-04-25 13:12:12.838341300 +0900
@@ -0,0 +1,5 @@
+Name: rudesession
+Description: Library (C++ API) for web applications
+Version: @VERSION@
+Libs: -L@libdir@ -lrudesession
+Cflags: -I@includedir@
