--- origsrc/cpp_redis-4.3.1/CMakeLists.txt	2018-02-01 15:32:34.000000000 +0900
+++ src/cpp_redis-4.3.1/CMakeLists.txt	2019-01-29 17:23:06.160091100 +0900
@@ -139,6 +139,10 @@ configure_file("cpp_redis.pc.in" "${CMAK
 add_library(${PROJECT} ${SOURCES})
 set_property(TARGET ${PROJECT} PROPERTY POSITION_INDEPENDENT_CODE ON)
 
+if(CYGWIN)
+  set_target_properties(${PROJECT} PROPERTIES SOVERSION 0)
+endif()
+
 if(WIN32)
   set_target_properties(${PROJECT}
                         PROPERTIES COMPILE_PDB_NAME ${PROJECT}
@@ -201,9 +205,12 @@ endif(BUILD_EXAMPLES)
 if(BUILD_TESTS)
   enable_testing()
   add_subdirectory(tests)
-  ExternalProject_Add("googletest"
-                      GIT_REPOSITORY "https://github.com/google/googletest.git"
-                      CMAKE_ARGS "-DCMAKE_INSTALL_PREFIX=${PROJECT_SOURCE_DIR}/deps")
+  find_package(gtest)
+  if(NOT GTEST_FOUND)
+    ExternalProject_Add("googletest"
+                        GIT_REPOSITORY "https://github.com/google/googletest.git"
+                        CMAKE_ARGS "-DCMAKE_INSTALL_PREFIX=${PROJECT_SOURCE_DIR}/deps")
+  endif()
   # Reset variable to false to ensure tacopie does no build tests
   set(BUILD_TESTS false)
 endif(BUILD_TESTS)
