--- origsrc/ticcltools-0.4/Makefile.am	2017-04-04 19:37:38.000000000 +0900
+++ src/ticcltools-0.4/Makefile.am	2018-02-10 21:27:59.322861700 +0900
@@ -9,3 +9,6 @@ EXTRA_DIST = bootstrap.sh AUTHORS TODO N
 
 ChangeLog: NEWS
 	git pull; git2cl > ChangeLog
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/ticcltools-0.4/configure.ac	2017-04-04 19:37:38.000000000 +0900
+++ src/ticcltools-0.4/configure.ac	2018-02-10 21:27:59.322861700 +0900
@@ -80,7 +80,7 @@ CXXFLAGS="$ticcutils_CFLAGS $CXXFLAGS"
 LIBS="$ticcutils_LIBS $LIBS"
 
 
-AC_CONFIG_FILES([
+AC_CONFIG_FILES([ticcltools.pc 
   Makefile
   m4/Makefile
   src/Makefile
--- origsrc/ticcltools-0.4/src/Makefile.am	2017-04-04 19:37:38.000000000 +0900
+++ src/ticcltools-0.4/src/Makefile.am	2018-02-10 21:27:59.339017400 +0900
@@ -8,7 +8,7 @@ bin_PROGRAMS = TICCL-indexer TICCL-index
 
 LDADD = libticcl.la
 lib_LTLIBRARIES = libticcl.la
-libticcl_la_LDFLAGS= -version-info 1:0:0
+libticcl_la_LDFLAGS= -no-undefined -version-info 1:0:0
 
 libticcl_la_SOURCES = word2vec.cxx
 
--- origsrc/ticcltools-0.4/ticcltools.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/ticcltools-0.4/ticcltools.pc.in	2018-02-10 21:27:59.339017400 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: ticcltools
+Description: Tools for TICCL
+Version: @VERSION@
+Libs: -L${libdir} -lticcl
+Cflags: -I${includedir}
+Requires: ticcutils
