--- origsrc/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/Makefile.am	2010-10-11 23:24:31.748113000 +0900
@@ -0,0 +1,5 @@
+lib_LTLIBRARIES = libpnglite.la
+libpnglite_la_SOURCES = pnglite.c
+libpnglite_la_LDFLAGS = -no-undefined
+libpnglite_la_LIBADD = @ZLIB_LIBS@
+include_HEADERS = pnglite.h
--- origsrc/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/configure.ac	2010-10-11 23:24:06.029363000 +0900
@@ -0,0 +1,27 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.65])
+AC_INIT([pnglite], [0.1.17])
+AM_INIT_AUTOMAKE
+AC_CONFIG_SRCDIR([pnglite.c])
+AC_CONFIG_HEADERS([config.h])
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT(win32-dll)
+
+# Checks for libraries.
+PKG_CHECK_MODULES(ZLIB, zlib)
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_CHECK_FUNCS([memset])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
