--- origsrc/pnglite-0.1.17/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/pnglite-0.1.17/Makefile.am	2012-03-24 22:00:17.286539300 +0900
@@ -0,0 +1,6 @@
+include_HEADERS = pnglite.h
+
+lib_LTLIBRARIES = libpnglite.la
+libpnglite_la_SOURCES = pnglite.c
+libpnglite_la_LDFLAGS = -no-undefined
+libpnglite_la_LIBADD = @ZLIB_LIBS@
--- origsrc/pnglite-0.1.17/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/pnglite-0.1.17/configure.ac	2012-03-24 22:01:18.786539300 +0900
@@ -0,0 +1,27 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([pnglite], [0.1.17])
+AC_CONFIG_SRCDIR([pnglite.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT([disable-static])
+
+# Checks for libraries.
+PKG_CHECK_MODULES(ZLIB, zlib)
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_CHECK_FUNCS([memset])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
