--- origsrc/libburn-1.3.2/Makefile.am	2013-08-07 19:09:27.000000000 +0900
+++ src/libburn-1.3.2/Makefile.am	2013-09-26 11:15:15.462571500 +0900
@@ -12,7 +12,7 @@ ACLOCAL_AMFLAGS = -I ./
 
 # Build libraries
 libburn_libburn_la_LDFLAGS = \
-	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
+	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -no-undefined
 # This causes undesired .o names
 # configure.ac appends -D options to variable CFLAG
 ### libburn_libburn_la_CFLAGS = $(LIBBURN_DVD_OBS_64K)
