--- origsrc/libburn-1.1.6/Makefile.am	2011-09-27 15:57:19.000000000 +0900
+++ src/libburn-1.1.6/Makefile.am	2011-10-25 09:05:33.592808200 +0900
@@ -12,7 +12,7 @@ ACLOCAL_AMFLAGS = -I ./
 
 # Build libraries
 libburn_libburn_la_LDFLAGS = \
-	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
+	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -no-undefined
 # This causes undesired .o names
 # configure.ac appends -D options to variable CFLAG
 ### libburn_libburn_la_CFLAGS = $(LIBBURN_DVD_OBS_64K)
