--- origsrc/libburn-1.4.0/Makefile.am	2015-05-17 19:15:45.000000000 +0900
+++ src/libburn-1.4.0/Makefile.am	2015-05-23 23:22:41.719322200 +0900
@@ -12,7 +12,7 @@ ACLOCAL_AMFLAGS = -I ./
 
 # Build libraries
 libburn_libburn_la_LDFLAGS = \
-	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
+	-no-undefined -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 # This causes undesired .o names
 # configure.ac appends -D options to variable CFLAG
 ### libburn_libburn_la_CFLAGS = $(LIBBURN_DVD_OBS_64K)
