--- origsrc/cyphesis-0.6.1/Doxyfile	2013-09-15 00:43:07.000000000 +0900
+++ src/cyphesis-0.6.1/Doxyfile	2013-09-25 20:21:14.354877500 +0900
@@ -610,13 +610,13 @@ WARN_LOGFILE           =
 # directories like "/usr/src/myproject". Separate the files or directories
 # with spaces.
 
-INPUT                  = ../data/ \
-                         ../client/ \
-                         ../common/ \
-                         ../rulesets/ \
-                         ../server/ \
-                         ../tools/ \
-                         ../dox/
+INPUT                  = /cygdrive/c/Users/fd0/src/cyphesis/cyphesis-0.6.1-1bl1/src/cyphesis-0.6.1/data/ \
+                         /cygdrive/c/Users/fd0/src/cyphesis/cyphesis-0.6.1-1bl1/src/cyphesis-0.6.1/client/ \
+                         /cygdrive/c/Users/fd0/src/cyphesis/cyphesis-0.6.1-1bl1/src/cyphesis-0.6.1/common/ \
+                         /cygdrive/c/Users/fd0/src/cyphesis/cyphesis-0.6.1-1bl1/src/cyphesis-0.6.1/rulesets/ \
+                         /cygdrive/c/Users/fd0/src/cyphesis/cyphesis-0.6.1-1bl1/src/cyphesis-0.6.1/server/ \
+                         /cygdrive/c/Users/fd0/src/cyphesis/cyphesis-0.6.1-1bl1/src/cyphesis-0.6.1/tools/ \
+                         /cygdrive/c/Users/fd0/src/cyphesis/cyphesis-0.6.1-1bl1/src/cyphesis-0.6.1/dox/
 
 # This tag can be used to specify the character encoding of the source files
 # that doxygen parses. Internally doxygen uses the UTF-8 encoding, which is
--- origsrc/cyphesis-0.6.1/configure.ac	2013-09-14 23:17:40.000000000 +0900
+++ src/cyphesis-0.6.1/configure.ac	2013-09-25 15:33:11.800302300 +0900
@@ -151,8 +151,11 @@ AC_COMPILE_IFELSE(
 
 dnl General header checks
 AC_CHECK_HEADERS(termios.h signal.h syslog.h sys/utsname.h sys/wait.h)
-AC_CHECK_HEADERS(winsock.h arpa/inet.h sys/un.h dirent.h io.h sys/ucred.h)
-AC_CHECK_HEADERS(windows.h getopt.h)
+AC_CHECK_HEADERS(arpa/inet.h sys/un.h dirent.h io.h sys/ucred.h)
+AC_CHECK_HEADERS(getopt.h)
+if test "$host_os" != "cygwin"; then
+    AC_CHECK_HEADERS(winsock.h windows.h)
+fi
 
 dnl then check for the header file Python.h and set
 dnl python_include_path and python_version
--- origsrc/cyphesis-0.6.1/metaserverapi/MetaServerPacket.hpp	2013-06-02 06:17:16.000000000 +0900
+++ src/cyphesis-0.6.1/metaserverapi/MetaServerPacket.hpp	2013-09-25 20:28:36.312516000 +0900
@@ -35,6 +35,8 @@
 #include <netinet/in.h> // htonl
 #include <array>
 #include <string>
+#include <cstdio>
+#include <cstdlib>
 
 #define MAX_PACKET_BYTES 1024
 #define MAX_UDP_OUT_BYTES 570
--- origsrc/cyphesis-0.6.1/tests/Makefile.am	2013-09-14 23:17:40.000000000 +0900
+++ src/cyphesis-0.6.1/tests/Makefile.am	2013-09-25 20:54:59.715836300 +0900
@@ -24,11 +24,12 @@ PHYSICS_TESTS = BBoxtest Vector3Dtest Qu
 MODULE_TESTS = Locationtest WorldTimetest EntityReftest DateTimetest \
                TerrainContexttest
 
+# AllPropertytest : multiple definition
 RULESETS_TESTS = LocatedEntitytest Entitytest Planttest \
                  Stackabletest Thingtest Worldtest \
                  Charactertest Creatortest ThingupdatePropertiestest \
                  Containertest Tasktest EntityPropertytest \
-                 AllPropertytest Scripttest Motiontest AreaPropertytest \
+                 Scripttest Motiontest AreaPropertytest \
                  BBoxPropertytest CalendarPropertytest \
                  LinePropertytest MindPropertytest \
                  OutfitPropertytest SolidPropertytest \
--- origsrc/cyphesis-0.6.1/tools/cypython.cpp	2013-07-20 06:29:51.000000000 +0900
+++ src/cyphesis-0.6.1/tools/cypython.cpp	2013-09-25 20:33:19.456070700 +0900
@@ -31,6 +31,8 @@
 
 #include <varconf/config.h>
 
+#include <cstdio>
+
 #ifndef READLINE_CXX_SANE   // defined in config.h
 extern "C" {
 #endif
