--- origsrc/uterus-0.2.6/m4/sxe-compiler.m4	2013-05-01 04:44:48.000000000 +0900
+++ src/uterus-0.2.6/m4/sxe-compiler.m4	2013-10-17 12:42:51.041680800 +0900
@@ -337,10 +337,10 @@ AC_DEFUN([SXE_FEATFLAGS], [dnl
 		featflags="${featflags} -static-intel"
 		XCCLDFLAGS="${XCCLDFLAGS} \${XCCFLAG} -static-intel"], [:],
 		[${SXE_CFLAGS}])
-	SXE_CHECK_COMPILER_FLAG([-static-libgcc], [
-		featflags="${featflags} -static-libgcc"
-		XCCLDFLAGS="${XCCLDFLAGS} \${XCCFLAG} -static-libgcc"], [:],
-		[${SXE_CFLAGS}])
+	dnl SXE_CHECK_COMPILER_FLAG([-static-libgcc], [
+	dnl	featflags="${featflags} -static-libgcc"
+	dnl	XCCLDFLAGS="${XCCLDFLAGS} \${XCCFLAG} -static-libgcc"], [:],
+	dnl	[${SXE_CFLAGS}])
 
 	AC_SUBST([XCCLDFLAGS])
 	AC_SUBST([XCCFLAG])
@@ -364,7 +364,7 @@ AC_DEFUN([SXE_CHECK_CFLAGS], [dnl
 
 	## Use either command line flag, environment var, or autodetection
 	CFLAGS=""
-	SXE_DEBUGFLAGS
+	dnl SXE_DEBUGFLAGS
 	SXE_WARNFLAGS
 	SXE_OPTIFLAGS
 	SXE_CFLAGS="$SXE_CFLAGS $debugflags $optiflags $warnflags"
--- origsrc/uterus-0.2.6/src/Makefile.am	2013-05-01 04:44:48.000000000 +0900
+++ src/uterus-0.2.6/src/Makefile.am	2013-10-17 13:03:56.099255500 +0900
@@ -38,7 +38,7 @@
 AM_CFLAGS = $(EXTRA_CFLAGS)
 AM_CPPFLAGS = -D_GNU_SOURCE -D_XOPEN_SOURCE=600 -D_POSIX_C_SOURCE=201001L
 AM_CPPFLAGS += -DDEBUG -DTIMESTAMPS
-AM_LDFLAGS = $(XCCLDFLAGS)
+AM_LDFLAGS = $(XCCLDFLAGS) -no-undefined
 
 ## headerdir and utedir are defined in c.ac
 
@@ -59,7 +59,7 @@ GGO_HELPERS = common.ggo.in
 DSO_LDFLAGS = -shared -avoid-version -module
 
 ## datrie stuff
-noinst_LIBRARIES += libdatrie.a
+#noinst_LIBRARIES += libdatrie.a
 libdatrie_a_SOURCES =
 libdatrie_a_CPPFLAGS = $(AM_CPPFLAGS) -DSTATIC_TRIE_GUTS
 libdatrie_a_SOURCES += trie.c trie.h
@@ -124,7 +124,7 @@ EXTRA_ute_SOURCES =
 ute_SOURCES += ute.c
 ute_LDADD = $(uterus_LIBS)
 ute_CPPFLAGS = $(AM_CPPFLAGS) -DUTEDIR=\"$(utedir)\"
-ute_LDFLAGS = $(AM_LDFLAGS) -static
+ute_LDFLAGS = $(AM_LDFLAGS)
 
 
 UTE_CMD_CPPFLAGS = -DSTANDALONE
@@ -132,14 +132,14 @@ UTE_CMD_CPPFLAGS = -DSTANDALONE
 ute_PROGRAMS += ute-shnot
 ute_shnot_SOURCES = ute-shnot.c ute-shnot.h ute-shnot-clo.ggo
 ute_shnot_CPPFLAGS = $(AM_CPPFLAGS) $(UTE_CMD_CPPFLAGS)
-ute_shnot_LDFLAGS = $(AM_LDFLAGS) -static
+ute_shnot_LDFLAGS = $(AM_LDFLAGS)
 ute_shnot_LDADD = $(uterus_LIBS)
 BUILT_SOURCES += ute-shnot-clo.c ute-shnot-clo.h
 
 ute_PROGRAMS += ute-chndl
 ute_chndl_SOURCES = ute-chndl.c ute-chndl.h ute-chndl-clo.ggo
 ute_chndl_CPPFLAGS = $(AM_CPPFLAGS) $(UTE_CMD_CPPFLAGS)
-ute_chndl_LDFLAGS = $(AM_LDFLAGS) -static
+ute_chndl_LDFLAGS = $(AM_LDFLAGS)
 ute_chndl_LDADD = $(uterus_LIBS)
 BUILT_SOURCES += ute-chndl-clo.c ute-chndl-clo.h
 
@@ -169,28 +169,28 @@ BUILT_SOURCES += ute-mux-clo.c ute-mux-c
 ute_PROGRAMS += ute-fsck
 ute_fsck_SOURCES = ute-fsck.c ute-fsck-clo.ggo
 ute_fsck_CPPFLAGS = $(AM_CPPFLAGS) $(UTE_CMD_CPPFLAGS)
-ute_fsck_LDFLAGS = $(AM_LDFLAGS) -static
+ute_fsck_LDFLAGS = $(AM_LDFLAGS)
 ute_fsck_LDADD = $(uterus_LIBS)
 BUILT_SOURCES += ute-fsck-clo.c ute-fsck-clo.h
 
 ute_PROGRAMS += ute-slut
 ute_slut_SOURCES = ute-slut.c ute-slut-clo.ggo
 ute_slut_CPPFLAGS = $(AM_CPPFLAGS) $(UTE_CMD_CPPFLAGS)
-ute_slut_LDFLAGS = $(AM_LDFLAGS) -static
+ute_slut_LDFLAGS = $(AM_LDFLAGS)
 ute_slut_LDADD = $(uterus_LIBS)
 BUILT_SOURCES += ute-slut-clo.c ute-slut-clo.h
 
 ute_PROGRAMS += ute-slab
 ute_slab_SOURCES = ute-slab.c ute-slab-clo.ggo
 ute_slab_CPPFLAGS = $(AM_CPPFLAGS) $(UTE_CMD_CPPFLAGS)
-ute_slab_LDFLAGS = $(AM_LDFLAGS) -static
+ute_slab_LDFLAGS = $(AM_LDFLAGS)
 ute_slab_LDADD = $(uterus_LIBS)
 BUILT_SOURCES += ute-slab-clo.c ute-slab-clo.h
 
 ute_PROGRAMS += ute-info
 ute_info_SOURCES = ute-info.c ute-info-clo.ggo
 ute_info_CPPFLAGS = $(AM_CPPFLAGS) $(UTE_CMD_CPPFLAGS)
-ute_info_LDFLAGS = $(AM_LDFLAGS) -static
+ute_info_LDFLAGS = $(AM_LDFLAGS)
 ute_info_LDADD = $(uterus_LIBS)
 EXTRA_ute_info_SOURCES = gbs.c gbs.h
 BUILT_SOURCES += ute-info-clo.c ute-info-clo.h
@@ -198,7 +198,7 @@ BUILT_SOURCES += ute-info-clo.c ute-info
 ute_PROGRAMS += ute-anal
 ute_anal_SOURCES = ute-anal.c ute-anal-clo.ggo
 ute_anal_CPPFLAGS = $(AM_CPPFLAGS) $(UTE_CMD_CPPFLAGS)
-ute_anal_LDFLAGS = $(AM_LDFLAGS) -static
+ute_anal_LDFLAGS = $(AM_LDFLAGS)
 if HAVE_LIBPNG
 ute_anal_LDFLAGS += -lpng
 endif  HAVE_LIBPNG
@@ -208,11 +208,14 @@ BUILT_SOURCES += ute-anal-clo.c ute-anal
 ## muxer/demuxer DSOs
 ute_LTLIBRARIES += uta.la
 uta_la_SOURCES = uta.c
-uta_la_LDFLAGS = $(DSO_LDFLAGS) $(XCCLDFLAGS)
+uta_la_LDFLAGS = $(DSO_LDFLAGS) $(AM_LDFLAGS)
+uta_la_LIBADD = libuterus.la
 
 ute_LTLIBRARIES += dukas.la
 dukas_la_SOURCES = dukas.c
-dukas_la_LDFLAGS = $(DSO_LDFLAGS) $(XCCLDFLAGS)
+dukas_la_LDFLAGS = $(DSO_LDFLAGS) $(AM_LDFLAGS)
+dukas_la_LIBADD = libuterus.la
+
 ## dukas has a binary too
 noinst_PROGRAMS += ute-mux-dukas
 ute_mux_dukas_SOURCES = dukas.c dukas-clo.ggo
@@ -223,36 +226,43 @@ BUILT_SOURCES += dukas-clo.c dukas-clo.h
 
 ute_LTLIBRARIES += mat.la
 mat_la_SOURCES = mat.c
-mat_la_LDFLAGS = $(DSO_LDFLAGS) $(XCCLDFLAGS)
+mat_la_LDFLAGS = $(DSO_LDFLAGS) $(AM_LDFLAGS)
+mat_la_LIBADD = libuterus.la
 
 if HAVE_HDF5
 ute_LTLIBRARIES += hdf5.la
 hdf5_la_SOURCES = hdf5.c hdf5-clo.ggo
 hdf5_la_CPPFLAGS = $(AM_CPPFLAGS) $(HDF5_CPPFLAGS)
-hdf5_la_LDFLAGS = $(DSO_LDFLAGS) $(XCCLDFLAGS) $(HDF5_LDFLAGS)
+hdf5_la_LDFLAGS = $(DSO_LDFLAGS) $(AM_LDFLAGS) $(HDF5_LDFLAGS)
+hdf5_la_LIBADD = libuterus.la
+
 endif  ## HAVE_HDF5
 BUILT_SOURCES += hdf5-clo.c hdf5-clo.h
 
 ute_LTLIBRARIES += ariva.la
 ariva_la_SOURCES = ariva.c
-ariva_la_LDFLAGS = $(DSO_LDFLAGS) $(XCCLDFLAGS)
+ariva_la_LDFLAGS = $(DSO_LDFLAGS) $(AM_LDFLAGS)
+ariva_la_LIBADD = libuterus.la
 
 if HAVE_EXPAT
 ute_LTLIBRARIES += ibhist.la
 ibhist_la_SOURCES = ibhist.c
 ibhist_la_SOURCES += expobuf.h
 ibhist_la_CPPFLAGS = $(AM_CPPFLAGS) $(expat_CFLAGS)
-ibhist_la_LDFLAGS = $(DSO_LDFLAGS) $(XCCLDFLAGS)
+ibhist_la_LDFLAGS = $(DSO_LDFLAGS) $(AM_LDFLAGS)
 ibhist_la_LDFLAGS += $(expat_LIBS)
+ibhist_la_LIBADD = libuterus.la
 endif
 
 ute_LTLIBRARIES += forexfactory.la
 forexfactory_la_SOURCES = forexfactory.c
-forexfactory_la_LDFLAGS = $(DSO_LDFLAGS) $(XCCLDFLAGS)
+forexfactory_la_LDFLAGS = $(DSO_LDFLAGS) $(AM_LDFLAGS)
+forexfactory_la_LIBADD = libuterus.la
 
 ute_LTLIBRARIES += fxstreet.la
 fxstreet_la_SOURCES = fxstreet.c
-fxstreet_la_LDFLAGS = $(DSO_LDFLAGS) $(XCCLDFLAGS)
+fxstreet_la_LDFLAGS = $(DSO_LDFLAGS) $(AM_LDFLAGS)
+fxstreet_la_LIBADD = libuterus.la
 
 ## ggo rule
 %.c %.h: %.ggo $(GGO_HELPERS)
--- origsrc/uterus-0.2.6/src/timegm.c	2013-05-01 04:44:48.000000000 +0900
+++ src/uterus-0.2.6/src/timegm.c	2013-10-17 12:55:06.628552400 +0900
@@ -119,7 +119,9 @@ ffff_localtime(struct tm *tm, time_t t,
 {
 	int32_t new_t = zif_local_time(z, t);
 	ffff_gmtime(tm, new_t);
+#if !defined(__CYGWIN__)
 	tm->tm_gmtoff = new_t - t;
+#endif
 	return;
 }
 
--- origsrc/uterus-0.2.6/src/ute-print.h	2013-05-01 04:44:48.000000000 +0900
+++ src/uterus-0.2.6/src/ute-print.h	2013-10-17 12:56:51.566052400 +0900
@@ -92,7 +92,11 @@ pr_tsmstz(char *restrict buf, uint32_t s
 	buf[21] = (char)(((msec / 10) % 10) + '0');
 	buf[22] = (char)(((msec / 1) % 10) + '0');
 	/* compute offset as HHMM */
+#if defined(__CYGWIN__)
+	if (1) {;
+#else
 	if ((off = tm.tm_gmtoff) == 0) {;
+#endif
 		buf[23] = '+';
 		buf[24] = '0';
 		buf[25] = '0';
