--- origsrc/scala-3.3.21/Makefile.am	2015-03-06 18:49:53.059760700 +0900
+++ src/scala-3.3.21/Makefile.am	2015-03-06 18:55:47.338081000 +0900
@@ -1,5 +1,5 @@
 
-LIBS = -lccp4f $(LAPACK_LIBS)
+scala_LDADD = $(LIBCCP4F_LIBS) $(LAPACK_LIBS)
 
 bin_PROGRAMS = scala
 
--- origsrc/scala-3.3.21/configure.ac	2015-03-06 18:49:53.044135700 +0900
+++ src/scala-3.3.21/configure.ac	2015-03-06 18:56:08.791206000 +0900
@@ -2,11 +2,12 @@
 # Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.59)
-AC_INIT(scala, https://bugs.gentoo.org/)
+AC_INIT([scala], [3.3.21])
 AC_CONFIG_SRCDIR([scala.f])
 AM_INIT_AUTOMAKE([foreign])
 
 PKG_CHECK_MODULES([LAPACK],[lapack])
+PKG_CHECK_MODULES([LIBCCP4F], [libccp4f])
 
 # Checks for programs.
 AC_PROG_FC
