--- origsrc/libsigscan-20240505/Makefile.am	2024-05-05 17:03:22.000000000 +0900
+++ src/libsigscan-20240505/Makefile.am	2025-05-20 19:16:52.840320300 +0900
@@ -3,20 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
 	libsigscan \
 	sigscantools \
 	pysigscan \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libsigscan-20240505/configure.ac	2024-05-05 17:03:22.000000000 +0900
+++ src/libsigscan-20240505/configure.ac	2025-05-20 19:16:52.840320300 +0900
@@ -151,16 +151,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
 AC_CONFIG_FILES([libsigscan/Makefile])
 AC_CONFIG_FILES([pysigscan/Makefile])
 AC_CONFIG_FILES([sigscantools/Makefile])
--- origsrc/libsigscan-20240505/pysigscan/Makefile.am	2024-05-05 17:03:22.000000000 +0900
+++ src/libsigscan-20240505/pysigscan/Makefile.am	2025-05-20 19:17:42.076179500 +0900
@@ -43,7 +43,7 @@ pysigscan_la_LIBADD = \
 	@LIBBFIO_LIBADD@
 
 pysigscan_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pysigscan_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pysigscan_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
