--- origsrc/libhildon-2.2.30/Makefile.am	2025-11-13 10:01:50.000000000 +0900
+++ src/libhildon-2.2.30/Makefile.am	2025-11-23 21:19:00.224746400 +0900
@@ -3,8 +3,7 @@ SUBDIRS 				= hildon					\
 					  pkgconfig 					\
 					  examples 					\
 					  tests						\
-					  doc						\
-					  po
+					  doc
 
 MAINTAINERCLEANFILES			= Makefile.in aclocal.m4 configure 		\
 			  		  config.h.in compile 				\
--- origsrc/libhildon-2.2.30/configure.ac	2025-11-13 10:01:50.000000000 +0900
+++ src/libhildon-2.2.30/configure.ac	2025-11-23 21:27:36.577867300 +0900
@@ -1,4 +1,4 @@
-AC_INIT([Hildon gtk widgets], [2.2.24], [hildon-devel@garage.maemo.org], [hildon])
+AC_INIT([Hildon gtk widgets], [2.2.30], [hildon-devel@garage.maemo.org], [hildon])
 AC_CANONICAL_SYSTEM
 
 AM_CONFIG_HEADER(config.h)
--- origsrc/libhildon-2.2.30/hildon/Makefile.am	2025-11-13 10:01:50.000000000 +0900
+++ src/libhildon-2.2.30/hildon/Makefile.am	2025-11-23 21:19:00.230803300 +0900
@@ -1,5 +1,6 @@
 INCLUDES 	= -DG_LOG_DOMAIN=\"hildon-@API_VERSION_MAJOR@\" \
-		  -DLOCALEDIR=\"$(localedir)\"
+		  -DLOCALEDIR=\"$(localedir)\" \
+		  -I$(top_srcdir)
 
 CLEANFILES = \
 		hildon-enum-types.c 				\
@@ -11,7 +12,7 @@ EXTRA_DIST = hildon-marshalers.list
 
 lib_LTLIBRARIES = libhildon-@API_VERSION_MAJOR@.la
 
-libhildon_@API_VERSION_MAJOR@_la_LDFLAGS = $(HILDON_WIDGETS_LT_LDFLAGS)
+libhildon_@API_VERSION_MAJOR@_la_LDFLAGS = -no-undefined $(HILDON_WIDGETS_LT_LDFLAGS)
 
 libhildon_@API_VERSION_MAJOR@_la_LIBADD = \
 		$(GTK_LIBS) 			\
@@ -241,8 +242,8 @@ hildon-enum-types.c: @REBUILD@ $(libhild
 	$(libhildon_$(API_VERSION_MAJOR)_public_headers)) > $@
 
 hildon-marshalers.h: hildon-marshalers.list
-	glib-genmarshal --prefix _hildon_marshal --header hildon-marshalers.list >hildon-marshalers.h
+	glib-genmarshal --prefix _hildon_marshal --header $(srcdir)/hildon-marshalers.list >hildon-marshalers.h
 
 hildon-marshalers.c: hildon-marshalers.list hildon-marshalers.h
 	echo '#include "hildon-marshalers.h"' >hildon-marshalers.c
-	glib-genmarshal --prefix _hildon_marshal --body hildon-marshalers.list >>hildon-marshalers.c
+	glib-genmarshal --prefix _hildon_marshal --body $(srcdir)/hildon-marshalers.list >>hildon-marshalers.c
--- origsrc/libhildon-2.2.30/hildon/hildon-calendar.c	2025-11-13 10:01:50.000000000 +0900
+++ src/libhildon-2.2.30/hildon/hildon-calendar.c	2025-11-23 21:19:00.232802600 +0900
@@ -986,8 +986,13 @@ hildon_calendar_init
     else if (strcmp (year_before, "calendar:MY") != 0)
         g_warning ("Whoever translated calendar:MY did so wrongly.\n");
 #endif
+#if defined(_NL_TIME_FIRST_WEEKDAY)
     langinfo = nl_langinfo (_NL_TIME_FIRST_WEEKDAY);
     first_weekday = langinfo[0];
+#else
+    first_weekday = 1;
+#endif
+#if defined(_NL_TIME_WEEK_1STDAY)
     langinfo = nl_langinfo (_NL_TIME_WEEK_1STDAY);
     week_origin = GPOINTER_TO_UINT (langinfo);
     if (week_origin == 19971130)
@@ -1006,6 +1011,7 @@ hildon_calendar_init
         g_date_free (date);
     }
     else
+#endif
     {
         g_warning ("Invalid value set for _NL_TIME_WEEK_1STDAY"); 
         week_1stday = G_DATE_SUNDAY;
--- origsrc/libhildon-2.2.30/hildon/hildon-weekday-picker.c	2025-11-13 10:01:50.000000000 +0900
+++ src/libhildon-2.2.30/hildon/hildon-weekday-picker.c	2025-11-23 21:19:00.234802700 +0900
@@ -212,8 +212,11 @@ hildon_weekday_picker_init
     sgroup = gtk_size_group_new (GTK_SIZE_GROUP_BOTH);
 
     /* Check our first weekday */
+#if defined(_NL_TIME_FIRST_WEEKDAY)
     day = *nl_langinfo (_NL_TIME_FIRST_WEEKDAY); 
-
+#else
+    day = 1;
+#endif
 
     /* Shift the days by one. This is done because GDateWeekday 
      * starts with Monday(1) and langinfo's first day is Sunday */
