--- origsrc/sv-1.0/Makefile.am	2017-05-23 14:57:06.000000000 +0900
+++ src/sv-1.0/Makefile.am	2018-12-05 15:08:39.100550800 +0900
@@ -30,7 +30,7 @@ libsv_AGE      = @libsv_VERSION_INTERFAC
 include_HEADERS = include/semver.h
 
 lib_LTLIBRARIES  = libsv.la
-libsv_la_LDFLAGS = -version-info $(libsv_CURRENT):$(libsv_REVISION):$(libsv_AGE)
+libsv_la_LDFLAGS = -no-undefined -version-info $(libsv_CURRENT):$(libsv_REVISION):$(libsv_AGE)
 libsv_la_SOURCES = \
 	src/comp.c     \
 	src/id.c       \
