--- origsrc/sv-1.2/Makefile.am	2023-10-07 13:27:44.000000000 +0900
+++ src/sv-1.2/Makefile.am	2025-04-04 18:46:32.534355000 +0900
@@ -30,7 +30,7 @@ libsv_AGE      = @libsv_VERSION_INTERFAC
 include_HEADERS = include/semver.h
 
 lib_LTLIBRARIES  = libsv.la
-libsv_la_LDFLAGS = -version-info $(libsv_CURRENT):$(libsv_REVISION):$(libsv_AGE)
+libsv_la_LDFLAGS = -no-undefined -version-info $(libsv_CURRENT):$(libsv_REVISION):$(libsv_AGE)
 libsv_la_SOURCES = \
 	src/comp.c     \
 	src/id.c       \
--- origsrc/sv-1.2/configure.ac	2023-10-07 13:27:44.000000000 +0900
+++ src/sv-1.2/configure.ac	2025-04-04 18:46:49.834891000 +0900
@@ -2,7 +2,7 @@ dnl @configure_input@
 dnl
 
 AC_PREREQ([2.69])
-AC_INIT([libsv],[0.0.1],,[libsv],[http://github.com/uael/sv])
+AC_INIT([libsv],[1.2],,[libsv],[http://github.com/uael/sv])
 AC_COPYRIGHT([This is free and unencumbered software released into the public domain.
 
 Anyone  is  free  to  copy,  modify, publish,  use,  compile,  sell,  or
--- origsrc/sv-1.2/scripts/libsv.pc.in	2023-10-07 13:27:44.000000000 +0900
+++ src/sv-1.2/scripts/libsv.pc.in	2025-04-04 18:59:17.359136600 +0900
@@ -20,7 +20,7 @@ sysconfdir=@sysconfdir@
 
 Name: @PACKAGE_NAME@
 Description: semantic versioning for the C language
-Version: @LIBSV_PKG_CONFIG_VERSION@
+Version: @VERSION@
 Libs: -L${libdir} -lsv
 Cflags: -I${includedir}
 
