--- origsrc/libaegis-0.1.23/CMakeLists.txt	2024-06-10 18:34:37.000000000 +0900
+++ src/libaegis-0.1.23/CMakeLists.txt	2024-09-16 09:54:18.150874400 +0900
@@ -22,20 +22,24 @@ target_include_directories("${PROJECT_NA
                            PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
                                   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
 
+if (DEFINED SOVERSION)
+  set_target_properties("${PROJECT_NAME}" PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 # Installation
 
 set(TARGETS_EXPORT_NAME "${PROJECT_NAME}-targets")
 
 install(TARGETS "${PROJECT_NAME}"
         EXPORT "${TARGETS_EXPORT_NAME}"
-        LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
         ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 
 install(FILES ${HEADER_FILES} DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
 
 # CMake find_package() support
 
-set(INSTALL_CONFIG_DIR "${CMAKE_INSTALL_DATADIR}/cmake/${PROJECT_NAME}")
+set(INSTALL_CONFIG_DIR "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
 
 set(TARGETS_EXPORT_FILE_NAME "${TARGETS_EXPORT_NAME}.cmake")
 set(TARGETS_EXPORT_FILE "${PROJECT_BINARY_DIR}/${TARGETS_EXPORT_FILE_NAME}")
@@ -57,3 +61,10 @@ install(FILES "${CONFIG_FILE}" DESTINATI
 export(EXPORT "${TARGETS_EXPORT_NAME}" FILE "${TARGETS_EXPORT_FILE}" NAMESPACE "${PROJECT_NAME}::")
 
 export(PACKAGE "${PROJECT_NAME}")
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libaegis.pc.in ${CMAKE_BINARY_DIR}/libaegis.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libaegis.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libaegis-0.1.23/libaegis.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libaegis-0.1.23/libaegis.pc.in	2024-09-16 09:53:34.577628900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libaegis
+Description: Portable C implementations of the AEGIS family of high-performance authenticated encryption algorithms
+Version: @VERSION@
+Libs: -L${libdir} -laegis
+Cflags: -I${includedir}
