--- origsrc/libaegis-0.9.0/CMakeLists.txt	2025-11-25 05:58:31.000000000 +0900
+++ src/libaegis-0.9.0/CMakeLists.txt	2026-01-15 18:39:50.834797600 +0900
@@ -38,14 +38,16 @@ set(TARGETS_EXPORT_NAME "${PROJECT_NAME}
 
 install(TARGETS "${PROJECT_NAME}"
         EXPORT "${TARGETS_EXPORT_NAME}"
-        LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
         ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 
 install(FILES ${HEADER_FILES} DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
 
 # CMake find_package() support
 
-set(INSTALL_CONFIG_DIR "${CMAKE_INSTALL_DATADIR}/cmake/${PROJECT_NAME}")
+if (NOT DEFINED INSTALL_CONFIG_DIR)
+  set(INSTALL_CONFIG_DIR "${CMAKE_INSTALL_DATADIR}/cmake/${PROJECT_NAME}")
+endif()
 
 set(TARGETS_EXPORT_FILE_NAME "${TARGETS_EXPORT_NAME}.cmake")
 set(TARGETS_EXPORT_FILE "${PROJECT_BINARY_DIR}/${TARGETS_EXPORT_FILE_NAME}")
@@ -70,3 +72,10 @@ install(FILES "${PROJECT_BINARY_DIR}/${P
 export(EXPORT "${TARGETS_EXPORT_NAME}" FILE "${TARGETS_EXPORT_FILE}" NAMESPACE "${PROJECT_NAME}::")
 
 export(PACKAGE "${PROJECT_NAME}")
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libaegis.pc.in ${CMAKE_BINARY_DIR}/libaegis.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libaegis.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libaegis-0.9.0/libaegis.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libaegis-0.9.0/libaegis.pc.in	2026-01-15 18:39:42.339661900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libaegis
+Description: Portable C implementations of the AEGIS family of high-performance authenticated encryption algorithms
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -laegis
+Cflags: -I${includedir}
