--- origsrc/libscep-0.09/CMakeLists.txt	2021-08-16 17:35:46.000000000 +0900
+++ src/libscep-0.09/CMakeLists.txt	2024-09-21 13:42:30.376004700 +0900
@@ -49,3 +49,10 @@ foreach (target ${TEST_TARGETS})
 endforeach (target)
 add_test (NAME test_message COMMAND test_message)
 endif ( DEFINED NOTEST )
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libscep.pc.in ${CMAKE_BINARY_DIR}/libscep.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libscep.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libscep-0.09/libscep.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libscep-0.09/libscep.pc.in	2024-09-21 13:39:47.886139400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libscep
+Description: Implementation of the SCEP protocol in C for both client and server
+Version: @libscep_VERSION_MAJOR@.@libscep_VERSION_MINOR@
+Libs: -L${libdir} -lscep
+Cflags: -I${includedir}
--- origsrc/libscep-0.09/src/CMakeLists.txt	2021-08-16 17:35:46.000000000 +0900
+++ src/libscep-0.09/src/CMakeLists.txt	2024-09-21 13:39:47.886139400 +0900
@@ -2,6 +2,9 @@ set (CMAKE_C_FLAGS "-Wall")
 
 add_library (scep SHARED scep.c util.c configuration.c message.c)
 target_link_libraries (scep crypto m pthread rt)
+if (CYGWIN)
+    set_target_properties(scep PROPERTIES SOVERSION ${libscep_VERSION_MAJOR})
+endif()
 if (BUILD_CLI)
 add_subdirectory (clients/cli)
 endif()
--- origsrc/libscep-0.09/src/clients/cli/CMakeLists.txt	2021-08-16 17:35:46.000000000 +0900
+++ src/libscep-0.09/src/clients/cli/CMakeLists.txt	2024-09-21 13:39:47.886139400 +0900
@@ -1,4 +1,4 @@
 add_executable (scep-client scep-client.c scep-client-util.c scep-client.h configuration.c configuration.h)
-target_link_libraries (scep-client scep curl uriparser)
+target_link_libraries (scep-client scep curl uriparser argp)
 
 install(TARGETS scep-client DESTINATION bin)
