--- origsrc/medida/CMakeLists.txt	2020-02-20 11:39:26.000000000 +0900
+++ src/medida/CMakeLists.txt	2020-02-20 11:47:47.006614100 +0900
@@ -97,7 +97,8 @@ set_target_properties(medida PROPERTIES
 )
 
 install(TARGETS medida
-  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin
+  ARCHIVE DESTINATION lib
 )
 
 install(FILES
@@ -169,12 +170,11 @@ if(BUILD_DOCS)
     COMMAND ${DOXYGEN_EXECUTABLE} ${PROJECT_BINARY_DIR}/doxygen.cfg
     SOURCES ${PROJECT_BINARY_DIR}/doxygen.cfg
   )
-endif()
 
-install(DIRECTORY ${PROJECT_BINARY_DIR}/doc/
+  install(DIRECTORY ${PROJECT_BINARY_DIR}/doc/
   DESTINATION share/doc/libmedida-dev/html/
-)
-
+  )
+endif()
 
 ## Examples
 
--- origsrc/medida/libmedida.pc.in	2020-02-20 11:39:26.000000000 +0900
+++ src/medida/libmedida.pc.in	2020-02-20 11:49:54.285493000 +0900
@@ -1,7 +1,7 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=@CMAKE_INSTALL_PREFIX@
-libdir=${prefix}/@LIB_INSTALL_DIR@
-includedir=${prefix}/@INCLUDE_INSTALL_ROOT_DIR@
+libdir=${prefix}/lib
+includedir=${prefix}/include
 
 Name: libmedida
 Description: Medida metrics library for C++
