--- origsrc/naludump-0.1.1/Makefile	2013-11-06 08:02:02.000000000 +0900
+++ src/naludump-0.1.1/Makefile	2019-04-06 09:02:44.369517500 +0900
@@ -27,7 +27,7 @@ $(DEPFILE): Makefile
 # The main program:
 
 naludump: $(OBJS)
-	$(CXX) $(CXXFLAGS) -rdynamic $(OBJS) -o naludump
+	$(CXX) $(CXXFLAGS) -rdynamic $(OBJS) -o naludump -liconv
 
 # Housekeeping:
 
--- origsrc/naludump-0.1.1/naludump.c	2013-11-06 08:02:02.000000000 +0900
+++ src/naludump-0.1.1/naludump.c	2019-04-06 09:02:26.303486800 +0900
@@ -26,6 +26,9 @@
 #include "remux.h"
 #include "tools.h"
 
+#if !defined(O_LARGEFILE)
+#define O_LARGEFILE (0)
+#endif
 
 long long int LastTotalPackets = 0;
 
