--- origsrc/libsemigroups-2.1.5/Makefile.am	2022-04-22 01:59:21.000000000 +0900
+++ src/libsemigroups-2.1.5/Makefile.am	2022-05-26 09:01:02.444964500 +0900
@@ -11,8 +11,10 @@ AM_CXXFLAGS += $(EIGEN3_CFLAGS)
 if LIBSEMIGROUPS_HPCOMBI_ENABLED
 AM_CXXFLAGS += $(HPCOMBI_CXXFLAGS)
 endif
+AM_CXXFLAGS += -I$(top_builddir)/include
+AM_CXXFLAGS += -I$(top_builddir)/include/libsemigroups
 
-AM_LDFLAGS  =  -no-undefined -lpthread $(FMT_LIBS)
+AM_LDFLAGS  =  -no-undefined $(FMT_LIBS)
 
 if LIBSEMIGROUPS_DEBUG
 AM_CPPFLAGS = -DDEBUG
