--- origsrc/libsemigroups-0.6.7/Makefile.am	2019-03-15 19:31:59.000000000 +0900
+++ src/libsemigroups-0.6.7/Makefile.am	2019-04-29 09:45:25.033149300 +0900
@@ -2,7 +2,7 @@
 # Automake configuration
 ##############################################################################
 
-AM_CXXFLAGS =  -I. -Wall -Wextra -pedantic -Wno-c++11-extensions -std=gnu++11
+AM_CXXFLAGS =  -I$(top_builddir)/src -I. -Wall -Wextra -pedantic -Wno-c++11-extensions -std=gnu++11
 AM_CXXFLAGS += -g -O3 -march=native
 AM_LDFLAGS  =  -no-undefined
 
