--- origsrc/libsemigroups-1.0.7/Makefile.am	2020-03-17 17:23:50.000000000 +0900
+++ src/libsemigroups-1.0.7/Makefile.am	2020-03-24 17:27:31.270720800 +0900
@@ -6,11 +6,13 @@ AM_CXXFLAGS =  -I$(abs_top_srcdir)
 AM_CXXFLAGS += -I$(abs_top_srcdir)/include
 AM_CXXFLAGS += -I$(abs_top_srcdir)/extern/HPCombi/include
 AM_CXXFLAGS += -I$(abs_top_srcdir)/extern/HPCombi/include/fallback
+AM_CXXFLAGS += -I$(top_builddir)/include
+AM_CXXFLAGS += -I$(top_builddir)/include/libsemigroups
 AM_CXXFLAGS += -std=gnu++11 -g -O3 -Wall -Wextra
 AM_CXXFLAGS += $(WARNING_CXXFLAGS)
 AM_CXXFLAGS += $(FMT_CFLAGS)
 
-AM_LDFLAGS  =  -no-undefined -lpthread
+AM_LDFLAGS  =  -no-undefined
 
 if LIBSEMIGROUPS_DEBUG
 AM_CPPFLAGS = -DDEBUG
@@ -105,6 +107,7 @@ pkginclude_HEADERS += extern/HPCombi/inc
 pkginclude_HEADERS += extern/HPCombi/include/vect_generic.hpp
 endif
 
+if !LIBSEMIGROUPS_FMT
 fmtincludedir = $(includedir)/libsemigroups/fmt
 fmtinclude_HEADERS =  extern/fmt-5.3.0/include/fmt/chrono.h  
 fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/color.h  
@@ -117,6 +120,7 @@ fmtinclude_HEADERS += extern/fmt-5.3.0/i
 fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/printf.h  
 fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/ranges.h  
 fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/time.h
+endif
 
 lib_LTLIBRARIES = libsemigroups.la
 
--- origsrc/libsemigroups-1.0.7/m4/ax_check_fmt.m4	2020-03-17 17:23:50.000000000 +0900
+++ src/libsemigroups-1.0.7/m4/ax_check_fmt.m4	2020-03-24 17:27:31.301969400 +0900
@@ -28,4 +28,5 @@ AC_DEFUN([AX_CHECK_FMT], [
 	AC_SUBST(FMT_CFLAGS, ['-I$(srcdir)/extern/fmt-5.3.0/include'])
         
   fi
+  AM_CONDITIONAL([LIBSEMIGROUPS_FMT], [test x"$with_external_fmt" = xyes])
 ])
