--- origsrc/libsemigroups-3.4.0/Makefile.am	2026-01-17 20:40:32.000000000 +0900
+++ src/libsemigroups-3.4.0/Makefile.am	2026-01-20 23:27:09.938373200 +0900
@@ -16,8 +16,10 @@ AM_CXXFLAGS += $(EIGEN3_CFLAGS)
 if LIBSEMIGROUPS_HPCOMBI_ENABLED
 AM_CXXFLAGS += $(HPCOMBI_CXXFLAGS)
 endif
+AM_CXXFLAGS += -I$(top_builddir)/include
+AM_CXXFLAGS += -I$(top_builddir)/include/libsemigroups
 
-AM_LDFLAGS  =  -no-undefined -lpthread $(FMT_LIBS)
+AM_LDFLAGS  =  -no-undefined $(FMT_LIBS)
 
 if LIBSEMIGROUPS_DEBUG
 AM_CPPFLAGS = -DDEBUG
