--- origsrc/libsemigroups-2.0.2/Makefile.am	2021-09-21 01:53:48.000000000 +0900
+++ src/libsemigroups-2.0.2/Makefile.am	2021-10-04 10:29:43.866171800 +0900
@@ -11,8 +11,10 @@ AM_CXXFLAGS += $(EIGEN3_CFLAGS)
 if LIBSEMIGROUPS_HPCOMBI_ENABLED
 AM_CXXFLAGS += $(HPCOMBI_CXXFLAGS)
 endif
+AM_CXXFLAGS += -I$(top_builddir)/include
+AM_CXXFLAGS += -I$(top_builddir)/include/libsemigroups
 
-AM_LDFLAGS  =  -no-undefined -lpthread $(FMT_LIBS)
+AM_LDFLAGS  =  -no-undefined $(FMT_LIBS)
 
 if LIBSEMIGROUPS_DEBUG
 AM_CPPFLAGS = -DDEBUG
