--- origsrc/libsemigroups-1.2.1/Makefile.am	2020-06-28 22:50:20.000000000 +0900
+++ src/libsemigroups-1.2.1/Makefile.am	2020-07-15 17:59:46.363799100 +0900
@@ -7,8 +7,10 @@ AM_CXXFLAGS += -I$(abs_top_srcdir)/exter
 AM_CXXFLAGS += -std=gnu++11 -g -O3 -Wall -Wextra
 AM_CXXFLAGS += $(WARNING_CXXFLAGS)
 AM_CXXFLAGS += $(FMT_CFLAGS)
+AM_CXXFLAGS += -I$(top_builddir)/include
+AM_CXXFLAGS += -I$(top_builddir)/include/libsemigroups
 
-AM_LDFLAGS  =  -no-undefined -lpthread
+AM_LDFLAGS  =  -no-undefined
 
 if LIBSEMIGROUPS_DEBUG
 AM_CPPFLAGS = -DDEBUG
@@ -112,6 +114,7 @@ hpcombifallbackinclude_HEADERS =  extern
 hpcombifallbackinclude_HEADERS += extern/HPCombi/include/fallback/seq.hpp
 endif
 
+if !LIBSEMIGROUPS_FMT
 fmtincludedir = $(includedir)/libsemigroups/fmt
 fmtinclude_HEADERS =  extern/fmt-5.3.0/include/fmt/chrono.h  
 fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/color.h  
@@ -124,6 +127,7 @@ fmtinclude_HEADERS += extern/fmt-5.3.0/i
 fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/printf.h  
 fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/ranges.h  
 fmtinclude_HEADERS += extern/fmt-5.3.0/include/fmt/time.h
+endif
 
 lib_LTLIBRARIES = libsemigroups.la
 
--- origsrc/libsemigroups-1.2.1/m4/ax_check_fmt.m4	2020-06-28 22:50:20.000000000 +0900
+++ src/libsemigroups-1.2.1/m4/ax_check_fmt.m4	2020-07-15 17:59:46.379417800 +0900
@@ -28,4 +28,5 @@ AC_DEFUN([AX_CHECK_FMT], [
 	AC_SUBST(FMT_CFLAGS, ['-I$(srcdir)/extern/fmt-5.3.0/include'])
         
   fi
+  AM_CONDITIONAL([LIBSEMIGROUPS_FMT], [test x"$with_external_fmt" = xyes])
 ])
