--- origsrc/libsemigroups-0.6.2/Makefile.am	2018-01-12 18:03:15.000000000 +0900
+++ src/libsemigroups-0.6.2/Makefile.am	2018-01-13 20:46:26.296000000 +0900
@@ -2,7 +2,7 @@
 # Automake configuration
 ##############################################################################
 
-AM_CXXFLAGS =  -I. -Wall -Wextra -pedantic -Wno-c++11-extensions -std=c++11 
+AM_CXXFLAGS =  -I$(top_builddir)/src -I. -Wall -Wextra -pedantic -Wno-c++11-extensions -std=c++11 
 AM_CXXFLAGS += -g -O3 -march=native
 AM_LDFLAGS  =  -no-undefined
 
@@ -151,3 +151,6 @@ doc:
 	@echo "Generating documentation . . ."; \
 	doxygen Doxyfile
 	@echo "See: html/index.html";
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libsemigroups-0.6.2/configure.ac	2018-01-12 18:03:15.000000000 +0900
+++ src/libsemigroups-0.6.2/configure.ac	2018-01-13 20:05:52.632920300 +0900
@@ -78,5 +78,5 @@ AX_CODE_COVERAGE()
 
 dnl Output configured files
 
-AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([libsemigroups.pc Makefile])
 AC_OUTPUT
--- origsrc/libsemigroups-0.6.2/libsemigroups.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libsemigroups-0.6.2/libsemigroups.pc.in	2018-01-13 20:05:52.601598800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libsemigroups
+Description: C++ library for semigroups and monoids
+Version: @VERSION@
+Libs: -L${libdir} -lsemigroups
+Cflags: -I${includedir}
