--- origsrc/ink-0.5.3/configure.in	2018-07-14 07:53:42.000000000 +0900
+++ src/ink-0.5.3/configure.in	2022-11-20 23:32:13.006936300 +0900
@@ -25,7 +25,7 @@ AC_DEFINE_UNQUOTED(LINUX,1,[Define value
 AC_DEFINE_UNQUOTED(FREEBSD,3,[Define value for FreeBSD OS])
 
 case $host in
-  *-*-linux*) AC_DEFINE_UNQUOTED(HOST_OS,LINUX,[Host OS]);;
+  *-*-linux*|*-*-cygwin*) AC_DEFINE_UNQUOTED(HOST_OS,LINUX,[Host OS]);;
   *-*-freebsd*) AC_DEFINE(HOST_OS,FREEBSD,[HOST OS])
                 CFLAGS="$CFLAGS -I/usr/local/include"
                 LDFLAGS="$LDFLAGS -L/usr/local/lib";;
