--- origsrc/libxlsxwriter-RELEASE_1.0.6/CMakeLists.txt	2021-05-28 15:05:13.000000000 +0000
+++ src/libxlsxwriter-RELEASE_1.0.6/CMakeLists.txt	2021-05-29 22:48:09.890953700 +0000
@@ -202,14 +202,24 @@ configure_file(dev/release/pkg-config.tx
 enable_language(CXX)
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
+find_package(pkgconfig REQUIRED)
+
 # ZLIB
-find_package(ZLIB REQUIRED "1.0")
+if (CYGWIN)
+    pkg_check_modules(ZLIB REQUIRED zlib>=1.0)
+else()
+    find_package(ZLIB REQUIRED "1.0")
+endif()
 list(APPEND LXW_PRIVATE_INCLUDE_DIRS ${ZLIB_INCLUDE_DIRS})
 message("zlib version: " ${ZLIB_VERSION})
 
 # MINIZIP
 if (USE_SYSTEM_MINIZIP)
-    find_package(MINIZIP REQUIRED "1.0")
+    if (CYGWIN)
+        pkg_check_modules(MINIZIP REQUIRED minizip>=1.0)
+    else()
+        find_package(MINIZIP REQUIRED "1.0")
+    endif()
     list(APPEND LXW_PRIVATE_INCLUDE_DIRS ${MINIZIP_INCLUDE_DIRS})
 endif()
 
@@ -274,6 +284,10 @@ target_include_directories(${PROJECT_NAM
     PUBLIC include include/xlsxwriter
 )
 
+if (CYGWIN)
+    set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION 0)
+endif()
+
 # TESTS
 # -----
 
--- origsrc/libxlsxwriter-RELEASE_1.0.6/cmake/FindPackage.cmake	2021-05-28 15:05:13.000000000 +0000
+++ src/libxlsxwriter-RELEASE_1.0.6/cmake/FindPackage.cmake	2021-05-29 22:48:09.890953700 +0000
@@ -43,7 +43,7 @@ macro(SetSuffixes packageName)
             set(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
         endif()
     else()
-        if(WIN32)
+        if(WIN32 OR CYGWIN)
             set(CMAKE_FIND_LIBRARY_SUFFIXES ".dll.a" ".dll" ".lib" ".a")
         else()
             set(CMAKE_FIND_LIBRARY_SUFFIXES ".so" ".a")
--- origsrc/libxlsxwriter-RELEASE_1.0.6/include/xlsxwriter/packager.h	2021-05-28 15:05:13.000000000 +0000
+++ src/libxlsxwriter-RELEASE_1.0.6/include/xlsxwriter/packager.h	2021-05-29 22:48:09.890953700 +0000
@@ -12,7 +12,12 @@
 #include <stdint.h>
 
 #ifdef USE_SYSTEM_MINIZIP
+#if defined(__CYGWIN__)
+#include "mz_compat.h"
+#include "zlib.h"
+#else
 #include "minizip/zip.h"
+#endif
 #else
 #include "third_party/zip.h"
 #endif
