--- origsrc/libibsimu-1.0.5b/src/Makefile.am	2011-10-07 19:40:01.000000000 +0900
+++ src/libibsimu-1.0.5b/src/Makefile.am	2014-07-21 13:33:07.250949500 +0900
@@ -1,6 +1,6 @@
 lib_LTLIBRARIES = libibsimu-1.0.5b.la
 
-libibsimu_1_0_5b_la_LDFLAGS = -version-info 0:1:0
+libibsimu_1_0_5b_la_LDFLAGS = -no-undefined -version-info 0:1:0
 
 AM_LDFLAGS = -Wall @rdynamic_flag@
 AM_CPPFLAGS = -Wall
--- origsrc/libibsimu-1.0.5b/src/error.cpp	2011-09-26 18:58:14.000000000 +0900
+++ src/libibsimu-1.0.5b/src/error.cpp	2014-07-21 13:33:07.250949500 +0900
@@ -47,6 +47,10 @@
 #include "error.hpp"
 #include "ibsimu.hpp"
 
+#if defined(__CYGWIN__)
+#undef _GNU_SOURCE
+#endif
+
 #if defined(_GNU_SOURCE) && defined(HAVE_SIGINFO_T)
 
 #include <execinfo.h>
--- origsrc/libibsimu-1.0.5b/src/fielddiagplot.hpp	2011-06-20 15:19:33.000000000 +0900
+++ src/libibsimu-1.0.5b/src/fielddiagplot.hpp	2014-07-21 13:33:07.250949500 +0900
@@ -52,6 +52,9 @@
 #include "scalarfield.hpp"
 #include "vectorfield.hpp"
 
+#if defined(__CYGWIN__)
+#undef _N
+#endif
 
 #define FIELDD_DIAG_NONE FIELD_NONE
 #define FIELDD_DIAG_EPOT FIELD_EPOT
--- origsrc/libibsimu-1.0.5b/src/ibsimu.cpp	2011-10-07 20:29:42.000000000 +0900
+++ src/libibsimu-1.0.5b/src/ibsimu.cpp	2014-07-21 13:33:07.266574500 +0900
@@ -49,7 +49,9 @@
 #include "timer.hpp"
 #include "error.hpp"
 
-
+#if defined(__CYGWIN__)
+#undef _GNU_SOURCE
+#endif
 
 IBSimu::IBSimu()
     : _hello(false), _verbose_output(0), _threadcount(1), _is_cout(true), _vout(&std::cout)
--- origsrc/libibsimu-1.0.5b/src/ilu0_precond.hpp	2011-06-20 15:19:34.000000000 +0900
+++ src/libibsimu-1.0.5b/src/ilu0_precond.hpp	2014-07-21 13:33:07.266574500 +0900
@@ -49,6 +49,10 @@
 #include "ccolmatrix.hpp"
 #include "precond.hpp"
 
+#if defined(__CYGWIN__)
+#undef _U
+#undef _L
+#endif
 
 /*! \brief Zero fill-in incomplete LU preconditioner class.
  */
--- origsrc/libibsimu-1.0.5b/src/random.hpp	2011-06-20 15:19:34.000000000 +0900
+++ src/libibsimu-1.0.5b/src/random.hpp	2014-07-21 13:33:07.282199500 +0900
@@ -51,7 +51,9 @@
 #include <gsl/gsl_qrng.h>
 #include <gsl/gsl_roots.h>
 
-
+#if defined(__CYGWIN__)
+#undef _N
+#endif
 
 /*! \brief Base class for non-uniform random variate transformation.
  */
--- origsrc/libibsimu-1.0.5b/src/readascii.hpp	2011-09-26 18:58:14.000000000 +0900
+++ src/libibsimu-1.0.5b/src/readascii.hpp	2014-07-21 13:33:07.297824500 +0900
@@ -47,6 +47,9 @@
 #include <vector>
 #include <stdint.h>
 
+#if defined(__CYGWIN__)
+#undef _N
+#endif
 
 /*! \brief Class for reading ASCII data files.
  *
--- origsrc/libibsimu-1.0.5b/src/trajectory.hpp	2011-06-20 15:19:34.000000000 +0900
+++ src/libibsimu-1.0.5b/src/trajectory.hpp	2014-07-21 13:33:07.297824500 +0900
@@ -43,6 +43,10 @@
 #ifndef TRAJECTORY_HPP
 #define TRAJECTORY_HPP 1
 
+#if defined(__CYGWIN__)
+#undef _B
+#undef _C
+#endif
 
 enum trajectory_rep_e {
     TRAJ_EMPTY = 0,
