--- origsrc/libtmx-0.4.0/src/lib/CMakeLists.txt	2015-08-19 07:14:53.000000000 +0900
+++ src/libtmx-0.4.0/src/lib/CMakeLists.txt	2025-04-21 22:11:18.080095500 +0900
@@ -28,6 +28,6 @@ set_target_properties(tmx0
 
 install(
   TARGETS tmx0
-  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin
   ARCHIVE DESTINATION lib
 )
--- origsrc/libtmx-0.4.0/src/lib/Parser.cc	2015-08-19 07:14:53.000000000 +0900
+++ src/libtmx-0.4.0/src/lib/Parser.cc	2025-04-21 22:11:18.080095500 +0900
@@ -39,6 +39,10 @@
 #include <tmx/TileLayer.h>
 #include <tmx/TileSet.h>
 
+#if defined(__CYGWIN__)
+#define XML_NO_ERROR XML_SUCCESS
+#endif
+
 #define INVALID static_cast<unsigned>(-1)
 
 namespace fs = boost::filesystem;
