--- origsrc/lmfit-5.1/demo/Makefile.am	2013-07-18 17:11:02.000000000 +0900
+++ src/lmfit-5.1/demo/Makefile.am	2013-09-19 08:47:11.302515900 +0900
@@ -1,6 +1,6 @@
 AM_CFLAGS = -I$(srcdir)/../lib
-noinst_PROGRAMS = curve1 nonlin1 surface1
+bin_PROGRAMS = curve1 nonlin1 surface1
 curve1_SOURCES = curve1.c
 nonlin1_SOURCES = nonlin1.c
 surface1_SOURCES = surface1.c
-AM_LDFLAGS = -lm -L../lib -llmfit
+AM_LDFLAGS = -lm $(top_builddir)/lib/liblmfit.la
--- origsrc/lmfit-5.1/lib/Makefile.am	2013-07-18 06:23:14.000000000 +0900
+++ src/lmfit-5.1/lib/Makefile.am	2013-09-19 08:44:06.969508100 +0900
@@ -1,4 +1,4 @@
 include_HEADERS = lmstruct.h lmmin.h lmcurve.h
 lib_LTLIBRARIES = liblmfit.la
 liblmfit_la_SOURCES = $(include_HEADERS) lmmin.c lmcurve.c
-liblmfit_la_LDFLAGS = -lm -version-info $(VERSION)
+liblmfit_la_LDFLAGS = -no-undefined -lm -version-info $(VERSION)
