--- origsrc/lmfit-3.5/lib/Makefile.am	2013-01-17 04:30:29.000000000 +0900
+++ src/lmfit-3.5/lib/Makefile.am	2013-06-09 09:54:52.165863000 +0900
@@ -1,4 +1,4 @@
 include_HEADERS = lmmin.h lmcurve.h
 lib_LTLIBRARIES = liblmmin.la
 liblmmin_la_SOURCES = lmmin.c lmmin.h lmcurve.c lmcurve.h
-liblmmin_la_LDFLAGS = -lm -version-info $(VERSION)
+liblmmin_la_LDFLAGS = -no-undefined -lm -version-info $(VERSION)
