--- origsrc/SILLY-0.1.0/configure.ac	2006-11-05 23:03:06.000000000 +0900
+++ src/SILLY-0.1.0/configure.ac	2010-11-28 16:59:46.886341000 +0900
@@ -33,9 +33,7 @@ AC_PROG_CXX
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
 dnl make libtool aware we may be building Win32 DLLs
-AC_LIBTOOL_WIN32_DLL
-AC_PROG_LIBTOOL
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 dnl is this is on mingw32?
 AC_MINGW32
--- origsrc/SILLY-0.1.0/src/Makefile.am	2006-11-04 23:27:14.000000000 +0900
+++ src/SILLY-0.1.0/src/Makefile.am	2010-11-28 17:00:02.719107400 +0900
@@ -3,7 +3,7 @@ INCLUDES=-I$(top_srcdir)/include
 lib_LTLIBRARIES = libSILLY.la
 libSILLY_la_CXXFLAGS= -DUSE_SILLYOPTIONS_H
 libSILLY_la_LIBADD=
-libSILLY_la_LDFLAGS= -version-info @SILLY_VERSION_INFO@
+libSILLY_la_LDFLAGS= -version-info @SILLY_VERSION_INFO@ -no-undefined
 libSILLY_la_SOURCES= \
   SILLYDataSource.cpp \
   SILLYFileDataSource.cpp \
@@ -17,7 +17,6 @@ libSILLY_la_SOURCES= \
 
 if SILLY_MINGW32
 libSILLY_la_CXXFLAGS += -DSILLY_EXPORTS
-libSILLY_la_LDFLAGS += -no-undefined
 endif
 
 if SILLY_HAVE_JPG
