--- origsrc/string_theory-3.8/test/CMakeLists.txt	2023-08-10 08:50:06.000000000 +0900
+++ src/string_theory-3.8/test/CMakeLists.txt	2024-01-14 09:03:39.573567600 +0900
@@ -32,11 +32,14 @@ set(gtest_hide_internal_symbols ON CACHE
 
 set(BUILD_GMOCK OFF CACHE INTERNAL "Override gtest default" FORCE)
 set(INSTALL_GTEST OFF CACHE INTERNAL "Override gtest default" FORCE)
-add_subdirectory(gtest EXCLUDE_FROM_ALL)
+find_package(gtest)
+if(NOT GTEST_FOUND)
+    add_subdirectory(gtest EXCLUDE_FROM_ALL)
 
-if(CMAKE_COMPILER_IS_GNUCXX)
-    # Squash an error in gtest that surfaced in GCC 12
-    target_compile_options(gtest PRIVATE -Wno-restrict)
+    if(CMAKE_COMPILER_IS_GNUCXX)
+        # Squash an error in gtest that surfaced in GCC 12
+        target_compile_options(gtest PRIVATE -Wno-restrict)
+    endif()
 endif()
 
 add_executable(st_gtests "")
