--- origsrc/glm/CMakeLists.txt	2016-02-21 22:44:18.000000000 +0900
+++ src/glm/CMakeLists.txt	2016-03-10 21:07:16.935959400 +0900
@@ -105,7 +105,7 @@ elseif(GLM_TEST_ENABLE_SIMD_SSE3)
 	elseif(GLM_USE_INTEL)
 		add_definitions(/QxSSE3)
 	elseif(MSVC)
-		add_definitions(/arch:SSE2) # VC doesn't support /arch:SSE3	
+		add_definitions(/arch:SSE2) # VC doesn't support /arch:SSE3
 	endif()
 elseif(GLM_TEST_ENABLE_SIMD_SSE2)
 	if(CMAKE_COMPILER_IS_GNUCXX)
@@ -126,11 +126,11 @@ if(GLM_TEST_ENABLE_FAST_MATH)
 	endif()
 
 	if(MSVC)
-		add_definitions(/fp:fast)	
+		add_definitions(/fp:fast)
 	endif()
 elseif(NOT GLM_TEST_ENABLE_FAST_MATH)
 	if(MSVC)
-		add_definitions(/fp:precise)	
+		add_definitions(/fp:precise)
 	endif()
 endif()
 
@@ -150,7 +150,7 @@ add_subdirectory(glm)
 add_subdirectory(test)
 
 set(GLM_INSTALL_CONFIGDIR "${CMAKE_INSTALL_LIBDIR}/cmake/glm")
-install(DIRECTORY glm DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(DIRECTORY glm DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} PATTERN CMakeLists.txt EXCLUDE)
 
 write_basic_package_version_file(
     "${CMAKE_CURRENT_BINARY_DIR}/glmVersion.cmake"
@@ -175,7 +175,7 @@ configure_package_config_file(
 )
 
 install(
-    FILES 
+    FILES
         "${CMAKE_CURRENT_BINARY_DIR}/${GLM_INSTALL_CONFIGDIR}/glmConfig.cmake"
         "${CMAKE_CURRENT_BINARY_DIR}/glmVersion.cmake"
     DESTINATION ${GLM_INSTALL_CONFIGDIR}
@@ -193,7 +193,7 @@ if (NOT CMAKE_VERSION VERSION_LESS "3.0"
         EXPORT glmTargets
         FILE "${CMAKE_CURRENT_BINARY_DIR}/glmTargets.cmake"
     )
-    
+
     install(
         EXPORT glmTargets FILE glmTargets.cmake
         DESTINATION ${GLM_INSTALL_CONFIGDIR}
