diff --strip-trailing-cr -uprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.log -x config.status -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/glm-0.9.4.3/CMakeLists.txt src/glm-0.9.4.3/CMakeLists.txt
--- origsrc/glm-0.9.4.3/CMakeLists.txt	2013-03-17 11:34:30.000000000 +0900
+++ src/glm-0.9.4.3/CMakeLists.txt	2013-04-28 22:46:41.531425500 +0900
@@ -90,4 +90,4 @@ include_directories("./test/external")
 add_subdirectory(glm)
 add_subdirectory(test)
 
-install(DIRECTORY glm DESTINATION include)
+install(DIRECTORY glm DESTINATION include PATTERN CMakeLists.txt EXCLUDE)
