--- origsrc/hayai-1.0.2/CMakeLists.txt	2018-03-02 03:34:23.000000000 +0900
+++ src/hayai-1.0.2/CMakeLists.txt	2018-03-15 10:17:29.021000000 +0900
@@ -46,10 +46,13 @@ add_subdirectory(src)
 add_subdirectory(sample)
 
 # Include tests if requested by the user passing -DTEST=true.
+option(USE_SYSTEM_GTEST "use system gtest" ON)
 if (${TEST})
   enable_testing()
 
-  add_subdirectory(vendor/gtest)
+  if (NOT USE_SYSTEM_GTEST)
+    add_subdirectory(vendor/gtest)
+  endif()
   add_subdirectory(tests)
 endif (${TEST})
 
@@ -93,3 +96,10 @@ install(
   DESTINATION "${INSTALL_CMAKE_DIR}"
   COMPONENT dev
 )
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/hayai.pc.in ${CMAKE_BINARY_DIR}/hayai.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/hayai.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/hayai-1.0.2/hayai.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/hayai-1.0.2/hayai.pc.in	2018-03-15 10:21:10.826000000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: hayai
+Description: C++ benchmarking framework inspired by the googletest framework
+Version: @VERSION@
+Libs: -L${libdir} -lhayai_main
+Cflags: -I${includedir}
--- origsrc/hayai-1.0.2/src/CMakeLists.txt	2018-03-02 03:34:23.000000000 +0900
+++ src/hayai-1.0.2/src/CMakeLists.txt	2018-03-15 10:15:20.654264300 +0900
@@ -17,7 +17,7 @@ file(GLOB hayai_headers
   hayai_main.hpp
 )
 
-add_library(hayai_main
+add_library(hayai_main SHARED
   hayai_posix_main.cpp
 )
 
@@ -25,6 +25,10 @@ set_target_properties(hayai_main PROPERT
   PUBLIC_HEADER "${hayai_headers}"
 )
 
+if (CYGWIN)
+  set_target_properties(hayai_main PROPERTIES SOVERSION 0)
+endif()
+
 install(
   TARGETS hayai_main
   EXPORT hayai-targets
--- origsrc/hayai-1.0.2/src/hayai.hpp	2018-03-02 03:34:23.000000000 +0900
+++ src/hayai-1.0.2/src/hayai.hpp	2018-03-15 10:17:47.560000000 +0900
@@ -10,7 +10,7 @@
 #include "hayai_junit_xml_outputter.hpp"
 
 
-#define HAYAI_VERSION "1.0.1"
+#define HAYAI_VERSION "1.0.2"
 
 
 #define BENCHMARK_CLASS_NAME_(fixture_name, benchmark_name) \
--- origsrc/hayai-1.0.2/tests/CMakeLists.txt	2018-03-02 03:34:23.000000000 +0900
+++ src/hayai-1.0.2/tests/CMakeLists.txt	2018-03-15 10:10:39.976000000 +0900
@@ -7,10 +7,11 @@ include_directories(
   ${PROJECT_SOURCE_DIR}/src
 )
 
-include_directories(
-  ${PROJECT_SOURCE_DIR}/src
-  ${PROJECT_SOURCE_DIR}/vendor/gtest/include
-)
+if (NOT USE_SYSTEM_GTEST)
+  include_directories(
+    ${PROJECT_SOURCE_DIR}/vendor/gtest/include
+  )
+endif()
 
 add_executable(tests
   hayai_test_parameter_descriptor.cpp
@@ -21,4 +22,8 @@ target_link_libraries(tests
   ${LIB_TIMING}
 )
 
+if (USE_SYSTEM_GTEST)
+  target_link_libraries(tests gtest)
+endif()
+
 add_test(HayaiTests tests)
