--- origsrc/libmbim-1.24.4/src/libmbim-glib/Makefile.am	2020-07-10 17:12:29.000000000 +0900
+++ src/libmbim-1.24.4/src/libmbim-glib/Makefile.am	2020-10-06 18:42:05.008527100 +0900
@@ -44,7 +44,7 @@ libmbim_glib_la_LIBADD = \
 	${top_builddir}/src/libmbim-glib/generated/libmbim-glib-generated.la \
 	$(NULL)
 
-libmbim_glib_la_LDFLAGS = \
+libmbim_glib_la_LDFLAGS = -no-undefined \
 	$(WARN_LDFLAGS) \
 	-version-info $(MBIM_GLIB_LT_CURRENT):$(MBIM_GLIB_LT_REVISION):$(MBIM_GLIB_LT_AGE) \
 	-export-symbols-regex '^mbim_.*' \
