--- origsrc/libmbim-1.24.2/src/libmbim-glib/Makefile.am	2020-05-13 17:03:13.000000000 +0900
+++ src/libmbim-1.24.2/src/libmbim-glib/Makefile.am	2020-07-15 16:35:12.987883700 +0900
@@ -44,7 +44,7 @@ libmbim_glib_la_LIBADD = \
 	${top_builddir}/src/libmbim-glib/generated/libmbim-glib-generated.la \
 	$(NULL)
 
-libmbim_glib_la_LDFLAGS = \
+libmbim_glib_la_LDFLAGS = -no-undefined \
 	$(WARN_LDFLAGS) \
 	-version-info $(MBIM_GLIB_LT_CURRENT):$(MBIM_GLIB_LT_REVISION):$(MBIM_GLIB_LT_AGE) \
 	-export-symbols-regex '^mbim_.*' \
