--- origsrc/libstomp/Makefile.am	2025-05-01 13:25:35.000000000 +0900
+++ src/libstomp/Makefile.am	2025-05-01 13:25:37.211116400 +0900
@@ -1 +1,4 @@
 SUBDIRS = src example
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libstomp/configure.ac	2025-05-01 13:25:35.000000000 +0900
+++ src/libstomp/configure.ac	2025-05-01 13:25:37.211116400 +0900
@@ -22,7 +22,7 @@ PKG_CHECK_MODULES([APR],[apr-1])
 AC_CHECK_FUNCS([strstr])
 
 
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([libstomp.pc Makefile
 		src/Makefile
 		example/Makefile])
 AC_OUTPUT
--- origsrc/libstomp/example/Makefile.am	2025-05-01 13:25:35.000000000 +0900
+++ src/libstomp/example/Makefile.am	2025-05-01 13:25:37.211116400 +0900
@@ -1,4 +1,4 @@
 bin_PROGRAMS = stomptest
-stomptest_CFLAGS = -I ../src @APR_CFLAGS@
+stomptest_CFLAGS = -I$(top_srcdir)/src @APR_CFLAGS@
 stomptest_LDADD = ../src/libstomp.la
 stomptest_SOURCES = main.c
--- origsrc/libstomp/libstomp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libstomp/libstomp.pc.in	2025-05-01 13:25:37.211116400 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libstomp
+Description: APR based C STOMP clientslibrary
+Version: @VERSION@
+Libs: -L${libdir} -lstomp
+Cflags: -I${includedir}
+Requires: apr-1
--- origsrc/libstomp/src/Makefile.am	2025-05-01 13:25:35.000000000 +0900
+++ src/libstomp/src/Makefile.am	2025-05-01 13:25:37.211116400 +0900
@@ -2,4 +2,5 @@ lib_LTLIBRARIES = libstomp.la
 libstomp_la_CFLAGS = @APR_CFLAGS@
 libstomp_la_LIBADD = @APR_LIBS@
 libstomp_la_SOURCES = stomp.c 
+libstomp_la_LDFLAGS = -no-undefined
 include_HEADERS = stomp.h
