--- origsrc/u5e/CMakeLists.txt	2023-11-20 23:01:40.000000000 +0900
+++ src/u5e/CMakeLists.txt	2023-11-20 23:13:30.626267800 +0900
@@ -11,7 +11,14 @@ add_library(u5e
   )
 target_include_directories(u5e PUBLIC include/)
 target_compile_options(u5e PUBLIC -std=c++1y -Wall -Werror)
+if (DEFINED SOVERSION)
+  set_target_properties(u5e PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+find_package(pkgconfig)
+pkg_check_modules(GTEST gtest)
+if (NOT GTEST_FOUND)
 add_subdirectory(googletest/googletest)
+endif()
 enable_testing()
 
 include_directories(${u5e_SOURCE_DIR}/include ${gtest_SOURCE_DIR}/include)
@@ -19,3 +26,12 @@ include_directories(${u5e_SOURCE_DIR}/in
 add_subdirectory(t)
 add_subdirectory(examples)
 
+install(TARGETS u5e RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
+install(DIRECTORY include/u5e DESTINATION include)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/u5e.pc.in ${CMAKE_BINARY_DIR}/u5e.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/u5e.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/u5e/u5e.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/u5e/u5e.pc.in	2023-11-20 23:12:44.025225800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: u5e
+Description: Unicode Text C++ library
+Version: @VERSION@
+Libs: -L${libdir} -lu5e
+Cflags: -I${includedir}
