--- origsrc/libeasyfc-0.14.0/configure.ac	2018-02-16 12:18:42.000000000 +0900
+++ src/libeasyfc-0.14.0/configure.ac	2020-08-24 14:43:03.852332000 +0900
@@ -24,7 +24,6 @@ AC_PROG_CC
 
 GNOME_COMPILE_WARNINGS
 
-GNOME_COMMON_INIT
 GNOME_MAINTAINER_MODE_DEFINES
 
 CFLAGS="$CFLAGS $WARN_CFLAGS"
--- origsrc/libeasyfc-0.14.0/libeasyfc/Makefile.am	2016-02-10 15:03:48.000000000 +0900
+++ src/libeasyfc-0.14.0/libeasyfc/Makefile.am	2020-08-24 14:36:57.417509300 +0900
@@ -101,7 +101,7 @@ libeasyfc_la_SOURCES =				\
 libeasyfc_la_CFLAGS =				\
 	-DG_LOG_DOMAIN="\"EasyFC\""		\
 	$(NULL)
-libeasyfc_la_LDFLAGS =						\
+libeasyfc_la_LDFLAGS = -no-undefined				\
 	$(LDFLAGS)						\
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)	\
 	$(NULL)
--- origsrc/libeasyfc-0.14.0/libeasyfc-gobject/Makefile.am	2018-02-16 12:14:25.000000000 +0900
+++ src/libeasyfc-0.14.0/libeasyfc-gobject/Makefile.am	2020-08-24 14:36:57.427509900 +0900
@@ -111,7 +111,7 @@ libeasyfc_gobject_la_SOURCES =				\
 libeasyfc_gobject_la_CFLAGS =				\
 	-DG_LOG_DOMAIN="\"EasyFC-GObject\""		\
 	$(NULL)
-libeasyfc_gobject_la_LDFLAGS =						\
+libeasyfc_gobject_la_LDFLAGS = -no-undefined				\
 	$(LDFLAGS)							\
 	-version-info $(LT_G_CURRENT):$(LT_G_REVISION):$(LT_G_AGE)	\
 	$(NULL)
