--- origsrc/libeasyfc-0.13.1/libeasyfc/Makefile.am	2015-01-19 18:25:07.000000000 +0900
+++ src/libeasyfc-0.13.1/libeasyfc/Makefile.am	2017-01-02 12:57:52.595664800 +0900
@@ -98,7 +98,7 @@ libeasyfc_la_SOURCES =				\
 libeasyfc_la_CFLAGS =				\
 	-DG_LOG_DOMAIN="\"EasyFC\""		\
 	$(NULL)
-libeasyfc_la_LDFLAGS =						\
+libeasyfc_la_LDFLAGS = -no-undefined				\
 	$(LDFLAGS)						\
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)	\
 	$(NULL)
--- origsrc/libeasyfc-0.13.1/libeasyfc-gobject/Makefile.am	2016-02-05 15:41:32.000000000 +0900
+++ src/libeasyfc-0.13.1/libeasyfc-gobject/Makefile.am	2017-01-02 12:57:52.599155000 +0900
@@ -109,7 +109,7 @@ libeasyfc_gobject_la_SOURCES =				\
 libeasyfc_gobject_la_CFLAGS =				\
 	-DG_LOG_DOMAIN="\"EasyFC-GObject\""		\
 	$(NULL)
-libeasyfc_gobject_la_LDFLAGS =						\
+libeasyfc_gobject_la_LDFLAGS = -no-undefined				\
 	$(LDFLAGS)							\
 	-version-info $(LT_G_CURRENT):$(LT_G_REVISION):$(LT_G_AGE)	\
 	$(NULL)
