--- origsrc/libeasyfc-0.7/libeasyfc/Makefile.am	2012-02-17 18:00:01.000000000 +0900
+++ src/libeasyfc-0.7/libeasyfc/Makefile.am	2012-03-18 13:52:53.982788100 +0900
@@ -96,7 +96,7 @@ libeasyfc_la_SOURCES =				\
 libeasyfc_la_CFLAGS =				\
 	-DG_LOG_DOMAIN="\"EasyFC\""		\
 	$(NULL)
-libeasyfc_la_LDFLAGS =						\
+libeasyfc_la_LDFLAGS = -no-undefined				\
 	$(LDFLAGS)						\
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)	\
 	$(NULL)
--- origsrc/libeasyfc-0.7/libeasyfc-gobject/Makefile.am	2012-02-28 19:25:02.000000000 +0900
+++ src/libeasyfc-0.7/libeasyfc-gobject/Makefile.am	2012-03-18 13:52:53.982788100 +0900
@@ -104,7 +104,7 @@ libeasyfc_gobject_la_SOURCES =				\
 libeasyfc_gobject_la_CFLAGS =				\
 	-DG_LOG_DOMAIN="\"EasyFC-GObject\""		\
 	$(NULL)
-libeasyfc_gobject_la_LDFLAGS =						\
+libeasyfc_gobject_la_LDFLAGS = -no-undefined				\
 	$(LDFLAGS)							\
 	-version-info $(LT_G_CURRENT):$(LT_G_REVISION):$(LT_G_AGE)	\
 	$(NULL)
