--- origsrc/libeasyfc-0.14.1/libeasyfc/Makefile.am	2016-02-10 15:03:48.000000000 +0900
+++ src/libeasyfc-0.14.1/libeasyfc/Makefile.am	2023-12-20 17:10:22.417683800 +0900
@@ -102,6 +102,7 @@ libeasyfc_la_CFLAGS =				\
 	-DG_LOG_DOMAIN="\"EasyFC\""		\
 	$(NULL)
 libeasyfc_la_LDFLAGS =						\
+	-no-undefined						\
 	$(LDFLAGS)						\
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)	\
 	$(NULL)
--- origsrc/libeasyfc-0.14.1/libeasyfc-gobject/Makefile.am	2018-02-16 12:14:25.000000000 +0900
+++ src/libeasyfc-0.14.1/libeasyfc-gobject/Makefile.am	2023-12-20 17:10:44.845056000 +0900
@@ -112,6 +112,7 @@ libeasyfc_gobject_la_CFLAGS =				\
 	-DG_LOG_DOMAIN="\"EasyFC-GObject\""		\
 	$(NULL)
 libeasyfc_gobject_la_LDFLAGS =						\
+	-no-undefined							\
 	$(LDFLAGS)							\
 	-version-info $(LT_G_CURRENT):$(LT_G_REVISION):$(LT_G_AGE)	\
 	$(NULL)
