--- origsrc/libfwps-20191221/Makefile.am	2019-12-21 19:35:09.000000000 +0900
+++ src/libfwps-20191221/Makefile.am	2020-09-03 12:07:07.450163200 +0900
@@ -3,20 +3,10 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libuna \
-	libfdatetime \
-	libfguid \
-	libfole \
 	libfwps \
 	pyfwps \
 	pyfwps-python2 \
 	pyfwps-python3 \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libfwps-20191221/configure.ac	2019-12-21 19:35:09.000000000 +0900
+++ src/libfwps-20191221/configure.ac	2020-09-03 11:55:08.230106000 +0900
@@ -158,15 +158,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfole/Makefile])
 AC_CONFIG_FILES([libfwps/Makefile])
 AC_CONFIG_FILES([pyfwps/Makefile])
 AC_CONFIG_FILES([pyfwps-python2/Makefile])
--- origsrc/libfwps-20191221/libfwps/Makefile.am	2019-12-21 19:35:09.000000000 +0900
+++ src/libfwps-20191221/libfwps/Makefile.am	2020-09-03 11:55:30.535665900 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libfwps-20191221/pyfwps/Makefile.am	2019-12-21 19:35:09.000000000 +0900
+++ src/libfwps-20191221/pyfwps/Makefile.am	2020-09-03 11:56:48.289787300 +0900
@@ -1,5 +1,6 @@
 if HAVE_PYTHON
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -34,7 +35,7 @@ pyfwps_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pyfwps_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyfwps_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyfwps_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
--- origsrc/libfwps-20191221/pyfwps-python2/Makefile.am	2019-12-21 19:35:09.000000000 +0900
+++ src/libfwps-20191221/pyfwps-python2/Makefile.am	2020-09-03 11:57:15.236824400 +0900
@@ -6,6 +6,7 @@ pyexecdir = $(pyexecdir2)
 
 if HAVE_PYTHON2
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -44,7 +45,7 @@ pyfwps_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pyfwps_la_CPPFLAGS = $(PYTHON2_CPPFLAGS)
-pyfwps_la_LDFLAGS  = -module -avoid-version $(PYTHON2_LDFLAGS)
+pyfwps_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON2_LDFLAGS)
 
 $(BUILT_SOURCES):
 	/bin/cp -f $(top_srcdir)/pyfwps/$@ $@
--- origsrc/libfwps-20191221/pyfwps-python3/Makefile.am	2019-12-21 19:35:09.000000000 +0900
+++ src/libfwps-20191221/pyfwps-python3/Makefile.am	2020-09-03 11:57:37.075669200 +0900
@@ -6,6 +6,7 @@ pyexecdir = $(pyexecdir3)
 
 if HAVE_PYTHON3
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -44,7 +45,7 @@ pyfwps_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pyfwps_la_CPPFLAGS = $(PYTHON3_CPPFLAGS)
-pyfwps_la_LDFLAGS  = -module -avoid-version $(PYTHON3_LDFLAGS)
+pyfwps_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON3_LDFLAGS)
 
 $(BUILT_SOURCES):
 	/bin/cp -f $(top_srcdir)/pyfwps/$@ $@
--- origsrc/libfwps-20191221/tests/Makefile.am	2019-12-21 19:35:09.000000000 +0900
+++ src/libfwps-20191221/tests/Makefile.am	2020-09-03 11:57:56.680219400 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
