--- origsrc/ttyvideo-0.0.1/handle.c	2017-05-30 03:14:58.000000000 +0900
+++ src/ttyvideo-0.0.1/handle.c	2025-05-20 22:07:36.259442400 +0900
@@ -21,6 +21,8 @@ static char** helpMessages = NULL;
 static char* defaultArg = NULL;
 static char* defaultArgHelp = NULL;
 
+int terminate;
+
 #define MAX_ARG_SIZE 200
 
 char* setDefaultArgument(char* helpText) {
--- origsrc/ttyvideo-0.0.1/handle.h	2017-05-30 03:14:58.000000000 +0900
+++ src/ttyvideo-0.0.1/handle.h	2025-05-20 22:08:42.349947800 +0900
@@ -5,8 +5,6 @@
 extern "C" {
 #endif
 
-char* filename;
-
 char* setDefaultArgument(char* helpText);
 
 char* addArgument(char* helpText, int numArguments, char* firstCall, char* secondCall);
@@ -16,7 +14,7 @@ void printUsage();
 int error(char* message);
 int handle(int argc, char** argv);
 
-int terminate;
+extern int terminate;
 
 #define TAKES_NO_ARGUMENTS 0
 #define TAKES_ONE_ARGUMENT 1
