--- origsrc/ign-cmake-ignition-cmake2_2.5.0/CMakeLists.txt	2020-09-09 08:18:08.000000000 +0900
+++ src/ign-cmake-ignition-cmake2_2.5.0/CMakeLists.txt	2020-09-25 11:01:08.536554300 +0900
@@ -53,7 +53,11 @@ install(DIRECTORY codecheck/
 set(ign_config_input  "${CMAKE_CURRENT_SOURCE_DIR}/config/ignition-cmake-config.cmake.in")
 set(ign_config_output "${PROJECT_NAME_LOWER}-config.cmake")
 set(ign_version_output "${PROJECT_NAME_LOWER}-config-version.cmake")
-set(ign_config_install_dir "${CMAKE_INSTALL_DATAROOTDIR}/cmake/${PROJECT_NAME_LOWER}")
+if(CYGWIN)
+  set(ign_config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME_LOWER}")
+else()
+  set(ign_config_install_dir "${CMAKE_INSTALL_DATAROOTDIR}/cmake/${PROJECT_NAME_LOWER}")
+endif()
 set(ign_pkgconfig_input "${CMAKE_CURRENT_SOURCE_DIR}/config/ignition-cmake.pc.in")
 set(ign_pkgconfig_output "${CMAKE_BINARY_DIR}/ignition-cmake${PROJECT_VERSION_MAJOR}.pc")
 set(ign_utilities_target ${PROJECT_EXPORT_NAME}-utilities)
