--- origsrc/gz-cmake-gz-cmake4_4.0.0/CMakeLists.txt	2024-09-26 07:52:35.000000000 +0900
+++ src/gz-cmake-gz-cmake4_4.0.0/CMakeLists.txt	2024-10-04 07:26:03.635897600 +0900
@@ -60,7 +60,7 @@ install(DIRECTORY tools/
 set(gz_config_input  "${CMAKE_CURRENT_SOURCE_DIR}/config/gz-cmake-config.cmake.in")
 set(gz_config_output "${PROJECT_NAME_LOWER}-config.cmake")
 set(gz_version_output "${PROJECT_NAME_LOWER}-config-version.cmake")
-set(gz_config_install_dir "${CMAKE_INSTALL_DATAROOTDIR}/cmake/${PROJECT_NAME_LOWER}")
+set(gz_config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME_LOWER}")
 set(gz_pkgconfig_input "${CMAKE_CURRENT_SOURCE_DIR}/config/gz-cmake.pc.in")
 set(gz_pkgconfig_output "${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc")
 set(gz_pkgconfig_install_dir "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
--- origsrc/gz-cmake-gz-cmake4_4.0.0/config/gz-cmake.pc.in	2024-09-26 07:52:35.000000000 +0900
+++ src/gz-cmake-gz-cmake4_4.0.0/config/gz-cmake.pc.in	2024-10-04 07:14:38.981320700 +0900
@@ -1,4 +1,4 @@
-prefix=${pcfiledir}/@GZ_PC_CONFIG_RELATIVE_PATH_TO_PREFIX@
+prefix=@CMAKE_INSTALL_PREFIX@
 includedir=@GZ_PC_INCLUDEDIR@
 
 Name: Gazebo @GZ_DESIGNATION@
