--- origsrc/libpfxtree-0.4.0/CMakeLists.txt	1970-01-01 09:00:00.000000000 +0900
+++ src/libpfxtree-0.4.0/CMakeLists.txt	2025-08-12 06:14:16.659130300 +0900
@@ -0,0 +1,23 @@
+cmake_minimum_required(VERSION 3.28)
+project(libpfxtree
+  VERSION 0.4.0
+  DESCRIPTION "Prefix tree (trie) library"
+  HOMEPAGE_URL "https://github.com/delwink/libpfxtree")
+
+include(GNUInstallDirs)
+include(CTest)
+
+add_library(pfxtree SHARED "pfxtree.c")
+set_target_properties(pfxtree PROPERTIES SOVERSION ${PROJECT_VERSION_MAJOR})
+
+install(TARGETS pfxtree
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+add_executable(pfxtree-test "pfxtree-test.c")
+target_link_libraries(pfxtree-test pfxtree)
+add_test(NAME pfxtree-test COMMAND pfxtree-test)
+
+install(FILES pfxtree.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+
+install(FILES pfxtree.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
--- origsrc/libpfxtree-0.4.0/pfxtree.pc	2020-06-02 06:26:01.000000000 +0900
+++ src/libpfxtree-0.4.0/pfxtree.pc	2025-08-12 06:11:12.190455200 +0900
@@ -3,7 +3,7 @@ exec_prefix = ${prefix}
 libdir = ${exec_prefix}/lib
 includedir = ${prefix}/include
 
-Version: 0.3.0
+Version: 0.4.0
 Cflags = -I${includedir}
 Description: Delwink prefix tree library
 Name: pfxtree
