--- origsrc/minilzo-2.10/CMakeLists.txt	1970-01-01 09:00:00.000000000 +0900
+++ src/minilzo-2.10/CMakeLists.txt	2025-08-15 07:13:12.868423200 +0900
@@ -0,0 +1,28 @@
+cmake_minimum_required(VERSION 3.28)
+project(minilzo
+  VERSION 2.10
+  DESCRIPTION "Lightweight subset of the LZO library"
+  HOMEPAGE_URL "https://www.oberhumer.com/opensource/lzo/#minilzo")
+
+include(GNUInstallDirs)
+include(CTest)
+
+add_library(minilzo SHARED minilzo.c)
+set_target_properties(minilzo PROPERTIES SOVERSION ${PROJECT_VERSION_MAJOR})
+
+install(TARGETS minilzo
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+add_executable(testmini testmini.c)
+target_link_libraries(testmini minilzo)
+add_test(NAME testmini COMMAND testmini)
+
+install(FILES lzoconf.h lzodefs.h minilzo.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/minilzo)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/minilzo.pc.in ${CMAKE_BINARY_DIR}/minilzo.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/minilzo.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/minilzo-2.10/minilzo.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/minilzo-2.10/minilzo.pc.in	2025-08-15 07:13:19.103304300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: minilzo
+Description: Lightweight subset of the LZO library
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lminilzo
+Cflags: -I${includedir}
