--- origsrc/vsqlite++-0.3.12/Makefile.am	2013-01-31 16:05:37.000000000 +0900
+++ src/vsqlite++-0.3.12/Makefile.am	2013-10-16 12:10:32.057683900 +0900
@@ -1,12 +1,14 @@
-LIBVSQLITE_VERSION=`cat VERSION`
+LIBVSQLITE_VERSION=$(VERSION)
 AUTOMAKE_OPTIONS = subdir-objects foreign
 ACLOCAL_AMFLAGS = ${ACLOCAL_FLAGS} -I m4
 AM_CXXFLAGS = -W -Wall -pedantic -O3 -W -std=c++03
 
 lib_LTLIBRARIES = libvsqlitepp.la
 
-libvsqlitepp_la_CXXFLAGS = -I include $(AM_CXXFLAGS)
-libvsqlitepp_la_LDFLAGS = -lsqlite3 -Wl,--as-needed -Wl,-soname -Wl,libvsqlitepp.so.3 -version-info 3:0:0
+libvsqlitepp_la_CPPFLAGS = -I$(top_srcdir)/include
+libvsqlitepp_la_CXXFLAGS = $(SQLITE3_CFLAGS) $(AM_CXXFLAGS)
+libvsqlitepp_la_LIBADD = $(SQLITE3_LIBS)
+libvsqlitepp_la_LDFLAGS = -no-undefined -version-info 3:0:0
 libvsqlitepp_la_SOURCES = \
 	src/sqlite/command.cpp \
 	src/sqlite/connection.cpp \
@@ -18,7 +20,8 @@ libvsqlitepp_la_SOURCES = \
 	$(NULL)
 
 noinst_PROGRAMS=vsqlitepp_example
-vsqlitepp_example_CXXFLAGS=-I include $(AM_CXXFLAGS)
+vsqlitepp_example_CPPFLAGS = -I$(top_srcdir)/include
+vsqlitepp_example_CXXFLAGS= $(AM_CXXFLAGS)
 vsqlitepp_example_LDADD = libvsqlitepp.la
 vsqlitepp_example_SOURCES = \
 	examples/sqlite_wrapper.cpp \
--- origsrc/vsqlite++-0.3.12/configure.ac	2013-01-31 17:01:31.000000000 +0900
+++ src/vsqlite++-0.3.12/configure.ac	2013-10-16 12:06:12.206121400 +0900
@@ -4,6 +4,7 @@ AM_INIT_AUTOMAKE
 AC_CONFIG_MACRO_DIR([m4])
 AC_PROG_CXX
 LT_INIT
+PKG_CHECK_MODULES([SQLITE3], [sqlite3])
 
 dnl -------------------------------------------
 dnl test "$prefix" = "NONE" && prefix=/usr/local
--- origsrc/vsqlite++-0.3.12/include/Makefile.am	2013-01-31 15:33:22.000000000 +0900
+++ src/vsqlite++-0.3.12/include/Makefile.am	2013-10-16 12:13:12.604558900 +0900
@@ -1,4 +1,4 @@
-nobase_include_HEADERS = \
+nobase_pkginclude_HEADERS = \
     sqlite/command.hpp \
     sqlite/database_exception.hpp \
     sqlite/query.hpp \
