diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/eblob-0.23.1/bindings/cpp/CMakeLists.txt src/eblob-0.23.1/bindings/cpp/CMakeLists.txt
--- origsrc/eblob-0.23.1/bindings/cpp/CMakeLists.txt	2015-08-04 22:21:45.000000000 +0900
+++ src/eblob-0.23.1/bindings/cpp/CMakeLists.txt	2015-12-28 16:22:30.692676000 +0900
@@ -11,6 +11,7 @@ set_target_properties(eblob_cpp PROPERTI
 target_link_libraries(eblob_cpp ${EBLOB_CPP_LIBRARIES} eblob)
 
 install(TARGETS eblob_cpp
+    RUNTIME DESTINATION bin
     LIBRARY DESTINATION lib${LIB_SUFFIX}
     ARCHIVE DESTINATION lib${LIB_SUFFIX}
     BUNDLE DESTINATION library
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/eblob-0.23.1/bindings/python/CMakeLists.txt src/eblob-0.23.1/bindings/python/CMakeLists.txt
--- origsrc/eblob-0.23.1/bindings/python/CMakeLists.txt	2015-08-04 22:21:45.000000000 +0900
+++ src/eblob-0.23.1/bindings/python/CMakeLists.txt	2015-12-28 16:22:51.129861200 +0900
@@ -2,6 +2,7 @@ add_library(eblob_python SHARED eblob_py
 target_link_libraries(eblob_python eblob eblob_cpp ${EBLOB_PYTHON_LIBRARIES})
 
 install(TARGETS eblob_python
+    RUNTIME DESTINATION bin
     LIBRARY DESTINATION lib${LIB_SUFFIX}
     ARCHIVE DESTINATION lib${LIB_SUFFIX}
     BUNDLE DESTINATION library
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/eblob-0.23.1/library/CMakeLists.txt src/eblob-0.23.1/library/CMakeLists.txt
--- origsrc/eblob-0.23.1/library/CMakeLists.txt	2015-08-04 22:21:45.000000000 +0900
+++ src/eblob-0.23.1/library/CMakeLists.txt	2015-12-28 16:25:01.229518000 +0900
@@ -26,9 +26,10 @@ set_target_properties(eblob PROPERTIES
     LINKER_LANGUAGE CXX
     )
 
-target_link_libraries(eblob ${EBLOB_LIBRARIES} ${Boost_SYSTEM_LIBRARIES} handystats rt)
+target_link_libraries(eblob ${EBLOB_LIBRARIES} ${Boost_SYSTEM_LIBRARIES} handystats rt ssp)
 
 install(TARGETS eblob
+    RUNTIME DESTINATION bin
     LIBRARY DESTINATION lib${LIB_SUFFIX}
     ARCHIVE DESTINATION lib${LIB_SUFFIX}
     BUNDLE DESTINATION library
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/eblob-0.23.1/library/mobjects.c src/eblob-0.23.1/library/mobjects.c
--- origsrc/eblob-0.23.1/library/mobjects.c	2015-08-04 22:21:45.000000000 +0900
+++ src/eblob-0.23.1/library/mobjects.c	2015-12-28 16:24:17.731588600 +0900
@@ -49,6 +49,13 @@
 
 #if !defined(_D_EXACT_NAMLEN) && (defined(__FreeBSD__) || defined(__APPLE__))
 #define _D_EXACT_NAMLEN(d) ((d)->d_namlen)
+#elif defined(__CYGWIN__)
+#define _D_EXACT_NAMLEN(d) strlen((d)->d_name)
+#endif
+
+#if defined(__CYGWIN__)
+#define dirent64 dirent
+#define readdir64(d) readdir(d)
 #endif
 
 static const char *eblob_get_base(const char *blob_base)
