--- origsrc/eblob-0.21.22/bindings/cpp/CMakeLists.txt	2013-10-22 04:25:08.000000000 +0900
+++ src/eblob-0.21.22/bindings/cpp/CMakeLists.txt	2013-11-02 18:22:46.823507200 +0900
@@ -12,6 +12,7 @@ set_target_properties(eblob_cpp PROPERTI
 target_link_libraries(eblob_cpp ${EBLOB_CPP_LIBRARIES} eblob)
 
 install(TARGETS eblob_cpp
+    RUNTIME DESTINATION bin
     LIBRARY DESTINATION lib${LIB_SUFFIX}
     ARCHIVE DESTINATION lib${LIB_SUFFIX}
     BUNDLE DESTINATION library
--- origsrc/eblob-0.21.22/library/CMakeLists.txt	2013-10-22 04:25:08.000000000 +0900
+++ src/eblob-0.21.22/library/CMakeLists.txt	2013-11-02 17:05:34.637179100 +0900
@@ -19,9 +19,10 @@ set_target_properties(eblob PROPERTIES
     SOVERSION ${EBLOB_VERSION_ABI}
     )
 
-target_link_libraries(eblob ${EBLOB_LIBRARIES} ${Boost_SYSTEM_LIBRARIES})
+target_link_libraries(eblob ${EBLOB_LIBRARIES} ${Boost_SYSTEM_LIBRARIES} ssp)
 
 install(TARGETS eblob
+    RUNTIME DESTINATION bin
     LIBRARY DESTINATION lib${LIB_SUFFIX}
     ARCHIVE DESTINATION lib${LIB_SUFFIX}
     BUNDLE DESTINATION library
--- origsrc/eblob-0.21.22/library/mobjects.c	2013-10-22 04:25:08.000000000 +0900
+++ src/eblob-0.21.22/library/mobjects.c	2013-11-02 17:01:47.128390100 +0900
@@ -40,8 +40,17 @@
 #include <string.h>
 #include <unistd.h>
 
-#if !defined(_D_EXACT_NAMLEN) && (defined(__FreeBSD__) || defined(__APPLE__))
+#if !defined(_D_EXACT_NAMLEN)
+#if defined(__FreeBSD__) || defined(__APPLE__)
 #define _D_EXACT_NAMLEN(d) ((d)->d_namlen)
+#elif defined(__CYGWIN__)
+#define _D_EXACT_NAMLEN(d) strlen((d)->d_name)
+#endif
+#endif
+
+#if defined(__CYGWIN__)
+#define dirent64 dirent
+#define readdir64(d) readdir(d)
 #endif
 
 static const char *eblob_get_base(const char *blob_base)
