--- origsrc/libdoge/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libdoge/Makefile.am	2024-11-29 19:25:52.875070900 +0900
@@ -0,0 +1,15 @@
+
+bin_PROGRAMS = doge
+doge_SOURCES = doge.c
+doge_LDADD = libdoge.la
+
+include_HEADERS = libdoge.h
+
+lib_LTLIBRARIES = libdoge.la
+libdoge_la_SOURCES = libdoge.c
+libdoge_la_LDFLAGS = -no-undefined -version-info 1
+
+TESTS = doge
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libdoge/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/libdoge/configure.ac	2024-11-29 19:23:23.207550700 +0900
@@ -0,0 +1,26 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([libdoge], [0])
+AC_CONFIG_SRCDIR([doge.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_LN_S
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+
+# Checks for library functions.
+
+AC_CONFIG_FILES([libdoge.pc Makefile])
+AC_OUTPUT
--- origsrc/libdoge/libdoge.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libdoge/libdoge.pc.in	2024-11-29 19:23:23.207550700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libdoge
+Description: Library for integrating doge into C/C++ applications
+Version: @VERSION@
+Libs: -L${libdir} -ldoge
+Cflags: -I${includedir}
