--- origsrc/exceptions4c/Makefile.am	2018-08-22 23:44:31.000000000 +0900
+++ src/exceptions4c/Makefile.am	2018-08-23 11:12:52.094253500 +0900
@@ -7,21 +7,21 @@
 
 AUTOMAKE_OPTIONS = foreign subdir-objects
 
-AM_CFLAGS = -Wall -Werror --pedantic -Wno-missing-braces -coverage -O0 -I$(E4C_PATH) -I$(E4C_LITE_PATH)
-
+AM_CFLAGS = -Wall --pedantic -Wno-missing-braces -coverage -O0 -I$(E4C_PATH) -I$(E4C_LITE_PATH)
+AM_CPPFLAGS = -I$(top_srcdir)/src -I$(top_srcdir)/lite/src
 
 # Library
 
 E4C_PATH            = src
-E4C_LIBRARY         = lib/libe4c.a
+E4C_LIBRARY         = lib/libe4c.la
 
 E4C_LITE_PATH       = lite/src
-E4C_LITE_LIBRARY    = lib/libe4c_lite.a
+E4C_LITE_LIBRARY    = lib/libe4c_lite.la
 
 
 # Install
 
-lib_LIBRARIES =                         \
+lib_LTLIBRARIES =                         \
     $(E4C_LIBRARY)                      \
     $(E4C_LITE_LIBRARY)
 
@@ -33,7 +33,7 @@ include_HEADERS =
 # Documentation
 
 docdir = $(datadir)/doc/exceptions4c
-doc_DATA = doc/*
+doc_DATA = $(top_srcdir)/doc/*
 
 
 # Cleanup
@@ -343,14 +343,16 @@ tests: check
 
 # exceptions4c library
 
-lib_libe4c_a_CFLAGS                     = -Wall -Werror --pedantic -Wno-missing-braces
-lib_libe4c_a_SOURCES                    = src/e4c.c
+lib_libe4c_la_CFLAGS                     = -Wall -Werror --pedantic -Wno-missing-braces
+lib_libe4c_la_SOURCES                    = src/e4c.c
+lib_libe4c_la_LDFLAGS                    = -no-undefined
 
 
 # exceptions4c library (lightweight version)
 
-lib_libe4c_lite_a_CFLAGS                = -Wall -Werror --pedantic -Wno-missing-braces
-lib_libe4c_lite_a_SOURCES               = lite/src/e4c_lite.c
+lib_libe4c_lite_la_CFLAGS                = -Wall -Werror --pedantic -Wno-missing-braces
+lib_libe4c_lite_la_SOURCES               = lite/src/e4c_lite.c
+lib_libe4c_lite_la_LDFLAGS               = -no-undefined
 
 
 # exceptions4c tests
--- origsrc/exceptions4c/configure.ac	2018-08-22 23:44:31.000000000 +0900
+++ src/exceptions4c/configure.ac	2018-08-22 23:48:32.562514300 +0900
@@ -30,8 +30,7 @@ AC_CONFIG_SRCDIR([src/e4c.c])
 # Checks for programs.
 AC_PROG_CC
 AC_PROG_CPP
-AC_PROG_RANLIB
-AM_PROG_AR
+LT_INIT
 
 
 # Checks for header files.
