--- origsrc/exceptions4c-4.0.0/Makefile.am	2025-02-22 23:44:46.000000000 +0900
+++ src/exceptions4c-4.0.0/Makefile.am	2025-08-08 18:22:47.375387700 +0900
@@ -7,26 +7,28 @@
 
 AUTOMAKE_OPTIONS = foreign subdir-objects
 
-AM_CFLAGS = -Wall -Werror --pedantic -Wno-missing-braces -Wno-dangling-else -coverage -O0 -I$(EXCEPTIONS4C_PATH)
+AM_CFLAGS = -Wall -Werror --pedantic -Wno-missing-braces -Wno-dangling-else -I$(EXCEPTIONS4C_PATH)
 
 
 # Library
 
-EXCEPTIONS4C_PATH = src
-EXCEPTIONS4C_LIBRARY = lib/libexceptions4c.a
+EXCEPTIONS4C_PATH = $(srcdir)/src
+EXCEPTIONS4C_LIBRARY = lib/libexceptions4c.la
 
 
 # Install
 
-lib_LIBRARIES = $(EXCEPTIONS4C_LIBRARY)
+lib_LTLIBRARIES = $(EXCEPTIONS4C_LIBRARY)
 
 include_HEADERS = src/exceptions4c.h
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = exceptions4c.pc
 
 # Documentation
 
-docsdir = $(datadir)/docs/exceptions4c
-docs_DATA = docs/*
+# docsdir = $(datadir)/docs/exceptions4c
+# docs_DATA = docs/*
 
 
 # Cleanup
@@ -141,9 +143,9 @@ tests: check
 
 # Library
 
-lib_libexceptions4c_a_CFLAGS                = -Wall -Werror --pedantic -Wno-missing-braces -I$(EXCEPTIONS4C_PATH)
-lib_libexceptions4c_a_SOURCES               = src/exceptions4c.c
-
+lib_libexceptions4c_la_CFLAGS                = -Wall -Werror --pedantic -Wno-missing-braces -I$(EXCEPTIONS4C_PATH)
+lib_libexceptions4c_la_SOURCES               = src/exceptions4c.c
+lib_libexceptions4c_la_LDFLAGS               = -no-undefined
 
 # Tests
 
--- origsrc/exceptions4c-4.0.0/configure.ac	2025-02-22 23:44:46.000000000 +0900
+++ src/exceptions4c-4.0.0/configure.ac	2025-08-08 17:52:42.285006300 +0900
@@ -30,8 +30,7 @@ AC_CONFIG_SRCDIR([src/exceptions4c.c])
 # Checks for programs.
 AC_PROG_CC
 AC_PROG_CPP
-AC_PROG_RANLIB
-AM_PROG_AR
+LT_INIT
 
 
 # Checks for header files.
--- origsrc/exceptions4c-4.0.0/exceptions4c.pc.in	2025-02-22 23:44:46.000000000 +0900
+++ src/exceptions4c-4.0.0/exceptions4c.pc.in	2025-08-08 19:54:44.364542400 +0900
@@ -6,5 +6,5 @@ includedir=${prefix}/include
 Name: exceptions4c
 Description: Exceptions for C
 Version: @VERSION@
-Libs: -L${libdir} -llibexceptions4c
+Libs: -L${libdir} -lexceptions4c
 Cflags: -I${includedir}
