--- origsrc/spatialite-tools-4.3.0/Makefile.am	2015-07-01 16:30:35.000000000 +0900
+++ src/spatialite-tools-4.3.0/Makefile.am	2018-03-15 23:08:47.356000000 +0900
@@ -70,7 +70,7 @@ spatialite_gml_LDADD = @LIBSPATIALITE_LI
 	-lexpat 
 spatialite_LDADD = @LIBSPATIALITE_LIBS@ \
 	@LIBFREEXL_LIBS@ \
-	@READLINE_LIBS@
+	@READLINE_LIBS@ $(LTLIBICONV)
 spatialite_xml_validator_LDADD = @LIBXML2_LIBS@
 spatialite_xml_load_LDADD = @LIBSPATIALITE_LIBS@ \
 	-lexpat
--- origsrc/spatialite-tools-4.3.0/configure.ac	2015-07-01 16:30:35.000000000 +0900
+++ src/spatialite-tools-4.3.0/configure.ac	2018-03-15 21:11:34.702000000 +0900
@@ -89,6 +89,7 @@ AC_STRUCT_TM
 AC_C_VOLATILE
 
 # Checks for library functions.
+AM_ICONV
 AC_FUNC_LSTAT
 AC_FUNC_LSTAT_FOLLOWS_SLASHED_SYMLINK
 AC_FUNC_MEMCMP
