--- origsrc/libcvautomation-Release_2.0/configure.ac	2012-10-12 06:13:57.000000000 +0900
+++ src/libcvautomation-Release_2.0/configure.ac	2017-07-15 19:09:51.550000000 +0900
@@ -36,7 +36,7 @@ if test "$use_opencv2" == "true"; then
 	AC_SUBST(CV_VERSION, "opencv2")
 	cv_headers=$srcdir/include/opencv2_includes
 	AC_SUBST(CV_CFLAGS, [`pkg-config --cflags opencv`])
-	AC_SUBST(CV_LIBS, ["-lopencv_core -lopencv_imgproc -lopencv_highgui"])
+	AC_SUBST(CV_LIBS, ["-lopencv_core -lopencv_imgproc -lopencv_highgui -lopencv_imgcodecs"])
 else
 	echo "could not find OpenCV version 2.0.0 or higher, checking for 1.0.0 or higher..."
 	PKG_CHECK_MODULES([OpenCV], [opencv >= 1.0.0])
--- origsrc/libcvautomation-Release_2.0/examples/cva-match.c	2012-10-12 06:13:57.000000000 +0900
+++ src/libcvautomation-Release_2.0/examples/cva-match.c	2017-07-15 19:09:09.938999800 +0900
@@ -20,6 +20,7 @@
 #include <unistd.h>
 #include <getopt.h>
 #include <limits.h>
+#include <math.h>
 
 #include <libcvautomation/libcvautomation.h>
 
--- origsrc/libcvautomation-Release_2.0/include/opencv2_includes	2012-10-12 06:13:57.000000000 +0900
+++ src/libcvautomation-Release_2.0/include/opencv2_includes	2017-07-15 19:03:39.219000200 +0900
@@ -2,4 +2,5 @@
  * from the OpenCV1 counterparts */
 #include <opencv2/core/core_c.h>
 #include <opencv2/highgui/highgui_c.h>
+#include <opencv2/imgcodecs/imgcodecs_c.h>
 #include <opencv2/imgproc/imgproc_c.h>
--- origsrc/libcvautomation-Release_2.0/libcvautomation/Makefile.am	2012-10-12 06:13:57.000000000 +0900
+++ src/libcvautomation-Release_2.0/libcvautomation/Makefile.am	2017-07-15 18:55:13.622173800 +0900
@@ -5,7 +5,7 @@ AM_LIBS = -shared
 #Lists the libraries to build
 lib_LTLIBRARIES = libcvautomation.la
 libcvautomation_la_SOURCES = libcvautomation-opencv.c libcvautomation-xlib.c libcvautomation-xtest.c
-libcvautomation_la_LDFLAGS = -version-info @LIBTOOL_VERSION@
+libcvautomation_la_LDFLAGS = -no-undefined -version-info @LIBTOOL_VERSION@
 libcvautomation_la_LIBADD = @X11_LIBS@ @XTEST_LIBS@ @CV_LIBS@
 libcvautomation_la_CFLAGS = -Wall -Werror -O2 -I$(top_srcdir)/include @X11_CFLAGS@ @XTEST_CFLAGS@ @CV_CFLAGS@
 
--- origsrc/libcvautomation-Release_2.0/python/Makefile.am	2012-10-12 06:13:57.000000000 +0900
+++ src/libcvautomation-Release_2.0/python/Makefile.am	2017-07-15 18:55:13.622173800 +0900
@@ -1,7 +1,7 @@
 pyexec_LTLIBRARIES = _libcvautomation.la
 _libcvautomation_la_SOURCES = libcvautomation-swig.c
-_libcvautomation_la_LDFLAGS = -avoid-version -module $(PYTHON_LDFLAGS) -L$(top_srcdir)/libcvautomation/.libs
-_libcvautomation_la_LIBADD = -lcvautomation
+_libcvautomation_la_LDFLAGS = -no-undefined -avoid-version -module $(PYTHON_LDFLAGS)
+_libcvautomation_la_LIBADD = $(top_builddir)/libcvautomation/libcvautomation.la
 _libcvautomation_la_CFLAGS = $(PYTHON_CPPFLAGS) -I$(top_srcdir)/include
 
 python_PYTHON = libcvautomation.py libcvautomation_funcs.py
