--- origsrc/libmobi-0.12/configure.ac	2024-06-17 15:10:16.000000000 +0900
+++ src/libmobi-0.12/configure.ac	2025-06-25 02:59:20.470385700 +0900
@@ -113,6 +113,12 @@ case "$host" in
         ISO99_SOURCE="-D_ISOC99_SOURCE=1"
         WIN32=yes
         ;;
+    *-*-cygwin*)
+        NO_UNDEFINED="-no-undefined"
+        AVOID_VERSION=
+        ISO99_SOURCE=
+        WIN32=no
+        ;;
     *)
         NO_UNDEFINED=
         AVOID_VERSION=
--- origsrc/libmobi-0.12/tests/test.sh.in	2024-06-17 15:10:16.000000000 +0900
+++ src/libmobi-0.12/tests/test.sh.in	2025-06-25 03:00:05.767232600 +0900
@@ -94,7 +94,7 @@ if [[ "${do_md5}" -eq "1" ]]; then
             cd "${tmp_dir}${separator}${markup_dir}" || die "Could not change directory to ${tmp_dir}${separator}${markup_dir}" $?
             rm -f "..${separator}${testfile_md5}.md5"
             ${md5prog} * > "..${separator}${testfile_md5}.md5"
-            diff -w "..${separator}${testfile_md5}.md5" "..${separator}..${separator}${md5file_markup}" || die "Wrong md5 checksum in ${markup_dir}" $?
+            diff -w "..${separator}${testfile_md5}.md5" "${md5file_markup}" || die "Wrong md5 checksum in ${markup_dir}" $?
             rm -f "..${separator}${testfile_md5}.md5"
         ) || exit $?
         log "Checksum correct"
--- origsrc/libmobi-0.12/tools/Makefile.am	2024-06-17 15:10:16.000000000 +0900
+++ src/libmobi-0.12/tools/Makefile.am	2025-06-25 03:04:28.062409800 +0900
@@ -7,26 +7,26 @@ AM_CPPFLAGS = -I$(top_srcdir)/src -I$(to
 bin_PROGRAMS = mobitool mobimeta
 man_MANS = mobitool.1 mobimeta.1
 
-noinst_LIBRARIES = libcommon.a
-libcommon_a_SOURCES = common.c common.h
+noinst_LTLIBRARIES = libcommon.la
+libcommon_la_SOURCES = common.c common.h
 
 if USE_INTERNAL_GETOPT
-libcommon_a_SOURCES += win32/getopt.c win32/getopt.h
+libcommon_la_SOURCES += win32/getopt.c win32/getopt.h
 endif
 
 mobitool_SOURCES = mobitool.c
-mobitool_DEPENDENCIES = $(top_builddir)/src/libmobi.la libcommon.a
-mobitool_LDADD = libcommon.a $(top_builddir)/src/libmobi.la
+mobitool_DEPENDENCIES = $(top_builddir)/src/libmobi.la libcommon.la
+mobitool_LDADD = libcommon.la $(top_builddir)/src/libmobi.la
 mobitool_CFLAGS = $(ISO99_SOURCE) $(DEBUG_CFLAGS) -D_POSIX_C_SOURCE=200112L
 mobitool_LDFLAGS = $(TOOLS_STATIC)
 
 if USE_XMLWRITER
 # miniz.c zip functions are needed for epub creation
-noinst_LIBRARIES += libminiz.a
-libminiz_a_SOURCES = ../src/miniz.c ../src/miniz.h
-libminiz_a_CFLAGS = $(MINIZ_CFLAGS) -DMINIZ_NO_ZLIB_COMPATIBLE_NAMES
-mobitool_DEPENDENCIES += libminiz.a
-mobitool_LDADD += libminiz.a
+noinst_LTLIBRARIES += libminiz.la
+libminiz_la_SOURCES = ../src/miniz.c ../src/miniz.h
+libminiz_la_CFLAGS = $(MINIZ_CFLAGS) -DMINIZ_NO_ZLIB_COMPATIBLE_NAMES
+mobitool_DEPENDENCIES += libminiz.la
+mobitool_LDADD += libminiz.la
 if USE_STATIC
 if USE_MINIZ
 mobitool_LDFLAGS += $(MOBI_ALLOW_MULTIPLE)
@@ -35,8 +35,8 @@ endif
 endif
 
 mobimeta_SOURCES = mobimeta.c
-mobimeta_DEPENDENCIES = $(top_builddir)/src/libmobi.la libcommon.a
-mobimeta_LDADD = libcommon.a $(top_builddir)/src/libmobi.la
+mobimeta_DEPENDENCIES = $(top_builddir)/src/libmobi.la libcommon.la
+mobimeta_LDADD = libcommon.la $(top_builddir)/src/libmobi.la
 mobimeta_CFLAGS = $(ISO99_SOURCE) $(DEBUG_CFLAGS) -D_POSIX_C_SOURCE=200112L
 mobimeta_LDFLAGS = $(TOOLS_STATIC)
 
@@ -44,8 +44,8 @@ if USE_ENCRYPTION
 bin_PROGRAMS += mobidrm
 man_MANS += mobidrm.1
 mobidrm_SOURCES = mobidrm.c
-mobidrm_DEPENDENCIES = $(top_builddir)/src/libmobi.la libcommon.a
-mobidrm_LDADD = libcommon.a $(top_builddir)/src/libmobi.la
+mobidrm_DEPENDENCIES = $(top_builddir)/src/libmobi.la libcommon.la
+mobidrm_LDADD = libcommon.la $(top_builddir)/src/libmobi.la
 mobidrm_CFLAGS = $(ISO99_SOURCE) $(DEBUG_CFLAGS) -D_POSIX_C_SOURCE=200112L
 mobidrm_LDFLAGS = $(TOOLS_STATIC)
 endif
