--- origsrc/libopennet-0.9.9/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libopennet-0.9.9/Makefile.am	2010-08-15 20:32:56.135507500 +0900
@@ -0,0 +1,11 @@
+lib_LTLIBRARIES = libopennet.la
+libopennet_la_SOURCES = net.c opennet.c
+libopennet_la_LDFLAGS = -no-undefined
+man_MANS = opennet.3
+
+libopennet_includedir = $(includedir)/libopennet
+libopennet_include_HEADERS = net.h opennet.h
+
+check_PROGRAMS = test
+test_SOURCES = test.c
+test_LDADD = libopennet.la
--- origsrc/libopennet-0.9.9/configure.ac	2006-12-18 22:31:52.000000000 +0900
+++ src/libopennet-0.9.9/configure.ac	2010-08-15 20:13:03.057382500 +0900
@@ -1,21 +1,20 @@
 AC_REVISION($Revision $)
 AC_INIT(libconfig, 0.9.9)
+AM_INIT_AUTOMAKE
 AC_CONFIG_HEADER(config.h)
 
 dnl Checks for programs.
+LT_INIT(win32-dll)
 AC_PROG_CC
 AC_PROG_MAKE_SET
 AC_PROG_INSTALL
 AC_PROG_LN_S
-AC_PROG_RANLIB
-AC_CHECK_TOOL(AR, ar, true)
-AC_AIX
 
 dnl Find out about the host OS (must be done before GET_SHOBJFLAGS)
-DC_CHK_OS_INFO
+#DC_CHK_OS_INFO
 
 dnl Must be done before checking headers and compiler stuff.
-DC_GET_SHOBJFLAGS
+#DC_GET_SHOBJFLAGS
 
 AC_HEADER_DIRENT
 AC_HEADER_STDC
@@ -33,23 +32,23 @@ AC_FUNC_FSEEKO
 
 dnl Checks for library functions.
 AC_CHECK_FUNCS(socket strcspn strdup strstr getpeername)
-AC_REPLACE_FUNCS(uname)
+#AC_REPLACE_FUNCS(uname)
 
-DC_DO_TYPE(uint64_t, unsigned, 8)
-DC_DO_TYPE(int64_t, signed, 8)
-DC_DO_TYPE(uint32_t, unsigned, 4)
-DC_DO_TYPE(int32_t, signed, 4)
-DC_DO_TYPE(uint16_t, unsigned, 2)
-DC_DO_TYPE(int16_t, signed, 2)
+#DC_DO_TYPE(uint64_t, unsigned, 8)
+#DC_DO_TYPE(int64_t, signed, 8)
+#DC_DO_TYPE(uint32_t, unsigned, 4)
+#DC_DO_TYPE(int32_t, signed, 4)
+#DC_DO_TYPE(uint16_t, unsigned, 2)
+#DC_DO_TYPE(int16_t, signed, 2)
 
 dnl Checks for Win32 specific things.
-DC_DO_NETSET
-DC_DO_WIN32
+#DC_DO_NETSET
+#DC_DO_WIN32
 
 dnl Must be done after modifying LIBOBJS
-DC_SYNC_SHLIBOBJS
+#DC_SYNC_SHLIBOBJS
 
 dnl This must be last.
-DC_SETLDRUNPATH($libdir)
+#DC_SETLDRUNPATH($libdir)
 
 AC_OUTPUT(Makefile fopen_net.3 fseek_net.3 lseek_net.3 open_net.3 opennet.3 read_net.3 fread_net.3)
