--- origsrc/bfast-0.6.4e/bfast/RGMatch.c	2010-08-19 06:02:02.000000000 +0900
+++ src/bfast-0.6.4e/bfast/RGMatch.c	2010-12-16 09:41:09.230156000 +0900
@@ -20,7 +20,7 @@ int32_t RGMatchRead(gzFile fp,
 	/* Read in the read length */
 	if(gzread64(fp, &m->readLength, sizeof(int32_t))!=sizeof(int32_t)||
 			gzread64(fp, &m->qualLength, sizeof(int32_t))!=sizeof(int32_t)) {
-		if(feof(fp) != 0) {
+		if(gzeof(fp) != 0) {
 			return EOF;
 		}
 		else {
--- origsrc/bfast-0.6.4e/tests/Makefile.am	2010-08-13 01:09:40.000000000 +0900
+++ src/bfast-0.6.4e/tests/Makefile.am	2010-12-16 10:09:03.027588300 +0900
@@ -7,6 +7,5 @@ TESTS = test.initialize.sh \
 		test.match.sh \
 		test.localalign.sh \
 		test.postprocess.sh \
-		test.solid2fastq.sh \
 		test.diff.sh \
 		test.cleanup.sh
--- origsrc/bfast-0.6.4e/tests/test.cleanup.sh	2010-08-13 01:09:40.000000000 +0900
+++ src/bfast-0.6.4e/tests/test.cleanup.sh	2010-12-16 09:41:09.245277200 +0900
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-. test.definitions.sh
+. ./test.definitions.sh
 
 echo "      Cleaning up files.";
 
--- origsrc/bfast-0.6.4e/tests/test.diff.sh	2010-08-13 01:09:40.000000000 +0900
+++ src/bfast-0.6.4e/tests/test.diff.sh	2010-12-16 09:41:09.252837800 +0900
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-. test.definitions.sh
+. ./test.definitions.sh
 
 #error()
 #{
--- origsrc/bfast-0.6.4e/tests/test.fasta2brg.sh	2010-08-13 01:09:40.000000000 +0900
+++ src/bfast-0.6.4e/tests/test.fasta2brg.sh	2010-12-16 09:41:09.252837800 +0900
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-. test.definitions.sh
+. ./test.definitions.sh
 TMP_DIR="tmp/";
 
 echo "      Building a reference genome.";
--- origsrc/bfast-0.6.4e/tests/test.index.sh	2010-08-13 01:09:40.000000000 +0900
+++ src/bfast-0.6.4e/tests/test.index.sh	2010-12-16 09:41:09.260398400 +0900
@@ -1,5 +1,5 @@
 #!/bin/sh 
-. test.definitions.sh
+. ./test.definitions.sh
 
 echo "      Building an index.";
 
--- origsrc/bfast-0.6.4e/tests/test.initialize.sh	2010-08-13 01:09:40.000000000 +0900
+++ src/bfast-0.6.4e/tests/test.initialize.sh	2010-12-16 09:41:09.260398400 +0900
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-. test.definitions.sh
+. ./test.definitions.sh
 
 echo "      Initializing data for tests.";
 
--- origsrc/bfast-0.6.4e/tests/test.localalign.sh	2010-08-13 01:09:40.000000000 +0900
+++ src/bfast-0.6.4e/tests/test.localalign.sh	2010-12-16 09:41:09.260398400 +0900
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-. test.definitions.sh
+. ./test.definitions.sh
 
 echo "      Running local alignment.";
 
--- origsrc/bfast-0.6.4e/tests/test.match.sh	2010-08-13 01:09:40.000000000 +0900
+++ src/bfast-0.6.4e/tests/test.match.sh	2010-12-16 09:41:09.275519600 +0900
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-. test.definitions.sh
+. ./test.definitions.sh
 
 echo "      Finding matches.";
 
--- origsrc/bfast-0.6.4e/tests/test.postprocess.sh	2010-08-13 01:09:40.000000000 +0900
+++ src/bfast-0.6.4e/tests/test.postprocess.sh	2010-12-16 09:41:09.283080200 +0900
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-. test.definitions.sh
+. ./test.definitions.sh
 
 echo "      Running postprocessing.";
 
