--- origsrc/picohttpparser/CMakeLists.txt	1970-01-01 09:00:00.000000000 +0900
+++ src/picohttpparser/CMakeLists.txt	2025-05-06 15:01:21.229787600 +0900
@@ -0,0 +1,28 @@
+cmake_minimum_required(VERSION 3.28)
+project(picohttpparser
+  VERSION 0
+  DESCRIPTION "Tiny HTTP parser"
+  HOMEPAGE_URL "https://github.com/h2o/picohttpparser")
+
+include(GNUInstallDirs)
+include(CTest)
+
+add_library(picohttpparser SHARED picohttpparser.c)
+set_target_properties(picohttpparser PROPERTIES SOVERSION 0)
+
+add_executable(test-bin test.c picotest/picotest.c)
+target_link_libraries(test-bin picohttpparser)
+add_test(test-bin test-bin)
+
+install(TARGETS picohttpparser
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+install(FILES picohttpparser.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/picohttpparser.pc.in ${CMAKE_BINARY_DIR}/picohttpparser.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/picohttpparser.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/picohttpparser/picohttpparser.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/picohttpparser/picohttpparser.pc.in	2025-05-06 15:01:29.351486600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: picohttpparser
+Description: Tiny HTTP parser
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lpicohttpparser
+Cflags: -I${includedir}
