--- origsrc/kaiju-1.10.1/src/makefile	2024-03-04 05:10:52.000000000 +0900
+++ src/kaiju-1.10.1/src/makefile	2024-10-21 16:37:19.378047800 +0900
@@ -1,7 +1,7 @@
-CC = gcc
-CXX = g++
-CFLAGS = -O3 -DNDEBUG
-CXXFLAGS = -O3 -pthread -std=c++11 -DNDEBUG
+CC ?= gcc
+CXX ?= g++
+CFLAGS ?= -O3 -DNDEBUG 
+CXXFLAGS ?= -O3 -pthread -std=c++11 -DNDEBUG
 LDLIBS = -lpthread -lz
 INCLUDES	= -I./include -I./include/ncbi-blast+
 
