--- origsrc/atasm-1.26/src/Makefile	2025-02-06 03:47:59.000000000 +0900
+++ src/atasm-1.26/src/Makefile	2025-03-12 08:32:25.975439300 +0900
@@ -17,7 +17,7 @@ UNIX    = -DUNIX
 
 # Compiler flags, if you are using egcs, pgcs, or gcc >2.8.1 use:
 #CFLAGS  = -g -Wall $(USEZ) $(DOS) $(UNIX) $(ARCH)
-CFLAGS  = -Wall $(USEZ) $(DOS) $(UNIX) -O3 -fomit-frame-pointer $(ARCH)
+CFLAGS  ?= -Wall $(USEZ) $(DOS) $(UNIX) -O3 -fomit-frame-pointer $(ARCH)
 
 L       =  $(ZLIB) -lm
 CC      = gcc
