--- origsrc/atasm-1.30/src/Makefile	2025-05-21 21:48:28.000000000 +0900
+++ src/atasm-1.30/src/Makefile	2025-05-25 22:03:33.403602600 +0900
@@ -17,7 +17,7 @@ UNIX    = -DUNIX
 
 # Compiler flags, if you are using egcs, pgcs, or gcc >2.8.1 use:
 #CFLAGS  = -g -Wall $(USEZ) $(DOS) $(UNIX) $(ARCH)
-CFLAGS  = -Wall $(USEZ) $(DOS) $(UNIX) -O3 -fomit-frame-pointer $(ARCH)
+CFLAGS  ?= -Wall $(USEZ) $(DOS) $(UNIX) -O3 -fomit-frame-pointer $(ARCH)
 
 L       =  $(ZLIB) -lm
 CC      = gcc
