--- origsrc/rnetclient-2016.0/Makefile.am	2016-04-06 19:46:29.000000000 +0900
+++ src/rnetclient-2016.0/Makefile.am	2017-08-09 22:52:35.968000000 +0900
@@ -3,6 +3,7 @@ SUBDIRS = doc data
 bin_PROGRAMS = rnetclient
 rnetclient_SOURCES = rnetclient.c decfile.c decfile.h pmhash.c pmhash.h \
 	rnet_message.c rnet_message.h rnet_encode.c rnet_encode.h
+rnetclient_LDADD = -largp
 
 check_PROGRAMS = t-parse
 t_parse_SOURCES = t-parse.c decfile.c decfile.h pmhash.c pmhash.h \
