--- origsrc/rnetclient-2015.1/Makefile.am	2015-03-03 05:29:16.000000000 +0900
+++ src/rnetclient-2015.1/Makefile.am	2015-04-23 00:05:08.436670500 +0900
@@ -3,3 +3,4 @@ SUBDIRS = doc
 bin_PROGRAMS = rnetclient
 rnetclient_SOURCES = rnetclient.c decfile.c decfile.h pmhash.c pmhash.h \
 	rnet_message.c rnet_message.h rnet_encode.c rnet_encode.h
+rnetclient_LDADD = -largp
--- origsrc/rnetclient-2015.1/rnetclient.c	2015-03-03 22:03:59.000000000 +0900
+++ src/rnetclient-2015.1/rnetclient.c	2015-04-23 00:04:44.495665200 +0900
@@ -31,6 +31,7 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <netdb.h>
+#include <libgen.h>
 #include <gnutls/gnutls.h>
 #include <zlib.h>
 #include <argp.h>
