--- origsrc/upoll/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/upoll/Makefile.am	2015-02-14 12:57:06.349918800 +0900
@@ -0,0 +1,10 @@
+
+include_HEADERS = src/include/up.h
+
+lib_LTLIBRARIES = libup.la
+libup_la_CPPFLAGS = -I$(top_srcdir)/include
+libup_la_SOURCES = src/upoll.c
+libup_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = upoll.pc
--- origsrc/upoll/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/upoll/configure.ac	2015-02-14 12:56:37.349918800 +0900
@@ -0,0 +1,28 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([upoll], [0])
+AC_CONFIG_SRCDIR([example/echo-server.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE([subdir-objects])
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_MAKE_SET
+LT_INIT
+
+# Checks for header files.
+AC_CHECK_HEADERS([arpa/inet.h fcntl.h netdb.h netinet/in.h stddef.h stdint.h stdlib.h string.h sys/socket.h sys/time.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+AC_TYPE_UINT16_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+
+# Checks for library functions.
+AC_CHECK_FUNCS([memset select socket])
+
+AC_CONFIG_FILES([Makefile upoll.pc])
+AC_OUTPUT
--- origsrc/upoll/upoll.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/upoll/upoll.pc.in	2015-02-14 12:56:46.115543800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: upoll
+Description: Epoll for the epoll impaired
+Version: @VERSION@
+Libs: -L${libdir} -lup
+Cflags: -I${includedir}
