--- origsrc/libfyaml-0.5.4/configure.ac	2020-02-12 04:11:35.000000000 +0900
+++ src/libfyaml-0.5.4/configure.ac	2020-03-06 09:36:50.294582500 +0900
@@ -1,7 +1,7 @@
 AC_PREREQ(2.61)
 
 AC_INIT([libfyaml],
-        m4_esyscmd([build-aux/git-version-gen .tarball-version]),
+        [0.5.4],
 	[pantelis.antoniou@konsulko.com])
 
 AC_CONFIG_SRCDIR([src/lib/fy-parse.c])
--- origsrc/libfyaml-0.5.4/src/Makefile.am	2020-02-12 04:11:35.000000000 +0900
+++ src/libfyaml-0.5.4/src/Makefile.am	2020-03-06 09:37:30.170808600 +0900
@@ -2,6 +2,7 @@ AM_CPPFLAGS = \
         -I$(top_srcdir)/src \
         -I$(top_srcdir)/include
 AM_CFLAGS =
+AM_LDFLAGS = -no-undefined
 
 lib_LTLIBRARIES = libfyaml-@MAJOR@.@MINOR@.la
 libfyaml_@MAJOR@_@MINOR@_la_SOURCES = \
