--- origsrc/gflags-2.2.1/CMakeLists.txt	2017-07-12 02:30:34.000000000 +0900
+++ src/gflags-2.2.1/CMakeLists.txt	2017-08-26 11:40:03.762000000 +0900
@@ -543,7 +543,11 @@ if (INSTALL_HEADERS)
   )
   install (EXPORT ${EXPORT_NAME} DESTINATION ${CONFIG_INSTALL_DIR})
   if (UNIX)
-    install (PROGRAMS src/gflags_completions.sh DESTINATION ${RUNTIME_INSTALL_DIR})
+    if (CYGWIN)
+      install (PROGRAMS src/gflags_completions.sh DESTINATION /etc/profile.d)
+    else ()
+      install (PROGRAMS src/gflags_completions.sh DESTINATION ${RUNTIME_INSTALL_DIR})
+    endif ()
   endif ()
 endif ()
 
