--- origsrc/gflags-2.2.2/CMakeLists.txt	2018-11-12 06:21:00.000000000 +0900
+++ src/gflags-2.2.2/CMakeLists.txt	2019-01-06 09:09:21.921183200 +0900
@@ -566,7 +566,11 @@ if (INSTALL_HEADERS)
     DESTINATION ${CONFIG_INSTALL_DIR}
   )
   if (UNIX)
-    install (PROGRAMS src/gflags_completions.sh DESTINATION ${RUNTIME_INSTALL_DIR})
+    if (CYGWIN)
+      install (PROGRAMS src/gflags_completions.sh DESTINATION /etc/profile.d)
+    else ()
+      install (PROGRAMS src/gflags_completions.sh DESTINATION ${RUNTIME_INSTALL_DIR})
+    endif ()
   endif ()
 endif ()
 
