--- origsrc/gflags-2.0/Makefile.am	2012-01-26 08:59:40.000000000 +0900
+++ src/gflags-2.0/Makefile.am	2014-02-03 07:24:26.978246800 +0900
@@ -65,13 +65,13 @@ WINDOWS_PROJECTS += vsprojects/libgflags
 libgflags_la_SOURCES = $(GFLAGS_SOURCES)
 libgflags_la_CXXFLAGS = $(PTHREAD_CFLAGS) -DNDEBUG
 # -version-info gets passed to libtool
-libgflags_la_LDFLAGS = $(PTHREAD_CFLAGS) -version-info @SO_VERSION@
+libgflags_la_LDFLAGS = $(PTHREAD_CFLAGS) -version-info @SO_VERSION@ $(AM_LDFLAGS)
 libgflags_la_LIBADD = $(PTHREAD_LIBS)
 
 lib_LTLIBRARIES += libgflags_nothreads.la
 libgflags_nothreads_la_SOURCES = $(GFLAGS_SOURCES)
 libgflags_nothreads_la_CXXFLAGS = -DNDEBUG -DNO_THREADS
-libgflags_nothreads_la_LDFLAGS = -version-info @SO_VERSION@
+libgflags_nothreads_la_LDFLAGS = -version-info @SO_VERSION@ $(AM_LDFLAGS)
 
 TESTS += gflags_unittest
 WINDOWS_PROJECTS += vsprojects/gflags_unittest/gflags_unittest.vcproj
