diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x libtool.m4 -x ltoptions.m4 -x ltsugar.m4 -x ltversion.m4 -x lt~obsolete.m4 -x ltmain.sh -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x Makefile.in origsrc/libtexpdf/Makefile.am src/libtexpdf/Makefile.am
--- origsrc/libtexpdf/Makefile.am	2016-01-11 13:26:09.000000000 +0900
+++ src/libtexpdf/Makefile.am	2016-01-11 13:35:05.927886400 +0900
@@ -197,3 +197,6 @@ pkginclude_HEADERS = \
 	tfm.h
 
 libtexpdf_la_LIBADD = $(LIBPNG_LIBS) $(ZLIB_LIBS) $(LIBPAPER_LIBS)
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x libtool.m4 -x ltoptions.m4 -x ltsugar.m4 -x ltversion.m4 -x lt~obsolete.m4 -x ltmain.sh -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x Makefile.in origsrc/libtexpdf/configure.ac src/libtexpdf/configure.ac
--- origsrc/libtexpdf/configure.ac	2016-01-11 13:26:09.000000000 +0900
+++ src/libtexpdf/configure.ac	2016-01-11 13:34:57.766129100 +0900
@@ -75,6 +75,6 @@ AC_DEFINE([HAVE_ZLIB], 1, [Define if you
 AC_DEFINE([HAVE_ZLIB_COMPRESS2], 1, [Define if your zlib has the compress2 function.])
 AC_DEFINE([HAVE_LIBPNG], 1, [Define if you have libpng and its headers.])
 
-AC_CONFIG_FILES([Makefile ])
+AC_CONFIG_FILES([libtexpdf.pc Makefile ])
 
 AC_OUTPUT
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x libtool.m4 -x ltoptions.m4 -x ltsugar.m4 -x ltversion.m4 -x lt~obsolete.m4 -x ltmain.sh -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x Makefile.in origsrc/libtexpdf/dpxfile.c src/libtexpdf/dpxfile.c
--- origsrc/libtexpdf/dpxfile.c	2016-01-11 13:26:09.000000000 +0900
+++ src/libtexpdf/dpxfile.c	2016-01-11 13:32:59.194504100 +0900
@@ -25,7 +25,7 @@
 #include <wchar.h>
 #endif
 
-#include "libtexpdf/libtexpdf.h"
+#include "libtexpdf.h"
 
 #include <time.h>
 #include <unistd.h>
@@ -805,4 +805,3 @@ qcheck_filetype (const char *fqpn, dpx_r
 
   return  r;
 }
-
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x libtool.m4 -x ltoptions.m4 -x ltsugar.m4 -x ltversion.m4 -x lt~obsolete.m4 -x ltmain.sh -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x Makefile.in origsrc/libtexpdf/libtexpdf.h src/libtexpdf/libtexpdf.h
--- origsrc/libtexpdf/libtexpdf.h	2016-01-11 13:26:09.000000000 +0900
+++ src/libtexpdf/libtexpdf.h	2016-01-11 13:32:41.461362500 +0900
@@ -3,7 +3,7 @@
 #define XETEX 1 /* We are all xetex now */
 
 #ifdef BUILDING_LIBTEXPDF
-#include <libtexpdf/config.h>
+#include "config.h"
 #endif
 
 #ifdef HAVE_STRING_H
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x libtool.m4 -x ltoptions.m4 -x ltsugar.m4 -x ltversion.m4 -x lt~obsolete.m4 -x ltmain.sh -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x Makefile.in origsrc/libtexpdf/libtexpdf.pc.in src/libtexpdf/libtexpdf.pc.in
--- origsrc/libtexpdf/libtexpdf.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libtexpdf/libtexpdf.pc.in	2016-01-11 13:36:25.205039400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libtexpdf
+Description: PDF library extracted from TeX's dvipdfmx
+Version: @VERSION@
+Libs: -L${libdir} -ltexpdf
+Cflags: -I${includedir}/libtexpdf
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x libtool.m4 -x ltoptions.m4 -x ltsugar.m4 -x ltversion.m4 -x lt~obsolete.m4 -x ltmain.sh -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x Makefile.in origsrc/libtexpdf/pngimage.c src/libtexpdf/pngimage.c
--- origsrc/libtexpdf/pngimage.c	2016-01-11 13:26:10.000000000 +0900
+++ src/libtexpdf/pngimage.c	2016-01-11 13:33:26.268056500 +0900
@@ -22,7 +22,7 @@
 #define XETEX 1 /* We are all xetex now */
 
 #ifdef BUILDING_LIBTEXPDF
-#include <libtexpdf/config.h>
+#include "config.h"
 #endif
 
 /*
