--- origsrc/libtexpdf/Makefile.am	2024-12-12 04:42:09.000000000 +0900
+++ src/libtexpdf/Makefile.am	2024-12-12 04:47:34.879545200 +0900
@@ -201,3 +201,6 @@ pkginclude_HEADERS = \
 	tfm.h
 
 libtexpdf_la_LIBADD = $(LIBPNG_LIBS) $(ZLIB_LIBS) $(LIBPAPER_LIBS)
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libtexpdf/configure.ac	2024-12-12 04:42:09.000000000 +0900
+++ src/libtexpdf/configure.ac	2024-12-12 04:47:34.895812500 +0900
@@ -78,4 +78,5 @@ AC_DEFINE([HAVE_LIBPNG], 1, [Define if y
 
 AC_CONFIG_FILES([Makefile ])
 
+AC_CONFIG_FILES([libtexpdf.pc])
 AC_OUTPUT
--- origsrc/libtexpdf/libtexpdf.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libtexpdf/libtexpdf.pc.in	2024-12-12 04:47:34.848291900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libtexpdf
+Description: PDF library extracted from TeX's dvipdfmx
+Version: @VERSION@
+Libs: -L${libdir} -ltexpdf
+Cflags: -I${includedir}
