--- origsrc/libukcprog-1.0.2/configure.in	2003-10-12 23:46:07.000000000 +0900
+++ src/libukcprog-1.0.2/configure.in	2010-10-10 14:14:32.012375000 +0900
@@ -18,7 +18,7 @@ AC_PREREQ(2.57)
 # Checks for programs
 AC_PROG_CC
 AC_PROG_INSTALL
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 # Checks for header files
 AC_HEADER_STDC
--- origsrc/libukcprog-1.0.2/src/Makefile.am	2003-10-10 06:01:12.000000000 +0900
+++ src/libukcprog-1.0.2/src/Makefile.am	2010-10-10 14:13:45.840500000 +0900
@@ -5,6 +5,6 @@
 include_HEADERS = ukcprog.h
 lib_LTLIBRARIES = libukcprog.la
 
-libukcprog_la_LDFLAGS = -version-info 0:1:0
+libukcprog_la_LDFLAGS = -version-info 0:1:0 -no-undefined
 
 libukcprog_la_SOURCES = alloc.c config.c e_malloc.c e_realloc.c ebuf.c errf.c formf.c fpgetline.c ip.c logf.c panic.c sccsdata.c ssplit.c strf.c strnf.c strsave.c strtol.c
--- origsrc/libukcprog-1.0.2/src/formf.c	2003-08-22 06:53:42.000000000 +0900
+++ src/libukcprog-1.0.2/src/formf.c	2010-10-10 14:13:45.840500000 +0900
@@ -30,7 +30,9 @@ char ukcprog_formf_sccsid[] = "$Id: form
 #ifndef __FreeBSD__
 #ifndef linux
 extern int sys_nerr, errno;
+#ifndef __CYGWIN__
 extern char *sys_errlist[];
+#endif // __CYGWIN__
 #endif /* !linux */
 #endif /* !__FreeBSD__ */
 #endif /* !MSDOS */
--- origsrc/libukcprog-1.0.2/src/logf.c	2003-03-30 01:30:33.000000000 +0900
+++ src/libukcprog-1.0.2/src/logf.c	2010-10-10 14:13:45.856125000 +0900
@@ -30,7 +30,9 @@ char ukcprog_logf_sccsid[] = "$Id: logf.
 #ifndef __FreeBSD__
 #ifndef linux
 extern int sys_nerr, errno;
+#ifndef __CYGWIN__
 extern char *sys_errlist[];
+#endif // __CYGWIN__
 #endif /* !linux */
 #endif /* !__FreeBSD__ */
 #endif /* !MSDOS */
