--- origsrc/aws-checksums-0.1.2/CMakeLists.txt	2019-01-10 03:17:59.000000000 +0900
+++ src/aws-checksums-0.1.2/CMakeLists.txt	2019-01-29 10:28:11.859595600 +0900
@@ -65,7 +65,7 @@ set_target_properties(aws-checksums PROP
 set(CMAKE_C_FLAGS_DEBUGOPT "")
 set_property(TARGET aws-checksums PROPERTY C_STANDARD 99)
 
-if(BUILD_SHARED_LIBS AND WIN32)
+if(BUILD_SHARED_LIBS)
     target_compile_definitions(aws-checksums PRIVATE "-DAWS_CHECKSUMS_EXPORTS")    
 endif()
 
@@ -100,6 +100,9 @@ target_link_libraries(aws-checksums-test
 set_target_properties(aws-checksums-tests PROPERTIES LINKER_LANGUAGE C)
 set_property(TARGET aws-checksums-tests PROPERTY C_STANDARD 99)
 
+if(CYGWIN)
+    set_target_properties(aws-checksums PROPERTIES SOVERSION 0)
+endif()
 
 include(CTest)
 enable_testing()
@@ -117,7 +120,7 @@ install(FILES ${AWS_CHECKSUMS_HEADERS} D
 aws_prepare_shared_lib_exports(${CMAKE_PROJECT_NAME})
 
 install(EXPORT "${CMAKE_PROJECT_NAME}-targets"
-    DESTINATION "${LIBRARY_DIRECTORY}/${CMAKE_PROJECT_NAME}/cmake/"
+    DESTINATION "${LIBRARY_DIRECTORY}/cmake/${CMAKE_PROJECT_NAME}/"
     NAMESPACE AWS::)
 
 configure_file("cmake/${CMAKE_PROJECT_NAME}-config.cmake"
@@ -125,5 +128,12 @@ configure_file("cmake/${CMAKE_PROJECT_NA
     @ONLY)
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_PROJECT_NAME}-config.cmake"
-    DESTINATION "${LIBRARY_DIRECTORY}/${CMAKE_PROJECT_NAME}/cmake/")
+    DESTINATION "${LIBRARY_DIRECTORY}/cmake/${CMAKE_PROJECT_NAME}/")
+
 
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/aws-checksums.pc.in ${CMAKE_BINARY_DIR}/aws-checksums.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/aws-checksums.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/aws-checksums-0.1.2/aws-checksums.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/aws-checksums-0.1.2/aws-checksums.pc.in	2019-01-29 10:28:11.823593100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: aws-checksums
+Description: Cross-Platform HW accelerated CRC32c and CRC32 with fallback to efficient SW implementations
+Version: @VERSION@
+Libs: -L${libdir} -laws-checksums
+Cflags: -I${includedir}
