--- origsrc/adplug-2.2.1/configure.in	2010-04-06 11:55:31.000000000 +0900
+++ src/adplug-2.2.1/configure.in	2010-07-17 20:34:31.266865800 +0900
@@ -5,6 +5,7 @@ AC_CONFIG_FILES([Makefile src/Makefile d
 AM_INIT_AUTOMAKE
 AM_MAINTAINER_MODE
 #AM_DISABLE_SHARED
+AC_LIBTOOL_WIN32_DLL
 AM_PROG_LIBTOOL
 AC_LANG(C++)
 
--- origsrc/adplug-2.2.1/src/Makefile.am	2010-01-01 17:26:26.000000000 +0900
+++ src/adplug-2.2.1/src/Makefile.am	2010-07-17 20:34:44.532490800 +0900
@@ -9,7 +9,7 @@ hyp.cpp psi.cpp rat.cpp u6m.cpp rol.cpp
 lds.cpp realopl.cpp analopl.cpp temuopl.cpp msc.cpp rix.cpp adl.cpp jbm.cpp \
 cmf.cpp surroundopl.cpp dro2.cpp
 
-libadplug_la_LDFLAGS = -release @VERSION@ -version-info 0 $(libbinio_LIBS)
+libadplug_la_LDFLAGS = -release @VERSION@ -version-info 0 $(libbinio_LIBS) -no-undefined
 
 # -Dstricmp=strcasecmp is a hack. Throughout AdPlug, stricmp() is used to do
 # caseless string comparations. UNIX libcs don't support stricmp(), but do
