--- origsrc/adplug-2.2.1/src/Makefile.am	2010-01-01 17:26:26.000000000 +0900
+++ src/adplug-2.2.1/src/Makefile.am	2011-09-22 01:10:28.695206000 +0900
@@ -9,7 +9,7 @@ hyp.cpp psi.cpp rat.cpp u6m.cpp rol.cpp
 lds.cpp realopl.cpp analopl.cpp temuopl.cpp msc.cpp rix.cpp adl.cpp jbm.cpp \
 cmf.cpp surroundopl.cpp dro2.cpp
 
-libadplug_la_LDFLAGS = -release @VERSION@ -version-info 0 $(libbinio_LIBS)
+libadplug_la_LDFLAGS = -release @VERSION@ -version-info 0 $(libbinio_LIBS) -no-undefined
 
 # -Dstricmp=strcasecmp is a hack. Throughout AdPlug, stricmp() is used to do
 # caseless string comparations. UNIX libcs don't support stricmp(), but do
