--- origsrc/adplug-2.3.1/Makefile.am	2018-04-22 18:08:52.000000000 +0900
+++ src/adplug-2.3.1/Makefile.am	2019-04-02 10:49:40.444572800 +0900
@@ -79,7 +79,7 @@ src_libadplug_la_SOURCES = \
 	src/xad.cpp \
 	src/xsm.cpp
 
-src_libadplug_la_LDFLAGS = -release $(PACKAGE_VERSION) -version-info 0
+src_libadplug_la_LDFLAGS = -no-undefined -release $(PACKAGE_VERSION) -version-info 0
 src_libadplug_la_LIBADD = $(libbinio_LIBS)
 
 # -Dstricmp=strcasecmp is a hack. Throughout AdPlug, stricmp() is used to do
--- origsrc/adplug-2.3.1/doc/libadplug.info	2018-04-22 18:13:12.000000000 +0900
+++ src/adplug-2.3.1/doc/libadplug.info	2019-04-02 11:10:28.571730700 +0900
@@ -1,4 +1,4 @@
-This is libadplug.info, produced by makeinfo version 6.5 from
+This is libadplug.info, produced by makeinfo version 6.6 from
 libadplug.texi.
 
 This manual documents the AdPlug core library, version 2.3.1.
--- origsrc/adplug-2.3.1/doc/stamp-vti	2018-04-22 18:13:11.000000000 +0900
+++ src/adplug-2.3.1/doc/stamp-vti	2019-04-02 10:52:24.397718700 +0900
@@ -1,4 +1,4 @@
-@set UPDATED 22 January 2018
+@set UPDATED 23 January 2018
 @set UPDATED-MONTH January 2018
 @set EDITION 2.3.1
 @set VERSION 2.3.1
--- origsrc/adplug-2.3.1/doc/version.texi	2018-04-22 18:13:11.000000000 +0900
+++ src/adplug-2.3.1/doc/version.texi	2019-04-02 10:52:24.100857200 +0900
@@ -1,4 +1,4 @@
-@set UPDATED 22 January 2018
+@set UPDATED 23 January 2018
 @set UPDATED-MONTH January 2018
 @set EDITION 2.3.1
 @set VERSION 2.3.1
