--- origsrc/cp2k/cmake/libcp2k.pc.in	2024-12-13 13:35:26.000000000 +0900
+++ src/cp2k/cmake/libcp2k.pc.in	2024-12-14 08:21:41.926950600 +0900
@@ -1,11 +1,11 @@
-prefix="@CMAKE_INSTALL_PREFIX@"
-exec_prefix="${prefix}"
-libdir="${prefix}/@CMAKE_INSTALL_LIBDIR@"
-includedir="${prefix}/@CMAKE_INSTALL_INCLUDEDIR@"
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
 
 Name: @PROJECT_NAME@
 Description: @CMAKE_PROJECT_DESCRIPTION@
 URL: @CMAKE_PROJECT_HOMEPAGE_URL@
 Version: @PROJECT_VERSION@
-Cflags: -I"${includedir}/cp2k" -I"${includedir}/cp2k/@CMAKE_Fortran_COMPILER_ID@-@CMAKE_Fortran_COMPILER_VERSION@"
-Libs: -L"${libdir}" -lcp2k
+Cflags: -I${includedir}/cp2k
+Libs: -L${libdir} -lcp2k
--- origsrc/cp2k/src/CMakeLists.txt	2024-12-13 13:35:27.000000000 +0900
+++ src/cp2k/src/CMakeLists.txt	2024-12-13 15:15:53.885196700 +0900
@@ -1566,6 +1566,10 @@ set_target_properties(
              POSITION_INDEPENDENT_CODE ON
              VERSION ${cp2k_VERSION})
 
+if (DEFINED SOVERSION)
+  set_target_properties(cp2k PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 if(CP2K_USE_ACCEL MATCHES "CUDA")
   set_property(TARGET cp2k PROPERTY CUDA_ARCHITECTURES
                                     ${CMAKE_CUDA_ARCHITECTURES})
