--- origsrc/soxr-0.1.0-Source/CMakeLists.txt	2013-02-24 20:25:07.901689200 +0900
+++ src/soxr-0.1.0-Source/CMakeLists.txt	2013-02-24 20:42:44.219782200 +0900
@@ -67,6 +67,10 @@ include (CheckIncludeFiles)
 include (CheckLibraryExists)
 include (TestBigEndian)
 
+if (CYGWIN)
+  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DNDEBUG")
+endif ()
+
 check_library_exists (m pow "" NEED_LIBM)
 if (NEED_LIBM)
   set (CMAKE_REQUIRED_LIBRARIES "m;${CMAKE_REQUIRED_LIBRARIES}")
