--- origsrc/soxr-0.1.1-Source/CMakeLists.txt	2013-09-17 21:46:10.808972700 +0900
+++ src/soxr-0.1.1-Source/CMakeLists.txt	2013-09-17 21:46:11.840222700 +0900
@@ -67,6 +67,10 @@ include (CheckIncludeFiles)
 include (CheckLibraryExists)
 include (TestBigEndian)
 
+if (CYGWIN)
+  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DNDEBUG")
+endif ()
+
 check_library_exists (m pow "" NEED_LIBM)
 if (NEED_LIBM)
   set (CMAKE_REQUIRED_LIBRARIES "m;${CMAKE_REQUIRED_LIBRARIES}")
