--- origsrc/libmicrohttpd-0.9.68/src/microhttpd/mhd_sockets.c	2019-10-18 06:10:22.000000000 +0900
+++ src/libmicrohttpd-0.9.68/src/microhttpd/mhd_sockets.c	2019-12-09 10:22:26.535051500 +0900
@@ -518,7 +518,7 @@ MHD_socket_cork_ (MHD_socket sock,
 #if defined(__FreeBSD__) &&  __FreeBSD__ + 0 >= 9
   /* FreeBSD do not need zero-send for flushing starting from version 9 */
   return 1;
-#elif defined(TCP_NOPUSH) && ! defined(TCP_CORK)
+#elif defined(TCP_NOPUSH) && ! defined(TCP_CORK) && !defined(__CYGWIN__)
   {
     const int dummy = 0;
     /* Force flush data with zero send otherwise Darwin and some BSD systems
