--- origsrc/libmicrohttpd-0.9.25/doc/Makefile.am	2012-09-09 04:12:06.000000000 +0900
+++ src/libmicrohttpd-0.9.25/doc/Makefile.am	2013-02-09 10:16:34.922994000 +0900
@@ -6,9 +6,9 @@ DISTCLEANFILES = \
   libmicrohttpd.dvi \
   libmicrohttpd-tutorial.cps \
   libmicrohttpd-tutorial.dvi
-info_TEXINFOS = \
-  libmicrohttpd.texi \
-  libmicrohttpd-tutorial.texi
+#info_TEXINFOS = \
+#  libmicrohttpd.texi \
+#  libmicrohttpd-tutorial.texi
 microhttpd_TEXINFOS = \
   chapters/basicauthentication.inc \
   chapters/bibliography.inc \
