--- origsrc/libmicrohttpd-0.9.31/configure.ac	2013-10-19 23:45:01.000000000 +0900
+++ src/libmicrohttpd-0.9.31/configure.ac	2013-11-17 14:31:50.873286500 +0900
@@ -211,11 +211,8 @@ fi
 AM_CONDITIONAL(ENABLE_EPOLL, [test "x$enable_epoll" != "xno"])
 
 
-CHECK_PTHREAD
+AX_PTHREAD
 LIBS="$PTHREAD_LIBS $LIBS"
-AC_SUBST(PTHREAD_LIBS)
-AC_SUBST(PTHREAD_LDFLAGS)
-AC_SUBST(PTHREAD_CPPFLAGS)
 
 # Check for headers that are ALWAYS required
 AC_CHECK_HEADERS([fcntl.h math.h errno.h limits.h stdio.h locale.h sys/stat.h sys/types.h pthread.h],,AC_MSG_ERROR([Compiling libmicrohttpd requires standard UNIX headers files]))
