--- origsrc/libmicrohttpd-0.9.32/configure.ac	2013-12-04 05:25:03.000000000 +0900
+++ src/libmicrohttpd-0.9.32/configure.ac	2013-12-08 12:48:20.675616500 +0900
@@ -211,11 +211,8 @@ fi
 AM_CONDITIONAL(ENABLE_EPOLL, [test "x$enable_epoll" != "xno"])
 
 
-CHECK_PTHREAD
+AX_PTHREAD
 LIBS="$PTHREAD_LIBS $LIBS"
-AC_SUBST(PTHREAD_LIBS)
-AC_SUBST(PTHREAD_LDFLAGS)
-AC_SUBST(PTHREAD_CPPFLAGS)
 
 # Check for headers that are ALWAYS required
 AC_CHECK_HEADERS([fcntl.h math.h errno.h limits.h stdio.h locale.h sys/stat.h sys/types.h pthread.h],,AC_MSG_ERROR([Compiling libmicrohttpd requires standard UNIX headers files]))
@@ -433,32 +430,7 @@ AC_DEFINE_UNQUOTED([MHD_GCRYPT_VERSION],
 
 # gnutls
 gnutls=0
-AC_MSG_CHECKING(for gnutls)
-AC_ARG_WITH(gnutls,
-   [  --with-gnutls=PFX   base of gnutls installation],
-   [AC_MSG_RESULT([$with_gnutls])
-    case $with_gnutls in
-      no)
-        ;;
-      yes)
-        AC_CHECK_HEADERS([gnutls/gnutls.h],
-            AC_CHECK_LIB([gnutls], [gnutls_priority_set],
-	     gnutls=true))
-        ;;
-      *)
-        LDFLAGS="-L$with_gnutls/lib $LDFLAGS"
-        CPPFLAGS="-I$with_gnutls/include $CPPFLAGS"
-        AC_CHECK_HEADERS([gnutls/gnutls.h],
-            AC_CHECK_LIB([gnutls], [gnutls_priority_set],
-              EXT_LIB_PATH="-L$with_gnutls/lib $EXT_LIB_PATH"
-              gnutls=true))
-        ;;
-    esac
-   ],
-   [AC_MSG_RESULT([--with-gnutls not specified])
-    AC_CHECK_HEADERS([gnutls/gnutls.h],
-        AC_CHECK_LIB([gnutls], [gnutls_priority_set],
-          gnutls=true))])
+PKG_CHECK_MODULES([GNUTLS], [gnutls], [gnutls=true])
 AM_CONDITIONAL(HAVE_GNUTLS, test x$gnutls = xtrue)
 AC_DEFINE_UNQUOTED([HAVE_GNUTLS], $gnutls, [We have gnutls])
 
