--- origsrc/kitchen_sync-0.42/src/backtrace.h	2015-06-16 17:32:26.000000000 +0900
+++ src/kitchen_sync-0.42/src/backtrace.h	2015-06-20 20:35:11.385402900 +0900
@@ -1,7 +1,9 @@
 #ifndef BACKTRACE_H
 #define BACKTRACE_H
 
+#if !defined(__CYGWIN__)
 #include <execinfo.h>
+#endif
 #include <stdlib.h> /* for free() */
 #include <iostream>
 
@@ -9,11 +11,13 @@ using namespace std;
 
 static void backtrace()
 {
+#if !defined(__CYGWIN__)
 	void *callers[100];
 	size_t size = backtrace(callers, 100);
 	char **symbols = backtrace_symbols(callers, size);
 	for (size_t i = 0; i < size; i++) cerr << symbols[i] << endl;
 	free(symbols);
+#endif
 }
 
 #endif
--- origsrc/kitchen_sync-0.42/src/options.h	2015-06-16 17:32:26.000000000 +0900
+++ src/kitchen_sync-0.42/src/options.h	2015-06-20 20:36:05.917839400 +0900
@@ -2,6 +2,7 @@
 #define OPTIONS_H
 
 #include <getopt.h>
+#include <cstdlib>
 #include <cstring>
 #include <stdexcept>
 #include "commit_level.h"
