--- origsrc/XQilla-2.3.3/Makefile.am	2025-04-30 17:15:39.312759400 +0900
+++ src/XQilla-2.3.3/Makefile.am	2025-04-30 17:16:02.192489100 +0900
@@ -662,3 +662,6 @@ winprojects.xq: $(libxqilla_la_SOURCES)
 	@echo "  order by \$$f/@name" >>winprojects.xq
 	@echo "  return ('&#10;        ', \$$f)" >>winprojects.xq
 	@echo "}&#10;      </filter>}" >>winprojects.xq
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = xqilla.pc
--- origsrc/XQilla-2.3.3/configure.in	2016-03-24 22:38:48.000000000 +0900
+++ src/XQilla-2.3.3/configure.in	2025-04-30 17:16:02.192489100 +0900
@@ -322,5 +322,5 @@ AC_FUNC_ERROR_AT_LINE
 AC_FUNC_MKTIME
 AC_CHECK_FUNCS([floor ftime gettimeofday memmove memset pow sqrt strchr strstr tzset localtime_r])
 
-AC_CONFIG_FILES([Makefile include/Makefile])
+AC_CONFIG_FILES([xqilla.pc Makefile include/Makefile])
 AC_OUTPUT
--- origsrc/XQilla-2.3.3/src/functions/FunctionParseHTML.cpp	2015-05-19 02:39:00.000000000 +0900
+++ src/XQilla-2.3.3/src/functions/FunctionParseHTML.cpp	2025-04-30 18:10:58.854934800 +0900
@@ -30,7 +30,7 @@
 
 #ifdef HAVE_LIBTIDY
 #include <tidy.h>
-#include <buffio.h>
+#include <tidybuffio.h>
 #endif
 
 #include <xercesc/util/XMLString.hpp>
--- origsrc/XQilla-2.3.3/xqilla.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/XQilla-2.3.3/xqilla.pc.in	2025-04-30 17:16:02.192489100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: XQilla
+Description: XQuery and XPath 2.0 library, built on top of Xerces-C
+Version: @VERSION@
+Libs: -L${libdir} -lxqilla
+Cflags: -I${includedir}
