--- origsrc/libchdr/CMakeLists.txt	2021-10-09 09:05:18.000000000 +0900
+++ src/libchdr/CMakeLists.txt	2021-10-09 09:14:12.397191100 +0900
@@ -86,13 +86,16 @@ if (BUILD_SHARED_LIBS)
 
   set_target_properties(chdr PROPERTIES PUBLIC_HEADER "include/libchdr/bitstream.h;include/libchdr/cdrom.h;include/libchdr/chd.h;include/libchdr/chdconfig.h;include/libchdr/coretypes.h;include/libchdr/flac.h;include/libchdr/huffman.h")
   set_target_properties(chdr PROPERTIES VERSION "${CHDR_VERSION_MAJOR}.${CHDR_VERSION_MINOR}")
+  if (CYGWIN)
+    set_target_properties(chdr PROPERTIES SOVERSION "${CHDR_VERSION_MAJOR}.${CHDR_VERSION_MINOR}")
+  endif()
 
   if (CMAKE_BUILD_TYPE MATCHES Release)
     #add_custom_command(TARGET chdr POST_BUILD COMMAND ${CMAKE_STRIP} libchdr.so)
   endif (CMAKE_BUILD_TYPE MATCHES Release)
 
   install(TARGETS chdr
-    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
     ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
     PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/libchdr"
   )
