--- origsrc/VTK-8.0.1/IO/EnSight/vtkEnSight6BinaryReader.cxx	2017-08-31 05:55:05.000000000 +0900
+++ src/VTK-8.0.1/IO/EnSight/vtkEnSight6BinaryReader.cxx	2017-12-14 11:14:55.921000000 +0900
@@ -40,6 +40,9 @@
 // FreeBSD and OSX now deprecate stat64
 # define VTK_STAT_STRUCT struct stat
 # define VTK_STAT_FUNC stat
+#elif defined(__CYGWIN__)
+# define VTK_STAT_STRUCT struct stat
+# define VTK_STAT_FUNC stat
 #else
 // here, we're relying on _FILE_OFFSET_BITS defined in vtkWin32Header.h to help
 // us on POSIX without resorting to using stat64.
--- origsrc/VTK-8.0.1/IO/EnSight/vtkEnSightGoldBinaryReader.cxx	2017-08-31 05:55:05.000000000 +0900
+++ src/VTK-8.0.1/IO/EnSight/vtkEnSightGoldBinaryReader.cxx	2017-12-14 11:36:42.402000000 +0900
@@ -41,6 +41,9 @@
 // FreeBSD and OSX now deprecate stat64
 # define VTK_STAT_STRUCT struct stat
 # define VTK_STAT_FUNC stat
+#elif defined(__CYGWIN__)
+# define VTK_STAT_STRUCT struct stat
+# define VTK_STAT_FUNC stat
 #else
 // here, we're relying on _FILE_OFFSET_BITS defined in vtkWin32Header.h to help
 // us on POSIX without resorting to using stat64.
--- origsrc/VTK-8.0.1/IO/LSDyna/private/LSDynaFamily.cxx	2017-08-31 05:55:05.000000000 +0900
+++ src/VTK-8.0.1/IO/LSDyna/private/LSDynaFamily.cxx	2017-12-14 01:48:28.802643300 +0900
@@ -35,7 +35,7 @@ namespace
 // Decide which of 3 stat varieties to use: stat, stat64, __stat64
 // Usually stat uses 32 bit fields, and stat64 (with underscores in Windows) uses 64 bit fields.
 // But on OS X and FreeBSD, stat uses 64 bit fields these days.
-#if (VTK_SIZEOF_ID_TYPE == 8) && !defined(_DARWIN_FEATURE_64_BIT_INODE) && !defined(__FreeBSD__)
+#if (VTK_SIZEOF_ID_TYPE == 8) && !defined(_DARWIN_FEATURE_64_BIT_INODE) && !defined(__FreeBSD__) && !defined(__CYGWIN__)
   #ifndef _WIN32
     #define USE_STAT_64
   #else
--- origsrc/VTK-8.0.1/ThirdParty/sqlite/vtksqlite/vtk_sqlite3.c	2017-08-31 05:55:05.000000000 +0900
+++ src/VTK-8.0.1/ThirdParty/sqlite/vtksqlite/vtk_sqlite3.c	2017-12-14 01:48:29.075059700 +0900
@@ -7858,7 +7858,7 @@ VTK_SQLITE_PRIVATE void vtk_sqlite3PCach
 #if !defined(VTK_SQLITE_OS_UNIX) && !defined(VTK_SQLITE_OS_OTHER)
 # define VTK_SQLITE_OS_OTHER 0
 # ifndef VTK_SQLITE_OS_WIN
-#   if defined(_WIN32) || defined(WIN32) || defined(__CYGWIN__) || defined(__MINGW32__) || defined(__BORLANDC__)
+#   if defined(_WIN32) || defined(WIN32) || defined(__MINGW32__) || defined(__BORLANDC__)
 #     define VTK_SQLITE_OS_WIN 1
 #     define VTK_SQLITE_OS_UNIX 0
 #     define VTK_SQLITE_OS_OS2 0
