--- origsrc/VTK-9.5.2/Common/Core/vtkLogger.cxx	2025-09-18 05:02:27.000000000 +0900
+++ src/VTK-9.5.2/Common/Core/vtkLogger.cxx	2025-12-10 11:07:30.466745200 +0900
@@ -4,6 +4,8 @@
 
 #include "vtkObjectFactory.h"
 
+#define LOGURU_WITH_STREAMS 1
+
 #if VTK_MODULE_ENABLE_VTK_loguru
 #include <vtk_loguru.h>
 #endif
--- origsrc/VTK-9.5.2/Rendering/GL2PSOpenGL2/CMakeLists.txt	2025-09-18 05:02:27.000000000 +0900
+++ src/VTK-9.5.2/Rendering/GL2PSOpenGL2/CMakeLists.txt	2025-12-10 11:07:30.466745200 +0900
@@ -14,3 +14,4 @@ vtk_module_add_module(VTK::RenderingGL2P
   SOURCES ${vtk_object_factory_source}
   PRIVATE_HEADERS ${vtk_object_factory_header})
 vtk_add_test_mangling(VTK::RenderingGL2PSOpenGL2)
+vtk_module_link(VTK::RenderingGL2PSOpenGL2 PRIVATE GL)
--- origsrc/VTK-9.5.2/ThirdParty/fast_float/CMakeLists.txt	2025-09-18 05:02:27.000000000 +0900
+++ src/VTK-9.5.2/ThirdParty/fast_float/CMakeLists.txt	2025-12-10 11:08:27.858860000 +0900
@@ -9,11 +9,11 @@ vtk_module_third_party(
     SPDX_DOWNLOAD_LOCATION
       "git+https://gitlab.kitware.com/third-party/fast_float.git@for/vtk-20250313-8.0.2"
     VERSION
-      "8.0.2"
+      "8.1.0"
     STANDARD_INCLUDE_DIRS
     HEADER_ONLY
   EXTERNAL
-    VERSION "7.0.0"
+    VERSION "8.1.0"
     PACKAGE FastFloat
     TARGETS FastFloat::fast_float
     STANDARD_INCLUDE_DIRS
--- origsrc/VTK-9.5.2/ThirdParty/ioss/vtkioss/Ioss_Utils.C	2025-09-18 05:02:27.000000000 +0900
+++ src/VTK-9.5.2/ThirdParty/ioss/vtkioss/Ioss_Utils.C	2025-12-10 16:16:48.368015500 +0900
@@ -162,13 +162,13 @@ std::ostream &Ioss::Utils::get_debug_str
 void Ioss::Utils::time_and_date(char *time_string, char *date_string, size_t length)
 {
   std::time_t t    = std::time(nullptr);
-  std::string time = fmt::format("{:%H:%M:%S}", fmt::localtime(t));
+  std::string time = fmt::format("{:%H:%M:%S}", *std::localtime(&t));
   std::string date;
   if (length >= 10) {
-    date = fmt::format("{:%Y/%m/%d}", fmt::localtime(t));
+    date = fmt::format("{:%Y/%m/%d}", *std::localtime(&t));
   }
   else {
-    date = fmt::format("{:%y/%m/%d}", fmt::localtime(t));
+    date = fmt::format("{:%y/%m/%d}", *std::localtime(&t));
   }
   copy_string(time_string, time, 9);
   copy_string(date_string, date, length + 1);
--- origsrc/VTK-9.5.2/ThirdParty/ioss/vtkioss/cgns/Iocgns_Utils.C	2025-09-18 05:02:27.000000000 +0900
+++ src/VTK-9.5.2/ThirdParty/ioss/vtkioss/cgns/Iocgns_Utils.C	2025-12-10 16:25:39.981666500 +0900
@@ -100,7 +100,7 @@
   } while (0)
 
 namespace {
-#if defined(__IOSS_WINDOWS__) || defined(__CYGWIN__)
+#if defined(__IOSS_WINDOWS__)
   const char *strcasestr(const char *haystack, const char *needle)
   {
     std::string lneedle(Ioss::Utils::lowercase(needle));
@@ -1076,8 +1076,8 @@ size_t Iocgns::Utils::common_write_metad
 
   CGERR(cg_goto(file_ptr, base, "end"));
   std::time_t t    = std::time(nullptr);
-  std::string date = fmt::format("{:%Y/%m/%d}", fmt::localtime(t));
-  std::string time = fmt::format("{:%H:%M:%S}", fmt::localtime(t));
+  std::string date = fmt::format("{:%Y/%m/%d}", *std::localtime(&t));
+  std::string time = fmt::format("{:%H:%M:%S}", *std::localtime(&t));
 
   std::string code_version = region.get_optional_property("code_version", "unknown");
   std::string code_name    = region.get_optional_property("code_name", "unknown");
--- origsrc/VTK-9.5.2/ThirdParty/loguru/CMakeLists.txt	2025-09-18 05:02:27.000000000 +0900
+++ src/VTK-9.5.2/ThirdParty/loguru/CMakeLists.txt	2025-12-10 11:07:30.482401000 +0900
@@ -1,4 +1,5 @@
-vtk_module_third_party_internal(
+vtk_module_third_party(
+  INTERNAL
     LICENSE_FILES
       "vtkloguru/LICENSE"
     SPDX_LICENSE_IDENTIFIER
@@ -13,6 +14,11 @@ vtk_module_third_party_internal(
       "PublicDomain-loguru"
     VERSION
       "2.1"
+    STANDARD_INCLUDE_DIRS
+  EXTERNAL
+    PACKAGE loguru
+    TARGETS loguru::loguru
+    VERSION "2.1.0"
     STANDARD_INCLUDE_DIRS)
 
 configure_file(
