--- origsrc/mpimemu-1.0/configure.ac	2013-09-23 03:08:02.000000000 +0900
+++ src/mpimemu-1.0/configure.ac	2015-03-07 16:05:03.921416000 +0900
@@ -33,14 +33,7 @@ AC_PROG_CC([mpicc cc])
 
 # check for adequate mpi support
 AC_MSG_CHECKING([if $CC can compile MPI applications])
-AC_LINK_IFELSE([
-    AC_LANG_PROGRAM(
-        [[#include <mpi.h>]],
-        [[MPI_Finalize();]])],
-    [AC_MSG_RESULT([yes])],
-    [AC_MSG_RESULT([no])
-     AC_MSG_ERROR([$CC cannot compile MPI applications. cannot continue.])]
-)
+PKG_CHECK_MODULES([MPI], [ompi])
 
 AM_PROG_CC_C_O
 AC_PROG_CC_C99
--- origsrc/mpimemu-1.0/src/Makefile.am	2013-09-23 03:06:43.000000000 +0900
+++ src/mpimemu-1.0/src/Makefile.am	2015-03-07 16:07:00.789580100 +0900
@@ -26,6 +26,7 @@ mmu_memory.h mmu_memory.c \
 mmu_process.h mmu_process.c \
 mmu_mpi.h mmu_mpi.c \
 mpimemu.c
+mpimemu_LDADD = $(MPI_LIBS)
 
 dist_bin_SCRIPTS = \
 mpimemu-run \
