--- origsrc/getdns-1.7.0/CMakeLists.txt	2021-06-05 03:53:14.000000000 +0900
+++ src/getdns-1.7.0/CMakeLists.txt	2021-06-07 13:40:07.841046000 +0900
@@ -93,7 +93,7 @@ include(TargetSharedLibraryExports)
 include(TargetSharedLibraryVersion)
 
 # Target Platform
-if (WIN32 OR MINGW OR MSYS OR CYGWIN)
+if (WIN32 OR MINGW OR MSYS)
   set(HOSTOS "windows")
   set(GETDNS_ON_WINDOWS 1)
   set(USE_WINSOCK 1)
@@ -289,10 +289,12 @@ check_include_file(sys/sysctl.h HAVE_SYS
 check_include_file(sys/time.h HAVE_SYS_TIME_H)
 check_include_file(sys/wait.h HAVE_SYS_WAIT_H)
 
-check_include_file(windows.h HAVE_WINDOWS_H)
-check_include_file(winsock.h HAVE_WINSOCK_H)
-check_include_file(winsock2.h HAVE_WINSOCK2_H)
-check_include_file(ws2tcpip.h HAVE_WS2TCPIP_H)
+if (NOT CYGWIN)
+  check_include_file(windows.h HAVE_WINDOWS_H)
+  check_include_file(winsock.h HAVE_WINSOCK_H)
+  check_include_file(winsock2.h HAVE_WINSOCK2_H)
+  check_include_file(ws2tcpip.h HAVE_WS2TCPIP_H)
+endif()
 
 # Check for include declarations
 check_symbol_exists(getentropy unistd.h HAVE_DECL_GETENTROPY)
@@ -1080,15 +1082,15 @@ if (ENABLE_STATIC)
   endif ()
 endif ()
 if (ENABLE_SHARED)
-  install(TARGETS getdns_shared LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+  install(TARGETS getdns_shared RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
   if (USE_LIBEV)
-    install(TARGETS getdns_ext_ev_shared LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+    install(TARGETS getdns_ext_ev_shared RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
   endif ()
   if (USE_LIBEVENT2)
-    install(TARGETS getdns_ext_event_shared LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+    install(TARGETS getdns_ext_event_shared RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
   endif ()
   if (USE_LIBUV)
-    install(TARGETS getdns_ext_uv_shared LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+    install(TARGETS getdns_ext_uv_shared RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
   endif ()
 endif ()
 if (BUILD_GETDNS_QUERY)
--- origsrc/getdns-1.7.0/src/server.c	2021-06-05 03:53:14.000000000 +0900
+++ src/getdns-1.7.0/src/server.c	2021-06-07 14:23:03.152200900 +0900
@@ -38,6 +38,8 @@
 #include <fcntl.h>
 #endif
 
+#include <netinet/tcp.h>
+
 #include "getdns/getdns_extra.h"
 #include "context.h"
 #include "types-internal.h"
--- origsrc/getdns-1.7.0/src/stub.c	2021-06-05 03:53:14.000000000 +0900
+++ src/getdns-1.7.0/src/stub.c	2021-06-07 14:24:55.413728100 +0900
@@ -57,6 +57,7 @@
 #include "general.h"
 #include "pubkey-pinning.h"
 
+#include <netinet/tcp.h>
 
 /* WSA TODO: 
  * STUB_TCP_RETRY added to deal with edge triggered event loops (versus
