--- origsrc/libfnr/Makefile.am	2015-03-24 21:01:44.000000000 +0900
+++ src/libfnr/Makefile.am	2015-03-24 21:07:44.763476600 +0900
@@ -12,3 +12,5 @@ test/raw-ips
 
 ACLOCAL_AMFLAGS = -I m4
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = libfnr.pc
--- origsrc/libfnr/configure.ac	2015-03-24 21:01:44.000000000 +0900
+++ src/libfnr/configure.ac	2015-03-24 21:07:22.932421900 +0900
@@ -66,7 +66,8 @@ AC_CHECK_LIB([ssl], [SSL_set_msg_callbac
 
 LT_INIT
 
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([libfnr.pc
+                 Makefile
                  src/Makefile
                  test/Makefile])
 
--- origsrc/libfnr/libfnr.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libfnr/libfnr.pc.in	2015-03-24 21:09:52.542773500 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libfnr
+Description: Small domain block cipher to encrypt small objects ( < 128 bits )
+Version: @VERSION@
+Libs: -L${libdir} -lfnr
+Cflags: -I${includedir}
+Requires: openssl
--- origsrc/libfnr/src/Makefile.am	2015-03-24 21:01:44.000000000 +0900
+++ src/libfnr/src/Makefile.am	2015-03-24 21:01:57.061328100 +0900
@@ -4,7 +4,7 @@ lib_LTLIBRARIES = libfnr.la
 
 include_HEADERS = fnr.h
 libfnr_la_SOURCES = fnr.c $(include_HEADERS)
-libfnr_la_LDFLAGS = -version-info 1:0:0
+libfnr_la_LDFLAGS = -no-undefined -version-info 1:0:0
 libfnr_la_LIBADD = $(LIBSSL)
 
 if DEBUG
