--- origsrc/quicklz-1.5.0/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/quicklz-1.5.0/Makefile.am	2011-03-30 23:25:32.779737500 +0900
@@ -0,0 +1,4 @@
+include_HEADERS = quicklz.h
+lib_LTLIBRARIES = libquicklz.la
+libquicklz_la_SOURCES = quicklz.c
+libquicklz_la_LDFLAGS = -no-undefined
--- origsrc/quicklz-1.5.0/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/quicklz-1.5.0/configure.ac	2011-03-30 23:26:26.086388700 +0900
@@ -0,0 +1,26 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([quicklz], [1.5.0], [])
+AC_CONFIG_SRCDIR([quicklz.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT([win32-dll disable-static])
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_CHECK_FUNCS([memset])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
