--- origsrc/mutter-3.0.2.1/src/Makefile.am	2011-03-27 04:14:54.000000000 +0900
+++ src/mutter-3.0.2.1/src/Makefile.am	2011-06-04 10:45:45.963675000 +0900
@@ -3,7 +3,7 @@
 
 lib_LTLIBRARIES = libmutter.la
 
-SUBDIRS=wm-tester tools compositor/plugins
+SUBDIRS=wm-tester tools . compositor/plugins
 
 INCLUDES=								\
 	$(MUTTER_CFLAGS)						\
--- origsrc/mutter-3.0.2.1/src/compositor/plugins/Makefile.am	2011-03-27 04:14:54.000000000 +0900
+++ src/mutter-3.0.2.1/src/compositor/plugins/Makefile.am	2011-06-04 10:44:00.577168800 +0900
@@ -3,10 +3,10 @@ pkglibdir=@MUTTER_PLUGIN_DIR@
 
 INCLUDES=@MUTTER_CFLAGS@ -I $(top_srcdir)/src -DMUTTER_LIBEXECDIR=\"$(libexecdir)\" -DHOST_ALIAS=\"@HOST_ALIAS@\" -DMUTTER_LOCALEDIR=\"$(prefix)/@DATADIRNAME@/locale\" -DMUTTER_PKGDATADIR=\"$(pkgdatadir)\" -DMUTTER_DATADIR=\"$(datadir)\" -DG_LOG_DOMAIN=\"mutter\" -DSN_API_NOT_YET_FROZEN=1 -DMUTTER_MAJOR_VERSION=$(MUTTER_MAJOR_VERSION) -DMUTTER_MINOR_VERSION=$(MUTTER_MINOR_VERSION) -DMUTTER_MICRO_VERSION=$(MUTTER_MICRO_VERSION) -DMUTTER_PLUGIN_API_VERSION=$(MUTTER_PLUGIN_API_VERSION) -DMUTTER_PLUGIN_DIR=\"@MUTTER_PLUGIN_DIR@\"
 
-default_la_CFLAGS   = -fPIC
+default_la_CFLAGS   = 
 default_la_SOURCES  = default.c
 default_la_LDFLAGS  = -module -avoid-version -no-undefined
-default_la_LIBADD   = @CLUTTER_LIBS@
+default_la_LIBADD   = @CLUTTER_LIBS@ $(top_builddir)/src/libmutter.la
 
 pkglib_LTLIBRARIES = default.la
 
