--- origsrc/mutter-3.8.4/src/Makefile.am	2013-03-08 22:38:59.000000000 +0900
+++ src/mutter-3.8.4/src/Makefile.am	2013-09-28 08:38:49.281171800 +0900
@@ -3,7 +3,7 @@
 
 lib_LTLIBRARIES = libmutter.la
 
-SUBDIRS=wm-tester tools compositor/plugins
+SUBDIRS=wm-tester tools . compositor/plugins
 
 INCLUDES=								\
 	-DCLUTTER_ENABLE_EXPERIMENTAL_API				\
--- origsrc/mutter-3.8.4/src/compositor/plugins/Makefile.am	2012-05-26 01:17:11.000000000 +0900
+++ src/mutter-3.8.4/src/compositor/plugins/Makefile.am	2013-09-28 08:38:49.296796800 +0900
@@ -6,7 +6,7 @@ INCLUDES=@MUTTER_CFLAGS@ -I $(top_srcdir
 default_la_CFLAGS   = -fPIC
 default_la_SOURCES  = default.c
 default_la_LDFLAGS  = -module -avoid-version -no-undefined
-default_la_LIBADD   = @CLUTTER_LIBS@
+default_la_LIBADD   = $(top_builddir)/src/libmutter.la @CLUTTER_LIBS@
 
 pkglib_LTLIBRARIES = default.la
 
