--- origsrc/snpfile-2.0.1/Makefile.am	2017-08-25 18:19:16.307907600 +0900
+++ src/snpfile-2.0.1/Makefile.am	2017-08-25 18:49:31.884530700 +0900
@@ -89,4 +89,7 @@ libsnpfile_la_SOURCES = common.hh matrix
 	input_archive.hh output_archive.hh snpfile.hh snpfile.cc serialization_trait.hh \
 	type_trait.hh
 libsnpfile_la_LIBADD = $(BOOST_PROGRAM_OPTIONS_LIB) $(BOOST_SIGNALS_LIB) -lstdc++
+libsnpfile_la_LDFLAGS = -no-undefined
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/snpfile-2.0.1/configure.in	2017-08-25 18:19:16.495359000 +0900
+++ src/snpfile-2.0.1/configure.in	2017-08-25 18:57:13.083000000 +0900
@@ -19,6 +19,7 @@ AX_BOOST_PROGRAM_OPTIONS
 AX_BOOST_SIGNALS
 
 AC_OUTPUT([
+    snpfile.pc
 	Makefile
 ])
 
--- origsrc/snpfile-2.0.1/snpfile.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/snpfile-2.0.1/snpfile.pc.in	2017-08-25 18:49:31.861301100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: snpfile
+Description: Library and API for manipulating large SNP datasets
+Version: @VERSION@
+Libs: -L${libdir} -lsnpfile
+Cflags: -I${includedir}
