--- origsrc/libngf/libngf/Makefile.am	2019-06-06 09:50:49.000000000 +0900
+++ src/libngf/libngf/Makefile.am	2019-06-06 09:50:57.731976600 +0900
@@ -9,4 +9,4 @@ libngf0_la_SOURCES	= ngf.h \
 			  proplist.h proplist.c
 libngf0_la_CPPFLAGS	= $(BASE_CFLAGS)
 libngf0_la_LIBADD	= $(BASE_LIBS)
-libngf0_la_LDFLAGS	= -version-info $(NGF_LIBRARY_VERSION) -release $(NGF_RELEASE)
+libngf0_la_LDFLAGS	= -no-undefined -version-info $(NGF_LIBRARY_VERSION) -release $(NGF_RELEASE)
--- origsrc/libngf/tests/test-proplist.c	2019-06-06 09:50:49.000000000 +0900
+++ src/libngf/tests/test-proplist.c	2019-06-06 09:50:57.741978900 +0900
@@ -18,7 +18,11 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
+#if defined(__CYGWIN__)
+#include <limits.h>
+#else
 #include <values.h>
+#endif
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdint.h>
