--- origsrc/arcs-1.2.5/Common/Uncompress.cpp	2023-01-07 01:46:07.000000000 +0900
+++ src/arcs-1.2.5/Common/Uncompress.cpp	2023-11-27 17:41:09.342212600 +0900
@@ -7,6 +7,7 @@
  * @author Shaun Jackman <sjackman@bcgsc.ca>
  */
 
+#if !defined(__CYGWIN__)
 #include "config.h"
 #if HAVE_LIBDL
 
@@ -205,11 +206,15 @@ int open(const char *path, int flags, mo
 } // extern "C"
 
 #endif // HAVE_LIBDL
+#endif // !defined(__CYGWIN__)
+
 /** Initialize the uncompress module. */
 bool uncompress_init()
 {
 #if HAVE_LIBDL
 	signalInit();
-#endif
 	return HAVE_LIBDL;
+#else
+	return 0;
+#endif
 }
