--- origsrc/ta-lib-0.5.0/Makefile.am	2008-02-03 17:02:00.000000000 +0900
+++ src/ta-lib-0.5.0/Makefile.am	2024-01-14 11:19:08.636463900 +0900
@@ -1,3 +1,6 @@
 AUTOMAKE_OPTIONS = foreign 1.4
 SUBDIRS = src src/tools
 bin_SCRIPTS = ta-lib-config
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/ta-lib-0.5.0/configure.in	2008-02-03 17:02:00.000000000 +0900
+++ src/ta-lib-0.5.0/configure.in	2024-01-14 11:39:00.259039500 +0900
@@ -3,10 +3,10 @@
 
 AC_PREREQ(2.59)
 AC_COPYRIGHT([(c) TicTacTec 2005, 2006])
-AC_INIT([ta-lib], [SVN], [http://sourceforge.net/tracker/?group_id=8903&atid=108903])
+AC_INIT([ta-lib], [0.5.0], [http://sourceforge.net/tracker/?group_id=8903&atid=108903])
 AC_CONFIG_SRCDIR([src/ta_func/ta_AD.c])
 AC_CONFIG_HEADER([include/ta_config.h])
-AM_INIT_AUTOMAKE([ta-lib], [SVN])
+AM_INIT_AUTOMAKE
 
 # Checks for programs.
 AC_PROG_CC
@@ -44,4 +44,5 @@ TALIB_LIBRARY_VERSION=0:0:0
 AC_SUBST(TALIB_LIBRARY_VERSION)
 
 AC_CONFIG_FILES([Makefile src/Makefile src/ta_abstract/Makefile src/ta_common/Makefile src/ta_func/Makefile src/tools/Makefile src/tools/gen_code/Makefile src/tools/ta_regtest/Makefile ta-lib-config ta-lib.spec ta-lib.dpkg])
+AC_CONFIG_FILES([ta-lib.pc])
 AC_OUTPUT
--- origsrc/ta-lib-0.5.0/src/Makefile.am	2008-02-03 17:02:00.000000000 +0900
+++ src/ta-lib-0.5.0/src/Makefile.am	2024-01-14 11:07:33.216436300 +0900
@@ -10,5 +10,5 @@ libta_lib_la_LIBADD = \
   ta_func/libta_func.la \
   ta_common/libta_common.la
 
-libta_lib_la_LDFLAGS = -version-info $(TALIB_LIBRARY_VERSION)
+libta_lib_la_LDFLAGS = -no-undefined -version-info $(TALIB_LIBRARY_VERSION)
 
--- origsrc/ta-lib-0.5.0/src/ta_abstract/Makefile.am	2008-02-03 17:02:00.000000000 +0900
+++ src/ta-lib-0.5.0/src/ta_abstract/Makefile.am	2024-01-14 11:07:33.216436300 +0900
@@ -38,7 +38,7 @@ libta_abstract_gc_la_SOURCES = $(libta_a
 libta_abstract_la_LDFLAGS = -version-info $(TALIB_LIBRARY_VERSION)
 libta_abstract_gc_la_LDFLAGS = $(libta_abstract_la_LDFLAGS)
 
-libta_abstract_la_CPPFLAGS = -I../ta_common/ -Iframes/
+libta_abstract_la_CPPFLAGS = -I$(top_srcdir)/src/ta_common/ -I$(top_srcdir)/src/ta_abstract/frames/ -I$(top_srcdir)/include
 
 # The 'gc' version is a minimal version used to just to compile gen_code
 libta_abstract_gc_la_CPPFLAGS = -DTA_GEN_CODE $(libta_abstract_la_CPPFLAGS)
--- origsrc/ta-lib-0.5.0/src/ta_common/Makefile.am	2008-02-03 17:02:00.000000000 +0900
+++ src/ta-lib-0.5.0/src/ta_common/Makefile.am	2024-01-14 11:07:33.216436300 +0900
@@ -1,6 +1,6 @@
 
 noinst_LTLIBRARIES = libta_common.la
-AM_CPPFLAGS = -I../ta_common/
+AM_CPPFLAGS = -I$(top_srcdir)/src/ta_common/ -I$(top_srcdir)/include
 
 libta_common_la_SOURCES = ta_global.c \
 	ta_retcode.c \
--- origsrc/ta-lib-0.5.0/src/ta_func/Makefile.am	2008-02-03 17:02:00.000000000 +0900
+++ src/ta-lib-0.5.0/src/ta_func/Makefile.am	2024-01-14 11:07:33.216436300 +0900
@@ -1,6 +1,6 @@
 
 noinst_LTLIBRARIES = libta_func.la
-AM_CPPFLAGS = -I../ta_common/
+AM_CPPFLAGS = -I$(top_srcdir)/src/ta_common -I$(top_srcdir)/include
 
 libta_func_la_SOURCES = ta_utility.c \
 	ta_ACCBANDS.c \
--- origsrc/ta-lib-0.5.0/src/tools/gen_code/Makefile.am	2008-02-03 17:02:00.000000000 +0900
+++ src/ta-lib-0.5.0/src/tools/gen_code/Makefile.am	2024-01-14 11:34:18.982138800 +0900
@@ -5,10 +5,10 @@ noinst_PROGRAMS = gen_code
 
 gen_code_SOURCES = gen_code.c
 
-gen_code_CPPFLAGS = -I../../ta_common
+gen_code_CPPFLAGS = -I$(top_srcdir)/src/ta_common -I$(top_srcdir)/include
 gen_code_LDFLAGS = -L../../ta_common -L../../ta_abstract -L../../ta_func
 gen_code_LDADD = -lta_common -lta_abstract_gc -lta_func -lm
 
 all-local:
-	$(MAKE) $(AM_MAKEFLAGS) gen_code
-	cp gen_code ../../../bin
+	$(MAKE) $(AM_MAKEFLAGS) gen_code$(EXEEXT)
+	cp gen_code$(EXEEXT) ../../../bin
--- origsrc/ta-lib-0.5.0/src/tools/gen_code/gen_code.c	2008-02-03 17:02:00.000000000 +0900
+++ src/ta-lib-0.5.0/src/tools/gen_code/gen_code.c	2024-01-14 11:14:03.685614300 +0900
@@ -3453,7 +3453,7 @@ static void printFuncHeaderDoc( FILE *ou
    int first;
 
    fprintf( out, "%sTA_%s - %s\n", prefix, funcInfo->name, funcInfo->hint );
-   fprintf( out, prefix );
+   fprintf( out, "%s", prefix );
 
    fprintf( out, "\n" );
    fprintf( out, "%sInput  = ", prefix );
--- origsrc/ta-lib-0.5.0/src/tools/ta_regtest/Makefile.am	2008-02-03 17:02:00.000000000 +0900
+++ src/ta-lib-0.5.0/src/tools/ta_regtest/Makefile.am	2024-01-14 11:07:33.216436300 +0900
@@ -1,7 +1,8 @@
 
 # ta_regrest is used for testing and should
 # not be installed.
-noinst_PROGRAMS = ta_regtest
+TESTS = ta_regtest
+check_PROGRAMS = ta_regtest
 
 ta_regtest_SOURCES = ta_regtest.c \
 	test_data.c \
@@ -29,10 +30,11 @@ ta_regtest_SOURCES = ta_regtest.c \
 	ta_test_func/test_trange.c \
 	test_internals.c
 
-ta_regtest_CPPFLAGS = -I../../ta_func \
-		      -I../../ta_common/trio \
-		      -I../../ta_common/mt \
-		      -I../../ta_common \
-		      -I../../ta_abstract
+ta_regtest_CPPFLAGS = -I$(top_srcdir)/src/ta_func \
+		      -I$(top_srcdir)/src/ta_common/trio \
+		      -I$(top_srcdir)/src/ta_common/mt \
+		      -I$(top_srcdir)/src/ta_common \
+		      -I$(top_srcdir)/src/ta_abstract \
+		      -I$(top_srcdir)/include
 ta_regtest_LDFLAGS = -L../.. -lta_lib \
 		     -lm
--- origsrc/ta-lib-0.5.0/ta-lib-config.in	2008-02-03 17:02:00.000000000 +0900
+++ src/ta-lib-0.5.0/ta-lib-config.in	2024-01-14 11:07:33.216436300 +0900
@@ -14,10 +14,10 @@ while test $# -gt 0; do
       echo @VERSION@
       ;;
     --cflags)
-      echo -I@includedir@/ta-lib @DEFS@
+      echo -I/usr/include/ta-lib
       ;;
     --libs)
-      echo -L@libdir@ @LIBS@ 
+      echo -lta_lib @LIBS@
       ;;
     *)
       echo "${usage}" 1>&2
--- origsrc/ta-lib-0.5.0/ta-lib.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/ta-lib-0.5.0/ta-lib.pc.in	2024-01-14 11:27:02.832820600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: ta-lib
+Description: Technical Analysis Library
+Version: @VERSION@
+Libs: -L${libdir} -lta_lib
+Cflags: -I${includedir}
