--- origsrc/ta-lib-0.6.4/src/Makefile.am	2025-01-11 13:49:26.000000000 +0900
+++ src/ta-lib-0.6.4/src/Makefile.am	2025-04-04 19:26:18.520617700 +0900
@@ -10,3 +10,4 @@ libta_lib_la_LIBADD = \
   ta_func/libta_func.la \
   ta_common/libta_common.la
 
+libta_lib_la_LDFLAGS = -no-undefined
--- origsrc/ta-lib-0.6.4/src/ta_abstract/Makefile.am	2025-01-11 13:49:26.000000000 +0900
+++ src/ta-lib-0.6.4/src/ta_abstract/Makefile.am	2025-04-04 19:26:18.520617700 +0900
@@ -37,7 +37,7 @@ libta_abstract_gc_la_SOURCES = $(libta_a
 
 libta_abstract_gc_la_LDFLAGS = $(libta_abstract_la_LDFLAGS)
 
-libta_abstract_la_CPPFLAGS = -I../ta_common/ -Iframes/
+libta_abstract_la_CPPFLAGS = -I$(top_srcdir)/src/ta_common/ -I$(top_srcdir)/src/ta_abstract/frames/ -I$(top_srcdir)/include
 
 # The 'gc' version is a minimal version used to just to compile gen_code
 libta_abstract_gc_la_CPPFLAGS = -DTA_GEN_CODE $(libta_abstract_la_CPPFLAGS)
--- origsrc/ta-lib-0.6.4/src/ta_common/Makefile.am	2025-01-11 13:49:26.000000000 +0900
+++ src/ta-lib-0.6.4/src/ta_common/Makefile.am	2025-04-04 19:26:18.520617700 +0900
@@ -1,6 +1,6 @@
 
 noinst_LTLIBRARIES = libta_common.la
-AM_CPPFLAGS = -I../ta_common/
+AM_CPPFLAGS = -I$(top_srcdir)/src/ta_common/ -I$(top_srcdir)/include
 
 libta_common_la_SOURCES = ta_global.c \
 	ta_retcode.c \
--- origsrc/ta-lib-0.6.4/src/ta_func/Makefile.am	2025-01-11 13:49:26.000000000 +0900
+++ src/ta-lib-0.6.4/src/ta_func/Makefile.am	2025-04-04 19:26:18.520617700 +0900
@@ -1,6 +1,6 @@
 
 noinst_LTLIBRARIES = libta_func.la
-AM_CPPFLAGS = -I../ta_common/
+AM_CPPFLAGS = -I$(top_srcdir)/src/ta_common -I$(top_srcdir)/include
 
 libta_func_la_SOURCES = ta_utility.c \
 	ta_ACCBANDS.c \
--- origsrc/ta-lib-0.6.4/src/tools/gen_code/Makefile.am	2025-01-11 13:49:26.000000000 +0900
+++ src/ta-lib-0.6.4/src/tools/gen_code/Makefile.am	2025-04-04 19:26:18.520617700 +0900
@@ -4,7 +4,7 @@ noinst_PROGRAMS = gen_code
 
 gen_code_SOURCES = gen_code.c
 
-gen_code_CPPFLAGS = -I../../ta_common
+gen_code_CPPFLAGS = -I$(top_srcdir)/src/ta_common -I$(top_srcdir)/include
 gen_code_LDFLAGS = -no-undefined
 gen_code_LDADD = \
 	../../ta_common/libta_common.la \
@@ -13,4 +13,4 @@ gen_code_LDADD = \
 	$(LIBM)
 
 all-local: gen_code
-	$(LIBTOOL) --mode=execute ../post-build-bin.sh gen_code
+	$(LIBTOOL) --mode=execute $(srcdir)/../post-build-bin.sh gen_code
--- origsrc/ta-lib-0.6.4/src/tools/ta_regtest/Makefile.am	2025-01-11 13:49:26.000000000 +0900
+++ src/ta-lib-0.6.4/src/tools/ta_regtest/Makefile.am	2025-04-04 19:26:18.520617700 +0900
@@ -31,13 +31,14 @@ ta_regtest_SOURCES = ta_regtest.c \
 	ta_test_func/test_avgdev.c \
 	test_internals.c
 
-ta_regtest_CPPFLAGS = -I../../ta_func \
-		      -I../../ta_common/trio \
-		      -I../../ta_common/mt \
-		      -I../../ta_common \
-		      -I../../ta_abstract
+ta_regtest_CPPFLAGS = -I$(top_srcdir)/src/ta_func \
+		      -I$(top_srcdir)/src/ta_common/trio \
+		      -I$(top_srcdir)/src/ta_common/mt \
+		      -I$(top_srcdir)/src/ta_common \
+		      -I$(top_srcdir)/src/ta_abstract \
+		      -I$(top_srcdir)/include
 ta_regtest_LDFLAGS = -no-undefined
 ta_regtest_LDADD = ../../libta-lib.la $(LIBM)
 
 all-local: ta_regtest
-	$(LIBTOOL) --mode=execute ../post-build-bin.sh ta_regtest
+	$(LIBTOOL) --mode=execute $(srcdir)/../post-build-bin.sh ta_regtest
