--- origsrc/libnavajo-1.5/CMakeLists.txt	2017-05-25 04:26:23.000000000 +0900
+++ src/libnavajo-1.5/CMakeLists.txt	2017-11-13 21:51:03.443000000 +0900
@@ -86,10 +86,15 @@ set(EXECUTABLE_OUTPUT_PATH "${CMAKE_BINA
 ###############      build the library     #####################
 
 add_library(navajo SHARED ${sources_lib} ${headers_lib} )
-add_library(navajoStatic STATIC ${sources_lib} ${headers_lib} )
+option(ENABLE_STATIC "enable static" OFF)
+if(ENABLE_STATIC)
+  add_library(navajoStatic STATIC ${sources_lib} ${headers_lib} )
+endif()
 
 set_target_properties (navajo PROPERTIES VERSION ${LIBNAVAJO_VERSION} SOVERSION ${LIBNAVAJO_MVERSION})
-set_target_properties(navajoStatic PROPERTIES OUTPUT_NAME navajo)
+if(ENABLE_STATIC)
+  set_target_properties(navajoStatic PROPERTIES OUTPUT_NAME navajo)
+endif()
 
 include_directories (${PROJECT_SOURCE_DIR}/include)
 
@@ -101,9 +106,11 @@ target_link_libraries(navajo ${ZLIB_LIBR
 
 install(FILES ${headers_lib} DESTINATION include/libnavajo COMPONENT headers)
 install(FILES ${mpfdHeaders_lib} DESTINATION include/MPFDParser COMPONENT headers)
-install(TARGETS navajo LIBRARY DESTINATION lib COMPONENT libraries)
+install(TARGETS navajo RUNTIME DESTINATION bin ARCHIVE DESTINATION lib COMPONENT libraries)
 
-install(TARGETS navajoStatic ARCHIVE DESTINATION lib LIBRARY DESTINATION lib COMPONENT libraries)
+if(ENABLE_STATIC)
+  install(TARGETS navajoStatic ARCHIVE DESTINATION lib LIBRARY DESTINATION lib COMPONENT libraries)
+endif()
 
 ############### navajoPrecompiler binary generation ###################
 file( GLOB  navajoPrecompiler_source ${PROJECT_SOURCE_DIR}/src/navajoPrecompiler.cc )
@@ -126,3 +133,11 @@ COMMENT "Generating API documentation wi
 )
 endif(DOXYGEN_FOUND)
 
+INSTALL(FILES ${CMAKE_SOURCE_DIR}/FindLibnavajo.cmake DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/cmake/libnavojo)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libnavajo.pc.in ${CMAKE_BINARY_DIR}/libnavajo.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libnavajo.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libnavajo-1.5/libnavajo.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libnavajo-1.5/libnavajo.pc.in	2017-11-13 21:48:08.665000000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libnavajo
+Description: Http server with local dynamic or precompiled repository containers
+Version: @LIBNAVAJO_VERSION@
+Libs: -L${libdir} -lnavajo
+Cflags: -I${includedir}
