--- origsrc/lodepng/CMakeLists.txt	1970-01-01 09:00:00.000000000 +0900
+++ src/lodepng/CMakeLists.txt	2025-04-23 20:02:20.470354900 +0900
@@ -0,0 +1,36 @@
+cmake_minimum_required(VERSION 3.28)
+project(lodepng
+  VERSION 0
+  DESCRIPTION "PNG image decoder and encoder"
+  HOMEPAGE_URL "https://github.com/lvandeve/lodepng")
+
+include(GNUInstallDirs)
+include(CTest)
+
+add_library(lodepng SHARED "lodepng.cpp")
+set_target_properties(lodepng PROPERTIES SOVERSION 0)
+install(TARGETS lodepng
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+install(FILES lodepng.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+
+add_executable(pngdetail "lodepng_util.cpp" "pngdetail.cpp")
+target_link_libraries(pngdetail PRIVATE lodepng)
+install(TARGETS pngdetail RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+
+find_package(SDL2 REQUIRED)
+
+add_executable(benchmark "lodepng_util.cpp" "lodepng_benchmark.cpp")
+target_link_libraries(benchmark PRIVATE lodepng SDL2::SDL2)
+
+add_executable(unittest "lodepng_util.cpp" "lodepng_unittest.cpp")
+target_link_libraries(unittest PRIVATE lodepng)
+add_test(NAME unittest COMMAND unittest)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/lodepng.pc.in ${CMAKE_BINARY_DIR}/lodepng.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/lodepng.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/lodepng/lodepng.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/lodepng/lodepng.pc.in	2025-04-23 20:02:29.970699400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: lodepng
+Description: PNG image decoder and encoder
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -llodepng
+Cflags: -I${includedir}
