--- origsrc/double-conversion-3.0.0/CMakeLists.txt	2017-08-05 23:39:46.000000000 +0900
+++ src/double-conversion-3.0.0/CMakeLists.txt	2018-02-14 01:17:10.572432600 +0900
@@ -68,3 +68,11 @@ install(FILES
 # generates install cmake files to find libraries in installation.
 install(EXPORT double-conversionLibraryDepends DESTINATION
   "${INSTALL_CMAKE_DIR}" COMPONENT dev)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "3.0.0")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/double-conversion.pc.in ${CMAKE_BINARY_DIR}/double-conversion.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/double-conversion.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/double-conversion-3.0.0/double-conversion.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/double-conversion-3.0.0/double-conversion.pc.in	2018-02-14 01:18:53.961000000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: double-conversion
+Description: Binary-decimal and decimal-binary routines for IEEE doubles
+Version: @VERSION@
+Libs: -L${libdir} -ldouble-conversion
+Cflags: -I${includedir}/double-conversion
