--- origsrc/double-conversion-2.0.1/CMakeLists.txt	2014-01-13 01:31:24.000000000 +0900
+++ src/double-conversion-2.0.1/CMakeLists.txt	2016-12-21 17:55:31.514766600 +0900
@@ -2,8 +2,7 @@ cmake_minimum_required(VERSION 2.8)
 project(double-conversion)
 
 # pick a version #
-set(double-conversion_VERSION
-  1.1.1)
+set(double-conversion_VERSION 2.0.1)
 
 # set paths for install -- empty initially
 # Offer the user the choice of overriding the installation directories
@@ -87,3 +86,10 @@ install(FILES
 # generates install cmake files to find libraries in installation.
 install(EXPORT double-conversionLibraryDepends DESTINATION
   "${INSTALL_CMAKE_DIR}" COMPONENT dev)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/${PROJECT_NAME}.pc.in ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/double-conversion-2.0.1/double-conversion.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/double-conversion-2.0.1/double-conversion.pc.in	2016-12-21 17:54:58.135809800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: double-conversion
+Description: Binary-decimal and decimal-binary routines for IEEE doubles
+Version: @double-conversion_VERSION@
+Libs: -L${libdir} -ldouble-conversion
+Cflags: -I${includedir}/double-conversion
--- origsrc/double-conversion-2.0.1/src/CMakeLists.txt	2014-01-13 01:31:24.000000000 +0900
+++ src/double-conversion-2.0.1/src/CMakeLists.txt	2016-12-21 17:50:25.273427500 +0900
@@ -27,7 +27,7 @@ ${headers}
 # associates the list of headers with the library
 # for the purposes of installation/import into other projects
 set_target_properties(double-conversion
-    PROPERTIES PUBLIC_HEADER "${headers}")
+    PROPERTIES PUBLIC_HEADER "${headers}" SOVERSION 0)
 
 #
 # install command to set up library install
@@ -35,7 +35,7 @@ set_target_properties(double-conversion
 # pulls along all the header files with the library.
 install(TARGETS double-conversion
   EXPORT double-conversionLibraryDepends
-  RUNTIME DESTINATION "${INSTALL_BIN_DIR}" COMPONENT bin
+  RUNTIME DESTINATION "${INSTALL_BIN_DIR}/bin" COMPONENT bin
   LIBRARY DESTINATION "${INSTALL_LIB_DIR}/lib" COMPONENT shlib
   ARCHIVE DESTINATION "${INSTALL_LIB_DIR}/lib" COMPONENT lib
   PUBLIC_HEADER DESTINATION "${INSTALL_INCLUDE_DIR}/include/double-conversion"
