--- origsrc/libfirm-1.21.0/include/libfirm/adt/xmalloc.h	2012-11-20 02:07:16.000000000 +0900
+++ src/libfirm-1.21.0/include/libfirm/adt/xmalloc.h	2014-07-21 20:01:21.319974800 +0900
@@ -35,7 +35,7 @@
 #ifdef _WIN32
 #include <malloc.h>
 #endif
-#if defined(__linux__) || defined(__APPLE__)
+#if defined(__linux__) || defined(__APPLE__) || defined(__CYGWIN__)
 #include <alloca.h>
 #endif
 
--- origsrc/libfirm-1.21.0/ir/tv/fltcalc.c	2012-11-20 02:07:17.000000000 +0900
+++ src/libfirm-1.21.0/ir/tv/fltcalc.c	2014-07-21 20:01:21.335599800 +0900
@@ -46,7 +46,7 @@
 
 static long double string_to_long_double(const char *str)
 {
-#if __STDC_VERSION__ >= 199901L || _POSIX_C_SOURCE >= 200112L
+#if !defined(__CYGWIN__) && (__STDC_VERSION__ >= 199901L || _POSIX_C_SOURCE >= 200112L)
 	return strtold(str, NULL);
 #else
 	return strtod(str, NULL);
