--- origsrc/iroha-ed25519-2.0.2/CMakeLists.txt	2019-04-11 19:24:40.000000000 +0900
+++ src/iroha-ed25519-2.0.2/CMakeLists.txt	2025-02-10 00:45:54.651593700 +0900
@@ -169,3 +169,10 @@ endif ()
 if (FUZZING)
   add_subdirectory(fuzzing)
 endif ()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/ed25519.pc.in ${CMAKE_BINARY_DIR}/ed25519.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/ed25519.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/iroha-ed25519-2.0.2/cmake/ed25519_add_test.cmake	2019-04-11 19:24:40.000000000 +0900
+++ src/iroha-ed25519-2.0.2/cmake/ed25519_add_test.cmake	2025-02-09 20:16:45.993208000 +0900
@@ -1,8 +1,8 @@
 function(ed25519_add_test test_name)
   add_executable(${test_name} ${ARGN})
   target_link_libraries(${test_name} PRIVATE
-    GTest::main
-    GTest::gtest
+    gtest
+    gtest_main
     )
   add_test(
     NAME    ${test_name}
--- origsrc/iroha-ed25519-2.0.2/ed25519.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/iroha-ed25519-2.0.2/ed25519.pc.in	2025-02-10 00:43:57.958801500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: ed25519
+Description: RFC8032 compatible Ed25519 implementation
+Version: @VERSION@
+Libs: -L${libdir} -led25519
+Cflags: -I${includedir}
