--- origsrc/libfds-0.6.0/CMakeModules/git_info.cmake	2024-08-05 21:02:42.000000000 +0900
+++ src/libfds-0.6.0/CMakeModules/git_info.cmake	2024-08-31 09:19:02.309696100 +0900
@@ -10,7 +10,7 @@ unset(GIT_USER_NAME)
 unset(GIT_USER_EMAIL)
 
 find_package(Git QUIET)
-if (GIT_FOUND)
+if (GIT_FOUND AND EXISTS "${CMAKE_SOURCE_DIR}/.git")
 	# Get an abbreviated commit hash of the latest commit
 	execute_process(
 		COMMAND ${GIT_EXECUTABLE} "log" "-1" "--format=%h"
--- origsrc/libfds-0.6.0/src/CMakeLists.txt	2024-08-05 21:02:42.000000000 +0900
+++ src/libfds-0.6.0/src/CMakeLists.txt	2024-08-31 09:18:01.877705000 +0900
@@ -87,6 +87,7 @@ set_target_properties(fds PROPERTIES
 
 # Installation targets
 install(
-	TARGETS fds LIBRARY
-	DESTINATION ${INSTALL_DIR_LIB}
+	TARGETS fds
+	RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION ${INSTALL_DIR_LIB}
 )
