--- origsrc/gssdp-0.14.6/libgssdp/Makefile.am	2013-10-30 22:57:17.000000000 +0900
+++ src/gssdp-0.14.6/libgssdp/Makefile.am	2014-01-04 16:08:56.446778200 +0900
@@ -50,10 +50,11 @@ libgssdp_1_0_la_SOURCES = $(introspectio
 
 libgssdp_1_0_la_LIBADD = $(LIBGSSDP_LIBS)
 
+libgssdp_1_0_la_LDFLAGS += -no-undefined
+
 if OS_WIN32
 
 libgssdp_1_0_la_LIBADD += -lws2_32 -liphlpapi
-libgssdp_1_0_la_LDFLAGS += -no-undefined
 endif
 
 if TARGET_ANDROID
