--- origsrc/gssdp-1.2.2/libgssdp/gssdp-net-posix.c	2020-01-02 20:44:00.000000000 +0900
+++ src/gssdp-1.2.2/libgssdp/gssdp-net-posix.c	2020-01-18 19:32:35.008829300 +0900
@@ -568,10 +568,11 @@ gssdp_net_list_devices (void)
                 goto out;
         }
 
+#if defined(AF_PACKET)
         for (ifa = ifa_list; ifa != NULL; ifa = ifa->ifa_next) {
                 g_hash_table_add (interfaces, g_strdup (ifa->ifa_name));
         }
-
+#endif
 
         freeifaddrs (ifa_list);
 
--- origsrc/gssdp-1.2.2/meson.build	2020-01-02 20:44:00.000000000 +0900
+++ src/gssdp-1.2.2/meson.build	2020-01-18 19:32:35.040128000 +0900
@@ -12,6 +12,9 @@ pktinfo_test = '''#define _GNU_SOURCE
 #ifdef __APPLE__
 #define __APPLE_USE_RFC_3542
 #endif
+#if defined(__CYGWIN__)
+#error "IPV6_RECVPKTINFO not defined in cygwin"
+#endif
 #include <netinet/ip.h>
 struct in_pktinfo pktinfo;
 struct in6_pktinfo pktinfo6;
