--- origsrc/gssdp-1.1.1/meson.build	2018-12-11 06:21:06.000000000 +0900
+++ src/gssdp-1.1.1/meson.build	2019-01-06 23:19:30.945416600 +0900
@@ -12,6 +12,9 @@ pktinfo_test = '''#define _GNU_SOURCE
 #ifdef __APPLE__
 #define __APPLE_USE_RFC_3542
 #endif
+#if defined(__CYGWIN__)
+#error "IPV6_RECVPKTINFO not defined in cygwin"
+#endif
 #include <netinet/ip.h>
 struct in_pktinfo pktinfo;
 struct in6_pktinfo pktinfo6;
