--- origsrc/gssdp-1.0.2/configure.ac	2017-03-20 20:43:12.000000000 +0900
+++ src/gssdp-1.0.2/configure.ac	2018-04-10 01:11:09.700000000 +0900
@@ -123,6 +123,8 @@ AC_MSG_RESULT([$target_android])
 AM_CONDITIONAL(TARGET_ANDROID, [test $target_android = yes])
 
 dnl Check whether in_pktinfo is available
+HAVE_PKTINFO=no
+if test "$host_os" != "cygwin"; then
 AC_CHECK_TYPE(struct in_pktinfo,
               [
                HAVE_PKTINFO=yes
@@ -131,6 +133,7 @@ AC_CHECK_TYPE(struct in_pktinfo,
               [
                HAVE_PKTINFO=no
               ], [#include <netinet/ip.h>])
+fi
 AM_CONDITIONAL([HAVE_PKTINFO], [test $HAVE_PKTINFO = yes], [])
 
 dnl Check for if_nametoindex
--- origsrc/gssdp-1.0.2/libgssdp/Makefile.am	2017-03-20 20:38:52.000000000 +0900
+++ src/gssdp-1.0.2/libgssdp/Makefile.am	2018-04-10 01:12:12.100000000 +0900
@@ -54,11 +54,10 @@ libgssdp_1_0_la_SOURCES += gssdp-pktinfo
 endif
 
 libgssdp_1_0_la_LIBADD = $(LIBGSSDP_LIBS)
+libgssdp_1_0_la_LDFLAGS += -no-undefined
 
 if OS_WIN32
-
 libgssdp_1_0_la_LIBADD += -lws2_32 -liphlpapi
-libgssdp_1_0_la_LDFLAGS += -no-undefined
 endif
 
 if TARGET_ANDROID
--- origsrc/gssdp-1.0.2/tests/Makefile.am	2017-03-20 20:38:52.000000000 +0900
+++ src/gssdp-1.0.2/tests/Makefile.am	2018-04-10 09:03:35.686000200 +0900
@@ -15,8 +15,8 @@ test_functional_SOURCES = test-functiona
 test_functional_LDFLAGS = $(WARN_LDFLAGS)
 
 LDADD = \
-	$(top_builddir)/libgssdp/libgssdp-1.0.la \
 	libtestutil.a \
+	$(top_builddir)/libgssdp/libgssdp-1.0.la \
 	$(LIBGSSDP_LIBS)
 
 AM_CFLAGS = \
