--- origsrc/gssdp-0.12.1/libgssdp/Makefile.am	2011-11-29 21:07:01.000000000 +0900
+++ src/gssdp-0.12.1/libgssdp/Makefile.am	2012-03-31 23:52:09.138520300 +0900
@@ -38,7 +38,7 @@ introspection_sources = $(libgssdpinc_HE
 			gssdp-resource-browser.c\
 			gssdp-resource-group.c
 
-libgssdp_1_0_la_LDFLAGS = -version-info $(LTVERSION)
+libgssdp_1_0_la_LDFLAGS = -version-info $(LTVERSION) -no-undefined
 libgssdp_1_0_la_SOURCES = $(introspection_sources)	\
 			  gssdp-client-private.h	\
 			  gssdp-protocol.h		\
@@ -53,7 +53,6 @@ libgssdp_1_0_la_LIBADD = $(LIBGSSDP_LIBS
 if OS_WIN32
 
 libgssdp_1_0_la_LIBADD += -lws2_32 -liphlpapi
-libgssdp_1_0_la_LDFLAGS += -no-undefined
 endif
 
 EXTRA_DIST = gssdp-marshal.list
