--- origsrc/gssdp-0.14.7/libgssdp/Makefile.am	2014-02-04 00:39:18.000000000 +0900
+++ src/gssdp-0.14.7/libgssdp/Makefile.am	2014-03-10 22:13:41.010557400 +0900
@@ -50,10 +50,11 @@ libgssdp_1_0_la_SOURCES = $(introspectio
 
 libgssdp_1_0_la_LIBADD = $(LIBGSSDP_LIBS)
 
+libgssdp_1_0_la_LDFLAGS += -no-undefined
+
 if OS_WIN32
 
 libgssdp_1_0_la_LIBADD += -lws2_32 -liphlpapi
-libgssdp_1_0_la_LDFLAGS += -no-undefined
 endif
 
 if TARGET_ANDROID
