--- origsrc/ETL-0.04.17/m4/subs.m4	2013-10-25 19:54:51.000000000 +0900
+++ src/ETL-0.04.17/m4/subs.m4	2013-11-26 18:45:32.635823300 +0900
@@ -87,8 +87,8 @@ AC_DEFUN([AC_ARG_DEBUG],
 	esac
 
 
-	CXXFLAGS="`echo $CXXFLAGS | sed s:-g::` $debug_flags"
-	CFLAGS="`echo $CFLAGS | sed s:-g::` $debug_flags"
+	CXXFLAGS="`echo $CXXFLAGS | sed 's:-g ::'` $debug_flags"
+	CFLAGS="`echo $CFLAGS | sed 's:-g ::'` $debug_flags"
 
 	AC_MSG_RESULT([$debug_flags])
 ])
@@ -214,7 +214,7 @@ case "$host" in
 dnl    LDFLAGS="$LDFLAGS -lole32 -no-undefined -Wl,--export-all-symbols -Wl,--enable-auto-import -Wl,--subsystem=console -Wl,--enable-runtime-pseudo-reloc"
     ;;
   *cygwin*)
-    LDFLAGS="$LDFLAGS -lole32 -no-undefined -Wl,--export-all-symbols"
+    LDFLAGS="$LDFLAGS -lole32 -Wl,-no-undefined -Wl,--export-all-symbols"
 dnl    LDFLAGS="$LDFLAGS -lole32 -no-undefined -Wl,--export-all-symbols -Wl,--enable-auto-import -Wl,--subsystem=console"
     CXXFLAGS="$CXXFLAGS -I/target/include"
     CFLAGS="$CFLAGS -I/target/include"
