--- origsrc/libease/Makefile.am	2015-04-29 22:25:38.000000000 +0900
+++ src/libease/Makefile.am	2015-04-29 22:28:59.929459100 +0900
@@ -6,3 +6,5 @@ dist_doc_DATA = README
 man_MANS = man/libease.3
 EXTRA_DIST = man/libease.3 src/ease.h
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libease/configure.ac	2015-04-29 22:25:38.000000000 +0900
+++ src/libease/configure.ac	2015-04-29 22:28:46.037841400 +0900
@@ -7,7 +7,7 @@ AC_CONFIG_MACRO_DIR([m4])
 # Set some information about the headers and makefiles.
 AC_CONFIG_SRCDIR([src/ease.h])
 AC_CONFIG_HEADERS([src/config.h])
-AC_CONFIG_FILES([Makefile src/Makefile])
+AC_CONFIG_FILES([libease.pc Makefile src/Makefile])
 
 # Checks for programs.
 AM_INIT_AUTOMAKE
@@ -18,4 +18,3 @@ AC_CHECK_HEADERS([stdio.h stdarg.h stdli
 AC_HEADER_STDC
 
 AC_OUTPUT
-
--- origsrc/libease/libease.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libease/libease.pc.in	2015-04-29 22:28:32.508060000 +0900
@@ -0,0 +1,5 @@
+Name: libease
+Description: Library for graphical effects or mathematical calculations
+Version: @VERSION@
+Libs: -L@libdir@ -lease
+Cflags: -I@includedir@
--- origsrc/libease/src/Makefile.am	2015-04-29 22:25:38.000000000 +0900
+++ src/libease/src/Makefile.am	2015-04-29 22:25:49.477322900 +0900
@@ -4,5 +4,4 @@ AM_FLAGS = -Wall
 lib_LTLIBRARIES = libease.la
 libease_la_SOURCES = libease.c
 include_HEADERS = ease.h
-libease_la_LDFLAGS = -version-info 0:0:0
-
+libease_la_LDFLAGS = -no-undefined -version-info 0:0:0
