--- origsrc/mp3val-0.1.8-src/Makefile.gcc	2007-03-13 01:50:16.000000000 +0900
+++ src/mp3val-0.1.8-src/Makefile.gcc	2015-05-06 14:43:34.691235800 +0900
@@ -1,5 +1,5 @@
 mp3val.exe: main.o mpegparse.o out.o report.o crossapi.o crc.o
-	g++ -Wall main.o mpegparse.o out.o report.o crossapi.o crc.o -o mp3val.exe
+	g++ -Wall main.o mpegparse.o out.o report.o crossapi.o crc.o -o mp3val.exe -O3
 	strip mp3val.exe
 main.o: main.cpp
 	g++ -Wall -c main.cpp
