--- origsrc/mp3val-0.1.8-src/Makefile.gcc	2007-03-13 01:50:16.000000000 +0900
+++ src/mp3val-0.1.8-src/Makefile.gcc	2024-11-08 08:26:44.561293600 +0900
@@ -1,18 +1,20 @@
+CXX ?= g++
+CXXFLAGS ?= -Wall
 mp3val.exe: main.o mpegparse.o out.o report.o crossapi.o crc.o
-	g++ -Wall main.o mpegparse.o out.o report.o crossapi.o crc.o -o mp3val.exe
+	${CXX} ${LDFLAGS} main.o mpegparse.o out.o report.o crossapi.o crc.o -o mp3val.exe
 	strip mp3val.exe
 main.o: main.cpp
-	g++ -Wall -c main.cpp
+	${CXX} ${CPPFLAGS} ${CXXFLAGS} -c main.cpp
 mpegparse.o: mpegparse.cpp
-	g++ -Wall -c mpegparse.cpp
+	${CXX} ${CPPFLAGS} ${CXXFLAGS} -c mpegparse.cpp
 out.o: out.cpp
-	g++ -Wall -c out.cpp
+	${CXX} ${CPPFLAGS} ${CXXFLAGS} -c out.cpp
 report.o: report.cpp
-	g++ -Wall -c report.cpp
+	${CXX} ${CPPFLAGS} ${CXXFLAGS} -c report.cpp
 crossapi.o: crossapi.cpp
-	g++ -Wall -c crossapi.cpp
+	${CXX} ${CPPFLAGS} ${CXXFLAGS} -c crossapi.cpp
 crc.o: crc.cpp crc.h
-	g++ -Wall -c crc.cpp
+	${CXX} ${CPPFLAGS} ${CXXFLAGS} -c crc.cpp
 clean:
 	rm *.o
 	rm *.exe
