--- origsrc/liboglappth-1.0.0/configure.ac	2011-10-13 02:21:59.000000000 +0900
+++ src/liboglappth-1.0.0/configure.ac	2014-03-15 01:53:08.963815100 +0900
@@ -62,12 +62,14 @@ LIBS="$LIBS -L/usr/X11R6/lib"
 ## ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 #AC_CHECK_LIB([GL], [glRenderMode],, AC_MSG_ERROR([GL library missing]))
 #AC_CHECK_LIB([GLU], [gluCylinder],, AC_MSG_ERROR([GLU library missing]))
+PKG_CHECK_MODULES(GL, gl)
+PKG_CHECK_MODULES(GLU, glu)
 
 # Checks for header files.
 # ^^^^^^^^^^^^^^^^^^^^^^^^
 
-AC_CHECK_HEADERS([GL/gl.h])
-AC_CHECK_HEADERS([GL/glu.h])
+#AC_CHECK_HEADERS([GL/gl.h])
+#AC_CHECK_HEADERS([GL/glu.h])
 
 # Checks for typedefs, structures, and compiler characteristics.
 # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
--- origsrc/liboglappth-1.0.0/src/Makefile.am	2011-10-13 02:24:27.000000000 +0900
+++ src/liboglappth-1.0.0/src/Makefile.am	2014-03-15 01:53:08.963815100 +0900
@@ -9,12 +9,12 @@ lib_LTLIBRARIES = liboglappth.la
 ## 5. If any interfaces have been added since the last public release, then increment age.
 ## 6. If any interfaces have been removed since the last public release, then set age to 0.
 
-liboglappth_la_LDFLAGS = -version-info 2:1:0
+liboglappth_la_LDFLAGS = -version-info 2:1:0 -no-undefined
 ##liboglappth_la_LDFLAGS = -version-info 2:1:0		; version 1.0.0
 ##liboglappth_la_LDFLAGS = -version-info 2:0:0		; version 0.98
 ##liboglappth_la_LDFLAGS = -version-info 1:0:0		; version 0.96
 
-##liboglappth_la_LIBADD = -lfoo_extra
+liboglappth_la_LIBADD = $(GL_LIBS) $(GLU_LIBS)
 
 liboglappth_la_SOURCES = \
 	oglv3d.h \
