--- origsrc/liboglappth-0.98/configure.ac	2008-07-31 19:11:55.000000000 +0900
+++ src/liboglappth-0.98/configure.ac	2010-11-27 01:00:45.600684100 +0900
@@ -44,7 +44,7 @@ AC_PROG_CXX
 AC_PROG_CPP
 AC_PROG_CXXCPP
 AC_PROG_INSTALL
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_PATH_PROG(AR, ar, ar)
 AC_PATH_PROG(SHELL, sh, /bin/sh)
 AC_PROG_MAKE_SET
@@ -62,12 +62,14 @@ LIBS="$LIBS -L/usr/X11R6/lib"
 ## ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 #AC_CHECK_LIB(GL,glRenderMode,,[AC_MSG_ERROR([GL library missing])])
 #AC_CHECK_LIB(GLU,gluCylinder,,[AC_MSG_ERROR([GLU library missing])])
+PKG_CHECK_MODULES(GL, gl)
+PKG_CHECK_MODULES(GLU, glu)
 
 # Checks for header files.
 # ^^^^^^^^^^^^^^^^^^^^^^^^
 
-AC_CHECK_HEADERS(GL/gl.h)
-AC_CHECK_HEADERS(GL/glu.h)
+#AC_CHECK_HEADERS(GL/gl.h)
+#AC_CHECK_HEADERS(GL/glu.h)
 
 # Checks for typedefs, structures, and compiler characteristics.
 # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
--- origsrc/liboglappth-0.98/src/Makefile.am	2008-07-31 19:12:59.000000000 +0900
+++ src/liboglappth-0.98/src/Makefile.am	2010-11-27 01:01:22.633935300 +0900
@@ -9,10 +9,11 @@ lib_LTLIBRARIES = liboglappth.la
 ## 5. If any interfaces have been added since the last public release, then increment age.
 ## 6. If any interfaces have been removed since the last public release, then set age to 0.
 
-liboglappth_la_LDFLAGS = -version-info 2:0:0
+liboglappth_la_LDFLAGS = -version-info 2:0:0 -no-undefined
 ##liboglappth_la_LDFLAGS = -version-info 2:0:0		; was 0.98
 ##liboglappth_la_LDFLAGS = -version-info 1:0:0		; was 0.96
 
+liboglappth_la_LIBADD = @GL_LIBS@ @GLU_LIBS@
 ##liboglappth_la_LIBADD = -lfoo_extra
 
 liboglappth_la_SOURCES = \
