--- origsrc/graphite2-1.0.3/gr2fonttest/CMakeLists.txt	2011-09-23 00:30:02.000000000 +0900
+++ src/graphite2-1.0.3/gr2fonttest/CMakeLists.txt	2011-10-15 14:53:25.068449200 +0900
@@ -31,8 +31,12 @@ if (NOT DISABLE_TRACING)
 endif (NOT DISABLE_TRACING)
 
 add_executable(gr2fonttest gr2FontTest.cpp)
-
 target_link_libraries(gr2fonttest graphite2)
+if (NOT ${ICONV_HEADER_PATH} STREQUAL "ICONV_HEADER_PATH-NOTFOUND")
+	find_library(LIBICONV iconv)
+	target_link_libraries(gr2fonttest ${LIBICONV})
+endif (NOT ${ICONV_HEADER_PATH} STREQUAL "ICONV_HEADER_PATH-NOTFOUND")
+
 add_dependencies(gr2fonttest graphite2)
 
 if (${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
@@ -53,7 +57,7 @@ endif  (${CMAKE_SYSTEM_NAME} STREQUAL "L
 
 # copy the DLL so that gr2fonttest can find it
 add_custom_target(copy_dll ALL 
-	COMMAND ${CMAKE_COMMAND} -E copy_if_different ${graphite2_core_BINARY_DIR}/${CMAKE_CFG_INTDIR}/${CMAKE_SHARED_LIBRARY_PREFIX}graphite2${CMAKE_SHARED_LIBRARY_SUFFIX} ${PROJECT_BINARY_DIR}/${CMAKE_CFG_INTDIR})
+	COMMAND ${CMAKE_COMMAND} -E copy_if_different ${graphite2_core_BINARY_DIR}/${CMAKE_CFG_INTDIR}/${CMAKE_SHARED_LIBRARY_PREFIX}graphite2*${CMAKE_SHARED_LIBRARY_SUFFIX} ${PROJECT_BINARY_DIR}/${CMAKE_CFG_INTDIR})
 
 if (${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
     if (NOT ${WIN_ICONV_PATH} STREQUAL "WIN_ICONV_PATH-NOTFOUND")
--- origsrc/graphite2-1.0.3/src/Face.cpp	2011-09-23 00:30:02.000000000 +0900
+++ src/graphite2-1.0.3/src/Face.cpp	2011-10-15 14:53:25.068449200 +0900
@@ -27,7 +27,7 @@ of the License or (at your option) any l
 #include <cstring>
 #include "graphite2/Segment.h"
 #include "Face.h"
-#include "Endian.h"
+#include "Endian_.h"
 #include "Segment.h"
 #include "CmapCache.h"
 #include "NameTable.h"
--- origsrc/graphite2-1.0.3/src/FeatureMap.cpp	2011-09-23 00:30:02.000000000 +0900
+++ src/graphite2-1.0.3/src/FeatureMap.cpp	2011-10-15 14:53:25.078463600 +0900
@@ -27,7 +27,7 @@ of the License or (at your option) any l
 #include <string.h>
 
 #include "Main.h"
-#include "Endian.h"
+#include "Endian_.h"
 #include "FeatureMap.h"
 #include "FeatureVal.h"
 #include "graphite2/Font.h"
--- origsrc/graphite2-1.0.3/src/GlyphFace.cpp	2011-09-23 00:30:02.000000000 +0900
+++ src/graphite2-1.0.3/src/GlyphFace.cpp	2011-10-15 14:53:25.078463600 +0900
@@ -29,7 +29,7 @@ of the License or (at your option) any l
 #include "XmlTraceLog.h"
 #include "GlyphFaceCache.h"
 #include "TtfUtil.h"
-#include "Endian.h"
+#include "Endian_.h"
 
 
 using namespace graphite2;
--- origsrc/graphite2-1.0.3/src/GlyphFaceCache.cpp	2011-09-23 00:30:02.000000000 +0900
+++ src/graphite2-1.0.3/src/GlyphFaceCache.cpp	2011-10-15 14:53:25.098492400 +0900
@@ -27,7 +27,7 @@ of the License or (at your option) any l
 #include "GlyphFaceCache.h"
 #include "graphite2/Font.h"
 #include "Face.h"     //for the tags
-#include "Endian.h"
+#include "Endian_.h"
 
 using namespace graphite2;
 
--- origsrc/graphite2-1.0.3/src/NameTable.cpp	2011-09-23 00:30:02.000000000 +0900
+++ src/graphite2-1.0.3/src/NameTable.cpp	2011-10-15 14:53:25.108506800 +0900
@@ -25,7 +25,7 @@ License, as published by the Free Softwa
 of the License or (at your option) any later version.
 */
 #include "Main.h"
-#include "Endian.h"
+#include "Endian_.h"
 
 #include "NameTable.h"
 #include "UtfCodec.h"
--- origsrc/graphite2-1.0.3/src/Pass.cpp	2011-09-23 00:30:02.000000000 +0900
+++ src/graphite2-1.0.3/src/Pass.cpp	2011-10-15 14:53:25.108506800 +0900
@@ -25,7 +25,7 @@ License, as published by the Free Softwa
 of the License or (at your option) any later version.
 */
 #include "Main.h"
-#include "Endian.h"
+#include "Endian_.h"
 #include "Pass.h"
 #include <string.h>
 #include <stdlib.h>
--- origsrc/graphite2-1.0.3/src/Silf.cpp	2011-09-23 00:30:02.000000000 +0900
+++ src/graphite2-1.0.3/src/Silf.cpp	2011-10-15 14:53:25.128535600 +0900
@@ -26,7 +26,7 @@ of the License or (at your option) any l
 */
 #include <cstdlib>
 #include "graphite2/Segment.h"
-#include "Endian.h"
+#include "Endian_.h"
 #include "Silf.h"
 #include "XmlTraceLog.h"
 #include "Segment.h"
--- origsrc/graphite2-1.0.3/src/TtfUtil.cpp	2011-09-23 00:30:02.000000000 +0900
+++ src/graphite2-1.0.3/src/TtfUtil.cpp	2011-10-15 14:53:25.138550000 +0900
@@ -51,7 +51,7 @@ Description
 // Module headers
 #include "TtfUtil.h"
 #include "TtfTypes.h"
-#include "Endian.h"
+#include "Endian_.h"
 
 /***********************************************************************************************
 	Forward declarations
--- origsrc/graphite2-1.0.3/tests/endian/endiantest.cpp	2011-09-23 00:30:02.000000000 +0900
+++ src/graphite2-1.0.3/tests/endian/endiantest.cpp	2011-10-15 14:53:25.158578800 +0900
@@ -33,7 +33,7 @@ integers in addition to 32, 16 and 8 bit
 -----------------------------------------------------------------------------*/
 #include <cstdlib>
 #include <string>
-#include "Endian.h"
+#include "Endian_.h"
 
 typedef unsigned char 		uint8;
 typedef signed char		     int8;
--- origsrc/graphite2-1.0.3/tests/featuremap/featuremaptest.cpp	2011-09-23 00:30:02.000000000 +0900
+++ src/graphite2-1.0.3/tests/featuremap/featuremaptest.cpp	2011-10-15 14:53:25.168593200 +0900
@@ -24,7 +24,7 @@
 #include <cstdlib>
 #include <cstring>
 #include "Main.h"
-#include "Endian.h"
+#include "Endian_.h"
 #include "Face.h"
 #include "FeatureMap.h"
 #include "TtfTypes.h"
--- origsrc/graphite2-1.0.3/tests/nametabletest/nametabletest.cpp	2011-09-23 00:30:02.000000000 +0900
+++ src/graphite2-1.0.3/tests/nametabletest/nametabletest.cpp	2011-10-15 14:53:25.168593200 +0900
@@ -25,7 +25,7 @@
 #include <cstring>
 #include <graphite2/Types.h>
 #include "Main.h"
-#include "Endian.h"
+#include "Endian_.h"
 #include "TtfTypes.h"
 #include "NameTable.h"
 
--- origsrc/graphite2-1.0.3/tests/segcache/CMakeLists.txt	2011-09-23 00:30:02.000000000 +0900
+++ src/graphite2-1.0.3/tests/segcache/CMakeLists.txt	2011-10-15 14:53:25.198636400 +0900
@@ -8,8 +8,8 @@ if  (${CMAKE_SYSTEM_NAME} STREQUAL "Wind
 endif (${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
 
 
-target_objects(OBJS graphite2)
-add_executable(grsegcachetest segcachetest.cpp ${OBJS})
+add_executable(grsegcachetest segcachetest.cpp)
 add_dependencies(grsegcachetest graphite2)
+target_link_libraries(grsegcachetest graphite2)
 
 add_test(NAME grsegcachetest COMMAND $<TARGET_FILE:grsegcachetest> ${PROJECT_SOURCE_DIR}/../fonts/Padauk.ttf)
--- origsrc/graphite2-1.0.3/tests/utftest/CMakeLists.txt	2011-09-23 00:30:02.000000000 +0900
+++ src/graphite2-1.0.3/tests/utftest/CMakeLists.txt	2011-10-15 14:53:25.198636400 +0900
@@ -8,8 +8,8 @@ if  (${CMAKE_SYSTEM_NAME} STREQUAL "Wind
 endif (${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
 
 
-target_objects(OBJS graphite2)
-add_executable(utftest utftest.cpp ${OBJS})
+add_executable(utftest utftest.cpp)
 add_dependencies(utftest graphite2)
+target_link_libraries(utftest graphite2)
 
 add_test(NAME utftest COMMAND $<TARGET_FILE:utftest>)
--- origsrc/graphite2-1.0.3/tests/vm/CMakeLists.txt	2011-09-23 00:30:02.000000000 +0900
+++ src/graphite2-1.0.3/tests/vm/CMakeLists.txt	2011-10-15 14:53:25.208650800 +0900
@@ -35,7 +35,7 @@ target_objects(OBJS graphite2
     TtfUtil.cpp
     UtfCodec.cpp
     ${TRACESUPPORT})
-add_library(vm-test-common STATIC basic_test.cpp ${OBJS})
+add_library(vm-test-common basic_test.cpp ${OBJS})
 add_dependencies(vm-test-common graphite2)
 
 if  (${CMAKE_COMPILER_IS_GNUCXX})
@@ -44,11 +44,11 @@ if  (${CMAKE_COMPILER_IS_GNUCXX})
 endif  (${CMAKE_COMPILER_IS_GNUCXX})
 
 add_executable(vm-test-call ${graphite2_core_SOURCE_DIR}/call_machine.cpp)
-target_link_libraries(vm-test-call vm-test-common)
+target_link_libraries(vm-test-call vm-test-common graphite2)
 
 if  (${CMAKE_COMPILER_IS_GNUCXX})
 	add_executable(vm-test-direct ${graphite2_core_SOURCE_DIR}/direct_machine.cpp)
-	target_link_libraries(vm-test-direct vm-test-common)
+	target_link_libraries(vm-test-direct vm-test-common graphite2)
 endif  (${CMAKE_COMPILER_IS_GNUCXX})
 
 if  (${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
