--- origsrc/gtk-engines-cleanice-2.4.1/configure.ac	2007-08-09 04:16:50.000000000 +0900
+++ src/gtk-engines-cleanice-2.4.1/configure.ac	2010-12-30 16:37:20.829310100 +0900
@@ -8,14 +8,10 @@ AC_COPYRIGHT([Copyright 2002-2007 Elysiu
 AM_INIT_AUTOMAKE
 AM_CONFIG_HEADER(config.h)
 
-AM_MAINTAINER_MODE
-AC_DISABLE_STATIC
-
 AC_ISC_POSIX
 AC_PROG_CC
 AC_STDC_HEADERS
-AC_LIBTOOL_WIN32_DLL
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 PKG_CHECK_MODULES(CLEANICE, gtk+-2.0 >= 2.4.0 gobject-2.0 >= 2.0,,
 	 AC_MSG_ERROR([GTK+ >= 2.4 is required]))
