--- origsrc/soci-3.2.3/CMakeLists.txt	2015-04-09 07:39:00.000000000 +0900
+++ src/soci-3.2.3/CMakeLists.txt	2015-04-17 17:06:48.820164300 +0900
@@ -73,7 +73,7 @@ include(SociDependencies)
 # Installation
 ###############################################################################
 
-if(APPLE OR CMAKE_SIZEOF_VOID_P EQUAL 4)
+if(APPLE OR CYGWIN OR CMAKE_SIZEOF_VOID_P EQUAL 4)
   set(SOCI_LIBDIR "lib")
 else()
   set(SOCI_LIBDIR "lib64")
--- origsrc/soci-3.2.3/cmake/SociConfig.cmake	2015-04-09 05:32:00.000000000 +0900
+++ src/soci-3.2.3/cmake/SociConfig.cmake	2015-04-17 17:09:22.909092400 +0900
@@ -32,10 +32,12 @@ else()
 	"-pedantic -ansi -Wall -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Wredundant-decls -Wno-long-long")
 
   if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
-        
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC ${SOCI_GCC_CLANG_COMMON_FLAGS}")
+    if (NOT CYGWIN)
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
+    endif()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${SOCI_GCC_CLANG_COMMON_FLAGS}")
     if (CMAKE_COMPILER_IS_GNUCXX)
-        if (CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+        if (CMAKE_SYSTEM_NAME MATCHES "FreeBSD" OR CYGWIN)
             set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++98")
         else()
             set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++98")
