--- origsrc/soci-4.0.1/cmake/SociBackend.cmake	2020-10-19 23:58:52.000000000 +0900
+++ src/soci-4.0.1/cmake/SociBackend.cmake	2020-10-27 18:30:14.152222700 +0900
@@ -229,7 +229,11 @@ macro(soci_backend NAME)
          )
       endif()
 
-      install(EXPORT SOCI NAMESPACE SOCI:: DESTINATION cmake)
+      if (CYGWIN)
+        install(EXPORT SOCI NAMESPACE SOCI:: DESTINATION lib/cmake/soci)
+      else()
+        install(EXPORT SOCI NAMESPACE SOCI:: DESTINATION cmake)
+      endif()
 
     else()
         colormsg(HIRED "${NAME}" RED "backend disabled, since")
