--- origsrc/ilbc-rfc3951/configure.ac	2005-05-11 23:19:58.000000000 +0900
+++ src/ilbc-rfc3951/configure.ac	2011-04-16 10:55:05.026430900 +0900
@@ -10,9 +10,7 @@ AM_INIT_AUTOMAKE($PACKAGE_NAME,$PACKAGE_
 
 # Checks for programs.
 AC_PROG_CC
-AC_PROG_LIBTOOL
-AC_ENABLE_SHARED(yes)
-AC_ENABLE_STATIC(no)
+LT_INIT([win32-dll disable-static])
 
 # Checks for libraries.
 
--- origsrc/ilbc-rfc3951/src/Makefile.am	2005-05-11 23:18:38.000000000 +0900
+++ src/ilbc-rfc3951/src/Makefile.am	2011-04-16 10:55:24.274107700 +0900
@@ -10,6 +10,8 @@ createCB.h   FrameClassify.h  hpInput.h
 doCPLC.c     gainquant.c      hpOutput.c  iLBC_define.h   lsf.c          syntFilter.c \
 doCPLC.h     gainquant.h      hpOutput.h  iLBC_encode.c   lsf.h          syntFilter.h 
 
+libilbc_la_LDFLAGS = -no-undefined
+
 ilbc_includedir=$(includedir)/ilbc
 ilbc_include_HEADERS=iLBC_decode.h iLBC_encode.h iLBC_define.h
 
