--- origsrc/jsoncpp-src-0.6.0-rc2/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/jsoncpp-src-0.6.0-rc2/Makefile.am	2014-03-09 10:47:08.933138300 +0900
@@ -0,0 +1,22 @@
+
+check_PROGRAMS = test_lib_json
+test_lib_json_CPPFLAGS = \
+-I$(top_srcdir)/include -I$(top_srcdir)/src/test_lib_json
+test_lib_json_LDADD = libjsoncpp.la
+test_lib_json_SOURCES = \
+src/test_lib_json/jsontest.cpp  src/test_lib_json/main.cpp
+
+TESTS = $(check_PROGRAMS)
+
+pkginclude_HEADERS = \
+include/json/autolink.h  include/json/config.h  include/json/features.h \
+include/json/forwards.h  include/json/json.h    include/json/reader.h \
+include/json/value.h     include/json/writer.h
+
+lib_LTLIBRARIES = libjsoncpp.la
+libjsoncpp_la_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/src/lib_json
+libjsoncpp_la_LDFLAGS = -no-undefined
+libjsoncpp_la_SOURCES = \
+src/lib_json/json_reader.cpp  src/lib_json/json_value.cpp \
+src/lib_json/json_writer.cpp
+
--- origsrc/jsoncpp-src-0.6.0-rc2/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/jsoncpp-src-0.6.0-rc2/configure.ac	2014-03-09 10:39:20.941357000 +0900
@@ -0,0 +1,30 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([jsoncpp], [0.6.0-rc2])
+AC_CONFIG_SRCDIR([include/json/autolink.h])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE([subdir-objects])
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+AC_C_INLINE
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([strchr strpbrk])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
