--- origsrc/libsmdev-20200210/Makefile.am	2020-02-11 00:13:15.000000000 +0900
+++ src/libsmdev-20200210/Makefile.am	2020-09-06 08:34:18.922979800 +0900
@@ -3,19 +3,11 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libuna \
-	libcfile \
 	libsmdev \
 	smdevtools \
 	pysmdev \
 	pysmdev-python2 \
 	pysmdev-python3 \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libsmdev-20200210/configure.ac	2020-02-11 00:13:15.000000000 +0900
+++ src/libsmdev-20200210/configure.ac	2020-09-06 08:34:19.723034100 +0900
@@ -158,13 +158,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
 AC_CONFIG_FILES([libsmdev/Makefile])
 AC_CONFIG_FILES([pysmdev/Makefile])
 AC_CONFIG_FILES([pysmdev-python2/Makefile])
--- origsrc/libsmdev-20200210/libsmdev/Makefile.am	2020-02-11 00:13:15.000000000 +0900
+++ src/libsmdev-20200210/libsmdev/Makefile.am	2020-09-06 08:34:19.818976200 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libsmdev-20200210/pysmdev/Makefile.am	2020-02-11 00:13:15.000000000 +0900
+++ src/libsmdev-20200210/pysmdev/Makefile.am	2020-09-06 08:34:20.185985500 +0900
@@ -1,5 +1,6 @@
 if HAVE_PYTHON
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -32,7 +33,7 @@ pysmdev_la_LIBADD = \
 	@LIBCFILE_LIBADD@
 
 pysmdev_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pysmdev_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pysmdev_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
--- origsrc/libsmdev-20200210/pysmdev-python2/Makefile.am	2020-02-11 00:13:15.000000000 +0900
+++ src/libsmdev-20200210/pysmdev-python2/Makefile.am	2020-09-06 08:34:20.296032400 +0900
@@ -6,6 +6,7 @@ pyexecdir = $(pyexecdir2)
 
 if HAVE_PYTHON2
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -43,7 +44,7 @@ pysmdev_la_LIBADD = \
 	@LIBCFILE_LIBADD@
 
 pysmdev_la_CPPFLAGS = $(PYTHON2_CPPFLAGS)
-pysmdev_la_LDFLAGS  = -module -avoid-version $(PYTHON2_LDFLAGS)
+pysmdev_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON2_LDFLAGS)
 
 $(BUILT_SOURCES):
 	/bin/cp -f $(top_srcdir)/pysmdev/$@ $@
--- origsrc/libsmdev-20200210/pysmdev-python3/Makefile.am	2020-02-11 00:13:15.000000000 +0900
+++ src/libsmdev-20200210/pysmdev-python3/Makefile.am	2020-09-06 08:34:20.930092600 +0900
@@ -6,6 +6,7 @@ pyexecdir = $(pyexecdir3)
 
 if HAVE_PYTHON3
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
@@ -43,7 +44,7 @@ pysmdev_la_LIBADD = \
 	@LIBCFILE_LIBADD@
 
 pysmdev_la_CPPFLAGS = $(PYTHON3_CPPFLAGS)
-pysmdev_la_LDFLAGS  = -module -avoid-version $(PYTHON3_LDFLAGS)
+pysmdev_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON3_LDFLAGS)
 
 $(BUILT_SOURCES):
 	/bin/cp -f $(top_srcdir)/pysmdev/$@ $@
--- origsrc/libsmdev-20200210/smdevtools/Makefile.am	2020-02-11 00:13:15.000000000 +0900
+++ src/libsmdev-20200210/smdevtools/Makefile.am	2020-09-06 08:34:21.875021900 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libsmdev-20200210/tests/Makefile.am	2020-02-11 00:13:15.000000000 +0900
+++ src/libsmdev-20200210/tests/Makefile.am	2020-09-06 08:34:22.329043400 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
