--- origsrc/libevocosm-3.3.1/Makefile.am	2008-04-01 23:29:52.000000000 +0900
+++ src/libevocosm-3.3.1/Makefile.am	2010-11-30 10:11:59.678408000 +0900
@@ -4,7 +4,10 @@ else
 DOC_DIR =
 endif
 
-SUBDIRS = libevocosm funcopt $(DOC_DIR)
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = libevocosm.pc
+
+SUBDIRS = libevocosm funcopt #$(DOC_DIR)
 
 EXTRA_DIST = LICENSE.GPL LICENSE.POLICY LICENSE.NON-FREE reconf
 
--- origsrc/libevocosm-3.3.1/configure.ac	2008-07-17 03:23:39.000000000 +0900
+++ src/libevocosm-3.3.1/configure.ac	2010-11-30 00:51:45.573268200 +0900
@@ -55,7 +55,7 @@ dnl ------------------------------------
 
 AC_PROG_CC
 AC_PROG_CXX
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AM_SANITY_CHECK
 AC_LANG_CPLUSPLUS
 
--- origsrc/libevocosm-3.3.1/libevocosm/Makefile.am	2008-04-01 23:29:52.000000000 +0900
+++ src/libevocosm-3.3.1/libevocosm/Makefile.am	2010-11-30 08:58:04.633329900 +0900
@@ -12,7 +12,7 @@ cpp_sources = evocommon.cpp evoreal.cpp
 lib_LTLIBRARIES = libevocosm.la
 
 libevocosm_la_SOURCES = $(h_sources) $(cpp_sources)
-libevocosm_la_LDFLAGS= -version-info $(GENERIC_LIBRARY_VERSION) -release $(GENERIC_RELEASE)
+libevocosm_la_LDFLAGS= -version-info $(GENERIC_LIBRARY_VERSION) -release $(GENERIC_RELEASE) -no-undefined
 
 library_includedir=$(includedir)/$(GENERIC_LIBRARY_NAME)
 library_include_HEADERS = $(h_sources)
--- origsrc/libevocosm-3.3.1/libevocosm.pc.in	2008-04-01 23:29:52.000000000 +0900
+++ src/libevocosm-3.3.1/libevocosm.pc.in	2010-11-30 10:10:21.947878400 +0900
@@ -8,6 +8,6 @@ Description: A framew3ork for evolutiona
 Author: Scott Robert Ladd
 URL: http://www.coyotegulch.com/libevocosm
 Version: @VERSION@
-Libs: -L${libdir} -l@GENERIC_LIBRARY_NAME
+Libs: -L${libdir} -l@GENERIC_LIBRARY_NAME@
 Cflags: -I${includedir}/@GENERIC_LIBRARY_NAME@ -I${libdir}/@GENERIC_LIBRARY_NAME@/include
 
