--- origsrc/libg722-v_1_0_2/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libg722-v_1_0_2/Makefile.am	2019-07-03 07:58:16.152140800 +0900
@@ -0,0 +1,13 @@
+
+check_PROGRAMS = g722-test
+g722_test_SOURCES = test.c
+g722_test_LDADD = libg722.la
+
+include_HEADERS = g722.h g722_private.h g722_encoder.h g722_decoder.h
+
+lib_LTLIBRARIES = libg722.la
+libg722_la_SOURCES = g722_decode.c g722_encode.c
+libg722_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libg722-v_1_0_2/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/libg722-v_1_0_2/configure.ac	2019-07-03 07:57:09.874960400 +0900
@@ -0,0 +1,32 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([libg722], [1.0.2])
+AC_CONFIG_SRCDIR([g722.h])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_LN_S
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([inttypes.h memory.h stdint.h stdlib.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_INT16_T
+AC_TYPE_INT32_T
+AC_TYPE_SIZE_T
+AC_TYPE_UINT8_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([memset])
+
+AC_CONFIG_FILES([libg722.pc Makefile])
+AC_OUTPUT
--- origsrc/libg722-v_1_0_2/libg722.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libg722-v_1_0_2/libg722.pc.in	2019-07-03 07:58:25.959918800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libg722
+Description: Implementation of the ITU G.722 specification
+Version: @VERSION@
+Libs: -L${libdir} -lg722
+Cflags: -I${includedir}
--- origsrc/libg722-v_1_0_2/test.c	2016-07-30 09:47:53.000000000 +0900
+++ src/libg722-v_1_0_2/test.c	2019-07-03 08:02:15.568615600 +0900
@@ -25,7 +25,7 @@
  *
  */
 
-#include <sys/endian.h>
+#include <endian.h>
 #include <getopt.h>
 #include <stdint.h>
 #include <stdio.h>
