--- origsrc/harfbuzz-0.9.4/configure.ac	2012-09-05 04:28:32.000000000 +0900
+++ src/harfbuzz-0.9.4/configure.ac	2012-09-23 19:57:01.341656000 +0900
@@ -53,7 +53,11 @@ dnl GOBJECT_INTROSPECTION_CHECK([0.9.0])
 dnl GTK_DOC_CHECK([1.15],[--flavour no-tmpl])
 
 # Functions and headers
-AC_CHECK_FUNCS(atexit mprotect sysconf getpagesize sched_yield mmap _setmode isatty)
+AC_CHECK_FUNCS(atexit mprotect sysconf getpagesize sched_yield mmap isatty)
+case "$host_os" in
+	cygwin*) ;;
+	*) AC_CHECK_FUNCS(_setmode);;
+esac
 AC_CHECK_HEADERS(unistd.h sys/mman.h sched.h io.h)
 
 # Compiler flags
