--- origsrc/harfbuzz/configure.ac	2010-12-24 17:06:19.000000000 +0900
+++ src/harfbuzz/configure.ac	2010-12-24 17:06:50.371571500 +0900
@@ -5,8 +5,7 @@ AC_CONFIG_HEADERS([config.h])
 AM_INIT_AUTOMAKE([1.9.6 gnu dist-bzip2 no-dist-gzip -Wall no-define])
 m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
 
-AC_LIBTOOL_WIN32_DLL
-AC_PROG_LIBTOOL dnl ([1.4]) Don't remove!
+LT_INIT(win32-dll)
 
 AC_PROG_CC
 AM_PROG_CC_C_O
--- origsrc/harfbuzz/src/Makefile.am	2010-12-24 17:06:19.000000000 +0900
+++ src/harfbuzz/src/Makefile.am	2010-12-24 17:07:29.467789100 +0900
@@ -111,6 +111,7 @@ CXXLINK = $(LINK)
 libharfbuzz_la_SOURCES = $(HBSOURCES) $(HBHEADERS)
 libharfbuzz_la_CPPFLAGS = $(HBCFLAGS)
 libharfbuzz_la_LIBADD = $(HBLIBS)
+libharfbuzz_la_LDFLAGS = -no-undefined
 pkginclude_HEADERS = $(HBHEADERS)
 
 
