--- origsrc/evcon/configure.ac	2025-08-08 11:54:14.000000000 +0900
+++ src/evcon/configure.ac	2025-08-08 11:55:49.606285200 +0900
@@ -9,7 +9,7 @@ AC_CONFIG_HEADERS([src/core/evcon-config
 
 AC_CONFIG_MACRO_DIR([m4])
 
-AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+AM_INIT_AUTOMAKE([-Wall foreign])
 
 m4_ifndef([PKG_PROG_PKG_CONFIG], [m4_fatal([pkg-config not installed])])
 m4_ifndef([AC_PROG_LIBTOOL], [m4_fatal([libtool not installed])])
--- origsrc/evcon/src/tests/Makefile.am	2025-08-08 11:54:14.000000000 +0900
+++ src/evcon/src/tests/Makefile.am	2025-08-08 12:05:19.789342600 +0900
@@ -30,4 +30,4 @@ EXTRA_DIST = evcon-echo.h
 check_PROGRAMS=$(test_binaries)
 
 TESTS=$(test_binaries)
-TESTS_ENVIRONMENT=gtester --verbose
+# TESTS_ENVIRONMENT=gtester --verbose
