--- origsrc/accounts-glib/libaccounts-glib/Makefile.am	2011-08-16 09:10:27.000000000 +0900
+++ src/accounts-glib/libaccounts-glib/Makefile.am	2011-08-16 09:12:17.624179000 +0900
@@ -4,7 +4,7 @@ lib_LTLIBRARIES = \
 libaccounts_glib_la_CFLAGS = \
 	$(LIBACCOUNTS_CFLAGS) $(AEGISCRYPTO_CFLAGS) -Wall -Werror
 libaccounts_glib_la_LIBADD = $(LIBACCOUNTS_LIBS) $(AEGISCRYPTO_LIBS) -lrt
-libaccounts_glib_la_LDFLAGS = \
+libaccounts_glib_la_LDFLAGS = -no-undefined \
 	-version-info 1:3:1 \
 	-export-symbols-regex '^ag_'
 
@@ -12,6 +12,8 @@ BUILT_SOURCES = \
 	ag-marshal.c \
 	ag-marshal.h
 
+INCLUDES = -I$(top_srcdir)
+
 CLEANFILES = $(BUILT_SOURCES)
 
 libaccounts_glib_la_SOURCES = \
