--- origsrc/sufary-2.3.8/Makefile.am	2002-01-23 09:29:14.000000000 +0900
+++ src/sufary-2.3.8/Makefile.am	2015-03-23 00:18:37.986435900 +0900
@@ -14,17 +14,6 @@ EXTRA_DIRS = doc misc kwicview perl ruby
 
 pkgdatadir = @datadir@/@PACKAGE@
 
-install-data-local:
-	$(mkinstalldirs) $(pkgdatadir)/etc
-	for subdir in $(EXTRA_DIRS); do \
-	    cp -rp $(srcdir)/$$subdir $(pkgdatadir); \
-	    rm -rf $(pkgdatadir)/$$subdir/CVS; \
-	    rm -rf $(pkgdatadir)/$$subdir/*/CVS; \
-	    rm -f $(pkgdatadir)/$$subdir/Makefile*; \
-	done
-	cp ChangeLog* COPYING NEWS README README-ja TODO \
-		$(pkgdatadir)/etc
-
 dist-hook:
 	for subdir in $(EXTRA_DIRS); do \
 	    cp -rp $$subdir $(distdir)/$$subdir; \
@@ -34,5 +23,5 @@ dist-hook:
 	    rm -rf $(distdir)/$$subdir/*/CVS; \
 	done
 
-##Bug in automake: Can't use `if MAINTAINER_MODE'
-@MAINTAINER_MODE_TRUE@include $(srcdir)/maintMakefile
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = sufary.pc
--- origsrc/sufary-2.3.8/configure.in	2002-01-21 12:53:00.000000000 +0900
+++ src/sufary-2.3.8/configure.in	2015-03-23 00:18:50.990195500 +0900
@@ -6,9 +6,9 @@ dnl Created  by Satoru Takabayashi <sato
 dnl
 
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(src/mkary.c)
+AC_INIT([sufary], [2.3.8])
 AC_PREREQ(2.13)               dnl Minimum Autoconf version required.
-AM_INIT_AUTOMAKE(sufary, 2.3.8)
+AM_INIT_AUTOMAKE
 AM_MAINTAINER_MODE
 
 dnl *
@@ -66,7 +66,7 @@ dnl * Output Files.
 dnl * 
 AM_CONFIG_HEADER(config.h)
 
-AC_OUTPUT([Makefile src/Makefile lib/Makefile man/Makefile src/mkdids], 
+AC_OUTPUT([Makefile src/Makefile lib/Makefile man/Makefile src/mkdids sufary.pc], 
   [chmod +x src/mkdids]) 
 
 # Local Variables:
--- origsrc/sufary-2.3.8/lib/Makefile.am	2000-11-28 16:56:42.000000000 +0900
+++ src/sufary-2.3.8/lib/Makefile.am	2015-03-23 00:12:47.529131600 +0900
@@ -16,4 +16,4 @@ noinst_PROGRAMS = test
 test_SOURCES = test.c
 test_LDADD   = libsufary.la
 
-libsufary_la_LDFLAGS = -version-info 0:0:0
+libsufary_la_LDFLAGS = -no-undefined -version-info 0:0:0
--- origsrc/sufary-2.3.8/src/Makefile.am	1999-12-05 08:42:06.000000000 +0900
+++ src/sufary-2.3.8/src/Makefile.am	2015-03-23 00:12:47.529131600 +0900
@@ -1,25 +1,25 @@
 # $Id: Makefile.am,v 1.7 1999/12/04 23:42:06 satoru-t Exp $ 
 AUTOMAKE_OPTIONS = no-dependencies
-INCLUDES = -I../lib
+INCLUDES = -I$(top_srcdir)/lib
 bin_PROGRAMS = sass af sang mkdid mkary # array
 bin_SCRIPTS  = mkdids
 
 sass_SOURCES = sass.c getopt.c getopt1.c getopt.h
-sass_LDADD   = $(top_srcdir)/lib/libsufary.la
+sass_LDADD   = $(top_builddir)/lib/libsufary.la
 
 af_SOURCES   = af.c
-af_LDADD     = $(top_srcdir)/lib/libsufary.la
+af_LDADD     = $(top_builddir)/lib/libsufary.la
 
 sang_SOURCES = sang.c
-sang_LDADD   = $(top_srcdir)/lib/libsufary.la
+sang_LDADD   = $(top_builddir)/lib/libsufary.la
 
 mkdid_SOURCES = mkdid.c
-mkdid_LDADD   = $(top_srcdir)/lib/libsufary.la
+mkdid_LDADD   = $(top_builddir)/lib/libsufary.la
 
 mkary_SOURCES = mkary.c
-mkary_LDADD   = $(top_srcdir)/lib/libsufary.la
+mkary_LDADD   = $(top_builddir)/lib/libsufary.la
 
 # array_SOURCES = array-main.c array-misc.c array-show.c array.h
-# array_LDADD   = $(top_srcdir)/lib/libsufary.la
+# array_LDADD   = $(top_builddir)/lib/libsufary.la
 # array_LIBS    = -lm
 
--- origsrc/sufary-2.3.8/sufary.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/sufary-2.3.8/sufary.pc.in	2015-03-23 00:18:20.070242700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: sufary
+Description: Suffix Array Library
+Version: @VERSION@
+Libs: -L${libdir} -lsufary
+Cflags: -I${includedir}
