--- origsrc/CGAL-4.5.2/CMakeLists.txt	2014-08-29 22:58:16.000000000 +0900
+++ src/CGAL-4.5.2/CMakeLists.txt	2015-02-28 12:38:50.297939100 +0900
@@ -725,11 +725,11 @@ set ( CGAL_INSTALL_BIN_DIR    "bin"
   CACHE STRING "The folder where CGAL user-side scripts will be installed, relative to CMAKE_INSTALL_PREFIX" 
   )
 
-set ( CGAL_INSTALL_CMAKE_DIR "${CGAL_INSTALL_LIB_DIR}/CGAL"
+set ( CGAL_INSTALL_CMAKE_DIR "${CGAL_INSTALL_LIB_DIR}/cmake/CGAL"
   CACHE STRING "The folder where CGAL CMake modules will be installed, relative to CMAKE_INSTALL_PREFIX" 
   )
 
-set ( CGAL_INSTALL_DOC_DIR "share/doc/${CGAL_VERSION_DIR}"  
+set ( CGAL_INSTALL_DOC_DIR "share/doc/CGAL"  
   CACHE STRING "The folder where CGAL documentation and license files will be installed, relative to CMAKE_INSTALL_PREFIX" 
   )
 
