--- origsrc/libphidget-2.1.8.20130320/Makefile.am	2013-03-21 05:32:01.000000000 +0900
+++ src/libphidget-2.1.8.20130320/Makefile.am	2013-06-09 11:10:08.937618900 +0900
@@ -17,7 +17,7 @@ lib_LTLIBRARIES = libphidget21.la
 #    if linking against the old version at runtime: set revision to 0, bump current and age.
 # 3. Programs may need to be changed, recompiled, relinked in order to use the new version. 
 #    Bump current, set revision and age to 0.
-libphidget21_la_LDFLAGS = -version-info 0:0:0
+libphidget21_la_LDFLAGS = -no-undefined -version-info 0:0:0
 
 libphidget21_la_SOURCES = cphidget.c \
 	cphidget.h \
--- origsrc/libphidget-2.1.8.20130320/utils/utils.c	2013-03-21 05:32:01.000000000 +0900
+++ src/libphidget-2.1.8.20130320/utils/utils.c	2013-06-09 11:13:06.397579900 +0900
@@ -15,7 +15,7 @@
 #else
 #include "wincompat.h"
 #endif
-#if defined(__CYGWIN__) || defined(__MINGW32_VERSION)
+#if defined(__MINGW32_VERSION)
 #include "getaddrinfo.h"
 #endif
 #ifndef _MSC_EXTENSIONS
