--- origsrc/libgenerics-1.2.1/source/Makefile.am	2007-06-18 01:18:58.000000000 +0900
+++ src/libgenerics-1.2.1/source/Makefile.am	2015-05-19 04:46:42.454897800 +0900
@@ -1,4 +1,4 @@
-INCLUDES = -I../include `pkg-config --cflags libxml-2.0`
+INCLUDES = -I$(top_srcdir)/include $(DEPS_CFLAGS)
 
 lib_LTLIBRARIES = libgenerics.la
 
@@ -15,5 +15,6 @@ libgenerics_la_SOURCES =cstring.cpp 			\
 		 	cmetamodule.cpp			\
 		 	cobject.cpp 
 
-libgenerics_la_LDFLAGS =  `pkg-config --libs libxml-2.0`
+libgenerics_la_LDFLAGS =  -no-undefined
+libgenerics_la_LIBADD = $(DEPS_LIBS)
 libgenerics_la_CPPFLAGS = -Wno-multichar
