--- origsrc/ua/Makefile.am	2019-11-28 16:12:43.000000000 +0900
+++ src/ua/Makefile.am	2019-11-28 16:12:46.544637200 +0900
@@ -1,7 +1,9 @@
 bin_PROGRAMS = ua kua
 
 ua_SOURCES = filei.cc filei.h ua.cc 
+ua_LDADD = $(OPENSSL_LIBADD)
 kua_SOURCES = filei.cc filei.h kua.cc
+kua_LDADD = $(OPENSSL_LIBADD)
 man_MANS = ua.1 kua.1
 
 EXTRA_DIST = $(man_MANS)
--- origsrc/ua/configure.in	2019-11-28 16:12:44.000000000 +0900
+++ src/ua/configure.in	2019-11-28 16:12:46.544637200 +0900
@@ -9,5 +9,6 @@ AC_PROG_INSTALL
 AC_CHECK_LIB(crypto, MD5_Init)
 AC_CHECK_LIB(crypto, MD5_Update)
 AC_CHECK_LIB(crypto, MD5_Final)
+PKG_CHECK_MODULES([OPENSSL], [openssl])
 
 AC_OUTPUT(Makefile)
--- origsrc/ua/filei.cc	2019-11-28 16:12:44.000000000 +0900
+++ src/ua/filei.cc	2019-11-28 16:14:45.070697600 +0900
@@ -36,6 +36,7 @@
 
 extern "C" {
 #include <stdlib.h>
+#include <string.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
@@ -51,7 +52,7 @@ char filei::_buffer[__UABUFFSIZE];
      
 filei::filei(const std::string& path, bool ic, bool iw, size_t m, size_t bs)
 throw(const char*):_path(path),_h(0)  {
-   ::bzero(_md5,16); // zero out
+   bzero(_md5,16); // zero out
    calc(ic,iw,bs,m);
 }
 
@@ -333,5 +334,3 @@ bool filei::md5eq::operator()(const file
    }
    return true;
 }
-
-
--- origsrc/ua/kua.cc	2019-11-28 16:12:44.000000000 +0900
+++ src/ua/kua.cc	2019-11-28 16:12:46.560264400 +0900
@@ -49,6 +49,7 @@
 
 extern "C" {
 #include <stdio.h>
+#include <getopt.h>
 }
 
 static char __help[] = 
--- origsrc/ua/ua.cc	2019-11-28 16:12:44.000000000 +0900
+++ src/ua/ua.cc	2019-11-28 16:12:46.575889200 +0900
@@ -57,6 +57,7 @@
 
 extern "C" {
 #include <stdio.h>
+#include <getopt.h>
 }
 
 static char __help[] = 
