--- origsrc/libkiwix-12.0.0/meson.build	2022-12-01 02:10:19.000000000 +0900
+++ src/libkiwix-12.0.0/meson.build	2023-04-06 07:17:43.230272400 +0900
@@ -73,7 +73,11 @@ if get_option('doc')
   subdir('docs')
 endif
 
-pkg_requires = ['libzim', 'icu-i18n', 'pugixml', 'libcurl', 'libmicrohttpd', 'xapian-core']
+pkg_requires = ['libzim', 'icu-i18n', 'libcurl', 'libmicrohttpd', 'xapian-core']
+if host_machine.system() != 'cygwin' # cygwin libpugixml-devel does not contain pugixml.pc
+  pkg_requires += 'pugixml'
+endif
+
 
 pkg_conf = configuration_data()
 pkg_conf.set('prefix', get_option('prefix'))
--- origsrc/libkiwix-12.0.0/src/meson.build	2022-12-01 02:10:19.000000000 +0900
+++ src/libkiwix-12.0.0/src/meson.build	2023-04-06 07:17:43.239213900 +0900
@@ -51,5 +51,4 @@ libkiwix = library('kiwix',
                    dependencies : all_deps,
                    link_args: extra_libs,
                    version: meson.project_version(),
-                   install: true,
-                   install_dir: install_dir)
+                   install: true)
--- origsrc/libkiwix-12.0.0/src/tools/networkTools.cpp	2022-12-01 02:10:19.000000000 +0900
+++ src/libkiwix-12.0.0/src/tools/networkTools.cpp	2023-04-06 07:17:43.244252800 +0900
@@ -133,7 +133,7 @@ std::map<std::string, std::string> kiwix
     /* some systems have ifr_addr.sa_len and adjust the length that
      * way, but not mine. weird */
     size_t len;
-#ifndef __linux__
+#if !defined(__linux__) && !defined(__CYGWIN__)
     len = IFNAMSIZ + ifreq->ifr_addr.sa_len;
 #else
     len = sizeof(*ifreq);
