--- origsrc/libkiwix-14.2.0/src/meson.build	2026-03-02 16:18:56.000000000 +0900
+++ src/libkiwix-14.2.0/src/meson.build	2026-03-21 03:32:51.224731100 +0900
@@ -56,5 +56,4 @@ libkiwix = library('kiwix',
                    dependencies : all_deps,
                    link_args: extra_libs,
                    version: meson.project_version(),
-                   install: true,
-                   install_dir: install_dir)
+                   install: true)
