--- origsrc/libeXosip2-3.3.0/Makefile.am	2008-09-18 20:50:19.000000000 +0900
+++ src/libeXosip2-3.3.0/Makefile.am	2010-12-07 10:51:39.876329700 +0900
@@ -3,6 +3,8 @@ EXTRA_DIST = README autogen.sh
 
 SUBDIRS = scripts platform src include tools help
 
+ACLOCAL_INCLUDE = -I scripts
+
 doxygen:
 	doxygen help/doxygen/DoxyFile
 
--- origsrc/libeXosip2-3.3.0/configure.in	2009-03-04 19:50:58.000000000 +0900
+++ src/libeXosip2-3.3.0/configure.in	2010-12-07 11:30:13.112600100 +0900
@@ -44,7 +44,6 @@ dnl Initialize automake stuff
 AC_CONFIG_AUX_DIR(scripts)
 AM_CONFIG_HEADER(config.h)
 AM_INIT_AUTOMAKE($PACKAGE, $VERSION)
-AM_ACLOCAL_INCLUDE(scripts)
 
 AC_CANONICAL_HOST
 
@@ -58,8 +57,8 @@ AC_PROG_CXX
 AC_PROG_INSTALL
 
 dnl enable --enable-shared and --disable-shared options
+LT_INIT(win32-dll)
 LT_AC_PROG_SED
-AC_PROG_LIBTOOL
 
 #AC_ENABLE_SHARED(yes)
 
@@ -192,7 +191,7 @@ AC_CHECK_LIB(nsl,nis_add,[EXOSIP_LIB="$E
 AC_CHECK_LIB(socket,sendto,[EXOSIP_LIB="$EXOSIP_LIB -lsocket"])
 AC_CHECK_LIB(rt,clock_gettime,[EXOSIP_LIB="$EXOSIP_LIB -lrt"])
 
-AC_CHECK_LIB(ssl,SSL_CTX_new,[EXOSIP_LIB="$EXOSIP_LIB -lssl"])
+PKG_CHECK_MODULES(SSL, openssl)
 
 case $OS in
   openbsd*)
--- origsrc/libeXosip2-3.3.0/src/Makefile.am	2008-09-18 20:50:19.000000000 +0900
+++ src/libeXosip2-3.3.0/src/Makefile.am	2010-12-07 11:31:03.324801700 +0900
@@ -47,7 +47,7 @@ eXrefer_api.c           jpublish.c
 sdp_offans.c
 endif
 
-libeXosip2_la_LDFLAGS = -version-info $(LIBEXOSIP_SO_VERSION)
-libeXosip2_la_LIBADD = @EXOSIP_LIB@ @PTHREAD_LIBS@ $(OSIP_LIBS)
+libeXosip2_la_LDFLAGS = -version-info $(LIBEXOSIP_SO_VERSION) -no-undefined
+libeXosip2_la_LIBADD = @EXOSIP_LIB@ @PTHREAD_LIBS@ $(OSIP_LIBS) @SSL_LIBS@
 
-INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/include $(OSIP_CFLAGS)
+INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/include $(OSIP_CFLAGS) @SSL_CFLAGS@
