--- origsrc/libexosip2-5.1.3/Makefile.am	2020-08-24 21:38:28.000000000 +0900
+++ src/libexosip2-5.1.3/Makefile.am	2020-08-27 10:59:53.035517100 +0900
@@ -20,3 +20,6 @@ indent:
 	rm -rf $(top_srcdir)/*/*/*.h~
 	rm -rf $(top_srcdir)/*/*.h~
 	rm -rf $(top_srcdir)/*/*.c~
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libexosip2-5.1.3/configure.ac	2020-08-25 21:49:26.000000000 +0900
+++ src/libexosip2-5.1.3/configure.ac	2020-08-27 10:59:53.005522500 +0900
@@ -358,7 +358,7 @@ if test "x$enable_srvrec" = "xyes" ; the
   AC_DEFINE([SRV_RECORD], [1], [Defined to 1 if compiled with NAPTR and SRV support.])
 fi
 
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([libexosip2.pc Makefile
  scripts/Makefile
  platform/Makefile
  platform/vsnet/Makefile
--- origsrc/libexosip2-5.1.3/libexosip2.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libexosip2-5.1.3/libexosip2.pc.in	2020-08-27 11:02:12.708658300 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libexosip2
+Description: eXtended osip library
+Version: @VERSION@
+Libs: -L${libdir} -lexosip2
+Cflags: -I${includedir}
+Requires: libosip2 >= 5.0
