--- origsrc/libexosip2-5.1.1/Makefile.am	2016-01-21 21:06:52.000000000 +0900
+++ src/libexosip2-5.1.1/Makefile.am	2020-01-19 20:01:55.538947200 +0900
@@ -19,3 +19,6 @@ indent:
 	rm -rf */*/*.h~
 	rm -rf */*.h~
 	rm -rf */*.c~
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libexosip2-5.1.1/configure.ac	2020-01-17 03:02:13.000000000 +0900
+++ src/libexosip2-5.1.1/configure.ac	2020-01-19 20:01:55.554571200 +0900
@@ -357,7 +357,7 @@ if test "x$enable_srvrec" = "xyes" ; the
   AC_DEFINE([SRV_RECORD], [1], [Defined to 1 if compiled with NAPTR and SRV support.])
 fi
 
-AC_CONFIG_FILES([Makefile
+AC_CONFIG_FILES([libexosip2.pc Makefile
  scripts/Makefile
  platform/Makefile
  platform/vsnet/Makefile
--- origsrc/libexosip2-5.1.1/libexosip2.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libexosip2-5.1.1/libexosip2.pc.in	2020-01-19 20:01:55.570399400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libexosip2
+Description: eXtended osip library
+Version: @VERSION@
+Libs: -L${libdir} -lexosip2
+Cflags: -I${includedir}
