--- origsrc/libgitimport/configure.ac	2014-03-10 23:57:24.000000000 +0900
+++ src/libgitimport/configure.ac	2014-03-11 00:02:27.014109100 +0900
@@ -6,7 +6,7 @@ dnl the file LICENSE are met.
 
 AC_INIT([libgitimport], [0.0.1], [itz@buug.org])
 AC_CONFIG_MACRO_DIR([m4])
-AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+AM_INIT_AUTOMAKE([-Wall foreign])
 LT_INIT([])
 AC_CONFIG_SRCDIR([src/ctx.c])
 AC_PROG_CC
--- origsrc/libgitimport/src/Makefile.am	2014-03-10 23:57:24.000000000 +0900
+++ src/libgitimport/src/Makefile.am	2014-03-11 00:11:13.468582100 +0900
@@ -8,7 +8,9 @@ lib_LTLIBRARIES = libgitimport.la
 pkginclude_HEADERS = gitimport.h
 nodist_pkgdata_DATA = en.cat
 
-libgitimport_la_LDFLAGS = -Wl,--version-script=$(srcdir)/libgitimport.sym
+libgitimport_la_CFLAGS = -mcmodel=large
+libgitimport_la_LIBADD = -lcatgets
+libgitimport_la_LDFLAGS = -no-undefined -Wl,--version-script=$(srcdir)/libgitimport.sym
 libgitimport_la_DEPENDENCIES = libgitimport.sym
 
 libgitimport_la_SOURCES = version.c ctx.c callback.c blob.c \
