--- origsrc/xxhash/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/xxhash/Makefile.am	2013-06-07 20:21:48.163485000 +0900
@@ -0,0 +1,13 @@
+
+AM_CFLAGS = -std=c99 -Wall -W -Wundef -Wno-implicit-function-declaration
+
+check_PROGRAMS = bench
+bench_SOURCES = bench.c
+bench_LDADD = libxxhash.la
+
+include_HEADERS = xxhash.h
+
+lib_LTLIBRARIES = libxxhash.la
+libxxhash_la_SOURCES = xxhash.c
+libxxhash_la_LDFLAGS = -no-undefined
+
--- origsrc/xxhash/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/xxhash/configure.ac	2013-06-07 20:20:55.991610000 +0900
@@ -0,0 +1,11 @@
+AC_PREREQ([2.69])
+AC_INIT([xxhash], [0+svn30])
+AC_CONFIG_SRCDIR([bench.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+AC_PROG_CC
+LT_INIT
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
