diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/xxHash-0.5.1/cmake_unofficial/CMakeLists.txt src/xxHash-0.5.1/cmake_unofficial/CMakeLists.txt
--- origsrc/xxHash-0.5.1/cmake_unofficial/CMakeLists.txt	2016-03-01 21:22:23.000000000 +0900
+++ src/xxHash-0.5.1/cmake_unofficial/CMakeLists.txt	2016-03-10 08:53:33.963746700 +0900
@@ -21,4 +21,4 @@ endif(BUILD_STATIC_LIBS)
 
 
 INSTALL(FILES ../xxhash.h DESTINATION include)
-INSTALL(TARGETS ${install_libs} DESTINATION lib)
+INSTALL(TARGETS ${install_libs} RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
