--- origsrc/xxhash/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/xxhash/Makefile.am	2014-03-15 02:27:31.317535600 +0900
@@ -0,0 +1,12 @@
+AM_CFLAGS = -std=c99 -Wall -W -Wundef -Wno-implicit-function-declaration -O3
+
+check_PROGRAMS = bench
+bench_SOURCES = bench.c
+bench_LDADD = libxxhash.la
+
+include_HEADERS = xxhash.h
+
+lib_LTLIBRARIES = libxxhash.la
+libxxhash_la_SOURCES = xxhash.c
+libxxhash_la_LDFLAGS = -no-undefined
+
--- origsrc/xxhash/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/xxhash/configure.ac	2014-03-15 02:28:08.803863800 +0900
@@ -0,0 +1,23 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([xxhash], [0+svn34])
+AC_CONFIG_SRCDIR([bench.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+
+# Checks for typedefs, structures, and compiler characteristics.
+
+# Checks for library functions.
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
