--- origsrc/libarchive-qt-v2.0.2/LibArchive-Qt.pro	2020-05-13 16:05:23.000000000 +0900
+++ src/libarchive-qt-v2.0.2/LibArchive-Qt.pro	2020-10-20 15:38:08.210389900 +0900
@@ -1,12 +1,9 @@
 TEMPLATE = subdirs
 
 # Directories
-SUBDIRS += shared static example
+SUBDIRS += shared example
 
 shared.file = lib/shared.pro
 
-static.file = lib/static.pro
-static.depends = shared
-
-example.depends = static
+example.depends = shared
 example.file = example/example.pro
--- origsrc/libarchive-qt-v2.0.2/example/example.pro	2020-05-13 16:05:23.000000000 +0900
+++ src/libarchive-qt-v2.0.2/example/example.pro	2020-10-20 15:38:08.229485300 +0900
@@ -4,7 +4,7 @@ TARGET = archiver
 DEPENDPATH += . ../lib/
 INCLUDEPATH += . ../lib/
 
-LIBS += ../lib/libarchiveqt5.a -larchive -lz -lbz2 -llzma
+LIBS += ../lib/libarchiveqt5.dll.a -larchive -lz -lbz2 -llzma
 
 contains ( DEFINES, HAVE_LZLIB ) {
 	LIBS += -llz
--- origsrc/libarchive-qt-v2.0.2/lib/pkgconfig/archiveqt.pc	2020-05-13 16:05:23.000000000 +0900
+++ src/libarchive-qt-v2.0.2/lib/pkgconfig/archiveqt.pc	2020-10-20 15:43:22.472886600 +0900
@@ -6,7 +6,7 @@ qt_config=lex yacc warn_on uic resources
 
 Name: libarchive-qt
 Description: A Qt based archiving solution with libarchive backend
-Version: 1.1.1
-Libs: -L${libdir} -larchiveqt
-Libs.private: -L/usr/lib/x86_64-linux-gnu  -larchive -lz -lbz2 -llzma -lQtCore -lpthread
+Version: 2.0.2
+Libs: -L${libdir} -larchiveqt5
+Libs.private: -larchive -lz -lbz2 -llzma -lQt5Core -lpthread
 Cflags:  -I${includedir}
--- origsrc/libarchive-qt-v2.0.2/lib/pkgconfig/archiveqt5.pc	2020-05-13 16:05:23.000000000 +0900
+++ src/libarchive-qt-v2.0.2/lib/pkgconfig/archiveqt5.pc	2020-10-20 15:43:32.594215400 +0900
@@ -6,7 +6,7 @@ qt_config=lex yacc warn_on uic resources
 
 Name: libarchive-qt
 Description: A Qt based archiving solution with libarchive backend
-Version: 1.1.1
+Version: 2.0.2
 Libs: -L${libdir} -larchiveqt5
-Libs.private: -L/usr/lib/x86_64-linux-gnu  -larchive -lz -lbz2 -llzma -lQtCore -lpthread
+Libs.private: -larchive -lz -lbz2 -llzma -lQt5Core -lpthread
 Cflags:  -I${includedir}
--- origsrc/libarchive-qt-v2.0.2/lib/shared.pro	2020-05-13 16:05:23.000000000 +0900
+++ src/libarchive-qt-v2.0.2/lib/shared.pro	2020-10-20 15:38:08.334459300 +0900
@@ -54,9 +54,6 @@ unix {
 	target.path			= $$PREFIX/lib/
 	includes.files		= libarchiveqt.h
 	includes.path		= $$PREFIX/include/
-
-	data.path = $$PREFIX/share/lib$$TARGET/
-	data.files = Changelog README
 }
 
 macx {
