--- origsrc/libarchive-qt-v2.0.6/CMakeLists.txt	2021-08-16 16:42:18.000000000 +0900
+++ src/libarchive-qt-v2.0.6/CMakeLists.txt	2021-08-30 13:04:20.966967600 +0900
@@ -34,6 +34,10 @@ include_directories( lib )
 add_subdirectory( lib )
 add_subdirectory( archiver )
 
-add_dependencies( archiver mylib_static )
+if (BUILD_SHARED_LIBS)
+    add_dependencies( archiver mylib_shared )
+else()
+    add_dependencies( archiver mylib_static )
+endif()
 
 message( "Installing to ${CMAKE_INSTALL_PREFIX}" )
--- origsrc/libarchive-qt-v2.0.6/archiver/CMakeLists.txt	2021-08-16 16:42:18.000000000 +0900
+++ src/libarchive-qt-v2.0.6/archiver/CMakeLists.txt	2021-08-30 13:08:03.339067100 +0900
@@ -25,11 +25,11 @@ endif()
 add_executable( archiver archiver.cpp  )
 
 if ( USEQT STREQUAL "Qt4" )
-	target_link_libraries( archiver  Qt4::QtCore ${CMAKE_CURRENT_BINARY_DIR}/../lib/libarchiveqt4.a archive z bz2 lzma )
+	target_link_libraries( archiver  Qt4::QtCore mylib_shared archive z bz2 lzma )
 elseif ( USEQT STREQUAL "Qt5" )
-	target_link_libraries( archiver  Qt5::Core ${CMAKE_CURRENT_BINARY_DIR}/../lib/libarchiveqt5.a archive z bz2 lzma )
+	target_link_libraries( archiver  Qt5::Core mylib_shared archive z bz2 lzma )
 elseif ( USEQT STREQUAL "Qt6" )
-	target_link_libraries( archiver  Qt::Core ${CMAKE_CURRENT_BINARY_DIR}/../lib/libarchiveqt6.a archive z bz2 lzma )
+	target_link_libraries( archiver  Qt::Core mylib_shared archive z bz2 lzma )
 endif()
 
 
--- origsrc/libarchive-qt-v2.0.6/lib/CMakeLists.txt	2021-08-16 16:42:18.000000000 +0900
+++ src/libarchive-qt-v2.0.6/lib/CMakeLists.txt	2021-08-30 13:13:03.148839600 +0900
@@ -110,16 +110,18 @@ elseif ( USEQT STREQUAL "Qt6" )
 	target_link_libraries( mylib_shared  Qt::Core archive z bz2 lzma )
 endif()
 
-add_library( mylib_static STATIC $<TARGET_OBJECTS:archiveqt> )
-if ( USEQT STREQUAL "Qt4" )
-	set_target_properties( mylib_static PROPERTIES OUTPUT_NAME archiveqt4 )
-	target_link_libraries( mylib_static  Qt4::QtCore archive z bz2 lzma )
-elseif ( USEQT STREQUAL "Qt5" )
-	set_target_properties( mylib_static PROPERTIES OUTPUT_NAME archiveqt5 )
-	target_link_libraries( mylib_static  Qt5::Core archive z bz2 lzma )
-elseif ( USEQT STREQUAL "Qt6" )
-	set_target_properties( mylib_static PROPERTIES OUTPUT_NAME archiveqt6 )
-	target_link_libraries( mylib_static  Qt::Core archive z bz2 lzma )
+if (NOT BUILD_SHARED_LIBS)
+	add_library( mylib_static STATIC $<TARGET_OBJECTS:archiveqt> )
+	if ( USEQT STREQUAL "Qt4" )
+		set_target_properties( mylib_static PROPERTIES OUTPUT_NAME archiveqt4 )
+		target_link_libraries( mylib_static  Qt4::QtCore archive z bz2 lzma )
+	elseif ( USEQT STREQUAL "Qt5" )
+		set_target_properties( mylib_static PROPERTIES OUTPUT_NAME archiveqt5 )
+		target_link_libraries( mylib_static  Qt5::Core archive z bz2 lzma )
+	elseif ( USEQT STREQUAL "Qt6" )
+		set_target_properties( mylib_static PROPERTIES OUTPUT_NAME archiveqt6 )
+		target_link_libraries( mylib_static  Qt::Core archive z bz2 lzma )
+	endif()
 endif()
 
 install( TARGETS mylib_shared EXPORT LIBRARY )
@@ -134,11 +136,11 @@ install( FILES libarchiveqt.h DESTINATIO
 ## Configure and install the pkg-config .pc files
 if ( USEQT STREQUAL "Qt4" )
     configure_file( pkgconfig/archiveqt.pc.in archiveqt.pc @ONLY )
-    install( FILES ${CMAKE_CURRENT_BINARY_DIR}/archiveqt.pc DESTINATION share/pkgconfig )
+    install( FILES ${CMAKE_CURRENT_BINARY_DIR}/archiveqt.pc DESTINATION lib/pkgconfig )
 elseif ( USEQT STREQUAL "Qt5" )
     configure_file( pkgconfig/archiveqt5.pc.in archiveqt5.pc @ONLY )
-    install( FILES ${CMAKE_CURRENT_BINARY_DIR}/archiveqt5.pc DESTINATION share/pkgconfig )
+    install( FILES ${CMAKE_CURRENT_BINARY_DIR}/archiveqt5.pc DESTINATION lib/pkgconfig )
 elseif ( USEQT STREQUAL "Qt6" )
     configure_file( pkgconfig/archiveqt6.pc.in archiveqt6.pc @ONLY )
-    install( FILES ${CMAKE_CURRENT_BINARY_DIR}/archiveqt6.pc DESTINATION share/pkgconfig )
+    install( FILES ${CMAKE_CURRENT_BINARY_DIR}/archiveqt6.pc DESTINATION lib/pkgconfig )
 endif()
