--- origsrc/libarchive-qt-v1.0.2/LibArchive-Qt.pro	2019-02-19 18:58:29.000000000 +0900
+++ src/libarchive-qt-v1.0.2/LibArchive-Qt.pro	2019-12-18 16:32:00.620783900 +0900
@@ -1,12 +1,9 @@
 TEMPLATE = subdirs
 
 # Directories
-SUBDIRS += shared static example
+SUBDIRS += shared example
 
 shared.file = lib/shared.pro
 
-static.file = lib/static.pro
-static.depends = shared
-
-example.depends = static
+example.depends = shared
 example.file = example/example.pro
--- origsrc/libarchive-qt-v1.0.2/example/example.pro	2019-02-19 18:58:29.000000000 +0900
+++ src/libarchive-qt-v1.0.2/example/example.pro	2019-12-18 16:33:18.242304900 +0900
@@ -4,7 +4,7 @@ TARGET = archiver
 DEPENDPATH += . ../lib/
 INCLUDEPATH += . ../lib/
 
-LIBS += ../lib/libarchiveqt5.a -larchive -lz -lbz2 -llzma
+LIBS += ../lib/libarchiveqt5.dll.a -larchive -lz -lbz2 -llzma
 
 # Input
 SOURCES += example.cpp
--- origsrc/libarchive-qt-v1.0.2/lib/pkgconfig/archiveqt.pc	2019-02-19 18:58:29.000000000 +0900
+++ src/libarchive-qt-v1.0.2/lib/pkgconfig/archiveqt.pc	2019-12-18 16:37:12.186088200 +0900
@@ -6,7 +6,7 @@ qt_config=lex yacc warn_on uic resources
 
 Name: libarchive-qt
 Description: A Qt based archiving solution with libarchive backend
-Version: 1.1.0
-Libs: -L${libdir} -larchiveqt
-Libs.private: -L/usr/lib/x86_64-linux-gnu  -larchive -lz -lbz2 -llzma -lQtCore -lpthread
+Version: 1.0.2
+Libs: -L${libdir} -larchiveqt5
+Libs.private: -larchive -lz -lbz2 -llzma -lQtCore -lpthread
 Cflags:  -I${includedir}
--- origsrc/libarchive-qt-v1.0.2/lib/shared.pro	2019-02-19 18:58:29.000000000 +0900
+++ src/libarchive-qt-v1.0.2/lib/shared.pro	2019-12-18 16:40:15.340230500 +0900
@@ -55,9 +55,6 @@ unix {
 	includes.files		= libarchiveqt.h
 	includes.path		= $$PREFIX/include/
 
-	data.path = $$PREFIX/share/lib$$TARGET/
-	data.files = Changelog README
-
 	QMAKE_PKGCONFIG_NAME = libarchive-qt
 	QMAKE_PKGCONFIG_DESCRIPTION = A Qt based archiving solution with libarchive backend
 	QMAKE_PKGCONFIG_PREFIX  = $$INSTALL_PREFIX
