--- origsrc/cidrmerge/Makefile	2008-07-08 18:45:33.000000000 +0900
+++ src/cidrmerge/Makefile	2024-11-05 14:19:35.596034300 +0900
@@ -2,9 +2,9 @@
 CC=gcc
 
 #Usual gcc flags
-CC_FLAGS=-O3 -fPIC -g -c -Wall -W
+#CC_FLAGS=-O3 -fPIC -g -c -Wall -W
 #Enforce ansi compatibility
-CC_FLAGS+= -ansi -pedantic
+#CC_FLAGS+= -ansi -pedantic
 #Pack struct: this will reduce significantly memory usage (- 30%), but some platform cannot handle not aligned to word integers (SIGBUS will be generated in this case). It can cause incompatibility with code compiled without this flag too and program to be slower in some cases, it needs test on target system.
 #CC_FLAGS+=-fpack-struct
 #cpu dependant, configure accordingly to final running machine. It can increase significantly performances
