--- origsrc/camp/CMakeLists.txt	2024-12-13 09:49:33.000000000 +0900
+++ src/camp/CMakeLists.txt	2024-12-13 09:49:46.403123400 +0900
@@ -147,6 +147,10 @@ set_target_properties (camp PROPERTIES
   INTERFACE_LIB_VERSION ${PROJECT_VERSION}
   INTERFACE_COMPILE_FEATURES cxx_std_14)
 
+if (DEFINED SOVERSION)
+  set_target_properties(camp PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 include(CMakePackageConfigHelpers)
 
 write_basic_package_version_file(
@@ -158,7 +162,7 @@ write_basic_package_version_file(
 install(TARGETS
   camp
   EXPORT campTargets
-  RUNTIME DESTINATION lib
+  RUNTIME DESTINATION bin
   LIBRARY DESTINATION lib
   ARCHIVE DESTINATION lib)
 
@@ -195,3 +199,9 @@ if(CAMP_ENABLE_DOCUMENTATION)
   add_subdirectory(docs)
 endif()
 
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/camp.pc.in ${CMAKE_BINARY_DIR}/camp.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/camp.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/camp/camp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/camp/camp.pc.in	2024-12-13 09:49:46.418712400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: camp
+Description: Concepts And Meta-Programming library
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lcamp
+Cflags: -I${includedir}
