--- origsrc/libisoburn-1.5.0/Makefile.am	2017-09-29 20:31:45.000000000 +0900
+++ src/libisoburn-1.5.0/Makefile.am	2018-11-13 11:17:07.852087700 +0900
@@ -11,7 +11,7 @@ ACLOCAL_AMFLAGS = -I ./
 ## ========================================================================= ##
 
 # Build libraries
-libisoburn_libisoburn_la_LDFLAGS = \
+libisoburn_libisoburn_la_LDFLAGS = -no-undefined \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LIBLDFLAGS)
 libisoburn_libisoburn_la_SOURCES = \
 	libisoburn/isoburn.h \
@@ -295,5 +295,3 @@ EXTRA_DIST = \
 	releng/run_all_auto \
 	releng/template_new \
 	$(man_MANS)
-
-
