--- origsrc/libisoburn-1.1.8/Makefile.am	2011-11-21 02:19:14.000000000 +0900
+++ src/libisoburn-1.1.8/Makefile.am	2012-01-05 23:37:40.395902300 +0900
@@ -12,7 +12,7 @@ ACLOCAL_AMFLAGS = -I ./
 
 # Build libraries
 libisoburn_libisoburn_la_LDFLAGS = \
-	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
+	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -no-undefined
 libisoburn_libisoburn_la_SOURCES = \
 	libisoburn/isoburn.h \
 	libisoburn/libisoburn.h \
