--- origsrc/ParMGridGen-1.0/MGridGen/Lib/Makefile.am	2015-03-25 08:52:07.295853700 +0900
+++ src/ParMGridGen-1.0/MGridGen/Lib/Makefile.am	2015-03-25 08:54:28.753861500 +0900
@@ -1,6 +1,6 @@
 ## Source directory
 
-INCLUDES = -I../IMlib
+INCLUDES = -I$(top_srcdir)/MGridGen/IMlib
 
 pkginclude_HEADERS = defs.h  macros.h  mgridgen.h  proto.h  struct.h
 pkgincludedir = $(includedir)/mgridgen
--- origsrc/ParMGridGen-1.0/MGridGen/Programs/Makefile.am	2015-03-25 08:52:07.217728700 +0900
+++ src/ParMGridGen-1.0/MGridGen/Programs/Makefile.am	2015-03-25 08:56:57.302689600 +0900
@@ -1,6 +1,6 @@
 ## Source directory
 
-INCLUDES = -I../IMlib -I../Lib
+INCLUDES = -I$(top_srcdir)/MGridGen/IMlib -I$(top_srcdir)/MGridGen/Lib
 
 bin_PROGRAMS = mgridgen
 mgridgen_SOURCES = io.c mgridgen.c
--- origsrc/ParMGridGen-1.0/ParMGridGen/IMParMetis-2.0/ParMETISLib/Makefile.am	2015-03-25 08:52:07.233353700 +0900
+++ src/ParMGridGen-1.0/ParMGridGen/IMParMetis-2.0/ParMETISLib/Makefile.am	2015-03-25 09:03:46.544877100 +0900
@@ -6,3 +6,4 @@ pkgincludedir = $(includedir)/imparmetis
 lib_LTLIBRARIES = libIMparmetis.la
 libIMparmetis_la_SOURCES = coarsen.c  comm.c  debug.c  diffuse.c  drivers.c  edge_refine.c  fused.c  grsetup.c  iidxsort.c  ikeysort.c  memory.c  remap.c  setup.c  timer.c  util.c  var.c
 libIMparmetis_la_LDFLAGS = -no-undefined -export-dynamic -version-info 1:0:0
+libIMparmetis_la_LIBADD = $(top_builddir)/MGridGen/IMlib/libIMlib.la -lmpi
--- origsrc/ParMGridGen-1.0/ParMGridGen/ParLib/Makefile.am	2015-03-25 08:52:07.295853700 +0900
+++ src/ParMGridGen-1.0/ParMGridGen/ParLib/Makefile.am	2015-03-25 09:04:59.055619300 +0900
@@ -1,6 +1,6 @@
 ## Source directory
 
-INCLUDES = -I$(top_builddir)/MGridGen/IMlib
+INCLUDES = -I$(top_srcdir)/MGridGen/IMlib
 
 pkginclude_HEADERS = defs.h  macros.h  parmgridgen.h  proto.h  rename.h  struct.h
 pkgincludedir = $(includedir)/parmgridgen
--- origsrc/ParMGridGen-1.0/ParMGridGen/Programs/Makefile.am	2015-03-25 08:52:07.233353700 +0900
+++ src/ParMGridGen-1.0/ParMGridGen/Programs/Makefile.am	2015-03-25 09:06:29.698197400 +0900
@@ -1,6 +1,6 @@
 ## Source directory
 
-INCLUDES = -I../ParLib -I$(top_builddir)/MGridGen/IMlib
+INCLUDES = -I$(top_srcdir)/ParMGridGen/ParLib -I$(top_srcdir)/MGridGen/IMlib
 
 bin_PROGRAMS = parmgridgen
 parmgridgen_SOURCES = io.c  main.c  parmgridgen.c
