--- origsrc/libchmxx-0.2/Makefile.am	2003-10-02 09:18:09.000000000 +0900
+++ src/libchmxx-0.2/Makefile.am	2024-10-14 14:51:16.548000900 +0900
@@ -3,3 +3,6 @@
 AUTOMAKE_OPTIONS = foreign 1.4
 
 SUBDIRS = src
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libchmxx-0.2/configure.in	2008-01-04 19:22:44.000000000 +0900
+++ src/libchmxx-0.2/configure.in	2024-10-14 14:51:16.548000900 +0900
@@ -8,5 +8,5 @@ AC_LANG([C++])
 AC_PROG_CXX
 AC_PROG_LIBTOOL
 
-AC_CONFIG_FILES([Makefile src/Makefile])
+AC_CONFIG_FILES([libchmxx.pc Makefile src/Makefile])
 AC_OUTPUT
--- origsrc/libchmxx-0.2/libchmxx.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libchmxx-0.2/libchmxx.pc.in	2024-10-14 14:51:16.548000900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libchmxx
+Description: C++ library for accessing the Microsoft(tm) CHM files
+Version: @VERSION@
+Libs: -L${libdir} -lchmxx
+Cflags: -I${includedir}
--- origsrc/libchmxx-0.2/src/Makefile.am	2008-01-04 19:23:40.000000000 +0900
+++ src/libchmxx-0.2/src/Makefile.am	2024-10-14 14:51:16.548000900 +0900
@@ -4,7 +4,7 @@ INCLUDES = $(all_includes)
 
 # the library search path.
 lib_LTLIBRARIES =  libchmxx.la
-libchmxx_la_LDFLAGS = 
+libchmxx_la_LDFLAGS = -no-undefined
 libchmxx_la_SOURCES = chm.cpp chmxx.h tagreader.cpp tagreader.h
 include_HEADERS =  chmxx.h
 bin_PROGRAMS = chmdissect
