diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/axe-0.2.6/CMakeLists.txt src/axe-0.2.6/CMakeLists.txt
--- origsrc/axe-0.2.6/CMakeLists.txt	2015-06-01 13:52:10.000000000 +0900
+++ src/axe-0.2.6/CMakeLists.txt	2015-06-14 18:33:25.378435100 +0900
@@ -69,8 +69,10 @@ INCLUDE_DIRECTORIES(${CMAKE_BINARY_DIR})
 CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/src/axe_config.h.in
                ${CMAKE_BINARY_DIR}/axe_config.h)
 
+INCLUDE(FindPkgConfig)
+PKG_CHECK_MODULES(LIBQES REQUIRED libqes)
+
+INCLUDE_DIRECTORIES(${LIBQES_INCLUDE_DIRS})
+
 ADD_SUBDIRECTORY(docs)
-ADD_SUBDIRECTORY(tests)
 ADD_SUBDIRECTORY(src)
-SET(LIBQES_AS_SUBMODULE True)  # stop libqes installing itself
-ADD_SUBDIRECTORY(src/libqes)
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/axe-0.2.6/src/CMakeLists.txt src/axe-0.2.6/src/CMakeLists.txt
--- origsrc/axe-0.2.6/src/CMakeLists.txt	2015-06-01 13:52:10.000000000 +0900
+++ src/axe-0.2.6/src/CMakeLists.txt	2015-06-14 18:33:45.055278600 +0900
@@ -1,13 +1,13 @@
 # Axe library (libaxe.a)
 FILE(GLOB DATRIE_SRCS ${CMAKE_CURRENT_SOURCE_DIR}/datrie/*.c)
 FILE(GLOB GSL_SRCS ${CMAKE_CURRENT_SOURCE_DIR}/gsl/*.c)
-SET(AXELIB_SRCS ${DATRIE_SRCS} ${GSL_SRCS} axe.c)
+SET(AXELIB_SRCS axe.c)
 
 ADD_LIBRARY(axelib STATIC ${AXELIB_SRCS})
-TARGET_LINK_LIBRARIES(axelib qes_static)
+TARGET_LINK_LIBRARIES(axelib qes)
 SET_TARGET_PROPERTIES(axelib PROPERTIES OUTPUT_NAME axe)
 
 # Executable
 ADD_EXECUTABLE(axe main.c)
-TARGET_LINK_LIBRARIES(axe ${AXE_DEPENDS_LIBS} axelib)
+TARGET_LINK_LIBRARIES(axe ${AXE_DEPENDS_LIBS} axelib datrie gsl)
 INSTALL(TARGETS axe DESTINATION "bin")
