--- origsrc/libfwsi-20240423/Makefile.am	2024-04-23 14:00:36.000000000 +0900
+++ src/libfwsi-20240423/Makefile.am	2025-05-18 06:07:48.236824900 +0900
@@ -3,19 +3,8 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libuna \
-	libfdatetime \
-	libfguid \
-	libfole \
-	libfwps \
 	libfwsi \
 	pyfwsi \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libfwsi-20240423/configure.ac	2024-04-23 14:00:36.000000000 +0900
+++ src/libfwsi-20240423/configure.ac	2025-05-18 06:07:48.236824900 +0900
@@ -142,16 +142,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfole/Makefile])
-AC_CONFIG_FILES([libfwps/Makefile])
 AC_CONFIG_FILES([libfwsi/Makefile])
 AC_CONFIG_FILES([pyfwsi/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
--- origsrc/libfwsi-20240423/pyfwsi/Makefile.am	2024-04-23 14:00:36.000000000 +0900
+++ src/libfwsi-20240423/pyfwsi/Makefile.am	2025-05-18 06:08:12.425203500 +0900
@@ -50,7 +50,7 @@ pyfwsi_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pyfwsi_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyfwsi_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyfwsi_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
