--- origsrc/libfwsi-20160110/Makefile.am	2016-01-10 17:49:43.000000000 +0900
+++ src/libfwsi-20160110/Makefile.am	2016-04-05 16:12:33.500973700 +0900
@@ -3,24 +3,11 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libuna \
-	libfdatetime \
-	libfguid \
-	libfole \
-	libfwps \
 	libfwsi \
 	pyfwsi \
 	pyfwsi2 \
 	pyfwsi3 \
-	po \
 	manuals \
-	libcfile \
 	tests \
 	msvscpp
 
@@ -117,4 +104,3 @@ splint:
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libcfile && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
-
--- origsrc/libfwsi-20160110/configure.ac	2016-01-10 17:49:43.000000000 +0900
+++ src/libfwsi-20160110/configure.ac	2016-04-05 16:12:40.677736800 +0900
@@ -163,17 +163,6 @@ dnl Have configure make the Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfole/Makefile])
-AC_CONFIG_FILES([libfwps/Makefile])
 AC_CONFIG_FILES([libfwsi/Makefile])
 AC_CONFIG_FILES([pyfwsi/Makefile])
 AC_CONFIG_FILES([pyfwsi2/Makefile])
@@ -181,7 +170,6 @@ AC_CONFIG_FILES([pyfwsi3/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
 AC_CONFIG_FILES([manuals/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
 AC_CONFIG_FILES([tests/Makefile])
 AC_CONFIG_FILES([msvscpp/Makefile])
 dnl Have configure make the include files
@@ -225,4 +213,3 @@ Features:
    Verbose output:                    $ac_cv_enable_verbose_output
    Debug output:                      $ac_cv_enable_debug_output
 ]);
-
--- origsrc/libfwsi-20160110/libfwsi/Makefile.am	2016-01-10 17:49:43.000000000 +0900
+++ src/libfwsi-20160110/libfwsi/Makefile.am	2016-04-05 16:13:05.886980300 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -100,4 +101,3 @@ distclean: clean
 splint:
 	@echo "Running splint on libfwsi ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(libfwsi_la_SOURCES)
-
--- origsrc/libfwsi-20160110/pyfwsi/Makefile.am	2016-01-10 17:49:43.000000000 +0900
+++ src/libfwsi-20160110/pyfwsi/Makefile.am	2016-04-05 16:13:27.117823400 +0900
@@ -1,5 +1,6 @@
 if HAVE_PYTHON
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -55,4 +56,3 @@ MAINTAINERCLEANFILES = \
 
 distclean: clean
 	/bin/rm -f Makefile
-
--- origsrc/libfwsi-20160110/pyfwsi2/Makefile.am	2016-01-10 17:49:43.000000000 +0900
+++ src/libfwsi-20160110/pyfwsi2/Makefile.am	2016-04-05 16:13:34.207219600 +0900
@@ -6,6 +6,7 @@ pyexecdir = $(pyexecdir2)
 
 if HAVE_PYTHON2
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -71,4 +72,3 @@ clean-local:
 
 distclean: clean
 	/bin/rm -f Makefile
-
--- origsrc/libfwsi-20160110/pyfwsi3/Makefile.am	2016-01-10 17:49:43.000000000 +0900
+++ src/libfwsi-20160110/pyfwsi3/Makefile.am	2016-04-05 16:13:40.933197800 +0900
@@ -6,6 +6,7 @@ pyexecdir = $(pyexecdir3)
 
 if HAVE_PYTHON3
 AM_CFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -71,4 +72,3 @@ clean-local:
 
 distclean: clean
 	/bin/rm -f Makefile
-
--- origsrc/libfwsi-20160110/tests/Makefile.am	2016-01-10 17:49:43.000000000 +0900
+++ src/libfwsi-20160110/tests/Makefile.am	2016-04-05 16:13:17.968610000 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -46,4 +47,3 @@ MAINTAINERCLEANFILES = \
 
 distclean: clean
 	/bin/rm -f Makefile
-
