--- origsrc/libfmapi-20180714/Makefile.am	2018-07-14 20:18:08.000000000 +0900
+++ src/libfmapi-20180714/Makefile.am	2020-09-02 20:13:44.018646100 +0900
@@ -3,16 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libcnotify \
-	libuna \
-	libfdatetime \
-	libfguid \
-	libfwnt \
 	libfmapi \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libfmapi-20180714/configure.ac	2018-07-14 20:18:08.000000000 +0900
+++ src/libfmapi-20180714/configure.ac	2020-09-02 20:13:44.031652900 +0900
@@ -136,14 +136,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfwnt/Makefile])
 AC_CONFIG_FILES([libfmapi/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
--- origsrc/libfmapi-20180714/libfmapi/Makefile.am	2018-07-14 20:18:08.000000000 +0900
+++ src/libfmapi-20180714/libfmapi/Makefile.am	2020-09-02 20:13:44.043660800 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
--- origsrc/libfmapi-20180714/tests/Makefile.am	2018-07-14 20:18:08.000000000 +0900
+++ src/libfmapi-20180714/tests/Makefile.am	2020-09-02 20:13:44.051644100 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCERROR_CPPFLAGS@ \
