--- origsrc/libitzam-6.0.4/Makefile.am	2011-07-28 13:17:06.000000000 +0900
+++ src/libitzam-6.0.4/Makefile.am	2013-06-02 14:14:36.126571600 +0900
@@ -2,6 +2,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = src test
 EXTRA_DIST = ItzamDocumentation.html LicenseOpenSource.txt LicenseClosedSource.txt reconf cleanup
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = libitzam.pc
+
 dist-hook:
 
 	if test -d ItzamC; then \
--- origsrc/libitzam-6.0.4/src/Makefile.am	2011-07-22 14:56:47.000000000 +0900
+++ src/libitzam-6.0.4/src/Makefile.am	2013-06-02 14:05:12.092391900 +0900
@@ -8,7 +8,7 @@ cpp_sources = itzam_util.c itzam_data.c
 lib_LTLIBRARIES = libitzam.la
 
 libitzam_la_SOURCES = $(h_sources) $(cpp_sources)
-libitzam_la_LDFLAGS= -version-info $(GENERIC_LIBRARY_VERSION) -release $(GENERIC_RELEASE)
+libitzam_la_LDFLAGS= -version-info $(GENERIC_LIBRARY_VERSION) -release $(GENERIC_RELEASE) -no-undefined
 
 library_includedir=$(includedir)/$(GENERIC_LIBRARY_NAME)
 library_include_HEADERS = $(h_sources)
--- origsrc/libitzam-6.0.4/src/itzam.h	2011-09-18 03:22:14.000000000 +0900
+++ src/libitzam-6.0.4/src/itzam.h	2013-06-02 14:03:50.711532500 +0900
@@ -132,7 +132,7 @@ extern "C" {
 
 #if defined(ITZAM_UNIX) || defined(ITZAM_LINUX)
     #include <unistd.h>
-    #if _POSIX_VERSION < 200809L
+    #if !defined(__CYGWIN__) && _POSIX_VERSION < 200809L
         #error "Itzam requires POSIX.1-2008 or later"
     #endif
 #endif
--- origsrc/libitzam-6.0.4/test/Makefile.am	2011-09-18 09:12:28.000000000 +0900
+++ src/libitzam-6.0.4/test/Makefile.am	2013-06-02 14:12:56.279891900 +0900
@@ -9,6 +9,8 @@ itzam_btree_test_stress_SOURCES = itzam_
 itzam_btree_test_threads_SOURCES = itzam_btree_test_threads.c
 itzam_btree_test_strvar_SOURCES = itzam_btree_test_strvar.c
 
-LIBS = -L../src -litzam -lpthread
+LIBS = ../src/libitzam.la -lpthread
 
 EXTRA_DIST = itzam_errors.h
+
+TESTS = $(bin_PROGRAMS)
--- origsrc/libitzam-6.0.4/test/itzam_btree_test_insert.c	2011-09-18 09:12:28.000000000 +0900
+++ src/libitzam-6.0.4/test/itzam_btree_test_insert.c	2013-06-02 14:09:48.123641900 +0900
@@ -114,7 +114,7 @@ itzam_bool test_btree_insert()
     record       rec;
     int n;
     char * filename      = "insert.itz";
-    itzam_int test_size  = 1000000;
+    itzam_int test_size  = 10000;
     int order            = 25;
 
     time_t start = time(NULL);
--- origsrc/libitzam-6.0.4/test/itzam_btree_test_stress.c	2011-07-28 12:35:53.000000000 +0900
+++ src/libitzam-6.0.4/test/itzam_btree_test_stress.c	2013-06-02 14:10:27.123641900 +0900
@@ -179,7 +179,7 @@ itzam_bool test_btree_stress()
     char *       tran_message     = NULL;
     char *       filename         = "stress.itz";
     itzam_int    maxkey           = 400000;
-    itzam_int    test_size        = 1000000;
+    itzam_int    test_size        = 1000;
     int order                     = 25;
     itzam_bool   verified_run     = itzam_false;
     itzam_bool   verbose          = itzam_false;
--- origsrc/libitzam-6.0.4/test/itzam_btree_test_threads.c	2011-07-28 12:29:37.000000000 +0900
+++ src/libitzam-6.0.4/test/itzam_btree_test_threads.c	2013-06-02 14:12:15.904891900 +0900
@@ -195,7 +195,7 @@ static itzam_bool test_threaded()
     char * mfilename = "threaded.multi";
     time_t start;
     int maxkey = 5000;
-    int test_size = 1000000;
+    int test_size = 10000;
     pthread_attr_t attr;
     pthread_t * thread;
     struct threadArgs * args;
