diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/fcl-0.4.0/CMakeModules/CompilerSettings.cmake src/fcl-0.4.0/CMakeModules/CompilerSettings.cmake
--- origsrc/fcl-0.4.0/CMakeModules/CompilerSettings.cmake	2016-01-16 19:57:11.000000000 +0900
+++ src/fcl-0.4.0/CMakeModules/CompilerSettings.cmake	2016-01-29 23:30:15.189911700 +0900
@@ -36,9 +36,9 @@ if(IS_XLC)
     set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -q64")
 endif(IS_XLC)
 
-if((CMAKE_COMPILER_IS_GNUCXX OR IS_ICPC) AND NOT MINGW)
+if((CMAKE_COMPILER_IS_GNUCXX OR IS_ICPC) AND NOT MINGW AND NOT CYGWIN)
     add_definitions(-fPIC)
-endif((CMAKE_COMPILER_IS_GNUCXX OR IS_ICPC) AND NOT MINGW)
+endif((CMAKE_COMPILER_IS_GNUCXX OR IS_ICPC) AND NOT MINGW AND NOT CYGWIN)
 
 # Set rpath http://www.paraview.org/Wiki/CMake_RPATH_handling
 set(CMAKE_SKIP_BUILD_RPATH FALSE)
