--- origsrc/libpciaccess-0.13.4/scanpci/Makefile.am	2015-05-01 13:44:47.000000000 +0900
+++ src/libpciaccess-0.13.4/scanpci/Makefile.am	2016-12-31 17:15:46.336473500 +0900
@@ -21,7 +21,7 @@
 # FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 # DEALINGS IN THE SOFTWARE.
 
-noinst_PROGRAMS = scanpci
+bin_PROGRAMS = scanpci
 
 AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/src
 LDADD =  $(top_builddir)/src/libpciaccess.la
