--- origsrc/libpciaccess/scanpci/Makefile.am	2015-01-09 11:27:19.000000000 +0900
+++ src/libpciaccess/scanpci/Makefile.am	2015-01-09 11:31:08.531281200 +0900
@@ -21,7 +21,7 @@
 # FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 # DEALINGS IN THE SOFTWARE.
 
-noinst_PROGRAMS = scanpci
+bin_PROGRAMS = scanpci
 
 AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/src
 LDADD =  $(top_builddir)/src/libpciaccess.la
