--- origsrc/libpciaccess-0.14/scanpci/Makefile.am	2017-10-24 00:48:29.000000000 +0900
+++ src/libpciaccess-0.14/scanpci/Makefile.am	2017-12-17 00:20:37.341618100 +0900
@@ -21,7 +21,7 @@
 # FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 # DEALINGS IN THE SOFTWARE.
 
-noinst_PROGRAMS = scanpci
+bin_PROGRAMS = scanpci
 
 AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/src
 LDADD =  $(top_builddir)/src/libpciaccess.la
