--- origsrc/libpciaccess-0.13.3/scanpci/Makefile.am	2015-02-04 08:59:14.000000000 +0900
+++ src/libpciaccess-0.13.3/scanpci/Makefile.am	2015-02-21 22:38:59.117389700 +0900
@@ -21,7 +21,7 @@
 # FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 # DEALINGS IN THE SOFTWARE.
 
-noinst_PROGRAMS = scanpci
+bin_PROGRAMS = scanpci
 
 AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/src
 LDADD =  $(top_builddir)/src/libpciaccess.la
