--- origsrc/wake-compiler-0.1.1/makefile	2015-02-16 14:17:26.000000000 +0900
+++ src/wake-compiler-0.1.1/makefile	2015-02-26 17:39:38.332073900 +0900
@@ -19,7 +19,7 @@ OPT=-O0 -g
 FLAGS=-Iinclude -Igen
 CC=cc $(FLAGS)
 CPP=g++ $(FLAGS)
-TEST=false
+TEST=true
 
 CPPNAMES= \
 	ClassSpaceSymbolTable.cpp \
@@ -107,13 +107,13 @@ TESTOBJS=$(addprefix bin/tests/, $(TESTN
 
 include wmake.mk
 
-prefix=/usr/local
+prefix=/usr
 
 install: bin/wake
-	mkdir -p $(prefix)/bin
+	mkdir -p $(DESTDIR)$(prefix)/bin
 	#mkdir -p $(prefix)/share/wake/stdtable
 	#mkdir -p $(prefix)/share/wake/stdobj
-	install -m 0755 bin/wake $(prefix)/bin
+	install -m 0755 bin/wake $(DESTDIR)$(prefix)/bin
 	#install -m 0644 $(WAKETABLEOBJS) $(prefix)/share/wake/stdtable
 	#install -m 0644 bin/wakeobj/std.o $(prefix)/share/wake/stdobj
 
