--- origsrc/sfsexp-1.4.1/examples/Makefile.am	2023-03-24 16:46:08.000000000 +0900
+++ src/sfsexp-1.4.1/examples/Makefile.am	2023-09-26 03:27:08.332588000 +0900
@@ -1,17 +1,17 @@
 CFLAGS = $(SFSEXP_CFLAGS)
-CPPFLAGS = -I../src $(SFSEXP_CPPFLAGS)
+CPPFLAGS = -I$(top_srcdir)/src $(SFSEXP_CPPFLAGS)
 LDFLAGS =
 EXTRA_DIST = testrc vis.in sexps.in edgelist.dat conttest
 
 noinst_PROGRAMS = binmode callbacks continuations packunpack paultest rcfile sexpvis simple_interp
-noinst_HEADERS = ../src/sexp.h
-LDADD = ../src/libsexp.la
-binmode_SOURCES = binmode.c ../src/sexp.h
-callbacks_SOURCES = callbacks.c ../src/sexp.h
-continuations_SOURCES = continuations.c ../src/sexp.h
-packunpack_SOURCES = packunpack.c ../src/sexp.h
-paultest_SOURCES = paultest.c ../src/sexp.h
-rcfile_SOURCES = rcfile.c ../src/sexp.h ../src/sexp_ops.h
-sexpvis_SOURCES = sexpvis.c ../src/sexp.h ../src/sexp_vis.h
-simple_interp_SOURCES = simple_interp.c ../src/sexp.h ../src/sexp_ops.h
+noinst_HEADERS = $(top_srcdir)/src/sexp.h
+LDADD = $(top_builddir)/src/libsexp.la
+binmode_SOURCES = binmode.c $(top_srcdir)/src/sexp.h
+callbacks_SOURCES = callbacks.c $(top_srcdir)/src/sexp.h
+continuations_SOURCES = continuations.c $(top_srcdir)/src/sexp.h
+packunpack_SOURCES = packunpack.c $(top_srcdir)/src/sexp.h
+paultest_SOURCES = paultest.c $(top_srcdir)/src/sexp.h
+rcfile_SOURCES = rcfile.c $(top_srcdir)/src/sexp.h $(top_srcdir)/src/sexp_ops.h
+sexpvis_SOURCES = sexpvis.c $(top_srcdir)/src/sexp.h $(top_srcdir)/src/sexp_vis.h
+simple_interp_SOURCES = simple_interp.c $(top_srcdir)/src/sexp.h $(top_srcdir)/src/sexp_ops.h
 
--- origsrc/sfsexp-1.4.1/src/Makefile.am	2023-03-24 16:46:08.000000000 +0900
+++ src/sfsexp-1.4.1/src/Makefile.am	2023-09-26 03:22:36.975474900 +0900
@@ -4,4 +4,4 @@ CPPFLAGS = $(SFSEXP_CPPFLAGS)
 lib_LTLIBRARIES = libsexp.la
 pkginclude_HEADERS = sexp.h sexp_vis.h sexp_ops.h sexp_memory.h sexp_errors.h cstring.h faststack.h
 libsexp_la_SOURCES = cstring.c cstring.h event_temp.c faststack.c faststack.h io.c parser.c sexp.c sexp.h sexp_memory.c sexp_memory.h sexp_errors.h sexp_ops.c sexp_ops.h sexp_vis.c sexp_vis.h
-libsexp_la_LDFLAGS = -version-info 1:0:0
+libsexp_la_LDFLAGS = -no-undefined -version-info 1:0:0
--- origsrc/sfsexp-1.4.1/tests/Makefile.am	2023-03-24 16:46:08.000000000 +0900
+++ src/sfsexp-1.4.1/tests/Makefile.am	2023-09-26 03:29:00.859510000 +0900
@@ -1,14 +1,14 @@
 CFLAGS = $(SFSEXP_CFLAGS)
-CPPFLAGS = -I../src $(SFSEXP_CPPFLAGS)
+CPPFLAGS = -I$(top_srcdir)/src $(SFSEXP_CPPFLAGS)
 LDFLAGS =
 EXTRA_DIST = test_expressions dotests.sh randsexp.pl
 
 noinst_PROGRAMS = bug ctest ctorture error_codes partial read_and_dump readtests vis_test
-LDADD = ../src/libsexp.la
-bug_SOURCES = bug.c ../src/sexp.h
-ctest_SOURCES = ctest.c ../src/sexp.h
-ctorture_SOURCES = ctorture.c ../src/sexp.h
-error_codes_SOURCES = error_codes.c ../src/sexp.h
-partial_SOURCES = partial.c ../src/sexp.h
-read_and_dump_SOURCES = read_and_dump.c ../src/sexp.h
-readtests_SOURCES = readtests.c ../src/sexp.h
+LDADD = $(top_builddir)/src/libsexp.la
+bug_SOURCES = bug.c $(top_srcdir)/src/sexp.h
+ctest_SOURCES = ctest.c $(top_srcdir)/src/sexp.h
+ctorture_SOURCES = ctorture.c $(top_srcdir)/src/sexp.h
+error_codes_SOURCES = error_codes.c $(top_srcdir)/src/sexp.h
+partial_SOURCES = partial.c $(top_srcdir)/src/sexp.h
+read_and_dump_SOURCES = read_and_dump.c $(top_srcdir)/src/sexp.h
+readtests_SOURCES = readtests.c $(top_srcdir)/src/sexp.h
