--- origsrc/gpds-1.9.0/CMakeLists.txt	2024-10-03 01:54:23.000000000 +0900
+++ src/gpds-1.9.0/CMakeLists.txt	2025-05-31 16:53:34.351701300 +0900
@@ -76,3 +76,10 @@ set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE
 set(CPACK_RESOURCE_FILE_README "${CMAKE_CURRENT_SOURCE_DIR}/readme.md")
 set(CPACK_INCLUDE_TOPLEVEL_DIRECTORY OFF)
 include(CPack)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/gpds.pc.in ${CMAKE_BINARY_DIR}/gpds.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/gpds.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/gpds-1.9.0/gpds.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/gpds-1.9.0/gpds.pc.in	2025-05-31 16:53:44.422890400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: gpds
+Description: General purpose data serializer
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lgpds
+Cflags: -I${includedir}
--- origsrc/gpds-1.9.0/lib/CMakeLists.txt	2024-10-03 01:54:23.000000000 +0900
+++ src/gpds-1.9.0/lib/CMakeLists.txt	2025-05-31 16:47:59.995209600 +0900
@@ -92,6 +92,10 @@ function(setup_target_common target)
 			VERSION ${PROJECT_VERSION}
 			POSITION_INDEPENDENT_CODE ON
 	)
+
+	if (DEFINED SOVERSION)
+		set_target_properties(${target} PROPERTIES SOVERSION "${SOVERSION}")
+	endif()
 endfunction()
 
 
