--- origsrc/simdjson-0.3.1/CMakeLists.txt	2020-04-03 08:24:43.000000000 +0900
+++ src/simdjson-0.3.1/CMakeLists.txt	2020-04-07 18:21:43.832548200 +0900
@@ -100,4 +100,9 @@ set(CPACK_SOURCE_GENERATOR "TGZ;ZIP")
 
 include(CPack)
 
-
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/simdjson.pc.in ${CMAKE_BINARY_DIR}/simdjson.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/simdjson.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/simdjson-0.3.1/simdjson.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/simdjson-0.3.1/simdjson.pc.in	2020-04-07 18:22:01.611971100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: simdjson
+Description: Parsing gigabytes of JSON per second
+Version: @SIMDJSON_LIB_VERSION@
+Libs: -L${libdir} -lsimdjson
+Cflags: -I${includedir}
