--- origsrc/libfreesrp-0.3.0/CMakeLists.txt	2025-03-16 08:31:35.072769500 +0900
+++ src/libfreesrp-0.3.0/CMakeLists.txt	2025-03-16 08:45:08.518237600 +0900
@@ -60,8 +60,15 @@ add_executable(freesrp-ctl ${FREESRP_CTL
 target_link_libraries(freesrp-ctl freesrp)
 
 # Install library
-install(TARGETS freesrp LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS freesrp RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(FILES ${LIBFREESRP_INCLUDE_FILES} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 # Install freesrp-ctl and freesrp-io utilities
 install(TARGETS freesrp-ctl freesrp-io RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libfreesrp.pc.in ${CMAKE_BINARY_DIR}/libfreesrp.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libfreesrp.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libfreesrp-0.3.0/libfreesrp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libfreesrp-0.3.0/libfreesrp.pc.in	2025-03-16 08:44:09.457760400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libfreesrp
+Description: Small C++ library to interface with the FreeSRP
+Version: @VERSION@
+Libs: -L${libdir} -lfreesrp
+Cflags: -I${includedir}
