--- origsrc/libframeworkd-glib/src/Makefile.am	2013-10-13 21:44:11.000000000 +0900
+++ src/libframeworkd-glib/src/Makefile.am	2013-10-13 21:50:02.547230800 +0900
@@ -12,6 +12,7 @@ BUILT_SOURCES= dialer-marshal.c dialer-m
 lib_LTLIBRARIES = libframeworkd-glib.la
 
 libframeworkd_glib_la_LDFLAGS = $(all_libraries) -version-info 0:0:0 -no-undefined
+libframeworkd_glib_la_LIBADD = $(AM_LDFLAGS)
 
 dialer-marshal.h: dialer-marshal.list
 	( @GLIB_GENMARSHAL@ $< --header > dialer-marshal.tmp \
