--- origsrc/libpillowfight-0.2.4/CMakeLists.txt	2018-04-11 04:06:37.000000000 +0900
+++ src/libpillowfight-0.2.4/CMakeLists.txt	2019-01-18 16:18:15.782821400 +0900
@@ -22,11 +22,21 @@ add_library(pillowfight SHARED
     src/pillowfight/_version.c
     src/pillowfight/util.c
 )
+if (CYGWIN)
+	set_target_properties(pillowfight PROPERTIES SOVERSION 0)
+endif()
 
-install (TARGETS pillowfight DESTINATION ${LIBDIR})
+install (TARGETS pillowfight RUNTIME DESTINATION bin ARCHIVE DESTINATION ${LIBDIR})
 install (FILES
     include/pillowfight/pillowfight.h
     include/pillowfight/util.h
     DESTINATION
     include/pillowfight
 )
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libpillowfight.pc.in ${CMAKE_BINARY_DIR}/libpillowfight.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libpillowfight.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libpillowfight-0.2.4/libpillowfight.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libpillowfight-0.2.4/libpillowfight.pc.in	2019-01-18 16:14:39.760648400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libpillowfight
+Description: Small library containing various image processing algorithms
+Version: @VERSION@
+Libs: -L${libdir} -lpillowfight
+Cflags: -I${includedir}
--- origsrc/libpillowfight-0.2.4/src/pillowfight/_version.h	1970-01-01 09:00:00.000000000 +0900
+++ src/libpillowfight-0.2.4/src/pillowfight/_version.h	2019-01-18 16:12:38.955390500 +0900
@@ -0,0 +1 @@
+#define INTERNAL_PILLOWFIGHT_VERSION "3a7ebba7"
