--- origsrc/onnx-1.18.0/CMakeLists.txt	2025-06-20 01:24:26.302703700 +0900
+++ src/onnx-1.18.0/CMakeLists.txt	2025-06-20 01:25:01.713540800 +0900
@@ -728,7 +728,9 @@ endif()
 
 install(TARGETS
   onnx onnx_proto
-  EXPORT ONNXTargets DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  EXPORT ONNXTargets
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 if(ONNX_BUILD_TESTS)
   include(${ONNX_ROOT}/cmake/unittest.cmake)
