--- origsrc/onnx-1.17.0/CMakeLists.txt	2025-03-22 16:32:38.334682000 +0900
+++ src/onnx-1.17.0/CMakeLists.txt	2025-03-22 17:02:27.802872500 +0900
@@ -748,7 +748,9 @@ endif()
 
 install(TARGETS
   onnx onnx_proto
-  EXPORT ONNXTargets DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  EXPORT ONNXTargets
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 if(ONNX_BUILD_TESTS)
   include(${ONNX_ROOT}/cmake/unittest.cmake)
