--- origsrc/onnx-1.16.0/CMakeLists.txt	2024-03-25 22:55:50.000000000 +0900
+++ src/onnx-1.16.0/CMakeLists.txt	2024-04-12 01:27:18.279445900 +0900
@@ -475,6 +475,9 @@ target_include_directories(onnx_proto PU
   $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
   $<INSTALL_INTERFACE:include>
   $<BUILD_INTERFACE:${PROTOBUF_INCLUDE_DIRS}>)
+if (DEFINED SOVERSION)
+  set_target_properties(onnx_proto PROPERTIES SOVERSION "${SOVERSION}")
+endif()
 
 if (MSVC)
   if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
@@ -534,6 +537,9 @@ target_include_directories(onnx PUBLIC
   $<INSTALL_INTERFACE:include>)
 target_link_libraries(onnx PUBLIC onnx_proto)
 add_onnx_global_defines(onnx)
+if (DEFINED SOVERSION)
+  set_target_properties(onnx PROPERTIES SOVERSION "${SOVERSION}")
+endif()
 
 if(BUILD_ONNX_PYTHON)
   if("${PY_EXT_SUFFIX}" STREQUAL "")
@@ -768,7 +774,10 @@ endif()
 
 install(TARGETS
   onnx onnx_proto
-  EXPORT ONNXTargets DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  EXPORT ONNXTargets 
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
 
 if(ONNX_BUILD_TESTS)
   include(${ONNX_ROOT}/cmake/unittest.cmake)
