--- origsrc/geohash/CMakeLists.txt	1970-01-01 09:00:00.000000000 +0900
+++ src/geohash/CMakeLists.txt	2025-06-06 21:33:49.439332300 +0900
@@ -0,0 +1,27 @@
+cmake_minimum_required(VERSION 3.28)
+project(geohash-nmmmnu
+  VERSION 0.0.0
+  DESCRIPTION "Geohash implementation in C++"
+  HOMEPAGE_URL "https://github.com/nmmmnu/geohash")
+
+include(GNUInstallDirs)
+include(CTest)
+
+add_library(geohash-nmmmnu SHARED geohash.cc)
+set_target_properties(geohash-nmmmnu PROPERTIES SOVERSION 0)
+
+install(TARGETS geohash-nmmmnu
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+add_executable(main main.cc)
+target_link_libraries(main geohash-nmmmnu)
+
+install(FILES geohash.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} RENAME geohash-nmmmnu.h)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/geohash-nmmmnu.pc.in ${CMAKE_BINARY_DIR}/geohash-nmmmnu.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/geohash-nmmmnu.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/geohash/geohash-nmmmnu.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/geohash/geohash-nmmmnu.pc.in	2025-06-06 21:34:59.165458200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: geohash-nmmmnu
+Description: Geohash implementation in C++
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lgeohash-nmmmnu
+Cflags: -I${includedir}
