--- origsrc/xsimd-14.0.0/test/CMakeLists.txt	2026-02-09 10:21:34.262709900 +0900
+++ src/xsimd-14.0.0/test/CMakeLists.txt	2026-02-09 10:21:36.135822200 +0900
@@ -122,7 +122,7 @@ set(XSIMD_TESTS
     test_api.cpp
     test_arch.cpp
     test_basic_math.cpp
-    test_batch.cpp
+    # test_batch.cpp # too many sections on cygwin
     test_batch_bool.cpp
     test_batch_cast.cpp
     test_batch_complex.cpp
@@ -143,7 +143,7 @@ set(XSIMD_TESTS
     test_extract_pair.cpp
     test_fp_manipulation.cpp
     test_hyperbolic.cpp
-    test_load_store.cpp
+    # test_load_store.cpp # too many sections on cygwin
     test_memory.cpp
     test_poly_evaluation.cpp
     test_power.cpp
