--- origsrc/xsimd-11.1.0/include/xsimd/config/xsimd_config.hpp	2023-05-14 00:49:21.000000000 +0900
+++ src/xsimd-11.1.0/include/xsimd/config/xsimd_config.hpp	2023-10-11 13:30:48.822877100 +0900
@@ -13,8 +13,8 @@
 #define XSIMD_CONFIG_HPP
 
 #define XSIMD_VERSION_MAJOR 11
-#define XSIMD_VERSION_MINOR 0
-#define XSIMD_VERSION_PATCH 1
+#define XSIMD_VERSION_MINOR 1
+#define XSIMD_VERSION_PATCH 0
 
 /**
  * high level free functions
