--- origsrc/libgutenfetch-1.2/src/Makefile.am	2004-07-06 04:24:35.000000000 +0900
+++ src/libgutenfetch-1.2/src/Makefile.am	2012-04-24 15:26:42.268673600 +0900
@@ -38,7 +38,7 @@ libgutenfetch_la_SOURCES = 		gutenfetch.
 	rb.c rb.h $(REGEX)
 
 
-libgutenfetch_la_LDFLAGS = @PACKAGE_CFLAGS@
+libgutenfetch_la_LDFLAGS = @PACKAGE_CFLAGS@ -no-undefined
 
 libgutenfetch_la_LIBADD = @PACKAGE_LIBS@ $(INTLLIBS)
 
