--- origsrc/libgutenfetch-1.2/Makefile.am	2004-07-20 17:15:59.000000000 +0900
+++ src/libgutenfetch-1.2/Makefile.am	2025-01-24 17:22:11.988809800 +0900
@@ -3,3 +3,6 @@ SUBDIRS = src doc po data
 ACLOCAL_AMFLAGS = -I m4
 
 EXTRA_DIST = config.rpath
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libgutenfetch-1.2/configure.in	2004-07-20 09:14:53.000000000 +0900
+++ src/libgutenfetch-1.2/configure.in	2025-01-24 17:22:11.988809800 +0900
@@ -172,7 +172,7 @@ dnl ------------------------------------
 dnl - Output our makefiles.
 dnl ---------------------------------------------------------------
 
-AC_CONFIG_FILES([
+AC_CONFIG_FILES([libgutenfetch.pc 
 Makefile
 src/Makefile
 doc/Makefile
--- origsrc/libgutenfetch-1.2/libgutenfetch.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libgutenfetch-1.2/libgutenfetch.pc.in	2025-01-24 17:22:11.988809800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libgutenfetch
+Description: Project Gutenberg Library
+Version: @VERSION@
+Libs: -L${libdir} -lgutenfetch
+Cflags: -I${includedir}
--- origsrc/libgutenfetch-1.2/src/Makefile.am	2004-07-06 04:24:35.000000000 +0900
+++ src/libgutenfetch-1.2/src/Makefile.am	2025-01-24 17:22:11.988809800 +0900
@@ -38,7 +38,7 @@ libgutenfetch_la_SOURCES = 		gutenfetch.
 	rb.c rb.h $(REGEX)
 
 
-libgutenfetch_la_LDFLAGS = @PACKAGE_CFLAGS@
+libgutenfetch_la_LDFLAGS = -no-undefined @PACKAGE_CFLAGS@
 
 libgutenfetch_la_LIBADD = @PACKAGE_LIBS@ $(INTLLIBS)
 
