--- origsrc/jabanner-0.0.9/configure.in	2008-08-23 21:05:07.000000000 +0900
+++ src/jabanner-0.0.9/configure.in	2011-03-14 22:05:15.893558000 +0900
@@ -17,7 +17,7 @@ AC_PROG_CXX
 # Checks for libraries.
 AC_CHECK_LIB(gd,gdImageStringTTF,,AC_MSG_ERROR(Test for libgd failed.))
 if test "$enable_ncurses" != "no"; then
-	AC_CHECK_LIB(ncurses,initscr,,AC_MSG_WARN(Test for libncurses failed.))
+	PKG_CHECK_MODULES(NCURSES, ncurses,, AC_MSG_WARN(Test for libncurses failed.))
 fi
 
 
--- origsrc/jabanner-0.0.9/src/Makefile.am	2008-04-09 11:28:58.000000000 +0900
+++ src/jabanner-0.0.9/src/Makefile.am	2011-03-14 21:51:44.490105100 +0900
@@ -1,3 +1,4 @@
 bin_PROGRAMS = jabanner
 jabanner_SOURCES = jabanner.cc intl.h
+jabanner_LDADD = @LTLIBINTL@
 AM_CPPFLAGS=-DLOCALEDIR=\""$(LOCALEDIR)"\"
