--- origsrc/bedtools2-2.27.1/Makefile	2017-12-15 02:15:02.000000000 +0900
+++ src/bedtools2-2.27.1/Makefile	2018-02-22 20:11:31.862000000 +0900
@@ -18,9 +18,9 @@ export SRC_DIR	= src
 export UTIL_DIR	= src/utils
 export CXX		= g++
 ifeq ($(DEBUG),1)
-export CXXFLAGS = -Wall -Wextra -DDEBUG -D_DEBUG -g -O0 -D_FILE_OFFSET_BITS=64 -fPIC $(INCLUDES)
+export CXXFLAGS = -Wall -Wextra -DDEBUG -D_DEBUG -g -O0 -D_FILE_OFFSET_BITS=64 $(INCLUDES) -D_GNU_SOURCE
 else
-export CXXFLAGS = -Wall -O2 -D_FILE_OFFSET_BITS=64 -fPIC $(INCLUDES)
+export CXXFLAGS = -Wall -O2 -D_FILE_OFFSET_BITS=64 $(INCLUDES) -D_GNU_SOURCE
 endif
 
 # If the user has specified to do so, tell the compile to use rand() (instead of mt19937).
@@ -33,7 +33,7 @@ endif
 export LIBS		= -lz
 export BT_ROOT  = src/utils/BamTools/
 
-prefix ?= /usr/local
+prefix ?= /usr
 
 SUBDIRS = $(SRC_DIR)/annotateBed \
 		  $(SRC_DIR)/bamToBed \
