--- origsrc/wandio-4.2.6-1/Makefile.am	2023-09-19 08:46:06.000000000 +0900
+++ src/wandio-4.2.6-1/Makefile.am	2025-09-05 22:14:25.760068300 +0900
@@ -11,3 +11,6 @@ bootstrap-clean:
 		mkinstalldirs stamp-h.in \
 		lib/Makefile.in replace/libreplace.la  replace/Makefile.in
 	$(RM) -rf autom4te.cache/
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/wandio-4.2.6-1/configure.ac	2023-09-19 08:46:06.000000000 +0900
+++ src/wandio-4.2.6-1/configure.ac	2025-09-05 22:14:25.773767800 +0900
@@ -30,7 +30,7 @@ AC_DEFINE([WANDIO_MID],${WANDIO_MID},[wa
 AC_DEFINE([WANDIO_MINOR],${WANDIO_MINOR},[wandio minor version])
 
 # These are all the files we want to be built for us by configure
-AC_CONFIG_FILES([Makefile lib/Makefile tools/wandiocat/Makefile])
+AC_CONFIG_FILES([wandio.pc Makefile lib/Makefile tools/wandiocat/Makefile])
 
 
 # Function that checks if the C++ compiler actually works - there's a bit of
@@ -81,7 +81,7 @@ AC_CHECK_HEADERS_ONCE([sys/time.h])
 AC_SYS_LARGEFILE
 
 ADD_INCLS=""
-ADD_LDFLAGS="$ADD_LDFLAGS -L\$(abs_top_srcdir)/lib"
+ADD_LDFLAGS="$ADD_LDFLAGS -no-undefined -L\$(abs_top_srcdir)/lib"
 LIBWANDIO_LIBS=""
 
 CFLAGS="$CFLAGS -Wall -Wmissing-prototypes -Wextra -DLT_BUILDING_DLL=1"
--- origsrc/wandio-4.2.6-1/tools/wandiocat/Makefile.am	2023-09-19 08:46:06.000000000 +0900
+++ src/wandio-4.2.6-1/tools/wandiocat/Makefile.am	2025-09-05 22:14:25.777808000 +0900
@@ -2,5 +2,5 @@ bin_PROGRAMS = wandiocat
 wandiocat_SOURCES = wcat.c
 wandiocat_CFLAGS = -I"$(top_srcdir)/lib"
 wandiocat_CXXFLAGS = -I"$(top_srcdir)/lib"
-wandiocat_LDFLAGS = -L"$(top_srcdir)/lib" -lwandio
+wandiocat_LDFLAGS = $(top_builddir)/lib/libwandio.la
 dist_man_MANS = wandiocat.1
--- origsrc/wandio-4.2.6-1/wandio.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/wandio-4.2.6-1/wandio.pc.in	2025-09-05 22:14:25.781839300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: wandio
+Description: Variety of supported compression methods
+Version: @VERSION@
+Libs: -L${libdir} -lwandio
+Cflags: -I${includedir}
