--- origsrc/libuv-1.13.1/configure.ac	2017-07-07 07:42:36.000000000 +0900
+++ src/libuv-1.13.1/configure.ac	2017-07-15 10:28:46.482000200 +0900
@@ -15,10 +15,9 @@
 AC_PREREQ(2.57)
 AC_INIT([libuv], [1.13.1], [https://github.com/libuv/libuv/issues])
 AC_CONFIG_MACRO_DIR([m4])
-m4_include([m4/libuv-extra-automake-flags.m4])
 m4_include([m4/as_case.m4])
 m4_include([m4/libuv-check-flags.m4])
-AM_INIT_AUTOMAKE([-Wall -Werror foreign subdir-objects] UV_EXTRA_AUTOMAKE_FLAGS)
+AM_INIT_AUTOMAKE([-Wall -Werror foreign subdir-objects])
 AC_CANONICAL_HOST
 AC_ENABLE_SHARED
 AC_ENABLE_STATIC
