--- origsrc/libuv-1.15.0/configure.ac	2017-10-03 08:30:27.000000000 +0900
+++ src/libuv-1.15.0/configure.ac	2017-10-09 14:08:18.057217100 +0900
@@ -15,10 +15,9 @@
 AC_PREREQ(2.57)
 AC_INIT([libuv], [1.15.0], [https://github.com/libuv/libuv/issues])
 AC_CONFIG_MACRO_DIR([m4])
-m4_include([m4/libuv-extra-automake-flags.m4])
 m4_include([m4/as_case.m4])
 m4_include([m4/libuv-check-flags.m4])
-AM_INIT_AUTOMAKE([-Wall -Werror foreign subdir-objects] UV_EXTRA_AUTOMAKE_FLAGS)
+AM_INIT_AUTOMAKE([-Wall -Werror foreign subdir-objects])
 AC_CANONICAL_HOST
 AC_ENABLE_SHARED
 AC_ENABLE_STATIC
--- origsrc/libuv-1.15.0/src/unix/bsd-ifaddrs.c	2017-10-03 08:30:27.000000000 +0900
+++ src/libuv-1.15.0/src/unix/bsd-ifaddrs.c	2017-10-09 14:08:18.073051300 +0900
@@ -36,6 +36,7 @@ static int uv__ifaddr_exclude(struct ifa
     return 1;
   if (ent->ifa_addr == NULL)
     return 1;
+#if defined(AF_LINK)
   /*
    * If `exclude_type` is `UV__EXCLUDE_IFPHYS`, just see whether `sa_family`
    * equals to `AF_LINK` or not. Otherwise, the result depends on the operation
@@ -43,6 +44,7 @@ static int uv__ifaddr_exclude(struct ifa
    */
   if (exclude_type == UV__EXCLUDE_IFPHYS)
     return (ent->ifa_addr->sa_family != AF_LINK);
+#endif
 #if defined(__APPLE__) || defined(__FreeBSD__) || defined(__DragonFly__)
   /*
    * On BSD getifaddrs returns information related to the raw underlying
