--- origsrc/zxcc-0.5.6/Makefile.am	2010-07-14 09:06:51.000000000 +0900
+++ src/zxcc-0.5.6/Makefile.am	2011-09-25 17:55:40.434643100 +0900
@@ -1,10 +1,8 @@
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS= . @LSUBDIRS@ Z80 bin 
+SUBDIRS= . Z80 bin 
 EXTRA_DIST= $(srcdir)/config/* $(srcdir)/zxcc.doc $(srcdir)/zxcc.html
 pkgdata_DATA = zxcc.doc
 
-ACLOCAL_AMFLAGS=-I m4
-
 zxcc.doc:	$(srcdir)/zxcc.html
 	lynx -dump $(srcdir)/zxcc.html > zxcc.doc
--- origsrc/zxcc-0.5.6/configure.in	2010-12-31 08:27:03.000000000 +0900
+++ src/zxcc-0.5.6/configure.in	2011-09-25 17:54:42.499279900 +0900
@@ -1,29 +1,14 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(bin/zxcc.c)
+AC_INIT([zxcc], [0.5.6])
 AC_CONFIG_AUX_DIR(config)
 AC_CONFIG_MACRO_DIR([m4])
 VERSION=0.5.6
 UPDATED='July 14, 2010'
-AM_INIT_AUTOMAKE(zxcc, $VERSION)
+AM_INIT_AUTOMAKE
 AM_CONFIG_HEADER(config.h)
 LT_INIT
 
-dnl Get an absolute version of $srcdir and .
-
-LSUBDIRS=""
-DSTROOT=`pwd`
-if test x"$srcdir" = x
-then
-	SRCROOT=`pwd`
-else
-	cd $srcdir
-	SRCROOT=`pwd`
-	cd $DSTROOT
-fi
-
-
 AC_ARG_WITH(readline,  [  --with-readline         Use GNU Readline for line input (no)])
-AC_ARG_WITH(cpmio,     [  --with-cpmio            Use CPMIO library for output (yes)])
 
 dnl Checks for programs.
 AC_PROG_CC
@@ -34,21 +19,9 @@ then
 	AC_CHECK_LIB(readline, readline)
 fi
 
-dnl If using CPMIO, add it to libs. Can't use AC_CHECK_LIB because
-dnl CPMIO hasn't been built yet. 
-if test x"$with_cpmio" != xno 
-then
-	AC_CHECK_LIB(ncurses, printw)
-	AC_CHECK_LIB(curses,  printw)
-	CPPFLAGS="$CPPFLAGS -I$SRCROOT/cpmio/include -DUSE_CPMIO"
-	LDFLAGS="$LDFLAGS -L$DSTROOT/cpmio/lib"
-	LSUBDIRS="$LSUBDIRS cpmio"
-fi
-
-dnl CPMREDIR isn't conditional, so just add it
-CPPFLAGS="$CPPFLAGS -I$SRCROOT/cpmredir/include"
-LDFLAGS="$LDFLAGS -L$DSTROOT/cpmredir/lib"
-LSUBDIRS="$LSUBDIRS cpmredir"
+PKG_CHECK_MODULES(NCURSES, ncurses)
+AC_CHECK_LIB(cpmio, cpm_scr_init, [CPP_FLAGS="$CPP_FLAGS -DUSE_CPMIO"])
+AC_CHECK_LIB(cpmredir, fcb_init)
 
 dnl General preprocessor symbols
 CPPFLAGS="$CPPFLAGS -DZXCC"
@@ -71,23 +44,11 @@ dnl Checks for library functions.
 AC_FUNC_VPRINTF
 AC_CHECK_FUNCS(strstr)
 
-
-AC_SUBST(LSUBDIRS)
-AC_CONFIG_SUBDIRS($LSUBDIRS)
-
 dnl Location of the ZXCC library directories
 dnl CPPFLAGS="$CPPFLAGS -DLIBROOT=\"$libdir/cpm/\""
 CPPFLAGS="$CPPFLAGS -DBINDIR80=\\\"$libdir/cpm/bin80/\\\""
 CPPFLAGS="$CPPFLAGS -DLIBDIR80=\\\"$libdir/cpm/lib80/\\\""
 CPPFLAGS="$CPPFLAGS -DINCDIR80=\\\"$libdir/cpm/include80/\\\""
 
-dnl Do this last, because adding "-lcpmio" to libraries when there isn't
-dnl in fact a CPMIO library built yet will make other tests go funny.
-if test x"$with_cpmio" != xno 
-then
-	LIBS="-lcpmio $LIBS"
-fi
-LIBS="-lcpmredir $LIBS"
-
 AC_OUTPUT(Makefile bin/Makefile Z80/Makefile )
 
