--- origsrc/tinynotify-send-1.2.1/Makefile.am	2012-01-03 15:46:55.000000000 +0900
+++ src/tinynotify-send-1.2.1/Makefile.am	2012-01-15 21:01:45.561170800 +0900
@@ -12,7 +12,7 @@ lib_LTLIBRARIES = libtinynotify-cli.la
 pkgconfig_DATA = libtinynotify-cli.pc
 include_HEADERS = lib/tinynotify-cli.h
 
-libtinynotify_cli_la_LDFLAGS = -version-info 3:1:2
+libtinynotify_cli_la_LDFLAGS = -version-info 3:1:2 -no-undefined
 libtinynotify_cli_la_CPPFLAGS = $(LIBTINYNOTIFY_CFLAGS)
 libtinynotify_cli_la_LIBADD = $(LIBTINYNOTIFY_LIBS)
 libtinynotify_cli_la_SOURCES = lib/tinynotify-cli.c lib/tinynotify-cli.h
