--- origsrc/libtexprintf-1.25/Makefile.am	2023-07-20 05:42:22.000000000 +0900
+++ src/libtexprintf-1.25/Makefile.am	2023-11-17 23:53:44.567515400 +0900
@@ -8,3 +8,6 @@ TESTS=testsuite.sh testeqs.sh testfonts.
 EXTRA_DIST = testsuite.sh testeqs.sh testfonts.sh testerrors.sh testtexsymbols.sh test/testsuite.txt test/testeqs.txt test/testfonts.txt test/test.awk test/README
 
 DISTCLEANFILES = configure~ src/config.h.in~
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libtexprintf-1.25/configure.ac	2023-07-20 05:42:22.000000000 +0900
+++ src/libtexprintf-1.25/configure.ac	2023-11-17 23:53:44.567515400 +0900
@@ -28,4 +28,5 @@ AC_CHECK_LIB([m],[cos])
 AC_CONFIG_FILES([
  Makefile src/Makefile
 ])
+AC_CONFIG_FILES([libtexprintf.pc])
 AC_OUTPUT
--- origsrc/libtexprintf-1.25/libtexprintf.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libtexprintf-1.25/libtexprintf.pc.in	2023-11-17 23:53:44.489394600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libtexprintf
+Description: Formatted Output with tex-like syntax support
+Version: @VERSION@
+Libs: -L${libdir} -ltexprintf
+Cflags: -I${includedir}
--- origsrc/libtexprintf-1.25/src/Makefile.am	2023-07-20 05:42:22.000000000 +0900
+++ src/libtexprintf-1.25/src/Makefile.am	2023-11-17 23:51:44.501758300 +0900
@@ -7,7 +7,7 @@ libtexprintf_la_SOURCES = boxes.c  drawb
 nodist_libtexprintf_la_SOURCES = errorflags.h errormessages.h texprintfsymbols
 CLEANFILES = errorflags.h errormessages.h texprintfsymbols
 libtexprintf_la_LIBADD = libstringutils.la
-libtexprintf_la_LDFLAGS = -export-symbols ${srcdir}/texprintfsymbols -version-info 1:0:0
+libtexprintf_la_LDFLAGS = -no-undefined -export-symbols ${builddir}/texprintfsymbols -version-info 1:0:0
 man3_MANS = texprintf.3
 include_HEADERS = texprintf.h
 
@@ -37,4 +37,5 @@ errormessages.h: errorflags.h
 	@test -f $@ || $(MAKE) $(AM_MAKEFLAGS) errorflags.h
 
 texprintfsymbols: gentexprintfsymbols.sh texprintf.h
-	${SHELL} ${srcdir}/gentexprintfsymbols.sh
+	grep 'extern' ${srcdir}/texprintf.h |grep -o -E '[^ ;]+;'|grep -o -E '[^;]+' >texprintfsymbols
+	grep -o -E '[^ \*\(]+\(' ${srcdir}/texprintf.h |grep -o -E '^[^\(]+' >>texprintfsymbols
--- origsrc/libtexprintf-1.25/src/drawbox.c	2023-07-20 05:42:22.000000000 +0900
+++ src/libtexprintf-1.25/src/drawbox.c	2023-11-17 23:50:30.993868100 +0900
@@ -85,7 +85,7 @@ char * DrawBox(box *b)
 							Na+=strlen(str)+2;
 							buffer=realloc(buffer, Na*sizeof(char));
 						}
-						sprintf(buffer+i, str);
+						sprintf(buffer+i, "%s", str);
 						i+=strlen(str);
 						free(str);
 					}
--- origsrc/libtexprintf-1.25/src/testfonts.c	2023-07-20 05:42:22.000000000 +0900
+++ src/libtexprintf-1.25/src/testfonts.c	2023-11-17 23:46:51.690701900 +0900
@@ -149,7 +149,7 @@ void MakeBlockString(int start, int end)
 		s+=k;
 		free(c);
 	}
-	printf(str);
+	printf("%s", str);
 	free(str);
 }
 
@@ -162,11 +162,11 @@ void TestFonts()
 	char *scale="|---------+---------+---------+---------+---------+---------+\n";
 	char *str;
 	j=0;
-	printf(col);
-	printf(scale);
+	printf("%s", col);
+	printf("%s", scale);
 	while (font_tests[j].str)
 	{
-		printf(font_tests[j].descr);
+		printf("%s", font_tests[j].descr);
 		str=AlignChars(j++);
 		root=ParseString(str, 0, styles[i]);
 		free(str);
@@ -174,8 +174,8 @@ void TestFonts()
 		PrintBox(&root);
 		FreeBox(&root);
 	}
-	printf(scale);
-	printf(col);
+	printf("%s", scale);
+	printf("%s", col);
 	printf("\n");
 	i++;
 	
