--- origsrc/libtexprintf-1.28/Makefile.am	2026-01-05 17:27:10.000000000 +0900
+++ src/libtexprintf-1.28/Makefile.am	2026-01-21 10:36:11.673245500 +0900
@@ -8,3 +8,6 @@ TESTS=testsuite.sh testeqs.sh testfonts.
 EXTRA_DIST = testsuite.sh testeqs.sh testfonts.sh testerrors.sh testtexsymbols.sh test/testsuite.txt test/testeqs.txt test/testfonts.txt test/test.awk test/README
 
 DISTCLEANFILES = configure~ src/config.h.in~
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libtexprintf-1.28/configure.ac	2026-01-05 17:27:10.000000000 +0900
+++ src/libtexprintf-1.28/configure.ac	2026-01-21 10:36:11.685715500 +0900
@@ -28,4 +28,5 @@ AC_CHECK_LIB([m],[cos])
 AC_CONFIG_FILES([
  Makefile src/Makefile
 ])
+AC_CONFIG_FILES([libtexprintf.pc])
 AC_OUTPUT
--- origsrc/libtexprintf-1.28/libtexprintf.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libtexprintf-1.28/libtexprintf.pc.in	2026-01-21 10:36:11.599552700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libtexprintf
+Description: Formatted Output with tex-like syntax support
+Version: @VERSION@
+Libs: -L${libdir} -ltexprintf
+Cflags: -I${includedir}
--- origsrc/libtexprintf-1.28/src/Makefile.am	2026-01-05 17:27:10.000000000 +0900
+++ src/libtexprintf-1.28/src/Makefile.am	2026-01-21 10:06:35.864914800 +0900
@@ -7,7 +7,7 @@ libtexprintf_la_SOURCES = boxes.c  drawb
 nodist_libtexprintf_la_SOURCES = errorflags.h errormessages.h texprintfsymbols
 CLEANFILES = errorflags.h errormessages.h texprintfsymbols
 libtexprintf_la_LIBADD = libstringutils.la
-libtexprintf_la_LDFLAGS = -export-symbols ${srcdir}/texprintfsymbols -version-info 1:0:0
+libtexprintf_la_LDFLAGS = -export-symbols ${builddir}/texprintfsymbols -version-info 1:0:0
 man3_MANS = texprintf.3
 include_HEADERS = texprintf.h
 
@@ -37,4 +37,4 @@ errormessages.h: errorflags.h
 	@test -f $@ || $(MAKE) $(AM_MAKEFLAGS) errorflags.h
 
 texprintfsymbols: gentexprintfsymbols.sh texprintf.h
-	${SHELL} ${srcdir}/gentexprintfsymbols.sh
+	srcdir=${srcdir} ${SHELL} ${srcdir}/gentexprintfsymbols.sh
--- origsrc/libtexprintf-1.28/src/gentexprintfsymbols.sh	2026-01-05 17:27:10.000000000 +0900
+++ src/libtexprintf-1.28/src/gentexprintfsymbols.sh	2026-01-21 10:05:38.475340000 +0900
@@ -3,5 +3,5 @@
 # parses texprintf.h to collect all relavant symbols
 
 # get global variables
-grep 'extern' texprintf.h |grep -o -E '[^ ;]+;'|grep -o -E '[^;]+' >texprintfsymbols
-grep -o -E '[^ \*\(]+\(' texprintf.h |grep -o -E '^[^\(]+' >>texprintfsymbols
+grep 'extern' ${srcdir}/texprintf.h |grep -o -E '[^ ;]+;'|grep -o -E '[^;]+' >texprintfsymbols
+grep -o -E '[^ \*\(]+\(' ${srcdir}/texprintf.h |grep -o -E '^[^\(]+' >>texprintfsymbols
