--- origsrc/x52lib/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/x52lib/Makefile.am	2024-12-10 01:27:33.519133700 +0900
@@ -0,0 +1,17 @@
+
+bin_PROGRAMS = x52output
+x52output_SOURCES = x52output.c
+x52output_LDADD = libx52pro.la
+
+include_HEADERS = x52pro.h
+
+lib_LTLIBRARIES = libx52pro.la
+libx52pro_la_CPPFLAGS = -D_REENTRANT
+libx52pro_la_SOURCES = x52pro.c
+libx52pro_la_LDFLAGS = -no-undefined
+libx52pro_la_LIBADD = $(USB_LIBS)
+
+man_MANS = x52output.1
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/x52lib/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/x52lib/configure.ac	2024-12-10 01:27:33.519133700 +0900
@@ -0,0 +1,29 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([x52pro], [0.1.2])
+AC_CONFIG_SRCDIR([x52output.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_LN_S
+LT_INIT
+
+# Checks for libraries.
+PKG_CHECK_MODULES([USB], [libusb])
+
+# Checks for header files.
+AC_CHECK_HEADERS([stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([memset strtol])
+
+AC_CONFIG_FILES([x52pro.pc Makefile])
+AC_OUTPUT
--- origsrc/x52lib/x52pro.c	2024-12-10 01:27:32.000000000 +0900
+++ src/x52lib/x52pro.c	2024-12-10 01:30:44.889626400 +0900
@@ -275,8 +275,7 @@ enum x52_type x52_gettype(struct x52* hd
 
 void x52_close(struct x52* x52) 
 {
-	int r;
-	r = usb_close(x52->hdl);
+	usb_close(x52->hdl);
 	free(x52);
 }
 
--- origsrc/x52lib/x52pro.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/x52lib/x52pro.pc.in	2024-12-10 01:27:33.519133700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: x52pro
+Description: x52 PRO linux library
+Version: @VERSION@
+Libs: -L${libdir} -lx52pro
+Cflags: -I${includedir}
