--- origsrc/matrixssl-3-2-2-open/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/matrixssl-3-2-2-open/Makefile.am	2011-10-19 00:41:12.493154500 +0900
@@ -0,0 +1,82 @@
+
+AM_CPPFLAGS = -DPOSIX
+
+lib_LTLIBRARIES = libmatrixssl.la
+libmatrixssl_la_SOURCES = \
+	core/corelib.c \
+	core/POSIX/osdep.c \
+	crypto/keyformat/asn1.c \
+	crypto/keyformat/base64.c \
+	crypto/keyformat/x509.c \
+	crypto/digest/md5.c \
+	crypto/digest/sha1.c \
+	crypto/digest/hmac.c \
+	crypto/math/pstm.c \
+	crypto/pubkey/pubkey.c \
+	crypto/pubkey/pkcs.c \
+	crypto/pubkey/rsa.c \
+	crypto/prng/prng.c \
+	crypto/prng/yarrow.c \
+	crypto/symmetric/aes.c \
+	crypto/symmetric/arc4.c \
+	crypto/symmetric/des3.c \
+	crypto/symmetric/rc2.c \
+	matrixssl/cipherSuite.c \
+	matrixssl/matrixssl.c \
+	matrixssl/matrixsslApi.c \
+	matrixssl/sslDecode.c \
+	matrixssl/sslEncode.c \
+	matrixssl/tls.c \
+	matrixssl/sslv3.c
+libmatrixssl_la_LDFLAGS = -no-undefined
+
+matrixssl_includedir = $(includedir)/matrixssl
+matrixssl_include_HEADERS = \
+	matrixssl/matrixsslApi.h \
+	matrixssl/matrixsslConfig.h \
+	matrixssl/matrixssllib.h \
+	matrixssl/version.h
+
+matrixssl_core_includedir = $(includedir)/matrixssl/core
+matrixssl_core_include_HEADERS = \
+	core/coreApi.h \
+	core/coreConfig.h \
+	core/list.h \
+	core/osdep.h \
+	core/psmalloc.h
+
+matrixssl_crypto_includedir = $(includedir)/matrixssl/crypto
+matrixssl_crypto_include_HEADERS = \
+	crypto/cryptoApi.h \
+	crypto/cryptoConfig.h \
+	crypto/cryptolib.h
+
+matrixssl_crypto_digest_includedir = $(includedir)/matrixssl/crypto/digest
+matrixssl_crypto_digest_include_HEADERS = \
+	crypto/digest/digest.h
+
+matrixssl_crypto_keyformat_includedir = $(includedir)/matrixssl/crypto/keyformat
+matrixssl_crypto_keyformat_include_HEADERS = \
+	crypto/keyformat/asn1.h \
+	crypto/keyformat/x509.h
+
+matrixssl_crypto_math_includedir = $(includedir)/matrixssl/crypto/math
+matrixssl_crypto_math_include_HEADERS = \
+	crypto/math/pstm.h
+
+matrixssl_crypto_prng_includedir = $(includedir)/matrixssl/crypto/prng
+matrixssl_crypto_prng_include_HEADERS = \
+	crypto/prng/prng.h
+
+matrixssl_crypto_pubkey_includedir = $(includedir)/matrixssl/crypto/pubkey
+matrixssl_crypto_pubkey_include_HEADERS = \
+	crypto/pubkey/pubkey.h
+
+matrixssl_crypto_symmetric_includedir = $(includedir)/matrixssl/crypto/symmetric
+matrixssl_crypto_symmetric_include_HEADERS = \
+	crypto/symmetric/symmetric.h
+
+TESTS = matrixssl/test/sslTest
+check_PROGRAMS = matrixssl/test/sslTest
+matrixssl_test_sslTest_SOURCES = matrixssl/test/sslTest.c
+matrixssl_test_sslTest_LDADD = libmatrixssl.la
--- origsrc/matrixssl-3-2-2-open/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/matrixssl-3-2-2-open/configure.ac	2011-10-19 00:47:19.490581200 +0900
@@ -0,0 +1,35 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([matrixssl], [3.2.2], [support@matrixssl.org])
+AC_CONFIG_SRCDIR([apps/app.h])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT([disable-static])
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([arpa/inet.h fcntl.h netdb.h stdint.h stdlib.h string.h sys/socket.h sys/time.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_C_INLINE
+AC_TYPE_INT16_T
+AC_TYPE_INT32_T
+AC_TYPE_INT64_T
+AC_TYPE_SIZE_T
+AC_TYPE_UINT16_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([gettimeofday localtime_r memmove memset select socket strstr])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
