--- origsrc/jpeginfo-1.7.1/configure.in	2023-10-30 03:59:53.000000000 +0900
+++ src/jpeginfo-1.7.1/configure.in	2023-11-19 15:16:05.695959200 +0900
@@ -8,8 +8,6 @@ AC_CONFIG_AUX_DIR(tools)
 
 
 AC_CANONICAL_HOST
-HOST_TYPE=$host
-AC_DEFINE_UNQUOTED(HOST_TYPE,"$host")
 
 dnl Checks for programs.
 AC_PROG_CC
@@ -60,23 +58,6 @@ AC_CHECK_FUNCS(getopt_long, break, [GNUG
 AC_SUBST(GNUGETOPT)
 
 
-dnl own tests
-
-AC_MSG_CHECKING([for broken jmorecfg.h (METHODDEF)])
-AC_CACHE_VAL(jpeginfo_cv_broken_methoddef,
-[AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <stdio.h>
-#include <jpeglib.h>
-METHODDEF(void) foo(void) { };
-int main(void)
-{
-  return 0;
-}]])],[jpeginfo_cv_broken_methoddef=no],[jpeginfo_cv_broken_methoddef=yes],[jpeginfo_cv_broken_methoddef=no])])
-if test $jpeginfo_cv_broken_methoddef = yes; then
-  AC_DEFINE(BROKEN_METHODDEF)
-fi
-AC_MSG_RESULT($jpeginfo_cv_broken_methoddef)
-
 
 AC_CONFIG_FILES([Makefile])
 AC_OUTPUT
