--- origsrc/libvips-8.10.1/Makefile.am	2020-09-04 20:41:19.000000000 +0900
+++ src/libvips-8.10.1/Makefile.am	2020-09-27 03:24:09.550926500 +0900
@@ -2,7 +2,6 @@ SUBDIRS = \
 	libvips \
 	cplusplus \
 	tools \
-	po \
 	man \
 	doc \
 	test \
--- origsrc/libvips-8.10.1/fuzz/test_fuzz.sh	2020-09-04 20:41:19.000000000 +0900
+++ src/libvips-8.10.1/fuzz/test_fuzz.sh	2020-09-27 03:24:09.565340200 +0900
@@ -13,7 +13,11 @@ export VIPS_WARNING=0
 
 ret=0
 
-for fuzzer in *_fuzzer; do
+if [[ `uname` =~ ^CYGWIN ]]; then
+  EXEEXT=.exe
+fi
+
+for fuzzer in *_fuzzer${EXEEXT}; do
   for file in common_fuzzer_corpus/*; do
     if ! ./$fuzzer $file; then
       echo FAIL $fuzzer $file
