--- origsrc/libvips-8.10.5/Makefile.am	2020-12-18 19:12:26.000000000 +0900
+++ src/libvips-8.10.5/Makefile.am	2021-01-12 19:28:38.555254500 +0900
@@ -2,7 +2,6 @@ SUBDIRS = \
 	libvips \
 	cplusplus \
 	tools \
-	po \
 	man \
 	doc \
 	test \
--- origsrc/libvips-8.10.5/fuzz/test_fuzz.sh	2020-12-18 19:12:26.000000000 +0900
+++ src/libvips-8.10.5/fuzz/test_fuzz.sh	2021-01-12 19:28:38.565267400 +0900
@@ -13,7 +13,11 @@ export VIPS_WARNING=0
 
 ret=0
 
-for fuzzer in *_fuzzer; do
+if [[ `uname` =~ ^CYGWIN ]]; then
+  EXEEXT=.exe
+fi
+
+for fuzzer in *_fuzzer${EXEEXT}; do
   for file in common_fuzzer_corpus/*; do
     if ! ./$fuzzer $file; then
       echo FAIL $fuzzer $file
