--- origsrc/libvips-8.15.4/meson.build	2024-10-03 21:44:10.000000000 +0900
+++ src/libvips-8.15.4/meson.build	2024-10-17 13:03:40.377612500 +0900
@@ -268,8 +268,13 @@ endif
 
 # quant package we use
 quantisation_package = disabler()
+quantisation_package_name = 'none'
 
-imagequant_dep = dependency('imagequant', required: get_option('imagequant'))
+imagequant_dep = dependency('imagequant', required: false)
+if not imagequant_dep.found()
+    imagequant_dep = cc.find_library('imagequant', required: get_option('imagequant'))
+    quantisation_package_name = 'imagequant'
+endif
 if imagequant_dep.found()
     libvips_deps += imagequant_dep
     cfg_var.set('HAVE_IMAGEQUANT', '1')
@@ -302,6 +307,10 @@ if quantisation_package.found()
     endif
 endif
 
+if quantisation_package.found() and quantisation_package_name == 'none'
+    quantisation_package_name = quantisation_package.name()
+endif
+
 libexif_dep = dependency('libexif', version: '>=0.6', required: get_option('exif'))
 if libexif_dep.found()
     libvips_deps += libexif_dep
@@ -729,7 +738,7 @@ build_summary = {
      'JPEG2000 load/save with OpenJPEG': [libopenjp2_dep.found()],
      'PNG load/save with libspng': [spng_found],
      'PNG load/save with libpng': [png_dep.found()],
-     'selected quantisation package': [quantisation_package.found() ? quantisation_package.name() : 'none'],
+     'selected quantisation package': [quantisation_package.found() ? quantisation_package_name : 'none'],
      'TIFF load/save with libtiff': [libtiff_dep.found()],
      'image pyramid save with libarchive': [libarchive_dep.found()],
      'HEIC/AVIF load/save with libheif': [libheif_dep.found(), ' (dynamic module: ', libheif_module, ')'],
