--- origsrc/libvips-8.11.4/Makefile.am	2021-09-13 22:40:20.000000000 +0900
+++ src/libvips-8.11.4/Makefile.am	2021-11-08 21:45:53.854082800 +0900
@@ -2,7 +2,6 @@ SUBDIRS = \
 	libvips \
 	cplusplus \
 	tools \
-	po \
 	man \
 	doc \
 	test \
--- origsrc/libvips-8.11.4/fuzz/test_fuzz.sh	2021-09-13 22:40:20.000000000 +0900
+++ src/libvips-8.11.4/fuzz/test_fuzz.sh	2021-11-08 21:45:53.863080000 +0900
@@ -13,7 +13,11 @@ export VIPS_WARNING=0
 
 ret=0
 
-for fuzzer in *_fuzzer; do
+if [[ `uname` =~ ^CYGWIN ]]; then
+  EXEEXT=.exe
+fi
+
+for fuzzer in *_fuzzer${EXEEXT}; do
   for file in common_fuzzer_corpus/*; do
     if ! ./$fuzzer $file; then
       echo FAIL $fuzzer $file
--- origsrc/libvips-8.11.4/libvips/foreign/fits.c	2021-09-13 22:40:20.000000000 +0900
+++ src/libvips-8.11.4/libvips/foreign/fits.c	2021-11-08 21:45:53.892047500 +0900
@@ -81,7 +81,7 @@
 #include <vips/internal.h>
 #include <vips/debug.h>
 
-#include <fitsio.h>
+#include <cfitsio/fitsio.h>
 
 #include "pforeign.h"
 
