--- origsrc/libvips/Makefile.am	2013-10-11 23:40:40.000000000 +0900
+++ src/libvips/Makefile.am	2013-10-12 01:50:05.266127400 +0900
@@ -22,7 +22,6 @@ endif
 SUBDIRS = \
 	libvips \
 	tools \
-	po \
 	man \
 	doc \
 	$(C_COMPILE_DIR) \
--- origsrc/libvips/configure.ac	2013-10-11 23:40:40.000000000 +0900
+++ src/libvips/configure.ac	2013-10-11 23:46:43.229295600 +0900
@@ -334,7 +334,7 @@ PKG_CHECK_MODULES(THREADS, glib-2.0 >= 2
 # after 2.36 the type system inits itself
 PKG_CHECK_MODULES(TYPE_INIT, glib-2.0 < 2.36,[
   AC_DEFINE(NEED_TYPE_INIT,1,[define if your glib needs g_type_init().])
-  ],[
+  ],[:
   ]
 )
 
--- origsrc/libvips/libvips/Makefile.am	2013-10-11 23:40:40.000000000 +0900
+++ src/libvips/libvips/Makefile.am	2013-10-12 01:44:30.116713400 +0900
@@ -88,7 +88,7 @@ introspection_sources = @vips_introspect
 # we make the vips8 API
 Vips-8.0.gir: libvips.la
 Vips_8_0_gir_INCLUDES = GObject-2.0
-Vips_8_0_gir_CFLAGS = $(INCLUDES) -I${top_srcdir}/libvips/include
+Vips_8_0_gir_CFLAGS = $(INCLUDES) -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include
 Vips_8_0_gir_LIBS = libvips.la
 Vips_8_0_gir_FILES = $(introspection_sources)
 INTROSPECTION_GIRS += Vips-8.0.gir
--- origsrc/libvips/libvips/arithmetic/Makefile.am	2013-10-11 23:40:40.000000000 +0900
+++ src/libvips/libvips/arithmetic/Makefile.am	2013-10-12 00:14:48.775630200 +0900
@@ -38,4 +38,4 @@ libarithmetic_la_SOURCES = \
 	math2.c \
 	round.c 
 
-AM_CPPFLAGS = -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
+AM_CPPFLAGS = -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
--- origsrc/libvips/libvips/cimg/Makefile.am	2013-10-11 23:40:40.000000000 +0900
+++ src/libvips/libvips/cimg/Makefile.am	2013-10-12 01:31:38.129408700 +0900
@@ -24,4 +24,4 @@ endif
 # used by the final libvips link rather than us
 EXTRA_DIST = dummy2.cc
 
-AM_CPPFLAGS += -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
+AM_CPPFLAGS += -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
--- origsrc/libvips/libvips/colour/Makefile.am	2013-10-11 23:40:40.000000000 +0900
+++ src/libvips/libvips/colour/Makefile.am	2013-10-12 01:24:42.450394900 +0900
@@ -30,4 +30,4 @@ libcolour_la_SOURCES = \
 	XYZ2scRGB.c \
 	scRGB2sRGB.c 
 
-AM_CPPFLAGS = -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
+AM_CPPFLAGS = -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
--- origsrc/libvips/libvips/conversion/Makefile.am	2013-10-11 23:40:40.000000000 +0900
+++ src/libvips/libvips/conversion/Makefile.am	2013-10-12 01:25:49.080580600 +0900
@@ -32,4 +32,4 @@ libconversion_la_SOURCES = \
 	subsample.c \
 	zoom.c
 
-AM_CPPFLAGS = -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@
+AM_CPPFLAGS = -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@
--- origsrc/libvips/libvips/convolution/Makefile.am	2013-10-11 23:40:40.000000000 +0900
+++ src/libvips/libvips/convolution/Makefile.am	2013-10-12 01:30:30.830580600 +0900
@@ -17,4 +17,4 @@ libconvolution_la_SOURCES = \
 	im_sharpen.c \
 	im_spcor.c 
 
-AM_CPPFLAGS = -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
+AM_CPPFLAGS = -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
--- origsrc/libvips/libvips/create/Makefile.am	2013-10-11 23:40:40.000000000 +0900
+++ src/libvips/libvips/create/Makefile.am	2013-10-12 01:40:30.302260300 +0900
@@ -20,4 +20,4 @@ libcreate_la_SOURCES = \
 	zone.c \
 	other_dispatch.c
 
-AM_CPPFLAGS = -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
+AM_CPPFLAGS = -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
--- origsrc/libvips/libvips/deprecated/Makefile.am	2013-10-11 23:40:40.000000000 +0900
+++ src/libvips/libvips/deprecated/Makefile.am	2013-10-12 01:27:23.787611800 +0900
@@ -76,4 +76,4 @@ libdeprecated_la_SOURCES = \
 	radiance.c \
 	raw.c
 
-AM_CPPFLAGS = -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
+AM_CPPFLAGS = -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
--- origsrc/libvips/libvips/foreign/Makefile.am	2013-10-11 23:40:40.000000000 +0900
+++ src/libvips/libvips/foreign/Makefile.am	2013-10-12 00:03:47.687739600 +0900
@@ -64,5 +64,5 @@ libforeign_la_SOURCES = \
 
 EXTRA_DIST = 
 
-AM_CPPFLAGS = -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@
+AM_CPPFLAGS = -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@
 
--- origsrc/libvips/libvips/freq_filt/Makefile.am	2013-10-11 23:40:41.000000000 +0900
+++ src/libvips/libvips/freq_filt/Makefile.am	2013-10-12 01:33:03.325697800 +0900
@@ -13,4 +13,4 @@ libfreq_filt_la_SOURCES = \
 	im_invfft.c \
 	im_invfftr.c 
 
-AM_CPPFLAGS = -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
+AM_CPPFLAGS = -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
--- origsrc/libvips/libvips/histogram/Makefile.am	2013-10-11 23:40:41.000000000 +0900
+++ src/libvips/libvips/histogram/Makefile.am	2013-10-12 01:34:20.772963400 +0900
@@ -16,4 +16,4 @@ libhistogram_la_SOURCES = \
 	hist_ismonotonic.c \
 	stdif.c 
 
-AM_CPPFLAGS = -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
+AM_CPPFLAGS = -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
--- origsrc/libvips/libvips/include/vips/Makefile.am	2013-10-11 23:40:41.000000000 +0900
+++ src/libvips/libvips/include/vips/Makefile.am	2013-10-12 00:02:35.137934900 +0900
@@ -68,7 +68,7 @@ vips_scan_headers = \
 	${top_srcdir}/libvips/include/vips/object.h
 
 enumtypes.h: $(vips_scan_headers) Makefile
-	glib-mkenums --template enumtemplate \
+	glib-mkenums --template $(srcdir)/enumtemplate \
 		$(vips_scan_headers) > xgen-geth && \
 	( cmp -s xgen-geth enumtypes.h || cp xgen-geth enumtypes.h ) && \
 	rm -f xgen-geth 
--- origsrc/libvips/libvips/inplace/Makefile.am	2013-10-11 23:40:41.000000000 +0900
+++ src/libvips/libvips/inplace/Makefile.am	2013-10-12 01:35:17.017104000 +0900
@@ -13,4 +13,4 @@ libinplace_la_SOURCES = \
 	im_draw_smudge.c \
 	inplace_dispatch.c
 
-AM_CPPFLAGS = -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
+AM_CPPFLAGS = -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
--- origsrc/libvips/libvips/iofuncs/Makefile.am	2013-10-11 23:40:41.000000000 +0900
+++ src/libvips/libvips/iofuncs/Makefile.am	2013-10-12 01:37:44.592299300 +0900
@@ -34,7 +34,7 @@ libiofuncs_la_SOURCES = \
 
 EXTRA_DIST = enumtemplate
 
-AM_CPPFLAGS = -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
+AM_CPPFLAGS = -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
 
 # the headers we scan for enums etc.
 # keep in sync with ../include/vips/Makefile.am
@@ -53,7 +53,7 @@ vips_scan_headers = \
 	${top_srcdir}/libvips/include/vips/object.h
 
 enumtypes.c: $(vips_scan_headers) Makefile
-	glib-mkenums --template enumtemplate \
+	glib-mkenums --template $(srcdir)/enumtemplate \
 		$(vips_scan_headers) > xgen-getc && \
 	( cmp -s xgen-getc enumtypes.c || cp xgen-getc enumtypes.c ) && \
 	rm -f xgen-getc
--- origsrc/libvips/libvips/morphology/Makefile.am	2013-10-11 23:40:41.000000000 +0900
+++ src/libvips/libvips/morphology/Makefile.am	2013-10-12 01:38:43.613783700 +0900
@@ -9,4 +9,4 @@ libmorphology_la_SOURCES = \
 	morph_dispatch.c \
 	im_label_regions.c 
 
-AM_CPPFLAGS = -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
+AM_CPPFLAGS = -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
--- origsrc/libvips/libvips/mosaicing/Makefile.am	2013-10-11 23:40:41.000000000 +0900
+++ src/libvips/libvips/mosaicing/Makefile.am	2013-10-12 01:39:33.584486800 +0900
@@ -23,4 +23,4 @@ libmosaicing_la_SOURCES = \
 	global_balance.h \
 	mosaic.h
 
-AM_CPPFLAGS = -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
+AM_CPPFLAGS = -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
--- origsrc/libvips/libvips/resample/Makefile.am	2013-10-11 23:40:41.000000000 +0900
+++ src/libvips/libvips/resample/Makefile.am	2013-10-12 01:23:45.629145800 +0900
@@ -42,4 +42,4 @@ endif
 
 noinst_LTLIBRARIES = libresample.la
 
-AM_CPPFLAGS = -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
+AM_CPPFLAGS = -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
--- origsrc/libvips/libvips/video/Makefile.am	2013-10-11 23:40:42.000000000 +0900
+++ src/libvips/libvips/video/Makefile.am	2013-10-12 01:41:21.667494600 +0900
@@ -5,4 +5,4 @@ libvideo_la_SOURCES = \
 	im_video_v4l1.c \
 	im_video_test.c 
 
-AM_CPPFLAGS = -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
+AM_CPPFLAGS = -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
--- origsrc/libvips/libvipsCC/Makefile.am	2013-10-11 23:40:42.000000000 +0900
+++ src/libvips/libvipsCC/Makefile.am	2013-10-12 01:51:55.298354000 +0900
@@ -2,6 +2,7 @@ SUBDIRS = \
 	include 
 
 AM_CPPFLAGS = \
+	-I$(top_builddir)/libvips/include \
 	-I$(top_srcdir)/libvips/include \
 	-I$(top_srcdir)/libvipsCC/include \
 	@VIPS_CFLAGS@ 
--- origsrc/libvips/swig/vipsCC/Makefile.am	2013-10-11 23:40:42.000000000 +0900
+++ src/libvips/swig/vipsCC/Makefile.am	2013-10-12 01:59:05.215346200 +0900
@@ -1,6 +1,7 @@
 # Let make substitute the value of PYTHON_INCLUDES rather than auto*
 # this makes it easier to support multiple python installs
 AM_CPPFLAGS = \
+	-I${top_builddir}/libvips/include \
 	-I${top_srcdir}/libvips/include \
 	-I${top_srcdir}/libvipsCC/include \
 	@VIPS_CFLAGS@ \
@@ -33,20 +34,20 @@ vipscc_LTLIBRARIES = vimagemodule.la vdi
 
 # maybe there's a clever way to avoid repeating the link stuff 4 times
 # vimagemodule uses the C API as well, so it needs libvips too
-vimagemodule_la_LDFLAGS = -module -avoid-version 
-vimagemodule_la_LIBADD = ../../libvipsCC/libvipsCC.la ../../libvips/libvips.la $(VIPS_LIBS)
+vimagemodule_la_LDFLAGS = -module -avoid-version -no-undefined
+vimagemodule_la_LIBADD = ../../libvipsCC/libvipsCC.la ../../libvips/libvips.la $(VIPS_LIBS) -lpython${PYTHON_VERSION}
 nodist_vimagemodule_la_SOURCES = vimagemodule.cxx 
 
-vdisplaymodule_la_LDFLAGS = -module -avoid-version 
-vdisplaymodule_la_LIBADD = ../../libvipsCC/libvipsCC.la $(VIPS_LIBS)
+vdisplaymodule_la_LDFLAGS = -module -avoid-version -no-undefined
+vdisplaymodule_la_LIBADD = ../../libvipsCC/libvipsCC.la $(VIPS_LIBS) -lpython${PYTHON_VERSION}
 nodist_vdisplaymodule_la_SOURCES = vdisplaymodule.cxx 
 
-verrormodule_la_LDFLAGS = -module -avoid-version 
-verrormodule_la_LIBADD = ../../libvipsCC/libvipsCC.la $(VIPS_LIBS)
+verrormodule_la_LDFLAGS = -module -avoid-version -no-undefined
+verrormodule_la_LIBADD = ../../libvipsCC/libvipsCC.la $(VIPS_LIBS) -lpython${PYTHON_VERSION}
 nodist_verrormodule_la_SOURCES = verrormodule.cxx 
 
-vmaskmodule_la_LDFLAGS = -module -avoid-version 
-vmaskmodule_la_LIBADD = ../../libvipsCC/libvipsCC.la $(VIPS_LIBS)
+vmaskmodule_la_LDFLAGS = -module -avoid-version -no-undefined
+vmaskmodule_la_LIBADD = ../../libvipsCC/libvipsCC.la $(VIPS_LIBS) -lpython${PYTHON_VERSION}
 nodist_vmaskmodule_la_SOURCES = vmaskmodule.cxx 
 
 CLEANFILES = VImage.h 
--- origsrc/libvips/tools/Makefile.am	2013-10-11 23:40:42.000000000 +0900
+++ src/libvips/tools/Makefile.am	2013-10-12 01:45:25.620619600 +0900
@@ -13,7 +13,7 @@ vipsthumbnail_SOURCES = vipsthumbnail.c
 find_mosaic_SOURCES = find_mosaic.c
 mergeup_SOURCES =  mergeup.c
 
-AM_CPPFLAGS = -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@
+AM_CPPFLAGS = -I${top_builddir}/libvips/include -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@
 AM_LDFLAGS = @LDFLAGS@ 
 LDADD = @VIPS_CFLAGS@ ${top_builddir}/libvips/libvips.la @VIPS_LIBS@
 if ENABLE_CXX
