--- origsrc/hjson-cpp-2.2/CMakeLists.txt	2021-04-15 01:17:16.000000000 +0900
+++ src/hjson-cpp-2.2/CMakeLists.txt	2021-06-02 09:29:40.190230400 +0900
@@ -28,14 +28,22 @@ set_property(CACHE HJSON_NUMBER_PARSER P
 option(BUILD_SHARED_LIBS "Build shared libraries" OFF)
 option(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS "Needed for shared libs on Windows" ON)
 
-if(HJSON_VERSIONED_INSTALL)
-  set(include_dest "include/hjson-${PROJECT_VERSION}")
-  set(cmake_dest "lib/hjson-${PROJECT_VERSION}")
-  set(lib_dest "lib/hjson-${PROJECT_VERSION}")
+if(CYGWIN)
+  include(GNUInstallDirs)
+  set(include_dest "${CMAKE_INSTALL_INCLUDEDIR}")
+  set(cmake_dest "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+  set(lib_dest "${CMAKE_INSTALL_LIBDIR}")
+  set(bin_dest "${CMAKE_INSTALL_BINDIR}")
 else()
-  set(include_dest "include/hjson")
-  set(cmake_dest "lib/hjson")
-  set(lib_dest "lib")
+  if(HJSON_VERSIONED_INSTALL)
+    set(include_dest "include/hjson-${PROJECT_VERSION}")
+    set(cmake_dest "lib/hjson-${PROJECT_VERSION}")
+    set(lib_dest "lib/hjson-${PROJECT_VERSION}")
+  else()
+    set(include_dest "include/hjson")
+    set(cmake_dest "lib/hjson")
+    set(lib_dest "lib")
+  endif()
 endif()
 
 add_subdirectory(src)
@@ -60,3 +68,10 @@ if(HJSON_ENABLE_INSTALL)
 
   install(EXPORT hjson DESTINATION ${cmake_dest})
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/hjson-cpp.pc.in ${CMAKE_BINARY_DIR}/hjson-cpp.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/hjson-cpp.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/hjson-cpp-2.2/hjson-cpp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/hjson-cpp-2.2/hjson-cpp.pc.in	2021-06-02 09:30:34.856646600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: hjson-cpp
+Description: Hjson for C++
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lhjson
+Cflags: -I${includedir}
--- origsrc/hjson-cpp-2.2/src/CMakeLists.txt	2021-04-15 01:17:16.000000000 +0900
+++ src/hjson-cpp-2.2/src/CMakeLists.txt	2021-06-02 09:17:36.558323500 +0900
@@ -33,6 +33,14 @@ set_target_properties(hjson PROPERTIES
 )
 
 if(HJSON_ENABLE_INSTALL)
-  install(TARGETS hjson EXPORT hjson DESTINATION ${lib_dest})
+  if(CYGWIN)
+    install(TARGETS hjson
+      EXPORT hjson
+      RUNTIME DESTINATION ${bin_dest}
+      ARCHIVE DESTINATION ${lib_dest}
+    )
+  else()
+    install(TARGETS hjson EXPORT hjson DESTINATION ${lib_dest})
+  endif()
   install(FILES ${header} DESTINATION ${include_dest})
 endif()
