--- origsrc/hjson-cpp-2.6/CMakeLists.txt	2025-04-21 00:10:46.000000000 +0900
+++ src/hjson-cpp-2.6/CMakeLists.txt	2026-01-11 16:21:29.472847400 +0900
@@ -28,14 +28,18 @@ set_property(CACHE HJSON_NUMBER_PARSER P
 option(BUILD_SHARED_LIBS "Build shared libraries" OFF)
 option(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS "Needed for shared libs on Windows" ON)
 
+include(GNUInstallDirs)
+
+set(bin_dest "${CMAKE_INSTALL_BINDIR}")
+
 if(HJSON_VERSIONED_INSTALL)
   set(include_dest "include/hjson-${PROJECT_VERSION}")
   set(cmake_dest "lib/hjson-${PROJECT_VERSION}")
   set(lib_dest "lib/hjson-${PROJECT_VERSION}")
 else()
-  set(include_dest "include/hjson")
-  set(cmake_dest "lib/hjson")
-  set(lib_dest "lib")
+  set(include_dest "${CMAKE_INSTALL_INCLUDEDIR}/hjson")
+  set(cmake_dest "${CMAKE_INSTALL_LIBDIR}/cmake/hjson")
+  set(lib_dest "${CMAKE_INSTALL_LIBDIR}")
 endif()
 
 add_subdirectory(src)
@@ -60,3 +64,10 @@ if(HJSON_ENABLE_INSTALL)
 
   install(EXPORT hjson DESTINATION ${cmake_dest})
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/${lib_dest})
+set(includedir ${CMAKE_INSTALL_PREFIX}/${include_dest})
+configure_file(${CMAKE_SOURCE_DIR}/hjson-cpp.pc.in ${CMAKE_BINARY_DIR}/hjson-cpp.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/hjson-cpp.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/hjson-cpp-2.6/hjson-cpp.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/hjson-cpp-2.6/hjson-cpp.pc.in	2026-01-11 16:20:47.962071300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: hjson-cpp
+Description: Hjson for C++
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lhjson
+Cflags: -I${includedir}
--- origsrc/hjson-cpp-2.6/src/CMakeLists.txt	2025-04-21 00:10:46.000000000 +0900
+++ src/hjson-cpp-2.6/src/CMakeLists.txt	2026-01-11 15:59:50.656109200 +0900
@@ -33,6 +33,6 @@ set_target_properties(hjson PROPERTIES
 )
 
 if(HJSON_ENABLE_INSTALL)
-  install(TARGETS hjson EXPORT hjson DESTINATION ${lib_dest})
+  install(TARGETS hjson EXPORT hjson RUNTIME DESTINATION ${bin_dest} ARCHIVE DESTINATION ${lib_dest})
   install(FILES ${header} DESTINATION ${include_dest})
 endif()
