--- origsrc/liblogdb/Makefile.am	2024-10-09 20:41:29.000000000 +0900
+++ src/liblogdb/Makefile.am	2024-10-09 20:41:30.865953800 +0900
@@ -34,6 +34,7 @@ liblogdb_la_SOURCES = \
     src/red_black_tree.c
 
 liblogdb_la_CFLAGS = -I$(top_srcdir)/include
+liblogdb_la_LDFLAGS = -no-undefined
 
 if USE_TESTS
 noinst_PROGRAMS = tests
--- origsrc/liblogdb/liblogdb.pc.in	2024-10-09 20:41:29.000000000 +0900
+++ src/liblogdb/liblogdb.pc.in	2024-10-09 20:41:30.881584800 +0900
@@ -5,7 +5,7 @@ includedir=@includedir@
 
 Name: liblogdb
 Description: A save and simple append only key-/value-database format written in C
-URL: https://github.com/logdb/logdb
+URL: https://github.com/liblogdb/liblogdb
 Version: @PACKAGE_VERSION@
 Cflags: -I${includedir}
 Libs: -L${libdir} -llogdb
