--- origsrc/SAASound-3.2/src/Makefile.am	2004-03-08 12:25:52.000000000 +0900
+++ src/SAASound-3.2/src/Makefile.am	2011-08-27 23:14:24.504286800 +0900
@@ -6,6 +6,6 @@ GENERAL_SRCS = SAAAmp.cpp SAAEnv.cpp SAA
 lib_LTLIBRARIES = libSAASound.la
 
 libSAASound_la_SOURCES = $(GENERAL_SRCS)
-libSAASound_la_LDFLAGS = 	\
+libSAASound_la_LDFLAGS = -no-undefined	\
         -release $(LT_RELEASE)	\
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
