diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/frag_find-1.0.0/Makefile.am src/frag_find-1.0.0/Makefile.am
--- origsrc/frag_find-1.0.0/Makefile.am	2013-04-07 01:12:16.000000000 +0900
+++ src/frag_find-1.0.0/Makefile.am	2015-12-27 19:06:36.495405200 +0900
@@ -1,8 +1,6 @@
-SUBDIRS = src doc java
+SUBDIRS = src doc # java
 
 AM_CFLAGS   = -Wall
 AM_CPPFLAGS = -Wall
 
 EXTRA_DIST = bloom.spec.in 
-
-
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/frag_find-1.0.0/configure.ac src/frag_find-1.0.0/configure.ac
--- origsrc/frag_find-1.0.0/configure.ac	2013-04-07 01:41:26.000000000 +0900
+++ src/frag_find-1.0.0/configure.ac	2015-12-27 19:02:58.081380900 +0900
@@ -26,6 +26,7 @@ AC_PROG_INSTALL
 
 # Try for Java?
 AM_PROG_GCJ
+AC_PATH_PROGS([JAVAC], [javac ecj])
 
 # Bring additional directories where things might be found into our
 # search path. I don't know why autoconf doesn't do this by default
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/frag_find-1.0.0/java/Makefile.am src/frag_find-1.0.0/java/Makefile.am
--- origsrc/frag_find-1.0.0/java/Makefile.am	2013-04-07 01:12:16.000000000 +0900
+++ src/frag_find-1.0.0/java/Makefile.am	2015-12-27 19:02:58.097005900 +0900
@@ -40,7 +40,6 @@ frag_find.jar$(EXEEXT): $(CLASSES)
 clean:
 	rm -f *~ *.class 
 
-JAVAC = javac
 SUFFIXES = .java
 JAVAC_ARGS = -d ${builddir} -sourcepath ${srcdir}:${builddir}:. -cp ${srcdir}:${builddir}:.
 
