--- origsrc/frag_find-1.0.0/configure.ac	2013-04-07 01:41:26.000000000 +0900
+++ src/frag_find-1.0.0/configure.ac	2013-06-29 00:37:44.940896100 +0900
@@ -26,6 +26,7 @@ AC_PROG_INSTALL
 
 # Try for Java?
 AM_PROG_GCJ
+AC_PATH_PROGS([JAVAC], [javac ecj])
 
 # Bring additional directories where things might be found into our
 # search path. I don't know why autoconf doesn't do this by default
--- origsrc/frag_find-1.0.0/java/Makefile.am	2013-04-07 01:12:16.000000000 +0900
+++ src/frag_find-1.0.0/java/Makefile.am	2013-06-29 00:39:29.941872700 +0900
@@ -40,7 +40,6 @@ frag_find.jar$(EXEEXT): $(CLASSES)
 clean:
 	rm -f *~ *.class 
 
-JAVAC = javac
 SUFFIXES = .java
 JAVAC_ARGS = -d ${builddir} -sourcepath ${srcdir}:${builddir}:. -cp ${srcdir}:${builddir}:.
 
