--- origsrc/net6-1.3.12/Makefile.am	2009-11-15 22:02:46.000000000 +0900
+++ src/net6-1.3.12/Makefile.am	2010-09-04 01:57:48.430875000 +0900
@@ -12,9 +12,10 @@ libnet6_la_LIBADD = $(libnet6_LIBS)
 
 if WIN32
 libnet6_la_LIBADD += -lws2_32 -lintl
-libnet6_la_LDFLAGS += -no-undefined
 endif
 
+libnet6_la_LDFLAGS += -no-undefined
+
 INCLUDES = $(libnet6_CFLAGS) -I$(srcdir)/inc
 
 pkginclude_HEADERS = \
