--- origsrc/libsbml-5.20.0/CMakeLists.txt	2023-04-03 19:48:28.000000000 +0900
+++ src/libsbml-5.20.0/CMakeLists.txt	2023-04-07 00:02:26.707246500 +0900
@@ -613,7 +613,9 @@ valid. It should contain the file zlib.h
 endif(WITH_ZLIB)
 
 # install find scripts only for used dependencies
-install(FILES ${LIBSBML_FIND_MODULES} DESTINATION share/cmake/Modules)
+if (NOT CYGWIN)
+    install(FILES ${LIBSBML_FIND_MODULES} DESTINATION share/cmake/Modules)
+endif()
 
 ###############################################################################
 #
@@ -996,7 +998,9 @@ set(DOCUMENTATION_FILES
         VERSION.txt
 )
 
-install(FILES ${DOCUMENTATION_FILES} DESTINATION ${MISC_PREFIX})
+if (NOT CYGWIN)
+    install(FILES ${DOCUMENTATION_FILES} DESTINATION ${MISC_PREFIX})
+endif()
 
 ###############################################################################
 #
--- origsrc/libsbml-5.20.0/examples/c/CMakeLists.txt	2023-04-03 19:48:28.000000000 +0900
+++ src/libsbml-5.20.0/examples/c/CMakeLists.txt	2023-04-07 00:02:26.717280200 +0900
@@ -69,8 +69,11 @@ foreach(example
 )
 	add_executable(example_c_${example} ${example}.c util.c)
 	set_target_properties(example_c_${example} PROPERTIES  OUTPUT_NAME ${example})
+    if (LIBSBML_SKIP_STATIC_LIBRARY)
+        target_link_libraries(example_c_${example} ${LIBSBML_LIBRARY})	
+    else()
 	target_link_libraries(example_c_${example} ${LIBSBML_LIBRARY}-static)	
-
+    endif()
 endforeach()
 
 # run examples as test 
--- origsrc/libsbml-5.20.0/examples/c++/CMakeLists.txt	2023-04-03 19:48:28.000000000 +0900
+++ src/libsbml-5.20.0/examples/c++/CMakeLists.txt	2023-04-07 00:02:26.719355800 +0900
@@ -79,14 +79,21 @@ foreach(example
 )
     add_executable(example_cpp_${example} ${example}.cpp util.c)
     set_target_properties(example_cpp_${example} PROPERTIES  OUTPUT_NAME ${example})
+    if (LIBSBML_SKIP_STATIC_LIBRARY)
+      target_link_libraries(example_cpp_${example} ${LIBSBML_LIBRARY})
+    else()
     target_link_libraries(example_cpp_${example} ${LIBSBML_LIBRARY}-static)
-
+    endif()
 endforeach()
 
 if (WITH_LIBXML)
   add_executable(example_cpp_rngvalidator rngvalidator.cpp util.c)
   set_target_properties(example_cpp_rngvalidator PROPERTIES  OUTPUT_NAME rngvalidator)
+  if (LIBSBML_SKIP_STATIC_LIBRARY)
+    target_link_libraries(example_cpp_rngvalidator ${LIBSBML_LIBRARY})
+  else()
   target_link_libraries(example_cpp_rngvalidator ${LIBSBML_LIBRARY}-static)
+  endif()
   target_link_libraries(example_cpp_rngvalidator LIBXML::LIBXML ${EXTRA_LIBS})
 endif()
 
