--- origsrc/tristripper-1.10/CMakeLists.txt	2014-07-28 08:33:52.000000000 +0900
+++ src/tristripper-1.10/CMakeLists.txt	2025-08-20 14:07:45.622101700 +0900
@@ -33,12 +33,12 @@ set_target_properties(tristripper
   VERSION "${MAJOR_VERSION}.${MINOR_VERSION}"
 )
 
-install(TARGETS tristripper DESTINATION lib LIBRARY)
+install(TARGETS tristripper RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 
 install(FILES
   include/public_types.h
   include/tri_stripper.h
-  DESTINATION include)
+  DESTINATION include/tristripper)
 
 install(FILES
   include/detail/cache_simulator.h
@@ -47,7 +47,7 @@ install(FILES
   include/detail/heap_array.h
   include/detail/policy.h
   include/detail/types.h
-  DESTINATION include/detail
+  DESTINATION include/tristripper/detail
 )
 
 ###############################################
@@ -75,3 +75,10 @@ if(TRISTRIPPER_BUILD_TEST)
   endif()
 
 endif()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/tristripper.pc.in ${CMAKE_BINARY_DIR}/tristripper.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/tristripper.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/tristripper-1.10/test/static_assert.h	2014-07-28 08:33:52.000000000 +0900
+++ src/tristripper-1.10/test/static_assert.h	2025-08-20 14:04:37.859855800 +0900
@@ -12,10 +12,12 @@ namespace _static_assert {
 }
 
 
-
+#if defined(__CYGWIN__)
+#define static_assert_msg(expr, msg)	static_assert(expr, msg)
+#else
 #define static_assert(expr)				{ _static_assert::compile_time_error<((expr) != 0)> ERROR_STATIC_ASSERT; (void) ERROR_STATIC_ASSERT; }
 #define static_assert_msg(expr, msg)	{ _static_assert::compile_time_error<((expr) != 0)> ERROR_##msg; (void) ERROR_##msg; }
-
+#endif
 
 
 #endif // HEADER_GUARD_STATIC_ASSERT_H
--- origsrc/tristripper-1.10/tristripper.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/tristripper-1.10/tristripper.pc.in	2025-08-20 14:08:10.739366200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: tristripper
+Description: Triangle stripification algorithm
+Version: @MAJOR_VERSION@.@MINOR_VERSION@
+Libs: -L${libdir} -ltristripper
+Cflags: -I${includedir}
