--- origsrc/libwebm/CMakeLists.txt	2022-04-25 17:27:32.000000000 +0900
+++ src/libwebm/CMakeLists.txt	2022-04-26 10:55:08.884292400 +0900
@@ -337,6 +337,10 @@ if (WIN32)
   set_target_properties(webm PROPERTIES PREFIX lib)
 endif ()
 
+if (CYGWIN)
+  set_target_properties(webm PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 add_executable(mkvparser_sample ${mkvparser_sample_sources})
 target_link_libraries(mkvparser_sample LINK_PUBLIC webm)
 
@@ -383,18 +387,21 @@ if (ENABLE_WEBMTS)
 endif ()
 
 if (ENABLE_TESTS)
-  set(GTEST_SRC_DIR "${LIBWEBM_SRC_DIR}/../googletest" CACHE PATH
-      "Path to Googletest git repository.")
-  # This directory is where libwebm will build googletest dependencies.
-  set(GTEST_BUILD_DIR "${CMAKE_BINARY_DIR}/googletest_build")
-
-  if (LIBWEBM_DISABLE_GTEST_CMAKE)
-    add_library(gtest STATIC "${GTEST_SRC_DIR}/googletest/src/gtest-all.cc")
-    include_directories("${GTEST_SRC_DIR}/googletest")
-  else ()
-    add_subdirectory("${GTEST_SRC_DIR}" "${GTEST_BUILD_DIR}")
-  endif ()
-  include_directories("${GTEST_SRC_DIR}/googletest/include")
+  find_package(gtest)
+  if (NOT GTEST_FOUND)
+    set(GTEST_SRC_DIR "${LIBWEBM_SRC_DIR}/../googletest" CACHE PATH
+        "Path to Googletest git repository.")
+    # This directory is where libwebm will build googletest dependencies.
+    set(GTEST_BUILD_DIR "${CMAKE_BINARY_DIR}/googletest_build")
+
+    if (LIBWEBM_DISABLE_GTEST_CMAKE)
+      add_library(gtest STATIC "${GTEST_SRC_DIR}/googletest/src/gtest-all.cc")
+      include_directories("${GTEST_SRC_DIR}/googletest")
+    else ()
+      add_subdirectory("${GTEST_SRC_DIR}" "${GTEST_BUILD_DIR}")
+    endif ()
+    include_directories("${GTEST_SRC_DIR}/googletest/include")
+  endif()
 
   add_executable(mkvmuxer_tests ${mkvmuxer_tests_sources})
   target_link_libraries(mkvmuxer_tests LINK_PUBLIC gtest webm)
@@ -451,3 +458,12 @@ if (ENABLE_IWYU)
     message(STATUS "  See README.libwebm for more information.")
   endif ()
 endif ()
+
+if (CYGWIN)
+  set(prefix ${CMAKE_INSTALL_PREFIX})
+  set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+  set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+  set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+  configure_file(${CMAKE_SOURCE_DIR}/libwebm.pc.in ${CMAKE_BINARY_DIR}/libwebm.pc @ONLY)
+  install(FILES ${CMAKE_BINARY_DIR}/libwebm.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+endif()
--- origsrc/libwebm/libwebm.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libwebm/libwebm.pc.in	2022-04-26 10:54:56.868971500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libwebm
+Description: WebM File Parser
+Version: @VERSION@
+Libs: -L${libdir} -lwebm
+Cflags: -I${includedir}
