--- origsrc/libwebm/CMakeLists.txt	2026-01-22 23:17:12.890856300 +0900
+++ src/libwebm/CMakeLists.txt	2026-01-22 23:58:12.557837000 +0900
@@ -499,3 +499,10 @@ install(FILES ${mkvmuxer_public_headers}
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/webm/mkvmuxer)
 install(FILES ${mkvparser_public_headers}
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/webm/mkvparser)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libwebm.pc.in ${CMAKE_BINARY_DIR}/libwebm.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libwebm.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libwebm/libwebm.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libwebm/libwebm.pc.in	2026-01-22 23:57:39.209985700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libwebm
+Description: WebM File Parser
+Version: @VERSION@
+Libs: -L${libdir} -lwebm
+Cflags: -I${includedir}
