--- origsrc/libansilove-1.4.2/CMakeLists.txt	2025-03-25 02:12:12.000000000 +0900
+++ src/libansilove-1.4.2/CMakeLists.txt	2025-04-14 08:32:57.434186900 +0900
@@ -43,7 +43,6 @@ add_library(objlib OBJECT ${SRC} ${LOADE
 set_property(TARGET objlib PROPERTY POSITION_INDEPENDENT_CODE ON)
 
 add_library(ansilove SHARED $<TARGET_OBJECTS:objlib>)
-add_library(ansilove-static STATIC $<TARGET_OBJECTS:objlib>)
 
 target_link_libraries(ansilove ${GD_LIBRARIES} m)
 
@@ -51,8 +50,9 @@ set_target_properties(ansilove PROPERTIE
   VERSION ${LIB_VERSION_STRING} SOVERSION ${LIB_VERSION_MAJOR}
   C_VISIBILITY_PRESET hidden)
 
-install(TARGETS ansilove DESTINATION ${CMAKE_INSTALL_LIBDIR})
-install(TARGETS ansilove-static DESTINATION ${CMAKE_INSTALL_LIBDIR})
+add_subdirectory(example)
+
+install(TARGETS ansilove RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(FILES include/ansilove.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 install(FILES man/libansilove.3 DESTINATION ${CMAKE_INSTALL_MANDIR}/man3/)
 install(FILES man/ansilove_clean.3 DESTINATION ${CMAKE_INSTALL_MANDIR}/man3/)
@@ -67,3 +67,10 @@ install(FILES man/ansilove_icedraw.3 DES
 install(FILES man/ansilove_pcboard.3 DESTINATION ${CMAKE_INSTALL_MANDIR}/man3/)
 install(FILES man/ansilove_tundra.3 DESTINATION ${CMAKE_INSTALL_MANDIR}/man3/)
 install(FILES man/ansilove_xbin.3 DESTINATION ${CMAKE_INSTALL_MANDIR}/man3/)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libansilove.pc.in ${CMAKE_BINARY_DIR}/libansilove.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libansilove.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libansilove-1.4.2/example/CMakeLists.txt	2025-03-25 02:12:12.000000000 +0900
+++ src/libansilove-1.4.2/example/CMakeLists.txt	2025-04-14 08:34:35.614271300 +0900
@@ -7,8 +7,8 @@ set(CMAKE_C_EXTENSIONS OFF)
 project(example C)
 
 # Ansilove library
-find_path(ANSILOVE_INCLUDE_DIRS ansilove.h)
-find_library(ANSILOVE_LIBRARIES NAMES ansilove REQUIRED)
+set(ANSILOVE_INCLUDE_DIRS ${CMAKE_SOURCE_DIR}/include)
+set(ANSILOVE_LIBRARIES ansilove)
 include_directories(${ANSILOVE_INCLUDE_DIRS})
 
 set(SRC example.c)
@@ -16,4 +16,5 @@ set(SRC example.c)
 add_definitions(-Wall -Wextra -pedantic)
 add_executable(example ${SRC})
 
+target_link_directories(example PRIVATE ${CMAKE_BINARY_DIR})
 target_link_libraries(example ${ANSILOVE_LIBRARIES})
--- origsrc/libansilove-1.4.2/libansilove.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libansilove-1.4.2/libansilove.pc.in	2025-04-14 08:31:18.248571300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libansilove
+Description: Library for converting ANSI, ASCII, and other formats to PNG
+Version: @LIB_VERSION_STRING@
+Libs: -L${libdir} -lansilove
+Cflags: -I${includedir}
