--- origsrc/ondir-0.2.4/Makefile	2013-09-20 06:59:09.000000000 +0900
+++ src/ondir-0.2.4/Makefile	2025-02-08 09:08:50.662962400 +0900
@@ -14,13 +14,13 @@ DESTDIR=
 # scripts.
 # **WARNING** This is not recommended at all.
 
-CC=cc
-CFLAGS=-O3 -DVERSION=\"$(VERSION)\" -DGLOBAL_CONF=\"$(CONF)\" -DUSE_ONENTERLEAVE
-CFLAGS=-Wall -c -g -DVERSION=\"$(VERSION)\" -DGLOBAL_CONF=\"$(CONF)\"
+#CC=cc
+#CFLAGS=-O3 -DVERSION=\"$(VERSION)\" -DGLOBAL_CONF=\"$(CONF)\" -DUSE_ONENTERLEAVE
+CFLAGS+=-DVERSION=\"$(VERSION)\" -DGLOBAL_CONF=\"$(CONF)\"
 
-LD=cc
-LDFLAGS=
-LDFLAGS=-g
+#LD=cc
+#LDFLAGS=
+#LDFLAGS=-g
 
 $(TARGET): $(OBJS)
 	$(LD) $(OBJS) $(LDFLAGS) -o $@
