--- origsrc/DromeAudio-0.2.1/src/DromeAudio/CMakeLists.txt	2010-11-14 14:25:30.000000000 +0900
+++ src/DromeAudio-0.2.1/src/DromeAudio/CMakeLists.txt	2013-07-21 22:35:13.991055200 +0900
@@ -79,15 +79,16 @@ if(VORBISFILE_FOUND)
 	add_definitions(-DWITH_VORBIS)
 endif(VORBISFILE_FOUND)
 
-add_library(DromeAudio STATIC ${SRCS})
+add_library(DromeAudio SHARED ${SRCS})
 target_link_libraries(
 	DromeAudio
 	${LIBS}
 )
+set_target_properties(DromeAudio PROPERTIES SOVERSION 0)
 
 install(
 	TARGETS DromeAudio
-	RUNTIME DESTINATION ${DROMEAUDIO_LIBRARY_DIR}
+	RUNTIME DESTINATION bin
 	LIBRARY DESTINATION ${DROMEAUDIO_LIBRARY_DIR}
 	ARCHIVE DESTINATION ${DROMEAUDIO_LIBRARY_DIR}
 )
