--- origsrc/libctl-4.5.1/examples/Makefile.am	2022-02-02 11:46:49.000000000 +0900
+++ src/libctl-4.5.1/examples/Makefile.am	2024-10-16 08:48:20.826923200 +0900
@@ -29,7 +29,7 @@ MY_SOURCES = example.c # plus any other
 
 MY_LIBS    =         # extra libs you need go here
 MY_LDFLAGS =         # extra -L flags go here
-MY_CPPFLAGS =        # extra -I include flags go here
+MY_CPPFLAGS =        -I$(top_srcdir)/src # extra -I include flags go here
 MY_DEFS =            # extra -D define flags go here
 
 # what is printed out when invoking your program with --version:
--- origsrc/libctl-4.5.1/utils/Makefile.am	2022-02-02 11:46:49.000000000 +0900
+++ src/libctl-4.5.1/utils/Makefile.am	2024-10-16 08:48:20.826923200 +0900
@@ -7,7 +7,7 @@ noinst_PROGRAMS = geomtst
 EXTRA_DIST = gen-ctl-io.in README geom.scm geom-ctl-io-defaults.c nlopt.c
 
 libctlgeom_la_SOURCES = geom.c $(top_srcdir)/src/ctl-math.c $(top_srcdir)/src/integrator.c geom-ctl-io.c ctlgeom-types.h
-libctlgeom_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@
+libctlgeom_la_LDFLAGS = -no-undefined -version-info @SHARED_VERSION_INFO@
 libctlgeom_la_CPPFLAGS = -DLIBCTLGEOM -I$(top_srcdir)/src
 
 geomtst_SOURCES = geomtst.c
