--- origsrc/VanitySearch-1.19/IntMod.cpp	2020-08-10 23:33:14.000000000 +0900
+++ src/VanitySearch-1.19/IntMod.cpp	2021-09-23 05:56:35.804931300 +0900
@@ -22,6 +22,10 @@
 #define MAX(x,y) (((x)>(y))?(x):(y))
 #define MIN(x,y) (((x)<(y))?(x):(y))
 
+#if defined(_P)
+#undef _P
+#endif
+
 static Int     _P;       // Field characteristic
 static Int     _R;       // Montgomery multiplication R
 static Int     _R2;      // Montgomery multiplication R2
