--- origsrc/libfishsound-1.0.0/Makefile.am	2009-03-28 12:40:03.000000000 +0900
+++ src/libfishsound-1.0.0/Makefile.am	2012-04-23 14:49:53.909445400 +0900
@@ -29,6 +29,8 @@
 
 ## Process this file with automake to produce Makefile.in
 
+ACLOCAL_AMFLAGS = -I m4
+
 DISTCHECK_CONFIGURE_FLAGS = --enable-gcc-werror
 
 SUBDIRS = doc include src
--- origsrc/libfishsound-1.0.0/src/libfishsound/Makefile.am	2010-02-03 22:34:02.000000000 +0900
+++ src/libfishsound-1.0.0/src/libfishsound/Makefile.am	2012-04-23 14:49:53.909445400 +0900
@@ -29,5 +29,5 @@ libfishsound_la_SOURCES = \
 	flac.c \
 	fs_vector.c
 
-libfishsound_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
+libfishsound_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@ -no-undefined
 libfishsound_la_LIBADD = $(VORBIS_LIBS) $(SPEEX_LIBS) $(FLAC_LIBS)
