--- origsrc/libfishsound-1.0.1/configure.ac	2025-02-15 16:49:54.000000000 +0900
+++ src/libfishsound-1.0.1/configure.ac	2025-03-08 13:28:37.665521200 +0900
@@ -48,9 +48,14 @@ AC_CHECK_TYPES([uintptr_t])
 AC_CHECK_PROG(HAVE_PKG_CONFIG, pkg-config, yes)
 PKG_PROG_PKG_CONFIG
 
+ac_enable_doc=no
+AC_ARG_ENABLE(doc,
+  AC_HELP_STRING([--enable-doc], [enable doc]),
+  [ ac_enable_doc=yes ])
+
 # Check for doxygen
 AC_CHECK_PROG(HAVE_DOXYGEN, doxygen, true, false)
-AM_CONDITIONAL(HAVE_DOXYGEN,$HAVE_DOXYGEN)
+AM_CONDITIONAL(HAVE_DOXYGEN, test $HAVE_DOXYGEN = true -a ac_enable_doc = yes)
 if test $HAVE_DOXYGEN = "false"; then
              AC_MSG_WARN([*** doxygen not found, docs will not be built])
 fi
--- origsrc/libfishsound-1.0.1/src/examples/Makefile.am	2025-02-15 16:49:54.000000000 +0900
+++ src/libfishsound-1.0.1/src/examples/Makefile.am	2025-03-08 13:32:29.201114500 +0900
@@ -25,8 +25,8 @@ endif
 endif
 
 # Programs to build
-noinst_PROGRAMS = $(oggz_examples) $(sndfile_examples) $(oggz_sndfile_examples)
-#bin_PROGRAMS = $(oggz_examples) $(sndfile_examples) $(oggz_sndfile_examples)
+#noinst_PROGRAMS = $(oggz_examples) $(sndfile_examples) $(oggz_sndfile_examples)
+bin_PROGRAMS = $(oggz_examples) $(sndfile_examples) $(oggz_sndfile_examples)
 
 fishsound_identify_SOURCES = fishsound-identify.c
 fishsound_identify_LDADD = $(FISHSOUND_LIBS) $(OGGZ_LIBS)
--- origsrc/libfishsound-1.0.1/src/libfishsound/Makefile.am	2025-02-15 16:49:54.000000000 +0900
+++ src/libfishsound-1.0.1/src/libfishsound/Makefile.am	2025-03-08 13:27:33.823232600 +0900
@@ -29,5 +29,5 @@ libfishsound_la_SOURCES = \
 	flac.c \
 	fs_vector.c
 
-libfishsound_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
+libfishsound_la_LDFLAGS = -no-undefined -version-info @SHARED_VERSION_INFO@ @SHLIB_VERSION_ARG@
 libfishsound_la_LIBADD = $(VORBIS_LIBS) $(SPEEX_LIBS) $(FLAC_LIBS)
