--- origsrc/tap-1.01/src/Makefile.am	2005-01-07 04:18:25.000000000 +0900
+++ src/tap-1.01/src/Makefile.am	2013-10-27 21:28:29.175588400 +0900
@@ -1,5 +1,6 @@
 lib_LTLIBRARIES = libtap.la
 libtap_la_SOURCES = tap.c tap.h
+libtap_la_LDFLAGS = -no-undefined
 
 man_MANS = tap.3
 EXTRA_DIST = $(man_MANS)
--- origsrc/tap-1.01/tests/diag/Makefile.am	2004-12-22 03:49:43.000000000 +0900
+++ src/tap-1.01/tests/diag/Makefile.am	2013-10-27 21:34:17.675437300 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/fail/Makefile.am	2004-12-22 03:49:47.000000000 +0900
+++ src/tap-1.01/tests/fail/Makefile.am	2013-10-27 21:34:29.396140400 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/ok/ok/Makefile.am	2004-12-22 03:50:11.000000000 +0900
+++ src/tap-1.01/tests/ok/ok/Makefile.am	2013-10-27 21:34:46.750632600 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/ok/ok-hash/Makefile.am	2004-12-22 03:50:02.000000000 +0900
+++ src/tap-1.01/tests/ok/ok-hash/Makefile.am	2013-10-27 21:35:02.205710700 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/ok/ok-numeric/Makefile.am	2004-12-22 03:50:07.000000000 +0900
+++ src/tap-1.01/tests/ok/ok-numeric/Makefile.am	2013-10-27 21:35:10.166648200 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/pass/Makefile.am	2004-12-22 03:49:52.000000000 +0900
+++ src/tap-1.01/tests/pass/Makefile.am	2013-10-27 21:35:19.346335700 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/plan/no-tests/Makefile.am	2004-12-22 03:50:13.000000000 +0900
+++ src/tap-1.01/tests/plan/no-tests/Makefile.am	2013-10-27 21:35:42.908835700 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/plan/no_plan/Makefile.am	2004-12-22 03:50:16.000000000 +0900
+++ src/tap-1.01/tests/plan/no_plan/Makefile.am	2013-10-27 21:35:27.935202900 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/plan/not-enough-tests/Makefile.am	2004-12-22 03:50:19.000000000 +0900
+++ src/tap-1.01/tests/plan/not-enough-tests/Makefile.am	2013-10-27 21:35:35.265281000 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/plan/sane/Makefile.am	2004-12-22 03:50:22.000000000 +0900
+++ src/tap-1.01/tests/plan/sane/Makefile.am	2013-10-27 21:35:50.275046700 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/plan/skip_all/Makefile.am	2004-12-22 03:50:24.000000000 +0900
+++ src/tap-1.01/tests/plan/skip_all/Makefile.am	2013-10-27 21:35:58.151999800 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/plan/too-many-plans/Makefile.am	2004-12-22 03:50:27.000000000 +0900
+++ src/tap-1.01/tests/plan/too-many-plans/Makefile.am	2013-10-27 21:36:05.888327900 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/plan/too-many-tests/Makefile.am	2004-12-22 03:50:30.000000000 +0900
+++ src/tap-1.01/tests/plan/too-many-tests/Makefile.am	2013-10-27 21:36:14.734031000 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/skip/Makefile.am	2004-12-22 03:49:56.000000000 +0900
+++ src/tap-1.01/tests/skip/Makefile.am	2013-10-27 21:36:22.752585700 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/todo/Makefile.am	2004-12-22 03:49:59.000000000 +0900
+++ src/tap-1.01/tests/todo/Makefile.am	2013-10-27 21:36:30.535788900 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
