--- origsrc/tap-1.01/Makefile.am	2004-12-20 07:23:22.000000000 +0900
+++ src/tap-1.01/Makefile.am	2024-09-26 18:47:19.471920400 +0900
@@ -3,3 +3,6 @@ SUBDIRS += tests
 
 prove:
 	prove -v -r
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/tap-1.01/configure.in	2005-01-12 03:12:44.000000000 +0900
+++ src/tap-1.01/configure.in	2024-09-26 18:47:19.471920400 +0900
@@ -49,4 +49,5 @@ AC_CONFIG_FILES([Makefile
 		 tests/skip/Makefile
 		 tests/todo/Makefile
 		])
+AC_CONFIG_FILES([tap.pc])
 AC_OUTPUT
--- origsrc/tap-1.01/src/Makefile.am	2005-01-07 04:18:25.000000000 +0900
+++ src/tap-1.01/src/Makefile.am	2024-09-26 17:37:20.889441800 +0900
@@ -1,5 +1,6 @@
 lib_LTLIBRARIES = libtap.la
 libtap_la_SOURCES = tap.c tap.h
+libtap_la_LDFLAGS = -no-undefined
 
 man_MANS = tap.3
 EXTRA_DIST = $(man_MANS)
--- origsrc/tap-1.01/tap.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/tap-1.01/tap.pc.in	2024-09-26 18:47:19.456293400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: tap
+Description: Testing library for C, implementing the Test Anything Protocol
+Version: @VERSION@
+Libs: -L${libdir} -ltap
+Cflags: -I${includedir}
--- origsrc/tap-1.01/tests/diag/Makefile.am	2004-12-22 03:49:43.000000000 +0900
+++ src/tap-1.01/tests/diag/Makefile.am	2024-09-26 17:37:20.889441800 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/fail/Makefile.am	2004-12-22 03:49:47.000000000 +0900
+++ src/tap-1.01/tests/fail/Makefile.am	2024-09-26 17:37:20.889441800 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/ok/ok/Makefile.am	2004-12-22 03:50:11.000000000 +0900
+++ src/tap-1.01/tests/ok/ok/Makefile.am	2024-09-26 17:37:20.905073800 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/ok/ok-hash/Makefile.am	2004-12-22 03:50:02.000000000 +0900
+++ src/tap-1.01/tests/ok/ok-hash/Makefile.am	2024-09-26 17:37:20.905073800 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/ok/ok-numeric/Makefile.am	2004-12-22 03:50:07.000000000 +0900
+++ src/tap-1.01/tests/ok/ok-numeric/Makefile.am	2024-09-26 17:37:20.905073800 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/pass/Makefile.am	2004-12-22 03:49:52.000000000 +0900
+++ src/tap-1.01/tests/pass/Makefile.am	2024-09-26 17:37:20.905073800 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/plan/no-tests/Makefile.am	2004-12-22 03:50:13.000000000 +0900
+++ src/tap-1.01/tests/plan/no-tests/Makefile.am	2024-09-26 17:37:20.905073800 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/plan/no_plan/Makefile.am	2004-12-22 03:50:16.000000000 +0900
+++ src/tap-1.01/tests/plan/no_plan/Makefile.am	2024-09-26 17:37:20.905073800 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/plan/not-enough-tests/Makefile.am	2004-12-22 03:50:19.000000000 +0900
+++ src/tap-1.01/tests/plan/not-enough-tests/Makefile.am	2024-09-26 17:37:20.905073800 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/plan/sane/Makefile.am	2004-12-22 03:50:22.000000000 +0900
+++ src/tap-1.01/tests/plan/sane/Makefile.am	2024-09-26 17:37:20.905073800 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/plan/skip_all/Makefile.am	2004-12-22 03:50:24.000000000 +0900
+++ src/tap-1.01/tests/plan/skip_all/Makefile.am	2024-09-26 17:37:20.905073800 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/plan/too-many-plans/Makefile.am	2004-12-22 03:50:27.000000000 +0900
+++ src/tap-1.01/tests/plan/too-many-plans/Makefile.am	2024-09-26 17:37:20.905073800 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/plan/too-many-tests/Makefile.am	2004-12-22 03:50:30.000000000 +0900
+++ src/tap-1.01/tests/plan/too-many-tests/Makefile.am	2024-09-26 17:37:20.920700600 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/skip/Makefile.am	2004-12-22 03:49:56.000000000 +0900
+++ src/tap-1.01/tests/skip/Makefile.am	2024-09-26 17:37:20.920700600 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
--- origsrc/tap-1.01/tests/todo/Makefile.am	2004-12-22 03:49:59.000000000 +0900
+++ src/tap-1.01/tests/todo/Makefile.am	2024-09-26 17:37:20.920700600 +0900
@@ -6,8 +6,8 @@ EXTRA_DIST = 		$(TESTS) test.pl
 
 check_PROGRAMS = 	test
 
-test_CFLAGS = 		-g -I../../src
+test_CFLAGS = 		-g -I$(top_srcdir)/src
 test_LDFLAGS = 		-L../../src
-test_LDADD = 		-ltap
+test_LDADD = 		$(top_builddir)/src/libtap.la
 
 CLEANFILES =	test.o test.c.out test.pl.out
