--- origsrc/liblogging-1.0.5/rfc3195/src/Makefile.am	2014-09-26 16:57:14.000000000 +0900
+++ src/liblogging-1.0.5/rfc3195/src/Makefile.am	2015-02-11 10:22:04.575284500 +0900
@@ -2,6 +2,7 @@ lib_LTLIBRARIES = liblogging-rfc3195.la
 
 liblogging_rfc3195_la_CFLAGS = ${AM_CFLAGS}
 liblogging_rfc3195_la_LDFLAGS = \
+	-no-undefined \
 	-version-info 0:0:0 \
 	-export-symbols-regex '(^(srAPI|srSLMG).*)'
 liblogging_rfc3195_la_SOURCES = \
--- origsrc/liblogging-1.0.5/rfc3195/src/oscallsUnix.c	2014-09-26 16:57:14.000000000 +0900
+++ src/liblogging-1.0.5/rfc3195/src/oscallsUnix.c	2015-02-11 10:22:04.590909500 +0900
@@ -76,7 +76,15 @@ srRetVal getCurrTime(int* year, int* mon
      * It is UTC - localtime, which is opposite sign of min east of GMT.
      */
     lBias = -(daylight ? altzone : timezone);
+#elif defined(__CYGWIN__)
+  {
+    lBias = timezone;
+    if (tm->tm_isdst) {
+      lBias += 3600;
+    }
+  }
 #else
-	lBias = tm->tm_gmtoff;#endif
+	lBias = tm->tm_gmtoff;
+#endif
 	if(lBias < 0)	{		*pcOffsetMode = '-';		lBias *= -1;	}	else		*pcOffsetMode = '+';	*pOffsetHour = lBias / 3600;	*pOffsetMinute = lBias % 3600;
 	return SR_RET_OK;}
--- origsrc/liblogging-1.0.5/stdlog/Makefile.am	2014-12-09 19:09:47.000000000 +0900
+++ src/liblogging-1.0.5/stdlog/Makefile.am	2015-02-11 10:22:04.590909500 +0900
@@ -5,6 +5,7 @@ liblogging_stdlog_la_CPPFLAGS =
 liblogging_stdlog_la_CFLAGS = ${AM_CFLAGS}
 liblogging_stdlog_la_LIBADD =  $(SOL_LIBS)
 liblogging_stdlog_la_LDFLAGS = \
+	-no-undefined \
 	-version-info 0:0:0 \
 	-export-symbols-regex '(^stdlog_.*)'
 # For instructions on how to increment --version-info see:
