--- origsrc/fastqcheck-1.1.1/Makefile.am	2016-11-08 23:31:42.000000000 +0900
+++ src/fastqcheck-1.1.1/Makefile.am	2025-12-26 19:23:21.408176800 +0900
@@ -1,2 +1,5 @@
 SUBDIRS=src
-ACLOCAL_AMFLAGS=-I m4
\ No newline at end of file
+ACLOCAL_AMFLAGS=-I m4
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/fastqcheck-1.1.1/VERSION	2016-11-08 23:31:42.000000000 +0900
+++ src/fastqcheck-1.1.1/VERSION	2025-12-26 19:33:12.546204600 +0900
@@ -1 +1 @@
-1.1
\ No newline at end of file
+1.1.1
\ No newline at end of file
--- origsrc/fastqcheck-1.1.1/configure.ac	2016-11-08 23:31:42.000000000 +0900
+++ src/fastqcheck-1.1.1/configure.ac	2025-12-26 19:14:10.211360600 +0900
@@ -10,6 +10,7 @@ AC_PROG_CC
 AC_PROG_CXX
 
 AC_CONFIG_FILES([Makefile src/Makefile])
+AC_CONFIG_FILES([fastqcheck.pc])
 AC_OUTPUT
 
 AC_CHECK_HEADERS([math.h])
--- origsrc/fastqcheck-1.1.1/fastqcheck.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/fastqcheck-1.1.1/fastqcheck.pc.in	2025-12-26 19:14:10.160195000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: fastqcheck
+Description: Generate statistics on and validate fastq files
+Version: @VERSION@
+Libs: -L${libdir} -lfastqcheck
+Cflags: -I${includedir}
--- origsrc/fastqcheck-1.1.1/src/Makefile.am	2016-11-08 23:31:42.000000000 +0900
+++ src/fastqcheck-1.1.1/src/Makefile.am	2025-12-26 19:08:57.277519500 +0900
@@ -1,5 +1,5 @@
 # ship these headers in the "make dist" target
-dist_data_DATA = readseq.h
+pkginclude_HEADERS = readseq.h
 
 # fastqcheck is our top level program
 bin_PROGRAMS = fastqcheck
@@ -9,4 +9,4 @@ fastqcheck_LDADD=-lm libfastqcheck.la
 # libfastqcheck.so is our library
 lib_LTLIBRARIES = libfastqcheck.la
 libfastqcheck_la_SOURCES = readseq.c
-libfastqcheck_la_LDFLAGS= -version-info 1:0
+libfastqcheck_la_LDFLAGS= -no-undefined -version-info 1:0
