--- origsrc/freealut/configure.ac	2024-10-10 22:51:01.000000000 +0900
+++ src/freealut/configure.ac	2024-10-10 22:58:00.273660600 +0900
@@ -214,6 +214,9 @@ fi
 
 AC_SUBST([AM_CFLAGS])
 
+extra_cflags=
+AC_SUBST([extra_cflags])
+
 # NOTE: Do not break the following line, otherwise we are into CR/LF vs. LF
 # trouble! This is a buglet in autoconf IMHO, but easy to work around.
 AC_CONFIG_FILES([Makefile admin/Makefile admin/pkgconfig/Makefile admin/pkgconfig/freealut-config admin/pkgconfig/freealut.pc admin/RPM/freealut.spec examples/Makefile include/Makefile src/Makefile test_suite/Makefile])
