--- origsrc/netcdf-4.1.1/configure.ac	2010-04-11 13:15:33.000000000 +0900
+++ src/netcdf-4.1.1/configure.ac	2011-01-01 14:34:03.141096900 +0900
@@ -46,9 +46,6 @@ AC_CONFIG_SRCDIR([libsrc/nc.c])
 
 AC_MSG_NOTICE([checking user options])
 
-# Turn off shared libraries by default
-AM_DISABLE_SHARED
-
 # For backward compatibility, check to see if the user has set CXX,
 # FC, and/or F90 to null. If so, treat that as the equivilent
 # --disable- option to configure. The autoconf manual suggests the
@@ -942,8 +939,7 @@ AM_CONDITIONAL(BUILD_CXX, [test "x$nc_bu
 # Set up libtool.
 AC_MSG_NOTICE([setting up libtool])
 LT_PREREQ([2.2.2])
-#LT_INIT(win32-dll)
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 AC_MSG_NOTICE([finding other utilities])
 
@@ -1234,7 +1230,7 @@ dnl 	 fi
 else
 # !netcdf-4
    # Modify netcdf.h to include defs for missing parallel stuff.
-   sed '/#include <errno.h>/{G;s/$/\/* These defs added by netCDF configure because parallel HDF5 is not present. *\//;G;s/$/#ifndef MPI_INCLUDED/;G;s/$/#define MPI_Comm int/;G;s/$/#define MPI_Info int/;G;s/$/#define MPI_COMM_WORLD 0/;G;s/$/#define MPI_INFO_NULL 0/;G;s/$/#endif/;}' libsrc4/netcdf_base.h > libsrc4/netcdf.h
+   sed '/#include <errno.h>/{G;s/$/\/* These defs added by netCDF configure because parallel HDF5 is not present. *\//;G;s/$/#ifndef MPI_INCLUDED/;G;s/$/#define MPI_Comm int/;G;s/$/#define MPI_Info int/;G;s/$/#define MPI_COMM_WORLD 0/;G;s/$/#define MPI_INFO_NULL 0/;G;s/$/#endif/;}' ${srcdir}/libsrc4/netcdf_base.h > ${srcdir}/libsrc4/netcdf.h
 fi
 
 # Restore LDFLAGS to it's old value.
--- origsrc/netcdf-4.1.1/cxx/Makefile.am	2009-12-09 05:58:10.000000000 +0900
+++ src/netcdf-4.1.1/cxx/Makefile.am	2010-12-31 15:12:24.478336700 +0900
@@ -13,7 +13,7 @@
 # c++ -o nctst -g -O2  nctst.o libnetcdf_c++.a -L../libsrc -lnetcdf 
 
 AM_CPPFLAGS =
-AM_LDFLAGS =
+AM_LDFLAGS = -no-undefined
 
 CLEANFILES = nctst_classic.nc nctst_64bit_offset.nc nctst_netcdf4.nc	\
 nctst_netcdf4_classic.nc nctst_classic.cdl nctst_64bit_offset.cdl	\
--- origsrc/netcdf-4.1.1/cxx4/Makefile.am	2010-03-30 23:18:26.000000000 +0900
+++ src/netcdf-4.1.1/cxx4/Makefile.am	2010-12-31 15:12:39.179475900 +0900
@@ -7,7 +7,7 @@
 
 # Point pre-preprocessor to netcdf-4 directory (libsrc4).
 AM_CPPFLAGS = -I$(top_srcdir)/libsrc4 
-AM_LDFLAGS =
+AM_LDFLAGS = -no-undefined
 
 # Link to the netCDF-4 and the HDF5 libraries, and the C++ library.
 LDADD = $(top_builddir)/cxx4/libnetcdf_c++4.la $(top_builddir)/libsrc4/libnetcdf.la -lhdf5_hl -lhdf5
--- origsrc/netcdf-4.1.1/f90/Makefile.am	2010-01-26 06:01:08.000000000 +0900
+++ src/netcdf-4.1.1/f90/Makefile.am	2010-12-31 15:12:57.786231100 +0900
@@ -11,7 +11,7 @@
 # $Id: Makefile.am,v 1.49 2010/01/25 21:01:08 ed Exp $
 
 AM_FCFLAGS =
-AM_LDFLAGS =
+AM_LDFLAGS = -no-undefined
 
 # If the user specified a root location for HDF4, use it.
 if USE_HDF4_DIR
--- origsrc/netcdf-4.1.1/fortran/Makefile.am	2009-12-27 22:24:00.000000000 +0900
+++ src/netcdf-4.1.1/fortran/Makefile.am	2010-12-31 15:13:13.759199100 +0900
@@ -6,7 +6,7 @@
 # $Id: Makefile.am,v 1.55 2009/12/27 13:24:00 ed Exp $
 
 AM_CPPFLAGS =
-AM_LDFLAGS =
+AM_LDFLAGS = -no-undefined
 
 # The F77 API man page.
 man_MANS = netcdf_f77.3
--- origsrc/netcdf-4.1.1/libncdap3/Makefile.am	2010-04-11 05:56:32.000000000 +0900
+++ src/netcdf-4.1.1/libncdap3/Makefile.am	2010-12-31 15:14:15.838464700 +0900
@@ -16,7 +16,7 @@ else
 AM_CPPFLAGS = -I$(top_srcdir)/libsrc
 endif
 
-AM_LDFLAGS = 
+AM_LDFLAGS = -no-undefined
 
 CURL_CFLAGS = @CURL_CFLAGS@
 
--- origsrc/netcdf-4.1.1/libncdap3/oc/Makefile.am	2010-04-11 05:56:38.000000000 +0900
+++ src/netcdf-4.1.1/libncdap3/oc/Makefile.am	2010-12-31 15:14:27.264895100 +0900
@@ -11,7 +11,7 @@
 AM_CPPFLAGS = -I$(top_srcdir)/libsrc
 
 # Linker needs to search current directory when building tests.
-AM_LDFLAGS = 
+AM_LDFLAGS = -no-undefined
 
 CURL_CFLAGS = @CURL_CFLAGS@
 CURL_LIBS = @CURL_LIBS@
--- origsrc/netcdf-4.1.1/libncdap4/Makefile.am	2009-12-06 05:46:28.000000000 +0900
+++ src/netcdf-4.1.1/libncdap4/Makefile.am	2010-12-31 15:14:53.242248700 +0900
@@ -12,7 +12,7 @@ AM_CPPFLAGS = -I$(top_srcdir)/libncdap3
 	      -I$(top_srcdir)/libsrc
 
 AM_CPPFLAGS += -I@HDF5DIR@/include -I@ZLIBDIR@/include
-AM_LDFLAGS = -L@HDF5DIR@/lib -L@ZLIBDIR@/lib
+AM_LDFLAGS = -L@HDF5DIR@/lib -L@ZLIBDIR@/lib -no-undefined
 
 # If the user specified a prefix for the HDF4 library location, use it.
 if USE_HDF4_DIR
--- origsrc/netcdf-4.1.1/libsrc/Makefile.am	2010-04-11 13:15:36.000000000 +0900
+++ src/netcdf-4.1.1/libsrc/Makefile.am	2011-01-01 13:24:49.669342400 +0900
@@ -49,7 +49,7 @@ LDADD=
 libnetcdf_la_LIBADD = 
 noinst_LTLIBRARIES = 
 AM_CPPFLAGS =
-AM_LDFLAGS =
+AM_LDFLAGS = -no-undefined
 libnetcdf_la_LDFLAGS =
 
 # Does  the user want the V2 API?
--- origsrc/netcdf-4.1.1/man4/Makefile.am	2010-01-20 00:03:38.000000000 +0900
+++ src/netcdf-4.1.1/man4/Makefile.am	2011-01-01 15:53:22.784474600 +0900
@@ -17,13 +17,13 @@ netcdf-f77.texi netcdf-f90.texi netcdf-c
 #netcdf-cxx4.texi
 
 # Get lists of the coresponding ps, info, and pdf files.
-html_docs = ${info_TEXINFOS:.texi=.html} 
-html_dirs = ${info_TEXINFOS:.texi=} 
-ps_docs = ${info_TEXINFOS:.texi=.ps} 
-dvi_docs = ${info_TEXINFOS:.texi=.dvi} 
-pdf_docs = ${info_TEXINFOS:.texi=.pdf}
-info_docs= ${info_TEXINFOS:.texi=.info}
-txt_docs= ${info_TEXINFOS:.texi=.txt}
+html_docs = #${info_TEXINFOS:.texi=.html} 
+html_dirs = #${info_TEXINFOS:.texi=} 
+ps_docs = #${info_TEXINFOS:.texi=.ps} 
+dvi_docs = #${info_TEXINFOS:.texi=.dvi} 
+pdf_docs = #${info_TEXINFOS:.texi=.pdf}
+info_docs= #${info_TEXINFOS:.texi=.info}
+txt_docs= #${info_TEXINFOS:.texi=.txt}
 
 # These are html versions of the man pages.
 html_mans = ncdump-man-1.html ncgen-man-1.html netcdf_f77-man.html	\
