--- origsrc/libSDL2pp-0.17.0/CMakeLists.txt	2022-04-29 03:47:26.000000000 +0900
+++ src/libSDL2pp-0.17.0/CMakeLists.txt	2022-05-25 17:08:47.050689100 +0900
@@ -24,7 +24,7 @@ endif()
 
 # depends
 find_package(SDL2 REQUIRED)
-set(SDL2_ALL_LIBRARIES SDL2::SDL2)
+set(SDL2_ALL_LIBRARIES SDL2)
 set(SDL2_ALL_PKGCONFIG_MODULES sdl2)
 
 if(TARGET SDL2::SDL2main)
@@ -246,9 +246,6 @@ if(CMAKE_SOURCE_DIR STREQUAL PROJECT_SOU
 	install(
 		FILES
 			${PROJECT_BINARY_DIR}/cmake/SDL2ppConfig.cmake
-			cmake/FindSDL2_image.cmake
-			cmake/FindSDL2_ttf.cmake
-			cmake/FindSDL2_mixer.cmake
 		DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/SDL2pp
 	)
 endif()
--- origsrc/libSDL2pp-0.17.0/sdl2pp.pc.in	2022-04-29 03:47:26.000000000 +0900
+++ src/libSDL2pp-0.17.0/sdl2pp.pc.in	2022-05-26 08:39:09.036755300 +0900
@@ -5,7 +5,7 @@ includedir=${prefix}/include
 
 Name: sdl2pp
 Description: C++11 bindings/wrappers for SDL2
-Version: @SDL2PP_VERSION@
+Version: @PROJECT_VERSION@
 Requires: @SDL2_ALL_PKGCONFIG_MODULES@
 Conflicts:
 Libs: @SDL2PP_EXTRA_PKGCONFIG_LIBRARIES@ -L${libdir} -lSDL2pp
