--- origsrc/libSDL2pp-0.7.1/tests/testing.h	2015-03-27 09:55:52.000000000 +0900
+++ src/libSDL2pp-0.7.1/tests/testing.h	2015-03-28 19:30:46.090269200 +0900
@@ -36,6 +36,8 @@
 #include <sstream>
 #include <functional>
 
+#include <boost/lexical_cast.hpp>
+
 //
 // Helper class for literal quoting / extra processing
 //
@@ -120,7 +122,7 @@ private:
 		default:
 			return str;
 		}
-		return std::string("\033[") + (bright ? "1" : "0") + ";" + std::to_string(30 + color) + "m" + str + "\033[0m";
+		return boost::lexical_cast<std::string>("\033[") + (bright ? "1" : "0") + ";" + boost::lexical_cast<std::string>(30 + color) + "m" + str + "\033[0m";
 	}
 
 	//
