diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libSDL2pp-0.8.0/CMakeLists.txt src/libSDL2pp-0.8.0/CMakeLists.txt
--- origsrc/libSDL2pp-0.8.0/CMakeLists.txt	2015-06-10 06:20:30.000000000 +0900
+++ src/libSDL2pp-0.8.0/CMakeLists.txt	2015-06-13 18:25:48.045095500 +0900
@@ -208,7 +208,7 @@ IF(CMAKE_SOURCE_DIR STREQUAL PROJECT_SOU
 	)
 
 	# cmake file
-	SET(CMAKEMODDIR share/cmake/Modules CACHE STRING "directory where to install cmake module")
+	SET(CMAKEMODDIR lib/cmake/libSDL2pp CACHE STRING "directory where to install cmake module")
 	CONFIGURE_FILE(
 		FindSDL2PP.cmake.in
 		FindSDL2PP.cmake
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libSDL2pp-0.8.0/examples/fill.cc src/libSDL2pp-0.8.0/examples/fill.cc
--- origsrc/libSDL2pp-0.8.0/examples/fill.cc	2015-06-10 06:20:30.000000000 +0900
+++ src/libSDL2pp-0.8.0/examples/fill.cc	2015-06-13 18:17:22.890546400 +0900
@@ -32,6 +32,10 @@
 #include <SDL2pp/Texture.hh>
 #include <SDL2pp/Surface.hh>
 
+#if !defined(M_PI)
+#define M_PI 3.14159265358979323846f
+#endif
+
 using namespace SDL2pp;
 
 int Run() {
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libSDL2pp-0.8.0/tests/testing.h src/libSDL2pp-0.8.0/tests/testing.h
--- origsrc/libSDL2pp-0.8.0/tests/testing.h	2015-06-10 06:20:30.000000000 +0900
+++ src/libSDL2pp-0.8.0/tests/testing.h	2015-06-13 18:14:02.681264600 +0900
@@ -36,6 +36,8 @@
 #include <sstream>
 #include <functional>
 
+#include <boost/lexical_cast.hpp>
+
 //
 // Helper class for literal quoting / extra processing
 //
@@ -120,7 +122,7 @@ private:
 		default:
 			return str;
 		}
-		return std::string("\033[") + (bright ? "1" : "0") + ";" + std::to_string(30 + color) + "m" + str + "\033[0m";
+		return boost::lexical_cast<std::string>("\033[") + (bright ? "1" : "0") + ";" + boost::lexical_cast<std::string>(30 + color) + "m" + str + "\033[0m";
 	}
 
 	//
