diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libSDL2pp-0.9.0/CMakeLists.txt src/libSDL2pp-0.9.0/CMakeLists.txt
--- origsrc/libSDL2pp-0.9.0/CMakeLists.txt	2015-09-07 22:22:57.000000000 +0900
+++ src/libSDL2pp-0.9.0/CMakeLists.txt	2015-09-08 19:13:16.342530600 +0900
@@ -236,7 +236,7 @@ IF(CMAKE_SOURCE_DIR STREQUAL PROJECT_SOU
 	)
 
 	# cmake file
-	SET(CMAKEMODDIR share/cmake/Modules CACHE STRING "directory where to install cmake module")
+	SET(CMAKEMODDIR lib/cmake/libSDL2pp CACHE STRING "directory where to install cmake module")
 	CONFIGURE_FILE(
 		FindSDL2PP.cmake.in
 		FindSDL2PP.cmake
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libSDL2pp-0.9.0/examples/fill.cc src/libSDL2pp-0.9.0/examples/fill.cc
--- origsrc/libSDL2pp-0.9.0/examples/fill.cc	2015-09-07 22:22:57.000000000 +0900
+++ src/libSDL2pp-0.9.0/examples/fill.cc	2015-09-08 19:17:54.023835000 +0900
@@ -32,6 +32,10 @@
 #include <SDL2pp/Texture.hh>
 #include <SDL2pp/Surface.hh>
 
+#if !defined(M_PI)
+#define M_PI 3.14159265358979323846f
+#endif
+
 using namespace SDL2pp;
 
 int Run() {
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/libSDL2pp-0.9.0/tests/testing.h src/libSDL2pp-0.9.0/tests/testing.h
--- origsrc/libSDL2pp-0.9.0/tests/testing.h	2015-09-07 22:22:57.000000000 +0900
+++ src/libSDL2pp-0.9.0/tests/testing.h	2015-09-08 19:15:59.934225500 +0900
@@ -36,6 +36,10 @@
 #include <sstream>
 #include <functional>
 
+#if defined(__CYGWIN__)
+#include <boost/lexical_cast.hpp>
+#endif
+
 #ifdef _WIN32
 #	define TESTING_NO_COLOR
 #endif
@@ -52,6 +56,17 @@ std::string QuoteLiteral(T value) {
 	return ss.str();
 }
 
+#if defined(__CYGWIN__)
+template<>
+std::string QuoteLiteral<std::string>(std::string value) {
+	return boost::lexical_cast<std::string>("\"") + value + boost::lexical_cast<std::string>("\"");
+}
+
+template<>
+std::string QuoteLiteral<const char*>(const char* value) {
+	return boost::lexical_cast<std::string>("\"") + value + boost::lexical_cast<std::string>("\"");
+}
+#else
 template<>
 std::string QuoteLiteral<std::string>(std::string value) {
 	return std::string("\"") + value + std::string("\"");
@@ -61,6 +76,7 @@ template<>
 std::string QuoteLiteral<const char*>(const char* value) {
 	return std::string("\"") + value + std::string("\"");
 }
+#endif
 
 //
 // Test registry
