--- origsrc/sentencepiece-0.1.99/src/CMakeLists.txt	2023-05-01 02:37:15.000000000 +0900
+++ src/sentencepiece-0.1.99/src/CMakeLists.txt	2023-08-25 12:42:23.710747400 +0900
@@ -222,16 +222,10 @@ if (SPM_ENABLE_SHARED)
   add_library(sentencepiece_train SHARED ${SPM_TRAIN_SRCS})
 endif()
 
-add_library(sentencepiece-static STATIC ${SPM_SRCS})
-add_library(sentencepiece_train-static STATIC ${SPM_TRAIN_SRCS})
-
-target_link_libraries(sentencepiece-static INTERFACE ${SPM_LIBS})
-target_link_libraries(sentencepiece_train-static INTERFACE sentencepiece-static ${SPM_LIBS})
-
 if (SPM_ENABLE_SHARED)
   target_link_libraries(sentencepiece ${SPM_LIBS})
   target_link_libraries(sentencepiece_train ${SPM_LIBS} sentencepiece)
-  set(SPM_INSTALLTARGETS sentencepiece sentencepiece_train sentencepiece-static sentencepiece_train-static)
+  set(SPM_INSTALLTARGETS sentencepiece sentencepiece_train)
   set_target_properties(sentencepiece sentencepiece_train PROPERTIES SOVERSION 0 VERSION 0.0.0)
   set_target_properties(sentencepiece PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS YES)
   set_target_properties(sentencepiece_train PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS YES)
@@ -242,15 +236,8 @@ if (SPM_ENABLE_SHARED)
     set_target_properties(sentencepiece PROPERTIES IMPORT_SUFFIX ".dll.a")
     set_target_properties(sentencepiece_train PROPERTIES IMPORT_SUFFIX ".dll.a")
   endif()
-else()
-  add_library(sentencepiece ALIAS sentencepiece-static)
-  add_library(sentencepiece_train ALIAS sentencepiece_train-static)
-  set(SPM_INSTALLTARGETS sentencepiece-static sentencepiece_train-static)
 endif()
 
-set_target_properties(sentencepiece-static PROPERTIES OUTPUT_NAME "sentencepiece")
-set_target_properties(sentencepiece_train-static PROPERTIES OUTPUT_NAME "sentencepiece_train")
-
 if (NOT MSVC)
   if (SPM_COVERAGE)
     set(CMAKE_CXX_FLAGS "-O0 -Wall -fPIC -coverage ${CMAKE_CXX_FLAGS}")
