--- origsrc/cctz-2.5/CMakeLists.txt	2025-05-20 23:15:26.000000000 +0900
+++ src/cctz-2.5/CMakeLists.txt	2025-06-30 06:39:39.028485300 +0900
@@ -100,6 +100,10 @@ if(APPLE)
 endif()
 add_library(cctz::cctz ALIAS cctz)
 
+if (DEFINED SOVERSION)
+  set_target_properties(cctz PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 if (BUILD_TOOLS)
   add_executable(time_tool src/time_tool.cc)
   cctz_target_set_cxx_standard(time_tool)
@@ -184,3 +188,10 @@ install(FILES cmake/${PROJECT_NAME}-conf
 set(quiet_on_empty QUIET_ON_EMPTY)
 
 feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES ${quiet_on_empty})
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/cctz.pc.in ${CMAKE_BINARY_DIR}/cctz.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/cctz.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/cctz-2.5/cctz.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/cctz-2.5/cctz.pc.in	2025-06-30 06:39:26.458496800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: cctz
+Description: C++ library for translating between absolute and civil times using the rules of a time zone
+Version: @VERSION@
+Libs: -L${libdir} -lcctz
+Cflags: -I${includedir}
