--- origsrc/cctz-2.4/CMakeLists.txt	2024-02-03 03:34:46.000000000 +0900
+++ src/cctz-2.4/CMakeLists.txt	2024-03-16 08:24:56.717790100 +0900
@@ -108,6 +108,10 @@ if(APPLE)
 endif()
 add_library(cctz::cctz ALIAS cctz)
 
+if (DEFINED SOVERSION)
+  set_target_properties(cctz PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 if (BUILD_TOOLS)
   add_executable(time_tool src/time_tool.cc)
   cctz_target_set_cxx_standard(time_tool)
@@ -170,7 +174,7 @@ include(GNUInstallDirs)
 install(TARGETS cctz
   EXPORT ${PROJECT_NAME}-targets
   PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/cctz
-  RUNTIME DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
   )
@@ -196,3 +200,10 @@ else()
 endif()
 
 feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES ${quiet_on_empty})
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/cctz.pc.in ${CMAKE_BINARY_DIR}/cctz.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/cctz.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/cctz-2.4/cctz.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/cctz-2.4/cctz.pc.in	2024-03-16 08:21:14.695324100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: cctz
+Description: C++ library for translating between absolute and civil times using the rules of a time zone
+Version: @VERSION@
+Libs: -L${libdir} -lcctz
+Cflags: -I${includedir}
