--- origsrc/prometheus-cpp-0.13.0/CMakeLists.txt	2021-09-10 00:56:07.000000000 +0900
+++ src/prometheus-cpp-0.13.0/CMakeLists.txt	2021-09-11 09:01:05.607923100 +0900
@@ -105,6 +105,11 @@ if(ENABLE_WARNINGS_AS_ERRORS AND NOT MSV
   )
 endif()
 
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+
 # prometheus-cpp
 
 add_subdirectory(core)
--- origsrc/prometheus-cpp-0.13.0/core/CMakeLists.txt	2021-09-10 00:56:07.000000000 +0900
+++ src/prometheus-cpp-0.13.0/core/CMakeLists.txt	2021-09-11 09:01:05.616941400 +0900
@@ -89,3 +89,6 @@ endif()
 if(benchmark_FOUND)
   add_subdirectory(benchmarks)
 endif()
+
+CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/prometheus-cpp-core.pc.in ${CMAKE_BINARY_DIR}/prometheus-cpp-core.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/prometheus-cpp-core.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/prometheus-cpp-0.13.0/core/prometheus-cpp-core.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/prometheus-cpp-0.13.0/core/prometheus-cpp-core.pc.in	2021-09-11 09:01:05.619940700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: prometheus-cpp-core
+Description: Prometheus Client Library for Modern C++
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lprometheus-cpp-core
+Cflags: -I${includedir}
--- origsrc/prometheus-cpp-0.13.0/pull/CMakeLists.txt	2021-09-10 00:56:07.000000000 +0900
+++ src/prometheus-cpp-0.13.0/pull/CMakeLists.txt	2021-09-11 09:01:05.623931700 +0900
@@ -118,3 +118,6 @@ if(ENABLE_TESTING)
 
   add_subdirectory(tests)
 endif()
+
+CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/prometheus-cpp-pull.pc.in ${CMAKE_BINARY_DIR}/prometheus-cpp-pull.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/prometheus-cpp-pull.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/prometheus-cpp-0.13.0/pull/prometheus-cpp-pull.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/prometheus-cpp-0.13.0/pull/prometheus-cpp-pull.pc.in	2021-09-11 09:01:05.664938500 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: prometheus-cpp-pull
+Description: Prometheus Client Library for Modern C++ (pull)
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lprometheus-cpp-pull
+Cflags: -I${includedir}
+Requires: prometheus-cpp-core civetweb civetweb-cpp zlib
--- origsrc/prometheus-cpp-0.13.0/push/CMakeLists.txt	2021-09-10 00:56:07.000000000 +0900
+++ src/prometheus-cpp-0.13.0/push/CMakeLists.txt	2021-09-11 09:01:05.668954600 +0900
@@ -70,3 +70,6 @@ endif()
 if(ENABLE_TESTING)
   add_subdirectory(tests)
 endif()
+
+CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/prometheus-cpp-push.pc.in ${CMAKE_BINARY_DIR}/prometheus-cpp-push.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/prometheus-cpp-push.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/prometheus-cpp-0.13.0/push/prometheus-cpp-push.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/prometheus-cpp-0.13.0/push/prometheus-cpp-push.pc.in	2021-09-11 09:01:05.686997600 +0900
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: prometheus-cpp-push
+Description: Prometheus Client Library for Modern C++ (push)
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lprometheus-cpp-push
+Cflags: -I${includedir}
+Requires: prometheus-cpp-core libcurl
