--- origsrc/subnetcalc-2.4.19/CMakeLists.txt	2021-11-09 04:10:38.000000000 +0900
+++ src/subnetcalc-2.4.19/CMakeLists.txt	2022-06-16 23:20:19.130752300 +0900
@@ -58,6 +58,10 @@ ELSEIF (${CMAKE_SYSTEM_NAME} MATCHES "Ne
     SET(CMAKE_LIBRARY_PATH "/usr/local/lib")
     INCLUDE_DIRECTORIES("/usr/pkg/include" "/usr/local/include")
 
+ELSEIF (CYGWIN)
+    MESSAGE(STATUS ${CMAKE_SYSTEM_NAME} " supported")
+    ADD_DEFINITIONS("-DLINUX")
+
 ELSE()
     MESSAGE(FATAL_ERROR ${CMAKE_SYSTEM_NAME} " not supported (yet?)")
 
