--- origsrc/subnetcalc-2.4.18/CMakeLists.txt	2021-03-07 07:40:21.000000000 +0900
+++ src/subnetcalc-2.4.18/CMakeLists.txt	2021-05-12 08:06:12.658999300 +0900
@@ -58,6 +58,10 @@ ELSEIF (${CMAKE_SYSTEM_NAME} MATCHES "Ne
     SET(CMAKE_LIBRARY_PATH "/usr/local/lib")
     INCLUDE_DIRECTORIES("/usr/pkg/include" "/usr/local/include")
 
+ELSEIF (CYGWIN)
+    MESSAGE(STATUS ${CMAKE_SYSTEM_NAME} " supported")
+    ADD_DEFINITIONS("-DLINUX")
+
 ELSE()
     MESSAGE(FATAL_ERROR ${CMAKE_SYSTEM_NAME} " not supported (yet?)")
 
