--- origsrc/subnetcalc-2.5.1/CMakeLists.txt	2024-02-13 05:36:39.000000000 +0900
+++ src/subnetcalc-2.5.1/CMakeLists.txt	2024-02-19 20:26:45.626940700 +0900
@@ -58,6 +58,10 @@ ELSEIF (${CMAKE_SYSTEM_NAME} MATCHES "Ne
     SET(CMAKE_LIBRARY_PATH "/usr/local/lib")
     INCLUDE_DIRECTORIES("/usr/pkg/include" "/usr/local/include")
 
+ELSEIF (CYGWIN)
+    MESSAGE(STATUS ${CMAKE_SYSTEM_NAME} " supported")
+    ADD_DEFINITIONS("-DLINUX")
+
 ELSE()
     MESSAGE(FATAL_ERROR ${CMAKE_SYSTEM_NAME} " not supported (yet?)")
 
