diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/forestdb-1.1/CMakeLists.txt src/forestdb-1.1/CMakeLists.txt
--- origsrc/forestdb-1.1/CMakeLists.txt	2016-03-16 15:13:59.000000000 +0900
+++ src/forestdb-1.1/CMakeLists.txt	2016-03-22 15:42:26.527721400 +0900
@@ -189,7 +189,7 @@ add_library(forestdb SHARED
 target_link_libraries(forestdb ${PTHREAD_LIB} ${LIBM} ${SNAPPY_LIBRARIES}
                       ${ASYNC_IO_LIB} ${MALLOC_LIBRARIES} ${LIBRT}
                       ${BACKTRACE_LIBRARIES} ${PLATFORM_LIBRARY} ${CRYPTO_LIB})
-
+set_target_properties(forestdb PROPERTIES SOVERSION 0)
 
 # Create objects libraries for the different "tools" parts forestdb, which can be reused
 # by various targets.
@@ -255,3 +255,11 @@ endif()
 # add top-level testing dir
 enable_testing()
 add_subdirectory("${PROJECT_SOURCE_DIR}/tests")
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "1.1")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/forestdb.pc.in ${CMAKE_BINARY_DIR}/forestdb.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/forestdb.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/forestdb-1.1/forestdb.pc.in src/forestdb-1.1/forestdb.pc.in
--- origsrc/forestdb-1.1/forestdb.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/forestdb-1.1/forestdb.pc.in	2016-03-22 15:42:35.850968900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: forestdb
+Description: Fast Key-Value Storage Engine Based on Hierarchical B+-Tree Trie
+Version: @VERSION@
+Libs: -L${libdir} -lforestdb
+Cflags: -I${includedir}/libforestdb
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/forestdb-1.1/src/filemgr_ops_linux.cc src/forestdb-1.1/src/filemgr_ops_linux.cc
--- origsrc/forestdb-1.1/src/filemgr_ops_linux.cc	2016-03-16 15:13:59.000000000 +0900
+++ src/forestdb-1.1/src/filemgr_ops_linux.cc	2016-03-22 15:35:03.841242600 +0900
@@ -364,7 +364,7 @@ struct tranfer_blk_ownership {
 };
 #endif // EXT4_IOC_TRANSFER_BLK_OWNERSHIP
 
-#ifndef __sun
+#if !defined(__sun) && !defined(__CYGWIN__)
 static
 int _filemgr_linux_ext4_share_blks(int src_fd, int dst_fd, uint64_t src_off,
                                    uint64_t dst_off, uint64_t len)
@@ -385,7 +385,7 @@ int _filemgr_linux_ext4_share_blks(int s
 
 int _filemgr_linux_get_fs_type(int src_fd)
 {
-#ifdef __sun
+#if defined(__sun) || defined(__CYGWIN__)
     // No support for ZFS
     return FILEMGR_FS_NO_COW;
 #else
@@ -419,7 +419,7 @@ int _filemgr_linux_copy_file_range(int f
                                    uint64_t dst_off, uint64_t len)
 {
     int ret = (int)FDB_RESULT_INVALID_ARGS;
-#ifndef __sun
+#if !defined(__sun) && !defined(__CYGWIN__)
     if (fs_type == FILEMGR_FS_BTRFS) {
         struct btrfs_ioctl_clone_range_args cr_args;
 
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/forestdb-1.1/utils/timing.cc src/forestdb-1.1/utils/timing.cc
--- origsrc/forestdb-1.1/utils/timing.cc	2016-03-16 15:13:59.000000000 +0900
+++ src/forestdb-1.1/utils/timing.cc	2016-03-22 15:25:45.049629100 +0900
@@ -261,7 +261,7 @@ ts_nsec get_monotonic_ts() {
     }
 
     ts = (double)time * timebase.numer / timebase.denom;
-#elif defined(__linux__) || defined(__sun) || defined(__FreeBSD__)
+#elif defined(__linux__) || defined(__sun) || defined(__FreeBSD__) || defined(__CYGWIN__)
     /* Linux and Solaris can use clock_gettime */
     struct timespec tm;
     if (clock_gettime(CLOCK_MONOTONIC, &tm) == -1) {
