--- origsrc/mstore-0.3.0/meson.build	2024-06-13 23:16:18.000000000 +0900
+++ src/mstore-0.3.0/meson.build	2025-09-09 17:48:41.577676200 +0900
@@ -68,7 +68,7 @@ if install
     install_dir: get_option('datadir')/'licenses'/meson.project_name()
   )
 
-  module_id = meson.project_name() / fc_id + '-' + fc.version()
+  module_id = meson.project_name() / 'modules'
   meson.add_install_script(
     find_program(files('config'/'install-mod.py')),
     get_option('includedir') / module_id,
