--- origsrc/mstore-0.3.0/CMakeLists.txt	2024-06-13 23:16:18.000000000 +0900
+++ src/mstore-0.3.0/CMakeLists.txt	2024-07-28 19:06:59.700118300 +0900
@@ -84,6 +84,7 @@ install(
   "${PROJECT_NAME}-lib"
   EXPORT
   "${PROJECT_NAME}-targets"
+  RUNTIME DESTINATION bin
   LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
   ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
 )
@@ -97,7 +98,7 @@ install(
 install(
   DIRECTORY
   "${PROJECT_BINARY_DIR}/include/"
-  DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${module-dir}"
+  DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
 )
 # Package license files
 install(
--- origsrc/mstore-0.3.0/config/template.pc	2024-06-13 23:16:18.000000000 +0900
+++ src/mstore-0.3.0/config/template.pc	2024-07-28 19:06:35.576868200 +0900
@@ -7,4 +7,4 @@ Description: @PROJECT_DESCRIPTION@
 Version: @PROJECT_VERSION@
 Requires: mctc-lib
 Libs: -L${libdir} -l@PROJECT_NAME@
-Cflags: -I${includedir} -I${includedir}/@module-dir@
+Cflags: -I${includedir} -I${includedir}
