--- origsrc/ugrep-3.8.3/Makefile.am	2022-06-29 10:13:43.000000000 +0900
+++ src/ugrep-3.8.3/Makefile.am	2022-08-03 16:58:38.222625700 +0900
@@ -28,7 +28,7 @@ install-exec-hook:
 
 install-data-hook:
 		mkdir -p $(DESTDIR)$(datadir)/ugrep && \
-		  cp -rf patterns $(DESTDIR)$(datadir)/ugrep/
+		  cp -rf $(srcdir)/patterns $(DESTDIR)$(datadir)/ugrep/
 		cd $(DESTDIR)$(mandir)/man1; \
 		  rm -f ug.1; \
 		  $(LN_S) ugrep.1 ug.1
