--- origsrc/ugrep-4.5.1/Makefile.am	2024-01-06 23:02:32.000000000 +0900
+++ src/ugrep-4.5.1/Makefile.am	2024-01-08 07:14:38.069527700 +0900
@@ -39,7 +39,7 @@ cp2bin:		$(top_builddir)/src/ugrep$(EXEE
 		@echo
 
 install-exec-hook:
-		cp -f $(top_builddir)/bin/ugrep+ $(top_builddir)/bin/ug+ $(DESTDIR)$(bindir)
+#		cp -f $(top_builddir)/bin/ugrep+ $(top_builddir)/bin/ug+ $(DESTDIR)$(bindir)
 		rm -f $(DESTDIR)$(bindir)/ug$(EXEEXT)
 		cd $(DESTDIR)$(bindir) && cp -f ugrep$(EXEEXT) ug$(EXEEXT)
 
--- origsrc/ugrep-4.5.1/src/Makefile.am	2024-01-06 23:02:32.000000000 +0900
+++ src/ugrep-4.5.1/src/Makefile.am	2024-01-08 07:19:16.727238100 +0900
@@ -1,4 +1,4 @@
 bin_PROGRAMS   = ugrep
-ugrep_CPPFLAGS = -I$(top_srcdir)/include $(EXTRA_CFLAGS) $(SIMD_FLAGS) $(PTHREAD_CFLAGS) -DPLATFORM=\"$(PLATFORM)\" -DGREP_PATH=\"$(GREP_PATH)\" -DWITH_NO_INDENT
+ugrep_CPPFLAGS = -I$(top_srcdir)/include $(EXTRA_CFLAGS) $(SIMD_FLAGS) $(PTHREAD_CFLAGS) -DPLATFORM=\"$(PLATFORM)\" -DGREP_PATH=\"$(GREP_PATH)\" -DWITH_NO_INDENT -I$(top_srcdir)/lzma/C
 ugrep_SOURCES  = ugrep.cpp cnf.hpp cnf.cpp flag.hpp glob.hpp glob.cpp mmap.hpp output.hpp output.cpp query.hpp query.cpp screen.hpp screen.cpp stats.hpp stats.cpp vkey.hpp vkey.cpp zstream.hpp zopen.h zopen.c
 ugrep_LDADD    = $(PTHREAD_LIBS) $(top_builddir)/lib/libreflex.a
