--- origsrc/ugrep-4.3.2/Makefile.am	2023-11-03 23:20:28.000000000 +0900
+++ src/ugrep-4.3.2/Makefile.am	2023-11-12 13:26:42.276570900 +0900
@@ -24,7 +24,7 @@ cp2bin:		$(top_builddir)/src/ugrep$(EXEE
 		@echo
 
 install-exec-hook:
-		cp -f $(top_builddir)/bin/ugrep+ $(top_builddir)/bin/ug+ $(DESTDIR)$(bindir)
+#		cp -f $(top_builddir)/bin/ugrep+ $(top_builddir)/bin/ug+ $(DESTDIR)$(bindir)
 		rm -f $(DESTDIR)$(bindir)/ug$(EXEEXT)
 		cd $(DESTDIR)$(bindir) && cp -f ugrep$(EXEEXT) ug$(EXEEXT)
 
