--- origsrc/liblightify/src/tests/test-lightify.c	2025-02-07 17:56:36.000000000 +0900
+++ src/liblightify/src/tests/test-lightify.c	2025-02-07 18:22:30.397663300 +0900
@@ -62,7 +62,7 @@ void print_protocol_mismatch_write(struc
 				(one == two ? '=' : '!'), two);
 		if (one != two) found =1;
 	}
-	if (found) ck_abort_msg(buf);
+	if (found) ck_abort_msg("%s", buf);
 }
 
 void print_protocol_mismatch_read(struct fake_socket *mfs, const unsigned char *should) {
@@ -78,7 +78,7 @@ void print_protocol_mismatch_read(struct
 				(one == two ? '=' : '!'), two);
 		if (one != two) found =1;
 	}
-	if (found) ck_abort_msg(buf);
+	if (found) ck_abort_msg("%s", buf);
 }
 
 
@@ -579,7 +579,7 @@ START_TEST(lightify_tst_manipulate_nodes
 					sprintf(buf + strlen(buf), " %02x%c=%02x", one,
 							(one == two ? '=' : '!'), two);
 				}
-				ck_abort_msg(buf);
+				ck_abort_msg("%s", buf);
 			}
 			ck_assert_int_eq(err, 0);
 
