--- origsrc/shash-0.2.6/src/Makefile.am	2002-08-31 22:49:24.000000000 +0900
+++ src/shash-0.2.6/src/Makefile.am	2024-10-31 13:33:00.489394700 +0900
@@ -4,7 +4,7 @@ bin_PROGRAMS = shash
 EXTRA_DIST = environ.h shash.h gaa.h gaaout.c errors.h defines.h extra.h functions.h shash.gaa random.h
 shash_SOURCES = shash.c errors.c extra.c functions.c random.c environ.c
 shash_LDADD = gaaout.o -lmhash
-gaaout.c: shash.gaa
-	gaa shash.gaa -o gaaout.c -i gaa.h
-gaaout.o: gaaout.c
-	$(CC) -o gaaout.o -c gaaout.c
+#gaaout.c: shash.gaa
+#	gaa shash.gaa -o gaaout.c -i gaa.h
+#gaaout.o: gaaout.c
+#	$(CC) -o gaaout.o -c gaaout.c
