--- origsrc/cogutil-2.0.3-stable/CMakeLists.txt	2016-12-12 12:19:48.000000000 +0900
+++ src/cogutil-2.0.3-stable/CMakeLists.txt	2025-08-12 11:22:55.383762100 +0900
@@ -371,3 +371,10 @@ SUMMARY_ADD("StackPrint" "Pretty printin
 SUMMARY_ADD("Unit tests" "Unit tests" CXXTEST_FOUND)
 SUMMARY_ADD("Util" "General utility library" HAVE_UTIL)
 SUMMARY_SHOW()
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/cogutil.pc.in ${CMAKE_BINARY_DIR}/cogutil.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/cogutil.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/cogutil-2.0.3-stable/cogutil.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/cogutil-2.0.3-stable/cogutil.pc.in	2025-08-12 11:23:00.155194200 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: cogutil
+Description: Very low-level C++ programming utilities used by several components
+Version: @VERSION@
+Libs: -L${libdir} -lcogutil
+Cflags: -I${includedir}
--- origsrc/cogutil-2.0.3-stable/opencog/util/CMakeLists.txt	2016-12-12 12:19:48.000000000 +0900
+++ src/cogutil-2.0.3-stable/opencog/util/CMakeLists.txt	2025-08-12 11:12:11.103889200 +0900
@@ -121,12 +121,12 @@ INSTALL(FILES
 	DESTINATION "include/opencog/util"
 )
 
+if (DEFINED SOVERSION)
+	set_target_properties(cogutil PROPERTIES SOVERSION ${SOVERSION})
+endif ()
+
 IF (CYGWIN)
-    INSTALL(TARGETS cogutil DESTINATION "bin/opencog")
-    ADD_CUSTOM_COMMAND (
-        TARGET cogutil 
-        COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_BINARY_DIR}/opencog/util/cygcogutil.dll ${CMAKE_BINARY_DIR}/tests/util/ 
-        MAIN_DEPENDENCY cogutil)
+	INSTALL(TARGETS cogutil RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 ELSE (CYGWIN) #Linux
     INSTALL(TARGETS cogutil LIBRARY DESTINATION "lib${LIB_DIR_SUFFIX}")
 ENDIF (CYGWIN)
