--- origsrc/lib2geom/CMakeLists.txt	2015-02-26 17:57:39.000000000 +0900
+++ src/lib2geom/CMakeLists.txt	2015-02-26 18:10:30.595232400 +0900
@@ -27,7 +27,7 @@ IF(gsl_FOUND)
 ENDIF(gsl_FOUND) 
 
 # Set Compiler Flags
-SET(CXX_WARNINGS_FLAGS "-Wall -Wformat -Wformat-security -W -Wpointer-arith -Wcast-align -Wsign-compare -Woverloaded-virtual -Wswitch")
+SET(CXX_WARNINGS_FLAGS "-Wall -Wformat -Wformat-security -W -Wpointer-arith -Wcast-align -Wsign-compare -Woverloaded-virtual -Wswitch -fpermissive")
 SET(CMAKE_CXX_FLAGS "${GTK2_CFLAGS} ${cairo_CFLAGS} ${gsl_CFLAGS} ${CXX_WARNINGS_FLAGS}")
 SET(CMAKE_MAKE_PROGRAM "${CMAKE_MAKE_PROGRAM} -j2")
 
