--- origsrc/libsigrokdecode-0.5.0/libsigrokdecode.h	2017-06-12 10:21:27.000000000 +0900
+++ src/libsigrokdecode-0.5.0/libsigrokdecode.h	2017-10-08 20:10:41.303000000 +0900
@@ -105,7 +105,9 @@ enum srd_loglevel {
  */
 
 /* Marks public libsigrokdecode API symbols. */
-#ifndef _WIN32
+#if defined(__CYGWIN__)
+#define SRD_API __declspec(dllexport)
+#elif !defined(_WIN32)
 #define SRD_API __attribute__((visibility("default")))
 #else
 #define SRD_API
