--- origsrc/libsigrokdecode-0.5.2/configure.ac	2018-09-13 05:39:03.000000000 +0900
+++ src/libsigrokdecode-0.5.2/configure.ac	2018-11-06 15:35:45.795313500 +0900
@@ -45,8 +45,10 @@ AM_PROG_CC_C_O
 
 # Set the standard the C library headers should conform to.
 AH_VERBATIM([_POSIX_C_SOURCE], [/* The targeted POSIX standard. */
+#if !defined(__CYGWIN__)
 #ifndef _POSIX_C_SOURCE
 # define _POSIX_C_SOURCE 200112L
+#endif
 #endif])
 
 # Get compiler versions.
--- origsrc/libsigrokdecode-0.5.2/libsigrokdecode.h	2018-09-13 05:39:03.000000000 +0900
+++ src/libsigrokdecode-0.5.2/libsigrokdecode.h	2018-11-06 15:35:45.813357600 +0900
@@ -106,7 +106,9 @@ enum srd_loglevel {
  */
 
 /* Marks public libsigrokdecode API symbols. */
-#ifndef _WIN32
+#if defined(__CYGWIN__)
+#define SRD_API __declspec(dllexport)
+#elif !defined(_WIN32)
 #define SRD_API __attribute__((visibility("default")))
 #else
 #define SRD_API
