--- origsrc/octetos-biblion-0.1.1-alpha/src/Makefile.am	2020-08-21 20:12:40.000000000 +0900
+++ src/octetos-biblion-0.1.1-alpha/src/Makefile.am	2020-09-17 06:31:03.133511100 +0900
@@ -19,21 +19,21 @@ libbiblion_la_CFLAGS = -fPIC  $(AM_CFLAG
 libbiblion_la_CPPFLAGS = -fPIC $(AM_CPPFLAGS)
 libbiblion_la_CXXFLAGS = -fPIC $(AM_CPPFLAGS)
 libbiblion_la_LDFLAGS = -shared -fPIC
+libbiblion_la_LIBADD = $(OCTETOS_BIBLION_LIBS)
 if PLATFORM_WIN32
 libbiblion_la_LDFLAGS += -no-undefined
 endif
 if NATIVE_WIN32
 libbiblion_la_LDFLAGS += -export-dynamic
-
-libbiblion_la_LIBADD = $(OCTETOS_BIBLION_LIBS)
 endif
-libbiblion_la_LIBDADD =
 libbiblion_la_SOURCES = \
 	cita.c \
 	cita.tab.c \
 	cita-lexer.cc \
 	biblion.cc
 
+octetos_biblion_includedir = $(includedir)/octetos/biblion
+octetos_biblion_include_HEADERS = biblion.hh cita-lexer.h
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = octetos-biblion.pc
