--- origsrc/libsidplayfp-1.1.0/Makefile.am	2013-09-17 05:02:13.000000000 +0900
+++ src/libsidplayfp-1.1.0/Makefile.am	2013-10-12 12:15:24.525091800 +0900
@@ -3,11 +3,6 @@ ACLOCAL_AMFLAGS = -I m4
 
 AUTOMAKE_OPTIONS = foreign subdir-objects
 
-
-if MINGW32
-  W32_LDFLAGS = -no-undefined
-endif
-
 if HAVE_VISIBILITY
   AM_CXXFLAGS = -fvisibility=hidden -fvisibility-inlines-hidden
 endif
@@ -134,7 +129,7 @@ utils/MD5/MD5.cpp \
 utils/MD5/MD5.h \
 utils/MD5/MD5_Defs.h
 
-sidplayfp_libsidplayfp_la_LDFLAGS = -version-info $(LIBSIDPLAYVERSION) $(W32_LDFLAGS)
+sidplayfp_libsidplayfp_la_LDFLAGS = -version-info $(LIBSIDPLAYVERSION) $(W32_LDFLAGS) -no-undefined
 
 sidplayfp_libsidplayfp_ladir = $(includedir)/sidplayfp
 
@@ -290,7 +285,7 @@ endif
 sidplayfp_libstilview_la_SOURCES = \
 utils/STILview/stil.cpp
 
-sidplayfp_libstilview_la_LDFLAGS = -version-info $(LIBSTILVIEWVERSION) $(W32_LDFLAGS)
+sidplayfp_libstilview_la_LDFLAGS = -version-info $(LIBSTILVIEWVERSION) -no-undefined
 
 sidplayfp_libstilview_ladir = $(includedir)/stilview
 
