--- origsrc/libnfc-libnfc-1.7.1/examples/Makefile.am	2013-02-01 09:42:00.000000000 +0900
+++ src/libnfc-libnfc-1.7.1/examples/Makefile.am	2015-02-10 08:45:07.148769200 +0900
@@ -28,61 +28,47 @@ AM_CPPFLAGS = $(all_includes) $(LIBNFC_C
 AM_CFLAGS = -I$(top_srcdir)/libnfc -I$(top_srcdir)
 
 nfc_poll_SOURCES = nfc-poll.c
-nfc_poll_LDADD = $(top_builddir)/libnfc/libnfc.la \
-		 $(top_builddir)/utils/libnfcutils.la
+nfc_poll_LDADD = $(top_builddir)/utils/libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 nfc_anticol_SOURCES = nfc-anticol.c
-nfc_anticol_LDADD = $(top_builddir)/libnfc/libnfc.la \
-		    $(top_builddir)/utils/libnfcutils.la
+nfc_anticol_LDADD = $(top_builddir)/utils/libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 nfc_relay_SOURCES = nfc-relay.c
-nfc_relay_LDADD = $(top_builddir)/libnfc/libnfc.la \
-		  $(top_builddir)/utils/libnfcutils.la
+nfc_relay_LDADD = $(top_builddir)/utils/libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 nfc_emulate_forum_tag2_SOURCES = nfc-emulate-forum-tag2.c
-nfc_emulate_forum_tag2_LDADD = $(top_builddir)/libnfc/libnfc.la \
-			$(top_builddir)/utils/libnfcutils.la
+nfc_emulate_forum_tag2_LDADD = $(top_builddir)/utils/libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 nfc_emulate_tag_SOURCES = nfc-emulate-tag.c
-nfc_emulate_tag_LDADD = $(top_builddir)/libnfc/libnfc.la \
-			$(top_builddir)/utils/libnfcutils.la
+nfc_emulate_tag_LDADD = $(top_builddir)/utils/libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 nfc_emulate_uid_SOURCES = nfc-emulate-uid.c
-nfc_emulate_uid_LDADD = $(top_builddir)/libnfc/libnfc.la \
-		        $(top_builddir)/utils/libnfcutils.la
+nfc_emulate_uid_LDADD = $(top_builddir)/utils/libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 nfc_dep_target_SOURCES = nfc-dep-target.c
-nfc_dep_target_LDADD = $(top_builddir)/libnfc/libnfc.la \
-		       $(top_builddir)/utils/libnfcutils.la
+nfc_dep_target_LDADD = $(top_builddir)/utils/libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 nfc_dep_initiator_SOURCES = nfc-dep-initiator.c
-nfc_dep_initiator_LDADD = $(top_builddir)/libnfc/libnfc.la \
-			  $(top_builddir)/utils/libnfcutils.la
+nfc_dep_initiator_LDADD = $(top_builddir)/utils/libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 nfc_mfsetuid_SOURCES = nfc-mfsetuid.c
-nfc_mfsetuid_LDADD = $(top_builddir)/libnfc/libnfc.la \
-			  $(top_builddir)/utils/libnfcutils.la
+nfc_mfsetuid_LDADD = $(top_builddir)/utils/libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 pn53x_diagnose_SOURCES = pn53x-diagnose.c
-pn53x_diagnose_LDADD = $(top_builddir)/libnfc/libnfc.la \
-		       $(top_builddir)/utils/libnfcutils.la
+pn53x_diagnose_LDADD = $(top_builddir)/utils/libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 pn53x_sam_SOURCES = pn53x-sam.c
-pn53x_sam_LDADD = $(top_builddir)/libnfc/libnfc.la \
-		  $(top_builddir)/utils/libnfcutils.la
+pn53x_sam_LDADD = $(top_builddir)/utils/libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 pn53x_tamashell_SOURCES = pn53x-tamashell.c
-pn53x_tamashell_LDADD = $(top_builddir)/libnfc/libnfc.la \
-		        $(top_builddir)/utils/libnfcutils.la
+pn53x_tamashell_LDADD = $(top_builddir)/utils/libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 pn53x_tamashell_LDFLAGS = @READLINE_LIBS@
 
 quick_start_example1_SOURCES = doc/quick_start_example1.c
-quick_start_example1_LDADD =  $(top_builddir)/libnfc/libnfc.la \
-		  $(top_builddir)/utils/libnfcutils.la
+quick_start_example1_LDADD =  $(top_builddir)/utils/libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 quick_start_example2_SOURCES = doc/quick_start_example2.c
-quick_start_example2_LDADD =  $(top_builddir)/libnfc/libnfc.la \
-		  $(top_builddir)/utils/libnfcutils.la
+quick_start_example2_LDADD =  $(top_builddir)/utils/libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 
 dist_man_MANS = \
--- origsrc/libnfc-libnfc-1.7.1/libnfc/buses/uart.c	2013-10-01 21:09:14.000000000 +0900
+++ src/libnfc-libnfc-1.7.1/libnfc/buses/uart.c	2015-02-10 08:37:17.390956700 +0900
@@ -77,7 +77,7 @@
 const char *serial_ports_device_radix[] = { "tty.SLAB_USBtoUART", "tty.usbserial-", NULL };
 #  elif defined (__FreeBSD__) || defined (__OpenBSD__) || defined(__FreeBSD_kernel__)
 const char *serial_ports_device_radix[] = { "cuaU", "cuau", NULL };
-#  elif defined (__linux__)
+#  elif defined (__linux__) || defined(__CYGWIN__)
 const char *serial_ports_device_radix[] = { "ttyUSB", "ttyS", "ttyACM", "ttyAMA", "ttyO", NULL };
 #  else
 #    error "Can't determine serial string for your system"
--- origsrc/libnfc-libnfc-1.7.1/utils/Makefile.am	2014-02-03 08:04:08.000000000 +0900
+++ src/libnfc-libnfc-1.7.1/utils/Makefile.am	2015-02-10 08:42:17.029628600 +0900
@@ -16,34 +16,28 @@ noinst_LTLIBRARIES = libnfcutils.la
 libnfcutils_la_SOURCES = nfc-utils.c
 
 nfc_emulate_forum_tag4_SOURCES = nfc-emulate-forum-tag4.c nfc-utils.h
-nfc_emulate_forum_tag4_LDADD = $(top_builddir)/libnfc/libnfc.la \
-			       libnfcutils.la
+nfc_emulate_forum_tag4_LDADD = libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 nfc_jewel_SOURCES = nfc-jewel.c jewel.c jewel.h nfc-utils.h
 nfc_jewel_LDADD = $(top_builddir)/libnfc/libnfc.la
 
 nfc_list_SOURCES = nfc-list.c nfc-utils.h
-nfc_list_LDADD = $(top_builddir)/libnfc/libnfc.la \
-		 libnfcutils.la
+nfc_list_LDADD = libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 nfc_mfclassic_SOURCES = nfc-mfclassic.c mifare.c mifare.h nfc-utils.h
-nfc_mfclassic_LDADD = $(top_builddir)/libnfc/libnfc.la \
-		    libnfcutils.la
+nfc_mfclassic_LDADD = libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 nfc_mfultralight_SOURCES = nfc-mfultralight.c mifare.c mifare.h nfc-utils.h
 nfc_mfultralight_LDADD = $(top_builddir)/libnfc/libnfc.la
 
 nfc_read_forum_tag3_SOURCES = nfc-read-forum-tag3.c nfc-utils.h
-nfc_read_forum_tag3_LDADD = $(top_builddir)/libnfc/libnfc.la \
-		            libnfcutils.la
+nfc_read_forum_tag3_LDADD = libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 nfc_relay_picc_SOURCES = nfc-relay-picc.c nfc-utils.h
-nfc_relay_picc_LDADD = $(top_builddir)/libnfc/libnfc.la \
-		       libnfcutils.la
+nfc_relay_picc_LDADD = libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 nfc_scan_device_SOURCES = nfc-scan-device.c nfc-utils.h
-nfc_scan_device_LDADD = $(top_builddir)/libnfc/libnfc.la \
-		 libnfcutils.la
+nfc_scan_device_LDADD = libnfcutils.la $(top_builddir)/libnfc/libnfc.la
 
 dist_man_MANS = \
 		nfc-emulate-forum-tag4.1 \
