--- origsrc/libghthash-0.6.2/Makefile.am	2011-11-27 16:34:19.000000000 +0900
+++ src/libghthash-0.6.2/Makefile.am	2025-09-18 13:23:52.438451500 +0900
@@ -1,4 +1,7 @@
 AUTOMAKE_OPTIONS = gnu
 SUBDIRS = src examples
-man_MANS = *.3
+man_MANS = $(top_srcdir)/*.3
 EXTRA_DIST = html/* Makefile.win $(man_MANS)
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libghthash-0.6.2/configure.in	2025-09-18 13:08:06.616849800 +0900
+++ src/libghthash-0.6.2/configure.in	2025-09-18 13:25:32.258603400 +0900
@@ -132,7 +132,7 @@ AC_ARG_ENABLE(debug,
 if test x$DEBUG_ON = xyes; then
       CFLAGS="$CFLAGS -g"
 else
-#      CFLAGS='echo $CFLAGS | sed "s/-g//g"'
+#      CFLAGS='echo $CFLAGS |ï¿½sed "s/-g//g"'
       CFLAGS="$CFLAGS -DNDEBUG"
 fi
 
@@ -143,3 +143,4 @@ fi
 AC_SUBST(CFLAGS)
 
 AC_OUTPUT(Makefile src/ght_hash_table.h src/Makefile examples/Makefile)
+AC_OUTPUT(libghthash.pc)
--- origsrc/libghthash-0.6.2/libghthash.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libghthash-0.6.2/libghthash.pc.in	2025-09-18 13:23:52.365588100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libghthash
+Description: Generic Hash Table
+Version: @VERSION@
+Libs: -L${libdir} -lghthash
+Cflags: -I${includedir}
--- origsrc/libghthash-0.6.2/src/Makefile.am	2011-11-27 16:34:19.000000000 +0900
+++ src/libghthash-0.6.2/src/Makefile.am	2025-09-18 13:09:01.091403000 +0900
@@ -5,6 +5,6 @@ libghthash_la_SOURCES = hash_table.c has
 include_HEADERS = ght_hash_table.h
 noinst_HEADERS =
 
-libghthash_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
+libghthash_la_LDFLAGS = -no-undefined -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 
 EXTRA_DIST = Makefile.win
