diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*.whl' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x Doxyfile -x eia608_from_utf8.c origsrc/libcaption-0.7/CMakeLists.txt src/libcaption-0.7/CMakeLists.txt
--- origsrc/libcaption-0.7/CMakeLists.txt	2017-10-13 07:11:01.000000000 +0900
+++ src/libcaption-0.7/CMakeLists.txt	2020-01-22 17:28:24.397733400 +0900
@@ -73,6 +73,10 @@ set(CAPTION_HEADERS
 
 add_library(caption ${CAPTION_SOURCES})
 
+if(CYGWIN)
+  set_target_properties(caption PROPERTIES SOVERSION 0)
+endif()
+
 if(ENABLE_RE2C)
   add_dependencies(caption re2c)
 endif()
@@ -93,7 +97,7 @@ endif()
 #add_executable(test_captions unit_tests/test_sei.c )
 #target_link_libraries(test_captions caption)
 
-install (TARGETS caption DESTINATION lib EXPORT caption-targets)
+install (TARGETS caption EXPORT caption-targets RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 install (FILES ${CAPTION_HEADERS} DESTINATION include/caption)
 
 find_package(Doxygen)
@@ -105,3 +109,10 @@ if(DOXYGEN_FOUND)
   COMMENT "Generating API documentation with Doxygen" VERBATIM
 )
 endif(DOXYGEN_FOUND)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libcaption.pc.in ${CMAKE_BINARY_DIR}/libcaption.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libcaption.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*.whl' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 -x Doxyfile -x eia608_from_utf8.c origsrc/libcaption-0.7/libcaption.pc.in src/libcaption-0.7/libcaption.pc.in
--- origsrc/libcaption-0.7/libcaption.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libcaption-0.7/libcaption.pc.in	2020-01-22 17:36:01.384996700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libcaption
+Description: Free open-source CEA608 / CEA708 closed-caption encoder/decoder
+Version: @VERSION@
+Libs: -L${libdir} -lcaption
+Cflags: -I${includedir}
