--- origsrc/libexe-20160110/Makefile.am	2016-01-10 17:35:10.000000000 +0900
+++ src/libexe-20160110/Makefile.am	2016-04-07 18:59:22.754206200 +0900
@@ -3,27 +3,11 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
 	libexe \
-	libcsystem \
 	exetools \
 	pyexe \
 	pyexe2 \
 	pyexe3 \
-	po \
 	manuals \
 	tests \
 	msvscpp
@@ -129,4 +113,3 @@ splint:
 	(cd $(srcdir)/pyexe && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
-
--- origsrc/libexe-20160110/configure.ac	2016-01-10 17:35:10.000000000 +0900
+++ src/libexe-20160110/configure.ac	2016-04-07 18:55:40.545207900 +0900
@@ -190,22 +190,7 @@ dnl Have configure make the Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
 AC_CONFIG_FILES([libexe/Makefile])
-AC_CONFIG_FILES([libcsystem/Makefile])
 AC_CONFIG_FILES([exetools/Makefile])
 AC_CONFIG_FILES([pyexe/Makefile])
 AC_CONFIG_FILES([pyexe2/Makefile])
@@ -261,5 +246,3 @@ Features:
    Verbose output:                           $ac_cv_enable_verbose_output
    Debug output:                             $ac_cv_enable_debug_output
 ]);
-
-
--- origsrc/libexe-20160110/exetools/Makefile.am	2016-01-10 17:35:10.000000000 +0900
+++ src/libexe-20160110/exetools/Makefile.am	2016-04-07 18:56:15.540846700 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -55,4 +56,3 @@ distclean: clean
 splint:
 	@echo "Running splint on exeinfo ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(exeinfo_SOURCES)
-
--- origsrc/libexe-20160110/libexe/Makefile.am	2016-01-10 17:35:10.000000000 +0900
+++ src/libexe-20160110/libexe/Makefile.am	2016-04-07 18:56:28.851028700 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -87,4 +88,3 @@ distclean: clean
 splint:
 	@echo "Running splint on libexe ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(libexe_la_SOURCES)
-
--- origsrc/libexe-20160110/tests/Makefile.am	2016-01-10 17:35:10.000000000 +0900
+++ src/libexe-20160110/tests/Makefile.am	2016-04-07 19:00:22.364557800 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -80,4 +81,3 @@ MAINTAINERCLEANFILES = \
 
 distclean: clean
 	/bin/rm -f Makefile
-
