--- origsrc/libexe-20240420/Makefile.am	2024-04-21 02:18:22.000000000 +0900
+++ src/libexe-20240420/Makefile.am	2025-05-18 12:07:42.644573400 +0900
@@ -3,23 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
 	libexe \
 	exetools \
 	pyexe \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libexe-20240420/configure.ac	2024-04-21 02:18:22.000000000 +0900
+++ src/libexe-20240420/configure.ac	2025-05-18 12:07:42.644573400 +0900
@@ -160,19 +160,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
 AC_CONFIG_FILES([libexe/Makefile])
 AC_CONFIG_FILES([pyexe/Makefile])
 AC_CONFIG_FILES([exetools/Makefile])
--- origsrc/libexe-20240420/pyexe/Makefile.am	2024-04-21 02:18:22.000000000 +0900
+++ src/libexe-20240420/pyexe/Makefile.am	2025-05-18 12:07:51.171949300 +0900
@@ -43,7 +43,7 @@ pyexe_la_LIBADD = \
 	@LIBBFIO_LIBADD@
 
 pyexe_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pyexe_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pyexe_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
--- origsrc/libexe-20240420/tests/Makefile.am	2024-04-21 02:18:22.000000000 +0900
+++ src/libexe-20240420/tests/Makefile.am	2025-05-18 12:17:32.656549800 +0900
@@ -3,6 +3,7 @@ AUTOMAKE_OPTIONS = subdir-objects
 AM_CPPFLAGS = \
 	-I../include -I$(top_srcdir)/include \
 	-I../common -I$(top_srcdir)/common \
+	-I$(top_builddir)/libexe \
 	@LIBCERROR_CPPFLAGS@ \
 	@LIBCTHREADS_CPPFLAGS@ \
 	@LIBCDATA_CPPFLAGS@ \
