--- origsrc/chachapoly/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/chachapoly/Makefile.am	2021-04-18 17:34:20.534703000 +0900
@@ -0,0 +1,15 @@
+
+check_PROGRAMS = chachapoly-test
+chachapoly_test_SOURCES = chachapoly-test.c
+chachapoly_test_LDADD = libchachapoly.la
+
+pkginclude_HEADERS = chacha.h chachapoly.h poly1305.h
+
+lib_LTLIBRARIES = libchachapoly.la
+libchachapoly_la_SOURCES = chacha.c chachapoly.c poly1305.c
+libchachapoly_la_LDFLAGS = -no-undefined
+
+TESTS = $(check_PROGRAMS)
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/chachapoly/chachapoly.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/chachapoly/chachapoly.pc.in	2021-04-18 17:34:20.471696700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: chachapoly
+Description: RFC 7539 compliant ChaCha20-Poly1305 AEAD implementation
+Version: @VERSION@
+Libs: -L${libdir} -lchachapoly
+Cflags: -I${includedir}
--- origsrc/chachapoly/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/chachapoly/configure.ac	2021-04-18 17:34:20.535705200 +0900
@@ -0,0 +1,28 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([chachapoly], [0])
+AC_CONFIG_SRCDIR([chacha.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([stddef.h stdint.h stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+
+# Checks for library functions.
+AC_CHECK_FUNCS([memset])
+
+AC_CONFIG_FILES([chachapoly.pc Makefile])
+AC_OUTPUT
