diff --strip-trailing-cr -uprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.log -x config.status -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x config.guess -x config.sub -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/csvprintf-1.0.3/configure.ac src/csvprintf-1.0.3/configure.ac
--- origsrc/csvprintf-1.0.3/configure.ac	2013-01-06 00:37:09.000000000 +0900
+++ src/csvprintf-1.0.3/configure.ac	2013-03-09 19:44:15.983753200 +0900
@@ -55,7 +55,8 @@ fi
 
 # Check for required libc functions
 AC_CHECK_LIB(c, iconv_open,,
-    [AC_MSG_ERROR([required function iconv_open missing])])
+    [AC_CHECK_LIB(iconv, libiconv_open,,
+        [AC_MSG_ERROR([required function iconv_open missing])])])
 
 # Check for required header files
 AC_HEADER_STDC
