--- origsrc/csvprintf-1.3.4/Makefile.am	2026-03-02 00:52:42.000000000 +0900
+++ src/csvprintf-1.3.4/Makefile.am	2026-03-16 11:04:18.192328900 +0900
@@ -52,5 +52,5 @@ uninstall-hook:
 			rm -f "$(DESTDIR)$(man1dir)"/xml2csv.1
 
 gitrev.c:
-			printf 'const char *const csvprintf_version = "%s";\n' "`git describe`" > gitrev.c
-
+			ver="`git describe 2>/dev/null || echo $(PACKAGE_VERSION)`"; \
+			printf 'const char *const csvprintf_version = "%s";\n' "$$ver" > gitrev.c
