--- origsrc/libmrss-0.19.2/configure.ac	2008-08-22 07:00:41.000000000 +0900
+++ src/libmrss-0.19.2/configure.ac	2010-10-31 14:44:07.247624200 +0900
@@ -15,8 +15,7 @@ AM_MAINTAINER_MODE
 AC_PROG_CC
 AC_GNU_SOURCE
 AC_ISC_POSIX
-AC_PROG_RANLIB
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 AC_PROG_INSTALL
 AM_PROG_CC_STDC
 
--- origsrc/libmrss-0.19.2/src/Makefile.am	2007-07-04 17:42:16.000000000 +0900
+++ src/libmrss-0.19.2/src/Makefile.am	2010-10-31 14:43:34.923277700 +0900
@@ -10,7 +10,7 @@ libmrss_la_SOURCES = \
 	mrss_options.c \
 	mrss_search.c
 
-libmrss_la_LDFLAGS = -version-info @LIBMRSS_VERSION_INFO@
+libmrss_la_LDFLAGS = -version-info @LIBMRSS_VERSION_INFO@ -no-undefined
 
 EXTRA_DIST = mrss.h.in mrss.h mrss_internal.h
 
