--- origsrc/libsquish-1.15/CMakeLists.txt	2017-03-14 16:40:13.000000000 +0900
+++ src/libsquish-1.15/CMakeLists.txt	2024-10-05 00:31:56.529968800 +0900
@@ -95,6 +95,7 @@ IF (BUILD_SQUISH_EXTRA)
     ADD_EXECUTABLE(squishtest ${SQUISHTEST_SRCS})
     SET_TARGET_PROPERTIES(squishtest PROPERTIES DEBUG_POSTFIX "d")
     TARGET_LINK_LIBRARIES(squishtest squish)
+    INSTALL(TARGETS squishtest DESTINATION bin)
 
     SET(SQUISHPNG_SRCS extra/squishpng.cpp)
 
@@ -106,12 +107,18 @@ IF (BUILD_SQUISH_EXTRA)
         ADD_EXECUTABLE(squishpng ${SQUISHPNG_SRCS})
         SET_TARGET_PROPERTIES(squishpng PROPERTIES DEBUG_POSTFIX "d")
         TARGET_LINK_LIBRARIES(squishpng squish ${PNG_LIBRARIES})
+        INSTALL(TARGETS squishpng DESTINATION bin)
     ENDIF (PNG_FOUND)
 ENDIF (BUILD_SQUISH_EXTRA)
 
 INSTALL(
     TARGETS squish
-    LIBRARY DESTINATION lib
+    RUNTIME DESTINATION bin
     ARCHIVE DESTINATION lib
     PUBLIC_HEADER DESTINATION include
     )
+
+SET(PREFIX ${CMAKE_INSTALL_PREFIX})
+SET(LIB_PATH lib)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libsquish.pc.in ${CMAKE_BINARY_DIR}/libsquish.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libsquish.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libsquish-1.15/libsquish.pc.in	2016-09-24 19:46:01.000000000 +0900
+++ src/libsquish-1.15/libsquish.pc.in	2024-10-05 00:31:56.529968800 +0900
@@ -6,8 +6,8 @@ includedir=${prefix}/include
 
 Name: libsquish
 Description: squish DXT library
-Version: 1.14
+Version: 1.15
 
 Requires:
-Libs: -L${libdir} -L${sharedlibdir} -llibsquish
+Libs: -L${libdir} -L${sharedlibdir} -lsquish
 Cflags: -I${includedir}
