--- origsrc/libmaxminddb-0.5.5/src/Makefile.am	2014-02-13 08:20:15.000000000 +0900
+++ src/libmaxminddb-0.5.5/src/Makefile.am	2014-07-11 23:26:29.873534000 +0900
@@ -3,5 +3,5 @@ include $(top_srcdir)/common.mk
 lib_LTLIBRARIES = libmaxminddb.la
 
 libmaxminddb_la_SOURCES = maxminddb.c
-libmaxminddb_la_LDFLAGS = -version-info 0:7:0
+libmaxminddb_la_LDFLAGS = -no-undefined -version-info 0:7:0
 include_HEADERS = $(top_srcdir)/include/maxminddb.h
