--- origsrc/libmaxminddb-0.5.2/src/Makefile.am	2013-11-10 05:54:18.000000000 +0900
+++ src/libmaxminddb-0.5.2/src/Makefile.am	2013-11-26 22:18:11.375760200 +0900
@@ -3,5 +3,5 @@ include $(top_srcdir)/common.mk
 lib_LTLIBRARIES = libmaxminddb.la
 
 libmaxminddb_la_SOURCES = maxminddb.c
-libmaxminddb_la_LDFLAGS = -version-info 0:6:0
+libmaxminddb_la_LDFLAGS = -no-undefined -version-info 0:6:0
 include_HEADERS = $(top_srcdir)/include/maxminddb.h
