--- origsrc/libmaxminddb-1.3.1/src/Makefile.am	2017-11-25 04:12:38.000000000 +0900
+++ src/libmaxminddb-1.3.1/src/Makefile.am	2017-11-27 08:38:54.301000200 +0900
@@ -4,7 +4,7 @@ lib_LTLIBRARIES = libmaxminddb.la
 
 libmaxminddb_la_SOURCES = maxminddb.c maxminddb-compat-util.h \
 	data-pool.c data-pool.h
-libmaxminddb_la_LDFLAGS = -version-info 0:7:0
+libmaxminddb_la_LDFLAGS = -no-undefined -version-info 0:7:0
 include_HEADERS = $(top_srcdir)/include/maxminddb.h
 
 pkgconfig_DATA = libmaxminddb.pc
@@ -15,11 +15,4 @@ check_PROGRAMS = test-data-pool
 
 test_data_pool_SOURCES = data-pool.c data-pool.h
 test_data_pool_CPPFLAGS = $(AM_CPPFLAGS) -I$(top_srcdir)/t -DTEST_DATA_POOL
-test_data_pool_LDADD = $(top_srcdir)/t/libmmdbtest.la \
-	$(top_srcdir)/t/libtap/libtap.a
-
-$(top_srcdir)/t/libmmdbtest.la:
-	$(MAKE) -C $(top_srcdir)/t libmmdbtest.la
-
-$(top_srcdir)/t/libtap/libtap.a:
-	$(MAKE) -C $(top_srcdir)/t/libtap libtap.a
+test_data_pool_LDADD = $(top_srcdir)/t/libmmdbtest.la
--- origsrc/libmaxminddb-1.3.1/t/Makefile.am	2017-11-25 04:12:38.000000000 +0900
+++ src/libmaxminddb-1.3.1/t/Makefile.am	2017-11-27 08:39:26.331000000 +0900
@@ -1,14 +1,9 @@
 include $(top_srcdir)/common.mk
 
-all-local:
-		cd libtap && $(MAKE) $(AM_MAKEFLAGS) all
-clean-local:
-		cd libtap && $(MAKE) $(AM_MAKEFLAGS) clean
-
 AM_LDFLAGS = $(top_builddir)/src/libmaxminddb.la
 
 noinst_LTLIBRARIES = libmmdbtest.la
-libmmdbtest_la_SOURCES = maxminddb_test_helper.c
+libmmdbtest_la_SOURCES = maxminddb_test_helper.c libtap/tap.c
 
 check_PROGRAMS = \
 	bad_pointers_t bad_databases_t basic_lookup_t data_entry_list_t \
@@ -21,4 +16,4 @@ threads_t_CFLAGS = $(CFLAGS) -pthread
 
 TESTS = $(check_PROGRAMS) compile_c++_t.pl mmdblookup_t.pl
 
-LDADD = libmmdbtest.la libtap/libtap.a
+LDADD = libmmdbtest.la
