--- origsrc/exhale-v1.2.2/CMakeLists.txt	2025-12-20 19:00:07.000000000 +0900
+++ src/exhale-v1.2.2/CMakeLists.txt	2026-01-07 10:50:16.219417800 +0900
@@ -49,3 +49,10 @@ endif()
 
 add_subdirectory(src/lib)
 add_subdirectory(src/app)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/exhale.pc.in ${CMAKE_BINARY_DIR}/exhale.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/exhale.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/exhale-v1.2.2/exhale.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/exhale-v1.2.2/exhale.pc.in	2026-01-07 10:50:16.219417800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: exhale
+Description: Open-source ISO/IEC 23003-3 (USAC, xHE-AAC) encoder
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lexhale
+Cflags: -I${includedir}
--- origsrc/exhale-v1.2.2/src/lib/CMakeLists.txt	2025-12-20 19:00:07.000000000 +0900
+++ src/exhale-v1.2.2/src/lib/CMakeLists.txt	2026-01-07 10:50:16.224860100 +0900
@@ -38,6 +38,10 @@ add_library(exhaleLib
 
 set_target_properties(exhaleLib PROPERTIES OUTPUT_NAME exhale)
 
+if (DEFINED SOVERSION)
+    set_target_properties(exhaleLib PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 if(TARGET Threads::Threads)
     target_link_libraries(exhaleLib PRIVATE Threads::Threads)
 endif()
@@ -56,6 +60,7 @@ endif()
 target_compile_definitions(exhaleLib PRIVATE EXHALE_DYN_LINK)
 
 install(TARGETS exhaleLib
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}
     LIBRARY DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}
     PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR})
