--- origsrc/libconfini-1.14.0/configure.ac	2020-03-02 10:24:10.000000000 +0900
+++ src/libconfini-1.14.0/configure.ac	2020-03-10 18:29:31.386675900 +0900
@@ -599,7 +599,7 @@ AM_COND_IF([PLATFORM_MSWIN], [
 
 # `LT_INIT([win32-dll])` should set the `${DLLTOOL}` variable
 AM_CONDITIONAL([HAVE_DLLTOOL],
-	[test "x${DLLTOOL}" != xfalse -a "x${DLLTOOL}" != x])
+	[test "x${DLLTOOL}" != xfalse -a "x${DLLTOOL}" != x -a "$host_os" != "cygwin"])
 
 # `LT_PROG_RC` should set the `${RC}` variable
 AM_CONDITIONAL([HAVE_RC],
