--- origsrc/libconfini-1.16.3/configure.ac	2021-11-09 14:26:21.000000000 +0900
+++ src/libconfini-1.16.3/configure.ac	2021-11-17 08:16:43.572559900 +0900
@@ -610,7 +610,7 @@ AM_COND_IF([PLATFORM_MSWIN], [
 
 # `LT_INIT([win32-dll])` should set the `${DLLTOOL}` variable
 AM_CONDITIONAL([HAVE_DLLTOOL],
-	[test "x${DLLTOOL}" != xfalse -a "x${DLLTOOL}" != x])
+	[test "x${DLLTOOL}" != xfalse -a "x${DLLTOOL}" != x -a "$host_os" != "cygwin"])
 
 # `LT_PROG_RC` should set the `${RC}` variable
 AM_CONDITIONAL([HAVE_RC],
