--- origsrc/libconfini-1.16.2/configure.ac	2021-09-25 10:12:16.000000000 +0900
+++ src/libconfini-1.16.2/configure.ac	2021-10-03 11:32:14.153333100 +0900
@@ -609,7 +609,7 @@ AM_COND_IF([PLATFORM_MSWIN], [
 
 # `LT_INIT([win32-dll])` should set the `${DLLTOOL}` variable
 AM_CONDITIONAL([HAVE_DLLTOOL],
-	[test "x${DLLTOOL}" != xfalse -a "x${DLLTOOL}" != x])
+	[test "x${DLLTOOL}" != xfalse -a "x${DLLTOOL}" != x -a "$host_os" != "cygwin"])
 
 # `LT_PROG_RC` should set the `${RC}` variable
 AM_CONDITIONAL([HAVE_RC],
