--- origsrc/libconfini-1.15.0/configure.ac	2020-10-06 22:45:58.000000000 +0900
+++ src/libconfini-1.15.0/configure.ac	2020-10-20 17:27:53.076825200 +0900
@@ -627,7 +627,7 @@ AM_COND_IF([PLATFORM_MSWIN], [
 
 # `LT_INIT([win32-dll])` should set the `${DLLTOOL}` variable
 AM_CONDITIONAL([HAVE_DLLTOOL],
-	[test "x${DLLTOOL}" != xfalse -a "x${DLLTOOL}" != x])
+	[test "x${DLLTOOL}" != xfalse -a "x${DLLTOOL}" != x -a "$host_os" != "cygwin"])
 
 # `LT_PROG_RC` should set the `${RC}` variable
 AM_CONDITIONAL([HAVE_RC],
