--- origsrc/timbl-6.10/src/Makefile.am	2024-12-16 19:01:24.000000000 +0900
+++ src/timbl-6.10/src/Makefile.am	2025-01-19 10:57:49.496357700 +0900
@@ -23,3 +23,5 @@ libtimbl_la_SOURCES = Common.cxx \
 	TimblExperiment.cxx IGExperiment.cxx Metrics.cxx Testers.cxx \
 	TRIBLExperiments.cxx LOOExperiment.cxx CVExperiment.cxx \
 	Types.cxx neighborSet.cxx Statistics.cxx BestArray.cxx
+
+libtimbl_la_LDFLAGS += -no-undefined
