--- origsrc/timbl-6.4.14/src/Makefile.am	2019-10-21 20:04:45.000000000 +0900
+++ src/timbl-6.4.14/src/Makefile.am	2019-12-31 21:57:15.645815600 +0900
@@ -13,7 +13,7 @@ LDADD = libtimbl.la
 
 timbl_SOURCES = Timbl.cxx
 lib_LTLIBRARIES = libtimbl.la
-libtimbl_la_LDFLAGS= -version-info 4:1:0
+libtimbl_la_LDFLAGS= -no-undefined -version-info 4:1:0
 
 libtimbl_la_SOURCES = Common.cxx \
 	GetOptClass.cxx IBtree.cxx IBprocs.cxx Instance.cxx \
