--- origsrc/timbl-6.6/src/Makefile.am	2021-12-15 20:58:49.000000000 +0900
+++ src/timbl-6.6/src/Makefile.am	2022-06-25 08:57:49.334984200 +0900
@@ -13,7 +13,7 @@ LDADD = libtimbl.la
 
 timbl_SOURCES = Timbl.cxx
 lib_LTLIBRARIES = libtimbl.la
-libtimbl_la_LDFLAGS= -version-info 6:0:0
+libtimbl_la_LDFLAGS= -no-undefined -version-info 6:0:0
 
 libtimbl_la_SOURCES = Common.cxx \
 	GetOptClass.cxx IBtree.cxx IBprocs.cxx Instance.cxx \
