--- origsrc/timbl-6.4.9/src/Makefile.am	2017-04-04 16:42:10.000000000 +0900
+++ src/timbl-6.4.9/src/Makefile.am	2017-07-19 19:53:14.487000000 +0900
@@ -13,7 +13,7 @@ LDADD = libtimbl.la
 
 timbl_SOURCES = Timbl.cxx
 lib_LTLIBRARIES = libtimbl.la
-libtimbl_la_LDFLAGS= -version-info 4:1:0
+libtimbl_la_LDFLAGS= -no-undefined -version-info 4:1:0
 
 libtimbl_la_SOURCES = Common.cxx \
 	GetOptClass.cxx IBtree.cxx IBprocs.cxx Instance.cxx \
