--- origsrc/timbl-6.7/src/Makefile.am	2022-07-22 07:17:26.000000000 +0900
+++ src/timbl-6.7/src/Makefile.am	2022-08-02 05:16:43.781607700 +0900
@@ -13,7 +13,7 @@ LDADD = libtimbl.la
 
 timbl_SOURCES = Timbl.cxx
 lib_LTLIBRARIES = libtimbl.la
-libtimbl_la_LDFLAGS= -version-info 6:0:0
+libtimbl_la_LDFLAGS= -no-undefined -version-info 6:0:0
 
 libtimbl_la_SOURCES = Common.cxx \
 	GetOptClass.cxx IBtree.cxx IBprocs.cxx Instance.cxx \
