--- origsrc/timbl-6.4.11/src/Makefile.am	2018-01-09 19:03:11.000000000 +0900
+++ src/timbl-6.4.11/src/Makefile.am	2018-02-11 12:04:19.891805600 +0900
@@ -13,7 +13,7 @@ LDADD = libtimbl.la
 
 timbl_SOURCES = Timbl.cxx
 lib_LTLIBRARIES = libtimbl.la
-libtimbl_la_LDFLAGS= -version-info 4:1:0
+libtimbl_la_LDFLAGS= -no-undefined -version-info 4:1:0
 
 libtimbl_la_SOURCES = Common.cxx \
 	GetOptClass.cxx IBtree.cxx IBprocs.cxx Instance.cxx \
