--- origsrc/libunity-7.1.4/configure.ac	2019-03-19 19:18:01.000000000 +0900
+++ src/libunity-7.1.4/configure.ac	2026-03-24 10:14:53.159016700 +0900
@@ -24,7 +24,7 @@ LIBUNITY_LT_VERSION="$LIBUNITY_LT_CURREN
 # There's some autoconf weirdness which causes loss of [] characters,
 # therefore they're doubled, the regex is to exclude unity_internal_.*
 # symbols, it's so strange cause there's no lookahead support.
-LIBUNITY_LT_LDFLAGS="-version-info $LIBUNITY_LT_VERSION -export-symbols-regex '^unity_([[^i]]|i[[^n]]|in[[^t]]|int[[^e]]).*'"
+LIBUNITY_LT_LDFLAGS="-no-undefined -version-info $LIBUNITY_LT_VERSION -export-symbols-regex '^unity_([[^i]]|i[[^n]]|in[[^t]]|int[[^e]]).*'"
 
 AC_SUBST(LIBUNITY_LT_CURRENT)
 AC_SUBST(LIBUNITY_LT_VERSION)
@@ -34,7 +34,7 @@ LIBUNITY_EXTRAS_LT_CURRENT=$LIBUNITY_LT_
 LIBUNITY_EXTRAS_LT_REV=$LIBUNITY_LT_REV
 LIBUNITY_EXTRAS_LT_AGE=$LIBUNITY_LT_AGE
 LIBUNITY_EXTRAS_LT_VERSION="$LIBUNITY_EXTRAS_LT_CURRENT:$LIBUNITY_EXTRAS_LT_REV:$LIBUNITY_EXTRAS_LT_AGE"
-LIBUNITY_EXTRAS_LT_LDFLAGS="-version-info $LIBUNITY_EXTRAS_LT_VERSION -export-symbols-regex '^unity_extras_.*'"
+LIBUNITY_EXTRAS_LT_LDFLAGS="-no-undefined -version-info $LIBUNITY_EXTRAS_LT_VERSION -export-symbols-regex '^unity_extras_.*'"
 
 AC_SUBST(LIBUNITY_EXTRAS_LT_CURRENT)
 AC_SUBST(LIBUNITY_EXTRAS_LT_VERSION)
@@ -44,7 +44,7 @@ LIBPROTOCOL_LT_CURRENT=0
 LIBPROTOCOL_LT_REV=0
 LIBPROTOCOL_LT_AGE=0
 LIBPROTOCOL_LT_VERSION="$LIBPROTOCOL_LT_CURRENT:$LIBPROTOCOL_LT_REV:$LIBPROTOCOL_LT_AGE"
-LIBPROTOCOL_LT_LDFLAGS="-version-info $LIBPROTOCOL_LT_VERSION -export-symbols-regex '^unity_protocol_.*'"
+LIBPROTOCOL_LT_LDFLAGS="-no-undefined -version-info $LIBPROTOCOL_LT_VERSION -export-symbols-regex '^unity_protocol_.*'"
 
 AC_SUBST(LIBPROTOCOL_LT_CURRENT)
 AC_SUBST(LIBPROTOCOL_LT_VERSION)
--- origsrc/libunity-7.1.4/extras/Makefile.am	2026-03-24 06:37:50.648682500 +0900
+++ src/libunity-7.1.4/extras/Makefile.am	2026-03-24 10:46:06.953252500 +0900
@@ -30,6 +30,7 @@ libunity_extras_la_VALAFLAGS = \
   --vapidir $(top_srcdir)/vapi \
   --vapidir $(top_builddir)/src \
   --vapidir $(top_builddir)/protocol \
+  --vapidir $(top_srcdir)/src \
   --pkg unity-protocol \
   --pkg config \
   --pkg unity-trace \
