--- origsrc/libunity-6.12.0/configure.ac	2012-11-08 15:26:47.000000000 +0900
+++ src/libunity-6.12.0/configure.ac	2017-08-11 14:16:18.387614300 +0900
@@ -21,7 +21,7 @@ LIBUNITY_LT_CURRENT=9
 LIBUNITY_LT_REV=2
 LIBUNITY_LT_AGE=0
 LIBUNITY_LT_VERSION="$LIBUNITY_LT_CURRENT:$LIBUNITY_LT_REV:$LIBUNITY_LT_AGE"
-LIBUNITY_LT_LDFLAGS="-version-info $LIBUNITY_LT_VERSION -export-symbols-regex '^unity_.*'"
+LIBUNITY_LT_LDFLAGS="-no-undefined -version-info $LIBUNITY_LT_VERSION -export-symbols-regex '^unity_.*'"
 
 AC_SUBST(LIBUNITY_LT_CURRENT)
 AC_SUBST(LIBUNITY_LT_VERSION)
@@ -31,7 +31,7 @@ LIBPROTOCOL_LT_CURRENT=0
 LIBPROTOCOL_LT_REV=0
 LIBPROTOCOL_LT_AGE=0
 LIBPROTOCOL_LT_VERSION="$LIBPROTOCOL_LT_CURRENT:$LIBPROTOCOL_LT_REV:$LIBPROTOCOL_LT_AGE"
-LIBPROTOCOL_LT_LDFLAGS="-version-info $LIBPROTOCOL_LT_VERSION -export-symbols-regex '^unity_protocol_.*'"
+LIBPROTOCOL_LT_LDFLAGS="-no-undefined -version-info $LIBPROTOCOL_LT_VERSION -export-symbols-regex '^unity_protocol_.*'"
 
 AC_SUBST(LIBPROTOCOL_LT_CURRENT)
 AC_SUBST(LIBPROTOCOL_LT_VERSION)
--- origsrc/libunity-6.12.0/tools/music-track-model-renderer.vala	2012-10-28 01:16:03.000000000 +0900
+++ src/libunity-6.12.0/tools/music-track-model-renderer.vala	2017-08-11 14:22:36.888000000 +0900
@@ -28,13 +28,13 @@ public class MusicTrackModelRenderer: Ob
     private Dee.ModelTag<int> track_model_tag;
     private int row_counter = 0;
     private ulong model_sync_sig_id = 0;
-    public ListStore track_view_model { get; construct; }
+    public Gtk.ListStore track_view_model { get; construct; }
 
     public signal void track_list_synchronized();
 
     construct
     {
-        track_view_model = new ListStore(6, typeof(string), typeof(int), typeof(string), typeof(uint), typeof(uint), typeof(double));
+        track_view_model = new Gtk.ListStore(6, typeof(string), typeof(int), typeof(string), typeof(uint), typeof(uint), typeof(double));
     }
 
     public MusicTrackModelRenderer(string dbus_model)
--- origsrc/libunity-6.12.0/tools/unity-tool-ui.vala	2012-10-28 01:16:03.000000000 +0900
+++ src/libunity-6.12.0/tools/unity-tool-ui.vala	2017-08-11 14:22:02.700000000 +0900
@@ -40,8 +40,8 @@ public class UnityToolUi: GLib.Object
             window.destroy.connect(Gtk.main_quit);
             window.show_all();
 
-            uimodel = builder.get_object("results_model") as ListStore;
-            ui_filter_model = builder.get_object("filters_model") as ListStore;
+            uimodel = builder.get_object("results_model") as Gtk.ListStore;
+            ui_filter_model = builder.get_object("filters_model") as Gtk.ListStore;
 
             notebook = builder.get_object("notebook") as Notebook;
             search_entry = builder.get_object("search_entry") as Entry;
@@ -154,7 +154,7 @@ public class UnityToolUi: GLib.Object
             builder.add_from_resource("/com/canonical/Unity/unity-tool/dbus-lens-connect.ui");
             builder.connect_signals(this);
             lens_discovery_spinner = builder.get_object("lens_discovery_spinner") as Spinner;
-            lens_list_model = builder.get_object("lens_list_model") as ListStore;
+            lens_list_model = builder.get_object("lens_list_model") as Gtk.ListStore;
             lens_list_combobox = builder.get_object("lens_list_combobox") as ComboBox;
             dbus_name_entry = builder.get_object("dbus_name_entry") as Entry;
             dbus_path_entry = builder.get_object("dbus_path_entry") as Entry;
@@ -779,9 +779,9 @@ public class UnityToolUi: GLib.Object
     private Gtk.Entry dbus_name_entry = null;
     private Gtk.Entry dbus_path_entry = null;
     private Gtk.ComboBox lens_list_combobox = null;
-    private ListStore uimodel = null;
-    private ListStore ui_filter_model = null;
-    private ListStore lens_list_model = null;
+    private Gtk.ListStore uimodel = null;
+    private Gtk.ListStore ui_filter_model = null;
+    private Gtk.ListStore lens_list_model = null;
     private ulong model_sync_sig_id;
     private ulong filters_model_sync_sig_id;
     private Dee.SharedModel? dee_results_model = null;
