--- origsrc/kplot-VERSION_0_1_15/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/kplot-VERSION_0_1_15/Makefile.am	2024-10-03 18:57:37.387426100 +0900
@@ -0,0 +1,70 @@
+include_HEADERS = kplot.h
+
+BUILT_SOURCES = compat.h
+
+lib_LTLIBRARIES = libkplot.la
+libkplot_la_CPPFLAGS = $(CAIRO_CFLAGS)
+libkplot_la_LDFLAGS = -no-undefined
+libkplot_la_LIBADD = $(CAIRO_LIBS)
+libkplot_la_SOURCES = \
+		  array.c \
+		  border.c \
+		  bucket.c \
+		  buffer.c \
+		  colours.c \
+		  draw.c \
+		  grid.c \
+		  hist.c \
+		  label.c \
+		  kdata.c \
+		  kplot.c \
+		  margin.c \
+		  mean.c \
+		  plotctx.c \
+		  reallocarray.c \
+		  stddev.c \
+		  tic.c \
+		  vector.c
+man_MANS = man/kdata_array_alloc.3 \
+		  man/kdata_array_fill.3 \
+		  man/kdata_bucket_alloc.3 \
+		  man/kdata_bucket_add.3 \
+		  man/kdata_buffer_alloc.3 \
+		  man/kdata_buffer_copy.3 \
+		  man/kdata_destroy.3 \
+		  man/kdata_get.3 \
+		  man/kdata_hist_alloc.3 \
+		  man/kdata_hist_add.3 \
+		  man/kdata_mean_alloc.3 \
+		  man/kdata_mean_attach.3 \
+		  man/kdata_pmfmean.3 \
+		  man/kdata_pmfstddev.3 \
+		  man/kdata_pmfvar.3 \
+		  man/kdata_stddev_alloc.3 \
+		  man/kdata_stddev_attach.3 \
+		  man/kdata_vector_append.3 \
+		  man/kdata_vector_alloc.3 \
+		  man/kdata_xmax.3 \
+		  man/kdata_xmean.3 \
+		  man/kdata_xmin.3 \
+		  man/kdata_xstddev.3 \
+		  man/kdatacfg_defaults.3 \
+		  man/kplot.3 \
+		  man/kplot_alloc.3 \
+		  man/kplot_attach_data.3 \
+		  man/kplot_attach_datas.3 \
+		  man/kplot_attach_smooth.3 \
+		  man/kplot_detach.3 \
+		  man/kplot_draw.3 \
+		  man/kplot_free.3 \
+		  man/kplot_get_datacfg.3 \
+		  man/kplot_get_plotcfg.3 \
+	 	  man/kplotcfg_defaults.3
+
+compat.h: $(srcdir)/compat.pre.h $(srcdir)/compat.post.h
+	( cat $(srcdir)/compat.pre.h ; \
+	  CC="$(CC)" CFLAGS="$(CFLAGS)" sh $(srcdir)/compat.sh reallocarray ; \
+	  cat $(srcdir)/compat.post.h ; ) >$@
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/kplot-VERSION_0_1_15/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/kplot-VERSION_0_1_15/configure.ac	2024-10-03 18:57:37.387426100 +0900
@@ -0,0 +1,31 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([kplot], [0.1.15])
+AC_CONFIG_SRCDIR([array.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+AC_PROG_INSTALL
+LT_INIT
+
+PKG_CHECK_MODULES([CAIRO], [cairo])
+
+# Checks for header files.
+AC_CHECK_HEADERS([float.h inttypes.h stdint.h stdlib.h string.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_C_INLINE
+AC_TYPE_SIZE_T
+AC_TYPE_SSIZE_T
+
+# Checks for library functions.
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([floor memmove memset sqrt])
+
+AC_CONFIG_FILES([kplot.pc Makefile])
+AC_OUTPUT
--- origsrc/kplot-VERSION_0_1_15/kplot.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/kplot-VERSION_0_1_15/kplot.pc.in	2024-10-03 18:57:37.387426100 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: kplot
+Description: Cairo plotting library
+Version: @VERSION@
+Libs: -L${libdir} -lkplot
+Cflags: -I${includedir}
