--- origsrc/libscca-20160108/Makefile.am	2016-01-08 23:26:23.000000000 +0900
+++ src/libscca-20160108/Makefile.am	2016-04-08 08:20:16.862097100 +0900
@@ -3,29 +3,11 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
-	libfvalue \
-	libfwnt \
 	libscca \
-	libcsystem \
 	sccatools \
 	pyscca \
 	pyscca2 \
 	pyscca3 \
-	po \
 	manuals \
 	tests \
 	msvscpp
@@ -135,4 +117,3 @@ splint:
 	(cd $(srcdir)/pyscca && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
-
--- origsrc/libscca-20160108/configure.ac	2016-01-08 23:26:23.000000000 +0900
+++ src/libscca-20160108/configure.ac	2016-04-08 08:20:05.323226500 +0900
@@ -196,24 +196,7 @@ dnl Have configure make the Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
-AC_CONFIG_FILES([libfwnt/Makefile])
 AC_CONFIG_FILES([libscca/Makefile])
-AC_CONFIG_FILES([libcsystem/Makefile])
 AC_CONFIG_FILES([sccatools/Makefile])
 AC_CONFIG_FILES([pyscca/Makefile])
 AC_CONFIG_FILES([pyscca2/Makefile])
@@ -272,4 +255,3 @@ Features:
    Verbose output:                            $ac_cv_enable_verbose_output
    Debug output:                              $ac_cv_enable_debug_output
 ]);
-
--- origsrc/libscca-20160108/libscca/Makefile.am	2016-01-08 23:26:23.000000000 +0900
+++ src/libscca-20160108/libscca/Makefile.am	2016-04-08 08:20:30.104131300 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -95,4 +96,3 @@ distclean: clean
 splint:
 	@echo "Running splint on libscca ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(libscca_la_SOURCES)
-
--- origsrc/libscca-20160108/sccatools/Makefile.am	2016-01-08 23:26:23.000000000 +0900
+++ src/libscca-20160108/sccatools/Makefile.am	2016-04-08 08:20:39.190903600 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -54,4 +55,3 @@ distclean: clean
 splint:
 	@echo "Running splint on sccainfo ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(sccainfo_SOURCES)
-
--- origsrc/libscca-20160108/tests/Makefile.am	2016-01-08 23:26:23.000000000 +0900
+++ src/libscca-20160108/tests/Makefile.am	2016-04-08 08:20:46.201040900 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -52,4 +53,3 @@ MAINTAINERCLEANFILES = \
 
 distclean: clean
 	/bin/rm -f Makefile
-
