--- origsrc/dcmtk-3.7.0/CMake/dcmtk.pc.in	2025-12-15 16:28:13.000000000 +0900
+++ src/dcmtk-3.7.0/CMake/dcmtk.pc.in	2026-01-03 23:23:13.035364200 +0900
@@ -1,7 +1,7 @@
-prefix="@CMAKE_INSTALL_PREFIX@"
-exec_prefix="${prefix}"
-libdir="@CMAKE_INSTALL_FULL_LIBDIR@"
-includedir="${prefix}/include/"
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=${prefix}/include/
 
 Name: DCMTK
 Description: DICOM Toolkit (DCMTK)
@@ -9,6 +9,6 @@ URL: https://dcmtk.org
 Version: @DCMTK_MAJOR_VERSION@.@DCMTK_MINOR_VERSION@.@DCMTK_BUILD_VERSION@
 Requires: @PKGCONF_REQ_PUB@
 Requires.private: @PKGCONF_REQ_PRIV@
-Cflags: -I"${includedir}"
-Libs: -L"${libdir}" @PKGCONF_LIBS@
+Cflags: -I${includedir}
+Libs: -L${libdir} @PKGCONF_LIBS@
 Libs.private: -L"${libdir}" @PKGCONF_LIBS_PRIV@
