--- origsrc/iddawc-0.9.6/CMakeLists.txt	2020-07-17 09:11:10.000000000 +0900
+++ src/iddawc-0.9.6/CMakeLists.txt	2020-08-11 10:09:30.176981100 +0900
@@ -123,14 +123,16 @@ set(JANSSON_MIN_VERSION 2.11)
 find_package(Jansson ${JANSSON_MIN_VERSION} REQUIRED)
 if (JANSSON_FOUND)
 	set(LIBS ${LIBS} ${JANSSON_LIBRARIES})
-	include_directories(${JANSSON_INCLUDE_DIRS})
+    include_directories(${JANSSON_INCLUDE_DIRS})
+    target_link_libraries(iddawc ${JANSSON_LIBRARIES})
 endif ()
 
 include(FindGnuTLS)
 find_package(GnuTLS REQUIRED)
 if (GNUTLS_FOUND)
 	set(LIBS ${LIBS} ${GNUTLS_LIBRARIES})
-	include_directories(${GNUTLS_INCLUDE_DIRS})
+    include_directories(${GNUTLS_INCLUDE_DIRS})
+    target_link_libraries(iddawc ${GNUTLS_LIBRARIES})
 endif ()
 
 include(FindNettle)
