--- origsrc/libpoly-0.1.7/CMakeLists.txt	2018-07-10 00:42:12.000000000 +0900
+++ src/libpoly-0.1.7/CMakeLists.txt	2019-05-12 16:51:04.179689000 +0900
@@ -9,7 +9,7 @@ project(libpoly C)
 
 option(LIBPOLY_BUILD_PYTHON_API "Build the Python API" ON)
 option(LIBPOLY_BUILD_STATIC_PIC "Build the static PIC library" ON)
-option(LIBPOLY_BUILD_STATIC "Build the static libraru" ON)
+option(LIBPOLY_BUILD_STATIC "Build the static library" ON)
 
 set(LIBPOLY_VERSION_MAJOR 0)
 set(LIBPOLY_VERSION_MINOR 1)
@@ -64,3 +64,10 @@ if(LIBPOLY_BUILD_PYTHON_API)
   add_subdirectory(test)
 
 endif()
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/libpoly.pc.in ${CMAKE_BINARY_DIR}/libpoly.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/libpoly.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libpoly-0.1.7/libpoly.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libpoly-0.1.7/libpoly.pc.in	2019-05-12 16:51:21.787678800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libpoly
+Description: C library for manipulating polynomials
+Version: @LIBPOLY_VERSION@
+Libs: -L${libdir} -lpoly
+Cflags: -I${includedir}
--- origsrc/libpoly-0.1.7/src/CMakeLists.txt	2018-07-10 00:42:12.000000000 +0900
+++ src/libpoly-0.1.7/src/CMakeLists.txt	2019-05-12 16:49:28.297811700 +0900
@@ -60,7 +60,7 @@ target_link_libraries(poly ${GMP_LIBRARY
 # cmake complains if the RUNTIME DESTINATION bin is missing
 #
 if (CYGWIN OR WIN32 OR MINGW)
- install(TARGETS poly LIBRARY DESTINATION lib RUNTIME DESTINATION bin)
+ install(TARGETS poly ARCHIVE DESTINATION lib RUNTIME DESTINATION bin)
 else()
  install(TARGETS poly LIBRARY DESTINATION lib)
 endif()
