--- origsrc/libpoly-0.2.1/src/CMakeLists.txt	2025-12-02 16:37:26.000000000 +0900
+++ src/libpoly-0.2.1/src/CMakeLists.txt	2026-01-20 10:38:18.323235200 +0900
@@ -97,8 +97,8 @@ target_link_libraries(polyxx poly ${GMP_
 # cmake complains if the RUNTIME DESTINATION bin is missing
 #
 if (CYGWIN OR WIN32 OR MINGW)
- install(TARGETS poly LIBRARY DESTINATION lib RUNTIME DESTINATION bin)
- install(TARGETS polyxx LIBRARY DESTINATION lib RUNTIME DESTINATION bin)
+ install(TARGETS poly ARCHIVE DESTINATION lib RUNTIME DESTINATION bin)
+ install(TARGETS polyxx ARCHIVE DESTINATION lib RUNTIME DESTINATION bin)
 else()
  install(TARGETS poly LIBRARY DESTINATION lib)
  install(TARGETS polyxx LIBRARY DESTINATION lib)
