--- origsrc/dosbox-0.74/configure.in	2010-05-11 03:58:57.000000000 +0900
+++ src/dosbox-0.74/configure.in	2010-12-18 09:44:04.913131400 +0900
@@ -17,7 +17,7 @@ AC_PROG_CC
 AC_PROG_CPP
 AC_PROG_CXX
 AC_PROG_INSTALL
-AC_PROG_RANLIB
+LT_INIT(win32-dll)
 
 dnl Some needed libaries for OS2
 dnl perharps join this with the other target depended checks. move them upwards
--- origsrc/dosbox-0.74/src/Makefile.am	2010-05-11 02:43:54.000000000 +0900
+++ src/dosbox-0.74/src/Makefile.am	2010-12-18 01:07:53.075780400 +0900
@@ -12,8 +12,8 @@ endif
 	$(WINDRES) -o $@ $<
 
 dosbox_SOURCES = dosbox.cpp $(ico_stuff)
-dosbox_LDADD = cpu/libcpu.a debug/libdebug.a dos/libdos.a fpu/libfpu.a  hardware/libhardware.a gui/libgui.a \
-               ints/libints.a misc/libmisc.a shell/libshell.a hardware/serialport/libserial.a libs/gui_tk/libgui_tk.a
+dosbox_LDADD = cpu/libcpu.la debug/libdebug.la dos/libdos.la fpu/libfpu.la  hardware/libhardware.la gui/libgui.la \
+               ints/libints.la misc/libmisc.la shell/libshell.la hardware/serialport/libserial.la libs/gui_tk/libgui_tk.la
 
 EXTRA_DIST = winres.rc dosbox.ico
 
--- origsrc/dosbox-0.74/src/cpu/Makefile.am	2010-05-11 02:43:54.000000000 +0900
+++ src/dosbox-0.74/src/cpu/Makefile.am	2010-12-18 01:03:54.202297200 +0900
@@ -1,7 +1,7 @@
 SUBDIRS = core_full core_normal core_dyn_x86 core_dynrec
 AM_CPPFLAGS = -I$(top_srcdir)/include
 
-noinst_LIBRARIES = libcpu.a
-libcpu_a_SOURCES = callback.cpp cpu.cpp flags.cpp modrm.cpp modrm.h core_full.cpp instructions.h	\
+noinst_LTLIBRARIES = libcpu.la
+libcpu_la_SOURCES = callback.cpp cpu.cpp flags.cpp modrm.cpp modrm.h core_full.cpp instructions.h	\
 		   paging.cpp lazyflags.h core_normal.cpp core_simple.cpp core_prefetch.cpp \
 		   core_dyn_x86.cpp core_dynrec.cpp
--- origsrc/dosbox-0.74/src/debug/Makefile.am	2010-05-11 02:43:54.000000000 +0900
+++ src/dosbox-0.74/src/debug/Makefile.am	2010-12-18 01:04:22.212574000 +0900
@@ -1,4 +1,4 @@
 AM_CPPFLAGS = -I$(top_srcdir)/include
 
-noinst_LIBRARIES = libdebug.a
-libdebug_a_SOURCES = debug.cpp debug_gui.cpp debug_disasm.cpp debug_inc.h disasm_tables.h debug_win32.cpp
\ No newline at end of file
+noinst_LTLIBRARIES = libdebug.la
+libdebug_la_SOURCES = debug.cpp debug_gui.cpp debug_disasm.cpp debug_inc.h disasm_tables.h debug_win32.cpp
--- origsrc/dosbox-0.74/src/dos/Makefile.am	2010-05-11 02:43:54.000000000 +0900
+++ src/dosbox-0.74/src/dos/Makefile.am	2010-12-18 01:04:56.191433200 +0900
@@ -1,8 +1,8 @@
 AM_CPPFLAGS = -I$(top_srcdir)/include
 
-noinst_LIBRARIES = libdos.a
+noinst_LTLIBRARIES = libdos.la
 EXTRA_DIST = scsidefs.h wnaspi32.h dos_codepages.h dos_keyboard_layout_data.h
-libdos_a_SOURCES = dos.cpp dos_devices.cpp dos_execute.cpp dos_files.cpp dos_ioctl.cpp dos_memory.cpp \
+libdos_la_SOURCES = dos.cpp dos_devices.cpp dos_execute.cpp dos_files.cpp dos_ioctl.cpp dos_memory.cpp \
                    dos_misc.cpp dos_classes.cpp dos_programs.cpp dos_tables.cpp \
 		   drives.cpp drives.h drive_virtual.cpp drive_local.cpp drive_cache.cpp drive_fat.cpp \
 		   drive_iso.cpp dev_con.h dos_mscdex.cpp dos_keyboard_layout.cpp \
--- origsrc/dosbox-0.74/src/fpu/Makefile.am	2010-05-11 02:43:54.000000000 +0900
+++ src/dosbox-0.74/src/fpu/Makefile.am	2010-12-18 01:04:50.473210800 +0900
@@ -1,5 +1,5 @@
 AM_CPPFLAGS = -I$(top_srcdir)/include
 
-noinst_LIBRARIES = libfpu.a
-libfpu_a_SOURCES = fpu.cpp fpu_instructions.h \
+noinst_LTLIBRARIES = libfpu.la
+libfpu_la_SOURCES = fpu.cpp fpu_instructions.h \
                    fpu_instructions_x86.h
--- origsrc/dosbox-0.74/src/gui/Makefile.am	2010-05-11 02:43:54.000000000 +0900
+++ src/dosbox-0.74/src/gui/Makefile.am	2010-12-18 01:05:14.207338800 +0900
@@ -1,7 +1,7 @@
 AM_CPPFLAGS = -I$(top_srcdir)/include
 
-noinst_LIBRARIES = libgui.a
-libgui_a_SOURCES = sdlmain.cpp sdl_mapper.cpp dosbox_logo.h \
+noinst_LTLIBRARIES = libgui.la
+libgui_la_SOURCES = sdlmain.cpp sdl_mapper.cpp dosbox_logo.h \
 	render.cpp render_scalers.cpp render_scalers.h \
 	render_templates.h render_loops.h render_simple.h \
 	render_templates_sai.h render_templates_hq.h \
--- origsrc/dosbox-0.74/src/hardware/Makefile.am	2010-05-11 02:43:54.000000000 +0900
+++ src/dosbox-0.74/src/hardware/Makefile.am	2010-12-18 01:05:30.791185200 +0900
@@ -4,9 +4,9 @@ SUBDIRS = serialport
 
 EXTRA_DIST = opl.cpp opl.h adlib.h dbopl.h
 
-noinst_LIBRARIES = libhardware.a
+noinst_LTLIBRARIES = libhardware.la
 
-libhardware_a_SOURCES = adlib.cpp dma.cpp gameblaster.cpp hardware.cpp iohandler.cpp joystick.cpp keyboard.cpp \
+libhardware_la_SOURCES = adlib.cpp dma.cpp gameblaster.cpp hardware.cpp iohandler.cpp joystick.cpp keyboard.cpp \
                         memory.cpp mixer.cpp pcspeaker.cpp pic.cpp sblaster.cpp tandy_sound.cpp timer.cpp \
 			vga.cpp vga_attr.cpp vga_crtc.cpp vga_dac.cpp vga_draw.cpp vga_gfx.cpp vga_other.cpp \
 			vga_memory.cpp vga_misc.cpp vga_seq.cpp vga_xga.cpp vga_s3.cpp vga_tseng.cpp vga_paradise.cpp \
--- origsrc/dosbox-0.74/src/hardware/pic.cpp	2010-05-11 02:43:54.000000000 +0900
+++ src/dosbox-0.74/src/hardware/pic.cpp	2010-12-18 11:07:39.542533700 +0900
@@ -541,12 +541,12 @@ void TIMER_AddTick(void) {
 }
 
 
-class PIC:public Module_base{
+class PIC_:public Module_base{
 private:
 	IO_ReadHandleObject ReadHandler[4];
 	IO_WriteHandleObject WriteHandler[4];
 public:
-	PIC(Section* configuration):Module_base(configuration){
+	PIC_(Section* configuration):Module_base(configuration){
 		/* Setup pic0 and pic1 with initial values like DOS has normally */
 		PIC_IRQCheck=0;
 		PIC_IRQActive=PIC_NOIRQ;
@@ -596,17 +596,17 @@ public:
 		pic_queue.free_entry=&pic_queue.entries[0];
 		pic_queue.next_entry=0;
 	}
-	~PIC(){
+	~PIC_(){
 	}
 };
 
-static PIC* test;
+static PIC_* test;
 
 void PIC_Destroy(Section* sec){
 	delete test;
 }
 
 void PIC_Init(Section* sec) {
-	test = new PIC(sec);
+	test = new PIC_(sec);
 	sec->AddDestroyFunction(&PIC_Destroy);
 }
--- origsrc/dosbox-0.74/src/hardware/serialport/Makefile.am	2010-05-11 02:43:54.000000000 +0900
+++ src/dosbox-0.74/src/hardware/serialport/Makefile.am	2010-12-18 01:05:43.659689200 +0900
@@ -1,8 +1,8 @@
 AM_CPPFLAGS = -I$(top_srcdir)/include
 
-noinst_LIBRARIES = libserial.a
+noinst_LTLIBRARIES = libserial.la
 
-libserial_a_SOURCES =	directserial.cpp directserial.h \
+libserial_la_SOURCES =	directserial.cpp directserial.h \
 						libserial.cpp libserial.h \
 						serialdummy.cpp serialdummy.h serialport.cpp \
 						softmodem.cpp softmodem.h misc_util.cpp misc_util.h \
--- origsrc/dosbox-0.74/src/hardware/serialport/libserial.cpp	2010-05-11 02:43:54.000000000 +0900
+++ src/dosbox-0.74/src/hardware/serialport/libserial.cpp	2010-12-18 11:31:19.304050500 +0900
@@ -254,7 +254,7 @@ bool SERIAL_setCommParameters(COMPORT po
 }
 #endif
 
-#if defined (LINUX) || defined (MACOSX) || defined (BSD)
+#if defined (LINUX) || defined (MACOSX) || defined (BSD) || defined(__CYGWIN__)
 
 #include <string.h> // strlen
 #include <stdlib.h>
--- origsrc/dosbox-0.74/src/ints/Makefile.am	2010-05-11 02:43:54.000000000 +0900
+++ src/dosbox-0.74/src/ints/Makefile.am	2010-12-18 01:06:04.880202800 +0900
@@ -1,7 +1,7 @@
 AM_CPPFLAGS = -I$(top_srcdir)/include
 
-noinst_LIBRARIES = libints.a
-libints_a_SOURCES = mouse.cpp xms.cpp xms.h ems.cpp \
+noinst_LTLIBRARIES = libints.la
+libints_la_SOURCES = mouse.cpp xms.cpp xms.h ems.cpp \
                     int10.cpp int10.h int10_char.cpp int10_memory.cpp int10_misc.cpp int10_modes.cpp \
                     int10_vesa.cpp int10_pal.cpp int10_put_pixel.cpp int10_video_state.cpp int10_vptable.cpp \
                     bios.cpp bios_disk.cpp bios_keyboard.cpp 
--- origsrc/dosbox-0.74/src/libs/gui_tk/Makefile.am	2010-05-11 02:43:54.000000000 +0900
+++ src/dosbox-0.74/src/libs/gui_tk/Makefile.am	2010-12-18 01:06:22.966209200 +0900
@@ -1,4 +1,4 @@
 AM_CPPFLAGS = -I$(top_srcdir)/include
 
-noinst_LIBRARIES = libgui_tk.a
-libgui_tk_a_SOURCES = gui_tk.cpp gui_tk.h
+noinst_LTLIBRARIES = libgui_tk.la
+libgui_tk_la_SOURCES = gui_tk.cpp gui_tk.h
--- origsrc/dosbox-0.74/src/misc/Makefile.am	2010-05-11 02:43:54.000000000 +0900
+++ src/dosbox-0.74/src/misc/Makefile.am	2010-12-18 01:06:46.530092400 +0900
@@ -1,4 +1,4 @@
 AM_CPPFLAGS = -I$(top_srcdir)/include
 
-noinst_LIBRARIES = libmisc.a
-libmisc_a_SOURCES = cross.cpp messages.cpp programs.cpp setup.cpp support.cpp
+noinst_LTLIBRARIES = libmisc.la
+libmisc_la_SOURCES = cross.cpp messages.cpp programs.cpp setup.cpp support.cpp
--- origsrc/dosbox-0.74/src/shell/Makefile.am	2010-05-11 02:43:54.000000000 +0900
+++ src/dosbox-0.74/src/shell/Makefile.am	2010-12-18 01:07:15.722068400 +0900
@@ -1,4 +1,4 @@
 AM_CPPFLAGS = -I$(top_srcdir)/include
 
-noinst_LIBRARIES = libshell.a
-libshell_a_SOURCES = shell.cpp shell_batch.cpp shell_cmds.cpp shell_misc.cpp
+noinst_LTLIBRARIES = libshell.la
+libshell_la_SOURCES = shell.cpp shell_batch.cpp shell_cmds.cpp shell_misc.cpp
