--- origsrc/rubberband-3.3.0/meson.build	2023-07-28 19:42:47.000000000 +0900
+++ src/rubberband-3.3.0/meson.build	2024-10-16 12:26:00.583908900 +0900
@@ -716,9 +716,11 @@ if have_lv2
       lv2_symbol_args,
     ],
     dependencies: [
-      rubberband_objlib_dep,
       general_dependencies,
     ],
+    link_with: [
+      rubberband_library,
+    ],
     name_prefix: '',
     install: true,
     install_dir: get_option('libdir') / 'lv2/rubberband.lv2',
@@ -779,10 +781,12 @@ if have_sndfile
       feature_libraries,
     ],
     dependencies: [
-      rubberband_objlib_dep,
       general_dependencies,
       sndfile_dep,
     ],
+    link_with: [
+      rubberband_library,
+    ],
     install: true,
   )
   target_summary += { 'Command-line utility (R3)': [ true, 'Name: ' + rubberband_program_name_r3 ] }
