--- origsrc/rubberband-4.0.0/meson.build	2024-10-26 01:26:08.000000000 +0900
+++ src/rubberband-4.0.0/meson.build	2025-09-16 12:19:01.171415800 +0900
@@ -726,9 +726,11 @@ if have_lv2
       lv2_symbol_args,
     ],
     dependencies: [
-      rubberband_objlib_dep,
       general_dependencies,
     ],
+    link_with: [
+      rubberband_library,
+    ],
     name_prefix: '',
     install: true,
     install_dir: get_option('libdir') / 'lv2/rubberband.lv2',
@@ -789,10 +791,12 @@ if have_sndfile
       feature_libraries,
     ],
     dependencies: [
-      rubberband_objlib_dep,
       general_dependencies,
       sndfile_dep,
     ],
+    link_with: [
+      rubberband_library,
+    ],
     install: true,
   )
   target_summary += { 'Command-line utility (R3)': [ true, 'Name: ' + rubberband_program_name_r3 ] }
