--- origsrc/libebb/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/libebb/Makefile.am	2012-02-11 20:24:28.741677700 +0900
@@ -0,0 +1,26 @@
+
+lib_LTLIBRARIES = libebb.la
+
+libebb_la_SOURCES = ebb.c ebb_request_parser.c rbtree.c
+libebb_la_CPPFLAGS = -DVERSION=\"0.1\" -DHAVE_GNUTLS ${GNUTLS_CFLAGS}
+libebb_la_LDFLAGS = -no-undefined
+libebb_la_LIBADD = ${GNUTLS_LIBS}
+
+TESTS = test_request_parser test_rbtree
+
+check_PROGRAMS = test_request_parser test_rbtree hello_world
+
+test_request_parser_SOURCES = test_request_parser.c
+test_request_parser_LDADD = libebb.la
+
+test_rbtree_SOURCES = test_rbtree.c
+test_rbtree_LDADD = libebb.la
+
+hello_world_SOURCES = examples/hello_world.c
+hello_world_LDADD = libebb.la
+
+include_HEADERS = ebb.h ebb_request_parser.h
+
+.rl.c:
+	ragel -s -G2 $< -o $@
+
--- origsrc/libebb/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/libebb/configure.ac	2012-02-11 20:19:17.801370400 +0900
@@ -0,0 +1,36 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([libebb], [0.1], [http://tinyclouds.org/libebb/])
+AC_CONFIG_SRCDIR([ebb.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_LN_S
+LT_INIT([disable-static])
+
+# Checks for libraries.
+PKG_CHECK_MODULES([GNUTLS], [gnutls])
+
+# FIXME: Replace `main' with a function in `-lev':
+AC_CHECK_LIB([ev], [ev_run])
+
+# Checks for header files.
+AC_CHECK_HEADERS([arpa/inet.h fcntl.h netinet/in.h stdlib.h string.h sys/socket.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+AC_TYPE_SSIZE_T
+
+# Checks for library functions.
+AC_FUNC_ERROR_AT_LINE
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([inet_ntoa memset socket])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
