--- origsrc/quesa-1.8/Makefile.am	2006-04-24 16:20:40.000000000 +0900
+++ src/quesa-1.8/Makefile.am	2024-11-18 17:35:11.811488500 +0900
@@ -277,3 +277,7 @@ libquesa_la_SOURCES=	${myIncludes}
 libquesa_la_CFLAGS= -x c++ -DQUESA_OS_UNIX=1 $(WARN) $(QUESAINCLUDES)
 libquesa_la_CPPFLAGS= -DQUESA_OS_UNIX=1 $(WARN) $(QUESAINCLUDES)
 libquesa_la_LIBADD= -lm -lc -lX11 -lGL -lGLU
+libquesa_la_LDFLAGS = -no-undefined
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/quesa-1.8/configure.in	2024-11-18 17:35:09.295866000 +0900
+++ src/quesa-1.8/configure.in	2024-11-18 17:35:11.811488500 +0900
@@ -36,4 +36,4 @@ fi
 fi
 
 
-AC_OUTPUT(Makefile)
+AC_OUTPUT([quesa.pc Makefile])
--- origsrc/quesa-1.8/quesa.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/quesa-1.8/quesa.pc.in	2024-11-18 17:35:11.811488500 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: quesa
+Description: Quesa QuickDraw 3D implementation
+Version: @VERSION@
+Libs: -L${libdir} -lquesa
+Cflags: -I${includedir}/quesa
