--- origsrc/quesa-1.8/Makefile.am	2006-04-24 16:20:40.000000000 +0900
+++ src/quesa-1.8/Makefile.am	2010-11-23 23:47:21.292159500 +0900
@@ -277,3 +277,4 @@ libquesa_la_SOURCES=	${myIncludes}
 libquesa_la_CFLAGS= -x c++ -DQUESA_OS_UNIX=1 $(WARN) $(QUESAINCLUDES)
 libquesa_la_CPPFLAGS= -DQUESA_OS_UNIX=1 $(WARN) $(QUESAINCLUDES)
 libquesa_la_LIBADD= -lm -lc -lX11 -lGL -lGLU
+libquesa_la_LDFLAGS = -no-undefined
--- origsrc/quesa-1.8/configure.in	2010-11-23 23:45:45.558873100 +0900
+++ src/quesa-1.8/configure.in	2010-11-23 23:46:25.075755600 +0900
@@ -3,13 +3,11 @@ AC_INIT(Quesa,1.8,quesa_list@designcommu
 
 AC_CONFIG_SRCDIR(Source/Core/Glue/QD3DCamera.c)
 AM_INIT_AUTOMAKE([no-dependencies])
-AM_MAINTAINER_MODE
 
 dnl Checks for programs.
 AC_PROG_CC
 AC_PROG_CXX
-AC_PROG_LIBTOOL
-AC_PROG_RANLIB
+LT_INIT(win32-dll)
 
 AC_C_BIGENDIAN_QUESA
 
