--- origsrc/lc0-0.28.2/src/neural/reader.cc	2021-12-13 11:26:08.000000000 +0900
+++ src/lc0-0.28.2/src/neural/reader.cc	2022-02-06 08:19:23.836084500 +0900
@@ -220,6 +220,9 @@ bool TrainingDataReader::ReadChunk(V6Tra
         if (read_size < 0) throw Exception("Corrupt read.");
         return read_size == v4_extra + v5_extra + v6_extra;
       }
+      default: {
+        throw Exception("Invalid version.");
+      }
     }
   }
 }
