--- origsrc/glpng/CMakeLists.txt	2025-11-12 17:38:07.000000000 +0900
+++ src/glpng/CMakeLists.txt	2025-11-12 17:55:03.988166900 +0900
@@ -10,11 +10,20 @@ set(VERSION ${SONAME_VERSION}.${MINOR_VE
 find_package(PNG REQUIRED)
 find_package(OpenGL REQUIRED COMPONENTS OpenGL)
 
-add_link_options(LINKER:-z,defs)
+if (NOT CYGWIN)
+    add_link_options(LINKER:-z,defs)
+endif()
 
 add_subdirectory(src)
 configure_file(FindGLPNG.cmake.in FindGLPNG.cmake @ONLY)
 
 install(FILES include/GL/glpng.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/GL/)
 install(FILES glpng.htm DESTINATION ${CMAKE_INSTALL_DOCDIR})
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/FindGLPNG.cmake DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/cmake/Modules)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/FindGLPNG.cmake DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/glpng/Modules)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/glpng.pc.in ${CMAKE_BINARY_DIR}/glpng.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/glpng.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/glpng/glpng.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/glpng/glpng.pc.in	2025-11-12 17:44:30.670194800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: glpng
+Description: PNG loader for OpenGL
+Version: @VERSION@
+Libs: -L${libdir} -lglpng
+Cflags: -I${includedir}
--- origsrc/glpng/src/CMakeLists.txt	2025-11-12 17:38:07.000000000 +0900
+++ src/glpng/src/CMakeLists.txt	2025-11-12 17:42:13.979962100 +0900
@@ -19,4 +19,4 @@ target_link_libraries(
 )
 
 set_target_properties(glpng PROPERTIES VERSION ${VERSION} SOVERSION ${SONAME_VERSION})
-install(TARGETS glpng DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS glpng RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
