--- origsrc/glpng/CMakeLists.txt	2025-01-31 07:11:53.000000000 +0900
+++ src/glpng/CMakeLists.txt	2025-01-31 07:25:30.375232200 +0900
@@ -18,4 +18,11 @@ configure_file(FindGLPNG.cmake.in FindGL
 
 install(FILES include/GL/glpng.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/GL/)
 install(FILES glpng.htm DESTINATION ${CMAKE_INSTALL_DOCDIR})
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/FindGLPNG.cmake DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/cmake/Modules)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/FindGLPNG.cmake DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/glpng/Modules)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/glpng.pc.in ${CMAKE_BINARY_DIR}/glpng.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/glpng.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/glpng/glpng.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/glpng/glpng.pc.in	2025-01-31 07:21:40.299736400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: glpng
+Description: PNG loader for OpenGL
+Version: @VERSION@
+Libs: -L${libdir} -lglpng
+Cflags: -I${includedir}
--- origsrc/glpng/src/CMakeLists.txt	2025-01-31 07:11:53.000000000 +0900
+++ src/glpng/src/CMakeLists.txt	2025-01-31 07:20:11.335982200 +0900
@@ -13,7 +13,7 @@ link_libraries(
 
 add_library(glpng SHARED glpng.c)
 set_target_properties(glpng PROPERTIES VERSION ${VERSION} SOVERSION ${SONAME_VERSION})
-install(TARGETS glpng DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS glpng RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 if(BUILD_STATIC_LIBS)
 	add_library(glpng-static STATIC glpng.c)
