--- origsrc/AsmJit-1.0-beta4/CMakeLists.txt	2012-08-14 22:26:44.361500000 +0900
+++ src/AsmJit-1.0-beta4/CMakeLists.txt	2012-08-14 22:31:01.013692700 +0900
@@ -71,7 +71,10 @@ EndIf()
 # Build AsmJit shared library?
 If(ASMJIT_BUILD_LIBRARY)
   Add_Library(AsmJit SHARED ${ASMJIT_SOURCES} ${ASMJIT_HEADERS})
-  Install(TARGETS AsmJit DESTINATION lib${LIB_SUFFIX})
+  Set_Target_Properties(AsmJit PROPERTIES SOVERSION 0)
+
+  Install(TARGETS AsmJit RUNTIME DESTINATION bin
+                         ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
   # Install header files.
   ForEach(i ${ASMJIT_HEADERS})
