--- origsrc/argdata-0.7/CMakeLists.txt	2017-08-23 01:09:59.000000000 +0900
+++ src/argdata-0.7/CMakeLists.txt	2017-08-26 18:19:54.362891500 +0900
@@ -62,10 +62,16 @@ set_property(
 		"1"
 )
 
+if (CYGWIN)
+	set_target_properties(argdata PROPERTIES SOVERSION 1)
+endif (CYGWIN)
+
 install(
 	TARGETS
 		argdata
-	DESTINATION
+	RUNTIME DESTINATION
+		${CMAKE_INSTALL_BINDIR}
+	ARCHIVE DESTINATION
 		${CMAKE_INSTALL_LIBDIR}
 )
 
@@ -76,3 +82,11 @@ install(
 	DESTINATION
 		${CMAKE_INSTALL_INCLUDEDIR}
 )
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "0.7")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/argdata.pc.in ${CMAKE_BINARY_DIR}/argdata.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/argdata.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/argdata-0.7/argdata.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/argdata-0.7/argdata.pc.in	2017-08-26 18:18:37.761000000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: argdata
+Description: Binary serialisation format inspired by YAML, MessagePack and nvlists
+Version: @VERSION@
+Libs: -L${libdir} -largdata
+Cflags: -I${includedir}
