--- origsrc/argdata-0.8/CMakeLists.txt	2019-01-02 06:04:58.000000000 +0900
+++ src/argdata-0.8/CMakeLists.txt	2025-08-10 15:55:59.062545800 +0900
@@ -62,10 +62,16 @@ set_property(
 		"1"
 )
 
+if (CYGWIN)
+	set_target_properties(argdata PROPERTIES SOVERSION 1)
+endif (CYGWIN)
+
 install(
 	TARGETS
 		argdata
-	DESTINATION
+	RUNTIME DESTINATION
+		${CMAKE_INSTALL_BINDIR}
+	ARCHIVE DESTINATION
 		${CMAKE_INSTALL_LIBDIR}
 )
 
@@ -76,3 +82,10 @@ install(
 	DESTINATION
 		${CMAKE_INSTALL_INCLUDEDIR}
 )
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/argdata.pc.in ${CMAKE_BINARY_DIR}/argdata.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/argdata.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/argdata-0.8/argdata.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/argdata-0.8/argdata.pc.in	2025-08-10 15:54:38.087983300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: argdata
+Description: Binary serialisation format inspired by YAML, MessagePack and nvlists
+Version: @VERSION@
+Libs: -L${libdir} -largdata
+Cflags: -I${includedir}
