--- origsrc/lastz-1.04.22/src/Makefile	2022-08-08 05:18:57.000000000 +0900
+++ src/lastz-1.04.22/src/Makefile	2023-01-09 20:20:04.613785500 +0900
@@ -54,7 +54,7 @@ builds = lastz lastz_D lastz_32
 #
 #---------
 
-definedForAll = -Wall -Wextra -Werror -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
+definedForAll = -Wall -Wextra -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 flagsFor32    = -Dmax_sequence_index=32 -Dmax_malloc_index=40 -Ddiag_hash_size=4194304
 
 allowBackToBackGaps ?= 0                       # by default allowBackToBackGaps
