--- origsrc/lastz-1.04.52/src/Makefile	2025-11-30 16:43:55.460463300 +0900
+++ src/lastz-1.04.52/src/Makefile	2025-11-30 16:44:39.381353000 +0900
@@ -55,7 +55,7 @@ builds = lastz lastz_D lastz_32 lastz_40
 #
 #---------
 
-definedForAll = -Wall -Wextra -Werror -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
+definedForAll = -Wall -Wextra -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 flagsFor32    = -Dmax_sequence_index=32 -Dmax_malloc_index=40 -Ddiag_hash_size=4194304
 flagsFor40    = -Dmax_sequence_index=63 -Dmax_malloc_index=40 -Ddiag_hash_size=4194304
 
