--- origsrc/gmt-6.0.0/src/CMakeLists.txt	2019-10-31 12:22:02.000000000 +0900
+++ src/gmt-6.0.0/src/CMakeLists.txt	2019-12-16 07:59:41.503223000 +0900
@@ -438,14 +438,23 @@ if (HAVE_ZLIB)
 endif (HAVE_ZLIB)
 
 # set the build version (VERSION) and the API version (SOVERSION)
-set_target_properties (pslib
-	PROPERTIES
-	OUTPUT_NAME postscriptlight
-	PREFIX "${LIB_PREFIX}"
-	IMPORT_PREFIX "${LIB_PREFIX}"
-	VERSION ${GMT_LIB_VERSION}
-	SOVERSION ${GMT_LIB_SOVERSION}
-	DEFINE_SYMBOL "LIBRARY_EXPORTS")
+if (CYGWIN)
+	set_target_properties (pslib
+		PROPERTIES
+		OUTPUT_NAME postscriptlight
+		VERSION ${GMT_LIB_VERSION}
+		SOVERSION ${GMT_LIB_SOVERSION}
+		DEFINE_SYMBOL "LIBRARY_EXPORTS")
+else ()
+	set_target_properties (pslib
+		PROPERTIES
+		OUTPUT_NAME postscriptlight
+		PREFIX "${LIB_PREFIX}"
+		IMPORT_PREFIX "${LIB_PREFIX}"
+		VERSION ${GMT_LIB_VERSION}
+		SOVERSION ${GMT_LIB_SOVERSION}
+		DEFINE_SYMBOL "LIBRARY_EXPORTS")
+endif ()
 
 # If a renaming of the gmtpslpsl dll has been set in ConfigUser.cmake
 if (WIN32 AND PSL_DLL_RENAME)
@@ -576,15 +585,25 @@ if (HAVE_M_LIBRARY)
 endif (HAVE_M_LIBRARY)
 
 # set the build version (VERSION) and the API version (SOVERSION)
-set_target_properties (gmtlib
-	PROPERTIES
-	OUTPUT_NAME gmt
-	PREFIX "${LIB_PREFIX}"
-	IMPORT_PREFIX "${LIB_PREFIX}"
-	VERSION ${GMT_LIB_VERSION}
-	SOVERSION ${GMT_LIB_SOVERSION}
-	LINK_FLAGS "${USER_GMTLIB_LINK_FLAGS}"
-	DEFINE_SYMBOL "LIBRARY_EXPORTS")
+if (CYGWIN)
+	set_target_properties (gmtlib
+		PROPERTIES
+		OUTPUT_NAME gmt
+		VERSION ${GMT_LIB_VERSION}
+		SOVERSION ${GMT_LIB_SOVERSION}
+		LINK_FLAGS "${USER_GMTLIB_LINK_FLAGS}"
+		DEFINE_SYMBOL "LIBRARY_EXPORTS")
+else()
+	set_target_properties (gmtlib
+		PROPERTIES
+		OUTPUT_NAME gmt
+		PREFIX "${LIB_PREFIX}"
+		IMPORT_PREFIX "${LIB_PREFIX}"
+		VERSION ${GMT_LIB_VERSION}
+		SOVERSION ${GMT_LIB_SOVERSION}
+		LINK_FLAGS "${USER_GMTLIB_LINK_FLAGS}"
+		DEFINE_SYMBOL "LIBRARY_EXPORTS")
+endif()
 
 # If a renaming of the gmt dll has been set in ConfigUser.cmake
 if (WIN32 AND GMT_DLL_RENAME)
