--- origsrc/gmt-6.6.0/CMakeLists.txt	2025-07-23 19:39:53.000000000 +0900
+++ src/gmt-6.6.0/CMakeLists.txt	2025-11-12 16:17:15.076620100 +0900
@@ -96,9 +96,9 @@ find_package (LATEX COMPONENTS DVIPS)
 add_subdirectory (src)
 add_subdirectory (share)      # share must be processed *after* src (GSHHG_PATH)
 add_subdirectory (doc)        # doc must be processed *after* src
-if (EXISTS ${GMT_SOURCE_DIR}/test/)
+if (DO_TESTS)
 	add_subdirectory (test)
-endif (EXISTS ${GMT_SOURCE_DIR}/test/)
+endif (DO_TESTS)
 add_subdirectory (cmake/dist) # make distribution bundles (always last)
 
 # Source release target
--- origsrc/gmt-6.6.0/src/CMakeLists.txt	2025-07-23 19:39:53.000000000 +0900
+++ src/gmt-6.6.0/src/CMakeLists.txt	2025-11-12 16:17:15.082370200 +0900
@@ -512,15 +512,25 @@ if (HAVE_DLADDR AND HAVE_LIBDL)
 endif (HAVE_DLADDR AND HAVE_LIBDL)
 
 # set the build version (VERSION) and the API version (SOVERSION)
-set_target_properties (pslib
-	PROPERTIES
-	OUTPUT_NAME postscriptlight
-	RUNTIME_OUTPUT_NAME postscriptlight
-	PREFIX "${LIB_PREFIX}"
-	IMPORT_PREFIX "${LIB_PREFIX}"
-	VERSION ${GMT_LIB_VERSION}
-	SOVERSION ${GMT_LIB_SOVERSION}
-	DEFINE_SYMBOL "LIBRARY_EXPORTS")
+if (CYGWIN)
+	set_target_properties (pslib
+		PROPERTIES
+		OUTPUT_NAME postscriptlight
+		RUNTIME_OUTPUT_NAME postscriptlight
+		VERSION ${GMT_LIB_VERSION}
+		SOVERSION ${GMT_LIB_SOVERSION}
+		DEFINE_SYMBOL "LIBRARY_EXPORTS")
+else()
+	set_target_properties (pslib
+		PROPERTIES
+		OUTPUT_NAME postscriptlight
+		RUNTIME_OUTPUT_NAME postscriptlight
+		PREFIX "${LIB_PREFIX}"
+		IMPORT_PREFIX "${LIB_PREFIX}"
+		VERSION ${GMT_LIB_VERSION}
+		SOVERSION ${GMT_LIB_SOVERSION}
+		DEFINE_SYMBOL "LIBRARY_EXPORTS")
+endif()
 
 # If a renaming of the psl dll has been set in ConfigUserAdvanced.cmake
 if (WIN32 AND PSL_DLL_RENAME)
@@ -560,16 +570,27 @@ if ((MSVC OR MINGW) AND FFTW3_FOUND)
 endif ((MSVC OR MINGW) AND FFTW3_FOUND)
 
 # set the build version (VERSION) and the API version (SOVERSION)
-set_target_properties (gmtlib
-	PROPERTIES
-	OUTPUT_NAME gmt
-	RUNTIME_OUTPUT_NAME gmt
-	PREFIX "${LIB_PREFIX}"
-	IMPORT_PREFIX "${LIB_PREFIX}"
-	VERSION ${GMT_LIB_VERSION}
-	SOVERSION ${GMT_LIB_SOVERSION}
-	LINK_FLAGS "${USER_GMTLIB_LINK_FLAGS}"
-	DEFINE_SYMBOL "LIBRARY_EXPORTS")
+if (CYGWIN)
+	set_target_properties (gmtlib
+		PROPERTIES
+		OUTPUT_NAME gmt
+		RUNTIME_OUTPUT_NAME gmt
+		VERSION ${GMT_LIB_VERSION}
+		SOVERSION ${GMT_LIB_SOVERSION}
+		LINK_FLAGS "${USER_GMTLIB_LINK_FLAGS}"
+		DEFINE_SYMBOL "LIBRARY_EXPORTS")
+else()
+	set_target_properties (gmtlib
+		PROPERTIES
+		OUTPUT_NAME gmt
+		RUNTIME_OUTPUT_NAME gmt
+		PREFIX "${LIB_PREFIX}"
+		IMPORT_PREFIX "${LIB_PREFIX}"
+		VERSION ${GMT_LIB_VERSION}
+		SOVERSION ${GMT_LIB_SOVERSION}
+		LINK_FLAGS "${USER_GMTLIB_LINK_FLAGS}"
+		DEFINE_SYMBOL "LIBRARY_EXPORTS")
+endif()
 
 # If a renaming of the gmt dll has been set in ConfigUserAdvanced.cmake
 if (WIN32 AND GMT_DLL_RENAME)
@@ -705,9 +726,9 @@ endif (BUILD_DEVELOPER)
 ##	Settings for GMT and non-GMT supplemental libraries
 ##
 if (BUILD_SUPPLEMENTS)
-	if (UNIX)
+	if (UNIX AND NOT CYGWIN)
 		set (CMAKE_SHARED_MODULE_SUFFIX .so)
-	endif (UNIX)
+	endif (UNIX AND NOT CYGWIN)
 
 	# supplement directories to be built as supplement libraries
 	# SUPPL_EXTRA_DIRS are directories for custom supplements.
