--- origsrc/gmt-6.1.1/src/CMakeLists.txt	2020-09-02 05:27:30.000000000 +0900
+++ src/gmt-6.1.1/src/CMakeLists.txt	2020-09-06 21:42:46.659433600 +0900
@@ -494,15 +494,25 @@ if (HAVE_DLADDR AND HAVE_LIBDL)
 endif (HAVE_DLADDR AND HAVE_LIBDL)
 
 # set the build version (VERSION) and the API version (SOVERSION)
-set_target_properties (pslib
-	PROPERTIES
-	OUTPUT_NAME postscriptlight
-	RUNTIME_OUTPUT_NAME postscriptlight
-	PREFIX "${LIB_PREFIX}"
-	IMPORT_PREFIX "${LIB_PREFIX}"
-	VERSION ${GMT_LIB_VERSION}
-	SOVERSION ${GMT_LIB_SOVERSION}
-	DEFINE_SYMBOL "LIBRARY_EXPORTS")
+if (CYGWIN)
+	set_target_properties (pslib
+		PROPERTIES
+		OUTPUT_NAME postscriptlight
+		RUNTIME_OUTPUT_NAME postscriptlight
+		VERSION ${GMT_LIB_VERSION}
+		SOVERSION ${GMT_LIB_SOVERSION}
+		DEFINE_SYMBOL "LIBRARY_EXPORTS")
+else()
+	set_target_properties (pslib
+		PROPERTIES
+		OUTPUT_NAME postscriptlight
+		RUNTIME_OUTPUT_NAME postscriptlight
+		PREFIX "${LIB_PREFIX}"
+		IMPORT_PREFIX "${LIB_PREFIX}"
+		VERSION ${GMT_LIB_VERSION}
+		SOVERSION ${GMT_LIB_SOVERSION}
+		DEFINE_SYMBOL "LIBRARY_EXPORTS")
+endif()
 
 # If a renaming of the psl dll has been set in ConfigUserAdvanced.cmake
 if (WIN32 AND PSL_DLL_RENAME)
@@ -542,16 +552,27 @@ if (MSVC AND FFTW3_FOUND)
 endif (MSVC AND FFTW3_FOUND)
 
 # set the build version (VERSION) and the API version (SOVERSION)
-set_target_properties (gmtlib
-	PROPERTIES
-	OUTPUT_NAME gmt
-	RUNTIME_OUTPUT_NAME gmt
-	PREFIX "${LIB_PREFIX}"
-	IMPORT_PREFIX "${LIB_PREFIX}"
-	VERSION ${GMT_LIB_VERSION}
-	SOVERSION ${GMT_LIB_SOVERSION}
-	LINK_FLAGS "${USER_GMTLIB_LINK_FLAGS}"
-	DEFINE_SYMBOL "LIBRARY_EXPORTS")
+if (CYGWIN)
+	set_target_properties (gmtlib
+		PROPERTIES
+		OUTPUT_NAME gmt
+		RUNTIME_OUTPUT_NAME gmt
+		VERSION ${GMT_LIB_VERSION}
+		SOVERSION ${GMT_LIB_SOVERSION}
+		LINK_FLAGS "${USER_GMTLIB_LINK_FLAGS}"
+		DEFINE_SYMBOL "LIBRARY_EXPORTS")
+else()
+	set_target_properties (gmtlib
+		PROPERTIES
+		OUTPUT_NAME gmt
+		RUNTIME_OUTPUT_NAME gmt
+		PREFIX "${LIB_PREFIX}"
+		IMPORT_PREFIX "${LIB_PREFIX}"
+		VERSION ${GMT_LIB_VERSION}
+		SOVERSION ${GMT_LIB_SOVERSION}
+		LINK_FLAGS "${USER_GMTLIB_LINK_FLAGS}"
+		DEFINE_SYMBOL "LIBRARY_EXPORTS")
+endif()
 
 # If a renaming of the gmt dll has been set in ConfigUserAdvanced.cmake
 if (WIN32 AND GMT_DLL_RENAME)
