--- origsrc/libphonenumber-8.12.39/cpp/CMakeLists.txt	2021-12-07 17:02:27.000000000 +0900
+++ src/libphonenumber-8.12.39/cpp/CMakeLists.txt	2024-09-23 00:44:33.098414000 +0900
@@ -351,7 +351,7 @@ add_metadata_gen_target (
   ${TEST_METADATA_TARGET}
   "${RESOURCES_DIR}/PhoneNumberMetadataForTesting.xml"
   "test_metadata"
-  "metadata"
+  "test_metadata"
 )
 list (APPEND TESTING_LIBRARY_SOURCES "src/phonenumbers/test_metadata.cc")
 
@@ -411,7 +411,6 @@ if (${BUILD_GEOCODER} STREQUAL "ON")
   set_target_properties (geocoding-shared
     PROPERTIES
       OUTPUT_NAME "geocoding"
-      PREFIX "lib"
       SOVERSION ${libphonenumber_VERSION_MAJOR}
       VERSION ${libphonenumber_VERSION_MAJOR}.${libphonenumber_VERSION_MINOR})
 endif ()
@@ -419,7 +418,7 @@ endif ()
 # Build a shared library (with -fPIC).
 set (BUILD_SHARED_LIB true)
 
-if (${USE_RE2} STREQUAL "ON")
+if (${USE_RE2} STREQUAL "ON" AND NOT CYGWIN)
   # RE2 is not always available as a shared library (e.g: package provided by
   # Ubuntu) therefore disable the shared library build in this case.
   if (${RE2_LIB} MATCHES ".*\\.a")
@@ -442,7 +441,6 @@ if (BUILD_SHARED_LIB)
   set_target_properties (phonenumber-shared
     PROPERTIES
       OUTPUT_NAME "phonenumber"
-      PREFIX "lib"
       SOVERSION ${libphonenumber_VERSION_MAJOR}
       VERSION ${libphonenumber_VERSION_MAJOR}.${libphonenumber_VERSION_MINOR})
 endif ()
@@ -487,8 +485,10 @@ if (${BUILD_GEOCODER} STREQUAL "ON")
   list (APPEND GEOCODER_DEPS ${COMMON_DEPS})
   # Note that the subset of base/ on which the geocoder relies is implemented
   # on top of Boost header-only libraries (e.g. scoped_ptr.hpp).
-  target_link_libraries (geocoding ${LIBRARY_DEPS})
-  target_link_libraries (geocoding-shared ${LIBRARY_DEPS})
+  if (${BUILD_STATIC_LIB} STREQUAL "ON")
+    target_link_libraries (geocoding ${LIBRARY_DEPS})
+  endif()
+  target_link_libraries (geocoding-shared ${LIBRARY_DEPS} phonenumber-shared)
 endif ()
 
 # Build a specific library for testing purposes.
@@ -649,13 +649,15 @@ if (${BUILD_STATIC_LIB} STREQUAL "ON")
 endif ()
 
 if (BUILD_SHARED_LIB)
-  install (TARGETS phonenumber-shared LIBRARY DESTINATION ${LIBDIR} ARCHIVE
+  install (TARGETS phonenumber-shared RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE
            DESTINATION ${LIBDIR})
 endif ()
 
 if (${BUILD_GEOCODER} STREQUAL "ON")
-  install (TARGETS geocoding LIBRARY DESTINATION ${LIBDIR} ARCHIVE DESTINATION ${LIBDIR})
-  install (TARGETS geocoding-shared LIBRARY DESTINATION ${LIBDIR} ARCHIVE
+  if (${BUILD_STATIC_LIB} STREQUAL "ON")
+    install (TARGETS geocoding LIBRARY DESTINATION ${LIBDIR} ARCHIVE DESTINATION ${LIBDIR})
+  endif()
+  install (TARGETS geocoding-shared RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} ARCHIVE
            DESTINATION ${LIBDIR})
 endif ()
 
@@ -666,7 +668,11 @@ if (${BUILD_GEOCODER} STREQUAL "ON")
     geocoding_test_program
     "test/phonenumbers/geocoding/geocoding_test_program.cc"
   )
-  target_link_libraries (geocoding_test_program geocoding phonenumber)
+  if (${BUILD_STATIC_LIB} STREQUAL "ON")
+    target_link_libraries (geocoding_test_program geocoding phonenumber)
+  else()
+    target_link_libraries (geocoding_test_program geocoding-shared phonenumber-shared)
+  endif()
 endif ()
 
 # Build an RPM
