--- origsrc/stiff-2.4.0/src/Makefile.am	2013-06-27 23:37:50.000000000 +0900
+++ src/stiff-2.4.0/src/Makefile.am	2019-10-30 10:24:26.043024700 +0900
@@ -33,6 +33,6 @@ stiff_SOURCES		= datamem.c field.c image
 			  datamem.h define.h field.h globals.h image.h \
 			  key.h prefs.h preflist.h threads.h tiff.h types.h \
 			  xml.h
-stiff_LDADD		= $(srcdir)/fits/libfits.a
+stiff_LDADD		= $(builddir)/fits/libfits.a
 DATE=`date +"%Y-%m-%d"`
 
