--- origsrc/x48-0.6.4/src/rpl.c	2011-11-15 12:03:40.000000000 +0900
+++ src/x48-0.6.4/src/rpl.c	2025-12-05 09:03:51.474236700 +0900
@@ -367,7 +367,7 @@ char    *string;
       *addr += 2;
       if (hp48_trans_tbl[c].trans)
         {
-          sprintf(p, hp48_trans_tbl[c].trans);
+          sprintf(p, "%s", hp48_trans_tbl[c].trans);
           p += strlen(p);
         }
       else
@@ -611,7 +611,7 @@ char    *string;
       c = read_nibbles(*addr + 2 * i + 7, 2);
       if (hp48_trans_tbl[c].trans)
         {
-          sprintf(p, hp48_trans_tbl[c].trans);
+          sprintf(p, "%s", hp48_trans_tbl[c].trans);
           p += strlen(p);
         }
       else
@@ -767,7 +767,7 @@ char    *string;
       *addr += 2;
       if (hp48_trans_tbl[c].trans)
         {
-          sprintf(p, hp48_trans_tbl[c].trans);
+          sprintf(p, "%s", hp48_trans_tbl[c].trans);
           p += strlen(p);
         }
       else
@@ -811,7 +811,7 @@ char    *string;
       *addr += 2;
       if (hp48_trans_tbl[c].trans)
         {
-          sprintf(p, hp48_trans_tbl[c].trans);
+          sprintf(p, "%s", hp48_trans_tbl[c].trans);
           p += strlen(p);
         }
       else
@@ -988,7 +988,7 @@ char *string;
                           name_addr += 2;
                           if (hp48_trans_tbl[c].trans)
                             {
-                              sprintf(p, hp48_trans_tbl[c].trans);
+                              sprintf(p, "%s", hp48_trans_tbl[c].trans);
                               p += strlen(p);
                             }
                           else
@@ -1181,7 +1181,7 @@ char    *string;
   *p++ = '\'';
   if (hp48_trans_tbl[c].trans)
     {
-      sprintf(p, hp48_trans_tbl[c].trans);
+      sprintf(p, "%s", hp48_trans_tbl[c].trans);
       p += strlen(p);
     }
   else
--- origsrc/x48-0.6.4/src/serial.c	2011-11-15 12:03:40.000000000 +0900
+++ src/x48-0.6.4/src/serial.c	2025-12-05 09:03:51.489981400 +0900
@@ -294,7 +294,7 @@ serial_init()
   ir_fd = -1;
   if (useSerial)
     {
-      sprintf(tty_dev_name, serialLine);
+      sprintf(tty_dev_name, "%s", serialLine);
       if ((ir_fd = open(tty_dev_name, O_RDWR | O_NDELAY)) >= 0)
         {
           if (verbose)
