--- origsrc/libriscv-1.12/CMakeLists.txt	2025-01-22 23:56:48.000000000 +0900
+++ src/libriscv-1.12/CMakeLists.txt	2025-05-31 13:27:49.478553900 +0900
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 3.5)
 get_directory_property(hasParent PARENT_DIRECTORY)
 
 if (NOT hasParent)
-	project (libriscv VERSION 1.11 LANGUAGES CXX)
+	project (libriscv VERSION 1.12 LANGUAGES CXX)
 
 	set(CPACK_PACKAGE_NAME libriscv-dev)
 	set(CPACK_DEBIAN_PACKAGE_MAINTAINER "fwsgonzo")
--- origsrc/libriscv-1.12/lib/CMakeLists.txt	2025-01-22 23:56:48.000000000 +0900
+++ src/libriscv-1.12/lib/CMakeLists.txt	2025-05-31 13:22:31.915593000 +0900
@@ -250,6 +250,10 @@ if (RISCV_BINARY_TRANSLATION AND RISCV_L
 	)
 endif()
 
+if (DEFINED SOVERSION)
+	set_target_properties(riscv PROPERTIES SOVERSION "${SOVERSION}")
+endif()
+
 if (${CMAKE_PROJECT_NAME} STREQUAL "libriscv")
 	install(TARGETS riscv)
 	install(FILES
--- origsrc/libriscv-1.12/lib/libriscv/linux/system_calls.cpp	2025-01-22 23:56:48.000000000 +0900
+++ src/libriscv-1.12/lib/libriscv/linux/system_calls.cpp	2025-05-31 13:16:35.299405200 +0900
@@ -30,7 +30,9 @@ extern "C" int dup3(int oldfd, int newfd
 #if __has_include(<termios.h>)
 #include <termios.h>
 #endif
+#if !defined(__CYGWIN__)
 #include <sys/syscall.h>
+#endif
 #ifndef EBADFD
 #define EBADFD EBADF  // OpenBSD, FreeBSD
 #endif
