--- origsrc/zlib-ng-2.0.4/CMakeLists.txt	2021-06-12 04:35:20.000000000 +0900
+++ src/zlib-ng-2.0.4/CMakeLists.txt	2021-06-12 23:35:51.953916000 +0900
@@ -1049,7 +1049,7 @@ if(NOT DEFINED BUILD_SHARED_LIBS OR BUIL
             set_target_properties(zlib PROPERTIES INSTALL_NAME_DIR "${LIB_INSTALL_DIR}")
         endif()
     endif()
-    if(MSYS OR CYGWIN)
+    if(MSYS)
         # Suppress version number from shared library name
         set(CMAKE_SHARED_LIBRARY_NAME_WITH_VERSION 0)
     elseif(WIN32)
