--- origsrc/libvdpau-0.5/src/vdpau_wrapper.c	2012-09-05 02:26:33.000000000 +0900
+++ src/libvdpau-0.5/src/vdpau_wrapper.c	2012-09-30 16:52:38.865591100 +0900
@@ -66,7 +66,11 @@ static void _vdp_wrapper_error_breakpoin
 
 #endif
 
-#define DRIVER_LIB_FORMAT "%slibvdpau_%s.so%s"
+#if defined(__CYGWIN__)
+#define DRIVER_LIB_FORMAT "%scygvdpau_%s-%s.dll"
+#else
+#define DRIVER_LIB_FORMAT "%slibvdpau_%s.so.%s"
+#endif
 
 static char * _vdp_get_driver_name_from_dri2(
     Display *             display,
@@ -118,7 +122,7 @@ static VdpStatus _vdp_open_driver(
     }
 
     if (snprintf(vdpau_driver_lib, sizeof(vdpau_driver_lib), DRIVER_LIB_FORMAT,
-                 VDPAU_MODULEDIR "/", vdpau_driver, ".1") >=
+                 VDPAU_MODULEDIR "/", vdpau_driver, "1") >=
             sizeof(vdpau_driver_lib)) {
         fprintf(stderr, "Failed to construct driver path: path too long\n");
         if (vdpau_driver_dri2) {
@@ -155,8 +159,13 @@ static VdpStatus _vdp_open_driver(
     if (vdpau_trace && atoi(vdpau_trace)) {
         SetDllHandle * set_dll_handle;
 
+#if defined(__CYGWIN__)
+        _vdp_trace_dll = dlopen(VDPAU_MODULEDIR "/cygvdpau_trace-1.dll",
+                                RTLD_NOW | RTLD_GLOBAL);
+#else
         _vdp_trace_dll = dlopen(VDPAU_MODULEDIR "/libvdpau_trace.so.1",
                                 RTLD_NOW | RTLD_GLOBAL);
+#endif
         if (!_vdp_trace_dll) {
             fprintf(stderr, "Failed to open VDPAU trace library %s\n", dlerror());
             _VDP_ERROR_BREAKPOINT();
