--- origsrc/mx-1.2.0/configure.ac	2011-06-03 23:27:19.000000000 +0900
+++ src/mx-1.2.0/configure.ac	2011-06-19 22:28:24.718101100 +0900
@@ -27,7 +27,7 @@ AC_CONFIG_HEADERS([config.h])
 AC_ISC_POSIX
 AC_HEADER_STDC
 AC_DISABLE_STATIC
-AC_PROG_LIBTOOL
+LT_INIT
 AM_PROG_CC_C_O
 AM_PATH_GLIB_2_0([2.26.0])
 AC_CHECK_FUNCS([localtime_r])
--- origsrc/mx-1.2.0/mx/Makefile.am	2011-05-18 19:36:17.000000000 +0900
+++ src/mx-1.2.0/mx/Makefile.am	2011-06-19 22:28:24.728115500 +0900
@@ -244,7 +244,7 @@ common_includes = \
 common_ldflags = \
 	-export-dynamic			\
 	-export-symbols-regex "^mx.*"	\
-	-rpath $(libdir)
+	-rpath $(libdir) -no-undefined
 
 libmx_@MX_API_VERSION@_la_CFLAGS =	\
 	$(common_includes)		\
--- origsrc/mx-1.2.0/mx-gtk/Makefile.am	2011-05-18 19:36:17.000000000 +0900
+++ src/mx-1.2.0/mx-gtk/Makefile.am	2011-06-19 22:28:24.798216300 +0900
@@ -35,6 +35,8 @@ libmx_gtk_@MX_API_VERSION@_la_CFLAGS =	\
 	$(GTK_CFLAGS)				\
 	$(NULL)
 
+common_ldflags = -no-undefined
+
 libmx_gtk_@MX_API_VERSION@_la_LDFLAGS = $(GTK_LT_LDFLAGS) $(common_ldflags)
 
 libmx_gtk_@MX_API_VERSION@_la_SOURCES =	\
