--- origsrc/mx-1.3.0/mx/Makefile.am	2011-05-19 20:07:41.000000000 +0900
+++ src/mx-1.3.0/mx/Makefile.am	2011-08-06 11:34:59.909536100 +0900
@@ -244,7 +244,7 @@ common_includes = \
 common_ldflags = \
 	-export-dynamic			\
 	-export-symbols-regex "^mx.*"	\
-	-rpath $(libdir)
+	-rpath $(libdir) -no-undefined
 
 libmx_@MX_API_VERSION@_la_CFLAGS =	\
 	$(common_includes)		\
--- origsrc/mx-1.3.0/mx-gtk/Makefile.am	2011-05-19 20:07:41.000000000 +0900
+++ src/mx-1.3.0/mx-gtk/Makefile.am	2011-08-06 11:35:39.586588900 +0900
@@ -18,6 +18,8 @@ mxinclude_DATA = 				\
 
 lib_LTLIBRARIES = libmx-gtk-@MX_API_VERSION@.la
 
+common_ldflags = -no-undefined
+
 #
 # Mx GTK+ Library
 #
