--- origsrc/OpenIGTLink/Source/igtlWin32Header.h	2013-02-22 21:23:58.000000000 +0900
+++ src/OpenIGTLink/Source/igtlWin32Header.h	2013-02-22 21:29:53.665458900 +0900
@@ -39,15 +39,6 @@
 
 // add in the Windows variants
 
-#if defined(__CYGWIN__)
-#ifndef WIN32
-#define WIN32 1
-#endif
-#ifndef _WIN32
-#define _WIN32 1
-#endif
-#endif
-
 #if defined(_WIN32)
   // Include the windows header here only if requested by user code.
 # if defined(IGTL_INCLUDE_WINDOWS_H)
