--- origsrc/OpenIGTLink-3.0/CMakeLists.txt	2017-06-28 19:28:12.000000000 +0900
+++ src/OpenIGTLink-3.0/CMakeLists.txt	2019-05-14 23:06:26.308283500 +0900
@@ -267,3 +267,11 @@ OPTION(BUILD_DOCUMENTATION "Build OpenIG
 IF(BUILD_DOCUMENTATION)
   ADD_SUBDIRECTORY( Documents/Doxygen )
 ENDIF(BUILD_DOCUMENTATION)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+SET(VERSION "${OpenIGTLink_VERSION_MAJOR}.${OpenIGTLink_VERSION_MINOR}.${OpenIGTLink_VERSION_PATCH}")
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/OpenIGTLink.pc.in ${CMAKE_BINARY_DIR}/OpenIGTLink.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/OpenIGTLink.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/OpenIGTLink-3.0/OpenIGTLink.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/OpenIGTLink-3.0/OpenIGTLink.pc.in	2019-05-14 23:03:16.945452300 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: OpenIGTLink
+Description: Open Network Interface for Image Guided Therapy
+Version: @VERSION@
+Libs: -L${libdir} -lOpenIGTLink
+Cflags: -I${includedir}
--- origsrc/OpenIGTLink-3.0/Source/CMakeLists.txt	2017-06-28 19:28:12.000000000 +0900
+++ src/OpenIGTLink-3.0/Source/CMakeLists.txt	2019-05-14 23:21:28.547712000 +0900
@@ -72,7 +72,7 @@ SET(OpenIGTLink_SOURCES
   )
 
 SET(OpenIGTLink_INCLUDE_FILES)
-IF( MSVC OR ${CMAKE_GENERATOR} MATCHES "Xcode" )
+IF( MSVC OR ${CMAKE_GENERATOR} MATCHES "Xcode" OR CYGWIN )
   LIST(APPEND OpenIGTLink_INCLUDE_FILES
     igtlutil/igtl_header.h
     igtlutil/igtl_image.h
@@ -150,7 +150,7 @@ IF (${OpenIGTLink_PROTOCOL_VERSION} GREA
     igtlBindMessage.cxx
     igtlNDArrayMessage.cxx
     )
-  IF( MSVC OR ${CMAKE_GENERATOR} MATCHES "Xcode" )
+  IF( MSVC OR ${CMAKE_GENERATOR} MATCHES "Xcode" OR CYGWIN )
     LIST(APPEND OpenIGTLink_INCLUDE_FILES
       igtlutil/igtl_colortable.h
       igtlutil/igtl_imgmeta.h
@@ -191,7 +191,7 @@ IF( ${OpenIGTLink_PROTOCOL_VERSION} GREA
     igtlutil/igtl_command.c
     igtlutil/igtl_query.c
     )
-  IF( MSVC OR ${CMAKE_GENERATOR} MATCHES "Xcode" )
+  IF( MSVC OR ${CMAKE_GENERATOR} MATCHES "Xcode" OR CYGWIN )
     LIST(APPEND OpenIGTLink_INCLUDE_FILES
       igtlCommandMessage.h
       igtlQueryMessage.h
--- origsrc/OpenIGTLink-3.0/Source/igtlWin32Header.h	2017-06-28 19:28:12.000000000 +0900
+++ src/OpenIGTLink-3.0/Source/igtlWin32Header.h	2019-05-14 22:52:47.917386900 +0900
@@ -37,15 +37,6 @@
 
 // add in the Windows variants
 
-#if defined(__CYGWIN__)
-#ifndef WIN32
-#define WIN32 1
-#endif
-#ifndef _WIN32
-#define _WIN32 1
-#endif
-#endif
-
 #if defined(_WIN32)
   // Include the windows header here only if requested by user code.
 # if defined(IGTL_INCLUDE_WINDOWS_H)
