--- origsrc/OpenIGTLink-2.1/Source/igtlWin32Header.h	2017-03-08 23:15:42.000000000 +0900
+++ src/OpenIGTLink-2.1/Source/igtlWin32Header.h	2017-03-28 18:26:09.681441100 +0900
@@ -25,8 +25,8 @@
   Portions of this code are covered under the VTK copyright.
   See VTKCopyright.txt or http://www.kitware.com/VTKCopyright.htm for details.
 
-     This software is distributed WITHOUT ANY WARRANTY; without even 
-     the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR 
+     This software is distributed WITHOUT ANY WARRANTY; without even
+     the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
      PURPOSE.  See the above copyright notices for more information.
 
 =========================================================================*/
@@ -37,15 +37,6 @@
 
 // add in the Windows variants
 
-#if defined(__CYGWIN__)
-#ifndef WIN32
-#define WIN32 1
-#endif
-#ifndef _WIN32
-#define _WIN32 1
-#endif
-#endif
-
 #if defined(_WIN32)
   // Include the windows header here only if requested by user code.
 # if defined(IGTL_INCLUDE_WINDOWS_H)
@@ -140,7 +131,7 @@
 // IGTL_EXPORT can not be used
 #define IGTL_EXPORT
 
-#if (defined(_WIN32) || defined(WIN32)) && !defined(IGTLSTATIC) 
+#if (defined(_WIN32) || defined(WIN32)) && !defined(IGTLSTATIC)
 # ifdef IGTLCommon_EXPORTS
 #  define IGTLCommon_EXPORT __declspec(dllexport)
 # else
@@ -148,7 +139,7 @@
 # endif  /* IGTLCommon_EXPORT */
 #else
 /* unix needs nothing */
-#define IGTLCommon_EXPORT 
+#define IGTLCommon_EXPORT
 #endif
 
 
--- origsrc/OpenIGTLink-2.1/Testing/CMakeLists.txt	2017-03-08 23:15:42.000000000 +0900
+++ src/OpenIGTLink-2.1/Testing/CMakeLists.txt	2017-03-28 18:36:33.929727800 +0900
@@ -7,11 +7,11 @@ include_directories(${OpenIGTLink_INCLUD
 link_directories(${OpenIGTLink_LIBRARY_DIRS})
 include_directories(${PROJECT_BINARY_DIR})
 OPTION(USE_GTEST "Build the testing tree." ON)
-IF(USE_GTEST)
+IF(USE_GTEST AND (NOT CYGWIN))
 	SET(OpenIGTLink_USE_GTEST "1")
 ELSE()
   SET(OpenIGTLink_USE_GTEST "0")
-ENDIF()	
+ENDIF()
 configure_file(${PROJECT_SOURCE_DIR}/igtlTestConfig.h.in ${PROJECT_BINARY_DIR}/igtlTestConfig.h)
 ENABLE_TESTING()
 ADD_SUBDIRECTORY( igtlutil )
@@ -24,10 +24,10 @@ IF(OpenIGTLink_USE_GTEST)
 	# Download and unpack googletest at configure time
 	configure_file(CMakeLists.txt.in
 								 googletest-download/CMakeLists.txt)
-	execute_process(COMMAND "${CMAKE_COMMAND}" -G "${CMAKE_GENERATOR}" . 
+	execute_process(COMMAND "${CMAKE_COMMAND}" -G "${CMAKE_GENERATOR}" .
 		WORKING_DIRECTORY "${PROJECT_BINARY_DIR}/googletest-download" )
 
-	execute_process(COMMAND "${CMAKE_COMMAND}" --build . 
+	execute_process(COMMAND "${CMAKE_COMMAND}" --build .
 		WORKING_DIRECTORY "${PROJECT_BINARY_DIR}/googletest-download" )
 
 	# Prevent GoogleTest from overriding our compiler/linker options
@@ -82,7 +82,7 @@ IF(USE_GTEST)
 ELSE()
   SET(GTEST_LINK OpenIGTLink)
 ENDIF()
-  
+
 TARGET_LINK_LIBRARIES(igtlMultiThreaderTest1 OpenIGTLink)
 TARGET_LINK_LIBRARIES(igtlMultiThreaderTest2 OpenIGTLink)
 TARGET_LINK_LIBRARIES(igtlMultiThreaderTest3 OpenIGTLink)
