--- origsrc/ftplib-4.0/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/ftplib-4.0/Makefile.am	2013-09-27 00:17:27.403073700 +0900
@@ -0,0 +1 @@
+SUBDIRS = src
--- origsrc/ftplib-4.0/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/ftplib-4.0/configure.ac	2013-09-27 00:16:58.482175300 +0900
@@ -0,0 +1,32 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([ftplib], [4.0])
+AC_CONFIG_SRCDIR([src/ftplib.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_LN_S
+LT_INIT
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([arpa/inet.h inttypes.h limits.h netdb.h netinet/in.h stdlib.h string.h sys/socket.h sys/time.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([gethostbyname gethostname getpass memset select socket strchr strdup strerror strstr])
+
+AC_CONFIG_FILES([Makefile
+                 src/Makefile])
+AC_OUTPUT
--- origsrc/ftplib-4.0/src/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/ftplib-4.0/src/Makefile.am	2013-09-27 00:13:26.200925300 +0900
@@ -0,0 +1,13 @@
+
+bin_PROGRAMS = qftp
+qftp_SOURCES = qftp.c
+qftp_LDADD = libftp.la
+
+
+include_HEADERS = ftplib.h
+
+
+lib_LTLIBRARIES = libftp.la
+libftp_la_SOURCES = ftplib.c
+libftp_la_LDFLAGS = -version-info 4:0:0 -no-undefined
+
