--- origsrc/ftplib-3.1-1/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/ftplib-3.1-1/Makefile.am	2010-11-23 11:38:24.049648900 +0900
@@ -0,0 +1 @@
+SUBDIRS = html src
--- origsrc/ftplib-3.1-1/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/ftplib-3.1-1/configure.ac	2010-11-23 11:31:50.263412100 +0900
@@ -0,0 +1,30 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.67])
+AC_INIT([ftplib], [3.1-1], [pfau@nbpfaus.net])
+AC_CONFIG_SRCDIR([old/ftpdir.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_LN_S
+LT_INIT(win32-dll)
+
+# Checks for header files.
+AC_CHECK_HEADERS([arpa/inet.h netdb.h netinet/in.h stdlib.h string.h sys/socket.h sys/time.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_ERROR_AT_LINE
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([gethostbyname gethostname getpass memset select socket strchr strdup strerror strstr])
+
+AC_CONFIG_FILES([Makefile
+                 html/Makefile
+                 src/Makefile])
+AC_OUTPUT
--- origsrc/ftplib-3.1-1/html/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/ftplib-3.1-1/html/Makefile.am	2010-11-23 12:09:39.262223300 +0900
@@ -0,0 +1,8 @@
+html_DATA = \
+FtpAccess.html   FtpGet.html           FtpOptions.html  FtpSite.html \
+FtpCDUp.html     FtpInit.html          FtpPut.html      FtpSize.html \
+FtpChdir.html    FtpLastResponse.html  FtpPwd.html      FtpSysType.html \
+FtpClose.html    FtpLogin.html         FtpQuit.html     FtpWrite.html \
+FtpConnect.html  FtpMkdir.html         FtpRead.html     ftplib.html \
+FtpDelete.html   FtpModDate.html       FtpRename.html   model.html \
+FtpDir.html      FtpNlst.html          FtpRmdir.html    qftp.html
--- origsrc/ftplib-3.1-1/src/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/ftplib-3.1-1/src/Makefile.am	2010-11-23 11:16:32.499637700 +0900
@@ -0,0 +1,10 @@
+lib_LTLIBRARIES = libftp.la
+libftp_la_SOURCES = ftplib.c
+libftp_la_LDFLAGS = -version-info 1:0:0 -no-undefined
+
+bin_PROGRAMS = qftp
+qftp_SOURCES = qftp.c
+qftp_LDADD = libftp.la
+
+include_HEADERS = ftplib.h
+
