diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/partio-1.1.0/CMakeLists.txt src/partio-1.1.0/CMakeLists.txt
--- origsrc/partio-1.1.0/CMakeLists.txt	2012-03-05 16:32:10.000000000 +0900
+++ src/partio-1.1.0/CMakeLists.txt	2018-04-15 17:13:14.993000000 +0900
@@ -43,7 +43,7 @@ SET ( CMAKE_VERBOSE_MAKEFILE TRUE )
 
 ## project name & version
 PROJECT( partio )
-SET( ${PROJECT_NAME}_MAJOR_VERSION 0 )
+SET( ${PROJECT_NAME}_MAJOR_VERSION 1 )
 SET( ${PROJECT_NAME}_MINOR_VERSION 1 )
 SET( ${PROJECT_NAME}_PATCH_LEVEL 0 )
 
@@ -74,10 +74,12 @@ IF(NOT CMAKE_BUILD_TYPE)
       FORCE)
 ENDIF(NOT CMAKE_BUILD_TYPE)
 
-## Set install location
-EXECUTE_PROCESS(COMMAND sh -c "echo `uname`-`uname -r | cut -d'-' -f1`-`uname -m`" OUTPUT_VARIABLE VARIANT_DIRECTORY OUTPUT_STRIP_TRAILING_WHITESPACE)
-#EXECUTE_PROCESS(COMMAND uname OUTPUT_VARIABLE VARIANT_DIRECTORY OUTPUT_STRIP_TRAILING_WHITESPACE)
-SET(CMAKE_INSTALL_PREFIX "${CMAKE_SOURCE_DIR}/${VARIANT_DIRECTORY}")
+IF(NOT CYGWIN)
+    ## Set install location
+    EXECUTE_PROCESS(COMMAND sh -c "echo `uname`-`uname -r | cut -d'-' -f1`-`uname -m`" OUTPUT_VARIABLE VARIANT_DIRECTORY OUTPUT_STRIP_TRAILING_WHITESPACE)
+    #EXECUTE_PROCESS(COMMAND uname OUTPUT_VARIABLE VARIANT_DIRECTORY OUTPUT_STRIP_TRAILING_WHITESPACE)
+    SET(CMAKE_INSTALL_PREFIX "${CMAKE_SOURCE_DIR}/${VARIANT_DIRECTORY}")
+ENDIF()
 
 ## Search for useful libraries
 find_package(GLUT REQUIRED)
@@ -103,4 +105,9 @@ ADD_SUBDIRECTORY (src/py)
 ADD_SUBDIRECTORY (src/tests)
 ADD_SUBDIRECTORY (src/doc)
 
-
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/partio.pc.in ${CMAKE_BINARY_DIR}/partio.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/partio.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/partio-1.1.0/partio.pc.in src/partio-1.1.0/partio.pc.in
--- origsrc/partio-1.1.0/partio.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/partio-1.1.0/partio.pc.in	2018-04-15 17:11:01.918604000 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: partio
+Description: Library for particle IO and manipulation
+Version: @VERSION@
+Libs: -L${libdir} -lpartio
+Cflags: -I${includedir}
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/partio-1.1.0/src/doc/CMakeLists.txt src/partio-1.1.0/src/doc/CMakeLists.txt
--- origsrc/partio-1.1.0/src/doc/CMakeLists.txt	2012-03-05 16:32:10.000000000 +0900
+++ src/partio-1.1.0/src/doc/CMakeLists.txt	2018-04-15 17:00:17.007999800 +0900
@@ -31,8 +31,9 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.
 
+option(ENABLE_DOC "enable doc" OFF)
 find_package(Doxygen)
-IF(DOXYGEN_FOUND)
+IF(ENABLE_DOC AND DOXYGEN_FOUND)
   configure_file(${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile.in ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile @ONLY)
   FILE(GLOB (DOCUMENTED_FILES ../lib/*.h))
   add_custom_target(doc ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR} 
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/partio-1.1.0/src/lib/CMakeLists.txt src/partio-1.1.0/src/lib/CMakeLists.txt
--- origsrc/partio-1.1.0/src/lib/CMakeLists.txt	2012-03-05 16:32:10.000000000 +0900
+++ src/partio-1.1.0/src/lib/CMakeLists.txt	2018-04-15 16:46:14.221999800 +0900
@@ -35,7 +35,12 @@ FILE(GLOB io_cpp "io/*.cpp")
 FILE(GLOB core_cpp "core/*.cpp")
 ADD_LIBRARY (partio ${io_cpp} ${core_cpp})
 
+TARGET_LINK_LIBRARIES(partio ${ZLIB_LIBRARY})
+IF(CYGWIN)
+    SET_TARGET_PROPERTIES(partio PROPERTIES SOVERSION 0)
+ENDIF()
+
 FILE(GLOB public_includes "*.h")
 
-INSTALL (TARGETS partio DESTINATION lib)
+INSTALL (TARGETS partio RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 INSTALL (FILES ${public_includes} DESTINATION include)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/partio-1.1.0/src/py/CMakeLists.txt src/partio-1.1.0/src/py/CMakeLists.txt
--- origsrc/partio-1.1.0/src/py/CMakeLists.txt	2012-03-05 16:32:10.000000000 +0900
+++ src/partio-1.1.0/src/py/CMakeLists.txt	2018-04-15 17:07:53.043000000 +0900
@@ -58,9 +58,14 @@ SET_SOURCE_FILES_PROPERTIES(partio.i PRO
 SWIG_ADD_MODULE(partio python partio.i)
 SWIG_LINK_LIBRARIES(partio ${PYTHON_LIBRARIES} ${ZLIB_LIBRARY} partio)
 
-SET(PYTHON_DEST "lib64/python${PYTHON_VERSION}/site-packages" )
-INSTALL(TARGETS _partio DESTINATION ${PYTHON_DEST})
-INSTALL(FILES ${CMAKE_BINARY_DIR}/${outdir}/src/py/partio.py  DESTINATION ${PYTHON_DEST})
+IF(CYGWIN)
+    INSTALL(TARGETS _partio DESTINATION "lib/python3.6/site-packages")
+    INSTALL(FILES ${CMAKE_SWIG_OUTDIR}/partio.py DESTINATION "lib/python3.6/site-packages")
+ELSE()
+    SET(PYTHON_DEST "lib64/python${PYTHON_VERSION}/site-packages" )
+    INSTALL(TARGETS _partio DESTINATION ${PYTHON_DEST})
+    INSTALL(FILES ${CMAKE_BINARY_DIR}/${outdir}/src/py/partio.py  DESTINATION ${PYTHON_DEST})
+ENDIF()
 
 ENDIF(PYTHONLIBS_FOUND)
 ENDIF(SWIG_FOUND)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x x86_64-pc-cygwin -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/partio-1.1.0/src/tests/CMakeLists.txt src/partio-1.1.0/src/tests/CMakeLists.txt
--- origsrc/partio-1.1.0/src/tests/CMakeLists.txt	2012-03-05 16:32:10.000000000 +0900
+++ src/partio-1.1.0/src/tests/CMakeLists.txt	2018-04-15 17:08:34.542000000 +0900
@@ -34,5 +34,5 @@
 foreach(item testiterator test testcache teststr makecircle makeline testkdtree)
     ADD_EXECUTABLE(${item} "${item}.cpp")
     target_link_libraries(${item} ${PARTIO_LIBRARIES})
-    install(TARGETS ${item} DESTINATION test)
+    install(TARGETS ${item} DESTINATION share/partio/test)
 endforeach(item)
