--- origsrc/fastrpc-8.0.15/src/Makefile.am	2018-12-03 22:01:27.000000000 +0900
+++ src/fastrpc-8.0.15/src/Makefile.am	2020-11-08 23:28:57.421399900 +0900
@@ -78,7 +78,7 @@ libfastrpc_la_SOURCES = frpcvalue.cc frp
                         frpcb64writer.cc frpcconfig.cc frpccompare.cc
 
 # with these flags (version info etc.)
-libfastrpc_la_LDFLAGS = @VERSION_INFO@ $(DEPS_LIBS)
+libfastrpc_la_LDFLAGS = -no-undefined @VERSION_INFO@ $(DEPS_LIBS)
 
 check_PROGRAMS=test-base64 test-protocol test-marshallers
 test_base64_SOURCES=../test/base64.cc
--- origsrc/fastrpc-8.0.15/src/frpcconnector.cc	2018-12-03 22:01:27.000000000 +0900
+++ src/fastrpc-8.0.15/src/frpcconnector.cc	2020-11-08 23:50:34.779381300 +0900
@@ -233,6 +233,8 @@ SimpleConnector_t::SimpleConnector_t(con
 
 #if (defined(sun) || defined(__sun)) && (defined(__SVR4) || defined(__svr4__))
     he = gethostbyname_r(url.host.c_str(), &h, tmpbuf, 1024, &errcode);
+#elif defined(__CYGWIN__)
+    he = gethostbyname(url.host.c_str());
 #else
     gethostbyname_r(url.host.c_str(), &h, tmpbuf, 1024, &he, &errcode);
 #endif
--- origsrc/fastrpc-8.0.15/test/marshallers.cc	2018-12-03 22:01:27.000000000 +0900
+++ src/fastrpc-8.0.15/test/marshallers.cc	2020-11-08 23:55:10.253789100 +0900
@@ -777,7 +777,11 @@ void runTests(const TestSettings_t &ts)
 }
 
 int main(int argc, const char *argv[]) {
+#if defined(__CYGWIN__)
+    doColor = ::isatty(::fileno(stderr));
+#else
     doColor = ::isatty(::fileno(::stderr));
+#endif
 
     Args_t args(argv + 1, argv + argc);
     TestSettings_t ts;
