--- origsrc/regex-opt-1.2.4/Makefile.sets	2017-11-14 04:53:59.000000000 +0900
+++ src/regex-opt-1.2.4/Makefile.sets	2026-07-21 08:28:53.823098200 +0900
@@ -1,9 +1,9 @@
 INSTALL=install
-CC=gcc
-CXX=g++
+CC?=gcc
+CXX?=g++
 
-CPP=$(CC)
-LDFLAGS=-g
+CPP?=$(CC)
+LDFLAGS?=-g
 #LDLIBS=-lstdc++
 
 CWARNINGS=-Wall -W -pipe -g \
@@ -12,11 +12,12 @@ CWARNINGS=-Wall -W -pipe -g \
     -Winline
 #   -Wtraditional  -fmessage-length=128
 
-CXXFLAGS=-pedantic -ffast-math $(CWARNINGS)
+CXXFLAGS?=-pedantic -ffast-math $(CWARNINGS)
 #CFLAGS=-O3 -fomit-frame-pointer -ffast-math $(CWARNINGS)
-CFLAGS=-O -g $(CWARNINGS)
+CFLAGS?=-O -g $(CWARNINGS)
 
-CPPFLAGS=-DVERSION=\"$(VERSION)\" \
+CPPFLAGS?=
+CPPFLAGS+=-DVERSION=\"$(VERSION)\" \
          -DCONFIG=\"$(CFGDIR)/$(CFGFILE)\" \
          -DBINDIR=\"$(BINDIR)\" $(DEFS)
 
