--- origsrc/protozero-1.8.1/CMakeLists.txt	2025-07-16 02:28:17.000000000 +0900
+++ src/protozero-1.8.1/CMakeLists.txt	2025-09-18 19:49:55.977594700 +0900
@@ -162,3 +162,10 @@ endif()
 add_subdirectory(tools)
 
 #-----------------------------------------------------------------------------
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/protozero.pc.in ${CMAKE_BINARY_DIR}/protozero.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/protozero.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/protozero-1.8.1/protozero.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/protozero-1.8.1/protozero.pc.in	2025-09-18 19:50:15.343163900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: protozero
+Description: Minimalist protocol buffer decoder and encoder in C++
+Version: @PROJECT_VERSION@
+Libs:
+Cflags: -I${includedir}
