--- origsrc/libvirt-glib-0.2.0/configure.ac	2014-12-16 01:17:48.000000000 +0900
+++ src/libvirt-glib-0.2.0/configure.ac	2014-12-20 13:48:08.650653200 +0900
@@ -122,31 +122,6 @@ AC_SUBST(GLIB_MKENUMS)
 
 GLIB_TESTS
 
-dnl Extra link-time flags for Cygwin.
-dnl Copied from libxml2 configure.in, but I removed mingw changes
-dnl for now since I'm not supporting mingw at present.  - RWMJ
-CYGWIN_EXTRA_LDFLAGS=
-CYGWIN_EXTRA_LIBADD=
-CYGWIN_EXTRA_PYTHON_LIBADD=
-MINGW_EXTRA_LDFLAGS=
-case "$host" in
-  *-*-cygwin*)
-    CYGWIN_EXTRA_LDFLAGS="-no-undefined"
-    CYGWIN_EXTRA_LIBADD="${INTLLIBS}"
-    if test "x$PYTHON_VERSION" != "x"; then
-      CYGWIN_EXTRA_PYTHON_LIBADD="-L/usr/lib/python${PYTHON_VERSION}/config -lpython${PYTHON_VERSION}"
-    fi
-    ;;
-  *-*-mingw*)
-    MINGW_EXTRA_LDFLAGS="-no-undefined"
-    ;;
-esac
-AC_SUBST([CYGWIN_EXTRA_LDFLAGS])
-AC_SUBST([CYGWIN_EXTRA_LIBADD])
-AC_SUBST([CYGWIN_EXTRA_PYTHON_LIBADD])
-AC_SUBST([MINGW_EXTRA_LDFLAGS])
-
-
 AC_ARG_ENABLE([test-coverage],
 [  --enable-test-coverage  turn on code coverage instrumentation],
 [case "${enableval}" in
@@ -245,6 +220,30 @@ AC_SUBST([PYTHON_VERSION])
 AC_SUBST([PYTHON_INCLUDES])
 AC_SUBST([PYTHON_SITE_PACKAGES])
 
+dnl Extra link-time flags for Cygwin.
+dnl Copied from libxml2 configure.in, but I removed mingw changes
+dnl for now since I'm not supporting mingw at present.  - RWMJ
+CYGWIN_EXTRA_LDFLAGS=
+CYGWIN_EXTRA_LIBADD=
+CYGWIN_EXTRA_PYTHON_LIBADD=
+MINGW_EXTRA_LDFLAGS=
+case "$host" in
+  *-*-cygwin*)
+    CYGWIN_EXTRA_LDFLAGS="-no-undefined"
+    CYGWIN_EXTRA_LIBADD="${INTLLIBS}"
+    if test "x$PYTHON_VERSION" != "x"; then
+      CYGWIN_EXTRA_PYTHON_LIBADD="-L/usr/lib/python${PYTHON_VERSION}/config -lpython${PYTHON_VERSION}"
+    fi
+    ;;
+  *-*-mingw*)
+    MINGW_EXTRA_LDFLAGS="-no-undefined"
+    ;;
+esac
+AC_SUBST([CYGWIN_EXTRA_LDFLAGS])
+AC_SUBST([CYGWIN_EXTRA_LIBADD])
+AC_SUBST([CYGWIN_EXTRA_PYTHON_LIBADD])
+AC_SUBST([MINGW_EXTRA_LDFLAGS])
+
 GOBJECT_INTROSPECTION_CHECK([$GOBJECT_INTROSPECTION_REQUIRED])
 
 with_linux=no
--- origsrc/libvirt-glib-0.2.0/python/Makefile.am	2014-12-16 01:17:48.000000000 +0900
+++ src/libvirt-glib-0.2.0/python/Makefile.am	2014-12-20 13:47:52.051033700 +0900
@@ -17,7 +17,7 @@ libvirtglibmod_la_CFLAGS = -Wno-redundan
 
 libvirtglibmod_la_LDFLAGS = -module -avoid-version \
 	$(CYGWIN_EXTRA_LDFLAGS)
-libvirtglibmod_la_LIBADD = ../libvirt-glib/libvirt-glib-1.0.la
+libvirtglibmod_la_LIBADD = ../libvirt-glib/libvirt-glib-1.0.la \
 	$(CYGWIN_EXTRA_LIBADD) $(CYGWIN_EXTRA_PYTHON_LIBADD)
 
 
