--- origsrc/nlopt-2.2.4/Makefile.am	2011-06-10 02:18:17.000000000 +0900
+++ src/nlopt-2.2.4/Makefile.am	2011-08-08 00:45:50.132519600 +0900
@@ -23,11 +23,7 @@ cobyla/libcobyla.la newuoa/libnewuoa.la
 auglag/libauglag.la bobyqa/libbobyqa.la isres/libisres.la		\
 slsqp/libslsqp.la api/libapi.la util/libutil.la
 
-if WITH_CXX
-libnlopt@NLOPT_SUFFIX@_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@
-else
 libnlopt@NLOPT_SUFFIX@_la_LDFLAGS = -no-undefined -version-info @SHARED_VERSION_INFO@
-endif
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = nlopt.pc
