--- origsrc/nlopt-2.8.0/cmake/FindGuile.cmake	2024-08-10 05:31:24.000000000 +0900
+++ src/nlopt-2.8.0/cmake/FindGuile.cmake	2024-10-16 08:51:47.288275100 +0900
@@ -13,30 +13,36 @@
 # GUILE_EXTENSION_DIR    - extension dir
 # GUILE_ROOT_DIR         - prefix dir
 
-# Look for the header file
-# Look for guile-2.2 first, then 2.0, then 1.8
-# Macports for OSX puts things in /opt/local
-find_path (GUILE_INCLUDE_DIR libguile.h
-  PATH_SUFFIXES
-    guile/3.0
-    guile/2.2
-    guile/2.0
-    guile/1.8
-    libguile
-    guile
-  HINTS /opt/local/include
-)
-
-# Look for the library
-find_library (GUILE_LIBRARY NAMES guile-3.0 guile-2.2 guile-2.0 guile
-  HINTS
-    /opt/local/lib
-)
-
-
-set (GUILE_LIBRARIES ${GUILE_LIBRARY})
-set (GUILE_INCLUDE_DIRS ${GUILE_INCLUDE_DIR})
 
+if (CYGWIN)
+  find_package(PkgConfig)
+  pkg_check_modules(GUILE guile-3.0)
+  set(GUILE_INCLUDE_DIR ${GUILE_INCLUDE_DIRS})
+  set(GUILE_LIBRARY ${GUILE_LIBRARIES})
+else()
+  # Look for the header file
+  # Look for guile-2.2 first, then 2.0, then 1.8
+  # Macports for OSX puts things in /opt/local
+  find_path (GUILE_INCLUDE_DIR libguile.h
+    PATH_SUFFIXES
+      guile/3.0
+      guile/2.2
+      guile/2.0
+      guile/1.8
+      libguile
+      guile
+    HINTS /opt/local/include
+  )
+
+  # Look for the library
+  find_library (GUILE_LIBRARY NAMES guile-3.0 guile-2.2 guile-2.0 guile
+    HINTS
+      /opt/local/lib
+  )
+  
+  set (GUILE_LIBRARIES ${GUILE_LIBRARY})
+  set (GUILE_INCLUDE_DIRS ${GUILE_INCLUDE_DIR})
+endif()
 
 # check guile's version if we're using cmake >= 2.6
 if (GUILE_INCLUDE_DIR)
--- origsrc/nlopt-2.8.0/src/octave/CMakeLists.txt	2024-08-10 05:31:24.000000000 +0900
+++ src/nlopt-2.8.0/src/octave/CMakeLists.txt	2024-10-16 09:22:12.430337800 +0900
@@ -49,7 +49,7 @@ if (OCTAVE_FOUND)
   if (NOT DEFINED INSTALL_OCT_DIR)
     file (RELATIVE_PATH INSTALL_OCT_DIR ${OCTAVE_ROOT_DIR} ${OCTAVE_OCT_SITE_DIR})
   endif ()
-  install (TARGETS nlopt_optimize DESTINATION ${INSTALL_OCT_DIR})
+  install (TARGETS nlopt_optimize RUNTIME DESTINATION ${INSTALL_OCT_DIR})
 
   if (NOT DEFINED INSTALL_M_DIR)
     file (RELATIVE_PATH INSTALL_M_DIR ${OCTAVE_ROOT_DIR} ${OCTAVE_M_SITE_DIR})
