--- origsrc/nlopt-2.7.1/cmake/FindGuile.cmake	2021-12-04 03:38:01.000000000 +0900
+++ src/nlopt-2.7.1/cmake/FindGuile.cmake	2022-06-03 11:55:04.702511300 +0900
@@ -13,30 +13,36 @@
 # GUILE_EXTENSION_DIR    - extension dir
 # GUILE_ROOT_DIR         - prefix dir
 
-# Look for the header file
-# Look for guile-2.2 first, then 2.0, then 1.8
-# Macports for OSX puts things in /opt/local
-find_path (GUILE_INCLUDE_DIR libguile.h
-  PATH_SUFFIXES
-    guile/3.0
-    guile/2.2
-    guile/2.0
-    guile/1.8
-    libguile
-    guile
-  HINTS /opt/local/include
-)
-
-# Look for the library
-find_library (GUILE_LIBRARY NAMES guile-3.0 guile-2.2 guile-2.0 guile
-  HINTS
-    /opt/local/lib
-)
-
-
-set (GUILE_LIBRARIES ${GUILE_LIBRARY})
-set (GUILE_INCLUDE_DIRS ${GUILE_INCLUDE_DIR})
 
+if (CYGWIN)
+  find_package(PkgConfig)
+  pkg_check_modules(GUILE guile-3.0)
+  set(GUILE_INCLUDE_DIR ${GUILE_INCLUDE_DIRS})
+  set(GUILE_LIBRARY ${GUILE_LIBRARIES})
+else()
+  # Look for the header file
+  # Look for guile-2.2 first, then 2.0, then 1.8
+  # Macports for OSX puts things in /opt/local
+  find_path (GUILE_INCLUDE_DIR libguile.h
+    PATH_SUFFIXES
+      guile/3.0
+      guile/2.2
+      guile/2.0
+      guile/1.8
+      libguile
+      guile
+    HINTS /opt/local/include
+  )
+
+  # Look for the library
+  find_library (GUILE_LIBRARY NAMES guile-3.0 guile-2.2 guile-2.0 guile
+    HINTS
+      /opt/local/lib
+  )
+  
+  set (GUILE_LIBRARIES ${GUILE_LIBRARY})
+  set (GUILE_INCLUDE_DIRS ${GUILE_INCLUDE_DIR})
+endif()
 
 # check guile's version if we're using cmake >= 2.6
 if (GUILE_INCLUDE_DIR)
