--- origsrc/nlopt-2.10.0/CMakeLists.txt	2025-02-05 03:29:30.000000000 +0900
+++ src/nlopt-2.10.0/CMakeLists.txt	2026-01-26 15:52:45.930343400 +0900
@@ -343,8 +343,16 @@ else ()
   set (INSTALL_PYTHON_DIR ${NLOPT_INSTALL_LIBDIR}/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages CACHE STRING "site packages dir")
 endif ()
 
+find_package(pkgconfig)
+
 if (NLOPT_GUILE)
-  find_package (Guile)
+  pkg_check_modules(GUILE guile-3.0)
+  pkg_get_variable(GUILE_ROOT_DIR guile-3.0 prefix)
+  pkg_get_variable(GUILE_SITE_DIR guile-3.0 sitedir)
+  pkg_get_variable(GUILE_EXTENSION_DIR guile-3.0 extensiondir)
+  if (NOT GUILE_FOUND)
+    find_package (Guile)
+  endif()
 endif ()
 
 if (NLOPT_JAVA)
--- origsrc/nlopt-2.10.0/src/octave/CMakeLists.txt	2025-02-05 03:29:30.000000000 +0900
+++ src/nlopt-2.10.0/src/octave/CMakeLists.txt	2026-01-26 15:43:39.604872700 +0900
@@ -45,7 +45,7 @@ if (OCTAVE_FOUND)
   if (NOT DEFINED INSTALL_OCT_DIR)
     file (RELATIVE_PATH INSTALL_OCT_DIR ${OCTAVE_ROOT_DIR} ${OCTAVE_OCT_SITE_DIR})
   endif ()
-  install (TARGETS nlopt_optimize DESTINATION ${INSTALL_OCT_DIR})
+  install (TARGETS nlopt_optimize RUNTIME DESTINATION ${INSTALL_OCT_DIR})
 
   if (NOT DEFINED INSTALL_M_DIR)
     file (RELATIVE_PATH INSTALL_M_DIR ${OCTAVE_ROOT_DIR} ${OCTAVE_M_SITE_DIR})
