--- origsrc/nlopt-2.4.2/Makefile.am	2013-11-03 02:39:41.000000000 +0900
+++ src/nlopt-2.4.2/Makefile.am	2014-06-09 23:42:25.670703500 +0900
@@ -17,13 +17,9 @@ cdirect/libcdirect.la $(CXX_LIBS) praxis
 luksan/libluksan.la crs/libcrs.la mlsl/libmlsl.la mma/libmma.la		\
 cobyla/libcobyla.la newuoa/libnewuoa.la neldermead/libneldermead.la	\
 auglag/libauglag.la bobyqa/libbobyqa.la isres/libisres.la		\
-slsqp/libslsqp.la esch/libesch.la api/libapi.la util/libutil.la
+slsqp/libslsqp.la esch/libesch.la api/libapi.la util/libutil.la -lstdc++
 
-if WITH_CXX
-libnlopt@NLOPT_SUFFIX@_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@
-else
 libnlopt@NLOPT_SUFFIX@_la_LDFLAGS = -no-undefined -version-info @SHARED_VERSION_INFO@
-endif
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = nlopt.pc
