--- origsrc/mbelib-1.2.5/CMakeLists.txt	2013-08-10 07:26:13.000000000 +0900
+++ src/mbelib-1.2.5/CMakeLists.txt	2013-10-06 10:49:49.583633500 +0900
@@ -5,15 +5,13 @@ FILE(GLOB SRCS *.c)
  
 include_directories("${PROJECT_SOURCE_DIR}")
  
-ADD_LIBRARY(mbe-static STATIC ${SRCS})
 ADD_LIBRARY(mbe-shared SHARED ${SRCS})
-TARGET_LINK_LIBRARIES(mbe-static m)
 TARGET_LINK_LIBRARIES(mbe-shared m)
  
-install(TARGETS mbe-static mbe-shared DESTINATION lib)
+install(TARGETS mbe-shared RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 install (FILES "${PROJECT_SOURCE_DIR}/mbelib.h" DESTINATION include)
  
-set_target_properties(mbe-static mbe-shared PROPERTIES OUTPUT_NAME mbe
+set_target_properties(mbe-shared PROPERTIES OUTPUT_NAME mbe
                       VERSION 1.0 SOVERSION 1 INSTALL_NAME_DIR ${CMAKE_INSTALL_PREFIX}/lib )
 
 # uninstall target
