--- origsrc/libieee1284-0_2_11/Makefile.am	2007-09-18 18:58:42.000000000 +0900
+++ src/libieee1284-0_2_11/Makefile.am	2022-07-04 10:32:35.322876600 +0900
@@ -8,7 +8,7 @@ libieee1284_la_SOURCES = src/detect.c sr
 	src/par_nt.h src/io.h src/conf.h src/conf.c libieee1284.sym
 # When rolling a release, remember to adjust the version info.
 # It's current:release:age.
-libieee1284_la_LDFLAGS = -version-info 5:2:2 \
+libieee1284_la_LDFLAGS = -no-undefined -version-info 5:2:2 \
 	-export-symbols $(top_srcdir)/libieee1284.sym
 include_HEADERS = include/ieee1284.h
 INCLUDES = -I$(top_srcdir)/include
@@ -34,9 +34,9 @@ libieee1284_test_LDADD = libieee1284.la
 src_ieee1284module_la_SOURCES = src/ieee1284module.c
 if PYTHON
 pyexec_LTLIBRARIES = src/ieee1284module.la
-src_ieee1284module_la_LDFLAGS = -module -avoid-version
+src_ieee1284module_la_LDFLAGS = -no-undefined -module -avoid-version
 src_ieee1284module_la_LIBADD = libieee1284.la
-src_ieee1284module_la_CFLAGS = -I/usr/include/python$(PYTHON_VERSION) -fpic
+src_ieee1284module_la_CFLAGS = -I/usr/include/python$(PYTHON_VERSION)
 endif
 
 man3_MANS = doc/libieee1284.3 \
@@ -93,3 +93,6 @@ MAINTAINERCLEANFILES = $(man3_MANS)
 
 tag:
 	cvs tag -c `echo V@VERSION@ | tr . _`
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libieee1284-0_2_11/configure.in	2007-09-18 18:58:42.000000000 +0900
+++ src/libieee1284-0_2_11/configure.in	2022-07-04 10:32:35.337753500 +0900
@@ -75,4 +75,5 @@ fi
 dnl Checks for library functions.
 
 AC_CONFIG_FILES(Makefile libieee1284.spec)
+AC_CONFIG_FILES([libieee1284.pc])
 AC_OUTPUT
--- origsrc/libieee1284-0_2_11/libieee1284.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libieee1284-0_2_11/libieee1284.pc.in	2022-07-04 10:32:35.239057900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libieee1284
+Description: Parallel port communication library
+Version: @VERSION@
+Libs: -L${libdir} -lieee1284
+Cflags: -I${includedir}
