--- origsrc/cdescent-0.6.0/Makefile.am	2015-11-11 11:26:29.000000000 +0900
+++ src/cdescent-0.6.0/Makefile.am	2025-08-15 19:09:16.167188200 +0900
@@ -2,16 +2,6 @@
 
 SUBDIRS = src include example
 
-libcdescentdocdir = ${prefix}/doc/cdescent
-libcdescentdoc_DATA = \
-	README\
-	COPYING\
-	AUTHORS\
-	ChangeLog\
-	INSTALL\
-	NEWS\
-	TODO
-
 EXTRA_DIST = $(libcdescentdoc_DATA)
 
 # Copy all the spec files. Of cource, only one is actually used.
@@ -22,3 +12,5 @@ dist-hook:
 		fi \
 	done
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/cdescent-0.6.0/example/Makefile.am	2015-11-11 11:26:29.000000000 +0900
+++ src/cdescent-0.6.0/example/Makefile.am	2025-08-15 19:19:39.776588500 +0900
@@ -13,9 +13,7 @@ libexample_la_CPPFLAGS =\
 	-I$(top_srcdir)/include\
 	-I$(top_srcdir)/example
 
-libexample_la_LDFLAGS = -L$(top_srcdir)/src
-
-libexample_la_LIBADD = -lcdescent
+libexample_la_LIBADD = $(top_builddir)/src/libcdescent.la
 
 # sample program
 elasticnet_SOURCES = elasticnet.c
@@ -24,7 +22,7 @@ elasticnet_CPPFLAGS =\
 	-I$(top_srcdir)/example
 
 elasticnet_LDADD =\
-	$(top_srcdir)/example/libexample.la
+	libexample.la
 
 # sample program
 adaelasticnet_SOURCES = adaelasticnet.c
@@ -33,4 +31,4 @@ adaelasticnet_CPPFLAGS =\
 	-I$(top_srcdir)/example
 
 adaelasticnet_LDADD =\
-	$(top_srcdir)/example/libexample.la
+	libexample.la
--- origsrc/cdescent-0.6.0/src/Makefile.am	2015-11-11 11:26:29.000000000 +0900
+++ src/cdescent-0.6.0/src/Makefile.am	2025-08-15 19:09:16.171215900 +0900
@@ -22,6 +22,6 @@ libcdescent_la_SOURCES =\
 
 libcdescent_la_LIBADD = -L$(top_srcdir)/src
 libcdescent_la_CPPFLAGS = -I$(top_srcdir)/include
-libcdescent_la_LDFLAGS = -version-info $(CDESCENT_LT_VERSION)
+libcdescent_la_LDFLAGS = -no-undefined -version-info $(CDESCENT_LT_VERSION)
 
 
