--- origsrc/libfguid-20240415/Makefile.am	2024-04-15 14:25:11.000000000 +0900
+++ src/libfguid-20240415/Makefile.am	2025-05-17 15:19:39.823271000 +0900
@@ -3,9 +3,7 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
 	libfguid \
-	po \
 	manuals \
 	tests \
 	msvscpp
--- origsrc/libfguid-20240415/configure.ac	2024-04-15 14:25:11.000000000 +0900
+++ src/libfguid-20240415/configure.ac	2025-05-17 15:19:39.823271000 +0900
@@ -91,7 +91,6 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
 AC_CONFIG_FILES([libfguid/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
