--- origsrc/redis-2.4.7/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/redis-2.4.7/Makefile.am	2012-02-10 10:16:57.140390300 +0900
@@ -0,0 +1 @@
+SUBDIRS = deps src
--- origsrc/redis-2.4.7/configure.ac	1970-01-01 09:00:00.000000000 +0900
+++ src/redis-2.4.7/configure.ac	2012-02-10 10:18:46.519489500 +0900
@@ -0,0 +1,56 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([redis], [2.4.7], [http://redis.io/])
+AC_CONFIG_SRCDIR([deps/hiredis/async.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_MAKE_SET
+LT_INIT([disable-static])
+
+# Checks for libraries.
+PKG_CHECK_MODULES(LIBEVENT, libevent)
+
+# Checks for header files.
+AC_CHECK_HEADERS([arpa/inet.h fcntl.h float.h inttypes.h limits.h netdb.h netinet/in.h stdint.h stdlib.h string.h strings.h sys/ioctl.h sys/socket.h sys/time.h syslog.h termios.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_C_INLINE
+AC_TYPE_INT16_T
+AC_TYPE_INT32_T
+AC_TYPE_INT64_T
+AC_TYPE_INT8_T
+AC_TYPE_MODE_T
+AC_TYPE_OFF_T
+AC_TYPE_PID_T
+AC_TYPE_SIZE_T
+AC_TYPE_SSIZE_T
+AC_TYPE_UINT16_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+AC_TYPE_UINT8_T
+
+# Checks for library functions.
+AC_FUNC_ERROR_AT_LINE
+AC_FUNC_FORK
+AC_FUNC_FSEEKO
+AC_FUNC_MALLOC
+AC_FUNC_MMAP
+AC_FUNC_REALLOC
+AC_FUNC_STRCOLL
+AC_FUNC_STRTOD
+AC_FUNC_WAIT3
+AC_CHECK_FUNCS([atexit bzero dup2 ftruncate getcwd gethostbyname gettimeofday inet_ntoa memmove memset munmap select socket strcasecmp strchr strdup strerror strncasecmp strstr strtol])
+
+AC_CONFIG_FILES([Makefile
+                 deps/Makefile
+                 deps/hiredis/Makefile
+                 deps/hiredis/adapters/Makefile
+                 deps/linenoise/Makefile
+                 src/Makefile])
+AC_OUTPUT
--- origsrc/redis-2.4.7/deps/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/redis-2.4.7/deps/Makefile.am	2012-02-10 10:16:57.147331100 +0900
@@ -0,0 +1 @@
+SUBDIRS = hiredis linenoise
--- origsrc/redis-2.4.7/deps/hiredis/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/redis-2.4.7/deps/hiredis/Makefile.am	2012-02-10 10:16:57.154271900 +0900
@@ -0,0 +1,19 @@
+SUBDIRS = adapters
+
+lib_LTLIBRARIES = libhiredis.la
+
+hiredis_includedir = $(includedir)/hiredis
+hiredis_include_HEADERS = hiredis.h async.h
+
+bin_PROGRAMS = hiredis-example-libevent hiredis-test
+
+TESTS = hiredis-test
+
+hiredis_example_libevent_SOURCES = example-libevent.c
+hiredis_example_libevent_LDADD = libhiredis.la $(LIBEVENT_LIBS)
+
+hiredis_test_SOURCES = test.c
+hiredis_test_LDADD = libhiredis.la
+
+libhiredis_la_SOURCES = net.c hiredis.c sds.c async.c
+libhiredis_la_LDFLAGS = -no-undefined
--- origsrc/redis-2.4.7/deps/hiredis/adapters/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/redis-2.4.7/deps/hiredis/adapters/Makefile.am	2012-02-10 10:16:57.154271900 +0900
@@ -0,0 +1,2 @@
+hiredis_adapters_includedir = $(includedir)/hiredis/adapters
+hiredis_adapters_include_HEADERS = ae.h libev.h libevent.h
--- origsrc/redis-2.4.7/deps/hiredis/adapters/libevent.h	2012-02-02 23:29:24.000000000 +0900
+++ src/redis-2.4.7/deps/hiredis/adapters/libevent.h	2012-02-10 10:16:57.161212700 +0900
@@ -52,7 +52,7 @@ int redisLibeventAttach(redisAsyncContex
     redisLibeventEvents *e;
 
     /* Nothing should be attached when something is already attached */
-    if (ac->_adapter_data != NULL)
+    if (ac->ev.data != NULL)
         return REDIS_ERR;
 
     /* Create container for context and r/w events */
@@ -60,12 +60,12 @@ int redisLibeventAttach(redisAsyncContex
     e->context = ac;
 
     /* Register functions to start/stop listening for events */
-    ac->evAddRead = redisLibeventAddRead;
-    ac->evDelRead = redisLibeventDelRead;
-    ac->evAddWrite = redisLibeventAddWrite;
-    ac->evDelWrite = redisLibeventDelWrite;
-    ac->evCleanup = redisLibeventCleanup;
-    ac->_adapter_data = e;
+    ac->ev.addRead = redisLibeventAddRead;
+    ac->ev.delRead = redisLibeventDelRead;
+    ac->ev.addWrite = redisLibeventAddWrite;
+    ac->ev.delWrite = redisLibeventDelWrite;
+    ac->ev.cleanup = redisLibeventCleanup;
+    ac->ev.data = e;
 
     /* Initialize and install read/write events */
     event_set(&e->rev,c->fd,EV_READ,redisLibeventReadEvent,e);
--- origsrc/redis-2.4.7/deps/linenoise/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/redis-2.4.7/deps/linenoise/Makefile.am	2012-02-10 10:16:57.161212700 +0900
@@ -0,0 +1,2 @@
+noinst_LTLIBRARIES = liblinenoise.la
+liblinenoise_la_SOURCES = linenoise.c
--- origsrc/redis-2.4.7/src/Makefile.am	1970-01-01 09:00:00.000000000 +0900
+++ src/redis-2.4.7/src/Makefile.am	2012-02-10 10:16:57.168153500 +0900
@@ -0,0 +1,15 @@
+INCLUDES = -I$(top_srcdir)/deps/hiredis -I$(top_srcdir)/deps/linenoise
+
+bin_PROGRAMS = redis-server redis-benchmark redis-cli redis-check-dump redis-check-aof
+
+redis_server_SOURCES = adlist.c ae.c anet.c dict.c redis.c sds.c zmalloc.c lzf_c.c lzf_d.c pqsort.c zipmap.c sha1.c ziplist.c release.c networking.c util.c object.c db.c replication.c rdb.c t_string.c t_list.c t_set.c t_zset.c t_hash.c config.c aof.c vm.c pubsub.c multi.c debug.c sort.c intset.c syncio.c slowlog.c bio.c
+
+redis_benchmark_SOURCES = ae.c anet.c redis-benchmark.c sds.c adlist.c zmalloc.c
+redis_benchmark_LDADD = $(top_builddir)/deps/hiredis/libhiredis.la
+
+redis_cli_SOURCES = anet.c sds.c adlist.c redis-cli.c zmalloc.c release.c
+redis_cli_LDADD = $(top_builddir)/deps/hiredis/libhiredis.la $(top_builddir)/deps/linenoise/liblinenoise.la
+
+redis_check_dump_SOURCES = redis-check-dump.c lzf_c.c lzf_d.c
+
+redis_check_aof_SOURCES = redis-check-aof.c
--- origsrc/redis-2.4.7/src/debug.c	2012-02-02 23:29:24.000000000 +0900
+++ src/redis-2.4.7/src/debug.c	2012-02-10 10:16:57.175094300 +0900
@@ -359,7 +359,9 @@ void redisLogObjectDebugInfo(robj *o) {
 }
 
 void _redisAssert(char *estr, char *file, int line) {
+#ifdef HAVE_BACKTRACE
     bugReportStart();
+#endif
     redisLog(REDIS_WARNING,"=== ASSERTION FAILED ===");
     redisLog(REDIS_WARNING,"==> %s:%d '%s' is not true",file,line,estr);
 #ifdef HAVE_BACKTRACE
@@ -372,7 +374,9 @@ void _redisAssert(char *estr, char *file
 }
 
 void _redisPanic(char *msg, char *file, int line) {
+#ifdef HAVE_BACKTRACE
     bugReportStart();
+#endif
     redisLog(REDIS_WARNING,"!!! Software Failure. Press left mouse button to continue");
     redisLog(REDIS_WARNING,"Guru Meditation: %s #%s:%d",msg,file,line);
 #ifdef HAVE_BACKTRACE
--- origsrc/redis-2.4.7/src/redis.c	2012-02-02 23:29:24.000000000 +0900
+++ src/redis-2.4.7/src/redis.c	2012-02-10 10:16:57.182035100 +0900
@@ -55,6 +55,10 @@
 #include <pthread.h>
 #include <sys/resource.h>
 
+#if !defined(SA_ONSTACK)
+#define SA_ONSTACK 0x00000000
+#endif
+
 /* Our shared "common" objects */
 
 struct sharedObjectsStruct shared;
--- origsrc/redis-2.4.7/src/release.h	1970-01-01 09:00:00.000000000 +0900
+++ src/redis-2.4.7/src/release.h	2012-02-10 10:16:57.182035100 +0900
@@ -0,0 +1,2 @@
+#define REDIS_GIT_SHA1 "00000000"
+#define REDIS_GIT_DIRTY "0"
