--- origsrc/redis-2.0.4/utils/redis_init_script	2010-09-09 23:51:41.000000000 +0900
+++ src/redis-2.0.4/utils/redis_init_script	2010-12-12 00:09:38.124816700 +0900
@@ -1,10 +1,10 @@
 #!/bin/sh
 
 REDISPORT=6379
-EXEC=/usr/local/bin/redis-server
+EXEC=/usr/bin/redis-server
 
 PIDFILE=/var/run/redis_${REDISPORT}.pid
-CONF="/etc/redis/${REDISPORT}.conf"
+CONF="/etc/redis.conf"
 
 case "$1" in
     start)
