--- origsrc/maxfetch-stable/maxfetch	2023-06-27 16:28:16.000000000 +0900
+++ src/maxfetch-stable/maxfetch	2023-09-06 23:45:25.067394300 +0900
@@ -34,12 +34,19 @@ up=$(uptime | sed -E 's/^[^,]*up *//; s/
 fetch() {
     echo "┌────────────┐ "
     echo "│  ${red}${normal} user    │ ${red}$(whoami)${normal}"
+    if test -f "/etc/hostname"; then
     echo "│  ${yellow}${normal} hname   │ ${yellow}$(cat /etc/hostname)${normal} "
+    elif command -v hostname &> /dev/null; then
+    echo "│  ${yellow}${normal} hname   │ ${yellow}$(hostname)${normal} "
+    fi
+    if test -f "/etc/os-release"; then
     echo "│  ${green}${normal} distro  │ ${green}$(sed -nE "s@PRETTY_NAME=\"([^\"]*)\"@\1@p" /etc/os-release)${normal} "
+    fi
     echo "│  ${cyan}漣${normal}kernel  │ ${cyan}$(uname -r)${normal} "
     echo "│  ${blue}${normal} de/wm   │ ${blue}$XDG_CURRENT_DESKTOP${normal} "
     echo "│  ${magenta}⏻${normal} uptime  │ ${magenta}${up}${normal} "
     echo "│  ${red}${normal} shell   │ ${red}$(echo ${SHELL##*/})${normal} "
+    ps -o comm= &> /dev/null && \
     echo "│  ${yellow}${normal} term    │ ${yellow}$(ps -o comm= -p "$(($(ps -o ppid= -p "$(($(ps -o sid= -p "$$")))")))")${normal} "
     echo "├────────────┤ "
     echo "│  ${yellow} ${normal}colors  │ ${red}$unicode${normal}${yellow}$unicode${normal}${green}$unicode${normal}${cyan}$unicode${normal}${blue}$unicode${normal}${magenta}$unicode${normal}"
