--- origsrc/clac-0.3.3/makefile	2019-11-17 22:43:15.000000000 +0900
+++ src/clac-0.3.3/makefile	2019-12-24 08:47:53.724936900 +0900
@@ -1,24 +1,16 @@
-FLAGS = -I./deps/linenoise -I./deps/sds
+FLAGS =
 PREFIX?=/usr/local
 MANPREFIX?=${PREFIX}/share/man
 STRIP?=strip
 
 default: clac
 
-deps/linenoise/linenoise.o:
-	@cd deps/linenoise && $(MAKE)
-
-deps/sds/sds.o:
-	@cd deps/sds && $(MAKE)
-
-clac: clac.c deps/linenoise/linenoise.o deps/sds/sds.o
-	$(CC) $(FLAGS) -Wall -Os -o clac clac.c -lm deps/linenoise/linenoise.o deps/sds/sds.o
+clac: clac.c
+	$(CC) $(FLAGS) -Wall -Os -o clac clac.c -lm -llinenoise -lsds
 
 clean:
 	@echo cleaning
 	@rm -f clac
-	@rm -f deps/sds/sds.o
-	@rm -f deps/linenoise/linenoise.o
 
 install: clac
 	@echo stripping executable
