--- origsrc/libwfut-0.2.3/libwfut/Makefile.am	2009-01-10 07:23:46.000000000 +0900
+++ src/libwfut-0.2.3/libwfut/Makefile.am	2015-02-12 22:47:44.910539700 +0900
@@ -1,6 +1,6 @@
 lib_LTLIBRARIES = libwfut-0.2.la
 
-libwfut_0_2_la_LDFLAGS = -version-info @LIBWFUT_VERSION_INFO@
+libwfut_0_2_la_LDFLAGS = -no-undefined -version-info @LIBWFUT_VERSION_INFO@
 #libwfut_0_2_la_LIBADD = ../tinyxml/libtinyxml.a
 
 INCLUDES=-I${top_srcdir} -I$(top_builddir)
--- origsrc/libwfut-0.2.3/python/Makefile.am	2011-01-08 23:51:24.000000000 +0900
+++ src/libwfut-0.2.3/python/Makefile.am	2015-02-12 22:50:59.567771200 +0900
@@ -25,8 +25,8 @@ endif
 pythonfvs_PYTHON =
 pythonfvs_LTLIBRARIES = 
 
-pyWFUT_la_LDFLAGS = -module -avoid-version
-pyWFUT_la_LIBADD = ../libwfut/libwfut-0.2.la @LIBWFUT_LIBS@
+pyWFUT_la_LDFLAGS = -no-undefined -module -avoid-version
+pyWFUT_la_LIBADD = ../libwfut/libwfut-0.2.la -lpython${PYTHON_VERSION}
 pyWFUT_la_SOURCES = WFUT.cxx 
 
 CLEANFILES = \
