--- origsrc/dmidecode-2.12/Makefile	2010-10-11 23:08:16.000000000 +0900
+++ src/dmidecode-2.12/Makefile	2015-05-08 23:28:32.481755300 +0900
@@ -19,14 +19,14 @@ CFLAGS  = -W -Wall -Wshadow -Wstrict-pro
 #CFLAGS += -DALIGNMENT_WORKAROUND
 
 # When debugging, disable -O2 and enable -g.
-CFLAGS += -O2
+CFLAGS += -O3
 #CFLAGS += -g
 
 # Pass linker flags here
 LDFLAGS =
 
 DESTDIR =
-prefix  = /usr/local
+prefix  = /usr
 sbindir = $(prefix)/sbin
 mandir  = $(prefix)/share/man
 man8dir = $(mandir)/man8
