diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/dmidecode-3.0/Makefile src/dmidecode-3.0/Makefile
--- origsrc/dmidecode-3.0/Makefile	2015-09-03 15:03:19.000000000 +0900
+++ src/dmidecode-3.0/Makefile	2015-12-27 15:45:02.546431500 +0900
@@ -23,14 +23,14 @@ CFLAGS += -D_FILE_OFFSET_BITS=64
 #CFLAGS += -DALIGNMENT_WORKAROUND
 
 # When debugging, disable -O2 and enable -g.
-CFLAGS += -O2
+CFLAGS += -O3
 #CFLAGS += -g
 
 # Pass linker flags here
 LDFLAGS =
 
 DESTDIR =
-prefix  = /usr/local
+prefix  = /usr
 sbindir = $(prefix)/sbin
 mandir  = $(prefix)/share/man
 man8dir = $(mandir)/man8
