--- origsrc/dmidecode-3.1/Makefile	2017-05-23 22:34:14.000000000 +0900
+++ src/dmidecode-3.1/Makefile	2017-06-09 20:49:17.473572000 +0900
@@ -23,14 +23,14 @@ CFLAGS += -D_FILE_OFFSET_BITS=64
 #CFLAGS += -DALIGNMENT_WORKAROUND
 
 # When debugging, disable -O2 and enable -g.
-CFLAGS += -O2
+CFLAGS += -O3
 #CFLAGS += -g
 
 # Pass linker flags here
 LDFLAGS =
 
 DESTDIR =
-prefix  = /usr/local
+prefix  = /usr
 sbindir = $(prefix)/sbin
 mandir  = $(prefix)/share/man
 man8dir = $(mandir)/man8
