diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/cmocka-1.1.1/DefineOptions.cmake src/cmocka-1.1.1/DefineOptions.cmake
--- origsrc/cmocka-1.1.1/DefineOptions.cmake	2015-02-13 00:59:22.000000000 +0900
+++ src/cmocka-1.1.1/DefineOptions.cmake	2017-04-30 13:50:09.150668900 +0900
@@ -2,6 +2,6 @@ option(WITH_STATIC_LIB "Build with a sta
 option(WITH_CMOCKERY_SUPPORT "Install a cmockery header" OFF)
 option(UNIT_TESTING "Build with unit testing" OFF)
 
-if (UNIT_TESTING)
+if (UNIT_TESTING AND NOT CYGWIN)
     set(WITH_STATIC_LIB ON)
 endif()
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/cmocka-1.1.1/cmake/Modules/DefineCompilerFlags.cmake src/cmocka-1.1.1/cmake/Modules/DefineCompilerFlags.cmake
--- origsrc/cmocka-1.1.1/cmake/Modules/DefineCompilerFlags.cmake	2016-11-12 20:14:13.000000000 +0900
+++ src/cmocka-1.1.1/cmake/Modules/DefineCompilerFlags.cmake	2017-04-30 13:50:09.154589800 +0900
@@ -16,10 +16,12 @@ if (UNIX AND NOT WIN32)
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wmissing-format-attribute -Wundef -Wstrict-prototypes")
 
         # with -fPIC
+        if (NOT CYGWIN)
         check_c_compiler_flag("-fPIC" WITH_FPIC)
         if (WITH_FPIC)
             set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
         endif (WITH_FPIC)
+        endif()
 
         check_c_compiler_flag_ssp("-fstack-protector" WITH_STACK_PROTECTOR)
         if (WITH_STACK_PROTECTOR)
diff -buprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/cmocka-1.1.1/tests/CMakeLists.txt src/cmocka-1.1.1/tests/CMakeLists.txt
--- origsrc/cmocka-1.1.1/tests/CMakeLists.txt	2017-04-05 15:40:50.000000000 +0900
+++ src/cmocka-1.1.1/tests/CMakeLists.txt	2017-04-30 13:50:09.158949600 +0900
@@ -25,14 +25,14 @@ set(CMOCKA_TESTS
     test_returns_fail)
 
 foreach(_CMOCKA_TEST ${CMOCKA_TESTS})
-    add_cmocka_test(${_CMOCKA_TEST} ${_CMOCKA_TEST}.c ${CMOCKA_STATIC_LIBRARY})
+    add_cmocka_test(${_CMOCKA_TEST} ${_CMOCKA_TEST}.c ${CMOCKA_SHARED_LIBRARY})
 endforeach()
 
 ### Special Cases
 if (${CMAKE_C_COMPILER_ID} MATCHES "(GNU|Clang)")
     set_source_files_properties(test_cmockery.c PROPERTIES COMPILE_FLAGS "-Wno-deprecated-declarations")
 endif()
-add_cmocka_test(test_cmockery test_cmockery.c ${CMOCKA_STATIC_LIBRARY})
+add_cmocka_test(test_cmockery test_cmockery.c ${CMOCKA_SHARED_LIBRARY})
 
 ### Exceptions
 
