--- origsrc/cmocka-1.0.0/cmake/Modules/DefineCompilerFlags.cmake	2015-02-11 15:57:11.000000000 +0900
+++ src/cmocka-1.0.0/cmake/Modules/DefineCompilerFlags.cmake	2015-02-27 23:03:34.070812500 +0900
@@ -16,11 +16,13 @@ if (UNIX AND NOT WIN32)
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wmissing-format-attribute -Wundef")
 
         # with -fPIC
-        check_c_compiler_flag("-fPIC" WITH_FPIC)
-        if (WITH_FPIC)
-            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
-        endif (WITH_FPIC)
-
+        if (NOT CYGWIN)
+            check_c_compiler_flag("-fPIC" WITH_FPIC)
+            if (WITH_FPIC)
+                set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
+            endif (WITH_FPIC)
+        endif()
+        
         check_c_compiler_flag_ssp("-fstack-protector" WITH_STACK_PROTECTOR)
         if (WITH_STACK_PROTECTOR)
             set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fstack-protector")
