--- origsrc/cmocka-1.1.5/DefineOptions.cmake	2019-02-20 00:13:00.000000000 +0900
+++ src/cmocka-1.1.5/DefineOptions.cmake	2019-04-05 13:36:11.060074700 +0900
@@ -4,10 +4,12 @@ option(WITH_EXAMPLES "Build examples" ON
 option(UNIT_TESTING "Build with unit testing" OFF)
 option(PICKY_DEVELOPER "Build with picky developer flags" OFF)
 
-if (WITH_STATIC_LIB)
+if (NOT CYGWIN)
+  if (WITH_STATIC_LIB)
     set(BUILD_STATIC_LIB ON)
-endif (WITH_STATIC_LIB)
+  endif (WITH_STATIC_LIB)
 
-if (UNIT_TESTING)
-  set(BUILD_STATIC_LIB ON)
-endif (UNIT_TESTING)
+  if (UNIT_TESTING)
+    set(BUILD_STATIC_LIB ON)
+  endif (UNIT_TESTING)
+endif()
--- origsrc/cmocka-1.1.5/src/CMakeLists.txt	2019-02-20 00:11:47.000000000 +0900
+++ src/cmocka-1.1.5/src/CMakeLists.txt	2019-04-05 13:36:11.060074700 +0900
@@ -74,6 +74,8 @@ endif (NOT WIN32)
 
 install(TARGETS
             ${CMOCKA_SHARED_LIBRARY}
+        RUNTIME DESTINATION
+            ${CMAKE_INSTALL_BINDIR}
         ARCHIVE DESTINATION
             ${CMAKE_INSTALL_LIBDIR}
         LIBRARY DESTINATION
--- origsrc/cmocka-1.1.5/tests/CMakeLists.txt	2019-03-28 21:33:50.000000000 +0900
+++ src/cmocka-1.1.5/tests/CMakeLists.txt	2019-04-05 13:36:11.075691900 +0900
@@ -38,7 +38,7 @@ foreach(_CMOCKA_TEST ${CMOCKA_TESTS})
     add_cmocka_test(${_CMOCKA_TEST}
                     SOURCES ${_CMOCKA_TEST}.c
                     COMPILE_OPTIONS ${DEFAULT_C_COMPILE_FLAGS}
-                    LINK_LIBRARIES ${CMOCKA_STATIC_LIBRARY}
+                    LINK_LIBRARIES ${CMOCKA_SHARED_LIBRARY}
                     LINK_OPTIONS ${DEFAULT_LINK_FLAGS})
     target_include_directories(${_CMOCKA_TEST} PRIVATE ${cmocka_BINARY_DIR})
 endforeach()
@@ -50,7 +50,7 @@ endif()
 add_cmocka_test(test_cmockery
                 SOURCES test_cmockery.c
                 COMPILE_OPTIONS ${DEFAULT_C_COMPILE_FLAGS}
-                LINK_LIBRARIES ${CMOCKA_STATIC_LIBRARY}
+                LINK_LIBRARIES ${CMOCKA_SHARED_LIBRARY}
                 LINK_OPTIONS ${DEFAULT_LINK_FLAGS})
 
 ### Exceptions
