--- origsrc/logstalgia-1.1.5/configure.ac	2026-03-04 10:55:53.000000000 +0900
+++ src/logstalgia-1.1.5/configure.ac	2026-07-28 10:58:46.706453400 +0900
@@ -45,8 +45,9 @@ AC_CHECK_FUNCS([IMG_LoadPNG_RW], , AC_MS
 AC_CHECK_FUNCS([IMG_LoadJPG_RW], , AC_MSG_ERROR([SDL2_image with JPEG support required. Please see INSTALL]))
 
 #BOOST
-AX_BOOST_BASE([1.69], , AC_MSG_ERROR(Boost Filesystem >= 1.69 is required. Please see INSTALL))
+AX_BOOST_BASE([1.66], , AC_MSG_ERROR(Boost Filesystem >= 1.66 is required. Please see INSTALL))
 AX_BOOST_FILESYSTEM
+AX_BOOST_SYSTEM
 
 CPPFLAGS="$CPPFLAGS $BOOST_CPPFLAGS"
 LIBS="$LIBS $BOOST_SYSTEM_LIB $BOOST_FILESYSTEM_LIB"
