--- origsrc/libbox0/libbox0/common.h	2019-06-12 08:10:00.000000000 +0900
+++ src/libbox0/libbox0/common.h	2019-06-12 08:12:49.923329600 +0900
@@ -65,6 +65,7 @@ __BEGIN_DECLS
 #define B0_PACKED __attribute__((packed))
 
 /* thanks Bionic */
+#if !defined(__restrict)
 #if defined(__STDC__VERSION__) && __STDC_VERSION__ >= 199901L
 # define __restrict	restrict
 #else
@@ -73,6 +74,7 @@ __BEGIN_DECLS
 #  define __restrict /* delete __restrict when not supported */
 # endif
 #endif
+#endif
 
 /* Just for readability */
 #define B0_IS_NULL(ptr) (ptr == NULL)
