diff --strip-trailing-cr -uprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.log -x config.status -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/countrycodes-1.0.5/src/Makefile src/countrycodes-1.0.5/src/Makefile
--- origsrc/countrycodes-1.0.5/src/Makefile	2003-02-02 00:56:08.000000000 +0900
+++ src/countrycodes-1.0.5/src/Makefile	2013-03-24 12:02:34.042286400 +0900
@@ -70,10 +70,10 @@ strip:
 	strip $(PROGRAM)
 
 install:
-	$(INSTALL) -g root -m $(BINMODE) -o root -s ${srcdir}/$(PROGRAM) ${bindir}
+	@test `uname -o` = Cygwin || $(INSTALL) -g root -m $(BINMODE) -o root -s ${srcdir}/$(PROGRAM) ${bindir}
 	@echo "Installing man page..."
 	@$(SEDBIN) $(SEDCMDS) ${docdir}/iso3166.1.in > ${mandir}/iso3166.1
-	@chown 0.0 ${mandir}/iso3166.1
+	@test `uname -o` = Cygwin || chown 0.0 ${mandir}/iso3166.1
 	@echo
 
 uninstall:
