--- origsrc/libisds-0.10/src/Makefile.am	2015-01-10 22:51:07.000000000 +0900
+++ src/libisds-0.10/src/Makefile.am	2015-02-11 17:11:51.515062500 +0900
@@ -48,5 +48,5 @@ libisds_la_HEADERS = isds.h
 libisds_la_CPPFLAGS = @XML_CPPFLAGS@ @LIBCURL_CPPFLAGS@ \
 					  -DLOCALEDIR=\"$(localedir)\"
 libisds_la_CFLAGS = @LIBGCRYPT_CFLAGS@ @GPGME_CFLAGS@
-libisds_la_LDFLAGS = -version-info 9:0:4 @XML_LIBS@ @LIBCURL@ @LIBGCRYPT_LIBS@ \
+libisds_la_LDFLAGS = -no-undefined -version-info 9:0:4 @XML_LIBS@ @LIBCURL@ @LIBGCRYPT_LIBS@ \
 	@GPGME_LIBS@ @EXPAT_LIBS@ @LTLIBINTL@
--- origsrc/libisds-0.10/src/isds_priv.h	2014-11-03 03:36:06.000000000 +0900
+++ src/libisds-0.10/src/isds_priv.h	2015-02-11 17:14:33.843187500 +0900
@@ -10,6 +10,10 @@
 #define _XOPEN_SOURCE 700
 #endif
 
+#if defined(__CYGWIN__)
+#include <sys/types.h>
+#endif
+
 #ifndef _POSIX_SOURCE
 /* defined: strtok_r */
 #define _POSIX_SOURCE   
--- origsrc/libisds-0.10/test/simline/server_cli.c	2015-01-03 21:43:01.000000000 +0900
+++ src/libisds-0.10/test/simline/server_cli.c	2015-02-11 17:21:20.871507800 +0900
@@ -9,6 +9,7 @@
 #include <sys/select.h>
 #include <sys/types.h> /* For pid_t */
 #include <string.h> /* memset() */
+#include <getopt.h>
 
 static const char *username = NULL;
 static const char *password = NULL;
