--- origsrc/libisds-0.8/src/Makefile.am	2013-10-14 03:52:52.000000000 +0900
+++ src/libisds-0.8/src/Makefile.am	2013-12-10 20:43:16.854419800 +0900
@@ -21,5 +21,5 @@ libisds_la_HEADERS = isds.h
 libisds_la_CPPFLAGS = @XML_CPPFLAGS@ @LIBCURL_CPPFLAGS@ \
 					  -DLOCALEDIR=\"$(localedir)\"
 libisds_la_CFLAGS = @LIBGCRYPT_CFLAGS@ @GPGME_CFLAGS@
-libisds_la_LDFLAGS = -version-info 7:0:2 @XML_LIBS@ @LIBCURL@ @LIBGCRYPT_LIBS@ \
+libisds_la_LDFLAGS = -no-undefined -version-info 7:0:2 @XML_LIBS@ @LIBCURL@ @LIBGCRYPT_LIBS@ \
 	@GPGME_LIBS@ @EXPAT_LIBS@ @LTLIBINTL@
--- origsrc/libisds-0.8/src/isds_priv.h	2012-10-22 03:58:20.000000000 +0900
+++ src/libisds-0.8/src/isds_priv.h	2013-12-10 21:16:56.777945800 +0900
@@ -10,6 +10,10 @@
 #define _XOPEN_SOURCE 700
 #endif
 
+#if defined(__CYGWIN__)
+#include <sys/types.h>
+#endif
+
 #ifndef _POSIX_SOURCE
 /* defined: strtok_r */
 #define _POSIX_SOURCE   
--- origsrc/libisds-0.8/test/simline/server_cli.c	2013-01-01 06:00:27.000000000 +0900
+++ src/libisds-0.8/test/simline/server_cli.c	2013-12-10 21:28:04.628380700 +0900
@@ -9,6 +9,7 @@
 #include <sys/select.h>
 #include <sys/types.h> /* For pid_t */
 #include <string.h> /* memset() */
+#include <getopt.h>
 
 static const char *username = NULL;
 static const char *password = NULL;
