--- origsrc/libisds-0.10.7/src/Makefile.am	2017-07-09 20:18:35.000000000 +0900
+++ src/libisds-0.10.7/src/Makefile.am	2017-08-28 18:15:23.356932800 +0900
@@ -48,5 +48,5 @@ libisds_la_HEADERS = isds.h
 libisds_la_CPPFLAGS = @XML_CPPFLAGS@ @LIBCURL_CPPFLAGS@ \
 					  -DLOCALEDIR=\"$(localedir)\"
 libisds_la_CFLAGS = @LIBGCRYPT_CFLAGS@ @GPGME_CFLAGS@
-libisds_la_LDFLAGS = -version-info 10:0:5 @XML_LIBS@ @LIBCURL@ @LIBGCRYPT_LIBS@ \
+libisds_la_LDFLAGS = -no-undefined -version-info 10:0:5 @XML_LIBS@ @LIBCURL@ @LIBGCRYPT_LIBS@ \
 	@GPGME_LIBS@ @EXPAT_LIBS@ @LTLIBINTL@
--- origsrc/libisds-0.10.7/test/simline/server_cli.c	2015-01-03 21:43:01.000000000 +0900
+++ src/libisds-0.10.7/test/simline/server_cli.c	2017-08-28 18:29:00.215000000 +0900
@@ -9,6 +9,7 @@
 #include <sys/select.h>
 #include <sys/types.h> /* For pid_t */
 #include <string.h> /* memset() */
+#include <getopt.h>
 
 static const char *username = NULL;
 static const char *password = NULL;
