--- origsrc/libisds-0.11/src/Makefile.am	2019-09-06 01:58:48.000000000 +0900
+++ src/libisds-0.11/src/Makefile.am	2019-10-03 09:09:18.478737700 +0900
@@ -48,5 +48,5 @@ libisds_la_HEADERS = isds.h
 libisds_la_CPPFLAGS = @XML_CPPFLAGS@ @LIBCURL_CPPFLAGS@ \
 					  -DLOCALEDIR=\"$(localedir)\"
 libisds_la_CFLAGS = @LIBGCRYPT_CFLAGS@ @GPGME_CFLAGS@
-libisds_la_LDFLAGS = -version-info 9:0:4 @XML_LIBS@ @LIBCURL@ @LIBGCRYPT_LIBS@ \
+libisds_la_LDFLAGS = -no-undefined -version-info 9:0:4 @XML_LIBS@ @LIBCURL@ @LIBGCRYPT_LIBS@ \
 	@GPGME_LIBS@ @EXPAT_LIBS@ @LTLIBINTL@
--- origsrc/libisds-0.11/test/simline/server_cli.c	2015-01-03 21:43:01.000000000 +0900
+++ src/libisds-0.11/test/simline/server_cli.c	2019-10-03 09:11:05.537785900 +0900
@@ -9,6 +9,7 @@
 #include <sys/select.h>
 #include <sys/types.h> /* For pid_t */
 #include <string.h> /* memset() */
+#include <getopt.h>
 
 static const char *username = NULL;
 static const char *password = NULL;
