--- origsrc/libmemcache-1.4.0.rc2/Makefile.am	2006-02-21 16:58:59.000000000 +0900
+++ src/libmemcache-1.4.0.rc2/Makefile.am	2023-10-03 19:01:30.875072800 +0900
@@ -19,3 +19,6 @@ test: check
 
 dist-hook:
 	rm -f $(distdir)/regen
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/libmemcache-1.4.0.rc2/configure.ac	2023-10-03 17:29:41.835687900 +0900
+++ src/libmemcache-1.4.0.rc2/configure.ac	2023-10-03 19:01:30.875072800 +0900
@@ -38,7 +38,7 @@ AC_SUBST(MEMCACHE_VERSION_NUMBER)
 
 AM_INIT_AUTOMAKE(libmemcache,${MEMCACHE_VERSION})
 
-dnl AM_PATH_CHECK()
+PKG_CHECK_MODULES([CHECK], [check >= 0.9.4])
 
 dnl Checks for programs.
 AC_C_CONST
@@ -378,4 +378,5 @@ AC_CONFIG_FILES([Makefile
 	test/regress/Makefile
 	test/redundant_server/Makefile
 	test/unit/Makefile])
+AC_CONFIG_FILES([libmemcache.pc])
 AC_OUTPUT
--- origsrc/libmemcache-1.4.0.rc2/include/memcache.h	2006-02-21 17:04:37.000000000 +0900
+++ src/libmemcache-1.4.0.rc2/include/memcache.h	2023-10-03 19:09:35.051895900 +0900
@@ -88,17 +88,17 @@ typedef uint32_t u_int32_t;
 
 /* A handful of #define's for callbacks that discourage developers
  * from improperly using the interface. */
-#define MCM_CALLBACK_CTXT	md597f78ab4a96c3e71c3bb9fe036f37367
-#define MCM_CALLBACK_KEY	md5783365dee46035fa36d8d9e6fe15da0f
-#define MCM_CALLBACK_LEN	md54c1f40392e708222b8a25a4337cd05fd
-#define MCM_CALLBACK_MC		md5646277084faa5a0f511793a728521ef5
-#define MCM_CALLBACK_PTR	md57f5686851aeded9e00761d4aaaac20ee
-#define MCM_CALLBACK_RES	md52e8a13a74753effc0f149a3ce900dc28
-#define MCM_ERR_MASK		md5fc9f4e2249ad88de0081136fef04defa
-#define MCM_ERR_FUNC_ERR_CTXT	md5096953205b8982964e25927bd8154148
-#define MCM_ERR_FUNC_MC_CTXT	md521e67c8edeaac774b00d960e230c8686
-#define MCM_KEY_VALID_KEY	md58926735b19e189407440aa1ba3ab1962
-#define MCM_KEY_VALID_LEN	md5d6077405f1a7b35ddeac18ccc2a8f4c7
+#define MCM_CALLBACK_CTXT	md5md517e9be250407e8a37fb60180a76510d7
+#define MCM_CALLBACK_KEY	md5md5c029c91da86138a5ad1489427e1d71b0
+#define MCM_CALLBACK_LEN	md5md5a07b51f993c7254fcb18197565700326
+#define MCM_CALLBACK_MC		md5md5897e5e464792802e78a18e0fdd437b7c
+#define MCM_CALLBACK_PTR	md5md57b4ac6d3ad66bcb14ca7584f9a3eff43
+#define MCM_CALLBACK_RES	md5md5e252857f6eaeebe7e61573a2a5bfa564
+#define MCM_ERR_MASK		md5md51ba931bcdf1b3f122e9b86617155853e
+#define MCM_ERR_FUNC_ERR_CTXT	md5md5314e69877a17018a14789a1a7fb8025c
+#define MCM_ERR_FUNC_MC_CTXT	md5md57c7c80036a426539d819e35ebab5192f
+#define MCM_KEY_VALID_KEY	md5md5537bb21da67eb92e5898f9911c607217
+#define MCM_KEY_VALID_LEN	md5md55ecd4a9c706b439f8eebaca09eeadacf
 
 /* Our initial read(2) buffer has to be long enough to read the
  * first line of the response.  ie:
@@ -923,7 +923,7 @@ int	mcMemGet(mcFreeFunc *freeFunc, mcMal
 		 mcMallocFunc *mallocAtomicFunc, mcReallocFunc *reallocFunc);
 
 /* Returns a pointer to the global context. */
-inline struct memcache_ctxt *mc_global_ctxt(void);
+struct memcache_ctxt *mc_global_ctxt(void);
 
 /* The next two functions are used to setup an error handler. */
 int	mcErrSetup(mcErrFunc errFunc);
--- origsrc/libmemcache-1.4.0.rc2/libmemcache.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libmemcache-1.4.0.rc2/libmemcache.pc.in	2023-10-03 19:01:30.827847400 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libmemcache
+Description: The C API for memcached(8)
+Version: @VERSION@
+Libs: -L${libdir} -lmemcache
+Cflags: -I${includedir}
--- origsrc/libmemcache-1.4.0.rc2/src/Makefile.am	2006-02-21 16:58:59.000000000 +0900
+++ src/libmemcache-1.4.0.rc2/src/Makefile.am	2023-10-03 17:31:41.161609200 +0900
@@ -1,5 +1,5 @@
 INCLUDES = -I${top_srcdir}/include
 lib_LTLIBRARIES = libmemcache.la
 libmemcache_la_SOURCES = memcache.c buffer.c
-libmemcache_la_LDFLAGS = -version-info @MEMCACHE_VERSION_INFO@
+libmemcache_la_LDFLAGS = -no-undefined -version-info @MEMCACHE_VERSION_INFO@
 noinst_HEADERS = crc32_table.h
--- origsrc/libmemcache-1.4.0.rc2/test/benchmark/Makefile.am	2006-02-21 16:58:55.000000000 +0900
+++ src/libmemcache-1.4.0.rc2/test/benchmark/Makefile.am	2023-10-03 17:32:57.653965600 +0900
@@ -1,5 +1,4 @@
 check_PROGRAMS = benchmark
 benchmark_SOURCES = benchmark.c
-benchmark_LDADD = -lmemcache
-benchmark_LDFLAGS = -L${top_srcdir}/src
+benchmark_LDADD = $(top_builldir)/src/libmemcache.la
 INCLUDES = -I${top_srcdir}/include
--- origsrc/libmemcache-1.4.0.rc2/test/buffer_recycle/Makefile.am	2006-02-21 16:58:54.000000000 +0900
+++ src/libmemcache-1.4.0.rc2/test/buffer_recycle/Makefile.am	2023-10-03 17:33:25.300731800 +0900
@@ -1,5 +1,4 @@
 check_PROGRAMS = buffer_recycle
 buffer_recycle_SOURCES = buffer_recycle.c
-buffer_recycle_LDADD = -lmemcache
-buffer_recycle_LDFLAGS = -L${top_srcdir}/src
+buffer_recycle_LDADD = $(top_builldir)/src/libmemcache.la
 INCLUDES = -I${top_srcdir}/include
--- origsrc/libmemcache-1.4.0.rc2/test/long_val/Makefile.am	2006-02-21 16:58:55.000000000 +0900
+++ src/libmemcache-1.4.0.rc2/test/long_val/Makefile.am	2023-10-03 17:33:47.086594800 +0900
@@ -1,5 +1,4 @@
 check_PROGRAMS = long_val
 long_val_SOURCES = long_val.c
-long_val_LDADD = -lmemcache
-long_val_LDFLAGS = -L${top_srcdir}/src
+long_val_LDADD = $(top_builldir)/src/libmemcache.la
 INCLUDES = -I${top_srcdir}/include
--- origsrc/libmemcache-1.4.0.rc2/test/redundant_server/Makefile.am	2006-02-21 16:58:56.000000000 +0900
+++ src/libmemcache-1.4.0.rc2/test/redundant_server/Makefile.am	2023-10-03 17:34:09.359398000 +0900
@@ -1,5 +1,4 @@
 check_PROGRAMS = redundant_server
 redundant_server_SOURCES = redundant_server.c
-redundant_server_LDADD = -lmemcache
-redundant_server_LDFLAGS = -L${top_srcdir}/src
+redundant_server_LDADD = $(top_builldir)/src/libmemcache.la
 INCLUDES = -I${top_srcdir}/include
--- origsrc/libmemcache-1.4.0.rc2/test/regress/Makefile.am	2006-02-21 16:58:56.000000000 +0900
+++ src/libmemcache-1.4.0.rc2/test/regress/Makefile.am	2023-10-03 17:34:32.442022400 +0900
@@ -1,5 +1,4 @@
 check_PROGRAMS = regress
 regress_SOURCES = regress.c
-regress_LDADD = -lmemcache
-regress_LDFLAGS = -L${top_srcdir}/src
+regress_LDADD = $(top_builldir)/src/libmemcache.la
 INCLUDES = -I${top_srcdir}/include
--- origsrc/libmemcache-1.4.0.rc2/test/unit/Makefile.am	2006-02-21 16:58:55.000000000 +0900
+++ src/libmemcache-1.4.0.rc2/test/unit/Makefile.am	2023-10-03 17:36:15.178559000 +0900
@@ -1,8 +1,6 @@
 check_PROGRAMS=check_buffer
 check_buffer_SOURCES=  check_buffer.c
-check_buffer_INCLUDES= @CHECK_CFLAGS@
-check_buffer_LIBS= @CHECK_LIBS@
-check_buffer_LDADD = -lmemcache
-check_buffer_LDFLAGS = -L${top_srcdir}/src
-INCLUDES = -I${top_srcdir}/include @CHECK_CFLAGS@
-LIBS = @CHECK_LIBS@
+check_buffer_INCLUDES= $(CHECK_CFLAGS)
+check_buffer_LDADD = $(top_builldir)/src/libmemcache.la
+INCLUDES = -I$(top_srcdir)/include $(CHECK_CFLAGS)
+LIBS = $(CHECK_LIBS)
