--- origsrc/ibus-libzhuyin-1.9.1/data/model/Makefile.am	2019-05-07 15:57:50.000000000 +0900
+++ src/ibus-libzhuyin-1.9.1/data/model/Makefile.am	2020-09-09 01:12:53.373140600 +0900
@@ -52,8 +52,8 @@ $(tablefiles) table.conf: interpolation2
 
 bigram.db: $(textual_model_data)
 	$(RM) $(binary_model_data)
-	gen_binary_files --table-dir $(top_srcdir)/data/model
-	import_interpolation --table-dir $(top_srcdir)/data/model < $(top_srcdir)/data/model/interpolation2.text
-	gen_unigram --table-dir $(top_srcdir)/data/model
+	pinyin-gen_binary_files --table-dir $(top_srcdir)/data/model
+	pinyin-import_interpolation --table-dir $(top_srcdir)/data/model < $(top_srcdir)/data/model/interpolation2.text
+	pinyin-gen_unigram --table-dir $(top_srcdir)/data/model
 
 addon_phrase_index.bin addon_pinyin_index.bin phrase_index.bin pinyin_index.bin $(binfiles): bigram.db
--- origsrc/ibus-libzhuyin-1.9.1/src/ZYTradSimpConverter.cc	2019-05-07 15:57:50.000000000 +0900
+++ src/ibus-libzhuyin-1.9.1/src/ZYTradSimpConverter.cc	2020-09-09 01:09:30.537667400 +0900
@@ -26,6 +26,7 @@
 #include "ZYTradSimpConverter.h"
 
 #ifdef HAVE_OPENCC
+#  include <string>
 #  include <opencc.h>
 #else
 #  include <cstring>
