--- origsrc/ibus-libzhuyin-1.10.4/configure.ac	2025-09-23 12:24:15.000000000 +0900
+++ src/ibus-libzhuyin-1.10.4/configure.ac	2026-01-12 15:43:12.524866300 +0900
@@ -118,7 +118,7 @@ GETTEXT_PACKAGE=ibus-libzhuyin
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE", [Gettext package])
 
-AM_GNU_GETTEXT_VERSION([0.21])
+AM_GNU_GETTEXT_VERSION([0.22.5])
 AM_GNU_GETTEXT([external])
 
 # Checks for GSettings.
