--- origsrc/cyassl-1.6.0/Makefile.am	2010-07-03 02:58:03.000000000 +0900
+++ src/cyassl-1.6.0/Makefile.am	2012-11-10 22:45:59.149132800 +0900
@@ -4,6 +4,37 @@ EXTRA_DIST = cyassl.dsp cyassl.dsw certs
     doc/*.pdf swig/README swig/*.i swig/cyassl_adds.c swig/*.sh swig/runme.* \
     swig/python_cyassl.vcproj swig/rsasign.py
 
+yassl_includedir = $(includedir)/yassl
+yassl_include_HEADERS = include/cyassl_error.h include/cyassl_int.h \
+    include/sniffer.h include/sniffer_error.h
+yassl_openssl_includedir = $(includedir)/yassl/openssl
+yassl_openssl_include_HEADERS = include/openssl/asn1.h include/openssl/bio.h \
+    include/openssl/bn.h include/openssl/conf.h include/openssl/crypto.h \
+    include/openssl/cyassl_callbacks.h include/openssl/des.h \
+    include/openssl/dh.h include/openssl/dsa.h include/openssl/ec.h \
+    include/openssl/ecdsa.h include/openssl/engine.h include/openssl/err.h \
+    include/openssl/evp.h include/openssl/hmac.h include/openssl/lhash.h \
+    include/openssl/md4.h include/openssl/md5.h include/openssl/ocsp.h \
+    include/openssl/opensslconf.h include/openssl/opensslv.h \
+    include/openssl/ossl_typ.h include/openssl/pem.h include/openssl/pkcs12.h \
+    include/openssl/rand.h include/openssl/rsa.h include/openssl/sha.h \
+    include/openssl/ssl.h include/openssl/stack.h include/openssl/ui.h \
+    include/openssl/x509.h include/openssl/x509v3.h
+yassl_ctaocrypt_includedir = $(includedir)/yassl/ctaocrypt
+yassl_ctaocrypt_include_HEADERS = ctaocrypt/include/aes.h \
+    ctaocrypt/include/arc4.h ctaocrypt/include/asn.h \
+    ctaocrypt/include/coding.h ctaocrypt/include/config.h \
+    ctaocrypt/include/des3.h ctaocrypt/include/dh.h ctaocrypt/include/dsa.h \
+    ctaocrypt/include/error.h ctaocrypt/include/hc128.h \
+    ctaocrypt/include/hmac.h ctaocrypt/include/integer.h \
+    ctaocrypt/include/md4.h ctaocrypt/include/md5.h ctaocrypt/include/misc.h \
+    ctaocrypt/include/mpi_class.h ctaocrypt/include/mpi_superclass.h \
+    ctaocrypt/include/pwdbased.h ctaocrypt/include/rabbit.h \
+    ctaocrypt/include/random.h ctaocrypt/include/ripemd.h \
+    ctaocrypt/include/rsa.h ctaocrypt/include/sha.h \
+    ctaocrypt/include/sha256.h ctaocrypt/include/sha512.h \
+    ctaocrypt/include/tfm.h ctaocrypt/include/types.h
+
 # !!!! first line of rule has to start with a hard (real) tab, not spaces
 basic:
 	cd src; $(MAKE); cd ../testsuite; $(MAKE); cd ../
@@ -15,14 +46,14 @@ openssl-links:
 
 #  !!! test -e with a .name like .libs then a * like *dylib fails so just
 #      look for the .dylib on OS X, and .so otherwise but copy all parts
-install:
-	$(mkinstalldirs) $(DESTDIR)$(includedir) $(DESTDIR)$(libdir); \
-    cp -fpR include/* $(DESTDIR)$(includedir); \
-    cp -fpR ctaocrypt/include/* $(DESTDIR)$(includedir); \
-    cp -fpR src/libcyassl.la $(DESTDIR)$(libdir); \
-    if test -e src/.libs/libcyassl.a; then \
-    cp -fp src/.libs/libcyassl.a $(DESTDIR)$(libdir); fi; \
-    if test -e src/.libs/libcyassl.so; then \
-    cp -fpR src/.libs/libcyassl.so* $(DESTDIR)$(libdir); fi; \
-    if test -e src/.libs/libcyassl.dylib; then \
-    cp -fpR src/.libs/libcyassl.*dylib $(DESTDIR)$(libdir); fi;
+#install:
+#	$(mkinstalldirs) $(DESTDIR)$(includedir) $(DESTDIR)$(libdir); \
+#    cp -fpR include/* $(DESTDIR)$(includedir); \
+#    cp -fpR ctaocrypt/include/* $(DESTDIR)$(includedir); \
+#    cp -fpR src/libcyassl.la $(DESTDIR)$(libdir); \
+#    if test -e src/.libs/libcyassl.a; then \
+#    cp -fp src/.libs/libcyassl.a $(DESTDIR)$(libdir); fi; \
+#    if test -e src/.libs/libcyassl.so; then \
+#    cp -fpR src/.libs/libcyassl.so* $(DESTDIR)$(libdir); fi; \
+#    if test -e src/.libs/libcyassl.dylib; then \
+#    cp -fpR src/.libs/libcyassl.*dylib $(DESTDIR)$(libdir); fi;
--- origsrc/cyassl-1.6.0/configure.in	2010-08-28 01:24:03.000000000 +0900
+++ src/cyassl-1.6.0/configure.in	2012-11-10 11:34:16.116343700 +0900
@@ -13,6 +13,7 @@ sinclude(acx_pthread.m4)
 CFLAGS="$CFLAGS $C_EXTRA_FLAGS"    
 
 AC_PROG_CC
+AM_PROG_CC_C_O
 AM_PROG_AS
 AC_PROG_INSTALL
 AC_LIBTOOL_WIN32_DLL
--- origsrc/cyassl-1.6.0/ctaocrypt/benchmark/Makefile.am	2009-01-09 05:41:02.000000000 +0900
+++ src/cyassl-1.6.0/ctaocrypt/benchmark/Makefile.am	2012-11-10 22:27:32.435265600 +0900
@@ -1,7 +1,8 @@
-INCLUDES = -I../include
-bin_PROGRAMS = benchmark
+INCLUDES = -I../include -I$(top_srcdir)/ctaocrypt/include
+check_PROGRAMS = benchmark
 benchmark_SOURCES = benchmark.c
 benchmark_LDFLAGS      = -L../src
 benchmark_LDADD        = ../../src/libcyassl.la -lm
 benchmark_DEPENDENCIES = ../../src/libcyassl.la
 EXTRA_DIST =  benchmark.dsp *.der benchmark.sln benchmark.vcproj
+TESTS = $(check_PROGRAMS)
--- origsrc/cyassl-1.6.0/ctaocrypt/test/Makefile.am	2010-07-29 04:01:06.000000000 +0900
+++ src/cyassl-1.6.0/ctaocrypt/test/Makefile.am	2012-11-10 22:30:43.119835900 +0900
@@ -1,5 +1,5 @@
-INCLUDES = -I../include -I../../include
-bin_PROGRAMS = test
+INCLUDES = -I../include -I../../include -I$(top_srcdir)/ctaocrypt/include
+check_PROGRAMS = test
 test_SOURCES = test.c
 test_LDFLAGS = -L../src
 test_LDADD        = ../../src/libcyassl.la -lm
--- origsrc/cyassl-1.6.0/examples/client/Makefile.am	2009-01-09 05:42:04.000000000 +0900
+++ src/cyassl-1.6.0/examples/client/Makefile.am	2012-11-10 14:27:56.028722000 +0900
@@ -1,5 +1,5 @@
-INCLUDES = -I../../include
-bin_PROGRAMS    = client
+INCLUDES = -I../../include -I$(top_srcdir)/ctaocrypt/include -I$(top_srcdir)/include
+check_PROGRAMS    = client
 client_SOURCES  = client.c
 client_CFLAGS = -Wall -Wno-unused
 client_LDFLAGS      = -L../../src
--- origsrc/cyassl-1.6.0/examples/echoclient/Makefile.am	2009-01-09 05:42:27.000000000 +0900
+++ src/cyassl-1.6.0/examples/echoclient/Makefile.am	2012-11-10 14:28:05.476206200 +0900
@@ -1,5 +1,5 @@
-INCLUDES = -I../../include
-bin_PROGRAMS    = echoclient
+INCLUDES = -I../../include -I$(top_srcdir)/ctaocrypt/include -I$(top_srcdir)/include
+check_PROGRAMS    = echoclient
 echoclient_SOURCES  = echoclient.c
 echoclient_CFLAGS = -Wall -Wno-unused
 echoclient_LDFLAGS     = -L../../src 
--- origsrc/cyassl-1.6.0/examples/echoserver/Makefile.am	2009-01-09 05:42:40.000000000 +0900
+++ src/cyassl-1.6.0/examples/echoserver/Makefile.am	2012-11-10 14:28:15.235999600 +0900
@@ -1,5 +1,5 @@
-INCLUDES = -I../../include
-bin_PROGRAMS    = echoserver
+INCLUDES = -I../../include -I$(top_srcdir)/ctaocrypt/include -I$(top_srcdir)/include
+check_PROGRAMS    = echoserver
 echoserver_SOURCES  = echoserver.c
 echoserver_CFLAGS = -Wall
 echoserver_LDFLAGS  = -L../../src
--- origsrc/cyassl-1.6.0/examples/server/Makefile.am	2009-01-09 05:42:17.000000000 +0900
+++ src/cyassl-1.6.0/examples/server/Makefile.am	2012-11-10 14:28:24.547940200 +0900
@@ -1,5 +1,5 @@
-INCLUDES = -I../../include
-bin_PROGRAMS    = server
+INCLUDES = -I../../include -I$(top_srcdir)/ctaocrypt/include -I$(top_srcdir)/include
+check_PROGRAMS    = server
 server_SOURCES  = server.c
 server_CFLAGS = -Wall
 server_LDFLAGS      = -L../../src
--- origsrc/cyassl-1.6.0/src/Makefile.am	2010-08-25 07:40:02.000000000 +0900
+++ src/cyassl-1.6.0/src/Makefile.am	2012-11-10 10:21:24.841498200 +0900
@@ -1,4 +1,4 @@
-INCLUDES = -I../include -I../ctaocrypt/include
+INCLUDES = -I../include -I../ctaocrypt/include -I$(top_srcdir)/include -I$(top_srcdir)/ctaocrypt/include
 
 lib_LTLIBRARIES = libcyassl.la
 libcyassl_la_SOURCES = ../ctaocrypt/src/integer.c \
--- origsrc/cyassl-1.6.0/sslSniffer/sslSnifferTest/Makefile.am	2010-05-12 09:00:56.000000000 +0900
+++ src/cyassl-1.6.0/sslSniffer/sslSnifferTest/Makefile.am	2012-11-10 22:20:33.754601500 +0900
@@ -1,8 +1,9 @@
 INCLUDES = -I../../include
-bin_PROGRAMS    = snifftest
+check_PROGRAMS    = snifftest
 snifftest_SOURCES  = snifftest.c
 snifftest_CFLAGS = -Wall -Wno-unused
 snifftest_LDFLAGS      = -L../../src
+TESTS = $(check_PROGRAMS)
 
 if BUILD_SNIFFER
 snifftest_LDADD        = ../../src/libcyassl.la -lm -lpcap
--- origsrc/cyassl-1.6.0/testsuite/Makefile.am	2009-01-09 05:41:23.000000000 +0900
+++ src/cyassl-1.6.0/testsuite/Makefile.am	2012-11-10 14:22:45.963409800 +0900
@@ -1,5 +1,5 @@
-INCLUDES = -I../include -I../ctaocrypt/include
-bin_PROGRAMS    = testsuite
+INCLUDES = -I../include -I../ctaocrypt/include -I$(top_srcdir)/ctaocrypt/include -I$(top_srcdir)/include
+check_PROGRAMS    = testsuite
 testsuite_SOURCES = testsuite.c ../ctaocrypt/test/test.c \
     ../examples/client/client.c ../examples/server/server.c \
     ../examples/echoclient/echoclient.c ../examples/echoserver/echoserver.c
@@ -8,3 +8,4 @@ testsuite_LDFLAGS      = -L../src
 testsuite_LDADD        = ../src/libcyassl.la -lm
 testsuite_DEPENDENCIES = ../src/libcyassl.la 
 EXTRA_DIST = testsuite.dsp input quit testsuite.sln testsuite.vcproj
+TESTS = $(check_PROGRAMS)
