--- origsrc/httrack-3.49.2/configure.ac	2017-05-20 17:45:07.000000000 +0900
+++ src/httrack-3.49.2/configure.ac	2021-11-16 08:17:54.296864800 +0900
@@ -212,9 +212,7 @@ AC_DEFINE(LIBSOCKET, 1,[Check for libsoc
 AC_SUBST(SOCKET_LIBS)
 
 ### Check for iconv
-AC_CHECK_LIB(iconv, iconv, [ICONV_LIBS="-liconv"
-AC_DEFINE(LIBICONV, 1,[Check for libiconv])], AC_MSG_RESULT([not necessary]))
-AC_SUBST(ICONV_LIBS)
+AM_ICONV
 
 ### Check for dlsym
 AC_CHECK_LIB(dl, dlopen, [DL_LIBS=-ldl
--- origsrc/httrack-3.49.2/html/Makefile.am	2016-04-27 02:10:23.000000000 +0900
+++ src/httrack-3.49.2/html/Makefile.am	2021-11-16 08:17:54.301842500 +0900
@@ -14,21 +14,21 @@ WebIcon48x48dir = $(datadir)/icons/hicol
 VFolderEntrydir = $(prefix)/share/applications
 
 HelpHtmlroot_DATA = ../httrack-doc.html ../history.txt
-HelpHtml_DATA = *.html
+HelpHtml_DATA = $(srcdir)/*.html
 HelpHtmldiv_DATA = div/search.sh
-HelpHtmlimg_DATA  = img/*
-HelpHtmlimages_DATA = images/*
+HelpHtmlimg_DATA  = $(srcdir)/img/*
+HelpHtmlimages_DATA = $(srcdir)/images/*
 HelpHtmlTxt_DATA = ../greetings.txt ../history.txt ../license.txt
-WebHtml_DATA = server/*.html server/*.js server/*.css
-WebHtmlimages_DATA = server/images/*
+WebHtml_DATA = $(srcdir)/server/*.html $(srcdir)/server/*.js $(srcdir)/server/*.css
+WebHtmlimages_DATA = $(srcdir)/server/images/*
 # note: converted & normalized by
 # ico2xpm favicon.ico -o httrack.xpm
 # mogrify -format xpm -map /usr/share/doc/menu/examples/cmap.xpm httrack.xpm
-WebPixmap_DATA = server/div/*.xpm
-WebIcon16x16_DATA = server/div/16x16/*.png
-WebIcon32x32_DATA = server/div/32x32/*.png
-WebIcon48x48_DATA = server/div/48x48/*.png
-VFolderEntry_DATA = server/div/*.desktop
+WebPixmap_DATA = $(srcdir)/server/div/*.xpm
+WebIcon16x16_DATA = $(srcdir)/server/div/16x16/*.png
+WebIcon32x32_DATA = $(srcdir)/server/div/32x32/*.png
+WebIcon48x48_DATA = $(srcdir)/server/div/48x48/*.png
+VFolderEntry_DATA = $(srcdir)/server/div/*.desktop
 
 EXTRA_DIST = $(HelpHtml_DATA) $(HelpHtmlimg_DATA) $(HelpHtmlimages_DATA) \
 	$(HelpHtmldiv_DATA) $(WebHtml_DATA) $(WebHtmlimages_DATA) \
--- origsrc/httrack-3.49.2/lang/Makefile.am	2016-04-27 02:10:23.000000000 +0900
+++ src/httrack-3.49.2/lang/Makefile.am	2021-11-16 08:17:54.319951500 +0900
@@ -1,6 +1,6 @@
 
 langdir = $(datadir)/httrack/lang
-lang_DATA = *.txt
+lang_DATA = $(srcdir)/*.txt
 langrootdir = $(datadir)/httrack
 langroot_DATA = ../lang.def ../lang.indexes
 
--- origsrc/httrack-3.49.2/libtest/Makefile.am	2017-05-20 17:45:09.000000000 +0900
+++ src/httrack-3.49.2/libtest/Makefile.am	2021-11-16 08:17:54.327953200 +0900
@@ -1,6 +1,6 @@
 
 exemplesdir = $(datadir)/httrack/libtest
-exemples_DATA = *.c *.h *.txt
+exemples_DATA = $(srcdir)/*.c $(srcdir)/*.h $(srcdir)/*.txt
 EXTRA_DIST = $(exemples_DATA) libtest.mak libtest.vcproj
 
 AM_CPPFLAGS = \
@@ -13,38 +13,38 @@ AM_CPPFLAGS = \
 	-DDATADIR=\""$(datadir)"\" \
 	-DLIBDIR=\""$(libdir)"\"
 
-AM_CPPFLAGS += -I../src
+AM_CPPFLAGS += -I../src -I$(top_srcdir)/src
 AM_LDFLAGS = -L../src
 
 # Examples
 libbaselinks_la_SOURCES =  callbacks-example-baselinks.c
 libbaselinks_la_LIBADD = $(THREADS_LIBS) $(SOCKET_LIBS) $(top_builddir)/src/libhttrack.la
-libbaselinks_la_LDFLAGS = -version-info 1:0:0
+libbaselinks_la_LDFLAGS = -module -no-undefined -version-info 1:0:0
 libchangecontent_la_SOURCES =  callbacks-example-changecontent.c
 libchangecontent_la_LIBADD = $(THREADS_LIBS) $(SOCKET_LIBS) $(top_builddir)/src/libhttrack.la
-libchangecontent_la_LDFLAGS = -version-info 1:0:0
+libchangecontent_la_LDFLAGS = -module -no-undefined -version-info 1:0:0
 libcontentfilter_la_SOURCES =  callbacks-example-contentfilter.c
 libcontentfilter_la_LIBADD = $(THREADS_LIBS) $(SOCKET_LIBS) $(top_builddir)/src/libhttrack.la
-libcontentfilter_la_LDFLAGS = -version-info 1:0:0
+libcontentfilter_la_LDFLAGS = -module -no-undefined -version-info 1:0:0
 libdisplayheader_la_SOURCES =  callbacks-example-displayheader.c
 libdisplayheader_la_LIBADD = $(THREADS_LIBS) $(SOCKET_LIBS) $(top_builddir)/src/libhttrack.la
-libdisplayheader_la_LDFLAGS = -version-info 1:0:0
+libdisplayheader_la_LDFLAGS = -module -no-undefined -version-info 1:0:0
 libfilename2_la_SOURCES =  callbacks-example-filename2.c
 libfilename2_la_LIBADD = $(THREADS_LIBS) $(SOCKET_LIBS) $(top_builddir)/src/libhttrack.la
-libfilename2_la_LDFLAGS = -version-info 1:0:0
+libfilename2_la_LDFLAGS = -module -no-undefined -version-info 1:0:0
 libfilename_la_SOURCES =  callbacks-example-filename.c
 libfilename_la_LIBADD = $(THREADS_LIBS) $(SOCKET_LIBS) $(top_builddir)/src/libhttrack.la
-libfilename_la_LDFLAGS = -version-info 1:0:0
+libfilename_la_LDFLAGS = -module -no-undefined -version-info 1:0:0
 libfilenameiisbug_la_SOURCES =  callbacks-example-filenameiisbug.c
 libfilenameiisbug_la_LIBADD = $(THREADS_LIBS) $(SOCKET_LIBS) $(top_builddir)/src/libhttrack.la
-libfilenameiisbug_la_LDFLAGS = -version-info 1:0:0
+libfilenameiisbug_la_LDFLAGS = -module -no-undefined -version-info 1:0:0
 liblistlinks_la_SOURCES =  callbacks-example-listlinks.c
 liblistlinks_la_LIBADD = $(THREADS_LIBS) $(SOCKET_LIBS) $(top_builddir)/src/libhttrack.la
-liblistlinks_la_LDFLAGS = -version-info 1:0:0
+liblistlinks_la_LDFLAGS = -module -no-undefined -version-info 1:0:0
 liblog_la_SOURCES =  callbacks-example-log.c
 liblog_la_LIBADD = $(THREADS_LIBS) $(SOCKET_LIBS) $(top_builddir)/src/libhttrack.la
-liblog_la_LDFLAGS = -version-info 1:0:0
+liblog_la_LDFLAGS = -module -no-undefined -version-info 1:0:0
 libsimple_la_SOURCES =  callbacks-example-simple.c
 libsimple_la_LIBADD = $(THREADS_LIBS) $(SOCKET_LIBS) $(top_builddir)/src/libhttrack.la
-libsimple_la_LDFLAGS = -version-info 1:0:0
+libsimple_la_LDFLAGS = -module -no-undefined -version-info 1:0:0
 pkglib_LTLIBRARIES = libbaselinks.la libchangecontent.la libcontentfilter.la libdisplayheader.la libfilename2.la libfilename.la libfilenameiisbug.la liblistlinks.la liblog.la libsimple.la
--- origsrc/httrack-3.49.2/src/Makefile.am	2016-04-27 02:10:23.000000000 +0900
+++ src/httrack-3.49.2/src/Makefile.am	2021-11-16 08:17:54.332959600 +0900
@@ -68,13 +68,13 @@ libhttrack_la_SOURCES =  htscore.c htspa
 	htsentities.h htsentities.sh htsbasiccharsets.sh htscodepages.h \
 	md5.h coucal/murmurhash3.h \
 	minizip/crypt.h minizip/ioapi.h minizip/mztools.h minizip/unzip.h minizip/zip.h
-libhttrack_la_LIBADD = $(THREADS_LIBS) $(ZLIB_LIBS) $(OPENSSL_LIBS) $(DL_LIBS) $(SOCKET_LIBS) $(ICONV_LIBS)
+libhttrack_la_LIBADD = $(THREADS_LIBS) $(ZLIB_LIBS) $(OPENSSL_LIBS) $(DL_LIBS) $(SOCKET_LIBS) $(LTLIBICONV)
 libhttrack_la_CFLAGS = $(AM_CFLAGS) -DLIBHTTRACK_EXPORTS -DZLIB_CONST
-libhttrack_la_LDFLAGS = -version-info $(VERSION_INFO)
+libhttrack_la_LDFLAGS = -no-undefined -version-info $(VERSION_INFO)
 
 libhtsjava_la_SOURCES = htsjava.c htsjava.h
 libhtsjava_la_LIBADD = $(THREADS_LIBS) $(DL_LIBS) libhttrack.la
-libhtsjava_la_LDFLAGS = -version-info $(VERSION_INFO)
+libhtsjava_la_LDFLAGS = -no-undefined -version-info $(VERSION_INFO)
 
 EXTRA_DIST = httrack.h webhttrack \
 		coucal/murmurhash3.h.diff \
--- origsrc/httrack-3.49.2/src/htsglobal.h	2017-05-20 17:33:03.000000000 +0900
+++ src/httrack-3.49.2/src/htsglobal.h	2021-11-16 08:17:54.336965000 +0900
@@ -167,7 +167,7 @@ Please visit our Website: http://www.htt
 #define HTS_USESWF 1
 #endif
 
-#ifdef _WIN32
+#ifdef __cdecl
 #else
 #define __cdecl
 #endif
@@ -326,7 +326,7 @@ typedef int T_SOC;
 /* IPV4, IPV6 and various unified structures */
 #define HTS_MAXADDRLEN 64
 
-#ifdef _WIN32
+#ifdef __cdecl
 #else
 #define __cdecl
 #endif
--- origsrc/httrack-3.49.2/src/htslib.c	2017-04-02 05:00:27.000000000 +0900
+++ src/httrack-3.49.2/src/htslib.c	2021-11-16 08:17:54.341916500 +0900
@@ -5388,6 +5388,10 @@ HTSEXT_API httrackp *hts_create_opt(void
   static const char *defaultModules[] = {
     "htsswf", "htsjava", "httrack-plugin", NULL
   };
+#elif defined(__CYGWIN__)
+  static const char *defaultModules[] = {
+    "cyghtsswf-1.dll", "cyghtsjava-2.dll", "httrack-plugin", NULL
+  };
 #else
   static const char *defaultModules[] = {
     "libhtsswf.so.1", "libhtsjava.so.2", "httrack-plugin", NULL
--- origsrc/httrack-3.49.2/src/htsmodules.c	2017-04-02 04:38:16.000000000 +0900
+++ src/httrack-3.49.2/src/htsmodules.c	2021-11-16 08:17:54.345929200 +0900
@@ -171,7 +171,11 @@ void *openFunctionLib(const char *file_)
 #else
   handle = dlopen(file, RTLD_LAZY);
   if (handle == NULL) {
+#if defined(__CYGWIN__)
+    sprintf(file, "cyg%s-1.dll", file_);
+#else
     sprintf(file, "lib%s.so", file_);
+#endif
     handle = dlopen(file, RTLD_LAZY);
   }
 #endif
