--- origsrc/httrack-3.49.6/html/Makefile.am	2022-11-21 03:28:12.000000000 +0900
+++ src/httrack-3.49.6/html/Makefile.am	2026-01-12 10:32:42.434074800 +0900
@@ -14,21 +14,21 @@ WebIcon48x48dir = $(datadir)/icons/hicol
 VFolderEntrydir = $(prefix)/share/applications
 
 HelpHtmlroot_DATA = ../httrack-doc.html ../history.txt
-HelpHtml_DATA = *.html
+HelpHtml_DATA = $(srcdir)/*.html
 HelpHtmldiv_DATA = div/search.sh
-HelpHtmlimg_DATA  = img/*
-HelpHtmlimages_DATA = images/*
+HelpHtmlimg_DATA  = $(srcdir)/img/*
+HelpHtmlimages_DATA = $(srcdir)/images/*
 HelpHtmlTxt_DATA = ../greetings.txt ../history.txt ../license.txt
-WebHtml_DATA = server/*.html server/*.js server/*.css
-WebHtmlimages_DATA = server/images/*
+WebHtml_DATA = $(srcdir)/server/*.html $(srcdir)/server/*.js $(srcdir)/server/*.css
+WebHtmlimages_DATA = $(srcdir)/server/images/*
 # note: converted & normalized by
 # ico2xpm favicon.ico -o httrack.xpm
 # mogrify -format xpm -map /usr/share/doc/menu/examples/cmap.xpm httrack.xpm
-WebPixmap_DATA = server/div/*.xpm
-WebIcon16x16_DATA = server/div/16x16/*.png
-WebIcon32x32_DATA = server/div/32x32/*.png
-WebIcon48x48_DATA = server/div/48x48/*.png
-VFolderEntry_DATA = server/div/*.desktop
+WebPixmap_DATA = $(srcdir)/server/div/*.xpm
+WebIcon16x16_DATA = $(srcdir)/server/div/16x16/*.png
+WebIcon32x32_DATA = $(srcdir)/server/div/32x32/*.png
+WebIcon48x48_DATA = $(srcdir)/server/div/48x48/*.png
+VFolderEntry_DATA = $(srcdir)/server/div/*.desktop
 
 EXTRA_DIST = $(HelpHtml_DATA) $(HelpHtmlimg_DATA) $(HelpHtmlimages_DATA) \
 	$(HelpHtmldiv_DATA) $(WebHtml_DATA) $(WebHtmlimages_DATA) \
--- origsrc/httrack-3.49.6/lang/Makefile.am	2022-11-21 03:28:12.000000000 +0900
+++ src/httrack-3.49.6/lang/Makefile.am	2026-01-12 10:31:25.228953800 +0900
@@ -1,6 +1,6 @@
 
 langdir = $(datadir)/httrack/lang
-lang_DATA = *.txt
+lang_DATA = $(srcdir)/*.txt
 langrootdir = $(datadir)/httrack
 langroot_DATA = ../lang.def ../lang.indexes
 
--- origsrc/httrack-3.49.6/libtest/Makefile.am	2023-01-14 23:29:24.000000000 +0900
+++ src/httrack-3.49.6/libtest/Makefile.am	2026-01-12 10:42:42.006635600 +0900
@@ -1,6 +1,6 @@
 
 exemplesdir = $(datadir)/httrack/libtest
-exemples_DATA = *.c *.h *.txt
+exemples_DATA = $(srcdir)/*.c $(srcdir)/*.h $(srcdir)/*.txt
 EXTRA_DIST = $(exemples_DATA) libtest.mak libtest.vcproj
 
 AM_CPPFLAGS = \
@@ -12,9 +12,9 @@ AM_CPPFLAGS = \
 	-DSYSCONFDIR=\""$(sysconfdir)"\" \
 	-DDATADIR=\""$(datadir)"\" \
 	-DLIBDIR=\""$(libdir)"\"
-AM_CPPFLAGS += -I../src
+AM_CPPFLAGS += -I../src -I$(top_srcdir)/src
 
-AM_LDFLAGS = \
+AM_LDFLAGS = -module -no-undefined \
 	@DEFAULT_LDFLAGS@ \
 	-L../src
 
--- origsrc/httrack-3.49.6/src/Makefile.am	2024-01-27 22:22:48.000000000 +0900
+++ src/httrack-3.49.6/src/Makefile.am	2026-01-12 09:23:07.853338100 +0900
@@ -79,11 +79,11 @@ libhttrack_la_SOURCES =  htscore.c htspa
 	minizip/crypt.h minizip/ioapi.h minizip/mztools.h minizip/unzip.h minizip/zip.h
 libhttrack_la_LIBADD = $(THREADS_LIBS) $(ZLIB_LIBS) $(OPENSSL_LIBS) $(DL_LIBS) $(SOCKET_LIBS) $(ICONV_LIBS)
 libhttrack_la_CFLAGS = $(AM_CFLAGS) -DLIBHTTRACK_EXPORTS -DZLIB_CONST
-libhttrack_la_LDFLAGS = $(AM_LDFLAGS) -version-info $(VERSION_INFO)
+libhttrack_la_LDFLAGS = $(AM_LDFLAGS) -no-undefined -version-info $(VERSION_INFO)
 
 libhtsjava_la_SOURCES = htsjava.c htsjava.h
 libhtsjava_la_LIBADD = $(THREADS_LIBS) $(DL_LIBS) libhttrack.la
-libhtsjava_la_LDFLAGS = $(AM_LDFLAGS) -version-info $(VERSION_INFO)
+libhtsjava_la_LDFLAGS = $(AM_LDFLAGS) -no-undefined -version-info $(VERSION_INFO)
 
 EXTRA_DIST = httrack.h webhttrack \
 		coucal/murmurhash3.h.diff \
--- origsrc/httrack-3.49.6/src/htsglobal.h	2025-03-12 03:40:02.000000000 +0900
+++ src/httrack-3.49.6/src/htsglobal.h	2026-01-12 09:28:04.924094000 +0900
@@ -165,8 +165,10 @@ Please visit our Website: http://www.htt
 
 #ifdef _WIN32
 #else
+#if !defined(__cdecl)
 #define __cdecl
 #endif
+#endif
 
 /* rc file */
 #ifdef _WIN32
@@ -324,8 +326,10 @@ typedef int T_SOC;
 
 #ifdef _WIN32
 #else
+#if !defined(__cdecl)
 #define __cdecl
 #endif
+#endif
 
 /* mode pour mkdir ET chmod (accès aux fichiers) */
 #define HTS_PROTECT_FOLDER (S_IRUSR|S_IWUSR|S_IXUSR)
--- origsrc/httrack-3.49.6/src/htslib.c	2023-01-15 01:21:58.000000000 +0900
+++ src/httrack-3.49.6/src/htslib.c	2026-01-12 10:35:03.558784700 +0900
@@ -5401,6 +5401,10 @@ HTSEXT_API httrackp *hts_create_opt(void
   static const char *defaultModules[] = {
     "htsswf", "htsjava", "httrack-plugin", NULL
   };
+#elif defined(__CYGWIN__)
+  static const char *defaultModules[] = {
+    "cyghtsswf-1.dll", "cyghtsjava-2.dll", "httrack-plugin.dll", NULL
+  };
 #else
   static const char *defaultModules[] = {
     "libhtsswf.so.1", "libhtsjava.so.2", "httrack-plugin", NULL
--- origsrc/httrack-3.49.6/src/htsmodules.c	2022-11-21 03:28:12.000000000 +0900
+++ src/httrack-3.49.6/src/htsmodules.c	2026-01-12 10:36:27.310223800 +0900
@@ -171,7 +171,11 @@ void *openFunctionLib(const char *file_)
 #else
   handle = dlopen(file, RTLD_LAZY);
   if (handle == NULL) {
+#if defined(__CYGWIN__)
+    sprintf(file, "cyg%s-1.dll", file_);
+#else
     sprintf(file, "lib%s.so", file_);
+#endif
     handle = dlopen(file, RTLD_LAZY);
   }
 #endif
--- origsrc/httrack-3.49.6/src/httrack-library.h	2022-11-21 03:28:12.000000000 +0900
+++ src/httrack-3.49.6/src/httrack-library.h	2026-01-12 09:31:05.903130500 +0900
@@ -288,6 +288,10 @@ HTSEXT_API int hts_findisdir(find_handle
 HTSEXT_API int hts_findisfile(find_handle find);
 HTSEXT_API int hts_findissystem(find_handle find);
 
+#if defined(FOPEN)
+#undef FOPEN
+#endif
+
 /* UTF-8 aware FILE API */
 #ifndef HTS_DEF_FILEAPI
 #ifdef _WIN32
