--- origsrc/OpenCC-ver.1.2.0/data/CMakeLists.txt	2026-01-22 18:05:03.000000000 +0900
+++ src/OpenCC-ver.1.2.0/data/CMakeLists.txt	2026-01-30 08:42:02.573029800 +0900
@@ -115,6 +115,7 @@ foreach(DICT ${DICTS})
     COMMENT
       "Building ${DICT}.ocd2"
     COMMAND
+      PATH=${PROJECT_BINARY_DIR}/src:${PROJECT_BINARY_DIR}/src/tools:$ENV{PATH}
       ${OPENCC_DICT_BIN}
         --input ${DICT_${DICT}_INPUT}
         --output ${DICT}.ocd2
