diff -buprN --no-dereference -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/OpenCC-ver.1.0.5/src/CMakeLists.txt src/OpenCC-ver.1.0.5/src/CMakeLists.txt
--- origsrc/OpenCC-ver.1.0.5/src/CMakeLists.txt	2017-02-07 07:18:02.000000000 +0900
+++ src/OpenCC-ver.1.0.5/src/CMakeLists.txt	2017-02-17 00:42:34.074798900 +0900
@@ -86,6 +86,7 @@ install(
   TARGETS libopencc
   LIBRARY DESTINATION ${DIR_LIBRARY}
   ARCHIVE DESTINATION ${DIR_LIBRARY}
+  RUNTIME DESTINATION bin
 )
 
 install(
