--- origsrc/OpenCC-ver.1.1.6/CMakeLists.txt	2022-12-08 16:44:48.000000000 +0900
+++ src/OpenCC-ver.1.1.6/CMakeLists.txt	2022-12-22 14:43:06.857339300 +0900
@@ -42,7 +42,7 @@ set (PACKAGE_URL https://github.com/BYVo
 set (PACKAGE_BUGREPORT https://github.com/BYVoid/Opencc/issues)
 set (OPENCC_VERSION_MAJOR 1)
 set (OPENCC_VERSION_MINOR 1)
-set (OPENCC_VERSION_REVISION 5)
+set (OPENCC_VERSION_REVISION 6)
 
 if (CMAKE_BUILD_TYPE MATCHES Debug)
   set (version_suffix .Debug)
--- origsrc/OpenCC-ver.1.1.6/data/CMakeLists.txt	2022-12-08 16:44:48.000000000 +0900
+++ src/OpenCC-ver.1.1.6/data/CMakeLists.txt	2022-12-22 14:42:13.946373400 +0900
@@ -136,6 +136,7 @@ foreach(DICT ${DICTS})
     COMMENT
       "Building ${DICT}.ocd2"
     COMMAND
+      PATH=${PROJECT_BINARY_DIR}/src:${PROJECT_BINARY_DIR}/src/tools:$ENV{PATH}
       ${OPENCC_DICT_BIN}
         --input ${DICT_${DICT}_INPUT}
         --output ${DICT}.ocd2
