--- origsrc/opencc-0.3.0/CMakeLists.txt	2011-12-02 01:44:09.000000000 +0900
+++ src/opencc-0.3.0/CMakeLists.txt	2012-01-15 21:55:33.062417700 +0900
@@ -62,7 +62,7 @@ endif (WIN32)
 ######## Directory
 
 set (DIR_PREFIX ${CMAKE_INSTALL_PREFIX})
-set (DIR_LIBRARY ${DIR_PREFIX}/${CMAKE_SHARED_LIBRARY_PREFIX})
+set (DIR_LIBRARY ${DIR_PREFIX}/${CMAKE_STATIC_LIBRARY_PREFIX})
 set (DIR_LIBRARY_STATIC ${DIR_PREFIX}/${CMAKE_STATIC_LIBRARY_PREFIX})
 set (DIR_INCLUDE ${DIR_PREFIX}/include)
 set (DIR_SHARE ${DIR_PREFIX}/share)
--- origsrc/opencc-0.3.0/src/CMakeLists.txt	2011-12-02 01:44:09.000000000 +0900
+++ src/opencc-0.3.0/src/CMakeLists.txt	2012-01-15 22:02:37.012691300 +0900
@@ -34,7 +34,6 @@ set(
 )
 
 set (LIBOPENCC_TARGET libopencc)
-set (LIBOPENCC_STATIC_TARGET libopencc_static)
 
 add_definitions(
 	-DPKGDATADIR="${DIR_SHARE_OPENCC}"
@@ -51,15 +50,8 @@ add_library(
 	${LIBOPENCC_SOURCES}
 )
 
-add_library(
-	${LIBOPENCC_STATIC_TARGET}
-	STATIC
-	${LIBOPENCC_SOURCES}
-)
-
 set_target_properties(
 	${LIBOPENCC_TARGET}
-	${LIBOPENCC_STATIC_TARGET}
 	PROPERTIES
 		OUTPUT_NAME
 			opencc
@@ -98,17 +90,14 @@ if (NOT WIN32)
 install(
 	TARGETS
 		${LIBOPENCC_TARGET}
+        RUNTIME DESTINATION
+		${DIR_BIN}
 	LIBRARY DESTINATION
 		${DIR_LIBRARY}
-)
-endif (NOT WIN32)
-
-install(
-	TARGETS
-		${LIBOPENCC_STATIC_TARGET}
 	ARCHIVE DESTINATION
-		${DIR_LIBRARY_STATIC}
+		${DIR_LIBRARY}
 )
+endif (NOT WIN32)
 
 install(
 	FILES
