--- origsrc/opencc-0.1.1/configure.ac	2010-08-10 22:20:01.000000000 +0900
+++ src/opencc-0.1.1/configure.ac	2010-09-04 13:09:25.644358300 +0900
@@ -44,13 +44,11 @@ AS_VERSION
 AS_NANO
 AM_SANITY_CHECK
 AM_MAINTAINER_MODE
-AM_DISABLE_STATIC
 AC_PROG_CC
 AM_PROG_CC_C_O
-AC_PROG_RANLIB
 AC_ISC_POSIX
 AC_HEADER_STDC
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 AC_C_BIGENDIAN
 if test "x$ac_cv_c_bigendian" = "xyes"; then
--- origsrc/opencc-0.1.1/src/Makefile.am	2010-08-10 21:48:16.000000000 +0900
+++ src/opencc-0.1.1/src/Makefile.am	2010-09-04 13:10:53.425608300 +0900
@@ -69,7 +69,10 @@ libopencc_la_CFLAGS = \
 	$(macro_cflags) \
 	$(NULL)
 
+libopencc_la_LIBADD = $(LTLIBINTL)
+
 libopencc_la_LDFLAGS = \
+	-no-undefined \
 	-export-symbols-regex "opencc_.*" \
 	-version-info @LT_VERSION_INFO@ \
 	$(NULL)
