--- origsrc/OpenCC-ver.1.1.9/data/CMakeLists.txt	2024-08-03 12:45:16.000000000 +0900
+++ src/OpenCC-ver.1.1.9/data/CMakeLists.txt	2024-09-05 19:53:37.282579600 +0900
@@ -136,6 +136,7 @@ foreach(DICT ${DICTS})
     COMMENT
       "Building ${DICT}.ocd2"
     COMMAND
+      PATH=${PROJECT_BINARY_DIR}/src:${PROJECT_BINARY_DIR}/src/tools:$ENV{PATH}
       ${OPENCC_DICT_BIN}
         --input ${DICT_${DICT}_INPUT}
         --output ${DICT}.ocd2
