--- origsrc/sfeed_curses-0.9.2/Makefile	2020-08-23 23:41:37.000000000 +0900
+++ src/sfeed_curses-0.9.2/Makefile	2020-09-28 17:44:10.870836400 +0900
@@ -14,7 +14,7 @@ SFEED_LDFLAGS = ${LDFLAGS} -lcurses
 SFEED_CPPFLAGS = -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_BSD_SOURCE
 
 # Linux: some distros use ncurses and require -lncurses.
-#SFEED_LDFLAGS = ${LDFLAGS} -lncurses
+SFEED_LDFLAGS = ${LDFLAGS} -lncursesw
 
 BIN = sfeed_curses
 SCRIPTS = sfeed_content sfeed_markread
@@ -37,7 +37,7 @@ OBJ = ${SRC:.c=.o}
 ${OBJ}:
 
 .o:
-	${CC} ${SFEED_LDFLAGS} -o $@ $<
+	${CC} -o $@ $< ${SFEED_LDFLAGS}
 
 .c.o:
 	${CC} ${SFEED_CFLAGS} ${SFEED_CPPFLAGS} -o $@ -c $<
