--- origsrc/po6-releases-0.8.0/Makefile.am	2015-07-10 06:01:54.000000000 +0900
+++ src/po6-releases-0.8.0/Makefile.am	2025-08-28 23:47:10.493605300 +0900
@@ -29,6 +29,7 @@ nobase_include_HEADERS += po6/threads/th
 lib_LTLIBRARIES = libpo6.la
 libpo6_la_SOURCES =
 libpo6_la_SOURCES += errno.cc
+libpo6_la_LDFLAGS = -no-undefined
 
 ##################################### Tests ####################################
 
