--- origsrc/vxlog-1.0.3/CMakeLists.txt	2010-10-18 22:23:04.000000000 +0900
+++ src/vxlog-1.0.3/CMakeLists.txt	2020-09-08 20:34:30.120977100 +0900
@@ -45,8 +45,11 @@ if(CMAKE_COMPILER_IS_GNUCXX)
   set(CMAKE_CXX_FLAGS_DEBUGFULL      "-g -fno-inline")
   set(CMAKE_CXX_FLAGS_PROFILE        "-g -fno-inline -ftest-coverage -fprofile-arcs")
 
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wnon-virtual-dtor -Wno-long-long -ansi -Wundef -Wcast-align -Wchar-subscripts -Wall -Wextra -Wpointer-arith -Wformat-security -fno-common -pedantic")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wnon-virtual-dtor -Wno-long-long -Wundef -Wcast-align -Wchar-subscripts -Wall -Wextra -Wpointer-arith -Wformat-security -fno-common -pedantic")
 
+  if(NOT CYGWIN)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ansi")
+  endif()
 endif(CMAKE_COMPILER_IS_GNUCXX)
 
 if(CMAKE_CXX_COMPILER MATCHES "clang")
@@ -89,3 +92,10 @@ if(TESTING)
   add_subdirectory(utests)
 endif(TESTING)
 
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/vxlog.pc.in ${CMAKE_BINARY_DIR}/vxlog.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/vxlog.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/vxlog-1.0.3/utests/logger-utest.cc	2010-10-18 22:23:04.000000000 +0900
+++ src/vxlog-1.0.3/utests/logger-utest.cc	2020-09-08 20:12:48.697523100 +0900
@@ -22,7 +22,6 @@
 
 #include "vxlog/logger.h"
 #include <vector>
-#include <vxlog/shared-ptr.h>
 #include "vxlog/macros.h"
 #include "utests/googletest.h"
 #include "vxlog/outputter-interface.h"
--- origsrc/vxlog-1.0.3/vxlog.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/vxlog-1.0.3/vxlog.pc.in	2020-09-08 20:34:30.073979900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: vxlog
+Description: C++ logging library
+Version: @VERSION@
+Libs: -L${libdir} -lvxlog
+Cflags: -I${includedir}
