--- origsrc/libsydneyaudio/Makefile.am	2015-05-19 07:30:49.000000000 +0900
+++ src/libsydneyaudio/Makefile.am	2015-05-19 07:30:51.675475400 +0900
@@ -33,6 +33,8 @@
 #
 # ***** END LICENSE BLOCK ***** *
 
+ACLOCAL_AMFLAGS = -I m4
+
 AUTOMAKE_OPTIONS = foreign
 
 SUBDIRS = src include
--- origsrc/libsydneyaudio/src/Makefile.am	2015-05-19 07:30:49.000000000 +0900
+++ src/libsydneyaudio/src/Makefile.am	2015-05-19 07:30:51.677409000 +0900
@@ -74,4 +74,4 @@ lib_LTLIBRARIES = libsydneyaudio.la
 libsydneyaudio_la_SOURCES = $(audio_sources)
 
 libsydneyaudio_la_LIBADD = $(audio_libs) $(PTHREAD_LIBS)
-
+libsydneyaudio_la_LDFLAGS = -no-undefined
