--- origsrc/aws-sdk-cpp-1.3.22/CMakeLists.txt	2017-12-13 06:33:53.000000000 +0900
+++ src/aws-sdk-cpp-1.3.22/CMakeLists.txt	2017-12-16 17:25:45.888000000 +0900
@@ -162,7 +162,7 @@ if (CMAKE_INSTALL_INCLUDEDIR)
     set(INCLUDE_DIRECTORY "${CMAKE_INSTALL_INCLUDEDIR}")
 endif()
 
-if(BUILD_SHARED_LIBS)
+if(BUILD_SHARED_LIBS AND NOT CYGWIN)
     set(ARCHIVE_DIRECTORY "${BINARY_DIRECTORY}")
 else()
     set(ARCHIVE_DIRECTORY "${LIBRARY_DIRECTORY}")
--- origsrc/aws-sdk-cpp-1.3.22/aws-cpp-sdk-core/CMakeLists.txt	2017-12-13 06:33:53.000000000 +0900
+++ src/aws-sdk-cpp-1.3.22/aws-cpp-sdk-core/CMakeLists.txt	2017-12-16 17:24:37.584447400 +0900
@@ -394,6 +394,10 @@ else()
     endif()
 endif()
 
+if(CYGWIN)
+    set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION 0)
+endif()
+
 if(SIMPLE_INSTALL)
     configure_file("${AWS_NATIVE_SDK_ROOT}/toolchains/pkg-config.pc.in" "${PROJECT_NAME}.pc" @ONLY)
 
--- origsrc/aws-sdk-cpp-1.3.22/cmake/compiler_settings.cmake	2017-12-13 06:33:53.000000000 +0900
+++ src/aws-sdk-cpp-1.3.22/cmake/compiler_settings.cmake	2017-12-14 23:29:52.787000000 +0900
@@ -33,7 +33,11 @@ endfunction()
 
 
 macro(set_gcc_flags)
-    list(APPEND AWS_COMPILER_FLAGS "-fno-exceptions" "-std=c++${CPP_STANDARD}")
+    if(CYGWIN)
+        list(APPEND AWS_COMPILER_FLAGS "-fno-exceptions" "-std=gnu++${CPP_STANDARD}")
+    else()
+        list(APPEND AWS_COMPILER_FLAGS "-fno-exceptions" "-std=c++${CPP_STANDARD}")
+    endif()
 
     if(NOT BUILD_SHARED_LIBS)
         list(APPEND AWS_COMPILER_FLAGS "-fPIC")
--- origsrc/aws-sdk-cpp-1.3.22/cmake/utilities.cmake	2017-12-13 06:33:53.000000000 +0900
+++ src/aws-sdk-cpp-1.3.22/cmake/utilities.cmake	2017-12-16 17:24:43.590289300 +0900
@@ -83,6 +83,9 @@ macro(setup_install)
                      RUNTIME DESTINATION ${BINARY_DIRECTORY}/${SDK_INSTALL_BINARY_PREFIX}/${PLATFORM_INSTALL_QUALIFIER}/\${CMAKE_INSTALL_CONFIG_NAME})
         endif()
     endif()
+    if(CYGWIN)
+        set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION 0)
+    endif()
 endmacro()
 
 macro(do_packaging)
