--- origsrc/aws-sdk-cpp-1.7.275/CMakeLists.txt	2020-02-15 05:54:09.000000000 +0900
+++ src/aws-sdk-cpp-1.7.275/CMakeLists.txt	2020-02-17 00:24:26.824228700 +0900
@@ -306,7 +306,7 @@ if (CMAKE_INSTALL_INCLUDEDIR)
     set(INCLUDE_DIRECTORY "${CMAKE_INSTALL_INCLUDEDIR}")
 endif()
 
-if(BUILD_SHARED_LIBS)
+if(BUILD_SHARED_LIBS AND NOT CYGWIN)
     set(ARCHIVE_DIRECTORY "${BINARY_DIRECTORY}")
 else()
     set(ARCHIVE_DIRECTORY "${LIBRARY_DIRECTORY}")
--- origsrc/aws-sdk-cpp-1.7.275/aws-cpp-sdk-core/CMakeLists.txt	2020-02-15 05:54:09.000000000 +0900
+++ src/aws-sdk-cpp-1.7.275/aws-cpp-sdk-core/CMakeLists.txt	2020-02-17 00:24:26.902345200 +0900
@@ -528,6 +528,10 @@ else()
     endif()
 endif()
 
+if(CYGWIN)
+    set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION 0)
+endif()
+
 if(SIMPLE_INSTALL)
     # These libs are needed by aws-cpp-sdk-core only.
     set(ALL_DEP_LIBS ${THIRD_PARTY_LIBS} ${PLATFORM_DEP_LIBS_ABSTRACT_NAME} ${CLIENT_LIBS_ABSTRACT_NAME} ${CRYPTO_LIBS_ABSTRACT_NAME})
--- origsrc/aws-sdk-cpp-1.7.275/cmake/compiler_settings.cmake	2020-02-15 05:54:09.000000000 +0900
+++ src/aws-sdk-cpp-1.7.275/cmake/compiler_settings.cmake	2020-02-17 00:24:26.964872800 +0900
@@ -33,7 +33,11 @@ endfunction()
 
 
 macro(set_gcc_flags)
+    if(CYGWIN)
+        list(APPEND AWS_COMPILER_FLAGS "-fno-exceptions" "-std=gnu++${CPP_STANDARD}")
+    else()
     list(APPEND AWS_COMPILER_FLAGS "-fno-exceptions" "-std=c++${CPP_STANDARD}")
+    endif()
 
     if(NOT BUILD_SHARED_LIBS)
         list(APPEND AWS_COMPILER_FLAGS "-fPIC")
@@ -53,7 +57,10 @@ macro(set_gcc_flags)
 endmacro()
 
 macro(set_gcc_warnings)
-    list(APPEND AWS_COMPILER_WARNINGS "-Wall" "-Werror" "-pedantic" "-Wextra")
+    if(NOT CYGWIN)
+        list(APPEND AWS_COMPILER_WARNINGS "-Werror")
+    endif()
+    list(APPEND AWS_COMPILER_WARNINGS "-Wall" "-pedantic" "-Wextra")
     if(COMPILER_CLANG)
         if(PLATFORM_ANDROID)
             # when using clang with libc and API lower than 21 we need to include Android support headers and ignore the gnu-include-next warning.
--- origsrc/aws-sdk-cpp-1.7.275/cmake/utilities.cmake	2020-02-15 05:54:09.000000000 +0900
+++ src/aws-sdk-cpp-1.7.275/cmake/utilities.cmake	2020-02-17 00:24:26.980473600 +0900
@@ -56,6 +56,9 @@ macro(setup_install)
                      RUNTIME DESTINATION ${BINARY_DIRECTORY}/${SDK_INSTALL_BINARY_PREFIX}/${PLATFORM_INSTALL_QUALIFIER}/\${CMAKE_INSTALL_CONFIG_NAME})
         endif()
     endif()
+    if(CYGWIN)
+        set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION 0)
+    endif()
 endmacro()
 
 macro(do_packaging)
