--- origsrc/dnswire/src/Makefile.am	2026-02-15 09:37:03.000000000 +0900
+++ src/dnswire/src/Makefile.am	2026-02-15 09:37:40.658157200 +0900
@@ -18,7 +18,7 @@ nobase_include_HEADERS = dnswire/decoder
   dnswire/dnswire.h dnswire/encoder.h dnswire/reader.h dnswire/writer.h
 nobase_nodist_include_HEADERS = dnswire/version.h dnswire/dnstap.pb-c.h \
   dnswire/dnstap-macros.h dnswire/trace.h
-libdnswire_la_LDFLAGS = -version-info $(DNSWIRE_LIBRARY_VERSION) \
+libdnswire_la_LDFLAGS = -no-undefined -version-info $(DNSWIRE_LIBRARY_VERSION) \
   $(protobuf_c_LIBS) \
   $(tinyframe_LIBS)
 
--- origsrc/dnswire/src/test/Makefile.am	2026-02-15 09:37:03.000000000 +0900
+++ src/dnswire/src/test/Makefile.am	2026-02-15 09:37:58.504429400 +0900
@@ -17,47 +17,47 @@ EXTRA_DIST = create_dnstap.c print_dnsta
 
 reader_read_SOURCES = reader_read.c
 reader_read_LDADD = ../libdnswire.la
-reader_read_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS) -static
+reader_read_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS)
 
 reader_push_SOURCES = reader_push.c
 reader_push_LDADD = ../libdnswire.la
-reader_push_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS) -static
+reader_push_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS)
 
 writer_write_SOURCES = writer_write.c
 writer_write_LDADD = ../libdnswire.la
-writer_write_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS) -static
+writer_write_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS)
 
 writer_pop_SOURCES = writer_pop.c
 writer_pop_LDADD = ../libdnswire.la
-writer_pop_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS) -static
+writer_pop_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS)
 
 reader_unixsock_SOURCES = reader_unixsock.c
 reader_unixsock_LDADD = ../libdnswire.la
-reader_unixsock_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS) -static
+reader_unixsock_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS)
 
 writer_unixsock_SOURCES = writer_unixsock.c
 writer_unixsock_LDADD = ../libdnswire.la
-writer_unixsock_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS) -static
+writer_unixsock_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS)
 
 test_dnstap_SOURCES = test_dnstap.c
 test_dnstap_LDADD = ../libdnswire.la
-test_dnstap_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS) -static
+test_dnstap_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS)
 
 test_encoder_SOURCES = test_encoder.c
 test_encoder_LDADD = ../libdnswire.la
-test_encoder_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS) -static
+test_encoder_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS)
 
 test_decoder_SOURCES = test_decoder.c
 test_decoder_LDADD = ../libdnswire.la
-test_decoder_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS) -static
+test_decoder_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS)
 
 test_reader_SOURCES = test_reader.c
 test_reader_LDADD = ../libdnswire.la
-test_reader_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS) -static
+test_reader_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS)
 
 test_writer_SOURCES = test_writer.c
 test_writer_LDADD = ../libdnswire.la
-test_writer_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS) -static
+test_writer_LDFLAGS = $(protobuf_c_LIBS) $(tinyframe_LIBS)
 
 if ENABLE_GCOV
 gcov-local:
