--- origsrc/libdsk-1.3.3/configure.in	2010-06-29 04:43:20.000000000 +0900
+++ src/libdsk-1.3.3/configure.in	2011-08-07 23:14:21.998236600 +0900
@@ -51,7 +51,11 @@ AC_HEADER_STDC
 AC_CHECK_HEADERS(errno.h limits.h sys/ioctl.h stat.h sys/stat.h sys/types.h)
 AC_CHECK_HEADERS(unistd.h termios.h)
 AC_CHECK_HEADERS(dirent.h fcntl.h utime.h pwd.h time.h dir.h direct.h)
-AC_CHECK_HEADERS(linux/fd.h linux/fdreg.h windows.h winioctl.h shlobj.h)
+AC_CHECK_HEADERS(linux/fd.h linux/fdreg.h winioctl.h shlobj.h)
+case "$host_os" in
+	cygwin*) ;;
+	*) AC_CHECK_HEADERS(windows.h) ;;
+esac
 
 dnl Checks for functions
 AC_CHECK_FUNCS(strcmpi stricmp strcasecmp, break)
--- origsrc/libdsk-1.3.3/lib/Makefile.am	2008-06-03 03:48:43.000000000 +0900
+++ src/libdsk-1.3.3/lib/Makefile.am	2011-08-07 23:14:34.411477800 +0900
@@ -30,7 +30,7 @@ EXTRA_DIST=$(CLASSDPRE)/Drive.java \
 # If interfaces have been removed set the right-hand number to 0.
 # 
 #
-libdsk_la_LDFLAGS = -version-info 5:4:2
+libdsk_la_LDFLAGS = -version-info 5:4:2 -no-undefined
 libdsk_la_SOURCES = dskread.c  dskwrite.c dskcheck.c dskstat.c \
 		   dsklphys.c dskfmt.c   dskopen.c  dskpars.c \
 		   dskerror.c dskseek.c  dsksecid.c dskgeom.c \
