--- origsrc/librtp-0.1/configure.in	2000-01-08 07:44:03.000000000 +0900
+++ src/librtp-0.1/configure.in	2010-11-28 18:00:23.779800200 +0900
@@ -2,7 +2,7 @@ dnl Process this file with autoconf to p
 AC_INIT(src/rtp-packet.c)
 AM_INIT_AUTOMAKE(librtp, 0.1)
 AM_CONFIG_HEADER(config.h)
-AM_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 dnl Checks for programs.
 AC_PROG_CC
@@ -10,7 +10,6 @@ AC_PROG_CPP
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
-AC_PROG_RANLIB
 
 AM_PATH_GLIB(1.2.0,
             ,
--- origsrc/librtp-0.1/src/Makefile.am	2001-05-08 02:35:39.000000000 +0900
+++ src/librtp-0.1/src/Makefile.am	2010-11-28 17:54:49.659358600 +0900
@@ -3,9 +3,9 @@ lib_LTLIBRARIES = librtp.la
 INCLUDES = -I$(top_srcdir)/include $(GLIB_CFLAGS)
 
 librtp_la_SOURCES = rtp.c random.c md5c.c rtp-packet.c rtcp-packet.c
-librtp_la_LDFLAGS = -version-info 0:0:0 $(GLIB_LIBS)
+librtp_la_LDFLAGS = -version-info 0:0:0 $(GLIB_LIBS) -no-undefined
 
-AM_CFLAGS = -g -Wall
+AM_CFLAGS = -Wall
 
 
 
