diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/beansdb-0.7.1.4/Makefile.am src/beansdb-0.7.1.4/Makefile.am
--- origsrc/beansdb-0.7.1.4/Makefile.am	2015-06-19 22:51:10.000000000 +0900
+++ src/beansdb-0.7.1.4/Makefile.am	2016-03-22 12:58:49.909024700 +0900
@@ -1,4 +1,4 @@
-SUBDIRS = third-party/zlog-1.2 src doc
+SUBDIRS = src doc
 DIST_SUBDIRS = python
 
 EXTRA_DIST = python src/crc32.c src/clock_gettime_stub.c src/ae_epoll.c src/ae_kqueue.c src/ae_select.c CREDITS AUTHORS LICENSE
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/beansdb-0.7.1.4/configure.ac src/beansdb-0.7.1.4/configure.ac
--- origsrc/beansdb-0.7.1.4/configure.ac	2015-06-19 22:51:10.000000000 +0900
+++ src/beansdb-0.7.1.4/configure.ac	2016-03-22 12:59:06.563570600 +0900
@@ -49,6 +49,5 @@ fi
 
 AC_C_SOCKLEN_T
 
-AC_CONFIG_SUBDIRS([third-party/zlog-1.2])
 AC_CONFIG_FILES([Makefile doc/Makefile src/Makefile])
 AC_OUTPUT
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/beansdb-0.7.1.4/src/Makefile.am src/beansdb-0.7.1.4/src/Makefile.am
--- origsrc/beansdb-0.7.1.4/src/Makefile.am	2015-06-19 22:51:10.000000000 +0900
+++ src/beansdb-0.7.1.4/src/Makefile.am	2016-03-22 12:59:23.407466400 +0900
@@ -1,7 +1,4 @@
 bin_PROGRAMS = beansdb
 #export JEMALLOC_PATH=${HOME}/local/jemalloc-3.6.0
 beansdb_SOURCES = beansdb.c item.c fnv1a.h  beansdb.h thread.c htree.h htree.c hint.h hint.c record.h record.c codec.h codec.c bitcask.h bitcask.c hstore.h hstore.c quicklz.h quicklz.c diskmgr.h diskmgr.c util.h const.h log.h log.c mfile.h mfile.c common.c
-beansdb_CPPFLAGS = -I ../third-party/zlog-1.2/ # -I${JEMALLOC_PATH}/include
-beansdb_LDFLAGS =  -L ../third-party/zlog-1.2/ # -L ${JEMALLOC_PATH}/lib -Wl,-rpath,${JEMALLOC_PATH}/lib
 LIBS += -lzlog # -ljemalloc
-
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/beansdb-0.7.1.4/src/ae_epoll.c src/beansdb-0.7.1.4/src/ae_epoll.c
--- origsrc/beansdb-0.7.1.4/src/ae_epoll.c	2015-06-19 22:51:10.000000000 +0900
+++ src/beansdb-0.7.1.4/src/ae_epoll.c	2016-03-22 13:01:48.855074600 +0900
@@ -6,6 +6,10 @@
 #include <errno.h>
 #include "log.h"
 
+#if !defined(EPOLLONESHOT)
+#define EPOLLONESHOT (0)
+#endif
+
 typedef struct aeApiState
 {
     int epfd;
diff -wuprN -x CYGWIN-PATCHES -x aclocal.m4~ -x aclocal.m4t -x autom4te.cache -x config.cache -x config.guess -x config.log -x config.status -x config.sub -x '*SlackBuild*' -x '*.egg-info' -x '*.dll' -x '*.exe' -x '*.o' -x '*.gch' -x '*.gem' -x .gitignore -x '*.jpg' -x '*.png' -x '*.qm' -x '*.svgz' -x '*.tif' -x '*.class' -x '*.jlc' -x '*.pyc' -x '*.pyo' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x aclocal.m4 -x configure -x config.h -x '*.h.in' -x COPYING -x INSTALL -x ansi2knr.c -x ar-lib -x config-ml.in -x elisp-comp -x mdate-sh -x mkinstalldirs -x py-compile -x symlink-tree -x test-driver -x texinfo.tex -x ylwrap -x Makefile.in -x GNUmakefile.in -x makefile.in -x compile -x depcomp -x install-sh -x missing -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x mate-doc-utils.make -x mate-doc-utils.m4 origsrc/beansdb-0.7.1.4/src/thread.c src/beansdb-0.7.1.4/src/thread.c
--- origsrc/beansdb-0.7.1.4/src/thread.c	2015-06-19 22:51:10.000000000 +0900
+++ src/beansdb-0.7.1.4/src/thread.c	2016-03-22 13:03:11.760276700 +0900
@@ -114,6 +114,9 @@ void mt_stats_unlock()
 
 /* Include the best multiplexing layer supported by this system.
  * The following should be ordered by performances, descending. */
+#if defined(__CYGWIN__)
+#include "ae_select.c"
+#else
 #ifdef HAVE_EPOLL
 #include "ae_epoll.c"
 #else
@@ -123,6 +126,7 @@ void mt_stats_unlock()
 #include "ae_select.c"
 #endif
 #endif
+#endif
 
 /*
  * Initializes the thread subsystem, creating various worker threads.
