--- origsrc/thames-0.1.1/configure.in	2012-09-12 06:05:45.000000000 +0900
+++ src/thames-0.1.1/configure.in	2020-08-02 22:12:42.365063800 +0900
@@ -1,27 +1,13 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(bin/thames.c)
+AC_INIT([thames], [0.1.1])
 AC_CONFIG_AUX_DIR(config)
 AC_CONFIG_MACRO_DIR([m4])
 VERSION=0.1.1
 UPDATED='September 11, 2012'
-AM_INIT_AUTOMAKE(thames, $VERSION)
+AM_INIT_AUTOMAKE
 AM_CONFIG_HEADER(config.h)
 LT_INIT
 
-dnl Get an absolute version of $srcdir and .
-
-LSUBDIRS=""
-DSTROOT=`pwd`
-if test x"$srcdir" = x
-then
-	SRCROOT=`pwd`
-else
-	cd $srcdir
-	SRCROOT=`pwd`
-	cd $DSTROOT
-fi
-
-
 dnl Checks for programs.
 AC_PROG_CC
 
@@ -38,4 +24,3 @@ AC_FUNC_VPRINTF
 AC_CHECK_FUNCS(strstr)
 
 AC_OUTPUT(Makefile bin/Makefile)
-
