--- origsrc/frog-0.31/src/Makefile.am	2023-10-22 02:11:16.000000000 +0900
+++ src/frog-0.31/src/Makefile.am	2023-12-02 11:45:16.418514500 +0900
@@ -9,7 +9,7 @@ ner_SOURCES = ner_prog.cxx
 
 LDADD = libfrog.la
 lib_LTLIBRARIES = libfrog.la
-libfrog_la_LDFLAGS = -version-info 3:0:0
+libfrog_la_LDFLAGS = -no-undefined -version-info 3:0:0
 
 libfrog_la_SOURCES = FrogAPI.cxx FrogData.cxx \
 	mbma_rule.cxx mbma_mod.cxx mbma_brackets.cxx clex.cxx \
