--- origsrc/cpl-7.3.2/configure.ac	2023-04-23 22:01:21.000000000 +0900
+++ src/cpl-7.3.2/configure.ac	2025-01-16 12:55:09.205638800 +0900
@@ -68,8 +68,8 @@ AC_CHECK_LIB(socket, socket, [LIBS="$LIB
 AC_CHECK_LIB(nsl, inet_ntoa, [LIBS="$LIBS -lnsl"])
 
 CPL_CONFIG_CEXT
-CPL_CONFIG_CFITSIO([3.35])
-CPL_CHECK_WCS([4.16])
+PKG_CHECK_MODULES([CFITSIO], [cfitsio])
+PKG_CHECK_MODULES([WCSLIB], [wcslib])
 CPL_CHECK_FFTW([3.3.3])
 
 # Checks for header files.
--- origsrc/cpl-7.3.2/cplcore/Makefile.am	2023-04-23 22:01:22.000000000 +0900
+++ src/cpl-7.3.2/cplcore/Makefile.am	2025-01-16 13:13:47.918342800 +0900
@@ -36,8 +36,8 @@ endif
 # obsolete and therefore unwanted CFITSIO installations
 
 AM_CPPFLAGS = -DCX_LOG_DOMAIN=\"CplCore\" \
-	          -I\$(top_srcdir)/cpldrs $(CX_INCLUDES) $(CFITSIO_INCLUDES) \
-	          $(WCS_INCLUDES) $(FFTW_INCLUDES) $(FFTWF_INCLUDES) \
+	          -I$(top_srcdir)/cpldrs $(CX_INCLUDES) $(CFITSIO_CFLAGS) \
+	          $(WCSLIB_CFLAGS) $(FFTW_INCLUDES) $(FFTWF_INCLUDES) \
 	          $(CPL_CFLAGS)
 
 EXTRA_DIST = cpl_version.h.top cpl_version.h.bot cpl_func.h.top cpl_func.h.bot \
@@ -143,8 +143,8 @@ libcplcore_la_SOURCES = cpl_array.c \
 
 # Place optional 3rd party components last since those locations may contain
 # obsolete and therefore unwanted CFITSIO installations
-libcplcore_la_LDFLAGS = $(CX_LDFLAGS) $(CFITSIO_LDFLAGS) $(FFTW_LDFLAGS) $(FFTWF_LDFLAGS) -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
-libcplcore_la_LIBADD =  $(LIBCEXT) $(LIBCFITSIO) $(LIBFFTW) $(LIBFFTWF) -lm
+libcplcore_la_LDFLAGS = -no-undefined $(CX_LDFLAGS) $(CFITSIO_LDFLAGS) $(FFTW_LDFLAGS) $(FFTWF_LDFLAGS) -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
+libcplcore_la_LIBADD =  $(LIBCEXT) $(CFITSIO_LIBS) $(LIBFFTW) $(LIBFFTWF) -lm
 libcplcore_la_DEPENDENCIES =
 
 
--- origsrc/cpl-7.3.2/cplcore/tests/Makefile.am	2025-01-16 12:54:39.929805600 +0900
+++ src/cpl-7.3.2/cplcore/tests/Makefile.am	2025-01-16 12:56:48.535792000 +0900
@@ -29,7 +29,7 @@ MAINTAINERCLEANFILES = $(srcdir)/Makefil
 
 endif
 
-AM_CPPFLAGS = $(CPLCORE_INCLUDES) $(CX_INCLUDES) $(CFITSIO_INCLUDES) \
+AM_CPPFLAGS = $(CPLCORE_INCLUDES) $(CX_INCLUDES) $(CFITSIO_CFLAGS) \
 	          $(CPL_CFLAGS)
 AM_LDFLAGS = $(CFITSIO_LDFLAGS)
 
--- origsrc/cpl-7.3.2/cpldfs/Makefile.am	2023-04-23 22:01:20.000000000 +0900
+++ src/cpl-7.3.2/cpldfs/Makefile.am	2025-01-16 12:58:01.520858400 +0900
@@ -32,7 +32,7 @@ SUBDIRS = tests
 
 AM_CPPFLAGS = -DCX_LOG_DOMAIN=\"CplDfs\" \
 	          $(CPLDFS_INCLUDES) $(CPLUI_INCLUDES) \
-              $(CPLCORE_INCLUDES) $(CX_INCLUDES) $(CFITSIO_INCLUDES)
+              $(CPLCORE_INCLUDES) $(CX_INCLUDES) $(CFITSIO_CFLAGS)
 
 include_HEADERS = cpl_dfs.h cpl_multiframe.h
 
@@ -42,9 +42,9 @@ lib_LTLIBRARIES = libcpldfs.la
 
 libcpldfs_la_SOURCES = cpl_dfs.c cpl_multiframe.c
 
-libcpldfs_la_LDFLAGS = $(CX_LDFLAGS) $(CFITSIO_LDFLAGS) -version-info \
+libcpldfs_la_LDFLAGS = -no-undefined $(CX_LDFLAGS) $(CFITSIO_LDFLAGS) -version-info \
                        $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
-libcpldfs_la_LIBADD =  $(LIBCPLUI) $(LIBCPLCORE) $(LIBCFITSIO) \
+libcpldfs_la_LIBADD =  $(LIBCPLUI) $(LIBCPLCORE) $(CFITSIO_LIBS) \
                        $(LIBCEXT) -lm
 libcpldfs_la_DEPENDENCIES = $(LIBCPLCORE)
 
--- origsrc/cpl-7.3.2/cpldfs/tests/Makefile.am	2025-01-16 12:54:39.938837600 +0900
+++ src/cpl-7.3.2/cpldfs/tests/Makefile.am	2025-01-16 12:59:44.195486600 +0900
@@ -31,7 +31,7 @@ endif
 
 
 AM_CPPFLAGS = $(CPLDFS_INCLUDES) $(CPLCORE_INCLUDES) \
-              $(CPLUI_INCLUDES) $(CX_INCLUDES) $(CFITSIO_INCLUDES)
+              $(CPLUI_INCLUDES) $(CX_INCLUDES) $(CFITSIO_CFLAGS)
 
 LDADD = $(LIBCPLDFS) $(LIBCPLUI) $(LIBCPLCORE) $(LIBCEXT)
 
--- origsrc/cpl-7.3.2/cpldrs/Makefile.am	2023-04-23 22:01:23.000000000 +0900
+++ src/cpl-7.3.2/cpldrs/Makefile.am	2025-01-16 13:01:09.067410900 +0900
@@ -36,7 +36,7 @@ SUBDIRS = tests
 
 AM_CPPFLAGS = -DCX_LOG_DOMAIN=\"CplDrs\" \
               $(CPLDRS_INCLUDES) $(CPLCORE_INCLUDES) $(CX_INCLUDES) \
-	          $(CFITSIO_INCLUDES) $(WCS_INCLUDES) $(FFTW_INCLUDES) \
+	          $(CFITSIO_CFLAGS) $(WCSLIB_CFLAGS) $(FFTW_INCLUDES) \
 	          $(FFTWF_INCLUDES)
 
 include_HEADERS =   cpl_apertures.h \
@@ -71,6 +71,6 @@ libcpldrs_la_SOURCES =  cpl_apertures.c
 
 # Place optional 3rd party components last since those locations may contain
 # obsolete and therefore unwanted CFITSIO installations
-libcpldrs_la_LDFLAGS = $(CX_LDFLAGS) $(CFITSIO_LDFLAGS) $(FFTW_LDFLAGS) $(FFTWF_LDFLAGS) $(WCS_LDFLAGS) -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
-libcpldrs_la_LIBADD = $(LIBCPLCORE) $(LIBCFITSIO) $(LIBWCS) $(LIBCEXT) $(LIBFFTW) $(LIBFFTWF) -lm
+libcpldrs_la_LDFLAGS = -no-undefined $(CX_LDFLAGS) $(CFITSIO_LDFLAGS) $(FFTW_LDFLAGS) $(FFTWF_LDFLAGS) $(WCS_LDFLAGS) -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
+libcpldrs_la_LIBADD = $(LIBCPLCORE) $(CFITSIO_LIBS) $(WCSLIB_LIBS) $(LIBCEXT) $(LIBFFTW) $(LIBFFTWF) -lm
 libcpldrs_la_DEPENDENCIES = $(LIBCPLCORE)
--- origsrc/cpl-7.3.2/cpldrs/tests/Makefile.am	2025-01-16 12:54:39.938837600 +0900
+++ src/cpl-7.3.2/cpldrs/tests/Makefile.am	2025-01-16 13:01:37.373969100 +0900
@@ -30,7 +30,7 @@ MAINTAINERCLEANFILES = $(srcdir)/Makefil
 endif
 
 AM_CPPFLAGS = $(CPLDRS_INCLUDES) $(CPLCORE_INCLUDES) $(CX_INCLUDES) \
-	          $(CFITSIO_INCLUDES) $(WCS_INCLUDES) \
+	          $(CFITSIO_CFLAGS) $(WCSLIB_CFLAGS) \
 	          $(CPL_CFLAGS)
 
 LDADD = $(LIBCPLDRS) $(LIBCPLCORE) $(LIBCEXT) $(LIBWCS) -lm
--- origsrc/cpl-7.3.2/cplui/Makefile.am	2023-04-23 22:01:23.000000000 +0900
+++ src/cpl-7.3.2/cplui/Makefile.am	2025-01-16 13:02:09.078875000 +0900
@@ -32,7 +32,7 @@ endif
 SUBDIRS = tests
 
 AM_CPPFLAGS = -DCX_LOG_DOMAIN=\"CplUi\" \
-              $(CPLCORE_INCLUDES) $(CX_INCLUDES) $(CFITSIO_INCLUDES)
+              $(CPLCORE_INCLUDES) $(CX_INCLUDES) $(CFITSIO_CFLAGS)
 
 include_HEADERS = cpl_frame.h cpl_frameset.h cpl_frameset_io.h \
                   cpl_parameter.h cpl_parameterlist.h cpl_plugin.h \
@@ -48,6 +48,6 @@ libcplui_la_SOURCES = cpl_frame.c cpl_fr
                       cpl_recipedefine.c  \
                       cpl_pluginlist.c cpl_recipeconfig.c cpl_framedata.c
 
-libcplui_la_LDFLAGS = $(CX_LDFLAGS) $(CFITSIO_LDFLAGS) -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
-libcplui_la_LIBADD = $(LIBCPLCORE) $(LIBCFITSIO) $(LIBCEXT) -lm
+libcplui_la_LDFLAGS = -no-undefined $(CX_LDFLAGS) $(CFITSIO_LDFLAGS) -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
+libcplui_la_LIBADD = $(LIBCPLCORE) $(CFITSIO_LIBS) $(LIBCEXT) -lm
 libcplui_la_DEPENDENCIES = $(LIBCPLCORE)
--- origsrc/cpl-7.3.2/cplui/tests/Makefile.am	2025-01-16 12:54:39.938837600 +0900
+++ src/cpl-7.3.2/cplui/tests/Makefile.am	2025-01-16 13:02:32.789104800 +0900
@@ -30,10 +30,10 @@ MAINTAINERCLEANFILES = $(srcdir)/Makefil
 endif
 
 
-LDADD = $(LIBCPLUI) $(LIBCPLCORE) $(LIBCFITSIO) $(LIBCEXT)
+LDADD = $(LIBCPLUI) $(LIBCPLCORE) $(CFITSIO_LIBS) $(LIBCEXT)
 
 AM_CPPFLAGS = $(CPLUI_INCLUDES) $(CPLCORE_INCLUDES) $(CX_INCLUDES) \
-              $(CFITSIO_INCLUDES)
+              $(CFITSIO_CFLAGS)
 AM_LDFLAGS = $(CFITSIO_LDFLAGS)
 
 
--- origsrc/cpl-7.3.2/libcext/cext/Makefile.am	2023-04-23 22:01:24.000000000 +0900
+++ src/cpl-7.3.2/libcext/cext/Makefile.am	2025-01-16 13:03:30.912399900 +0900
@@ -49,7 +49,7 @@ libcext_la_SOURCES = cxfileutils.c cxlis
 
 EXTRA_libcext_la_SOURCES = snprintf.h snprintf.c
 
-libcext_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
+libcext_la_LDFLAGS = -no-undefined -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 libcext_la_LIBADD = @SNPRINTF@
 libcext_la_DEPENDENCIES = @SNPRINTF@
 
