--- origsrc/abyss-2.3.7/Common/Uncompress.cpp	2023-05-17 07:53:03.000000000 +0900
+++ src/abyss-2.3.7/Common/Uncompress.cpp	2023-11-26 17:07:35.261279300 +0900
@@ -7,6 +7,7 @@
  * @author Shaun Jackman <sjackman@bcgsc.ca>
  */
 
+#if !defined(__CYGWIN__)
 #include "config.h"
 #if HAVE_LIBDL
 
@@ -214,12 +215,15 @@ int open(const char *path, int flags, mo
 } // extern "C"
 
 #endif // HAVE_LIBDL
+#endif // !defined(__CYGWIN__)
 
 /** Initialize the uncompress module. */
 bool uncompress_init()
 {
 #if HAVE_LIBDL
 	signalInit();
-#endif
 	return HAVE_LIBDL;
+#else
+	return 0;
+#endif
 }
