--- origsrc/libgta-1.0.7/CMakeLists.txt	2014-12-18 14:23:56.093752600 +0900
+++ src/libgta-1.0.7/CMakeLists.txt	2015-01-06 00:38:50.949442000 +0900
@@ -53,7 +53,7 @@ if(GTA_BUILD_SHARED_LIB)
   set_target_properties(libgta_shared PROPERTIES DEFINE_SYMBOL DLL_EXPORT)
   set_target_properties(libgta_shared PROPERTIES OUTPUT_NAME gta)
   set_target_properties(libgta_shared PROPERTIES VERSION ${GTA_LIB_VERSION})
-  set_target_properties(libgta_shared PROPERTIES SOVERSION ${GTA_LIB_SOVERSION})  
+  set_target_properties(libgta_shared PROPERTIES SOVERSION ${GTA_LIB_SOVERSION})
   install(TARGETS libgta_shared
     RUNTIME DESTINATION bin
     LIBRARY DESTINATION "lib${LIB_SUFFIX}"
@@ -96,7 +96,7 @@ if(GTA_BUILD_DOCUMENTATION)
   )
   add_custom_target(doc ALL DEPENDS "${CMAKE_BINARY_DIR}/doc/reference/index.html")
   install(DIRECTORY "${CMAKE_BINARY_DIR}/doc/reference" DESTINATION share/doc/libgta)
-  install(FILES "${CMAKE_SOURCE_DIR}/cmake/FindGTA.cmake" DESTINATION share/libgta/cmake)
+  install(FILES "${CMAKE_SOURCE_DIR}/cmake/FindGTA.cmake" DESTINATION lib/cmake/libgta)
 endif()
 
 # Extra target: 'make dist' for making
--- origsrc/libgta-1.0.7/Makefile.am	2014-12-18 14:22:24.189756400 +0900
+++ src/libgta-1.0.7/Makefile.am	2015-01-06 00:41:09.168192000 +0900
@@ -9,7 +9,7 @@ SUBDIRS = doc src tests
 
 EXTRA_DIST = CMakeLists.txt cmake/FindGTA.cmake
 
-cmakedir = $(datadir)/$(PACKAGE)/cmake
+cmakedir = $(libdir)/cmake/$(PACKAGE)
 cmake_DATA = cmake/FindGTA.cmake
 
 fuzztest:
