--- origsrc/alembic-1.8.7/CMakeLists.txt	2024-10-08 02:12:28.000000000 +0900
+++ src/alembic-1.8.7/CMakeLists.txt	2024-10-23 13:52:59.751269000 +0900
@@ -327,3 +327,10 @@ set(CPACK_PACKAGE_CONTACT "Lucas Miller"
 # the future (the example is for Ubuntu 22.04)
 # set(CPACK_DEBIAN_PACKAGE_DEPENDS "libimath-3-1-29")
 include(CPack)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/alembic.pc.in ${CMAKE_BINARY_DIR}/alembic.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/alembic.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/alembic-1.8.7/alembic.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/alembic-1.8.7/alembic.pc.in	2024-10-23 13:52:59.766892700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: alembic
+Description: Open framework for storing and sharing scene data
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lAlembic
+Cflags: -I${includedir}
