--- origsrc/alembic-1.7.11/CMakeLists.txt	2019-05-17 03:26:44.000000000 +0900
+++ src/alembic-1.7.11/CMakeLists.txt	2019-05-28 08:41:37.392554300 +0900
@@ -343,3 +343,10 @@ info_cfg_option(ALEMBIC_SHARED_LIBS)
 info_cfg_option(ALEMBIC_LIB_USES_BOOST)
 info_cfg_option(ALEMBIC_LIB_USES_TR1)
 MESSAGE("${_config_msg}")
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/alembic.pc.in ${CMAKE_BINARY_DIR}/alembic.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/alembic.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/alembic-1.7.11/alembic.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/alembic-1.7.11/alembic.pc.in	2019-05-28 08:41:37.408231600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: alembic
+Description: Open framework for storing and sharing scene data
+Version: @VERSION@
+Libs: -L${libdir} -lAlembic
+Cflags: -I${includedir}
--- origsrc/alembic-1.7.11/lib/Alembic/CMakeLists.txt	2019-05-17 03:26:44.000000000 +0900
+++ src/alembic-1.7.11/lib/Alembic/CMakeLists.txt	2019-05-28 08:41:37.423901600 +0900
@@ -90,7 +90,7 @@ INSTALL(TARGETS Alembic
         EXPORT AlembicTargets
         LIBRARY DESTINATION ${ALEMBIC_LIB_INSTALL_DIR}
         ARCHIVE DESTINATION ${ALEMBIC_LIB_INSTALL_DIR}
-        RUNTIME DESTINATION ${ALEMBIC_LIB_INSTALL_DIR})
+        RUNTIME DESTINATION bin)
 
 #-******************************************************************************
 # PACKAGE EXPORTS
