--- origsrc/libmdbx-0.13.10/CMakeLists.txt	2025-12-17 23:57:38.000000000 +0900
+++ src/libmdbx-0.13.10/CMakeLists.txt	2026-01-19 12:20:53.238986800 +0900
@@ -984,6 +984,9 @@ if(MDBX_BUILD_SHARED_LIBRARY)
   endif()
   list(APPEND MDBX_BUILD_FLAGS ${CMAKE_SHARED_LINKER_FLAGS})
   target_include_directories(mdbx INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}")
+  if (CYGWIN)
+    set_target_properties(mdbx PROPERTIES SOVERSION "${MDBX_VERSION_MAJOR}")
+  endif()
 endif()
 
 if(MDBX_BUILD_SHARED_LIBRARY AND MDBX_LINK_TOOLS_NONSTATIC)
@@ -1064,7 +1067,7 @@ if(MDBX_BUILD_SHARED_LIBRARY)
     install(
       TARGETS mdbx
       EXPORT libmdbx
-      LIBRARY DESTINATION ${MDBX_DLL_INSTALL_DESTINATION} COMPONENT runtime
+      RUNTIME DESTINATION ${MDBX_DLL_INSTALL_DESTINATION} COMPONENT runtime
       ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT devel
       PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} COMPONENT devel
       INCLUDES
@@ -1074,9 +1077,8 @@ if(MDBX_BUILD_SHARED_LIBRARY)
     install(
       TARGETS mdbx
       EXPORT libmdbx
-      LIBRARY DESTINATION ${MDBX_DLL_INSTALL_DESTINATION}
+      RUNTIME DESTINATION ${MDBX_DLL_INSTALL_DESTINATION}
               COMPONENT runtime
-              NAMELINK_COMPONENT devel
       OBJECTS DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT devel
       ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT devel
       PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} COMPONENT devel
@@ -1285,3 +1287,10 @@ if(NOT SUBPROJECT)
 endif()
 
 cmake_policy(POP)
+
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+configure_file(${CMAKE_SOURCE_DIR}/libmdbx.pc.in ${CMAKE_BINARY_DIR}/libmdbx.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libmdbx.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/libmdbx-0.13.10/libmdbx.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/libmdbx-0.13.10/libmdbx.pc.in	2026-01-19 12:20:53.238986800 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libmdbx
+Description: One of the fastest embeddable key-value ACID database
+Version: @MDBX_VERSION@
+Libs: -L${libdir} -lmdbx
+Cflags: -I${includedir}
--- origsrc/libmdbx-0.13.10/mdbx.c	2025-12-17 23:57:38.000000000 +0900
+++ src/libmdbx-0.13.10/mdbx.c	2026-01-19 12:20:53.253353000 +0900
@@ -193,6 +193,10 @@
 #include <string.h>
 #include <time.h>
 
+#if defined(__CYGWIN__)
+#include <sys/vfs.h>
+#endif
+
 /*----------------------------------------------------------------------------*/
 /* feature testing */
 
@@ -1196,7 +1200,7 @@ typedef struct osal_mmap {
 } osal_mmap_t;
 
 #ifndef MDBX_HAVE_PWRITEV
-#if defined(_WIN32) || defined(_WIN64)
+#if defined(_WIN32) || defined(_WIN64) || defined(__CYGWIN__)
 
 #define MDBX_HAVE_PWRITEV 0
 
@@ -27908,6 +27912,10 @@ __extern_C void __assert(const char *fun
     ;
 #define __assert_fail(assertion, file, line, function) __assert(function, file, line, assertion)
 
+#elif defined(__CYGWIN__)
+#define __assert_fail(assertion, file, line, function)                         \
+  __assert_func(file, line, function, assertion)
+
 #endif /* __assert_fail */
 
 __cold void mdbx_assert_fail(const MDBX_env *env, const char *msg, const char *func, unsigned line) {
