--- origsrc/urlparse/Makefile.am	2026-01-27 12:05:03.000000000 +0900
+++ src/urlparse/Makefile.am	2026-01-27 12:27:57.628338000 +0900
@@ -66,6 +66,9 @@ test_SOURCES = urlparse_test_main.c urlp
 test_CFLAGS = $(WARNCFLAGS) -I$(top_srcdir)
 test_CPPFLAGS = -I$(srcdir) -I$(srcdir)/munit
 test_LDADD = $(top_builddir)/liburlparse.la $(top_builddir)/libhttp-parser.la
-test_LDFLAGS = -static
+test_LDFLAGS =
 
 TESTS = test
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/urlparse/configure.ac	2026-01-27 12:05:03.000000000 +0900
+++ src/urlparse/configure.ac	2026-01-27 12:27:57.646536700 +0900
@@ -25,7 +25,7 @@
 # OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 # WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 AC_PREREQ(2.61)
-AC_INIT([urlparse], [0.0.0-DEV], [t-tujikawa@users.sourceforge.net])
+AC_INIT([urlparse], [0.0.0], [t-tujikawa@users.sourceforge.net])
 AC_CONFIG_AUX_DIR([.])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADERS([config.h])
@@ -157,6 +157,7 @@ AC_CONFIG_FILES([
   doc/Makefile
   doc/source/conf.py
 ])
+AC_CONFIG_FILES([urlparse.pc])
 AC_OUTPUT
 
 AC_MSG_NOTICE([summary of build options:
--- origsrc/urlparse/urlparse.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/urlparse/urlparse.pc.in	2026-01-27 12:28:06.494656600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: urlparse
+Description: URL parser compatible to url_parser_parse_url() from nodejs/http-parser
+Version: @VERSION@
+Libs: -L${libdir} -lurlparse
+Cflags: -I${includedir}
