--- origsrc/slapt-get-0.11.11/meson.build	2025-01-05 04:30:04.000000000 +0900
+++ src/slapt-get-0.11.11/meson.build	2025-04-04 09:16:04.505051700 +0900
@@ -20,7 +20,7 @@ prefix = get_option('prefix')
 localedir = join_paths(prefix, get_option('localedir'))
 datadir = join_paths(prefix, get_option('datadir'))
 pkgdatadir = join_paths(datadir, package)
-docdatadir = join_paths(datadir, 'doc', package + '-' + version)
+docdatadir = join_paths(datadir, 'doc', package)
 sbindir = join_paths(get_option('prefix'), get_option('sbindir'))
 rcdir = join_paths(get_option('sysconfdir'), 'slapt-get')
 rcpath = join_paths(rcdir, 'slapt-getrc')
--- origsrc/slapt-get-0.11.11/src/meson.build	2025-01-05 04:30:04.000000000 +0900
+++ src/slapt-get-0.11.11/src/meson.build	2025-04-04 09:14:11.485773400 +0900
@@ -33,7 +33,7 @@ slapt_h = custom_target('slapt.h',
   install_dir: get_option('includedir')
 )
 
-libslapt = library('libslapt', lib_sources, dependencies : deps, install: true, version: version, soversion: 0, name_prefix: '')
+libslapt = library('slapt', lib_sources, dependencies : deps, install: true, version: version, soversion: 0)
 
 # slapt_get = executable('slapt-get', main_sources + lib_sources, dependencies : deps, install: true, install_dir: get_option('sbindir'))
 slapt_get = executable('slapt-get', main_sources, link_with: libslapt, dependencies : deps, install: true, install_dir: get_option('sbindir'))
