--- origsrc/prng-3.0.2/src/Makefile.am	2012-05-03 19:41:41.437248700 +0900
+++ src/prng-3.0.2/src/Makefile.am	2012-05-03 19:41:43.188034100 +0900
@@ -22,6 +22,7 @@ libprng_la_SOURCES = \
  anti.c \
  support.c 
 
+libprng_la_LDFLAGS = -no-undefined
 
 include_HEADERS = \
  prng.h
--- origsrc/prng-3.0.2/tests/Makefile.am	2003-07-30 23:31:26.000000000 +0900
+++ src/prng-3.0.2/tests/Makefile.am	2012-05-03 19:41:43.209127900 +0900
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 ## $Id: Makefile.am,v 1.4 2001/02/27 12:22:09 leydold Exp $
 
-LDADD = $(top_builddir)/src/libprng.a -lm
+LDADD = $(top_builddir)/src/libprng.la -lm
 
 INCLUDES = \
  -I$(top_builddir) \
