--- origsrc/prng-3.0.2/configure.ac	2014-11-08 12:08:11.910187500 +0900
+++ src/prng-3.0.2/configure.ac	2014-11-08 12:09:59.789093700 +0900
@@ -20,10 +20,9 @@ AC_PROG_CC
 AC_PROG_CPP
 AC_CHECK_TOOL(RANLIB, ranlib, :)
 AC_CHECK_TOOL(AR, ar, :)
-AC_PROG_RANLIB
 AC_PROG_INSTALL
 AC_PROG_LN_S
-AC_PROG_LIBTOOL
+LT_INIT
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
--- origsrc/prng-3.0.2/doc/prng.texi	2001-03-13 01:04:49.000000000 +0900
+++ src/prng-3.0.2/doc/prng.texi	2014-11-08 12:21:11.598664000 +0900
@@ -90,7 +90,7 @@ Please let us also know about your exper
 
 @c ....................................................................
 
-@unnumberedsec Features
+@heading Features
 
 @itemize @bullet
 
@@ -231,14 +231,14 @@ you made, so that I can include them in
  
 @c ..............................................
 
-@unnumberedsec Documentation
+@heading Documentation
 
 A manual can be found in directory @file{doc} in various formats, including
 PS, PDF, HTML, Info and plain text.
 
 @c ..............................................
 
-@unnumberedsec Profiling and Verification
+@heading Profiling and Verification
 
 Do
 
--- origsrc/prng-3.0.2/src/Makefile.am	2014-11-08 12:08:11.925812500 +0900
+++ src/prng-3.0.2/src/Makefile.am	2014-11-08 12:08:13.160187500 +0900
@@ -22,6 +22,7 @@ libprng_la_SOURCES = \
  anti.c \
  support.c 
 
+libprng_la_LDFLAGS = -no-undefined
 
 include_HEADERS = \
  prng.h
--- origsrc/prng-3.0.2/tests/Makefile.am	2003-07-30 23:31:26.000000000 +0900
+++ src/prng-3.0.2/tests/Makefile.am	2014-11-08 12:08:13.160187500 +0900
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 ## $Id: Makefile.am,v 1.4 2001/02/27 12:22:09 leydold Exp $
 
-LDADD = $(top_builddir)/src/libprng.a -lm
+LDADD = $(top_builddir)/src/libprng.la -lm
 
 INCLUDES = \
  -I$(top_builddir) \
