--- origsrc/libjson-rpc-cpp-1.1.0/CMakeLists.txt	2018-01-04 23:19:57.000000000 +0900
+++ src/libjson-rpc-cpp-1.1.0/CMakeLists.txt	2018-01-07 20:10:00.221999800 +0900
@@ -22,7 +22,7 @@ if (${CMAKE_MAJOR_VERSION} GREATER 2)
 endif()
 
 set(MAJOR_VERSION 1)
-set(MINOR_VERSION 0)
+set(MINOR_VERSION 1)
 set(PATCH_VERSION 0)
 set(SO_VERSION    1)
 
@@ -134,7 +134,11 @@ add_custom_target(
 set(package_name "libjson-rpc-cpp")
 include(CMakePackageConfigHelpers)
 include(GNUInstallDirs)
-set(config_install_destination ${CMAKE_INSTALL_LIBDIR}/${package_name}/cmake)
+if (CYGWIN)
+    set(config_install_destination ${CMAKE_INSTALL_LIBDIR}/cmake/${package_name})
+else()
+    set(config_install_destination ${CMAKE_INSTALL_LIBDIR}/${package_name}/cmake)
+endif()
 configure_package_config_file(
 	cmake/${package_name}Config.cmake.in ${package_name}Config.cmake
 	INSTALL_DESTINATION ${config_install_destination}
--- origsrc/libjson-rpc-cpp-1.1.0/src/jsonrpccpp/server/connectors/httpserver.h	2018-01-04 23:19:57.000000000 +0900
+++ src/libjson-rpc-cpp-1.1.0/src/jsonrpccpp/server/connectors/httpserver.h	2018-01-07 19:52:02.592000000 +0900
@@ -25,6 +25,10 @@ typedef intptr_t ssize_t;
 #include <sys/socket.h>
 #endif
 
+#if defined(__CYGWIN__)
+#include <sys/select.h> 
+#endif
+
 #include <map>
 #include <microhttpd.h>
 #include "../abstractserverconnector.h"
--- origsrc/libjson-rpc-cpp-1.1.0/src/test/test_stubgenerator.cpp	2018-01-04 23:19:57.000000000 +0900
+++ src/libjson-rpc-cpp-1.1.0/src/test/test_stubgenerator.cpp	2018-01-07 20:07:49.141000000 +0900
@@ -25,18 +25,18 @@ using namespace std;
 
 namespace teststubgen {
 struct F {
-  FILE *stdout;
-  FILE *stderr;
+  FILE *_stdout;
+  FILE *_stderr;
   vector<StubGenerator *> stubgens;
   vector<Procedure> procedures;
   F() {
-    stdout = fopen("stdout.txt", "w");
-    stderr = fopen("stderr.txt", "w");
+    _stdout = fopen("stdout.txt", "w");
+    _stderr = fopen("stderr.txt", "w");
   }
 
   ~F() {
-    fclose(stdout);
-    fclose(stderr);
+    fclose(_stdout);
+    fclose(_stderr);
   }
 };
 } // namespace teststubgen
