--- origsrc/libjson-rpc-cpp-1.2.0/CMakeLists.txt	2019-03-29 10:25:29.000000000 +0900
+++ src/libjson-rpc-cpp-1.2.0/CMakeLists.txt	2019-04-25 05:57:28.322267400 +0900
@@ -134,7 +134,11 @@ add_custom_target(
 # CMake package
 set(package_name "libjson-rpc-cpp")
 include(CMakePackageConfigHelpers)
-set(config_install_destination ${CMAKE_INSTALL_LIBDIR}/${package_name}/cmake)
+if (CYGWIN)
+    set(config_install_destination ${CMAKE_INSTALL_LIBDIR}/cmake/${package_name})
+else()
+    set(config_install_destination ${CMAKE_INSTALL_LIBDIR}/${package_name}/cmake)
+endif()
 configure_package_config_file(
 	cmake/${package_name}Config.cmake.in ${package_name}Config.cmake
 	INSTALL_DESTINATION ${config_install_destination}
--- origsrc/libjson-rpc-cpp-1.2.0/src/test/test_stubgenerator.cpp	2019-03-29 10:25:29.000000000 +0900
+++ src/libjson-rpc-cpp-1.2.0/src/test/test_stubgenerator.cpp	2019-04-25 05:53:40.196204500 +0900
@@ -20,6 +20,13 @@
 
 #include <sstream>
 
+#if defined(stdout)
+#undef stdout
+#endif
+#if defined(stderr)
+#undef stderr
+#endif
+
 using namespace jsonrpc;
 using namespace std;
 
