--- origsrc/libjson-rpc-cpp-0.2.1/src/jsonrpc/CMakeLists.txt	2013-07-27 23:09:18.000000000 +0900
+++ src/libjson-rpc-cpp-0.2.1/src/jsonrpc/CMakeLists.txt	2013-10-13 17:43:13.006215200 +0900
@@ -4,22 +4,18 @@ file(GLOB connector_header connectors/*.
 file(GLOB jsoncpp_header json/*.h)
 
 add_library(jsonrpc SHARED ${jsonrpc_source})
-add_library(jsonrpcStatic STATIC ${jsonrpc_source})
-
-set_target_properties(jsonrpcStatic PROPERTIES OUTPUT_NAME jsonrpc)
 
 set(VERSION_STRING ${MAJOR_VERSION}.${MINOR_VERSION}.${PATCH_VERSION}) 
-set_target_properties(jsonrpc jsonrpcStatic PROPERTIES VERSION "${VERSION_STRING}" SOVERSION "${VERSION_MAJOR}")
+set_target_properties(jsonrpc PROPERTIES VERSION "${VERSION_STRING}" SOVERSION "${MAJOR_VERSION}")
 
 
 target_link_libraries(jsonrpc ${CURL_LIBRARIES})
-target_link_libraries(jsonrpcStatic ${CURL_LIBRARIES})
 
 install(FILES ${jsonrpc_header} DESTINATION include/jsonrpc) 
 install(FILES ${connector_header} DESTINATION include/jsonrpc/connectors)
 install(FILES ${jsoncpp_header} DESTINATION include/jsonrpc/json) 
 
-install(TARGETS jsonrpc jsonrpcStatic LIBRARY DESTINATION lib
+install(TARGETS jsonrpc LIBRARY DESTINATION lib
                       ARCHIVE DESTINATION lib
                       RUNTIME DESTINATION bin)
 
