--- origsrc/SKA-1.0/Makefile	2018-10-25 23:24:05.000000000 +0900
+++ src/SKA-1.0/Makefile	2025-06-10 21:06:54.970218600 +0900
@@ -1,5 +1,5 @@
-CXX = g++
-CXXFLAGS = -g -std=c++0x -O2
+CXX ?= g++
+CXXFLAGS ?=
 
 RM = rm -f
 MKDIR = mkdir
--- origsrc/SKA-1.0/tests/Makefile	2018-10-25 23:24:05.000000000 +0900
+++ src/SKA-1.0/tests/Makefile	2025-06-10 21:09:03.454620000 +0900
@@ -22,15 +22,16 @@ USER_TEST_DIR = .
 # Where to find user code.
 USER_DIR = ../src/
 
-CXX = g++ -std=c++0x
+CXX ?= g++ 
+CXXFLAGS ?=
 
 # Flags passed to the preprocessor.
 # Set Google Test's header directory as a system directory, such that
 # the compiler doesn't generate warnings in Google Test headers.
-CPPFLAGS += -isystem $(GTEST_DIR)/include
+CPPFLAGS += -isystem $(GTEST_DIR)/include -D_GNU_SOURCE
 
 # Flags passed to the C++ compiler.
-CXXFLAGS += -g -Wall -Wextra -pthread
+CXXFLAGS += -pthread
 
 # All tests produced by this Makefile.  Remember to add new tests you
 # created to the list.
@@ -78,34 +79,34 @@ gtest_main.a : gtest-all.o gtest_main.o
 # gtest_main.a, depending on whether it defines its own main()
 # function.
 
-general.o : $(USER_DIR)/general.cpp $(USER_DIR)/general.hpp $(GTEST_HEADERS)
+general.o : $(USER_DIR)/general.cpp $(USER_DIR)/general.hpp
 	 $(CXX) $(CPPFLAGS) $(CXXFLAGS) -c $(USER_DIR)/general.cpp
 
 general_unittest.o : $(USER_TEST_DIR)/general_unittest.cpp \
                      $(USER_DIR)/general.hpp $(GTEST_HEADERS)
 	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c $(USER_TEST_DIR)/general_unittest.cpp
 
-general_unittest : general.o general_unittest.cpp gtest_main.a 
-	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -lpthread $^ -o $@ -lz
+general_unittest : general.o general_unittest.cpp
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -lpthread $^ -o $@ -lz -lgtest -lgtest_main
 
 
-DNA.o : $(USER_DIR)/DNA.cpp $(USER_DIR)/DNA.hpp $(GTEST_HEADERS)
+DNA.o : $(USER_DIR)/DNA.cpp $(USER_DIR)/DNA.hpp
 	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c $(USER_DIR)/DNA.cpp
 
 DNA_unittest.o : $(USER_TEST_DIR)/DNA_unittest.cpp \
                      $(USER_DIR)/DNA.hpp $(GTEST_HEADERS)
 	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c $(USER_TEST_DIR)/DNA_unittest.cpp
 
-DNA_unittest : DNA.o DNA_unittest.cpp gtest_main.a
-	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -lpthread $^ -o $@  -lz
+DNA_unittest : DNA.o DNA_unittest.cpp
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -lpthread $^ -o $@  -lz -lgtest -lgtest_main
 
 
-kmers.o : $(USER_DIR)/kmers.cpp $(USER_DIR)/kmers.hpp $(GTEST_HEADERS)
+kmers.o : $(USER_DIR)/kmers.cpp $(USER_DIR)/kmers.hpp
 	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c $(USER_DIR)/kmers.cpp
 
 kmers_unittest.o : $(USER_TEST_DIR)/kmers_unittest.cpp \
                      $(USER_DIR)/kmers.hpp $(GTEST_HEADERS)
 	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c $(USER_TEST_DIR)/kmers_unittest.cpp
 
-kmers_unittest : kmers.o DNA.o general.o kmers_unittest.cpp gtest_main.a
-	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -lpthread $^ -o $@ -lz
+kmers_unittest : kmers.o DNA.o general.o kmers_unittest.cpp
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -lpthread $^ -o $@ -lz -lgtest -lgtest_main
