--- origsrc/gwcalc-0.3.1/src/gui.c	2010-06-05 14:07:26.000000000 +0900
+++ src/gwcalc-0.3.1/src/gui.c	2025-12-05 10:05:29.713913900 +0900
@@ -70,7 +70,7 @@ error_dialog(char *ttl, char *msg)
                 GTK_BUTTONS_OK,  
                 ttl, NULL)); 
   gtk_window_set_position(GTK_WINDOW(dlg), GTK_WIN_POS_CENTER);
-  gtk_message_dialog_format_secondary_text(dlg, msg);
+  gtk_message_dialog_format_secondary_text(dlg, "%s", msg);
   gtk_dialog_run(GTK_DIALOG(dlg));
   gtk_widget_destroy(GTK_WIDGET(dlg));
 }
