--- origsrc/libnk2-20150927/Makefile.am	2015-09-28 00:29:45.000000000 +0900
+++ src/libnk2-20150927/Makefile.am	2016-04-07 19:40:54.734949800 +0900
@@ -3,27 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfdatetime \
-	libfguid \
-	libfvalue \
-	libfwnt \
-	libfmapi \
 	libnk2 \
-	libcsystem \
 	nk2tools \
 	pynk2 \
-	po \
 	manuals \
 	tests \
 	msvscpp
@@ -127,4 +109,3 @@ splint:
 	(cd $(srcdir)/pynk2 && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
-
--- origsrc/libnk2-20150927/configure.ac	2015-09-28 00:29:45.000000000 +0900
+++ src/libnk2-20150927/configure.ac	2016-04-07 19:40:43.351134500 +0900
@@ -193,24 +193,7 @@ dnl Have configure make the Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
-AC_CONFIG_FILES([libfwnt/Makefile])
-AC_CONFIG_FILES([libfmapi/Makefile])
 AC_CONFIG_FILES([libnk2/Makefile])
-AC_CONFIG_FILES([libcsystem/Makefile])
 AC_CONFIG_FILES([nk2tools/Makefile])
 AC_CONFIG_FILES([pynk2/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
@@ -266,5 +249,3 @@ Features:
    Verbose output:                           $ac_cv_enable_verbose_output
    Debug output:                             $ac_cv_enable_debug_output
 ]);
-
-
--- origsrc/libnk2-20150927/libnk2/Makefile.am	2015-09-28 00:29:45.000000000 +0900
+++ src/libnk2-20150927/libnk2/Makefile.am	2016-04-07 19:41:07.068494900 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -85,4 +86,3 @@ distclean: clean
 splint:
 	@echo "Running splint on libnk2 ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(libnk2_la_SOURCES)
-
--- origsrc/libnk2-20150927/nk2tools/Makefile.am	2015-09-28 00:29:45.000000000 +0900
+++ src/libnk2-20150927/nk2tools/Makefile.am	2016-04-07 19:41:18.191448700 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -107,4 +108,3 @@ splint:
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(nk2export_SOURCES)
 	@echo "Running splint on nk2info ..."
 	-splint -preproc -redef $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(nk2info_SOURCES)
-
--- origsrc/libnk2-20150927/tests/Makefile.am	2015-09-28 00:29:45.000000000 +0900
+++ src/libnk2-20150927/tests/Makefile.am	2016-04-07 19:41:24.829605500 +0900
@@ -1,4 +1,5 @@
 AM_CPPFLAGS = \
+	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/common \
 	@LIBCSTRING_CPPFLAGS@ \
@@ -70,4 +71,3 @@ MAINTAINERCLEANFILES = \
 
 distclean: clean
 	/bin/rm -f Makefile
-
