--- origsrc/libnk2-20240426/Makefile.am	2024-04-26 12:42:55.000000000 +0900
+++ src/libnk2-20240426/Makefile.am	2025-05-18 15:10:56.277733600 +0900
@@ -3,25 +3,9 @@ ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfdatetime \
-	libfguid \
-	libfwnt \
-	libfmapi \
 	libnk2 \
-	libfvalue \
 	nk2tools \
 	pynk2 \
-	po \
 	manuals \
 	tests \
 	ossfuzz \
--- origsrc/libnk2-20240426/configure.ac	2024-04-26 12:42:55.000000000 +0900
+++ src/libnk2-20240426/configure.ac	2025-05-18 15:10:56.277733600 +0900
@@ -173,23 +173,8 @@ dnl Generate Makefiles
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfwnt/Makefile])
-AC_CONFIG_FILES([libfmapi/Makefile])
 AC_CONFIG_FILES([libnk2/Makefile])
 AC_CONFIG_FILES([pynk2/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
 AC_CONFIG_FILES([nk2tools/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
--- origsrc/libnk2-20240426/nk2tools/export_handle.c	2024-04-26 12:42:55.000000000 +0900
+++ src/libnk2-20240426/nk2tools/export_handle.c	2025-05-18 15:10:56.277733600 +0900
@@ -1574,12 +1574,14 @@ int export_handle_export_alias(
 		     log_handle,
 		     error ) != 1 )
 		{
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
 				 "%s: unable to export item values.\n",
 				 function );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
@@ -1740,6 +1742,7 @@ int export_handle_export_items(
 			 item_index + 1,
 			 number_of_items );
 
+#if defined( HAVE_DEBUG_OUTPUT )
 			if( libcnotify_verbose != 0 )
 			{
 				libcnotify_printf(
@@ -1747,6 +1750,7 @@ int export_handle_export_items(
 				 function,
 				 item_index + 1 );
 			}
+#endif
 			if( ( error != NULL )
 			 && ( *error != NULL ) )
 			{
--- origsrc/libnk2-20240426/pynk2/Makefile.am	2024-04-26 12:42:55.000000000 +0900
+++ src/libnk2-20240426/pynk2/Makefile.am	2025-05-18 15:11:29.557531000 +0900
@@ -49,7 +49,7 @@ pynk2_la_LIBADD = \
 	@LIBFGUID_LIBADD@
 
 pynk2_la_CPPFLAGS = $(PYTHON_CPPFLAGS)
-pynk2_la_LDFLAGS  = -module -avoid-version $(PYTHON_LDFLAGS)
+pynk2_la_LDFLAGS  = -no-undefined -module -avoid-version $(PYTHON_LDFLAGS)
 
 endif
 
