--- origsrc/SPR-3.3.2/src/Makefile.am	2009-04-16 05:03:46.000000000 +0900
+++ src/SPR-3.3.2/src/Makefile.am	2015-03-20 10:09:51.190179700 +0900
@@ -3,7 +3,7 @@ lib_LTLIBRARIES = libSPR.la
 AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)
 AM_LDFLAGS  = -L/usr/lib -L$(top_srcdir)/src
 AM_LIBS     = -lSPR 
-APPLDADD    = -lSPR
+APPLDADD    = libSPR.la
 if WITH_ROOT
    AM_CXXFLAGS += $(ROOTCXXFLAGS) -O2
    AM_CPPFLAGS += $(ROOTCPPFLAGS) -DSPRROOTTUPLE
@@ -14,7 +14,7 @@ else
    AM_CXXFLAGS += -O4
 endif
 
-libSPR_la_LIBADD = $(top_srcdir)/math/libSPRmath.la
+libSPR_la_LIBADD = $(top_builddir)/math/libSPRmath.la
 libSPR_la_SOURCES  = \
   SprDataMoments.cc                 SprDecisionTree.cc \
   SprFisher.cc                      SprFomCalculator.cc \
@@ -55,7 +55,7 @@ if WITH_ROOT
    libSPR_la_SOURCES  += SprRootReader.cc SprRootWriter.cc SprAdapterDict.C SprRootAdapter.cc
 endif
 
-
+libSPR_la_LDFLAGS = -no-undefined
 
 #####################################################################
 
