--- origsrc/xmlroff-0.6.2/xmlroff/Makefile.am	2009-03-12 06:08:57.000000000 +0900
+++ src/xmlroff-0.6.2/xmlroff/Makefile.am	2014-12-05 14:12:06.890062500 +0900
@@ -8,9 +8,7 @@ endif
 
 SUBDIRS = libfo xmlroff . examples docs $(CUNIT_DIRS)
 
-if PLATFORM_WIN32
 no_undefined = -no-undefined
-endif
 
 INCLUDES = \
 	-DG_LOG_DOMAIN=\"libfo\"	\
--- origsrc/xmlroff-0.6.2/xmlroff/cunit/Makefile.am	2009-03-12 06:08:57.000000000 +0900
+++ src/xmlroff-0.6.2/xmlroff/cunit/Makefile.am	2014-12-05 14:13:11.141039000 +0900
@@ -4,10 +4,12 @@
 INCLUDES = \
 	-DG_LOG_DOMAIN=\"libfo\" \
 	$(CUNIT_CFLAGS)	\
-	$(PANGO_CFLAGS)
+	$(PANGO_CFLAGS) \
+	-I$(top_srcdir)
 
 if ENABLE_CUNIT
-  noinst_PROGRAMS = test
+  check_PROGRAMS = test
+  TESTS = test
 endif
 
 test_SOURCES = \
@@ -29,9 +31,7 @@ test_SOURCES = \
 	test-libfo-version.c	\
 	test-libfo-version.h
 
-test_LDFLAGS = -static
-
 test_LDADD = \
 	$(CUNIT_LIBS)		\
 	$(PANGO_LIBS)		\
-	../libfo/libfo-0.6.la
\ No newline at end of file
+	../libfo/libfo-0.6.la
--- origsrc/xmlroff-0.6.2/xmlroff/docs/Makefile.am	2009-03-12 06:08:57.000000000 +0900
+++ src/xmlroff-0.6.2/xmlroff/docs/Makefile.am	2014-12-08 08:49:43.356897600 +0900
@@ -227,9 +227,9 @@ dist-hook-local:
 # Location of DocBook XSL stylesheet.
 DOCBOOK_XSL=http://docbook.sourceforge.net/release/xsl/current/fo/docbook.xsl
 
-xmlroff.fo:	xmlroff.xml ../libfo-compat.xsl
-	xsltproc $(DOCBOOK_XSL) xmlroff.xml > xmlroff.tmp
-	xsltproc ../libfo-compat.xsl xmlroff.tmp > xmlroff.fo
+xmlroff.fo:	$(srcdir)/xmlroff.xml $(top_srcdir)/libfo-compat.xsl
+	xsltproc $(DOCBOOK_XSL) $(srcdir)/xmlroff.xml > xmlroff.tmp
+	xsltproc $(top_srcdir)/libfo-compat.xsl xmlroff.tmp > xmlroff.fo
 	rm -f xmlroff.tmp
 
 ####################################
--- origsrc/xmlroff-0.6.2/xmlroff/libfo/Makefile.am	2009-03-12 06:08:57.000000000 +0900
+++ src/xmlroff-0.6.2/xmlroff/libfo/Makefile.am	2014-12-08 08:46:28.971155400 +0900
@@ -6,7 +6,8 @@ INCLUDES = \
 	-DG_LOG_DOMAIN=\"libfo\" \
 	$(LIBXSLT_CFLAGS)	\
 	$(LIBRSVG_CFLAGS)	\
-	$(PANGO_CFLAGS)
+	$(PANGO_CFLAGS)		\
+	-I$(top_srcdir)
 
 LDADDS = @STRIP_BEGIN@	\
 	@GLIB_LIBS@	\
@@ -76,7 +77,7 @@ libfo_libfo_la_SOURCES =	\
 	fo-xml-node.c		\
 	fo-xslt-transformer.c	\
 	fo-xsl-formatter.c	\
-	libfo-compat.c		\
+	$(top_builddir)/libfo/libfo-compat.c		\
 	libfo-pango.c		\
 	libfo-pango.h
 
@@ -103,7 +104,7 @@ fo_base_cfiles = \
 
 libfo_0_6_la_LDFLAGS = \
 	-version-info $(LT_VERSION_INFO)	\
-	$(no_undefined)
+	-no-undefined
 
 libfo_0_6_includedir = $(includedir)/libfo-0.6/libfo
 
@@ -141,10 +142,10 @@ libfo_0_6_la_LIBADD = \
 libfo-features.h:
 	cd $(top_builddir) && ./config.status libfo/libfo-features.h
 
-libfo-compat.c: s-libfo-compat-c
+$(top_builddir)/libfo/libfo-compat.c: s-libfo-compat-c
 	@true
-s-libfo-compat-c: ../libfo-compat.xsl libfo-compat.c.tmpl ../tools/insert-file-as-string.pl
-	$(PERL) ../tools/insert-file-as-string.pl --file ../libfo-compat.xsl libfo-compat.c.tmpl > libfo-compat.c \
+s-libfo-compat-c: $(top_srcdir)/libfo-compat.xsl $(srcdir)/libfo-compat.c.tmpl $(top_srcdir)/tools/insert-file-as-string.pl
+	$(PERL) $(top_srcdir)/tools/insert-file-as-string.pl --file $(top_srcdir)/libfo-compat.xsl $(srcdir)/libfo-compat.c.tmpl > $(top_builddir)/libfo/libfo-compat.c \
 	&& echo timestamp > $(@F)
 
 EXTRA_DIST = \
--- origsrc/xmlroff-0.6.2/xmlroff/libfo/area/Makefile.am	2009-03-12 06:08:57.000000000 +0900
+++ src/xmlroff-0.6.2/xmlroff/libfo/area/Makefile.am	2014-12-08 08:04:23.343560800 +0900
@@ -3,7 +3,7 @@
 INCLUDES = \
 	-DG_LOG_DOMAIN=\"libfo\" \
 	$(PANGO_CFLAGS)	\
-	-I..
+	-I.. -I$(top_srcdir) -I$(top_srcdir)/libfo
 
 LDADDS = @STRIP_BEGIN@	\
 	@GLIB_LIBS@	\
--- origsrc/xmlroff-0.6.2/xmlroff/libfo/datatype/Makefile.am	2009-03-12 06:08:57.000000000 +0900
+++ src/xmlroff-0.6.2/xmlroff/libfo/datatype/Makefile.am	2014-12-08 08:04:38.759576400 +0900
@@ -3,7 +3,7 @@
 INCLUDES = \
 	-DG_LOG_DOMAIN=\"libfo\" \
 	$(GLIB_CFLAGS)		\
-	-I..
+	-I.. -I$(top_srcdir) -I$(top_srcdir)/libfo
 
 LDADDS = @STRIP_BEGIN@	\
 	@GLIB_LIBS@	\
--- origsrc/xmlroff-0.6.2/xmlroff/libfo/expr/Makefile.am	2009-03-12 06:08:57.000000000 +0900
+++ src/xmlroff-0.6.2/xmlroff/libfo/expr/Makefile.am	2014-12-08 08:04:48.044732700 +0900
@@ -3,7 +3,7 @@
 INCLUDES = \
 	-DG_LOG_DOMAIN=\"libfo\" \
 	$(GLIB_CFLAGS)	\
-	-I..
+	-I.. -I$(top_srcdir) -I$(top_srcdir)/libfo
 
 LDADDS = @STRIP_BEGIN@	\
 	@GLIB_LIBS@	\
--- origsrc/xmlroff-0.6.2/xmlroff/libfo/fo/Makefile.am	2009-03-12 06:08:57.000000000 +0900
+++ src/xmlroff-0.6.2/xmlroff/libfo/fo/Makefile.am	2014-12-08 08:05:03.313287300 +0900
@@ -5,7 +5,7 @@ INCLUDES = \
 	-DFO_DISABLE_DEPRECATED		\
 	$(LIBXSLT_CFLAGS)	\
 	$(PANGO_CFLAGS)	\
-	-I..
+	-I.. -I$(top_srcdir) -I$(top_srcdir)/libfo
 
 LDADDS = @STRIP_BEGIN@	\
 	@GLIB_LIBS@	\
--- origsrc/xmlroff-0.6.2/xmlroff/libfo/fo-libfo-basic.h	2009-03-12 06:08:57.000000000 +0900
+++ src/xmlroff-0.6.2/xmlroff/libfo/fo-libfo-basic.h	2014-12-08 08:38:55.728967900 +0900
@@ -11,8 +11,7 @@
 #define __FO_LIBFO_BASIC_H__
 
 #include <stdio.h>
-#include <glib/gtypes.h>
-#include <glib/gerror.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
--- origsrc/xmlroff-0.6.2/xmlroff/libfo/property/Makefile.am	2009-03-12 06:08:57.000000000 +0900
+++ src/xmlroff-0.6.2/xmlroff/libfo/property/Makefile.am	2014-12-08 08:05:12.912896700 +0900
@@ -3,7 +3,7 @@
 INCLUDES = \
 	-DG_LOG_DOMAIN=\"libfo\" \
 	$(GLIB_CFLAGS)	\
-	-I..
+	-I.. -I$(top_srcdir) -I$(top_srcdir)/libfo
 
 LDADDS = @STRIP_BEGIN@	\
 	@GLIB_LIBS@	\
--- origsrc/xmlroff-0.6.2/xmlroff/libfo/util/Makefile.am	2009-03-12 06:08:57.000000000 +0900
+++ src/xmlroff-0.6.2/xmlroff/libfo/util/Makefile.am	2014-12-08 08:05:20.934381100 +0900
@@ -5,7 +5,7 @@ INCLUDES = \
 	$(GLIB_CFLAGS)		\
 	$(LIBXSLT_CFLAGS)	\
 	$(GDKPIXBUF_CFLAGS)	\
-	-I..
+	-I.. -I$(top_srcdir) -I$(top_srcdir)/libfo
 
 LDADDS = @STRIP_BEGIN@	\
 	@GLIB_LIBS@	\
--- origsrc/xmlroff-0.6.2/xmlroff/libfo-0.6.pc.in	2009-03-12 06:08:57.000000000 +0900
+++ src/xmlroff-0.6.2/xmlroff/libfo-0.6.pc.in	2014-12-05 14:11:37.186937500 +0900
@@ -10,7 +10,7 @@ compat_stylesheet=@prefix@/share/xml/lib
 Name: libfo
 Description: XSL formatter library for xmlroff
 Version: @VERSION@
-Requires: @GNOMEPRINT_REQUIRES@ glib-2.0 gobject-2.0 gmodule-2.0 libxslt gdk-pixbuf-2.0
+Requires: @GNOMEPRINT_REQUIRES@ glib-2.0 gobject-2.0 gmodule-2.0 libxml-2.0 gdk-pixbuf-2.0
 Libs: -L${libdir} -lfo-@LT_RELEASE@ @GNOMEPRINT_LIBS@
 Cflags: -I${pkgincludedir} @GNOMEPRINT_CFLAGS@
 
--- origsrc/xmlroff-0.6.2/xmlroff/xmlroff/Makefile.am	2009-03-12 06:08:57.000000000 +0900
+++ src/xmlroff-0.6.2/xmlroff/xmlroff/Makefile.am	2014-12-08 08:06:12.539849800 +0900
@@ -1,13 +1,12 @@
 # xmlroff Makefile.am
 
-if PLATFORM_WIN32
 no_undefined = -no-undefined
-endif
 
 INCLUDES = \
 	-DG_LOG_DOMAIN=\"libfo\"	\
 	$(LIBXSLT_CFLAGS)	\
-	$(GLIB_CFLAGS)
+	$(GLIB_CFLAGS)		\
+	-I$(top_srcdir)
 
 bin_PROGRAMS = \
 	xmlroff
@@ -18,10 +17,8 @@ bin_PROGRAMS = \
 xmlroff_SOURCES = \
 	xmlroff.c
 
-xmlroff_LDFLAGS = -static
-
 xmlroff_LDADD = \
-	../libfo/libfo-0.6.la \
+	$(top_builddir)/libfo/libfo-0.6.la \
 	$(PANGO_LIBS) \
 	$(GNOMEPRINT_LIBS) \
 	$(LIBXSLT_LIBS) \
