--- origsrc/gravit-0.5.1/Makefile.am	2014-03-06 08:02:28.000000000 +0900
+++ src/gravit-0.5.1/Makefile.am	2025-06-05 12:32:54.936079100 +0900
@@ -11,10 +11,10 @@ skybox2dir=@datadir@/gravit/data/skybox/
 spawndir=@datadir@/gravit/spawn
 
 sysconf_DATA = cfg/gravit.cfg cfg/screensaver.cfg
-misc_DATA =$(shell echo data/*.??*)
-skybox1_DATA =$(shell echo data/skybox/*.??*)
-skybox2_DATA =$(shell echo data/skybox/purplenebula/*.??*)
-spawn_DATA =$(shell echo spawn/*)
+misc_DATA =$(shell echo $(srcdir)/data/*.??*)
+skybox1_DATA =$(shell echo $(srcdir)/data/skybox/*.??*)
+skybox2_DATA =$(shell echo $(srcdir)/data/skybox/purplenebula/*.??*)
+spawn_DATA =$(shell echo $(srcdir)/spawn/*)
 
 bin_PROGRAMS=gravit
 gravit_SOURCES=src/color.c src/command.c src/command.h src/config.c src/console.c src/font.c src/font.h src/fps.c src/frame-ot.c src/frame-pp.c src/frame-pp_sse.c src/frame.c src/gfx.c src/gravit.h src/input.c src/main.c src/osd.c src/sdlk.h src/spawn.c src/texture.c src/timer.c src/tool.c src/png_save.c
--- origsrc/gravit-0.5.1/configure.in	2014-03-06 08:02:28.000000000 +0900
+++ src/gravit-0.5.1/configure.in	2025-06-05 12:28:29.609422000 +0900
@@ -66,7 +66,6 @@ AC_CHECK_HEADERS([omp.h],[
 
 # check for supported SIMD extensions (gcc specific)
 # if SSE is supported, HAVE_SSE will be set
-AX_EXT_SSE
 CFLAGS="$CFLAGS $SIMD_FLAGS -I/usr/local/include"
 LDFLAGS="$SIMD_FLAGS $LDFLAGS -L/usr/local/lib"
 
--- origsrc/gravit-0.5.1/src/gravit.h	2014-03-06 08:02:28.000000000 +0900
+++ src/gravit-0.5.1/src/gravit.h	2025-06-05 12:29:59.706071800 +0900
@@ -666,8 +666,6 @@ typedef struct pib_s {
 
 } pib_t;
 
-// main.c
-#ifdef WIN32
 
 #ifndef NO_GUI
 extern video_t video;
@@ -675,15 +673,6 @@ extern video_t video;
 extern state_t state;
 extern view_t view;
 
-#else
-
-#ifndef NO_GUI
-video_t video;
-#endif
-state_t state;
-view_t view;
-
-#endif
 
 void cleanMemory();
 void runVideo();
