--- origsrc/gpp4-1.3.1/src/Makefile.am	2011-02-14 18:05:01.000000000 +0900
+++ src/gpp4-1.3.1/src/Makefile.am	2012-04-26 09:41:49.665815900 +0900
@@ -15,7 +15,7 @@ cmap_symop.c pack_c.c gpp4_open_symop_fi
 # Read http://sourceware.org/autobook/autobook/autobook_91.html 
 # before changing the version-info (read: don't change it)
 
-libgpp4_la_LDFLAGS = -version-info 0:0:0
+libgpp4_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 
 library_include_HEADERS = ccp4_array.h ccp4_unitcell.h cmap_stats.h	\
 ccp4_errno.h ccp4_utils.h cmtzlib.h ccp4_file_err.h ccp4_vars.h		\
--- origsrc/gpp4-1.3.1/src_f/Makefile.am	2009-09-17 20:48:18.000000000 +0900
+++ src/gpp4-1.3.1/src_f/Makefile.am	2012-04-26 09:42:12.650190900 +0900
@@ -23,7 +23,7 @@ libgpp4f_la_CFLAGS = $(AM_CFLAGS) @MMDB_
 # Read http://sourceware.org/autobook/autobook/autobook_91.html 
 # before changing the version-info (read: don't change it)
 
-libgpp4f_la_LDFLAGS = -version-info 0:0:0
+libgpp4f_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 
 unix.f: unix.m4
 	m4 < unix.m4 > unix.f
