--- origsrc/midgard-core-12.09.1/Makefile.am	2012-09-26 17:39:27.000000000 +0900
+++ src/midgard-core-12.09.1/Makefile.am	2015-03-18 21:17:07.378734400 +0900
@@ -139,7 +139,7 @@ $(PROGRAMS): $(LDADD)
 src_libmidgard2_la_CPPFLAGS = \
 	-I$(top_srcdir)/src -DG_LOG_DOMAIN=\"midgard-core\"
 
-AM_CFLAGS = -I$(top_srcdir)/src @MIDGARD_CFLAGS@
+AM_CFLAGS = -I$(top_builddir)/src -I$(top_srcdir)/src @MIDGARD_CFLAGS@
 AM_LDFLAGS = @MIDGARD_LIBS@ 
 
 src_libmidgard2_la_LDFLAGS = -version-info 21:0:9 $(no_undefined)
@@ -304,7 +304,7 @@ Midgard-12.09.gir: $(lib_LTLIBRARIES)
 Midgard_12_09_gir_INCLUDES = GObject-2.0 libxml2-2.0 $(GDA_GIR_INCLUDE)
 Midgard_12_09_gir_CFLAGS = $(AM_CFLAGS) `pkg-config --cflags libxml-2.0`
 Midgard_12_09_gir_LIBS = $(lib_LTLIBRARIES) 
-Midgard_12_09_gir_FILES = $(addprefix $(srcdir)/,$(introspection_sources))
+Midgard_12_09_gir_FILES = $(introspection_sources)
 Midgard_12_09_gir_PACKAGES = gobject-2.0 libxml-2.0 $(GDA_GIR_PACKAGE)
 INTROSPECTION_GIRS = Midgard-12.09.gir
 INTROSPECTION_LIBS += -lxml2
--- origsrc/midgard-core-12.09.1/configure.in	2012-09-26 17:39:27.000000000 +0900
+++ src/midgard-core-12.09.1/configure.in	2015-03-18 21:06:11.982250000 +0900
@@ -18,7 +18,7 @@ AC_CONFIG_SRCDIR(src/midgard.c)
 
 dnl Automake is needed for the build system.
 dnl
-AM_INIT_AUTOMAKE([-Wno-portability -Wall])
+AM_INIT_AUTOMAKE([-Wno-portability -Wall subdir-objects])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 AM_MAINTAINER_MODE
 GOBJECT_INTROSPECTION_CHECK([0.6.5])
@@ -56,7 +56,7 @@ if test "$_MIDGARD_DBUS_SUPPORT" == "no"
 	dbus_libs=""
 fi
 
-PKG_CHECK_MODULES(MIDGARD, glib-2.0 gobject-2.0 libxml-2.0 $gdamodule $dbus_libs)
+PKG_CHECK_MODULES(MIDGARD, glib-2.0 gobject-2.0 gio-2.0 libxml-2.0 $gdamodule $dbus_libs)
 if test "$gdamodule" == "libgda-4.0"; then
     MIDGARD_CFLAGS="${MIDGARD_CFLAGS} -DHAVE_LIBGDA_4"
 fi
