--- origsrc/io_lib-1.13.7/io_lib/Makefile.am	2014-04-09 01:49:28.000000000 +0900
+++ src/io_lib-1.13.7/io_lib/Makefile.am	2014-06-06 21:34:44.177368800 +0900
@@ -133,7 +133,7 @@ libstaden_read_la_SOURCES = \
 
 libstaden_read_la_CPPFLAGS = @LIBCURL_CPPFLAGS@
 
-libstaden_read_la_LDFLAGS = -version-info @VERS_CURRENT@:@VERS_REVISION@:@VERS_AGE@ 
+libstaden_read_la_LDFLAGS = -no-undefined -version-info @VERS_CURRENT@:@VERS_REVISION@:@VERS_AGE@ 
 libstaden_read_la_LIBADD = @LIBZ@ @LIBCURL@
 
 INCLUDES = -I${top_srcdir}
