--- origsrc/libhdcd-1.4/Makefile.am	2017-07-06 14:05:11.000000000 +0900
+++ src/libhdcd-1.4/Makefile.am	2025-03-16 20:18:16.073508500 +0900
@@ -34,7 +34,7 @@ libhdcd_la_SOURCES = src/hdcd_decode2.c
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libhdcd.pc
 
-libhdcd_la_LDFLAGS = -version-info @HDCD_LIBTOOL_VERSION@ -export-symbols-regex "^hdcd_"
+libhdcd_la_LDFLAGS = -no-undefined -version-info @HDCD_LIBTOOL_VERSION@ -export-symbols-regex "^hdcd_"
 
 LDADD = libhdcd.la
 
