--- origsrc/pialign-0.2.4/Makefile.am	2012-11-07 10:32:31.000000000 +0900
+++ src/pialign-0.2.4/Makefile.am	2015-04-21 10:50:30.200049500 +0900
@@ -17,3 +17,6 @@ script/reversealign.pl \
 script/time.pl \
 script/make-moses-config.pl \
 script/visualize.pl
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/pialign-0.2.4/configure.ac	2012-10-29 11:09:54.000000000 +0900
+++ src/pialign-0.2.4/configure.ac	2015-04-21 10:50:44.408993300 +0900
@@ -12,6 +12,7 @@ AC_CONFIG_FILES([
     src/lib/Makefile
     src/bin/Makefile
     src/test/Makefile
+    pialign.pc
 ])
 
 AM_INIT_AUTOMAKE([-Wall])
--- origsrc/pialign-0.2.4/pialign.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/pialign-0.2.4/pialign.pc.in	2015-04-21 10:45:35.203619700 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: pialign
+Description: Phrasal ITG Aligner
+Version: @VERSION@
+Libs: -L${libdir} -lpialign
+Cflags: -I${includedir}
--- origsrc/pialign-0.2.4/src/lib/Makefile.am	2012-09-02 10:04:48.000000000 +0900
+++ src/pialign-0.2.4/src/lib/Makefile.am	2015-04-21 10:38:39.255417400 +0900
@@ -6,4 +6,4 @@ AM_CPPFLAGS = -I$(srcdir)/../include
 lib_LTLIBRARIES = libpialign.la
 
 libpialign_la_SOURCES = ${PIACPP}
-libpialign_la_LDFLAGS = -version-info 0:0:0
+libpialign_la_LDFLAGS = -no-undefined -version-info 0:0:0
