--- origsrc/pialign-0.2.5/Makefile.am	2019-04-29 23:11:26.000000000 +0900
+++ src/pialign-0.2.5/Makefile.am	2023-11-06 19:05:32.016744900 +0900
@@ -18,3 +18,6 @@ script/reversealign.pl \
 script/time.pl \
 script/make-moses-config.pl \
 script/visualize.pl
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
--- origsrc/pialign-0.2.5/configure.ac	2019-04-29 23:17:22.000000000 +0900
+++ src/pialign-0.2.5/configure.ac	2023-11-06 19:05:32.016744900 +0900
@@ -12,6 +12,7 @@ AC_CONFIG_FILES([
     src/lib/Makefile
     src/bin/Makefile
     src/test/Makefile
+    pialign.pc
 ])
 AC_CONFIG_MACRO_DIRS([m4])
 
--- origsrc/pialign-0.2.5/pialign.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/pialign-0.2.5/pialign.pc.in	2023-11-06 19:05:32.016744900 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: pialign
+Description: Phrasal ITG Aligner
+Version: @VERSION@
+Libs: -L${libdir} -lpialign
+Cflags: -I${includedir}
--- origsrc/pialign-0.2.5/src/lib/Makefile.am	2019-04-29 23:11:26.000000000 +0900
+++ src/pialign-0.2.5/src/lib/Makefile.am	2023-11-06 19:05:32.016744900 +0900
@@ -6,4 +6,4 @@ AM_CPPFLAGS = -I$(srcdir)/../include
 lib_LTLIBRARIES = libpialign.la
 
 libpialign_la_SOURCES = ${PIACPP}
-libpialign_la_LDFLAGS = -version-info 0:0:0
+libpialign_la_LDFLAGS = -no-undefined -version-info 0:0:0
