--- origsrc/mainline/Makefile.am	2010-10-10 16:05:59.000000000 +0900
+++ src/mainline/Makefile.am	2010-10-10 16:10:40.021013000 +0900
@@ -14,7 +14,7 @@
 
 SUBDIRS = flicksoup examples
 
-ACLOCAL_AMFLAGS=-I m4
+ACLOCAL_AMFLAGS=
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = flicksoup.pc
--- origsrc/mainline/configure.ac	2010-10-10 16:05:59.000000000 +0900
+++ src/mainline/configure.ac	2010-10-10 16:06:25.005388000 +0900
@@ -24,7 +24,7 @@ AC_SUBST(LIBRARY_VERSION)
 dnl Required programs
 
 AC_PROG_CC
-AC_PROG_LIBTOOL
+LT_INIT(win32-dll)
 
 
 dnl Default compilation flags
--- origsrc/mainline/flicksoup/Makefile.am	2010-10-10 16:05:59.000000000 +0900
+++ src/mainline/flicksoup/Makefile.am	2010-10-10 16:06:45.661638000 +0900
@@ -42,7 +42,7 @@ libflicksoup_la_LDFLAGS =		 \
 	$(GLIB_LIBS)			 \
 	$(LIBSOUP_LIBS)			 \
 	-version-info $(LIBRARY_VERSION) \
-	-release $(RELEASE)
+	-release $(RELEASE) -no-undefined
 
 hdir = $(includedir)/flicksoup
 h_DATA = 			\
@@ -50,4 +50,4 @@ h_DATA = 			\
 	fsp-data.h		\
 	fsp-error.h 		\
 	fsp-photos-mgr.h	\
-	fsp-session.h
\ No newline at end of file
+	fsp-session.h
