--- origsrc/velocypack/CMakeLists.txt	2019-06-11 00:01:28.000000000 +0900
+++ src/velocypack/CMakeLists.txt	2019-06-11 08:50:22.539256400 +0900
@@ -55,10 +55,20 @@ set(VELOCY_SOURCE
     src/fpconv.cpp
 )
 
+add_library(velocypack SHARED ${VELOCY_SOURCE})
+
+include(FindPkgConfig)
+
 #Use xxhash or fasthash?
 if(HashType STREQUAL "xxhash")
-    list(APPEND VELOCY_SOURCE src/xxhash.cpp)
-    add_definitions("-DVELOCYPACK_XXHASH=1")
+    pkg_check_modules(XXHASH xxhash QUIET)
+    if(XXHASH_FOUND)
+        include_directories(${XXHASH_INCLUDE_DIRS})
+        target_link_libraries(velocypack ${XXHASH_LIBRARIES})
+    else()
+        list(APPEND VELOCY_SOURCE src/xxhash.cpp)
+        add_definitions("-DVELOCYPACK_XXHASH=1")
+    endif()
 elseif(HashType STREQUAL "fasthash")
     list(APPEND VELOCY_SOURCE src/fasthash.cpp)
     add_definitions("-DVELOCYPACK_FASTHASH=1")
@@ -67,9 +77,11 @@ else()
 endif()
 message(STATUS "Building with hash type: ${HashType}")
 
-add_library(velocypack STATIC ${VELOCY_SOURCE})
 target_include_directories(velocypack PRIVATE src)
 target_include_directories(velocypack PUBLIC include)
+if(CYGWIN)
+    set_target_properties(velocypack PROPERTIES SOVERSION 0)
+endif()
 
 if(Maintainer)
     add_executable(buildVersion scripts/build-version.cpp)
@@ -83,11 +95,21 @@ endif()
 
 add_subdirectory(tools)
 add_subdirectory(examples)
+
+enable_testing()
+
 add_subdirectory(tests)
 
-install(TARGETS velocypack DESTINATION lib)
+install(TARGETS velocypack RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
 install(
     DIRECTORY   "include/velocypack"
     DESTINATION include
 )
 include(CPack)
+
+SET(prefix ${CMAKE_INSTALL_PREFIX})
+SET(exec_prefix ${CMAKE_INSTALL_PREFIX})
+SET(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+SET(includedir ${CMAKE_INSTALL_PREFIX}/include)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/velocypack.pc.in ${CMAKE_BINARY_DIR}/velocypack.pc @ONLY)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/velocypack.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
--- origsrc/velocypack/velocypack.pc.in	1970-01-01 09:00:00.000000000 +0900
+++ src/velocypack/velocypack.pc.in	2019-06-11 08:49:38.801870600 +0900
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: velocypack
+Description: Fast and compact format for serialization and storage
+Version: @VELOCYPACK_VERSION@
+Libs: -L${libdir} -lvelocypack
+Cflags: -I${includedir}
