--- origsrc/sbpl-1.3.1/CMakeLists.txt	2018-08-06 23:28:02.000000000 +0900
+++ src/sbpl-1.3.1/CMakeLists.txt	2025-08-16 22:09:19.859040100 +0900
@@ -34,6 +34,10 @@ add_library(sbpl SHARED
   src/utils/config.cpp
   )
 
+if (DEFINED SOVERSION)
+  set_target_properties(sbpl PROPERTIES SOVERSION ${SOVERSION})
+endif()
+
 set(SBPL_INCLUDE_DIR "${CMAKE_INSTALL_PREFIX}/include")
 set(SBPL_LIB_DIR "${CMAKE_INSTALL_PREFIX}/lib")
 
@@ -54,7 +58,7 @@ configure_file(sbpl-config-version.cmake
 install(FILES
   "${PROJECT_BINARY_DIR}/InstallFiles/sbpl-config.cmake"
   "${PROJECT_BINARY_DIR}/InstallFiles/sbpl-config-version.cmake"
-DESTINATION share/sbpl/)
+DESTINATION lib/cmake/sbpl/)
 
 # support for build tree export
 option(SBPL_BUILD_TREE_EXPORT "Enable build tree exports" OFF)
@@ -71,7 +75,8 @@ endif()
 install(DIRECTORY src/include/sbpl DESTINATION include
 	FILES_MATCHING PATTERN "*.h")
 install(TARGETS sbpl
-	LIBRARY DESTINATION lib)
+  RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION lib)
 
 add_executable(test_sbpl src/test/main.cpp)
 target_link_libraries(test_sbpl sbpl)
